{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e5562930",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "6c169afe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>rsID</th>\n",
       "      <th>SAS_Allele_Frequencies</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:946247</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>rs2272757</td>\n",
       "      <td>0.441893</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:946538</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>0.029353</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1:978953</td>\n",
       "      <td>C/G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>rs13302979</td>\n",
       "      <td>0.556310</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1:979472</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>rs13303368</td>\n",
       "      <td>0.512580</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1:979560</td>\n",
       "      <td>T/C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>rs13303033</td>\n",
       "      <td>0.509585</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26280</th>\n",
       "      <td>X:156005349</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>rs150608249</td>\n",
       "      <td>0.057708</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26281</th>\n",
       "      <td>Y:3019783</td>\n",
       "      <td>A/C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>LINC00278</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>0.269669</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26282</th>\n",
       "      <td>Y:6885470</td>\n",
       "      <td>A/G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>AMELY</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>rs9785971</td>\n",
       "      <td>0.270199</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26283</th>\n",
       "      <td>Y:7000077</td>\n",
       "      <td>A/G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TBL1Y</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>rs1865680</td>\n",
       "      <td>0.239735</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26285</th>\n",
       "      <td>Y:21389038</td>\n",
       "      <td>C/A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>rs2033003</td>\n",
       "      <td>0.165298</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14208 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Chr:Pos Ref/Alt      Zygosity  Gene Name         Consequence  \\\n",
       "3         1:946247     G/A    Homozygous      NOC2L  synonymous_variant   \n",
       "4         1:946538     G/A  Heterozygous      NOC2L    missense_variant   \n",
       "15        1:978953     C/G    Homozygous      PERM1    missense_variant   \n",
       "16        1:979472     G/C    Homozygous      PERM1    missense_variant   \n",
       "18        1:979560     T/C    Homozygous      PERM1  synonymous_variant   \n",
       "...            ...     ...           ...        ...                 ...   \n",
       "26280  X:156005349     C/T  Heterozygous       IL9R  synonymous_variant   \n",
       "26281    Y:3019783     A/C    Homozygous  LINC00278      intron_variant   \n",
       "26282    Y:6885470     A/G    Homozygous      AMELY      intron_variant   \n",
       "26283    Y:7000077     A/G    Homozygous      TBL1Y      intron_variant   \n",
       "26285   Y:21389038     C/A    Homozygous      PRORY      intron_variant   \n",
       "\n",
       "              rsID  SAS_Allele_Frequencies CLIN_SIG  \n",
       "3        rs2272757                0.441893   Benign  \n",
       "4       rs35471880                0.029353   Benign  \n",
       "15      rs13302979                0.556310   Benign  \n",
       "16      rs13303368                0.512580   Benign  \n",
       "18      rs13303033                0.509585   Benign  \n",
       "...            ...                     ...      ...  \n",
       "26280  rs150608249                0.057708   Benign  \n",
       "26281    rs9786184                0.269669   Benign  \n",
       "26282    rs9785971                0.270199   Benign  \n",
       "26283    rs1865680                0.239735   Benign  \n",
       "26285    rs2033003                0.165298   Benign  \n",
       "\n",
       "[14208 rows x 8 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHGLBS622.xlsx')\n",
    "vcf['SAS_Allele_Frequencies'] = vcf['SAS_Allele_Frequencies'].replace('', 0).astype(float)\n",
    "vcf = vcf[vcf['SAS_Allele_Frequencies'] <= 0.6]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "6d6e1cff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946247</td>\n",
       "      <td>rs2272757</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946538</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>978953</td>\n",
       "      <td>rs13302979</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979472</td>\n",
       "      <td>rs13303368</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979560</td>\n",
       "      <td>rs13303033</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26280</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156005349</td>\n",
       "      <td>rs150608249</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26281</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>LINC00278</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26282</th>\n",
       "      <td>chrY</td>\n",
       "      <td>6885470</td>\n",
       "      <td>rs9785971</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>AMELY</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26283</th>\n",
       "      <td>chrY</td>\n",
       "      <td>7000077</td>\n",
       "      <td>rs1865680</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TBL1Y</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26285</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21389038</td>\n",
       "      <td>rs2033003</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14208 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT      Zygosity  Gene Name  \\\n",
       "3      chr1     946247    rs2272757   G   A    Homozygous      NOC2L   \n",
       "4      chr1     946538   rs35471880   G   A  Heterozygous      NOC2L   \n",
       "15     chr1     978953   rs13302979   C   G    Homozygous      PERM1   \n",
       "16     chr1     979472   rs13303368   G   C    Homozygous      PERM1   \n",
       "18     chr1     979560   rs13303033   T   C    Homozygous      PERM1   \n",
       "...     ...        ...          ...  ..  ..           ...        ...   \n",
       "26280  chrX  156005349  rs150608249   C   T  Heterozygous       IL9R   \n",
       "26281  chrY    3019783    rs9786184   A   C    Homozygous  LINC00278   \n",
       "26282  chrY    6885470    rs9785971   A   G    Homozygous      AMELY   \n",
       "26283  chrY    7000077    rs1865680   A   G    Homozygous      TBL1Y   \n",
       "26285  chrY   21389038    rs2033003   C   A    Homozygous      PRORY   \n",
       "\n",
       "              Consequence CLIN_SIG         consequence  \n",
       "3      synonymous_variant   Benign  synonymous variant  \n",
       "4        missense_variant   Benign    missense variant  \n",
       "15       missense_variant   Benign    missense variant  \n",
       "16       missense_variant   Benign    missense variant  \n",
       "18     synonymous_variant   Benign  synonymous variant  \n",
       "...                   ...      ...                 ...  \n",
       "26280  synonymous_variant   Benign  synonymous variant  \n",
       "26281      intron_variant   Benign      intron variant  \n",
       "26282      intron_variant   Benign      intron variant  \n",
       "26283      intron_variant   Benign      intron variant  \n",
       "26285      intron_variant   Benign      intron variant  \n",
       "\n",
       "[14208 rows x 10 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CHROM'] = 'chr' + vcf['Chr:Pos'].str.split(':').str[0]\n",
    "vcf['POS'] = vcf['Chr:Pos'].str.split(':').str[1]\n",
    "vcf['REF'] = vcf['Ref/Alt'].str.split('/').str[0]\n",
    "vcf['ALT'] = vcf['Ref/Alt'].str.split('/').str[1]\n",
    "vcf = vcf[['CHROM', 'POS','rsID', 'REF', 'ALT', 'Zygosity', 'Gene Name', 'Consequence', 'CLIN_SIG']]\n",
    "vcf['consequence'] = vcf['Consequence'].str.split(',').str[0].str.replace('_', ' ')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "2f9f16be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "e8ad3e28",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946247</td>\n",
       "      <td>rs2272757</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946538</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>978953</td>\n",
       "      <td>rs13302979</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979472</td>\n",
       "      <td>rs13303368</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979560</td>\n",
       "      <td>rs13303033</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14203</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156005349</td>\n",
       "      <td>rs150608249</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14204</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>LINC00278</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14205</th>\n",
       "      <td>chrY</td>\n",
       "      <td>6885470</td>\n",
       "      <td>rs9785971</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>AMELY</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14206</th>\n",
       "      <td>chrY</td>\n",
       "      <td>7000077</td>\n",
       "      <td>rs1865680</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TBL1Y</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14207</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21389038</td>\n",
       "      <td>rs2033003</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14208 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT      Zygosity  Gene Name  \\\n",
       "0      chr1     946247    rs2272757   G   A    Homozygous      NOC2L   \n",
       "1      chr1     946538   rs35471880   G   A  Heterozygous      NOC2L   \n",
       "2      chr1     978953   rs13302979   C   G    Homozygous      PERM1   \n",
       "3      chr1     979472   rs13303368   G   C    Homozygous      PERM1   \n",
       "4      chr1     979560   rs13303033   T   C    Homozygous      PERM1   \n",
       "...     ...        ...          ...  ..  ..           ...        ...   \n",
       "14203  chrX  156005349  rs150608249   C   T  Heterozygous       IL9R   \n",
       "14204  chrY    3019783    rs9786184   A   C    Homozygous  LINC00278   \n",
       "14205  chrY    6885470    rs9785971   A   G    Homozygous      AMELY   \n",
       "14206  chrY    7000077    rs1865680   A   G    Homozygous      TBL1Y   \n",
       "14207  chrY   21389038    rs2033003   C   A    Homozygous      PRORY   \n",
       "\n",
       "              Consequence CLIN_SIG         consequence Consequence_score  \n",
       "0      synonymous_variant   Benign  synonymous variant              3/10  \n",
       "1        missense_variant   Benign    missense variant              7/10  \n",
       "2        missense_variant   Benign    missense variant              7/10  \n",
       "3        missense_variant   Benign    missense variant              7/10  \n",
       "4      synonymous_variant   Benign  synonymous variant              3/10  \n",
       "...                   ...      ...                 ...               ...  \n",
       "14203  synonymous_variant   Benign  synonymous variant              3/10  \n",
       "14204      intron_variant   Benign      intron variant              2/10  \n",
       "14205      intron_variant   Benign      intron variant              2/10  \n",
       "14206      intron_variant   Benign      intron variant              2/10  \n",
       "14207      intron_variant   Benign      intron variant              2/10  \n",
       "\n",
       "[14208 rows x 11 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(vcf, df_1, on='consequence', how='left', sort=False)\n",
    "#merged_1.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/data_287.xlsx', index=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "578f52fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3/10     6192\n",
       "7/10     4703\n",
       "2/10     1967\n",
       "4/10      591\n",
       "1/10      356\n",
       "NaN       165\n",
       "6/10      112\n",
       "10/10     103\n",
       "8/10       15\n",
       "9/10        4\n",
       "Name: Consequence_score, dtype: int64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.Consequence_score.value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "4c3d989d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946538</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>978953</td>\n",
       "      <td>rs13302979</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979472</td>\n",
       "      <td>rs13303368</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999842</td>\n",
       "      <td>rs2298214</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>HES4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1087202</td>\n",
       "      <td>rs371815645</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C1orf159</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14180</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153770343</td>\n",
       "      <td>rs2266879</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PLXNB3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14185</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909841</td>\n",
       "      <td>rs2070099</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14188</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153954909</td>\n",
       "      <td>rs1051152</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>HCFC1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14191</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153982271</td>\n",
       "      <td>rs2266890</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TMEM187</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14192</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153982294</td>\n",
       "      <td>rs7350355</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TMEM187</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4937 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT      Zygosity Gene Name  \\\n",
       "1      chr1     946538   rs35471880   G   A  Heterozygous     NOC2L   \n",
       "2      chr1     978953   rs13302979   C   G    Homozygous     PERM1   \n",
       "3      chr1     979472   rs13303368   G   C    Homozygous     PERM1   \n",
       "5      chr1     999842    rs2298214   C   A    Homozygous      HES4   \n",
       "9      chr1    1087202  rs371815645   C   A  Heterozygous  C1orf159   \n",
       "...     ...        ...          ...  ..  ..           ...       ...   \n",
       "14180  chrX  153770343    rs2266879   G   A  Heterozygous    PLXNB3   \n",
       "14185  chrX  153909841    rs2070099   C   A    Homozygous   ARHGAP4   \n",
       "14188  chrX  153954909    rs1051152   A   G    Homozygous     HCFC1   \n",
       "14191  chrX  153982271    rs2266890   C   T    Homozygous   TMEM187   \n",
       "14192  chrX  153982294    rs7350355   A   G    Homozygous   TMEM187   \n",
       "\n",
       "            Consequence CLIN_SIG       consequence Consequence_score  \n",
       "1      missense_variant   Benign  missense variant              7/10  \n",
       "2      missense_variant   Benign  missense variant              7/10  \n",
       "3      missense_variant   Benign  missense variant              7/10  \n",
       "5      missense_variant   Benign  missense variant              7/10  \n",
       "9      missense_variant   Benign  missense variant              7/10  \n",
       "...                 ...      ...               ...               ...  \n",
       "14180  missense_variant   Benign  missense variant              7/10  \n",
       "14185  missense_variant   Benign  missense variant              7/10  \n",
       "14188  missense_variant   Benign  missense variant              7/10  \n",
       "14191  missense_variant   Benign  missense variant              7/10  \n",
       "14192  missense_variant   Benign  missense variant              7/10  \n",
       "\n",
       "[4937 rows x 11 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2.fillna('0/10').astype('str')\n",
    "merged_2 = merged_2[merged_2['Consequence_score'].apply(lambda x: eval(x) >= 6/10)]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "2acf738e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>Gene_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946538</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>978953</td>\n",
       "      <td>rs13302979</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979472</td>\n",
       "      <td>rs13303368</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999842</td>\n",
       "      <td>rs2298214</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>HES4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1087202</td>\n",
       "      <td>rs371815645</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C1orf159</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4932</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153770343</td>\n",
       "      <td>rs2266879</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PLXNB3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4933</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909841</td>\n",
       "      <td>rs2070099</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4934</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153954909</td>\n",
       "      <td>rs1051152</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>HCFC1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4935</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153982271</td>\n",
       "      <td>rs2266890</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TMEM187</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4936</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153982294</td>\n",
       "      <td>rs7350355</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TMEM187</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4937 rows Ã— 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS         rsID REF ALT      Zygosity Gene Name  \\\n",
       "0     chr1     946538   rs35471880   G   A  Heterozygous     NOC2L   \n",
       "1     chr1     978953   rs13302979   C   G    Homozygous     PERM1   \n",
       "2     chr1     979472   rs13303368   G   C    Homozygous     PERM1   \n",
       "3     chr1     999842    rs2298214   C   A    Homozygous      HES4   \n",
       "4     chr1    1087202  rs371815645   C   A  Heterozygous  C1orf159   \n",
       "...    ...        ...          ...  ..  ..           ...       ...   \n",
       "4932  chrX  153770343    rs2266879   G   A  Heterozygous    PLXNB3   \n",
       "4933  chrX  153909841    rs2070099   C   A    Homozygous   ARHGAP4   \n",
       "4934  chrX  153954909    rs1051152   A   G    Homozygous     HCFC1   \n",
       "4935  chrX  153982271    rs2266890   C   T    Homozygous   TMEM187   \n",
       "4936  chrX  153982294    rs7350355   A   G    Homozygous   TMEM187   \n",
       "\n",
       "           Consequence CLIN_SIG       consequence Consequence_score  \\\n",
       "0     missense_variant   Benign  missense variant              7/10   \n",
       "1     missense_variant   Benign  missense variant              7/10   \n",
       "2     missense_variant   Benign  missense variant              7/10   \n",
       "3     missense_variant   Benign  missense variant              7/10   \n",
       "4     missense_variant   Benign  missense variant              7/10   \n",
       "...                ...      ...               ...               ...   \n",
       "4932  missense_variant   Benign  missense variant              7/10   \n",
       "4933  missense_variant   Benign  missense variant              7/10   \n",
       "4934  missense_variant   Benign  missense variant              7/10   \n",
       "4935  missense_variant   Benign  missense variant              7/10   \n",
       "4936  missense_variant   Benign  missense variant              7/10   \n",
       "\n",
       "     Gene Match Matched_Gene Condition Headings 21_Conditions_list Gene_Score  \n",
       "0            No                     No       No                 No         No  \n",
       "1            No                     No       No                 No         No  \n",
       "2            No                     No       No                 No         No  \n",
       "3            No                     No       No                 No         No  \n",
       "4            No                     No       No                 No         No  \n",
       "...         ...          ...       ...      ...                ...        ...  \n",
       "4932         No                     No       No                 No         No  \n",
       "4933         No                     No       No                 No         No  \n",
       "4934         No                     No       No                 No         No  \n",
       "4935         No                     No       No                 No         No  \n",
       "4936         No                     No       No                 No         No  \n",
       "\n",
       "[4937 rows x 17 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/genes/Conditions_final_genes.xlsx')\n",
    "\n",
    "merged_2['Gene Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "df_gene = df_gene.rename({'Gene Name':'Matched_Gene'}, axis=1)\n",
    "\n",
    "#merged_2 = merged_2.drop(columns=['Gene Match'], axis=1)\n",
    "\n",
    "merged_2 = pd.merge(merged_2, df_gene, on= 'Matched_Gene', how = 'left', sort = False)\n",
    "merged_2['Condition'] = merged_2['Condition'].fillna('No')\n",
    "merged_2['Headings'] = merged_2['Headings'].fillna('No')\n",
    "merged_2['21_Conditions_list'] = merged_2['21_Conditions_list'].fillna('No')\n",
    "merged_2['Gene_Score'] = merged_2['Gene_Score'].fillna('No')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "954d218c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>gene</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946538</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>978953</td>\n",
       "      <td>rs13302979</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979472</td>\n",
       "      <td>rs13303368</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999842</td>\n",
       "      <td>rs2298214</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>HES4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1087202</td>\n",
       "      <td>rs371815645</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C1orf159</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4932</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153770343</td>\n",
       "      <td>rs2266879</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PLXNB3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4933</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909841</td>\n",
       "      <td>rs2070099</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4934</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153954909</td>\n",
       "      <td>rs1051152</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>HCFC1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4935</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153982271</td>\n",
       "      <td>rs2266890</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TMEM187</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4936</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153982294</td>\n",
       "      <td>rs7350355</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TMEM187</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4937 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS         rsID REF ALT      Zygosity Gene Name  \\\n",
       "0     chr1     946538   rs35471880   G   A  Heterozygous     NOC2L   \n",
       "1     chr1     978953   rs13302979   C   G    Homozygous     PERM1   \n",
       "2     chr1     979472   rs13303368   G   C    Homozygous     PERM1   \n",
       "3     chr1     999842    rs2298214   C   A    Homozygous      HES4   \n",
       "4     chr1    1087202  rs371815645   C   A  Heterozygous  C1orf159   \n",
       "...    ...        ...          ...  ..  ..           ...       ...   \n",
       "4932  chrX  153770343    rs2266879   G   A  Heterozygous    PLXNB3   \n",
       "4933  chrX  153909841    rs2070099   C   A    Homozygous   ARHGAP4   \n",
       "4934  chrX  153954909    rs1051152   A   G    Homozygous     HCFC1   \n",
       "4935  chrX  153982271    rs2266890   C   T    Homozygous   TMEM187   \n",
       "4936  chrX  153982294    rs7350355   A   G    Homozygous   TMEM187   \n",
       "\n",
       "           Consequence CLIN_SIG       consequence Consequence_score  \\\n",
       "0     missense_variant   Benign  missense variant              7/10   \n",
       "1     missense_variant   Benign  missense variant              7/10   \n",
       "2     missense_variant   Benign  missense variant              7/10   \n",
       "3     missense_variant   Benign  missense variant              7/10   \n",
       "4     missense_variant   Benign  missense variant              7/10   \n",
       "...                ...      ...               ...               ...   \n",
       "4932  missense_variant   Benign  missense variant              7/10   \n",
       "4933  missense_variant   Benign  missense variant              7/10   \n",
       "4934  missense_variant   Benign  missense variant              7/10   \n",
       "4935  missense_variant   Benign  missense variant              7/10   \n",
       "4936  missense_variant   Benign  missense variant              7/10   \n",
       "\n",
       "     Gene Match Gene Condition Headings 21_Conditions_list Gene_Score gene  \\\n",
       "0            No             No       No                 No         No  NaN   \n",
       "1            No             No       No                 No         No  NaN   \n",
       "2            No             No       No                 No         No  NaN   \n",
       "3            No             No       No                 No         No  NaN   \n",
       "4            No             No       No                 No         No  NaN   \n",
       "...         ...  ...       ...      ...                ...        ...  ...   \n",
       "4932         No             No       No                 No         No  NaN   \n",
       "4933         No             No       No                 No         No  NaN   \n",
       "4934         No             No       No                 No         No  NaN   \n",
       "4935         No             No       No                 No         No  NaN   \n",
       "4936         No             No       No                 No         No  NaN   \n",
       "\n",
       "     Literature  \n",
       "0            No  \n",
       "1            No  \n",
       "2            No  \n",
       "3            No  \n",
       "4            No  \n",
       "...         ...  \n",
       "4932         No  \n",
       "4933         No  \n",
       "4934         No  \n",
       "4935         No  \n",
       "4936         No  \n",
       "\n",
       "[4937 rows x 19 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/new_final_output_concatenated.xlsx')\n",
    "\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "\n",
    "merged_2 = merged_2.rename({'Matched_Gene':'Gene'}, axis=1)\n",
    "\n",
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "d64a20f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>gene</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20650507</td>\n",
       "      <td>rs1043424</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PINK1,PINK1-AS</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>Parkisons Disease</td>\n",
       "      <td>Parkisons Disease</td>\n",
       "      <td>Parkinsons</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30874473</td>\n",
       "      <td>rs2282440</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SDC3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>SDC3</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33479392</td>\n",
       "      <td>rs41265901</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>Diabetic Neuropathy</td>\n",
       "      <td>Diabetic Nephropathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150512511</td>\n",
       "      <td>rs13294</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ECM1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ECM1</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut_Health</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152305958</td>\n",
       "      <td>rs61816760</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FLG</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>FLG</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4841</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43927042</td>\n",
       "      <td>rs2076213</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty_Liver</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4842</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43927090</td>\n",
       "      <td>rs2076212</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty_Liver</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4863</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50525807</td>\n",
       "      <td>rs11479</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SCO2,TYMP</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>SCO2</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4867</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50572947</td>\n",
       "      <td>rs8142477</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CHKB-CPT1B,CPT1B</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CHKB-CPT1B</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4868</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50577409</td>\n",
       "      <td>rs3213445</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CHKB-CPT1B,CPT1B</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CHKB-CPT1B</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS        rsID REF ALT      Zygosity         Gene Name  \\\n",
       "123    chr1   20650507   rs1043424   A   C  Heterozygous    PINK1,PINK1-AS   \n",
       "164    chr1   30874473   rs2282440   G   A  Heterozygous              SDC3   \n",
       "169    chr1   33479392  rs41265901   A   G  Heterozygous           ZSCAN20   \n",
       "242    chr1  150512511     rs13294   G   A  Heterozygous              ECM1   \n",
       "257    chr1  152305958  rs61816760   T   G  Heterozygous               FLG   \n",
       "...     ...        ...         ...  ..  ..           ...               ...   \n",
       "4841  chr22   43927042   rs2076213   T   G  Heterozygous            PNPLA3   \n",
       "4842  chr22   43927090   rs2076212   G   T  Heterozygous            PNPLA3   \n",
       "4863  chr22   50525807     rs11479   G   A  Heterozygous         SCO2,TYMP   \n",
       "4867  chr22   50572947   rs8142477   G   C  Heterozygous  CHKB-CPT1B,CPT1B   \n",
       "4868  chr22   50577409   rs3213445   T   C  Heterozygous  CHKB-CPT1B,CPT1B   \n",
       "\n",
       "           Consequence       CLIN_SIG       consequence Consequence_score  \\\n",
       "123   missense_variant  Benign,Benign  missense variant              7/10   \n",
       "164   missense_variant         Benign  missense variant              7/10   \n",
       "169   missense_variant         Benign  missense variant              7/10   \n",
       "242   missense_variant         Benign  missense variant              7/10   \n",
       "257   missense_variant         Benign  missense variant              7/10   \n",
       "...                ...            ...               ...               ...   \n",
       "4841  missense_variant         Benign  missense variant              7/10   \n",
       "4842  missense_variant         Benign  missense variant              7/10   \n",
       "4863  missense_variant  Benign,Benign  missense variant              7/10   \n",
       "4867  missense_variant  Benign,Benign  missense variant              7/10   \n",
       "4868  missense_variant  Benign,Benign  missense variant              7/10   \n",
       "\n",
       "     Gene Match        Gene              Condition              Headings  \\\n",
       "123         Yes       PINK1      Parkisons Disease     Parkisons Disease   \n",
       "164         Yes        SDC3                Obesity               Obesity   \n",
       "169         Yes     ZSCAN20    Diabetic Neuropathy  Diabetic Nephropathy   \n",
       "242         Yes        ECM1             Gut Health            Gut Health   \n",
       "257         Yes         FLG  Dry Skin/fragile skin           Skin Health   \n",
       "...         ...         ...                    ...                   ...   \n",
       "4841        Yes      PNPLA3            Fatty liver           Fatty liver   \n",
       "4842        Yes      PNPLA3            Fatty liver           Fatty liver   \n",
       "4863        Yes        SCO2        Muscular_health       Muscular_health   \n",
       "4867        Yes  CHKB-CPT1B        Muscular_health       Muscular_health   \n",
       "4868        Yes  CHKB-CPT1B        Muscular_health       Muscular_health   \n",
       "\n",
       "     21_Conditions_list Gene_Score gene Literature  \n",
       "123          Parkinsons        8.0  NaN         No  \n",
       "164             Obesity         No  NaN         No  \n",
       "169            Diabetes        4.0  NaN         No  \n",
       "242          Gut_Health        6.0  NaN         No  \n",
       "257         Skin_Health        8.0  NaN         No  \n",
       "...                 ...        ...  ...        ...  \n",
       "4841        Fatty_Liver        8.0  NaN         No  \n",
       "4842        Fatty_Liver        8.0  NaN         No  \n",
       "4863    Muscular_health        4.0  NaN         No  \n",
       "4867    Muscular_health         No  NaN         No  \n",
       "4868    Muscular_health         No  NaN         No  \n",
       "\n",
       "[223 rows x 19 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "condition_filter = merged_3[merged_3['21_Conditions_list'] != 'No']\n",
    "condition_filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "7086abee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Parkisons Disease</td>\n",
       "      <td>Parkisons Disease</td>\n",
       "      <td>Parkinsons</td>\n",
       "      <td>PINK1,PINK1-AS</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>rs1043424</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>20650507</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>SDC3</td>\n",
       "      <td>SDC3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2282440</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>30874473</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Diabetic Neuropathy</td>\n",
       "      <td>Diabetic Nephropathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>4.0</td>\n",
       "      <td>rs41265901</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>33479392</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut_Health</td>\n",
       "      <td>ECM1</td>\n",
       "      <td>ECM1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>rs13294</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>150512511</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>FLG</td>\n",
       "      <td>FLG</td>\n",
       "      <td>8.0</td>\n",
       "      <td>rs61816760</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152305958</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty_Liver</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>8.0</td>\n",
       "      <td>rs2076213</td>\n",
       "      <td>No</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43927042</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty liver</td>\n",
       "      <td>Fatty_Liver</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>8.0</td>\n",
       "      <td>rs2076212</td>\n",
       "      <td>No</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43927090</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>SCO2,TYMP</td>\n",
       "      <td>SCO2</td>\n",
       "      <td>4.0</td>\n",
       "      <td>rs11479</td>\n",
       "      <td>No</td>\n",
       "      <td>chr22</td>\n",
       "      <td>50525807</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>CHKB-CPT1B,CPT1B</td>\n",
       "      <td>CHKB-CPT1B</td>\n",
       "      <td>No</td>\n",
       "      <td>rs8142477</td>\n",
       "      <td>No</td>\n",
       "      <td>chr22</td>\n",
       "      <td>50572947</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>Muscular_health</td>\n",
       "      <td>CHKB-CPT1B,CPT1B</td>\n",
       "      <td>CHKB-CPT1B</td>\n",
       "      <td>No</td>\n",
       "      <td>rs3213445</td>\n",
       "      <td>No</td>\n",
       "      <td>chr22</td>\n",
       "      <td>50577409</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Condition              Headings 21_Conditions_list  \\\n",
       "0        Parkisons Disease     Parkisons Disease         Parkinsons   \n",
       "1                  Obesity               Obesity            Obesity   \n",
       "2      Diabetic Neuropathy  Diabetic Nephropathy           Diabetes   \n",
       "3               Gut Health            Gut Health         Gut_Health   \n",
       "4    Dry Skin/fragile skin           Skin Health        Skin_Health   \n",
       "..                     ...                   ...                ...   \n",
       "218            Fatty liver           Fatty liver        Fatty_Liver   \n",
       "219            Fatty liver           Fatty liver        Fatty_Liver   \n",
       "220        Muscular_health       Muscular_health    Muscular_health   \n",
       "221        Muscular_health       Muscular_health    Muscular_health   \n",
       "222        Muscular_health       Muscular_health    Muscular_health   \n",
       "\n",
       "            Gene Name        Gene Gene_Score        rsID Literature  CHROM  \\\n",
       "0      PINK1,PINK1-AS       PINK1        8.0   rs1043424         No   chr1   \n",
       "1                SDC3        SDC3         No   rs2282440         No   chr1   \n",
       "2             ZSCAN20     ZSCAN20        4.0  rs41265901         No   chr1   \n",
       "3                ECM1        ECM1        6.0     rs13294         No   chr1   \n",
       "4                 FLG         FLG        8.0  rs61816760         No   chr1   \n",
       "..                ...         ...        ...         ...        ...    ...   \n",
       "218            PNPLA3      PNPLA3        8.0   rs2076213         No  chr22   \n",
       "219            PNPLA3      PNPLA3        8.0   rs2076212         No  chr22   \n",
       "220         SCO2,TYMP        SCO2        4.0     rs11479         No  chr22   \n",
       "221  CHKB-CPT1B,CPT1B  CHKB-CPT1B         No   rs8142477         No  chr22   \n",
       "222  CHKB-CPT1B,CPT1B  CHKB-CPT1B         No   rs3213445         No  chr22   \n",
       "\n",
       "           POS REF ALT      Zygosity       Consequence Consequence_score  \\\n",
       "0     20650507   A   C  Heterozygous  missense_variant              7/10   \n",
       "1     30874473   G   A  Heterozygous  missense_variant              7/10   \n",
       "2     33479392   A   G  Heterozygous  missense_variant              7/10   \n",
       "3    150512511   G   A  Heterozygous  missense_variant              7/10   \n",
       "4    152305958   T   G  Heterozygous  missense_variant              7/10   \n",
       "..         ...  ..  ..           ...               ...               ...   \n",
       "218   43927042   T   G  Heterozygous  missense_variant              7/10   \n",
       "219   43927090   G   T  Heterozygous  missense_variant              7/10   \n",
       "220   50525807   G   A  Heterozygous  missense_variant              7/10   \n",
       "221   50572947   G   C  Heterozygous  missense_variant              7/10   \n",
       "222   50577409   T   C  Heterozygous  missense_variant              7/10   \n",
       "\n",
       "          CLIN_SIG       consequence Gene Match gene  \n",
       "0    Benign,Benign  missense variant        Yes  NaN  \n",
       "1           Benign  missense variant        Yes  NaN  \n",
       "2           Benign  missense variant        Yes  NaN  \n",
       "3           Benign  missense variant        Yes  NaN  \n",
       "4           Benign  missense variant        Yes  NaN  \n",
       "..             ...               ...        ...  ...  \n",
       "218         Benign  missense variant        Yes  NaN  \n",
       "219         Benign  missense variant        Yes  NaN  \n",
       "220  Benign,Benign  missense variant        Yes  NaN  \n",
       "221  Benign,Benign  missense variant        Yes  NaN  \n",
       "222  Benign,Benign  missense variant        Yes  NaN  \n",
       "\n",
       "[223 rows x 19 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "condition_filter = condition_filter.set_index(['Gene Name', 'Gene', 'Gene_Score', 'rsID', 'Literature', 'CHROM', 'POS', 'REF',\n",
    "       'ALT', 'Zygosity', 'Consequence', 'Consequence_score', 'CLIN_SIG', 'Condition', 'Headings', '21_Conditions_list']).apply(lambda x: x.str.split('; ').explode()).reset_index()\n",
    "\n",
    "# Reorder the DataFrame\n",
    "columns_to_move = ['Condition', 'Headings', '21_Conditions_list']\n",
    "condition_filter = condition_filter[columns_to_move + [col for col in condition_filter.columns if col not in columns_to_move]]\n",
    "condition_filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "19551a99",
   "metadata": {},
   "outputs": [],
   "source": [
    "drop_duplicates_filter = condition_filter.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "1712b7ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sample list of condition names\n",
    "conditions_list = [\n",
    "    'Diabetes',\n",
    "    'High_Blood_Pressure',\n",
    "    'Cardiac_Health',\n",
    "    'Cholesterol_Disorders',\n",
    "    'Thyroid_Disorders',\n",
    "    'Parkinsons',\n",
    "    'Dementia',\n",
    "    'Headaches',\n",
    "    'Allergies',\n",
    "    'Anemia',\n",
    "    'Fatty_Liver',\n",
    "    'Gall_stones',\n",
    "    'Pancreatic_Disorders',\n",
    "    'Gut_Health',\n",
    "    'Gastritis',\n",
    "    'Glomerular_Diseases',\n",
    "    'Interstitial_Nephritis',\n",
    "    'Renal_stones',\n",
    "    'Skin_Health',\n",
    "    'Osteoporosis',\n",
    "    'Arthritis_Degenerative_Joint',\n",
    "    'Muscular_Dystrophy',\n",
    "    'Fatigue',\n",
    "    'Mood_Disorders',\n",
    "    'Obesity',\n",
    "    'Seizures',\n",
    "    'Bone_Joint_health',\n",
    "    'Muscular_health'\n",
    "]\n",
    "\n",
    "# Create an Excel writer object\n",
    "excel_writer = pd.ExcelWriter(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/KHGLBS622_final_output.xlsx', engine='xlsxwriter')\n",
    "\n",
    "# Define a workbook\n",
    "workbook = excel_writer.book\n",
    "\n",
    "# Define the green and pink cell formats\n",
    "green_format = workbook.add_format({'bg_color': '#00FF00'})  # Green color\n",
    "pink_format = workbook.add_format({'bg_color': '#FF00FF'})   # Pink color\n",
    "\n",
    "# A flag to keep track of whether at least one group was found\n",
    "found_at_least_one_group = False\n",
    "\n",
    "# Iterate through each condition and save it as a separate sheet\n",
    "for condition in conditions_list:\n",
    "    group = drop_duplicates_filter[drop_duplicates_filter['21_Conditions_list'] == condition]\n",
    "    if not group.empty:\n",
    "        found_at_least_one_group = True\n",
    "        # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "        sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', condition)[:31]\n",
    "\n",
    "        # Sort the group first by the \"Condition\" column and then by the \"Literature\" column\n",
    "        group_sorted = group.sort_values(by=['Condition', 'Literature'], ascending=[True, False])\n",
    "\n",
    "        # Delete the '21_Conditions_list' column\n",
    "        group_sorted = group_sorted.drop(columns=['21_Conditions_list', 'Gene Name'])\n",
    "\n",
    "        columns_to_keep = ['Condition', 'Headings', 'Gene', 'Gene_Score', 'Zygosity', 'CLIN_SIG','Consequence', 'Consequence_score', 'CHROM', 'POS','rsID','Literature', 'REF', 'ALT']\n",
    "\n",
    "        # Select only the desired columns\n",
    "        group_sorted = group_sorted[columns_to_keep]\n",
    "\n",
    "        # Write each group to a separate sheet with the modified sheet name\n",
    "        group_sorted.to_excel(excel_writer, sheet_name=sheet_name, index=False)\n",
    "\n",
    "        # Access the worksheet to apply conditional formatting\n",
    "        worksheet = excel_writer.sheets[sheet_name]\n",
    "\n",
    "        # Apply conditional formatting based on 'Literature' column values\n",
    "        for i, value in enumerate(group_sorted['Literature']):\n",
    "            if value == 'Yes':\n",
    "                worksheet.write(i + 1, columns_to_keep.index('Literature'), value, green_format)\n",
    "            elif value == 'No':\n",
    "                worksheet.write(i + 1, columns_to_keep.index('Literature'), value, pink_format)\n",
    "\n",
    "# If no groups were found, create a sheet named \"Condition not found\" and write the message in cell A1\n",
    "for condition in conditions_list:\n",
    "    if condition not in excel_writer.sheets:\n",
    "        # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "        sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', condition)[:31]\n",
    "        worksheet = excel_writer.book.add_worksheet(sheet_name)\n",
    "        worksheet.write(0, 0, \"Variants not found\")\n",
    "\n",
    "# Save the Excel file\n",
    "excel_writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c7cc51d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b5263b5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebe8fa71",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "52ec42bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/kalyani_mam_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGHGPTTL287-WESPOOL160_S70_L003_R.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGHGPTTL287-WESPOOL160_S70_L003_R.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9e15c852",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>AD</th>\n",
       "      <th>DP</th>\n",
       "      <th>GQ</th>\n",
       "      <th>PGT</th>\n",
       "      <th>PID</th>\n",
       "      <th>PL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>67.28</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>95,9,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1964.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=68;ExcessHet=3.0103;FS=0.000...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0,68</td>\n",
       "      <td>68</td>\n",
       "      <td>99</td>\n",
       "      <td>1993,204,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>31.74</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=2;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>59,6,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>931131</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>CCCCT</td>\n",
       "      <td>1090.73</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;BaseQRankSum=-0.671;ClippingRan...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1,27</td>\n",
       "      <td>28</td>\n",
       "      <td>58</td>\n",
       "      <td>1128,58,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1290.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=38;ExcessHet=3.0103;FS=0.000...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0,38</td>\n",
       "      <td>38</td>\n",
       "      <td>99</td>\n",
       "      <td>1319,114,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48809</th>\n",
       "      <td>chr9_KI270719v1_random</td>\n",
       "      <td>154634</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>268.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.5;AN=2;BaseQRankSum=-2.188;ClippingR...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>60,11</td>\n",
       "      <td>71</td>\n",
       "      <td>99</td>\n",
       "      <td>0|1</td>\n",
       "      <td>154555_G_A</td>\n",
       "      <td>297,0,2794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48810</th>\n",
       "      <td>chr9_KI270719v1_random</td>\n",
       "      <td>154676</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>156.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.5;AN=2;BaseQRankSum=-0.329;ClippingR...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>61,8</td>\n",
       "      <td>69</td>\n",
       "      <td>99</td>\n",
       "      <td>0|1</td>\n",
       "      <td>154555_G_A</td>\n",
       "      <td>185,0,2630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48811</th>\n",
       "      <td>chr9_KI270719v1_random</td>\n",
       "      <td>154703</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>44.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.5;AN=2;BaseQRankSum=1.987;ClippingRa...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69,7</td>\n",
       "      <td>76</td>\n",
       "      <td>73</td>\n",
       "      <td>0|1</td>\n",
       "      <td>154555_G_A</td>\n",
       "      <td>73,0,2715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48812</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>1459097</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>63.28</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>91,9,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48813</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>1592156</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>107.28</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1592115_A_T</td>\n",
       "      <td>135,9,0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48814 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         CHROM      POS rsID REF    ALT     QUAL FILTER  \\\n",
       "0                         chr1    69270    .   A      G    67.28      .   \n",
       "1                         chr1    69511    .   A      G  1964.77      .   \n",
       "2                         chr1    69897    .   T      C    31.74      .   \n",
       "3                         chr1   931131    .   C  CCCCT  1090.73      .   \n",
       "4                         chr1   941119    .   A      G  1290.77      .   \n",
       "...                        ...      ...  ...  ..    ...      ...    ...   \n",
       "48809   chr9_KI270719v1_random   154634    .   G      T   268.77      .   \n",
       "48810   chr9_KI270719v1_random   154676    .   G      T   156.77      .   \n",
       "48811   chr9_KI270719v1_random   154703    .   C      T    44.77      .   \n",
       "48812  chr16_KI270728v1_random  1459097    .   G      A    63.28      .   \n",
       "48813  chr16_KI270728v1_random  1592156    .   T      C   107.28      .   \n",
       "\n",
       "                                                    INFO   GT     AD  DP  GQ  \\\n",
       "0      AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...  1/1    0,3   3   9   \n",
       "1      AC=2;AF=1;AN=2;DP=68;ExcessHet=3.0103;FS=0.000...  1/1   0,68  68  99   \n",
       "2      AC=2;AF=1;AN=2;DP=2;ExcessHet=3.0103;FS=0.000;...  1/1    0,2   2   6   \n",
       "3      AC=2;AF=1;AN=2;BaseQRankSum=-0.671;ClippingRan...  1/1   1,27  28  58   \n",
       "4      AC=2;AF=1;AN=2;DP=38;ExcessHet=3.0103;FS=0.000...  1/1   0,38  38  99   \n",
       "...                                                  ...  ...    ...  ..  ..   \n",
       "48809  AC=1;AF=0.5;AN=2;BaseQRankSum=-2.188;ClippingR...  0/1  60,11  71  99   \n",
       "48810  AC=1;AF=0.5;AN=2;BaseQRankSum=-0.329;ClippingR...  0/1   61,8  69  99   \n",
       "48811  AC=1;AF=0.5;AN=2;BaseQRankSum=1.987;ClippingRa...  0/1   69,7  76  73   \n",
       "48812  AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...  1/1    0,3   3   9   \n",
       "48813  AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...  1/1    0,3   3   9   \n",
       "\n",
       "              PGT          PID          PL  \n",
       "0          95,9,0          NaN         NaN  \n",
       "1      1993,204,0          NaN         NaN  \n",
       "2          59,6,0          NaN         NaN  \n",
       "3       1128,58,0          NaN         NaN  \n",
       "4      1319,114,0          NaN         NaN  \n",
       "...           ...          ...         ...  \n",
       "48809         0|1   154555_G_A  297,0,2794  \n",
       "48810         0|1   154555_G_A  185,0,2630  \n",
       "48811         0|1   154555_G_A   73,0,2715  \n",
       "48812      91,9,0          NaN         NaN  \n",
       "48813         0|1  1592115_A_T     135,9,0  \n",
       "\n",
       "[48814 rows x 15 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGHGPTTL287-WESPOOL160_S70_L003_R.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "vcf['GT'] = vcf['SAMPLE'].str.split(':').str[0]\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2]\n",
    "vcf['GQ'] = vcf['SAMPLE'].str.split(':').str[3]\n",
    "vcf['PGT'] = vcf['SAMPLE'].str.split(':').str[4]\n",
    "vcf['PID'] = vcf['SAMPLE'].str.split(':').str[5]\n",
    "vcf['PL'] = vcf['SAMPLE'].str.split(':').str[6]\n",
    "\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'AD', 'DP', 'GQ', 'PGT', 'PID', 'PL']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e478ea2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "aa1af371",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13273</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>43.74</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=2;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,2:2:6:71,6,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13417</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>CGAGA</td>\n",
       "      <td>166.73</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.5;AN=2;BaseQRankSum=-0.967;ClippingR...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:1,5:6:40:204,0,40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14522</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>91.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.5;AN=2;BaseQRankSum=-0.000;ClippingR...</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>0/1:2,3:5:75:0|1:14522_G_A:120,0,75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14542</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>91.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.5;AN=2;BaseQRankSum=-0.000;ClippingR...</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>0/1:2,3:5:75:0|1:14522_G_A:120,0,75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14574</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>133.77</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.5;AN=2;BaseQRankSum=-0.000;ClippingR...</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>0/1:2,4:6:99:0|1:14522_G_A:162,0,102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135350</th>\n",
       "      <td>chrUn_JTFH01000319v1_decoy</td>\n",
       "      <td>1429</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>21.77</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=7;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,2:2:6:49,6,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135351</th>\n",
       "      <td>chrUn_JTFH01000981v1_decoy</td>\n",
       "      <td>2235</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>15.65</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=8;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,1:1:3:42,3,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135352</th>\n",
       "      <td>chrUn_JTFH01000981v1_decoy</td>\n",
       "      <td>2372</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>15.65</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=4;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,1:1:3:42,3,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135353</th>\n",
       "      <td>chrUn_JTFH01001016v1_decoy</td>\n",
       "      <td>1437</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>107.28</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>1/1:0,3:3:9:0|1:1437_G_C:135,9,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135354</th>\n",
       "      <td>chrUn_JTFH01001016v1_decoy</td>\n",
       "      <td>1448</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>107.28</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>1/1:0,3:3:9:0|1:1437_G_C:135,9,0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>135355 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             CHROM    POS rsID REF    ALT    QUAL   FILTER  \\\n",
       "0                             chr1  13273    .   G      C   43.74        .   \n",
       "1                             chr1  13417    .   C  CGAGA  166.73        .   \n",
       "2                             chr1  14522    .   G      A   91.77        .   \n",
       "3                             chr1  14542    .   A      G   91.77        .   \n",
       "4                             chr1  14574    .   A      G  133.77        .   \n",
       "...                            ...    ...  ...  ..    ...     ...      ...   \n",
       "135350  chrUn_JTFH01000319v1_decoy   1429    .   G      T   21.77  LowQual   \n",
       "135351  chrUn_JTFH01000981v1_decoy   2235    .   T      C   15.65  LowQual   \n",
       "135352  chrUn_JTFH01000981v1_decoy   2372    .   A      C   15.65  LowQual   \n",
       "135353  chrUn_JTFH01001016v1_decoy   1437    .   G      C  107.28        .   \n",
       "135354  chrUn_JTFH01001016v1_decoy   1448    .   T      C  107.28        .   \n",
       "\n",
       "                                                     INFO  \\\n",
       "0       AC=2;AF=1;AN=2;DP=2;ExcessHet=3.0103;FS=0.000;...   \n",
       "1       AC=1;AF=0.5;AN=2;BaseQRankSum=-0.967;ClippingR...   \n",
       "2       AC=1;AF=0.5;AN=2;BaseQRankSum=-0.000;ClippingR...   \n",
       "3       AC=1;AF=0.5;AN=2;BaseQRankSum=-0.000;ClippingR...   \n",
       "4       AC=1;AF=0.5;AN=2;BaseQRankSum=-0.000;ClippingR...   \n",
       "...                                                   ...   \n",
       "135350  AC=2;AF=1;AN=2;DP=7;ExcessHet=3.0103;FS=0.000;...   \n",
       "135351  AC=2;AF=1;AN=2;DP=8;ExcessHet=3.0103;FS=0.000;...   \n",
       "135352  AC=2;AF=1;AN=2;DP=4;ExcessHet=3.0103;FS=0.000;...   \n",
       "135353  AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...   \n",
       "135354  AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;...   \n",
       "\n",
       "                        FORMAT                                SAMPLE  \n",
       "0               GT:AD:DP:GQ:PL                    1/1:0,2:2:6:71,6,0  \n",
       "1               GT:AD:DP:GQ:PL                 0/1:1,5:6:40:204,0,40  \n",
       "2       GT:AD:DP:GQ:PGT:PID:PL   0/1:2,3:5:75:0|1:14522_G_A:120,0,75  \n",
       "3       GT:AD:DP:GQ:PGT:PID:PL   0/1:2,3:5:75:0|1:14522_G_A:120,0,75  \n",
       "4       GT:AD:DP:GQ:PGT:PID:PL  0/1:2,4:6:99:0|1:14522_G_A:162,0,102  \n",
       "...                        ...                                   ...  \n",
       "135350          GT:AD:DP:GQ:PL                    1/1:0,2:2:6:49,6,0  \n",
       "135351          GT:AD:DP:GQ:PL                    1/1:0,1:1:3:42,3,0  \n",
       "135352          GT:AD:DP:GQ:PL                    1/1:0,1:1:3:42,3,0  \n",
       "135353  GT:AD:DP:GQ:PGT:PID:PL      1/1:0,3:3:9:0|1:1437_G_C:135,9,0  \n",
       "135354  GT:AD:DP:GQ:PGT:PID:PL      1/1:0,3:3:9:0|1:1437_G_C:135,9,0  \n",
       "\n",
       "[135355 rows x 10 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGHGPTTL287-WESPOOL160_S70_L003_R.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "63873863",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'AC=2;AF=1;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;MLEAC=2;MLEAF=1;MQ=27.15;QD=22.43;SOR=2.833'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = vcf[vcf['POS'] == 69270]\n",
    "x.INFO.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d4106ed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04113f78",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "5049cce5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>haplotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2,rs3</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs4,rs5</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      rsID haplotype\n",
       "0      rs1         A\n",
       "1  rs2,rs3         B\n",
       "2  rs4,rs5         C"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming you have two dataframes: df1 and df2\n",
    "# Replace 'haplotype' with the actual column name you want to map\n",
    "\n",
    "# Example data\n",
    "data1 = {'rsID': ['rs1', 'rs2,rs3', 'rs4,rs5'], 'haplotype': ['A', 'B', 'C']}\n",
    "data2 = {'rsID': ['rs3', 'rs1', 'rs5,rs4'], 'haplotype': ['X', 'Y', 'Z']}\n",
    "\n",
    "df1 = pd.DataFrame(data1)\n",
    "df2 = pd.DataFrame(data2)\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "2da76e63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  data\n",
       "0  rs1799930,rs1041983\n",
       "1  rs1799930,rs1041983\n",
       "2  rs2279343,rs3745274"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming 'data' is the column containing the comma-separated values\n",
    "data = ['rs1799930,rs1041983', 'rs1799930,rs1041983', 'rs2279343,rs3745274']\n",
    "\n",
    "# Convert the data column to a DataFrame\n",
    "df = pd.DataFrame({'data': data})\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "2985d5ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  data\n",
       "0  rs1041983,rs1799930\n",
       "1  rs1041983,rs1799930\n",
       "2  rs2279343,rs3745274"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Split each row into a list of values, sort the values, and join them back into a string\n",
    "df['data'] = df['data'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "\n",
    "# Print the sorted DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5017e444",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
