{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3d7b2926",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "a043688b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/srinivas_sir_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/clinvar_20240206.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/clinvar_20240206.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "185ff46f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2206917;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2232003;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2200058;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2226217;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=32774;CLNDISDB=MedGen:C3661900|Human_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  CHROM     POS     rsID REF ALT QUAL FILTER  \\\n",
       "0                     1   69134  2205837   A   G    .      .   \n",
       "1                     1   69581  2252161   C   G    .      .   \n",
       "2                     1   69682  2396347   G   A    .      .   \n",
       "3                     1   69769  2288999   T   C    .      .   \n",
       "4                     1   69995  2351346   G   C    .      .   \n",
       "...                 ...     ...      ...  ..  ..  ...    ...   \n",
       "2353858     NT_187693.1  273806  2219599   G   A    .      .   \n",
       "2353859     NT_187693.1  273866  2237818   A   C    .      .   \n",
       "2353860     NT_187693.1  274366  2206666   G   C    .      .   \n",
       "2353861     NT_187693.1  275068  2241971   T   C    .      .   \n",
       "2353862  NW_009646201.1   83614    17735  TC   T    .      .   \n",
       "\n",
       "                                                      INFO  \n",
       "0        ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "1        ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "2        ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "3        ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "4        ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "...                                                    ...  \n",
       "2353858  ALLELEID=2206917;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "2353859  ALLELEID=2232003;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "2353860  ALLELEID=2200058;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "2353861  ALLELEID=2226217;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "2353862  ALLELEID=32774;CLNDISDB=MedGen:C3661900|Human_...  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/clinvar_20240206.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8d48cfb6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2206917;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LILRA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2232003;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LILRA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2200058;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LILRA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2226217;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=32774;CLNDISDB=MedGen:C3661900|Human_...</td>\n",
       "      <td>ABO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  CHROM     POS     rsID REF ALT QUAL FILTER  \\\n",
       "0                     1   69134  2205837   A   G    .      .   \n",
       "1                     1   69581  2252161   C   G    .      .   \n",
       "2                     1   69682  2396347   G   A    .      .   \n",
       "3                     1   69769  2288999   T   C    .      .   \n",
       "4                     1   69995  2351346   G   C    .      .   \n",
       "...                 ...     ...      ...  ..  ..  ...    ...   \n",
       "2353858     NT_187693.1  273806  2219599   G   A    .      .   \n",
       "2353859     NT_187693.1  273866  2237818   A   C    .      .   \n",
       "2353860     NT_187693.1  274366  2206666   G   C    .      .   \n",
       "2353861     NT_187693.1  275068  2241971   T   C    .      .   \n",
       "2353862  NW_009646201.1   83614    17735  TC   T    .      .   \n",
       "\n",
       "                                                      INFO  \\\n",
       "0        ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "1        ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2        ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "3        ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "4        ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "...                                                    ...   \n",
       "2353858  ALLELEID=2206917;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353859  ALLELEID=2232003;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353860  ALLELEID=2200058;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353861  ALLELEID=2226217;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353862  ALLELEID=32774;CLNDISDB=MedGen:C3661900|Human_...   \n",
       "\n",
       "                        Gene  \n",
       "0                      OR4F5  \n",
       "1                      OR4F5  \n",
       "2                      OR4F5  \n",
       "3                      OR4F5  \n",
       "4                      OR4F5  \n",
       "...                      ...  \n",
       "2353858               LILRA3  \n",
       "2353859               LILRA3  \n",
       "2353860               LILRA3  \n",
       "2353861  LOC126057115,LILRA3  \n",
       "2353862                  ABO  \n",
       "\n",
       "[2353863 rows x 9 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene'] = vcf['Gene'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7ef38856",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2206917;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2232003;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2200058;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2226217;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=32774;CLNDISDB=MedGen:C3661900|Human_...</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  CHROM     POS     rsID REF ALT QUAL FILTER  \\\n",
       "0                     1   69134  2205837   A   G    .      .   \n",
       "1                     1   69581  2252161   C   G    .      .   \n",
       "2                     1   69682  2396347   G   A    .      .   \n",
       "3                     1   69769  2288999   T   C    .      .   \n",
       "4                     1   69995  2351346   G   C    .      .   \n",
       "...                 ...     ...      ...  ..  ..  ...    ...   \n",
       "2353858     NT_187693.1  273806  2219599   G   A    .      .   \n",
       "2353859     NT_187693.1  273866  2237818   A   C    .      .   \n",
       "2353860     NT_187693.1  274366  2206666   G   C    .      .   \n",
       "2353861     NT_187693.1  275068  2241971   T   C    .      .   \n",
       "2353862  NW_009646201.1   83614    17735  TC   T    .      .   \n",
       "\n",
       "                                                      INFO  \\\n",
       "0        ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "1        ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2        ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "3        ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "4        ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "...                                                    ...   \n",
       "2353858  ALLELEID=2206917;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353859  ALLELEID=2232003;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353860  ALLELEID=2200058;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353861  ALLELEID=2226217;CLNDISDB=MeSH:D030342,MedGen:...   \n",
       "2353862  ALLELEID=32774;CLNDISDB=MedGen:C3661900|Human_...   \n",
       "\n",
       "                        Gene                  CLNSIG  \n",
       "0                      OR4F5           Likely_benign  \n",
       "1                      OR4F5  Uncertain_significance  \n",
       "2                      OR4F5  Uncertain_significance  \n",
       "3                      OR4F5  Uncertain_significance  \n",
       "4                      OR4F5  Uncertain_significance  \n",
       "...                      ...                     ...  \n",
       "2353858               LILRA3           Likely_benign  \n",
       "2353859               LILRA3  Uncertain_significance  \n",
       "2353860               LILRA3  Uncertain_significance  \n",
       "2353861  LOC126057115,LILRA3  Uncertain_significance  \n",
       "2353862                  ABO                  Benign  \n",
       "\n",
       "[2353863 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"CLNSIG\"] = vcf[\"INFO\"].str.extract('CLNSIG=(?P<CLNSIG>.+?);')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1e438c6b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  CHROM     POS     rsID REF ALT                 Gene  \\\n",
       "0                     1   69134  2205837   A   G                OR4F5   \n",
       "1                     1   69581  2252161   C   G                OR4F5   \n",
       "2                     1   69682  2396347   G   A                OR4F5   \n",
       "3                     1   69769  2288999   T   C                OR4F5   \n",
       "4                     1   69995  2351346   G   C                OR4F5   \n",
       "...                 ...     ...      ...  ..  ..                  ...   \n",
       "2353858     NT_187693.1  273806  2219599   G   A               LILRA3   \n",
       "2353859     NT_187693.1  273866  2237818   A   C               LILRA3   \n",
       "2353860     NT_187693.1  274366  2206666   G   C               LILRA3   \n",
       "2353861     NT_187693.1  275068  2241971   T   C  LOC126057115,LILRA3   \n",
       "2353862  NW_009646201.1   83614    17735  TC   T                  ABO   \n",
       "\n",
       "                         CLNSIG  \n",
       "0                 Likely_benign  \n",
       "1        Uncertain_significance  \n",
       "2        Uncertain_significance  \n",
       "3        Uncertain_significance  \n",
       "4        Uncertain_significance  \n",
       "...                         ...  \n",
       "2353858           Likely_benign  \n",
       "2353859  Uncertain_significance  \n",
       "2353860  Uncertain_significance  \n",
       "2353861  Uncertain_significance  \n",
       "2353862                  Benign  \n",
       "\n",
       "[2353863 rows x 7 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'Gene', 'CLNSIG']]\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e6e30ec4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65811</td>\n",
       "      <td>65993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69461</td>\n",
       "      <td>69620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>785981</td>\n",
       "      <td>786159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786130</td>\n",
       "      <td>786446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230714</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190028</td>\n",
       "      <td>57190328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230715</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190299</td>\n",
       "      <td>57190439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230716</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190874</td>\n",
       "      <td>57191014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230717</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191846</td>\n",
       "      <td>57192058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230718</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194102</td>\n",
       "      <td>57194243</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230719 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos\n",
       "0            chr1      65489     65645\n",
       "1            chr1      65811     65993\n",
       "2            chr1      69461     69620\n",
       "3            chr1     785981    786159\n",
       "4            chr1     786130    786446\n",
       "...           ...        ...       ...\n",
       "230714       chrY   57190028  57190328\n",
       "230715       chrY   57190299  57190439\n",
       "230716       chrY   57190874  57191014\n",
       "230717       chrY   57191846  57192058\n",
       "230718       chrY   57194102  57194243\n",
       "\n",
       "[230719 rows x 3 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/srinivas_sir_covered.bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5a3eba56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  CHROM     POS     rsID REF ALT                 Gene  \\\n",
       "0                     1   69134  2205837   A   G                OR4F5   \n",
       "1                     1   69581  2252161   C   G                OR4F5   \n",
       "2                     1   69682  2396347   G   A                OR4F5   \n",
       "3                     1   69769  2288999   T   C                OR4F5   \n",
       "4                     1   69995  2351346   G   C                OR4F5   \n",
       "...                 ...     ...      ...  ..  ..                  ...   \n",
       "2353858     NT_187693.1  273806  2219599   G   A               LILRA3   \n",
       "2353859     NT_187693.1  273866  2237818   A   C               LILRA3   \n",
       "2353860     NT_187693.1  274366  2206666   G   C               LILRA3   \n",
       "2353861     NT_187693.1  275068  2241971   T   C  LOC126057115,LILRA3   \n",
       "2353862  NW_009646201.1   83614    17735  TC   T                  ABO   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "0                 Likely_benign    Not_Covered  \n",
       "1        Uncertain_significance    Not_Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "...                         ...            ...  \n",
       "2353858           Likely_benign    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "2353862                  Benign    Not_Covered  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Start_pos']\n",
    "    end_pos = row['End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in data\n",
    "vcf_new['Covered_status'] = vcf_new.apply(check_coverage, axis=1)\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "555e5303",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chr2                    222933\n",
       "chr1                    206078\n",
       "chr17                   147671\n",
       "chr11                   139443\n",
       "chr3                    126372\n",
       "chr16                   124927\n",
       "chr19                   123537\n",
       "chr5                    119884\n",
       "chr7                    112893\n",
       "chr12                   105629\n",
       "chr9                    105004\n",
       "chr6                    104971\n",
       "chrX                     87730\n",
       "chr10                    87560\n",
       "chr15                    82402\n",
       "chr4                     80960\n",
       "chr8                     79867\n",
       "chr14                    74855\n",
       "chr22                    51217\n",
       "chr13                    50780\n",
       "chr20                    47570\n",
       "chr18                    41051\n",
       "chr21                    27517\n",
       "chrMT                     2910\n",
       "chrY                        78\n",
       "chr19_KI270938v1_alt         9\n",
       "chr17_KI270909v1_alt         7\n",
       "chr22_KI270879v1_alt         6\n",
       "chrUn_GL000218v1             1\n",
       "chr9_KN196479v1_fix          1\n",
       "Name: CHROM, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new.CHROM.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e20d54c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>chr9_KN196479v1_fix</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CHROM     POS     rsID REF ALT                 Gene  \\\n",
       "0                           1   69134  2205837   A   G                OR4F5   \n",
       "1                           1   69581  2252161   C   G                OR4F5   \n",
       "2                           1   69682  2396347   G   A                OR4F5   \n",
       "3                           1   69769  2288999   T   C                OR4F5   \n",
       "4                           1   69995  2351346   G   C                OR4F5   \n",
       "...                       ...     ...      ...  ..  ..                  ...   \n",
       "2353858  chr19_KI270938v1_alt  273806  2219599   G   A               LILRA3   \n",
       "2353859  chr19_KI270938v1_alt  273866  2237818   A   C               LILRA3   \n",
       "2353860  chr19_KI270938v1_alt  274366  2206666   G   C               LILRA3   \n",
       "2353861  chr19_KI270938v1_alt  275068  2241971   T   C  LOC126057115,LILRA3   \n",
       "2353862   chr9_KN196479v1_fix   83614    17735  TC   T                  ABO   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "0                 Likely_benign    Not_Covered  \n",
       "1        Uncertain_significance    Not_Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "...                         ...            ...  \n",
       "2353858           Likely_benign    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "2353862                  Benign    Not_Covered  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new['CHROM'] = vcf_new['CHROM'].replace({'NT_187693.1': 'chr19_KI270938v1_alt', 'NT_187661.1': 'chr17_KI270909v1_alt', \n",
    "                                            'NT_187633.1': 'chr22_KI270879v1_alt', 'NT_113889.1': 'chrUn_GL000218v1', \n",
    "                                            'NW_009646201.1': 'chr9_KN196479v1_fix'})\n",
    "vcf_new "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "9a768377",
   "metadata": {},
   "outputs": [],
   "source": [
    "#NT_187693.1 = chr19_KI270938v1_alt\n",
    "#NT_187661.1 = chr17_KI270909v1_alt\n",
    "#NT_187633.1 = chr22_KI270879v1_alt\n",
    "#NT_113889.1 = chrUn_GL000218v1\n",
    "#NW_009646201.1 = chr9_KN196479v1_fix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c1a5d804",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>chrchr19_KI270938v1_alt</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chrchr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chrchr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chrchr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>chrchr9_KN196479v1_fix</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           CHROM     POS     rsID REF ALT  \\\n",
       "0                           chr1   69134  2205837   A   G   \n",
       "1                           chr1   69581  2252161   C   G   \n",
       "2                           chr1   69682  2396347   G   A   \n",
       "3                           chr1   69769  2288999   T   C   \n",
       "4                           chr1   69995  2351346   G   C   \n",
       "...                          ...     ...      ...  ..  ..   \n",
       "2353858  chrchr19_KI270938v1_alt  273806  2219599   G   A   \n",
       "2353859  chrchr19_KI270938v1_alt  273866  2237818   A   C   \n",
       "2353860  chrchr19_KI270938v1_alt  274366  2206666   G   C   \n",
       "2353861  chrchr19_KI270938v1_alt  275068  2241971   T   C   \n",
       "2353862   chrchr9_KN196479v1_fix   83614    17735  TC   T   \n",
       "\n",
       "                        Gene                  CLNSIG Covered_status  \n",
       "0                      OR4F5           Likely_benign    Not_Covered  \n",
       "1                      OR4F5  Uncertain_significance    Not_Covered  \n",
       "2                      OR4F5  Uncertain_significance    Not_Covered  \n",
       "3                      OR4F5  Uncertain_significance    Not_Covered  \n",
       "4                      OR4F5  Uncertain_significance    Not_Covered  \n",
       "...                      ...                     ...            ...  \n",
       "2353858               LILRA3           Likely_benign    Not_Covered  \n",
       "2353859               LILRA3  Uncertain_significance    Not_Covered  \n",
       "2353860               LILRA3  Uncertain_significance    Not_Covered  \n",
       "2353861  LOC126057115,LILRA3  Uncertain_significance    Not_Covered  \n",
       "2353862                  ABO                  Benign    Not_Covered  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new['CHROM'] = 'chr' + vcf_new['CHROM']\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "3962db13",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>chr9_KN196479v1_fix</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CHROM     POS     rsID REF ALT                 Gene  \\\n",
       "0                        chr1   69134  2205837   A   G                OR4F5   \n",
       "1                        chr1   69581  2252161   C   G                OR4F5   \n",
       "2                        chr1   69682  2396347   G   A                OR4F5   \n",
       "3                        chr1   69769  2288999   T   C                OR4F5   \n",
       "4                        chr1   69995  2351346   G   C                OR4F5   \n",
       "...                       ...     ...      ...  ..  ..                  ...   \n",
       "2353858  chr19_KI270938v1_alt  273806  2219599   G   A               LILRA3   \n",
       "2353859  chr19_KI270938v1_alt  273866  2237818   A   C               LILRA3   \n",
       "2353860  chr19_KI270938v1_alt  274366  2206666   G   C               LILRA3   \n",
       "2353861  chr19_KI270938v1_alt  275068  2241971   T   C  LOC126057115,LILRA3   \n",
       "2353862   chr9_KN196479v1_fix   83614    17735  TC   T                  ABO   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "0                 Likely_benign    Not_Covered  \n",
       "1        Uncertain_significance    Not_Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "...                         ...            ...  \n",
       "2353858           Likely_benign    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "2353862                  Benign    Not_Covered  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new['CHROM'] = vcf_new['CHROM'].replace({'chrchr19_KI270938v1_alt': 'chr19_KI270938v1_alt', 'chrchr17_KI270909v1_alt': 'chr17_KI270909v1_alt', \n",
    "                                            'chrchr22_KI270879v1_alt': 'chr22_KI270879v1_alt', 'chrchrUn_GL000218v1': 'chrUn_GL000218v1', \n",
    "                                            'chrchr9_KN196479v1_fix': 'chr9_KN196479v1_fix'})\n",
    "vcf_new "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e56abfe5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>chr9_KN196479v1_fix</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CHROM     POS     rsID REF ALT                 Gene  \\\n",
       "0                        chr1   69134  2205837   A   G                OR4F5   \n",
       "1                        chr1   69581  2252161   C   G                OR4F5   \n",
       "2                        chr1   69682  2396347   G   A                OR4F5   \n",
       "3                        chr1   69769  2288999   T   C                OR4F5   \n",
       "4                        chr1   69995  2351346   G   C                OR4F5   \n",
       "...                       ...     ...      ...  ..  ..                  ...   \n",
       "2353858  chr19_KI270938v1_alt  273806  2219599   G   A               LILRA3   \n",
       "2353859  chr19_KI270938v1_alt  273866  2237818   A   C               LILRA3   \n",
       "2353860  chr19_KI270938v1_alt  274366  2206666   G   C               LILRA3   \n",
       "2353861  chr19_KI270938v1_alt  275068  2241971   T   C  LOC126057115,LILRA3   \n",
       "2353862   chr9_KN196479v1_fix   83614    17735  TC   T                  ABO   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "0                 Likely_benign    Not_Covered  \n",
       "1        Uncertain_significance        Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "...                         ...            ...  \n",
       "2353858           Likely_benign    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "2353862                  Benign    Not_Covered  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Start_pos']\n",
    "    end_pos = row['End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in data\n",
    "vcf_new['Covered_status'] = vcf_new.apply(check_coverage, axis=1)\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "6624e57c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>rs2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>rs2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69682</td>\n",
       "      <td>rs2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69769</td>\n",
       "      <td>rs2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69995</td>\n",
       "      <td>rs2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273806</td>\n",
       "      <td>rs2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>rs2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>rs2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>rs2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>chr9_KN196479v1_fix</td>\n",
       "      <td>83614</td>\n",
       "      <td>rs17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CHROM     POS       rsID REF ALT                 Gene  \\\n",
       "0                        chr1   69134  rs2205837   A   G                OR4F5   \n",
       "1                        chr1   69581  rs2252161   C   G                OR4F5   \n",
       "2                        chr1   69682  rs2396347   G   A                OR4F5   \n",
       "3                        chr1   69769  rs2288999   T   C                OR4F5   \n",
       "4                        chr1   69995  rs2351346   G   C                OR4F5   \n",
       "...                       ...     ...        ...  ..  ..                  ...   \n",
       "2353858  chr19_KI270938v1_alt  273806  rs2219599   G   A               LILRA3   \n",
       "2353859  chr19_KI270938v1_alt  273866  rs2237818   A   C               LILRA3   \n",
       "2353860  chr19_KI270938v1_alt  274366  rs2206666   G   C               LILRA3   \n",
       "2353861  chr19_KI270938v1_alt  275068  rs2241971   T   C  LOC126057115,LILRA3   \n",
       "2353862   chr9_KN196479v1_fix   83614    rs17735  TC   T                  ABO   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "0                 Likely_benign    Not_Covered  \n",
       "1        Uncertain_significance        Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "...                         ...            ...  \n",
       "2353858           Likely_benign    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "2353862                  Benign    Not_Covered  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new['rsID'] = 'rs' + vcf_new['rsID'].astype('str')\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "666cc3d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>rs2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>rs2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69682</td>\n",
       "      <td>rs2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69769</td>\n",
       "      <td>rs2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69995</td>\n",
       "      <td>rs2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273806</td>\n",
       "      <td>rs2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>rs2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>rs2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>rs2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353862</th>\n",
       "      <td>chr9_KN196479v1_fix</td>\n",
       "      <td>83614</td>\n",
       "      <td>rs17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>ABO</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353863 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CHROM     POS       rsID REF ALT                 Gene  \\\n",
       "0                        chr1   69134  rs2205837   A   G                OR4F5   \n",
       "1                        chr1   69581  rs2252161   C   G                OR4F5   \n",
       "2                        chr1   69682  rs2396347   G   A                OR4F5   \n",
       "3                        chr1   69769  rs2288999   T   C                OR4F5   \n",
       "4                        chr1   69995  rs2351346   G   C                OR4F5   \n",
       "...                       ...     ...        ...  ..  ..                  ...   \n",
       "2353858  chr19_KI270938v1_alt  273806  rs2219599   G   A               LILRA3   \n",
       "2353859  chr19_KI270938v1_alt  273866  rs2237818   A   C               LILRA3   \n",
       "2353860  chr19_KI270938v1_alt  274366  rs2206666   G   C               LILRA3   \n",
       "2353861  chr19_KI270938v1_alt  275068  rs2241971   T   C  LOC126057115,LILRA3   \n",
       "2353862   chr9_KN196479v1_fix   83614    rs17735  TC   T                  ABO   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "0                 Likely_benign    Not_Covered  \n",
       "1        Uncertain_significance        Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "...                         ...            ...  \n",
       "2353858           Likely_benign    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "2353862                  Benign    Not_Covered  \n",
       "\n",
       "[2353863 rows x 8 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/clinvar_20240206.tsv', sep = '\\t')\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "9ac4a555",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>rs2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>rs2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69682</td>\n",
       "      <td>rs2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69769</td>\n",
       "      <td>rs2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69995</td>\n",
       "      <td>rs2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353857</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273753</td>\n",
       "      <td>rs2650447</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353858</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273806</td>\n",
       "      <td>rs2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>rs2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>rs2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>rs2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1183949 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CHROM     POS       rsID REF ALT                 Gene  \\\n",
       "0                        chr1   69134  rs2205837   A   G                OR4F5   \n",
       "1                        chr1   69581  rs2252161   C   G                OR4F5   \n",
       "2                        chr1   69682  rs2396347   G   A                OR4F5   \n",
       "3                        chr1   69769  rs2288999   T   C                OR4F5   \n",
       "4                        chr1   69995  rs2351346   G   C                OR4F5   \n",
       "...                       ...     ...        ...  ..  ..                  ...   \n",
       "2353857  chr19_KI270938v1_alt  273753  rs2650447   G   A               LILRA3   \n",
       "2353858  chr19_KI270938v1_alt  273806  rs2219599   G   A               LILRA3   \n",
       "2353859  chr19_KI270938v1_alt  273866  rs2237818   A   C               LILRA3   \n",
       "2353860  chr19_KI270938v1_alt  274366  rs2206666   G   C               LILRA3   \n",
       "2353861  chr19_KI270938v1_alt  275068  rs2241971   T   C  LOC126057115,LILRA3   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "0                 Likely_benign    Not_Covered  \n",
       "1        Uncertain_significance        Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "...                         ...            ...  \n",
       "2353857                  Benign    Not_Covered  \n",
       "2353858           Likely_benign    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "\n",
       "[1183949 rows x 8 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows_with_Tp53 = vcf_new[vcf_new['CHROM'].str.contains('chr1', case=False, na=False)]\n",
    "rows_with_Tp53"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "12940a12",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Uncertain_significance                                              1137676\n",
       "Likely_benign                                                        624662\n",
       "Benign                                                               193028\n",
       "Pathogenic                                                           135155\n",
       "Conflicting_classifications_of_pathogenicity                         107971\n",
       "                                                                     ...   \n",
       "Conflicting_classifications_of_pathogenicity|drug_response|other          1\n",
       "Benign/Likely_benign|risk_factor                                          1\n",
       "Benign/Likely_benign|other|risk_factor                                    1\n",
       "other|risk_factor                                                         1\n",
       "Likely_benign|risk_factor                                                 1\n",
       "Name: CLNSIG, Length: 90, dtype: int64"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new.CLNSIG.value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "4e1b319b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930200</td>\n",
       "      <td>rs1043045</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>chr1</td>\n",
       "      <td>935839</td>\n",
       "      <td>rs1085785</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939117</td>\n",
       "      <td>rs1427749</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941284</td>\n",
       "      <td>rs1006858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943995</td>\n",
       "      <td>rs950448</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353780</th>\n",
       "      <td>chrMT</td>\n",
       "      <td>15923</td>\n",
       "      <td>rs39575</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353812</th>\n",
       "      <td>chrMT</td>\n",
       "      <td>15958</td>\n",
       "      <td>rs870596</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353814</th>\n",
       "      <td>chrMT</td>\n",
       "      <td>15967</td>\n",
       "      <td>rs9572</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353822</th>\n",
       "      <td>chrMT</td>\n",
       "      <td>15990</td>\n",
       "      <td>rs9570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353826</th>\n",
       "      <td>chrMT</td>\n",
       "      <td>16002</td>\n",
       "      <td>rs690271</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>341887 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM     POS       rsID REF ALT    Gene  \\\n",
       "35        chr1  930200  rs1043045   G   A  SAMD11   \n",
       "147       chr1  935839  rs1085785   C   T  SAMD11   \n",
       "203       chr1  939117  rs1427749   G   A  SAMD11   \n",
       "312       chr1  941284  rs1006858   G   A  SAMD11   \n",
       "819       chr1  943995   rs950448   C   T  SAMD11   \n",
       "...        ...     ...        ...  ..  ..     ...   \n",
       "2353780  chrMT   15923    rs39575   A   G   MT-TT   \n",
       "2353812  chrMT   15958   rs870596   A   T   MT-TP   \n",
       "2353814  chrMT   15967     rs9572   G   A   MT-TP   \n",
       "2353822  chrMT   15990     rs9570   C   T   MT-TP   \n",
       "2353826  chrMT   16002   rs690271   T   C   MT-TP   \n",
       "\n",
       "                                               CLNSIG Covered_status  \n",
       "35       Conflicting_classifications_of_pathogenicity        Covered  \n",
       "147      Conflicting_classifications_of_pathogenicity        Covered  \n",
       "203      Conflicting_classifications_of_pathogenicity        Covered  \n",
       "312      Conflicting_classifications_of_pathogenicity        Covered  \n",
       "819                                        Pathogenic        Covered  \n",
       "...                                               ...            ...  \n",
       "2353780  Conflicting_classifications_of_pathogenicity    Not_Covered  \n",
       "2353812                                    Pathogenic    Not_Covered  \n",
       "2353814                                    Pathogenic    Not_Covered  \n",
       "2353822                             Likely_pathogenic    Not_Covered  \n",
       "2353826                  Pathogenic/Likely_pathogenic    Not_Covered  \n",
       "\n",
       "[341887 rows x 8 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows_with_Tp53 = vcf_new[vcf_new['CLNSIG'].str.contains('Pathogenic', case=False, na=False)]\n",
    "rows_with_Tp53"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "a135bf66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943995</td>\n",
       "      <td>rs950448</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1068</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976215</td>\n",
       "      <td>rs1320032</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1103</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013983</td>\n",
       "      <td>rs1028857</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1139</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>rs183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1173</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1014316</td>\n",
       "      <td>rs161455</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350894</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2787551</td>\n",
       "      <td>rs9754</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350895</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2787592</td>\n",
       "      <td>rs9751</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>SRY</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350897</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2787600</td>\n",
       "      <td>rs9753</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SRY</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350901</th>\n",
       "      <td>chrY</td>\n",
       "      <td>7063898</td>\n",
       "      <td>rs625467</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>LOC126057105,TBL1Y</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350905</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12735724</td>\n",
       "      <td>rs9757</td>\n",
       "      <td>TTAAG</td>\n",
       "      <td>T</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>126511 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM       POS       rsID    REF ALT                Gene      CLNSIG  \\\n",
       "819      chr1    943995   rs950448      C   T              SAMD11  Pathogenic   \n",
       "1068     chr1    976215  rs1320032      A   G               PERM1  Pathogenic   \n",
       "1103     chr1   1013983  rs1028857      G   A               ISG15  Pathogenic   \n",
       "1139     chr1   1014143   rs183381      C   T               ISG15  Pathogenic   \n",
       "1173     chr1   1014316   rs161455      C  CG               ISG15  Pathogenic   \n",
       "...       ...       ...        ...    ...  ..                 ...         ...   \n",
       "2350894  chrY   2787551     rs9754      C   T                 NaN  Pathogenic   \n",
       "2350895  chrY   2787592     rs9751      A   T                 SRY  Pathogenic   \n",
       "2350897  chrY   2787600     rs9753      G   A                 SRY  Pathogenic   \n",
       "2350901  chrY   7063898   rs625467      A   T  LOC126057105,TBL1Y  Pathogenic   \n",
       "2350905  chrY  12735724     rs9757  TTAAG   T               USP9Y  Pathogenic   \n",
       "\n",
       "        Covered_status  \n",
       "819            Covered  \n",
       "1068           Covered  \n",
       "1103           Covered  \n",
       "1139           Covered  \n",
       "1173           Covered  \n",
       "...                ...  \n",
       "2350894        Covered  \n",
       "2350895        Covered  \n",
       "2350897        Covered  \n",
       "2350901        Covered  \n",
       "2350905        Covered  \n",
       "\n",
       "[126511 rows x 8 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_result = rows_with_Tp53.loc[(rows_with_Tp53['CLNSIG'] == 'Pathogenic') & (rows_with_Tp53['Covered_status'] == 'Covered')]\n",
    "query_result "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "123e812a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Covered        2041980\n",
       "Not_Covered     311883\n",
       "Name: Covered_status, dtype: int64"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new.Covered_status.value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "48858a80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Uncertain_significance                                              1137676\n",
       "Likely_benign                                                        624662\n",
       "Benign                                                               193028\n",
       "Pathogenic                                                           135155\n",
       "Conflicting_classifications_of_pathogenicity                         107971\n",
       "                                                                     ...   \n",
       "Conflicting_classifications_of_pathogenicity|drug_response|other          1\n",
       "Benign/Likely_benign|risk_factor                                          1\n",
       "Benign/Likely_benign|other|risk_factor                                    1\n",
       "other|risk_factor                                                         1\n",
       "Likely_benign|risk_factor                                                 1\n",
       "Name: CLNSIG, Length: 90, dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new.CLNSIG.value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "999922e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>rs2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69682</td>\n",
       "      <td>rs2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69769</td>\n",
       "      <td>rs2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69995</td>\n",
       "      <td>rs2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925946</td>\n",
       "      <td>rs1924157</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353853</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273340</td>\n",
       "      <td>rs2249453</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353855</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273632</td>\n",
       "      <td>rs2360545</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353859</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>273866</td>\n",
       "      <td>rs2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353860</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>274366</td>\n",
       "      <td>rs2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2353861</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>275068</td>\n",
       "      <td>rs2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC126057115,LILRA3</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1137676 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CHROM     POS       rsID REF ALT                 Gene  \\\n",
       "1                        chr1   69581  rs2252161   C   G                OR4F5   \n",
       "2                        chr1   69682  rs2396347   G   A                OR4F5   \n",
       "3                        chr1   69769  rs2288999   T   C                OR4F5   \n",
       "4                        chr1   69995  rs2351346   G   C                OR4F5   \n",
       "5                        chr1  925946  rs1924157   C   G               SAMD11   \n",
       "...                       ...     ...        ...  ..  ..                  ...   \n",
       "2353853  chr19_KI270938v1_alt  273340  rs2249453   T   A               LILRA3   \n",
       "2353855  chr19_KI270938v1_alt  273632  rs2360545   G   A               LILRA3   \n",
       "2353859  chr19_KI270938v1_alt  273866  rs2237818   A   C               LILRA3   \n",
       "2353860  chr19_KI270938v1_alt  274366  rs2206666   G   C               LILRA3   \n",
       "2353861  chr19_KI270938v1_alt  275068  rs2241971   T   C  LOC126057115,LILRA3   \n",
       "\n",
       "                         CLNSIG Covered_status  \n",
       "1        Uncertain_significance        Covered  \n",
       "2        Uncertain_significance    Not_Covered  \n",
       "3        Uncertain_significance    Not_Covered  \n",
       "4        Uncertain_significance    Not_Covered  \n",
       "5        Uncertain_significance        Covered  \n",
       "...                         ...            ...  \n",
       "2353853  Uncertain_significance    Not_Covered  \n",
       "2353855  Uncertain_significance    Not_Covered  \n",
       "2353859  Uncertain_significance    Not_Covered  \n",
       "2353860  Uncertain_significance    Not_Covered  \n",
       "2353861  Uncertain_significance    Not_Covered  \n",
       "\n",
       "[1137676 rows x 8 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new[vcf_new['CLNSIG'] == 'Uncertain_significance']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "97422885",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69581</td>\n",
       "      <td>rs2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925946</td>\n",
       "      <td>rs1924157</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925952</td>\n",
       "      <td>rs1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925961</td>\n",
       "      <td>rs2069387</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925976</td>\n",
       "      <td>rs1362713</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350913</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12840450</td>\n",
       "      <td>rs2438505</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350914</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12842370</td>\n",
       "      <td>rs2627757</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350921</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14622379</td>\n",
       "      <td>rs559565</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NLGN4Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350924</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14830121</td>\n",
       "      <td>rs391879</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NLGN4Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350925</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14840423</td>\n",
       "      <td>rs2689591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NLGN4Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1035626 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM       POS       rsID REF ALT    Gene                  CLNSIG  \\\n",
       "1        chr1     69581  rs2252161   C   G   OR4F5  Uncertain_significance   \n",
       "5        chr1    925946  rs1924157   C   G  SAMD11  Uncertain_significance   \n",
       "6        chr1    925952  rs1019397   G   A  SAMD11  Uncertain_significance   \n",
       "8        chr1    925961  rs2069387   A   T  SAMD11  Uncertain_significance   \n",
       "10       chr1    925976  rs1362713   T   C  SAMD11  Uncertain_significance   \n",
       "...       ...       ...        ...  ..  ..     ...                     ...   \n",
       "2350913  chrY  12840450  rs2438505   C   T   USP9Y  Uncertain_significance   \n",
       "2350914  chrY  12842370  rs2627757   G   T   USP9Y  Uncertain_significance   \n",
       "2350921  chrY  14622379   rs559565   G   A  NLGN4Y  Uncertain_significance   \n",
       "2350924  chrY  14830121   rs391879   C   A  NLGN4Y  Uncertain_significance   \n",
       "2350925  chrY  14840423  rs2689591   C   T  NLGN4Y  Uncertain_significance   \n",
       "\n",
       "        Covered_status  \n",
       "1              Covered  \n",
       "5              Covered  \n",
       "6              Covered  \n",
       "8              Covered  \n",
       "10             Covered  \n",
       "...                ...  \n",
       "2350913        Covered  \n",
       "2350914        Covered  \n",
       "2350921        Covered  \n",
       "2350924        Covered  \n",
       "2350925        Covered  \n",
       "\n",
       "[1035626 rows x 8 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_result = vcf_new.loc[(vcf_new['CLNSIG'] == 'Uncertain_significance') & (vcf_new['Covered_status'] == 'Covered')]\n",
    "query_result "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "a95202c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2350907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12793053</td>\n",
       "      <td>rs2438504</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12818466</td>\n",
       "      <td>rs2499164</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350912</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12839999</td>\n",
       "      <td>rs2661889</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350913</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12840450</td>\n",
       "      <td>rs2438505</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350914</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12842370</td>\n",
       "      <td>rs2627757</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM       POS       rsID REF ALT   Gene                  CLNSIG  \\\n",
       "2350907  chrY  12793053  rs2438504   C   G  USP9Y  Uncertain_significance   \n",
       "2350910  chrY  12818466  rs2499164   T   C  USP9Y  Uncertain_significance   \n",
       "2350912  chrY  12839999  rs2661889   G   T  USP9Y  Uncertain_significance   \n",
       "2350913  chrY  12840450  rs2438505   C   T  USP9Y  Uncertain_significance   \n",
       "2350914  chrY  12842370  rs2627757   G   T  USP9Y  Uncertain_significance   \n",
       "\n",
       "        Covered_status  \n",
       "2350907        Covered  \n",
       "2350910        Covered  \n",
       "2350912        Covered  \n",
       "2350913        Covered  \n",
       "2350914        Covered  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_result = vcf_new.loc[(vcf_new['CLNSIG'] == 'Uncertain_significance') & (vcf_new['Gene'] == 'USP9Y') & (vcf_new['Covered_status'] == 'Covered')]\n",
    "query_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c55b3ed9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1137676"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "102050 + 1035626"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87e720c8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3067082f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66fdef4e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "d2468ad4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NaN                    21261\n",
       "BRCA2                  17174\n",
       "TTN-AS1,TTN            14138\n",
       "APC                    12955\n",
       "NF1                    11802\n",
       "                       ...  \n",
       "LOC124819405,SLC8B1        1\n",
       "LOC130008814,OAS3          1\n",
       "MIR6761,ALDH2              1\n",
       "CUX2,SH2B3                 1\n",
       "LOC126057115,LILRA3        1\n",
       "Name: Gene, Length: 27969, dtype: int64"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new.Gene.value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "448578fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1469884</th>\n",
       "      <td>chr12</td>\n",
       "      <td>111799896</td>\n",
       "      <td>rs728511</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>MIR6761,ALDH2</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM        POS      rsID REF ALT           Gene         CLNSIG  \\\n",
       "1469884  chr12  111799896  rs728511   C   T  MIR6761,ALDH2  Likely_benign   \n",
       "\n",
       "        Covered_status  \n",
       "1469884        Covered  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows_with_Tp53 = vcf_new[vcf_new['Gene'].str.contains('MIR6761', case=False, na=False)]\n",
    "rows_with_Tp53"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "61bd3ba5",
   "metadata": {},
   "outputs": [],
   "source": [
    "rows_with_Tp53.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/clinvar_20240206_IGHV.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9acaaf7e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "a777ce5d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tp53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>POLE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IGHV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene\n",
       "0  Tp53\n",
       "1  POLE\n",
       "2  IGHV"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/clinvar_20240206_genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "394d311a",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[38], line 9\u001b[0m\n\u001b[0;32m      7\u001b[0m         match \u001b[38;5;241m=\u001b[39m \u001b[38;5;28many\u001b[39m(gene \u001b[38;5;129;01min\u001b[39;00m df_gene[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGene\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mvalues \u001b[38;5;28;01mfor\u001b[39;00m gene \u001b[38;5;129;01min\u001b[39;00m gene_list)  \u001b[38;5;66;03m# Check if any gene in the list exists in df2\u001b[39;00m\n\u001b[0;32m      8\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m match:\n\u001b[1;32m----> 9\u001b[0m             vcf_new\u001b[38;5;241m.\u001b[39mloc[\u001b[43mvcf_new\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mGene\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m==\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mgenes\u001b[49m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGene_Match\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mYes\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m     11\u001b[0m vcf_new \u001b[38;5;241m=\u001b[39m vcf_new[vcf_new[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGene_Match\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mYes\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m     12\u001b[0m vcf_new\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\ops\\common.py:72\u001b[0m, in \u001b[0;36m_unpack_zerodim_and_defer.<locals>.new_method\u001b[1;34m(self, other)\u001b[0m\n\u001b[0;32m     68\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mNotImplemented\u001b[39m\n\u001b[0;32m     70\u001b[0m other \u001b[38;5;241m=\u001b[39m item_from_zerodim(other)\n\u001b[1;32m---> 72\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mmethod\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mother\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\arraylike.py:42\u001b[0m, in \u001b[0;36mOpsMixin.__eq__\u001b[1;34m(self, other)\u001b[0m\n\u001b[0;32m     40\u001b[0m \u001b[38;5;129m@unpack_zerodim_and_defer\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m__eq__\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     41\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__eq__\u001b[39m(\u001b[38;5;28mself\u001b[39m, other):\n\u001b[1;32m---> 42\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cmp_method\u001b[49m\u001b[43m(\u001b[49m\u001b[43mother\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moperator\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43meq\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:6243\u001b[0m, in \u001b[0;36mSeries._cmp_method\u001b[1;34m(self, other, op)\u001b[0m\n\u001b[0;32m   6240\u001b[0m rvalues \u001b[38;5;241m=\u001b[39m extract_array(other, extract_numpy\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, extract_range\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m   6242\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m np\u001b[38;5;241m.\u001b[39merrstate(\u001b[38;5;28mall\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[1;32m-> 6243\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m \u001b[43mops\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcomparison_op\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mop\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   6245\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_construct_result(res_values, name\u001b[38;5;241m=\u001b[39mres_name)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\ops\\array_ops.py:287\u001b[0m, in \u001b[0;36mcomparison_op\u001b[1;34m(left, right, op)\u001b[0m\n\u001b[0;32m    284\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m invalid_comparison(lvalues, rvalues, op)\n\u001b[0;32m    286\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m is_object_dtype(lvalues\u001b[38;5;241m.\u001b[39mdtype) \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(rvalues, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m--> 287\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m \u001b[43mcomp_method_OBJECT_ARRAY\u001b[49m\u001b[43m(\u001b[49m\u001b[43mop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrvalues\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    289\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    290\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m _na_arithmetic_op(lvalues, rvalues, op, is_cmp\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\ops\\array_ops.py:76\u001b[0m, in \u001b[0;36mcomp_method_OBJECT_ARRAY\u001b[1;34m(op, x, y)\u001b[0m\n\u001b[0;32m     74\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m     75\u001b[0m     result \u001b[38;5;241m=\u001b[39m libops\u001b[38;5;241m.\u001b[39mscalar_compare(x\u001b[38;5;241m.\u001b[39mravel(), y, op)\n\u001b[1;32m---> 76\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mresult\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreshape\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshape\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "vcf_new['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in vcf_new['Gene']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            vcf_new.loc[vcf_new['Gene'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "vcf_new = vcf_new[vcf_new['Gene_Match'] == 'Yes']\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "14ded4c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_new.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/clinvar_20240206_Tp53_PLOE_IGHV.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a6088c97",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "005d1e58",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Covered_status</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CHROM, POS, rsID, REF, ALT, Gene, CLNSIG, Covered_status, Gene_Match]\n",
       "Index: []"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05b0439c",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
