{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4dee9123",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading the Librabies completed\n",
      "Covered/Not_Covered completed\n",
      "Data Loading completed and Depth columns splitted\n",
      "Zygosity extraction completed\n",
      "Gene extraction completed\n",
      "CSQ splitting completed\n",
      "Required columns extraction completed\n",
      "Protein_HGVSc_HGVSp_extraction completed\n",
      "\"not_specified\", \"not_provided\" completed\n",
      "Scores added\n",
      "Specific Genes Mapped\n",
      "Lit Variants Mapped\n",
      "Exporting to excel\n",
      "VCF processing Completed and Saved as Excel File\n",
      "Filter portion started\n",
      "filtering completed\n",
      "Sheets saved: Got the final output\n"
     ]
    }
   ],
   "source": [
    "##################################### IMPORTING THE REQUIRED LIBRARIES ##########################################################\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "import warnings\n",
    "import xlsxwriter\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "print(\"Loading the Librabies completed\")\n",
    "\n",
    "######################################### GETTING THE COVERED POSITIONS FROM THE MOTHER VCF FILE FROM THE BED COORDINATES ####\n",
    "\n",
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/kalyani_mam_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHSTAFFGPTTL2_final.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHSTAFFGPTTL2_final.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n",
    "\n",
    "print(\"Covered/Not_Covered completed\")\n",
    "\n",
    "########################################### IMPORTING THE VCF DATA AND EXPANDING THE DEPTH COLUMNS ##########################\n",
    "\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHSTAFFGPTTL2_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "\n",
    "print('Data Loading completed and Depth columns splitted')\n",
    "\n",
    "######################################### EXTRACTING THE ZYGOSITY FROM THE INFO COLUMN OF THE EACH VARIANT ##############\n",
    "\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "\n",
    "print('Zygosity extraction completed')\n",
    "\n",
    "######################################## EXTRACTING THE GENEINFO FROM THE INFO COLUMN ####################################\n",
    "\n",
    "vcf[\"Gene_Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "\n",
    "print('Gene extraction completed')\n",
    "\n",
    "####################################### SPLITTING AND EXPLODING THE CSQ COLUMN FOR THE REQUIRED PARAMETERS ##############\n",
    "\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "\n",
    "print('CSQ splitting completed')\n",
    "\n",
    "###################################### EXTRACTION OF THE REQUIRED KEY-VALUE PAIRS FROM THE CSQ ##########################\n",
    "\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "\n",
    "print('Required columns extraction completed')\n",
    "\n",
    "############################################### Protein Position and Amino Acid Change ##################################\n",
    "\n",
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "\n",
    "############################################### HGVSc AND HGVSp TRANSCRIPTS EXTRACTION ###################################\n",
    "\n",
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf_final = vcf.copy()\n",
    "\n",
    "print('Protein_HGVSc_HGVSp_extraction completed')\n",
    "\n",
    "############################################### REMOVING THE [\"not_specified\", \"not_provided\"] FROM THE COLUMNS ##########\n",
    "\n",
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "print('\"not_specified\", \"not_provided\" completed')\n",
    "\n",
    "#################################################### REMOVING THE UNNESESSERY CHARACTERS FROM THE COLUMNS ###################\n",
    "\n",
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "\n",
    "############################################ CONSEQUENCE SCORES AND IMPACT SCORES MAPPING ###################################\n",
    "\n",
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "\n",
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "\n",
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "\n",
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "\n",
    "print('Scores added')\n",
    "\n",
    "############################################# CONDITION GENES MAPPING TO THE MAIN VCF ######################################\n",
    "\n",
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/genes/Conditions_final_genes.xlsx')\n",
    "\n",
    "merged_2['Gene Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "df_gene = df_gene.rename({'Gene Name':'Matched_Gene'}, axis=1)\n",
    "\n",
    "#merged_2 = merged_2.drop(columns=['Gene Match'], axis=1)\n",
    "\n",
    "merged_2 = pd.merge(merged_2, df_gene, on= 'Matched_Gene', how = 'left', sort = False)\n",
    "merged_2['Condition'] = merged_2['Condition'].fillna('No')\n",
    "merged_2['Headings'] = merged_2['Headings'].fillna('No')\n",
    "merged_2['21_Conditions_list'] = merged_2['21_Conditions_list'].fillna('No')\n",
    "merged_2['Gene_Score'] = merged_2['Gene_Score'].fillna('No')\n",
    "\n",
    "print(\"Specific Genes Mapped\")\n",
    "\n",
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/new_final_output_concatenated.xlsx')\n",
    "\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "\n",
    "merged_2 = merged_2.rename({'Matched_Gene':'Gene'}, axis=1)\n",
    "\n",
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "\n",
    "print(\"Lit Variants Mapped\")\n",
    "\n",
    "merged_3 = merged_3[['Gene Name','Gene', 'Gene_Score', 'Condition', 'Headings', '21_Conditions_list', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "\n",
    "print(\"Exporting to excel\")\n",
    "\n",
    "#merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/Gene_scores_KHAIGHGPTTL265_depth_vcf_processed.xlsx', index=False)\n",
    "\n",
    "print('VCF processing Completed and Saved as Excel File')\n",
    "\n",
    "print('Filter portion started')\n",
    "\n",
    "condition_filter = merged_3[merged_3['21_Conditions_list'] != 'No']\n",
    "\n",
    "consequence_filter = condition_filter[condition_filter['Consequence_score'].apply(lambda x: eval(x) >= 6/10)]\n",
    "\n",
    "consequence_filter['DP'] = consequence_filter['DP'].astype('int64')\n",
    "\n",
    "dp_filter = consequence_filter[consequence_filter['DP'] >= 15]\n",
    "\n",
    "dp_filter['gnomADe_AF'] = dp_filter['gnomADe_AF'].replace('', '0').astype(float)\n",
    "\n",
    "gnomADe_AF_filter = dp_filter[dp_filter['gnomADe_AF'] <= 0.6]\n",
    "\n",
    "gnomADe_AF_filter['gnomADe_SAS_AF'] = gnomADe_AF_filter['gnomADe_SAS_AF'].replace('', '0').astype(float)\n",
    "\n",
    "gnomADe_SAS_AF_filter = gnomADe_AF_filter[gnomADe_AF_filter['gnomADe_SAS_AF'] <= 0.6]\n",
    "\n",
    "df = gnomADe_SAS_AF_filter.copy()\n",
    "\n",
    "#########################################################################################################################\n",
    "\n",
    "# Replace empty strings with '0/0' and convert numerical parts to integers\n",
    "df['EXON'] = df['EXON'].replace('', '0/0')\n",
    "df['EXON_Numerator'] = df['EXON'].apply(lambda x: x.split('/')[0] if '/' in x else 0)\n",
    "df['EXON_Denominator'] = df['EXON'].apply(lambda x: x.split('/')[1] if '/' in x else 0)\n",
    "\n",
    "df['INTRON'] = df['INTRON'].replace('', '0/0')\n",
    "df['INTRON_Numerator'] = df['INTRON'].apply(lambda x: x.split('/')[0] if '/' in x else 0)\n",
    "df['INTRON_Denominator'] = df['INTRON'].apply(lambda x: x.split('/')[1] if '/' in x else 0)\n",
    "\n",
    "# Convert the data types of numerator and denominator columns to integers\n",
    "df['EXON_Numerator'] = df['EXON_Numerator'].astype(int)\n",
    "df['EXON_Denominator'] = df['EXON_Denominator'].astype(int)\n",
    "\n",
    "df['INTRON_Numerator'] = df['INTRON_Numerator'].astype(int)\n",
    "df['INTRON_Denominator'] = df['INTRON_Denominator'].astype(int)\n",
    "\n",
    "# Initialize an empty DataFrame to store the final result\n",
    "result_df = pd.DataFrame()\n",
    "\n",
    "# Iterate over unique combinations of CHROM, POS, rsID, REF, ALT\n",
    "for _, group_df in df.groupby(['CHROM', 'POS', 'REF', 'ALT']):\n",
    "    # Check if EXON column has values\n",
    "    if not group_df['EXON_Numerator'].eq(0).all():\n",
    "        # Prioritize rows with values in EXON column\n",
    "        result_df = pd.concat([result_df, group_df.sort_values(by=['EXON_Numerator'], ascending=False).head(1)])\n",
    "    else:\n",
    "        # If EXON is empty, prioritize rows with values in INTRON column\n",
    "        if not group_df['INTRON_Numerator'].eq(0).all():\n",
    "            result_df = pd.concat([result_df, group_df.sort_values(by=['INTRON_Numerator'], ascending=False).head(1)])\n",
    "        else:\n",
    "            # If both EXON and INTRON are empty, just concatenate the first row\n",
    "            result_df = pd.concat([result_df, group_df.head(1)])\n",
    "\n",
    "# Now result_df contains the rows you're looking for based on the specified logic\n",
    "result_df = result_df.drop(['EXON_Numerator', 'EXON_Denominator', 'INTRON_Numerator', 'INTRON_Denominator'], axis=1)\n",
    "\n",
    "result_df['EXON'] = result_df['EXON'].astype('str')\n",
    "result_df['INTRON'] = result_df['INTRON'].astype('str')\n",
    "                                  \n",
    "########################################################################################################################\n",
    "\n",
    "drop_duplicates_filter = result_df.copy()\n",
    "\n",
    "drop_duplicates_filter = drop_duplicates_filter.set_index(['Gene Name', 'Gene', 'Gene_Score', 'rsID', 'Literature', 'CHROM', 'POS', 'REF',\n",
    "       'ALT', 'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT',\n",
    "       'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "       'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)',\n",
    "       'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR',\n",
    "       'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF',\n",
    "       'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF',\n",
    "       'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF',\n",
    "       'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF',\n",
    "       'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF',\n",
    "       'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF',\n",
    "       'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']).apply(lambda x: x.str.split('; ').explode()).reset_index()\n",
    "# Specify the column names to move to the beginning\n",
    "columns_to_move = ['Condition', 'Headings', '21_Conditions_list']\n",
    "\n",
    "# Reorder the DataFrame to move specified columns to the beginning\n",
    "drop_duplicates_filter = drop_duplicates_filter[columns_to_move + [col for col in drop_duplicates_filter.columns if col not in columns_to_move]]\n",
    "\n",
    "print('filtering completed')\n",
    "\n",
    "#######################################################################################################################################\n",
    "\n",
    "# Sample list of condition names\n",
    "conditions_list = [\n",
    "    'Diabetes',\n",
    "    'High_Blood_Pressure',\n",
    "    'Cardiac_Health',\n",
    "    'Cholesterol_Disorders',\n",
    "    'Thyroid_Disorders',\n",
    "    'Parkinsons',\n",
    "    'Dementia',\n",
    "    'Headaches',\n",
    "    'Allergies',\n",
    "    'Anemia',\n",
    "    'Fatty_Liver',\n",
    "    'Gall_stones',\n",
    "    'Pancreatic_Disorders',\n",
    "    'Gut_Health',\n",
    "    'Gastritis',\n",
    "    'Glomerular_Diseases',\n",
    "    'Interstitial_Nephritis',\n",
    "    'Renal_stones',\n",
    "    'Skin_Health',\n",
    "    'Osteoporosis',\n",
    "    'Arthritis_Degenerative_Joint',\n",
    "    'Muscular_Dystrophy',\n",
    "    'Fatigue',\n",
    "    'Mood_Disorders',\n",
    "    'Obesity',\n",
    "    'Seizures',\n",
    "    'Bone_Joint_health',\n",
    "    'Muscular_health'\n",
    "]\n",
    "\n",
    "# Create an Excel writer object\n",
    "excel_writer = pd.ExcelWriter(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/KHSTAFFGPTTL2_output_new.xlsx', engine='xlsxwriter')\n",
    "\n",
    "# Define a workbook\n",
    "workbook = excel_writer.book\n",
    "\n",
    "# Define the green and pink cell formats\n",
    "green_format = workbook.add_format({'bg_color': '#00FF00'})  # Green color\n",
    "pink_format = workbook.add_format({'bg_color': '#FF00FF'})   # Pink color\n",
    "\n",
    "# A flag to keep track of whether at least one group was found\n",
    "found_at_least_one_group = False\n",
    "\n",
    "# Iterate through each condition and save it as a separate sheet\n",
    "for condition in conditions_list:\n",
    "    group = drop_duplicates_filter[drop_duplicates_filter['21_Conditions_list'] == condition]\n",
    "    if not group.empty:\n",
    "        found_at_least_one_group = True\n",
    "        # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "        sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', condition)[:31]\n",
    "\n",
    "        # Sort the group first by the \"Condition\" column and then by the \"Literature\" column\n",
    "        group_sorted = group.sort_values(by=['Condition', 'Literature'], ascending=[True, False])\n",
    "\n",
    "        # Delete the '21_Conditions_list' column\n",
    "        group_sorted = group_sorted.drop(columns=['21_Conditions_list', 'Gene Name'])\n",
    "\n",
    "        columns_to_keep = ['Condition', 'Headings', 'Gene', 'Gene_Score', 'Zygosity', 'CLIN_SIG','Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score', 'CHROM', 'POS','rsID','Literature', 'REF', 'ALT']\n",
    "\n",
    "        # Select only the desired columns\n",
    "        group_sorted = group_sorted[columns_to_keep]\n",
    "\n",
    "        # Write each group to a separate sheet with the modified sheet name\n",
    "        group_sorted.to_excel(excel_writer, sheet_name=sheet_name, index=False)\n",
    "\n",
    "        # Access the worksheet to apply conditional formatting\n",
    "        worksheet = excel_writer.sheets[sheet_name]\n",
    "\n",
    "        # Apply conditional formatting based on 'Literature' column values\n",
    "        for i, value in enumerate(group_sorted['Literature']):\n",
    "            if value == 'Yes':\n",
    "                worksheet.write(i + 1, columns_to_keep.index('Literature'), value, green_format)\n",
    "            elif value == 'No':\n",
    "                worksheet.write(i + 1, columns_to_keep.index('Literature'), value, pink_format)\n",
    "\n",
    "# If no groups were found, create a sheet named \"Condition not found\" and write the message in cell A1\n",
    "for condition in conditions_list:\n",
    "    if condition not in excel_writer.sheets:\n",
    "        # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "        sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', condition)[:31]\n",
    "        worksheet = excel_writer.book.add_worksheet(sheet_name)\n",
    "        worksheet.write(0, 0, \"Condition not found\")\n",
    "\n",
    "# Save the Excel file\n",
    "excel_writer.save()\n",
    "\n",
    "print('Sheets saved: Got the final output')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfbf0606",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b5b076e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a01a0449",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "5a7a88f9",
   "metadata": {},
   "source": [
    "# OLD code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72bd8d89",
   "metadata": {},
   "outputs": [],
   "source": [
    "##################################### IMPORTING THE REQUIRED LIBRARIES ##########################################################\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "import warnings\n",
    "import xlsxwriter\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "print(\"Loading the Librabies completed\")\n",
    "\n",
    "######################################### GETTING THE COVERED POSITIONS FROM THE MOTHER VCF FILE FROM THE BED COORDINATES ####\n",
    "\n",
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/kalyani_mam_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHSTAFFGPTTL2_final.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHSTAFFGPTTL2_final.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n",
    "\n",
    "print(\"Covered/Not_Covered completed\")\n",
    "\n",
    "########################################### IMPORTING THE VCF DATA AND EXPANDING THE DEPTH COLUMNS ##########################\n",
    "\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHSTAFFGPTTL2_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "\n",
    "print('Data Loading completed and Depth columns splitted')\n",
    "\n",
    "######################################### EXTRACTING THE ZYGOSITY FROM THE INFO COLUMN OF THE EACH VARIANT ##############\n",
    "\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "\n",
    "print('Zygosity extraction completed')\n",
    "\n",
    "######################################## EXTRACTING THE GENEINFO FROM THE INFO COLUMN ####################################\n",
    "\n",
    "vcf[\"Gene_Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "\n",
    "print('Gene extraction completed')\n",
    "\n",
    "####################################### SPLITTING AND EXPLODING THE CSQ COLUMN FOR THE REQUIRED PARAMETERS ##############\n",
    "\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "\n",
    "print('CSQ splitting completed')\n",
    "\n",
    "###################################### EXTRACTION OF THE REQUIRED KEY-VALUE PAIRS FROM THE CSQ ##########################\n",
    "\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "\n",
    "print('Required columns extraction completed')\n",
    "\n",
    "############################################### Protein Position and Amino Acid Change ##################################\n",
    "\n",
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "\n",
    "############################################### HGVSc AND HGVSp TRANSCRIPTS EXTRACTION ###################################\n",
    "\n",
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf_final = vcf.copy()\n",
    "\n",
    "print('Protein_HGVSc_HGVSp_extraction completed')\n",
    "\n",
    "############################################### REMOVING THE [\"not_specified\", \"not_provided\"] FROM THE COLUMNS ##########\n",
    "\n",
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "print('\"not_specified\", \"not_provided\" completed')\n",
    "\n",
    "#################################################### REMOVING THE UNNESESSERY CHARACTERS FROM THE COLUMNS ###################\n",
    "\n",
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "\n",
    "############################################ CONSEQUENCE SCORES AND IMPACT SCORES MAPPING ###################################\n",
    "\n",
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "\n",
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "\n",
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "\n",
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "\n",
    "print('Scores added')\n",
    "\n",
    "############################################# CONDITION GENES MAPPING TO THE MAIN VCF ######################################\n",
    "\n",
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/genes/Conditions_final_genes.xlsx')\n",
    "\n",
    "merged_2['Gene Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "df_gene = df_gene.rename({'Gene Name':'Matched_Gene'}, axis=1)\n",
    "\n",
    "#merged_2 = merged_2.drop(columns=['Gene Match'], axis=1)\n",
    "\n",
    "merged_2 = pd.merge(merged_2, df_gene, on= 'Matched_Gene', how = 'left', sort = False)\n",
    "merged_2['Condition'] = merged_2['Condition'].fillna('No')\n",
    "merged_2['Headings'] = merged_2['Headings'].fillna('No')\n",
    "merged_2['21_Conditions_list'] = merged_2['21_Conditions_list'].fillna('No')\n",
    "merged_2['Gene_Score'] = merged_2['Gene_Score'].fillna('No')\n",
    "\n",
    "print(\"Specific Genes Mapped\")\n",
    "\n",
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/new_final_output_concatenated.xlsx')\n",
    "\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "\n",
    "merged_2 = merged_2.rename({'Matched_Gene':'Gene'}, axis=1)\n",
    "\n",
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "\n",
    "print(\"Lit Variants Mapped\")\n",
    "\n",
    "merged_3 = merged_3[['Gene Name','Gene', 'Gene_Score', 'Condition', 'Headings', '21_Conditions_list', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "\n",
    "print(\"Exporting to excel\")\n",
    "\n",
    "#merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/Gene_scores_KHAIGHGPTTL265_depth_vcf_processed.xlsx', index=False)\n",
    "\n",
    "print('VCF processing Completed and Saved as Excel File')\n",
    "\n",
    "print('Filter portion started')\n",
    "\n",
    "condition_filter = merged_3[merged_3['21_Conditions_list'] != 'No']\n",
    "\n",
    "consequence_filter = condition_filter[condition_filter['Consequence_score'].apply(lambda x: eval(x) >= 6/10)]\n",
    "\n",
    "consequence_filter['DP'] = consequence_filter['DP'].astype('int64')\n",
    "\n",
    "dp_filter = consequence_filter[consequence_filter['DP'] >= 15]\n",
    "\n",
    "dp_filter['gnomADe_AF'] = dp_filter['gnomADe_AF'].replace('', '0').astype(float)\n",
    "\n",
    "gnomADe_AF_filter = dp_filter[dp_filter['gnomADe_AF'] <= 0.6]\n",
    "\n",
    "gnomADe_AF_filter['gnomADe_SAS_AF'] = gnomADe_AF_filter['gnomADe_SAS_AF'].replace('', '0').astype(float)\n",
    "\n",
    "gnomADe_SAS_AF_filter = gnomADe_AF_filter[gnomADe_AF_filter['gnomADe_SAS_AF'] <= 0.6]\n",
    "\n",
    "drop_duplicates_filter = gnomADe_SAS_AF_filter.drop_duplicates(subset=['Gene', 'rsID', 'CHROM', 'POS', 'REF', 'ALT'])\n",
    "\n",
    "drop_duplicates_filter = drop_duplicates_filter.set_index(['Gene Name', 'Gene', 'Gene_Score', 'rsID', 'Literature', 'CHROM', 'POS', 'REF',\n",
    "       'ALT', 'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT',\n",
    "       'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "       'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)',\n",
    "       'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR',\n",
    "       'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF',\n",
    "       'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF',\n",
    "       'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF',\n",
    "       'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF',\n",
    "       'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF',\n",
    "       'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF',\n",
    "       'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']).apply(lambda x: x.str.split('; ').explode()).reset_index()\n",
    "# Specify the column names to move to the beginning\n",
    "columns_to_move = ['Condition', 'Headings', '21_Conditions_list']\n",
    "\n",
    "# Reorder the DataFrame to move specified columns to the beginning\n",
    "drop_duplicates_filter = drop_duplicates_filter[columns_to_move + [col for col in drop_duplicates_filter.columns if col not in columns_to_move]]\n",
    "\n",
    "print('filtering completed')\n",
    "\n",
    "#######################################################################################################################################\n",
    "\n",
    "# Sample list of condition names\n",
    "conditions_list = [\n",
    "    'Diabetes',\n",
    "    'High_Blood_Pressure',\n",
    "    'Cardiac_Health',\n",
    "    'Cholesterol_Disorders',\n",
    "    'Thyroid_Disorders',\n",
    "    'Parkinsons',\n",
    "    'Dementia',\n",
    "    'Headaches',\n",
    "    'Allergies',\n",
    "    'Anemia',\n",
    "    'Fatty_Liver',\n",
    "    'Gall_stones',\n",
    "    'Pancreatic_Disorders',\n",
    "    'Gut_Health',\n",
    "    'Gastritis',\n",
    "    'Glomerular_Diseases',\n",
    "    'Interstitial_Nephritis',\n",
    "    'Renal_stones',\n",
    "    'Skin_Health',\n",
    "    'Osteoporosis',\n",
    "    'Arthritis_Degenerative_Joint',\n",
    "    'Muscular_Dystrophy',\n",
    "    'Fatigue',\n",
    "    'Mood_Disorders',\n",
    "    'Obesity',\n",
    "    'Seizures',\n",
    "    'Bone_Joint_health',\n",
    "    'Muscular_health'\n",
    "]\n",
    "\n",
    "# Create an Excel writer object\n",
    "excel_writer = pd.ExcelWriter(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/KHSTAFFGPTTL2_output.xlsx', engine='xlsxwriter')\n",
    "\n",
    "# Define a workbook\n",
    "workbook = excel_writer.book\n",
    "\n",
    "# Define the green and pink cell formats\n",
    "green_format = workbook.add_format({'bg_color': '#00FF00'})  # Green color\n",
    "pink_format = workbook.add_format({'bg_color': '#FF00FF'})   # Pink color\n",
    "\n",
    "# A flag to keep track of whether at least one group was found\n",
    "found_at_least_one_group = False\n",
    "\n",
    "# Iterate through each condition and save it as a separate sheet\n",
    "for condition in conditions_list:\n",
    "    group = drop_duplicates_filter[drop_duplicates_filter['21_Conditions_list'] == condition]\n",
    "    if not group.empty:\n",
    "        found_at_least_one_group = True\n",
    "        # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "        sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', condition)[:31]\n",
    "\n",
    "        # Sort the group first by the \"Condition\" column and then by the \"Literature\" column\n",
    "        group_sorted = group.sort_values(by=['Condition', 'Literature'], ascending=[True, False])\n",
    "\n",
    "        # Delete the '21_Conditions_list' column\n",
    "        group_sorted = group_sorted.drop(columns=['21_Conditions_list', 'Gene Name'])\n",
    "\n",
    "        columns_to_keep = ['Condition', 'Headings', 'Gene', 'Gene_Score', 'Zygosity', 'CLIN_SIG','Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score', 'CHROM', 'POS','rsID','Literature', 'REF', 'ALT']\n",
    "\n",
    "        # Select only the desired columns\n",
    "        group_sorted = group_sorted[columns_to_keep]\n",
    "\n",
    "        # Write each group to a separate sheet with the modified sheet name\n",
    "        group_sorted.to_excel(excel_writer, sheet_name=sheet_name, index=False)\n",
    "\n",
    "        # Access the worksheet to apply conditional formatting\n",
    "        worksheet = excel_writer.sheets[sheet_name]\n",
    "\n",
    "        # Apply conditional formatting based on 'Literature' column values\n",
    "        for i, value in enumerate(group_sorted['Literature']):\n",
    "            if value == 'Yes':\n",
    "                worksheet.write(i + 1, columns_to_keep.index('Literature'), value, green_format)\n",
    "            elif value == 'No':\n",
    "                worksheet.write(i + 1, columns_to_keep.index('Literature'), value, pink_format)\n",
    "\n",
    "# If no groups were found, create a sheet named \"Condition not found\" and write the message in cell A1\n",
    "for condition in conditions_list:\n",
    "    if condition not in excel_writer.sheets:\n",
    "        # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "        sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', condition)[:31]\n",
    "        worksheet = excel_writer.book.add_worksheet(sheet_name)\n",
    "        worksheet.write(0, 0, \"Condition not found\")\n",
    "\n",
    "# Save the Excel file\n",
    "excel_writer.save()\n",
    "\n",
    "print('Sheets saved: Got the final output')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
