{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "99224e9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "018b13f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  \n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5  \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...  \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...  \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...  \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...  \n",
       "...                                                   ...  \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...  \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...  \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...  \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...  \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...  \n",
       "\n",
       "[208911 rows x 4 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "088ba8dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  Extended_Start_pos  \\\n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5               65489   \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...               69007   \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...              450710   \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...              685686   \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...              924401   \n",
       "...                                                   ...                 ...   \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25038781   \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25041746   \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25043888   \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...            25622413   \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...            25624397   \n",
       "\n",
       "        Extended_End_pos  \n",
       "0                  65649  \n",
       "1                  70037  \n",
       "2                 451706  \n",
       "3                 686682  \n",
       "4                 924977  \n",
       "...                  ...  \n",
       "208906          25038941  \n",
       "208907          25041906  \n",
       "208908          25044048  \n",
       "208909          25624093  \n",
       "208910          25624557  \n",
       "\n",
       "[208911 rows x 6 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "690d9c22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  Extended_Start_pos  \\\n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5               65489   \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...               69007   \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...              450710   \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...              685686   \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...              924401   \n",
       "...                                                   ...                 ...   \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25038781   \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25041746   \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25043888   \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...            25622413   \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...            25624397   \n",
       "\n",
       "        Extended_End_pos    Gene  \n",
       "0                  65649   OR4F5  \n",
       "1                  70037   OR4F5  \n",
       "2                 451706  OR4F29  \n",
       "3                 686682  OR4F16  \n",
       "4                 924977  SAMD11  \n",
       "...                  ...     ...  \n",
       "208906          25038941   BPY2C  \n",
       "208907          25041906   BPY2C  \n",
       "208908          25044048   BPY2C  \n",
       "208909          25624093    CDY1  \n",
       "208910          25624557    CDY1  \n",
       "\n",
       "[208911 rows x 7 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Gene'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "30c5f7f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Extended_Start_pos  Extended_End_pos  \\\n",
       "0            chr1               65489             65649   \n",
       "1            chr1               69007             70037   \n",
       "2            chr1              450710            451706   \n",
       "3            chr1              685686            686682   \n",
       "4            chr1              924401            924977   \n",
       "...           ...                 ...               ...   \n",
       "208906       chrY            25038781          25038941   \n",
       "208907       chrY            25041746          25041906   \n",
       "208908       chrY            25043888          25044048   \n",
       "208909       chrY            25622413          25624093   \n",
       "208910       chrY            25624397          25624557   \n",
       "\n",
       "                                                     INFO    Gene  \n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5   OR4F5  \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...   OR4F5  \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...  OR4F29  \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...  OR4F16  \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...  SAMD11  \n",
       "...                                                   ...     ...  \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...   BPY2C  \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...   BPY2C  \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...   BPY2C  \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...    CDY1  \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...    CDY1  \n",
       "\n",
       "[208911 rows x 5 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'Gene']]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "66b69e3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>37892</th>\n",
       "      <td>chr3</td>\n",
       "      <td>11259007</td>\n",
       "      <td>11260529</td>\n",
       "      <td>ccds_id=CCDS2604.1;ensembl_gene_id=ENSG0000019...</td>\n",
       "      <td>HRH1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  Extended_Start_pos  Extended_End_pos  \\\n",
       "37892       chr3            11259007          11260529   \n",
       "\n",
       "                                                    INFO  Gene  \n",
       "37892  ccds_id=CCDS2604.1;ensembl_gene_id=ENSG0000019...  HRH1  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Value to check\n",
    "value = 11259021\n",
    "\n",
    "# Filter rows where the value lies between start and end positions\n",
    "filtered_df = df[(df['Extended_Start_pos'] <= value) & (df['Extended_End_pos'] >= value)]\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9968c06",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a8668069",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8347cd5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61f81c3b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43f60fbd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b47c81d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d6959dca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17-7674953-T-C, 17-7670654-T-C,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17-7674921-C-A, 17-7670703-C-A, 17-7670682-C-A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17-7674917-T-C, 17-7673323-T-C, 17-7676011-T-C...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17-7674894-G-A, 17-7674209-G-A, 17-7673300-G-A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>17-7578205-A-AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>17-7578397-G-GGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>17-7577070-A-AGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>17-7577095-A-ATCTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>17-7579470-G-GG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>424 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   POS\n",
       "0     17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A, \n",
       "1                     17-7674953-T-C, 17-7670654-T-C, \n",
       "2    17-7674921-C-A, 17-7670703-C-A, 17-7670682-C-A...\n",
       "3    17-7674917-T-C, 17-7673323-T-C, 17-7676011-T-C...\n",
       "4    17-7674894-G-A, 17-7674209-G-A, 17-7673300-G-A...\n",
       "..                                                 ...\n",
       "419                                    17-7578205-A-AT\n",
       "420                                   17-7578397-G-GGG\n",
       "421                                   17-7577070-A-AGA\n",
       "422                                 17-7577095-A-ATCTC\n",
       "423                                    17-7579470-G-GG\n",
       "\n",
       "[424 rows x 1 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_4/KHAIGHGPTTL190/POS_1.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "27a6d0af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>POS_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td>17-7676154-G-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td>17-7676154-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td>17-7676154-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17-7674953-T-C, 17-7670654-T-C,</td>\n",
       "      <td>17-7674953-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>17-7578205-A-AT</td>\n",
       "      <td>17-7578205-A-AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>17-7578397-G-GGG</td>\n",
       "      <td>17-7578397-G-GGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>17-7577070-A-AGA</td>\n",
       "      <td>17-7577070-A-AGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>17-7577095-A-ATCTC</td>\n",
       "      <td>17-7577095-A-ATCTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>17-7579470-G-GG</td>\n",
       "      <td>17-7579470-G-GG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>673 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  POS             POS_new\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,       17-7676154-G-C\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,       17-7676154-G-T\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,       17-7676154-G-A\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,                     \n",
       "1                    17-7674953-T-C, 17-7670654-T-C,       17-7674953-T-C\n",
       "..                                                ...                 ...\n",
       "419                                   17-7578205-A-AT     17-7578205-A-AT\n",
       "420                                  17-7578397-G-GGG    17-7578397-G-GGG\n",
       "421                                  17-7577070-A-AGA    17-7577070-A-AGA\n",
       "422                                17-7577095-A-ATCTC  17-7577095-A-ATCTC\n",
       "423                                   17-7579470-G-GG     17-7579470-G-GG\n",
       "\n",
       "[673 rows x 2 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['POS_new'] = df['POS'].str.split(', ')\n",
    "df = df.explode('POS_new')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7395cc66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>POS_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td>17-7676154-G-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td>17-7676154-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td>17-7676154-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17-7674953-T-C, 17-7670654-T-C,</td>\n",
       "      <td>17-7674953-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>17-7578205-A-AT</td>\n",
       "      <td>17-7578205-A-AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>17-7578397-G-GGG</td>\n",
       "      <td>17-7578397-G-GGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>17-7577070-A-AGA</td>\n",
       "      <td>17-7577070-A-AGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>17-7577095-A-ATCTC</td>\n",
       "      <td>17-7577095-A-ATCTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>17-7579470-G-GG</td>\n",
       "      <td>17-7579470-G-GG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>673 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  POS             POS_new\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,       17-7676154-G-C\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,       17-7676154-G-T\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,       17-7676154-G-A\n",
       "0    17-7676154-G-C, 17-7676154-G-T, 17-7676154-G-A,                     \n",
       "1                    17-7674953-T-C, 17-7670654-T-C,       17-7674953-T-C\n",
       "..                                                ...                 ...\n",
       "419                                   17-7578205-A-AT     17-7578205-A-AT\n",
       "420                                  17-7578397-G-GGG    17-7578397-G-GGG\n",
       "421                                  17-7577070-A-AGA    17-7577070-A-AGA\n",
       "422                                17-7577095-A-ATCTC  17-7577095-A-ATCTC\n",
       "423                                   17-7579470-G-GG     17-7579470-G-GG\n",
       "\n",
       "[673 rows x 2 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.POS_new.dropna()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "feac7c7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/scc_pos.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "85ecd609",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9-104858586-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9-104858586-C-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9-104826974-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9-104826974-C-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9-104824472-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2152</th>\n",
       "      <td>18-31595256-GTG-ATA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2153</th>\n",
       "      <td>18-31598691-GTG-ATT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2154</th>\n",
       "      <td>18-31595256-GTG-ATT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2155</th>\n",
       "      <td>18-31598691-GTG-ATC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>18-31595256-GTG-ATC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2157 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  POS_new\n",
       "0         9-104858586-C-T\n",
       "1         9-104858586-C-G\n",
       "2         9-104826974-C-T\n",
       "3         9-104826974-C-G\n",
       "4         9-104824472-T-C\n",
       "...                   ...\n",
       "2152  18-31595256-GTG-ATA\n",
       "2153  18-31598691-GTG-ATT\n",
       "2154  18-31595256-GTG-ATT\n",
       "2155  18-31598691-GTG-ATC\n",
       "2156  18-31595256-GTG-ATC\n",
       "\n",
       "[2157 rows x 1 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/Cardiac_pos.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d6370ae9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS_new</th>\n",
       "      <th>POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9-104858586-C-T</td>\n",
       "      <td>104858586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9-104858586-C-G</td>\n",
       "      <td>104858586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9-104826974-C-T</td>\n",
       "      <td>104826974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9-104826974-C-G</td>\n",
       "      <td>104826974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9-104824472-T-C</td>\n",
       "      <td>104824472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2152</th>\n",
       "      <td>18-31595256-GTG-ATA</td>\n",
       "      <td>31595256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2153</th>\n",
       "      <td>18-31598691-GTG-ATT</td>\n",
       "      <td>31598691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2154</th>\n",
       "      <td>18-31595256-GTG-ATT</td>\n",
       "      <td>31595256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2155</th>\n",
       "      <td>18-31598691-GTG-ATC</td>\n",
       "      <td>31598691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>18-31595256-GTG-ATC</td>\n",
       "      <td>31595256</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2157 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  POS_new        POS\n",
       "0         9-104858586-C-T  104858586\n",
       "1         9-104858586-C-G  104858586\n",
       "2         9-104826974-C-T  104826974\n",
       "3         9-104826974-C-G  104826974\n",
       "4         9-104824472-T-C  104824472\n",
       "...                   ...        ...\n",
       "2152  18-31595256-GTG-ATA   31595256\n",
       "2153  18-31598691-GTG-ATT   31598691\n",
       "2154  18-31595256-GTG-ATT   31595256\n",
       "2155  18-31598691-GTG-ATC   31598691\n",
       "2156  18-31595256-GTG-ATC   31595256\n",
       "\n",
       "[2157 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['POS'] = df['POS_new'].str.split('-').str[1]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "945a0545",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/Cardiac_pos.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "197953f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE  \n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  \n",
       "...                                                 ...  \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  \n",
       "\n",
       "[5413 rows x 10 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_4/KHHSPTGPCSP6/KHHSPTGPCSP6_annotated_indel.vcf', comment = '#', sep=\"\\t\", low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ab3618d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>6.192E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995E-5</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>43.33%</td>\n",
       "      <td>4.5813E-10</td>\n",
       "      <td>48</td>\n",
       "      <td>52</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>43</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>54.05%</td>\n",
       "      <td>2.7004E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.3461E-6</td>\n",
       "      <td>74</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.5723E-4</td>\n",
       "      <td>61</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.3988E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.6826E-4</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>93.75%</td>\n",
       "      <td>2.8282E-8</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE   GT  GQ SDP  DP  RD  \\\n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  0/1  22  11  11   5   \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  0/1  40  15  15   5   \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  0/1  93  60  60  34   \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  0/1  22  11  11   4   \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  0/1  75  37  37  17   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  1/1  58  16  16   1   \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  1/1  34  10  10   2   \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  0/1  24  18  18   9   \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  1/1  37  12  12   3   \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  1/1  75  16  16   1   \n",
       "\n",
       "      AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR  \n",
       "0      6  54.55%    6.192E-3  32  46   2   3   4   2  \n",
       "1     10  66.67%    9.995E-5  44  22   5   0  10   0  \n",
       "2     26  43.33%  4.5813E-10  48  52  21  13  17   9  \n",
       "3      6  54.55%    5.418E-3  43  55   4   0   6   0  \n",
       "4     20  54.05%   2.7004E-8  54  49   9   8  16   4  \n",
       "...   ..     ...         ...  ..  ..  ..  ..  ..  ..  \n",
       "5408  12     75%   1.3461E-6  74  60   1   0   9   3  \n",
       "5409   8     80%   3.5723E-4  61  52   2   0   8   0  \n",
       "5410   7  38.89%   3.3988E-3  32  35   9   0   7   0  \n",
       "5411   9     75%   1.6826E-4  41  31   3   0   8   1  \n",
       "5412  15  93.75%   2.8282E-8  37  55   1   0  12   3  \n",
       "\n",
       "[5413 rows x 24 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f5e0ec41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>6.192E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995E-5</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>43.33%</td>\n",
       "      <td>4.5813E-10</td>\n",
       "      <td>48</td>\n",
       "      <td>52</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>43</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>54.05%</td>\n",
       "      <td>2.7004E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.3461E-6</td>\n",
       "      <td>74</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.5723E-4</td>\n",
       "      <td>61</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.3988E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.6826E-4</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>93.75%</td>\n",
       "      <td>2.8282E-8</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE   GT  GQ SDP  DP  RD  \\\n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  0/1  22  11  11   5   \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  0/1  40  15  15   5   \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  0/1  93  60  60  34   \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  0/1  22  11  11   4   \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  0/1  75  37  37  17   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  1/1  58  16  16   1   \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  1/1  34  10  10   2   \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  0/1  24  18  18   9   \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  1/1  37  12  12   3   \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  1/1  75  16  16   1   \n",
       "\n",
       "      AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR HET HOM      Zygosity  \n",
       "0      6  54.55%    6.192E-3  32  46   2   3   4   2   1   0  Heterozygous  \n",
       "1     10  66.67%    9.995E-5  44  22   5   0  10   0   1   0  Heterozygous  \n",
       "2     26  43.33%  4.5813E-10  48  52  21  13  17   9   1   0  Heterozygous  \n",
       "3      6  54.55%    5.418E-3  43  55   4   0   6   0   1   0  Heterozygous  \n",
       "4     20  54.05%   2.7004E-8  54  49   9   8  16   4   1   0  Heterozygous  \n",
       "...   ..     ...         ...  ..  ..  ..  ..  ..  ..  ..  ..           ...  \n",
       "5408  12     75%   1.3461E-6  74  60   1   0   9   3   0   1    Homozygous  \n",
       "5409   8     80%   3.5723E-4  61  52   2   0   8   0   0   1    Homozygous  \n",
       "5410   7  38.89%   3.3988E-3  32  35   9   0   7   0   1   0  Heterozygous  \n",
       "5411   9     75%   1.6826E-4  41  31   3   0   8   1   0   1    Homozygous  \n",
       "5412  15  93.75%   2.8282E-8  37  55   1   0  12   3   0   1    Homozygous  \n",
       "\n",
       "[5413 rows x 27 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0491a138",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ADP=39;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.769,0.231,.,.;COMMON=1;G5;G5A;GENEINFO=HEPACAM:220296|HEPN1:641654|LOC107984406:107984406;INT;KGPhase3;LSD;NOV;PM;RS=560243987;RSPOS=124919336;SAO=0;SSR=0;TOPMED=0.76814156472986748,0.23183454383282364,0.00001592762487257,0.00000796381243628;U3;U5;VC=DIV;VLD;VP=0x050060c8000517002e100204;WGT=1;dbSNPBuildID=136'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.INFO.iloc[900]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2372d990",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>6.192E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995E-5</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>AGRN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>43.33%</td>\n",
       "      <td>4.5813E-10</td>\n",
       "      <td>48</td>\n",
       "      <td>52</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SCNN1D:6339</td>\n",
       "      <td>SCNN1D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>43</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MIR6808:102466740|DVL1:1855</td>\n",
       "      <td>MIR6808,DVL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>54.05%</td>\n",
       "      <td>2.7004E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.3461E-6</td>\n",
       "      <td>74</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.5723E-4</td>\n",
       "      <td>61</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY:100533178</td>\n",
       "      <td>PRORY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.3988E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.6826E-4</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>93.75%</td>\n",
       "      <td>2.8282E-8</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE   GT  GQ SDP  DP  RD  \\\n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  0/1  22  11  11   5   \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  0/1  40  15  15   5   \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  0/1  93  60  60  34   \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  0/1  22  11  11   4   \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  0/1  75  37  37  17   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  1/1  58  16  16   1   \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  1/1  34  10  10   2   \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  0/1  24  18  18   9   \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  1/1  37  12  12   3   \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  1/1  75  16  16   1   \n",
       "\n",
       "      AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      6  54.55%    6.192E-3  32  46   2   3   4   2   1   0  Heterozygous   \n",
       "1     10  66.67%    9.995E-5  44  22   5   0  10   0   1   0  Heterozygous   \n",
       "2     26  43.33%  4.5813E-10  48  52  21  13  17   9   1   0  Heterozygous   \n",
       "3      6  54.55%    5.418E-3  43  55   4   0   6   0   1   0  Heterozygous   \n",
       "4     20  54.05%   2.7004E-8  54  49   9   8  16   4   1   0  Heterozygous   \n",
       "...   ..     ...         ...  ..  ..  ..  ..  ..  ..  ..  ..           ...   \n",
       "5408  12     75%   1.3461E-6  74  60   1   0   9   3   0   1    Homozygous   \n",
       "5409   8     80%   3.5723E-4  61  52   2   0   8   0   0   1    Homozygous   \n",
       "5410   7  38.89%   3.3988E-3  32  35   9   0   7   0   1   0  Heterozygous   \n",
       "5411   9     75%   1.6826E-4  41  31   3   0   8   1   0   1    Homozygous   \n",
       "5412  15  93.75%   2.8282E-8  37  55   1   0  12   3   0   1    Homozygous   \n",
       "\n",
       "                        Gene_Name     Gene Name  \n",
       "0                   SAMD11:148398        SAMD11  \n",
       "1                     AGRN:375790          AGRN  \n",
       "2                     SCNN1D:6339        SCNN1D  \n",
       "3     MIR6808:102466740|DVL1:1855  MIR6808,DVL1  \n",
       "4                     MXRA8:54587         MXRA8  \n",
       "...                           ...           ...  \n",
       "5408                          NaN                \n",
       "5409              PRORY:100533178         PRORY  \n",
       "5410                          NaN                \n",
       "5411                          NaN                \n",
       "5412                          NaN                \n",
       "\n",
       "[5413 rows x 29 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene_Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "2ed67071",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>6.192E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995E-5</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>43.33%</td>\n",
       "      <td>4.5813E-10</td>\n",
       "      <td>48</td>\n",
       "      <td>52</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SCNN1D:6339</td>\n",
       "      <td>SCNN1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>43</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MIR6808:102466740|DVL1:1855</td>\n",
       "      <td>MIR6808,DVL1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>54.05%</td>\n",
       "      <td>2.7004E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.3461E-6</td>\n",
       "      <td>74</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.5723E-4</td>\n",
       "      <td>61</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY:100533178</td>\n",
       "      <td>PRORY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.3988E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.6826E-4</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>93.75%</td>\n",
       "      <td>2.8282E-8</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE   GT  GQ SDP  DP  RD  \\\n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  0/1  22  11  11   5   \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  0/1  40  15  15   5   \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  0/1  93  60  60  34   \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  0/1  22  11  11   4   \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  0/1  75  37  37  17   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  1/1  58  16  16   1   \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  1/1  34  10  10   2   \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  0/1  24  18  18   9   \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  1/1  37  12  12   3   \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  1/1  75  16  16   1   \n",
       "\n",
       "      AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      6  54.55%    6.192E-3  32  46   2   3   4   2   1   0  Heterozygous   \n",
       "1     10  66.67%    9.995E-5  44  22   5   0  10   0   1   0  Heterozygous   \n",
       "2     26  43.33%  4.5813E-10  48  52  21  13  17   9   1   0  Heterozygous   \n",
       "3      6  54.55%    5.418E-3  43  55   4   0   6   0   1   0  Heterozygous   \n",
       "4     20  54.05%   2.7004E-8  54  49   9   8  16   4   1   0  Heterozygous   \n",
       "...   ..     ...         ...  ..  ..  ..  ..  ..  ..  ..  ..           ...   \n",
       "5408  12     75%   1.3461E-6  74  60   1   0   9   3   0   1    Homozygous   \n",
       "5409   8     80%   3.5723E-4  61  52   2   0   8   0   0   1    Homozygous   \n",
       "5410   7  38.89%   3.3988E-3  32  35   9   0   7   0   1   0  Heterozygous   \n",
       "5411   9     75%   1.6826E-4  41  31   3   0   8   1   0   1    Homozygous   \n",
       "5412  15  93.75%   2.8282E-8  37  55   1   0  12   3   0   1    Homozygous   \n",
       "\n",
       "                        Gene_Name     Gene Name  CSQ  csq  \n",
       "0                   SAMD11:148398        SAMD11  NaN  NaN  \n",
       "1                     AGRN:375790          AGRN  NaN  NaN  \n",
       "2                     SCNN1D:6339        SCNN1D  NaN  NaN  \n",
       "3     MIR6808:102466740|DVL1:1855  MIR6808,DVL1  NaN  NaN  \n",
       "4                     MXRA8:54587         MXRA8  NaN  NaN  \n",
       "...                           ...           ...  ...  ...  \n",
       "5408                          NaN                NaN  NaN  \n",
       "5409              PRORY:100533178         PRORY  NaN  NaN  \n",
       "5410                          NaN                NaN  NaN  \n",
       "5411                          NaN                NaN  NaN  \n",
       "5412                          NaN                NaN  NaN  \n",
       "\n",
       "[5413 rows x 31 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "1072074f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>SYMBOL / Gene Name</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>6.192E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995E-5</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>43.33%</td>\n",
       "      <td>4.5813E-10</td>\n",
       "      <td>48</td>\n",
       "      <td>52</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SCNN1D:6339</td>\n",
       "      <td>SCNN1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>43</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MIR6808:102466740|DVL1:1855</td>\n",
       "      <td>MIR6808,DVL1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>54.05%</td>\n",
       "      <td>2.7004E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.3461E-6</td>\n",
       "      <td>74</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.5723E-4</td>\n",
       "      <td>61</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY:100533178</td>\n",
       "      <td>PRORY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.3988E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.6826E-4</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>93.75%</td>\n",
       "      <td>2.8282E-8</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE   GT  GQ SDP  DP  RD  \\\n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  0/1  22  11  11   5   \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  0/1  40  15  15   5   \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  0/1  93  60  60  34   \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  0/1  22  11  11   4   \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  0/1  75  37  37  17   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  1/1  58  16  16   1   \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  1/1  34  10  10   2   \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  0/1  24  18  18   9   \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  1/1  37  12  12   3   \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  1/1  75  16  16   1   \n",
       "\n",
       "      AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      6  54.55%    6.192E-3  32  46   2   3   4   2   1   0  Heterozygous   \n",
       "1     10  66.67%    9.995E-5  44  22   5   0  10   0   1   0  Heterozygous   \n",
       "2     26  43.33%  4.5813E-10  48  52  21  13  17   9   1   0  Heterozygous   \n",
       "3      6  54.55%    5.418E-3  43  55   4   0   6   0   1   0  Heterozygous   \n",
       "4     20  54.05%   2.7004E-8  54  49   9   8  16   4   1   0  Heterozygous   \n",
       "...   ..     ...         ...  ..  ..  ..  ..  ..  ..  ..  ..           ...   \n",
       "5408  12     75%   1.3461E-6  74  60   1   0   9   3   0   1    Homozygous   \n",
       "5409   8     80%   3.5723E-4  61  52   2   0   8   0   0   1    Homozygous   \n",
       "5410   7  38.89%   3.3988E-3  32  35   9   0   7   0   1   0  Heterozygous   \n",
       "5411   9     75%   1.6826E-4  41  31   3   0   8   1   0   1    Homozygous   \n",
       "5412  15  93.75%   2.8282E-8  37  55   1   0  12   3   0   1    Homozygous   \n",
       "\n",
       "                        Gene_Name     Gene Name  CSQ  csq SYMBOL / Gene Name  \\\n",
       "0                   SAMD11:148398        SAMD11  NaN  NaN                NaN   \n",
       "1                     AGRN:375790          AGRN  NaN  NaN                NaN   \n",
       "2                     SCNN1D:6339        SCNN1D  NaN  NaN                NaN   \n",
       "3     MIR6808:102466740|DVL1:1855  MIR6808,DVL1  NaN  NaN                NaN   \n",
       "4                     MXRA8:54587         MXRA8  NaN  NaN                NaN   \n",
       "...                           ...           ...  ...  ...                ...   \n",
       "5408                          NaN                NaN  NaN                NaN   \n",
       "5409              PRORY:100533178         PRORY  NaN  NaN                NaN   \n",
       "5410                          NaN                NaN  NaN                NaN   \n",
       "5411                          NaN                NaN  NaN                NaN   \n",
       "5412                          NaN                NaN  NaN                NaN   \n",
       "\n",
       "     ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar HGVSC HGVSP PolyPhen  \\\n",
       "0              NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "1              NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "2              NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "3              NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "4              NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "...            ...      ...                ...     ...   ...   ...      ...   \n",
       "5408           NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "5409           NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "5410           NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "5411           NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "5412           NaN      NaN                NaN     NaN   NaN   NaN      NaN   \n",
       "\n",
       "     BIOTYPE EXON INTRON Protein_position Amino_acids Codons STRAND PUBMED  \\\n",
       "0        NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "1        NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "2        NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "3        NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "4        NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "...      ...  ...    ...              ...         ...    ...    ...    ...   \n",
       "5408     NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "5409     NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "5410     NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "5411     NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "5412     NaN  NaN    NaN              NaN         NaN    NaN    NaN    NaN   \n",
       "\n",
       "     Consequence IMPACT SIFT  \n",
       "0            NaN    NaN  NaN  \n",
       "1            NaN    NaN  NaN  \n",
       "2            NaN    NaN  NaN  \n",
       "3            NaN    NaN  NaN  \n",
       "4            NaN    NaN  NaN  \n",
       "...          ...    ...  ...  \n",
       "5408         NaN    NaN  NaN  \n",
       "5409         NaN    NaN  NaN  \n",
       "5410         NaN    NaN  NaN  \n",
       "5411         NaN    NaN  NaN  \n",
       "5412         NaN    NaN  NaN  \n",
       "\n",
       "[5413 rows x 50 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['SYMBOL / Gene Name'] = vcf['csq'].str.split('|').str[3]\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "00609748",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>SYMBOL / Gene Name</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>6.192E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995E-5</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>AGRN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>43.33%</td>\n",
       "      <td>4.5813E-10</td>\n",
       "      <td>48</td>\n",
       "      <td>52</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SCNN1D:6339</td>\n",
       "      <td>SCNN1D</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>43</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MIR6808:102466740|DVL1:1855</td>\n",
       "      <td>MIR6808,DVL1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>54.05%</td>\n",
       "      <td>2.7004E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.3461E-6</td>\n",
       "      <td>74</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.5723E-4</td>\n",
       "      <td>61</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY:100533178</td>\n",
       "      <td>PRORY</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.3988E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.6826E-4</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>93.75%</td>\n",
       "      <td>2.8282E-8</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE   GT  GQ SDP  DP  RD  \\\n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  0/1  22  11  11   5   \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  0/1  40  15  15   5   \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  0/1  93  60  60  34   \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  0/1  22  11  11   4   \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  0/1  75  37  37  17   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  1/1  58  16  16   1   \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  1/1  34  10  10   2   \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  0/1  24  18  18   9   \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  1/1  37  12  12   3   \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  1/1  75  16  16   1   \n",
       "\n",
       "      AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      6  54.55%    6.192E-3  32  46   2   3   4   2   1   0  Heterozygous   \n",
       "1     10  66.67%    9.995E-5  44  22   5   0  10   0   1   0  Heterozygous   \n",
       "2     26  43.33%  4.5813E-10  48  52  21  13  17   9   1   0  Heterozygous   \n",
       "3      6  54.55%    5.418E-3  43  55   4   0   6   0   1   0  Heterozygous   \n",
       "4     20  54.05%   2.7004E-8  54  49   9   8  16   4   1   0  Heterozygous   \n",
       "...   ..     ...         ...  ..  ..  ..  ..  ..  ..  ..  ..           ...   \n",
       "5408  12     75%   1.3461E-6  74  60   1   0   9   3   0   1    Homozygous   \n",
       "5409   8     80%   3.5723E-4  61  52   2   0   8   0   0   1    Homozygous   \n",
       "5410   7  38.89%   3.3988E-3  32  35   9   0   7   0   1   0  Heterozygous   \n",
       "5411   9     75%   1.6826E-4  41  31   3   0   8   1   0   1    Homozygous   \n",
       "5412  15  93.75%   2.8282E-8  37  55   1   0  12   3   0   1    Homozygous   \n",
       "\n",
       "                        Gene_Name     Gene Name CSQ csq SYMBOL / Gene Name  \\\n",
       "0                   SAMD11:148398        SAMD11                              \n",
       "1                     AGRN:375790          AGRN                              \n",
       "2                     SCNN1D:6339        SCNN1D                              \n",
       "3     MIR6808:102466740|DVL1:1855  MIR6808,DVL1                              \n",
       "4                     MXRA8:54587         MXRA8                              \n",
       "...                           ...           ...  ..  ..                ...   \n",
       "5408                                                                         \n",
       "5409              PRORY:100533178         PRORY                              \n",
       "5410                                                                         \n",
       "5411                                                                         \n",
       "5412                                                                         \n",
       "\n",
       "     ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar HGVSC HGVSP PolyPhen  \\\n",
       "0                                                                             \n",
       "1                                                                             \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4                                                                             \n",
       "...            ...      ...                ...     ...   ...   ...      ...   \n",
       "5408                                                                          \n",
       "5409                                                                          \n",
       "5410                                                                          \n",
       "5411                                                                          \n",
       "5412                                                                          \n",
       "\n",
       "     BIOTYPE EXON INTRON Protein_position Amino_acids Codons STRAND PUBMED  \\\n",
       "0                                                                            \n",
       "1                                                                            \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...      ...  ...    ...              ...         ...    ...    ...    ...   \n",
       "5408                                                                         \n",
       "5409                                                                         \n",
       "5410                                                                         \n",
       "5411                                                                         \n",
       "5412                                                                         \n",
       "\n",
       "     Consequence IMPACT SIFT Protein Position and Amino Acid  \n",
       "0                                                             \n",
       "1                                                             \n",
       "2                                                             \n",
       "3                                                             \n",
       "4                                                             \n",
       "...          ...    ...  ...                             ...  \n",
       "5408                                                          \n",
       "5409                                                          \n",
       "5410                                                          \n",
       "5411                                                          \n",
       "5412                                                          \n",
       "\n",
       "[5413 rows x 51 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.fillna('', inplace=True)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "affb0e24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>SYMBOL / Gene Name</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943686</td>\n",
       "      <td>rs539654690</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>6.192E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995E-5</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>AGRN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1286108</td>\n",
       "      <td>rs555847974</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>43.33%</td>\n",
       "      <td>4.5813E-10</td>\n",
       "      <td>48</td>\n",
       "      <td>52</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SCNN1D:6339</td>\n",
       "      <td>SCNN1D</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341593</td>\n",
       "      <td>rs145370195</td>\n",
       "      <td>G</td>\n",
       "      <td>GACAC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>43</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MIR6808:102466740|DVL1:1855</td>\n",
       "      <td>MIR6808,DVL1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>54.05%</td>\n",
       "      <td>2.7004E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.3461E-6</td>\n",
       "      <td>74</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21431502</td>\n",
       "      <td>rs113251312</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.5723E-4</td>\n",
       "      <td>61</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PRORY:100533178</td>\n",
       "      <td>PRORY</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23188123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.3988E-3</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56855218</td>\n",
       "      <td>rs148562705;rs869025766</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.6826E-4</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5412</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57172007</td>\n",
       "      <td>.</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>93.75%</td>\n",
       "      <td>2.8282E-8</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5413 rows × 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                     rsID   REF    ALT QUAL FILTER  \\\n",
       "0     chr1    943686              rs539654690     T     TC    .   PASS   \n",
       "1     chr1   1043223               rs35881187   CCT      C    .   PASS   \n",
       "2     chr1   1286108              rs555847974   CAA      C    .   PASS   \n",
       "3     chr1   1341593              rs145370195     G  GACAC    .   PASS   \n",
       "4     chr1   1353987              rs140777846   CTG      C    .   PASS   \n",
       "...    ...       ...                      ...   ...    ...  ...    ...   \n",
       "5408  chrY  21311242              rs202147921     A     AT    .   PASS   \n",
       "5409  chrY  21431502              rs113251312     G     GA    .   PASS   \n",
       "5410  chrY  23188123                        .     G     GA    .   PASS   \n",
       "5411  chrY  56855218  rs148562705;rs869025766  TATA      T    .   PASS   \n",
       "5412  chrY  57172007                        .    TC      T    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9994,0....   \n",
       "1     ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5208,0....   \n",
       "2     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9978,0....   \n",
       "3     ADP=11;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.121,.,0...   \n",
       "4     ADP=37;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "...                                                 ...   \n",
       "5408  ADP=16;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "5409  ADP=10;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=PROR...   \n",
       "5410                       ADP=18;WT=0;HET=1;HOM=0;NC=0   \n",
       "5411  ADP=12;WT=0;HET=0;HOM=1;NC=0;ASP;RS=148562705,...   \n",
       "5412                       ADP=16;WT=0;HET=0;HOM=1;NC=0   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "5408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5412  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE   GT  GQ SDP  DP  RD  \\\n",
       "0        0/1:22:11:11:5:6:54.55%:6.192E-3:32:46:2:3:4:2  0/1  22  11  11   5   \n",
       "1      0/1:40:15:15:5:10:66.67%:9.995E-5:44:22:5:0:10:0  0/1  40  15  15   5   \n",
       "2     0/1:93:60:60:34:26:43.33%:4.5813E-10:48:52:21:...  0/1  93  60  60  34   \n",
       "3        0/1:22:11:11:4:6:54.55%:5.418E-3:43:55:4:0:6:0  0/1  22  11  11   4   \n",
       "4     0/1:75:37:37:17:20:54.05%:2.7004E-8:54:49:9:8:...  0/1  75  37  37  17   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "5408      1/1:58:16:16:1:12:75%:1.3461E-6:74:60:1:0:9:3  1/1  58  16  16   1   \n",
       "5409       1/1:34:10:10:2:8:80%:3.5723E-4:61:52:2:0:8:0  1/1  34  10  10   2   \n",
       "5410    0/1:24:18:18:9:7:38.89%:3.3988E-3:32:35:9:0:7:0  0/1  24  18  18   9   \n",
       "5411       1/1:37:12:12:3:9:75%:1.6826E-4:41:31:3:0:8:1  1/1  37  12  12   3   \n",
       "5412  1/1:75:16:16:1:15:93.75%:2.8282E-8:37:55:1:0:12:3  1/1  75  16  16   1   \n",
       "\n",
       "      AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      6  54.55%    6.192E-3  32  46   2   3   4   2   1   0  Heterozygous   \n",
       "1     10  66.67%    9.995E-5  44  22   5   0  10   0   1   0  Heterozygous   \n",
       "2     26  43.33%  4.5813E-10  48  52  21  13  17   9   1   0  Heterozygous   \n",
       "3      6  54.55%    5.418E-3  43  55   4   0   6   0   1   0  Heterozygous   \n",
       "4     20  54.05%   2.7004E-8  54  49   9   8  16   4   1   0  Heterozygous   \n",
       "...   ..     ...         ...  ..  ..  ..  ..  ..  ..  ..  ..           ...   \n",
       "5408  12     75%   1.3461E-6  74  60   1   0   9   3   0   1    Homozygous   \n",
       "5409   8     80%   3.5723E-4  61  52   2   0   8   0   0   1    Homozygous   \n",
       "5410   7  38.89%   3.3988E-3  32  35   9   0   7   0   1   0  Heterozygous   \n",
       "5411   9     75%   1.6826E-4  41  31   3   0   8   1   0   1    Homozygous   \n",
       "5412  15  93.75%   2.8282E-8  37  55   1   0  12   3   0   1    Homozygous   \n",
       "\n",
       "                        Gene_Name     Gene Name CSQ csq SYMBOL / Gene Name  \\\n",
       "0                   SAMD11:148398        SAMD11                              \n",
       "1                     AGRN:375790          AGRN                              \n",
       "2                     SCNN1D:6339        SCNN1D                              \n",
       "3     MIR6808:102466740|DVL1:1855  MIR6808,DVL1                              \n",
       "4                     MXRA8:54587         MXRA8                              \n",
       "...                           ...           ...  ..  ..                ...   \n",
       "5408                                                                         \n",
       "5409              PRORY:100533178         PRORY                              \n",
       "5410                                                                         \n",
       "5411                                                                         \n",
       "5412                                                                         \n",
       "\n",
       "     ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar HGVSC HGVSP PolyPhen  \\\n",
       "0                                                                             \n",
       "1                                                                             \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4                                                                             \n",
       "...            ...      ...                ...     ...   ...   ...      ...   \n",
       "5408                                                                          \n",
       "5409                                                                          \n",
       "5410                                                                          \n",
       "5411                                                                          \n",
       "5412                                                                          \n",
       "\n",
       "     BIOTYPE EXON INTRON Protein_position Amino_acids Codons STRAND PUBMED  \\\n",
       "0                                                                            \n",
       "1                                                                            \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...      ...  ...    ...              ...         ...    ...    ...    ...   \n",
       "5408                                                                         \n",
       "5409                                                                         \n",
       "5410                                                                         \n",
       "5411                                                                         \n",
       "5412                                                                         \n",
       "\n",
       "     Consequence IMPACT SIFT Protein Position and Amino Acid  \n",
       "0                                                        NaN  \n",
       "1                                                        NaN  \n",
       "2                                                        NaN  \n",
       "3                                                        NaN  \n",
       "4                                                        NaN  \n",
       "...          ...    ...  ...                             ...  \n",
       "5408                                                     NaN  \n",
       "5409                                                     NaN  \n",
       "5410                                                     NaN  \n",
       "5411                                                     NaN  \n",
       "5412                                                     NaN  \n",
       "\n",
       "[5413 rows x 51 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6afe3ade",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Row</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs539769126, CM110656, COSV54627430, -</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs146496725, CM110660, -, COSV99074759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-, rs1040877016, CM110671</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Row\n",
       "0  rs539769126, CM110656, COSV54627430, -\n",
       "1  rs146496725, CM110660, -, COSV99074759\n",
       "2               -, rs1040877016, CM110671"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a DataFrame with the given rows\n",
    "data = {\n",
    "    'Row': ['rs539769126, CM110656, COSV54627430, -',\n",
    "            'rs146496725, CM110660, -, COSV99074759',\n",
    "            '-, rs1040877016, CM110671']\n",
    "}\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "10c4c544",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Row</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs539769126-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs146496725-,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-, rs1040877016,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Row\n",
       "0       rs539769126-\n",
       "1     rs146496725-, \n",
       "2  -, rs1040877016, "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# Drop the words that start with \"C\"\n",
    "df['Row'] = df['Row'].str.replace(r'\\bC\\w*,?\\s*', '', regex=True)\n",
    "\n",
    "# Print the updated DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0952afe5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs146496725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs123456789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs987654321</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Value\n",
       "0  rs146496725\n",
       "1  rs123456789\n",
       "2  rs987654321"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a DataFrame with the column containing the values\n",
    "data = {'Value': ['rs146496725, ', 'rs123456789, ', 'rs987654321, ']}\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Remove the comma at the end of the values in the 'ID' column\n",
    "df['Value'] = df['Value'].str.replace(r',\\s*$', '', regex=True)\n",
    "\n",
    "# Print the updated DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c56f0755",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f8c1f070",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>, rs146496725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>, rs789456123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>, rs987654321</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID          Value\n",
       "0   1  , rs146496725\n",
       "1   2  , rs789456123\n",
       "2   3  , rs987654321"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a sample DataFrame\n",
    "data = {'ID': [1, 2, 3],\n",
    "        'Value': [', rs146496725', ', rs789456123', ', rs987654321']}\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Remove the comma and space before \"rsid\" in the 'Value' column\n",
    "df['Value'] = df['Value'].str.replace(r',\\s*(rsid)', r'\\1', regex=True)\n",
    "\n",
    "# Display the updated DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "146e41af",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51f51ab3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4bd1a036",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Nucleotide change</th>\n",
       "      <th>Amino acid change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRF1</td>\n",
       "      <td>c.272C&gt;T</td>\n",
       "      <td>Ala91Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRF1</td>\n",
       "      <td>c.386G &gt; C</td>\n",
       "      <td>p.Trp129Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRF1</td>\n",
       "      <td>c.50DelT</td>\n",
       "      <td>p.Leu17fsTer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRF1</td>\n",
       "      <td>c.190C&gt;T</td>\n",
       "      <td>p.Gln64Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRF1</td>\n",
       "      <td>c.148G&gt;A</td>\n",
       "      <td>p.Val50Met</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>STXBP2</td>\n",
       "      <td>c.795-4C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>RHOG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>CDC42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.R186C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>CDC42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.C188Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>CDC42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.*192C*24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Nucleotide change Amino acid change\n",
       "0     PRF1          c.272C>T          Ala91Val\n",
       "1     PRF1        c.386G > C       p.Trp129Ser\n",
       "2     PRF1          c.50DelT      p.Leu17fsTer\n",
       "3     PRF1          c.190C>T        p.Gln64Ter\n",
       "4     PRF1          c.148G>A        p.Val50Met\n",
       "..     ...               ...               ...\n",
       "73  STXBP2        c.795-4C>T               NaN\n",
       "74    RHOG               NaN               NaN\n",
       "75   CDC42               NaN           p.R186C\n",
       "76   CDC42               NaN           p.C188Y\n",
       "77   CDC42               NaN        p.*192C*24\n",
       "\n",
       "[78 rows x 3 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/HLH_Lit_variants (2).xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e55be56",
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Nucleotide change'].fillna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2232b1d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c29a03d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c8cb9be",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd8c2dae",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66ac0803",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8659fb3c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
