{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0da9435",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "2bcb0c59",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Haplotype</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Genomic_POS</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CYP2A6*2</td>\n",
       "      <td>rs1801272</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40848628.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>g.40848628A&gt;T</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CYP2A6*7</td>\n",
       "      <td>rs5031016</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40843869.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>g.40843869A&gt;G</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP2A6*7</td>\n",
       "      <td>rs59322119</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40843749.0</td>\n",
       "      <td>CC</td>\n",
       "      <td>C</td>\n",
       "      <td>g.40843750del</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CYP2A6*7</td>\n",
       "      <td>rs28399469</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40843746.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>g.40843746T&gt;C</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CYP2A6*7</td>\n",
       "      <td>rs28399470</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40843742.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>g.40843742G&gt;A</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724</th>\n",
       "      <td>G6PD*Musashino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154536019.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>g.154536019G&gt;A</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>725</th>\n",
       "      <td>G6PD*Kambos</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154536151.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>g.154536151G&gt;A</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>G6PD*Kamiube, Keelung</td>\n",
       "      <td>rs1557229502</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154532258.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>g.154532258G&gt;A</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>727</th>\n",
       "      <td>G6PD*Lages</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154546116.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>g.154546116C&gt;T</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>728</th>\n",
       "      <td>G6PD*Insuli</td>\n",
       "      <td>rs868950643</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154533004.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>g.154533004C&gt;T</td>\n",
       "      <td>Haplo</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>729 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Haplotype          rsID  CHROM          POS REF ALT  \\\n",
       "0                 CYP2A6*2     rs1801272  chr19   40848628.0   A   T   \n",
       "1                 CYP2A6*7     rs5031016  chr19   40843869.0   A   G   \n",
       "2                 CYP2A6*7    rs59322119  chr19   40843749.0  CC   C   \n",
       "3                 CYP2A6*7    rs28399469  chr19   40843746.0   T   C   \n",
       "4                 CYP2A6*7    rs28399470  chr19   40843742.0   G   A   \n",
       "..                     ...           ...    ...          ...  ..  ..   \n",
       "724         G6PD*Musashino           NaN   chrX  154536019.0   G   A   \n",
       "725            G6PD*Kambos           NaN   chrX  154536151.0   G   A   \n",
       "726  G6PD*Kamiube, Keelung  rs1557229502   chrX  154532258.0   G   A   \n",
       "727             G6PD*Lages           NaN   chrX  154546116.0   C   T   \n",
       "728            G6PD*Insuli   rs868950643   chrX  154533004.0   C   T   \n",
       "\n",
       "        Genomic_POS   Type  \n",
       "0     g.40848628A>T  Haplo  \n",
       "1     g.40843869A>G  Haplo  \n",
       "2     g.40843750del  Haplo  \n",
       "3     g.40843746T>C  Haplo  \n",
       "4     g.40843742G>A  Haplo  \n",
       "..              ...    ...  \n",
       "724  g.154536019G>A  Haplo  \n",
       "725  g.154536151G>A  Haplo  \n",
       "726  g.154532258G>A  Haplo  \n",
       "727  g.154546116C>T  Haplo  \n",
       "728  g.154533004C>T  Haplo  \n",
       "\n",
       "[729 rows x 8 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Haplotypes_1_12_2023/main_new_data.xlsx')\n",
    "df1\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGPRX1351_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR', 'Zygosity']]\n",
    "vcf\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos']]\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in data\n",
    "vcf['Covered/Not_Covered'] = vcf.apply(check_coverage, axis=1)\n",
    "vcf\n",
    "\n",
    "merged = pd.merge(vcf, df1, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort=False)\n",
    "merged\n",
    "\n",
    "df_1 = vcf.copy()\n",
    "df_2 = merged.copy()\n",
    "\n",
    "df_2['Gene'] = df_2['Haplotype'].str.split('*').str[0]\n",
    "df_2\n",
    "\n",
    "df_2 = df_2.drop_duplicates(subset=['Gene', 'Haplotype'], keep='first').reset_index(drop=True)\n",
    "df_2['Haplotype_new'] = \"*\" + df_2['Haplotype'].str.split('*').str[1]\n",
    "df_2\n",
    "\n",
    "import pandas as pd\n",
    "from itertools import combinations_with_replacement\n",
    "# Function to generate combinations\n",
    "def get_combinations(group):\n",
    "    return list(combinations_with_replacement(group, 2))\n",
    "\n",
    "# Apply the function to each group of rsID\n",
    "result = df_2.groupby('Gene')['Haplotype'].apply(get_combinations).explode().reset_index(name='Diplotype')\n",
    "\n",
    "# Extract the relevant portions of the haplotypes and keep the asterisk (*)\n",
    "result['Diplotype'] = result['Diplotype'].apply(lambda x: '/'.join([hap.replace(hap.split('*')[0], '') if len(hap.split('*')) > 1 else hap for hap in x]))\n",
    "\n",
    "# Filter out reversed combinations\n",
    "result = result[result['Diplotype'].apply(lambda x: x[0] != x[2])]\n",
    "\n",
    "# Drop duplicates\n",
    "result = result.drop_duplicates(subset=['Gene', 'Diplotype'], keep='first').reset_index(drop=True)\n",
    "\n",
    "# Print the result\n",
    "result\n",
    "\n",
    "df2 = df_2.copy()\n",
    "df2 = df2.rename(columns={'Haplotype_new': 'Diplotype', 'Covered/Not_Covered':'Coverage'})\n",
    "df2 = df2[['Gene', 'Diplotype', 'Coverage', 'Zygosity']]\n",
    "df2\n",
    "\n",
    "# Create an empty 'Coverage' column in dataset1\n",
    "result['Coverage'] = ''\n",
    "result['Zygosity'] = ''\n",
    "\n",
    "# Iterate over rows in dataset1\n",
    "for index, row in result.iterrows():\n",
    "    gene = row['Gene']\n",
    "    diplotype_parts = row['Diplotype'].split('/')\n",
    "    \n",
    "    # Check if each part of the diplotype is covered in dataset2\n",
    "    coverages = []\n",
    "    zygosities = []\n",
    "    for part in diplotype_parts:\n",
    "        coverage = df2.loc[(df2['Gene'] == gene) & (df2['Diplotype'] == part), 'Coverage'].values\n",
    "        zygosity = df2.loc[(df2['Gene'] == gene) & (df2['Diplotype'] == part), 'Zygosity'].values\n",
    "        \n",
    "        coverages.append(coverage[0] if coverage else 'Not Covered')\n",
    "        zygosities.append(zygosity[0] if zygosity else 'Not Specified')\n",
    "\n",
    "    # Set the 'Coverage' and 'Zygosity' column values in dataset1\n",
    "    result.at[index, 'Coverage'] = '/'.join(coverages)\n",
    "    result.at[index, 'Zygosity'] = '/'.join(zygosities)\n",
    "\n",
    "result[['Gene', 'Diplotype', 'Coverage', 'Zygosity']]\n",
    "result\n",
    "\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Diplotype_phenotype.xlsx')\n",
    "data\n",
    "\n",
    "result.Gene.value_counts()\n",
    "\n",
    "df_merged = pd.merge(result, data, on = ['Gene', 'Diplotype'], how = 'left', sort = False)\n",
    "df_merged\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Star_allele_function.xlsx')\n",
    "df_1\n",
    "\n",
    "# Function to map Diplotype to Function\n",
    "def map_diplotype_to_function(row):\n",
    "    gene = row['Gene']\n",
    "    diplotype_parts = row['Diplotype'].split('/')\n",
    "    \n",
    "    function_parts = [\n",
    "        df_1.loc[(df_1['Gene'] == gene) & (df_1['Allele'] == part), 'Function'].values[0] \n",
    "        if not df_1.loc[(df_1['Gene'] == gene) & (df_1['Allele'] == part), 'Function'].empty \n",
    "        else '-' for part in diplotype_parts\n",
    "    ]\n",
    "    \n",
    "    return '/'.join(function_parts)\n",
    "\n",
    "# Apply the mapping function to create a new column\n",
    "df_merged['New_Column'] = df_merged.apply(map_diplotype_to_function, axis=1)\n",
    "\n",
    "# Display the result\n",
    "result_df = df_merged[['Gene', 'Diplotype', 'Coverage','New_Column', 'Zygosity', 'Coded Diplotype/Phenotype Summary']]\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47473b69",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c2be065",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54873507",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "8d9fcfe9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Diplotype</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>New_Column</th>\n",
       "      <th>Coded Diplotype/Phenotype Summary</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>*7/*7</td>\n",
       "      <td>Not_Covered/Not_Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>*7/*9</td>\n",
       "      <td>Not_Covered/Not_Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>*7/*10</td>\n",
       "      <td>Not_Covered/Not_Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>*7/*19</td>\n",
       "      <td>Not_Covered/Not_Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>*7/*23</td>\n",
       "      <td>Not_Covered/Not_Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>*12A/*12A</td>\n",
       "      <td>Covered/Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>NUDT15</td>\n",
       "      <td>*2/*2</td>\n",
       "      <td>Covered/Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>NUDT15</td>\n",
       "      <td>*2/*3</td>\n",
       "      <td>Covered/Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>NUDT15</td>\n",
       "      <td>*3/*3</td>\n",
       "      <td>Covered/Covered</td>\n",
       "      <td>-/-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>SLCO1B1</td>\n",
       "      <td>*14/*14</td>\n",
       "      <td>Covered/Covered</td>\n",
       "      <td>Increased function/Increased function</td>\n",
       "      <td>Increased Function</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>161 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene  Diplotype                 Coverage  \\\n",
       "0     CYP2A6      *7/*7  Not_Covered/Not_Covered   \n",
       "1     CYP2A6      *7/*9  Not_Covered/Not_Covered   \n",
       "2     CYP2A6     *7/*10  Not_Covered/Not_Covered   \n",
       "3     CYP2A6     *7/*19  Not_Covered/Not_Covered   \n",
       "4     CYP2A6     *7/*23  Not_Covered/Not_Covered   \n",
       "..       ...        ...                      ...   \n",
       "156     NAT2  *12A/*12A          Covered/Covered   \n",
       "157   NUDT15      *2/*2          Covered/Covered   \n",
       "158   NUDT15      *2/*3          Covered/Covered   \n",
       "159   NUDT15      *3/*3          Covered/Covered   \n",
       "160  SLCO1B1    *14/*14          Covered/Covered   \n",
       "\n",
       "                                New_Column Coded Diplotype/Phenotype Summary  \n",
       "0                                      -/-                               NaN  \n",
       "1                                      -/-                               NaN  \n",
       "2                                      -/-                               NaN  \n",
       "3                                      -/-                               NaN  \n",
       "4                                      -/-                               NaN  \n",
       "..                                     ...                               ...  \n",
       "156                                    -/-                               NaN  \n",
       "157                                    -/-                               NaN  \n",
       "158                                    -/-                               NaN  \n",
       "159                                    -/-                               NaN  \n",
       "160  Increased function/Increased function                Increased Function  \n",
       "\n",
       "[161 rows x 5 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Haplotypes_1_12_2023/main_new_data.xlsx')\n",
    "df1\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGPRX1351_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR', 'Zygosity']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos']]\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in data\n",
    "vcf['Covered/Not_Covered'] = vcf.apply(check_coverage, axis=1)\n",
    "#vcf = vcf[vcf['Covered/Not_Covered'] == 'Covered']\n",
    "#vcf[\"Gene Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "#vcf['Gene Name'] = vcf['Gene Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "#vcf.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGPRX1331_snp_pos.xlsx', index=False)\n",
    "vcf\n",
    "\n",
    "merged = pd.merge(vcf, df1, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort=False)\n",
    "merged\n",
    "\n",
    "df_1 = vcf.copy()\n",
    "df_2 = merged.copy()\n",
    "\n",
    "df_2['Gene'] = df_2['Haplotype'].str.split('*').str[0]\n",
    "df_2\n",
    "\n",
    "df_2 = df_2.drop_duplicates(subset=['Gene', 'Haplotype'], keep='first').reset_index(drop=True)\n",
    "df_2['Haplotype_new'] = \"*\" + df_2['Haplotype'].str.split('*').str[1]\n",
    "df_2\n",
    "\n",
    "import pandas as pd\n",
    "from itertools import combinations_with_replacement\n",
    "# Function to generate combinations\n",
    "def get_combinations(group):\n",
    "    return list(combinations_with_replacement(group, 2))\n",
    "\n",
    "# Apply the function to each group of rsID\n",
    "result = df_2.groupby('Gene')['Haplotype'].apply(get_combinations).explode().reset_index(name='Diplotype')\n",
    "\n",
    "# Extract the relevant portions of the haplotypes and keep the asterisk (*)\n",
    "result['Diplotype'] = result['Diplotype'].apply(lambda x: '/'.join([hap.replace(hap.split('*')[0], '') if len(hap.split('*')) > 1 else hap for hap in x]))\n",
    "\n",
    "# Filter out reversed combinations\n",
    "result = result[result['Diplotype'].apply(lambda x: x[0] != x[2])]\n",
    "\n",
    "# Drop duplicates\n",
    "result = result.drop_duplicates(subset=['Gene', 'Diplotype'], keep='first').reset_index(drop=True)\n",
    "\n",
    "# Print the result\n",
    "result\n",
    "\n",
    "df2 = df_2.copy()\n",
    "df2 = df2.rename(columns={'Haplotype_new': 'Diplotype', 'Covered/Not_Covered':'Coverage'})\n",
    "df2 = df2[['Gene', 'Diplotype', 'Coverage', 'Zygosity']]\n",
    "df2\n",
    "\n",
    "# Create an empty 'Coverage' column in dataset1\n",
    "result['Coverage'] = ''\n",
    "\n",
    "# Iterate over rows in dataset1\n",
    "for index, row in result.iterrows():\n",
    "    gene = row['Gene']\n",
    "    diplotype_parts = row['Diplotype'].split('/')\n",
    "    \n",
    "    # Check if each part of the diplotype is covered in dataset2\n",
    "    coverages = []\n",
    "    for part in diplotype_parts:\n",
    "        coverage = df2.loc[(df2['Gene'] == gene) & (df2['Diplotype'] == part), 'Coverage'].values\n",
    "        coverages.append(coverage[0] if coverage else 'Not Covered')\n",
    "\n",
    "    # Set the 'Coverage' column value in dataset1\n",
    "    result.at[index, 'Coverage'] = '/'.join(coverages)\n",
    "    \n",
    "result\n",
    "\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Diplotype_phenotype.xlsx')\n",
    "data\n",
    "\n",
    "result.Gene.value_counts()\n",
    "\n",
    "df_merged = pd.merge(result, data, on = ['Gene', 'Diplotype'], how = 'left', sort = False)\n",
    "df_merged\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Star_allele_function.xlsx')\n",
    "df_1\n",
    "\n",
    "# Function to map Diplotype to Function\n",
    "def map_diplotype_to_function(row):\n",
    "    gene = row['Gene']\n",
    "    diplotype_parts = row['Diplotype'].split('/')\n",
    "    \n",
    "    function_parts = [\n",
    "        df_1.loc[(df_1['Gene'] == gene) & (df_1['Allele'] == part), 'Function'].values[0] \n",
    "        if not df_1.loc[(df_1['Gene'] == gene) & (df_1['Allele'] == part), 'Function'].empty \n",
    "        else '-' for part in diplotype_parts\n",
    "    ]\n",
    "    \n",
    "    return '/'.join(function_parts)\n",
    "\n",
    "# Apply the mapping function to create a new column\n",
    "df_merged['New_Column'] = df_merged.apply(map_diplotype_to_function, axis=1)\n",
    "\n",
    "# Display the result\n",
    "result_df = df_merged[['Gene', 'Diplotype', 'Coverage','New_Column', 'Coded Diplotype/Phenotype Summary']]\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "d055054a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14653</td>\n",
       "      <td>rs62635297</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14907</td>\n",
       "      <td>rs6682375</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>45%</td>\n",
       "      <td>3.1919E-7</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.3347E-8</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14937</td>\n",
       "      <td>rs11807796</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>36</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>28</td>\n",
       "      <td>11</td>\n",
       "      <td>28.21%</td>\n",
       "      <td>2.1547E-4</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=36;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>110</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>10</td>\n",
       "      <td>26</td>\n",
       "      <td>72.22%</td>\n",
       "      <td>9.2118E-12</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209365</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>36%</td>\n",
       "      <td>8.1542E-4</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74652</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=0;HOM=1;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>115</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>92%</td>\n",
       "      <td>2.7767E-12</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74653</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209764</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>55</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>85.71%</td>\n",
       "      <td>2.9913E-6</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74654</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211706</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>14</td>\n",
       "      <td>7</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>4.3101E-3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74655</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211712</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=23;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>30.43%</td>\n",
       "      <td>4.5803E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74656 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS        rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14653  rs62635297   C   T    .   PASS   \n",
       "1      chr1     14907   rs6682375   A   G    .   PASS   \n",
       "2      chr1     14930   rs6682385   A   G    .   PASS   \n",
       "3      chr1     14937  rs11807796   T   C    .   PASS   \n",
       "4      chr1     16495   rs3210724   G   C    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "74651  chrY  57209365           .   T   C    .   PASS   \n",
       "74652  chrY  57209613           .   A   G    .   PASS   \n",
       "74653  chrY  57209764           .   C   T    .   PASS   \n",
       "74654  chrY  57211706           .   G   C    .   PASS   \n",
       "74655  chrY  57211712           .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=26;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1   23  26  26   \n",
       "1      ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1   64  40  40   \n",
       "2      ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....  0/1   73  41  41   \n",
       "3      ADP=39;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...  0/1   36  39  39   \n",
       "4      ADP=36;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...  0/1  110  36  36   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "74651  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1   30  25  25   \n",
       "74652  ADP=25;WT=0;HET=0;HOM=1;NC=0;CSQ=G|non_coding_...  1/1  115  25  25   \n",
       "74653  ADP=14;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_vari...  1/1   55  14  14   \n",
       "74654  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1   23  21  21   \n",
       "74655  ADP=23;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...  0/1   23  23  23   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR      Zygosity  \n",
       "0      19   7  26.92%   4.9169E-3  15   4   6   1  Heterozygous  \n",
       "1      22  18     45%   3.1919E-7   8  14  16   2  Heterozygous  \n",
       "2      21  20  48.78%   4.3347E-8   8  13  15   5  Heterozygous  \n",
       "3      28  11  28.21%   2.1547E-4  20   8   2   9  Heterozygous  \n",
       "4      10  26  72.22%  9.2118E-12  10   0  25   1  Heterozygous  \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..           ...  \n",
       "74651  16   9     36%   8.1542E-4  13   3   2   7  Heterozygous  \n",
       "74652   2  23     92%  2.7767E-12   0   2  18   5    Homozygous  \n",
       "74653   2  12  85.71%   2.9913E-6   2   0   2  10    Homozygous  \n",
       "74654  14   7  33.33%   4.3101E-3   7   7   5   2  Heterozygous  \n",
       "74655  16   7  30.43%   4.5803E-3  10   6   4   3  Heterozygous  \n",
       "\n",
       "[74656 rows x 21 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGPRX1351_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR', 'Zygosity']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62c99374",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
