{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "44c9aec1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "8d532195",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/srinivas_sir_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHCDPRGPTTL125_final.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL125_final.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "579c11c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.9117E-30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943996</td>\n",
       "      <td>rs374426973</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=45;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9996,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>95</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>20</td>\n",
       "      <td>25</td>\n",
       "      <td>55.56%</td>\n",
       "      <td>2.7295E-10</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.07907,0...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37638</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5100327</td>\n",
       "      <td>rs2524543</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37639</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5100614</td>\n",
       "      <td>rs2563389</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>254</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.8097E-26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37640</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12816280</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=48;WT=0;HET=0;HOM=1;NC=0;CSQ=T|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.554E-28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37643 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "2      chr1    942451     rs6672356   T   C    .   PASS   \n",
       "3      chr1    943996   rs374426973   G   A    .   PASS   \n",
       "4      chr1    944858     rs3748592   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "37638  chrY   5100327     rs2524543   G   T    .   PASS   \n",
       "37639  chrY   5100614     rs2563389   T   G    .   PASS   \n",
       "37640  chrY  12816280             .   C   T    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37642  chrY  56961295             .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "2      ADP=50;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...  1/1  255  50  50   \n",
       "3      ADP=45;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9996,0....  0/1   95  45  45   \n",
       "4      ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.07907,0...  1/1  164  29  29   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "37638  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1  182  32  32   \n",
       "37639  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1  254  44  44   \n",
       "37640  ADP=48;WT=0;HET=0;HOM=1;NC=0;CSQ=T|missense_va...  1/1  255  48  48   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37642  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...  1/1   46   9   9   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR  \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1  \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9  \n",
       "2       0  50    100%  9.9117E-30   0   0  34  16  \n",
       "3      20  25  55.56%  2.7295E-10  15   5  20   5  \n",
       "4       0  29    100%  3.3259E-17   0   0  17  12  \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  \n",
       "37638   0  32    100%  5.4567E-19   0   0  17  15  \n",
       "37639   0  44    100%  3.8097E-26   0   0  24  20  \n",
       "37640   0  48    100%   1.554E-28   0   0  23  25  \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5  \n",
       "37642   0   9    100%   2.0568E-5   0   0   6   3  \n",
       "\n",
       "[37643 rows x 20 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL125_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "fdfd6b25",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.9117E-30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943996</td>\n",
       "      <td>rs374426973</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=45;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9996,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>95</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>20</td>\n",
       "      <td>25</td>\n",
       "      <td>55.56%</td>\n",
       "      <td>2.7295E-10</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.07907,0...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37638</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5100327</td>\n",
       "      <td>rs2524543</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37639</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5100614</td>\n",
       "      <td>rs2563389</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>254</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.8097E-26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37640</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12816280</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=48;WT=0;HET=0;HOM=1;NC=0;CSQ=T|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.554E-28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37643 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "2      chr1    942451     rs6672356   T   C    .   PASS   \n",
       "3      chr1    943996   rs374426973   G   A    .   PASS   \n",
       "4      chr1    944858     rs3748592   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "37638  chrY   5100327     rs2524543   G   T    .   PASS   \n",
       "37639  chrY   5100614     rs2563389   T   G    .   PASS   \n",
       "37640  chrY  12816280             .   C   T    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37642  chrY  56961295             .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "2      ADP=50;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...  1/1  255  50  50   \n",
       "3      ADP=45;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9996,0....  0/1   95  45  45   \n",
       "4      ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.07907,0...  1/1  164  29  29   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "37638  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1  182  32  32   \n",
       "37639  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1  254  44  44   \n",
       "37640  ADP=48;WT=0;HET=0;HOM=1;NC=0;CSQ=T|missense_va...  1/1  255  48  48   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37642  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...  1/1   46   9   9   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous  \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous  \n",
       "2       0  50    100%  9.9117E-30   0   0  34  16   0   1    Homozygous  \n",
       "3      20  25  55.56%  2.7295E-10  15   5  20   5   1   0  Heterozygous  \n",
       "4       0  29    100%  3.3259E-17   0   0  17  12   0   1    Homozygous  \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  ..  ..           ...  \n",
       "37638   0  32    100%  5.4567E-19   0   0  17  15   0   1    Homozygous  \n",
       "37639   0  44    100%  3.8097E-26   0   0  24  20   0   1    Homozygous  \n",
       "37640   0  48    100%   1.554E-28   0   0  23  25   0   1    Homozygous  \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous  \n",
       "37642   0   9    100%   2.0568E-5   0   0   6   3   0   1    Homozygous  \n",
       "\n",
       "[37643 rows x 23 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "4f4a7fbb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.9117E-30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>943996</td>\n",
       "      <td>rs374426973</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=45;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9996,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>95</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>20</td>\n",
       "      <td>25</td>\n",
       "      <td>55.56%</td>\n",
       "      <td>2.7295E-10</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NOC2L,SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.07907,0...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NOC2L,SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37638</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5100327</td>\n",
       "      <td>rs2524543</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PCDH11Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37639</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5100614</td>\n",
       "      <td>rs2563389</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>254</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.8097E-26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PCDH11Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37640</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12816280</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=48;WT=0;HET=0;HOM=1;NC=0;CSQ=T|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.554E-28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37643 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "2      chr1    942451     rs6672356   T   C    .   PASS   \n",
       "3      chr1    943996   rs374426973   G   A    .   PASS   \n",
       "4      chr1    944858     rs3748592   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "37638  chrY   5100327     rs2524543   G   T    .   PASS   \n",
       "37639  chrY   5100614     rs2563389   T   G    .   PASS   \n",
       "37640  chrY  12816280             .   C   T    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37642  chrY  56961295             .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "2      ADP=50;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...  1/1  255  50  50   \n",
       "3      ADP=45;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9996,0....  0/1   95  45  45   \n",
       "4      ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.07907,0...  1/1  164  29  29   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "37638  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1  182  32  32   \n",
       "37639  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1  254  44  44   \n",
       "37640  ADP=48;WT=0;HET=0;HOM=1;NC=0;CSQ=T|missense_va...  1/1  255  48  48   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37642  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...  1/1   46   9   9   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "2       0  50    100%  9.9117E-30   0   0  34  16   0   1    Homozygous   \n",
       "3      20  25  55.56%  2.7295E-10  15   5  20   5   1   0  Heterozygous   \n",
       "4       0  29    100%  3.3259E-17   0   0  17  12   0   1    Homozygous   \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "37638   0  32    100%  5.4567E-19   0   0  17  15   0   1    Homozygous   \n",
       "37639   0  44    100%  3.8097E-26   0   0  24  20   0   1    Homozygous   \n",
       "37640   0  48    100%   1.554E-28   0   0  23  25   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37642   0   9    100%   2.0568E-5   0   0   6   3   0   1    Homozygous   \n",
       "\n",
       "          Gene Name  \n",
       "0                    \n",
       "1            SAMD11  \n",
       "2            SAMD11  \n",
       "3      NOC2L,SAMD11  \n",
       "4      NOC2L,SAMD11  \n",
       "...             ...  \n",
       "37638       PCDH11Y  \n",
       "37639       PCDH11Y  \n",
       "37640                \n",
       "37641                \n",
       "37642                \n",
       "\n",
       "[37643 rows x 24 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "69ef7c63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER||ENSG000002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|intron_variant|MODIFIER|SAMD11|ENSG000001876...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37642  chrY  56961295             .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37642  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...  1/1   46   9   9   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37642   0   9    100%   2.0568E-5   0   0   6   3   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "...         ...                                                ...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37642            T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "\n",
       "                                                     csq  \n",
       "0      T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...  \n",
       "0      T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...  \n",
       "0      T|downstream_gene_variant|MODIFIER||ENSG000002...  \n",
       "1      G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...  \n",
       "1      G|intron_variant|MODIFIER|SAMD11|ENSG000001876...  \n",
       "...                                                  ...  \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...  \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...  \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...  \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...  \n",
       "37642  T|intergenic_variant|MODIFIER|||||||||||||||||...  \n",
       "\n",
       "[333685 rows x 26 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "6b412ebf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER||ENSG000002...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|intron_variant|MODIFIER|SAMD11|ENSG000001876...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8:c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1:c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1:p.Ile1228Val</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>1228</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1:c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1:p.Ile570Val</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>570</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1:c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1:p.Ile1273Val</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>1273</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0.02)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1:c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1:p.Ile937Val</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>937</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37642  chrY  56961295             .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37642  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...  1/1   46   9   9   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37642   0   9    100%   2.0568E-5   0   0   6   3   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "...         ...                                                ...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37642            T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...                 \n",
       "0      T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...                 \n",
       "0      T|downstream_gene_variant|MODIFIER||ENSG000002...                 \n",
       "1      G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...                 \n",
       "1      G|intron_variant|MODIFIER|SAMD11|ENSG000001876...                 \n",
       "...                                                  ...           ...   \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37642  T|intergenic_variant|MODIFIER|||||||||||||||||...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                          HGVSC  \\\n",
       "0                                                                          \n",
       "0                                             ENST00000641515.2:c.-36C>T   \n",
       "0                                                                          \n",
       "1                                                                          \n",
       "1                                          ENST00000341065.8:c.709-75C>G   \n",
       "...        ...                ...     ...                            ...   \n",
       "37641                                        ENST00000682216.1:c.3682A>G   \n",
       "37641                                        ENST00000682761.1:c.1708A>G   \n",
       "37641                                        ENST00000682913.1:c.3817A>G   \n",
       "37641                                        ENST00000684226.1:c.2809A>G   \n",
       "37642                                                                      \n",
       "\n",
       "                                HGVSP       PolyPhen  \\\n",
       "0                                                      \n",
       "0                                                      \n",
       "0                                                      \n",
       "1                                                      \n",
       "1                                                      \n",
       "...                               ...            ...   \n",
       "37641  ENSP00000507656.1:p.Ile1228Val  benign(0.067)   \n",
       "37641   ENSP00000506900.1:p.Ile570Val  benign(0.044)   \n",
       "37641  ENSP00000507736.1:p.Ile1273Val  benign(0.124)   \n",
       "37641   ENSP00000507919.1:p.Ile937Val  benign(0.031)   \n",
       "37642                                                  \n",
       "\n",
       "                                  BIOTYPE   EXON INTRON Protein_position  \\\n",
       "0      transcribed_unprocessed_pseudogene                                  \n",
       "0                          protein_coding    2/3                           \n",
       "0                                  lncRNA                                  \n",
       "1                          protein_coding                                  \n",
       "1                          protein_coding          7/11                    \n",
       "...                                   ...    ...    ...              ...   \n",
       "37641                      non_stop_decay  25/28                    1228   \n",
       "37641                      non_stop_decay  10/12                     570   \n",
       "37641                      protein_coding  26/29                    1273   \n",
       "37641                      non_stop_decay  16/19                     937   \n",
       "37642                                                                      \n",
       "\n",
       "      Amino_acids   Codons STRAND PUBMED              Consequence    IMPACT  \\\n",
       "0                               1         downstream_gene_variant  MODIFIER   \n",
       "0                               1             5_prime_UTR_variant  MODIFIER   \n",
       "0                               1         downstream_gene_variant  MODIFIER   \n",
       "1                              -1         downstream_gene_variant  MODIFIER   \n",
       "1                               1                  intron_variant  MODIFIER   \n",
       "...           ...      ...    ...    ...                      ...       ...   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37642                                          intergenic_variant  MODIFIER   \n",
       "\n",
       "                                   SIFT      AF  AFR_AF  AMR_AF  EAS_AF  \\\n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "1                                        0.8880  0.7935  0.9092  0.9256   \n",
       "1                                        0.8880  0.7935  0.9092  0.9256   \n",
       "...                                 ...     ...     ...     ...     ...   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37641  deleterious_low_confidence(0.02)                                   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37642                                                                     \n",
       "\n",
       "       EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                                                                               \n",
       "0                                                                               \n",
       "0                                                                               \n",
       "1      0.9443  0.9039                                                           \n",
       "1      0.9443  0.9039                                                           \n",
       "...       ...     ...        ...            ...            ...            ...   \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37642                                                                           \n",
       "\n",
       "      gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "1                                                                   \n",
       "1                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "1                        0.8921         0.7967         0.8813          0.913   \n",
       "1                        0.8921         0.7967         0.8813          0.913   \n",
       "...              ...        ...            ...            ...            ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0                0.5              0              0              0   \n",
       "0                0.5              0              0              0   \n",
       "0                0.5              0              0              0   \n",
       "1             0.9116         0.9261         0.9422         0.9114   \n",
       "1             0.9116         0.9261         0.9422         0.9114   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ  \n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ  \n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ  \n",
       "1              0.933          0.902         0.9047  0.9443          EUR  \n",
       "1              0.933          0.902         0.9047  0.9443          EUR  \n",
       "...              ...            ...            ...     ...          ...  \n",
       "37641                                                                    \n",
       "37641                                                                    \n",
       "37641                                                                    \n",
       "37641                                                                    \n",
       "37642                                                                    \n",
       "\n",
       "[333685 rows x 72 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "########################################################### Required columns extraction from the CSQ column ####################\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "################################################## Frequency columns extraction ################################################\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "7d2b31f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER||ENSG000002...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|intron_variant|MODIFIER|SAMD11|ENSG000001876...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8:c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1:c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1:p.Ile1228Val</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>1228</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1:c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1:p.Ile570Val</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>570</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1:c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1:p.Ile1273Val</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>1273</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0.02)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1:c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1:p.Ile937Val</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>937</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37642  chrY  56961295             .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37642  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...  1/1   46   9   9   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37642   0   9    100%   2.0568E-5   0   0   6   3   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "...         ...                                                ...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37642            T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...                 \n",
       "0      T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...                 \n",
       "0      T|downstream_gene_variant|MODIFIER||ENSG000002...                 \n",
       "1      G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...                 \n",
       "1      G|intron_variant|MODIFIER|SAMD11|ENSG000001876...                 \n",
       "...                                                  ...           ...   \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37642  T|intergenic_variant|MODIFIER|||||||||||||||||...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                          HGVSC  \\\n",
       "0                                                                          \n",
       "0                                             ENST00000641515.2:c.-36C>T   \n",
       "0                                                                          \n",
       "1                                                                          \n",
       "1                                          ENST00000341065.8:c.709-75C>G   \n",
       "...        ...                ...     ...                            ...   \n",
       "37641                                        ENST00000682216.1:c.3682A>G   \n",
       "37641                                        ENST00000682761.1:c.1708A>G   \n",
       "37641                                        ENST00000682913.1:c.3817A>G   \n",
       "37641                                        ENST00000684226.1:c.2809A>G   \n",
       "37642                                                                      \n",
       "\n",
       "                                HGVSP       PolyPhen  \\\n",
       "0                                                      \n",
       "0                                                      \n",
       "0                                                      \n",
       "1                                                      \n",
       "1                                                      \n",
       "...                               ...            ...   \n",
       "37641  ENSP00000507656.1:p.Ile1228Val  benign(0.067)   \n",
       "37641   ENSP00000506900.1:p.Ile570Val  benign(0.044)   \n",
       "37641  ENSP00000507736.1:p.Ile1273Val  benign(0.124)   \n",
       "37641   ENSP00000507919.1:p.Ile937Val  benign(0.031)   \n",
       "37642                                                  \n",
       "\n",
       "                                  BIOTYPE   EXON INTRON Protein_position  \\\n",
       "0      transcribed_unprocessed_pseudogene                                  \n",
       "0                          protein_coding    2/3                           \n",
       "0                                  lncRNA                                  \n",
       "1                          protein_coding                                  \n",
       "1                          protein_coding          7/11                    \n",
       "...                                   ...    ...    ...              ...   \n",
       "37641                      non_stop_decay  25/28                    1228   \n",
       "37641                      non_stop_decay  10/12                     570   \n",
       "37641                      protein_coding  26/29                    1273   \n",
       "37641                      non_stop_decay  16/19                     937   \n",
       "37642                                                                      \n",
       "\n",
       "      Amino_acids   Codons STRAND PUBMED              Consequence    IMPACT  \\\n",
       "0                               1         downstream_gene_variant  MODIFIER   \n",
       "0                               1             5_prime_UTR_variant  MODIFIER   \n",
       "0                               1         downstream_gene_variant  MODIFIER   \n",
       "1                              -1         downstream_gene_variant  MODIFIER   \n",
       "1                               1                  intron_variant  MODIFIER   \n",
       "...           ...      ...    ...    ...                      ...       ...   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37642                                          intergenic_variant  MODIFIER   \n",
       "\n",
       "                                   SIFT      AF  AFR_AF  AMR_AF  EAS_AF  \\\n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "1                                        0.8880  0.7935  0.9092  0.9256   \n",
       "1                                        0.8880  0.7935  0.9092  0.9256   \n",
       "...                                 ...     ...     ...     ...     ...   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37641  deleterious_low_confidence(0.02)                                   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37642                                                                     \n",
       "\n",
       "       EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                                                                               \n",
       "0                                                                               \n",
       "0                                                                               \n",
       "1      0.9443  0.9039                                                           \n",
       "1      0.9443  0.9039                                                           \n",
       "...       ...     ...        ...            ...            ...            ...   \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37642                                                                           \n",
       "\n",
       "      gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "1                                                                   \n",
       "1                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "1                        0.8921         0.7967         0.8813          0.913   \n",
       "1                        0.8921         0.7967         0.8813          0.913   \n",
       "...              ...        ...            ...            ...            ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0                0.5              0              0              0   \n",
       "0                0.5              0              0              0   \n",
       "0                0.5              0              0              0   \n",
       "1             0.9116         0.9261         0.9422         0.9114   \n",
       "1             0.9116         0.9261         0.9422         0.9114   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ   \n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ   \n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ   \n",
       "1              0.933          0.902         0.9047  0.9443          EUR   \n",
       "1              0.933          0.902         0.9047  0.9443          EUR   \n",
       "...              ...            ...            ...     ...          ...   \n",
       "37641                                                                     \n",
       "37641                                                                     \n",
       "37641                                                                     \n",
       "37641                                                                     \n",
       "37642                                                                     \n",
       "\n",
       "      Protein Position and Amino Acid  \n",
       "0                                 NaN  \n",
       "0                                 NaN  \n",
       "0                                 NaN  \n",
       "1                                 NaN  \n",
       "1                                 NaN  \n",
       "...                               ...  \n",
       "37641                          I1228V  \n",
       "37641                           I570V  \n",
       "37641                          I1273V  \n",
       "37641                           I937V  \n",
       "37642                             NaN  \n",
       "\n",
       "[333685 rows x 73 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "ecb17e67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER||ENSG000002...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>G|intron_variant|MODIFIER|SAMD11|ENSG000001876...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8:c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1:c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1:p.Ile1228Val</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>1228</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1:c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1:p.Ile570Val</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>570</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1:c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1:p.Ile1273Val</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>1273</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0.02)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td>C|missense_variant|MODERATE|UTY|ENSG0000018387...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1:c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1:p.Ile937Val</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>937</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "1      chr1    942335     rs6605066   C   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37641  chrY  13297744             .   T   C    .   PASS   \n",
       "37642  chrY  56961295             .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "0      ADP=43;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   32  43  43   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "1      ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.112,0.8...  1/1  224  39  39   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37641  ADP=31;WT=0;HET=0;HOM=1;NC=0;CSQ=C|missense_va...  1/1  161  31  31   \n",
       "37642  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_v...  1/1   46   9   9   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "0      33  10  23.26%   5.4148E-4  26   7   9   1   1   0  Heterozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "1       0  39    100%  3.6742E-23   0   0  30   9   0   1    Homozygous   \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37641   1  30  96.77%  6.8754E-17   1   0  25   5   0   1    Homozygous   \n",
       "37642   0   9    100%   2.0568E-5   0   0   6   3   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "0                T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "1        SAMD11  G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "...         ...                                                ...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37641            C|missense_variant|MODERATE|UTY|ENSG0000018387...   \n",
       "37642            T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|OR4G11P|ENS...                 \n",
       "0      T|5_prime_UTR_variant|MODIFIER|OR4F5|ENSG00000...                 \n",
       "0      T|downstream_gene_variant|MODIFIER||ENSG000002...                 \n",
       "1      G|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...                 \n",
       "1      G|intron_variant|MODIFIER|SAMD11|ENSG000001876...                 \n",
       "...                                                  ...           ...   \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37641  C|missense_variant|MODERATE|UTY|ENSG0000018387...                 \n",
       "37642  T|intergenic_variant|MODIFIER|||||||||||||||||...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                          HGVSC  \\\n",
       "0                                                                          \n",
       "0                                             ENST00000641515.2:c.-36C>T   \n",
       "0                                                                          \n",
       "1                                                                          \n",
       "1                                          ENST00000341065.8:c.709-75C>G   \n",
       "...        ...                ...     ...                            ...   \n",
       "37641                                        ENST00000682216.1:c.3682A>G   \n",
       "37641                                        ENST00000682761.1:c.1708A>G   \n",
       "37641                                        ENST00000682913.1:c.3817A>G   \n",
       "37641                                        ENST00000684226.1:c.2809A>G   \n",
       "37642                                                                      \n",
       "\n",
       "                                HGVSP       PolyPhen  \\\n",
       "0                                                      \n",
       "0                                                      \n",
       "0                                                      \n",
       "1                                                      \n",
       "1                                                      \n",
       "...                               ...            ...   \n",
       "37641  ENSP00000507656.1:p.Ile1228Val  benign(0.067)   \n",
       "37641   ENSP00000506900.1:p.Ile570Val  benign(0.044)   \n",
       "37641  ENSP00000507736.1:p.Ile1273Val  benign(0.124)   \n",
       "37641   ENSP00000507919.1:p.Ile937Val  benign(0.031)   \n",
       "37642                                                  \n",
       "\n",
       "                                  BIOTYPE   EXON INTRON Protein_position  \\\n",
       "0      transcribed_unprocessed_pseudogene                                  \n",
       "0                          protein_coding    2/3                           \n",
       "0                                  lncRNA                                  \n",
       "1                          protein_coding                                  \n",
       "1                          protein_coding          7/11                    \n",
       "...                                   ...    ...    ...              ...   \n",
       "37641                      non_stop_decay  25/28                    1228   \n",
       "37641                      non_stop_decay  10/12                     570   \n",
       "37641                      protein_coding  26/29                    1273   \n",
       "37641                      non_stop_decay  16/19                     937   \n",
       "37642                                                                      \n",
       "\n",
       "      Amino_acids   Codons STRAND PUBMED              Consequence    IMPACT  \\\n",
       "0                               1         downstream_gene_variant  MODIFIER   \n",
       "0                               1             5_prime_UTR_variant  MODIFIER   \n",
       "0                               1         downstream_gene_variant  MODIFIER   \n",
       "1                              -1         downstream_gene_variant  MODIFIER   \n",
       "1                               1                  intron_variant  MODIFIER   \n",
       "...           ...      ...    ...    ...                      ...       ...   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37641         I/V  Atc/Gtc     -1                missense_variant  MODERATE   \n",
       "37642                                          intergenic_variant  MODIFIER   \n",
       "\n",
       "                                   SIFT      AF  AFR_AF  AMR_AF  EAS_AF  \\\n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "1                                        0.8880  0.7935  0.9092  0.9256   \n",
       "1                                        0.8880  0.7935  0.9092  0.9256   \n",
       "...                                 ...     ...     ...     ...     ...   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37641  deleterious_low_confidence(0.02)                                   \n",
       "37641     deleterious_low_confidence(0)                                   \n",
       "37642                                                                     \n",
       "\n",
       "       EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                                                                               \n",
       "0                                                                               \n",
       "0                                                                               \n",
       "1      0.9443  0.9039                                                           \n",
       "1      0.9443  0.9039                                                           \n",
       "...       ...     ...        ...            ...            ...            ...   \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37641                                                                           \n",
       "37642                                                                           \n",
       "\n",
       "      gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "1                                                                   \n",
       "1                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "0                       0.03299        0.02406              0        0.01429   \n",
       "1                        0.8921         0.7967         0.8813          0.913   \n",
       "1                        0.8921         0.7967         0.8813          0.913   \n",
       "...              ...        ...            ...            ...            ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0                0.5              0              0              0   \n",
       "0                0.5              0              0              0   \n",
       "0                0.5              0              0              0   \n",
       "1             0.9116         0.9261         0.9422         0.9114   \n",
       "1             0.9116         0.9261         0.9422         0.9114   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ   \n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ   \n",
       "0             0.1071              0              0     0.5  gnomADg_ASJ   \n",
       "1              0.933          0.902         0.9047  0.9443          EUR   \n",
       "1              0.933          0.902         0.9047  0.9443          EUR   \n",
       "...              ...            ...            ...     ...          ...   \n",
       "37641                                                                     \n",
       "37641                                                                     \n",
       "37641                                                                     \n",
       "37641                                                                     \n",
       "37642                                                                     \n",
       "\n",
       "      Protein Position and Amino Acid              HGVSc HGVSc (Transcript)  \\\n",
       "0                                 NaN                                  None   \n",
       "0                                 NaN  ENST00000641515.2           c.-36C>T   \n",
       "0                                 NaN                                  None   \n",
       "1                                 NaN                                  None   \n",
       "1                                 NaN  ENST00000341065.8        c.709-75C>G   \n",
       "...                               ...                ...                ...   \n",
       "37641                          I1228V  ENST00000682216.1          c.3682A>G   \n",
       "37641                           I570V  ENST00000682761.1          c.1708A>G   \n",
       "37641                          I1273V  ENST00000682913.1          c.3817A>G   \n",
       "37641                           I937V  ENST00000684226.1          c.2809A>G   \n",
       "37642                             NaN                                  None   \n",
       "\n",
       "                   HGVSp HGVSp (Transcript)  \n",
       "0                                      None  \n",
       "0                                      None  \n",
       "0                                      None  \n",
       "1                                      None  \n",
       "1                                      None  \n",
       "...                  ...                ...  \n",
       "37641  ENSP00000507656.1       p.Ile1228Val  \n",
       "37641  ENSP00000506900.1        p.Ile570Val  \n",
       "37641  ENSP00000507736.1       p.Ile1273Val  \n",
       "37641  ENSP00000507919.1        p.Ile937Val  \n",
       "37642                                  None  \n",
       "\n",
       "[333685 rows x 77 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "ccde4da5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT',\n",
       "       'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
       "       'ADR', 'HET', 'HOM', 'Zygosity', 'Gene Name', 'CSQ', 'csq',\n",
       "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSC',\n",
       "       'HGVSP', 'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON', 'Protein_position',\n",
       "       'Amino_acids', 'Codons', 'STRAND', 'PUBMED', 'Consequence', 'IMPACT',\n",
       "       'SIFT', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
       "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
       "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
       "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
       "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
       "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
       "       'MAX_AF', 'MAX_AF_POPS', 'Protein Position and Amino Acid', 'HGVSc',\n",
       "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "e34f2be0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...         ...           ...   ...       ...  ..  ..           ...   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37642                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                   Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0      downstream_gene_variant  MODIFIER                          \n",
       "0          5_prime_UTR_variant  MODIFIER                          \n",
       "0      downstream_gene_variant  MODIFIER                          \n",
       "1      downstream_gene_variant  MODIFIER                          \n",
       "1               intron_variant  MODIFIER                          \n",
       "...                        ...       ...           ...      ...   \n",
       "37641         missense_variant  MODERATE                          \n",
       "37641         missense_variant  MODERATE                          \n",
       "37641         missense_variant  MODERATE                          \n",
       "37641         missense_variant  MODERATE                          \n",
       "37642       intergenic_variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                                                 None   \n",
       "0                                 ENST00000641515.2           c.-36C>T   \n",
       "0                                                                 None   \n",
       "1                                                                 None   \n",
       "1                                 ENST00000341065.8        c.709-75C>G   \n",
       "...                  ...     ...                ...                ...   \n",
       "37641                             ENST00000682216.1          c.3682A>G   \n",
       "37641                             ENST00000682761.1          c.1708A>G   \n",
       "37641                             ENST00000682913.1          c.3817A>G   \n",
       "37641                             ENST00000684226.1          c.2809A>G   \n",
       "37642                                                             None   \n",
       "\n",
       "                   HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ  \\\n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "...                  ...                ...  ...  ...  ..  ..  ..  ..     ...   \n",
       "37641  ENSP00000507656.1       p.Ile1228Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000506900.1        p.Ile570Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507736.1       p.Ile1273Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507919.1        p.Ile937Val  1/1  161  31  31   1  30  96.77%   \n",
       "37642                                  None  1/1   46   9   9   0   9    100%   \n",
       "\n",
       "             PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "...           ...  ..  ..  ..  ..                               ...   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious_low_confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious_low_confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5  deleterious_low_confidence(0.02)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious_low_confidence(0)   \n",
       "37642   2.0568E-5   0   0   6   3                                     \n",
       "\n",
       "            PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0                                                                      \n",
       "0                                                                      \n",
       "0                                                                      \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "...              ...     ...     ...     ...     ...     ...     ...   \n",
       "37641  benign(0.067)                                                   \n",
       "37641  benign(0.044)                                                   \n",
       "37641  benign(0.124)                                                   \n",
       "37641  benign(0.031)                                                   \n",
       "37642                                                                  \n",
       "\n",
       "      gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                              \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "1                                                                              \n",
       "1                                                                              \n",
       "...          ...            ...            ...            ...            ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "1                                                                     0.8921   \n",
       "1                                                                     0.8921   \n",
       "...              ...            ...            ...            ...        ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                  0              0     0.5  gnomADg_ASJ   \n",
       "0                  0              0     0.5  gnomADg_ASJ   \n",
       "0                  0              0     0.5  gnomADg_ASJ   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "...              ...            ...     ...          ...   \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37642                                                      \n",
       "\n",
       "                                  BIOTYPE   EXON INTRON  \\\n",
       "0      transcribed_unprocessed_pseudogene                 \n",
       "0                          protein_coding    2/3          \n",
       "0                                  lncRNA                 \n",
       "1                          protein_coding                 \n",
       "1                          protein_coding          7/11   \n",
       "...                                   ...    ...    ...   \n",
       "37641                      non_stop_decay  25/28          \n",
       "37641                      non_stop_decay  10/12          \n",
       "37641                      protein_coding  26/29          \n",
       "37641                      non_stop_decay  16/19          \n",
       "37642                                                     \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                 NaN               1         \n",
       "0                                 NaN               1         \n",
       "0                                 NaN               1         \n",
       "1                                 NaN              -1         \n",
       "1                                 NaN               1         \n",
       "...                               ...      ...    ...    ...  \n",
       "37641                          I1228V  Atc/Gtc     -1         \n",
       "37641                           I570V  Atc/Gtc     -1         \n",
       "37641                          I1273V  Atc/Gtc     -1         \n",
       "37641                           I937V  Atc/Gtc     -1         \n",
       "37642                             NaN                         \n",
       "\n",
       "[333685 rows x 66 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf[['Gene Name', 'rsID','CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "a41ddc7b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non_stop_decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...         ...           ...   ...       ...  ..  ..           ...   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37642                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                   Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0      downstream_gene_variant  MODIFIER                          \n",
       "0          5_prime_UTR_variant  MODIFIER                          \n",
       "0      downstream_gene_variant  MODIFIER                          \n",
       "1      downstream_gene_variant  MODIFIER                          \n",
       "1               intron_variant  MODIFIER                          \n",
       "...                        ...       ...           ...      ...   \n",
       "37641         missense_variant  MODERATE                          \n",
       "37641         missense_variant  MODERATE                          \n",
       "37641         missense_variant  MODERATE                          \n",
       "37641         missense_variant  MODERATE                          \n",
       "37642       intergenic_variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                                                 None   \n",
       "0                                 ENST00000641515.2           c.-36C>T   \n",
       "0                                                                 None   \n",
       "1                                                                 None   \n",
       "1                                 ENST00000341065.8        c.709-75C>G   \n",
       "...                  ...     ...                ...                ...   \n",
       "37641                             ENST00000682216.1          c.3682A>G   \n",
       "37641                             ENST00000682761.1          c.1708A>G   \n",
       "37641                             ENST00000682913.1          c.3817A>G   \n",
       "37641                             ENST00000684226.1          c.2809A>G   \n",
       "37642                                                             None   \n",
       "\n",
       "                   HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ  \\\n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "...                  ...                ...  ...  ...  ..  ..  ..  ..     ...   \n",
       "37641  ENSP00000507656.1       p.Ile1228Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000506900.1        p.Ile570Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507736.1       p.Ile1273Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507919.1        p.Ile937Val  1/1  161  31  31   1  30  96.77%   \n",
       "37642                                  None  1/1   46   9   9   0   9    100%   \n",
       "\n",
       "             PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "...           ...  ..  ..  ..  ..                               ...   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious_low_confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious_low_confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5  deleterious_low_confidence(0.02)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious_low_confidence(0)   \n",
       "37642   2.0568E-5   0   0   6   3                                     \n",
       "\n",
       "            PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0                                                                      \n",
       "0                                                                      \n",
       "0                                                                      \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "...              ...     ...     ...     ...     ...     ...     ...   \n",
       "37641  benign(0.067)                                                   \n",
       "37641  benign(0.044)                                                   \n",
       "37641  benign(0.124)                                                   \n",
       "37641  benign(0.031)                                                   \n",
       "37642                                                                  \n",
       "\n",
       "      gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                              \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "1                                                                              \n",
       "1                                                                              \n",
       "...          ...            ...            ...            ...            ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "1                                                                     0.8921   \n",
       "1                                                                     0.8921   \n",
       "...              ...            ...            ...            ...        ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                  0              0     0.5  gnomADg_ASJ   \n",
       "0                  0              0     0.5  gnomADg_ASJ   \n",
       "0                  0              0     0.5  gnomADg_ASJ   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "...              ...            ...     ...          ...   \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37642                                                      \n",
       "\n",
       "                                  BIOTYPE   EXON INTRON  \\\n",
       "0      transcribed_unprocessed_pseudogene                 \n",
       "0                          protein_coding    2/3          \n",
       "0                                  lncRNA                 \n",
       "1                          protein_coding                 \n",
       "1                          protein_coding          7/11   \n",
       "...                                   ...    ...    ...   \n",
       "37641                      non_stop_decay  25/28          \n",
       "37641                      non_stop_decay  10/12          \n",
       "37641                      protein_coding  26/29          \n",
       "37641                      non_stop_decay  16/19          \n",
       "37642                                                     \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                 NaN               1         \n",
       "0                                 NaN               1         \n",
       "0                                 NaN               1         \n",
       "1                                 NaN              -1         \n",
       "1                                 NaN               1         \n",
       "...                               ...      ...    ...    ...  \n",
       "37641                          I1228V  Atc/Gtc     -1         \n",
       "37641                           I570V  Atc/Gtc     -1         \n",
       "37641                          I1273V  Atc/Gtc     -1         \n",
       "37641                           I937V  Atc/Gtc     -1         \n",
       "37642                             NaN                         \n",
       "\n",
       "[333685 rows x 66 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "# Print the modified DataFrame\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "d3ba7f20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...         ...           ...   ...       ...  ..  ..           ...   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37642                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                   Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0      downstream gene variant  MODIFIER                          \n",
       "0          5 prime UTR variant  MODIFIER                          \n",
       "0      downstream gene variant  MODIFIER                          \n",
       "1      downstream gene variant  MODIFIER                          \n",
       "1               intron variant  MODIFIER                          \n",
       "...                        ...       ...           ...      ...   \n",
       "37641         missense variant  MODERATE                          \n",
       "37641         missense variant  MODERATE                          \n",
       "37641         missense variant  MODERATE                          \n",
       "37641         missense variant  MODERATE                          \n",
       "37642       intergenic variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                                                 None   \n",
       "0                                 ENST00000641515.2           c.-36C>T   \n",
       "0                                                                 None   \n",
       "1                                                                 None   \n",
       "1                                 ENST00000341065.8        c.709-75C>G   \n",
       "...                  ...     ...                ...                ...   \n",
       "37641                             ENST00000682216.1          c.3682A>G   \n",
       "37641                             ENST00000682761.1          c.1708A>G   \n",
       "37641                             ENST00000682913.1          c.3817A>G   \n",
       "37641                             ENST00000684226.1          c.2809A>G   \n",
       "37642                                                             None   \n",
       "\n",
       "                   HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ  \\\n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "...                  ...                ...  ...  ...  ..  ..  ..  ..     ...   \n",
       "37641  ENSP00000507656.1       p.Ile1228Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000506900.1        p.Ile570Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507736.1       p.Ile1273Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507919.1        p.Ile937Val  1/1  161  31  31   1  30  96.77%   \n",
       "37642                                  None  1/1   46   9   9   0   9    100%   \n",
       "\n",
       "             PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "...           ...  ..  ..  ..  ..                               ...   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5  deleterious low confidence(0.02)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "37642   2.0568E-5   0   0   6   3                                     \n",
       "\n",
       "            PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0                                                                      \n",
       "0                                                                      \n",
       "0                                                                      \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "...              ...     ...     ...     ...     ...     ...     ...   \n",
       "37641  benign(0.067)                                                   \n",
       "37641  benign(0.044)                                                   \n",
       "37641  benign(0.124)                                                   \n",
       "37641  benign(0.031)                                                   \n",
       "37642                                                                  \n",
       "\n",
       "      gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                              \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "1                                                                              \n",
       "1                                                                              \n",
       "...          ...            ...            ...            ...            ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "1                                                                     0.8921   \n",
       "1                                                                     0.8921   \n",
       "...              ...            ...            ...            ...        ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                  0              0     0.5  gnomADg ASJ   \n",
       "0                  0              0     0.5  gnomADg ASJ   \n",
       "0                  0              0     0.5  gnomADg ASJ   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "...              ...            ...     ...          ...   \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37642                                                      \n",
       "\n",
       "                                  BIOTYPE   EXON INTRON  \\\n",
       "0      transcribed unprocessed pseudogene                 \n",
       "0                          protein coding    2/3          \n",
       "0                                  lncRNA                 \n",
       "1                          protein coding                 \n",
       "1                          protein coding          7/11   \n",
       "...                                   ...    ...    ...   \n",
       "37641                      non stop decay  25/28          \n",
       "37641                      non stop decay  10/12          \n",
       "37641                      protein coding  26/29          \n",
       "37641                      non stop decay  16/19          \n",
       "37642                                                     \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                 nan               1         \n",
       "0                                 nan               1         \n",
       "0                                 nan               1         \n",
       "1                                 nan              -1         \n",
       "1                                 nan               1         \n",
       "...                               ...      ...    ...    ...  \n",
       "37641                          I1228V  Atc/Gtc     -1         \n",
       "37641                           I570V  Atc/Gtc     -1         \n",
       "37641                          I1273V  Atc/Gtc     -1         \n",
       "37641                           I937V  Atc/Gtc     -1         \n",
       "37642                             nan                         \n",
       "\n",
       "[333685 rows x 66 columns]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "81edf288",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37641</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37642</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intergenic variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 67 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "0                rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "1        SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...         ...           ...   ...       ...  ..  ..           ...   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37641                       .  chrY  13297744   T   C    Homozygous   \n",
       "37642                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                   Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0      downstream gene variant  MODIFIER                          \n",
       "0          5 prime UTR variant  MODIFIER                          \n",
       "0      downstream gene variant  MODIFIER                          \n",
       "1      downstream gene variant  MODIFIER                          \n",
       "1               intron variant  MODIFIER                          \n",
       "...                        ...       ...           ...      ...   \n",
       "37641         missense variant  MODERATE                          \n",
       "37641         missense variant  MODERATE                          \n",
       "37641         missense variant  MODERATE                          \n",
       "37641         missense variant  MODERATE                          \n",
       "37642       intergenic variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                                                 None   \n",
       "0                                 ENST00000641515.2           c.-36C>T   \n",
       "0                                                                 None   \n",
       "1                                                                 None   \n",
       "1                                 ENST00000341065.8        c.709-75C>G   \n",
       "...                  ...     ...                ...                ...   \n",
       "37641                             ENST00000682216.1          c.3682A>G   \n",
       "37641                             ENST00000682761.1          c.1708A>G   \n",
       "37641                             ENST00000682913.1          c.3817A>G   \n",
       "37641                             ENST00000684226.1          c.2809A>G   \n",
       "37642                                                             None   \n",
       "\n",
       "                   HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ  \\\n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "0                                      None  0/1   32  43  43  33  10  23.26%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "1                                      None  1/1  224  39  39   0  39    100%   \n",
       "...                  ...                ...  ...  ...  ..  ..  ..  ..     ...   \n",
       "37641  ENSP00000507656.1       p.Ile1228Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000506900.1        p.Ile570Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507736.1       p.Ile1273Val  1/1  161  31  31   1  30  96.77%   \n",
       "37641  ENSP00000507919.1        p.Ile937Val  1/1  161  31  31   1  30  96.77%   \n",
       "37642                                  None  1/1   46   9   9   0   9    100%   \n",
       "\n",
       "             PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "0       5.4148E-4  26   7   9   1                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "1      3.6742E-23   0   0  30   9                                     \n",
       "...           ...  ..  ..  ..  ..                               ...   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "37641  6.8754E-17   1   0  25   5  deleterious low confidence(0.02)   \n",
       "37641  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "37642   2.0568E-5   0   0   6   3                                     \n",
       "\n",
       "            PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0                                                                      \n",
       "0                                                                      \n",
       "0                                                                      \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "1                     0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "...              ...     ...     ...     ...     ...     ...     ...   \n",
       "37641  benign(0.067)                                                   \n",
       "37641  benign(0.044)                                                   \n",
       "37641  benign(0.124)                                                   \n",
       "37641  benign(0.031)                                                   \n",
       "37642                                                                  \n",
       "\n",
       "      gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                              \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "1                                                                              \n",
       "1                                                                              \n",
       "...          ...            ...            ...            ...            ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "0                                                                    0.03299   \n",
       "1                                                                     0.8921   \n",
       "1                                                                     0.8921   \n",
       "...              ...            ...            ...            ...        ...   \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37641                                                                          \n",
       "37642                                                                          \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "0            0.02406              0        0.01429            0.5   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "1             0.7967         0.8813          0.913         0.9116   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "0                  0              0              0         0.1071   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "1             0.9261         0.9422         0.9114          0.933   \n",
       "...              ...            ...            ...            ...   \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37641                                                               \n",
       "37642                                                               \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                  0              0     0.5  gnomADg ASJ   \n",
       "0                  0              0     0.5  gnomADg ASJ   \n",
       "0                  0              0     0.5  gnomADg ASJ   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "1              0.902         0.9047  0.9443          EUR   \n",
       "...              ...            ...     ...          ...   \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37641                                                      \n",
       "37642                                                      \n",
       "\n",
       "                                  BIOTYPE   EXON INTRON  \\\n",
       "0      transcribed unprocessed pseudogene                 \n",
       "0                          protein coding    2/3          \n",
       "0                                  lncRNA                 \n",
       "1                          protein coding                 \n",
       "1                          protein coding          7/11   \n",
       "...                                   ...    ...    ...   \n",
       "37641                      non stop decay  25/28          \n",
       "37641                      non stop decay  10/12          \n",
       "37641                      protein coding  26/29          \n",
       "37641                      non stop decay  16/19          \n",
       "37642                                                     \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                 nan               1          \n",
       "0                                 nan               1          \n",
       "0                                 nan               1          \n",
       "1                                 nan              -1          \n",
       "1                                 nan               1          \n",
       "...                               ...      ...    ...    ...   \n",
       "37641                          I1228V  Atc/Gtc     -1          \n",
       "37641                           I570V  Atc/Gtc     -1          \n",
       "37641                          I1273V  Atc/Gtc     -1          \n",
       "37641                           I937V  Atc/Gtc     -1          \n",
       "37642                             nan                          \n",
       "\n",
       "                   consequence  \n",
       "0      downstream gene variant  \n",
       "0          5 prime UTR variant  \n",
       "0      downstream gene variant  \n",
       "1      downstream gene variant  \n",
       "1               intron variant  \n",
       "...                        ...  \n",
       "37641         missense variant  \n",
       "37641         missense variant  \n",
       "37641         missense variant  \n",
       "37641         missense variant  \n",
       "37642       intergenic variant  \n",
       "\n",
       "[333685 rows x 67 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "c0d6ce31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "410dbadd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333680</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333681</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333682</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333683</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333684</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "4         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "333680                       .  chrY  13297744   T   C    Homozygous   \n",
       "333681                       .  chrY  13297744   T   C    Homozygous   \n",
       "333682                       .  chrY  13297744   T   C    Homozygous   \n",
       "333683                       .  chrY  13297744   T   C    Homozygous   \n",
       "333684                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                    Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       downstream gene variant  MODIFIER                          \n",
       "1           5 prime UTR variant  MODIFIER                          \n",
       "2       downstream gene variant  MODIFIER                          \n",
       "3       downstream gene variant  MODIFIER                          \n",
       "4                intron variant  MODIFIER                          \n",
       "...                         ...       ...           ...      ...   \n",
       "333680         missense variant  MODERATE                          \n",
       "333681         missense variant  MODERATE                          \n",
       "333682         missense variant  MODERATE                          \n",
       "333683         missense variant  MODERATE                          \n",
       "333684       intergenic variant  MODIFIER                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                                                  None   \n",
       "1                                  ENST00000641515.2           c.-36C>T   \n",
       "2                                                                  None   \n",
       "3                                                                  None   \n",
       "4                                  ENST00000341065.8        c.709-75C>G   \n",
       "...                   ...     ...                ...                ...   \n",
       "333680                             ENST00000682216.1          c.3682A>G   \n",
       "333681                             ENST00000682761.1          c.1708A>G   \n",
       "333682                             ENST00000682913.1          c.3817A>G   \n",
       "333683                             ENST00000684226.1          c.2809A>G   \n",
       "333684                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD  \\\n",
       "0                                       None  0/1   32  43  43  33  10   \n",
       "1                                       None  0/1   32  43  43  33  10   \n",
       "2                                       None  0/1   32  43  43  33  10   \n",
       "3                                       None  1/1  224  39  39   0  39   \n",
       "4                                       None  1/1  224  39  39   0  39   \n",
       "...                   ...                ...  ...  ...  ..  ..  ..  ..   \n",
       "333680  ENSP00000507656.1       p.Ile1228Val  1/1  161  31  31   1  30   \n",
       "333681  ENSP00000506900.1        p.Ile570Val  1/1  161  31  31   1  30   \n",
       "333682  ENSP00000507736.1       p.Ile1273Val  1/1  161  31  31   1  30   \n",
       "333683  ENSP00000507919.1        p.Ile937Val  1/1  161  31  31   1  30   \n",
       "333684                                  None  1/1   46   9   9   0   9   \n",
       "\n",
       "          FREQ        PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       23.26%   5.4148E-4  26   7   9   1                                     \n",
       "1       23.26%   5.4148E-4  26   7   9   1                                     \n",
       "2       23.26%   5.4148E-4  26   7   9   1                                     \n",
       "3         100%  3.6742E-23   0   0  30   9                                     \n",
       "4         100%  3.6742E-23   0   0  30   9                                     \n",
       "...        ...         ...  ..  ..  ..  ..                               ...   \n",
       "333680  96.77%  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "333681  96.77%  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "333682  96.77%  6.8754E-17   1   0  25   5  deleterious low confidence(0.02)   \n",
       "333683  96.77%  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "333684    100%   2.0568E-5   0   0   6   3                                     \n",
       "\n",
       "             PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0                                                                       \n",
       "1                                                                       \n",
       "2                                                                       \n",
       "3                      0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "4                      0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "...               ...     ...     ...     ...     ...     ...     ...   \n",
       "333680  benign(0.067)                                                   \n",
       "333681  benign(0.044)                                                   \n",
       "333682  benign(0.124)                                                   \n",
       "333683  benign(0.031)                                                   \n",
       "333684                                                                  \n",
       "\n",
       "       gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                               \n",
       "1                                                                               \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...           ...            ...            ...            ...            ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03299   \n",
       "1                                                                     0.03299   \n",
       "2                                                                     0.03299   \n",
       "3                                                                      0.8921   \n",
       "4                                                                      0.8921   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02406              0        0.01429            0.5   \n",
       "1             0.02406              0        0.01429            0.5   \n",
       "2             0.02406              0        0.01429            0.5   \n",
       "3              0.7967         0.8813          0.913         0.9116   \n",
       "4              0.7967         0.8813          0.913         0.9116   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0                   0              0              0         0.1071   \n",
       "1                   0              0              0         0.1071   \n",
       "2                   0              0              0         0.1071   \n",
       "3              0.9261         0.9422         0.9114          0.933   \n",
       "4              0.9261         0.9422         0.9114          0.933   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                   0              0     0.5  gnomADg ASJ   \n",
       "1                   0              0     0.5  gnomADg ASJ   \n",
       "2                   0              0     0.5  gnomADg ASJ   \n",
       "3               0.902         0.9047  0.9443          EUR   \n",
       "4               0.902         0.9047  0.9443          EUR   \n",
       "...               ...            ...     ...          ...   \n",
       "333680                                                      \n",
       "333681                                                      \n",
       "333682                                                      \n",
       "333683                                                      \n",
       "333684                                                      \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding                 \n",
       "4                           protein coding          7/11   \n",
       "...                                    ...    ...    ...   \n",
       "333680                      non stop decay  25/28          \n",
       "333681                      non stop decay  10/12          \n",
       "333682                      protein coding  26/29          \n",
       "333683                      non stop decay  16/19          \n",
       "333684                                                     \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  nan               1          \n",
       "1                                  nan               1          \n",
       "2                                  nan               1          \n",
       "3                                  nan              -1          \n",
       "4                                  nan               1          \n",
       "...                                ...      ...    ...    ...   \n",
       "333680                          I1228V  Atc/Gtc     -1          \n",
       "333681                           I570V  Atc/Gtc     -1          \n",
       "333682                          I1273V  Atc/Gtc     -1          \n",
       "333683                           I937V  Atc/Gtc     -1          \n",
       "333684                             nan                          \n",
       "\n",
       "                    consequence Consequence_score  \n",
       "0       downstream gene variant              2/10  \n",
       "1           5 prime UTR variant              3/10  \n",
       "2       downstream gene variant              2/10  \n",
       "3       downstream gene variant              2/10  \n",
       "4                intron variant              2/10  \n",
       "...                         ...               ...  \n",
       "333680         missense variant              7/10  \n",
       "333681         missense variant              7/10  \n",
       "333682         missense variant              7/10  \n",
       "333683         missense variant              7/10  \n",
       "333684       intergenic variant              1/10  \n",
       "\n",
       "[333685 rows x 68 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "bdbb47eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     IMPACT  IMPACT_score\n",
       "0      HIGH          10.0\n",
       "1  MODERATE           5.0\n",
       "2       LOW           2.5\n",
       "3  MODIFIER           1.5"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "68cdd8fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333680</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333681</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333682</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333683</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333684</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "4         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "333680                       .  chrY  13297744   T   C    Homozygous   \n",
       "333681                       .  chrY  13297744   T   C    Homozygous   \n",
       "333682                       .  chrY  13297744   T   C    Homozygous   \n",
       "333683                       .  chrY  13297744   T   C    Homozygous   \n",
       "333684                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                    Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       downstream gene variant  MODIFIER                          \n",
       "1           5 prime UTR variant  MODIFIER                          \n",
       "2       downstream gene variant  MODIFIER                          \n",
       "3       downstream gene variant  MODIFIER                          \n",
       "4                intron variant  MODIFIER                          \n",
       "...                         ...       ...           ...      ...   \n",
       "333680         missense variant  MODERATE                          \n",
       "333681         missense variant  MODERATE                          \n",
       "333682         missense variant  MODERATE                          \n",
       "333683         missense variant  MODERATE                          \n",
       "333684       intergenic variant  MODIFIER                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                                                  None   \n",
       "1                                  ENST00000641515.2           c.-36C>T   \n",
       "2                                                                  None   \n",
       "3                                                                  None   \n",
       "4                                  ENST00000341065.8        c.709-75C>G   \n",
       "...                   ...     ...                ...                ...   \n",
       "333680                             ENST00000682216.1          c.3682A>G   \n",
       "333681                             ENST00000682761.1          c.1708A>G   \n",
       "333682                             ENST00000682913.1          c.3817A>G   \n",
       "333683                             ENST00000684226.1          c.2809A>G   \n",
       "333684                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD  \\\n",
       "0                                       None  0/1   32  43  43  33  10   \n",
       "1                                       None  0/1   32  43  43  33  10   \n",
       "2                                       None  0/1   32  43  43  33  10   \n",
       "3                                       None  1/1  224  39  39   0  39   \n",
       "4                                       None  1/1  224  39  39   0  39   \n",
       "...                   ...                ...  ...  ...  ..  ..  ..  ..   \n",
       "333680  ENSP00000507656.1       p.Ile1228Val  1/1  161  31  31   1  30   \n",
       "333681  ENSP00000506900.1        p.Ile570Val  1/1  161  31  31   1  30   \n",
       "333682  ENSP00000507736.1       p.Ile1273Val  1/1  161  31  31   1  30   \n",
       "333683  ENSP00000507919.1        p.Ile937Val  1/1  161  31  31   1  30   \n",
       "333684                                  None  1/1   46   9   9   0   9   \n",
       "\n",
       "          FREQ        PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       23.26%   5.4148E-4  26   7   9   1                                     \n",
       "1       23.26%   5.4148E-4  26   7   9   1                                     \n",
       "2       23.26%   5.4148E-4  26   7   9   1                                     \n",
       "3         100%  3.6742E-23   0   0  30   9                                     \n",
       "4         100%  3.6742E-23   0   0  30   9                                     \n",
       "...        ...         ...  ..  ..  ..  ..                               ...   \n",
       "333680  96.77%  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "333681  96.77%  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "333682  96.77%  6.8754E-17   1   0  25   5  deleterious low confidence(0.02)   \n",
       "333683  96.77%  6.8754E-17   1   0  25   5     deleterious low confidence(0)   \n",
       "333684    100%   2.0568E-5   0   0   6   3                                     \n",
       "\n",
       "             PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0                                                                       \n",
       "1                                                                       \n",
       "2                                                                       \n",
       "3                      0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "4                      0.8880  0.7935  0.9092  0.9256  0.9443  0.9039   \n",
       "...               ...     ...     ...     ...     ...     ...     ...   \n",
       "333680  benign(0.067)                                                   \n",
       "333681  benign(0.044)                                                   \n",
       "333682  benign(0.124)                                                   \n",
       "333683  benign(0.031)                                                   \n",
       "333684                                                                  \n",
       "\n",
       "       gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                               \n",
       "1                                                                               \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...           ...            ...            ...            ...            ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03299   \n",
       "1                                                                     0.03299   \n",
       "2                                                                     0.03299   \n",
       "3                                                                      0.8921   \n",
       "4                                                                      0.8921   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02406              0        0.01429            0.5   \n",
       "1             0.02406              0        0.01429            0.5   \n",
       "2             0.02406              0        0.01429            0.5   \n",
       "3              0.7967         0.8813          0.913         0.9116   \n",
       "4              0.7967         0.8813          0.913         0.9116   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0                   0              0              0         0.1071   \n",
       "1                   0              0              0         0.1071   \n",
       "2                   0              0              0         0.1071   \n",
       "3              0.9261         0.9422         0.9114          0.933   \n",
       "4              0.9261         0.9422         0.9114          0.933   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                   0              0     0.5  gnomADg ASJ   \n",
       "1                   0              0     0.5  gnomADg ASJ   \n",
       "2                   0              0     0.5  gnomADg ASJ   \n",
       "3               0.902         0.9047  0.9443          EUR   \n",
       "4               0.902         0.9047  0.9443          EUR   \n",
       "...               ...            ...     ...          ...   \n",
       "333680                                                      \n",
       "333681                                                      \n",
       "333682                                                      \n",
       "333683                                                      \n",
       "333684                                                      \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding                 \n",
       "4                           protein coding          7/11   \n",
       "...                                    ...    ...    ...   \n",
       "333680                      non stop decay  25/28          \n",
       "333681                      non stop decay  10/12          \n",
       "333682                      protein coding  26/29          \n",
       "333683                      non stop decay  16/19          \n",
       "333684                                                     \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  nan               1          \n",
       "1                                  nan               1          \n",
       "2                                  nan               1          \n",
       "3                                  nan              -1          \n",
       "4                                  nan               1          \n",
       "...                                ...      ...    ...    ...   \n",
       "333680                          I1228V  Atc/Gtc     -1          \n",
       "333681                           I570V  Atc/Gtc     -1          \n",
       "333682                          I1273V  Atc/Gtc     -1          \n",
       "333683                           I937V  Atc/Gtc     -1          \n",
       "333684                             nan                          \n",
       "\n",
       "                    consequence Consequence_score  IMPACT_score  \n",
       "0       downstream gene variant              2/10           1.5  \n",
       "1           5 prime UTR variant              3/10           1.5  \n",
       "2       downstream gene variant              2/10           1.5  \n",
       "3       downstream gene variant              2/10           1.5  \n",
       "4                intron variant              2/10           1.5  \n",
       "...                         ...               ...           ...  \n",
       "333680         missense variant              7/10           5.0  \n",
       "333681         missense variant              7/10           5.0  \n",
       "333682         missense variant              7/10           5.0  \n",
       "333683         missense variant              7/10           5.0  \n",
       "333684       intergenic variant              1/10           1.5  \n",
       "\n",
       "[333685 rows x 69 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "f59ad4dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
       "       'Consequence', 'IMPACT', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
       "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
       "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
       "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
       "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
       "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
       "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
       "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
       "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
       "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
       "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
       "       'PUBMED', 'consequence', 'Consequence_score', 'IMPACT_score'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "ee044ab0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333680</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333681</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333682</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333683</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333684</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "4         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "333680                       .  chrY  13297744   T   C    Homozygous   \n",
       "333681                       .  chrY  13297744   T   C    Homozygous   \n",
       "333682                       .  chrY  13297744   T   C    Homozygous   \n",
       "333683                       .  chrY  13297744   T   C    Homozygous   \n",
       "333684                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1           5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "4                intron variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "333680         missense variant              7/10  MODERATE           5.0   \n",
       "333681         missense variant              7/10  MODERATE           5.0   \n",
       "333682         missense variant              7/10  MODERATE           5.0   \n",
       "333683         missense variant              7/10  MODERATE           5.0   \n",
       "333684       intergenic variant              1/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                                                             \n",
       "1                                                         ENST00000641515.2   \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4                                                         ENST00000341065.8   \n",
       "...              ...      ...                ...     ...                ...   \n",
       "333680                                                    ENST00000682216.1   \n",
       "333681                                                    ENST00000682761.1   \n",
       "333682                                                    ENST00000682913.1   \n",
       "333683                                                    ENST00000684226.1   \n",
       "333684                                                                        \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ SDP  \\\n",
       "0                    None                                  None  0/1   32  43   \n",
       "1                c.-36C>T                                  None  0/1   32  43   \n",
       "2                    None                                  None  0/1   32  43   \n",
       "3                    None                                  None  1/1  224  39   \n",
       "4             c.709-75C>G                                  None  1/1  224  39   \n",
       "...                   ...                ...                ...  ...  ...  ..   \n",
       "333680          c.3682A>G  ENSP00000507656.1       p.Ile1228Val  1/1  161  31   \n",
       "333681          c.1708A>G  ENSP00000506900.1        p.Ile570Val  1/1  161  31   \n",
       "333682          c.3817A>G  ENSP00000507736.1       p.Ile1273Val  1/1  161  31   \n",
       "333683          c.2809A>G  ENSP00000507919.1        p.Ile937Val  1/1  161  31   \n",
       "333684               None                                  None  1/1   46   9   \n",
       "\n",
       "        DP  RD  AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       43  33  10  23.26%   5.4148E-4  26   7   9   1   \n",
       "1       43  33  10  23.26%   5.4148E-4  26   7   9   1   \n",
       "2       43  33  10  23.26%   5.4148E-4  26   7   9   1   \n",
       "3       39   0  39    100%  3.6742E-23   0   0  30   9   \n",
       "4       39   0  39    100%  3.6742E-23   0   0  30   9   \n",
       "...     ..  ..  ..     ...         ...  ..  ..  ..  ..   \n",
       "333680  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333681  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333682  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333683  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333684   9   0   9    100%   2.0568E-5   0   0   6   3   \n",
       "\n",
       "                                    SIFT       PolyPhen      AF  AFR_AF  \\\n",
       "0                                                                         \n",
       "1                                                                         \n",
       "2                                                                         \n",
       "3                                                        0.8880  0.7935   \n",
       "4                                                        0.8880  0.7935   \n",
       "...                                  ...            ...     ...     ...   \n",
       "333680     deleterious low confidence(0)  benign(0.067)                   \n",
       "333681     deleterious low confidence(0)  benign(0.044)                   \n",
       "333682  deleterious low confidence(0.02)  benign(0.124)                   \n",
       "333683     deleterious low confidence(0)  benign(0.031)                   \n",
       "333684                                                                    \n",
       "\n",
       "        AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                  \n",
       "1                                                                  \n",
       "2                                                                  \n",
       "3       0.9092  0.9256  0.9443  0.9039                             \n",
       "4       0.9092  0.9256  0.9443  0.9039                             \n",
       "...        ...     ...     ...     ...        ...            ...   \n",
       "333680                                                             \n",
       "333681                                                             \n",
       "333682                                                             \n",
       "333683                                                             \n",
       "333684                                                             \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03299        0.02406   \n",
       "1                                                      0.03299        0.02406   \n",
       "2                                                      0.03299        0.02406   \n",
       "3                                                       0.8921         0.7967   \n",
       "4                                                       0.8921         0.7967   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0                   0        0.01429            0.5              0   \n",
       "1                   0        0.01429            0.5              0   \n",
       "2                   0        0.01429            0.5              0   \n",
       "3              0.8813          0.913         0.9116         0.9261   \n",
       "4              0.8813          0.913         0.9116         0.9261   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0              0         0.1071              0   \n",
       "1                   0              0         0.1071              0   \n",
       "2                   0              0         0.1071              0   \n",
       "3              0.9422         0.9114          0.933          0.902   \n",
       "4              0.9422         0.9114          0.933          0.902   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                   0     0.5  gnomADg ASJ   \n",
       "1                   0     0.5  gnomADg ASJ   \n",
       "2                   0     0.5  gnomADg ASJ   \n",
       "3              0.9047  0.9443          EUR   \n",
       "4              0.9047  0.9443          EUR   \n",
       "...               ...     ...          ...   \n",
       "333680                                       \n",
       "333681                                       \n",
       "333682                                       \n",
       "333683                                       \n",
       "333684                                       \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding                 \n",
       "4                           protein coding          7/11   \n",
       "...                                    ...    ...    ...   \n",
       "333680                      non stop decay  25/28          \n",
       "333681                      non stop decay  10/12          \n",
       "333682                      protein coding  26/29          \n",
       "333683                      non stop decay  16/19          \n",
       "333684                                                     \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  nan               1         \n",
       "1                                  nan               1         \n",
       "2                                  nan               1         \n",
       "3                                  nan              -1         \n",
       "4                                  nan               1         \n",
       "...                                ...      ...    ...    ...  \n",
       "333680                          I1228V  Atc/Gtc     -1         \n",
       "333681                           I570V  Atc/Gtc     -1         \n",
       "333682                          I1273V  Atc/Gtc     -1         \n",
       "333683                           I937V  Atc/Gtc     -1         \n",
       "333684                             nan                         \n",
       "\n",
       "[333685 rows x 68 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "317dcd8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merged_2.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/Backup_files/KHCDPRGPTTL116_final.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "e6e3dd75",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LDLR-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NOS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRKAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>TECRL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>CASQ2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>TRDN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>PCSK9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>APOB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name\n",
       "0       LDLR\n",
       "1   LDLR-AS1\n",
       "2       NOS3\n",
       "3     PRKAG2\n",
       "4       LRP6\n",
       "..       ...\n",
       "74     TECRL\n",
       "75     CASQ2\n",
       "76      TRDN\n",
       "77     PCSK9\n",
       "78      APOB\n",
       "\n",
       "[79 rows x 1 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/OneDrive/Desktop/cardiac_genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd9c5fe3",
   "metadata": {},
   "source": [
    "# Multiple genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "1b682144",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>233</td>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "      <td>34</td>\n",
       "      <td>58</td>\n",
       "      <td>63.04%</td>\n",
       "      <td>4.4849E-24</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>233</td>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "      <td>34</td>\n",
       "      <td>58</td>\n",
       "      <td>63.04%</td>\n",
       "      <td>4.4849E-24</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "      <td>1</td>\n",
       "      <td>144</td>\n",
       "      <td>99.31%</td>\n",
       "      <td>1.5678E-84</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>108</td>\n",
       "      <td>36</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11,LOC107985728</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0935E-50</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>36</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7498</td>\n",
       "      <td>0.4039</td>\n",
       "      <td>0.9366</td>\n",
       "      <td>0.6944</td>\n",
       "      <td>0.9811</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8215</td>\n",
       "      <td>0.4795</td>\n",
       "      <td>0.9693</td>\n",
       "      <td>0.9213</td>\n",
       "      <td>0.9732</td>\n",
       "      <td>0.6886</td>\n",
       "      <td>0.9618</td>\n",
       "      <td>0.9427</td>\n",
       "      <td>0.9802</td>\n",
       "      <td>0.8549</td>\n",
       "      <td>0.9098</td>\n",
       "      <td>0.9811</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309004</th>\n",
       "      <td>RAB39B</td>\n",
       "      <td>rs3105275</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155259147</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369454.4</td>\n",
       "      <td>c.*1656T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>83</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>19</td>\n",
       "      <td>22</td>\n",
       "      <td>53.66%</td>\n",
       "      <td>4.8137E-9</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.5960</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>0.6202</td>\n",
       "      <td>0.7251</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>0.7201</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6161</td>\n",
       "      <td>0.2976</td>\n",
       "      <td>0.6866</td>\n",
       "      <td>0.639</td>\n",
       "      <td>0.762</td>\n",
       "      <td>0.7219</td>\n",
       "      <td>0.7682</td>\n",
       "      <td>0.7881</td>\n",
       "      <td>0.7577</td>\n",
       "      <td>0.6284</td>\n",
       "      <td>0.7201</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309005</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309006</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369449.7</td>\n",
       "      <td>c.*19C&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309007</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309008</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>309009 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene Name         rsID CHROM        POS REF ALT  \\\n",
       "0                     OR4F5  rs201219564  chr1      69270   A   G   \n",
       "1                     OR4F5  rs201219564  chr1      69270   A   G   \n",
       "2                     OR4F5    rs2691305  chr1      69511   A   G   \n",
       "3                     OR4F5  rs200676709  chr1      69897   T   C   \n",
       "4       SAMD11,LOC107985728  rs112703963  chr1     924533   A   G   \n",
       "...                     ...          ...   ...        ...  ..  ..   \n",
       "309004               RAB39B    rs3105275  chrX  155259147   A   G   \n",
       "309005                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309006                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309007                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309008                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "\n",
       "            Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous         synonymous variant              3/10       LOW   \n",
       "1       Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2         Homozygous           missense variant              7/10  MODERATE   \n",
       "3       Heterozygous         synonymous variant              3/10       LOW   \n",
       "4         Homozygous      upstream gene variant              2/10  MODIFIER   \n",
       "...              ...                        ...               ...       ...   \n",
       "309004  Heterozygous        3 prime UTR variant              3/10  MODIFIER   \n",
       "309005  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "309006  Heterozygous        3 prime UTR variant              3/10  MODIFIER   \n",
       "309007  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "309008  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                2.5                                                     \n",
       "1                1.5                                                     \n",
       "2                5.0                                                     \n",
       "3                2.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "309004           1.5                 benign                              \n",
       "309005           1.5                 benign                              \n",
       "309006           1.5                 benign                              \n",
       "309007           1.5                 benign                              \n",
       "309008           1.5                 benign                              \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                       None                      \n",
       "2       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "4                                       None                      \n",
       "...                   ...                ...                ...   \n",
       "309004  ENST00000369454.4         c.*1656T>C                      \n",
       "309005                                  None                      \n",
       "309006  ENST00000369449.7           c.*19C>A                      \n",
       "309007                                  None                      \n",
       "309008                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL  \\\n",
       "0              p.Ser81%3D  0/1  233   92   92  34   58  63.04%  4.4849E-24   \n",
       "1                    None  0/1  233   92   92  34   58  63.04%  4.4849E-24   \n",
       "2             p.Thr162Ala  1/1  255  145  145   1  144  99.31%  1.5678E-84   \n",
       "3             p.Ser290%3D  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "4                    None  1/1  255   85   85   0   85    100%  1.0935E-50   \n",
       "...                   ...  ...  ...  ...  ...  ..  ...     ...         ...   \n",
       "309004               None  0/1   83   41   41  19   22  53.66%   4.8137E-9   \n",
       "309005               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309006               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309007               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309008               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "\n",
       "       RDF RDR  ADF ADR             SIFT   PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0       27   7   38  20                                                       \n",
       "1       27   7   38  20                                                       \n",
       "2        1   0  108  36  tolerated(0.92)  benign(0)                           \n",
       "3        6  12    5   5                              0.6881   0.407  0.6254   \n",
       "4        0   0   49  36                              0.7498  0.4039  0.9366   \n",
       "...     ..  ..  ...  ..              ...        ...     ...     ...     ...   \n",
       "309004  12   7   16   6                              0.5960  0.2383  0.6202   \n",
       "309005  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309006  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309007  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309008  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "\n",
       "        EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                                   0.838         0.3591         0.7932   \n",
       "1                                   0.838         0.3591         0.7932   \n",
       "2                                  0.9497         0.6075         0.9514   \n",
       "3        0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "4       0.6944  0.9811  0.9039                                            \n",
       "...        ...     ...     ...        ...            ...            ...   \n",
       "309004  0.7251  0.8029  0.7201                                            \n",
       "309005  0.7277  0.8055  0.7173                                            \n",
       "309006  0.7277  0.8055  0.7173                                            \n",
       "309007  0.7277  0.8055  0.7173                                            \n",
       "309008  0.7277  0.8055  0.7173                                            \n",
       "\n",
       "       gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0              0.8482         0.9984         0.8821         0.9146   \n",
       "1              0.8482         0.9984         0.8821         0.9146   \n",
       "2              0.9767         0.9994         0.9916         0.9726   \n",
       "3              0.8131         0.7793         0.8652         0.8438   \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "309004                                                               \n",
       "309005                                                               \n",
       "309006                                                               \n",
       "309007                                                               \n",
       "309008                                                               \n",
       "\n",
       "       gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0               0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "1               0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "2              0.9506         0.9854      0.846         0.5948          0.998   \n",
       "3              0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "4                                        0.8215         0.4795         0.9693   \n",
       "...               ...            ...        ...            ...            ...   \n",
       "309004                                   0.6161         0.2976         0.6866   \n",
       "309005                                   0.6616         0.4499         0.6906   \n",
       "309006                                   0.6616         0.4499         0.6906   \n",
       "309007                                   0.6616         0.4499         0.6906   \n",
       "309008                                   0.6616         0.4499         0.6906   \n",
       "\n",
       "       gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0              0.6945         0.8033         0.9961         0.9294   \n",
       "1              0.6945         0.8033         0.9961         0.9294   \n",
       "2              0.8951         0.9784         0.9998         0.9907   \n",
       "3              0.4782         0.6714         0.6047         0.6987   \n",
       "4              0.9213         0.9732         0.6886         0.9618   \n",
       "...               ...            ...            ...            ...   \n",
       "309004          0.639          0.762         0.7219         0.7682   \n",
       "309005         0.6639         0.7619         0.7254         0.7679   \n",
       "309006         0.6639         0.7619         0.7254         0.7679   \n",
       "309007         0.6639         0.7619         0.7254         0.7679   \n",
       "309008         0.6639         0.7619         0.7254         0.7679   \n",
       "\n",
       "       gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  \\\n",
       "0              0.7857         0.8798         0.6087         0.8855  0.9984   \n",
       "1              0.7857         0.8798         0.6087         0.8855  0.9984   \n",
       "2                 0.9         0.9674         0.8624         0.9772  0.9998   \n",
       "3              0.6441         0.6751         0.4659          0.622   0.876   \n",
       "4              0.9427         0.9802         0.8549         0.9098  0.9811   \n",
       "...               ...            ...            ...            ...     ...   \n",
       "309004         0.7881         0.7577         0.6284         0.7201  0.8029   \n",
       "309005         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309006         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309007         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309008         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "\n",
       "        MAX_AF_POPS                         BIOTYPE EXON INTRON  \\\n",
       "0       gnomADe EAS                  protein coding  3/3          \n",
       "1       gnomADe EAS                 TF binding site               \n",
       "2       gnomADg EAS                  protein coding  3/3          \n",
       "3               EAS                  protein coding  3/3          \n",
       "4               EUR                  protein coding               \n",
       "...             ...                             ...  ...    ...   \n",
       "309004          EUR                  protein coding  2/2          \n",
       "309005          EUR                  protein coding               \n",
       "309006          EUR                  protein coding  6/6          \n",
       "309007          EUR  protein coding CDS not defined               \n",
       "309008          EUR  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Gene Match  \\\n",
       "0                                  S81  tcA/tcG      1                No   \n",
       "1                                  nan                                No   \n",
       "2                                T162A  Aca/Gca      1                No   \n",
       "3                                 S290  tcT/tcC      1                No   \n",
       "4                                  nan               1                No   \n",
       "...                                ...      ...    ...    ...        ...   \n",
       "309004                             nan              -1                No   \n",
       "309005                             nan              -1                No   \n",
       "309006                             nan              -1                No   \n",
       "309007                             nan              -1                No   \n",
       "309008                             nan              -1                No   \n",
       "\n",
       "       Matched_Gene  \n",
       "0                    \n",
       "1                    \n",
       "2                    \n",
       "3                    \n",
       "4                    \n",
       "...             ...  \n",
       "309004               \n",
       "309005               \n",
       "309006               \n",
       "309007               \n",
       "309008               \n",
       "\n",
       "[309009 rows x 70 columns]"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "70fb9bbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           307043\n",
       "ABCC8         615\n",
       "STAT3         203\n",
       "HNF1A         191\n",
       "WFS1          140\n",
       "KCNJ11         86\n",
       "ERBB3          71\n",
       "INS            65\n",
       "BLK            64\n",
       "PARK7          56\n",
       "PRKAG2         53\n",
       "C1QTNF6        47\n",
       "INSR           41\n",
       "GCK            33\n",
       "LIPC           28\n",
       "TAGAP          28\n",
       "TCF7L2         26\n",
       "BACH2          26\n",
       "ACADS          24\n",
       "KLF11          22\n",
       "PAX4           21\n",
       "HNF4A          20\n",
       "IL2RA          20\n",
       "MLKL           12\n",
       "PPP1R3A        12\n",
       "IRS1           10\n",
       "CCR5           10\n",
       "PTPN22          9\n",
       "WRN             9\n",
       "NEUROD1         9\n",
       "SLC2A2          6\n",
       "HNF1B           5\n",
       "SH2B3           4\n",
       "Name: Matched_Gene, dtype: int64"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.Matched_Gene.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "c1a5b012",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS</td>\n",
       "      <td>Type_1, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type_1, Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CEL</td>\n",
       "      <td>Type_1, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HLADQ</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HLADR</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PTPN22</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IL2</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ERBB3</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BACH2</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>C1QTNF6</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PRKQ</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>RGS1</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CCR5</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TAGAP</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>STAT3</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FOXP3</td>\n",
       "      <td>Type_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>IRS1</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SLC2A2</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>WFS1</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCK</td>\n",
       "      <td>Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PAX4</td>\n",
       "      <td>Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>WRN</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ABCC8</td>\n",
       "      <td>Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PDX1</td>\n",
       "      <td>Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>AKT2</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HNF4A</td>\n",
       "      <td>Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>KLF11</td>\n",
       "      <td>Type_2, Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ENPP1</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PPP1R3A</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>ACADS</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>LIPC</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>INSR</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>Type_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>HNF1B</td>\n",
       "      <td>Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MLKL</td>\n",
       "      <td>Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NEUROD1</td>\n",
       "      <td>Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>BLK</td>\n",
       "      <td>Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>Mody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PRKAG2</td>\n",
       "      <td>Mody</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Matched_Gene            Gene Match\n",
       "0         CTLA4                Type_1\n",
       "1           INS          Type_1, Mody\n",
       "2         IL2RA                Type_1\n",
       "3         HNF1A  Type_1, Type_2, Mody\n",
       "4           CEL          Type_1, Mody\n",
       "5         HLADQ                Type_1\n",
       "6         HLADR                Type_1\n",
       "7        PTPN22                Type_1\n",
       "8           IL2                Type_1\n",
       "9         ERBB3                Type_1\n",
       "10        BACH2                Type_1\n",
       "11      C1QTNF6                Type_1\n",
       "12         PRKQ                Type_1\n",
       "13         RGS1                Type_1\n",
       "14      IL18RAP                Type_1\n",
       "15         CCR5                Type_1\n",
       "16        TAGAP                Type_1\n",
       "17        SH2B3                Type_1\n",
       "18        STAT3                Type_1\n",
       "19        FOXP3                Type_1\n",
       "20         IRS1                Type_2\n",
       "21       SLC2A2                Type_2\n",
       "22         WFS1                Type_2\n",
       "23          GCK          Type_2, Mody\n",
       "24         PAX4          Type_2, Mody\n",
       "25          WRN                Type_2\n",
       "26       KCNJ11          Type_2, Mody\n",
       "27        ABCC8          Type_2, Mody\n",
       "28         PDX1          Type_2, Mody\n",
       "29         AKT2                Type_2\n",
       "30        HNF4A          Type_2, Mody\n",
       "31        KLF11          Type_2, Mody\n",
       "32        ENPP1                Type_2\n",
       "33      PPP1R3A                Type_2\n",
       "34        ACADS                Type_2\n",
       "35         LIPC                Type_2\n",
       "36         INSR                Type_2\n",
       "37       TCF7L2                Type_2\n",
       "38        HNF1B                  Mody\n",
       "39         MLKL                  Mody\n",
       "40      NEUROD1                  Mody\n",
       "41          BLK                  Mody\n",
       "42        PARK7                  Mody\n",
       "43       PRKAG2                  Mody"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = df_gene.rename({'Gene Name':'Matched_Gene'}, axis=1)\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "08c27a5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>233</td>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "      <td>34</td>\n",
       "      <td>58</td>\n",
       "      <td>63.04%</td>\n",
       "      <td>4.4849E-24</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>233</td>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "      <td>34</td>\n",
       "      <td>58</td>\n",
       "      <td>63.04%</td>\n",
       "      <td>4.4849E-24</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "      <td>1</td>\n",
       "      <td>144</td>\n",
       "      <td>99.31%</td>\n",
       "      <td>1.5678E-84</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>108</td>\n",
       "      <td>36</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11,LOC107985728</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0935E-50</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>36</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7498</td>\n",
       "      <td>0.4039</td>\n",
       "      <td>0.9366</td>\n",
       "      <td>0.6944</td>\n",
       "      <td>0.9811</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8215</td>\n",
       "      <td>0.4795</td>\n",
       "      <td>0.9693</td>\n",
       "      <td>0.9213</td>\n",
       "      <td>0.9732</td>\n",
       "      <td>0.6886</td>\n",
       "      <td>0.9618</td>\n",
       "      <td>0.9427</td>\n",
       "      <td>0.9802</td>\n",
       "      <td>0.8549</td>\n",
       "      <td>0.9098</td>\n",
       "      <td>0.9811</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309004</th>\n",
       "      <td>RAB39B</td>\n",
       "      <td>rs3105275</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155259147</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369454.4</td>\n",
       "      <td>c.*1656T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>83</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>19</td>\n",
       "      <td>22</td>\n",
       "      <td>53.66%</td>\n",
       "      <td>4.8137E-9</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.5960</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>0.6202</td>\n",
       "      <td>0.7251</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>0.7201</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6161</td>\n",
       "      <td>0.2976</td>\n",
       "      <td>0.6866</td>\n",
       "      <td>0.639</td>\n",
       "      <td>0.762</td>\n",
       "      <td>0.7219</td>\n",
       "      <td>0.7682</td>\n",
       "      <td>0.7881</td>\n",
       "      <td>0.7577</td>\n",
       "      <td>0.6284</td>\n",
       "      <td>0.7201</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309005</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309006</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369449.7</td>\n",
       "      <td>c.*19C&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309007</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309008</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>309009 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene Name         rsID CHROM        POS REF ALT  \\\n",
       "0                     OR4F5  rs201219564  chr1      69270   A   G   \n",
       "1                     OR4F5  rs201219564  chr1      69270   A   G   \n",
       "2                     OR4F5    rs2691305  chr1      69511   A   G   \n",
       "3                     OR4F5  rs200676709  chr1      69897   T   C   \n",
       "4       SAMD11,LOC107985728  rs112703963  chr1     924533   A   G   \n",
       "...                     ...          ...   ...        ...  ..  ..   \n",
       "309004               RAB39B    rs3105275  chrX  155259147   A   G   \n",
       "309005                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309006                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309007                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309008                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "\n",
       "            Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous         synonymous variant              3/10       LOW   \n",
       "1       Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2         Homozygous           missense variant              7/10  MODERATE   \n",
       "3       Heterozygous         synonymous variant              3/10       LOW   \n",
       "4         Homozygous      upstream gene variant              2/10  MODIFIER   \n",
       "...              ...                        ...               ...       ...   \n",
       "309004  Heterozygous        3 prime UTR variant              3/10  MODIFIER   \n",
       "309005  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "309006  Heterozygous        3 prime UTR variant              3/10  MODIFIER   \n",
       "309007  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "309008  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                2.5                                                     \n",
       "1                1.5                                                     \n",
       "2                5.0                                                     \n",
       "3                2.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "309004           1.5                 benign                              \n",
       "309005           1.5                 benign                              \n",
       "309006           1.5                 benign                              \n",
       "309007           1.5                 benign                              \n",
       "309008           1.5                 benign                              \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                       None                      \n",
       "2       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "4                                       None                      \n",
       "...                   ...                ...                ...   \n",
       "309004  ENST00000369454.4         c.*1656T>C                      \n",
       "309005                                  None                      \n",
       "309006  ENST00000369449.7           c.*19C>A                      \n",
       "309007                                  None                      \n",
       "309008                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL  \\\n",
       "0              p.Ser81%3D  0/1  233   92   92  34   58  63.04%  4.4849E-24   \n",
       "1                    None  0/1  233   92   92  34   58  63.04%  4.4849E-24   \n",
       "2             p.Thr162Ala  1/1  255  145  145   1  144  99.31%  1.5678E-84   \n",
       "3             p.Ser290%3D  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "4                    None  1/1  255   85   85   0   85    100%  1.0935E-50   \n",
       "...                   ...  ...  ...  ...  ...  ..  ...     ...         ...   \n",
       "309004               None  0/1   83   41   41  19   22  53.66%   4.8137E-9   \n",
       "309005               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309006               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309007               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309008               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "\n",
       "       RDF RDR  ADF ADR             SIFT   PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0       27   7   38  20                                                       \n",
       "1       27   7   38  20                                                       \n",
       "2        1   0  108  36  tolerated(0.92)  benign(0)                           \n",
       "3        6  12    5   5                              0.6881   0.407  0.6254   \n",
       "4        0   0   49  36                              0.7498  0.4039  0.9366   \n",
       "...     ..  ..  ...  ..              ...        ...     ...     ...     ...   \n",
       "309004  12   7   16   6                              0.5960  0.2383  0.6202   \n",
       "309005  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309006  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309007  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309008  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "\n",
       "        EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                                   0.838         0.3591         0.7932   \n",
       "1                                   0.838         0.3591         0.7932   \n",
       "2                                  0.9497         0.6075         0.9514   \n",
       "3        0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "4       0.6944  0.9811  0.9039                                            \n",
       "...        ...     ...     ...        ...            ...            ...   \n",
       "309004  0.7251  0.8029  0.7201                                            \n",
       "309005  0.7277  0.8055  0.7173                                            \n",
       "309006  0.7277  0.8055  0.7173                                            \n",
       "309007  0.7277  0.8055  0.7173                                            \n",
       "309008  0.7277  0.8055  0.7173                                            \n",
       "\n",
       "       gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0              0.8482         0.9984         0.8821         0.9146   \n",
       "1              0.8482         0.9984         0.8821         0.9146   \n",
       "2              0.9767         0.9994         0.9916         0.9726   \n",
       "3              0.8131         0.7793         0.8652         0.8438   \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "309004                                                               \n",
       "309005                                                               \n",
       "309006                                                               \n",
       "309007                                                               \n",
       "309008                                                               \n",
       "\n",
       "       gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0               0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "1               0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "2              0.9506         0.9854      0.846         0.5948          0.998   \n",
       "3              0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "4                                        0.8215         0.4795         0.9693   \n",
       "...               ...            ...        ...            ...            ...   \n",
       "309004                                   0.6161         0.2976         0.6866   \n",
       "309005                                   0.6616         0.4499         0.6906   \n",
       "309006                                   0.6616         0.4499         0.6906   \n",
       "309007                                   0.6616         0.4499         0.6906   \n",
       "309008                                   0.6616         0.4499         0.6906   \n",
       "\n",
       "       gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0              0.6945         0.8033         0.9961         0.9294   \n",
       "1              0.6945         0.8033         0.9961         0.9294   \n",
       "2              0.8951         0.9784         0.9998         0.9907   \n",
       "3              0.4782         0.6714         0.6047         0.6987   \n",
       "4              0.9213         0.9732         0.6886         0.9618   \n",
       "...               ...            ...            ...            ...   \n",
       "309004          0.639          0.762         0.7219         0.7682   \n",
       "309005         0.6639         0.7619         0.7254         0.7679   \n",
       "309006         0.6639         0.7619         0.7254         0.7679   \n",
       "309007         0.6639         0.7619         0.7254         0.7679   \n",
       "309008         0.6639         0.7619         0.7254         0.7679   \n",
       "\n",
       "       gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  \\\n",
       "0              0.7857         0.8798         0.6087         0.8855  0.9984   \n",
       "1              0.7857         0.8798         0.6087         0.8855  0.9984   \n",
       "2                 0.9         0.9674         0.8624         0.9772  0.9998   \n",
       "3              0.6441         0.6751         0.4659          0.622   0.876   \n",
       "4              0.9427         0.9802         0.8549         0.9098  0.9811   \n",
       "...               ...            ...            ...            ...     ...   \n",
       "309004         0.7881         0.7577         0.6284         0.7201  0.8029   \n",
       "309005         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309006         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309007         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309008         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "\n",
       "        MAX_AF_POPS                         BIOTYPE EXON INTRON  \\\n",
       "0       gnomADe EAS                  protein coding  3/3          \n",
       "1       gnomADe EAS                 TF binding site               \n",
       "2       gnomADg EAS                  protein coding  3/3          \n",
       "3               EAS                  protein coding  3/3          \n",
       "4               EUR                  protein coding               \n",
       "...             ...                             ...  ...    ...   \n",
       "309004          EUR                  protein coding  2/2          \n",
       "309005          EUR                  protein coding               \n",
       "309006          EUR                  protein coding  6/6          \n",
       "309007          EUR  protein coding CDS not defined               \n",
       "309008          EUR  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \n",
       "0                                  S81  tcA/tcG      1                      \n",
       "1                                  nan                                      \n",
       "2                                T162A  Aca/Gca      1                      \n",
       "3                                 S290  tcT/tcC      1                      \n",
       "4                                  nan               1                      \n",
       "...                                ...      ...    ...    ...          ...  \n",
       "309004                             nan              -1                      \n",
       "309005                             nan              -1                      \n",
       "309006                             nan              -1                      \n",
       "309007                             nan              -1                      \n",
       "309008                             nan              -1                      \n",
       "\n",
       "[309009 rows x 69 columns]"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2.drop(columns=['Gene Match'], axis=1)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "31ca44c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>233</td>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "      <td>34</td>\n",
       "      <td>58</td>\n",
       "      <td>63.04%</td>\n",
       "      <td>4.4849E-24</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>233</td>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "      <td>34</td>\n",
       "      <td>58</td>\n",
       "      <td>63.04%</td>\n",
       "      <td>4.4849E-24</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "      <td>1</td>\n",
       "      <td>144</td>\n",
       "      <td>99.31%</td>\n",
       "      <td>1.5678E-84</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>108</td>\n",
       "      <td>36</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11,LOC107985728</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0935E-50</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>36</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7498</td>\n",
       "      <td>0.4039</td>\n",
       "      <td>0.9366</td>\n",
       "      <td>0.6944</td>\n",
       "      <td>0.9811</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8215</td>\n",
       "      <td>0.4795</td>\n",
       "      <td>0.9693</td>\n",
       "      <td>0.9213</td>\n",
       "      <td>0.9732</td>\n",
       "      <td>0.6886</td>\n",
       "      <td>0.9618</td>\n",
       "      <td>0.9427</td>\n",
       "      <td>0.9802</td>\n",
       "      <td>0.8549</td>\n",
       "      <td>0.9098</td>\n",
       "      <td>0.9811</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309004</th>\n",
       "      <td>RAB39B</td>\n",
       "      <td>rs3105275</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155259147</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369454.4</td>\n",
       "      <td>c.*1656T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>83</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>19</td>\n",
       "      <td>22</td>\n",
       "      <td>53.66%</td>\n",
       "      <td>4.8137E-9</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.5960</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>0.6202</td>\n",
       "      <td>0.7251</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>0.7201</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6161</td>\n",
       "      <td>0.2976</td>\n",
       "      <td>0.6866</td>\n",
       "      <td>0.639</td>\n",
       "      <td>0.762</td>\n",
       "      <td>0.7219</td>\n",
       "      <td>0.7682</td>\n",
       "      <td>0.7881</td>\n",
       "      <td>0.7577</td>\n",
       "      <td>0.6284</td>\n",
       "      <td>0.7201</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309005</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309006</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369449.7</td>\n",
       "      <td>c.*19C&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309007</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309008</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.4499</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.6639</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.7254</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>0.7176</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>309009 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene Name         rsID CHROM        POS REF ALT  \\\n",
       "0                     OR4F5  rs201219564  chr1      69270   A   G   \n",
       "1                     OR4F5  rs201219564  chr1      69270   A   G   \n",
       "2                     OR4F5    rs2691305  chr1      69511   A   G   \n",
       "3                     OR4F5  rs200676709  chr1      69897   T   C   \n",
       "4       SAMD11,LOC107985728  rs112703963  chr1     924533   A   G   \n",
       "...                     ...          ...   ...        ...  ..  ..   \n",
       "309004               RAB39B    rs3105275  chrX  155259147   A   G   \n",
       "309005                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309006                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309007                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "309008                CLIC2     rs559165  chrX  155277884   G   T   \n",
       "\n",
       "            Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous         synonymous variant              3/10       LOW   \n",
       "1       Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2         Homozygous           missense variant              7/10  MODERATE   \n",
       "3       Heterozygous         synonymous variant              3/10       LOW   \n",
       "4         Homozygous      upstream gene variant              2/10  MODIFIER   \n",
       "...              ...                        ...               ...       ...   \n",
       "309004  Heterozygous        3 prime UTR variant              3/10  MODIFIER   \n",
       "309005  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "309006  Heterozygous        3 prime UTR variant              3/10  MODIFIER   \n",
       "309007  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "309008  Heterozygous    downstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                2.5                                                     \n",
       "1                1.5                                                     \n",
       "2                5.0                                                     \n",
       "3                2.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "309004           1.5                 benign                              \n",
       "309005           1.5                 benign                              \n",
       "309006           1.5                 benign                              \n",
       "309007           1.5                 benign                              \n",
       "309008           1.5                 benign                              \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                       None                      \n",
       "2       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "4                                       None                      \n",
       "...                   ...                ...                ...   \n",
       "309004  ENST00000369454.4         c.*1656T>C                      \n",
       "309005                                  None                      \n",
       "309006  ENST00000369449.7           c.*19C>A                      \n",
       "309007                                  None                      \n",
       "309008                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL  \\\n",
       "0              p.Ser81%3D  0/1  233   92   92  34   58  63.04%  4.4849E-24   \n",
       "1                    None  0/1  233   92   92  34   58  63.04%  4.4849E-24   \n",
       "2             p.Thr162Ala  1/1  255  145  145   1  144  99.31%  1.5678E-84   \n",
       "3             p.Ser290%3D  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "4                    None  1/1  255   85   85   0   85    100%  1.0935E-50   \n",
       "...                   ...  ...  ...  ...  ...  ..  ...     ...         ...   \n",
       "309004               None  0/1   83   41   41  19   22  53.66%   4.8137E-9   \n",
       "309005               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309006               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309007               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "309008               None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "\n",
       "       RDF RDR  ADF ADR             SIFT   PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0       27   7   38  20                                                       \n",
       "1       27   7   38  20                                                       \n",
       "2        1   0  108  36  tolerated(0.92)  benign(0)                           \n",
       "3        6  12    5   5                              0.6881   0.407  0.6254   \n",
       "4        0   0   49  36                              0.7498  0.4039  0.9366   \n",
       "...     ..  ..  ...  ..              ...        ...     ...     ...     ...   \n",
       "309004  12   7   16   6                              0.5960  0.2383  0.6202   \n",
       "309005  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309006  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309007  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "309008  18   0   10   0                              0.6458  0.4108   0.645   \n",
       "\n",
       "        EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                                   0.838         0.3591         0.7932   \n",
       "1                                   0.838         0.3591         0.7932   \n",
       "2                                  0.9497         0.6075         0.9514   \n",
       "3        0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "4       0.6944  0.9811  0.9039                                            \n",
       "...        ...     ...     ...        ...            ...            ...   \n",
       "309004  0.7251  0.8029  0.7201                                            \n",
       "309005  0.7277  0.8055  0.7173                                            \n",
       "309006  0.7277  0.8055  0.7173                                            \n",
       "309007  0.7277  0.8055  0.7173                                            \n",
       "309008  0.7277  0.8055  0.7173                                            \n",
       "\n",
       "       gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0              0.8482         0.9984         0.8821         0.9146   \n",
       "1              0.8482         0.9984         0.8821         0.9146   \n",
       "2              0.9767         0.9994         0.9916         0.9726   \n",
       "3              0.8131         0.7793         0.8652         0.8438   \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "309004                                                               \n",
       "309005                                                               \n",
       "309006                                                               \n",
       "309007                                                               \n",
       "309008                                                               \n",
       "\n",
       "       gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0               0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "1               0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "2              0.9506         0.9854      0.846         0.5948          0.998   \n",
       "3              0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "4                                        0.8215         0.4795         0.9693   \n",
       "...               ...            ...        ...            ...            ...   \n",
       "309004                                   0.6161         0.2976         0.6866   \n",
       "309005                                   0.6616         0.4499         0.6906   \n",
       "309006                                   0.6616         0.4499         0.6906   \n",
       "309007                                   0.6616         0.4499         0.6906   \n",
       "309008                                   0.6616         0.4499         0.6906   \n",
       "\n",
       "       gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0              0.6945         0.8033         0.9961         0.9294   \n",
       "1              0.6945         0.8033         0.9961         0.9294   \n",
       "2              0.8951         0.9784         0.9998         0.9907   \n",
       "3              0.4782         0.6714         0.6047         0.6987   \n",
       "4              0.9213         0.9732         0.6886         0.9618   \n",
       "...               ...            ...            ...            ...   \n",
       "309004          0.639          0.762         0.7219         0.7682   \n",
       "309005         0.6639         0.7619         0.7254         0.7679   \n",
       "309006         0.6639         0.7619         0.7254         0.7679   \n",
       "309007         0.6639         0.7619         0.7254         0.7679   \n",
       "309008         0.6639         0.7619         0.7254         0.7679   \n",
       "\n",
       "       gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  \\\n",
       "0              0.7857         0.8798         0.6087         0.8855  0.9984   \n",
       "1              0.7857         0.8798         0.6087         0.8855  0.9984   \n",
       "2                 0.9         0.9674         0.8624         0.9772  0.9998   \n",
       "3              0.6441         0.6751         0.4659          0.622   0.876   \n",
       "4              0.9427         0.9802         0.8549         0.9098  0.9811   \n",
       "...               ...            ...            ...            ...     ...   \n",
       "309004         0.7881         0.7577         0.6284         0.7201  0.8029   \n",
       "309005         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309006         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309007         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "309008         0.7778          0.758         0.6775         0.7176  0.8055   \n",
       "\n",
       "        MAX_AF_POPS                         BIOTYPE EXON INTRON  \\\n",
       "0       gnomADe EAS                  protein coding  3/3          \n",
       "1       gnomADe EAS                 TF binding site               \n",
       "2       gnomADg EAS                  protein coding  3/3          \n",
       "3               EAS                  protein coding  3/3          \n",
       "4               EUR                  protein coding               \n",
       "...             ...                             ...  ...    ...   \n",
       "309004          EUR                  protein coding  2/2          \n",
       "309005          EUR                  protein coding               \n",
       "309006          EUR                  protein coding  6/6          \n",
       "309007          EUR  protein coding CDS not defined               \n",
       "309008          EUR  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \\\n",
       "0                                  S81  tcA/tcG      1                       \n",
       "1                                  nan                                       \n",
       "2                                T162A  Aca/Gca      1                       \n",
       "3                                 S290  tcT/tcC      1                       \n",
       "4                                  nan               1                       \n",
       "...                                ...      ...    ...    ...          ...   \n",
       "309004                             nan              -1                       \n",
       "309005                             nan              -1                       \n",
       "309006                             nan              -1                       \n",
       "309007                             nan              -1                       \n",
       "309008                             nan              -1                       \n",
       "\n",
       "       Gene Match  \n",
       "0              No  \n",
       "1              No  \n",
       "2              No  \n",
       "3              No  \n",
       "4              No  \n",
       "...           ...  \n",
       "309004         No  \n",
       "309005         No  \n",
       "309006         No  \n",
       "309007         No  \n",
       "309008         No  \n",
       "\n",
       "[309009 rows x 70 columns]"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_2, df_gene, on= ['Matched_Gene'], how = 'left', sort = False)\n",
    "merged_2['Gene Match'] = merged_2['Gene Match'].fillna('No')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "b6e1684a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No                      307043\n",
       "Type_2, Mody               797\n",
       "Type_1                     418\n",
       "Type_2                     296\n",
       "Mody                       199\n",
       "Type_1, Type_2, Mody       191\n",
       "Type_1, Mody                65\n",
       "Name: Gene Match, dtype: int64"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene Match'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e0426ae",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "50a454a3",
   "metadata": {},
   "source": [
    "# single gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "623802ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333680</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333681</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333682</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333683</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333684</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "4         SAMD11     rs6605066  chr1    942335   C   G    Homozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "333680                       .  chrY  13297744   T   C    Homozygous   \n",
       "333681                       .  chrY  13297744   T   C    Homozygous   \n",
       "333682                       .  chrY  13297744   T   C    Homozygous   \n",
       "333683                       .  chrY  13297744   T   C    Homozygous   \n",
       "333684                       .  chrY  56961295   G   T    Homozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1           5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "4                intron variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "333680         missense variant              7/10  MODERATE           5.0   \n",
       "333681         missense variant              7/10  MODERATE           5.0   \n",
       "333682         missense variant              7/10  MODERATE           5.0   \n",
       "333683         missense variant              7/10  MODERATE           5.0   \n",
       "333684       intergenic variant              1/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                                                             \n",
       "1                                                         ENST00000641515.2   \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4                                                         ENST00000341065.8   \n",
       "...              ...      ...                ...     ...                ...   \n",
       "333680                                                    ENST00000682216.1   \n",
       "333681                                                    ENST00000682761.1   \n",
       "333682                                                    ENST00000682913.1   \n",
       "333683                                                    ENST00000684226.1   \n",
       "333684                                                                        \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ SDP  \\\n",
       "0                    None                                  None  0/1   32  43   \n",
       "1                c.-36C>T                                  None  0/1   32  43   \n",
       "2                    None                                  None  0/1   32  43   \n",
       "3                    None                                  None  1/1  224  39   \n",
       "4             c.709-75C>G                                  None  1/1  224  39   \n",
       "...                   ...                ...                ...  ...  ...  ..   \n",
       "333680          c.3682A>G  ENSP00000507656.1       p.Ile1228Val  1/1  161  31   \n",
       "333681          c.1708A>G  ENSP00000506900.1        p.Ile570Val  1/1  161  31   \n",
       "333682          c.3817A>G  ENSP00000507736.1       p.Ile1273Val  1/1  161  31   \n",
       "333683          c.2809A>G  ENSP00000507919.1        p.Ile937Val  1/1  161  31   \n",
       "333684               None                                  None  1/1   46   9   \n",
       "\n",
       "        DP  RD  AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       43  33  10  23.26%   5.4148E-4  26   7   9   1   \n",
       "1       43  33  10  23.26%   5.4148E-4  26   7   9   1   \n",
       "2       43  33  10  23.26%   5.4148E-4  26   7   9   1   \n",
       "3       39   0  39    100%  3.6742E-23   0   0  30   9   \n",
       "4       39   0  39    100%  3.6742E-23   0   0  30   9   \n",
       "...     ..  ..  ..     ...         ...  ..  ..  ..  ..   \n",
       "333680  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333681  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333682  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333683  31   1  30  96.77%  6.8754E-17   1   0  25   5   \n",
       "333684   9   0   9    100%   2.0568E-5   0   0   6   3   \n",
       "\n",
       "                                    SIFT       PolyPhen      AF  AFR_AF  \\\n",
       "0                                                                         \n",
       "1                                                                         \n",
       "2                                                                         \n",
       "3                                                        0.8880  0.7935   \n",
       "4                                                        0.8880  0.7935   \n",
       "...                                  ...            ...     ...     ...   \n",
       "333680     deleterious low confidence(0)  benign(0.067)                   \n",
       "333681     deleterious low confidence(0)  benign(0.044)                   \n",
       "333682  deleterious low confidence(0.02)  benign(0.124)                   \n",
       "333683     deleterious low confidence(0)  benign(0.031)                   \n",
       "333684                                                                    \n",
       "\n",
       "        AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                  \n",
       "1                                                                  \n",
       "2                                                                  \n",
       "3       0.9092  0.9256  0.9443  0.9039                             \n",
       "4       0.9092  0.9256  0.9443  0.9039                             \n",
       "...        ...     ...     ...     ...        ...            ...   \n",
       "333680                                                             \n",
       "333681                                                             \n",
       "333682                                                             \n",
       "333683                                                             \n",
       "333684                                                             \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03299        0.02406   \n",
       "1                                                      0.03299        0.02406   \n",
       "2                                                      0.03299        0.02406   \n",
       "3                                                       0.8921         0.7967   \n",
       "4                                                       0.8921         0.7967   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0                   0        0.01429            0.5              0   \n",
       "1                   0        0.01429            0.5              0   \n",
       "2                   0        0.01429            0.5              0   \n",
       "3              0.8813          0.913         0.9116         0.9261   \n",
       "4              0.8813          0.913         0.9116         0.9261   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0              0         0.1071              0   \n",
       "1                   0              0         0.1071              0   \n",
       "2                   0              0         0.1071              0   \n",
       "3              0.9422         0.9114          0.933          0.902   \n",
       "4              0.9422         0.9114          0.933          0.902   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                   0     0.5  gnomADg ASJ   \n",
       "1                   0     0.5  gnomADg ASJ   \n",
       "2                   0     0.5  gnomADg ASJ   \n",
       "3              0.9047  0.9443          EUR   \n",
       "4              0.9047  0.9443          EUR   \n",
       "...               ...     ...          ...   \n",
       "333680                                       \n",
       "333681                                       \n",
       "333682                                       \n",
       "333683                                       \n",
       "333684                                       \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding                 \n",
       "4                           protein coding          7/11   \n",
       "...                                    ...    ...    ...   \n",
       "333680                      non stop decay  25/28          \n",
       "333681                      non stop decay  10/12          \n",
       "333682                      protein coding  26/29          \n",
       "333683                      non stop decay  16/19          \n",
       "333684                                                     \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Gene_Match  \n",
       "0                                  nan               1                No  \n",
       "1                                  nan               1                No  \n",
       "2                                  nan               1                No  \n",
       "3                                  nan              -1                No  \n",
       "4                                  nan               1                No  \n",
       "...                                ...      ...    ...    ...        ...  \n",
       "333680                          I1228V  Atc/Gtc     -1                No  \n",
       "333681                           I570V  Atc/Gtc     -1                No  \n",
       "333682                          I1273V  Atc/Gtc     -1                No  \n",
       "333683                           I937V  Atc/Gtc     -1                No  \n",
       "333684                             nan                                No  \n",
       "\n",
       "[333685 rows x 69 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in merged_2['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            merged_2.loc[merged_2['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "2b57ffa1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     329665\n",
       "Yes      4020\n",
       "Name: Gene_Match, dtype: int64"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "3f1594be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333680</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333681</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333682</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333683</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333684</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match          rsID CHROM       POS REF ALT  \\\n",
       "0                        No  rs1473003496  chr1     65529   C   T   \n",
       "1                        No  rs1473003496  chr1     65529   C   T   \n",
       "2                        No  rs1473003496  chr1     65529   C   T   \n",
       "3         SAMD11         No     rs6605066  chr1    942335   C   G   \n",
       "4         SAMD11         No     rs6605066  chr1    942335   C   G   \n",
       "...          ...        ...           ...   ...       ...  ..  ..   \n",
       "333680                   No             .  chrY  13297744   T   C   \n",
       "333681                   No             .  chrY  13297744   T   C   \n",
       "333682                   No             .  chrY  13297744   T   C   \n",
       "333683                   No             .  chrY  13297744   T   C   \n",
       "333684                   No             .  chrY  56961295   G   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous      5 prime UTR variant              3/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3         Homozygous  downstream gene variant              2/10  MODIFIER   \n",
       "4         Homozygous           intron variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "333680    Homozygous         missense variant              7/10  MODERATE   \n",
       "333681    Homozygous         missense variant              7/10  MODERATE   \n",
       "333682    Homozygous         missense variant              7/10  MODERATE   \n",
       "333683    Homozygous         missense variant              7/10  MODERATE   \n",
       "333684    Homozygous       intergenic variant              1/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                1.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "333680           5.0                                                     \n",
       "333681           5.0                                                     \n",
       "333682           5.0                                                     \n",
       "333683           5.0                                                     \n",
       "333684           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                                       None                      \n",
       "1       ENST00000641515.2           c.-36C>T                      \n",
       "2                                       None                      \n",
       "3                                       None                      \n",
       "4       ENST00000341065.8        c.709-75C>G                      \n",
       "...                   ...                ...                ...   \n",
       "333680  ENST00000682216.1          c.3682A>G  ENSP00000507656.1   \n",
       "333681  ENST00000682761.1          c.1708A>G  ENSP00000506900.1   \n",
       "333682  ENST00000682913.1          c.3817A>G  ENSP00000507736.1   \n",
       "333683  ENST00000684226.1          c.2809A>G  ENSP00000507919.1   \n",
       "333684                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ        PVAL RDF  \\\n",
       "0                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "1                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "2                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "3                    None  1/1  224  39  39   0  39    100%  3.6742E-23   0   \n",
       "4                    None  1/1  224  39  39   0  39    100%  3.6742E-23   0   \n",
       "...                   ...  ...  ...  ..  ..  ..  ..     ...         ...  ..   \n",
       "333680       p.Ile1228Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333681        p.Ile570Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333682       p.Ile1273Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333683        p.Ile937Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333684               None  1/1   46   9   9   0   9    100%   2.0568E-5   0   \n",
       "\n",
       "       RDR ADF ADR                              SIFT       PolyPhen      AF  \\\n",
       "0        7   9   1                                                            \n",
       "1        7   9   1                                                            \n",
       "2        7   9   1                                                            \n",
       "3        0  30   9                                                   0.8880   \n",
       "4        0  30   9                                                   0.8880   \n",
       "...     ..  ..  ..                               ...            ...     ...   \n",
       "333680   0  25   5     deleterious low confidence(0)  benign(0.067)           \n",
       "333681   0  25   5     deleterious low confidence(0)  benign(0.044)           \n",
       "333682   0  25   5  deleterious low confidence(0.02)  benign(0.124)           \n",
       "333683   0  25   5     deleterious low confidence(0)  benign(0.031)           \n",
       "333684   0   6   3                                                            \n",
       "\n",
       "        AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                          \n",
       "1                                                                          \n",
       "2                                                                          \n",
       "3       0.7935  0.9092  0.9256  0.9443  0.9039                             \n",
       "4       0.7935  0.9092  0.9256  0.9443  0.9039                             \n",
       "...        ...     ...     ...     ...     ...        ...            ...   \n",
       "333680                                                                     \n",
       "333681                                                                     \n",
       "333682                                                                     \n",
       "333683                                                                     \n",
       "333684                                                                     \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03299        0.02406   \n",
       "1                                                      0.03299        0.02406   \n",
       "2                                                      0.03299        0.02406   \n",
       "3                                                       0.8921         0.7967   \n",
       "4                                                       0.8921         0.7967   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0                   0        0.01429            0.5              0   \n",
       "1                   0        0.01429            0.5              0   \n",
       "2                   0        0.01429            0.5              0   \n",
       "3              0.8813          0.913         0.9116         0.9261   \n",
       "4              0.8813          0.913         0.9116         0.9261   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0              0         0.1071              0   \n",
       "1                   0              0         0.1071              0   \n",
       "2                   0              0         0.1071              0   \n",
       "3              0.9422         0.9114          0.933          0.902   \n",
       "4              0.9422         0.9114          0.933          0.902   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                   0     0.5  gnomADg ASJ   \n",
       "1                   0     0.5  gnomADg ASJ   \n",
       "2                   0     0.5  gnomADg ASJ   \n",
       "3              0.9047  0.9443          EUR   \n",
       "4              0.9047  0.9443          EUR   \n",
       "...               ...     ...          ...   \n",
       "333680                                       \n",
       "333681                                       \n",
       "333682                                       \n",
       "333683                                       \n",
       "333684                                       \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding                 \n",
       "4                           protein coding          7/11   \n",
       "...                                    ...    ...    ...   \n",
       "333680                      non stop decay  25/28          \n",
       "333681                      non stop decay  10/12          \n",
       "333682                      protein coding  26/29          \n",
       "333683                      non stop decay  16/19          \n",
       "333684                                                     \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  nan               1         \n",
       "1                                  nan               1         \n",
       "2                                  nan               1         \n",
       "3                                  nan              -1         \n",
       "4                                  nan               1         \n",
       "...                                ...      ...    ...    ...  \n",
       "333680                          I1228V  Atc/Gtc     -1         \n",
       "333681                           I570V  Atc/Gtc     -1         \n",
       "333682                          I1273V  Atc/Gtc     -1         \n",
       "333683                           I937V  Atc/Gtc     -1         \n",
       "333684                             nan                         \n",
       "\n",
       "[333685 rows x 69 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "6a3185a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr9</td>\n",
       "      <td>104826974</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr9</td>\n",
       "      <td>104824472</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr9</td>\n",
       "      <td>104858554</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr9</td>\n",
       "      <td>104794495</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1861</th>\n",
       "      <td>chr18</td>\n",
       "      <td>31592980</td>\n",
       "      <td>GTG</td>\n",
       "      <td>ATC</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1862</th>\n",
       "      <td>chr18</td>\n",
       "      <td>31595223</td>\n",
       "      <td>CTT</td>\n",
       "      <td>ATG</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1863</th>\n",
       "      <td>chr18</td>\n",
       "      <td>31593004</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1864</th>\n",
       "      <td>chr18</td>\n",
       "      <td>31598691</td>\n",
       "      <td>GTG</td>\n",
       "      <td>ATA</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1865</th>\n",
       "      <td>chr18</td>\n",
       "      <td>31595256</td>\n",
       "      <td>GTG</td>\n",
       "      <td>ATA</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1597 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS  REF  ALT Literature Covered/Not_Covered\n",
       "0      chr9  104858586    C    T        Yes             Covered\n",
       "1      chr9  104826974    C    T        Yes             Covered\n",
       "2      chr9  104824472    T    C        Yes             Covered\n",
       "3      chr9  104858554    G    A        Yes             Covered\n",
       "4      chr9  104794495    T    G        Yes             Covered\n",
       "...     ...        ...  ...  ...        ...                 ...\n",
       "1861  chr18   31592980  GTG  ATC        Yes             Covered\n",
       "1862  chr18   31595223  CTT  ATG        Yes             Covered\n",
       "1863  chr18   31593004    A    G        Yes             Covered\n",
       "1864  chr18   31598691  GTG  ATA        Yes             Covered\n",
       "1865  chr18   31595256  GTG  ATA        Yes             Covered\n",
       "\n",
       "[1597 rows x 6 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/OneDrive/Desktop/Cardiac_Lit_final_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "x['REF'] = x['chrom'].str.split('-').str[2]\n",
    "x['ALT'] = x['chrom'].str.split('-').str[3]\n",
    "\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS', 'REF', 'ALT']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'REF', 'ALT', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "baf288f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333680</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333681</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333682</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333683</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333684</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match          rsID CHROM       POS REF ALT  \\\n",
       "0                        No  rs1473003496  chr1     65529   C   T   \n",
       "1                        No  rs1473003496  chr1     65529   C   T   \n",
       "2                        No  rs1473003496  chr1     65529   C   T   \n",
       "3         SAMD11         No     rs6605066  chr1    942335   C   G   \n",
       "4         SAMD11         No     rs6605066  chr1    942335   C   G   \n",
       "...          ...        ...           ...   ...       ...  ..  ..   \n",
       "333680                   No             .  chrY  13297744   T   C   \n",
       "333681                   No             .  chrY  13297744   T   C   \n",
       "333682                   No             .  chrY  13297744   T   C   \n",
       "333683                   No             .  chrY  13297744   T   C   \n",
       "333684                   No             .  chrY  56961295   G   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous      5 prime UTR variant              3/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3         Homozygous  downstream gene variant              2/10  MODIFIER   \n",
       "4         Homozygous           intron variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "333680    Homozygous         missense variant              7/10  MODERATE   \n",
       "333681    Homozygous         missense variant              7/10  MODERATE   \n",
       "333682    Homozygous         missense variant              7/10  MODERATE   \n",
       "333683    Homozygous         missense variant              7/10  MODERATE   \n",
       "333684    Homozygous       intergenic variant              1/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                1.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "333680           5.0                                                     \n",
       "333681           5.0                                                     \n",
       "333682           5.0                                                     \n",
       "333683           5.0                                                     \n",
       "333684           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                                       None                      \n",
       "1       ENST00000641515.2           c.-36C>T                      \n",
       "2                                       None                      \n",
       "3                                       None                      \n",
       "4       ENST00000341065.8        c.709-75C>G                      \n",
       "...                   ...                ...                ...   \n",
       "333680  ENST00000682216.1          c.3682A>G  ENSP00000507656.1   \n",
       "333681  ENST00000682761.1          c.1708A>G  ENSP00000506900.1   \n",
       "333682  ENST00000682913.1          c.3817A>G  ENSP00000507736.1   \n",
       "333683  ENST00000684226.1          c.2809A>G  ENSP00000507919.1   \n",
       "333684                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ        PVAL RDF  \\\n",
       "0                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "1                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "2                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "3                    None  1/1  224  39  39   0  39    100%  3.6742E-23   0   \n",
       "4                    None  1/1  224  39  39   0  39    100%  3.6742E-23   0   \n",
       "...                   ...  ...  ...  ..  ..  ..  ..     ...         ...  ..   \n",
       "333680       p.Ile1228Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333681        p.Ile570Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333682       p.Ile1273Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333683        p.Ile937Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333684               None  1/1   46   9   9   0   9    100%   2.0568E-5   0   \n",
       "\n",
       "       RDR ADF ADR                              SIFT       PolyPhen      AF  \\\n",
       "0        7   9   1                                                            \n",
       "1        7   9   1                                                            \n",
       "2        7   9   1                                                            \n",
       "3        0  30   9                                                   0.8880   \n",
       "4        0  30   9                                                   0.8880   \n",
       "...     ..  ..  ..                               ...            ...     ...   \n",
       "333680   0  25   5     deleterious low confidence(0)  benign(0.067)           \n",
       "333681   0  25   5     deleterious low confidence(0)  benign(0.044)           \n",
       "333682   0  25   5  deleterious low confidence(0.02)  benign(0.124)           \n",
       "333683   0  25   5     deleterious low confidence(0)  benign(0.031)           \n",
       "333684   0   6   3                                                            \n",
       "\n",
       "        AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                          \n",
       "1                                                                          \n",
       "2                                                                          \n",
       "3       0.7935  0.9092  0.9256  0.9443  0.9039                             \n",
       "4       0.7935  0.9092  0.9256  0.9443  0.9039                             \n",
       "...        ...     ...     ...     ...     ...        ...            ...   \n",
       "333680                                                                     \n",
       "333681                                                                     \n",
       "333682                                                                     \n",
       "333683                                                                     \n",
       "333684                                                                     \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03299        0.02406   \n",
       "1                                                      0.03299        0.02406   \n",
       "2                                                      0.03299        0.02406   \n",
       "3                                                       0.8921         0.7967   \n",
       "4                                                       0.8921         0.7967   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0                   0        0.01429            0.5              0   \n",
       "1                   0        0.01429            0.5              0   \n",
       "2                   0        0.01429            0.5              0   \n",
       "3              0.8813          0.913         0.9116         0.9261   \n",
       "4              0.8813          0.913         0.9116         0.9261   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0              0         0.1071              0   \n",
       "1                   0              0         0.1071              0   \n",
       "2                   0              0         0.1071              0   \n",
       "3              0.9422         0.9114          0.933          0.902   \n",
       "4              0.9422         0.9114          0.933          0.902   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                   0     0.5  gnomADg ASJ   \n",
       "1                   0     0.5  gnomADg ASJ   \n",
       "2                   0     0.5  gnomADg ASJ   \n",
       "3              0.9047  0.9443          EUR   \n",
       "4              0.9047  0.9443          EUR   \n",
       "...               ...     ...          ...   \n",
       "333680                                       \n",
       "333681                                       \n",
       "333682                                       \n",
       "333683                                       \n",
       "333684                                       \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding                 \n",
       "4                           protein coding          7/11   \n",
       "...                                    ...    ...    ...   \n",
       "333680                      non stop decay  25/28          \n",
       "333681                      non stop decay  10/12          \n",
       "333682                      protein coding  26/29          \n",
       "333683                      non stop decay  16/19          \n",
       "333684                                                     \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Literature  \\\n",
       "0                                  nan               1                No   \n",
       "1                                  nan               1                No   \n",
       "2                                  nan               1                No   \n",
       "3                                  nan              -1                No   \n",
       "4                                  nan               1                No   \n",
       "...                                ...      ...    ...    ...        ...   \n",
       "333680                          I1228V  Atc/Gtc     -1                No   \n",
       "333681                           I570V  Atc/Gtc     -1                No   \n",
       "333682                          I1273V  Atc/Gtc     -1                No   \n",
       "333683                           I937V  Atc/Gtc     -1                No   \n",
       "333684                             nan                                No   \n",
       "\n",
       "       Covered/Not_Covered  \n",
       "0                      NaN  \n",
       "1                      NaN  \n",
       "2                      NaN  \n",
       "3                      NaN  \n",
       "4                      NaN  \n",
       "...                    ...  \n",
       "333680                 NaN  \n",
       "333681                 NaN  \n",
       "333682                 NaN  \n",
       "333683                 NaN  \n",
       "333684                 NaN  \n",
       "\n",
       "[333685 rows x 71 columns]"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "5e0683b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     333637\n",
       "Yes        48\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "d3c7da39",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.4148E-4</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03299</td>\n",
       "      <td>0.02406</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1071</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6605066</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942335</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.709-75C&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.7935</td>\n",
       "      <td>0.9092</td>\n",
       "      <td>0.9256</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>0.9039</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8921</td>\n",
       "      <td>0.7967</td>\n",
       "      <td>0.8813</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.9116</td>\n",
       "      <td>0.9261</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.9047</td>\n",
       "      <td>0.9443</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/11</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333680</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682216.1</td>\n",
       "      <td>c.3682A&gt;G</td>\n",
       "      <td>ENSP00000507656.1</td>\n",
       "      <td>p.Ile1228Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.067)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>25/28</td>\n",
       "      <td></td>\n",
       "      <td>I1228V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333681</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682761.1</td>\n",
       "      <td>c.1708A&gt;G</td>\n",
       "      <td>ENSP00000506900.1</td>\n",
       "      <td>p.Ile570Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.044)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>10/12</td>\n",
       "      <td></td>\n",
       "      <td>I570V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333682</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682913.1</td>\n",
       "      <td>c.3817A&gt;G</td>\n",
       "      <td>ENSP00000507736.1</td>\n",
       "      <td>p.Ile1273Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0.02)</td>\n",
       "      <td>benign(0.124)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td></td>\n",
       "      <td>I1273V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333683</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chrY</td>\n",
       "      <td>13297744</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000684226.1</td>\n",
       "      <td>c.2809A&gt;G</td>\n",
       "      <td>ENSP00000507919.1</td>\n",
       "      <td>p.Ile937Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>161</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>96.77%</td>\n",
       "      <td>6.8754E-17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.031)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non stop decay</td>\n",
       "      <td>16/19</td>\n",
       "      <td></td>\n",
       "      <td>I937V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333684</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.0568E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333685 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match          rsID Literature CHROM       POS REF ALT  \\\n",
       "0                        No  rs1473003496         No  chr1     65529   C   T   \n",
       "1                        No  rs1473003496         No  chr1     65529   C   T   \n",
       "2                        No  rs1473003496         No  chr1     65529   C   T   \n",
       "3         SAMD11         No     rs6605066         No  chr1    942335   C   G   \n",
       "4         SAMD11         No     rs6605066         No  chr1    942335   C   G   \n",
       "...          ...        ...           ...        ...   ...       ...  ..  ..   \n",
       "333680                   No             .         No  chrY  13297744   T   C   \n",
       "333681                   No             .         No  chrY  13297744   T   C   \n",
       "333682                   No             .         No  chrY  13297744   T   C   \n",
       "333683                   No             .         No  chrY  13297744   T   C   \n",
       "333684                   No             .         No  chrY  56961295   G   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous      5 prime UTR variant              3/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3         Homozygous  downstream gene variant              2/10  MODIFIER   \n",
       "4         Homozygous           intron variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "333680    Homozygous         missense variant              7/10  MODERATE   \n",
       "333681    Homozygous         missense variant              7/10  MODERATE   \n",
       "333682    Homozygous         missense variant              7/10  MODERATE   \n",
       "333683    Homozygous         missense variant              7/10  MODERATE   \n",
       "333684    Homozygous       intergenic variant              1/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                1.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "333680           5.0                                                     \n",
       "333681           5.0                                                     \n",
       "333682           5.0                                                     \n",
       "333683           5.0                                                     \n",
       "333684           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                                       None                      \n",
       "1       ENST00000641515.2           c.-36C>T                      \n",
       "2                                       None                      \n",
       "3                                       None                      \n",
       "4       ENST00000341065.8        c.709-75C>G                      \n",
       "...                   ...                ...                ...   \n",
       "333680  ENST00000682216.1          c.3682A>G  ENSP00000507656.1   \n",
       "333681  ENST00000682761.1          c.1708A>G  ENSP00000506900.1   \n",
       "333682  ENST00000682913.1          c.3817A>G  ENSP00000507736.1   \n",
       "333683  ENST00000684226.1          c.2809A>G  ENSP00000507919.1   \n",
       "333684                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ        PVAL RDF  \\\n",
       "0                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "1                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "2                    None  0/1   32  43  43  33  10  23.26%   5.4148E-4  26   \n",
       "3                    None  1/1  224  39  39   0  39    100%  3.6742E-23   0   \n",
       "4                    None  1/1  224  39  39   0  39    100%  3.6742E-23   0   \n",
       "...                   ...  ...  ...  ..  ..  ..  ..     ...         ...  ..   \n",
       "333680       p.Ile1228Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333681        p.Ile570Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333682       p.Ile1273Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333683        p.Ile937Val  1/1  161  31  31   1  30  96.77%  6.8754E-17   1   \n",
       "333684               None  1/1   46   9   9   0   9    100%   2.0568E-5   0   \n",
       "\n",
       "       RDR ADF ADR                              SIFT       PolyPhen      AF  \\\n",
       "0        7   9   1                                                            \n",
       "1        7   9   1                                                            \n",
       "2        7   9   1                                                            \n",
       "3        0  30   9                                                   0.8880   \n",
       "4        0  30   9                                                   0.8880   \n",
       "...     ..  ..  ..                               ...            ...     ...   \n",
       "333680   0  25   5     deleterious low confidence(0)  benign(0.067)           \n",
       "333681   0  25   5     deleterious low confidence(0)  benign(0.044)           \n",
       "333682   0  25   5  deleterious low confidence(0.02)  benign(0.124)           \n",
       "333683   0  25   5     deleterious low confidence(0)  benign(0.031)           \n",
       "333684   0   6   3                                                            \n",
       "\n",
       "        AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                          \n",
       "1                                                                          \n",
       "2                                                                          \n",
       "3       0.7935  0.9092  0.9256  0.9443  0.9039                             \n",
       "4       0.7935  0.9092  0.9256  0.9443  0.9039                             \n",
       "...        ...     ...     ...     ...     ...        ...            ...   \n",
       "333680                                                                     \n",
       "333681                                                                     \n",
       "333682                                                                     \n",
       "333683                                                                     \n",
       "333684                                                                     \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03299        0.02406   \n",
       "1                                                      0.03299        0.02406   \n",
       "2                                                      0.03299        0.02406   \n",
       "3                                                       0.8921         0.7967   \n",
       "4                                                       0.8921         0.7967   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "333680                                                                          \n",
       "333681                                                                          \n",
       "333682                                                                          \n",
       "333683                                                                          \n",
       "333684                                                                          \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0                   0        0.01429            0.5              0   \n",
       "1                   0        0.01429            0.5              0   \n",
       "2                   0        0.01429            0.5              0   \n",
       "3              0.8813          0.913         0.9116         0.9261   \n",
       "4              0.8813          0.913         0.9116         0.9261   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0              0         0.1071              0   \n",
       "1                   0              0         0.1071              0   \n",
       "2                   0              0         0.1071              0   \n",
       "3              0.9422         0.9114          0.933          0.902   \n",
       "4              0.9422         0.9114          0.933          0.902   \n",
       "...               ...            ...            ...            ...   \n",
       "333680                                                               \n",
       "333681                                                               \n",
       "333682                                                               \n",
       "333683                                                               \n",
       "333684                                                               \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0                   0     0.5  gnomADg ASJ   \n",
       "1                   0     0.5  gnomADg ASJ   \n",
       "2                   0     0.5  gnomADg ASJ   \n",
       "3              0.9047  0.9443          EUR   \n",
       "4              0.9047  0.9443          EUR   \n",
       "...               ...     ...          ...   \n",
       "333680                                       \n",
       "333681                                       \n",
       "333682                                       \n",
       "333683                                       \n",
       "333684                                       \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding                 \n",
       "4                           protein coding          7/11   \n",
       "...                                    ...    ...    ...   \n",
       "333680                      non stop decay  25/28          \n",
       "333681                      non stop decay  10/12          \n",
       "333682                      protein coding  26/29          \n",
       "333683                      non stop decay  16/19          \n",
       "333684                                                     \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  nan               1         \n",
       "1                                  nan               1         \n",
       "2                                  nan               1         \n",
       "3                                  nan              -1         \n",
       "4                                  nan               1         \n",
       "...                                ...      ...    ...    ...  \n",
       "333680                          I1228V  Atc/Gtc     -1         \n",
       "333681                           I570V  Atc/Gtc     -1         \n",
       "333682                          I1273V  Atc/Gtc     -1         \n",
       "333683                           I937V  Atc/Gtc     -1         \n",
       "333684                             nan                         \n",
       "\n",
       "[333685 rows x 70 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = merged_3[['Gene Name', 'Gene_Match', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "aae64773",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL125_final_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "469cbe27",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e44ed87",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cb7b2c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "092d77e2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1d06842",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31510422",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83de0cd5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "124f6ed7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "282a9859",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "469f702c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7dab2584",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a8e0d6e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21183687",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88569366",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07954f0f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c020c8ca",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c3d3269",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7be73536",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a57cff7a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "594615e1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a07f424e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5bdc8b9f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5eab78c7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54f9df02",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c67fbc67",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "828b45a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e672c85",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f8ae486",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1fb7604",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd104981",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c07d0f08",
   "metadata": {},
   "source": [
    "# Only condition sheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "15fc7774",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NPPA,NPPA-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs5063</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11847591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Atrial fibrillation, familial, 6</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1169995</td>\n",
       "      <td>ENST00000376480.7</td>\n",
       "      <td>c.94G&gt;A</td>\n",
       "      <td>ENSP00000365663.3</td>\n",
       "      <td>p.Val32Met</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>36%</td>\n",
       "      <td>8.1542E-4</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>tolerated(0.22)</td>\n",
       "      <td>benign(0.057)</td>\n",
       "      <td>0.0735</td>\n",
       "      <td>0.0613</td>\n",
       "      <td>0.0476</td>\n",
       "      <td>0.1151</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>0.1053</td>\n",
       "      <td>0.055860</td>\n",
       "      <td>0.06115</td>\n",
       "      <td>0.04429</td>\n",
       "      <td>0.02202</td>\n",
       "      <td>0.115</td>\n",
       "      <td>0.02818</td>\n",
       "      <td>0.04568</td>\n",
       "      <td>0.05033</td>\n",
       "      <td>0.100200</td>\n",
       "      <td>0.05381</td>\n",
       "      <td>0.06202</td>\n",
       "      <td>0.01535</td>\n",
       "      <td>0.0484</td>\n",
       "      <td>0.02046</td>\n",
       "      <td>0.1154</td>\n",
       "      <td>0.02903</td>\n",
       "      <td>0.01582</td>\n",
       "      <td>0.04731</td>\n",
       "      <td>0.05306</td>\n",
       "      <td>0.1146</td>\n",
       "      <td>0.1154</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/3</td>\n",
       "      <td></td>\n",
       "      <td>V32M</td>\n",
       "      <td>Gtg/Atg</td>\n",
       "      <td>-1</td>\n",
       "      <td>18513389,19131662,19330901,19559392,20577119,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs34967813</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>237678090</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Cardiomyopathy,Catech...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>36750</td>\n",
       "      <td>ENST00000366574.7</td>\n",
       "      <td>c.8873A&gt;G</td>\n",
       "      <td>ENSP00000355533.2</td>\n",
       "      <td>p.Gln2958Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>76</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>24</td>\n",
       "      <td>21</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>2.2206E-8</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td>benign(0.033)</td>\n",
       "      <td>0.0992</td>\n",
       "      <td>0.0113</td>\n",
       "      <td>0.1988</td>\n",
       "      <td>0</td>\n",
       "      <td>0.2942</td>\n",
       "      <td>0.0491</td>\n",
       "      <td>0.214100</td>\n",
       "      <td>0.04827</td>\n",
       "      <td>0.198</td>\n",
       "      <td>0.2216</td>\n",
       "      <td>0.0007356</td>\n",
       "      <td>0.2341</td>\n",
       "      <td>0.3069</td>\n",
       "      <td>0.2428</td>\n",
       "      <td>0.077920</td>\n",
       "      <td>0.2014</td>\n",
       "      <td>0.05371</td>\n",
       "      <td>0.1327</td>\n",
       "      <td>0.2132</td>\n",
       "      <td>0.212</td>\n",
       "      <td>0.001731</td>\n",
       "      <td>0.2301</td>\n",
       "      <td>0.1804</td>\n",
       "      <td>0.3095</td>\n",
       "      <td>0.1864</td>\n",
       "      <td>0.07601</td>\n",
       "      <td>0.3095</td>\n",
       "      <td>gnomADg NFE</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>61/105</td>\n",
       "      <td></td>\n",
       "      <td>Q2958R</td>\n",
       "      <td>cAa/cGa</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24033266,23861362,23757202,11157710,1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MYPN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs10823148</td>\n",
       "      <td>No</td>\n",
       "      <td>chr10</td>\n",
       "      <td>68166577</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>MYPN-related myopathy,Cardiovascular phenotype...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>31804</td>\n",
       "      <td>ENST00000354393.7</td>\n",
       "      <td>c.1059C&gt;G</td>\n",
       "      <td>ENSP00000346369.2</td>\n",
       "      <td>p.Phe353Leu</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.925E-29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>19</td>\n",
       "      <td>deleterious low confidence(0.04)</td>\n",
       "      <td></td>\n",
       "      <td>0.3165</td>\n",
       "      <td>0.1104</td>\n",
       "      <td>0.4049</td>\n",
       "      <td>0.2163</td>\n",
       "      <td>0.5099</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.439800</td>\n",
       "      <td>0.1566</td>\n",
       "      <td>0.431</td>\n",
       "      <td>0.3278</td>\n",
       "      <td>0.1951</td>\n",
       "      <td>0.544</td>\n",
       "      <td>0.5135</td>\n",
       "      <td>0.4431</td>\n",
       "      <td>0.436200</td>\n",
       "      <td>0.396</td>\n",
       "      <td>0.1671</td>\n",
       "      <td>0.5318</td>\n",
       "      <td>0.4337</td>\n",
       "      <td>0.3138</td>\n",
       "      <td>0.2117</td>\n",
       "      <td>0.5462</td>\n",
       "      <td>0.2595</td>\n",
       "      <td>0.5179</td>\n",
       "      <td>0.3914</td>\n",
       "      <td>0.4359</td>\n",
       "      <td>0.5462</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/20</td>\n",
       "      <td></td>\n",
       "      <td>F353L</td>\n",
       "      <td>ttC/ttG</td>\n",
       "      <td>1</td>\n",
       "      <td>24033266,23861362,18006477,22286171,22892539,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MYPN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs10997975</td>\n",
       "      <td>No</td>\n",
       "      <td>chr10</td>\n",
       "      <td>68174164</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>MYPN-related myopathy,Cardiovascular phenotype...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>31797</td>\n",
       "      <td>ENST00000354393.7</td>\n",
       "      <td>c.1247G&gt;A</td>\n",
       "      <td>ENSP00000346369.2</td>\n",
       "      <td>p.Ser416Asn</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.9137E-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td></td>\n",
       "      <td>0.3359</td>\n",
       "      <td>0.1762</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.2649</td>\n",
       "      <td>0.4851</td>\n",
       "      <td>0.4233</td>\n",
       "      <td>0.430900</td>\n",
       "      <td>0.2045</td>\n",
       "      <td>0.4192</td>\n",
       "      <td>0.329</td>\n",
       "      <td>0.247</td>\n",
       "      <td>0.5298</td>\n",
       "      <td>0.4872</td>\n",
       "      <td>0.4319</td>\n",
       "      <td>0.428800</td>\n",
       "      <td>0.3943</td>\n",
       "      <td>0.2123</td>\n",
       "      <td>0.4232</td>\n",
       "      <td>0.419</td>\n",
       "      <td>0.3185</td>\n",
       "      <td>0.2627</td>\n",
       "      <td>0.5309</td>\n",
       "      <td>0.2532</td>\n",
       "      <td>0.4898</td>\n",
       "      <td>0.392</td>\n",
       "      <td>0.4301</td>\n",
       "      <td>0.5309</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>11/20</td>\n",
       "      <td></td>\n",
       "      <td>S416N</td>\n",
       "      <td>aGc/aAc</td>\n",
       "      <td>1</td>\n",
       "      <td>24033266,23861362,23164068,20395239,18006477,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MYPN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs7916821</td>\n",
       "      <td>No</td>\n",
       "      <td>chr10</td>\n",
       "      <td>68174212</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,MYPN-related myopathy...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>31798</td>\n",
       "      <td>ENST00000354393.7</td>\n",
       "      <td>c.1295G&gt;A</td>\n",
       "      <td>ENSP00000346369.2</td>\n",
       "      <td>p.Ser432Asn</td>\n",
       "      <td>1/1</td>\n",
       "      <td>236</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>41</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.3541E-24</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>6</td>\n",
       "      <td>tolerated(0.2)</td>\n",
       "      <td></td>\n",
       "      <td>0.3251</td>\n",
       "      <td>0.1762</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.2163</td>\n",
       "      <td>0.4851</td>\n",
       "      <td>0.4182</td>\n",
       "      <td>0.426200</td>\n",
       "      <td>0.2041</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>0.329</td>\n",
       "      <td>0.1982</td>\n",
       "      <td>0.5277</td>\n",
       "      <td>0.4867</td>\n",
       "      <td>0.4275</td>\n",
       "      <td>0.426900</td>\n",
       "      <td>0.3915</td>\n",
       "      <td>0.2115</td>\n",
       "      <td>0.4242</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>0.3179</td>\n",
       "      <td>0.215</td>\n",
       "      <td>0.5281</td>\n",
       "      <td>0.2532</td>\n",
       "      <td>0.4892</td>\n",
       "      <td>0.3901</td>\n",
       "      <td>0.424</td>\n",
       "      <td>0.5281</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>11/20</td>\n",
       "      <td></td>\n",
       "      <td>S432N</td>\n",
       "      <td>aGt/aAt</td>\n",
       "      <td>1</td>\n",
       "      <td>24033266,23861362,16385451,28445147,18006477,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MYPN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs3814182</td>\n",
       "      <td>No</td>\n",
       "      <td>chr10</td>\n",
       "      <td>68174501</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>31808</td>\n",
       "      <td>ENST00000354393.7</td>\n",
       "      <td>c.1584C&gt;G</td>\n",
       "      <td>ENSP00000346369.2</td>\n",
       "      <td>p.Ser528Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0149E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>25</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td></td>\n",
       "      <td>0.4714</td>\n",
       "      <td>0.447</td>\n",
       "      <td>0.4597</td>\n",
       "      <td>0.3512</td>\n",
       "      <td>0.5666</td>\n",
       "      <td>0.5389</td>\n",
       "      <td>0.518700</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.4634</td>\n",
       "      <td>0.4303</td>\n",
       "      <td>0.3666</td>\n",
       "      <td>0.6307</td>\n",
       "      <td>0.5571</td>\n",
       "      <td>0.5117</td>\n",
       "      <td>0.512000</td>\n",
       "      <td>0.5179</td>\n",
       "      <td>0.4582</td>\n",
       "      <td>0.4112</td>\n",
       "      <td>0.5022</td>\n",
       "      <td>0.4231</td>\n",
       "      <td>0.3678</td>\n",
       "      <td>0.634</td>\n",
       "      <td>0.3829</td>\n",
       "      <td>0.5583</td>\n",
       "      <td>0.5226</td>\n",
       "      <td>0.5118</td>\n",
       "      <td>0.634</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>11/20</td>\n",
       "      <td></td>\n",
       "      <td>S528R</td>\n",
       "      <td>agC/agG</td>\n",
       "      <td>1</td>\n",
       "      <td>24033266,23861362,18006477,22286171,22892539,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MYPN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs7079481</td>\n",
       "      <td>No</td>\n",
       "      <td>chr10</td>\n",
       "      <td>68199485</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>MYPN-related myopathy,Cardiovascular phenotype...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>31800</td>\n",
       "      <td>ENST00000354393.7</td>\n",
       "      <td>c.2578C&gt;A</td>\n",
       "      <td>ENSP00000346369.2</td>\n",
       "      <td>p.Pro860Thr</td>\n",
       "      <td>1/1</td>\n",
       "      <td>194</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.5146E-20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td></td>\n",
       "      <td>0.3403</td>\n",
       "      <td>0.2163</td>\n",
       "      <td>0.402</td>\n",
       "      <td>0.2391</td>\n",
       "      <td>0.4841</td>\n",
       "      <td>0.4202</td>\n",
       "      <td>0.432000</td>\n",
       "      <td>0.2546</td>\n",
       "      <td>0.4187</td>\n",
       "      <td>0.3334</td>\n",
       "      <td>0.2185</td>\n",
       "      <td>0.5268</td>\n",
       "      <td>0.4873</td>\n",
       "      <td>0.4257</td>\n",
       "      <td>0.430500</td>\n",
       "      <td>0.4062</td>\n",
       "      <td>0.2613</td>\n",
       "      <td>0.4211</td>\n",
       "      <td>0.4166</td>\n",
       "      <td>0.325</td>\n",
       "      <td>0.2359</td>\n",
       "      <td>0.5269</td>\n",
       "      <td>0.2484</td>\n",
       "      <td>0.4896</td>\n",
       "      <td>0.3922</td>\n",
       "      <td>0.4287</td>\n",
       "      <td>0.5269</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>17/20</td>\n",
       "      <td></td>\n",
       "      <td>P860T</td>\n",
       "      <td>Cca/Aca</td>\n",
       "      <td>1</td>\n",
       "      <td>24033266,23861362,18006477,22286171,22892539,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PKP2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1046116</td>\n",
       "      <td>No</td>\n",
       "      <td>chr12</td>\n",
       "      <td>32869000</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Arrhythmogenic right ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>45006</td>\n",
       "      <td>ENST00000070846.11</td>\n",
       "      <td>c.1097T&gt;C</td>\n",
       "      <td>ENSP00000070846.6</td>\n",
       "      <td>p.Leu366Pro</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.4343E-27</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated(0.89)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.1514</td>\n",
       "      <td>0.2103</td>\n",
       "      <td>0.1614</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.2157</td>\n",
       "      <td>0.1227</td>\n",
       "      <td>0.180000</td>\n",
       "      <td>0.1991</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1672</td>\n",
       "      <td>0.02936</td>\n",
       "      <td>0.1942</td>\n",
       "      <td>0.2201</td>\n",
       "      <td>0.1914</td>\n",
       "      <td>0.163500</td>\n",
       "      <td>0.1979</td>\n",
       "      <td>0.1951</td>\n",
       "      <td>0.1505</td>\n",
       "      <td>0.1663</td>\n",
       "      <td>0.1802</td>\n",
       "      <td>0.03272</td>\n",
       "      <td>0.1984</td>\n",
       "      <td>0.2468</td>\n",
       "      <td>0.2238</td>\n",
       "      <td>0.1937</td>\n",
       "      <td>0.1545</td>\n",
       "      <td>0.2468</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/14</td>\n",
       "      <td></td>\n",
       "      <td>L366P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,17556197,17521752,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MYH6</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs573489857</td>\n",
       "      <td>No</td>\n",
       "      <td>chr14</td>\n",
       "      <td>23407154</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Cardiomyopathy,Hypert...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>164260</td>\n",
       "      <td>ENST00000405093.9</td>\n",
       "      <td>c.70C&gt;A</td>\n",
       "      <td>ENSP00000386041.3</td>\n",
       "      <td>p.Leu24Ile</td>\n",
       "      <td>0/1</td>\n",
       "      <td>98</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>30</td>\n",
       "      <td>27</td>\n",
       "      <td>47.37%</td>\n",
       "      <td>1.2661E-10</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>deleterious low confidence(0.04)</td>\n",
       "      <td>benign(0.022)</td>\n",
       "      <td>0.0012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0061</td>\n",
       "      <td>0.000632</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8.791e-06</td>\n",
       "      <td>0.0001629</td>\n",
       "      <td>0.005128</td>\n",
       "      <td>0.0001445</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004547</td>\n",
       "      <td>0.0061</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/39</td>\n",
       "      <td></td>\n",
       "      <td>L24I</td>\n",
       "      <td>Cta/Ata</td>\n",
       "      <td>-1</td>\n",
       "      <td>24033266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>DSG2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2230234</td>\n",
       "      <td>No</td>\n",
       "      <td>chr18</td>\n",
       "      <td>31524751</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Cardiomyopathy,Arrhyt...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>44331</td>\n",
       "      <td>ENST00000261590.13</td>\n",
       "      <td>c.877A&gt;G</td>\n",
       "      <td>ENSP00000261590.8</td>\n",
       "      <td>p.Ile293Val</td>\n",
       "      <td>0/1</td>\n",
       "      <td>77</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "      <td>22</td>\n",
       "      <td>21</td>\n",
       "      <td>48.84%</td>\n",
       "      <td>1.8298E-8</td>\n",
       "      <td>17</td>\n",
       "      <td>5</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td></td>\n",
       "      <td>0.0323</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.036</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0368</td>\n",
       "      <td>0.068660</td>\n",
       "      <td>0.01292</td>\n",
       "      <td>0.03952</td>\n",
       "      <td>0.07701</td>\n",
       "      <td>0.0002781</td>\n",
       "      <td>0.1453</td>\n",
       "      <td>0.08646</td>\n",
       "      <td>0.0838</td>\n",
       "      <td>0.044350</td>\n",
       "      <td>0.0646</td>\n",
       "      <td>0.01532</td>\n",
       "      <td>0.2061</td>\n",
       "      <td>0.04894</td>\n",
       "      <td>0.07896</td>\n",
       "      <td>0.0009608</td>\n",
       "      <td>0.1516</td>\n",
       "      <td>0.08228</td>\n",
       "      <td>0.08845</td>\n",
       "      <td>0.06274</td>\n",
       "      <td>0.04122</td>\n",
       "      <td>0.2061</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/15</td>\n",
       "      <td></td>\n",
       "      <td>I293V</td>\n",
       "      <td>Ata/Gta</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24033266,23861362,20829228,18678517,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LDLR</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs576530433</td>\n",
       "      <td>No</td>\n",
       "      <td>chr19</td>\n",
       "      <td>11090668</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000252444.9</td>\n",
       "      <td>c.71G&gt;A</td>\n",
       "      <td>ENSP00000252444.5</td>\n",
       "      <td>p.Arg24His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>5.5097E-10</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.099)</td>\n",
       "      <td>0.0018</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0092</td>\n",
       "      <td>0.000488</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001011</td>\n",
       "      <td>0.002595</td>\n",
       "      <td>0.0001183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00352</td>\n",
       "      <td>0.0092</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/18</td>\n",
       "      <td></td>\n",
       "      <td>R24H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>APOB</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1042031</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>21002881</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Familial hypobetalipo...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>128417</td>\n",
       "      <td>ENST00000233242.5</td>\n",
       "      <td>c.12541G&gt;A</td>\n",
       "      <td>ENSP00000233242.1</td>\n",
       "      <td>p.Glu4181Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>28</td>\n",
       "      <td>19</td>\n",
       "      <td>40.43%</td>\n",
       "      <td>1.9354E-7</td>\n",
       "      <td>15</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>tolerated(0.22)</td>\n",
       "      <td>benign(0.179)</td>\n",
       "      <td>0.1278</td>\n",
       "      <td>0.1611</td>\n",
       "      <td>0.1196</td>\n",
       "      <td>0.0446</td>\n",
       "      <td>0.2127</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>0.152900</td>\n",
       "      <td>0.154</td>\n",
       "      <td>0.122</td>\n",
       "      <td>0.1499</td>\n",
       "      <td>0.04842</td>\n",
       "      <td>0.2115</td>\n",
       "      <td>0.1816</td>\n",
       "      <td>0.1621</td>\n",
       "      <td>0.101700</td>\n",
       "      <td>0.1648</td>\n",
       "      <td>0.1526</td>\n",
       "      <td>0.1316</td>\n",
       "      <td>0.1452</td>\n",
       "      <td>0.1489</td>\n",
       "      <td>0.04582</td>\n",
       "      <td>0.2218</td>\n",
       "      <td>0.1242</td>\n",
       "      <td>0.1829</td>\n",
       "      <td>0.1793</td>\n",
       "      <td>0.09298</td>\n",
       "      <td>0.2218</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>29/29</td>\n",
       "      <td></td>\n",
       "      <td>E4181K</td>\n",
       "      <td>Gaa/Aaa</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,27589735,28659664,16642433,22328972,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>APOB</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs575505383</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>21019079</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Hypercholesterolemia, familial, 1,Familial hyp...</td>\n",
       "      <td>uncertain significance,benign,likely benign</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>334160</td>\n",
       "      <td>ENST00000233242.5</td>\n",
       "      <td>c.3034G&gt;A</td>\n",
       "      <td>ENSP00000233242.1</td>\n",
       "      <td>p.Glu1012Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>90</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>22</td>\n",
       "      <td>24</td>\n",
       "      <td>52.17%</td>\n",
       "      <td>9.761E-10</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(0.63)</td>\n",
       "      <td>benign(0.196)</td>\n",
       "      <td>0.0016</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0082</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.519e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.005520</td>\n",
       "      <td>0.0002433</td>\n",
       "      <td>4.831e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.94e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.006849</td>\n",
       "      <td>0.0082</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>20/29</td>\n",
       "      <td></td>\n",
       "      <td>E1012K</td>\n",
       "      <td>Gag/Aag</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs3829747</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178532834</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47663</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.77162G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Arg25721His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>60</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>23</td>\n",
       "      <td>17</td>\n",
       "      <td>42.5%</td>\n",
       "      <td>8.743E-7</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.0666</td>\n",
       "      <td>0.1484</td>\n",
       "      <td>0.4484</td>\n",
       "      <td>0.1392</td>\n",
       "      <td>0.2679</td>\n",
       "      <td>0.175600</td>\n",
       "      <td>0.07002</td>\n",
       "      <td>0.145</td>\n",
       "      <td>0.1704</td>\n",
       "      <td>0.4299</td>\n",
       "      <td>0.1525</td>\n",
       "      <td>0.1437</td>\n",
       "      <td>0.1577</td>\n",
       "      <td>0.253000</td>\n",
       "      <td>0.1413</td>\n",
       "      <td>0.07479</td>\n",
       "      <td>0.4736</td>\n",
       "      <td>0.1279</td>\n",
       "      <td>0.1702</td>\n",
       "      <td>0.4391</td>\n",
       "      <td>0.1628</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>0.1454</td>\n",
       "      <td>0.1264</td>\n",
       "      <td>0.2562</td>\n",
       "      <td>0.4736</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>186/191</td>\n",
       "      <td></td>\n",
       "      <td>R25721H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,23757202,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs4894028</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178539023</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47610</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.72293G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Arg24098His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>54</td>\n",
       "      <td>33</td>\n",
       "      <td>32</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "      <td>46.88%</td>\n",
       "      <td>3.5464E-6</td>\n",
       "      <td>13</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0769</td>\n",
       "      <td>0.0113</td>\n",
       "      <td>0.1484</td>\n",
       "      <td>0.0248</td>\n",
       "      <td>0.0477</td>\n",
       "      <td>0.1984</td>\n",
       "      <td>0.074200</td>\n",
       "      <td>0.01188</td>\n",
       "      <td>0.1943</td>\n",
       "      <td>0.02975</td>\n",
       "      <td>0.02762</td>\n",
       "      <td>0.05772</td>\n",
       "      <td>0.03227</td>\n",
       "      <td>0.05908</td>\n",
       "      <td>0.181000</td>\n",
       "      <td>0.04453</td>\n",
       "      <td>0.01337</td>\n",
       "      <td>0.04934</td>\n",
       "      <td>0.1514</td>\n",
       "      <td>0.02968</td>\n",
       "      <td>0.02372</td>\n",
       "      <td>0.05839</td>\n",
       "      <td>0.06646</td>\n",
       "      <td>0.03035</td>\n",
       "      <td>0.04019</td>\n",
       "      <td>0.1765</td>\n",
       "      <td>0.1984</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>181/191</td>\n",
       "      <td></td>\n",
       "      <td>R24098H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs3731749</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178541464</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47586</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.70994G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Arg23665His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>45</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>13</td>\n",
       "      <td>39.39%</td>\n",
       "      <td>2.8029E-5</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.0666</td>\n",
       "      <td>0.1484</td>\n",
       "      <td>0.4514</td>\n",
       "      <td>0.1382</td>\n",
       "      <td>0.2689</td>\n",
       "      <td>0.175400</td>\n",
       "      <td>0.07031</td>\n",
       "      <td>0.1448</td>\n",
       "      <td>0.1706</td>\n",
       "      <td>0.4294</td>\n",
       "      <td>0.1516</td>\n",
       "      <td>0.1436</td>\n",
       "      <td>0.1587</td>\n",
       "      <td>0.253200</td>\n",
       "      <td>0.1416</td>\n",
       "      <td>0.07537</td>\n",
       "      <td>0.4726</td>\n",
       "      <td>0.128</td>\n",
       "      <td>0.1703</td>\n",
       "      <td>0.4404</td>\n",
       "      <td>0.1627</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>0.1455</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0.2568</td>\n",
       "      <td>0.4726</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>178/191</td>\n",
       "      <td></td>\n",
       "      <td>R23665H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs62621236</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178543986</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47566</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.69539T&gt;C</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Ile23180Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>110</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>18</td>\n",
       "      <td>28</td>\n",
       "      <td>60.87%</td>\n",
       "      <td>8.7676E-12</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>23</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1158</td>\n",
       "      <td>0.146</td>\n",
       "      <td>0.1643</td>\n",
       "      <td>0.0258</td>\n",
       "      <td>0.0517</td>\n",
       "      <td>0.1994</td>\n",
       "      <td>0.085240</td>\n",
       "      <td>0.146</td>\n",
       "      <td>0.2031</td>\n",
       "      <td>0.02983</td>\n",
       "      <td>0.0275</td>\n",
       "      <td>0.05769</td>\n",
       "      <td>0.03472</td>\n",
       "      <td>0.06653</td>\n",
       "      <td>0.182900</td>\n",
       "      <td>0.08327</td>\n",
       "      <td>0.1441</td>\n",
       "      <td>0.04934</td>\n",
       "      <td>0.1652</td>\n",
       "      <td>0.02968</td>\n",
       "      <td>0.02492</td>\n",
       "      <td>0.05836</td>\n",
       "      <td>0.08228</td>\n",
       "      <td>0.03306</td>\n",
       "      <td>0.07122</td>\n",
       "      <td>0.1781</td>\n",
       "      <td>0.2031</td>\n",
       "      <td>gnomADe AMR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>174/191</td>\n",
       "      <td></td>\n",
       "      <td>I23180T</td>\n",
       "      <td>aTt/aCt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs747122</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178549591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47519</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.65512G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Val21838Met</td>\n",
       "      <td>0/1</td>\n",
       "      <td>65</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>47.37%</td>\n",
       "      <td>2.68E-7</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1502</td>\n",
       "      <td>0.2731</td>\n",
       "      <td>0.1787</td>\n",
       "      <td>0.0248</td>\n",
       "      <td>0.0487</td>\n",
       "      <td>0.1973</td>\n",
       "      <td>0.092810</td>\n",
       "      <td>0.259</td>\n",
       "      <td>0.2066</td>\n",
       "      <td>0.03542</td>\n",
       "      <td>0.02751</td>\n",
       "      <td>0.05775</td>\n",
       "      <td>0.03343</td>\n",
       "      <td>0.06858</td>\n",
       "      <td>0.182100</td>\n",
       "      <td>0.1142</td>\n",
       "      <td>0.2553</td>\n",
       "      <td>0.04934</td>\n",
       "      <td>0.175</td>\n",
       "      <td>0.03602</td>\n",
       "      <td>0.02445</td>\n",
       "      <td>0.0588</td>\n",
       "      <td>0.08861</td>\n",
       "      <td>0.03148</td>\n",
       "      <td>0.09004</td>\n",
       "      <td>0.1777</td>\n",
       "      <td>0.2731</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>166/191</td>\n",
       "      <td></td>\n",
       "      <td>V21838M</td>\n",
       "      <td>Gtg/Atg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs9808377</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178556967</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47475</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.61568T&gt;C</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Ile20523Thr</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "      <td>60</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.035E-35</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>37</td>\n",
       "      <td>23</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.5072</td>\n",
       "      <td>0.5666</td>\n",
       "      <td>0.4049</td>\n",
       "      <td>0.7153</td>\n",
       "      <td>0.2495</td>\n",
       "      <td>0.5501</td>\n",
       "      <td>0.350700</td>\n",
       "      <td>0.5434</td>\n",
       "      <td>0.4241</td>\n",
       "      <td>0.2798</td>\n",
       "      <td>0.7039</td>\n",
       "      <td>0.2775</td>\n",
       "      <td>0.2239</td>\n",
       "      <td>0.3055</td>\n",
       "      <td>0.514500</td>\n",
       "      <td>0.3603</td>\n",
       "      <td>0.5417</td>\n",
       "      <td>0.5626</td>\n",
       "      <td>0.3812</td>\n",
       "      <td>0.2824</td>\n",
       "      <td>0.699</td>\n",
       "      <td>0.2849</td>\n",
       "      <td>0.2739</td>\n",
       "      <td>0.2233</td>\n",
       "      <td>0.3258</td>\n",
       "      <td>0.5153</td>\n",
       "      <td>0.7153</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>158/191</td>\n",
       "      <td></td>\n",
       "      <td>I20523T</td>\n",
       "      <td>aTc/aCc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs3829746</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178562809</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign/likely benign,benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47425</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.56704A&gt;G</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Ile18902Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>176</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.1486E-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.5084</td>\n",
       "      <td>0.5658</td>\n",
       "      <td>0.4078</td>\n",
       "      <td>0.7153</td>\n",
       "      <td>0.2535</td>\n",
       "      <td>0.5511</td>\n",
       "      <td>0.352200</td>\n",
       "      <td>0.5447</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.2812</td>\n",
       "      <td>0.7037</td>\n",
       "      <td>0.2773</td>\n",
       "      <td>0.2259</td>\n",
       "      <td>0.3083</td>\n",
       "      <td>0.515300</td>\n",
       "      <td>0.3617</td>\n",
       "      <td>0.5414</td>\n",
       "      <td>0.5604</td>\n",
       "      <td>0.3841</td>\n",
       "      <td>0.2843</td>\n",
       "      <td>0.7002</td>\n",
       "      <td>0.2855</td>\n",
       "      <td>0.2753</td>\n",
       "      <td>0.2254</td>\n",
       "      <td>0.3281</td>\n",
       "      <td>0.5157</td>\n",
       "      <td>0.7153</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>154/191</td>\n",
       "      <td></td>\n",
       "      <td>I18902V</td>\n",
       "      <td>Ata/Gta</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,23757202,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs3731746</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178566270</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47382</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.53243C&gt;T</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Thr17748Met</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>18</td>\n",
       "      <td>14</td>\n",
       "      <td>43.75%</td>\n",
       "      <td>9.8512E-6</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3423</td>\n",
       "      <td>0.3442</td>\n",
       "      <td>0.2061</td>\n",
       "      <td>0.629</td>\n",
       "      <td>0.171</td>\n",
       "      <td>0.317</td>\n",
       "      <td>0.232300</td>\n",
       "      <td>0.3372</td>\n",
       "      <td>0.1854</td>\n",
       "      <td>0.2299</td>\n",
       "      <td>0.6132</td>\n",
       "      <td>0.1726</td>\n",
       "      <td>0.1652</td>\n",
       "      <td>0.2081</td>\n",
       "      <td>0.303400</td>\n",
       "      <td>0.2396</td>\n",
       "      <td>0.3371</td>\n",
       "      <td>0.4813</td>\n",
       "      <td>0.1862</td>\n",
       "      <td>0.2284</td>\n",
       "      <td>0.6138</td>\n",
       "      <td>0.1817</td>\n",
       "      <td>0.1835</td>\n",
       "      <td>0.1664</td>\n",
       "      <td>0.2227</td>\n",
       "      <td>0.3061</td>\n",
       "      <td>0.629</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>154/191</td>\n",
       "      <td></td>\n",
       "      <td>T17748M</td>\n",
       "      <td>aCg/aTg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,23757202,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs744426</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178571293</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47329</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.48220C&gt;T</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Arg16074Cys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>70</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>51.35%</td>\n",
       "      <td>8.2514E-8</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2081</td>\n",
       "      <td>0.0666</td>\n",
       "      <td>0.1455</td>\n",
       "      <td>0.4504</td>\n",
       "      <td>0.1372</td>\n",
       "      <td>0.2669</td>\n",
       "      <td>0.175000</td>\n",
       "      <td>0.07013</td>\n",
       "      <td>0.1448</td>\n",
       "      <td>0.1686</td>\n",
       "      <td>0.4282</td>\n",
       "      <td>0.152</td>\n",
       "      <td>0.1432</td>\n",
       "      <td>0.1573</td>\n",
       "      <td>0.252100</td>\n",
       "      <td>0.1406</td>\n",
       "      <td>0.07529</td>\n",
       "      <td>0.4669</td>\n",
       "      <td>0.1268</td>\n",
       "      <td>0.1681</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.1612</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>0.1444</td>\n",
       "      <td>0.1254</td>\n",
       "      <td>0.2543</td>\n",
       "      <td>0.4669</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>154/191</td>\n",
       "      <td></td>\n",
       "      <td>R16074C</td>\n",
       "      <td>Cgc/Tgc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,23757202,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs10164753</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178574139</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47305</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.45374G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Arg15125His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>30%</td>\n",
       "      <td>9.678E-4</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0978</td>\n",
       "      <td>0.0877</td>\n",
       "      <td>0.1542</td>\n",
       "      <td>0.0258</td>\n",
       "      <td>0.0467</td>\n",
       "      <td>0.1984</td>\n",
       "      <td>0.079530</td>\n",
       "      <td>0.07788</td>\n",
       "      <td>0.2002</td>\n",
       "      <td>0.02979</td>\n",
       "      <td>0.0276</td>\n",
       "      <td>0.05774</td>\n",
       "      <td>0.03258</td>\n",
       "      <td>0.0637</td>\n",
       "      <td>0.181600</td>\n",
       "      <td>0.06355</td>\n",
       "      <td>0.07948</td>\n",
       "      <td>0.04945</td>\n",
       "      <td>0.1572</td>\n",
       "      <td>0.02967</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>0.05881</td>\n",
       "      <td>0.07911</td>\n",
       "      <td>0.03061</td>\n",
       "      <td>0.05641</td>\n",
       "      <td>0.1766</td>\n",
       "      <td>0.2002</td>\n",
       "      <td>gnomADe AMR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>154/191</td>\n",
       "      <td></td>\n",
       "      <td>R15125H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2303838</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178580212</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47247</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.40456G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Val13486Ile</td>\n",
       "      <td>0/1</td>\n",
       "      <td>58</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>47.06%</td>\n",
       "      <td>1.4997E-6</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3512</td>\n",
       "      <td>0.3434</td>\n",
       "      <td>0.2147</td>\n",
       "      <td>0.631</td>\n",
       "      <td>0.1849</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.239900</td>\n",
       "      <td>0.3351</td>\n",
       "      <td>0.1923</td>\n",
       "      <td>0.2368</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.1728</td>\n",
       "      <td>0.1737</td>\n",
       "      <td>0.2168</td>\n",
       "      <td>0.326300</td>\n",
       "      <td>0.245</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.4856</td>\n",
       "      <td>0.2001</td>\n",
       "      <td>0.2367</td>\n",
       "      <td>0.6135</td>\n",
       "      <td>0.1819</td>\n",
       "      <td>0.2057</td>\n",
       "      <td>0.1742</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.328</td>\n",
       "      <td>0.631</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>146/191</td>\n",
       "      <td></td>\n",
       "      <td>V13486I</td>\n",
       "      <td>Gtc/Atc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs554828165</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178580532</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Hypertrophic cardiomyopathy</td>\n",
       "      <td>uncertain significance</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>978758</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.40228G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Val13410Ile</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>16</td>\n",
       "      <td>19</td>\n",
       "      <td>54.29%</td>\n",
       "      <td>6.3952E-8</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>1.315e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0004137</td>\n",
       "      <td>0.001</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>145/191</td>\n",
       "      <td></td>\n",
       "      <td>V13410I</td>\n",
       "      <td>Gta/Ata</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2042996</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178586693</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47209</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.37589C&gt;T</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Thr12530Ile</td>\n",
       "      <td>1/1</td>\n",
       "      <td>224</td>\n",
       "      <td>40</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.6742E-23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.5076</td>\n",
       "      <td>0.5582</td>\n",
       "      <td>0.4078</td>\n",
       "      <td>0.7212</td>\n",
       "      <td>0.2485</td>\n",
       "      <td>0.5562</td>\n",
       "      <td>0.351100</td>\n",
       "      <td>0.5294</td>\n",
       "      <td>0.4264</td>\n",
       "      <td>0.2848</td>\n",
       "      <td>0.7085</td>\n",
       "      <td>0.2774</td>\n",
       "      <td>0.2238</td>\n",
       "      <td>0.3061</td>\n",
       "      <td>0.518500</td>\n",
       "      <td>0.357</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.3796</td>\n",
       "      <td>0.2877</td>\n",
       "      <td>0.7061</td>\n",
       "      <td>0.286</td>\n",
       "      <td>0.2753</td>\n",
       "      <td>0.2232</td>\n",
       "      <td>0.3265</td>\n",
       "      <td>0.5202</td>\n",
       "      <td>0.7212</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>136/191</td>\n",
       "      <td></td>\n",
       "      <td>T12530I</td>\n",
       "      <td>aCt/aTt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs16866406</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178592420</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47152</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.32966C&gt;T</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Pro10989Leu</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>53.85%</td>\n",
       "      <td>5.4595E-6</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2131</td>\n",
       "      <td>0.0666</td>\n",
       "      <td>0.1527</td>\n",
       "      <td>0.4573</td>\n",
       "      <td>0.1392</td>\n",
       "      <td>0.2781</td>\n",
       "      <td>0.177000</td>\n",
       "      <td>0.07051</td>\n",
       "      <td>0.1475</td>\n",
       "      <td>0.1751</td>\n",
       "      <td>0.4321</td>\n",
       "      <td>0.1519</td>\n",
       "      <td>0.144</td>\n",
       "      <td>0.1594</td>\n",
       "      <td>0.257800</td>\n",
       "      <td>0.1424</td>\n",
       "      <td>0.07557</td>\n",
       "      <td>0.4714</td>\n",
       "      <td>0.1294</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.4458</td>\n",
       "      <td>0.1632</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>0.1457</td>\n",
       "      <td>0.1292</td>\n",
       "      <td>0.2649</td>\n",
       "      <td>0.4714</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>129/191</td>\n",
       "      <td></td>\n",
       "      <td>P10989L</td>\n",
       "      <td>cCa/cTa</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,27681629,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs542259178</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178597807</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.30656T&gt;C</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Leu10219Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>72</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>25</td>\n",
       "      <td>20</td>\n",
       "      <td>44.44%</td>\n",
       "      <td>6.2178E-8</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>1.315e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0004142</td>\n",
       "      <td>0.001</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>122/191</td>\n",
       "      <td></td>\n",
       "      <td>L10219P</td>\n",
       "      <td>cTt/cCt</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TTN,TTN-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1001238</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178599800</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign/likely benign,benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47110</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.29482A&gt;G</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Asn9828Asp</td>\n",
       "      <td>1/1</td>\n",
       "      <td>218</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4508E-22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>13</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.5128</td>\n",
       "      <td>0.5688</td>\n",
       "      <td>0.4078</td>\n",
       "      <td>0.7252</td>\n",
       "      <td>0.2535</td>\n",
       "      <td>0.5593</td>\n",
       "      <td>0.355300</td>\n",
       "      <td>0.5438</td>\n",
       "      <td>0.432</td>\n",
       "      <td>0.2897</td>\n",
       "      <td>0.7126</td>\n",
       "      <td>0.2787</td>\n",
       "      <td>0.2275</td>\n",
       "      <td>0.3113</td>\n",
       "      <td>0.522900</td>\n",
       "      <td>0.3631</td>\n",
       "      <td>0.5402</td>\n",
       "      <td>0.5593</td>\n",
       "      <td>0.3872</td>\n",
       "      <td>0.2906</td>\n",
       "      <td>0.7088</td>\n",
       "      <td>0.2873</td>\n",
       "      <td>0.2816</td>\n",
       "      <td>0.227</td>\n",
       "      <td>0.3309</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.7252</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>117/191</td>\n",
       "      <td></td>\n",
       "      <td>N9828D</td>\n",
       "      <td>Aat/Gat</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2562847</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178653276</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Autosomal recessive limb-girdle muscular dystr...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>535163</td>\n",
       "      <td>ENST00000589042.5</td>\n",
       "      <td>c.38753T&gt;C</td>\n",
       "      <td>ENSP00000467141.1</td>\n",
       "      <td>p.Leu12918Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>78</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "      <td>46</td>\n",
       "      <td>23</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>1.315E-8</td>\n",
       "      <td>33</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1170</td>\n",
       "      <td>0.0416</td>\n",
       "      <td>0.1816</td>\n",
       "      <td>0.125</td>\n",
       "      <td>0.0636</td>\n",
       "      <td>0.2198</td>\n",
       "      <td>0.102200</td>\n",
       "      <td>0.04336</td>\n",
       "      <td>0.2238</td>\n",
       "      <td>0.04196</td>\n",
       "      <td>0.1291</td>\n",
       "      <td>0.104</td>\n",
       "      <td>0.04924</td>\n",
       "      <td>0.08419</td>\n",
       "      <td>0.194400</td>\n",
       "      <td>0.06855</td>\n",
       "      <td>0.04321</td>\n",
       "      <td>0.07269</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.04493</td>\n",
       "      <td>0.1194</td>\n",
       "      <td>0.1014</td>\n",
       "      <td>0.04839</td>\n",
       "      <td>0.04785</td>\n",
       "      <td>0.05673</td>\n",
       "      <td>0.1891</td>\n",
       "      <td>0.2238</td>\n",
       "      <td>gnomADe AMR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>198/363</td>\n",
       "      <td></td>\n",
       "      <td>L12918S</td>\n",
       "      <td>tTg/tCg</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs12994774</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178663311</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Autosomal recessive limb-girdle muscular dystr...</td>\n",
       "      <td>benign,uncertain significance</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>220982</td>\n",
       "      <td>ENST00000589042.5</td>\n",
       "      <td>c.36655T&gt;G</td>\n",
       "      <td>ENSP00000467141.1</td>\n",
       "      <td>p.Leu12219Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>81</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>100%</td>\n",
       "      <td>6.4467E-9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.099750</td>\n",
       "      <td>0.1372</td>\n",
       "      <td>0.1658</td>\n",
       "      <td>0.0592</td>\n",
       "      <td>0.3329</td>\n",
       "      <td>0.02273</td>\n",
       "      <td>0.05033</td>\n",
       "      <td>0.1178</td>\n",
       "      <td>0.154700</td>\n",
       "      <td>0.1768</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.4255</td>\n",
       "      <td>0.2221</td>\n",
       "      <td>0.1839</td>\n",
       "      <td>0.4479</td>\n",
       "      <td>0.1409</td>\n",
       "      <td>0.2077</td>\n",
       "      <td>0.1303</td>\n",
       "      <td>0.171</td>\n",
       "      <td>0.3094</td>\n",
       "      <td>0.4479</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>173/363</td>\n",
       "      <td></td>\n",
       "      <td>L12219V</td>\n",
       "      <td>Ttg/Gtg</td>\n",
       "      <td>-1</td>\n",
       "      <td>24033266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2163008</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178663651</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Dilated cardiomyopathy 1G,Autosomal recessive ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>192215</td>\n",
       "      <td>ENST00000425332.2</td>\n",
       "      <td>c.700G&gt;A</td>\n",
       "      <td>ENSP00000396805.2</td>\n",
       "      <td>p.Glu234Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>176</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "      <td>44</td>\n",
       "      <td>47</td>\n",
       "      <td>51.65%</td>\n",
       "      <td>2.0677E-18</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "      <td>23</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.2592</td>\n",
       "      <td>0.503</td>\n",
       "      <td>0.2277</td>\n",
       "      <td>0.1339</td>\n",
       "      <td>0.0755</td>\n",
       "      <td>0.2699</td>\n",
       "      <td>0.146000</td>\n",
       "      <td>0.4722</td>\n",
       "      <td>0.2531</td>\n",
       "      <td>0.06936</td>\n",
       "      <td>0.1345</td>\n",
       "      <td>0.1212</td>\n",
       "      <td>0.05948</td>\n",
       "      <td>0.1151</td>\n",
       "      <td>0.238900</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.4629</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.2271</td>\n",
       "      <td>0.07147</td>\n",
       "      <td>0.1297</td>\n",
       "      <td>0.1211</td>\n",
       "      <td>0.1184</td>\n",
       "      <td>0.05838</td>\n",
       "      <td>0.1643</td>\n",
       "      <td>0.2401</td>\n",
       "      <td>0.503</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>E234K</td>\n",
       "      <td>Gag/Aag</td>\n",
       "      <td>-1</td>\n",
       "      <td>23861362,34427245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2244492</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178689578</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46859</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.28132G&gt;A</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Gly9378Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>8.3837E-6</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.4133</td>\n",
       "      <td>0.6067</td>\n",
       "      <td>0.3501</td>\n",
       "      <td>0.1677</td>\n",
       "      <td>0.3827</td>\n",
       "      <td>0.4816</td>\n",
       "      <td>0.379800</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>0.3621</td>\n",
       "      <td>0.2732</td>\n",
       "      <td>0.1677</td>\n",
       "      <td>0.4981</td>\n",
       "      <td>0.3565</td>\n",
       "      <td>0.3488</td>\n",
       "      <td>0.453200</td>\n",
       "      <td>0.4381</td>\n",
       "      <td>0.6011</td>\n",
       "      <td>0.3051</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.2769</td>\n",
       "      <td>0.1701</td>\n",
       "      <td>0.5046</td>\n",
       "      <td>0.3481</td>\n",
       "      <td>0.3705</td>\n",
       "      <td>0.3781</td>\n",
       "      <td>0.4592</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>120/312</td>\n",
       "      <td></td>\n",
       "      <td>G9378R</td>\n",
       "      <td>Gga/Aga</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2042995</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178693639</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign/likely benign,benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46852</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.27832A&gt;G</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Ile9278Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>141</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>100%</td>\n",
       "      <td>7.9107E-15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td>benign(0.06)</td>\n",
       "      <td>0.4778</td>\n",
       "      <td>0.5877</td>\n",
       "      <td>0.4193</td>\n",
       "      <td>0.5685</td>\n",
       "      <td>0.2455</td>\n",
       "      <td>0.5164</td>\n",
       "      <td>0.336600</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.4297</td>\n",
       "      <td>0.2666</td>\n",
       "      <td>0.5737</td>\n",
       "      <td>0.2517</td>\n",
       "      <td>0.227</td>\n",
       "      <td>0.2987</td>\n",
       "      <td>0.479600</td>\n",
       "      <td>0.3595</td>\n",
       "      <td>0.5524</td>\n",
       "      <td>0.5571</td>\n",
       "      <td>0.3902</td>\n",
       "      <td>0.2671</td>\n",
       "      <td>0.5723</td>\n",
       "      <td>0.2605</td>\n",
       "      <td>0.2785</td>\n",
       "      <td>0.229</td>\n",
       "      <td>0.328</td>\n",
       "      <td>0.484</td>\n",
       "      <td>0.5877</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>116/312</td>\n",
       "      <td></td>\n",
       "      <td>I9278V</td>\n",
       "      <td>Att/Gtt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs4893852</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178706721</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46818</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.25421T&gt;C</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Ile8474Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>43</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>46.15%</td>\n",
       "      <td>4.6796E-5</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.1274</td>\n",
       "      <td>0.2231</td>\n",
       "      <td>0.1556</td>\n",
       "      <td>0.0258</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>0.1769</td>\n",
       "      <td>0.082390</td>\n",
       "      <td>0.2039</td>\n",
       "      <td>0.1942</td>\n",
       "      <td>0.04102</td>\n",
       "      <td>0.02855</td>\n",
       "      <td>0.05694</td>\n",
       "      <td>0.02692</td>\n",
       "      <td>0.06223</td>\n",
       "      <td>0.168600</td>\n",
       "      <td>0.09388</td>\n",
       "      <td>0.2014</td>\n",
       "      <td>0.04276</td>\n",
       "      <td>0.1518</td>\n",
       "      <td>0.03604</td>\n",
       "      <td>0.02616</td>\n",
       "      <td>0.05751</td>\n",
       "      <td>0.0538</td>\n",
       "      <td>0.02601</td>\n",
       "      <td>0.07534</td>\n",
       "      <td>0.1608</td>\n",
       "      <td>0.2231</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>99/312</td>\n",
       "      <td></td>\n",
       "      <td>I8474T</td>\n",
       "      <td>aTt/aCt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs13398235</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178713977</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46786</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.22949C&gt;T</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Pro7650Leu</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>56%</td>\n",
       "      <td>4.7528E-6</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.1114</td>\n",
       "      <td>0.1649</td>\n",
       "      <td>0.1527</td>\n",
       "      <td>0.0258</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>0.1759</td>\n",
       "      <td>0.078440</td>\n",
       "      <td>0.1524</td>\n",
       "      <td>0.1903</td>\n",
       "      <td>0.04086</td>\n",
       "      <td>0.02849</td>\n",
       "      <td>0.05763</td>\n",
       "      <td>0.02637</td>\n",
       "      <td>0.05979</td>\n",
       "      <td>0.167200</td>\n",
       "      <td>0.07946</td>\n",
       "      <td>0.1521</td>\n",
       "      <td>0.04286</td>\n",
       "      <td>0.1452</td>\n",
       "      <td>0.03576</td>\n",
       "      <td>0.02623</td>\n",
       "      <td>0.05845</td>\n",
       "      <td>0.04747</td>\n",
       "      <td>0.02517</td>\n",
       "      <td>0.06897</td>\n",
       "      <td>0.1624</td>\n",
       "      <td>0.1903</td>\n",
       "      <td>gnomADe AMR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>89/312</td>\n",
       "      <td></td>\n",
       "      <td>P7650L</td>\n",
       "      <td>cCg/cTg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs12693164</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178714366</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign/likely benign,benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46777</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.22676A&gt;G</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Asn7559Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>60%</td>\n",
       "      <td>2.2548E-5</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.2470</td>\n",
       "      <td>0.1936</td>\n",
       "      <td>0.1859</td>\n",
       "      <td>0.4365</td>\n",
       "      <td>0.1561</td>\n",
       "      <td>0.2607</td>\n",
       "      <td>0.179700</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.176</td>\n",
       "      <td>0.1271</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.09591</td>\n",
       "      <td>0.1503</td>\n",
       "      <td>0.1672</td>\n",
       "      <td>0.231700</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>0.1636</td>\n",
       "      <td>0.4714</td>\n",
       "      <td>0.1739</td>\n",
       "      <td>0.1308</td>\n",
       "      <td>0.4344</td>\n",
       "      <td>0.0981</td>\n",
       "      <td>0.1234</td>\n",
       "      <td>0.1541</td>\n",
       "      <td>0.1686</td>\n",
       "      <td>0.2336</td>\n",
       "      <td>0.4714</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>88/312</td>\n",
       "      <td></td>\n",
       "      <td>N7559S</td>\n",
       "      <td>aAt/aGt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs13390491</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178717600</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46756</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.21542G&gt;A</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Ser7181Asn</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>10</td>\n",
       "      <td>17</td>\n",
       "      <td>62.96%</td>\n",
       "      <td>1.7891E-7</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.2438</td>\n",
       "      <td>0.1936</td>\n",
       "      <td>0.183</td>\n",
       "      <td>0.4355</td>\n",
       "      <td>0.1571</td>\n",
       "      <td>0.2464</td>\n",
       "      <td>0.177500</td>\n",
       "      <td>0.1652</td>\n",
       "      <td>0.1732</td>\n",
       "      <td>0.1222</td>\n",
       "      <td>0.4297</td>\n",
       "      <td>0.09552</td>\n",
       "      <td>0.1497</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.221700</td>\n",
       "      <td>0.1679</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.4715</td>\n",
       "      <td>0.1712</td>\n",
       "      <td>0.1256</td>\n",
       "      <td>0.4321</td>\n",
       "      <td>0.09874</td>\n",
       "      <td>0.1203</td>\n",
       "      <td>0.1537</td>\n",
       "      <td>0.1668</td>\n",
       "      <td>0.2237</td>\n",
       "      <td>0.4715</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>84/312</td>\n",
       "      <td></td>\n",
       "      <td>S7181N</td>\n",
       "      <td>aGc/aAc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2627043</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178717810</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46750</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.21332C&gt;A</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Ala7111Glu</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.4449</td>\n",
       "      <td>0.4637</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0.6577</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.4734</td>\n",
       "      <td>0.308400</td>\n",
       "      <td>0.4227</td>\n",
       "      <td>0.4115</td>\n",
       "      <td>0.2003</td>\n",
       "      <td>0.6675</td>\n",
       "      <td>0.2016</td>\n",
       "      <td>0.2025</td>\n",
       "      <td>0.2687</td>\n",
       "      <td>0.430400</td>\n",
       "      <td>0.305</td>\n",
       "      <td>0.423</td>\n",
       "      <td>0.5463</td>\n",
       "      <td>0.3579</td>\n",
       "      <td>0.2031</td>\n",
       "      <td>0.6483</td>\n",
       "      <td>0.2037</td>\n",
       "      <td>0.1867</td>\n",
       "      <td>0.2051</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.4305</td>\n",
       "      <td>0.6675</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>84/312</td>\n",
       "      <td></td>\n",
       "      <td>A7111E</td>\n",
       "      <td>gCg/gAg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,27182706,18414213,23861362,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs16866465</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178718769</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46731</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.20699A&gt;C</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Glu6900Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>43</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>29</td>\n",
       "      <td>13</td>\n",
       "      <td>30.95%</td>\n",
       "      <td>4.0777E-5</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.2388</td>\n",
       "      <td>0.1732</td>\n",
       "      <td>0.183</td>\n",
       "      <td>0.4375</td>\n",
       "      <td>0.1571</td>\n",
       "      <td>0.2464</td>\n",
       "      <td>0.177300</td>\n",
       "      <td>0.1521</td>\n",
       "      <td>0.1734</td>\n",
       "      <td>0.1231</td>\n",
       "      <td>0.431</td>\n",
       "      <td>0.09594</td>\n",
       "      <td>0.1502</td>\n",
       "      <td>0.1647</td>\n",
       "      <td>0.222500</td>\n",
       "      <td>0.1647</td>\n",
       "      <td>0.1536</td>\n",
       "      <td>0.4715</td>\n",
       "      <td>0.1701</td>\n",
       "      <td>0.1259</td>\n",
       "      <td>0.4332</td>\n",
       "      <td>0.09868</td>\n",
       "      <td>0.1203</td>\n",
       "      <td>0.1541</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.2238</td>\n",
       "      <td>0.4715</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>81/312</td>\n",
       "      <td></td>\n",
       "      <td>E6900A</td>\n",
       "      <td>gAa/gCa</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs12693166</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178722403</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46698</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.18652G&gt;C</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Asp6218His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>27</td>\n",
       "      <td>19</td>\n",
       "      <td>41.3%</td>\n",
       "      <td>1.8161E-7</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.2470</td>\n",
       "      <td>0.202</td>\n",
       "      <td>0.1844</td>\n",
       "      <td>0.4375</td>\n",
       "      <td>0.1581</td>\n",
       "      <td>0.2474</td>\n",
       "      <td>0.178500</td>\n",
       "      <td>0.171</td>\n",
       "      <td>0.174</td>\n",
       "      <td>0.1233</td>\n",
       "      <td>0.431</td>\n",
       "      <td>0.09591</td>\n",
       "      <td>0.1501</td>\n",
       "      <td>0.1649</td>\n",
       "      <td>0.222400</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.1717</td>\n",
       "      <td>0.4757</td>\n",
       "      <td>0.1729</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0.4341</td>\n",
       "      <td>0.09826</td>\n",
       "      <td>0.1203</td>\n",
       "      <td>0.1541</td>\n",
       "      <td>0.1687</td>\n",
       "      <td>0.2251</td>\n",
       "      <td>0.4757</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>74/312</td>\n",
       "      <td></td>\n",
       "      <td>D6218H</td>\n",
       "      <td>Gat/Cat</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs17355446</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178724514</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46677</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.17129C&gt;T</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Ala5710Val</td>\n",
       "      <td>0/1</td>\n",
       "      <td>35</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>43.48%</td>\n",
       "      <td>2.8066E-4</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.0515</td>\n",
       "      <td>0.0023</td>\n",
       "      <td>0.1441</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.0298</td>\n",
       "      <td>0.1258</td>\n",
       "      <td>0.056800</td>\n",
       "      <td>0.005844</td>\n",
       "      <td>0.1838</td>\n",
       "      <td>0.01864</td>\n",
       "      <td>0.0007289</td>\n",
       "      <td>0.05548</td>\n",
       "      <td>0.01931</td>\n",
       "      <td>0.04699</td>\n",
       "      <td>0.128000</td>\n",
       "      <td>0.03257</td>\n",
       "      <td>0.006592</td>\n",
       "      <td>0.04276</td>\n",
       "      <td>0.1303</td>\n",
       "      <td>0.01701</td>\n",
       "      <td>0.00155</td>\n",
       "      <td>0.05668</td>\n",
       "      <td>0.02215</td>\n",
       "      <td>0.0194</td>\n",
       "      <td>0.02874</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.1838</td>\n",
       "      <td>gnomADe AMR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>69/312</td>\n",
       "      <td></td>\n",
       "      <td>A5710V</td>\n",
       "      <td>gCa/gTa</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,19268276,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs11888217</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178728625</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46656</td>\n",
       "      <td>ENST00000342992.11</td>\n",
       "      <td>c.15569G&gt;A</td>\n",
       "      <td>ENSP00000343764.6</td>\n",
       "      <td>p.Ser5190Asn</td>\n",
       "      <td>0/1</td>\n",
       "      <td>94</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>15</td>\n",
       "      <td>24</td>\n",
       "      <td>61.54%</td>\n",
       "      <td>3.1797E-10</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>0.0932</td>\n",
       "      <td>0.087</td>\n",
       "      <td>0.1513</td>\n",
       "      <td>0.0238</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>0.1912</td>\n",
       "      <td>0.074130</td>\n",
       "      <td>0.083</td>\n",
       "      <td>0.1859</td>\n",
       "      <td>0.03685</td>\n",
       "      <td>0.0264</td>\n",
       "      <td>0.0563</td>\n",
       "      <td>0.02651</td>\n",
       "      <td>0.05835</td>\n",
       "      <td>0.174700</td>\n",
       "      <td>0.05955</td>\n",
       "      <td>0.08205</td>\n",
       "      <td>0.04276</td>\n",
       "      <td>0.1389</td>\n",
       "      <td>0.03228</td>\n",
       "      <td>0.02414</td>\n",
       "      <td>0.05725</td>\n",
       "      <td>0.0443</td>\n",
       "      <td>0.02558</td>\n",
       "      <td>0.04585</td>\n",
       "      <td>0.1672</td>\n",
       "      <td>0.1912</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>63/312</td>\n",
       "      <td></td>\n",
       "      <td>S5190N</td>\n",
       "      <td>aGt/aAt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2627037</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178741811</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47807</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.10909C&gt;T</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Pro3637Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>37</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>36.67%</td>\n",
       "      <td>1.594E-4</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2732</td>\n",
       "      <td>0.4531</td>\n",
       "      <td>0.3602</td>\n",
       "      <td>0.2133</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>0.2209</td>\n",
       "      <td>0.182000</td>\n",
       "      <td>0.4225</td>\n",
       "      <td>0.4465</td>\n",
       "      <td>0.1041</td>\n",
       "      <td>0.2335</td>\n",
       "      <td>0.1093</td>\n",
       "      <td>0.0742</td>\n",
       "      <td>0.145</td>\n",
       "      <td>0.212900</td>\n",
       "      <td>0.2045</td>\n",
       "      <td>0.4158</td>\n",
       "      <td>0.09121</td>\n",
       "      <td>0.313</td>\n",
       "      <td>0.09879</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.1095</td>\n",
       "      <td>0.09177</td>\n",
       "      <td>0.07399</td>\n",
       "      <td>0.1699</td>\n",
       "      <td>0.2132</td>\n",
       "      <td>0.4531</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>45/191</td>\n",
       "      <td></td>\n",
       "      <td>P3637S</td>\n",
       "      <td>Cca/Tca</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,31872004,31827228,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs4893853</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178767769</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign/likely benign,benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47648</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.9323A&gt;G</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Lys3108Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>62</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>59.26%</td>\n",
       "      <td>6.1806E-7</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0861</td>\n",
       "      <td>0.0749</td>\n",
       "      <td>0.1542</td>\n",
       "      <td>0.0208</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>0.1728</td>\n",
       "      <td>0.070410</td>\n",
       "      <td>0.07095</td>\n",
       "      <td>0.1855</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.0252</td>\n",
       "      <td>0.0553</td>\n",
       "      <td>0.0232</td>\n",
       "      <td>0.05554</td>\n",
       "      <td>0.169200</td>\n",
       "      <td>0.05353</td>\n",
       "      <td>0.07005</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1348</td>\n",
       "      <td>0.02592</td>\n",
       "      <td>0.02172</td>\n",
       "      <td>0.05701</td>\n",
       "      <td>0.02848</td>\n",
       "      <td>0.02214</td>\n",
       "      <td>0.03776</td>\n",
       "      <td>0.1618</td>\n",
       "      <td>0.1855</td>\n",
       "      <td>gnomADe AMR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>38/191</td>\n",
       "      <td></td>\n",
       "      <td>K3108R</td>\n",
       "      <td>aAg/aGg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs4894048</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178773994</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47377</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.7036G&gt;A</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Gly2346Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>34.15%</td>\n",
       "      <td>1.6028E-5</td>\n",
       "      <td>16</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0875</td>\n",
       "      <td>0.0915</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.0149</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>0.1585</td>\n",
       "      <td>0.070020</td>\n",
       "      <td>0.09463</td>\n",
       "      <td>0.1874</td>\n",
       "      <td>0.03398</td>\n",
       "      <td>0.01873</td>\n",
       "      <td>0.05609</td>\n",
       "      <td>0.02334</td>\n",
       "      <td>0.05666</td>\n",
       "      <td>0.152300</td>\n",
       "      <td>0.0595</td>\n",
       "      <td>0.09199</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1388</td>\n",
       "      <td>0.03141</td>\n",
       "      <td>0.01619</td>\n",
       "      <td>0.05785</td>\n",
       "      <td>0.02848</td>\n",
       "      <td>0.02228</td>\n",
       "      <td>0.04159</td>\n",
       "      <td>0.1464</td>\n",
       "      <td>0.1874</td>\n",
       "      <td>gnomADe AMR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>29/191</td>\n",
       "      <td></td>\n",
       "      <td>G2346S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,29897334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>TTN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs16866538</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178795185</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Dilated cardiomyopath...</td>\n",
       "      <td>benign,benign/likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47703</td>\n",
       "      <td>ENST00000342175.11</td>\n",
       "      <td>c.982C&gt;T</td>\n",
       "      <td>ENSP00000340554.6</td>\n",
       "      <td>p.Arg328Cys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>61</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>68.18%</td>\n",
       "      <td>7.4178E-7</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2360</td>\n",
       "      <td>0.1974</td>\n",
       "      <td>0.098</td>\n",
       "      <td>0.5883</td>\n",
       "      <td>0.0477</td>\n",
       "      <td>0.2168</td>\n",
       "      <td>0.123700</td>\n",
       "      <td>0.1689</td>\n",
       "      <td>0.1012</td>\n",
       "      <td>0.06619</td>\n",
       "      <td>0.5975</td>\n",
       "      <td>0.04886</td>\n",
       "      <td>0.05134</td>\n",
       "      <td>0.08445</td>\n",
       "      <td>0.188000</td>\n",
       "      <td>0.1112</td>\n",
       "      <td>0.1669</td>\n",
       "      <td>0.1272</td>\n",
       "      <td>0.09281</td>\n",
       "      <td>0.06167</td>\n",
       "      <td>0.5812</td>\n",
       "      <td>0.05353</td>\n",
       "      <td>0.01582</td>\n",
       "      <td>0.05249</td>\n",
       "      <td>0.09273</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.5975</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/191</td>\n",
       "      <td></td>\n",
       "      <td>R328C</td>\n",
       "      <td>Cgt/Tgt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,31641117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>TMEM43</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs4685076</td>\n",
       "      <td>No</td>\n",
       "      <td>chr3</td>\n",
       "      <td>14132927</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Emery-Dreifuss muscul...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46148</td>\n",
       "      <td>ENST00000306077.5</td>\n",
       "      <td>c.504A&gt;T</td>\n",
       "      <td>ENSP00000303992.5</td>\n",
       "      <td>p.Lys168Asn</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>48.72%</td>\n",
       "      <td>1.0268E-7</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>tolerated(0.1)</td>\n",
       "      <td>benign(0.127)</td>\n",
       "      <td>0.3526</td>\n",
       "      <td>0.3026</td>\n",
       "      <td>0.366</td>\n",
       "      <td>0.4613</td>\n",
       "      <td>0.2545</td>\n",
       "      <td>0.3998</td>\n",
       "      <td>0.324900</td>\n",
       "      <td>0.2839</td>\n",
       "      <td>0.4204</td>\n",
       "      <td>0.2495</td>\n",
       "      <td>0.4401</td>\n",
       "      <td>0.3008</td>\n",
       "      <td>0.2832</td>\n",
       "      <td>0.295</td>\n",
       "      <td>0.372400</td>\n",
       "      <td>0.2952</td>\n",
       "      <td>0.2803</td>\n",
       "      <td>0.239</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.2367</td>\n",
       "      <td>0.4286</td>\n",
       "      <td>0.2894</td>\n",
       "      <td>0.2437</td>\n",
       "      <td>0.2855</td>\n",
       "      <td>0.2939</td>\n",
       "      <td>0.3849</td>\n",
       "      <td>0.4613</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/12</td>\n",
       "      <td></td>\n",
       "      <td>K168N</td>\n",
       "      <td>aaA/aaT</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,23757202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>TMEM43</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2340917</td>\n",
       "      <td>No</td>\n",
       "      <td>chr3</td>\n",
       "      <td>14133762</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Emery-Dreifuss muscul...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>46150</td>\n",
       "      <td>ENST00000306077.5</td>\n",
       "      <td>c.536T&gt;C</td>\n",
       "      <td>ENSP00000303992.5</td>\n",
       "      <td>p.Met179Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>93</td>\n",
       "      <td>59</td>\n",
       "      <td>59</td>\n",
       "      <td>33</td>\n",
       "      <td>26</td>\n",
       "      <td>44.07%</td>\n",
       "      <td>4.2407E-10</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4633</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5058</td>\n",
       "      <td>0.6756</td>\n",
       "      <td>0.2624</td>\n",
       "      <td>0.4376</td>\n",
       "      <td>0.378900</td>\n",
       "      <td>0.4165</td>\n",
       "      <td>0.5574</td>\n",
       "      <td>0.2534</td>\n",
       "      <td>0.6696</td>\n",
       "      <td>0.3141</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.409700</td>\n",
       "      <td>0.3565</td>\n",
       "      <td>0.4154</td>\n",
       "      <td>0.2451</td>\n",
       "      <td>0.4354</td>\n",
       "      <td>0.2409</td>\n",
       "      <td>0.6482</td>\n",
       "      <td>0.3006</td>\n",
       "      <td>0.2468</td>\n",
       "      <td>0.2926</td>\n",
       "      <td>0.338</td>\n",
       "      <td>0.4331</td>\n",
       "      <td>0.6756</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/12</td>\n",
       "      <td></td>\n",
       "      <td>M179T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,23757202,1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SCN5A</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1805124</td>\n",
       "      <td>No</td>\n",
       "      <td>chr3</td>\n",
       "      <td>38603929</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiac arrhythmia,Cardiovascular phenotype,Co...</td>\n",
       "      <td>benign/likely benign,pathogenic,benign,likely ...</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>48289</td>\n",
       "      <td>ENST00000333535.9</td>\n",
       "      <td>c.1673A&gt;G</td>\n",
       "      <td>ENSP00000328968.4</td>\n",
       "      <td>p.His558Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>138</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>29</td>\n",
       "      <td>36</td>\n",
       "      <td>55.38%</td>\n",
       "      <td>1.5685E-14</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td></td>\n",
       "      <td>0.2304</td>\n",
       "      <td>0.3094</td>\n",
       "      <td>0.2277</td>\n",
       "      <td>0.1012</td>\n",
       "      <td>0.2167</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.220400</td>\n",
       "      <td>0.2922</td>\n",
       "      <td>0.2158</td>\n",
       "      <td>0.2014</td>\n",
       "      <td>0.09327</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2301</td>\n",
       "      <td>0.2281</td>\n",
       "      <td>0.247000</td>\n",
       "      <td>0.2452</td>\n",
       "      <td>0.2932</td>\n",
       "      <td>0.3392</td>\n",
       "      <td>0.2302</td>\n",
       "      <td>0.1885</td>\n",
       "      <td>0.1011</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.2215</td>\n",
       "      <td>0.2399</td>\n",
       "      <td>0.239</td>\n",
       "      <td>0.2414</td>\n",
       "      <td>0.3392</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>12/28</td>\n",
       "      <td></td>\n",
       "      <td>H558R</td>\n",
       "      <td>cAc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21056700,24033266,27182706,20850564,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>PITX2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2278783</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>110620968</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1296541</td>\n",
       "      <td>ENST00000557119.2</td>\n",
       "      <td>c.607C&gt;T</td>\n",
       "      <td>ENSP00000475617.1</td>\n",
       "      <td>p.Arg203Cys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>42.86%</td>\n",
       "      <td>3.7193E-7</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>deleterious low confidence(0)</td>\n",
       "      <td>benign(0.152)</td>\n",
       "      <td>0.2554</td>\n",
       "      <td>0.3782</td>\n",
       "      <td>0.1628</td>\n",
       "      <td>0.2718</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.2331</td>\n",
       "      <td>0.184800</td>\n",
       "      <td>0.3529</td>\n",
       "      <td>0.1444</td>\n",
       "      <td>0.1724</td>\n",
       "      <td>0.2477</td>\n",
       "      <td>0.1401</td>\n",
       "      <td>0.1604</td>\n",
       "      <td>0.1885</td>\n",
       "      <td>0.223300</td>\n",
       "      <td>0.2162</td>\n",
       "      <td>0.3456</td>\n",
       "      <td>0.1107</td>\n",
       "      <td>0.1815</td>\n",
       "      <td>0.1702</td>\n",
       "      <td>0.2567</td>\n",
       "      <td>0.1486</td>\n",
       "      <td>0.2342</td>\n",
       "      <td>0.1547</td>\n",
       "      <td>0.2173</td>\n",
       "      <td>0.2385</td>\n",
       "      <td>0.3782</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>R203C</td>\n",
       "      <td>Cgc/Tgc</td>\n",
       "      <td>-1</td>\n",
       "      <td>28911203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>PITX2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2278782</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>110620998</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1238490</td>\n",
       "      <td>ENST00000557119.2</td>\n",
       "      <td>c.577C&gt;T</td>\n",
       "      <td>ENSP00000475617.1</td>\n",
       "      <td>p.Gln193Ter</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>26</td>\n",
       "      <td>18</td>\n",
       "      <td>40.91%</td>\n",
       "      <td>4.2568E-7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1542</td>\n",
       "      <td>0.1331</td>\n",
       "      <td>0.1297</td>\n",
       "      <td>0.1736</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.1738</td>\n",
       "      <td>0.154600</td>\n",
       "      <td>0.1357</td>\n",
       "      <td>0.1297</td>\n",
       "      <td>0.1608</td>\n",
       "      <td>0.152</td>\n",
       "      <td>0.119</td>\n",
       "      <td>0.1582</td>\n",
       "      <td>0.166</td>\n",
       "      <td>0.184300</td>\n",
       "      <td>0.1472</td>\n",
       "      <td>0.1385</td>\n",
       "      <td>0.1107</td>\n",
       "      <td>0.1481</td>\n",
       "      <td>0.1544</td>\n",
       "      <td>0.1588</td>\n",
       "      <td>0.1215</td>\n",
       "      <td>0.1994</td>\n",
       "      <td>0.1523</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.1866</td>\n",
       "      <td>0.1994</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>Q193*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>-1</td>\n",
       "      <td>18300947,28911203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>LOC107986633,LAMA4</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs200177134</td>\n",
       "      <td>No</td>\n",
       "      <td>chr6</td>\n",
       "      <td>112114132</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Dilated cardiomyopathy 1JJ,Cardiovascular phen...</td>\n",
       "      <td>uncertain significance,likely benign,conflicti...</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>44404</td>\n",
       "      <td>ENST00000230538.12</td>\n",
       "      <td>c.5270C&gt;T</td>\n",
       "      <td>ENSP00000230538.7</td>\n",
       "      <td>p.Pro1757Leu</td>\n",
       "      <td>0/1</td>\n",
       "      <td>74</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>50%</td>\n",
       "      <td>3.8991E-8</td>\n",
       "      <td>17</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably damaging(0.983)</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.000442</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001158</td>\n",
       "      <td>0.0006949</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002466</td>\n",
       "      <td>0.0001633</td>\n",
       "      <td>0.002319</td>\n",
       "      <td>0.0001776</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001311</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002499</td>\n",
       "      <td>0.000478</td>\n",
       "      <td>0.001247</td>\n",
       "      <td>0.002319</td>\n",
       "      <td>gnomADe SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>38/39</td>\n",
       "      <td></td>\n",
       "      <td>P1757L</td>\n",
       "      <td>cCc/cTc</td>\n",
       "      <td>-1</td>\n",
       "      <td>24033266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>LOC107986633,LAMA4</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1050349</td>\n",
       "      <td>No</td>\n",
       "      <td>chr6</td>\n",
       "      <td>112136181</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Dilated cardiomyopathy 1JJ</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>44377</td>\n",
       "      <td>ENST00000230538.12</td>\n",
       "      <td>c.3356C&gt;G</td>\n",
       "      <td>ENSP00000230538.7</td>\n",
       "      <td>p.Pro1119Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>176</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.1486E-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>4</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably damaging(0.991)</td>\n",
       "      <td>0.2105</td>\n",
       "      <td>0.1233</td>\n",
       "      <td>0.2104</td>\n",
       "      <td>0.3601</td>\n",
       "      <td>0.2316</td>\n",
       "      <td>0.1524</td>\n",
       "      <td>0.225300</td>\n",
       "      <td>0.1433</td>\n",
       "      <td>0.1705</td>\n",
       "      <td>0.288</td>\n",
       "      <td>0.3506</td>\n",
       "      <td>0.1732</td>\n",
       "      <td>0.2561</td>\n",
       "      <td>0.2411</td>\n",
       "      <td>0.154100</td>\n",
       "      <td>0.2156</td>\n",
       "      <td>0.1475</td>\n",
       "      <td>0.2259</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.2885</td>\n",
       "      <td>0.3402</td>\n",
       "      <td>0.1684</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.2598</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.1647</td>\n",
       "      <td>0.3601</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>25/39</td>\n",
       "      <td></td>\n",
       "      <td>P1119R</td>\n",
       "      <td>cCt/cGt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,23861362,35284542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>NOS3</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs199688227</td>\n",
       "      <td>No</td>\n",
       "      <td>chr7</td>\n",
       "      <td>150993922</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000297494.8</td>\n",
       "      <td>c.119G&gt;A</td>\n",
       "      <td>ENSP00000297494.3</td>\n",
       "      <td>p.Arg40Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>79</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>18</td>\n",
       "      <td>21</td>\n",
       "      <td>53.85%</td>\n",
       "      <td>1.1402E-8</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated low confidence(0.67)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000150</td>\n",
       "      <td>0.001091</td>\n",
       "      <td>0.0001759</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6.383e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000269</td>\n",
       "      <td>0.0004209</td>\n",
       "      <td>0.001425</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.413e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001425</td>\n",
       "      <td>gnomADg AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/27</td>\n",
       "      <td></td>\n",
       "      <td>R40Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>PRKAG2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs79474211</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr7</td>\n",
       "      <td>151781320</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>PRKAG2 cardiac syndrome,Cardiovascular phenoty...</td>\n",
       "      <td>benign,conflicting interpretations of pathogen...</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>36697</td>\n",
       "      <td>ENST00000287878.9</td>\n",
       "      <td>c.298G&gt;A</td>\n",
       "      <td>ENSP00000287878.3</td>\n",
       "      <td>p.Gly100Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>114</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>25</td>\n",
       "      <td>30</td>\n",
       "      <td>54.55%</td>\n",
       "      <td>3.6885E-12</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>tolerated low confidence(0.21)</td>\n",
       "      <td></td>\n",
       "      <td>0.0146</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0347</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.0378</td>\n",
       "      <td>0.007360</td>\n",
       "      <td>0.0005567</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.0008945</td>\n",
       "      <td>0.03179</td>\n",
       "      <td>0.0001858</td>\n",
       "      <td>0.0008907</td>\n",
       "      <td>0.005554</td>\n",
       "      <td>0.034860</td>\n",
       "      <td>0.003092</td>\n",
       "      <td>0.000725</td>\n",
       "      <td>0.009868</td>\n",
       "      <td>0.0009823</td>\n",
       "      <td>0.0008661</td>\n",
       "      <td>0.03591</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01911</td>\n",
       "      <td>0.0006619</td>\n",
       "      <td>0.005258</td>\n",
       "      <td>0.03425</td>\n",
       "      <td>0.0378</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/16</td>\n",
       "      <td></td>\n",
       "      <td>G100S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,24938736,27930701,23778007,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs374190304</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104829118</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Tangier disease,Hypoalphalipoproteinemia, prim...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>364441</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.1913G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg638Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>53.57%</td>\n",
       "      <td>2.3037E-6</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>deleterious low confidence(0.05)</td>\n",
       "      <td>probably damaging(0.957)</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0102</td>\n",
       "      <td>0.001856</td>\n",
       "      <td>0.0001231</td>\n",
       "      <td>0</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.000598</td>\n",
       "      <td>0</td>\n",
       "      <td>8.814e-06</td>\n",
       "      <td>0.000163</td>\n",
       "      <td>0.014700</td>\n",
       "      <td>0.0004929</td>\n",
       "      <td>0.0001449</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003854</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.01365</td>\n",
       "      <td>0.0147</td>\n",
       "      <td>gnomADe SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/50</td>\n",
       "      <td></td>\n",
       "      <td>R638Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>-1</td>\n",
       "      <td>15297675,25215231</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene Name Gene_Match         rsID Literature  CHROM        POS  \\\n",
       "0        NPPA,NPPA-AS1        Yes       rs5063        Yes   chr1   11847591   \n",
       "1                 RYR2        Yes   rs34967813         No   chr1  237678090   \n",
       "2                 MYPN        Yes   rs10823148         No  chr10   68166577   \n",
       "3                 MYPN        Yes   rs10997975         No  chr10   68174164   \n",
       "4                 MYPN        Yes    rs7916821         No  chr10   68174212   \n",
       "5                 MYPN        Yes    rs3814182         No  chr10   68174501   \n",
       "6                 MYPN        Yes    rs7079481         No  chr10   68199485   \n",
       "7                 PKP2        Yes    rs1046116         No  chr12   32869000   \n",
       "8                 MYH6        Yes  rs573489857         No  chr14   23407154   \n",
       "9                 DSG2        Yes    rs2230234         No  chr18   31524751   \n",
       "10                LDLR        Yes  rs576530433         No  chr19   11090668   \n",
       "11                APOB        Yes    rs1042031         No   chr2   21002881   \n",
       "12                APOB        Yes  rs575505383         No   chr2   21019079   \n",
       "13         TTN,TTN-AS1        Yes    rs3829747         No   chr2  178532834   \n",
       "14         TTN,TTN-AS1        Yes    rs4894028         No   chr2  178539023   \n",
       "15         TTN,TTN-AS1        Yes    rs3731749         No   chr2  178541464   \n",
       "16         TTN,TTN-AS1        Yes   rs62621236         No   chr2  178543986   \n",
       "17         TTN,TTN-AS1        Yes     rs747122         No   chr2  178549591   \n",
       "18         TTN,TTN-AS1        Yes    rs9808377         No   chr2  178556967   \n",
       "19         TTN,TTN-AS1        Yes    rs3829746         No   chr2  178562809   \n",
       "20         TTN,TTN-AS1        Yes    rs3731746         No   chr2  178566270   \n",
       "21         TTN,TTN-AS1        Yes     rs744426         No   chr2  178571293   \n",
       "22         TTN,TTN-AS1        Yes   rs10164753         No   chr2  178574139   \n",
       "23         TTN,TTN-AS1        Yes    rs2303838         No   chr2  178580212   \n",
       "24         TTN,TTN-AS1        Yes  rs554828165         No   chr2  178580532   \n",
       "25         TTN,TTN-AS1        Yes    rs2042996         No   chr2  178586693   \n",
       "26         TTN,TTN-AS1        Yes   rs16866406         No   chr2  178592420   \n",
       "27         TTN,TTN-AS1        Yes  rs542259178         No   chr2  178597807   \n",
       "28         TTN,TTN-AS1        Yes    rs1001238         No   chr2  178599800   \n",
       "29                 TTN        Yes    rs2562847         No   chr2  178653276   \n",
       "30                 TTN        Yes   rs12994774         No   chr2  178663311   \n",
       "31                 TTN        Yes    rs2163008         No   chr2  178663651   \n",
       "32                 TTN        Yes    rs2244492         No   chr2  178689578   \n",
       "33                 TTN        Yes    rs2042995         No   chr2  178693639   \n",
       "34                 TTN        Yes    rs4893852         No   chr2  178706721   \n",
       "35                 TTN        Yes   rs13398235         No   chr2  178713977   \n",
       "36                 TTN        Yes   rs12693164         No   chr2  178714366   \n",
       "37                 TTN        Yes   rs13390491         No   chr2  178717600   \n",
       "38                 TTN        Yes    rs2627043         No   chr2  178717810   \n",
       "39                 TTN        Yes   rs16866465         No   chr2  178718769   \n",
       "40                 TTN        Yes   rs12693166         No   chr2  178722403   \n",
       "41                 TTN        Yes   rs17355446         No   chr2  178724514   \n",
       "42                 TTN        Yes   rs11888217         No   chr2  178728625   \n",
       "43                 TTN        Yes    rs2627037         No   chr2  178741811   \n",
       "44                 TTN        Yes    rs4893853         No   chr2  178767769   \n",
       "45                 TTN        Yes    rs4894048         No   chr2  178773994   \n",
       "46                 TTN        Yes   rs16866538         No   chr2  178795185   \n",
       "47              TMEM43        Yes    rs4685076         No   chr3   14132927   \n",
       "48              TMEM43        Yes    rs2340917         No   chr3   14133762   \n",
       "49               SCN5A        Yes    rs1805124         No   chr3   38603929   \n",
       "50               PITX2        Yes    rs2278783         No   chr4  110620968   \n",
       "51               PITX2        Yes    rs2278782         No   chr4  110620998   \n",
       "52  LOC107986633,LAMA4        Yes  rs200177134         No   chr6  112114132   \n",
       "53  LOC107986633,LAMA4        Yes    rs1050349         No   chr6  112136181   \n",
       "54                NOS3        Yes  rs199688227         No   chr7  150993922   \n",
       "55              PRKAG2        Yes   rs79474211        Yes   chr7  151781320   \n",
       "56               ABCA1        Yes  rs374190304         No   chr9  104829118   \n",
       "\n",
       "   REF ALT      Zygosity                             Consequence  \\\n",
       "0    C   T  Heterozygous                        missense variant   \n",
       "1    A   G  Heterozygous                        missense variant   \n",
       "2    C   G    Homozygous                        missense variant   \n",
       "3    G   A    Homozygous                        missense variant   \n",
       "4    G   A    Homozygous                        missense variant   \n",
       "5    C   G    Homozygous                        missense variant   \n",
       "6    C   A    Homozygous                        missense variant   \n",
       "7    A   G    Homozygous                        missense variant   \n",
       "8    G   T  Heterozygous                        missense variant   \n",
       "9    A   G  Heterozygous                        missense variant   \n",
       "10   G   A  Heterozygous                        missense variant   \n",
       "11   C   T  Heterozygous                        missense variant   \n",
       "12   C   T  Heterozygous                        missense variant   \n",
       "13   C   T  Heterozygous                        missense variant   \n",
       "14   C   T  Heterozygous                        missense variant   \n",
       "15   C   T  Heterozygous                        missense variant   \n",
       "16   A   G  Heterozygous                        missense variant   \n",
       "17   C   T  Heterozygous                        missense variant   \n",
       "18   A   G    Homozygous                        missense variant   \n",
       "19   T   C    Homozygous                        missense variant   \n",
       "20   G   A  Heterozygous                        missense variant   \n",
       "21   G   A  Heterozygous                        missense variant   \n",
       "22   C   T  Heterozygous                        missense variant   \n",
       "23   C   T  Heterozygous                        missense variant   \n",
       "24   C   T  Heterozygous                        missense variant   \n",
       "25   G   A    Homozygous                        missense variant   \n",
       "26   G   A  Heterozygous                        missense variant   \n",
       "27   A   G  Heterozygous                        missense variant   \n",
       "28   T   C    Homozygous                        missense variant   \n",
       "29   A   G  Heterozygous                        missense variant   \n",
       "30   A   C    Homozygous                        missense variant   \n",
       "31   C   T  Heterozygous                        missense variant   \n",
       "32   C   T  Heterozygous                        missense variant   \n",
       "33   T   C    Homozygous                        missense variant   \n",
       "34   A   G  Heterozygous                        missense variant   \n",
       "35   G   A  Heterozygous                        missense variant   \n",
       "36   T   C  Heterozygous                        missense variant   \n",
       "37   C   T  Heterozygous                        missense variant   \n",
       "38   G   T    Homozygous  missense variant,splice region variant   \n",
       "39   T   G  Heterozygous                        missense variant   \n",
       "40   C   G  Heterozygous                        missense variant   \n",
       "41   G   A  Heterozygous                        missense variant   \n",
       "42   C   T  Heterozygous                        missense variant   \n",
       "43   G   A  Heterozygous                        missense variant   \n",
       "44   T   C  Heterozygous                        missense variant   \n",
       "45   C   T  Heterozygous                        missense variant   \n",
       "46   G   A  Heterozygous                        missense variant   \n",
       "47   A   T  Heterozygous                        missense variant   \n",
       "48   T   C  Heterozygous                        missense variant   \n",
       "49   T   C  Heterozygous                        missense variant   \n",
       "50   G   A  Heterozygous                        missense variant   \n",
       "51   G   A  Heterozygous                             stop gained   \n",
       "52   G   A  Heterozygous                        missense variant   \n",
       "53   G   C    Homozygous                        missense variant   \n",
       "54   G   A  Heterozygous                        missense variant   \n",
       "55   C   T  Heterozygous                        missense variant   \n",
       "56   C   T  Heterozygous                        missense variant   \n",
       "\n",
       "   Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0               7/10  MODERATE           5.0   \n",
       "1               7/10  MODERATE           5.0   \n",
       "2               7/10  MODERATE           5.0   \n",
       "3               7/10  MODERATE           5.0   \n",
       "4               7/10  MODERATE           5.0   \n",
       "5               7/10  MODERATE           5.0   \n",
       "6               7/10  MODERATE           5.0   \n",
       "7               7/10  MODERATE           5.0   \n",
       "8               7/10  MODERATE           5.0   \n",
       "9               7/10  MODERATE           5.0   \n",
       "10              7/10  MODERATE           5.0   \n",
       "11              7/10  MODERATE           5.0   \n",
       "12              7/10  MODERATE           5.0   \n",
       "13              7/10  MODERATE           5.0   \n",
       "14              7/10  MODERATE           5.0   \n",
       "15              7/10  MODERATE           5.0   \n",
       "16              7/10  MODERATE           5.0   \n",
       "17              7/10  MODERATE           5.0   \n",
       "18              7/10  MODERATE           5.0   \n",
       "19              7/10  MODERATE           5.0   \n",
       "20              7/10  MODERATE           5.0   \n",
       "21              7/10  MODERATE           5.0   \n",
       "22              7/10  MODERATE           5.0   \n",
       "23              7/10  MODERATE           5.0   \n",
       "24              7/10  MODERATE           5.0   \n",
       "25              7/10  MODERATE           5.0   \n",
       "26              7/10  MODERATE           5.0   \n",
       "27              7/10  MODERATE           5.0   \n",
       "28              7/10  MODERATE           5.0   \n",
       "29              7/10  MODERATE           5.0   \n",
       "30              7/10  MODERATE           5.0   \n",
       "31              7/10  MODERATE           5.0   \n",
       "32              7/10  MODERATE           5.0   \n",
       "33              7/10  MODERATE           5.0   \n",
       "34              7/10  MODERATE           5.0   \n",
       "35              7/10  MODERATE           5.0   \n",
       "36              7/10  MODERATE           5.0   \n",
       "37              7/10  MODERATE           5.0   \n",
       "38              7/10  MODERATE           5.0   \n",
       "39              7/10  MODERATE           5.0   \n",
       "40              7/10  MODERATE           5.0   \n",
       "41              7/10  MODERATE           5.0   \n",
       "42              7/10  MODERATE           5.0   \n",
       "43              7/10  MODERATE           5.0   \n",
       "44              7/10  MODERATE           5.0   \n",
       "45              7/10  MODERATE           5.0   \n",
       "46              7/10  MODERATE           5.0   \n",
       "47              7/10  MODERATE           5.0   \n",
       "48              7/10  MODERATE           5.0   \n",
       "49              7/10  MODERATE           5.0   \n",
       "50              7/10  MODERATE           5.0   \n",
       "51             10/10      HIGH          10.0   \n",
       "52              7/10  MODERATE           5.0   \n",
       "53              7/10  MODERATE           5.0   \n",
       "54              7/10  MODERATE           5.0   \n",
       "55              7/10  MODERATE           5.0   \n",
       "56              7/10  MODERATE           5.0   \n",
       "\n",
       "                                        ClinVar_CLNDN  \\\n",
       "0                    Atrial fibrillation, familial, 6   \n",
       "1   Cardiovascular phenotype,Cardiomyopathy,Catech...   \n",
       "2   MYPN-related myopathy,Cardiovascular phenotype...   \n",
       "3   MYPN-related myopathy,Cardiovascular phenotype...   \n",
       "4   Cardiovascular phenotype,MYPN-related myopathy...   \n",
       "5   Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "6   MYPN-related myopathy,Cardiovascular phenotype...   \n",
       "7   Cardiovascular phenotype,Arrhythmogenic right ...   \n",
       "8   Cardiovascular phenotype,Cardiomyopathy,Hypert...   \n",
       "9   Cardiovascular phenotype,Cardiomyopathy,Arrhyt...   \n",
       "10                                                      \n",
       "11  Cardiovascular phenotype,Familial hypobetalipo...   \n",
       "12  Hypercholesterolemia, familial, 1,Familial hyp...   \n",
       "13  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "14  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "15  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "16  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "17  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "18  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "19  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "20  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "21  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "22  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "23  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "24                        Hypertrophic cardiomyopathy   \n",
       "25  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "26  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "27                                                      \n",
       "28  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "29  Autosomal recessive limb-girdle muscular dystr...   \n",
       "30  Autosomal recessive limb-girdle muscular dystr...   \n",
       "31  Dilated cardiomyopathy 1G,Autosomal recessive ...   \n",
       "32  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "33  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "34  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "35  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "36  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "37  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "38  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "39  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "40  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "41  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "42  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "43  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "44  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "45  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "46  Cardiovascular phenotype,Dilated cardiomyopath...   \n",
       "47  Cardiovascular phenotype,Emery-Dreifuss muscul...   \n",
       "48  Cardiovascular phenotype,Emery-Dreifuss muscul...   \n",
       "49  Cardiac arrhythmia,Cardiovascular phenotype,Co...   \n",
       "50                                       not provided   \n",
       "51                                       not provided   \n",
       "52  Dilated cardiomyopathy 1JJ,Cardiovascular phen...   \n",
       "53                         Dilated cardiomyopathy 1JJ   \n",
       "54                                                      \n",
       "55  PRKAG2 cardiac syndrome,Cardiovascular phenoty...   \n",
       "56  Tangier disease,Hypoalphalipoproteinemia, prim...   \n",
       "\n",
       "                                             CLIN_SIG  \\\n",
       "0                                              benign   \n",
       "1                                benign,likely benign   \n",
       "2                                              benign   \n",
       "3                                              benign   \n",
       "4                                              benign   \n",
       "5                                              benign   \n",
       "6                                              benign   \n",
       "7                                              benign   \n",
       "8                                benign,likely benign   \n",
       "9                                              benign   \n",
       "10                                                      \n",
       "11                        benign,benign/likely benign   \n",
       "12        uncertain significance,benign,likely benign   \n",
       "13                        benign,benign/likely benign   \n",
       "14                                             benign   \n",
       "15                        benign,benign/likely benign   \n",
       "16                                             benign   \n",
       "17                        benign,benign/likely benign   \n",
       "18                        benign,benign/likely benign   \n",
       "19                        benign/likely benign,benign   \n",
       "20                        benign,benign/likely benign   \n",
       "21                        benign,benign/likely benign   \n",
       "22                                             benign   \n",
       "23                        benign,benign/likely benign   \n",
       "24                             uncertain significance   \n",
       "25                        benign,benign/likely benign   \n",
       "26                        benign,benign/likely benign   \n",
       "27                                                      \n",
       "28                        benign/likely benign,benign   \n",
       "29                                             benign   \n",
       "30                      benign,uncertain significance   \n",
       "31                                             benign   \n",
       "32                        benign,benign/likely benign   \n",
       "33                        benign/likely benign,benign   \n",
       "34                        benign,benign/likely benign   \n",
       "35                        benign,benign/likely benign   \n",
       "36                        benign/likely benign,benign   \n",
       "37                        benign,benign/likely benign   \n",
       "38                        benign,benign/likely benign   \n",
       "39                        benign,benign/likely benign   \n",
       "40                        benign,benign/likely benign   \n",
       "41                                             benign   \n",
       "42                        benign,benign/likely benign   \n",
       "43                                             benign   \n",
       "44                        benign/likely benign,benign   \n",
       "45                        benign,benign/likely benign   \n",
       "46                        benign,benign/likely benign   \n",
       "47                                             benign   \n",
       "48                                             benign   \n",
       "49  benign/likely benign,pathogenic,benign,likely ...   \n",
       "50                                             benign   \n",
       "51                                             benign   \n",
       "52  uncertain significance,likely benign,conflicti...   \n",
       "53                                             benign   \n",
       "54                                                      \n",
       "55  benign,conflicting interpretations of pathogen...   \n",
       "56                               benign,likely benign   \n",
       "\n",
       "                                   ClinVar_CLNREVSTAT  ClinVar  \\\n",
       "0   criteria provided, multiple submitters, no con...  1169995   \n",
       "1   criteria provided, multiple submitters, no con...    36750   \n",
       "2   criteria provided, multiple submitters, no con...    31804   \n",
       "3   criteria provided, multiple submitters, no con...    31797   \n",
       "4   criteria provided, multiple submitters, no con...    31798   \n",
       "5   criteria provided, multiple submitters, no con...    31808   \n",
       "6   criteria provided, multiple submitters, no con...    31800   \n",
       "7   criteria provided, multiple submitters, no con...    45006   \n",
       "8   criteria provided, multiple submitters, no con...   164260   \n",
       "9   criteria provided, multiple submitters, no con...    44331   \n",
       "10                                                               \n",
       "11  criteria provided, multiple submitters, no con...   128417   \n",
       "12     criteria provided, conflicting interpretations   334160   \n",
       "13  criteria provided, multiple submitters, no con...    47663   \n",
       "14  criteria provided, multiple submitters, no con...    47610   \n",
       "15  criteria provided, multiple submitters, no con...    47586   \n",
       "16  criteria provided, multiple submitters, no con...    47566   \n",
       "17  criteria provided, multiple submitters, no con...    47519   \n",
       "18  criteria provided, multiple submitters, no con...    47475   \n",
       "19  criteria provided, multiple submitters, no con...    47425   \n",
       "20  criteria provided, multiple submitters, no con...    47382   \n",
       "21  criteria provided, multiple submitters, no con...    47329   \n",
       "22  criteria provided, multiple submitters, no con...    47305   \n",
       "23  criteria provided, multiple submitters, no con...    47247   \n",
       "24                criteria provided, single submitter   978758   \n",
       "25  criteria provided, multiple submitters, no con...    47209   \n",
       "26  criteria provided, multiple submitters, no con...    47152   \n",
       "27                                                               \n",
       "28  criteria provided, multiple submitters, no con...    47110   \n",
       "29  criteria provided, multiple submitters, no con...   535163   \n",
       "30     criteria provided, conflicting interpretations   220982   \n",
       "31  criteria provided, multiple submitters, no con...   192215   \n",
       "32  criteria provided, multiple submitters, no con...    46859   \n",
       "33  criteria provided, multiple submitters, no con...    46852   \n",
       "34  criteria provided, multiple submitters, no con...    46818   \n",
       "35  criteria provided, multiple submitters, no con...    46786   \n",
       "36  criteria provided, multiple submitters, no con...    46777   \n",
       "37  criteria provided, multiple submitters, no con...    46756   \n",
       "38  criteria provided, multiple submitters, no con...    46750   \n",
       "39  criteria provided, multiple submitters, no con...    46731   \n",
       "40  criteria provided, multiple submitters, no con...    46698   \n",
       "41  criteria provided, multiple submitters, no con...    46677   \n",
       "42  criteria provided, multiple submitters, no con...    46656   \n",
       "43  criteria provided, multiple submitters, no con...    47807   \n",
       "44  criteria provided, multiple submitters, no con...    47648   \n",
       "45  criteria provided, multiple submitters, no con...    47377   \n",
       "46  criteria provided, multiple submitters, no con...    47703   \n",
       "47  criteria provided, multiple submitters, no con...    46148   \n",
       "48  criteria provided, multiple submitters, no con...    46150   \n",
       "49  criteria provided, multiple submitters, no con...    48289   \n",
       "50                criteria provided, single submitter  1296541   \n",
       "51                criteria provided, single submitter  1238490   \n",
       "52     criteria provided, conflicting interpretations    44404   \n",
       "53  criteria provided, multiple submitters, no con...    44377   \n",
       "54                                                               \n",
       "55     criteria provided, conflicting interpretations    36697   \n",
       "56  criteria provided, multiple submitters, no con...   364441   \n",
       "\n",
       "                 HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0    ENST00000376480.7            c.94G>A  ENSP00000365663.3   \n",
       "1    ENST00000366574.7          c.8873A>G  ENSP00000355533.2   \n",
       "2    ENST00000354393.7          c.1059C>G  ENSP00000346369.2   \n",
       "3    ENST00000354393.7          c.1247G>A  ENSP00000346369.2   \n",
       "4    ENST00000354393.7          c.1295G>A  ENSP00000346369.2   \n",
       "5    ENST00000354393.7          c.1584C>G  ENSP00000346369.2   \n",
       "6    ENST00000354393.7          c.2578C>A  ENSP00000346369.2   \n",
       "7   ENST00000070846.11          c.1097T>C  ENSP00000070846.6   \n",
       "8    ENST00000405093.9            c.70C>A  ENSP00000386041.3   \n",
       "9   ENST00000261590.13           c.877A>G  ENSP00000261590.8   \n",
       "10   ENST00000252444.9            c.71G>A  ENSP00000252444.5   \n",
       "11   ENST00000233242.5         c.12541G>A  ENSP00000233242.1   \n",
       "12   ENST00000233242.5          c.3034G>A  ENSP00000233242.1   \n",
       "13  ENST00000342175.11         c.77162G>A  ENSP00000340554.6   \n",
       "14  ENST00000342175.11         c.72293G>A  ENSP00000340554.6   \n",
       "15  ENST00000342175.11         c.70994G>A  ENSP00000340554.6   \n",
       "16  ENST00000342175.11         c.69539T>C  ENSP00000340554.6   \n",
       "17  ENST00000342175.11         c.65512G>A  ENSP00000340554.6   \n",
       "18  ENST00000342175.11         c.61568T>C  ENSP00000340554.6   \n",
       "19  ENST00000342175.11         c.56704A>G  ENSP00000340554.6   \n",
       "20  ENST00000342175.11         c.53243C>T  ENSP00000340554.6   \n",
       "21  ENST00000342175.11         c.48220C>T  ENSP00000340554.6   \n",
       "22  ENST00000342175.11         c.45374G>A  ENSP00000340554.6   \n",
       "23  ENST00000342175.11         c.40456G>A  ENSP00000340554.6   \n",
       "24  ENST00000342175.11         c.40228G>A  ENSP00000340554.6   \n",
       "25  ENST00000342175.11         c.37589C>T  ENSP00000340554.6   \n",
       "26  ENST00000342175.11         c.32966C>T  ENSP00000340554.6   \n",
       "27  ENST00000342175.11         c.30656T>C  ENSP00000340554.6   \n",
       "28  ENST00000342175.11         c.29482A>G  ENSP00000340554.6   \n",
       "29   ENST00000589042.5         c.38753T>C  ENSP00000467141.1   \n",
       "30   ENST00000589042.5         c.36655T>G  ENSP00000467141.1   \n",
       "31   ENST00000425332.2           c.700G>A  ENSP00000396805.2   \n",
       "32  ENST00000342992.11         c.28132G>A  ENSP00000343764.6   \n",
       "33  ENST00000342992.11         c.27832A>G  ENSP00000343764.6   \n",
       "34  ENST00000342992.11         c.25421T>C  ENSP00000343764.6   \n",
       "35  ENST00000342992.11         c.22949C>T  ENSP00000343764.6   \n",
       "36  ENST00000342992.11         c.22676A>G  ENSP00000343764.6   \n",
       "37  ENST00000342992.11         c.21542G>A  ENSP00000343764.6   \n",
       "38  ENST00000342992.11         c.21332C>A  ENSP00000343764.6   \n",
       "39  ENST00000342992.11         c.20699A>C  ENSP00000343764.6   \n",
       "40  ENST00000342992.11         c.18652G>C  ENSP00000343764.6   \n",
       "41  ENST00000342992.11         c.17129C>T  ENSP00000343764.6   \n",
       "42  ENST00000342992.11         c.15569G>A  ENSP00000343764.6   \n",
       "43  ENST00000342175.11         c.10909C>T  ENSP00000340554.6   \n",
       "44  ENST00000342175.11          c.9323A>G  ENSP00000340554.6   \n",
       "45  ENST00000342175.11          c.7036G>A  ENSP00000340554.6   \n",
       "46  ENST00000342175.11           c.982C>T  ENSP00000340554.6   \n",
       "47   ENST00000306077.5           c.504A>T  ENSP00000303992.5   \n",
       "48   ENST00000306077.5           c.536T>C  ENSP00000303992.5   \n",
       "49   ENST00000333535.9          c.1673A>G  ENSP00000328968.4   \n",
       "50   ENST00000557119.2           c.607C>T  ENSP00000475617.1   \n",
       "51   ENST00000557119.2           c.577C>T  ENSP00000475617.1   \n",
       "52  ENST00000230538.12          c.5270C>T  ENSP00000230538.7   \n",
       "53  ENST00000230538.12          c.3356C>G  ENSP00000230538.7   \n",
       "54   ENST00000297494.8           c.119G>A  ENSP00000297494.3   \n",
       "55   ENST00000287878.9           c.298G>A  ENSP00000287878.3   \n",
       "56   ENST00000374736.8          c.1913G>A  ENSP00000363868.3   \n",
       "\n",
       "   HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD    FREQ        PVAL RDF RDR  \\\n",
       "0          p.Val32Met  0/1   30  25  25  16   9     36%   8.1542E-4   9   7   \n",
       "1        p.Gln2958Arg  0/1   76  45  45  24  21  46.67%   2.2206E-8  14  10   \n",
       "2         p.Phe353Leu  1/1  255  49  49   0  49    100%   3.925E-29   0   0   \n",
       "3         p.Ser416Asn  1/1  200  35  35   0  35    100%  8.9137E-21   0   0   \n",
       "4         p.Ser432Asn  1/1  236  41  41   0  41    100%  2.3541E-24   0   0   \n",
       "5         p.Ser528Arg  1/1  255  55  55   0  55    100%  1.0149E-32   0   0   \n",
       "6         p.Pro860Thr  1/1  194  34  34   0  34    100%  3.5146E-20   0   0   \n",
       "7         p.Leu366Pro  1/1  255  46  46   0  46    100%  2.4343E-27   0   0   \n",
       "8          p.Leu24Ile  0/1   98  57  57  30  27  47.37%  1.2661E-10  22   8   \n",
       "9         p.Ile293Val  0/1   77  43  43  22  21  48.84%   1.8298E-8  17   5   \n",
       "10         p.Arg24His  0/1   92  42  42  18  24  57.14%  5.5097E-10  16   2   \n",
       "11       p.Glu4181Lys  0/1   67  47  47  28  19  40.43%   1.9354E-7  15  13   \n",
       "12       p.Glu1012Lys  0/1   90  46  46  22  24  52.17%   9.761E-10  14   8   \n",
       "13      p.Arg25721His  0/1   60  40  40  23  17   42.5%    8.743E-7  15   8   \n",
       "14      p.Arg24098His  0/1   54  33  32  17  15  46.88%   3.5464E-6  13   4   \n",
       "15      p.Arg23665His  0/1   45  33  33  20  13  39.39%   2.8029E-5  16   4   \n",
       "16      p.Ile23180Thr  0/1  110  46  46  18  28  60.87%  8.7676E-12  13   5   \n",
       "17      p.Val21838Met  0/1   65  38  38  20  18  47.37%     2.68E-7  16   4   \n",
       "18      p.Ile20523Thr  1/1  255  60  60   0  60    100%   1.035E-35   0   0   \n",
       "19      p.Ile18902Val  1/1  176  31  31   0  31    100%  2.1486E-18   0   0   \n",
       "20      p.Thr17748Met  0/1   50  32  32  18  14  43.75%   9.8512E-6  15   3   \n",
       "21      p.Arg16074Cys  0/1   70  37  37  18  19  51.35%   8.2514E-8  13   5   \n",
       "22      p.Arg15125His  0/1   30  30  30  21   9     30%    9.678E-4  12   9   \n",
       "23      p.Val13486Ile  0/1   58  34  34  18  16  47.06%   1.4997E-6  12   6   \n",
       "24      p.Val13410Ile  0/1   71  35  35  16  19  54.29%   6.3952E-8   9   7   \n",
       "25      p.Thr12530Ile  1/1  224  40  39   0  39    100%  3.6742E-23   0   0   \n",
       "26      p.Pro10989Leu  0/1   52  26  26  12  14  53.85%   5.4595E-6  11   1   \n",
       "27      p.Leu10219Pro  0/1   72  45  45  25  20  44.44%   6.2178E-8  14  11   \n",
       "28       p.Asn9828Asp  1/1  218  38  38   0  38    100%  1.4508E-22   0   0   \n",
       "29      p.Leu12918Ser  0/1   78  69  69  46  23  33.33%    1.315E-8  33  13   \n",
       "30      p.Leu12219Val  1/1   81  15  15   0  15    100%   6.4467E-9   0   0   \n",
       "31        p.Glu234Lys  0/1  176  91  91  44  47  51.65%  2.0677E-18  24  20   \n",
       "32       p.Gly9378Arg  0/1   50  30  30  16  14  46.67%   8.3837E-6  11   5   \n",
       "33       p.Ile9278Val  1/1  141  25  25   0  25    100%  7.9107E-15   0   0   \n",
       "34       p.Ile8474Thr  0/1   43  26  26  14  12  46.15%   4.6796E-5   8   6   \n",
       "35       p.Pro7650Leu  0/1   53  25  25  11  14     56%   4.7528E-6   9   2   \n",
       "36       p.Asn7559Ser  0/1   46  20  20   8  12     60%   2.2548E-5   6   2   \n",
       "37       p.Ser7181Asn  0/1   67  27  27  10  17  62.96%   1.7891E-7   9   1   \n",
       "38       p.Ala7111Glu  1/1  123  22  22   0  22    100%  4.7526E-13   0   0   \n",
       "39       p.Glu6900Ala  0/1   43  42  42  29  13  30.95%   4.0777E-5  22   7   \n",
       "40       p.Asp6218His  0/1   67  46  46  27  19   41.3%   1.8161E-7  22   5   \n",
       "41       p.Ala5710Val  0/1   35  23  23  13  10  43.48%   2.8066E-4   7   6   \n",
       "42       p.Ser5190Asn  0/1   94  39  39  15  24  61.54%  3.1797E-10  10   5   \n",
       "43       p.Pro3637Ser  0/1   37  30  30  19  11  36.67%    1.594E-4  14   5   \n",
       "44       p.Lys3108Arg  0/1   62  27  27  11  16  59.26%   6.1806E-7   6   5   \n",
       "45       p.Gly2346Ser  0/1   47  41  41  27  14  34.15%   1.6028E-5  16  11   \n",
       "46        p.Arg328Cys  0/1   61  22  22   7  15  68.18%   7.4178E-7   2   5   \n",
       "47        p.Lys168Asn  0/1   69  39  39  20  19  48.72%   1.0268E-7  11   9   \n",
       "48        p.Met179Thr  0/1   93  59  59  33  26  44.07%  4.2407E-10  24   9   \n",
       "49        p.His558Arg  0/1  138  65  65  29  36  55.38%  1.5685E-14  23   6   \n",
       "50        p.Arg203Cys  0/1   64  42  42  24  18  42.86%   3.7193E-7  16   8   \n",
       "51        p.Gln193Ter  0/1   63  44  44  26  18  40.91%   4.2568E-7  13  13   \n",
       "52       p.Pro1757Leu  0/1   74  40  40  20  20     50%   3.8991E-8  17   3   \n",
       "53       p.Pro1119Arg  1/1  176  31  31   0  31    100%  2.1486E-18   0   0   \n",
       "54         p.Arg40Gln  0/1   79  39  39  18  21  53.85%   1.1402E-8  12   6   \n",
       "55        p.Gly100Ser  0/1  114  55  55  25  30  54.55%  3.6885E-12  16   9   \n",
       "56        p.Arg638Gln  0/1   56  28  28  13  15  53.57%   2.3037E-6   9   4   \n",
       "\n",
       "   ADF ADR                              SIFT                  PolyPhen  \\\n",
       "0    7   2                   tolerated(0.22)             benign(0.057)   \n",
       "1   12   9                                               benign(0.033)   \n",
       "2   30  19  deleterious low confidence(0.04)                             \n",
       "3   27   8                      tolerated(1)                             \n",
       "4   35   6                    tolerated(0.2)                             \n",
       "5   30  25                 deleterious(0.04)                             \n",
       "6   23  11                    deleterious(0)                             \n",
       "7   34  12                   tolerated(0.89)                 benign(0)   \n",
       "8   18   9  deleterious low confidence(0.04)             benign(0.022)   \n",
       "9   19   2                    deleterious(0)                             \n",
       "10  18   6     deleterious low confidence(0)             benign(0.099)   \n",
       "11  13   6                   tolerated(0.22)             benign(0.179)   \n",
       "12  16   8                   tolerated(0.63)             benign(0.196)   \n",
       "13  12   5                                                               \n",
       "14   8   7                                                               \n",
       "15   5   8                                                               \n",
       "16  23   5                                                               \n",
       "17  13   5                                                               \n",
       "18  37  23                                                               \n",
       "19  23   8                                                               \n",
       "20  10   4                                                               \n",
       "21   7  12                                                               \n",
       "22   6   3                                                               \n",
       "23   8   8                                                               \n",
       "24  15   4                                                               \n",
       "25  31   8                                                               \n",
       "26  13   1                                                               \n",
       "27  13   7                                                               \n",
       "28  25  13                                                               \n",
       "29  12  11                                                               \n",
       "30  12   3                                                               \n",
       "31  24  23                      tolerated(1)                 benign(0)   \n",
       "32   7   7                                                  unknown(0)   \n",
       "33  22   3                                                benign(0.06)   \n",
       "34   8   4                                                  unknown(0)   \n",
       "35  10   4                                                  unknown(0)   \n",
       "36   9   3                                                  unknown(0)   \n",
       "37  15   2                                                  unknown(0)   \n",
       "38  13   9                                                  unknown(0)   \n",
       "39  12   1                                                  unknown(0)   \n",
       "40  10   9                                                  unknown(0)   \n",
       "41   9   1                                                  unknown(0)   \n",
       "42  14  10                                                  unknown(0)   \n",
       "43   3   8                                                               \n",
       "44  15   1                                                               \n",
       "45  12   2                                                               \n",
       "46  10   5                                                               \n",
       "47  13   6                    tolerated(0.1)             benign(0.127)   \n",
       "48  17   9                      tolerated(1)                 benign(0)   \n",
       "49  24  12       tolerated low confidence(1)                             \n",
       "50  11   7     deleterious low confidence(0)             benign(0.152)   \n",
       "51  13   5                                                               \n",
       "52  18   2                    deleterious(0)  probably damaging(0.983)   \n",
       "53  27   4                    deleterious(0)  probably damaging(0.991)   \n",
       "54  17   4    tolerated low confidence(0.67)                 benign(0)   \n",
       "55  19  11    tolerated low confidence(0.21)                             \n",
       "56  11   4  deleterious low confidence(0.05)  probably damaging(0.957)   \n",
       "\n",
       "        AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0   0.0735  0.0613  0.0476  0.1151  0.0348  0.1053    0.055860        0.06115   \n",
       "1   0.0992  0.0113  0.1988       0  0.2942  0.0491    0.214100        0.04827   \n",
       "2   0.3165  0.1104  0.4049  0.2163  0.5099  0.4366    0.439800         0.1566   \n",
       "3   0.3359  0.1762  0.4035  0.2649  0.4851  0.4233    0.430900         0.2045   \n",
       "4   0.3251  0.1762  0.4035  0.2163  0.4851  0.4182    0.426200         0.2041   \n",
       "5   0.4714   0.447  0.4597  0.3512  0.5666  0.5389    0.518700         0.4599   \n",
       "6   0.3403  0.2163   0.402  0.2391  0.4841  0.4202    0.432000         0.2546   \n",
       "7   0.1514  0.2103  0.1614  0.0308  0.2157  0.1227    0.180000         0.1991   \n",
       "8   0.0012       0       0       0       0  0.0061    0.000632              0   \n",
       "9   0.0323   0.003   0.036   0.001  0.0954  0.0368    0.068660        0.01292   \n",
       "10  0.0018       0       0       0       0  0.0092    0.000488              0   \n",
       "11  0.1278  0.1611  0.1196  0.0446  0.2127  0.0869    0.152900          0.154   \n",
       "12  0.0016       0       0       0       0  0.0082    0.000688              0   \n",
       "13  0.2087  0.0666  0.1484  0.4484  0.1392  0.2679    0.175600        0.07002   \n",
       "14  0.0769  0.0113  0.1484  0.0248  0.0477  0.1984    0.074200        0.01188   \n",
       "15  0.2093  0.0666  0.1484  0.4514  0.1382  0.2689    0.175400        0.07031   \n",
       "16  0.1158   0.146  0.1643  0.0258  0.0517  0.1994    0.085240          0.146   \n",
       "17  0.1502  0.2731  0.1787  0.0248  0.0487  0.1973    0.092810          0.259   \n",
       "18  0.5072  0.5666  0.4049  0.7153  0.2495  0.5501    0.350700         0.5434   \n",
       "19  0.5084  0.5658  0.4078  0.7153  0.2535  0.5511    0.352200         0.5447   \n",
       "20  0.3423  0.3442  0.2061   0.629   0.171   0.317    0.232300         0.3372   \n",
       "21  0.2081  0.0666  0.1455  0.4504  0.1372  0.2669    0.175000        0.07013   \n",
       "22  0.0978  0.0877  0.1542  0.0258  0.0467  0.1984    0.079530        0.07788   \n",
       "23  0.3512  0.3434  0.2147   0.631  0.1849  0.3415    0.239900         0.3351   \n",
       "24  0.0002       0       0       0       0   0.001    0.000028              0   \n",
       "25  0.5076  0.5582  0.4078  0.7212  0.2485  0.5562    0.351100         0.5294   \n",
       "26  0.2131  0.0666  0.1527  0.4573  0.1392  0.2781    0.177000        0.07051   \n",
       "27  0.0002       0       0       0       0   0.001    0.000024              0   \n",
       "28  0.5128  0.5688  0.4078  0.7252  0.2535  0.5593    0.355300         0.5438   \n",
       "29  0.1170  0.0416  0.1816   0.125  0.0636  0.2198    0.102200        0.04336   \n",
       "30                                                    0.099750         0.1372   \n",
       "31  0.2592   0.503  0.2277  0.1339  0.0755  0.2699    0.146000         0.4722   \n",
       "32  0.4133  0.6067  0.3501  0.1677  0.3827  0.4816    0.379800         0.6069   \n",
       "33  0.4778  0.5877  0.4193  0.5685  0.2455  0.5164    0.336600         0.5581   \n",
       "34  0.1274  0.2231  0.1556  0.0258  0.0358  0.1769    0.082390         0.2039   \n",
       "35  0.1114  0.1649  0.1527  0.0258  0.0358  0.1759    0.078440         0.1524   \n",
       "36  0.2470  0.1936  0.1859  0.4365  0.1561  0.2607    0.179700          0.163   \n",
       "37  0.2438  0.1936   0.183  0.4355  0.1571  0.2464    0.177500         0.1652   \n",
       "38  0.4449  0.4637   0.379  0.6577  0.2247  0.4734    0.308400         0.4227   \n",
       "39  0.2388  0.1732   0.183  0.4375  0.1571  0.2464    0.177300         0.1521   \n",
       "40  0.2470   0.202  0.1844  0.4375  0.1581  0.2474    0.178500          0.171   \n",
       "41  0.0515  0.0023  0.1441   0.002  0.0298  0.1258    0.056800       0.005844   \n",
       "42  0.0932   0.087  0.1513  0.0238  0.0358  0.1912    0.074130          0.083   \n",
       "43  0.2732  0.4531  0.3602  0.2133  0.0875  0.2209    0.182000         0.4225   \n",
       "44  0.0861  0.0749  0.1542  0.0208  0.0348  0.1728    0.070410        0.07095   \n",
       "45  0.0875  0.0915  0.1599  0.0149  0.0358  0.1585    0.070020        0.09463   \n",
       "46  0.2360  0.1974   0.098  0.5883  0.0477  0.2168    0.123700         0.1689   \n",
       "47  0.3526  0.3026   0.366  0.4613  0.2545  0.3998    0.324900         0.2839   \n",
       "48  0.4633  0.4508  0.5058  0.6756  0.2624  0.4376    0.378900         0.4165   \n",
       "49  0.2304  0.3094  0.2277  0.1012  0.2167   0.273    0.220400         0.2922   \n",
       "50  0.2554  0.3782  0.1628  0.2718   0.163  0.2331    0.184800         0.3529   \n",
       "51  0.1542  0.1331  0.1297  0.1736    0.16  0.1738    0.154600         0.1357   \n",
       "52  0.0006       0       0       0   0.001   0.002    0.000442              0   \n",
       "53  0.2105  0.1233  0.2104  0.3601  0.2316  0.1524    0.225300         0.1433   \n",
       "54  0.0004  0.0008  0.0014       0       0       0    0.000150       0.001091   \n",
       "55  0.0146       0       0  0.0347   0.001  0.0378    0.007360      0.0005567   \n",
       "56  0.0020       0       0       0       0  0.0102    0.001856      0.0001231   \n",
       "\n",
       "   gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0         0.04429        0.02202          0.115        0.02818        0.04568   \n",
       "1           0.198         0.2216      0.0007356         0.2341         0.3069   \n",
       "2           0.431         0.3278         0.1951          0.544         0.5135   \n",
       "3          0.4192          0.329          0.247         0.5298         0.4872   \n",
       "4          0.4168          0.329         0.1982         0.5277         0.4867   \n",
       "5          0.4634         0.4303         0.3666         0.6307         0.5571   \n",
       "6          0.4187         0.3334         0.2185         0.5268         0.4873   \n",
       "7          0.1269         0.1672        0.02936         0.1942         0.2201   \n",
       "8               0              0              0              0      8.791e-06   \n",
       "9         0.03952        0.07701      0.0002781         0.1453        0.08646   \n",
       "10              0              0              0              0              0   \n",
       "11          0.122         0.1499        0.04842         0.2115         0.1816   \n",
       "12              0              0              0              0      3.519e-05   \n",
       "13          0.145         0.1704         0.4299         0.1525         0.1437   \n",
       "14         0.1943        0.02975        0.02762        0.05772        0.03227   \n",
       "15         0.1448         0.1706         0.4294         0.1516         0.1436   \n",
       "16         0.2031        0.02983         0.0275        0.05769        0.03472   \n",
       "17         0.2066        0.03542        0.02751        0.05775        0.03343   \n",
       "18         0.4241         0.2798         0.7039         0.2775         0.2239   \n",
       "19          0.426         0.2812         0.7037         0.2773         0.2259   \n",
       "20         0.1854         0.2299         0.6132         0.1726         0.1652   \n",
       "21         0.1448         0.1686         0.4282          0.152         0.1432   \n",
       "22         0.2002        0.02979         0.0276        0.05774        0.03258   \n",
       "23         0.1923         0.2368         0.6106         0.1728         0.1737   \n",
       "24              0              0              0              0              0   \n",
       "25         0.4264         0.2848         0.7085         0.2774         0.2238   \n",
       "26         0.1475         0.1751         0.4321         0.1519          0.144   \n",
       "27              0              0              0              0              0   \n",
       "28          0.432         0.2897         0.7126         0.2787         0.2275   \n",
       "29         0.2238        0.04196         0.1291          0.104        0.04924   \n",
       "30         0.1658         0.0592         0.3329        0.02273        0.05033   \n",
       "31         0.2531        0.06936         0.1345         0.1212        0.05948   \n",
       "32         0.3621         0.2732         0.1677         0.4981         0.3565   \n",
       "33         0.4297         0.2666         0.5737         0.2517          0.227   \n",
       "34         0.1942        0.04102        0.02855        0.05694        0.02692   \n",
       "35         0.1903        0.04086        0.02849        0.05763        0.02637   \n",
       "36          0.176         0.1271         0.4309        0.09591         0.1503   \n",
       "37         0.1732         0.1222         0.4297        0.09552         0.1497   \n",
       "38         0.4115         0.2003         0.6675         0.2016         0.2025   \n",
       "39         0.1734         0.1231          0.431        0.09594         0.1502   \n",
       "40          0.174         0.1233          0.431        0.09591         0.1501   \n",
       "41         0.1838        0.01864      0.0007289        0.05548        0.01931   \n",
       "42         0.1859        0.03685         0.0264         0.0563        0.02651   \n",
       "43         0.4465         0.1041         0.2335         0.1093         0.0742   \n",
       "44         0.1855        0.02941         0.0252         0.0553         0.0232   \n",
       "45         0.1874        0.03398        0.01873        0.05609        0.02334   \n",
       "46         0.1012        0.06619         0.5975        0.04886        0.05134   \n",
       "47         0.4204         0.2495         0.4401         0.3008         0.2832   \n",
       "48         0.5574         0.2534         0.6696         0.3141           0.29   \n",
       "49         0.2158         0.2014        0.09327            0.2         0.2301   \n",
       "50         0.1444         0.1724         0.2477         0.1401         0.1604   \n",
       "51         0.1297         0.1608          0.152          0.119         0.1582   \n",
       "52      0.0001158      0.0006949              0              0      0.0002466   \n",
       "53         0.1705          0.288         0.3506         0.1732         0.2561   \n",
       "54      0.0001759              0              0              0      6.383e-05   \n",
       "55         0.0011      0.0008945        0.03179      0.0001858      0.0008907   \n",
       "56              0      9.934e-05       0.000598              0      8.814e-06   \n",
       "\n",
       "   gnomADe_OTH_AF  gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0         0.05033        0.100200    0.05381        0.06202        0.01535   \n",
       "1          0.2428        0.077920     0.2014        0.05371         0.1327   \n",
       "2          0.4431        0.436200      0.396         0.1671         0.5318   \n",
       "3          0.4319        0.428800     0.3943         0.2123         0.4232   \n",
       "4          0.4275        0.426900     0.3915         0.2115         0.4242   \n",
       "5          0.5117        0.512000     0.5179         0.4582         0.4112   \n",
       "6          0.4257        0.430500     0.4062         0.2613         0.4211   \n",
       "7          0.1914        0.163500     0.1979         0.1951         0.1505   \n",
       "8       0.0001629        0.005128  0.0001445              0              0   \n",
       "9          0.0838        0.044350     0.0646        0.01532         0.2061   \n",
       "10       0.001011        0.002595  0.0001183              0              0   \n",
       "11         0.1621        0.101700     0.1648         0.1526         0.1316   \n",
       "12              0        0.005520  0.0002433      4.831e-05              0   \n",
       "13         0.1577        0.253000     0.1413        0.07479         0.4736   \n",
       "14        0.05908        0.181000    0.04453        0.01337        0.04934   \n",
       "15         0.1587        0.253200     0.1416        0.07537         0.4726   \n",
       "16        0.06653        0.182900    0.08327         0.1441        0.04934   \n",
       "17        0.06858        0.182100     0.1142         0.2553        0.04934   \n",
       "18         0.3055        0.514500     0.3603         0.5417         0.5626   \n",
       "19         0.3083        0.515300     0.3617         0.5414         0.5604   \n",
       "20         0.2081        0.303400     0.2396         0.3371         0.4813   \n",
       "21         0.1573        0.252100     0.1406        0.07529         0.4669   \n",
       "22         0.0637        0.181600    0.06355        0.07948        0.04945   \n",
       "23         0.2168        0.326300      0.245         0.3352         0.4856   \n",
       "24              0        0.000229  1.315e-05              0              0   \n",
       "25         0.3061        0.518500      0.357          0.528         0.5603   \n",
       "26         0.1594        0.257800     0.1424        0.07557         0.4714   \n",
       "27              0        0.000196  1.315e-05              0              0   \n",
       "28         0.3113        0.522900     0.3631         0.5402         0.5593   \n",
       "29        0.08419        0.194400    0.06855        0.04321        0.07269   \n",
       "30         0.1178        0.154700     0.1768           0.21         0.4255   \n",
       "31         0.1151        0.238900        0.2         0.4629           0.08   \n",
       "32         0.3488        0.453200     0.4381         0.6011         0.3051   \n",
       "33         0.2987        0.479600     0.3595         0.5524         0.5571   \n",
       "34        0.06223        0.168600    0.09388         0.2014        0.04276   \n",
       "35        0.05979        0.167200    0.07946         0.1521        0.04286   \n",
       "36         0.1672        0.231700     0.1683         0.1636         0.4714   \n",
       "37         0.1654        0.221700     0.1679         0.1654         0.4715   \n",
       "38         0.2687        0.430400      0.305          0.423         0.5463   \n",
       "39         0.1647        0.222500     0.1647         0.1536         0.4715   \n",
       "40         0.1649        0.222400       0.17         0.1717         0.4757   \n",
       "41        0.04699        0.128000    0.03257       0.006592        0.04276   \n",
       "42        0.05835        0.174700    0.05955        0.08205        0.04276   \n",
       "43          0.145        0.212900     0.2045         0.4158        0.09121   \n",
       "44        0.05554        0.169200    0.05353        0.07005              0   \n",
       "45        0.05666        0.152300     0.0595        0.09199              0   \n",
       "46        0.08445        0.188000     0.1112         0.1669         0.1272   \n",
       "47          0.295        0.372400     0.2952         0.2803          0.239   \n",
       "48         0.3326        0.409700     0.3565         0.4154         0.2451   \n",
       "49         0.2281        0.247000     0.2452         0.2932         0.3392   \n",
       "50         0.1885        0.223300     0.2162         0.3456         0.1107   \n",
       "51          0.166        0.184300     0.1472         0.1385         0.1107   \n",
       "52      0.0001633        0.002319  0.0001776              0              0   \n",
       "53         0.2411        0.154100     0.2156         0.1475         0.2259   \n",
       "54              0        0.000269  0.0004209       0.001425              0   \n",
       "55       0.005554        0.034860   0.003092       0.000725       0.009868   \n",
       "56       0.000163        0.014700  0.0004929      0.0001449              0   \n",
       "\n",
       "   gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0          0.0484        0.02046         0.1154        0.02903        0.01582   \n",
       "1          0.2132          0.212       0.001731         0.2301         0.1804   \n",
       "2          0.4337         0.3138         0.2117         0.5462         0.2595   \n",
       "3           0.419         0.3185         0.2627         0.5309         0.2532   \n",
       "4          0.4168         0.3179          0.215         0.5281         0.2532   \n",
       "5          0.5022         0.4231         0.3678          0.634         0.3829   \n",
       "6          0.4166          0.325         0.2359         0.5269         0.2484   \n",
       "7          0.1663         0.1802        0.03272         0.1984         0.2468   \n",
       "8               0              0              0              0              0   \n",
       "9         0.04894        0.07896      0.0009608         0.1516        0.08228   \n",
       "10              0              0              0              0              0   \n",
       "11         0.1452         0.1489        0.04582         0.2218         0.1242   \n",
       "12              0              0              0              0              0   \n",
       "13         0.1279         0.1702         0.4391         0.1628         0.1139   \n",
       "14         0.1514        0.02968        0.02372        0.05839        0.06646   \n",
       "15          0.128         0.1703         0.4404         0.1627         0.1139   \n",
       "16         0.1652        0.02968        0.02492        0.05836        0.08228   \n",
       "17          0.175        0.03602        0.02445         0.0588        0.08861   \n",
       "18         0.3812         0.2824          0.699         0.2849         0.2739   \n",
       "19         0.3841         0.2843         0.7002         0.2855         0.2753   \n",
       "20         0.1862         0.2284         0.6138         0.1817         0.1835   \n",
       "21         0.1268         0.1681         0.4397         0.1612         0.1139   \n",
       "22         0.1572        0.02967         0.0244        0.05881        0.07911   \n",
       "23         0.2001         0.2367         0.6135         0.1819         0.2057   \n",
       "24              0              0              0              0              0   \n",
       "25         0.3796         0.2877         0.7061          0.286         0.2753   \n",
       "26         0.1294         0.1744         0.4458         0.1632         0.1139   \n",
       "27              0              0              0              0              0   \n",
       "28         0.3872         0.2906         0.7088         0.2873         0.2816   \n",
       "29         0.1599        0.04493         0.1194         0.1014        0.04839   \n",
       "30         0.2221         0.1839         0.4479         0.1409         0.2077   \n",
       "31         0.2271        0.07147         0.1297         0.1211         0.1184   \n",
       "32           0.39         0.2769         0.1701         0.5046         0.3481   \n",
       "33         0.3902         0.2671         0.5723         0.2605         0.2785   \n",
       "34         0.1518        0.03604        0.02616        0.05751         0.0538   \n",
       "35         0.1452        0.03576        0.02623        0.05845        0.04747   \n",
       "36         0.1739         0.1308         0.4344         0.0981         0.1234   \n",
       "37         0.1712         0.1256         0.4321        0.09874         0.1203   \n",
       "38         0.3579         0.2031         0.6483         0.2037         0.1867   \n",
       "39         0.1701         0.1259         0.4332        0.09868         0.1203   \n",
       "40         0.1729          0.126         0.4341        0.09826         0.1203   \n",
       "41         0.1303        0.01701        0.00155        0.05668        0.02215   \n",
       "42         0.1389        0.03228        0.02414        0.05725         0.0443   \n",
       "43          0.313        0.09879         0.2111         0.1095        0.09177   \n",
       "44         0.1348        0.02592        0.02172        0.05701        0.02848   \n",
       "45         0.1388        0.03141        0.01619        0.05785        0.02848   \n",
       "46        0.09281        0.06167         0.5812        0.05353        0.01582   \n",
       "47         0.3271         0.2367         0.4286         0.2894         0.2437   \n",
       "48         0.4354         0.2409         0.6482         0.3006         0.2468   \n",
       "49         0.2302         0.1885         0.1011          0.197         0.2215   \n",
       "50         0.1815         0.1702         0.2567         0.1486         0.2342   \n",
       "51         0.1481         0.1544         0.1588         0.1215         0.1994   \n",
       "52      0.0001311       0.000288              0              0              0   \n",
       "53           0.19         0.2885         0.3402         0.1684           0.25   \n",
       "54       0.000131              0              0              0              0   \n",
       "55      0.0009823      0.0008661        0.03591              0        0.01911   \n",
       "56              0              0      0.0003854              0              0   \n",
       "\n",
       "   gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF    MAX_AF  MAX_AF_POPS  \\\n",
       "0         0.04731        0.05306         0.1146    0.1154  gnomADg EAS   \n",
       "1          0.3095         0.1864        0.07601    0.3095  gnomADg NFE   \n",
       "2          0.5179         0.3914         0.4359    0.5462  gnomADg FIN   \n",
       "3          0.4898          0.392         0.4301    0.5309  gnomADg FIN   \n",
       "4          0.4892         0.3901          0.424    0.5281  gnomADg FIN   \n",
       "5          0.5583         0.5226         0.5118     0.634  gnomADg FIN   \n",
       "6          0.4896         0.3922         0.4287    0.5269  gnomADg FIN   \n",
       "7          0.2238         0.1937         0.1545    0.2468  gnomADg MID   \n",
       "8               0              0       0.004547    0.0061          SAS   \n",
       "9         0.08845        0.06274        0.04122    0.2061  gnomADg AMI   \n",
       "10       1.47e-05              0        0.00352    0.0092          SAS   \n",
       "11         0.1829         0.1793        0.09298    0.2218  gnomADg FIN   \n",
       "12       2.94e-05              0       0.006849    0.0082          SAS   \n",
       "13         0.1454         0.1264         0.2562    0.4736  gnomADg AMI   \n",
       "14        0.03035        0.04019         0.1765    0.1984          SAS   \n",
       "15         0.1455          0.126         0.2568    0.4726  gnomADg AMI   \n",
       "16        0.03306        0.07122         0.1781    0.2031  gnomADe AMR   \n",
       "17        0.03148        0.09004         0.1777    0.2731          AFR   \n",
       "18         0.2233         0.3258         0.5153    0.7153          EAS   \n",
       "19         0.2254         0.3281         0.5157    0.7153          EAS   \n",
       "20         0.1664         0.2227         0.3061     0.629          EAS   \n",
       "21         0.1444         0.1254         0.2543    0.4669  gnomADg AMI   \n",
       "22        0.03061        0.05641         0.1766    0.2002  gnomADe AMR   \n",
       "23         0.1742          0.233          0.328     0.631          EAS   \n",
       "24              0              0      0.0004137     0.001          SAS   \n",
       "25         0.2232         0.3265         0.5202    0.7212          EAS   \n",
       "26         0.1457         0.1292         0.2649    0.4714  gnomADg AMI   \n",
       "27              0              0      0.0004142     0.001          SAS   \n",
       "28          0.227         0.3309         0.5207    0.7252          EAS   \n",
       "29        0.04785        0.05673         0.1891    0.2238  gnomADe AMR   \n",
       "30         0.1303          0.171         0.3094    0.4479  gnomADg EAS   \n",
       "31        0.05838         0.1643         0.2401     0.503          AFR   \n",
       "32         0.3705         0.3781         0.4592    0.6069  gnomADe AFR   \n",
       "33          0.229          0.328          0.484    0.5877          AFR   \n",
       "34        0.02601        0.07534         0.1608    0.2231          AFR   \n",
       "35        0.02517        0.06897         0.1624    0.1903  gnomADe AMR   \n",
       "36         0.1541         0.1686         0.2336    0.4714  gnomADg AMI   \n",
       "37         0.1537         0.1668         0.2237    0.4715  gnomADg AMI   \n",
       "38         0.2051           0.29         0.4305    0.6675  gnomADe EAS   \n",
       "39         0.1541         0.1654         0.2238    0.4715  gnomADg AMI   \n",
       "40         0.1541         0.1687         0.2251    0.4757  gnomADg AMI   \n",
       "41         0.0194        0.02874         0.1243    0.1838  gnomADe AMR   \n",
       "42        0.02558        0.04585         0.1672    0.1912          SAS   \n",
       "43        0.07399         0.1699         0.2132    0.4531          AFR   \n",
       "44        0.02214        0.03776         0.1618    0.1855  gnomADe AMR   \n",
       "45        0.02228        0.04159         0.1464    0.1874  gnomADe AMR   \n",
       "46        0.05249        0.09273           0.19    0.5975  gnomADe EAS   \n",
       "47         0.2855         0.2939         0.3849    0.4613          EAS   \n",
       "48         0.2926          0.338         0.4331    0.6756          EAS   \n",
       "49         0.2399          0.239         0.2414    0.3392  gnomADg AMI   \n",
       "50         0.1547         0.2173         0.2385    0.3782          AFR   \n",
       "51         0.1523         0.1538         0.1866    0.1994  gnomADg MID   \n",
       "52      0.0002499       0.000478       0.001247  0.002319  gnomADe SAS   \n",
       "53         0.2598          0.233         0.1647    0.3601          EAS   \n",
       "54      4.413e-05              0              0  0.001425  gnomADg AFR   \n",
       "55      0.0006619       0.005258        0.03425    0.0378          SAS   \n",
       "56       1.47e-05              0        0.01365    0.0147  gnomADe SAS   \n",
       "\n",
       "           BIOTYPE     EXON INTRON Protein Position and Amino Acid   Codons  \\\n",
       "0   protein coding      1/3                                   V32M  Gtg/Atg   \n",
       "1   protein coding   61/105                                 Q2958R  cAa/cGa   \n",
       "2   protein coding    10/20                                  F353L  ttC/ttG   \n",
       "3   protein coding    11/20                                  S416N  aGc/aAc   \n",
       "4   protein coding    11/20                                  S432N  aGt/aAt   \n",
       "5   protein coding    11/20                                  S528R  agC/agG   \n",
       "6   protein coding    17/20                                  P860T  Cca/Aca   \n",
       "7   protein coding     4/14                                  L366P  cTg/cCg   \n",
       "8   protein coding     3/39                                   L24I  Cta/Ata   \n",
       "9   protein coding     8/15                                  I293V  Ata/Gta   \n",
       "10  protein coding     1/18                                   R24H  cGc/cAc   \n",
       "11  protein coding    29/29                                 E4181K  Gaa/Aaa   \n",
       "12  protein coding    20/29                                 E1012K  Gag/Aag   \n",
       "13  protein coding  186/191                                R25721H  cGc/cAc   \n",
       "14  protein coding  181/191                                R24098H  cGc/cAc   \n",
       "15  protein coding  178/191                                R23665H  cGc/cAc   \n",
       "16  protein coding  174/191                                I23180T  aTt/aCt   \n",
       "17  protein coding  166/191                                V21838M  Gtg/Atg   \n",
       "18  protein coding  158/191                                I20523T  aTc/aCc   \n",
       "19  protein coding  154/191                                I18902V  Ata/Gta   \n",
       "20  protein coding  154/191                                T17748M  aCg/aTg   \n",
       "21  protein coding  154/191                                R16074C  Cgc/Tgc   \n",
       "22  protein coding  154/191                                R15125H  cGt/cAt   \n",
       "23  protein coding  146/191                                V13486I  Gtc/Atc   \n",
       "24  protein coding  145/191                                V13410I  Gta/Ata   \n",
       "25  protein coding  136/191                                T12530I  aCt/aTt   \n",
       "26  protein coding  129/191                                P10989L  cCa/cTa   \n",
       "27  protein coding  122/191                                L10219P  cTt/cCt   \n",
       "28  protein coding  117/191                                 N9828D  Aat/Gat   \n",
       "29  protein coding  198/363                                L12918S  tTg/tCg   \n",
       "30  protein coding  173/363                                L12219V  Ttg/Gtg   \n",
       "31  protein coding      9/9                                  E234K  Gag/Aag   \n",
       "32  protein coding  120/312                                 G9378R  Gga/Aga   \n",
       "33  protein coding  116/312                                 I9278V  Att/Gtt   \n",
       "34  protein coding   99/312                                 I8474T  aTt/aCt   \n",
       "35  protein coding   89/312                                 P7650L  cCg/cTg   \n",
       "36  protein coding   88/312                                 N7559S  aAt/aGt   \n",
       "37  protein coding   84/312                                 S7181N  aGc/aAc   \n",
       "38  protein coding   84/312                                 A7111E  gCg/gAg   \n",
       "39  protein coding   81/312                                 E6900A  gAa/gCa   \n",
       "40  protein coding   74/312                                 D6218H  Gat/Cat   \n",
       "41  protein coding   69/312                                 A5710V  gCa/gTa   \n",
       "42  protein coding   63/312                                 S5190N  aGt/aAt   \n",
       "43  protein coding   45/191                                 P3637S  Cca/Tca   \n",
       "44  protein coding   38/191                                 K3108R  aAg/aGg   \n",
       "45  protein coding   29/191                                 G2346S  Ggc/Agc   \n",
       "46  protein coding    6/191                                  R328C  Cgt/Tgt   \n",
       "47  protein coding     6/12                                  K168N  aaA/aaT   \n",
       "48  protein coding     7/12                                  M179T  aTg/aCg   \n",
       "49  protein coding    12/28                                  H558R  cAc/cGc   \n",
       "50  protein coding      2/2                                  R203C  Cgc/Tgc   \n",
       "51  protein coding      2/2                                  Q193*  Cag/Tag   \n",
       "52  protein coding    38/39                                 P1757L  cCc/cTc   \n",
       "53  protein coding    25/39                                 P1119R  cCt/cGt   \n",
       "54  protein coding     2/27                                   R40Q  cGg/cAg   \n",
       "55  protein coding     3/16                                  G100S  Ggc/Agc   \n",
       "56  protein coding    15/50                                  R638Q  cGg/cAg   \n",
       "\n",
       "   STRAND                                             PUBMED  \n",
       "0      -1  18513389,19131662,19330901,19559392,20577119,2...  \n",
       "1       1  25741868,24033266,23861362,23757202,11157710,1...  \n",
       "2       1  24033266,23861362,18006477,22286171,22892539,3...  \n",
       "3       1  24033266,23861362,23164068,20395239,18006477,2...  \n",
       "4       1  24033266,23861362,16385451,28445147,18006477,2...  \n",
       "5       1  24033266,23861362,18006477,22286171,22892539,3...  \n",
       "6       1  24033266,23861362,18006477,22286171,22892539,3...  \n",
       "7      -1  25741868,24033266,23861362,17556197,17521752,2...  \n",
       "8      -1                                           24033266  \n",
       "9       1  25741868,24033266,23861362,20829228,18678517,2...  \n",
       "10      1                                                     \n",
       "11     -1  25741868,27589735,28659664,16642433,22328972,2...  \n",
       "12     -1                                                     \n",
       "13     -1  25741868,24033266,18414213,23861362,23757202,2...  \n",
       "14     -1                25741868,24033266,23861362,29897334  \n",
       "15     -1  25741868,24033266,18414213,23861362,27681629,3...  \n",
       "16     -1                25741868,24033266,23861362,29897334  \n",
       "17     -1                25741868,24033266,23861362,29897334  \n",
       "18     -1  25741868,24033266,18414213,23861362,27681629,2...  \n",
       "19     -1  25741868,24033266,18414213,23861362,23757202,2...  \n",
       "20     -1  25741868,24033266,18414213,23861362,23757202,2...  \n",
       "21     -1  25741868,24033266,18414213,23861362,23757202,2...  \n",
       "22     -1                25741868,24033266,23861362,29897334  \n",
       "23     -1  25741868,24033266,18414213,23861362,27681629,2...  \n",
       "24     -1                                                     \n",
       "25     -1  25741868,24033266,18414213,23861362,27681629,2...  \n",
       "26     -1       25741868,24033266,18414213,27681629,29897334  \n",
       "27     -1                                                     \n",
       "28     -1  25741868,24033266,18414213,23861362,27681629,2...  \n",
       "29     -1                                                     \n",
       "30     -1                                           24033266  \n",
       "31     -1                                  23861362,34427245  \n",
       "32     -1  25741868,24033266,18414213,23861362,27681629,2...  \n",
       "33     -1  25741868,24033266,18414213,23861362,27681629,2...  \n",
       "34     -1                25741868,24033266,18414213,23861362  \n",
       "35     -1       25741868,24033266,18414213,23861362,29897334  \n",
       "36     -1                25741868,24033266,18414213,23861362  \n",
       "37     -1  25741868,24033266,18414213,23861362,27681629,2...  \n",
       "38     -1  25741868,24033266,27182706,18414213,23861362,2...  \n",
       "39     -1                25741868,24033266,18414213,23861362  \n",
       "40     -1       25741868,24033266,18414213,23861362,27681629  \n",
       "41     -1       25741868,24033266,23861362,19268276,29897334  \n",
       "42     -1                25741868,24033266,23861362,29897334  \n",
       "43     -1  25741868,24033266,23861362,31872004,31827228,2...  \n",
       "44     -1                         25741868,24033266,23861362  \n",
       "45     -1       25741868,24033266,18414213,23861362,29897334  \n",
       "46     -1       25741868,24033266,18414213,23861362,31641117  \n",
       "47      1       25741868,24033266,18414213,23861362,23757202  \n",
       "48      1  25741868,24033266,18414213,23861362,23757202,1...  \n",
       "49     -1  25741868,21056700,24033266,27182706,20850564,2...  \n",
       "50     -1                                           28911203  \n",
       "51     -1                                  18300947,28911203  \n",
       "52     -1                                           24033266  \n",
       "53     -1                25741868,24033266,23861362,35284542  \n",
       "54      1                                                     \n",
       "55     -1  25741868,24033266,24938736,27930701,23778007,2...  \n",
       "56     -1                                  15297675,25215231  "
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "condition_filter = merged_3[merged_3['Gene_Match'] != 'No']\n",
    "\n",
    "consequence_filter = condition_filter[condition_filter['Consequence_score'].apply(lambda x: eval(x) >= 6/10)]\n",
    "\n",
    "consequence_filter['DP'] = consequence_filter['DP'].astype('int64')\n",
    "\n",
    "dp_filter = consequence_filter[consequence_filter['DP'] >= 15]\n",
    "\n",
    "dp_filter['gnomADe_AF'] = dp_filter['gnomADe_AF'].replace('', '0').astype(float)\n",
    "\n",
    "gnomADe_AF_filter = dp_filter[dp_filter['gnomADe_AF'] <= 0.6]\n",
    "\n",
    "gnomADe_AF_filter['gnomADe_SAS_AF'] = gnomADe_AF_filter['gnomADe_SAS_AF'].replace('', '0').astype(float)\n",
    "\n",
    "gnomADe_SAS_AF_filter = gnomADe_AF_filter[gnomADe_AF_filter['gnomADe_SAS_AF'] <= 0.6]\n",
    "\n",
    "drop_duplicates_filter = gnomADe_SAS_AF_filter.drop_duplicates(subset=['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT'])\n",
    "\n",
    "drop_duplicates_filter = drop_duplicates_filter.set_index(['Gene Name', 'Gene_Match', 'rsID', 'Literature', 'CHROM', 'POS', 'REF',\n",
    "       'ALT', 'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT',\n",
    "       'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "       'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)',\n",
    "       'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR',\n",
    "       'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF',\n",
    "       'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF',\n",
    "       'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF',\n",
    "       'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF',\n",
    "       'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF',\n",
    "       'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF',\n",
    "       'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']).apply(lambda x: x.str.split('; ').explode()).reset_index()\n",
    "\n",
    "drop_duplicates_filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "ee3e0549",
   "metadata": {},
   "outputs": [],
   "source": [
    "drop_duplicates_filter.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL40_Cardiac.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ae55176",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c91fa8c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "281598ba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebb44a03",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50a9cf00",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee5e6993",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f1a3974",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "724bdb2c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "185d94c9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d4c385d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "88e1a3c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>152</td>\n",
       "      <td>149</td>\n",
       "      <td>149</td>\n",
       "      <td>104</td>\n",
       "      <td>45</td>\n",
       "      <td>30.2%</td>\n",
       "      <td>5.6076E-16</td>\n",
       "      <td>80</td>\n",
       "      <td>24</td>\n",
       "      <td>38</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>152</td>\n",
       "      <td>149</td>\n",
       "      <td>149</td>\n",
       "      <td>104</td>\n",
       "      <td>45</td>\n",
       "      <td>30.2%</td>\n",
       "      <td>5.6076E-16</td>\n",
       "      <td>80</td>\n",
       "      <td>24</td>\n",
       "      <td>38</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Name Gene_Match         rsID Literature CHROM    POS Literature REF  \\\n",
       "0     OR4F5         No  rs201219564         No  chr1  69270         No   A   \n",
       "1     OR4F5         No  rs201219564         No  chr1  69270         No   A   \n",
       "\n",
       "  ALT      Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0   G  Heterozygous         synonymous variant              3/10       LOW   \n",
       "1   G  Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "\n",
       "   IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0           2.5                                                     \n",
       "1           1.5                                                     \n",
       "\n",
       "               HGVSc HGVSc (Transcript)              HGVSp HGVSp (Transcript)  \\\n",
       "0  ENST00000641515.2           c.243A>G  ENSP00000493376.2         p.Ser81%3D   \n",
       "1                                  None                                  None   \n",
       "\n",
       "    GT   GQ  SDP   DP   RD  AD   FREQ        PVAL RDF RDR ADF ADR SIFT  \\\n",
       "0  0/1  152  149  149  104  45  30.2%  5.6076E-16  80  24  38   7        \n",
       "1  0/1  152  149  149  104  45  30.2%  5.6076E-16  80  24  38   7        \n",
       "\n",
       "  PolyPhen AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                     0.838         0.3591   \n",
       "1                                                     0.838         0.3591   \n",
       "\n",
       "  gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0         0.7932         0.8482         0.9984         0.8821         0.9146   \n",
       "1         0.7932         0.8482         0.9984         0.8821         0.9146   \n",
       "\n",
       "  gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0          0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "1          0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "\n",
       "  gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0         0.6945         0.8033         0.9961         0.9294         0.7857   \n",
       "1         0.6945         0.8033         0.9961         0.9294         0.7857   \n",
       "\n",
       "  gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0         0.8798         0.6087         0.8855  0.9984  gnomADe EAS   \n",
       "1         0.8798         0.6087         0.8855  0.9984  gnomADe EAS   \n",
       "\n",
       "           BIOTYPE EXON INTRON Protein Position and Amino Acid   Codons  \\\n",
       "0   protein coding  3/3                                    S81  tcA/tcG   \n",
       "1  TF binding site                                         nan            \n",
       "\n",
       "  STRAND PUBMED  \n",
       "0      1         \n",
       "1                "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3[merged_3['POS'] == 69270]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b890148c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "7c93c7e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>152</td>\n",
       "      <td>149</td>\n",
       "      <td>149</td>\n",
       "      <td>104</td>\n",
       "      <td>45</td>\n",
       "      <td>30.2%</td>\n",
       "      <td>5.6076E-16</td>\n",
       "      <td>80</td>\n",
       "      <td>24</td>\n",
       "      <td>38</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>152</td>\n",
       "      <td>149</td>\n",
       "      <td>149</td>\n",
       "      <td>104</td>\n",
       "      <td>45</td>\n",
       "      <td>30.2%</td>\n",
       "      <td>5.6076E-16</td>\n",
       "      <td>80</td>\n",
       "      <td>24</td>\n",
       "      <td>38</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>178</td>\n",
       "      <td>178</td>\n",
       "      <td>0</td>\n",
       "      <td>176</td>\n",
       "      <td>98.88%</td>\n",
       "      <td>2.565E-105</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>144</td>\n",
       "      <td>32</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>144</td>\n",
       "      <td>186</td>\n",
       "      <td>186</td>\n",
       "      <td>142</td>\n",
       "      <td>44</td>\n",
       "      <td>23.66%</td>\n",
       "      <td>3.1648E-15</td>\n",
       "      <td>81</td>\n",
       "      <td>61</td>\n",
       "      <td>25</td>\n",
       "      <td>19</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>99</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>56</td>\n",
       "      <td>29</td>\n",
       "      <td>34.12%</td>\n",
       "      <td>1.0363E-10</td>\n",
       "      <td>40</td>\n",
       "      <td>16</td>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Name Gene_Match         rsID Literature CHROM     POS Literature REF  \\\n",
       "0     OR4F5         No  rs201219564         No  chr1   69270         No   A   \n",
       "1     OR4F5         No  rs201219564         No  chr1   69270         No   A   \n",
       "2     OR4F5         No    rs2691305         No  chr1   69511         No   A   \n",
       "3     OR4F5         No  rs200676709         No  chr1   69897         No   T   \n",
       "4                   No            .         No  chr1  451290         No   A   \n",
       "\n",
       "  ALT      Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0   G  Heterozygous         synonymous variant              3/10       LOW   \n",
       "1   G  Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2   G    Homozygous           missense variant              7/10  MODERATE   \n",
       "3   C  Heterozygous         synonymous variant              3/10       LOW   \n",
       "4   C  Heterozygous           missense variant              7/10  MODERATE   \n",
       "\n",
       "   IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0           2.5                                                     \n",
       "1           1.5                                                     \n",
       "2           5.0                                                     \n",
       "3           2.5                                                     \n",
       "4           5.0                                                     \n",
       "\n",
       "               HGVSc HGVSc (Transcript)              HGVSp HGVSp (Transcript)  \\\n",
       "0  ENST00000641515.2           c.243A>G  ENSP00000493376.2         p.Ser81%3D   \n",
       "1                                  None                                  None   \n",
       "2  ENST00000641515.2           c.484A>G  ENSP00000493376.2        p.Thr162Ala   \n",
       "3  ENST00000641515.2           c.870T>C  ENSP00000493376.2        p.Ser290%3D   \n",
       "4  ENST00000426406.4           c.389T>G  ENSP00000409316.1        p.Leu130Arg   \n",
       "\n",
       "    GT   GQ  SDP   DP   RD   AD    FREQ        PVAL RDF RDR  ADF ADR  \\\n",
       "0  0/1  152  149  149  104   45   30.2%  5.6076E-16  80  24   38   7   \n",
       "1  0/1  152  149  149  104   45   30.2%  5.6076E-16  80  24   38   7   \n",
       "2  1/1  255  178  178    0  176  98.88%  2.565E-105   0   0  144  32   \n",
       "3  0/1  144  186  186  142   44  23.66%  3.1648E-15  81  61   25  19   \n",
       "4  0/1   99   85   85   56   29  34.12%  1.0363E-10  40  16   25   4   \n",
       "\n",
       "                               SIFT       PolyPhen      AF AFR_AF  AMR_AF  \\\n",
       "0                                                                           \n",
       "1                                                                           \n",
       "2                   tolerated(0.92)      benign(0)                          \n",
       "3                                                   0.6881  0.407  0.6254   \n",
       "4  deleterious low confidence(0.01)  benign(0.336)                          \n",
       "\n",
       "  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                             0.838         0.3591         0.7932   \n",
       "1                             0.838         0.3591         0.7932   \n",
       "2                            0.9497         0.6075         0.9514   \n",
       "3  0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "4                                                                   \n",
       "\n",
       "  gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0         0.8482         0.9984         0.8821         0.9146          0.868   \n",
       "1         0.8482         0.9984         0.8821         0.9146          0.868   \n",
       "2         0.9767         0.9994         0.9916         0.9726         0.9506   \n",
       "3         0.8131         0.7793         0.8652         0.8438         0.7932   \n",
       "4                                                                              \n",
       "\n",
       "  gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0         0.9005     0.6291         0.3544         0.8158         0.6945   \n",
       "1         0.9005     0.6291         0.3544         0.8158         0.6945   \n",
       "2         0.9854      0.846         0.5948          0.998         0.8951   \n",
       "3         0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "4                                                                          \n",
       "\n",
       "  gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0         0.8033         0.9961         0.9294         0.7857         0.8798   \n",
       "1         0.8033         0.9961         0.9294         0.7857         0.8798   \n",
       "2         0.9784         0.9998         0.9907            0.9         0.9674   \n",
       "3         0.6714         0.6047         0.6987         0.6441         0.6751   \n",
       "4                                                                              \n",
       "\n",
       "  gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS          BIOTYPE EXON  \\\n",
       "0         0.6087         0.8855  0.9984  gnomADe EAS   protein coding  3/3   \n",
       "1         0.6087         0.8855  0.9984  gnomADe EAS  TF binding site        \n",
       "2         0.8624         0.9772  0.9998  gnomADg EAS   protein coding  3/3   \n",
       "3         0.4659          0.622   0.876          EAS   protein coding  3/3   \n",
       "4                                                      protein coding  1/1   \n",
       "\n",
       "  INTRON Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                    S81  tcA/tcG      1         \n",
       "1                                    nan                         \n",
       "2                                  T162A  Aca/Gca      1         \n",
       "3                                   S290  tcT/tcC      1         \n",
       "4                                  L130R  cTc/cGc     -1         "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged  = merged_3.head()\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "c8ebe699",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant, regulatory region variant</td>\n",
       "      <td>3/10, 2/10</td>\n",
       "      <td>LOW, MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2,</td>\n",
       "      <td>c.243A&gt;G, None</td>\n",
       "      <td>ENSP00000493376.2,</td>\n",
       "      <td>p.Ser81%3D, None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>152</td>\n",
       "      <td>149</td>\n",
       "      <td>149</td>\n",
       "      <td>104</td>\n",
       "      <td>45</td>\n",
       "      <td>30.2%</td>\n",
       "      <td>5.6076E-16</td>\n",
       "      <td>80</td>\n",
       "      <td>24</td>\n",
       "      <td>38</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.6945</td>\n",
       "      <td>0.8033</td>\n",
       "      <td>0.9961</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.6087</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding, TF binding site</td>\n",
       "      <td>3/3,</td>\n",
       "      <td></td>\n",
       "      <td>S81, nan</td>\n",
       "      <td>tcA/tcG,</td>\n",
       "      <td>1,</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>178</td>\n",
       "      <td>178</td>\n",
       "      <td>0</td>\n",
       "      <td>176</td>\n",
       "      <td>98.88%</td>\n",
       "      <td>2.565E-105</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>144</td>\n",
       "      <td>32</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>144</td>\n",
       "      <td>186</td>\n",
       "      <td>186</td>\n",
       "      <td>142</td>\n",
       "      <td>44</td>\n",
       "      <td>23.66%</td>\n",
       "      <td>3.1648E-15</td>\n",
       "      <td>81</td>\n",
       "      <td>61</td>\n",
       "      <td>25</td>\n",
       "      <td>19</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>99</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>56</td>\n",
       "      <td>29</td>\n",
       "      <td>34.12%</td>\n",
       "      <td>1.0363E-10</td>\n",
       "      <td>40</td>\n",
       "      <td>16</td>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS         rsID Gene Name Gene_Match Literature Literature REF  \\\n",
       "0  chr1   69270  rs201219564     OR4F5         No         No         No   A   \n",
       "1  chr1   69511    rs2691305     OR4F5         No         No         No   A   \n",
       "2  chr1   69897  rs200676709     OR4F5         No         No         No   T   \n",
       "3  chr1  451290            .                   No         No         No   A   \n",
       "\n",
       "  ALT      Zygosity                                    Consequence  \\\n",
       "0   G  Heterozygous  synonymous variant, regulatory region variant   \n",
       "1   G    Homozygous                               missense variant   \n",
       "2   C  Heterozygous                             synonymous variant   \n",
       "3   C  Heterozygous                               missense variant   \n",
       "\n",
       "  Consequence_score         IMPACT ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0        3/10, 2/10  LOW, MODIFIER                                             \n",
       "1              7/10       MODERATE                                             \n",
       "2              3/10            LOW                                             \n",
       "3              7/10       MODERATE                                             \n",
       "\n",
       "  ClinVar                HGVSc HGVSc (Transcript)                HGVSp  \\\n",
       "0          ENST00000641515.2,      c.243A>G, None  ENSP00000493376.2,    \n",
       "1            ENST00000641515.2           c.484A>G    ENSP00000493376.2   \n",
       "2            ENST00000641515.2           c.870T>C    ENSP00000493376.2   \n",
       "3            ENST00000426406.4           c.389T>G    ENSP00000409316.1   \n",
       "\n",
       "  HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ        PVAL RDF  \\\n",
       "0   p.Ser81%3D, None  0/1  152  149  149  104   45   30.2%  5.6076E-16  80   \n",
       "1        p.Thr162Ala  1/1  255  178  178    0  176  98.88%  2.565E-105   0   \n",
       "2        p.Ser290%3D  0/1  144  186  186  142   44  23.66%  3.1648E-15  81   \n",
       "3        p.Leu130Arg  0/1   99   85   85   56   29  34.12%  1.0363E-10  40   \n",
       "\n",
       "  RDR  ADF ADR                              SIFT       PolyPhen      AF  \\\n",
       "0  24   38   7                                                            \n",
       "1   0  144  32                   tolerated(0.92)      benign(0)           \n",
       "2  61   25  19                                                   0.6881   \n",
       "3  16   25   4  deleterious low confidence(0.01)  benign(0.336)           \n",
       "\n",
       "  AFR_AF  AMR_AF EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                            0.838         0.3591   \n",
       "1                                           0.9497         0.6075   \n",
       "2  0.407  0.6254  0.876  0.7942  0.8098     0.7209          0.307   \n",
       "3                                                                   \n",
       "\n",
       "  gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0         0.7932         0.8482         0.9984         0.8821         0.9146   \n",
       "1         0.9514         0.9767         0.9994         0.9916         0.9726   \n",
       "2         0.6749         0.8131         0.7793         0.8652         0.8438   \n",
       "3                                                                              \n",
       "\n",
       "  gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0          0.868         0.9005     0.6291         0.3544         0.8158   \n",
       "1         0.9506         0.9854      0.846         0.5948          0.998   \n",
       "2         0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "3                                                                          \n",
       "\n",
       "  gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0         0.6945         0.8033         0.9961         0.9294         0.7857   \n",
       "1         0.8951         0.9784         0.9998         0.9907            0.9   \n",
       "2         0.4782         0.6714         0.6047         0.6987         0.6441   \n",
       "3                                                                              \n",
       "\n",
       "  gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0         0.8798         0.6087         0.8855  0.9984  gnomADe EAS   \n",
       "1         0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "2         0.6751         0.4659          0.622   0.876          EAS   \n",
       "3                                                                     \n",
       "\n",
       "                           BIOTYPE   EXON INTRON  \\\n",
       "0  protein coding, TF binding site  3/3,           \n",
       "1                   protein coding    3/3          \n",
       "2                   protein coding    3/3          \n",
       "3                   protein coding    1/1          \n",
       "\n",
       "  Protein Position and Amino Acid     Codons STRAND PUBMED  \n",
       "0                        S81, nan  tcA/tcG,     1,          \n",
       "1                           T162A    Aca/Gca      1         \n",
       "2                            S290    tcT/tcC      1         \n",
       "3                           L130R    cTc/cGc     -1         "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_data = merged.groupby(['CHROM', 'POS', 'rsID']).agg(lambda x: ', '.join(x.unique())).reset_index()\n",
    "grouped_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "762b92f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ENST00000641515.2, '"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_data.HGVSc.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa8fd0af",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe1505fd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "897ee340",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "321a053e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d591aa96",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a211a595",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "7263e75c",
   "metadata": {},
   "source": [
    "## DataBase wise mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "100a5c53",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "4f0239d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Amino acid change</th>\n",
       "      <th>Uploaded_variant</th>\n",
       "      <th>Chr-Position-Ref-Alt</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF_ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Gene Info</th>\n",
       "      <th>Publication</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G551D</td>\n",
       "      <td>CFTR:p.Gly551Asp</td>\n",
       "      <td>7-117587806-G-A</td>\n",
       "      <td>7.0</td>\n",
       "      <td>117587806</td>\n",
       "      <td>G-A</td>\n",
       "      <td>rs75527207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.N1303K</td>\n",
       "      <td>CFTR:p.Asn1303Lys</td>\n",
       "      <td>7-117652877-C-G</td>\n",
       "      <td>7.0</td>\n",
       "      <td>117652877</td>\n",
       "      <td>C-G</td>\n",
       "      <td>rs80034486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R117H</td>\n",
       "      <td>CFTR:p.Arg117His</td>\n",
       "      <td>7-117710365-GA-AT</td>\n",
       "      <td>7.0</td>\n",
       "      <td>117710365</td>\n",
       "      <td>GA-AT</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G85E</td>\n",
       "      <td>CFTR:p.Gly85Glu</td>\n",
       "      <td>7-117509123-G-A</td>\n",
       "      <td>7.0</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G-A</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R334W</td>\n",
       "      <td>CFTR:p.Arg334Trp</td>\n",
       "      <td>7-117540230-C-T</td>\n",
       "      <td>7.0</td>\n",
       "      <td>117540230</td>\n",
       "      <td>C-T</td>\n",
       "      <td>rs121909011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>TRPV6</td>\n",
       "      <td>p.A626P</td>\n",
       "      <td>TRPV6:p.Ala626Pro</td>\n",
       "      <td>7-142873480-C-G</td>\n",
       "      <td>7.0</td>\n",
       "      <td>142873480</td>\n",
       "      <td>C-G</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Transient receptor potential vanilloid subfami...</td>\n",
       "      <td>PMID: 33610740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>CLDN2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLDN2 gene product belongs to the claudin prot...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>CEL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The enzyme carboxyl ester lipase (CEL), hydrol...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>CASR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The CASR allows regulation of parathyroid horm...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>GGT1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGT1 encodes for the membrane-bound enzyme gam...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>213 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name Amino acid change   Uploaded_variant Chr-Position-Ref-Alt  \\\n",
       "0        CFTR           p.G551D   CFTR:p.Gly551Asp      7-117587806-G-A   \n",
       "1        CFTR          p.N1303K  CFTR:p.Asn1303Lys      7-117652877-C-G   \n",
       "2        CFTR           p.R117H   CFTR:p.Arg117His    7-117710365-GA-AT   \n",
       "3        CFTR            p.G85E    CFTR:p.Gly85Glu      7-117509123-G-A   \n",
       "4        CFTR           p.R334W   CFTR:p.Arg334Trp      7-117540230-C-T   \n",
       "..        ...               ...                ...                  ...   \n",
       "208     TRPV6           p.A626P  TRPV6:p.Ala626Pro      7-142873480-C-G   \n",
       "209     CLDN2                 -                  -                    -   \n",
       "210       CEL                 -                  -                    -   \n",
       "211      CASR                 -                  -                    -   \n",
       "212      GGT1                 -                  -                    -   \n",
       "\n",
       "     CHROM        POS REF_ALT         rsID Variant mechanism  \\\n",
       "0      7.0  117587806     G-A   rs75527207               NaN   \n",
       "1      7.0  117652877     C-G   rs80034486               NaN   \n",
       "2      7.0  117710365   GA-AT   rs78655421               NaN   \n",
       "3      7.0  117509123     G-A   rs75961395               NaN   \n",
       "4      7.0  117540230     C-T  rs121909011               NaN   \n",
       "..     ...        ...     ...          ...               ...   \n",
       "208    7.0  142873480     C-G            -               NaN   \n",
       "209    NaN          0     NaN            -               NaN   \n",
       "210    NaN          0     NaN            -               NaN   \n",
       "211    NaN          0     NaN            -               NaN   \n",
       "212    NaN          0     NaN            -               NaN   \n",
       "\n",
       "                                             Gene Info  \\\n",
       "0    Cystic fibrosis transmembrane conductance regu...   \n",
       "1    Cystic fibrosis transmembrane conductance regu...   \n",
       "2    Cystic fibrosis transmembrane conductance regu...   \n",
       "3    Cystic fibrosis transmembrane conductance regu...   \n",
       "4    Cystic fibrosis transmembrane conductance regu...   \n",
       "..                                                 ...   \n",
       "208  Transient receptor potential vanilloid subfami...   \n",
       "209  CLDN2 gene product belongs to the claudin prot...   \n",
       "210  The enzyme carboxyl ester lipase (CEL), hydrol...   \n",
       "211  The CASR allows regulation of parathyroid horm...   \n",
       "212  GGT1 encodes for the membrane-bound enzyme gam...   \n",
       "\n",
       "                                        Publication  \n",
       "0    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "1    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "2    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "3    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "4    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "..                                              ...  \n",
       "208                                  PMID: 33610740  \n",
       "209                                             NaN  \n",
       "210                                             NaN  \n",
       "211                                             NaN  \n",
       "212                                             NaN  \n",
       "\n",
       "[213 rows x 11 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Final_Pancreatitis_data.xlsx')\n",
    "df['POS'] = df['POS'].fillna(0)\n",
    "df['POS'] = df['POS'].astype('int64')\n",
    "df = df.rename({'Gene':'Gene Name'}, axis=1)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "95a8b0e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Publication</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000415508.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440066.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/20</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303235</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000449750.7</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303236</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000634662.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000489430.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/23</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303237</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682740.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000508087.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/29</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>303238 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3          OR4F5   rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "4          OR4F5   rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "303233                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303234                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303235                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303236                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303237                       .  chrY  24853196   A   G  Heterozygous   \n",
       "\n",
       "                      Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1             5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3              synonymous variant              3/10       LOW           2.5   \n",
       "4       regulatory region variant              2/10  MODIFIER           1.5   \n",
       "...                           ...               ...       ...           ...   \n",
       "303233             intron variant              2/10  MODIFIER           1.5   \n",
       "303234             intron variant              2/10  MODIFIER           1.5   \n",
       "303235             intron variant              2/10  MODIFIER           1.5   \n",
       "303236         synonymous variant              3/10       LOW           2.5   \n",
       "303237         synonymous variant              3/10       LOW           2.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                                                             \n",
       "1                                                         ENST00000641515.2   \n",
       "2                                                                             \n",
       "3                                                         ENST00000641515.2   \n",
       "4                                                                             \n",
       "...              ...      ...                ...     ...                ...   \n",
       "303233                                                    ENST00000415508.6   \n",
       "303234                                                    ENST00000440066.6   \n",
       "303235                                                    ENST00000449750.7   \n",
       "303236                                                    ENST00000634662.1   \n",
       "303237                                                    ENST00000682740.1   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                    None                                  None  0/1   23   \n",
       "1                c.-36C>T                                  None  0/1   23   \n",
       "2                    None                                  None  0/1   23   \n",
       "3                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  132   \n",
       "4                    None                                  None  0/1  132   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "303233      c.499-1637A>G                                  None  0/1   53   \n",
       "303234      c.499-1637A>G                                  None  0/1   53   \n",
       "303235      c.499-1637A>G                                  None  0/1   53   \n",
       "303236           c.846A>G  ENSP00000489430.1        p.Gln282%3D  0/1   53   \n",
       "303237           c.846A>G  ENSP00000508087.1        p.Gln282%3D  0/1   53   \n",
       "\n",
       "        SDP   DP  RD  AD    FREQ        PVAL RDF RDR ADF ADR SIFT PolyPhen  \\\n",
       "0        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "1        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "2        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "3       102  102  64  38  37.25%  5.1448E-14  41  23  28  10                 \n",
       "4       102  102  64  38  37.25%  5.1448E-14  41  23  28  10                 \n",
       "...     ...  ...  ..  ..     ...         ...  ..  ..  ..  ..  ...      ...   \n",
       "303233   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303234   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303235   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303236   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303237   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding    3/3          \n",
       "4                          TF binding site                 \n",
       "...                                    ...    ...    ...   \n",
       "303233                      protein coding          6/15   \n",
       "303234                      protein coding          6/20   \n",
       "303235                      protein coding          6/16   \n",
       "303236                      protein coding  10/23          \n",
       "303237                      protein coding  10/29          \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  nan               1          \n",
       "1                                  nan               1          \n",
       "2                                  nan               1          \n",
       "3                                  S81  tcA/tcG      1          \n",
       "4                                  nan                          \n",
       "...                                ...      ...    ...    ...   \n",
       "303233                             nan               1          \n",
       "303234                             nan               1          \n",
       "303235                             nan               1          \n",
       "303236                            Q282  caA/caG      1          \n",
       "303237                            Q282  caA/caG      1          \n",
       "\n",
       "       Variant mechanism Publication  \n",
       "0                    NaN         NaN  \n",
       "1                    NaN         NaN  \n",
       "2                    NaN         NaN  \n",
       "3                    NaN         NaN  \n",
       "4                    NaN         NaN  \n",
       "...                  ...         ...  \n",
       "303233               NaN         NaN  \n",
       "303234               NaN         NaN  \n",
       "303235               NaN         NaN  \n",
       "303236               NaN         NaN  \n",
       "303237               NaN         NaN  \n",
       "\n",
       "[303238 rows x 42 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Specify the columns you want to keep from df dataset\n",
    "columns_to_keep_1 = ['POS', 'Variant mechanism', 'Publication']\n",
    "\n",
    "# Perform the left merge and select the desired columns\n",
    "merged_3 = pd.merge(merged_2, df[columns_to_keep_1], on='POS', how='left', sort=False)\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "a2c9a456",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PMID: 32948427, PMID: 31860051    3\n",
       "Name: Publication, dtype: int64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Publication.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "8ecb202f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: Variant mechanism, dtype: int64)"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3['Variant mechanism'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "37c7c71b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Publication</th>\n",
       "      <th>Gene Info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000415508.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440066.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/20</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303235</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000449750.7</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303236</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000634662.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000489430.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/23</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303237</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682740.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000508087.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/29</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>303238 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3          OR4F5   rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "4          OR4F5   rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "303233                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303234                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303235                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303236                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303237                       .  chrY  24853196   A   G  Heterozygous   \n",
       "\n",
       "                      Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1             5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3              synonymous variant              3/10       LOW           2.5   \n",
       "4       regulatory region variant              2/10  MODIFIER           1.5   \n",
       "...                           ...               ...       ...           ...   \n",
       "303233             intron variant              2/10  MODIFIER           1.5   \n",
       "303234             intron variant              2/10  MODIFIER           1.5   \n",
       "303235             intron variant              2/10  MODIFIER           1.5   \n",
       "303236         synonymous variant              3/10       LOW           2.5   \n",
       "303237         synonymous variant              3/10       LOW           2.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                                                             \n",
       "1                                                         ENST00000641515.2   \n",
       "2                                                                             \n",
       "3                                                         ENST00000641515.2   \n",
       "4                                                                             \n",
       "...              ...      ...                ...     ...                ...   \n",
       "303233                                                    ENST00000415508.6   \n",
       "303234                                                    ENST00000440066.6   \n",
       "303235                                                    ENST00000449750.7   \n",
       "303236                                                    ENST00000634662.1   \n",
       "303237                                                    ENST00000682740.1   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                    None                                  None  0/1   23   \n",
       "1                c.-36C>T                                  None  0/1   23   \n",
       "2                    None                                  None  0/1   23   \n",
       "3                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  132   \n",
       "4                    None                                  None  0/1  132   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "303233      c.499-1637A>G                                  None  0/1   53   \n",
       "303234      c.499-1637A>G                                  None  0/1   53   \n",
       "303235      c.499-1637A>G                                  None  0/1   53   \n",
       "303236           c.846A>G  ENSP00000489430.1        p.Gln282%3D  0/1   53   \n",
       "303237           c.846A>G  ENSP00000508087.1        p.Gln282%3D  0/1   53   \n",
       "\n",
       "        SDP   DP  RD  AD    FREQ        PVAL RDF RDR ADF ADR SIFT PolyPhen  \\\n",
       "0        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "1        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "2        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "3       102  102  64  38  37.25%  5.1448E-14  41  23  28  10                 \n",
       "4       102  102  64  38  37.25%  5.1448E-14  41  23  28  10                 \n",
       "...     ...  ...  ..  ..     ...         ...  ..  ..  ..  ..  ...      ...   \n",
       "303233   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303234   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303235   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303236   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303237   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding    3/3          \n",
       "4                          TF binding site                 \n",
       "...                                    ...    ...    ...   \n",
       "303233                      protein coding          6/15   \n",
       "303234                      protein coding          6/20   \n",
       "303235                      protein coding          6/16   \n",
       "303236                      protein coding  10/23          \n",
       "303237                      protein coding  10/29          \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  nan               1          \n",
       "1                                  nan               1          \n",
       "2                                  nan               1          \n",
       "3                                  S81  tcA/tcG      1          \n",
       "4                                  nan                          \n",
       "...                                ...      ...    ...    ...   \n",
       "303233                             nan               1          \n",
       "303234                             nan               1          \n",
       "303235                             nan               1          \n",
       "303236                            Q282  caA/caG      1          \n",
       "303237                            Q282  caA/caG      1          \n",
       "\n",
       "       Variant mechanism Publication Gene Info  \n",
       "0                    NaN         NaN       NaN  \n",
       "1                    NaN         NaN       NaN  \n",
       "2                    NaN         NaN       NaN  \n",
       "3                    NaN         NaN       NaN  \n",
       "4                    NaN         NaN       NaN  \n",
       "...                  ...         ...       ...  \n",
       "303233               NaN         NaN       NaN  \n",
       "303234               NaN         NaN       NaN  \n",
       "303235               NaN         NaN       NaN  \n",
       "303236               NaN         NaN       NaN  \n",
       "303237               NaN         NaN       NaN  \n",
       "\n",
       "[303238 rows x 43 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = df.drop_duplicates(subset='Gene Name', keep='first')\n",
    "\n",
    "# Specify the columns you want to keep from df dataset\n",
    "columns_to_keep_2 = ['Gene Name', 'Gene Info']\n",
    "\n",
    "# Perform the left merge and select the desired columns\n",
    "merged_4 = pd.merge(merged_3, df_1[columns_to_keep_2], on='Gene Name', how='left', sort=False)\n",
    "merged_4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "6c67cd45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_4['Gene Info'].value_counts().count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "1cd3f96c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Publication</th>\n",
       "      <th>Gene Info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000415508.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440066.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/20</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303235</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000449750.7</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303236</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000634662.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000489430.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/23</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303237</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682740.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000508087.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/29</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>303238 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2                 rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3          OR4F5   rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "4          OR4F5   rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "303233                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303234                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303235                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303236                       .  chrY  24853196   A   G  Heterozygous   \n",
       "303237                       .  chrY  24853196   A   G  Heterozygous   \n",
       "\n",
       "                      Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1             5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3              synonymous variant              3/10       LOW           2.5   \n",
       "4       regulatory region variant              2/10  MODIFIER           1.5   \n",
       "...                           ...               ...       ...           ...   \n",
       "303233             intron variant              2/10  MODIFIER           1.5   \n",
       "303234             intron variant              2/10  MODIFIER           1.5   \n",
       "303235             intron variant              2/10  MODIFIER           1.5   \n",
       "303236         synonymous variant              3/10       LOW           2.5   \n",
       "303237         synonymous variant              3/10       LOW           2.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                                                             \n",
       "1                                                         ENST00000641515.2   \n",
       "2                                                                             \n",
       "3                                                         ENST00000641515.2   \n",
       "4                                                                             \n",
       "...              ...      ...                ...     ...                ...   \n",
       "303233                                                    ENST00000415508.6   \n",
       "303234                                                    ENST00000440066.6   \n",
       "303235                                                    ENST00000449750.7   \n",
       "303236                                                    ENST00000634662.1   \n",
       "303237                                                    ENST00000682740.1   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                    None                                  None  0/1   23   \n",
       "1                c.-36C>T                                  None  0/1   23   \n",
       "2                    None                                  None  0/1   23   \n",
       "3                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  132   \n",
       "4                    None                                  None  0/1  132   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "303233      c.499-1637A>G                                  None  0/1   53   \n",
       "303234      c.499-1637A>G                                  None  0/1   53   \n",
       "303235      c.499-1637A>G                                  None  0/1   53   \n",
       "303236           c.846A>G  ENSP00000489430.1        p.Gln282%3D  0/1   53   \n",
       "303237           c.846A>G  ENSP00000508087.1        p.Gln282%3D  0/1   53   \n",
       "\n",
       "        SDP   DP  RD  AD    FREQ        PVAL RDF RDR ADF ADR SIFT PolyPhen  \\\n",
       "0        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "1        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "2        22   22  15   7  31.82%   4.4505E-3  14   1   6   1                 \n",
       "3       102  102  64  38  37.25%  5.1448E-14  41  23  28  10                 \n",
       "4       102  102  64  38  37.25%  5.1448E-14  41  23  28  10                 \n",
       "...     ...  ...  ..  ..     ...         ...  ..  ..  ..  ..  ...      ...   \n",
       "303233   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303234   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303235   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303236   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303237   55   54  38  16  29.63%   4.1444E-6  28  10  14   2                 \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding    3/3          \n",
       "4                          TF binding site                 \n",
       "...                                    ...    ...    ...   \n",
       "303233                      protein coding          6/15   \n",
       "303234                      protein coding          6/20   \n",
       "303235                      protein coding          6/16   \n",
       "303236                      protein coding  10/23          \n",
       "303237                      protein coding  10/29          \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  nan               1          \n",
       "1                                  nan               1          \n",
       "2                                  nan               1          \n",
       "3                                  S81  tcA/tcG      1          \n",
       "4                                  nan                          \n",
       "...                                ...      ...    ...    ...   \n",
       "303233                             nan               1          \n",
       "303234                             nan               1          \n",
       "303235                             nan               1          \n",
       "303236                            Q282  caA/caG      1          \n",
       "303237                            Q282  caA/caG      1          \n",
       "\n",
       "       Variant mechanism Publication Gene Info  \n",
       "0                     No          No        No  \n",
       "1                     No          No        No  \n",
       "2                     No          No        No  \n",
       "3                     No          No        No  \n",
       "4                     No          No        No  \n",
       "...                  ...         ...       ...  \n",
       "303233                No          No        No  \n",
       "303234                No          No        No  \n",
       "303235                No          No        No  \n",
       "303236                No          No        No  \n",
       "303237                No          No        No  \n",
       "\n",
       "[303238 rows x 43 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns_to_fill = ['Variant mechanism', 'Publication', 'Gene Info']\n",
    "merged_4[columns_to_fill] = merged_4[columns_to_fill].fillna('No')\n",
    "merged_4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "7b152192",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
       "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
       "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
       "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
       "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
       "       'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Variant mechanism', 'Publication', 'Gene Info'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_4.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "10936f14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Info</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Publication</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000415508.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440066.6</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/20</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303235</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000449750.7</td>\n",
       "      <td>c.499-1637A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303236</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000634662.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000489430.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/23</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303237</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000682740.1</td>\n",
       "      <td>c.846A&gt;G</td>\n",
       "      <td>ENSP00000508087.1</td>\n",
       "      <td>p.Gln282%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/29</td>\n",
       "      <td></td>\n",
       "      <td>Q282</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>303238 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene Info          rsID CHROM       POS Variant mechanism  \\\n",
       "0                       No  rs1473003496  chr1     65529                No   \n",
       "1                       No  rs1473003496  chr1     65529                No   \n",
       "2                       No  rs1473003496  chr1     65529                No   \n",
       "3          OR4F5        No   rs201219564  chr1     69270                No   \n",
       "4          OR4F5        No   rs201219564  chr1     69270                No   \n",
       "...          ...       ...           ...   ...       ...               ...   \n",
       "303233                  No             .  chrY  24853196                No   \n",
       "303234                  No             .  chrY  24853196                No   \n",
       "303235                  No             .  chrY  24853196                No   \n",
       "303236                  No             .  chrY  24853196                No   \n",
       "303237                  No             .  chrY  24853196                No   \n",
       "\n",
       "       Publication REF ALT      Zygosity                Consequence  \\\n",
       "0               No   C   T  Heterozygous    downstream gene variant   \n",
       "1               No   C   T  Heterozygous        5 prime UTR variant   \n",
       "2               No   C   T  Heterozygous    downstream gene variant   \n",
       "3               No   A   G  Heterozygous         synonymous variant   \n",
       "4               No   A   G  Heterozygous  regulatory region variant   \n",
       "...            ...  ..  ..           ...                        ...   \n",
       "303233          No   A   G  Heterozygous             intron variant   \n",
       "303234          No   A   G  Heterozygous             intron variant   \n",
       "303235          No   A   G  Heterozygous             intron variant   \n",
       "303236          No   A   G  Heterozygous         synonymous variant   \n",
       "303237          No   A   G  Heterozygous         synonymous variant   \n",
       "\n",
       "       Consequence_score    IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                   2/10  MODIFIER           1.5                          \n",
       "1                   3/10  MODIFIER           1.5                          \n",
       "2                   2/10  MODIFIER           1.5                          \n",
       "3                   3/10       LOW           2.5                          \n",
       "4                   2/10  MODIFIER           1.5                          \n",
       "...                  ...       ...           ...           ...      ...   \n",
       "303233              2/10  MODIFIER           1.5                          \n",
       "303234              2/10  MODIFIER           1.5                          \n",
       "303235              2/10  MODIFIER           1.5                          \n",
       "303236              3/10       LOW           2.5                          \n",
       "303237              3/10       LOW           2.5                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                                                  None   \n",
       "1                                  ENST00000641515.2           c.-36C>T   \n",
       "2                                                                  None   \n",
       "3                                  ENST00000641515.2           c.243A>G   \n",
       "4                                                                  None   \n",
       "...                   ...     ...                ...                ...   \n",
       "303233                             ENST00000415508.6      c.499-1637A>G   \n",
       "303234                             ENST00000440066.6      c.499-1637A>G   \n",
       "303235                             ENST00000449750.7      c.499-1637A>G   \n",
       "303236                             ENST00000634662.1           c.846A>G   \n",
       "303237                             ENST00000682740.1           c.846A>G   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP  RD  AD  \\\n",
       "0                                       None  0/1   23   22   22  15   7   \n",
       "1                                       None  0/1   23   22   22  15   7   \n",
       "2                                       None  0/1   23   22   22  15   7   \n",
       "3       ENSP00000493376.2         p.Ser81%3D  0/1  132  102  102  64  38   \n",
       "4                                       None  0/1  132  102  102  64  38   \n",
       "...                   ...                ...  ...  ...  ...  ...  ..  ..   \n",
       "303233                                  None  0/1   53   55   54  38  16   \n",
       "303234                                  None  0/1   53   55   54  38  16   \n",
       "303235                                  None  0/1   53   55   54  38  16   \n",
       "303236  ENSP00000489430.1        p.Gln282%3D  0/1   53   55   54  38  16   \n",
       "303237  ENSP00000508087.1        p.Gln282%3D  0/1   53   55   54  38  16   \n",
       "\n",
       "          FREQ        PVAL RDF RDR ADF ADR SIFT PolyPhen  \\\n",
       "0       31.82%   4.4505E-3  14   1   6   1                 \n",
       "1       31.82%   4.4505E-3  14   1   6   1                 \n",
       "2       31.82%   4.4505E-3  14   1   6   1                 \n",
       "3       37.25%  5.1448E-14  41  23  28  10                 \n",
       "4       37.25%  5.1448E-14  41  23  28  10                 \n",
       "...        ...         ...  ..  ..  ..  ..  ...      ...   \n",
       "303233  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303234  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303235  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303236  29.63%   4.1444E-6  28  10  14   2                 \n",
       "303237  29.63%   4.1444E-6  28  10  14   2                 \n",
       "\n",
       "                                   BIOTYPE   EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene                 \n",
       "1                           protein coding    2/3          \n",
       "2                                   lncRNA                 \n",
       "3                           protein coding    3/3          \n",
       "4                          TF binding site                 \n",
       "...                                    ...    ...    ...   \n",
       "303233                      protein coding          6/15   \n",
       "303234                      protein coding          6/20   \n",
       "303235                      protein coding          6/16   \n",
       "303236                      protein coding  10/23          \n",
       "303237                      protein coding  10/29          \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  nan               1         \n",
       "1                                  nan               1         \n",
       "2                                  nan               1         \n",
       "3                                  S81  tcA/tcG      1         \n",
       "4                                  nan                         \n",
       "...                                ...      ...    ...    ...  \n",
       "303233                             nan               1         \n",
       "303234                             nan               1         \n",
       "303235                             nan               1         \n",
       "303236                            Q282  caA/caG      1         \n",
       "303237                            Q282  caA/caG      1         \n",
       "\n",
       "[303238 rows x 43 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_4 = merged_4[['Gene Name', 'Gene Info', 'rsID', 'CHROM', 'POS', 'Variant mechanism', 'Publication', \n",
    "                    'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "f248e6c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_4.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/KHAIGHGPTTL187_final.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e444d6a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Haplotype_Gene</th>\n",
       "      <th>Haplotype_Star_Allele</th>\n",
       "      <th>HET_Mean(%)</th>\n",
       "      <th>HOM_Mean(%)</th>\n",
       "      <th>Gender_M</th>\n",
       "      <th>Gender_F</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>296</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>558</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>558</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>98</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>153</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Count Haplotype_Gene Haplotype_Star_Allele  HET_Mean(%)  HOM_Mean(%)  \\\n",
       "0      1        CYP2C19                    *3            1            0   \n",
       "1      5        CYP2C19                    *3            1            0   \n",
       "2    296        CYP2C19                    *3            0            1   \n",
       "3    558        CYP2C19                    *3            0            1   \n",
       "4     98        CYP2C19                    *3            1            0   \n",
       "5    153        CYP2C19                    *3            1            0   \n",
       "\n",
       "   Gender_M  Gender_F  \n",
       "0         0         1  \n",
       "1         5         0  \n",
       "2         0       296  \n",
       "3       558         0  \n",
       "4         0        98  \n",
       "5       153         0  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Your data\n",
    "data = {\n",
    "    'Count': [1, 5, 296, 558, 98, 153],\n",
    "    'Haplotype_Gene': ['CYP2C19', 'CYP2C19', 'CYP2C19', 'CYP2C19', 'CYP2C19', 'CYP2C19'],\n",
    "    'Haplotype_Star_Allele': ['*3', '*3', '*3', '*3', '*3', '*3'],\n",
    "    'HET_Mean(%)': [1, 1, 0, 0, 1, 1],\n",
    "    'HOM_Mean(%)': [0, 0, 1, 1, 0, 0],\n",
    "    'Gender_M': [0, 5, 0, 558, 0, 153],\n",
    "    'Gender_F': [1, 0, 296, 0, 98, 0]\n",
    "}\n",
    "\n",
    "# Creating a DataFrame\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "0fd69054",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Haplotype_Gene</th>\n",
       "      <th>Haplotype_Star_Allele</th>\n",
       "      <th>Count</th>\n",
       "      <th>Male_HET</th>\n",
       "      <th>Male_HOM</th>\n",
       "      <th>Female_HET</th>\n",
       "      <th>Female_HOM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>1111</td>\n",
       "      <td>158</td>\n",
       "      <td>558</td>\n",
       "      <td>99</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Haplotype_Gene Haplotype_Star_Allele  Count  Male_HET  Male_HOM  Female_HET  \\\n",
       "0        CYP2C19                    *3   1111       158       558          99   \n",
       "\n",
       "   Female_HOM  \n",
       "0         296  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Creating separate conditions for HET and HOM\n",
    "het_condition = df['HET_Mean(%)'] == 1\n",
    "hom_condition = df['HOM_Mean(%)'] == 1\n",
    "\n",
    "# Creating new columns for male and female counts based on conditions\n",
    "df['Male_HET'] = df['Gender_M'].where(het_condition, 0)\n",
    "df['Male_HOM'] = df['Gender_M'].where(hom_condition, 0)\n",
    "df['Female_HET'] = df['Gender_F'].where(het_condition, 0)\n",
    "df['Female_HOM'] = df['Gender_F'].where(hom_condition, 0)\n",
    "\n",
    "# Grouping by Haplotype_Gene and Haplotype_Star_Allele and aggregating the counts\n",
    "result = df.groupby(['Haplotype_Gene', 'Haplotype_Star_Allele']).agg({\n",
    "    'Count': 'sum',\n",
    "    'Male_HET': 'sum',\n",
    "    'Male_HOM': 'sum',\n",
    "    'Female_HET': 'sum',\n",
    "    'Female_HOM': 'sum'\n",
    "}).reset_index()\n",
    "\n",
    "# Renaming columns\n",
    "result.columns = ['Haplotype_Gene', 'Haplotype_Star_Allele', 'Count', 'Male_HET', 'Male_HOM', 'Female_HET', 'Female_HOM']\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "42f4a630",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Haplotype_Gene</th>\n",
       "      <th>Haplotype_Star_Allele</th>\n",
       "      <th>Count</th>\n",
       "      <th>HET_Mean(%)</th>\n",
       "      <th>HOM_Mean(%)</th>\n",
       "      <th>Gender_M</th>\n",
       "      <th>Gender_F</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>*3</td>\n",
       "      <td>1111</td>\n",
       "      <td>158</td>\n",
       "      <td>558</td>\n",
       "      <td>716</td>\n",
       "      <td>395</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Haplotype_Gene Haplotype_Star_Allele  Count  HET_Mean(%)  HOM_Mean(%)  \\\n",
       "0        CYP2C19                    *3   1111          158          558   \n",
       "\n",
       "   Gender_M  Gender_F  \n",
       "0       716       395  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "cba9bda6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>New_Column</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A(24)</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>T(56)</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C(10)</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G(30)</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Column1 New_Column\n",
       "0   A(24)          A\n",
       "1   T(56)          T\n",
       "2   C(10)          C\n",
       "3   G(30)          G"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a sample DataFrame\n",
    "data = {'Column1': ['A(24)', 'T(56)', 'C(10)', 'G(30)']}\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Extract only the letters from the values\n",
    "df['New_Column'] = df['Column1'].str.extract(r'([A-Za-z])')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "314aba6c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>.</td>\n",
       "      <td>Value1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A</td>\n",
       "      <td>Value2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>Value3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B</td>\n",
       "      <td>Value4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>Value5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Column1 Column2\n",
       "0       .  Value1\n",
       "1       A  Value2\n",
       "2       .  Value3\n",
       "3       B  Value4\n",
       "4       .  Value5"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a sample DataFrame\n",
    "data = {'Column1': ['.', 'A', '.', 'B', '.'],\n",
    "        'Column2': ['Value1', 'Value2', 'Value3', 'Value4', 'Value5']}\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "07767d66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Value1</td>\n",
       "      <td>Value1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A</td>\n",
       "      <td>Value2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Value3</td>\n",
       "      <td>Value3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B</td>\n",
       "      <td>Value4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Value5</td>\n",
       "      <td>Value5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Column1 Column2\n",
       "0  Value1  Value1\n",
       "1       A  Value2\n",
       "2  Value3  Value3\n",
       "3       B  Value4\n",
       "4  Value5  Value5"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Identify rows with \".\" in Column1\n",
    "mask = df['Column1'] == '.'\n",
    "\n",
    "# Replace rows with \".\" in Column1 with values from Column2\n",
    "df.loc[mask, 'Column1'] = df.loc[mask, 'Column2']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "56de3a65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAxoAAAKQCAYAAAAYHAxYAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAAEAAElEQVR4nOzdd3hT5dvA8W+SjnTv3VJG2avspWwUZImgiPID3IMXAfdEXIgoomxQAVFZyhQEZQ/ZQhllldEB3Xu3Gef9oxKobSkjJW25P9eVC3LOc55zn7RNcp9nqRRFURBCCCGEEEIIM1JbOgAhhBBCCCFE9SOJhhBCCCGEEMLsJNEQQgghhBBCmJ0kGkIIIYQQQgizk0RDCCGEEEIIYXaSaAghhBBCCCHMThINIYQQQgghhNlJoiGEEEIIIYQwO0k0hBBCCCGEEGYniYYQosIsWrQIlUrF4cOHS93fr18/atasWeFxqFQqJk6ceMvHxcbGMnHiRMLCwsweU0XZs2cPw4YNo0aNGtja2uLg4EDjxo157bXXOHPmjKXDu6Grvy9XH1qtFl9fX7p168bnn39OYmJiiWMmTpyISqW6pfPk5uYyceJEduzYcUvHlXaumjVr0q9fv1uqpzxLlizhm2++KXXf7f4uCyGEJUiiIYQQZYiNjeWjjz6qMonG+++/z/33309UVBTvv/8+mzZtYs2aNTz99NNs3ryZhg0bYjAYLB1muRYuXMi+ffvYvHkzs2bNIjQ0lC+++IKGDRuyZcuWYmWfffZZ9u3bd0v15+bm8tFHH91yonE757odN0o09u3bx7PPPlvhMQghhDlYWToAIYQQd27p0qV89tlnvPjii8yePbvYnfdevXrx6quvMnv2bAtGePOaNGlC69atTc8HDx7M+PHjue+++3jkkUeIiIjAx8cHgMDAQAIDAys0ntzcXOzt7e/KucrTvn17i55fCCFuhbRoCCEqlVmzZtG5c2e8vb1xcHCgadOmTJkyBZ1OV6xc165dadKkCbt376Z9+/bY2dkREBDABx98cFN37U+ePMnAgQNxc3NDq9USGhrKjz/+aNq/Y8cO2rRpA8BTTz1l6s4zceJEfvrpJ1QqVal3tz/++GOsra2JjY295TgLCwv59NNPadCgAba2tnh5efHUU0+RlJRU7vV8+umneHp6Mm3atFK7EqlUKkaPHo1Goym2fcuWLfTo0QNnZ2fs7e3p1KkTW7duLVbmapeh8PBwhg0bhouLCz4+Pjz99NNkZGQUK6soCrNnzyY0NBQ7Ozvc3NwYMmQIFy9eLPcabqRGjRpMnTqVrKws5s2bVyK2623bto2uXbvi4eGBnZ0dNWrUYPDgweTm5hIZGYmXlxcAH330kennOmrUqGL1HTlyhCFDhuDm5kadOnXKPNdVq1evplmzZmi1WmrXrs306dOL7b/aLSwyMrLY9h07dqBSqUytK127dmXDhg1ERUUV60Z2VWldp8r7Xb7+PEuXLuW9997D398fZ2dnevbsydmzZ8t+4YUQ4g5IoiGEqHAGgwG9Xl/ioShKibIXLlzgiSee4KeffmL9+vU888wzfPnll7zwwgslysbHx/P444/z5JNPsnbtWoYMGcKnn37K2LFjbxjP2bNn6dixI+Hh4UyfPp1Vq1bRqFEjRo0axZQpUwBo2bIlCxcuBIq6JO3bt8/UbWXo0KH4+voya9asYvXq9XrmzZvHoEGD8Pf3v6U4jUYjAwcOZPLkyTzxxBNs2LCByZMns3nzZrp27UpeXl6Z1xMbG8upU6fo1asXWq32htd+vZ9//pkHHngAZ2dnfvzxR1asWIG7uzsPPvhgiWQDiloW6tWrx8qVK3n77bdZsmQJ48ePL1bmhRdeYNy4cfTs2ZM1a9Ywe/ZswsPD6dixIwkJCTcdW2keeughNBoNu3btKrNMZGQkffv2xcbGhgULFrBp0yYmT56Mg4MDhYWF+Pn5sWnTJgCeeeYZ08/1gw8+KFbPI488QkhICL/++itz5869YVxhYWGMGzeO8ePHs3r1ajp27MjYsWP56quvbvkaZ8+eTadOnfD19TXFdqPuWjfzu3y9d999l6ioKL7//nvmz59PREQE/fv3rxJd6oQQVZAihBAVZOHChQpww0dwcHCZxxsMBkWn0ymLFy9WNBqNkpqaatrXpUsXBVDWrl1b7JjnnntOUavVSlRUlGkboHz44Yem548//rhia2urREdHFzu2T58+ir29vZKenq4oiqIcOnRIAZSFCxeWiO3DDz9UbGxslISEBNO25cuXK4Cyc+fOW45z6dKlCqCsXLmyWLmrMcyePbvM12n//v0KoLz99tsl9un1ekWn05keRqNRURRFycnJUdzd3ZX+/fsXK28wGJTmzZsrbdu2LXatgDJlypRiZV9++WVFq9Wa6ty3b58CKFOnTi1WLiYmRrGzs1PefPPNMq9BUa79vhw6dKjMMj4+PkrDhg1LxHbVb7/9pgBKWFhYmXUkJSWV+J34b30TJkwoc9/1goODFZVKVeJ8vXr1UpydnZWcnJxi13bp0qVi5bZv364Ayvbt203b+vbtW+bfxe3+Ll89z0MPPVSs3IoVKxRA2bdvX6nnE0KIOyEtGkKICrd48WIOHTpU4nHfffeVKHv06FEGDBiAh4cHGo0Ga2trRowYgcFg4Ny5c8XKOjk5MWDAgGLbnnjiCYxG4w3vem/bto0ePXoQFBRUbPuoUaPIzc29qQG/L730EgDfffedadvMmTNp2rQpnTt3vuU4169fj6urK/379y/W6hMaGoqvr+8tD1y+ysPDA2tra9Nj5cqVAOzdu5fU1FRGjhxZ7HxGo5HevXtz6NAhcnJyitX132to1qwZ+fn5ptmg1q9fj0qlYvjw4cXq9PX1pXnz5rd9DddTSmkFu15oaCg2NjY8//zz/Pjjj7fdZWvw4ME3XbZx48Y0b9682LYnnniCzMxMjhw5clvnv1m3+rtc2s8QICoqqkLjFELcm2QwuBCiwjVs2LDY4N6rXFxciImJMT2Pjo7m/vvvp379+nz77bfUrFkTrVbLwYMHGT16dInuQ1cHBF/P19cXgJSUlDLjSUlJwc/Pr8T2q92dbnTs9eceOnQo8+bN4+233yY8PJzdu3cXGz9wK3EmJCSQnp6OjY1NqedLTk4uM5arXzJL+7K4Y8cO9Ho9//zzDy+++KJp+9VuTEOGDCmz3tTUVBwcHEzPPTw8iu23tbUFMP1cEhISUBSl1OsFqF27dpnnuhk5OTmkpKTQtGnTMsvUqVOHLVu2MGXKFEaPHk1OTg61a9fmlVdeKbdL3fVK+/0oy9WfZWnbbuZ36U7c6u9yeT9DIYQwJ0k0hBCVxpo1a8jJyWHVqlUEBwebtpc1vWxpff7j4+OBkl+orufh4UFcXFyJ7VcHcHt6et5UvGPHjuWnn35i7dq1bNq0CVdXV5588snbitPT0xMPDw/T+IH/cnJyKjMOf39/GjduzObNm8nPzy82TiM0NBSA7OzsYsdcvcYZM2aUOZNRWQlDWTw9PVGpVOzevdv0BfZ6pW27FRs2bMBgMNC1a9cblrv//vu5//77MRgMHD58mBkzZjBu3Dh8fHx4/PHHb+pct7I2x9WfZWnbrv58r/5MCgoKipW7UQJ5M8z1uyyEEBVBuk4JISqNq1/urv9CqihKse5J18vKymLdunXFti1ZsgS1Wl2i+9L1evTowbZt20xfxq5avHgx9vb2pi/e5d3tbdWqFR07duSLL77gl19+YdSoUcVaAG4lzn79+pGSkoLBYKB169YlHvXr1y/zegDee+89kpOTefXVV8vtXgTQqVMnXF1dOXXqVKnna926dZmtK2Xp168fiqJw5cqVUuu7UUtEeaKjo3n99ddxcXEpdWKA0mg0Gtq1a2catH+1G5O57+KHh4dz7NixYtuWLFmCk5MTLVu2BDAtTHn8+PFi5f77e3E1vpuN7WZ/l4UQwhKkRUMIUWn06tULGxsbhg0bxptvvkl+fj5z5swhLS2t1PIeHh689NJLREdHU69ePf744w++++47XnrpJWrUqFHmeT788EPWr19Pt27dmDBhAu7u7vzyyy9s2LCBKVOm4OLiAhR1w7Gzs+OXX36hYcOGODo64u/vX2xGqbFjxzJ06FBUKhUvv/zybcf5+OOP88svv/DQQw8xduxY2rZti7W1NZcvX2b79u0MHDiQQYMGlXlNw4YNIzw8nM8++4xjx44xatQo6tati9FoJCYmhp9++gm41jLi6OjIjBkzGDlyJKmpqQwZMgRvb2+SkpI4duwYSUlJzJkz5wY/rZI6derE888/z1NPPcXhw4fp3LkzDg4OxMXFsWfPHpo2bWoa23IjJ0+eNI3vSExMZPfu3SxcuBCNRsPq1atN09OWZu7cuWzbto2+fftSo0YN8vPzWbBgAQA9e/Y0vQbBwcGsXbuWHj164O7ujqen522vUu/v78+AAQOYOHEifn5+/Pzzz2zevJkvvvgCe3t7ANq0aUP9+vV5/fXX0ev1uLm5sXr1avbs2VOivqZNm7Jq1SrmzJlDq1atUKvVpXY9hJv/XRZCCIuw6FB0IUS1Vt4sQqXNrvP7778rzZs3V7RarRIQEKC88cYbysaNG0vMzNOlSxelcePGyo4dO5TWrVsrtra2ip+fn/Luu+8qOp2uWJ2UMsPQiRMnlP79+ysuLi6KjY2N0rx581Jnl1q6dKnSoEEDxdrautR6CgoKFFtbW6V3796lXuOtxKnT6ZSvvvrKdP2Ojo5KgwYNlBdeeEGJiIgotf7/2rVrlzJ06FAlMDBQsba2Vuzt7ZVGjRopL730knL48OES5Xfu3Kn07dtXcXd3V6ytrZWAgAClb9++yq+//moqc3W2paSkpGLHljWT0oIFC5R27dopDg4Oip2dnVKnTh1lxIgRpZ6/tPquPmxsbBRvb2+lS5cuyqRJk5TExMQSx/x3Jqh9+/YpgwYNUoKDgxVbW1vFw8ND6dKli7Ju3bpix23ZskVp0aKFYmtrqwDKyJEjb3itpZ1LUYpmnerbt6/y22+/KY0bN1ZsbGyUmjVrKl9//XWJ48+dO6c88MADirOzs+Ll5aWMGTNG2bBhQ4nf7dTUVGXIkCGKq6urolKpip3zdn+Xr846df3PVVEU5dKlS2XOrCaEEHdKpSg30cYuhBCVTNeuXUlOTubkyZMWjeP3339nwIABbNiwgYceeqjE/soSpxBCCHG3SdcpIYS4DadOnSIqKorXXnuN0NBQ+vTpY+mQhBBCiEpFBoMLIcRtePnllxkwYABubm4sXbr0lmYpulckJibywgsvUKNGDWxtbfH19eXBBx80re1Qs2ZNvvnmmxvWsXLlSrp27YqLiwuOjo40a9aMjz/+mNTUVFOZvLw8PvzwQ+rXr4+trS2enp4MGTKE8PDwYnWFh4czePBgatasiUqlKvXco0aNQqVSmR4eHh707t27xCDuzz77jI4dO2Jvb4+rq+ttvT5CCFHdSaIhhKiSduzYYdHuSDt27ECn03HgwAEaNGhww3L3arepwYMHc+zYMX788UfOnTvHunXr6Nq1a7Ek4Ubee+89hg4dSps2bdi4cSMnT55k6tSpHDt2zDS4vaCggJ49e7JgwQI++eQTzp07xx9//IHBYKBdu3bs37/fVF9ubi61a9dm8uTJpa59cVXv3r2Ji4sjLi6OrVu3YmVlRb9+/YqVKSws5NFHH72pwe1CCHGvkjEaQgghzC49PR03Nzd27NhBly5dSi1Ts2ZNxo0bx7hx40rsO3jwIO3ateObb74pdaG99PR0XF1d+eKLL3jnnXc4evRosdW5jUYj7dq1Izc3l5MnT5ZocSrr3KNGjSI9PZ01a9aYtu3evZvOnTuTmJhYYsarRYsWMW7cONLT02/8ggghxD1IWjSEEEKYnaOjI46OjqxZs6bEInU345dffsHR0bHMKYOvdldasmQJvXr1KpZkAKjVasaPH8+pU6dKrHFxK7Kzs/nll18ICQm54SKQQgghSpJEQwghhNlZWVmxaNEifvzxR1xdXenUqRPvvvtuibEOZYmIiKB27dpYW1vfsNy5c+do2LBhqfuubj937twtxb5+/XpTouTk5MS6detYvnw5arV8ZAohxK2Qd00hhBAVYvDgwcTGxrJu3ToefPBBduzYQcuWLVm0aFG5xyqKcscD7K/2DL7Verp160ZYWBhhYWEcOHCABx54gD59+hAVFXVH8QghxL1GEg0hhBAVRqvV0qtXLyZMmMDevXsZNWoUH374YbnH1atXjwsXLqDT6cotd+rUqVL3nTlzBoC6deveUswODg6EhIQQEhJC27Zt+eGHH8jJyeG77767pXqEEOJeJ4mGEEKIu6ZRo0bk5OSUW+6JJ54gOzub2bNnl7r/6uDrxx9/nC1btpQYh2E0Gpk2bRqNGjUqMX7jVqlUKtRqNXl5eXdUjxBC3GtkwT4hhBBml5KSwqOPPsrTTz9Ns2bNcHJy4vDhw0yZMoWBAweayl25coWwsLBix9aoUYN27drx5ptv8tprr3HlyhUGDRqEv78/58+fZ+7cudx3332MHTuW8ePHs3btWvr378/UqVNp164dCQkJTJo0idOnT7NlyxZT16nCwkJT60dhYaHp3I6OjoSEhJjOX1BQQHx8PABpaWnMnDmT7Oxs+vfvbyoTHR1Namoq0dHRGAwG0zWEhITg6OhYES+pEEJUOTK9rRBCCLMrKChg4sSJ/PXXX6YuUEFBQTz66KO8++672NnZUbNmzVLHPSxcuJBRo0YBsGLFCmbNmsXRo0cxGo3UqVOHIUOGMGbMGNPMU7m5uXz++ecsW7aMqKgonJyc6NatGxMnTqRJkyameiMjI6lVq1aJ83Xp0oUdO3YARdPb/vjjj6Z9Tk5ONGjQgLfeeovBgwebtv+33FXbt2+na9eut/GKCSFE9SOJhhBCCCGEEMLsZIyGEEIIIYQQwuwk0RBCCCGEEEKYnSQaQgghhBBCCLOTREMIIYQQQghhdpJoCCGEEEIIIcxOEg0hhBBCCCGE2UmiIYQQQgghhDA7STSEEEIIIYQQZieJhhBCCCGEEMLsJNEQQgghhBBCmJ0kGkIIIYQQQgizk0RDCCGEEEIIYXaSaAghhBBCCCHMThINIYQQQgghhNlJoiGEEEIIIYQwO0k0hBBCCCGEEGYniYYQQgghhBDC7CTREEIIIYQQQpidJBpCCCGEEEIIs5NEQwghhBBCCGF2kmgIIYQQQgghzE4SDSGEEEIIIYTZSaIhhBDiliiKgmJULB2GEEKISs7K0gEIIYQwr4I8PXmZheTn6tDlGSjI01OYr6cw7+rDQEG+Hl2evmhfnh5dgQGjQcFoLEoiiv3fqKBc91y5mmOoQK1RodGoUWtURf+3UqOxVmNlrcHKRo2VtRqNtQZrWzW29tZoHayxdbBC62CN1v66///70FjL/S8hhKguVIqiyG0pIYSoAgrz9ORkFJCTUUhOegE5GQXkZhQWbUv/9/+ZhegLDJYO9bZZWauxdbDGzskaRzctjm62OLkX/Xv1uYObLRqNJCRCCFHZSaIhhBCVSG5mIRmJuWQk5ZGemEtGYh4ZSXlkJOZSmF91EwhzUqnA3tkGx+sSEGdPO9x87XHztcfRTWvpEIUQQiCJhhBC3HX6QgMpsTmkxeUUTygkmTALa1sNbr72uPra4+bjgJtf0b8uPnbSEiKEEHeRJBpCCFGBcjMLSb6cRXJMNsmXs0mOySI9MU8GU1uAWq3CyVOLm68DnoGOeAc74VXDGUc3W0uHJoQQ1ZIkGkIIYQaKopCRmEdSzNWkoujf3MxCS4cmymHvbINXsBPeNZzwCnbGO9gJBxdJPoQQ4k5JoiGEELdBrzOQGJlF3IV04i9kEHcxg4IcvaXDEmZi72KDd7AzXjWc8KnpjG8dF2ztZKJGIYS4FZJoCCHETcjLKiTuQgZxFzKIv5BOYnQWRr28fd4rVCrwDHLCP8QV/3qu+Nd1RetgbemwhBCiUpNEQwghSpGTXkD0qVRiz6cTdz6djMQ8S4ckKhMVePg74F/XDf+6RYmHvbONpaMSQohKRRINIYSgqCtUbEQ60adSiTmVSmpsjqVDElWMm689/vXcCGrgRlBDd2ykq5UQ4h4niYYQ4p6VciWbmNOpRS0XEekYdEZLhySqCbVGhV+IC8GNPQlu6oG7n4OlQxJCiLtOEg0hxD2jME9PVHiKqdUiJ73A0iGJe4Szp5bgxh7UaOJBYH03rGw0lg5JCCEqnCQaQohqLS+7kEvHkrl4NImYM6kygFtYnJW1moD6bgQ38aBmM0+c3GUlcyFE9SSJhhCi2slJL+DC0SQuhiUSG5Ehi+OJSs27pjN1W3sT0sobRzdJOoQQ1YckGkKIaiEzOY8LR5K4cDSRhMhMkHc2UdWowLeWCyH/Jh2yaKAQoqqTREMIUWXlpBdw9kA8EYcTSI7JtnQ4QpiNSgV+Ia6EtPKmTktvmTpXCFElSaIhhKhS9IUGLoYlcWZ/PJfPpEm3KFHtqVTgX8+Vuq19CGntIyuUCyGqDEk0hBCVnqIoxEakc3Z/POePJKLLN1g6JCEswspaTe0WXjTs5E9APVdUKpWlQxJCiDJJoiGEqLQyknI5sy+ecwfjyUzOt3Q4QlQqzp5aGnb0o0EHPxlELoSolCTREEJUKnqdgYhDiZz+O5a4CxmWDkeISk+lgqBG7jTs6E+t5p5orNSWDkkIIQBJNIQQlURmch4nd17h9N448nN0lg5HiCpJ62hN/ba+NOzkh0eAo6XDEULc4yTREEJYjKIoRIencmLnZaJPpiDvRkKYT0A9V5p1D6JWM09UahnLIYS4+yTREELcdfk5Ok7vjePkritkJuVZOhwhqjVnTy1NuwbSqJM/NjJjlRDiLpJEQwhx1yRFZ3Fix2UiDiWg1xktHY4Q9xRrWw0NOvjRrFsgrj72lg5HCHEPkERDCFHhok6mcOTPKGIj0i0dihBCBcFNPGjeLYigRu6WjkYIUY1JoiGEqBBGo8L5fxI48mc0KZdl1W4hKiN3fweadQukQQc/ma1KCGF2kmgIIcxKrzNwZm8cRzdHy9oXQlQRjm62hPaqQeP7/LGy0Vg6HCFENSGJhhDCLAry9JzceZlj2y6Tl1lo6XCEELfBztmG0J5BNOkcgI1WBo4LIe6MJBpCiDuSk1HA8W0xnNx5hcJ8g6XDEUKYgdbBmmbdA2nWPQhbmalKCHGbJNEQQtyWvOxC/tkUxcmdVzDIDFJCVEs2dlY07RpAaI8aaB2tLR2OEKKKkURDCHFLCvL0hG2O5tjWGHQF0oIhxL3AylZDk/v9afFAMPbONpYORwhRRcgUE6JKUalUN3yMGjWqRDknJydat27NqlWrTPVMnDgRlUpF7969S5xjypQpqFQqunbtCkDNmjVveM6r5ao7XaGBfzZF8tP7ezn8R6QkGULcQ/QFBsK2xPDzB/s48PtFCvP1lg5JCFEFSMdLUaXExcWZ/r98+XImTJjA2bNnTdvs7OxM/1+4cCG9e/cmPT2dL7/8kkcffZQ9e/bQoUMHAPz8/Ni+fTuXL18mMDCw2HE1atQwPT906BAGQ9GX6r179zJ48GDOnj2Ls7MzADY21fvunkFvJHz3Ff7ZGEWuDPIW4p6mKzBweEMk4buu0PqhmjTuHIBGI/cshRClk0RDVCm+vr6m/7u4uKBSqYptu56rqyu+vr74+voyd+5cli1bxrp160yJhre3N61ateLHH3/kvffeA4oSieTkZB599FFOnToFgJeXl6lOd3d307Gurq4VcYmVhtGocHZ/HIfWR5KVKtPUCiGuycvSsXt5BMe2Xab9gNqEtPZGpVJZOiwhRCUjtyHEPcHa2horKyt0Ol2x7U8//TSLFi0yPV+wYAFPPvlktW+lKM+Fo4ks/egA2xafkSRDCFGmzKQ8/vohnF8/P0zMmVRLhyOEqGQk0RDVXkFBAZ9++imZmZn06NGj2L5+/fqRmZnJrl27yMnJYcWKFTz99NMWitTyki9ns2baETbNO0l6Qq6lwxFCVBFJ0Vms+yaMdd8eJSk6y9LhCCEqCek6JaqtYcOGodFoyMvLw8XFha+++oo+ffoUK2Ntbc3w4cNZuHAhFy9epF69ejRr1sxCEVtOXnYhB9Zd4tSeWBSjTER3L/jz6BKOXdpDQno01hpbavs2YmC75/FxDTKVCbu4mz2n1xOTfI6c/EzeHjyPQM+QcuvOLcjm94M/cCxyD7kFWXg4+fFIhxdpXKMdAIcitrD2wPcU6vPpUL8Pgzq8YDo2JSuemRve5M1H5mBn42D+CxcVKuZ0GjFnDlGvjQ8dBoXg6GZr6ZCEEBYkiYaotqZNm0bPnj1xdnbG29u7zHJPP/007dq14+TJk/dca4bBYOTkjisc2nCJglyZReZecj72OJ0bDyDYqwEGxcDvB39g5oY3ef+xBdhaF02qUKjPp45vY1rW7sySXV/fVL16g46ZG97Eyc6VZ3p9iJuDF2nZidja2AOQnZfBkp1TGd71TTyd/Ziz8T3q+jenSXB7AJbv/oaBbZ+TJKMqU+DcwQQuHUum9UM1ad4zSAaMC3GPkkRDVFu+vr6EhJR/97Vx48Y0btyY48eP88QTT9yFyCqHqPAU/v41grR46SJ1Lxrdd3Kx58O7vsk7iwcTkxRBiH9Rq17ber2AolaGm7Xv7CZyCzJ5beB0NJqijxh3Jx/T/uSsOLQ2DrQK6QZAPf9Q4tOiaBLcnkMRW9GorQmtff8dXZuoHHQFBvatvsDpvXHcP7QuNRp5WDokIcRdJomGEMC2bdvQ6XTVfiYpgPSEXPb8GkHUyRRLhyIqkfzCHADstU53VM+JyL3U8m7E8j3TORH1N45aV1qHdKdX6OOo1Rq8XQLQ6QuISY7A3dGHqKSztG/Qm5z8TDYcXsTY/lPNcTmiEklPyOX36ceoHepFp0dDcPawK/8gIUS1IG2ZQgAODg7VPsnQFRj4e+V5ln58QJIMUYyiKKzcN4c6vk3wd691R3WlZMVx9NIuFMXAS30+58GWT7L1+K/8efQXAOxtnfhft7dYvP0Lvlw9mrb1etEoqA2r98+jS5OHScmMZ/JvL/DZimc4enGnOS5PVBIXw5JYOvEAhzZcwqAzWjqcCjVq1CjToq5WVlbUqFGDl156iYSEBBo3bszzzz9f4pg333yT4OBgMjMzWbRoUbGFYX18fOjfvz/h4eHFjtm1axf9+/fH398flUrFmjVr7tIVCnFzpEVDVFmjRo0yrQT+X4py4wHNEydOZOLEiWXu/+abb0rd3rVr13LrroyiTqawc8lZmapWlGrFnunEplxk/MBv77guo2LEyc6NYZ1fRa3WUMOrHhm5KWw9toI+rUYA0LzWfTSvdZ/pmHOxYcSmXuKxTmOYuGwET/V4D2d7d75cPZoQv2Y42bndcVyictDrjBz8/RJn9sdz/6N1qdnM09IhVZjevXuzcOFC9Ho9p06d4umnnyY9PZ3FixfToUMHHnnkEXr37g3A/v37mTZtGn/99ZdpMVhnZ2fOnj2LoihcuXKFN998k759+3Lu3DnTFOw5OTk0b96cp556isGDB1vsWoUoiyQaQlRjuZmF7FlxjojDiZYORVRSK/bM4ETUPsYNmIabo1f5B5TDxd4DjdoKtVpj2ubrWoPM3FT0Bh1WGuti5XWGQlbs/paR3d8hKfMKRqOBuv7NAfB2CSQy4TRNa3a847hE5ZKZlMeG2cep2dSDzsPq4+SutXRIZmdra2taUDYwMJChQ4eyaNEili5dynvvvcezzz7LyZMn0Wq1PPXUU4wePZpu3bqZjr9+QVo/Pz/Gjx/PgAEDOHv2LE2bNgWgT58+JWZTFKIyka5TQlRTp/6OZcnE/ZJkiFIpisKKPdM5dmk3r/T/Ck9nP7PUW9unMUkZVzAq17rGJGZcxtneo0SSAbDpn59pVKMtQV71MCpGjIrBtM9g1BerR1Q/kSdSWPrxAU7suFwlW4tvxtVuVD///DPW1kV/A++99x52dna4ubnx/vvvk5GRwcKFC0s93tXVldmzZ7NkyRIAmjVrZupS5eDgQN26dUu07l/fdausB9xc16vS6mrfvr35XiBRrUmiIUQ1k56Qy5ppR9j+0xmZslaUacWe6RyK2MKoHu+htbYnMzeVzNxUCvUFpjI5+ZlcTj5PfFoUAAnpMVxOPk9m7rUVoBdvm8zaA9+bnt/feAA5BZn89vcsEtJjOBm1n7+OLqFz44ElYohLjeTIhR30bT0KAB/XGqhUKvae+YOTUftJSI8m2Lt+Bb0CorLQ5RvYtewca74+Snpi9ZkFb/369Tg6OvLzzz8DkJiYyJgxYwCwsrLilVdeAWDGjBk8++yzpi//V2VkZODo6EhGRgajR49m2bJlDBgwAICFCxcSFxdHeHg4s2bNIjs7G4Dt27cD8O233xIXF2d6XH/M9duudr2aOXPmDa+ld+/exY79448/zPESiXuAdJ0SopowGIwc/TOawxsjq/1AS3Hndp9aB8C3v79abPvwrm/Qvn5Rv/ETUXv5eceXpn0Lt34KQJ9WI+jbeiQAqdmJxb4guTl6M/qhL1i1bw6f//Ycrg6edG3yCL1CHy92HkVRWLrrax7p+JJp3Q4bK1uGd32TFXumozfoeKzTGFwd7rw7l6gaYiPSWf7JQdr2r03znkGo1aryD6rEunXrxpw5c3j11Vc5ceIESUlJpKWlmfYHBRUtjtmrVy9q165d4ngnJyeOHDlCixYteOSRR/j777+ZO3cu69atw9XV1dStqmbNmjzwwAOoVCrmz5/PxIkTcXNzw8XFpVh91x9z1c12vbq+G5gQt0ISDSGqgfhLGez4+QwpV3IsHYqoIma+sLXcMu3r9zYlHWUZN6DkQn61fRvz+qAb3yFVqVS8+vD0EtubBnegaXCHcmMT1ZNeZ2TvqvOcP5JI9xEN8PB3tHRIt83BwYGQkBDc3d1p3rw5Fy5c4JtvvmHcuHEEBgaayllZlf5VTK1WExISgkajoVu3btSuXZuhQ4fe8Jz5+fls3ryZxx57zKzXsmPHDry9vXF1daVLly589tlnN1wIV4irpOuUEFWYwWBk/5oLrPryiCQZQohqIzEykxWTDhVNhWuoHi20M2bMwGg08sYbb9zW8ePHj+fYsWPllouMjLyt+svSp08ffvnlF7Zt28bUqVM5dOgQ3bt3p6CgoPyDxT1PEg0hqqjUuBxWfvEP/2yKQjFWz0GUQoh7l1GvcPD3S/z6+WGSorMsHc4d69q1K/Xq1WP58uWcOnXqlo93dnbm2WefBa5N4Z6dnU1YWBhhYWGmcnFxcURHR5slZoChQ4fSt29fmjRpQv/+/dm4cSPnzp1jw4YNZjuHqL4k0RCiilEUhWPbYlgx6VC1+PAVQogbSbmczW+TD3P4j8gqf1NlwoQJqFQqxo8fX2y7s7Mz2dnZGAyGYtsNBgPZ2dmm8RZjx44FYO/evQAcPnyYFi1a0KJFC9Mx33zzDRMmTKiwa/Dz8yM4OJiIiIgKO4eoPmSMhhBVSHZaAdsWnyLmdFr5hYUQopowGhUOrLtIzOlUej7VqNKvu7Fo0aJStz/xxBM0bdqU0NBQmjcvWi/G1dWVmTNnYjAYGDhwYLGWgqvjN7Taouu9mkDEx8cD1xaRXbNmDYMGDcLZ2ZmoqCjCwsJQqVSkpaXh6upaIo5Vq1YxadIkzp8/j06no27dujd9bSkpKcTExODnZ54psUX1Ji0aQlQREYcTWPbJAUkyhBD3rNiIdJZ/epDz/1Td9YGaNm3Kk08+yYwZM0zbnJ2d6dOnD3v37qV169YcOHCA5cuX07BhQ7p3784DDzxQrI6VK1dy5swZoqKi2Lx5M1OmTAFgzpw5pSYW/+Xu7s57773Hli1bWLZsGb169QJg48aNhIWFmbpeZWdn8/rrr7Nv3z4iIyPZsWMH/fv3x9PTk0GDBpnpFRHVmbRoCFHJFeTq2Ln0HBGHEiwdihBCWFxBrp4/vztJdLgf9w+th7WtpvyDKplPPvmEFStWFNu2bNkyOnTowOnTp+nSpQuBgYH069ePiRMnotEUv8a8vDwaNmyIVqslICCAGjVqAEUtJjeja9euQNFsUlfX5gCYP38+8+fPZ+TIkSxatAiNRsOJEydYvHgx6enp+Pn50a1bN5YvX46Tk9MdvALiXiEtGve4rl27Mm7cuBLb16xZU2xu/MLCQr788ktatmyJg4MDLi4uNG/enPfff5/Y2FgmTpxY7iqkkZGRhIeHM3jwYGrWrIlKpeKbb74pM7a9e/ei0Wjo3bv06TXHjh1Lq1atsLW1JTQ09A5ficrpytk0ln1yUJIMIYT4j9N741gx6RCJUZmWDuWGFi1aVGLF7eDgYPLz84uthu7s7EybNm3o2bMneXl5REREMG3atBLrYQwcOJBVq1ah1WqJiIjg/PnzpsX/yqIoCg8//HCJ7V27dsVoNLJlyxbs7e3566+/UBTF1PXLzs6OP//8k8TERAoLC4mKimLRokWmNUCEKI8kGqJcBQUF9OrVi0mTJjFq1Ch27drFP//8w5QpU0hJSWHGjBm8/vrrxVYNDQwM5OOPPy62LSgoiNzcXGrXrs3kyZPLXfxnwYIFjBkzhj179pQ6g4aiKDz99NPlziteFSlGhYPrL7H2m6Nkp8kUgkIIUZr0hFxWTvmHI39GFfvSXpVdXVH86uPRRx8tUWbQoEGEhoby4Ycf3vZ5rq48bmNjQ9++fZkxY4apC5UQ5iJdp0S5pk2bxp49e0yzW1wVEhLCgw8+iKIoqFQqHB2vLayk0WhwcnIqkUy0adOGNm3aAPD222+Xec6cnBxWrFjBoUOHiI+PZ9GiRSVm0Zg+vWixr6SkJI4fP37H11lZ5GYWsnlBOJfPyFgMIYQoj9GgsG/1haKB4qMa4eBqa+mQ7sjVFcWvcnBwKLXcF198Qffu3Xnttddu6zxOTk6EhYWRnZ3N1q1befXVV6ldu7apW5UQ5iAtGqJcS5cupVevXsWSjOtd38XKXJYvX079+vWpX78+w4cPZ+HChdXmbtWNXD6bxvJPD0qSIYQQt+jymTSWf3aQy2er9vvn1RXFrz7Kmt2pc+fOPPjgg7z77ru3dZ6rK4+Hhoby2muvMWTIED7//PM7CV2IEiTREOU6d+4c9evXL7Zt0KBBpmbdjh07mv2cP/zwA8OHDwegd+/epjsu1ZWiKBxat4aNcw6Tm1lo6XCEEKJKysvSse7bMI78FWXpUO6KyZMn8/vvv5vW1bgTiqLIat/C7KTrlLgp/221mD17Njk5OUyfPp1du3aZ9Vxnz57l4MGDrFq1CiiaR3zo0KEsWLCAnj17mvVclUFedhYbZ3zFpbB/8K/fmtT8zpYOSQghqizFqLBv1QUSL2XSfWRDbLTV96tOaVPlXu/EiRMlZocKDQ3l888/p3Xr1tSpU4fCwkL++OMPFi9eXKzLlhDmUH3/+sRNcXZ2JiMjo8T29PR0nJ2dAahbty5nzpwptv9qU667u7vZY/rhhx/Q6/UEBASYtimKgrW1NWlpabi5uZn9nJZSeCWbvUsWcynsHwBizx6mRrNAEmNqWzgyIYSo2i4cTSI1Loc+LzbFzbf0cQ7VQWlT5V7VuXPJG1eKopCTk8PLL7/M5cuXsbOzo0GDBvz888/VcnIVYVmSaNzjGjRowMaNG0tsP3TokKm71LBhw3j//fc5evRomeM0zEWv17N48WKmTp1aYoGiwYMH88svv/B///d/FRrD3ZJzJIH01edpoG3JJZcjZGQUTWF7OXw93iEjSU90KacGIYQQN5IWn8uvkw/TY0RD6rT0tnQ4pbp+FfGyVhS/0f6rU+Ve7+qK4WX59NNP+fTTT28lTCFui4zRuMe9/PLLXLhwgdGjR3Ps2DHOnTvHrFmz+OGHH3jjjTcAGD9+PB06dKB79+58++23HDlyhEuXLvHnn3+ycePGEgsJ3UhhYSFhYWGEhYVRWFjIlStXCAsL4/z580DRtH5paWk888wzNGnSpNhjyJAh/PDDD6a6zp8/T1hYGPHx8eTl5RWrtzJTDArp6y6QtuIcis6IkqXnwUbPoFYXvY5Gg56sxLXYaPUWjlQIIao+Xb6BTfNPsnfVeYzG6j+piBCViSQa97iaNWuye/duLly4wAMPPECbNm1YtGgRixYtMs3drdVq2bp1K2+//TYLFy7kvvvuo2HDhowbN45OnTqVWIjoRmJjY2nRogUtWrQgLi6Or776ihYtWvDss88CRd2mevbsWWKBIihq0QgLC+PIkSMAPPvss7Ro0YJ58+Zx7tw5U72xsbF3/sJUEEN2IUnfnyB7b/EYVfEGend4yfQ8Jy0ZO/vtKMiHohBCmMPRv6JZ920YeVmV+2bUzYiPj2fs2LGEhISg1Wrx8fHhvvvuY+7cueTm5prK7du3j+7du+Pg4ICrqytdu3YlLy/PtP/6RXWdnJxo3bq1aXwkUGwxXisrKzw9PencuTPffPONDBwXN0Wl3AtzhgpRCRTGZJHy8ykMGWV/yF10PsWhY7+bntdo1pfEmPpllhdCCHFrHN1seeilZnjVcCq/cCV08eJFOnXqhKurKx999BFNmzZFr9dz7tw5FixYwAsvvMCAAQPYt28fvXv35p133qF///7Y2Nhw7Ngx+vfvj61t0VojKpWKhQsX0rt3b9LT0/nyyy9ZtGgRe/bsoUOHDkycOJHffvuNLVu2YDQaSUlJYceOHXz66acEBgayY8eOEoPNhbieJBpC3AW5J5JIXX4O9MYbllPZqNmdvYYrV04XPVer8as/gtR48w+6F0KIe5WVrYZeTzWidqiXpUO5Zb179yY8PJwzZ86Uupjf1UV027dvT69evfjkk0/KrEulUrF69WoefvhhAHQ6Ha6urrzyyit8/vnnTJw4kTVr1hAWFlbsuDNnztC8eXPeeOMNGeshbki6TglRwbJ2XSZ1yZlykwwApdDI/b6D0doVzfilGI2kXVmD1qHqN/ULIURloS8wsHHeCY78WbXW20hJSeGvv/5i9OjRZa4YrlKpSExM5MCBA3h7e9OxY0d8fHzo0qULe/bsuWH91tbWWFlZodPpbliuQYMG9OnTp1g3KyFKI4mGEBVEMSqkrTlPxh+XuJWhFkqqjr4tr43XyMtMx9pqM6ik8VEIIcxGgX2rL7D1x1MYbuJGUGVw/vx5FEUpsYiup6enaRHdt956i4sXLwJFYyyee+45Nm3aRMuWLenRowcRERGl1l1QUMCnn35KZmYmPXr0KDeWBg0aEBkZecfXJKo3STSEqADGQgMpi0+Rsz/uto63ioUe7UeZnidFnsU74KSZohNCCHFVxKEE5vw5j4yCkmtKVVb/XUT34MGDhIWF0bhxYwoKCjAaixKnF154gaeeeooWLVowbdo06tevz4IFC4odO2zYMBwdHbG3t+frr7/mq6++ok+fPuXGcLWLlhA3IomGEGZmyCokad5x8s+k3lE9nkm+NKp/bbGl6JNb8PBPutPwhBBCXCe7VTjzkmcz/I/hxGTGWDqcGwoJCUGlUpVYRLd27dqEhIRgZ2cHXFtUt1GjRsXKNWzYkOjo6GLbpk2bRlhYGHFxcaSmpvLaa6/dVCynT5+mVq1at3sp4h4hiYYQZqRLyCFxVhi6K9l3XplRoZn1/Xi4BxY9VxSSI1dj75x/4+OEEELcFG2LVH5UzwMgMjOS4RuHczzpuIWjKpuHhwe9evVi5syZ5OTklFmuZs2a+Pv7c/bs2WLbz507R3BwcLFtvr6+hISE4O198wsanjlzhk2bNjF48OBbuwBxz5FEQwgzyb+QTuKc4xjSzTe3uJKjp2fdkWg0NgAU5GSjMmxCpaka/YmFEKKycmlgYLr242LbUvNTefavZ9kRs8MiMd2M2bNno9frad26NcuXL+f06dOcPXuWn3/+mTNnzqDRaFCpVLzxxhtMnz6d3377jfPnz/PBBx9w5swZnnnmmVs6n16vJz4+ntjYWE6cOMGMGTPo0qULoaGhpoV9hSiLlaUDEKI6yPkngbRVEWCogAHbiXoe6vASv+/5FoCUyxcJanKUpCutzH8uIYS4B7gEWTHb7V2MpczUkafPY9z2cbzX/j0erfeoBaK7sTp16nD06FEmTZrEO++8w+XLl7G1taVRo0a8/vrrvPzyywCMGzeO/Px8xo8fT2pqKs2bN2fz5s3UqVPnls4XHh6On58fGo0GFxcXGjVqxDvvvMNLL71kWo9DiLLIOhpC3KHMLVFkbokuv+AdinKNYP/Ra1MJBjUdRtJlvwo/rxBCVCcOHtb8Wm8y0ZT/vj2lxav0afbUXYhKiOpJuk4JcZsUo0Lqr+fuSpIBEJxTn+AazU3P4yNW4+iad1fOLYQQ1YGtvYad9RfeVJLR3LkOXde/CxteA6N0VxXidkiiIcRtUPRGUn85Te4/CXfvpDojHTz64+DgVvQ0Px993no01oa7F4MQQlRRaisV50O3cFg5VG7ZOo6BzIoIw64wFw59DyufBsONF7ETQpQkiYYQt0jRGUhefIq88JS7f+50HX2avwD/zl2eHh+Dq8fBux6HEEJUNXmtzrBBWVNuOR87T+bGROGSm3ZtY/hqWPo4FOZWXIBCVEOSaAhxC4wFepIWnKTgXFr5hSuIJk6hV/tnTc+vnN6Hd9Dd6b4lhBBVkTY0jYXq2eWWc7FxZl5yNr7pV0ruPL8Ffn4E8jMrIEIhqidJNIS4ScZcHUnfnaDwkuU/ZNwT3GnWqIfp+ZXT63D2NMPaHUIIUc241Dcy8z/T2JbGTqNlZq4VdRLPlV0oeh/82B9y72xBVksYNWoUKpUKlUqFtbU1Pj4+9OrViwULFphWEoeiNTiulrO3t6dJkybMmzcPRVHo2bMnDz74YIm6Z8+ejYuLC9HR0ezYscN0vFqtxsXFhRYtWvDmm28SFxd3Ny9ZVAKSaFQje/fuRaPR0Lt372LbIyMjTX/0KpUKGxsbQkJC+PTTT/nvpGPnz5/nqaeeIjAwEFtbW2rVqsWwYcM4fPhwsXLbt2+nX79+eHl5odVqqVOnDkOHDmXXrl2mMjt27GDgwIH4+fnh4OBAaGgov/zyS8W9ABXIkFVI0vzj6C5Xki/zCjSiHV5eNQEw6ArJT1+HtY3esnEJIUQl4hJoxRz3D9CrbjyY20plxVeKJ6ExYeVXGhcGi/pCdpJZYrybevfuTVxcHJGRkWzcuJFu3boxduxY+vXrh15/7fPj448/Ji4ujuPHj/Pwww/z4osvsmLFChYuXMiBAweYN2+eqeylS5d46623+Pbbb6lRo4Zp+9mzZ4mNjeXQoUO89dZbbNmyhSZNmnDixIm7es3CsiTRqEYWLFjAmDFj2LNnD9HRJbvSbNmyhbi4OCIiIvjoo4/47LPPWLBggWn/4cOHadWqFefOnWPevHmcOnWK1atX06BBA1577TVTudmzZ9OjRw88PDxMiwX99NNPdOzYkfHjx5vK7d27l2bNmrFy5UqOHz/O008/zYgRI/j9998r9oUwM0Pmv0lGfOXqm6vkGehe60msrbUAZCbF4+jyt4WjEkKIysHB3ZplQVPIpPwbRBNta9H5wt6brzzxFJG/TyYhM/8OIrz7bG1t8fX1JSAggJYtW/Luu++ydu1aNm7cyKJFi0zlnJycTCuGf/rpp9StW5c1a9YQFBTEt99+y+uvv86lS5dQFIVnnnmGHj16MGrUqGLn8vb2xtfXl3r16vH444/z999/4+XlxUsvvXR3L1pYlCQaFhQfH8+YMWOoXbs2tra2BAUF0b9/f7Zu3Vqs3KRJk9BoNEyePLnMel588UUWLFjA7NmzMRqN9OrVq0Q9PXv2JDIykuDgYJ588kk6duzI9OnT6dq1K4qiMGrUKOrWrcvu3btp3rw5DRs25PHHH+fDDz9k7dq1AERHRzNu3Dg6dOhAREQE/fr1o0WLFnTs2JGxY8cWa/moX78+f/31F3369CE0NJR58+YRHBzM6tWrzfxKVhxDRgFJ84+jT6qk08gm6Xmo3cump3Hn/sE76KIFAxJCCMuzsdOwu+GPRBFVbtlxTo0ZeHprueWuF+/fiwdPdOXx+fuJz6haycZ/de/enebNm7Nq1aoyy2i1WnS6olm3Ro4cSY8ePXjqqaeYOXMmJ0+eZP78+eWex87OjhdffJG///6bxMREs8UvKjdJNCwkMjKSVq1asW3bNqZMmcKJEyfYtGkT3bp1Y/To0cXKLly4kDfffLNY68N/6/n999+pVasWJ0+e5MsvvyQpKalEPba2trz11lvEx8czdOhQduzYwYkTJ9i/fz9dunQhPDyc1157jdq1axMUFIROp+Ps2bOoVCqaNGnCxIkTCQ4ORqfTsXfvXvbt20deXh4ZGRmmbllRUVFMnDgRlUrFkCFDOHToELa2tnTt2pXx48eTn5+Pu7t7hb625qJPLyBx/nH0yZU0yfiX9oo197V+3PT8cvh6XL0zLBiREEJYjlqj4lKLbRw0Hii37P9cm/HM8Y23VH+Sfzd6RI2kwKjmUnIOQ+fvIza9cn9OlKdBgwZERkaW2K7X61m0aBEnTpygR49r4wLnz5/PqVOnGDduHPPmzcPb2/umzwOUei5RPUmiYSEvv/wyKpWKgwcPMmTIEOrVq0fjxo159dVX2b9/v6nczp07ycvL4+OPPyYnJ6fYGIjr6wkKCuKVV16hXr16vPDCC6W2gOj1enbt2oWfnx8rVqyge/fu/O9//6Np06bUrFkTuPYm4ObmxpIlSxg9ejRDhw7l6NGjvP7664wYMQJHR0fi4uIIDAxk0KBB2NnZmR6NGzcmISEBDw8POnTowOXLl1m5ciVRUVFMmTKFlJQUnnqq8q+yqk/LJ2n+cQwpVeNOVUBGTWrXagWA0aAnK3EtNloZryGEuPfkt4rgd6Xsu/NXPeTWhDeObrilulP97qdr9NPkGK59fYpKyeXx+fu5UoWTDUVRUP07bTrAW2+9haOjI3Z2dowePZo33niDF154wbTf29ub559/noYNGzJo0KBbOg9Q7FyiepNEwwJSU1PZtGkTo0ePxsHBocR+V1dX0/9/+OEHhg0bhrW1NcOGDeOHH34oUc9jjz3GP//8w+OPF93VtrKyYujQoaxYsaJYvaNGjSIgIAArKysWLlzIiRMnCA8Px8HBgb59+wJFf/z5+fno9Xoee+wxXnrpJdavX49Wq8XR0RF7e3s0Gg2+vr5oNBq0Wi02NjYcP36cjRs3kpubi6Io2NjYcP78eVJTU7n//vvp2rUrERERTJ8+ncaNG1fAq2o++tR8kuYdx5BaNZIMAPQKbZ174+TkCUBOWjJ29tsB5cbHCSFENWIXmskCzYxyy3VybcCnYZtR3cJ7ZLpvB7pdfo4cvabEvujUXB6fv4+4jKqZbJw+fZpatWqZnr/xxhuEhYURFRVFdnY2U6ZMQa0u/pXRysoKKyurWz4PYLq5Kao/STQs4Pz58yiKYmo9KEtmZiYrV65k+PDhAAwfPpzffvuNzMzMYvVcvHgRvV5vSiKsrKyYM2cOq1atIi3t2noPWq2W2NhY1Go1Go2GcePGceTIEYxGI/Xq1QOK3gSys7Np0aIFGo2Gxo0bExISwvLlywGoW7cuGRkZxMfHF4s1JCSE4OBg03NPT0/atGlD06ZN8fLyYubMmahUKh599NE7fwErkCGzkKTvT2BIL7B0KLdMydTTu+lzqFRFf9YJF07gHXTWwlEJIcTd4VJPYYb2w3LLNXWuzdfhf2NtvPmVvrO8W9Mj9kUydGV/sY5JzePJ7w6QmFWFblIB27Zt48SJEwwePNi0zdPTk5CQEPz9/c3W+pCXl8f8+fPp3LkzXl5eZqlTVH6SaFjAzTYdLlmyhNq1a9O8eXMAQkNDqV27NsuWLStWz44dO5g6dSphYWGmx7FjxwgODi42nWx6ejqKovDII48wYcIEFEUxPUJDQ2nUqBFffPEFOTk57Nu3D0dHRxwdHTl9+jSffPIJAEOGDMHa2povvvjihrGHh4ezc+dObGxsSE5ORlEUrK2t6datG7m5lWv2pquMuTqSfjhRtVoy/kMdZ+TBjteat2NObMLdr+rN9y6EELfCOcCKuZ4Typ3GtqaDP7POn8C+MOem6872akGPhDGkFFqXW/Zicg7Dvz9Aak7hTdd/NxUUFBAfH8+VK1c4cuQIkyZNYuDAgfTr148RI0aY9VyJiYnEx8cTERHBsmXL6NSpE8nJycyZM8es5xGVmyQaFlC3bl1UKpWpCbEsCxYsIDw83NRKYWVlRXh4uKn71NV6srOzeeaZZ2jSpEmxx5AhQ4p1tcrLK2rSve+++8jIyOCLL74gMDAQjUaDSqVi4cKFnDlzBgCdTkdeXh55eXkUFhYSHR3NqVOnqFGjBlOnTuXbb78lOTmZhIQEjEYjR44cYfr06UBRAlW/fn3mzZuHWq2mS5cuNGnShD///JNTp04Vi6myMBYaSF4Ujj6hciZB1/vxyGo6zn2MkK968tCiZzkQc6zYfpc4Z1o0KVpL5XjMFT6eP4a3Fw/i9QX9+Wr1/3Eq5lCx8qcvH+ajZSN4feEAFm+fjN5w7S5fXkE2Hy0bQWpWQsVfmBBC3AZ7N2t+Df6KDOXGi6l6az2YdyUWt5yUm64717MpvZLGklhQfpJx1bmEbIZ/f4CM3JtvMblbNm3ahJ+fHzVr1qR3795s376d6dOns3btWjSakl3C7kT9+vXx9/enVatWTJ48mZ49e3Ly5EkaNWpk1vOIyk2l/HfFNnFX9OnThxMnTnD27NkS4zTS09OJiYmhefPm7Nixo9gsTenp6XTu3Jnjx4/TpEkTvL29ycrKIjk5uUQ9O3fupGvXrvzzzz+0atXKtF2lUuHi4kJ+fj6DBg0iNjaWHTt2ANCkSRPOnTuHjY0NBQUFeHl5ERoaSlJSEl26dOGrr74CitbkGDBgAAaDgcLCQnx8fOjQoQMvvvgi+/btY82aNYSGhvLjjz+WuPaQkBAiIiLM9VLeMUVvJHlROAXn0y0dSrnWnd7KuPWf8dkDr9I6oAm/hK1j6fENbHt2MQHOPqZyKq2GnWm/MnfjGpzttLRu1hL0D7DvzJ9sPf4rrw+aSZBnXYyKkXcXP0qvFo/TKLAN32/+iM6NB9KlycMALN01DW/XQHo0q9xd3oQQ9yYbrYYDrX7hgHHfDcs5WTuyKMNIvYQzN113vkcjeqa8weV829uKrUUNV35+ph0Otrc2jkGI6kRaNCxk9uzZGAwG2rZty8qVK4mIiOD06dNMnz6dDh068MMPP9C2bVs6d+5crJXivvvuM+0HOHDgAK6urqXW8+KLL6IoCi1btiQ4OJhp06bRu3dv/P39uXTpEgEBAcXWtNi9ezfh4eF4eXnx6aefotPpiI2N5Y8//uD5559n8eLFpnm069Wrh6urK927d8fR0ZFNmzbx4Ycf0qlTJ6BobQ9vb2+2b9/OxYsXWbduHSqVCltbW9avX3/3X/AyKEaFlKVnqkSSAfDdoRUMbdaXYc37UdezJhN7voK/kxc/HV1TrJySb6BL4FAebdeGbg3q4FSYQePG6Qxo9yxeLgGcjCr6UM7JzyA7P53OjQbi516TpsEdiE8rmnf+QvxJopPO0a3JI3f7MoUQolxqtYqoFjvKTTJsNbbMyLe7pSSjwK0+D6W9fttJBsDR6HSe/+kwBXrDbdchRFUniYaF1KpViyNHjtCtWzdee+01mjRpYlpk79tvv+Xnn38uNjDreoMHD+bnn3+msLDwhvWU1g/yaoLTqVMn+vfvT35+Pjk5OUyfPp2BAwfSqFEjrK1LNhE//PDDpKammlb1njBhAnFxcWzatMk0eLxFixamBfscHR25ePEiI0aMoEGDBjz33HN4e3vTsmVL6tevb8ZX8vYpikLaygjyw2++Gd2SCg06TsSfo3OtNsW2d67VhsNXTpYor6To6Nv22loq0Se34OabQIEuD3tbZwActa4423tw+vJhCvUFXIg/QYBHbfQGHct3f8PjncehVpu3OV0IIcyhsPUF1vLbDctoVBqmqHxpFf3PzdfrWof+mW9yMVd7pyHy9/kU/m/JUfSGG48dEaK6kq5T96C4uDg+++wz1q9fT1xcHF5eXrRq1Yrx48fTtWtXatasybhx4xg3btwN6ymr3MSJE1mzZg1hYWHFtu/du5dOnTqxf/9+2rVrZ96Lug3p6y+SveeKpcO4afFZybSZ/Qirn5xF68Cmpu0z9v3Ebyc3sfO5X0o/ziuWnQd/AmD3xctsOXmR9x9biJOdGwAX4k6wct8ccvIzaBTUliEdR/NX2BJyC7Lp2OAhlu76muz8DLo0GWTqUiWEEJZk3zyLr+3fL7fch/b1GRK++abr1bnU5OHc9wjPKjn1/J0Y1CKArx9rLutHiHuOJBrinpS5NZrMzVGWDuOWXE001gyfTauAJqbt0/cuZlX4X+x47ufSD9SoOMpOlm/7jRWHTvDKwH7U8f0/jIbSGzQT0mOYu/E93h4yj2nrxtGt6WAaBbXhsxXPMqbfFAI86lTE5QkhxE1xqQtfe7yKTnXjLkmjnZvw4rE/brpevXMQjxV8wJEMxzsNsVTP3V+L9/rKQGhxb5GuU+Kek703tsolGQDu9i5oVBoSc4pPV5uSm4ang1vZBxoULp838uuhE/yvQwv8rPR4+BwptaiiKCzdNY1BHV7EqBi5nHyeFrU642TnRl3/ZkTEHTfnJQkhxC1x9rdivueH5SYZw1yb3VqS4RTAE4XvVViSAfDd7kt8v/tihdUvRGUkUyGIe0rO0UTSf79g6TBui43Gmqa+9dgdeZg+9Tqbtu+OPMwDde8r87g1p7bw+sbJzBrxKVl5+zAaDcSE7yKoaQBJl/2Kld135g8ctM40q9mR3IIsAAxGvelfxSj9jIUQN29X+Fq2HltBRm4Kfm41GdzxZUL8mpVZPiL2GKv2zSEuLRIXe096hg7l/kb9AbB3tWJu1jscfusw+kw9zi2d8X/KH7VV0T1TQ66BCx9dYORnD/B22M0nGQYHH0bq3+dguvOdXexN+OyP03g7axnQ3L/CzyVEZSAtGuKekXcmlbRfz0EV7iz4XJvHWHZsPcuObyAiOZKJW2dwJTOR4aEDAZi8cx7j1n9mKr/m1BbGb/iMD7qNpqV9A1o3HkpmXj55hTriI1bj5HZt3ZCsvDQ2HfmFRzsVDSC3t3XC17UG20+s4mJ8OGevHKWWb+O7e8FCiCrrn/PbWbl3Ng+2eIK3B8+jjm9TZv/xTpnr8iRnxjFn47vU8W3K24Pn8WCLYfz290yOXtyFtVbDvoZLODDvAO7d3Kn9fm3yLuaRtiPNdHz8inhCH6rPrNj9qJWbuylitPfkOSbwd5qLWa65PIoCr684xv5oWUhV3BukRUPcEwpjs0ldcgaMVTjLAAY07EFaXibf/v0jiTkp1PesxY+PfkGgiy8ACdkpXMm89iH+S9g69EYD72+exvubp5m2t64ZyONtm6PLXY/GejAGnYbf/p5Fj+aP4ergZSo3vNtb/LT9C3acXEXP5o9R07vB3btYIUSVtu3Eb3Ro0IeODfsCMKTTaE5fPszuU78zsN2zJcrvOfU7bo7eDPn3ZoevWzBRSefYdnwF/sOt2J2xE0OWAffu7qht1Di1cKIgtgCAnIgciDGyuWcK1rqbW5XbaOfOy5oP2ZZ0g66nFaB5Ey9GRl5mjbsdDR3t7uq5hbjbZDC4qPYMmQUkzgrDkHFzHz7VncpGzZ7stVy+cgqAgIbtSYnvaOGohBDVid6g49UfHuKZXh/SvFZR185d4WtZd+B78nV5BHmGlOhGNW3tOII8QxjS6f84FxvG9N9fK1GvxklDwNMBODZ25MLEC+iz9Bh1RjQaDetG+dDHs6jLZ0a+Qpvvctgywp4aLiU7bxi1roy1+YjfE71K7KsoahW0bOvPHteimacCbK3Z0KoevrY3v+q4EFWNdJ0S1Zqx0EDyj6ckybiOUmjkPp9H0NoV9Ue+cno/3kHRFo5KCFGdZOdnYFSMpmm0r3ajqh/QAncn71K7UWXmpZrKX/XiU68DUOejOtT/pj5Bo4NIWpdExLsRFCYV4tnHE7/7fXFQFP4+lkSnBTnUn5lN3yU5vNjautQkQ7F14g3bD+9qkmFrpabxfYGmJAPgSoGO4ccvkiML+olqTBINUW0pRoXUZWfRXcm2dCiVjpKmo2/Ll0zPr5xeh/O/dwKFEMLcrnajCvKqj7XGhiGdRuPm6M3uU78XL/ifdSb+clsJgJWLFdau1jg2cKTOh3WoM6EOik7Bu4U3VmFpPNVUzbwjhbzQypqpvWw5cMVIt5olFxtVbBx4x34iKxN8Kuxa/8vFzprA+wM4ZF+yA8nJ7DxePBWFUTqXiGpKEg1RbWX8cYn8U1Vj1W9LsIqFHu2fAsCgKyQ/bR3WtnJnTQhx5xy1LqhVarLy0tAbdMQknaNhYGuy89JMrRYNA1txKSHcdIyznTtZuUWDpB3ci5KEqO+KWltj5saQffraTSONkwYrFytyZl7kqy4q/o4xkJwLA+pb8e62AroEa9gTXXxAuGJtz0THiSyLKz7bXkXyddVi18mHcJuyB6dvTsnk4wuxdy0mIe4mSTREtbT0YDSbr6TJb3g5PJN8aFS/aKrczOQEHJ13WzgiIUR1YKWxJsirHmcu/1OsG9WZy/9Qy6do9jonOzcyc6/NvlTLpxFnLv+DnYsVhxpuwH+UPw71HbANsEXrryVySiQ5Z3MA0Kg1DLo/mNSUHN7ckk9T76I3+yl/F9KjlhWKAl/vL6D+zGxmHixEsdLyucsEfowNuGuvQR1fR7LbeBKpKb+1Ym5MEr/Eyo0xUf3I1zBR7ey/mMKEtSd541I8i/yswbZk87n4l1GhmfX9eLgHAhB37gjeNarmOiNCiMqle9Mh7D3zB/+c3w7AjhOrSM1ONK2LcSrmMJm5qewKX8uHS55k+4mVJKTH8NmOp4k0XkKlVpF1NAvvh73xH+GPUzMnkjcmY9QZCViey6oNF8gpLBpk3amGFXXcVMz7R4efI2yLNBCbpdDcR83HOwt4LfcZ5l+ugbEghyvzn0efmVih196klhuXmrmQrL75LlFvn7vMvnTp6iuqF5l1SlQrMam5DJi5h7RcnWlbF3dHPjVoUcmA8LJ5W/Hb4S8xGApRazR4h4wiPfHuzCsvhKi+doWvZXPYMtKyE/F08mN41zcJ8S+aaeqT5U+RmpWAUTEw9L5XqOPfhN/OT+X0P+Go1Cqs3KzwesgL9+7uACSuSyRjXwb1Aj2JPRnL402seL2jLYk5CjqDkTEbC0jMUYjNUlCAQ8/a8781BWRpfclt0A/nVv1J+XMm1m4BOLcdVGHX3KqRFwcCrdH/Z7zJzXC31rCpVT1q2NlWQGRC3H3SoiGqjVxdLq//+nexJANgZ2o2w3WZFPra3/WYFh9dQ68Fo2g4rTcNp/Vm4E8vsf3C/jLLH7x8nEE/v0zTb/sRMrUnXb8bzneHVhQrs+vSITrPf4JG0/owfsNnFBquXW9mQTad5z9RbC2Nm5Ko56H2LwNgNBjISlyDjZ2unIOEEOLGOjceyCdPLiXYuwENg9qYkgwAFWBn62Baa8OnpxrN/6mw9bXFo7cH9b+qb0oyAPKj83G1tSP6+BXOjXFk5kN21HRV0zZAw+lkhVpuao684GBakzXEy4YaLTsTn6UHo5H8y6cojD+PU+sBFXa9bVr58neQzW0lGQCpOgMjT1ySmahEtSEL9olqI/n9iTzgl02q//1ExDoX2xeVW0i//ER+CfbFK+ruNU37OXnxTpcXqOlW1DXp15ObeGbVu2wc9QP1vWqVKG9vrWVky0do6FUHexsthy6f4O0/v8LeWsuToQMwKkZeWf8JL7d7ki612vLimg9Ycux3RrV8BIDPd8xleOhAApxvfUYV+1hbOrQczL4jK8lJS8HHfTuFeb0o+joghBC3r3vTISzePpkaXvWo5dOIv09vIDU7EZ2+gIaBrdl4cQ5/n9hC4POBODZxJONgBva17bENsEUxKKTvTSfzcCbt61hh8Ncw5e8Cfjquw8FaRc/aGjZE6Nn7tANe9ir8HFVo1PC/sFZsP3IeQ3YqNn4hpP41G48+Y1Gpzd+d1kqtomkHf3Y73nldp3PyGXsmmu+blPyMEKKqkURDVAsp339Pztrf6ahSUWNQAQsatWXfqcBiZXKNRgZFxTK9lh8tI3PgLnQa7BXSqdjztzo/x09H13A0NrzURKOJTz2a+NQzPQ9y8WPjuZ0cvHycJ0MHkJqbQUpuOiNaPozWypZedTsRkRwJwKHLJzgWf5ZPe42/7XhrZNcjtkZzoqKPkXDhJDWaBZEY0/C26xNCCIBWId3IKchk4z8/kZmbip97Tf7X7S1+2PwRviFurDy2lcKUou6tGmcNxjwj8cvj0aXpUNuo8ajhypr/uTBnbzZ7og1orVSsHmpPcq6RR3/No6m3mgDnok4ayx+15/GN9qz/cweo1Li0G0J+1HG0wc1RWdkQ//MbGPIycWpZ1J3qTtnbaKjRwY99WvN9qKxPymBGVAJjgu/eNLxCVARJNESVl7N3L4nTvil6oigErtrLW+2z+KF3KOuPNUNRit+Rf+VSHGMCPRmaqIfCsqccNDeD0cD6MzvI0+XTMqDJTR1zMuEc/1wJ5437nwXAw94Vb0cPdl06xP0123Aw5jhDmvam0KDj3b+m8lWft9Hcyd06nZEOHv1JTokmJyeNmBN/4tfAh9R49/KPFUKIG+jceCCdGw80PU/PSQZgvccC/J+7bjYoBTSOGupNLrrpUs+xBosiTuCUn8Gsv4uW2vjlETtctCpAw0+DYMiKPPJ0ClprNUkd3sbauwXB/1anS71C4m8T8Rs1nYQlb+PUegB2tVoRu2A02qAm2HjffsuBh6MNDm29CbM2/52ryZfiaOZkTxd3J7PXLcTdIomGqNJ0V65w5dXXwFC8P6vt/hOMjk/D/cl0Vl3oRFZ+8V/1GZeTOevpzIQCa8iq2LEIp5Mu8PBPL1OgL8TBxo7vBn1KPc+aNzymzazBpOalozcaGN/pKYY17weASqVizsCP+GjrTCZunU632u0Z2rQvs/b/TKfgVmitbBn088uk5mXwVMtHGNVq8C3Hq6Tr6NP8BX7b9wWKYiTt8iq0riPIz7a5ncsXQohSeXl7oFKrSElPxZlr3V0NWQasXIreswPsfZgbdR6n/AwA/JzUBDgZ/00yijT0VKMAlzONnGn+Oq+eb2HapygKKZtm4NbtWVAUChMuYF+/E2prLdqgJuTHnLztRCPIw56clu6cu4WZpW6Fo0bNpcg5tLMfjlbrXyHnEKKiSaIhqixjQQGXx7yCIT291P1K5GWGTU/Hb2QuywruJzKpeOfZv5IziXKwZZ63K1aJeRUWZx33Gmx66gcy87PZeG4n4zdM4tcnZtww2Vj55AxydHkcvXKKz3fOo6ZbAA836glA28BmbBg531T2YmoMK8P/ZNOoHxiyZAzPtH6UrrXa0XPBSNoFhdLQu84tx6yJU3ig/XP8tW8+eVmZOLr/RYGqb4nWISGEuB1WNmqONV+NtqaW7PBsnFtdSzSyw7NxauGEu60r8xJS8MqMN+3rFKTh13Ad2YUKjjZF70fnUoyoVXCm4f8x5nyrYufJPv4XGjtn7Ou2w5D/7/g8Y9GNKcVoAOPttWo3CHQmsqET6RWUZDS0UzHGMBGX9GOcOHmAVi2XolbLzR5R9cisU6LKip/wIfmnTt2wjJKVTZfZ23kzcwut65WcielsTgEDUpNIq2GGEXxlsNFYU8stkOZ+DXi7yws08g5hweFfb3hMDVd/GnrV4YnQ/jzb5lGm/b2w1HKKovDWpi/5oNtojIrCyYQI+tbviqeDG+2DQtkfE3bbcbsluNHs3+QmKeocXgEnbrsuIYS4SqWChFYH2KVsw/NBT9J2ppG2K4382HzilsShS9ER0CuAWVkKc5edYsTqazeCnmhqjYe9iqfW5nEqycCuKD1vbC6gz33NGRN9f7HzGHLSydi7HLeezwOg0Tpi7RFE5qG1FFw5TX7UMWwDGtxy/KF1PTjTyIn0CvoG1c85i7fzRuJScAyAzMwwIs5PqpiTCVHBJNEQVVLasuVkrF17c4WNRmot38M7+7bQu3nJxCRTb2RAdCzhtRzMHGXpFBQKDDffXUtRoFBfevllxzfgZufMA3Xvw6AU3ZnTGfWmfw3GO5giUYFGtMX7324F0Se24BlQsYtcCSHuAa2u8Cs/A+DSzgXfJ3xJXJvIhQkXyDmbQ+3XajPb1ZsmV04Ql61wIsFA/6W5+E/NwunzLN7qZEN6vkLr+Tk8uSqPVq1acLzNBFP1uWf3krDsfS7PGYUhJ42kNZ+Td/EfADweGkfO6V0krJiASm1FwvIPSN7wNcp178k3WtSvTXMfDtayJb8CGnetVSrGuYQxLGMUVkpWsX2XL/9EQsJ6859UiAomC/aJKqfgwgUuDR6Ckp9/6weH1OTnYS1YebYd+YUlew6+XsOLh2N1oDfPIPHJO+fTrXY7/J29yS7MZd3pbcze/ws/PfolnWu1YfLOecRnJfNNv/cAWHRkFQHOPoS41wDg4OUTfLR1BqNaPcKbnZ8rVndyThr9F7/AquGz8HPyAqD79/+jf4PudK7VhidXvMbSodNo4d/ojq5B5WnNyrCv0OnysbV3wNF7JDkZ2juqUwhxb3Jsms9Xjm+VuV+Fism2tXnozHbTto0ROv6OMdDST8PgFXmsHmrHww2sAThc42mGnOtZrI7ULfPROLmjrdEMta0D2Se2kHlwNX4jpmLjUwdFMXJ55v9waTcEba2WJK35HOdW/XBqWTQWrrRF/VQqaNXWnz2uFdN91MdGzetWP+CfU3YyodE40Kb1ahwcbr07rBCWImM0RJViLCzkymuv316SAXA+kv/NSMNnVA7LMu4nNs2u2O6vopM45+3CmznWkHPng8STc1IZt/4zEnNScLJ1oKFXHVOSAZCQnVJscT1FUZi8cz4xGXFYqTQEu/nzdtcXGB5acoGpD7dO54W2Q01JBsDUh97h1Q2TWPDPSl5sO+yOkwwAJVnHQ21Hs/bvqRTk5uCg24haMxCjQRpEhRA3z6W2im8c379hmTcdG/LQiU3FtvWpa02futb/PrvWjep40P9KJBkA7v92lbrKrctI8iIOkHv+IDY+dTDmZmLMzcCpZV9UVjbY121HYXIMgGlRP/deL5mOt7FS06CDP3vsK+a+bGsHIy8UvI59zqUbljMYcjhxcjRtWq9Go7G7YVkhKgtp0RBVSsLnn5P64+I7r8jKirOPd2SBVyeOX/Iosbupk5aZts5okm8zoamGrnhEs+fwUgCCGncmKba1hSMSQlQVTj7WLKo9gSSSyyzztEtTxodtuGE9qo8yWT3Ujnq9RvBQxM2tgaEoRq7MeQbndo/g3Ko/iqJwZfZI3B8YjbZmKInL3sOhaQ8cm/Yk7sfxePQZi61fXQCc7azwbu/LSZuK+ao0zCWBhzLGoVYKb/oYX5+Hadx4aoXEI4S5yS1JUWVk795N6uKfzFOZXk/9n3fx3vHN9GhyvsTuE1n5DMpIITPw7ozbqAoCMoIJ+bclJiZ8F16BsRaOSAhRFWidrFhfZ+YNk4yBbuUnGVfFenak7/l+N33+zIOrUXT5ODQoGiyuUqnwHPgWGXuXEfvDy1j71MGxaS8y9v9WbFG/hB9eJDNpR4UkGfZqFe85badf+su3lGQAxCes4fKVJWaPSYiKIC0aokrQp6RwceDDGJLL/qC6XUrDOix+tBUrw1uhMxRf7E4D/FDLn5BL2WY/b1WkcrZiw6X5ZGUmYW2rxSVgFNlp9pYOSwhRSVnZqDnReg07la1lluni2pBvjm3F6t+JLG5E9VEmXo+8i33djjd1/pxTO0nZNB2vRz7ArmZomeX+u6hfnR6PYRzUi3PPDcbty7lY16l3U+e7GbW0asbzFR55f992HWq1DW3arMPRoa7Z4hKiIkiLhqj0FEUh9p13KiTJAFCdvsBTs7fwUs3teDoVFNtnAEZdiuWvYC1oZA0JJVNP78bPoVKp0RXko89Zj8b6Dma2EkJUWyoVJLY6dMMkI9S5Dl+d3HVTSUZM4L+tGDe5nk/O6V2kbJyO58C3b5hklLaoX+6T/Un3cMOmWSt0x/65qfPdjG5OBXxQ8NwdJRkARmMhp069gfEmXjchLEkSDVHppf30Ezm7dlfoOYzJqfSZvpW3rDbTIDC9xP6PoxKZ4aUGe5k/QR1voHeHFwBIT7iMq/t+C0ckhKiMVK3iWEHZY+rqOAYy89xRtLryF0yNDehN94vDbvrcOad2kvLHN3j2fx37Om1uWPb6Rf2a13MDINfw7xd4gx7lNhf1u54aeN7lPM9kPomt0Tw3zbKyThAVNdcsdQlRUSTREJVa/tmzJH51lwa96XQ0WbiDD87/RedGkSV2L49PY5x1PkZ3mdrVOd6ZFk37AHDlzAG8g6IsHJEQojJxbFLAbOvJZe73tfNibkwULnnp5dZ1yb0LHfZ3JCeuaFYmfUYChQkXTetcpO1cRPL6a58TOad2krzha9y6PYOtfwMM2WkYstMwFuSUqPv6Rf3atPLlUANPNMG1yF25hMLwYxQeOYhN4+a3ePXFuVmp+dhxFV3S30KFeXurX4qcRVb2GbPWKYQ5yRgNUWkZ8/O5NGQIhecv3PVzG5rVZ+HDrVlzMhSDsXgzvY+NFT95eWF/peSH1r1EZathV8ZvxMadQ2NtjWfNp8hIrrgV1oUQVYNLLTXf+r5OPgWl7ne1ceHHtHxqJ0aUW1eKXxda7W5P1C/vldjn0KQHnn3Hk7xhGvqMBHyfKEps4pe8TUHMyTLLXy9p3RTsAxtx35gX2Pfv25fu9EkyvpiAMT0V+0eG4TjihXLjLEtje/g//Uc4Fxy/7TrK4+jYkDatV6NWW5dfWIi7TBINUWnFffQR6UuXWez8Kh9v1o5syy8xncjMK/4Gbq1S8WNNP2rc44PEVR7WrD45jYL8HJw8fVBZD6UwX7qXCXGvcvK25sc6E0mk5KraAHYaLd8V2NE85li5daX5dqLz5RfI0lfce4q9jYYaHfwI05r/q9AA50wGZ47BSqn4z4maNf+POrXHl19QiLtMuk6JSilr2zaLJhkASkIiA6dv4U2nLdTxzSy2T6coPHEpll017UB97w4SV1J09G31MgBZyQk4OO2xcERCCEvROlnxR8jsMpMMK5UVXymeN5VkZPi0p+uVik0yPBxt8LzP/EmGjUrFq87/MDTjqbuSZABERc0lM/PEXTnX7YiPj2fMmDHUrl0bW1tbgoKC6N+/Px999BHW1tbs2VP8syMnJ4fatWszfnxR8tS1a1dUKhUqlQpbW1vq1avHpEmTMBhKTkZy/vx5nJyccHV1LbZ90aJFpjquf3z//fcArFq1il69euHl5YWzszMdOnTgzz//rJgX5B4iiYaodHQJicS9W7KZ3BKU/Hxazd/KhCtbaFev5LoR70Ym8L2vBrSaUo6+N1jHquna9n8AxJ07gndQyXVJhBDVm8ZazbFmawlXSnZZAlCh4iPbWnS+sLfcurK8W9M97iUydLeXZGQd2cDluc8Q9dUg4haNJb+UblRBHvYoHbw5Z62Qu2Y5yaMeIaF3e5JHPEzeX78XK1tweD/JIwaS2P9+MiZ/gKLTmfYZs7NIHjEQQ0IcAH42aibZfU+rjEm3FfvtUhQ9p06/gdFYenc1S4qMjKRVq1Zs27aNKVOmcOLECTZt2kS3bt1YunQpY8aMYdSoUeTkXOuO/Oabb2Jra8vnn39u2vbcc88RFxfH2bNneeWVV3j//ff56quvip1Lp9MxbNgw7r///lJjcXZ2Ji4urtjjySefBGDXrl306tWLP/74g3/++Ydu3brRv39/jh49WgGvyr1Duk6JSif6hRfI2bnL0mGwNC2NBWmpJOn1hNjY8Mb97Tkyqg+/n2zCf/9q6mZcJG3j91xIiCRPn0+gsy9Phg7guTaPmcrsunSI9zdPIzknjQfr3ccXvd/ERlPUJSuzIJt+Pz7P0senEeDsczcv0zw0Ko6yk3Pn96HWaPAJGUVaoouloxJC3A0qSG8fxjIWlllkvFNjnj6+sdyqcrxC6ZE4jvgCm9sKJef0LpLXf437Ay+hDWhEVthGso//hf+zs7Fy9gagfqAzUQ2dSFdD7toVZH83HefXPsCqfmP0Z06SOfUTXN6bhG3HLihGI0lDeuIw7ClsW3ck/aM3sH94KPYPDwUgc9qnaAKDcXj0f7R1NPBc3pvY6yNvK3ZzCK7xAiEhb1rs/KV56KGHOH78OGfPnsXBofgiuOnp6Wi1Wlq2bEn37t2ZOXMm27dv58EHH2Tv3r20bt0aKGrRCA0N5ZtvvjEd26tXL7Kzs9m3b59p21tvvUVsbCw9evRg3LhxpKenm/YtWrSoxLbyNG7cmKFDhzJhwoTbunYhLRqikslYv6FSJBnvx8XxSWICCTodQdbWBFpb8/LWXTw8ey3P1tuDvW3xucuPREZwMiOBHF0eHnautAxozJe7v+eXsHUA7Lx0kP/9+gaxmYm0rxHK0SunWHKs6K5ZZkE27ec8Sv8G3atmkgFgUGhp1x1XVz+MBgOZCauxtdeVf5wQosrTtEy4YZIxwrXZTSUZeZ5NeCBp7G0nGQCZh9bg2KwXTs0fxNozCPeez6Nx8iTr6B8AhNb14GyjoiQDIH/zBuz6DUbb7UGs/APRdu+NXZ+HyVm2CAAlIx0lPQ37gY9hVasOth27oI+6CEDhyTB0Z09h/8gTPOESzyvZT1g0yQCIjvmejIzKcwc+NTWVTZs2MXr06BJJBoCrqytarZbFixczf/581qxZw9NPP827775rSjLKYmdnh+661qVt27bx66+/MmvWLLPEbjQaycrKwt3d3Sz13ask0RCVhiE9nYTrmkktZWNmJqsyM7BRqVChIs1gYEd2Nl5WViw9c5bBM/7idfetBHgUNfPq0uNJ3TwXXWE+OhRylUJWnvyT+p61OHj5OEbFyItrJmBUjGjUGvZHh+Hr5ElEciQA/7f2Y3IL8xna7CELXvWdU7L1PNDgKTQaK3LSU9HabgMzT+UohKhcnBoXMtum7Pftfm5NeP3ohnLryXdvSJ+UV7mSb3vbsSgGHYXx57Gr1aLYdrtaLSi4coY2zX04WMuW/OuG1Sk6HSqb/yQ2trbozpxE0etQubqh9vCk4NA+lIJ8dMePYFW7LopOR9Y3k/B57X0+cN1J3/TRqBTLL56nKAZOnX4TgyHf0qEAReMlFEWhQYMGNyzXunVr3nnnHQYPHoyHhwfvv/9+mWWNRiObNm3izz//pEePHgCkpKQwatQoFi1ahLOzc5nHZmRk4OjoaHr4+vqWWXbq1Knk5OTw2GOPlVlGlE8SDVFpJEz5EkNKiqXD4JvkJAAecXZhVc2a9HV2RgfYqtSE5eWh5ObSYe4WJqZvJrROIpn7fwODHoeGXfAfNQP1ff9DQeF4/FnaB4WSnJNGdmEuDjZ2vN3leTRqNcfiztDQuw57o4+yK/IgI1sMooarv2Uv3AxUCQZ6t38JgISL4XgHyfzuQlRXLjU1zHB6H6WMGwqdXBvwcdjmcteOKHCrR9/014nMu7M1igy5maAYUdu7FduucXDD2pDFbl8rjKrik3fYtOlA3h9r0J07haIo6M6Gk79pLej1GDPSUalUuEyYQs7P35H81GCs6jbArs9AcpYuwLNNO151WMbcUe8yamQMa9Zk3FH85pKbe5ELF+/S+lPluNo7X6Uqf9KU999/H6PRyNtvv42VVcnxObNnz8bR0RGtVsuAAQMYPnw4H374IVA0fuOJJ56gc+fONzyHk5MTYWFhpsfevaWPGVq6dCkTJ05k+fLleHt7lxu7KJvMQykqhZz9+8lYtcrSYVCoKMT82xTb38WZOra2vOPtw9qMDOJ0heiUf6e5VRQCV+5hYocstsTsQ611wL3n81yeNRJDXgYoCgoKQ1r1JT09DYCX2g1jxYmNZBbk4OXgxqBGD9Bu9hAcbR0Y2qwvg35+mdS8DJ5q+QijWg221Etwxxxj7WnbfAAHj60j5sSf+DX0JjXOw9JhCSHMyNHLmh/9Pi5zrYymzrX5OvxvrI037kJZ6FqbgVlvciHXzmyxXf+d1sZKjU+wE+cvqSjZcQcc//ccxtQUUkePBEVB7eaO9sEB5C5bhEpdNMmHTdMWeMz5xXSMPiYK1db1zJznw9uvnOCRwS60aWPPc8/G0Kypltp1br9VxlxiYhbh5fUAbq43XhW9otWtWxeVSsXp06d5+OGHb1jW2rro87W0JAPgySef5L333sPW1hZ/f380mmuTsGzbto1169aZBocrioLRaMTKyor58+fz9NNPA6BWqwkJCblhHMuXL+eZZ57h119/pWfPnjd7qaIM0qIhLM5YUEDcv3clLC1Jf+1DUcW1T6vaNjbkKQr/vSlju+8Y+vQs/ANcsbEy4PPkF/iN/AZtSHsAHj/xG55+PrjZOfPriT9Jy8vAw96VMR1GMGnHXLIKs/mw+/8x6JeXiEqP5dEmffhm74+cTrz7ixSaU+38xgQFNEZRjKTFrEbrWGjpkIQQZmLraMWmenOJJ77U/TUdAph1/gT2hTde1FTnHMzgnHc4k21vlrg09s6gUmPIKbq542xnRY37/YnOSUXtVno/e5WtFpc3J+K9cS+eSzfguWwjGh8/VPYOqFxcS5RXKwqO09/ljef1WOtTOH++kM6dHXBz09CsmR3HjleOLktg5PTpNzEYci0ahbu7Ow8++CCzZs0qNqvUVbcyMNvFxYWQkBCCgoKKJRkA+/btK9ZS8fHHH5taLwYNGnTT51i6dCmjRo1iyZIl9O3b96aPE2WTRENYXPKs2eiioi0dBgAZ/87JrQKS9df629qp1SiAh6bkNLYGo5HgpExe89tBQLALNl41savdEoDDO5YxypDF3Gen4qJ1Qq1S0adeZ9oHNWfJsXV0qdWWmft+JleXj52VLTP2/UQjrxD2x4TdhautOEqhkU4+g7C3dyYvKxNr9V+oVDJeQ4iqTmOt5mTz3zlpLH2la2+tJ/OvXMYt58bdYPVOAQwteJcTWaW1M9welcYaG98Q8iLD8HXRYt/Rl5M2CoX/7Me6cfMbH2tljcbLB5VGQ/72P7Ftfz8qdfGvSO5Wah7Y+j5B9pF06miP0Vj0nnb1o0KvVzAazXY5dywvL5rz56dYOgxmz56NwWCgbdu2rFy5koiICE6fPs306dPp0KGDWc7RsGFDmjRpYnoEBASgVqtp0qQJbm5u5VdAUZIxYsQIpk6dSvv27YmPjyc+Pp6MjMrRJa6qkkRDWFT+2bOkLFhg6TBMnDVFTbbeGg17c6/dfck1GlGAULuSzfsalYqcwkI6z/qTiQVbaBycipKfBYBiMHIht5CPnAL4ccIv7H1xBZ/2Gs+o395Go9IwpfdbXEyLAeDnoVPRG/XEZMZhMJZchKiqUdJ09GlRNF4jKeocngGlfzERQlQRKkhrfYytxk2l7naydmRuag5+/76nlcXg6MeTug84kuFk9hCd2zxMzvG/uJyym/NXLpA16yuMCfHY9x8CQNZ308n4/NpAY31MFHmbN6C/HIXu9EnSP3kLfeQFHJ8dU6zeJvbwZvobbPrhL/5vtCcATk4aatSwZtWqDE6F53P0aB6NG1u+29T1Ll/5mdTU8tcuqUi1atXiyJEjdOvWjddee40mTZrQq1cvtm7dypw5cywa2/XmzZuHXq9n9OjR+Pn5mR5jx461dGhVmqyjISxGMRqJfHwY+ccrzxfQQkUh9NxZ6tvYcKGwkA99fQnV2jE48hKFwObadQiwtubrpEQS9Xom+/nT7cJ5MgwGpvr7U8vGlvw2jXj678OkX4nHqd2juHcdaap/Vi1f9i9fwKTtc5jS5y06Bbeiw9xHqekawCONH2DhkVVk5mexevhsWvg3stwLYUYpPkls2V+UTAY2Hk5yrAysE6IqsmqVxEybT0vdZ6uxZX6hEy2jj9ywDoODN08pE9mV6loBEUKTmm7sP72etOWLMKYmY1UzBKeXX8OmeSsAMr6YgCE+FvdpRatB66MukvHZu0XjLqyssAltjeNzY7GqUdNU58MuGTyS8X9M/uQSjZtoefjha2sEnTmTz5QvkkhLM/DIIy78b8TN3T2/m7TaANq1/QMrK0dLhyLuQZJoCItJXbyYhEmWn872v3pfvECMTkc/J2f25+aQYjBgBLo7ODIzMJCvkxL5MysLXysrfqwRzMT4OFZkZOCiVpNvNIJKRYGiYO/kgO/oJRiUa92tDDnpxM0ZRW23QLY9s5j0/CyaftuXwY0fZH90GFeyEmjp35i1/6s8d3numBpOWh8g/MwObOzscfIZSU6G+QZ+CiEqnlMjHV+5vFHqDFMalYavNYF0j9h9wzqMdp48p57I1pSKWZegVSMvDgRao7+JGY5uhq1Kxf85H6JleuX7nLpVNYKeoW7ddy0dhrgHSdcpYRG62FiSvvnW0mGUaqynFypgd042aQYDrhoNtioV7/gULaa3MzubXKORH2sEA/Csuwc2KhUalQoDReM51MDnvv6MC96Nq8O1gdApf85EsdbS4aWp4GSNq9aJIBdftl/cT0ZBFtYaKyb2GFMyqKrMCE01nfDwCKIwLxdFtxG1VSXqyCyEuCGXYA0zXcqexnaCtk75SYbWjdFWEyosyWjT0o+/g2zMlmT426iZpJ1fLZIMgJjLi8nJOW/pMMQ9SBINYRHxH32MMdeys2GUpY+zM+95+2CvVqNSqfC1suK7wCAC/p16r7FWS+3rFngKtLHhu8AgfKysUKlUOKjVvO/jQy+1hp4zNzFRvYW6/pkAqKxscLnvSXYW2vKsko3e245ZAybibueCldqKVzqMqDZdpq6n5OrpWWcEVlY2pF6JxN37H0uHJIS4CY6e1iz2/4w8Sp9NaYxzEx45teWGdSi2Loy3/ZCNSZ5mj89KraJlxwB2e5nv60w7RwMfG8bim1v6WJSqSFF0nDv3saXDEPcg6Tol7rrMP/7gyquvWTqMuyqzZxtmtOrEnrOBxbY7W6lZ4u+Da/SNp4GsLvICCli35xsAgpoNJSkmwLIBCSHKZOtgxfbm33FcCSt1/xNuzXjnyPob1qHYOPKm/cf8Gl/2Csy3y85GQ80OfhzVmu9rzP9cYnkwY3ylWOW7IjRtMhtv7wctHYa4h0iiIe4qQ3Y2F3r3wZCcbOlQ7jqlUV0WDerIr6eaoCjXmvdVwHe1/Ghw6d5INmLcLrD3yG9Y22pxCRxFdqp55tAXQpiPxkrFubZ/slnZUOr+3m6N+eLon6iVsrtBKtYOvO/4Eb/E+Zs9PndHGxzbeXPOyjxfYRw1Ksbb/0WDzLlmqa+y0moDaN/uLzSaO1uFXYibJV2nxF2VMm/ePZlkAKhORfDMvN8ZU+tvnLTX7pYpwLOX4thQwxaszNO/uDILyq5LreBQdAX56LJ/R2Nd9afyFaK6yWh9sswko51rPSYd23rjJMPKjk+cJ1RIkhHkYQ8dzJdk1NGqmWT1ZbVPMgDy868QFTXP0mGIe4i0aIi7pvDyFS4+9BBK4T2+SrS1NSeHd2Wm9X1EJRZfrGqQjwuvZVtBjq6Mg6sHlYs168/PIjs7jYAG7UhJ6GTpkIQQ/7JumcIM29L78zdyqsmCs0dxKMgq83hFY8sU94nMiQk2e2z1A52JauhEupluk/Zwymd49lhsjPfODTC12pb27f7Czi6w/MJC3CFp0RB3TdLXX0uSAaDT0WThZibFbKJNnfhiu1YnZPCyJheDZ/Vu1lYydPRu9iKoVFw5cwDvoChLhySEAJwaGphpU3qSUcPej9kXT5eTZNjwjccHFZJkhNb14Gwj8yQZGuAll7M8nfnkPZVkABiNBUSc/8zSYYh7hCQa4q7IO3aMzD/+sHQYlYr7xn18uGU9Dzc6U2z78cw8HslIITvQoYwjqwdNnJEHOzwHwJXT63DxLPvLixCi4jnX0DDb9X2UUnpwetq6Mzc+Ho/spDKPV9RWzPF8j2+ja5s9tjbNfDhYy5Z8M/QudbdS84nDr9yXfu+uK5GU9BepqX9bOgxxD5BEQ9wVCZO/sHQIlZLm2BleXLCOl+ruR2tzbdxGis5A38txXKxVvVdydY13o3njXhh0OnJT12KjrZ4zvQhR2Tl4WPNLwOfkUHLacUdrB+ZkFBKUUnbLo6LS8IP3e0yJqmvWuFQqaN3Wn91+VhjNsEZGU3uYpHqX4OxlZoiuaos4PwnlBuNshDAHSTREhcvctIm8o0ctHUalpcQnMnDGej5w3Ymf67UPeQMw4lIsW4PtQFNNB4kr0FBpg493HbJSEnFw3FXmomBCiIphY69ha/3vucKVkvvUNkwvdKRB3Kkyj1dUan7yfYdPI+ubNy4rNU3vC2SPm3ne/wa5pPNG7v9wKjxtlvqquuzsM8TF/WbpMEQ1J4mGqFBKYSGJU7+2dBiVnpKfT8v5G5mc+hfNgov3F/4wKoE53iqws7JQdBVLyTPQLXgYNtZ2xEWE4RN0wdIhCXHPUFupONviT8KUIyX3qdRM1vjTJvJQmccrqFjh9yYTLpl3oVFnOytq3O/PQfs7v/Fgq1LxuvNBhqQ/g0apnAvFWsqFi9MwGOQ1ERVHEg1RoVJ//gVdTIylw6gyvNft4dO/19OnQfEv27/EpfOabT5Gd1sLRVaxlGQdD7V7GYDL4Rtw80m3bEBC3COyWp/iT+Pvpe57z74evc7tuuHxawNf462Lzcwak6+LFvuOvpy0ufMkI8BGzSTtPFpkSPfd0hQWJhIVNd/SYYhqTKa3FRVGn5bGhQd7Y8zMtHQoVY4q0J/fnujCjxdaoDNcux/gZ2vFYg9P7GKr5x2oWI9odh9eir2rO9YOT1CQa2PpkISotmxapjHddmKp+150acrosNLX0bjqj8DxvHy+jVljqu3jSFIzF5LMcBu0g6OBZ/Jex04ffeeVVWNqtR0dOmxBa2v+1duFkBYNUWGSZ82WJOM2KZdjGTJzLe/67MLDqcC0Pa5AT7+4BC5X00Hi/hnBhNRqQ256Kra220DGawhRIZwbGJhp+1Gp+x51Kz/J2BI0xuxJRpOarkQ1N0+SMcLlCqOzn5Ak4yYYjXlcvCBdnEXFkERDVIiCS5dIW77c0mFUaUpuLu3nbOCL3L9oEJBh2l6gKDx+KZY9teyq31+wXqG10wM4OXuRePEU3kEyaFMIc3MOsmK2+wcYS0nke7o15v2jG294/K6gl3g2ooNZY2rV0Iuwevbk3OG4b0eNiglOf/Jg+iuoFJnF7mbFxa8mKyvc0mGIaqi6fU0RlUTiV1NBV71Xt74rFIWAlTuZfOR3etQrfmfu7UsJLPC1AluNhYKrGEqWnt6Nn0OlUhNz4i88/FIsHZIQ1YaDuzVLA78gW8kpsa+1S12+OLYN9Q2mPD0Q9BwjIu43a0xtWvrxdw0b9Hc4fW2IVsVkqy+onyljDm6dkYiISZYOQlRDkmgIs8s5eJDsrVstHUa1YrsvjDd+XcOI+sfQqK/dhVwQm8o7DjoU1+o1SFwdb6B3hxdRFCMp0auxu677mBDi9tjYadjWcCExlOxOVN8pmOlnDmNjKPtv7UjQKIZGdDNbPFZqFS07BrDb686/ivRyzuP9gmdxyz9ghsjuTWnp+0lO2WHpMEQ1I4mGMLvEr6ZaOoRqSbkUzROz1/BmwB6c7ApN23enZvNkQToFfvYWjM78nOOdaNnsIfKzM9HwFyq1LCwlxO1Sa1REtNjCUWPJqWoD7H2YExmBU35GKUcWORn0JI9EPGC2eOxsNNS9P4C9TndWjwZ42eUMozKGY21MNUts97LISzMtHYKoZiTREGaVvXsP+cePWzqMakvJyqLzrN/5gq3U9sk2bY/O09E/IZGEmtVokLgC9QpDCfCvT3J0BJ5+8nslxO3Kbn2WjcqaEtvdbV2Zl5CCV2Z8mceeDRpKv4i+ZovF3dEGr/v9OKq9s8kePKzVfOqwjE7p75kpMpGReZTUtH2WDkNUI5JoCLNKnjvX0iFUf0YjtZZuZcqZ9XSsE2fanGs0Mjgyln9q2UM1WUhcKTBwv9+j2GodiDm5Dc+ABEuHJESVYxuazmL1rBLb7a3smZ1pJDj5YpnHng8aTO/zA8wWS5CHPar23pyzurMko7k9TOIdamT/aqbIxFWRkSV/V4S4XZJoCLPJOXCQvH/+sXQY9wyHnYf54PfVDKt/iuvHUI69FM/SABuwqR5/3kqqjr6tihbzS7ywGnuXfAtHJETV4VzfyAztxBLbrdXWfKN3pnHsyTKPjQocwAPnB6Eo5rlzUT/QmaSWblzR3FmS8YhLKq/lDsex8IxZ4hLFpaXtIyPjqKXDENVE9fgmIiqF5DlzLB3CvefcRUbOX8Wrwfuwt702leOsy8l85GwE5+qx4J11rJpu7UZSmJeLUrgBtZWM1xCiPM6BVsz2+ACjqvgXexUqPrMOosOlg2UeeznwIXpeHIpRMc/XhNC6Hpxt5ET6HVSnVat4w3k/g9OfQ6PkmSUuUbrIyNmWDkFUE5JoCLPIPXKU3P37LR3GPUlJS6fnzLVMst1GgMe1KSs3J2cyUp+FzsfOgtGZj3eKHw3qdiItNgp3r8OWDkeISs3e3ZplNaaQrWSX2PeWY0P6nNlR5rHxAb3ocfEJdEbztGS0aebDwVq25N9BdQE2aibZziE040uzxCRuLDllG1lZpywdhqgGJNEQZpE8V1ozLEqvp/7iv/g6aiOtayaaNl/ILaB/chIpwdVgkLhBIVTbFXc3fy6f2oNX4BVLRyREpWSt1bCr4Y9EK1El9j3r2pQnT2wq89gk/+50ixxBgfHOvx6oVNC6rT+7/aww3sEaGR0d9XxsGINP7uY7jslclixJo2ePi8yelXzDcoWFCgt+SOWJYdH06X2R/w2PZuPGTNP+fw7nMnJEDAMGXOKLyYnodNdan7KzjYwcEUNCgmUWHoyMks91ceck0RB3LO/ESXJ27bZ0GAJw3rKfjzevYVC986Zt2QYjA6NiOVnLocoPEley9fSq/xQajRVxZ9fg5F5y0TEh7mVqtYqLLbdz2FhyPYlH3Joy9uiGMo9N8etM16inyTPc+SKgNlZqmt4XyB6323/TUQEjXWJ4KesJtPrLdxyTuZw5k88fG7KoXbv8rqmffJLAkaN5vPa6JwsXBfHe+97UqFF0nNGo8PnnifTv78T06QGcPVvAHxuuJSHff5dC//5O+PhYVdi13Ehi4iZyci5Y5Nyi+pBEQ9wxmWmqclGFn+PFBb8yptYhbK4by/DipThWB9mCdRX/s0/Q06f9y+gLCyjM/h0ra4OlIxKi0shrE8EGZWWJ7V3dGjLhaNktGem+Heka8yw5hjt/f3C2s6LG/f4ctL/9Qd9OGjUTnDbyQPo41FSev/G8PCOfT0pi/KueODrd+LU6eDCX48fymTTJl1at7PH1taZBAy2NG2sByMgwkp5uZMBAZ2rWtKFDB3uionQAnDyZz7lzBQx6xKXCr6lsRmnVEHesin/jEJaWf/Ys2du2WToM8R/G5BT6zlrNZ8478L5ulqap0Ul84WYER2sLRnfnHGLtaBv6MBkJsbi4y5zvQgBoQ7NYqJ5RYnsLlxC+PLELjVL6F/ZMn7Z0u/IiWfo7v3Pu66LFvqMvJ21uP8moa6fic6tJ1Mv8/o7jMbfp3ybTrr0drVqVv0Dqvn251Ktvy/Ll6Qx9LIqRI2KYNzeFgoKiG0Curmo8PDQcPpxHQYGREyfyqV3bBp1O4dtvkhk7zguNxrLN0AkJv5OXF2PRGETVJomGuCPJc+eCcmdTFYqKoRQW0mTBH0xL2kiToDTT9t8TM3meHPReWgtGd+dq5zWkRmATrpw5iHdQpKXDEcKinOspzNBOKLE9xDGIGWePoNWVPktTtncresSNJk1350lGbR9Hctp6EHkHa2Q84JzLe/nP4JZfcgVzS9u+LZuI8wU8+6z7TZWPi9Nx8kQ+kZd0fPSxDy+P9mDXrhymTy8a16FSqXj/Ax9+/jmdZ56+TEiILb37OLFsaTotWmqxtVUx9pUrjBoZw5o1Za/aXpEURU9U9HyLnFtUD5JoiNtWcPEiWX/+ZekwRDnc//ibybtX81DdSNO2U9n5PJyWQkaQg+UCu0NKoZGO3g9jb+/MlVPrcPHKsnRIQliEc4AVcz0+wKAqPu2zn50Xc6Mv4ZKXXupxuZ7N6ZkwhqTCO2/hbFLTlajmLiTd5rcKKxWMdjnNyIz/YW1MK/+AuywxUc+sWSm88443Nje5RpFiLBoQ/8673jRooKVdO3tefMmDv/7MNrVqNG2qZfbsAH7+pQavjPUkPk7Hli1ZPPWUO5M/T6RvP2emfePPzz+lcfFCQUVeYpni4n6joKByLZYaHx/PmDFjqF27Nra2tgQFBdG/f38++ugjrP+fvfsOj6Jc+zj+ne0tm94TQkJvIfTeBRTEAlZQxN5RxAMW7K9i12PDhgVFT7ErHrBiw05TQErokN57tsz7RzAYQwlhk8nu3p/r4lJ2n5n5LYTdufdpRiPffvttg/YVFRWkpaUxZ84cAEaPHo2iKCiKgtlspnPnztx33314PHW9fps3b2bMmDHExsZisVhIS0tjwYIFuFyuRlmqqqoIDw8nIiKCqqrGBf1fr/Xnr3POOacF/lTaJik0RLPlP/sseGU/A3+gX7uR2a//lys6rMWgq/u2sdjt4eQ9WWxJ9eNio8jFpD5X4nG7qSh4D5NFm9VZhNCKLczAf9o9TCkNC+0wUyjP5pcQW7L/kMdVRfZkQv71ZNcc/147/bpFs7azjYpmjvKJMur4P9ubDC1ecNxZWsrWLTUUF3u48op9TBi/nQnjt7N+XTXvvlvKhPHb8Xga9+JEROiJitLjcBy81WrXzoiqQl5e42Fsqqry6GP5XH5FJF4vbNtWy8iRdsLD9aSnW1m3XpvNSr3eWnbtbjvD2Hbu3Em/fv344osvePDBB/ntt99Yvnw5Y8aM4c033+Taa69l1qxZVFQcXCxk3rx5mM1mFi5cWP/YpZdeSlZWFps3b2b27NksWLCAhx9+GACj0cjMmTP55JNP2Lx5M48//jgvvPACd9xxR6M8b7/9Nj179qR79+688847h8z857X+/PXcc8/5+E+l7dJmKQPh92p376Z02cdaxxDHQM3K4bSn3yJ5ZgkPVQyhuMKECly0I4tbU2I4aW8NHOLDsq3T74cTBl/MZz8sJq7T19RUj0Hx9+W1hGgCo0XPt91fZ6e6o8HjVr2Fpyt0pOVuO+RxNRFdOKnwBvZWm487w4C+8XwT3fzvLDPsKlfW3oSjfPNxZ2lJffpaeeHFpAaPPfRQHu2SjZx9Ttgh51L06Gnh668rqKryYrXW/Rnt3etCp4Po6MYre/3v4zKcTh1Dh9opK6srRNwHvjtxu1VNv9fbt+9N2qdcicnUtGFjLemqq65CURR++ukn7PaDX5T16NGDiy66CIvFwvLly5k/fz5PPfUUX375JS+88AKrVq3CYjk4ZNhmsxEXFwfANddcw/vvv897773H/PnzSUtLIy0trb5tSkoKK1eu5JtvGq+wuXjxYs477zxUVWXx4sXMmDGjUZu/XivYSI+GaJaCFxeDp+2sBCKaRq2upu/zH/J4+Sd0iju4jOK9u3L5Z5QCNv/87iEyL4oeXUeTvXUtscmHvrkSIpDodAo7+n7FT2rDxRAMioFH1EjS96475HG1YR05uWQeO6uOb46WQafQd2jicRUZZ4QWckPF+Thq23aRAWCz6UhNNTX4ZbEoOJ11jwO8+GIh999/cB+jceMcOJ06Hnowj107a1m/vornnytk4okhmM0N/9yKijwsXVrMNVdHARASoqddOyPvvFPCxg3VrFlTRY8ex18YNpfXW8WePS9pdv0/FRYWsnz5cq6++uoGRcafwsLCsFgsLFmyhOeff5733nuPiy66iFtuuYX+/fsf8dxWq/WQQ6MAtm3bxvLlyxk1alSDxzMzM/n+++8566yzOOuss1i1ahXbt29vdPzSpUuJioqiR48e3HjjjZSVBc9QXyk0xDHzlJRQ8uGHWscQxyH2/ZU88ut7jO1wcNO7/+YUM9tYhTfSDyeJe6GXfhhRUe3Yu+FjwmOLtU4kRIuqHpDJR+p/GzymoHC3uT0jMg+9EpsrNI1Ty29ia4X1uK5tNenpNCKJVSHNO96iU5jvXMXpxZeiVw89Sd0fFRa4yc09OHzTatXxwIPxlJd7ueqqfSy8L5chQ2xcc01ko2OfeTqfM88KJSr64Jc98+ZHs/LLcm69NZuzzgqja1dt35v37nsdt1vbG+Rt27ahqipdu3Y9Yrv+/ftz8803M23aNCIjI1mw4PDD8rxeL8uXL2fFihWMGzeuwXNDhw7FYrHQqVMnRowYwd13393g+ZdeeomTTjqpfo7GiSeeyEsvNSzIZsyYwZtvvsnKlSu57bbbePvtt5k6deoxvnL/paiqLBkkjk3By6+Q+8ADWscQPqAkJ/Lvs8fzSmYPVLWu6z/WZGBJdDT2fX64GV60gbdXP4TJbscYch41Ff69jK8Qh2LtXc5jtlsbPT43pDuz1h96rwy3sx1TqxewvtRxXNeOcJhwDIphSzNXlkoy65irPEVM1efHlUNoIy3tBlLbX63Z9X/88UcGDx7Mu+++y2mnnXbEti6XC5PJxH//+1/OOOOMBs+NHj2aVatWYTKZqK2tBeD888/nySefxGY7uHTxnj17KCsrY926dfzjH/9g9uzZzJs3DwCPx0NKSgr//Oc/mTZtGgBvvfUWc+bMYefOnej1h9748tdff6V///78+uuv9O3bt7l/FH5DejTEMVFVlaI339Q6hvARdc8+zl70X26P+54QS12XcU6tm5P3Z7PTHyeJ57mZNOhqKkuKMBs/B+R7FBFYnJ3gCWvjZWwvCOt1+CIjJJFzam497iIjKdKGMrj5RcYwh5u7XNdIkeHH9ux5BY9Hm0npAJ06dUJRFDZt2nTUtkZj3RdNBsOhhwTPmDGDtWvXkpmZSVVVFYsXL25QZAAkJyfTvXt3zj33XO6//37uvPPO+pWpVqxYwb59+zj77LMxGAwYDAbOOecc9u7dyyefHH5Fzr59+2I0Gtm6dWtTX7Zfk0JDHJOKb77BtXu31jGED6kVFQx+9l0edX9GSlQ5AC5V5bwdWXzZ3goabxh1rKz7TQztdya5OzYSk7xR6zhC+ExIgoHnom7HozScHzclvBdz1xx6cQ6PPY7z3Qv4paSZ45wO6JLopKBfOPv0x15kKMCs0N1cUTYdi2ffUduLtsvlKiQn5yPNrh8REcHEiRN5+umnG6wq9afi4uImnys0NJSOHTuSnJx82N6Hv1JVFZfLxZ8DgRYvXsw555zD2rVrG/yaMWMGixcvPux5NmzYgMvlIj4+vslZ/Zl/zvwUmila+obWEURLUFWS3vqcx4cV8VD/E1i1KwaA23bmcF58OFcUAdX+M/k/uawjqSkZ7PjtExK7x1KwP0rrSEIcF2uogbfbP0qJ2nDjtuFhXbl77QqUQ/TeeW3RXMRtfF8UelzXzugUydpUC9XN+M7Bqdcx1/YhHYu1n0gsfGPfvqUkJJxx9IYt5JlnnmHo0KEMHDiQu+++m/T0dNxuN59++imLFi1qUm/H0SxduhSj0UivXr0wm838+uuv3HzzzfW9F3l5eXz44Yd88MEH9OzZs8GxF1xwAZMnTyYvL4/S0lKWLl3KpEmTiIqKYuPGjcydO5c+ffowbNiw487pD6RHQzRZ7Z49lB9iaTcROCzfreb2D//NuR231D/2elYRN1prUcO1W/HkmLm8DAo/GYc9jIJd72IN0WajKyF8wWDWsarnm2xXMxs8nu5M45EN32HwNt4/xmuN5HLdHXxVEH5c1x6QHstPqeZmFRldrAoL9f9Hx1IpMgJJadl6Skt/0+z6qamprF69mjFjxjB37lx69uzJ+PHj+fzzz1m0aJFPrmEwGHjggQcYOHAg6enp3HnnnVx99dW8+GLdfiJLlizBbrc3mjwOMGbMGEJCQnjttdcwmUx8/vnnTJw4kS5dujB79mwmTJjAZ5991qRelEAgk8FFk+U8+BCFL8kHRjBQnE5WzjyRx3P7UV1b92aYaDHySkQk1v2VGqdrOm+8jv9+fz9RSR2oqDwZ1SvfrQj/ouhg/+DveV/9V4PHU+2JLNm+mbDKwkbHeC3hXGO8i4/zmt+TpyjQb0AC34Y3b+jkic4Kzim9FuPfemBEYEiIP4tu3RYevaEIevKpK5rEW11Nydtvax1DtBK1tJRRi97mUf0XxIXVLT+5r9rFyVk57G9/fBNKW5Muy8vEIZeRv2cbUfHrtY4jxDFz9d/VqMiIsUTx3L69hywyVLOTG8x3HFeRYTLo6DU8qVlFhkGBa0N/5/ySmVJkBLDsnA9xuUqP3lAEPSk0RJOULluGp0Q+NIKKx0Pqmyt4aueH9E0qAKBGVTlr535+SLX5zbtHWHYYGT0msOf3L4hOzNE6jhBNZu1dyYv6Rxs85jSF8FxhBfFFexq1V00ObrLdyXs5Mc2+ptNqoN2IBH6yHftgh2ijjnutrzO4+I5mX1/4B6+3iuzsd7SOIfyAn9wqCK3JJPDgZV/5E/d++m+mpR3c7fTGHdm8Gm8Asx+MMVWhq7c/cbEdyMl8F3tY4GwQJgKXs4PCk9bbGjxm0Zt5qtJEx5zGO2mrRjt3OO7k31lxzb5mbKgF+9B4fjcde5HRx65yrzqfpIp3m3194V/27pOl7sXRSaEhjqpyzRqqN8oyoUFt83YuffkNbmi3BqPeC8AL+wpZ4HChhpo0Dnd0arWHUe3OBbeKt+ZjdAav1pGEOKyQOAMvxN6BWzk4yVuv6HmIWPrsWdOovWqwcm/o7SzZn9Dsa6bFOqgcGMmOZvzbOCs0nzkV52F3bTl6YxEwKiu3UVT0g9YxRBsnhYY4qqI35FsLAWpRMeOf+TcP278m8sAqTisLypnpKqU2znaUo9uAfBeTBl1N0f5dRET/onUaIQ7JGmrg3bTHKVKLGjx+h6UDo7d926i9qjfzYPhtvLg3udnX7NE+jF29Q8k7xjsCi07hJue3nFp8OXpVu03chHb27pPRDuLIpNAQR+QuKKBs+aF3mxVByO2m85KPeDprGT3iiwHYUVnLybm55KW0/Uni5n16Rg6Yzt6N3xKTtFfrOEI0YDDp+KHnf9imNtwxeLazB6dv/KxRe1Vn5InI21i0p32zr9mvWzTrO9upOMZ53+3MOu43PUmvkseafW3hv0ymBGz6c6heNwZPea3WcUQbJsvbiiPKf/ZZ8h7/p9YxRBvk7dGFZyacxLIdSfWPPZkaT5+dFRxi77C2Q6+wWv2SHXvWEp48i7JCu9aJhEBRIGvIj7ynNvyGeEZYOjetabwTs6oz8Gz0bTywq1OzrzmgbxzfRB/7PKsRIS5mVczB4slq9rWF/1EUIxbzQHSFQ7BldiEiywhA6OQ0QkYkapxOtFVSaIgj2jZ+Aq49jVc3EQJAiY7iw+kns2h3d7xqXQfptUlRnJ3rhtq2Ow9CCTHwv50vgMWARzkTd60fTGoXAc0zYA8vGB5u8NhJ4T15YPX/Gu36rSp6Xo67hbt3dGvWtfQ6hd6DE1gVcmzHKcCFobsYU/wPdHiadW3hf8ymZPTuEeh39yM6MwJjTcPuL0Osjbg5/TRKJ9o6KTTEYVWuWcOuc6drHUO0cYrJxPrzJnFP1UDKquomhk+IcnJ7jRHKXBqnOzw1Ts9bPz5AXOe+FOaM0DqOCGK2XlU86ripwWNDwrrw9PqvMHoaDktRFR1vxN/Erdt7NutaVpOe9kMSWGM5ti8CQvU6brB+QMeyl5t1XeFfFMWExTwEXcEQHFs7EJZrPGL7mGsyMCUdY+UqgoIUGuKwsu/5P4qWLtU6hvAThZNHcUfCKLblOQHoYjfznD0MQ27bXU62LL6cj1c9Tbv008ndk6p1HBGEnB0UHou+AddfVpjq4UzlpT9+xVZT3qCtisJ/E+cxL7N3s64V4TDhGBTDFsOxfex3sSpc57mH0JrGK16JwGI2t8dQOxLdrj7EZoahdzVt8o59cDzhp3Vs4XTCH0mhIQ5JdbvZOnIUnsLGO88KcTiejB48MnISX+6OBcBp0LE0IY7w3eVHOVIjCmxz/M7ajSuISptFyYEiSYjWEBJnZHHaAgrVg++zKfYEluzYRkRFfqP2HyTNZfa25g1RSYq0UdUngn36Y/vIP8lZwdml18ou3wFMp7NgNg5FnzeEkK1pOAuOfSipYjGQcOsgFKOsMSQakkJDHFL511+z57LLtY4h/JAuIY7/njWFxbu6oKp147qfS42n+44KraMdkmLW8U3pO5RUF6JYzsFVdeQhAkL4gsVp4KNeT7DNe3DzvWhLBEtyCkkq3N2o/fKk67hi26BmXatLopPdPUIoOoaVpYyKwhXO9QwuvrNZ1xRtn9ncEUPNCPQ7MojZ4UTvPsalx/4m4pwu2DKavyu9CExSaIhD2vePeZR++KHWMYSfUqwWfjnvZO4r7U9ljQGAee2iOWW/C9xtb5K4EmHkvQ2PE5bUgaKCsSgc3weuEEdiMOlYPfBtVnlW1j8WYnTwcil0yW68OeoXyVdz0dZhzbpWRqdI1qZaqD6GH+kYo44bja+QWPF+s64p2i69zobZOBxd7mDCtqRgL/LdQhjmjmFEX9LLZ+cTgUEKDdGIt6qKLcOGo1ZWah1F+Lmc08axIGIkew8sIXtqTCj/qDBARdubJO5KUHnnuwdpl34yuXs6ax1HBChFgZwhP/OO+nr9YyadiWfdYQzY1Xgjye+SL2fG1lHNutaA9Fi+i9PjVZpeZfSzq1xeMx+7K7NZ1xRtk8XSFUPVcIyZGUTttKPztsCXKQrEzRuAIdzi+3MLvyWFhmik5KNl7L/xRq1jiADhGpjBwoEn8v3eKAB6hVh4yuxEn9/2dhLOjclm5c+vEdf5AopywrWOIwKQd8B+njc8UP97naLjEUMyJ2z5plHbn5Iv5qyt4475GooC/QYk8G34sd1Mnh2ax+SS62WX7wCh1zsw60egzxlE2JZ22Epafv5E6EmphIxKOnpDETRk1o5oRIZMCV8y/rSWOz54nRkpdd+Q/lZWzdSSAsqS2t5GeTEF8XTtOJSS/e9itre9Xhfh32w9qxsUGQC3WjsdsshYmzyzWUWGyaCj1/CkYyoyrDqFm53fcErxFVJktKD33y/hvBm7OenEHVx5xV5+W3/kFfnWraviyiv2ctKJOzjvvN18+GFpg+d//aWSC2bu4ZRTdvDA/bm4XHXfG1ssPVDKLuCCswpQ3hhFws/tW6XIAKja0HgRAxHcpNAQDbiLiij/7jutY4gAo+7ey/mLl3Bb3C+YDF4KXB5O3ptFZmobKzY8Kr3No7DobZiNn4EiHb7CN5xpOp4MWdDgsatCe3HWhk8btd2QPJ3Ttp54zNcIsRhoNyKBn2xN/7lNMeu43/xPepY8fszXE0335ZflLHqmgOnTw3j2uUR69bJw883Z5OS4D9k+K8vFrbdk06uXhWefS2T6uWE8/VQ+X39dt4Kf16uycGEuU6aE8MQTiWze7OKzFe1w5v8fsV/M5eU7f+WCHqeR5IxrzZdJ7Z4yPKU1rXpN0bZJoSEaKP3f/8B96Dc+IY6HWl7B0Of+zVN8QYyzGg9wwY4sPk2xgL7tTL5Wy92M73whBbu3EpO4Qes4IgA4Yo0sjrsLFwd7yc4O78WVa5c1arsl+Uwmbz35mK8RG2rBMSye301NLzJGhtRyh+sKoiq/OubriWPz9lslnHhSCJMmO0lJMXHV1VHExBga9VL86aMPS4mJMXDV1VGkpJiYNNnJiSeG8N//1C0zXFLipbjYy1lnDaNX0hzGdjmFvK/CiV+dxG9/bGBd9mYu7n9Ga77EOipUbSho/euKNksKDdFA6YcfaR1BBDJVJfmt5TyX+R594ooAuGtXLk9G68Bm0DjcX+S6OWnwVez+/VMiE2QogGg+S4iBDzo8SYF68OdofHgPblnzv0ZttyedzsRtpx3zNVJjHVQNjGKHoWkruumAS0J3cFnpdMyenGO+njg2LpfKli019O9va/B4v35WNm449FC1jRtr6NfP2uCx/gNsbNlSA4SRGD2VqLBwtr4wmPAV6fy2YSPdojtQ63FxyyePsHDCXPQ6360odSyqfpf3THGQFBqiXu3efVStXat1DBEELN/9wn2fvM605F0A/Du7iOuN1Xgj2s5qJfb9Vgb1PpX8ne9gDZFx6+LY6Y06fk1/ly3qpvrHBoR24v51X6BTGxYFu5JOYXzmNFT12Hr3erQPY3fvUHJ1TevJCDXouMvxPmOKb0RBhga2hpISD14vhIc3vPEPD9dTWOg55DGFhZ6/tVeIi+2GxwOs+gfJH5/Oc5Pv4ekvX2Pc4vPpEduJs3tN5pkfljIspR8Wg5nTX7+KUS/M4JVf3265F3cINTtK8bTBlQWFNtrQV4hCa6UffQSyCJloLdt3cemrr9Jhxqk8mpPOLyWVnGmq5bXEaGz72sbmfqmV3cgK30aFugJFNwXVK9/NiCZSILv/L3zj/az+oa4hKTzxx8+YPA3HsO9PPJETtp+N5xiLjH7dovkp2YSriYd1tSrM9txNaNmaY7qO8I2//zWp1K0QdqQDjMYoTOooDHsHEv5jDrCS8GwLOGBgUjrLLni+vvn2wj28vWEFy2ct5ow3ruXi/mcyOnUQJ7x0AYOSM+gW06EFXtUheFWqNxVg79+680NE2ySfmqJeyUey2pRoXWppKWOef4N/mr8izF5LTq2byfuz2Z3q0DoaAKrLy9Do06gqyCcqfp3WcYQ/6ZfF27xa/9skWxyLdmzBUd1wTH52wnjG7DgP1zHuazCgbxzftWt6kTHZWcbNVRcQWiNFRmsLDdWj00FhUcPei+Kiv/da/ElHVFQYFXnpRPx6H0kfTyFufSwlxcUYdHrCraGNjlBVlfnLH+K2MVfjVVV+z9nK5C6jibKHMzg5gx/2rG2ZF3cYVb/LPA1RRwoNAUDNjh3UbpMNmoQGPB7S3vyIF/d/SNfoUlyqyvQd+/m6vRV02k8SV4tdTOpzJXs2rCQqMVvrOMIP2HvW8qzx/vrfR5jDeS4nj6iyhvMh8hNGM27XBdQcQ0+ZXqfQd2gi30Q3bfy9UVG4PnQd00tmYVDLmnwd4TtGo0LnzmZ+/bXhcra//lpF9x6Wv7SLxWE4i7B9DzLYdgKrv9tJWLap/vmvd/xMelxXjPrGg1H+tX4Z4VYnEzoNx3NgWJ7L667/r8d76CFaLaV6WxHemta9pmibpNAQAJR/JauOCG3Zv/yex75+g8nJ+wG4ZWcOL8bpwaLNhMa/0u9XOWHwxeRmvoc97Mhr34vg5myv48mQW+t/bzfYWFTqpl3+jgbtCuOGM2r3xVR4mv4xbDXp6TIikVUhTWsfa9Jxn/VlBhTf3eRriJYx7YxQ/vdxKf/7Xym7dtXyzDP55Oa6OWVKODbLEF5b1IlH5jhI/HgSsRuimJl+KntLc7jr86fYmr+Tf61fxr/XL+PygWc3Ond+RRFPrFrCXSdcB0CYJYROkSks/vm//Lrvd77b9Sv9Enu27gt2q1T/Udi61xRtkuwMLgDYdeGFVH7/g9YxhEAXEc5H55zOU1ndUFWFoeF2HlBtKMUar82ugw2mn9hXnEmt+3Q8bu0LING2OKKNvNrpTvLUXACMOiNPeyMZsuOnBu2K44Ywat+VlLiaPk0ywm4iZFAMm41N+8juZ/dyec1N2F3SU91WvP9+Cf/5dwmFhW5SU+3ccPVUBoROJXJbBDe+t5C9Jdn8d/oT9e2/372Wu794ki35O4l1RHLloBmc3+fURue9+oO7GJDYk1n9ptU/tmb/Rm5Ydh/5lcVc3P8Mrh82qzVeYgPW9Cgip3dr9euKtkUKDYG3ooItg4egumSVCNFGGAz8ce4UbncNoqzaQJLVyCthkViyKjWNpdj0fJ7zBuaYRPKzBmmaRbQtFoeB5b0X8Yf3dwB0io4HjO05cfPKBu3KYvozOvtaCmqNTT53UqSNqj4R7NM37eP6nNBcJpdch06tbfI1RMtTFCNW82CMBUOx/9GJkILAXo9HMetJuG0wikEGzwQz+dsXlK9aJUWGaFvcbrq+9i4vFnxMakQ5e6tcnJyTQ3Z7bSeJq5UexqbNIHvLz0Qn79U0i2g79EYdq3u/X19kAMy3d2lUZJRH92FczrEVGZ0TnRT0C29SkWHTKdwaspIpxVdKkdGGmE3JhOjOJ2LHQyQsu4i477oFfJEBoNZ4qN5apHUMoTEpNITMzxBtlvPTr3nqp38xNj6Xaq/KGTv381OqrfE6ka0pz83kwVeT9cd7OCPbxjK8QkMK5PZfzdfeT+ofujS0F9N/W9GgWWVUL8bnXUduTdOLjN4dI9jSI4SiJvy8p5h1LDQ9RvfSJ5t8ftFyFMWE3TKasLJbiPr2ThKWjyNqSxh6j/YLXLQmWX1KSKEhqPjqa60jCHFYut//YP7bL3NZ4lYAbtiRzdJEI5i1myNh2WdiUK/TqSn9AINJVlYJZkq/XP7Ly/W/nxrei9lrlzVoUx3ZnRML5pBVbfr74Yc1oFcMP6dZqG7CfenokFrudF1OVNU3TT6/aBlmcypO5UIiMh8m4aMLiP2+M47i4J3PVb2pANWr/Qj97Oxsrr32WtLS0jCbzSQnJzNlyhTuuusujEYj3377bYP2FRUVpKWlMWfOHABGjx6NoigoioLZbKZz587cd999eDx17/8rV67k1FNPJT4+HrvdTkZGBkuXLm1wzldeeaX+HHq9nvDwcAYNGsTdd99NSUlJo8z79u3jvPPOIzIyEpvNRkZGBr/++msL/Qm1nMDvuxNHVL1xI+68PK1jCHFE3tw8pr74CmnTT+XOin4s2lvAlsgQ7jKboVSbISJJpalk2bZRbf6OwtyRmmQQ2rL3cPGI8d76348O687ta5Y3aFMT3oVJRTeyu6ppu94rCvQbkMA34UevMHTAxaHbGVU8T3b51pBOZ8FqHIYxdxjOTSlYS4O3sPg7b6Wbmu0lWDqGaZZh586dDBs2jLCwMB588EHS09NxuVysWLGC559/nmuvvZZZs2axbt067HY7APPmzcNsNrNw4cL681x66aXcfffdVFdX89FHHzF79mz0ej3z589n1apVpKenM3/+fGJjY1m2bBkzZ87E6XQyZcqU+nM4nU42b96MqqoUFxezatUqFi5cyMsvv8x3331HQkICAEVFRQwbNowxY8bwv//9j5iYGDIzMwkLC2vVPztfkMngQS5/0SLy/vnE0RsK0UYUTB7LvMjR7C+10cFm4gVnOKZsbZacVUINLNv2LOEdRpK7O1WTDEIbzhQ9/4yfS41Stxpa39COPLfheyyugz+LtWEdOLnsFrZUWJt0TpNBR7fBCfxoP/rHcphBx42Wd0gte615L0AcN4u5E+aaURi3ZhCx045yjDu7Bwv7kHjCT+2o2fUnTZrE+vXr2bx5c30h8afi4mIsFgt9+/Zl7NixPPXUU3z55ZdMnDiRVatW0b9/f6CuRyMjI4PHH3+8/tjx48dTXl7O999/f8jrTp48mdjYWF566SWgrkfj+uuvp7i4uEG73NxcevTowcSJE3n99dcBuOmmm/juu+/45hv/76WUoVNBrnylzM8Q/iVy2Rc899t/GBJbRGZlLVPy8shPsR/9wBaglrg5Mf0y9m1cRmh06dEPEAHBHm3klYR76ouMjo5kntz8a4MiwxXantMrbmpykRFiMdBuRNOKjO42uE93pxQZGtDrbDjME4kovJPYL24m7pPhRO5wSJFxBNUbtZunUVhYyPLly7n66qsbFRkAYWFhWCwWlixZwvPPP897773HRRddxC233FJfZByO1WrFdYSFdEpKSoiIiDhqxpiYGGbMmMEHH3xQPxTrgw8+oH///px55pnExMTQp08fXnjhhaOeqy2SQiOIuQsLqfrtN61jCHHMDGt+544PX+L8xJ1UeLyctiuLdRpNEtdleRk/4CIq8t/DaJXV2wKd2W7gf52fJZe6Xb4TrDE8t3s7zqqDY6zdIUmcXX0LG8qaVgDHhlpwDIvnd9PRi4wpzlLmV84itGZd816AaBaLpSth3suJ3PgQCR+dQ/Qv7bFUyC1UU3hKanHlabM0+bZt21BVla5dux6xXf/+/bn55puZNm0akZGRLFiw4LBtvV4vy5cvZ8WKFYwbN+6Qbd566y1+/vlnLrzwwibl7Nq1K2VlZRQU1BVl27dvZ9GiRXTq1IkVK1ZwxRVXMHv2bJYsWdKk87Ul8q8kiFV88w14vVrHEKJZ1P1ZzHhpMXdGrcao93L1jmzeSjKBsfXf1kKzQ+mY0A+77StUGSsfsPQGhXUZH7LRux6AMFMoz+YVEVOSVd/G40hgumsBq0uathRzaqyDqoFR7DAc+b3YpCjMCV3DOSUXYlDLmv8iRJPp9Q4c5klE5P8f8Z/OI/azIYTvliFSzVGzrViT6/45O0BRjv53tmDBArxeLzfddBMGQ+MpzM888wwOhwOLxcIpp5zCeeedxx133NGo3cqVK5k1axYvvPACPXr0aFZOr9dL3759ue++++jTpw+XX345l156KYsWLWrS+doSKTSCWLmsNiX8nFpVxaAX3+B5z+dEOap5fE8+C8O84Gj6EqK+CQJdPf2grILY5C2te23RavIHrGelt26yt9Vg5ZkKhdS8gztve+yxzPTcxk/Fziadr0f7MHb3DiVXd+TiNM6k417rS/Qv/r/mhxdNZrX0IMx9NdHrHybxw7OIXp2EqUpul46HVoVGp06dUBSFTZs2HbWt0Vj3uXGoIgNgxowZrF27lszMTKqqqli8eDE2m61Bm6+++oopU6bw6KOPMnPmzCbn3LRpE06nk8jISADi4+Pp3r17gzbdunVj9+7dTT5nWyH/coKU6vFQ/t13WscQwifi3lvB4i3vkBFVyrK8Ui6hHHd008bG+4pa7WF08rnkbPmKiDjZpCrQ6Prl8W9eBMCgM/CYJ5xee9fXP++1RXEpt/NdUWiTztevWzTrO9upOMoXrQPsXu7x3kBCxUfNzi6OzmAIJcR0CpG5C4n/ZC6xXwwgdF/TVgoTR1ezo0STZW4jIiKYOHEiTz/9NBUVjfc9+vvE7CMJDQ2lY8eOJCcno9c3Xlls5cqVTJ48mfvvv5/LLrusyefNzc3ljTfe4LTTTkOnq7stHzZsGJs3b27QbsuWLaSkpDT5vG2FFBpBqmrNGryHWLdZCH9l+mk193/6MmfG7+WP8hpOK8qnJLl1J4mrBS4m9b+C4n3vYrHLzsyBwtHdzSJT3TK2Cgp3m9ozbPsP9c97rRFcpb+DLwrCm3S+AX3j+K6dCddRiozpoTnMrpiBzbWj2dnFkVktGYS7ZhO9+kESPppK1Np4jNVya+Rr3ko3rixtNjh95pln8Hg8DBw4kLfffputW7eyadMmnnjiCYYMGeKTa/xZZMyePZtp06aRnZ1NdnY2hYWFDdqpqkp2djZZWVls2rSJl156iaFDhxIaGsr9999f327OnDn88MMP3HfffWzbto033niD559/nquvvtoneVuT/GsKUhWrVmkdQQifU3ft4eLXXuDm6N8p9bg4eU8Wm1Nbt9gw7dPTv8tkjIbPQJH5Gv7OmaLnKeet9XNv5oZ0Z8qmL+qf91rCuM54B8vzIo96Lr1Ood/QRL6JPvI+Czadwq0hXzK5+Cp0qhSsvmYwhOM0TiUy6wESll9HzJd9cWabtY4V8LQaPpWamsrq1asZM2YMc+fOpWfPnowfP57PP//cZ3MeXnnlFSorK1m4cCHx8fH1v6ZOndqgXWlpKfHx8SQmJjJkyBCee+45LrjgAtasWUN8fHx9uwEDBvDuu+/y5ptv0rNnT+655x4ef/xxZsyY4ZO8rUn20QhSu847n8pfftE6hhAtQ6djz9TJzDUMpazayM3topm8vxbcrfR2p1dYrX5JjT2U3D09W+eawufsUUaWdrqHbOome88K68XcNQd3/VbNIdxouZu3c2KPei6rSU/qkHhWW478M5hq0TGHh4mskqGtvqVgs/TFXDoC64buhOQ1fZd24RvmzuFEXyTvh8FGCo0g5K2tZcuAgag1NVpHEaJFVQ8byE1dJrG50MHU2FBuKNdDhbtVrq2EGFi+ezH2xPEU7I9ulWsK3zHb9Xye8SK/e9cCcEp4L/5v9cf1O3CrJjs32+/mX1nxRzhLnQi7iZBBsWw2HnllqTEhNZxfPhuzN/+484s6RmMUNnUMxl0DCd0chcEtK0ZpRTHqSLhjCIpBBtMEEyk0glDlL7+w67zztY4hROvokMqT48/g4+xYejutPGEKQZ9f3SqXVuP0fPj701jDz6OyTIZm+AudQeGPwcv5wvMxACPCuvHEus8xeOuKVNVo486Qu3h1f+JRz5UUYaOqbwT79If/qNUBl4RuY2TxTfWFjDgeOuyW/piLRmLb0BV74aFXERKtL/qKdMztm7ZggggM8q8vCMmQKRFUMncwO/85upx9No/ndWGqoZalSTE49rb8xEQl28OY9PP4cfdyFP0UVI98k+cPigZsqC8y0p0deGTDNweLDIOFhaG38+reoxcZnROd7OkRQtER5uqEG3TMtbxFavFS34QPYiZjHFbvGEw7BhC2NRydR3ov2pqaHaVSaAQZ6dEIQrsvvoQKWdpWBBu9nm1nnsJ8dTA1NXpeTk0gbUd5q1w6M2QDOd5y8vb1bZXriebT9yvkadNdAKQ5kliSuYnQyrrlilW9iUcj7+DJ3alHPU/vjhGsS7NSfYR73R42uMZ9B86a332SPRgpigGbeSCWghHYfu+ErUS+P23LLF3CibpQ5mkEEyk0gozq8bBlwEC8lZVaRxFCExVjhjO33QR2ldi4KyWWcXurwdOyb4OKWcc3Ze+hi+hD3t64Fr2WaD5HNw+PhM1FRSXWGsXr+3OJK94LgKoz8lTU7Tyyu8NRzzOgVwzfxRvwHmE34lOcpUwrvRaD2jrFbqAxmRKwucdiyhxAaKYTnVd6L/yBYtGTcPsQFJ38fQULKTSCTNVvv7PzzDO1jiGEptSunXhoxBl8mRfJjPgwrizWQVXLThJXIox8nPkcpogzKC9q3c0ExdE5k/U8mfQPqqjCaQphSZGLDrl1u7yrOgMvxCzgvp2dj3gORYF+AxL4NvzwN1EmReGakF/oV3KfT/MHA0UxYjcPwZI3DNvvHbGWHXmZYNE2xczugynBoXUM0UpkwHCQqVqzWusIQmhO+WMrN73zHFfEZLI0q5i55mq84S07WVstdHFCjwvxVC1Db/C06LXEsbFHGlmSvJAqqrDozTxdaTpYZCg6Xo29+ahFhlGvkD4s6YhFRrxJx33WF6XIOEZmUzvClQuI3voICR9eSOT3XYKiyHjq+9dJfmAkd372xGHb/G/zV0z/1w30fmIK3R47kVNfu5KV239q0ObrHT8z8vnpdH/sJOYsu5daj6v+udKackY+P519pTkt9jr+rnZnaatdS2hPCo0gU7V2ndYRhGgTvAWFnPrSCzxs/5k15eWcXVlEVYKtRa9p3K/Qp91YwqJ+btHriKYz2fR82vVFstV9GBQDDxNDxp41AKgo/Ct+Pnfu6HbEc4RYDLQfmcSP9sMPEBjk8HC3Zw7xFR/7NH+gUhQTDssYokpuI/7Lu4hZMYbw7U4UNTiG3KzN2sQb6z6gW/SRh+r9uGcdI1L78+qZD/LxBS8wpF0fLnr7Jn7PqSuUvaqX2R/dw3kZp/Luec+wdv8m3lj3Yf3xC1c+y3kZp5LoPPpeML5Ss7Ok1a4ltCezpoJM1dq1WkcQou1wu+mx9N+8OiGPObHjODnLw6vt40na2XLj5qPz48jSZ6JP3kPunuQWu444Op1eYWOfFaz31vX03m5JY9TGz4C6IuPdxBu5ObPXEc8R67SgGxjNb/rD75ExIzSLk0quR1FbZw8Xf2Yxp2KtGYv5jz6E7grO4TUVtZXM/vAeHjhxHk+sWnLEtneeMLvB728adRmfbv2Wz7atomdsZworSyioLGZm39OwGMyM7zSMrfk7Afh572+sy97M/42f01Iv5ZBqpEcjqEiPRhBx5ebi2r9f6xhCtDlhn3zB4rVL6RNZyDk79/Nde2vLvTt6VXqbRlKd8wvOyJZfYlccXsmATXzmrft29zpnT04/UGQALEuaww2ZfY54fGqsg6pBUew4TJFh1ysscH7OpOJrpMg4Ap3OQoh5PDFFdxL/6e3EfDIiaIsMgAWfPsbYDkMY0b7/MR/rVb2U11YSZgkBINIWRowjkq93/EyVq4af9qynW0wHaj0ubvnkERZOmIte17rD0LyltbgLqlr1mkI70qMRRKrWybApIQ5H99tG7szN441TzmX+Trg4MZIL871Q4/v5FGqFmzHtp/PJriUYTafjqpW34tZm7FPEUmURAOeFpXPJmo/qn/skaTbXbDvyTV6PlDA2d7ZTrjv0cKk0i445PExEiSwlfjgWSyfslWMwb8ggZF/LDlv0F+9v/Jzfsrfw0QXPN+v453/6N5Wuak7uOhYARVFYdOpd3PX5U9z5+ROMSRvM2b0m8/QPrzMspR8Wg5nTX7+KwqoSLuw7lVn9pvny5RxWze4yDJGyKEYwkE+3ICLDpoQ4MjUnj+lLnqPz2WdyZ1ZvtoSFcJ/FglJS6/uL5boZ3mEqv+Z/R2HuSN+fXxxWSBcvD5vr9sqYFN6TeauX1T+3MvkqLts6+IjH9+0Wxc/JZlyHmS4wzlnNjLLrMHvzfZY5UOh1NuzGUVj2DcWxIQljTXDMuWiK/aU53Pn5Eyw9+xEshmNfnOK9jZ/x6Hcvs3jqfUTZw+sfH5iUzrK/FC7bC/fw9oYVLJ+1mDPeuJaL+5/J6NRBnPDSBQxKzqBbzNGXcD5erqxy6BPT4tcR2pNCI4hIj4YQR6fW1NBvyeu8PCmP2VWjmIGdl+MjMWf5fu8Z234L7cLaYUjeTu6eNJ+fXzTmTDbwdMQ8VFSGhHXh/9Z+ikJdr8T3yZcxa+vwIx4/oE8c38QceqiJHrgkdCsjim+uP6eoY7V0w1Y+GsvG3jiyLVrHaZPWZ28hv7KISa9cWv+YR/Xw4551vLL6XTJv/Oyww5w+2PQ5//jfAzx72t1HHHKlqirzlz/EbWOuxquq/J6zlcldRmM1WhicnMEPe9a2UqEhw0aDhRQaQUL1eKj+fYPWMYTwG1Efr2BJn2wW9D6VKTkeXmsfR2wLTBJvX9GFnNrlhMVEUpwb6vPzi4NsEUbeSF5IhVpBD2cqj2/8HqO3bqnPX5Mv5Nytow97rF6nkDE4gW9CDv18hEHHjeb/kFL8Zgsk9096fQgO/WgsewYTsjEB/eG6gAQAw1P68elFrzR4bO7H99Mxsh1XDpp+2CLjvY2fceP/7uepKXcwrsOQI17jX+uXEW51MqHTcIqrywBwed1YD/zX422dpbdd2VJoBAspNIJE7a5dqNXVWscQwq8Y1qzjgZxcFk8+l2k7vfwzNY5+Oyvx5ZfVqsvLoPDJfJbzBibLydRWy9tySzBa9Xze7SX2evfQ3p7AM9s2YKupKxzXJ5/PtK3jD3us1aQndUg831kO/Rff0wbXuBYQUi5f5gDYLL2wlY3GsqYn9ryW3Z8mkDjMNrpGN+zZtBkthFuc9Y/f/9VzZJfl8/jJtwJ1RcacZfdy57jZ9E3oTm55AQAWoxmnueGE+vyKIp5YtYR3znsagDBLCJ0iU1j8838ZmTqA73b9yjWDz2vplwmAt8yFp7wWvcPUKtcT2pFPtCBRs2WL1hGE8Evq/iwufn0RXc86h+t3qlyVGM25uW6oPfxypsd8jWIXo1PP4tvs/1FTPRYF+ebXl3Q6hc19P2ed9xeiLRE8u38/ERV18yc2Jp/LKVtPOuyxEXYTIYNiWW089N/3aaHFTC25Fr3q+6F1/sRgCMWhjMGyewiOTTHo3fIz3BJyygsabK63dO0HuL0eFnz6GAs+faz+8TN6nshjk29pcOwdnz/B5QPPJj4kuv6xRybdzA3L7uOlX9/mioHn0iehe8u/iANcWRXoO0mhEegUVVVlIGkQyHviSfKfeUbrGEL4L0Vh/ymTud4ynMEhEdxRa4JS304SL4wrYHNFFrl7uvj0vMGuYtAWXtM9TYjRwculKl2yNwGwNfkMJmw7HfUwm8AlRdio6hvBPn3jj0mzonBtyE/0Kbm/RbO3dTZLBvaSUVh/64G1UG4aRdOFTk4lZESS1jFEC5MejSBRs3Wr1hGE8G+qSsL7H/H6oGzmdT2FC9zhvBgbhjHHd+vBR+RGEmreiTuukMLsCJ+dN5gZ+5Twmu5pzHozT9RY6ZL9KwA7kk47YpHROdHJnh4hFCmNi4wEk465+ueIK1neotnbKoMhnBBlHKbtg3BujkLnld4LcexkQnhwkEIjSEihIYRvmH78hX9m5/LECdOZku/ijXaxROz20QemF3owkO+qPsRiH0V1hXxDfDxCOqs8YrkTvaLnASWO/ru+AWBP0mROyDzjsEVG744RrEuzUn2Ipwc7PFxSNQdrze6WjN4GKdgt/bAXjcL6WzcsxXL7II6PTAgPDjJ0Kgh4a2rY3LcfeFpnNQkhgoES4uDTadN5tKQzi9rH03Nnhe8miUcb+Dr/Q4qKx8FhbobFkYUkGViUPJ9yyrnd1oUzN3wKQFbiREbumInrMN/CD+gVw3fxBrxK4+fPD93PxJI5QbXLt9EYTYg6Fsu2gTi2RqDIz6PwFYNC4t3DUHTyMxXIpNAIAlUbNrBz2hlaxxAi8Oh07Dz9NOboB3F1XBynZ7nA5ZtJ4tUJLtaU/kbu3l4+OV8wsYUb+U/3B9jr3cVVzp5cue5jAHITxjFi54XUeHWNjlEU6Nc/gW8jGt/0OPQKN9hW0KX0uRbP3jbocFgGYisYiW1dF8zlh15WVYjjFTunL8ZYu9YxRAuSvs8gIMOmhGghXi/t336HN4ZnM8c4iS3hMcyvNEK567hPbdlvJDkyAk9CHgX7o49+gADAaNHzZfdX2evdxdnhvbjywK7fBfGjGLPr0EWGUa/QfUgi39obf+/WwaLjevUBIkp/aPHsWjMZ43B4xmHZOgDH9lDpvRAtzpVdIYVGgJNCIwhIoSFEy7J+u4pns7J5cOS5XK7E83S0E0Pe8e9bk1SSSr7uW6qcIVSWym7KR6PTKWzt9wVrvD8yPrwHt6z5HwBFccMYvediKjyNi4wQi4G4IfH8aGrcE3WCs5oZZddh8ua3eHatKIoBh3kwtrwR2NZ1xFQpvRei9biyKqG31ilES5JCIwhIoSFEK8jczvyCp3j/9POZWtyB15JjCN1znJMd3Sp9QkfxnXc5VfoRqIe4URYHVQzYynLvOwwM7cz9675Ap3opjR3E6H2XU+Zu/HEX67SgHxDFb4aGRYYeuCx0M8OLb2l0TKAwmxJxuMZh2TwAx87DbHcuRAuTCeGBT+ZoBIGtY8bizsrSOoYQwcFgYPPUaczz9OfZdgl03nH8H6TeeB0/lv5A7r4+PggYmEy9y3jCtoBuISm8tGUdjupSymL6Mzb7WvJqjY3ap8Y6KEwPI0fX8CMw0qhjrulNUsr/01rRW42iGAkxD8WWPQLr+g6YDrWslhCtSB9qIv7mQVrHEC1ICo0A5ykrY8uAgVrHECLolI0Zxez4CVwcncxJe2vAc3xvtSXxpWwoKyZvb7yPEgaOkE7wSNQNJNiieW3XTqLKc6mIzmBc7vVk1zReIrhHShibO9sp/1sHUS8bXO1aQEjtplZK3jrMpnaE1J6AZWNf7HsdWscRooGE2wejszX+MkAEBhk6FeBk2JQQ2gj58ite6prN3YPPZktUO64r00Nl85dFDc12EuPcR1VYFeXFVh8m9W8hCQYWRd1EmNnJ89m5RJXnUhXVkwl51x2yyOjbLYqfk824/vZl/umhxZxeci16tbKVkrcsRTHhNI/Atn8EtvUpGGql90K0Ta7sCsxpYVrHEC1EBvwGuJotUmgIoRXlj83c9dHT2Gt/Z7axCm/kcUzoVqFTTU/CrL+jN8qeOADWMAP/Tn0Y1eBhUamb5IKdVEd046SCG9hXbW7UfkCfOFa1a1hkWHQKNzp/4oziiwOiyLCY04j2XkrS+seJ++B8nL+0lyJDtGmyQ3hgkx6NACc9GkJoy5tfwNmvP02XaedwZk1fliTGYN/XvA9WtdpD/8hR/BD7Ldl7+/o4qX8xWvR83XMpWepeFrkj6L7/Z2rDOzG5+EZ2VjUs6PQ6hYxBCXzjbHiORJOOufpniC35tBWT+55OZ8FpHIV17zDsvyej/3t3jRBtmCvH/wt8cXhSaAS42h07tI4ghHC5yPjXazw2PpvzGMdjqe1ov6O8WadSC1z0SRzAr8oOcve083FQ/6DoILPfV6zx/sCDxhQGbf6K2rA0TimbT2Zlw2FlVqOe1KHxfGdpOEdmqMPNxVVzsdTsbc3oPmU1dyKkchzWNb2x5MlwOuGf3EXHvxS4aLuk0AhwLlltSog2I/zTT3m5Vxa3mM7gzPYdGLO7GrzHPknctE9HWqyd6qhySvODb3Jv1YAdfOz9Lzc7ujHxtxW4nClMq7iZP8ptDdpF2E2EDIpltfHg8rUKMDN0LycU34AO/xuCptfZcBrGYN0zFNvvCeg90nsh/JunqEbrCKIFSaER4FzZ2VpHEEL8hf6333kwN4+XTjqfLbHduLwIqD72G97Y/HgKHeuoKu2EqzZ43srNvSt4Vvc4l4b2YvraZbhDEjm75hZ+K2u4u3BShI3qfhFs1h0sMkL0Om6wfUzn4hdaO/Zxs1m64ygbg3V9OpaCxvNPhPBX7uJqVFVFUaRoDkSyvG0AcxcVsXXIUK1jCCEOQbFY+H7aDD419ecBbCjN+FZPcRj4VfcTe3b3aoGEbU9IJ3g0ai6nhXfjztXL8Djime6+gx+LG06+6JzgZE/PEIr+ct/S0apwvfcBwqt/bOXUzafXh+DUj8G6Yyj2TbHovHIjJgJT/C2D0DsbrxIn/F/wfA0WhNzSmyFEm6VWVzN46WJST8rmoshxLEqIwbL/2CZFquVu+sQOoDb5D3L2tG+ZoG1ESLyR56JvZlRYZ25bsxyPPYYLvbc1KjJ6d4hgXQcrf92LboKziullszF6C1s5dfPYLb1wlIzFuq475mLpvRCBz11ULYVGgJJCI4DJ/Awh2r7Y/y3jkb5ZXN/9dO5o34H4ncc2SVzJ8dAzMZGamBKKc0NbKKW2rKEG/pP2CB1Donlw/UoUSziXcjtfF4Y1aNe/Vwyr4g38+cW/HrgidBNDixe0euZjZTCEEqqMw5I5GPvmaBRVei9E8PAUVUOK8+gNhd+RQiOASaEhhH8wrl7NwznZPDXufEan9mLwrkrwHv24Pzn22UiNyWdDqZ3a6sB6WzeYdHzT6w1MVhdPbv4No8HK1Ybb+Twvor6NokD/AQl8E37w5jzKqGOuaSntit/SInaTOSx9CSkcjXVtN4zlsjuyCE7uYpkQHqgC6xNJNCBDp4TwH+q+/Vzz1pN8fvr5vBo/iAvyVahp+iTxlOL2lCX8zrbtHalbW8n/KQrsHPAtWeZtLNm9lxAvXG+5g//lRNW3MeoVug9J5Bv7wemGvW1wletmHOV/aBH7qIyGCEI5AcuWQdgyI6T3QgQ9jyxxG7Ck0Ahgrv3SoyGEP1ErKxn7xvPsmZLLnbHjuaPWhlJS27Rja730pDdV7Teyb2f7lg3aSqoH7OY7w3KW5FYSXVXGPNvdvJ8dU/98iMVA3JB4fjQd7P6ZFlrIqSWz0atVWkQ+AoUQS39C8kdjWdsFY6V8/Iogp4BiN6BaoaK2mHCt84gWIe90AUyWthXCD6kqyR+8yzWDspjf/Qz+Ly4OU3bTJomrhS4yEjpRFZ9LYZZ/f2xb0it51fwML5WbSCnOZoHjLv6bFVf/fKzTgn5AFL8Z6ooMi05htmMVvYsf1iryIRmN0YR5TsC6eSDWHf79dyJEkymg2OqKCI/RTY2uhip3KWU1RZSU51JQvI/Cwr14PG4AorJSuODsARqHFi1BCo0A5srar3UEIUQzmX/8gdtzsnho1EwuS+lG9K6mTRI37IceiWZ+Ka2hqsI/VywK6ajwVMjdPOYJp0fOOu4JvZOl+xLqn0+NcVDYO4wcXd1wqSSzjrm6p4kp+UyryH+jw2kZhCN3FNa1nTBU67UOJIRPKVY92HS4TW5qdTVUecoorymipCKPwpL9FBTsxeVq+nCo8sKCFkwrtCT7aAQo1evlj/Te4HZrHUUIcRyUEAcfnXohXaIH02dnBTTlHVunsD16J79viUf1s/H/jjgDi9NuZ54pnMlbv+PBiDtZtCel/vkeKWFs7mynXFf3+6EONxdXzcXi3qtN4L8wGeMIdY/HumkA1t2ygo7wT4pZD3YdHrOHWl0N1d5yymuKKanMpagki/zCvdRUV/j8ute99g4GkyxxG2ikRyNAufPypMgQIgCoZeVMfuNptp2Sw1vtJnFGthtqj7IklVelQ3kHytO2sSMzsXWC+oDFaeDtjv/kEmsskzd+weORt7No98Eio2/XKH5uZ8al1E13vyB0D+OK56Lj2HdW9xVFMeA0D8GeNRLrujQMtdJ7IdouxaQDux6P2YtLX0O1t4Ly2mJKK/MpKskir3AP1VWlmmQrLywgLC5ek2uLliOFRoBy7ZdhU0IEDK+Xju/9h4ThWTzV6xyuKbdBmeuIh6gVbtLtHSlP3Efevogjtm0LDCYd36f/m9EhZs5ft4JF0bfxz11p9c8P6BPHNzF1N/FOvY4bbMvoVPyiVnExmxIJrR2PdX0/LFkhmuUQop5Rh2LX47V4celrqfZWUOEqobSqgKLSbAoKd1NeXqR1ysOSQiMwSaERoGRpWyECj+3bb7ggO4t/jr6Yq61JGHKPsrJSrps+yZF8W1ZFZamldUI2g6LA7gHfE+csYs7aFSyOuZUHd3YCQK9T6D0ogW8OjETqZFW4znMv4aW/aJDTSKhpOI79IzCvS8Xg9q9hacKP6RUUux7VSl0RQSUVrlJKq/IpLsuhsHAvJaW5Wqc8LmVFMk8jEEmhEaBcWVJoCBGQtm3j4vwH+M8pl3BSuz6E7z7yWGnrHgO9kl38/IcJr0fXSiGPTe2AvRQ71/D4ui95PXY+/7ejCwBWo560IfGsstZNTJnorOTcstkYva37razFnEJo1QnYfuuHKcfWqtcWQUD3lyLC4KKGKirdpZRVFVBclkN+0V6KS7IhwKfUVsiE8IDUNj91xHHzFBVqHUEI0ULU4mKmvPE4f+xdzqZU+1Hbx+VG0aVj23xPsPaq4pfwd3how7e8E3sDt+3oAUC43UTsiAR+taoYFLgmdCMzS85vtSJDUUyEm8eTnHsXyR/fSdhnI6TIEMdOAcVhgGgD7gSoSKwiLzaHHc5NrNF/xWelb/DfHQ/xr/X38u8f7+Wd7x5k2XdP8uWPr/LL+o/YtuNniouzAr7IAKgs02ZuyJ+ys7O59tprSUtLw2w2k5yczJQpU7jrrrswGo18++23DdpXVFSQlpbGnDlzABg9ejSKoqAoCmazmc6dO3Pffffh8dTNIVu5ciWnnnoq8fHx2O12MjIyWLp0aaMctbW1PPjgg/Tu3RubzUZUVBTDhg3j5ZdfxuWqGzJbVlbG9ddfT0pKClarlaFDh/Lzzz+38J9Q80iPRoDylJVpHUEI0ZLcbnq+8xolo/fzScYMJuz1gvswk8RdXrrUtKMkZT/7d4W1aswjCUlT+CD2WRZv/Z1PYq5k/rZ0AJIibFT3i+APnZdoo465pqUkF7/VKpms5jRCK8ZjXZeBqcDaKtcUfkppuMxrjVJ9iL0i9uPxNG3TzWBXU960Jbxbws6dOxk2bBhhYWE8+OCDpKen43K5WLFiBc8//zzXXnsts2bNYt26ddjtdV/uzJs3D7PZzMKFC+vPc+mll3L33XdTXV3NRx99xOzZs9Hr9cyfP59Vq1aRnp7O/PnziY2NZdmyZcycOROn08mUKVOAuiJj4sSJrFu3jnvuuYdhw4bhdDr54YcfePjhh+nTpw8ZGRlccskl/P7777z22mskJCTw+uuvc8IJJ7Bx40YSE9vWAiCyvG2A2jf3RkqXLdM6hhCiFajdurFizBWcWeKEisNPEvck6PlqfxFlxdrfQDtijXzY4zGe2r+Z1Y4zuWpb3WZdnROc7OkZQpECfewqV9TeiqN2c4tm0emshBlG49g5DPPGRHRemXsh/rZXhFJDlbf8wF4Ruc3aK0IcWedBw5hyw82aXHvSpEmsX7+ezZs31xcSfyouLsZisdC3b1/Gjh3LU089xZdffsnEiRNZtWoV/fv3B+p6NDIyMnj88cfrjx0/fjzl5eV8//33h7zu5MmTiY2N5aWXXgLgwQcf5Oabb+aXX36hT58+Ddq6XC5qa2vR6XSEhITw/vvvM3ny5PrnMzIyOPnkk/m///s/X/yR+Iz0aAQoT7n0aAgRLJRNmzgp7x7eO+0qplg7os8/9M2Pfr+HASkWvqrw4HFptwyrJcTAZ91e5IH8PWwIOZ2rttYVGb07RLCug5VqBc4MLWBKyXXo1aNMeD8ONnMXQsvGYVmbjqm47U6WF753+L0i8g7sFbGnRfaKEIdXXaHNfUthYSHLly/n3nvvbVRkAISFhQGwZMkShg4dygknnMCcOXO45ZZb6ouMw7FarRQVHX64Z0lJCd26dav//dKlSznhhBMaFRkARqMRo9FIWVkZHo8Hi6Xhe5bVam00vKstkEIjQHnLtOuCFEK0PjU/n3GvLuTHaRfRK2kkIXsPfZMUsttE7y4lrP5dm/kGeqOONX3eZ27FTvaZx3Hx1iEA9O8Vw6p4Aya9wnzHt6QXP9oy19fbCNONxb5jKJZN8Sh+tqGhOLqj7RVRULSHykpt5wMcq8y8Alb+sZ19RSWUVtcwa1g/eibGHfEYt8fDpxu38uuu/ZRV1xBmtTCuW0cGpiUDsCU7j3dWb6C8poYeCbGc2T8dg75u6m5VrYt/fvYdl48aRLi9dXpAqzQaOrVt2zZUVaVr165HbNe/f39uvvlmpk2bRp8+fViwYMFh23q9Xj755BNWrFjB9ddff8g2b731Fj///DPPPfdc/WNbt25l9OjRR8wREhLCkCFDuOeee+jWrRuxsbG8+eab/Pjjj3Tq1OmIx2pBCo0A5ZU5GkIEH5eLXv96jvwTdlPQawbtd1Q2bqNCu4IIijsUsD3z6BPJfUqB7IG/cJZnIzUMYObWESgK9OufwLcRCslmHTcoTxJT8oXPL223dCe0eBzW1b0wlMvuw37LoKA4DAf3ilArqagt9pu9Ipqr1u0hIczJwNQkXl21uknHvPb9GsqqazhrQDpRDhvl1bV41bp5XF5VZemPaxnbtQNd4qJZsupXfty+m2Gd2gOwbP0fDOnQrtWKDICaCm0KjT9nECjK0b90WLBgAXfffTc33XQTBkPjW+hnnnmGF198kdraunk5559/PnfccUejditXrmTWrFm88MIL9OjRo0GWpuR47bXXuOiii0hMTESv19O3b1+mT5/O6tVN+9loTVJoBCiPhpOqhBDaivrsf3hy9vHLuGvpvwfwNJyKp1Z56OWIpjCqgOL81ruR8PTfTx/TKhzlnZm6dQxGvUL3IYl8a1cZHuLiwoq5WDz7fHY9vT6EcN047FuHYNkS67PzihaiV1AcelRLw70iyqryKS7LpaBwj9/vFdFc3eJj6BYf0+T2f2TlkplXwC2TxmAz1xXWEfaDvZgVNbVU1NQytGMKRr2e7gmx5JTW3TfsyC9kb1EJU/v29O2LOIpqjYZ8d+rUCUVR2LRpE6eddtoR2xqNRoBDFhkAM2bM4NZbb8VsNpOQkIBe33iI6ldffcWUKVN49NFHmTlzZoPnOnfuzKZNm46auUOHDnz11VdUVFRQWlpKfHw8Z599NqmpqUc9trVJoRGgvFJoCBHU9L+tp2vebXw7dS7DiyKhyt2wQZ6LIe3C+KysEldNy8/XsPasgfCP6FAUx8lbJxBiMRA/OJ6fzF4uDN3N2OIb0eHxybUclnRCC8dhWd0dQ6XRJ+cUx+koe0UUFO2lKAj2imgtG/bnkBweypebt/Prrr2Y9AZ6JMRwYs8uGA16HGYTTouZLdn5dI6NYkd+If3bJ+H2eHnn1985a0A6Ol3rDiusrapC9XpRdK2780JERAQTJ07k6aefZvbs2YecDP7nPI2jCQ0NpWPHjod9fuXKlZx88sk88MADXHbZZY2enz59Orfccgtr1qxpNE/D7XZTU1PTIJ/dbsdut1NUVMSKFSt48MEHm5SzNUmhEYBUVcVbIZPYhAh2anY2GS/dxq9nXU0fazq6wpoGz5t2exnUVc+361o2R0iajvykN5lc4GTi1snEOM0YBkSz0wx32pbRsfil476GwRBKhDoe++YhmLZH+iC1aDIFFLsB1Qpuo5saqqjylFFWXUhxeQ4FRfsoKtqP1+ubQlIcXWFFFTvyizDo9cwa2p+K2lre+fV3KmtdnD2wN4qicP6Qvry/diPvrd1At7gYBqYm8/mmTDrGRGHU63nq81VU1NYyrGN7hh8YUtXSaqurMNtaeUgndUOehg4dysCBA7n77rtJT0/H7Xbz6aefsmjRoib1MhzNypUrmTx5Mtdddx3Tpk0jO7tuY2WTyURERAQA119/PcuWLWPcuHHcc889DB8+nJCQEH755RceeOABFi9eTEZGBitWrEBVVbp06cK2bdv4xz/+QZcuXbjwwguPO6evSaERgLwVFeA9zHr6QoigolZX03nJI+yafBZx7U/Buq/hvI3IfSa6d61g4x8t882/I8bI7s6vcXaBgQnbTqF9jJ2i3uFY7bDQ83+Elf56XOcPsfQjNG80ljXd0FfLR5rPKaBYDWBTGu0VUVqRR0HxPgoK9sleEW2MqqqgwPRBGVhNdf+2T8noxpJVq5natydGg57U6AiuHz+8/pi8snJW79rLnPEjeObL7xnROZUucdE8vOJr0qIjSAhztnju2iptCo3U1FRWr17Nvffey9y5c8nKyiI6Opp+/fqxaNEin1zjlVdeobKykoULFzbYe2PUqFGsXLkSALPZzKeffspjjz3Gc889x4033ojNZqNbt27Mnj2bnj3rhrOVlJRw8803s3fvXiIiIpg2bRr33ntv/dCutkT20QhArqwsto0Zq3UMIUQb4+o7AM/Ia4ne2XD5W8Vp4HtXKTlZvv2QMjsM5Ax9l7OKi5i4dRpd20WwpbOd4WEVnFN6LUa1pFnnNRoiCPeOx75xEKbdET7NHGwO7hXhoVZXTZXnr3tFZFFQsEf2imhjbvzPsqOuOvXmj2vZWVDEzZPG1D+WU1rGQ8u/Zv5Jo4kOaXgzr6oqi1b+wKjOaXSIiWDBu59w39QTMRn0LFn1K2nREQzv1PLj/2c98gyRSe1a/Dqi9cjXPwFIdgUXQhyKcfXPmHJvYee0W2m/xwDeuu+Z1FI3gxLCWF5cQm2Vbz4W9EYdhQNXckZJEZO2nUFGl2hWp5i5NPQ3Bhff2YwzKjgtAwjNGY1lTWd0tfLxdTSN94qoOLBXRK7sFRHgUqMiWL83ixqXG7Ox7t9KXlkFigJh1sZ7xvy0Yw92k4keibFU1tZt+unxegE9Hq/651tFi6utarl9c4Q25J06AMlEcCHE4ah79xL1wly2n3czaaUJUF03bl7ZX8uoLg4+XVsFHOckUAXcgzdyUtUOTtk6jYze8fyRYOAu02skFb97TKcyGaMJd03AtmEgpn2hx5crgChGHTj+vldECWVV+QeKiN1+t1eEOLwal5v88oNFYWF5JfuKSrCZTITbrXy8/g9Kqqo5d1AGAH3aJfDpxq38++d1TOjRmYraWj5a9wcD2ydjNDRc/KGsuobPNm7jmrFDAbCZjMQ4HXyzdQedY6PZlpvPuG6Hn+Ds09dZdYgluYVfk0IjAMkeGkKII1ErK4l+4Xb2T7uY+LARKMV1k8Rtuzz062ng19+Pb9KudUA+6d7VnLN1Kj0GJVMer3JvzTzsFduaeAYdoeYhhGaNwrSuA3oNdzHXhKFuhSavRcVlaLhXRHFpNgWFeykrL9A6pWhFe4pKeHblD/W//2Bd3eTk/u2TOGdgb0qrayiqPNgbYDYauHzUIN5ds4F/fvYtNpOJ3snxnNSzS6Nzv79mI6O6pBFqO9jTcc6A3vzrp7V8u3Uno7p0oF1kWMu9uL9wSY9GwJE5GgGo5KNl7L/xRq1jCCH8gGfoWOy9Z2HMqhuHr1j0rDZWs3tP884X2sNFu8i3mbPlVOIGJtEhLo+TS65Drx59nL/ZGEd47QTs6wdgyAlpXoC27hB7RVS6Sw8UETlBvVeEECddfQPdR8oc00AiPRoBSIZOCSGaSr/qC2r278I15VZse9yo1R762u0UhJRRUXZs69mHttcTHfMuCzJPJ3JYIuMjvqZX8WNHPEZRDISZhuLcNxLT2jR03tZdQ9+ndAeXef37XhEl5bkUFO2lsDhL9ooQ4jC8HlkCOdBIoRGAVI/76I2EEOIAdWcm3leuo2TmPYTut6EW1DI6NZSPfytFVZs2X8MebcSZ9hH/3DuF0OHRzDA+QXTJl4dtbzElEVE1Htu6fujzHb56KS1HAcVmQLX9uVdENVWe0vq9IgqL9lEoe0UIcVyk0Ag8UmgIIYRALStDt+gGSmbcSGhtFwy7qxney8E364++KpHZbiCk19e8nj+B2KE2zqi5FktVVqN2imIk3DQC566RGH9rh05tI70Xf98rgmqqvGWUV9ct8yp7RQjROjzyRWnAkUJDCCFEHa8X3WsPUjl+Gvb4k4nY56JrBzN/ZNYc9hC9QSG0/+98VDmUrgOLGFl6OQoNhwZZTSmEV0zEuqYPhmJrS7+KRg63V0RpRR6FJfvJl70ihGgTVOnRCDhSaAQiGf4rhDgOnk/fpqJ7JrZh19G1CnLDdRQWeRs3VCB8UDZf67sxpPs3dCh9tf4pnc5MuGEUzh0jMG1MbrH3pbq9IhQ8Zu8h9orIJr9wt+wVIYSf8EihEXCk0BBCCNGIZ+NaKrL/ge3s+xjhcPBhaSl/n34Q3beGNaERjIl8mtDyNQDYzB0ILx2PbU1vdKXH13vReK+ISipcxZRWyl4RQgQimaMReKTQCESyookQwge8hfmUv3AV9gvuYVzPdny6rrj+uahuBna2z2OU4QHMrloijJNwbhuGcXNi005uUFAcBrxmr+wVIYKWouhQdAqKToei/Pnfg/+PoqBTdKDoDrb5y3P1bZU/f//X/6977s//hwPH8Zc2KHXn/rPdgd/Xbdr553HUH4+i/O25A4+pDZ87+OvPeVgNH1PrzoDKgWMP/L/Z3qm1/uhFK5FCQwghxOG53VQsvhn7aVeQ3mUI6zeXE5ZiprLXKkaavyS8+Dysq3ujqzAdPEZft+GcaqGuiKCKSneJ7BXRmg7cdOp0uvobyUY3s4e4cT14U6ugcOBmVPf3m9mDN6mNHvvzZvOvN7cNbmr/fpN64Eb2rze4f7b5682p8ucN68Gb4EP/OtBGPTBTSNEdGLZX91/1z3bqgZvcA1dXD9zs4j3YRj1wnKpysL0Kqnqwvddbd1XVW3fzjBe8B85f95yC6gWvWnesVwW8OrzqgdfcBA0GLaoHfh1iJGMg6NA3VOsIwsek0BBCCHFUFe89S1y/zRSkn0to+u8kZSdjLLiFGqWKwpBSyi3FlJbnUVSWTWl5LuTS8FvYv96sWo2E2ZIO+81r3f//pf1fb1wb3MAqB//b4LG/3Nj+7ea27ptUHSoNb3zrv3lVFFD/elNL/fN//RZWqb9RPfitrIJSdyP553F/vTnlzxtepX7JYNX75/GAt65t3U1t3c2tWt8eVFV34Ca47jiVA+3++tifN8FeDvw5HNmf96yNfyNaWtNKDCH8nxQagUiGTgkhWoD71y/xdjqLje93YiMAtYAeCD3wK6Wu4d9XrVUBGXrdappQYwjRRskPb6BpI4uYCyGEaOuq+4wjKztAx2wIIYTwOSk0hBBCNMmebqdrHUEIIYQfkUIjIMnQKSGEb9V2HcCeLBnWIIRoQfIWE3Ck0BBCCHFU+/rOkO8whBBCHBMpNAKRTAYXQviQO6U7u7JNR28ohBBC/IUUGkIIIY4oa+gsvF75AkMI0bJ0ehk7FWik0AhEivy1CiF8wx2bwvb8EK1jCCGCgMkiuy4EGrkjDUA6u03rCEKIAJE3+lI8LlnSVgjR8kwWvdYRhI9JoRGAdHaH1hGEEAHAGxpFZkmU1jGEEEFCejQCjxQaAUjnkEJDCHH88sdfQW21bOkthGgdJqv0aAQaKTQCkN5h1zqCEMLPea0OtlUlax1DCBFEjGbp0Qg0UmgEIOnREEIcr6IJl1Nd4dY6hhAiiEiPRuCRQiMASaEhhDgeXpOZTDprHUMIEWRkjkbgkUIjAEmhIYQ4HqUnXER5ifRmCCFalxQagUcKjQCks9tBkU1vhBDHTlUUtlv7aB1DCBFkdAYFvVFuSwON/I0GIEVR0NlkLw0hxLErH3s+xQUurWMIIYKM9GYEJik0ApQMnxJCNMeOyGFaRxBCBCHZrC8wSaERoKTQEEIcq8php5OfK3MzhBCtzyg9GgFJCo0ApZdCQwhxjHYmT9A6ghAiSEmPRmCSQiNASY+GEOJYVPc9gexsr9YxhBBBymSVHo1AJIVGgJJCQwhxLPZ0PU3rCEKIICaTwQOTFBoBSuewax1BCOEnarsNYk+WLIkthNCODJ0KTFJoBCh9WJjWEYQQfmJfn3NB1TqFECKYyWTwwCSFRoAyxsZpHUEI4QfcKd3ZlW3SOoYQIsjZQuR9KBBJoRGgDHGxWkcQQviBrKGz8HqlO0MIoS1HhFnrCKIFSKERoIzx8VpHEEK0ce7YFLbnh2gdQwghCIm0aB1BtAApNAKUIVZ6NIQQR5Y7+lI8LlnSVgihvZAIKTQCkRQaAcoQFQVGo9YxhBBtlDc0iu0lUVrHEEIIdAYFm1PmaAQiKTQClKLTYYyJ0TqGEKKNyh9/JbXVHq1jCCEEjjAziiJLbAciKTQCmCFOVp4SQjTmtTrYVpWkdQwhhABk2FQgk0IjgBml0BBCHELRxCuornBrHUMIIQBwSKERsKTQCGCyxK0Q4u+8JjOZaietYwghRD3p0QhcUmgEMGOcLHErhGiodNxFlJdIb4YQou1whMseGoFKCo0AJj0aQoi/UnV6Mq19tI4hhBANyB4agUsKjQAmPRpCiL8qHzODkkKX1jGEEKIBGToVuKTQCGDGeJkMLoQ4aEfkMK0jCCFEIzIZPHBJoRHA9JGRKLJpnxACqBw2lfxcmZshhGhbLHYjRpNe6xiihUihEcAURcEQK/M0hBCwM3m81hGEEKIRR4RMBA9kUmgEOFNKitYRhBAaq+47nuxsr9YxhBCiEZmfEdik0Ahw5o4dtY4ghNDY7q6naR1BCCEOSeZnBDYpNAKcubNszCVEMKvtNoi9WVqnEEKIQ3PK0rYBTQqNACc9GkIEt319zgVV6xRCCHFokQkOrSOIFiSFRoAzdegIiqJ1DCGEBlzte7Ar26R1DCGEOKzIJCk0ApkUGgFO77BjjJeN+4QIRtlDZ+H1SneGEKJtsjlN2JzyZUggk0IjCJg6yfApIYKNOzaF7XnyTaEQou2S3ozAJ4VGEJB5GkIEn9zRl+JxyZK2Qoi2KypRCo1AZ9A6gGh55k4tt/LU8wUFfFZexvaaWiw6hQyrlbnR0aSaDm7AU+H18lheLp+Xl1Ps8ZBoNHJeWDjnhIcf8dxLCgv5V3ExWW4X4Xo9E0JCmBMVjVlXVx9/WFrCY3l5VHq9TAsN4x8xMfXH7nPVcsmePfw3pT0Ovew4KoKLJzyGzOJIQAoNIUTbJT0agU8KjSBg7thyhcYvlZWcGxZGT4sVj6ryz/w8Ltmzhw9T07AdKAgeyM3hx8pKHoiPJ9Fo5LuKSu7JySbaYGBcSMghz/thaQmP5ufxf3Fx9LFa2Vnr4pasujU6b4qJpcjt5vbsbO6LiyfJaOTKfXsZaLMxylH3pnVXTg43RMdIkSGCUsG4y3HlS5EhhGjboqTQCHgydCoImDukga5l/qqfT07m9NAwOpnNdLVYuDcuniy3m43V1fVt1lZVcZozlIE2O4lGE2eFhdHFbGZDTfVhz7uuqoo+VisnO0NJNJoYZrczyRnC7wfOu8flwqHTcZLTSS+rlYE2G9tqawD4qLQEo6Iw/jBFjBCBzGt3sq0qSesYQghxRDqDQlicTesYooVJoREEdFYrxqTWufEo89Z9ixr6l56EvlYbX1aUk+NyoaoqP1ZWsLPWxTCb/bDn6Wu1sbG6mvVVVQDsqa3lm4oKRtnrvv1IMZmoVlU2VldT7PHwe3U1Xcxmij0enszPZ0FMbAu+SiHarqLxl1Nd4dY6hhBCHFFEvB29Xm5DA50MnQoS5o4dce3e3aLXUFWVB3Nz6Wu10sl8cI7GLbGx3JGdxZjtmRgARVG4JzaOfrbDf5Mxyemk0OPmvN27AHAD54SFcWlkJFBXyCyMi+fmrCyqVS+nOJ0Mtzu4NSuL88LD2edycfW+vbhVlaujopgY4mzJly5Em+A1mdnm7UjdvxghhGi7ImUieFCQQiNImDt2pPyLL1r0Gv+Xm8Pmmmpeb5fS4PHXiwpZV1XN04mJJBiM/FJVxd05OUQZDAy1H7pX46fKCp4rKOD22DjSrRZ217q4LzeHaH0+V0ZFAXBCSAgn/GV41E+VFWytrWFBbCwnbt/OwwkJRBn0nL1rF/2tNiIN8uMuAlvpCRdTUSpFhhCi7ZP5GcFB7ryCREuuPAXwfznZfFlezpLkdsQZjfWPV3u9PJ6Xx5OJSfUTtbtYLPxRU80rhYWHLTSeyM/nFGcoZ4SFAdDZbKHS6+XOnGwuj4xE97fdzmu9Xu7OyeHB+AR219biQWXAgR6T9iYT66urGOOQORsicKk6PZmWDKh0aR1FCCGOSlacCg4yOC5ImDt3bpHzqqrK/+Vk81l5OS8ltyPJ1HCHT7eq4gaUvx2nQ8HL4XcsrvZ60f3tIL2ioMIhj1pUUMAIu53uFgueA9f9k0tV8cjmyCLAlY89j5JCKTKEEP5BejSCg/RoBAlzxw4oNhtqZaVPz3tPbg7LSkt5KjEJu05Hnrtu2EaITodFp8Oh1zPAauXhvFwsOoUEg5Gfqyr5oLSE+dEH9724KWs/MQYDNxx4bLTDwatFRXQzW0i3WNjtcvFEfh5jHA70f+vN2FpTw//KSnmnfSoAaSYTOkXh7eJiogwGdtTW0sti8enrFqKt2R4+FPJk2JQQou2zhZqwOkxHbyj8nhQaQULR67H26EHlzz/79Lz/Ki4G4II9DSea3xsXx+mhYQA8nJDIY3l5zMvKosTjIcFo5LqoaM4+MCwKIMvlatC9dkVkFAoK/8zPI9ftJlyvZ4zDwXVR0Q2uo6oqd2Znc1NMbP2+HRadjvvi4rknJ5taVWVBTCyxfxnOJUSgqRg2lQIpMoQQfkJ6M4KHoqqqDCoJErmPPELBCy9qHUMI4WMbz3mW7GyP1jGEEKJJ+kxox9CpHbWOIVqBzNEIItbevbWOIITwseq+46XIEEL4FenRCB5SaAQRKTSECDy7u56mdQQhhDgmcWmhWkcQrUQKjSBiiI7GmJCgdQwhhI/UdhvM3iytUwghRNM5Isw4o6xaxxCtRAqNIGPNkF4NIQLF3j7nHnq9ZyGEaKMSO4VrHUG0Iik0gow1o4/WEYQQPuBq34Nd2bJwoBDCvyR0DtM6gmhFfltoZGdnc+2115KWlobZbCY5OZkpU6Zw1113YTQa+fbbbxu0r6ioIC0tjTlz5gAwevRoFEVBURTMZjOdO3fmvvvuw+Opm1S5cuVKTj31VOLj47Hb7WRkZLB06dIG53zllVfqz6EoCg6Hg379+vHOO+80yrtp0yZOOeUUQkNDCQkJYfDgwezevbtRu5Zm7de31a8phPC9rCGzUL1apxBCiGOTKIVGUPHLQmPnzp3069ePL774ggcffJDffvuN5cuXM2bMGN58802uvfZaZs2aRUVFRf0x8+bNw2w2s3DhwvrHLr30UrKysti8eTOzZ89mwYIFPPzwwwCsWrWK9PR03n77bdavX89FF13EzJkz+fDDDxtkcTqdZGVlkZWVxZo1a5g4cSJnnXUWmzdvrm+TmZnJ8OHD6dq1KytXrmTdunXcdtttWDTYRM7StSs6u73VryuE8B13fHt25Mm/YyGEf3GEmwmNtmkdQ7Qiv9xHY9KkSaxfv57Nmzdj/9tNc3FxMRaLhb59+zJ27FieeuopvvzySyZOnMiqVavo378/UNejkZGRweOPP15/7Pjx4ykvL+f7778/5HUnT55MbGwsL730ElDXo3H99ddTfGDTOgCv14vFYmHp0qWceeaZAJxzzjkYjUZee+01H/4pNN/uSy6l4m89PkII/7H/nHv5IztM6xhCCHFMOg+MZfxFPbSOIVqR3/VoFBYWsnz5cq6++upGRQZAWFgYFouFJUuW8Pzzz/Pee+9x0UUXccstt9QXGYdjtVpxuVyHfb6kpISIiIjDPu/xeHj11VcB6Nu3boiS1+tl2bJldO7cmYkTJxITE8OgQYN47733mvBqW4atfz/Nri2EOD6e8Bgyiw7/PiSEEG1VYmeZCB5s/K7Q2LZtG6qq0rVr1yO269+/PzfffDPTpk0jMjKSBQsWHLat1+tl+fLlrFixgnHjxh2yzVtvvcXPP//MhRde2ODxkpISHA4HDocDk8nElVdeyfPPP0+HDh0AyM3Npby8nPvvv58TTzyRTz75hNNPP52pU6fy1VdfHeOr9w1bPyk0hPBX+SdcjqtGJmcIIfxPQqcwrSOIVuZ3S5b8OdJLUZSjtl2wYAF33303N910EwZD45f6zDPP8OKLL1JbWwvA+eefzx133NGo3cqVK5k1axYvvPACPXo07PILCQlh9erVAFRWVvLZZ59x+eWXExkZyZQpU/B6624ITj311PqJ6BkZGaxatYpnn32WUaNGHcOr9w1LejqK0Yh6hN4bIUTb47U7yaxMAtxaRxFCiGNiDzMTFivzM4KN3/VodOrUCUVR2LRp01HbGo1GgEMWGQAzZsxg7dq1ZGZmUlVVxeLFi7HZGv4j+Oqrr5gyZQqPPvooM2fObHQOnU5Hx44d6dixI+np6dxwww2MGTOGBx54AICoqCgMBgPdu3dvcFy3bt00WXUKQGc2Y+nVS5NrCyGar3D85VRXSJEhhPA/0psRnPyu0IiIiGDixIk8/fTTDVaV+tNfJ2YfTWhoKB07diQ5ORm9Xt/o+ZUrVzJ58mTuv/9+LrvssiafV6/XU1VVBYDJZGLAgAENVqEC2LJlCykpKU0+p6/ZhwzR7NpCiGPnNZnJ9HbUOoYQQjSLLGsbnPyu0IC6IU8ej4eBAwfy9ttvs3XrVjZt2sQTTzzBEB/dQP9ZZMyePZtp06aRnZ1NdnY2hYWFDdqpqlr/3I4dO3j++edZsWIFp556an2bf/zjH/z73//mhRdeYNu2bTz11FN8+OGHXHXVVT7J2hyOkSM0u7YQ4tiVjr+EilLpzRBC+CeZCB6c/G6OBkBqaiqrV6/m3nvvZe7cuWRlZREdHU2/fv1YtGiRT67xyiuvUFlZycKFCxvsvTFq1ChWrlxZ//vS0lLi4+MBMJvNpKSkcPfddzN//vz6NqeffjrPPvssCxcuZPbs2XTp0oW3336b4cOH+yRrc1h69UIfEYHnb4WTEKLtUXV6Ms29oULmVQkh/I8t1CTzM4KUX+6jIXxj37x5lH7w4dEbCiE0VXbCBfzsHqh1DCGEaJZO/WOYcElPrWMIDfjl0CnhG44RI7WOIIRogu3hQ7WOIIQQzZYgw6aClhQaQcwxYjjo5EdAiLasYvg0CvJkboYQwn/JRPDgJXeZQUwfFoY1PV3rGEKII9iZdILWEYQQotlCIiyEx9m1jiE0IoVGkHOMkuFTQrRV1f0mkJMtu4ALIfxXWka01hGEhqTQCHL2kVJoCNFW7e5y6tEbCSFEG5bWJ0rrCEJDUmgEOUv37uij5U1AiLamtvsQ9mZpnUIIIZrPGmIkvkOY1jGEhqTQCHKKouAYLpv3CdHW7M04B2TxcSGEH2ufHoWiU7SOITQkhYaQeRpCtDGu1J7syvbL/VSFEKKezM8QUmgI7MOGgUFuaoRoK7IGz0KVOeBCCD9mtOhJ7hqhdQyhMSk0BPqQEGwZGVrHEEIAnvhUduTZtI4hhBDHJaVHJHqj3GYGO/kJEADYZfiUEG1CzqhL8LhlcoYQwr/JsCkBUmiIA0LGjdM6ghBBzxMeQ2aRDDUQQvg3nUEhpWek1jFEGyCFhgDAnJaGuVs3rWMIEdTyx12Bq0YmZwgh/FtSlwhMVpn7KaTQEH8RevJkrSMIEbS8dieZVYlaxxBCiOOWliH7c4k6UmiIes5Jk0CR9a6F0ELh+MuprnBrHUMIIY6LokBqb5mfIepIoSHqGePjsfbrq3UMIYKO12Qm09tR6xhCCHHc4jqEYnOatI4h2ggpNEQDoSefrHUEIYJOyfhLqCiV3gwhhP+T1abEX0mhIRoImTgRjEatYwgRNFSdnu3m3lrHEEIIn5BCQ/yVFBqiAUN4OI6hQ7WOIUTQKBt7PiWFLq1jCCHEcYtMcuCMsmodQ7QhUmiIRpwyfEqIVrMjfIjWEYQQwie6DIrTOoJoY6TQEI2EjBuLYrNpHUOIgFcx4gwK8mRuhhDC/+n0Cl0HS6EhGpJCQzSis9kIGTNG6xhCBLydieO0jiCEED7RPj0Ka4isNiUakkJDHJJTNu8TokVV9Z9ITrbsAi6ECAzdhsZrHUG0QVJoiENyDB+OPjRU6xhCBKw9nU/ROoIQQviEI9xMux6RWscQbZAUGuKQFKOxbqlbIYTP1XYfwt4srVMIIYRvdB0Sj06naB1DtEFSaIjDkuFTQrSMvRnngKp1CiGE8AFFhk2Jw5NCQxyWbcAAjMnJWscQIqC40tLZlW3QOoYQQvhEYudw2TtDHJYUGuKwFEUh/JyztY4hREDJGjQTVeaACyECRPdh0pshDk8KDXFEoVOnopjNWscQIiB44lPZkSd71AghAoPZZiCtT7TWMUQbJoWGOCJDeDjOk07SOoYQASFn1KV43DI5QwgRGDoPiMVg1GsdQ7RhUmiIowqfMV3rCEL4PU9ELJlF4VrHEEIIn+k2LEHrCKKNk0JDHJW1Vy8sPXtqHUMIv5Y/7nJcNTI5QwgRGKKSHUS3C9E6hmjjpNAQTRJ+7rlaRxDCb3kdoWRWyDd/QojA0V16M0QTSKEhmsQ5eZLsFC5EMxWNv5zqSo/WMYQQwif0Rh2dB8ZqHUP4ASk0RJPoLBZCp07VOoYQfkc1Wdjm6aB1DCGE8JkOfaIx24xaxxB+QAoN0WTh554DiqJ1DCH8SvH4i6kodWsdQwghfKbHiEStIwg/IYWGaDJTu3bYhw/XOoYQfkPV6dlu7q11DCGE8JnYVCcJncK0jiH8hBQa4piET5dJ4UI0Vdm4mZQUurSOIYQQPtNnQjutIwg/IoWGOCaOUaMwJkqXqRBNsSNssNYRhBDCZ8JibaT1lp3ARdNJoSGOiaLTEXbO2VrHEKLNqxhxJgV5MjdDCBE4Mk5IRtHJXE3RdFJoiGMWdsYZKGaz1jGEaNN2Jo7TOoIQQviMzWmi6+B4rWMIPyOFhjhmhvBwQqeernUMIdqsqv4TycmWfTOEEIEjfWwSeqPcNopjIz8xolmiLrkEjLKGthCHsrvTKVpHEEIInzFa9PQcKfMzxbGTQkM0izExkdApU7SOIUSbU9NjKPuytE4hhBC+02N4gmzQJ5pFCg3RbFGXXQp6vdYxhGhT9vWWxRKEEIFDp1foPU6WtBXNI4WGaDZT+/Y4TzxR6xhCtBmutHR2ZRu0jiGEED7TeUAsjnBZAEY0jxQa4rhEXn4ZKLLUnRAAWYNnonq1TiGEED6iQJ8JKVqnEH5MCg1xXCydO+MYN1brGEJozhOfyo5cm9YxhBDCZ9r3jCQiwa51DOHHpNAQxy3qiiu1jiCE5rJHXorHrWodQwghfEZ6M8TxkkJDHDdrzx7YR4zQOoYQmvFGxLG9OFzrGEII4TNxaU4SOoVpHUP4OSk0hE9EXXmF1hGE0EzeuMtx1cjkDCFE4JDeDOELUmgIn7D17YttwACtYwjR6ryOUDIr4rWOIYQQPhMebye1d5TWMUQAkEJD+Iz0aohgVDj+cqorPVrHEEIInxl0SiqKrCgpfEAKDeEz9qFDsfburXUMIVqNarKQ6emgdQwhhPCZmJQQOvSJ0TqGCBBSaAifirzicq0jCNFqiidcQkWpW+sYQgjhM4NPly9PhO9IoSF8KmTMGKwZGVrHEKLFqTo9mcZ0rWMIIYTPJHUNJ7lrhNYxRACRQkP4XMz8eVpHEKLFlY2bSWmRS+sYQgjhM4NPld4M4VtSaAifs/XpQ8jEiVrHEKJFbQ8brHUEIYTwmbSMaGJTnVrHEAFGCg3RImJunItiNGodQ4gWUT7iLArzZG6GECIwKAoMOiVN6xgiAEmhIVqEKTmZ8OnTtY4hRIvYlThW6whCCOEzXQbHEZFg1zqGCEBSaIgWE3XVlehCQ7WOIYRPVQ04kZxs2TdDCBEYDCYdg06RuRmiZUihIVqMPjSUqCtkEz8RWHZ3mqJ1BCGE8JmME9rhCDdrHUMEKCk0RIuKmDEdY3Ky1jGE8ImansPZt1/rFEII4RtWp4k+E9ppHUMEMCk0RItSTCZibpijdQwhfGJv+plaRxBCCJ8ZeHIqJotB6xgigEmhIVqc86STsPburXUMIY6LKy2d3dnygSyECAwRCXa6D0/QOoYIcFJoiFYRM3++1hGEOC77B12A6tU6hRBC+MaQ0zug0ylaxxABTgoN0SpsffsQMmGC1jGEaBZ3Qho7cq1axxBCCJ9I6hpO+15RWscQQUAKDdFqYm6cC7KJn/BDOSMvwetRtY4hhBDHTadXGH5WJ61jiCAhhYZoNaZ27YiQTfyEn/FGxLG9KFzrGEII4RMZ49sRmeDQOoYIElJoiFYVde01GGJjtY4hRJPljbscV41MzhBC+D9ntJUBk9trHUMEESk0RKvSOxzELrhV6xhCNInXEcq2initYwghhE+MPrcLBqNe6xgiiEihIVqdc/x4HCeM0zqGEEdVOP5yaio9WscQQojj1mlALMndI7SOIYKMFBpCE3ELFqCz27WOIcRhqSYLmZ4OWscQQojjZrYZGH6mTAAXrU8KDaEJY1wc0ddfr3UMIQ6reMKlVJS6tY4hhBDHbejUjticJq1jiCAkhYbQTPiM6Vh6p2sdQ4hGVJ2eTGMvrWMIIcRxi+8YSrdhMtdMaEMKDaEZRacj/u57wGDQOooQDZSNu4DSIpfWMYQQ4rjoDAqjZ3RFUWQHcKENKTSEpixdOhN54SytYwjRwPawQVpHEEKI49Z3QgoR8TIfUmhHCg2huairr8aYnKx1DCEAKB95FoV5MjdDCOHfQmOs9D+pvdYxRJCTQkNoTmexEHfnHVrHEAKAXfFjtY4ghBDHbfT0LuiNcpsntCU/gaJNcAwbhvOUKVrHEEGuauAkcnJk3wwhhH/rMjiOpK6yZ4bQnhQaos2Ivekm9GFhWscQQWx3x8laRxBCiONisRsZdkZHrWMIAUihIdoQQ0QEMfPmaR1DBKmansPZt1/rFEIIcXyGTuuI1SF7Zoi2QQoN0aaETT0d+6iRWscQQWhv+plaRxBCiOPSrkck3YbKnhmi7ZBCQ7Q5Cffdhz4yUusYIojUdkhnd7bs5yKE8F9Wp4lxF3TTOoYQDUihIdocQ2QkCffdq3UMEUSyBl2A6tU6hRBCNJMCJ1zQDZtThkyJtkUKDdEmOUaNIvy887SOIYKAOyGNHTlWrWMIIUSz9R6TTLseMhJAtD1SaIg2K+YfN2Lu1EnrGCLA5Yy8FK9H1TqGEEI0S2SSgyGnd9A6hhCHJIWGaLN0ZjMJDz+MYjZrHUUEKG9EHNuLwrSOIYQQzWIw6phwcQ/ZmE+0WfKTKdo0S5fOxMydq3UMEaByx12Bq0YmZwgh/NOwMzsREW/XOoYQhyWFhmjzImaej33kCK1jiADjdYSRWRGndQwhhGiW1N5R9ByZqHUMIY5ICg3hF2TJW+FrheMvp6bSo3UMIYQ4ZvYwM2PPl6VsRdsnhYbwC4aoKOLv/T+tY4gAoZosbHOnaR1DCCGOmaLACbO6YXEYtY4ixFFJoSH8Rsjo0YRPn651DBEAiidcRmWZW+sYQghxzDLGtyOpa4TWMYRoEik0hF+JmT8Pc6eOWscQfkzVG8g09tQ6hhBCHLOYlBAGnSq9scJ/SKEh/ErdkrePoJhk91PRPGXjZlJa5NI6hhBCHBOjWc/4i3qg18utm/Af8tMq/I6lS2dib71V6xjCT213DtI6ghBCHLMRZ3ciLNamdQwhjokUGsIvhZ99FqHTpmodQ/iZ8lFnU5gvczOEEP6l04BYug1N0DqGEMdMCg3ht+Juvx1LTxlrL5puV9wYrSMIIcQxiW4Xwtjzu2odQ4hmkUJD+C2d2UzSE/9EHx6udRThB6oGTiInR/bNEEL4D6vTxKQre2Ew6bWOIkSzSKEh/JoxIYHERx4GvbwJiyPb3eFkrSMIIUST6QwKJ13eC0e4ResoQjSbFBrC79mHDiX6+uu0jiHasJpeI9iXpWodQwghmmzUuV2I7xCqdQwhjosUGiIgRF16KSEnnqh1DNFG7e11ptYRhBCiydLHJNF9mEz+Fv5PCg0RMBIW3oe5WzetY4g2xtUhg91ZMrROCOEfkrqGM+zMTlrHEMInpNAQAUNntZL89FPoIyO1jiLakP2DzkeVUVNCCD8QGm1l4qU90ekUraMI4RNSaIiAYkxIIOmJf4LRqHUU0Qa4EzuyI8eqdQwhhDgqo0XPpCvTsdjl80sEDik0RMCx9etH3G0LtI4h2oCcERfj9Uh3hhCibVMUmHBRDyIS7FpHEcKnpNAQASn8rLMInz5d6xhCQ57IeDILw7SOIYQQRzXo1DTap0dpHUMIn5NCQwSs2Ftuxj50qNYxhEbyxl6Ou9ardQwhhDiiTv1j6Hdie61jCNEipNAQAUsxGEh84gks3btrHUW0Mq8jjMyKOK1jCCHEEUW3C2HsTFktUQQuKTREQNM77CS/8DzGdu20jiJaUeGEK6ip9GgdQwghDsvqNDHpyl4YTLL8tghcUmiIgGeIjKTdiy/IsrdBQjVZ2OZK1TqGEEIcltGsZ/JV6TjCLVpHEaJFSaEhgoKpXTuSn3sOnc2mdRTRwoonXEZlmVvrGEIIcUh6g45JV/Yitr1T6yhCtDgpNETQsPbsQeITT8geGwFM1RvINPbUOoYQQhySolOYcEkPkrpGaB1FiFYhhYYIKo7hw0i47966RctFwCkbdwGlRS6tYwghRGMKjD2/K2kZ0VonEaLVSKEhgk7olCnE3Hij1jFEC9juHKh1BCGEOKThZ3Si65B4rWMI0aqk0BBBKfLii4iYNUvrGMKHykedTWG+zM0QQrQ9/Se1p/e4ZK1jCNHqpNAQQStm/jycJ5+sdQzhIzvjxmgdQQghGuk1OolBp6RpHUMITUihIYKWoigkLLxPdg8PAFUDJ5GbI/tmCCHals4DYxlxdietYwihGSk0RFBTjEaSnnwCS48eWkcRx2F3B+mZEkK0Le17RTLugm4osviICGJSaIigp7PX7R5u7tRR6yiiGarTR7AvS9U6hhBC1EvoFMbEy3qi08ttlghu8i9ACMAQEUG7V1/F3Em6uP3Nvp5nah1BCCHqRbcLYfJV6RiMeq2jCKE5RVVV+SpQiAPcRUXsnnUhNZs3ax1FNIGrQwbftrsUeRcT4tC+2fAB32z8gMKyHADiwlM4qd/59Gg3CI/HzYc/v8SGPT9RUJqFxWSna2JfThl0CWH2qMOe87tNy/hpyyfsL9wJQLvozkwZeDHtY7rWt/l562e8/+OL1LqrGdLlJE4fcnn9cwVl2Ty1bB7zpi7CarK3zAvXSFisjak39sUaYtI6ihBtghQaQvyNu6iI3RdeRM0ff2gdRRzFrumPkLnfonUMIdqs33auQqfTE+VMAODHLZ/w+br/cNO05wizR/Hip3cxrNtkEiM7UFlTxturnsHj9TB/2qLDnvOVz+8jLa4HabE9MOhNfLbu36zb8Q23nrWYMHs05VUl3Lb0HM4bPY8oZzyL/ncr543+Bz1TBgPwzMc3MbTrZDLSRrTKn0FrcUSYmXpjP0Ii5D1JiD/J0Ckh/sYQHk7KKy9j7t5N6yjiCNyJHdmRY9U6hhBtWq/2Q+nRbhCxYcnEhiVzysCLMRut7MjdiNXs4NqTH6Jvh9HEhiWTGtudM4ddw578LfU9IIcya9wtjOxxKklRHYkLb8f0kTegqiqb960BIL+srnekX8cxpMR0pXNCBtlFuwD4eevn6HXGgCsyQiItnDanjxQZQvyNFBpCHII+LIyUl17C0r271lHEYeSMuBivRzpkhWgqr9fDL9u+oNZVTWrsod/bqmorUFCwmh1NPm+tuwaP143NHAJATGgiLncNe/K3UlFdyq68zSREplFRXcqyX17hrOHX+uT1tBV/DpcKjbZpHUWINsegdQAh2ip9WBjtXn6J3RddTPWGDVrHEX/hiYwnszAM8GodRYg2b1/Bdh5571rcnlrMRiuXTryL+PD2jdq53LW8/+OL9O849pjmTrz/4wuE2qPomtgPAJs5hPPHzGfJlw/gctcwsPN4uicP4PWVDzGq52kUlGbz3PLb8HjdTOo/kz5po3z1UltdZJKDU2ZnYHPKnAwhDkXmaAhxFJ7S0rpi4/fftY4iDsg+60425kZrHUMIv+D2uCgqz6Wytpy127/h+z8+5rpTHm1QbHg8bhZ/dheF5blcN+XRJhcan679F5+u/RfXTXmExMgOh223Zf9a3vvhea6f8ih3/msmF467FactgofevZo7znmVEGv48b7MVheb6uTka3pjsRu1jiJEmyVDp4Q4Cr3TSbuXX8KSnq51FAF4HWFklsVpHUMIv2HQG4kOTSQlugunDrqExMgOrPztnfrn64qMuykozeaayQ82ucj4bN1/+GTNG1w9+YEjFhkuTy3/+eafnDvievJK9+H1euiU0JvYsGRiQpPYmbPpuF9ja0vsEsYp12VIkSHEUUihIUQT6ENCaPfSYqy9e2sdJegVTLiCmiqP1jGE8FuqquL2uICDRUZeyT6uOfkhHJbQJp3js7X/Zvnq17lq0v2kRHc5Ytvlv75O93YDSY7ujFf14lUP/vv1eN14Vf8aApnSK5KTr+mNySKjz4U4Gik0hGgivcNB8uIXsWZkaB0laKlmK5muVK1jCOE3PvjxRbZlraegLJt9Bdv54KfFbM1aR/9O4/B4Pbz46V3sztvCBeNuQVW9lFYWUlpZWF+IACz54n7e//HF+t9/uvZffPTzy8wYdSORIXH1x9S4qhpdP6twJ6szVzK5/ywAYsPaoSgKq/74mN93/UBO8W5SYo5cqLQlHfvFcNIVvWQzPiGaSMpxIY6B3uEg+cUX2XvNNVT+8IPWcYJO/6Zp1gAAFU5JREFU8YTLqCxzax1DCL9RVlXEki/up7SyEIvJTmJkGldNWki3pP4UlGXz265VANz/1mUNjps95RE6J2QAUFiei6Io9c99s+ED3F4Xiz+9q8ExJ/WbyeT+F9T/XlVV3vz6UaYOvRKzsW4papPBzHmj5/Gfb5/A7XFx1rBrCbP7x3yrrkPjGXNeV3Q65eiNhRCATAYXolnU2lr233wLpcuWaR0laKh6A7+e8gylRa6jNxZCCB/qNSaJEWd1alBwCSGOToZOCdEMislEwsMPEXnJxVpHCRplJ8ySIkMI0er6nZjCyLM7S5EhRDNIoSFEMymKQsyNNxK7YAHo5J9SS9seMkDrCEKIIDP4tDQGn3b4FbWEEEcmd0dCHKeI82aQ+PhjKGaz1lECVsWocyjMl7kZQohWosDIczrT78T2WicRwq9JoSGEDzgnTKDdyy+jD23a0pDi2OyIG611BCFEkNAbdYy/qDu9RidpHUUIvyeFhhA+Yuvbh5Q338SYmKh1lIBSNXAyuTmyb4YQouXZnCZOv6EvnQfIpqBC+IIUGkL4kDktlfb/ehNL9+5aRwkYuztM1jqCECIIRCU7OPPm/sSmOrWOIkTAkEJDCB8zREeT8toS7CNGaB3F71Wnj2BflqzALYRoWWl9opn6j344wi1aRxEioEihIUQL0NntJC96htCpU7WO4tf29TxT6whCiADX78QUTrysJ0aT7PYthK/JzuBCtBDFYCDhvnsxxsWR/8wzWsfxO66OfdidJR/8QoiWoTfqGHNeV7oMkvkYQrQU6dEQooVFz76WxMcfR2ezaR3Fr+wfeD6qjJoSQrQAq9PEaXP6SJEhRAuTQkOIVuA8cSLt//NvTO3bax3FL7iTOrEjR8ZKCyF8LyrZwZk39ScuTZYjF6KlSaEhRCsxd+xI+//+B8fYsVpHafNyRlyM1yPdGUII30rtHcXUG/sREiFfZAjRGqTQEKIV6UNCSHr6KaJmXws6+ed3KJ7IeDIL5JtGIYRv9T0xhZOu6IXRLHO/hGgtiqrKKGghtFD+9dfs+8c8vCUlWkdpU7LPupONudFaxxBCBAi9QceY82XStxBakK9UhdCIY+RIUv/7H8xdumgdpc3whkSQWSY3A0II37CHmTntBpn0LYRWpNAQQkOmdu1o/683cZ58stZR2oSC8ZdTU+XROoYQIgC07xXJ2QsGyKRvITQk+2gIoTGd1Uriww9h7dWTnIceBrdb60iaUM1WMl3tgeB8/UII39AZFIae3pHe45K1jiJE0JM5GkK0IZU//8zeOTfgyc/XOkqrK5xyPWvLOmkdQwjhx0JjrEy8pCfR7UK0jiKEQIZOCdGm2AYMIPXtt7D27at1lFal6g1k6rtrHUMI4ce6DIrjrFsGSJEhRBsiPRpCtEGqx0PBCy+Q9/Qz4HJpHafFlUy8lF9rMrSOIYTwQ0aznlHndqbL4Hitowgh/kYKDSHasKrffmf/vHnU7tihdZQWoyoKa6c9R1F+4BdUQgjfikp2MPGSnoTF2rSOIoQ4BCk0hGjjvFVV5Dz4IMVv/kvrKC2ifPS5/MRwrWMIIfxM+pgkhk7riN4go8CFaKuk0BDCT5R/9RX7b10QcBPFfz/7WXJzZElbIUTTWOxGxl7QjdT0KK2jCCGOQgoNIfyIu7CQrAW3Uf7FF1pH8YmqgZP53jZJ6xhCCD+R0CmM8Rf1wBFu1jqKEKIJpNAQwg8V/ec/5Nz/AGplpdZRjssf5z7D/ix5CxJCHJmiUxgwuT39T2qPolO0jiOEaCIpNITwU7W7drFv3jyq163XOkqz1KSP5LuIs7WOIYRo48Lj7Yw9v6vs8C2EH5JCQwg/prrd5C96lvznnvO7HcW3TX+S3ftlEqcQ4tB0OoU+E9sxYFIqeqO8Vwjhj6TQECIAVK1bx/6bbvabZXBrO/Xlu6SLkXcfIcShRCU7GDuzG9HJsvmeEP5MCg0hAoRaW0v+Cy9Q8NzzqLW1Wsc5ol3THyVzv0zmFEI0pDfo6D+5PX0ntEOnl14MIfydFBpCBJjanTvJuusuKr//Qesoh+RO7sK3na/D65G3HiHEQbGpTsbO7EZEvF3rKEIIH5FCQ4gAVfLhh+Tc/wCeggKtozSw99z72ZIlwyGEEHVMFj2DT+tAz5GJsqKUEAFGCg0hApinpITcRx6l+L//pS1MiPBEJfLd/7d3p8FRlXsex3+nO71k65BOmpgmCwlLEtCQBNwIV1EjQTEld6i6ztURsVxeTBWrlghiWcIgBeVEqwbh4gKUt7gvrlpqlVQJOGX0IjpjBoIoi6BRlixk3zrpTieZF5HUMAFFOEln+X6qUt3pc85z/icvTuXXz/OcJ2+NgoHuUJcCYAiYkOfRHx6YrMgYhlICIxFBAxgFfAcPqeqFF+Q/eTKkdVT96UUdPc9qvsBoFx3n1G3/PFnjb+B+AIxkBA1glOgJBlW/c6dqXtuinvb2QT9/d7RbX856Sf72rkE/N4ChwWIxNO2uZN1YlCab3RrqcgAMMIIGMMoEzp5T9bp1av3ss0E9b80/rdKR+qRBPSeAoSNxQoxu+3OG4pOiQl0KgEFC0ABGqeY9e1W9fr2C588P+Ll6HOH6rzmvytcyvBYVBHDtYsaG69Y/TtCE3LGhLgXAICNoAKNYd1ub6nbsVP327er2+QbsPPVFy1TWMmnA2gcw9DgjbZoxb7yuv32crKyJAYxKBA0ACtbWqmbzZjW++54UNLfXoccaptKiLWpp7DS1XQBDk9VmUfYdSZp+z3g5wsNCXQ6AECJoAOjj/7FcNa8Uq2XfJ6a12VT4hP7Hn2NaewCGKEOaNCNBt8xPlysuPNTVABgCCBoA+vEdPKTzL7+s9oMHr6mdHsNQ2YJtaqilNwMYycZNHqOZCyZqbKor1KUAGEIIGgAuq+WTT3T+34sVKC+/quNbZz+o/1a+yVUBGCpir4vQrX+coLRpnlCXAmAIImgA+FU9waAa331PNa9tVldN7e869sgDf1FNNetmACNNeLRNN92XpimzvLIw0RvAZRA0AFyRbp9Pddt3XPETqnw3F+mr8LmDUBmAwRJms2haQbLyClNldzLRG8CvI2gA+F2CtbWq3foXNb7zjnoCgcvud/zPW1RRye0FGAnC7BZNmeVV7t2piop1hLocAMMEQQPAVek8f171b21Xw9//rp729ou2+bNv0xfuB0JUGQCzOCLCdMPsJGXfmaTwKHuoywEwzBA0AFyTYEOD6nfsVMPf/qbu1lZJ0qkH/0OnKxi3DQxXETF25dyVoqm3eRkiBeCqETQAmKKruVn1f/2rqr84os9j/iTuLMDw44p3KndOqrJuTZTVxpcFAK4NQQOAqQIdQR3dX6HD/3lGrQ3+UJcD4ArEjYtU3txUTZyeIIvFCHU5AEYIvq4AYCq7M0w5BSn6l3+7VQWLshQ3LjLUJQG4jOvSYzTvX7P1wJqbNPnG60ZEyKiqqtLixYuVnp4uh8Oh5ORkFRUV6cUXX5TNZtP+/fsv2r+trU3p6elavny5JGn27NkyDEOGYcjhcGjy5Ml66aWX1NXV+6jukpIS3X///UpMTFRkZKRycnK0a9eufnUEAgFt2rRJ06ZNU0REhOLj45Wfn68dO3aos7N3EdPPP/9cRUVF8nq9MgxDH3zwQb92qqurtWjRInm9XkVERGju3Lk6efKkyX81YGAw8BLAgLBaLcq4JVEZtyTq52/rdGjvzzr3fWOoywIgKWWqW9Pnpso7KTbUpZjqp59+Un5+vsaMGaNNmzYpOztbnZ2d2rNnj15//XUtXrxYixYt0uHDhxUZ2fslyDPPPCOHw6ENGzb0tfPEE09o7dq16ujo0EcffaQlS5bIarVq5cqVOnDggLKzs7Vy5UolJCRo9+7dWrhwoVwul4qKiiT1hozCwkIdPnxY69atU35+vlwul7766iu9/PLLys3NVU5Ojtra2jRt2jQ9+uijWrBgQb/r6enp0fz582Wz2fThhx/K5XKpuLhYBQUFOnr0aN81AEMVQ6cADJrqn5p1aO9plZfVqLubWw8wmCwWQ+m5HuUVpsqTEh3qcgbEvffeq2+++UYnTpzo9094Y2OjnE6n8vLydOedd2rz5s369NNPVVhYqAMHDmjGjBmSens0cnJy9Oqrr/Yde/fdd6u1tVVffvnlJc87b948JSQkaPv27ZKkTZs2adWqVSotLVVubu5F+3Z2dioQCPSrzzAMvf/++5o/f37fZ99//70yMjL07bffaurUqZKkrq4ujR07Vhs3btTjjz9+VX8nYLDQowFg0CSMd2nuk9errcmvYwcqdeyLCjXXdoS6LGBEc8U7lZXvVdbMREXGjNw1MOrr6/Xxxx9r/fr1l/ymf8yYMZKkt99+WzNnzlRBQYGWL1+u1atX94WMywkPD1dDQ8Nltzc1NSkrK6vv9127dqmgoKBfyJAkm80mm812Rdfk9/fOc3M6nX2fWa1W2e127d+/n6CBIY+gAWDQRcY4NOOe8Zo+N1VnjtXr6D8qVP5Nrbq76OUAzGCxGkrLjteUP3iVnOWWYQz/uRe/5dSpU+rp6VFmZuav7jdjxgytWrVKCxYsUG5urtasWXPZfbu7u7V3717t2bNHy5Ytu+Q+7777rr7++mtt27at77OTJ09q9uzZV3MZF8nMzFRqaqpWrVqlbdu2KTIyUsXFxaqqqlJlZeU1tw8MNIIGgJAxDEMpU+KUMiVOvuaAjh2o0NEvKtVc0/7bBwPox+UJ15T8RGXN9CrCNboW2LswEvxKQtWaNWu0du1aPfvsswoL6/+v0JYtW/Tmm28qEAhIkh5++GG98MIL/fYrKSnRokWL9MYbb/QNbbpQixnhzmaz6b333tNjjz0mt9stq9WqgoIC3XPPPdfcNjAYCBoAhoQIl13T545XXmGqzh5v0Hf/qFD54Rp6OYDfYLEaSpvm0dQ/eJWUGTsqei8uZdKkSTIMQ8eOHbtonsOlXBi6dKmQIUkPPfSQnnvuOTkcDnm9Xlmt1n77fPbZZyoqKlJxcbEWLlx40bbJkyfr2LFjV3ch/8/06dNVVlampqYmBQIBeTwe3Xzzzb853AsYCggaAIYUwzCUnOVWcpZbvuaAjn9ZqaP7K9RELwdwkZix4ZqS71XmrYmjrvfiUtxutwoLC/Xaa69pyZIll5wMfmGexm+JiYnRxIkTL7u9pKRE9913nzZu3Kgnn3yy3/YHH3xQq1ev1qFDh/rN0wgGg/L7/b/7iVExMTGSeodllZaWat26db/reCAUCBoAhqwIl115hanKnZOicyca9N3+Cv1YVqPuIL0cGJ0sYYbSczyaOsurcRmjt/ficrZs2aKZM2fqpptu0tq1a5Wdna1gMKh9+/Zp69atpvQylJSUaN68eVq6dKkWLFigqqoqSZLdbpfb7ZYkLVu2TLt379Zdd92ldevWadasWYqOjlZpaak2btyot956Szk5OWptbdWpU6f62i4vL1dZWZncbrdSUlIkSe+88448Ho9SUlJ05MgRLV26VPPnz9ecOXOu+VqAgUbQADDkGYahpEy3kjLdam8N6OTX1frhYI0qTzWKB3RjpDMMKXHiGE3I82jSjQkKj6L34nLS0tJ08OBBrV+/Xk899ZQqKyvl8Xg0ffp0bd261ZRz7Ny5Uz6fTxs2bLho7Y3bb79dJSUlkiSHw6F9+/bplVde0bZt2/T0008rIiJCWVlZWrJkia6//npJUmlpqe64446+NlasWCFJeuSRR7Rz505JUmVlpVasWKHq6molJiZq4cKFev755025FmCgsY4GgGHL1xzQj2U1+uHgeZ37vlE9rM2BEcJiMeSdPEYT8sYqPcfD0CgAwxJBA8CI0N4aUPnhWv1w8LzOHm9gEjmGHYu1t+duQp5H6dM8ckZd2VoLADBUETQAjDh+X6fKv6nVDwdrdOZovbqC3aEuCbgka5hFyVN6w0VadrwcEYQLACMHQQPAiBboCOqnI72h4/R3dQoGCB0IrTC7RalT45Se59H4G+JldzJdEsDIRNAAMGp0Brr085E6/XjovE4frZffFwx1SRglHBFhSpniVnruWKXeECebvf+6DAAw0hA0AIxKPd09qjnTorPHG3TuRIMqfmhS0N8V6rIwQoQ5rPJOjNG4jFglZcTKkxwtw8KjaAGMLgQNAJDU1dWt6vJmnTvRoLPHG1RV3sR6Hbhi1jCLEtJcSsqM1biMWCWkuWS1WkJdFgCEFEEDAC4hGOhS5akmnT3RoLMnGlRzuoXH56KPYTHkSYlWUkaskjJjlTghRmEMhwKAixA0AOAKBNqDOneyUeeO9waPuopWibvn6GFIcd4oJWXEalxmrMZNGiN7OJO4AeDXEDQA4Cq0twZU9WOzak63qPZMi2rOtKi13h/qsmCSKLdDnuRoeVKiFZ8crevSXAqPZtE8APg9CBoAYJKO1k7VnG1R7elW1ZzpDSCN1T5xlx26DEMakxCh+ORoeZKjFZ8SJU9ytJyRrGcBANeKoAEAA6jT36W6c63/p+ejVfUVbSwiGAKWMENx3ijFJ0f19VbEjYuSzcHcCgAYCAQNABhkXV3daqhsU83pVtWda1VzbbuaazvUUteuQAeP2L1WdqdV0XHhcsU75YoLV1xSpOKTo+X2RvIkKAAYRAQNABhCOto61VLXcVH4aK7rUHNd73tWNu9dWTs6LlyuOKdccc6+UBEd55QrPpxhTwAwRBA0AGAY8TUH1FzXrpbaDjX/EkJa6jrkawrI396pgC+ogL9r2D4RyxpmkTMyTM4omyJcdkXHXwgU4Yr+pYciwsWkbAAYDggaADDCdHf3KNAelN8XlN/XKX97UAHfhd+D8rd39r3v3a+z770ZS4VYDMnmDJPdae17tTvDZPvl1R7e++qMsskZ+cvPL++ZLwEAIwdBAwAAAIDpmBUHAAAAwHQEDQAAAACmI2gAAAAAMB1BAwAAAIDpCBoAAAAATEfQAAAAAGA6ggYAAAAA0xE0AAAAAJiOoAEAAADAdAQNAAAAAKYjaAAAAAAwHUEDAAAAgOkIGgAAAABMR9AAAAAAYDqCBgAAAADTETQAAAAAmI6gAQAAAMB0BA0AAAAApiNoAAAAADAdQQMAAACA6QgaAAAAAExH0AAAAABgOoIGAAAAANMRNAAAAACYjqABAAAAwHQEDQAAAACmI2gAAAAAMB1BAwAAAIDpCBoAAAAATEfQAAAAAGA6ggYAAAAA0xE0AAAAAJiOoAEAAADAdAQNAAAAAKYjaAAAAAAwHUEDAAAAgOkIGgAAAABM978n+Pl2mk59YAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 1000x800 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Create a sample DataFrame\n",
    "data = {'Haplotype_Gene': ['ABCG2', 'CACNA1S', 'CFTR', 'CYP2B6', 'CYP2C19', 'CYP2C9', 'CYP2D6', 'CYP3A5', 'CYP4F2', 'DPYD', 'G6PD', 'IFNL3', 'NUDT15', 'RYR1', 'SLCO1B1', 'TPMT', 'UGT1A1'],\n",
    "        'Count': [297, 1, 30, 31935, 35731, 1815, 4655, 20, 5100, 1015, 1339, 622, 514, 3, 24001, 213, 3698]}\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Create a pie plot with labels outside\n",
    "plt.figure(figsize=(10, 8))\n",
    "plt.pie(df['Count'], labels=df['Haplotype_Gene'], autopct='%1.1f%%', startangle=140,\n",
    "        pctdistance=0.85, labeldistance=1.1)  # Adjust labeldistance and pctdistance\n",
    "plt.axis('equal')  # Equal aspect ratio ensures that pie is drawn as a circle.\n",
    "plt.title('Haplotype Gene Distribution')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "f0f800e4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pandas in c:\\users\\genepowerx_madhu\\anaconda3\\lib\\site-packages (1.5.3)\n",
      "Collecting xlsxwriter\n",
      "  Downloading XlsxWriter-3.1.2-py3-none-any.whl (153 kB)\n",
      "     -------------------------------------- 153.0/153.0 kB 1.8 MB/s eta 0:00:00\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in c:\\users\\genepowerx_madhu\\anaconda3\\lib\\site-packages (from pandas) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\users\\genepowerx_madhu\\anaconda3\\lib\\site-packages (from pandas) (2022.7)\n",
      "Requirement already satisfied: numpy>=1.21.0 in c:\\users\\genepowerx_madhu\\anaconda3\\lib\\site-packages (from pandas) (1.23.5)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\genepowerx_madhu\\anaconda3\\lib\\site-packages (from python-dateutil>=2.8.1->pandas) (1.16.0)\n",
      "Installing collected packages: xlsxwriter\n",
      "Successfully installed xlsxwriter-3.1.2\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install pandas xlsxwriter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "5add8aac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA90AAAJOCAYAAACqS2TfAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAAB7sUlEQVR4nOzdd1yV9f//8ecBERkCLgQMBULc2zK0civuVVouXFmOyrRl5cqVluNTuRquylGmDTP3SMttpiblwlGK5kBcoMD1+6Mv5+cJUDhyvA76uN9u53bjvN/vc13Pc0Dqxft9vS+LYRiGAAAAAABAjnMxOwAAAAAAAPcqim4AAAAAAByEohsAAAAAAAeh6AYAAAAAwEEougEAAAAAcBCKbgAAAAAAHISiGwAAAAAAB6HoBgAAAADAQSi6AQAAAABwEIpuADDZ1q1b1aZNGxUvXlzu7u4qWrSoIiMjNWjQIJtxU6dO1ezZsx2eJyQkRM2bN3foOYYPHy6LxXLbcd26dZPFYrE+XF1d9cADD6h9+/bat2+fQzP+V1JSkqZMmaLatWurUKFCcnNzU6FChVSnTh3NmDFDly5duqt5bjZ79mxZLBbt2LHD4edauHChypUrJw8PD1ksFu3evTvDcevXr5fFYtGiRYsy7O/fv3+WfgbuVEhIiLp165bt1129elXDhw/X+vXrczyTo+zZs0fdu3dXaGio8uXLJ29vb1WtWlXjx4/X+fPnzY4nSZo3b54mT55sdgwAuKvymB0AAO5nP/zwg1q2bKk6depo/PjxCgwM1KlTp7Rjxw4tWLBAEyZMsI6dOnWqChcubFcBkZt5eHho7dq1kqTk5GQdOnRIo0aNUs2aNRUTE6NixYo5PMM///yjqKgo7du3T9HR0XrhhRfk7++vc+fOae3atXr11Ve1adMmffbZZw7PYqZ//vlHXbp0UVRUlKZOnSp3d3dFRESYHcshrl69qhEjRkiS6tSpY26YLPj444/Vt29flSpVSq+88orKli2rGzduaMeOHZo+fbo2b96sJUuWmB1T8+bN0759+zRgwACzowDAXUPRDQAmGj9+vEJDQ7VixQrlyfP/fyU/9dRTGj9+vInJnIeLi4seeeQR6/NHH31UxYsXV/369fXDDz+od+/ed3yOlJQUJScny93dPcP+zp07a+/evVq9erUef/xxm77WrVtr2LBh+vHHH+/oHLnBgQMHdOPGDXXu3Fm1a9c2Ow7+z+bNm9WnTx81bNhQ33zzjc3PWMOGDTVo0CAtX77cxIQAcH9jeTkAmOjcuXMqXLiwTcGdxsXl//+KDgkJ0e+//64NGzZYl1qHhIRIkhITEzVo0CBVrlxZvr6+KliwoCIjI/Xtt9+mO2Zqaqo++OADVa5cWR4eHvLz89Mjjzyi77777pY5p06dqjx58mjYsGHWttWrV6t+/fry8fGRp6enatWqpTVr1qR77Q8//KDKlSvL3d1doaGheu+997L68WTK19dXkuTm5mZt++eff9S3b1+VLVtW3t7e8vf3V7169bRx40ab1x49elQWi0Xjx4/XqFGjFBoaKnd3d61bty7Dc23fvl0rV65U79690xXcaQoVKqTOnTtn6RzZ+X5ZLBb1799fM2bMUEREhNzd3VW2bFktWLAgwxyXLl1Snz59VLhwYRUqVEht27bVyZMnb/1h/p/vvvtOkZGR8vT0VP78+dWwYUNt3rzZ2t+tWzc9+uijkqQOHTrIYrHk+AzwwoUL1ahRIwUGBsrDw0NlypTR66+/ritXrtiM69atm7y9vfX777+rfv368vLyUpEiRdS/f39dvXr1tuc5fvy4OnfuLH9/f7m7u6tMmTKaMGGCUlNTJf37/StSpIgkacSIEdZ/c926ddPGjRtlsVg0f/78dMedO3euLBaLtm/fnu2chmFo6tSp1n+bBQoU0BNPPKEjR47c9v2MGTNGFotFH330UYZ/1MmbN69atmxpfZ6amqrx48erdOnScnd3l7+/v7p27aq//vrL5nWZLc2vU6eOzfc+7VKC+fPn680331RQUJB8fHzUoEED/fnnnzav++GHH3Ts2DGby0bSTJs2TZUqVZK3t7fy58+v0qVL64033rjt+wcAp2cAAEzTq1cvQ5Lx/PPPG1u2bDGuX7+e4bhdu3YZYWFhRpUqVYzNmzcbmzdvNnbt2mUYhmHEx8cb3bp1Mz777DNj7dq1xvLly42XX37ZcHFxMebMmWNznC5duhgWi8Xo1auX8e233xo//vijMXr0aON///ufdUyJEiWMZs2aGYZhGKmpqcagQYMMNzc3Y9asWdYxn332mWGxWIzWrVsbixcvNr7//nujefPmhqurq7F69WrruNWrVxuurq7Go48+aixevNj46quvjIceesgoXry4kZX/BEVHRxteXl7GjRs3jBs3bhjXrl0z9u7da9StW9coUKCAcfr0aevYP/74w+jTp4+xYMECY/369cbSpUuNnj17Gi4uLsa6deus42JjYw1JRrFixYy6desaixYtMlauXGnExsZmmGH06NGGJGPFihW3zZuVc2Tn+yXJCA4ONsqWLWvMnz/f+O6774yoqChDkvHVV19Zx82aNcuQZISFhRnPP/+8sWLFCuOTTz4xChQoYNStW/e2eb/44gtDktGoUSPjm2++MRYuXGhUq1bNyJs3r7Fx40bDMAzj0KFDxpQpUwxJxpgxY4zNmzcbv//+e6bHXLdunSHJWLhwofX7d/Ojb9++6X4GRo4caUyaNMn44YcfjPXr1xvTp083QkND072H6OhoI2/evEbx4sWN0aNHGytXrjSGDx9u5MmTx2jevLnN2BIlShjR0dHW52fOnDGKFStmFClSxJg+fbqxfPlyo3///oYko0+fPoZhGEZiYqKxfPlyQ5LRs2dP67+5Q4cOGYZhGFWqVDFq1aqV7j0/9NBDxkMPPWRXzmeeecZwc3MzBg0aZCxfvtyYN2+eUbp0aaNo0aJGXFxcpp9zcnKy4enpadSoUSPTMf/Vu3dvQ5LRv39/Y/ny5cb06dONIkWKGMHBwcY///yT6WeXpnbt2kbt2rWtz9O+1yEhIUanTp2MH374wZg/f75RvHhxo2TJkkZycrJhGIbx+++/G7Vq1TICAgKsn+nmzZsNwzCM+fPnW38Xrly50li9erUxffp044UXXsjy+wIAZ0XRDQAmOnv2rPHoo48akgxJhpubm1GzZk1j7NixxqVLl2zGlitXzuZ/dDOTnJxs3Lhxw+jZs6dRpUoVa/tPP/1kSDLefPPNW74+rei+evWq0a5dO8PX19emkL5y5YpRsGBBo0WLFjavS0lJMSpVqmQ8/PDD1rYaNWoYQUFBxrVr16xtCQkJRsGCBbNcdKd9Njc/AgMDjU2bNmXpc6hfv77Rpk0ba3taQfzggw9m+keOmz333HOGJOOPP/6waU9NTbUpItMKi+yeI7Pvl2H8W3R7eHjYFF3JyclG6dKljfDwcGtbWtHdt29fm9ePHz/ekGScOnUq0/OnpKQYQUFBRoUKFYyUlBRr+6VLlwx/f3+jZs2a1ra04urmgj8zaWNv98hM2ue7YcMGQ5Lx22+/WfvSfi5u/mORYfz/P5Dc/LPx38Lx9ddfNyQZW7dutXltnz59DIvFYvz555+GYRjGP//8Y0gyhg0bli5b2uf966+/Wtu2bdtmSLL5w0lWc27evNmQZEyYMMFm3IkTJwwPDw/j1VdfzfRziouLMyQZTz31VKZjbhYTE5Phz8rWrVsNScYbb7xhbctu0d20aVObcV9++aUhyVpYG4ZhNGvWzChRokS6Y/bv39/w8/PL0nsAgNyG5eUAYKJChQpp48aN2r59u9555x21atVKBw4c0ODBg1WhQgWdPXs2S8f56quvVKtWLXl7eytPnjxyc3PTp59+qpiYGOuYtGuO+/Xrd9vjnTt3TvXq1dO2bdu0adMm1a9f39r3yy+/6Pz584qOjlZycrL1kZqaqqioKG3fvl1XrlzRlStXtH37drVt21b58uWzvj5//vxq0aJFVj8ieXh4aPv27dq+fbu2bt2qxYsXKyIiQk2bNrVZ/ixJ06dPV9WqVZUvXz7r57BmzRqbzyFNy5YtbZanZ9e3334rNzc36yNtyXtWzpGV71ea+vXrq2jRotbnrq6u6tChgw4dOpRuOfDNS4glqWLFipKkY8eOZfo+/vzzT508eVJdunSxuaTB29tb7dq105YtW7K0ZDsz48aNs37/bn60b98+3dgjR46oY8eOCggIkKurq9zc3KzXjmf02XTq1MnmeceOHSUp00sFJGnt2rUqW7asHn74YZv2bt26yTAM66Z9t/L000/L399fU6ZMsbZ98MEHKlKkiDp06JDtnEuXLpXFYlHnzp1t/k0FBASoUqVKObqDeto5/7ts/OGHH1aZMmUyvEQkq+z5+bv5/PHx8Xr66af17bffZvl3HwDkBhTdAOAEqlevrtdee01fffWVTp48qZdeeklHjx7N0mZqixcvVvv27VWsWDF9/vnn2rx5s7Zv364ePXooMTHROu6ff/6Rq6urAgICbnvMAwcOaOvWrWrSpInKly9v03f69GlJ0hNPPGFTdLq5uWncuHEyDEPnz5/XhQsXlJqamuH5spIhjYuLi6pXr67q1avr4YcfVps2bbRs2TLlyZNHAwcOtI6bOHGi+vTpoxo1aujrr7/Wli1btH37dkVFRenatWvpjhsYGJil8xcvXlxS+sKhTp061gIys1usZXSOrH6/0tzq8zt37pxNe6FChWyep13fm9H7T5N2jIyyBgUFKTU1VRcuXMj09bcTFhZm/f7d/Ei7ZjrN5cuX9dhjj2nr1q0aNWqU1q9fr+3bt2vx4sUZvoc8efKke7+ZfS43O3fuXKbv9XavTePu7q5nn31W8+bNU3x8vP755x99+eWX6tWrV7prqrOS8/Tp0zIMQ0WLFk33b2rLli23LEALFy4sT09PxcbG3jb3zefM7DPIyvvPjD0/f2m6dOmimTNn6tixY2rXrp38/f1Vo0YNrVq1yu48AOAs2L0cAJyMm5ubhg0bpkmTJmXpXtSff/65QkNDtXDhQptNiZKSkmzGFSlSRCkpKYqLi7ttwRkZGaknn3xSPXv2lPTvBkdps6CFCxeW9O/M3s27it+saNGiunHjhiwWi+Li4tL1Z9SWHZ6ennrwwQf122+/Wds+//xz1alTR9OmTbMZm9n9s7N6j+iGDRvqjTfe0HfffadGjRpZ2/38/FS9enVJ6YuNW50jq9+vNLf6/DI7b3akHePUqVPp+k6ePCkXFxcVKFDgjs9zO2vXrtXJkye1fv16m53R4+PjMxyfnJysc+fO2XwGWflcChUqlOl7lf7/z/ft9OnTR++8845mzpypxMREJScn67nnnrMrZ+HChWWxWLRx48YMN0K71Y73rq6uql+/vn788Uf99ddfeuCBB26Z++bv93/Hnjx50ub958uXL8Ofy7Nnz2b5c8qO7t27q3v37rpy5Yp++uknDRs2TM2bN9eBAwdUokSJHD8fANwtzHQDgIky+p9/6f8vpU2bfZP+/R/vjGaMLBaL8ubNa1PAxcXFpdsNu0mTJpKUrijNTHR0tBYsWKBZs2apa9euSklJkSTVqlVLfn5+2r9/f4YzmNWrV1fevHnl5eWlhx9+WIsXL7aZwb106ZK+//77LGXIzOXLl3Xo0CH5+/tb2ywWS7riZM+ePemWoGdX9erV1ahRI3388cfpdkK3R1a/X2nWrFljXV0g/XvrsYULF+rBBx+8bYGVFaVKlVKxYsU0b948GYZhbb9y5Yq+/vpr647mjpb2efz3ezhjxoxMX/PFF1/YPJ83b56kW99Xu379+tq/f7927dpl056283jdunVtcmQ2SxsYGKgnn3xSU6dO1fTp09WiRQvrqojs5mzevLkMw9Dff/+d4b+nChUqZPp+JGnw4MEyDEPPPPOMrl+/nq7/xo0b1n9z9erVk/TvH39utn37dsXExNhcShISEqI9e/bYjDtw4IDNjuTZldnvsZt5eXmpSZMmevPNN3X9+nX9/vvvdp8PAJwBM90AYKLGjRvrgQceUIsWLVS6dGmlpqZq9+7dmjBhgry9vfXiiy9ax1aoUEELFizQwoULFRYWpnz58qlChQpq3ry5Fi9erL59++qJJ57QiRMnNHLkSAUGBurgwYPW1z/22GPq0qWLRo0apdOnT6t58+Zyd3fXr7/+Kk9PTz3//PPp8j3xxBPy9PTUE088oWvXrmn+/Pny9vbWBx98oOjoaJ0/f15PPPGE/P399c8//+i3337TP//8Yy3sR44cqaioKOu9glNSUjRu3Dh5eXnp/PnzWfqMUlNTtWXLFuvXf//9t95//31duHBBw4cPt45r3ry5Ro4cqWHDhql27dr6888/9fbbbys0NFTJycn2fHusPv/8czVu3FgNGjRQt27d1LhxY/n7+yshIUF79uzR6tWr5ePjk6VjZfX7laZw4cKqV6+ehgwZIi8vL02dOlV//PFHprcNyy4XFxeNHz9enTp1UvPmzfXss88qKSlJ7777ruLj4/XOO+/kyHlup2bNmipQoICee+45DRs2TG5ubvriiy9sVjPcLG/evJowYYIuX76shx56SL/88otGjRqlJk2aWG9tlpGXXnpJc+fOVbNmzfT222+rRIkS+uGHHzR16lT16dNHERERkv7de6BEiRL69ttvVb9+fRUsWFCFCxe23qpPkl588UXVqFFDkjRr1iy7c9aqVUu9e/dW9+7dtWPHDj3++OPy8vLSqVOntGnTJlWoUEF9+vTJ9D1FRkZq2rRp6tu3r6pVq6Y+ffqoXLlyunHjhn799Vd99NFHKl++vFq0aKFSpUqpd+/e+uCDD+Ti4qImTZro6NGjGjJkiIKDg/XSSy9Zj9ulSxd17txZffv2Vbt27XTs2DGNHz8+3aUB2VGhQgUtXrxY06ZNU7Vq1ayXjzzzzDPy8PBQrVq1FBgYqLi4OI0dO1a+vr566KGH7D4fADgFM3dxA4D73cKFC42OHTsaJUuWNLy9vQ03NzejePHiRpcuXYz9+/fbjD169KjRqFEjI3/+/IYkmx2A33nnHSMkJMRwd3c3ypQpY3z88cfGsGHD0u0OnZKSYkyaNMkoX768kTdvXsPX19eIjIw0vv/+e+uYm28ZlmbdunWGt7e3ERUVZVy9etUwDMPYsGGD0axZM6NgwYKGm5ubUaxYMaNZs2bpdrb+7rvvjIoVK1pvnfTOO+9kmC0jGe1e7u/vb9SuXdtYsmSJzdikpCTj5ZdfNooVK2bky5fPqFq1qvHNN98Y0dHRNp9V2s7i77777m3Pf7PExETjgw8+MB599FHDz8/PyJMnj1GwYEHjscceM8aNG2ecO3cuy+fI6vdLktGvXz9j6tSpxoMPPmi4ubkZpUuXNr744gubcWm7aW/fvt2mPW1X6ZtvmZaZb775xqhRo4aRL18+w8vLy6hfv77x888/Z3i87OxentnYfv36pXu/v/zyixEZGWl4enoaRYoUMXr16mXs2rXLkGRzy7q0W8nt2bPHqFOnjuHh4WEULFjQ6NOnj3H58mWbY2a0A/exY8eMjh07GoUKFTLc3NyMUqVKGe+++67N7u2G8e8t76pUqWK4u7sbkjLcyTskJMQoU6ZMhu8xOzkNwzBmzpxp1KhRw/Dy8jI8PDyMBx980OjatauxY8eODI//X7t37zaio6ON4sWLG3nz5jW8vLyMKlWqGEOHDjXOnDljHZeSkmKMGzfOiIiIMNzc3IzChQsbnTt3Nk6cOGFzvNTUVGP8+PFGWFiYkS9fPqN69erG2rVrM929/L/f67R/Bzd/786fP2888cQThp+fn2GxWKw/A3PmzDHq1q1rFC1a1MibN68RFBRktG/f3tizZ0+W3jsAODOLYdy0lgwAADgNi8Wifv366cMPPzQ7ilPp1q2bFi1apMuXL5uaY8+ePapUqZKmTJmivn37put3lpwAAHOxvBwAACAbDh8+rGPHjumNN95QYGBguttvAQBwMzZSAwAAyIaRI0eqYcOGunz5sr766qu7stEcACD3Ynk5AAAAAAAOwkw3AAAAAAAOQtENAAAAAICDUHQDAAAAAOAg9/zu5ampqTp58qTy588vi8VidhwAAAAAwD3AMAxdunRJQUFBcnHJfD77ni+6T548qeDgYLNjAAAAAADuQSdOnNADDzyQaf89X3Tnz59f0r8fhI+Pj8lpAAAAAAD3goSEBAUHB1trzszc80V32pJyHx8fim4AAAAAQI663WXMbKQGAAAAAICDUHQDAAAAAOAgFN0AAAAAADjIPX9NNwAAAAD8V2pqqq5fv252DDgxNzc3ubq63vFxKLoBAAAA3FeuX7+u2NhYpaammh0FTs7Pz08BAQG33SztVkwtuqdNm6Zp06bp6NGjkqRy5cpp6NChatKkiaR/bzY+YsQIffTRR7pw4YJq1KihKVOmqFy5ciamBgAAAJBbGYahU6dOydXVVcHBwXJx4YpbpGcYhq5evaozZ85IkgIDA+0+lqlF9wMPPKB33nlH4eHhkqQ5c+aoVatW+vXXX1WuXDmNHz9eEydO1OzZsxUREaFRo0apYcOG+vPPP297LzQAAAAA+K/k5GRdvXpVQUFB8vT0NDsOnJiHh4ck6cyZM/L397d7qbmpf9Zp0aKFmjZtqoiICEVERGj06NHy9vbWli1bZBiGJk+erDfffFNt27ZV+fLlNWfOHF29elXz5s0zMzYAAACAXColJUWSlDdvXpOTIDdI+8PMjRs37D6G06ylSElJ0YIFC3TlyhVFRkYqNjZWcXFxatSokXWMu7u7ateurV9++SXT4yQlJSkhIcHmAQAAAAA3u5NrdHH/yImfE9M3Utu7d68iIyOVmJgob29vLVmyRGXLlrUW1kWLFrUZX7RoUR07dizT440dO1YjRoxwaOb7wZTn1podAfeBftPrmR0BAAAAcCjTZ7pLlSql3bt3a8uWLerTp4+io6O1f/9+a/9//7JgGMYt/9owePBgXbx40fo4ceKEw7IDAAAAAHArps90582b17qRWvXq1bV9+3b973//02uvvSZJiouLs9kp7syZM+lmv2/m7u4ud3d3x4YGAAAAcE8Jef2Hu3q+o+80s+t1cXFxGj16tH744Qf9/fff8vf3V+XKlTVgwADVr18/h1NmzmKxaMmSJWrduvVdO2duZfpM938ZhqGkpCSFhoYqICBAq1atsvZdv35dGzZsUM2aNU1MCAAAAAB339GjR1WtWjWtXbtW48eP1969e7V8+XLVrVtX/fr1MzseMmHqTPcbb7yhJk2aKDg4WJcuXdKCBQu0fv16LV++XBaLRQMGDNCYMWNUsmRJlSxZUmPGjJGnp6c6duxoZmwAAAAAuOv69u0ri8Wibdu2ycvLy9perlw59ejRQ5J0/PhxPf/881qzZo1cXFwUFRWlDz74wLpauFu3boqPj9c333xjff2AAQO0e/durV+/XpJUp04dVaxYUfny5dMnn3yivHnz6rnnntPw4cMlSSEhIZKkNm3aSJJKlCiho0ePOvS952amFt2nT59Wly5ddOrUKfn6+qpixYpavny5GjZsKEl69dVXde3aNfXt21cXLlxQjRo1tHLlSu7RDQAAAOC+cv78eS1fvlyjR4+2KbjT+Pn5yTAMtW7dWl5eXtqwYYOSk5PVt29fdejQwVpQZ9WcOXM0cOBAbd26VZs3b1a3bt1Uq1YtNWzYUNu3b5e/v79mzZqlqKgou+9ffb8wtej+9NNPb9lvsVg0fPhw619UAAAAAOB+dOjQIRmGodKlS2c6ZvXq1dqzZ49iY2MVHBwsSfrss89Urlw5bd++XQ899FCWz1exYkUNGzZMklSyZEl9+OGHWrNmjRo2bKgiRYpI+rfQDwgIuIN3dX9wumu6AQAAAAC2DMOQdOv7RsfExCg4ONhacEtS2bJl5efnp5iYmGydr2LFijbPAwMDdebMmWwdA/+i6AYAAAAAJ1eyZElZLJZbFs+Z3V755nYXFxdrAZ/mxo0b6V7j5uZm89xisSg1NdWe6Pc9im4AAAAAcHIFCxZU48aNNWXKFF25ciVdf3x8vMqWLavjx4/rxIkT1vb9+/fr4sWLKlOmjCSpSJEiOnXqlM1rd+/ene08bm5uSklJyfbr7kcU3QAAAACQC0ydOlUpKSl6+OGH9fXXX+vgwYOKiYnR+++/r8jISDVo0EAVK1ZUp06dtGvXLm3btk1du3ZV7dq1Vb16dUlSvXr1tGPHDs2dO1cHDx7UsGHDtG/fvmxnCQkJ0Zo1axQXF6cLFy7k9Fu9p1B0AwAAAEAuEBoaql27dqlu3boaNGiQypcvr4YNG2rNmjWaNm2aLBaLvvnmGxUoUECPP/64GjRooLCwMC1cuNB6jMaNG2vIkCF69dVX9dBDD+nSpUvq2rVrtrNMmDBBq1atUnBwsKpUqZKTb/OeYzH+u6D/HpOQkCBfX19dvHhRPj4+ZsfJNaY8t9bsCLgP9Jtez+wIAADgPpOYmKjY2FiFhoYqX758ZseBk7vVz0tWa01mugEAAAAAcBCKbgAAAAAAHISiGwAAAAAAB6HoBgAAAADAQSi6AQAAAABwEIpuAAAAAAAchKIbAAAAAAAHoegGAAAAAMBBKLoBAAAAAHAQim4AAAAAuI8cPXpUFotFu3fvNjvKfSGP2QEAAAAAwHTDfe/y+S5ma3i3bt00Z84cPfvss5o+fbpNX9++fTVt2jRFR0dr9uzZORgy69ni4+P1zTff2LSvX79edevW1YULF+Tn5ydJSklJ0fvvv69Zs2bpwIEDypcvnyIjI/XWW2+pVq1a1tfOnj1b3bt3V+nSpRUTE2Nz3C+//FIdOnRQiRIldPToUQe/uzvHTDcAAAAA5ALBwcFasGCBrl27Zm1LTEzU/PnzVbx4cROTZY1hGHrqqaf09ttv64UXXlBMTIw2bNig4OBg1alTJ13R7uXlpTNnzmjz5s027TNnzswV7zcNRTcAAAAA5AJVq1ZV8eLFtXjxYmvb4sWLFRwcrCpVqljbli9frkcffVR+fn4qVKiQmjdvrsOHD9/y2Pv371fTpk3l7e2tokWLqkuXLjp79myO5v/yyy+1aNEizZ07V7169VJoaKgqVaqkjz76SC1btlSvXr105coV6/g8efKoY8eOmjlzprXtr7/+0vr169WxY8cczeZIFN0AAAAAkEt0795ds2bNsj6fOXOmevToYTPmypUrGjhwoLZv3641a9bIxcVFbdq0UWpqaobHPHXqlGrXrq3KlStrx44dWr58uU6fPq327dvnaPZ58+YpIiJCLVq0SNc3aNAgnTt3TqtWrbJp79mzpxYuXKirV69K+nfZeVRUlIoWLZqj2RyJa7oBAAAAIJfo0qWLBg8ebN0M7eeff9aCBQu0fv1665h27drZvObTTz+Vv7+/9u/fr/Lly6c75rRp01S1alWNGTPG2jZz5kwFBwfrwIEDioiIuG2upUuXytvb26YtJSXF5vmBAwdUpkyZDF+f1n7gwAGb9sqVK+vBBx/UokWL1KVLF82ePVsTJ07UkSNHbpvJWVB0AwAAAEAuUbhwYTVr1kxz5syRYRhq1qyZChcubDPm8OHDGjJkiLZs2aKzZ89aZ7iPHz+eYdG9c+dOrVu3Ll3RnHasrBTddevW1bRp02zatm7dqs6dO2fn7clisaRr69Gjh2bNmqXixYvr8uXLatq0qT788MNsHddMFN0AAAAAkIv06NFD/fv3lyRNmTIlXX+LFi0UHBysjz/+WEFBQUpNTVX58uV1/fr1DI+XmpqqFi1aaNy4cen6AgMDs5TJy8tL4eHhNm1//fWXzfOIiAjt378/w9en7VBesmTJdH2dOnXSq6++quHDh6tr167Kkyd3lbFc0w0AAAAAuUhUVJSuX7+u69evq3HjxjZ9586dU0xMjN566y3Vr19fZcqU0YULF255vKpVq+r3339XSEiIwsPDbR5eXl45lvupp57SwYMH9f3336frmzBhggoVKqSGDRum6ytYsKBatmypDRs2pLt+PTeg6AYAAACAXMTV1VUxMTGKiYmRq6urTV+BAgVUqFAhffTRRzp06JDWrl2rgQMH3vJ4/fr10/nz5/X0009r27ZtOnLkiFauXKkePXqkuy77Tjz11FNq06aNoqOj9emnn+ro0aPas2ePnn32WX333Xf65JNPMi3yZ8+erbNnz6p06dI5luduoegGAAAAgFzGx8dHPj4+6dpdXFy0YMEC7dy5U+XLl9dLL72kd99995bHCgoK0s8//6yUlBQ1btxY5cuX14svvihfX1+5uORcyWixWPTll1/qzTff1KRJk1S6dGk99thjOnbsmNatW6fWrVtn+loPDw8VKlQox7LcTRbDMAyzQzhSQkKCfH19dfHixQx/KJGxKc+tNTsC7gP9ptczOwIAALjPJCYmKjY2VqGhocqXL5/ZceDkbvXzktVak5luAAAAAAAchKIbAAAAAJCh48ePy9vbO9PH8ePHzY7o9Ezda33s2LFavHix/vjjD3l4eKhmzZoaN26cSpUqZR3TrVs3zZkzx+Z1NWrU0JYtW+52XAAAAAC4rwQFBWn37t237MetmVp0b9iwQf369dNDDz2k5ORkvfnmm2rUqJH2799vs2tdVFSUZs2aZX2eN29eM+ICAAAAwH0lT5486e6/jewxtehevny5zfNZs2bJ399fO3fu1OOPP25td3d3V0BAwN2OBwAAAADAHXGqa7ovXrwo6d+bn99s/fr18vf3V0REhJ555hmdOXMm02MkJSUpISHB5gEAAAAAgBmcpug2DEMDBw7Uo48+qvLly1vbmzRpoi+++EJr167VhAkTtH37dtWrV09JSUkZHmfs2LHy9fW1PoKDg+/WWwAAAAAAwIapy8tv1r9/f+3Zs0ebNm2yae/QoYP16/Lly6t69eoqUaKEfvjhB7Vt2zbdcQYPHqyBAwdanyckJFB4AwAAAABM4RRF9/PPP6/vvvtOP/30kx544IFbjg0MDFSJEiV08ODBDPvd3d3l7u7uiJgAAAAAAGSLqcvLDcNQ//79tXjxYq1du1ahoaG3fc25c+d04sQJBQYG3oWEAAAAAHBvOXr0qCwWyy1vBYacY+pMd79+/TRv3jx9++23yp8/v+Li4iRJvr6+8vDw0OXLlzV8+HC1a9dOgYGBOnr0qN544w0VLlxYbdq0MTM6AAAAgHtIhTkV7ur59kbvzdb4bt26ac6cOXr22Wc1ffp0m76+fftq2rRpio6O1uzZs3MwZdazxcfH65tvvrFpX79+verWrasLFy7Iz8/P+jwjp06d0iOPPKJjx45lep7atWtr/fr1OZj87jC16J42bZokqU6dOjbts2bNUrdu3eTq6qq9e/dq7ty5io+PV2BgoOrWrauFCxcqf/78JiQGAAAAAHMEBwdrwYIFmjRpkjw8PCRJiYmJmj9/vooXL25yuqz7888/5ePjY9Pm7++v7du3KyUlRZL0yy+/qF27djZj8+bNe9ez5gRTi27DMG7Z7+HhoRUrVtylNAAAAADgvKpWraojR45o8eLF6tSpkyRp8eLFCg4OVlhYmHXc8uXLNWrUKO3bt0+urq6KjIzU//73Pz344IOZHnv//v16+eWX9dNPP8nLy0uNGjXSpEmTVLhw4Rx/H/7+/vLz80vXXqRIEevXabeRzmxsbuI0twwDAAAAANxa9+7dNWvWLOvzmTNnqkePHjZjrly5ooEDB2r79u1as2aNXFxc1KZNG6WmpmZ4zFOnTql27dqqXLmyduzYoeXLl+v06dNq3769Q9/L/cIpdi8HAAAAANxely5dNHjwYOtmaD///LMWLFhgc61zu3btbF7z6aefyt/fX/v371f58uXTHXPatGmqWrWqxowZY22bOXOmgoODdeDAAUVERNw219KlS+Xt7W3TlrZU/L/+e8eqYsWK6c8//7ztOXIrim4AAAAAyCUKFy6sZs2aac6cOTIMQ82aNUu3BPzw4cMaMmSItmzZorNnz1pnuI8fP55h0b1z506tW7cuXdGcdqysFN1169a17tmVZuvWrercuXO6sRs3brTZoytPnnu7LL233x0AAAAA3GN69Oih/v37S5KmTJmSrr9FixYKDg7Wxx9/rKCgIKWmpqp8+fK6fv16hsdLTU1VixYtNG7cuHR9Wb1Vs5eXl8LDw23a/vrrrwzHhoaG5vrrtLODohsAAAAAcpGoqChrAd24cWObvnPnzikmJkYzZszQY489JknatGnTLY9XtWpVff311woJCbnnZ53NwEZqAAAAAJCLuLq6KiYmRjExMXJ1dbXpK1CggAoVKqSPPvpIhw4d0tq1azVw4MBbHq9fv346f/68nn76aW3btk1HjhzRypUr1aNHj0yvy74TZ86cUVxcnM3jxo0bOX4eZ0HRDQAAAAC5jI+PT7p7XUuSi4uLFixYoJ07d6p8+fJ66aWX9O67797yWEFBQfr555+VkpKixo0bq3z58nrxxRfl6+srF5ecLxlLlSqlwMBAm8fOnTtz/DzOwmLc7mbZuVxCQoJ8fX118eLFDH8okbEpz601OwLuA/2m1zM7AgAAuM8kJiYqNjZWoaGhypcvn9lx4ORu9fOS1VqTmW4AAAAAAByEq+SRoXrr+5kdAfeFGLMDAAAA4BaOHz+usmXLZtq/f/9+FS9e/C4myn0ougEAAAAAGQoKCtLu3btv2Y9bo+gGAAAAAGQoT5486e6/jezhmm4AAAAAAByEohsAAAAAAAeh6AYAAAAAwEEougEAAAAAcBCKbgAAAAAAHISiGwAAAACQbSEhIZo8ebLZMZwetwwDAAAAcN+LKV3mrp6vzB8x2RrfrVs3zZkzJ137wYMH74lbes2ePVsDBgxQfHx8uj6LxaIlS5aodevW1ralS5fqvffe086dO5WSkqJy5cqpX79+6tatm3XM0aNHFRoaKldXVx07dkzFihWz9p06dUrBwcFKSUlRbGysQkJCHPbemOkGAAAAgFwgKipKp06dsnmEhoaaHeuu++CDD9SqVSvVrFlTW7du1Z49e/TUU0/pueee08svv5xufFBQkObOnWvTNmfOHJsi3JEougEAAAAgF3B3d1dAQIDNw9XVVd9//72qVaumfPnyKSwsTCNGjFBycrL1dRaLRTNmzFDz5s3l6empMmXKaPPmzTp06JDq1KkjLy8vRUZG6vDhw9bXHD58WK1atVLRokXl7e2thx56SKtXr75lvosXL6p3797y9/eXj4+P6tWrp99++y1HP4MTJ05o0KBBGjBggMaMGaOyZcsqPDxcgwYN0rvvvqsJEyZo69atNq+Jjo7WrFmzbNpmz56t6OjoHM2WGYpuAAAAAMilVqxYoc6dO+uFF17Q/v37NWPGDM2ePVujR4+2GTdy5Eh17dpVu3fvVunSpdWxY0c9++yzGjx4sHbs2CFJ6t+/v3X85cuX1bRpU61evVq//vqrGjdurBYtWuj48eMZ5jAMQ82aNVNcXJyWLVumnTt3qmrVqqpfv77Onz+fY+930aJFunHjRoYz2s8++6y8vb01f/58m/aWLVvqwoUL2rRpkyRp06ZNOn/+vFq0aJFjuW6FohsAAAAAcoGlS5fK29vb+njyySc1evRovf7664qOjlZYWJgaNmyokSNHasaMGTav7d69u9q3b6+IiAi99tprOnr0qDp16qTGjRurTJkyevHFF7V+/Xrr+EqVKunZZ59VhQoVVLJkSY0aNUphYWH67rvvMsy2bt067d27V1999ZWqV6+ukiVL6r333pOfn58WLVqUpfd38eJFm/eX9rjZgQMH5Ovrq8DAwHSvz5s3r8LCwnTgwAGbdjc3N3Xu3FkzZ86UJM2cOVOdO3eWm5tblnLdKTZSAwAAAIBcoG7dupo2bZr1uZeXl8LDw7V9+3abme2UlBQlJibq6tWr8vT0lCRVrFjR2l+0aFFJUoUKFWzaEhMTlZCQIB8fH125ckUjRozQ0qVLdfLkSSUnJ+vatWuZznTv3LlTly9fVqFChWzar127ZrNs/Vby58+vXbt2pWsvWbJkll4v/TvjbrFY0rX37NlTkZGRGjNmjL766itt3rzZZgm+I1F0AwAAAEAukFZk3yw1NVUjRoxQ27Zt043Ply+f9eubZ3XTitKM2lJTUyVJr7zyilasWKH33ntP4eHh8vDw0BNPPKHr169nmC01NVWBgYE2s+Vp/Pz8svT+XFxcbrsTe0REhC5evKiTJ08qKCjIpu/69es6cuSI6tWrl+515cuXV+nSpfX000+rTJkyKl++vHbv3p2lXHeKohsZaj+YHw043l6zAwAAAORyVatW1Z9//pnjtw3buHGjunXrpjZt2kj69xrvo0eP3jJHXFyc8uTJ49Dbb7Vr106vvvqqJkyYoAkTJtj0TZ8+XVeuXNHTTz+d4Wt79Oihvn372qwWuBuorAAAAAAglxo6dKiaN2+u4OBgPfnkk3JxcdGePXu0d+9ejRo1yu7jhoeHa/HixWrRooUsFouGDBlinQXPSIMGDRQZGanWrVtr3LhxKlWqlE6ePKlly5apdevWql69ut1Zbla8eHGNHz9eL7/8svLly6cuXbrIzc1N3377rd544w0NGjRINWrUyPC1zzzzjJ588sksz7znFDZSAwAAAIBcqnHjxlq6dKlWrVqlhx56SI888ogmTpyoEiVK3NFxJ02apAIFCqhmzZpq0aKFGjdurKpVq2Y63mKxaNmyZXr88cfVo0cPRURE6KmnntLRo0et15DnlJdeeklLlizRxo0bVb16dZUvX17z5s3TtGnT9N5772X6ujx58qhw4cLKk+fuzj1bDMMw7uoZbzJ27FgtXrxYf/zxhzw8PFSzZk3rX0XSGIahESNG6KOPPtKFCxdUo0YNTZkyReXKlcvSORISEuTr66uLFy/Kx8fHUW/lnlNhToXbDwLu0N5oFpgDAIC7KzExUbGxsQoNDbW55hnIyK1+XrJaa5o6071hwwb169dPW7Zs0apVq5ScnKxGjRrpypUr1jHjx4/XxIkT9eGHH2r79u0KCAhQw4YNdenSJROTAwAAAABwe6YW3cuXL1e3bt1Urlw5VapUSbNmzdLx48e1c+dOSf/Ock+ePFlvvvmm2rZtq/Lly2vOnDm6evWq5s2bZ2Z0AAAAAEAWlStXLsN7cHt7e+uLL74wO55DOdVGahcvXpQkFSxYUJIUGxuruLg4NWrUyDrG3d1dtWvX1i+//KJnn33WlJwAAAAAgKxbtmyZbty4kWFfTl/z7Wycpug2DEMDBw7Uo48+qvLly0uS4uLiJKX/JhQtWlTHjh3L8DhJSUlKSkqyPk9ISHBQYgAAAABAVtzpxm65mdMU3f3799eePXu0adOmdH1pN2pPYxhGurY0Y8eO1YgRIxySEQCQ+0x5bq3ZEXCP6ze9ntkRAABOzCluGfb888/ru+++07p16/TAAw9Y2wMCAiT9/xnvNGfOnMl0CcLgwYN18eJF6+PEiROOCw4AAAAAwC2YWnQbhqH+/ftr8eLFWrt2rUJDQ236Q0NDFRAQoFWrVlnbrl+/rg0bNqhmzZoZHtPd3V0+Pj42DwAAAAAAzGDq8vJ+/fpp3rx5+vbbb5U/f37rjLavr688PDxksVg0YMAAjRkzRiVLllTJkiU1ZswYeXp6qmPHjmZGBwAAAADgtkwtuqdNmyZJqlOnjk37rFmz1K1bN0nSq6++qmvXrqlv3766cOGCatSooZUrVyp//vx3OS0AAAAAANljatFtGMZtx1gsFg0fPlzDhw93fCAAAAAAQJaEhIRowIABGjBggNlRnJrT7F4OAAAAAGa523e7yO6dD7p166Y5c+akaz948KDCw8NzKpZpZs+erQEDBig+Pj5dn8Vi0ZIlS9S6dWvr84zMnz9fy5cvz/BzullWJn9zEkU3AAAAAOQCUVFRmjVrlk1bkSJFTEpjrlmzZikqKsqmzc/PT02aNNE777xjbQsMDMxw7N3kFLcMAwAAAADcmru7uwICAmwerq6u+v7771WtWjXly5dPYWFhGjFihJKTk62vs1gsmjFjhpo3by5PT0+VKVNGmzdv1qFDh1SnTh15eXkpMjJShw8ftr7m8OHDatWqlYoWLSpvb2899NBDWr169S3zXbx4Ub1795a/v798fHxUr149/fbbbw75LPz8/NJ9Fvny5ZOvr69NW0Zj7zaKbgAAAADIpVasWKHOnTvrhRde0P79+zVjxgzNnj1bo0ePthk3cuRIde3aVbt371bp0qXVsWNHPfvssxo8eLB27NghSerfv791/OXLl9W0aVOtXr1av/76qxo3bqwWLVro+PHjGeYwDEPNmjVTXFycli1bpp07d6pq1aqqX7++zp8/77gPIBdgeTkAAAAA5AJLly6Vt7e39XmTJk10+vRpvf7664qOjpYkhYWFaeTIkXr11Vc1bNgw69ju3burffv2kqTXXntNkZGRGjJkiBo3bixJevHFF9W9e3fr+EqVKqlSpUrW56NGjdKSJUv03Xff2RTnadatW6e9e/fqzJkzcnd3lyS99957+uabb7Ro0SL17t37tu/v4sWLNu/vVp5++mm5urratO3Zs0dhYWFZev3dRNENAAAAALlA3bp1rbddliQvLy+Fh4dr+/btNjPbKSkpSkxM1NWrV+Xp6SlJqlixorW/aNGikqQKFSrYtCUmJiohIUE+Pj66cuWKRowYoaVLl+rkyZNKTk7WtWvXMp3p3rlzpy5fvqxChQrZtF+7ds1m2fqt5M+fX7t27UrXXrJkyXRtkyZNUoMGDWzagoODs3Seu42iGwAAAABygbQi+2apqakaMWKE2rZtm258vnz5rF+7ublZv07b/TujttTUVEnSK6+8ohUrVui9995TeHi4PDw89MQTT+j69esZZktNTVVgYKDWr1+frs/Pzy9L78/FxSXLO7EHBATkml3bKboBAAAAIJeqWrWq/vzzzxwvQDdu3Khu3bqpTZs2kv69xvvo0aO3zBEXF6c8efIoJCQkR7PkdhTdAAAAAJBLDR06VM2bN1dwcLCefPJJubi4aM+ePdq7d69GjRpl93HDw8O1ePFitWjRQhaLRUOGDLHOgmekQYMGioyMVOvWrTVu3DiVKlVKJ0+e1LJly9S6dWtVr17d7iwZiY+PV1xcnE1b/vz55eXllaPnyQl27V4eFhamc+fOpWuPj493ygvXAQAAAOBe1LhxYy1dulSrVq3SQw89pEceeUQTJ05UiRIl7ui4kyZNUoECBVSzZk21aNFCjRs3VtWqVTMdb7FYtGzZMj3++OPq0aOHIiIi9NRTT+no0aPWa8hzUvfu3RUYGGjz+OCDD3L8PDnBYhiGkd0Xubi4KC4uTv7+/jbtp0+fVvHixZWUlJRjAe9UQkKCfH19dfHiRfn4+JgdJ9eoMKfC7QcBd2hv9F6zI+A+MOW5tWZHwD2u3/R6ZkcAkA2JiYmKjY1VaGiozTXPQEZu9fOS1VozW8vLv/vuO+vXK1askK+vr/V5SkqK1qxZw/p9AAAAAAD+T7aK7tatW0v6d+lA2n3g0ri5uSkkJEQTJkzIsXAAAAAAgNyvXLlyOnbsWIZ9M2bMUKdOne5yorsnW0V32oXzoaGh2r59uwoXLuyQUAAAAACAe8eyZct048aNDPsccc23M7Fr9/LY2NiczgEAAAAAuEfd6cZuuZndtwxbs2aN1qxZozNnzqTbOn7mzJl3HAwAAAAAgNzOrqJ7xIgRevvtt1W9enUFBgbKYrHkdC4AAAAAcBg7buKE+1BO/JzYVXRPnz5ds2fPVpcuXe44AAAAAADcLa6urpKk69evy8PDw+Q0cHZXr16V9O/G4fayq+i+fv26atasafdJAQAAAMAMefLkkaenp/755x+5ubnJxcXF7EhwQoZh6OrVqzpz5oz8/Pysf6yxh11Fd69evTRv3jwNGTLE7hMDAAAAwN1msVgUGBio2NjYTG9hBaTx8/NTQEDAHR3DrqI7MTFRH330kVavXq2KFSumm2qfOHHiHYUCAAAAAEfJmzevSpYsqevXr5sdBU7Mzc3tjma409hVdO/Zs0eVK1eWJO3bt8+mj03VAAAAADg7FxcX5cuXz+wYuA/YVXSvW7cup3MAAAAAAHDPYdcAAAAAAAAcxK6Z7rp1695yGfnatWvtDgQAAAAAwL3CrqI77XruNDdu3NDu3bu1b98+RUdH50QuAAAAAAByPbuK7kmTJmXYPnz4cF2+fPmOAgEAkJPqre9ndgTc82LMDgAAcGI5ek13586dNXPmzJw8JAAAAAAAuVaOFt2bN29m230AAAAAAP6PXcvL27Zta/PcMAydOnVKO3bs0JAhQ7J8nJ9++knvvvuudu7cqVOnTmnJkiVq3bq1tb9bt26aM2eOzWtq1KihLVu22BMbAAAAAIC7yq6i29fX1+a5i4uLSpUqpbfffluNGjXK8nGuXLmiSpUqqXv37mrXrl2GY6KiojRr1izr87x589oTGQAAAACAu86uovvmIvhONGnSRE2aNLnlGHd3dwUEBOTI+QAAAAAAuJvsKrrT7Ny5UzExMbJYLCpbtqyqVKmSU7ms1q9fL39/f/n5+al27doaPXq0/P39c/w8AAAAAADkNLuK7jNnzuipp57S+vXr5efnJ8MwdPHiRdWtW1cLFixQkSJFciRckyZN9OSTT6pEiRKKjY3VkCFDVK9ePe3cuVPu7u4ZviYpKUlJSUnW5wkJCTmSBQAAAACA7LKr6H7++eeVkJCg33//XWXKlJEk7d+/X9HR0XrhhRc0f/78HAnXoUMH69fly5dX9erVVaJECf3www/pNnNLM3bsWI0YMSJHzn8/uxTzjtkRAAAAACDXs+uWYcuXL9e0adOsBbcklS1bVlOmTNGPP/6YY+H+KzAwUCVKlNDBgwczHTN48GBdvHjR+jhx4oTD8gAAAAAAcCt2zXSnpqbKzc0tXbubm5tSU1PvOFRmzp07pxMnTigwMDDTMe7u7pkuPQcAAAAA4G6ya6a7Xr16evHFF3Xy5Elr299//62XXnpJ9evXz/JxLl++rN27d2v37t2SpNjYWO3evVvHjx/X5cuX9fLLL2vz5s06evSo1q9frxYtWqhw4cJq06aNPbEBAAAAALir7Cq6P/zwQ126dEkhISF68MEHFR4ertDQUF26dEkffPBBlo+zY8cOValSxbrr+cCBA1WlShUNHTpUrq6u2rt3r1q1aqWIiAhFR0crIiJCmzdvVv78+e2JDQAAAADAXWXX8vLg4GDt2rVLq1at0h9//CHDMFS2bFk1aNAgW8epU6eODMPItH/FihX2xAMAAAAAwClka6Z77dq1Klu2rPU2XA0bNtTzzz+vF154QQ899JDKlSunjRs3OiQoAAAAAAC5TbaK7smTJ+uZZ56Rj49Puj5fX189++yzmjhxYo6FAwAAAAAgN8vW8vLffvtN48aNy7S/UaNGeu+99+44FAAAOaX9YLuupAKybK/ZAQAATi1bM92nT5/O8FZhafLkyaN//vnnjkMBAAAAAHAvyFbRXaxYMe3dm/nfc/fs2XPLe2gDAAAAAHA/yVbR3bRpUw0dOlSJiYnp+q5du6Zhw4apefPmORYOAAAAAIDcLFsXur311ltavHixIiIi1L9/f5UqVUoWi0UxMTGaMmWKUlJS9OabbzoqKwAAAAAAuUq2iu6iRYvql19+UZ8+fTR48GDrPbYtFosaN26sqVOnqmjRog4JCgAAAABAbpPtLV1LlCihZcuW6cKFCzp06JAMw1DJkiVVoEABR+QDAAAAACDXsvs+KgUKFNBDDz2Uk1kAAAAAALinZGsjNQAAAAAAkHUU3QAAAAAAOAhFNwAAAAAADkLRDQAAAACAg1B0AwAAAADgIBTdAAAAAAA4CEU3AAAAAAAOQtENAAAAAICDUHQDAAAAAOAgFN0AAAAAADgIRTcAAAAAAA5C0Q0AAAAAgINQdAMAAAAA4CAU3QAAAAAAOAhFNwAAAAAADkLRDQAAAACAg1B0AwAAAADgIBTdAAAAAAA4CEU3AAAAAAAOYmrR/dNPP6lFixYKCgqSxWLRN998Y9NvGIaGDx+uoKAgeXh4qE6dOvr999/NCQsAAAAAQDaZWnRfuXJFlSpV0ocffphh//jx4zVx4kR9+OGH2r59uwICAtSwYUNdunTpLicFAAAAACD78ph58iZNmqhJkyYZ9hmGocmTJ+vNN99U27ZtJUlz5sxR0aJFNW/ePD377LN3MyoAAAAAANnmtNd0x8bGKi4uTo0aNbK2ubu7q3bt2vrll18yfV1SUpISEhJsHgAAAAAAmMHUme5biYuLkyQVLVrUpr1o0aI6duxYpq8bO3asRowY4dBsAIDc41LMO2ZHAIA7NuW5tWZHwH2g3/R6Zke4JzntTHcai8Vi89wwjHRtNxs8eLAuXrxofZw4ccLREQEAAAAAyJDTznQHBARI+nfGOzAw0Np+5syZdLPfN3N3d5e7u7vD8wEAAAAAcDtOO9MdGhqqgIAArVq1ytp2/fp1bdiwQTVr1jQxGQAAAAAAWWPqTPfly5d16NAh6/PY2Fjt3r1bBQsWVPHixTVgwACNGTNGJUuWVMmSJTVmzBh5enqqY8eOJqYGAAAAACBrTC26d+zYobp161qfDxw4UJIUHR2t2bNn69VXX9W1a9fUt29fXbhwQTVq1NDKlSuVP39+syIDAAAAAJBlphbdderUkWEYmfZbLBYNHz5cw4cPv3uhAAAAAADIIU57TTcAAAAAALkdRTcAAAAAAA5C0Q0AAAAAgINQdAMAAAAA4CAU3QAAAAAAOAhFNwAAAAAADkLRDQAAAACAg1B0AwAAAADgIBTdAAAAAAA4CEU3AAAAAAAOQtENAAAAAICDUHQDAAAAAOAgFN0AAAAAADgIRTcAAAAAAA5C0Q0AAAAAgINQdAMAAAAA4CAU3QAAAAAAOEgeswMAAAAAuLXpkS+aHQH3gX7aa3aEexIz3QAAAAAAOAhFNwAAAAAADkLRDQAAAACAg1B0AwAAAADgIBTdAAAAAAA4CEU3AAAAAAAOQtENAAAAAICDUHQDAAAAAOAgFN0AAAAAADgIRTcAAAAAAA5C0Q0AAAAAgIM4ddE9fPhwWSwWm0dAQIDZsQAAAAAAyJI8Zge4nXLlymn16tXW566uriamAQAAAAAg65y+6M6TJw+z2wAAAACAXMmpl5dL0sGDBxUUFKTQ0FA99dRTOnLkyC3HJyUlKSEhweYBAAAAAIAZnHqmu0aNGpo7d64iIiJ0+vRpjRo1SjVr1tTvv/+uQoUKZfiasWPHasSIEXc5KQAAAOA4l2LeMTsCADs59Ux3kyZN1K5dO1WoUEENGjTQDz/8IEmaM2dOpq8ZPHiwLl68aH2cOHHibsUFAAAAAMCGU890/5eXl5cqVKiggwcPZjrG3d1d7u7udzEVAAAAAAAZc+qZ7v9KSkpSTEyMAgMDzY4CAAAAAMBtOXXR/fLLL2vDhg2KjY3V1q1b9cQTTyghIUHR0dFmRwMAAAAA4Lacenn5X3/9paefflpnz55VkSJF9Mgjj2jLli0qUaKE2dEAAAAAALgtpy66FyxYYHYEAAAAAADs5tTLywEAAAAAyM0ougEAAAAAcBCKbgAAAAAAHISiGwAAAAAAB6HoBgAAAADAQSi6AQAAAABwEIpuAAAAAAAchKIbAAAAAAAHoegGAAAAAMBBKLoBAAAAAHAQim4AAAAAAByEohsAAAAAAAeh6AYAAAAAwEEougEAAAAAcBCKbgAAAAAAHISiGwAAAAAAB6HoBgAAAADAQSi6AQAAAABwEIpuAAAAAAAchKIbAAAAAAAHoegGAAAAAMBBKLoBAAAAAHAQim4AAAAAAByEohsAAAAAAAeh6AYAAAAAwEEougEAAAAAcBCKbgAAAAAAHISiGwAAAAAAB8kVRffUqVMVGhqqfPnyqVq1atq4caPZkQAAAAAAuC2nL7oXLlyoAQMG6M0339Svv/6qxx57TE2aNNHx48fNjgYAAAAAwC05fdE9ceJE9ezZU7169VKZMmU0efJkBQcHa9q0aWZHAwAAAADglpy66L5+/bp27typRo0a2bQ3atRIv/zyi0mpAAAAAADImjxmB7iVs2fPKiUlRUWLFrVpL1q0qOLi4jJ8TVJSkpKSkqzPL168KElKSEhwXNB7UGrSVbMj4D7Av0vcDfw+g6Pxuwx3A7/LcDfw+yx70j4vwzBuOc6pi+40FovF5rlhGOna0owdO1YjRoxI1x4cHOyQbADs5zvZ7AQAcOf4XQbgXsHvM/tcunRJvr6+mfY7ddFduHBhubq6ppvVPnPmTLrZ7zSDBw/WwIEDrc9TU1N1/vx5FSpUKNNCHbhTCQkJCg4O1okTJ+Tj42N2HACwG7/PANwL+F2Gu8EwDF26dElBQUG3HOfURXfevHlVrVo1rVq1Sm3atLG2r1q1Sq1atcrwNe7u7nJ3d7dp8/Pzc2RMwMrHx4df7ADuCfw+A3Av4HcZHO1WM9xpnLrolqSBAweqS5cuql69uiIjI/XRRx/p+PHjeu6558yOBgAAAADALTl90d2hQwedO3dOb7/9tk6dOqXy5ctr2bJlKlGihNnRAAAAAAC4JacvuiWpb9++6tu3r9kxgEy5u7tr2LBh6S5tAIDcht9nAO4F/C6DM7EYt9vfHAAAAAAA2MXF7AAAAAAAANyrKLoBAAAAAHAQim4AAAAAAByEohsAAAAAAAeh6AYAAAAAwEEougEAAADkesuXL9emTZusz6dMmaLKlSurY8eOunDhgonJcL+j6AbuwGeffaZatWopKChIx44dkyRNnjxZ3377rcnJACB79uzZo0WLFum7777TH3/8YXYcAMi2V155RQkJCZKkvXv3atCgQWratKmOHDmigQMHmpwO9zOKbsBO06ZN08CBA9W0aVPFx8crJSVFkuTn56fJkyebGw4Asmjbtm2qUKGCqlSpovbt26t169YqV66cHnnkEZvi+/z58yamBIDbi42NVdmyZSVJX3/9tZo3b64xY8Zo6tSp+vHHH01Oh/sZRTdgpw8++EAff/yx3nzzTbm6ulrbq1evrr1795qYDACyZv/+/apfv748PDz0+eefa9euXdq5c6c+++wzpaSkqGbNmjp58qSmTp2qqVOnmh0XAG4pb968unr1qiRp9erVatSokSSpYMGC1hlwwAx5zA4A5FaxsbGqUqVKunZ3d3dduXLFhEQAkD3Dhg1Tw4YN9fXXX8tisVjbq1Spoqefflpt27ZV3bp1deLECWaJADi9WrVqaeDAgapVq5a2bdumhQsXSpIOHDigBx54wOR0uJ8x0w3YKTQ0VLt3707X/uOPP1qXNgGAM1u/fr3eeOMNm4I7jcVi0RtvvKGDBw9qyZIlql27tgkJASDrpkyZIjc3Ny1atEjTpk1TsWLFJP37/2ZRUVEmp8P9jJluwE6vvPKK+vXrp8TERBmGoW3btmn+/PkaO3asPvnkE7PjAcBtXbp0SUWLFs20PyAgQG5ubmrcuPFdTAUA2ZecnKx169bpo48+UmBgoE3fpEmTTEoF/IuiG7BT9+7dlZycrFdffVVXr15Vx44dVaxYMf3vf//TU089ZXY8ALitkJAQbdu2TcHBwRn2b926VSVKlLjLqQAg+/LkyaM+ffooJibG7ChAOiwvB+7AM888o2PHjunMmTOKi4vTiRMn1LNnT7NjAUCWdOjQQQMHDtS+ffvS9e3du1cvv/wyf0QEkGvUqFFDv/76q9kxgHQshmEYZocAcqPY2FglJyerZMmSNu0HDx6Um5ubQkJCzAkGAFmUmJio+vXra+vWrWrYsKHKlCkj6d9dzVevXq2HH35Ya9askYeHh8lJAeD2vvrqK73++ut66aWXVK1aNXl5edn0V6xY0aRkuN9RdAN2ql27tnr06KHo6Gib9s8//1yffPKJ1q9fb04wAMiG69eva9KkSZo/f74OHDggSSpZsqSefvppvfTSS3J3dzc5IQBkjYtL+kW8FotFhmHIYrEoJSXFhFQARTdgNx8fH+3atUvh4eE27YcOHVL16tUVHx9vTjAAAID70LFjx27Zzx4VMAsbqQF2slgsunTpUrr2ixcv8pdUALnOsWPHFBcXJ4vFoqJFi/I/pwByHX5vwVmxkRpgp8cee0xjx461KbBTUlI0duxYPfrooyYmA4CsmzRpkoKDgxUWFqbIyEg98sgjCgsLU3BwsCZPnmx2PADIls8++0y1atVSUFCQdeZ78uTJ+vbbb01OhvsZM92AncaPH6/HH39cpUqV0mOPPSZJ2rhxoxISErR27VqT0wHA7Y0cOVLvvfee3njjDTVu3FhFixaVYRg6c+aMVqxYoeHDh+vy5ct66623zI4KALc1bdo0DR06VAMGDNDo0aOtEyN+fn6aPHmyWrVqZXJC3K+4phu4AydPntSHH36o3377TR4eHqpYsaL69++vggULmh0NAG4rODhYH3zwgVq3bp1h/5IlS9S/f3/9/fffdzcYANihbNmyGjNmjFq3bq38+fPrt99+U1hYmPbt26c6dero7NmzZkfEfYqZbuAOBAUFacyYMWbHAAC7nDt3TqVKlcq0PyIiQhcuXLiLiQDAfrGxsapSpUq6dnd3d125csWERMC/KLqBOxAfH69t27bpzJkzSk1Ntenr2rWrSakAIGsefvhhjR49WrNnz1aePLb/S5CcnKwxY8bo4YcfNikdAGRPaGiodu/enW5DtR9//FFly5Y1KRVA0Q3Y7fvvv1enTp105coV5c+fXxaLxdpnsVgougE4vQ8++ECNGjWSv7+/ateuraJFi8pisSguLk4//fST3N3dtWrVKrNjAkCWvPLKK+rXr58SExNlGIa2bdum+fPna+zYsfrkk0/Mjof7GNd0A3aKiIhQ06ZNNWbMGHl6epodBwDscunSJX3++efasmWL4uLiJEkBAQGKjIxUx44d5ePjY3JCAMi6jz/+WKNGjdKJEyckScWKFdPw4cPVs2dPk5PhfkbRDdjJy8tLe/fuVVhYmNlRAAAAcJOzZ88qNTVV/v7+ZkcBuE83YK/GjRtrx44dZscAALuxky+Ae1XhwoUpuOE0uKYbsFOzZs30yiuvaP/+/apQoYLc3Nxs+lu2bGlSMgDImqJFi6pOnTrq2bOn2rVrJ3d3d7MjAYDdTp8+rZdffllr1qzRmTNn9N8FvWn37QbuNpaXA3Zyccl8oYjFYuEXOwCn5+LiosaNG2vt2rXy8vJSp06d1LNnT1WuXNnsaACQbU2aNNHx48fVv39/BQYG2mxyK0mtWrUyKRnudxTdAADcp1xcXBQXFycXFxfNmTNHs2bNUkxMjCpXrqxevXqpY8eO8vX1NTsmAGRJ/vz5tXHjRv5wCKfDNd1ADkhMTDQ7AgDYrXDhwho0aJD27dunTZs2qXLlynrttdcUFBTE7Q8B5BrBwcHplpQDzoCiG7BTSkqKRo4cqWLFisnb21tHjhyRJA0ZMkSffvqpyekA4Pb+u/RSkiIjI/Xpp5/q1KlTev/993X48GETkgFA9k2ePFmvv/66jh49anYUwAbLywE7vf3225ozZ47efvttPfPMM9q3b5/CwsL05ZdfatKkSdq8ebPZEQHgltKWl7PDL4DcqkCBAjZ/QLxy5YqSk5Pl6emZbpPb8+fP3+14gCR2LwfsNnfuXH300UeqX7++nnvuOWt7xYoV9ccff5iYDACyZtasWVyzDSBXmzx5stkRgNui6Abs9Pfffys8PDxde2pqqm7cuGFCIgDInujoaLMjAMAd4fcYcgOKbsBO5cqV08aNG1WiRAmb9q+++kpVqlQxKRUAZN/Bgwf1yy+/KC4uThaLRUWLFlXNmjVVsmRJs6MBQJa5urrq1KlT6S6ZOXfunPz9/bmdK0xD0Q3YadiwYerSpYv+/vtvpaamavHixfrzzz81d+5cLV261Ox4AHBbFy9eVNeuXfX999/L19dX/v7+MgxD//zzjxISEtSiRQvNnTtXPj4+ZkcFgNvKbKuqpKQk5c2b9y6nAf4/im7ATi1atNDChQs1ZswYWSwWDR06VFWrVtX333+vhg0bmh0PAG7r+eefV2xsrDZv3qwaNWrY9G3dulW9e/fW888/rzlz5piUEABu7/3335f07x0ZPvnkE3l7e1v7UlJS9NNPP6l06dJmxQPYvRwAgPuVn5+fVqxYka7gTrNlyxZFRUUpPj7+7gYDgGwIDQ2VJB07dkwPPPCAXF1drX158+ZVSEiI3n777Ux/1wGOxkw3AAD3sYzu1Z2VPgBwFrGxsZKkunXravHixUpOTpaLi4sKFSpkcjLgXy5mBwByKxcXF7m6umb6AABn16JFCz3zzDPasWNHur4dO3boueeeU8uWLU1IBgDZEx8frzJlyqhkyZIKCAiQv7+/ChcurP79+7NaB6ZjeTlgp2+//dbm+Y0bN/Trr79qzpw5GjFihHr27GlSMgDImvj4eD399NNasWKF/Pz85O/vL4vFotOnT+vixYtq3Lix5s2bJz8/P7OjAkCmzp8/r8jISP3999/q1KmTypQpI8MwFBMTo3nz5ik4OFi//PKLChQoYHZU3KcouoEcNm/ePC1cuDBdUQ4AzuqPP/7Q5s2bFRcXJ0kKCAhQZGQkGw8ByBUGDBigNWvWaPXq1SpatKhNX1xcnBo1aqT69etr0qRJJiXE/Y6iG8hhhw8fVsWKFXXlyhWzowAAANzzQkJCNGPGDDVu3DjD/uXLl+u5557T0aNH724w4P9wTTeQg65du6YPPvhADzzwgNlRAOCOXbhwQXPnzjU7BgDc0qlTp1SuXLlM+8uXL29dyQOYgd3LATsVKFDAZmdfwzB06dIleXp66vPPPzcxGQDkjOPHj6t79+7q2rWr2VEAIFOFCxfW0aNHM530iI2NZSdzmIqiG7DTpEmTbIpuFxcXFSlSRDVq1GCjDgC5QkJCwi37L126dJeSAID9oqKi9Oabb2rVqlXKmzevTV9SUpKGDBmiqKgok9IBXNMNAMB9y8XF5Zb34jYMQxaLRSkpKXcxFQBkz19//aXq1avL3d1d/fr1s24CuX//fk2dOlVJSUnasWOHgoODTU6K+xVFN2CnPXv2ZHlsxYoVHZgEAOzj6+urN998UzVq1Miw/+DBg3r22WcpugE4vdjYWPXt21crV65UWnljsVjUsGFDffjhhwoPDzc5Ie5nLC8H7FS5cuVbzhBJzBIBcG5Vq1aVJNWuXTvDfj8/P/G3eQC5QWhoqH788UdduHBBBw8elCSFh4erYMGCJicDKLoBuy1evFgvv/yyXnnlFUVGRkqSNm/erAkTJmj8+PGqUqWKyQkB4NY6duyoa9euZdofEBCgYcOG3cVEAHBnChQooIcfftjsGIANlpcDdnr44Yc1fPhwNW3a1KZ92bJlGjJkiHbu3GlSMgDIOX///beKFStmdgwAAHIt7tMN2Gnv3r0KDQ1N1x4aGqr9+/ebkAgAck5cXJyef/55roMEAOAOUXQDdipTpoxGjRqlxMREa1tSUpJGjRqlMmXKmJgMALImPj5enTp1UpEiRRQUFKT3339fqampGjp0qMLCwrRlyxbNnDnT7JgAAORqLC8H7LRt2za1aNFCqampqlSpkiTpt99+k8Vi0dKlS7meCIDT69u3r77//nt16NBBy5cvV0xMjBo3bqzExEQNGzYs0w3WAABA1lF0A3fg6tWr+vzzz/XHH3/IMAyVLVtWHTt2lJeXl9nRAOC2SpQooU8//VQNGjTQkSNHFB4erhdeeEGTJ082OxoAAPcMim4AAO5Tbm5uOnbsmIKCgiRJnp6e2rZtm8qXL29yMgAA7h1c0w3cgc8++0yPPvqogoKCdOzYMUnSpEmT9O2335qcDABuLzU1VW5ubtbnrq6urNQBACCHcZ9uwE7Tpk3T0KFDNWDAAI0aNUopKSmS/r0/5OTJk9WqVSuTEwLArRmGoW7dusnd3V2SlJiYqOeeey5d4b148WIz4gEAcE9geTlgp7Jly2rMmDFq3bq18ufPr99++01hYWHat2+f6tSpo7Nnz5odEQBuqXv37lkaN2vWLAcnAQDg3sVMN2Cn2NhYValSJV27u7u7rly5YkIiAMgeimkAAByPa7oBO4WGhmr37t3p2n/88UeVLVv27gcCAAAA4HSY6Qbs9Morr6hfv35KTEyUYRjatm2b5s+fr7Fjx+qTTz4xOx4AAAAAJ8A13cAd+PjjjzVq1CidOHFCklSsWDENHz5cPXv2NDkZAAAAAGdA0Q3kgLNnzyo1NVX+/v5mRwEAAADgRLimG7DTtWvXdPXqVUlS4cKFde3aNU2ePFkrV640ORkAAAAAZ0HRDdipVatWmjt3riQpPj5eDz/8sCZMmKBWrVpp2rRpJqcDAAAA4AwougE77dq1S4899pgkadGiRQoICNCxY8c0d+5cvf/++yanAwAAAOAMKLoBO129elX58+eXJK1cuVJt27aVi4uLHnnkER07dszkdAAAAACcAUU3YKfw8HB98803OnHihFasWKFGjRpJks6cOSMfHx+T0wEAAABwBhTdgJ2GDh2ql19+WSEhIapRo4YiIyMl/TvrXaVKFZPTAQAAAHAG3DIMuANxcXE6deqUKlWqJBeXf/+GtW3bNvn4+Kh06dImpwMAAABgNopuwAEWLVqkJ554wuwYAAAAAEzG8nLADsnJyfr999914MABm/Zvv/1WlSpVUqdOnUxKBgAAAMCZUHQD2bR//35FRESoYsWKKlOmjNq2bavTp0+rdu3aio6OVsOGDXXo0CGzYwIAAABwAiwvB7KpZcuWunLlil566SV98cUXWrhwocLDw9W5c2e99NJL1tuIAQAAAABFN5BNAQEBWrZsmapWrar4+HgVLFhQM2bM0DPPPGN2NAAAAABOhuXlQDadOXNGxYoVkyT5+fnJ09NTtWvXNjkVAAAAAGdE0Q1kk8Visd4eTJJcXFzk5uZmYiIAAAAAzorl5UA2ubi4yNfXVxaLRZIUHx8vHx8fm0Jcks6fP29GPAAAAABOJI/ZAYDcZtasWWZHAAAAAJBLUHQD2RQaGqqaNWsqTx7++QAAAAC4NZaXA9nk6uqqU6dOyd/f3+woAAAAAJwcG6kB2cTfqQAAAABkFUU3YIe0TdQAAAAA4FZYXg5kk4uLi3r37i1PT89bjps4ceJdSgQAAADAWbETFGCHvXv3Km/evJn2MxMOAAAAQGKmG8g2FxcXxcXFsZEaAAAAgNvimm4gm5jFBgAAAJBVFN1ANrE4BAAAAEBWUXQD2fTxxx/L19fX7BgAAAAAcgGKbiCbevfurYsXL1qfv/LKKzp//ryJiQAAAAA4K4puIJv+u7x8xowZio+PNycMAAAAAKdG0Q3cIa7xBgAAAJAZim4AAAAAABwkj9kBgNxo6NCh8vT0lCRdv35do0ePTre52sSJE82IBgAAAMCJWAzWxgLZUqdOndveq9tisWjt2rV3KREAAAAAZ0XRDQAAAACAg3BNN2CHhIQEpaampmtPTU1VQkKCCYkAAAAAOCOKbiCblixZourVqysxMTFdX2Jioh566CF9//33JiQDAAAA4GwouoFsmjZtml599VXrRmo38/T01GuvvaYPP/zQhGQAAAAAnA1FN5BN+/btU506dTLtf/zxx7V37967FwgAAACA06LoBrLpwoULSk5OzrT/xo0bunDhwl1MBAAAAMBZUXQD2RQSEqIdO3Zk2r9jxw6VKFHiLiYCAAAA4KwouoFsatu2rd58802dPn06XV9cXJzeeusttWvXzoRkAAAAAJwN9+kGsunSpUuKjIzU8ePH1blzZ5UqVUoWi0UxMTH64osvFBwcrC1btih//vxmRwUAAABgMopuwA4XL17U4MGDtXDhQuv12wUKFFCHDh00ZswY+fn5mRsQAAAAgFOg6AbugGEYOnv2rAzDUJEiRWSxWMyOBAAAAMCJcE03cAdSUlL022+/acmSJbp8+bIk6eTJk9avAQAAANzfmOkG7HTs2DFFRUXp+PHjSkpK0oEDBxQWFqYBAwYoMTFR06dPNzsiAAAAAJMx0w3Y6cUXX1T16tV14cIFeXh4WNvbtGmjNWvWmJgMAAAAgLPIY3YAILfatGmTfv75Z+XNm9emvUSJEvr7779NSgUAAADAmTDTDdgpNTVVKSkp6dr/+usvbhcGAAAAQBJFN2C3hg0bavLkydbnFotFly9f1rBhw9S0aVPzggEAAABwGmykBtjp5MmTqlu3rlxdXXXw4EFVr15dBw8eVOHChfXTTz/J39/f7IgAAAAATEbRDdyBa9euaf78+dq1a5dSU1NVtWpVderUyWZjNQAAAAD3L4puwE5Xr16Vp6en2TEAAAAAODGu6Qbs5O/vr86dO2vFihVKTU01Ow4AAAAAJ0TRDdhp7ty5SkpKUps2bRQUFKQXX3xR27dvNzsWAAAAACfC8nLgDl26dEmLFi3S/PnztW7dOoWGhqpz584aOnSo2dEAAAAAmIyiG8hB+/fvV6dOnbRnz54M7+ENAAAA4P7C8nLgDiUmJurLL79U69atVbVqVZ07d04vv/yy2bEAAAAAOIE8ZgcAcquVK1fqiy++0DfffCNXV1c98cQTWrFihWrXrm12NAAAAABOguXlgJ08PT3VrFkzderUSc2aNZObm5vZkQAAAAA4GYpuwE4JCQny8fExOwYAAAAAJ8byciAb/ltoJyQkZDqWghwAAAAAM91ANri6uurUqVPy9/eXi4uLLBZLujGGYchisbB7OQAAAABmuoHsWLt2rQoWLChJWrdunclpAAAAADg7ZroBAAAAAHAQ7tMN2Gn58uXatGmT9fmUKVNUuXJldezYURcuXDAxGQAAAABnQdEN2OmVV16xbqS2d+9eDRw4UE2bNtWRI0c0cOBAk9MBAAAAcAZc0w3YKTY2VmXLlpUkff3112rRooXGjBmjXbt2qWnTpianAwAAAOAMmOkG7JQ3b15dvXpVkrR69Wo1atRIklSwYMFb3koMAAAAwP2DmW7ATo8++qgGDhyoWrVqadu2bVq4cKEk6cCBA3rggQdMTgcAAADAGTDTDdjpww8/VJ48ebRo0SJNmzZNxYoVkyT9+OOPioqKMjkdAAAAAGfALcMAAAAAAHAQlpcDdyA1NVWHDh3SmTNnlJqaatP3+OOPm5QKAAAAgLOg6AbstGXLFnXs2FHHjh3TfxeMWCwWpaSkmJQMAAAAgLNgeTlgp8qVKysiIkIjRoxQYGCgLBaLTb+vr69JyQAAAAA4C4puwE5eXl767bffFB4ebnYUAAAAAE6K3csBO9WoUUOHDh0yOwYAAAAAJ8Y13YCdnn/+eQ0aNEhxcXGqUKGC3NzcbPorVqxoUjIAAAAAzoLl5YCdXFzSLxSxWCwyDION1AAAAABIYqYbsFtsbKzZEQAAAAA4OWa6AQAAAABwEGa6gTtw+PBhTZ48WTExMbJYLCpTpoxefPFFPfjgg2ZHAwAAAOAE2L0csNOKFStUtmxZbdu2TRUrVlT58uW1detWlStXTqtWrTI7HgAAAAAnwPJywE5VqlRR48aN9c4779i0v/7661q5cqV27dplUjIAAAAAzoKiG7BTvnz5tHfvXpUsWdKm/cCBA6pYsaISExNNSgYAAADAWbC8HLBTkSJFtHv37nTtu3fvlr+//90PBAAAAMDpsJEaYKdnnnlGvXv31pEjR1SzZk1ZLBZt2rRJ48aN06BBg8yOBwAAAMAJsLwcsJNhGJo8ebImTJigkydPSpKCgoL0yiuv6IUXXpDFYjE5IQAAAACzUXQDOeDSpUuSpPz585ucBAAAAIAzYXk5cIfOnDmjP//8UxaLRaVKlVKRIkXMjgQAAADASbCRGmCnhIQEdenSRUFBQapdu7Yef/xxBQUFqXPnzrp48aLZ8QAAAAA4AYpuwE69evXS1q1b9cMPPyg+Pl4XL17U0qVLtWPHDj3zzDNmxwMAAADgBLimG7CTl5eXVqxYoUcffdSmfePGjYqKitKVK1dMSgYAAADAWTDTDdipUKFC8vX1Tdfu6+urAgUKmJAIAAAAgLOh6Abs9NZbb2ngwIE6deqUtS0uLk6vvPKKhgwZYmIyAAAAAM6C5eWAnapUqaJDhw4pKSlJxYsXlyQdP35c7u7uKlmypM3YXbt2mRERAAAAgMm4ZRhgp9atW5sdAQAAAICTY6YbAAAAAAAH4ZpuwE4nTpzQX3/9ZX2+bds2DRgwQB999JGJqQAAAAA4E4puwE4dO3bUunXrJP27gVqDBg20bds2vfHGG3r77bdNTgcAAADAGVB0A3bat2+fHn74YUnSl19+qQoVKuiXX37RvHnzNHv2bHPDAQAAAHAKFN2AnW7cuCF3d3dJ0urVq9WyZUtJUunSpW1uIwYAAADg/kXRDdipXLlymj59ujZu3KhVq1YpKipKknTy5EkVKlTI5HQAAAAAnAFFN2CncePGacaMGapTp46efvppVapUSZL03XffWZedAwAAALi/ccsw4A6kpKQoISFBBQoUsLYdPXpUnp6e8vf3NzEZAAAAAGfATDdwBwzD0M6dOzVjxgxdunRJkpQ3b155enqanAwAAACAM2CmG7DTsWPHFBUVpePHjyspKUkHDhxQWFiYBgwYoMTERE2fPt3siAAAAABMxkw3YKcXX3xR1atX14ULF+Th4WFtb9OmjdasWWNiMgAAAADOIo/ZAYDcatOmTfr555+VN29em/YSJUro77//NikVAAAAAGfCTDdgp9TUVKWkpKRr/+uvv5Q/f34TEgEAAABwNhTdgJ0aNmyoyZMnW59bLBZdvnxZw4YNU9OmTc0LBgAAAMBpsJEaYKe///5b9erVk6urqw4ePKjq1avr4MGDKly4sH766SduGQYAAACAohu4E9euXdOCBQu0c+dOpaamqmrVqurUqZPNxmoAAAAA7l8U3YAdbty4oVKlSmnp0qUqW7as2XEAAAAAOCmu6Qbs4ObmpqSkJFksFrOjAAAAAHBiFN2AnZ5//nmNGzdOycnJZkcBAAAA4KRYXg7YqU2bNlqzZo28vb1VoUIFeXl52fQvXrzYpGQAAAAAnEUeswMAuZWfn5/atWtndgwAAAAAToyZbgAAAAAAHIRruoEc8M477yg+Pt7sGAAAAACcDDPdQA7w8fHR7t27FRYWZnYUAAAAAE6EmW4gB/C3KwAAAAAZoegGAAAAAMBBWF4O5IATJ04oKChIrq6uZkcBAAAA4EQouoE7dP36dZ05c0apqak27cWLFzcpEQAAAABnwX26ATsdPHhQPXr00C+//GLTbhiGLBaLUlJSTEoGAAAAwFlQdAN26tatm/LkyaOlS5cqMDBQFovF7EgAAAAAnAzLywE7eXl5aefOnSpdurTZUQAAAAA4KXYvB+xUtmxZnT171uwYAAAAAJwYRTdgp3HjxunVV1/V+vXrde7cOSUkJNg8AAAAAIDl5YCdXFz+/ZvVf6/lZiM1AAAAAGnYSA2w07p168yOAAAAAMDJMdMNAAAAAICDMNMN3IH4+Hh9+umniomJkcViUdmyZdWjRw/5+vqaHQ0AAACAE2CmG7DTjh071LhxY3l4eOjhhx+WYRjasWOHrl27ppUrV6pq1apmRwQAAABgMopuwE6PPfaYwsPD9fHHHytPnn8XjSQnJ6tXr146cuSIfvrpJ5MTAgAAADAbRTdgJw8PD/36668qXbq0Tfv+/ftVvXp1Xb161aRkAAAAAJwF9+kG7OTj46Pjx4+naz9x4oTy589vQiIAAAAAzoaiG7BThw4d1LNnTy1cuFAnTpzQX3/9pQULFqhXr156+umnzY4HAAAAwAmwezlgp/fee08Wi0Vdu3ZVcnKyJMnNzU19+vTRO++8Y3I6AAAAAM6Aa7qBO3T16lUdPnxYhmEoPDxcnp6eZkcCAAAA4CQougEAAAAAcBCWlwN2unLlit555x2tWbNGZ86cUWpqqk3/kSNHTEoGAAAAwFlQdAN26tWrlzZs2KAuXbooMDBQFovF7EgAAAAAnAzLywE7+fn56YcfflCtWrXMjgIAAADASXHLMMBOBQoUUMGCBc2OAQAAAMCJUXQDdho5cqSGDh2qq1evmh0FAAAAgJNieTlgpypVqlhvFRYSEiI3Nzeb/l27dpmUDAAAAICzYCM1wE6tW7c2OwIAAAAAJ0fRDdghOTlZktSjRw8FBwebnAYAAACAs2J5OWCn/Pnza+/evQoJCTE7CgAAAAAnxUZqgJ3q16+v9evXmx0DAAAAgBNjeTlgpyZNmmjw4MHat2+fqlWrJi8vL5v+li1bmpQMAAAAgLNgeTlgJxeXzBeKWCwWpaSk3MU0AAAAAJwRRTcAAAAAAA7CNd0AAAAAADgIRTdwBzZs2KAWLVooPDxcJUuWVMuWLbVx40azYwEAAABwEhTdgJ0+//xzNWjQQJ6ennrhhRfUv39/eXh4qH79+po3b57Z8QAAAAA4Aa7pBuxUpkwZ9e7dWy+99JJN+8SJE/Xxxx8rJibGpGQAAAAAnAVFN2And3d3/f777woPD7dpP3TokMqXL6/ExESTkgEAAABwFiwvB+wUHBysNWvWpGtfs2aNgoODTUgEAAAAwNnkMTsAkFsNGjRIL7zwgnbv3q2aNWvKYrFo06ZNmj17tv73v/+ZHQ8AAACAE2B5OXAHlixZogkTJliv3y5TpoxeeeUVtWrVyuRkAAAAAJwBRTcAAAAAAA7CNd2AncLCwnTu3Ll07fHx8QoLCzMhEQAAAABnQ9EN2Ono0aNKSUlJ156UlKS///7bhEQAAAAAnA0bqQHZ9N1331m/XrFihXx9fa3PU1JStGbNGoWEhJiQDAAAAICz4ZpuIJtcXDJfIOLm5qaQkBBNmDBBzZs3v4upAAAAADgjZrqBbNizZ49u3LghV1dXhYaGavv27SpcuLDZsQAAAAA4Ka7pBrKhSpUqOn/+vCTJYrHIYrGYnAgAAACAM6PoBrLBz89PR44ckSQdO3ZMqampJicCAAAA4MxYXg5kQ7t27VS7dm0FBgZKkqpXry5XV9cMx6YV5wAAAADuXxTdQDZ89NFHatu2rQ4dOqQXXnhBzzzzjPLnz292LAAAAABOit3LATt1795d77//PkU3AAAAgExRdAMAAAAA4CAsLwfuwPbt2/XVV1/p+PHjun79uk3f4sWLTUoFAAAAwFmwezlgpwULFqhWrVrav3+/lixZohs3bmj//v1au3atfH19zY4HAAAAwAlQdAN2GjNmjCZNmqSlS5cqb968+t///qeYmBi1b99exYsXNzseAAAAACdA0Q3Y6fDhw2rWrJkkyd3dXVeuXJHFYtFLL72kjz76yOR0AAAAAJwBRTdgp4IFC+rSpUuSpGLFimnfvn2SpPj4eF29etXMaAAAAACcBBupAXZ67LHHtGrVKlWoUEHt27fXiy++qLVr12rVqlWqX7++2fEAAAAAOAFuGQbY6fz580pMTFRQUJBSU1P13nvvadOmTQoPD9eQIUNUoEABsyMCAAAAMBlFNwAAAAAADsI13YCdli1bphUrVqRrX7lypX788UcTEgEAAABwNhTdgJ1ef/11paSkpGtPTU3V66+/bkIiAAAAAM6Gohuw08GDB1W2bNl07aVLl9ahQ4dMSAQAAADA2VB0A3by9fXVkSNH0rUfOnRIXl5eJiQCAAAA4GwougE7tWzZUgMGDNDhw4etbYcOHdKgQYPUsmVLE5MBAAAAcBbsXg7Y6eLFi4qKitKOHTv0wAMPSJL++usvPfbYY1q8eLH8/PzMDQgAAADAdBTdwB0wDEOrVq3Sb7/9Jg8PD1WsWFGPP/642bEAAAAAOAmKbgAAAAAAHCSP2QGA3OzKlSvasGGDjh8/ruvXr9v0vfDCCyalAgAAAOAsmOkG7PTrr7+qadOmunr1qq5cuaKCBQvq7Nmz8vT0lL+/f4Y7mwMAAAC4v7B7OWCnl156SS1atND58+fl4eGhLVu26NixY6pWrZree+89s+MBAAAAcALMdAN28vPz09atW1WqVCn5+flp8+bNKlOmjLZu3aro6Gj98ccfZkcEAAAAYDJmugE7ubm5yWKxSJKKFi2q48ePS5J8fX2tXwMAAAC4v7GRGmCnKlWqaMeOHYqIiFDdunU1dOhQnT17Vp999pkqVKhgdjwAAAAAToDl5YCdduzYoUuXLqlu3br6559/FB0drU2bNik8PFyzZs1SpUqVzI4IAAAAwGQU3QAAAAAAOAjXdAN2GjFihA4fPmx2DAAAAABOjKIbsNPXX3+tiIgIPfLII/rwww/1zz//mB0JAAAAgJOh6AbstGfPHu3Zs0f16tXTxIkTVaxYMTVt2lTz5s3T1atXzY4HAAAAwAlwTTeQQ37++WfNmzdPX331lRITE5WQkGB2JAAAAAAmY6YbyCFeXl7y8PBQ3rx5dePGDbPjAAAAAHACFN3AHYiNjdXo0aNVtmxZVa9eXbt27dLw4cMVFxdndjQAAAAAToDl5YCdIiMjtW3bNlWoUEGdOnVSx44dVaxYMbNjAQAAAHAiecwOAORWdevW1SeffKJy5cqZHQUAAACAk6LoBrJh4MCBGjlypLy8vHTt2jV9+umnmY6dOHHiXUwGAAAAwBlRdAPZ8Ouvv1o3Sdu9e3em4ywWy11KBAAAAMCZcU03AAAAAAAOwu7lAAAAAAA4CEU3AAAAAAAOQtENAAAAAICDUHQDAAAAAOAgFN0AAOSwkJAQTZ482ewY96X169fLYrEoPj5ekjR79mz5+fll6xgWi0XffPNNjmcDANyfKLoBAPeFbt26qXXr1una/1ukmWH48OGqXLmyaee/lRkzZqhSpUry8vKSn5+fqlSponHjxln7M/tcHemXX36Rq6uroqKi7up5AQCwB/fpBgAAGfr00081cOBAvf/++6pdu7aSkpK0Z88e7d+/P8fPdePGDbm5uWVp7MyZM/X888/rk08+0fHjx1W8ePEczwMAQE5hphsAgP9z7tw5Pf3003rggQfk6empChUqaP78+TZj6tSpo/79+6t///7y8/NToUKF9NZbb8kwjEyPe/z4cbVq1Ure3t7y8fFR+/btdfr0aUn/Ln8eMWKEfvvtN1ksFlksFs2ePVs9evRQ8+bNbY6TnJysgIAAzZw5M8tZrl+/rldffVXFihWTl5eXatSoofXr12fp8/j+++/Vvn179ezZU+Hh4SpXrpyefvppjRw5UtK/M/Rz5szRt99+a82eduzXXntNERER8vT0VFhYmIYMGaIbN25Yj502uz9z5kyFhYXJ3d39lp9hmitXrujLL79Unz591Lx5c82ePTtL7+W/76tatWrKly+fwsLCNGLECCUnJ2c6/u+//1aHDh1UoEABFSpUSK1atdLRo0ezfV4AwP2JohsAgP+TmJioatWqaenSpdq3b5969+6tLl26aOvWrTbj5syZozx58mjr1q16//33NWnSJH3yyScZHtMwDLVu3Vrnz5/Xhg0btGrVKh0+fFgdOnSQJHXo0EGDBg1SuXLldOrUKZ06dUodOnRQr169tHz5cp06dcp6rGXLluny5ctq3759lrN0795dP//8sxYsWKA9e/boySefVFRUlA4ePHjbzyMgIEBbtmzRsWPHMux/+eWX1b59e0VFRVmz16xZU5KUP39+zZ49W/v379f//vc/ffzxx5o0aZLN6w8dOqQvv/xSX3/9tXbv3n3bPJK0cOFClSpVSqVKlVLnzp01a9asLBXraVasWKHOnTvrhRde0P79+zVjxgzNnj1bo0ePznD81atXVbduXXl7e+unn37Spk2b5O3traioKF2/fj3L5wUA3McMAADuA9HR0Yarq6vh5eVl88iXL58hybhw4UKGr2vatKkxaNAg6/PatWsbZcqUMVJTU61tr732mlGmTBnr8xIlShiTJk0yDMMwVq5cabi6uhrHjx+39v/++++GJGPbtm2GYRjGsGHDjEqVKqU7d9myZY1x48ZZn7du3dro1q1blrMcOnTIsFgsxt9//21z3Pr16xuDBw/O7KOyOnnypPHII48YkoyIiAgjOjraWLhwoZGSkmIdEx0dbbRq1eq2xxo/frxRrVo16/Nhw4YZbm5uxpkzZ2772pvVrFnTmDx5smEYhnHjxg2jcOHCxqpVq6z969ats/l+zpo1y/D19bX2P/bYY8aYMWNsjvnZZ58ZgYGB1ueSjCVLlhiGYRiffvqpUapUKZvPOCkpyfDw8DBWrFiRrewAgPsTM90AgPtG3bp1tXv3bpvHzbPCKSkpGj16tCpWrKhChQrJ29tbK1eu1PHjx22O88gjj8hisVifR0ZG6uDBg0pJSUl3zpiYGAUHBys4ONjaVrZsWfn5+SkmJuaWeXv16qVZs2ZJks6cOaMffvhBPXr0yHKWXbt2yTAMRUREyNvb2/rYsGGDDh8+fNvPKzAwUJs3b9bevXv1wgsv6MaNG4qOjlZUVJRSU1Nv+dpFixbp0UcfVUBAgLy9vTVkyJB0n2OJEiVUpEiR2+ZI8+eff2rbtm166qmnJEl58uRRhw4drMvts2Lnzp16++23bT6PZ555RqdOndLVq1czHH/o0CHlz5/fOr5gwYJKTEzM0mcIAAAbqQEA7hteXl4KDw+3afvrr7+sX0+YMEGTJk3S5MmTVaFCBXl5eWnAgAF3tIzYMAybovh27Tfr2rWrXn/9dW3evFmbN29WSEiIHnvssSyfOzU1Va6urtq5c6dcXV1t+ry9vbN8nPLly6t8+fLq16+fNm3apMcee0wbNmxQ3bp1Mxy/ZcsWPfXUUxoxYoQaN24sX19fLViwQBMmTLAZ5+XlleUM0r8buyUnJ6tYsWLWNsMw5ObmpgsXLqhAgQK3PUZqaqpGjBihtm3bpuvLly9fhuOrVaumL774Il1fdv5gAAC4f1F0AwDwfzZu3KhWrVqpc+fOkv4tuA4ePKgyZcrYjNuyZUu65yVLlkxX2Er/zmofP35cJ06csM5279+/XxcvXrQeN2/evBnOkhcqVEitW7fWrFmztHnzZnXv3j3dmFtlqVKlilJSUnTmzJlsFeu3UrZsWUn/bmiWWfaff/5ZJUqU0Jtvvmlty+y68KxKTk7W3LlzNWHCBDVq1Mimr127dvriiy/Uv3//2x6natWq+vPPP9P98eVW4xcuXCh/f3/5+PjYlR0AcH9jeTkAAP8nPDxcq1at0i+//KKYmBg9++yziouLSzfuxIkTGjhwoP7880/Nnz9fH3zwgV588cUMj9mgQQNVrFhRnTp10q5du7Rt2zZ17dpVtWvXVvXq1SVJISEhio2N1e7du3X27FklJSVZX9+rVy/NmTNHMTExio6OzlaWiIgIderUSV27dtXixYsVGxur7du3a9y4cVq2bNltP48+ffpo5MiR+vnnn3Xs2DFt2bJFXbt2VZEiRRQZGWnNvmfPHv355586e/asbty4ofDwcB0/flwLFizQ4cOH9f7772vJkiW3/wbcwtKlS3XhwgX17NnTOvOe9njiiSf06aefZuk4Q4cO1dy5czV8+HD9/vvviomJ0cKFC/XWW29lOL5Tp04qXLiwWrVqpY0bNyo2NlYbNmzQiy++aLNKAgCAzFB0AwDwf4YMGaKqVauqcePGqlOnjgICAtS6det047p27apr167p4YcfVr9+/fT888+rd+/eGR7TYrHom2++UYECBfT444+rQYMGCgsL08KFC61j2rVrp6ioKNWtW1dFihSxuU1ZgwYNFBgYqMaNGysoKCjbWWbNmqWuXbtq0KBBKlWqlFq2bKmtW7faXGOemQYNGmjLli168sknFRERoXbt2inf/2vv7lEai8IwAL8DFiKBQIKNAbULdikMWKYQJI2C7kBIl8YF2KYzuAGzgNR2YiPEUlyCBARXYJqg084Qf8bBm3HweeBW3+V+X/ueew5ncTGXl5epVqtJkk6nk3q9ns3NzSwvL2c0GmVvby9HR0fpdrtpNBq5vr7O8fHxu/3ecnZ2lu3t7ZTL5ZnawcFBbm9vc3Nz8+53dnZ2cn5+nouLizSbzWxtbaXf72dtbe3F95eWlnJ1dZXV1dXs7+9nY2Mjh4eHmUwm/nwD8Ed+PD9/4J4NAPjmWq1WGo1GTk9P59Lv8fExKysrGQwGM+eQ5z0LAPBxznQDwBf09PSUh4eHnJycpFwuZ3d391+PBAD8BdvLAeALGo/HqdVqGQ6HGQwGWVj4/HXydrv929VZvz69Xu/T+71lPB6/OkupVJq5bgwA/he2lwPAN3V/f5/JZPJirVKppFKpzG2W6XSau7u7V+vr6+uFLDwAQNGEbgAAACiI7eUAAABQEKEbAAAACiJ0AwAAQEGEbgAAACiI0A0AAAAFEboBAACgIEI3AAAAFEToBgAAgIL8BMZmZqElbO/fAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1000x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Your data\n",
    "data = {\n",
    "    'Haplotype_Star_Allele': ['ivacaftornon-responsiveCFTRsequence', 'R1070Q', 'Others'],\n",
    "    'Count': [15, 10, 5],\n",
    "    'Male_HOM': [0, 0, 0],\n",
    "    'Male_HET': [6, 4, 2],\n",
    "    'Female_HOM': [2, 2, 0],\n",
    "    'Female_HET': [7, 4, 3]\n",
    "}\n",
    "\n",
    "skd_data = pd.DataFrame(data)\n",
    "\n",
    "# Set 'Haplotype_Star_Allele' column as index\n",
    "skd_data.set_index('Haplotype_Star_Allele', inplace=True)\n",
    "\n",
    "# Plot the stacked bar graph\n",
    "ax = skd_data.plot(kind='bar', stacked=True, figsize=(10, 6))\n",
    "ax.set_ylabel('Count')\n",
    "ax.set_title('Stacked Bar Graph of Haplotype Counts')\n",
    "\n",
    "# Show the plot\n",
    "plt.tight_layout()\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5cadc0c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Sample_Name</th>\n",
       "      <th>EXON</th>\n",
       "      <th>IMPACT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs3842753</td>\n",
       "      <td>HET</td>\n",
       "      <td>INS</td>\n",
       "      <td>12652700</td>\n",
       "      <td>3/3</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs3842753</td>\n",
       "      <td>HET</td>\n",
       "      <td>INS</td>\n",
       "      <td>12652700</td>\n",
       "      <td>2/3</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3842753</td>\n",
       "      <td>HET</td>\n",
       "      <td>DEL</td>\n",
       "      <td>12652700</td>\n",
       "      <td>1/3</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1234567</td>\n",
       "      <td>HOM</td>\n",
       "      <td>DEL</td>\n",
       "      <td>45678901</td>\n",
       "      <td>2/5</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1234567</td>\n",
       "      <td>HOM</td>\n",
       "      <td>DEL</td>\n",
       "      <td>45678901</td>\n",
       "      <td>4/5</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID Zygosity Matched_Gene  Sample_Name EXON    IMPACT\n",
       "0  rs3842753      HET          INS     12652700  3/3  MODIFIER\n",
       "1  rs3842753      HET          INS     12652700  2/3  MODIFIER\n",
       "2  rs3842753      HET          DEL     12652700  1/3      HIGH\n",
       "3  rs1234567      HOM          DEL     45678901  2/5  MODERATE\n",
       "4  rs1234567      HOM          DEL     45678901  4/5  MODERATE"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming your data is stored in a DataFrame called 'df'\n",
    "# Replace this with your actual data loading method\n",
    "\n",
    "# Sample data\n",
    "data = {\n",
    "    'ID': ['rs3842753', 'rs3842753', 'rs3842753', 'rs1234567', 'rs1234567'],\n",
    "    'Zygosity': ['HET', 'HET', 'HET', 'HOM', 'HOM'],\n",
    "    'Matched_Gene': ['INS', 'INS', 'DEL', 'DEL', 'DEL'],\n",
    "    'Sample_Name': [12652700, 12652700, 12652700, 45678901, 45678901],\n",
    "    'EXON': ['3/3', '2/3', '1/3', '2/5', '4/5'],\n",
    "    'IMPACT': ['MODIFIER', 'MODIFIER', 'HIGH', 'MODERATE', 'MODERATE']\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ffb0b3f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Sample_Name</th>\n",
       "      <th>EXON</th>\n",
       "      <th>IMPACT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs3842753</td>\n",
       "      <td>HET</td>\n",
       "      <td>INS</td>\n",
       "      <td>12652700</td>\n",
       "      <td>3/3</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1234567</td>\n",
       "      <td>HOM</td>\n",
       "      <td>DEL</td>\n",
       "      <td>45678901</td>\n",
       "      <td>2/5</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID Zygosity Matched_Gene  Sample_Name EXON    IMPACT\n",
       "0  rs3842753      HET          INS     12652700  3/3  MODIFIER\n",
       "3  rs1234567      HOM          DEL     45678901  2/5  MODERATE"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Convert the denominator part of EXON to integer for comparison\n",
    "df['EXON_Denominator'] = df['EXON'].apply(lambda x: int(x.split('/')[1]))\n",
    "\n",
    "# Sort the DataFrame by Sample_Name and EXON_Denominator in descending order\n",
    "df.sort_values(by=['Sample_Name', 'EXON_Denominator'], ascending=[True, False], inplace=True)\n",
    "\n",
    "# Drop duplicates based on Sample_Name, keeping the first (highest) EXON value\n",
    "df.drop_duplicates(subset='Sample_Name', keep='first', inplace=True)\n",
    "\n",
    "# Drop the temporary EXON_Denominator column\n",
    "df.drop(columns=['EXON_Denominator'], inplace=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "71c40223",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>IMPACT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sample1</td>\n",
       "      <td>rs123</td>\n",
       "      <td>GeneA</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sample1</td>\n",
       "      <td>rs123</td>\n",
       "      <td>GeneA</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs456</td>\n",
       "      <td>GeneB</td>\n",
       "      <td>LOW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs789</td>\n",
       "      <td>GeneC</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs789</td>\n",
       "      <td>GeneC</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample   rsID   Gene    IMPACT\n",
       "0  sample1  rs123  GeneA  MODERATE\n",
       "1  sample1  rs123  GeneA      HIGH\n",
       "2  sample2  rs456  GeneB       LOW\n",
       "3  sample2  rs789  GeneC  MODERATE\n",
       "4  sample2  rs789  GeneC  MODIFIER"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample data\n",
    "data = {\n",
    "    'sample': ['sample1', 'sample1', 'sample2', 'sample2', 'sample2'],\n",
    "    'rsID': ['rs123', 'rs123', 'rs456', 'rs789', 'rs789'],\n",
    "    'Gene': ['GeneA', 'GeneA', 'GeneB', 'GeneC', 'GeneC'],\n",
    "    'IMPACT': ['MODERATE', 'HIGH', 'LOW', 'MODERATE', 'MODIFIER']\n",
    "}\n",
    "\n",
    "# Create a DataFrame\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "1c495cd2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>IMPACT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sample1</td>\n",
       "      <td>rs123</td>\n",
       "      <td>GeneA</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs456</td>\n",
       "      <td>GeneB</td>\n",
       "      <td>LOW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs789</td>\n",
       "      <td>GeneC</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample   rsID   Gene    IMPACT\n",
       "1  sample1  rs123  GeneA      HIGH\n",
       "2  sample2  rs456  GeneB       LOW\n",
       "3  sample2  rs789  GeneC  MODERATE"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the priority order\n",
    "impact_order = ['HIGH', 'MODERATE', 'LOW', 'MODIFIER']\n",
    "\n",
    "# Sort the DataFrame based on the priority\n",
    "df['IMPACT'] = pd.Categorical(df['IMPACT'], categories=impact_order, ordered=True)\n",
    "df.sort_values(['sample', 'rsID', 'Gene', 'IMPACT'], ascending=[True, True, True, True], inplace=True)\n",
    "\n",
    "# Drop duplicates while keeping the first occurrence\n",
    "filtered_df = df.drop_duplicates(subset=['sample', 'rsID', 'Gene'], keep='first')\n",
    "\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "014ae235",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7edf337a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "345ac4a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>IMPACT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sample1</td>\n",
       "      <td>rs123</td>\n",
       "      <td>GeneA</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sample1</td>\n",
       "      <td>rs123</td>\n",
       "      <td>GeneA</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs456</td>\n",
       "      <td>GeneB</td>\n",
       "      <td>LOW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs789</td>\n",
       "      <td>GeneC</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sample2</td>\n",
       "      <td>rs789</td>\n",
       "      <td>GeneC</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample   rsID   Gene    IMPACT\n",
       "0  sample1  rs123  GeneA  MODERATE\n",
       "1  sample1  rs123  GeneA      HIGH\n",
       "2  sample2  rs456  GeneB       LOW\n",
       "4  sample2  rs789  GeneC  MODIFIER\n",
       "3  sample2  rs789  GeneC  MODERATE"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the priority order\n",
    "impact_order = ['HIGH', 'MODERATE', 'LOW', 'MODIFIER']\n",
    "\n",
    "# Sort the DataFrame based on the priority\n",
    "df['IMPACT'] = pd.Categorical(df['IMPACT'], categories=impact_order, ordered=True)\n",
    "df.sort_values(['sample', 'rsID', 'Gene', 'IMPACT'], ascending=[True, True, True, False], inplace=True)\n",
    "df\n",
    "# Drop duplicates while keeping the first occurrence\n",
    "#filtered_df = df.drop_duplicates(subset=['sample', 'rsID', 'Gene'], keep='first')\n",
    "\n",
    "#filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4c1df416",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "      <th>Concatenated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>A,B,C</td>\n",
       "      <td>A,B,C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>D,E</td>\n",
       "      <td>X,Y,Z</td>\n",
       "      <td>D,E,X,Y,Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>F</td>\n",
       "      <td>P,Q,R</td>\n",
       "      <td>F,P,Q,R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G,H,I,J</td>\n",
       "      <td>K,L</td>\n",
       "      <td>G,H,I,J,K,L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Column1 Column2 Concatenated\n",
       "0        A   A,B,C        A,B,C\n",
       "1      D,E   X,Y,Z    D,E,X,Y,Z\n",
       "2        F   P,Q,R      F,P,Q,R\n",
       "3  G,H,I,J     K,L  G,H,I,J,K,L"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a sample DataFrame\n",
    "data = {'Column1': ['A', 'D,E', 'F', 'G,H,I,J'],\n",
    "        'Column2': ['A,B,C', 'X,Y,Z', 'P,Q,R', 'K,L']}\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Concatenate the two columns with commas\n",
    "df['Concatenated'] = df['Column1'] + ',' + df['Column2']\n",
    "\n",
    "# Split the concatenated values into sets of values in a new column and sort them\n",
    "df['Concatenated'] = df['Concatenated'].str.split(',').apply(lambda x: ','.join(sorted(set(x))))\n",
    "\n",
    "# Display the updated DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "7fbdc899",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCC8,BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS,BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCD4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         genes\n",
       "0  ABCC8,BRCA1\n",
       "1    INS,BRCA2\n",
       "2        BRCA2\n",
       "3        ABCD4"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a sample DataFrame\n",
    "data = {'genes': ['ABCC8,BRCA1', 'INS,BRCA2', ' ,BRCA2', 'ABCD4']}\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Replace rows that start with a space with an empty string\n",
    "df['genes'] = df['genes'].str.replace(r'^\\s+,', '', regex=True)\n",
    "\n",
    "# Display the updated DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe3a7471",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3d93683",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "1e30ed0b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>31.03%</td>\n",
       "      <td>9.404800e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0500</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>31.03%</td>\n",
       "      <td>9.404800e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0500</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>31.03%</td>\n",
       "      <td>9.404800e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0500</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>94</td>\n",
       "      <td>95</td>\n",
       "      <td>95</td>\n",
       "      <td>67</td>\n",
       "      <td>28</td>\n",
       "      <td>29.47%</td>\n",
       "      <td>3.598000e-10</td>\n",
       "      <td>63</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.62910</td>\n",
       "      <td>0.35440</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.69450</td>\n",
       "      <td>0.80330</td>\n",
       "      <td>0.996100</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.87980</td>\n",
       "      <td>0.60870</td>\n",
       "      <td>0.88550</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>94</td>\n",
       "      <td>95</td>\n",
       "      <td>95</td>\n",
       "      <td>67</td>\n",
       "      <td>28</td>\n",
       "      <td>29.47%</td>\n",
       "      <td>3.598000e-10</td>\n",
       "      <td>63</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.3591</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8482</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>0.8821</td>\n",
       "      <td>0.9146</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>0.62910</td>\n",
       "      <td>0.35440</td>\n",
       "      <td>0.8158</td>\n",
       "      <td>0.69450</td>\n",
       "      <td>0.80330</td>\n",
       "      <td>0.996100</td>\n",
       "      <td>0.9294</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>0.87980</td>\n",
       "      <td>0.60870</td>\n",
       "      <td>0.88550</td>\n",
       "      <td>0.9984</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>TF binding site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315111</th>\n",
       "      <td>RAB39B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs3105275</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155259147</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000369454.4</td>\n",
       "      <td>c.*1656T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.594300e-31</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5960</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>0.6202</td>\n",
       "      <td>0.7251</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>0.7201</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.61610</td>\n",
       "      <td>0.29760</td>\n",
       "      <td>0.6866</td>\n",
       "      <td>0.63900</td>\n",
       "      <td>0.76200</td>\n",
       "      <td>0.721900</td>\n",
       "      <td>0.7682</td>\n",
       "      <td>0.7881</td>\n",
       "      <td>0.75770</td>\n",
       "      <td>0.62840</td>\n",
       "      <td>0.72010</td>\n",
       "      <td>0.8029</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315112</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.752600e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.6450</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66160</td>\n",
       "      <td>0.44990</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.66390</td>\n",
       "      <td>0.76190</td>\n",
       "      <td>0.725400</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.75800</td>\n",
       "      <td>0.67750</td>\n",
       "      <td>0.71760</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315113</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000369449.7</td>\n",
       "      <td>c.*19C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.752600e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.6450</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66160</td>\n",
       "      <td>0.44990</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.66390</td>\n",
       "      <td>0.76190</td>\n",
       "      <td>0.725400</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.75800</td>\n",
       "      <td>0.67750</td>\n",
       "      <td>0.71760</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315114</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.752600e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.6450</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66160</td>\n",
       "      <td>0.44990</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.66390</td>\n",
       "      <td>0.76190</td>\n",
       "      <td>0.725400</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.75800</td>\n",
       "      <td>0.67750</td>\n",
       "      <td>0.71760</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315115</th>\n",
       "      <td>CLIC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.752600e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6458</td>\n",
       "      <td>0.4108</td>\n",
       "      <td>0.6450</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>0.7173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66160</td>\n",
       "      <td>0.44990</td>\n",
       "      <td>0.6906</td>\n",
       "      <td>0.66390</td>\n",
       "      <td>0.76190</td>\n",
       "      <td>0.725400</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.7778</td>\n",
       "      <td>0.75800</td>\n",
       "      <td>0.67750</td>\n",
       "      <td>0.71760</td>\n",
       "      <td>0.8055</td>\n",
       "      <td>EUR</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>315116 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene Gene_Score Condition Headings 21_Conditions_list  \\\n",
       "0            NaN  NaN         No        No       No                 No   \n",
       "1            NaN  NaN         No        No       No                 No   \n",
       "2            NaN  NaN         No        No       No                 No   \n",
       "3          OR4F5  NaN         No        No       No                 No   \n",
       "4          OR4F5  NaN         No        No       No                 No   \n",
       "...          ...  ...        ...       ...      ...                ...   \n",
       "315111    RAB39B  NaN         No        No       No                 No   \n",
       "315112     CLIC2  NaN         No        No       No                 No   \n",
       "315113     CLIC2  NaN         No        No       No                 No   \n",
       "315114     CLIC2  NaN         No        No       No                 No   \n",
       "315115     CLIC2  NaN         No        No       No                 No   \n",
       "\n",
       "                rsID Literature CHROM        POS REF ALT      Zygosity  \\\n",
       "0       rs1416222198         No  chr1      65591   C   T  Heterozygous   \n",
       "1       rs1416222198         No  chr1      65591   C   T  Heterozygous   \n",
       "2       rs1416222198         No  chr1      65591   C   T  Heterozygous   \n",
       "3        rs201219564         No  chr1      69270   A   G  Heterozygous   \n",
       "4        rs201219564         No  chr1      69270   A   G  Heterozygous   \n",
       "...              ...        ...   ...        ...  ..  ..           ...   \n",
       "315111     rs3105275         No  chrX  155259147   A   G    Homozygous   \n",
       "315112      rs559165         No  chrX  155277884   G   T    Homozygous   \n",
       "315113      rs559165         No  chrX  155277884   G   T    Homozygous   \n",
       "315114      rs559165         No  chrX  155277884   G   T    Homozygous   \n",
       "315115      rs559165         No  chrX  155277884   G   T    Homozygous   \n",
       "\n",
       "                      Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1                  intron variant              2/10  MODIFIER           1.5   \n",
       "2         downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3              synonymous variant              3/10       LOW           2.5   \n",
       "4       regulatory region variant              2/10  MODIFIER           1.5   \n",
       "...                           ...               ...       ...           ...   \n",
       "315111        3 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "315112    downstream gene variant              2/10  MODIFIER           1.5   \n",
       "315113        3 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "315114    downstream gene variant              2/10  MODIFIER           1.5   \n",
       "315115    downstream gene variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "0                NaN      NaN                NaN      NaN                NaN   \n",
       "1                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "2                NaN      NaN                NaN      NaN                NaN   \n",
       "3                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "4                NaN      NaN                NaN      NaN                NaN   \n",
       "...              ...      ...                ...      ...                ...   \n",
       "315111           NaN   benign                NaN      NaN  ENST00000369454.4   \n",
       "315112           NaN   benign                NaN      NaN                NaN   \n",
       "315113           NaN   benign                NaN      NaN  ENST00000369449.7   \n",
       "315114           NaN   benign                NaN      NaN                NaN   \n",
       "315115           NaN   benign                NaN      NaN                NaN   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                    None                NaN               None  0/1   30   \n",
       "1               c.9+18C>T                NaN               None  0/1   30   \n",
       "2                    None                NaN               None  0/1   30   \n",
       "3                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1   94   \n",
       "4                    None                NaN               None  0/1   94   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "315111         c.*1656T>C                NaN               None  1/1  255   \n",
       "315112               None                NaN               None  1/1  123   \n",
       "315113           c.*19C>A                NaN               None  1/1  123   \n",
       "315114               None                NaN               None  1/1  123   \n",
       "315115               None                NaN               None  1/1  123   \n",
       "\n",
       "        SDP  DP  RD  AD    FREQ          PVAL  RDF  RDR  ADF  ADR SIFT  \\\n",
       "0        29  29  20   9  31.03%  9.404800e-04   11    9    7    2  NaN   \n",
       "1        29  29  20   9  31.03%  9.404800e-04   11    9    7    2  NaN   \n",
       "2        29  29  20   9  31.03%  9.404800e-04   11    9    7    2  NaN   \n",
       "3        95  95  67  28  29.47%  3.598000e-10   63    4   22    6  NaN   \n",
       "4        95  95  67  28  29.47%  3.598000e-10   63    4   22    6  NaN   \n",
       "...     ...  ..  ..  ..     ...           ...  ...  ...  ...  ...  ...   \n",
       "315111   53  53   0  53    100%  1.594300e-31    0    0   38   15  NaN   \n",
       "315112   22  22   0  22    100%  4.752600e-13    0    0   19    3  NaN   \n",
       "315113   22  22   0  22    100%  4.752600e-13    0    0   19    3  NaN   \n",
       "315114   22  22   0  22    100%  4.752600e-13    0    0   19    3  NaN   \n",
       "315115   22  22   0  22    100%  4.752600e-13    0    0   19    3  NaN   \n",
       "\n",
       "       PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  \\\n",
       "0           NaN     NaN     NaN     NaN     NaN     NaN     NaN         NaN   \n",
       "1           NaN     NaN     NaN     NaN     NaN     NaN     NaN         NaN   \n",
       "2           NaN     NaN     NaN     NaN     NaN     NaN     NaN         NaN   \n",
       "3           NaN     NaN     NaN     NaN     NaN     NaN     NaN       0.838   \n",
       "4           NaN     NaN     NaN     NaN     NaN     NaN     NaN       0.838   \n",
       "...         ...     ...     ...     ...     ...     ...     ...         ...   \n",
       "315111      NaN  0.5960  0.2383  0.6202  0.7251  0.8029  0.7201         NaN   \n",
       "315112      NaN  0.6458  0.4108  0.6450  0.7277  0.8055  0.7173         NaN   \n",
       "315113      NaN  0.6458  0.4108  0.6450  0.7277  0.8055  0.7173         NaN   \n",
       "315114      NaN  0.6458  0.4108  0.6450  0.7277  0.8055  0.7173         NaN   \n",
       "315115      NaN  0.6458  0.4108  0.6450  0.7277  0.8055  0.7173         NaN   \n",
       "\n",
       "        gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  \\\n",
       "0                  NaN             NaN             NaN             NaN   \n",
       "1                  NaN             NaN             NaN             NaN   \n",
       "2                  NaN             NaN             NaN             NaN   \n",
       "3               0.3591          0.7932          0.8482          0.9984   \n",
       "4               0.3591          0.7932          0.8482          0.9984   \n",
       "...                ...             ...             ...             ...   \n",
       "315111             NaN             NaN             NaN             NaN   \n",
       "315112             NaN             NaN             NaN             NaN   \n",
       "315113             NaN             NaN             NaN             NaN   \n",
       "315114             NaN             NaN             NaN             NaN   \n",
       "315115             NaN             NaN             NaN             NaN   \n",
       "\n",
       "        gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  \\\n",
       "0                  NaN             NaN             NaN             NaN   \n",
       "1                  NaN             NaN             NaN             NaN   \n",
       "2                  NaN             NaN             NaN             NaN   \n",
       "3               0.8821          0.9146           0.868          0.9005   \n",
       "4               0.8821          0.9146           0.868          0.9005   \n",
       "...                ...             ...             ...             ...   \n",
       "315111             NaN             NaN             NaN             NaN   \n",
       "315112             NaN             NaN             NaN             NaN   \n",
       "315113             NaN             NaN             NaN             NaN   \n",
       "315114             NaN             NaN             NaN             NaN   \n",
       "315115             NaN             NaN             NaN             NaN   \n",
       "\n",
       "        gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "0          0.03416         0.02306          0.1667         0.02711   \n",
       "1          0.03416         0.02306          0.1667         0.02711   \n",
       "2          0.03416         0.02306          0.1667         0.02711   \n",
       "3          0.62910         0.35440          0.8158         0.69450   \n",
       "4          0.62910         0.35440          0.8158         0.69450   \n",
       "...            ...             ...             ...             ...   \n",
       "315111     0.61610         0.29760          0.6866         0.63900   \n",
       "315112     0.66160         0.44990          0.6906         0.66390   \n",
       "315113     0.66160         0.44990          0.6906         0.66390   \n",
       "315114     0.66160         0.44990          0.6906         0.66390   \n",
       "315115     0.66160         0.44990          0.6906         0.66390   \n",
       "\n",
       "        gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "0              0.03333        0.001639          0.0000          0.0500   \n",
       "1              0.03333        0.001639          0.0000          0.0500   \n",
       "2              0.03333        0.001639          0.0000          0.0500   \n",
       "3              0.80330        0.996100          0.9294          0.7857   \n",
       "4              0.80330        0.996100          0.9294          0.7857   \n",
       "...                ...             ...             ...             ...   \n",
       "315111         0.76200        0.721900          0.7682          0.7881   \n",
       "315112         0.76190        0.725400          0.7679          0.7778   \n",
       "315113         0.76190        0.725400          0.7679          0.7778   \n",
       "315114         0.76190        0.725400          0.7679          0.7778   \n",
       "315115         0.76190        0.725400          0.7679          0.7778   \n",
       "\n",
       "        gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.09167         0.05357         0.02083  0.1667  gnomADg AMI   \n",
       "1              0.09167         0.05357         0.02083  0.1667  gnomADg AMI   \n",
       "2              0.09167         0.05357         0.02083  0.1667  gnomADg AMI   \n",
       "3              0.87980         0.60870         0.88550  0.9984  gnomADe EAS   \n",
       "4              0.87980         0.60870         0.88550  0.9984  gnomADe EAS   \n",
       "...                ...             ...             ...     ...          ...   \n",
       "315111         0.75770         0.62840         0.72010  0.8029          EUR   \n",
       "315112         0.75800         0.67750         0.71760  0.8055          EUR   \n",
       "315113         0.75800         0.67750         0.71760  0.8055          EUR   \n",
       "315114         0.75800         0.67750         0.71760  0.8055          EUR   \n",
       "315115         0.75800         0.67750         0.71760  0.8055          EUR   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene  NaN    NaN   \n",
       "1                           protein coding  NaN    2/2   \n",
       "2                                   lncRNA  NaN    NaN   \n",
       "3                           protein coding  3/3    NaN   \n",
       "4                          TF binding site  NaN    NaN   \n",
       "...                                    ...  ...    ...   \n",
       "315111                      protein coding  2/2    NaN   \n",
       "315112                      protein coding  NaN    NaN   \n",
       "315113                      protein coding  6/6    NaN   \n",
       "315114      protein coding CDS not defined  NaN    NaN   \n",
       "315115      protein coding CDS not defined  NaN    NaN   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED  \n",
       "0                                  NaN      NaN     1.0    NaN  \n",
       "1                                  NaN      NaN     1.0    NaN  \n",
       "2                                  NaN      NaN     1.0    NaN  \n",
       "3                                  S81  tcA/tcG     1.0    NaN  \n",
       "4                                  NaN      NaN     NaN    NaN  \n",
       "...                                ...      ...     ...    ...  \n",
       "315111                             NaN      NaN    -1.0    NaN  \n",
       "315112                             NaN      NaN    -1.0    NaN  \n",
       "315113                             NaN      NaN    -1.0    NaN  \n",
       "315114                             NaN      NaN    -1.0    NaN  \n",
       "315115                             NaN      NaN    -1.0    NaN  \n",
       "\n",
       "[315116 rows x 74 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/Gene_scores_KHGLBS583_depth_vcf_processed.xlsx')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a60ea926",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2303</th>\n",
       "      <td>NPHP4</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>8</td>\n",
       "      <td>Focal segmental glomerulosclerosis; Tubular In...</td>\n",
       "      <td>Glomerular Diseases; Tubular Interstitial dise...</td>\n",
       "      <td>Glomerular_Diseases; Interstitial_Nephritis</td>\n",
       "      <td>rs555164</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5866447</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Nephronophthisis,Nephronophthisis 4,Senior-Lok...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>95683.0</td>\n",
       "      <td>ENST00000378156.9</td>\n",
       "      <td>c.3570A&gt;G</td>\n",
       "      <td>ENSP00000367398.4</td>\n",
       "      <td>p.Glu1190%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.385200e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3197</td>\n",
       "      <td>0.1619</td>\n",
       "      <td>0.3790</td>\n",
       "      <td>0.2837</td>\n",
       "      <td>0.4135</td>\n",
       "      <td>0.4315</td>\n",
       "      <td>0.3926</td>\n",
       "      <td>0.20020</td>\n",
       "      <td>0.3706</td>\n",
       "      <td>0.4699</td>\n",
       "      <td>0.2558</td>\n",
       "      <td>0.42560</td>\n",
       "      <td>0.4222</td>\n",
       "      <td>0.4067</td>\n",
       "      <td>0.4302</td>\n",
       "      <td>0.35380</td>\n",
       "      <td>0.20830</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>0.3694</td>\n",
       "      <td>0.4614</td>\n",
       "      <td>0.2711</td>\n",
       "      <td>0.42760</td>\n",
       "      <td>0.3924</td>\n",
       "      <td>0.41970</td>\n",
       "      <td>0.3617</td>\n",
       "      <td>0.4301</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E1190</td>\n",
       "      <td>gaA/gaG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>25741868,23757202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2304</th>\n",
       "      <td>NPHP4</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>8</td>\n",
       "      <td>Focal segmental glomerulosclerosis; Tubular In...</td>\n",
       "      <td>Glomerular Diseases; Tubular Interstitial dise...</td>\n",
       "      <td>Glomerular_Diseases; Interstitial_Nephritis</td>\n",
       "      <td>rs555164</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5866447</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Nephronophthisis,Nephronophthisis 4,Senior-Lok...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>95683.0</td>\n",
       "      <td>ENST00000378161.5</td>\n",
       "      <td>n.734A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.385200e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3197</td>\n",
       "      <td>0.1619</td>\n",
       "      <td>0.3790</td>\n",
       "      <td>0.2837</td>\n",
       "      <td>0.4135</td>\n",
       "      <td>0.4315</td>\n",
       "      <td>0.3926</td>\n",
       "      <td>0.20020</td>\n",
       "      <td>0.3706</td>\n",
       "      <td>0.4699</td>\n",
       "      <td>0.2558</td>\n",
       "      <td>0.42560</td>\n",
       "      <td>0.4222</td>\n",
       "      <td>0.4067</td>\n",
       "      <td>0.4302</td>\n",
       "      <td>0.35380</td>\n",
       "      <td>0.20830</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>0.3694</td>\n",
       "      <td>0.4614</td>\n",
       "      <td>0.2711</td>\n",
       "      <td>0.42760</td>\n",
       "      <td>0.3924</td>\n",
       "      <td>0.41970</td>\n",
       "      <td>0.3617</td>\n",
       "      <td>0.4301</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>25741868,23757202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2305</th>\n",
       "      <td>NPHP4</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>8</td>\n",
       "      <td>Focal segmental glomerulosclerosis; Tubular In...</td>\n",
       "      <td>Glomerular Diseases; Tubular Interstitial dise...</td>\n",
       "      <td>Glomerular_Diseases; Interstitial_Nephritis</td>\n",
       "      <td>rs555164</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5866447</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>3 prime UTR variant,NMD transcript variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Nephronophthisis,Nephronophthisis 4,Senior-Lok...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>95683.0</td>\n",
       "      <td>ENST00000378169.7</td>\n",
       "      <td>c.*2471A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.385200e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3197</td>\n",
       "      <td>0.1619</td>\n",
       "      <td>0.3790</td>\n",
       "      <td>0.2837</td>\n",
       "      <td>0.4135</td>\n",
       "      <td>0.4315</td>\n",
       "      <td>0.3926</td>\n",
       "      <td>0.20020</td>\n",
       "      <td>0.3706</td>\n",
       "      <td>0.4699</td>\n",
       "      <td>0.2558</td>\n",
       "      <td>0.42560</td>\n",
       "      <td>0.4222</td>\n",
       "      <td>0.4067</td>\n",
       "      <td>0.4302</td>\n",
       "      <td>0.35380</td>\n",
       "      <td>0.20830</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>0.3694</td>\n",
       "      <td>0.4614</td>\n",
       "      <td>0.2711</td>\n",
       "      <td>0.42760</td>\n",
       "      <td>0.3924</td>\n",
       "      <td>0.41970</td>\n",
       "      <td>0.3617</td>\n",
       "      <td>0.4301</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>23/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>25741868,23757202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2306</th>\n",
       "      <td>NPHP4</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>8</td>\n",
       "      <td>Focal segmental glomerulosclerosis; Tubular In...</td>\n",
       "      <td>Glomerular Diseases; Tubular Interstitial dise...</td>\n",
       "      <td>Glomerular_Diseases; Interstitial_Nephritis</td>\n",
       "      <td>rs555164</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5866447</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Nephronophthisis,Nephronophthisis 4,Senior-Lok...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>95683.0</td>\n",
       "      <td>ENST00000460696.1</td>\n",
       "      <td>n.635A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.385200e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3197</td>\n",
       "      <td>0.1619</td>\n",
       "      <td>0.3790</td>\n",
       "      <td>0.2837</td>\n",
       "      <td>0.4135</td>\n",
       "      <td>0.4315</td>\n",
       "      <td>0.3926</td>\n",
       "      <td>0.20020</td>\n",
       "      <td>0.3706</td>\n",
       "      <td>0.4699</td>\n",
       "      <td>0.2558</td>\n",
       "      <td>0.42560</td>\n",
       "      <td>0.4222</td>\n",
       "      <td>0.4067</td>\n",
       "      <td>0.4302</td>\n",
       "      <td>0.35380</td>\n",
       "      <td>0.20830</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>0.3694</td>\n",
       "      <td>0.4614</td>\n",
       "      <td>0.2711</td>\n",
       "      <td>0.42760</td>\n",
       "      <td>0.3924</td>\n",
       "      <td>0.41970</td>\n",
       "      <td>0.3617</td>\n",
       "      <td>0.4301</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>25741868,23757202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2307</th>\n",
       "      <td>NPHP4</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>8</td>\n",
       "      <td>Focal segmental glomerulosclerosis; Tubular In...</td>\n",
       "      <td>Glomerular Diseases; Tubular Interstitial dise...</td>\n",
       "      <td>Glomerular_Diseases; Interstitial_Nephritis</td>\n",
       "      <td>rs555164</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5866447</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Nephronophthisis,Nephronophthisis 4,Senior-Lok...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>95683.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.385200e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3197</td>\n",
       "      <td>0.1619</td>\n",
       "      <td>0.3790</td>\n",
       "      <td>0.2837</td>\n",
       "      <td>0.4135</td>\n",
       "      <td>0.4315</td>\n",
       "      <td>0.3926</td>\n",
       "      <td>0.20020</td>\n",
       "      <td>0.3706</td>\n",
       "      <td>0.4699</td>\n",
       "      <td>0.2558</td>\n",
       "      <td>0.42560</td>\n",
       "      <td>0.4222</td>\n",
       "      <td>0.4067</td>\n",
       "      <td>0.4302</td>\n",
       "      <td>0.35380</td>\n",
       "      <td>0.20830</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>0.3694</td>\n",
       "      <td>0.4614</td>\n",
       "      <td>0.2711</td>\n",
       "      <td>0.42760</td>\n",
       "      <td>0.3924</td>\n",
       "      <td>0.41970</td>\n",
       "      <td>0.3617</td>\n",
       "      <td>0.4301</td>\n",
       "      <td>0.5493</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>25741868,23757202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315089</th>\n",
       "      <td>DKC1</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>6</td>\n",
       "      <td>Other Skin Conditions</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1800533</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154776873</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Dyskeratosis congenita, X-linked</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>166995.0</td>\n",
       "      <td>ENST00000696590.1</td>\n",
       "      <td>n.2577G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>90</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>32</td>\n",
       "      <td>25</td>\n",
       "      <td>43.86%</td>\n",
       "      <td>9.995900e-10</td>\n",
       "      <td>27</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1383</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.1832</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.0731</td>\n",
       "      <td>0.3593</td>\n",
       "      <td>0.1466</td>\n",
       "      <td>0.02632</td>\n",
       "      <td>0.2593</td>\n",
       "      <td>0.1915</td>\n",
       "      <td>0.1008</td>\n",
       "      <td>0.06731</td>\n",
       "      <td>0.0950</td>\n",
       "      <td>0.1424</td>\n",
       "      <td>0.3534</td>\n",
       "      <td>0.09163</td>\n",
       "      <td>0.02709</td>\n",
       "      <td>0.1146</td>\n",
       "      <td>0.1901</td>\n",
       "      <td>0.1937</td>\n",
       "      <td>0.1028</td>\n",
       "      <td>0.06056</td>\n",
       "      <td>0.1477</td>\n",
       "      <td>0.09269</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>gnomADg SAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315090</th>\n",
       "      <td>DKC1</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>6</td>\n",
       "      <td>Other Skin Conditions</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1800533</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154776873</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Dyskeratosis congenita, X-linked</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>166995.0</td>\n",
       "      <td>ENST00000696591.1</td>\n",
       "      <td>n.900G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>90</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>32</td>\n",
       "      <td>25</td>\n",
       "      <td>43.86%</td>\n",
       "      <td>9.995900e-10</td>\n",
       "      <td>27</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1383</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.1832</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.0731</td>\n",
       "      <td>0.3593</td>\n",
       "      <td>0.1466</td>\n",
       "      <td>0.02632</td>\n",
       "      <td>0.2593</td>\n",
       "      <td>0.1915</td>\n",
       "      <td>0.1008</td>\n",
       "      <td>0.06731</td>\n",
       "      <td>0.0950</td>\n",
       "      <td>0.1424</td>\n",
       "      <td>0.3534</td>\n",
       "      <td>0.09163</td>\n",
       "      <td>0.02709</td>\n",
       "      <td>0.1146</td>\n",
       "      <td>0.1901</td>\n",
       "      <td>0.1937</td>\n",
       "      <td>0.1028</td>\n",
       "      <td>0.06056</td>\n",
       "      <td>0.1477</td>\n",
       "      <td>0.09269</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>gnomADg SAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>7/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315091</th>\n",
       "      <td>DKC1</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>6</td>\n",
       "      <td>Other Skin Conditions</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1800533</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154776873</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Dyskeratosis congenita, X-linked</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>166995.0</td>\n",
       "      <td>ENST00000696592.1</td>\n",
       "      <td>n.4306G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>90</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>32</td>\n",
       "      <td>25</td>\n",
       "      <td>43.86%</td>\n",
       "      <td>9.995900e-10</td>\n",
       "      <td>27</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1383</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.1832</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.0731</td>\n",
       "      <td>0.3593</td>\n",
       "      <td>0.1466</td>\n",
       "      <td>0.02632</td>\n",
       "      <td>0.2593</td>\n",
       "      <td>0.1915</td>\n",
       "      <td>0.1008</td>\n",
       "      <td>0.06731</td>\n",
       "      <td>0.0950</td>\n",
       "      <td>0.1424</td>\n",
       "      <td>0.3534</td>\n",
       "      <td>0.09163</td>\n",
       "      <td>0.02709</td>\n",
       "      <td>0.1146</td>\n",
       "      <td>0.1901</td>\n",
       "      <td>0.1937</td>\n",
       "      <td>0.1028</td>\n",
       "      <td>0.06056</td>\n",
       "      <td>0.1477</td>\n",
       "      <td>0.09269</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>gnomADg SAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315092</th>\n",
       "      <td>DKC1</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>6</td>\n",
       "      <td>Other Skin Conditions</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1800533</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154776873</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant,NMD transcript variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Dyskeratosis congenita, X-linked</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>166995.0</td>\n",
       "      <td>ENST00000696627.1</td>\n",
       "      <td>c.*377G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>90</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>32</td>\n",
       "      <td>25</td>\n",
       "      <td>43.86%</td>\n",
       "      <td>9.995900e-10</td>\n",
       "      <td>27</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1383</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.1832</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.0731</td>\n",
       "      <td>0.3593</td>\n",
       "      <td>0.1466</td>\n",
       "      <td>0.02632</td>\n",
       "      <td>0.2593</td>\n",
       "      <td>0.1915</td>\n",
       "      <td>0.1008</td>\n",
       "      <td>0.06731</td>\n",
       "      <td>0.0950</td>\n",
       "      <td>0.1424</td>\n",
       "      <td>0.3534</td>\n",
       "      <td>0.09163</td>\n",
       "      <td>0.02709</td>\n",
       "      <td>0.1146</td>\n",
       "      <td>0.1901</td>\n",
       "      <td>0.1937</td>\n",
       "      <td>0.1028</td>\n",
       "      <td>0.06056</td>\n",
       "      <td>0.1477</td>\n",
       "      <td>0.09269</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>gnomADg SAS</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>15/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315093</th>\n",
       "      <td>DKC1</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>6</td>\n",
       "      <td>Other Skin Conditions</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1800533</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154776873</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Dyskeratosis congenita, X-linked</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>166995.0</td>\n",
       "      <td>ENST00000696628.1</td>\n",
       "      <td>c.*23G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>90</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>32</td>\n",
       "      <td>25</td>\n",
       "      <td>43.86%</td>\n",
       "      <td>9.995900e-10</td>\n",
       "      <td>27</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1383</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.1832</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.0731</td>\n",
       "      <td>0.3593</td>\n",
       "      <td>0.1466</td>\n",
       "      <td>0.02632</td>\n",
       "      <td>0.2593</td>\n",
       "      <td>0.1915</td>\n",
       "      <td>0.1008</td>\n",
       "      <td>0.06731</td>\n",
       "      <td>0.0950</td>\n",
       "      <td>0.1424</td>\n",
       "      <td>0.3534</td>\n",
       "      <td>0.09163</td>\n",
       "      <td>0.02709</td>\n",
       "      <td>0.1146</td>\n",
       "      <td>0.1901</td>\n",
       "      <td>0.1937</td>\n",
       "      <td>0.1028</td>\n",
       "      <td>0.06056</td>\n",
       "      <td>0.1477</td>\n",
       "      <td>0.09269</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>gnomADg SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19751 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name   Gene Gene_Score  \\\n",
       "2303       NPHP4  NPHP4          8   \n",
       "2304       NPHP4  NPHP4          8   \n",
       "2305       NPHP4  NPHP4          8   \n",
       "2306       NPHP4  NPHP4          8   \n",
       "2307       NPHP4  NPHP4          8   \n",
       "...          ...    ...        ...   \n",
       "315089      DKC1   DKC1          6   \n",
       "315090      DKC1   DKC1          6   \n",
       "315091      DKC1   DKC1          6   \n",
       "315092      DKC1   DKC1          6   \n",
       "315093      DKC1   DKC1          6   \n",
       "\n",
       "                                                Condition  \\\n",
       "2303    Focal segmental glomerulosclerosis; Tubular In...   \n",
       "2304    Focal segmental glomerulosclerosis; Tubular In...   \n",
       "2305    Focal segmental glomerulosclerosis; Tubular In...   \n",
       "2306    Focal segmental glomerulosclerosis; Tubular In...   \n",
       "2307    Focal segmental glomerulosclerosis; Tubular In...   \n",
       "...                                                   ...   \n",
       "315089                             Other Skin Conditions    \n",
       "315090                             Other Skin Conditions    \n",
       "315091                             Other Skin Conditions    \n",
       "315092                             Other Skin Conditions    \n",
       "315093                             Other Skin Conditions    \n",
       "\n",
       "                                                 Headings  \\\n",
       "2303    Glomerular Diseases; Tubular Interstitial dise...   \n",
       "2304    Glomerular Diseases; Tubular Interstitial dise...   \n",
       "2305    Glomerular Diseases; Tubular Interstitial dise...   \n",
       "2306    Glomerular Diseases; Tubular Interstitial dise...   \n",
       "2307    Glomerular Diseases; Tubular Interstitial dise...   \n",
       "...                                                   ...   \n",
       "315089                                        Skin Health   \n",
       "315090                                        Skin Health   \n",
       "315091                                        Skin Health   \n",
       "315092                                        Skin Health   \n",
       "315093                                        Skin Health   \n",
       "\n",
       "                                 21_Conditions_list       rsID Literature  \\\n",
       "2303    Glomerular_Diseases; Interstitial_Nephritis   rs555164         No   \n",
       "2304    Glomerular_Diseases; Interstitial_Nephritis   rs555164         No   \n",
       "2305    Glomerular_Diseases; Interstitial_Nephritis   rs555164         No   \n",
       "2306    Glomerular_Diseases; Interstitial_Nephritis   rs555164         No   \n",
       "2307    Glomerular_Diseases; Interstitial_Nephritis   rs555164         No   \n",
       "...                                             ...        ...        ...   \n",
       "315089                                  Skin_Health  rs1800533         No   \n",
       "315090                                  Skin_Health  rs1800533         No   \n",
       "315091                                  Skin_Health  rs1800533         No   \n",
       "315092                                  Skin_Health  rs1800533         No   \n",
       "315093                                  Skin_Health  rs1800533         No   \n",
       "\n",
       "       CHROM        POS REF ALT      Zygosity  \\\n",
       "2303    chr1    5866447   T   C    Homozygous   \n",
       "2304    chr1    5866447   T   C    Homozygous   \n",
       "2305    chr1    5866447   T   C    Homozygous   \n",
       "2306    chr1    5866447   T   C    Homozygous   \n",
       "2307    chr1    5866447   T   C    Homozygous   \n",
       "...      ...        ...  ..  ..           ...   \n",
       "315089  chrX  154776873   G   A  Heterozygous   \n",
       "315090  chrX  154776873   G   A  Heterozygous   \n",
       "315091  chrX  154776873   G   A  Heterozygous   \n",
       "315092  chrX  154776873   G   A  Heterozygous   \n",
       "315093  chrX  154776873   G   A  Heterozygous   \n",
       "\n",
       "                                       Consequence Consequence_score  \\\n",
       "2303                            synonymous variant              3/10   \n",
       "2304            non coding transcript exon variant              2/10   \n",
       "2305    3 prime UTR variant,NMD transcript variant              3/10   \n",
       "2306            non coding transcript exon variant              2/10   \n",
       "2307                       downstream gene variant              2/10   \n",
       "...                                            ...               ...   \n",
       "315089          non coding transcript exon variant              2/10   \n",
       "315090          non coding transcript exon variant              2/10   \n",
       "315091          non coding transcript exon variant              2/10   \n",
       "315092  3 prime UTR variant,NMD transcript variant              3/10   \n",
       "315093                         3 prime UTR variant              3/10   \n",
       "\n",
       "          IMPACT  IMPACT_score  \\\n",
       "2303         LOW           2.5   \n",
       "2304    MODIFIER           1.5   \n",
       "2305    MODIFIER           1.5   \n",
       "2306    MODIFIER           1.5   \n",
       "2307    MODIFIER           1.5   \n",
       "...          ...           ...   \n",
       "315089  MODIFIER           1.5   \n",
       "315090  MODIFIER           1.5   \n",
       "315091  MODIFIER           1.5   \n",
       "315092  MODIFIER           1.5   \n",
       "315093  MODIFIER           1.5   \n",
       "\n",
       "                                            ClinVar_CLNDN CLIN_SIG  \\\n",
       "2303    Nephronophthisis,Nephronophthisis 4,Senior-Lok...   benign   \n",
       "2304    Nephronophthisis,Nephronophthisis 4,Senior-Lok...   benign   \n",
       "2305    Nephronophthisis,Nephronophthisis 4,Senior-Lok...   benign   \n",
       "2306    Nephronophthisis,Nephronophthisis 4,Senior-Lok...   benign   \n",
       "2307    Nephronophthisis,Nephronophthisis 4,Senior-Lok...   benign   \n",
       "...                                                   ...      ...   \n",
       "315089                   Dyskeratosis congenita, X-linked   benign   \n",
       "315090                   Dyskeratosis congenita, X-linked   benign   \n",
       "315091                   Dyskeratosis congenita, X-linked   benign   \n",
       "315092                   Dyskeratosis congenita, X-linked   benign   \n",
       "315093                   Dyskeratosis congenita, X-linked   benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT   ClinVar  \\\n",
       "2303    criteria provided, multiple submitters, no con...   95683.0   \n",
       "2304    criteria provided, multiple submitters, no con...   95683.0   \n",
       "2305    criteria provided, multiple submitters, no con...   95683.0   \n",
       "2306    criteria provided, multiple submitters, no con...   95683.0   \n",
       "2307    criteria provided, multiple submitters, no con...   95683.0   \n",
       "...                                                   ...       ...   \n",
       "315089  criteria provided, multiple submitters, no con...  166995.0   \n",
       "315090  criteria provided, multiple submitters, no con...  166995.0   \n",
       "315091  criteria provided, multiple submitters, no con...  166995.0   \n",
       "315092  criteria provided, multiple submitters, no con...  166995.0   \n",
       "315093  criteria provided, multiple submitters, no con...  166995.0   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "2303    ENST00000378156.9          c.3570A>G  ENSP00000367398.4   \n",
       "2304    ENST00000378161.5           n.734A>G                NaN   \n",
       "2305    ENST00000378169.7         c.*2471A>G                NaN   \n",
       "2306    ENST00000460696.1           n.635A>G                NaN   \n",
       "2307                  NaN               None                NaN   \n",
       "...                   ...                ...                ...   \n",
       "315089  ENST00000696590.1          n.2577G>A                NaN   \n",
       "315090  ENST00000696591.1           n.900G>A                NaN   \n",
       "315091  ENST00000696592.1          n.4306G>A                NaN   \n",
       "315092  ENST00000696627.1          c.*377G>A                NaN   \n",
       "315093  ENST00000696628.1           c.*23G>A                NaN   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "2303         p.Glu1190%3D  1/1  188   33  33   0  33    100%  1.385200e-19   \n",
       "2304                 None  1/1  188   33  33   0  33    100%  1.385200e-19   \n",
       "2305                 None  1/1  188   33  33   0  33    100%  1.385200e-19   \n",
       "2306                 None  1/1  188   33  33   0  33    100%  1.385200e-19   \n",
       "2307                 None  1/1  188   33  33   0  33    100%  1.385200e-19   \n",
       "...                   ...  ...  ...  ...  ..  ..  ..     ...           ...   \n",
       "315089               None  0/1   90   57  57  32  25  43.86%  9.995900e-10   \n",
       "315090               None  0/1   90   57  57  32  25  43.86%  9.995900e-10   \n",
       "315091               None  0/1   90   57  57  32  25  43.86%  9.995900e-10   \n",
       "315092               None  0/1   90   57  57  32  25  43.86%  9.995900e-10   \n",
       "315093               None  0/1   90   57  57  32  25  43.86%  9.995900e-10   \n",
       "\n",
       "        RDF  RDR  ADF  ADR SIFT PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  \\\n",
       "2303      0    0   24    9  NaN      NaN  0.3197  0.1619  0.3790  0.2837   \n",
       "2304      0    0   24    9  NaN      NaN  0.3197  0.1619  0.3790  0.2837   \n",
       "2305      0    0   24    9  NaN      NaN  0.3197  0.1619  0.3790  0.2837   \n",
       "2306      0    0   24    9  NaN      NaN  0.3197  0.1619  0.3790  0.2837   \n",
       "2307      0    0   24    9  NaN      NaN  0.3197  0.1619  0.3790  0.2837   \n",
       "...     ...  ...  ...  ...  ...      ...     ...     ...     ...     ...   \n",
       "315089   27    5   21    4  NaN      NaN  0.1383  0.0169  0.1832  0.1243   \n",
       "315090   27    5   21    4  NaN      NaN  0.1383  0.0169  0.1832  0.1243   \n",
       "315091   27    5   21    4  NaN      NaN  0.1383  0.0169  0.1832  0.1243   \n",
       "315092   27    5   21    4  NaN      NaN  0.1383  0.0169  0.1832  0.1243   \n",
       "315093   27    5   21    4  NaN      NaN  0.1383  0.0169  0.1832  0.1243   \n",
       "\n",
       "        EUR_AF  SAS_AF  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  \\\n",
       "2303    0.4135  0.4315      0.3926         0.20020          0.3706   \n",
       "2304    0.4135  0.4315      0.3926         0.20020          0.3706   \n",
       "2305    0.4135  0.4315      0.3926         0.20020          0.3706   \n",
       "2306    0.4135  0.4315      0.3926         0.20020          0.3706   \n",
       "2307    0.4135  0.4315      0.3926         0.20020          0.3706   \n",
       "...        ...     ...         ...             ...             ...   \n",
       "315089  0.0731  0.3593      0.1466         0.02632          0.2593   \n",
       "315090  0.0731  0.3593      0.1466         0.02632          0.2593   \n",
       "315091  0.0731  0.3593      0.1466         0.02632          0.2593   \n",
       "315092  0.0731  0.3593      0.1466         0.02632          0.2593   \n",
       "315093  0.0731  0.3593      0.1466         0.02632          0.2593   \n",
       "\n",
       "        gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  \\\n",
       "2303            0.4699          0.2558         0.42560          0.4222   \n",
       "2304            0.4699          0.2558         0.42560          0.4222   \n",
       "2305            0.4699          0.2558         0.42560          0.4222   \n",
       "2306            0.4699          0.2558         0.42560          0.4222   \n",
       "2307            0.4699          0.2558         0.42560          0.4222   \n",
       "...                ...             ...             ...             ...   \n",
       "315089          0.1915          0.1008         0.06731          0.0950   \n",
       "315090          0.1915          0.1008         0.06731          0.0950   \n",
       "315091          0.1915          0.1008         0.06731          0.0950   \n",
       "315092          0.1915          0.1008         0.06731          0.0950   \n",
       "315093          0.1915          0.1008         0.06731          0.0950   \n",
       "\n",
       "        gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  \\\n",
       "2303            0.4067          0.4302     0.35380         0.20830   \n",
       "2304            0.4067          0.4302     0.35380         0.20830   \n",
       "2305            0.4067          0.4302     0.35380         0.20830   \n",
       "2306            0.4067          0.4302     0.35380         0.20830   \n",
       "2307            0.4067          0.4302     0.35380         0.20830   \n",
       "...                ...             ...         ...             ...   \n",
       "315089          0.1424          0.3534     0.09163         0.02709   \n",
       "315090          0.1424          0.3534     0.09163         0.02709   \n",
       "315091          0.1424          0.3534     0.09163         0.02709   \n",
       "315092          0.1424          0.3534     0.09163         0.02709   \n",
       "315093          0.1424          0.3534     0.09163         0.02709   \n",
       "\n",
       "        gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  \\\n",
       "2303            0.5493          0.3694          0.4614          0.2711   \n",
       "2304            0.5493          0.3694          0.4614          0.2711   \n",
       "2305            0.5493          0.3694          0.4614          0.2711   \n",
       "2306            0.5493          0.3694          0.4614          0.2711   \n",
       "2307            0.5493          0.3694          0.4614          0.2711   \n",
       "...                ...             ...             ...             ...   \n",
       "315089          0.1146          0.1901          0.1937          0.1028   \n",
       "315090          0.1146          0.1901          0.1937          0.1028   \n",
       "315091          0.1146          0.1901          0.1937          0.1028   \n",
       "315092          0.1146          0.1901          0.1937          0.1028   \n",
       "315093          0.1146          0.1901          0.1937          0.1028   \n",
       "\n",
       "        gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  \\\n",
       "2303           0.42760          0.3924         0.41970          0.3617   \n",
       "2304           0.42760          0.3924         0.41970          0.3617   \n",
       "2305           0.42760          0.3924         0.41970          0.3617   \n",
       "2306           0.42760          0.3924         0.41970          0.3617   \n",
       "2307           0.42760          0.3924         0.41970          0.3617   \n",
       "...                ...             ...             ...             ...   \n",
       "315089         0.06056          0.1477         0.09269          0.1181   \n",
       "315090         0.06056          0.1477         0.09269          0.1181   \n",
       "315091         0.06056          0.1477         0.09269          0.1181   \n",
       "315092         0.06056          0.1477         0.09269          0.1181   \n",
       "315093         0.06056          0.1477         0.09269          0.1181   \n",
       "\n",
       "        gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS                  BIOTYPE   EXON  \\\n",
       "2303            0.4301  0.5493  gnomADg AMI           protein coding  26/30   \n",
       "2304            0.4301  0.5493  gnomADg AMI          retained intron    3/3   \n",
       "2305            0.4301  0.5493  gnomADg AMI  nonsense mediated decay  23/27   \n",
       "2306            0.4301  0.5493  gnomADg AMI          retained intron    2/3   \n",
       "2307            0.4301  0.5493  gnomADg AMI          retained intron    NaN   \n",
       "...                ...     ...          ...                      ...    ...   \n",
       "315089          0.3679  0.3679  gnomADg SAS          retained intron    6/6   \n",
       "315090          0.3679  0.3679  gnomADg SAS          retained intron    7/7   \n",
       "315091          0.3679  0.3679  gnomADg SAS          retained intron    2/2   \n",
       "315092          0.3679  0.3679  gnomADg SAS  nonsense mediated decay  15/15   \n",
       "315093          0.3679  0.3679  gnomADg SAS           protein coding  15/15   \n",
       "\n",
       "       INTRON Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "2303      NaN                           E1190  gaA/gaG    -1.0   \n",
       "2304      NaN                             NaN      NaN    -1.0   \n",
       "2305      NaN                             NaN      NaN    -1.0   \n",
       "2306      NaN                             NaN      NaN    -1.0   \n",
       "2307      NaN                             NaN      NaN    -1.0   \n",
       "...       ...                             ...      ...     ...   \n",
       "315089    NaN                             NaN      NaN     1.0   \n",
       "315090    NaN                             NaN      NaN     1.0   \n",
       "315091    NaN                             NaN      NaN     1.0   \n",
       "315092    NaN                             NaN      NaN     1.0   \n",
       "315093    NaN                             NaN      NaN     1.0   \n",
       "\n",
       "                   PUBMED  \n",
       "2303    25741868,23757202  \n",
       "2304    25741868,23757202  \n",
       "2305    25741868,23757202  \n",
       "2306    25741868,23757202  \n",
       "2307    25741868,23757202  \n",
       "...                   ...  \n",
       "315089                NaN  \n",
       "315090                NaN  \n",
       "315091                NaN  \n",
       "315092                NaN  \n",
       "315093                NaN  \n",
       "\n",
       "[19751 rows x 74 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = vcf[vcf['21_Conditions_list'] != 'No']\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "e4523ab1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5222</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5223</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5224</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.196)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5225</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.9)</td>\n",
       "      <td>benign(0.332)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6559</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypercholesterolemia,...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>68.75%</td>\n",
       "      <td>3.385400e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>0.4349</td>\n",
       "      <td>0.3207</td>\n",
       "      <td>0.5101</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.5239</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.5054</td>\n",
       "      <td>0.3509</td>\n",
       "      <td>0.5516</td>\n",
       "      <td>0.5811</td>\n",
       "      <td>0.4477</td>\n",
       "      <td>0.5044</td>\n",
       "      <td>0.5345</td>\n",
       "      <td>0.5366</td>\n",
       "      <td>0.4310</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.3543</td>\n",
       "      <td>0.4901</td>\n",
       "      <td>0.5202</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.4747</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.5573</td>\n",
       "      <td>0.5379</td>\n",
       "      <td>0.5249</td>\n",
       "      <td>0.4449</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.76)</td>\n",
       "      <td>probably damaging(0.998)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.3342</td>\n",
       "      <td>0.2071</td>\n",
       "      <td>0.2480</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>0.3327</td>\n",
       "      <td>0.3263</td>\n",
       "      <td>0.2208</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>0.3170</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303239</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs2488602</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123371268</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1167061.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.2126T&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Val709Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.609000e-36</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>15</td>\n",
       "      <td>tolerated(0.71)</td>\n",
       "      <td>benign(0.003)</td>\n",
       "      <td>0.9966</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9914</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9978</td>\n",
       "      <td>0.9921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>V709A</td>\n",
       "      <td>gTg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303240</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs2488602</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123371268</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1167061.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.2126T&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Val709Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.609000e-36</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>15</td>\n",
       "      <td>tolerated(0.72)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.9966</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9914</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9978</td>\n",
       "      <td>0.9921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>V709A</td>\n",
       "      <td>gTg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303241</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs2488602</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123371268</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant,NMD transcript variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1167061.0</td>\n",
       "      <td>ENST00000460253.1</td>\n",
       "      <td>c.1130T&gt;C</td>\n",
       "      <td>ENSP00000435279.1</td>\n",
       "      <td>p.Val377Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.609000e-36</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>15</td>\n",
       "      <td>tolerated(0.66)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.9966</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9914</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9978</td>\n",
       "      <td>0.9921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>3/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>V377A</td>\n",
       "      <td>gTg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314496</th>\n",
       "      <td>AVPR2</td>\n",
       "      <td>AVPR2</td>\n",
       "      <td>8</td>\n",
       "      <td>Diabetic Nephropathy; Diabetic kidney disease</td>\n",
       "      <td>Diabetic Nephropathy; Glomerular Diseases</td>\n",
       "      <td>Diabetes; Glomerular_Diseases</td>\n",
       "      <td>rs5201</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153906539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Nephrogenic syndrome of inappropriate antidiur...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>254775.0</td>\n",
       "      <td>ENST00000430697.1</td>\n",
       "      <td>c.839A&gt;G</td>\n",
       "      <td>ENSP00000393513.1</td>\n",
       "      <td>p.Tyr280Cys</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>71</td>\n",
       "      <td>71</td>\n",
       "      <td>0</td>\n",
       "      <td>71</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.683500e-42</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>24</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>possibly damaging(0.634)</td>\n",
       "      <td>0.6885</td>\n",
       "      <td>0.9372</td>\n",
       "      <td>0.6450</td>\n",
       "      <td>0.7186</td>\n",
       "      <td>0.3185</td>\n",
       "      <td>0.7354</td>\n",
       "      <td>0.4753</td>\n",
       "      <td>0.8754</td>\n",
       "      <td>0.6834</td>\n",
       "      <td>0.3524</td>\n",
       "      <td>0.7177</td>\n",
       "      <td>0.3212</td>\n",
       "      <td>0.2905</td>\n",
       "      <td>0.4248</td>\n",
       "      <td>0.7028</td>\n",
       "      <td>0.5095</td>\n",
       "      <td>0.8647</td>\n",
       "      <td>0.0784</td>\n",
       "      <td>0.5989</td>\n",
       "      <td>0.3470</td>\n",
       "      <td>0.7088</td>\n",
       "      <td>0.3187</td>\n",
       "      <td>0.4852</td>\n",
       "      <td>0.2976</td>\n",
       "      <td>0.5175</td>\n",
       "      <td>0.7193</td>\n",
       "      <td>0.9372</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y280C</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1635 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name     Gene Gene_Score  \\\n",
       "5222       PADI4    PADI4         No   \n",
       "5223       PADI4    PADI4         No   \n",
       "5224       PADI4    PADI4         No   \n",
       "5225       PADI4    PADI4         No   \n",
       "6559     LDLRAP1  LDLRAP1          8   \n",
       "...          ...      ...        ...   \n",
       "303234      CRB2     CRB2          6   \n",
       "303239      CRB2     CRB2          6   \n",
       "303240      CRB2     CRB2          6   \n",
       "303241      CRB2     CRB2          6   \n",
       "314496     AVPR2    AVPR2          8   \n",
       "\n",
       "                                                Condition  \\\n",
       "5222                                 Rheumatoid Arthritis   \n",
       "5223                                 Rheumatoid Arthritis   \n",
       "5224                                 Rheumatoid Arthritis   \n",
       "5225                                 Rheumatoid Arthritis   \n",
       "6559    Polygenic (Familial combined, small dense LDL,...   \n",
       "...                                                   ...   \n",
       "303234                 Focal segmental glomerulosclerosis   \n",
       "303239                 Focal segmental glomerulosclerosis   \n",
       "303240                 Focal segmental glomerulosclerosis   \n",
       "303241                 Focal segmental glomerulosclerosis   \n",
       "314496      Diabetic Nephropathy; Diabetic kidney disease   \n",
       "\n",
       "                                         Headings  \\\n",
       "5222                          Autoimmune Disorder   \n",
       "5223                          Autoimmune Disorder   \n",
       "5224                          Autoimmune Disorder   \n",
       "5225                          Autoimmune Disorder   \n",
       "6559                        Cholesterol disorders   \n",
       "...                                           ...   \n",
       "303234                        Glomerular Diseases   \n",
       "303239                        Glomerular Diseases   \n",
       "303240                        Glomerular Diseases   \n",
       "303241                        Glomerular Diseases   \n",
       "314496  Diabetic Nephropathy; Glomerular Diseases   \n",
       "\n",
       "                          21_Conditions_list        rsID Literature CHROM  \\\n",
       "5222    Arthritis_Degenerative_Joint_Disease  rs11203366         No  chr1   \n",
       "5223    Arthritis_Degenerative_Joint_Disease  rs11203366         No  chr1   \n",
       "5224    Arthritis_Degenerative_Joint_Disease    rs874881         No  chr1   \n",
       "5225    Arthritis_Degenerative_Joint_Disease    rs874881         No  chr1   \n",
       "6559                   Cholesterol_Disorders   rs6687605         No  chr1   \n",
       "...                                      ...         ...        ...   ...   \n",
       "303234                   Glomerular_Diseases   rs1105222         No  chr9   \n",
       "303239                   Glomerular_Diseases   rs2488602         No  chr9   \n",
       "303240                   Glomerular_Diseases   rs2488602         No  chr9   \n",
       "303241                   Glomerular_Diseases   rs2488602         No  chr9   \n",
       "314496         Diabetes; Glomerular_Diseases      rs5201        Yes  chrX   \n",
       "\n",
       "              POS REF ALT      Zygosity  \\\n",
       "5222     17331039   G   A  Heterozygous   \n",
       "5223     17331039   G   A  Heterozygous   \n",
       "5224     17334004   G   C  Heterozygous   \n",
       "5225     17334004   G   C  Heterozygous   \n",
       "6559     25563141   T   C  Heterozygous   \n",
       "...           ...  ..  ..           ...   \n",
       "303234  123365974   G   C  Heterozygous   \n",
       "303239  123371268   T   C    Homozygous   \n",
       "303240  123371268   T   C    Homozygous   \n",
       "303241  123371268   T   C    Homozygous   \n",
       "314496  153906539   A   G    Homozygous   \n",
       "\n",
       "                                    Consequence Consequence_score    IMPACT  \\\n",
       "5222                           missense variant              7/10  MODERATE   \n",
       "5223                           missense variant              7/10  MODERATE   \n",
       "5224                           missense variant              7/10  MODERATE   \n",
       "5225                           missense variant              7/10  MODERATE   \n",
       "6559                           missense variant              7/10  MODERATE   \n",
       "...                                         ...               ...       ...   \n",
       "303234                         missense variant              7/10  MODERATE   \n",
       "303239                         missense variant              7/10  MODERATE   \n",
       "303240                         missense variant              7/10  MODERATE   \n",
       "303241  missense variant,NMD transcript variant              7/10  MODERATE   \n",
       "314496                         missense variant              7/10  MODERATE   \n",
       "\n",
       "        IMPACT_score                                      ClinVar_CLNDN  \\\n",
       "5222             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "5223             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "5224             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "5225             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "6559             5.0  Cardiovascular phenotype,Hypercholesterolemia,...   \n",
       "...              ...                                                ...   \n",
       "303234           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "303239           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "303240           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "303241           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "314496           5.0  Nephrogenic syndrome of inappropriate antidiur...   \n",
       "\n",
       "           CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "5222    association                     no assertion criteria provided   \n",
       "5223    association                     no assertion criteria provided   \n",
       "5224    association                     no assertion criteria provided   \n",
       "5225    association                     no assertion criteria provided   \n",
       "6559         benign  criteria provided, multiple submitters, no con...   \n",
       "...             ...                                                ...   \n",
       "303234       benign  criteria provided, multiple submitters, no con...   \n",
       "303239       benign  criteria provided, multiple submitters, no con...   \n",
       "303240       benign  criteria provided, multiple submitters, no con...   \n",
       "303241       benign  criteria provided, multiple submitters, no con...   \n",
       "314496       benign  criteria provided, multiple submitters, no con...   \n",
       "\n",
       "          ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "5222     972892.0  ENST00000375448.4           c.163G>A  ENSP00000364597.4   \n",
       "5223     972892.0  ENST00000375453.5           c.163G>A  ENSP00000364602.1   \n",
       "5224     972895.0  ENST00000375448.4           c.335G>C  ENSP00000364597.4   \n",
       "5225     972895.0  ENST00000375453.5           c.335G>C  ENSP00000364602.1   \n",
       "6559     296982.0  ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "...           ...                ...                ...                ...   \n",
       "303234  1166114.0  ENST00000373631.8           c.476G>C  ENSP00000362734.3   \n",
       "303239  1167061.0  ENST00000359999.7          c.2126T>C  ENSP00000353092.3   \n",
       "303240  1167061.0  ENST00000373631.8          c.2126T>C  ENSP00000362734.3   \n",
       "303241  1167061.0  ENST00000460253.1          c.1130T>C  ENSP00000435279.1   \n",
       "314496   254775.0  ENST00000430697.1           c.839A>G  ENSP00000393513.1   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "5222           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5223           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5224          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03   \n",
       "5225          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03   \n",
       "6559          p.Ser202Pro  0/1   44   16  16   5  11  68.75%  3.385400e-05   \n",
       "...                   ...  ...  ...  ...  ..  ..  ..     ...           ...   \n",
       "303234        p.Gly159Ala  0/1   73   41  41  21  20  48.78%  4.334700e-08   \n",
       "303239        p.Val709Ala  1/1  255   61  61   0  61    100%  2.609000e-36   \n",
       "303240        p.Val709Ala  1/1  255   61  61   0  61    100%  2.609000e-36   \n",
       "303241        p.Val377Ala  1/1  255   61  61   0  61    100%  2.609000e-36   \n",
       "314496        p.Tyr280Cys  1/1  255   71  71   0  71    100%  2.683500e-42   \n",
       "\n",
       "        RDF  RDR  ADF  ADR               SIFT                  PolyPhen  \\\n",
       "5222     29    3    7    8       tolerated(1)                 benign(0)   \n",
       "5223     29    3    7    8       tolerated(1)                 benign(0)   \n",
       "5224      7    0    5    3    tolerated(0.67)             benign(0.196)   \n",
       "5225      7    0    5    3     tolerated(0.9)             benign(0.332)   \n",
       "6559      3    2    5    6  deleterious(0.04)             benign(0.037)   \n",
       "...     ...  ...  ...  ...                ...                       ...   \n",
       "303234   12    9   16    4    tolerated(0.76)  probably damaging(0.998)   \n",
       "303239    0    0   46   15    tolerated(0.71)             benign(0.003)   \n",
       "303240    0    0   46   15    tolerated(0.72)                 benign(0)   \n",
       "303241    0    0   46   15    tolerated(0.66)             benign(0.001)   \n",
       "314496    0    0   47   24    tolerated(0.16)  possibly damaging(0.634)   \n",
       "\n",
       "            AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  \\\n",
       "5222    0.5246  0.4508  0.5432  0.5635  0.5755  0.5184      0.5589   \n",
       "5223    0.5246  0.4508  0.5432  0.5635  0.5755  0.5184      0.5589   \n",
       "5224    0.5220  0.4607  0.5403  0.5585  0.5537  0.5215      0.5453   \n",
       "5225    0.5220  0.4607  0.5403  0.5585  0.5537  0.5215      0.5453   \n",
       "6559    0.4349  0.3207  0.5101  0.4425  0.5239  0.4366      0.5054   \n",
       "...        ...     ...     ...     ...     ...     ...         ...   \n",
       "303234  0.3159  0.1989  0.3314  0.3889  0.3728  0.3292      0.3342   \n",
       "303239  0.9966  0.9871  1.0000  1.0000  1.0000  1.0000      0.9994   \n",
       "303240  0.9966  0.9871  1.0000  1.0000  1.0000  1.0000      0.9994   \n",
       "303241  0.9966  0.9871  1.0000  1.0000  1.0000  1.0000      0.9994   \n",
       "314496  0.6885  0.9372  0.6450  0.7186  0.3185  0.7354      0.4753   \n",
       "\n",
       "        gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  \\\n",
       "5222            0.4713          0.5003          0.5952          0.5767   \n",
       "5223            0.4713          0.5003          0.5952          0.5767   \n",
       "5224            0.4790          0.4973          0.5764          0.5781   \n",
       "5225            0.4790          0.4973          0.5764          0.5781   \n",
       "6559            0.3509          0.5516          0.5811          0.4477   \n",
       "...                ...             ...             ...             ...   \n",
       "303234          0.2071          0.2480          0.3128          0.3752   \n",
       "303239          0.9914          0.9999          1.0000          1.0000   \n",
       "303240          0.9914          0.9999          1.0000          1.0000   \n",
       "303241          0.9914          0.9999          1.0000          1.0000   \n",
       "314496          0.8754          0.6834          0.3524          0.7177   \n",
       "\n",
       "        gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  \\\n",
       "5222            0.5898          0.5858          0.5713          0.5220   \n",
       "5223            0.5898          0.5858          0.5713          0.5220   \n",
       "5224            0.5581          0.5635          0.5577          0.5253   \n",
       "5225            0.5581          0.5635          0.5577          0.5253   \n",
       "6559            0.5044          0.5345          0.5366          0.4310   \n",
       "...                ...             ...             ...             ...   \n",
       "303234          0.3427          0.3740          0.3400          0.3327   \n",
       "303239          1.0000          0.9999          0.9997          1.0000   \n",
       "303240          1.0000          0.9999          0.9997          1.0000   \n",
       "303241          1.0000          0.9999          0.9997          1.0000   \n",
       "314496          0.3212          0.2905          0.4248          0.7028   \n",
       "\n",
       "        gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "5222        0.5508          0.4763          0.6490          0.5484   \n",
       "5223        0.5508          0.4763          0.6490          0.5484   \n",
       "5224        0.5361          0.4791          0.6355          0.5395   \n",
       "5225        0.5361          0.4791          0.6355          0.5395   \n",
       "6559        0.4791          0.3543          0.4901          0.5202   \n",
       "...            ...             ...             ...             ...   \n",
       "303234      0.3263          0.2208          0.3352          0.3230   \n",
       "303239      0.9978          0.9921          1.0000          0.9996   \n",
       "303240      0.9978          0.9921          1.0000          0.9996   \n",
       "303241      0.9978          0.9921          1.0000          0.9996   \n",
       "314496      0.5095          0.8647          0.0784          0.5989   \n",
       "\n",
       "        gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "5222            0.6176          0.5931          0.5930          0.5759   \n",
       "5223            0.6176          0.5931          0.5930          0.5759   \n",
       "5224            0.5988          0.5903          0.5603          0.5759   \n",
       "5225            0.5988          0.5903          0.5603          0.5759   \n",
       "6559            0.5858          0.4747          0.5000          0.5573   \n",
       "...                ...             ...             ...             ...   \n",
       "303234          0.3170          0.3788          0.3561          0.3354   \n",
       "303239          1.0000          1.0000          1.0000          1.0000   \n",
       "303240          1.0000          1.0000          1.0000          1.0000   \n",
       "303241          1.0000          1.0000          1.0000          1.0000   \n",
       "314496          0.3470          0.7088          0.3187          0.4852   \n",
       "\n",
       "        gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF  \\\n",
       "5222            0.5840          0.5311          0.5319  0.6490   \n",
       "5223            0.5840          0.5311          0.5319  0.6490   \n",
       "5224            0.5579          0.5207          0.5347  0.6355   \n",
       "5225            0.5579          0.5207          0.5347  0.6355   \n",
       "6559            0.5379          0.5249          0.4449  0.5858   \n",
       "...                ...             ...             ...     ...   \n",
       "303234          0.3825          0.3250          0.3326  0.3889   \n",
       "303239          1.0000          0.9995          1.0000  1.0000   \n",
       "303240          1.0000          0.9995          1.0000  1.0000   \n",
       "303241          1.0000          0.9995          1.0000  1.0000   \n",
       "314496          0.2976          0.5175          0.7193  0.9372   \n",
       "\n",
       "                                              MAX_AF_POPS  \\\n",
       "5222                                          gnomADg AMI   \n",
       "5223                                          gnomADg AMI   \n",
       "5224                                          gnomADg AMI   \n",
       "5225                                          gnomADg AMI   \n",
       "6559                                          gnomADg ASJ   \n",
       "...                                                   ...   \n",
       "303234                                                EAS   \n",
       "303239  AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...   \n",
       "303240  AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...   \n",
       "303241  AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...   \n",
       "314496                                                AFR   \n",
       "\n",
       "                        BIOTYPE  EXON INTRON Protein Position and Amino Acid  \\\n",
       "5222             protein coding  2/16    NaN                            G55S   \n",
       "5223             protein coding   2/4    NaN                            G55S   \n",
       "5224             protein coding  3/16    NaN                           G112A   \n",
       "5225             protein coding   3/4    NaN                           G112A   \n",
       "6559             protein coding   6/9    NaN                           S202P   \n",
       "...                         ...   ...    ...                             ...   \n",
       "303234           protein coding  3/13    NaN                           G159A   \n",
       "303239           protein coding  8/10    NaN                           V709A   \n",
       "303240           protein coding  8/13    NaN                           V709A   \n",
       "303241  nonsense mediated decay   3/9    NaN                           V377A   \n",
       "314496           protein coding   3/3    NaN                           Y280C   \n",
       "\n",
       "         Codons  STRAND                                             PUBMED  \n",
       "5222    Ggc/Agc     1.0  23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5223    Ggc/Agc     1.0  23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5224    gGg/gCg     1.0  23251581,26843965,21698003,18576335,20537173,2...  \n",
       "5225    gGg/gCg     1.0  23251581,26843965,21698003,18576335,20537173,2...  \n",
       "6559    Tcc/Ccc     1.0       31227787,24906453,31397093,26129832,24067191  \n",
       "...         ...     ...                                                ...  \n",
       "303234  gGc/gCc     1.0                                                NaN  \n",
       "303239  gTg/gCg     1.0                                                NaN  \n",
       "303240  gTg/gCg     1.0                                                NaN  \n",
       "303241  gTg/gCg     1.0                                                NaN  \n",
       "314496  tAc/tGc     1.0                                                NaN  \n",
       "\n",
       "[1635 rows x 74 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cons_x = x[x['Consequence_score'].apply(lambda x: eval(x) >= 6/10)]\n",
    "cons_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "69780031",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5222</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5223</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5224</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.196)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5225</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.9)</td>\n",
       "      <td>benign(0.332)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6559</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypercholesterolemia,...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>68.75%</td>\n",
       "      <td>3.385400e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>0.4349</td>\n",
       "      <td>0.3207</td>\n",
       "      <td>0.5101</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.5239</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.5054</td>\n",
       "      <td>0.3509</td>\n",
       "      <td>0.5516</td>\n",
       "      <td>0.5811</td>\n",
       "      <td>0.4477</td>\n",
       "      <td>0.5044</td>\n",
       "      <td>0.5345</td>\n",
       "      <td>0.5366</td>\n",
       "      <td>0.4310</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.3543</td>\n",
       "      <td>0.4901</td>\n",
       "      <td>0.5202</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.4747</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.5573</td>\n",
       "      <td>0.5379</td>\n",
       "      <td>0.5249</td>\n",
       "      <td>0.4449</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.76)</td>\n",
       "      <td>probably damaging(0.998)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.3342</td>\n",
       "      <td>0.2071</td>\n",
       "      <td>0.2480</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>0.3327</td>\n",
       "      <td>0.3263</td>\n",
       "      <td>0.2208</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>0.3170</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303239</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs2488602</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123371268</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1167061.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.2126T&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Val709Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.609000e-36</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>15</td>\n",
       "      <td>tolerated(0.71)</td>\n",
       "      <td>benign(0.003)</td>\n",
       "      <td>0.9966</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9914</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9978</td>\n",
       "      <td>0.9921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>V709A</td>\n",
       "      <td>gTg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303240</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs2488602</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123371268</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1167061.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.2126T&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Val709Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.609000e-36</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>15</td>\n",
       "      <td>tolerated(0.72)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.9966</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9914</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9978</td>\n",
       "      <td>0.9921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>V709A</td>\n",
       "      <td>gTg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303241</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs2488602</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123371268</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant,NMD transcript variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1167061.0</td>\n",
       "      <td>ENST00000460253.1</td>\n",
       "      <td>c.1130T&gt;C</td>\n",
       "      <td>ENSP00000435279.1</td>\n",
       "      <td>p.Val377Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.609000e-36</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>15</td>\n",
       "      <td>tolerated(0.66)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.9966</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9914</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9978</td>\n",
       "      <td>0.9921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>3/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>V377A</td>\n",
       "      <td>gTg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314496</th>\n",
       "      <td>AVPR2</td>\n",
       "      <td>AVPR2</td>\n",
       "      <td>8</td>\n",
       "      <td>Diabetic Nephropathy; Diabetic kidney disease</td>\n",
       "      <td>Diabetic Nephropathy; Glomerular Diseases</td>\n",
       "      <td>Diabetes; Glomerular_Diseases</td>\n",
       "      <td>rs5201</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153906539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Nephrogenic syndrome of inappropriate antidiur...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>254775.0</td>\n",
       "      <td>ENST00000430697.1</td>\n",
       "      <td>c.839A&gt;G</td>\n",
       "      <td>ENSP00000393513.1</td>\n",
       "      <td>p.Tyr280Cys</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>71</td>\n",
       "      <td>71</td>\n",
       "      <td>0</td>\n",
       "      <td>71</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.683500e-42</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>24</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>possibly damaging(0.634)</td>\n",
       "      <td>0.6885</td>\n",
       "      <td>0.9372</td>\n",
       "      <td>0.6450</td>\n",
       "      <td>0.7186</td>\n",
       "      <td>0.3185</td>\n",
       "      <td>0.7354</td>\n",
       "      <td>0.4753</td>\n",
       "      <td>0.8754</td>\n",
       "      <td>0.6834</td>\n",
       "      <td>0.3524</td>\n",
       "      <td>0.7177</td>\n",
       "      <td>0.3212</td>\n",
       "      <td>0.2905</td>\n",
       "      <td>0.4248</td>\n",
       "      <td>0.7028</td>\n",
       "      <td>0.5095</td>\n",
       "      <td>0.8647</td>\n",
       "      <td>0.0784</td>\n",
       "      <td>0.5989</td>\n",
       "      <td>0.3470</td>\n",
       "      <td>0.7088</td>\n",
       "      <td>0.3187</td>\n",
       "      <td>0.4852</td>\n",
       "      <td>0.2976</td>\n",
       "      <td>0.5175</td>\n",
       "      <td>0.7193</td>\n",
       "      <td>0.9372</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y280C</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1603 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name     Gene Gene_Score  \\\n",
       "5222       PADI4    PADI4         No   \n",
       "5223       PADI4    PADI4         No   \n",
       "5224       PADI4    PADI4         No   \n",
       "5225       PADI4    PADI4         No   \n",
       "6559     LDLRAP1  LDLRAP1          8   \n",
       "...          ...      ...        ...   \n",
       "303234      CRB2     CRB2          6   \n",
       "303239      CRB2     CRB2          6   \n",
       "303240      CRB2     CRB2          6   \n",
       "303241      CRB2     CRB2          6   \n",
       "314496     AVPR2    AVPR2          8   \n",
       "\n",
       "                                                Condition  \\\n",
       "5222                                 Rheumatoid Arthritis   \n",
       "5223                                 Rheumatoid Arthritis   \n",
       "5224                                 Rheumatoid Arthritis   \n",
       "5225                                 Rheumatoid Arthritis   \n",
       "6559    Polygenic (Familial combined, small dense LDL,...   \n",
       "...                                                   ...   \n",
       "303234                 Focal segmental glomerulosclerosis   \n",
       "303239                 Focal segmental glomerulosclerosis   \n",
       "303240                 Focal segmental glomerulosclerosis   \n",
       "303241                 Focal segmental glomerulosclerosis   \n",
       "314496      Diabetic Nephropathy; Diabetic kidney disease   \n",
       "\n",
       "                                         Headings  \\\n",
       "5222                          Autoimmune Disorder   \n",
       "5223                          Autoimmune Disorder   \n",
       "5224                          Autoimmune Disorder   \n",
       "5225                          Autoimmune Disorder   \n",
       "6559                        Cholesterol disorders   \n",
       "...                                           ...   \n",
       "303234                        Glomerular Diseases   \n",
       "303239                        Glomerular Diseases   \n",
       "303240                        Glomerular Diseases   \n",
       "303241                        Glomerular Diseases   \n",
       "314496  Diabetic Nephropathy; Glomerular Diseases   \n",
       "\n",
       "                          21_Conditions_list        rsID Literature CHROM  \\\n",
       "5222    Arthritis_Degenerative_Joint_Disease  rs11203366         No  chr1   \n",
       "5223    Arthritis_Degenerative_Joint_Disease  rs11203366         No  chr1   \n",
       "5224    Arthritis_Degenerative_Joint_Disease    rs874881         No  chr1   \n",
       "5225    Arthritis_Degenerative_Joint_Disease    rs874881         No  chr1   \n",
       "6559                   Cholesterol_Disorders   rs6687605         No  chr1   \n",
       "...                                      ...         ...        ...   ...   \n",
       "303234                   Glomerular_Diseases   rs1105222         No  chr9   \n",
       "303239                   Glomerular_Diseases   rs2488602         No  chr9   \n",
       "303240                   Glomerular_Diseases   rs2488602         No  chr9   \n",
       "303241                   Glomerular_Diseases   rs2488602         No  chr9   \n",
       "314496         Diabetes; Glomerular_Diseases      rs5201        Yes  chrX   \n",
       "\n",
       "              POS REF ALT      Zygosity  \\\n",
       "5222     17331039   G   A  Heterozygous   \n",
       "5223     17331039   G   A  Heterozygous   \n",
       "5224     17334004   G   C  Heterozygous   \n",
       "5225     17334004   G   C  Heterozygous   \n",
       "6559     25563141   T   C  Heterozygous   \n",
       "...           ...  ..  ..           ...   \n",
       "303234  123365974   G   C  Heterozygous   \n",
       "303239  123371268   T   C    Homozygous   \n",
       "303240  123371268   T   C    Homozygous   \n",
       "303241  123371268   T   C    Homozygous   \n",
       "314496  153906539   A   G    Homozygous   \n",
       "\n",
       "                                    Consequence Consequence_score    IMPACT  \\\n",
       "5222                           missense variant              7/10  MODERATE   \n",
       "5223                           missense variant              7/10  MODERATE   \n",
       "5224                           missense variant              7/10  MODERATE   \n",
       "5225                           missense variant              7/10  MODERATE   \n",
       "6559                           missense variant              7/10  MODERATE   \n",
       "...                                         ...               ...       ...   \n",
       "303234                         missense variant              7/10  MODERATE   \n",
       "303239                         missense variant              7/10  MODERATE   \n",
       "303240                         missense variant              7/10  MODERATE   \n",
       "303241  missense variant,NMD transcript variant              7/10  MODERATE   \n",
       "314496                         missense variant              7/10  MODERATE   \n",
       "\n",
       "        IMPACT_score                                      ClinVar_CLNDN  \\\n",
       "5222             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "5223             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "5224             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "5225             5.0  Rheumatoid arthritis,Abnormal pulmonary inters...   \n",
       "6559             5.0  Cardiovascular phenotype,Hypercholesterolemia,...   \n",
       "...              ...                                                ...   \n",
       "303234           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "303239           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "303240           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "303241           5.0             Ventriculomegaly-cystic kidney disease   \n",
       "314496           5.0  Nephrogenic syndrome of inappropriate antidiur...   \n",
       "\n",
       "           CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "5222    association                     no assertion criteria provided   \n",
       "5223    association                     no assertion criteria provided   \n",
       "5224    association                     no assertion criteria provided   \n",
       "5225    association                     no assertion criteria provided   \n",
       "6559         benign  criteria provided, multiple submitters, no con...   \n",
       "...             ...                                                ...   \n",
       "303234       benign  criteria provided, multiple submitters, no con...   \n",
       "303239       benign  criteria provided, multiple submitters, no con...   \n",
       "303240       benign  criteria provided, multiple submitters, no con...   \n",
       "303241       benign  criteria provided, multiple submitters, no con...   \n",
       "314496       benign  criteria provided, multiple submitters, no con...   \n",
       "\n",
       "          ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "5222     972892.0  ENST00000375448.4           c.163G>A  ENSP00000364597.4   \n",
       "5223     972892.0  ENST00000375453.5           c.163G>A  ENSP00000364602.1   \n",
       "5224     972895.0  ENST00000375448.4           c.335G>C  ENSP00000364597.4   \n",
       "5225     972895.0  ENST00000375453.5           c.335G>C  ENSP00000364602.1   \n",
       "6559     296982.0  ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "...           ...                ...                ...                ...   \n",
       "303234  1166114.0  ENST00000373631.8           c.476G>C  ENSP00000362734.3   \n",
       "303239  1167061.0  ENST00000359999.7          c.2126T>C  ENSP00000353092.3   \n",
       "303240  1167061.0  ENST00000373631.8          c.2126T>C  ENSP00000362734.3   \n",
       "303241  1167061.0  ENST00000460253.1          c.1130T>C  ENSP00000435279.1   \n",
       "314496   254775.0  ENST00000430697.1           c.839A>G  ENSP00000393513.1   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "5222           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5223           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5224          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03   \n",
       "5225          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03   \n",
       "6559          p.Ser202Pro  0/1   44   16  16   5  11  68.75%  3.385400e-05   \n",
       "...                   ...  ...  ...  ...  ..  ..  ..     ...           ...   \n",
       "303234        p.Gly159Ala  0/1   73   41  41  21  20  48.78%  4.334700e-08   \n",
       "303239        p.Val709Ala  1/1  255   61  61   0  61    100%  2.609000e-36   \n",
       "303240        p.Val709Ala  1/1  255   61  61   0  61    100%  2.609000e-36   \n",
       "303241        p.Val377Ala  1/1  255   61  61   0  61    100%  2.609000e-36   \n",
       "314496        p.Tyr280Cys  1/1  255   71  71   0  71    100%  2.683500e-42   \n",
       "\n",
       "        RDF  RDR  ADF  ADR               SIFT                  PolyPhen  \\\n",
       "5222     29    3    7    8       tolerated(1)                 benign(0)   \n",
       "5223     29    3    7    8       tolerated(1)                 benign(0)   \n",
       "5224      7    0    5    3    tolerated(0.67)             benign(0.196)   \n",
       "5225      7    0    5    3     tolerated(0.9)             benign(0.332)   \n",
       "6559      3    2    5    6  deleterious(0.04)             benign(0.037)   \n",
       "...     ...  ...  ...  ...                ...                       ...   \n",
       "303234   12    9   16    4    tolerated(0.76)  probably damaging(0.998)   \n",
       "303239    0    0   46   15    tolerated(0.71)             benign(0.003)   \n",
       "303240    0    0   46   15    tolerated(0.72)                 benign(0)   \n",
       "303241    0    0   46   15    tolerated(0.66)             benign(0.001)   \n",
       "314496    0    0   47   24    tolerated(0.16)  possibly damaging(0.634)   \n",
       "\n",
       "            AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  \\\n",
       "5222    0.5246  0.4508  0.5432  0.5635  0.5755  0.5184      0.5589   \n",
       "5223    0.5246  0.4508  0.5432  0.5635  0.5755  0.5184      0.5589   \n",
       "5224    0.5220  0.4607  0.5403  0.5585  0.5537  0.5215      0.5453   \n",
       "5225    0.5220  0.4607  0.5403  0.5585  0.5537  0.5215      0.5453   \n",
       "6559    0.4349  0.3207  0.5101  0.4425  0.5239  0.4366      0.5054   \n",
       "...        ...     ...     ...     ...     ...     ...         ...   \n",
       "303234  0.3159  0.1989  0.3314  0.3889  0.3728  0.3292      0.3342   \n",
       "303239  0.9966  0.9871  1.0000  1.0000  1.0000  1.0000      0.9994   \n",
       "303240  0.9966  0.9871  1.0000  1.0000  1.0000  1.0000      0.9994   \n",
       "303241  0.9966  0.9871  1.0000  1.0000  1.0000  1.0000      0.9994   \n",
       "314496  0.6885  0.9372  0.6450  0.7186  0.3185  0.7354      0.4753   \n",
       "\n",
       "        gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  \\\n",
       "5222            0.4713          0.5003          0.5952          0.5767   \n",
       "5223            0.4713          0.5003          0.5952          0.5767   \n",
       "5224            0.4790          0.4973          0.5764          0.5781   \n",
       "5225            0.4790          0.4973          0.5764          0.5781   \n",
       "6559            0.3509          0.5516          0.5811          0.4477   \n",
       "...                ...             ...             ...             ...   \n",
       "303234          0.2071          0.2480          0.3128          0.3752   \n",
       "303239          0.9914          0.9999          1.0000          1.0000   \n",
       "303240          0.9914          0.9999          1.0000          1.0000   \n",
       "303241          0.9914          0.9999          1.0000          1.0000   \n",
       "314496          0.8754          0.6834          0.3524          0.7177   \n",
       "\n",
       "        gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  \\\n",
       "5222            0.5898          0.5858          0.5713          0.5220   \n",
       "5223            0.5898          0.5858          0.5713          0.5220   \n",
       "5224            0.5581          0.5635          0.5577          0.5253   \n",
       "5225            0.5581          0.5635          0.5577          0.5253   \n",
       "6559            0.5044          0.5345          0.5366          0.4310   \n",
       "...                ...             ...             ...             ...   \n",
       "303234          0.3427          0.3740          0.3400          0.3327   \n",
       "303239          1.0000          0.9999          0.9997          1.0000   \n",
       "303240          1.0000          0.9999          0.9997          1.0000   \n",
       "303241          1.0000          0.9999          0.9997          1.0000   \n",
       "314496          0.3212          0.2905          0.4248          0.7028   \n",
       "\n",
       "        gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "5222        0.5508          0.4763          0.6490          0.5484   \n",
       "5223        0.5508          0.4763          0.6490          0.5484   \n",
       "5224        0.5361          0.4791          0.6355          0.5395   \n",
       "5225        0.5361          0.4791          0.6355          0.5395   \n",
       "6559        0.4791          0.3543          0.4901          0.5202   \n",
       "...            ...             ...             ...             ...   \n",
       "303234      0.3263          0.2208          0.3352          0.3230   \n",
       "303239      0.9978          0.9921          1.0000          0.9996   \n",
       "303240      0.9978          0.9921          1.0000          0.9996   \n",
       "303241      0.9978          0.9921          1.0000          0.9996   \n",
       "314496      0.5095          0.8647          0.0784          0.5989   \n",
       "\n",
       "        gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "5222            0.6176          0.5931          0.5930          0.5759   \n",
       "5223            0.6176          0.5931          0.5930          0.5759   \n",
       "5224            0.5988          0.5903          0.5603          0.5759   \n",
       "5225            0.5988          0.5903          0.5603          0.5759   \n",
       "6559            0.5858          0.4747          0.5000          0.5573   \n",
       "...                ...             ...             ...             ...   \n",
       "303234          0.3170          0.3788          0.3561          0.3354   \n",
       "303239          1.0000          1.0000          1.0000          1.0000   \n",
       "303240          1.0000          1.0000          1.0000          1.0000   \n",
       "303241          1.0000          1.0000          1.0000          1.0000   \n",
       "314496          0.3470          0.7088          0.3187          0.4852   \n",
       "\n",
       "        gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF  \\\n",
       "5222            0.5840          0.5311          0.5319  0.6490   \n",
       "5223            0.5840          0.5311          0.5319  0.6490   \n",
       "5224            0.5579          0.5207          0.5347  0.6355   \n",
       "5225            0.5579          0.5207          0.5347  0.6355   \n",
       "6559            0.5379          0.5249          0.4449  0.5858   \n",
       "...                ...             ...             ...     ...   \n",
       "303234          0.3825          0.3250          0.3326  0.3889   \n",
       "303239          1.0000          0.9995          1.0000  1.0000   \n",
       "303240          1.0000          0.9995          1.0000  1.0000   \n",
       "303241          1.0000          0.9995          1.0000  1.0000   \n",
       "314496          0.2976          0.5175          0.7193  0.9372   \n",
       "\n",
       "                                              MAX_AF_POPS  \\\n",
       "5222                                          gnomADg AMI   \n",
       "5223                                          gnomADg AMI   \n",
       "5224                                          gnomADg AMI   \n",
       "5225                                          gnomADg AMI   \n",
       "6559                                          gnomADg ASJ   \n",
       "...                                                   ...   \n",
       "303234                                                EAS   \n",
       "303239  AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...   \n",
       "303240  AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...   \n",
       "303241  AMR,EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomAD...   \n",
       "314496                                                AFR   \n",
       "\n",
       "                        BIOTYPE  EXON INTRON Protein Position and Amino Acid  \\\n",
       "5222             protein coding  2/16    NaN                            G55S   \n",
       "5223             protein coding   2/4    NaN                            G55S   \n",
       "5224             protein coding  3/16    NaN                           G112A   \n",
       "5225             protein coding   3/4    NaN                           G112A   \n",
       "6559             protein coding   6/9    NaN                           S202P   \n",
       "...                         ...   ...    ...                             ...   \n",
       "303234           protein coding  3/13    NaN                           G159A   \n",
       "303239           protein coding  8/10    NaN                           V709A   \n",
       "303240           protein coding  8/13    NaN                           V709A   \n",
       "303241  nonsense mediated decay   3/9    NaN                           V377A   \n",
       "314496           protein coding   3/3    NaN                           Y280C   \n",
       "\n",
       "         Codons  STRAND                                             PUBMED  \n",
       "5222    Ggc/Agc     1.0  23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5223    Ggc/Agc     1.0  23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5224    gGg/gCg     1.0  23251581,26843965,21698003,18576335,20537173,2...  \n",
       "5225    gGg/gCg     1.0  23251581,26843965,21698003,18576335,20537173,2...  \n",
       "6559    Tcc/Ccc     1.0       31227787,24906453,31397093,26129832,24067191  \n",
       "...         ...     ...                                                ...  \n",
       "303234  gGc/gCc     1.0                                                NaN  \n",
       "303239  gTg/gCg     1.0                                                NaN  \n",
       "303240  gTg/gCg     1.0                                                NaN  \n",
       "303241  gTg/gCg     1.0                                                NaN  \n",
       "314496  tAc/tGc     1.0                                                NaN  \n",
       "\n",
       "[1603 rows x 74 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dp_x = cons_x[cons_x['DP'] >= 15]\n",
    "dp_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "654881ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5222</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5223</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5224</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.196)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5225</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.9)</td>\n",
       "      <td>benign(0.332)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6559</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypercholesterolemia,...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>68.75%</td>\n",
       "      <td>3.385400e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>0.4349</td>\n",
       "      <td>0.3207</td>\n",
       "      <td>0.5101</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.5239</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.5054</td>\n",
       "      <td>0.3509</td>\n",
       "      <td>0.5516</td>\n",
       "      <td>0.5811</td>\n",
       "      <td>0.4477</td>\n",
       "      <td>0.5044</td>\n",
       "      <td>0.5345</td>\n",
       "      <td>0.5366</td>\n",
       "      <td>0.4310</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.3543</td>\n",
       "      <td>0.4901</td>\n",
       "      <td>0.5202</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.4747</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.5573</td>\n",
       "      <td>0.5379</td>\n",
       "      <td>0.5249</td>\n",
       "      <td>0.4449</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303230</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105223</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365932</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166113.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.434T&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Met145Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>44.12%</td>\n",
       "      <td>4.183500e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3754</td>\n",
       "      <td>0.3714</td>\n",
       "      <td>0.3516</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.3559</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.2612</td>\n",
       "      <td>0.3304</td>\n",
       "      <td>0.3927</td>\n",
       "      <td>0.3463</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.3567</td>\n",
       "      <td>0.3565</td>\n",
       "      <td>0.3758</td>\n",
       "      <td>0.3675</td>\n",
       "      <td>0.3454</td>\n",
       "      <td>0.3448</td>\n",
       "      <td>0.3361</td>\n",
       "      <td>0.3984</td>\n",
       "      <td>0.3581</td>\n",
       "      <td>0.3576</td>\n",
       "      <td>0.3932</td>\n",
       "      <td>0.3634</td>\n",
       "      <td>0.3570</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M145T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28146080,29273807,28869550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303231</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105223</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365932</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166113.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.434T&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Met145Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>44.12%</td>\n",
       "      <td>4.183500e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3754</td>\n",
       "      <td>0.3714</td>\n",
       "      <td>0.3516</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.3559</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.2612</td>\n",
       "      <td>0.3304</td>\n",
       "      <td>0.3927</td>\n",
       "      <td>0.3463</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.3567</td>\n",
       "      <td>0.3565</td>\n",
       "      <td>0.3758</td>\n",
       "      <td>0.3675</td>\n",
       "      <td>0.3454</td>\n",
       "      <td>0.3448</td>\n",
       "      <td>0.3361</td>\n",
       "      <td>0.3984</td>\n",
       "      <td>0.3581</td>\n",
       "      <td>0.3576</td>\n",
       "      <td>0.3932</td>\n",
       "      <td>0.3634</td>\n",
       "      <td>0.3570</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M145T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28146080,29273807,28869550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.7)</td>\n",
       "      <td>possibly damaging(0.52)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.3342</td>\n",
       "      <td>0.2071</td>\n",
       "      <td>0.2480</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>0.3327</td>\n",
       "      <td>0.3263</td>\n",
       "      <td>0.2208</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>0.3170</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.76)</td>\n",
       "      <td>probably damaging(0.998)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.3342</td>\n",
       "      <td>0.2071</td>\n",
       "      <td>0.2480</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>0.3327</td>\n",
       "      <td>0.3263</td>\n",
       "      <td>0.2208</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>0.3170</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314496</th>\n",
       "      <td>AVPR2</td>\n",
       "      <td>AVPR2</td>\n",
       "      <td>8</td>\n",
       "      <td>Diabetic Nephropathy; Diabetic kidney disease</td>\n",
       "      <td>Diabetic Nephropathy; Glomerular Diseases</td>\n",
       "      <td>Diabetes; Glomerular_Diseases</td>\n",
       "      <td>rs5201</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153906539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Nephrogenic syndrome of inappropriate antidiur...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>254775.0</td>\n",
       "      <td>ENST00000430697.1</td>\n",
       "      <td>c.839A&gt;G</td>\n",
       "      <td>ENSP00000393513.1</td>\n",
       "      <td>p.Tyr280Cys</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>71</td>\n",
       "      <td>71</td>\n",
       "      <td>0</td>\n",
       "      <td>71</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.683500e-42</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>24</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>possibly damaging(0.634)</td>\n",
       "      <td>0.6885</td>\n",
       "      <td>0.9372</td>\n",
       "      <td>0.6450</td>\n",
       "      <td>0.7186</td>\n",
       "      <td>0.3185</td>\n",
       "      <td>0.7354</td>\n",
       "      <td>0.4753</td>\n",
       "      <td>0.8754</td>\n",
       "      <td>0.6834</td>\n",
       "      <td>0.3524</td>\n",
       "      <td>0.7177</td>\n",
       "      <td>0.3212</td>\n",
       "      <td>0.2905</td>\n",
       "      <td>0.4248</td>\n",
       "      <td>0.7028</td>\n",
       "      <td>0.5095</td>\n",
       "      <td>0.8647</td>\n",
       "      <td>0.0784</td>\n",
       "      <td>0.5989</td>\n",
       "      <td>0.3470</td>\n",
       "      <td>0.7088</td>\n",
       "      <td>0.3187</td>\n",
       "      <td>0.4852</td>\n",
       "      <td>0.2976</td>\n",
       "      <td>0.5175</td>\n",
       "      <td>0.7193</td>\n",
       "      <td>0.9372</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y280C</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1156 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name     Gene Gene_Score  \\\n",
       "5222       PADI4    PADI4         No   \n",
       "5223       PADI4    PADI4         No   \n",
       "5224       PADI4    PADI4         No   \n",
       "5225       PADI4    PADI4         No   \n",
       "6559     LDLRAP1  LDLRAP1          8   \n",
       "...          ...      ...        ...   \n",
       "303230      CRB2     CRB2          6   \n",
       "303231      CRB2     CRB2          6   \n",
       "303233      CRB2     CRB2          6   \n",
       "303234      CRB2     CRB2          6   \n",
       "314496     AVPR2    AVPR2          8   \n",
       "\n",
       "                                                Condition  \\\n",
       "5222                                 Rheumatoid Arthritis   \n",
       "5223                                 Rheumatoid Arthritis   \n",
       "5224                                 Rheumatoid Arthritis   \n",
       "5225                                 Rheumatoid Arthritis   \n",
       "6559    Polygenic (Familial combined, small dense LDL,...   \n",
       "...                                                   ...   \n",
       "303230                 Focal segmental glomerulosclerosis   \n",
       "303231                 Focal segmental glomerulosclerosis   \n",
       "303233                 Focal segmental glomerulosclerosis   \n",
       "303234                 Focal segmental glomerulosclerosis   \n",
       "314496      Diabetic Nephropathy; Diabetic kidney disease   \n",
       "\n",
       "                                         Headings  \\\n",
       "5222                          Autoimmune Disorder   \n",
       "5223                          Autoimmune Disorder   \n",
       "5224                          Autoimmune Disorder   \n",
       "5225                          Autoimmune Disorder   \n",
       "6559                        Cholesterol disorders   \n",
       "...                                           ...   \n",
       "303230                        Glomerular Diseases   \n",
       "303231                        Glomerular Diseases   \n",
       "303233                        Glomerular Diseases   \n",
       "303234                        Glomerular Diseases   \n",
       "314496  Diabetic Nephropathy; Glomerular Diseases   \n",
       "\n",
       "                          21_Conditions_list        rsID Literature CHROM  \\\n",
       "5222    Arthritis_Degenerative_Joint_Disease  rs11203366         No  chr1   \n",
       "5223    Arthritis_Degenerative_Joint_Disease  rs11203366         No  chr1   \n",
       "5224    Arthritis_Degenerative_Joint_Disease    rs874881         No  chr1   \n",
       "5225    Arthritis_Degenerative_Joint_Disease    rs874881         No  chr1   \n",
       "6559                   Cholesterol_Disorders   rs6687605         No  chr1   \n",
       "...                                      ...         ...        ...   ...   \n",
       "303230                   Glomerular_Diseases   rs1105223        Yes  chr9   \n",
       "303231                   Glomerular_Diseases   rs1105223        Yes  chr9   \n",
       "303233                   Glomerular_Diseases   rs1105222         No  chr9   \n",
       "303234                   Glomerular_Diseases   rs1105222         No  chr9   \n",
       "314496         Diabetes; Glomerular_Diseases      rs5201        Yes  chrX   \n",
       "\n",
       "              POS REF ALT      Zygosity       Consequence Consequence_score  \\\n",
       "5222     17331039   G   A  Heterozygous  missense variant              7/10   \n",
       "5223     17331039   G   A  Heterozygous  missense variant              7/10   \n",
       "5224     17334004   G   C  Heterozygous  missense variant              7/10   \n",
       "5225     17334004   G   C  Heterozygous  missense variant              7/10   \n",
       "6559     25563141   T   C  Heterozygous  missense variant              7/10   \n",
       "...           ...  ..  ..           ...               ...               ...   \n",
       "303230  123365932   T   C  Heterozygous  missense variant              7/10   \n",
       "303231  123365932   T   C  Heterozygous  missense variant              7/10   \n",
       "303233  123365974   G   C  Heterozygous  missense variant              7/10   \n",
       "303234  123365974   G   C  Heterozygous  missense variant              7/10   \n",
       "314496  153906539   A   G    Homozygous  missense variant              7/10   \n",
       "\n",
       "          IMPACT  IMPACT_score  \\\n",
       "5222    MODERATE           5.0   \n",
       "5223    MODERATE           5.0   \n",
       "5224    MODERATE           5.0   \n",
       "5225    MODERATE           5.0   \n",
       "6559    MODERATE           5.0   \n",
       "...          ...           ...   \n",
       "303230  MODERATE           5.0   \n",
       "303231  MODERATE           5.0   \n",
       "303233  MODERATE           5.0   \n",
       "303234  MODERATE           5.0   \n",
       "314496  MODERATE           5.0   \n",
       "\n",
       "                                            ClinVar_CLNDN     CLIN_SIG  \\\n",
       "5222    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "5223    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "5224    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "5225    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "6559    Cardiovascular phenotype,Hypercholesterolemia,...       benign   \n",
       "...                                                   ...          ...   \n",
       "303230             Ventriculomegaly-cystic kidney disease       benign   \n",
       "303231             Ventriculomegaly-cystic kidney disease       benign   \n",
       "303233             Ventriculomegaly-cystic kidney disease       benign   \n",
       "303234             Ventriculomegaly-cystic kidney disease       benign   \n",
       "314496  Nephrogenic syndrome of inappropriate antidiur...       benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT    ClinVar  \\\n",
       "5222                       no assertion criteria provided   972892.0   \n",
       "5223                       no assertion criteria provided   972892.0   \n",
       "5224                       no assertion criteria provided   972895.0   \n",
       "5225                       no assertion criteria provided   972895.0   \n",
       "6559    criteria provided, multiple submitters, no con...   296982.0   \n",
       "...                                                   ...        ...   \n",
       "303230  criteria provided, multiple submitters, no con...  1166113.0   \n",
       "303231  criteria provided, multiple submitters, no con...  1166113.0   \n",
       "303233  criteria provided, multiple submitters, no con...  1166114.0   \n",
       "303234  criteria provided, multiple submitters, no con...  1166114.0   \n",
       "314496  criteria provided, multiple submitters, no con...   254775.0   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "5222    ENST00000375448.4           c.163G>A  ENSP00000364597.4   \n",
       "5223    ENST00000375453.5           c.163G>A  ENSP00000364602.1   \n",
       "5224    ENST00000375448.4           c.335G>C  ENSP00000364597.4   \n",
       "5225    ENST00000375453.5           c.335G>C  ENSP00000364602.1   \n",
       "6559    ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "...                   ...                ...                ...   \n",
       "303230  ENST00000359999.7           c.434T>C  ENSP00000353092.3   \n",
       "303231  ENST00000373631.8           c.434T>C  ENSP00000362734.3   \n",
       "303233  ENST00000359999.7           c.476G>C  ENSP00000353092.3   \n",
       "303234  ENST00000373631.8           c.476G>C  ENSP00000362734.3   \n",
       "314496  ENST00000430697.1           c.839A>G  ENSP00000393513.1   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "5222           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5223           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5224          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03   \n",
       "5225          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03   \n",
       "6559          p.Ser202Pro  0/1   44   16  16   5  11  68.75%  3.385400e-05   \n",
       "...                   ...  ...  ...  ...  ..  ..  ..     ...           ...   \n",
       "303230        p.Met145Thr  0/1   53   34  34  19  15  44.12%  4.183500e-06   \n",
       "303231        p.Met145Thr  0/1   53   34  34  19  15  44.12%  4.183500e-06   \n",
       "303233        p.Gly159Ala  0/1   73   41  41  21  20  48.78%  4.334700e-08   \n",
       "303234        p.Gly159Ala  0/1   73   41  41  21  20  48.78%  4.334700e-08   \n",
       "314496        p.Tyr280Cys  1/1  255   71  71   0  71    100%  2.683500e-42   \n",
       "\n",
       "        RDF  RDR  ADF  ADR               SIFT                  PolyPhen  \\\n",
       "5222     29    3    7    8       tolerated(1)                 benign(0)   \n",
       "5223     29    3    7    8       tolerated(1)                 benign(0)   \n",
       "5224      7    0    5    3    tolerated(0.67)             benign(0.196)   \n",
       "5225      7    0    5    3     tolerated(0.9)             benign(0.332)   \n",
       "6559      3    2    5    6  deleterious(0.04)             benign(0.037)   \n",
       "...     ...  ...  ...  ...                ...                       ...   \n",
       "303230   10    9   11    4    tolerated(0.82)                 benign(0)   \n",
       "303231   10    9   11    4    tolerated(0.67)                 benign(0)   \n",
       "303233   12    9   16    4     tolerated(0.7)   possibly damaging(0.52)   \n",
       "303234   12    9   16    4    tolerated(0.76)  probably damaging(0.998)   \n",
       "314496    0    0   47   24    tolerated(0.16)  possibly damaging(0.634)   \n",
       "\n",
       "            AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  \\\n",
       "5222    0.5246  0.4508  0.5432  0.5635  0.5755  0.5184      0.5589   \n",
       "5223    0.5246  0.4508  0.5432  0.5635  0.5755  0.5184      0.5589   \n",
       "5224    0.5220  0.4607  0.5403  0.5585  0.5537  0.5215      0.5453   \n",
       "5225    0.5220  0.4607  0.5403  0.5585  0.5537  0.5215      0.5453   \n",
       "6559    0.4349  0.3207  0.5101  0.4425  0.5239  0.4366      0.5054   \n",
       "...        ...     ...     ...     ...     ...     ...         ...   \n",
       "303230  0.3754  0.3714  0.3516  0.4107  0.3817  0.3548      0.3559   \n",
       "303231  0.3754  0.3714  0.3516  0.4107  0.3817  0.3548      0.3559   \n",
       "303233  0.3159  0.1989  0.3314  0.3889  0.3728  0.3292      0.3342   \n",
       "303234  0.3159  0.1989  0.3314  0.3889  0.3728  0.3292      0.3342   \n",
       "314496  0.6885  0.9372  0.6450  0.7186  0.3185  0.7354      0.4753   \n",
       "\n",
       "        gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  \\\n",
       "5222            0.4713          0.5003          0.5952          0.5767   \n",
       "5223            0.4713          0.5003          0.5952          0.5767   \n",
       "5224            0.4790          0.4973          0.5764          0.5781   \n",
       "5225            0.4790          0.4973          0.5764          0.5781   \n",
       "6559            0.3509          0.5516          0.5811          0.4477   \n",
       "...                ...             ...             ...             ...   \n",
       "303230          0.3561          0.2612          0.3304          0.3927   \n",
       "303231          0.3561          0.2612          0.3304          0.3927   \n",
       "303233          0.2071          0.2480          0.3128          0.3752   \n",
       "303234          0.2071          0.2480          0.3128          0.3752   \n",
       "314496          0.8754          0.6834          0.3524          0.7177   \n",
       "\n",
       "        gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  \\\n",
       "5222            0.5898          0.5858          0.5713          0.5220   \n",
       "5223            0.5898          0.5858          0.5713          0.5220   \n",
       "5224            0.5581          0.5635          0.5577          0.5253   \n",
       "5225            0.5581          0.5635          0.5577          0.5253   \n",
       "6559            0.5044          0.5345          0.5366          0.4310   \n",
       "...                ...             ...             ...             ...   \n",
       "303230          0.3463          0.3835          0.3567          0.3565   \n",
       "303231          0.3463          0.3835          0.3567          0.3565   \n",
       "303233          0.3427          0.3740          0.3400          0.3327   \n",
       "303234          0.3427          0.3740          0.3400          0.3327   \n",
       "314496          0.3212          0.2905          0.4248          0.7028   \n",
       "\n",
       "        gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "5222        0.5508          0.4763          0.6490          0.5484   \n",
       "5223        0.5508          0.4763          0.6490          0.5484   \n",
       "5224        0.5361          0.4791          0.6355          0.5395   \n",
       "5225        0.5361          0.4791          0.6355          0.5395   \n",
       "6559        0.4791          0.3543          0.4901          0.5202   \n",
       "...            ...             ...             ...             ...   \n",
       "303230      0.3758          0.3675          0.3454          0.3448   \n",
       "303231      0.3758          0.3675          0.3454          0.3448   \n",
       "303233      0.3263          0.2208          0.3352          0.3230   \n",
       "303234      0.3263          0.2208          0.3352          0.3230   \n",
       "314496      0.5095          0.8647          0.0784          0.5989   \n",
       "\n",
       "        gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "5222            0.6176          0.5931          0.5930          0.5759   \n",
       "5223            0.6176          0.5931          0.5930          0.5759   \n",
       "5224            0.5988          0.5903          0.5603          0.5759   \n",
       "5225            0.5988          0.5903          0.5603          0.5759   \n",
       "6559            0.5858          0.4747          0.5000          0.5573   \n",
       "...                ...             ...             ...             ...   \n",
       "303230          0.3361          0.3984          0.3581          0.3576   \n",
       "303231          0.3361          0.3984          0.3581          0.3576   \n",
       "303233          0.3170          0.3788          0.3561          0.3354   \n",
       "303234          0.3170          0.3788          0.3561          0.3354   \n",
       "314496          0.3470          0.7088          0.3187          0.4852   \n",
       "\n",
       "        gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "5222            0.5840          0.5311          0.5319  0.6490  gnomADg AMI   \n",
       "5223            0.5840          0.5311          0.5319  0.6490  gnomADg AMI   \n",
       "5224            0.5579          0.5207          0.5347  0.6355  gnomADg AMI   \n",
       "5225            0.5579          0.5207          0.5347  0.6355  gnomADg AMI   \n",
       "6559            0.5379          0.5249          0.4449  0.5858  gnomADg ASJ   \n",
       "...                ...             ...             ...     ...          ...   \n",
       "303230          0.3932          0.3634          0.3570  0.4107          EAS   \n",
       "303231          0.3932          0.3634          0.3570  0.4107          EAS   \n",
       "303233          0.3825          0.3250          0.3326  0.3889          EAS   \n",
       "303234          0.3825          0.3250          0.3326  0.3889          EAS   \n",
       "314496          0.2976          0.5175          0.7193  0.9372          AFR   \n",
       "\n",
       "               BIOTYPE  EXON INTRON Protein Position and Amino Acid   Codons  \\\n",
       "5222    protein coding  2/16    NaN                            G55S  Ggc/Agc   \n",
       "5223    protein coding   2/4    NaN                            G55S  Ggc/Agc   \n",
       "5224    protein coding  3/16    NaN                           G112A  gGg/gCg   \n",
       "5225    protein coding   3/4    NaN                           G112A  gGg/gCg   \n",
       "6559    protein coding   6/9    NaN                           S202P  Tcc/Ccc   \n",
       "...                ...   ...    ...                             ...      ...   \n",
       "303230  protein coding  3/10    NaN                           M145T  aTg/aCg   \n",
       "303231  protein coding  3/13    NaN                           M145T  aTg/aCg   \n",
       "303233  protein coding  3/10    NaN                           G159A  gGc/gCc   \n",
       "303234  protein coding  3/13    NaN                           G159A  gGc/gCc   \n",
       "314496  protein coding   3/3    NaN                           Y280C  tAc/tGc   \n",
       "\n",
       "        STRAND                                             PUBMED  \n",
       "5222       1.0  23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5223       1.0  23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5224       1.0  23251581,26843965,21698003,18576335,20537173,2...  \n",
       "5225       1.0  23251581,26843965,21698003,18576335,20537173,2...  \n",
       "6559       1.0       31227787,24906453,31397093,26129832,24067191  \n",
       "...        ...                                                ...  \n",
       "303230     1.0                         28146080,29273807,28869550  \n",
       "303231     1.0                         28146080,29273807,28869550  \n",
       "303233     1.0                                                NaN  \n",
       "303234     1.0                                                NaN  \n",
       "314496     1.0                                                NaN  \n",
       "\n",
       "[1156 rows x 74 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomADe_AF_x = dp_x[dp_x['gnomADe_AF'] <= 0.6]\n",
    "gnomADe_AF_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "7c785a7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5222</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5223</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5224</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.196)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5225</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375453.5</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364602.1</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.9)</td>\n",
       "      <td>benign(0.332)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6559</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypercholesterolemia,...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>68.75%</td>\n",
       "      <td>3.385400e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>0.4349</td>\n",
       "      <td>0.3207</td>\n",
       "      <td>0.5101</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.5239</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.5054</td>\n",
       "      <td>0.3509</td>\n",
       "      <td>0.5516</td>\n",
       "      <td>0.5811</td>\n",
       "      <td>0.4477</td>\n",
       "      <td>0.5044</td>\n",
       "      <td>0.5345</td>\n",
       "      <td>0.5366</td>\n",
       "      <td>0.4310</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.3543</td>\n",
       "      <td>0.4901</td>\n",
       "      <td>0.5202</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.4747</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.5573</td>\n",
       "      <td>0.5379</td>\n",
       "      <td>0.5249</td>\n",
       "      <td>0.4449</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301570</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cardiac_Health; Cholesterol_Disorders; Gall_st...</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000678995.1</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000504612.1</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.6301</td>\n",
       "      <td>0.3249</td>\n",
       "      <td>0.2806</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.2196</td>\n",
       "      <td>0.2742</td>\n",
       "      <td>0.3178</td>\n",
       "      <td>0.3676</td>\n",
       "      <td>0.3838</td>\n",
       "      <td>0.6250</td>\n",
       "      <td>0.4473</td>\n",
       "      <td>0.3462</td>\n",
       "      <td>0.2764</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.2218</td>\n",
       "      <td>0.3956</td>\n",
       "      <td>0.2729</td>\n",
       "      <td>0.3768</td>\n",
       "      <td>0.3726</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303230</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105223</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365932</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166113.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.434T&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Met145Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>44.12%</td>\n",
       "      <td>4.183500e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3754</td>\n",
       "      <td>0.3714</td>\n",
       "      <td>0.3516</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.3559</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.2612</td>\n",
       "      <td>0.3304</td>\n",
       "      <td>0.3927</td>\n",
       "      <td>0.3463</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.3567</td>\n",
       "      <td>0.3565</td>\n",
       "      <td>0.3758</td>\n",
       "      <td>0.3675</td>\n",
       "      <td>0.3454</td>\n",
       "      <td>0.3448</td>\n",
       "      <td>0.3361</td>\n",
       "      <td>0.3984</td>\n",
       "      <td>0.3581</td>\n",
       "      <td>0.3576</td>\n",
       "      <td>0.3932</td>\n",
       "      <td>0.3634</td>\n",
       "      <td>0.3570</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M145T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28146080,29273807,28869550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303231</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105223</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365932</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166113.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.434T&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Met145Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>44.12%</td>\n",
       "      <td>4.183500e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3754</td>\n",
       "      <td>0.3714</td>\n",
       "      <td>0.3516</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.3559</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.2612</td>\n",
       "      <td>0.3304</td>\n",
       "      <td>0.3927</td>\n",
       "      <td>0.3463</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.3567</td>\n",
       "      <td>0.3565</td>\n",
       "      <td>0.3758</td>\n",
       "      <td>0.3675</td>\n",
       "      <td>0.3454</td>\n",
       "      <td>0.3448</td>\n",
       "      <td>0.3361</td>\n",
       "      <td>0.3984</td>\n",
       "      <td>0.3581</td>\n",
       "      <td>0.3576</td>\n",
       "      <td>0.3932</td>\n",
       "      <td>0.3634</td>\n",
       "      <td>0.3570</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M145T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28146080,29273807,28869550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.7)</td>\n",
       "      <td>possibly damaging(0.52)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.3342</td>\n",
       "      <td>0.2071</td>\n",
       "      <td>0.2480</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>0.3327</td>\n",
       "      <td>0.3263</td>\n",
       "      <td>0.2208</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>0.3170</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303234</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000373631.8</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000362734.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.76)</td>\n",
       "      <td>probably damaging(0.998)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.3342</td>\n",
       "      <td>0.2071</td>\n",
       "      <td>0.2480</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>0.3327</td>\n",
       "      <td>0.3263</td>\n",
       "      <td>0.2208</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>0.3170</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1157 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name     Gene Gene_Score  \\\n",
       "5222       PADI4    PADI4         No   \n",
       "5223       PADI4    PADI4         No   \n",
       "5224       PADI4    PADI4         No   \n",
       "5225       PADI4    PADI4         No   \n",
       "6559     LDLRAP1  LDLRAP1          8   \n",
       "...          ...      ...        ...   \n",
       "301570     ABCA1    ABCA1          7   \n",
       "303230      CRB2     CRB2          6   \n",
       "303231      CRB2     CRB2          6   \n",
       "303233      CRB2     CRB2          6   \n",
       "303234      CRB2     CRB2          6   \n",
       "\n",
       "                                                Condition  \\\n",
       "5222                                 Rheumatoid Arthritis   \n",
       "5223                                 Rheumatoid Arthritis   \n",
       "5224                                 Rheumatoid Arthritis   \n",
       "5225                                 Rheumatoid Arthritis   \n",
       "6559    Polygenic (Familial combined, small dense LDL,...   \n",
       "...                                                   ...   \n",
       "301570  Coronary artery disease risk; Polygenic (Famil...   \n",
       "303230                 Focal segmental glomerulosclerosis   \n",
       "303231                 Focal segmental glomerulosclerosis   \n",
       "303233                 Focal segmental glomerulosclerosis   \n",
       "303234                 Focal segmental glomerulosclerosis   \n",
       "\n",
       "                                          Headings  \\\n",
       "5222                           Autoimmune Disorder   \n",
       "5223                           Autoimmune Disorder   \n",
       "5224                           Autoimmune Disorder   \n",
       "5225                           Autoimmune Disorder   \n",
       "6559                         Cholesterol disorders   \n",
       "...                                            ...   \n",
       "301570  Cardiac; Cholesterol disorders; Gallstones   \n",
       "303230                         Glomerular Diseases   \n",
       "303231                         Glomerular Diseases   \n",
       "303233                         Glomerular Diseases   \n",
       "303234                         Glomerular Diseases   \n",
       "\n",
       "                                       21_Conditions_list        rsID  \\\n",
       "5222                 Arthritis_Degenerative_Joint_Disease  rs11203366   \n",
       "5223                 Arthritis_Degenerative_Joint_Disease  rs11203366   \n",
       "5224                 Arthritis_Degenerative_Joint_Disease    rs874881   \n",
       "5225                 Arthritis_Degenerative_Joint_Disease    rs874881   \n",
       "6559                                Cholesterol_Disorders   rs6687605   \n",
       "...                                                   ...         ...   \n",
       "301570  Cardiac_Health; Cholesterol_Disorders; Gall_st...   rs2230806   \n",
       "303230                                Glomerular_Diseases   rs1105223   \n",
       "303231                                Glomerular_Diseases   rs1105223   \n",
       "303233                                Glomerular_Diseases   rs1105222   \n",
       "303234                                Glomerular_Diseases   rs1105222   \n",
       "\n",
       "       Literature CHROM        POS REF ALT      Zygosity       Consequence  \\\n",
       "5222           No  chr1   17331039   G   A  Heterozygous  missense variant   \n",
       "5223           No  chr1   17331039   G   A  Heterozygous  missense variant   \n",
       "5224           No  chr1   17334004   G   C  Heterozygous  missense variant   \n",
       "5225           No  chr1   17334004   G   C  Heterozygous  missense variant   \n",
       "6559           No  chr1   25563141   T   C  Heterozygous  missense variant   \n",
       "...           ...   ...        ...  ..  ..           ...               ...   \n",
       "301570        Yes  chr9  104858586   C   T  Heterozygous  missense variant   \n",
       "303230        Yes  chr9  123365932   T   C  Heterozygous  missense variant   \n",
       "303231        Yes  chr9  123365932   T   C  Heterozygous  missense variant   \n",
       "303233         No  chr9  123365974   G   C  Heterozygous  missense variant   \n",
       "303234         No  chr9  123365974   G   C  Heterozygous  missense variant   \n",
       "\n",
       "       Consequence_score    IMPACT  IMPACT_score  \\\n",
       "5222                7/10  MODERATE           5.0   \n",
       "5223                7/10  MODERATE           5.0   \n",
       "5224                7/10  MODERATE           5.0   \n",
       "5225                7/10  MODERATE           5.0   \n",
       "6559                7/10  MODERATE           5.0   \n",
       "...                  ...       ...           ...   \n",
       "301570              7/10  MODERATE           5.0   \n",
       "303230              7/10  MODERATE           5.0   \n",
       "303231              7/10  MODERATE           5.0   \n",
       "303233              7/10  MODERATE           5.0   \n",
       "303234              7/10  MODERATE           5.0   \n",
       "\n",
       "                                            ClinVar_CLNDN     CLIN_SIG  \\\n",
       "5222    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "5223    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "5224    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "5225    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "6559    Cardiovascular phenotype,Hypercholesterolemia,...       benign   \n",
       "...                                                   ...          ...   \n",
       "301570  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...       benign   \n",
       "303230             Ventriculomegaly-cystic kidney disease       benign   \n",
       "303231             Ventriculomegaly-cystic kidney disease       benign   \n",
       "303233             Ventriculomegaly-cystic kidney disease       benign   \n",
       "303234             Ventriculomegaly-cystic kidney disease       benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT    ClinVar  \\\n",
       "5222                       no assertion criteria provided   972892.0   \n",
       "5223                       no assertion criteria provided   972892.0   \n",
       "5224                       no assertion criteria provided   972895.0   \n",
       "5225                       no assertion criteria provided   972895.0   \n",
       "6559    criteria provided, multiple submitters, no con...   296982.0   \n",
       "...                                                   ...        ...   \n",
       "301570  criteria provided, multiple submitters, no con...     9506.0   \n",
       "303230  criteria provided, multiple submitters, no con...  1166113.0   \n",
       "303231  criteria provided, multiple submitters, no con...  1166113.0   \n",
       "303233  criteria provided, multiple submitters, no con...  1166114.0   \n",
       "303234  criteria provided, multiple submitters, no con...  1166114.0   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "5222    ENST00000375448.4           c.163G>A  ENSP00000364597.4   \n",
       "5223    ENST00000375453.5           c.163G>A  ENSP00000364602.1   \n",
       "5224    ENST00000375448.4           c.335G>C  ENSP00000364597.4   \n",
       "5225    ENST00000375453.5           c.335G>C  ENSP00000364602.1   \n",
       "6559    ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "...                   ...                ...                ...   \n",
       "301570  ENST00000678995.1           c.656G>A  ENSP00000504612.1   \n",
       "303230  ENST00000359999.7           c.434T>C  ENSP00000353092.3   \n",
       "303231  ENST00000373631.8           c.434T>C  ENSP00000362734.3   \n",
       "303233  ENST00000359999.7           c.476G>C  ENSP00000353092.3   \n",
       "303234  ENST00000373631.8           c.476G>C  ENSP00000362734.3   \n",
       "\n",
       "       HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "5222           p.Gly55Ser  0/1  50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5223           p.Gly55Ser  0/1  50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5224          p.Gly112Ala  0/1  29   15  15   7   8  53.33%  1.099500e-03   \n",
       "5225          p.Gly112Ala  0/1  29   15  15   7   8  53.33%  1.099500e-03   \n",
       "6559          p.Ser202Pro  0/1  44   16  16   5  11  68.75%  3.385400e-05   \n",
       "...                   ...  ...  ..  ...  ..  ..  ..     ...           ...   \n",
       "301570        p.Arg219Lys  0/1  69   53  53  33  20  37.74%  1.040500e-07   \n",
       "303230        p.Met145Thr  0/1  53   34  34  19  15  44.12%  4.183500e-06   \n",
       "303231        p.Met145Thr  0/1  53   34  34  19  15  44.12%  4.183500e-06   \n",
       "303233        p.Gly159Ala  0/1  73   41  41  21  20  48.78%  4.334700e-08   \n",
       "303234        p.Gly159Ala  0/1  73   41  41  21  20  48.78%  4.334700e-08   \n",
       "\n",
       "        RDF  RDR  ADF  ADR                         SIFT  \\\n",
       "5222     29    3    7    8                 tolerated(1)   \n",
       "5223     29    3    7    8                 tolerated(1)   \n",
       "5224      7    0    5    3              tolerated(0.67)   \n",
       "5225      7    0    5    3               tolerated(0.9)   \n",
       "6559      3    2    5    6            deleterious(0.04)   \n",
       "...     ...  ...  ...  ...                          ...   \n",
       "301570   29    4   15    5  tolerated low confidence(1)   \n",
       "303230   10    9   11    4              tolerated(0.82)   \n",
       "303231   10    9   11    4              tolerated(0.67)   \n",
       "303233   12    9   16    4               tolerated(0.7)   \n",
       "303234   12    9   16    4              tolerated(0.76)   \n",
       "\n",
       "                        PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  \\\n",
       "5222                   benign(0)  0.5246  0.4508  0.5432  0.5635  0.5755   \n",
       "5223                   benign(0)  0.5246  0.4508  0.5432  0.5635  0.5755   \n",
       "5224               benign(0.196)  0.5220  0.4607  0.5403  0.5585  0.5537   \n",
       "5225               benign(0.332)  0.5220  0.4607  0.5403  0.5585  0.5537   \n",
       "6559               benign(0.037)  0.4349  0.3207  0.5101  0.4425  0.5239   \n",
       "...                          ...     ...     ...     ...     ...     ...   \n",
       "301570                 benign(0)  0.4397  0.7103  0.3487  0.4147  0.2425   \n",
       "303230                 benign(0)  0.3754  0.3714  0.3516  0.4107  0.3817   \n",
       "303231                 benign(0)  0.3754  0.3714  0.3516  0.4107  0.3817   \n",
       "303233   possibly damaging(0.52)  0.3159  0.1989  0.3314  0.3889  0.3728   \n",
       "303234  probably damaging(0.998)  0.3159  0.1989  0.3314  0.3889  0.3728   \n",
       "\n",
       "        SAS_AF  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  \\\n",
       "5222    0.5184      0.5589          0.4713          0.5003          0.5952   \n",
       "5223    0.5184      0.5589          0.4713          0.5003          0.5952   \n",
       "5224    0.5215      0.5453          0.4790          0.4973          0.5764   \n",
       "5225    0.5215      0.5453          0.4790          0.4973          0.5764   \n",
       "6559    0.4366      0.5054          0.3509          0.5516          0.5811   \n",
       "...        ...         ...             ...             ...             ...   \n",
       "301570  0.3671      0.3236          0.6301          0.3249          0.2806   \n",
       "303230  0.3548      0.3559          0.3561          0.2612          0.3304   \n",
       "303231  0.3548      0.3559          0.3561          0.2612          0.3304   \n",
       "303233  0.3292      0.3342          0.2071          0.2480          0.3128   \n",
       "303234  0.3292      0.3342          0.2071          0.2480          0.3128   \n",
       "\n",
       "        gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  \\\n",
       "5222            0.5767          0.5898          0.5858          0.5713   \n",
       "5223            0.5767          0.5898          0.5858          0.5713   \n",
       "5224            0.5781          0.5581          0.5635          0.5577   \n",
       "5225            0.5781          0.5581          0.5635          0.5577   \n",
       "6559            0.4477          0.5044          0.5345          0.5366   \n",
       "...                ...             ...             ...             ...   \n",
       "301570          0.4309          0.2196          0.2742          0.3178   \n",
       "303230          0.3927          0.3463          0.3835          0.3567   \n",
       "303231          0.3927          0.3463          0.3835          0.3567   \n",
       "303233          0.3752          0.3427          0.3740          0.3400   \n",
       "303234          0.3752          0.3427          0.3740          0.3400   \n",
       "\n",
       "        gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  \\\n",
       "5222            0.5220      0.5508          0.4763          0.6490   \n",
       "5223            0.5220      0.5508          0.4763          0.6490   \n",
       "5224            0.5253      0.5361          0.4791          0.6355   \n",
       "5225            0.5253      0.5361          0.4791          0.6355   \n",
       "6559            0.4310      0.4791          0.3543          0.4901   \n",
       "...                ...         ...             ...             ...   \n",
       "301570          0.3676      0.3838          0.6250          0.4473   \n",
       "303230          0.3565      0.3758          0.3675          0.3454   \n",
       "303231          0.3565      0.3758          0.3675          0.3454   \n",
       "303233          0.3327      0.3263          0.2208          0.3352   \n",
       "303234          0.3327      0.3263          0.2208          0.3352   \n",
       "\n",
       "        gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  \\\n",
       "5222            0.5484          0.6176          0.5931          0.5930   \n",
       "5223            0.5484          0.6176          0.5931          0.5930   \n",
       "5224            0.5395          0.5988          0.5903          0.5603   \n",
       "5225            0.5395          0.5988          0.5903          0.5603   \n",
       "6559            0.5202          0.5858          0.4747          0.5000   \n",
       "...                ...             ...             ...             ...   \n",
       "301570          0.3462          0.2764          0.4273          0.2218   \n",
       "303230          0.3448          0.3361          0.3984          0.3581   \n",
       "303231          0.3448          0.3361          0.3984          0.3581   \n",
       "303233          0.3230          0.3170          0.3788          0.3561   \n",
       "303234          0.3230          0.3170          0.3788          0.3561   \n",
       "\n",
       "        gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  \\\n",
       "5222            0.5759          0.5840          0.5311          0.5319   \n",
       "5223            0.5759          0.5840          0.5311          0.5319   \n",
       "5224            0.5759          0.5579          0.5207          0.5347   \n",
       "5225            0.5759          0.5579          0.5207          0.5347   \n",
       "6559            0.5573          0.5379          0.5249          0.4449   \n",
       "...                ...             ...             ...             ...   \n",
       "301570          0.3956          0.2729          0.3768          0.3726   \n",
       "303230          0.3576          0.3932          0.3634          0.3570   \n",
       "303231          0.3576          0.3932          0.3634          0.3570   \n",
       "303233          0.3354          0.3825          0.3250          0.3326   \n",
       "303234          0.3354          0.3825          0.3250          0.3326   \n",
       "\n",
       "        MAX_AF  MAX_AF_POPS         BIOTYPE  EXON INTRON  \\\n",
       "5222    0.6490  gnomADg AMI  protein coding  2/16    NaN   \n",
       "5223    0.6490  gnomADg AMI  protein coding   2/4    NaN   \n",
       "5224    0.6355  gnomADg AMI  protein coding  3/16    NaN   \n",
       "5225    0.6355  gnomADg AMI  protein coding   3/4    NaN   \n",
       "6559    0.5858  gnomADg ASJ  protein coding   6/9    NaN   \n",
       "...        ...          ...             ...   ...    ...   \n",
       "301570  0.7103          AFR  protein coding  7/50    NaN   \n",
       "303230  0.4107          EAS  protein coding  3/10    NaN   \n",
       "303231  0.4107          EAS  protein coding  3/13    NaN   \n",
       "303233  0.3889          EAS  protein coding  3/10    NaN   \n",
       "303234  0.3889          EAS  protein coding  3/13    NaN   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "5222                              G55S  Ggc/Agc     1.0   \n",
       "5223                              G55S  Ggc/Agc     1.0   \n",
       "5224                             G112A  gGg/gCg     1.0   \n",
       "5225                             G112A  gGg/gCg     1.0   \n",
       "6559                             S202P  Tcc/Ccc     1.0   \n",
       "...                                ...      ...     ...   \n",
       "301570                           R219K  aGg/aAg    -1.0   \n",
       "303230                           M145T  aTg/aCg     1.0   \n",
       "303231                           M145T  aTg/aCg     1.0   \n",
       "303233                           G159A  gGc/gCc     1.0   \n",
       "303234                           G159A  gGc/gCc     1.0   \n",
       "\n",
       "                                                   PUBMED  \n",
       "5222    23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5223    23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5224    23251581,26843965,21698003,18576335,20537173,2...  \n",
       "5225    23251581,26843965,21698003,18576335,20537173,2...  \n",
       "6559         31227787,24906453,31397093,26129832,24067191  \n",
       "...                                                   ...  \n",
       "301570  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "303230                         28146080,29273807,28869550  \n",
       "303231                         28146080,29273807,28869550  \n",
       "303233                                                NaN  \n",
       "303234                                                NaN  \n",
       "\n",
       "[1157 rows x 74 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomADe_SAS_AF_x = dp_x[dp_x['gnomADe_SAS_AF'] <= 0.6]\n",
    "gnomADe_SAS_AF_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "c60bf90a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5222</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.55890</td>\n",
       "      <td>0.471300</td>\n",
       "      <td>0.50030</td>\n",
       "      <td>0.59520</td>\n",
       "      <td>0.57670</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.57130</td>\n",
       "      <td>0.52200</td>\n",
       "      <td>0.55080</td>\n",
       "      <td>0.476300</td>\n",
       "      <td>0.64900</td>\n",
       "      <td>0.54840</td>\n",
       "      <td>0.61760</td>\n",
       "      <td>0.59310</td>\n",
       "      <td>0.59300</td>\n",
       "      <td>0.575900</td>\n",
       "      <td>0.58400</td>\n",
       "      <td>0.53110</td>\n",
       "      <td>0.53190</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5224</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.196)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.54530</td>\n",
       "      <td>0.479000</td>\n",
       "      <td>0.49730</td>\n",
       "      <td>0.57640</td>\n",
       "      <td>0.57810</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.55770</td>\n",
       "      <td>0.52530</td>\n",
       "      <td>0.53610</td>\n",
       "      <td>0.479100</td>\n",
       "      <td>0.63550</td>\n",
       "      <td>0.53950</td>\n",
       "      <td>0.59880</td>\n",
       "      <td>0.59030</td>\n",
       "      <td>0.56030</td>\n",
       "      <td>0.575900</td>\n",
       "      <td>0.55790</td>\n",
       "      <td>0.52070</td>\n",
       "      <td>0.53470</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6559</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypercholesterolemia,...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>68.75%</td>\n",
       "      <td>3.385400e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>0.4349</td>\n",
       "      <td>0.3207</td>\n",
       "      <td>0.5101</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.5239</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.50540</td>\n",
       "      <td>0.350900</td>\n",
       "      <td>0.55160</td>\n",
       "      <td>0.58110</td>\n",
       "      <td>0.44770</td>\n",
       "      <td>0.5044</td>\n",
       "      <td>0.5345</td>\n",
       "      <td>0.53660</td>\n",
       "      <td>0.43100</td>\n",
       "      <td>0.47910</td>\n",
       "      <td>0.354300</td>\n",
       "      <td>0.49010</td>\n",
       "      <td>0.52020</td>\n",
       "      <td>0.58580</td>\n",
       "      <td>0.47470</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.557300</td>\n",
       "      <td>0.53790</td>\n",
       "      <td>0.52490</td>\n",
       "      <td>0.44490</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7839</th>\n",
       "      <td>LOC105378642,GJA4</td>\n",
       "      <td>LOC105378642</td>\n",
       "      <td>7</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1764391</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>34795168</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000342280.5</td>\n",
       "      <td>c.955C&gt;T</td>\n",
       "      <td>ENSP00000343676.4</td>\n",
       "      <td>p.Pro319Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>51.52%</td>\n",
       "      <td>4.101300e-07</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.5847</td>\n",
       "      <td>0.2723</td>\n",
       "      <td>0.1726</td>\n",
       "      <td>0.3320</td>\n",
       "      <td>0.2055</td>\n",
       "      <td>0.30320</td>\n",
       "      <td>0.539200</td>\n",
       "      <td>0.23980</td>\n",
       "      <td>0.29240</td>\n",
       "      <td>0.17210</td>\n",
       "      <td>0.4258</td>\n",
       "      <td>0.3102</td>\n",
       "      <td>0.30210</td>\n",
       "      <td>0.22030</td>\n",
       "      <td>0.36740</td>\n",
       "      <td>0.539800</td>\n",
       "      <td>0.18500</td>\n",
       "      <td>0.27440</td>\n",
       "      <td>0.29680</td>\n",
       "      <td>0.17750</td>\n",
       "      <td>0.42990</td>\n",
       "      <td>0.382900</td>\n",
       "      <td>0.30580</td>\n",
       "      <td>0.33810</td>\n",
       "      <td>0.21710</td>\n",
       "      <td>0.5847</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P319S</td>\n",
       "      <td>Ccc/Tcc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>21894447,17975119,29631604,18182569,22305353,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11032</th>\n",
       "      <td>MYSM1</td>\n",
       "      <td>MYSM1</td>\n",
       "      <td>4</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>rs12139511</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>58682254</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Bone marrow failure syndrome 4</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1169781.0</td>\n",
       "      <td>ENST00000472487.6</td>\n",
       "      <td>c.790A&gt;G</td>\n",
       "      <td>ENSP00000418734.1</td>\n",
       "      <td>p.Thr264Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>218</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.450800e-22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>11</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6320</td>\n",
       "      <td>0.6589</td>\n",
       "      <td>0.6643</td>\n",
       "      <td>0.6290</td>\n",
       "      <td>0.5875</td>\n",
       "      <td>0.6217</td>\n",
       "      <td>0.60850</td>\n",
       "      <td>0.633400</td>\n",
       "      <td>0.63730</td>\n",
       "      <td>0.50940</td>\n",
       "      <td>0.65300</td>\n",
       "      <td>0.7167</td>\n",
       "      <td>0.5814</td>\n",
       "      <td>0.58540</td>\n",
       "      <td>0.59880</td>\n",
       "      <td>0.61220</td>\n",
       "      <td>0.635600</td>\n",
       "      <td>0.52960</td>\n",
       "      <td>0.61660</td>\n",
       "      <td>0.51850</td>\n",
       "      <td>0.63900</td>\n",
       "      <td>0.71510</td>\n",
       "      <td>0.439900</td>\n",
       "      <td>0.58740</td>\n",
       "      <td>0.56280</td>\n",
       "      <td>0.61000</td>\n",
       "      <td>0.7167</td>\n",
       "      <td>gnomADe FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T264A</td>\n",
       "      <td>Act/Gct</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300489</th>\n",
       "      <td>LOC102723989,FRMD3</td>\n",
       "      <td>FRMD3</td>\n",
       "      <td>No</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>rs4877747</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>83248259</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000304195.8</td>\n",
       "      <td>c.1453G&gt;T</td>\n",
       "      <td>ENSP00000303508.3</td>\n",
       "      <td>p.Asp485Tyr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>28</td>\n",
       "      <td>14</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>1.665600e-05</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.125)</td>\n",
       "      <td>0.0309</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.0694</td>\n",
       "      <td>0.0239</td>\n",
       "      <td>0.0399</td>\n",
       "      <td>0.03218</td>\n",
       "      <td>0.004714</td>\n",
       "      <td>0.03574</td>\n",
       "      <td>0.01539</td>\n",
       "      <td>0.06352</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>0.0270</td>\n",
       "      <td>0.02606</td>\n",
       "      <td>0.03732</td>\n",
       "      <td>0.02449</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>0.02961</td>\n",
       "      <td>0.02593</td>\n",
       "      <td>0.01729</td>\n",
       "      <td>0.06757</td>\n",
       "      <td>0.05042</td>\n",
       "      <td>0.006329</td>\n",
       "      <td>0.02848</td>\n",
       "      <td>0.01434</td>\n",
       "      <td>0.03776</td>\n",
       "      <td>0.0694</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>14/14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D485Y</td>\n",
       "      <td>Gat/Tat</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301556</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cardiac_Health; Cholesterol_Disorders; Gall_st...</td>\n",
       "      <td>rs2066714</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104824472</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Tangier disease,Hypoa...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>364426.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.2649A&gt;G</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Ile883Met</td>\n",
       "      <td>0/1</td>\n",
       "      <td>58</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>47.06%</td>\n",
       "      <td>1.499700e-06</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>tolerated low confidence(0.12)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>0.5083</td>\n",
       "      <td>0.2233</td>\n",
       "      <td>0.6865</td>\n",
       "      <td>0.1352</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.21140</td>\n",
       "      <td>0.480300</td>\n",
       "      <td>0.22320</td>\n",
       "      <td>0.19120</td>\n",
       "      <td>0.71850</td>\n",
       "      <td>0.1012</td>\n",
       "      <td>0.1299</td>\n",
       "      <td>0.19460</td>\n",
       "      <td>0.14100</td>\n",
       "      <td>0.25390</td>\n",
       "      <td>0.476900</td>\n",
       "      <td>0.09231</td>\n",
       "      <td>0.21930</td>\n",
       "      <td>0.18840</td>\n",
       "      <td>0.69450</td>\n",
       "      <td>0.10450</td>\n",
       "      <td>0.212000</td>\n",
       "      <td>0.12850</td>\n",
       "      <td>0.24830</td>\n",
       "      <td>0.15960</td>\n",
       "      <td>0.7185</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>18/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>I883M</td>\n",
       "      <td>atA/atG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23039238,29615537,23168575,19041386,26891315,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301568</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cardiac_Health; Cholesterol_Disorders; Gall_st...</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.32360</td>\n",
       "      <td>0.630100</td>\n",
       "      <td>0.32490</td>\n",
       "      <td>0.28060</td>\n",
       "      <td>0.43090</td>\n",
       "      <td>0.2196</td>\n",
       "      <td>0.2742</td>\n",
       "      <td>0.31780</td>\n",
       "      <td>0.36760</td>\n",
       "      <td>0.38380</td>\n",
       "      <td>0.625000</td>\n",
       "      <td>0.44730</td>\n",
       "      <td>0.34620</td>\n",
       "      <td>0.27640</td>\n",
       "      <td>0.42730</td>\n",
       "      <td>0.22180</td>\n",
       "      <td>0.395600</td>\n",
       "      <td>0.27290</td>\n",
       "      <td>0.37680</td>\n",
       "      <td>0.37260</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303230</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105223</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365932</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166113.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.434T&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Met145Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>44.12%</td>\n",
       "      <td>4.183500e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3754</td>\n",
       "      <td>0.3714</td>\n",
       "      <td>0.3516</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.35590</td>\n",
       "      <td>0.356100</td>\n",
       "      <td>0.26120</td>\n",
       "      <td>0.33040</td>\n",
       "      <td>0.39270</td>\n",
       "      <td>0.3463</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.35670</td>\n",
       "      <td>0.35650</td>\n",
       "      <td>0.37580</td>\n",
       "      <td>0.367500</td>\n",
       "      <td>0.34540</td>\n",
       "      <td>0.34480</td>\n",
       "      <td>0.33610</td>\n",
       "      <td>0.39840</td>\n",
       "      <td>0.35810</td>\n",
       "      <td>0.357600</td>\n",
       "      <td>0.39320</td>\n",
       "      <td>0.36340</td>\n",
       "      <td>0.35700</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M145T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28146080,29273807,28869550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.7)</td>\n",
       "      <td>possibly damaging(0.52)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.33420</td>\n",
       "      <td>0.207100</td>\n",
       "      <td>0.24800</td>\n",
       "      <td>0.31280</td>\n",
       "      <td>0.37520</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.34000</td>\n",
       "      <td>0.33270</td>\n",
       "      <td>0.32630</td>\n",
       "      <td>0.220800</td>\n",
       "      <td>0.33520</td>\n",
       "      <td>0.32300</td>\n",
       "      <td>0.31700</td>\n",
       "      <td>0.37880</td>\n",
       "      <td>0.35610</td>\n",
       "      <td>0.335400</td>\n",
       "      <td>0.38250</td>\n",
       "      <td>0.32500</td>\n",
       "      <td>0.33260</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>258 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene Name          Gene Gene_Score  \\\n",
       "5222                 PADI4         PADI4         No   \n",
       "5224                 PADI4         PADI4         No   \n",
       "6559               LDLRAP1       LDLRAP1          8   \n",
       "7839     LOC105378642,GJA4  LOC105378642          7   \n",
       "11032                MYSM1         MYSM1          4   \n",
       "...                    ...           ...        ...   \n",
       "300489  LOC102723989,FRMD3         FRMD3         No   \n",
       "301556               ABCA1         ABCA1          7   \n",
       "301568               ABCA1         ABCA1          7   \n",
       "303230                CRB2          CRB2          6   \n",
       "303233                CRB2          CRB2          6   \n",
       "\n",
       "                                                Condition  \\\n",
       "5222                                 Rheumatoid Arthritis   \n",
       "5224                                 Rheumatoid Arthritis   \n",
       "6559    Polygenic (Familial combined, small dense LDL,...   \n",
       "7839                                Dry Skin/fragile skin   \n",
       "11032                                Diabetic Retinopathy   \n",
       "...                                                   ...   \n",
       "300489                               Diabetic Retinopathy   \n",
       "301556  Coronary artery disease risk; Polygenic (Famil...   \n",
       "301568  Coronary artery disease risk; Polygenic (Famil...   \n",
       "303230                 Focal segmental glomerulosclerosis   \n",
       "303233                 Focal segmental glomerulosclerosis   \n",
       "\n",
       "                                          Headings  \\\n",
       "5222                           Autoimmune Disorder   \n",
       "5224                           Autoimmune Disorder   \n",
       "6559                         Cholesterol disorders   \n",
       "7839                                   Skin Health   \n",
       "11032                         Diabetic Retinopathy   \n",
       "...                                            ...   \n",
       "300489                        Diabetic Retinopathy   \n",
       "301556  Cardiac; Cholesterol disorders; Gallstones   \n",
       "301568  Cardiac; Cholesterol disorders; Gallstones   \n",
       "303230                         Glomerular Diseases   \n",
       "303233                         Glomerular Diseases   \n",
       "\n",
       "                                       21_Conditions_list        rsID  \\\n",
       "5222                 Arthritis_Degenerative_Joint_Disease  rs11203366   \n",
       "5224                 Arthritis_Degenerative_Joint_Disease    rs874881   \n",
       "6559                                Cholesterol_Disorders   rs6687605   \n",
       "7839                                          Skin_Health   rs1764391   \n",
       "11032                                            Diabetes  rs12139511   \n",
       "...                                                   ...         ...   \n",
       "300489                                           Diabetes   rs4877747   \n",
       "301556  Cardiac_Health; Cholesterol_Disorders; Gall_st...   rs2066714   \n",
       "301568  Cardiac_Health; Cholesterol_Disorders; Gall_st...   rs2230806   \n",
       "303230                                Glomerular_Diseases   rs1105223   \n",
       "303233                                Glomerular_Diseases   rs1105222   \n",
       "\n",
       "       Literature CHROM        POS REF ALT      Zygosity       Consequence  \\\n",
       "5222           No  chr1   17331039   G   A  Heterozygous  missense variant   \n",
       "5224           No  chr1   17334004   G   C  Heterozygous  missense variant   \n",
       "6559           No  chr1   25563141   T   C  Heterozygous  missense variant   \n",
       "7839           No  chr1   34795168   C   T  Heterozygous  missense variant   \n",
       "11032          No  chr1   58682254   T   C    Homozygous  missense variant   \n",
       "...           ...   ...        ...  ..  ..           ...               ...   \n",
       "300489         No  chr9   83248259   C   A  Heterozygous  missense variant   \n",
       "301556        Yes  chr9  104824472   T   C  Heterozygous  missense variant   \n",
       "301568        Yes  chr9  104858586   C   T  Heterozygous  missense variant   \n",
       "303230        Yes  chr9  123365932   T   C  Heterozygous  missense variant   \n",
       "303233         No  chr9  123365974   G   C  Heterozygous  missense variant   \n",
       "\n",
       "       Consequence_score    IMPACT  IMPACT_score  \\\n",
       "5222                7/10  MODERATE           5.0   \n",
       "5224                7/10  MODERATE           5.0   \n",
       "6559                7/10  MODERATE           5.0   \n",
       "7839                7/10  MODERATE           5.0   \n",
       "11032               7/10  MODERATE           5.0   \n",
       "...                  ...       ...           ...   \n",
       "300489              7/10  MODERATE           5.0   \n",
       "301556              7/10  MODERATE           5.0   \n",
       "301568              7/10  MODERATE           5.0   \n",
       "303230              7/10  MODERATE           5.0   \n",
       "303233              7/10  MODERATE           5.0   \n",
       "\n",
       "                                            ClinVar_CLNDN     CLIN_SIG  \\\n",
       "5222    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "5224    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "6559    Cardiovascular phenotype,Hypercholesterolemia,...       benign   \n",
       "7839                                                  NaN          NaN   \n",
       "11032                      Bone marrow failure syndrome 4       benign   \n",
       "...                                                   ...          ...   \n",
       "300489                                                NaN          NaN   \n",
       "301556  Cardiovascular phenotype,Tangier disease,Hypoa...       benign   \n",
       "301568  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...       benign   \n",
       "303230             Ventriculomegaly-cystic kidney disease       benign   \n",
       "303233             Ventriculomegaly-cystic kidney disease       benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT    ClinVar  \\\n",
       "5222                       no assertion criteria provided   972892.0   \n",
       "5224                       no assertion criteria provided   972895.0   \n",
       "6559    criteria provided, multiple submitters, no con...   296982.0   \n",
       "7839                                                  NaN        NaN   \n",
       "11032   criteria provided, multiple submitters, no con...  1169781.0   \n",
       "...                                                   ...        ...   \n",
       "300489                                                NaN        NaN   \n",
       "301556  criteria provided, multiple submitters, no con...   364426.0   \n",
       "301568  criteria provided, multiple submitters, no con...     9506.0   \n",
       "303230  criteria provided, multiple submitters, no con...  1166113.0   \n",
       "303233  criteria provided, multiple submitters, no con...  1166114.0   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "5222    ENST00000375448.4           c.163G>A  ENSP00000364597.4   \n",
       "5224    ENST00000375448.4           c.335G>C  ENSP00000364597.4   \n",
       "6559    ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "7839    ENST00000342280.5           c.955C>T  ENSP00000343676.4   \n",
       "11032   ENST00000472487.6           c.790A>G  ENSP00000418734.1   \n",
       "...                   ...                ...                ...   \n",
       "300489  ENST00000304195.8          c.1453G>T  ENSP00000303508.3   \n",
       "301556  ENST00000374736.8          c.2649A>G  ENSP00000363868.3   \n",
       "301568  ENST00000374736.8           c.656G>A  ENSP00000363868.3   \n",
       "303230  ENST00000359999.7           c.434T>C  ENSP00000353092.3   \n",
       "303233  ENST00000359999.7           c.476G>C  ENSP00000353092.3   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "5222           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   \n",
       "5224          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03   \n",
       "6559          p.Ser202Pro  0/1   44   16  16   5  11  68.75%  3.385400e-05   \n",
       "7839          p.Pro319Ser  0/1   63   33  33  15  17  51.52%  4.101300e-07   \n",
       "11032         p.Thr264Ala  1/1  218   38  38   0  38    100%  1.450800e-22   \n",
       "...                   ...  ...  ...  ...  ..  ..  ..     ...           ...   \n",
       "300489        p.Asp485Tyr  0/1   47   42  42  28  14  33.33%  1.665600e-05   \n",
       "301556        p.Ile883Met  0/1   58   34  34  18  16  47.06%  1.499700e-06   \n",
       "301568        p.Arg219Lys  0/1   69   53  53  33  20  37.74%  1.040500e-07   \n",
       "303230        p.Met145Thr  0/1   53   34  34  19  15  44.12%  4.183500e-06   \n",
       "303233        p.Gly159Ala  0/1   73   41  41  21  20  48.78%  4.334700e-08   \n",
       "\n",
       "        RDF  RDR  ADF  ADR                            SIFT  \\\n",
       "5222     29    3    7    8                    tolerated(1)   \n",
       "5224      7    0    5    3                 tolerated(0.67)   \n",
       "6559      3    2    5    6               deleterious(0.04)   \n",
       "7839      9    6    8    9                 tolerated(0.82)   \n",
       "11032     0    0   27   11                    tolerated(1)   \n",
       "...     ...  ...  ...  ...                             ...   \n",
       "300489   21    7    7    7                  deleterious(0)   \n",
       "301556   17    1   15    1  tolerated low confidence(0.12)   \n",
       "301568   29    4   15    5     tolerated low confidence(1)   \n",
       "303230   10    9   11    4                 tolerated(0.82)   \n",
       "303233   12    9   16    4                  tolerated(0.7)   \n",
       "\n",
       "                       PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  \\\n",
       "5222                  benign(0)  0.5246  0.4508  0.5432  0.5635  0.5755   \n",
       "5224              benign(0.196)  0.5220  0.4607  0.5403  0.5585  0.5537   \n",
       "6559              benign(0.037)  0.4349  0.3207  0.5101  0.4425  0.5239   \n",
       "7839                  benign(0)  0.3337  0.5847  0.2723  0.1726  0.3320   \n",
       "11032                       NaN  0.6320  0.6589  0.6643  0.6290  0.5875   \n",
       "...                         ...     ...     ...     ...     ...     ...   \n",
       "300489            benign(0.125)  0.0309  0.0008  0.0303  0.0694  0.0239   \n",
       "301556            benign(0.001)  0.3568  0.5083  0.2233  0.6865  0.1352   \n",
       "301568                benign(0)  0.4397  0.7103  0.3487  0.4147  0.2425   \n",
       "303230                benign(0)  0.3754  0.3714  0.3516  0.4107  0.3817   \n",
       "303233  possibly damaging(0.52)  0.3159  0.1989  0.3314  0.3889  0.3728   \n",
       "\n",
       "        SAS_AF  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  \\\n",
       "5222    0.5184     0.55890        0.471300         0.50030         0.59520   \n",
       "5224    0.5215     0.54530        0.479000         0.49730         0.57640   \n",
       "6559    0.4366     0.50540        0.350900         0.55160         0.58110   \n",
       "7839    0.2055     0.30320        0.539200         0.23980         0.29240   \n",
       "11032   0.6217     0.60850        0.633400         0.63730         0.50940   \n",
       "...        ...         ...             ...             ...             ...   \n",
       "300489  0.0399     0.03218        0.004714         0.03574         0.01539   \n",
       "301556  0.1350     0.21140        0.480300         0.22320         0.19120   \n",
       "301568  0.3671     0.32360        0.630100         0.32490         0.28060   \n",
       "303230  0.3548     0.35590        0.356100         0.26120         0.33040   \n",
       "303233  0.3292     0.33420        0.207100         0.24800         0.31280   \n",
       "\n",
       "        gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  \\\n",
       "5222           0.57670          0.5898          0.5858         0.57130   \n",
       "5224           0.57810          0.5581          0.5635         0.55770   \n",
       "6559           0.44770          0.5044          0.5345         0.53660   \n",
       "7839           0.17210          0.4258          0.3102         0.30210   \n",
       "11032          0.65300          0.7167          0.5814         0.58540   \n",
       "...                ...             ...             ...             ...   \n",
       "300489         0.06352          0.0495          0.0270         0.02606   \n",
       "301556         0.71850          0.1012          0.1299         0.19460   \n",
       "301568         0.43090          0.2196          0.2742         0.31780   \n",
       "303230         0.39270          0.3463          0.3835         0.35670   \n",
       "303233         0.37520          0.3427          0.3740         0.34000   \n",
       "\n",
       "        gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  \\\n",
       "5222           0.52200     0.55080        0.476300         0.64900   \n",
       "5224           0.52530     0.53610        0.479100         0.63550   \n",
       "6559           0.43100     0.47910        0.354300         0.49010   \n",
       "7839           0.22030     0.36740        0.539800         0.18500   \n",
       "11032          0.59880     0.61220        0.635600         0.52960   \n",
       "...                ...         ...             ...             ...   \n",
       "300489         0.03732     0.02449        0.004995         0.02961   \n",
       "301556         0.14100     0.25390        0.476900         0.09231   \n",
       "301568         0.36760     0.38380        0.625000         0.44730   \n",
       "303230         0.35650     0.37580        0.367500         0.34540   \n",
       "303233         0.33270     0.32630        0.220800         0.33520   \n",
       "\n",
       "        gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  \\\n",
       "5222           0.54840         0.61760         0.59310         0.59300   \n",
       "5224           0.53950         0.59880         0.59030         0.56030   \n",
       "6559           0.52020         0.58580         0.47470         0.50000   \n",
       "7839           0.27440         0.29680         0.17750         0.42990   \n",
       "11032          0.61660         0.51850         0.63900         0.71510   \n",
       "...                ...             ...             ...             ...   \n",
       "300489         0.02593         0.01729         0.06757         0.05042   \n",
       "301556         0.21930         0.18840         0.69450         0.10450   \n",
       "301568         0.34620         0.27640         0.42730         0.22180   \n",
       "303230         0.34480         0.33610         0.39840         0.35810   \n",
       "303233         0.32300         0.31700         0.37880         0.35610   \n",
       "\n",
       "        gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  \\\n",
       "5222          0.575900         0.58400         0.53110         0.53190   \n",
       "5224          0.575900         0.55790         0.52070         0.53470   \n",
       "6559          0.557300         0.53790         0.52490         0.44490   \n",
       "7839          0.382900         0.30580         0.33810         0.21710   \n",
       "11032         0.439900         0.58740         0.56280         0.61000   \n",
       "...                ...             ...             ...             ...   \n",
       "300489        0.006329         0.02848         0.01434         0.03776   \n",
       "301556        0.212000         0.12850         0.24830         0.15960   \n",
       "301568        0.395600         0.27290         0.37680         0.37260   \n",
       "303230        0.357600         0.39320         0.36340         0.35700   \n",
       "303233        0.335400         0.38250         0.32500         0.33260   \n",
       "\n",
       "        MAX_AF  MAX_AF_POPS         BIOTYPE   EXON INTRON  \\\n",
       "5222    0.6490  gnomADg AMI  protein coding   2/16    NaN   \n",
       "5224    0.6355  gnomADg AMI  protein coding   3/16    NaN   \n",
       "6559    0.5858  gnomADg ASJ  protein coding    6/9    NaN   \n",
       "7839    0.5847          AFR  protein coding    2/2    NaN   \n",
       "11032   0.7167  gnomADe FIN  protein coding   8/20    NaN   \n",
       "...        ...          ...             ...    ...    ...   \n",
       "300489  0.0694          EAS  protein coding  14/14    NaN   \n",
       "301556  0.7185  gnomADe EAS  protein coding  18/50    NaN   \n",
       "301568  0.7103          AFR  protein coding   7/50    NaN   \n",
       "303230  0.4107          EAS  protein coding   3/10    NaN   \n",
       "303233  0.3889          EAS  protein coding   3/10    NaN   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "5222                              G55S  Ggc/Agc     1.0   \n",
       "5224                             G112A  gGg/gCg     1.0   \n",
       "6559                             S202P  Tcc/Ccc     1.0   \n",
       "7839                             P319S  Ccc/Tcc     1.0   \n",
       "11032                            T264A  Act/Gct    -1.0   \n",
       "...                                ...      ...     ...   \n",
       "300489                           D485Y  Gat/Tat    -1.0   \n",
       "301556                           I883M  atA/atG    -1.0   \n",
       "301568                           R219K  aGg/aAg    -1.0   \n",
       "303230                           M145T  aTg/aCg     1.0   \n",
       "303233                           G159A  gGc/gCc     1.0   \n",
       "\n",
       "                                                   PUBMED  \n",
       "5222    23251581,26843965,23164236,18576335,20537173,2...  \n",
       "5224    23251581,26843965,21698003,18576335,20537173,2...  \n",
       "6559         31227787,24906453,31397093,26129832,24067191  \n",
       "7839    21894447,17975119,29631604,18182569,22305353,2...  \n",
       "11032                                                 NaN  \n",
       "...                                                   ...  \n",
       "300489                                                NaN  \n",
       "301556  23039238,29615537,23168575,19041386,26891315,2...  \n",
       "301568  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "303230                         28146080,29273807,28869550  \n",
       "303233                                                NaN  \n",
       "\n",
       "[258 rows x 74 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drop_dup = gnomADe_SAS_AF_x.drop_duplicates(subset=['Gene', 'rsID', 'CHROM', 'POS', 'REF', 'ALT'])\n",
    "df = drop_dup.copy()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "cf0e5da8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17331039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972892.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.163G&gt;A</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly55Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>31.91%</td>\n",
       "      <td>8.086400e-06</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5246</td>\n",
       "      <td>0.4508</td>\n",
       "      <td>0.5432</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>0.5184</td>\n",
       "      <td>0.5589</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>0.5003</td>\n",
       "      <td>0.5952</td>\n",
       "      <td>0.5767</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.5713</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.5508</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>0.5484</td>\n",
       "      <td>0.6176</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.5311</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>0.6490</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G55S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,23164236,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>No</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17334004</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Rheumatoid arthritis,Abnormal pulmonary inters...</td>\n",
       "      <td>association</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>972895.0</td>\n",
       "      <td>ENST00000375448.4</td>\n",
       "      <td>c.335G&gt;C</td>\n",
       "      <td>ENSP00000364597.4</td>\n",
       "      <td>p.Gly112Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.196)</td>\n",
       "      <td>0.5220</td>\n",
       "      <td>0.4607</td>\n",
       "      <td>0.5403</td>\n",
       "      <td>0.5585</td>\n",
       "      <td>0.5537</td>\n",
       "      <td>0.5215</td>\n",
       "      <td>0.5453</td>\n",
       "      <td>0.4790</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>0.5781</td>\n",
       "      <td>0.5581</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>0.5577</td>\n",
       "      <td>0.5253</td>\n",
       "      <td>0.5361</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>0.5395</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.5903</td>\n",
       "      <td>0.5603</td>\n",
       "      <td>0.5759</td>\n",
       "      <td>0.5579</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>0.5347</td>\n",
       "      <td>0.6355</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G112A</td>\n",
       "      <td>gGg/gCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23251581,26843965,21698003,18576335,20537173,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypercholesterolemia,...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>68.75%</td>\n",
       "      <td>3.385400e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>0.4349</td>\n",
       "      <td>0.3207</td>\n",
       "      <td>0.5101</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.5239</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.5054</td>\n",
       "      <td>0.3509</td>\n",
       "      <td>0.5516</td>\n",
       "      <td>0.5811</td>\n",
       "      <td>0.4477</td>\n",
       "      <td>0.5044</td>\n",
       "      <td>0.5345</td>\n",
       "      <td>0.5366</td>\n",
       "      <td>0.4310</td>\n",
       "      <td>0.4791</td>\n",
       "      <td>0.3543</td>\n",
       "      <td>0.4901</td>\n",
       "      <td>0.5202</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>0.4747</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.5573</td>\n",
       "      <td>0.5379</td>\n",
       "      <td>0.5249</td>\n",
       "      <td>0.4449</td>\n",
       "      <td>0.5858</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>LOC105378642,GJA4</td>\n",
       "      <td>LOC105378642</td>\n",
       "      <td>7</td>\n",
       "      <td>rs1764391</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>34795168</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000342280.5</td>\n",
       "      <td>c.955C&gt;T</td>\n",
       "      <td>ENSP00000343676.4</td>\n",
       "      <td>p.Pro319Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>51.52%</td>\n",
       "      <td>4.101300e-07</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.5847</td>\n",
       "      <td>0.2723</td>\n",
       "      <td>0.1726</td>\n",
       "      <td>0.3320</td>\n",
       "      <td>0.2055</td>\n",
       "      <td>0.3032</td>\n",
       "      <td>0.5392</td>\n",
       "      <td>0.2398</td>\n",
       "      <td>0.2924</td>\n",
       "      <td>0.1721</td>\n",
       "      <td>0.4258</td>\n",
       "      <td>0.3102</td>\n",
       "      <td>0.3021</td>\n",
       "      <td>0.2203</td>\n",
       "      <td>0.3674</td>\n",
       "      <td>0.5398</td>\n",
       "      <td>0.1850</td>\n",
       "      <td>0.2744</td>\n",
       "      <td>0.2968</td>\n",
       "      <td>0.1775</td>\n",
       "      <td>0.4299</td>\n",
       "      <td>0.3829</td>\n",
       "      <td>0.3058</td>\n",
       "      <td>0.3381</td>\n",
       "      <td>0.2171</td>\n",
       "      <td>0.5847</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P319S</td>\n",
       "      <td>Ccc/Tcc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>21894447,17975119,29631604,18182569,22305353,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>MYSM1</td>\n",
       "      <td>MYSM1</td>\n",
       "      <td>4</td>\n",
       "      <td>rs12139511</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>58682254</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Bone marrow failure syndrome 4</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1169781.0</td>\n",
       "      <td>ENST00000472487.6</td>\n",
       "      <td>c.790A&gt;G</td>\n",
       "      <td>ENSP00000418734.1</td>\n",
       "      <td>p.Thr264Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>218</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.450800e-22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>11</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6320</td>\n",
       "      <td>0.6589</td>\n",
       "      <td>0.6643</td>\n",
       "      <td>0.6290</td>\n",
       "      <td>0.5875</td>\n",
       "      <td>0.6217</td>\n",
       "      <td>0.6085</td>\n",
       "      <td>0.6334</td>\n",
       "      <td>0.6373</td>\n",
       "      <td>0.5094</td>\n",
       "      <td>0.6530</td>\n",
       "      <td>0.7167</td>\n",
       "      <td>0.5814</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>0.5988</td>\n",
       "      <td>0.6122</td>\n",
       "      <td>0.6356</td>\n",
       "      <td>0.5296</td>\n",
       "      <td>0.6166</td>\n",
       "      <td>0.5185</td>\n",
       "      <td>0.6390</td>\n",
       "      <td>0.7151</td>\n",
       "      <td>0.4399</td>\n",
       "      <td>0.5874</td>\n",
       "      <td>0.5628</td>\n",
       "      <td>0.6100</td>\n",
       "      <td>0.7167</td>\n",
       "      <td>gnomADe FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T264A</td>\n",
       "      <td>Act/Gct</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>Coronary artery disease risk</td>\n",
       "      <td>Cardiac</td>\n",
       "      <td>Cardiac_Health</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.6301</td>\n",
       "      <td>0.3249</td>\n",
       "      <td>0.2806</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.2196</td>\n",
       "      <td>0.2742</td>\n",
       "      <td>0.3178</td>\n",
       "      <td>0.3676</td>\n",
       "      <td>0.3838</td>\n",
       "      <td>0.6250</td>\n",
       "      <td>0.4473</td>\n",
       "      <td>0.3462</td>\n",
       "      <td>0.2764</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.2218</td>\n",
       "      <td>0.3956</td>\n",
       "      <td>0.2729</td>\n",
       "      <td>0.3768</td>\n",
       "      <td>0.3726</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>390</th>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.6301</td>\n",
       "      <td>0.3249</td>\n",
       "      <td>0.2806</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.2196</td>\n",
       "      <td>0.2742</td>\n",
       "      <td>0.3178</td>\n",
       "      <td>0.3676</td>\n",
       "      <td>0.3838</td>\n",
       "      <td>0.6250</td>\n",
       "      <td>0.4473</td>\n",
       "      <td>0.3462</td>\n",
       "      <td>0.2764</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.2218</td>\n",
       "      <td>0.3956</td>\n",
       "      <td>0.2729</td>\n",
       "      <td>0.3768</td>\n",
       "      <td>0.3726</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>Gallstones</td>\n",
       "      <td>Gallstones</td>\n",
       "      <td>Gall_stones</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.6301</td>\n",
       "      <td>0.3249</td>\n",
       "      <td>0.2806</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.2196</td>\n",
       "      <td>0.2742</td>\n",
       "      <td>0.3178</td>\n",
       "      <td>0.3676</td>\n",
       "      <td>0.3838</td>\n",
       "      <td>0.6250</td>\n",
       "      <td>0.4473</td>\n",
       "      <td>0.3462</td>\n",
       "      <td>0.2764</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.2218</td>\n",
       "      <td>0.3956</td>\n",
       "      <td>0.2729</td>\n",
       "      <td>0.3768</td>\n",
       "      <td>0.3726</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>rs1105223</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365932</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166113.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.434T&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Met145Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>44.12%</td>\n",
       "      <td>4.183500e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3754</td>\n",
       "      <td>0.3714</td>\n",
       "      <td>0.3516</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.3559</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.2612</td>\n",
       "      <td>0.3304</td>\n",
       "      <td>0.3927</td>\n",
       "      <td>0.3463</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.3567</td>\n",
       "      <td>0.3565</td>\n",
       "      <td>0.3758</td>\n",
       "      <td>0.3675</td>\n",
       "      <td>0.3454</td>\n",
       "      <td>0.3448</td>\n",
       "      <td>0.3361</td>\n",
       "      <td>0.3984</td>\n",
       "      <td>0.3581</td>\n",
       "      <td>0.3576</td>\n",
       "      <td>0.3932</td>\n",
       "      <td>0.3634</td>\n",
       "      <td>0.3570</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M145T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28146080,29273807,28869550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.7)</td>\n",
       "      <td>possibly damaging(0.52)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.3342</td>\n",
       "      <td>0.2071</td>\n",
       "      <td>0.2480</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.3427</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>0.3327</td>\n",
       "      <td>0.3263</td>\n",
       "      <td>0.2208</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>0.3170</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.3326</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>394 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Condition               Headings  \\\n",
       "0                                 Rheumatoid Arthritis    Autoimmune Disorder   \n",
       "1                                 Rheumatoid Arthritis    Autoimmune Disorder   \n",
       "2    Polygenic (Familial combined, small dense LDL,...  Cholesterol disorders   \n",
       "3                                Dry Skin/fragile skin            Skin Health   \n",
       "4                                 Diabetic Retinopathy   Diabetic Retinopathy   \n",
       "..                                                 ...                    ...   \n",
       "389                       Coronary artery disease risk                Cardiac   \n",
       "390  Polygenic (Familial combined, small dense LDL,...  Cholesterol disorders   \n",
       "391                                         Gallstones             Gallstones   \n",
       "392                 Focal segmental glomerulosclerosis    Glomerular Diseases   \n",
       "393                 Focal segmental glomerulosclerosis    Glomerular Diseases   \n",
       "\n",
       "                       21_Conditions_list          Gene Name          Gene  \\\n",
       "0    Arthritis_Degenerative_Joint_Disease              PADI4         PADI4   \n",
       "1    Arthritis_Degenerative_Joint_Disease              PADI4         PADI4   \n",
       "2                   Cholesterol_Disorders            LDLRAP1       LDLRAP1   \n",
       "3                             Skin_Health  LOC105378642,GJA4  LOC105378642   \n",
       "4                                Diabetes              MYSM1         MYSM1   \n",
       "..                                    ...                ...           ...   \n",
       "389                        Cardiac_Health              ABCA1         ABCA1   \n",
       "390                 Cholesterol_Disorders              ABCA1         ABCA1   \n",
       "391                           Gall_stones              ABCA1         ABCA1   \n",
       "392                   Glomerular_Diseases               CRB2          CRB2   \n",
       "393                   Glomerular_Diseases               CRB2          CRB2   \n",
       "\n",
       "    Gene_Score        rsID Literature CHROM        POS REF ALT      Zygosity  \\\n",
       "0           No  rs11203366         No  chr1   17331039   G   A  Heterozygous   \n",
       "1           No    rs874881         No  chr1   17334004   G   C  Heterozygous   \n",
       "2            8   rs6687605         No  chr1   25563141   T   C  Heterozygous   \n",
       "3            7   rs1764391         No  chr1   34795168   C   T  Heterozygous   \n",
       "4            4  rs12139511         No  chr1   58682254   T   C    Homozygous   \n",
       "..         ...         ...        ...   ...        ...  ..  ..           ...   \n",
       "389          7   rs2230806        Yes  chr9  104858586   C   T  Heterozygous   \n",
       "390          7   rs2230806        Yes  chr9  104858586   C   T  Heterozygous   \n",
       "391          7   rs2230806        Yes  chr9  104858586   C   T  Heterozygous   \n",
       "392          6   rs1105223        Yes  chr9  123365932   T   C  Heterozygous   \n",
       "393          6   rs1105222         No  chr9  123365974   G   C  Heterozygous   \n",
       "\n",
       "          Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0    missense variant              7/10  MODERATE           5.0   \n",
       "1    missense variant              7/10  MODERATE           5.0   \n",
       "2    missense variant              7/10  MODERATE           5.0   \n",
       "3    missense variant              7/10  MODERATE           5.0   \n",
       "4    missense variant              7/10  MODERATE           5.0   \n",
       "..                ...               ...       ...           ...   \n",
       "389  missense variant              7/10  MODERATE           5.0   \n",
       "390  missense variant              7/10  MODERATE           5.0   \n",
       "391  missense variant              7/10  MODERATE           5.0   \n",
       "392  missense variant              7/10  MODERATE           5.0   \n",
       "393  missense variant              7/10  MODERATE           5.0   \n",
       "\n",
       "                                         ClinVar_CLNDN     CLIN_SIG  \\\n",
       "0    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "1    Rheumatoid arthritis,Abnormal pulmonary inters...  association   \n",
       "2    Cardiovascular phenotype,Hypercholesterolemia,...       benign   \n",
       "3                                                  NaN          NaN   \n",
       "4                       Bone marrow failure syndrome 4       benign   \n",
       "..                                                 ...          ...   \n",
       "389  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...       benign   \n",
       "390  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...       benign   \n",
       "391  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...       benign   \n",
       "392             Ventriculomegaly-cystic kidney disease       benign   \n",
       "393             Ventriculomegaly-cystic kidney disease       benign   \n",
       "\n",
       "                                    ClinVar_CLNREVSTAT    ClinVar  \\\n",
       "0                       no assertion criteria provided   972892.0   \n",
       "1                       no assertion criteria provided   972895.0   \n",
       "2    criteria provided, multiple submitters, no con...   296982.0   \n",
       "3                                                  NaN        NaN   \n",
       "4    criteria provided, multiple submitters, no con...  1169781.0   \n",
       "..                                                 ...        ...   \n",
       "389  criteria provided, multiple submitters, no con...     9506.0   \n",
       "390  criteria provided, multiple submitters, no con...     9506.0   \n",
       "391  criteria provided, multiple submitters, no con...     9506.0   \n",
       "392  criteria provided, multiple submitters, no con...  1166113.0   \n",
       "393  criteria provided, multiple submitters, no con...  1166114.0   \n",
       "\n",
       "                 HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0    ENST00000375448.4           c.163G>A  ENSP00000364597.4   \n",
       "1    ENST00000375448.4           c.335G>C  ENSP00000364597.4   \n",
       "2    ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "3    ENST00000342280.5           c.955C>T  ENSP00000343676.4   \n",
       "4    ENST00000472487.6           c.790A>G  ENSP00000418734.1   \n",
       "..                 ...                ...                ...   \n",
       "389  ENST00000374736.8           c.656G>A  ENSP00000363868.3   \n",
       "390  ENST00000374736.8           c.656G>A  ENSP00000363868.3   \n",
       "391  ENST00000374736.8           c.656G>A  ENSP00000363868.3   \n",
       "392  ENST00000359999.7           c.434T>C  ENSP00000353092.3   \n",
       "393  ENST00000359999.7           c.476G>C  ENSP00000353092.3   \n",
       "\n",
       "    HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  RDF  \\\n",
       "0           p.Gly55Ser  0/1   50   47  47  32  15  31.91%  8.086400e-06   29   \n",
       "1          p.Gly112Ala  0/1   29   15  15   7   8  53.33%  1.099500e-03    7   \n",
       "2          p.Ser202Pro  0/1   44   16  16   5  11  68.75%  3.385400e-05    3   \n",
       "3          p.Pro319Ser  0/1   63   33  33  15  17  51.52%  4.101300e-07    9   \n",
       "4          p.Thr264Ala  1/1  218   38  38   0  38    100%  1.450800e-22    0   \n",
       "..                 ...  ...  ...  ...  ..  ..  ..     ...           ...  ...   \n",
       "389        p.Arg219Lys  0/1   69   53  53  33  20  37.74%  1.040500e-07   29   \n",
       "390        p.Arg219Lys  0/1   69   53  53  33  20  37.74%  1.040500e-07   29   \n",
       "391        p.Arg219Lys  0/1   69   53  53  33  20  37.74%  1.040500e-07   29   \n",
       "392        p.Met145Thr  0/1   53   34  34  19  15  44.12%  4.183500e-06   10   \n",
       "393        p.Gly159Ala  0/1   73   41  41  21  20  48.78%  4.334700e-08   12   \n",
       "\n",
       "     RDR  ADF  ADR                         SIFT                 PolyPhen  \\\n",
       "0      3    7    8                 tolerated(1)                benign(0)   \n",
       "1      0    5    3              tolerated(0.67)            benign(0.196)   \n",
       "2      2    5    6            deleterious(0.04)            benign(0.037)   \n",
       "3      6    8    9              tolerated(0.82)                benign(0)   \n",
       "4      0   27   11                 tolerated(1)                      NaN   \n",
       "..   ...  ...  ...                          ...                      ...   \n",
       "389    4   15    5  tolerated low confidence(1)                benign(0)   \n",
       "390    4   15    5  tolerated low confidence(1)                benign(0)   \n",
       "391    4   15    5  tolerated low confidence(1)                benign(0)   \n",
       "392    9   11    4              tolerated(0.82)                benign(0)   \n",
       "393    9   16    4               tolerated(0.7)  possibly damaging(0.52)   \n",
       "\n",
       "         AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  \\\n",
       "0    0.5246  0.4508  0.5432  0.5635  0.5755  0.5184      0.5589   \n",
       "1    0.5220  0.4607  0.5403  0.5585  0.5537  0.5215      0.5453   \n",
       "2    0.4349  0.3207  0.5101  0.4425  0.5239  0.4366      0.5054   \n",
       "3    0.3337  0.5847  0.2723  0.1726  0.3320  0.2055      0.3032   \n",
       "4    0.6320  0.6589  0.6643  0.6290  0.5875  0.6217      0.6085   \n",
       "..      ...     ...     ...     ...     ...     ...         ...   \n",
       "389  0.4397  0.7103  0.3487  0.4147  0.2425  0.3671      0.3236   \n",
       "390  0.4397  0.7103  0.3487  0.4147  0.2425  0.3671      0.3236   \n",
       "391  0.4397  0.7103  0.3487  0.4147  0.2425  0.3671      0.3236   \n",
       "392  0.3754  0.3714  0.3516  0.4107  0.3817  0.3548      0.3559   \n",
       "393  0.3159  0.1989  0.3314  0.3889  0.3728  0.3292      0.3342   \n",
       "\n",
       "     gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  \\\n",
       "0            0.4713          0.5003          0.5952          0.5767   \n",
       "1            0.4790          0.4973          0.5764          0.5781   \n",
       "2            0.3509          0.5516          0.5811          0.4477   \n",
       "3            0.5392          0.2398          0.2924          0.1721   \n",
       "4            0.6334          0.6373          0.5094          0.6530   \n",
       "..              ...             ...             ...             ...   \n",
       "389          0.6301          0.3249          0.2806          0.4309   \n",
       "390          0.6301          0.3249          0.2806          0.4309   \n",
       "391          0.6301          0.3249          0.2806          0.4309   \n",
       "392          0.3561          0.2612          0.3304          0.3927   \n",
       "393          0.2071          0.2480          0.3128          0.3752   \n",
       "\n",
       "     gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  \\\n",
       "0            0.5898          0.5858          0.5713          0.5220   \n",
       "1            0.5581          0.5635          0.5577          0.5253   \n",
       "2            0.5044          0.5345          0.5366          0.4310   \n",
       "3            0.4258          0.3102          0.3021          0.2203   \n",
       "4            0.7167          0.5814          0.5854          0.5988   \n",
       "..              ...             ...             ...             ...   \n",
       "389          0.2196          0.2742          0.3178          0.3676   \n",
       "390          0.2196          0.2742          0.3178          0.3676   \n",
       "391          0.2196          0.2742          0.3178          0.3676   \n",
       "392          0.3463          0.3835          0.3567          0.3565   \n",
       "393          0.3427          0.3740          0.3400          0.3327   \n",
       "\n",
       "     gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "0        0.5508          0.4763          0.6490          0.5484   \n",
       "1        0.5361          0.4791          0.6355          0.5395   \n",
       "2        0.4791          0.3543          0.4901          0.5202   \n",
       "3        0.3674          0.5398          0.1850          0.2744   \n",
       "4        0.6122          0.6356          0.5296          0.6166   \n",
       "..          ...             ...             ...             ...   \n",
       "389      0.3838          0.6250          0.4473          0.3462   \n",
       "390      0.3838          0.6250          0.4473          0.3462   \n",
       "391      0.3838          0.6250          0.4473          0.3462   \n",
       "392      0.3758          0.3675          0.3454          0.3448   \n",
       "393      0.3263          0.2208          0.3352          0.3230   \n",
       "\n",
       "     gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "0            0.6176          0.5931          0.5930          0.5759   \n",
       "1            0.5988          0.5903          0.5603          0.5759   \n",
       "2            0.5858          0.4747          0.5000          0.5573   \n",
       "3            0.2968          0.1775          0.4299          0.3829   \n",
       "4            0.5185          0.6390          0.7151          0.4399   \n",
       "..              ...             ...             ...             ...   \n",
       "389          0.2764          0.4273          0.2218          0.3956   \n",
       "390          0.2764          0.4273          0.2218          0.3956   \n",
       "391          0.2764          0.4273          0.2218          0.3956   \n",
       "392          0.3361          0.3984          0.3581          0.3576   \n",
       "393          0.3170          0.3788          0.3561          0.3354   \n",
       "\n",
       "     gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0            0.5840          0.5311          0.5319  0.6490  gnomADg AMI   \n",
       "1            0.5579          0.5207          0.5347  0.6355  gnomADg AMI   \n",
       "2            0.5379          0.5249          0.4449  0.5858  gnomADg ASJ   \n",
       "3            0.3058          0.3381          0.2171  0.5847          AFR   \n",
       "4            0.5874          0.5628          0.6100  0.7167  gnomADe FIN   \n",
       "..              ...             ...             ...     ...          ...   \n",
       "389          0.2729          0.3768          0.3726  0.7103          AFR   \n",
       "390          0.2729          0.3768          0.3726  0.7103          AFR   \n",
       "391          0.2729          0.3768          0.3726  0.7103          AFR   \n",
       "392          0.3932          0.3634          0.3570  0.4107          EAS   \n",
       "393          0.3825          0.3250          0.3326  0.3889          EAS   \n",
       "\n",
       "            BIOTYPE  EXON INTRON Protein Position and Amino Acid   Codons  \\\n",
       "0    protein coding  2/16    NaN                            G55S  Ggc/Agc   \n",
       "1    protein coding  3/16    NaN                           G112A  gGg/gCg   \n",
       "2    protein coding   6/9    NaN                           S202P  Tcc/Ccc   \n",
       "3    protein coding   2/2    NaN                           P319S  Ccc/Tcc   \n",
       "4    protein coding  8/20    NaN                           T264A  Act/Gct   \n",
       "..              ...   ...    ...                             ...      ...   \n",
       "389  protein coding  7/50    NaN                           R219K  aGg/aAg   \n",
       "390  protein coding  7/50    NaN                           R219K  aGg/aAg   \n",
       "391  protein coding  7/50    NaN                           R219K  aGg/aAg   \n",
       "392  protein coding  3/10    NaN                           M145T  aTg/aCg   \n",
       "393  protein coding  3/10    NaN                           G159A  gGc/gCc   \n",
       "\n",
       "     STRAND                                             PUBMED  \n",
       "0       1.0  23251581,26843965,23164236,18576335,20537173,2...  \n",
       "1       1.0  23251581,26843965,21698003,18576335,20537173,2...  \n",
       "2       1.0       31227787,24906453,31397093,26129832,24067191  \n",
       "3       1.0  21894447,17975119,29631604,18182569,22305353,2...  \n",
       "4      -1.0                                                NaN  \n",
       "..      ...                                                ...  \n",
       "389    -1.0  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "390    -1.0  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "391    -1.0  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "392     1.0                         28146080,29273807,28869550  \n",
       "393     1.0                                                NaN  \n",
       "\n",
       "[394 rows x 74 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = df.set_index(['Gene Name', 'Gene', 'Gene_Score', 'rsID', 'Literature', 'CHROM', 'POS', 'REF',\n",
    "       'ALT', 'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT',\n",
    "       'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "       'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)',\n",
    "       'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR',\n",
    "       'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF',\n",
    "       'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF',\n",
    "       'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF',\n",
    "       'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF',\n",
    "       'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF',\n",
    "       'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF',\n",
    "       'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']).apply(lambda x: x.str.split('; ').explode()).reset_index()\n",
    "# Specify the column names to move to the beginning\n",
    "columns_to_move = ['Condition', 'Headings', '21_Conditions_list']\n",
    "\n",
    "# Reorder the DataFrame to move specified columns to the beginning\n",
    "df_1 = df_1[columns_to_move + [col for col in df_1.columns if col not in columns_to_move]]\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "688f2b3c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eee28a86",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5bbfc12",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "13607eb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6559</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypercholesterolemia,...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>68.75%</td>\n",
       "      <td>3.385400e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>0.4349</td>\n",
       "      <td>0.3207</td>\n",
       "      <td>0.5101</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.5239</td>\n",
       "      <td>0.4366</td>\n",
       "      <td>0.50540</td>\n",
       "      <td>0.35090</td>\n",
       "      <td>0.55160</td>\n",
       "      <td>0.58110</td>\n",
       "      <td>0.4477</td>\n",
       "      <td>0.50440</td>\n",
       "      <td>0.53450</td>\n",
       "      <td>0.53660</td>\n",
       "      <td>0.43100</td>\n",
       "      <td>0.47910</td>\n",
       "      <td>0.35430</td>\n",
       "      <td>0.49010</td>\n",
       "      <td>0.52020</td>\n",
       "      <td>0.58580</td>\n",
       "      <td>0.4747</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.5573</td>\n",
       "      <td>0.5379</td>\n",
       "      <td>0.52490</td>\n",
       "      <td>0.44490</td>\n",
       "      <td>0.58580</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7839</th>\n",
       "      <td>LOC105378642,GJA4</td>\n",
       "      <td>LOC105378642</td>\n",
       "      <td>7</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1764391</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>34795168</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000342280.5</td>\n",
       "      <td>c.955C&gt;T</td>\n",
       "      <td>ENSP00000343676.4</td>\n",
       "      <td>p.Pro319Ser</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>51.52%</td>\n",
       "      <td>4.101300e-07</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.5847</td>\n",
       "      <td>0.2723</td>\n",
       "      <td>0.1726</td>\n",
       "      <td>0.3320</td>\n",
       "      <td>0.2055</td>\n",
       "      <td>0.30320</td>\n",
       "      <td>0.53920</td>\n",
       "      <td>0.23980</td>\n",
       "      <td>0.29240</td>\n",
       "      <td>0.1721</td>\n",
       "      <td>0.42580</td>\n",
       "      <td>0.31020</td>\n",
       "      <td>0.30210</td>\n",
       "      <td>0.22030</td>\n",
       "      <td>0.36740</td>\n",
       "      <td>0.53980</td>\n",
       "      <td>0.18500</td>\n",
       "      <td>0.27440</td>\n",
       "      <td>0.29680</td>\n",
       "      <td>0.1775</td>\n",
       "      <td>0.42990</td>\n",
       "      <td>0.3829</td>\n",
       "      <td>0.3058</td>\n",
       "      <td>0.33810</td>\n",
       "      <td>0.21710</td>\n",
       "      <td>0.58470</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P319S</td>\n",
       "      <td>Ccc/Tcc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>21894447,17975119,29631604,18182569,22305353,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15130</th>\n",
       "      <td>ECM1</td>\n",
       "      <td>ECM1</td>\n",
       "      <td>6</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut_Health</td>\n",
       "      <td>rs3737240</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>150510879</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Lipid proteinosis</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1222321.0</td>\n",
       "      <td>ENST00000346569.6</td>\n",
       "      <td>c.389C&gt;T</td>\n",
       "      <td>ENSP00000271630.6</td>\n",
       "      <td>p.Thr130Met</td>\n",
       "      <td>1/1</td>\n",
       "      <td>99</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.101900e-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated(0.07)</td>\n",
       "      <td>possibly damaging(0.471)</td>\n",
       "      <td>0.2238</td>\n",
       "      <td>0.0325</td>\n",
       "      <td>0.2493</td>\n",
       "      <td>0.2371</td>\n",
       "      <td>0.3757</td>\n",
       "      <td>0.2945</td>\n",
       "      <td>0.33920</td>\n",
       "      <td>0.08034</td>\n",
       "      <td>0.21870</td>\n",
       "      <td>0.49530</td>\n",
       "      <td>0.2337</td>\n",
       "      <td>0.41010</td>\n",
       "      <td>0.40110</td>\n",
       "      <td>0.38060</td>\n",
       "      <td>0.33810</td>\n",
       "      <td>0.29750</td>\n",
       "      <td>0.08901</td>\n",
       "      <td>0.42970</td>\n",
       "      <td>0.27530</td>\n",
       "      <td>0.50090</td>\n",
       "      <td>0.2453</td>\n",
       "      <td>0.40510</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.4010</td>\n",
       "      <td>0.33240</td>\n",
       "      <td>0.32730</td>\n",
       "      <td>0.50090</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T130M</td>\n",
       "      <td>aCg/aTg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18438406,21049557,31892796,30568945,28699600,1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15144</th>\n",
       "      <td>ECM1</td>\n",
       "      <td>ECM1</td>\n",
       "      <td>6</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut_Health</td>\n",
       "      <td>rs13294</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>150512511</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Lipid proteinosis</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1255367.0</td>\n",
       "      <td>ENST00000346569.6</td>\n",
       "      <td>c.868G&gt;A</td>\n",
       "      <td>ENSP00000271630.6</td>\n",
       "      <td>p.Gly290Ser</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.2666</td>\n",
       "      <td>0.1747</td>\n",
       "      <td>0.2709</td>\n",
       "      <td>0.2371</td>\n",
       "      <td>0.3827</td>\n",
       "      <td>0.2986</td>\n",
       "      <td>0.35210</td>\n",
       "      <td>0.19920</td>\n",
       "      <td>0.22870</td>\n",
       "      <td>0.50560</td>\n",
       "      <td>0.2338</td>\n",
       "      <td>0.41520</td>\n",
       "      <td>0.40580</td>\n",
       "      <td>0.39120</td>\n",
       "      <td>0.34090</td>\n",
       "      <td>0.33360</td>\n",
       "      <td>0.20380</td>\n",
       "      <td>0.43060</td>\n",
       "      <td>0.29310</td>\n",
       "      <td>0.50810</td>\n",
       "      <td>0.2456</td>\n",
       "      <td>0.41030</td>\n",
       "      <td>0.3861</td>\n",
       "      <td>0.4048</td>\n",
       "      <td>0.36130</td>\n",
       "      <td>0.33000</td>\n",
       "      <td>0.50810</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G290S</td>\n",
       "      <td>Ggt/Agt</td>\n",
       "      <td>1.0</td>\n",
       "      <td>27417569,18438406,21049557,26738999,31892796,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16357</th>\n",
       "      <td>FLG</td>\n",
       "      <td>FLG</td>\n",
       "      <td>8</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs7518080</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152304195</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1247942.0</td>\n",
       "      <td>ENST00000368799.2</td>\n",
       "      <td>c.10691G&gt;T</td>\n",
       "      <td>ENSP00000357789.1</td>\n",
       "      <td>p.Arg3564Leu</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>156</td>\n",
       "      <td>156</td>\n",
       "      <td>77</td>\n",
       "      <td>79</td>\n",
       "      <td>50.64%</td>\n",
       "      <td>2.371700e-30</td>\n",
       "      <td>56</td>\n",
       "      <td>21</td>\n",
       "      <td>54</td>\n",
       "      <td>25</td>\n",
       "      <td>tolerated(0.08)</td>\n",
       "      <td>benign(0.389)</td>\n",
       "      <td>0.0164</td>\n",
       "      <td>0.0083</td>\n",
       "      <td>0.0202</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0497</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>0.04346</td>\n",
       "      <td>0.01662</td>\n",
       "      <td>0.02979</td>\n",
       "      <td>0.06244</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.01551</td>\n",
       "      <td>0.06256</td>\n",
       "      <td>0.04835</td>\n",
       "      <td>0.02078</td>\n",
       "      <td>0.04517</td>\n",
       "      <td>0.02553</td>\n",
       "      <td>0.04348</td>\n",
       "      <td>0.04381</td>\n",
       "      <td>0.05373</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.01464</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.0617</td>\n",
       "      <td>0.05093</td>\n",
       "      <td>0.01697</td>\n",
       "      <td>0.06256</td>\n",
       "      <td>gnomADe NFE</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R3564L</td>\n",
       "      <td>cGt/cTt</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301568</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cardiac_Health</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.32360</td>\n",
       "      <td>0.63010</td>\n",
       "      <td>0.32490</td>\n",
       "      <td>0.28060</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.21960</td>\n",
       "      <td>0.27420</td>\n",
       "      <td>0.31780</td>\n",
       "      <td>0.36760</td>\n",
       "      <td>0.38380</td>\n",
       "      <td>0.62500</td>\n",
       "      <td>0.44730</td>\n",
       "      <td>0.34620</td>\n",
       "      <td>0.27640</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.22180</td>\n",
       "      <td>0.3956</td>\n",
       "      <td>0.2729</td>\n",
       "      <td>0.37680</td>\n",
       "      <td>0.37260</td>\n",
       "      <td>0.71030</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301568</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.32360</td>\n",
       "      <td>0.63010</td>\n",
       "      <td>0.32490</td>\n",
       "      <td>0.28060</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.21960</td>\n",
       "      <td>0.27420</td>\n",
       "      <td>0.31780</td>\n",
       "      <td>0.36760</td>\n",
       "      <td>0.38380</td>\n",
       "      <td>0.62500</td>\n",
       "      <td>0.44730</td>\n",
       "      <td>0.34620</td>\n",
       "      <td>0.27640</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.22180</td>\n",
       "      <td>0.3956</td>\n",
       "      <td>0.2729</td>\n",
       "      <td>0.37680</td>\n",
       "      <td>0.37260</td>\n",
       "      <td>0.71030</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301568</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>ABCA1</td>\n",
       "      <td>7</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Gall_stones</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>69</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>37.74%</td>\n",
       "      <td>1.040500e-07</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>0.3487</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.2425</td>\n",
       "      <td>0.3671</td>\n",
       "      <td>0.32360</td>\n",
       "      <td>0.63010</td>\n",
       "      <td>0.32490</td>\n",
       "      <td>0.28060</td>\n",
       "      <td>0.4309</td>\n",
       "      <td>0.21960</td>\n",
       "      <td>0.27420</td>\n",
       "      <td>0.31780</td>\n",
       "      <td>0.36760</td>\n",
       "      <td>0.38380</td>\n",
       "      <td>0.62500</td>\n",
       "      <td>0.44730</td>\n",
       "      <td>0.34620</td>\n",
       "      <td>0.27640</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.22180</td>\n",
       "      <td>0.3956</td>\n",
       "      <td>0.2729</td>\n",
       "      <td>0.37680</td>\n",
       "      <td>0.37260</td>\n",
       "      <td>0.71030</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303230</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105223</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365932</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166113.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.434T&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Met145Thr</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>44.12%</td>\n",
       "      <td>4.183500e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.82)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.3754</td>\n",
       "      <td>0.3714</td>\n",
       "      <td>0.3516</td>\n",
       "      <td>0.4107</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.35590</td>\n",
       "      <td>0.35610</td>\n",
       "      <td>0.26120</td>\n",
       "      <td>0.33040</td>\n",
       "      <td>0.3927</td>\n",
       "      <td>0.34630</td>\n",
       "      <td>0.38350</td>\n",
       "      <td>0.35670</td>\n",
       "      <td>0.35650</td>\n",
       "      <td>0.37580</td>\n",
       "      <td>0.36750</td>\n",
       "      <td>0.34540</td>\n",
       "      <td>0.34480</td>\n",
       "      <td>0.33610</td>\n",
       "      <td>0.3984</td>\n",
       "      <td>0.35810</td>\n",
       "      <td>0.3576</td>\n",
       "      <td>0.3932</td>\n",
       "      <td>0.36340</td>\n",
       "      <td>0.35700</td>\n",
       "      <td>0.41070</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M145T</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28146080,29273807,28869550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303233</th>\n",
       "      <td>CRB2</td>\n",
       "      <td>CRB2</td>\n",
       "      <td>6</td>\n",
       "      <td>Focal segmental glomerulosclerosis</td>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "      <td>rs1105222</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123365974</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Ventriculomegaly-cystic kidney disease</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1166114.0</td>\n",
       "      <td>ENST00000359999.7</td>\n",
       "      <td>c.476G&gt;C</td>\n",
       "      <td>ENSP00000353092.3</td>\n",
       "      <td>p.Gly159Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>73</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>48.78%</td>\n",
       "      <td>4.334700e-08</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.7)</td>\n",
       "      <td>possibly damaging(0.52)</td>\n",
       "      <td>0.3159</td>\n",
       "      <td>0.1989</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3728</td>\n",
       "      <td>0.3292</td>\n",
       "      <td>0.33420</td>\n",
       "      <td>0.20710</td>\n",
       "      <td>0.24800</td>\n",
       "      <td>0.31280</td>\n",
       "      <td>0.3752</td>\n",
       "      <td>0.34270</td>\n",
       "      <td>0.37400</td>\n",
       "      <td>0.34000</td>\n",
       "      <td>0.33270</td>\n",
       "      <td>0.32630</td>\n",
       "      <td>0.22080</td>\n",
       "      <td>0.33520</td>\n",
       "      <td>0.32300</td>\n",
       "      <td>0.31700</td>\n",
       "      <td>0.3788</td>\n",
       "      <td>0.35610</td>\n",
       "      <td>0.3354</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>0.32500</td>\n",
       "      <td>0.33260</td>\n",
       "      <td>0.38890</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G159A</td>\n",
       "      <td>gGc/gCc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>238 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene Name          Gene Gene_Score  \\\n",
       "6559              LDLRAP1       LDLRAP1          8   \n",
       "7839    LOC105378642,GJA4  LOC105378642          7   \n",
       "15130                ECM1          ECM1          6   \n",
       "15144                ECM1          ECM1          6   \n",
       "16357                 FLG           FLG          8   \n",
       "...                   ...           ...        ...   \n",
       "301568              ABCA1         ABCA1          7   \n",
       "301568              ABCA1         ABCA1          7   \n",
       "301568              ABCA1         ABCA1          7   \n",
       "303230               CRB2          CRB2          6   \n",
       "303233               CRB2          CRB2          6   \n",
       "\n",
       "                                                Condition  \\\n",
       "6559    Polygenic (Familial combined, small dense LDL,...   \n",
       "7839                                Dry Skin/fragile skin   \n",
       "15130                                          Gut Health   \n",
       "15144                                          Gut Health   \n",
       "16357                               Dry Skin/fragile skin   \n",
       "...                                                   ...   \n",
       "301568  Coronary artery disease risk; Polygenic (Famil...   \n",
       "301568  Coronary artery disease risk; Polygenic (Famil...   \n",
       "301568  Coronary artery disease risk; Polygenic (Famil...   \n",
       "303230                 Focal segmental glomerulosclerosis   \n",
       "303233                 Focal segmental glomerulosclerosis   \n",
       "\n",
       "                                          Headings     21_Conditions_list  \\\n",
       "6559                         Cholesterol disorders  Cholesterol_Disorders   \n",
       "7839                                   Skin Health            Skin_Health   \n",
       "15130                                   Gut Health             Gut_Health   \n",
       "15144                                   Gut Health             Gut_Health   \n",
       "16357                                  Skin Health            Skin_Health   \n",
       "...                                            ...                    ...   \n",
       "301568  Cardiac; Cholesterol disorders; Gallstones         Cardiac_Health   \n",
       "301568  Cardiac; Cholesterol disorders; Gallstones  Cholesterol_Disorders   \n",
       "301568  Cardiac; Cholesterol disorders; Gallstones            Gall_stones   \n",
       "303230                         Glomerular Diseases    Glomerular_Diseases   \n",
       "303233                         Glomerular Diseases    Glomerular_Diseases   \n",
       "\n",
       "             rsID Literature CHROM        POS REF ALT      Zygosity  \\\n",
       "6559    rs6687605         No  chr1   25563141   T   C  Heterozygous   \n",
       "7839    rs1764391         No  chr1   34795168   C   T  Heterozygous   \n",
       "15130   rs3737240         No  chr1  150510879   C   T    Homozygous   \n",
       "15144     rs13294         No  chr1  150512511   G   A    Homozygous   \n",
       "16357   rs7518080         No  chr1  152304195   C   A  Heterozygous   \n",
       "...           ...        ...   ...        ...  ..  ..           ...   \n",
       "301568  rs2230806        Yes  chr9  104858586   C   T  Heterozygous   \n",
       "301568  rs2230806        Yes  chr9  104858586   C   T  Heterozygous   \n",
       "301568  rs2230806        Yes  chr9  104858586   C   T  Heterozygous   \n",
       "303230  rs1105223        Yes  chr9  123365932   T   C  Heterozygous   \n",
       "303233  rs1105222         No  chr9  123365974   G   C  Heterozygous   \n",
       "\n",
       "             Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "6559    missense variant              7/10  MODERATE           5.0   \n",
       "7839    missense variant              7/10  MODERATE           5.0   \n",
       "15130   missense variant              7/10  MODERATE           5.0   \n",
       "15144   missense variant              7/10  MODERATE           5.0   \n",
       "16357   missense variant              7/10  MODERATE           5.0   \n",
       "...                  ...               ...       ...           ...   \n",
       "301568  missense variant              7/10  MODERATE           5.0   \n",
       "301568  missense variant              7/10  MODERATE           5.0   \n",
       "301568  missense variant              7/10  MODERATE           5.0   \n",
       "303230  missense variant              7/10  MODERATE           5.0   \n",
       "303233  missense variant              7/10  MODERATE           5.0   \n",
       "\n",
       "                                            ClinVar_CLNDN CLIN_SIG  \\\n",
       "6559    Cardiovascular phenotype,Hypercholesterolemia,...   benign   \n",
       "7839                                                  NaN      NaN   \n",
       "15130                                   Lipid proteinosis   benign   \n",
       "15144                                   Lipid proteinosis   benign   \n",
       "16357                                        not provided   benign   \n",
       "...                                                   ...      ...   \n",
       "301568  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...   benign   \n",
       "301568  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...   benign   \n",
       "301568  Cardiovascular phenotype,RECLASSIFIED - ABCA1 ...   benign   \n",
       "303230             Ventriculomegaly-cystic kidney disease   benign   \n",
       "303233             Ventriculomegaly-cystic kidney disease   benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT    ClinVar  \\\n",
       "6559    criteria provided, multiple submitters, no con...   296982.0   \n",
       "7839                                                  NaN        NaN   \n",
       "15130   criteria provided, multiple submitters, no con...  1222321.0   \n",
       "15144                 criteria provided, single submitter  1255367.0   \n",
       "16357                 criteria provided, single submitter  1247942.0   \n",
       "...                                                   ...        ...   \n",
       "301568  criteria provided, multiple submitters, no con...     9506.0   \n",
       "301568  criteria provided, multiple submitters, no con...     9506.0   \n",
       "301568  criteria provided, multiple submitters, no con...     9506.0   \n",
       "303230  criteria provided, multiple submitters, no con...  1166113.0   \n",
       "303233  criteria provided, multiple submitters, no con...  1166114.0   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "6559    ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "7839    ENST00000342280.5           c.955C>T  ENSP00000343676.4   \n",
       "15130   ENST00000346569.6           c.389C>T  ENSP00000271630.6   \n",
       "15144   ENST00000346569.6           c.868G>A  ENSP00000271630.6   \n",
       "16357   ENST00000368799.2         c.10691G>T  ENSP00000357789.1   \n",
       "...                   ...                ...                ...   \n",
       "301568  ENST00000374736.8           c.656G>A  ENSP00000363868.3   \n",
       "301568  ENST00000374736.8           c.656G>A  ENSP00000363868.3   \n",
       "301568  ENST00000374736.8           c.656G>A  ENSP00000363868.3   \n",
       "303230  ENST00000359999.7           c.434T>C  ENSP00000353092.3   \n",
       "303233  ENST00000359999.7           c.476G>C  ENSP00000353092.3   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP  RD  AD    FREQ          PVAL  \\\n",
       "6559          p.Ser202Pro  0/1   44   16   16   5  11  68.75%  3.385400e-05   \n",
       "7839          p.Pro319Ser  0/1   63   33   33  15  17  51.52%  4.101300e-07   \n",
       "15130         p.Thr130Met  1/1   99   18   18   0  18    100%  1.101900e-10   \n",
       "15144         p.Gly290Ser  1/1  200   35   35   0  35    100%  8.913700e-21   \n",
       "16357        p.Arg3564Leu  0/1  255  156  156  77  79  50.64%  2.371700e-30   \n",
       "...                   ...  ...  ...  ...  ...  ..  ..     ...           ...   \n",
       "301568        p.Arg219Lys  0/1   69   53   53  33  20  37.74%  1.040500e-07   \n",
       "301568        p.Arg219Lys  0/1   69   53   53  33  20  37.74%  1.040500e-07   \n",
       "301568        p.Arg219Lys  0/1   69   53   53  33  20  37.74%  1.040500e-07   \n",
       "303230        p.Met145Thr  0/1   53   34   34  19  15  44.12%  4.183500e-06   \n",
       "303233        p.Gly159Ala  0/1   73   41   41  21  20  48.78%  4.334700e-08   \n",
       "\n",
       "        RDF  RDR  ADF  ADR                         SIFT  \\\n",
       "6559      3    2    5    6            deleterious(0.04)   \n",
       "7839      9    6    8    9              tolerated(0.82)   \n",
       "15130     0    0   13    5              tolerated(0.07)   \n",
       "15144     0    0   23   12                 tolerated(1)   \n",
       "16357    56   21   54   25              tolerated(0.08)   \n",
       "...     ...  ...  ...  ...                          ...   \n",
       "301568   29    4   15    5  tolerated low confidence(1)   \n",
       "301568   29    4   15    5  tolerated low confidence(1)   \n",
       "301568   29    4   15    5  tolerated low confidence(1)   \n",
       "303230   10    9   11    4              tolerated(0.82)   \n",
       "303233   12    9   16    4               tolerated(0.7)   \n",
       "\n",
       "                        PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  \\\n",
       "6559               benign(0.037)  0.4349  0.3207  0.5101  0.4425  0.5239   \n",
       "7839                   benign(0)  0.3337  0.5847  0.2723  0.1726  0.3320   \n",
       "15130   possibly damaging(0.471)  0.2238  0.0325  0.2493  0.2371  0.3757   \n",
       "15144              benign(0.001)  0.2666  0.1747  0.2709  0.2371  0.3827   \n",
       "16357              benign(0.389)  0.0164  0.0083  0.0202  0.0000  0.0497   \n",
       "...                          ...     ...     ...     ...     ...     ...   \n",
       "301568                 benign(0)  0.4397  0.7103  0.3487  0.4147  0.2425   \n",
       "301568                 benign(0)  0.4397  0.7103  0.3487  0.4147  0.2425   \n",
       "301568                 benign(0)  0.4397  0.7103  0.3487  0.4147  0.2425   \n",
       "303230                 benign(0)  0.3754  0.3714  0.3516  0.4107  0.3817   \n",
       "303233   possibly damaging(0.52)  0.3159  0.1989  0.3314  0.3889  0.3728   \n",
       "\n",
       "        SAS_AF  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  \\\n",
       "6559    0.4366     0.50540         0.35090         0.55160         0.58110   \n",
       "7839    0.2055     0.30320         0.53920         0.23980         0.29240   \n",
       "15130   0.2945     0.33920         0.08034         0.21870         0.49530   \n",
       "15144   0.2986     0.35210         0.19920         0.22870         0.50560   \n",
       "16357   0.0072     0.04346         0.01662         0.02979         0.06244   \n",
       "...        ...         ...             ...             ...             ...   \n",
       "301568  0.3671     0.32360         0.63010         0.32490         0.28060   \n",
       "301568  0.3671     0.32360         0.63010         0.32490         0.28060   \n",
       "301568  0.3671     0.32360         0.63010         0.32490         0.28060   \n",
       "303230  0.3548     0.35590         0.35610         0.26120         0.33040   \n",
       "303233  0.3292     0.33420         0.20710         0.24800         0.31280   \n",
       "\n",
       "        gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  \\\n",
       "6559            0.4477         0.50440         0.53450         0.53660   \n",
       "7839            0.1721         0.42580         0.31020         0.30210   \n",
       "15130           0.2337         0.41010         0.40110         0.38060   \n",
       "15144           0.2338         0.41520         0.40580         0.39120   \n",
       "16357           0.0001         0.01551         0.06256         0.04835   \n",
       "...                ...             ...             ...             ...   \n",
       "301568          0.4309         0.21960         0.27420         0.31780   \n",
       "301568          0.4309         0.21960         0.27420         0.31780   \n",
       "301568          0.4309         0.21960         0.27420         0.31780   \n",
       "303230          0.3927         0.34630         0.38350         0.35670   \n",
       "303233          0.3752         0.34270         0.37400         0.34000   \n",
       "\n",
       "        gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  \\\n",
       "6559           0.43100     0.47910         0.35430         0.49010   \n",
       "7839           0.22030     0.36740         0.53980         0.18500   \n",
       "15130          0.33810     0.29750         0.08901         0.42970   \n",
       "15144          0.34090     0.33360         0.20380         0.43060   \n",
       "16357          0.02078     0.04517         0.02553         0.04348   \n",
       "...                ...         ...             ...             ...   \n",
       "301568         0.36760     0.38380         0.62500         0.44730   \n",
       "301568         0.36760     0.38380         0.62500         0.44730   \n",
       "301568         0.36760     0.38380         0.62500         0.44730   \n",
       "303230         0.35650     0.37580         0.36750         0.34540   \n",
       "303233         0.33270     0.32630         0.22080         0.33520   \n",
       "\n",
       "        gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  \\\n",
       "6559           0.52020         0.58580          0.4747         0.50000   \n",
       "7839           0.27440         0.29680          0.1775         0.42990   \n",
       "15130          0.27530         0.50090          0.2453         0.40510   \n",
       "15144          0.29310         0.50810          0.2456         0.41030   \n",
       "16357          0.04381         0.05373          0.0000         0.01464   \n",
       "...                ...             ...             ...             ...   \n",
       "301568         0.34620         0.27640          0.4273         0.22180   \n",
       "301568         0.34620         0.27640          0.4273         0.22180   \n",
       "301568         0.34620         0.27640          0.4273         0.22180   \n",
       "303230         0.34480         0.33610          0.3984         0.35810   \n",
       "303233         0.32300         0.31700          0.3788         0.35610   \n",
       "\n",
       "        gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  \\\n",
       "6559            0.5573          0.5379         0.52490         0.44490   \n",
       "7839            0.3829          0.3058         0.33810         0.21710   \n",
       "15130           0.3671          0.4010         0.33240         0.32730   \n",
       "15144           0.3861          0.4048         0.36130         0.33000   \n",
       "16357           0.0303          0.0617         0.05093         0.01697   \n",
       "...                ...             ...             ...             ...   \n",
       "301568          0.3956          0.2729         0.37680         0.37260   \n",
       "301568          0.3956          0.2729         0.37680         0.37260   \n",
       "301568          0.3956          0.2729         0.37680         0.37260   \n",
       "303230          0.3576          0.3932         0.36340         0.35700   \n",
       "303233          0.3354          0.3825         0.32500         0.33260   \n",
       "\n",
       "         MAX_AF  MAX_AF_POPS         BIOTYPE  EXON INTRON  \\\n",
       "6559    0.58580  gnomADg ASJ  protein coding   6/9    NaN   \n",
       "7839    0.58470          AFR  protein coding   2/2    NaN   \n",
       "15130   0.50090  gnomADg ASJ  protein coding   6/9    NaN   \n",
       "15144   0.50810  gnomADg ASJ  protein coding   7/9    NaN   \n",
       "16357   0.06256  gnomADe NFE  protein coding   3/3    NaN   \n",
       "...         ...          ...             ...   ...    ...   \n",
       "301568  0.71030          AFR  protein coding  7/50    NaN   \n",
       "301568  0.71030          AFR  protein coding  7/50    NaN   \n",
       "301568  0.71030          AFR  protein coding  7/50    NaN   \n",
       "303230  0.41070          EAS  protein coding  3/10    NaN   \n",
       "303233  0.38890          EAS  protein coding  3/10    NaN   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "6559                             S202P  Tcc/Ccc     1.0   \n",
       "7839                             P319S  Ccc/Tcc     1.0   \n",
       "15130                            T130M  aCg/aTg     1.0   \n",
       "15144                            G290S  Ggt/Agt     1.0   \n",
       "16357                           R3564L  cGt/cTt    -1.0   \n",
       "...                                ...      ...     ...   \n",
       "301568                           R219K  aGg/aAg    -1.0   \n",
       "301568                           R219K  aGg/aAg    -1.0   \n",
       "301568                           R219K  aGg/aAg    -1.0   \n",
       "303230                           M145T  aTg/aCg     1.0   \n",
       "303233                           G159A  gGc/gCc     1.0   \n",
       "\n",
       "                                                   PUBMED  \n",
       "6559         31227787,24906453,31397093,26129832,24067191  \n",
       "7839    21894447,17975119,29631604,18182569,22305353,2...  \n",
       "15130   18438406,21049557,31892796,30568945,28699600,1...  \n",
       "15144   27417569,18438406,21049557,26738999,31892796,2...  \n",
       "16357                                                 NaN  \n",
       "...                                                   ...  \n",
       "301568  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "301568  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "301568  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "303230                         28146080,29273807,28869550  \n",
       "303233                                                NaN  \n",
       "\n",
       "[238 rows x 74 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['21_Conditions_list'] = df['21_Conditions_list'].str.split('; ')\n",
    "df = df.explode('21_Conditions_list')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "57b039b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75779</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>8</td>\n",
       "      <td>Type 1 Diabetes; Type 2 Diabetes; Maturity ons...</td>\n",
       "      <td>Predisposition for diabetes; Risk factors for ...</td>\n",
       "      <td>nan</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>No</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Maturity onset diabetes mellitus in young,Nonp...</td>\n",
       "      <td>risk factor,benign</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>14937.0</td>\n",
       "      <td>ENST00000257555.11</td>\n",
       "      <td>c.79A&gt;C</td>\n",
       "      <td>ENSP00000257555.5</td>\n",
       "      <td>p.Ile27Leu</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>0</td>\n",
       "      <td>57</td>\n",
       "      <td>100%</td>\n",
       "      <td>6.457200e-34</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>26</td>\n",
       "      <td>tolerated(0.11)</td>\n",
       "      <td>benign(0.282)</td>\n",
       "      <td>0.2985</td>\n",
       "      <td>0.0832</td>\n",
       "      <td>0.3300</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.3390</td>\n",
       "      <td>0.4325</td>\n",
       "      <td>0.35490</td>\n",
       "      <td>0.117900</td>\n",
       "      <td>0.366600</td>\n",
       "      <td>0.46660</td>\n",
       "      <td>0.414700</td>\n",
       "      <td>0.39920</td>\n",
       "      <td>0.33650</td>\n",
       "      <td>0.36130</td>\n",
       "      <td>0.42680</td>\n",
       "      <td>0.28880</td>\n",
       "      <td>0.124800</td>\n",
       "      <td>0.3998</td>\n",
       "      <td>0.34440</td>\n",
       "      <td>0.45790</td>\n",
       "      <td>0.411300</td>\n",
       "      <td>0.40380</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>0.32790</td>\n",
       "      <td>0.31090</td>\n",
       "      <td>0.41680</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>I27L</td>\n",
       "      <td>Atc/Ctc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>25741868,21094359,21647738,21931794,21937998,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75815</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>8</td>\n",
       "      <td>Type 1 Diabetes; Type 2 Diabetes; Maturity ons...</td>\n",
       "      <td>Predisposition for diabetes; Risk factors for ...</td>\n",
       "      <td>nan</td>\n",
       "      <td>rs2464196</td>\n",
       "      <td>No</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997624</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Nonpapillary renal cell carcinoma,Maturity ons...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129230.0</td>\n",
       "      <td>ENST00000257555.11</td>\n",
       "      <td>c.1460G&gt;A</td>\n",
       "      <td>ENSP00000257555.5</td>\n",
       "      <td>p.Ser487Asn</td>\n",
       "      <td>1/1</td>\n",
       "      <td>206</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.259800e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>tolerated(0.19)</td>\n",
       "      <td>benign(0.086)</td>\n",
       "      <td>0.3177</td>\n",
       "      <td>0.1120</td>\n",
       "      <td>0.3501</td>\n",
       "      <td>0.4732</td>\n",
       "      <td>0.3161</td>\n",
       "      <td>0.4141</td>\n",
       "      <td>0.33740</td>\n",
       "      <td>0.119400</td>\n",
       "      <td>0.371100</td>\n",
       "      <td>0.46200</td>\n",
       "      <td>0.486200</td>\n",
       "      <td>0.31240</td>\n",
       "      <td>0.30630</td>\n",
       "      <td>0.33760</td>\n",
       "      <td>0.41660</td>\n",
       "      <td>0.27300</td>\n",
       "      <td>0.128100</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>0.34490</td>\n",
       "      <td>0.45940</td>\n",
       "      <td>0.486900</td>\n",
       "      <td>0.32140</td>\n",
       "      <td>0.5158</td>\n",
       "      <td>0.29930</td>\n",
       "      <td>0.30540</td>\n",
       "      <td>0.40360</td>\n",
       "      <td>0.5158</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S487N</td>\n",
       "      <td>aGc/aAc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>25741868,18439548,21647738,24763700,20416077,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193383</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>CTLA4</td>\n",
       "      <td>8</td>\n",
       "      <td>Type 1 Diabetes; Hyperthyroidism (Graves Disea...</td>\n",
       "      <td>Predisposition for diabetes; Thyroid; Gastriti...</td>\n",
       "      <td>nan</td>\n",
       "      <td>rs231775</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr2</td>\n",
       "      <td>203867991</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>TYPE 1 DIABETES MELLITUS 12, SUSCEPTIBILITY TO...</td>\n",
       "      <td>risk factor,benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>16921.0</td>\n",
       "      <td>ENST00000295854.10</td>\n",
       "      <td>c.49A&gt;G</td>\n",
       "      <td>ENSP00000295854.6</td>\n",
       "      <td>p.Thr17Ala</td>\n",
       "      <td>0/1</td>\n",
       "      <td>136</td>\n",
       "      <td>67</td>\n",
       "      <td>67</td>\n",
       "      <td>31</td>\n",
       "      <td>36</td>\n",
       "      <td>53.73%</td>\n",
       "      <td>2.098000e-14</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>29</td>\n",
       "      <td>7</td>\n",
       "      <td>tolerated low confidence(0.29)</td>\n",
       "      <td>benign(0.015)</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>0.3880</td>\n",
       "      <td>0.4625</td>\n",
       "      <td>0.6369</td>\n",
       "      <td>0.3588</td>\n",
       "      <td>0.3098</td>\n",
       "      <td>0.41510</td>\n",
       "      <td>0.373800</td>\n",
       "      <td>0.461900</td>\n",
       "      <td>0.30160</td>\n",
       "      <td>0.666400</td>\n",
       "      <td>0.52790</td>\n",
       "      <td>0.38190</td>\n",
       "      <td>0.38980</td>\n",
       "      <td>0.31930</td>\n",
       "      <td>0.39960</td>\n",
       "      <td>0.374200</td>\n",
       "      <td>0.4286</td>\n",
       "      <td>0.41190</td>\n",
       "      <td>0.30020</td>\n",
       "      <td>0.645000</td>\n",
       "      <td>0.52970</td>\n",
       "      <td>0.2771</td>\n",
       "      <td>0.38560</td>\n",
       "      <td>0.35920</td>\n",
       "      <td>0.31960</td>\n",
       "      <td>0.6664</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T17A</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19131662,19263529,19330901,19559392,25474356,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216244</th>\n",
       "      <td>C1QTNF6</td>\n",
       "      <td>C1QTNF6</td>\n",
       "      <td>No</td>\n",
       "      <td>Type 1 Diabetes; Gastritis</td>\n",
       "      <td>Predisposition for diabetes; Gastritis</td>\n",
       "      <td>nan</td>\n",
       "      <td>rs229527</td>\n",
       "      <td>No</td>\n",
       "      <td>chr22</td>\n",
       "      <td>37185445</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000337843.7</td>\n",
       "      <td>c.62G&gt;T</td>\n",
       "      <td>ENSP00000338812.2</td>\n",
       "      <td>p.Gly21Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>152</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.136300e-16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>11</td>\n",
       "      <td>tolerated low confidence(0.15)</td>\n",
       "      <td>possibly damaging(0.455)</td>\n",
       "      <td>0.4501</td>\n",
       "      <td>0.3457</td>\n",
       "      <td>0.3934</td>\n",
       "      <td>0.6548</td>\n",
       "      <td>0.4225</td>\n",
       "      <td>0.4489</td>\n",
       "      <td>0.43200</td>\n",
       "      <td>0.339500</td>\n",
       "      <td>0.369300</td>\n",
       "      <td>0.49230</td>\n",
       "      <td>0.661300</td>\n",
       "      <td>0.38950</td>\n",
       "      <td>0.42410</td>\n",
       "      <td>0.43220</td>\n",
       "      <td>0.45200</td>\n",
       "      <td>0.40680</td>\n",
       "      <td>0.343900</td>\n",
       "      <td>0.4031</td>\n",
       "      <td>0.40540</td>\n",
       "      <td>0.50430</td>\n",
       "      <td>0.651600</td>\n",
       "      <td>0.38120</td>\n",
       "      <td>0.5348</td>\n",
       "      <td>0.42080</td>\n",
       "      <td>0.43370</td>\n",
       "      <td>0.45640</td>\n",
       "      <td>0.6613</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G21V</td>\n",
       "      <td>gGg/gTg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20410501,26870082,19474294,20203524,29070082,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268147</th>\n",
       "      <td>TAGAP,LOC105378083</td>\n",
       "      <td>TAGAP</td>\n",
       "      <td>No</td>\n",
       "      <td>Type 1 Diabetes; Gastritis</td>\n",
       "      <td>Predisposition for diabetes; Gastritis</td>\n",
       "      <td>nan</td>\n",
       "      <td>rs41267765</td>\n",
       "      <td>No</td>\n",
       "      <td>chr6</td>\n",
       "      <td>159041392</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000338313.5</td>\n",
       "      <td>c.439G&gt;A</td>\n",
       "      <td>ENSP00000340217.5</td>\n",
       "      <td>p.Glu147Lys</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>37.5%</td>\n",
       "      <td>6.061800e-06</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(0.62)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.0106</td>\n",
       "      <td>0.0083</td>\n",
       "      <td>0.0086</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.0133</td>\n",
       "      <td>0.01861</td>\n",
       "      <td>0.005906</td>\n",
       "      <td>0.009887</td>\n",
       "      <td>0.01825</td>\n",
       "      <td>0.000326</td>\n",
       "      <td>0.02934</td>\n",
       "      <td>0.02332</td>\n",
       "      <td>0.02558</td>\n",
       "      <td>0.01986</td>\n",
       "      <td>0.01615</td>\n",
       "      <td>0.005939</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.01375</td>\n",
       "      <td>0.01816</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.02884</td>\n",
       "      <td>0.0443</td>\n",
       "      <td>0.02163</td>\n",
       "      <td>0.02199</td>\n",
       "      <td>0.02056</td>\n",
       "      <td>0.0443</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E147K</td>\n",
       "      <td>Gag/Aag</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23261300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene Name     Gene Gene_Score  \\\n",
       "75779                HNF1A    HNF1A          8   \n",
       "75815                HNF1A    HNF1A          8   \n",
       "193383               CTLA4    CTLA4          8   \n",
       "216244             C1QTNF6  C1QTNF6         No   \n",
       "268147  TAGAP,LOC105378083    TAGAP         No   \n",
       "\n",
       "                                                Condition  \\\n",
       "75779   Type 1 Diabetes; Type 2 Diabetes; Maturity ons...   \n",
       "75815   Type 1 Diabetes; Type 2 Diabetes; Maturity ons...   \n",
       "193383  Type 1 Diabetes; Hyperthyroidism (Graves Disea...   \n",
       "216244                        Type 1 Diabetes; Gastritis    \n",
       "268147                        Type 1 Diabetes; Gastritis    \n",
       "\n",
       "                                                 Headings 21_Conditions_list  \\\n",
       "75779   Predisposition for diabetes; Risk factors for ...                nan   \n",
       "75815   Predisposition for diabetes; Risk factors for ...                nan   \n",
       "193383  Predisposition for diabetes; Thyroid; Gastriti...                nan   \n",
       "216244            Predisposition for diabetes; Gastritis                 nan   \n",
       "268147            Predisposition for diabetes; Gastritis                 nan   \n",
       "\n",
       "              rsID Literature  CHROM        POS REF ALT      Zygosity  \\\n",
       "75779    rs1169288         No  chr12  120978847   A   C    Homozygous   \n",
       "75815    rs2464196         No  chr12  120997624   G   A    Homozygous   \n",
       "193383    rs231775        Yes   chr2  203867991   A   G  Heterozygous   \n",
       "216244    rs229527         No  chr22   37185445   C   A    Homozygous   \n",
       "268147  rs41267765         No   chr6  159041392   C   T  Heterozygous   \n",
       "\n",
       "             Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "75779   missense variant              7/10  MODERATE           5.0   \n",
       "75815   missense variant              7/10  MODERATE           5.0   \n",
       "193383  missense variant              7/10  MODERATE           5.0   \n",
       "216244  missense variant              7/10  MODERATE           5.0   \n",
       "268147  missense variant              7/10  MODERATE           5.0   \n",
       "\n",
       "                                            ClinVar_CLNDN            CLIN_SIG  \\\n",
       "75779   Maturity onset diabetes mellitus in young,Nonp...  risk factor,benign   \n",
       "75815   Nonpapillary renal cell carcinoma,Maturity ons...              benign   \n",
       "193383  TYPE 1 DIABETES MELLITUS 12, SUSCEPTIBILITY TO...  risk factor,benign   \n",
       "216244                                                NaN                 NaN   \n",
       "268147                                                NaN                 NaN   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT   ClinVar  \\\n",
       "75779      criteria provided, conflicting interpretations   14937.0   \n",
       "75815   criteria provided, multiple submitters, no con...  129230.0   \n",
       "193383  criteria provided, multiple submitters, no con...   16921.0   \n",
       "216244                                                NaN       NaN   \n",
       "268147                                                NaN       NaN   \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "75779   ENST00000257555.11            c.79A>C  ENSP00000257555.5   \n",
       "75815   ENST00000257555.11          c.1460G>A  ENSP00000257555.5   \n",
       "193383  ENST00000295854.10            c.49A>G  ENSP00000295854.6   \n",
       "216244   ENST00000337843.7            c.62G>T  ENSP00000338812.2   \n",
       "268147   ENST00000338313.5           c.439G>A  ENSP00000340217.5   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "75779          p.Ile27Leu  1/1  255   57  57   0  57    100%  6.457200e-34   \n",
       "75815         p.Ser487Asn  1/1  206   36  36   0  36    100%  2.259800e-21   \n",
       "193383         p.Thr17Ala  0/1  136   67  67  31  36  53.73%  2.098000e-14   \n",
       "216244         p.Gly21Val  1/1  152   27  27   0  27    100%  5.136300e-16   \n",
       "268147        p.Glu147Lys  0/1   52   40  40  25  15   37.5%  6.061800e-06   \n",
       "\n",
       "        RDF  RDR  ADF  ADR                            SIFT  \\\n",
       "75779     0    0   31   26                 tolerated(0.11)   \n",
       "75815     0    0   28    8                 tolerated(0.19)   \n",
       "193383   26    5   29    7  tolerated low confidence(0.29)   \n",
       "216244    0    0   16   11  tolerated low confidence(0.15)   \n",
       "268147   24    1   11    4                 tolerated(0.62)   \n",
       "\n",
       "                        PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  \\\n",
       "75779              benign(0.282)  0.2985  0.0832  0.3300  0.3889  0.3390   \n",
       "75815              benign(0.086)  0.3177  0.1120  0.3501  0.4732  0.3161   \n",
       "193383             benign(0.015)  0.4273  0.3880  0.4625  0.6369  0.3588   \n",
       "216244  possibly damaging(0.455)  0.4501  0.3457  0.3934  0.6548  0.4225   \n",
       "268147                 benign(0)  0.0106  0.0083  0.0086  0.0000  0.0229   \n",
       "\n",
       "        SAS_AF  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  \\\n",
       "75779   0.4325     0.35490        0.117900        0.366600         0.46660   \n",
       "75815   0.4141     0.33740        0.119400        0.371100         0.46200   \n",
       "193383  0.3098     0.41510        0.373800        0.461900         0.30160   \n",
       "216244  0.4489     0.43200        0.339500        0.369300         0.49230   \n",
       "268147  0.0133     0.01861        0.005906        0.009887         0.01825   \n",
       "\n",
       "        gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  \\\n",
       "75779         0.414700         0.39920         0.33650         0.36130   \n",
       "75815         0.486200         0.31240         0.30630         0.33760   \n",
       "193383        0.666400         0.52790         0.38190         0.38980   \n",
       "216244        0.661300         0.38950         0.42410         0.43220   \n",
       "268147        0.000326         0.02934         0.02332         0.02558   \n",
       "\n",
       "        gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  \\\n",
       "75779          0.42680     0.28880        0.124800          0.3998   \n",
       "75815          0.41660     0.27300        0.128100          0.3568   \n",
       "193383         0.31930     0.39960        0.374200          0.4286   \n",
       "216244         0.45200     0.40680        0.343900          0.4031   \n",
       "268147         0.01986     0.01615        0.005939          0.0000   \n",
       "\n",
       "        gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  \\\n",
       "75779          0.34440         0.45790        0.411300         0.40380   \n",
       "75815          0.34490         0.45940        0.486900         0.32140   \n",
       "193383         0.41190         0.30020        0.645000         0.52970   \n",
       "216244         0.40540         0.50430        0.651600         0.38120   \n",
       "268147         0.01375         0.01816        0.000385         0.02884   \n",
       "\n",
       "        gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  \\\n",
       "75779           0.5222         0.32790         0.31090         0.41680   \n",
       "75815           0.5158         0.29930         0.30540         0.40360   \n",
       "193383          0.2771         0.38560         0.35920         0.31960   \n",
       "216244          0.5348         0.42080         0.43370         0.45640   \n",
       "268147          0.0443         0.02163         0.02199         0.02056   \n",
       "\n",
       "        MAX_AF  MAX_AF_POPS         BIOTYPE  EXON INTRON  \\\n",
       "75779   0.5222  gnomADg MID  protein coding  1/10    NaN   \n",
       "75815   0.5158  gnomADg MID  protein coding  7/10    NaN   \n",
       "193383  0.6664  gnomADe EAS  protein coding   1/3    NaN   \n",
       "216244  0.6613  gnomADe EAS  protein coding   2/3    NaN   \n",
       "268147  0.0443  gnomADg MID  protein coding   6/8    NaN   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "75779                             I27L  Atc/Ctc     1.0   \n",
       "75815                            S487N  aGc/aAc     1.0   \n",
       "193383                            T17A  Acc/Gcc     1.0   \n",
       "216244                            G21V  gGg/gTg    -1.0   \n",
       "268147                           E147K  Gag/Aag    -1.0   \n",
       "\n",
       "                                                   PUBMED  \n",
       "75779   25741868,21094359,21647738,21931794,21937998,2...  \n",
       "75815   25741868,18439548,21647738,24763700,20416077,2...  \n",
       "193383  19131662,19263529,19330901,19559392,25474356,2...  \n",
       "216244  20410501,26870082,19474294,20203524,29070082,2...  \n",
       "268147                                           23261300  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['21_Conditions_list'] == \"nan\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "ce21c0ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group the data by the '21_Conditions_list' column\n",
    "grouped = df_1.groupby('21_Conditions_list')\n",
    "\n",
    "# Create an Excel writer object\n",
    "excel_writer = pd.ExcelWriter(r'C:/Users/GenepoweRx_Madhu/Documents/583_output.xlsx', engine='xlsxwriter')\n",
    "\n",
    "# Iterate through each group and save it as a separate sheet\n",
    "for name, group in grouped:\n",
    "    # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "    sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', name)[:31]\n",
    "    \n",
    "    # Sort the group based on the \"Literature\" column\n",
    "    group_sorted = group.sort_values(by='Condition', ascending=True)\n",
    "    group_sorted = group.sort_values(by='Literature', ascending=False)\n",
    "    \n",
    "    # Write each group to a separate sheet with the modified sheet name\n",
    "    group_sorted.to_excel(excel_writer, sheet_name=sheet_name, index=False)\n",
    "\n",
    "# Save the Excel file\n",
    "excel_writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "3a4feb79",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group the data by the '21_Conditions_list' column\n",
    "grouped = df_1.groupby('21_Conditions_list')\n",
    "\n",
    "# Create an Excel writer object\n",
    "excel_writer = pd.ExcelWriter(r'C:/Users/GenepoweRx_Madhu/Documents/583_output.xlsx', engine='xlsxwriter')\n",
    "\n",
    "# Iterate through each group and save it as a separate sheet\n",
    "for name, group in grouped:\n",
    "    # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "    sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', name)[:31]\n",
    "    \n",
    "    # Sort the group first by the \"Condition\" column and then by the \"Literature\" column\n",
    "    group_sorted = group.sort_values(by=['Condition', 'Literature'], ascending=[True, False])\n",
    "    \n",
    "    # Delete the '21_Conditions_list' column\n",
    "    group_sorted = group_sorted.drop(columns=['21_Conditions_list', 'Gene Name'])\n",
    "    \n",
    "    columns_to_keep = ['Condition', 'Headings', 'Gene', 'Gene_Score', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score']\n",
    "\n",
    "    # Select only the desired columns\n",
    "    group_sorted = group_sorted[columns_to_keep]\n",
    "    \n",
    "    # Write each group to a separate sheet with the modified sheet name\n",
    "    group_selected.to_excel(excel_writer, sheet_name=sheet_name, index=False)\n",
    "\n",
    "# Save the Excel file\n",
    "excel_writer.save()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "3c1415e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import xlsxwriter\n",
    "\n",
    "# Group the data by the '21_Conditions_list' column\n",
    "grouped = df_1.groupby('21_Conditions_list')\n",
    "\n",
    "# Create an Excel writer object\n",
    "excel_writer = pd.ExcelWriter(r'C:/Users/GenepoweRx_Madhu/Documents/583_output.xlsx', engine='xlsxwriter')\n",
    "\n",
    "# Define a workbook\n",
    "workbook = excel_writer.book\n",
    "\n",
    "# Define the green and pink cell formats\n",
    "green_format = workbook.add_format({'bg_color': '#00FF00'})  # Green color\n",
    "pink_format = workbook.add_format({'bg_color': '#FF00FF'})   # Pink color\n",
    "\n",
    "# Iterate through each group and save it as a separate sheet\n",
    "for name, group in grouped:\n",
    "    # Modify the name to make it fit within the Excel sheet name limit (31 characters)\n",
    "    sheet_name = re.sub(r'[\\/:*?\"<>|]', '_', name)[:31]\n",
    "    \n",
    "    # Sort the group first by the \"Condition\" column and then by the \"Literature\" column\n",
    "    group_sorted = group.sort_values(by=['Condition', 'Literature'], ascending=[True, False])\n",
    "    \n",
    "    # Delete the '21_Conditions_list' column\n",
    "    group_sorted = group_sorted.drop(columns=['21_Conditions_list', 'Gene Name'])\n",
    "    \n",
    "    columns_to_keep = ['Condition', 'Headings', 'Gene', 'Gene_Score', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score']\n",
    "    \n",
    "    # Select only the desired columns\n",
    "    group_selected = group_sorted[columns_to_keep]\n",
    "    \n",
    "    # Write each group to a separate sheet with the modified sheet name\n",
    "    group_selected.to_excel(excel_writer, sheet_name=sheet_name, index=False)\n",
    "    \n",
    "    # Access the worksheet to apply conditional formatting\n",
    "    worksheet = excel_writer.sheets[sheet_name]\n",
    "    \n",
    "    # Apply conditional formatting based on 'Literature' column values\n",
    "    for i, value in enumerate(group_selected['Literature']):\n",
    "        if value == 'Yes':\n",
    "            worksheet.write(i + 1, columns_to_keep.index('Literature'), value, green_format)\n",
    "        elif value == 'No':\n",
    "            worksheet.write(i + 1, columns_to_keep.index('Literature'), value, pink_format)\n",
    "\n",
    "# Save the Excel file\n",
    "excel_writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ded77294",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "ddf0e681",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/final_583rows.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "ec3f6251",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes present in VCF file have been saved to C:/Users/GenepoweRx_Madhu/Downloads/biotype_final_csq_specific_gene.xlsx\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Define the genes to check\n",
    "genes_to_check = [\"ABCG2\", \"CACNA1S\", \"CFTR\", \"CYP2B6\", \"CYP2C19\", \"CYP2C9\", \"CYP2D6\", \"CYP3A5\", \"DPYD\", \"G6PD\", \"NUDT15\", \"RYR1\", \"SLCO1B1\", \"TPMT\", \"UGT1A1\", \"OR4F5\"]\n",
    "\n",
    "# Initialize an empty list to store data\n",
    "gene_info_list = []\n",
    "\n",
    "# Open the VCF file\n",
    "with open('C:/Users/GenepoweRx_Madhu/Downloads/718_final.vcf', 'r') as vcf_file:\n",
    "    lines = vcf_file.readlines()\n",
    "\n",
    "    # Initialize a record flag\n",
    "    is_in_record = False\n",
    "\n",
    "    # Initialize a dictionary to store record data\n",
    "    record_data = {}\n",
    "\n",
    "    for line in lines:\n",
    "        line = line.strip()\n",
    "\n",
    "        if line.startswith('#'):\n",
    "            continue  # Skip header lines\n",
    "\n",
    "        if line == \"\":\n",
    "            if is_in_record:\n",
    "                gene_symbols = record_data.get(\"INFO\", {}).get(\"GENEINFO\", \"\").split('|')\n",
    "                for gene_symbol in gene_symbols:\n",
    "                    gene_data = gene_symbol.split(':')\n",
    "                    gene_name = gene_data[0]\n",
    "\n",
    "                    if gene_name in genes_to_check:\n",
    "                        # Extract relevant fields and store them as a dictionary\n",
    "                        data = {\n",
    "                            \"Gene\": gene_name,\n",
    "                            \"#CHROM\": record_data.get(\"#CHROM\", \"\"),\n",
    "                            \"POS\": record_data.get(\"POS\", \"\"),\n",
    "                            \"ID\": record_data.get(\"ID\", \"\"),\n",
    "                            \"REF\": record_data.get(\"REF\", \"\"),\n",
    "                            \"ALT\": record_data.get(\"ALT\", \"\"),\n",
    "                            \"QUAL\": record_data.get(\"QUAL\", \"\"),\n",
    "                            \"FILTER\": record_data.get(\"FILTER\", \"\"),\n",
    "                            \"INFO\": record_data.get(\"INFO\", \"\"),\n",
    "                            \"FORMAT\": record_data.get(\"FORMAT\", \"\"),\n",
    "                            \"Sample1\": record_data.get(\"Sample1\", \"\"),\n",
    "                            \"GT\": record_data.get(\"GT\", \"\"),\n",
    "                            \"GQ\": record_data.get(\"GQ\", \"\"),\n",
    "                            \"SDP\": record_data.get(\"SDP\", \"\"),\n",
    "                            \"DP\": record_data.get(\"DP\", \"\"),\n",
    "                            \"RD\": record_data.get(\"RD\", \"\"),\n",
    "                            \"AD\": record_data.get(\"AD\", \"\"),\n",
    "                            \"FREQ\": record_data.get(\"FREQ\", \"\"),\n",
    "                            \"PVAL\": record_data.get(\"PVAL\", \"\"),\n",
    "                            \"RBQ\": record_data.get(\"RBQ\", \"\"),\n",
    "                            \"ABQ\": record_data.get(\"ABQ\", \"\"),\n",
    "                            \"RDF\": record_data.get(\"RDF\", \"\"),\n",
    "                            \"RDR\": record_data.get(\"RDR\", \"\"),\n",
    "                            \"ADF\": record_data.get(\"ADF\", \"\"),\n",
    "                            \"ADR\": record_data.get(\"ADR\", \"\"),\n",
    "                        }\n",
    "\n",
    "                        # Extract CSQ if it's present in INFO\n",
    "                        if \"CSQ\" in record_data.get(\"INFO\", {}):\n",
    "                            csq_values_list = record_data[\"INFO\"][\"CSQ\"].split(',')\n",
    "                            csq_values = csq_values_list[0].split('|')  # Corrected this line\n",
    "\n",
    "                            csq_columns = [\n",
    "                                # ... (rest of the columns as before)\n",
    "                            ]\n",
    "\n",
    "                            csq_data = {column: value if value != '' else 'NA' for column, value in\n",
    "                                        zip(csq_columns, csq_values)}\n",
    "                            data.update({\"CSQ_\" + column: csq_data.get(column, \"NA\") for column in csq_columns})\n",
    "\n",
    "                            # Extract the \"Biotype\" values and concatenate them\n",
    "                            biotype_values = []\n",
    "                            for transcript_csq in csq_values_list:\n",
    "                                transcript_values = transcript_csq.split('|')\n",
    "                                if len(transcript_values) >= 8:\n",
    "                                    biotype = transcript_values[7]\n",
    "                                    biotype_values.append(biotype)\n",
    "\n",
    "                            # Concatenate the Biotype values with commas\n",
    "                            data[\"Biotype\"] = \",\".join(biotype_values)\n",
    "\n",
    "                            gene_info_list.append(data)\n",
    "\n",
    "                # Reset the record_data dictionary and flag\n",
    "                record_data = {}\n",
    "                is_in_record = False\n",
    "        else:\n",
    "            # Split the line into fields\n",
    "            fields = line.split('\\t')\n",
    "            if len(fields) >= 10:\n",
    "                # Store the fields in the record_data dictionary\n",
    "                record_data[\"#CHROM\"] = fields[0]\n",
    "                record_data[\"POS\"] = fields[1]\n",
    "                record_data[\"ID\"] = fields[2]\n",
    "                record_data[\"REF\"] = fields[3]\n",
    "                record_data[\"ALT\"] = fields[4]\n",
    "                record_data[\"QUAL\"] = fields[5]\n",
    "                record_data[\"FILTER\"] = fields[6]\n",
    "                record_data[\"INFO\"] = fields[7]\n",
    "                record_data[\"FORMAT\"] = fields[8]\n",
    "                record_data[\"Sample1\"] = fields[9]\n",
    "\n",
    "                # Extract GT, GQ, SDP, DP, RD, AD, FREQ, PVAL, RBQ, ABQ, RDF, RDR, ADF, ADR values\n",
    "                format_fields = fields[8].split(':')\n",
    "                sample_fields = fields[9].split(':')\n",
    "                record_data[\"GT\"] = sample_fields[format_fields.index(\"GT\")]\n",
    "                record_data[\"GQ\"] = sample_fields[format_fields.index(\"GQ\")]\n",
    "                record_data[\"SDP\"] = sample_fields[format_fields.index(\"SDP\")]\n",
    "                record_data[\"DP\"] = sample_fields[format_fields.index(\"DP\")]\n",
    "                record_data[\"RD\"] = sample_fields[format_fields.index(\"RD\")]\n",
    "                record_data[\"AD\"] = sample_fields[format_fields.index(\"AD\")]\n",
    "                record_data[\"FREQ\"] = sample_fields[format_fields.index(\"FREQ\")]\n",
    "                record_data[\"PVAL\"] = sample_fields[format_fields.index(\"PVAL\")]\n",
    "                record_data[\"RBQ\"] = sample_fields[format_fields.index(\"RBQ\")]\n",
    "                record_data[\"ABQ\"] = sample_fields[format_fields.index(\"ABQ\")]\n",
    "                record_data[\"RDF\"] = sample_fields[format_fields.index(\"RDF\")]\n",
    "                record_data[\"RDR\"] = sample_fields[format_fields.index(\"RDR\")]\n",
    "                record_data[\"ADF\"] = sample_fields[format_fields.index(\"ADF\")]\n",
    "                record_data[\"ADR\"] = sample_fields[format_fields.index(\"ADR\")]\n",
    "\n",
    "                is_in_record = True\n",
    "\n",
    "# Create a DataFrame from the list\n",
    "gene_info = pd.DataFrame(gene_info_list)\n",
    "\n",
    "# Specify the correct file path for the Excel file\n",
    "output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/biotype_final_csq_specific_gene.xlsx'\n",
    "gene_info.to_excel(output_file, sheet_name='Genes', index=False)\n",
    "\n",
    "print(f\"Genes present in VCF file have been saved to {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "8930dd42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:92:51:51:26:25:49.02%:5.8079E-10:41:51:10:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15118</td>\n",
       "      <td>rs11580262</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:23:20:20:13:7:35%:4.158E-3:42:45:8:5:4:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:107:34:34:9:25:73.53%:1.9819E-11:37:44:8:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16949</td>\n",
       "      <td>rs199745162</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:55:85:85:68:17:20%:3.1403E-6:52:39:55:13:11:6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24786</td>\n",
       "      <td>rs11266852</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:190:93:93:43:50:53.76%:9.1368E-20:63:55:39...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71451</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209899</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:21:21:10:11:52.38%:8.2399E-5:62:54:10:0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71452</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:33:16:16:7:9:56.25%:4.0786E-4:74:64:7:0:8:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71453</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210723</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:29:37:37:28:9:24.32%:1.1256E-3:68:68:28:0:9:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71454</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210771</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:30:30:30:21:9:30%:9.678E-4:59:65:17:4:8:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71455</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211636</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:59:19:19:5:14:73.68%:1.2025E-6:66:52:4:1:10:4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71456 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14930    rs6682385   A   G    .   PASS   \n",
       "1      chr1     15118   rs11580262   A   G    .   PASS   \n",
       "2      chr1     16495    rs3210724   G   C    .   PASS   \n",
       "3      chr1     16949  rs199745162   A   C    .   PASS   \n",
       "4      chr1     24786   rs11266852   G   A    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "71451  chrY  57209899            .   C   T    .   PASS   \n",
       "71452  chrY  57210016            .   C   T    .   PASS   \n",
       "71453  chrY  57210723            .   G   A    .   PASS   \n",
       "71454  chrY  57210771            .   A   C    .   PASS   \n",
       "71455  chrY  57211636            .   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....   \n",
       "1      ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...   \n",
       "2      ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "3      ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...   \n",
       "4      ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...   \n",
       "...                                                  ...   \n",
       "71451  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...   \n",
       "71452  ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "71453  ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...   \n",
       "71454  ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...   \n",
       "71455  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "71451  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71452  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71453  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71454  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71455  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      0/1:92:51:51:26:25:49.02%:5.8079E-10:41:51:10:...  \n",
       "1           0/1:23:20:20:13:7:35%:4.158E-3:42:45:8:5:4:3  \n",
       "2      0/1:107:34:34:9:25:73.53%:1.9819E-11:37:44:8:1...  \n",
       "3      0/1:55:85:85:68:17:20%:3.1403E-6:52:39:55:13:11:6  \n",
       "4      0/1:190:93:93:43:50:53.76%:9.1368E-20:63:55:39...  \n",
       "...                                                  ...  \n",
       "71451  0/1:40:21:21:10:11:52.38%:8.2399E-5:62:54:10:0...  \n",
       "71452    0/1:33:16:16:7:9:56.25%:4.0786E-4:74:64:7:0:8:1  \n",
       "71453  0/1:29:37:37:28:9:24.32%:1.1256E-3:68:68:28:0:9:0  \n",
       "71454      0/1:30:30:30:21:9:30%:9.678E-4:59:65:17:4:8:1  \n",
       "71455  0/1:59:19:19:5:14:73.68%:1.2025E-6:66:52:4:1:10:4  \n",
       "\n",
       "[71456 rows x 10 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/718_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False, skiprows=80)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "c62d0640",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>49.02%</td>\n",
       "      <td>5.8079E-10</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15118</td>\n",
       "      <td>rs11580262</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>35%</td>\n",
       "      <td>4.158E-3</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>107</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>73.53%</td>\n",
       "      <td>1.9819E-11</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16949</td>\n",
       "      <td>rs199745162</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>55</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>68</td>\n",
       "      <td>17</td>\n",
       "      <td>20%</td>\n",
       "      <td>3.1403E-6</td>\n",
       "      <td>55</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24786</td>\n",
       "      <td>rs11266852</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>190</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "      <td>43</td>\n",
       "      <td>50</td>\n",
       "      <td>53.76%</td>\n",
       "      <td>9.1368E-20</td>\n",
       "      <td>39</td>\n",
       "      <td>4</td>\n",
       "      <td>37</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71451</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209899</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>52.38%</td>\n",
       "      <td>8.2399E-5</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71452</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>56.25%</td>\n",
       "      <td>4.0786E-4</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71453</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210723</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>28</td>\n",
       "      <td>9</td>\n",
       "      <td>24.32%</td>\n",
       "      <td>1.1256E-3</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71454</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210771</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>30%</td>\n",
       "      <td>9.678E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71455</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211636</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>59</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>73.68%</td>\n",
       "      <td>1.2025E-6</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71456 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14930    rs6682385   A   G    .   PASS   \n",
       "1      chr1     15118   rs11580262   A   G    .   PASS   \n",
       "2      chr1     16495    rs3210724   G   C    .   PASS   \n",
       "3      chr1     16949  rs199745162   A   C    .   PASS   \n",
       "4      chr1     24786   rs11266852   G   A    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "71451  chrY  57209899            .   C   T    .   PASS   \n",
       "71452  chrY  57210016            .   C   T    .   PASS   \n",
       "71453  chrY  57210723            .   G   A    .   PASS   \n",
       "71454  chrY  57210771            .   A   C    .   PASS   \n",
       "71455  chrY  57211636            .   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....  0/1   92  51  51   \n",
       "1      ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...  0/1   23  20  20   \n",
       "2      ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...  0/1  107  34  34   \n",
       "3      ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...  0/1   55  85  85   \n",
       "4      ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...  0/1  190  93  93   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "71451  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...  0/1   40  21  21   \n",
       "71452  ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...  0/1   33  16  16   \n",
       "71453  ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...  0/1   29  37  37   \n",
       "71454  ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1   30  30  30   \n",
       "71455  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...  0/1   59  19  19   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR  \n",
       "0      26  25  49.02%  5.8079E-10  10  16  17   8  \n",
       "1      13   7     35%    4.158E-3   8   5   4   3  \n",
       "2       9  25  73.53%  1.9819E-11   8   1  24   1  \n",
       "3      68  17     20%   3.1403E-6  55  13  11   6  \n",
       "4      43  50  53.76%  9.1368E-20  39   4  37  13  \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  \n",
       "71451  10  11  52.38%   8.2399E-5  10   0   9   2  \n",
       "71452   7   9  56.25%   4.0786E-4   7   0   8   1  \n",
       "71453  28   9  24.32%   1.1256E-3  28   0   9   0  \n",
       "71454  21   9     30%    9.678E-4  17   4   8   1  \n",
       "71455   5  14  73.68%   1.2025E-6   4   1  10   4  \n",
       "\n",
       "[71456 rows x 20 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/718_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False, skiprows=80)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "ec180ec5",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "not enough values to unpack (expected 10, got 8)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[45], line 23\u001b[0m\n\u001b[0;32m     21\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m line \u001b[38;5;129;01min\u001b[39;00m lines:\n\u001b[0;32m     22\u001b[0m     line \u001b[38;5;241m=\u001b[39m line\u001b[38;5;241m.\u001b[39mstrip()\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m---> 23\u001b[0m     chrom, pos, rsid, ref, alt, qual, filt, info, fmt, samples \u001b[38;5;241m=\u001b[39m line\n\u001b[0;32m     25\u001b[0m     \u001b[38;5;66;03m# Split the 'SAMPLE' column\u001b[39;00m\n\u001b[0;32m     26\u001b[0m     sample_values \u001b[38;5;241m=\u001b[39m samples\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[1;31mValueError\u001b[0m: not enough values to unpack (expected 10, got 8)"
     ]
    }
   ],
   "source": [
    "# Initialize variables to store data\n",
    "vcf_data = []\n",
    "additional_columns = ['ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID', 'SLO',\n",
    "                      'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV', 'TPA', 'CFL', 'GNO',\n",
    "                      'VLD', 'ASP', 'ASS', 'REF', 'U3', 'U5', 'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC',\n",
    "                      'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO',\n",
    "                      'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar',\n",
    "                      'ClinVar_CLNSIG']\n",
    "\n",
    "# Read the VCF file line by line (skip first 80 lines)\n",
    "with open(r'C:/Users/GenepoweRx_Madhu/Downloads/718_final.vcf', 'r') as vcf_file:\n",
    "    lines = vcf_file.readlines()\n",
    "    lines = lines[80:]  # Skip first 80 lines\n",
    "\n",
    "    # Initialize variables for sample columns\n",
    "    sample_columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "    # Initialize variables for samples data\n",
    "    sample_data = []\n",
    "\n",
    "    for line in lines:\n",
    "        line = line.strip().split('\\t')\n",
    "        chrom, pos, rsid, ref, alt, qual, filt, info, fmt, samples = line\n",
    "\n",
    "        # Split the 'SAMPLE' column\n",
    "        sample_values = samples.split(':')\n",
    "\n",
    "        # Create a dictionary to store the sample data\n",
    "        sample_dict = {col: value for col, value in zip(sample_columns, sample_values)}\n",
    "\n",
    "        # Append the sample data to the list\n",
    "        sample_data.append(sample_dict)\n",
    "\n",
    "        # Create a dictionary to store the additional columns\n",
    "        additional_dict = {col: '' for col in additional_columns}\n",
    "\n",
    "        # Split the 'INFO' field into key-value pairs\n",
    "        info_fields = info.split(';')\n",
    "        for field in info_fields:\n",
    "            key_value = field.split('=')\n",
    "            key = key_value[0]\n",
    "            if key in additional_columns:\n",
    "                if len(key_value) > 1:\n",
    "                    value = f\"{key}={key_value[1]}\"\n",
    "                else:\n",
    "                    value = key\n",
    "                additional_dict[key] = value\n",
    "\n",
    "        # Append all the data to the VCF data list\n",
    "        vcf_data.append([chrom, pos, rsid, ref, alt, qual, filt, info] +\n",
    "                         sample_values + list(additional_dict.values()))\n",
    "\n",
    "# Define column names\n",
    "columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO'] + sample_columns + additional_columns\n",
    "\n",
    "# Print the resulting data\n",
    "for row in [columns] + vcf_data:\n",
    "    print('\\t'.join(row))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "13aa88a6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'New_Column'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3802\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3801\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:138\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:165\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5745\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5753\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'New_Column'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[41], line 14\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[38;5;66;03m# Populate columns based on 'info' values\u001b[39;00m\n\u001b[0;32m     13\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i, row \u001b[38;5;129;01min\u001b[39;00m vcf\u001b[38;5;241m.\u001b[39miterrows():\n\u001b[1;32m---> 14\u001b[0m     info \u001b[38;5;241m=\u001b[39m \u001b[43mrow\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mNew_Column\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[0;32m     15\u001b[0m     items \u001b[38;5;241m=\u001b[39m info\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m;\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m     16\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m item \u001b[38;5;129;01min\u001b[39;00m items:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:981\u001b[0m, in \u001b[0;36mSeries.__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m    978\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[key]\n\u001b[0;32m    980\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m key_is_scalar:\n\u001b[1;32m--> 981\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_value\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    983\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_hashable(key):\n\u001b[0;32m    984\u001b[0m     \u001b[38;5;66;03m# Otherwise index.get_value will raise InvalidIndexError\u001b[39;00m\n\u001b[0;32m    985\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    986\u001b[0m         \u001b[38;5;66;03m# For labels that don't resolve as scalars like tuples and frozensets\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:1089\u001b[0m, in \u001b[0;36mSeries._get_value\u001b[1;34m(self, label, takeable)\u001b[0m\n\u001b[0;32m   1086\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[label]\n\u001b[0;32m   1088\u001b[0m \u001b[38;5;66;03m# Similar to Index.get_value, but we do not fall back to positional\u001b[39;00m\n\u001b[1;32m-> 1089\u001b[0m loc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlabel\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1090\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39m_get_values_for_loc(\u001b[38;5;28mself\u001b[39m, loc, label)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3804\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m-> 3804\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[0;32m   3805\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[0;32m   3806\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[0;32m   3807\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[0;32m   3808\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[0;32m   3809\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[1;31mKeyError\u001b[0m: 'New_Column'"
     ]
    }
   ],
   "source": [
    "# Create empty columns\n",
    "columns = ['ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID', 'SLO',\n",
    "           'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV', 'TPA', 'CFL', 'GNO',\n",
    "           'VLD', 'ASP', 'ASS', 'REF', 'U3', 'U5', 'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC',\n",
    "           'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO',\n",
    "           'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar',\n",
    "           'ClinVar_CLNSIG']\n",
    "\n",
    "for col in columns:\n",
    "    vcf[col] = ''\n",
    "\n",
    "# Populate columns based on 'info' values\n",
    "for i, row in vcf.iterrows():\n",
    "    info = row['New_Column']\n",
    "    items = info.split(';')\n",
    "    for item in items:\n",
    "        key_value = item.split('=')\n",
    "        key = key_value[0]\n",
    "        if key in columns:\n",
    "            if len(key_value) > 1:\n",
    "                value = key_value[1]\n",
    "                vcf.at[i, key] = f\"{key}={value}\"\n",
    "            else:\n",
    "                vcf.at[i, key] = key\n",
    "        else:\n",
    "            vcf.at[i, key] = 'null'\n",
    "\n",
    "# Print the resulting DataFrame\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3c1df33",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "2bbc704b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>HET_new</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:92:51:51:26:25:49.02%:5.8079E-10:41:51:10:...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>49.02%</td>\n",
       "      <td>5.8079E-10</td>\n",
       "      <td>41</td>\n",
       "      <td>51</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ADP=51</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=116</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=6682385</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.54678739806320081,0.45321260193679918</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.5178,0.4822</td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005140126000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14930</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15118</td>\n",
       "      <td>rs11580262</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:23:20:20:13:7:35%:4.158E-3:42:45:8:5:4:3</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>35%</td>\n",
       "      <td>4.158E-3</td>\n",
       "      <td>42</td>\n",
       "      <td>45</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ADP=20</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=120</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=11580262</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.60931128950050968,0.39068871049949031</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005040102000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=15118</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:107:34:34:9:25:73.53%:1.9819E-11:37:44:8:1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>107</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>73.53%</td>\n",
       "      <td>1.9819E-11</td>\n",
       "      <td>37</td>\n",
       "      <td>44</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ADP=34</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=105</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>G5A</td>\n",
       "      <td></td>\n",
       "      <td>RS=3210724</td>\n",
       "      <td>RV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.51763188073394495,0.48230440876656472...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005070002000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=16495</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16949</td>\n",
       "      <td>rs199745162</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:55:85:85:68:17:20%:3.1403E-6:52:39:55:13:11:6</td>\n",
       "      <td>0/1</td>\n",
       "      <td>55</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>68</td>\n",
       "      <td>17</td>\n",
       "      <td>20%</td>\n",
       "      <td>3.1403E-6</td>\n",
       "      <td>52</td>\n",
       "      <td>39</td>\n",
       "      <td>55</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>MIR6859-1,WASH7P</td>\n",
       "      <td>ADP=85</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=199745162</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.76095024209989806,0.23904975790010193</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.986,0.01398</td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000040005150026000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635|MIR6859-1:102466751</td>\n",
       "      <td></td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=16949</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24786</td>\n",
       "      <td>rs11266852</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:190:93:93:43:50:53.76%:9.1368E-20:63:55:39...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>190</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "      <td>43</td>\n",
       "      <td>50</td>\n",
       "      <td>53.76%</td>\n",
       "      <td>9.1368E-20</td>\n",
       "      <td>63</td>\n",
       "      <td>55</td>\n",
       "      <td>39</td>\n",
       "      <td>4</td>\n",
       "      <td>37</td>\n",
       "      <td>13</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ADP=93</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=120</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=11266852</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100000005000102000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=24786</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71451</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209899</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:40:21:21:10:11:52.38%:8.2399E-5:62:54:10:0...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>52.38%</td>\n",
       "      <td>8.2399E-5</td>\n",
       "      <td>62</td>\n",
       "      <td>54</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71452</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:33:16:16:7:9:56.25%:4.0786E-4:74:64:7:0:8:1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>56.25%</td>\n",
       "      <td>4.0786E-4</td>\n",
       "      <td>74</td>\n",
       "      <td>64</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ADP=16</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71453</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210723</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:29:37:37:28:9:24.32%:1.1256E-3:68:68:28:0:9:0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>28</td>\n",
       "      <td>9</td>\n",
       "      <td>24.32%</td>\n",
       "      <td>1.1256E-3</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>ADP=37</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71454</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210771</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:30:30:30:21:9:30%:9.678E-4:59:65:17:4:8:1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>30%</td>\n",
       "      <td>9.678E-4</td>\n",
       "      <td>59</td>\n",
       "      <td>65</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ADP=30</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71455</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211636</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:59:19:19:5:14:73.68%:1.2025E-6:66:52:4:1:10:4</td>\n",
       "      <td>0/1</td>\n",
       "      <td>59</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>73.68%</td>\n",
       "      <td>1.2025E-6</td>\n",
       "      <td>66</td>\n",
       "      <td>52</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td>ADP=19</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71456 rows × 81 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14930    rs6682385       G    .   PASS   \n",
       "1      chr1     15118   rs11580262       G    .   PASS   \n",
       "2      chr1     16495    rs3210724       C    .   PASS   \n",
       "3      chr1     16949  rs199745162       C    .   PASS   \n",
       "4      chr1     24786   rs11266852       A    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "71451  chrY  57209899            .       T    .   PASS   \n",
       "71452  chrY  57210016            .       T    .   PASS   \n",
       "71453  chrY  57210723            .       A    .   PASS   \n",
       "71454  chrY  57210771            .       C    .   PASS   \n",
       "71455  chrY  57211636            .       A    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....   \n",
       "1      ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...   \n",
       "2      ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "3      ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...   \n",
       "4      ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...   \n",
       "...                                                  ...   \n",
       "71451  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...   \n",
       "71452  ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "71453  ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...   \n",
       "71454  ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...   \n",
       "71455  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "71451  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71452  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71453  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71454  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "71455  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   GT   GQ SDP  DP  \\\n",
       "0      0/1:92:51:51:26:25:49.02%:5.8079E-10:41:51:10:...  0/1   92  51  51   \n",
       "1           0/1:23:20:20:13:7:35%:4.158E-3:42:45:8:5:4:3  0/1   23  20  20   \n",
       "2      0/1:107:34:34:9:25:73.53%:1.9819E-11:37:44:8:1...  0/1  107  34  34   \n",
       "3      0/1:55:85:85:68:17:20%:3.1403E-6:52:39:55:13:11:6  0/1   55  85  85   \n",
       "4      0/1:190:93:93:43:50:53.76%:9.1368E-20:63:55:39...  0/1  190  93  93   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "71451  0/1:40:21:21:10:11:52.38%:8.2399E-5:62:54:10:0...  0/1   40  21  21   \n",
       "71452    0/1:33:16:16:7:9:56.25%:4.0786E-4:74:64:7:0:8:1  0/1   33  16  16   \n",
       "71453  0/1:29:37:37:28:9:24.32%:1.1256E-3:68:68:28:0:9:0  0/1   29  37  37   \n",
       "71454      0/1:30:30:30:21:9:30%:9.678E-4:59:65:17:4:8:1  0/1   30  30  30   \n",
       "71455  0/1:59:19:19:5:14:73.68%:1.2025E-6:66:52:4:1:10:4  0/1   59  19  19   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RBQ ABQ RDF RDR ADF ADR              Gene  \\\n",
       "0      26  25  49.02%  5.8079E-10  41  51  10  16  17   8            WASH7P   \n",
       "1      13   7     35%    4.158E-3  42  45   8   5   4   3            WASH7P   \n",
       "2       9  25  73.53%  1.9819E-11  37  44   8   1  24   1            WASH7P   \n",
       "3      68  17     20%   3.1403E-6  52  39  55  13  11   6  MIR6859-1,WASH7P   \n",
       "4      43  50  53.76%  9.1368E-20  63  55  39   4  37  13            WASH7P   \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..  ..  ..               ...   \n",
       "71451  10  11  52.38%   8.2399E-5  62  54  10   0   9   2                     \n",
       "71452   7   9  56.25%   4.0786E-4  74  64   7   0   8   1                     \n",
       "71453  28   9  24.32%   1.1256E-3  68  68  28   0   9   0                     \n",
       "71454  21   9     30%    9.678E-4  59  65  17   4   8   1                     \n",
       "71455   5  14  73.68%   1.2025E-6  66  52   4   1  10   4                     \n",
       "\n",
       "          ADP    WT    HET    HOM    NC CDA OTH S3D WTD      dbSNPBuildID  \\\n",
       "0      ADP=51  WT=0  HET=1  HOM=0  NC=0                  dbSNPBuildID=116   \n",
       "1      ADP=20  WT=0  HET=1  HOM=0  NC=0                  dbSNPBuildID=120   \n",
       "2      ADP=34  WT=0  HET=1  HOM=0  NC=0                  dbSNPBuildID=105   \n",
       "3      ADP=85  WT=0  HET=1  HOM=0  NC=0                  dbSNPBuildID=137   \n",
       "4      ADP=93  WT=0  HET=1  HOM=0  NC=0                  dbSNPBuildID=120   \n",
       "...       ...   ...    ...    ...   ...  ..  ..  ..  ..               ...   \n",
       "71451  ADP=21  WT=0  HET=1  HOM=0  NC=0                                     \n",
       "71452  ADP=16  WT=0  HET=1  HOM=0  NC=0                                     \n",
       "71453  ADP=37  WT=0  HET=1  HOM=0  NC=0                                     \n",
       "71454  ADP=30  WT=0  HET=1  HOM=0  NC=0                                     \n",
       "71455  ADP=19  WT=0  HET=1  HOM=0  NC=0                                     \n",
       "\n",
       "       SLO NSF  R3 R5 NSN NSM  G5A    COMMON            RS  RV TPA CFL  GNO  \\\n",
       "0      SLO                          COMMON=1    RS=6682385              GNO   \n",
       "1      SLO                                     RS=11580262              GNO   \n",
       "2      SLO                     G5A              RS=3210724  RV                \n",
       "3               R3                  COMMON=1  RS=199745162                    \n",
       "4      SLO                                     RS=11266852              GNO   \n",
       "...    ...  ..  .. ..  ..  ..  ...       ...           ...  ..  ..  ..  ...   \n",
       "71451                                                                         \n",
       "71452                                                                         \n",
       "71453                                                                         \n",
       "71454                                                                         \n",
       "71455                                                                         \n",
       "\n",
       "       VLD  ASP ASS U3 U5                                             TOPMED  \\\n",
       "0      VLD  ASP               TOPMED=0.54678739806320081,0.45321260193679918   \n",
       "1      VLD  ASP               TOPMED=0.60931128950050968,0.39068871049949031   \n",
       "2      VLD  ASP            TOPMED=0.51763188073394495,0.48230440876656472...   \n",
       "3      VLD  ASP               TOPMED=0.76095024209989806,0.23904975790010193   \n",
       "4           ASP                                                                \n",
       "...    ...  ...  .. .. ..                                                ...   \n",
       "71451                                                                          \n",
       "71452                                                                          \n",
       "71453                                                                          \n",
       "71454                                                                          \n",
       "71455                                                                          \n",
       "\n",
       "         WGT MTP LSD NOC DSS SYN  KGPhase3                CAF      VC MUT  \\\n",
       "0      WGT=1                      KGPhase3  CAF=0.5178,0.4822  VC=SNV       \n",
       "1      WGT=1                                                   VC=SNV       \n",
       "2      WGT=1                                                   VC=SNV       \n",
       "3      WGT=1                      KGPhase3  CAF=0.986,0.01398  VC=SNV       \n",
       "4      WGT=1                                                   VC=SNV       \n",
       "...      ...  ..  ..  ..  ..  ..       ...                ...     ...  ..   \n",
       "71451                                                                       \n",
       "71452                                                                       \n",
       "71453                                                                       \n",
       "71454                                                                       \n",
       "71455                                                                       \n",
       "\n",
       "      KGPhase1 NOV                             VP    SAO  \\\n",
       "0                   VP=0x050100080005140126000100  SAO=0   \n",
       "1                   VP=0x050100080005040102000100  SAO=0   \n",
       "2                   VP=0x050100080005070002000100  SAO=0   \n",
       "3                   VP=0x050000040005150026000100  SAO=0   \n",
       "4                   VP=0x050100000005000102000100  SAO=0   \n",
       "...        ...  ..                            ...    ...   \n",
       "71451                                                      \n",
       "71452                                                      \n",
       "71453                                                      \n",
       "71454                                                      \n",
       "71455                                                      \n",
       "\n",
       "                                         GENEINFO  INT  G5 OM PMC    SSR  \\\n",
       "0                          GENEINFO=WASH7P:653635  INT             SSR=0   \n",
       "1                          GENEINFO=WASH7P:653635  INT             SSR=0   \n",
       "2                          GENEINFO=WASH7P:653635  INT  G5         SSR=0   \n",
       "3      GENEINFO=WASH7P:653635|MIR6859-1:102466751       G5         SSR=0   \n",
       "4                          GENEINFO=WASH7P:653635                  SSR=0   \n",
       "...                                           ...  ...  .. ..  ..    ...   \n",
       "71451                                                                      \n",
       "71452                                                                      \n",
       "71453                                                                      \n",
       "71454                                                                      \n",
       "71455                                                                      \n",
       "\n",
       "             RSPOS HD PM ClinVar ClinVar_CLNSIG  HET_new      Zygosity  \n",
       "0      RSPOS=14930                                     1  Heterozygous  \n",
       "1      RSPOS=15118                                     1  Heterozygous  \n",
       "2      RSPOS=16495                                     1  Heterozygous  \n",
       "3      RSPOS=16949                                     1  Heterozygous  \n",
       "4      RSPOS=24786                                     1  Heterozygous  \n",
       "...            ... .. ..     ...            ...      ...           ...  \n",
       "71451                                                  1  Heterozygous  \n",
       "71452                                                  1  Heterozygous  \n",
       "71453                                                  1  Heterozygous  \n",
       "71454                                                  1  Heterozygous  \n",
       "71455                                                  1  Heterozygous  \n",
       "\n",
       "[71456 rows x 81 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Read the VCF file into a Pandas DataFrame (skip first 80 rows as shown in a previous answer)\n",
    "vcf = pd.read_csv(\n",
    "    r'C:/Users/GenepoweRx_Madhu/Downloads/718_final.vcf',\n",
    "    comment='#',\n",
    "    sep='\\t',\n",
    "    header=None,\n",
    "    low_memory=False,\n",
    "    skiprows=80\n",
    ")\n",
    "\n",
    "# Assign appropriate column names\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "\n",
    "vcf[\"Gene\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene'] = vcf['Gene'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "\n",
    "# Create empty columns\n",
    "columns = ['ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID', 'SLO',\n",
    "           'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV', 'TPA', 'CFL', 'GNO',\n",
    "           'VLD', 'ASP', 'ASS', 'REF', 'U3', 'U5', 'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC',\n",
    "           'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO',\n",
    "           'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar',\n",
    "           'ClinVar_CLNSIG']\n",
    "\n",
    "for col in columns:\n",
    "    vcf[col] = ''\n",
    "\n",
    "# Populate columns based on 'INFO' values\n",
    "for i, row in vcf.iterrows():\n",
    "    info = row['INFO']\n",
    "    items = info.split(';')\n",
    "    for item in items:\n",
    "        key_value = item.split('=')\n",
    "        key = key_value[0]\n",
    "        if key in columns:\n",
    "            if len(key_value) > 1:\n",
    "                value = key_value[1]\n",
    "                vcf.at[i, key] = f\"{key}={value}\"\n",
    "            else:\n",
    "                vcf.at[i, key] = key\n",
    "\n",
    "# Extract the numeric part of the HET column and convert it to an integer\n",
    "vcf['HET_new'] = vcf['HET'].str.extract('(\\d+)').astype(int)\n",
    "\n",
    "# Create the Zygosity column based on the HET column\n",
    "vcf['Zygosity'] = vcf['HET_new'].apply(lambda x: 'Heterozygous' if x == 1 else 'Homozygous')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "e4339971",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>HET_new</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>49.02%</td>\n",
       "      <td>5.8079E-10</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>ADP=51</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=116</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=6682385</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.54678739806320081,0.45321260193679918</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.5178,0.4822</td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005140126000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14930</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15118</td>\n",
       "      <td>rs11580262</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>35%</td>\n",
       "      <td>4.158E-3</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>ADP=20</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=120</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=11580262</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.60931128950050968,0.39068871049949031</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005040102000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=15118</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>107</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>73.53%</td>\n",
       "      <td>1.9819E-11</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=34</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=105</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>G5A</td>\n",
       "      <td></td>\n",
       "      <td>RS=3210724</td>\n",
       "      <td>RV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.51763188073394495,0.48230440876656472...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005070002000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=16495</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16949</td>\n",
       "      <td>rs199745162</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>55</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "      <td>68</td>\n",
       "      <td>17</td>\n",
       "      <td>20%</td>\n",
       "      <td>3.1403E-6</td>\n",
       "      <td>55</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>ADP=85</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=199745162</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.76095024209989806,0.23904975790010193</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.986,0.01398</td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000040005150026000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635|MIR6859-1:102466751</td>\n",
       "      <td></td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=16949</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24786</td>\n",
       "      <td>rs11266852</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>190</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "      <td>43</td>\n",
       "      <td>50</td>\n",
       "      <td>53.76%</td>\n",
       "      <td>9.1368E-20</td>\n",
       "      <td>39</td>\n",
       "      <td>4</td>\n",
       "      <td>37</td>\n",
       "      <td>13</td>\n",
       "      <td>ADP=93</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=120</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=11266852</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100000005000102000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=24786</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71451</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209899</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>52.38%</td>\n",
       "      <td>8.2399E-5</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71452</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>56.25%</td>\n",
       "      <td>4.0786E-4</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=16</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71453</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210723</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>28</td>\n",
       "      <td>9</td>\n",
       "      <td>24.32%</td>\n",
       "      <td>1.1256E-3</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=37</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71454</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210771</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>30%</td>\n",
       "      <td>9.678E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=30</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71455</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211636</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>59</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>73.68%</td>\n",
       "      <td>1.2025E-6</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>ADP=19</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71456 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14930    rs6682385       G    .   PASS   \n",
       "1      chr1     15118   rs11580262       G    .   PASS   \n",
       "2      chr1     16495    rs3210724       C    .   PASS   \n",
       "3      chr1     16949  rs199745162       C    .   PASS   \n",
       "4      chr1     24786   rs11266852       A    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "71451  chrY  57209899            .       T    .   PASS   \n",
       "71452  chrY  57210016            .       T    .   PASS   \n",
       "71453  chrY  57210723            .       A    .   PASS   \n",
       "71454  chrY  57210771            .       C    .   PASS   \n",
       "71455  chrY  57211636            .       A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ SDP  DP  \\\n",
       "0      ADP=51;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5178,0....  0/1   92  51  51   \n",
       "1      ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...  0/1   23  20  20   \n",
       "2      ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...  0/1  107  34  34   \n",
       "3      ADP=85;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.986,0.0...  0/1   55  85  85   \n",
       "4      ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...  0/1  190  93  93   \n",
       "...                                                  ...  ...  ...  ..  ..   \n",
       "71451  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|non_coding_...  0/1   40  21  21   \n",
       "71452  ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...  0/1   33  16  16   \n",
       "71453  ADP=37;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...  0/1   29  37  37   \n",
       "71454  ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1   30  30  30   \n",
       "71455  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...  0/1   59  19  19   \n",
       "\n",
       "       RD  AD    FREQ        PVAL RDF RDR ADF ADR     ADP    WT    HET    HOM  \\\n",
       "0      26  25  49.02%  5.8079E-10  10  16  17   8  ADP=51  WT=0  HET=1  HOM=0   \n",
       "1      13   7     35%    4.158E-3   8   5   4   3  ADP=20  WT=0  HET=1  HOM=0   \n",
       "2       9  25  73.53%  1.9819E-11   8   1  24   1  ADP=34  WT=0  HET=1  HOM=0   \n",
       "3      68  17     20%   3.1403E-6  55  13  11   6  ADP=85  WT=0  HET=1  HOM=0   \n",
       "4      43  50  53.76%  9.1368E-20  39   4  37  13  ADP=93  WT=0  HET=1  HOM=0   \n",
       "...    ..  ..     ...         ...  ..  ..  ..  ..     ...   ...    ...    ...   \n",
       "71451  10  11  52.38%   8.2399E-5  10   0   9   2  ADP=21  WT=0  HET=1  HOM=0   \n",
       "71452   7   9  56.25%   4.0786E-4   7   0   8   1  ADP=16  WT=0  HET=1  HOM=0   \n",
       "71453  28   9  24.32%   1.1256E-3  28   0   9   0  ADP=37  WT=0  HET=1  HOM=0   \n",
       "71454  21   9     30%    9.678E-4  17   4   8   1  ADP=30  WT=0  HET=1  HOM=0   \n",
       "71455   5  14  73.68%   1.2025E-6   4   1  10   4  ADP=19  WT=0  HET=1  HOM=0   \n",
       "\n",
       "         NC CDA OTH S3D WTD      dbSNPBuildID  SLO NSF  R3 R5 NSN NSM  G5A  \\\n",
       "0      NC=0                  dbSNPBuildID=116  SLO                           \n",
       "1      NC=0                  dbSNPBuildID=120  SLO                           \n",
       "2      NC=0                  dbSNPBuildID=105  SLO                     G5A   \n",
       "3      NC=0                  dbSNPBuildID=137           R3                   \n",
       "4      NC=0                  dbSNPBuildID=120  SLO                           \n",
       "...     ...  ..  ..  ..  ..               ...  ...  ..  .. ..  ..  ..  ...   \n",
       "71451  NC=0                                                                  \n",
       "71452  NC=0                                                                  \n",
       "71453  NC=0                                                                  \n",
       "71454  NC=0                                                                  \n",
       "71455  NC=0                                                                  \n",
       "\n",
       "         COMMON            RS  RV TPA CFL  GNO  VLD  ASP ASS U3 U5  \\\n",
       "0      COMMON=1    RS=6682385              GNO  VLD  ASP             \n",
       "1                 RS=11580262              GNO  VLD  ASP             \n",
       "2                  RS=3210724  RV               VLD  ASP             \n",
       "3      COMMON=1  RS=199745162                   VLD  ASP             \n",
       "4                 RS=11266852              GNO       ASP             \n",
       "...         ...           ...  ..  ..  ..  ...  ...  ...  .. .. ..   \n",
       "71451                                                                \n",
       "71452                                                                \n",
       "71453                                                                \n",
       "71454                                                                \n",
       "71455                                                                \n",
       "\n",
       "                                                  TOPMED    WGT MTP LSD NOC  \\\n",
       "0         TOPMED=0.54678739806320081,0.45321260193679918  WGT=1               \n",
       "1         TOPMED=0.60931128950050968,0.39068871049949031  WGT=1               \n",
       "2      TOPMED=0.51763188073394495,0.48230440876656472...  WGT=1               \n",
       "3         TOPMED=0.76095024209989806,0.23904975790010193  WGT=1               \n",
       "4                                                         WGT=1               \n",
       "...                                                  ...    ...  ..  ..  ..   \n",
       "71451                                                                         \n",
       "71452                                                                         \n",
       "71453                                                                         \n",
       "71454                                                                         \n",
       "71455                                                                         \n",
       "\n",
       "      DSS SYN  KGPhase3                CAF      VC MUT KGPhase1 NOV  \\\n",
       "0              KGPhase3  CAF=0.5178,0.4822  VC=SNV                    \n",
       "1                                           VC=SNV                    \n",
       "2                                           VC=SNV                    \n",
       "3              KGPhase3  CAF=0.986,0.01398  VC=SNV                    \n",
       "4                                           VC=SNV                    \n",
       "...    ..  ..       ...                ...     ...  ..      ...  ..   \n",
       "71451                                                                 \n",
       "71452                                                                 \n",
       "71453                                                                 \n",
       "71454                                                                 \n",
       "71455                                                                 \n",
       "\n",
       "                                  VP    SAO  \\\n",
       "0      VP=0x050100080005140126000100  SAO=0   \n",
       "1      VP=0x050100080005040102000100  SAO=0   \n",
       "2      VP=0x050100080005070002000100  SAO=0   \n",
       "3      VP=0x050000040005150026000100  SAO=0   \n",
       "4      VP=0x050100000005000102000100  SAO=0   \n",
       "...                              ...    ...   \n",
       "71451                                         \n",
       "71452                                         \n",
       "71453                                         \n",
       "71454                                         \n",
       "71455                                         \n",
       "\n",
       "                                         GENEINFO  INT  G5 OM PMC    SSR  \\\n",
       "0                          GENEINFO=WASH7P:653635  INT             SSR=0   \n",
       "1                          GENEINFO=WASH7P:653635  INT             SSR=0   \n",
       "2                          GENEINFO=WASH7P:653635  INT  G5         SSR=0   \n",
       "3      GENEINFO=WASH7P:653635|MIR6859-1:102466751       G5         SSR=0   \n",
       "4                          GENEINFO=WASH7P:653635                  SSR=0   \n",
       "...                                           ...  ...  .. ..  ..    ...   \n",
       "71451                                                                      \n",
       "71452                                                                      \n",
       "71453                                                                      \n",
       "71454                                                                      \n",
       "71455                                                                      \n",
       "\n",
       "             RSPOS HD PM ClinVar ClinVar_CLNSIG  HET_new      Zygosity  \n",
       "0      RSPOS=14930                                     1  Heterozygous  \n",
       "1      RSPOS=15118                                     1  Heterozygous  \n",
       "2      RSPOS=16495                                     1  Heterozygous  \n",
       "3      RSPOS=16949                                     1  Heterozygous  \n",
       "4      RSPOS=24786                                     1  Heterozygous  \n",
       "...            ... .. ..     ...            ...      ...           ...  \n",
       "71451                                                  1  Heterozygous  \n",
       "71452                                                  1  Heterozygous  \n",
       "71453                                                  1  Heterozygous  \n",
       "71454                                                  1  Heterozygous  \n",
       "71455                                                  1  Heterozygous  \n",
       "\n",
       "[71456 rows x 76 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract the numeric part of the HET column and convert it to an integer\n",
    "vcf['HET_new'] = vcf['HET'].str.extract('(\\d+)').astype(int)\n",
    "\n",
    "# Create the Zygosity column based on the HET column\n",
    "vcf['Zygosity'] = vcf['HET_new'].apply(lambda x: 'Heterozygous' if x == 1 else 'Homozygous')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "867b1c9d",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs991811</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99589702</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129625</td>\n",
       "      <td>ENST00000265517.10</td>\n",
       "      <td>c.453T&gt;C</td>\n",
       "      <td>ENSP00000265517.5</td>\n",
       "      <td>p.Gly151%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.812300e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5443</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>0.3084</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3141</td>\n",
       "      <td>0.4407</td>\n",
       "      <td>0.3950</td>\n",
       "      <td>0.7687</td>\n",
       "      <td>0.2648</td>\n",
       "      <td>0.5787</td>\n",
       "      <td>0.6482</td>\n",
       "      <td>0.2079</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3900</td>\n",
       "      <td>0.4184</td>\n",
       "      <td>0.4712</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>0.4693</td>\n",
       "      <td>0.3664</td>\n",
       "      <td>0.5739</td>\n",
       "      <td>0.6295</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3510</td>\n",
       "      <td>0.4588</td>\n",
       "      <td>0.4163</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G151</td>\n",
       "      <td>ggT/ggC</td>\n",
       "      <td>1</td>\n",
       "      <td>27170061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs991811</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99589702</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129625</td>\n",
       "      <td>ENST00000457717.6</td>\n",
       "      <td>c.453T&gt;C</td>\n",
       "      <td>ENSP00000400821.1</td>\n",
       "      <td>p.Gly151%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.812300e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5443</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>0.3084</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3141</td>\n",
       "      <td>0.4407</td>\n",
       "      <td>0.3950</td>\n",
       "      <td>0.7687</td>\n",
       "      <td>0.2648</td>\n",
       "      <td>0.5787</td>\n",
       "      <td>0.6482</td>\n",
       "      <td>0.2079</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3900</td>\n",
       "      <td>0.4184</td>\n",
       "      <td>0.4712</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>0.4693</td>\n",
       "      <td>0.3664</td>\n",
       "      <td>0.5739</td>\n",
       "      <td>0.6295</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3510</td>\n",
       "      <td>0.4588</td>\n",
       "      <td>0.4163</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G151</td>\n",
       "      <td>ggT/ggC</td>\n",
       "      <td>1</td>\n",
       "      <td>27170061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs991811</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99589702</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129625</td>\n",
       "      <td>ENST00000511045.6</td>\n",
       "      <td>c.204T&gt;C</td>\n",
       "      <td>ENSP00000427679.2</td>\n",
       "      <td>p.Gly68%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.812300e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5443</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>0.3084</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3141</td>\n",
       "      <td>0.4407</td>\n",
       "      <td>0.3950</td>\n",
       "      <td>0.7687</td>\n",
       "      <td>0.2648</td>\n",
       "      <td>0.5787</td>\n",
       "      <td>0.6482</td>\n",
       "      <td>0.2079</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3900</td>\n",
       "      <td>0.4184</td>\n",
       "      <td>0.4712</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>0.4693</td>\n",
       "      <td>0.3664</td>\n",
       "      <td>0.5739</td>\n",
       "      <td>0.6295</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3510</td>\n",
       "      <td>0.4588</td>\n",
       "      <td>0.4163</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G68</td>\n",
       "      <td>ggT/ggC</td>\n",
       "      <td>1</td>\n",
       "      <td>27170061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs2306985</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99594865</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129628</td>\n",
       "      <td>ENST00000265517.10</td>\n",
       "      <td>c.891C&gt;G</td>\n",
       "      <td>ENSP00000265517.5</td>\n",
       "      <td>p.His297Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>100</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>64.1%</td>\n",
       "      <td>8.832600e-11</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated(0.54)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5423</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>0.3069</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3151</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.3944</td>\n",
       "      <td>0.7585</td>\n",
       "      <td>0.2649</td>\n",
       "      <td>0.5788</td>\n",
       "      <td>0.6480</td>\n",
       "      <td>0.2082</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3901</td>\n",
       "      <td>0.4186</td>\n",
       "      <td>0.4684</td>\n",
       "      <td>0.7444</td>\n",
       "      <td>0.4702</td>\n",
       "      <td>0.3636</td>\n",
       "      <td>0.5747</td>\n",
       "      <td>0.6325</td>\n",
       "      <td>0.2057</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3508</td>\n",
       "      <td>0.4578</td>\n",
       "      <td>0.4164</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>H297Q</td>\n",
       "      <td>caC/caG</td>\n",
       "      <td>1</td>\n",
       "      <td>19878569,24496996,27170061,31321587,31397093,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs2306985</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99594865</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129628</td>\n",
       "      <td>ENST00000457717.6</td>\n",
       "      <td>c.891C&gt;G</td>\n",
       "      <td>ENSP00000400821.1</td>\n",
       "      <td>p.His297Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>100</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>64.1%</td>\n",
       "      <td>8.832600e-11</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated(0.54)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5423</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>0.3069</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3151</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.3944</td>\n",
       "      <td>0.7585</td>\n",
       "      <td>0.2649</td>\n",
       "      <td>0.5788</td>\n",
       "      <td>0.6480</td>\n",
       "      <td>0.2082</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3901</td>\n",
       "      <td>0.4186</td>\n",
       "      <td>0.4684</td>\n",
       "      <td>0.7444</td>\n",
       "      <td>0.4702</td>\n",
       "      <td>0.3636</td>\n",
       "      <td>0.5747</td>\n",
       "      <td>0.6325</td>\n",
       "      <td>0.2057</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3508</td>\n",
       "      <td>0.4578</td>\n",
       "      <td>0.4164</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>H297Q</td>\n",
       "      <td>caC/caG</td>\n",
       "      <td>1</td>\n",
       "      <td>19878569,24496996,27170061,31321587,31397093,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs2306985</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99594865</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129628</td>\n",
       "      <td>ENST00000508578.1</td>\n",
       "      <td>n.293G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>100</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>64.1%</td>\n",
       "      <td>8.832600e-11</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5423</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>0.3069</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3151</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.3944</td>\n",
       "      <td>0.7585</td>\n",
       "      <td>0.2649</td>\n",
       "      <td>0.5788</td>\n",
       "      <td>0.6480</td>\n",
       "      <td>0.2082</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3901</td>\n",
       "      <td>0.4186</td>\n",
       "      <td>0.4684</td>\n",
       "      <td>0.7444</td>\n",
       "      <td>0.4702</td>\n",
       "      <td>0.3636</td>\n",
       "      <td>0.5747</td>\n",
       "      <td>0.6325</td>\n",
       "      <td>0.2057</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3508</td>\n",
       "      <td>0.4578</td>\n",
       "      <td>0.4164</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>AFR</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>4/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>19878569,24496996,27170061,31321587,31397093,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs2306985</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99594865</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129628</td>\n",
       "      <td>ENST00000511045.6</td>\n",
       "      <td>c.642C&gt;G</td>\n",
       "      <td>ENSP00000427679.2</td>\n",
       "      <td>p.His214Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>100</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>64.1%</td>\n",
       "      <td>8.832600e-11</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated(0.54)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5423</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>0.3069</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3151</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.3944</td>\n",
       "      <td>0.7585</td>\n",
       "      <td>0.2649</td>\n",
       "      <td>0.5788</td>\n",
       "      <td>0.6480</td>\n",
       "      <td>0.2082</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3901</td>\n",
       "      <td>0.4186</td>\n",
       "      <td>0.4684</td>\n",
       "      <td>0.7444</td>\n",
       "      <td>0.4702</td>\n",
       "      <td>0.3636</td>\n",
       "      <td>0.5747</td>\n",
       "      <td>0.6325</td>\n",
       "      <td>0.2057</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3508</td>\n",
       "      <td>0.4578</td>\n",
       "      <td>0.4164</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>H214Q</td>\n",
       "      <td>caC/caG</td>\n",
       "      <td>1</td>\n",
       "      <td>19878569,24496996,27170061,31321587,31397093,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs1061271</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99623894</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>347062</td>\n",
       "      <td>ENST00000265517.10</td>\n",
       "      <td>c.*1046C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>50%</td>\n",
       "      <td>2.928300e-09</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6877</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>0.3905</td>\n",
       "      <td>0.7966</td>\n",
       "      <td>0.4384</td>\n",
       "      <td>0.6646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5941</td>\n",
       "      <td>0.8924</td>\n",
       "      <td>0.6077</td>\n",
       "      <td>0.4730</td>\n",
       "      <td>0.7056</td>\n",
       "      <td>0.7732</td>\n",
       "      <td>0.3049</td>\n",
       "      <td>0.6456</td>\n",
       "      <td>0.4623</td>\n",
       "      <td>0.5984</td>\n",
       "      <td>0.6239</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>18/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs1061271</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99623894</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>347062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>50%</td>\n",
       "      <td>2.928300e-09</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6877</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>0.3905</td>\n",
       "      <td>0.7966</td>\n",
       "      <td>0.4384</td>\n",
       "      <td>0.6646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5941</td>\n",
       "      <td>0.8924</td>\n",
       "      <td>0.6077</td>\n",
       "      <td>0.4730</td>\n",
       "      <td>0.7056</td>\n",
       "      <td>0.7732</td>\n",
       "      <td>0.3049</td>\n",
       "      <td>0.6456</td>\n",
       "      <td>0.4623</td>\n",
       "      <td>0.5984</td>\n",
       "      <td>0.6239</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs1061271</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99623894</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>347062</td>\n",
       "      <td>ENST00000508578.1</td>\n",
       "      <td>n.32+1988G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>50%</td>\n",
       "      <td>2.928300e-09</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6877</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>0.3905</td>\n",
       "      <td>0.7966</td>\n",
       "      <td>0.4384</td>\n",
       "      <td>0.6646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5941</td>\n",
       "      <td>0.8924</td>\n",
       "      <td>0.6077</td>\n",
       "      <td>0.4730</td>\n",
       "      <td>0.7056</td>\n",
       "      <td>0.7732</td>\n",
       "      <td>0.3049</td>\n",
       "      <td>0.6456</td>\n",
       "      <td>0.4623</td>\n",
       "      <td>0.5984</td>\n",
       "      <td>0.6239</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>AFR</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs1061271</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99623894</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>347062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>50%</td>\n",
       "      <td>2.928300e-09</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6877</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>0.3905</td>\n",
       "      <td>0.7966</td>\n",
       "      <td>0.4384</td>\n",
       "      <td>0.6646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5941</td>\n",
       "      <td>0.8924</td>\n",
       "      <td>0.6077</td>\n",
       "      <td>0.4730</td>\n",
       "      <td>0.7056</td>\n",
       "      <td>0.7732</td>\n",
       "      <td>0.3049</td>\n",
       "      <td>0.6456</td>\n",
       "      <td>0.4623</td>\n",
       "      <td>0.5984</td>\n",
       "      <td>0.6239</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name  Gene  Gene_Score  \\\n",
       "0       MTTP  MTTP           8   \n",
       "1       MTTP  MTTP           8   \n",
       "2       MTTP  MTTP           8   \n",
       "3       MTTP  MTTP           8   \n",
       "4       MTTP  MTTP           8   \n",
       "5       MTTP  MTTP           8   \n",
       "6       MTTP  MTTP           8   \n",
       "7       MTTP  MTTP           8   \n",
       "8       MTTP  MTTP           8   \n",
       "9       MTTP  MTTP           8   \n",
       "10      MTTP  MTTP           8   \n",
       "\n",
       "                                            Condition  \\\n",
       "0   Polygenic (Familial combined, small dense LDL,...   \n",
       "1   Polygenic (Familial combined, small dense LDL,...   \n",
       "2   Polygenic (Familial combined, small dense LDL,...   \n",
       "3   Polygenic (Familial combined, small dense LDL,...   \n",
       "4   Polygenic (Familial combined, small dense LDL,...   \n",
       "5   Polygenic (Familial combined, small dense LDL,...   \n",
       "6   Polygenic (Familial combined, small dense LDL,...   \n",
       "7   Polygenic (Familial combined, small dense LDL,...   \n",
       "8   Polygenic (Familial combined, small dense LDL,...   \n",
       "9   Polygenic (Familial combined, small dense LDL,...   \n",
       "10  Polygenic (Familial combined, small dense LDL,...   \n",
       "\n",
       "                             Headings                  21_Conditions_list  \\\n",
       "0   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "1   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "2   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "3   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "4   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "5   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "6   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "7   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "8   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "9   Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "10  Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "\n",
       "         rsID Literature CHROM       POS REF ALT      Zygosity  \\\n",
       "0    rs991811         No  chr4  99589702   T   C  Heterozygous   \n",
       "1    rs991811         No  chr4  99589702   T   C  Heterozygous   \n",
       "2    rs991811         No  chr4  99589702   T   C  Heterozygous   \n",
       "3   rs2306985        Yes  chr4  99594865   C   G  Heterozygous   \n",
       "4   rs2306985        Yes  chr4  99594865   C   G  Heterozygous   \n",
       "5   rs2306985        Yes  chr4  99594865   C   G  Heterozygous   \n",
       "6   rs2306985        Yes  chr4  99594865   C   G  Heterozygous   \n",
       "7   rs1061271         No  chr4  99623894   C   G  Heterozygous   \n",
       "8   rs1061271         No  chr4  99623894   C   G  Heterozygous   \n",
       "9   rs1061271         No  chr4  99623894   C   G  Heterozygous   \n",
       "10  rs1061271         No  chr4  99623894   C   G  Heterozygous   \n",
       "\n",
       "                                     Consequence Consequence_score    IMPACT  \\\n",
       "0                             synonymous variant              3/10       LOW   \n",
       "1                             synonymous variant              3/10       LOW   \n",
       "2                             synonymous variant              3/10       LOW   \n",
       "3                               missense variant              7/10  MODERATE   \n",
       "4                               missense variant              7/10  MODERATE   \n",
       "5             non coding transcript exon variant              2/10  MODIFIER   \n",
       "6                               missense variant              7/10  MODERATE   \n",
       "7                            3 prime UTR variant              3/10  MODIFIER   \n",
       "8                        downstream gene variant              2/10  MODIFIER   \n",
       "9   intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "10                       downstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "    IMPACT_score          ClinVar_CLNDN              CLIN_SIG  \\\n",
       "0            2.5  Abetalipoproteinaemia                benign   \n",
       "1            2.5  Abetalipoproteinaemia                benign   \n",
       "2            2.5  Abetalipoproteinaemia                benign   \n",
       "3            5.0  Abetalipoproteinaemia  benign,likely benign   \n",
       "4            5.0  Abetalipoproteinaemia  benign,likely benign   \n",
       "5            1.5  Abetalipoproteinaemia  benign,likely benign   \n",
       "6            5.0  Abetalipoproteinaemia  benign,likely benign   \n",
       "7            1.5  Abetalipoproteinaemia                benign   \n",
       "8            1.5  Abetalipoproteinaemia                benign   \n",
       "9            1.5  Abetalipoproteinaemia                benign   \n",
       "10           1.5  Abetalipoproteinaemia                benign   \n",
       "\n",
       "                                   ClinVar_CLNREVSTAT  ClinVar  \\\n",
       "0   criteria provided, multiple submitters, no con...   129625   \n",
       "1   criteria provided, multiple submitters, no con...   129625   \n",
       "2   criteria provided, multiple submitters, no con...   129625   \n",
       "3   criteria provided, multiple submitters, no con...   129628   \n",
       "4   criteria provided, multiple submitters, no con...   129628   \n",
       "5   criteria provided, multiple submitters, no con...   129628   \n",
       "6   criteria provided, multiple submitters, no con...   129628   \n",
       "7                 criteria provided, single submitter   347062   \n",
       "8                 criteria provided, single submitter   347062   \n",
       "9                 criteria provided, single submitter   347062   \n",
       "10                criteria provided, single submitter   347062   \n",
       "\n",
       "                 HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0   ENST00000265517.10           c.453T>C  ENSP00000265517.5   \n",
       "1    ENST00000457717.6           c.453T>C  ENSP00000400821.1   \n",
       "2    ENST00000511045.6           c.204T>C  ENSP00000427679.2   \n",
       "3   ENST00000265517.10           c.891C>G  ENSP00000265517.5   \n",
       "4    ENST00000457717.6           c.891C>G  ENSP00000400821.1   \n",
       "5    ENST00000508578.1           n.293G>C                NaN   \n",
       "6    ENST00000511045.6           c.642C>G  ENSP00000427679.2   \n",
       "7   ENST00000265517.10         c.*1046C>G                NaN   \n",
       "8                  NaN               None                NaN   \n",
       "9    ENST00000508578.1       n.32+1988G>C                NaN   \n",
       "10                 NaN               None                NaN   \n",
       "\n",
       "   HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  RDF  \\\n",
       "0         p.Gly151%3D  0/1   24   18  18  11   7  38.89%  3.812300e-03   11   \n",
       "1         p.Gly151%3D  0/1   24   18  18  11   7  38.89%  3.812300e-03   11   \n",
       "2          p.Gly68%3D  0/1   24   18  18  11   7  38.89%  3.812300e-03   11   \n",
       "3         p.His297Gln  0/1  100   39  39  14  25   64.1%  8.832600e-11   11   \n",
       "4         p.His297Gln  0/1  100   39  39  14  25   64.1%  8.832600e-11   11   \n",
       "5                None  0/1  100   39  39  14  25   64.1%  8.832600e-11   11   \n",
       "6         p.His214Gln  0/1  100   39  39  14  25   64.1%  8.832600e-11   11   \n",
       "7                None  0/1   85   46  46  23  23     50%  2.928300e-09   23   \n",
       "8                None  0/1   85   46  46  23  23     50%  2.928300e-09   23   \n",
       "9                None  0/1   85   46  46  23  23     50%  2.928300e-09   23   \n",
       "10               None  0/1   85   46  46  23  23     50%  2.928300e-09   23   \n",
       "\n",
       "    RDR  ADF  ADR             SIFT   PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  \\\n",
       "0     0    5    2              NaN        NaN  0.5443  0.8238  0.3084  0.6706   \n",
       "1     0    5    2              NaN        NaN  0.5443  0.8238  0.3084  0.6706   \n",
       "2     0    5    2              NaN        NaN  0.5443  0.8238  0.3084  0.6706   \n",
       "3     3   13   12  tolerated(0.54)  benign(0)  0.5423  0.8169  0.3069  0.6706   \n",
       "4     3   13   12  tolerated(0.54)  benign(0)  0.5423  0.8169  0.3069  0.6706   \n",
       "5     3   13   12              NaN        NaN  0.5423  0.8169  0.3069  0.6706   \n",
       "6     3   13   12  tolerated(0.54)  benign(0)  0.5423  0.8169  0.3069  0.6706   \n",
       "7     0   20    3              NaN        NaN  0.6877  0.9675  0.3905  0.7966   \n",
       "8     0   20    3              NaN        NaN  0.6877  0.9675  0.3905  0.7966   \n",
       "9     0   20    3              NaN        NaN  0.6877  0.9675  0.3905  0.7966   \n",
       "10    0   20    3              NaN        NaN  0.6877  0.9675  0.3905  0.7966   \n",
       "\n",
       "    EUR_AF  SAS_AF  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  \\\n",
       "0   0.3141  0.4407      0.3950          0.7687          0.2648   \n",
       "1   0.3141  0.4407      0.3950          0.7687          0.2648   \n",
       "2   0.3141  0.4407      0.3950          0.7687          0.2648   \n",
       "3   0.3151  0.4397      0.3944          0.7585          0.2649   \n",
       "4   0.3151  0.4397      0.3944          0.7585          0.2649   \n",
       "5   0.3151  0.4397      0.3944          0.7585          0.2649   \n",
       "6   0.3151  0.4397      0.3944          0.7585          0.2649   \n",
       "7   0.4384  0.6646         NaN             NaN             NaN   \n",
       "8   0.4384  0.6646         NaN             NaN             NaN   \n",
       "9   0.4384  0.6646         NaN             NaN             NaN   \n",
       "10  0.4384  0.6646         NaN             NaN             NaN   \n",
       "\n",
       "    gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  \\\n",
       "0           0.5787          0.6482          0.2079          0.3533   \n",
       "1           0.5787          0.6482          0.2079          0.3533   \n",
       "2           0.5787          0.6482          0.2079          0.3533   \n",
       "3           0.5788          0.6480          0.2082          0.3533   \n",
       "4           0.5788          0.6480          0.2082          0.3533   \n",
       "5           0.5788          0.6480          0.2082          0.3533   \n",
       "6           0.5788          0.6480          0.2082          0.3533   \n",
       "7              NaN             NaN             NaN             NaN   \n",
       "8              NaN             NaN             NaN             NaN   \n",
       "9              NaN             NaN             NaN             NaN   \n",
       "10             NaN             NaN             NaN             NaN   \n",
       "\n",
       "    gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  \\\n",
       "0           0.3900          0.4184      0.4712          0.7535   \n",
       "1           0.3900          0.4184      0.4712          0.7535   \n",
       "2           0.3900          0.4184      0.4712          0.7535   \n",
       "3           0.3901          0.4186      0.4684          0.7444   \n",
       "4           0.3901          0.4186      0.4684          0.7444   \n",
       "5           0.3901          0.4186      0.4684          0.7444   \n",
       "6           0.3901          0.4186      0.4684          0.7444   \n",
       "7              NaN             NaN      0.5941          0.8924   \n",
       "8              NaN             NaN      0.5941          0.8924   \n",
       "9              NaN             NaN      0.5941          0.8924   \n",
       "10             NaN             NaN      0.5941          0.8924   \n",
       "\n",
       "    gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  \\\n",
       "0           0.4693          0.3664          0.5739          0.6295   \n",
       "1           0.4693          0.3664          0.5739          0.6295   \n",
       "2           0.4693          0.3664          0.5739          0.6295   \n",
       "3           0.4702          0.3636          0.5747          0.6325   \n",
       "4           0.4702          0.3636          0.5747          0.6325   \n",
       "5           0.4702          0.3636          0.5747          0.6325   \n",
       "6           0.4702          0.3636          0.5747          0.6325   \n",
       "7           0.6077          0.4730          0.7056          0.7732   \n",
       "8           0.6077          0.4730          0.7056          0.7732   \n",
       "9           0.6077          0.4730          0.7056          0.7732   \n",
       "10          0.6077          0.4730          0.7056          0.7732   \n",
       "\n",
       "    gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  \\\n",
       "0           0.2062          0.5190          0.3510          0.4588   \n",
       "1           0.2062          0.5190          0.3510          0.4588   \n",
       "2           0.2062          0.5190          0.3510          0.4588   \n",
       "3           0.2057          0.5190          0.3508          0.4578   \n",
       "4           0.2057          0.5190          0.3508          0.4578   \n",
       "5           0.2057          0.5190          0.3508          0.4578   \n",
       "6           0.2057          0.5190          0.3508          0.4578   \n",
       "7           0.3049          0.6456          0.4623          0.5984   \n",
       "8           0.3049          0.6456          0.4623          0.5984   \n",
       "9           0.3049          0.6456          0.4623          0.5984   \n",
       "10          0.3049          0.6456          0.4623          0.5984   \n",
       "\n",
       "    gnomADg_SAS_AF  MAX_AF MAX_AF_POPS         BIOTYPE   EXON INTRON  \\\n",
       "0           0.4163  0.8238         AFR  protein coding   4/18    NaN   \n",
       "1           0.4163  0.8238         AFR  protein coding   5/19    NaN   \n",
       "2           0.4163  0.8238         AFR  protein coding   4/18    NaN   \n",
       "3           0.4164  0.8169         AFR  protein coding   7/18    NaN   \n",
       "4           0.4164  0.8169         AFR  protein coding   8/19    NaN   \n",
       "5           0.4164  0.8169         AFR          lncRNA    4/4    NaN   \n",
       "6           0.4164  0.8169         AFR  protein coding   7/18    NaN   \n",
       "7           0.6239  0.9675         AFR  protein coding  18/18    NaN   \n",
       "8           0.6239  0.9675         AFR  protein coding    NaN    NaN   \n",
       "9           0.6239  0.9675         AFR          lncRNA    NaN    1/3   \n",
       "10          0.6239  0.9675         AFR  protein coding    NaN    NaN   \n",
       "\n",
       "   Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "0                             G151  ggT/ggC       1   \n",
       "1                             G151  ggT/ggC       1   \n",
       "2                              G68  ggT/ggC       1   \n",
       "3                            H297Q  caC/caG       1   \n",
       "4                            H297Q  caC/caG       1   \n",
       "5                              NaN      NaN      -1   \n",
       "6                            H214Q  caC/caG       1   \n",
       "7                              NaN      NaN       1   \n",
       "8                              NaN      NaN       1   \n",
       "9                              NaN      NaN      -1   \n",
       "10                             NaN      NaN       1   \n",
       "\n",
       "                                               PUBMED  \n",
       "0                                            27170061  \n",
       "1                                            27170061  \n",
       "2                                            27170061  \n",
       "3   19878569,24496996,27170061,31321587,31397093,3...  \n",
       "4   19878569,24496996,27170061,31321587,31397093,3...  \n",
       "5   19878569,24496996,27170061,31321587,31397093,3...  \n",
       "6   19878569,24496996,27170061,31321587,31397093,3...  \n",
       "7                                                 NaN  \n",
       "8                                                 NaN  \n",
       "9                                                 NaN  \n",
       "10                                                NaN  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/pcsk9_genes.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b8583c46",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs991811</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99589702</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129625</td>\n",
       "      <td>ENST00000265517.10</td>\n",
       "      <td>c.453T&gt;C</td>\n",
       "      <td>ENSP00000265517.5</td>\n",
       "      <td>p.Gly151%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>38.89%</td>\n",
       "      <td>3.812300e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5443</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>0.3084</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3141</td>\n",
       "      <td>0.4407</td>\n",
       "      <td>0.3950</td>\n",
       "      <td>0.7687</td>\n",
       "      <td>0.2648</td>\n",
       "      <td>0.5787</td>\n",
       "      <td>0.6482</td>\n",
       "      <td>0.2079</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3900</td>\n",
       "      <td>0.4184</td>\n",
       "      <td>0.4712</td>\n",
       "      <td>0.7535</td>\n",
       "      <td>0.4693</td>\n",
       "      <td>0.3664</td>\n",
       "      <td>0.5739</td>\n",
       "      <td>0.6295</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3510</td>\n",
       "      <td>0.4588</td>\n",
       "      <td>0.4163</td>\n",
       "      <td>0.8238</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G151</td>\n",
       "      <td>ggT/ggC</td>\n",
       "      <td>1</td>\n",
       "      <td>27170061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs2306985</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99594865</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>129628</td>\n",
       "      <td>ENST00000265517.10</td>\n",
       "      <td>c.891C&gt;G</td>\n",
       "      <td>ENSP00000265517.5</td>\n",
       "      <td>p.His297Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>100</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>64.1%</td>\n",
       "      <td>8.832600e-11</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated(0.54)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.5423</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>0.3069</td>\n",
       "      <td>0.6706</td>\n",
       "      <td>0.3151</td>\n",
       "      <td>0.4397</td>\n",
       "      <td>0.3944</td>\n",
       "      <td>0.7585</td>\n",
       "      <td>0.2649</td>\n",
       "      <td>0.5788</td>\n",
       "      <td>0.6480</td>\n",
       "      <td>0.2082</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.3901</td>\n",
       "      <td>0.4186</td>\n",
       "      <td>0.4684</td>\n",
       "      <td>0.7444</td>\n",
       "      <td>0.4702</td>\n",
       "      <td>0.3636</td>\n",
       "      <td>0.5747</td>\n",
       "      <td>0.6325</td>\n",
       "      <td>0.2057</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.3508</td>\n",
       "      <td>0.4578</td>\n",
       "      <td>0.4164</td>\n",
       "      <td>0.8169</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>H297Q</td>\n",
       "      <td>caC/caG</td>\n",
       "      <td>1</td>\n",
       "      <td>19878569,24496996,27170061,31321587,31397093,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MTTP</td>\n",
       "      <td>MTTP</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cholesterol_Disorders; Gall_stones</td>\n",
       "      <td>rs1061271</td>\n",
       "      <td>No</td>\n",
       "      <td>chr4</td>\n",
       "      <td>99623894</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Abetalipoproteinaemia</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>347062</td>\n",
       "      <td>ENST00000265517.10</td>\n",
       "      <td>c.*1046C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>50%</td>\n",
       "      <td>2.928300e-09</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6877</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>0.3905</td>\n",
       "      <td>0.7966</td>\n",
       "      <td>0.4384</td>\n",
       "      <td>0.6646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5941</td>\n",
       "      <td>0.8924</td>\n",
       "      <td>0.6077</td>\n",
       "      <td>0.4730</td>\n",
       "      <td>0.7056</td>\n",
       "      <td>0.7732</td>\n",
       "      <td>0.3049</td>\n",
       "      <td>0.6456</td>\n",
       "      <td>0.4623</td>\n",
       "      <td>0.5984</td>\n",
       "      <td>0.6239</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>18/18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Name  Gene  Gene_Score  \\\n",
       "0      MTTP  MTTP           8   \n",
       "3      MTTP  MTTP           8   \n",
       "7      MTTP  MTTP           8   \n",
       "\n",
       "                                           Condition  \\\n",
       "0  Polygenic (Familial combined, small dense LDL,...   \n",
       "3  Polygenic (Familial combined, small dense LDL,...   \n",
       "7  Polygenic (Familial combined, small dense LDL,...   \n",
       "\n",
       "                            Headings                  21_Conditions_list  \\\n",
       "0  Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "3  Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "7  Cholesterol disorders; Gallstones  Cholesterol_Disorders; Gall_stones   \n",
       "\n",
       "        rsID Literature CHROM       POS REF ALT      Zygosity  \\\n",
       "0   rs991811         No  chr4  99589702   T   C  Heterozygous   \n",
       "3  rs2306985        Yes  chr4  99594865   C   G  Heterozygous   \n",
       "7  rs1061271         No  chr4  99623894   C   G  Heterozygous   \n",
       "\n",
       "           Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0   synonymous variant              3/10       LOW           2.5   \n",
       "3     missense variant              7/10  MODERATE           5.0   \n",
       "7  3 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "\n",
       "           ClinVar_CLNDN              CLIN_SIG  \\\n",
       "0  Abetalipoproteinaemia                benign   \n",
       "3  Abetalipoproteinaemia  benign,likely benign   \n",
       "7  Abetalipoproteinaemia                benign   \n",
       "\n",
       "                                  ClinVar_CLNREVSTAT  ClinVar  \\\n",
       "0  criteria provided, multiple submitters, no con...   129625   \n",
       "3  criteria provided, multiple submitters, no con...   129628   \n",
       "7                criteria provided, single submitter   347062   \n",
       "\n",
       "                HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0  ENST00000265517.10           c.453T>C  ENSP00000265517.5   \n",
       "3  ENST00000265517.10           c.891C>G  ENSP00000265517.5   \n",
       "7  ENST00000265517.10         c.*1046C>G                NaN   \n",
       "\n",
       "  HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  RDF  \\\n",
       "0        p.Gly151%3D  0/1   24   18  18  11   7  38.89%  3.812300e-03   11   \n",
       "3        p.His297Gln  0/1  100   39  39  14  25   64.1%  8.832600e-11   11   \n",
       "7               None  0/1   85   46  46  23  23     50%  2.928300e-09   23   \n",
       "\n",
       "   RDR  ADF  ADR             SIFT   PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  \\\n",
       "0    0    5    2              NaN        NaN  0.5443  0.8238  0.3084  0.6706   \n",
       "3    3   13   12  tolerated(0.54)  benign(0)  0.5423  0.8169  0.3069  0.6706   \n",
       "7    0   20    3              NaN        NaN  0.6877  0.9675  0.3905  0.7966   \n",
       "\n",
       "   EUR_AF  SAS_AF  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  \\\n",
       "0  0.3141  0.4407      0.3950          0.7687          0.2648          0.5787   \n",
       "3  0.3151  0.4397      0.3944          0.7585          0.2649          0.5788   \n",
       "7  0.4384  0.6646         NaN             NaN             NaN             NaN   \n",
       "\n",
       "   gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  \\\n",
       "0          0.6482          0.2079          0.3533          0.3900   \n",
       "3          0.6480          0.2082          0.3533          0.3901   \n",
       "7             NaN             NaN             NaN             NaN   \n",
       "\n",
       "   gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "0          0.4184      0.4712          0.7535          0.4693          0.3664   \n",
       "3          0.4186      0.4684          0.7444          0.4702          0.3636   \n",
       "7             NaN      0.5941          0.8924          0.6077          0.4730   \n",
       "\n",
       "   gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "0          0.5739          0.6295          0.2062          0.5190   \n",
       "3          0.5747          0.6325          0.2057          0.5190   \n",
       "7          0.7056          0.7732          0.3049          0.6456   \n",
       "\n",
       "   gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF MAX_AF_POPS  \\\n",
       "0          0.3510          0.4588          0.4163  0.8238         AFR   \n",
       "3          0.3508          0.4578          0.4164  0.8169         AFR   \n",
       "7          0.4623          0.5984          0.6239  0.9675         AFR   \n",
       "\n",
       "          BIOTYPE   EXON INTRON Protein Position and Amino Acid   Codons  \\\n",
       "0  protein coding   4/18    NaN                            G151  ggT/ggC   \n",
       "3  protein coding   7/18    NaN                           H297Q  caC/caG   \n",
       "7  protein coding  18/18    NaN                             NaN      NaN   \n",
       "\n",
       "   STRAND                                             PUBMED  \n",
       "0       1                                           27170061  \n",
       "3       1  19878569,24496996,27170061,31321587,31397093,3...  \n",
       "7       1                                                NaN  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.drop_duplicates(subset='rsID', keep=\"first\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9539b3d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/AIGPRX1351_sample_MTTP_gene.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90679aed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c072ec5",
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_file_path = 'your_output_file.bed'\n",
    "df.to_csv(bed_file_path, sep='\\t', header=False, index=False, columns=['chromosome', 'start', 'end', 'name', 'score', 'strand'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "32fa374b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55038957</td>\n",
       "      <td>rs1644576644</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55038958</td>\n",
       "      <td>rs1281656568</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55038965</td>\n",
       "      <td>rs915165350</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55038966</td>\n",
       "      <td>rs1160766107</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55038968</td>\n",
       "      <td>rs1644576734</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1687</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55064679</td>\n",
       "      <td>rs1644788787</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1688</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55064681</td>\n",
       "      <td>rs551315846</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1689</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55064682</td>\n",
       "      <td>rs1644788868</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1690</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55064683</td>\n",
       "      <td>rs2100350182</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1691</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55064685</td>\n",
       "      <td>rs903883328</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1692 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS          rsID REF  ALT\n",
       "0     chr1  55038957  rs1644576644   T    C\n",
       "1     chr1  55038958  rs1281656568   C    T\n",
       "2     chr1  55038965   rs915165350   T    C\n",
       "3     chr1  55038966  rs1160766107   A    G\n",
       "4     chr1  55038968  rs1644576734   C    A\n",
       "...    ...       ...           ...  ..  ...\n",
       "1687  chr1  55064679  rs1644788787   C    T\n",
       "1688  chr1  55064681   rs551315846   C  A,T\n",
       "1689  chr1  55064682  rs1644788868   C    G\n",
       "1690  chr1  55064683  rs2100350182   C    T\n",
       "1691  chr1  55064685   rs903883328   A    G\n",
       "\n",
       "[1692 rows x 5 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/covered_bed_variants.tsv', sep = '\\t')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "7efd7940",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/data_new.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "b367a865",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55038956</td>\n",
       "      <td>55038997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55039486</td>\n",
       "      <td>55039527</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     START       END\n",
       "0  chr1  55038956  55038997\n",
       "1  chr1  55039486  55039527"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/chrom_data.xlsx')\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "aab23245",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Can only compare identically-labeled Series objects",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[29], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m merged_data \u001b[38;5;241m=\u001b[39m df_1[(\u001b[43mdf_1\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mPOS\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m>\u001b[39;49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mSTART\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m) \u001b[38;5;241m&\u001b[39m (df_1[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPOS\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m<\u001b[39m\u001b[38;5;241m=\u001b[39m df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mEND\u001b[39m\u001b[38;5;124m'\u001b[39m])]\n\u001b[0;32m      3\u001b[0m \u001b[38;5;66;03m# Create a new column in df and fill it with rsID values\u001b[39;00m\n\u001b[0;32m      4\u001b[0m df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mnew_column\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m merged_data[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mrsID\u001b[39m\u001b[38;5;124m'\u001b[39m]\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\ops\\common.py:72\u001b[0m, in \u001b[0;36m_unpack_zerodim_and_defer.<locals>.new_method\u001b[1;34m(self, other)\u001b[0m\n\u001b[0;32m     68\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mNotImplemented\u001b[39m\n\u001b[0;32m     70\u001b[0m other \u001b[38;5;241m=\u001b[39m item_from_zerodim(other)\n\u001b[1;32m---> 72\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mmethod\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mother\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\arraylike.py:62\u001b[0m, in \u001b[0;36mOpsMixin.__ge__\u001b[1;34m(self, other)\u001b[0m\n\u001b[0;32m     60\u001b[0m \u001b[38;5;129m@unpack_zerodim_and_defer\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m__ge__\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     61\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__ge__\u001b[39m(\u001b[38;5;28mself\u001b[39m, other):\n\u001b[1;32m---> 62\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cmp_method\u001b[49m\u001b[43m(\u001b[49m\u001b[43mother\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moperator\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mge\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:6237\u001b[0m, in \u001b[0;36mSeries._cmp_method\u001b[1;34m(self, other, op)\u001b[0m\n\u001b[0;32m   6234\u001b[0m res_name \u001b[38;5;241m=\u001b[39m ops\u001b[38;5;241m.\u001b[39mget_op_result_name(\u001b[38;5;28mself\u001b[39m, other)\n\u001b[0;32m   6236\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(other, Series) \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_indexed_same(other):\n\u001b[1;32m-> 6237\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCan only compare identically-labeled Series objects\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m   6239\u001b[0m lvalues \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values\n\u001b[0;32m   6240\u001b[0m rvalues \u001b[38;5;241m=\u001b[39m extract_array(other, extract_numpy\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, extract_range\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "\u001b[1;31mValueError\u001b[0m: Can only compare identically-labeled Series objects"
     ]
    }
   ],
   "source": [
    "merged_data = df_1[(df_1['POS'] >= df['START']) & (df_1['POS'] <= df['END'])]\n",
    "\n",
    "# Create a new column in df and fill it with rsID values\n",
    "df['new_column'] = merged_data['rsID']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "a2d545f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genotype/Allele</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AG</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GG</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CC</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Genotype/Allele REF  ALT\n",
       "0              AA   G    A\n",
       "1              AG   G    A\n",
       "2              GG   G    A\n",
       "3              CC   T    C\n",
       "4              GT   G    T\n",
       "5              GT   G    T\n",
       "6              GT   G    T\n",
       "7              CA   C    A\n",
       "8              CA   C    A\n",
       "9              CA   C  A,T"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample data\n",
    "data = {'Genotype/Allele': ['AA', 'AG', 'GG', 'CC', 'GT', 'GT', 'GT', 'CA', 'CA', 'CA'],\n",
    "        'REF': ['G', 'G', 'G', 'T', 'G', 'G', 'G', 'C', 'C', 'C'],\n",
    "        'ALT': ['A', 'A', 'A', 'C', 'T', 'T', 'T', 'A', 'A', 'A,T']}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "bdf109a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genotype/Allele</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AG</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GG</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CC</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Genotype/Allele REF  ALT      Zygosity\n",
       "0              AA   G    A    Homozygous\n",
       "1              AG   G    A  Heterozygous\n",
       "2              GG   G    A    Homozygous\n",
       "3              CC   T    C    Homozygous\n",
       "4              GT   G    T  Heterozygous\n",
       "5              GT   G    T  Heterozygous\n",
       "6              GT   G    T  Heterozygous\n",
       "7              CA   C    A  Heterozygous\n",
       "8              CA   C    A  Heterozygous\n",
       "9              CA   C  A,T  Heterozygous"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# Function to determine Zygosity\n",
    "def determine_zygosity(row):\n",
    "    if ',' in row['ALT']:\n",
    "        # If ALT has a comma, take the first part\n",
    "        alt = row['ALT'].split(',')[0]\n",
    "    else:\n",
    "        alt = row['ALT']\n",
    "\n",
    "    if row['Genotype/Allele'] == row['REF'] + row['REF']:\n",
    "        return 'Homozygous'\n",
    "    elif row['Genotype/Allele'] == row['REF'] + alt or row['Genotype/Allele'] == alt + row['REF']:\n",
    "        return 'Heterozygous'\n",
    "    elif row['Genotype/Allele'] == alt + alt:\n",
    "        return 'Homozygous'\n",
    "    else:\n",
    "        return 'Unknown'\n",
    "\n",
    "# Apply the function to create the Zygosity column\n",
    "df['Zygosity'] = df.apply(determine_zygosity, axis=1)\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "50a87cc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>POMGNT1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>46195898</td>\n",
       "      <td>AA</td>\n",
       "      <td>TT</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>POMGNT1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>46194637</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>chr1</td>\n",
       "      <td>156134829</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>chr1</td>\n",
       "      <td>156130744</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>chr1</td>\n",
       "      <td>156135197</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3826</th>\n",
       "      <td>MYH7</td>\n",
       "      <td>chr14</td>\n",
       "      <td>23415801</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3827</th>\n",
       "      <td>MYH7</td>\n",
       "      <td>chr14</td>\n",
       "      <td>23415778</td>\n",
       "      <td>TCTC</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3828</th>\n",
       "      <td>MYH7</td>\n",
       "      <td>chr14</td>\n",
       "      <td>23415478</td>\n",
       "      <td>T</td>\n",
       "      <td>TTCT</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3829</th>\n",
       "      <td>MYH7</td>\n",
       "      <td>chr14</td>\n",
       "      <td>23415153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3830</th>\n",
       "      <td>MYH7</td>\n",
       "      <td>chr14</td>\n",
       "      <td>23415174</td>\n",
       "      <td>GCAG</td>\n",
       "      <td>G</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3172 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  CHROM        POS   REF   ALT Literature Covered/Not_Covered\n",
       "0     POMGNT1   chr1   46195898    AA    TT        Yes             Covered\n",
       "1     POMGNT1   chr1   46194637     C     T        Yes             Covered\n",
       "2        LMNA   chr1  156134829     C     T        Yes             Covered\n",
       "3        LMNA   chr1  156130744     C     T        Yes             Covered\n",
       "4        LMNA   chr1  156135197     C     T        Yes             Covered\n",
       "...       ...    ...        ...   ...   ...        ...                 ...\n",
       "3826     MYH7  chr14   23415801     C     G        Yes             Covered\n",
       "3827     MYH7  chr14   23415778  TCTC     T        Yes             Covered\n",
       "3828     MYH7  chr14   23415478     T  TTCT        Yes             Covered\n",
       "3829     MYH7  chr14   23415153     C     T        Yes             Covered\n",
       "3830     MYH7  chr14   23415174  GCAG     G        Yes             Covered\n",
       "\n",
       "[3172 rows x 7 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Muscularhealth_final_Lit_Positions_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "x['REF'] = x['chrom'].str.split('-').str[2]\n",
    "x['ALT'] = x['chrom'].str.split('-').str[3]\n",
    "\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['Gene', 'CHROM', 'POS', 'REF', 'ALT']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['Gene', 'CHROM', 'POS', 'REF', 'ALT', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a1803d15",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/Muscularhealth_lit.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df50b30b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "528e91e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Index 1 is out-of-bounds in filtered_df1.\n",
      "Error: Index 1 is out-of-bounds in filtered_df1.\n",
      "Error: Index 2 is out-of-bounds in filtered_df1.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\GenepoweRx_Madhu\\AppData\\Local\\Temp\\ipykernel_6312\\427133746.py:35: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  result_df = result_df.append({'CHROM': row2['CHROM'], 'POS': nearest_row['POS'], 'Gene': row2['Gene']}, ignore_index=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100</td>\n",
       "      <td>GeneA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM  POS   Gene\n",
       "0  chr1  100  GeneA"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample DataFrames\n",
    "data1 = {'CHROM': ['chr1', 'chr2', 'chr3'],\n",
    "         'POS': [100, 200, 300],\n",
    "         'Gene': ['GeneA', 'GeneB', 'GeneC']}\n",
    "\n",
    "data2 = {'CHROM': ['chr1', 'chr2', 'chr2', 'chr3'],\n",
    "         'POS': [90, 180, 220, 310],\n",
    "         'Gene': ['GeneA', 'GeneB', 'GeneB', 'GeneC']}\n",
    "\n",
    "df1 = pd.DataFrame(data1)\n",
    "df2 = pd.DataFrame(data2)\n",
    "\n",
    "# Initialize an empty DataFrame to store the results\n",
    "result_df = pd.DataFrame(columns=df2.columns)\n",
    "\n",
    "# Iterate through rows in dataset2\n",
    "for index2, row2 in df2.iterrows():\n",
    "    chrom_filter = df1['CHROM'] == row2['CHROM']\n",
    "    gene_filter = df1['Gene'] == row2['Gene']\n",
    "\n",
    "    # Filter dataset1 based on CHROM and Gene\n",
    "    filtered_df1 = df1[chrom_filter & gene_filter]\n",
    "\n",
    "    # Check if the filtered DataFrame is not empty\n",
    "    if not filtered_df1.empty:\n",
    "        # Find the nearest POS in filtered dataset1\n",
    "        min_pos_index = (filtered_df1['POS'] - row2['POS']).abs().idxmin()\n",
    "        \n",
    "        if min_pos_index < len(filtered_df1):\n",
    "            nearest_row = filtered_df1.iloc[min_pos_index]\n",
    "            \n",
    "            # Append the result to the result DataFrame\n",
    "            result_df = result_df.append({'CHROM': row2['CHROM'], 'POS': nearest_row['POS'], 'Gene': row2['Gene']}, ignore_index=True)\n",
    "        else:\n",
    "            print(f\"Error: Index {min_pos_index} is out-of-bounds in filtered_df1.\")\n",
    "    else:\n",
    "        # Handle the case where the filtered DataFrame is empty\n",
    "        print(f\"No matching rows in dataset1 for {row2['Gene']} and {row2['CHROM']} in dataset2\")\n",
    "\n",
    "# Display the result\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "149d124d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55074009_novel_C_T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55074009_novel_C_T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>KHHSPTGPONC31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Variant      Zygosity         Sample\n",
       "0  chr4_55074009_novel_C_T  Heterozygous  KHHSPTGPONC31\n",
       "1  chr4_55074009_novel_C_T    Homozygous  KHHSPTGPONC31"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example DataFrame\n",
    "data = {'Variant': ['chr4_55074009_novel_C_T', 'chr4_55074009_novel_C_T'],\n",
    "        'Zygosity': ['Heterozygous', 'Homozygous'],\n",
    "        'Sample': ['KHHSPTGPONC31', 'KHHSPTGPONC31']}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d337eaca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55074009_novel_C_T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Variant      Zygosity         Sample\n",
       "0  chr4_55074009_novel_C_T  Heterozygous  KHHSPTGPONC31"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example DataFrame\n",
    "data = {'Variant': ['chr4_55074009_novel_C_T', 'chr4_55074009_novel_C_T'],\n",
    "        'Zygosity': ['Heterozygous', 'Homozygous'],\n",
    "        'Sample': ['KHHSPTGPONC31', 'KHHSPTGPONC31']}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Identify and remove rows with the same \"Variant\" and \"Sample\" but \"Zygosity\" is Homozygous\n",
    "mask = (df.duplicated(subset=['Variant', 'Sample'], keep=False) & (df['Zygosity'] == 'Homozygous'))\n",
    "filtered_df = df[~mask]\n",
    "\n",
    "# Resetting the index if needed\n",
    "filtered_df = filtered_df.reset_index(drop=True)\n",
    "\n",
    "# Displaying the resulting DataFrame\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "fa432da3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loadings of each rsID for the first few principal components:\n",
      "                 PC1       PC2       PC3\n",
      "rsID                                    \n",
      "rs2231142   0.159636  0.287004 -0.479993\n",
      "rs4961     -0.039098  0.339753 -0.682084\n",
      "rs1042713  -0.329768 -0.017079 -0.146838\n",
      "rs7412      0.298918  0.119173 -0.025294\n",
      "rs4673993  -0.223284  0.440982  0.291636\n",
      "rs71647871  0.399261  0.056007  0.079850\n",
      "rs78769542  0.402465  0.049328  0.077809\n",
      "rs16969968  0.034646  0.377825  0.310084\n",
      "rs12777823 -0.154137  0.279994 -0.087696\n",
      "rs3745274  -0.178643  0.162863  0.163454\n",
      "rs2242480  -0.199838 -0.325096 -0.014796\n",
      "rs2108622  -0.186463  0.463836  0.209581\n",
      "rs1801160   0.331703  0.128803  0.060702\n",
      "rs3918290   0.402546  0.050728  0.079352\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABjUAAAJOCAYAAAD/KYUYAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAABt7ElEQVR4nOzde5yVVb0/8M9mZmAQmVFAURQQtdJCkYsgmllZpKmFlUoWeC2vPzNPF+mmWR06nTI7GZYnL2GmpillmUVH84qKCKZHrUwTFVBBmEERGIb9+8PD1DQMzODcNvN+v177VXvt9ez93ZuH7bP47LVWoVgsFgMAAAAAANDF9ejsAgAAAAAAAFpCqAEAAAAAAJQEoQYAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlQagBAAAAAACUBKEGAAAAAABQEoQaAAAAAABASRBqAAAAAAAAJUGoAbAFufLKK1MoFBpu5eXl2XnnnXPCCSfk+eefb9L/qaeeyplnnpk3v/nN6d27d7baaqu87W1vy5e+9KUN9k+SD33oQykUCjnzzDObreNLX/pSDj/88Oy0004pFAo5/vjj2+otvmHnn39+o8+oZ8+eGTZsWD71qU9l+fLlTfr/6U9/ygknnJBhw4alsrIyW2+9dUaNGpVvfetbefnllzf4GqNGjUqhUMi3v/3tDT5+22235cQTT8wee+yRPn36ZKeddsoHP/jBzJ07ty3fKgAAvCHGF5vWFcYX8+fPz2GHHZYhQ4akd+/e6devX8aPH5+f/vSnbflWAboMoQbAFuiKK67I7NmzM2vWrHziE5/INddckwMPPDCvvvpqQ59f//rX2XvvvfPrX/86n/zkJ/PrX/+64f/ffPPNOfzww5s874svvphf//rXSZKrr746q1at2uDrf/e7383SpUvzgQ98ID179myfN/kG3XrrrZk9e3Z+85vfZOLEifn+97+fQw89NMVisaHPf//3f2f06NGZM2dOPvvZz+bWW2/NTTfdlKOOOio//OEPc9JJJzV53vnz52fevHlJkssuu2yDr33JJZfk73//ez71qU/llltuyfe+9728+OKL2W+//XLbbbe1zxsGAIDNZHyxaZ05vli+fHkGDx6cf//3f88tt9ySGTNmZJdddsnkyZPz9a9/vX3eMEBnKgKwxbjiiiuKSYpz5sxp1P7lL3+5mKT405/+tFgsFotPPfVUsU+fPsWRI0cWly9f3uR51q1bV/zFL37RpP0///M/i0mKhx12WDFJ8eqrr95gHfX19Q3/v0+fPsXjjjvuDbyrtnXeeecVkxRfeumlRu2TJ08uJinefffdxWKxWLz33nuLZWVlxUMOOaS4atWqJs+zevXq4i9/+csm7WeccUajz+iee+5p0ueFF15o0rZixYriwIEDiwcffPDmvjUAAGhTxheb1hXGF80ZN25ccfDgwa18RwBdn5kaAN3AfvvtlyR55plnkiQXXnhhXn311UyfPj3V1dVN+hcKhXzoQx9q0n755Zdn4MCB+clPfpLevXvn8ssv3+Dr9eixef95mTlzZgqFQv7nf/6nyWOXXHJJCoVC/vSnPyV5fWr7pEmTMmjQoPTq1SsDBw7MwQcfnPnz52/Wa//rZ/Tv//7vKRQKufTSS9OrV68m/Xv27JkPfOADjdpWrVqVn/3sZxk9enS++93vJskGP6Ptt9++SdvWW2+dt771rXn22Wc3q34AAOgoxheb1pHji+YMGDAg5eXlm1U/QFcm1ADoBp588skkyXbbbZck+f3vf5+BAwc2XGi3xL333pvHH388U6ZMSf/+/fPhD384t912W55++uk2q/Pwww/P9ttvnyuuuKLJY1deeWVGjRqVvffeO0ny/ve/P3Pnzs23vvWtzJo1K5dccklGjhy5wXVrW+KfP6P6+vrcdtttGT16dAYPHtzi57jxxhuzbNmynHjiiXnTm96Ut7/97bnuuuvyyiuvbPLYmpqaPPTQQ3nb2962WfUDAEBHMb7YtM4YX6xbty5r167NSy+9lOnTp+d3v/tdPv/5z29W/QBdmVADYAtUX1+ftWvX5pVXXslvfvObfP3rX0/fvn0bfvmzYMGCDBs2rFXPuX791hNPPDFJctJJJ6VYLG5wgLC5ysvL8/GPfzw33nhjampqGtoff/zxPPDAAznhhBOSJEuXLs2f//znnHnmmfn4xz+ed7zjHfnQhz6Ub3/723nnO9/Zotda/xktX748V199dX74wx9m8ODBOfDAA7NkyZKsXLlysz6jysrKHHvssUle/4xeeeWV/PznP9/ksWeccUZeffXVfPGLX2zVawIAQHszvti0rjC+OP3001NRUZHtt98+n/70p/Nf//VfOeWUU1r1mgClQKgBsAXab7/9UlFRkb59++bwww/PDjvskN/+9rcZOHDgZj3f+gvn/fffP3vssUeS5KCDDspuu+2WK6+8MuvWrWuz2k888cS89tprue666xrarrjiivTq1avhYr5fv37Zbbfd8p//+Z+58MILM2/evFbXsMMOO6SioiLbbrttPv7xj2fUqFG59dZbU1lZuVl1P/3007n99tvzoQ99KNtss02S5Kijjkrfvn03OUX8y1/+cq6++up897vfzejRozfr9QEAoL0YX2xaVxhffOELX8icOXPym9/8JieeeGLOPPPMfPvb396s1wfoyoQaAFugGTNmZM6cOZk3b14WLlyYP/3pTznggAMaHh8yZEirpnWvn+J89NFHZ/ny5Vm+fHlqampy9NFH59lnn82sWbParPa3ve1t2XfffRt+oVVfX5+f/vSn+eAHP5h+/folScO6uO973/vyrW99K6NGjcp2222Xs846KytWrGjR6/zhD3/InDlzMn/+/CxZsiR333133vrWtyZ5fe3ZrbbaqlWf0eWXX55isZiPfOQjDZ9RXV1dPvCBD+See+7JE088scHjvvrVr+brX/96vvGNb+TMM89s8esBAEBHMb7YtK4wvhgyZEjGjBmT97///bnkkkvyyU9+MlOnTs1LL73U4tcFKAVCDYAt0J577pkxY8Zkn332yY477tjk8fe973154YUXct9997Xo+dZPDT/77LOz7bbbNtymTZvW6PG2csIJJ+S+++7L448/nltvvTWLFi1qmBq+3tChQ3PZZZdl8eLF+fOf/5xPf/rTmT59ej772c+26DVGjBiRMWPGZMSIEenfv3+jx8rKynLwwQdn7ty5ee655zb5XOvWrcuVV16ZJPnQhz7U6DO6+uqrk2x4Q7+vfvWrOf/883P++efnC1/4QovqBgCAjmZ8sWldYXzxr8aOHZu1a9fmqaeeatF7ACgVQg2AbujTn/50+vTpk9NPP73R2rLrFYvF3HTTTUleX2929uzZ+fCHP5zbb7+9ye3ggw/OL3/5yyxdurTN6vvoRz+aysrKXHnllbnyyiuz0047ZcKECc32f/Ob35wvfelL2WuvvfLQQw+1SQ1Tp05NsVjMJz7xiaxZs6bJ43V1dbn55puTJL/73e/y3HPP5YwzztjgZ/S2t70tM2bMyNq1axuO/9rXvpbzzz8/X/rSl3Leeee1Sc0AANAZjC82rb3HFxty++23p0ePHtl1113b5D0AdBXlnV0AAB1v2LBhufbaa3PMMcdkn332yZlnnpmRI0cmSR577LGGqc5HHnlkw6+kPve5z2Xs2LFNnmvFihX5n//5n/z0pz/Npz71qSTJHXfc0TDFub6+Ps8880xuuOGGJK+vlbvddttttL5tttkmRx55ZK688sosX748n/nMZ9Kjxz9y+D/96U8588wzc9RRR+VNb3pTevbsmdtuuy1/+tOfcu65577xDyjJ+PHjc8kll+T000/P6NGjc9ppp+Vtb3tb6urqMm/evFx66aUZPnx4jjjiiFx22WUpLy/PF77whQwaNKjJc51yyik566yz8pvf/CYf/OAH853vfCdf+cpXcsghh+Swww5r8ou2/fbbr03eAwAAdATji01rz/HFJz/5yVRVVWXs2LEZOHBglixZkuuvvz7XXXddPvvZz27y8wEoOUUAthhXXHFFMUlxzpw5Ler/t7/9rXj66acXd99992KvXr2KvXv3Lr71rW8tnnPOOcWnn366uGbNmuL2229f3GeffZp9jrVr1xZ33nnn4l577dXQdtBBBxWTbPB2++23t6i23//+9w3H/OUvf2n02AsvvFA8/vjji3vssUexT58+xa233rq49957F7/73e8W165du9HnPe+884pJii+99FKL6pg/f37xuOOOKw4ZMqTYs2fPYp8+fYojR44sfuUrXym++OKLxZdeeqnYs2fP4sSJE5t9jmXLlhV79+5dPOKII4rF4sY/H/9pBgCgqzC+KI3xxeWXX1488MADiwMGDCiWl5cXt9lmm+JBBx1UvOqqq1pUE0CpKRSLxWIH5ScAAAAAAACbzZ4aAAAAAABASRBqAAAAAAAAJUGoAQAAAAAAlAShBgAAAAAAUBKEGgAAAAAAQEkQagAAAAAAACWhvLML6OrWrVuXhQsXpm/fvikUCp1dDgAAdAnFYjErVqzIoEGD0qOH30q1hjEGAAA01dIxhlBjExYuXJjBgwd3dhkAANAlPfvss9l55507u4ySYowBAADN29QYQ6ixCX379k3y+gdZVVXVydUAAEDXUFtbm8GDBzdcL9NyxhgAANBUS8cYQo1NWD8dvKqqyoADAAD+heWTWs8YAwAAmrepMYbFbwEAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEpCeWcXAABA56hbtzbzlv09tXWvpaqid0Zuu0sqerg8BAAoJXV19Xnk4QVZUfta+lb1zl4jhqSioqyzywJoN0atAADdTN26tbnyqTvy82dmp6ZuZUN7dcVWOXro+By/60HCDQCALq6urj7XzrgnM2+Yk9qa1xraq6p7Z+JH9s2kKQcIN4AtktEqAEA3Urdubc6ZOyMPLP1biik2eqymbmV+/ORt+dOyZ3Lh6CmCDQCALqqurj5f/uy1eWjO0yk2vqRLbc1rueryO/O/jzybr/3nJMEGsMWxpwYAQDdy5VN3bDDQWK+YYh5Y+rdc+dQdHVwZAAAtde2MezYYaKxXLCYPzXk61864p2MLA+gAQg0AgG6ibt3a/PyZ2c0GGusVU8z1z8xO3bq1HVQZAAAtVVdXn5k3zGk20FivWExm3jAndXX1HVMYQAcRagAAdBPzlv290R4aG7O8bmXmL3umnSsCAKC1Hnl4QaM9NDamtua1PPrwgnauCKBjCTUAALqJ2rqWDX7/0b9lAQgAAB1nRW3rrulW1K5qp0oAOodQAwCgm6iq6N3K/lu1UyUAAGyuvlWtu6brW1XZTpUAdA6hBgBANzFy211S3cKgYpuKrbLPtkPbuSIAAFprrxFDUlXdsmCjqrp3ho8Y0s4VAXQsoQYAQDdR0aM8Rw8dn0IKG+1XSCFHDR2fih7lHVQZAAAtVVFRlokf2TeFjV/SpVBIJn5k31RUlHVMYQAdRKgBANCNHL/rQRnbf7dmg41CChnbf7ccv+tBHVwZAAAtNWnKARm177Bmg41CIRm177BMmnJAxxYG0AGEGgAA3UhFj/JcOHpKTt793U2Woqqu2Con7/7uXDh6ilkaAABdWEVFWb72n5My+cR3NFmKqqq6dyaf+I587T8nmaUBbJEKxWKx2NlFdGW1tbWprq5OTU1NqqqqOrscAIA2U7dubeYveya1dStT9X97aAgzaCnXyZvPZwdAW6qrq8+jDy/IitpV6VtVmeEjhggzgJLU0utko1YAgG6qokd59u2/W2eXAQDAG1BRUZaRY4Z1dhkAHcbyUwAAAAAAQEkQagAAAAAAACVBqAEAAAAAAJQEoQYAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlQagBAAAAAACUBKEGAAAAAABQEoQaAAAAAABASRBqAAAAAAAAJUGoAQAAAAAAlAShBgAAAAAAUBKEGgAAAAAAQEkQagAAAAAAACVBqAEAAAAAAJQEoQYAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlQagBAAAAAACUBKEGAAAAAABQEoQaAAAAAABASSi5UGP69OkZNmxYKisrM3r06Nx1110b7X/11VdnxIgR2WqrrbLjjjvmhBNOyNKlSzuoWgAAAAAAoK2UVKhx3XXX5eyzz84Xv/jFzJs3LwceeGAOPfTQLFiwYIP977777kyZMiUnnXRS/vd//zfXX3995syZk5NPPrmDKwcAAAAAAN6okgo1Lrzwwpx00kk5+eSTs+eee+aiiy7K4MGDc8kll2yw/3333ZdddtklZ511VoYNG5a3v/3tOeWUU/Lggw92cOUAAAAAAMAbVTKhxpo1azJ37txMmDChUfuECRNy7733bvCY/fffP88991xuueWWFIvFvPDCC7nhhhty2GGHNfs6q1evTm1tbaMbAAAAAADQ+Uom1FiyZEnq6+szcODARu0DBw7M4sWLN3jM/vvvn6uvvjrHHHNMevbsmR122CHbbLNNvv/97zf7OtOmTUt1dXXDbfDgwW36PgAAAAAAgM1TMqHGeoVCodH9YrHYpG29xx57LGeddVa+8pWvZO7cubn11lvz9NNP59RTT232+adOnZqampqG27PPPtum9QMAAAAAAJunvLMLaKkBAwakrKysyayMF198scnsjfWmTZuWAw44IJ/97GeTJHvvvXf69OmTAw88MF//+tez4447NjmmV69e6dWrV9u/AQAAAAAA4A0pmZkaPXv2zOjRozNr1qxG7bNmzcr++++/wWNWrlyZHj0av8WysrIkr8/wAAAAAAAASkfJhBpJcs455+THP/5xLr/88jz++OP59Kc/nQULFjQsJzV16tRMmTKlof8RRxyRG2+8MZdcckmeeuqp3HPPPTnrrLMyduzYDBo0qLPeBgAAAAAAsBlKKtQ45phjctFFF+WCCy7IPvvskzvvvDO33HJLhg4dmiRZtGhRFixY0ND/+OOPz4UXXpiLL744w4cPz1FHHZW3vOUtufHGGzvrLQAAAF3M9OnTM2zYsFRWVmb06NG56667WnTcPffck/Ly8uyzzz7tWyAAANCgULQO00bV1tamuro6NTU1qaqq6uxyAACgS9hSrpOvu+66TJ48OdOnT88BBxyQH/3oR/nxj3+cxx57LEOGDGn2uJqamowaNSq77757XnjhhcyfP7/Fr7mlfHYAANCWWnqdXFIzNQAAANrShRdemJNOOiknn3xy9txzz1x00UUZPHhwLrnkko0ed8opp+TYY4/N+PHjO6hSAAAgEWoAAADd1Jo1azJ37txMmDChUfuECRNy7733NnvcFVdckb/97W8577zz2rtEAADgX5R3dgEAAACdYcmSJamvr8/AgQMbtQ8cODCLFy/e4DF//etfc+655+auu+5KeXnLhlOrV6/O6tWrG+7X1tZuftEAANDNmakBAAB0a4VCodH9YrHYpC1J6uvrc+yxx+arX/1q3vzmN7f4+adNm5bq6uqG2+DBg99wzQAA0F0JNQAAgG5pwIABKSsrazIr48UXX2wyeyNJVqxYkQcffDBnnnlmysvLU15engsuuCAPP/xwysvLc9ttt23wdaZOnZqampqG27PPPtsu7wcAALoDy08BAADdUs+ePTN69OjMmjUrRx55ZEP7rFmz8sEPfrBJ/6qqqjzyyCON2qZPn57bbrstN9xwQ4YNG7bB1+nVq1d69erVtsUDAEA3JdQAAAC6rXPOOSeTJ0/OmDFjMn78+Fx66aVZsGBBTj311CSvz7J4/vnnM2PGjPTo0SPDhw9vdPz222+fysrKJu0AAED7EGoAAADd1jHHHJOlS5fmggsuyKJFizJ8+PDccsstGTp0aJJk0aJFWbBgQSdXCQAArFcoFovFzi6iK6utrU11dXVqampSVVXV2eUAAECX4Dp58/nsAACgqZZeJ9soHAAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASkJ5ZxcAAAAA0Bnq1tTlkbueyIqXX0nffltnrwP3SEXPis4uCwDYCKEGAAAA0K3UranLtdNmZubFv03t0hUN7VX9+2bimYdm0tSJwg0A6KKEGgAAAEC3UbemLl8+4pt56A+PpFgsNnqsdumKXHXB9fnfe5/I124+V7ABAF2QPTUAAACAbuPaaTM3GGisVywW89AfHsm102Z2bGEAQIsINQAAAIBuoW5NXWZe/NtmA431isViZv7gt6lbU9dBlQEALSXUAAAAALqFR+56otEeGhtTu2RFHr37iXauCABoLaEGAAAA0C2sePmVdu0PALQ/oQYAAADQLfTtt3W79gcA2p9QAwAAAOgW9jpwj1T179uivlUD+mb42/do54oAgNYSagAAAADdQkXPikw889AUCoWN9isUCpl4xqGp6FnRQZUBAC0l1AAAAAC6jUlTJ2bUe/ZqNtgoFAoZ9Z69MmnqxI4tDABoEaEGAAAA0G1U9KzI124+N5O/clSTpaiq+vfN5K8cla/dfK5ZGgDQRZV3dgEAAAAAHamiZ0Umn3dUJk2dmEfvfiIrXn4lffttneFv30OYAQBdXMnN1Jg+fXqGDRuWysrKjB49OnfddddG+69evTpf/OIXM3To0PTq1Su77bZbLr/88g6qFgAAAOiqKnpWZOS798o7PjI+I9+9l0ADAEpASc3UuO6663L22Wdn+vTpOeCAA/KjH/0ohx56aB577LEMGTJkg8ccffTReeGFF3LZZZdl9913z4svvpi1a9d2cOUAAAAAAMAbVSgWi8XOLqKlxo0bl1GjRuWSSy5paNtzzz0zceLETJs2rUn/W2+9NZMmTcpTTz2Vfv36bdZr1tbWprq6OjU1Namqqtrs2gEAYEviOnnz+ewAAKCpll4nl8zyU2vWrMncuXMzYcKERu0TJkzIvffeu8FjfvWrX2XMmDH51re+lZ122ilvfvOb85nPfCavvfZaR5QMAAAAAAC0oZJZfmrJkiWpr6/PwIEDG7UPHDgwixcv3uAxTz31VO6+++5UVlbmpptuypIlS3L66afn5ZdfbnZfjdWrV2f16tUN92tra9vuTQAAAAAAAJutZGZqrFcoFBrdLxaLTdrWW7duXQqFQq6++uqMHTs273//+3PhhRfmyiuvbHa2xrRp01JdXd1wGzx4cJu/BwAAAAAAoPVKJtQYMGBAysrKmszKePHFF5vM3lhvxx13zE477ZTq6uqGtj333DPFYjHPPffcBo+ZOnVqampqGm7PPvts270JAAAAAABgs5VMqNGzZ8+MHj06s2bNatQ+a9as7L///hs85oADDsjChQvzyiuvNLT95S9/SY8ePbLzzjtv8JhevXqlqqqq0Q0AAAAAAOh8JRNqJMk555yTH//4x7n88svz+OOP59Of/nQWLFiQU089NcnrsyymTJnS0P/YY49N//79c8IJJ+Sxxx7LnXfemc9+9rM58cQT07t37856GwAAQBcyffr0DBs2LJWVlRk9enTuuuuuZvvefffdOeCAA9K/f//07t07e+yxR7773e92YLUAANC9lcxG4UlyzDHHZOnSpbnggguyaNGiDB8+PLfcckuGDh2aJFm0aFEWLFjQ0H/rrbfOrFmz8v/+3//LmDFj0r9//xx99NH5+te/3llvAQAA6EKuu+66nH322Zk+fXoOOOCA/OhHP8qhhx6axx57LEOGDGnSv0+fPjnzzDOz9957p0+fPrn77rtzyimnpE+fPvnkJz/ZCe8AAAC6l0KxWCx2dhFdWW1tbaqrq1NTU2MpKgAA+D9bynXyuHHjMmrUqFxyySUNbXvuuWcmTpyYadOmteg5PvShD6VPnz656qqrWtR/S/nsAACgLbX0Ormklp8CAABoK2vWrMncuXMzYcKERu0TJkzIvffe26LnmDdvXu69994cdNBBzfZZvXp1amtrG90AAIDNI9QAAAC6pSVLlqS+vj4DBw5s1D5w4MAsXrx4o8fuvPPO6dWrV8aMGZMzzjgjJ598crN9p02blurq6obb4MGD26R+AADojoQaAABAt1YoFBrdLxaLTdr+1V133ZUHH3wwP/zhD3PRRRflmmuuabbv1KlTU1NT03B79tln26RuAADojkpqo3AAAIC2MmDAgJSVlTWZlfHiiy82mb3xr4YNG5Yk2WuvvfLCCy/k/PPPz0c/+tEN9u3Vq1d69erVNkUDAEA3Z6YGAADQLfXs2TOjR4/OrFmzGrXPmjUr+++/f4ufp1gsZvXq1W1dHgAAsAFmagAAAN3WOeeck8mTJ2fMmDEZP358Lr300ixYsCCnnnpqkteXjnr++eczY8aMJMkPfvCDDBkyJHvssUeS5O677863v/3t/L//9/867T0AAEB3ItQAAAC6rWOOOSZLly7NBRdckEWLFmX48OG55ZZbMnTo0CTJokWLsmDBgob+69aty9SpU/P000+nvLw8u+22W775zW/mlFNO6ay3AAAA3UqhWCwWO7uIrqy2tjbV1dWpqalJVVVVZ5cDAABdguvkzeezAwCAplp6nWxPDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKQnlnFwAAALRO3dr6PPTk86l9dVWq+lRm1O47paK8rLPLAgAAaHdCDQAAKBF1a+tz+a0P5No/zk/Nq6sa2qv7VGbSO/fJiYeMFW4AAABbNKEGAACUgLq19fnU9Jm5/4kFKRYbP1bz6qpcest9efiphfne6RMFGwAAwBbLnhoAAFACLr/1gQ0GGusVi8n9TyzI5bc+0LGFAQAAdCChBgAAdHF1a+tz7R/nNxtorFcsJtfd8XDq1tZ3TGEAAAAdTKgBAABd3ENPPt9oD42NWf7Ka5n35PPtXBEAAEDnEGoAAEAXV9vCQKOh/8rW9QcAACgVQg0AAOjiqvpUtq7/Vq3rDwAAUCqEGgAA0MWN2n2nVLcw2Nhm694ZuftO7VwRAABA5xBqAABAF1dRXpZJ79wnhcLG+xUKyTEHjUhFeVnHFAYAANDBhBoAAFACTjxkbMbtMaTZYKNQSMbtMSQnHjK2YwsDAADoQEINAAAoARXlZfne6RPzyffv12Qpquo+lfnk+/fL906faJYGAACwRSvv7AIAAICWqSgvyymHj8+Jh4zNvCefT+3KVanaqjIjd99JmAEAAHQLQg0AACgxFeVlGbvHkM4uAwAAoMNZfgoAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJJRcqDF9+vQMGzYslZWVGT16dO66664WHXfPPfekvLw8++yzT/sWCAAAAAAAtIuSCjWuu+66nH322fniF7+YefPm5cADD8yhhx6aBQsWbPS4mpqaTJkyJQcffHAHVQoAAAAAALS1kgo1Lrzwwpx00kk5+eSTs+eee+aiiy7K4MGDc8kll2z0uFNOOSXHHntsxo8f30GVAgAAAAAAba1kQo01a9Zk7ty5mTBhQqP2CRMm5N577232uCuuuCJ/+9vfct5557XodVavXp3a2tpGNwAAAAAAoPOVTKixZMmS1NfXZ+DAgY3aBw4cmMWLF2/wmL/+9a8599xzc/XVV6e8vLxFrzNt2rRUV1c33AYPHvyGawcAAAAAAN64kgk11isUCo3uF4vFJm1JUl9fn2OPPTZf/epX8+Y3v7nFzz916tTU1NQ03J599tk3XDMAANB1TZ8+PcOGDUtlZWVGjx6du+66q9m+N954Y9773vdmu+22S1VVVcaPH5/f/e53HVgtAAB0byUTagwYMCBlZWVNZmW8+OKLTWZvJMmKFSvy4IMP5swzz0x5eXnKy8tzwQUX5OGHH055eXluu+22Db5Or169UlVV1egGAABsma677rqcffbZ+eIXv5h58+blwAMPzKGHHpoFCxZssP+dd96Z9773vbnlllsyd+7cvOtd78oRRxyRefPmdXDlAADQPRWKxWKxs4toqXHjxmX06NGZPn16Q9tb3/rWfPCDH8y0adMa9V23bl0ee+yxRm3Tp0/PbbfdlhtuuCHDhg1Lnz59NvmatbW1qa6uTk1NjYADAAD+z5ZynTxu3LiMGjUql1xySUPbnnvumYkTJzYZYzTnbW97W4455ph85StfaVH/LeWzAwCAttTS6+SWbTTRRZxzzjmZPHlyxowZk/Hjx+fSSy/NggULcuqppyZ5femo559/PjNmzEiPHj0yfPjwRsdvv/32qaysbNIOAAB0P2vWrMncuXNz7rnnNmqfMGFC7r333hY9x7p167JixYr069ev2T6rV6/O6tWrG+7X1tZuXsEAAEBphRrHHHNMli5dmgsuuCCLFi3K8OHDc8stt2To0KFJkkWLFjU7TRwAAOCfLVmyJPX19U2Wsx04cGCTZW+b853vfCevvvpqjj766Gb7TJs2LV/96lffUK0AAMDrSmr5qc5gajgAADS1JVwnL1y4MDvttFPuvffejB8/vqH9G9/4Rq666qo88cQTGz3+mmuuycknn5xf/vKXec973tNsvw3N1Bg8eHBJf3YAANDWtsjlpwAAANrKgAEDUlZW1mRWxosvvthk9sa/uu6663LSSSfl+uuv32igkSS9evVKr1693nC9AABA0qOzCwAAAOgMPXv2zOjRozNr1qxG7bNmzcr+++/f7HHXXHNNjj/++PzsZz/LYYcd1t5lAgAA/8RMDQAAoNs655xzMnny5IwZMybjx4/PpZdemgULFuTUU09NkkydOjXPP/98ZsyYkeT1QGPKlCn53ve+l/32269hlkfv3r1TXV3dae8DAAC6C6EGAADQbR1zzDFZunRpLrjggixatCjDhw/PLbfckqFDhyZJFi1alAULFjT0/9GPfpS1a9fmjDPOyBlnnNHQftxxx+XKK6/s6PIBAKDbsVH4JmwJGyACAEBbc528+Xx2AADQVEuvk+2pAQAAAAAAlAShBgAAAAAAUBKEGgAAAAAAQEkQagAAAAAAACVBqAEAAAAAAJQEoQYAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlQagBAAAAAACUBKEGAAAAAABQEoQaAAAAAABASRBqAAAAAAAAJUGoAQAAAAAAlAShBgAAAAAAUBKEGgAAAAAAQEkQagAAAAAAACVBqAEAAAAAAJQEoQYAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlQagBAAAAAACUBKEGAAAAAABQEoQaAAAAAABASRBqAAAAAAAAJUGoAQAAAAAAlAShBgAAAAAAUBKEGgAAAAAAQEkQagAAAAAAACVBqAEAAAAAAJQEoQYAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlYbNCjXXr1jXbvmDBgjdUEAAA0L0Ui8U8/fTTWbt2bZJkzZo1ue666zJjxowsWbKkk6sDAAC6klaFGrW1tTn66KPTp0+fDBw4MOedd17q6+sbHn/ppZcybNiwNi8SAADYMv35z3/OsGHDsvvuu2fPPffM008/nf333z8nnXRSTjvttOy5557561//2tllAgAAXUSrQo0vf/nLefjhh3PVVVflG9/4Rn7yk5/kgx/8YNasWdPQp1gstnmRAADAlunzn/98RowYkfnz5+fwww/P4Ycfnp133jnLli3LsmXLcsABB+SCCy7o7DIBAIAuolBsRQoxdOjQ/OQnP8k73/nOJMnSpUtz2GGHpbq6Or/61a+yfPnyDBo0qNHsjVJXW1ub6urq1NTUpKqqqrPLAQCALqGtrpO33377/P73v88+++yTV199NX379s2dd96Zt7/97UmS2bNnZ9KkSXnmmWfaqvROZ4wBAABNtfQ6uVUzNZYsWZKhQ4c23O/fv39mzZqVFStW5P3vf39Wrly5+RUDAADdziuvvJJ+/folSfr06ZM+ffpkxx13bHh85513zgsvvNBZ5QEAAF1Mq0KNwYMH5/HHH2/U1rdv3/z+97/Pa6+9liOPPLJNiwMAALZsgwYNyoIFCxruf+tb38r222/fcP+ll17Ktttu2xmlAQAAXVCrQo0JEybkiiuuaNK+9dZb53e/+10qKyvbrDAAAGDL9573vCdPPPFEw/3TTjstffv2bbj/+9//PqNGjeqM0gAAgC6oVXtqLFu2LAsXLszb3va2DT7+yiuvZO7cuTnooIParMDOZr1bAABoqqOuk59++ulUVlY2WpKq1BljAABAU+2yp8a2227bbKCRJKtXr868efNa85QAAADNqqqqynXXXdfZZQAAAF1Eq0KNDSkWi/nd736Xo48+OoMGDco3vvGNtqgLAADopowxAACA5mx2qPH3v/89X/nKVzJ06NC8//3vT2VlZX7zm99k8eLFbVkfAADQTRhjAAAAm9KqUGP16tW55pprcvDBB2fPPffMo48+mgsvvDA9evTIueeem/e85z0pKytrr1oBAIAtjDEGAADQGuWt6bzTTjvlrW99az7+8Y/nhhtuyLbbbpsk+ehHP9ouxQEAAFs2YwwAAKA1WjVTo76+PoVCIYVCwa+lAACAN8wYAwAAaI1WhRqLFi3KJz/5yVxzzTXZYYcd8uEPfzg33XRTCoVCe9UHAABswYwxAACA1mhVqFFZWZmPfexjue222/LII49kzz33zFlnnZW1a9fmG9/4RmbNmpX6+vr2qhUAANjCGGMAAACt0apQ45/ttttu+frXv55nnnkmv/nNb7J69eocfvjh2X777duyPgAAoJswxgAAADalVRuFb0iPHj1y6KGH5tBDD82SJUsyY8aMtqgLAADopowxAACA5rRqpsayZcvy/e9/P7W1tU0eq6mpyTXXXJOTTz65zYoDAAC2bMYYAABAa7Qq1Lj44otz5513pqqqqslj1dXVueuuu3LxxRe3WXEAAMCWzRgDAABojVaFGr/4xS9y6qmnNvv4Kaeckuuvv/4NF8U/1K2py0P/80juuH52HvqfR1K3pq6zSwIAgDZjjAEAALRGq0KNv/3tb3nTm97U7ONvetOb8re//e0NF7Ux06dPz7Bhw1JZWZnRo0fnrrvuarbvjTfemPe+973ZbrvtUlVVlfHjx+d3v/tdu9bXVurW1OWqr16fSTudks+/94J8/ZgL8/n3XpBJO52Sq756vXADAIAtQlcYYwAAAKWjVaFGWVlZFi5c2OzjCxcuTI8erXrKVrnuuuty9tln54tf/GLmzZuXAw88MIceemgWLFiwwf533nln3vve9+aWW27J3Llz8653vStHHHFE5s2b1241toW6NXX58hHfzFUXXJ/al19JqrZO+lUnVVun9uVXctUF1+fLR3xTsAEAQMnr7DEGAABQWlo1Ohg5cmRmzpzZ7OM33XRTRo4c+UZrataFF16Yk046KSeffHL23HPPXHTRRRk8eHAuueSSDfa/6KKL8rnPfS777rtv3vSmN+Xf//3f86Y3vSk333xzu9XYFq6dNjNz/+fRFAdtn4x6a/LW3ZI37/L6/456a4qDts/c/3k0106b2dmlAgDAG9LZYwwAAKC0tCrUOPPMM/Od73wnF198cerr6xva6+vr8/3vfz/f/e53c8YZZ7R5kUmyZs2azJ07NxMmTGjUPmHChNx7770teo5169ZlxYoV6devX3uU2Cbq1tTlpot/m7x5aLLzwKSivHGHivLX2988NDf94LdmawAAUNI6c4yxXmuWuF20aFGOPfbYvOUtb0mPHj1y9tlnt2ttAABAY60KNT784Q/nc5/7XM4666z069cvI0eOzKhRo9KvX7+cffbZOeecc/KRj3ykXQpdsmRJ6uvrM3DgwEbtAwcOzOLFi1v0HN/5znfy6quv5uijj262z+rVq1NbW9vo1pEeueuJrKjcKqnumxQKG+5UKCTVfbOi11Z59O4nOrQ+AABoS505xkhav8Tt6tWrs9122+WLX/xiRowY0W51AQAAG9bqxWm/8Y1v5P77788JJ5yQQYMGZYcddsgJJ5yQ2bNn55vf/GZ71NhI4V/+ob9YLDZp25Brrrkm559/fq677rpsv/32zfabNm1aqqurG26DBw9+wzW3xvKXapMdBjQfaKxXKCQ7DHi9PwAAlLDOHGO0donbXXbZJd/73vcyZcqUVFdXt2ttAABAU+Wb7vIPK1euzGc/+9nMnDkzdXV1Ofjgg/P9738/AwYMaK/6GgwYMCBlZWVNZmW8+OKLTWZv/KvrrrsuJ510Uq6//vq85z3v2WjfqVOn5pxzzmm4X1tb26HBxpLa1U2XnGpORfnr/QEAoER15hhj/RK35557bqP21ixx2xKrV6/O6tX/uG7v6NngAACwJWnVTI3zzjsvV155ZQ477LB89KMfzR/+8Iecdtpp7VVbIz179szo0aMza9asRu2zZs3K/vvv3+xx11xzTY4//vj87Gc/y2GHHbbJ1+nVq1eqqqoa3TrSgF22a13/oa3rDwAAXUlnjjHaYonblujs2eAAALAladVMjRtvvDGXXXZZJk2alCT52Mc+lgMOOCD19fUpKytrlwL/2TnnnJPJkydnzJgxGT9+fC699NIsWLAgp556apLXZ1k8//zzmTFjRpLXA40pU6bke9/7Xvbbb7+GgUnv3r277FTxbfpt3a79AQCgK+nsMUay+UvctlRnzwYHAIAtSatCjWeffTYHHnhgw/2xY8emvLw8Cxcu7JCL8mOOOSZLly7NBRdckEWLFmX48OG55ZZbMnTo0CTJokWLGm3o96Mf/Shr167NGWeckTPOOKOh/bjjjsuVV17Z7vVujr1GDEnfqt5ZUfvaJvtW9l2XPffesQOqAgCA9tGZY4w3ssRta/Tq1Su9evVqs+cDAIDurFXLT9XX16dnz56N2srLy7N27do2LWpjTj/99Pz973/P6tWrM3fu3LzjHe9oeOzKK6/MH//4x4b7f/zjH1MsFpvcumqgkSQVFWU58qh9k039MKxQzJB3L8j82p93SF0AANAeOnOMsblL3AIAAJ2nVTM1isVijj/++Ea/Mlq1alVOPfXU9OnTp6HtxhtvbLsKu6GjJo/N7+77dV54rG9S3EC6UShmwFuXZffDnsnDy36Zfft/NGWFio4vFAAA3qDOHmO0donbJJk/f36S5JVXXslLL72U+fPnp2fPnnnrW9/aLjUCAAD/0KpQ47jjjmvS9vGPf7zNiuF1L9Y9ltGfmpcnfzMkT//Pzql75R+BRcXWdRl28HPZ/bAF6VFezKr6mixc+WgG9xnZiRUDAMDm6ewxRmuXuE2SkSP/ce09d+7c/OxnP8vQoUPz97//vcPqBgCA7qpQLBaLnV1EV1ZbW5vq6urU1NSkqqqqQ17zr7V35JaFX0uSrFtbyMt/rc6aVyvSs09d+r2pJj3KG/+RvX/QV/Kmqnds6KkAAKBddMZ18pbCZwcAAE219Dq5VTM16Bi9yvo2/P8e5cUM2HP5RvtX/lN/AAAAAABojbq19XnoyedT++qqVPWpzKjdd0pFeVlnl7VBQo0uaKet9kplj6qsWle7yb6VZdUZtNXwDqgKAAAAAIAtSd3a+lx+6wO59o/zU/Pqqob26j6VmfTOfXLiIWO7XLjRo7MLoKmyQkVG9JuYZAObhDdSyIhtP2iTcAAAAAAAWqVubX0+NX1mLr3lvkaBRpLUvLoql95yXz41fWbq1tZ3UoUbJtToovbt/9EM2WpUmg82Chmy1ajs2/+jHVkWAAAAAABbgMtvfSD3P7Egze26XSwm9z+xIJff+kDHFrYJQo0uqqxQkQ8M/nrGDZicyrLGm6JUllVl3IDJ+cDgr5ulAQAAAABAq9Strc+1f5zfbKCxXrGYXHfHw11qtoY9NbqwskJF9hswJfv2/2gWrnw0q+pXpLKsbwZtNVyYAQAAAADAZnnoyeebLDnVnOWvvJZ5Tz6fsXsMaeeqWkaoUQLKChUZ3GdkZ5cBAAAAAMAWoLaFgUZD/5Wt69+eLD8FAAAAAADdSFWfytb136p1/duTUAMAAAAAALqRUbvvlOoWBhvbbN07I3ffqZ0rajmhBgAAAAAAdCMV5WWZ9M59UihsvF+hkBxz0IhUlJd1TGEtINQAAAAAAIBu5sRDxmbcHkOaDTYKhWTcHkNy4iFjO7awTRBqAAAAAABAN1NRXpbvnT4xn3z/fk2WoqruU5lPvn+/fO/0iV1qlkaSlHd2AQAAAAAAQMerKC/LKYePz4mHjM28J59P7cpVqdqqMiN336nLhRnrCTUAAAAAAP5F3dr6PPTk86l9dVWq+lRmVBf+R154oyrKyzJ2jyGdXUaLCDUAAAAAAP5P3dr6XH7rA7n2j/NT8+qqhvbqPpWZ9M59cuIhY4Ub0ImEGgAAAAAAeT3Q+NT0mbn/iQUpFhs/VvPqqlx6y315+KmFXXKfAegubBQOAAAAAJDk8lsf2GCgsV6xmNz/xIJcfusDHVsY0ECoAQAAAAB0e3Vr63PtH+c3G2isVywm193xcOrW1ndMYUAjQg0AAAAAoNt76MnnG+2hsTHLX3kt8558vp0rAjZEqAEAAAAAdHu1LQw0GvqvbF1/oG0INQAAAACAbq+qT2Xr+m/Vuv5A2xBqAAAAAADd3qjdd0p1C4ONbbbunZG779TOFQEbItQAAAAAALq9ivKyTHrnPikUNt6vUEiOOWhEKsrLOqYwoBGhBgAAAABAkhMPGZtxewxpNtgoFJJxewzJiYeM7djCgAZCDQAAAACAvD5b43unT8wn379fk6WoqvtU5pPv3y/fO32iWRrQico7uwAAAAAAgK6iorwspxw+PiceMjbznnw+tStXpWqryozcfSdhBnQBQg0AAAAAgH9RUV6WsXsM6ewygH9h+SkAAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAklDe2QUAAABAV7amvj5zFj6XmlWrUl1ZmX0H7ZyeZWWdXRYAQLck1AAAAIANWFNfn0sevD8zHp6XZatWNbRvW1mZKSNG5rQx44QbAAAdTKgBAAAA/2JNfX1Ovvmm3LPgmRT/5bFlq1blv+6fnbmLFubHRxwp2AAA6ED21AAAAIB/ccmD928w0FivmOSeBc/kkgfv78iyAAC6PaEGAAAA/JM19fWZ8fC8ZgON9YpJZjw8P2vq6zuiLAAAItQAAACARuYsfK7RHhobs2zVa3lw4fPtXBEAAOsJNQAAAOCf1LQw0FhveSv7AwCw+YQaAAAA8E+qKytb1X+bVvYHAGDzCTUAAADgn+w7aOds28KgYtvK3hkzaKd2rggAgPWEGgAAAPBPepaVZcqIkSlsol8hyZQR+6RnWVlHlAUAQIQaAAAA0MRpY8blgCFDmw02CkkOGDI0p40Z15FlAQB0eyUXakyfPj3Dhg1LZWVlRo8enbvuumuj/e+4446MHj06lZWV2XXXXfPDH/6wgyoFAABKgTEGG9KzrCw/PuLInDVufJOlqLatrMxZ48bnx0ccaZYGAEAHK+/sAlrjuuuuy9lnn53p06fngAMOyI9+9KMceuiheeyxxzJkyJAm/Z9++um8//3vzyc+8Yn89Kc/zT333JPTTz892223XT784Q93wjsAAAC6EmMMNqZnWVk+NW7/nDZmXB5c+HyWr1qVbSorM2bQTsIMAIBOUigWi8XOLqKlxo0bl1GjRuWSSy5paNtzzz0zceLETJs2rUn/z3/+8/nVr36Vxx9/vKHt1FNPzcMPP5zZs2e36DVra2tTXV2dmpqaVFVVvfE30Y3VranLI3c9kRUvv5K+/bbOXgfukYqeFZ1dFgAAm2FLuU42xgAAgK6hpdfJJTNTY82aNZk7d27OPffcRu0TJkzIvffeu8FjZs+enQkTJjRqe9/73pfLLrssdXV1qaho+g/qq1evzurVqxvu19bWtkH13VvdmrpcO21mZl7829QuXdHQXtW/byaeeWgmTZ0o3AAAoMN11BgDAABoOyWzp8aSJUtSX1+fgQMHNmofOHBgFi9evMFjFi9evMH+a9euzZIlSzZ4zLRp01JdXd1wGzx4cNu8gW6qbk1dvnzEN3PVBdc3CjSSpHbpilx1wfX58hHfTN2auk6qEACA7qqjxhirV69ObW1toxsAALB5SibUWK9QKDS6XywWm7Rtqv+G2tebOnVqampqGm7PPvvsG6y4e7t22sw89IdH0twqZ8ViMQ/94ZFcO21mxxYGAAD/p73HGH44BQB0F3Xr1uaBpU/mD4sfyQNLn0zdurWdXRJboJJZfmrAgAEpKytr8oupF198sckvpdbbYYcdNti/vLw8/fv33+AxvXr1Sq9evdqm6G6ubk1dZl7822YDjfWKxWJm/uC3lqECAKBDddQYY+rUqTnnnHMa7tfW1go2AIAtSt26tbnyqTvy82dmp6ZuZUN7dcVWOXro+By/60Gp6FEy/xRNF1cyMzV69uyZ0aNHZ9asWY3aZ82alf3333+Dx4wfP75J/9///vcZM2aMtW47wCN3PdFkyanm1C5ZkUfvfqKdKwIAgH/oqDFGr169UlVV1egGALClqFu3NufMnZEfP3lbo0AjSWrqVubHT96Wc+bOMGuDNlMyoUaSnHPOOfnxj3+cyy+/PI8//ng+/elPZ8GCBTn11FOTvP4LqClTpjT0P/XUU/PMM8/knHPOyeOPP57LL788l112WT7zmc901lvoVla8/Eq79gcAgDfKGAMA4I258qk78sDSv6WYZpafTzEPLP1brnzqjg6ujC1VSc35OeaYY7J06dJccMEFWbRoUYYPH55bbrklQ4cOTZIsWrQoCxYsaOg/bNiw3HLLLfn0pz+dH/zgBxk0aFD+67/+Kx/+8Ic76y10K337bd2u/Zuzpr4+cxY+l5pVq1JdWZl9B+2cnmVlbfLcAABsWYwxAAA2X926tfn5M7ObDTTWK6aY65+ZbRkq2kShuKkND7q52traVFdXp6amxjTxVqpbU5dJO53SoiWoqgb0zbXP/egN7amxpr4+lzx4f2Y8PC/LVq1qaN+2sjJTRozMaWPGCTcAANqI6+TN57MDALYUDyx9MmfOubzF/X+w70nZt/9u7VgRpayl18kltfwUpaWiZ0UmnnloCoXCRvsVCoVMPOPQNxxonHzzTfmv+2c3CjSSZNmqVfmv+2fn5Jtvypr6+s1+DQAAAADgH2rrXmtl/5Wb7gSbINSgXU2aOjGj3rNXs8FGoVDIqPfslUlTJ76h17nkwftzz4Jnmp3oVkxyz4JncsmD97+h1wEAAAAAXldV0buV/bdqp0roToQatKuKnhX52s3nZvJXjkpV/76NHqvq3zeTv3JUvnbzuW94lsaMh+dtYuW+14ONGQ/PN1sDAAAAANrAyG13SXULg4ptKrbKPtsObeeK6A7sykK7q+hZkcnnHZVJUyfm0bufyIqXX0nffltn+Nv3eENhxnpzFj7XZMmp5ixb9VoeXPh89h885A2/LgAAAAB0ZxU9ynP00PH58ZO3bXSz8EIKOWroeJuE0yacRXSYip4VGfnuvdr8eWtaGGist7yV/QEAAACADTt+14Pyp2XP5IGlf9tgsFFIIWP775bjdz2oE6pjS2T5KUpedWVlq/pv08r+AAAAAMCGVfQoz4Wjp+Tk3d/dZCmq6oqtcvLu786Fo6eYpUGbcSZR8vYdtHO2raxs0RJU21b2zphBO3VAVQAAAADQPVT0KM8ndj84x+96UOYveya1dStT9X97aAgzaGvOKEpez7KyTBkxMv91/+yNbhZeSDJlxD7pWVbWUaUBAAAAQLdR0aM8+/bfrbPLYAtn+Sm2CKeNGZcDhgxNoZnHC0kOGDI0p40Z15FlAQAAAADQhoQabBF6lpXlx0ccmbPGjc+2/7JnxraVlTlr3Pj8+IgjzdIAAAAAAChhlp9ii9GzrCyfGrd/ThszLg8ufD7LV63KNpWVGTNoJ2EGAAAAAMAWQKjBFqdnWVn2Hzyks8sAAAAAAKCNWX4KAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJJQ3tkFAAAAwOaoW1OXR+56IitefiV9+22dvQ7cIxU9Kzq7LAAA2pFQAwAAgJJSt6Yu106bmZkX/za1S1c0tFf175uJZx6aSVMnCjcAALZQQg0AAABKRt2aunz5iG/moT88kmKx2Oix2qUrctUF1+d/730iX7v5XMEGAMAWyJ4aAAAAlIxrp83cYKCxXrFYzEN/eCTXTpvZsYUBANAhhBoAAACUhLo1dZl58W+bDTTWKxaLmfmD36ZuTV0HVQYAQEcRagAAtNCa+vrc8+wzueWvf849zz6TNfX1nV0SQLfyyF1PNNpDY2Nql6zIo3c/0c4VAQDQ0eypAQCwCWvq63PJg/dnxsPzsmzVqob2bSsrM2XEyJw2Zlx6lpV1YoUA3cOKl19p1/4AAHR9Qg0AgI1YU1+fk2++KfcseCb/utjJslWr8l/3z87cRQvz4yOOFGwAtLO+/bZu1/4AAHR9lp8CANiISx68f4OBxnrFJPcseCaXPHh/R5YF0C3tdeAeqerft0V9qwb0zfC379HOFQEA0NGEGgAAzVhTX58ZD89rNtBYr5hkxsPz7bEB0M4qelZk4pmHplAobLRfoVDIxDMOTUXPig6qDACAjiLUAABoxpyFzzXaQ2Njlq16LQ8ufL6dKwJg0tSJGfWevZoNNgqFQka9Z69MmjqxYwsDAKBDCDUAAJpR08JAY73lrewPQOtV9KzI124+N5O/clSTpaiq+vfN5K8cla/dfK5ZGgAAWygbhQMANKO6srJV/bdpZX+AjlZXV5/5Dz6VR+//a2pXLczQPXpl74P2yNBt9klZoXRCgIqeFZl83lGZNHViHr37iax4+ZX07bd1hr99D2EGAMAWTqgBANCMfQftnG0rK1u0BNW2lb0zZtBOHVAVQOvV1dXn6ivuzA1Xz87qunX/eOAXa9Pjs/flLe97IUd/4fCM3/FjJRdujHz3Xp1dBgAAHcjyUwAAzehZVpYpI0Zm49vRJoUkU0bsk55lZR1RFkCr1NXV54v/dk2uvvLuxoFGklSUZ92AHfL4bbvm4qN+mZv+9sXUF+s6p1AAAGgBoQYAwEacNmZcDhgytNlgo5DkgCFDc9qYcR1ZFkCLXTvjnsx78O9Jc99khUJS3TdLnx2U27/3WOYsvaYjywMAgFYRagAAbETPsrL8+Igjc9a48dn2X/bM2LayMmeNG58fH3GkWRpAl1RXV5+brp+z6Y6FQjJwuzz9s8rMe+GXZmsAANBl2VMDAGATepaV5VPj9s9pY8blwYXPZ/mqVdmmsjJjBu0kzAC6tEceXpAVta+1rHNFeerq+2bhnEVZuMujGdxnZPsWBwAAm0GoAQDQQj3LyrL/4CGdXQZAi7U40FivvCxragpZVb+ifQoCAIA3yPJTAAAAW6i+Vb1bd8Da+vSsLqayrG/7FAQAAG+QUAMAAGALtdeIIS0PNurWpqJsRQbt2yeDthrevoUBAMBmEmoAAABsoSoqynLkUftuumOxmLzwUoYduyojB34wZYWK9i8OAAA2g1ADAABgCzZpygEZOWaXJMUNdygWk5oV6T94Yd71qbdm3/4f7cjyAACgVYQaAAAAW7CKirJ84zsfzceOf3t6VfzLELBubXosWZw9D34qZ17/wRy52zfM0gAAoEsr7+wCAAAAaF8VFWU5/pPvysdOeEfmz306/3v/X7Ji1aIMeUuv7H3Q4RmyzQhhBgAAJUGoAQAA0E1UVJRl3/12z7777d7ZpQAAwGax/BQAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlQagBAAAAAACUBKEGAAAAAABQEoQaAAAAAABASRBqAAAAAAAAJUGoAQAAAAAAlAShBgAAAAAAUBKEGgAAAAAAQEkQagAAAAAAACVBqAEAAHRLy5Yty+TJk1NdXZ3q6upMnjw5y5cv3+gxN954Y973vvdlwIABKRQKmT9/fofUCgAAvE6oAQAAdEvHHnts5s+fn1tvvTW33npr5s+fn8mTJ2/0mFdffTUHHHBAvvnNb3ZQlQAAwD8r7+wCAAAAOtrjjz+eW2+9Nffdd1/GjRuXJPnv//7vjB8/Pn/+85/zlre8ZYPHrQ89/v73v3dUqQAAwD8pmZkarZ0aXldXl89//vPZa6+90qdPnwwaNChTpkzJwoULO65oAACgS5o9e3aqq6sbAo0k2W+//VJdXZ177723EysDAAA2pmRCjdZODV+5cmUeeuihfPnLX85DDz2UG2+8MX/5y1/ygQ98oAOrBgAAuqLFixdn++23b9K+/fbbZ/HixW36WqtXr05tbW2jGwAAsHlKYvmpzZkaXl1dnVmzZjVq+/73v5+xY8dmwYIFGTJkSIfUDgAAdJzzzz8/X/3qVzfaZ86cOUmSQqHQ5LFisbjB9jdi2rRpm6wJAABomZIINTY1Nby59W7/VU1NTQqFQrbZZptm+6xevTqrV69uuO9XVAAAUDrOPPPMTJo0aaN9dtlll/zpT3/KCy+80OSxl156KQMHDmzTmqZOnZpzzjmn4X5tbW0GDx7cpq8BAADdRUmEGm0xNXzVqlU599xzc+yxx6aqqqrZfn5FBQAApWvAgAEZMGDAJvuNHz8+NTU1eeCBBzJ27Ngkyf3335+amprsv//+bVpTr1690qtXrzZ9TgAA6K46dU+N888/P4VCYaO3Bx98MMkbmxpeV1eXSZMmZd26dZk+ffpG+06dOjU1NTUNt2effXbz3hwAANBl7bnnnjnkkEPyiU98Ivfdd1/uu+++fOITn8jhhx/eaCb4HnvskZtuuqnh/ssvv5z58+fnscceS5L8+c9/zvz589t8Hw4AoHtYU1+fe559Jrf89c+559lnsqa+vrNLgi6vU2dqdMTU8Lq6uhx99NF5+umnc9ttt210lkbiV1QAANBdXH311TnrrLMyYcKEJMkHPvCBXHzxxY36/PnPf05NTU3D/V/96lc54YQTGu6vH8+cd955Of/889u/aABgi7Cmvj6XPHh/Zjw8L8tWrWpo37ayMlNGjMxpY8alZ1lZJ1YIXVehWCwWO7uITXn88cfz1re+Nffff3+jqeH77bdfnnjiiWb31FgfaPz1r3/N7bffnu22267Vr11bW5vq6urU1NRsMhABgJaoq6vPIw8vyIra19K3qnf2GjEkFRUuVoHS4jp58/nsAKB7W1Nfn5Nvvin3LHgmG/qH2UKSA4YMzY+POFKwQbfS0uvkkthT45+nhv/oRz9Kknzyk5/c4NTwadOm5cgjj8zatWvzkY98JA899FB+/etfp76+vmFKeL9+/dKzZ89OeS8AdF91dfW5dsY9mXnDnNTWvNbQXlXdOxM/sm8mTTlAuAEAALCFu+TB+5sNNJKkmOSeBc/kkgfvz6fGte1eX7Al6NQ9NVrj6quvzl577ZUJEyZkwoQJ2XvvvXPVVVc16vPPU8Ofe+65/OpXv8pzzz2XffbZJzvuuGPD7d577+2MtwBAN1ZXV58vf/baXHX5nY0CjSSprXktV11+Z7782WtTV2f9VAAAgC3Vmvr6zHh4XrOBxnrFJDMenm+PDdiAkpipkbw+u+KnP/3pRvv880pau+yyS0pgZS0AuolrZ9yTh+Y8neb+01QsJg/NeTrXzrgnk096R8cWBwAAQIeYs/C5RntobMyyVa/lwYXPZ//BQ9q5KigtJTNTAwBKVV1dfWbeMKfZQGO9YjGZecMcszUAAAC2UDUtDDTWW97K/tAdCDUAoJ098vCCJktONae25rU8+vCCdq4IAACAzlBdWdmq/tu0sj90B0INAGhnK2pbFmj8o79f4gAAAGyJ9h20c7ZtYVCxbWXvjBm0UztXBKVHqAGUjLo1dXnofx7JHdfPzkP/80jq1tR1dknQIn2rereyv1/iAAAAbIl6lpVlyoiRKWyiXyHJlBH7pGdZWUeUBSWlZDYKB7qvujV1uXbazMy8+LepXbqiob2qf99MPPPQTJo6MRU9KzqxQti4vUYMSVV17xYtQVVV3TvDR9gEDgAAYEt12phxmbtoYe5Z8Ew2tPViIckBQ4bmtDHjOro0KAlmagBdWt2aunz5iG/mqguubxRoJEnt0hW56oLr8+UjvmnWBl1aRUVZJn5k3xQ28VOcQiGZ+JF9U1HhlzgAAABbqp5lZfnxEUfmrHHjmyxFtW1lZc4aNz4/PuJIszSgGWZqAF3atdNm5qE/PJJicUO/XUiKxWIe+sMjuXbazEw+76gOrg5abtKUA/K/jzybh+Y8nQ2dzoVCMmrfYZk05YCOLw4AAIAO1bOsLJ8at39OGzMuDy58PstXrco2lZUZM2gnYQZsgpkaQJdVt6YuMy/+bbOBxnrFYjEzf/BbszXo0ioqyvK1/5yUySe+I1XVjffYqKruncknviNf+89JZmkAAAB0Iz3LyrL/4CF5/5venP0HDxFoQAuYqQF0WY/c9USTJaeaU7tkRR69+4mMfPde7VwVbL6KirJMPukdmTTlgDz68IKsqF2VvlWVGT5iiDADAAAAoAWEGkCXteLlV9q1P3SWioqyjBwzrLPLAAAAACg5lp8Cuqy+/bZu1/4AAAAAQGkRagBd1l4H7pGq/n1b1LdqQN8Mf/se7VwRAAAAANCZhBpAl1XRsyITzzw0hUJho/0KhUImnnFoKnpWdFBlAAAAAEBnEGoAXdqkqRMz6j17NRtsFAqFjHrPXpk0dWLHFgYAAAAAdDihBtClVfSsyNduPjeTv3JUk6Woqvr3zeSvHJWv3XyuWRoAAAAA0A2Ud3YBAJtS0bMik887KpOmTsyjdz+RFS+/kr79ts7wt+8hzAAAAACAbkSoAZSMip4VGfnuvTq7DAAAAACgk1h+CgAAAAAAKAlCDQAAAAAAoCRYfgqADlVXV59HHl6QFbWvpW9V7+w1YkgqKso6uyy2EHVr6/PQk8+n9tVVqepTmVG775SKcucXAABQ+uqLdXl+5SNZXb8ivcr6Zqet9kpZwV6jdD9CDQA6RF1dfa6dcU9m3jAntTWvNbRXVffOxI/sm0lTDhBusNnq1tbn8lsfyLV/nJ+aV1c1tFf3qcykd+6TEw8ZK9wAAABKUn2xLnOWXpOHX56ZVetqG9ore1RlRL+J2bf/R4UbdCtCDQDaXV1dfb782Wvz0JynUyw2fqy25rVcdfmd+d9Hns3X/nOSYINWq1tbn09Nn5n7n1jQ5PyqeXVVLr3lvjz81MJ87/SJgg0AAKCk1Bfr8qtnv5QFKx9K0njAs2pdbe5fclUWrfzffGDw1wUbdBv21ACg3V07454NBhrrFYvJQ3OezrUz7unYwtgiXH7rAxsMNNYrFpP7n1iQy299oGMLAwAAeIPmLL1mg4HGPxSzYOVDmbP0mo4sCzqVUAOAdlVXV5+ZN8xp9h+c1ysWk5k3zEldXX3HFMYWoW5tfa794/wWnV/X3fFw6tY6vwAAgNJQX6zLwy/PTPOBxnrFPLzsl6kv1nVAVdD5hBoAtKtHHl7QaA+NjamteS2PPrygnStiS/LQk8832kNjY5a/8lrmPfl8O1cEAADQNp5f+UijPTQ2ZlV9TRaufLSdK4KuQagBQLtaUduyQOMf/Vv2D9SQJLUtDDQa+q90fgEAAKVhdf2KVvVf1cr+UKqEGgC0q75VvVvZv7KdKmFLVNWndedL1VbOLwAAoDT0Kuvbqv6VrewPpUqoAUC72mvEkFRVtyzYqKruneEjhrRzRWxJRu2+U6pbGGxss3XvjNx9p3auCAAAoG3stNVeqexR1aK+lWXVGbTV8HauCLoGoQYA7aqioiwTP7JvCoWN9ysUkokf2TcVFWUdUxhbhIryskx65z4tOr+OOWhEKsqdXwAAQGkoK1RkRL+JSTYx4EkhI7b9YMoKFR1QFXQ+oQYA7W7SlAMyat9hzf7Dc6GQjNp3WCZNOaBjC2OLcOIhYzNujyEbPb/G7TEkJx4ytmML68Lq1q3NA0ufzB8WP5IHlj6ZunVrO7skAABgA/bt/9EM2WpUmg82Chmy1ajs2/+jHVkWdKpCsVgsdnYRXVltbW2qq6tTU1OTqqqWTfcCoKm6uvpcO+OezLxhTmpr/rF5eFV170z8yL6ZNOUAszTYbHVr63P5rQ/k2j/OT80/bR5e3acyk965T048ZKxZGnk9zLjyqTvy82dmp6ZuZUN7dcVWOXro+By/60Gp6FHeiRVSSlwnbz6fHQDQGvXFusxZek0eXjYzq+prG9ory6oyYtuJ2bf/R83SYIvQ0utkocYmGHAAtK26uvo8+vCCrKhdlb5VlRk+YogwgzZTt7Y+8558PrUrV6Vqq8qM3H0nYcb/qVu3NufMnZEHlv4txTS9/CukkLH9d8uFo6cINmgR18mbz2cHAGyO+mJdFq58NKvqV6SyrG8GbTVcmMEWpaXXyUasAHSoioqyjBwzrLPLYAtVUV6WsXvYbH5DrnzqjmYDjSQpppgHlv4tVz51Rz6x+8EdXB0AALApZYWKDO4zsrPLgE5nTw0AgC1c3bq1+fkzs5sNNNYrppjrn5ltjw0AAAC6LKEGAMAWbt6yvzfaQ2NjltetzPxlz7RzRQAAALB5hBoAAFu42rrXWtm/ZQEIAAAAdDShBgDAFq6qoncr+2/VTpUAAADAGyPUAADYwo3cdpdUtzCo2KZiq+yz7dB2rggAAAA2j1ADAGALV9GjPEcPHZ9CChvtV0ghRw0dn4oe5R1UGQAAALSOUAMAoBs4fteDMrb/bs0GG4UUMrb/bjl+14M6uDIAAABoOaEGAEA3UNGjPBeOnpKTd393k6Woqiu2ysm7vzsXjp5ilgYAAABdmlErAEA3UdGjPJ/Y/eAcv+tBmb/smdTWrUzV/+2hIcwAAACgFBi9AgB0MxU9yrNv/906uwwAAABoNctPAQAAAAAAJUGoAQAAAAAAlAShBgAAAAAAUBKEGgAAAAAAQEkQagAAAAAAACVBqAEAAAAAAJQEoQYAAAAAAFAShBoAAAAAAEBJEGoAAAAAAAAlQagBAAAAAACUBKEGAADQLS1btiyTJ09OdXV1qqurM3ny5CxfvrzZ/nV1dfn85z+fvfbaK3369MmgQYMyZcqULFy4sOOKBgCAbk6oAQAAdEvHHnts5s+fn1tvvTW33npr5s+fn8mTJzfbf+XKlXnooYfy5S9/OQ899FBuvPHG/OUvf8kHPvCBDqwaAAC6t/LOLgAAAKCjPf7447n11ltz3333Zdy4cUmS//7v/8748ePz5z//OW95y1uaHFNdXZ1Zs2Y1avv+97+fsWPHZsGCBRkyZEiH1A4AAN2ZmRoAAEC3M3v27FRXVzcEGkmy3377pbq6Ovfee2+Ln6empiaFQiHbbLNNO1QJAAD8KzM1AACAbmfx4sXZfvvtm7Rvv/32Wbx4cYueY9WqVTn33HNz7LHHpqqqqtl+q1evzurVqxvu19bWtr5gAAAgSQnN1GjtJn7/6pRTTkmhUMhFF13UbjUCAACd6/zzz0+hUNjo7cEHH0ySFAqFJscXi8UNtv+rurq6TJo0KevWrcv06dM32nfatGkN45jq6uoMHjx4894cAABQOjM1jj322Dz33HO59dZbkySf/OQnM3ny5Nx8882bPHbmzJm5//77M2jQoPYuEwAA6ERnnnlmJk2atNE+u+yyS/70pz/lhRdeaPLYSy+9lIEDB270+Lq6uhx99NF5+umnc9ttt210lkaSTJ06Neecc07D/draWsEGAABsppIINTZnE7/1nn/++Zx55pn53e9+l8MOO6yjSgYAADrBgAEDMmDAgE32Gz9+fGpqavLAAw9k7NixSZL7778/NTU12X///Zs9bn2g8de//jW33357+vfvv8nX6tWrV3r16tXyNwEAUGLq1q3NvGV/T23da6mq6J2R2+6Sih4l8U/PlKCSOLM2tYlfc6HGunXrMnny5Hz2s5/N2972to4qFwAA6OL23HPPHHLIIfnEJz6RH/3oR0lenw1++OGHNxpf7LHHHpk2bVqOPPLIrF27Nh/5yEfy0EMP5de//nXq6+sb9t/o169fevbs2SnvBQCgs9StW5srn7ojP39mdmrqVja0V1dslaOHjs/xux4k3KDNlcQZtbmb+P3Hf/xHysvLc9ZZZ7X4tWziBwAA3cPVV1+ds846KxMmTEiSfOADH8jFF1/cqM+f//zn1NTUJEmee+65/OpXv0qS7LPPPo363X777XnnO9/Z7jUDAHQVdevW5py5M/LA0r+lmGKjx2rqVubHT96WPy17JheOniLYoE116tl0/vnn56tf/epG+8yZMydJ6zfxmzt3br73ve/loYceatFGf+tNmzZtkzUBAAClr1+/fvnpT3+60T7F4j8G6Lvsskuj+wAA3dmVT92xwUBjvWKKeWDp33LlU3fkE7sf3MHVsSXr1FCjPTfxu+uuu/Liiy9myJAhDW319fX5t3/7t1x00UX5+9//vsHjbOIHAAAAANC8unVr8/NnZjcbaKxXTDHXPzPbMlS0qU49k9pzE7/JkyfnPe95T6O2973vfZk8eXJOOOGEZl/LJn4AAAAAAM2bt+zvjfbQ2JjldSszf9kz2bf/bu1cFd1FScRjm7OJX//+/dO/f/9Gz1NRUZEddtih2Y3FAQAAAADYuNq611rZv2UBCLRESYQaSes38QMA6Gz1xbo8v/KRrK5fkV5lfbPTVnulrFDR2WUBAGwx6urq88jDC7Ki9rX0reqdvUYMSUVFWWeXBVu8qorerey/VTtVQndUMqFGazfx25Dm9tEAAGhL9cW6zFl6TR5+eWZWrattaK/sUZUR/SZm3/4fFW4AALwBdXX1uXbGPZl5w5zU1vzjF+NV1b0z8SP7ZtKUA4Qb0I5GbrtLqiu2atESVNtUbJV9th3aAVXRXfTo7AIAALYk9cW6/OrZL+X+JVc1CjSSZNW62ty/5Kr86tkvpb5Y10kVAgCUtrq6+nz5s9fmqsvvbBRoJEltzWu56vI78+XPXpu6uvpOqhC2fBU9ynP00PEppLDRfoUUctTQ8TYJp00JNQAA2tCcpddkwcqHkjQ3g7SYBSsfypyl13RkWQAAW4xrZ9yTh+Y8neYW7CgWk4fmPJ1rZ9zTsYVBN3P8rgdlbP/dmg02CilkbP/dcvyuB3VwZWzphBoAAG2kvliXh1+emeYDjfWKeXjZL83WAABopbq6+sy8YU6zgcZ6xWIy84Y5ZmtAO6roUZ4LR0/Jybu/O9X/smdGdcVWOXn3d+fC0VPM0qDNOaMAANrI8ysfabLkVHNW1ddk4cpHM7jPyHauCgBgy/HIwwuaLDnVnNqa1/Lowwsycsywdq4Kuq+KHuX5xO4H5/hdD8r8Zc+ktm5lqv5vDw1hBu3FmQUA0EZW169oVf9VrewPANDdrahtWaDxj/6r2qkS4J9V9CjPvv136+wy6CYsPwUA0EZ6lfVtVf/KVvYHAOju+lb1bmX/ynaqBIDOItQAAGgjO221Vyp7VLWob2VZdQZtNbydKwIA2LLsNWJIqqpbFmxUVffO8BFD2rkiADqaUAMAoI2UFSoyot/EJIVN9CxkxLYfTFmhogOqAgDYclRUlGXiR/ZNYROXW4VCMvEj+6aioqxjCgOgwwg1AADa0L79P5ohW41K88FGIUO2GpV9+3+0I8sCANhiTJpyQEbtO6zZYKNQSEbtOyyTphzQsYUB0CGEGgAAbaisUJEPDP56xg2YnMqyxktRVZZVZdyAyfnA4K+bpQEAsJkqKsrytf+clMknvqPJUlRV1b0z+cR35Gv/OcksDYAtVKFYLBY7u4iurLa2NtXV1ampqUlVVcvWyAYASJL6Yl0Wrnw0q+pXpLKsbwZtNVyYwRbDdfLm89kBtJ26uvo8+vCCrKhdlb5VlRk+YogwA6BEtfQ6ubwDawIA6FbKChUZ3GdkZ5cBALDFqqgoy8gxwzq7DAA6kOWnAAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUAAAAAAICSINQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1AAAAAACAkiDUAAAAAAAASkJ5ZxfQ1RWLxSRJbW1tJ1cCAABdx/rr4/XXy7ScMQYAADTV0jGGUGMTVqxYkSQZPHhwJ1cCAABdz4oVK1JdXd3ZZZQUYwwAAGjepsYYhaKfVm3UunXrsnDhwvTt2zeFQqGzy+lyamtrM3jw4Dz77LOpqqrq7HLoJM4DEucBzgFe5zzoPorFYlasWJFBgwalRw+r2raGMUbzfIeQOA94nfOAxHnA65wH3UdLxxhmamxCjx49svPOO3d2GV1eVVWVLxWcByRxHuAc4HXOg+7BDI3NY4yxab5DSJwHvM55QOI84HXOg+6hJWMMP6kCAAAAAABKglADAAAAAAAoCUIN3pBevXrlvPPOS69evTq7FDqR84DEeYBzgNc5D4A3wncIifOA1zkPSJwHvM55wL+yUTgAAAAAAFASzNQAAAAAAABKglADAAAAAAAoCUINAAAAAACgJAg1aJVvfOMb2X///bPVVltlm222adExxWIx559/fgYNGpTevXvnne98Z/73f/+3fQulXS1btiyTJ09OdXV1qqurM3ny5Cxfvnyjxxx//PEpFAqNbvvtt1/HFEybmD59eoYNG5bKysqMHj06d91110b733HHHRk9enQqKyuz66675oc//GEHVUp7as158Mc//rHJ3/tCoZAnnniiAyumrd1555054ogjMmjQoBQKhcycOXOTx/g+ADbGGIPEGKO7MsYgMcbo7owv2BxCDVplzZo1Oeqoo3Laaae1+JhvfetbufDCC3PxxRdnzpw52WGHHfLe9743K1asaMdKaU/HHnts5s+fn1tvvTW33npr5s+fn8mTJ2/yuEMOOSSLFi1quN1yyy0dUC1t4brrrsvZZ5+dL37xi5k3b14OPPDAHHrooVmwYMEG+z/99NN5//vfnwMPPDDz5s3LF77whZx11ln5xS9+0cGV05Zaex6s9+c//7nR3/03velNHVQx7eHVV1/NiBEjcvHFF7eov+8DYFOMMUiMMbojYwwSYwyML9hMRdgMV1xxRbG6unqT/datW1fcYYcdit/85jcb2latWlWsrq4u/vCHP2zHCmkvjz32WDFJ8b777mtomz17djFJ8Yknnmj2uOOOO674wQ9+sAMqpD2MHTu2eOqppzZq22OPPYrnnnvuBvt/7nOfK+6xxx6N2k455ZTifvvt12410v5aex7cfvvtxSTFZcuWdUB1dIYkxZtuummjfXwfAC1ljNF9GWN0T8YYFIvGGDRmfEFLmalBu3r66aezePHiTJgwoaGtV69eOeigg3Lvvfd2YmVsrtmzZ6e6ujrjxo1raNtvv/1SXV29yT/TP/7xj9l+++3z5je/OZ/4xCfy4osvtne5tIE1a9Zk7ty5jf4eJ8mECROa/TOfPXt2k/7ve9/78uCDD6aurq7daqX9bM55sN7IkSOz44475uCDD87tt9/enmXSBfk+ANqaMcaWxxij+zHGIDHGYPP4LiCx/BTtbPHixUmSgQMHNmofOHBgw2OUlsWLF2f77bdv0r799ttv9M/00EMPzdVXX53bbrst3/nOdzJnzpy8+93vzurVq9uzXNrAkiVLUl9f36q/x4sXL95g/7Vr12bJkiXtVivtZ3POgx133DGXXnppfvGLX+TGG2/MW97ylhx88MG58847O6JkugjfB0BbM8bY8hhjdD/GGCTGGGwe3wUkSXlnF0DnO//88/PVr351o33mzJmTMWPGbPZrFAqFRveLxWKTNjpXS8+DpOmfZ7LpP9Njjjmm4f8PHz48Y8aMydChQ/Ob3/wmH/rQhzazajpSa/8eb6j/htopLa05D97ylrfkLW95S8P98ePH59lnn823v/3tvOMd72jXOulafB9A92OMQWKMwaYZY5AYY9B6vgsQapAzzzwzkyZN2mifXXbZZbOee4cddkjyeoq64447NrS/+OKLTVJVOldLz4M//elPeeGFF5o89tJLL7Xqz3THHXfM0KFD89e//rXVtdKxBgwYkLKysia/lNnY3+Mddthhg/3Ly8vTv3//dquV9rM558GG7LfffvnpT3/a1uXRhfk+gO7JGIPEGIPmGWOQGGOweXwXkAg1yOv/ERkwYEC7PPewYcOyww47ZNasWRk5cmSS19dMvOOOO/If//Ef7fKabJ6Wngfjx49PTU1NHnjggYwdOzZJcv/996empib7779/i19v6dKlefbZZxsNROmaevbsmdGjR2fWrFk58sgjG9pnzZqVD37wgxs8Zvz48bn55psbtf3+97/PmDFjUlFR0a710j425zzYkHnz5vl73834PoDuyRiDxBiD5hljkBhjsHl8F5Ak6Zz9ySlVzzzzTHHevHnFr371q8Wtt966OG/evOK8efOKK1asaOjzlre8pXjjjTc23P/mN79ZrK6uLt54443FRx55pPjRj360uOOOOxZra2s74y3QBg455JDi3nvvXZw9e3Zx9uzZxb322qt4+OGHN+rzz+fBihUriv/2b/9WvPfee4tPP/108fbbby+OHz++uNNOOzkPSsS1115brKioKF522WXFxx57rHj22WcX+/TpU/z73/9eLBaLxXPPPbc4efLkhv5PPfVUcauttip++tOfLj722GPFyy67rFhRUVG84YYbOust0AZaex5897vfLd50003Fv/zlL8VHH320eO655xaTFH/xi1901lugDaxYsaLhv/9JihdeeGFx3rx5xWeeeaZYLPo+AFrPGINi0RijOzLGoFg0xsD4gs0j1KBVjjvuuGKSJrfbb7+9oU+S4hVXXNFwf926dcXzzjuvuMMOOxR79epVfMc73lF85JFHOr542szSpUuLH/vYx4p9+/Yt9u3bt/ixj32suGzZskZ9/vk8WLlyZXHChAnF7bbbrlhRUVEcMmRI8bjjjisuWLCg44tns/3gBz8oDh06tNizZ8/iqFGjinfccUfDY8cdd1zxoIMOatT/j3/8Y3HkyJHFnj17FnfZZZfiJZdc0sEV0x5acx78x3/8R3G33XYrVlZWFrfddtvi29/+9uJvfvObTqiatnT77bdv8FrguOOOKxaLvg+A1jPGoFg0xuiujDEoFo0xujvjCzZHoVj8v51UAAAAAAAAurAenV0AAAAAAABASwg1AAAAAACAkiDUAAAAAAAASoJQAwAAAAAAKAlCDQAAAAAAoCQINQAAAAAAgJIg1AAAAAAAAEqCUAMAAAAAACgJQg0AAAAAAKAkCDUA6DDHH398CoVCCoVCKioqsuuuu+Yzn/lMXn311YY+v/jFL/LOd74z1dXV2XrrrbP33nvnggsuyMsvv9zouV577bVsu+226devX1577bUmr3XppZfmne98Z6qqqlIoFLJ8+fL2fnsAAEAHM8YA6H6EGgB0qEMOOSSLFi3KU089la9//euZPn16PvOZzyRJvvjFL+aYY47Jvvvum9/+9rd59NFH853vfCcPP/xwrrrqqkbP84tf/CLDhw/PW9/61tx4441NXmflypU55JBD8oUvfKFD3hcAANA5jDEAupdCsVgsdnYRAHQPxx9/fJYvX56ZM2c2tH3iE5/Ir3/96/zyl7/MuHHjctFFF+VTn/pUk2OXL1+ebbbZpuH+u971rkyaNCnFYjE///nPc9ttt23wNf/4xz/mXe96V5YtW9boeAAAoPQZYwB0P+WdXQAA3Vvv3r1TV1eXq6++OltvvXVOP/30Dfb758HC3/72t8yePTs33nhjisVizj777Dz11FPZddddO6hqAACgqzLGANiyWX4KgE7zwAMP5Gc/+1kOPvjg/PWvf82uu+6aioqKTR53+eWX59BDD21Y7/aQQw7J5Zdf3gEVAwAAXZkxBsCWT6gBQIf69a9/na233jqVlZUZP3583vGOd+T73/9+isViCoXCJo+vr6/PT37yk3z84x9vaPv4xz+en/zkJ6mvr2/P0gEAgC7IGAOge7H8FAAd6l3velcuueSSVFRUZNCgQQ2/mnrzm9+cu+++O3V1dRv9JdXvfve7PP/88znmmGMatdfX1+f3v/99Dj300HatHwAA6FqMMQC6FzM1AOhQffr0ye67756hQ4c2Glgce+yxeeWVVzJ9+vQNHrd8+fIkyWWXXZZJkyZl/vz5jW4f+9jHctlll3XEWwAAALoQYwyA7sVMDQC6hHHjxuVzn/tc/u3f/i3PP/98jjzyyAwaNChPPvlkfvjDH+btb397jj322Nx888351a9+leHDhzc6/rjjjsthhx2Wl156Kdttt10WL16cxYsX58knn0ySPPLII+nbt2+GDBmSfv36dcZbBAAAOpAxBsCWyUwNALqM//iP/8jPfvaz3H///Xnf+96Xt73tbTnnnHOy995757jjjsuMGTPSp0+fHHzwwf+/fTu2YRiGoSj4M4CG4QIGPIx2Uu8NvKAyQIp0MRjc9QTEknjQx+x5nhlj5LquJMlaK1WVOWeS5DiOVFXu+/7pTgAAwHPcGAD/57X33k8/AgAAAAAA4Bs/NQAAAAAAgBZEDQAAAAAAoAVRAwAAAAAAaEHUAAAAAAAAWhA1AAAAAACAFkQNAAAAAACgBVEDAAAAAABoQdQAAAAAAABaEDUAAAAAAIAWRA0AAAAAAKAFUQMAAAAAAGhB1AAAAAAAAFp4A1nUuQhr+IoKAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1600x600 with 2 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from sklearn.decomposition import PCA\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.metrics import pairwise_distances\n",
    "\n",
    "# Your data\n",
    "data = {\n",
    "    'rsID': ['rs2231142', 'rs4961', 'rs1042713', 'rs7412', 'rs4673993', 'rs71647871', 'rs78769542', 'rs16969968', 'rs12777823', 'rs3745274', 'rs2242480', 'rs2108622', 'rs1801160', 'rs3918290'],\n",
    "    'Alt_Freq_MAF_K&H': [0.102678571, 0.194399351, 0.373782468, 0.034902597, 0.430194805, 0.003652597, 0.003246753, 0.172077922, 0.36525974, 0.27637987, 0.32711039, 0.421266234, 0.043425325, 0.002435065],\n",
    "    'Global': [0.1119, 0.1984, 0.3883, 0.07424, 0.3195, 0.01176, 0.0003074, 0.3016, 0.1905, 0.2564, 0.1533, 0.2875, 0.04675, 0.005061],\n",
    "    'East_Asian': [0.2982, 0.508, 0.5276, 0.0605, 0.2617, 0.00006684, 0.0001561, 0.02676, 0.3119, 0.1905, 0.2521, 0.2667, 0.01862, 0],\n",
    "    'European_non_Finnish': [0.1109, 0.1989, 0.3652, 0.0783, 0.328, 0.01297, 0.00002289, 0.3354, 0.1513, 0.2408, 0.09031, 0.2978, 0.045, 0.004832],\n",
    "    'South_Asian': [0.09018, 0.175, 0.4483, 0.04164, 0.4725, 0.006593, 0.004678, 0.2376, 0.3345, 0.3869, 0.321, 0.3903, 0.09625, 0.003996],\n",
    "    'African_African_American': [0.02677, 0.07598, 0.4935, 0.1056, 0.1208, 0.002414, 0.0002002, 0.05608, 0.2496, 0.3771, 0.7415, 0.09773, 0.02498, 0.0004799]\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Extract rsID and MAF columns\n",
    "rsIDs = df['rsID']\n",
    "maf_columns = ['Alt_Freq_MAF_K&H', 'Global', 'East_Asian', 'European_non_Finnish', 'South_Asian', 'African_African_American']\n",
    "df_maf = df[maf_columns]\n",
    "\n",
    "# Calculate pairwise genetic distances\n",
    "genetic_distances = pairwise_distances(df_maf, metric='euclidean')\n",
    "\n",
    "# Perform PCA on the distance matrix\n",
    "pca = PCA(n_components=3)\n",
    "pca_result = pca.fit_transform(genetic_distances)\n",
    "\n",
    "# Create a DataFrame for visualization\n",
    "pca_df = pd.DataFrame(data=pca_result, columns=['PC1', 'PC2', 'PC3'])\n",
    "\n",
    "# Display the loadings of each rsID for the first few principal components\n",
    "loadings_df = pd.DataFrame(data=pca.components_.T, columns=['PC1', 'PC2', 'PC3'], index=rsIDs)\n",
    "print(\"Loadings of each rsID for the first few principal components:\")\n",
    "print(loadings_df)\n",
    "# Create a color map for MAF columns\n",
    "color_map = {column: plt.cm.get_cmap('viridis')(i / len(maf_columns)) for i, column in enumerate(maf_columns)}\n",
    "\n",
    "# Visualize PCA results\n",
    "fig = plt.figure(figsize=(16, 6))\n",
    "\n",
    "# Plot PCA1 against PCA2\n",
    "ax1 = fig.add_subplot(121)\n",
    "for i, rsID in enumerate(rsIDs):\n",
    "    label_color = color_map[maf_columns[i % len(maf_columns)]]  # Corrected color assignment\n",
    "    ax1.scatter(pca_df.loc[i, 'PC1'], pca_df.loc[i, 'PC2'], s=55, color=label_color)  # Set the size to 50\n",
    "\n",
    "ax1.set_xlabel('PCA1')\n",
    "ax1.set_ylabel('PCA2')\n",
    "ax1.set_title('PCA1 vs PCA2')\n",
    "\n",
    "# Plot PCA1 against PCA3\n",
    "ax2 = fig.add_subplot(122)\n",
    "for i, rsID in enumerate(rsIDs):\n",
    "    label_color = color_map[maf_columns[i % len(maf_columns)]]  # Corrected color assignment\n",
    "    ax2.scatter(pca_df.loc[i, 'PC1'], pca_df.loc[i, 'PC3'], s=55, color=label_color)  # Set the size to 50\n",
    "\n",
    "ax2.set_xlabel('PCA1')\n",
    "ax2.set_ylabel('PCA3')\n",
    "ax2.set_title('PCA1 vs PCA3')\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c5cec3ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bar</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>cLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A</td>\n",
       "      <td>ciLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>B</td>\n",
       "      <td>cLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C</td>\n",
       "      <td>fLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D</td>\n",
       "      <td>ciLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A</td>\n",
       "      <td>fLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>B</td>\n",
       "      <td>fLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>E</td>\n",
       "      <td>cLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>F</td>\n",
       "      <td>ciLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>H</td>\n",
       "      <td>fLAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  bar   type\n",
       "0   A   cLAD\n",
       "1   A  ciLAD\n",
       "2   B   cLAD\n",
       "3   C   fLAD\n",
       "4   D  ciLAD\n",
       "5   A   fLAD\n",
       "6   B   fLAD\n",
       "7   E   cLAD\n",
       "8   F  ciLAD\n",
       "9   H   fLAD"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "data = {\n",
    "    'bar': ['A', 'A', 'B', 'C', 'D', 'A', 'B', 'E', 'F', 'H'],\n",
    "    'type': ['cLAD', 'ciLAD', 'cLAD', 'fLAD', 'ciLAD', 'fLAD', 'fLAD', 'cLAD', 'ciLAD', 'fLAD']\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "f832404d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bar</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>ciLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B</td>\n",
       "      <td>cLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C</td>\n",
       "      <td>fLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D</td>\n",
       "      <td>ciLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>E</td>\n",
       "      <td>cLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>F</td>\n",
       "      <td>ciLAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>H</td>\n",
       "      <td>fLAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  bar   type\n",
       "0   A  ciLAD\n",
       "1   B   cLAD\n",
       "2   C   fLAD\n",
       "3   D  ciLAD\n",
       "4   E   cLAD\n",
       "5   F  ciLAD\n",
       "6   H   fLAD"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def prioritize_types(types):\n",
    "    if 'ciLAD' in types.values:\n",
    "        return 'ciLAD'\n",
    "    elif 'cLAD' in types.values:\n",
    "        return 'cLAD'\n",
    "    elif 'fLAD' in types.values:\n",
    "        return 'fLAD'\n",
    "    elif 'cLAD' in types.values and 'ciLAD' in types.values:\n",
    "        return 'ciLAD'\n",
    "    elif 'ciLAD' in types.values and 'fLAD' in types.values:\n",
    "        return 'ciLAD'\n",
    "    elif 'cLAD' in types.values and 'fLAD' in types.values:\n",
    "        return 'cLAD'\n",
    "    elif 'cLAD' in types.values and 'ciLAD' in types.values and 'fLAD' in types.values:\n",
    "        return 'ciLAD'\n",
    "\n",
    "df = df.groupby('bar')['type'].agg(prioritize_types).reset_index()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "eab54cde",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\GenepoweRx_Madhu\\AppData\\Local\\Temp\\ipykernel_5868\\1991291127.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  new_df = df.append(new_rows, ignore_index=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_df1</th>\n",
       "      <th>end_df1</th>\n",
       "      <th>chrom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000000</td>\n",
       "      <td>3100000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3100000</td>\n",
       "      <td>3200000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3200000</td>\n",
       "      <td>3300000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3300000</td>\n",
       "      <td>3400000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3400000</td>\n",
       "      <td>3500000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3600000</td>\n",
       "      <td>3700000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3700000</td>\n",
       "      <td>3800000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>3800000</td>\n",
       "      <td>3900000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3900000</td>\n",
       "      <td>4000000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3900000</td>\n",
       "      <td>4000000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4000000</td>\n",
       "      <td>4100000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4100000</td>\n",
       "      <td>4200000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4200000</td>\n",
       "      <td>4300000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4300000</td>\n",
       "      <td>4400000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>4400000</td>\n",
       "      <td>4500000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4600000</td>\n",
       "      <td>4700000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>4700000</td>\n",
       "      <td>4800000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4800000</td>\n",
       "      <td>4900000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>4900000</td>\n",
       "      <td>5000000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>5000000</td>\n",
       "      <td>5100000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>5100000</td>\n",
       "      <td>5200000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>5200000</td>\n",
       "      <td>5300000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    start_df1  end_df1 chrom\n",
       "0     3000000  3100000  chr1\n",
       "1     3100000  3200000  chr1\n",
       "2     3200000  3300000  chr1\n",
       "3     3300000  3400000  chr1\n",
       "4     3400000  3500000  chr1\n",
       "5     3600000  3700000  chr1\n",
       "6     3700000  3800000  chr1\n",
       "7     3800000  3900000  chr1\n",
       "8     3900000  4000000  chr1\n",
       "9     3900000  4000000  chr1\n",
       "10    4000000  4100000  chr1\n",
       "11    4100000  4200000  chr1\n",
       "12    4200000  4300000  chr1\n",
       "13    4300000  4400000  chr1\n",
       "14    4400000  4500000  chr1\n",
       "15    4600000  4700000  chr1\n",
       "16    4700000  4800000  chr1\n",
       "17    4800000  4900000  chr1\n",
       "18    4900000  5000000  chr1\n",
       "19    5000000  5100000  chr1\n",
       "20    5100000  5200000  chr1\n",
       "21    5200000  5300000  chr1"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Your initial DataFrame\n",
    "data = {'start_df1': [3000000, 3100000, 3200000, 3300000, 3400000, 3600000, 3900000, 4400000, 4600000, 4700000, 4800000, 4900000, 5000000, 5100000, 5200000],\n",
    "        'end_df1': [3100000, 3200000, 3300000, 3400000, 3500000, 3700000, 4000000, 4500000, 4700000, 4800000, 4900000, 5000000, 5100000, 5200000, 5300000],\n",
    "        'chrom': ['chr1']*15}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Create a new DataFrame with the missing rows filled in\n",
    "new_rows = []\n",
    "for chrom, group in df.groupby('chrom'):\n",
    "    cumulative_diff = 0\n",
    "    for i in range(1, len(group)):\n",
    "        diff = group['start_df1'].iloc[i] - group['end_df1'].iloc[i - 1]\n",
    "        cumulative_diff += diff\n",
    "        if cumulative_diff > 100000:\n",
    "            # Insert missing rows\n",
    "            num_missing_rows = cumulative_diff // 100000\n",
    "            for j in range(num_missing_rows):\n",
    "                new_row = {\n",
    "                    'start_df1': group['end_df1'].iloc[i - 1] + 100000 * j,\n",
    "                    'end_df1': group['end_df1'].iloc[i - 1] + 100000 * (j + 1),\n",
    "                    'chrom': chrom\n",
    "                }\n",
    "                new_rows.append(new_row)\n",
    "            cumulative_diff = 0  # reset cumulative difference after inserting rows\n",
    "\n",
    "# Append the new rows to the original DataFrame\n",
    "new_df = df.append(new_rows, ignore_index=True)\n",
    "\n",
    "# Sort the DataFrame based on 'chrom' and 'start_df1'\n",
    "new_df.sort_values(by=['chrom', 'start_df1'], inplace=True)\n",
    "\n",
    "# Reset the index\n",
    "new_df.reset_index(drop=True, inplace=True)\n",
    "new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "377b797c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_df1</th>\n",
       "      <th>end_df1</th>\n",
       "      <th>chrom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000000</td>\n",
       "      <td>3100000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3100000</td>\n",
       "      <td>3200000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3200000</td>\n",
       "      <td>3300000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3300000</td>\n",
       "      <td>3400000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3400000</td>\n",
       "      <td>3500000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3500000</td>\n",
       "      <td>3600000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3600000</td>\n",
       "      <td>3700000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>3700000</td>\n",
       "      <td>3800000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3800000</td>\n",
       "      <td>3900000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>3900000</td>\n",
       "      <td>4000000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4000000</td>\n",
       "      <td>4100000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>4100000</td>\n",
       "      <td>4200000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4200000</td>\n",
       "      <td>4300000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>4300000</td>\n",
       "      <td>4400000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4400000</td>\n",
       "      <td>4500000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>4500000</td>\n",
       "      <td>4600000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4600000</td>\n",
       "      <td>4700000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>4700000</td>\n",
       "      <td>4800000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>4800000</td>\n",
       "      <td>4900000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>4900000</td>\n",
       "      <td>5000000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>5000000</td>\n",
       "      <td>5100000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>5100000</td>\n",
       "      <td>5200000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>5200000</td>\n",
       "      <td>5300000</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   start_df1  end_df1 chrom\n",
       "0    3000000  3100000  chr1\n",
       "2    3100000  3200000  chr1\n",
       "4    3200000  3300000  chr1\n",
       "6    3300000  3400000  chr1\n",
       "8    3400000  3500000  chr1\n",
       "10   3500000  3600000  chr1\n",
       "11   3600000  3700000  chr1\n",
       "13   3700000  3800000  chr1\n",
       "14   3800000  3900000  chr1\n",
       "15   3900000  4000000  chr1\n",
       "17   4000000  4100000  chr1\n",
       "18   4100000  4200000  chr1\n",
       "19   4200000  4300000  chr1\n",
       "20   4300000  4400000  chr1\n",
       "21   4400000  4500000  chr1\n",
       "23   4500000  4600000  chr1\n",
       "24   4600000  4700000  chr1\n",
       "26   4700000  4800000  chr1\n",
       "28   4800000  4900000  chr1\n",
       "30   4900000  5000000  chr1\n",
       "32   5000000  5100000  chr1\n",
       "34   5100000  5200000  chr1\n",
       "36   5200000  5300000  chr1"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Your initial DataFrame\n",
    "data = {\n",
    "    'start_df1': [3000000, 3100000, 3200000, 3300000, 3400000, 3600000, 3900000, 4400000, 4600000, 4700000, 4800000, 4900000, 5000000, 5100000, 5200000],\n",
    "    'end_df1': [3100000, 3200000, 3300000, 3400000, 3500000, 3700000, 4000000, 4500000, 4700000, 4800000, 4900000, 5000000, 5100000, 5200000, 5300000],\n",
    "    'chrom': ['chr1']*15\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Function to impute missing rows\n",
    "def impute_missing_rows(df):\n",
    "    result = pd.DataFrame(columns=df.columns)\n",
    "    \n",
    "    for chrom, group in df.groupby('chrom'):\n",
    "        start = group['start_df1'].min()\n",
    "        end = group['end_df1'].max()\n",
    "        \n",
    "        # Generate new rows based on 100000 difference\n",
    "        new_rows = pd.DataFrame({\n",
    "            'start_df1': range(start, end, 100000),\n",
    "            'end_df1': range(start + 100000, end + 100000, 100000),\n",
    "            'chrom': [chrom]*len(range(start, end, 100000))\n",
    "        })\n",
    "        \n",
    "        result = pd.concat([result, new_rows], ignore_index=True)\n",
    "    \n",
    "    return result\n",
    "\n",
    "# Concatenate the original DataFrame with the imputed rows\n",
    "result_df = pd.concat([df, impute_missing_rows(df)], ignore_index=True)\n",
    "\n",
    "# Sort the DataFrame based on 'chrom' and 'start_df1'\n",
    "result_df = result_df.sort_values(by=['chrom', 'start_df1']).reset_index(drop=True)\n",
    "result_df.drop_duplicates(subset = ['start_df1', 'end_df1', 'chrom'], inplace=True)\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3fd48252",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_df1</th>\n",
       "      <th>end_df1</th>\n",
       "      <th>chrom</th>\n",
       "      <th>col1</th>\n",
       "      <th>col2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000000</td>\n",
       "      <td>3100000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3100000</td>\n",
       "      <td>3200000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3200000</td>\n",
       "      <td>3300000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3300000</td>\n",
       "      <td>3400000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3400000</td>\n",
       "      <td>3500000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5</td>\n",
       "      <td>E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3500000</td>\n",
       "      <td>3600000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3600000</td>\n",
       "      <td>3700000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>3700000</td>\n",
       "      <td>3800000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3800000</td>\n",
       "      <td>3900000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>3900000</td>\n",
       "      <td>4000000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>4000000</td>\n",
       "      <td>4100000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>4100000</td>\n",
       "      <td>4200000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>4200000</td>\n",
       "      <td>4300000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>4300000</td>\n",
       "      <td>4400000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4400000</td>\n",
       "      <td>4500000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8</td>\n",
       "      <td>H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>4500000</td>\n",
       "      <td>4600000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>4600000</td>\n",
       "      <td>4700000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>9</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>4700000</td>\n",
       "      <td>4800000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10</td>\n",
       "      <td>J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>4800000</td>\n",
       "      <td>4900000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11</td>\n",
       "      <td>K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>4900000</td>\n",
       "      <td>5000000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>12</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>5000000</td>\n",
       "      <td>5100000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13</td>\n",
       "      <td>M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>5100000</td>\n",
       "      <td>5200000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>5200000</td>\n",
       "      <td>5300000</td>\n",
       "      <td>chr1</td>\n",
       "      <td>15</td>\n",
       "      <td>O</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   start_df1  end_df1 chrom col1 col2\n",
       "0    3000000  3100000  chr1    1    A\n",
       "2    3100000  3200000  chr1    2    B\n",
       "4    3200000  3300000  chr1    3    C\n",
       "6    3300000  3400000  chr1    4    D\n",
       "8    3400000  3500000  chr1    5    E\n",
       "10   3500000  3600000  chr1    1    A\n",
       "11   3600000  3700000  chr1    6    F\n",
       "13   3700000  3800000  chr1    1    A\n",
       "14   3800000  3900000  chr1    1    A\n",
       "15   3900000  4000000  chr1    7    G\n",
       "17   4000000  4100000  chr1    1    A\n",
       "18   4100000  4200000  chr1    1    A\n",
       "19   4200000  4300000  chr1    1    A\n",
       "20   4300000  4400000  chr1    1    A\n",
       "21   4400000  4500000  chr1    8    H\n",
       "23   4500000  4600000  chr1    1    A\n",
       "24   4600000  4700000  chr1    9    I\n",
       "26   4700000  4800000  chr1   10    J\n",
       "28   4800000  4900000  chr1   11    K\n",
       "30   4900000  5000000  chr1   12    L\n",
       "32   5000000  5100000  chr1   13    M\n",
       "34   5100000  5200000  chr1   14    N\n",
       "36   5200000  5300000  chr1   15    O"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Your initial DataFrame\n",
    "data = {\n",
    "    'start_df1': [3000000, 3100000, 3200000, 3300000, 3400000, 3600000, 3900000, 4400000, 4600000, 4700000, 4800000, 4900000, 5000000, 5100000, 5200000],\n",
    "    'end_df1': [3100000, 3200000, 3300000, 3400000, 3500000, 3700000, 4000000, 4500000, 4700000, 4800000, 4900000, 5000000, 5100000, 5200000, 5300000],\n",
    "    'chrom': ['chr1']*15,\n",
    "    'col1': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15],\n",
    "    'col2': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O']\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Function to impute missing rows\n",
    "def impute_missing_rows(df):\n",
    "    result = pd.DataFrame(columns=df.columns)\n",
    "    \n",
    "    for chrom, group in df.groupby('chrom'):\n",
    "        start = group['start_df1'].min()\n",
    "        end = group['end_df1'].max()\n",
    "        \n",
    "        # Generate new rows based on 100000 difference\n",
    "        new_rows = pd.DataFrame({\n",
    "            'start_df1': range(start, end, 100000),\n",
    "            'end_df1': range(start + 100000, end + 100000, 100000),\n",
    "            'chrom': [chrom]*len(range(start, end, 100000)),\n",
    "            'col1': [group['col1'].iloc[0]]*len(range(start, end, 100000)),  # Assuming col1 doesn't change within a chrom group\n",
    "            'col2': [group['col2'].iloc[0]]*len(range(start, end, 100000))   # Assuming col2 doesn't change within a chrom group\n",
    "        })\n",
    "        \n",
    "        result = pd.concat([result, new_rows], ignore_index=True)\n",
    "    \n",
    "    return result\n",
    "\n",
    "# Concatenate the original DataFrame with the imputed rows\n",
    "result_df = pd.concat([df, impute_missing_rows(df)], ignore_index=True)\n",
    "\n",
    "# Sort the DataFrame based on 'chrom' and 'start_df1'\n",
    "result_df = result_df.sort_values(by=['chrom', 'start_df1']).reset_index(drop=True)\n",
    "result_df.drop_duplicates(subset=['start_df1', 'end_df1', 'chrom'], inplace=True)\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b3241d44",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>your_column_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    your_column_name\n",
       "0                1.0\n",
       "1                2.0\n",
       "2                2.0\n",
       "3                2.0\n",
       "4                5.0\n",
       "5                6.0\n",
       "6                6.0\n",
       "7                8.0\n",
       "8                8.0\n",
       "9                8.0\n",
       "10              11.0"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming your data is in a DataFrame named 'df'\n",
    "# Replace 'your_column_name' with the actual column name you want to fill null values for\n",
    "\n",
    "# Example DataFrame\n",
    "data = {\n",
    "    'your_column_name': [1, 2, None, None, 5, 6, None, 8, None, None, 11],\n",
    "    # add other columns if necessary\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Fill null rows with the mean of the previous non-null row\n",
    "df['your_column_name'] = df['your_column_name'].ffill().fillna(df['your_column_name'].mean())\n",
    "\n",
    "# Print the resulting DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "577264a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start1</th>\n",
       "      <th>end1</th>\n",
       "      <th>range_start</th>\n",
       "      <th>range_end</th>\n",
       "      <th>bound</th>\n",
       "      <th>difference</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>A</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>B</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>15</td>\n",
       "      <td>B</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>15</td>\n",
       "      <td>20</td>\n",
       "      <td>B</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>25</td>\n",
       "      <td>B</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>18</td>\n",
       "      <td>26</td>\n",
       "      <td>15</td>\n",
       "      <td>20</td>\n",
       "      <td>C</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>18</td>\n",
       "      <td>26</td>\n",
       "      <td>20</td>\n",
       "      <td>25</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>18</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>30</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   start1  end1  range_start  range_end bound  difference\n",
       "0       4     8            0          5     A           2\n",
       "1       4     8            5         10     A           4\n",
       "2       9    21            5         10     B           2\n",
       "3       9    21           10         15     B           6\n",
       "4       9    21           15         20     B           6\n",
       "5       9    21           20         25     B           2\n",
       "6      18    26           15         20     C           3\n",
       "7      18    26           20         25     C           6\n",
       "8      18    26           25         30     C           2"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample data\n",
    "data = {'start1': [4, 4, 9, 9, 9, 9, 18, 18, 18],\n",
    "        'end1': [8, 8, 21, 21, 21, 21, 26, 26, 26],\n",
    "        'range_start': [0, 5, 5, 10, 15, 20, 15, 20, 25],\n",
    "        'range_end': [5, 10, 10, 15, 20, 25, 20, 25, 30],\n",
    "        'bound': ['A', 'A', 'B', 'B', 'B', 'B', 'C', 'C', 'C']}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Calculate the shared portion and add it to a new 'difference' column\n",
    "df['difference'] = df.apply(lambda row: len(range(max(row['start1'], row['range_start']), min(row['end1'], row['range_end']) + 1)), axis=1)\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a06a9f30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Group</th>\n",
       "      <th>Value1</th>\n",
       "      <th>Value2</th>\n",
       "      <th>Value3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>80</td>\n",
       "      <td>65</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B</td>\n",
       "      <td>70</td>\n",
       "      <td>60</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C</td>\n",
       "      <td>60</td>\n",
       "      <td>55</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Group  Value1  Value2  Value3\n",
       "0     A      80      65       8\n",
       "1     B      70      60       7\n",
       "2     C      60      55       6"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample DataFrame\n",
    "data = {'Group': ['A', 'A', 'B', 'B', 'A', 'C'],\n",
    "        'Value1': [10, 20, 30, 40, 50, 60],\n",
    "        'Value2': [5, 15, 25, 35, 45, 55],\n",
    "        'Value3': [1, 2, 3, 4, 5, 6]}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Calculate the sum of unique values in 'Value1', 'Value2', and 'Value3' for each group\n",
    "grouped_df = df.groupby('Group').agg({\n",
    "    'Value1': lambda x: pd.Series.unique(x).sum(),\n",
    "    'Value2': lambda x: pd.Series.unique(x).sum(),\n",
    "    'Value3': lambda x: pd.Series.unique(x).sum()\n",
    "}).reset_index()\n",
    "\n",
    "grouped_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "51e23f9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sample1</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sample2</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sample3</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   samples variant\n",
       "0  Sample1       A\n",
       "2  Sample2       A\n",
       "4  Sample3       A"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example DataFrame\n",
    "df = pd.DataFrame({\n",
    "    'samples': ['Sample1', 'Sample1', 'Sample2', 'Sample2', 'Sample3'],\n",
    "    'variant': ['A', 'B', 'A', 'C', 'A']\n",
    "})\n",
    "\n",
    "# Counting the occurrence of each variant\n",
    "variant_counts = df['variant'].value_counts()\n",
    "\n",
    "# Filtering variants that are present in all samples\n",
    "variants_in_all_samples = variant_counts[variant_counts == df['samples'].nunique()].index.tolist()\n",
    "\n",
    "# Filtering the DataFrame to include only rows with variants present in all samples\n",
    "rows_in_all_samples = df[df['variant'].isin(variants_in_all_samples)]\n",
    "rows_in_all_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00a1f15c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
