{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "57248292",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "b6eb5e24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>rsID_updated</th>\n",
       "      <th>Haplotype</th>\n",
       "      <th>rsID_mapped</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*4</td>\n",
       "      <td>rs1799930,rs1041983,rs1208,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous,Homozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>rs1799930,rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6O</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7G</td>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*13A</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:38:28:28:17:11:39.29%:1.4422E-4:67:60:14:3...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*6B</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400806</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400806</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400806</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>NAT2*12A</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400860</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9227,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:41:41:27:13:31.71%:3.8184E-5:68:59:24:3...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>NAT2*7A</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94775367</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=53;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.7716,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:86:53:53:29:24:45.28%:2.005E-9:66:56:26:3:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48045719</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>NUDT15*2</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48045719</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>NUDT15*3</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*4</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "1    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "2    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "3    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "4    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "5    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "6    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "9    chr8  18400593    rs1799930   G   A    .   PASS   \n",
       "13   chr8  18400806       rs1208   G   A    .   PASS   \n",
       "14   chr8  18400806       rs1208   G   A    .   PASS   \n",
       "15   chr8  18400806       rs1208   G   A    .   PASS   \n",
       "18   chr8  18400860    rs1799931   G   A    .   PASS   \n",
       "21  chr10  94775367   rs12769205   A   G    .   PASS   \n",
       "23  chr13  48045719  rs116855232   C   T    .   PASS   \n",
       "24  chr13  48045719  rs116855232   C   T    .   PASS   \n",
       "25  chr19  41009358    rs2279343   A   G    .   PASS   \n",
       "26  chr19  41009358    rs2279343   A   G    .   PASS   \n",
       "27  chr19  41009358    rs2279343   A   G    .   PASS   \n",
       "\n",
       "                                                 INFO  \\\n",
       "0   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "1   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "2   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "3   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "4   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "5   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "6   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "9   ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "13  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....   \n",
       "14  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....   \n",
       "15  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....   \n",
       "18  ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9227,0....   \n",
       "21  ADP=53;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.7716,0....   \n",
       "23  ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....   \n",
       "24  ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....   \n",
       "25  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "26  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "27  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "\n",
       "                                               FORMAT  \\\n",
       "0   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "6   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "9   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "13  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "14  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "15  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "18  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "21  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "23  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "24  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "25  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "26  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "27  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                               SAMPLE      Zygosity  \\\n",
       "0    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "1    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "2    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "3    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "4    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "5    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "6    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "9   0/1:38:28:28:17:11:39.29%:1.4422E-4:67:60:14:3...  Heterozygous   \n",
       "13   1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5    Homozygous   \n",
       "14   1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5    Homozygous   \n",
       "15   1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5    Homozygous   \n",
       "18  0/1:44:41:41:27:13:31.71%:3.8184E-5:68:59:24:3...  Heterozygous   \n",
       "21  0/1:86:53:53:29:24:45.28%:2.005E-9:66:56:26:3:...  Heterozygous   \n",
       "23   0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1  Heterozygous   \n",
       "24   0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1  Heterozygous   \n",
       "25    1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5    Homozygous   \n",
       "26    1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5    Homozygous   \n",
       "27    1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5    Homozygous   \n",
       "\n",
       "   rsID_updated  Haplotype                           rsID_mapped  \\\n",
       "0     rs1041983     NAT2*4  rs1799930,rs1041983,rs1208,rs1799931   \n",
       "1     rs1041983    NAT2*6A                   rs1799930,rs1041983   \n",
       "2     rs1041983    NAT2*6J         rs1799930,rs1041983,rs1799931   \n",
       "3     rs1041983    NAT2*6O                   rs1799930,rs1041983   \n",
       "4     rs1041983    NAT2*7B                   rs1041983,rs1799931   \n",
       "5     rs1041983    NAT2*7G                   rs1041983,rs1799931   \n",
       "6     rs1041983   NAT2*13A                             rs1041983   \n",
       "9     rs1799930    NAT2*6B                             rs1799930   \n",
       "13       rs1208    NAT2*5B                                rs1208   \n",
       "14       rs1208    NAT2*5C                                rs1208   \n",
       "15       rs1208   NAT2*12A                                rs1208   \n",
       "18    rs1799931    NAT2*7A                             rs1799931   \n",
       "21   rs12769205  CYP2C19*2                  rs12769205,rs4244285   \n",
       "23  rs116855232   NUDT15*2                           rs116855232   \n",
       "24  rs116855232   NUDT15*3                           rs116855232   \n",
       "25    rs2279343   CYP2B6*4                             rs2279343   \n",
       "26    rs2279343   CYP2B6*6                             rs2279343   \n",
       "27    rs2279343  CYP2B6*26                             rs2279343   \n",
       "\n",
       "                                    Zygosity_mapped  \n",
       "0   Homozygous,Heterozygous,Homozygous,Heterozygous  \n",
       "1                           Homozygous,Heterozygous  \n",
       "2              Homozygous,Heterozygous,Heterozygous  \n",
       "3                           Homozygous,Heterozygous  \n",
       "4                           Homozygous,Heterozygous  \n",
       "5                           Homozygous,Heterozygous  \n",
       "6                                        Homozygous  \n",
       "9                                      Heterozygous  \n",
       "13                                       Homozygous  \n",
       "14                                       Homozygous  \n",
       "15                                       Homozygous  \n",
       "18                                     Heterozygous  \n",
       "21                        Heterozygous,Heterozygous  \n",
       "23                                     Heterozygous  \n",
       "24                                     Heterozygous  \n",
       "25                                       Homozygous  \n",
       "26                                       Homozygous  \n",
       "27                                       Homozygous  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGPRX959_final_DP.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "df.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "df['HET'] = df['INFO'].str.extract(r'HET=(\\d)')\n",
    "df['HOM'] = df['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "df['Zygosity'] = ''\n",
    "df.loc[df['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "df.loc[df['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "df = df.drop(columns=['HET', 'HOM'], axis = 1)\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/RSIDS_POSITION_DBSNP.xlsx')\n",
    "df = pd.merge(df, data, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort = False)\n",
    "df2 = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/13_Genes_Key_Variants_Coordinates.csv', sep = '\\t')\n",
    "df2 = df2[['CHROM', 'POS', 'REF', 'ALT', 'Haplotype']]\n",
    "merged = pd.merge(df, df2, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "madhu = merged.copy()\n",
    "madhu['rsID_mapped'] = madhu.groupby('Haplotype')['rsID'].transform(lambda x: ','.join(set(x)))\n",
    "madhu['Zygosity_mapped'] = madhu.groupby(['Haplotype'])['Zygosity'].transform(lambda x: ','.join(x))\n",
    "madhu.drop_duplicates(subset='Haplotype', inplace=True)\n",
    "madhu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "880e3841",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>rsID_updated</th>\n",
       "      <th>Haplotype</th>\n",
       "      <th>rsID_mapped</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*4</td>\n",
       "      <td>rs1799930,rs1041983,rs1208,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous,Homozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1208,rs1799930,rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>rs1799930,rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930,rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6O</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7G</td>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*13A</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:38:28:28:17:11:39.29%:1.4422E-4:67:60:14:3...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*6B</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400806</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400806</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400806</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>NAT2*12A</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs1208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400860</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9227,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:41:41:27:13:31.71%:3.8184E-5:68:59:24:3...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>NAT2*7A</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94775367</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=53;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.7716,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:86:53:53:29:24:45.28%:2.005E-9:66:56:26:3:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48045719</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>NUDT15*2</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs116855232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48045719</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>NUDT15*3</td>\n",
       "      <td>rs116855232</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs116855232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*4</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "1    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "2    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "3    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "4    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "5    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "6    chr8  18400285    rs1041983   C   T    .   PASS   \n",
       "9    chr8  18400593    rs1799930   G   A    .   PASS   \n",
       "13   chr8  18400806       rs1208   G   A    .   PASS   \n",
       "14   chr8  18400806       rs1208   G   A    .   PASS   \n",
       "15   chr8  18400806       rs1208   G   A    .   PASS   \n",
       "18   chr8  18400860    rs1799931   G   A    .   PASS   \n",
       "21  chr10  94775367   rs12769205   A   G    .   PASS   \n",
       "23  chr13  48045719  rs116855232   C   T    .   PASS   \n",
       "24  chr13  48045719  rs116855232   C   T    .   PASS   \n",
       "25  chr19  41009358    rs2279343   A   G    .   PASS   \n",
       "26  chr19  41009358    rs2279343   A   G    .   PASS   \n",
       "27  chr19  41009358    rs2279343   A   G    .   PASS   \n",
       "\n",
       "                                                 INFO  \\\n",
       "0   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "1   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "2   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "3   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "4   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "5   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "6   ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6026,0....   \n",
       "9   ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "13  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....   \n",
       "14  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....   \n",
       "15  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.3229,0....   \n",
       "18  ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9227,0....   \n",
       "21  ADP=53;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.7716,0....   \n",
       "23  ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....   \n",
       "24  ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9605,0....   \n",
       "25  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "26  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "27  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "\n",
       "                                               FORMAT  \\\n",
       "0   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "6   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "9   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "13  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "14  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "15  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "18  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "21  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "23  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "24  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "25  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "26  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "27  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                               SAMPLE      Zygosity  \\\n",
       "0    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "1    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "2    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "3    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "4    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "5    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "6    1/1:182:33:32:0:32:100%:5.4567E-19:0:57:0:0:24:8    Homozygous   \n",
       "9   0/1:38:28:28:17:11:39.29%:1.4422E-4:67:60:14:3...  Heterozygous   \n",
       "13   1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5    Homozygous   \n",
       "14   1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5    Homozygous   \n",
       "15   1/1:254:44:44:0:44:100%:3.8097E-26:0:63:0:0:39:5    Homozygous   \n",
       "18  0/1:44:41:41:27:13:31.71%:3.8184E-5:68:59:24:3...  Heterozygous   \n",
       "21  0/1:86:53:53:29:24:45.28%:2.005E-9:66:56:26:3:...  Heterozygous   \n",
       "23   0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1  Heterozygous   \n",
       "24   0/1:44:24:24:12:12:50%:3.8815E-5:68:61:12:0:11:1  Heterozygous   \n",
       "25    1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5    Homozygous   \n",
       "26    1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5    Homozygous   \n",
       "27    1/1:99:18:18:0:18:100%:1.1019E-10:0:55:0:0:13:5    Homozygous   \n",
       "\n",
       "   rsID_updated  Haplotype                           rsID_mapped  \\\n",
       "0     rs1041983     NAT2*4  rs1799930,rs1041983,rs1208,rs1799931   \n",
       "1     rs1041983    NAT2*6A                   rs1799930,rs1041983   \n",
       "2     rs1041983    NAT2*6J         rs1799930,rs1041983,rs1799931   \n",
       "3     rs1041983    NAT2*6O                   rs1799930,rs1041983   \n",
       "4     rs1041983    NAT2*7B                   rs1041983,rs1799931   \n",
       "5     rs1041983    NAT2*7G                   rs1041983,rs1799931   \n",
       "6     rs1041983   NAT2*13A                             rs1041983   \n",
       "9     rs1799930    NAT2*6B                             rs1799930   \n",
       "13       rs1208    NAT2*5B                                rs1208   \n",
       "14       rs1208    NAT2*5C                                rs1208   \n",
       "15       rs1208   NAT2*12A                                rs1208   \n",
       "18    rs1799931    NAT2*7A                             rs1799931   \n",
       "21   rs12769205  CYP2C19*2                  rs12769205,rs4244285   \n",
       "23  rs116855232   NUDT15*2                           rs116855232   \n",
       "24  rs116855232   NUDT15*3                           rs116855232   \n",
       "25    rs2279343   CYP2B6*4                             rs2279343   \n",
       "26    rs2279343   CYP2B6*6                             rs2279343   \n",
       "27    rs2279343  CYP2B6*26                             rs2279343   \n",
       "\n",
       "                                    Zygosity_mapped  \\\n",
       "0   Homozygous,Heterozygous,Homozygous,Heterozygous   \n",
       "1                           Homozygous,Heterozygous   \n",
       "2              Homozygous,Heterozygous,Heterozygous   \n",
       "3                           Homozygous,Heterozygous   \n",
       "4                           Homozygous,Heterozygous   \n",
       "5                           Homozygous,Heterozygous   \n",
       "6                                        Homozygous   \n",
       "9                                      Heterozygous   \n",
       "13                                       Homozygous   \n",
       "14                                       Homozygous   \n",
       "15                                       Homozygous   \n",
       "18                                     Heterozygous   \n",
       "21                        Heterozygous,Heterozygous   \n",
       "23                                     Heterozygous   \n",
       "24                                     Heterozygous   \n",
       "25                                       Homozygous   \n",
       "26                                       Homozygous   \n",
       "27                                       Homozygous   \n",
       "\n",
       "                      rsID_mapped_sorted  \n",
       "0   rs1041983,rs1208,rs1799930,rs1799931  \n",
       "1                    rs1041983,rs1799930  \n",
       "2          rs1041983,rs1799930,rs1799931  \n",
       "3                    rs1041983,rs1799930  \n",
       "4                    rs1041983,rs1799931  \n",
       "5                    rs1041983,rs1799931  \n",
       "6                              rs1041983  \n",
       "9                              rs1799930  \n",
       "13                                rs1208  \n",
       "14                                rs1208  \n",
       "15                                rs1208  \n",
       "18                             rs1799931  \n",
       "21                  rs12769205,rs4244285  \n",
       "23                           rs116855232  \n",
       "24                           rs116855232  \n",
       "25                             rs2279343  \n",
       "26                             rs2279343  \n",
       "27                             rs2279343  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "madhu['rsID_mapped_sorted'] = madhu['rsID_mapped'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "madhu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "02de3318",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>Haplotype_updated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1041983,rs1799930,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              rsID_mapped_sorted                       Zygosity_mapped  \\\n",
       "0            rs1041983,rs1799930               Homozygous,Heterozygous   \n",
       "2  rs1041983,rs1799930,rs1799931  Homozygous,Heterozygous,Heterozygous   \n",
       "3            rs1041983,rs1799931               Homozygous,Heterozygous   \n",
       "5           rs12769205,rs4244285             Heterozygous,Heterozygous   \n",
       "\n",
       "  Haplotype_updated  \n",
       "0           NAT2*6A  \n",
       "2           NAT2*6J  \n",
       "3           NAT2*7B  \n",
       "5         CYP2C19*2  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGPRX959_final_DP.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "df.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "df['HET'] = df['INFO'].str.extract(r'HET=(\\d)')\n",
    "df['HOM'] = df['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "df['Zygosity'] = ''\n",
    "df.loc[df['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "df.loc[df['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "df = df.drop(columns=['HET', 'HOM'], axis = 1)\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/RSIDS_POSITION_DBSNP.xlsx')\n",
    "df = pd.merge(df, data, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort = False)\n",
    "df2 = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/13_Genes_Key_Variants_Coordinates.csv', sep = '\\t')\n",
    "df2 = df2[['CHROM', 'POS', 'REF', 'ALT', 'Haplotype']]\n",
    "merged = pd.merge(df, df2, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "madhu = merged.copy()\n",
    "madhu['rsID_mapped'] = madhu.groupby('Haplotype')['rsID'].transform(lambda x: ','.join(set(x)))\n",
    "madhu['Zygosity_mapped'] = madhu.groupby(['Haplotype'])['Zygosity'].transform(lambda x: ','.join(x))\n",
    "madhu.drop_duplicates(subset='Haplotype', inplace=True)\n",
    "\n",
    "madhu['rsID_mapped_sorted'] = madhu['rsID_mapped'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "madhu = madhu.drop(columns=['rsID_updated', 'Haplotype'], axis=1)\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Multiple_Positions.xlsx')\n",
    "data = data.rename({'Haplotype':'Haplotype_updated', 'rsID':'rsID_mapped'}, axis = 1)\n",
    "data['rsID_mapped_sorted'] = data['rsID_mapped'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "data = data.drop(columns=['rsID_mapped'], axis=1)\n",
    "mapped_df = pd.merge(madhu, data, on = 'rsID_mapped_sorted', how = 'inner', sort = False)\n",
    "mapped_df.drop_duplicates(subset='Haplotype_updated', inplace=True)\n",
    "mapped_df = mapped_df.drop(columns=['Zygosity'], axis=1)\n",
    "mapped_df = mapped_df[['rsID_mapped_sorted', 'Zygosity_mapped', 'Haplotype_updated']]\n",
    "mapped_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d58f3ffc",
   "metadata": {},
   "source": [
    "## check the sorting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cdcb8c87",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b400dae",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "4da52357",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID_mapped</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>Haplotype_updated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1799930,rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     rsID_mapped                       Zygosity_mapped  \\\n",
       "0            rs1799930,rs1041983               Homozygous,Heterozygous   \n",
       "2  rs1799930,rs1041983,rs1799931  Homozygous,Heterozygous,Heterozygous   \n",
       "3            rs1041983,rs1799931               Homozygous,Heterozygous   \n",
       "5           rs12769205,rs4244285             Heterozygous,Heterozygous   \n",
       "\n",
       "  Haplotype_updated  \n",
       "0           NAT2*6A  \n",
       "2           NAT2*6J  \n",
       "3           NAT2*7B  \n",
       "5         CYP2C19*2  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGPRX959_final_DP.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "df.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "df['HET'] = df['INFO'].str.extract(r'HET=(\\d)')\n",
    "df['HOM'] = df['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "df['Zygosity'] = ''\n",
    "df.loc[df['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "df.loc[df['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "df = df.drop(columns=['HET', 'HOM'], axis = 1)\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/RSIDS_POSITION_DBSNP.xlsx')\n",
    "df = pd.merge(df, data, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort = False)\n",
    "df2 = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/13_Genes_Key_Variants_Coordinates.csv', sep = '\\t')\n",
    "df2 = df2[['CHROM', 'POS', 'REF', 'ALT', 'Haplotype']]\n",
    "merged = pd.merge(df, df2, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "madhu = merged.copy()\n",
    "madhu['rsID_mapped'] = madhu.groupby('Haplotype')['rsID'].transform(lambda x: ','.join(set(x)))\n",
    "madhu['Zygosity_mapped'] = madhu.groupby(['Haplotype'])['Zygosity'].transform(lambda x: ','.join(x))\n",
    "madhu.drop_duplicates(subset='Haplotype', inplace=True)\n",
    "\n",
    "madhu['rsID_mapped_sorted'] = madhu['rsID_mapped'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "madhu = madhu.drop(columns=['rsID_updated', 'Haplotype'], axis=1)\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Multiple_Positions.xlsx')\n",
    "data = data.rename({'Haplotype':'Haplotype_updated', 'rsID':'rsID_mapped'}, axis = 1)\n",
    "data['rsID_mapped_sorted'] = data['rsID_mapped'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "data = data.drop(columns=['rsID_mapped'], axis=1)\n",
    "mapped_df = pd.merge(madhu, data, on = 'rsID_mapped_sorted', how = 'inner', sort = False)\n",
    "mapped_df.drop_duplicates(subset='Haplotype_updated', inplace=True)\n",
    "mapped_df = mapped_df.drop(columns=['Zygosity'], axis=1)\n",
    "mapped_df = mapped_df[['rsID_mapped', 'Zygosity_mapped', 'Haplotype_updated']]\n",
    "mapped_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adfdb505",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba0cbf12",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0943a802",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c19a0b16",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "b7c8949b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>rsID_updated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:38:28:28:17:11:39.29%:1.4422E-4:67:60:14:3...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "1  chr8  18400593  rs1799930   G   A    .   PASS   \n",
       "\n",
       "                                                INFO  \\\n",
       "1  ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "\n",
       "                                              FORMAT  \\\n",
       "1  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                              SAMPLE      Zygosity  \\\n",
       "1  0/1:38:28:28:17:11:39.29%:1.4422E-4:67:60:14:3...  Heterozygous   \n",
       "\n",
       "  rsID_updated  \n",
       "1    rs1799930  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGPRX959_final_DP.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "df.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "df['HET'] = df['INFO'].str.extract(r'HET=(\\d)')\n",
    "df['HOM'] = df['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "df['Zygosity'] = ''\n",
    "df.loc[df['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "df.loc[df['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "df = df.drop(columns=['HET', 'HOM'], axis = 1)\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/RSIDS_POSITION_DBSNP.xlsx')\n",
    "df = pd.merge(df, data, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort = False)\n",
    "df[df['rsID_updated'] == 'rs1799930']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7e8336a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7340415f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dec2f5a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2f7593e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77bb9e33",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c62316bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>rsID_updated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94775367</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs12769205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS        rsID REF ALT QUAL FILTER  \\\n",
       "0   chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "1   chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "2  chr10  94775367  rs12769205   A   G    .   PASS   \n",
       "3  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "4  chr19  41009358   rs2279343   A   G    .   PASS   \n",
       "\n",
       "                                                INFO  \\\n",
       "0  ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "1  ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "2  ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....   \n",
       "3  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "4  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "\n",
       "                                              FORMAT  \\\n",
       "0  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                              SAMPLE      Zygosity  \\\n",
       "0  0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "1  0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "2  1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...    Homozygous   \n",
       "3  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "4  0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5  Heterozygous   \n",
       "\n",
       "  rsID_updated  \n",
       "0    rs1041983  \n",
       "1    rs1799930  \n",
       "2   rs12769205  \n",
       "3    rs3745274  \n",
       "4    rs2279343  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGPRX981_final_DP.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "df.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "df['HET'] = df['INFO'].str.extract(r'HET=(\\d)')\n",
    "df['HOM'] = df['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "df['Zygosity'] = ''\n",
    "df.loc[df['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "df.loc[df['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "df = df.drop(columns=['HET', 'HOM'], axis = 1)\n",
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/RSIDS_POSITION_DBSNP.xlsx')\n",
    "df = pd.merge(df, data, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort = False)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9c93cf61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Haplotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40848628.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>CYP2A6*2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40843869.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>CYP2A6*7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40843749.0</td>\n",
       "      <td>CC</td>\n",
       "      <td>C</td>\n",
       "      <td>CYP2A6*7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40843746.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CYP2A6*7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40843742.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CYP2A6*7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154536019.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G6PD*Musashino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>725</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154536151.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G6PD*Kambos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154532258.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G6PD*Kamiube, Keelung</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>727</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154546116.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>G6PD*Lages</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>728</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154533004.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>G6PD*Insuli</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>729 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM          POS REF ALT              Haplotype\n",
       "0    chr19   40848628.0   A   T               CYP2A6*2\n",
       "1    chr19   40843869.0   A   G               CYP2A6*7\n",
       "2    chr19   40843749.0  CC   C               CYP2A6*7\n",
       "3    chr19   40843746.0   T   C               CYP2A6*7\n",
       "4    chr19   40843742.0   G   A               CYP2A6*7\n",
       "..     ...          ...  ..  ..                    ...\n",
       "724   chrX  154536019.0   G   A         G6PD*Musashino\n",
       "725   chrX  154536151.0   G   A            G6PD*Kambos\n",
       "726   chrX  154532258.0   G   A  G6PD*Kamiube, Keelung\n",
       "727   chrX  154546116.0   C   T             G6PD*Lages\n",
       "728   chrX  154533004.0   C   T            G6PD*Insuli\n",
       "\n",
       "[729 rows x 5 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/13_Genes_Key_Variants_Coordinates.csv', sep = '\\t')\n",
    "df2 = df2[['CHROM', 'POS', 'REF', 'ALT', 'Haplotype']]\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6f9c71fa",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>rsID_updated</th>\n",
       "      <th>Haplotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6O</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*13A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*6A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*6B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*6J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*6O</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94775367</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>CYP2B6*9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS        rsID REF ALT QUAL FILTER  \\\n",
       "0    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "1    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "2    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "3    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "4    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "5    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "6    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "7    chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "8    chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "9    chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "10   chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "11   chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "12  chr10  94775367  rs12769205   A   G    .   PASS   \n",
       "13  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "14  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "15  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "16  chr19  41009358   rs2279343   A   G    .   PASS   \n",
       "17  chr19  41009358   rs2279343   A   G    .   PASS   \n",
       "18  chr19  41009358   rs2279343   A   G    .   PASS   \n",
       "\n",
       "                                                 INFO  \\\n",
       "0   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "1   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "2   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "3   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "4   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "5   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "6   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "7   ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "8   ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "9   ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "10  ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "11  ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "12  ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....   \n",
       "13  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "14  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "15  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "16  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "17  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "18  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "\n",
       "                                               FORMAT  \\\n",
       "0   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "6   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "7   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "8   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "9   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "10  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "11  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "12  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "13  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "14  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "15  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "16  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "17  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "18  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                               SAMPLE      Zygosity  \\\n",
       "0   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "1   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "2   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "3   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "4   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "5   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "6   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "7   0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "8   0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "9   0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "10  0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "11  0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "12  1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...    Homozygous   \n",
       "13  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "14  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "15  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "16  0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5  Heterozygous   \n",
       "17  0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5  Heterozygous   \n",
       "18  0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5  Heterozygous   \n",
       "\n",
       "   rsID_updated  Haplotype  \n",
       "0     rs1041983     NAT2*4  \n",
       "1     rs1041983    NAT2*6A  \n",
       "2     rs1041983    NAT2*6J  \n",
       "3     rs1041983    NAT2*6O  \n",
       "4     rs1041983    NAT2*7B  \n",
       "5     rs1041983    NAT2*7G  \n",
       "6     rs1041983   NAT2*13A  \n",
       "7     rs1799930     NAT2*4  \n",
       "8     rs1799930    NAT2*6A  \n",
       "9     rs1799930    NAT2*6B  \n",
       "10    rs1799930    NAT2*6J  \n",
       "11    rs1799930    NAT2*6O  \n",
       "12   rs12769205  CYP2C19*2  \n",
       "13    rs3745274   CYP2B6*6  \n",
       "14    rs3745274   CYP2B6*9  \n",
       "15    rs3745274  CYP2B6*26  \n",
       "16    rs2279343   CYP2B6*4  \n",
       "17    rs2279343   CYP2B6*6  \n",
       "18    rs2279343  CYP2B6*26  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.merge(df, df2, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "54164d42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>rsID_updated</th>\n",
       "      <th>Haplotype</th>\n",
       "      <th>rsID_mapped</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*4</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*6O</td>\n",
       "      <td>rs1799930,rs1041983</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*7G</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>NAT2*13A</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2*6B</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94775367</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>CYP2B6*9</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>CYP2B6*4</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS        rsID REF ALT QUAL FILTER  \\\n",
       "0    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "1    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "2    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "3    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "4    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "5    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "6    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "9    chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "12  chr10  94775367  rs12769205   A   G    .   PASS   \n",
       "13  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "14  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "15  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "16  chr19  41009358   rs2279343   A   G    .   PASS   \n",
       "\n",
       "                                                 INFO  \\\n",
       "0   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "1   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "2   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "3   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "4   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "5   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "6   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "9   ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "12  ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....   \n",
       "13  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "14  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "15  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "16  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "\n",
       "                                               FORMAT  \\\n",
       "0   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "6   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "9   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "12  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "13  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "14  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "15  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "16  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                               SAMPLE      Zygosity  \\\n",
       "0   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "1   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "2   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "3   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "4   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "5   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "6   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "9   0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "12  1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...    Homozygous   \n",
       "13  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "14  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "15  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "16  0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5  Heterozygous   \n",
       "\n",
       "   rsID_updated  Haplotype          rsID_mapped            Zygosity_mapped  \n",
       "0     rs1041983     NAT2*4  rs1799930,rs1041983  Heterozygous,Heterozygous  \n",
       "1     rs1041983    NAT2*6A  rs1799930,rs1041983  Heterozygous,Heterozygous  \n",
       "2     rs1041983    NAT2*6J  rs1799930,rs1041983  Heterozygous,Heterozygous  \n",
       "3     rs1041983    NAT2*6O  rs1799930,rs1041983  Heterozygous,Heterozygous  \n",
       "4     rs1041983    NAT2*7B            rs1041983               Heterozygous  \n",
       "5     rs1041983    NAT2*7G            rs1041983               Heterozygous  \n",
       "6     rs1041983   NAT2*13A            rs1041983               Heterozygous  \n",
       "9     rs1799930    NAT2*6B            rs1799930               Heterozygous  \n",
       "12   rs12769205  CYP2C19*2           rs12769205                 Homozygous  \n",
       "13    rs3745274   CYP2B6*6  rs2279343,rs3745274  Heterozygous,Heterozygous  \n",
       "14    rs3745274   CYP2B6*9            rs3745274               Heterozygous  \n",
       "15    rs3745274  CYP2B6*26  rs2279343,rs3745274  Heterozygous,Heterozygous  \n",
       "16    rs2279343   CYP2B6*4            rs2279343               Heterozygous  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "madhu = merged.copy()\n",
    "madhu['rsID_mapped'] = madhu.groupby('Haplotype')['rsID'].transform(lambda x: ','.join(set(x)))\n",
    "madhu['Zygosity_mapped'] = madhu.groupby(['Haplotype'])['Zygosity'].transform(lambda x: ','.join(x))\n",
    "madhu.drop_duplicates(subset='Haplotype', inplace=True)\n",
    "madhu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "ff9455b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1041983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400593</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94775367</td>\n",
       "      <td>rs12769205</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>rs12769205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs3745274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41009358</td>\n",
       "      <td>rs2279343</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>rs2279343</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS        rsID REF ALT QUAL FILTER  \\\n",
       "0    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "1    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "2    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "3    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "4    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "5    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "6    chr8  18400285   rs1041983   C   T    .   PASS   \n",
       "9    chr8  18400593   rs1799930   G   A    .   PASS   \n",
       "12  chr10  94775367  rs12769205   A   G    .   PASS   \n",
       "13  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "14  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "15  chr19  41006936   rs3745274   G   T    .   PASS   \n",
       "16  chr19  41009358   rs2279343   A   G    .   PASS   \n",
       "\n",
       "                                                 INFO  \\\n",
       "0   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "1   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "2   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "3   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "4   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "5   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "6   ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "9   ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.735,0.2...   \n",
       "12  ADP=51;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7716,0....   \n",
       "13  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "14  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "15  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "16  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "\n",
       "                                               FORMAT  \\\n",
       "0   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "6   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "9   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "12  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "13  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "14  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "15  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "16  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                               SAMPLE      Zygosity  \\\n",
       "0   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "1   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "2   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "3   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "4   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "5   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "6   0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...  Heterozygous   \n",
       "9   0/1:57:35:35:19:16:45.71%:1.637E-6:60:54:15:4:...  Heterozygous   \n",
       "12  1/1:255:51:51:1:50:98.04%:1.3013E-28:74:51:1:0...    Homozygous   \n",
       "13  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "14  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "15  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...  Heterozygous   \n",
       "16  0/1:31:22:22:13:9:40.91%:7.0165E-4:60:52:12:1:4:5  Heterozygous   \n",
       "\n",
       "              Zygosity_mapped   rsID_mapped_sorted  \n",
       "0   Heterozygous,Heterozygous  rs1041983,rs1799930  \n",
       "1   Heterozygous,Heterozygous  rs1041983,rs1799930  \n",
       "2   Heterozygous,Heterozygous  rs1041983,rs1799930  \n",
       "3   Heterozygous,Heterozygous  rs1041983,rs1799930  \n",
       "4                Heterozygous            rs1041983  \n",
       "5                Heterozygous            rs1041983  \n",
       "6                Heterozygous            rs1041983  \n",
       "9                Heterozygous            rs1799930  \n",
       "12                 Homozygous           rs12769205  \n",
       "13  Heterozygous,Heterozygous  rs2279343,rs3745274  \n",
       "14               Heterozygous            rs3745274  \n",
       "15  Heterozygous,Heterozygous  rs2279343,rs3745274  \n",
       "16               Heterozygous            rs2279343  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "madhu['rsID_mapped_sorted'] = madhu['rsID_mapped'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "madhu = madhu.drop(columns=['rsID_mapped', 'rsID_updated', 'Haplotype'], axis=1)\n",
    "madhu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "e9158c77",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Haplotype_updated</th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NAT2*7G</td>\n",
       "      <td>chr8_18400229_T_G,rs1041983,rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP2D6*10</td>\n",
       "      <td>rs1058164,rs1065852,rs1135840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CYP2D6*2</td>\n",
       "      <td>rs1058164,rs1135840,rs16947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>rs1041983,rs1799930,rs1799931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NAT2*6O</td>\n",
       "      <td>rs1041983,rs1799930,rs56393504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TPMT*3A</td>\n",
       "      <td>rs1142345,rs1800460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>rs1208,rs1801280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SLCO1B1*14</td>\n",
       "      <td>rs11045819,rs2306283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>rs2306283,rs4149056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CYP2A6*9</td>\n",
       "      <td>rs28399433,rs7248240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CYP2A6*25</td>\n",
       "      <td>rs28399440,rs7248240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CYP2A6*23</td>\n",
       "      <td>rs56256500,rs7248240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NUDT15*2</td>\n",
       "      <td>rs116855232,rs746071566</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Haplotype_updated                     rsID_mapped_sorted\n",
       "0            NAT2*7B                    rs1041983,rs1799931\n",
       "1            NAT2*7G  chr8_18400229_T_G,rs1041983,rs1799931\n",
       "2          CYP2D6*10          rs1058164,rs1065852,rs1135840\n",
       "3           CYP2D6*2            rs1058164,rs1135840,rs16947\n",
       "4            NAT2*6A                    rs1041983,rs1799930\n",
       "5            NAT2*6J          rs1041983,rs1799930,rs1799931\n",
       "6            NAT2*6O         rs1041983,rs1799930,rs56393504\n",
       "7            TPMT*3A                    rs1142345,rs1800460\n",
       "8            NAT2*5C                       rs1208,rs1801280\n",
       "9            NAT2*5A                    rs1799929,rs1801280\n",
       "10           NAT2*5B             rs1208,rs1799929,rs1801280\n",
       "11        SLCO1B1*14                   rs11045819,rs2306283\n",
       "12        SLCO1B1*15                    rs2306283,rs4149056\n",
       "13          CYP2A6*9                   rs28399433,rs7248240\n",
       "14         CYP2A6*25                   rs28399440,rs7248240\n",
       "15          CYP2B6*6                    rs2279343,rs3745274\n",
       "16         CYP2B6*26                    rs2279343,rs3745274\n",
       "17         CYP2C19*2                   rs12769205,rs4244285\n",
       "18         CYP2A6*23                   rs56256500,rs7248240\n",
       "19          NUDT15*2                rs116855232,rs746071566"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Multiple_Positions.xlsx')\n",
    "data = data.rename({'Haplotype':'Haplotype_updated', 'rsID':'rsID_mapped'}, axis = 1)\n",
    "data['rsID_mapped_sorted'] = data['rsID_mapped'].apply(lambda x: ','.join(sorted(x.split(','))))\n",
    "data = data.drop(columns=['rsID_mapped'], axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "67e9cbed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "      <th>Haplotype_updated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400285</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>NAT2*6A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006936</td>\n",
       "      <td>rs3745274</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0   chr8  18400285  rs1041983   C   T    .   PASS   \n",
       "4  chr19  41006936  rs3745274   G   T    .   PASS   \n",
       "5  chr19  41006936  rs3745274   G   T    .   PASS   \n",
       "\n",
       "                                                INFO  \\\n",
       "0  ADP=34;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6026,0....   \n",
       "4  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "5  ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.684...   \n",
       "\n",
       "                                              FORMAT  \\\n",
       "0  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                              SAMPLE  \\\n",
       "0  0/1:77:34:34:14:20:58.82%:1.6951E-8:58:56:13:1...   \n",
       "4  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...   \n",
       "5  0/1:35:22:22:12:10:45.45%:2.6061E-4:61:55:11:1...   \n",
       "\n",
       "             Zygosity_mapped   rsID_mapped_sorted Haplotype_updated  \n",
       "0  Heterozygous,Heterozygous  rs1041983,rs1799930           NAT2*6A  \n",
       "4  Heterozygous,Heterozygous  rs2279343,rs3745274          CYP2B6*6  \n",
       "5  Heterozygous,Heterozygous  rs2279343,rs3745274         CYP2B6*26  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapped_df = pd.merge(madhu, data, on = 'rsID_mapped_sorted', how = 'inner', sort = False)\n",
    "mapped_df.drop_duplicates(subset='Haplotype_updated', inplace=True)\n",
    "mapped_df = mapped_df.drop(columns=['Zygosity'], axis=1)\n",
    "mapped_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af985724",
   "metadata": {},
   "source": [
    "# zygosity classes excel files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "6cc30d05",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excel files saved successfully.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Assuming df is your DataFrame with 'Zygosity' column\n",
    "# You may need to replace 'your_data.csv' with your actual data file\n",
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/All_1232_updated_haplo_data_16_12_2023.xlsx')\n",
    "\n",
    "# Get unique values in the 'Zygosity' column\n",
    "zygosity_values = df['Zygosity_mapped'].unique()\n",
    "\n",
    "# Create a folder to save the Excel files\n",
    "output_folder = r'C:/Users/GenepoweRx_Madhu/Downloads/new_haplo_data_16_12_2023/'\n",
    "os.makedirs(output_folder, exist_ok=True)\n",
    "\n",
    "# Iterate through unique zygosity values\n",
    "for zygosity_value in zygosity_values:\n",
    "    # Create a subset DataFrame for each zygosity value\n",
    "    subset_df = df[df['Zygosity_mapped'] == zygosity_value]\n",
    "\n",
    "    # Save the subset as an Excel file\n",
    "    output_filename = os.path.join(output_folder, f'{zygosity_value}.xlsx')\n",
    "    subset_df.to_excel(output_filename, index=False)\n",
    "\n",
    "print(\"Excel files saved successfully.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "26b2a171",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>Haplotype_updated</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1208,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3763</th>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX1005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3764</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX1002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3765</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3766</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3767</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3768 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              rsID_mapped_sorted                         Zygosity_mapped  \\\n",
       "0            rs1041983,rs1799931               Heterozygous,Heterozygous   \n",
       "1            rs1799929,rs1801280               Heterozygous,Heterozygous   \n",
       "2     rs1208,rs1799929,rs1801280  Heterozygous,Heterozygous,Heterozygous   \n",
       "3               rs1208,rs1801280               Heterozygous,Heterozygous   \n",
       "4            rs1041983,rs1799930               Heterozygous,Heterozygous   \n",
       "...                          ...                                     ...   \n",
       "3763        rs12769205,rs4244285               Heterozygous,Heterozygous   \n",
       "3764         rs1799929,rs1801280                   Homozygous,Homozygous   \n",
       "3765         rs1041983,rs1799931               Heterozygous,Heterozygous   \n",
       "3766         rs1041983,rs1799930               Heterozygous,Heterozygous   \n",
       "3767         rs1799929,rs1801280               Heterozygous,Heterozygous   \n",
       "\n",
       "     Haplotype_updated        Sample  \n",
       "0              NAT2*7B     KHAIGPRX9  \n",
       "1              NAT2*5A     KHAIGPRX9  \n",
       "2              NAT2*5B     KHAIGPRX9  \n",
       "3              NAT2*5C     KHAIGPRX9  \n",
       "4              NAT2*6A    KHAIGPRX99  \n",
       "...                ...           ...  \n",
       "3763         CYP2C19*2  KHAIGPRX1005  \n",
       "3764           NAT2*5A  KHAIGPRX1002  \n",
       "3765           NAT2*7B  KHAIGPRX1001  \n",
       "3766           NAT2*6A  KHAIGPRX1000  \n",
       "3767           NAT2*5A  KHAIGPRX1000  \n",
       "\n",
       "[3768 rows x 4 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/All_1232_updated_haplo_data_16_12_2023.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "2dff59a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>Haplotype_updated</th>\n",
       "      <th>Sample</th>\n",
       "      <th>rsID_new</th>\n",
       "      <th>Zygosity_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "      <td>rs1799929</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX9</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3765</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX1001</td>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3766</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX1000</td>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3766</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX1000</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3767</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX1000</td>\n",
       "      <td>rs1799929</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3767</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX1000</td>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5649 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              rsID_mapped_sorted                         Zygosity_mapped  \\\n",
       "0            rs1041983,rs1799931               Heterozygous,Heterozygous   \n",
       "0            rs1041983,rs1799931               Heterozygous,Heterozygous   \n",
       "1            rs1799929,rs1801280               Heterozygous,Heterozygous   \n",
       "1            rs1799929,rs1801280               Heterozygous,Heterozygous   \n",
       "2     rs1208,rs1799929,rs1801280  Heterozygous,Heterozygous,Heterozygous   \n",
       "...                          ...                                     ...   \n",
       "3765         rs1041983,rs1799931               Heterozygous,Heterozygous   \n",
       "3766         rs1041983,rs1799930               Heterozygous,Heterozygous   \n",
       "3766         rs1041983,rs1799930               Heterozygous,Heterozygous   \n",
       "3767         rs1799929,rs1801280               Heterozygous,Heterozygous   \n",
       "3767         rs1799929,rs1801280               Heterozygous,Heterozygous   \n",
       "\n",
       "     Haplotype_updated        Sample   rsID_new  Zygosity_new  \n",
       "0              NAT2*7B     KHAIGPRX9  rs1041983  Heterozygous  \n",
       "0              NAT2*7B     KHAIGPRX9  rs1799931  Heterozygous  \n",
       "1              NAT2*5A     KHAIGPRX9  rs1799929  Heterozygous  \n",
       "1              NAT2*5A     KHAIGPRX9  rs1801280  Heterozygous  \n",
       "2              NAT2*5B     KHAIGPRX9     rs1208  Heterozygous  \n",
       "...                ...           ...        ...           ...  \n",
       "3765           NAT2*7B  KHAIGPRX1001  rs1799931  Heterozygous  \n",
       "3766           NAT2*6A  KHAIGPRX1000  rs1041983  Heterozygous  \n",
       "3766           NAT2*6A  KHAIGPRX1000  rs1799930  Heterozygous  \n",
       "3767           NAT2*5A  KHAIGPRX1000  rs1799929  Heterozygous  \n",
       "3767           NAT2*5A  KHAIGPRX1000  rs1801280  Heterozygous  \n",
       "\n",
       "[5649 rows x 6 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df = df.copy()\n",
    "new_df['rsID_new'] = new_df['rsID_mapped_sorted'].str.split(',')\n",
    "new_df['Zygosity_new'] = new_df['Zygosity_mapped'].str.split(',')\n",
    "# Explode the lists into separate rows\n",
    "new_df = new_df.explode('rsID_new')\n",
    "new_df = new_df.explode('Zygosity_new')\n",
    "new_df.drop_duplicates(subset=['rsID_new', 'Zygosity_new', 'Sample'], inplace=True)\n",
    "new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "655e7302",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID_new</th>\n",
       "      <th>Zygosity_new</th>\n",
       "      <th>Haplotype_updated</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Sample_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX99,KHAIGPRX997,KHAIGPRX995,KHAIGPRX991...</td>\n",
       "      <td>488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX9,KHAIGPRX983,KHAIGPRX97,KHAIGPRX951,K...</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX999,KHAIGPRX998,KHAIGPRX989,KHAIGPRX97...</td>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs1041983</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX97,KHAIGPRX90,KHAIGPRX758,KHAIGPRX662,...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs11045819</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLCO1B1*14</td>\n",
       "      <td>KHAIGPRX952_RS,KHAIGPRX94,KHAIGPRX932,KHAIGPRX...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs11045819</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SLCO1B1*14</td>\n",
       "      <td>KHAIGPRX882,KHAIGPRX852,KHAIGPRX821,KHAIGPRX77...</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs1142345</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TPMT*3A</td>\n",
       "      <td>KHAIGPRX636,KHAIGPRX503,KHAIGPRX446,KHAIGPRX37...</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs1208</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX9,KHAIGPRX99,KHAIGPRX997,KHAIGPRX992,K...</td>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>rs1208</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX996,KHAIGPRX984,KHAIGPRX977,KHAIGPRX97...</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX937,KHAIGPRX910,KHAIGPRX760,KHAIGPRX64...</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>rs1208</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX380</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>rs12769205</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX99,KHAIGPRX997,KHAIGPRX989,KHAIGPRX986...</td>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>rs12769205</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX935,KHAIGPRX855,KHAIGPRX81,KHAIGPRX818...</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>rs1799929</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX9,KHAIGPRX99,KHAIGPRX997,KHAIGPRX992,K...</td>\n",
       "      <td>343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>rs1799929</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX949,KHAIGPRX937,KHAIGPRX933,KHAIGPRX93...</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>rs1799929</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX993,KHAIGPRX985,KHAIGPRX967,KHAIGPRX95...</td>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>rs1799929</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX937,KHAIGPRX910,KHAIGPRX760,KHAIGPRX64...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX99,KHAIGPRX997,KHAIGPRX995,KHAIGPRX991...</td>\n",
       "      <td>488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX999,KHAIGPRX998,KHAIGPRX989,KHAIGPRX97...</td>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>rs1799930</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>KHAIGPRX989,KHAIGPRX959,KHAIGPRX909,KHAIGPRX87...</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX9,KHAIGPRX983,KHAIGPRX97,KHAIGPRX951,K...</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>KHAIGPRX989,KHAIGPRX959,KHAIGPRX909,KHAIGPRX87...</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>rs1799931</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX97,KHAIGPRX90,KHAIGPRX758,KHAIGPRX662,...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>rs1800460</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TPMT*3A</td>\n",
       "      <td>KHAIGPRX636,KHAIGPRX503,KHAIGPRX446,KHAIGPRX37...</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX9,KHAIGPRX99,KHAIGPRX997,KHAIGPRX992,K...</td>\n",
       "      <td>343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX949,KHAIGPRX937,KHAIGPRX933,KHAIGPRX93...</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX996,KHAIGPRX984,KHAIGPRX977,KHAIGPRX97...</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX993,KHAIGPRX985,KHAIGPRX967,KHAIGPRX95...</td>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX937,KHAIGPRX910,KHAIGPRX760,KHAIGPRX64...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>rs1801280</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX380</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX99,KHAIGPRX990,KHAIGPRX98,KHAIGPRX981,...</td>\n",
       "      <td>366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>rs2279343</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX971,KHAIGPRX955,KHAIGPRX942,KHAIGPRX89...</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>rs2306283</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLCO1B1*14</td>\n",
       "      <td>KHAIGPRX952_RS,KHAIGPRX94,KHAIGPRX932,KHAIGPRX...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>rs2306283</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>KHAIGPRX972,KHAIGPRX941,KHAIGPRX940,KHAIGPRX92...</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>rs2306283</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SLCO1B1*14</td>\n",
       "      <td>KHAIGPRX882,KHAIGPRX852,KHAIGPRX821,KHAIGPRX77...</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>rs2306283</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>KHAIGPRX972,KHAIGPRX95,KHAIGPRX941,KHAIGPRX928...</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>rs3745274</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX99,KHAIGPRX990,KHAIGPRX98,KHAIGPRX981,...</td>\n",
       "      <td>366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>rs3745274</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX971,KHAIGPRX955,KHAIGPRX942,KHAIGPRX89...</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>rs4149056</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>KHAIGPRX972,KHAIGPRX94,KHAIGPRX941,KHAIGPRX940...</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>rs4149056</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>KHAIGPRX972,KHAIGPRX95,KHAIGPRX941,KHAIGPRX928...</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>rs4244285</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX99,KHAIGPRX997,KHAIGPRX989,KHAIGPRX986...</td>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>rs4244285</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX935,KHAIGPRX855,KHAIGPRX81,KHAIGPRX818...</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>rs56393504</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NAT2*6O</td>\n",
       "      <td>KHAIGPRX370</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      rsID_new  Zygosity_new Haplotype_updated  \\\n",
       "0    rs1041983  Heterozygous           NAT2*6A   \n",
       "1    rs1041983  Heterozygous           NAT2*6J   \n",
       "2    rs1041983  Heterozygous           NAT2*7B   \n",
       "3    rs1041983    Homozygous           NAT2*6A   \n",
       "4    rs1041983    Homozygous           NAT2*6J   \n",
       "5    rs1041983    Homozygous           NAT2*7B   \n",
       "6   rs11045819  Heterozygous        SLCO1B1*14   \n",
       "7   rs11045819    Homozygous        SLCO1B1*14   \n",
       "8    rs1142345  Heterozygous           TPMT*3A   \n",
       "9       rs1208  Heterozygous           NAT2*5B   \n",
       "10      rs1208  Heterozygous           NAT2*5C   \n",
       "11      rs1208    Homozygous           NAT2*5B   \n",
       "12      rs1208    Homozygous           NAT2*5C   \n",
       "13  rs12769205  Heterozygous         CYP2C19*2   \n",
       "14  rs12769205    Homozygous         CYP2C19*2   \n",
       "15   rs1799929  Heterozygous           NAT2*5A   \n",
       "16   rs1799929  Heterozygous           NAT2*5B   \n",
       "17   rs1799929    Homozygous           NAT2*5A   \n",
       "18   rs1799929    Homozygous           NAT2*5B   \n",
       "19   rs1799930  Heterozygous           NAT2*6A   \n",
       "20   rs1799930  Heterozygous           NAT2*6J   \n",
       "21   rs1799930    Homozygous           NAT2*6A   \n",
       "22   rs1799930    Homozygous           NAT2*6J   \n",
       "23   rs1799931  Heterozygous           NAT2*6J   \n",
       "24   rs1799931  Heterozygous           NAT2*7B   \n",
       "25   rs1799931    Homozygous           NAT2*6J   \n",
       "26   rs1799931    Homozygous           NAT2*7B   \n",
       "27   rs1800460  Heterozygous           TPMT*3A   \n",
       "28   rs1801280  Heterozygous           NAT2*5A   \n",
       "29   rs1801280  Heterozygous           NAT2*5B   \n",
       "30   rs1801280  Heterozygous           NAT2*5C   \n",
       "31   rs1801280    Homozygous           NAT2*5A   \n",
       "32   rs1801280    Homozygous           NAT2*5B   \n",
       "33   rs1801280    Homozygous           NAT2*5C   \n",
       "34   rs2279343  Heterozygous          CYP2B6*6   \n",
       "35   rs2279343    Homozygous          CYP2B6*6   \n",
       "36   rs2306283  Heterozygous        SLCO1B1*14   \n",
       "37   rs2306283  Heterozygous        SLCO1B1*15   \n",
       "38   rs2306283    Homozygous        SLCO1B1*14   \n",
       "39   rs2306283    Homozygous        SLCO1B1*15   \n",
       "40   rs3745274  Heterozygous          CYP2B6*6   \n",
       "41   rs3745274    Homozygous          CYP2B6*6   \n",
       "42   rs4149056  Heterozygous        SLCO1B1*15   \n",
       "43   rs4149056    Homozygous        SLCO1B1*15   \n",
       "44   rs4244285  Heterozygous         CYP2C19*2   \n",
       "45   rs4244285    Homozygous         CYP2C19*2   \n",
       "46  rs56393504  Heterozygous           NAT2*6O   \n",
       "\n",
       "                                               Sample  Sample_count  \n",
       "0   KHAIGPRX99,KHAIGPRX997,KHAIGPRX995,KHAIGPRX991...           488  \n",
       "1   KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049             4  \n",
       "2   KHAIGPRX9,KHAIGPRX983,KHAIGPRX97,KHAIGPRX951,K...            89  \n",
       "3   KHAIGPRX999,KHAIGPRX998,KHAIGPRX989,KHAIGPRX97...           178  \n",
       "4   KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049             4  \n",
       "5   KHAIGPRX97,KHAIGPRX90,KHAIGPRX758,KHAIGPRX662,...             7  \n",
       "6   KHAIGPRX952_RS,KHAIGPRX94,KHAIGPRX932,KHAIGPRX...            33  \n",
       "7   KHAIGPRX882,KHAIGPRX852,KHAIGPRX821,KHAIGPRX77...            18  \n",
       "8   KHAIGPRX636,KHAIGPRX503,KHAIGPRX446,KHAIGPRX37...             6  \n",
       "9   KHAIGPRX9,KHAIGPRX99,KHAIGPRX997,KHAIGPRX992,K...           422  \n",
       "10  KHAIGPRX996,KHAIGPRX984,KHAIGPRX977,KHAIGPRX97...            45  \n",
       "11  KHAIGPRX937,KHAIGPRX910,KHAIGPRX760,KHAIGPRX64...            13  \n",
       "12                                        KHAIGPRX380             1  \n",
       "13  KHAIGPRX99,KHAIGPRX997,KHAIGPRX989,KHAIGPRX986...           422  \n",
       "14  KHAIGPRX935,KHAIGPRX855,KHAIGPRX81,KHAIGPRX818...            35  \n",
       "15  KHAIGPRX9,KHAIGPRX99,KHAIGPRX997,KHAIGPRX992,K...           343  \n",
       "16  KHAIGPRX949,KHAIGPRX937,KHAIGPRX933,KHAIGPRX93...           137  \n",
       "17  KHAIGPRX993,KHAIGPRX985,KHAIGPRX967,KHAIGPRX95...           135  \n",
       "18  KHAIGPRX937,KHAIGPRX910,KHAIGPRX760,KHAIGPRX64...            12  \n",
       "19  KHAIGPRX99,KHAIGPRX997,KHAIGPRX995,KHAIGPRX991...           488  \n",
       "20  KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049             4  \n",
       "21  KHAIGPRX999,KHAIGPRX998,KHAIGPRX989,KHAIGPRX97...           178  \n",
       "22  KHAIGPRX559,KHAIGPRX544,KHAIGPRX1078,KHAIGPRX1049             4  \n",
       "23  KHAIGPRX989,KHAIGPRX959,KHAIGPRX909,KHAIGPRX87...            52  \n",
       "24  KHAIGPRX9,KHAIGPRX983,KHAIGPRX97,KHAIGPRX951,K...            89  \n",
       "25  KHAIGPRX989,KHAIGPRX959,KHAIGPRX909,KHAIGPRX87...            52  \n",
       "26  KHAIGPRX97,KHAIGPRX90,KHAIGPRX758,KHAIGPRX662,...             7  \n",
       "27  KHAIGPRX636,KHAIGPRX503,KHAIGPRX446,KHAIGPRX37...             6  \n",
       "28  KHAIGPRX9,KHAIGPRX99,KHAIGPRX997,KHAIGPRX992,K...           343  \n",
       "29  KHAIGPRX949,KHAIGPRX937,KHAIGPRX933,KHAIGPRX93...           137  \n",
       "30  KHAIGPRX996,KHAIGPRX984,KHAIGPRX977,KHAIGPRX97...            45  \n",
       "31  KHAIGPRX993,KHAIGPRX985,KHAIGPRX967,KHAIGPRX95...           135  \n",
       "32  KHAIGPRX937,KHAIGPRX910,KHAIGPRX760,KHAIGPRX64...            12  \n",
       "33                                        KHAIGPRX380             1  \n",
       "34  KHAIGPRX99,KHAIGPRX990,KHAIGPRX98,KHAIGPRX981,...           366  \n",
       "35  KHAIGPRX971,KHAIGPRX955,KHAIGPRX942,KHAIGPRX89...           102  \n",
       "36  KHAIGPRX952_RS,KHAIGPRX94,KHAIGPRX932,KHAIGPRX...            33  \n",
       "37  KHAIGPRX972,KHAIGPRX941,KHAIGPRX940,KHAIGPRX92...            81  \n",
       "38  KHAIGPRX882,KHAIGPRX852,KHAIGPRX821,KHAIGPRX77...            18  \n",
       "39  KHAIGPRX972,KHAIGPRX95,KHAIGPRX941,KHAIGPRX928...            46  \n",
       "40  KHAIGPRX99,KHAIGPRX990,KHAIGPRX98,KHAIGPRX981,...           366  \n",
       "41  KHAIGPRX971,KHAIGPRX955,KHAIGPRX942,KHAIGPRX89...           102  \n",
       "42  KHAIGPRX972,KHAIGPRX94,KHAIGPRX941,KHAIGPRX940...            84  \n",
       "43  KHAIGPRX972,KHAIGPRX95,KHAIGPRX941,KHAIGPRX928...            48  \n",
       "44  KHAIGPRX99,KHAIGPRX997,KHAIGPRX989,KHAIGPRX986...           422  \n",
       "45  KHAIGPRX935,KHAIGPRX855,KHAIGPRX81,KHAIGPRX818...            35  \n",
       "46                                        KHAIGPRX370             1  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_new = new_df.groupby(['rsID_new', 'Zygosity_new', 'Haplotype_updated']).agg({'Sample': lambda x: ','.join(x.unique())}).reset_index()\n",
    "grouped_new['Sample_count'] = grouped_new['Sample'].apply(lambda x: len(x.split(',')))\n",
    "grouped_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "37700b44",
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_new.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/haplotype_rsID_data.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14285562",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "e6087cea",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID_mapped_sorted</th>\n",
       "      <th>Zygosity_mapped</th>\n",
       "      <th>Haplotype_updated</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Sample_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX99, KHAIGPRX997, KHAIGPRX995, KHAIGPRX...</td>\n",
       "      <td>434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Heterozygous,Homozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX431, KHAIGPRX279</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX989, KHAIGPRX959, KHAIGPRX909, KHAIGPR...</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1041983,rs1799930</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>NAT2*6A</td>\n",
       "      <td>KHAIGPRX999, KHAIGPRX998, KHAIGPRX972, KHAIGPR...</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1041983,rs1799930,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6J</td>\n",
       "      <td>KHAIGPRX989, KHAIGPRX959, KHAIGPRX909, KHAIGPR...</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs1041983,rs1799930,rs56393504</td>\n",
       "      <td>Heterozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*6O</td>\n",
       "      <td>KHAIGPRX370</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX9, KHAIGPRX983, KHAIGPRX951, KHAIGPRX9...</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX989, KHAIGPRX97, KHAIGPRX959, KHAIGPRX...</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs1041983,rs1799931</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>NAT2*7B</td>\n",
       "      <td>KHAIGPRX90, KHAIGPRX758, KHAIGPRX662, KHAIGPRX456</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs11045819,rs2306283</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>SLCO1B1*14</td>\n",
       "      <td>KHAIGPRX952_RS, KHAIGPRX94, KHAIGPRX932, KHAIG...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>rs11045819,rs2306283</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>SLCO1B1*14</td>\n",
       "      <td>KHAIGPRX882, KHAIGPRX852, KHAIGPRX821, KHAIGPR...</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rs1142345,rs1800460</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>TPMT*3A</td>\n",
       "      <td>KHAIGPRX636, KHAIGPRX503, KHAIGPRX446, KHAIGPR...</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX9, KHAIGPRX99, KHAIGPRX997, KHAIGPRX99...</td>\n",
       "      <td>410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous,Homozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX937, KHAIGPRX910, KHAIGPRX286, KHAIGPR...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX760, KHAIGPRX239, KHAIGPRX1005</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Homozygous,Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX288</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Homozygous,Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX647, KHAIGPRX570, KHAIGPRX524, KHAIGPR...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>rs1208,rs1799929,rs1801280</td>\n",
       "      <td>Homozygous,Homozygous,Homozygous</td>\n",
       "      <td>NAT2*5B</td>\n",
       "      <td>KHAIGPRX1006_RS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>rs1208,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX9, KHAIGPRX99, KHAIGPRX997, KHAIGPRX99...</td>\n",
       "      <td>457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>rs1208,rs1801280</td>\n",
       "      <td>Heterozygous,Homozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX937, KHAIGPRX910, KHAIGPRX380, KHAIGPR...</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>rs1208,rs1801280</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX647, KHAIGPRX570, KHAIGPRX524, KHAIGPR...</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>rs1208,rs1801280</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>NAT2*5C</td>\n",
       "      <td>KHAIGPRX1006_RS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX99, KHAIGPRX997, KHAIGPRX989, KHAIGPRX...</td>\n",
       "      <td>418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Heterozygous,Homozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX715, KHAIGPRX568, KHAIGPRX1058</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX175</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>rs12769205,rs4244285</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>CYP2C19*2</td>\n",
       "      <td>KHAIGPRX935, KHAIGPRX855, KHAIGPRX81, KHAIGPRX...</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX9, KHAIGPRX99, KHAIGPRX997, KHAIGPRX99...</td>\n",
       "      <td>458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Heterozygous,Homozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX760, KHAIGPRX239, KHAIGPRX1005</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX985, KHAIGPRX921, KHAIGPRX906, KHAIGPR...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rs1799929,rs1801280</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>NAT2*5A</td>\n",
       "      <td>KHAIGPRX993, KHAIGPRX967, KHAIGPRX956, KHAIGPR...</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>KHAIGPRX99, KHAIGPRX990, KHAIGPRX98, KHAIGPRX9...</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX99, KHAIGPRX990, KHAIGPRX98, KHAIGPRX9...</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Heterozygous,Homozygous</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>KHAIGPRX897, KHAIGPRX894, KHAIGPRX815, KHAIGPR...</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Heterozygous,Homozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX897, KHAIGPRX894, KHAIGPRX815, KHAIGPR...</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>KHAIGPRX704, KHAIGPRX119</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX704, KHAIGPRX119</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>CYP2B6*26</td>\n",
       "      <td>KHAIGPRX971, KHAIGPRX955, KHAIGPRX942, KHAIGPR...</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>rs2279343,rs3745274</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>CYP2B6*6</td>\n",
       "      <td>KHAIGPRX971, KHAIGPRX955, KHAIGPRX942, KHAIGPR...</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>rs2306283,rs4149056</td>\n",
       "      <td>Heterozygous,Heterozygous</td>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>KHAIGPRX94, KHAIGPRX940, KHAIGPRX91, KHAIGPRX9...</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>rs2306283,rs4149056</td>\n",
       "      <td>Homozygous,Heterozygous</td>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>KHAIGPRX972, KHAIGPRX941, KHAIGPRX928, KHAIGPR...</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>rs2306283,rs4149056</td>\n",
       "      <td>Homozygous,Homozygous</td>\n",
       "      <td>SLCO1B1*15</td>\n",
       "      <td>KHAIGPRX95, KHAIGPRX414, KHAIGPRX409, KHAIGPRX...</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                rsID_mapped_sorted                         Zygosity_mapped  \\\n",
       "0              rs1041983,rs1799930               Heterozygous,Heterozygous   \n",
       "1              rs1041983,rs1799930                 Heterozygous,Homozygous   \n",
       "2              rs1041983,rs1799930                 Homozygous,Heterozygous   \n",
       "3              rs1041983,rs1799930                   Homozygous,Homozygous   \n",
       "4    rs1041983,rs1799930,rs1799931    Homozygous,Heterozygous,Heterozygous   \n",
       "5   rs1041983,rs1799930,rs56393504  Heterozygous,Heterozygous,Heterozygous   \n",
       "6              rs1041983,rs1799931               Heterozygous,Heterozygous   \n",
       "7              rs1041983,rs1799931                 Homozygous,Heterozygous   \n",
       "8              rs1041983,rs1799931                   Homozygous,Homozygous   \n",
       "9             rs11045819,rs2306283               Heterozygous,Heterozygous   \n",
       "10            rs11045819,rs2306283                 Homozygous,Heterozygous   \n",
       "11             rs1142345,rs1800460               Heterozygous,Heterozygous   \n",
       "12      rs1208,rs1799929,rs1801280  Heterozygous,Heterozygous,Heterozygous   \n",
       "13      rs1208,rs1799929,rs1801280    Heterozygous,Heterozygous,Homozygous   \n",
       "14      rs1208,rs1799929,rs1801280    Heterozygous,Homozygous,Heterozygous   \n",
       "15      rs1208,rs1799929,rs1801280    Homozygous,Heterozygous,Heterozygous   \n",
       "16      rs1208,rs1799929,rs1801280      Homozygous,Homozygous,Heterozygous   \n",
       "17      rs1208,rs1799929,rs1801280        Homozygous,Homozygous,Homozygous   \n",
       "18                rs1208,rs1801280               Heterozygous,Heterozygous   \n",
       "19                rs1208,rs1801280                 Heterozygous,Homozygous   \n",
       "20                rs1208,rs1801280                 Homozygous,Heterozygous   \n",
       "21                rs1208,rs1801280                   Homozygous,Homozygous   \n",
       "22            rs12769205,rs4244285               Heterozygous,Heterozygous   \n",
       "23            rs12769205,rs4244285                 Heterozygous,Homozygous   \n",
       "24            rs12769205,rs4244285                 Homozygous,Heterozygous   \n",
       "25            rs12769205,rs4244285                   Homozygous,Homozygous   \n",
       "26             rs1799929,rs1801280               Heterozygous,Heterozygous   \n",
       "27             rs1799929,rs1801280                 Heterozygous,Homozygous   \n",
       "28             rs1799929,rs1801280                 Homozygous,Heterozygous   \n",
       "29             rs1799929,rs1801280                   Homozygous,Homozygous   \n",
       "30             rs2279343,rs3745274               Heterozygous,Heterozygous   \n",
       "31             rs2279343,rs3745274               Heterozygous,Heterozygous   \n",
       "32             rs2279343,rs3745274                 Heterozygous,Homozygous   \n",
       "33             rs2279343,rs3745274                 Heterozygous,Homozygous   \n",
       "34             rs2279343,rs3745274                 Homozygous,Heterozygous   \n",
       "35             rs2279343,rs3745274                 Homozygous,Heterozygous   \n",
       "36             rs2279343,rs3745274                   Homozygous,Homozygous   \n",
       "37             rs2279343,rs3745274                   Homozygous,Homozygous   \n",
       "38             rs2306283,rs4149056               Heterozygous,Heterozygous   \n",
       "39             rs2306283,rs4149056                 Homozygous,Heterozygous   \n",
       "40             rs2306283,rs4149056                   Homozygous,Homozygous   \n",
       "\n",
       "   Haplotype_updated                                             Sample  \\\n",
       "0            NAT2*6A  KHAIGPRX99, KHAIGPRX997, KHAIGPRX995, KHAIGPRX...   \n",
       "1            NAT2*6A                           KHAIGPRX431, KHAIGPRX279   \n",
       "2            NAT2*6A  KHAIGPRX989, KHAIGPRX959, KHAIGPRX909, KHAIGPR...   \n",
       "3            NAT2*6A  KHAIGPRX999, KHAIGPRX998, KHAIGPRX972, KHAIGPR...   \n",
       "4            NAT2*6J  KHAIGPRX989, KHAIGPRX959, KHAIGPRX909, KHAIGPR...   \n",
       "5            NAT2*6O                                        KHAIGPRX370   \n",
       "6            NAT2*7B  KHAIGPRX9, KHAIGPRX983, KHAIGPRX951, KHAIGPRX9...   \n",
       "7            NAT2*7B  KHAIGPRX989, KHAIGPRX97, KHAIGPRX959, KHAIGPRX...   \n",
       "8            NAT2*7B  KHAIGPRX90, KHAIGPRX758, KHAIGPRX662, KHAIGPRX456   \n",
       "9         SLCO1B1*14  KHAIGPRX952_RS, KHAIGPRX94, KHAIGPRX932, KHAIG...   \n",
       "10        SLCO1B1*14  KHAIGPRX882, KHAIGPRX852, KHAIGPRX821, KHAIGPR...   \n",
       "11           TPMT*3A  KHAIGPRX636, KHAIGPRX503, KHAIGPRX446, KHAIGPR...   \n",
       "12           NAT2*5B  KHAIGPRX9, KHAIGPRX99, KHAIGPRX997, KHAIGPRX99...   \n",
       "13           NAT2*5B  KHAIGPRX937, KHAIGPRX910, KHAIGPRX286, KHAIGPR...   \n",
       "14           NAT2*5B             KHAIGPRX760, KHAIGPRX239, KHAIGPRX1005   \n",
       "15           NAT2*5B                                        KHAIGPRX288   \n",
       "16           NAT2*5B  KHAIGPRX647, KHAIGPRX570, KHAIGPRX524, KHAIGPR...   \n",
       "17           NAT2*5B                                    KHAIGPRX1006_RS   \n",
       "18           NAT2*5C  KHAIGPRX9, KHAIGPRX99, KHAIGPRX997, KHAIGPRX99...   \n",
       "19           NAT2*5C  KHAIGPRX937, KHAIGPRX910, KHAIGPRX380, KHAIGPR...   \n",
       "20           NAT2*5C  KHAIGPRX647, KHAIGPRX570, KHAIGPRX524, KHAIGPR...   \n",
       "21           NAT2*5C                                    KHAIGPRX1006_RS   \n",
       "22         CYP2C19*2  KHAIGPRX99, KHAIGPRX997, KHAIGPRX989, KHAIGPRX...   \n",
       "23         CYP2C19*2             KHAIGPRX715, KHAIGPRX568, KHAIGPRX1058   \n",
       "24         CYP2C19*2                                        KHAIGPRX175   \n",
       "25         CYP2C19*2  KHAIGPRX935, KHAIGPRX855, KHAIGPRX81, KHAIGPRX...   \n",
       "26           NAT2*5A  KHAIGPRX9, KHAIGPRX99, KHAIGPRX997, KHAIGPRX99...   \n",
       "27           NAT2*5A             KHAIGPRX760, KHAIGPRX239, KHAIGPRX1005   \n",
       "28           NAT2*5A  KHAIGPRX985, KHAIGPRX921, KHAIGPRX906, KHAIGPR...   \n",
       "29           NAT2*5A  KHAIGPRX993, KHAIGPRX967, KHAIGPRX956, KHAIGPR...   \n",
       "30         CYP2B6*26  KHAIGPRX99, KHAIGPRX990, KHAIGPRX98, KHAIGPRX9...   \n",
       "31          CYP2B6*6  KHAIGPRX99, KHAIGPRX990, KHAIGPRX98, KHAIGPRX9...   \n",
       "32         CYP2B6*26  KHAIGPRX897, KHAIGPRX894, KHAIGPRX815, KHAIGPR...   \n",
       "33          CYP2B6*6  KHAIGPRX897, KHAIGPRX894, KHAIGPRX815, KHAIGPR...   \n",
       "34         CYP2B6*26                           KHAIGPRX704, KHAIGPRX119   \n",
       "35          CYP2B6*6                           KHAIGPRX704, KHAIGPRX119   \n",
       "36         CYP2B6*26  KHAIGPRX971, KHAIGPRX955, KHAIGPRX942, KHAIGPR...   \n",
       "37          CYP2B6*6  KHAIGPRX971, KHAIGPRX955, KHAIGPRX942, KHAIGPR...   \n",
       "38        SLCO1B1*15  KHAIGPRX94, KHAIGPRX940, KHAIGPRX91, KHAIGPRX9...   \n",
       "39        SLCO1B1*15  KHAIGPRX972, KHAIGPRX941, KHAIGPRX928, KHAIGPR...   \n",
       "40        SLCO1B1*15  KHAIGPRX95, KHAIGPRX414, KHAIGPRX409, KHAIGPRX...   \n",
       "\n",
       "    Sample_count  \n",
       "0            434  \n",
       "1              2  \n",
       "2             56  \n",
       "3            124  \n",
       "4             52  \n",
       "5              1  \n",
       "6             86  \n",
       "7             55  \n",
       "8              4  \n",
       "9             15  \n",
       "10            18  \n",
       "11             6  \n",
       "12           410  \n",
       "13             4  \n",
       "14             3  \n",
       "15             1  \n",
       "16             4  \n",
       "17             1  \n",
       "18           457  \n",
       "19             5  \n",
       "20             5  \n",
       "21             1  \n",
       "22           418  \n",
       "23             3  \n",
       "24             1  \n",
       "25            31  \n",
       "26           458  \n",
       "27             3  \n",
       "28            15  \n",
       "29           125  \n",
       "30           338  \n",
       "31           338  \n",
       "32            26  \n",
       "33            26  \n",
       "34             2  \n",
       "35             2  \n",
       "36            74  \n",
       "37            74  \n",
       "38            42  \n",
       "39            42  \n",
       "40             6  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped = df.groupby(['rsID_mapped_sorted', 'Zygosity_mapped', 'Haplotype_updated']).agg({'Sample': lambda x: ', '.join(x.unique())}).reset_index()\n",
    "grouped['Sample_count'] = grouped['Sample'].apply(lambda x: len(x.split(',')))\n",
    "grouped.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/new_haplo_data_16_12_2023/1232_samples_Haplotype_updated.xlsx', index=False)\n",
    "grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fcb71a72",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "ead1080d",
   "metadata": {},
   "source": [
    "# condition3_ haplotype_new mapped genes positions along with mapped data... "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "d9691432",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11786195</td>\n",
       "      <td>rs4846048</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.2935,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:51:28:28:14:14:50%:6.911E-6:55:44:10:4:8:6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11786390</td>\n",
       "      <td>rs4845884</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.06669,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:158:28:28:0:28:100%:1.3074E-16:0:53:0:0:23:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11787392</td>\n",
       "      <td>rs3737967</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9257,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:39:25:25:14:11:44%:1.1933E-4:54:47:14:0:9:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11787715</td>\n",
       "      <td>rs3820192</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8876,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:27:24:24:16:8:33.33%:1.949E-3:47:50:8:8:3:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11788011</td>\n",
       "      <td>rs1537514</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8824,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:46:29:29:16:13:44.83%:2.1506E-5:59:50:11:5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>chr21</td>\n",
       "      <td>45532180</td>\n",
       "      <td>rs914231</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4856,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:43:17:17:6:11:64.71%:4.3258E-5:47:47:4:2:5:6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>chr21</td>\n",
       "      <td>45534483</td>\n",
       "      <td>rs150745916</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=23;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.998,0.0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:54:23:23:9:14:60.87%:3.4067E-6:42:50:7:2:13:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154531643</td>\n",
       "      <td>rs1050757</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5846,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:158:28:28:0:28:100%:1.3074E-16:0:44:0:0:16:12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154532293</td>\n",
       "      <td>rs2071429</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5852,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:105:19:19:0:19:100%:2.8292E-11:0:47:0:0:16:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154532439</td>\n",
       "      <td>rs2230037</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7841,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:152:27:27:0:27:100%:5.1363E-16:0:50:0:0:16:11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>113 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0     chr1   11786195    rs4846048   G   A    .   PASS   \n",
       "1     chr1   11786390    rs4845884   G   A    .   PASS   \n",
       "2     chr1   11787392    rs3737967   G   A    .   PASS   \n",
       "3     chr1   11787715    rs3820192   G   T    .   PASS   \n",
       "4     chr1   11788011    rs1537514   G   C    .   PASS   \n",
       "..     ...        ...          ...  ..  ..  ...    ...   \n",
       "108  chr21   45532180     rs914231   T   C    .   PASS   \n",
       "109  chr21   45534483  rs150745916   A   C    .   PASS   \n",
       "110   chrX  154531643    rs1050757   C   T    .   PASS   \n",
       "111   chrX  154532293    rs2071429   G   A    .   PASS   \n",
       "112   chrX  154532439    rs2230037   A   G    .   PASS   \n",
       "\n",
       "                                                  INFO  \\\n",
       "0    ADP=28;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.2935,0....   \n",
       "1    ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.06669,0...   \n",
       "2    ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9257,0....   \n",
       "3    ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8876,0....   \n",
       "4    ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8824,0....   \n",
       "..                                                 ...   \n",
       "108  ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4856,0....   \n",
       "109  ADP=23;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.998,0.0...   \n",
       "110  ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5846,0....   \n",
       "111  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5852,0....   \n",
       "112  ADP=27;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7841,0....   \n",
       "\n",
       "                                                FORMAT  \\\n",
       "0    GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1    GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2    GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3    GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4    GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "..                                                 ...   \n",
       "108  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "109  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "110  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "111  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "112  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                SAMPLE  \n",
       "0       0/1:51:28:28:14:14:50%:6.911E-6:55:44:10:4:8:6  \n",
       "1     1/1:158:28:28:0:28:100%:1.3074E-16:0:53:0:0:23:5  \n",
       "2      0/1:39:25:25:14:11:44%:1.1933E-4:54:47:14:0:9:2  \n",
       "3      0/1:27:24:24:16:8:33.33%:1.949E-3:47:50:8:8:3:5  \n",
       "4    0/1:46:29:29:16:13:44.83%:2.1506E-5:59:50:11:5...  \n",
       "..                                                 ...  \n",
       "108   0/1:43:17:17:6:11:64.71%:4.3258E-5:47:47:4:2:5:6  \n",
       "109  0/1:54:23:23:9:14:60.87%:3.4067E-6:42:50:7:2:13:1  \n",
       "110  1/1:158:28:28:0:28:100%:1.3074E-16:0:44:0:0:16:12  \n",
       "111   1/1:105:19:19:0:19:100%:2.8292E-11:0:47:0:0:16:3  \n",
       "112  1/1:152:27:27:0:27:100%:5.1363E-16:0:50:0:0:16:11  \n",
       "\n",
       "[113 rows x 10 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "variants = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/haplo_20_samples/KHAIGPRX1_final_DP.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "variants.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "d6315e9c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GENE</td>\n",
       "      <td>CHR</td>\n",
       "      <td>Start_POS</td>\n",
       "      <td>END_POS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40843541</td>\n",
       "      <td>40850447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP2B6</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40991282</td>\n",
       "      <td>41018398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>chr10</td>\n",
       "      <td>94762681</td>\n",
       "      <td>94855547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CYP2C9</td>\n",
       "      <td>chr10</td>\n",
       "      <td>94938658</td>\n",
       "      <td>94990091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>chr22</td>\n",
       "      <td>42126499</td>\n",
       "      <td>42130810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CYP3A4</td>\n",
       "      <td>chr7</td>\n",
       "      <td>99756967</td>\n",
       "      <td>99784184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CYP3A5</td>\n",
       "      <td>chr7</td>\n",
       "      <td>99648194</td>\n",
       "      <td>99679996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>chr8</td>\n",
       "      <td>18391282</td>\n",
       "      <td>18401218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NUDT15</td>\n",
       "      <td>chr13</td>\n",
       "      <td>48037726</td>\n",
       "      <td>48047221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SLCO1B1</td>\n",
       "      <td>chr12</td>\n",
       "      <td>21131194</td>\n",
       "      <td>21239796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TPMT</td>\n",
       "      <td>chr6</td>\n",
       "      <td>18128311</td>\n",
       "      <td>18155077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>UGT1A1</td>\n",
       "      <td>chr2</td>\n",
       "      <td>233760270</td>\n",
       "      <td>233773300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>G6PD</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154531391</td>\n",
       "      <td>154546846</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene chromosome Extended_Start_pos Extended_End_pos\n",
       "0      GENE        CHR          Start_POS          END_POS\n",
       "1    CYP2A6      chr19           40843541         40850447\n",
       "2    CYP2B6      chr19           40991282         41018398\n",
       "3   CYP2C19      chr10           94762681         94855547\n",
       "4    CYP2C9      chr10           94938658         94990091\n",
       "5    CYP2D6      chr22           42126499         42130810\n",
       "6    CYP3A4       chr7           99756967         99784184\n",
       "7    CYP3A5       chr7           99648194         99679996\n",
       "8      NAT2       chr8           18391282         18401218\n",
       "9    NUDT15      chr13           48037726         48047221\n",
       "10  SLCO1B1      chr12           21131194         21239796\n",
       "11     TPMT       chr6           18128311         18155077\n",
       "12   UGT1A1       chr2          233760270        233773300\n",
       "13     G6PD       chrX          154531391        154546846"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/haplo_20_samples/13genes_coordinates_haplotypes_07122023.xlsx', header = None)\n",
    "df.columns = ['Gene', 'chromosome', 'Extended_Start_pos', 'Extended_End_pos']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "c7c97dda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Covered_Chromosome</th>\n",
       "      <th>Covered_Start_Pos</th>\n",
       "      <th>Covered_End_Pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr6</td>\n",
       "      <td>18128428</td>\n",
       "      <td>rs7886</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.2372,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:17:17:10:7:41.18%:3.6151E-3:40:47:1:9:3:4</td>\n",
       "      <td>Covered</td>\n",
       "      <td>TPMT</td>\n",
       "      <td>chr6</td>\n",
       "      <td>18128311.0</td>\n",
       "      <td>18155077.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>chr7</td>\n",
       "      <td>99662739</td>\n",
       "      <td>rs4646453</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8546,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:34:15:15:6:9:60%:3.4983E-4:55:43:5:1:8:1</td>\n",
       "      <td>Covered</td>\n",
       "      <td>CYP3A5</td>\n",
       "      <td>chr7</td>\n",
       "      <td>99648194.0</td>\n",
       "      <td>99679996.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>chr7</td>\n",
       "      <td>99763843</td>\n",
       "      <td>rs2242480</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5783,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:146:26:26:0:26:100%:2.0165E-15:0:49:0:0:21:5</td>\n",
       "      <td>Covered</td>\n",
       "      <td>CYP3A4</td>\n",
       "      <td>chr7</td>\n",
       "      <td>99756967.0</td>\n",
       "      <td>99784184.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400344</td>\n",
       "      <td>rs1801280</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.7073,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:25:25:12:13:52%:1.4654E-5:49:61:9:3:10:3</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>chr8</td>\n",
       "      <td>18391282.0</td>\n",
       "      <td>18401218.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chr8</td>\n",
       "      <td>18400806</td>\n",
       "      <td>rs1208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=23;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3229,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:49:23:23:10:13:56.52%:1.1242E-5:54:57:8:2:...</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>chr8</td>\n",
       "      <td>18391282.0</td>\n",
       "      <td>18401218.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94762804</td>\n",
       "      <td>rs17885098</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=23;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9081,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:129:23:23:0:23:100%:1.2146E-13:0:54:0:0:17:6</td>\n",
       "      <td>Covered</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>chr10</td>\n",
       "      <td>94762681.0</td>\n",
       "      <td>94855547.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94942093</td>\n",
       "      <td>rs9332120</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8558,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:254:44:44:0:44:100%:3.8097E-26:0:52:0:0:31:13</td>\n",
       "      <td>Covered</td>\n",
       "      <td>CYP2C9</td>\n",
       "      <td>chr10</td>\n",
       "      <td>94938658.0</td>\n",
       "      <td>94990091.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>chr12</td>\n",
       "      <td>21239628</td>\n",
       "      <td>rs4149087</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5523,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:28:19:19:11:8:42.11%:1.5455E-3:43:44:8:3:6:2</td>\n",
       "      <td>Covered</td>\n",
       "      <td>SLCO1B1</td>\n",
       "      <td>chr12</td>\n",
       "      <td>21131194.0</td>\n",
       "      <td>21239796.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>chr12</td>\n",
       "      <td>21239652</td>\n",
       "      <td>rs4149088</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5515,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:20:17:17:11:6:35.29%:9.2021E-3:56:57:8:3:5:1</td>\n",
       "      <td>Covered</td>\n",
       "      <td>SLCO1B1</td>\n",
       "      <td>chr12</td>\n",
       "      <td>21131194.0</td>\n",
       "      <td>21239796.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48040905</td>\n",
       "      <td>rs747497670</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=NUDT...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:18:18:11:7:38.89%:3.8123E-3:56:41:11:0:7:0</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NUDT15</td>\n",
       "      <td>chr13</td>\n",
       "      <td>48037726.0</td>\n",
       "      <td>48047221.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40849944</td>\n",
       "      <td>rs2302990</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=84;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CYP2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:58:84:84:66:18:21.43%:1.3771E-6:50:66:45:2...</td>\n",
       "      <td>Covered</td>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40843541.0</td>\n",
       "      <td>40850447.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40849991</td>\n",
       "      <td>rs376970015</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=73;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CYP2...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:55:73:73:56:17:23.29%:2.6617E-6:54:54:39:1...</td>\n",
       "      <td>Covered</td>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40843541.0</td>\n",
       "      <td>40850447.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41006887</td>\n",
       "      <td>rs4803419</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7109,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:105:19:19:0:19:100%:2.8292E-11:0:51:0:0:17:2</td>\n",
       "      <td>Covered</td>\n",
       "      <td>CYP2B6</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40991282.0</td>\n",
       "      <td>41018398.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154531643</td>\n",
       "      <td>rs1050757</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5846,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:158:28:28:0:28:100%:1.3074E-16:0:44:0:0:16:12</td>\n",
       "      <td>Covered</td>\n",
       "      <td>G6PD</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154531391.0</td>\n",
       "      <td>154546846.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154532293</td>\n",
       "      <td>rs2071429</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5852,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:105:19:19:0:19:100%:2.8292E-11:0:47:0:0:16:3</td>\n",
       "      <td>Covered</td>\n",
       "      <td>G6PD</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154531391.0</td>\n",
       "      <td>154546846.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154532439</td>\n",
       "      <td>rs2230037</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7841,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:152:27:27:0:27:100%:5.1363E-16:0:50:0:0:16:11</td>\n",
       "      <td>Covered</td>\n",
       "      <td>G6PD</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154531391.0</td>\n",
       "      <td>154546846.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "45    chr6   18128428       rs7886   T   A    .   PASS   \n",
       "52    chr7   99662739    rs4646453   C   A    .   PASS   \n",
       "53    chr7   99763843    rs2242480   C   T    .   PASS   \n",
       "56    chr8   18400344    rs1801280   T   C    .   PASS   \n",
       "57    chr8   18400806       rs1208   G   A    .   PASS   \n",
       "58   chr10   94762804   rs17885098   C   T    .   PASS   \n",
       "59   chr10   94942093    rs9332120   T   C    .   PASS   \n",
       "60   chr12   21239628    rs4149087   T   G    .   PASS   \n",
       "61   chr12   21239652    rs4149088   A   G    .   PASS   \n",
       "62   chr13   48040905  rs747497670   G   C    .   PASS   \n",
       "98   chr19   40849944    rs2302990   A   G    .   PASS   \n",
       "99   chr19   40849991  rs376970015   T   C    .   PASS   \n",
       "100  chr19   41006887    rs4803419   C   T    .   PASS   \n",
       "110   chrX  154531643    rs1050757   C   T    .   PASS   \n",
       "111   chrX  154532293    rs2071429   G   A    .   PASS   \n",
       "112   chrX  154532439    rs2230037   A   G    .   PASS   \n",
       "\n",
       "                                                  INFO  \\\n",
       "45   ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.2372,0....   \n",
       "52   ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8546,0....   \n",
       "53   ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5783,0....   \n",
       "56   ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.7073,0....   \n",
       "57   ADP=23;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3229,0....   \n",
       "58   ADP=23;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9081,0....   \n",
       "59   ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8558,0....   \n",
       "60   ADP=19;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5523,0....   \n",
       "61   ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5515,0....   \n",
       "62   ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=NUDT...   \n",
       "98   ADP=84;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CYP2...   \n",
       "99   ADP=73;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CYP2...   \n",
       "100  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7109,0....   \n",
       "110  ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5846,0....   \n",
       "111  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5852,0....   \n",
       "112  ADP=27;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7841,0....   \n",
       "\n",
       "                                                FORMAT  \\\n",
       "45   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "52   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "53   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "56   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "57   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "58   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "59   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "60   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "61   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "62   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "98   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "99   GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "100  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "110  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "111  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "112  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                SAMPLE Covered/Not_Covered  \\\n",
       "45    0/1:24:17:17:10:7:41.18%:3.6151E-3:40:47:1:9:3:4             Covered   \n",
       "52        0/1:34:15:15:6:9:60%:3.4983E-4:55:43:5:1:8:1             Covered   \n",
       "53    1/1:146:26:26:0:26:100%:2.0165E-15:0:49:0:0:21:5             Covered   \n",
       "56     0/1:48:25:25:12:13:52%:1.4654E-5:49:61:9:3:10:3             Covered   \n",
       "57   0/1:49:23:23:10:13:56.52%:1.1242E-5:54:57:8:2:...             Covered   \n",
       "58    1/1:129:23:23:0:23:100%:1.2146E-13:0:54:0:0:17:6             Covered   \n",
       "59   1/1:254:44:44:0:44:100%:3.8097E-26:0:52:0:0:31:13             Covered   \n",
       "60    0/1:28:19:19:11:8:42.11%:1.5455E-3:43:44:8:3:6:2             Covered   \n",
       "61    0/1:20:17:17:11:6:35.29%:9.2021E-3:56:57:8:3:5:1             Covered   \n",
       "62   0/1:24:18:18:11:7:38.89%:3.8123E-3:56:41:11:0:7:0             Covered   \n",
       "98   0/1:58:84:84:66:18:21.43%:1.3771E-6:50:66:45:2...             Covered   \n",
       "99   0/1:55:73:73:56:17:23.29%:2.6617E-6:54:54:39:1...             Covered   \n",
       "100   1/1:105:19:19:0:19:100%:2.8292E-11:0:51:0:0:17:2             Covered   \n",
       "110  1/1:158:28:28:0:28:100%:1.3074E-16:0:44:0:0:16:12             Covered   \n",
       "111   1/1:105:19:19:0:19:100%:2.8292E-11:0:47:0:0:16:3             Covered   \n",
       "112  1/1:152:27:27:0:27:100%:5.1363E-16:0:50:0:0:16:11             Covered   \n",
       "\n",
       "        Gene Covered_Chromosome  Covered_Start_Pos  Covered_End_Pos  \n",
       "45      TPMT               chr6         18128311.0       18155077.0  \n",
       "52    CYP3A5               chr7         99648194.0       99679996.0  \n",
       "53    CYP3A4               chr7         99756967.0       99784184.0  \n",
       "56      NAT2               chr8         18391282.0       18401218.0  \n",
       "57      NAT2               chr8         18391282.0       18401218.0  \n",
       "58   CYP2C19              chr10         94762681.0       94855547.0  \n",
       "59    CYP2C9              chr10         94938658.0       94990091.0  \n",
       "60   SLCO1B1              chr12         21131194.0       21239796.0  \n",
       "61   SLCO1B1              chr12         21131194.0       21239796.0  \n",
       "62    NUDT15              chr13         48037726.0       48047221.0  \n",
       "98    CYP2A6              chr19         40843541.0       40850447.0  \n",
       "99    CYP2A6              chr19         40843541.0       40850447.0  \n",
       "100   CYP2B6              chr19         40991282.0       41018398.0  \n",
       "110     G6PD               chrX        154531391.0      154546846.0  \n",
       "111     G6PD               chrX        154531391.0      154546846.0  \n",
       "112     G6PD               chrX        154531391.0      154546846.0  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    gene = row['Gene']  # Assuming 'Gene' is the name of your gene column\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos, gene))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end, gene in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered', start, end, gene\n",
    "    return 'Not_Covered', None, None, None  # Return None for start, end, and gene if not covered\n",
    "\n",
    "# Step 3: Apply the function to create the new columns in data\n",
    "variants['Covered/Not_Covered'], variants['Start_Pos_Covered'], variants['End_Pos_Covered'], variants['Gene'] = zip(*variants.apply(check_coverage, axis=1))\n",
    "\n",
    "# Step 4: Create new columns for Covered rows\n",
    "covered_rows = variants['Covered/Not_Covered'] == 'Covered'\n",
    "variants.loc[covered_rows, 'Covered_Chromosome'] = variants.loc[covered_rows, 'CHROM']\n",
    "variants.loc[covered_rows, 'Covered_Start_Pos'] = variants.loc[covered_rows, 'Start_Pos_Covered']\n",
    "variants.loc[covered_rows, 'Covered_End_Pos'] = variants.loc[covered_rows, 'End_Pos_Covered']\n",
    "variants.loc[covered_rows, 'Gene'] = variants.loc[covered_rows, 'Gene']\n",
    "\n",
    "# Drop temporary columns\n",
    "variants.drop(['Start_Pos_Covered', 'End_Pos_Covered'], axis=1, inplace=True)\n",
    "\n",
    "# Display the DataFrame\n",
    "variants = variants[variants['Covered/Not_Covered'] == 'Covered']\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e3d6cfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "RSIDS_POSITION_DBSNP"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
