{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7b588921",
   "metadata": {},
   "source": [
    "# MAIN SCRIPT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3de07af6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Libraries Loaded\n",
      "Covered rows Extracted\n",
      "Zygosity Extracted\n",
      "Required columns extracted\n",
      "Genes Mapped\n",
      "Literature variants Mapped\n",
      "EXON and INTRON filter applied\n",
      "Dataframe Exported as excel\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "print('Libraries Loaded')\n",
    "\n",
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/srinivas_sir_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHCDPRGPTTL125_final.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL125_final.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n",
    "\n",
    "print('Covered rows Extracted')    \n",
    "    \n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL125_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "\n",
    "print('Zygosity Extracted')\n",
    "\n",
    "vcf[\"Gene Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf\n",
    "\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf\n",
    "\n",
    "########################################################### Required columns extraction from the CSQ column ####################\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "################################################## Frequency columns extraction ################################################\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "\n",
    "print('Required columns extracted')\n",
    "\n",
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "\n",
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "\n",
    "vcf_final = vcf[['Gene Name', 'rsID','CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "\n",
    "\n",
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "\n",
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "\n",
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "\n",
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "\n",
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "\n",
    "merged_2 = merged_2[['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "\n",
    "\n",
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/OneDrive/Desktop/cardiac_genes.xlsx')\n",
    "\n",
    "merged_2['Gene_Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene_Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "    \n",
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'Matched_Gene', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "\n",
    "print('Genes Mapped')\n",
    "\n",
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/OneDrive/Desktop/Cardiac_Lit_final_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "x['REF'] = x['chrom'].str.split('-').str[2]\n",
    "x['ALT'] = x['chrom'].str.split('-').str[3]\n",
    "\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS', 'REF', 'ALT']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'REF', 'ALT', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "\n",
    "print('Literature variants Mapped')\n",
    "\n",
    "merged_3 = merged_3[['Gene Name', 'Gene_Match', 'Matched_Gene', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "\n",
    "# EXON & INTRON filtering\n",
    "#merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL1_depth_vcf_processed.xlsx', index=False)\n",
    "\n",
    "df = merged_3[merged_3['Gene_Match'] == 'Yes']\n",
    "\n",
    "# Replace empty strings with '0/0' and convert numerical parts to integers\n",
    "df['EXON'] = df['EXON'].replace('', '0/0')\n",
    "df['EXON_Numerator'] = df['EXON'].apply(lambda x: x.split('/')[0] if '/' in x else 0)\n",
    "df['EXON_Denominator'] = df['EXON'].apply(lambda x: x.split('/')[1] if '/' in x else 0)\n",
    "\n",
    "df['INTRON'] = df['INTRON'].replace('', '0/0')\n",
    "df['INTRON_Numerator'] = df['INTRON'].apply(lambda x: x.split('/')[0] if '/' in x else 0)\n",
    "df['INTRON_Denominator'] = df['INTRON'].apply(lambda x: x.split('/')[1] if '/' in x else 0)\n",
    "\n",
    "# Convert the data types of numerator and denominator columns to integers\n",
    "df['EXON_Numerator'] = df['EXON_Numerator'].astype(int)\n",
    "df['EXON_Denominator'] = df['EXON_Denominator'].astype(int)\n",
    "\n",
    "df['INTRON_Numerator'] = df['INTRON_Numerator'].astype(int)\n",
    "df['INTRON_Denominator'] = df['INTRON_Denominator'].astype(int)\n",
    "\n",
    "# Initialize an empty DataFrame to store the final result\n",
    "result_df = pd.DataFrame()\n",
    "\n",
    "# Iterate over unique combinations of CHROM, POS, rsID, REF, ALT\n",
    "for _, group_df in df.groupby(['CHROM', 'POS', 'REF', 'ALT']):\n",
    "    # Check if EXON column has values\n",
    "    if not group_df['EXON_Numerator'].eq(0).all():\n",
    "        # Prioritize rows with values in EXON column\n",
    "        result_df = pd.concat([result_df, group_df.sort_values(by=['EXON_Numerator'], ascending=False).head(1)])\n",
    "    else:\n",
    "        # If EXON is empty, prioritize rows with values in INTRON column\n",
    "        if not group_df['INTRON_Numerator'].eq(0).all():\n",
    "            result_df = pd.concat([result_df, group_df.sort_values(by=['INTRON_Numerator'], ascending=False).head(1)])\n",
    "        else:\n",
    "            # If both EXON and INTRON are empty, just concatenate the first row\n",
    "            result_df = pd.concat([result_df, group_df.head(1)])\n",
    "            \n",
    "print('EXON and INTRON filter applied')\n",
    "\n",
    "# Now result_df contains the rows you're looking for based on the specified logic\n",
    "result_df = result_df.drop(['Gene_Match', 'Gene Name', 'EXON_Numerator', 'EXON_Denominator', 'INTRON_Numerator', 'INTRON_Denominator'], axis=1)\n",
    "\n",
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL125_final.xlsx', index=False)\n",
    "\n",
    "print('Dataframe Exported as excel')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b37fd37",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d26a20b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eab68bae",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff417353",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "444a332e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c837ca9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09d83cfa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7afedee6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da2437a8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "ced58cf4",
   "metadata": {},
   "source": [
    "# Changing the file name _final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c010b57",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "def remove_final_suffix(folder_path):\n",
    "    # List all files in the specified folder\n",
    "    files = os.listdir(folder_path)\n",
    "\n",
    "    for file_name in files:\n",
    "        # Check if the file name contains \"_final\"\n",
    "        if \"_final\" in file_name:\n",
    "            # Create the new file name by replacing \"_final\" with an empty string\n",
    "            new_file_name = file_name.replace(\"_final\", \"\")\n",
    "\n",
    "            # Create the full paths for the old and new file names\n",
    "            old_path = os.path.join(folder_path, file_name)\n",
    "            new_path = os.path.join(folder_path, new_file_name)\n",
    "\n",
    "            # Rename the file\n",
    "            os.rename(old_path, new_path)\n",
    "\n",
    "            print(f'Renamed: {file_name} -> {new_file_name}')\n",
    "\n",
    "# Replace 'your_folder_path' with the actual path of your folder\n",
    "remove_final_suffix(r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/latest_outputs/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbdbf8a1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2bd32d9a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3447de6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a75426ed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e00917a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "807fe5c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f411e4c9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fe6d765",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b34f0113",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a87ccef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "585d90c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d74c9e04",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cf81394",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db94e64e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "292959e3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "260e9b79",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d373c012",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "631dfece",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da7cbea4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a25498f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "813bb81c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5def2dff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f36435ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/kalyani_mam_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/KHCDPRGPTTL38.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL38.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n",
    "\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL38.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf\n",
    "\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "vcf\n",
    "\n",
    "vcf[\"Gene Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf\n",
    "\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf\n",
    "\n",
    "########################################################### Required columns extraction from the CSQ column ####################\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "################################################## Frequency columns extraction ################################################\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "vcf\n",
    "\n",
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf\n",
    "\n",
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf\n",
    "\n",
    "vcf_final = vcf[['Gene Name', 'rsID','CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "vcf_final\n",
    "\n",
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "# Print the modified DataFrame\n",
    "vcf_final\n",
    "\n",
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "vcf_final\n",
    "\n",
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "vcf_final\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1\n",
    "\n",
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1\n",
    "\n",
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "\n",
    "\n",
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "\n",
    "\n",
    "merged_2 = merged_2[['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "\n",
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/cardiac_genes.xlsx')\n",
    "\n",
    "merged_2['Gene_Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene_Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "\n",
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'Matched_Gene', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "\n",
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Cardiac_Lit_final_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "x['REF'] = x['chrom'].str.split('-').str[2]\n",
    "x['ALT'] = x['chrom'].str.split('-').str[3]\n",
    "\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS', 'REF', 'ALT']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'REF', 'ALT', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "\n",
    "merged_3 = merged_3[['Gene Name', 'Gene_Match', 'Matched_Gene', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "\n",
    "# EXON & INTRON filtering\n",
    "#merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHHSPTGPCSP26_depth_vcf_processed.xlsx', index=False)\n",
    "\n",
    "df = merged_3[merged_3['Gene_Match'] == 'Yes']\n",
    "df['EXON'] = df['EXON'].replace('', '0/0')\n",
    "df['INTRON'] = df['INTRON'].replace('', '0/0')\n",
    "\n",
    "# Replace empty strings with '0/0'\n",
    "df['EXON'] = df['EXON'].replace('', '0/0')\n",
    "df['INTRON'] = df['INTRON'].replace('', '0/0')\n",
    "\n",
    "# Convert the data types of EXON and INTRON to integers\n",
    "df['EXON'] = df['EXON'].astype(str).astype(int, errors='ignore')\n",
    "df['INTRON'] = df['INTRON'].astype(str).astype(int, errors='ignore')\n",
    "\n",
    "# Initialize an empty DataFrame to store the final result\n",
    "result_df = pd.DataFrame()\n",
    "\n",
    "# Iterate over unique combinations of CHROM, POS, rsID, REF, ALT\n",
    "for _, group_df in df.groupby(['CHROM', 'POS', 'REF', 'ALT']):\n",
    "    # Check if EXON column has values\n",
    "    if not group_df['EXON'].isnull().all():\n",
    "        # Prioritize rows with values in EXON column\n",
    "        result_df = pd.concat([result_df, group_df.sort_values(by=['EXON'], ascending=False).head(1)])\n",
    "    else:\n",
    "        # If EXON is empty, prioritize rows with values in INTRON column\n",
    "        if not group_df['INTRON'].isnull().all():\n",
    "            result_df = pd.concat([result_df, group_df.sort_values(by=['INTRON'], ascending=True).head(1)])\n",
    "        else:\n",
    "            # If both EXON and INTRON are empty, just concatenate the first row\n",
    "            result_df = pd.concat([result_df, group_df.head(1)])\n",
    "\n",
    "# Now result_df contains the rows you're looking for based on the specified logic\n",
    "result_df = result_df.drop(['Gene_Match', 'Gene Name'], axis=1)\n",
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/KHCDPRGPTTL38.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c92fd0b",
   "metadata": {},
   "source": [
    "# main code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2726109f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f2c19a3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "779aefb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/kalyani_mam_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/KHCDPRGPTTL39.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL39.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n",
    "\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL39.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf\n",
    "\n",
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "vcf\n",
    "\n",
    "vcf[\"Gene Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf\n",
    "\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf\n",
    "\n",
    "########################################################### Required columns extraction from the CSQ column ####################\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "################################################## Frequency columns extraction ################################################\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "vcf\n",
    "\n",
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf\n",
    "\n",
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf\n",
    "\n",
    "vcf_final = vcf[['Gene Name', 'rsID','CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "vcf_final\n",
    "\n",
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "# Print the modified DataFrame\n",
    "vcf_final\n",
    "\n",
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "vcf_final\n",
    "\n",
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "vcf_final\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1\n",
    "\n",
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1\n",
    "\n",
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "\n",
    "\n",
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "\n",
    "\n",
    "merged_2 = merged_2[['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "\n",
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/cardiac_genes.xlsx')\n",
    "\n",
    "merged_2['Gene_Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene_Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "\n",
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'Matched_Gene', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "\n",
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Cardiac_Lit_final_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "x['REF'] = x['chrom'].str.split('-').str[2]\n",
    "x['ALT'] = x['chrom'].str.split('-').str[3]\n",
    "\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS', 'REF', 'ALT']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'REF', 'ALT', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "\n",
    "merged_3 = merged_3[['Gene Name', 'Gene_Match', 'Matched_Gene', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "\n",
    "# EXON & INTRON filtering\n",
    "#merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL1_depth_vcf_processed.xlsx', index=False)\n",
    "\n",
    "df = merged_3[merged_3['Gene_Match'] == 'Yes']\n",
    "\n",
    "# Replace empty strings with '0/0' and convert numerical parts to integers\n",
    "df['EXON'] = df['EXON'].replace('', '0/0')\n",
    "df['EXON_Numerator'] = df['EXON'].apply(lambda x: x.split('/')[0] if '/' in x else 0)\n",
    "df['EXON_Denominator'] = df['EXON'].apply(lambda x: x.split('/')[1] if '/' in x else 0)\n",
    "\n",
    "df['INTRON'] = df['INTRON'].replace('', '0/0')\n",
    "df['INTRON_Numerator'] = df['INTRON'].apply(lambda x: x.split('/')[0] if '/' in x else 0)\n",
    "df['INTRON_Denominator'] = df['INTRON'].apply(lambda x: x.split('/')[1] if '/' in x else 0)\n",
    "\n",
    "# Convert the data types of numerator and denominator columns to integers\n",
    "df['EXON_Numerator'] = df['EXON_Numerator'].astype(int)\n",
    "df['EXON_Denominator'] = df['EXON_Denominator'].astype(int)\n",
    "\n",
    "df['INTRON_Numerator'] = df['INTRON_Numerator'].astype(int)\n",
    "df['INTRON_Denominator'] = df['INTRON_Denominator'].astype(int)\n",
    "\n",
    "# Initialize an empty DataFrame to store the final result\n",
    "result_df = pd.DataFrame()\n",
    "\n",
    "# Iterate over unique combinations of CHROM, POS, rsID, REF, ALT\n",
    "for _, group_df in df.groupby(['CHROM', 'POS', 'REF', 'ALT']):\n",
    "    # Check if EXON column has values\n",
    "    if not group_df['EXON_Numerator'].eq(0).all():\n",
    "        # Prioritize rows with values in EXON column\n",
    "        result_df = pd.concat([result_df, group_df.sort_values(by=['EXON_Numerator'], ascending=False).head(1)])\n",
    "    else:\n",
    "        # If EXON is empty, prioritize rows with values in INTRON column\n",
    "        if not group_df['INTRON_Numerator'].eq(0).all():\n",
    "            result_df = pd.concat([result_df, group_df.sort_values(by=['INTRON_Numerator'], ascending=False).head(1)])\n",
    "        else:\n",
    "            # If both EXON and INTRON are empty, just concatenate the first row\n",
    "            result_df = pd.concat([result_df, group_df.head(1)])\n",
    "\n",
    "# Now result_df contains the rows you're looking for based on the specified logic\n",
    "result_df = result_df.drop(['Gene_Match', 'Gene Name', 'EXON_Numerator', 'EXON_Denominator', 'INTRON_Numerator', 'INTRON_Denominator'], axis=1)\n",
    "\n",
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL39.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2c4b29a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d5695b9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d1c926a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d801ff2f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0e58492",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16de15fd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d339dfb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "39578807",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs45448095</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55039774</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2934</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/12</td>\n",
       "      <td>0/0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,19191301,12730697,29748315,33889589,3...</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs11583680</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55039995</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2941</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/12</td>\n",
       "      <td>0/0</td>\n",
       "      <td>A53V</td>\n",
       "      <td>gCc/gTc</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,28966647,25266949,30473376,31036026,2...</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs2495482</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55040059</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9816</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,17142622</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs625619</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55052493</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant,NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8046</td>\n",
       "      <td>EAS</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>0/0</td>\n",
       "      <td>5/5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,18300938</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs509504</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55057360</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/12</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Q342</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25343</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230808</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104800523</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8451</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>35/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>K1587R</td>\n",
       "      <td>aAa/aGa</td>\n",
       "      <td>-1</td>\n",
       "      <td>21467728,22833659,23656756,19041386,19878569,3...</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25344</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2066716</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104806424</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>31/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>T1427</td>\n",
       "      <td>acG/acA</td>\n",
       "      <td>-1</td>\n",
       "      <td>23018797,23168575,20167577,17855807,35213538</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25345</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2246841</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104840385</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2685</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G316</td>\n",
       "      <td>ggG/ggA</td>\n",
       "      <td>-1</td>\n",
       "      <td>17553166</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25346</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25347</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4887</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25348 rows  70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Matched_Gene        rsID Literature CHROM        POS REF ALT  \\\n",
       "0            PCSK9  rs45448095         No  chr1   55039774   C   T   \n",
       "1            PCSK9  rs11583680         No  chr1   55039995   C   T   \n",
       "2            PCSK9   rs2495482         No  chr1   55040059   A   G   \n",
       "3            PCSK9    rs625619         No  chr1   55052493   G   A   \n",
       "4            PCSK9    rs509504         No  chr1   55057360   A   G   \n",
       "...            ...         ...        ...   ...        ...  ..  ..   \n",
       "25343        ABCA1   rs2230808         No  chr9  104800523   T   C   \n",
       "25344        ABCA1   rs2066716         No  chr9  104806424   C   T   \n",
       "25345        ABCA1   rs2246841         No  chr9  104840385   C   T   \n",
       "25346        ABCA1   rs2230806        Yes  chr9  104858586   C   T   \n",
       "25347        ABCA1   rs2230805         No  chr9  104861748   C   T   \n",
       "\n",
       "           Zygosity                            Consequence Consequence_score  \\\n",
       "0      Heterozygous                    5 prime UTR variant              3/10   \n",
       "1      Heterozygous                       missense variant              7/10   \n",
       "2        Homozygous                         intron variant              2/10   \n",
       "3        Homozygous  intron variant,NMD transcript variant              2/10   \n",
       "4        Homozygous                     synonymous variant              3/10   \n",
       "...             ...                                    ...               ...   \n",
       "25343  Heterozygous                       missense variant              7/10   \n",
       "25344  Heterozygous                     synonymous variant              3/10   \n",
       "25345  Heterozygous                     synonymous variant              3/10   \n",
       "25346  Heterozygous                       missense variant              7/10   \n",
       "25347  Heterozygous                     synonymous variant              3/10   \n",
       "\n",
       "       ...  MAX_AF                                        MAX_AF_POPS  \\\n",
       "0      ...  0.2934                                        gnomADg AMI   \n",
       "1      ...  0.2941                                        gnomADg AMI   \n",
       "2      ...  0.9816                                                SAS   \n",
       "3      ...  0.8046                                                EAS   \n",
       "4      ...  1.0000  EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...   \n",
       "...    ...     ...                                                ...   \n",
       "25343  ...  0.8451                                        gnomADg FIN   \n",
       "25344  ...  0.2938                                        gnomADg EAS   \n",
       "25345  ...  0.2685                                                AFR   \n",
       "25346  ...  0.7103                                                AFR   \n",
       "25347  ...  0.4887                                                AFR   \n",
       "\n",
       "                       BIOTYPE   EXON INTRON  Protein Position and Amino Acid  \\\n",
       "0               protein coding   1/12    0/0                              NaN   \n",
       "1               protein coding   1/12    0/0                             A53V   \n",
       "2               protein coding    0/0   1/11                              NaN   \n",
       "3      nonsense mediated decay    0/0    5/5                              NaN   \n",
       "4               protein coding   7/12    0/0                             Q342   \n",
       "...                        ...    ...    ...                              ...   \n",
       "25343           protein coding  35/50    0/0                           K1587R   \n",
       "25344           protein coding  31/50    0/0                            T1427   \n",
       "25345           protein coding   9/50    0/0                             G316   \n",
       "25346           protein coding   7/50    0/0                            R219K   \n",
       "25347           protein coding   6/50    0/0                             L158   \n",
       "\n",
       "        Codons STRAND                                             PUBMED  \\\n",
       "0          NaN      1  25741868,19191301,12730697,29748315,33889589,3...   \n",
       "1      gCc/gTc      1  25741868,28966647,25266949,30473376,31036026,2...   \n",
       "2          NaN      1                                  25741868,17142622   \n",
       "3          NaN      1                                  25741868,18300938   \n",
       "4      caA/caG      1                                           25741868   \n",
       "...        ...    ...                                                ...   \n",
       "25343  aAa/aGa     -1  21467728,22833659,23656756,19041386,19878569,3...   \n",
       "25344  acG/acA     -1       23018797,23168575,20167577,17855807,35213538   \n",
       "25345  ggG/ggA     -1                                           17553166   \n",
       "25346  aGg/aAg     -1  23021345,23497168,28690685,20185793,30584432,2...   \n",
       "25347  ctG/ctA     -1  20167577,22675311,29156910,19606474,22369239,1...   \n",
       "\n",
       "             Sample  \n",
       "0      KHCDPRGPTTL1  \n",
       "1      KHCDPRGPTTL1  \n",
       "2      KHCDPRGPTTL1  \n",
       "3      KHCDPRGPTTL1  \n",
       "4      KHCDPRGPTTL1  \n",
       "...             ...  \n",
       "25343  KHCDPRGPTTL9  \n",
       "25344  KHCDPRGPTTL9  \n",
       "25345  KHCDPRGPTTL9  \n",
       "25346  KHCDPRGPTTL9  \n",
       "25347  KHCDPRGPTTL9  \n",
       "\n",
       "[25348 rows x 70 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Set the path to the folder containing your Excel files\n",
    "folder_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/latest_outputs/'\n",
    "\n",
    "# Initialize an empty list to store individual DataFrames\n",
    "dfs = []\n",
    "\n",
    "# Iterate over each file in the folder\n",
    "for filename in os.listdir(folder_path):\n",
    "    if filename.endswith(\".xlsx\"):\n",
    "        file_path = os.path.join(folder_path, filename)\n",
    "        \n",
    "        # Read the Excel file into a DataFrame\n",
    "        df = pd.read_excel(file_path)\n",
    "        \n",
    "        # Extract the \"Sample\" value from the base name of the file\n",
    "        sample_name = os.path.splitext(filename)[0]\n",
    "        \n",
    "        # Add a new column \"Sample\" with the extracted value\n",
    "        df['Sample'] = sample_name\n",
    "                \n",
    "        # Append the DataFrame to the list\n",
    "        dfs.append(df)\n",
    "\n",
    "# Concatenate all DataFrames in the list\n",
    "concatenated_data = pd.concat(dfs, ignore_index=True)\n",
    "\n",
    "# Save the final DataFrame to a new Excel file\n",
    "output_file_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/latest_outputs/All_samples_concatenated_data.xlsx'\n",
    "concatenated_data.to_excel(output_file_path, index=False)\n",
    "\n",
    "# Print the concatenated DataFrame\n",
    "concatenated_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ee6194ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs45448095</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55039774</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2934</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/12</td>\n",
       "      <td>0/0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,19191301,12730697,29748315,33889589,3...</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs11583680</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55039995</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2941</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/12</td>\n",
       "      <td>0/0</td>\n",
       "      <td>A53V</td>\n",
       "      <td>gCc/gTc</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,28966647,25266949,30473376,31036026,2...</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs2495482</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55040059</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9816</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,17142622</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs625619</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55052493</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant,NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8046</td>\n",
       "      <td>EAS</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>0/0</td>\n",
       "      <td>5/5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,18300938</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>rs509504</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55057360</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/12</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Q342</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25343</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230808</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104800523</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8451</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>35/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>K1587R</td>\n",
       "      <td>aAa/aGa</td>\n",
       "      <td>-1</td>\n",
       "      <td>21467728,22833659,23656756,19041386,19878569,3...</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25344</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2066716</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104806424</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>31/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>T1427</td>\n",
       "      <td>acG/acA</td>\n",
       "      <td>-1</td>\n",
       "      <td>23018797,23168575,20167577,17855807,35213538</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25345</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2246841</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104840385</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2685</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G316</td>\n",
       "      <td>ggG/ggA</td>\n",
       "      <td>-1</td>\n",
       "      <td>17553166</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25346</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>Yes</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7103</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25347</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>No</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4887</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/50</td>\n",
       "      <td>0/0</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25348 rows  70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Matched_Gene        rsID Literature CHROM        POS REF ALT  \\\n",
       "0            PCSK9  rs45448095         No  chr1   55039774   C   T   \n",
       "1            PCSK9  rs11583680         No  chr1   55039995   C   T   \n",
       "2            PCSK9   rs2495482         No  chr1   55040059   A   G   \n",
       "3            PCSK9    rs625619         No  chr1   55052493   G   A   \n",
       "4            PCSK9    rs509504         No  chr1   55057360   A   G   \n",
       "...            ...         ...        ...   ...        ...  ..  ..   \n",
       "25343        ABCA1   rs2230808         No  chr9  104800523   T   C   \n",
       "25344        ABCA1   rs2066716         No  chr9  104806424   C   T   \n",
       "25345        ABCA1   rs2246841         No  chr9  104840385   C   T   \n",
       "25346        ABCA1   rs2230806        Yes  chr9  104858586   C   T   \n",
       "25347        ABCA1   rs2230805         No  chr9  104861748   C   T   \n",
       "\n",
       "           Zygosity                            Consequence Consequence_score  \\\n",
       "0      Heterozygous                    5 prime UTR variant              3/10   \n",
       "1      Heterozygous                       missense variant              7/10   \n",
       "2        Homozygous                         intron variant              2/10   \n",
       "3        Homozygous  intron variant,NMD transcript variant              2/10   \n",
       "4        Homozygous                     synonymous variant              3/10   \n",
       "...             ...                                    ...               ...   \n",
       "25343  Heterozygous                       missense variant              7/10   \n",
       "25344  Heterozygous                     synonymous variant              3/10   \n",
       "25345  Heterozygous                     synonymous variant              3/10   \n",
       "25346  Heterozygous                       missense variant              7/10   \n",
       "25347  Heterozygous                     synonymous variant              3/10   \n",
       "\n",
       "       ...  MAX_AF                                        MAX_AF_POPS  \\\n",
       "0      ...  0.2934                                        gnomADg AMI   \n",
       "1      ...  0.2941                                        gnomADg AMI   \n",
       "2      ...  0.9816                                                SAS   \n",
       "3      ...  0.8046                                                EAS   \n",
       "4      ...  1.0000  EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...   \n",
       "...    ...     ...                                                ...   \n",
       "25343  ...  0.8451                                        gnomADg FIN   \n",
       "25344  ...  0.2938                                        gnomADg EAS   \n",
       "25345  ...  0.2685                                                AFR   \n",
       "25346  ...  0.7103                                                AFR   \n",
       "25347  ...  0.4887                                                AFR   \n",
       "\n",
       "                       BIOTYPE   EXON INTRON  Protein Position and Amino Acid  \\\n",
       "0               protein coding   1/12    0/0                              NaN   \n",
       "1               protein coding   1/12    0/0                             A53V   \n",
       "2               protein coding    0/0   1/11                              NaN   \n",
       "3      nonsense mediated decay    0/0    5/5                              NaN   \n",
       "4               protein coding   7/12    0/0                             Q342   \n",
       "...                        ...    ...    ...                              ...   \n",
       "25343           protein coding  35/50    0/0                           K1587R   \n",
       "25344           protein coding  31/50    0/0                            T1427   \n",
       "25345           protein coding   9/50    0/0                             G316   \n",
       "25346           protein coding   7/50    0/0                            R219K   \n",
       "25347           protein coding   6/50    0/0                             L158   \n",
       "\n",
       "        Codons STRAND                                             PUBMED  \\\n",
       "0          NaN      1  25741868,19191301,12730697,29748315,33889589,3...   \n",
       "1      gCc/gTc      1  25741868,28966647,25266949,30473376,31036026,2...   \n",
       "2          NaN      1                                  25741868,17142622   \n",
       "3          NaN      1                                  25741868,18300938   \n",
       "4      caA/caG      1                                           25741868   \n",
       "...        ...    ...                                                ...   \n",
       "25343  aAa/aGa     -1  21467728,22833659,23656756,19041386,19878569,3...   \n",
       "25344  acG/acA     -1       23018797,23168575,20167577,17855807,35213538   \n",
       "25345  ggG/ggA     -1                                           17553166   \n",
       "25346  aGg/aAg     -1  23021345,23497168,28690685,20185793,30584432,2...   \n",
       "25347  ctG/ctA     -1  20167577,22675311,29156910,19606474,22369239,1...   \n",
       "\n",
       "             Sample  \n",
       "0      KHCDPRGPTTL1  \n",
       "1      KHCDPRGPTTL1  \n",
       "2      KHCDPRGPTTL1  \n",
       "3      KHCDPRGPTTL1  \n",
       "4      KHCDPRGPTTL1  \n",
       "...             ...  \n",
       "25343  KHCDPRGPTTL9  \n",
       "25344  KHCDPRGPTTL9  \n",
       "25345  KHCDPRGPTTL9  \n",
       "25346  KHCDPRGPTTL9  \n",
       "25347  KHCDPRGPTTL9  \n",
       "\n",
       "[25348 rows x 70 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenated_data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/latest_outputs/All_samples_concatenated_data.xlsx')\n",
    "concatenated_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "046deb0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Sample_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs138880920</td>\n",
       "      <td>KHCDPRGPTTL34, KHCDPRGPTTL47, KHCDPRGPTTL60, K...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs142877738</td>\n",
       "      <td>KHCDPRGPTTL67, KHCDPRGPTTL81</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs150520154</td>\n",
       "      <td>KHCDPRGPTTL67</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs1800978</td>\n",
       "      <td>KHCDPRGPTTL10, KHCDPRGPTTL11, KHCDPRGPTTL12, K...</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs182899836</td>\n",
       "      <td>KHCDPRGPTTL65</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1832</th>\n",
       "      <td>VCL</td>\n",
       "      <td>rs548487697</td>\n",
       "      <td>KHCDPRGPTTL17, KHCDPRGPTTL27</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1833</th>\n",
       "      <td>VCL</td>\n",
       "      <td>rs551463686</td>\n",
       "      <td>KHCDPRGPTTL36</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1834</th>\n",
       "      <td>VCL</td>\n",
       "      <td>rs703258</td>\n",
       "      <td>KHCDPRGPTTL1, KHCDPRGPTTL10, KHCDPRGPTTL11, KH...</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1835</th>\n",
       "      <td>VCL</td>\n",
       "      <td>rs767809</td>\n",
       "      <td>KHCDPRGPTTL1, KHCDPRGPTTL10, KHCDPRGPTTL11, KH...</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1836</th>\n",
       "      <td>VCL</td>\n",
       "      <td>rs8255</td>\n",
       "      <td>KHCDPRGPTTL30, KHCDPRGPTTL36, KHCDPRGPTTL43, K...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1837 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Matched_Gene         rsID  \\\n",
       "0           ABCA1  rs138880920   \n",
       "1           ABCA1  rs142877738   \n",
       "2           ABCA1  rs150520154   \n",
       "3           ABCA1    rs1800978   \n",
       "4           ABCA1  rs182899836   \n",
       "...           ...          ...   \n",
       "1832          VCL  rs548487697   \n",
       "1833          VCL  rs551463686   \n",
       "1834          VCL     rs703258   \n",
       "1835          VCL     rs767809   \n",
       "1836          VCL       rs8255   \n",
       "\n",
       "                                                 Sample  Sample_count  \n",
       "0     KHCDPRGPTTL34, KHCDPRGPTTL47, KHCDPRGPTTL60, K...             4  \n",
       "1                          KHCDPRGPTTL67, KHCDPRGPTTL81             2  \n",
       "2                                         KHCDPRGPTTL67             1  \n",
       "3     KHCDPRGPTTL10, KHCDPRGPTTL11, KHCDPRGPTTL12, K...            42  \n",
       "4                                         KHCDPRGPTTL65             1  \n",
       "...                                                 ...           ...  \n",
       "1832                       KHCDPRGPTTL17, KHCDPRGPTTL27             2  \n",
       "1833                                      KHCDPRGPTTL36             1  \n",
       "1834  KHCDPRGPTTL1, KHCDPRGPTTL10, KHCDPRGPTTL11, KH...            58  \n",
       "1835  KHCDPRGPTTL1, KHCDPRGPTTL10, KHCDPRGPTTL11, KH...            51  \n",
       "1836  KHCDPRGPTTL30, KHCDPRGPTTL36, KHCDPRGPTTL43, K...            12  \n",
       "\n",
       "[1837 rows x 4 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_summary = concatenated_data.groupby(['Matched_Gene', 'rsID']).agg({'Sample': lambda x: ', '.join(x.unique())}).reset_index()\n",
    "gene_summary['Sample_count'] = gene_summary['Sample'].apply(lambda x: len(x.split(',')))\n",
    "gene_summary = gene_summary[['Matched_Gene', 'rsID', 'Sample', 'Sample_count']]\n",
    "gene_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "51261f6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_summary.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/latest_outputs/Gene_rsID_sample_counts.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1aeb4d22",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "1095f8ea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Combined</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>rs707189</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>237638470</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Cardiovascular phenotype,Cardiac arrhythmia,Ca...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>43819.0</td>\n",
       "      <td>ENST00000366574.7</td>\n",
       "      <td>c.6906T&gt;C</td>\n",
       "      <td>ENSP00000355533.2</td>\n",
       "      <td>p.Leu2302%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.325900e-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9549</td>\n",
       "      <td>0.8359</td>\n",
       "      <td>0.9870</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.8674</td>\n",
       "      <td>0.9936</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9965</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>0.9642</td>\n",
       "      <td>0.8757</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9855</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>0.9761</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADg AM...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>45/105</td>\n",
       "      <td>0/0</td>\n",
       "      <td>L2302</td>\n",
       "      <td>ctT/ctC</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24033266</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "      <td>RYR2_rs707189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>rs2797436</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>237700418</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Cardiovascular phenotype,Cardiac arrhythmia,Ca...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>43836.0</td>\n",
       "      <td>ENST00000366574.7</td>\n",
       "      <td>c.9318T&gt;G</td>\n",
       "      <td>ENSP00000355533.2</td>\n",
       "      <td>p.Ser3106%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>158</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.307400e-16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9692</td>\n",
       "      <td>0.8880</td>\n",
       "      <td>0.9914</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9928</td>\n",
       "      <td>0.9012</td>\n",
       "      <td>0.9953</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9951</td>\n",
       "      <td>0.9993</td>\n",
       "      <td>0.9717</td>\n",
       "      <td>0.9021</td>\n",
       "      <td>0.9989</td>\n",
       "      <td>0.9887</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9968</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>0.9813</td>\n",
       "      <td>0.9990</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EAS,EUR,SAS,gnomADe ASJ,gnomADe FIN,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>65/105</td>\n",
       "      <td>0/0</td>\n",
       "      <td>S3106</td>\n",
       "      <td>tcT/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24033266,30019117</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "      <td>RYR2_rs2797436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>RBM20</td>\n",
       "      <td>rs6585012</td>\n",
       "      <td>No</td>\n",
       "      <td>chr10</td>\n",
       "      <td>110783459</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Dilated cardiomyopathy 1DD</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>671182.0</td>\n",
       "      <td>ENST00000369519.4</td>\n",
       "      <td>c.1337+32G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>76</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.492700e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7881</td>\n",
       "      <td>0.5840</td>\n",
       "      <td>0.8256</td>\n",
       "      <td>0.9673</td>\n",
       "      <td>0.7575</td>\n",
       "      <td>0.8845</td>\n",
       "      <td>0.8181</td>\n",
       "      <td>0.6227</td>\n",
       "      <td>0.8327</td>\n",
       "      <td>0.7746</td>\n",
       "      <td>0.9687</td>\n",
       "      <td>0.8326</td>\n",
       "      <td>0.7968</td>\n",
       "      <td>0.7885</td>\n",
       "      <td>0.8677</td>\n",
       "      <td>0.7589</td>\n",
       "      <td>0.6219</td>\n",
       "      <td>0.8121</td>\n",
       "      <td>0.7978</td>\n",
       "      <td>0.7679</td>\n",
       "      <td>0.9717</td>\n",
       "      <td>0.8285</td>\n",
       "      <td>0.7310</td>\n",
       "      <td>0.7988</td>\n",
       "      <td>0.7304</td>\n",
       "      <td>0.8628</td>\n",
       "      <td>0.9717</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>0/0</td>\n",
       "      <td>3/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "      <td>RBM20_rs6585012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>MYBPC3</td>\n",
       "      <td>rs896818</td>\n",
       "      <td>No</td>\n",
       "      <td>chr11</td>\n",
       "      <td>47343211</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant,NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1232915.0</td>\n",
       "      <td>ENST00000544791.1</td>\n",
       "      <td>c.1226+49T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>87</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.663700e-09</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9641</td>\n",
       "      <td>0.8729</td>\n",
       "      <td>0.9827</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9911</td>\n",
       "      <td>0.8885</td>\n",
       "      <td>0.9925</td>\n",
       "      <td>0.9958</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9940</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9692</td>\n",
       "      <td>0.8952</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9851</td>\n",
       "      <td>0.9965</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9589</td>\n",
       "      <td>0.9993</td>\n",
       "      <td>0.9765</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EAS,EUR,SAS,gnomADg AMI,gnomADg EAS,gnomADg FIN</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>0/0</td>\n",
       "      <td>14/26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>18414213</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "      <td>MYBPC3_rs896818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>SCN3B</td>\n",
       "      <td>rs1148110</td>\n",
       "      <td>No</td>\n",
       "      <td>chr11</td>\n",
       "      <td>123634251</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant,NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Brugada syndrome 7</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>670950.0</td>\n",
       "      <td>ENST00000667790.1</td>\n",
       "      <td>c.*566-45G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>87</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.663700e-09</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7506</td>\n",
       "      <td>0.9720</td>\n",
       "      <td>0.6225</td>\n",
       "      <td>0.7222</td>\n",
       "      <td>0.5467</td>\n",
       "      <td>0.7812</td>\n",
       "      <td>0.6196</td>\n",
       "      <td>0.9125</td>\n",
       "      <td>0.5962</td>\n",
       "      <td>0.6414</td>\n",
       "      <td>0.7529</td>\n",
       "      <td>0.4855</td>\n",
       "      <td>0.5526</td>\n",
       "      <td>0.5865</td>\n",
       "      <td>0.7499</td>\n",
       "      <td>0.6672</td>\n",
       "      <td>0.9045</td>\n",
       "      <td>0.5923</td>\n",
       "      <td>0.6404</td>\n",
       "      <td>0.6413</td>\n",
       "      <td>0.7482</td>\n",
       "      <td>0.4801</td>\n",
       "      <td>0.6646</td>\n",
       "      <td>0.5469</td>\n",
       "      <td>0.6579</td>\n",
       "      <td>0.7698</td>\n",
       "      <td>0.9720</td>\n",
       "      <td>AFR</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>0/0</td>\n",
       "      <td>6/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KHCDPRGPTTL1</td>\n",
       "      <td>SCN3B_rs1148110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11566</th>\n",
       "      <td>PKP2</td>\n",
       "      <td>rs6488090</td>\n",
       "      <td>No</td>\n",
       "      <td>chr12</td>\n",
       "      <td>32791480</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Arrhythmogenic right ventricular dysplasia 9</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>308481.0</td>\n",
       "      <td>ENST00000070846.11</td>\n",
       "      <td>c.*944C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.631400e-27</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8992</td>\n",
       "      <td>0.8729</td>\n",
       "      <td>0.8617</td>\n",
       "      <td>0.8829</td>\n",
       "      <td>0.9483</td>\n",
       "      <td>0.9274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9204</td>\n",
       "      <td>0.8802</td>\n",
       "      <td>0.8531</td>\n",
       "      <td>0.8701</td>\n",
       "      <td>0.9326</td>\n",
       "      <td>0.8890</td>\n",
       "      <td>0.9542</td>\n",
       "      <td>0.9051</td>\n",
       "      <td>0.9522</td>\n",
       "      <td>0.9168</td>\n",
       "      <td>0.9416</td>\n",
       "      <td>0.9542</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>14/14</td>\n",
       "      <td>0/0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "      <td>PKP2_rs6488090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11609</th>\n",
       "      <td>APOB</td>\n",
       "      <td>rs584542</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>21009931</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Familial hypobetalipoproteinemia 1,Hypercholes...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>1168905.0</td>\n",
       "      <td>ENST00000233242.5</td>\n",
       "      <td>c.6937A&gt;G</td>\n",
       "      <td>ENSP00000233242.1</td>\n",
       "      <td>p.Ile2313Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.631400e-27</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>11</td>\n",
       "      <td>tolerated(0.57)</td>\n",
       "      <td>benign(0.015)</td>\n",
       "      <td>0.9862</td>\n",
       "      <td>0.9523</td>\n",
       "      <td>0.9957</td>\n",
       "      <td>0.9980</td>\n",
       "      <td>0.9990</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9958</td>\n",
       "      <td>0.9471</td>\n",
       "      <td>0.9971</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9989</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>0.9980</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9848</td>\n",
       "      <td>0.9475</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9954</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9987</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>SAS,gnomADe FIN,gnomADg AMI,gnomADg ASJ,gnomAD...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>26/29</td>\n",
       "      <td>0/0</td>\n",
       "      <td>I2313V</td>\n",
       "      <td>Att/Gtt</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,29036232</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "      <td>APOB_rs584542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11625</th>\n",
       "      <td>TTN</td>\n",
       "      <td>rs4145333</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178580041</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Autosomal recessive l...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47250.0</td>\n",
       "      <td>ENST00000589042.5</td>\n",
       "      <td>c.67246G&gt;C</td>\n",
       "      <td>ENSP00000467141.1</td>\n",
       "      <td>p.Ala22416Pro</td>\n",
       "      <td>1/1</td>\n",
       "      <td>254</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.809700e-26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9946</td>\n",
       "      <td>0.9811</td>\n",
       "      <td>0.9971</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9987</td>\n",
       "      <td>0.9847</td>\n",
       "      <td>0.9988</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>0.9993</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9960</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9988</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9905</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>0.9981</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>318/363</td>\n",
       "      <td>0/0</td>\n",
       "      <td>A22416P</td>\n",
       "      <td>Gct/Cct</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,27681629,2...</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "      <td>TTN_rs4145333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11657</th>\n",
       "      <td>TTN</td>\n",
       "      <td>rs10803917</td>\n",
       "      <td>No</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178750225</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>not specified</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>47743.0</td>\n",
       "      <td>ENST00000360870.10</td>\n",
       "      <td>c.12175T&gt;C</td>\n",
       "      <td>ENSP00000354117.4</td>\n",
       "      <td>p.Leu4059%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>170</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.455600e-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9750</td>\n",
       "      <td>0.9070</td>\n",
       "      <td>0.9971</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9942</td>\n",
       "      <td>0.9197</td>\n",
       "      <td>0.9969</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9974</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>0.9765</td>\n",
       "      <td>0.9176</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9939</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9937</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>0.9808</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>46/46</td>\n",
       "      <td>0/0</td>\n",
       "      <td>L4059</td>\n",
       "      <td>Tta/Cta</td>\n",
       "      <td>-1</td>\n",
       "      <td>24033266</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "      <td>TTN_rs10803917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11686</th>\n",
       "      <td>SCN10A</td>\n",
       "      <td>rs6599241</td>\n",
       "      <td>No</td>\n",
       "      <td>chr3</td>\n",
       "      <td>38698083</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Brugada syndrome,Brugada syndrome 1,Cardiovasc...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>259999.0</td>\n",
       "      <td>ENST00000449082.3</td>\n",
       "      <td>c.5137A&gt;G</td>\n",
       "      <td>ENSP00000390600.2</td>\n",
       "      <td>p.Met1713Val</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.100100e-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>106</td>\n",
       "      <td>14</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AFR,AMR,EAS,EUR,SAS,gnomADe AFR,gnomADe AMR,gn...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>28/28</td>\n",
       "      <td>0/0</td>\n",
       "      <td>M1713V</td>\n",
       "      <td>Atg/Gtg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,31309012,34357026</td>\n",
       "      <td>KHCDPRGPTTL9</td>\n",
       "      <td>SCN10A_rs6599241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>520 rows  71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Matched_Gene        rsID Literature  CHROM        POS REF ALT  \\\n",
       "31            RYR2    rs707189         No   chr1  237638470   T   C   \n",
       "33            RYR2   rs2797436         No   chr1  237700418   T   G   \n",
       "51           RBM20   rs6585012         No  chr10  110783459   G   A   \n",
       "60          MYBPC3    rs896818         No  chr11   47343211   A   G   \n",
       "67           SCN3B   rs1148110         No  chr11  123634251   C   G   \n",
       "...            ...         ...        ...    ...        ...  ..  ..   \n",
       "11566         PKP2   rs6488090         No  chr12   32791480   G   T   \n",
       "11609         APOB    rs584542         No   chr2   21009931   T   C   \n",
       "11625          TTN   rs4145333         No   chr2  178580041   C   G   \n",
       "11657          TTN  rs10803917         No   chr2  178750225   A   G   \n",
       "11686       SCN10A   rs6599241         No   chr3   38698083   T   C   \n",
       "\n",
       "         Zygosity                            Consequence Consequence_score  \\\n",
       "31     Homozygous                     synonymous variant              3/10   \n",
       "33     Homozygous                     synonymous variant              3/10   \n",
       "51     Homozygous                         intron variant              2/10   \n",
       "60     Homozygous  intron variant,NMD transcript variant              2/10   \n",
       "67     Homozygous  intron variant,NMD transcript variant              2/10   \n",
       "...           ...                                    ...               ...   \n",
       "11566  Homozygous                    3 prime UTR variant              3/10   \n",
       "11609  Homozygous                       missense variant              7/10   \n",
       "11625  Homozygous                       missense variant              7/10   \n",
       "11657  Homozygous                     synonymous variant              3/10   \n",
       "11686  Homozygous                       missense variant              7/10   \n",
       "\n",
       "         IMPACT  IMPACT_score  \\\n",
       "31          LOW           2.5   \n",
       "33          LOW           2.5   \n",
       "51     MODIFIER           1.5   \n",
       "60     MODIFIER           1.5   \n",
       "67     MODIFIER           1.5   \n",
       "...         ...           ...   \n",
       "11566  MODIFIER           1.5   \n",
       "11609  MODERATE           5.0   \n",
       "11625  MODERATE           5.0   \n",
       "11657       LOW           2.5   \n",
       "11686  MODERATE           5.0   \n",
       "\n",
       "                                           ClinVar_CLNDN CLIN_SIG  \\\n",
       "31     Cardiovascular phenotype,Cardiac arrhythmia,Ca...   benign   \n",
       "33     Cardiovascular phenotype,Cardiac arrhythmia,Ca...   benign   \n",
       "51                            Dilated cardiomyopathy 1DD   benign   \n",
       "60                                          not provided   benign   \n",
       "67                                    Brugada syndrome 7   benign   \n",
       "...                                                  ...      ...   \n",
       "11566       Arrhythmogenic right ventricular dysplasia 9   benign   \n",
       "11609  Familial hypobetalipoproteinemia 1,Hypercholes...   benign   \n",
       "11625  Cardiovascular phenotype,Autosomal recessive l...   benign   \n",
       "11657                                      not specified   benign   \n",
       "11686  Brugada syndrome,Brugada syndrome 1,Cardiovasc...   benign   \n",
       "\n",
       "                                      ClinVar_CLNREVSTAT    ClinVar  \\\n",
       "31     criteria provided, multiple submitters, no con...    43819.0   \n",
       "33     criteria provided, multiple submitters, no con...    43836.0   \n",
       "51     criteria provided, multiple submitters, no con...   671182.0   \n",
       "60                   criteria provided, single submitter  1232915.0   \n",
       "67     criteria provided, multiple submitters, no con...   670950.0   \n",
       "...                                                  ...        ...   \n",
       "11566                criteria provided, single submitter   308481.0   \n",
       "11609  criteria provided, multiple submitters, no con...  1168905.0   \n",
       "11625  criteria provided, multiple submitters, no con...    47250.0   \n",
       "11657  criteria provided, multiple submitters, no con...    47743.0   \n",
       "11686  criteria provided, multiple submitters, no con...   259999.0   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "31      ENST00000366574.7          c.6906T>C  ENSP00000355533.2   \n",
       "33      ENST00000366574.7          c.9318T>G  ENSP00000355533.2   \n",
       "51      ENST00000369519.4       c.1337+32G>A                NaN   \n",
       "60      ENST00000544791.1       c.1226+49T>C                NaN   \n",
       "67      ENST00000667790.1       c.*566-45G>C                NaN   \n",
       "...                   ...                ...                ...   \n",
       "11566  ENST00000070846.11          c.*944C>A                NaN   \n",
       "11609   ENST00000233242.5          c.6937A>G  ENSP00000233242.1   \n",
       "11625   ENST00000589042.5         c.67246G>C  ENSP00000467141.1   \n",
       "11657  ENST00000360870.10         c.12175T>C  ENSP00000354117.4   \n",
       "11686   ENST00000449082.3          c.5137A>G  ENSP00000390600.2   \n",
       "\n",
       "      HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD  FREQ          PVAL  \\\n",
       "31          p.Leu2302%3D  1/1  164   29   29   0   29  100%  3.325900e-17   \n",
       "33          p.Ser3106%3D  1/1  158   28   28   0   28  100%  1.307400e-16   \n",
       "51                  None  1/1   76   14   14   0   14  100%  2.492700e-08   \n",
       "60                  None  1/1   87   16   16   0   16  100%  1.663700e-09   \n",
       "67                  None  1/1   87   16   16   0   16  100%  1.663700e-09   \n",
       "...                  ...  ...  ...  ...  ...  ..  ...   ...           ...   \n",
       "11566               None  1/1  255   45   45   0   45  100%  9.631400e-27   \n",
       "11609       p.Ile2313Val  1/1  255   45   45   0   45  100%  9.631400e-27   \n",
       "11625      p.Ala22416Pro  1/1  254   44   44   0   44  100%  3.809700e-26   \n",
       "11657       p.Leu4059%3D  1/1  170   30   30   0   30  100%  8.455600e-18   \n",
       "11686       p.Met1713Val  1/1  255  120  120   0  120  100%  1.100100e-71   \n",
       "\n",
       "       RDF  RDR  ADF  ADR                         SIFT       PolyPhen      AF  \\\n",
       "31       0    0   21    8                          NaN            NaN  0.9549   \n",
       "33       0    0   24    4                          NaN            NaN  0.9692   \n",
       "51       0    0   10    4                          NaN            NaN  0.7881   \n",
       "60       0    0   11    5                          NaN            NaN  0.9641   \n",
       "67       0    0   10    6                          NaN            NaN  0.7506   \n",
       "...    ...  ...  ...  ...                          ...            ...     ...   \n",
       "11566    0    0   38    7                          NaN            NaN  0.8992   \n",
       "11609    0    0   34   11              tolerated(0.57)  benign(0.015)  0.9862   \n",
       "11625    0    0   32   12                          NaN            NaN  0.9946   \n",
       "11657    0    0   25    5                          NaN            NaN  0.9750   \n",
       "11686    0    0  106   14  tolerated low confidence(1)            NaN  1.0000   \n",
       "\n",
       "       AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  gnomADe_AFR_AF  \\\n",
       "31     0.8359  0.9870  1.0000  1.0000  1.0000      0.9906          0.8674   \n",
       "33     0.8880  0.9914  1.0000  1.0000  1.0000      0.9928          0.9012   \n",
       "51     0.5840  0.8256  0.9673  0.7575  0.8845      0.8181          0.6227   \n",
       "60     0.8729  0.9827  1.0000  1.0000  1.0000      0.9911          0.8885   \n",
       "67     0.9720  0.6225  0.7222  0.5467  0.7812      0.6196          0.9125   \n",
       "...       ...     ...     ...     ...     ...         ...             ...   \n",
       "11566  0.8729  0.8617  0.8829  0.9483  0.9274         NaN             NaN   \n",
       "11609  0.9523  0.9957  0.9980  0.9990  1.0000      0.9958          0.9471   \n",
       "11625  0.9811  0.9971  1.0000  1.0000  1.0000      0.9987          0.9847   \n",
       "11657  0.9070  0.9971  1.0000  1.0000  1.0000      0.9942          0.9197   \n",
       "11686  1.0000  1.0000  1.0000  1.0000  1.0000      1.0000          1.0000   \n",
       "\n",
       "       gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  \\\n",
       "31             0.9936          1.0000          1.0000          0.9999   \n",
       "33             0.9953          1.0000          0.9997          1.0000   \n",
       "51             0.8327          0.7746          0.9687          0.8326   \n",
       "60             0.9925          0.9958          0.9999          0.9999   \n",
       "67             0.5962          0.6414          0.7529          0.4855   \n",
       "...               ...             ...             ...             ...   \n",
       "11566             NaN             NaN             NaN             NaN   \n",
       "11609          0.9971          0.9999          0.9989          1.0000   \n",
       "11625          0.9988          1.0000          1.0000          1.0000   \n",
       "11657          0.9969          1.0000          1.0000          1.0000   \n",
       "11686          1.0000          0.9999          1.0000          1.0000   \n",
       "\n",
       "       gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  \\\n",
       "31             0.9998          0.9965          0.9997      0.9642   \n",
       "33             0.9994          0.9951          0.9993      0.9717   \n",
       "51             0.7968          0.7885          0.8677      0.7589   \n",
       "60             0.9992          0.9940          0.9994      0.9692   \n",
       "67             0.5526          0.5865          0.7499      0.6672   \n",
       "...               ...             ...             ...         ...   \n",
       "11566             NaN             NaN             NaN      0.9204   \n",
       "11609          0.9995          0.9980          0.9994      0.9848   \n",
       "11625          0.9997          0.9993          1.0000      0.9960   \n",
       "11657          0.9998          0.9974          0.9997      0.9765   \n",
       "11686          1.0000          1.0000          1.0000      1.0000   \n",
       "\n",
       "       gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  \\\n",
       "31             0.8757          1.0000          0.9855          1.0000   \n",
       "33             0.9021          0.9989          0.9887          1.0000   \n",
       "51             0.6219          0.8121          0.7978          0.7679   \n",
       "60             0.8952          1.0000          0.9851          0.9965   \n",
       "67             0.9045          0.5923          0.6404          0.6413   \n",
       "...               ...             ...             ...             ...   \n",
       "11566          0.8802          0.8531          0.8701          0.9326   \n",
       "11609          0.9475          1.0000          0.9954          1.0000   \n",
       "11625          0.9865          1.0000          0.9988          1.0000   \n",
       "11657          0.9176          1.0000          0.9939          1.0000   \n",
       "11686          1.0000          1.0000          1.0000          1.0000   \n",
       "\n",
       "       gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  \\\n",
       "31             1.0000          1.0000          1.0000          0.9997   \n",
       "33             1.0000          1.0000          0.9968          0.9995   \n",
       "51             0.9717          0.8285          0.7310          0.7988   \n",
       "60             1.0000          1.0000          0.9589          0.9993   \n",
       "67             0.7482          0.4801          0.6646          0.5469   \n",
       "...               ...             ...             ...             ...   \n",
       "11566          0.8890          0.9542          0.9051          0.9522   \n",
       "11609          0.9987          1.0000          1.0000          0.9995   \n",
       "11625          0.9998          1.0000          0.9905          0.9997   \n",
       "11657          1.0000          1.0000          0.9937          0.9997   \n",
       "11686          1.0000          1.0000          1.0000          1.0000   \n",
       "\n",
       "       gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF  \\\n",
       "31             0.9761          1.0000  1.0000   \n",
       "33             0.9813          0.9990  1.0000   \n",
       "51             0.7304          0.8628  0.9717   \n",
       "60             0.9765          0.9998  1.0000   \n",
       "67             0.6579          0.7698  0.9720   \n",
       "...               ...             ...     ...   \n",
       "11566          0.9168          0.9416  0.9542   \n",
       "11609          0.9871          0.9994  1.0000   \n",
       "11625          0.9981          1.0000  1.0000   \n",
       "11657          0.9808          0.9998  1.0000   \n",
       "11686          1.0000          1.0000  1.0000   \n",
       "\n",
       "                                             MAX_AF_POPS  \\\n",
       "31     EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADg AM...   \n",
       "33     EAS,EUR,SAS,gnomADe ASJ,gnomADe FIN,gnomADg AS...   \n",
       "51                                           gnomADg EAS   \n",
       "60       EAS,EUR,SAS,gnomADg AMI,gnomADg EAS,gnomADg FIN   \n",
       "67                                                   AFR   \n",
       "...                                                  ...   \n",
       "11566                                        gnomADg FIN   \n",
       "11609  SAS,gnomADe FIN,gnomADg AMI,gnomADg ASJ,gnomAD...   \n",
       "11625  EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...   \n",
       "11657  EAS,EUR,SAS,gnomADe ASJ,gnomADe EAS,gnomADe FI...   \n",
       "11686  AFR,AMR,EAS,EUR,SAS,gnomADe AFR,gnomADe AMR,gn...   \n",
       "\n",
       "                       BIOTYPE     EXON INTRON  \\\n",
       "31              protein coding   45/105    0/0   \n",
       "33              protein coding   65/105    0/0   \n",
       "51              protein coding      0/0   3/13   \n",
       "60     nonsense mediated decay      0/0  14/26   \n",
       "67     nonsense mediated decay      0/0    6/7   \n",
       "...                        ...      ...    ...   \n",
       "11566           protein coding    14/14    0/0   \n",
       "11609           protein coding    26/29    0/0   \n",
       "11625           protein coding  318/363    0/0   \n",
       "11657           protein coding    46/46    0/0   \n",
       "11686           protein coding    28/28    0/0   \n",
       "\n",
       "      Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "31                              L2302  ctT/ctC       1   \n",
       "33                              S3106  tcT/tcG       1   \n",
       "51                                NaN      NaN       1   \n",
       "60                                NaN      NaN      -1   \n",
       "67                                NaN      NaN      -1   \n",
       "...                               ...      ...     ...   \n",
       "11566                             NaN      NaN      -1   \n",
       "11609                          I2313V  Att/Gtt      -1   \n",
       "11625                         A22416P  Gct/Cct      -1   \n",
       "11657                           L4059  Tta/Cta      -1   \n",
       "11686                          M1713V  Atg/Gtg      -1   \n",
       "\n",
       "                                                  PUBMED        Sample  \\\n",
       "31                                     25741868,24033266  KHCDPRGPTTL1   \n",
       "33                            25741868,24033266,30019117  KHCDPRGPTTL1   \n",
       "51                                                   NaN  KHCDPRGPTTL1   \n",
       "60                                              18414213  KHCDPRGPTTL1   \n",
       "67                                                   NaN  KHCDPRGPTTL1   \n",
       "...                                                  ...           ...   \n",
       "11566                                                NaN  KHCDPRGPTTL9   \n",
       "11609                                  25741868,29036232  KHCDPRGPTTL9   \n",
       "11625  25741868,24033266,18414213,23861362,27681629,2...  KHCDPRGPTTL9   \n",
       "11657                                           24033266  KHCDPRGPTTL9   \n",
       "11686                         25741868,31309012,34357026  KHCDPRGPTTL9   \n",
       "\n",
       "               Combined  \n",
       "31        RYR2_rs707189  \n",
       "33       RYR2_rs2797436  \n",
       "51      RBM20_rs6585012  \n",
       "60      MYBPC3_rs896818  \n",
       "67      SCN3B_rs1148110  \n",
       "...                 ...  \n",
       "11566    PKP2_rs6488090  \n",
       "11609     APOB_rs584542  \n",
       "11625     TTN_rs4145333  \n",
       "11657    TTN_rs10803917  \n",
       "11686  SCN10A_rs6599241  \n",
       "\n",
       "[520 rows x 71 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming concatenated_data is your DataFrame\n",
    "df = concatenated_data.copy()\n",
    "\n",
    "# Create a new column by concatenating \"Matched_Gene\" and \"rsID\"\n",
    "df['Combined'] = df['Matched_Gene'].astype(str) + '_' + df['rsID'].astype(str)\n",
    "\n",
    "# Get the set of unique combinations for each sample\n",
    "sample_combinations = df.groupby('Sample')['Combined'].unique()\n",
    "\n",
    "# Find the common combinations across all samples\n",
    "common_combinations = set(sample_combinations.iloc[0]).intersection(*sample_combinations[1:])\n",
    "\n",
    "# Filter the original DataFrame to include only the common combinations\n",
    "result_df = df[df['Combined'].isin(common_combinations)]\n",
    "\n",
    "# Print or save the result\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a3d94561",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/Common_genes_rsID_combinations.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a57c8b82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a425b0b2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "d360e986",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Set the path to the folder containing your Excel files\n",
    "folder_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/'\n",
    "\n",
    "# Initialize an empty list to store individual DataFrames\n",
    "dfs = []\n",
    "\n",
    "# Iterate over each file in the folder\n",
    "for filename in os.listdir(folder_path):\n",
    "    if filename.endswith(\".xlsx\"):\n",
    "        file_path = os.path.join(folder_path, filename)\n",
    "\n",
    "        # Read the Excel file into a DataFrame\n",
    "        df = pd.read_excel(file_path)\n",
    "\n",
    "        # Extract the \"Matched_Gene\" and \"rsID\" columns\n",
    "        subset_df = df[['Matched_Gene', 'rsID']]\n",
    "\n",
    "        # Add a new column \"Sample\" with the extracted value\n",
    "        subset_df['Sample'] = os.path.splitext(filename)[0]\n",
    "\n",
    "        # Append the subset DataFrame to the list\n",
    "        dfs.append(subset_df)\n",
    "\n",
    "# Concatenate all subset DataFrames in the list\n",
    "concatenated_subset_data = pd.concat(dfs, ignore_index=True)\n",
    "\n",
    "# Find common values in \"Matched_Gene\" and \"rsID\" columns across all samples\n",
    "common_genes = set(concatenated_subset_data['Matched_Gene'])\n",
    "common_rsids = set(concatenated_subset_data['rsID'])\n",
    "\n",
    "# Convert sets to DataFrames\n",
    "common_genes_df = pd.DataFrame(list(common_genes), columns=['Common_Genes'])\n",
    "common_rsids_df = pd.DataFrame(list(common_rsids), columns=['Common_rsIDs'])\n",
    "\n",
    "# Save DataFrames to Excel files\n",
    "common_genes_file_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/common_genes.xlsx'\n",
    "common_rsids_file_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL_VCFs/OUTPUT_data/new_output/common_rsids.xlsx'\n",
    "\n",
    "common_genes_df.to_excel(common_genes_file_path, index=False)\n",
    "common_rsids_df.to_excel(common_rsids_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f4ef201",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18c4d636",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe29057b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "33c27022",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Other_Column</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>200</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>300</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>400</td>\n",
       "      <td>E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>500</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3</td>\n",
       "      <td>100</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM  POS Other_Column\n",
       "1  chr1  200            B\n",
       "2  chr1  300            C\n",
       "4  chr2  400            E\n",
       "0  chr2  500            A\n",
       "3  chr3  100            D"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample DataFrame\n",
    "data = {'CHROM': ['chr2', 'chr1', 'chr1', 'chr3', 'chr2'],\n",
    "        'POS': [500, 200, 300, 100, 400],\n",
    "        'Other_Column': ['A', 'B', 'C', 'D', 'E']}\n",
    "df = pd.DataFrame(data)\n",
    "# Sort based on 'CHROM' and 'POS'\n",
    "df.sort_values(by=['CHROM', 'POS'], inplace=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "97c1f44a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>distance_1</th>\n",
       "      <th>distance_2</th>\n",
       "      <th>distance_3</th>\n",
       "      <th>distance_4</th>\n",
       "      <th>distance_5</th>\n",
       "      <th>nearest_distance_column</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>distance_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>distance_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>distance_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs4</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>distance_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs5</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>17</td>\n",
       "      <td>21</td>\n",
       "      <td>distance_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  rsID  distance_1  distance_2  distance_3  distance_4  distance_5  \\\n",
       "0  rs1          10          12          11          13           9   \n",
       "1  rs2           5           6           7           8           4   \n",
       "2  rs3           8           9          10          11           7   \n",
       "3  rs4          15          14          13          12          16   \n",
       "4  rs5          20          18          19          17          21   \n",
       "\n",
       "  nearest_distance_column  \n",
       "0              distance_5  \n",
       "1              distance_5  \n",
       "2              distance_5  \n",
       "3              distance_4  \n",
       "4              distance_4  "
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example data\n",
    "data = {\n",
    "    'rsID': ['rs1', 'rs2', 'rs3', 'rs4', 'rs5'],\n",
    "    'distance_1': [10, 5, 8, 15, 20],\n",
    "    'distance_2': [12, 6, 9, 14, 18],\n",
    "    'distance_3': [11, 7, 10, 13, 19],\n",
    "    'distance_4': [13, 8, 11, 12, 17],\n",
    "    'distance_5': [9, 4, 7, 16, 21]\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Extracting distance column names\n",
    "distance_columns = df.columns[1:]\n",
    "\n",
    "# Create a new column for the nearest distance column name\n",
    "df['nearest_distance_column'] = df.apply(lambda row: min(distance_columns, key=lambda col: row[col]), axis=1)\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "481c1418",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
