{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a924bd59",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6ae233c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Desktop/Covered_regions.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGHGPTTL207_annotated_indel.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGHGPTTL207_annotated_indel.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d6e7a175",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>97</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>52%</td>\n",
       "      <td>1.7375E-10</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1331945</td>\n",
       "      <td>rs200330269</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>34.15%</td>\n",
       "      <td>1.5388E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>219</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "      <td>19</td>\n",
       "      <td>51</td>\n",
       "      <td>72.86%</td>\n",
       "      <td>1.2075E-22</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1355779</td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>88</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>72.41%</td>\n",
       "      <td>1.2841E-9</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1657358</td>\n",
       "      <td>rs377230281</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>78</td>\n",
       "      <td>112</td>\n",
       "      <td>112</td>\n",
       "      <td>88</td>\n",
       "      <td>24</td>\n",
       "      <td>21.43%</td>\n",
       "      <td>1.4999E-8</td>\n",
       "      <td>65</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2759</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153462077</td>\n",
       "      <td>rs138590468;rs369801831</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>40%</td>\n",
       "      <td>2.3315E-6</td>\n",
       "      <td>20</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2760</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153692201</td>\n",
       "      <td>rs199498083</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>113</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>40.51%</td>\n",
       "      <td>4.4277E-12</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2761</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909413</td>\n",
       "      <td>rs11405331;rs34046683;rs869136125</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>170</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.4556E-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2762</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>108</td>\n",
       "      <td>108</td>\n",
       "      <td>7</td>\n",
       "      <td>101</td>\n",
       "      <td>93.52%</td>\n",
       "      <td>7.6722E-54</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>76</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2763</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>46.88%</td>\n",
       "      <td>2.3037E-6</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2764 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                               rsID  REF ALT QUAL FILTER  \\\n",
       "0     chr1    1299382                        rs143128930   AG   A    .   PASS   \n",
       "1     chr1    1331945                        rs200330269    G  GC    .   PASS   \n",
       "2     chr1    1353987                        rs140777846  CTG   C    .   PASS   \n",
       "3     chr1    1355779                        rs201260508   GA   G    .   PASS   \n",
       "4     chr1    1657358                        rs377230281    T  TA    .   PASS   \n",
       "...    ...        ...                                ...  ...  ..  ...    ...   \n",
       "2759  chrX  153462077            rs138590468;rs369801831   TA   T    .   PASS   \n",
       "2760  chrX  153692201                        rs199498083    C  CA    .   PASS   \n",
       "2761  chrX  153909413  rs11405331;rs34046683;rs869136125    T  TG    .   PASS   \n",
       "2762  chrX  153913005               rs5904376;rs77485258    G  GC    .   PASS   \n",
       "2763  chrX  155228363                        rs781912204    A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO   GT   GQ  SDP   DP  \\\n",
       "0     ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....  0/1   97   50   50   \n",
       "1     ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....  0/1   48   41   41   \n",
       "2     ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....  0/1  219   70   70   \n",
       "3     ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...  0/1   88   29   29   \n",
       "4     ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...  0/1   78  112  112   \n",
       "...                                                 ...  ...  ...  ...  ...   \n",
       "2759  ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....  0/1   56   40   40   \n",
       "2760  ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....  0/1  113   79   79   \n",
       "2761  ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...  1/1  170   30   30   \n",
       "2762  ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...  1/1  255  108  108   \n",
       "2763  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...  0/1   56   32   32   \n",
       "\n",
       "      RD   AD    FREQ        PVAL RDF RDR ADF ADR  \n",
       "0     24   26     52%  1.7375E-10  18   6  17   9  \n",
       "1     26   14  34.15%   1.5388E-5  12  14   2  12  \n",
       "2     19   51  72.86%  1.2075E-22  12   7  34  17  \n",
       "3      8   21  72.41%   1.2841E-9   7   1   8  13  \n",
       "4     88   24  21.43%   1.4999E-8  65  23   4  20  \n",
       "...   ..  ...     ...         ...  ..  ..  ..  ..  \n",
       "2759  24   16     40%   2.3315E-6  20   4  13   3  \n",
       "2760  47   32  40.51%  4.4277E-12  30  17  23   9  \n",
       "2761   0   30    100%  8.4556E-18   0   0  16  14  \n",
       "2762   7  101  93.52%  7.6722E-54   4   3  76  25  \n",
       "2763  13   15  46.88%   2.3037E-6  12   1  12   3  \n",
       "\n",
       "[2764 rows x 20 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGHGPTTL207_annotated_indel.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7d532a25",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>97</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>52%</td>\n",
       "      <td>1.7375E-10</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1331945</td>\n",
       "      <td>rs200330269</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>34.15%</td>\n",
       "      <td>1.5388E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>219</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "      <td>19</td>\n",
       "      <td>51</td>\n",
       "      <td>72.86%</td>\n",
       "      <td>1.2075E-22</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1355779</td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>88</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>72.41%</td>\n",
       "      <td>1.2841E-9</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1657358</td>\n",
       "      <td>rs377230281</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>78</td>\n",
       "      <td>112</td>\n",
       "      <td>112</td>\n",
       "      <td>88</td>\n",
       "      <td>24</td>\n",
       "      <td>21.43%</td>\n",
       "      <td>1.4999E-8</td>\n",
       "      <td>65</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2759</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153462077</td>\n",
       "      <td>rs138590468;rs369801831</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>40%</td>\n",
       "      <td>2.3315E-6</td>\n",
       "      <td>20</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2760</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153692201</td>\n",
       "      <td>rs199498083</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>113</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>40.51%</td>\n",
       "      <td>4.4277E-12</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2761</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909413</td>\n",
       "      <td>rs11405331;rs34046683;rs869136125</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>170</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.4556E-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2762</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>108</td>\n",
       "      <td>108</td>\n",
       "      <td>7</td>\n",
       "      <td>101</td>\n",
       "      <td>93.52%</td>\n",
       "      <td>7.6722E-54</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>76</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2763</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>46.88%</td>\n",
       "      <td>2.3037E-6</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2764 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                               rsID  REF ALT QUAL FILTER  \\\n",
       "0     chr1    1299382                        rs143128930   AG   A    .   PASS   \n",
       "1     chr1    1331945                        rs200330269    G  GC    .   PASS   \n",
       "2     chr1    1353987                        rs140777846  CTG   C    .   PASS   \n",
       "3     chr1    1355779                        rs201260508   GA   G    .   PASS   \n",
       "4     chr1    1657358                        rs377230281    T  TA    .   PASS   \n",
       "...    ...        ...                                ...  ...  ..  ...    ...   \n",
       "2759  chrX  153462077            rs138590468;rs369801831   TA   T    .   PASS   \n",
       "2760  chrX  153692201                        rs199498083    C  CA    .   PASS   \n",
       "2761  chrX  153909413  rs11405331;rs34046683;rs869136125    T  TG    .   PASS   \n",
       "2762  chrX  153913005               rs5904376;rs77485258    G  GC    .   PASS   \n",
       "2763  chrX  155228363                        rs781912204    A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO   GT   GQ  SDP   DP  \\\n",
       "0     ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....  0/1   97   50   50   \n",
       "1     ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....  0/1   48   41   41   \n",
       "2     ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....  0/1  219   70   70   \n",
       "3     ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...  0/1   88   29   29   \n",
       "4     ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...  0/1   78  112  112   \n",
       "...                                                 ...  ...  ...  ...  ...   \n",
       "2759  ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....  0/1   56   40   40   \n",
       "2760  ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....  0/1  113   79   79   \n",
       "2761  ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...  1/1  170   30   30   \n",
       "2762  ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...  1/1  255  108  108   \n",
       "2763  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...  0/1   56   32   32   \n",
       "\n",
       "      RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \n",
       "0     24   26     52%  1.7375E-10  18   6  17   9   1   0  Heterozygous  \n",
       "1     26   14  34.15%   1.5388E-5  12  14   2  12   1   0  Heterozygous  \n",
       "2     19   51  72.86%  1.2075E-22  12   7  34  17   1   0  Heterozygous  \n",
       "3      8   21  72.41%   1.2841E-9   7   1   8  13   1   0  Heterozygous  \n",
       "4     88   24  21.43%   1.4999E-8  65  23   4  20   1   0  Heterozygous  \n",
       "...   ..  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...  \n",
       "2759  24   16     40%   2.3315E-6  20   4  13   3   1   0  Heterozygous  \n",
       "2760  47   32  40.51%  4.4277E-12  30  17  23   9   1   0  Heterozygous  \n",
       "2761   0   30    100%  8.4556E-18   0   0  16  14   0   1    Homozygous  \n",
       "2762   7  101  93.52%  7.6722E-54   4   3  76  25   0   1    Homozygous  \n",
       "2763  13   15  46.88%   2.3037E-6  12   1  12   3   1   0  Heterozygous  \n",
       "\n",
       "[2764 rows x 23 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "eaa97694",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>97</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>52%</td>\n",
       "      <td>1.7375E-10</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ACAP3:116983</td>\n",
       "      <td>ACAP3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1331945</td>\n",
       "      <td>rs200330269</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>34.15%</td>\n",
       "      <td>1.5388E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TAS1R3:83756</td>\n",
       "      <td>TAS1R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>219</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "      <td>19</td>\n",
       "      <td>51</td>\n",
       "      <td>72.86%</td>\n",
       "      <td>1.2075E-22</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1355779</td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>88</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>72.41%</td>\n",
       "      <td>1.2841E-9</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1657358</td>\n",
       "      <td>rs377230281</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>78</td>\n",
       "      <td>112</td>\n",
       "      <td>112</td>\n",
       "      <td>88</td>\n",
       "      <td>24</td>\n",
       "      <td>21.43%</td>\n",
       "      <td>1.4999E-8</td>\n",
       "      <td>65</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CDK11B:984</td>\n",
       "      <td>CDK11B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2759</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153462077</td>\n",
       "      <td>rs138590468;rs369801831</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>40%</td>\n",
       "      <td>2.3315E-6</td>\n",
       "      <td>20</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HAUS7:55559,HAUS7:55559</td>\n",
       "      <td>HAUS7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2760</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153692201</td>\n",
       "      <td>rs199498083</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>113</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>40.51%</td>\n",
       "      <td>4.4277E-12</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLC6A8:6535</td>\n",
       "      <td>SLC6A8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2761</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909413</td>\n",
       "      <td>rs11405331;rs34046683;rs869136125</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>170</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.4556E-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4:393,ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>ARHGAP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2762</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>108</td>\n",
       "      <td>108</td>\n",
       "      <td>7</td>\n",
       "      <td>101</td>\n",
       "      <td>93.52%</td>\n",
       "      <td>7.6722E-54</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>76</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>ARHGAP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2763</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>46.88%</td>\n",
       "      <td>2.3037E-6</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>VBP1:7411</td>\n",
       "      <td>VBP1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2764 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                               rsID  REF ALT QUAL FILTER  \\\n",
       "0     chr1    1299382                        rs143128930   AG   A    .   PASS   \n",
       "1     chr1    1331945                        rs200330269    G  GC    .   PASS   \n",
       "2     chr1    1353987                        rs140777846  CTG   C    .   PASS   \n",
       "3     chr1    1355779                        rs201260508   GA   G    .   PASS   \n",
       "4     chr1    1657358                        rs377230281    T  TA    .   PASS   \n",
       "...    ...        ...                                ...  ...  ..  ...    ...   \n",
       "2759  chrX  153462077            rs138590468;rs369801831   TA   T    .   PASS   \n",
       "2760  chrX  153692201                        rs199498083    C  CA    .   PASS   \n",
       "2761  chrX  153909413  rs11405331;rs34046683;rs869136125    T  TG    .   PASS   \n",
       "2762  chrX  153913005               rs5904376;rs77485258    G  GC    .   PASS   \n",
       "2763  chrX  155228363                        rs781912204    A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO   GT   GQ  SDP   DP  \\\n",
       "0     ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....  0/1   97   50   50   \n",
       "1     ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....  0/1   48   41   41   \n",
       "2     ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....  0/1  219   70   70   \n",
       "3     ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...  0/1   88   29   29   \n",
       "4     ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...  0/1   78  112  112   \n",
       "...                                                 ...  ...  ...  ...  ...   \n",
       "2759  ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....  0/1   56   40   40   \n",
       "2760  ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....  0/1  113   79   79   \n",
       "2761  ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...  1/1  170   30   30   \n",
       "2762  ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...  1/1  255  108  108   \n",
       "2763  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...  0/1   56   32   32   \n",
       "\n",
       "      RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0     24   26     52%  1.7375E-10  18   6  17   9   1   0  Heterozygous   \n",
       "1     26   14  34.15%   1.5388E-5  12  14   2  12   1   0  Heterozygous   \n",
       "2     19   51  72.86%  1.2075E-22  12   7  34  17   1   0  Heterozygous   \n",
       "3      8   21  72.41%   1.2841E-9   7   1   8  13   1   0  Heterozygous   \n",
       "4     88   24  21.43%   1.4999E-8  65  23   4  20   1   0  Heterozygous   \n",
       "...   ..  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "2759  24   16     40%   2.3315E-6  20   4  13   3   1   0  Heterozygous   \n",
       "2760  47   32  40.51%  4.4277E-12  30  17  23   9   1   0  Heterozygous   \n",
       "2761   0   30    100%  8.4556E-18   0   0  16  14   0   1    Homozygous   \n",
       "2762   7  101  93.52%  7.6722E-54   4   3  76  25   0   1    Homozygous   \n",
       "2763  13   15  46.88%   2.3037E-6  12   1  12   3   1   0  Heterozygous   \n",
       "\n",
       "                                Gene_Name Gene Name  \n",
       "0                            ACAP3:116983     ACAP3  \n",
       "1                            TAS1R3:83756    TAS1R3  \n",
       "2                             MXRA8:54587     MXRA8  \n",
       "3                             MXRA8:54587     MXRA8  \n",
       "4                              CDK11B:984    CDK11B  \n",
       "...                                   ...       ...  \n",
       "2759              HAUS7:55559,HAUS7:55559     HAUS7  \n",
       "2760                          SLC6A8:6535    SLC6A8  \n",
       "2761  ARHGAP4:393,ARHGAP4:393,ARHGAP4:393   ARHGAP4  \n",
       "2762              ARHGAP4:393,ARHGAP4:393   ARHGAP4  \n",
       "2763                            VBP1:7411      VBP1  \n",
       "\n",
       "[2764 rows x 25 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene_Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d7aa5ab2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>97</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>52%</td>\n",
       "      <td>1.7375E-10</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ACAP3:116983</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1331945</td>\n",
       "      <td>rs200330269</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>34.15%</td>\n",
       "      <td>1.5388E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TAS1R3:83756</td>\n",
       "      <td>TAS1R3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>219</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "      <td>19</td>\n",
       "      <td>51</td>\n",
       "      <td>72.86%</td>\n",
       "      <td>1.2075E-22</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1355779</td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>88</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>72.41%</td>\n",
       "      <td>1.2841E-9</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1657358</td>\n",
       "      <td>rs377230281</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>78</td>\n",
       "      <td>112</td>\n",
       "      <td>112</td>\n",
       "      <td>88</td>\n",
       "      <td>24</td>\n",
       "      <td>21.43%</td>\n",
       "      <td>1.4999E-8</td>\n",
       "      <td>65</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CDK11B:984</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2759</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153462077</td>\n",
       "      <td>rs138590468;rs369801831</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>40%</td>\n",
       "      <td>2.3315E-6</td>\n",
       "      <td>20</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HAUS7:55559,HAUS7:55559</td>\n",
       "      <td>HAUS7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2760</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153692201</td>\n",
       "      <td>rs199498083</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>113</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>40.51%</td>\n",
       "      <td>4.4277E-12</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLC6A8:6535</td>\n",
       "      <td>SLC6A8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2761</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909413</td>\n",
       "      <td>rs11405331;rs34046683;rs869136125</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>170</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.4556E-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4:393,ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2762</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>108</td>\n",
       "      <td>108</td>\n",
       "      <td>7</td>\n",
       "      <td>101</td>\n",
       "      <td>93.52%</td>\n",
       "      <td>7.6722E-54</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>76</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2763</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>46.88%</td>\n",
       "      <td>2.3037E-6</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>VBP1:7411</td>\n",
       "      <td>VBP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2764 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                               rsID  REF ALT QUAL FILTER  \\\n",
       "0     chr1    1299382                        rs143128930   AG   A    .   PASS   \n",
       "1     chr1    1331945                        rs200330269    G  GC    .   PASS   \n",
       "2     chr1    1353987                        rs140777846  CTG   C    .   PASS   \n",
       "3     chr1    1355779                        rs201260508   GA   G    .   PASS   \n",
       "4     chr1    1657358                        rs377230281    T  TA    .   PASS   \n",
       "...    ...        ...                                ...  ...  ..  ...    ...   \n",
       "2759  chrX  153462077            rs138590468;rs369801831   TA   T    .   PASS   \n",
       "2760  chrX  153692201                        rs199498083    C  CA    .   PASS   \n",
       "2761  chrX  153909413  rs11405331;rs34046683;rs869136125    T  TG    .   PASS   \n",
       "2762  chrX  153913005               rs5904376;rs77485258    G  GC    .   PASS   \n",
       "2763  chrX  155228363                        rs781912204    A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO   GT   GQ  SDP   DP  \\\n",
       "0     ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....  0/1   97   50   50   \n",
       "1     ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....  0/1   48   41   41   \n",
       "2     ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....  0/1  219   70   70   \n",
       "3     ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...  0/1   88   29   29   \n",
       "4     ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...  0/1   78  112  112   \n",
       "...                                                 ...  ...  ...  ...  ...   \n",
       "2759  ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....  0/1   56   40   40   \n",
       "2760  ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....  0/1  113   79   79   \n",
       "2761  ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...  1/1  170   30   30   \n",
       "2762  ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...  1/1  255  108  108   \n",
       "2763  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...  0/1   56   32   32   \n",
       "\n",
       "      RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0     24   26     52%  1.7375E-10  18   6  17   9   1   0  Heterozygous   \n",
       "1     26   14  34.15%   1.5388E-5  12  14   2  12   1   0  Heterozygous   \n",
       "2     19   51  72.86%  1.2075E-22  12   7  34  17   1   0  Heterozygous   \n",
       "3      8   21  72.41%   1.2841E-9   7   1   8  13   1   0  Heterozygous   \n",
       "4     88   24  21.43%   1.4999E-8  65  23   4  20   1   0  Heterozygous   \n",
       "...   ..  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "2759  24   16     40%   2.3315E-6  20   4  13   3   1   0  Heterozygous   \n",
       "2760  47   32  40.51%  4.4277E-12  30  17  23   9   1   0  Heterozygous   \n",
       "2761   0   30    100%  8.4556E-18   0   0  16  14   0   1    Homozygous   \n",
       "2762   7  101  93.52%  7.6722E-54   4   3  76  25   0   1    Homozygous   \n",
       "2763  13   15  46.88%   2.3037E-6  12   1  12   3   1   0  Heterozygous   \n",
       "\n",
       "                                Gene_Name Gene Name  CSQ  csq  \n",
       "0                            ACAP3:116983     ACAP3  NaN  NaN  \n",
       "1                            TAS1R3:83756    TAS1R3  NaN  NaN  \n",
       "2                             MXRA8:54587     MXRA8  NaN  NaN  \n",
       "3                             MXRA8:54587     MXRA8  NaN  NaN  \n",
       "4                              CDK11B:984    CDK11B  NaN  NaN  \n",
       "...                                   ...       ...  ...  ...  \n",
       "2759              HAUS7:55559,HAUS7:55559     HAUS7  NaN  NaN  \n",
       "2760                          SLC6A8:6535    SLC6A8  NaN  NaN  \n",
       "2761  ARHGAP4:393,ARHGAP4:393,ARHGAP4:393   ARHGAP4  NaN  NaN  \n",
       "2762              ARHGAP4:393,ARHGAP4:393   ARHGAP4  NaN  NaN  \n",
       "2763                            VBP1:7411      VBP1  NaN  NaN  \n",
       "\n",
       "[2764 rows x 27 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "918ffab3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>97</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>52%</td>\n",
       "      <td>1.7375E-10</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ACAP3:116983</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1331945</td>\n",
       "      <td>rs200330269</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>34.15%</td>\n",
       "      <td>1.5388E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TAS1R3:83756</td>\n",
       "      <td>TAS1R3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>219</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "      <td>19</td>\n",
       "      <td>51</td>\n",
       "      <td>72.86%</td>\n",
       "      <td>1.2075E-22</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1355779</td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>88</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>72.41%</td>\n",
       "      <td>1.2841E-9</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1657358</td>\n",
       "      <td>rs377230281</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>78</td>\n",
       "      <td>112</td>\n",
       "      <td>112</td>\n",
       "      <td>88</td>\n",
       "      <td>24</td>\n",
       "      <td>21.43%</td>\n",
       "      <td>1.4999E-8</td>\n",
       "      <td>65</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CDK11B:984</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2759</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153462077</td>\n",
       "      <td>rs138590468;rs369801831</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>40%</td>\n",
       "      <td>2.3315E-6</td>\n",
       "      <td>20</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HAUS7:55559,HAUS7:55559</td>\n",
       "      <td>HAUS7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2760</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153692201</td>\n",
       "      <td>rs199498083</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>113</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>47</td>\n",
       "      <td>32</td>\n",
       "      <td>40.51%</td>\n",
       "      <td>4.4277E-12</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLC6A8:6535</td>\n",
       "      <td>SLC6A8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2761</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153909413</td>\n",
       "      <td>rs11405331;rs34046683;rs869136125</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>170</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.4556E-18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4:393,ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2762</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>108</td>\n",
       "      <td>108</td>\n",
       "      <td>7</td>\n",
       "      <td>101</td>\n",
       "      <td>93.52%</td>\n",
       "      <td>7.6722E-54</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>76</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2763</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>46.88%</td>\n",
       "      <td>2.3037E-6</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>VBP1:7411</td>\n",
       "      <td>VBP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2764 rows × 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                               rsID  REF ALT QUAL FILTER  \\\n",
       "0     chr1    1299382                        rs143128930   AG   A    .   PASS   \n",
       "1     chr1    1331945                        rs200330269    G  GC    .   PASS   \n",
       "2     chr1    1353987                        rs140777846  CTG   C    .   PASS   \n",
       "3     chr1    1355779                        rs201260508   GA   G    .   PASS   \n",
       "4     chr1    1657358                        rs377230281    T  TA    .   PASS   \n",
       "...    ...        ...                                ...  ...  ..  ...    ...   \n",
       "2759  chrX  153462077            rs138590468;rs369801831   TA   T    .   PASS   \n",
       "2760  chrX  153692201                        rs199498083    C  CA    .   PASS   \n",
       "2761  chrX  153909413  rs11405331;rs34046683;rs869136125    T  TG    .   PASS   \n",
       "2762  chrX  153913005               rs5904376;rs77485258    G  GC    .   PASS   \n",
       "2763  chrX  155228363                        rs781912204    A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO   GT   GQ  SDP   DP  \\\n",
       "0     ADP=50;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....  0/1   97   50   50   \n",
       "1     ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8105,0....  0/1   48   41   41   \n",
       "2     ADP=70;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....  0/1  219   70   70   \n",
       "3     ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0,1;COMMO...  0/1   88   29   29   \n",
       "4     ADP=112;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=CDK...  0/1   78  112  112   \n",
       "...                                                 ...  ...  ...  ...  ...   \n",
       "2759  ADP=40;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5624,0....  0/1   56   40   40   \n",
       "2760  ADP=79;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8972,0....  0/1  113   79   79   \n",
       "2761  ADP=30;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,.,1,0,1...  1/1  170   30   30   \n",
       "2762  ADP=108;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.000264...  1/1  255  108  108   \n",
       "2763  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...  0/1   56   32   32   \n",
       "\n",
       "      RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0     24   26     52%  1.7375E-10  18   6  17   9   1   0  Heterozygous   \n",
       "1     26   14  34.15%   1.5388E-5  12  14   2  12   1   0  Heterozygous   \n",
       "2     19   51  72.86%  1.2075E-22  12   7  34  17   1   0  Heterozygous   \n",
       "3      8   21  72.41%   1.2841E-9   7   1   8  13   1   0  Heterozygous   \n",
       "4     88   24  21.43%   1.4999E-8  65  23   4  20   1   0  Heterozygous   \n",
       "...   ..  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "2759  24   16     40%   2.3315E-6  20   4  13   3   1   0  Heterozygous   \n",
       "2760  47   32  40.51%  4.4277E-12  30  17  23   9   1   0  Heterozygous   \n",
       "2761   0   30    100%  8.4556E-18   0   0  16  14   0   1    Homozygous   \n",
       "2762   7  101  93.52%  7.6722E-54   4   3  76  25   0   1    Homozygous   \n",
       "2763  13   15  46.88%   2.3037E-6  12   1  12   3   1   0  Heterozygous   \n",
       "\n",
       "                                Gene_Name Gene Name  CSQ  csq ClinVar_CLNDN  \\\n",
       "0                            ACAP3:116983     ACAP3  NaN  NaN           NaN   \n",
       "1                            TAS1R3:83756    TAS1R3  NaN  NaN           NaN   \n",
       "2                             MXRA8:54587     MXRA8  NaN  NaN           NaN   \n",
       "3                             MXRA8:54587     MXRA8  NaN  NaN           NaN   \n",
       "4                              CDK11B:984    CDK11B  NaN  NaN           NaN   \n",
       "...                                   ...       ...  ...  ...           ...   \n",
       "2759              HAUS7:55559,HAUS7:55559     HAUS7  NaN  NaN           NaN   \n",
       "2760                          SLC6A8:6535    SLC6A8  NaN  NaN           NaN   \n",
       "2761  ARHGAP4:393,ARHGAP4:393,ARHGAP4:393   ARHGAP4  NaN  NaN           NaN   \n",
       "2762              ARHGAP4:393,ARHGAP4:393   ARHGAP4  NaN  NaN           NaN   \n",
       "2763                            VBP1:7411      VBP1  NaN  NaN           NaN   \n",
       "\n",
       "     CLIN_SIG ClinVar_CLNREVSTAT ClinVar HGVSC HGVSP PolyPhen BIOTYPE EXON  \\\n",
       "0         NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "1         NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "2         NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "3         NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "4         NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "...       ...                ...     ...   ...   ...      ...     ...  ...   \n",
       "2759      NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "2760      NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "2761      NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "2762      NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "2763      NaN                NaN     NaN   NaN   NaN      NaN     NaN  NaN   \n",
       "\n",
       "     INTRON Protein_position Amino_acids Codons STRAND PUBMED Consequence  \\\n",
       "0       NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "1       NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "2       NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "3       NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "4       NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "...     ...              ...         ...    ...    ...    ...         ...   \n",
       "2759    NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "2760    NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "2761    NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "2762    NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "2763    NaN              NaN         NaN    NaN    NaN    NaN         NaN   \n",
       "\n",
       "     IMPACT SIFT   AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF  \\\n",
       "0       NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "1       NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "2       NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "3       NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "4       NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "...     ...  ...  ...    ...    ...    ...    ...    ...        ...   \n",
       "2759    NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "2760    NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "2761    NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "2762    NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "2763    NaN  NaN  NaN    NaN    NaN    NaN    NaN    NaN        NaN   \n",
       "\n",
       "     gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0               NaN            NaN            NaN            NaN   \n",
       "1               NaN            NaN            NaN            NaN   \n",
       "2               NaN            NaN            NaN            NaN   \n",
       "3               NaN            NaN            NaN            NaN   \n",
       "4               NaN            NaN            NaN            NaN   \n",
       "...             ...            ...            ...            ...   \n",
       "2759            NaN            NaN            NaN            NaN   \n",
       "2760            NaN            NaN            NaN            NaN   \n",
       "2761            NaN            NaN            NaN            NaN   \n",
       "2762            NaN            NaN            NaN            NaN   \n",
       "2763            NaN            NaN            NaN            NaN   \n",
       "\n",
       "     gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0               NaN            NaN            NaN            NaN        NaN   \n",
       "1               NaN            NaN            NaN            NaN        NaN   \n",
       "2               NaN            NaN            NaN            NaN        NaN   \n",
       "3               NaN            NaN            NaN            NaN        NaN   \n",
       "4               NaN            NaN            NaN            NaN        NaN   \n",
       "...             ...            ...            ...            ...        ...   \n",
       "2759            NaN            NaN            NaN            NaN        NaN   \n",
       "2760            NaN            NaN            NaN            NaN        NaN   \n",
       "2761            NaN            NaN            NaN            NaN        NaN   \n",
       "2762            NaN            NaN            NaN            NaN        NaN   \n",
       "2763            NaN            NaN            NaN            NaN        NaN   \n",
       "\n",
       "     gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0               NaN            NaN            NaN            NaN   \n",
       "1               NaN            NaN            NaN            NaN   \n",
       "2               NaN            NaN            NaN            NaN   \n",
       "3               NaN            NaN            NaN            NaN   \n",
       "4               NaN            NaN            NaN            NaN   \n",
       "...             ...            ...            ...            ...   \n",
       "2759            NaN            NaN            NaN            NaN   \n",
       "2760            NaN            NaN            NaN            NaN   \n",
       "2761            NaN            NaN            NaN            NaN   \n",
       "2762            NaN            NaN            NaN            NaN   \n",
       "2763            NaN            NaN            NaN            NaN   \n",
       "\n",
       "     gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0               NaN            NaN            NaN            NaN   \n",
       "1               NaN            NaN            NaN            NaN   \n",
       "2               NaN            NaN            NaN            NaN   \n",
       "3               NaN            NaN            NaN            NaN   \n",
       "4               NaN            NaN            NaN            NaN   \n",
       "...             ...            ...            ...            ...   \n",
       "2759            NaN            NaN            NaN            NaN   \n",
       "2760            NaN            NaN            NaN            NaN   \n",
       "2761            NaN            NaN            NaN            NaN   \n",
       "2762            NaN            NaN            NaN            NaN   \n",
       "2763            NaN            NaN            NaN            NaN   \n",
       "\n",
       "     gnomADg_OTH_AF gnomADg_SAS_AF MAX_AF MAX_AF_POPS  \n",
       "0               NaN            NaN    NaN         NaN  \n",
       "1               NaN            NaN    NaN         NaN  \n",
       "2               NaN            NaN    NaN         NaN  \n",
       "3               NaN            NaN    NaN         NaN  \n",
       "4               NaN            NaN    NaN         NaN  \n",
       "...             ...            ...    ...         ...  \n",
       "2759            NaN            NaN    NaN         NaN  \n",
       "2760            NaN            NaN    NaN         NaN  \n",
       "2761            NaN            NaN    NaN         NaN  \n",
       "2762            NaN            NaN    NaN         NaN  \n",
       "2763            NaN            NaN    NaN         NaN  \n",
       "\n",
       "[2764 rows x 73 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "337b6a73",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>38352177</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>38352183</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>38352181</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>38352256</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrX</td>\n",
       "      <td>38352332</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1882</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915692</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1883</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107916864</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1884</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915532</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1885</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917349</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1886</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917418</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1887 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS Literature\n",
       "0     chrX   38352177        Yes\n",
       "1     chrX   38352183        Yes\n",
       "2     chrX   38352181        Yes\n",
       "3     chrX   38352256        Yes\n",
       "4     chrX   38352332        Yes\n",
       "...    ...        ...        ...\n",
       "1882  chr7  107915692        Yes\n",
       "1883  chr7  107916864        Yes\n",
       "1884  chr7  107915532        Yes\n",
       "1885  chr7  107917349        Yes\n",
       "1886  chr7  107917418        Yes\n",
       "\n",
       "[1887 rows x 3 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Cerebellarataxia_Lit_final_Positions_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'Literature']]\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d145a714",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Cerebellarataxia.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "00fd1dce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7962863</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7984930</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7984981</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7984954</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7970951</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3146</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915692</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3147</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107916864</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3148</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915532</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3149</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917349</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3150</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917418</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3151 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS        Literature\n",
       "0     chr1    7962863         Parkinson\n",
       "1     chr1    7984930         Parkinson\n",
       "2     chr1    7984981         Parkinson\n",
       "3     chr1    7984954         Parkinson\n",
       "4     chr1    7970951         Parkinson\n",
       "...    ...        ...               ...\n",
       "3146  chr7  107915692  Cerebellarataxia\n",
       "3147  chr7  107916864  Cerebellarataxia\n",
       "3148  chr7  107915532  Cerebellarataxia\n",
       "3149  chr7  107917349  Cerebellarataxia\n",
       "3150  chr7  107917418  Cerebellarataxia\n",
       "\n",
       "[3151 rows x 3 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/parkin_Cerebellarataxia_lit_pos.xlsx')\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "0354dca5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYO6</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MRE11</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRRK2</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CLN1</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>CLN2</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>CLN3</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>PHYH</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name        Gene_Match\n",
       "0       MYO6         Parkinson\n",
       "1       GBA1         Parkinson\n",
       "2       PRKN         Parkinson\n",
       "3      MRE11         Parkinson\n",
       "4      LRRK2         Parkinson\n",
       "..       ...               ...\n",
       "79      CLN1  Cerebellarataxia\n",
       "80      CLN2  Cerebellarataxia\n",
       "81      CLN3  Cerebellarataxia\n",
       "82      PHYH  Cerebellarataxia\n",
       "83      PEX7  Cerebellarataxia\n",
       "\n",
       "[84 rows x 2 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Park_cere_genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0820df31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=120;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Thr162Ala</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>T162A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2:p.Ser290%3D</td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>S290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2:n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5:n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38700</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>C|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>C|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe_AFR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11:c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5:p.Arg365His</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>365</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>R365H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|3_prime_UTR_variant|MODIFIER|IL9R|ENSG000001...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7:c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|WASIR1|ENSG...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|IL9R|ENSG00...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "1      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "2      chr1     685694            .   T   C    .   PASS   \n",
       "2      chr1     685694            .   T   C    .   PASS   \n",
       "2      chr1     685694            .   T   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "38700  chrX  156003433    rs2037999   T   C    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=120;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  120  120   \n",
       "1      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1   82   98   98   \n",
       "2      ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   34   28   28   \n",
       "2      ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   34   28   28   \n",
       "2      ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   34   28   28   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "38700  ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0....  1/1   58   11   11   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "\n",
       "       RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0       0  120    100%  1.1001E-71   0   0  86  34   0   1    Homozygous   \n",
       "1      73   25  25.51%   5.1517E-9  41  32  11  14   1   0  Heterozygous   \n",
       "2      18   10  35.71%   3.6855E-4  17   1   9   1   1   0  Heterozygous   \n",
       "2      18   10  35.71%   3.6855E-4  17   1   9   1   1   0  Heterozygous   \n",
       "2      18   10  35.71%   3.6855E-4  17   1   9   1   1   0  Heterozygous   \n",
       "...    ..  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "38700   0   11    100%   1.4176E-6   0   0   7   4   0   1    Homozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "\n",
       "         Gene_Name Gene Name  \\\n",
       "0      OR4F5:79501     OR4F5   \n",
       "1      OR4F5:79501     OR4F5   \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "...            ...       ...   \n",
       "38700    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "1      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...                                                  ...   \n",
       "38700  C|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      G|missense_variant|MODERATE|OR4F5|ENSG00000186...                 \n",
       "1      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...                 \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...                 \n",
       "2      C|intron_variant&non_coding_transcript_variant...                 \n",
       "2      C|intron_variant&non_coding_transcript_variant...                 \n",
       "...                                                  ...           ...   \n",
       "38700  C|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...  not_provided   \n",
       "38701  A|3_prime_UTR_variant|MODIFIER|IL9R|ENSG000001...  not_provided   \n",
       "38701  A|downstream_gene_variant|MODIFIER|WASIR1|ENSG...  not_provided   \n",
       "38701  A|downstream_gene_variant|MODIFIER|IL9R|ENSG00...  not_provided   \n",
       "\n",
       "      CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                             \n",
       "1                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "...        ...                                  ...     ...   \n",
       "38700                                                         \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "\n",
       "                                  HGVSC                          HGVSP  \\\n",
       "0            ENST00000641515.2:c.484A>G  ENSP00000493376.2:p.Thr162Ala   \n",
       "1            ENST00000641515.2:c.870T>C  ENSP00000493376.2:p.Ser290%3D   \n",
       "2                                                                        \n",
       "2      ENST00000419394.2:n.480+17991A>G                                  \n",
       "2      ENST00000440200.5:n.169+34338A>G                                  \n",
       "...                                 ...                            ...   \n",
       "38700                                                                    \n",
       "38701      ENST00000244174.11:c.1094G>A  ENSP00000244174.5:p.Arg365His   \n",
       "38701        ENST00000369423.7:c.*85G>A                                  \n",
       "38701                                                                    \n",
       "38701                                                                    \n",
       "\n",
       "        PolyPhen                         BIOTYPE EXON INTRON Protein_position  \\\n",
       "0      benign(0)                  protein_coding  3/3                     162   \n",
       "1                                 protein_coding  3/3                     290   \n",
       "2                                 protein_coding                                \n",
       "2                           processed_transcript         3/3                    \n",
       "2                           processed_transcript         1/2                    \n",
       "...          ...                             ...  ...    ...              ...   \n",
       "38700                                   enhancer                                \n",
       "38701  benign(0)                  protein_coding  9/9                     365   \n",
       "38701                             protein_coding  9/9                           \n",
       "38701                                     lncRNA                                \n",
       "38701             protein_coding_CDS_not_defined                                \n",
       "\n",
       "      Amino_acids   Codons STRAND PUBMED  \\\n",
       "0             T/A  Aca/Gca      1          \n",
       "1               S  tcT/tcC      1          \n",
       "2                              -1          \n",
       "2                              -1          \n",
       "2                              -1          \n",
       "...           ...      ...    ...    ...   \n",
       "38700                                      \n",
       "38701         R/H  cGt/cAt      1          \n",
       "38701                           1          \n",
       "38701                          -1          \n",
       "38701                           1          \n",
       "\n",
       "                                        Consequence    IMPACT  \\\n",
       "0                                  missense_variant  MODERATE   \n",
       "1                                synonymous_variant       LOW   \n",
       "2                           downstream_gene_variant  MODIFIER   \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "...                                             ...       ...   \n",
       "38700                     regulatory_region_variant  MODIFIER   \n",
       "38701                              missense_variant  MODERATE   \n",
       "38701                           3_prime_UTR_variant  MODIFIER   \n",
       "38701                       downstream_gene_variant  MODIFIER   \n",
       "38701                       downstream_gene_variant  MODIFIER   \n",
       "\n",
       "                  SIFT      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0      tolerated(0.92)                                                 \n",
       "1                       0.6881  0.407  0.6254  0.876  0.7942  0.8098   \n",
       "2                                                                      \n",
       "2                                                                      \n",
       "2                                                                      \n",
       "...                ...     ...    ...     ...    ...     ...     ...   \n",
       "38700                                                                  \n",
       "38701  tolerated(0.56)                                                 \n",
       "38701                                                                  \n",
       "38701                                                                  \n",
       "38701                                                                  \n",
       "\n",
       "      gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0         0.9497         0.6075         0.9514         0.9767         0.9994   \n",
       "1         0.7209          0.307         0.6749         0.8131         0.7793   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...          ...            ...            ...            ...            ...   \n",
       "38700      0.441         0.8217         0.3483         0.4304          0.702   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0             0.9916         0.9726         0.9506         0.9854      0.846   \n",
       "1             0.8652         0.8438         0.7932         0.8049     0.4864   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...              ...            ...            ...            ...        ...   \n",
       "38700         0.5274         0.3271         0.4035         0.5578     0.5073   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.5948          0.998         0.8951         0.9784   \n",
       "1             0.2916         0.6511         0.4782         0.6714   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700          0.809          0.356         0.3772         0.4374   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0             0.9998         0.9907            0.9         0.9674   \n",
       "1             0.6047         0.6987         0.6441         0.6751   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700         0.7099         0.5742         0.4209         0.3323   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.8624         0.9772  0.9998  gnomADg_EAS   \n",
       "1             0.4659          0.622   0.876          EAS   \n",
       "2                                                          \n",
       "2                                                          \n",
       "2                                                          \n",
       "...              ...            ...     ...          ...   \n",
       "38700         0.4516         0.5465  0.8217  gnomADe_AFR   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "\n",
       "      Protein Position and Amino Acid  \n",
       "0                               T162A  \n",
       "1                                S290  \n",
       "2                                 NaN  \n",
       "2                                 NaN  \n",
       "2                                 NaN  \n",
       "...                               ...  \n",
       "38700                             NaN  \n",
       "38701                           R365H  \n",
       "38701                             NaN  \n",
       "38701                             NaN  \n",
       "38701                             NaN  \n",
       "\n",
       "[340657 rows x 74 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d2ca7fc1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=120;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Thr162Ala</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>T162A</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2:p.Ser290%3D</td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>S290</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2:n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5:n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38700</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>C|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>C|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe_AFR</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11:c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5:p.Arg365His</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>365</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>R365H</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|3_prime_UTR_variant|MODIFIER|IL9R|ENSG000001...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7:c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|WASIR1|ENSG...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R:3581</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>A|missense_variant|MODERATE|IL9R|ENSG000001243...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|IL9R|ENSG00...</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "1      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "2      chr1     685694            .   T   C    .   PASS   \n",
       "2      chr1     685694            .   T   C    .   PASS   \n",
       "2      chr1     685694            .   T   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "38700  chrX  156003433    rs2037999   T   C    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "38701  chrX  156009937  rs147385831   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=120;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  120  120   \n",
       "1      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1   82   98   98   \n",
       "2      ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   34   28   28   \n",
       "2      ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   34   28   28   \n",
       "2      ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   34   28   28   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "38700  ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0....  1/1   58   11   11   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "38701  ADP=20;WT=0;HET=1;HOM=0;NC=0;ASP;CFL;GENEINFO=...  0/1   27   20   20   \n",
       "\n",
       "       RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0       0  120    100%  1.1001E-71   0   0  86  34   0   1    Homozygous   \n",
       "1      73   25  25.51%   5.1517E-9  41  32  11  14   1   0  Heterozygous   \n",
       "2      18   10  35.71%   3.6855E-4  17   1   9   1   1   0  Heterozygous   \n",
       "2      18   10  35.71%   3.6855E-4  17   1   9   1   1   0  Heterozygous   \n",
       "2      18   10  35.71%   3.6855E-4  17   1   9   1   1   0  Heterozygous   \n",
       "...    ..  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "38700   0   11    100%   1.4176E-6   0   0   7   4   0   1    Homozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "38701  12    8     40%    1.638E-3  12   0   8   0   1   0  Heterozygous   \n",
       "\n",
       "         Gene_Name Gene Name  \\\n",
       "0      OR4F5:79501     OR4F5   \n",
       "1      OR4F5:79501     OR4F5   \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "...            ...       ...   \n",
       "38700    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "38701    IL9R:3581      IL9R   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "1      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...                                                  ...   \n",
       "38700  C|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      G|missense_variant|MODERATE|OR4F5|ENSG00000186...                 \n",
       "1      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...                 \n",
       "2      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...                 \n",
       "2      C|intron_variant&non_coding_transcript_variant...                 \n",
       "2      C|intron_variant&non_coding_transcript_variant...                 \n",
       "...                                                  ...           ...   \n",
       "38700  C|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "38701  A|missense_variant|MODERATE|IL9R|ENSG000001243...  not_provided   \n",
       "38701  A|3_prime_UTR_variant|MODIFIER|IL9R|ENSG000001...  not_provided   \n",
       "38701  A|downstream_gene_variant|MODIFIER|WASIR1|ENSG...  not_provided   \n",
       "38701  A|downstream_gene_variant|MODIFIER|IL9R|ENSG00...  not_provided   \n",
       "\n",
       "      CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                             \n",
       "1                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "...        ...                                  ...     ...   \n",
       "38700                                                         \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "\n",
       "                                  HGVSC                          HGVSP  \\\n",
       "0            ENST00000641515.2:c.484A>G  ENSP00000493376.2:p.Thr162Ala   \n",
       "1            ENST00000641515.2:c.870T>C  ENSP00000493376.2:p.Ser290%3D   \n",
       "2                                                                        \n",
       "2      ENST00000419394.2:n.480+17991A>G                                  \n",
       "2      ENST00000440200.5:n.169+34338A>G                                  \n",
       "...                                 ...                            ...   \n",
       "38700                                                                    \n",
       "38701      ENST00000244174.11:c.1094G>A  ENSP00000244174.5:p.Arg365His   \n",
       "38701        ENST00000369423.7:c.*85G>A                                  \n",
       "38701                                                                    \n",
       "38701                                                                    \n",
       "\n",
       "        PolyPhen                         BIOTYPE EXON INTRON Protein_position  \\\n",
       "0      benign(0)                  protein_coding  3/3                     162   \n",
       "1                                 protein_coding  3/3                     290   \n",
       "2                                 protein_coding                                \n",
       "2                           processed_transcript         3/3                    \n",
       "2                           processed_transcript         1/2                    \n",
       "...          ...                             ...  ...    ...              ...   \n",
       "38700                                   enhancer                                \n",
       "38701  benign(0)                  protein_coding  9/9                     365   \n",
       "38701                             protein_coding  9/9                           \n",
       "38701                                     lncRNA                                \n",
       "38701             protein_coding_CDS_not_defined                                \n",
       "\n",
       "      Amino_acids   Codons STRAND PUBMED  \\\n",
       "0             T/A  Aca/Gca      1          \n",
       "1               S  tcT/tcC      1          \n",
       "2                              -1          \n",
       "2                              -1          \n",
       "2                              -1          \n",
       "...           ...      ...    ...    ...   \n",
       "38700                                      \n",
       "38701         R/H  cGt/cAt      1          \n",
       "38701                           1          \n",
       "38701                          -1          \n",
       "38701                           1          \n",
       "\n",
       "                                        Consequence    IMPACT  \\\n",
       "0                                  missense_variant  MODERATE   \n",
       "1                                synonymous_variant       LOW   \n",
       "2                           downstream_gene_variant  MODIFIER   \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "...                                             ...       ...   \n",
       "38700                     regulatory_region_variant  MODIFIER   \n",
       "38701                              missense_variant  MODERATE   \n",
       "38701                           3_prime_UTR_variant  MODIFIER   \n",
       "38701                       downstream_gene_variant  MODIFIER   \n",
       "38701                       downstream_gene_variant  MODIFIER   \n",
       "\n",
       "                  SIFT      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  SAS_AF  \\\n",
       "0      tolerated(0.92)                                                 \n",
       "1                       0.6881  0.407  0.6254  0.876  0.7942  0.8098   \n",
       "2                                                                      \n",
       "2                                                                      \n",
       "2                                                                      \n",
       "...                ...     ...    ...     ...    ...     ...     ...   \n",
       "38700                                                                  \n",
       "38701  tolerated(0.56)                                                 \n",
       "38701                                                                  \n",
       "38701                                                                  \n",
       "38701                                                                  \n",
       "\n",
       "      gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0         0.9497         0.6075         0.9514         0.9767         0.9994   \n",
       "1         0.7209          0.307         0.6749         0.8131         0.7793   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...          ...            ...            ...            ...            ...   \n",
       "38700      0.441         0.8217         0.3483         0.4304          0.702   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "38701    0.01405        0.02867       0.007828        0.01496        0.01587   \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0             0.9916         0.9726         0.9506         0.9854      0.846   \n",
       "1             0.8652         0.8438         0.7932         0.8049     0.4864   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...              ...            ...            ...            ...        ...   \n",
       "38700         0.5274         0.3271         0.4035         0.5578     0.5073   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "38701       0.005438       0.008185        0.02348         0.0411    0.02864   \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.5948          0.998         0.8951         0.9784   \n",
       "1             0.2916         0.6511         0.4782         0.6714   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700          0.809          0.356         0.3772         0.4374   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "38701        0.03691        0.04154        0.01196        0.03051   \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0             0.9998         0.9907            0.9         0.9674   \n",
       "1             0.6047         0.6987         0.6441         0.6751   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700         0.7099         0.5742         0.4209         0.3323   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "38701        0.02488         0.0722        0.03478        0.02331   \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.8624         0.9772  0.9998  gnomADg_EAS   \n",
       "1             0.4659          0.622   0.876          EAS   \n",
       "2                                                          \n",
       "2                                                          \n",
       "2                                                          \n",
       "...              ...            ...     ...          ...   \n",
       "38700         0.4516         0.5465  0.8217  gnomADe_AFR   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "\n",
       "      Protein Position and Amino Acid               HGVSc HGVSc (Transcript)  \\\n",
       "0                               T162A   ENST00000641515.2           c.484A>G   \n",
       "1                                S290   ENST00000641515.2           c.870T>C   \n",
       "2                                 NaN                                   None   \n",
       "2                                 NaN   ENST00000419394.2     n.480+17991A>G   \n",
       "2                                 NaN   ENST00000440200.5     n.169+34338A>G   \n",
       "...                               ...                 ...                ...   \n",
       "38700                             NaN                                   None   \n",
       "38701                           R365H  ENST00000244174.11          c.1094G>A   \n",
       "38701                             NaN   ENST00000369423.7           c.*85G>A   \n",
       "38701                             NaN                                   None   \n",
       "38701                             NaN                                   None   \n",
       "\n",
       "                   HGVSp HGVSp (Transcript)  \n",
       "0      ENSP00000493376.2        p.Thr162Ala  \n",
       "1      ENSP00000493376.2        p.Ser290%3D  \n",
       "2                                      None  \n",
       "2                                      None  \n",
       "2                                      None  \n",
       "...                  ...                ...  \n",
       "38700                                  None  \n",
       "38701  ENSP00000244174.5        p.Arg365His  \n",
       "38701                                  None  \n",
       "38701                                  None  \n",
       "38701                                  None  \n",
       "\n",
       "[340657 rows x 78 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "163289de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38700</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe_AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0         OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1         OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "...         ...          ...   ...        ...  ..  ..           ...   \n",
       "38700      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                  missense_variant  MODERATE                 \n",
       "1                                synonymous_variant       LOW                 \n",
       "2                           downstream_gene_variant  MODIFIER                 \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER                 \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "38700                     regulatory_region_variant  MODIFIER                 \n",
       "38701                              missense_variant  MODERATE  not_provided   \n",
       "38701                           3_prime_UTR_variant  MODIFIER  not_provided   \n",
       "38701                       downstream_gene_variant  MODIFIER  not_provided   \n",
       "38701                       downstream_gene_variant  MODIFIER  not_provided   \n",
       "\n",
       "      CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                             \n",
       "1                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "...        ...                                  ...     ...   \n",
       "38700                                                         \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                       None                      \n",
       "2       ENST00000419394.2     n.480+17991A>G                      \n",
       "2       ENST00000440200.5     n.169+34338A>G                      \n",
       "...                   ...                ...                ...   \n",
       "38700                                   None                      \n",
       "38701  ENST00000244174.11          c.1094G>A  ENSP00000244174.5   \n",
       "38701   ENST00000369423.7           c.*85G>A                      \n",
       "38701                                   None                      \n",
       "38701                                   None                      \n",
       "\n",
       "      HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL RDF  \\\n",
       "0            p.Thr162Ala  1/1  255  120  120   0  120    100%  1.1001E-71   0   \n",
       "1            p.Ser290%3D  0/1   82   98   98  73   25  25.51%   5.1517E-9  41   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "...                  ...  ...  ...  ...  ...  ..  ...     ...         ...  ..   \n",
       "38700               None  1/1   58   11   11   0   11    100%   1.4176E-6   0   \n",
       "38701        p.Arg365His  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "\n",
       "      RDR ADF ADR             SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  \\\n",
       "0       0  86  34  tolerated(0.92)  benign(0)                                 \n",
       "1      32  11  14                              0.6881  0.407  0.6254  0.876   \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "...    ..  ..  ..              ...        ...     ...    ...     ...    ...   \n",
       "38700   0   7   4                                                             \n",
       "38701   0   8   0  tolerated(0.56)  benign(0)                                 \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "\n",
       "       EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                         0.9497         0.6075         0.9514         0.9767   \n",
       "1      0.7942  0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "...       ...     ...        ...            ...            ...            ...   \n",
       "38700                      0.441         0.8217         0.3483         0.4304   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "      gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0             0.9994         0.9916         0.9726         0.9506   \n",
       "1             0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700          0.702         0.5274         0.3271         0.4035   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "      gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0             0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1             0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...              ...        ...            ...            ...            ...   \n",
       "38700         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "      gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0             0.9784         0.9998         0.9907            0.9   \n",
       "1             0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700         0.4374         0.7099         0.5742         0.4209   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "      gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.9674         0.8624         0.9772  0.9998  gnomADg_EAS   \n",
       "1             0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "...              ...            ...            ...     ...          ...   \n",
       "38700         0.3323         0.4516         0.5465  0.8217  gnomADe_AFR   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "\n",
       "                              BIOTYPE EXON INTRON  \\\n",
       "0                      protein_coding  3/3          \n",
       "1                      protein_coding  3/3          \n",
       "2                      protein_coding               \n",
       "2                processed_transcript         3/3   \n",
       "2                processed_transcript         1/2   \n",
       "...                               ...  ...    ...   \n",
       "38700                        enhancer               \n",
       "38701                  protein_coding  9/9          \n",
       "38701                  protein_coding  9/9          \n",
       "38701                          lncRNA               \n",
       "38701  protein_coding_CDS_not_defined               \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                               T162A  Aca/Gca      1         \n",
       "1                                S290  tcT/tcC      1         \n",
       "2                                 NaN              -1         \n",
       "2                                 NaN              -1         \n",
       "2                                 NaN              -1         \n",
       "...                               ...      ...    ...    ...  \n",
       "38700                             NaN                         \n",
       "38701                           R365H  cGt/cAt      1         \n",
       "38701                             NaN               1         \n",
       "38701                             NaN              -1         \n",
       "38701                             NaN               1         \n",
       "\n",
       "[340657 rows x 66 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf[['Gene Name', 'rsID','CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2033b7a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38700</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe_AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0         OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1         OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "...         ...          ...   ...        ...  ..  ..           ...   \n",
       "38700      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                  missense_variant  MODERATE                 \n",
       "1                                synonymous_variant       LOW                 \n",
       "2                           downstream_gene_variant  MODIFIER                 \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER                 \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "38700                     regulatory_region_variant  MODIFIER                 \n",
       "38701                              missense_variant  MODERATE  not_provided   \n",
       "38701                           3_prime_UTR_variant  MODIFIER  not_provided   \n",
       "38701                       downstream_gene_variant  MODIFIER  not_provided   \n",
       "38701                       downstream_gene_variant  MODIFIER  not_provided   \n",
       "\n",
       "      CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                             \n",
       "1                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "...        ...                                  ...     ...   \n",
       "38700                                                         \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "38701   benign  criteria_provided&_single_submitter  771064   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                       None                      \n",
       "2       ENST00000419394.2     n.480+17991A>G                      \n",
       "2       ENST00000440200.5     n.169+34338A>G                      \n",
       "...                   ...                ...                ...   \n",
       "38700                                   None                      \n",
       "38701  ENST00000244174.11          c.1094G>A  ENSP00000244174.5   \n",
       "38701   ENST00000369423.7           c.*85G>A                      \n",
       "38701                                   None                      \n",
       "38701                                   None                      \n",
       "\n",
       "      HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL RDF  \\\n",
       "0            p.Thr162Ala  1/1  255  120  120   0  120    100%  1.1001E-71   0   \n",
       "1            p.Ser290%3D  0/1   82   98   98  73   25  25.51%   5.1517E-9  41   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "...                  ...  ...  ...  ...  ...  ..  ...     ...         ...  ..   \n",
       "38700               None  1/1   58   11   11   0   11    100%   1.4176E-6   0   \n",
       "38701        p.Arg365His  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "\n",
       "      RDR ADF ADR             SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  \\\n",
       "0       0  86  34  tolerated(0.92)  benign(0)                                 \n",
       "1      32  11  14                              0.6881  0.407  0.6254  0.876   \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "...    ..  ..  ..              ...        ...     ...    ...     ...    ...   \n",
       "38700   0   7   4                                                             \n",
       "38701   0   8   0  tolerated(0.56)  benign(0)                                 \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "\n",
       "       EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                         0.9497         0.6075         0.9514         0.9767   \n",
       "1      0.7942  0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "...       ...     ...        ...            ...            ...            ...   \n",
       "38700                      0.441         0.8217         0.3483         0.4304   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "      gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0             0.9994         0.9916         0.9726         0.9506   \n",
       "1             0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700          0.702         0.5274         0.3271         0.4035   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "      gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0             0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1             0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...              ...        ...            ...            ...            ...   \n",
       "38700         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "      gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0             0.9784         0.9998         0.9907            0.9   \n",
       "1             0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700         0.4374         0.7099         0.5742         0.4209   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "      gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.9674         0.8624         0.9772  0.9998  gnomADg_EAS   \n",
       "1             0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "...              ...            ...            ...     ...          ...   \n",
       "38700         0.3323         0.4516         0.5465  0.8217  gnomADe_AFR   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg_FIN   \n",
       "\n",
       "                              BIOTYPE EXON INTRON  \\\n",
       "0                      protein_coding  3/3          \n",
       "1                      protein_coding  3/3          \n",
       "2                      protein_coding               \n",
       "2                processed_transcript         3/3   \n",
       "2                processed_transcript         1/2   \n",
       "...                               ...  ...    ...   \n",
       "38700                        enhancer               \n",
       "38701                  protein_coding  9/9          \n",
       "38701                  protein_coding  9/9          \n",
       "38701                          lncRNA               \n",
       "38701  protein_coding_CDS_not_defined               \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                               T162A  Aca/Gca      1         \n",
       "1                                S290  tcT/tcC      1         \n",
       "2                                 NaN              -1         \n",
       "2                                 NaN              -1         \n",
       "2                                 NaN              -1         \n",
       "...                               ...      ...    ...    ...  \n",
       "38700                             NaN                         \n",
       "38701                           R365H  cGt/cAt      1         \n",
       "38701                             NaN               1         \n",
       "38701                             NaN              -1         \n",
       "38701                             NaN               1         \n",
       "\n",
       "[340657 rows x 66 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "# Print the modified DataFrame\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cd6260d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38700</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0         OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1         OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "...         ...          ...   ...        ...  ..  ..           ...   \n",
       "38700      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                  missense variant  MODERATE                 \n",
       "1                                synonymous variant       LOW                 \n",
       "2                           downstream gene variant  MODIFIER                 \n",
       "2      intron variant,non coding transcript variant  MODIFIER                 \n",
       "2      intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "38700                     regulatory region variant  MODIFIER                 \n",
       "38701                              missense variant  MODERATE  not provided   \n",
       "38701                           3 prime UTR variant  MODIFIER  not provided   \n",
       "38701                       downstream gene variant  MODIFIER  not provided   \n",
       "38701                       downstream gene variant  MODIFIER  not provided   \n",
       "\n",
       "      CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                             \n",
       "1                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "...        ...                                  ...     ...   \n",
       "38700                                                         \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                       None                      \n",
       "2       ENST00000419394.2     n.480+17991A>G                      \n",
       "2       ENST00000440200.5     n.169+34338A>G                      \n",
       "...                   ...                ...                ...   \n",
       "38700                                   None                      \n",
       "38701  ENST00000244174.11          c.1094G>A  ENSP00000244174.5   \n",
       "38701   ENST00000369423.7           c.*85G>A                      \n",
       "38701                                   None                      \n",
       "38701                                   None                      \n",
       "\n",
       "      HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL RDF  \\\n",
       "0            p.Thr162Ala  1/1  255  120  120   0  120    100%  1.1001E-71   0   \n",
       "1            p.Ser290%3D  0/1   82   98   98  73   25  25.51%   5.1517E-9  41   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "...                  ...  ...  ...  ...  ...  ..  ...     ...         ...  ..   \n",
       "38700               None  1/1   58   11   11   0   11    100%   1.4176E-6   0   \n",
       "38701        p.Arg365His  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "\n",
       "      RDR ADF ADR             SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  \\\n",
       "0       0  86  34  tolerated(0.92)  benign(0)                                 \n",
       "1      32  11  14                              0.6881  0.407  0.6254  0.876   \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "...    ..  ..  ..              ...        ...     ...    ...     ...    ...   \n",
       "38700   0   7   4                                                             \n",
       "38701   0   8   0  tolerated(0.56)  benign(0)                                 \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "\n",
       "       EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                         0.9497         0.6075         0.9514         0.9767   \n",
       "1      0.7942  0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "...       ...     ...        ...            ...            ...            ...   \n",
       "38700                      0.441         0.8217         0.3483         0.4304   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "      gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0             0.9994         0.9916         0.9726         0.9506   \n",
       "1             0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700          0.702         0.5274         0.3271         0.4035   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "      gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0             0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1             0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...              ...        ...            ...            ...            ...   \n",
       "38700         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "      gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0             0.9784         0.9998         0.9907            0.9   \n",
       "1             0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700         0.4374         0.7099         0.5742         0.4209   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "      gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1             0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "...              ...            ...            ...     ...          ...   \n",
       "38700         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                              BIOTYPE EXON INTRON  \\\n",
       "0                      protein coding  3/3          \n",
       "1                      protein coding  3/3          \n",
       "2                      protein coding               \n",
       "2                processed transcript         3/3   \n",
       "2                processed transcript         1/2   \n",
       "...                               ...  ...    ...   \n",
       "38700                        enhancer               \n",
       "38701                  protein coding  9/9          \n",
       "38701                  protein coding  9/9          \n",
       "38701                          lncRNA               \n",
       "38701  protein coding CDS not defined               \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                               T162A  Aca/Gca      1         \n",
       "1                                S290  tcT/tcC      1         \n",
       "2                                 nan              -1         \n",
       "2                                 nan              -1         \n",
       "2                                 nan              -1         \n",
       "...                               ...      ...    ...    ...  \n",
       "38700                             nan                         \n",
       "38701                           R365H  cGt/cAt      1         \n",
       "38701                             nan               1         \n",
       "38701                             nan              -1         \n",
       "38701                             nan               1         \n",
       "\n",
       "[340657 rows x 66 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "#vcf_final.loc[:, vcf_final.columns != 'CHROM'] = vcf_final.loc[:, vcf_final.columns != 'CHROM'].astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "25b2d80c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38700</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38701</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 67 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0         OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1         OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "2                          .  chr1     685694   T   C  Heterozygous   \n",
       "...         ...          ...   ...        ...  ..  ..           ...   \n",
       "38700      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "38701      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                  missense variant  MODERATE                 \n",
       "1                                synonymous variant       LOW                 \n",
       "2                           downstream gene variant  MODIFIER                 \n",
       "2      intron variant,non coding transcript variant  MODIFIER                 \n",
       "2      intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "38700                     regulatory region variant  MODIFIER                 \n",
       "38701                              missense variant  MODERATE  not provided   \n",
       "38701                           3 prime UTR variant  MODIFIER  not provided   \n",
       "38701                       downstream gene variant  MODIFIER  not provided   \n",
       "38701                       downstream gene variant  MODIFIER  not provided   \n",
       "\n",
       "      CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                             \n",
       "1                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "2                                                             \n",
       "...        ...                                  ...     ...   \n",
       "38700                                                         \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "38701   benign  criteria provided, single submitter  771064   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                       None                      \n",
       "2       ENST00000419394.2     n.480+17991A>G                      \n",
       "2       ENST00000440200.5     n.169+34338A>G                      \n",
       "...                   ...                ...                ...   \n",
       "38700                                   None                      \n",
       "38701  ENST00000244174.11          c.1094G>A  ENSP00000244174.5   \n",
       "38701   ENST00000369423.7           c.*85G>A                      \n",
       "38701                                   None                      \n",
       "38701                                   None                      \n",
       "\n",
       "      HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL RDF  \\\n",
       "0            p.Thr162Ala  1/1  255  120  120   0  120    100%  1.1001E-71   0   \n",
       "1            p.Ser290%3D  0/1   82   98   98  73   25  25.51%   5.1517E-9  41   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "2                   None  0/1   34   28   28  18   10  35.71%   3.6855E-4  17   \n",
       "...                  ...  ...  ...  ...  ...  ..  ...     ...         ...  ..   \n",
       "38700               None  1/1   58   11   11   0   11    100%   1.4176E-6   0   \n",
       "38701        p.Arg365His  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "38701               None  0/1   27   20   20  12    8     40%    1.638E-3  12   \n",
       "\n",
       "      RDR ADF ADR             SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  \\\n",
       "0       0  86  34  tolerated(0.92)  benign(0)                                 \n",
       "1      32  11  14                              0.6881  0.407  0.6254  0.876   \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "2       1   9   1                                                             \n",
       "...    ..  ..  ..              ...        ...     ...    ...     ...    ...   \n",
       "38700   0   7   4                                                             \n",
       "38701   0   8   0  tolerated(0.56)  benign(0)                                 \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "38701   0   8   0                                                             \n",
       "\n",
       "       EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                         0.9497         0.6075         0.9514         0.9767   \n",
       "1      0.7942  0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "2                                                                               \n",
       "...       ...     ...        ...            ...            ...            ...   \n",
       "38700                      0.441         0.8217         0.3483         0.4304   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "38701                    0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "      gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0             0.9994         0.9916         0.9726         0.9506   \n",
       "1             0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700          0.702         0.5274         0.3271         0.4035   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "38701        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "      gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0             0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1             0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "2                                                                              \n",
       "...              ...        ...            ...            ...            ...   \n",
       "38700         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "38701         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "      gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0             0.9784         0.9998         0.9907            0.9   \n",
       "1             0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "2                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "38700         0.4374         0.7099         0.5742         0.4209   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "38701        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "      gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0             0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1             0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "2                                                                         \n",
       "...              ...            ...            ...     ...          ...   \n",
       "38700         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "38701        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                              BIOTYPE EXON INTRON  \\\n",
       "0                      protein coding  3/3          \n",
       "1                      protein coding  3/3          \n",
       "2                      protein coding               \n",
       "2                processed transcript         3/3   \n",
       "2                processed transcript         1/2   \n",
       "...                               ...  ...    ...   \n",
       "38700                        enhancer               \n",
       "38701                  protein coding  9/9          \n",
       "38701                  protein coding  9/9          \n",
       "38701                          lncRNA               \n",
       "38701  protein coding CDS not defined               \n",
       "\n",
       "      Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                               T162A  Aca/Gca      1          \n",
       "1                                S290  tcT/tcC      1          \n",
       "2                                 nan              -1          \n",
       "2                                 nan              -1          \n",
       "2                                 nan              -1          \n",
       "...                               ...      ...    ...    ...   \n",
       "38700                             nan                          \n",
       "38701                           R365H  cGt/cAt      1          \n",
       "38701                             nan               1          \n",
       "38701                             nan              -1          \n",
       "38701                             nan               1          \n",
       "\n",
       "                     consequence  \n",
       "0               missense variant  \n",
       "1             synonymous variant  \n",
       "2        downstream gene variant  \n",
       "2                 intron variant  \n",
       "2                 intron variant  \n",
       "...                          ...  \n",
       "38700  regulatory region variant  \n",
       "38701           missense variant  \n",
       "38701        3 prime UTR variant  \n",
       "38701    downstream gene variant  \n",
       "38701    downstream gene variant  \n",
       "\n",
       "[340657 rows x 67 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "30a13b6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b1e64086",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                   missense variant  MODERATE                 \n",
       "1                                 synonymous variant       LOW                 \n",
       "2                            downstream gene variant  MODIFIER                 \n",
       "3       intron variant,non coding transcript variant  MODIFIER                 \n",
       "4       intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                              ...       ...           ...   \n",
       "340652                     regulatory region variant  MODIFIER                 \n",
       "340653                              missense variant  MODERATE  not provided   \n",
       "340654                           3 prime UTR variant  MODIFIER  not provided   \n",
       "340655                       downstream gene variant  MODIFIER  not provided   \n",
       "340656                       downstream gene variant  MODIFIER  not provided   \n",
       "\n",
       "       CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                              \n",
       "1                                                              \n",
       "2                                                              \n",
       "3                                                              \n",
       "4                                                              \n",
       "...         ...                                  ...     ...   \n",
       "340652                                                         \n",
       "340653   benign  criteria provided, single submitter  771064   \n",
       "340654   benign  criteria provided, single submitter  771064   \n",
       "340655   benign  criteria provided, single submitter  771064   \n",
       "340656   benign  criteria provided, single submitter  771064   \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0        ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1        ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                        None                      \n",
       "3        ENST00000419394.2     n.480+17991A>G                      \n",
       "4        ENST00000440200.5     n.169+34338A>G                      \n",
       "...                    ...                ...                ...   \n",
       "340652                                   None                      \n",
       "340653  ENST00000244174.11          c.1094G>A  ENSP00000244174.5   \n",
       "340654   ENST00000369423.7           c.*85G>A                      \n",
       "340655                                   None                      \n",
       "340656                                   None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL  \\\n",
       "0             p.Thr162Ala  1/1  255  120  120   0  120    100%  1.1001E-71   \n",
       "1             p.Ser290%3D  0/1   82   98   98  73   25  25.51%   5.1517E-9   \n",
       "2                    None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "3                    None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "4                    None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "...                   ...  ...  ...  ...  ...  ..  ...     ...         ...   \n",
       "340652               None  1/1   58   11   11   0   11    100%   1.4176E-6   \n",
       "340653        p.Arg365His  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "340654               None  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "340655               None  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "340656               None  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "\n",
       "       RDF RDR ADF ADR             SIFT   PolyPhen      AF AFR_AF  AMR_AF  \\\n",
       "0        0   0  86  34  tolerated(0.92)  benign(0)                          \n",
       "1       41  32  11  14                              0.6881  0.407  0.6254   \n",
       "2       17   1   9   1                                                      \n",
       "3       17   1   9   1                                                      \n",
       "4       17   1   9   1                                                      \n",
       "...     ..  ..  ..  ..              ...        ...     ...    ...     ...   \n",
       "340652   0   0   7   4                                                      \n",
       "340653  12   0   8   0  tolerated(0.56)  benign(0)                          \n",
       "340654  12   0   8   0                                                      \n",
       "340655  12   0   8   0                                                      \n",
       "340656  12   0   8   0                                                      \n",
       "\n",
       "       EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                                 0.9497         0.6075         0.9514   \n",
       "1       0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...       ...     ...     ...        ...            ...            ...   \n",
       "340652                             0.441         0.8217         0.3483   \n",
       "340653                           0.01405        0.02867       0.007828   \n",
       "340654                           0.01405        0.02867       0.007828   \n",
       "340655                           0.01405        0.02867       0.007828   \n",
       "340656                           0.01405        0.02867       0.007828   \n",
       "\n",
       "       gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0              0.9767         0.9994         0.9916         0.9726   \n",
       "1              0.8131         0.7793         0.8652         0.8438   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4304          0.702         0.5274         0.3271   \n",
       "340653        0.01496        0.01587       0.005438       0.008185   \n",
       "340654        0.01496        0.01587       0.005438       0.008185   \n",
       "340655        0.01496        0.01587       0.005438       0.008185   \n",
       "340656        0.01496        0.01587       0.005438       0.008185   \n",
       "\n",
       "       gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0              0.9506         0.9854      0.846         0.5948          0.998   \n",
       "1              0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...            ...        ...            ...            ...   \n",
       "340652         0.4035         0.5578     0.5073          0.809          0.356   \n",
       "340653        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "340654        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "340655        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "340656        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "\n",
       "       gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0              0.8951         0.9784         0.9998         0.9907   \n",
       "1              0.4782         0.6714         0.6047         0.6987   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.3772         0.4374         0.7099         0.5742   \n",
       "340653        0.01196        0.03051        0.02488         0.0722   \n",
       "340654        0.01196        0.03051        0.02488         0.0722   \n",
       "340655        0.01196        0.03051        0.02488         0.0722   \n",
       "340656        0.01196        0.03051        0.02488         0.0722   \n",
       "\n",
       "       gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  \\\n",
       "0                 0.9         0.9674         0.8624         0.9772  0.9998   \n",
       "1              0.6441         0.6751         0.4659          0.622   0.876   \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...               ...            ...            ...            ...     ...   \n",
       "340652         0.4209         0.3323         0.4516         0.5465  0.8217   \n",
       "340653        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "340654        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "340655        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "340656        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "\n",
       "        MAX_AF_POPS                         BIOTYPE EXON INTRON  \\\n",
       "0       gnomADg EAS                  protein coding  3/3          \n",
       "1               EAS                  protein coding  3/3          \n",
       "2                                    protein coding               \n",
       "3                              processed transcript         3/3   \n",
       "4                              processed transcript         1/2   \n",
       "...             ...                             ...  ...    ...   \n",
       "340652  gnomADe AFR                        enhancer               \n",
       "340653  gnomADg FIN                  protein coding  9/9          \n",
       "340654  gnomADg FIN                  protein coding  9/9          \n",
       "340655  gnomADg FIN                          lncRNA               \n",
       "340656  gnomADg FIN  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                T162A  Aca/Gca      1          \n",
       "1                                 S290  tcT/tcC      1          \n",
       "2                                  nan              -1          \n",
       "3                                  nan              -1          \n",
       "4                                  nan              -1          \n",
       "...                                ...      ...    ...    ...   \n",
       "340652                             nan                          \n",
       "340653                           R365H  cGt/cAt      1          \n",
       "340654                             nan               1          \n",
       "340655                             nan              -1          \n",
       "340656                             nan               1          \n",
       "\n",
       "                      consequence Consequence_score  \n",
       "0                missense variant              7/10  \n",
       "1              synonymous variant              3/10  \n",
       "2         downstream gene variant              2/10  \n",
       "3                  intron variant              2/10  \n",
       "4                  intron variant              2/10  \n",
       "...                           ...               ...  \n",
       "340652  regulatory region variant              2/10  \n",
       "340653           missense variant              7/10  \n",
       "340654        3 prime UTR variant              3/10  \n",
       "340655    downstream gene variant              2/10  \n",
       "340656    downstream gene variant              2/10  \n",
       "\n",
       "[340657 rows x 68 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "9ec0d189",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     IMPACT  IMPACT_score\n",
       "0      HIGH          10.0\n",
       "1  MODERATE           5.0\n",
       "2       LOW           2.5\n",
       "3  MODIFIER           1.5"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "33968543",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                   missense variant  MODERATE                 \n",
       "1                                 synonymous variant       LOW                 \n",
       "2                            downstream gene variant  MODIFIER                 \n",
       "3       intron variant,non coding transcript variant  MODIFIER                 \n",
       "4       intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                              ...       ...           ...   \n",
       "340652                     regulatory region variant  MODIFIER                 \n",
       "340653                              missense variant  MODERATE  not provided   \n",
       "340654                           3 prime UTR variant  MODIFIER  not provided   \n",
       "340655                       downstream gene variant  MODIFIER  not provided   \n",
       "340656                       downstream gene variant  MODIFIER  not provided   \n",
       "\n",
       "       CLIN_SIG                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                                                              \n",
       "1                                                              \n",
       "2                                                              \n",
       "3                                                              \n",
       "4                                                              \n",
       "...         ...                                  ...     ...   \n",
       "340652                                                         \n",
       "340653   benign  criteria provided, single submitter  771064   \n",
       "340654   benign  criteria provided, single submitter  771064   \n",
       "340655   benign  criteria provided, single submitter  771064   \n",
       "340656   benign  criteria provided, single submitter  771064   \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0        ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1        ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                        None                      \n",
       "3        ENST00000419394.2     n.480+17991A>G                      \n",
       "4        ENST00000440200.5     n.169+34338A>G                      \n",
       "...                    ...                ...                ...   \n",
       "340652                                   None                      \n",
       "340653  ENST00000244174.11          c.1094G>A  ENSP00000244174.5   \n",
       "340654   ENST00000369423.7           c.*85G>A                      \n",
       "340655                                   None                      \n",
       "340656                                   None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP  RD   AD    FREQ        PVAL  \\\n",
       "0             p.Thr162Ala  1/1  255  120  120   0  120    100%  1.1001E-71   \n",
       "1             p.Ser290%3D  0/1   82   98   98  73   25  25.51%   5.1517E-9   \n",
       "2                    None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "3                    None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "4                    None  0/1   34   28   28  18   10  35.71%   3.6855E-4   \n",
       "...                   ...  ...  ...  ...  ...  ..  ...     ...         ...   \n",
       "340652               None  1/1   58   11   11   0   11    100%   1.4176E-6   \n",
       "340653        p.Arg365His  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "340654               None  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "340655               None  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "340656               None  0/1   27   20   20  12    8     40%    1.638E-3   \n",
       "\n",
       "       RDF RDR ADF ADR             SIFT   PolyPhen      AF AFR_AF  AMR_AF  \\\n",
       "0        0   0  86  34  tolerated(0.92)  benign(0)                          \n",
       "1       41  32  11  14                              0.6881  0.407  0.6254   \n",
       "2       17   1   9   1                                                      \n",
       "3       17   1   9   1                                                      \n",
       "4       17   1   9   1                                                      \n",
       "...     ..  ..  ..  ..              ...        ...     ...    ...     ...   \n",
       "340652   0   0   7   4                                                      \n",
       "340653  12   0   8   0  tolerated(0.56)  benign(0)                          \n",
       "340654  12   0   8   0                                                      \n",
       "340655  12   0   8   0                                                      \n",
       "340656  12   0   8   0                                                      \n",
       "\n",
       "       EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                                 0.9497         0.6075         0.9514   \n",
       "1       0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...       ...     ...     ...        ...            ...            ...   \n",
       "340652                             0.441         0.8217         0.3483   \n",
       "340653                           0.01405        0.02867       0.007828   \n",
       "340654                           0.01405        0.02867       0.007828   \n",
       "340655                           0.01405        0.02867       0.007828   \n",
       "340656                           0.01405        0.02867       0.007828   \n",
       "\n",
       "       gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0              0.9767         0.9994         0.9916         0.9726   \n",
       "1              0.8131         0.7793         0.8652         0.8438   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4304          0.702         0.5274         0.3271   \n",
       "340653        0.01496        0.01587       0.005438       0.008185   \n",
       "340654        0.01496        0.01587       0.005438       0.008185   \n",
       "340655        0.01496        0.01587       0.005438       0.008185   \n",
       "340656        0.01496        0.01587       0.005438       0.008185   \n",
       "\n",
       "       gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0              0.9506         0.9854      0.846         0.5948          0.998   \n",
       "1              0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...            ...        ...            ...            ...   \n",
       "340652         0.4035         0.5578     0.5073          0.809          0.356   \n",
       "340653        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "340654        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "340655        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "340656        0.02348         0.0411    0.02864        0.03691        0.04154   \n",
       "\n",
       "       gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0              0.8951         0.9784         0.9998         0.9907   \n",
       "1              0.4782         0.6714         0.6047         0.6987   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.3772         0.4374         0.7099         0.5742   \n",
       "340653        0.01196        0.03051        0.02488         0.0722   \n",
       "340654        0.01196        0.03051        0.02488         0.0722   \n",
       "340655        0.01196        0.03051        0.02488         0.0722   \n",
       "340656        0.01196        0.03051        0.02488         0.0722   \n",
       "\n",
       "       gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  \\\n",
       "0                 0.9         0.9674         0.8624         0.9772  0.9998   \n",
       "1              0.6441         0.6751         0.4659          0.622   0.876   \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...               ...            ...            ...            ...     ...   \n",
       "340652         0.4209         0.3323         0.4516         0.5465  0.8217   \n",
       "340653        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "340654        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "340655        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "340656        0.03478        0.02331        0.02312        0.04483  0.0722   \n",
       "\n",
       "        MAX_AF_POPS                         BIOTYPE EXON INTRON  \\\n",
       "0       gnomADg EAS                  protein coding  3/3          \n",
       "1               EAS                  protein coding  3/3          \n",
       "2                                    protein coding               \n",
       "3                              processed transcript         3/3   \n",
       "4                              processed transcript         1/2   \n",
       "...             ...                             ...  ...    ...   \n",
       "340652  gnomADe AFR                        enhancer               \n",
       "340653  gnomADg FIN                  protein coding  9/9          \n",
       "340654  gnomADg FIN                  protein coding  9/9          \n",
       "340655  gnomADg FIN                          lncRNA               \n",
       "340656  gnomADg FIN  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                T162A  Aca/Gca      1          \n",
       "1                                 S290  tcT/tcC      1          \n",
       "2                                  nan              -1          \n",
       "3                                  nan              -1          \n",
       "4                                  nan              -1          \n",
       "...                                ...      ...    ...    ...   \n",
       "340652                             nan                          \n",
       "340653                           R365H  cGt/cAt      1          \n",
       "340654                             nan               1          \n",
       "340655                             nan              -1          \n",
       "340656                             nan               1          \n",
       "\n",
       "                      consequence Consequence_score  IMPACT_score  \n",
       "0                missense variant              7/10           5.0  \n",
       "1              synonymous variant              3/10           2.5  \n",
       "2         downstream gene variant              2/10           1.5  \n",
       "3                  intron variant              2/10           1.5  \n",
       "4                  intron variant              2/10           1.5  \n",
       "...                           ...               ...           ...  \n",
       "340652  regulatory region variant              2/10           1.5  \n",
       "340653           missense variant              7/10           5.0  \n",
       "340654        3 prime UTR variant              3/10           1.5  \n",
       "340655    downstream gene variant              2/10           1.5  \n",
       "340656    downstream gene variant              2/10           1.5  \n",
       "\n",
       "[340657 rows x 69 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "57a4af1d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "340652                     regulatory region variant              2/10   \n",
       "340653                              missense variant              7/10   \n",
       "340654                           3 prime UTR variant              3/10   \n",
       "340655                       downstream gene variant              2/10   \n",
       "340656                       downstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       MODERATE           5.0                          \n",
       "1            LOW           2.5                          \n",
       "2       MODIFIER           1.5                          \n",
       "3       MODIFIER           1.5                          \n",
       "4       MODIFIER           1.5                          \n",
       "...          ...           ...           ...      ...   \n",
       "340652  MODIFIER           1.5                          \n",
       "340653  MODERATE           5.0  not provided   benign   \n",
       "340654  MODIFIER           1.5  not provided   benign   \n",
       "340655  MODIFIER           1.5  not provided   benign   \n",
       "340656  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                T162A  Aca/Gca      1         \n",
       "1                                 S290  tcT/tcC      1         \n",
       "2                                  nan              -1         \n",
       "3                                  nan              -1         \n",
       "4                                  nan              -1         \n",
       "...                                ...      ...    ...    ...  \n",
       "340652                             nan                         \n",
       "340653                           R365H  cGt/cAt      1         \n",
       "340654                             nan               1         \n",
       "340655                             nan              -1         \n",
       "340656                             nan               1         \n",
       "\n",
       "[340657 rows x 68 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "57fb42d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYO6</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MRE11</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRRK2</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CLN1</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>CLN2</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>CLN3</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>PHYH</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name        Gene_Match\n",
       "0       MYO6         Parkinson\n",
       "1       GBA1         Parkinson\n",
       "2       PRKN         Parkinson\n",
       "3      MRE11         Parkinson\n",
       "4      LRRK2         Parkinson\n",
       "..       ...               ...\n",
       "79      CLN1  Cerebellarataxia\n",
       "80      CLN2  Cerebellarataxia\n",
       "81      CLN3  Cerebellarataxia\n",
       "82      PHYH  Cerebellarataxia\n",
       "83      PEX7  Cerebellarataxia\n",
       "\n",
       "[84 rows x 2 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Park_cere_genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "2b901682",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "340652                     regulatory region variant              2/10   \n",
       "340653                              missense variant              7/10   \n",
       "340654                           3 prime UTR variant              3/10   \n",
       "340655                       downstream gene variant              2/10   \n",
       "340656                       downstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       MODERATE           5.0                          \n",
       "1            LOW           2.5                          \n",
       "2       MODIFIER           1.5                          \n",
       "3       MODIFIER           1.5                          \n",
       "4       MODIFIER           1.5                          \n",
       "...          ...           ...           ...      ...   \n",
       "340652  MODIFIER           1.5                          \n",
       "340653  MODERATE           5.0  not provided   benign   \n",
       "340654  MODIFIER           1.5  not provided   benign   \n",
       "340655  MODIFIER           1.5  not provided   benign   \n",
       "340656  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Gene Match  \\\n",
       "0                                T162A  Aca/Gca      1                No   \n",
       "1                                 S290  tcT/tcC      1                No   \n",
       "2                                  nan              -1                No   \n",
       "3                                  nan              -1                No   \n",
       "4                                  nan              -1                No   \n",
       "...                                ...      ...    ...    ...        ...   \n",
       "340652                             nan                                No   \n",
       "340653                           R365H  cGt/cAt      1                No   \n",
       "340654                             nan               1                No   \n",
       "340655                             nan              -1                No   \n",
       "340656                             nan               1                No   \n",
       "\n",
       "       Matched_Gene  \n",
       "0                    \n",
       "1                    \n",
       "2                    \n",
       "3                    \n",
       "4                    \n",
       "...             ...  \n",
       "340652               \n",
       "340653               \n",
       "340654               \n",
       "340655               \n",
       "340656               \n",
       "\n",
       "[340657 rows x 70 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b1ab2cb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "bf3dc643",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYO6</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MRE11</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRRK2</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CLN1</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>CLN2</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>CLN3</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>PHYH</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Matched_Gene        Gene Match\n",
       "0          MYO6         Parkinson\n",
       "1          GBA1         Parkinson\n",
       "2          PRKN         Parkinson\n",
       "3         MRE11         Parkinson\n",
       "4         LRRK2         Parkinson\n",
       "..          ...               ...\n",
       "79         CLN1  Cerebellarataxia\n",
       "80         CLN2  Cerebellarataxia\n",
       "81         CLN3  Cerebellarataxia\n",
       "82         PHYH  Cerebellarataxia\n",
       "83         PEX7  Cerebellarataxia\n",
       "\n",
       "[84 rows x 2 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = df_gene.rename({'Gene Name':'Matched_Gene', 'Gene_Match':'Gene Match'}, axis=1)\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "e2509b57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
       "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
       "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
       "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
       "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
       "       'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
       "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
       "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
       "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
       "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
       "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
       "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Gene_Match', 'Matched_Gene', 'Gene Match_x', 'Gene Match_y'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "761a100f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match_x</th>\n",
       "      <th>Gene Match_y</th>\n",
       "      <th>Gene Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "340652                     regulatory region variant              2/10   \n",
       "340653                              missense variant              7/10   \n",
       "340654                           3 prime UTR variant              3/10   \n",
       "340655                       downstream gene variant              2/10   \n",
       "340656                       downstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       MODERATE           5.0                          \n",
       "1            LOW           2.5                          \n",
       "2       MODIFIER           1.5                          \n",
       "3       MODIFIER           1.5                          \n",
       "4       MODIFIER           1.5                          \n",
       "...          ...           ...           ...      ...   \n",
       "340652  MODIFIER           1.5                          \n",
       "340653  MODERATE           5.0  not provided   benign   \n",
       "340654  MODIFIER           1.5  not provided   benign   \n",
       "340655  MODIFIER           1.5  not provided   benign   \n",
       "340656  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Gene_Match  \\\n",
       "0                                T162A  Aca/Gca      1                No   \n",
       "1                                 S290  tcT/tcC      1                No   \n",
       "2                                  nan              -1                No   \n",
       "3                                  nan              -1                No   \n",
       "4                                  nan              -1                No   \n",
       "...                                ...      ...    ...    ...        ...   \n",
       "340652                             nan                                No   \n",
       "340653                           R365H  cGt/cAt      1                No   \n",
       "340654                             nan               1                No   \n",
       "340655                             nan              -1                No   \n",
       "340656                             nan               1                No   \n",
       "\n",
       "       Matched_Gene Gene Match_x Gene Match_y Gene Match  \n",
       "0                            NaN          NaN        NaN  \n",
       "1                            NaN          NaN        NaN  \n",
       "2                            NaN          NaN        NaN  \n",
       "3                            NaN          NaN        NaN  \n",
       "4                            NaN          NaN        NaN  \n",
       "...             ...          ...          ...        ...  \n",
       "340652                       NaN          NaN        NaN  \n",
       "340653                       NaN          NaN        NaN  \n",
       "340654                       NaN          NaN        NaN  \n",
       "340655                       NaN          NaN        NaN  \n",
       "340656                       NaN          NaN        NaN  \n",
       "\n",
       "[340657 rows x 73 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "2e4eb02f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3631</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>rs2640906</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7969449</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1267406</td>\n",
       "      <td>ENST00000338639.10</td>\n",
       "      <td>c.252+45G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>126</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>96%</td>\n",
       "      <td>2.0568E-13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.3689</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>0.2549</td>\n",
       "      <td>0.3386</td>\n",
       "      <td>0.4742</td>\n",
       "      <td>0.08915</td>\n",
       "      <td>0.6243</td>\n",
       "      <td>0.2772</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.2089</td>\n",
       "      <td>0.1092</td>\n",
       "      <td>0.2623</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.06872</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.09267</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.1007</td>\n",
       "      <td>0.1774</td>\n",
       "      <td>0.2253</td>\n",
       "      <td>0.1582</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>4/6</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>PARK7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3632</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>rs2640906</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7969449</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1267406</td>\n",
       "      <td>ENST00000377488.5</td>\n",
       "      <td>c.252+45G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>126</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>96%</td>\n",
       "      <td>2.0568E-13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.3689</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>0.2549</td>\n",
       "      <td>0.3386</td>\n",
       "      <td>0.4742</td>\n",
       "      <td>0.08915</td>\n",
       "      <td>0.6243</td>\n",
       "      <td>0.2772</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.2089</td>\n",
       "      <td>0.1092</td>\n",
       "      <td>0.2623</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.06872</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.09267</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.1007</td>\n",
       "      <td>0.1774</td>\n",
       "      <td>0.2253</td>\n",
       "      <td>0.1582</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>4/6</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>PARK7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3633</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>rs2640906</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7969449</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1267406</td>\n",
       "      <td>ENST00000377491.5</td>\n",
       "      <td>c.252+45G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>126</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>96%</td>\n",
       "      <td>2.0568E-13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.3689</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>0.2549</td>\n",
       "      <td>0.3386</td>\n",
       "      <td>0.4742</td>\n",
       "      <td>0.08915</td>\n",
       "      <td>0.6243</td>\n",
       "      <td>0.2772</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.2089</td>\n",
       "      <td>0.1092</td>\n",
       "      <td>0.2623</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.06872</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.09267</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.1007</td>\n",
       "      <td>0.1774</td>\n",
       "      <td>0.2253</td>\n",
       "      <td>0.1582</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>4/6</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>PARK7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3634</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>rs2640906</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7969449</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1267406</td>\n",
       "      <td>ENST00000377493.9</td>\n",
       "      <td>c.193-1445G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>126</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>96%</td>\n",
       "      <td>2.0568E-13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.3689</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>0.2549</td>\n",
       "      <td>0.3386</td>\n",
       "      <td>0.4742</td>\n",
       "      <td>0.08915</td>\n",
       "      <td>0.6243</td>\n",
       "      <td>0.2772</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.2089</td>\n",
       "      <td>0.1092</td>\n",
       "      <td>0.2623</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.06872</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.09267</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.1007</td>\n",
       "      <td>0.1774</td>\n",
       "      <td>0.2253</td>\n",
       "      <td>0.1582</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>3/5</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>PARK7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3635</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>rs2640906</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7969449</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1267406</td>\n",
       "      <td>ENST00000460192.5</td>\n",
       "      <td>n.412+45G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>126</td>\n",
       "      <td>27</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>96%</td>\n",
       "      <td>2.0568E-13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.3689</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>0.2549</td>\n",
       "      <td>0.3386</td>\n",
       "      <td>0.4742</td>\n",
       "      <td>0.08915</td>\n",
       "      <td>0.6243</td>\n",
       "      <td>0.2772</td>\n",
       "      <td>0.1654</td>\n",
       "      <td>0.2089</td>\n",
       "      <td>0.1092</td>\n",
       "      <td>0.2623</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.06872</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.09267</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.1007</td>\n",
       "      <td>0.1774</td>\n",
       "      <td>0.2253</td>\n",
       "      <td>0.1582</td>\n",
       "      <td>0.6508</td>\n",
       "      <td>EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>PARK7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339719</th>\n",
       "      <td>ABCD1</td>\n",
       "      <td>rs11803</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153744612</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Adrenoleukodystrophy</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>368066</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>0</td>\n",
       "      <td>68</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6809E-40</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>22</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7311</td>\n",
       "      <td>0.8784</td>\n",
       "      <td>0.7309</td>\n",
       "      <td>0.6008</td>\n",
       "      <td>0.6149</td>\n",
       "      <td>0.7883</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6966</td>\n",
       "      <td>0.8398</td>\n",
       "      <td>0.5541</td>\n",
       "      <td>0.6935</td>\n",
       "      <td>0.5757</td>\n",
       "      <td>0.6118</td>\n",
       "      <td>0.6302</td>\n",
       "      <td>0.6667</td>\n",
       "      <td>0.6329</td>\n",
       "      <td>0.662</td>\n",
       "      <td>0.7649</td>\n",
       "      <td>0.8784</td>\n",
       "      <td>AFR</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>ABCD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339720</th>\n",
       "      <td>ABCD1</td>\n",
       "      <td>rs1055847</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153744629</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Adrenoleukodystrophy</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>368067</td>\n",
       "      <td>ENST00000218104.6</td>\n",
       "      <td>c.*894G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>122</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>26</td>\n",
       "      <td>32</td>\n",
       "      <td>55.17%</td>\n",
       "      <td>5.693E-13</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.1944</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>0.4921</td>\n",
       "      <td>0.5196</td>\n",
       "      <td>0.6253</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4496</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.5332</td>\n",
       "      <td>0.5703</td>\n",
       "      <td>0.5165</td>\n",
       "      <td>0.5175</td>\n",
       "      <td>0.5434</td>\n",
       "      <td>0.5805</td>\n",
       "      <td>0.5324</td>\n",
       "      <td>0.457</td>\n",
       "      <td>0.606</td>\n",
       "      <td>0.6253</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/10</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>ABCD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339721</th>\n",
       "      <td>ABCD1</td>\n",
       "      <td>rs1055847</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153744629</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Adrenoleukodystrophy</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>368067</td>\n",
       "      <td>ENST00000434284.1</td>\n",
       "      <td>n.72-6051C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>122</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>26</td>\n",
       "      <td>32</td>\n",
       "      <td>55.17%</td>\n",
       "      <td>5.693E-13</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.1944</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>0.4921</td>\n",
       "      <td>0.5196</td>\n",
       "      <td>0.6253</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4496</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.5332</td>\n",
       "      <td>0.5703</td>\n",
       "      <td>0.5165</td>\n",
       "      <td>0.5175</td>\n",
       "      <td>0.5434</td>\n",
       "      <td>0.5805</td>\n",
       "      <td>0.5324</td>\n",
       "      <td>0.457</td>\n",
       "      <td>0.606</td>\n",
       "      <td>0.6253</td>\n",
       "      <td>SAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>ABCD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339722</th>\n",
       "      <td>ABCD1</td>\n",
       "      <td>rs1055847</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153744629</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Adrenoleukodystrophy</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>368067</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>122</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>26</td>\n",
       "      <td>32</td>\n",
       "      <td>55.17%</td>\n",
       "      <td>5.693E-13</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.1944</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>0.4921</td>\n",
       "      <td>0.5196</td>\n",
       "      <td>0.6253</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4496</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.5332</td>\n",
       "      <td>0.5703</td>\n",
       "      <td>0.5165</td>\n",
       "      <td>0.5175</td>\n",
       "      <td>0.5434</td>\n",
       "      <td>0.5805</td>\n",
       "      <td>0.5324</td>\n",
       "      <td>0.457</td>\n",
       "      <td>0.606</td>\n",
       "      <td>0.6253</td>\n",
       "      <td>SAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>ABCD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339723</th>\n",
       "      <td>ABCD1</td>\n",
       "      <td>rs1055847</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153744629</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Adrenoleukodystrophy</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>368067</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>122</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>26</td>\n",
       "      <td>32</td>\n",
       "      <td>55.17%</td>\n",
       "      <td>5.693E-13</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.1944</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>0.4921</td>\n",
       "      <td>0.5196</td>\n",
       "      <td>0.6253</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.4496</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.5332</td>\n",
       "      <td>0.5703</td>\n",
       "      <td>0.5165</td>\n",
       "      <td>0.5175</td>\n",
       "      <td>0.5434</td>\n",
       "      <td>0.5805</td>\n",
       "      <td>0.5324</td>\n",
       "      <td>0.457</td>\n",
       "      <td>0.606</td>\n",
       "      <td>0.6253</td>\n",
       "      <td>SAS</td>\n",
       "      <td>CTCF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Yes</td>\n",
       "      <td>ABCD1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2716 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name       rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "3631       PARK7  rs2640906  chr1    7969449   G   A    Homozygous   \n",
       "3632       PARK7  rs2640906  chr1    7969449   G   A    Homozygous   \n",
       "3633       PARK7  rs2640906  chr1    7969449   G   A    Homozygous   \n",
       "3634       PARK7  rs2640906  chr1    7969449   G   A    Homozygous   \n",
       "3635       PARK7  rs2640906  chr1    7969449   G   A    Homozygous   \n",
       "...          ...        ...   ...        ...  ..  ..           ...   \n",
       "339719     ABCD1    rs11803  chrX  153744612   C   T    Homozygous   \n",
       "339720     ABCD1  rs1055847  chrX  153744629   G   A  Heterozygous   \n",
       "339721     ABCD1  rs1055847  chrX  153744629   G   A  Heterozygous   \n",
       "339722     ABCD1  rs1055847  chrX  153744629   G   A  Heterozygous   \n",
       "339723     ABCD1  rs1055847  chrX  153744629   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "3631                                  intron variant              2/10   \n",
       "3632                                  intron variant              2/10   \n",
       "3633                                  intron variant              2/10   \n",
       "3634                                  intron variant              2/10   \n",
       "3635    intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "339719                       TF binding site variant              2/10   \n",
       "339720                           3 prime UTR variant              3/10   \n",
       "339721  intron variant,non coding transcript variant              2/10   \n",
       "339722                       downstream gene variant              2/10   \n",
       "339723                     regulatory region variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score         ClinVar_CLNDN CLIN_SIG  \\\n",
       "3631    MODIFIER           1.5          not provided   benign   \n",
       "3632    MODIFIER           1.5          not provided   benign   \n",
       "3633    MODIFIER           1.5          not provided   benign   \n",
       "3634    MODIFIER           1.5          not provided   benign   \n",
       "3635    MODIFIER           1.5          not provided   benign   \n",
       "...          ...           ...                   ...      ...   \n",
       "339719  MODIFIER           1.5  Adrenoleukodystrophy   benign   \n",
       "339720  MODIFIER           1.5  Adrenoleukodystrophy   benign   \n",
       "339721  MODIFIER           1.5  Adrenoleukodystrophy   benign   \n",
       "339722  MODIFIER           1.5  Adrenoleukodystrophy   benign   \n",
       "339723  MODIFIER           1.5  Adrenoleukodystrophy   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT  ClinVar               HGVSc  \\\n",
       "3631    criteria provided, single submitter  1267406  ENST00000338639.10   \n",
       "3632    criteria provided, single submitter  1267406   ENST00000377488.5   \n",
       "3633    criteria provided, single submitter  1267406   ENST00000377491.5   \n",
       "3634    criteria provided, single submitter  1267406   ENST00000377493.9   \n",
       "3635    criteria provided, single submitter  1267406   ENST00000460192.5   \n",
       "...                                     ...      ...                 ...   \n",
       "339719  criteria provided, single submitter   368066                       \n",
       "339720  criteria provided, single submitter   368067   ENST00000218104.6   \n",
       "339721  criteria provided, single submitter   368067   ENST00000434284.1   \n",
       "339722  criteria provided, single submitter   368067                       \n",
       "339723  criteria provided, single submitter   368067                       \n",
       "\n",
       "       HGVSc (Transcript) HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD  \\\n",
       "3631          c.252+45G>A                     None  1/1  126  27  25   1  24   \n",
       "3632          c.252+45G>A                     None  1/1  126  27  25   1  24   \n",
       "3633          c.252+45G>A                     None  1/1  126  27  25   1  24   \n",
       "3634        c.193-1445G>A                     None  1/1  126  27  25   1  24   \n",
       "3635          n.412+45G>A                     None  1/1  126  27  25   1  24   \n",
       "...                   ...   ...                ...  ...  ...  ..  ..  ..  ..   \n",
       "339719               None                     None  1/1  255  68  68   0  68   \n",
       "339720          c.*894G>A                     None  0/1  122  58  58  26  32   \n",
       "339721       n.72-6051C>T                     None  0/1  122  58  58  26  32   \n",
       "339722               None                     None  0/1  122  58  58  26  32   \n",
       "339723               None                     None  0/1  122  58  58  26  32   \n",
       "\n",
       "          FREQ        PVAL RDF RDR ADF ADR SIFT PolyPhen      AF  AFR_AF  \\\n",
       "3631       96%  2.0568E-13   1   0  13  11                0.3323  0.3472   \n",
       "3632       96%  2.0568E-13   1   0  13  11                0.3323  0.3472   \n",
       "3633       96%  2.0568E-13   1   0  13  11                0.3323  0.3472   \n",
       "3634       96%  2.0568E-13   1   0  13  11                0.3323  0.3472   \n",
       "3635       96%  2.0568E-13   1   0  13  11                0.3323  0.3472   \n",
       "...        ...         ...  ..  ..  ..  ..  ...      ...     ...     ...   \n",
       "339719    100%  1.6809E-40   0   0  46  22                0.7311  0.8784   \n",
       "339720  55.17%   5.693E-13  19   7  20  12                0.4599  0.1944   \n",
       "339721  55.17%   5.693E-13  19   7  20  12                0.4599  0.1944   \n",
       "339722  55.17%   5.693E-13  19   7  20  12                0.4599  0.1944   \n",
       "339723  55.17%   5.693E-13  19   7  20  12                0.4599  0.1944   \n",
       "\n",
       "        AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "3631    0.3689  0.6508   0.163  0.1319     0.2549         0.3386   \n",
       "3632    0.3689  0.6508   0.163  0.1319     0.2549         0.3386   \n",
       "3633    0.3689  0.6508   0.163  0.1319     0.2549         0.3386   \n",
       "3634    0.3689  0.6508   0.163  0.1319     0.2549         0.3386   \n",
       "3635    0.3689  0.6508   0.163  0.1319     0.2549         0.3386   \n",
       "...        ...     ...     ...     ...        ...            ...   \n",
       "339719  0.7309  0.6008  0.6149  0.7883                             \n",
       "339720  0.6069  0.4921  0.5196  0.6253                             \n",
       "339721  0.6069  0.4921  0.5196  0.6253                             \n",
       "339722  0.6069  0.4921  0.5196  0.6253                             \n",
       "339723  0.6069  0.4921  0.5196  0.6253                             \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "3631           0.4742        0.08915         0.6243         0.2772   \n",
       "3632           0.4742        0.08915         0.6243         0.2772   \n",
       "3633           0.4742        0.08915         0.6243         0.2772   \n",
       "3634           0.4742        0.08915         0.6243         0.2772   \n",
       "3635           0.4742        0.08915         0.6243         0.2772   \n",
       "...               ...            ...            ...            ...   \n",
       "339719                                                               \n",
       "339720                                                               \n",
       "339721                                                               \n",
       "339722                                                               \n",
       "339723                                                               \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "3631           0.1654         0.2089         0.1092     0.2623         0.3352   \n",
       "3632           0.1654         0.2089         0.1092     0.2623         0.3352   \n",
       "3633           0.1654         0.2089         0.1092     0.2623         0.3352   \n",
       "3634           0.1654         0.2089         0.1092     0.2623         0.3352   \n",
       "3635           0.1654         0.2089         0.1092     0.2623         0.3352   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "339719                                                  0.6966         0.8398   \n",
       "339720                                                  0.4496         0.2174   \n",
       "339721                                                  0.4496         0.2174   \n",
       "339722                                                  0.4496         0.2174   \n",
       "339723                                                  0.4496         0.2174   \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "3631          0.06872          0.352        0.09267          0.645   \n",
       "3632          0.06872          0.352        0.09267          0.645   \n",
       "3633          0.06872          0.352        0.09267          0.645   \n",
       "3634          0.06872          0.352        0.09267          0.645   \n",
       "3635          0.06872          0.352        0.09267          0.645   \n",
       "...               ...            ...            ...            ...   \n",
       "339719         0.5541         0.6935         0.5757         0.6118   \n",
       "339720         0.5332         0.5703         0.5165         0.5175   \n",
       "339721         0.5332         0.5703         0.5165         0.5175   \n",
       "339722         0.5332         0.5703         0.5165         0.5175   \n",
       "339723         0.5332         0.5703         0.5165         0.5175   \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "3631           0.3337         0.1007         0.1774         0.2253   \n",
       "3632           0.3337         0.1007         0.1774         0.2253   \n",
       "3633           0.3337         0.1007         0.1774         0.2253   \n",
       "3634           0.3337         0.1007         0.1774         0.2253   \n",
       "3635           0.3337         0.1007         0.1774         0.2253   \n",
       "...               ...            ...            ...            ...   \n",
       "339719         0.6302         0.6667         0.6329          0.662   \n",
       "339720         0.5434         0.5805         0.5324          0.457   \n",
       "339721         0.5434         0.5805         0.5324          0.457   \n",
       "339722         0.5434         0.5805         0.5324          0.457   \n",
       "339723         0.5434         0.5805         0.5324          0.457   \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF MAX_AF_POPS                         BIOTYPE  \\\n",
       "3631           0.1582  0.6508         EAS                  protein coding   \n",
       "3632           0.1582  0.6508         EAS                  protein coding   \n",
       "3633           0.1582  0.6508         EAS                  protein coding   \n",
       "3634           0.1582  0.6508         EAS                  protein coding   \n",
       "3635           0.1582  0.6508         EAS                 retained intron   \n",
       "...               ...     ...         ...                             ...   \n",
       "339719         0.7649  0.8784         AFR                                   \n",
       "339720          0.606  0.6253         SAS                  protein coding   \n",
       "339721          0.606  0.6253         SAS                          lncRNA   \n",
       "339722          0.606  0.6253         SAS  protein coding CDS not defined   \n",
       "339723          0.606  0.6253         SAS               CTCF binding site   \n",
       "\n",
       "         EXON INTRON Protein Position and Amino Acid Codons STRAND PUBMED  \\\n",
       "3631             4/6                             nan             1          \n",
       "3632             4/6                             nan             1          \n",
       "3633             4/6                             nan             1          \n",
       "3634             3/5                             nan             1          \n",
       "3635             5/5                             nan             1          \n",
       "...       ...    ...                             ...    ...    ...    ...   \n",
       "339719                                           nan            -1          \n",
       "339720  10/10                                    nan             1          \n",
       "339721           1/2                             nan            -1          \n",
       "339722                                           nan             1          \n",
       "339723                                           nan                        \n",
       "\n",
       "       Gene Match Matched_Gene  \n",
       "3631          Yes        PARK7  \n",
       "3632          Yes        PARK7  \n",
       "3633          Yes        PARK7  \n",
       "3634          Yes        PARK7  \n",
       "3635          Yes        PARK7  \n",
       "...           ...          ...  \n",
       "339719        Yes        ABCD1  \n",
       "339720        Yes        ABCD1  \n",
       "339721        Yes        ABCD1  \n",
       "339722        Yes        ABCD1  \n",
       "339723        Yes        ABCD1  \n",
       "\n",
       "[2716 rows x 70 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2[merged_2['Gene Match'] == 'Yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "6b261d4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
       "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
       "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
       "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
       "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
       "       'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
       "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
       "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
       "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
       "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
       "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
       "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Gene Match', 'Matched_Gene'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "06ff5176",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "340652                     regulatory region variant              2/10   \n",
       "340653                              missense variant              7/10   \n",
       "340654                           3 prime UTR variant              3/10   \n",
       "340655                       downstream gene variant              2/10   \n",
       "340656                       downstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       MODERATE           5.0                          \n",
       "1            LOW           2.5                          \n",
       "2       MODIFIER           1.5                          \n",
       "3       MODIFIER           1.5                          \n",
       "4       MODIFIER           1.5                          \n",
       "...          ...           ...           ...      ...   \n",
       "340652  MODIFIER           1.5                          \n",
       "340653  MODERATE           5.0  not provided   benign   \n",
       "340654  MODIFIER           1.5  not provided   benign   \n",
       "340655  MODIFIER           1.5  not provided   benign   \n",
       "340656  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \n",
       "0                                T162A  Aca/Gca      1                      \n",
       "1                                 S290  tcT/tcC      1                      \n",
       "2                                  nan              -1                      \n",
       "3                                  nan              -1                      \n",
       "4                                  nan              -1                      \n",
       "...                                ...      ...    ...    ...          ...  \n",
       "340652                             nan                                      \n",
       "340653                           R365H  cGt/cAt      1                      \n",
       "340654                             nan               1                      \n",
       "340655                             nan              -1                      \n",
       "340656                             nan               1                      \n",
       "\n",
       "[340657 rows x 69 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2.drop(columns=['Gene Match'], axis=1)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "8c412818",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "340652                     regulatory region variant              2/10   \n",
       "340653                              missense variant              7/10   \n",
       "340654                           3 prime UTR variant              3/10   \n",
       "340655                       downstream gene variant              2/10   \n",
       "340656                       downstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       MODERATE           5.0                          \n",
       "1            LOW           2.5                          \n",
       "2       MODIFIER           1.5                          \n",
       "3       MODIFIER           1.5                          \n",
       "4       MODIFIER           1.5                          \n",
       "...          ...           ...           ...      ...   \n",
       "340652  MODIFIER           1.5                          \n",
       "340653  MODERATE           5.0  not provided   benign   \n",
       "340654  MODIFIER           1.5  not provided   benign   \n",
       "340655  MODIFIER           1.5  not provided   benign   \n",
       "340656  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \\\n",
       "0                                T162A  Aca/Gca      1                       \n",
       "1                                 S290  tcT/tcC      1                       \n",
       "2                                  nan              -1                       \n",
       "3                                  nan              -1                       \n",
       "4                                  nan              -1                       \n",
       "...                                ...      ...    ...    ...          ...   \n",
       "340652                             nan                                       \n",
       "340653                           R365H  cGt/cAt      1                       \n",
       "340654                             nan               1                       \n",
       "340655                             nan              -1                       \n",
       "340656                             nan               1                       \n",
       "\n",
       "       Gene Match  \n",
       "0              No  \n",
       "1              No  \n",
       "2              No  \n",
       "3              No  \n",
       "4              No  \n",
       "...           ...  \n",
       "340652         No  \n",
       "340653         No  \n",
       "340654         No  \n",
       "340655         No  \n",
       "340656         No  \n",
       "\n",
       "[340657 rows x 70 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_2, df_gene, on= 'Matched_Gene', how = 'left', sort = False)\n",
    "merged_2['Gene Match'] = merged_2['Gene Match'].fillna('No')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "5471f654",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No                  337941\n",
       "Cerebellarataxia      1873\n",
       "Parkinson              843\n",
       "Name: Gene Match, dtype: int64"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene Match'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "a377ed84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7962863</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7984930</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7984981</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7984954</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7970951</td>\n",
       "      <td>Parkinson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3146</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915692</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3147</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107916864</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3148</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915532</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3149</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917349</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3150</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917418</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3151 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS        Literature\n",
       "0     chr1    7962863         Parkinson\n",
       "1     chr1    7984930         Parkinson\n",
       "2     chr1    7984981         Parkinson\n",
       "3     chr1    7984954         Parkinson\n",
       "4     chr1    7970951         Parkinson\n",
       "...    ...        ...               ...\n",
       "3146  chr7  107915692  Cerebellarataxia\n",
       "3147  chr7  107916864  Cerebellarataxia\n",
       "3148  chr7  107915532  Cerebellarataxia\n",
       "3149  chr7  107917349  Cerebellarataxia\n",
       "3150  chr7  107917418  Cerebellarataxia\n",
       "\n",
       "[3151 rows x 3 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "e624e039",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "merged_2['CHROM'] data type: object\n",
      "merged_2['POS'] data type: int64\n",
      "df_3['CHROM'] data type: object\n",
      "df_3['POS'] data type: int64\n"
     ]
    }
   ],
   "source": [
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "\n",
    "print(\"merged_2['CHROM'] data type:\", merged_2['CHROM'].dtype)\n",
    "print(\"merged_2['POS'] data type:\", merged_2['POS'].dtype)\n",
    "print(\"df_3['CHROM'] data type:\", df_3['CHROM'].dtype)\n",
    "print(\"df_3['POS'] data type:\", df_3['POS'].dtype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "6d109149",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "340652                     regulatory region variant              2/10   \n",
       "340653                              missense variant              7/10   \n",
       "340654                           3 prime UTR variant              3/10   \n",
       "340655                       downstream gene variant              2/10   \n",
       "340656                       downstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       MODERATE           5.0                          \n",
       "1            LOW           2.5                          \n",
       "2       MODIFIER           1.5                          \n",
       "3       MODIFIER           1.5                          \n",
       "4       MODIFIER           1.5                          \n",
       "...          ...           ...           ...      ...   \n",
       "340652  MODIFIER           1.5                          \n",
       "340653  MODERATE           5.0  not provided   benign   \n",
       "340654  MODIFIER           1.5  not provided   benign   \n",
       "340655  MODIFIER           1.5  not provided   benign   \n",
       "340656  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \\\n",
       "0                                T162A  Aca/Gca      1                       \n",
       "1                                 S290  tcT/tcC      1                       \n",
       "2                                  nan              -1                       \n",
       "3                                  nan              -1                       \n",
       "4                                  nan              -1                       \n",
       "...                                ...      ...    ...    ...          ...   \n",
       "340652                             nan                                       \n",
       "340653                           R365H  cGt/cAt      1                       \n",
       "340654                             nan               1                       \n",
       "340655                             nan              -1                       \n",
       "340656                             nan               1                       \n",
       "\n",
       "       Gene Match Literature  \n",
       "0              No         No  \n",
       "1              No         No  \n",
       "2              No         No  \n",
       "3              No         No  \n",
       "4              No         No  \n",
       "...           ...        ...  \n",
       "340652         No         No  \n",
       "340653         No         No  \n",
       "340654         No         No  \n",
       "340655         No         No  \n",
       "340656         No         No  \n",
       "\n",
       "[340657 rows x 71 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "5ecae4ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No                  340591\n",
       "Parkinson               40\n",
       "Cerebellarataxia        26\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "4aad4c46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
       "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
       "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
       "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
       "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
       "       'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
       "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
       "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
       "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
       "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
       "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
       "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Matched_Gene', 'Gene Match', 'Literature'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "d9f4d941",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>No</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene Match         rsID Literature CHROM        POS REF ALT  \\\n",
       "0          OR4F5         No    rs2691305         No  chr1      69511   A   G   \n",
       "1          OR4F5         No  rs200676709         No  chr1      69897   T   C   \n",
       "2                        No            .         No  chr1     685694   T   C   \n",
       "3                        No            .         No  chr1     685694   T   C   \n",
       "4                        No            .         No  chr1     685694   T   C   \n",
       "...          ...        ...          ...        ...   ...        ...  ..  ..   \n",
       "340652      IL9R         No    rs2037999         No  chrX  156003433   T   C   \n",
       "340653      IL9R         No  rs147385831         No  chrX  156009937   G   A   \n",
       "340654      IL9R         No  rs147385831         No  chrX  156009937   G   A   \n",
       "340655      IL9R         No  rs147385831         No  chrX  156009937   G   A   \n",
       "340656      IL9R         No  rs147385831         No  chrX  156009937   G   A   \n",
       "\n",
       "            Zygosity                                   Consequence  \\\n",
       "0         Homozygous                              missense variant   \n",
       "1       Heterozygous                            synonymous variant   \n",
       "2       Heterozygous                       downstream gene variant   \n",
       "3       Heterozygous  intron variant,non coding transcript variant   \n",
       "4       Heterozygous  intron variant,non coding transcript variant   \n",
       "...              ...                                           ...   \n",
       "340652    Homozygous                     regulatory region variant   \n",
       "340653  Heterozygous                              missense variant   \n",
       "340654  Heterozygous                           3 prime UTR variant   \n",
       "340655  Heterozygous                       downstream gene variant   \n",
       "340656  Heterozygous                       downstream gene variant   \n",
       "\n",
       "       Consequence_score    IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                   7/10  MODERATE           5.0                          \n",
       "1                   3/10       LOW           2.5                          \n",
       "2                   2/10  MODIFIER           1.5                          \n",
       "3                   2/10  MODIFIER           1.5                          \n",
       "4                   2/10  MODIFIER           1.5                          \n",
       "...                  ...       ...           ...           ...      ...   \n",
       "340652              2/10  MODIFIER           1.5                          \n",
       "340653              7/10  MODERATE           5.0  not provided   benign   \n",
       "340654              3/10  MODIFIER           1.5  not provided   benign   \n",
       "340655              2/10  MODIFIER           1.5  not provided   benign   \n",
       "340656              2/10  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                T162A  Aca/Gca      1         \n",
       "1                                 S290  tcT/tcC      1         \n",
       "2                                  nan              -1         \n",
       "3                                  nan              -1         \n",
       "4                                  nan              -1         \n",
       "...                                ...      ...    ...    ...  \n",
       "340652                             nan                         \n",
       "340653                           R365H  cGt/cAt      1         \n",
       "340654                             nan               1         \n",
       "340655                             nan              -1         \n",
       "340656                             nan               1         \n",
       "\n",
       "[340657 rows x 70 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = merged_3[['Gene Name', 'Gene Match', 'rsID', 'Literature', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "a5393724",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHAIGHGPTTL207_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "248ca8f9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "bbd560e8",
   "metadata": {},
   "source": [
    "# Literature check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "cf02ec94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16</td>\n",
       "      <td>31779</td>\n",
       "      <td>rs141542623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12652713</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17751397</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17751406</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12652700</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165543</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248855917</td>\n",
       "      <td>rs13313088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165544</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248856006</td>\n",
       "      <td>rs13313009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165545</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248913954</td>\n",
       "      <td>rs12025760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165546</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248916897</td>\n",
       "      <td>rs74157349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165547</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248917420</td>\n",
       "      <td>rs77124333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>165548 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples    GENE  CHROM        POS           ID\n",
       "0       12652713     NaN  chr16      31779  rs141542623\n",
       "1       12652713   TUBB8  chr10      49286    rs6560829\n",
       "2       17751397   TUBB8  chr10      49286    rs6560829\n",
       "3       17751406   TUBB8  chr10      49286    rs6560829\n",
       "4       12652700   TUBB8  chr10      49286    rs6560829\n",
       "...          ...     ...    ...        ...          ...\n",
       "165543  17751406  ZNF692   chr1  248855917   rs13313088\n",
       "165544  17751406  ZNF692   chr1  248856006   rs13313009\n",
       "165545  17751397   PGBD2   chr1  248913954   rs12025760\n",
       "165546  17751397   PGBD2   chr1  248916897   rs74157349\n",
       "165547  17751397   PGBD2   chr1  248917420   rs77124333\n",
       "\n",
       "[165548 rows x 5 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_covered_list.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "2e18a36b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65811</td>\n",
       "      <td>65993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69461</td>\n",
       "      <td>69620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>785981</td>\n",
       "      <td>786159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786130</td>\n",
       "      <td>786446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230714</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190028</td>\n",
       "      <td>57190328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230715</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190299</td>\n",
       "      <td>57190439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230716</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190874</td>\n",
       "      <td>57191014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230717</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191846</td>\n",
       "      <td>57192058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230718</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194102</td>\n",
       "      <td>57194243</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230719 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos\n",
       "0            chr1      65489     65645\n",
       "1            chr1      65811     65993\n",
       "2            chr1      69461     69620\n",
       "3            chr1     785981    786159\n",
       "4            chr1     786130    786446\n",
       "...           ...        ...       ...\n",
       "230714       chrY   57190028  57190328\n",
       "230715       chrY   57190299  57190439\n",
       "230716       chrY   57190874  57191014\n",
       "230717       chrY   57191846  57192058\n",
       "230718       chrY   57194102  57194243\n",
       "\n",
       "[230719 rows x 3 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/Depth_files/srinivas_sir_covered.bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "54e54a8c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  Extended_Start_pos  \\\n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5               65489   \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...               69007   \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...              450710   \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...              685686   \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...              924401   \n",
       "...                                                   ...                 ...   \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25038781   \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25041746   \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25043888   \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...            25622413   \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...            25624397   \n",
       "\n",
       "        Extended_End_pos  \n",
       "0                  65649  \n",
       "1                  70037  \n",
       "2                 451706  \n",
       "3                 686682  \n",
       "4                 924977  \n",
       "...                  ...  \n",
       "208906          25038941  \n",
       "208907          25041906  \n",
       "208908          25044048  \n",
       "208909          25624093  \n",
       "208910          25624557  \n",
       "\n",
       "[208911 rows x 6 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "bf80ac6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  Extended_Start_pos  \\\n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5               65489   \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...               69007   \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...              450710   \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...              685686   \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...              924401   \n",
       "...                                                   ...                 ...   \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25038781   \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25041746   \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25043888   \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...            25622413   \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...            25624397   \n",
       "\n",
       "        Extended_End_pos gene_symbol  \n",
       "0                  65649       OR4F5  \n",
       "1                  70037       OR4F5  \n",
       "2                 451706      OR4F29  \n",
       "3                 686682      OR4F16  \n",
       "4                 924977      SAMD11  \n",
       "...                  ...         ...  \n",
       "208906          25038941       BPY2C  \n",
       "208907          25041906       BPY2C  \n",
       "208908          25044048       BPY2C  \n",
       "208909          25624093        CDY1  \n",
       "208910          25624557        CDY1  \n",
       "\n",
       "[208911 rows x 7 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "98b1b88f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Extended_Start_pos  Extended_End_pos  \\\n",
       "0            chr1               65489             65649   \n",
       "1            chr1               69007             70037   \n",
       "2            chr1              450710            451706   \n",
       "3            chr1              685686            686682   \n",
       "4            chr1              924401            924977   \n",
       "...           ...                 ...               ...   \n",
       "208906       chrY            25038781          25038941   \n",
       "208907       chrY            25041746          25041906   \n",
       "208908       chrY            25043888          25044048   \n",
       "208909       chrY            25622413          25624093   \n",
       "208910       chrY            25624397          25624557   \n",
       "\n",
       "                                                     INFO gene_symbol  \n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5       OR4F5  \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...       OR4F5  \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...      OR4F29  \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...      OR4F16  \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...      SAMD11  \n",
       "...                                                   ...         ...  \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...       BPY2C  \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...       BPY2C  \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...       BPY2C  \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...        CDY1  \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...        CDY1  \n",
       "\n",
       "[208911 rows x 5 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "42e46958",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16</td>\n",
       "      <td>31779</td>\n",
       "      <td>rs141542623</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12652713</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17751397</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17751406</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12652700</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165543</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248855917</td>\n",
       "      <td>rs13313088</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165544</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248856006</td>\n",
       "      <td>rs13313009</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165545</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248913954</td>\n",
       "      <td>rs12025760</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165546</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248916897</td>\n",
       "      <td>rs74157349</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165547</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248917420</td>\n",
       "      <td>rs77124333</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>165548 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples    GENE  CHROM        POS           ID Covered/Not_Covered\n",
       "0       12652713     NaN  chr16      31779  rs141542623             Covered\n",
       "1       12652713   TUBB8  chr10      49286    rs6560829             Covered\n",
       "2       17751397   TUBB8  chr10      49286    rs6560829             Covered\n",
       "3       17751406   TUBB8  chr10      49286    rs6560829             Covered\n",
       "4       12652700   TUBB8  chr10      49286    rs6560829             Covered\n",
       "...          ...     ...    ...        ...          ...                 ...\n",
       "165543  17751406  ZNF692   chr1  248855917   rs13313088             Covered\n",
       "165544  17751406  ZNF692   chr1  248856006   rs13313009             Covered\n",
       "165545  17751397   PGBD2   chr1  248913954   rs12025760             Covered\n",
       "165546  17751397   PGBD2   chr1  248916897   rs74157349             Covered\n",
       "165547  17751397   PGBD2   chr1  248917420   rs77124333             Covered\n",
       "\n",
       "[165548 rows x 6 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Start_pos']\n",
    "    end_pos = row['End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "data['Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "e55996ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Covered    165548\n",
       "Name: Covered/Not_Covered, dtype: int64"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['Covered/Not_Covered'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "642e4bfb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978691</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978908</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120997590</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120996531</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978928</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2206</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17394331</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2207</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17396955</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2208</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17395249</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2209</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17413396</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17408487</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1685 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS Literature Covered_status\n",
       "0     chr12  120978691        Yes        Covered\n",
       "1     chr12  120978908        Yes        Covered\n",
       "2     chr12  120997590        Yes        Covered\n",
       "3     chr12  120996531        Yes        Covered\n",
       "4     chr12  120978928        Yes        Covered\n",
       "...     ...        ...        ...            ...\n",
       "2206  chr11   17394331        Yes        Covered\n",
       "2207  chr11   17396955        Yes        Covered\n",
       "2208  chr11   17395249        Yes        Covered\n",
       "2209  chr11   17413396        Yes        Covered\n",
       "2210  chr11   17408487        Yes        Covered\n",
       "\n",
       "[1685 rows x 4 columns]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Mody_Lit_final_Positions_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered_status'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered_status'] == 'Covered']\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "da78fe95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16</td>\n",
       "      <td>31779</td>\n",
       "      <td>rs141542623</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12652713</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17751397</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17751406</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12652700</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165543</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248855917</td>\n",
       "      <td>rs13313088</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165544</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248856006</td>\n",
       "      <td>rs13313009</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165545</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248913954</td>\n",
       "      <td>rs12025760</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165546</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248916897</td>\n",
       "      <td>rs74157349</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165547</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248917420</td>\n",
       "      <td>rs77124333</td>\n",
       "      <td>Covered</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>165548 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples    GENE  CHROM        POS           ID Covered/Not_Covered  \\\n",
       "0       12652713     NaN  chr16      31779  rs141542623             Covered   \n",
       "1       12652713   TUBB8  chr10      49286    rs6560829             Covered   \n",
       "2       17751397   TUBB8  chr10      49286    rs6560829             Covered   \n",
       "3       17751406   TUBB8  chr10      49286    rs6560829             Covered   \n",
       "4       12652700   TUBB8  chr10      49286    rs6560829             Covered   \n",
       "...          ...     ...    ...        ...          ...                 ...   \n",
       "165543  17751406  ZNF692   chr1  248855917   rs13313088             Covered   \n",
       "165544  17751406  ZNF692   chr1  248856006   rs13313009             Covered   \n",
       "165545  17751397   PGBD2   chr1  248913954   rs12025760             Covered   \n",
       "165546  17751397   PGBD2   chr1  248916897   rs74157349             Covered   \n",
       "165547  17751397   PGBD2   chr1  248917420   rs77124333             Covered   \n",
       "\n",
       "       Literature Covered_status  \n",
       "0             NaN            NaN  \n",
       "1             NaN            NaN  \n",
       "2             NaN            NaN  \n",
       "3             NaN            NaN  \n",
       "4             NaN            NaN  \n",
       "...           ...            ...  \n",
       "165543        NaN            NaN  \n",
       "165544        NaN            NaN  \n",
       "165545        NaN            NaN  \n",
       "165546        NaN            NaN  \n",
       "165547        NaN            NaN  \n",
       "\n",
       "[165548 rows x 8 columns]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(data, df_3, on=['CHROM', 'POS'], how='left', sort=False)\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "43178bcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Yes    9\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "2303bfe9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28075</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr11</td>\n",
       "      <td>17394397</td>\n",
       "      <td>.</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128673</th>\n",
       "      <td>12652713</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128674</th>\n",
       "      <td>17751397</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120979061</td>\n",
       "      <td>rs1800574</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128675</th>\n",
       "      <td>17751406</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120979061</td>\n",
       "      <td>rs1800574</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128678</th>\n",
       "      <td>12652700</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120989017</td>\n",
       "      <td>rs1057520291</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128682</th>\n",
       "      <td>12652700</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120994314</td>\n",
       "      <td>rs56348580</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128685</th>\n",
       "      <td>17751406</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128686</th>\n",
       "      <td>17751397</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128687</th>\n",
       "      <td>12652713</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples   GENE  CHROM        POS            ID Covered/Not_Covered  \\\n",
       "28075   12652713    NaN  chr11   17394397             .             Covered   \n",
       "128673  12652713  HNF1A  chr12  120978847     rs1169288             Covered   \n",
       "128674  17751397  HNF1A  chr12  120979061     rs1800574             Covered   \n",
       "128675  17751406  HNF1A  chr12  120979061     rs1800574             Covered   \n",
       "128678  12652700  HNF1A  chr12  120989017  rs1057520291             Covered   \n",
       "128682  12652700  HNF1A  chr12  120994314    rs56348580             Covered   \n",
       "128685  17751406  HNF1A  chr12  120997672     rs2464195             Covered   \n",
       "128686  17751397  HNF1A  chr12  120997672     rs2464195             Covered   \n",
       "128687  12652713  HNF1A  chr12  120997672     rs2464195             Covered   \n",
       "\n",
       "       Literature Covered_status  \n",
       "28075         Yes        Covered  \n",
       "128673        Yes        Covered  \n",
       "128674        Yes        Covered  \n",
       "128675        Yes        Covered  \n",
       "128678        Yes        Covered  \n",
       "128682        Yes        Covered  \n",
       "128685        Yes        Covered  \n",
       "128686        Yes        Covered  \n",
       "128687        Yes        Covered  "
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x= merged_3[merged_3['Literature'] == 'Yes']\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "a17b358e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28075</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr11</td>\n",
       "      <td>17394397</td>\n",
       "      <td>.</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128673</th>\n",
       "      <td>12652713</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128674</th>\n",
       "      <td>17751397</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120979061</td>\n",
       "      <td>rs1800574</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128675</th>\n",
       "      <td>17751406</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120979061</td>\n",
       "      <td>rs1800574</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128678</th>\n",
       "      <td>12652700</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120989017</td>\n",
       "      <td>rs1057520291</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128682</th>\n",
       "      <td>12652700</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120994314</td>\n",
       "      <td>rs56348580</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128685</th>\n",
       "      <td>17751406</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128686</th>\n",
       "      <td>17751397</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128687</th>\n",
       "      <td>12652713</td>\n",
       "      <td>HNF1A</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>Covered</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples   GENE  CHROM        POS            ID Covered/Not_Covered  \\\n",
       "28075   12652713    NaN  chr11   17394397             .             Covered   \n",
       "128673  12652713  HNF1A  chr12  120978847     rs1169288             Covered   \n",
       "128674  17751397  HNF1A  chr12  120979061     rs1800574             Covered   \n",
       "128675  17751406  HNF1A  chr12  120979061     rs1800574             Covered   \n",
       "128678  12652700  HNF1A  chr12  120989017  rs1057520291             Covered   \n",
       "128682  12652700  HNF1A  chr12  120994314    rs56348580             Covered   \n",
       "128685  17751406  HNF1A  chr12  120997672     rs2464195             Covered   \n",
       "128686  17751397  HNF1A  chr12  120997672     rs2464195             Covered   \n",
       "128687  12652713  HNF1A  chr12  120997672     rs2464195             Covered   \n",
       "\n",
       "       Literature Covered_status  \n",
       "28075         Yes        Covered  \n",
       "128673        Yes        Covered  \n",
       "128674        Yes        Covered  \n",
       "128675        Yes        Covered  \n",
       "128678        Yes        Covered  \n",
       "128682        Yes        Covered  \n",
       "128685        Yes        Covered  \n",
       "128686        Yes        Covered  \n",
       "128687        Yes        Covered  "
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/lit_mody.xlsx', index=False)\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "c6fc34a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Covered        2353\n",
       "Not_Covered     798\n",
       "Name: Covered/Not_Covered, dtype: int64"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['Covered/Not_Covered'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "fb4abfe2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7962863</td>\n",
       "      <td>Parkinson</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7970951</td>\n",
       "      <td>Parkinson</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7985014</td>\n",
       "      <td>Parkinson</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7985041</td>\n",
       "      <td>Parkinson</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7965425</td>\n",
       "      <td>Parkinson</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3146</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915692</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3147</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107916864</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3148</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107915532</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3149</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917349</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3150</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107917418</td>\n",
       "      <td>Cerebellarataxia</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2353 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS        Literature Covered/Not_Covered\n",
       "0     chr1    7962863         Parkinson             Covered\n",
       "4     chr1    7970951         Parkinson             Covered\n",
       "5     chr1    7985014         Parkinson             Covered\n",
       "8     chr1    7985041         Parkinson             Covered\n",
       "9     chr1    7965425         Parkinson             Covered\n",
       "...    ...        ...               ...                 ...\n",
       "3146  chr7  107915692  Cerebellarataxia             Covered\n",
       "3147  chr7  107916864  Cerebellarataxia             Covered\n",
       "3148  chr7  107915532  Cerebellarataxia             Covered\n",
       "3149  chr7  107917349  Cerebellarataxia             Covered\n",
       "3150  chr7  107917418  Cerebellarataxia             Covered\n",
       "\n",
       "[2353 rows x 4 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[data['Covered/Not_Covered'] == 'Covered']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "0dccdddc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>0</td>\n",
       "      <td>120</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.1001E-71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>34</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>82</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>73</td>\n",
       "      <td>25</td>\n",
       "      <td>25.51%</td>\n",
       "      <td>5.1517E-9</td>\n",
       "      <td>41</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000419394.2</td>\n",
       "      <td>n.480+17991A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000440200.5</td>\n",
       "      <td>n.169+34338A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>3.6855E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>processed transcript</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340652</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4304</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.5274</td>\n",
       "      <td>0.3271</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.5578</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>0.809</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>0.4374</td>\n",
       "      <td>0.7099</td>\n",
       "      <td>0.5742</td>\n",
       "      <td>0.4209</td>\n",
       "      <td>0.3323</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>0.5465</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>gnomADe AFR</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340653</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.1094G&gt;A</td>\n",
       "      <td>ENSP00000244174.5</td>\n",
       "      <td>p.Arg365His</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>tolerated(0.56)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>R365H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340654</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.*85G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340655</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340656</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs147385831</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156009937</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>771064</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.01405</td>\n",
       "      <td>0.02867</td>\n",
       "      <td>0.007828</td>\n",
       "      <td>0.01496</td>\n",
       "      <td>0.01587</td>\n",
       "      <td>0.005438</td>\n",
       "      <td>0.008185</td>\n",
       "      <td>0.02348</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.02864</td>\n",
       "      <td>0.03691</td>\n",
       "      <td>0.04154</td>\n",
       "      <td>0.01196</td>\n",
       "      <td>0.03051</td>\n",
       "      <td>0.02488</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>0.03478</td>\n",
       "      <td>0.02331</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>0.04483</td>\n",
       "      <td>0.0722</td>\n",
       "      <td>gnomADg FIN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340657 rows × 72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1      69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1      69897   T   C  Heterozygous   \n",
       "2                           .  chr1     685694   T   C  Heterozygous   \n",
       "3                           .  chr1     685694   T   C  Heterozygous   \n",
       "4                           .  chr1     685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...        ...  ..  ..           ...   \n",
       "340652      IL9R    rs2037999  chrX  156003433   T   C    Homozygous   \n",
       "340653      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340654      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340655      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "340656      IL9R  rs147385831  chrX  156009937   G   A  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "340652                     regulatory region variant              2/10   \n",
       "340653                              missense variant              7/10   \n",
       "340654                           3 prime UTR variant              3/10   \n",
       "340655                       downstream gene variant              2/10   \n",
       "340656                       downstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       MODERATE           5.0                          \n",
       "1            LOW           2.5                          \n",
       "2       MODIFIER           1.5                          \n",
       "3       MODIFIER           1.5                          \n",
       "4       MODIFIER           1.5                          \n",
       "...          ...           ...           ...      ...   \n",
       "340652  MODIFIER           1.5                          \n",
       "340653  MODERATE           5.0  not provided   benign   \n",
       "340654  MODIFIER           1.5  not provided   benign   \n",
       "340655  MODIFIER           1.5  not provided   benign   \n",
       "340656  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                     ENST00000641515.2   \n",
       "1                                                     ENST00000641515.2   \n",
       "2                                                                         \n",
       "3                                                     ENST00000419394.2   \n",
       "4                                                     ENST00000440200.5   \n",
       "...                                     ...     ...                 ...   \n",
       "340652                                                                    \n",
       "340653  criteria provided, single submitter  771064  ENST00000244174.11   \n",
       "340654  criteria provided, single submitter  771064   ENST00000369423.7   \n",
       "340655  criteria provided, single submitter  771064                       \n",
       "340656  criteria provided, single submitter  771064                       \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "1                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   82   \n",
       "2                    None                                  None  0/1   34   \n",
       "3          n.480+17991A>G                                  None  0/1   34   \n",
       "4          n.169+34338A>G                                  None  0/1   34   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "340652               None                                  None  1/1   58   \n",
       "340653          c.1094G>A  ENSP00000244174.5        p.Arg365His  0/1   27   \n",
       "340654           c.*85G>A                                  None  0/1   27   \n",
       "340655               None                                  None  0/1   27   \n",
       "340656               None                                  None  0/1   27   \n",
       "\n",
       "        SDP   DP  RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       120  120   0  120    100%  1.1001E-71   0   0  86  34   \n",
       "1        98   98  73   25  25.51%   5.1517E-9  41  32  11  14   \n",
       "2        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "3        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "4        28   28  18   10  35.71%   3.6855E-4  17   1   9   1   \n",
       "...     ...  ...  ..  ...     ...         ...  ..  ..  ..  ..   \n",
       "340652   11   11   0   11    100%   1.4176E-6   0   0   7   4   \n",
       "340653   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340654   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340655   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "340656   20   20  12    8     40%    1.638E-3  12   0   8   0   \n",
       "\n",
       "                   SIFT   PolyPhen      AF AFR_AF  AMR_AF EAS_AF  EUR_AF  \\\n",
       "0       tolerated(0.92)  benign(0)                                         \n",
       "1                                   0.6881  0.407  0.6254  0.876  0.7942   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                 ...        ...     ...    ...     ...    ...     ...   \n",
       "340652                                                                     \n",
       "340653  tolerated(0.56)  benign(0)                                         \n",
       "340654                                                                     \n",
       "340655                                                                     \n",
       "340656                                                                     \n",
       "\n",
       "        SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF  \\\n",
       "0                  0.9497         0.6075         0.9514         0.9767   \n",
       "1       0.8098     0.7209          0.307         0.6749         0.8131   \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...        ...        ...            ...            ...            ...   \n",
       "340652              0.441         0.8217         0.3483         0.4304   \n",
       "340653            0.01405        0.02867       0.007828        0.01496   \n",
       "340654            0.01405        0.02867       0.007828        0.01496   \n",
       "340655            0.01405        0.02867       0.007828        0.01496   \n",
       "340656            0.01405        0.02867       0.007828        0.01496   \n",
       "\n",
       "       gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0              0.9994         0.9916         0.9726         0.9506   \n",
       "1              0.7793         0.8652         0.8438         0.7932   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652          0.702         0.5274         0.3271         0.4035   \n",
       "340653        0.01587       0.005438       0.008185        0.02348   \n",
       "340654        0.01587       0.005438       0.008185        0.02348   \n",
       "340655        0.01587       0.005438       0.008185        0.02348   \n",
       "340656        0.01587       0.005438       0.008185        0.02348   \n",
       "\n",
       "       gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0              0.9854      0.846         0.5948          0.998         0.8951   \n",
       "1              0.8049     0.4864         0.2916         0.6511         0.4782   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...        ...            ...            ...            ...   \n",
       "340652         0.5578     0.5073          0.809          0.356         0.3772   \n",
       "340653         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340654         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340655         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "340656         0.0411    0.02864        0.03691        0.04154        0.01196   \n",
       "\n",
       "       gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0              0.9784         0.9998         0.9907            0.9   \n",
       "1              0.6714         0.6047         0.6987         0.6441   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "340652         0.4374         0.7099         0.5742         0.4209   \n",
       "340653        0.03051        0.02488         0.0722        0.03478   \n",
       "340654        0.03051        0.02488         0.0722        0.03478   \n",
       "340655        0.03051        0.02488         0.0722        0.03478   \n",
       "340656        0.03051        0.02488         0.0722        0.03478   \n",
       "\n",
       "       gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS  \\\n",
       "0              0.9674         0.8624         0.9772  0.9998  gnomADg EAS   \n",
       "1              0.6751         0.4659          0.622   0.876          EAS   \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...               ...            ...            ...     ...          ...   \n",
       "340652         0.3323         0.4516         0.5465  0.8217  gnomADe AFR   \n",
       "340653        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340654        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340655        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "340656        0.02331        0.02312        0.04483  0.0722  gnomADg FIN   \n",
       "\n",
       "                               BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                       protein coding  3/3          \n",
       "2                       protein coding               \n",
       "3                 processed transcript         3/3   \n",
       "4                 processed transcript         1/2   \n",
       "...                                ...  ...    ...   \n",
       "340652                        enhancer               \n",
       "340653                  protein coding  9/9          \n",
       "340654                  protein coding  9/9          \n",
       "340655                          lncRNA               \n",
       "340656  protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \\\n",
       "0                                T162A  Aca/Gca      1                       \n",
       "1                                 S290  tcT/tcC      1                       \n",
       "2                                  nan              -1                       \n",
       "3                                  nan              -1                       \n",
       "4                                  nan              -1                       \n",
       "...                                ...      ...    ...    ...          ...   \n",
       "340652                             nan                                       \n",
       "340653                           R365H  cGt/cAt      1                       \n",
       "340654                             nan               1                       \n",
       "340655                             nan              -1                       \n",
       "340656                             nan               1                       \n",
       "\n",
       "       Gene Match Literature Covered/Not_Covered  \n",
       "0              No         No                 NaN  \n",
       "1              No         No                 NaN  \n",
       "2              No         No                 NaN  \n",
       "3              No         No                 NaN  \n",
       "4              No         No                 NaN  \n",
       "...           ...        ...                 ...  \n",
       "340652         No         No                 NaN  \n",
       "340653         No         No                 NaN  \n",
       "340654         No         No                 NaN  \n",
       "340655         No         No                 NaN  \n",
       "340656         No         No                 NaN  \n",
       "\n",
       "[340657 rows x 72 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(merged_2, data, on=['CHROM', 'POS'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "31fa2885",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Covered    66\n",
       "Name: Covered/Not_Covered, dtype: int64"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3['Covered/Not_Covered'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81c4463d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
