{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51a2e8e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ac1cfb89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>326917</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24301931</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DHRS1,CIDEB,LTB4R2,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326918</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24302473</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>DHRS1,CIDEB,LTB4R2,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326919</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24302637</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DHRS1,CIDEB,LTB4R2,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326920</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24302733</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DHRS1,CIDEB,LTB4R2,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326921</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24306640</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>LTB4R2,CIDEB,LTB4R,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326922</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24308205</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LTB4R2,CIDEB,LTB4R,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326923</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24308220</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>LTB4R2,CIDEB,LTB4R,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326924</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24312106</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>LTB4R2,CIDEB,LTB4R,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326925</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24312124</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LTB4R2,CIDEB,LTB4R,NOP9</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326926</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24314911</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>LTB4R,CIDEB,ADCY4,LTB4R2</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM       POS rsID REF ALT                 Gene Name      Zygosity  \\\n",
       "326917  chr14  24301931    .   G   A   DHRS1,CIDEB,LTB4R2,NOP9  Heterozygous   \n",
       "326918  chr14  24302473    .   T   A   DHRS1,CIDEB,LTB4R2,NOP9  Heterozygous   \n",
       "326919  chr14  24302637    .   A   G   DHRS1,CIDEB,LTB4R2,NOP9  Heterozygous   \n",
       "326920  chr14  24302733    .   T   C   DHRS1,CIDEB,LTB4R2,NOP9  Heterozygous   \n",
       "326921  chr14  24306640    .   C   T   LTB4R2,CIDEB,LTB4R,NOP9  Heterozygous   \n",
       "326922  chr14  24308205    .   A   C   LTB4R2,CIDEB,LTB4R,NOP9  Heterozygous   \n",
       "326923  chr14  24308220    .   T   A   LTB4R2,CIDEB,LTB4R,NOP9  Heterozygous   \n",
       "326924  chr14  24312106    .   C   T   LTB4R2,CIDEB,LTB4R,NOP9  Heterozygous   \n",
       "326925  chr14  24312124    .   G   A   LTB4R2,CIDEB,LTB4R,NOP9  Heterozygous   \n",
       "326926  chr14  24314911    .   C   T  LTB4R,CIDEB,ADCY4,LTB4R2  Heterozygous   \n",
       "\n",
       "        DP  \n",
       "326917   2  \n",
       "326918   1  \n",
       "326919   2  \n",
       "326920   1  \n",
       "326921   3  \n",
       "326922   5  \n",
       "326923   4  \n",
       "326924   2  \n",
       "326925   2  \n",
       "326926   1  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "################################# Import the required Libraries ############################################################\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "\n",
    "################################### Import the VCF data and assign the column names ########################################\n",
    "\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/KHHSPTGPONC63_Mutect2_noindels.vcf', comment= '#', sep = '\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "################################### Extract the Zygosity of the variants from the Sample column of the vcf file ############\n",
    "\n",
    "vcf['Zygosity'] = vcf['SAMPLE'].str.split(':').str[0]\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].astype('int')\n",
    "#vcf = vcf[vcf['DP'] >= 5]\n",
    "\n",
    "# Iterate through each row in the DataFrame\n",
    "for index, row in vcf.iterrows():\n",
    "    # Check if the value is '1/1', if yes, set it to 'Homozygous', else set it to 'Heterozygous'\n",
    "    vcf.at[index, 'Zygosity'] = 'Homozygous' if row['Zygosity'] == '1/1' else 'Heterozygous'\n",
    "    \n",
    "###################### Extract the Gene Name from the INFO column (CSQ) ###################################################\n",
    "\n",
    "# Define a regular expression to extract the gene name from the CSQ field\n",
    "gene_regex = r'([\\w.-]+)\\|(ENSG\\d+|ENSG-\\d+[A-Z]?\\.\\d+)\\|'\n",
    "\n",
    "# Define a function to extract the gene name(s) from the INFO column\n",
    "def extract_gene_name(info):\n",
    "    csq_field = [field for field in info.split(';') if field.startswith('CSQ=')]\n",
    "    if csq_field:\n",
    "        csq_field = csq_field[0]\n",
    "        matches = re.findall(gene_regex, csq_field)\n",
    "        if len(set(matches)) > 1:\n",
    "            return ','.join(set([match[0] for match in matches]))\n",
    "        elif matches:\n",
    "            return matches[0][0]\n",
    "    return 'NA'\n",
    "\n",
    "vcf['Gene Name'] = vcf['INFO'].apply(extract_gene_name)\n",
    "\n",
    "################################## Create the DateFrame with KDR gene in the Gene Name column #############################\n",
    "\n",
    "data = {'Gene Name': ['CIDEB']}\n",
    "df_gene = pd.DataFrame(data)\n",
    "\n",
    "################################# Map the KDR gene with the extracted gene column in the DataFrame ########################\n",
    "\n",
    "vcf['Gene_Match'] = 'No'\n",
    "# Iterate through each gene in vcf\n",
    "for genes in vcf['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list) # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            vcf.loc[vcf['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "vcf = vcf[vcf['Gene_Match'] == 'Yes']\n",
    "\n",
    "###################### Finally pool the required columns from the vcf file and explort the dataframe as excel file #######\n",
    "\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'Gene Name', 'Zygosity', 'DP']]\n",
    "#vcf.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/KHAIGHGP0NC8_Haplotype_noindels_updated.xlsx', index=False)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "77e689e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(10, 8)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b042327f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT</th>\n",
       "      <th>HOM</th>\n",
       "      <th>HET</th>\n",
       "      <th>COUNT</th>\n",
       "      <th>HOM WILD</th>\n",
       "      <th>Ref_Allele_count</th>\n",
       "      <th>Alt_Allele_count</th>\n",
       "      <th>Totaol_Allele_count</th>\n",
       "      <th>Ref_Allele_Freq</th>\n",
       "      <th>KH_MAF</th>\n",
       "      <th>Prevalence</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Ncid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55076834_T_C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55076834</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000004.12:g.55076834T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55077777_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55077777</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077777T&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4_55077783_G_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55077783</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55077783G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4_55077786_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55077786</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077786T&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4_55077790_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55077790</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077790T&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>chr4_55127139_G_A</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>91</td>\n",
       "      <td>182</td>\n",
       "      <td>46</td>\n",
       "      <td>228</td>\n",
       "      <td>0.798246</td>\n",
       "      <td>0.201754</td>\n",
       "      <td>20.175439</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55127139</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127139G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>chr4_55127146_T_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55127146</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127146T&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>chr4_55127162_CC_GG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55127162</td>\n",
       "      <td>CC</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC_000004.12:g.55127162CC&gt;GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>chr4_55127210_T_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55127210</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127210T&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>chr4_55128099_A_T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.877193</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55128099</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.55128099A&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>287 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 VARIANT   HOM  HET  COUNT  HOM WILD  Ref_Allele_count  \\\n",
       "0      chr4_55076834_T_C   1.0    0      1       113               226   \n",
       "1      chr4_55077777_T_G   1.0    0      1       113               226   \n",
       "2      chr4_55077783_G_A   1.0    0      1       113               226   \n",
       "3      chr4_55077786_T_G   1.0    0      1       113               226   \n",
       "4      chr4_55077790_T_G   1.0    0      1       113               226   \n",
       "..                   ...   ...  ...    ...       ...               ...   \n",
       "282    chr4_55127139_G_A  23.0    0     23        91               182   \n",
       "283    chr4_55127146_T_A   1.0    0      1       113               226   \n",
       "284  chr4_55127162_CC_GG   1.0    0      1       113               226   \n",
       "285    chr4_55127210_T_A   1.0    0      1       113               226   \n",
       "286    chr4_55128099_A_T   1.0    0      1       113               226   \n",
       "\n",
       "     Alt_Allele_count  Totaol_Allele_count  Ref_Allele_Freq    KH_MAF  \\\n",
       "0                   2                  228         0.991228  0.008772   \n",
       "1                   2                  228         0.991228  0.008772   \n",
       "2                   2                  228         0.991228  0.008772   \n",
       "3                   2                  228         0.991228  0.008772   \n",
       "4                   2                  228         0.991228  0.008772   \n",
       "..                ...                  ...              ...       ...   \n",
       "282                46                  228         0.798246  0.201754   \n",
       "283                 2                  228         0.991228  0.008772   \n",
       "284                 2                  228         0.991228  0.008772   \n",
       "285                 2                  228         0.991228  0.008772   \n",
       "286                 2                  228         0.991228  0.008772   \n",
       "\n",
       "     Prevalence CHROM       POS REF ALT                          Ncid  \n",
       "0      0.877193  chr4  55076834   T   C    NC_000004.12:g.55076834T>C  \n",
       "1      0.877193  chr4  55077777   T   G    NC_000004.12:g.55077777T>G  \n",
       "2      0.877193  chr4  55077783   G   A    NC_000004.12:g.55077783G>A  \n",
       "3      0.877193  chr4  55077786   T   G    NC_000004.12:g.55077786T>G  \n",
       "4      0.877193  chr4  55077790   T   G    NC_000004.12:g.55077790T>G  \n",
       "..          ...   ...       ...  ..  ..                           ...  \n",
       "282   20.175439  chr4  55127139   G   A    NC_000004.12:g.55127139G>A  \n",
       "283    0.877193  chr4  55127146   T   A    NC_000004.12:g.55127146T>A  \n",
       "284    0.877193  chr4  55127162  CC  GG  NC_000004.12:g.55127162CC>GG  \n",
       "285    0.877193  chr4  55127210   T   A    NC_000004.12:g.55127210T>A  \n",
       "286    0.877193  chr4  55128099   A   T    NC_000004.12:g.55128099A>T  \n",
       "\n",
       "[287 rows x 16 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/MAF_KDR Known variants_287.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "35f0bf0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55076834</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55077777</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55077783</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55077786</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55077790</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55127121</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55127139</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55127146</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55127210</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55128099</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>263 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS REF ALT                                   Consequence  \\\n",
       "0    chr4  55076834   T   C  intron_variant&non_coding_transcript_variant   \n",
       "1    chr4  55077777   T   G  intron_variant&non_coding_transcript_variant   \n",
       "2    chr4  55077783   G   A  intron_variant&non_coding_transcript_variant   \n",
       "3    chr4  55077786   T   G  intron_variant&non_coding_transcript_variant   \n",
       "4    chr4  55077790   T   G  intron_variant&non_coding_transcript_variant   \n",
       "..    ...       ...  ..  ..                                           ...   \n",
       "258  chr4  55127121   G   T                            intergenic_variant   \n",
       "259  chr4  55127139   G   A                            intergenic_variant   \n",
       "260  chr4  55127146   T   A                            intergenic_variant   \n",
       "261  chr4  55127210   T   A                            intergenic_variant   \n",
       "262  chr4  55128099   A   T                            intergenic_variant   \n",
       "\n",
       "       IMPACT EXON INTRON PolyPhen SIFT  \n",
       "0    MODIFIER         3/3                \n",
       "1    MODIFIER         3/3                \n",
       "2    MODIFIER         3/3                \n",
       "3    MODIFIER         3/3                \n",
       "4    MODIFIER         3/3                \n",
       "..        ...  ...    ...      ...  ...  \n",
       "258  MODIFIER                            \n",
       "259  MODIFIER                            \n",
       "260  MODIFIER                            \n",
       "261  MODIFIER                            \n",
       "262  MODIFIER                            \n",
       "\n",
       "[263 rows x 10 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/263_KDR.vcf', comment= '#', sep = '\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO']\n",
    "vcf['CHROM'] = 'chr' + vcf['CHROM'].astype(str)\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['Consequence'] = vcf['CSQ'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['CSQ'].str.split('|').str[2]\n",
    "vcf['EXON'] = vcf['CSQ'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['CSQ'].str.split('|').str[9]\n",
    "vcf['PolyPhen'] = vcf['CSQ'].str.split('|').str[30]\n",
    "vcf['SIFT'] = vcf['CSQ'].str.split('|').str[29]\n",
    "vcf = vcf[['CHROM', 'POS', 'REF', 'ALT', 'Consequence', 'IMPACT', 'EXON', 'INTRON', 'PolyPhen', 'SIFT']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "7d04b5de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT</th>\n",
       "      <th>HOM</th>\n",
       "      <th>HET</th>\n",
       "      <th>COUNT</th>\n",
       "      <th>HOM WILD</th>\n",
       "      <th>Ref_Allele_count</th>\n",
       "      <th>Alt_Allele_count</th>\n",
       "      <th>Totaol_Allele_count</th>\n",
       "      <th>Ref_Allele_Freq</th>\n",
       "      <th>KH_MAF</th>\n",
       "      <th>...</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Ncid</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55076834_T_C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55076834</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000004.12:g.55076834T&gt;C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55077777_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077777</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077777T&gt;G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4_55077783_G_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077783</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55077783G&gt;A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4_55077786_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077786</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077786T&gt;G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4_55077790_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077790</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077790T&gt;G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>chr4_55127139_G_A</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>91</td>\n",
       "      <td>182</td>\n",
       "      <td>46</td>\n",
       "      <td>228</td>\n",
       "      <td>0.798246</td>\n",
       "      <td>0.201754</td>\n",
       "      <td>...</td>\n",
       "      <td>55127139</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127139G&gt;A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>chr4_55127146_T_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55127146</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127146T&gt;A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>chr4_55127162_CC_GG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55127162</td>\n",
       "      <td>CC</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC_000004.12:g.55127162CC&gt;GG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>chr4_55127210_T_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55127210</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127210T&gt;A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>chr4_55128099_A_T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55128099</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.55128099A&gt;T</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>287 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 VARIANT   HOM  HET  COUNT  HOM WILD  Ref_Allele_count  \\\n",
       "0      chr4_55076834_T_C   1.0    0      1       113               226   \n",
       "1      chr4_55077777_T_G   1.0    0      1       113               226   \n",
       "2      chr4_55077783_G_A   1.0    0      1       113               226   \n",
       "3      chr4_55077786_T_G   1.0    0      1       113               226   \n",
       "4      chr4_55077790_T_G   1.0    0      1       113               226   \n",
       "..                   ...   ...  ...    ...       ...               ...   \n",
       "282    chr4_55127139_G_A  23.0    0     23        91               182   \n",
       "283    chr4_55127146_T_A   1.0    0      1       113               226   \n",
       "284  chr4_55127162_CC_GG   1.0    0      1       113               226   \n",
       "285    chr4_55127210_T_A   1.0    0      1       113               226   \n",
       "286    chr4_55128099_A_T   1.0    0      1       113               226   \n",
       "\n",
       "     Alt_Allele_count  Totaol_Allele_count  Ref_Allele_Freq    KH_MAF  ...  \\\n",
       "0                   2                  228         0.991228  0.008772  ...   \n",
       "1                   2                  228         0.991228  0.008772  ...   \n",
       "2                   2                  228         0.991228  0.008772  ...   \n",
       "3                   2                  228         0.991228  0.008772  ...   \n",
       "4                   2                  228         0.991228  0.008772  ...   \n",
       "..                ...                  ...              ...       ...  ...   \n",
       "282                46                  228         0.798246  0.201754  ...   \n",
       "283                 2                  228         0.991228  0.008772  ...   \n",
       "284                 2                  228         0.991228  0.008772  ...   \n",
       "285                 2                  228         0.991228  0.008772  ...   \n",
       "286                 2                  228         0.991228  0.008772  ...   \n",
       "\n",
       "          POS REF  ALT                          Ncid  \\\n",
       "0    55076834   T    C    NC_000004.12:g.55076834T>C   \n",
       "1    55077777   T    G    NC_000004.12:g.55077777T>G   \n",
       "2    55077783   G    A    NC_000004.12:g.55077783G>A   \n",
       "3    55077786   T    G    NC_000004.12:g.55077786T>G   \n",
       "4    55077790   T    G    NC_000004.12:g.55077790T>G   \n",
       "..        ...  ..  ...                           ...   \n",
       "282  55127139   G    A    NC_000004.12:g.55127139G>A   \n",
       "283  55127146   T    A    NC_000004.12:g.55127146T>A   \n",
       "284  55127162  CC   GG  NC_000004.12:g.55127162CC>GG   \n",
       "285  55127210   T    A    NC_000004.12:g.55127210T>A   \n",
       "286  55128099   A    T    NC_000004.12:g.55128099A>T   \n",
       "\n",
       "                                      Consequence    IMPACT EXON INTRON  \\\n",
       "0    intron_variant&non_coding_transcript_variant  MODIFIER         3/3   \n",
       "1    intron_variant&non_coding_transcript_variant  MODIFIER         3/3   \n",
       "2    intron_variant&non_coding_transcript_variant  MODIFIER         3/3   \n",
       "3    intron_variant&non_coding_transcript_variant  MODIFIER         3/3   \n",
       "4    intron_variant&non_coding_transcript_variant  MODIFIER         3/3   \n",
       "..                                            ...       ...  ...    ...   \n",
       "282                            intergenic_variant  MODIFIER               \n",
       "283                            intergenic_variant  MODIFIER               \n",
       "284                                           NaN       NaN  NaN    NaN   \n",
       "285                            intergenic_variant  MODIFIER               \n",
       "286                            intergenic_variant  MODIFIER               \n",
       "\n",
       "    PolyPhen SIFT  \n",
       "0                  \n",
       "1                  \n",
       "2                  \n",
       "3                  \n",
       "4                  \n",
       "..       ...  ...  \n",
       "282                \n",
       "283                \n",
       "284      NaN  NaN  \n",
       "285                \n",
       "286                \n",
       "\n",
       "[287 rows x 22 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.merge(df, vcf, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort=False)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4b0c5d75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT</th>\n",
       "      <th>HOM</th>\n",
       "      <th>HET</th>\n",
       "      <th>COUNT</th>\n",
       "      <th>HOM WILD</th>\n",
       "      <th>Ref_Allele_count</th>\n",
       "      <th>Alt_Allele_count</th>\n",
       "      <th>Totaol_Allele_count</th>\n",
       "      <th>Ref_Allele_Freq</th>\n",
       "      <th>KH_MAF</th>\n",
       "      <th>...</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Ncid</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55076834_T_C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55076834</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000004.12:g.55076834T&gt;C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55077777_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077777</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077777T&gt;G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4_55077783_G_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077783</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55077783G&gt;A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4_55077786_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077786</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077786T&gt;G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4_55077790_T_G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55077790</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000004.12:g.55077790T&gt;G</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>chr4_55127139_G_A</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>91</td>\n",
       "      <td>182</td>\n",
       "      <td>46</td>\n",
       "      <td>228</td>\n",
       "      <td>0.798246</td>\n",
       "      <td>0.201754</td>\n",
       "      <td>...</td>\n",
       "      <td>55127139</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127139G&gt;A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>chr4_55127146_T_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55127146</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127146T&gt;A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>chr4_55127162_CC_GG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55127162</td>\n",
       "      <td>CC</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC_000004.12:g.55127162CC&gt;GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>chr4_55127210_T_A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55127210</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.55127210T&gt;A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>chr4_55128099_A_T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>113</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0.991228</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>...</td>\n",
       "      <td>55128099</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.55128099A&gt;T</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>287 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 VARIANT   HOM  HET  COUNT  HOM WILD  Ref_Allele_count  \\\n",
       "0      chr4_55076834_T_C   1.0    0      1       113               226   \n",
       "1      chr4_55077777_T_G   1.0    0      1       113               226   \n",
       "2      chr4_55077783_G_A   1.0    0      1       113               226   \n",
       "3      chr4_55077786_T_G   1.0    0      1       113               226   \n",
       "4      chr4_55077790_T_G   1.0    0      1       113               226   \n",
       "..                   ...   ...  ...    ...       ...               ...   \n",
       "282    chr4_55127139_G_A  23.0    0     23        91               182   \n",
       "283    chr4_55127146_T_A   1.0    0      1       113               226   \n",
       "284  chr4_55127162_CC_GG   1.0    0      1       113               226   \n",
       "285    chr4_55127210_T_A   1.0    0      1       113               226   \n",
       "286    chr4_55128099_A_T   1.0    0      1       113               226   \n",
       "\n",
       "     Alt_Allele_count  Totaol_Allele_count  Ref_Allele_Freq    KH_MAF  ...  \\\n",
       "0                   2                  228         0.991228  0.008772  ...   \n",
       "1                   2                  228         0.991228  0.008772  ...   \n",
       "2                   2                  228         0.991228  0.008772  ...   \n",
       "3                   2                  228         0.991228  0.008772  ...   \n",
       "4                   2                  228         0.991228  0.008772  ...   \n",
       "..                ...                  ...              ...       ...  ...   \n",
       "282                46                  228         0.798246  0.201754  ...   \n",
       "283                 2                  228         0.991228  0.008772  ...   \n",
       "284                 2                  228         0.991228  0.008772  ...   \n",
       "285                 2                  228         0.991228  0.008772  ...   \n",
       "286                 2                  228         0.991228  0.008772  ...   \n",
       "\n",
       "          POS REF  ALT                          Ncid  \\\n",
       "0    55076834   T    C    NC_000004.12:g.55076834T>C   \n",
       "1    55077777   T    G    NC_000004.12:g.55077777T>G   \n",
       "2    55077783   G    A    NC_000004.12:g.55077783G>A   \n",
       "3    55077786   T    G    NC_000004.12:g.55077786T>G   \n",
       "4    55077790   T    G    NC_000004.12:g.55077790T>G   \n",
       "..        ...  ..  ...                           ...   \n",
       "282  55127139   G    A    NC_000004.12:g.55127139G>A   \n",
       "283  55127146   T    A    NC_000004.12:g.55127146T>A   \n",
       "284  55127162  CC   GG  NC_000004.12:g.55127162CC>GG   \n",
       "285  55127210   T    A    NC_000004.12:g.55127210T>A   \n",
       "286  55128099   A    T    NC_000004.12:g.55128099A>T   \n",
       "\n",
       "                                      Consequence    IMPACT EXON INTRON  \\\n",
       "0    intron_variant&non_coding_transcript_variant  MODIFIER    -    3/3   \n",
       "1    intron_variant&non_coding_transcript_variant  MODIFIER    -    3/3   \n",
       "2    intron_variant&non_coding_transcript_variant  MODIFIER    -    3/3   \n",
       "3    intron_variant&non_coding_transcript_variant  MODIFIER    -    3/3   \n",
       "4    intron_variant&non_coding_transcript_variant  MODIFIER    -    3/3   \n",
       "..                                            ...       ...  ...    ...   \n",
       "282                            intergenic_variant  MODIFIER    -      -   \n",
       "283                            intergenic_variant  MODIFIER    -      -   \n",
       "284                                             -         -    -      -   \n",
       "285                            intergenic_variant  MODIFIER    -      -   \n",
       "286                            intergenic_variant  MODIFIER    -      -   \n",
       "\n",
       "    PolyPhen SIFT  \n",
       "0          -    -  \n",
       "1          -    -  \n",
       "2          -    -  \n",
       "3          -    -  \n",
       "4          -    -  \n",
       "..       ...  ...  \n",
       "282        -    -  \n",
       "283        -    -  \n",
       "284        -    -  \n",
       "285        -    -  \n",
       "286        -    -  \n",
       "\n",
       "[287 rows x 22 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.fillna('-', inplace=True)\n",
    "merged.replace('', '-', inplace=True)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d9d2f5e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/287_rsID_CON_IMP.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca9a760b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b67fd57b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "f3a101e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>DP</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55079914</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>53</td>\n",
       "      <td>KHAIGHGP0NC8_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55081941</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>53</td>\n",
       "      <td>KHAIGHGP0NC8_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55089065</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>24</td>\n",
       "      <td>KHAIGHGP0NC8_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55101886</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>160</td>\n",
       "      <td>KHAIGHGP0NC8_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55105679</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>6</td>\n",
       "      <td>KHAIGHGP0NC8_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1533</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55118982</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>32</td>\n",
       "      <td>KHULTRGPONC1_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1534</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55119042</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>17</td>\n",
       "      <td>KHULTRGPONC1_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1535</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55120835</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>14</td>\n",
       "      <td>KHULTRGPONC1_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1536</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55121546</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>6</td>\n",
       "      <td>KHULTRGPONC1_Mutect2_output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1537</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55127139</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>8</td>\n",
       "      <td>KHULTRGPONC1_Mutect2_output</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1538 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS rsID REF ALT Gene Name      Zygosity   DP  \\\n",
       "0     chr4  55079914    .   A   G       KDR  Heterozygous   53   \n",
       "1     chr4  55081941    .   A   G       KDR  Heterozygous   53   \n",
       "2     chr4  55089065    .   T   C       KDR  Heterozygous   24   \n",
       "3     chr4  55101886    .   A   C       KDR  Heterozygous  160   \n",
       "4     chr4  55105679    .   G   T       KDR  Heterozygous    6   \n",
       "...    ...       ...  ...  ..  ..       ...           ...  ...   \n",
       "1533  chr4  55118982    .   A   G       KDR  Heterozygous   32   \n",
       "1534  chr4  55119042    .   C   G       KDR  Heterozygous   17   \n",
       "1535  chr4  55120835    .   G   T       KDR  Heterozygous   14   \n",
       "1536  chr4  55121546    .   T   C       KDR  Heterozygous    6   \n",
       "1537  chr4  55127139    .   G   A       KDR  Heterozygous    8   \n",
       "\n",
       "                           Sample  \n",
       "0     KHAIGHGP0NC8_Mutect2_output  \n",
       "1     KHAIGHGP0NC8_Mutect2_output  \n",
       "2     KHAIGHGP0NC8_Mutect2_output  \n",
       "3     KHAIGHGP0NC8_Mutect2_output  \n",
       "4     KHAIGHGP0NC8_Mutect2_output  \n",
       "...                           ...  \n",
       "1533  KHULTRGPONC1_Mutect2_output  \n",
       "1534  KHULTRGPONC1_Mutect2_output  \n",
       "1535  KHULTRGPONC1_Mutect2_output  \n",
       "1536  KHULTRGPONC1_Mutect2_output  \n",
       "1537  KHULTRGPONC1_Mutect2_output  \n",
       "\n",
       "[1538 rows x 9 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Set the path to the folder containing your Excel files\n",
    "folder_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/KDR_112_samples_Mutect2_outputs/Mutect2_outputs/'\n",
    "\n",
    "# Initialize an empty list to store individual DataFrames\n",
    "dfs = []\n",
    "\n",
    "# Iterate over each file in the folder\n",
    "for filename in os.listdir(folder_path):\n",
    "    if filename.endswith(\".xlsx\"):\n",
    "        file_path = os.path.join(folder_path, filename)\n",
    "        \n",
    "        # Read the Excel file into a DataFrame\n",
    "        df = pd.read_excel(file_path)\n",
    "        \n",
    "        # Extract the \"Sample\" value from the base name of the file\n",
    "        sample_name = os.path.splitext(filename)[0]\n",
    "        \n",
    "        # Add a new column \"Sample\" with the extracted value\n",
    "        df['Sample'] = sample_name\n",
    "                \n",
    "        # Append the DataFrame to the list\n",
    "        dfs.append(df)\n",
    "\n",
    "# Concatenate all DataFrames in the list\n",
    "concatenated_data = pd.concat(dfs, ignore_index=True)\n",
    "\n",
    "# Save the final DataFrame to a new Excel file\n",
    "output_file_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/KDR_112_samples_Mutect2_outputs/Mutect2_outputs/All_MUTECT2_concatenated_data.xlsx'\n",
    "concatenated_data.to_excel(output_file_path, index=False)\n",
    "\n",
    "# Print the concatenated DataFrame\n",
    "concatenated_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "fd34e3b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>DP</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55079914</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>53</td>\n",
       "      <td>KHAIGHGP0NC8</td>\n",
       "      <td>chr4_55079914_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55081941</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>53</td>\n",
       "      <td>KHAIGHGP0NC8</td>\n",
       "      <td>chr4_55081941_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55089065</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>24</td>\n",
       "      <td>KHAIGHGP0NC8</td>\n",
       "      <td>chr4_55089065_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55101886</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>160</td>\n",
       "      <td>KHAIGHGP0NC8</td>\n",
       "      <td>chr4_55101886_A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55105679</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>6</td>\n",
       "      <td>KHAIGHGP0NC8</td>\n",
       "      <td>chr4_55105679_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1533</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55118982</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>32</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "      <td>chr4_55118982_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1534</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55119042</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>17</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "      <td>chr4_55119042_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1535</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55120835</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>14</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "      <td>chr4_55120835_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1536</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55121546</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>6</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "      <td>chr4_55121546_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1537</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55127139</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>8</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "      <td>chr4_55127139_G_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1538 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS rsID REF ALT Gene Name      Zygosity   DP        Sample  \\\n",
       "0     chr4  55079914    .   A   G       KDR  Heterozygous   53  KHAIGHGP0NC8   \n",
       "1     chr4  55081941    .   A   G       KDR  Heterozygous   53  KHAIGHGP0NC8   \n",
       "2     chr4  55089065    .   T   C       KDR  Heterozygous   24  KHAIGHGP0NC8   \n",
       "3     chr4  55101886    .   A   C       KDR  Heterozygous  160  KHAIGHGP0NC8   \n",
       "4     chr4  55105679    .   G   T       KDR  Heterozygous    6  KHAIGHGP0NC8   \n",
       "...    ...       ...  ...  ..  ..       ...           ...  ...           ...   \n",
       "1533  chr4  55118982    .   A   G       KDR  Heterozygous   32  KHULTRGPONC1   \n",
       "1534  chr4  55119042    .   C   G       KDR  Heterozygous   17  KHULTRGPONC1   \n",
       "1535  chr4  55120835    .   G   T       KDR  Heterozygous   14  KHULTRGPONC1   \n",
       "1536  chr4  55121546    .   T   C       KDR  Heterozygous    6  KHULTRGPONC1   \n",
       "1537  chr4  55127139    .   G   A       KDR  Heterozygous    8  KHULTRGPONC1   \n",
       "\n",
       "                Variant  \n",
       "0     chr4_55079914_A_G  \n",
       "1     chr4_55081941_A_G  \n",
       "2     chr4_55089065_T_C  \n",
       "3     chr4_55101886_A_C  \n",
       "4     chr4_55105679_G_T  \n",
       "...                 ...  \n",
       "1533  chr4_55118982_A_G  \n",
       "1534  chr4_55119042_C_G  \n",
       "1535  chr4_55120835_G_T  \n",
       "1536  chr4_55121546_T_C  \n",
       "1537  chr4_55127139_G_A  \n",
       "\n",
       "[1538 rows x 10 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/KDR_112_samples_Mutect2_outputs/Mutect2_outputs/All_MUTECT2_concatenated_data.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e30df043",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Sample_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55076834_T_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC33</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55077777_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4_55077783_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC91</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4_55077786_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4_55077790_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>chr4_55127121_G_T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>chr4_55127139_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC12,KHAIGHGPONC16,KHAIGHGPONC17,KHAI...</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>chr4_55127146_T_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>chr4_55127162_CC_GG</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC37</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>chr4_55127210_T_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>262 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Variant      Zygosity  \\\n",
       "0      chr4_55076834_T_C  Heterozygous   \n",
       "1      chr4_55077777_T_G  Heterozygous   \n",
       "2      chr4_55077783_G_A  Heterozygous   \n",
       "3      chr4_55077786_T_G  Heterozygous   \n",
       "4      chr4_55077790_T_G  Heterozygous   \n",
       "..                   ...           ...   \n",
       "257    chr4_55127121_G_T  Heterozygous   \n",
       "258    chr4_55127139_G_A  Heterozygous   \n",
       "259    chr4_55127146_T_A  Heterozygous   \n",
       "260  chr4_55127162_CC_GG  Heterozygous   \n",
       "261    chr4_55127210_T_A  Heterozygous   \n",
       "\n",
       "                                                Sample  Sample_count  \n",
       "0                                        KHHSPTGPONC33             1  \n",
       "1                                        KHHSPTGPONC89             1  \n",
       "2                                        KHHSPTGPONC91             1  \n",
       "3                                        KHHSPTGPONC89             1  \n",
       "4                                        KHHSPTGPONC89             1  \n",
       "..                                                 ...           ...  \n",
       "257                                       KHAIGHGPONC5             1  \n",
       "258  KHAIGHGPONC12,KHAIGHGPONC16,KHAIGHGPONC17,KHAI...            23  \n",
       "259                                      KHAIGHGPONC18             1  \n",
       "260                                      KHHSPTGPONC37             1  \n",
       "261                                      KHHSPTGPONC89             1  \n",
       "\n",
       "[262 rows x 4 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group by specified columns and aggregate using a dictionary\n",
    "agg_dict = {'Sample': lambda x: ','.join(x.unique())}\n",
    "\n",
    "result_df = df.groupby(['Variant', 'Zygosity']).agg(agg_dict).reset_index()\n",
    "result_df['Sample_count'] = result_df['Sample'].apply(lambda x: len(x.split(',')))\n",
    "\n",
    "# Print the result\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "20d35c75",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/KDR_112_samples_Mutect2_outputs/Mutect2_outputs/Grouped_MUTECT2_variant_counts.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "33d7ec2c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Concatenate</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55079914_A_G_KHAIGHGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55081901_G_A_KHAIGHGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4_55081941_A_G_KHAIGHGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4_55089065_T_C_KHAIGHGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4_55101639_A_G_KHAIGHGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2946</th>\n",
       "      <td>chr4_55118982_A_G_KHULTRGPONC1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2947</th>\n",
       "      <td>chr4_55119042_C_G_KHULTRGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2948</th>\n",
       "      <td>chr4_55120835_G_T_KHULTRGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949</th>\n",
       "      <td>chr4_55121546_T_C_KHULTRGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950</th>\n",
       "      <td>chr4_55127139_G_A_KHULTRGPONC1</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2951 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Concatenate      Zygosity\n",
       "0     chr4_55079914_A_G_KHAIGHGPONC1  Heterozygous\n",
       "1     chr4_55081901_G_A_KHAIGHGPONC1  Heterozygous\n",
       "2     chr4_55081941_A_G_KHAIGHGPONC1  Heterozygous\n",
       "3     chr4_55089065_T_C_KHAIGHGPONC1  Heterozygous\n",
       "4     chr4_55101639_A_G_KHAIGHGPONC1  Heterozygous\n",
       "...                              ...           ...\n",
       "2946  chr4_55118982_A_G_KHULTRGPONC1    Homozygous\n",
       "2947  chr4_55119042_C_G_KHULTRGPONC1  Heterozygous\n",
       "2948  chr4_55120835_G_T_KHULTRGPONC1  Heterozygous\n",
       "2949  chr4_55121546_T_C_KHULTRGPONC1  Heterozygous\n",
       "2950  chr4_55127139_G_A_KHULTRGPONC1  Heterozygous\n",
       "\n",
       "[2951 rows x 2 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/Variant_Zygosity_sample_data.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "e4a8a89b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55079914_A_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55081901_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4_55081941_A_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4_55089065_T_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4_55101639_A_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2932</th>\n",
       "      <td>chr4_55118982_A_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2933</th>\n",
       "      <td>chr4_55119042_C_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2934</th>\n",
       "      <td>chr4_55120835_G_T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2935</th>\n",
       "      <td>chr4_55121546_T_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2936</th>\n",
       "      <td>chr4_55127139_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHULTRGPONC1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1545 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant      Zygosity        Sample\n",
       "0     chr4_55079914_A_G  Heterozygous  KHAIGHGPONC1\n",
       "1     chr4_55081901_G_A  Heterozygous  KHAIGHGPONC1\n",
       "2     chr4_55081941_A_G  Heterozygous  KHAIGHGPONC1\n",
       "3     chr4_55089065_T_C  Heterozygous  KHAIGHGPONC1\n",
       "4     chr4_55101639_A_G  Heterozygous  KHAIGHGPONC1\n",
       "...                 ...           ...           ...\n",
       "2932  chr4_55118982_A_G  Heterozygous  KHULTRGPONC1\n",
       "2933  chr4_55119042_C_G  Heterozygous  KHULTRGPONC1\n",
       "2934  chr4_55120835_G_T  Heterozygous  KHULTRGPONC1\n",
       "2935  chr4_55121546_T_C  Heterozygous  KHULTRGPONC1\n",
       "2936  chr4_55127139_G_A  Heterozygous  KHULTRGPONC1\n",
       "\n",
       "[1545 rows x 3 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.drop_duplicates(subset=['Concatenate'], keep='first')\n",
    "df['CHROM'] = df['Concatenate'].str.split('_').str[0]\n",
    "df['POS'] = df['Concatenate'].str.split('_').str[1]\n",
    "df['REF'] = df['Concatenate'].str.split('_').str[2]\n",
    "df['ALT'] = df['Concatenate'].str.split('_').str[3]\n",
    "df['Sample'] = df['Concatenate'].str.split('_').str[4]\n",
    "df['Variant'] = df['CHROM'] + \"_\" + df['POS'] + \"_\" + df['REF'] + \"_\" + df['ALT']\n",
    "df = df[['Variant', 'Zygosity', 'Sample']]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "6eb9a55f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Sample_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_55076834_T_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC33</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4_55077777_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4_55077783_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC91</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4_55077786_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4_55077790_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>chr4_55127139_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC12,KHAIGHGPONC16,KHAIGHGPONC17,KHAI...</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>chr4_55127146_T_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHAIGHGPONC18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>chr4_55127162_CC_GG</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC37</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>chr4_55127210_T_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC89</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>chr4_55128099_A_T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KHHSPTGPONC45</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>299 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Variant      Zygosity  \\\n",
       "0      chr4_55076834_T_C  Heterozygous   \n",
       "1      chr4_55077777_T_G  Heterozygous   \n",
       "2      chr4_55077783_G_A  Heterozygous   \n",
       "3      chr4_55077786_T_G  Heterozygous   \n",
       "4      chr4_55077790_T_G  Heterozygous   \n",
       "..                   ...           ...   \n",
       "294    chr4_55127139_G_A  Heterozygous   \n",
       "295    chr4_55127146_T_A  Heterozygous   \n",
       "296  chr4_55127162_CC_GG  Heterozygous   \n",
       "297    chr4_55127210_T_A  Heterozygous   \n",
       "298    chr4_55128099_A_T  Heterozygous   \n",
       "\n",
       "                                                Sample  Sample_count  \n",
       "0                                        KHHSPTGPONC33             1  \n",
       "1                                        KHHSPTGPONC89             1  \n",
       "2                                        KHHSPTGPONC91             1  \n",
       "3                                        KHHSPTGPONC89             1  \n",
       "4                                        KHHSPTGPONC89             1  \n",
       "..                                                 ...           ...  \n",
       "294  KHAIGHGPONC12,KHAIGHGPONC16,KHAIGHGPONC17,KHAI...            23  \n",
       "295                                      KHAIGHGPONC18             1  \n",
       "296                                      KHHSPTGPONC37             1  \n",
       "297                                      KHHSPTGPONC89             1  \n",
       "298                                      KHHSPTGPONC45             1  \n",
       "\n",
       "[299 rows x 4 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group by specified columns and aggregate using a dictionary\n",
    "agg_dict = {'Sample': lambda x: ','.join(x.unique())}\n",
    "\n",
    "result_df = df.groupby(['Variant', 'Zygosity']).agg(agg_dict).reset_index()\n",
    "result_df['Sample_count'] = result_df['Sample'].apply(lambda x: len(x.split(',')))\n",
    "\n",
    "# Print the result\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "8dac1ac7",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/Final_counts_updated.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0d803c3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3de71cb5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e4747b2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6ed3142",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71b8501f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "89c8b808",
   "metadata": {},
   "source": [
    "# Concatenate the 1,2,4,5 columns in my excel files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "cd1f3b58",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Cannot set a DataFrame with multiple columns to the single column Variant",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[34], line 25\u001b[0m\n\u001b[0;32m     23\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m filename\u001b[38;5;241m.\u001b[39mendswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m.xlsx\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[0;32m     24\u001b[0m             file_path \u001b[38;5;241m=\u001b[39m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(directory_path, filename)\n\u001b[1;32m---> 25\u001b[0m             \u001b[43mconcatenate_columns\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfile_path\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     26\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mData files created with the new variant concatenated column\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "Cell \u001b[1;32mIn[34], line 12\u001b[0m, in \u001b[0;36mconcatenate_columns\u001b[1;34m(file_path)\u001b[0m\n\u001b[0;32m      9\u001b[0m concatenated_column \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39miloc[:, [\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m1\u001b[39m, \u001b[38;5;241m3\u001b[39m, \u001b[38;5;241m4\u001b[39m]]\u001b[38;5;241m.\u001b[39mastype(\u001b[38;5;28mstr\u001b[39m)\u001b[38;5;241m.\u001b[39mapply(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin, axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[0;32m     11\u001b[0m \u001b[38;5;66;03m# Add the new concatenated column to the DataFrame\u001b[39;00m\n\u001b[1;32m---> 12\u001b[0m df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mVariant\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m concatenated_column\n\u001b[0;32m     14\u001b[0m \u001b[38;5;66;03m# Save the updated DataFrame back to the Excel file\u001b[39;00m\n\u001b[0;32m     15\u001b[0m df\u001b[38;5;241m.\u001b[39mto_excel(file_path, index\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py:3970\u001b[0m, in \u001b[0;36mDataFrame.__setitem__\u001b[1;34m(self, key, value)\u001b[0m\n\u001b[0;32m   3968\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_setitem_array(key, value)\n\u001b[0;32m   3969\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(value, DataFrame):\n\u001b[1;32m-> 3970\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_set_item_frame_value\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   3971\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m (\n\u001b[0;32m   3972\u001b[0m     is_list_like(value)\n\u001b[0;32m   3973\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mis_unique\n\u001b[0;32m   3974\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;241m1\u001b[39m \u001b[38;5;241m<\u001b[39m \u001b[38;5;28mlen\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mget_indexer_for([key])) \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mlen\u001b[39m(value)\n\u001b[0;32m   3975\u001b[0m ):\n\u001b[0;32m   3976\u001b[0m     \u001b[38;5;66;03m# Column to set is duplicated\u001b[39;00m\n\u001b[0;32m   3977\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_setitem_array([key], value)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py:4125\u001b[0m, in \u001b[0;36mDataFrame._set_item_frame_value\u001b[1;34m(self, key, value)\u001b[0m\n\u001b[0;32m   4122\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m\n\u001b[0;32m   4124\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(value\u001b[38;5;241m.\u001b[39mcolumns) \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m-> 4125\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[0;32m   4126\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCannot set a DataFrame with multiple columns to the single \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   4127\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcolumn \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mkey\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   4128\u001b[0m     )\n\u001b[0;32m   4130\u001b[0m \u001b[38;5;28mself\u001b[39m[key] \u001b[38;5;241m=\u001b[39m value[value\u001b[38;5;241m.\u001b[39mcolumns[\u001b[38;5;241m0\u001b[39m]]\n",
      "\u001b[1;31mValueError\u001b[0m: Cannot set a DataFrame with multiple columns to the single column Variant"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "def concatenate_columns(file_path):\n",
    "    # Read the Excel file into a DataFrame\n",
    "    df = pd.read_excel(file_path)\n",
    "\n",
    "    # Concatenate the specified columns with \"_\"\n",
    "    concatenated_column = df.iloc[:, [0, 1, 3, 4]].astype(str).apply('_'.join, axis=1)\n",
    "\n",
    "    # Add the new concatenated column to the DataFrame\n",
    "    df['Variant'] = concatenated_column\n",
    "\n",
    "    # Save the updated DataFrame back to the Excel file\n",
    "    df.to_excel(file_path, index=False)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    # Specify the directory containing your Excel files\n",
    "    directory_path = r'C:/Users/GenepoweRx_Madhu/Downloads/KDR_MAF/KDR_113_samples_Haplo_outputs/Haplo_outputs/'\n",
    "\n",
    "    # Loop through each file in the directory\n",
    "    for filename in os.listdir(directory_path):\n",
    "        if filename.endswith(\".xlsx\"):\n",
    "            file_path = os.path.join(directory_path, filename)\n",
    "            concatenate_columns(file_path)\n",
    "print('Data files created with the new variant concatenated column')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4665c83",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22ab5f5e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19203093",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "63e650c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>613196</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55079914</td>\n",
       "      <td>rs4421048</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613197</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55081941</td>\n",
       "      <td>rs2412617</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613202</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55085619</td>\n",
       "      <td>rs555692700</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613203</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55089065</td>\n",
       "      <td>rs1531289</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613206</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55097934</td>\n",
       "      <td>rs6838752</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613208</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55098398</td>\n",
       "      <td>rs3816584</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613209</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55098438</td>\n",
       "      <td>rs2305946</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613211</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55099267</td>\n",
       "      <td>rs58415820</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613212</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55102839</td>\n",
       "      <td>rs13136007</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613214</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55106807</td>\n",
       "      <td>rs1870377</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613215</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55108123</td>\n",
       "      <td>rs6846151</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613220</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55114072</td>\n",
       "      <td>rs7692791</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613221</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55115516</td>\n",
       "      <td>rs41477053</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613223</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55118975</td>\n",
       "      <td>rs1962460</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613224</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55118982</td>\n",
       "      <td>rs1962459</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613225</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55119042</td>\n",
       "      <td>rs1962458</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613226</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55120835</td>\n",
       "      <td>rs2054246</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613228</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55125564</td>\n",
       "      <td>rs7667298</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613229</th>\n",
       "      <td>chr4</td>\n",
       "      <td>55125612</td>\n",
       "      <td>rs6824124</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>KDR</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS         rsID REF ALT Gene Name      Zygosity\n",
       "613196  chr4  55079914    rs4421048   A   G       KDR  Heterozygous\n",
       "613197  chr4  55081941    rs2412617   A   G       KDR  Heterozygous\n",
       "613202  chr4  55085619  rs555692700   G   A       KDR  Heterozygous\n",
       "613203  chr4  55089065    rs1531289   T   C       KDR  Heterozygous\n",
       "613206  chr4  55097934    rs6838752   T   C       KDR  Heterozygous\n",
       "613208  chr4  55098398    rs3816584   A   G       KDR  Heterozygous\n",
       "613209  chr4  55098438    rs2305946   C   T       KDR  Heterozygous\n",
       "613211  chr4  55099267   rs58415820   G   A       KDR  Heterozygous\n",
       "613212  chr4  55102839   rs13136007   C   A       KDR  Heterozygous\n",
       "613214  chr4  55106807    rs1870377   T   A       KDR  Heterozygous\n",
       "613215  chr4  55108123    rs6846151   C   G       KDR  Heterozygous\n",
       "613220  chr4  55114072    rs7692791   C   T       KDR  Heterozygous\n",
       "613221  chr4  55115516   rs41477053   G   T       KDR  Heterozygous\n",
       "613223  chr4  55118975    rs1962460   A   G       KDR  Heterozygous\n",
       "613224  chr4  55118982    rs1962459   A   G       KDR  Heterozygous\n",
       "613225  chr4  55119042    rs1962458   C   G       KDR  Heterozygous\n",
       "613226  chr4  55120835    rs2054246   G   T       KDR  Heterozygous\n",
       "613228  chr4  55125564    rs7667298   T   C       KDR  Heterozygous\n",
       "613229  chr4  55125612    rs6824124   T   A       KDR  Heterozygous"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/ONCO_samples/KHAIGHGP0NC7_MINI.Mutect2_filtered_annotated.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "#vcf['VC'] = vcf['INFO'].str.extract('VC=([^;]+)')\n",
    "# Filter rows where 'VC' is equal to 'SNV'\n",
    "#vcf = vcf[vcf['VC'] == 'SNV']\n",
    "vcf[\"Gene Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf['Zygosity'] = vcf['SAMPLE'].str.split(':').str[0]\n",
    "data = {'Gene Name': ['KDR']}\n",
    "df_gene = pd.DataFrame(data)\n",
    "vcf['Gene_Match'] = 'No'\n",
    "# Iterate through each gene in vcf\n",
    "for genes in vcf['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            vcf.loc[vcf['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "vcf = vcf[vcf['Gene_Match'] == 'Yes']\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'Gene Name', 'Zygosity']]\n",
    "vcf['Zygosity'] = vcf['Zygosity'].replace({'1/1': 'Homozygous', '0/1': 'Heterozygous', '0|1': 'Heterozygous', '1|1': 'Homozygous'})\n",
    "#vcf.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/ONCO_samples/KHAIGHGP0NC7_MINI.Mutect2_filtered_annotated.xlsx', index=False)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "ed6d4034",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Heterozygous    19\n",
       "Name: Zygosity, dtype: int64"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Zygosity'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "476b195a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "baf3d23a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05514acc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ce2d7b1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
