{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "86728fbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>Ref</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>rsid</th>\n",
       "      <th>allele</th>\n",
       "      <th>zygosity</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987.0</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>No</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>81</td>\n",
       "      <td>81</td>\n",
       "      <td>1</td>\n",
       "      <td>80</td>\n",
       "      <td>98.77%</td>\n",
       "      <td>2.241E-46</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>18</td>\n",
       "      <td>ADP=81</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>G5A</td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=140777846</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.13763857033639143,0.86236142966360856</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.1793,0.8207</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005170126000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1353988</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1355779.0</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>0/1</td>\n",
       "      <td>59</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>73.68%</td>\n",
       "      <td>1.2025E-6</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>ADP=19</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=0</td>\n",
       "      <td>RS=201260508</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.22555109582059123,0.77444890417940876</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0,1</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05000008000500003e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1355780</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1402505.0</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>CCCGGGGAA</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>20</td>\n",
       "      <td>10</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>3.9851E-4</td>\n",
       "      <td>19</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=30</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>45003826.0</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>22.22%</td>\n",
       "      <td>2.5282E-3</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>ADP=36</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>147757520.0</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>AAGGTG</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>7</td>\n",
       "      <td>28%</td>\n",
       "      <td>4.8126E-3</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=24</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5406</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>STIL</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs75930208</td>\n",
       "      <td>['TAA', T]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Primary_Microcephaly', '_Recessive</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5407</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>GNPTAB</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs76300806</td>\n",
       "      <td>['AGCC', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Mucolipidosis_type_II|Pseudo-Hurler_polydystro...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>LHFPL5</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs796097780</td>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Nonsyndromic_Hearing_Loss', '_Recessive</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs796117827</td>\n",
       "      <td>['GT', G]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Medium-chain_acyl-coenzyme_A_dehydrogenase_def...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5410</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>SOS1</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs869215095</td>\n",
       "      <td>['G', GT]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5411 rows × 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM          POS Gene Name Gene_Match         rsID        REF     ALT  \\\n",
       "0     chr1    1353987.0     MXRA8         No  rs140777846        CTG       C   \n",
       "1     chr1    1355779.0     MXRA8         No  rs201260508         GA       G   \n",
       "2     chr1    1402505.0                   No            .  CCCGGGGAA       C   \n",
       "3     chr1   45003826.0                   No            .          G      GC   \n",
       "4     chr1  147757520.0                   No            .          A  AAGGTG   \n",
       "...    ...          ...       ...        ...          ...        ...     ...   \n",
       "5406    NA           NA      STIL         No           NA         NA      NA   \n",
       "5407    NA           NA    GNPTAB         No           NA         NA      NA   \n",
       "5408    NA           NA    LHFPL5         No           NA         NA      NA   \n",
       "5409    NA           NA     ACADM         No           NA         NA      NA   \n",
       "5410    NA           NA      SOS1         No           NA         NA      NA   \n",
       "\n",
       "       GT   GQ SDP  DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR     ADP  \\\n",
       "0     1/1  255  81  81   1  80  98.77%  2.241E-46   1   0  62  18  ADP=81   \n",
       "1     0/1   59  19  19   5  14  73.68%  1.2025E-6   5   0  10   4  ADP=19   \n",
       "2     0/1   33  30  30  20  10  33.33%  3.9851E-4  19   1   9   1  ADP=30   \n",
       "3     0/1   25  36  36  28   8  22.22%  2.5282E-3  21   7   3   5  ADP=36   \n",
       "4     0/1   23  25  24  18   7     28%  4.8126E-3  18   0   5   2  ADP=24   \n",
       "...   ...  ...  ..  ..  ..  ..     ...        ...  ..  ..  ..  ..     ...   \n",
       "5406   NA   NA  NA  NA  NA  NA      NA         NA  NA  NA  NA  NA      NA   \n",
       "5407   NA   NA  NA  NA  NA  NA      NA         NA  NA  NA  NA  NA      NA   \n",
       "5408   NA   NA  NA  NA  NA  NA      NA         NA  NA  NA  NA  NA      NA   \n",
       "5409   NA   NA  NA  NA  NA  NA      NA         NA  NA  NA  NA  NA      NA   \n",
       "5410   NA   NA  NA  NA  NA  NA      NA         NA  NA  NA  NA  NA      NA   \n",
       "\n",
       "        WT    HET    HOM    NC CDA OTH S3D WTD      dbSNPBuildID  SLO NSF  R3  \\\n",
       "0     WT=0  HET=0  HOM=1  NC=0                  dbSNPBuildID=134  SLO           \n",
       "1     WT=0  HET=1  HOM=0  NC=0                  dbSNPBuildID=137                \n",
       "2     WT=0  HET=1  HOM=0  NC=0                                                  \n",
       "3     WT=0  HET=1  HOM=0  NC=0                                                  \n",
       "4     WT=0  HET=1  HOM=0  NC=0                                                  \n",
       "...    ...    ...    ...   ...  ..  ..  ..  ..               ...  ...  ..  ..   \n",
       "5406    NA     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA   \n",
       "5407    NA     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA   \n",
       "5408    NA     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA   \n",
       "5409    NA     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA   \n",
       "5410    NA     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA   \n",
       "\n",
       "      R5 NSN NSM  G5A    COMMON            RS  RV TPA CFL  GNO  VLD  ASP ASS  \\\n",
       "0                 G5A  COMMON=1  RS=140777846              GNO  VLD  ASP       \n",
       "1                      COMMON=0  RS=201260508                        ASP       \n",
       "2                                                                              \n",
       "3                                                                              \n",
       "4                                                                              \n",
       "...   ..  ..  ..  ...       ...           ...  ..  ..  ..  ...  ...  ...  ..   \n",
       "5406  NA  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA   \n",
       "5407  NA  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA   \n",
       "5408  NA  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA   \n",
       "5409  NA  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA   \n",
       "5410  NA  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA   \n",
       "\n",
       "     Ref  U3  U5                                          TOPMED    WGT MTP  \\\n",
       "0                 TOPMED=0.13763857033639143,0.86236142966360856  WGT=1       \n",
       "1                 TOPMED=0.22555109582059123,0.77444890417940876  WGT=1       \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4                                                                             \n",
       "...   ..  ..  ..                                             ...    ...  ..   \n",
       "5406  NA  NA  NA                                              NA     NA  NA   \n",
       "5407  NA  NA  NA                                              NA     NA  NA   \n",
       "5408  NA  NA  NA                                              NA     NA  NA   \n",
       "5409  NA  NA  NA                                              NA     NA  NA   \n",
       "5410  NA  NA  NA                                              NA     NA  NA   \n",
       "\n",
       "     LSD NOC DSS SYN  KGPhase3                CAF      VC MUT  KGPhase1 NOV  \\\n",
       "0                     KGPhase3  CAF=0.1793,0.8207  VC=DIV                     \n",
       "1                     KGPhase3            CAF=0,1  VC=DIV      KGPhase1       \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4                                                                             \n",
       "...   ..  ..  ..  ..       ...                ...     ...  ..       ...  ..   \n",
       "5406  NA  NA  NA  NA        NA                 NA      NA  NA        NA  NA   \n",
       "5407  NA  NA  NA  NA        NA                 NA      NA  NA        NA  NA   \n",
       "5408  NA  NA  NA  NA        NA                 NA      NA  NA        NA  NA   \n",
       "5409  NA  NA  NA  NA        NA                 NA      NA  NA        NA  NA   \n",
       "5410  NA  NA  NA  NA        NA                 NA      NA  NA        NA  NA   \n",
       "\n",
       "                                 VP    SAO  INT  G5  OM PMC    SSR  \\\n",
       "0     VP=0x050100080005170126000200  SAO=0  INT  G5          SSR=0   \n",
       "1     VP=0x05000008000500003e000200  SAO=0  INT              SSR=0   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...                             ...    ...  ...  ..  ..  ..    ...   \n",
       "5406                             NA     NA   NA  NA  NA  NA     NA   \n",
       "5407                             NA     NA   NA  NA  NA  NA     NA   \n",
       "5408                             NA     NA   NA  NA  NA  NA     NA   \n",
       "5409                             NA     NA   NA  NA  NA  NA     NA   \n",
       "5410                             NA     NA   NA  NA  NA  NA     NA   \n",
       "\n",
       "              RSPOS  HD  PM ClinVar ClinVar_CLNSIG         rsid       allele  \\\n",
       "0     RSPOS=1353988                                 rs140777846           NA   \n",
       "1     RSPOS=1355780                                 rs201260508           NA   \n",
       "2                                                             .           NA   \n",
       "3                                                             .           NA   \n",
       "4                                                             .           NA   \n",
       "...             ...  ..  ..     ...            ...          ...          ...   \n",
       "5406             NA  NA  NA      NA             NA   rs75930208   ['TAA', T]   \n",
       "5407             NA  NA  NA      NA             NA   rs76300806  ['AGCC', A]   \n",
       "5408             NA  NA  NA      NA             NA  rs796097780    ['AT', A]   \n",
       "5409             NA  NA  NA      NA             NA  rs796117827    ['GT', G]   \n",
       "5410             NA  NA  NA      NA             NA  rs869215095    ['G', GT]   \n",
       "\n",
       "          zygosity          consequence Consequence_score  \\\n",
       "0               NA                   NA                NA   \n",
       "1               NA                   NA                NA   \n",
       "2               NA                   NA                NA   \n",
       "3               NA                   NA                NA   \n",
       "4               NA                   NA                NA   \n",
       "...            ...                  ...               ...   \n",
       "5406  Heterozygous  3_prime_UTR_variant              3/10   \n",
       "5407  Heterozygous  5_prime_UTR_variant              3/10   \n",
       "5408  Heterozygous  3_prime_UTR_variant              3/10   \n",
       "5409    Homozygous       intron_variant              2/10   \n",
       "5410  Heterozygous       intron_variant              2/10   \n",
       "\n",
       "     clinical_significance                                associated_diseases  \\\n",
       "0                       NA                                                 NA   \n",
       "1                       NA                                                 NA   \n",
       "2                       NA                                                 NA   \n",
       "3                       NA                                                 NA   \n",
       "4                       NA                                                 NA   \n",
       "...                    ...                                                ...   \n",
       "5406                Benign                 Primary_Microcephaly', '_Recessive   \n",
       "5407                Benign  Mucolipidosis_type_II|Pseudo-Hurler_polydystro...   \n",
       "5408         Likely_benign            Nonsyndromic_Hearing_Loss', '_Recessive   \n",
       "5409                Benign  Medium-chain_acyl-coenzyme_A_dehydrogenase_def...   \n",
       "5410                Benign                         not_specified|not_provided   \n",
       "\n",
       "                                          review_status    origin  \\\n",
       "0                                                    NA        NA   \n",
       "1                                                    NA        NA   \n",
       "2                                                    NA        NA   \n",
       "3                                                    NA        NA   \n",
       "4                                                    NA        NA   \n",
       "...                                                 ...       ...   \n",
       "5406             criteria_provided', '_single_submitter  germline   \n",
       "5407  criteria_provided', '_multiple_submitters', '_...  germline   \n",
       "5408             criteria_provided', '_single_submitter  germline   \n",
       "5409  criteria_provided', '_multiple_submitters', '_...  germline   \n",
       "5410  criteria_provided', '_multiple_submitters', '_...  germline   \n",
       "\n",
       "     variant_type variant_subtype Phargkb_ann_exists is_mutation  \\\n",
       "0              NA              NA                 NA          NA   \n",
       "1              NA              NA                 NA          NA   \n",
       "2              NA              NA                 NA          NA   \n",
       "3              NA              NA                 NA          NA   \n",
       "4              NA              NA                 NA          NA   \n",
       "...           ...             ...                ...         ...   \n",
       "5406        indel        Deletion           MayBe No          No   \n",
       "5407        indel        Deletion           MayBe No          No   \n",
       "5408        indel        Deletion           MayBe No          No   \n",
       "5409        indel        Deletion           MayBe No          No   \n",
       "5410        indel       Insertion           MayBe No          No   \n",
       "\n",
       "     Variant_is_precious  \n",
       "0                     NA  \n",
       "1                     NA  \n",
       "2                     NA  \n",
       "3                     NA  \n",
       "4                     NA  \n",
       "...                  ...  \n",
       "5406                  No  \n",
       "5407                 Yes  \n",
       "5408                  No  \n",
       "5409                 Yes  \n",
       "5410                 Yes  \n",
       "\n",
       "[5411 rows x 87 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "######################################## Loading the Libraries #############################################################\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE\n",
    "\n",
    "################################## Covered rows extraction ###############################################################\n",
    "\n",
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Desktop/Covered_regions.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/Madhu/KHCDPRGPTTL13/KHCDPRGPTTL13_annotated_indel.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL13_annotated_indel.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n",
    "\n",
    "\n",
    "####################### Loading the Data #################################################################################\n",
    "\n",
    "data = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL13_annotated_indel.vcf', comment='#', sep= '\\t', header=None, low_memory=False)\n",
    "data.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "sample_cols = data['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "data = pd.concat([data, sample_cols], axis=1)\n",
    "data = data[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "\n",
    "######################################## Extracting the Total columns ###################################################\n",
    "\n",
    "# Create empty columns\n",
    "columns = ['ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID', 'SLO',\n",
    "           'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV', 'TPA', 'CFL', 'GNO',\n",
    "           'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5', 'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC',\n",
    "           'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO',\n",
    "           'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar',\n",
    "           'ClinVar_CLNSIG']\n",
    "\n",
    "for col in columns:\n",
    "    data[col] = ''\n",
    "\n",
    "# Populate columns based on 'INFO' values\n",
    "for i, row in data.iterrows():\n",
    "    info = row['INFO']\n",
    "    items = info.split(';')\n",
    "    for item in items:\n",
    "        key_value = item.split('=')\n",
    "        key = key_value[0]\n",
    "        if key in columns:\n",
    "            if len(key_value) > 1:\n",
    "                value = key_value[1]\n",
    "                data.at[i, key] = f\"{key}={value}\"\n",
    "            else:\n",
    "                data.at[i, key] = key\n",
    "        else:\n",
    "            data.at[i, key] = 'null'\n",
    "\n",
    "\n",
    "            \n",
    "data[\"Gene_Name\"] = data[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "data['Gene Name'] = data['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "\n",
    "\n",
    "data = data[['CHROM', 'POS', 'Gene Name', 'rsID', 'REF', 'ALT', 'GT',\n",
    "       'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD',\n",
    "       'dbSNPBuildID', 'SLO', 'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON',\n",
    "       'RS', 'RV', 'TPA', 'CFL', 'GNO', 'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5',\n",
    "       'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC', 'DSS', 'SYN', 'KGPhase3', 'CAF',\n",
    "       'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO', 'INT', 'G5',\n",
    "       'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar', 'ClinVar_CLNSIG']]\n",
    "\n",
    "data['rsid'] = data['rsID'].str.split(';')\n",
    "data = data.explode('rsid')\n",
    "\n",
    "\n",
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL10_11_12_13_clnsig_files/KHCDPRGPTTL13/KHCDPRGPTTL13_indel_clinical_significance.xlsx')\n",
    "df = df.rename(columns={'gene_name': 'Gene name', 'zygocity':'zygosity'})\n",
    "\n",
    "merged = pd.merge(data, df, on = 'rsid', how = 'outer', sort=False)\n",
    "\n",
    "merged['Gene Name'] = merged['Gene Name'].fillna(merged['Gene name'])\n",
    "\n",
    "cond_genes = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/cardiac genes.xlsx')\n",
    "\n",
    "merged['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in merged['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in cond_genes['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            merged.loc[merged['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "            \n",
    "merged['Consequence'] = merged['consequence'].astype(str).apply(lambda x: x.replace('&', ',').replace('_', ' ').replace(\"'\", '').replace(\"-\", ' '))\n",
    "merged['Consequence'] = merged['Consequence'].str.split(',').str[0]\n",
    "\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1 = df_1.rename({'consequence':'Consequence'}, axis =1)\n",
    "\n",
    "\n",
    "merged_1 = pd.merge(merged, df_1, on='Consequence', how='left', sort=False)\n",
    "\n",
    "\n",
    "merged_1 = merged_1[['CHROM', 'POS', 'Gene Name', 'Gene_Match', 'rsID', 'REF', 'ALT', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'ADP',\n",
    "       'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID',\n",
    "       'SLO', 'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV',\n",
    "       'TPA', 'CFL', 'GNO', 'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5', 'TOPMED',\n",
    "       'WGT', 'MTP', 'LSD', 'NOC', 'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC',\n",
    "       'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO', 'INT', 'G5', 'OM', 'PMC', 'SSR',\n",
    "       'RSPOS', 'HD', 'PM', 'ClinVar', 'ClinVar_CLNSIG', 'rsid', 'allele',\n",
    "       'zygosity', 'consequence', 'Consequence_score', 'clinical_significance',\n",
    "       'associated_diseases', 'review_status', 'origin', 'variant_type',\n",
    "       'variant_subtype', 'Phargkb_ann_exists', 'is_mutation',\n",
    "       'Variant_is_precious']]\n",
    "merged_1 = merged_1.fillna('NA')\n",
    "merged_1\n",
    "\n",
    "#merged_1.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/InDel_files/KHAIGHGPTTL207_InDel_final.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c8bea7f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
