{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2487c112",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>Ref</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>rsid</th>\n",
       "      <th>allele</th>\n",
       "      <th>zygosity</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382.0</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>0/1</td>\n",
       "      <td>97</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>52%</td>\n",
       "      <td>1.7375E-10</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>ADP=50</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=143128930</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.91126720183486238,0.08873279816513761</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.8806,0.1194</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x0500000a000515003e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1299383</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1331945.0</td>\n",
       "      <td>TAS1R3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs200330269</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>34.15%</td>\n",
       "      <td>1.5388E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>ADP=41</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=200330269</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.82975758154943934,0.17023445463812436...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.8105,0.1895,.</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05000008000515003e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1331945</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs200330269</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987.0</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>No</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>219</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "      <td>19</td>\n",
       "      <td>51</td>\n",
       "      <td>72.86%</td>\n",
       "      <td>1.2075E-22</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>17</td>\n",
       "      <td>ADP=70</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>G5A</td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=140777846</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.13763857033639143,0.86236142966360856</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.1793,0.8207</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005170126000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1353988</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1355779.0</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>0/1</td>\n",
       "      <td>88</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>72.41%</td>\n",
       "      <td>1.2841E-9</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>ADP=29</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=0</td>\n",
       "      <td>RS=201260508</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.22555109582059123,0.77444890417940876</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0,1</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05000008000500003e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1355780</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs201260508</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1657358.0</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>No</td>\n",
       "      <td>rs377230281</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0/1</td>\n",
       "      <td>78</td>\n",
       "      <td>112</td>\n",
       "      <td>112</td>\n",
       "      <td>88</td>\n",
       "      <td>24</td>\n",
       "      <td>21.43%</td>\n",
       "      <td>1.4999E-8</td>\n",
       "      <td>65</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>ADP=112</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=138</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=377230281</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.98587219673802242,0.01412780326197757</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000080005000002000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1657358</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs377230281</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4953</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>KAT6B</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs71929101</td>\n",
       "      <td>['GGAA', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>6/10</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Genitopatellar_syndrome|not_specified|not_prov...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4954</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>KAT6B</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs71929101</td>\n",
       "      <td>['GGAA', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>6/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Genitopatellar_syndrome|not_provided</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4955</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>GNPTAB</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs76300806</td>\n",
       "      <td>['AGCC', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Mucolipidosis_type_II|Pseudo-Hurler_polydystro...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4956</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>EEF1A2</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs77467883</td>\n",
       "      <td>['AG', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4957</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>SOS1</td>\n",
       "      <td>No</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>rs869215095</td>\n",
       "      <td>['G', GT]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4958 rows × 87 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS Gene Name Gene Match         rsID  REF ALT   GT   GQ  \\\n",
       "0     chr1  1299382.0     ACAP3         No  rs143128930   AG   A  0/1   97   \n",
       "1     chr1  1331945.0    TAS1R3         No  rs200330269    G  GC  0/1   48   \n",
       "2     chr1  1353987.0     MXRA8         No  rs140777846  CTG   C  0/1  219   \n",
       "3     chr1  1355779.0     MXRA8         No  rs201260508   GA   G  0/1   88   \n",
       "4     chr1  1657358.0    CDK11B         No  rs377230281    T  TA  0/1   78   \n",
       "...    ...        ...       ...        ...          ...  ...  ..  ...  ...   \n",
       "4953    NA         NA     KAT6B         No           NA   NA  NA   NA   NA   \n",
       "4954    NA         NA     KAT6B         No           NA   NA  NA   NA   NA   \n",
       "4955    NA         NA    GNPTAB         No           NA   NA  NA   NA   NA   \n",
       "4956    NA         NA    EEF1A2         No           NA   NA  NA   NA   NA   \n",
       "4957    NA         NA      SOS1         No           NA   NA  NA   NA   NA   \n",
       "\n",
       "      SDP   DP  RD  AD    FREQ        PVAL RDF RDR ADF ADR      ADP    WT  \\\n",
       "0      50   50  24  26     52%  1.7375E-10  18   6  17   9   ADP=50  WT=0   \n",
       "1      41   41  26  14  34.15%   1.5388E-5  12  14   2  12   ADP=41  WT=0   \n",
       "2      70   70  19  51  72.86%  1.2075E-22  12   7  34  17   ADP=70  WT=0   \n",
       "3      29   29   8  21  72.41%   1.2841E-9   7   1   8  13   ADP=29  WT=0   \n",
       "4     112  112  88  24  21.43%   1.4999E-8  65  23   4  20  ADP=112  WT=0   \n",
       "...   ...  ...  ..  ..     ...         ...  ..  ..  ..  ..      ...   ...   \n",
       "4953   NA   NA  NA  NA      NA          NA  NA  NA  NA  NA       NA    NA   \n",
       "4954   NA   NA  NA  NA      NA          NA  NA  NA  NA  NA       NA    NA   \n",
       "4955   NA   NA  NA  NA      NA          NA  NA  NA  NA  NA       NA    NA   \n",
       "4956   NA   NA  NA  NA      NA          NA  NA  NA  NA  NA       NA    NA   \n",
       "4957   NA   NA  NA  NA      NA          NA  NA  NA  NA  NA       NA    NA   \n",
       "\n",
       "        HET    HOM    NC CDA OTH S3D WTD      dbSNPBuildID  SLO NSF  R3  R5  \\\n",
       "0     HET=1  HOM=0  NC=0                  dbSNPBuildID=134               R5   \n",
       "1     HET=1  HOM=0  NC=0                  dbSNPBuildID=137                    \n",
       "2     HET=1  HOM=0  NC=0                  dbSNPBuildID=134  SLO               \n",
       "3     HET=1  HOM=0  NC=0                  dbSNPBuildID=137                    \n",
       "4     HET=1  HOM=0  NC=0                  dbSNPBuildID=138                    \n",
       "...     ...    ...   ...  ..  ..  ..  ..               ...  ...  ..  ..  ..   \n",
       "4953     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA  NA   \n",
       "4954     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA  NA   \n",
       "4955     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA  NA   \n",
       "4956     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA  NA   \n",
       "4957     NA     NA    NA  NA  NA  NA  NA                NA   NA  NA  NA  NA   \n",
       "\n",
       "     NSN NSM  G5A    COMMON            RS  RV TPA CFL  GNO  VLD  ASP ASS Ref  \\\n",
       "0                  COMMON=1  RS=143128930                   VLD  ASP           \n",
       "1                  COMMON=1  RS=200330269                   VLD  ASP           \n",
       "2             G5A  COMMON=1  RS=140777846              GNO  VLD  ASP           \n",
       "3                  COMMON=0  RS=201260508                        ASP           \n",
       "4                            RS=377230281                        ASP           \n",
       "...   ..  ..  ...       ...           ...  ..  ..  ..  ...  ...  ...  ..  ..   \n",
       "4953  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA  NA   \n",
       "4954  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA  NA   \n",
       "4955  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA  NA   \n",
       "4956  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA  NA   \n",
       "4957  NA  NA   NA        NA            NA  NA  NA  NA   NA   NA   NA  NA  NA   \n",
       "\n",
       "      U3  U5                                             TOPMED    WGT MTP  \\\n",
       "0                TOPMED=0.91126720183486238,0.08873279816513761  WGT=1       \n",
       "1             TOPMED=0.82975758154943934,0.17023445463812436...  WGT=1       \n",
       "2                TOPMED=0.13763857033639143,0.86236142966360856  WGT=1       \n",
       "3                TOPMED=0.22555109582059123,0.77444890417940876  WGT=1       \n",
       "4                TOPMED=0.98587219673802242,0.01412780326197757  WGT=1       \n",
       "...   ..  ..                                                ...    ...  ..   \n",
       "4953  NA  NA                                                 NA     NA  NA   \n",
       "4954  NA  NA                                                 NA     NA  NA   \n",
       "4955  NA  NA                                                 NA     NA  NA   \n",
       "4956  NA  NA                                                 NA     NA  NA   \n",
       "4957  NA  NA                                                 NA     NA  NA   \n",
       "\n",
       "     LSD NOC DSS SYN  KGPhase3                  CAF      VC MUT  KGPhase1 NOV  \\\n",
       "0                     KGPhase3    CAF=0.8806,0.1194  VC=DIV      KGPhase1       \n",
       "1                     KGPhase3  CAF=0.8105,0.1895,.  VC=DIV      KGPhase1       \n",
       "2                     KGPhase3    CAF=0.1793,0.8207  VC=DIV                     \n",
       "3                     KGPhase3              CAF=0,1  VC=DIV      KGPhase1       \n",
       "4                                                    VC=DIV                     \n",
       "...   ..  ..  ..  ..       ...                  ...     ...  ..       ...  ..   \n",
       "4953  NA  NA  NA  NA        NA                   NA      NA  NA        NA  NA   \n",
       "4954  NA  NA  NA  NA        NA                   NA      NA  NA        NA  NA   \n",
       "4955  NA  NA  NA  NA        NA                   NA      NA  NA        NA  NA   \n",
       "4956  NA  NA  NA  NA        NA                   NA      NA  NA        NA  NA   \n",
       "4957  NA  NA  NA  NA        NA                   NA      NA  NA        NA  NA   \n",
       "\n",
       "                                 VP    SAO  INT  G5  OM PMC    SSR  \\\n",
       "0     VP=0x0500000a000515003e000200  SAO=0  INT  G5          SSR=0   \n",
       "1     VP=0x05000008000515003e000200  SAO=0  INT  G5          SSR=0   \n",
       "2     VP=0x050100080005170126000200  SAO=0  INT  G5          SSR=0   \n",
       "3     VP=0x05000008000500003e000200  SAO=0  INT              SSR=0   \n",
       "4     VP=0x050000080005000002000200  SAO=0  INT              SSR=0   \n",
       "...                             ...    ...  ...  ..  ..  ..    ...   \n",
       "4953                             NA     NA   NA  NA  NA  NA     NA   \n",
       "4954                             NA     NA   NA  NA  NA  NA     NA   \n",
       "4955                             NA     NA   NA  NA  NA  NA     NA   \n",
       "4956                             NA     NA   NA  NA  NA  NA     NA   \n",
       "4957                             NA     NA   NA  NA  NA  NA     NA   \n",
       "\n",
       "              RSPOS  HD  PM ClinVar ClinVar_CLNSIG         rsid       allele  \\\n",
       "0     RSPOS=1299383                                 rs143128930           NA   \n",
       "1     RSPOS=1331945                                 rs200330269           NA   \n",
       "2     RSPOS=1353988                                 rs140777846           NA   \n",
       "3     RSPOS=1355780                                 rs201260508           NA   \n",
       "4     RSPOS=1657358                                 rs377230281           NA   \n",
       "...             ...  ..  ..     ...            ...          ...          ...   \n",
       "4953             NA  NA  NA      NA             NA   rs71929101  ['GGAA', G]   \n",
       "4954             NA  NA  NA      NA             NA   rs71929101  ['GGAA', G]   \n",
       "4955             NA  NA  NA      NA             NA   rs76300806  ['AGCC', A]   \n",
       "4956             NA  NA  NA      NA             NA   rs77467883    ['AG', A]   \n",
       "4957             NA  NA  NA      NA             NA  rs869215095    ['G', GT]   \n",
       "\n",
       "          zygosity          consequence Consequence_score  \\\n",
       "0               NA                   NA                NA   \n",
       "1               NA                   NA                NA   \n",
       "2               NA                   NA                NA   \n",
       "3               NA                   NA                NA   \n",
       "4               NA                   NA                NA   \n",
       "...            ...                  ...               ...   \n",
       "4953  Heterozygous     inframe_deletion              6/10   \n",
       "4954  Heterozygous     inframe_deletion              6/10   \n",
       "4955  Heterozygous  5_prime_UTR_variant              3/10   \n",
       "4956    Homozygous       intron_variant              2/10   \n",
       "4957    Homozygous       intron_variant              2/10   \n",
       "\n",
       "     clinical_significance                                associated_diseases  \\\n",
       "0                       NA                                                 NA   \n",
       "1                       NA                                                 NA   \n",
       "2                       NA                                                 NA   \n",
       "3                       NA                                                 NA   \n",
       "4                       NA                                                 NA   \n",
       "...                    ...                                                ...   \n",
       "4953  Benign/Likely_benign  Genitopatellar_syndrome|not_specified|not_prov...   \n",
       "4954                Benign               Genitopatellar_syndrome|not_provided   \n",
       "4955                Benign  Mucolipidosis_type_II|Pseudo-Hurler_polydystro...   \n",
       "4956                Benign                                       not_provided   \n",
       "4957                Benign                         not_specified|not_provided   \n",
       "\n",
       "                                          review_status    origin  \\\n",
       "0                                                    NA        NA   \n",
       "1                                                    NA        NA   \n",
       "2                                                    NA        NA   \n",
       "3                                                    NA        NA   \n",
       "4                                                    NA        NA   \n",
       "...                                                 ...       ...   \n",
       "4953  criteria_provided', '_multiple_submitters', '_...  germline   \n",
       "4954             criteria_provided', '_single_submitter  germline   \n",
       "4955  criteria_provided', '_multiple_submitters', '_...  germline   \n",
       "4956             criteria_provided', '_single_submitter  germline   \n",
       "4957  criteria_provided', '_multiple_submitters', '_...  germline   \n",
       "\n",
       "     variant_type variant_subtype Phargkb_ann_exists is_mutation  \\\n",
       "0              NA              NA                 NA          NA   \n",
       "1              NA              NA                 NA          NA   \n",
       "2              NA              NA                 NA          NA   \n",
       "3              NA              NA                 NA          NA   \n",
       "4              NA              NA                 NA          NA   \n",
       "...           ...             ...                ...         ...   \n",
       "4953        indel        Deletion           MayBe No          No   \n",
       "4954        indel        Deletion           MayBe No          No   \n",
       "4955        indel        Deletion           MayBe No          No   \n",
       "4956        indel        Deletion           MayBe No          No   \n",
       "4957        indel       Insertion           MayBe No          No   \n",
       "\n",
       "     Variant_is_precious  \n",
       "0                     NA  \n",
       "1                     NA  \n",
       "2                     NA  \n",
       "3                     NA  \n",
       "4                     NA  \n",
       "...                  ...  \n",
       "4953                 Yes  \n",
       "4954                 Yes  \n",
       "4955                 Yes  \n",
       "4956                  No  \n",
       "4957                 Yes  \n",
       "\n",
       "[4958 rows x 87 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE\n",
    "\n",
    "\n",
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Desktop/Covered_regions.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/Madhu/KHCDPRGPTTL13/KHCDPRGPTTL13_annotated_indel.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL13_annotated_indel.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n",
    "\n",
    "    \n",
    "data = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGHGPTTL207_annotated_indel.vcf', comment='#', sep= '\\t', header=None, low_memory=False)\n",
    "data.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = data['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "data = pd.concat([data, sample_cols], axis=1)\n",
    "data = data[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "\n",
    "# Create empty columns\n",
    "columns = ['ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID', 'SLO',\n",
    "           'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV', 'TPA', 'CFL', 'GNO',\n",
    "           'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5', 'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC',\n",
    "           'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO',\n",
    "           'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar',\n",
    "           'ClinVar_CLNSIG']\n",
    "\n",
    "for col in columns:\n",
    "    data[col] = ''\n",
    "\n",
    "# Populate columns based on 'INFO' values\n",
    "for i, row in data.iterrows():\n",
    "    info = row['INFO']\n",
    "    items = info.split(';')\n",
    "    for item in items:\n",
    "        key_value = item.split('=')\n",
    "        key = key_value[0]\n",
    "        if key in columns:\n",
    "            if len(key_value) > 1:\n",
    "                value = key_value[1]\n",
    "                data.at[i, key] = f\"{key}={value}\"\n",
    "            else:\n",
    "                data.at[i, key] = key\n",
    "        else:\n",
    "            data.at[i, key] = 'null'\n",
    "\n",
    "\n",
    "            \n",
    "data[\"Gene_Name\"] = data[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "data['Gene Name'] = data['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "\n",
    "\n",
    "data = data[['CHROM', 'POS', 'Gene Name', 'rsID', 'REF', 'ALT', 'GT',\n",
    "       'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD',\n",
    "       'dbSNPBuildID', 'SLO', 'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON',\n",
    "       'RS', 'RV', 'TPA', 'CFL', 'GNO', 'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5',\n",
    "       'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC', 'DSS', 'SYN', 'KGPhase3', 'CAF',\n",
    "       'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO', 'INT', 'G5',\n",
    "       'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar', 'ClinVar_CLNSIG']]\n",
    "\n",
    "data['rsid'] = data['rsID'].str.split(';')\n",
    "data = data.explode('rsid')\n",
    "\n",
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHAIGHGPTTL207_indel_clinical_significance.xlsx')\n",
    "df = df.rename(columns={'gene_name': 'Gene name', 'zygocity':'zygosity'})\n",
    "\n",
    "merged = pd.merge(data, df, on = 'rsid', how = 'outer', sort=False)\n",
    "\n",
    "merged['Gene Name'] = merged['Gene Name'].fillna(merged['Gene name'])\n",
    "\n",
    "cond_genes = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Park_cere_genes.xlsx')\n",
    "\n",
    "merged['Gene Match'] = 'No'\n",
    "merged['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in cond_genes['Gene Name'].values:\n",
    "                merged.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "cond_genes = cond_genes.rename({'Gene Name':'Matched_Gene', 'Gene_Match':'Gene Match'}, axis=1)\n",
    "cond_genes\n",
    "\n",
    "merged = merged.drop(columns=['Gene Match'], axis=1)\n",
    "merged_2 = pd.merge(merged, cond_genes, on= 'Matched_Gene', how = 'left', sort = False)\n",
    "merged_2['Gene Match'] = merged_2['Gene Match'].fillna('No')\n",
    "merged_2\n",
    "\n",
    "merged_2['Consequence'] = merged_2['consequence'].astype(str).apply(lambda x: x.replace('&', ',').replace('_', ' ').replace(\"'\", '').replace(\"-\", ' '))\n",
    "merged_2['Consequence'] = merged_2['Consequence'].str.split(',').str[0]\n",
    "\n",
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1 = df_1.rename({'consequence':'Consequence'}, axis =1)\n",
    "\n",
    "\n",
    "merged_1 = pd.merge(merged_2, df_1, on='Consequence', how='left', sort=False)\n",
    "\n",
    "\n",
    "merged_1 = merged_1[['CHROM', 'POS', 'Gene Name', 'Gene Match', 'rsID', 'REF', 'ALT', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'ADP',\n",
    "       'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID',\n",
    "       'SLO', 'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV',\n",
    "       'TPA', 'CFL', 'GNO', 'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5', 'TOPMED',\n",
    "       'WGT', 'MTP', 'LSD', 'NOC', 'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC',\n",
    "       'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO', 'INT', 'G5', 'OM', 'PMC', 'SSR',\n",
    "       'RSPOS', 'HD', 'PM', 'ClinVar', 'ClinVar_CLNSIG', 'rsid', 'allele',\n",
    "       'zygosity', 'consequence', 'Consequence_score', 'clinical_significance',\n",
    "       'associated_diseases', 'review_status', 'origin', 'variant_type',\n",
    "       'variant_subtype', 'Phargkb_ann_exists', 'is_mutation',\n",
    "       'Variant_is_precious']]\\\n",
    "\n",
    "merged_1 = merged_1.fillna('NA')\n",
    "merged_1\n",
    "\n",
    "#merged_1.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/InDel_files/KHAIGHGPTTL207_InDel_final.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90db4ba9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
