{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cb9214f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0171915",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "988df16c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba1192de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32aa792f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e8b35445",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>118.3520</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:145,255,0:230:0:0,201:1,28:1,229:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,255,0:93:0:0,38:0,55:0,93:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,178,0:59:0:0,53:0,6:0,59:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>205.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:235,51,0:17:0:0,15:0,2:0,17:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,6,0:2:0:0,2:0,0:0,2:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,166,0:55:0:0,14:0,41:0,55:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79443</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1418109</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>25.4267</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:55,255,0:99:0:0,26:0,73:0,99:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>26.4242</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:56,208,0:69:0:0,17:0,52:0,69:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1428198</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,148,0:49:0:0,12:0,37:0,49:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2221159</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>28.4205</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:58,255,0:153:0:0,42:0,111:0,153:0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79447 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM      POS       rsID REF ALT      QUAL FILTER  \\\n",
       "0      chr1   817514  rs3131971   T   C  118.3520      .   \n",
       "1      chr1   826893  rs3115849   G   A   34.4159      .   \n",
       "2      chr1   930939  rs9988021   G   A  225.4170      .   \n",
       "3      chr1   941119  rs4372192   A   G  205.4170      .   \n",
       "4      chr1   942451  rs6672356   T   C   34.4159      .   \n",
       "...     ...      ...        ...  ..  ..       ...    ...   \n",
       "79442  chrY  1387425          .   C   G   23.4340      .   \n",
       "79443  chrY  1418109          .   C   G   25.4267      .   \n",
       "79444  chrY  1425778          .   G   A   26.4242      .   \n",
       "79445  chrY  1428198          .   C   A   23.4340      .   \n",
       "79446  chrY  2221159          .   T   C   28.4205      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...   \n",
       "1      DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...   \n",
       "2      DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...   \n",
       "3      DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...   \n",
       "4      DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...   \n",
       "...                                                  ...   \n",
       "79442  DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...   \n",
       "79443  DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...   \n",
       "79444  DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...   \n",
       "79445  DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...   \n",
       "79446  DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...   \n",
       "\n",
       "                           FORMAT                                  SAMPLE  \n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR  1/1:145,255,0:230:0:0,201:1,28:1,229:0  \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:64,255,0:93:0:0,38:0,55:0,93:0  \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,178,0:59:0:0,53:0,6:0,59:0  \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR       1/1:235,51,0:17:0:0,15:0,2:0,17:0  \n",
       "4      GT:PL:DP:SP:ADF:ADR:AD:SCR            1/1:64,6,0:2:0:0,2:0,0:0,2:0  \n",
       "...                           ...                                     ...  \n",
       "79442  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,166,0:55:0:0,14:0,41:0,55:0  \n",
       "79443  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:55,255,0:99:0:0,26:0,73:0,99:0  \n",
       "79444  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:56,208,0:69:0:0,17:0,52:0,69:0  \n",
       "79445  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,148,0:49:0:0,12:0,37:0,49:0  \n",
       "79446  GT:PL:DP:SP:ADF:ADR:AD:SCR   1/1:58,255,0:153:0:0,42:0,111:0,153:0  \n",
       "\n",
       "[79447 rows x 10 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_BCFTOOL.vcf', comment= '#', sep = '\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0b6c1639",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>118.3520</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:145,255,0:230:0:0,201:1,28:1,229:0</td>\n",
       "      <td>1,229</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.995652</td>\n",
       "      <td>230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,255,0:93:0:0,38:0,55:0,93:0</td>\n",
       "      <td>0,93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,178,0:59:0:0,53:0,6:0,59:0</td>\n",
       "      <td>0,59</td>\n",
       "      <td>0</td>\n",
       "      <td>59</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>205.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:235,51,0:17:0:0,15:0,2:0,17:0</td>\n",
       "      <td>0,17</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,6,0:2:0:0,2:0,0:0,2:0</td>\n",
       "      <td>0,2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,166,0:55:0:0,14:0,41:0,55:0</td>\n",
       "      <td>0,55</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79443</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1418109</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>25.4267</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:55,255,0:99:0:0,26:0,73:0,99:0</td>\n",
       "      <td>0,99</td>\n",
       "      <td>0</td>\n",
       "      <td>99</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>26.4242</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:56,208,0:69:0:0,17:0,52:0,69:0</td>\n",
       "      <td>0,69</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1428198</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,148,0:49:0:0,12:0,37:0,49:0</td>\n",
       "      <td>0,49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2221159</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>28.4205</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:58,255,0:153:0:0,42:0,111:0,153:0</td>\n",
       "      <td>0,153</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79447 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM      POS       rsID REF ALT      QUAL FILTER  \\\n",
       "0      chr1   817514  rs3131971   T   C  118.3520      .   \n",
       "1      chr1   826893  rs3115849   G   A   34.4159      .   \n",
       "2      chr1   930939  rs9988021   G   A  225.4170      .   \n",
       "3      chr1   941119  rs4372192   A   G  205.4170      .   \n",
       "4      chr1   942451  rs6672356   T   C   34.4159      .   \n",
       "...     ...      ...        ...  ..  ..       ...    ...   \n",
       "79442  chrY  1387425          .   C   G   23.4340      .   \n",
       "79443  chrY  1418109          .   C   G   25.4267      .   \n",
       "79444  chrY  1425778          .   G   A   26.4242      .   \n",
       "79445  chrY  1428198          .   C   A   23.4340      .   \n",
       "79446  chrY  2221159          .   T   C   28.4205      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...   \n",
       "1      DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...   \n",
       "2      DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...   \n",
       "3      DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...   \n",
       "4      DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...   \n",
       "...                                                  ...   \n",
       "79442  DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...   \n",
       "79443  DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...   \n",
       "79444  DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...   \n",
       "79445  DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...   \n",
       "79446  DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...   \n",
       "\n",
       "                           FORMAT                                  SAMPLE  \\\n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR  1/1:145,255,0:230:0:0,201:1,28:1,229:0   \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:64,255,0:93:0:0,38:0,55:0,93:0   \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,178,0:59:0:0,53:0,6:0,59:0   \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR       1/1:235,51,0:17:0:0,15:0,2:0,17:0   \n",
       "4      GT:PL:DP:SP:ADF:ADR:AD:SCR            1/1:64,6,0:2:0:0,2:0,0:0,2:0   \n",
       "...                           ...                                     ...   \n",
       "79442  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,166,0:55:0:0,14:0,41:0,55:0   \n",
       "79443  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:55,255,0:99:0:0,26:0,73:0,99:0   \n",
       "79444  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:56,208,0:69:0:0,17:0,52:0,69:0   \n",
       "79445  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,148,0:49:0:0,12:0,37:0,49:0   \n",
       "79446  GT:PL:DP:SP:ADF:ADR:AD:SCR   1/1:58,255,0:153:0:0,42:0,111:0,153:0   \n",
       "\n",
       "          AD  RD  A_D       VAF   DP  \n",
       "0      1,229   1  229  0.995652  230  \n",
       "1       0,93   0   93  1.000000   93  \n",
       "2       0,59   0   59  1.000000   59  \n",
       "3       0,17   0   17  1.000000   17  \n",
       "4        0,2   0    2  1.000000    2  \n",
       "...      ...  ..  ...       ...  ...  \n",
       "79442   0,55   0   55  1.000000   55  \n",
       "79443   0,99   0   99  1.000000   99  \n",
       "79444   0,69   0   69  1.000000   69  \n",
       "79445   0,49   0   49  1.000000   49  \n",
       "79446  0,153   0  153  1.000000  153  \n",
       "\n",
       "[79447 rows x 15 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[6]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2].fillna('0').astype(int)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0bf9ba5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(79447, 15)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1 = vcf[vcf['VAF'] >= 0.1]\n",
    "vcf_1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e22b7508",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7850a4b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39e864d3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "501054f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>Allele_Count</th>\n",
       "      <th>VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>34.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.2682,0.7318;COMMON=1;CSQ=A...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:64,255,0</td>\n",
       "      <td>97</td>\n",
       "      <td>1</td>\n",
       "      <td>0.010309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.04653,0.9535,.;COMMON=1;CS...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:255,178,0</td>\n",
       "      <td>67</td>\n",
       "      <td>1</td>\n",
       "      <td>0.014925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>205.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.08686,0.9131;COMMON=1;CSQ=...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:235,51,0</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>0.043478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>34.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0,1;COMMON=0;CSQ=C|downstrea...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:64,6,0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944296</td>\n",
       "      <td>rs6605067</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>71.41</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.114,0.886;COMMON=1;CSQ=A|3...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:101,9,0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79440</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154929926</td>\n",
       "      <td>rs1800292</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>222.38</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AN=2;ASP;BQB=0.999581;CAF=0.8395,0.1605;C...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>0/1:255,0,255</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>0.011628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79441</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1286017</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>27.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|CSF2RA...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:57,151,0</td>\n",
       "      <td>54</td>\n",
       "      <td>1</td>\n",
       "      <td>0.018519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>23.43</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=G|downstream_gene_variant|MODIFI...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:53,166,0</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "      <td>0.015385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>26.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:56,208,0</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>0.012658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1428198</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>23.43</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:53,148,0</td>\n",
       "      <td>57</td>\n",
       "      <td>1</td>\n",
       "      <td>0.017544</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66295 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS       rsID REF ALT    QUAL FILTER  \\\n",
       "1      chr1     826893  rs3115849   G   A   34.42      .   \n",
       "2      chr1     930939  rs9988021   G   A  225.42      .   \n",
       "3      chr1     941119  rs4372192   A   G  205.42      .   \n",
       "4      chr1     942451  rs6672356   T   C   34.42      .   \n",
       "5      chr1     944296  rs6605067   G   A   71.41      .   \n",
       "...     ...        ...        ...  ..  ..     ...    ...   \n",
       "79440  chrX  154929926  rs1800292   T   G  222.38      .   \n",
       "79441  chrY    1286017          .   C   A   27.42      .   \n",
       "79442  chrY    1387425          .   C   G   23.43      .   \n",
       "79444  chrY    1425778          .   G   A   26.42      .   \n",
       "79445  chrY    1428198          .   C   A   23.43      .   \n",
       "\n",
       "                                                    INFO FORMAT  \\\n",
       "1      AC=2;AN=2;ASP;CAF=0.2682,0.7318;COMMON=1;CSQ=A...  GT:PL   \n",
       "2      AC=2;AN=2;ASP;CAF=0.04653,0.9535,.;COMMON=1;CS...  GT:PL   \n",
       "3      AC=2;AN=2;ASP;CAF=0.08686,0.9131;COMMON=1;CSQ=...  GT:PL   \n",
       "4      AC=2;AN=2;ASP;CAF=0,1;COMMON=0;CSQ=C|downstrea...  GT:PL   \n",
       "5      AC=2;AN=2;ASP;CAF=0.114,0.886;COMMON=1;CSQ=A|3...  GT:PL   \n",
       "...                                                  ...    ...   \n",
       "79440  AC=1;AN=2;ASP;BQB=0.999581;CAF=0.8395,0.1605;C...  GT:PL   \n",
       "79441  AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|CSF2RA...  GT:PL   \n",
       "79442  AC=2;AN=2;CSQ=G|downstream_gene_variant|MODIFI...  GT:PL   \n",
       "79444  AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...  GT:PL   \n",
       "79445  AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...  GT:PL   \n",
       "\n",
       "              SAMPLE  DP  Allele_Count       VAF  \n",
       "1       1/1:64,255,0  97             1  0.010309  \n",
       "2      1/1:255,178,0  67             1  0.014925  \n",
       "3       1/1:235,51,0  23             1  0.043478  \n",
       "4         1/1:64,6,0   2             1  0.500000  \n",
       "5        1/1:101,9,0   3             1  0.333333  \n",
       "...              ...  ..           ...       ...  \n",
       "79440  0/1:255,0,255  86             1  0.011628  \n",
       "79441   1/1:57,151,0  54             1  0.018519  \n",
       "79442   1/1:53,166,0  65             1  0.015385  \n",
       "79444   1/1:56,208,0  79             1  0.012658  \n",
       "79445   1/1:53,148,0  57             1  0.017544  \n",
       "\n",
       "[66295 rows x 13 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['DP'] = vcf['INFO'].str.extract(r'DP=(\\d+)')[0].fillna('0').astype(int)\n",
    "vcf['Allele_Count'] = vcf['ALT'].apply(lambda x: len(x))\n",
    "vcf['VAF'] = vcf['Allele_Count'] / vcf['DP']\n",
    "vcf_new = vcf[vcf['VAF'] >= 0.01]\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "87afb7ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>Allele_Count</th>\n",
       "      <th>VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>34.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.2682,0.7318;COMMON=1;CSQ=A...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:64,255,0</td>\n",
       "      <td>97</td>\n",
       "      <td>1</td>\n",
       "      <td>0.010309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.04653,0.9535,.;COMMON=1;CS...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:255,178,0</td>\n",
       "      <td>67</td>\n",
       "      <td>1</td>\n",
       "      <td>0.014925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>205.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.08686,0.9131;COMMON=1;CSQ=...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:235,51,0</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>0.043478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>225.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.07907,0.9209;COMMON=1;CSQ=...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:255,36,0</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0.071429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946653</td>\n",
       "      <td>rs2272756</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>191.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;ASP;CAF=0.8632,0.1368,.;COMMON=1;CSQ...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:221,72,0</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>0.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79440</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154929926</td>\n",
       "      <td>rs1800292</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>222.38</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AN=2;ASP;BQB=0.999581;CAF=0.8395,0.1605;C...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>0/1:255,0,255</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>0.011628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79441</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1286017</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>27.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|CSF2RA...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:57,151,0</td>\n",
       "      <td>54</td>\n",
       "      <td>1</td>\n",
       "      <td>0.018519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>23.43</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=G|downstream_gene_variant|MODIFI...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:53,166,0</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "      <td>0.015385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>26.42</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:56,208,0</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>0.012658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1428198</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>23.43</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...</td>\n",
       "      <td>GT:PL</td>\n",
       "      <td>1/1:53,148,0</td>\n",
       "      <td>57</td>\n",
       "      <td>1</td>\n",
       "      <td>0.017544</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>60927 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS       rsID REF ALT    QUAL FILTER  \\\n",
       "1      chr1     826893  rs3115849   G   A   34.42      .   \n",
       "2      chr1     930939  rs9988021   G   A  225.42      .   \n",
       "3      chr1     941119  rs4372192   A   G  205.42      .   \n",
       "7      chr1     944858  rs3748592   A   G  225.42      .   \n",
       "8      chr1     946653  rs2272756   G   A  191.42      .   \n",
       "...     ...        ...        ...  ..  ..     ...    ...   \n",
       "79440  chrX  154929926  rs1800292   T   G  222.38      .   \n",
       "79441  chrY    1286017          .   C   A   27.42      .   \n",
       "79442  chrY    1387425          .   C   G   23.43      .   \n",
       "79444  chrY    1425778          .   G   A   26.42      .   \n",
       "79445  chrY    1428198          .   C   A   23.43      .   \n",
       "\n",
       "                                                    INFO FORMAT  \\\n",
       "1      AC=2;AN=2;ASP;CAF=0.2682,0.7318;COMMON=1;CSQ=A...  GT:PL   \n",
       "2      AC=2;AN=2;ASP;CAF=0.04653,0.9535,.;COMMON=1;CS...  GT:PL   \n",
       "3      AC=2;AN=2;ASP;CAF=0.08686,0.9131;COMMON=1;CSQ=...  GT:PL   \n",
       "7      AC=2;AN=2;ASP;CAF=0.07907,0.9209;COMMON=1;CSQ=...  GT:PL   \n",
       "8      AC=2;AN=2;ASP;CAF=0.8632,0.1368,.;COMMON=1;CSQ...  GT:PL   \n",
       "...                                                  ...    ...   \n",
       "79440  AC=1;AN=2;ASP;BQB=0.999581;CAF=0.8395,0.1605;C...  GT:PL   \n",
       "79441  AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|CSF2RA...  GT:PL   \n",
       "79442  AC=2;AN=2;CSQ=G|downstream_gene_variant|MODIFI...  GT:PL   \n",
       "79444  AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...  GT:PL   \n",
       "79445  AC=2;AN=2;CSQ=A|intron_variant|MODIFIER|ASMTL|...  GT:PL   \n",
       "\n",
       "              SAMPLE  DP  Allele_Count       VAF  \n",
       "1       1/1:64,255,0  97             1  0.010309  \n",
       "2      1/1:255,178,0  67             1  0.014925  \n",
       "3       1/1:235,51,0  23             1  0.043478  \n",
       "7       1/1:255,36,0  14             1  0.071429  \n",
       "8       1/1:221,72,0  24             1  0.041667  \n",
       "...              ...  ..           ...       ...  \n",
       "79440  0/1:255,0,255  86             1  0.011628  \n",
       "79441   1/1:57,151,0  54             1  0.018519  \n",
       "79442   1/1:53,166,0  65             1  0.015385  \n",
       "79444   1/1:56,208,0  79             1  0.012658  \n",
       "79445   1/1:53,148,0  57             1  0.017544  \n",
       "\n",
       "[60927 rows x 13 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new = vcf_new[vcf_new['DP'] >= 10]\n",
    "vcf_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75097cf7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2967ef19",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "68964b8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>REF_Count</th>\n",
       "      <th>ALT_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>935662</td>\n",
       "      <td>rs199537431</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>42.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CAF=0.9916,0.008387;COMMON=1;GENEINFO=SAMD...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/1:24:51:19,32:0.627451:41,0,24</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939398</td>\n",
       "      <td>rs568340123</td>\n",
       "      <td>GCCTCCCCAGCCACGGTGAGGACCCACCCTGGCATGATCCCCCTCATCA</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;GENEINFO=SAMD11:148398;RS=568340123;RSPOS=...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:23:97:78,19:0.195876:0,23,39</td>\n",
       "      <td>49</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945259</td>\n",
       "      <td>rs35916504;rs397844509</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>46.2</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CAF=0.8119,0.1881,0.8119,0.1881;COMMON=1,1...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:35:26:1,25:0.961538:46,34,0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>948728</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>0.2</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>CSQ=C|intron_variant|MODIFIER|NOC2L|ENSG000001...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:14:13:11,2:0.153846:0,14,34</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>948730</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>CCT</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>CSQ=CT|intron_variant|MODIFIER|NOC2L|ENSG00000...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:28:12:10,2:0.166667:0,29,33</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113459</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155089365</td>\n",
       "      <td>rs1434044365</td>\n",
       "      <td>ATG</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;GENEINFO=BRCC3:79184;INT;RS=1434044365;RSP...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:30:85:52,33:0.388235:0,30,44</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113460</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155099185</td>\n",
       "      <td>rs782693515</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;GENEINFO=BRCC3:79184;INT;RS=782693515;RSPO...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:30:14:11,2:0.142857:0,29,50</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113461</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;GENEINFO=VBP1:7411;INT;NOV;RS=781912204;RS...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:33:37:29,7:0.189189:0,33,57</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113462</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155491679</td>\n",
       "      <td>.</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>CSQ=-|splice_polypyrimidine_tract_variant&amp;intr...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:24:151:97,41:0.271523:0,25,28</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113467</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5623220</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>0.5</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>CSQ=A|intron_variant|MODIFIER|PCDH11Y|ENSG0000...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:10:17:5,4:0.235294:0,14,10</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21324 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS                    rsID  \\\n",
       "9       chr1     935662             rs199537431   \n",
       "10      chr1     939398             rs568340123   \n",
       "17      chr1     945259  rs35916504;rs397844509   \n",
       "22      chr1     948728                       .   \n",
       "23      chr1     948730                       .   \n",
       "...      ...        ...                     ...   \n",
       "113459  chrX  155089365            rs1434044365   \n",
       "113460  chrX  155099185             rs782693515   \n",
       "113461  chrX  155228363             rs781912204   \n",
       "113462  chrX  155491679                       .   \n",
       "113467  chrY    5623220                       .   \n",
       "\n",
       "                                                      REF  ALT  QUAL   FILTER  \\\n",
       "9                                                       C   CA  42.0     PASS   \n",
       "10      GCCTCCCCAGCCACGGTGAGGACCCACCCTGGCATGATCCCCCTCATCA    G   0.0  RefCall   \n",
       "17                                                     TC    T  46.2     PASS   \n",
       "22                                                      T   TC   0.2  RefCall   \n",
       "23                                                      C  CCT   0.0  RefCall   \n",
       "...                                                   ...  ...   ...      ...   \n",
       "113459                                                ATG    A   0.0  RefCall   \n",
       "113460                                                  G   GT   0.0  RefCall   \n",
       "113461                                                  A   AT   0.0  RefCall   \n",
       "113462                                                 CA    C   0.0  RefCall   \n",
       "113467                                                  C   CA   0.5  RefCall   \n",
       "\n",
       "                                                     INFO              FORMAT  \\\n",
       "9       ASP;CAF=0.9916,0.008387;COMMON=1;GENEINFO=SAMD...  GT:GQ:DP:AD:VAF:PL   \n",
       "10      ASP;GENEINFO=SAMD11:148398;RS=568340123;RSPOS=...  GT:GQ:DP:AD:VAF:PL   \n",
       "17      ASP;CAF=0.8119,0.1881,0.8119,0.1881;COMMON=1,1...  GT:GQ:DP:AD:VAF:PL   \n",
       "22      CSQ=C|intron_variant|MODIFIER|NOC2L|ENSG000001...  GT:GQ:DP:AD:VAF:PL   \n",
       "23      CSQ=CT|intron_variant|MODIFIER|NOC2L|ENSG00000...  GT:GQ:DP:AD:VAF:PL   \n",
       "...                                                   ...                 ...   \n",
       "113459  ASP;GENEINFO=BRCC3:79184;INT;RS=1434044365;RSP...  GT:GQ:DP:AD:VAF:PL   \n",
       "113460  ASP;GENEINFO=BRCC3:79184;INT;RS=782693515;RSPO...  GT:GQ:DP:AD:VAF:PL   \n",
       "113461  ASP;GENEINFO=VBP1:7411;INT;NOV;RS=781912204;RS...  GT:GQ:DP:AD:VAF:PL   \n",
       "113462  CSQ=-|splice_polypyrimidine_tract_variant&intr...  GT:GQ:DP:AD:VAF:PL   \n",
       "113467  CSQ=A|intron_variant|MODIFIER|PCDH11Y|ENSG0000...  GT:GQ:DP:AD:VAF:PL   \n",
       "\n",
       "                                   SAMPLE  REF_Count  ALT_Count  \n",
       "9        0/1:24:51:19,32:0.627451:41,0,24          1          2  \n",
       "10       0/0:23:97:78,19:0.195876:0,23,39         49          1  \n",
       "17        1/1:35:26:1,25:0.961538:46,34,0          2          1  \n",
       "22        ./.:14:13:11,2:0.153846:0,14,34          1          2  \n",
       "23        0/0:28:12:10,2:0.166667:0,29,33          1          3  \n",
       "...                                   ...        ...        ...  \n",
       "113459   0/0:30:85:52,33:0.388235:0,30,44          3          1  \n",
       "113460    0/0:30:14:11,2:0.142857:0,29,50          1          2  \n",
       "113461    0/0:33:37:29,7:0.189189:0,33,57          1          2  \n",
       "113462  0/0:24:151:97,41:0.271523:0,25,28          2          1  \n",
       "113467     ./.:10:17:5,4:0.235294:0,14,10          1          2  \n",
       "\n",
       "[21324 rows x 12 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['REF_Count'] = vcf['REF'].apply(lambda x: len(x))\n",
    "vcf['ALT_Count'] = vcf['ALT'].apply(lambda x: len(x))\n",
    "non_matching_rows = vcf[vcf['REF_Count'] != vcf['ALT_Count']]\n",
    "non_matching_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "271224cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = str('GGAGAGAGAGAGAGAGAGAG')\n",
    "len(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "63804252",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69181</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57084531</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69182</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57128393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69183</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184462</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69184</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57189953</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69185</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191772</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69186 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69511  rs2691305   A   G    .   PASS   \n",
       "1      chr1    817514  rs3131971   T   C    .   PASS   \n",
       "2      chr1    826893  rs3115849   G   A    .   PASS   \n",
       "3      chr1    827209  rs3115848   G   C    .   PASS   \n",
       "4      chr1    827212  rs3131950   C   G    .   PASS   \n",
       "...     ...       ...        ...  ..  ..  ...    ...   \n",
       "69181  chrY  57084531          .   A   G    .   PASS   \n",
       "69182  chrY  57128393          .   G   C    .   PASS   \n",
       "69183  chrY  57184462          .   T   G    .   PASS   \n",
       "69184  chrY  57189953          .   T   C    .   PASS   \n",
       "69185  chrY  57191772          .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "1      ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...   \n",
       "2      ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....   \n",
       "3      ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "4      ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "...                                                  ...   \n",
       "69181  ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69182  ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...   \n",
       "69183  ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69184  ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...   \n",
       "69185  ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "69181  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69182  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69183  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69184  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69185  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...  \n",
       "1      1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...  \n",
       "2      1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55  \n",
       "3      1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...  \n",
       "4      1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...  \n",
       "...                                                  ...  \n",
       "69181  0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...  \n",
       "69182  0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...  \n",
       "69183  0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...  \n",
       "69184   0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9  \n",
       "69185  0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...  \n",
       "\n",
       "[69186 rows x 10 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_VARSCAN2.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f80e66ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...</td>\n",
       "      <td>368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...</td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69181</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57084531</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69182</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57128393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69183</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184462</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69184</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57189953</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69185</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191772</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69186 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69511  rs2691305   A   G    .   PASS   \n",
       "1      chr1    817514  rs3131971   T   C    .   PASS   \n",
       "2      chr1    826893  rs3115849   G   A    .   PASS   \n",
       "3      chr1    827209  rs3115848   G   C    .   PASS   \n",
       "4      chr1    827212  rs3131950   C   G    .   PASS   \n",
       "...     ...       ...        ...  ..  ..  ...    ...   \n",
       "69181  chrY  57084531          .   A   G    .   PASS   \n",
       "69182  chrY  57128393          .   G   C    .   PASS   \n",
       "69183  chrY  57184462          .   T   G    .   PASS   \n",
       "69184  chrY  57189953          .   T   C    .   PASS   \n",
       "69185  chrY  57191772          .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "1      ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...   \n",
       "2      ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....   \n",
       "3      ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "4      ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "...                                                  ...   \n",
       "69181  ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69182  ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...   \n",
       "69183  ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69184  ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...   \n",
       "69185  ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "69181  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69182  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69183  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69184  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69185  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   DP  \n",
       "0      1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...  211  \n",
       "1      1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...  368  \n",
       "2      1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55   91  \n",
       "3      1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...  125  \n",
       "4      1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...  128  \n",
       "...                                                  ...  ...  \n",
       "69181  0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...   62  \n",
       "69182  0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...  223  \n",
       "69183  0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...   43  \n",
       "69184   0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9   50  \n",
       "69185  0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...   76  \n",
       "\n",
       "[69186 rows x 11 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d3a2ba45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>Allele_Count</th>\n",
       "      <th>VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...</td>\n",
       "      <td>211</td>\n",
       "      <td>1</td>\n",
       "      <td>0.004739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55</td>\n",
       "      <td>91</td>\n",
       "      <td>1</td>\n",
       "      <td>0.010989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...</td>\n",
       "      <td>125</td>\n",
       "      <td>1</td>\n",
       "      <td>0.008000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...</td>\n",
       "      <td>128</td>\n",
       "      <td>1</td>\n",
       "      <td>0.007812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69181</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57084531</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...</td>\n",
       "      <td>62</td>\n",
       "      <td>1</td>\n",
       "      <td>0.016129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69182</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57128393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...</td>\n",
       "      <td>223</td>\n",
       "      <td>1</td>\n",
       "      <td>0.004484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69183</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184462</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...</td>\n",
       "      <td>43</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69184</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57189953</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9</td>\n",
       "      <td>50</td>\n",
       "      <td>1</td>\n",
       "      <td>0.020000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69185</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191772</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...</td>\n",
       "      <td>76</td>\n",
       "      <td>1</td>\n",
       "      <td>0.013158</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69186 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69511  rs2691305   A   G    .   PASS   \n",
       "1      chr1    817514  rs3131971   T   C    .   PASS   \n",
       "2      chr1    826893  rs3115849   G   A    .   PASS   \n",
       "3      chr1    827209  rs3115848   G   C    .   PASS   \n",
       "4      chr1    827212  rs3131950   C   G    .   PASS   \n",
       "...     ...       ...        ...  ..  ..  ...    ...   \n",
       "69181  chrY  57084531          .   A   G    .   PASS   \n",
       "69182  chrY  57128393          .   G   C    .   PASS   \n",
       "69183  chrY  57184462          .   T   G    .   PASS   \n",
       "69184  chrY  57189953          .   T   C    .   PASS   \n",
       "69185  chrY  57191772          .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "1      ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...   \n",
       "2      ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....   \n",
       "3      ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "4      ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "...                                                  ...   \n",
       "69181  ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69182  ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...   \n",
       "69183  ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69184  ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...   \n",
       "69185  ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "69181  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69182  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69183  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69184  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69185  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   DP  Allele_Count  \\\n",
       "0      1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...  211             1   \n",
       "1      1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...  368             1   \n",
       "2      1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55   91             1   \n",
       "3      1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...  125             1   \n",
       "4      1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...  128             1   \n",
       "...                                                  ...  ...           ...   \n",
       "69181  0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...   62             1   \n",
       "69182  0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...  223             1   \n",
       "69183  0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...   43             1   \n",
       "69184   0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9   50             1   \n",
       "69185  0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...   76             1   \n",
       "\n",
       "            VAF  \n",
       "0      0.004739  \n",
       "1      0.002717  \n",
       "2      0.010989  \n",
       "3      0.008000  \n",
       "4      0.007812  \n",
       "...         ...  \n",
       "69181  0.016129  \n",
       "69182  0.004484  \n",
       "69183  0.023256  \n",
       "69184  0.020000  \n",
       "69185  0.013158  \n",
       "\n",
       "[69186 rows x 13 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Allele_Count'] = vcf['ALT'].apply(lambda x: len(x))\n",
    "vcf['VAF'] = vcf['Allele_Count'] / vcf['DP']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "1a2cf8b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(57360, 13)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_new = vcf[vcf['VAF'] >= 0.01]\n",
    "vcf_new.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "62311844",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "G                                                        20415\n",
       "C                                                        20250\n",
       "A                                                        19801\n",
       "T                                                        19534\n",
       "CT                                                          98\n",
       "                                                         ...  \n",
       "GGAGAGAGAGAGAGAGAGAG                                         1\n",
       "GTATTATTATTA                                                 1\n",
       "AGAATG                                                       1\n",
       "CCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTCTTC        1\n",
       "AATATATATAT                                                  1\n",
       "Name: ALT, Length: 3012, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.ALT.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9e9e762",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cb406a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "181ffc74",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5faf8fe2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "69c1cf0b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample1</th>\n",
       "      <th>Barcode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>24899</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr16</td>\n",
       "      <td>61216</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_KI270762v1_alt</td>\n",
       "      <td>69758</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233294415</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>chr2</td>\n",
       "      <td>233671871</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>chr1</td>\n",
       "      <td>236900958</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238150925</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1248</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238435902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1249 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    CHROM        POS ID REF ALT QUAL FILTER  \\\n",
       "0                    chr9      24899  .   G   A    .   PASS   \n",
       "1     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "2     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "3                   chr16      61216  .   G   C    .   PASS   \n",
       "4     chr1_KI270762v1_alt      69758  .   G   A    .   PASS   \n",
       "...                   ...        ... ..  ..  ..  ...    ...   \n",
       "1244                 chr1  233294415  .   T   C    .   PASS   \n",
       "1245                 chr2  233671871  .   C   T    .   PASS   \n",
       "1246                 chr1  236900958  .   A   G    .   PASS   \n",
       "1247                 chr2  238150925  .   A   C    .   PASS   \n",
       "1248                 chr2  238435902  .   A   G    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...   \n",
       "1     ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "2     ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "3     ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "4     ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "...                                                 ...   \n",
       "1244  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "1245  ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "1246  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...   \n",
       "1247  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "1248  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "1244  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1245  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1246  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1247  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1248  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                Sample1      Barcode  \n",
       "0     0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...  KHCNCGPCSP3  \n",
       "1     0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...  KHCNCGPCSP1  \n",
       "2       0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8  KHCNCGPCSP2  \n",
       "3     0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...  KHCNCGPCSP2  \n",
       "4     0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...  KHCNCGPCSP4  \n",
       "...                                                 ...          ...  \n",
       "1244   0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1  KHCNCGPCSP4  \n",
       "1245  0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...  KHCNCGPCSP4  \n",
       "1246  0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...  KHCNCGPCSP3  \n",
       "1247  0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...  KHCNCGPCSP1  \n",
       "1248  0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...  KHCNCGPCSP2  \n",
       "\n",
       "[1249 rows x 11 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Epilepsy_cases_novel_variants.xlsx')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "0cd023d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample1</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>CSQ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>24899</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|MIR1302-9HG|E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr16</td>\n",
       "      <td>61216</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>C|missense_variant|MODERATE|RHBDF1|ENSG0000000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_KI270762v1_alt</td>\n",
       "      <td>69758</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>A|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233294415</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>C|intron_variant|MODIFIER|PCNX2|ENSG0000013574...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>chr2</td>\n",
       "      <td>233671871</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>T|intron_variant|MODIFIER|UGT1A10|ENSG00000242...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>chr1</td>\n",
       "      <td>236900958</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|MTR|ENSG000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238150925</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td>C|missense_variant|MODERATE|KLHL30|ENSG0000016...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1248</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238435902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>G|missense_variant|MODERATE|ASB1|ENSG000000658...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1249 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    CHROM        POS ID REF ALT QUAL FILTER  \\\n",
       "0                    chr9      24899  .   G   A    .   PASS   \n",
       "1     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "2     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "3                   chr16      61216  .   G   C    .   PASS   \n",
       "4     chr1_KI270762v1_alt      69758  .   G   A    .   PASS   \n",
       "...                   ...        ... ..  ..  ..  ...    ...   \n",
       "1244                 chr1  233294415  .   T   C    .   PASS   \n",
       "1245                 chr2  233671871  .   C   T    .   PASS   \n",
       "1246                 chr1  236900958  .   A   G    .   PASS   \n",
       "1247                 chr2  238150925  .   A   C    .   PASS   \n",
       "1248                 chr2  238435902  .   A   G    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...   \n",
       "1     ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "2     ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "3     ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "4     ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "...                                                 ...   \n",
       "1244  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "1245  ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "1246  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...   \n",
       "1247  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "1248  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "1244  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1245  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1246  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1247  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1248  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                Sample1      Barcode  \\\n",
       "0     0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...  KHCNCGPCSP3   \n",
       "1     0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...  KHCNCGPCSP1   \n",
       "2       0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8  KHCNCGPCSP2   \n",
       "3     0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...  KHCNCGPCSP2   \n",
       "4     0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...  KHCNCGPCSP4   \n",
       "...                                                 ...          ...   \n",
       "1244   0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1  KHCNCGPCSP4   \n",
       "1245  0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...  KHCNCGPCSP4   \n",
       "1246  0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...  KHCNCGPCSP3   \n",
       "1247  0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...  KHCNCGPCSP1   \n",
       "1248  0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...  KHCNCGPCSP2   \n",
       "\n",
       "                                                    CSQ  \n",
       "0     A|upstream_gene_variant|MODIFIER|MIR1302-9HG|E...  \n",
       "1     T|intergenic_variant|MODIFIER|||||||||||||||||...  \n",
       "2     T|intergenic_variant|MODIFIER|||||||||||||||||...  \n",
       "3     C|missense_variant|MODERATE|RHBDF1|ENSG0000000...  \n",
       "4     A|intergenic_variant|MODIFIER|||||||||||||||||...  \n",
       "...                                                 ...  \n",
       "1244  C|intron_variant|MODIFIER|PCNX2|ENSG0000013574...  \n",
       "1245  T|intron_variant|MODIFIER|UGT1A10|ENSG00000242...  \n",
       "1246  G|downstream_gene_variant|MODIFIER|MTR|ENSG000...  \n",
       "1247  C|missense_variant|MODERATE|KLHL30|ENSG0000016...  \n",
       "1248  G|missense_variant|MODERATE|ASB1|ENSG000000658...  \n",
       "\n",
       "[1249 rows x 12 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "555b5a67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample1</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>24899</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|MIR1302-9HG|E...</td>\n",
       "      <td>MIR1302-9HG,WASHC1,WASHC1,WASHC1,WASHC1,WASHC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr16</td>\n",
       "      <td>61216</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>C|missense_variant|MODERATE|RHBDF1|ENSG0000000...</td>\n",
       "      <td>RHBDF1,SNRNP25,SNRNP25,SNRNP25,RHBDF1,SNRNP25,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_KI270762v1_alt</td>\n",
       "      <td>69758</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>A|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233294415</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>C|intron_variant|MODIFIER|PCNX2|ENSG0000013574...</td>\n",
       "      <td>PCNX2,RPS7P3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>chr2</td>\n",
       "      <td>233671871</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>T|intron_variant|MODIFIER|UGT1A10|ENSG00000242...</td>\n",
       "      <td>UGT1A10,UGT1A9,UGT1A10,UGT1A8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>chr1</td>\n",
       "      <td>236900958</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|MTR|ENSG000...</td>\n",
       "      <td>MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238150925</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td>C|missense_variant|MODERATE|KLHL30|ENSG0000016...</td>\n",
       "      <td>KLHL30,KLHL30-AS1,KLHL30-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1248</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238435902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>G|missense_variant|MODERATE|ASB1|ENSG000000658...</td>\n",
       "      <td>ASB1,ASB1,ASB1,ASB1,ASB1,ASB1,ASB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1249 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    CHROM        POS ID REF ALT QUAL FILTER  \\\n",
       "0                    chr9      24899  .   G   A    .   PASS   \n",
       "1     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "2     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "3                   chr16      61216  .   G   C    .   PASS   \n",
       "4     chr1_KI270762v1_alt      69758  .   G   A    .   PASS   \n",
       "...                   ...        ... ..  ..  ..  ...    ...   \n",
       "1244                 chr1  233294415  .   T   C    .   PASS   \n",
       "1245                 chr2  233671871  .   C   T    .   PASS   \n",
       "1246                 chr1  236900958  .   A   G    .   PASS   \n",
       "1247                 chr2  238150925  .   A   C    .   PASS   \n",
       "1248                 chr2  238435902  .   A   G    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...   \n",
       "1     ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "2     ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "3     ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "4     ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "...                                                 ...   \n",
       "1244  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "1245  ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "1246  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...   \n",
       "1247  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "1248  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "1244  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1245  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1246  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1247  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1248  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                Sample1      Barcode  \\\n",
       "0     0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...  KHCNCGPCSP3   \n",
       "1     0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...  KHCNCGPCSP1   \n",
       "2       0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8  KHCNCGPCSP2   \n",
       "3     0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...  KHCNCGPCSP2   \n",
       "4     0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...  KHCNCGPCSP4   \n",
       "...                                                 ...          ...   \n",
       "1244   0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1  KHCNCGPCSP4   \n",
       "1245  0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...  KHCNCGPCSP4   \n",
       "1246  0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...  KHCNCGPCSP3   \n",
       "1247  0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...  KHCNCGPCSP1   \n",
       "1248  0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...  KHCNCGPCSP2   \n",
       "\n",
       "                                                    CSQ  \\\n",
       "0     A|upstream_gene_variant|MODIFIER|MIR1302-9HG|E...   \n",
       "1     T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "2     T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "3     C|missense_variant|MODERATE|RHBDF1|ENSG0000000...   \n",
       "4     A|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "...                                                 ...   \n",
       "1244  C|intron_variant|MODIFIER|PCNX2|ENSG0000013574...   \n",
       "1245  T|intron_variant|MODIFIER|UGT1A10|ENSG00000242...   \n",
       "1246  G|downstream_gene_variant|MODIFIER|MTR|ENSG000...   \n",
       "1247  C|missense_variant|MODERATE|KLHL30|ENSG0000016...   \n",
       "1248  G|missense_variant|MODERATE|ASB1|ENSG000000658...   \n",
       "\n",
       "                                                   Gene  \n",
       "0        MIR1302-9HG,WASHC1,WASHC1,WASHC1,WASHC1,WASHC1  \n",
       "1                                                        \n",
       "2                                                        \n",
       "3     RHBDF1,SNRNP25,SNRNP25,SNRNP25,RHBDF1,SNRNP25,...  \n",
       "4                                                        \n",
       "...                                                 ...  \n",
       "1244                                       PCNX2,RPS7P3  \n",
       "1245                      UGT1A10,UGT1A9,UGT1A10,UGT1A8  \n",
       "1246  MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MT...  \n",
       "1247                       KLHL30,KLHL30-AS1,KLHL30-AS1  \n",
       "1248                 ASB1,ASB1,ASB1,ASB1,ASB1,ASB1,ASB1  \n",
       "\n",
       "[1249 rows x 13 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_fourth_position(row):\n",
    "    elements = row.split(',')\n",
    "    fourth_element = ''\n",
    "    for item in elements:\n",
    "        sub_elements = item.split('|')\n",
    "        if len(sub_elements) >= 1:\n",
    "            fourth_element += sub_elements[3] + ','\n",
    "    return fourth_element.rstrip(',')\n",
    "\n",
    "vcf['Gene'] = vcf['CSQ'].apply(get_fourth_position)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e23ec336",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'T|intergenic_variant|MODIFIER|||||||||||||||||||||||||'"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.CSQ.iloc[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "6c1c7cca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample1</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>Gene</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>24899</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|MIR1302-9HG|E...</td>\n",
       "      <td>MIR1302-9HG,WASHC1,WASHC1,WASHC1,WASHC1,WASHC1</td>\n",
       "      <td>MIR1302-9HG,WASHC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr16</td>\n",
       "      <td>61216</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>C|missense_variant|MODERATE|RHBDF1|ENSG0000000...</td>\n",
       "      <td>RHBDF1,SNRNP25,SNRNP25,SNRNP25,RHBDF1,SNRNP25,...</td>\n",
       "      <td>RHBDF1,SNRNP25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_KI270762v1_alt</td>\n",
       "      <td>69758</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>A|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233294415</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>C|intron_variant|MODIFIER|PCNX2|ENSG0000013574...</td>\n",
       "      <td>PCNX2,RPS7P3</td>\n",
       "      <td>PCNX2,RPS7P3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>chr2</td>\n",
       "      <td>233671871</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>T|intron_variant|MODIFIER|UGT1A10|ENSG00000242...</td>\n",
       "      <td>UGT1A10,UGT1A9,UGT1A10,UGT1A8</td>\n",
       "      <td>UGT1A10,UGT1A8,UGT1A9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>chr1</td>\n",
       "      <td>236900958</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|MTR|ENSG000...</td>\n",
       "      <td>MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MT...</td>\n",
       "      <td>MTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238150925</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td>C|missense_variant|MODERATE|KLHL30|ENSG0000016...</td>\n",
       "      <td>KLHL30,KLHL30-AS1,KLHL30-AS1</td>\n",
       "      <td>KLHL30,KLHL30-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1248</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238435902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>G|missense_variant|MODERATE|ASB1|ENSG000000658...</td>\n",
       "      <td>ASB1,ASB1,ASB1,ASB1,ASB1,ASB1,ASB1</td>\n",
       "      <td>ASB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1249 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    CHROM        POS ID REF ALT QUAL FILTER  \\\n",
       "0                    chr9      24899  .   G   A    .   PASS   \n",
       "1     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "2     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "3                   chr16      61216  .   G   C    .   PASS   \n",
       "4     chr1_KI270762v1_alt      69758  .   G   A    .   PASS   \n",
       "...                   ...        ... ..  ..  ..  ...    ...   \n",
       "1244                 chr1  233294415  .   T   C    .   PASS   \n",
       "1245                 chr2  233671871  .   C   T    .   PASS   \n",
       "1246                 chr1  236900958  .   A   G    .   PASS   \n",
       "1247                 chr2  238150925  .   A   C    .   PASS   \n",
       "1248                 chr2  238435902  .   A   G    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...   \n",
       "1     ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "2     ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "3     ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "4     ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "...                                                 ...   \n",
       "1244  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "1245  ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "1246  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...   \n",
       "1247  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "1248  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "1244  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1245  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1246  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1247  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1248  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                Sample1      Barcode  \\\n",
       "0     0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...  KHCNCGPCSP3   \n",
       "1     0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...  KHCNCGPCSP1   \n",
       "2       0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8  KHCNCGPCSP2   \n",
       "3     0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...  KHCNCGPCSP2   \n",
       "4     0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...  KHCNCGPCSP4   \n",
       "...                                                 ...          ...   \n",
       "1244   0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1  KHCNCGPCSP4   \n",
       "1245  0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...  KHCNCGPCSP4   \n",
       "1246  0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...  KHCNCGPCSP3   \n",
       "1247  0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...  KHCNCGPCSP1   \n",
       "1248  0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...  KHCNCGPCSP2   \n",
       "\n",
       "                                                    CSQ  \\\n",
       "0     A|upstream_gene_variant|MODIFIER|MIR1302-9HG|E...   \n",
       "1     T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "2     T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "3     C|missense_variant|MODERATE|RHBDF1|ENSG0000000...   \n",
       "4     A|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "...                                                 ...   \n",
       "1244  C|intron_variant|MODIFIER|PCNX2|ENSG0000013574...   \n",
       "1245  T|intron_variant|MODIFIER|UGT1A10|ENSG00000242...   \n",
       "1246  G|downstream_gene_variant|MODIFIER|MTR|ENSG000...   \n",
       "1247  C|missense_variant|MODERATE|KLHL30|ENSG0000016...   \n",
       "1248  G|missense_variant|MODERATE|ASB1|ENSG000000658...   \n",
       "\n",
       "                                                   Gene                   GENE  \n",
       "0        MIR1302-9HG,WASHC1,WASHC1,WASHC1,WASHC1,WASHC1     MIR1302-9HG,WASHC1  \n",
       "1                                                                               \n",
       "2                                                                               \n",
       "3     RHBDF1,SNRNP25,SNRNP25,SNRNP25,RHBDF1,SNRNP25,...         RHBDF1,SNRNP25  \n",
       "4                                                                               \n",
       "...                                                 ...                    ...  \n",
       "1244                                       PCNX2,RPS7P3           PCNX2,RPS7P3  \n",
       "1245                      UGT1A10,UGT1A9,UGT1A10,UGT1A8  UGT1A10,UGT1A8,UGT1A9  \n",
       "1246  MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MTR,MT...                    MTR  \n",
       "1247                       KLHL30,KLHL30-AS1,KLHL30-AS1      KLHL30,KLHL30-AS1  \n",
       "1248                 ASB1,ASB1,ASB1,ASB1,ASB1,ASB1,ASB1                   ASB1  \n",
       "\n",
       "[1249 rows x 14 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['GENE'] = vcf['Gene'].apply(lambda x: ','.join(sorted(set(x.split(',')))))\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "ead51363",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT',\n",
       "       'Sample1', 'Barcode', 'CSQ', 'Gene', 'GENE'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ec08863a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Sample1</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>24899</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>MIR1302-9HG,WASHC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr16</td>\n",
       "      <td>61216</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>RHBDF1,SNRNP25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_KI270762v1_alt</td>\n",
       "      <td>69758</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233294415</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>PCNX2,RPS7P3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>chr2</td>\n",
       "      <td>233671871</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...</td>\n",
       "      <td>KHCNCGPCSP4</td>\n",
       "      <td>UGT1A10,UGT1A8,UGT1A9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>chr1</td>\n",
       "      <td>236900958</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...</td>\n",
       "      <td>KHCNCGPCSP3</td>\n",
       "      <td>MTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238150925</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...</td>\n",
       "      <td>KHCNCGPCSP1</td>\n",
       "      <td>KLHL30,KLHL30-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1248</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238435902</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...</td>\n",
       "      <td>KHCNCGPCSP2</td>\n",
       "      <td>ASB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1249 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    CHROM        POS ID REF ALT QUAL FILTER  \\\n",
       "0                    chr9      24899  .   G   A    .   PASS   \n",
       "1     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "2     chr2_KI270894v1_alt      41636  .   C   T    .   PASS   \n",
       "3                   chr16      61216  .   G   C    .   PASS   \n",
       "4     chr1_KI270762v1_alt      69758  .   G   A    .   PASS   \n",
       "...                   ...        ... ..  ..  ..  ...    ...   \n",
       "1244                 chr1  233294415  .   T   C    .   PASS   \n",
       "1245                 chr2  233671871  .   C   T    .   PASS   \n",
       "1246                 chr1  236900958  .   A   G    .   PASS   \n",
       "1247                 chr2  238150925  .   A   C    .   PASS   \n",
       "1248                 chr2  238435902  .   A   G    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=95;WT=0;HET=1;HOM=0;NC=0;CSQ=A|upstream_ge...   \n",
       "1     ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "2     ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "3     ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "4     ADP=40;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "...                                                 ...   \n",
       "1244  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "1245  ADP=32;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "1246  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|downstream_...   \n",
       "1247  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "1248  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "1244  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1245  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1246  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1247  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1248  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                Sample1      Barcode  \\\n",
       "0     0/1:65:95:95:75:20:21.05%:3.1216E-7:56:52:47:2...  KHCNCGPCSP3   \n",
       "1     0/1:140:62:62:26:36:58.06%:9.6098E-15:58:63:19...  KHCNCGPCSP1   \n",
       "2       0/1:48:75:75:60:15:20%:1.404E-5:55:46:41:19:7:8  KHCNCGPCSP2   \n",
       "3     0/1:63:33:33:16:17:51.52%:4.6376E-7:52:51:9:7:...  KHCNCGPCSP2   \n",
       "4     0/1:44:40:40:27:13:32.5%:3.8184E-5:56:63:20:7:...  KHCNCGPCSP4   \n",
       "...                                                 ...          ...   \n",
       "1244   0/1:48:25:25:12:13:52%:1.4654E-5:51:44:11:1:12:1  KHCNCGPCSP4   \n",
       "1245  0/1:45:32:32:19:13:40.62%:2.6443E-5:47:49:15:4...  KHCNCGPCSP4   \n",
       "1246  0/1:132:52:52:19:33:63.46%:5.4755E-14:59:48:15...  KHCNCGPCSP3   \n",
       "1247  0/1:144:52:52:17:35:67.31%:3.7678E-15:52:59:13...  KHCNCGPCSP1   \n",
       "1248  0/1:154:58:58:20:38:65.52%:3.2245E-16:49:48:11...  KHCNCGPCSP2   \n",
       "\n",
       "                       GENE  \n",
       "0        MIR1302-9HG,WASHC1  \n",
       "1                            \n",
       "2                            \n",
       "3            RHBDF1,SNRNP25  \n",
       "4                            \n",
       "...                     ...  \n",
       "1244           PCNX2,RPS7P3  \n",
       "1245  UGT1A10,UGT1A8,UGT1A9  \n",
       "1246                    MTR  \n",
       "1247      KLHL30,KLHL30-AS1  \n",
       "1248                   ASB1  \n",
       "\n",
       "[1249 rows x 12 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = vcf[['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT',\n",
    "       'Sample1', 'Barcode', 'GENE']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "2e98571d",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Epilepsy_cases_novel_variants_new.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fdef148a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69568737</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>40</td>\n",
       "      <td>22.5%</td>\n",
       "      <td>LRRC7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>103688940</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>28</td>\n",
       "      <td>46.43%</td>\n",
       "      <td>AMY1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>149860814</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>142</td>\n",
       "      <td>141</td>\n",
       "      <td>29.79%</td>\n",
       "      <td>H4C15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217438</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217444</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>chrX</td>\n",
       "      <td>102366132</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>69</td>\n",
       "      <td>23.19%</td>\n",
       "      <td>NXF2B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265039</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265043</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265051</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>99</td>\n",
       "      <td>87</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13470222</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1/1</td>\n",
       "      <td>76</td>\n",
       "      <td>14</td>\n",
       "      <td>100%</td>\n",
       "      <td>UTY</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS ID REF ALT FILTER   GT   GQ   DP    FREQ          GENE\n",
       "0    chr1   69568737  .   C   G   PASS  0/1   29   40   22.5%         LRRC7\n",
       "1    chr1  103688940  .   T   C   PASS  0/1   47   28  46.43%         AMY1B\n",
       "2    chr1  149860814  .   G   T   PASS  0/1  142  141  29.79%         H4C15\n",
       "3    chr1  152217438  .   C   G   PASS  0/1   32   11  72.73%  FLG-AS1,HRNR\n",
       "4    chr1  152217444  .   T   G   PASS  0/1   32   15  53.33%  FLG-AS1,HRNR\n",
       "..    ...        ... ..  ..  ..    ...  ...  ...  ...     ...           ...\n",
       "225  chrX  102366132  .   G   A   PASS  0/1   52   69  23.19%         NXF2B\n",
       "226  chrX  154265039  .   A   G   PASS  0/1   92   83  32.53%       OPN1MW3\n",
       "227  chrX  154265043  .   A   C   PASS  0/1   92   83  32.53%       OPN1MW3\n",
       "228  chrX  154265051  .   C   G   PASS  0/1   99   87  33.33%       OPN1MW3\n",
       "229  chrY   13470222  .   C   T   PASS  1/1   76   14    100%           UTY\n",
       "\n",
       "[230 rows x 11 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_cols = vcf['SAMPLE1'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'ID', 'REF', 'ALT', 'FILTER', 'GT', 'GQ', 'DP', 'FREQ', 'GENE']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "474f5906",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>GENE</th>\n",
       "      <th>dbsnp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69568737</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>40</td>\n",
       "      <td>22.5%</td>\n",
       "      <td>LRRC7</td>\n",
       "      <td>chr1_69568737_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>103688940</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>28</td>\n",
       "      <td>46.43%</td>\n",
       "      <td>AMY1B</td>\n",
       "      <td>chr1_103688940_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>149860814</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>142</td>\n",
       "      <td>141</td>\n",
       "      <td>29.79%</td>\n",
       "      <td>H4C15</td>\n",
       "      <td>chr1_149860814_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217438</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "      <td>chr1_152217438_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217444</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "      <td>chr1_152217444_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>chrX</td>\n",
       "      <td>102366132</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>69</td>\n",
       "      <td>23.19%</td>\n",
       "      <td>NXF2B</td>\n",
       "      <td>chrX_102366132_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265039</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265039_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265043</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265043_A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265051</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>99</td>\n",
       "      <td>87</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265051_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13470222</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1/1</td>\n",
       "      <td>76</td>\n",
       "      <td>14</td>\n",
       "      <td>100%</td>\n",
       "      <td>UTY</td>\n",
       "      <td>chrY_13470222_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS ID REF ALT FILTER   GT   GQ   DP    FREQ          GENE  \\\n",
       "0    chr1   69568737  .   C   G   PASS  0/1   29   40   22.5%         LRRC7   \n",
       "1    chr1  103688940  .   T   C   PASS  0/1   47   28  46.43%         AMY1B   \n",
       "2    chr1  149860814  .   G   T   PASS  0/1  142  141  29.79%         H4C15   \n",
       "3    chr1  152217438  .   C   G   PASS  0/1   32   11  72.73%  FLG-AS1,HRNR   \n",
       "4    chr1  152217444  .   T   G   PASS  0/1   32   15  53.33%  FLG-AS1,HRNR   \n",
       "..    ...        ... ..  ..  ..    ...  ...  ...  ...     ...           ...   \n",
       "225  chrX  102366132  .   G   A   PASS  0/1   52   69  23.19%         NXF2B   \n",
       "226  chrX  154265039  .   A   G   PASS  0/1   92   83  32.53%       OPN1MW3   \n",
       "227  chrX  154265043  .   A   C   PASS  0/1   92   83  32.53%       OPN1MW3   \n",
       "228  chrX  154265051  .   C   G   PASS  0/1   99   87  33.33%       OPN1MW3   \n",
       "229  chrY   13470222  .   C   T   PASS  1/1   76   14    100%           UTY   \n",
       "\n",
       "                  dbsnp  \n",
       "0     chr1_69568737_C_G  \n",
       "1    chr1_103688940_T_C  \n",
       "2    chr1_149860814_G_T  \n",
       "3    chr1_152217438_C_G  \n",
       "4    chr1_152217444_T_G  \n",
       "..                  ...  \n",
       "225  chrX_102366132_G_A  \n",
       "226  chrX_154265039_A_G  \n",
       "227  chrX_154265043_A_C  \n",
       "228  chrX_154265051_C_G  \n",
       "229   chrY_13470222_C_T  \n",
       "\n",
       "[230 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['dbsnp'] = vcf.apply(lambda row: '_'.join([str(row['CHROM']), str(row['POS']), str(row['REF']), str(row['ALT'])]), axis=1)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4d6b59a0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>GENE</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69568737</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>40</td>\n",
       "      <td>22.5%</td>\n",
       "      <td>LRRC7</td>\n",
       "      <td>chr1_69568737_C_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>103688940</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>28</td>\n",
       "      <td>46.43%</td>\n",
       "      <td>AMY1B</td>\n",
       "      <td>chr1_103688940_T_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>149860814</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>142</td>\n",
       "      <td>141</td>\n",
       "      <td>29.79%</td>\n",
       "      <td>H4C15</td>\n",
       "      <td>chr1_149860814_G_T</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217438</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "      <td>chr1_152217438_C_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217444</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "      <td>chr1_152217444_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>chrX</td>\n",
       "      <td>102366132</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>69</td>\n",
       "      <td>23.19%</td>\n",
       "      <td>NXF2B</td>\n",
       "      <td>chrX_102366132_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265039</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265039_A_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265043</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265043_A_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265051</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>99</td>\n",
       "      <td>87</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265051_C_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13470222</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1/1</td>\n",
       "      <td>76</td>\n",
       "      <td>14</td>\n",
       "      <td>100%</td>\n",
       "      <td>UTY</td>\n",
       "      <td>chrY_13470222_C_T</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS ID REF ALT FILTER   GT   GQ   DP    FREQ          GENE  \\\n",
       "0    chr1   69568737  .   C   G   PASS  0/1   29   40   22.5%         LRRC7   \n",
       "1    chr1  103688940  .   T   C   PASS  0/1   47   28  46.43%         AMY1B   \n",
       "2    chr1  149860814  .   G   T   PASS  0/1  142  141  29.79%         H4C15   \n",
       "3    chr1  152217438  .   C   G   PASS  0/1   32   11  72.73%  FLG-AS1,HRNR   \n",
       "4    chr1  152217444  .   T   G   PASS  0/1   32   15  53.33%  FLG-AS1,HRNR   \n",
       "..    ...        ... ..  ..  ..    ...  ...  ...  ...     ...           ...   \n",
       "225  chrX  102366132  .   G   A   PASS  0/1   52   69  23.19%         NXF2B   \n",
       "226  chrX  154265039  .   A   G   PASS  0/1   92   83  32.53%       OPN1MW3   \n",
       "227  chrX  154265043  .   A   C   PASS  0/1   92   83  32.53%       OPN1MW3   \n",
       "228  chrX  154265051  .   C   G   PASS  0/1   99   87  33.33%       OPN1MW3   \n",
       "229  chrY   13470222  .   C   T   PASS  1/1   76   14    100%           UTY   \n",
       "\n",
       "                  dbsnp      Zygosity  \n",
       "0     chr1_69568737_C_G  Heterozygous  \n",
       "1    chr1_103688940_T_C  Heterozygous  \n",
       "2    chr1_149860814_G_T  Heterozygous  \n",
       "3    chr1_152217438_C_G  Heterozygous  \n",
       "4    chr1_152217444_T_G  Heterozygous  \n",
       "..                  ...           ...  \n",
       "225  chrX_102366132_G_A  Heterozygous  \n",
       "226  chrX_154265039_A_G  Heterozygous  \n",
       "227  chrX_154265043_A_C  Heterozygous  \n",
       "228  chrX_154265051_C_G  Heterozygous  \n",
       "229   chrY_13470222_C_T    Homozygous  \n",
       "\n",
       "[230 rows x 13 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['GT'] == '1/1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['GT'] == '0/1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d2df04cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/variants_ADHD_31_noval_format.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "882e07e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>GENE</th>\n",
       "      <th>dbsnp</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>new_column</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69568737</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>40</td>\n",
       "      <td>22.5%</td>\n",
       "      <td>LRRC7</td>\n",
       "      <td>chr1_69568737_C_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chr1-69568737_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>103688940</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>28</td>\n",
       "      <td>46.43%</td>\n",
       "      <td>AMY1B</td>\n",
       "      <td>chr1_103688940_T_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chr1-103688940_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>149860814</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>142</td>\n",
       "      <td>141</td>\n",
       "      <td>29.79%</td>\n",
       "      <td>H4C15</td>\n",
       "      <td>chr1_149860814_G_T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chr1-149860814_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217438</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "      <td>chr1_152217438_C_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chr1-152217438_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152217444</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>FLG-AS1,HRNR</td>\n",
       "      <td>chr1_152217444_T_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chr1-152217444_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>chrX</td>\n",
       "      <td>102366132</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>69</td>\n",
       "      <td>23.19%</td>\n",
       "      <td>NXF2B</td>\n",
       "      <td>chrX_102366132_G_A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chrX-102366132_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265039</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265039_A_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chrX-154265039_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265043</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>92</td>\n",
       "      <td>83</td>\n",
       "      <td>32.53%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265043_A_C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chrX-154265043_A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154265051</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0/1</td>\n",
       "      <td>99</td>\n",
       "      <td>87</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>OPN1MW3</td>\n",
       "      <td>chrX_154265051_C_G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>chrX-154265051_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13470222</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1/1</td>\n",
       "      <td>76</td>\n",
       "      <td>14</td>\n",
       "      <td>100%</td>\n",
       "      <td>UTY</td>\n",
       "      <td>chrY_13470222_C_T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>chrY-13470222_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS ID REF ALT FILTER   GT   GQ   DP    FREQ          GENE  \\\n",
       "0    chr1   69568737  .   C   G   PASS  0/1   29   40   22.5%         LRRC7   \n",
       "1    chr1  103688940  .   T   C   PASS  0/1   47   28  46.43%         AMY1B   \n",
       "2    chr1  149860814  .   G   T   PASS  0/1  142  141  29.79%         H4C15   \n",
       "3    chr1  152217438  .   C   G   PASS  0/1   32   11  72.73%  FLG-AS1,HRNR   \n",
       "4    chr1  152217444  .   T   G   PASS  0/1   32   15  53.33%  FLG-AS1,HRNR   \n",
       "..    ...        ... ..  ..  ..    ...  ...  ...  ...     ...           ...   \n",
       "225  chrX  102366132  .   G   A   PASS  0/1   52   69  23.19%         NXF2B   \n",
       "226  chrX  154265039  .   A   G   PASS  0/1   92   83  32.53%       OPN1MW3   \n",
       "227  chrX  154265043  .   A   C   PASS  0/1   92   83  32.53%       OPN1MW3   \n",
       "228  chrX  154265051  .   C   G   PASS  0/1   99   87  33.33%       OPN1MW3   \n",
       "229  chrY   13470222  .   C   T   PASS  1/1   76   14    100%           UTY   \n",
       "\n",
       "                  dbsnp      Zygosity          new_column  \n",
       "0     chr1_69568737_C_G  Heterozygous   chr1-69568737_C_G  \n",
       "1    chr1_103688940_T_C  Heterozygous  chr1-103688940_T_C  \n",
       "2    chr1_149860814_G_T  Heterozygous  chr1-149860814_G_T  \n",
       "3    chr1_152217438_C_G  Heterozygous  chr1-152217438_C_G  \n",
       "4    chr1_152217444_T_G  Heterozygous  chr1-152217444_T_G  \n",
       "..                  ...           ...                 ...  \n",
       "225  chrX_102366132_G_A  Heterozygous  chrX-102366132_G_A  \n",
       "226  chrX_154265039_A_G  Heterozygous  chrX-154265039_A_G  \n",
       "227  chrX_154265043_A_C  Heterozygous  chrX-154265043_A_C  \n",
       "228  chrX_154265051_C_G  Heterozygous  chrX-154265051_C_G  \n",
       "229   chrY_13470222_C_T    Homozygous   chrY-13470222_C_T  \n",
       "\n",
       "[230 rows x 14 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['new_column'] = vcf.apply(lambda row: '{}-{}_{}_{}'.format(row['CHROM'], row['POS'], row['REF'], row['ALT']), axis=1)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c07e6f8f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d82462d6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "8fdb518e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44143213</td>\n",
       "      <td>44198170</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  Start_pos   End_pos\n",
       "0       chr7   44143213  44198170"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/krishna_bed.xlsx')\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos']\n",
    "df.to_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/krishna_bed.bed', header=None, sep='\\t', index=False)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b340546",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c7dd5de7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65811</td>\n",
       "      <td>65993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69461</td>\n",
       "      <td>69620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>785981</td>\n",
       "      <td>786159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786130</td>\n",
       "      <td>786446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230714</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190028</td>\n",
       "      <td>57190328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230715</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190299</td>\n",
       "      <td>57190439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230716</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190874</td>\n",
       "      <td>57191014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230717</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191846</td>\n",
       "      <td>57192058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230718</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194102</td>\n",
       "      <td>57194243</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230719 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos\n",
       "0            chr1      65489     65645\n",
       "1            chr1      65811     65993\n",
       "2            chr1      69461     69620\n",
       "3            chr1     785981    786159\n",
       "4            chr1     786130    786446\n",
       "...           ...        ...       ...\n",
       "230714       chrY   57190028  57190328\n",
       "230715       chrY   57190299  57190439\n",
       "230716       chrY   57190874  57191014\n",
       "230717       chrY   57191846  57192058\n",
       "230718       chrY   57194102  57194243\n",
       "\n",
       "[230719 rows x 3 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/srinivas_sir_covered.bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f25ce5fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65811</td>\n",
       "      <td>65993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69461</td>\n",
       "      <td>69620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>785981</td>\n",
       "      <td>786159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786130</td>\n",
       "      <td>786446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230715</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190299</td>\n",
       "      <td>57190439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230716</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57190874</td>\n",
       "      <td>57191014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230717</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191846</td>\n",
       "      <td>57192058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230718</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194102</td>\n",
       "      <td>57194243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230719</th>\n",
       "      <td>chrM</td>\n",
       "      <td>1</td>\n",
       "      <td>16569</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230720 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos\n",
       "0            chr1      65489     65645\n",
       "1            chr1      65811     65993\n",
       "2            chr1      69461     69620\n",
       "3            chr1     785981    786159\n",
       "4            chr1     786130    786446\n",
       "...           ...        ...       ...\n",
       "230715       chrY   57190299  57190439\n",
       "230716       chrY   57190874  57191014\n",
       "230717       chrY   57191846  57192058\n",
       "230718       chrY   57194102  57194243\n",
       "230719       chrM          1     16569\n",
       "\n",
       "[230720 rows x 3 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Add a new row at the bottom\n",
    "new_row = {'chromosome': 'chrM', 'Start_pos': 1, 'End_pos': 16569}\n",
    "df = df.append(new_row, ignore_index=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "692dda32",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6052137",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "9b4850c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  \n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5  \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...  \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...  \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...  \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...  \n",
       "...                                                   ...  \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...  \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...  \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...  \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...  \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...  \n",
       "\n",
       "[208911 rows x 4 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "3dc3a880",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  Extended_Start_pos  \\\n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5               65489   \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...               69007   \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...              450710   \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...              685686   \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...              924401   \n",
       "...                                                   ...                 ...   \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25038781   \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25041746   \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25043888   \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...            25622413   \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...            25624397   \n",
       "\n",
       "        Extended_End_pos  \n",
       "0                  65649  \n",
       "1                  70037  \n",
       "2                 451706  \n",
       "3                 686682  \n",
       "4                 924977  \n",
       "...                  ...  \n",
       "208906          25038941  \n",
       "208907          25041906  \n",
       "208908          25044048  \n",
       "208909          25624093  \n",
       "208910          25624557  \n",
       "\n",
       "[208911 rows x 6 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "3ba3fc73",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65509</td>\n",
       "      <td>65629</td>\n",
       "      <td>ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69027</td>\n",
       "      <td>70017</td>\n",
       "      <td>ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450730</td>\n",
       "      <td>451686</td>\n",
       "      <td>ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685706</td>\n",
       "      <td>686662</td>\n",
       "      <td>ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924421</td>\n",
       "      <td>924957</td>\n",
       "      <td>ensembl_gene_id=ENSG00000187634;gene_symbol=SA...</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038801</td>\n",
       "      <td>25038921</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041766</td>\n",
       "      <td>25041886</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043908</td>\n",
       "      <td>25044028</td>\n",
       "      <td>ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "      <td>BPY2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622433</td>\n",
       "      <td>25624073</td>\n",
       "      <td>ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624417</td>\n",
       "      <td>25624537</td>\n",
       "      <td>ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "      <td>CDY1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos  \\\n",
       "0            chr1      65509     65629   \n",
       "1            chr1      69027     70017   \n",
       "2            chr1     450730    451686   \n",
       "3            chr1     685706    686662   \n",
       "4            chr1     924421    924957   \n",
       "...           ...        ...       ...   \n",
       "208906       chrY   25038801  25038921   \n",
       "208907       chrY   25041766  25041886   \n",
       "208908       chrY   25043908  25044028   \n",
       "208909       chrY   25622433  25624073   \n",
       "208910       chrY   25624417  25624537   \n",
       "\n",
       "                                                     INFO  Extended_Start_pos  \\\n",
       "0       ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5               65489   \n",
       "1       ccds_id=CCDS30547.1;ensembl_gene_id=ENSG000001...               69007   \n",
       "2       ccds_id=CCDS72675.1;ensembl_gene_id=ENSG000002...              450710   \n",
       "3       ccds_id=CCDS41221.1;ensembl_gene_id=ENSG000002...              685686   \n",
       "4       ensembl_gene_id=ENSG00000187634;gene_symbol=SA...              924401   \n",
       "...                                                   ...                 ...   \n",
       "208906  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25038781   \n",
       "208907  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25041746   \n",
       "208908  ccds_id=CCDS44030.1;ensembl_gene_id=ENSG000001...            25043888   \n",
       "208909  ccds_id=CCDS14801.1,CCDS14802.1;ensembl_gene_i...            25622413   \n",
       "208910  ccds_id=CCDS14801.1;ensembl_gene_id=ENSG000001...            25624397   \n",
       "\n",
       "        Extended_End_pos    Gene  \n",
       "0                  65649   OR4F5  \n",
       "1                  70037   OR4F5  \n",
       "2                 451706  OR4F29  \n",
       "3                 686682  OR4F16  \n",
       "4                 924977  SAMD11  \n",
       "...                  ...     ...  \n",
       "208906          25038941   BPY2C  \n",
       "208907          25041906   BPY2C  \n",
       "208908          25044048   BPY2C  \n",
       "208909          25624093    CDY1  \n",
       "208910          25624557    CDY1  \n",
       "\n",
       "[208911 rows x 7 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Gene'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df['Gene'] = df['Gene'].str.split(',').str[0]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "028943f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80561</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44144154</td>\n",
       "      <td>44144314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80562</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44145106</td>\n",
       "      <td>44145308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80563</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44145464</td>\n",
       "      <td>44145768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80564</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44146028</td>\n",
       "      <td>44146188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80565</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44146430</td>\n",
       "      <td>44146654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80566</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44147616</td>\n",
       "      <td>44147862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80567</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44149727</td>\n",
       "      <td>44150093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80568</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44150931</td>\n",
       "      <td>44151091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80569</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44152235</td>\n",
       "      <td>44152461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80570</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44153255</td>\n",
       "      <td>44153503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80571</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44158013</td>\n",
       "      <td>44158173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80572</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44159015</td>\n",
       "      <td>44159175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80573</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44188858</td>\n",
       "      <td>44189018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80574</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44188990</td>\n",
       "      <td>44189150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80575</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44189342</td>\n",
       "      <td>44189502</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  Extended_Start_pos  Extended_End_pos\n",
       "80561       chr7            44144154          44144314\n",
       "80562       chr7            44145106          44145308\n",
       "80563       chr7            44145464          44145768\n",
       "80564       chr7            44146028          44146188\n",
       "80565       chr7            44146430          44146654\n",
       "80566       chr7            44147616          44147862\n",
       "80567       chr7            44149727          44150093\n",
       "80568       chr7            44150931          44151091\n",
       "80569       chr7            44152235          44152461\n",
       "80570       chr7            44153255          44153503\n",
       "80571       chr7            44158013          44158173\n",
       "80572       chr7            44159015          44159175\n",
       "80573       chr7            44188858          44189018\n",
       "80574       chr7            44188990          44189150\n",
       "80575       chr7            44189342          44189502"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = df[df['Gene'] == 'GCK']\n",
    "data = data[['chromosome', 'Extended_Start_pos', 'Extended_End_pos']]\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "2a461ffd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80561</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44144154</td>\n",
       "      <td>44144314</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80562</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44145106</td>\n",
       "      <td>44145308</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80563</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44145464</td>\n",
       "      <td>44145768</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80564</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44146028</td>\n",
       "      <td>44146188</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80565</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44146430</td>\n",
       "      <td>44146654</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80566</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44147616</td>\n",
       "      <td>44147862</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80567</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44149727</td>\n",
       "      <td>44150093</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80568</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44150931</td>\n",
       "      <td>44151091</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80569</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44152235</td>\n",
       "      <td>44152461</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80570</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44153255</td>\n",
       "      <td>44153503</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80571</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44158013</td>\n",
       "      <td>44158173</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80572</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44159015</td>\n",
       "      <td>44159175</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80573</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44188858</td>\n",
       "      <td>44189018</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80574</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44188990</td>\n",
       "      <td>44189150</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80575</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44189342</td>\n",
       "      <td>44189502</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  Extended_Start_pos  Extended_End_pos coverage\n",
       "80561       chr7            44144154          44144314  covered\n",
       "80562       chr7            44145106          44145308  covered\n",
       "80563       chr7            44145464          44145768  covered\n",
       "80564       chr7            44146028          44146188  covered\n",
       "80565       chr7            44146430          44146654  covered\n",
       "80566       chr7            44147616          44147862  covered\n",
       "80567       chr7            44149727          44150093  covered\n",
       "80568       chr7            44150931          44151091  covered\n",
       "80569       chr7            44152235          44152461  covered\n",
       "80570       chr7            44153255          44153503  covered\n",
       "80571       chr7            44158013          44158173  covered\n",
       "80572       chr7            44159015          44159175  covered\n",
       "80573       chr7            44188858          44189018  covered\n",
       "80574       chr7            44188990          44189150  covered\n",
       "80575       chr7            44189342          44189502  covered"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Function to check if a row satisfies the conditions\n",
    "def check_coverage(row):\n",
    "    if row['chromosome'] == 'chr7' and 44143213 <= row['Extended_Start_pos'] <= 44198170 and 44143213 <= row['Extended_End_pos'] <= 44198170:\n",
    "        return 'covered'\n",
    "    else:\n",
    "        return 'not_covered'\n",
    "\n",
    "# Apply the function to create a new 'coverage' column\n",
    "data['coverage'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "1d0aada4",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/GCK_covered_positions.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb9b870f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "60941384",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Extended_Start_pos  Extended_End_pos\n",
       "0            chr1               65489             65649\n",
       "1            chr1               69007             70037\n",
       "2            chr1              450710            451706\n",
       "3            chr1              685686            686682\n",
       "4            chr1              924401            924977\n",
       "...           ...                 ...               ...\n",
       "208906       chrY            25038781          25038941\n",
       "208907       chrY            25041746          25041906\n",
       "208908       chrY            25043888          25044048\n",
       "208909       chrY            25622413          25624093\n",
       "208910       chrY            25624397          25624557\n",
       "\n",
       "[208911 rows x 3 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos']]\n",
    "#new_row = {'chromosome': 'chrM', 'Extended_Start_pos': 1, 'Extended_End_pos': 16569}\n",
    "\n",
    "# Append the new row to the DataFrame\n",
    "#df = df.append(new_row, ignore_index=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14c448c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "44149722"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "db1f9cb6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chromosome, Start_pos, End_pos]\n",
       "Index: []"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Value to check\n",
    "value = 44146159\n",
    "\n",
    "# Filter rows where the value lies between start and end positions\n",
    "filtered_df = df[(df['Start_pos'] <= value) & (df['End_pos'] >= value)]\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "9d9bd607",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>DP</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1091327</td>\n",
       "      <td>rs4074137</td>\n",
       "      <td>22</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1676661</td>\n",
       "      <td>rs770523611</td>\n",
       "      <td>36</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1734736</td>\n",
       "      <td>rs61777509</td>\n",
       "      <td>55</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7830024</td>\n",
       "      <td>rs1357370388</td>\n",
       "      <td>56</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11098048</td>\n",
       "      <td>rs62623443</td>\n",
       "      <td>18</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>chrX</td>\n",
       "      <td>141589872</td>\n",
       "      <td>rs1397854713</td>\n",
       "      <td>33</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>chrX</td>\n",
       "      <td>141698301</td>\n",
       "      <td>rs782600137</td>\n",
       "      <td>40</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>chrX</td>\n",
       "      <td>141906704</td>\n",
       "      <td>rs59853306</td>\n",
       "      <td>90</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153807864</td>\n",
       "      <td>rs58306331</td>\n",
       "      <td>23</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>618</th>\n",
       "      <td>chrY</td>\n",
       "      <td>22513968</td>\n",
       "      <td>rs1189368309</td>\n",
       "      <td>23</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>619 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS          rsID  DP REF ALT\n",
       "0    chr1    1091327     rs4074137  22   C   A\n",
       "1    chr1    1676661   rs770523611  36   C   G\n",
       "2    chr1    1734736    rs61777509  55   C   T\n",
       "3    chr1    7830024  rs1357370388  56   A   C\n",
       "4    chr1   11098048    rs62623443  18   C   T\n",
       "..    ...        ...           ...  ..  ..  ..\n",
       "614  chrX  141589872  rs1397854713  33   G   A\n",
       "615  chrX  141698301   rs782600137  40   C   T\n",
       "616  chrX  141906704    rs59853306  90   G   A\n",
       "617  chrX  153807864    rs58306331  23   G   A\n",
       "618  chrY   22513968  rs1189368309  23   T   C\n",
       "\n",
       "[619 rows x 6 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/sainee/sample_4.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9fb4668f",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'POS'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3802\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3801\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:138\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:165\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5745\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5753\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'POS'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[12], line 23\u001b[0m\n\u001b[0;32m     20\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNot_Covered\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m     22\u001b[0m \u001b[38;5;66;03m# Step 3: Apply the function to create the new column in data\u001b[39;00m\n\u001b[1;32m---> 23\u001b[0m data[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCovered/Not_Covered\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mdata\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcheck_coverage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m     24\u001b[0m data\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py:9568\u001b[0m, in \u001b[0;36mDataFrame.apply\u001b[1;34m(self, func, axis, raw, result_type, args, **kwargs)\u001b[0m\n\u001b[0;32m   9557\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcore\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mapply\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m frame_apply\n\u001b[0;32m   9559\u001b[0m op \u001b[38;5;241m=\u001b[39m frame_apply(\n\u001b[0;32m   9560\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m   9561\u001b[0m     func\u001b[38;5;241m=\u001b[39mfunc,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   9566\u001b[0m     kwargs\u001b[38;5;241m=\u001b[39mkwargs,\n\u001b[0;32m   9567\u001b[0m )\n\u001b[1;32m-> 9568\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mop\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39m__finalize__(\u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mapply\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:764\u001b[0m, in \u001b[0;36mFrameApply.apply\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    761\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mraw:\n\u001b[0;32m    762\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mapply_raw()\n\u001b[1;32m--> 764\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply_standard\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:891\u001b[0m, in \u001b[0;36mFrameApply.apply_standard\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    890\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mapply_standard\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[1;32m--> 891\u001b[0m     results, res_index \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply_series_generator\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    893\u001b[0m     \u001b[38;5;66;03m# wrap results\u001b[39;00m\n\u001b[0;32m    894\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mwrap_results(results, res_index)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:907\u001b[0m, in \u001b[0;36mFrameApply.apply_series_generator\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    904\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m option_context(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmode.chained_assignment\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m    905\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m i, v \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(series_gen):\n\u001b[0;32m    906\u001b[0m         \u001b[38;5;66;03m# ignore SettingWithCopy here in case the user mutates\u001b[39;00m\n\u001b[1;32m--> 907\u001b[0m         results[i] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[43mv\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    908\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(results[i], ABCSeries):\n\u001b[0;32m    909\u001b[0m             \u001b[38;5;66;03m# If we have a view on v, we need to make a copy because\u001b[39;00m\n\u001b[0;32m    910\u001b[0m             \u001b[38;5;66;03m#  series_generator will swap out the underlying data\u001b[39;00m\n\u001b[0;32m    911\u001b[0m             results[i] \u001b[38;5;241m=\u001b[39m results[i]\u001b[38;5;241m.\u001b[39mcopy(deep\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "Cell \u001b[1;32mIn[12], line 13\u001b[0m, in \u001b[0;36mcheck_coverage\u001b[1;34m(row)\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcheck_coverage\u001b[39m(row):\n\u001b[1;32m---> 13\u001b[0m     pos \u001b[38;5;241m=\u001b[39m \u001b[43mrow\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mPOS\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[0;32m     14\u001b[0m     chromosome \u001b[38;5;241m=\u001b[39m row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCHROM\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m     15\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m chromosome \u001b[38;5;129;01min\u001b[39;00m chromosome_dict:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:981\u001b[0m, in \u001b[0;36mSeries.__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m    978\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[key]\n\u001b[0;32m    980\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m key_is_scalar:\n\u001b[1;32m--> 981\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_value\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    983\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_hashable(key):\n\u001b[0;32m    984\u001b[0m     \u001b[38;5;66;03m# Otherwise index.get_value will raise InvalidIndexError\u001b[39;00m\n\u001b[0;32m    985\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    986\u001b[0m         \u001b[38;5;66;03m# For labels that don't resolve as scalars like tuples and frozensets\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:1089\u001b[0m, in \u001b[0;36mSeries._get_value\u001b[1;34m(self, label, takeable)\u001b[0m\n\u001b[0;32m   1086\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[label]\n\u001b[0;32m   1088\u001b[0m \u001b[38;5;66;03m# Similar to Index.get_value, but we do not fall back to positional\u001b[39;00m\n\u001b[1;32m-> 1089\u001b[0m loc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlabel\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1090\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39m_get_values_for_loc(\u001b[38;5;28mself\u001b[39m, loc, label)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3804\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m-> 3804\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[0;32m   3805\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[0;32m   3806\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[0;32m   3807\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[0;32m   3808\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[0;32m   3809\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[1;31mKeyError\u001b[0m: 'POS'"
     ]
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in data\n",
    "data['Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ef98fb8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "834735a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/sainee/sample_4_updated.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "395dac7b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5319238f",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/srinivas_sir_covered_MITO.bed', header=None, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c6f71138",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chromosome, Extended_Start_pos, Extended_End_pos]\n",
       "Index: []"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Value to check\n",
    "value = 9360634\n",
    "\n",
    "# Filter rows where the value lies between start and end positions\n",
    "filtered_df = df[(df['Extended_Start_pos'] <= value) & (df['Extended_End_pos'] >= value)]\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "be80335d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SLC22A5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CPT1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SLC25A20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EIF2AK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HSPA5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HSPA5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PDIA4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PPP1R15A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>DDIT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MAPK8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CALD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HADHA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ACOX1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HMGB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MT-ATP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MT-ATP8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MT-CYB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene\n",
       "0    SLC22A5\n",
       "1      CPT1A\n",
       "2   SLC25A20\n",
       "3    EIF2AK3\n",
       "4      HSPA5\n",
       "5      HSPA5\n",
       "6      PDIA4\n",
       "7   PPP1R15A\n",
       "8      DDIT3\n",
       "9      MAPK8\n",
       "10     CALD1\n",
       "11     HADHA\n",
       "12     ACOX1\n",
       "13     HMGB1\n",
       "14   MT-ATP6\n",
       "15   MT-ATP8\n",
       "16      MCAT\n",
       "17    MT-CYB"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/some_genes_covered.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2dd84f94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SLC22A5</td>\n",
       "      <td>chr5</td>\n",
       "      <td>132369824</td>\n",
       "      <td>132370394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SLC22A5</td>\n",
       "      <td>chr5</td>\n",
       "      <td>132378118</td>\n",
       "      <td>132378278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SLC22A5</td>\n",
       "      <td>chr5</td>\n",
       "      <td>132378340</td>\n",
       "      <td>132378500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SLC22A5</td>\n",
       "      <td>chr5</td>\n",
       "      <td>132384115</td>\n",
       "      <td>132384331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SLC22A5</td>\n",
       "      <td>chr5</td>\n",
       "      <td>132385297</td>\n",
       "      <td>132385571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>HMGB1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>30462630</td>\n",
       "      <td>30462790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>MCAT</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43133012</td>\n",
       "      <td>43133514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>MCAT</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43137051</td>\n",
       "      <td>43137327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>MCAT</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43140997</td>\n",
       "      <td>43141277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>MCAT</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43142896</td>\n",
       "      <td>43143376</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>181 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene chromosome  Extended_Start_pos  Extended_End_pos\n",
       "0    SLC22A5       chr5           132369824         132370394\n",
       "1    SLC22A5       chr5           132378118         132378278\n",
       "2    SLC22A5       chr5           132378340         132378500\n",
       "3    SLC22A5       chr5           132384115         132384331\n",
       "4    SLC22A5       chr5           132385297         132385571\n",
       "..       ...        ...                 ...               ...\n",
       "176    HMGB1      chr13            30462630          30462790\n",
       "177     MCAT      chr22            43133012          43133514\n",
       "178     MCAT      chr22            43137051          43137327\n",
       "179     MCAT      chr22            43140997          43141277\n",
       "180     MCAT      chr22            43142896          43143376\n",
       "\n",
       "[181 rows x 4 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_genes = pd.merge(data, df, how = 'inner', on = 'Gene', sort=False)\n",
    "merged_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4d74f65e",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_genes.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/genes_covered.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "977e48be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>RET</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs75527207</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117587806.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs4149056</td>\n",
       "      <td>chr12</td>\n",
       "      <td>21178615.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1799971</td>\n",
       "      <td>chr6</td>\n",
       "      <td>154039662.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs141033578</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117606695.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs141033578</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117606695.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615</th>\n",
       "      <td>rs1059513</td>\n",
       "      <td>chr12</td>\n",
       "      <td>57095926.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3616</th>\n",
       "      <td>rs12885713</td>\n",
       "      <td>chr14</td>\n",
       "      <td>90397013.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3617</th>\n",
       "      <td>rs12885713</td>\n",
       "      <td>chr14</td>\n",
       "      <td>90397013.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3618</th>\n",
       "      <td>rs12885713</td>\n",
       "      <td>chr14</td>\n",
       "      <td>90397013.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3619</th>\n",
       "      <td>rs2874116</td>\n",
       "      <td>chr18</td>\n",
       "      <td>58792367.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3620 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             rsID  CHROM          POS RET ALT Covered/Not_Covered\n",
       "0      rs75527207   chr7  117587806.0   G   A             Covered\n",
       "1       rs4149056  chr12   21178615.0   T   C             Covered\n",
       "2       rs1799971   chr6  154039662.0   A   G             Covered\n",
       "3     rs141033578   chr7  117606695.0   C   G             Covered\n",
       "4     rs141033578   chr7  117606695.0   C   T             Covered\n",
       "...           ...    ...          ...  ..  ..                 ...\n",
       "3615    rs1059513  chr12   57095926.0   T   C         Not_Covered\n",
       "3616   rs12885713  chr14   90397013.0   C   G         Not_Covered\n",
       "3617   rs12885713  chr14   90397013.0   C   T         Not_Covered\n",
       "3618   rs12885713  chr14   90397013.0   C   A         Not_Covered\n",
       "3619    rs2874116  chr18   58792367.0   G   A         Not_Covered\n",
       "\n",
       "[3620 rows x 6 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in data\n",
    "data['Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "cbeb99d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHBSTLGPTTL1_updated.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "843a9193",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "63cd0699",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Gene'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3802\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3801\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:138\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:165\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5745\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5753\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'Gene'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[21], line 30\u001b[0m\n\u001b[0;32m     27\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNot_Covered\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m     29\u001b[0m \u001b[38;5;66;03m# Step 3: Apply the function to create the new column in dataset2\u001b[39;00m\n\u001b[1;32m---> 30\u001b[0m data[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCovered/Not_Covered\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mdata\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcheck_coverage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m     31\u001b[0m data\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py:9568\u001b[0m, in \u001b[0;36mDataFrame.apply\u001b[1;34m(self, func, axis, raw, result_type, args, **kwargs)\u001b[0m\n\u001b[0;32m   9557\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcore\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mapply\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m frame_apply\n\u001b[0;32m   9559\u001b[0m op \u001b[38;5;241m=\u001b[39m frame_apply(\n\u001b[0;32m   9560\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m   9561\u001b[0m     func\u001b[38;5;241m=\u001b[39mfunc,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   9566\u001b[0m     kwargs\u001b[38;5;241m=\u001b[39mkwargs,\n\u001b[0;32m   9567\u001b[0m )\n\u001b[1;32m-> 9568\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mop\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39m__finalize__(\u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mapply\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:764\u001b[0m, in \u001b[0;36mFrameApply.apply\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    761\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mraw:\n\u001b[0;32m    762\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mapply_raw()\n\u001b[1;32m--> 764\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply_standard\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:891\u001b[0m, in \u001b[0;36mFrameApply.apply_standard\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    890\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mapply_standard\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[1;32m--> 891\u001b[0m     results, res_index \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply_series_generator\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    893\u001b[0m     \u001b[38;5;66;03m# wrap results\u001b[39;00m\n\u001b[0;32m    894\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mwrap_results(results, res_index)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:907\u001b[0m, in \u001b[0;36mFrameApply.apply_series_generator\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    904\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m option_context(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmode.chained_assignment\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m    905\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m i, v \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(series_gen):\n\u001b[0;32m    906\u001b[0m         \u001b[38;5;66;03m# ignore SettingWithCopy here in case the user mutates\u001b[39;00m\n\u001b[1;32m--> 907\u001b[0m         results[i] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[43mv\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    908\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(results[i], ABCSeries):\n\u001b[0;32m    909\u001b[0m             \u001b[38;5;66;03m# If we have a view on v, we need to make a copy because\u001b[39;00m\n\u001b[0;32m    910\u001b[0m             \u001b[38;5;66;03m#  series_generator will swap out the underlying data\u001b[39;00m\n\u001b[0;32m    911\u001b[0m             results[i] \u001b[38;5;241m=\u001b[39m results[i]\u001b[38;5;241m.\u001b[39mcopy(deep\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "Cell \u001b[1;32mIn[21], line 19\u001b[0m, in \u001b[0;36mcheck_coverage\u001b[1;34m(row)\u001b[0m\n\u001b[0;32m     17\u001b[0m pos \u001b[38;5;241m=\u001b[39m row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPOS\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m     18\u001b[0m chromosome \u001b[38;5;241m=\u001b[39m row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCHROM\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m---> 19\u001b[0m gene \u001b[38;5;241m=\u001b[39m \u001b[43mrow\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mGene\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m  \u001b[38;5;66;03m# Assuming 'GENE' is the column name for gene information\u001b[39;00m\n\u001b[0;32m     20\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chromosome \u001b[38;5;129;01min\u001b[39;00m chromosome_dict:\n\u001b[0;32m     21\u001b[0m     ranges \u001b[38;5;241m=\u001b[39m chromosome_dict[chromosome]\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:981\u001b[0m, in \u001b[0;36mSeries.__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m    978\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[key]\n\u001b[0;32m    980\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m key_is_scalar:\n\u001b[1;32m--> 981\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_value\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    983\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_hashable(key):\n\u001b[0;32m    984\u001b[0m     \u001b[38;5;66;03m# Otherwise index.get_value will raise InvalidIndexError\u001b[39;00m\n\u001b[0;32m    985\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    986\u001b[0m         \u001b[38;5;66;03m# For labels that don't resolve as scalars like tuples and frozensets\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:1089\u001b[0m, in \u001b[0;36mSeries._get_value\u001b[1;34m(self, label, takeable)\u001b[0m\n\u001b[0;32m   1086\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[label]\n\u001b[0;32m   1088\u001b[0m \u001b[38;5;66;03m# Similar to Index.get_value, but we do not fall back to positional\u001b[39;00m\n\u001b[1;32m-> 1089\u001b[0m loc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlabel\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1090\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39m_get_values_for_loc(\u001b[38;5;28mself\u001b[39m, loc, label)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3804\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m-> 3804\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[0;32m   3805\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[0;32m   3806\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[0;32m   3807\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[0;32m   3808\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[0;32m   3809\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[1;31mKeyError\u001b[0m: 'Gene'"
     ]
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "gene_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    gene = row['Gene']  # Assuming 'Gene' is the column name for gene information\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "        gene_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "    gene_dict[chromosome].append(gene)\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    gene = row['Gene']  # Assuming 'GENE' is the column name for gene information\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        genes = gene_dict[chromosome]\n",
    "        for i, (start, end) in enumerate(ranges):\n",
    "            if start <= pos <= end:\n",
    "                if gene == genes[i]:\n",
    "                    return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "data['Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b133e80",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4a4f821b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515787</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs11615</td>\n",
       "      <td>ERCC1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>45420395</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs699517</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>chr18</td>\n",
       "      <td>673016</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs3218592</td>\n",
       "      <td>REV3L</td>\n",
       "      <td>chr6</td>\n",
       "      <td>111322635</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>chr7</td>\n",
       "      <td>87509329</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>rs9389568</td>\n",
       "      <td>PERP</td>\n",
       "      <td>chr6</td>\n",
       "      <td>138145853</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>rs2960436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7</td>\n",
       "      <td>45937683</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>rs117458836</td>\n",
       "      <td>CYP2C8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95013625</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>rs118088833</td>\n",
       "      <td>TP53AIP1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>128937143</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>rs1760217</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97137438</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>176 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            rsID      Gene  CHROM        POS Covered/Not_Covered\n",
       "0     rs55886062      DPYD   chr1   97515787             Covered\n",
       "1        rs11615     ERCC1  chr19   45420395             Covered\n",
       "2       rs699517      TYMS  chr18     673016         Not_Covered\n",
       "3      rs3218592     REV3L   chr6  111322635             Covered\n",
       "4      rs1045642     ABCB1   chr7   87509329             Covered\n",
       "..           ...       ...    ...        ...                 ...\n",
       "171    rs9389568      PERP   chr6  138145853         Not_Covered\n",
       "172    rs2960436       NaN   chr7   45937683         Not_Covered\n",
       "173  rs117458836    CYP2C8  chr10   95013625         Not_Covered\n",
       "174  rs118088833  TP53AIP1  chr11  128937143         Not_Covered\n",
       "175    rs1760217      DPYD   chr1   97137438         Not_Covered\n",
       "\n",
       "[176 rows x 5 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "data['Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "91732f50",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PharmGKB ID</th>\n",
       "      <th>Level</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Drugs</th>\n",
       "      <th>Phenotype Categories</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Pediatric</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1451276160</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1445401125</td>\n",
       "      <td>3</td>\n",
       "      <td>rs11615</td>\n",
       "      <td>ERCC1</td>\n",
       "      <td>capecitabine; radiotherapy</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Rectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1448616922</td>\n",
       "      <td>3</td>\n",
       "      <td>rs699517</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Asthenia; Nausea; Neoplasms; Vomiting</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1444667292</td>\n",
       "      <td>3</td>\n",
       "      <td>rs3218592</td>\n",
       "      <td>REV3L</td>\n",
       "      <td>capecitabine; fluorouracil</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Neoplasm Metastasis</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>981204466</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>1184986201</td>\n",
       "      <td>3</td>\n",
       "      <td>rs11479</td>\n",
       "      <td>TYMP</td>\n",
       "      <td>capecitabine; fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>981204864</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1801131</td>\n",
       "      <td>MTHFR</td>\n",
       "      <td>capecitabine; fluorouracil; leucovorin; oxalip...</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>1449270937</td>\n",
       "      <td>3</td>\n",
       "      <td>rs117458836</td>\n",
       "      <td>CYP2C8</td>\n",
       "      <td>cyclophosphamide; epirubicin; fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Breast Neoplasms; Neutropenia</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>1449270943</td>\n",
       "      <td>3</td>\n",
       "      <td>rs118088833</td>\n",
       "      <td>TP53AIP1</td>\n",
       "      <td>cyclophosphamide; epirubicin; fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Breast Neoplasms; Neutropenia</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>1447953072</td>\n",
       "      <td>3</td>\n",
       "      <td>rs183205964</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>capecitabine; fluorouracil; tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>261 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     PharmGKB ID Level         rsID      Gene  \\\n",
       "0     1451276160    1A   rs55886062      DPYD   \n",
       "1     1445401125     3      rs11615     ERCC1   \n",
       "2     1448616922     3     rs699517      TYMS   \n",
       "3     1444667292     3    rs3218592     REV3L   \n",
       "4      981204466     3    rs1045642     ABCB1   \n",
       "..           ...   ...          ...       ...   \n",
       "256   1184986201     3      rs11479      TYMP   \n",
       "257    981204864     3    rs1801131     MTHFR   \n",
       "258   1449270937     3  rs117458836    CYP2C8   \n",
       "259   1449270943     3  rs118088833  TP53AIP1   \n",
       "260   1447953072     3  rs183205964      TYMS   \n",
       "\n",
       "                                                 Drugs Phenotype Categories  \\\n",
       "0                                         capecitabine             Toxicity   \n",
       "1                           capecitabine; radiotherapy             Efficacy   \n",
       "2                                         capecitabine             Toxicity   \n",
       "3                           capecitabine; fluorouracil             Efficacy   \n",
       "4                                         capecitabine             Toxicity   \n",
       "..                                                 ...                  ...   \n",
       "256                         capecitabine; fluorouracil             Toxicity   \n",
       "257  capecitabine; fluorouracil; leucovorin; oxalip...             Efficacy   \n",
       "258         cyclophosphamide; epirubicin; fluorouracil             Toxicity   \n",
       "259         cyclophosphamide; epirubicin; fluorouracil             Toxicity   \n",
       "260                capecitabine; fluorouracil; tegafur             Toxicity   \n",
       "\n",
       "                                 Phenotype  Pediatric  \n",
       "0                                Neoplasms      False  \n",
       "1                         Rectal Neoplasms      False  \n",
       "2    Asthenia; Nausea; Neoplasms; Vomiting      False  \n",
       "3                      Neoplasm Metastasis      False  \n",
       "4                                Neoplasms      False  \n",
       "..                                     ...        ...  \n",
       "256                              Neoplasms      False  \n",
       "257                              Neoplasms      False  \n",
       "258          Breast Neoplasms; Neutropenia      False  \n",
       "259          Breast Neoplasms; Neutropenia      False  \n",
       "260                              Neoplasms      False  \n",
       "\n",
       "[261 rows x 8 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/krishna_covered_pos/DYPD gene_Capcetabine, flurouracil, oxaliplatin.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "aa0e913d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PharmGKB ID</th>\n",
       "      <th>Level</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene_x</th>\n",
       "      <th>Drugs</th>\n",
       "      <th>Phenotype Categories</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Pediatric</th>\n",
       "      <th>Gene_y</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1451276160</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515787.0</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1445401125</td>\n",
       "      <td>3</td>\n",
       "      <td>rs11615</td>\n",
       "      <td>ERCC1</td>\n",
       "      <td>capecitabine; radiotherapy</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Rectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>ERCC1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>45420395.0</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1448616922</td>\n",
       "      <td>3</td>\n",
       "      <td>rs699517</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Asthenia; Nausea; Neoplasms; Vomiting</td>\n",
       "      <td>False</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>chr18</td>\n",
       "      <td>673016.0</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1444667292</td>\n",
       "      <td>3</td>\n",
       "      <td>rs3218592</td>\n",
       "      <td>REV3L</td>\n",
       "      <td>capecitabine; fluorouracil</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Neoplasm Metastasis</td>\n",
       "      <td>False</td>\n",
       "      <td>REV3L</td>\n",
       "      <td>chr6</td>\n",
       "      <td>111322635.0</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>981204466</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>chr7</td>\n",
       "      <td>87509329.0</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>1184986201</td>\n",
       "      <td>3</td>\n",
       "      <td>rs11479</td>\n",
       "      <td>TYMP</td>\n",
       "      <td>capecitabine; fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>TYMP</td>\n",
       "      <td>chr22</td>\n",
       "      <td>50525807.0</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>981204864</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1801131</td>\n",
       "      <td>MTHFR</td>\n",
       "      <td>capecitabine; fluorouracil; leucovorin; oxalip...</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>MTHFR</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11794419.0</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>1449270937</td>\n",
       "      <td>3</td>\n",
       "      <td>rs117458836</td>\n",
       "      <td>CYP2C8</td>\n",
       "      <td>cyclophosphamide; epirubicin; fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Breast Neoplasms; Neutropenia</td>\n",
       "      <td>False</td>\n",
       "      <td>CYP2C8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95013625.0</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>1449270943</td>\n",
       "      <td>3</td>\n",
       "      <td>rs118088833</td>\n",
       "      <td>TP53AIP1</td>\n",
       "      <td>cyclophosphamide; epirubicin; fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Breast Neoplasms; Neutropenia</td>\n",
       "      <td>False</td>\n",
       "      <td>TP53AIP1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>128937143.0</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>1447953072</td>\n",
       "      <td>3</td>\n",
       "      <td>rs183205964</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>capecitabine; fluorouracil; tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>TYMS</td>\n",
       "      <td>chr18</td>\n",
       "      <td>657657.0</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>261 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     PharmGKB ID Level         rsID    Gene_x  \\\n",
       "0     1451276160    1A   rs55886062      DPYD   \n",
       "1     1445401125     3      rs11615     ERCC1   \n",
       "2     1448616922     3     rs699517      TYMS   \n",
       "3     1444667292     3    rs3218592     REV3L   \n",
       "4      981204466     3    rs1045642     ABCB1   \n",
       "..           ...   ...          ...       ...   \n",
       "256   1184986201     3      rs11479      TYMP   \n",
       "257    981204864     3    rs1801131     MTHFR   \n",
       "258   1449270937     3  rs117458836    CYP2C8   \n",
       "259   1449270943     3  rs118088833  TP53AIP1   \n",
       "260   1447953072     3  rs183205964      TYMS   \n",
       "\n",
       "                                                 Drugs Phenotype Categories  \\\n",
       "0                                         capecitabine             Toxicity   \n",
       "1                           capecitabine; radiotherapy             Efficacy   \n",
       "2                                         capecitabine             Toxicity   \n",
       "3                           capecitabine; fluorouracil             Efficacy   \n",
       "4                                         capecitabine             Toxicity   \n",
       "..                                                 ...                  ...   \n",
       "256                         capecitabine; fluorouracil             Toxicity   \n",
       "257  capecitabine; fluorouracil; leucovorin; oxalip...             Efficacy   \n",
       "258         cyclophosphamide; epirubicin; fluorouracil             Toxicity   \n",
       "259         cyclophosphamide; epirubicin; fluorouracil             Toxicity   \n",
       "260                capecitabine; fluorouracil; tegafur             Toxicity   \n",
       "\n",
       "                                 Phenotype  Pediatric    Gene_y  CHROM  \\\n",
       "0                                Neoplasms      False      DPYD   chr1   \n",
       "1                         Rectal Neoplasms      False     ERCC1  chr19   \n",
       "2    Asthenia; Nausea; Neoplasms; Vomiting      False      TYMS  chr18   \n",
       "3                      Neoplasm Metastasis      False     REV3L   chr6   \n",
       "4                                Neoplasms      False     ABCB1   chr7   \n",
       "..                                     ...        ...       ...    ...   \n",
       "256                              Neoplasms      False      TYMP  chr22   \n",
       "257                              Neoplasms      False     MTHFR   chr1   \n",
       "258          Breast Neoplasms; Neutropenia      False    CYP2C8  chr10   \n",
       "259          Breast Neoplasms; Neutropenia      False  TP53AIP1  chr11   \n",
       "260                              Neoplasms      False      TYMS  chr18   \n",
       "\n",
       "             POS Covered/Not_Covered  \n",
       "0     97515787.0             Covered  \n",
       "1     45420395.0             Covered  \n",
       "2       673016.0         Not_Covered  \n",
       "3    111322635.0             Covered  \n",
       "4     87509329.0             Covered  \n",
       "..           ...                 ...  \n",
       "256   50525807.0             Covered  \n",
       "257   11794419.0             Covered  \n",
       "258   95013625.0         Not_Covered  \n",
       "259  128937143.0         Not_Covered  \n",
       "260     657657.0         Not_Covered  \n",
       "\n",
       "[261 rows x 12 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(df_1, data, on = 'rsID', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c62b8a4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PharmGKB ID</th>\n",
       "      <th>Level</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Drugs</th>\n",
       "      <th>Phenotype Categories</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Pediatric</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1451276160</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1451286020</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801160</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>981201535</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs2297595</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1447989706</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801266</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1449575662</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs148994843</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1451275020</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1449575656</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs59086055</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1451287440</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs17376848</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1451266065</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1449575668</td>\n",
       "      <td>3</td>\n",
       "      <td>rs748620513</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1447989723</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801268</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1451286040</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801160</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1451274140</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs115232898</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1449575674</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1184321568</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1184986174</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1447989739</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs78060119</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1447989732</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs72549306</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1444608182</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12022243</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Colorectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1451286320</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1449575639</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1335150891</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1451274165</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1444608175</td>\n",
       "      <td>3</td>\n",
       "      <td>rs76387818</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Colorectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1451286326</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801158</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1449575633</td>\n",
       "      <td>3</td>\n",
       "      <td>rs188052243</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1449575645</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs56005131</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>1451289660</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs59086055</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1451274045</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>1451287360</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs2297595</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>1183630664</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>981201981</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801265</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>1183679818</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1760217</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>antineoplastic agents</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Pancreatic Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>1451287240</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801265</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>1451266120</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>1451288269</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs56038477</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>981201962</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1451275220</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>1183679833</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs17376848</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1447989784</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs56038477</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>1451274200</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs78060119</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>981203618</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1447989777</td>\n",
       "      <td>3</td>\n",
       "      <td>rs115632870</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>1447989678</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs72549309</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>981201946</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801158</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>1451266020</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1451289700</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>1183703784</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs115232898</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>1445401793</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801160</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>1451274984</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>1447989671</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs72549303</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>1451274090</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>1447989796</td>\n",
       "      <td>3</td>\n",
       "      <td>rs72728438</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>1449575690</td>\n",
       "      <td>3</td>\n",
       "      <td>rs143879757</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>1448640301</td>\n",
       "      <td>4</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>827843617</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1449575696</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1212037891</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>1448616865</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12119882</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Hyperbilirubinemia; Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>1444608224</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12132152</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Colorectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PharmGKB ID Level          rsID  Gene                  Drugs  \\\n",
       "0    1451276160    1A    rs55886062  DPYD           capecitabine   \n",
       "1    1451286020    1A     rs1801160  DPYD           fluorouracil   \n",
       "2     981201535    1A     rs2297595  DPYD           fluorouracil   \n",
       "3    1447989706    1A     rs1801266  DPYD           fluorouracil   \n",
       "4    1449575662    1A   rs148994843  DPYD           fluorouracil   \n",
       "5    1451275020    1A     rs3918290  DPYD           fluorouracil   \n",
       "6    1449575656    1A    rs59086055  DPYD           fluorouracil   \n",
       "7    1451287440    1A    rs17376848  DPYD           capecitabine   \n",
       "8    1451266065    1A     rs3918290  DPYD                tegafur   \n",
       "9    1449575668     3   rs748620513  DPYD           fluorouracil   \n",
       "10   1447989723    1A     rs1801268  DPYD           fluorouracil   \n",
       "11   1451286040    1A     rs1801160  DPYD           capecitabine   \n",
       "12   1451274140    1A   rs115232898  DPYD           fluorouracil   \n",
       "13   1449575674     3  rs1184321568  DPYD           fluorouracil   \n",
       "14   1184986174    1A    rs75017182  DPYD           fluorouracil   \n",
       "15   1447989739    1A    rs78060119  DPYD           fluorouracil   \n",
       "16   1447989732    1A    rs72549306  DPYD           fluorouracil   \n",
       "17   1444608182     3    rs12022243  DPYD           capecitabine   \n",
       "18   1451286320    1A     rs1801159  DPYD           capecitabine   \n",
       "19   1449575639     3  rs1335150891  DPYD           fluorouracil   \n",
       "20   1451274165    1A    rs75017182  DPYD           capecitabine   \n",
       "21   1444608175     3    rs76387818  DPYD           capecitabine   \n",
       "22   1451286326    1A     rs1801158  DPYD           capecitabine   \n",
       "23   1449575633     3   rs188052243  DPYD           fluorouracil   \n",
       "24   1449575645    1A    rs56005131  DPYD           fluorouracil   \n",
       "25   1451289660    1A    rs59086055  DPYD           fluorouracil   \n",
       "26   1451274045    1A    rs67376798  DPYD           fluorouracil   \n",
       "27   1451287360    1A     rs2297595  DPYD           capecitabine   \n",
       "28   1183630664    1A    rs55886062  DPYD           fluorouracil   \n",
       "29    981201981    1A     rs1801265  DPYD           fluorouracil   \n",
       "30   1183679818     3     rs1760217  DPYD  antineoplastic agents   \n",
       "31   1451287240    1A     rs1801265  DPYD           capecitabine   \n",
       "32   1451266120    1A    rs67376798  DPYD                tegafur   \n",
       "33   1451288269    1A    rs56038477  DPYD           capecitabine   \n",
       "34    981201962    1A     rs1801159  DPYD           fluorouracil   \n",
       "35   1451275220    1A    rs55886062  DPYD           fluorouracil   \n",
       "36   1183679833    1A    rs17376848  DPYD           fluorouracil   \n",
       "37   1447989784    1A    rs56038477  DPYD           fluorouracil   \n",
       "38   1451274200    1A    rs78060119  DPYD           fluorouracil   \n",
       "39    981203618    1A    rs67376798  DPYD           fluorouracil   \n",
       "40   1447989777     3   rs115632870  DPYD           fluorouracil   \n",
       "41   1447989678    1A    rs72549309  DPYD           fluorouracil   \n",
       "42    981201946    1A     rs1801158  DPYD           fluorouracil   \n",
       "43   1451266020    1A    rs55886062  DPYD                tegafur   \n",
       "44   1451289700    1A    rs75017182  DPYD           fluorouracil   \n",
       "45   1183703784    1A   rs115232898  DPYD           fluorouracil   \n",
       "46   1445401793    1A     rs1801160  DPYD           fluorouracil   \n",
       "47   1451274984    1A     rs3918290  DPYD           capecitabine   \n",
       "48   1447989671    1A    rs72549303  DPYD           fluorouracil   \n",
       "49   1451274090    1A    rs67376798  DPYD           capecitabine   \n",
       "50   1447989796     3    rs72728438  DPYD           fluorouracil   \n",
       "51   1449575690     3   rs143879757  DPYD           fluorouracil   \n",
       "52   1448640301     4     rs3918290  DPYD           fluorouracil   \n",
       "53    827843617    1A     rs3918290  DPYD           fluorouracil   \n",
       "54   1449575696     3  rs1212037891  DPYD           fluorouracil   \n",
       "55   1448616865     3    rs12119882  DPYD           capecitabine   \n",
       "56   1444608224     3    rs12132152  DPYD           capecitabine   \n",
       "\n",
       "   Phenotype Categories                      Phenotype  Pediatric  \n",
       "0              Toxicity                      Neoplasms      False  \n",
       "1              Toxicity                      Neoplasms       True  \n",
       "2              Toxicity                      Neoplasms      False  \n",
       "3                 Other                            NaN      False  \n",
       "4                 Other                            NaN      False  \n",
       "5                 Other                            NaN       True  \n",
       "6                 Other                            NaN      False  \n",
       "7              Toxicity                      Neoplasms      False  \n",
       "8              Toxicity                      Neoplasms      False  \n",
       "9                 Other                            NaN      False  \n",
       "10                Other                            NaN      False  \n",
       "11             Toxicity                      Neoplasms      False  \n",
       "12                Other                            NaN      False  \n",
       "13                Other                            NaN      False  \n",
       "14             Toxicity                      Neoplasms      False  \n",
       "15             Toxicity                      Neoplasms       True  \n",
       "16                Other                            NaN      False  \n",
       "17             Toxicity           Colorectal Neoplasms      False  \n",
       "18             Toxicity                      Neoplasms      False  \n",
       "19                Other                            NaN      False  \n",
       "20             Toxicity                      Neoplasms      False  \n",
       "21             Toxicity           Colorectal Neoplasms      False  \n",
       "22             Toxicity                      Neoplasms      False  \n",
       "23                Other                            NaN      False  \n",
       "24                Other                            NaN      False  \n",
       "25             Toxicity                      Neoplasms      False  \n",
       "26                Other                            NaN       True  \n",
       "27             Toxicity                      Neoplasms      False  \n",
       "28             Toxicity                      Neoplasms      False  \n",
       "29             Toxicity                      Neoplasms       True  \n",
       "30             Efficacy           Pancreatic Neoplasms      False  \n",
       "31             Toxicity                      Neoplasms      False  \n",
       "32             Toxicity                      Neoplasms      False  \n",
       "33             Toxicity                      Neoplasms      False  \n",
       "34             Toxicity                      Neoplasms      False  \n",
       "35                Other                            NaN       True  \n",
       "36             Toxicity                      Neoplasms      False  \n",
       "37             Toxicity                      Neoplasms      False  \n",
       "38                Other                            NaN      False  \n",
       "39             Toxicity                      Neoplasms      False  \n",
       "40                Other                            NaN      False  \n",
       "41                Other                            NaN      False  \n",
       "42             Toxicity                      Neoplasms      False  \n",
       "43             Toxicity                      Neoplasms      False  \n",
       "44                Other                            NaN      False  \n",
       "45             Toxicity                      Neoplasms      False  \n",
       "46                Other                            NaN      False  \n",
       "47             Toxicity                      Neoplasms      False  \n",
       "48                Other                            NaN      False  \n",
       "49             Toxicity                      Neoplasms      False  \n",
       "50                Other                            NaN      False  \n",
       "51                Other                            NaN      False  \n",
       "52             Efficacy                      Neoplasms       True  \n",
       "53             Toxicity                      Neoplasms       True  \n",
       "54                Other                            NaN      False  \n",
       "55             Toxicity  Hyperbilirubinemia; Neoplasms      False  \n",
       "56             Toxicity           Colorectal Neoplasms      False  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/krishna_covered_pos/DYPD_gene.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2eb5a99b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PharmGKB ID</th>\n",
       "      <th>Level</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene_x</th>\n",
       "      <th>Drugs</th>\n",
       "      <th>Phenotype Categories</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Pediatric</th>\n",
       "      <th>Gene_y</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1451276160</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515787</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1451286020</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801160</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97305364</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>981201535</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs2297595</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97699535</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1447989706</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801266</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97691776</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1449575662</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs148994843</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515923</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1451275020</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450058</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1449575656</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs59086055</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450190</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1451287440</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs17376848</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450068</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1451266065</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450058</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1449575668</td>\n",
       "      <td>3</td>\n",
       "      <td>rs748620513</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97573799</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1447989723</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801268</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97079071</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1451286040</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801160</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97305364</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1451274140</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs115232898</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97699474</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1449575674</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1184321568</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97593251</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1184986174</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97579893</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1447989739</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs78060119</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97573943</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1447989732</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs72549306</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97593343</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1444608182</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12022243</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Colorectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97397224</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1451286320</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515839</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1449575639</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1335150891</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97234874</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1451274165</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97579893</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1444608175</td>\n",
       "      <td>3</td>\n",
       "      <td>rs76387818</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Colorectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97073844</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1451286326</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801158</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515865</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1449575633</td>\n",
       "      <td>3</td>\n",
       "      <td>rs188052243</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97098577</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1449575645</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs56005131</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97234991</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>1451289660</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs59086055</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450190</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1451274045</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97082391</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>1451287360</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs2297595</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97699535</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>1183630664</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515787</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>981201981</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801265</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97883329</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>1183679818</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1760217</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>antineoplastic agents</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Pancreatic Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97137438</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>1451287240</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801265</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97883329</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>1451266120</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97082391</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>1451288269</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs56038477</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97573863</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>981201962</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515839</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1451275220</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515787</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>1183679833</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs17376848</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450068</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1447989784</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs56038477</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97573863</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>1451274200</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs78060119</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97573943</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>981203618</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97082391</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1447989777</td>\n",
       "      <td>3</td>\n",
       "      <td>rs115632870</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97828265</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>1447989678</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs72549309</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97740411</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>981201946</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801158</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515865</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>1451266020</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs55886062</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>tegafur</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515787</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1451289700</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97579893</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>1183703784</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs115232898</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97699474</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>1445401793</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs1801160</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97305364</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>1451274984</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450058</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>1447989671</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs72549303</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450066</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>1451274090</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs67376798</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97082391</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>1447989796</td>\n",
       "      <td>3</td>\n",
       "      <td>rs72728438</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97382318</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>1449575690</td>\n",
       "      <td>3</td>\n",
       "      <td>rs143879757</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97595124</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>1448640301</td>\n",
       "      <td>4</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450058</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>827843617</td>\n",
       "      <td>1A</td>\n",
       "      <td>rs3918290</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Neoplasms</td>\n",
       "      <td>True</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97450058</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1449575696</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1212037891</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97721668</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>1448616865</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12119882</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Hyperbilirubinemia; Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97696806</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>1444608224</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12132152</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Colorectal Neoplasms</td>\n",
       "      <td>False</td>\n",
       "      <td>DPYD</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97057448</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PharmGKB ID Level          rsID Gene_x                  Drugs  \\\n",
       "0    1451276160    1A    rs55886062   DPYD           capecitabine   \n",
       "1    1451286020    1A     rs1801160   DPYD           fluorouracil   \n",
       "2     981201535    1A     rs2297595   DPYD           fluorouracil   \n",
       "3    1447989706    1A     rs1801266   DPYD           fluorouracil   \n",
       "4    1449575662    1A   rs148994843   DPYD           fluorouracil   \n",
       "5    1451275020    1A     rs3918290   DPYD           fluorouracil   \n",
       "6    1449575656    1A    rs59086055   DPYD           fluorouracil   \n",
       "7    1451287440    1A    rs17376848   DPYD           capecitabine   \n",
       "8    1451266065    1A     rs3918290   DPYD                tegafur   \n",
       "9    1449575668     3   rs748620513   DPYD           fluorouracil   \n",
       "10   1447989723    1A     rs1801268   DPYD           fluorouracil   \n",
       "11   1451286040    1A     rs1801160   DPYD           capecitabine   \n",
       "12   1451274140    1A   rs115232898   DPYD           fluorouracil   \n",
       "13   1449575674     3  rs1184321568   DPYD           fluorouracil   \n",
       "14   1184986174    1A    rs75017182   DPYD           fluorouracil   \n",
       "15   1447989739    1A    rs78060119   DPYD           fluorouracil   \n",
       "16   1447989732    1A    rs72549306   DPYD           fluorouracil   \n",
       "17   1444608182     3    rs12022243   DPYD           capecitabine   \n",
       "18   1451286320    1A     rs1801159   DPYD           capecitabine   \n",
       "19   1449575639     3  rs1335150891   DPYD           fluorouracil   \n",
       "20   1451274165    1A    rs75017182   DPYD           capecitabine   \n",
       "21   1444608175     3    rs76387818   DPYD           capecitabine   \n",
       "22   1451286326    1A     rs1801158   DPYD           capecitabine   \n",
       "23   1449575633     3   rs188052243   DPYD           fluorouracil   \n",
       "24   1449575645    1A    rs56005131   DPYD           fluorouracil   \n",
       "25   1451289660    1A    rs59086055   DPYD           fluorouracil   \n",
       "26   1451274045    1A    rs67376798   DPYD           fluorouracil   \n",
       "27   1451287360    1A     rs2297595   DPYD           capecitabine   \n",
       "28   1183630664    1A    rs55886062   DPYD           fluorouracil   \n",
       "29    981201981    1A     rs1801265   DPYD           fluorouracil   \n",
       "30   1183679818     3     rs1760217   DPYD  antineoplastic agents   \n",
       "31   1451287240    1A     rs1801265   DPYD           capecitabine   \n",
       "32   1451266120    1A    rs67376798   DPYD                tegafur   \n",
       "33   1451288269    1A    rs56038477   DPYD           capecitabine   \n",
       "34    981201962    1A     rs1801159   DPYD           fluorouracil   \n",
       "35   1451275220    1A    rs55886062   DPYD           fluorouracil   \n",
       "36   1183679833    1A    rs17376848   DPYD           fluorouracil   \n",
       "37   1447989784    1A    rs56038477   DPYD           fluorouracil   \n",
       "38   1451274200    1A    rs78060119   DPYD           fluorouracil   \n",
       "39    981203618    1A    rs67376798   DPYD           fluorouracil   \n",
       "40   1447989777     3   rs115632870   DPYD           fluorouracil   \n",
       "41   1447989678    1A    rs72549309   DPYD           fluorouracil   \n",
       "42    981201946    1A     rs1801158   DPYD           fluorouracil   \n",
       "43   1451266020    1A    rs55886062   DPYD                tegafur   \n",
       "44   1451289700    1A    rs75017182   DPYD           fluorouracil   \n",
       "45   1183703784    1A   rs115232898   DPYD           fluorouracil   \n",
       "46   1445401793    1A     rs1801160   DPYD           fluorouracil   \n",
       "47   1451274984    1A     rs3918290   DPYD           capecitabine   \n",
       "48   1447989671    1A    rs72549303   DPYD           fluorouracil   \n",
       "49   1451274090    1A    rs67376798   DPYD           capecitabine   \n",
       "50   1447989796     3    rs72728438   DPYD           fluorouracil   \n",
       "51   1449575690     3   rs143879757   DPYD           fluorouracil   \n",
       "52   1448640301     4     rs3918290   DPYD           fluorouracil   \n",
       "53    827843617    1A     rs3918290   DPYD           fluorouracil   \n",
       "54   1449575696     3  rs1212037891   DPYD           fluorouracil   \n",
       "55   1448616865     3    rs12119882   DPYD           capecitabine   \n",
       "56   1444608224     3    rs12132152   DPYD           capecitabine   \n",
       "\n",
       "   Phenotype Categories                      Phenotype  Pediatric Gene_y  \\\n",
       "0              Toxicity                      Neoplasms      False   DPYD   \n",
       "1              Toxicity                      Neoplasms       True   DPYD   \n",
       "2              Toxicity                      Neoplasms      False   DPYD   \n",
       "3                 Other                            NaN      False   DPYD   \n",
       "4                 Other                            NaN      False   DPYD   \n",
       "5                 Other                            NaN       True   DPYD   \n",
       "6                 Other                            NaN      False   DPYD   \n",
       "7              Toxicity                      Neoplasms      False   DPYD   \n",
       "8              Toxicity                      Neoplasms      False   DPYD   \n",
       "9                 Other                            NaN      False   DPYD   \n",
       "10                Other                            NaN      False   DPYD   \n",
       "11             Toxicity                      Neoplasms      False   DPYD   \n",
       "12                Other                            NaN      False   DPYD   \n",
       "13                Other                            NaN      False   DPYD   \n",
       "14             Toxicity                      Neoplasms      False   DPYD   \n",
       "15             Toxicity                      Neoplasms       True   DPYD   \n",
       "16                Other                            NaN      False   DPYD   \n",
       "17             Toxicity           Colorectal Neoplasms      False   DPYD   \n",
       "18             Toxicity                      Neoplasms      False   DPYD   \n",
       "19                Other                            NaN      False   DPYD   \n",
       "20             Toxicity                      Neoplasms      False   DPYD   \n",
       "21             Toxicity           Colorectal Neoplasms      False   DPYD   \n",
       "22             Toxicity                      Neoplasms      False   DPYD   \n",
       "23                Other                            NaN      False   DPYD   \n",
       "24                Other                            NaN      False   DPYD   \n",
       "25             Toxicity                      Neoplasms      False   DPYD   \n",
       "26                Other                            NaN       True   DPYD   \n",
       "27             Toxicity                      Neoplasms      False   DPYD   \n",
       "28             Toxicity                      Neoplasms      False   DPYD   \n",
       "29             Toxicity                      Neoplasms       True   DPYD   \n",
       "30             Efficacy           Pancreatic Neoplasms      False   DPYD   \n",
       "31             Toxicity                      Neoplasms      False   DPYD   \n",
       "32             Toxicity                      Neoplasms      False   DPYD   \n",
       "33             Toxicity                      Neoplasms      False   DPYD   \n",
       "34             Toxicity                      Neoplasms      False   DPYD   \n",
       "35                Other                            NaN       True   DPYD   \n",
       "36             Toxicity                      Neoplasms      False   DPYD   \n",
       "37             Toxicity                      Neoplasms      False   DPYD   \n",
       "38                Other                            NaN      False   DPYD   \n",
       "39             Toxicity                      Neoplasms      False   DPYD   \n",
       "40                Other                            NaN      False   DPYD   \n",
       "41                Other                            NaN      False   DPYD   \n",
       "42             Toxicity                      Neoplasms      False   DPYD   \n",
       "43             Toxicity                      Neoplasms      False   DPYD   \n",
       "44                Other                            NaN      False   DPYD   \n",
       "45             Toxicity                      Neoplasms      False   DPYD   \n",
       "46                Other                            NaN      False   DPYD   \n",
       "47             Toxicity                      Neoplasms      False   DPYD   \n",
       "48                Other                            NaN      False   DPYD   \n",
       "49             Toxicity                      Neoplasms      False   DPYD   \n",
       "50                Other                            NaN      False   DPYD   \n",
       "51                Other                            NaN      False   DPYD   \n",
       "52             Efficacy                      Neoplasms       True   DPYD   \n",
       "53             Toxicity                      Neoplasms       True   DPYD   \n",
       "54                Other                            NaN      False   DPYD   \n",
       "55             Toxicity  Hyperbilirubinemia; Neoplasms      False   DPYD   \n",
       "56             Toxicity           Colorectal Neoplasms      False   DPYD   \n",
       "\n",
       "   CHROM       POS Covered/Not_Covered  \n",
       "0   chr1  97515787             Covered  \n",
       "1   chr1  97305364             Covered  \n",
       "2   chr1  97699535             Covered  \n",
       "3   chr1  97691776             Covered  \n",
       "4   chr1  97515923             Covered  \n",
       "5   chr1  97450058             Covered  \n",
       "6   chr1  97450190             Covered  \n",
       "7   chr1  97450068             Covered  \n",
       "8   chr1  97450058             Covered  \n",
       "9   chr1  97573799             Covered  \n",
       "10  chr1  97079071             Covered  \n",
       "11  chr1  97305364             Covered  \n",
       "12  chr1  97699474             Covered  \n",
       "13  chr1  97593251             Covered  \n",
       "14  chr1  97579893         Not_Covered  \n",
       "15  chr1  97573943             Covered  \n",
       "16  chr1  97593343             Covered  \n",
       "17  chr1  97397224         Not_Covered  \n",
       "18  chr1  97515839             Covered  \n",
       "19  chr1  97234874             Covered  \n",
       "20  chr1  97579893         Not_Covered  \n",
       "21  chr1  97073844         Not_Covered  \n",
       "22  chr1  97515865             Covered  \n",
       "23  chr1  97098577             Covered  \n",
       "24  chr1  97234991             Covered  \n",
       "25  chr1  97450190             Covered  \n",
       "26  chr1  97082391             Covered  \n",
       "27  chr1  97699535             Covered  \n",
       "28  chr1  97515787             Covered  \n",
       "29  chr1  97883329             Covered  \n",
       "30  chr1  97137438         Not_Covered  \n",
       "31  chr1  97883329             Covered  \n",
       "32  chr1  97082391             Covered  \n",
       "33  chr1  97573863             Covered  \n",
       "34  chr1  97515839             Covered  \n",
       "35  chr1  97515787             Covered  \n",
       "36  chr1  97450068             Covered  \n",
       "37  chr1  97573863             Covered  \n",
       "38  chr1  97573943             Covered  \n",
       "39  chr1  97082391             Covered  \n",
       "40  chr1  97828265         Not_Covered  \n",
       "41  chr1  97740411             Covered  \n",
       "42  chr1  97515865             Covered  \n",
       "43  chr1  97515787             Covered  \n",
       "44  chr1  97579893         Not_Covered  \n",
       "45  chr1  97699474             Covered  \n",
       "46  chr1  97305364             Covered  \n",
       "47  chr1  97450058             Covered  \n",
       "48  chr1  97450066             Covered  \n",
       "49  chr1  97082391             Covered  \n",
       "50  chr1  97382318         Not_Covered  \n",
       "51  chr1  97595124             Covered  \n",
       "52  chr1  97450058             Covered  \n",
       "53  chr1  97450058             Covered  \n",
       "54  chr1  97721668             Covered  \n",
       "55  chr1  97696806         Not_Covered  \n",
       "56  chr1  97057448         Not_Covered  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(df_2, data, on = 'rsID', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8e3ecdd4",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_1.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/krishna_covered_pos/DYPD_gene_Capcetabine_flurouracil_oxaliplatin_new.xlsx', index=False)\n",
    "merged_2.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/krishna_covered_pos/DYPD_gene_Capcetabine_flurouracil_oxaliplatin_2.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75513976",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21a0bfe3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "eb4d5e40",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIF7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CWC27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MANBA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4482</th>\n",
       "      <td>KCNJ1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4483</th>\n",
       "      <td>MIR140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4484</th>\n",
       "      <td>NCKAP1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4485</th>\n",
       "      <td>SOCS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4486</th>\n",
       "      <td>NFE2L2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4487 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_symbol\n",
       "0           KIF7\n",
       "1            AGA\n",
       "2          CWC27\n",
       "3          SUMF1\n",
       "4          MANBA\n",
       "...          ...\n",
       "4482       KCNJ1\n",
       "4483      MIR140\n",
       "4484     NCKAP1L\n",
       "4485       SOCS1\n",
       "4486      NFE2L2\n",
       "\n",
       "[4487 rows x 1 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Genes for rare disorders (Orphanet DB).xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "14b605c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>Matching</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIF7</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGA</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CWC27</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SUMF1</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MANBA</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4482</th>\n",
       "      <td>KCNJ1</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4483</th>\n",
       "      <td>MIR140</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4484</th>\n",
       "      <td>NCKAP1L</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4485</th>\n",
       "      <td>SOCS1</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4486</th>\n",
       "      <td>NFE2L2</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4487 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_symbol Matching\n",
       "0           KIF7      Yes\n",
       "1            AGA      Yes\n",
       "2          CWC27      Yes\n",
       "3          SUMF1      Yes\n",
       "4          MANBA      Yes\n",
       "...          ...      ...\n",
       "4482       KCNJ1      Yes\n",
       "4483      MIR140       No\n",
       "4484     NCKAP1L      Yes\n",
       "4485       SOCS1      Yes\n",
       "4486      NFE2L2      Yes\n",
       "\n",
       "[4487 rows x 2 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['Matching'] = data['gene_symbol'].apply(lambda gene: 'Yes' if gene in df['gene_symbol'].values else 'No')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c20993e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/4487_genes_status.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c813aa97",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>37354</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4361101</td>\n",
       "      <td>4361381</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37355</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4361397</td>\n",
       "      <td>4361753</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37356</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4361895</td>\n",
       "      <td>4362291</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37357</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4376282</td>\n",
       "      <td>4376442</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37358</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4410840</td>\n",
       "      <td>4411000</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37359</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4417104</td>\n",
       "      <td>4417264</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37360</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4417990</td>\n",
       "      <td>4418150</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37361</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4420018</td>\n",
       "      <td>4420178</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37362</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4449217</td>\n",
       "      <td>4449377</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37363</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4452845</td>\n",
       "      <td>4453085</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37364</th>\n",
       "      <td>chr3</td>\n",
       "      <td>4466945</td>\n",
       "      <td>4467299</td>\n",
       "      <td>SUMF1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  Extended_Start_pos  Extended_End_pos gene_symbol\n",
       "37354       chr3             4361101           4361381       SUMF1\n",
       "37355       chr3             4361397           4361753       SUMF1\n",
       "37356       chr3             4361895           4362291       SUMF1\n",
       "37357       chr3             4376282           4376442       SUMF1\n",
       "37358       chr3             4410840           4411000       SUMF1\n",
       "37359       chr3             4417104           4417264       SUMF1\n",
       "37360       chr3             4417990           4418150       SUMF1\n",
       "37361       chr3             4420018           4420178       SUMF1\n",
       "37362       chr3             4449217           4449377       SUMF1\n",
       "37363       chr3             4452845           4453085       SUMF1\n",
       "37364       chr3             4466945           4467299       SUMF1"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['gene_symbol'] == 'SUMF1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "372a1a49",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39c352aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d7dd629",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "918cd833",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PharmGKB ID</th>\n",
       "      <th>Level</th>\n",
       "      <th>Variant</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Drugs</th>\n",
       "      <th>Phenotype Categories</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Pediatric</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1449167328</td>\n",
       "      <td>3</td>\n",
       "      <td>rs4902333</td>\n",
       "      <td>chr14</td>\n",
       "      <td>64909368.0</td>\n",
       "      <td>CHURC1</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1449167200</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12302749</td>\n",
       "      <td>chr12</td>\n",
       "      <td>6867132.0</td>\n",
       "      <td>SPSB2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1449167584</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2013169</td>\n",
       "      <td>chr5</td>\n",
       "      <td>140118020.0</td>\n",
       "      <td>PURA</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1447964236</td>\n",
       "      <td>3</td>\n",
       "      <td>rs6551665</td>\n",
       "      <td>chr4</td>\n",
       "      <td>61873823.0</td>\n",
       "      <td>ADGRL3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1450376443</td>\n",
       "      <td>3</td>\n",
       "      <td>rs28386840</td>\n",
       "      <td>chr16</td>\n",
       "      <td>55652906.0</td>\n",
       "      <td>SLC6A2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1449167211</td>\n",
       "      <td>3</td>\n",
       "      <td>rs10420097</td>\n",
       "      <td>chr19</td>\n",
       "      <td>57633193.0</td>\n",
       "      <td>ZNF211</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1449167339</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2295490</td>\n",
       "      <td>chr20</td>\n",
       "      <td>388261.0</td>\n",
       "      <td>TRIB3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1449167469</td>\n",
       "      <td>3</td>\n",
       "      <td>rs11552708</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7559238.0</td>\n",
       "      <td>SENP3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1451420300</td>\n",
       "      <td>3</td>\n",
       "      <td>CYP2D6*4; CYP2D6*5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>atomoxetine; methylphenidate</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1450374881</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2032582</td>\n",
       "      <td>chr7</td>\n",
       "      <td>87531302.0</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1447964250</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1947275</td>\n",
       "      <td>chr4</td>\n",
       "      <td>61878827.0</td>\n",
       "      <td>ADGRL3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1449167350</td>\n",
       "      <td>3</td>\n",
       "      <td>rs4627790</td>\n",
       "      <td>chr3</td>\n",
       "      <td>32259860.0</td>\n",
       "      <td>CMTM8</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1183689114</td>\n",
       "      <td>3</td>\n",
       "      <td>CYP2D6*2; CYP2D6*10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1449167480</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1061735</td>\n",
       "      <td>chr7</td>\n",
       "      <td>158733764.0</td>\n",
       "      <td>ESYT2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1449167228</td>\n",
       "      <td>3</td>\n",
       "      <td>rs3810818</td>\n",
       "      <td>chr16</td>\n",
       "      <td>4382028.0</td>\n",
       "      <td>CORO7</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1450374895</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2070762</td>\n",
       "      <td>chr11</td>\n",
       "      <td>2165105.0</td>\n",
       "      <td>TH</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1444608441</td>\n",
       "      <td>3</td>\n",
       "      <td>rs6280</td>\n",
       "      <td>chr3</td>\n",
       "      <td>114171968.0</td>\n",
       "      <td>DRD3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Autism Spectrum Disorder</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1449167300</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2799018</td>\n",
       "      <td>chr10</td>\n",
       "      <td>31913141.0</td>\n",
       "      <td>ARHGAP12</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1449167557</td>\n",
       "      <td>3</td>\n",
       "      <td>rs3761372</td>\n",
       "      <td>chr21</td>\n",
       "      <td>39371919.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1447964264</td>\n",
       "      <td>3</td>\n",
       "      <td>rs6813183</td>\n",
       "      <td>chr4</td>\n",
       "      <td>61903413.0</td>\n",
       "      <td>ADGRL3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1449167431</td>\n",
       "      <td>3</td>\n",
       "      <td>rs7858</td>\n",
       "      <td>chr9</td>\n",
       "      <td>14087770.0</td>\n",
       "      <td>NFIB</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1449167178</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2071421</td>\n",
       "      <td>chr22</td>\n",
       "      <td>50625988.0</td>\n",
       "      <td>ARSA</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1448104059</td>\n",
       "      <td>4</td>\n",
       "      <td>rs5569</td>\n",
       "      <td>chr16</td>\n",
       "      <td>55697923.0</td>\n",
       "      <td>SLC6A2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1449167568</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1061115</td>\n",
       "      <td>chr10</td>\n",
       "      <td>98417292.0</td>\n",
       "      <td>PYROXD2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1449167442</td>\n",
       "      <td>3</td>\n",
       "      <td>rs929740</td>\n",
       "      <td>chr5</td>\n",
       "      <td>73621913.0</td>\n",
       "      <td>ARHGEF28</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>1449167317</td>\n",
       "      <td>3</td>\n",
       "      <td>rs12409352</td>\n",
       "      <td>chr1</td>\n",
       "      <td>160030645.0</td>\n",
       "      <td>PIGM</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1447964278</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1355368</td>\n",
       "      <td>chr4</td>\n",
       "      <td>61914104.0</td>\n",
       "      <td>ADGRL3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>1449167453</td>\n",
       "      <td>3</td>\n",
       "      <td>rs9032</td>\n",
       "      <td>chr9</td>\n",
       "      <td>123104493.0</td>\n",
       "      <td>STRBP</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>1449167267</td>\n",
       "      <td>3</td>\n",
       "      <td>rs4805162</td>\n",
       "      <td>chr19</td>\n",
       "      <td>36183403.0</td>\n",
       "      <td>ZNF565</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>1449167529</td>\n",
       "      <td>3</td>\n",
       "      <td>rs757978</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241431686.0</td>\n",
       "      <td>FARP2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>1447964292</td>\n",
       "      <td>3</td>\n",
       "      <td>rs734644</td>\n",
       "      <td>chr4</td>\n",
       "      <td>61935010.0</td>\n",
       "      <td>ADGRL3</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>1449167147</td>\n",
       "      <td>3</td>\n",
       "      <td>rs10413455</td>\n",
       "      <td>chr19</td>\n",
       "      <td>57620255.0</td>\n",
       "      <td>ZNF134</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>1449167278</td>\n",
       "      <td>3</td>\n",
       "      <td>rs4562</td>\n",
       "      <td>chr19</td>\n",
       "      <td>7260420.0</td>\n",
       "      <td>ELP5</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>1451346385</td>\n",
       "      <td>3</td>\n",
       "      <td>rs71647871</td>\n",
       "      <td>chr16</td>\n",
       "      <td>55823658.0</td>\n",
       "      <td>CES1</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Dosage</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>981202090</td>\n",
       "      <td>3</td>\n",
       "      <td>rs71647871</td>\n",
       "      <td>chr16</td>\n",
       "      <td>55823658.0</td>\n",
       "      <td>CES1</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1449167409</td>\n",
       "      <td>3</td>\n",
       "      <td>rs751655</td>\n",
       "      <td>chr11</td>\n",
       "      <td>132623600.0</td>\n",
       "      <td>OPCML</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>1449167289</td>\n",
       "      <td>3</td>\n",
       "      <td>rs17712523</td>\n",
       "      <td>chr10</td>\n",
       "      <td>71777857.0</td>\n",
       "      <td>CDH23</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1449167546</td>\n",
       "      <td>3</td>\n",
       "      <td>rs17279558</td>\n",
       "      <td>chr8</td>\n",
       "      <td>63015187.0</td>\n",
       "      <td>GGH</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>1450376745</td>\n",
       "      <td>4</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>chr22</td>\n",
       "      <td>19963748.0</td>\n",
       "      <td>COMT</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1449167420</td>\n",
       "      <td>3</td>\n",
       "      <td>rs11559290</td>\n",
       "      <td>chr4</td>\n",
       "      <td>158680524.0</td>\n",
       "      <td>ETFDH</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1449167361</td>\n",
       "      <td>3</td>\n",
       "      <td>rs1048786</td>\n",
       "      <td>chr16</td>\n",
       "      <td>286916.0</td>\n",
       "      <td>PDIA2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>1449167106</td>\n",
       "      <td>3</td>\n",
       "      <td>rs17685420</td>\n",
       "      <td>chr8</td>\n",
       "      <td>22927888.0</td>\n",
       "      <td>PEBP4</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>1183703019</td>\n",
       "      <td>4</td>\n",
       "      <td>rs1800544</td>\n",
       "      <td>chr10</td>\n",
       "      <td>111076745.0</td>\n",
       "      <td>ADRA2A</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>1449167495</td>\n",
       "      <td>3</td>\n",
       "      <td>rs11553441</td>\n",
       "      <td>chr22</td>\n",
       "      <td>42516091.0</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1449167245</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2886059</td>\n",
       "      <td>chr3</td>\n",
       "      <td>126146923.0</td>\n",
       "      <td>ALDH1L1</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>1449167120</td>\n",
       "      <td>3</td>\n",
       "      <td>rs3210967</td>\n",
       "      <td>chr16</td>\n",
       "      <td>84975857.0</td>\n",
       "      <td>ZDHHC7</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>1444608238</td>\n",
       "      <td>3</td>\n",
       "      <td>rs4532</td>\n",
       "      <td>chr5</td>\n",
       "      <td>175443147.0</td>\n",
       "      <td>DRD1</td>\n",
       "      <td>dextroamphetamine; methylphenidate</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>1450374917</td>\n",
       "      <td>3</td>\n",
       "      <td>rs2283265</td>\n",
       "      <td>chr11</td>\n",
       "      <td>113414814.0</td>\n",
       "      <td>DRD2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>1449167256</td>\n",
       "      <td>3</td>\n",
       "      <td>rs9901675</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7581494.0</td>\n",
       "      <td>FXR2</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>1449167512</td>\n",
       "      <td>3</td>\n",
       "      <td>rs362272</td>\n",
       "      <td>chr4</td>\n",
       "      <td>3233253.0</td>\n",
       "      <td>HTT</td>\n",
       "      <td>methylphenidate</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>Attention Deficit Disorder with Hyperactivity</td>\n",
       "      <td>True</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PharmGKB ID  Level              Variant  CHROM          POS      Gene  \\\n",
       "0    1449167328      3            rs4902333  chr14   64909368.0    CHURC1   \n",
       "1    1449167200      3           rs12302749  chr12    6867132.0     SPSB2   \n",
       "2    1449167584      3            rs2013169   chr5  140118020.0      PURA   \n",
       "3    1447964236      3            rs6551665   chr4   61873823.0    ADGRL3   \n",
       "4    1450376443      3           rs28386840  chr16   55652906.0    SLC6A2   \n",
       "5    1449167211      3           rs10420097  chr19   57633193.0    ZNF211   \n",
       "6    1449167339      3            rs2295490  chr20     388261.0     TRIB3   \n",
       "7    1449167469      3           rs11552708  chr17    7559238.0     SENP3   \n",
       "8    1451420300      3   CYP2D6*4; CYP2D6*5    NaN          NaN    CYP2D6   \n",
       "9    1450374881      3            rs2032582   chr7   87531302.0     ABCB1   \n",
       "10   1447964250      3            rs1947275   chr4   61878827.0    ADGRL3   \n",
       "11   1449167350      3            rs4627790   chr3   32259860.0     CMTM8   \n",
       "12   1183689114      3  CYP2D6*2; CYP2D6*10    NaN          NaN    CYP2D6   \n",
       "13   1449167480      3            rs1061735   chr7  158733764.0     ESYT2   \n",
       "14   1449167228      3            rs3810818  chr16    4382028.0     CORO7   \n",
       "15   1450374895      3            rs2070762  chr11    2165105.0        TH   \n",
       "16   1444608441      3               rs6280   chr3  114171968.0      DRD3   \n",
       "17   1449167300      3            rs2799018  chr10   31913141.0  ARHGAP12   \n",
       "18   1449167557      3            rs3761372  chr21   39371919.0       NaN   \n",
       "19   1447964264      3            rs6813183   chr4   61903413.0    ADGRL3   \n",
       "20   1449167431      3               rs7858   chr9   14087770.0      NFIB   \n",
       "21   1449167178      3            rs2071421  chr22   50625988.0      ARSA   \n",
       "22   1448104059      4               rs5569  chr16   55697923.0    SLC6A2   \n",
       "23   1449167568      3            rs1061115  chr10   98417292.0   PYROXD2   \n",
       "24   1449167442      3             rs929740   chr5   73621913.0  ARHGEF28   \n",
       "25   1449167317      3           rs12409352   chr1  160030645.0      PIGM   \n",
       "26   1447964278      3            rs1355368   chr4   61914104.0    ADGRL3   \n",
       "27   1449167453      3               rs9032   chr9  123104493.0     STRBP   \n",
       "28   1449167267      3            rs4805162  chr19   36183403.0    ZNF565   \n",
       "29   1449167529      3             rs757978   chr2  241431686.0     FARP2   \n",
       "30   1447964292      3             rs734644   chr4   61935010.0    ADGRL3   \n",
       "31   1449167147      3           rs10413455  chr19   57620255.0    ZNF134   \n",
       "32   1449167278      3               rs4562  chr19    7260420.0      ELP5   \n",
       "33   1451346385      3           rs71647871  chr16   55823658.0      CES1   \n",
       "34    981202090      3           rs71647871  chr16   55823658.0      CES1   \n",
       "35   1449167409      3             rs751655  chr11  132623600.0     OPCML   \n",
       "36   1449167289      3           rs17712523  chr10   71777857.0     CDH23   \n",
       "37   1449167546      3           rs17279558   chr8   63015187.0       GGH   \n",
       "38   1450376745      4               rs4680  chr22   19963748.0      COMT   \n",
       "39   1449167420      3           rs11559290   chr4  158680524.0     ETFDH   \n",
       "40   1449167361      3            rs1048786  chr16     286916.0     PDIA2   \n",
       "41   1449167106      3           rs17685420   chr8   22927888.0     PEBP4   \n",
       "42   1183703019      4            rs1800544  chr10  111076745.0    ADRA2A   \n",
       "43   1449167495      3           rs11553441  chr22   42516091.0     RRP7A   \n",
       "44   1449167245      3            rs2886059   chr3  126146923.0   ALDH1L1   \n",
       "45   1449167120      3            rs3210967  chr16   84975857.0    ZDHHC7   \n",
       "46   1444608238      3               rs4532   chr5  175443147.0      DRD1   \n",
       "47   1450374917      3            rs2283265  chr11  113414814.0      DRD2   \n",
       "48   1449167256      3            rs9901675  chr17    7581494.0      FXR2   \n",
       "49   1449167512      3             rs362272   chr4    3233253.0       HTT   \n",
       "\n",
       "                                 Drugs Phenotype Categories  \\\n",
       "0                      methylphenidate             Efficacy   \n",
       "1                      methylphenidate             Efficacy   \n",
       "2                      methylphenidate             Efficacy   \n",
       "3                      methylphenidate             Efficacy   \n",
       "4                      methylphenidate             Efficacy   \n",
       "5                      methylphenidate             Efficacy   \n",
       "6                      methylphenidate             Efficacy   \n",
       "7                      methylphenidate             Efficacy   \n",
       "8         atomoxetine; methylphenidate        Metabolism/PK   \n",
       "9                      methylphenidate             Toxicity   \n",
       "10                     methylphenidate             Efficacy   \n",
       "11                     methylphenidate             Efficacy   \n",
       "12                     methylphenidate             Toxicity   \n",
       "13                     methylphenidate             Efficacy   \n",
       "14                     methylphenidate             Efficacy   \n",
       "15                     methylphenidate             Efficacy   \n",
       "16                     methylphenidate             Efficacy   \n",
       "17                     methylphenidate             Efficacy   \n",
       "18                     methylphenidate             Efficacy   \n",
       "19                     methylphenidate             Efficacy   \n",
       "20                     methylphenidate             Efficacy   \n",
       "21                     methylphenidate             Efficacy   \n",
       "22                     methylphenidate             Efficacy   \n",
       "23                     methylphenidate             Efficacy   \n",
       "24                     methylphenidate             Efficacy   \n",
       "25                     methylphenidate             Efficacy   \n",
       "26                     methylphenidate             Efficacy   \n",
       "27                     methylphenidate             Efficacy   \n",
       "28                     methylphenidate             Efficacy   \n",
       "29                     methylphenidate             Efficacy   \n",
       "30                     methylphenidate             Efficacy   \n",
       "31                     methylphenidate             Efficacy   \n",
       "32                     methylphenidate             Efficacy   \n",
       "33                     methylphenidate               Dosage   \n",
       "34                     methylphenidate        Metabolism/PK   \n",
       "35                     methylphenidate             Efficacy   \n",
       "36                     methylphenidate             Efficacy   \n",
       "37                     methylphenidate             Efficacy   \n",
       "38                     methylphenidate             Efficacy   \n",
       "39                     methylphenidate             Efficacy   \n",
       "40                     methylphenidate             Efficacy   \n",
       "41                     methylphenidate             Efficacy   \n",
       "42                     methylphenidate             Efficacy   \n",
       "43                     methylphenidate             Efficacy   \n",
       "44                     methylphenidate             Efficacy   \n",
       "45                     methylphenidate             Efficacy   \n",
       "46  dextroamphetamine; methylphenidate             Toxicity   \n",
       "47                     methylphenidate             Toxicity   \n",
       "48                     methylphenidate             Efficacy   \n",
       "49                     methylphenidate             Efficacy   \n",
       "\n",
       "                                        Phenotype  Pediatric  \\\n",
       "0   Attention Deficit Disorder with Hyperactivity       True   \n",
       "1   Attention Deficit Disorder with Hyperactivity       True   \n",
       "2   Attention Deficit Disorder with Hyperactivity       True   \n",
       "3   Attention Deficit Disorder with Hyperactivity       True   \n",
       "4   Attention Deficit Disorder with Hyperactivity       True   \n",
       "5   Attention Deficit Disorder with Hyperactivity       True   \n",
       "6   Attention Deficit Disorder with Hyperactivity       True   \n",
       "7   Attention Deficit Disorder with Hyperactivity       True   \n",
       "8                                             NaN      False   \n",
       "9   Attention Deficit Disorder with Hyperactivity       True   \n",
       "10  Attention Deficit Disorder with Hyperactivity       True   \n",
       "11  Attention Deficit Disorder with Hyperactivity       True   \n",
       "12  Attention Deficit Disorder with Hyperactivity       True   \n",
       "13  Attention Deficit Disorder with Hyperactivity       True   \n",
       "14  Attention Deficit Disorder with Hyperactivity       True   \n",
       "15  Attention Deficit Disorder with Hyperactivity       True   \n",
       "16                       Autism Spectrum Disorder       True   \n",
       "17  Attention Deficit Disorder with Hyperactivity       True   \n",
       "18  Attention Deficit Disorder with Hyperactivity       True   \n",
       "19  Attention Deficit Disorder with Hyperactivity       True   \n",
       "20  Attention Deficit Disorder with Hyperactivity       True   \n",
       "21  Attention Deficit Disorder with Hyperactivity       True   \n",
       "22  Attention Deficit Disorder with Hyperactivity       True   \n",
       "23  Attention Deficit Disorder with Hyperactivity       True   \n",
       "24  Attention Deficit Disorder with Hyperactivity       True   \n",
       "25  Attention Deficit Disorder with Hyperactivity       True   \n",
       "26  Attention Deficit Disorder with Hyperactivity       True   \n",
       "27  Attention Deficit Disorder with Hyperactivity       True   \n",
       "28  Attention Deficit Disorder with Hyperactivity       True   \n",
       "29  Attention Deficit Disorder with Hyperactivity       True   \n",
       "30  Attention Deficit Disorder with Hyperactivity       True   \n",
       "31  Attention Deficit Disorder with Hyperactivity       True   \n",
       "32  Attention Deficit Disorder with Hyperactivity       True   \n",
       "33  Attention Deficit Disorder with Hyperactivity       True   \n",
       "34                                            NaN      False   \n",
       "35  Attention Deficit Disorder with Hyperactivity       True   \n",
       "36  Attention Deficit Disorder with Hyperactivity       True   \n",
       "37  Attention Deficit Disorder with Hyperactivity       True   \n",
       "38  Attention Deficit Disorder with Hyperactivity       True   \n",
       "39  Attention Deficit Disorder with Hyperactivity       True   \n",
       "40  Attention Deficit Disorder with Hyperactivity       True   \n",
       "41  Attention Deficit Disorder with Hyperactivity       True   \n",
       "42  Attention Deficit Disorder with Hyperactivity       True   \n",
       "43  Attention Deficit Disorder with Hyperactivity       True   \n",
       "44  Attention Deficit Disorder with Hyperactivity       True   \n",
       "45  Attention Deficit Disorder with Hyperactivity       True   \n",
       "46  Attention Deficit Disorder with Hyperactivity       True   \n",
       "47  Attention Deficit Disorder with Hyperactivity       True   \n",
       "48  Attention Deficit Disorder with Hyperactivity       True   \n",
       "49  Attention Deficit Disorder with Hyperactivity       True   \n",
       "\n",
       "   Covered/Not_Covered  \n",
       "0          Not_Covered  \n",
       "1          Not_Covered  \n",
       "2          Not_Covered  \n",
       "3          Not_Covered  \n",
       "4          Not_Covered  \n",
       "5          Not_Covered  \n",
       "6              Covered  \n",
       "7              Covered  \n",
       "8          Not_Covered  \n",
       "9              Covered  \n",
       "10         Not_Covered  \n",
       "11         Not_Covered  \n",
       "12         Not_Covered  \n",
       "13         Not_Covered  \n",
       "14             Covered  \n",
       "15         Not_Covered  \n",
       "16             Covered  \n",
       "17         Not_Covered  \n",
       "18         Not_Covered  \n",
       "19         Not_Covered  \n",
       "20         Not_Covered  \n",
       "21             Covered  \n",
       "22             Covered  \n",
       "23         Not_Covered  \n",
       "24         Not_Covered  \n",
       "25             Covered  \n",
       "26         Not_Covered  \n",
       "27         Not_Covered  \n",
       "28             Covered  \n",
       "29             Covered  \n",
       "30             Covered  \n",
       "31             Covered  \n",
       "32         Not_Covered  \n",
       "33             Covered  \n",
       "34             Covered  \n",
       "35         Not_Covered  \n",
       "36             Covered  \n",
       "37         Not_Covered  \n",
       "38             Covered  \n",
       "39             Covered  \n",
       "40             Covered  \n",
       "41         Not_Covered  \n",
       "42         Not_Covered  \n",
       "43             Covered  \n",
       "44             Covered  \n",
       "45         Not_Covered  \n",
       "46             Covered  \n",
       "47         Not_Covered  \n",
       "48             Covered  \n",
       "49             Covered  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "data['Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "6ecdfdb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(R'C:/Users/GenepoweRx_Madhu/Downloads/Methylphenidate_covered_not_covered.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "776dcd9e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "efd8d2df",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80ca0dd6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "8f3e4d55",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/kalyani_mam_covered.bed', header=None, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a17b8e0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65489</td>\n",
       "      <td>65649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69007</td>\n",
       "      <td>70037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450710</td>\n",
       "      <td>451706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685686</td>\n",
       "      <td>686682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924401</td>\n",
       "      <td>924977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038781</td>\n",
       "      <td>25038941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041746</td>\n",
       "      <td>25041906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043888</td>\n",
       "      <td>25044048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622413</td>\n",
       "      <td>25624093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624397</td>\n",
       "      <td>25624557</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208911 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos\n",
       "0            chr1      65489     65649\n",
       "1            chr1      69007     70037\n",
       "2            chr1     450710    451706\n",
       "3            chr1     685686    686682\n",
       "4            chr1     924401    924977\n",
       "...           ...        ...       ...\n",
       "208906       chrY   25038781  25038941\n",
       "208907       chrY   25041746  25041906\n",
       "208908       chrY   25043888  25044048\n",
       "208909       chrY   25622413  25624093\n",
       "208910       chrY   25624397  25624557\n",
       "\n",
       "[208911 rows x 3 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/kalyani_mam_covered.bed', header=None, sep='\\t', error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "40bca555",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>155302</th>\n",
       "      <td>chr16</td>\n",
       "      <td>4380848</td>\n",
       "      <td>4382928</td>\n",
       "      <td>VASN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Extended_Start_pos  Extended_End_pos gene_symbol\n",
       "155302      chr16             4380848           4382928        VASN"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Value to check\n",
    "value = 4382028\n",
    "\n",
    "# Filter rows where the value lies between start and end positions\n",
    "filtered_df = df[(df['Extended_Start_pos'] <= value) & (df['Extended_End_pos'] >= value)]\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "63bf5b24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ccds_id=CCDS30547.1;ensembl_gene_id=ENSG00000186092;gene_symbol=OR4F5'"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_df.INFO.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2e50345",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "88f56a06",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/covered_clingeno_bed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40bc3e09",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "68ebd739",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chr1     20980\n",
       "chr2     16263\n",
       "chr3     12382\n",
       "chr17    12130\n",
       "chr12    12015\n",
       "chr11    11675\n",
       "chr19    11373\n",
       "chr6     10469\n",
       "chr7      9921\n",
       "chr5      9467\n",
       "chr10     8780\n",
       "chr16     8717\n",
       "chr9      8493\n",
       "chr4      8436\n",
       "chr15     7524\n",
       "chrX      7213\n",
       "chr8      7059\n",
       "chr14     6618\n",
       "chr20     5088\n",
       "chr22     4418\n",
       "chr13     3802\n",
       "chr18     3392\n",
       "chr21     2276\n",
       "chrY       420\n",
       "Name: chromosome, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.chromosome.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "93b68b94",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16</td>\n",
       "      <td>31779</td>\n",
       "      <td>rs141542623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12652713</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17751397</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17751406</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12652700</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165543</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248855917</td>\n",
       "      <td>rs13313088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165544</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248856006</td>\n",
       "      <td>rs13313009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165545</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248913954</td>\n",
       "      <td>rs12025760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165546</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248916897</td>\n",
       "      <td>rs74157349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165547</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248917420</td>\n",
       "      <td>rs77124333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>165548 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples    GENE  CHROM        POS           ID\n",
       "0       12652713     NaN  chr16      31779  rs141542623\n",
       "1       12652713   TUBB8  chr10      49286    rs6560829\n",
       "2       17751397   TUBB8  chr10      49286    rs6560829\n",
       "3       17751406   TUBB8  chr10      49286    rs6560829\n",
       "4       12652700   TUBB8  chr10      49286    rs6560829\n",
       "...          ...     ...    ...        ...          ...\n",
       "165543  17751406  ZNF692   chr1  248855917   rs13313088\n",
       "165544  17751406  ZNF692   chr1  248856006   rs13313009\n",
       "165545  17751397   PGBD2   chr1  248913954   rs12025760\n",
       "165546  17751397   PGBD2   chr1  248916897   rs74157349\n",
       "165547  17751397   PGBD2   chr1  248917420   rs77124333\n",
       "\n",
       "[165548 rows x 5 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_covered_list.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e8cf9ad",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0ee2ebe0",
   "metadata": {},
   "source": [
    "# check a dataset positions Covered/ not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d24b9432",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>Madhu_Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16</td>\n",
       "      <td>31779</td>\n",
       "      <td>rs141542623</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12652713</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17751397</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17751406</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12652700</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49286</td>\n",
       "      <td>rs6560829</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165543</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248855917</td>\n",
       "      <td>rs13313088</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165544</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248856006</td>\n",
       "      <td>rs13313009</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165545</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248913954</td>\n",
       "      <td>rs12025760</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165546</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248916897</td>\n",
       "      <td>rs74157349</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165547</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248917420</td>\n",
       "      <td>rs77124333</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>165548 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples    GENE  CHROM        POS           ID  \\\n",
       "0       12652713     NaN  chr16      31779  rs141542623   \n",
       "1       12652713   TUBB8  chr10      49286    rs6560829   \n",
       "2       17751397   TUBB8  chr10      49286    rs6560829   \n",
       "3       17751406   TUBB8  chr10      49286    rs6560829   \n",
       "4       12652700   TUBB8  chr10      49286    rs6560829   \n",
       "...          ...     ...    ...        ...          ...   \n",
       "165543  17751406  ZNF692   chr1  248855917   rs13313088   \n",
       "165544  17751406  ZNF692   chr1  248856006   rs13313009   \n",
       "165545  17751397   PGBD2   chr1  248913954   rs12025760   \n",
       "165546  17751397   PGBD2   chr1  248916897   rs74157349   \n",
       "165547  17751397   PGBD2   chr1  248917420   rs77124333   \n",
       "\n",
       "       Madhu_Covered/Not_Covered  \n",
       "0                    Not_Covered  \n",
       "1                        Covered  \n",
       "2                        Covered  \n",
       "3                        Covered  \n",
       "4                        Covered  \n",
       "...                          ...  \n",
       "165543                   Covered  \n",
       "165544               Not_Covered  \n",
       "165545               Not_Covered  \n",
       "165546                   Covered  \n",
       "165547                   Covered  \n",
       "\n",
       "[165548 rows x 6 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "data['Madhu_Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f26c12c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>Madhu_Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12652713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16</td>\n",
       "      <td>31779</td>\n",
       "      <td>rs141542623</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>17751406</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49289</td>\n",
       "      <td>rs6560830</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>12652700</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49289</td>\n",
       "      <td>rs6560830</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>17751397</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49289</td>\n",
       "      <td>rs6560830</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>12652713</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>chr10</td>\n",
       "      <td>49289</td>\n",
       "      <td>rs6560830</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165539</th>\n",
       "      <td>12652713</td>\n",
       "      <td>SH3BP5L</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248826485</td>\n",
       "      <td>rs11205415</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165540</th>\n",
       "      <td>17751406</td>\n",
       "      <td>SH3BP5L</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248826485</td>\n",
       "      <td>rs11205415</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165541</th>\n",
       "      <td>17751397</td>\n",
       "      <td>SH3BP5L</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248826485</td>\n",
       "      <td>rs11205415</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165544</th>\n",
       "      <td>17751406</td>\n",
       "      <td>ZNF692</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248856006</td>\n",
       "      <td>rs13313009</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165545</th>\n",
       "      <td>17751397</td>\n",
       "      <td>PGBD2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>248913954</td>\n",
       "      <td>rs12025760</td>\n",
       "      <td>Not_Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50465 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         samples     GENE  CHROM        POS           ID  \\\n",
       "0       12652713      NaN  chr16      31779  rs141542623   \n",
       "5       17751406    TUBB8  chr10      49289    rs6560830   \n",
       "6       12652700    TUBB8  chr10      49289    rs6560830   \n",
       "7       17751397    TUBB8  chr10      49289    rs6560830   \n",
       "8       12652713    TUBB8  chr10      49289    rs6560830   \n",
       "...          ...      ...    ...        ...          ...   \n",
       "165539  12652713  SH3BP5L   chr1  248826485   rs11205415   \n",
       "165540  17751406  SH3BP5L   chr1  248826485   rs11205415   \n",
       "165541  17751397  SH3BP5L   chr1  248826485   rs11205415   \n",
       "165544  17751406   ZNF692   chr1  248856006   rs13313009   \n",
       "165545  17751397    PGBD2   chr1  248913954   rs12025760   \n",
       "\n",
       "       Madhu_Covered/Not_Covered  \n",
       "0                    Not_Covered  \n",
       "5                    Not_Covered  \n",
       "6                    Not_Covered  \n",
       "7                    Not_Covered  \n",
       "8                    Not_Covered  \n",
       "...                          ...  \n",
       "165539               Not_Covered  \n",
       "165540               Not_Covered  \n",
       "165541               Not_Covered  \n",
       "165544               Not_Covered  \n",
       "165545               Not_Covered  \n",
       "\n",
       "[50465 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = data[data['Madhu_Covered/Not_Covered'] == 'Not_Covered']\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "cfe55fe9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24153\n"
     ]
    }
   ],
   "source": [
    "data1 = set(x['ID'])\n",
    "print(len(data1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64ed2c64",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65e41927",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8fb3df11",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Updated_genomic_db.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "34c056c9",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[26], line 9\u001b[0m\n\u001b[0;32m      6\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCovered\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m is_covered \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNot_Covered\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m      8\u001b[0m \u001b[38;5;66;03m# Apply the function to create the new column in dataset2\u001b[39;00m\n\u001b[1;32m----> 9\u001b[0m data[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMadhu_Covered/Not_Covered\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mdata\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcheck_coverage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m     10\u001b[0m data\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py:9568\u001b[0m, in \u001b[0;36mDataFrame.apply\u001b[1;34m(self, func, axis, raw, result_type, args, **kwargs)\u001b[0m\n\u001b[0;32m   9557\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcore\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mapply\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m frame_apply\n\u001b[0;32m   9559\u001b[0m op \u001b[38;5;241m=\u001b[39m frame_apply(\n\u001b[0;32m   9560\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m   9561\u001b[0m     func\u001b[38;5;241m=\u001b[39mfunc,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   9566\u001b[0m     kwargs\u001b[38;5;241m=\u001b[39mkwargs,\n\u001b[0;32m   9567\u001b[0m )\n\u001b[1;32m-> 9568\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mop\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39m__finalize__(\u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mapply\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:764\u001b[0m, in \u001b[0;36mFrameApply.apply\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    761\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mraw:\n\u001b[0;32m    762\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mapply_raw()\n\u001b[1;32m--> 764\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply_standard\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:891\u001b[0m, in \u001b[0;36mFrameApply.apply_standard\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    890\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mapply_standard\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[1;32m--> 891\u001b[0m     results, res_index \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply_series_generator\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    893\u001b[0m     \u001b[38;5;66;03m# wrap results\u001b[39;00m\n\u001b[0;32m    894\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mwrap_results(results, res_index)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\apply.py:907\u001b[0m, in \u001b[0;36mFrameApply.apply_series_generator\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    904\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m option_context(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmode.chained_assignment\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m    905\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m i, v \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(series_gen):\n\u001b[0;32m    906\u001b[0m         \u001b[38;5;66;03m# ignore SettingWithCopy here in case the user mutates\u001b[39;00m\n\u001b[1;32m--> 907\u001b[0m         results[i] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[43mv\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    908\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(results[i], ABCSeries):\n\u001b[0;32m    909\u001b[0m             \u001b[38;5;66;03m# If we have a view on v, we need to make a copy because\u001b[39;00m\n\u001b[0;32m    910\u001b[0m             \u001b[38;5;66;03m#  series_generator will swap out the underlying data\u001b[39;00m\n\u001b[0;32m    911\u001b[0m             results[i] \u001b[38;5;241m=\u001b[39m results[i]\u001b[38;5;241m.\u001b[39mcopy(deep\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "Cell \u001b[1;32mIn[26], line 5\u001b[0m, in \u001b[0;36mcheck_coverage\u001b[1;34m(row)\u001b[0m\n\u001b[0;32m      3\u001b[0m pos \u001b[38;5;241m=\u001b[39m row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpos\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m      4\u001b[0m chrom \u001b[38;5;241m=\u001b[39m row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mchromosome\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m----> 5\u001b[0m is_covered \u001b[38;5;241m=\u001b[39m \u001b[38;5;28many\u001b[39m((chrom \u001b[38;5;241m==\u001b[39m df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mchromosome\u001b[39m\u001b[38;5;124m'\u001b[39m]) \u001b[38;5;241m&\u001b[39m (\u001b[43mpos\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m>\u001b[39;49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mExtended_Start_pos\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m) \u001b[38;5;241m&\u001b[39m (pos \u001b[38;5;241m<\u001b[39m\u001b[38;5;241m=\u001b[39m df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mExtended_End_pos\u001b[39m\u001b[38;5;124m'\u001b[39m]))\n\u001b[0;32m      6\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCovered\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m is_covered \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNot_Covered\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\ops\\common.py:72\u001b[0m, in \u001b[0;36m_unpack_zerodim_and_defer.<locals>.new_method\u001b[1;34m(self, other)\u001b[0m\n\u001b[0;32m     68\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mNotImplemented\u001b[39m\n\u001b[0;32m     70\u001b[0m other \u001b[38;5;241m=\u001b[39m item_from_zerodim(other)\n\u001b[1;32m---> 72\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mmethod\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mother\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\arraylike.py:54\u001b[0m, in \u001b[0;36mOpsMixin.__le__\u001b[1;34m(self, other)\u001b[0m\n\u001b[0;32m     52\u001b[0m \u001b[38;5;129m@unpack_zerodim_and_defer\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m__le__\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     53\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__le__\u001b[39m(\u001b[38;5;28mself\u001b[39m, other):\n\u001b[1;32m---> 54\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cmp_method\u001b[49m\u001b[43m(\u001b[49m\u001b[43mother\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moperator\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mle\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py:6243\u001b[0m, in \u001b[0;36mSeries._cmp_method\u001b[1;34m(self, other, op)\u001b[0m\n\u001b[0;32m   6240\u001b[0m rvalues \u001b[38;5;241m=\u001b[39m extract_array(other, extract_numpy\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, extract_range\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m   6242\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m np\u001b[38;5;241m.\u001b[39merrstate(\u001b[38;5;28mall\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[1;32m-> 6243\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m \u001b[43mops\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcomparison_op\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mop\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   6245\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_construct_result(res_values, name\u001b[38;5;241m=\u001b[39mres_name)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\ops\\array_ops.py:290\u001b[0m, in \u001b[0;36mcomparison_op\u001b[1;34m(left, right, op)\u001b[0m\n\u001b[0;32m    287\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m comp_method_OBJECT_ARRAY(op, lvalues, rvalues)\n\u001b[0;32m    289\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 290\u001b[0m     res_values \u001b[38;5;241m=\u001b[39m \u001b[43m_na_arithmetic_op\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mis_cmp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[0;32m    292\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m res_values\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\ops\\array_ops.py:165\u001b[0m, in \u001b[0;36m_na_arithmetic_op\u001b[1;34m(left, right, op, is_cmp)\u001b[0m\n\u001b[0;32m    162\u001b[0m     func \u001b[38;5;241m=\u001b[39m partial(expressions\u001b[38;5;241m.\u001b[39mevaluate, op)\n\u001b[0;32m    164\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 165\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mleft\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mright\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    166\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[0;32m    167\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m is_cmp \u001b[38;5;129;01mand\u001b[39;00m (is_object_dtype(left\u001b[38;5;241m.\u001b[39mdtype) \u001b[38;5;129;01mor\u001b[39;00m is_object_dtype(right)):\n\u001b[0;32m    168\u001b[0m         \u001b[38;5;66;03m# For object dtype, fallback to a masked operation (only operating\u001b[39;00m\n\u001b[0;32m    169\u001b[0m         \u001b[38;5;66;03m#  on the non-missing values)\u001b[39;00m\n\u001b[0;32m    170\u001b[0m         \u001b[38;5;66;03m# Don't do this for comparisons, as that will handle complex numbers\u001b[39;00m\n\u001b[0;32m    171\u001b[0m         \u001b[38;5;66;03m#  incorrectly, see GH#32047\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\computation\\expressions.py:241\u001b[0m, in \u001b[0;36mevaluate\u001b[1;34m(op, a, b, use_numexpr)\u001b[0m\n\u001b[0;32m    238\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m op_str \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m    239\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m use_numexpr:\n\u001b[0;32m    240\u001b[0m         \u001b[38;5;66;03m# error: \"None\" not callable\u001b[39;00m\n\u001b[1;32m--> 241\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_evaluate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mop_str\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43ma\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mb\u001b[49m\u001b[43m)\u001b[49m  \u001b[38;5;66;03m# type: ignore[misc]\u001b[39;00m\n\u001b[0;32m    242\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m _evaluate_standard(op, op_str, a, b)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\computation\\expressions.py:129\u001b[0m, in \u001b[0;36m_evaluate_numexpr\u001b[1;34m(op, op_str, a, b)\u001b[0m\n\u001b[0;32m    126\u001b[0m     _store_test_result(result \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[0;32m    128\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m result \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m--> 129\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43m_evaluate_standard\u001b[49m\u001b[43m(\u001b[49m\u001b[43mop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mop_str\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43ma\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mb\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    131\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m result\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\computation\\expressions.py:70\u001b[0m, in \u001b[0;36m_evaluate_standard\u001b[1;34m(op, op_str, a, b)\u001b[0m\n\u001b[0;32m     68\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m _TEST_MODE:\n\u001b[0;32m     69\u001b[0m     _store_test_result(\u001b[38;5;28;01mFalse\u001b[39;00m)\n\u001b[1;32m---> 70\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mop\u001b[49m\u001b[43m(\u001b[49m\u001b[43ma\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mb\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# Function to check if 'pos' is covered or not_covered\n",
    "def check_coverage(row):\n",
    "    pos = row['pos']\n",
    "    chrom = row['chromosome']\n",
    "    is_covered = any((chrom == df['chromosome']) & (pos >= df['Extended_Start_pos']) & (pos <= df['Extended_End_pos']))\n",
    "    return 'Covered' if is_covered else 'Not_Covered'\n",
    "\n",
    "# Apply the function to create the new column in dataset2\n",
    "data['Madhu_Covered/Not_Covered'] = data.apply(check_coverage, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5682c1fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71575ee2",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = data[['Covered/Not_Covered', 'Madhu_Covered/Not_Covered', 'geneinfo', 'rsid', 'clndn', 'clinical_significance',\n",
    "            'review_status', 'molecular_consequence', 'clinvar_id', 'chromosome', 'pos', 'ref_allele', 'alt_allele', \n",
    "            'disease_name', 'conflicting_clinical_signifincance', 'clnsig_inclusive', 'dbvar_id', 'origin', 'is_indel', \n",
    "            'is_snp', 'is_structural_variant', 'is_transition', 'var_type', 'var_subtype', 'CLNDN Include/Exclude', \n",
    "            'Group\\n(Adult/Neo,Pediatric/Both', 'one liner', 'Section Keyword']]\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c7a2b34",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Updated_genomic_db.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "047572dd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "49a7a582",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "      <th>geneinfo</th>\n",
       "      <th>rsid</th>\n",
       "      <th>clndn</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>review_status</th>\n",
       "      <th>molecular_consequence</th>\n",
       "      <th>clinvar_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>disease_name</th>\n",
       "      <th>conflicting_clinical_signifincance</th>\n",
       "      <th>clnsig_inclusive</th>\n",
       "      <th>dbvar_id</th>\n",
       "      <th>origin</th>\n",
       "      <th>is_indel</th>\n",
       "      <th>is_snp</th>\n",
       "      <th>is_structural_variant</th>\n",
       "      <th>is_transition</th>\n",
       "      <th>var_type</th>\n",
       "      <th>var_subtype</th>\n",
       "      <th>CLNDN Include/Exclude</th>\n",
       "      <th>Group\\n(Adult/Neo,Pediatric/Both</th>\n",
       "      <th>one liner</th>\n",
       "      <th>Section Keyword</th>\n",
       "      <th>coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7143</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>CFTR-related disorders</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Neonatal/Pediatric</td>\n",
       "      <td>None</td>\n",
       "      <td>Lung Health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7143</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>Lung Health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7143</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Congenital bilateral aplasia of vas deferens f...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>Miscellaneous</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7143</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Inborn genetic diseases</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>DD</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7143</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>not provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>Miscellaneous</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>CFTR-related disorders</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Neonatal/Pediatric</td>\n",
       "      <td>None</td>\n",
       "      <td>Lung Health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Obstructive azoospermia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>None</td>\n",
       "      <td>RH_M</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Hereditary pancreatitis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Adult</td>\n",
       "      <td>Pancreatitis</td>\n",
       "      <td>GH_Pancrea</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Bronchiectasis with or without elevated sweat ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>Bronchiectasis</td>\n",
       "      <td>Lung Health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>Cystic fibrosis</td>\n",
       "      <td>Lung Health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Congenital bilateral aplasia of vas deferens f...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>Miscellaneous</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>Inborn genetic diseases</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>DD</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>ivacaftor response - Efficacy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>Miscellaneous</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>['Cystic_fibrosis']</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>7109</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117530975</td>\n",
       "      <td>G</td>\n",
       "      <td>[A]</td>\n",
       "      <td>not provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['209047:Pathogenic']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>Miscellaneous</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Covered</td>\n",
       "      <td>CFTR:1080</td>\n",
       "      <td>rs78756941</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['practice_guideline']</td>\n",
       "      <td>['SO:0001575|splice_donor_variant']</td>\n",
       "      <td>38799</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117531115</td>\n",
       "      <td>G</td>\n",
       "      <td>[T]</td>\n",
       "      <td>CFTR-related disorders</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transversion</td>\n",
       "      <td>Include</td>\n",
       "      <td>Neonatal/Pediatric</td>\n",
       "      <td>None</td>\n",
       "      <td>Lung Health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>100092</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515839</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>capecitabine response - Toxicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>100092</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515839</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>fluorouracil response - Toxicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>100092</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515839</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>Dihydropyrimidine dehydrogenase deficiency</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>None</td>\n",
       "      <td>FG_MDM</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>100092</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515839</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>not specified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001583|missense_variant']</td>\n",
       "      <td>100092</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97515839</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>not provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Exclude</td>\n",
       "      <td>--</td>\n",
       "      <td>None</td>\n",
       "      <td>Miscellaneous</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001627|intron_variant']</td>\n",
       "      <td>635263</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97579893</td>\n",
       "      <td>G</td>\n",
       "      <td>[C]</td>\n",
       "      <td>capecitabine response - Toxicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001627|intron_variant']</td>\n",
       "      <td>635263</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97579893</td>\n",
       "      <td>G</td>\n",
       "      <td>[C]</td>\n",
       "      <td>fluorouracil response - Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>DPYD:1806</td>\n",
       "      <td>rs75017182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>drug response</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001627|intron_variant']</td>\n",
       "      <td>635263</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97579893</td>\n",
       "      <td>G</td>\n",
       "      <td>[C]</td>\n",
       "      <td>fluorouracil response - Toxicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['germline']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>USH2A:7399</td>\n",
       "      <td>rs397518039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001574|splice_acceptor_variant']</td>\n",
       "      <td>48604</td>\n",
       "      <td>chr1</td>\n",
       "      <td>215877882</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>Retinitis pigmentosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>Retinitis pigmentosa</td>\n",
       "      <td>Eye / ENT health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>USH2A:7399</td>\n",
       "      <td>rs397518039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001574|splice_acceptor_variant']</td>\n",
       "      <td>48604</td>\n",
       "      <td>chr1</td>\n",
       "      <td>215877882</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>Retinal dystrophy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>Retinal dystrophy</td>\n",
       "      <td>Eye / ENT health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>USH2A:7399</td>\n",
       "      <td>rs397518039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001574|splice_acceptor_variant']</td>\n",
       "      <td>48604</td>\n",
       "      <td>chr1</td>\n",
       "      <td>215877882</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>Usher syndrome type 2A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>USH2A:7399</td>\n",
       "      <td>rs397518039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>['reviewed_by_expert_panel']</td>\n",
       "      <td>['SO:0001574|splice_acceptor_variant']</td>\n",
       "      <td>48604</td>\n",
       "      <td>chr1</td>\n",
       "      <td>215877882</td>\n",
       "      <td>T</td>\n",
       "      <td>[C]</td>\n",
       "      <td>Retinitis pigmentosa 39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['NA']</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>snp</td>\n",
       "      <td>Transition</td>\n",
       "      <td>Include</td>\n",
       "      <td>Both</td>\n",
       "      <td>Retinitis pigmentosa</td>\n",
       "      <td>Eye / ENT health</td>\n",
       "      <td>covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Covered/Not_Covered    geneinfo         rsid                clndn  \\\n",
       "0              Covered   CFTR:1080   rs75961395                  NaN   \n",
       "1              Covered   CFTR:1080   rs75961395                  NaN   \n",
       "2              Covered   CFTR:1080   rs75961395                  NaN   \n",
       "3              Covered   CFTR:1080   rs75961395                  NaN   \n",
       "4              Covered   CFTR:1080   rs75961395                  NaN   \n",
       "5              Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "6              Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "7              Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "8              Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "9              Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "10             Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "11             Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "12             Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "13             Covered   CFTR:1080   rs78655421  ['Cystic_fibrosis']   \n",
       "14             Covered   CFTR:1080   rs78756941                  NaN   \n",
       "15         Not_Covered   DPYD:1806    rs1801159                  NaN   \n",
       "16         Not_Covered   DPYD:1806    rs1801159                  NaN   \n",
       "17         Not_Covered   DPYD:1806    rs1801159                  NaN   \n",
       "18         Not_Covered   DPYD:1806    rs1801159                  NaN   \n",
       "19         Not_Covered   DPYD:1806    rs1801159                  NaN   \n",
       "20         Not_Covered   DPYD:1806   rs75017182                  NaN   \n",
       "21         Not_Covered   DPYD:1806   rs75017182                  NaN   \n",
       "22         Not_Covered   DPYD:1806   rs75017182                  NaN   \n",
       "23         Not_Covered  USH2A:7399  rs397518039                  NaN   \n",
       "24         Not_Covered  USH2A:7399  rs397518039                  NaN   \n",
       "25         Not_Covered  USH2A:7399  rs397518039                  NaN   \n",
       "26         Not_Covered  USH2A:7399  rs397518039                  NaN   \n",
       "\n",
       "   clinical_significance                 review_status  \\\n",
       "0             Pathogenic        ['practice_guideline']   \n",
       "1             Pathogenic        ['practice_guideline']   \n",
       "2             Pathogenic        ['practice_guideline']   \n",
       "3             Pathogenic        ['practice_guideline']   \n",
       "4             Pathogenic        ['practice_guideline']   \n",
       "5             Pathogenic        ['practice_guideline']   \n",
       "6             Pathogenic        ['practice_guideline']   \n",
       "7             Pathogenic        ['practice_guideline']   \n",
       "8             Pathogenic        ['practice_guideline']   \n",
       "9             Pathogenic        ['practice_guideline']   \n",
       "10            Pathogenic        ['practice_guideline']   \n",
       "11            Pathogenic        ['practice_guideline']   \n",
       "12            Pathogenic        ['practice_guideline']   \n",
       "13            Pathogenic        ['practice_guideline']   \n",
       "14            Pathogenic        ['practice_guideline']   \n",
       "15         drug response  ['reviewed_by_expert_panel']   \n",
       "16         drug response  ['reviewed_by_expert_panel']   \n",
       "17         drug response  ['reviewed_by_expert_panel']   \n",
       "18         drug response  ['reviewed_by_expert_panel']   \n",
       "19         drug response  ['reviewed_by_expert_panel']   \n",
       "20         drug response  ['reviewed_by_expert_panel']   \n",
       "21         drug response  ['reviewed_by_expert_panel']   \n",
       "22         drug response  ['reviewed_by_expert_panel']   \n",
       "23            Pathogenic  ['reviewed_by_expert_panel']   \n",
       "24            Pathogenic  ['reviewed_by_expert_panel']   \n",
       "25            Pathogenic  ['reviewed_by_expert_panel']   \n",
       "26            Pathogenic  ['reviewed_by_expert_panel']   \n",
       "\n",
       "                     molecular_consequence  clinvar_id chromosome        pos  \\\n",
       "0          ['SO:0001583|missense_variant']        7143       chr7  117509123   \n",
       "1          ['SO:0001583|missense_variant']        7143       chr7  117509123   \n",
       "2          ['SO:0001583|missense_variant']        7143       chr7  117509123   \n",
       "3          ['SO:0001583|missense_variant']        7143       chr7  117509123   \n",
       "4          ['SO:0001583|missense_variant']        7143       chr7  117509123   \n",
       "5          ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "6          ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "7          ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "8          ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "9          ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "10         ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "11         ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "12         ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "13         ['SO:0001583|missense_variant']        7109       chr7  117530975   \n",
       "14     ['SO:0001575|splice_donor_variant']       38799       chr7  117531115   \n",
       "15         ['SO:0001583|missense_variant']      100092       chr1   97515839   \n",
       "16         ['SO:0001583|missense_variant']      100092       chr1   97515839   \n",
       "17         ['SO:0001583|missense_variant']      100092       chr1   97515839   \n",
       "18         ['SO:0001583|missense_variant']      100092       chr1   97515839   \n",
       "19         ['SO:0001583|missense_variant']      100092       chr1   97515839   \n",
       "20           ['SO:0001627|intron_variant']      635263       chr1   97579893   \n",
       "21           ['SO:0001627|intron_variant']      635263       chr1   97579893   \n",
       "22           ['SO:0001627|intron_variant']      635263       chr1   97579893   \n",
       "23  ['SO:0001574|splice_acceptor_variant']       48604       chr1  215877882   \n",
       "24  ['SO:0001574|splice_acceptor_variant']       48604       chr1  215877882   \n",
       "25  ['SO:0001574|splice_acceptor_variant']       48604       chr1  215877882   \n",
       "26  ['SO:0001574|splice_acceptor_variant']       48604       chr1  215877882   \n",
       "\n",
       "   ref_allele alt_allele                                       disease_name  \\\n",
       "0           G        [A]                             CFTR-related disorders   \n",
       "1           G        [A]                                    Cystic fibrosis   \n",
       "2           G        [A]  Congenital bilateral aplasia of vas deferens f...   \n",
       "3           G        [A]                            Inborn genetic diseases   \n",
       "4           G        [A]                                       not provided   \n",
       "5           G        [A]                             CFTR-related disorders   \n",
       "6           G        [A]                            Obstructive azoospermia   \n",
       "7           G        [A]                            Hereditary pancreatitis   \n",
       "8           G        [A]  Bronchiectasis with or without elevated sweat ...   \n",
       "9           G        [A]                                    Cystic fibrosis   \n",
       "10          G        [A]  Congenital bilateral aplasia of vas deferens f...   \n",
       "11          G        [A]                            Inborn genetic diseases   \n",
       "12          G        [A]                      ivacaftor response - Efficacy   \n",
       "13          G        [A]                                       not provided   \n",
       "14          G        [T]                             CFTR-related disorders   \n",
       "15          T        [C]                   capecitabine response - Toxicity   \n",
       "16          T        [C]                   fluorouracil response - Toxicity   \n",
       "17          T        [C]         Dihydropyrimidine dehydrogenase deficiency   \n",
       "18          T        [C]                                      not specified   \n",
       "19          T        [C]                                       not provided   \n",
       "20          G        [C]                   capecitabine response - Toxicity   \n",
       "21          G        [C]                      fluorouracil response - Other   \n",
       "22          G        [C]                   fluorouracil response - Toxicity   \n",
       "23          T        [C]                               Retinitis pigmentosa   \n",
       "24          T        [C]                                  Retinal dystrophy   \n",
       "25          T        [C]                             Usher syndrome type 2A   \n",
       "26          T        [C]                            Retinitis pigmentosa 39   \n",
       "\n",
       "    conflicting_clinical_signifincance       clnsig_inclusive  dbvar_id  \\\n",
       "0                                  NaN                    NaN       NaN   \n",
       "1                                  NaN                    NaN       NaN   \n",
       "2                                  NaN                    NaN       NaN   \n",
       "3                                  NaN                    NaN       NaN   \n",
       "4                                  NaN                    NaN       NaN   \n",
       "5                                  NaN  ['209047:Pathogenic']       NaN   \n",
       "6                                  NaN  ['209047:Pathogenic']       NaN   \n",
       "7                                  NaN  ['209047:Pathogenic']       NaN   \n",
       "8                                  NaN  ['209047:Pathogenic']       NaN   \n",
       "9                                  NaN  ['209047:Pathogenic']       NaN   \n",
       "10                                 NaN  ['209047:Pathogenic']       NaN   \n",
       "11                                 NaN  ['209047:Pathogenic']       NaN   \n",
       "12                                 NaN  ['209047:Pathogenic']       NaN   \n",
       "13                                 NaN  ['209047:Pathogenic']       NaN   \n",
       "14                                 NaN                    NaN       NaN   \n",
       "15                                 NaN                    NaN       NaN   \n",
       "16                                 NaN                    NaN       NaN   \n",
       "17                                 NaN                    NaN       NaN   \n",
       "18                                 NaN                    NaN       NaN   \n",
       "19                                 NaN                    NaN       NaN   \n",
       "20                                 NaN                    NaN       NaN   \n",
       "21                                 NaN                    NaN       NaN   \n",
       "22                                 NaN                    NaN       NaN   \n",
       "23                                 NaN                    NaN       NaN   \n",
       "24                                 NaN                    NaN       NaN   \n",
       "25                                 NaN                    NaN       NaN   \n",
       "26                                 NaN                    NaN       NaN   \n",
       "\n",
       "          origin  is_indel  is_snp  is_structural_variant  is_transition  \\\n",
       "0         ['NA']         0       1                      0              1   \n",
       "1         ['NA']         0       1                      0              1   \n",
       "2         ['NA']         0       1                      0              1   \n",
       "3         ['NA']         0       1                      0              1   \n",
       "4         ['NA']         0       1                      0              1   \n",
       "5         ['NA']         0       1                      0              1   \n",
       "6         ['NA']         0       1                      0              1   \n",
       "7         ['NA']         0       1                      0              1   \n",
       "8         ['NA']         0       1                      0              1   \n",
       "9         ['NA']         0       1                      0              1   \n",
       "10        ['NA']         0       1                      0              1   \n",
       "11        ['NA']         0       1                      0              1   \n",
       "12        ['NA']         0       1                      0              1   \n",
       "13        ['NA']         0       1                      0              1   \n",
       "14        ['NA']         0       1                      0              0   \n",
       "15  ['germline']         0       1                      0              1   \n",
       "16  ['germline']         0       1                      0              1   \n",
       "17  ['germline']         0       1                      0              1   \n",
       "18  ['germline']         0       1                      0              1   \n",
       "19  ['germline']         0       1                      0              1   \n",
       "20  ['germline']         0       1                      0              0   \n",
       "21  ['germline']         0       1                      0              0   \n",
       "22  ['germline']         0       1                      0              0   \n",
       "23        ['NA']         0       1                      0              1   \n",
       "24        ['NA']         0       1                      0              1   \n",
       "25        ['NA']         0       1                      0              1   \n",
       "26        ['NA']         0       1                      0              1   \n",
       "\n",
       "   var_type   var_subtype CLNDN Include/Exclude  \\\n",
       "0       snp    Transition               Include   \n",
       "1       snp    Transition               Include   \n",
       "2       snp    Transition               Exclude   \n",
       "3       snp    Transition               Exclude   \n",
       "4       snp    Transition               Exclude   \n",
       "5       snp    Transition               Include   \n",
       "6       snp    Transition               Include   \n",
       "7       snp    Transition               Include   \n",
       "8       snp    Transition               Include   \n",
       "9       snp    Transition               Include   \n",
       "10      snp    Transition               Exclude   \n",
       "11      snp    Transition               Exclude   \n",
       "12      snp    Transition               Exclude   \n",
       "13      snp    Transition               Exclude   \n",
       "14      snp  Transversion               Include   \n",
       "15      snp    Transition                   NaN   \n",
       "16      snp    Transition                   NaN   \n",
       "17      snp    Transition               Include   \n",
       "18      snp    Transition                   NaN   \n",
       "19      snp    Transition               Exclude   \n",
       "20      snp  Transversion                   NaN   \n",
       "21      snp  Transversion                   NaN   \n",
       "22      snp  Transversion                   NaN   \n",
       "23      snp    Transition               Include   \n",
       "24      snp    Transition               Include   \n",
       "25      snp    Transition                   NaN   \n",
       "26      snp    Transition               Include   \n",
       "\n",
       "   Group\\n(Adult/Neo,Pediatric/Both             one liner   Section Keyword  \\\n",
       "0                Neonatal/Pediatric                  None       Lung Health   \n",
       "1                              Both       Cystic fibrosis       Lung Health   \n",
       "2                                --                  None     Miscellaneous   \n",
       "3                                --                  None                DD   \n",
       "4                                --                  None     Miscellaneous   \n",
       "5                Neonatal/Pediatric                  None       Lung Health   \n",
       "6                              Both                  None              RH_M   \n",
       "7                             Adult          Pancreatitis        GH_Pancrea   \n",
       "8                              Both        Bronchiectasis       Lung Health   \n",
       "9                              Both       Cystic fibrosis       Lung Health   \n",
       "10                               --                  None     Miscellaneous   \n",
       "11                               --                  None                DD   \n",
       "12                               --                  None     Miscellaneous   \n",
       "13                               --                  None     Miscellaneous   \n",
       "14               Neonatal/Pediatric                  None       Lung Health   \n",
       "15                              NaN                   NaN               NaN   \n",
       "16                              NaN                   NaN               NaN   \n",
       "17                             Both                  None            FG_MDM   \n",
       "18                              NaN                   NaN               NaN   \n",
       "19                               --                  None     Miscellaneous   \n",
       "20                              NaN                   NaN               NaN   \n",
       "21                              NaN                   NaN               NaN   \n",
       "22                              NaN                   NaN               NaN   \n",
       "23                             Both  Retinitis pigmentosa  Eye / ENT health   \n",
       "24                             Both     Retinal dystrophy  Eye / ENT health   \n",
       "25                              NaN                   NaN               NaN   \n",
       "26                             Both  Retinitis pigmentosa  Eye / ENT health   \n",
       "\n",
       "       coverage  \n",
       "0       covered  \n",
       "1       covered  \n",
       "2       covered  \n",
       "3       covered  \n",
       "4       covered  \n",
       "5       covered  \n",
       "6       covered  \n",
       "7       covered  \n",
       "8       covered  \n",
       "9       covered  \n",
       "10      covered  \n",
       "11      covered  \n",
       "12      covered  \n",
       "13      covered  \n",
       "14      covered  \n",
       "15      covered  \n",
       "16      covered  \n",
       "17      covered  \n",
       "18      covered  \n",
       "19      covered  \n",
       "20  not_covered  \n",
       "21  not_covered  \n",
       "22  not_covered  \n",
       "23      covered  \n",
       "24      covered  \n",
       "25      covered  \n",
       "26      covered  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Function to check if 'pos' is covered or not_covered\n",
    "def check_coverage(row):\n",
    "    pos = row['pos']\n",
    "    chrom = row['chromosome']\n",
    "    is_covered = any((chrom == df['chromosome']) & (pos >= df['Extended_Start_pos']) & (pos <= df['Extended_End_pos']))\n",
    "    return 'covered' if is_covered else 'not_covered'\n",
    "\n",
    "# Apply the function to create the new column in dataset2\n",
    "data['coverage'] = data.apply(check_coverage, axis=1)\n",
    "\n",
    "data\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af85177b",
   "metadata": {},
   "source": [
    "## wrong"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b70de6f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to check if 'pos' is covered or not_covered\n",
    "def check_coverage(row):\n",
    "    pos = row['pos']\n",
    "    is_covered = any((pos >= df['Extended_Start_pos']) & (pos <= df['Extended_End_pos']))\n",
    "    return 'covered' if is_covered else 'not_covered'\n",
    "\n",
    "# Apply the function to create the new column in dataset2\n",
    "data['coverage'] = data.apply(check_coverage, axis=1)\n",
    "\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2549168c",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Updated_genomic_db.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d6aab81",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0eb61704",
   "metadata": {},
   "source": [
    "# check the single position Covered/ not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9c080365",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Extended_Start_pos</th>\n",
       "      <th>Extended_End_pos</th>\n",
       "      <th>INFO</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chromosome, Extended_Start_pos, Extended_End_pos, INFO, gene_symbol]\n",
       "Index: []"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Value to check\n",
    "value = 31779\n",
    "\n",
    "# Filter rows where the value lies between start and end positions\n",
    "filtered_df = df[(df['Extended_Start_pos'] <= value) & (df['Extended_End_pos'] >= value)]\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16b60890",
   "metadata": {},
   "source": [
    "# check the multiple positions Covered/ not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1fb69a00",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "627579 Covered\n",
      "\n",
      "1436793 Covered\n",
      "\n",
      "1447646 Covered\n",
      "\n",
      "1447739 Covered\n",
      "\n",
      "1448375 Covered\n",
      "\n",
      "1588020 Covered\n",
      "\n",
      "1588028 Covered\n",
      "\n",
      "1588032 Covered\n",
      "\n",
      "1589645 Covered\n",
      "\n",
      "1589647 Covered\n",
      "\n",
      "1589661 Covered\n",
      "\n",
      "1589693 Covered\n",
      "\n",
      "1592156 Covered\n",
      "\n",
      "1816872 Covered\n",
      "\n",
      "1816897 Covered\n",
      "\n",
      "505927 Covered\n",
      "\n",
      "169394 Covered\n",
      "\n",
      "240834 Covered\n",
      "\n",
      "240886 Covered\n",
      "\n",
      "432137 Covered\n",
      "\n",
      "196140 Covered\n",
      "\n",
      "196153 Covered\n",
      "\n",
      "196161 Covered\n",
      "\n",
      "724415 Covered\n",
      "\n",
      "1979403 Covered\n",
      "\n",
      "2025553 Covered\n",
      "\n",
      "57035 Covered\n",
      "\n",
      "138279 Covered\n",
      "\n",
      "1781290 Covered\n",
      "\n",
      "2387652 Covered\n",
      "\n",
      "501653 Covered\n",
      "\n",
      "748202 Covered\n",
      "\n",
      "748266 Covered\n",
      "\n",
      "748361 Covered\n",
      "\n",
      "748445 Covered\n",
      "\n",
      "2835920 Covered\n",
      "\n",
      "196129 Covered\n",
      "\n",
      "196142 Covered\n",
      "\n",
      "196150 Covered\n",
      "\n",
      "292900 Covered\n",
      "\n",
      "548240 Covered\n",
      "\n",
      "2891829 Covered\n",
      "\n",
      "3255108 Covered\n",
      "\n",
      "3255111 Covered\n",
      "\n",
      "3283757 Covered\n",
      "\n",
      "3439362 Covered\n",
      "\n",
      "3224872 Covered\n",
      "\n",
      "500147 Covered\n",
      "\n",
      "577842 Covered\n",
      "\n",
      "615982 Covered\n",
      "\n",
      "616026 Covered\n",
      "\n",
      "650242 Covered\n",
      "\n"
     ]
    }
   ],
   "source": [
    "values = [627579,1436793,1447646,1447739,1448375,1588020,1588028,1588032,1589645,1589647,1589661,1589693,1592156,1816872,1816897,505927,169394,240834,240886,432137,196140,196153,196161,724415,1979403,2025553,57035,138279,1781290,2387652,501653,748202,748266,748361,748445,2835920,196129,196142,196150,292900,548240,2891829,3255108,3255111,3283757,3439362,3224872,500147,577842,615982,616026,650242]\n",
    "\n",
    "# Iterate over each value and filter rows\n",
    "for value in values:\n",
    "    filtered_df = df[(df['Extended_Start_pos'] <= value) & (df['Extended_End_pos'] >= value)]\n",
    "    if not filtered_df.empty:\n",
    "        print(f\"{value} Covered\")\n",
    "        filtered_df\n",
    "    else:\n",
    "        print(f\"{value} Not Covered\")\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab04f6fd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03ce44b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample data\n",
    "data = {\n",
    "    'start': [100, 200, 300, 400],\n",
    "    'end': [250, 350, 400, 500]\n",
    "}\n",
    "\n",
    "# Create a DataFrame from the data\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# List of values to check\n",
    "values = [275, 150, 450]\n",
    "\n",
    "# Iterate over each value and filter rows\n",
    "for value in values:\n",
    "    filtered_df = df[(df['start'] <= value) & (df['end'] >= value)]\n",
    "    if not filtered_df.empty:\n",
    "        print(f\"Value {value} lies between start and end positions in the DataFrame:\")\n",
    "        print(filtered_df)\n",
    "    else:\n",
    "        print(f\"Value {value} does not lie between start and end positions in the DataFrame.\")\n",
    "    print()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "fc2591d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>New_Change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135686:G:A</td>\n",
       "      <td>16_2135686_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135686:G:C</td>\n",
       "      <td>16_2135686_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135677:C:T</td>\n",
       "      <td>16_2135677_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135673:G:T</td>\n",
       "      <td>16_2135673_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135668:G:GGGCGGGCGC</td>\n",
       "      <td>16_2135668_G_GGGCGGGCGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089782:C:T</td>\n",
       "      <td>16_2089782_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089773:C:A</td>\n",
       "      <td>16_2089773_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089750:T:C</td>\n",
       "      <td>16_2089750_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1139</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089747:C:T</td>\n",
       "      <td>16_2089747_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1140</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089741:G:A</td>\n",
       "      <td>16_2089741_G_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1141 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene                    Change               New_Change\n",
       "0     PKD1           16:2135686:G:A           16_2135686_G_A\n",
       "1     PKD1           16:2135686:G:C           16_2135686_G_C\n",
       "2     PKD1           16:2135677:C:T           16_2135677_C_T\n",
       "3     PKD1           16:2135673:G:T           16_2135673_G_T\n",
       "4     PKD1  16:2135668:G:GGGCGGGCGC  16_2135668_G_GGGCGGGCGC\n",
       "...    ...                      ...                      ...\n",
       "1136  PKD1           16:2089782:C:T           16_2089782_C_T\n",
       "1137  PKD1           16:2089773:C:A           16_2089773_C_A\n",
       "1138  PKD1           16:2089750:T:C           16_2089750_T_C\n",
       "1139  PKD1           16:2089747:C:T           16_2089747_C_T\n",
       "1140  PKD1           16:2089741:G:A           16_2089741_G_A\n",
       "\n",
       "[1141 rows x 3 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/16chrom.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f429a1ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>New_Change</th>\n",
       "      <th>HGVSc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2135686_G_A</td>\n",
       "      <td>c.4C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2135686_G_C</td>\n",
       "      <td>c.4C&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2135683_G_T</td>\n",
       "      <td>c.7C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2135682_G_C</td>\n",
       "      <td>c.8C&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2135677_C_T</td>\n",
       "      <td>c.13G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3553</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2089747_C_T</td>\n",
       "      <td>c.12892G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3554</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2089741_G_A</td>\n",
       "      <td>c.12898C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3555</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2089731_G_A</td>\n",
       "      <td>c.12908C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3556</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>DEL_16_1984218_2112963</td>\n",
       "      <td>c.CNV_Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3557</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>DEL_16_2089726_2097241</td>\n",
       "      <td>c.CNV_Del</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3558 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene              New_Change       HGVSc\n",
       "0     PKD1          16_2135686_G_A      c.4C>T\n",
       "1     PKD1          16_2135686_G_C      c.4C>G\n",
       "2     PKD1          16_2135683_G_T      c.7C>A\n",
       "3     PKD1          16_2135682_G_C      c.8C>G\n",
       "4     PKD1          16_2135677_C_T     c.13G>A\n",
       "...    ...                     ...         ...\n",
       "3553  PKD1          16_2089747_C_T  c.12892G>A\n",
       "3554  PKD1          16_2089741_G_A  c.12898C>T\n",
       "3555  PKD1          16_2089731_G_A  c.12908C>T\n",
       "3556  PKD1  DEL_16_1984218_2112963   c.CNV_Del\n",
       "3557  PKD1  DEL_16_2089726_2097241   c.CNV_Del\n",
       "\n",
       "[3558 rows x 3 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/PKD2_PKD1.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "6d7ac46e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>New_Change</th>\n",
       "      <th>Gene</th>\n",
       "      <th>HGVSc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135686:G:A</td>\n",
       "      <td>16_2135686_G_A</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.4C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135686:G:C</td>\n",
       "      <td>16_2135686_G_C</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.4C&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135677:C:T</td>\n",
       "      <td>16_2135677_C_T</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.13G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135673:G:T</td>\n",
       "      <td>16_2135673_G_T</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.17C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2135668:G:GGGCGGGCGC</td>\n",
       "      <td>16_2135668_G_GGGCGGGCGC</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.13_21dup</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089782:C:T</td>\n",
       "      <td>16_2089782_C_T</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.12857G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089773:C:A</td>\n",
       "      <td>16_2089773_C_A</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.12866G&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089750:T:C</td>\n",
       "      <td>16_2089750_T_C</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.12889A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1139</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089747:C:T</td>\n",
       "      <td>16_2089747_C_T</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.12892G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1140</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16:2089741:G:A</td>\n",
       "      <td>16_2089741_G_A</td>\n",
       "      <td>PKD1</td>\n",
       "      <td>c.12898C&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1141 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene                    Change               New_Change  Gene       HGVSc\n",
       "0     PKD1           16:2135686:G:A           16_2135686_G_A  PKD1      c.4C>T\n",
       "1     PKD1           16:2135686:G:C           16_2135686_G_C  PKD1      c.4C>G\n",
       "2     PKD1           16:2135677:C:T           16_2135677_C_T  PKD1     c.13G>A\n",
       "3     PKD1           16:2135673:G:T           16_2135673_G_T  PKD1     c.17C>A\n",
       "4     PKD1  16:2135668:G:GGGCGGGCGC  16_2135668_G_GGGCGGGCGC  PKD1  c.13_21dup\n",
       "...    ...                      ...                      ...   ...         ...\n",
       "1136  PKD1           16:2089782:C:T           16_2089782_C_T  PKD1  c.12857G>A\n",
       "1137  PKD1           16:2089773:C:A           16_2089773_C_A  PKD1  c.12866G>T\n",
       "1138  PKD1           16:2089750:T:C           16_2089750_T_C  PKD1  c.12889A>G\n",
       "1139  PKD1           16:2089747:C:T           16_2089747_C_T  PKD1  c.12892G>A\n",
       "1140  PKD1           16:2089741:G:A           16_2089741_G_A  PKD1  c.12898C>T\n",
       "\n",
       "[1141 rows x 5 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.merge(data, df, on = 'New_Change', how = 'left', sort=False)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "d4992156",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>New_Change</th>\n",
       "      <th>HGVSc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3552</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>16_2089750_T_C</td>\n",
       "      <td>c.12889A&gt;G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene      New_Change       HGVSc\n",
       "3552  PKD1  16_2089750_T_C  c.12889A>G"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['New_Change'] == '16_2089750_T_C']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "2f5cfe35",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/CHROM_4.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec7b470f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "a932f9a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>227303904</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>227307878</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>227314020</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>227263944</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>227290810</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3233</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17033058</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3233</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17028723</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3233</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17028652</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3234</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17024076</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3234</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17024010</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6841 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS Literature\n",
       "0     chr2  227303904        Yes\n",
       "2     chr2  227307878        Yes\n",
       "2     chr2  227314020        Yes\n",
       "3     chr2  227263944        Yes\n",
       "4     chr2  227290810        Yes\n",
       "...    ...        ...        ...\n",
       "3233  chr1   17033058        Yes\n",
       "3233  chr1   17028723        Yes\n",
       "3233  chr1   17028652        Yes\n",
       "3234  chr1   17024076        Yes\n",
       "3234  chr1   17024010        Yes\n",
       "\n",
       "[6841 rows x 3 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Nephropathy_Lit_final_Positions_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5350ca55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1754</th>\n",
       "      <td>chr16</td>\n",
       "      <td>2114176</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM      POS Literature\n",
       "1754  chr16  2114176        Yes"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3[df_3['POS'] == \"2114176\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "58528177",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Alterations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1742P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2062V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2067D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A59S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C2488Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>895</th>\n",
       "      <td>V1398D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>K1423E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>897</th>\n",
       "      <td>K1436Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>S1463F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>899</th>\n",
       "      <td>R2429*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>900 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Alterations\n",
       "0        A1742P\n",
       "1        A2062V\n",
       "2        A2067D\n",
       "3          A59S\n",
       "4        C2488Y\n",
       "..          ...\n",
       "895      V1398D\n",
       "896      K1423E\n",
       "897      K1436Q\n",
       "898      S1463F\n",
       "899      R2429*\n",
       "\n",
       "[900 rows x 1 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Protein position_Onckokb.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "47f7d7f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Alterations</th>\n",
       "      <th>before_number</th>\n",
       "      <th>after_number</th>\n",
       "      <th>number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1742P</td>\n",
       "      <td>A</td>\n",
       "      <td>P</td>\n",
       "      <td>1742</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2062V</td>\n",
       "      <td>A</td>\n",
       "      <td>V</td>\n",
       "      <td>2062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2067D</td>\n",
       "      <td>A</td>\n",
       "      <td>D</td>\n",
       "      <td>2067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A59S</td>\n",
       "      <td>A</td>\n",
       "      <td>S</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C2488Y</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>2488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>895</th>\n",
       "      <td>V1398D</td>\n",
       "      <td>V</td>\n",
       "      <td>D</td>\n",
       "      <td>1398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>K1423E</td>\n",
       "      <td>K</td>\n",
       "      <td>E</td>\n",
       "      <td>1423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>897</th>\n",
       "      <td>K1436Q</td>\n",
       "      <td>K</td>\n",
       "      <td>Q</td>\n",
       "      <td>1436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>S1463F</td>\n",
       "      <td>S</td>\n",
       "      <td>F</td>\n",
       "      <td>1463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>899</th>\n",
       "      <td>R2429*</td>\n",
       "      <td>R</td>\n",
       "      <td>*</td>\n",
       "      <td>2429</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>900 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Alterations before_number after_number number\n",
       "0        A1742P             A            P   1742\n",
       "1        A2062V             A            V   2062\n",
       "2        A2067D             A            D   2067\n",
       "3          A59S             A            S     59\n",
       "4        C2488Y             C            Y   2488\n",
       "..          ...           ...          ...    ...\n",
       "895      V1398D             V            D   1398\n",
       "896      K1423E             K            E   1423\n",
       "897      K1436Q             K            Q   1436\n",
       "898      S1463F             S            F   1463\n",
       "899      R2429*             R            *   2429\n",
       "\n",
       "[900 rows x 4 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['before_number', 'after_number']] = df['Alterations'].str.split(r'(\\d+)', expand=True).iloc[:, [0, 2]]\n",
    "df['number'] = df['Alterations'].str.extract(r'(\\d+)')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "43f2a6b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/final_df.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a5a88f4d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Alterations</th>\n",
       "      <th>change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1742P</td>\n",
       "      <td>p.Ala1742Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2062V</td>\n",
       "      <td>p.Ala2062Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2067D</td>\n",
       "      <td>p.Ala2067Asp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A59S</td>\n",
       "      <td>p.Ala59Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C2488Y</td>\n",
       "      <td>p.Cys2488Tyr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>865</th>\n",
       "      <td>R1391S</td>\n",
       "      <td>p.Arg1391Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>V1398D</td>\n",
       "      <td>p.Val1398Asp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>K1423E</td>\n",
       "      <td>p.Lys1423Glu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>K1436Q</td>\n",
       "      <td>p.Lys1436Gln</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>869</th>\n",
       "      <td>S1463F</td>\n",
       "      <td>p.Ser1463Phe</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>870 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Alterations        change\n",
       "0        A1742P  p.Ala1742Pro\n",
       "1        A2062V  p.Ala2062Val\n",
       "2        A2067D  p.Ala2067Asp\n",
       "3          A59S    p.Ala59Ser\n",
       "4        C2488Y  p.Cys2488Tyr\n",
       "..          ...           ...\n",
       "865      R1391S  p.Arg1391Ser\n",
       "866      V1398D  p.Val1398Asp\n",
       "867      K1423E  p.Lys1423Glu\n",
       "868      K1436Q  p.Lys1436Gln\n",
       "869      S1463F  p.Ser1463Phe\n",
       "\n",
       "[870 rows x 2 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/final_df.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "515c8b10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A1742P</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A1742P mutation is located outside of ...</td>\n",
       "      <td>Mutations in the ATM gene lead to impaired ove...</td>\n",
       "      <td>PMID: 16014569</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A2062V</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2062V mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A2067D</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2067D mutation is located in the FAT ...</td>\n",
       "      <td>A-TWinnipeg: Pathogenesis of rare ATM missense...</td>\n",
       "      <td>PMID: 25077176</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A59S</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A59S mutation is located outside of a ...</td>\n",
       "      <td>Loss of DNA Damage Response in Neuroblastoma a...</td>\n",
       "      <td>PMID: 29059438</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>C2488Y</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM C2488Y mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>NF1</td>\n",
       "      <td>V1398D</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 V1398D mutation is located in the GTPa...</td>\n",
       "      <td>Assessment of the potential pathogenicity of m...</td>\n",
       "      <td>PMID: 22807134</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>NF1</td>\n",
       "      <td>K1423E</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>The NF1 K1423E mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>NF1</td>\n",
       "      <td>K1436Q</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 K1436Q mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>NF1</td>\n",
       "      <td>S1463F</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 S1463F mutation is located in the GTPa...</td>\n",
       "      <td>Exploring the somatic NF1 mutational spectrum ...</td>\n",
       "      <td>PMID: 22108604, PMID: 22807134</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>885</th>\n",
       "      <td>NF1</td>\n",
       "      <td>R2429*</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 R2429* mutation is located in the FAK ...</td>\n",
       "      <td>A genome-scale RNA interference screen implica...</td>\n",
       "      <td>PMID: 23288408, PMID: 14722917</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>886 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Alterations         Oncogenic          Mutation effect  \\\n",
       "0    ATM      A1742P  Likely Oncogenic  Likely Loss-of-function   \n",
       "1    ATM      A2062V  Likely Oncogenic  Likely Loss-of-function   \n",
       "2    ATM      A2067D  Likely Oncogenic  Likely Loss-of-function   \n",
       "3    ATM        A59S  Likely Oncogenic  Likely Loss-of-function   \n",
       "4    ATM      C2488Y  Likely Oncogenic  Likely Loss-of-function   \n",
       "..   ...         ...               ...                      ...   \n",
       "881  NF1      V1398D  Likely Oncogenic  Likely Loss-of-function   \n",
       "882  NF1      K1423E         Oncogenic         Loss-of-function   \n",
       "883  NF1      K1436Q  Likely Oncogenic  Likely Loss-of-function   \n",
       "884  NF1      S1463F  Likely Oncogenic  Likely Loss-of-function   \n",
       "885  NF1      R2429*  Likely Oncogenic  Likely Loss-of-function   \n",
       "\n",
       "                      Description from OncoKB_Database  \\\n",
       "0    The ATM A1742P mutation is located outside of ...   \n",
       "1    The ATM A2062V mutation is located in the FAT ...   \n",
       "2    The ATM A2067D mutation is located in the FAT ...   \n",
       "3    The ATM A59S mutation is located outside of a ...   \n",
       "4    The ATM C2488Y mutation is located in the FAT ...   \n",
       "..                                                 ...   \n",
       "881  The NF1 V1398D mutation is located in the GTPa...   \n",
       "882  The NF1 K1423E mutation is located in the GTPa...   \n",
       "883  The NF1 K1436Q mutation is located in the GTPa...   \n",
       "884  The NF1 S1463F mutation is located in the GTPa...   \n",
       "885  The NF1 R2429* mutation is located in the FAK ...   \n",
       "\n",
       "                                             Citations  \\\n",
       "0    Mutations in the ATM gene lead to impaired ove...   \n",
       "1    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "2    A-TWinnipeg: Pathogenesis of rare ATM missense...   \n",
       "3    Loss of DNA Damage Response in Neuroblastoma a...   \n",
       "4    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "..                                                 ...   \n",
       "881  Assessment of the potential pathogenicity of m...   \n",
       "882  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "883  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "884  Exploring the somatic NF1 mutational spectrum ...   \n",
       "885  A genome-scale RNA interference screen implica...   \n",
       "\n",
       "                                              PMID  Count  \n",
       "0                                   PMID: 16014569    1.0  \n",
       "1                                   PMID: 23585524    1.0  \n",
       "2                                   PMID: 25077176    1.0  \n",
       "3                                   PMID: 29059438    1.0  \n",
       "4                                   PMID: 23585524    1.0  \n",
       "..                                             ...    ...  \n",
       "881                                 PMID: 22807134    1.0  \n",
       "882  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  \n",
       "883  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  \n",
       "884                 PMID: 22108604, PMID: 22807134    2.0  \n",
       "885                 PMID: 23288408, PMID: 14722917    2.0  \n",
       "\n",
       "[886 rows x 8 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Oncokb Database.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5f6fa791",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "      <th>change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A1742P</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A1742P mutation is located outside of ...</td>\n",
       "      <td>Mutations in the ATM gene lead to impaired ove...</td>\n",
       "      <td>PMID: 16014569</td>\n",
       "      <td>1.0</td>\n",
       "      <td>p.Ala1742Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A2062V</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2062V mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "      <td>p.Ala2062Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A2067D</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2067D mutation is located in the FAT ...</td>\n",
       "      <td>A-TWinnipeg: Pathogenesis of rare ATM missense...</td>\n",
       "      <td>PMID: 25077176</td>\n",
       "      <td>1.0</td>\n",
       "      <td>p.Ala2067Asp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A59S</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A59S mutation is located outside of a ...</td>\n",
       "      <td>Loss of DNA Damage Response in Neuroblastoma a...</td>\n",
       "      <td>PMID: 29059438</td>\n",
       "      <td>1.0</td>\n",
       "      <td>p.Ala59Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>C2488Y</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM C2488Y mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "      <td>p.Cys2488Tyr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864</th>\n",
       "      <td>NF1</td>\n",
       "      <td>R1391S</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 R1391S mutation is located in the GTPa...</td>\n",
       "      <td>Mutational and functional analysis of the neur...</td>\n",
       "      <td>PMID: 9003501, PMID: 16513807, PMID: 9668168</td>\n",
       "      <td>3.0</td>\n",
       "      <td>p.Arg1391Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>865</th>\n",
       "      <td>NF1</td>\n",
       "      <td>V1398D</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 V1398D mutation is located in the GTPa...</td>\n",
       "      <td>Assessment of the potential pathogenicity of m...</td>\n",
       "      <td>PMID: 22807134</td>\n",
       "      <td>1.0</td>\n",
       "      <td>p.Val1398Asp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>NF1</td>\n",
       "      <td>K1423E</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>The NF1 K1423E mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "      <td>p.Lys1423Glu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>NF1</td>\n",
       "      <td>K1436Q</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 K1436Q mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "      <td>p.Lys1436Gln</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>NF1</td>\n",
       "      <td>S1463F</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 S1463F mutation is located in the GTPa...</td>\n",
       "      <td>Exploring the somatic NF1 mutational spectrum ...</td>\n",
       "      <td>PMID: 22108604, PMID: 22807134</td>\n",
       "      <td>2.0</td>\n",
       "      <td>p.Ser1463Phe</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>869 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Alterations         Oncogenic          Mutation effect  \\\n",
       "0    ATM      A1742P  Likely Oncogenic  Likely Loss-of-function   \n",
       "1    ATM      A2062V  Likely Oncogenic  Likely Loss-of-function   \n",
       "2    ATM      A2067D  Likely Oncogenic  Likely Loss-of-function   \n",
       "3    ATM        A59S  Likely Oncogenic  Likely Loss-of-function   \n",
       "4    ATM      C2488Y  Likely Oncogenic  Likely Loss-of-function   \n",
       "..   ...         ...               ...                      ...   \n",
       "864  NF1      R1391S  Likely Oncogenic  Likely Loss-of-function   \n",
       "865  NF1      V1398D  Likely Oncogenic  Likely Loss-of-function   \n",
       "866  NF1      K1423E         Oncogenic         Loss-of-function   \n",
       "867  NF1      K1436Q  Likely Oncogenic  Likely Loss-of-function   \n",
       "868  NF1      S1463F  Likely Oncogenic  Likely Loss-of-function   \n",
       "\n",
       "                      Description from OncoKB_Database  \\\n",
       "0    The ATM A1742P mutation is located outside of ...   \n",
       "1    The ATM A2062V mutation is located in the FAT ...   \n",
       "2    The ATM A2067D mutation is located in the FAT ...   \n",
       "3    The ATM A59S mutation is located outside of a ...   \n",
       "4    The ATM C2488Y mutation is located in the FAT ...   \n",
       "..                                                 ...   \n",
       "864  The NF1 R1391S mutation is located in the GTPa...   \n",
       "865  The NF1 V1398D mutation is located in the GTPa...   \n",
       "866  The NF1 K1423E mutation is located in the GTPa...   \n",
       "867  The NF1 K1436Q mutation is located in the GTPa...   \n",
       "868  The NF1 S1463F mutation is located in the GTPa...   \n",
       "\n",
       "                                             Citations  \\\n",
       "0    Mutations in the ATM gene lead to impaired ove...   \n",
       "1    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "2    A-TWinnipeg: Pathogenesis of rare ATM missense...   \n",
       "3    Loss of DNA Damage Response in Neuroblastoma a...   \n",
       "4    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "..                                                 ...   \n",
       "864  Mutational and functional analysis of the neur...   \n",
       "865  Assessment of the potential pathogenicity of m...   \n",
       "866  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "867  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "868  Exploring the somatic NF1 mutational spectrum ...   \n",
       "\n",
       "                                              PMID  Count        change  \n",
       "0                                   PMID: 16014569    1.0  p.Ala1742Pro  \n",
       "1                                   PMID: 23585524    1.0  p.Ala2062Val  \n",
       "2                                   PMID: 25077176    1.0  p.Ala2067Asp  \n",
       "3                                   PMID: 29059438    1.0    p.Ala59Ser  \n",
       "4                                   PMID: 23585524    1.0  p.Cys2488Tyr  \n",
       "..                                             ...    ...           ...  \n",
       "864   PMID: 9003501, PMID: 16513807, PMID: 9668168    3.0  p.Arg1391Ser  \n",
       "865                                 PMID: 22807134    1.0  p.Val1398Asp  \n",
       "866  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  p.Lys1423Glu  \n",
       "867  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  p.Lys1436Gln  \n",
       "868                 PMID: 22108604, PMID: 22807134    2.0  p.Ser1463Phe  \n",
       "\n",
       "[869 rows x 9 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.merge(data, df, on = 'Alterations', how = 'inner', sort=False)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "19123088",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/final_onco.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fb138f1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45e7ae63",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "2727ab9a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala1742Pro</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A1742P mutation is located outside of ...</td>\n",
       "      <td>Mutations in the ATM gene lead to impaired ove...</td>\n",
       "      <td>PMID: 16014569</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala2062Val</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2062V mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala2067Asp</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2067D mutation is located in the FAT ...</td>\n",
       "      <td>A-TWinnipeg: Pathogenesis of rare ATM missense...</td>\n",
       "      <td>PMID: 25077176</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala59Ser</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A59S mutation is located outside of a ...</td>\n",
       "      <td>Loss of DNA Damage Response in Neuroblastoma a...</td>\n",
       "      <td>PMID: 29059438</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Cys2488Tyr</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM C2488Y mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Names   Alterations         Oncogenic          Mutation effect  \\\n",
       "0        ATM  p.Ala1742Pro  Likely Oncogenic  Likely Loss-of-function   \n",
       "1        ATM  p.Ala2062Val  Likely Oncogenic  Likely Loss-of-function   \n",
       "2        ATM  p.Ala2067Asp  Likely Oncogenic  Likely Loss-of-function   \n",
       "3        ATM    p.Ala59Ser  Likely Oncogenic  Likely Loss-of-function   \n",
       "4        ATM  p.Cys2488Tyr  Likely Oncogenic  Likely Loss-of-function   \n",
       "\n",
       "                    Description from OncoKB_Database  \\\n",
       "0  The ATM A1742P mutation is located outside of ...   \n",
       "1  The ATM A2062V mutation is located in the FAT ...   \n",
       "2  The ATM A2067D mutation is located in the FAT ...   \n",
       "3  The ATM A59S mutation is located outside of a ...   \n",
       "4  The ATM C2488Y mutation is located in the FAT ...   \n",
       "\n",
       "                                           Citations            PMID  Count  \n",
       "0  Mutations in the ATM gene lead to impaired ove...  PMID: 16014569    1.0  \n",
       "1  ATM mutations uniformly lead to ATM dysfunctio...  PMID: 23585524    1.0  \n",
       "2  A-TWinnipeg: Pathogenesis of rare ATM missense...  PMID: 25077176    1.0  \n",
       "3  Loss of DNA Damage Response in Neuroblastoma a...  PMID: 29059438    1.0  \n",
       "4  ATM mutations uniformly lead to ATM dysfunctio...  PMID: 23585524    1.0  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/final_onco.xlsx')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a180fca",
   "metadata": {},
   "source": [
    "# ONCO samples code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bac47ba0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Identifier</th>\n",
       "      <th>Variants to track</th>\n",
       "      <th>Filter</th>\n",
       "      <th>Allelic Depths (AD)</th>\n",
       "      <th>Variant Allele Fraction</th>\n",
       "      <th>Genotype Qualities (GQ)</th>\n",
       "      <th>0/1 Genotypes (GT)</th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Sequence Ontology (Combined)</th>\n",
       "      <th>Effect (Combined)</th>\n",
       "      <th>N of 4 Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Transcript Name (Clinically Relevant)</th>\n",
       "      <th>HGVS c. (Clinically Relevant)</th>\n",
       "      <th>HGVS p. (Clinically Relevant)</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27192</th>\n",
       "      <td>13:32332592</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39,32</td>\n",
       "      <td>0.450704</td>\n",
       "      <td>99</td>\n",
       "      <td>0/1</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>NM_000059.4:c.1114A&gt;C</td>\n",
       "      <td>NP_000050.3:p.Asn372His</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>p.Asn372His</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>The BRCA2 N372H mutation is located in the P/C...</td>\n",
       "      <td>Functional assays for classification of BRCA2 ...</td>\n",
       "      <td>PMID: 18451181, PMID: 15695382</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27198</th>\n",
       "      <td>13:32355250</td>\n",
       "      <td>T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0,101</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>99</td>\n",
       "      <td>1/1</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>NM_000059.4:c.7397T&gt;C</td>\n",
       "      <td>NP_000050.3:p.Val2466Ala</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>p.Val2466Ala</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>The BRCA2 V2466A mutation is located in the FA...</td>\n",
       "      <td>Assessment of the Clinical Relevance of BRCA2 ...</td>\n",
       "      <td>PMID: 29394989</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33666</th>\n",
       "      <td>17:43071077</td>\n",
       "      <td>T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0,104</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>99</td>\n",
       "      <td>1/1</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>NM_007294.4:c.4837A&gt;G</td>\n",
       "      <td>NP_009225.1:p.Ser1613Gly</td>\n",
       "      <td>NM_007298.3,NM_007299.4,NM_007297.4,NM_007300....</td>\n",
       "      <td>p.Ser1613Gly</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>The BRCA1 S1613G mutation is likely neutral. I...</td>\n",
       "      <td>Human BRCA1 inhibits growth in yeast: potentia...</td>\n",
       "      <td>PMID: 9159158</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Chr:Pos Ref/Alt  Identifier  Variants to track Filter  \\\n",
       "27192  13:32332592     A/C         NaN              False    NaN   \n",
       "27198  13:32355250     T/C         NaN              False    NaN   \n",
       "33666  17:43071077     T/C         NaN              False    NaN   \n",
       "\n",
       "      Allelic Depths (AD)  Variant Allele Fraction  Genotype Qualities (GQ)  \\\n",
       "27192               39,32                 0.450704                       99   \n",
       "27198               0,101                 1.000000                       99   \n",
       "33666               0,104                 1.000000                       99   \n",
       "\n",
       "      0/1 Genotypes (GT) Gene Names Sequence Ontology (Combined)  \\\n",
       "27192                0/1      BRCA2             missense_variant   \n",
       "27198                1/1      BRCA2             missense_variant   \n",
       "33666                1/1      BRCA1             missense_variant   \n",
       "\n",
       "      Effect (Combined) N of 4 Predicted Splicing Disrupted (Combined)  \\\n",
       "27192          Missense                                            NaN   \n",
       "27198          Missense                                            NaN   \n",
       "33666          Missense                                            NaN   \n",
       "\n",
       "      Predicted Splicing Disrupted (Combined)  \\\n",
       "27192                                     NaN   \n",
       "27198                                     NaN   \n",
       "33666                                     NaN   \n",
       "\n",
       "      Transcript Name (Clinically Relevant) HGVS c. (Clinically Relevant)  \\\n",
       "27192                           NM_000059.4         NM_000059.4:c.1114A>C   \n",
       "27198                           NM_000059.4         NM_000059.4:c.7397T>C   \n",
       "33666                           NM_007294.4         NM_007294.4:c.4837A>G   \n",
       "\n",
       "      HGVS p. (Clinically Relevant)  \\\n",
       "27192       NP_000050.3:p.Asn372His   \n",
       "27198      NP_000050.3:p.Val2466Ala   \n",
       "33666      NP_009225.1:p.Ser1613Gly   \n",
       "\n",
       "                                         Transcript Name   Alterations  \\\n",
       "27192                                        NM_000059.4   p.Asn372His   \n",
       "27198                                        NM_000059.4  p.Val2466Ala   \n",
       "33666  NM_007298.3,NM_007299.4,NM_007297.4,NM_007300....  p.Ser1613Gly   \n",
       "\n",
       "            Oncogenic Mutation effect  \\\n",
       "27192  Likely Neutral  Likely Neutral   \n",
       "27198  Likely Neutral  Likely Neutral   \n",
       "33666  Likely Neutral  Likely Neutral   \n",
       "\n",
       "                        Description from OncoKB_Database  \\\n",
       "27192  The BRCA2 N372H mutation is located in the P/C...   \n",
       "27198  The BRCA2 V2466A mutation is located in the FA...   \n",
       "33666  The BRCA1 S1613G mutation is likely neutral. I...   \n",
       "\n",
       "                                               Citations  \\\n",
       "27192  Functional assays for classification of BRCA2 ...   \n",
       "27198  Assessment of the Clinical Relevance of BRCA2 ...   \n",
       "33666  Human BRCA1 inhibits growth in yeast: potentia...   \n",
       "\n",
       "                                 PMID  Count  \n",
       "27192  PMID: 18451181, PMID: 15695382    1.0  \n",
       "27198                  PMID: 29394989    1.0  \n",
       "33666                   PMID: 9159158    1.0  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/final_onco.xlsx')\n",
    "\n",
    "data1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/ONCO_17_samples/input/Germline_KHULTRGPCSP1 - KHULTRGPCSP1_S76_L004_R.xlsx', header=1, usecols=list(range(18)))\n",
    "data1['Alterations'] = data1['HGVS p. (Clinically Relevant)'].str.split(':').str[1]\n",
    "\n",
    "data1['Gene Match'] = 'No'\n",
    "data1['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in data1['Gene Names'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df['Gene Names'].values:\n",
    "                data1.at[index, 'Gene Match'] = 'Yes'\n",
    "                data1.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "df = df.rename({'Gene Names':'Matched_Gene'}, axis=1)\n",
    "mer = pd.merge(data1, df, on = ['Matched_Gene', 'Alterations'], how = 'left', sort=False)\n",
    "mer = mer.dropna(subset=['Oncogenic'])\n",
    "mer = mer.drop(['Gene Match', 'Matched_Gene'], axis=1)\n",
    "mer.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/ONCO_17_samples/Germline_KHULTRGPCSP1 - KHULTRGPCSP1_S76_L004_R.xlsx', index=False)\n",
    "mer"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a9ebb26",
   "metadata": {},
   "source": [
    "# --------------------------------------------------------------------------------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d4b28bc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f7bdadc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "e50c13ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13, 25)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mer.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dbf15ee",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8af50e5a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f028508a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala1742Pro</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A1742P mutation is located outside of ...</td>\n",
       "      <td>Mutations in the ATM gene lead to impaired ove...</td>\n",
       "      <td>PMID: 16014569</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala2062Val</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2062V mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala2067Asp</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2067D mutation is located in the FAT ...</td>\n",
       "      <td>A-TWinnipeg: Pathogenesis of rare ATM missense...</td>\n",
       "      <td>PMID: 25077176</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala59Ser</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A59S mutation is located outside of a ...</td>\n",
       "      <td>Loss of DNA Damage Response in Neuroblastoma a...</td>\n",
       "      <td>PMID: 29059438</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Cys2488Tyr</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM C2488Y mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Arg1391Ser</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 R1391S mutation is located in the GTPa...</td>\n",
       "      <td>Mutational and functional analysis of the neur...</td>\n",
       "      <td>PMID: 9003501, PMID: 16513807, PMID: 9668168</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>865</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Val1398Asp</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 V1398D mutation is located in the GTPa...</td>\n",
       "      <td>Assessment of the potential pathogenicity of m...</td>\n",
       "      <td>PMID: 22807134</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Lys1423Glu</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>The NF1 K1423E mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Lys1436Gln</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 K1436Q mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Ser1463Phe</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 S1463F mutation is located in the GTPa...</td>\n",
       "      <td>Exploring the somatic NF1 mutational spectrum ...</td>\n",
       "      <td>PMID: 22108604, PMID: 22807134</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>869 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Names   Alterations         Oncogenic          Mutation effect  \\\n",
       "0          ATM  p.Ala1742Pro  Likely Oncogenic  Likely Loss-of-function   \n",
       "1          ATM  p.Ala2062Val  Likely Oncogenic  Likely Loss-of-function   \n",
       "2          ATM  p.Ala2067Asp  Likely Oncogenic  Likely Loss-of-function   \n",
       "3          ATM    p.Ala59Ser  Likely Oncogenic  Likely Loss-of-function   \n",
       "4          ATM  p.Cys2488Tyr  Likely Oncogenic  Likely Loss-of-function   \n",
       "..         ...           ...               ...                      ...   \n",
       "864        NF1  p.Arg1391Ser  Likely Oncogenic  Likely Loss-of-function   \n",
       "865        NF1  p.Val1398Asp  Likely Oncogenic  Likely Loss-of-function   \n",
       "866        NF1  p.Lys1423Glu         Oncogenic         Loss-of-function   \n",
       "867        NF1  p.Lys1436Gln  Likely Oncogenic  Likely Loss-of-function   \n",
       "868        NF1  p.Ser1463Phe  Likely Oncogenic  Likely Loss-of-function   \n",
       "\n",
       "                      Description from OncoKB_Database  \\\n",
       "0    The ATM A1742P mutation is located outside of ...   \n",
       "1    The ATM A2062V mutation is located in the FAT ...   \n",
       "2    The ATM A2067D mutation is located in the FAT ...   \n",
       "3    The ATM A59S mutation is located outside of a ...   \n",
       "4    The ATM C2488Y mutation is located in the FAT ...   \n",
       "..                                                 ...   \n",
       "864  The NF1 R1391S mutation is located in the GTPa...   \n",
       "865  The NF1 V1398D mutation is located in the GTPa...   \n",
       "866  The NF1 K1423E mutation is located in the GTPa...   \n",
       "867  The NF1 K1436Q mutation is located in the GTPa...   \n",
       "868  The NF1 S1463F mutation is located in the GTPa...   \n",
       "\n",
       "                                             Citations  \\\n",
       "0    Mutations in the ATM gene lead to impaired ove...   \n",
       "1    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "2    A-TWinnipeg: Pathogenesis of rare ATM missense...   \n",
       "3    Loss of DNA Damage Response in Neuroblastoma a...   \n",
       "4    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "..                                                 ...   \n",
       "864  Mutational and functional analysis of the neur...   \n",
       "865  Assessment of the potential pathogenicity of m...   \n",
       "866  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "867  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "868  Exploring the somatic NF1 mutational spectrum ...   \n",
       "\n",
       "                                              PMID  Count  \n",
       "0                                   PMID: 16014569    1.0  \n",
       "1                                   PMID: 23585524    1.0  \n",
       "2                                   PMID: 25077176    1.0  \n",
       "3                                   PMID: 29059438    1.0  \n",
       "4                                   PMID: 23585524    1.0  \n",
       "..                                             ...    ...  \n",
       "864   PMID: 9003501, PMID: 16513807, PMID: 9668168    3.0  \n",
       "865                                 PMID: 22807134    1.0  \n",
       "866  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  \n",
       "867  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  \n",
       "868                 PMID: 22108604, PMID: 22807134    2.0  \n",
       "\n",
       "[869 rows x 8 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/final_onco.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "af9f5b5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Identifier</th>\n",
       "      <th>Filter</th>\n",
       "      <th>Variant Allele Fraction</th>\n",
       "      <th>Allelic Depths (AD)</th>\n",
       "      <th>Read Depths (DP)</th>\n",
       "      <th>Genotype Qualities (GQ)</th>\n",
       "      <th>0/1 Genotypes (GT)</th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Sequence Ontology (Combined)</th>\n",
       "      <th>Effect (Combined)</th>\n",
       "      <th>N of 4 Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Transcript Name (Clinically Relevant)</th>\n",
       "      <th>HGVS c. (Clinically Relevant)</th>\n",
       "      <th>HGVS p. (Clinically Relevant)</th>\n",
       "      <th>Transcript Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:14522</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1609C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:14542</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1589T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:14574</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1557T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:14590</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1541C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:14599</td>\n",
       "      <td>T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1532A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164045</th>\n",
       "      <td>Un_KI270756v1:2828</td>\n",
       "      <td>A/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164046</th>\n",
       "      <td>Un_KI270757v1:23426</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164047</th>\n",
       "      <td>Un_KI270757v1:23500</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164048</th>\n",
       "      <td>Un_KI270757v1:52356</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164049</th>\n",
       "      <td>Un_KI270757v1:68921</td>\n",
       "      <td>T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>164050 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Chr:Pos Ref/Alt  Identifier   Filter  \\\n",
       "0                   1:14522     G/A         NaN      NaN   \n",
       "1                   1:14542     A/G         NaN      NaN   \n",
       "2                   1:14574     A/G         NaN      NaN   \n",
       "3                   1:14590     G/A         NaN      NaN   \n",
       "4                   1:14599     T/A         NaN      NaN   \n",
       "...                     ...     ...         ...      ...   \n",
       "164045   Un_KI270756v1:2828     A/T         NaN      NaN   \n",
       "164046  Un_KI270757v1:23426     G/A         NaN      NaN   \n",
       "164047  Un_KI270757v1:23500     A/C         NaN      NaN   \n",
       "164048  Un_KI270757v1:52356     A/G         NaN  LowQual   \n",
       "164049  Un_KI270757v1:68921     T/A         NaN  LowQual   \n",
       "\n",
       "       Variant Allele Fraction Allelic Depths (AD)  Read Depths (DP)  \\\n",
       "0                            1                 0,2                 2   \n",
       "1                            1                 0,3                 3   \n",
       "2                            1                 0,3                 3   \n",
       "3                            1                 0,3                 3   \n",
       "4                            1                 0,3                 3   \n",
       "...                        ...                 ...               ...   \n",
       "164045                       1                 0,2                 2   \n",
       "164046                       1                 0,2                 2   \n",
       "164047                       1                 0,2                 2   \n",
       "164048                       1                 0,2                 2   \n",
       "164049                       1                 0,2                 2   \n",
       "\n",
       "        Genotype Qualities (GQ) 0/1 Genotypes (GT) Gene Names  \\\n",
       "0                             6                1/1     WASH7P   \n",
       "1                             9                1/1     WASH7P   \n",
       "2                             9                1/1     WASH7P   \n",
       "3                             9                1/1     WASH7P   \n",
       "4                             9                1/1     WASH7P   \n",
       "...                         ...                ...        ...   \n",
       "164045                        6                1/1        NaN   \n",
       "164046                        6                1/1        NaN   \n",
       "164047                        6                1/1        NaN   \n",
       "164048                        6                1/1        NaN   \n",
       "164049                        6                1/1        NaN   \n",
       "\n",
       "       Sequence Ontology (Combined) Effect (Combined)  \\\n",
       "0           non_coding_exon_variant             Other   \n",
       "1           non_coding_exon_variant             Other   \n",
       "2           non_coding_exon_variant             Other   \n",
       "3           non_coding_exon_variant             Other   \n",
       "4           non_coding_exon_variant             Other   \n",
       "...                             ...               ...   \n",
       "164045           intergenic_variant             Other   \n",
       "164046           intergenic_variant             Other   \n",
       "164047           intergenic_variant             Other   \n",
       "164048           intergenic_variant             Other   \n",
       "164049           intergenic_variant             Other   \n",
       "\n",
       "       N of 4 Predicted Splicing Disrupted (Combined)  \\\n",
       "0                                                 NaN   \n",
       "1                                                 NaN   \n",
       "2                                                 NaN   \n",
       "3                                                 NaN   \n",
       "4                                                 NaN   \n",
       "...                                               ...   \n",
       "164045                                            NaN   \n",
       "164046                                            NaN   \n",
       "164047                                            NaN   \n",
       "164048                                            NaN   \n",
       "164049                                            NaN   \n",
       "\n",
       "       Predicted Splicing Disrupted (Combined)  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                          NaN   \n",
       "...                                        ...   \n",
       "164045                                     NaN   \n",
       "164046                                     NaN   \n",
       "164047                                     NaN   \n",
       "164048                                     NaN   \n",
       "164049                                     NaN   \n",
       "\n",
       "       Transcript Name (Clinically Relevant) HGVS c. (Clinically Relevant)  \\\n",
       "0                                NR_024540.1         NR_024540.1:n.1609C>T   \n",
       "1                                NR_024540.1         NR_024540.1:n.1589T>C   \n",
       "2                                NR_024540.1         NR_024540.1:n.1557T>C   \n",
       "3                                NR_024540.1         NR_024540.1:n.1541C>T   \n",
       "4                                NR_024540.1         NR_024540.1:n.1532A>T   \n",
       "...                                      ...                           ...   \n",
       "164045                                   NaN                           NaN   \n",
       "164046                                   NaN                           NaN   \n",
       "164047                                   NaN                           NaN   \n",
       "164048                                   NaN                           NaN   \n",
       "164049                                   NaN                           NaN   \n",
       "\n",
       "       HGVS p. (Clinically Relevant) Transcript Name  \n",
       "0                                NaN     NR_024540.1  \n",
       "1                                NaN     NR_024540.1  \n",
       "2                                NaN     NR_024540.1  \n",
       "3                                NaN     NR_024540.1  \n",
       "4                                NaN     NR_024540.1  \n",
       "...                              ...             ...  \n",
       "164045                           NaN             NaN  \n",
       "164046                           NaN             NaN  \n",
       "164047                           NaN             NaN  \n",
       "164048                           NaN             NaN  \n",
       "164049                           NaN             NaN  \n",
       "\n",
       "[164050 rows x 18 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Germline_KHSCQTGPTTL2 - KHSCQTGPTTL2_S31_L004_R.xlsx', header=1, usecols=list(range(18)))\n",
    "data1['Alterations'] = data1['HGVS p. (Clinically Relevant)'].str.split(':').str[1]\n",
    "data1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "47188ef5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "p.Leu12=       8\n",
       "p.Leu89=       8\n",
       "p.Ala63=       8\n",
       "p.Leu101=      7\n",
       "p.Leu189=      7\n",
       "              ..\n",
       "p.Ile287Leu    1\n",
       "p.Asp158=      1\n",
       "p.His61Arg     1\n",
       "p.Arg232Ser    1\n",
       "p.Gly465=      1\n",
       "Name: Alterations, Length: 22441, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data1['Alterations'] = data1['HGVS p. (Clinically Relevant)'].str.split(':').str[1]\n",
    "data1.Alterations.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "30b1d9d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Identifier</th>\n",
       "      <th>Filter</th>\n",
       "      <th>Variant Allele Fraction</th>\n",
       "      <th>Allelic Depths (AD)</th>\n",
       "      <th>Read Depths (DP)</th>\n",
       "      <th>Genotype Qualities (GQ)</th>\n",
       "      <th>0/1 Genotypes (GT)</th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Sequence Ontology (Combined)</th>\n",
       "      <th>Effect (Combined)</th>\n",
       "      <th>N of 4 Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Transcript Name (Clinically Relevant)</th>\n",
       "      <th>HGVS c. (Clinically Relevant)</th>\n",
       "      <th>HGVS p. (Clinically Relevant)</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:14522</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1609C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:14542</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1589T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:14574</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1557T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:14590</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1541C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:14599</td>\n",
       "      <td>T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1532A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164045</th>\n",
       "      <td>Un_KI270756v1:2828</td>\n",
       "      <td>A/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164046</th>\n",
       "      <td>Un_KI270757v1:23426</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164047</th>\n",
       "      <td>Un_KI270757v1:23500</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164048</th>\n",
       "      <td>Un_KI270757v1:52356</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164049</th>\n",
       "      <td>Un_KI270757v1:68921</td>\n",
       "      <td>T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>164050 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Chr:Pos Ref/Alt  Identifier   Filter  \\\n",
       "0                   1:14522     G/A         NaN      NaN   \n",
       "1                   1:14542     A/G         NaN      NaN   \n",
       "2                   1:14574     A/G         NaN      NaN   \n",
       "3                   1:14590     G/A         NaN      NaN   \n",
       "4                   1:14599     T/A         NaN      NaN   \n",
       "...                     ...     ...         ...      ...   \n",
       "164045   Un_KI270756v1:2828     A/T         NaN      NaN   \n",
       "164046  Un_KI270757v1:23426     G/A         NaN      NaN   \n",
       "164047  Un_KI270757v1:23500     A/C         NaN      NaN   \n",
       "164048  Un_KI270757v1:52356     A/G         NaN  LowQual   \n",
       "164049  Un_KI270757v1:68921     T/A         NaN  LowQual   \n",
       "\n",
       "       Variant Allele Fraction Allelic Depths (AD)  Read Depths (DP)  \\\n",
       "0                            1                 0,2                 2   \n",
       "1                            1                 0,3                 3   \n",
       "2                            1                 0,3                 3   \n",
       "3                            1                 0,3                 3   \n",
       "4                            1                 0,3                 3   \n",
       "...                        ...                 ...               ...   \n",
       "164045                       1                 0,2                 2   \n",
       "164046                       1                 0,2                 2   \n",
       "164047                       1                 0,2                 2   \n",
       "164048                       1                 0,2                 2   \n",
       "164049                       1                 0,2                 2   \n",
       "\n",
       "        Genotype Qualities (GQ) 0/1 Genotypes (GT) Gene Names  \\\n",
       "0                             6                1/1     WASH7P   \n",
       "1                             9                1/1     WASH7P   \n",
       "2                             9                1/1     WASH7P   \n",
       "3                             9                1/1     WASH7P   \n",
       "4                             9                1/1     WASH7P   \n",
       "...                         ...                ...        ...   \n",
       "164045                        6                1/1        NaN   \n",
       "164046                        6                1/1        NaN   \n",
       "164047                        6                1/1        NaN   \n",
       "164048                        6                1/1        NaN   \n",
       "164049                        6                1/1        NaN   \n",
       "\n",
       "       Sequence Ontology (Combined) Effect (Combined)  \\\n",
       "0           non_coding_exon_variant             Other   \n",
       "1           non_coding_exon_variant             Other   \n",
       "2           non_coding_exon_variant             Other   \n",
       "3           non_coding_exon_variant             Other   \n",
       "4           non_coding_exon_variant             Other   \n",
       "...                             ...               ...   \n",
       "164045           intergenic_variant             Other   \n",
       "164046           intergenic_variant             Other   \n",
       "164047           intergenic_variant             Other   \n",
       "164048           intergenic_variant             Other   \n",
       "164049           intergenic_variant             Other   \n",
       "\n",
       "       N of 4 Predicted Splicing Disrupted (Combined)  \\\n",
       "0                                                 NaN   \n",
       "1                                                 NaN   \n",
       "2                                                 NaN   \n",
       "3                                                 NaN   \n",
       "4                                                 NaN   \n",
       "...                                               ...   \n",
       "164045                                            NaN   \n",
       "164046                                            NaN   \n",
       "164047                                            NaN   \n",
       "164048                                            NaN   \n",
       "164049                                            NaN   \n",
       "\n",
       "       Predicted Splicing Disrupted (Combined)  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                          NaN   \n",
       "...                                        ...   \n",
       "164045                                     NaN   \n",
       "164046                                     NaN   \n",
       "164047                                     NaN   \n",
       "164048                                     NaN   \n",
       "164049                                     NaN   \n",
       "\n",
       "       Transcript Name (Clinically Relevant) HGVS c. (Clinically Relevant)  \\\n",
       "0                                NR_024540.1         NR_024540.1:n.1609C>T   \n",
       "1                                NR_024540.1         NR_024540.1:n.1589T>C   \n",
       "2                                NR_024540.1         NR_024540.1:n.1557T>C   \n",
       "3                                NR_024540.1         NR_024540.1:n.1541C>T   \n",
       "4                                NR_024540.1         NR_024540.1:n.1532A>T   \n",
       "...                                      ...                           ...   \n",
       "164045                                   NaN                           NaN   \n",
       "164046                                   NaN                           NaN   \n",
       "164047                                   NaN                           NaN   \n",
       "164048                                   NaN                           NaN   \n",
       "164049                                   NaN                           NaN   \n",
       "\n",
       "       HGVS p. (Clinically Relevant) Transcript Name Alterations Gene Match  \\\n",
       "0                                NaN     NR_024540.1         NaN         No   \n",
       "1                                NaN     NR_024540.1         NaN         No   \n",
       "2                                NaN     NR_024540.1         NaN         No   \n",
       "3                                NaN     NR_024540.1         NaN         No   \n",
       "4                                NaN     NR_024540.1         NaN         No   \n",
       "...                              ...             ...         ...        ...   \n",
       "164045                           NaN             NaN         NaN         No   \n",
       "164046                           NaN             NaN         NaN         No   \n",
       "164047                           NaN             NaN         NaN         No   \n",
       "164048                           NaN             NaN         NaN         No   \n",
       "164049                           NaN             NaN         NaN         No   \n",
       "\n",
       "       Matched_Gene  \n",
       "0                    \n",
       "1                    \n",
       "2                    \n",
       "3                    \n",
       "4                    \n",
       "...             ...  \n",
       "164045               \n",
       "164046               \n",
       "164047               \n",
       "164048               \n",
       "164049               \n",
       "\n",
       "[164050 rows x 21 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data1['Gene Match'] = 'No'\n",
    "data1['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in data1['Gene Names'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df['Gene Names'].values:\n",
    "                data1.at[index, 'Gene Match'] = 'Yes'\n",
    "                data1.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "data1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c6f069e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "          163870\n",
       "NF1           32\n",
       "BRCA2         30\n",
       "ATM           24\n",
       "BARD1         22\n",
       "BRIP1         16\n",
       "MSH2          14\n",
       "MSH6           8\n",
       "MRE11          8\n",
       "CHEK2          8\n",
       "CDH1           7\n",
       "CDKN2A         4\n",
       "MUTYH          3\n",
       "BRCA1          2\n",
       "HOXB13         2\n",
       "Name: Matched_Gene, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data1.Matched_Gene.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "3c23ea36",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala1742Pro</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A1742P mutation is located outside of ...</td>\n",
       "      <td>Mutations in the ATM gene lead to impaired ove...</td>\n",
       "      <td>PMID: 16014569</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala2062Val</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2062V mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala2067Asp</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A2067D mutation is located in the FAT ...</td>\n",
       "      <td>A-TWinnipeg: Pathogenesis of rare ATM missense...</td>\n",
       "      <td>PMID: 25077176</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Ala59Ser</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM A59S mutation is located outside of a ...</td>\n",
       "      <td>Loss of DNA Damage Response in Neuroblastoma a...</td>\n",
       "      <td>PMID: 29059438</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>p.Cys2488Tyr</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The ATM C2488Y mutation is located in the FAT ...</td>\n",
       "      <td>ATM mutations uniformly lead to ATM dysfunctio...</td>\n",
       "      <td>PMID: 23585524</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Arg1391Ser</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 R1391S mutation is located in the GTPa...</td>\n",
       "      <td>Mutational and functional analysis of the neur...</td>\n",
       "      <td>PMID: 9003501, PMID: 16513807, PMID: 9668168</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>865</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Val1398Asp</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 V1398D mutation is located in the GTPa...</td>\n",
       "      <td>Assessment of the potential pathogenicity of m...</td>\n",
       "      <td>PMID: 22807134</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Lys1423Glu</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>The NF1 K1423E mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Lys1436Gln</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 K1436Q mutation is located in the GTPa...</td>\n",
       "      <td>Somatic mutations in the neurofibromatosis 1 g...</td>\n",
       "      <td>PMID: 1568247, PMID: 16513807, PMID: 32697994</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>NF1</td>\n",
       "      <td>p.Ser1463Phe</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>The NF1 S1463F mutation is located in the GTPa...</td>\n",
       "      <td>Exploring the somatic NF1 mutational spectrum ...</td>\n",
       "      <td>PMID: 22108604, PMID: 22807134</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>869 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Matched_Gene   Alterations         Oncogenic          Mutation effect  \\\n",
       "0            ATM  p.Ala1742Pro  Likely Oncogenic  Likely Loss-of-function   \n",
       "1            ATM  p.Ala2062Val  Likely Oncogenic  Likely Loss-of-function   \n",
       "2            ATM  p.Ala2067Asp  Likely Oncogenic  Likely Loss-of-function   \n",
       "3            ATM    p.Ala59Ser  Likely Oncogenic  Likely Loss-of-function   \n",
       "4            ATM  p.Cys2488Tyr  Likely Oncogenic  Likely Loss-of-function   \n",
       "..           ...           ...               ...                      ...   \n",
       "864          NF1  p.Arg1391Ser  Likely Oncogenic  Likely Loss-of-function   \n",
       "865          NF1  p.Val1398Asp  Likely Oncogenic  Likely Loss-of-function   \n",
       "866          NF1  p.Lys1423Glu         Oncogenic         Loss-of-function   \n",
       "867          NF1  p.Lys1436Gln  Likely Oncogenic  Likely Loss-of-function   \n",
       "868          NF1  p.Ser1463Phe  Likely Oncogenic  Likely Loss-of-function   \n",
       "\n",
       "                      Description from OncoKB_Database  \\\n",
       "0    The ATM A1742P mutation is located outside of ...   \n",
       "1    The ATM A2062V mutation is located in the FAT ...   \n",
       "2    The ATM A2067D mutation is located in the FAT ...   \n",
       "3    The ATM A59S mutation is located outside of a ...   \n",
       "4    The ATM C2488Y mutation is located in the FAT ...   \n",
       "..                                                 ...   \n",
       "864  The NF1 R1391S mutation is located in the GTPa...   \n",
       "865  The NF1 V1398D mutation is located in the GTPa...   \n",
       "866  The NF1 K1423E mutation is located in the GTPa...   \n",
       "867  The NF1 K1436Q mutation is located in the GTPa...   \n",
       "868  The NF1 S1463F mutation is located in the GTPa...   \n",
       "\n",
       "                                             Citations  \\\n",
       "0    Mutations in the ATM gene lead to impaired ove...   \n",
       "1    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "2    A-TWinnipeg: Pathogenesis of rare ATM missense...   \n",
       "3    Loss of DNA Damage Response in Neuroblastoma a...   \n",
       "4    ATM mutations uniformly lead to ATM dysfunctio...   \n",
       "..                                                 ...   \n",
       "864  Mutational and functional analysis of the neur...   \n",
       "865  Assessment of the potential pathogenicity of m...   \n",
       "866  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "867  Somatic mutations in the neurofibromatosis 1 g...   \n",
       "868  Exploring the somatic NF1 mutational spectrum ...   \n",
       "\n",
       "                                              PMID  Count  \n",
       "0                                   PMID: 16014569    1.0  \n",
       "1                                   PMID: 23585524    1.0  \n",
       "2                                   PMID: 25077176    1.0  \n",
       "3                                   PMID: 29059438    1.0  \n",
       "4                                   PMID: 23585524    1.0  \n",
       "..                                             ...    ...  \n",
       "864   PMID: 9003501, PMID: 16513807, PMID: 9668168    3.0  \n",
       "865                                 PMID: 22807134    1.0  \n",
       "866  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  \n",
       "867  PMID: 1568247, PMID: 16513807, PMID: 32697994    3.0  \n",
       "868                 PMID: 22108604, PMID: 22807134    2.0  \n",
       "\n",
       "[869 rows x 8 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.rename({'Gene Names':'Matched_Gene'}, axis=1)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "36689b2c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Identifier</th>\n",
       "      <th>Filter</th>\n",
       "      <th>Variant Allele Fraction</th>\n",
       "      <th>Allelic Depths (AD)</th>\n",
       "      <th>Read Depths (DP)</th>\n",
       "      <th>Genotype Qualities (GQ)</th>\n",
       "      <th>0/1 Genotypes (GT)</th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Sequence Ontology (Combined)</th>\n",
       "      <th>Effect (Combined)</th>\n",
       "      <th>N of 4 Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Transcript Name (Clinically Relevant)</th>\n",
       "      <th>HGVS c. (Clinically Relevant)</th>\n",
       "      <th>HGVS p. (Clinically Relevant)</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:14522</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1609C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:14542</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1589T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:14574</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1557T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:14590</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1541C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:14599</td>\n",
       "      <td>T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>1/1</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NR_024540.1:n.1532A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164045</th>\n",
       "      <td>Un_KI270756v1:2828</td>\n",
       "      <td>A/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164046</th>\n",
       "      <td>Un_KI270757v1:23426</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164047</th>\n",
       "      <td>Un_KI270757v1:23500</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164048</th>\n",
       "      <td>Un_KI270757v1:52356</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164049</th>\n",
       "      <td>Un_KI270757v1:68921</td>\n",
       "      <td>T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LowQual</td>\n",
       "      <td>1</td>\n",
       "      <td>0,2</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>164050 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Chr:Pos Ref/Alt  Identifier   Filter  \\\n",
       "0                   1:14522     G/A         NaN      NaN   \n",
       "1                   1:14542     A/G         NaN      NaN   \n",
       "2                   1:14574     A/G         NaN      NaN   \n",
       "3                   1:14590     G/A         NaN      NaN   \n",
       "4                   1:14599     T/A         NaN      NaN   \n",
       "...                     ...     ...         ...      ...   \n",
       "164045   Un_KI270756v1:2828     A/T         NaN      NaN   \n",
       "164046  Un_KI270757v1:23426     G/A         NaN      NaN   \n",
       "164047  Un_KI270757v1:23500     A/C         NaN      NaN   \n",
       "164048  Un_KI270757v1:52356     A/G         NaN  LowQual   \n",
       "164049  Un_KI270757v1:68921     T/A         NaN  LowQual   \n",
       "\n",
       "       Variant Allele Fraction Allelic Depths (AD)  Read Depths (DP)  \\\n",
       "0                            1                 0,2                 2   \n",
       "1                            1                 0,3                 3   \n",
       "2                            1                 0,3                 3   \n",
       "3                            1                 0,3                 3   \n",
       "4                            1                 0,3                 3   \n",
       "...                        ...                 ...               ...   \n",
       "164045                       1                 0,2                 2   \n",
       "164046                       1                 0,2                 2   \n",
       "164047                       1                 0,2                 2   \n",
       "164048                       1                 0,2                 2   \n",
       "164049                       1                 0,2                 2   \n",
       "\n",
       "        Genotype Qualities (GQ) 0/1 Genotypes (GT) Gene Names  \\\n",
       "0                             6                1/1     WASH7P   \n",
       "1                             9                1/1     WASH7P   \n",
       "2                             9                1/1     WASH7P   \n",
       "3                             9                1/1     WASH7P   \n",
       "4                             9                1/1     WASH7P   \n",
       "...                         ...                ...        ...   \n",
       "164045                        6                1/1        NaN   \n",
       "164046                        6                1/1        NaN   \n",
       "164047                        6                1/1        NaN   \n",
       "164048                        6                1/1        NaN   \n",
       "164049                        6                1/1        NaN   \n",
       "\n",
       "       Sequence Ontology (Combined) Effect (Combined)  \\\n",
       "0           non_coding_exon_variant             Other   \n",
       "1           non_coding_exon_variant             Other   \n",
       "2           non_coding_exon_variant             Other   \n",
       "3           non_coding_exon_variant             Other   \n",
       "4           non_coding_exon_variant             Other   \n",
       "...                             ...               ...   \n",
       "164045           intergenic_variant             Other   \n",
       "164046           intergenic_variant             Other   \n",
       "164047           intergenic_variant             Other   \n",
       "164048           intergenic_variant             Other   \n",
       "164049           intergenic_variant             Other   \n",
       "\n",
       "       N of 4 Predicted Splicing Disrupted (Combined)  \\\n",
       "0                                                 NaN   \n",
       "1                                                 NaN   \n",
       "2                                                 NaN   \n",
       "3                                                 NaN   \n",
       "4                                                 NaN   \n",
       "...                                               ...   \n",
       "164045                                            NaN   \n",
       "164046                                            NaN   \n",
       "164047                                            NaN   \n",
       "164048                                            NaN   \n",
       "164049                                            NaN   \n",
       "\n",
       "       Predicted Splicing Disrupted (Combined)  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                          NaN   \n",
       "...                                        ...   \n",
       "164045                                     NaN   \n",
       "164046                                     NaN   \n",
       "164047                                     NaN   \n",
       "164048                                     NaN   \n",
       "164049                                     NaN   \n",
       "\n",
       "       Transcript Name (Clinically Relevant) HGVS c. (Clinically Relevant)  \\\n",
       "0                                NR_024540.1         NR_024540.1:n.1609C>T   \n",
       "1                                NR_024540.1         NR_024540.1:n.1589T>C   \n",
       "2                                NR_024540.1         NR_024540.1:n.1557T>C   \n",
       "3                                NR_024540.1         NR_024540.1:n.1541C>T   \n",
       "4                                NR_024540.1         NR_024540.1:n.1532A>T   \n",
       "...                                      ...                           ...   \n",
       "164045                                   NaN                           NaN   \n",
       "164046                                   NaN                           NaN   \n",
       "164047                                   NaN                           NaN   \n",
       "164048                                   NaN                           NaN   \n",
       "164049                                   NaN                           NaN   \n",
       "\n",
       "       HGVS p. (Clinically Relevant) Transcript Name Alterations Gene Match  \\\n",
       "0                                NaN     NR_024540.1         NaN         No   \n",
       "1                                NaN     NR_024540.1         NaN         No   \n",
       "2                                NaN     NR_024540.1         NaN         No   \n",
       "3                                NaN     NR_024540.1         NaN         No   \n",
       "4                                NaN     NR_024540.1         NaN         No   \n",
       "...                              ...             ...         ...        ...   \n",
       "164045                           NaN             NaN         NaN         No   \n",
       "164046                           NaN             NaN         NaN         No   \n",
       "164047                           NaN             NaN         NaN         No   \n",
       "164048                           NaN             NaN         NaN         No   \n",
       "164049                           NaN             NaN         NaN         No   \n",
       "\n",
       "       Matched_Gene Oncogenic Mutation effect  \\\n",
       "0                         NaN             NaN   \n",
       "1                         NaN             NaN   \n",
       "2                         NaN             NaN   \n",
       "3                         NaN             NaN   \n",
       "4                         NaN             NaN   \n",
       "...             ...       ...             ...   \n",
       "164045                    NaN             NaN   \n",
       "164046                    NaN             NaN   \n",
       "164047                    NaN             NaN   \n",
       "164048                    NaN             NaN   \n",
       "164049                    NaN             NaN   \n",
       "\n",
       "       Description from OncoKB_Database Citations PMID  Count  \n",
       "0                                   NaN       NaN  NaN    NaN  \n",
       "1                                   NaN       NaN  NaN    NaN  \n",
       "2                                   NaN       NaN  NaN    NaN  \n",
       "3                                   NaN       NaN  NaN    NaN  \n",
       "4                                   NaN       NaN  NaN    NaN  \n",
       "...                                 ...       ...  ...    ...  \n",
       "164045                              NaN       NaN  NaN    NaN  \n",
       "164046                              NaN       NaN  NaN    NaN  \n",
       "164047                              NaN       NaN  NaN    NaN  \n",
       "164048                              NaN       NaN  NaN    NaN  \n",
       "164049                              NaN       NaN  NaN    NaN  \n",
       "\n",
       "[164050 rows x 27 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mer = pd.merge(data1, df, on = ['Matched_Gene', 'Alterations'], how = 'left', sort=False)\n",
    "mer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "564b5b0b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Identifier</th>\n",
       "      <th>Filter</th>\n",
       "      <th>Variant Allele Fraction</th>\n",
       "      <th>Allelic Depths (AD)</th>\n",
       "      <th>Read Depths (DP)</th>\n",
       "      <th>Genotype Qualities (GQ)</th>\n",
       "      <th>0/1 Genotypes (GT)</th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Sequence Ontology (Combined)</th>\n",
       "      <th>Effect (Combined)</th>\n",
       "      <th>N of 4 Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Predicted Splicing Disrupted (Combined)</th>\n",
       "      <th>Transcript Name (Clinically Relevant)</th>\n",
       "      <th>HGVS c. (Clinically Relevant)</th>\n",
       "      <th>HGVS p. (Clinically Relevant)</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Alterations</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "      <th>Description from OncoKB_Database</th>\n",
       "      <th>Citations</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>103734</th>\n",
       "      <td>13:32332592</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.377358</td>\n",
       "      <td>33,20</td>\n",
       "      <td>53</td>\n",
       "      <td>99</td>\n",
       "      <td>0/1</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>NM_000059.4:c.1114A&gt;C</td>\n",
       "      <td>NP_000050.3:p.Asn372His</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>p.Asn372His</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>The BRCA2 N372H mutation is located in the P/C...</td>\n",
       "      <td>Functional assays for classification of BRCA2 ...</td>\n",
       "      <td>PMID: 18451181, PMID: 15695382</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103744</th>\n",
       "      <td>13:32355250</td>\n",
       "      <td>T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0,45</td>\n",
       "      <td>45</td>\n",
       "      <td>99</td>\n",
       "      <td>1/1</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>NM_000059.4:c.7397T&gt;C</td>\n",
       "      <td>NP_000050.3:p.Val2466Ala</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>p.Val2466Ala</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>The BRCA2 V2466A mutation is located in the FA...</td>\n",
       "      <td>Assessment of the Clinical Relevance of BRCA2 ...</td>\n",
       "      <td>PMID: 29394989</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Chr:Pos Ref/Alt  Identifier Filter Variant Allele Fraction  \\\n",
       "103734  13:32332592     A/C         NaN    NaN                0.377358   \n",
       "103744  13:32355250     T/C         NaN    NaN                       1   \n",
       "\n",
       "       Allelic Depths (AD)  Read Depths (DP)  Genotype Qualities (GQ)  \\\n",
       "103734               33,20                53                       99   \n",
       "103744                0,45                45                       99   \n",
       "\n",
       "       0/1 Genotypes (GT) Gene Names Sequence Ontology (Combined)  \\\n",
       "103734                0/1      BRCA2             missense_variant   \n",
       "103744                1/1      BRCA2             missense_variant   \n",
       "\n",
       "       Effect (Combined) N of 4 Predicted Splicing Disrupted (Combined)  \\\n",
       "103734          Missense                                            NaN   \n",
       "103744          Missense                                            NaN   \n",
       "\n",
       "       Predicted Splicing Disrupted (Combined)  \\\n",
       "103734                                     NaN   \n",
       "103744                                     NaN   \n",
       "\n",
       "       Transcript Name (Clinically Relevant) HGVS c. (Clinically Relevant)  \\\n",
       "103734                           NM_000059.4         NM_000059.4:c.1114A>C   \n",
       "103744                           NM_000059.4         NM_000059.4:c.7397T>C   \n",
       "\n",
       "       HGVS p. (Clinically Relevant) Transcript Name   Alterations  \\\n",
       "103734       NP_000050.3:p.Asn372His     NM_000059.4   p.Asn372His   \n",
       "103744      NP_000050.3:p.Val2466Ala     NM_000059.4  p.Val2466Ala   \n",
       "\n",
       "             Oncogenic Mutation effect  \\\n",
       "103734  Likely Neutral  Likely Neutral   \n",
       "103744  Likely Neutral  Likely Neutral   \n",
       "\n",
       "                         Description from OncoKB_Database  \\\n",
       "103734  The BRCA2 N372H mutation is located in the P/C...   \n",
       "103744  The BRCA2 V2466A mutation is located in the FA...   \n",
       "\n",
       "                                                Citations  \\\n",
       "103734  Functional assays for classification of BRCA2 ...   \n",
       "103744  Assessment of the Clinical Relevance of BRCA2 ...   \n",
       "\n",
       "                                  PMID  Count  \n",
       "103734  PMID: 18451181, PMID: 15695382    1.0  \n",
       "103744                  PMID: 29394989    1.0  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mer = mer.dropna(subset=['Oncogenic'])\n",
    "mer = mer.drop(['Gene Match', 'Matched_Gene'], axis=1)\n",
    "mer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d268a621",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "13122ca5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>21_conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cholesterol Disorders</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Skin Health</td>\n",
       "      <td>rs1764391</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>34795168</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gut Health (Irritable Bowel Disease, Inflammat...</td>\n",
       "      <td>rs3737240</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>150510879</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gut Health (Irritable Bowel Disease, Inflammat...</td>\n",
       "      <td>rs13294</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>150512511</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Skin Health</td>\n",
       "      <td>rs7518080</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152304195</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Skin Health</td>\n",
       "      <td>rs61816760</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152305958</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Skin Health</td>\n",
       "      <td>rs192116923</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152306930</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Skin Health</td>\n",
       "      <td>rs34806697</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152309145</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Diabetes, Cardiac Health(CAD, Arrythmias, Card...</td>\n",
       "      <td>rs513043</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>156129878</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Gut Health (Irritable Bowel Disease, Inflammat...</td>\n",
       "      <td>rs1801274</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>161509955</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Gut Health (Irritable Bowel Disease, Inflammat...</td>\n",
       "      <td>rs382627</td>\n",
       "      <td>No</td>\n",
       "      <td>chr1</td>\n",
       "      <td>161518015</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   21_conditions_list         rsID Literature  \\\n",
       "0                               Cholesterol Disorders    rs6687605         No   \n",
       "1                                         Skin Health    rs1764391         No   \n",
       "2   Gut Health (Irritable Bowel Disease, Inflammat...    rs3737240         No   \n",
       "3   Gut Health (Irritable Bowel Disease, Inflammat...      rs13294         No   \n",
       "4                                         Skin Health    rs7518080         No   \n",
       "5                                         Skin Health   rs61816760         No   \n",
       "6                                         Skin Health  rs192116923         No   \n",
       "7                                         Skin Health   rs34806697         No   \n",
       "8   Diabetes, Cardiac Health(CAD, Arrythmias, Card...     rs513043         No   \n",
       "9   Gut Health (Irritable Bowel Disease, Inflammat...    rs1801274         No   \n",
       "10  Gut Health (Irritable Bowel Disease, Inflammat...     rs382627         No   \n",
       "\n",
       "   CHROM        POS REF ALT  \n",
       "0   chr1   25563141   T   C  \n",
       "1   chr1   34795168   C   T  \n",
       "2   chr1  150510879   C   T  \n",
       "3   chr1  150512511   G   A  \n",
       "4   chr1  152304195   C   A  \n",
       "5   chr1  152305958   T   G  \n",
       "6   chr1  152306930   T   G  \n",
       "7   chr1  152309145   T   C  \n",
       "8   chr1  156129878   T   G  \n",
       "9   chr1  161509955   A   G  \n",
       "10  chr1  161518015   T   C  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import xlsxwriter\n",
    "\n",
    "# Sample DataFrame\n",
    "data = {\n",
    "    '21_conditions_list': [\n",
    "        'Cholesterol Disorders',\n",
    "        'Skin Health',\n",
    "        'Gut Health (Irritable Bowel Disease, Inflammatory Bowel Syndrome)',\n",
    "        'Gut Health (Irritable Bowel Disease, Inflammatory Bowel Syndrome)',\n",
    "        'Skin Health',\n",
    "        'Skin Health',\n",
    "        'Skin Health',\n",
    "        'Skin Health',\n",
    "        'Diabetes, Cardiac Health(CAD, Arrythmias, Cardiomyopathy), Cholesterol Disorders',\n",
    "        'Gut Health (Irritable Bowel Disease, Inflammatory Bowel Syndrome)',\n",
    "        'Gut Health (Irritable Bowel Disease, Inflammatory Bowel Syndrome)'\n",
    "    ],\n",
    "    'rsID': [\n",
    "        'rs6687605',\n",
    "        'rs1764391',\n",
    "        'rs3737240',\n",
    "        'rs13294',\n",
    "        'rs7518080',\n",
    "        'rs61816760',\n",
    "        'rs192116923',\n",
    "        'rs34806697',\n",
    "        'rs513043',\n",
    "        'rs1801274',\n",
    "        'rs382627'\n",
    "    ],\n",
    "    'Literature': [\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No',\n",
    "        'No'\n",
    "    ],\n",
    "    'CHROM': [\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1',\n",
    "        'chr1'\n",
    "    ],\n",
    "    'POS': [\n",
    "        25563141,\n",
    "        34795168,\n",
    "        150510879,\n",
    "        150512511,\n",
    "        152304195,\n",
    "        152305958,\n",
    "        152306930,\n",
    "        152309145,\n",
    "        156129878,\n",
    "        161509955,\n",
    "        161518015\n",
    "    ],\n",
    "    'REF': [\n",
    "        'T',\n",
    "        'C',\n",
    "        'C',\n",
    "        'G',\n",
    "        'C',\n",
    "        'T',\n",
    "        'T',\n",
    "        'T',\n",
    "        'T',\n",
    "        'A',\n",
    "        'T'\n",
    "    ],\n",
    "    'ALT': [\n",
    "        'C',\n",
    "        'T',\n",
    "        'T',\n",
    "        'A',\n",
    "        'A',\n",
    "        'G',\n",
    "        'G',\n",
    "        'C',\n",
    "        'G',\n",
    "        'G',\n",
    "        'C'\n",
    "    ]\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "caa26c9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create an Excel writer object\n",
    "output_path = r'C:/Users/GenepoweRx_Madhu/Downloads/sample_output.xlsx'\n",
    "workbook = xlsxwriter.Workbook(output_path)\n",
    "worksheet_names = []  # To keep track of worksheet names\n",
    "\n",
    "# Iterate through unique conditions\n",
    "for condition in df['21_conditions_list'].explode().unique():\n",
    "    # Filter rows for the current condition, including rows with multiple conditions\n",
    "    condition_df = df[df['21_conditions_list'].apply(lambda x: condition in x)]\n",
    "    \n",
    "    # Truncate long worksheet names to a maximum of 31 characters\n",
    "    sheet_name = condition[:31]\n",
    "    \n",
    "    # Ensure that the sheet name is unique\n",
    "    suffix = 1\n",
    "    while sheet_name in worksheet_names:\n",
    "        suffix += 1\n",
    "        sheet_name = condition[:28] + f'_{suffix}'\n",
    "    \n",
    "    worksheet_names.append(sheet_name)\n",
    "    \n",
    "    # Create a new worksheet with the truncated sheet name\n",
    "    worksheet = workbook.add_worksheet(sheet_name)\n",
    "    \n",
    "    # Write the filtered DataFrame to the worksheet\n",
    "    for row_num, (_, row_data) in enumerate(condition_df.iterrows(), start=1):\n",
    "        for col_num, value in enumerate(row_data, start=1):\n",
    "            if isinstance(value, list):\n",
    "                # Convert lists to a comma-separated string\n",
    "                value = ', '.join(value)\n",
    "            worksheet.write(row_num, col_num, value)\n",
    "\n",
    "# Close the workbook to save it\n",
    "workbook.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "75399578",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>EXON</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>rs1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>5/6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B</td>\n",
       "      <td>rs2</td>\n",
       "      <td>chr2</td>\n",
       "      <td>200</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>6/8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C</td>\n",
       "      <td>rs1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>300</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6/8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D</td>\n",
       "      <td>rs3</td>\n",
       "      <td>chr3</td>\n",
       "      <td>400</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2/5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>E</td>\n",
       "      <td>rs1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>500</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>6/8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene rsID CHROM  POS REF ALT EXON\n",
       "0    A  rs1  chr1  100   A   T  5/6\n",
       "1    B  rs2  chr2  200   C   G  6/8\n",
       "2    C  rs1  chr1  300   T   C  6/8\n",
       "3    D  rs3  chr3  400   G   A  2/5\n",
       "4    E  rs1  chr1  500   A   T  6/8"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample DataFrame\n",
    "data = {\n",
    "    'Gene': ['A', 'B', 'C', 'D', 'E'],\n",
    "    'rsID': ['rs1', 'rs2', 'rs1', 'rs3', 'rs1'],\n",
    "    'CHROM': ['chr1', 'chr2', 'chr1', 'chr3', 'chr1'],\n",
    "    'POS': [100, 200, 300, 400, 500],\n",
    "    'REF': ['A', 'C', 'T', 'G', 'A'],\n",
    "    'ALT': ['T', 'G', 'C', 'A', 'T'],\n",
    "    'EXON': ['5/6', '6/8', '6/8', '2/5', '6/8']\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9dc48fda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>EXON</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>rs1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C</td>\n",
       "      <td>rs1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>300</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene rsID CHROM  POS REF ALT EXON\n",
       "0    A  rs1  chr1  100   A   T    6\n",
       "2    C  rs1  chr1  300   T   C    8"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Identify rows with duplicate 'rsID' values\n",
    "duplicate_rsID = df[df['rsID'].duplicated(keep=False)]\n",
    "\n",
    "# Step 2: Select the row with the highest denominator in the 'EXON' column\n",
    "duplicate_rsID['EXON'] = duplicate_rsID['EXON'].apply(lambda x: max(x.split('/'), key=int))\n",
    "\n",
    "# Step 3: Keep the first occurrence of rows with the same 'rsID' and 'EXON' values\n",
    "result_df = duplicate_rsID.drop_duplicates(subset=['rsID', 'EXON'], keep='first')\n",
    "\n",
    "# Print or manipulate the result DataFrame as needed\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "20221846",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genes</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_conditions_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CEL</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>RYR1</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>758</th>\n",
       "      <td>SIM1</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>759</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>760</th>\n",
       "      <td>SCN1A</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>NTRK2</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>762 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Genes        Condition                     Headings 21_conditions_list\n",
       "0    CTLA4  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "1      INS  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "2    IL2RA  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "3    HNF1A  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "4      CEL  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "..     ...              ...                          ...                ...\n",
       "757   RYR1          Obesity                      Obesity            Obesity\n",
       "758   SIM1          Obesity                      Obesity            Obesity\n",
       "759  BRCA2          Obesity                      Obesity            Obesity\n",
       "760  SCN1A          Obesity                      Obesity            Obesity\n",
       "761  NTRK2          Obesity                      Obesity            Obesity\n",
       "\n",
       "[762 rows x 4 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/21 Conditions+Genes info (1).xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "7e1d0ede",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>21_conditions_list</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>High Blood Pressure</td>\n",
       "      <td>High_Blood_Pressure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cardiac Health(CAD, Arrythmias, Cardiomyopathy)</td>\n",
       "      <td>Cardiac_Health</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cholesterol Disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Thyroid Disorders</td>\n",
       "      <td>Thyroid_Disorders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Parkinson's</td>\n",
       "      <td>Parkinsons</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Dementia</td>\n",
       "      <td>Dementia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Headaches</td>\n",
       "      <td>Headaches</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Allergies (Respiratory, Food)</td>\n",
       "      <td>Allergies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Anemia</td>\n",
       "      <td>Anemia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Fatty Liver</td>\n",
       "      <td>Fatty_Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Gall stones</td>\n",
       "      <td>Gall_stones</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Pancreatic Disorders</td>\n",
       "      <td>Pancreatic_Disorders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Gut Health (Irritable Bowel Disease, Inflammat...</td>\n",
       "      <td>Gut_Health</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Gastritis</td>\n",
       "      <td>Gastritis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Glomerular Diseases</td>\n",
       "      <td>Glomerular_Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Interstitial Nephritis</td>\n",
       "      <td>Interstitial_Nephritis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Renal stones</td>\n",
       "      <td>Renal_stones</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Osteoporosis</td>\n",
       "      <td>Osteoporosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Arthritis/ Degenerative Joint Disease</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Muscular Dystrophy</td>\n",
       "      <td>Muscular_Dystrophy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Fatigue (Muscular, Arthritis, Osteoporosis)</td>\n",
       "      <td>Fatigue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Mood Disorders (Anxiety, Schizophrenia, Depres...</td>\n",
       "      <td>Mood_Disorders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Seizures</td>\n",
       "      <td>Seizures</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   21_conditions_list  \\\n",
       "0                                            Diabetes   \n",
       "1                                 High Blood Pressure   \n",
       "2     Cardiac Health(CAD, Arrythmias, Cardiomyopathy)   \n",
       "3                               Cholesterol Disorders   \n",
       "4                                   Thyroid Disorders   \n",
       "5                                         Parkinson's   \n",
       "6                                            Dementia   \n",
       "7                                           Headaches   \n",
       "8                       Allergies (Respiratory, Food)   \n",
       "9                                              Anemia   \n",
       "10                                        Fatty Liver   \n",
       "11                                        Gall stones   \n",
       "12                               Pancreatic Disorders   \n",
       "13  Gut Health (Irritable Bowel Disease, Inflammat...   \n",
       "14                                          Gastritis   \n",
       "15                                Glomerular Diseases   \n",
       "16                             Interstitial Nephritis   \n",
       "17                                       Renal stones   \n",
       "18                                        Skin Health   \n",
       "19                                       Osteoporosis   \n",
       "20              Arthritis/ Degenerative Joint Disease   \n",
       "21                                 Muscular Dystrophy   \n",
       "22        Fatigue (Muscular, Arthritis, Osteoporosis)   \n",
       "23  Mood Disorders (Anxiety, Schizophrenia, Depres...   \n",
       "24                                            Obesity   \n",
       "25                                           Seizures   \n",
       "\n",
       "                      21_Conditions_list  \n",
       "0                               Diabetes  \n",
       "1                    High_Blood_Pressure  \n",
       "2                         Cardiac_Health  \n",
       "3                  Cholesterol_Disorders  \n",
       "4                      Thyroid_Disorders  \n",
       "5                             Parkinsons  \n",
       "6                               Dementia  \n",
       "7                              Headaches  \n",
       "8                              Allergies  \n",
       "9                                 Anemia  \n",
       "10                           Fatty_Liver  \n",
       "11                           Gall_stones  \n",
       "12                  Pancreatic_Disorders  \n",
       "13                            Gut_Health  \n",
       "14                             Gastritis  \n",
       "15                   Glomerular_Diseases  \n",
       "16                Interstitial_Nephritis  \n",
       "17                          Renal_stones  \n",
       "18                           Skin_Health  \n",
       "19                          Osteoporosis  \n",
       "20  Arthritis_Degenerative_Joint_Disease  \n",
       "21                    Muscular_Dystrophy  \n",
       "22                               Fatigue  \n",
       "23                        Mood_Disorders  \n",
       "24                               Obesity  \n",
       "25                              Seizures  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Documents/Processed_vcf_files/21-cond.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "032a2066",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genes</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CEL</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>RYR1</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>758</th>\n",
       "      <td>SIM1</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>759</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>760</th>\n",
       "      <td>SCN1A</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>NTRK2</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>762 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Genes        Condition                     Headings 21_Conditions_list\n",
       "0    CTLA4  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "1      INS  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "2    IL2RA  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "3    HNF1A  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "4      CEL  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "..     ...              ...                          ...                ...\n",
       "757   RYR1          Obesity                      Obesity            Obesity\n",
       "758   SIM1          Obesity                      Obesity            Obesity\n",
       "759  BRCA2          Obesity                      Obesity            Obesity\n",
       "760  SCN1A          Obesity                      Obesity            Obesity\n",
       "761  NTRK2          Obesity                      Obesity            Obesity\n",
       "\n",
       "[762 rows x 4 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.merge(df, data, on = '21_conditions_list', how = 'left', sort=False)\n",
    "merged_df = merged_df[['Genes', 'Condition', 'Headings', '21_Conditions_list']]\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "7d13098f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genes</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CEL</td>\n",
       "      <td>Type 1 Diabetes</td>\n",
       "      <td>Predisposition for diabetes</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>RYR1</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>758</th>\n",
       "      <td>SIM1</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>759</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>760</th>\n",
       "      <td>SCN1A</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>NTRK2</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>Obesity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>762 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Genes        Condition                     Headings 21_Conditions_list\n",
       "0    CTLA4  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "1      INS  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "2    IL2RA  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "3    HNF1A  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "4      CEL  Type 1 Diabetes  Predisposition for diabetes           Diabetes\n",
       "..     ...              ...                          ...                ...\n",
       "757   RYR1          Obesity                      Obesity            Obesity\n",
       "758   SIM1          Obesity                      Obesity            Obesity\n",
       "759  BRCA2          Obesity                      Obesity            Obesity\n",
       "760  SCN1A          Obesity                      Obesity            Obesity\n",
       "761  NTRK2          Obesity                      Obesity            Obesity\n",
       "\n",
       "[762 rows x 4 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/dummy_genes.xlsx')\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "cb2f2fbf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>Other Skin Conditions</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cardiac_Health; Cholesterol_Disorders; Gall_st...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA12</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCB4</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>XPC</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>ZMPSTE24</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>ZNF804A</td>\n",
       "      <td>Schizophrenia</td>\n",
       "      <td>Mood Disorder</td>\n",
       "      <td>Mood_Disorders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>Diabetic Neuropathy</td>\n",
       "      <td>Diabetic Nephropathy</td>\n",
       "      <td>Diabetes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>584 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name                                          Condition  \\\n",
       "0       A2ML1                             Other Skin Conditions    \n",
       "1       AAGAB                              Dry Skin/fragile skin   \n",
       "2       ABCA1  Coronary artery disease risk; Polygenic (Famil...   \n",
       "3      ABCA12                              Dry Skin/fragile skin   \n",
       "4       ABCB4  Polygenic (Familial combined, small dense LDL,...   \n",
       "..        ...                                                ...   \n",
       "579       XPC                              Dry Skin/fragile skin   \n",
       "580  ZMPSTE24  Polygenic (Familial combined, small dense LDL,...   \n",
       "581   ZNF804A                                      Schizophrenia   \n",
       "582     ZNRF1                               Diabetic Retinopathy   \n",
       "583   ZSCAN20                                Diabetic Neuropathy   \n",
       "\n",
       "                                       Headings  \\\n",
       "0                                   Skin Health   \n",
       "1                                   Skin Health   \n",
       "2    Cardiac; Cholesterol disorders; Gallstones   \n",
       "3                                   Skin Health   \n",
       "4                         Cholesterol disorders   \n",
       "..                                          ...   \n",
       "579                                 Skin Health   \n",
       "580                       Cholesterol disorders   \n",
       "581                               Mood Disorder   \n",
       "582                        Diabetic Retinopathy   \n",
       "583                        Diabetic Nephropathy   \n",
       "\n",
       "                                    21_Conditions_list  \n",
       "0                                          Skin_Health  \n",
       "1                                          Skin_Health  \n",
       "2    Cardiac_Health; Cholesterol_Disorders; Gall_st...  \n",
       "3                                          Skin_Health  \n",
       "4                                Cholesterol_Disorders  \n",
       "..                                                 ...  \n",
       "579                                        Skin_Health  \n",
       "580                              Cholesterol_Disorders  \n",
       "581                                     Mood_Disorders  \n",
       "582                                           Diabetes  \n",
       "583                                           Diabetes  \n",
       "\n",
       "[584 rows x 4 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result = merged_df.groupby('Genes').agg({\n",
    "    'Condition': lambda x: '; '.join(x),\n",
    "    'Headings': lambda x: '; '.join(x),\n",
    "    '21_Conditions_list': lambda x: '; '.join(x.astype(str))\n",
    "}).reset_index()\n",
    "result = result.rename({'Genes': 'Gene Name'}, axis=1)\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "24ff16af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene Name, Condition, Headings, 21_Conditions_list]\n",
       "Index: []"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result[result['21_Conditions_list'] == 'Diabetes; nan']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "0bca47d3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRDN-AS1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMS2</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AASS</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA13</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCA7</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3935</th>\n",
       "      <td>MLYCD</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3936</th>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3937</th>\n",
       "      <td>PDHX</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3938</th>\n",
       "      <td>LOC105378457</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3939</th>\n",
       "      <td>EP300</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3940 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene Name  Gene_Score\n",
       "0         TRDN-AS1           6\n",
       "1             PMS2           8\n",
       "2             AASS           4\n",
       "3           ABCA13           4\n",
       "4            ABCA7           8\n",
       "...            ...         ...\n",
       "3935         MLYCD           6\n",
       "3936        ZSWIM6           4\n",
       "3937          PDHX           6\n",
       "3938  LOC105378457           4\n",
       "3939         EP300           6\n",
       "\n",
       "[3940 rows x 2 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scores = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/genes/genes_scores.xlsx')\n",
    "scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "800db80f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>Gene_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>Other Skin Conditions</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Coronary artery disease risk; Polygenic (Famil...</td>\n",
       "      <td>Cardiac; Cholesterol disorders; Gallstones</td>\n",
       "      <td>Cardiac_Health; Cholesterol_Disorders; Gall_st...</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA12</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCB4</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>XPC</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>ZMPSTE24</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>ZNF804A</td>\n",
       "      <td>Schizophrenia</td>\n",
       "      <td>Mood Disorder</td>\n",
       "      <td>Mood_Disorders</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>Diabetic Neuropathy</td>\n",
       "      <td>Diabetic Nephropathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>584 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name                                          Condition  \\\n",
       "0       A2ML1                             Other Skin Conditions    \n",
       "1       AAGAB                              Dry Skin/fragile skin   \n",
       "2       ABCA1  Coronary artery disease risk; Polygenic (Famil...   \n",
       "3      ABCA12                              Dry Skin/fragile skin   \n",
       "4       ABCB4  Polygenic (Familial combined, small dense LDL,...   \n",
       "..        ...                                                ...   \n",
       "579       XPC                              Dry Skin/fragile skin   \n",
       "580  ZMPSTE24  Polygenic (Familial combined, small dense LDL,...   \n",
       "581   ZNF804A                                      Schizophrenia   \n",
       "582     ZNRF1                               Diabetic Retinopathy   \n",
       "583   ZSCAN20                                Diabetic Neuropathy   \n",
       "\n",
       "                                       Headings  \\\n",
       "0                                   Skin Health   \n",
       "1                                   Skin Health   \n",
       "2    Cardiac; Cholesterol disorders; Gallstones   \n",
       "3                                   Skin Health   \n",
       "4                         Cholesterol disorders   \n",
       "..                                          ...   \n",
       "579                                 Skin Health   \n",
       "580                       Cholesterol disorders   \n",
       "581                               Mood Disorder   \n",
       "582                        Diabetic Retinopathy   \n",
       "583                        Diabetic Nephropathy   \n",
       "\n",
       "                                    21_Conditions_list  Gene_Score  \n",
       "0                                          Skin_Health         6.0  \n",
       "1                                          Skin_Health         6.0  \n",
       "2    Cardiac_Health; Cholesterol_Disorders; Gall_st...         7.0  \n",
       "3                                          Skin_Health         6.0  \n",
       "4                                Cholesterol_Disorders         6.0  \n",
       "..                                                 ...         ...  \n",
       "579                                        Skin_Health         6.0  \n",
       "580                              Cholesterol_Disorders         4.0  \n",
       "581                                     Mood_Disorders         NaN  \n",
       "582                                           Diabetes         NaN  \n",
       "583                                           Diabetes         4.0  \n",
       "\n",
       "[584 rows x 5 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(result, scores, on = 'Gene Name', how = 'left', sort = False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fee6bc4e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "6b73a6ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>Gene_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene Name, Condition, Headings, 21_Conditions_list, Gene_Score]\n",
       "Index: []"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nan_rows = merged_2[pd.isna(merged_2['21_Conditions_list'])]\n",
    "nan_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "d9d0aaa7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Skin_Health                                                                                          83\n",
       "Cardiac_Health                                                                                       68\n",
       "Cholesterol_Disorders                                                                                64\n",
       "Diabetes                                                                                             53\n",
       "Glomerular_Diseases                                                                                  39\n",
       "Allergies                                                                                            29\n",
       "Obesity                                                                                              25\n",
       "Thyroid_Disorders                                                                                    22\n",
       "High_Blood_Pressure                                                                                  22\n",
       "Parkinsons                                                                                           22\n",
       "Gut_Health                                                                                           20\n",
       "Diabetes; Gastritis                                                                                  16\n",
       "Gastritis                                                                                            15\n",
       "Renal_stones                                                                                         13\n",
       "Mood_Disorders                                                                                       10\n",
       "Pancreatic_Disorders                                                                                  9\n",
       "Arthritis_Degenerative_Joint_Disease                                                                  8\n",
       "Headaches                                                                                             7\n",
       "Dementia                                                                                              6\n",
       "Diabetes; Glomerular_Diseases                                                                         5\n",
       "Fatty_Liver                                                                                           5\n",
       "Cholesterol_Disorders; Gall_stones                                                                    4\n",
       "Interstitial_Nephritis                                                                                3\n",
       "High_Blood_Pressure; Glomerular_Diseases                                                              3\n",
       "Diabetes; Obesity                                                                                     3\n",
       "Cardiac_Health; Cholesterol_Disorders; Gall_stones                                                    2\n",
       "High_Blood_Pressure; Cardiac_Health                                                                   2\n",
       "Diabetes; Parkinsons; Glomerular_Diseases                                                             1\n",
       "High_Blood_Pressure; Glomerular_Diseases; Obesity                                                     1\n",
       "Cholesterol_Disorders; Obesity                                                                        1\n",
       "Glomerular_Diseases; Interstitial_Nephritis                                                           1\n",
       "Headaches; Obesity                                                                                    1\n",
       "Diabetes; Cholesterol_Disorders; Obesity                                                              1\n",
       "Diabetes; High_Blood_Pressure; Cardiac_Health; Glomerular_Diseases                                    1\n",
       "Diabetes; Gastritis; Arthritis_Degenerative_Joint_Disease                                             1\n",
       "Diabetes; Skin_Health                                                                                 1\n",
       "Diabetes; Cholesterol_Disorders                                                                       1\n",
       "Parkinsons; Dementia                                                                                  1\n",
       "Diabetes; Cardiac_Health; Cholesterol_Disorders                                                       1\n",
       "Diabetes; Thyroid_Disorders; Gastritis; Glomerular_Diseases; Arthritis_Degenerative_Joint_Disease     1\n",
       "Cardiac_Health; Cholesterol_Disorders                                                                 1\n",
       "Gall_stones                                                                                           1\n",
       "Cardiac_Health; Obesity                                                                               1\n",
       "Diabetes; Gut_Health                                                                                  1\n",
       "Cholesterol_Disorders; Fatty_Liver                                                                    1\n",
       "High_Blood_Pressure; Parkinsons                                                                       1\n",
       "Gut_Health; Arthritis_Degenerative_Joint_Disease                                                      1\n",
       "Cholesterol_Disorders; Skin_Health                                                                    1\n",
       "Cardiac_Health; Skin_Health                                                                           1\n",
       "Diabetes; Pancreatic_Disorders; Gastritis                                                             1\n",
       "Parkinsons; Headaches                                                                                 1\n",
       "Cholesterol_Disorders; Dementia; Gall_stones                                                          1\n",
       "Diabetes; Gut_Health; Gastritis                                                                       1\n",
       "Name: 21_Conditions_list, dtype: int64"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['21_Conditions_list'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "e8ed69f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_2.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/condition_specific/Condition_specific_all_lit_genes/Final_lit/last_conditions/genes/Conditions_final_genes.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "f34b29f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/dummy_genes.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f22bedc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffd571d0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "d143829c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>0</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs11203366</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PADI4</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>0</td>\n",
       "      <td>Rheumatoid Arthritis</td>\n",
       "      <td>Autoimmune Disorder</td>\n",
       "      <td>Arthritis_Degenerative_Joint_Disease</td>\n",
       "      <td>rs874881</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>8</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td></td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LOC105378642,GJA4</td>\n",
       "      <td>LOC105378642</td>\n",
       "      <td>7</td>\n",
       "      <td>Dry Skin/fragile skin</td>\n",
       "      <td>Skin Health</td>\n",
       "      <td>Skin_Health</td>\n",
       "      <td>rs1764391</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MYSM1</td>\n",
       "      <td>MYSM1</td>\n",
       "      <td>4</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>rs12139511</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>LMNA</td>\n",
       "      <td>6</td>\n",
       "      <td>Risk of Heart Disease; Hypertrophic cardiomyop...</td>\n",
       "      <td>Risk of Heart Disease; Cardiac; Arrhythmia; Ca...</td>\n",
       "      <td>Diabetes; Cardiac_Health; Cardiac_Health; Card...</td>\n",
       "      <td>rs513043</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>4</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut_Health</td>\n",
       "      <td>rs1801274</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>4</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut Health</td>\n",
       "      <td>Gut_Health</td>\n",
       "      <td>rs382627</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SELP</td>\n",
       "      <td>SELP</td>\n",
       "      <td>4</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>rs6136</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SELP</td>\n",
       "      <td>SELP</td>\n",
       "      <td>4</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetic Retinopathy</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>rs6127</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LOC102724528,C1orf105,PIGC</td>\n",
       "      <td>PIGC</td>\n",
       "      <td>4</td>\n",
       "      <td>Polygenic (Familial combined, small dense LDL,...</td>\n",
       "      <td>Cholesterol disorders</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs1063412</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CACNA1S</td>\n",
       "      <td>CACNA1S</td>\n",
       "      <td>6</td>\n",
       "      <td>Migraines and other headaches</td>\n",
       "      <td>Headaches</td>\n",
       "      <td>Headaches</td>\n",
       "      <td>rs3850625</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene Name          Gene  Gene_Score  \\\n",
       "0                        PADI4         PADI4           0   \n",
       "1                        PADI4         PADI4           0   \n",
       "2                      LDLRAP1       LDLRAP1           8   \n",
       "3            LOC105378642,GJA4  LOC105378642           7   \n",
       "4                        MYSM1         MYSM1           4   \n",
       "5                         LMNA          LMNA           6   \n",
       "6                       FCGR2A        FCGR2A           4   \n",
       "7                       FCGR2A        FCGR2A           4   \n",
       "8                         SELP          SELP           4   \n",
       "9                         SELP          SELP           4   \n",
       "10  LOC102724528,C1orf105,PIGC          PIGC           4   \n",
       "11                     CACNA1S       CACNA1S           6   \n",
       "\n",
       "                                            Condition  \\\n",
       "0                                Rheumatoid Arthritis   \n",
       "1                                Rheumatoid Arthritis   \n",
       "2   Polygenic (Familial combined, small dense LDL,...   \n",
       "3                               Dry Skin/fragile skin   \n",
       "4                                Diabetic Retinopathy   \n",
       "5   Risk of Heart Disease; Hypertrophic cardiomyop...   \n",
       "6                                          Gut Health   \n",
       "7                                          Gut Health   \n",
       "8                                Diabetic Retinopathy   \n",
       "9                                Diabetic Retinopathy   \n",
       "10  Polygenic (Familial combined, small dense LDL,...   \n",
       "11                      Migraines and other headaches   \n",
       "\n",
       "                                             Headings  \\\n",
       "0                                 Autoimmune Disorder   \n",
       "1                                 Autoimmune Disorder   \n",
       "2                               Cholesterol disorders   \n",
       "3                                         Skin Health   \n",
       "4                                Diabetic Retinopathy   \n",
       "5   Risk of Heart Disease; Cardiac; Arrhythmia; Ca...   \n",
       "6                                          Gut Health   \n",
       "7                                          Gut Health   \n",
       "8                                Diabetic Retinopathy   \n",
       "9                                Diabetic Retinopathy   \n",
       "10                              Cholesterol disorders   \n",
       "11                                          Headaches   \n",
       "\n",
       "                                   21_Conditions_list        rsID Literature  \n",
       "0                Arthritis_Degenerative_Joint_Disease  rs11203366         No  \n",
       "1                Arthritis_Degenerative_Joint_Disease    rs874881         No  \n",
       "2                                                       rs6687605         No  \n",
       "3                                         Skin_Health   rs1764391         No  \n",
       "4                                            Diabetes  rs12139511         No  \n",
       "5   Diabetes; Cardiac_Health; Cardiac_Health; Card...    rs513043         No  \n",
       "6                                          Gut_Health   rs1801274         No  \n",
       "7                                          Gut_Health    rs382627         No  \n",
       "8                                            Diabetes      rs6136         No  \n",
       "9                                            Diabetes      rs6127         No  \n",
       "10                              Cholesterol_Disorders   rs1063412         No  \n",
       "11                                          Headaches   rs3850625         No  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "data = {\n",
    "    'Gene Name': ['PADI4', 'PADI4', 'LDLRAP1', 'LOC105378642,GJA4', 'MYSM1', 'LMNA', 'FCGR2A', 'FCGR2A', 'SELP', 'SELP', 'LOC102724528,C1orf105,PIGC', 'CACNA1S'],\n",
    "    'Gene': ['PADI4', 'PADI4', 'LDLRAP1', 'LOC105378642', 'MYSM1', 'LMNA', 'FCGR2A', 'FCGR2A', 'SELP', 'SELP', 'PIGC', 'CACNA1S'],\n",
    "    'Gene_Score': [0, 0, 8, 7, 4, 6, 4, 4, 4, 4, 4, 6],\n",
    "    'Condition': ['Rheumatoid Arthritis', 'Rheumatoid Arthritis', 'Polygenic (Familial combined, small dense LDL, Miscellaneous)', 'Dry Skin/fragile skin', 'Diabetic Retinopathy', 'Risk of Heart Disease; Hypertrophic cardiomyopathy risk; Long QT syndrome; Cardiac Channelopathies; Polygenic (Familial combined, small dense LDL, Miscellaneous)', 'Gut Health', 'Gut Health', 'Diabetic Retinopathy', 'Diabetic Retinopathy', 'Polygenic (Familial combined, small dense LDL, Miscellaneous)', 'Migraines and other headaches'],\n",
    "    'Headings': ['Autoimmune Disorder', 'Autoimmune Disorder', 'Cholesterol disorders', 'Skin Health', 'Diabetic Retinopathy', 'Risk of Heart Disease; Cardiac; Arrhythmia; Cardiac; Cholesterol disorders', 'Gut Health', 'Gut Health', 'Diabetic Retinopathy', 'Diabetic Retinopathy', 'Cholesterol disorders', 'Headaches'],\n",
    "    '21_Conditions_list': ['Arthritis_Degenerative_Joint_Disease', 'Arthritis_Degenerative_Joint_Disease', '', 'Skin_Health', 'Diabetes', 'Diabetes; Cardiac_Health; Cardiac_Health; Cardiac_Health; Cholesterol_Disorders', 'Gut_Health', 'Gut_Health', 'Diabetes', 'Diabetes', 'Cholesterol_Disorders', 'Headaches'],\n",
    "    'rsID': ['rs11203366', 'rs874881', 'rs6687605', 'rs1764391', 'rs12139511', 'rs513043', 'rs1801274', 'rs382627', 'rs6136', 'rs6127', 'rs1063412', 'rs3850625'],\n",
    "    'Literature': ['No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No']\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "03b4d500",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_Score</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Headings</th>\n",
       "      <th>21_Conditions_list</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>LMNA</td>\n",
       "      <td>6</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Diabetes</td>\n",
       "      <td>rs513043</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>LMNA</td>\n",
       "      <td>6</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Cardiac_Health</td>\n",
       "      <td>rs513043</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>LMNA</td>\n",
       "      <td>6</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Cardiac_Health</td>\n",
       "      <td>rs513043</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>LMNA</td>\n",
       "      <td>6</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Cardiac_Health</td>\n",
       "      <td>rs513043</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>LMNA</td>\n",
       "      <td>6</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "      <td>Cholesterol_Disorders</td>\n",
       "      <td>rs513043</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>4</td>\n",
       "      <td>[Gut Health]</td>\n",
       "      <td>[Gut Health]</td>\n",
       "      <td>[Gut_Health]</td>\n",
       "      <td>rs382627</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>SELP</td>\n",
       "      <td>SELP</td>\n",
       "      <td>4</td>\n",
       "      <td>[Diabetic Retinopathy]</td>\n",
       "      <td>[Diabetic Retinopathy]</td>\n",
       "      <td>[Diabetes]</td>\n",
       "      <td>rs6136</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>SELP</td>\n",
       "      <td>SELP</td>\n",
       "      <td>4</td>\n",
       "      <td>[Diabetic Retinopathy]</td>\n",
       "      <td>[Diabetic Retinopathy]</td>\n",
       "      <td>[Diabetes]</td>\n",
       "      <td>rs6127</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>LOC102724528,C1orf105,PIGC</td>\n",
       "      <td>PIGC</td>\n",
       "      <td>4</td>\n",
       "      <td>[Polygenic (Familial combined, small dense LDL...</td>\n",
       "      <td>[Cholesterol disorders]</td>\n",
       "      <td>[Cholesterol_Disorders]</td>\n",
       "      <td>rs1063412</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>CACNA1S</td>\n",
       "      <td>CACNA1S</td>\n",
       "      <td>6</td>\n",
       "      <td>[Migraines and other headaches]</td>\n",
       "      <td>[Headaches]</td>\n",
       "      <td>[Headaches]</td>\n",
       "      <td>rs3850625</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>136 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Gene Name     Gene  Gene_Score  \\\n",
       "0                          LMNA     LMNA           6   \n",
       "1                          LMNA     LMNA           6   \n",
       "2                          LMNA     LMNA           6   \n",
       "3                          LMNA     LMNA           6   \n",
       "4                          LMNA     LMNA           6   \n",
       "..                          ...      ...         ...   \n",
       "131                      FCGR2A   FCGR2A           4   \n",
       "132                        SELP     SELP           4   \n",
       "133                        SELP     SELP           4   \n",
       "134  LOC102724528,C1orf105,PIGC     PIGC           4   \n",
       "135                     CACNA1S  CACNA1S           6   \n",
       "\n",
       "                                             Condition  \\\n",
       "0                                Risk of Heart Disease   \n",
       "1                                Risk of Heart Disease   \n",
       "2                                Risk of Heart Disease   \n",
       "3                                Risk of Heart Disease   \n",
       "4                                Risk of Heart Disease   \n",
       "..                                                 ...   \n",
       "131                                       [Gut Health]   \n",
       "132                             [Diabetic Retinopathy]   \n",
       "133                             [Diabetic Retinopathy]   \n",
       "134  [Polygenic (Familial combined, small dense LDL...   \n",
       "135                    [Migraines and other headaches]   \n",
       "\n",
       "                    Headings       21_Conditions_list       rsID Literature  \n",
       "0      Risk of Heart Disease                 Diabetes   rs513043         No  \n",
       "1      Risk of Heart Disease           Cardiac_Health   rs513043         No  \n",
       "2      Risk of Heart Disease           Cardiac_Health   rs513043         No  \n",
       "3      Risk of Heart Disease           Cardiac_Health   rs513043         No  \n",
       "4      Risk of Heart Disease    Cholesterol_Disorders   rs513043         No  \n",
       "..                       ...                      ...        ...        ...  \n",
       "131             [Gut Health]             [Gut_Health]   rs382627         No  \n",
       "132   [Diabetic Retinopathy]               [Diabetes]     rs6136         No  \n",
       "133   [Diabetic Retinopathy]               [Diabetes]     rs6127         No  \n",
       "134  [Cholesterol disorders]  [Cholesterol_Disorders]  rs1063412         No  \n",
       "135              [Headaches]              [Headaches]  rs3850625         No  \n",
       "\n",
       "[136 rows x 8 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Split the 'Condition', 'Headings', and '21_Conditions_list' columns into lists\n",
    "df['Condition'] = df['Condition'].str.split('; ')\n",
    "df['Headings'] = df['Headings'].str.split('; ')\n",
    "df['21_Conditions_list'] = df['21_Conditions_list'].str.split('; ')\n",
    "\n",
    "# Duplicate rows only if they contain a semicolon\n",
    "df_with_semicolon = df[df['Condition'].apply(lambda x: len(x) > 1)]\n",
    "df_without_semicolon = df[df['Condition'].apply(lambda x: len(x) == 1)]\n",
    "\n",
    "# Duplicate rows with semicolons and explode the lists\n",
    "df_with_semicolon = df_with_semicolon.explode('Condition').explode('Headings').explode('21_Conditions_list')\n",
    "\n",
    "# Concatenate the dataframes back together\n",
    "df = pd.concat([df_with_semicolon, df_without_semicolon], ignore_index=True)\n",
    "\n",
    "# Reset the index\n",
    "df.reset_index(drop=True, inplace=True)\n",
    "\n",
    "# Print the DataFrame\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "d7ff9901",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>column1</th>\n",
       "      <th>column2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>data; gene1</td>\n",
       "      <td>gene4; gene6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gene; gene1; gene2</td>\n",
       "      <td>gene6; gene7; gene88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gene3</td>\n",
       "      <td>gene99</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              column1               column2\n",
       "0         data; gene1          gene4; gene6\n",
       "1  gene; gene1; gene2  gene6; gene7; gene88\n",
       "2               gene3                gene99"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "data = {\n",
    "    'column1': ['data; gene1', 'gene; gene1; gene2', 'gene3'],\n",
    "    'column2': ['gene4; gene6', 'gene6; gene7; gene88', 'gene99']\n",
    "}\n",
    "\n",
    "# Create a DataFrame\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "3b0201c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>column1</th>\n",
       "      <th>column2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>data</td>\n",
       "      <td>gene4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gene1</td>\n",
       "      <td>gene6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gene</td>\n",
       "      <td>gene6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gene1</td>\n",
       "      <td>gene7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gene2</td>\n",
       "      <td>gene88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>gene3</td>\n",
       "      <td>gene99</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  column1 column2\n",
       "0    data   gene4\n",
       "1   gene1   gene6\n",
       "2    gene   gene6\n",
       "3   gene1   gene7\n",
       "4   gene2  gene88\n",
       "5   gene3  gene99"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Split the cells in each column by commas\n",
    "df['column1'] = df['column1'].str.split('; ')\n",
    "df['column2'] = df['column2'].str.split('; ')\n",
    "\n",
    "# Initialize an empty list to store the result\n",
    "result = []\n",
    "\n",
    "# Iterate through the rows and combine elements from both columns\n",
    "for index, row in df.iterrows():\n",
    "    column1_elements = row['column1']\n",
    "    column2_elements = row['column2']\n",
    "    \n",
    "    # Pair elements from both columns\n",
    "    pairs = zip(column1_elements, column2_elements)\n",
    "    \n",
    "    # Append pairs to the result list\n",
    "    result.extend(pairs)\n",
    "\n",
    "# Create a new DataFrame from the result list\n",
    "result_df = pd.DataFrame(result, columns=['column1', 'column2'])\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "8bb7b39c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "      <th>Column3</th>\n",
       "      <th>OtherData1</th>\n",
       "      <th>OtherData2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A;B</td>\n",
       "      <td>1;2</td>\n",
       "      <td>X;Y</td>\n",
       "      <td>foo</td>\n",
       "      <td>apple</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C;D</td>\n",
       "      <td>3;4</td>\n",
       "      <td>Z;W</td>\n",
       "      <td>bar</td>\n",
       "      <td>banana</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>E</td>\n",
       "      <td>5</td>\n",
       "      <td>V</td>\n",
       "      <td>baz</td>\n",
       "      <td>cherry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>F;G;H</td>\n",
       "      <td>6;7;8</td>\n",
       "      <td>U;T;S</td>\n",
       "      <td>qux</td>\n",
       "      <td>date</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Column1 Column2 Column3 OtherData1 OtherData2\n",
       "0     A;B     1;2     X;Y        foo      apple\n",
       "1     C;D     3;4     Z;W        bar     banana\n",
       "2       E       5       V        baz     cherry\n",
       "3   F;G;H   6;7;8   U;T;S        qux       date"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample DataFrame\n",
    "data = {'Column1': ['A;B', 'C;D', 'E', 'F;G;H'],\n",
    "        'Column2': ['1;2', '3;4', '5', '6;7;8'],\n",
    "        'Column3': ['X;Y', 'Z;W', 'V', 'U;T;S'],\n",
    "        'OtherData1': ['foo', 'bar', 'baz', 'qux'],\n",
    "        'OtherData2': ['apple', 'banana', 'cherry', 'date']}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "dc60db9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OtherData1</th>\n",
       "      <th>OtherData2</th>\n",
       "      <th>Column1</th>\n",
       "      <th>Column2</th>\n",
       "      <th>Column3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>foo</td>\n",
       "      <td>apple</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>foo</td>\n",
       "      <td>apple</td>\n",
       "      <td>B</td>\n",
       "      <td>2</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bar</td>\n",
       "      <td>banana</td>\n",
       "      <td>C</td>\n",
       "      <td>3</td>\n",
       "      <td>Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bar</td>\n",
       "      <td>banana</td>\n",
       "      <td>D</td>\n",
       "      <td>4</td>\n",
       "      <td>W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>baz</td>\n",
       "      <td>cherry</td>\n",
       "      <td>E</td>\n",
       "      <td>5</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>qux</td>\n",
       "      <td>date</td>\n",
       "      <td>F</td>\n",
       "      <td>6</td>\n",
       "      <td>U</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>qux</td>\n",
       "      <td>date</td>\n",
       "      <td>G</td>\n",
       "      <td>7</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>qux</td>\n",
       "      <td>date</td>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  OtherData1 OtherData2 Column1 Column2 Column3\n",
       "0        foo      apple       A       1       X\n",
       "1        foo      apple       B       2       Y\n",
       "2        bar     banana       C       3       Z\n",
       "3        bar     banana       D       4       W\n",
       "4        baz     cherry       E       5       V\n",
       "5        qux       date       F       6       U\n",
       "6        qux       date       G       7       T\n",
       "7        qux       date       H       8       S"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Split the columns and stack them\n",
    "df = df.set_index(['OtherData1', 'OtherData2']).apply(lambda x: x.str.split(';').explode()).reset_index()\n",
    "\n",
    "# Rename the columns\n",
    "#df.columns = ['Column1', 'Column2', 'Column3', 'OtherData1', 'OtherData2']\n",
    "\n",
    "# Display the result\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a59255f8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7fa60129",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>POS</th>\n",
       "      <th>Distance</th>\n",
       "      <th>Nearest_POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>40</td>\n",
       "      <td>2</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID  POS Distance Nearest_POS\n",
       "0   1   10        2          12\n",
       "1   2   25        1          24\n",
       "2   3   40        2          42"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Assuming you have two DataFrames: df1 and df2\n",
    "# Replace 'POS' with the actual column name in your datasets\n",
    "\n",
    "# Sample data for illustration purposes\n",
    "data1 = {'ID': [1, 2, 3], 'POS': [10, 25, 40]}\n",
    "data2 = {'ID': [4, 5, 6,7], 'POS': [12, 24, 42, 55]}\n",
    "\n",
    "df1 = pd.DataFrame(data1)\n",
    "df2 = pd.DataFrame(data2)\n",
    "\n",
    "# Create an empty 'Distance' column in df1\n",
    "df1['Distance'] = None\n",
    "df1['Nearest_POS'] = None\n",
    "\n",
    "# Loop through each row in df1\n",
    "for index1, row1 in df1.iterrows():\n",
    "    # Calculate absolute differences between POS values\n",
    "    differences = np.abs(df2['POS'] - row1['POS'])\n",
    "    \n",
    "    # Find the index of the minimum difference\n",
    "    min_index = np.argmin(differences)\n",
    "    \n",
    "    # Assign the minimum difference and nearest POS to the corresponding row in df1\n",
    "    df1.at[index1, 'Distance'] = differences[min_index]\n",
    "    df1.at[index1, 'Nearest_POS'] = df2.at[min_index, 'POS']\n",
    "\n",
    "# Print the updated df1\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "626e83aa",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
