{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "344d9f5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8a78d25a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Zygosity_father</th>\n",
       "      <th>DP_father</th>\n",
       "      <th>RD_father</th>\n",
       "      <th>AD_father</th>\n",
       "      <th>VAF_father</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>140</td>\n",
       "      <td>88</td>\n",
       "      <td>52</td>\n",
       "      <td>0.371429</td>\n",
       "      <td>0.8380</td>\n",
       "      <td>0.9005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>147</td>\n",
       "      <td>1</td>\n",
       "      <td>146</td>\n",
       "      <td>0.993197</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.9854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>137</td>\n",
       "      <td>99</td>\n",
       "      <td>38</td>\n",
       "      <td>0.277372</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.8049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11,LOC107985728</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>80</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31681</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155383098</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>87</td>\n",
       "      <td>59</td>\n",
       "      <td>28</td>\n",
       "      <td>0.321839</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31682</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010159</td>\n",
       "      <td>rs200413398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>0.656250</td>\n",
       "      <td>0.1634</td>\n",
       "      <td>0.2709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31683</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010162</td>\n",
       "      <td>rs150178903</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>33</td>\n",
       "      <td>13</td>\n",
       "      <td>20</td>\n",
       "      <td>0.606061</td>\n",
       "      <td>0.1568</td>\n",
       "      <td>0.2628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31684</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LINC00278</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31685</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>RBMY1E</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>78</td>\n",
       "      <td>58</td>\n",
       "      <td>20</td>\n",
       "      <td>0.256410</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31686 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT            Gene_Name  \\\n",
       "0      chr1      69270  rs201219564   A   G                OR4F5   \n",
       "1      chr1      69511    rs2691305   A   G                OR4F5   \n",
       "2      chr1      69897  rs200676709   T   C                OR4F5   \n",
       "3      chr1     924533  rs112703963   A   G  SAMD11,LOC107985728   \n",
       "4      chr1     942451    rs6672356   T   C               SAMD11   \n",
       "...     ...        ...          ...  ..  ..                  ...   \n",
       "31681  chrX  155383098            .   T   C                        \n",
       "31682  chrX  156010159  rs200413398   A   G                 IL9R   \n",
       "31683  chrX  156010162  rs150178903   A   G                 IL9R   \n",
       "31684  chrY    3019783    rs9786184   A   C            LINC00278   \n",
       "31685  chrY   21904151     rs162865   G   A               RBMY1E   \n",
       "\n",
       "      Zygosity_father  DP_father  RD_father  AD_father  VAF_father  \\\n",
       "0        Heterozygous        140         88         52    0.371429   \n",
       "1          Homozygous        147          1        146    0.993197   \n",
       "2        Heterozygous        137         99         38    0.277372   \n",
       "3          Homozygous         93          0         93    1.000000   \n",
       "4          Homozygous         80          0         80    1.000000   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "31681    Heterozygous         87         59         28    0.321839   \n",
       "31682    Heterozygous         32         11         21    0.656250   \n",
       "31683    Heterozygous         33         13         20    0.606061   \n",
       "31684      Homozygous         39          0         39    1.000000   \n",
       "31685    Heterozygous         78         58         20    0.256410   \n",
       "\n",
       "       gnomADe_AF  gnomADe_SAS_AF  \n",
       "0          0.8380          0.9005  \n",
       "1          0.9497          0.9854  \n",
       "2          0.7209          0.8049  \n",
       "3          0.0000          0.0000  \n",
       "4          0.9999          0.9997  \n",
       "...           ...             ...  \n",
       "31681      0.0000          0.0000  \n",
       "31682      0.1634          0.2709  \n",
       "31683      0.1568          0.2628  \n",
       "31684      0.0000          0.0000  \n",
       "31685      0.0000          0.0000  \n",
       "\n",
       "[31686 rows x 13 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "father = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Covered_father_new.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "father.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "father['DP_father'] = father['SAMPLE'].str.split(':').str[3].fillna('0').replace('', 0).astype(int)\n",
    "father['RD_father'] = father['SAMPLE'].str.split(':').str[4].fillna('0').replace('', 0).astype(int)\n",
    "father['AD_father'] = father['SAMPLE'].str.split(':').str[5].fillna('0').replace('', 0).astype(int)\n",
    "father['VAF_father'] = father['AD_father'] / (father['RD_father'] + father['AD_father']).replace('', 0).astype('float')\n",
    "father['CSQ'] = father['INFO'].str.extract(r'CSQ=(.*)')\n",
    "father['gnomADe_AF'] = father['CSQ'].str.split('|').str[48].replace('', 0).astype('float')\n",
    "father['gnomADe_SAS_AF'] = father['CSQ'].str.split('|').str[56].replace('', 0).astype('float')\n",
    "father[\"Gene_Name\"] = father[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "\n",
    "father['HET'] = father['INFO'].str.extract(r'HET=(\\d)')\n",
    "father['HOM'] = father['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "father['Zygosity_father'] = ''\n",
    "father.loc[father['HOM'] == '1', 'Zygosity_father'] = 'Homozygous'\n",
    "father.loc[father['HET'] == '1', 'Zygosity_father'] = 'Heterozygous'\n",
    "\n",
    "father['Gene_Name'] = father['Gene_Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "father = father[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'Gene_Name', 'Zygosity_father', 'DP_father', 'RD_father', 'AD_father', 'VAF_father', 'gnomADe_AF', 'gnomADe_SAS_AF']]\n",
    "father"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "7bde391e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity_mother</th>\n",
       "      <th>DP_mother</th>\n",
       "      <th>RD_mother</th>\n",
       "      <th>AD_mother</th>\n",
       "      <th>VAF_mother</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>28</td>\n",
       "      <td>0.636364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>133</td>\n",
       "      <td>0</td>\n",
       "      <td>133</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69761</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>66</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>0.272727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>46</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>0.369565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>0.975000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29803</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653251</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>41</td>\n",
       "      <td>26</td>\n",
       "      <td>15</td>\n",
       "      <td>0.365854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29804</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>72</td>\n",
       "      <td>49</td>\n",
       "      <td>23</td>\n",
       "      <td>0.319444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29805</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29806</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154792236</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>30</td>\n",
       "      <td>0.638298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29807</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155997799</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>18</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>0.611111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29808 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT Zygosity_mother  DP_mother  RD_mother  \\\n",
       "0      chr1      69270   A   G    Heterozygous         44         16   \n",
       "1      chr1      69511   A   G      Homozygous        133          0   \n",
       "2      chr1      69761   A   T    Heterozygous         66         48   \n",
       "3      chr1      69897   T   C    Heterozygous         46         29   \n",
       "4      chr1     924533   A   G      Homozygous         40          1   \n",
       "...     ...        ...  ..  ..             ...        ...        ...   \n",
       "29803  chrX  154653251   C   G    Heterozygous         41         26   \n",
       "29804  chrX  154653499   C   T    Heterozygous         72         49   \n",
       "29805  chrX  154766321   G   T      Homozygous         44          0   \n",
       "29806  chrX  154792236   C   T    Heterozygous         47         17   \n",
       "29807  chrX  155997799   C   T    Heterozygous         18          7   \n",
       "\n",
       "       AD_mother  VAF_mother  \n",
       "0             28    0.636364  \n",
       "1            133    1.000000  \n",
       "2             18    0.272727  \n",
       "3             17    0.369565  \n",
       "4             39    0.975000  \n",
       "...          ...         ...  \n",
       "29803         15    0.365854  \n",
       "29804         23    0.319444  \n",
       "29805         43    1.000000  \n",
       "29806         30    0.638298  \n",
       "29807         11    0.611111  \n",
       "\n",
       "[29808 rows x 9 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mother = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Covered_mother_new.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "mother.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "mother['DP_mother'] = mother['SAMPLE'].str.split(':').str[3].fillna('0').replace('', 0).astype(int)\n",
    "mother['RD_mother'] = mother['SAMPLE'].str.split(':').str[4].fillna('0').replace('', 0).astype(int)\n",
    "mother['AD_mother'] = mother['SAMPLE'].str.split(':').str[5].fillna('0').replace('', 0).astype(int)\n",
    "mother['VAF_mother'] = mother['AD_mother'] / (mother['RD_mother'] + mother['AD_mother']).replace('', 0).astype('float')\n",
    "\n",
    "mother['HET'] = mother['INFO'].str.extract(r'HET=(\\d)')\n",
    "mother['HOM'] = mother['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "mother['Zygosity_mother'] = ''\n",
    "mother.loc[mother['HOM'] == '1', 'Zygosity_mother'] = 'Homozygous'\n",
    "mother.loc[mother['HET'] == '1', 'Zygosity_mother'] = 'Heterozygous'\n",
    "\n",
    "mother['CSQ'] = mother['INFO'].str.extract(r'CSQ=(.*)')\n",
    "mother['gnomADe_AF'] = mother['CSQ'].str.split('|').str[48].replace('', 0).astype('float')\n",
    "mother['gnomADe_SAS_AF'] = mother['CSQ'].str.split('|').str[56].replace('', 0).astype('float')\n",
    "mother = mother[['CHROM', 'POS', 'REF', 'ALT', 'Zygosity_mother', 'DP_mother', 'RD_mother', 'AD_mother', 'VAF_mother']]\n",
    "mother"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "24901542",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity_son</th>\n",
       "      <th>DP_son</th>\n",
       "      <th>RD_son</th>\n",
       "      <th>AD_son</th>\n",
       "      <th>VAF_son</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>104</td>\n",
       "      <td>30</td>\n",
       "      <td>74</td>\n",
       "      <td>0.711538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>205</td>\n",
       "      <td>0</td>\n",
       "      <td>205</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>93</td>\n",
       "      <td>28</td>\n",
       "      <td>65</td>\n",
       "      <td>0.698925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>98</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32709</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32710</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32711</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32712</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12914512</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32713</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>52</td>\n",
       "      <td>41</td>\n",
       "      <td>11</td>\n",
       "      <td>0.211538</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32714 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT  Zygosity_son  DP_son  RD_son  AD_son   VAF_son\n",
       "0      chr1      69270   A   G  Heterozygous     104      30      74  0.711538\n",
       "1      chr1      69511   A   G    Homozygous     205       0     205  1.000000\n",
       "2      chr1      69897   T   C  Heterozygous      93      28      65  0.698925\n",
       "3      chr1     924533   A   G    Homozygous      81       0      80  1.000000\n",
       "4      chr1     942451   T   C    Homozygous      98       0      98  1.000000\n",
       "...     ...        ...  ..  ..           ...     ...     ...     ...       ...\n",
       "32709  chrX  155277884   G   T    Homozygous      28       0      28  1.000000\n",
       "32710  chrX  156003433   T   C  Heterozygous       9       3       6  0.666667\n",
       "32711  chrY    3019783   A   C    Homozygous      42       0      42  1.000000\n",
       "32712  chrY   12914512   C   A    Homozygous      18       0      18  1.000000\n",
       "32713  chrY   21904151   G   A  Heterozygous      52      41      11  0.211538\n",
       "\n",
       "[32714 rows x 9 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "son = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Covered_son_new.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "son.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "son['DP_son'] = son['SAMPLE'].str.split(':').str[3].fillna('0').replace('', 0).astype(int)\n",
    "son['RD_son'] = son['SAMPLE'].str.split(':').str[4].fillna('0').replace('', 0).astype(int)\n",
    "son['AD_son'] = son['SAMPLE'].str.split(':').str[5].fillna('0').replace('', 0).astype(int)\n",
    "son['VAF_son'] = son['AD_son'] / (son['RD_son'] + son['AD_son']).replace('', 0).astype('float')\n",
    "\n",
    "son['HET'] = son['INFO'].str.extract(r'HET=(\\d)')\n",
    "son['HOM'] = son['INFO'].str.extract(r'HOM=(\\d)')\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "son['Zygosity_son'] = ''\n",
    "son.loc[son['HOM'] == '1', 'Zygosity_son'] = 'Homozygous'\n",
    "son.loc[son['HET'] == '1', 'Zygosity_son'] = 'Heterozygous'\n",
    "\n",
    "son['CSQ'] = son['INFO'].str.extract(r'CSQ=(.*)')\n",
    "son['gnomADe_AF'] = son['CSQ'].str.split('|').str[48].replace('', 0).astype('float')\n",
    "son['gnomADe_SAS_AF'] = son['CSQ'].str.split('|').str[56].replace('', 0).astype('float')\n",
    "son = son[['CHROM', 'POS', 'REF', 'ALT', 'Zygosity_son', 'DP_son', 'RD_son', 'AD_son', 'VAF_son']]\n",
    "son"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "80346063",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15198</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154485448</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15199</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15200</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15203 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT\n",
       "0      chr1      69270   A   G\n",
       "1      chr1      69511   A   G\n",
       "2      chr1      69897   T   C\n",
       "3      chr1     924533   A   G\n",
       "4      chr1     942451   T   C\n",
       "...     ...        ...  ..  ..\n",
       "15198  chrX  154485448   T   C\n",
       "15199  chrX  154563953   C   T\n",
       "15200  chrX  154652556   C   A\n",
       "15201  chrX  154653499   C   T\n",
       "15202  chrX  154766321   G   T\n",
       "\n",
       "[15203 rows x 4 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/latest_trio/intersected_variants.xlsx')\n",
    "intersection = intersection[['CHROM', 'POS', 'REF', 'ALT']]\n",
    "intersection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "fd6d5e00",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Zygosity_father</th>\n",
       "      <th>DP_father</th>\n",
       "      <th>RD_father</th>\n",
       "      <th>AD_father</th>\n",
       "      <th>VAF_father</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>140</td>\n",
       "      <td>88</td>\n",
       "      <td>52</td>\n",
       "      <td>0.371429</td>\n",
       "      <td>0.8380</td>\n",
       "      <td>0.9005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>147</td>\n",
       "      <td>1</td>\n",
       "      <td>146</td>\n",
       "      <td>0.993197</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.9854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>137</td>\n",
       "      <td>99</td>\n",
       "      <td>38</td>\n",
       "      <td>0.277372</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.8049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>SAMD11,LOC107985728</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>80</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15198</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154485448</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs7057286</td>\n",
       "      <td>UBL4A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15199</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>IKBKG</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>59</td>\n",
       "      <td>15</td>\n",
       "      <td>44</td>\n",
       "      <td>0.745763</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15200</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs4326559</td>\n",
       "      <td>CTAG2,LOC105373387</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>0.975000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>CTAG2,LOC105373387</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>59</td>\n",
       "      <td>28</td>\n",
       "      <td>31</td>\n",
       "      <td>0.525424</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.9979</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15203 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT                   rsID            Gene_Name  \\\n",
       "0      chr1      69270   A   G            rs201219564                OR4F5   \n",
       "1      chr1      69511   A   G              rs2691305                OR4F5   \n",
       "2      chr1      69897   T   C            rs200676709                OR4F5   \n",
       "3      chr1     924533   A   G            rs112703963  SAMD11,LOC107985728   \n",
       "4      chr1     942451   T   C              rs6672356               SAMD11   \n",
       "...     ...        ...  ..  ..                    ...                  ...   \n",
       "15198  chrX  154485448   T   C              rs7057286                UBL4A   \n",
       "15199  chrX  154563953   C   T  rs201709278;rs5945206                IKBKG   \n",
       "15200  chrX  154652556   C   A              rs4326559   CTAG2,LOC105373387   \n",
       "15201  chrX  154653499   C   T             rs17855367   CTAG2,LOC105373387   \n",
       "15202  chrX  154766321   G   T              rs2728532                 DKC1   \n",
       "\n",
       "      Zygosity_father  DP_father  RD_father  AD_father  VAF_father  \\\n",
       "0        Heterozygous        140         88         52    0.371429   \n",
       "1          Homozygous        147          1        146    0.993197   \n",
       "2        Heterozygous        137         99         38    0.277372   \n",
       "3          Homozygous         93          0         93    1.000000   \n",
       "4          Homozygous         80          0         80    1.000000   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "15198      Homozygous         49          0         49    1.000000   \n",
       "15199    Heterozygous         59         15         44    0.745763   \n",
       "15200      Homozygous         40          1         39    0.975000   \n",
       "15201    Heterozygous         59         28         31    0.525424   \n",
       "15202      Homozygous         36          0         36    1.000000   \n",
       "\n",
       "       gnomADe_AF  gnomADe_SAS_AF  \n",
       "0          0.8380          0.9005  \n",
       "1          0.9497          0.9854  \n",
       "2          0.7209          0.8049  \n",
       "3          0.0000          0.0000  \n",
       "4          0.9999          0.9997  \n",
       "...           ...             ...  \n",
       "15198      0.0000          0.0000  \n",
       "15199      0.0000          0.0000  \n",
       "15200      0.0000          0.0000  \n",
       "15201      0.0000          0.0000  \n",
       "15202      0.9979          1.0000  \n",
       "\n",
       "[15203 rows x 13 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection_father = pd.merge(intersection, father, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "intersection_father"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "8f3a2784",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Zygosity_father</th>\n",
       "      <th>DP_father</th>\n",
       "      <th>RD_father</th>\n",
       "      <th>AD_father</th>\n",
       "      <th>VAF_father</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>Zygosity_mother</th>\n",
       "      <th>DP_mother</th>\n",
       "      <th>RD_mother</th>\n",
       "      <th>AD_mother</th>\n",
       "      <th>VAF_mother</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>140</td>\n",
       "      <td>88</td>\n",
       "      <td>52</td>\n",
       "      <td>0.371429</td>\n",
       "      <td>0.8380</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>28</td>\n",
       "      <td>0.636364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>147</td>\n",
       "      <td>1</td>\n",
       "      <td>146</td>\n",
       "      <td>0.993197</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>133</td>\n",
       "      <td>0</td>\n",
       "      <td>133</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>137</td>\n",
       "      <td>99</td>\n",
       "      <td>38</td>\n",
       "      <td>0.277372</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>46</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>0.369565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>SAMD11,LOC107985728</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>0.975000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>80</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15198</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154485448</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs7057286</td>\n",
       "      <td>UBL4A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>75</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15199</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>IKBKG</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>59</td>\n",
       "      <td>15</td>\n",
       "      <td>44</td>\n",
       "      <td>0.745763</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15200</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs4326559</td>\n",
       "      <td>CTAG2,LOC105373387</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>0.975000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>39</td>\n",
       "      <td>23</td>\n",
       "      <td>16</td>\n",
       "      <td>0.410256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>CTAG2,LOC105373387</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>59</td>\n",
       "      <td>28</td>\n",
       "      <td>31</td>\n",
       "      <td>0.525424</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>72</td>\n",
       "      <td>49</td>\n",
       "      <td>23</td>\n",
       "      <td>0.319444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.9979</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15203 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT                   rsID            Gene_Name  \\\n",
       "0      chr1      69270   A   G            rs201219564                OR4F5   \n",
       "1      chr1      69511   A   G              rs2691305                OR4F5   \n",
       "2      chr1      69897   T   C            rs200676709                OR4F5   \n",
       "3      chr1     924533   A   G            rs112703963  SAMD11,LOC107985728   \n",
       "4      chr1     942451   T   C              rs6672356               SAMD11   \n",
       "...     ...        ...  ..  ..                    ...                  ...   \n",
       "15198  chrX  154485448   T   C              rs7057286                UBL4A   \n",
       "15199  chrX  154563953   C   T  rs201709278;rs5945206                IKBKG   \n",
       "15200  chrX  154652556   C   A              rs4326559   CTAG2,LOC105373387   \n",
       "15201  chrX  154653499   C   T             rs17855367   CTAG2,LOC105373387   \n",
       "15202  chrX  154766321   G   T              rs2728532                 DKC1   \n",
       "\n",
       "      Zygosity_father  DP_father  RD_father  AD_father  VAF_father  \\\n",
       "0        Heterozygous        140         88         52    0.371429   \n",
       "1          Homozygous        147          1        146    0.993197   \n",
       "2        Heterozygous        137         99         38    0.277372   \n",
       "3          Homozygous         93          0         93    1.000000   \n",
       "4          Homozygous         80          0         80    1.000000   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "15198      Homozygous         49          0         49    1.000000   \n",
       "15199    Heterozygous         59         15         44    0.745763   \n",
       "15200      Homozygous         40          1         39    0.975000   \n",
       "15201    Heterozygous         59         28         31    0.525424   \n",
       "15202      Homozygous         36          0         36    1.000000   \n",
       "\n",
       "       gnomADe_AF  gnomADe_SAS_AF Zygosity_mother  DP_mother  RD_mother  \\\n",
       "0          0.8380          0.9005    Heterozygous         44         16   \n",
       "1          0.9497          0.9854      Homozygous        133          0   \n",
       "2          0.7209          0.8049    Heterozygous         46         29   \n",
       "3          0.0000          0.0000      Homozygous         40          1   \n",
       "4          0.9999          0.9997      Homozygous         38          0   \n",
       "...           ...             ...             ...        ...        ...   \n",
       "15198      0.0000          0.0000      Homozygous         75          0   \n",
       "15199      0.0000          0.0000      Homozygous         35          0   \n",
       "15200      0.0000          0.0000    Heterozygous         39         23   \n",
       "15201      0.0000          0.0000    Heterozygous         72         49   \n",
       "15202      0.9979          1.0000      Homozygous         44          0   \n",
       "\n",
       "       AD_mother  VAF_mother  \n",
       "0             28    0.636364  \n",
       "1            133    1.000000  \n",
       "2             17    0.369565  \n",
       "3             39    0.975000  \n",
       "4             38    1.000000  \n",
       "...          ...         ...  \n",
       "15198         75    1.000000  \n",
       "15199         35    1.000000  \n",
       "15200         16    0.410256  \n",
       "15201         23    0.319444  \n",
       "15202         43    1.000000  \n",
       "\n",
       "[15203 rows x 18 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection_mother = pd.merge(intersection_father, mother, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "intersection_mother"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "e5740f20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Zygosity_father</th>\n",
       "      <th>DP_father</th>\n",
       "      <th>RD_father</th>\n",
       "      <th>AD_father</th>\n",
       "      <th>VAF_father</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>Zygosity_mother</th>\n",
       "      <th>DP_mother</th>\n",
       "      <th>RD_mother</th>\n",
       "      <th>AD_mother</th>\n",
       "      <th>VAF_mother</th>\n",
       "      <th>Zygosity_son</th>\n",
       "      <th>DP_son</th>\n",
       "      <th>RD_son</th>\n",
       "      <th>AD_son</th>\n",
       "      <th>VAF_son</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>140</td>\n",
       "      <td>88</td>\n",
       "      <td>52</td>\n",
       "      <td>0.371429</td>\n",
       "      <td>0.8380</td>\n",
       "      <td>0.9005</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>28</td>\n",
       "      <td>0.636364</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>104</td>\n",
       "      <td>30</td>\n",
       "      <td>74</td>\n",
       "      <td>0.711538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>147</td>\n",
       "      <td>1</td>\n",
       "      <td>146</td>\n",
       "      <td>0.993197</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>133</td>\n",
       "      <td>0</td>\n",
       "      <td>133</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>205</td>\n",
       "      <td>0</td>\n",
       "      <td>205</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>137</td>\n",
       "      <td>99</td>\n",
       "      <td>38</td>\n",
       "      <td>0.277372</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>46</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>0.369565</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>93</td>\n",
       "      <td>28</td>\n",
       "      <td>65</td>\n",
       "      <td>0.698925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>SAMD11,LOC107985728</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>0.975000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>80</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>98</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15198</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154485448</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs7057286</td>\n",
       "      <td>UBL4A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>75</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>67</td>\n",
       "      <td>0</td>\n",
       "      <td>67</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15199</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>IKBKG</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>59</td>\n",
       "      <td>15</td>\n",
       "      <td>44</td>\n",
       "      <td>0.745763</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15200</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs4326559</td>\n",
       "      <td>CTAG2,LOC105373387</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>0.975000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>39</td>\n",
       "      <td>23</td>\n",
       "      <td>16</td>\n",
       "      <td>0.410256</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>CTAG2,LOC105373387</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>59</td>\n",
       "      <td>28</td>\n",
       "      <td>31</td>\n",
       "      <td>0.525424</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>72</td>\n",
       "      <td>49</td>\n",
       "      <td>23</td>\n",
       "      <td>0.319444</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>86</td>\n",
       "      <td>41</td>\n",
       "      <td>45</td>\n",
       "      <td>0.523256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.9979</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>57</td>\n",
       "      <td>0</td>\n",
       "      <td>57</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15203 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT                   rsID            Gene_Name  \\\n",
       "0      chr1      69270   A   G            rs201219564                OR4F5   \n",
       "1      chr1      69511   A   G              rs2691305                OR4F5   \n",
       "2      chr1      69897   T   C            rs200676709                OR4F5   \n",
       "3      chr1     924533   A   G            rs112703963  SAMD11,LOC107985728   \n",
       "4      chr1     942451   T   C              rs6672356               SAMD11   \n",
       "...     ...        ...  ..  ..                    ...                  ...   \n",
       "15198  chrX  154485448   T   C              rs7057286                UBL4A   \n",
       "15199  chrX  154563953   C   T  rs201709278;rs5945206                IKBKG   \n",
       "15200  chrX  154652556   C   A              rs4326559   CTAG2,LOC105373387   \n",
       "15201  chrX  154653499   C   T             rs17855367   CTAG2,LOC105373387   \n",
       "15202  chrX  154766321   G   T              rs2728532                 DKC1   \n",
       "\n",
       "      Zygosity_father  DP_father  RD_father  AD_father  VAF_father  \\\n",
       "0        Heterozygous        140         88         52    0.371429   \n",
       "1          Homozygous        147          1        146    0.993197   \n",
       "2        Heterozygous        137         99         38    0.277372   \n",
       "3          Homozygous         93          0         93    1.000000   \n",
       "4          Homozygous         80          0         80    1.000000   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "15198      Homozygous         49          0         49    1.000000   \n",
       "15199    Heterozygous         59         15         44    0.745763   \n",
       "15200      Homozygous         40          1         39    0.975000   \n",
       "15201    Heterozygous         59         28         31    0.525424   \n",
       "15202      Homozygous         36          0         36    1.000000   \n",
       "\n",
       "       gnomADe_AF  gnomADe_SAS_AF Zygosity_mother  DP_mother  RD_mother  \\\n",
       "0          0.8380          0.9005    Heterozygous         44         16   \n",
       "1          0.9497          0.9854      Homozygous        133          0   \n",
       "2          0.7209          0.8049    Heterozygous         46         29   \n",
       "3          0.0000          0.0000      Homozygous         40          1   \n",
       "4          0.9999          0.9997      Homozygous         38          0   \n",
       "...           ...             ...             ...        ...        ...   \n",
       "15198      0.0000          0.0000      Homozygous         75          0   \n",
       "15199      0.0000          0.0000      Homozygous         35          0   \n",
       "15200      0.0000          0.0000    Heterozygous         39         23   \n",
       "15201      0.0000          0.0000    Heterozygous         72         49   \n",
       "15202      0.9979          1.0000      Homozygous         44          0   \n",
       "\n",
       "       AD_mother  VAF_mother  Zygosity_son  DP_son  RD_son  AD_son   VAF_son  \n",
       "0             28    0.636364  Heterozygous     104      30      74  0.711538  \n",
       "1            133    1.000000    Homozygous     205       0     205  1.000000  \n",
       "2             17    0.369565  Heterozygous      93      28      65  0.698925  \n",
       "3             39    0.975000    Homozygous      81       0      80  1.000000  \n",
       "4             38    1.000000    Homozygous      98       0      98  1.000000  \n",
       "...          ...         ...           ...     ...     ...     ...       ...  \n",
       "15198         75    1.000000    Homozygous      67       0      67  1.000000  \n",
       "15199         35    1.000000    Homozygous      33       0      33  1.000000  \n",
       "15200         16    0.410256    Homozygous      43       0      43  1.000000  \n",
       "15201         23    0.319444  Heterozygous      86      41      45  0.523256  \n",
       "15202         43    1.000000    Homozygous      57       0      57  1.000000  \n",
       "\n",
       "[15203 rows x 23 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection_all = pd.merge(intersection_mother, son, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "intersection_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c9d30184",
   "metadata": {},
   "outputs": [],
   "source": [
    "intersection_all.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Intersection_FMS_DP_VAF_Zygosity.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6cd6af3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "40bc9ec4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>DP</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>181</td>\n",
       "      <td>140</td>\n",
       "      <td>88</td>\n",
       "      <td>52</td>\n",
       "      <td>37.14%</td>\n",
       "      <td>6.374300e-19</td>\n",
       "      <td>59</td>\n",
       "      <td>49</td>\n",
       "      <td>77</td>\n",
       "      <td>11</td>\n",
       "      <td>43</td>\n",
       "      <td>9</td>\n",
       "      <td>140</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>0.8380</td>\n",
       "      <td>0.9005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>147</td>\n",
       "      <td>1</td>\n",
       "      <td>146</td>\n",
       "      <td>99.32%</td>\n",
       "      <td>1.000100e-85</td>\n",
       "      <td>37</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>115</td>\n",
       "      <td>31</td>\n",
       "      <td>147</td>\n",
       "      <td>0.8460</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.9854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>137</td>\n",
       "      <td>99</td>\n",
       "      <td>38</td>\n",
       "      <td>27.74%</td>\n",
       "      <td>1.838500e-13</td>\n",
       "      <td>56</td>\n",
       "      <td>50</td>\n",
       "      <td>63</td>\n",
       "      <td>36</td>\n",
       "      <td>27</td>\n",
       "      <td>11</td>\n",
       "      <td>137</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.6220</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.8049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.745100e-55</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>70</td>\n",
       "      <td>23</td>\n",
       "      <td>93</td>\n",
       "      <td>0.8215</td>\n",
       "      <td>0.9098</td>\n",
       "      <td>G|upstream_gene_variant|MODIFIER|SAMD11|ENSG00...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>80</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.086400e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>26</td>\n",
       "      <td>80</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS         rsID REF ALT QUAL FILTER  \\\n",
       "0  chr1   69270  rs201219564   A   G    .   PASS   \n",
       "1  chr1   69511    rs2691305   A   G    .   PASS   \n",
       "2  chr1   69897  rs200676709   T   C    .   PASS   \n",
       "3  chr1  924533  rs112703963   A   G    .   PASS   \n",
       "4  chr1  942451    rs6672356   T   C    .   PASS   \n",
       "\n",
       "                                                INFO   GT   GQ  SDP  RD   AD  \\\n",
       "0  ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  181  140  88   52   \n",
       "1  ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  147   1  146   \n",
       "2  ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...  0/1  127  137  99   38   \n",
       "3  ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1  255   93   0   93   \n",
       "4  ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...  1/1  255   80   0   80   \n",
       "\n",
       "     FREQ          PVAL  RBQ  ABQ  RDF  RDR  ADF  ADR   DP  gnomADg_AF  \\\n",
       "0  37.14%  6.374300e-19   59   49   77   11   43    9  140      0.6291   \n",
       "1  99.32%  1.000100e-85   37   55    1    0  115   31  147      0.8460   \n",
       "2  27.74%  1.838500e-13   56   50   63   36   27   11  137      0.4864   \n",
       "3    100%  1.745100e-55    0   54    0    0   70   23   93      0.8215   \n",
       "4    100%  1.086400e-47    0   51    0    0   54   26   80      0.9998   \n",
       "\n",
       "   gnomADg_SAS_AF                                                CSQ  \\\n",
       "0          0.8855  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1          0.9772  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2          0.6220  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3          0.9098  G|upstream_gene_variant|MODIFIER|SAMD11|ENSG00...   \n",
       "4          0.9996  C|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "\n",
       "   gnomADe_AF  gnomADe_SAS_AF  \n",
       "0      0.8380          0.9005  \n",
       "1      0.9497          0.9854  \n",
       "2      0.7209          0.8049  \n",
       "3      0.0000          0.0000  \n",
       "4      0.9999          0.9997  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "father = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/latest_trio/Father_data_columns.xlsx')\n",
    "father['CSQ'] = father['INFO'].str.extract(r'CSQ=(.*)')\n",
    "father['gnomADe_AF'] = father['CSQ'].str.split('|').str[48].replace('', 0).astype('float')\n",
    "father['gnomADe_SAS_AF'] = father['CSQ'].str.split('|').str[56].replace('', 0).astype('float')\n",
    "father.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "77f8012e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>DP</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>112</td>\n",
       "      <td>44</td>\n",
       "      <td>16</td>\n",
       "      <td>28</td>\n",
       "      <td>63.64%</td>\n",
       "      <td>5.699700e-12</td>\n",
       "      <td>66</td>\n",
       "      <td>47</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "      <td>6</td>\n",
       "      <td>44</td>\n",
       "      <td>0.62910</td>\n",
       "      <td>0.88550</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>0.83800</td>\n",
       "      <td>0.90050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=133;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>133</td>\n",
       "      <td>0</td>\n",
       "      <td>133</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.725600e-79</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>108</td>\n",
       "      <td>25</td>\n",
       "      <td>133</td>\n",
       "      <td>0.84600</td>\n",
       "      <td>0.97720</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>0.94970</td>\n",
       "      <td>0.98540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69761</td>\n",
       "      <td>rs200505207</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=66;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9633,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>60</td>\n",
       "      <td>66</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>27.27%</td>\n",
       "      <td>9.974000e-07</td>\n",
       "      <td>57</td>\n",
       "      <td>61</td>\n",
       "      <td>35</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>66</td>\n",
       "      <td>0.06088</td>\n",
       "      <td>0.04502</td>\n",
       "      <td>T|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>0.09168</td>\n",
       "      <td>0.07638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=46;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>59</td>\n",
       "      <td>46</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>36.96%</td>\n",
       "      <td>1.241300e-06</td>\n",
       "      <td>62</td>\n",
       "      <td>59</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>46</td>\n",
       "      <td>0.48640</td>\n",
       "      <td>0.62200</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>0.72090</td>\n",
       "      <td>0.80490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>214</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>97.5%</td>\n",
       "      <td>3.813700e-22</td>\n",
       "      <td>62</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>40</td>\n",
       "      <td>0.82150</td>\n",
       "      <td>0.90980</td>\n",
       "      <td>G|upstream_gene_variant|MODIFIER|SAMD11|ENSG00...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS         rsID REF ALT QUAL FILTER  \\\n",
       "0  chr1   69270  rs201219564   A   G    .   PASS   \n",
       "1  chr1   69511    rs2691305   A   G    .   PASS   \n",
       "2  chr1   69761  rs200505207   A   T    .   PASS   \n",
       "3  chr1   69897  rs200676709   T   C    .   PASS   \n",
       "4  chr1  924533  rs112703963   A   G    .   PASS   \n",
       "\n",
       "                                                INFO   GT   GQ  SDP  RD   AD  \\\n",
       "0  ADP=44;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...  0/1  112   44  16   28   \n",
       "1  ADP=133;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  133   0  133   \n",
       "2  ADP=66;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9633,0....  0/1   60   66  48   18   \n",
       "3  ADP=46;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1   59   46  29   17   \n",
       "4  ADP=40;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1  214   40   1   39   \n",
       "\n",
       "     FREQ          PVAL  RBQ  ABQ  RDF  RDR  ADF  ADR   DP  gnomADg_AF  \\\n",
       "0  63.64%  5.699700e-12   66   47   15    1   22    6   44     0.62910   \n",
       "1    100%  1.725600e-79    0   55    0    0  108   25  133     0.84600   \n",
       "2  27.27%  9.974000e-07   57   61   35   13   14    4   66     0.06088   \n",
       "3  36.96%  1.241300e-06   62   59   22    7   15    2   46     0.48640   \n",
       "4   97.5%  3.813700e-22   62   56    1    0   30    9   40     0.82150   \n",
       "\n",
       "   gnomADg_SAS_AF                                                CSQ  \\\n",
       "0         0.88550  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1         0.97720  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2         0.04502  T|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "3         0.62200  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "4         0.90980  G|upstream_gene_variant|MODIFIER|SAMD11|ENSG00...   \n",
       "\n",
       "   gnomADe_AF  gnomADe_SAS_AF  \n",
       "0     0.83800         0.90050  \n",
       "1     0.94970         0.98540  \n",
       "2     0.09168         0.07638  \n",
       "3     0.72090         0.80490  \n",
       "4     0.00000         0.00000  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mother = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/latest_trio/Mother_data_columns.xlsx')\n",
    "mother['CSQ'] = mother['INFO'].str.extract(r'CSQ=(.*)')\n",
    "mother['gnomADe_AF'] = mother['CSQ'].str.split('|').str[48].replace('', 0).astype('float')\n",
    "mother['gnomADe_SAS_AF'] = mother['CSQ'].str.split('|').str[56].replace('', 0).astype('float')\n",
    "mother.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "55383b43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RBQ</th>\n",
       "      <th>ABQ</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>DP</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=104;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>104</td>\n",
       "      <td>30</td>\n",
       "      <td>74</td>\n",
       "      <td>71.15%</td>\n",
       "      <td>3.209300e-32</td>\n",
       "      <td>58</td>\n",
       "      <td>49</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>59</td>\n",
       "      <td>15</td>\n",
       "      <td>104</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>0.8380</td>\n",
       "      <td>0.9005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=205;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>205</td>\n",
       "      <td>0</td>\n",
       "      <td>205</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.603300e-123</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>60</td>\n",
       "      <td>205</td>\n",
       "      <td>0.8460</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.9854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>93</td>\n",
       "      <td>28</td>\n",
       "      <td>65</td>\n",
       "      <td>69.89%</td>\n",
       "      <td>4.005100e-28</td>\n",
       "      <td>53</td>\n",
       "      <td>51</td>\n",
       "      <td>15</td>\n",
       "      <td>13</td>\n",
       "      <td>49</td>\n",
       "      <td>16</td>\n",
       "      <td>93</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.6220</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.8049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=81;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>98.77%</td>\n",
       "      <td>1.086400e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>52</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>32</td>\n",
       "      <td>81</td>\n",
       "      <td>0.8215</td>\n",
       "      <td>0.9098</td>\n",
       "      <td>G|upstream_gene_variant|MODIFIER|SAMD11|ENSG00...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>98</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.749300e-58</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>98</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS         rsID REF ALT QUAL FILTER  \\\n",
       "0  chr1   69270  rs201219564   A   G    .   PASS   \n",
       "1  chr1   69511    rs2691305   A   G    .   PASS   \n",
       "2  chr1   69897  rs200676709   T   C    .   PASS   \n",
       "3  chr1  924533  rs112703963   A   G    .   PASS   \n",
       "4  chr1  942451    rs6672356   T   C    .   PASS   \n",
       "\n",
       "                                                INFO   GT   GQ  SDP  RD   AD  \\\n",
       "0  ADP=104;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  255  104  30   74   \n",
       "1  ADP=205;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  205   0  205   \n",
       "2  ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  255   93  28   65   \n",
       "3  ADP=81;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1  255   81   0   80   \n",
       "4  ADP=98;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...  1/1  255   98   0   98   \n",
       "\n",
       "     FREQ           PVAL  RBQ  ABQ  RDF  RDR  ADF  ADR   DP  gnomADg_AF  \\\n",
       "0  71.15%   3.209300e-32   58   49   21    9   59   15  104      0.6291   \n",
       "1    100%  9.603300e-123    0   54    0    0  145   60  205      0.8460   \n",
       "2  69.89%   4.005100e-28   53   51   15   13   49   16   93      0.4864   \n",
       "3  98.77%   1.086400e-47    0   52    0    0   48   32   81      0.8215   \n",
       "4    100%   1.749300e-58    0   48    0    0   66   32   98      0.9998   \n",
       "\n",
       "   gnomADg_SAS_AF                                                CSQ  \\\n",
       "0          0.8855  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1          0.9772  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2          0.6220  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3          0.9098  G|upstream_gene_variant|MODIFIER|SAMD11|ENSG00...   \n",
       "4          0.9996  C|downstream_gene_variant|MODIFIER|NOC2L|ENSG0...   \n",
       "\n",
       "   gnomADe_AF  gnomADe_SAS_AF  \n",
       "0      0.8380          0.9005  \n",
       "1      0.9497          0.9854  \n",
       "2      0.7209          0.8049  \n",
       "3      0.0000          0.0000  \n",
       "4      0.9999          0.9997  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "son = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/latest_trio/Son_data_columns.xlsx')\n",
    "son['CSQ'] = son['INFO'].str.extract(r'CSQ=(.*)')\n",
    "son['gnomADe_AF'] = son['CSQ'].str.split('|').str[48].replace('', 0).astype('float')\n",
    "son['gnomADe_SAS_AF'] = son['CSQ'].str.split('|').str[56].replace('', 0).astype('float')\n",
    "son.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6b1524ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs201219564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs2691305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs200676709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs112703963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs6672356</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS REF ALT         rsID\n",
       "0  chr1   69270   A   G  rs201219564\n",
       "1  chr1   69511   A   G    rs2691305\n",
       "2  chr1   69897   T   C  rs200676709\n",
       "3  chr1  924533   A   G  rs112703963\n",
       "4  chr1  942451   T   C    rs6672356"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/latest_trio/intersected_variants.xlsx')\n",
    "intersection.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7871c8f1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be118f39",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "623d0bda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31681</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155383098</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31682</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010159</td>\n",
       "      <td>rs200413398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31683</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010162</td>\n",
       "      <td>rs150178903</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31684</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31685</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31686 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT  \\\n",
       "0      chr1      69270  rs201219564   A   G   \n",
       "1      chr1      69511    rs2691305   A   G   \n",
       "2      chr1      69897  rs200676709   T   C   \n",
       "3      chr1     924533  rs112703963   A   G   \n",
       "4      chr1     942451    rs6672356   T   C   \n",
       "...     ...        ...          ...  ..  ..   \n",
       "31681  chrX  155383098            .   T   C   \n",
       "31682  chrX  156010159  rs200413398   A   G   \n",
       "31683  chrX  156010162  rs150178903   A   G   \n",
       "31684  chrY    3019783    rs9786184   A   C   \n",
       "31685  chrY   21904151     rs162865   G   A   \n",
       "\n",
       "                                                    INFO  \n",
       "0      ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  \n",
       "1      ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  \n",
       "2      ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...  \n",
       "3      ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  \n",
       "4      ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...  \n",
       "...                                                  ...  \n",
       "31681  ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  \n",
       "31682  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...  \n",
       "31683  ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....  \n",
       "31684  ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....  \n",
       "31685  ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...  \n",
       "\n",
       "[31686 rows x 6 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "father_dummy = father.copy()\n",
    "father_dummy = father_dummy[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'INFO']]\n",
    "father_dummy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "dfa2d4ec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31681</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155383098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31682</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010159</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31683</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010162</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31684</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LINC00278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31685</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>RBMY1E</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31686 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT            Gene_Name\n",
       "0      chr1      69270   A   G                OR4F5\n",
       "1      chr1      69511   A   G                OR4F5\n",
       "2      chr1      69897   T   C                OR4F5\n",
       "3      chr1     924533   A   G  LOC107985728,SAMD11\n",
       "4      chr1     942451   T   C               SAMD11\n",
       "...     ...        ...  ..  ..                  ...\n",
       "31681  chrX  155383098   T   C                     \n",
       "31682  chrX  156010159   A   G                 IL9R\n",
       "31683  chrX  156010162   A   G                 IL9R\n",
       "31684  chrY    3019783   A   C            LINC00278\n",
       "31685  chrY   21904151   G   A               RBMY1E\n",
       "\n",
       "[31686 rows x 5 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "father_dummy[\"Gene_Name\"] = father_dummy[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "father_dummy['Gene_Name'] = father_dummy['Gene_Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "father_dummy = father_dummy[['CHROM', 'POS', 'REF', 'ALT', 'Gene_Name']]\n",
    "father_dummy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "fdd6ac88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15198</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154485448</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs7057286</td>\n",
       "      <td>UBL4A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15199</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>IKBKG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15200</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs4326559</td>\n",
       "      <td>LOC105373387,CTAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>LOC105373387,CTAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>DKC1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15203 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT                   rsID            Gene_Name\n",
       "0      chr1      69270   A   G            rs201219564                OR4F5\n",
       "1      chr1      69511   A   G              rs2691305                OR4F5\n",
       "2      chr1      69897   T   C            rs200676709                OR4F5\n",
       "3      chr1     924533   A   G            rs112703963  LOC107985728,SAMD11\n",
       "4      chr1     942451   T   C              rs6672356               SAMD11\n",
       "...     ...        ...  ..  ..                    ...                  ...\n",
       "15198  chrX  154485448   T   C              rs7057286                UBL4A\n",
       "15199  chrX  154563953   C   T  rs201709278;rs5945206                IKBKG\n",
       "15200  chrX  154652556   C   A              rs4326559   LOC105373387,CTAG2\n",
       "15201  chrX  154653499   C   T             rs17855367   LOC105373387,CTAG2\n",
       "15202  chrX  154766321   G   T              rs2728532                 DKC1\n",
       "\n",
       "[15203 rows x 6 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection_data = pd.merge(intersection, father_dummy, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "intersection_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "e03167a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>DP</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>104</td>\n",
       "      <td>0.6291</td>\n",
       "      <td>0.8855</td>\n",
       "      <td>0.8380</td>\n",
       "      <td>0.9005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>205</td>\n",
       "      <td>0.8460</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.9854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>93</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.6220</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.8049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>81</td>\n",
       "      <td>0.8215</td>\n",
       "      <td>0.9098</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>98</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9996</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15198</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154485448</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs7057286</td>\n",
       "      <td>UBL4A</td>\n",
       "      <td>67</td>\n",
       "      <td>0.6669</td>\n",
       "      <td>0.5396</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15199</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>IKBKG</td>\n",
       "      <td>33</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15200</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs4326559</td>\n",
       "      <td>LOC105373387,CTAG2</td>\n",
       "      <td>43</td>\n",
       "      <td>0.4733</td>\n",
       "      <td>0.4243</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>LOC105373387,CTAG2</td>\n",
       "      <td>86</td>\n",
       "      <td>0.4428</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>57</td>\n",
       "      <td>0.9924</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9979</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15203 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT                   rsID            Gene_Name  \\\n",
       "0      chr1      69270   A   G            rs201219564                OR4F5   \n",
       "1      chr1      69511   A   G              rs2691305                OR4F5   \n",
       "2      chr1      69897   T   C            rs200676709                OR4F5   \n",
       "3      chr1     924533   A   G            rs112703963  LOC107985728,SAMD11   \n",
       "4      chr1     942451   T   C              rs6672356               SAMD11   \n",
       "...     ...        ...  ..  ..                    ...                  ...   \n",
       "15198  chrX  154485448   T   C              rs7057286                UBL4A   \n",
       "15199  chrX  154563953   C   T  rs201709278;rs5945206                IKBKG   \n",
       "15200  chrX  154652556   C   A              rs4326559   LOC105373387,CTAG2   \n",
       "15201  chrX  154653499   C   T             rs17855367   LOC105373387,CTAG2   \n",
       "15202  chrX  154766321   G   T              rs2728532                 DKC1   \n",
       "\n",
       "        DP  gnomADg_AF  gnomADg_SAS_AF  gnomADe_AF  gnomADe_SAS_AF  \n",
       "0      104      0.6291          0.8855      0.8380          0.9005  \n",
       "1      205      0.8460          0.9772      0.9497          0.9854  \n",
       "2       93      0.4864          0.6220      0.7209          0.8049  \n",
       "3       81      0.8215          0.9098      0.0000          0.0000  \n",
       "4       98      0.9998          0.9996      0.9999          0.9997  \n",
       "...    ...         ...             ...         ...             ...  \n",
       "15198   67      0.6669          0.5396      0.0000          0.0000  \n",
       "15199   33      0.0000          0.0000      0.0000          0.0000  \n",
       "15200   43      0.4733          0.4243      0.0000          0.0000  \n",
       "15201   86      0.4428          0.3835      0.0000          0.0000  \n",
       "15202   57      0.9924          1.0000      0.9979          1.0000  \n",
       "\n",
       "[15203 rows x 11 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "father_mapped = pd.merge(intersection_data, son, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort=False)\n",
    "father_mapped = father_mapped[['CHROM', 'POS', 'REF', 'ALT', 'rsID_x', 'Gene_Name', 'DP', 'gnomADg_AF', 'gnomADg_SAS_AF', 'gnomADe_AF', 'gnomADe_SAS_AF']]\n",
    "father_mapped.rename(columns={'rsID_x': 'rsID'}, inplace=True)\n",
    "father_mapped.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/latest_trio/Son_intersection_mapped_new.xlsx', index=False)\n",
    "father_mapped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b749cfb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = father_mapped.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "938c5401",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOC107985728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NOC2L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KLHL17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8269</th>\n",
       "      <td>UBL4A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8270</th>\n",
       "      <td>IKBKG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8271</th>\n",
       "      <td>LOC105373387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8272</th>\n",
       "      <td>CTAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8273</th>\n",
       "      <td>DKC1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8274 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 0\n",
       "0            OR4F5\n",
       "1           SAMD11\n",
       "2     LOC107985728\n",
       "3            NOC2L\n",
       "4           KLHL17\n",
       "...            ...\n",
       "8269         UBL4A\n",
       "8270         IKBKG\n",
       "8271  LOC105373387\n",
       "8272         CTAG2\n",
       "8273          DKC1\n",
       "\n",
       "[8274 rows x 1 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes['Gene_Name'] = genes['Gene_Name'].str.split(',')\n",
    "genes = genes.explode('Gene_Name')\n",
    "\n",
    "# Drop rows with null values\n",
    "genes.dropna(subset=['Gene_Name'], inplace=True)\n",
    "\n",
    "# Extract unique gene names\n",
    "unique_genes = pd.DataFrame(genes['Gene_Name'].unique())\n",
    "unique_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "0075eace",
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_genes.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/latest_trio/unique_genes_new.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f9ad9fd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6bc0bc93",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/KAPA_HyperExome_hg38_primary_targets_extended.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGPRX6_final_son.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGPRX6_final_son_covered.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7c9920bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65544</td>\n",
       "      <td>65593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69016</td>\n",
       "      <td>70028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450719</td>\n",
       "      <td>451698</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685695</td>\n",
       "      <td>686674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924411</td>\n",
       "      <td>924968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242132</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25038788</td>\n",
       "      <td>25038934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242133</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25041748</td>\n",
       "      <td>25041906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242134</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25043925</td>\n",
       "      <td>25044043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242135</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25622422</td>\n",
       "      <td>25624085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242136</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25624434</td>\n",
       "      <td>25624547</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>242137 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos\n",
       "0            chr1      65544     65593\n",
       "1            chr1      69016     70028\n",
       "2            chr1     450719    451698\n",
       "3            chr1     685695    686674\n",
       "4            chr1     924411    924968\n",
       "...           ...        ...       ...\n",
       "242132       chrY   25038788  25038934\n",
       "242133       chrY   25041748  25041906\n",
       "242134       chrY   25043925  25044043\n",
       "242135       chrY   25622422  25624085\n",
       "242136       chrY   25624434  25624547\n",
       "\n",
       "[242137 rows x 3 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/KAPA_HyperExome_hg38_primary_targets_extended.bed', sep = '\\t', header = None, error_bad_lines=False)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos']\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ec160210",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>Start_pos</th>\n",
       "      <th>End_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>163897</th>\n",
       "      <td>chr14</td>\n",
       "      <td>57209932</td>\n",
       "      <td>57210081</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chromosome  Start_pos   End_pos\n",
       "163897      chr14   57209932  57210081"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Value to check\n",
    "value = 57209963\n",
    "\n",
    "# Filter rows where the value lies between start and end positions\n",
    "filtered_df = df[(df['Start_pos'] <= value) & (df['End_pos'] >= value)]\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "657b6ca3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b3093c11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12284</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:29:16:16:8:8:50%:1.2236E-3:49:58:7:1:5:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:21:12:12:6:6:50%:6.865E-3:43:37:6:0:6:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14599</td>\n",
       "      <td>rs707680</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:34:15:15:6:9:60%:3.4983E-4:43:34:6:0:9:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14604</td>\n",
       "      <td>rs541940975</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:38:17:17:7:10:58.82%:1.4831E-4:51:35:7:0:10:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14610</td>\n",
       "      <td>rs878986575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:42:18:18:7:11:61.11%:5.2969E-5:59:37:7:0:11:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82471</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209963</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:26:27:27:19:8:29.63%:2.1337E-3:59:64:15:4:7:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82472</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210759</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:39:52:52:40:12:23.08%:1.1939E-4:67:61:36:4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82473</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211544</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=T|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:64:58:58:39:19:32.76%:3.2669E-7:59:59:33:6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82474</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211636</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=63;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:110:63:63:33:30:47.62%:9.545E-12:53:55:27:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82475</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211706</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:39:52:52:40:12:23.08%:1.1939E-4:51:55:32:8...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>82476 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     12284            .   A   C    .   PASS   \n",
       "1      chr1     14590     rs707679   G   A    .   PASS   \n",
       "2      chr1     14599     rs707680   T   A    .   PASS   \n",
       "3      chr1     14604  rs541940975   A   G    .   PASS   \n",
       "4      chr1     14610  rs878986575   T   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "82471  chrY  57209963            .   G   A    .   PASS   \n",
       "82472  chrY  57210759            .   C   G    .   PASS   \n",
       "82473  chrY  57211544            .   C   T    .   PASS   \n",
       "82474  chrY  57211636            .   G   A    .   PASS   \n",
       "82475  chrY  57211706            .   G   C    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "1      ADP=12;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...   \n",
       "2      ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....   \n",
       "3      ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...   \n",
       "4      ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...   \n",
       "...                                                  ...   \n",
       "82471  ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...   \n",
       "82472  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...   \n",
       "82473  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=T|splice_poly...   \n",
       "82474  ADP=63;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...   \n",
       "82475  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "82471  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82472  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82473  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82474  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82475  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0           0/1:29:16:16:8:8:50%:1.2236E-3:49:58:7:1:5:3  \n",
       "1            0/1:21:12:12:6:6:50%:6.865E-3:43:37:6:0:6:0  \n",
       "2           0/1:34:15:15:6:9:60%:3.4983E-4:43:34:6:0:9:0  \n",
       "3      0/1:38:17:17:7:10:58.82%:1.4831E-4:51:35:7:0:10:0  \n",
       "4      0/1:42:18:18:7:11:61.11%:5.2969E-5:59:37:7:0:11:0  \n",
       "...                                                  ...  \n",
       "82471  0/1:26:27:27:19:8:29.63%:2.1337E-3:59:64:15:4:7:1  \n",
       "82472  0/1:39:52:52:40:12:23.08%:1.1939E-4:67:61:36:4...  \n",
       "82473  0/1:64:58:58:39:19:32.76%:3.2669E-7:59:59:33:6...  \n",
       "82474  0/1:110:63:63:33:30:47.62%:9.545E-12:53:55:27:...  \n",
       "82475  0/1:39:52:52:40:12:23.08%:1.1939E-4:51:55:32:8...  \n",
       "\n",
       "[82476 rows x 10 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGPRX5_final_father.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "61e25e4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31721</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155383098</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31722</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010159</td>\n",
       "      <td>rs200413398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31723</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010162</td>\n",
       "      <td>rs150178903</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31724</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31725</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31726 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1     924533  rs112703963   A   G    .   PASS   \n",
       "4      chr1     942451    rs6672356   T   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "31721  chrX  155383098            .   T   C    .   PASS   \n",
       "31722  chrX  156010159  rs200413398   A   G    .   PASS   \n",
       "31723  chrX  156010162  rs150178903   A   G    .   PASS   \n",
       "31724  chrY    3019783    rs9786184   A   C    .   PASS   \n",
       "31725  chrY   21904151     rs162865   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...   \n",
       "1      ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2      ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...   \n",
       "3      ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "4      ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...   \n",
       "...                                                  ...   \n",
       "31721  ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...   \n",
       "31722  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...   \n",
       "31723  ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....   \n",
       "31724  ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....   \n",
       "31725  ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "31721  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31722  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31723  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31724  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31725  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...  \n",
       "1      1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...  \n",
       "2      0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...  \n",
       "3      1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23  \n",
       "4      1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26  \n",
       "...                                                  ...  \n",
       "31721  0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...  \n",
       "31722  0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...  \n",
       "31723  0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...  \n",
       "31724   1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6  \n",
       "31725  0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...  \n",
       "\n",
       "[31726 rows x 10 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1 = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGPRX5_final_father_covered.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf_1.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "abe8de90",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID_x</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL_x</th>\n",
       "      <th>FILTER_x</th>\n",
       "      <th>INFO_x</th>\n",
       "      <th>FORMAT_x</th>\n",
       "      <th>SAMPLE_x</th>\n",
       "      <th>rsID_y</th>\n",
       "      <th>QUAL_y</th>\n",
       "      <th>FILTER_y</th>\n",
       "      <th>INFO_y</th>\n",
       "      <th>FORMAT_y</th>\n",
       "      <th>SAMPLE_y</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31721</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155383098</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31722</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010159</td>\n",
       "      <td>rs200413398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...</td>\n",
       "      <td>rs200413398</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31723</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010162</td>\n",
       "      <td>rs150178903</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...</td>\n",
       "      <td>rs150178903</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31724</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31725</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31726 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS       rsID_x REF ALT QUAL_x FILTER_x  \\\n",
       "0      chr1      69270  rs201219564   A   G      .     PASS   \n",
       "1      chr1      69511    rs2691305   A   G      .     PASS   \n",
       "2      chr1      69897  rs200676709   T   C      .     PASS   \n",
       "3      chr1     924533  rs112703963   A   G      .     PASS   \n",
       "4      chr1     942451    rs6672356   T   C      .     PASS   \n",
       "...     ...        ...          ...  ..  ..    ...      ...   \n",
       "31721  chrX  155383098            .   T   C      .     PASS   \n",
       "31722  chrX  156010159  rs200413398   A   G      .     PASS   \n",
       "31723  chrX  156010162  rs150178903   A   G      .     PASS   \n",
       "31724  chrY    3019783    rs9786184   A   C      .     PASS   \n",
       "31725  chrY   21904151     rs162865   G   A      .     PASS   \n",
       "\n",
       "                                                  INFO_x  \\\n",
       "0      ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...   \n",
       "1      ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2      ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...   \n",
       "3      ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "4      ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...   \n",
       "...                                                  ...   \n",
       "31721  ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...   \n",
       "31722  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...   \n",
       "31723  ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....   \n",
       "31724  ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....   \n",
       "31725  ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...   \n",
       "\n",
       "                                                FORMAT_x  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "31721  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31722  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31723  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31724  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31725  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                SAMPLE_x       rsID_y QUAL_y  \\\n",
       "0      0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...  rs201219564      .   \n",
       "1      1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...    rs2691305      .   \n",
       "2      0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...  rs200676709      .   \n",
       "3      1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23  rs112703963      .   \n",
       "4      1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26    rs6672356      .   \n",
       "...                                                  ...          ...    ...   \n",
       "31721  0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...            .      .   \n",
       "31722  0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...  rs200413398      .   \n",
       "31723  0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...  rs150178903      .   \n",
       "31724   1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6    rs9786184      .   \n",
       "31725  0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...     rs162865      .   \n",
       "\n",
       "      FILTER_y                                             INFO_y  \\\n",
       "0         PASS  ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...   \n",
       "1         PASS  ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2         PASS  ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...   \n",
       "3         PASS  ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "4         PASS  ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...   \n",
       "...        ...                                                ...   \n",
       "31721     PASS  ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...   \n",
       "31722     PASS  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...   \n",
       "31723     PASS  ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....   \n",
       "31724     PASS  ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....   \n",
       "31725     PASS  ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...   \n",
       "\n",
       "                                                FORMAT_y  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "31721  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31722  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31723  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31724  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31725  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                SAMPLE_y _merge  \n",
       "0      0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...   both  \n",
       "1      1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...   both  \n",
       "2      0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...   both  \n",
       "3      1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23   both  \n",
       "4      1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26   both  \n",
       "...                                                  ...    ...  \n",
       "31721  0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...   both  \n",
       "31722  0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...   both  \n",
       "31723  0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...   both  \n",
       "31724   1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6   both  \n",
       "31725  0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...   both  \n",
       "\n",
       "[31726 rows x 17 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_vcf = pd.merge(vcf, vcf_1, on=['CHROM', 'POS', 'REF', 'ALT'], how='inner', indicator=True)\n",
    "merged_vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bf3a40fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID_x</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL_x</th>\n",
       "      <th>FILTER_x</th>\n",
       "      <th>INFO_x</th>\n",
       "      <th>FORMAT_x</th>\n",
       "      <th>SAMPLE_x</th>\n",
       "      <th>rsID_y</th>\n",
       "      <th>QUAL_y</th>\n",
       "      <th>FILTER_y</th>\n",
       "      <th>INFO_y</th>\n",
       "      <th>FORMAT_y</th>\n",
       "      <th>SAMPLE_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12284</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:29:16:16:8:8:50%:1.2236E-3:49:58:7:1:5:3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:21:12:12:6:6:50%:6.865E-3:43:37:6:0:6:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14599</td>\n",
       "      <td>rs707680</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:34:15:15:6:9:60%:3.4983E-4:43:34:6:0:9:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14604</td>\n",
       "      <td>rs541940975</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:38:17:17:7:10:58.82%:1.4831E-4:51:35:7:0:10:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14610</td>\n",
       "      <td>rs878986575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:42:18:18:7:11:61.11%:5.2969E-5:59:37:7:0:11:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82471</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209963</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:26:27:27:19:8:29.63%:2.1337E-3:59:64:15:4:7:1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82472</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210759</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:39:52:52:40:12:23.08%:1.1939E-4:67:61:36:4...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82473</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211544</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=T|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:64:58:58:39:19:32.76%:3.2669E-7:59:59:33:6...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82474</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211636</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=63;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:110:63:63:33:30:47.62%:9.545E-12:53:55:27:...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82475</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211706</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:39:52:52:40:12:23.08%:1.1939E-4:51:55:32:8...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50750 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID_x REF ALT QUAL_x FILTER_x  \\\n",
       "0      chr1     12284            .   A   C      .     PASS   \n",
       "1      chr1     14590     rs707679   G   A      .     PASS   \n",
       "2      chr1     14599     rs707680   T   A      .     PASS   \n",
       "3      chr1     14604  rs541940975   A   G      .     PASS   \n",
       "4      chr1     14610  rs878986575   T   C      .     PASS   \n",
       "...     ...       ...          ...  ..  ..    ...      ...   \n",
       "82471  chrY  57209963            .   G   A      .     PASS   \n",
       "82472  chrY  57210759            .   C   G      .     PASS   \n",
       "82473  chrY  57211544            .   C   T      .     PASS   \n",
       "82474  chrY  57211636            .   G   A      .     PASS   \n",
       "82475  chrY  57211706            .   G   C      .     PASS   \n",
       "\n",
       "                                                  INFO_x  \\\n",
       "0      ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "1      ADP=12;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...   \n",
       "2      ADP=15;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....   \n",
       "3      ADP=17;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...   \n",
       "4      ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...   \n",
       "...                                                  ...   \n",
       "82471  ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=A|non_coding_...   \n",
       "82472  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...   \n",
       "82473  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=T|splice_poly...   \n",
       "82474  ADP=63;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...   \n",
       "82475  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "\n",
       "                                                FORMAT_x  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "82471  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82472  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82473  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82474  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "82475  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                SAMPLE_x rsID_y QUAL_y  \\\n",
       "0           0/1:29:16:16:8:8:50%:1.2236E-3:49:58:7:1:5:3    NaN    NaN   \n",
       "1            0/1:21:12:12:6:6:50%:6.865E-3:43:37:6:0:6:0    NaN    NaN   \n",
       "2           0/1:34:15:15:6:9:60%:3.4983E-4:43:34:6:0:9:0    NaN    NaN   \n",
       "3      0/1:38:17:17:7:10:58.82%:1.4831E-4:51:35:7:0:10:0    NaN    NaN   \n",
       "4      0/1:42:18:18:7:11:61.11%:5.2969E-5:59:37:7:0:11:0    NaN    NaN   \n",
       "...                                                  ...    ...    ...   \n",
       "82471  0/1:26:27:27:19:8:29.63%:2.1337E-3:59:64:15:4:7:1    NaN    NaN   \n",
       "82472  0/1:39:52:52:40:12:23.08%:1.1939E-4:67:61:36:4...    NaN    NaN   \n",
       "82473  0/1:64:58:58:39:19:32.76%:3.2669E-7:59:59:33:6...    NaN    NaN   \n",
       "82474  0/1:110:63:63:33:30:47.62%:9.545E-12:53:55:27:...    NaN    NaN   \n",
       "82475  0/1:39:52:52:40:12:23.08%:1.1939E-4:51:55:32:8...    NaN    NaN   \n",
       "\n",
       "      FILTER_y INFO_y FORMAT_y SAMPLE_y  \n",
       "0          NaN    NaN      NaN      NaN  \n",
       "1          NaN    NaN      NaN      NaN  \n",
       "2          NaN    NaN      NaN      NaN  \n",
       "3          NaN    NaN      NaN      NaN  \n",
       "4          NaN    NaN      NaN      NaN  \n",
       "...        ...    ...      ...      ...  \n",
       "82471      NaN    NaN      NaN      NaN  \n",
       "82472      NaN    NaN      NaN      NaN  \n",
       "82473      NaN    NaN      NaN      NaN  \n",
       "82474      NaN    NaN      NaN      NaN  \n",
       "82475      NaN    NaN      NaN      NaN  \n",
       "\n",
       "[50750 rows x 16 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the two DataFrames based on the specified columns ('CHROM', 'POS', 'REF', 'ALT')\n",
    "merged_vcf = pd.merge(vcf, vcf_1, on=['CHROM', 'POS', 'REF', 'ALT'], how='outer', indicator=True)\n",
    "\n",
    "# Get the non-common rows from the merged DataFrame\n",
    "non_common_rows = merged_vcf[merged_vcf['_merge'] != 'both']\n",
    "\n",
    "# Drop the '_merge' column, which was used for indicating commonality\n",
    "non_common_rows.drop(columns='_merge', inplace=True)\n",
    "non_common_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c95a3439",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "82476"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "50750 + 31726 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "485627f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31681</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155383098</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31682</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010159</td>\n",
       "      <td>rs200413398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31683</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010162</td>\n",
       "      <td>rs150178903</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31684</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31685</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31686 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1     924533  rs112703963   A   G    .   PASS   \n",
       "4      chr1     942451    rs6672356   T   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "31681  chrX  155383098            .   T   C    .   PASS   \n",
       "31682  chrX  156010159  rs200413398   A   G    .   PASS   \n",
       "31683  chrX  156010162  rs150178903   A   G    .   PASS   \n",
       "31684  chrY    3019783    rs9786184   A   C    .   PASS   \n",
       "31685  chrY   21904151     rs162865   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...   \n",
       "1      ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2      ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...   \n",
       "3      ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "4      ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...   \n",
       "...                                                  ...   \n",
       "31681  ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...   \n",
       "31682  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...   \n",
       "31683  ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....   \n",
       "31684  ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....   \n",
       "31685  ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "31681  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31682  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31683  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31684  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31685  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...  \n",
       "1      1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...  \n",
       "2      0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...  \n",
       "3      1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23  \n",
       "4      1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26  \n",
       "...                                                  ...  \n",
       "31681  0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...  \n",
       "31682  0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...  \n",
       "31683  0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...  \n",
       "31684   1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6  \n",
       "31685  0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...  \n",
       "\n",
       "[31686 rows x 10 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_3= pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/matching_positions_father.vcf.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf_3.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "3f62f607",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "40"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "31726 - 31686 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "da6709d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID_x</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL_x</th>\n",
       "      <th>FILTER_x</th>\n",
       "      <th>INFO_x</th>\n",
       "      <th>FORMAT_x</th>\n",
       "      <th>SAMPLE_x</th>\n",
       "      <th>rsID_y</th>\n",
       "      <th>QUAL_y</th>\n",
       "      <th>FILTER_y</th>\n",
       "      <th>INFO_y</th>\n",
       "      <th>FORMAT_y</th>\n",
       "      <th>SAMPLE_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31686</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>661635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:29:15:15:7:8:53.33%:1.0995E-3:58:57:7:0:8:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31687</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>661744</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:24:17:17:10:7:41.18%:3.6151E-3:55:51:7:3:3:4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31688</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>661775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=13;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:21:13:13:7:6:46.15%:7.4534E-3:46:55:5:2:3:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31689</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>1436517</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:55:29:29:14:15:51.72%:2.6068E-6:46:34:14:0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31690</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>1436793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:67:27:27:10:17:62.96%:1.7891E-7:43:39:3:7:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31691</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>1447573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=46;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:80:46:46:24:22:47.83%:8.5409E-9:57:52:23:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31692</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>1447646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=70;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:103:70:70:40:29:41.43%:4.3774E-11:63:60:37...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31693</th>\n",
       "      <td>chr16_KI270728v1_random</td>\n",
       "      <td>1448375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:22:27:27:20:7:25.93%:5.0143E-3:46:54:6:14:4:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31694</th>\n",
       "      <td>chr4_GL000257v2_alt</td>\n",
       "      <td>505927</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:200:35:35:0:35:100%:8.9137E-21:0:62:0:0:30:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31695</th>\n",
       "      <td>chr7_KI270803v1_alt</td>\n",
       "      <td>290188</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=C|intergenic_v...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:46:9:9:0:9:100%:2.0568E-5:0:50:0:0:7:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31696</th>\n",
       "      <td>chr7_KI270803v1_alt</td>\n",
       "      <td>510876</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:39:16:16:6:10:62.5%:1.2413E-4:49:46:6:0:10:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31697</th>\n",
       "      <td>chr7_KI270803v1_alt</td>\n",
       "      <td>510972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=67;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:113:67:67:36:31:46.27%:4.8717E-12:53:51:33...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31698</th>\n",
       "      <td>chr7_KI270803v1_alt</td>\n",
       "      <td>887410</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=51;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:102:51:51:23:27:52.94%:5.635E-11:54:55:18:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31699</th>\n",
       "      <td>chr8_KI270821v1_alt</td>\n",
       "      <td>240886</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:170:30:30:0:30:100%:8.4556E-18:0:48:0:0:21:9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31700</th>\n",
       "      <td>chr12_GL877876v1_alt</td>\n",
       "      <td>196140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:39:24:24:13:11:45.83%:1.1047E-4:64:53:12:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31701</th>\n",
       "      <td>chr16_KI270853v1_alt</td>\n",
       "      <td>565193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:71:35:35:16:19:54.29%:6.3952E-8:53:46:10:6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31702</th>\n",
       "      <td>chr16_KI270853v1_alt</td>\n",
       "      <td>565195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:76:35:35:15:20:57.14%:2.0063E-8:55:42:10:5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31703</th>\n",
       "      <td>chr16_KI270853v1_alt</td>\n",
       "      <td>1978064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=46;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:32:46:46:36:10:21.74%:5.6532E-4:55:51:26:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31704</th>\n",
       "      <td>chr16_KI270853v1_alt</td>\n",
       "      <td>2025467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:30:31:31:22:9:29.03%:9.9376E-4:58:52:18:4:6:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31705</th>\n",
       "      <td>chr16_GL383556v1_alt</td>\n",
       "      <td>47491</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:59:31:31:15:16:51.61%:1.0995E-6:66:66:14:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31706</th>\n",
       "      <td>chr17_KI270861v1_alt</td>\n",
       "      <td>138279</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:37:12:12:3:9:75%:1.6826E-4:49:43:3:0:3:6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31707</th>\n",
       "      <td>chr17_KI270857v1_alt</td>\n",
       "      <td>1781279</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=47;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:80:47:47:25:22:46.81%:9.3895E-9:51:53:20:5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31708</th>\n",
       "      <td>chr17_KI270857v1_alt</td>\n",
       "      <td>1781290</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=47;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:47:47:0:47:100%:6.1512E-28:0:51:0:0:37:10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31709</th>\n",
       "      <td>chr17_KI270857v1_alt</td>\n",
       "      <td>2387450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:55:30:30:15:15:50%:2.9161E-6:50:58:14:1:11:4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31710</th>\n",
       "      <td>chr2_KI270894v1_alt</td>\n",
       "      <td>41636</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=113;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:113:113:83:30:26.55%:1.0094E-10:56:52:6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31711</th>\n",
       "      <td>chr5_KI270897v1_alt</td>\n",
       "      <td>748320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:20:19:19:13:6:31.58%:9.828E-3:53:49:11:2:5:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31712</th>\n",
       "      <td>chr5_KI270897v1_alt</td>\n",
       "      <td>748361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:42:27:27:15:12:44.44%:5.0681E-5:57:58:12:3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31713</th>\n",
       "      <td>chr5_KI270897v1_alt</td>\n",
       "      <td>748445</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:59:31:31:15:16:51.61%:1.0995E-6:64:61:13:2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31714</th>\n",
       "      <td>chr5_KI270897v1_alt</td>\n",
       "      <td>748452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=34;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:49:34:34:20:14:41.18%:1.1295E-5:56:66:14:6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31715</th>\n",
       "      <td>chr12_KI270904v1_alt</td>\n",
       "      <td>196129</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:27:21:21:13:8:38.1%:1.7241E-3:59:47:12:1:5:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31716</th>\n",
       "      <td>chr12_KI270904v1_alt</td>\n",
       "      <td>196142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:27:21:21:13:8:38.1%:1.7241E-3:56:47:11:2:3:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31717</th>\n",
       "      <td>chr12_KI270904v1_alt</td>\n",
       "      <td>196150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:27:24:24:16:8:33.33%:1.949E-3:53:47:13:3:3:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31718</th>\n",
       "      <td>chr19_GL949748v2_alt</td>\n",
       "      <td>1007679</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:64:12:12:0:12:100%:3.698E-7:0:42:0:0:9:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31719</th>\n",
       "      <td>chr19_GL949748v2_alt</td>\n",
       "      <td>1007690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=13;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:70:13:13:0:13:100%:9.6148E-8:0:48:0:0:10:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31720</th>\n",
       "      <td>chr19_GL949748v2_alt</td>\n",
       "      <td>1043030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=12;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:26:12:12:5:7:58.33%:2.2883E-3:51:57:2:3:4:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31721</th>\n",
       "      <td>chr19_GL949749v2_alt</td>\n",
       "      <td>990312</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:30:14:14:6:8:57.14%:9.6618E-4:64:56:5:1:6:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31722</th>\n",
       "      <td>chr19_GL949750v2_alt</td>\n",
       "      <td>964861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_v...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:23:9:9:3:6:66.67%:4.5249E-3:49:61:2:1:5:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31723</th>\n",
       "      <td>chr19_GL949751v2_alt</td>\n",
       "      <td>966712</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=0;HOM=1;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:52:10:10:0:10:100%:5.4125E-6:0:61:0:0:8:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31724</th>\n",
       "      <td>chr19_GL949753v2_alt</td>\n",
       "      <td>694950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=13;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:36:13:13:4:9:69.23%:2.2883E-4:64:39:4:0:6:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31725</th>\n",
       "      <td>chr19_KI270938v1_alt</td>\n",
       "      <td>519384</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:53:19:19:6:13:68.42%:5.0106E-6:55:61:3:3:10:3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         CHROM      POS rsID_x REF ALT QUAL_x FILTER_x INFO_x  \\\n",
       "31686  chr16_KI270728v1_random   661635    NaN   G   C    NaN      NaN    NaN   \n",
       "31687  chr16_KI270728v1_random   661744    NaN   A   C    NaN      NaN    NaN   \n",
       "31688  chr16_KI270728v1_random   661775    NaN   T   C    NaN      NaN    NaN   \n",
       "31689  chr16_KI270728v1_random  1436517    NaN   G   C    NaN      NaN    NaN   \n",
       "31690  chr16_KI270728v1_random  1436793    NaN   G   T    NaN      NaN    NaN   \n",
       "31691  chr16_KI270728v1_random  1447573    NaN   A   C    NaN      NaN    NaN   \n",
       "31692  chr16_KI270728v1_random  1447646    NaN   C   T    NaN      NaN    NaN   \n",
       "31693  chr16_KI270728v1_random  1448375    NaN   G   T    NaN      NaN    NaN   \n",
       "31694      chr4_GL000257v2_alt   505927    NaN   C   T    NaN      NaN    NaN   \n",
       "31695      chr7_KI270803v1_alt   290188    NaN   A   C    NaN      NaN    NaN   \n",
       "31696      chr7_KI270803v1_alt   510876    NaN   G   C    NaN      NaN    NaN   \n",
       "31697      chr7_KI270803v1_alt   510972    NaN   C   T    NaN      NaN    NaN   \n",
       "31698      chr7_KI270803v1_alt   887410    NaN   G   A    NaN      NaN    NaN   \n",
       "31699      chr8_KI270821v1_alt   240886    NaN   A   G    NaN      NaN    NaN   \n",
       "31700     chr12_GL877876v1_alt   196140    NaN   G   T    NaN      NaN    NaN   \n",
       "31701     chr16_KI270853v1_alt   565193    NaN   C   A    NaN      NaN    NaN   \n",
       "31702     chr16_KI270853v1_alt   565195    NaN   C   T    NaN      NaN    NaN   \n",
       "31703     chr16_KI270853v1_alt  1978064    NaN   G   A    NaN      NaN    NaN   \n",
       "31704     chr16_KI270853v1_alt  2025467    NaN   C   T    NaN      NaN    NaN   \n",
       "31705     chr16_GL383556v1_alt    47491    NaN   G   A    NaN      NaN    NaN   \n",
       "31706     chr17_KI270861v1_alt   138279    NaN   A   C    NaN      NaN    NaN   \n",
       "31707     chr17_KI270857v1_alt  1781279    NaN   G   A    NaN      NaN    NaN   \n",
       "31708     chr17_KI270857v1_alt  1781290    NaN   C   G    NaN      NaN    NaN   \n",
       "31709     chr17_KI270857v1_alt  2387450    NaN   T   C    NaN      NaN    NaN   \n",
       "31710      chr2_KI270894v1_alt    41636    NaN   C   T    NaN      NaN    NaN   \n",
       "31711      chr5_KI270897v1_alt   748320    NaN   C   A    NaN      NaN    NaN   \n",
       "31712      chr5_KI270897v1_alt   748361    NaN   G   T    NaN      NaN    NaN   \n",
       "31713      chr5_KI270897v1_alt   748445    NaN   G   A    NaN      NaN    NaN   \n",
       "31714      chr5_KI270897v1_alt   748452    NaN   C   T    NaN      NaN    NaN   \n",
       "31715     chr12_KI270904v1_alt   196129    NaN   G   T    NaN      NaN    NaN   \n",
       "31716     chr12_KI270904v1_alt   196142    NaN   G   T    NaN      NaN    NaN   \n",
       "31717     chr12_KI270904v1_alt   196150    NaN   C   T    NaN      NaN    NaN   \n",
       "31718     chr19_GL949748v2_alt  1007679    NaN   A   G    NaN      NaN    NaN   \n",
       "31719     chr19_GL949748v2_alt  1007690    NaN   T   G    NaN      NaN    NaN   \n",
       "31720     chr19_GL949748v2_alt  1043030    NaN   G   T    NaN      NaN    NaN   \n",
       "31721     chr19_GL949749v2_alt   990312    NaN   G   A    NaN      NaN    NaN   \n",
       "31722     chr19_GL949750v2_alt   964861    NaN   G   A    NaN      NaN    NaN   \n",
       "31723     chr19_GL949751v2_alt   966712    NaN   G   A    NaN      NaN    NaN   \n",
       "31724     chr19_GL949753v2_alt   694950    NaN   G   A    NaN      NaN    NaN   \n",
       "31725     chr19_KI270938v1_alt   519384    NaN   C   T    NaN      NaN    NaN   \n",
       "\n",
       "      FORMAT_x SAMPLE_x rsID_y QUAL_y FILTER_y  \\\n",
       "31686      NaN      NaN      .      .     PASS   \n",
       "31687      NaN      NaN      .      .     PASS   \n",
       "31688      NaN      NaN      .      .     PASS   \n",
       "31689      NaN      NaN      .      .     PASS   \n",
       "31690      NaN      NaN      .      .     PASS   \n",
       "31691      NaN      NaN      .      .     PASS   \n",
       "31692      NaN      NaN      .      .     PASS   \n",
       "31693      NaN      NaN      .      .     PASS   \n",
       "31694      NaN      NaN      .      .     PASS   \n",
       "31695      NaN      NaN      .      .     PASS   \n",
       "31696      NaN      NaN      .      .     PASS   \n",
       "31697      NaN      NaN      .      .     PASS   \n",
       "31698      NaN      NaN      .      .     PASS   \n",
       "31699      NaN      NaN      .      .     PASS   \n",
       "31700      NaN      NaN      .      .     PASS   \n",
       "31701      NaN      NaN      .      .     PASS   \n",
       "31702      NaN      NaN      .      .     PASS   \n",
       "31703      NaN      NaN      .      .     PASS   \n",
       "31704      NaN      NaN      .      .     PASS   \n",
       "31705      NaN      NaN      .      .     PASS   \n",
       "31706      NaN      NaN      .      .     PASS   \n",
       "31707      NaN      NaN      .      .     PASS   \n",
       "31708      NaN      NaN      .      .     PASS   \n",
       "31709      NaN      NaN      .      .     PASS   \n",
       "31710      NaN      NaN      .      .     PASS   \n",
       "31711      NaN      NaN      .      .     PASS   \n",
       "31712      NaN      NaN      .      .     PASS   \n",
       "31713      NaN      NaN      .      .     PASS   \n",
       "31714      NaN      NaN      .      .     PASS   \n",
       "31715      NaN      NaN      .      .     PASS   \n",
       "31716      NaN      NaN      .      .     PASS   \n",
       "31717      NaN      NaN      .      .     PASS   \n",
       "31718      NaN      NaN      .      .     PASS   \n",
       "31719      NaN      NaN      .      .     PASS   \n",
       "31720      NaN      NaN      .      .     PASS   \n",
       "31721      NaN      NaN      .      .     PASS   \n",
       "31722      NaN      NaN      .      .     PASS   \n",
       "31723      NaN      NaN      .      .     PASS   \n",
       "31724      NaN      NaN      .      .     PASS   \n",
       "31725      NaN      NaN      .      .     PASS   \n",
       "\n",
       "                                                  INFO_y  \\\n",
       "31686  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...   \n",
       "31687  ADP=17;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...   \n",
       "31688  ADP=13;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...   \n",
       "31689  ADP=29;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...   \n",
       "31690  ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31691  ADP=46;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...   \n",
       "31692  ADP=70;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31693  ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31694  ADP=35;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intergenic_...   \n",
       "31695  ADP=9;WT=0;HET=0;HOM=1;NC=0;CSQ=C|intergenic_v...   \n",
       "31696  ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...   \n",
       "31697  ADP=67;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31698  ADP=51;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31699  ADP=30;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...   \n",
       "31700  ADP=24;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31701  ADP=35;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31702  ADP=35;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31703  ADP=46;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31704  ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31705  ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31706  ADP=12;WT=0;HET=0;HOM=1;NC=0;CSQ=C|intergenic_...   \n",
       "31707  ADP=47;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31708  ADP=47;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...   \n",
       "31709  ADP=30;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intergenic_...   \n",
       "31710  ADP=113;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic...   \n",
       "31711  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31712  ADP=27;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31713  ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31714  ADP=34;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31715  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31716  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31717  ADP=24;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31718  ADP=12;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...   \n",
       "31719  ADP=13;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intergenic_...   \n",
       "31720  ADP=12;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "31721  ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31722  ADP=9;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_v...   \n",
       "31723  ADP=10;WT=0;HET=0;HOM=1;NC=0;CSQ=A|intergenic_...   \n",
       "31724  ADP=13;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intergenic_...   \n",
       "31725  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "\n",
       "                                                FORMAT_y  \\\n",
       "31686  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31687  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31688  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31689  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31690  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31691  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31692  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31693  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31694  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31695  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31696  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31697  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31698  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31699  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31700  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31701  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31702  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31703  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31704  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31705  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31706  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31707  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31708  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31709  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31710  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31711  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31712  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31713  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31714  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31715  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31716  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31717  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31718  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31719  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31720  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31721  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31722  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31723  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31724  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31725  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                SAMPLE_y  \n",
       "31686    0/1:29:15:15:7:8:53.33%:1.0995E-3:58:57:7:0:8:0  \n",
       "31687   0/1:24:17:17:10:7:41.18%:3.6151E-3:55:51:7:3:3:4  \n",
       "31688    0/1:21:13:13:7:6:46.15%:7.4534E-3:46:55:5:2:3:3  \n",
       "31689  0/1:55:29:29:14:15:51.72%:2.6068E-6:46:34:14:0...  \n",
       "31690  0/1:67:27:27:10:17:62.96%:1.7891E-7:43:39:3:7:...  \n",
       "31691  0/1:80:46:46:24:22:47.83%:8.5409E-9:57:52:23:1...  \n",
       "31692  0/1:103:70:70:40:29:41.43%:4.3774E-11:63:60:37...  \n",
       "31693  0/1:22:27:27:20:7:25.93%:5.0143E-3:46:54:6:14:4:3  \n",
       "31694   1/1:200:35:35:0:35:100%:8.9137E-21:0:62:0:0:30:5  \n",
       "31695         1/1:46:9:9:0:9:100%:2.0568E-5:0:50:0:0:7:2  \n",
       "31696   0/1:39:16:16:6:10:62.5%:1.2413E-4:49:46:6:0:10:0  \n",
       "31697  0/1:113:67:67:36:31:46.27%:4.8717E-12:53:51:33...  \n",
       "31698  0/1:102:51:51:23:27:52.94%:5.635E-11:54:55:18:...  \n",
       "31699   1/1:170:30:30:0:30:100%:8.4556E-18:0:48:0:0:21:9  \n",
       "31700  0/1:39:24:24:13:11:45.83%:1.1047E-4:64:53:12:1...  \n",
       "31701  0/1:71:35:35:16:19:54.29%:6.3952E-8:53:46:10:6...  \n",
       "31702  0/1:76:35:35:15:20:57.14%:2.0063E-8:55:42:10:5...  \n",
       "31703  0/1:32:46:46:36:10:21.74%:5.6532E-4:55:51:26:1...  \n",
       "31704  0/1:30:31:31:22:9:29.03%:9.9376E-4:58:52:18:4:6:3  \n",
       "31705  0/1:59:31:31:15:16:51.61%:1.0995E-6:66:66:14:1...  \n",
       "31706       1/1:37:12:12:3:9:75%:1.6826E-4:49:43:3:0:3:6  \n",
       "31707  0/1:80:47:47:25:22:46.81%:9.3895E-9:51:53:20:5...  \n",
       "31708  1/1:255:47:47:0:47:100%:6.1512E-28:0:51:0:0:37:10  \n",
       "31709   0/1:55:30:30:15:15:50%:2.9161E-6:50:58:14:1:11:4  \n",
       "31710  0/1:99:113:113:83:30:26.55%:1.0094E-10:56:52:6...  \n",
       "31711   0/1:20:19:19:13:6:31.58%:9.828E-3:53:49:11:2:5:1  \n",
       "31712  0/1:42:27:27:15:12:44.44%:5.0681E-5:57:58:12:3...  \n",
       "31713  0/1:59:31:31:15:16:51.61%:1.0995E-6:64:61:13:2...  \n",
       "31714  0/1:49:34:34:20:14:41.18%:1.1295E-5:56:66:14:6...  \n",
       "31715   0/1:27:21:21:13:8:38.1%:1.7241E-3:59:47:12:1:5:3  \n",
       "31716   0/1:27:21:21:13:8:38.1%:1.7241E-3:56:47:11:2:3:5  \n",
       "31717   0/1:27:24:24:16:8:33.33%:1.949E-3:53:47:13:3:3:5  \n",
       "31718       1/1:64:12:12:0:12:100%:3.698E-7:0:42:0:0:9:3  \n",
       "31719     1/1:70:13:13:0:13:100%:9.6148E-8:0:48:0:0:10:3  \n",
       "31720    0/1:26:12:12:5:7:58.33%:2.2883E-3:51:57:2:3:4:3  \n",
       "31721    0/1:30:14:14:6:8:57.14%:9.6618E-4:64:56:5:1:6:2  \n",
       "31722      0/1:23:9:9:3:6:66.67%:4.5249E-3:49:61:2:1:5:1  \n",
       "31723      1/1:52:10:10:0:10:100%:5.4125E-6:0:61:0:0:8:2  \n",
       "31724    0/1:36:13:13:4:9:69.23%:2.2883E-4:64:39:4:0:6:3  \n",
       "31725  0/1:53:19:19:6:13:68.42%:5.0106E-6:55:61:3:3:10:3  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the two DataFrames based on the specified columns ('CHROM', 'POS', 'REF', 'ALT')\n",
    "merged_vcf = pd.merge(vcf_3, vcf_1, on=['CHROM', 'POS', 'REF', 'ALT'], how='outer', indicator=True)\n",
    "\n",
    "# Get the non-common rows from the merged DataFrame\n",
    "non_common_rows = merged_vcf[merged_vcf['_merge'] != 'both']\n",
    "\n",
    "# Drop the '_merge' column, which was used for indicating commonality\n",
    "non_common_rows.drop(columns='_merge', inplace=True)\n",
    "non_common_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e1d016d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "82ecabc8",
   "metadata": {},
   "source": [
    "# Covered Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5779acad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31681</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155383098</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31682</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010159</td>\n",
       "      <td>rs200413398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31683</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156010162</td>\n",
       "      <td>rs150178903</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31684</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31685</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31686 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1     924533  rs112703963   A   G    .   PASS   \n",
       "4      chr1     942451    rs6672356   T   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "31681  chrX  155383098            .   T   C    .   PASS   \n",
       "31682  chrX  156010159  rs200413398   A   G    .   PASS   \n",
       "31683  chrX  156010162  rs150178903   A   G    .   PASS   \n",
       "31684  chrY    3019783    rs9786184   A   C    .   PASS   \n",
       "31685  chrY   21904151     rs162865   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=140;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...   \n",
       "1      ADP=147;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2      ADP=137;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0...   \n",
       "3      ADP=93;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "4      ADP=80;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...   \n",
       "...                                                  ...   \n",
       "31681  ADP=87;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...   \n",
       "31682  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.858,0.1...   \n",
       "31683  ADP=33;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8956,0....   \n",
       "31684  ADP=39;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....   \n",
       "31685  ADP=78;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "31681  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31682  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31683  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31684  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "31685  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      0/1:181:140:140:88:52:37.14%:6.3743E-19:59:49:...  \n",
       "1      1/1:255:147:147:1:146:99.32%:1.0001E-85:37:55:...  \n",
       "2      0/1:127:137:137:99:38:27.74%:1.8385E-13:56:50:...  \n",
       "3      1/1:255:93:93:0:93:100%:1.7451E-55:0:54:0:0:70:23  \n",
       "4      1/1:255:80:80:0:80:100%:1.0864E-47:0:51:0:0:54:26  \n",
       "...                                                  ...  \n",
       "31681  0/1:95:87:87:59:28:32.18%:2.7793E-10:52:61:46:...  \n",
       "31682  0/1:85:33:32:11:21:65.62%:3.1387E-9:56:52:8:3:...  \n",
       "31683  0/1:78:34:33:13:20:60.61%:1.4096E-8:53:54:9:4:...  \n",
       "31684   1/1:224:39:39:0:39:100%:3.6742E-23:0:63:0:0:33:6  \n",
       "31685  0/1:66:78:78:58:20:25.64%:2.3595E-7:58:49:42:1...  \n",
       "\n",
       "[31686 rows x 10 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "father = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Covered_father_new.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "father.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "father"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "104bca90",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:112:44:44:16:28:63.64%:5.6997E-12:66:47:15...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=133;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:133:133:0:133:100%:1.7256E-79:0:55:0:0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69761</td>\n",
       "      <td>rs200505207</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=66;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9633,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:60:66:66:48:18:27.27%:9.974E-7:57:61:35:13...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=46;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:59:46:46:29:17:36.96%:1.2413E-6:62:59:22:7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=40;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:214:40:40:1:39:97.5%:3.8137E-22:62:56:1:0:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29803</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653251</td>\n",
       "      <td>rs17328091</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4906,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:51:41:41:26:15:36.59%:6.364E-6:44:52:26:0:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29804</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=72;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5428,.,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:78:72:72:49:23:31.94%:1.4681E-8:61:46:37:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29805</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.007417,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:248:44:44:0:43:97.73%:1.5066E-25:0:55:0:0:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29806</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154792236</td>\n",
       "      <td>rs782318569</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=47;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9995,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:120:47:47:17:30:63.83%:8.4848E-13:47:48:13...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29807</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155997799</td>\n",
       "      <td>rs6642401</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:42:18:18:7:11:61.11%:5.2969E-5:52:56:5:2:7:4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29808 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69761  rs200505207   A   T    .   PASS   \n",
       "3      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "4      chr1     924533  rs112703963   A   G    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "29803  chrX  154653251   rs17328091   C   G    .   PASS   \n",
       "29804  chrX  154653499   rs17855367   C   T    .   PASS   \n",
       "29805  chrX  154766321    rs2728532   G   T    .   PASS   \n",
       "29806  chrX  154792236  rs782318569   C   T    .   PASS   \n",
       "29807  chrX  155997799    rs6642401   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=44;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "1      ADP=133;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2      ADP=66;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9633,0....   \n",
       "3      ADP=46;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....   \n",
       "4      ADP=40;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "...                                                  ...   \n",
       "29803  ADP=41;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4906,0....   \n",
       "29804  ADP=72;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5428,.,...   \n",
       "29805  ADP=44;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.007417,...   \n",
       "29806  ADP=47;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9995,0....   \n",
       "29807  ADP=18;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "29803  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "29804  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "29805  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "29806  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "29807  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      0/1:112:44:44:16:28:63.64%:5.6997E-12:66:47:15...  \n",
       "1      1/1:255:133:133:0:133:100%:1.7256E-79:0:55:0:0...  \n",
       "2      0/1:60:66:66:48:18:27.27%:9.974E-7:57:61:35:13...  \n",
       "3      0/1:59:46:46:29:17:36.96%:1.2413E-6:62:59:22:7...  \n",
       "4      1/1:214:40:40:1:39:97.5%:3.8137E-22:62:56:1:0:...  \n",
       "...                                                  ...  \n",
       "29803  0/1:51:41:41:26:15:36.59%:6.364E-6:44:52:26:0:...  \n",
       "29804  0/1:78:72:72:49:23:31.94%:1.4681E-8:61:46:37:1...  \n",
       "29805  1/1:248:44:44:0:43:97.73%:1.5066E-25:0:55:0:0:...  \n",
       "29806  0/1:120:47:47:17:30:63.83%:8.4848E-13:47:48:13...  \n",
       "29807   0/1:42:18:18:7:11:61.11%:5.2969E-5:52:56:5:2:7:4  \n",
       "\n",
       "[29808 rows x 10 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mother = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Covered_mother_new.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "mother.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "mother"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "69e4224e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=104;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:104:104:30:74:71.15%:3.2093E-32:58:49:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=205;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:205:205:0:205:100%:9.6033E-123:0:54:0:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:93:93:28:65:69.89%:4.0051E-28:53:51:15...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=81;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:81:81:0:80:98.77%:1.0864E-47:0:52:0:0:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:98:98:0:98:100%:1.7493E-58:0:48:0:0:66:32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32709</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6458,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:158:28:28:0:28:100%:1.3074E-16:0:54:0:0:25:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32710</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4079,0.5...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:23:9:9:3:6:66.67%:4.5249E-3:61:67:3:0:6:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32711</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=42;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:242:42:42:0:42:100%:5.9562E-25:0:54:0:0:31:11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32712</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12914512</td>\n",
       "      <td>rs2032624</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7575,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:18:18:0:18:100%:1.1019E-10:0:56:0:0:15:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32713</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:35:52:52:41:11:21.15%:2.7081E-4:53:43:28:1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32714 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1     924533  rs112703963   A   G    .   PASS   \n",
       "4      chr1     942451    rs6672356   T   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "32709  chrX  155277884     rs559165   G   T    .   PASS   \n",
       "32710  chrX  156003433    rs2037999   T   C    .   PASS   \n",
       "32711  chrY    3019783    rs9786184   A   C    .   PASS   \n",
       "32712  chrY   12914512    rs2032624   C   A    .   PASS   \n",
       "32713  chrY   21904151     rs162865   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=104;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...   \n",
       "1      ADP=205;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2      ADP=93;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....   \n",
       "3      ADP=81;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "4      ADP=98;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...   \n",
       "...                                                  ...   \n",
       "32709  ADP=28;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.6458,0....   \n",
       "32710  ADP=9;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4079,0.5...   \n",
       "32711  ADP=42;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1744,0....   \n",
       "32712  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.7575,0....   \n",
       "32713  ADP=52;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=RBMY...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "32709  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "32710  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "32711  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "32712  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "32713  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      0/1:255:104:104:30:74:71.15%:3.2093E-32:58:49:...  \n",
       "1      1/1:255:205:205:0:205:100%:9.6033E-123:0:54:0:...  \n",
       "2      0/1:255:93:93:28:65:69.89%:4.0051E-28:53:51:15...  \n",
       "3      1/1:255:81:81:0:80:98.77%:1.0864E-47:0:52:0:0:...  \n",
       "4      1/1:255:98:98:0:98:100%:1.7493E-58:0:48:0:0:66:32  \n",
       "...                                                  ...  \n",
       "32709   1/1:158:28:28:0:28:100%:1.3074E-16:0:54:0:0:25:3  \n",
       "32710      0/1:23:9:9:3:6:66.67%:4.5249E-3:61:67:3:0:6:0  \n",
       "32711  1/1:242:42:42:0:42:100%:5.9562E-25:0:54:0:0:31:11  \n",
       "32712    1/1:99:18:18:0:18:100%:1.1019E-10:0:56:0:0:15:3  \n",
       "32713  0/1:35:52:52:41:11:21.15%:2.7081E-4:53:43:28:1...  \n",
       "\n",
       "[32714 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "son = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Covered_son_new.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "son.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "son"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "6ce97790",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22170</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154556836</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22171</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22172</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>rs4326559</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22173</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22174</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22175 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS                   rsID REF ALT\n",
       "0      chr1      69270            rs201219564   A   G\n",
       "1      chr1      69511              rs2691305   A   G\n",
       "2      chr1      69897            rs200676709   T   C\n",
       "3      chr1     924533            rs112703963   A   G\n",
       "4      chr1     942451              rs6672356   T   C\n",
       "...     ...        ...                    ...  ..  ..\n",
       "22170  chrX  154556836                      .   C   T\n",
       "22171  chrX  154563953  rs201709278;rs5945206   C   T\n",
       "22172  chrX  154652556              rs4326559   C   A\n",
       "22173  chrX  154653499             rs17855367   C   T\n",
       "22174  chrX  154766321              rs2728532   G   T\n",
       "\n",
       "[22175 rows x 5 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mother_son = pd.merge(mother, son, on = ['CHROM', 'POS', 'rsID', 'REF', 'ALT'], how = 'inner', sort=False)\n",
    "mother_son = mother_son[['CHROM', 'POS', 'rsID', 'REF', 'ALT']]\n",
    "mother_son"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "3618a63b",
   "metadata": {},
   "outputs": [],
   "source": [
    "father_son.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Mother_Son_common_variants.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5becd435",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953778</td>\n",
       "      <td>rs13303056</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953779</td>\n",
       "      <td>rs13302945</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1020239</td>\n",
       "      <td>rs201073369</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1022260</td>\n",
       "      <td>rs6657048</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1047342</td>\n",
       "      <td>rs142416636</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32709</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32710</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32711</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3019783</td>\n",
       "      <td>rs9786184</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32712</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12914512</td>\n",
       "      <td>rs2032624</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32713</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21904151</td>\n",
       "      <td>rs162865</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10539 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT\n",
       "10     chr1     953778   rs13303056   G   C\n",
       "11     chr1     953779   rs13302945   A   C\n",
       "29     chr1    1020239  rs201073369   G   C\n",
       "30     chr1    1022260    rs6657048   C   T\n",
       "33     chr1    1047342  rs142416636   A   G\n",
       "...     ...        ...          ...  ..  ..\n",
       "32709  chrX  155277884     rs559165   G   T\n",
       "32710  chrX  156003433    rs2037999   T   C\n",
       "32711  chrY    3019783    rs9786184   A   C\n",
       "32712  chrY   12914512    rs2032624   C   A\n",
       "32713  chrY   21904151     rs162865   G   A\n",
       "\n",
       "[10539 rows x 5 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the two DataFrames on the common column with an indicator\n",
    "merged = son.merge(mother_son, on=['CHROM', 'POS', 'rsID', 'REF', 'ALT'], how='left', indicator=True)\n",
    "\n",
    "# Filter rows where the indicator column is set to 'left_only'\n",
    "non_common_rows_df1 = merged[merged['_merge'] == 'left_only']\n",
    "\n",
    "# Drop the indicator column if you don't need it\n",
    "non_common_rows_df1 = non_common_rows_df1.drop(columns=['_merge'])\n",
    "non_common_rows_df1 = non_common_rows_df1[['CHROM', 'POS', 'rsID', 'REF', 'ALT']]\n",
    "non_common_rows_df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "41f79603",
   "metadata": {},
   "outputs": [],
   "source": [
    "non_common_rows_df1.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Son_only_variants_(M&S_analysis).xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94ab8cc8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d4a296e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5081509",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "db3a80ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:827209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.314C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894115</td>\n",
       "      <td>7952</td>\n",
       "      <td>7110</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:827212</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.311G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894327</td>\n",
       "      <td>7826</td>\n",
       "      <td>6999</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481215</th>\n",
       "      <td>X:154641461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154641461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481219</th>\n",
       "      <td>X:154773313</td>\n",
       "      <td>A/T</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>1736</td>\n",
       "      <td>NM_001363.5</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001363.5:c.1155+64A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>PM2,BP7</td>\n",
       "      <td>DKC1:Variant is missing from all sub populatio...</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154773313</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481223</th>\n",
       "      <td>X:154785208</td>\n",
       "      <td>-/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.678-51_678-50insG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.0492715</td>\n",
       "      <td>3775</td>\n",
       "      <td>186</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154785208</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481224</th>\n",
       "      <td>X:154785208</td>\n",
       "      <td>A/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.678-51T&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.132715</td>\n",
       "      <td>3775</td>\n",
       "      <td>501</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154785208</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481239</th>\n",
       "      <td>Y:11153779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11153779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194203 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Chr:Pos Ref/Alt  Gene Name     Gene Inheritance Entrez Gene ID  \\\n",
       "0           1:69511     A/G      OR4F5  Default (Recessive)          79501   \n",
       "1          1:817514     T/C     FAM87B  Default (Recessive)         400728   \n",
       "2          1:826893     G/A  LINC00115  Default (Recessive)          79854   \n",
       "3          1:827209     G/C  LINC00115  Default (Recessive)          79854   \n",
       "4          1:827212     C/G  LINC00115  Default (Recessive)          79854   \n",
       "...             ...     ...        ...                  ...            ...   \n",
       "481215  X:154641461     NaN        NaN                  NaN            NaN   \n",
       "481219  X:154773313     A/T       DKC1            Recessive           1736   \n",
       "481223  X:154785208     -/C       MPP1  Default (Recessive)           4354   \n",
       "481224  X:154785208     A/C       MPP1  Default (Recessive)           4354   \n",
       "481239   Y:11153779     NaN        NaN                  NaN            NaN   \n",
       "\n",
       "       Transcript Name Transcript Strand                        HGVS cDot  \\\n",
       "0       NM_001005484.2                 +          NM_001005484.2:c.484A>G   \n",
       "1          NR_103536.1                 +             NR_103536.1:n.144T>C   \n",
       "2          NR_024321.1                 -             NR_024321.1:n.630C>T   \n",
       "3          NR_024321.1                 -             NR_024321.1:n.314C>G   \n",
       "4          NR_024321.1                 -             NR_024321.1:n.311G>C   \n",
       "...                ...               ...                              ...   \n",
       "481215             NaN               NaN                              NaN   \n",
       "481219     NM_001363.5                 +         NM_001363.5:c.1155+64A>T   \n",
       "481223     NM_002436.4                 -  NM_002436.4:c.678-51_678-50insG   \n",
       "481224     NM_002436.4                 -          NM_002436.4:c.678-51T>G   \n",
       "481239             NaN               NaN                              NaN   \n",
       "\n",
       "                         HGVS pDot        Sequence Ontology  \\\n",
       "0       NP_001005484.2:p.Thr162Ala         missense_variant   \n",
       "1                              NaN  non_coding_exon_variant   \n",
       "2                              NaN  non_coding_exon_variant   \n",
       "3                              NaN  non_coding_exon_variant   \n",
       "4                              NaN  non_coding_exon_variant   \n",
       "...                            ...                      ...   \n",
       "481215                         NaN                      NaN   \n",
       "481219                         NaN           intron_variant   \n",
       "481223                         NaN           intron_variant   \n",
       "481224                         NaN           intron_variant   \n",
       "481239                         NaN                      NaN   \n",
       "\n",
       "                  Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                   Single Transcript Gene                      BA1,BP4   \n",
       "1                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "3                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "4                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                                    ...                          ...   \n",
       "481215                                 NaN                          NaN   \n",
       "481219  Other Transcripts Have Same Effect                      PM2,BP7   \n",
       "481223  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "481224  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "481239                                 NaN                          NaN   \n",
       "\n",
       "                 ACMG Classification Criteria Description   Classification  \\\n",
       "0       OR4F5:Allele frequency is above 0.01 recessive...           Benign   \n",
       "1       FAM87B:Allele frequency is above 0.01 recessiv...           Benign   \n",
       "2       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "3       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "4       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "...                                                   ...              ...   \n",
       "481215                                                NaN              NaN   \n",
       "481219  DKC1:Variant is missing from all sub populatio...  VUS/Conflicting   \n",
       "481223  MPP1:Allele frequency is above 0.01 recessive ...           Benign   \n",
       "481224  MPP1:Allele frequency is above 0.01 recessive ...           Benign   \n",
       "481239                                                NaN              NaN   \n",
       "\n",
       "       Auto Classification Previous Classification Count  \\\n",
       "0                   Benign                           NaN   \n",
       "1                   Benign                           NaN   \n",
       "2                   Benign                           NaN   \n",
       "3                   Benign                           NaN   \n",
       "4                   Benign                           NaN   \n",
       "...                    ...                           ...   \n",
       "481215                 NaN                           NaN   \n",
       "481219     VUS/Conflicting                           NaN   \n",
       "481223              Benign                           NaN   \n",
       "481224              Benign                           NaN   \n",
       "481239                 NaN                           NaN   \n",
       "\n",
       "       Previous Classification Last Classification Date  \\\n",
       "0                          NaN                      NaN   \n",
       "1                          NaN                      NaN   \n",
       "2                          NaN                      NaN   \n",
       "3                          NaN                      NaN   \n",
       "4                          NaN                      NaN   \n",
       "...                        ...                      ...   \n",
       "481215                     NaN                      NaN   \n",
       "481219                     NaN                      NaN   \n",
       "481223                     NaN                      NaN   \n",
       "481224                     NaN                      NaN   \n",
       "481239                     NaN                      NaN   \n",
       "\n",
       "       Max Sub Population Freq Group Name    Max MAF Max Allele Number  \\\n",
       "0             Annotated gnomAD East Asian   0.999461             16698   \n",
       "1                       Annotated 1kG All   0.753195              5008   \n",
       "2             Annotated gnomAD East Asian   0.896992             10106   \n",
       "3             Annotated gnomAD East Asian   0.894115              7952   \n",
       "4             Annotated gnomAD East Asian   0.894327              7826   \n",
       "...                                   ...        ...               ...   \n",
       "481215                                NaN        NaN               NaN   \n",
       "481219                                NaN        NaN               NaN   \n",
       "481223                  Annotated 1kG All  0.0492715              3775   \n",
       "481224                  Annotated 1kG All   0.132715              3775   \n",
       "481239                                NaN        NaN               NaN   \n",
       "\n",
       "       Max Allele Count CHROM        POS  REF  ALT  \n",
       "0                 16689  chr1      69511    A    G  \n",
       "1                  3772  chr1     817514    T    C  \n",
       "2                  9065  chr1     826893    G    A  \n",
       "3                  7110  chr1     827209    G    C  \n",
       "4                  6999  chr1     827212    C    G  \n",
       "...                 ...   ...        ...  ...  ...  \n",
       "481215              NaN  chrX  154641461  NaN  NaN  \n",
       "481219              NaN  chrX  154773313    A    T  \n",
       "481223              186  chrX  154785208    -    C  \n",
       "481224              501  chrX  154785208    A    C  \n",
       "481239              NaN  chrY   11153779  NaN  NaN  \n",
       "\n",
       "[194203 rows x 26 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_new_05_10_2023/All_5samples.xlsx')\n",
    "df['CHROM'] = 'chr' + df['Chr:Pos'].str.split(':').str[0]\n",
    "df['POS'] = df['Chr:Pos'].str.split(':').str[1].astype(int)\n",
    "df['REF'] = df['Ref/Alt'].str.split('/').str[0]\n",
    "df['ALT'] = df['Ref/Alt'].str.split('/').str[1]\n",
    "columns_to_check = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "# Use drop_duplicates to remove duplicated rows based on the specified columns\n",
    "df = df.drop_duplicates(subset=columns_to_check)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "deb4f03f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22170</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154556836</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22171</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22172</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154652556</td>\n",
       "      <td>rs4326559</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>X:154652556</td>\n",
       "      <td>C/A</td>\n",
       "      <td>CTAG2</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>30848</td>\n",
       "      <td>NM_172377.5</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_172377.5:c.345G&gt;T</td>\n",
       "      <td>NP_758965.2:p.Pro115=</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP4,BP7,BP6</td>\n",
       "      <td>CTAG2:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.799206</td>\n",
       "      <td>12844</td>\n",
       "      <td>10265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22173</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154653499</td>\n",
       "      <td>rs17855367</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22174</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154766321</td>\n",
       "      <td>rs2728532</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>X:154766321</td>\n",
       "      <td>G/T</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>1736</td>\n",
       "      <td>NM_001363.5</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001363.5:c.369G&gt;T</td>\n",
       "      <td>NP_001354.1:p.Thr123=</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP4,BP7,BP6</td>\n",
       "      <td>DKC1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Annotated gnomAD Ashkenazi Jewish</td>\n",
       "      <td>1</td>\n",
       "      <td>7293</td>\n",
       "      <td>7293</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22175 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS                   rsID REF ALT      Chr:Pos Ref/Alt  \\\n",
       "0      chr1      69270            rs201219564   A   G            -       -   \n",
       "1      chr1      69511              rs2691305   A   G      1:69511     A/G   \n",
       "2      chr1      69897            rs200676709   T   C            -       -   \n",
       "3      chr1     924533            rs112703963   A   G            -       -   \n",
       "4      chr1     942451              rs6672356   T   C            -       -   \n",
       "...     ...        ...                    ...  ..  ..          ...     ...   \n",
       "22170  chrX  154556836                      .   C   T            -       -   \n",
       "22171  chrX  154563953  rs201709278;rs5945206   C   T            -       -   \n",
       "22172  chrX  154652556              rs4326559   C   A  X:154652556     C/A   \n",
       "22173  chrX  154653499             rs17855367   C   T            -       -   \n",
       "22174  chrX  154766321              rs2728532   G   T  X:154766321     G/T   \n",
       "\n",
       "      Gene Name     Gene Inheritance Entrez Gene ID Transcript Name  \\\n",
       "0             -                    -              -               -   \n",
       "1         OR4F5  Default (Recessive)          79501  NM_001005484.2   \n",
       "2             -                    -              -               -   \n",
       "3             -                    -              -               -   \n",
       "4             -                    -              -               -   \n",
       "...         ...                  ...            ...             ...   \n",
       "22170         -                    -              -               -   \n",
       "22171         -                    -              -               -   \n",
       "22172     CTAG2  Default (Recessive)          30848     NM_172377.5   \n",
       "22173         -                    -              -               -   \n",
       "22174      DKC1            Recessive           1736     NM_001363.5   \n",
       "\n",
       "      Transcript Strand                HGVS cDot                   HGVS pDot  \\\n",
       "0                     -                        -                           -   \n",
       "1                     +  NM_001005484.2:c.484A>G  NP_001005484.2:p.Thr162Ala   \n",
       "2                     -                        -                           -   \n",
       "3                     -                        -                           -   \n",
       "4                     -                        -                           -   \n",
       "...                 ...                      ...                         ...   \n",
       "22170                 -                        -                           -   \n",
       "22171                 -                        -                           -   \n",
       "22172                 -     NM_172377.5:c.345G>T       NP_758965.2:p.Pro115=   \n",
       "22173                 -                        -                           -   \n",
       "22174                 +     NM_001363.5:c.369G>T       NP_001354.1:p.Thr123=   \n",
       "\n",
       "        Sequence Ontology            Other Transcript Effects  \\\n",
       "0                       -                                   -   \n",
       "1        missense_variant              Single Transcript Gene   \n",
       "2                       -                                   -   \n",
       "3                       -                                   -   \n",
       "4                       -                                   -   \n",
       "...                   ...                                 ...   \n",
       "22170                   -                                   -   \n",
       "22171                   -                                   -   \n",
       "22172  synonymous_variant  Other Transcripts Have Same Effect   \n",
       "22173                   -                                   -   \n",
       "22174  synonymous_variant  Other Transcripts Have Same Effect   \n",
       "\n",
       "      ACMG Classification Criteria  \\\n",
       "0                                -   \n",
       "1                          BA1,BP4   \n",
       "2                                -   \n",
       "3                                -   \n",
       "4                                -   \n",
       "...                            ...   \n",
       "22170                            -   \n",
       "22171                            -   \n",
       "22172          BA1,BS2,BP4,BP7,BP6   \n",
       "22173                            -   \n",
       "22174          BA1,BS2,BP4,BP7,BP6   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0                                                      -           None   \n",
       "1      OR4F5:Allele frequency is above 0.01 recessive...         Benign   \n",
       "2                                                      -           None   \n",
       "3                                                      -           None   \n",
       "4                                                      -           None   \n",
       "...                                                  ...            ...   \n",
       "22170                                                  -           None   \n",
       "22171                                                  -           None   \n",
       "22172  CTAG2:Allele frequency is above 0.01 recessive...         Benign   \n",
       "22173                                                  -           None   \n",
       "22174  DKC1:Allele frequency is above 0.01 recessive ...         Benign   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                       -                             -   \n",
       "1                  Benign                             -   \n",
       "2                       -                             -   \n",
       "3                       -                             -   \n",
       "4                       -                             -   \n",
       "...                   ...                           ...   \n",
       "22170                   -                             -   \n",
       "22171                   -                             -   \n",
       "22172              Benign                             -   \n",
       "22173                   -                             -   \n",
       "22174              Benign                             -   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                           -                        -   \n",
       "1                           -                        -   \n",
       "2                           -                        -   \n",
       "3                           -                        -   \n",
       "4                           -                        -   \n",
       "...                       ...                      ...   \n",
       "22170                       -                        -   \n",
       "22171                       -                        -   \n",
       "22172                       -                        -   \n",
       "22173                       -                        -   \n",
       "22174                       -                        -   \n",
       "\n",
       "      Max Sub Population Freq Group Name   Max MAF Max Allele Number  \\\n",
       "0                                      -         -                 -   \n",
       "1            Annotated gnomAD East Asian  0.999461             16698   \n",
       "2                                      -         -                 -   \n",
       "3                                      -         -                 -   \n",
       "4                                      -         -                 -   \n",
       "...                                  ...       ...               ...   \n",
       "22170                                  -         -                 -   \n",
       "22171                                  -         -                 -   \n",
       "22172        Annotated gnomAD East Asian  0.799206             12844   \n",
       "22173                                  -         -                 -   \n",
       "22174  Annotated gnomAD Ashkenazi Jewish         1              7293   \n",
       "\n",
       "      Max Allele Count  \n",
       "0                    -  \n",
       "1                16689  \n",
       "2                    -  \n",
       "3                    -  \n",
       "4                    -  \n",
       "...                ...  \n",
       "22170                -  \n",
       "22171                -  \n",
       "22172            10265  \n",
       "22173                -  \n",
       "22174             7293  \n",
       "\n",
       "[22175 rows x 27 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mother_son['POS'] = mother_son['POS'].astype(int)\n",
    "\n",
    "merged = pd.merge(mother_son, df, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged = merged.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "\n",
    "# To fill NaN values in the 'Classification' column with 'NA', you need to assign the result back to the column.\n",
    "merged['Classification'] = merged['Classification'].fillna('NA')\n",
    "merged= merged.fillna('-')\n",
    "# List of your specified priorities\n",
    "priorities = ['Pathogenic', 'Likely Pathogenic', 'VUS/Weak Pathogenic', 'VUS/Conflicting', 'VUS', 'VUS/Weak Benign', 'Likely Benign', 'Benign']\n",
    "\n",
    "# Function to select the highest ranked value from Original_Column\n",
    "def select_highest_ranked(row):\n",
    "    values = row['Classification'].split(',')\n",
    "    for rank in priorities:\n",
    "        if rank in values:\n",
    "            return rank\n",
    "    return None  # Return None if no rank found\n",
    "\n",
    "# Apply the function to create the final selected rank column\n",
    "merged['Classification'] = merged.apply(select_highest_ranked, axis=1)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "a71afe53",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Mother_Son_ACMG.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "32890eb4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant_classification</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Benign</td>\n",
       "      <td>16252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>5432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VUS/Weak Pathogenic</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>VUS/Weak Benign</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>VUS</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Variant_classification  Counts\n",
       "0                 Benign   16252\n",
       "1                   None    5432\n",
       "2          Likely Benign     212\n",
       "3        VUS/Conflicting     184\n",
       "4    VUS/Weak Pathogenic      86\n",
       "5        VUS/Weak Benign       6\n",
       "6                    VUS       2\n",
       "7      Likely Pathogenic       1"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "value_counts_with_null_bcftools = merged['Classification'].value_counts(dropna=False)\n",
    "#result_df = pd.DataFrame({'Counts': value_counts_with_null_bcftools})\n",
    "result_df = pd.DataFrame({'Variant_classification': value_counts_with_null_bcftools.index, 'Counts': value_counts_with_null_bcftools.values})\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "75414fac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22175"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_df.Counts.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "e0e87766",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Trio_analysis/new_trio_files/Mother_Son_Classification_counts.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3340ec7f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bf4253d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "04db0092",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "file_directory = r'C:/Users/GenepoweRx_Madhu/Downloads/TRIO/'\n",
    "\n",
    "# Define a function to read and process an Excel file\n",
    "def read_excel_file(file_path):\n",
    "    # Construct the full file path\n",
    "    full_file_path = os.path.join(file_directory, file_path)\n",
    "\n",
    "    # Read the Excel file into a Pandas DataFrame, specifying header and usecols\n",
    "    excel_df = pd.read_excel(full_file_path, header=1, usecols=[0, 1])\n",
    "    return excel_df\n",
    "\n",
    "# List of Excel file names (without full paths)\n",
    "excel_files = ['Germline_KHAFAMGPCSP1.xlsx', 'Germline_KHAFAMGPCSP2.xlsx']  # Replace with your Excel file names\n",
    "\n",
    "# Initialize the result DataFrame with the data from the first Excel file\n",
    "result_df = read_excel_file(excel_files[0])\n",
    "\n",
    "# Define the columns to join on\n",
    "join_columns = ['Chr:Pos', 'Ref/Alt']\n",
    "\n",
    "# Iterate through the rest of the Excel files and merge with the result based on the join columns\n",
    "for file_name in excel_files[1:]:\n",
    "    excel_df = read_excel_file(file_name)\n",
    "    result_df = pd.merge(result_df, excel_df, on=join_columns, how='inner')\n",
    "\n",
    "# Specify the full file path for the output Excel file\n",
    "output_excel_path = os.path.join(file_directory, 'intersected_data.xlsx')\n",
    "\n",
    "# Save the result to an Excel file\n",
    "result_df.to_excel(output_excel_path, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c842ab2c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
