{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14653</td>\n",
       "      <td>rs62635297</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:30:14:14:6:8:57.14%:9.6618E-4:55:44:6:0:8:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14937</td>\n",
       "      <td>rs11807796</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:23:24:24:17:7:29.17%:4.7006E-3:53:45:14:3:2:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=13;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:42:13:13:3:10:76.92%:5.3843E-5:37:40:3:0:10:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24786</td>\n",
       "      <td>rs11266852</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=49;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:49:49:24:25:51.02%:4.6489E-10:54:58:19:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24912</td>\n",
       "      <td>rs1382803901</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:53:25:25:11:14:56%:4.7528E-6:56:46:6:5:6:8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56407</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209184</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:26:26:26:18:8:30.77%:2.076E-3:51:49:14:4:7:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56408</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209365</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:20:19:19:13:6:31.58%:9.828E-3:62:42:11:2:1:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56409</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209591</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=G|non_coding_t...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:31:8:8:1:7:87.5%:6.993E-4:74:42:1:0:7:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56410</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=G|non_coding_t...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:41:8:8:0:8:100%:7.77E-5:0:50:0:0:6:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56411</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210759</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:47:28:28:15:13:46.43%:1.9812E-5:59:62:11:4...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>56412 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14653    rs62635297   C   T    .   PASS   \n",
       "1      chr1     14937    rs11807796   T   C    .   PASS   \n",
       "2      chr1     16495     rs3210724   G   C    .   PASS   \n",
       "3      chr1     24786    rs11266852   G   A    .   PASS   \n",
       "4      chr1     24912  rs1382803901   G   A    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "56407  chrY  57209184             .   T   C    .   PASS   \n",
       "56408  chrY  57209365             .   T   C    .   PASS   \n",
       "56409  chrY  57209591             .   A   G    .   PASS   \n",
       "56410  chrY  57209613             .   A   G    .   PASS   \n",
       "56411  chrY  57210759             .   C   G    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=14;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...   \n",
       "1      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...   \n",
       "2      ADP=13;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...   \n",
       "3      ADP=49;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...   \n",
       "4      ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=WASH...   \n",
       "...                                                  ...   \n",
       "56407  ADP=26;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...   \n",
       "56408  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...   \n",
       "56409  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=G|non_coding_t...   \n",
       "56410  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=G|non_coding_t...   \n",
       "56411  ADP=28;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "56407  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "56408  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "56409  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "56410  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "56411  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0        0/1:30:14:14:6:8:57.14%:9.6618E-4:55:44:6:0:8:0  \n",
       "1      0/1:23:24:24:17:7:29.17%:4.7006E-3:53:45:14:3:2:5  \n",
       "2      1/1:42:13:13:3:10:76.92%:5.3843E-5:37:40:3:0:10:0  \n",
       "3      0/1:93:49:49:24:25:51.02%:4.6489E-10:54:58:19:...  \n",
       "4         0/1:53:25:25:11:14:56%:4.7528E-6:56:46:6:5:6:8  \n",
       "...                                                  ...  \n",
       "56407   0/1:26:26:26:18:8:30.77%:2.076E-3:51:49:14:4:7:1  \n",
       "56408   0/1:20:19:19:13:6:31.58%:9.828E-3:62:42:11:2:1:5  \n",
       "56409        1/1:31:8:8:1:7:87.5%:6.993E-4:74:42:1:0:7:0  \n",
       "56410           1/1:41:8:8:0:8:100%:7.77E-5:0:50:0:0:6:2  \n",
       "56411  0/1:47:28:28:15:13:46.43%:1.9812E-5:59:62:11:4...  \n",
       "\n",
       "[56412 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_4/KHCDPRGPTTL1/KHCDPRGPTTL1_final.vcf', sep=\"\\t\", comment=\"#\", header= None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1014228</td>\n",
       "      <td>rs1921</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6611,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:56:38:38:22:16:42.11%:2.0535E-6:53:48:19:3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS    rsID REF ALT QUAL FILTER  \\\n",
       "60  chr1  1014228  rs1921   G   A    .   PASS   \n",
       "\n",
       "                                                 INFO  \\\n",
       "60  ADP=38;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6611,0....   \n",
       "\n",
       "                                               FORMAT  \\\n",
       "60  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                               SAMPLE  \n",
       "60  0/1:56:38:38:22:16:42.11%:2.0535E-6:53:48:19:3...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[vcf['rsID'] == 'rs1921']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>5.609500e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14907</td>\n",
       "      <td>rs6682375</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>76</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>65.52%</td>\n",
       "      <td>2.114400e-08</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>87</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>76.92%</td>\n",
       "      <td>1.827300e-09</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17365</td>\n",
       "      <td>rs369606208</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>46</td>\n",
       "      <td>16</td>\n",
       "      <td>25.81%</td>\n",
       "      <td>5.029900e-06</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47664</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208581</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>60%</td>\n",
       "      <td>2.254800e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47665</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209184</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>23</td>\n",
       "      <td>10</td>\n",
       "      <td>30.3%</td>\n",
       "      <td>4.387200e-04</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47666</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209591</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>9.661800e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47667</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>6.993000e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47669 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS           ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752   rs17855866   G   A    .   PASS  \\\n",
       "1      chr1     14907    rs6682375   A   G    .   PASS   \n",
       "2      chr1     14930    rs6682385   A   G    .   PASS   \n",
       "3      chr1     16495    rs3210724   G   C    .   PASS   \n",
       "4      chr1     17365  rs369606208   C   G    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "47664  chrY  57208581            .   G   C    .   PASS   \n",
       "47665  chrY  57209184            .   T   C    .   PASS   \n",
       "47666  chrY  57209591            .   A   G    .   PASS   \n",
       "47667  chrY  57209613            .   A   G    .   PASS   \n",
       "47668  chrY  57210016            .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP  DP   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   35  35  \\\n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  76   29  29   \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....  1/1  87   26  26   \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1  71   17  17   \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...  0/1  52   62  62   \n",
       "...                                                  ...  ...  ..  ...  ..   \n",
       "47664  ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  46   20  20   \n",
       "47665  ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  33   33  33   \n",
       "47666  ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...  0/1  30   14  14   \n",
       "47667  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...  0/1  29   15  15   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...  1/1  31    8   8   \n",
       "\n",
       "       RD  AD     RDF           RDR  ADF  ADR  FREQ  PVAL  \n",
       "0      28   7     20%  5.609500e-03   28    0     7     0  \n",
       "1      10  19  65.52%  2.114400e-08    5    5    16     3  \n",
       "2       6  20  76.92%  1.827300e-09    4    2    15     5  \n",
       "3       2  15  88.24%  7.327700e-08    2    0    15     0  \n",
       "4      46  16  25.81%  5.029900e-06   28   18    15     1  \n",
       "...    ..  ..     ...           ...  ...  ...   ...   ...  \n",
       "47664   8  12     60%  2.254800e-05    8    0    12     0  \n",
       "47665  23  10   30.3%  4.387200e-04   23    0    10     0  \n",
       "47666   6   8  57.14%  9.661800e-04    4    2     7     1  \n",
       "47667   7   8  53.33%  1.099500e-03    5    2     7     1  \n",
       "47668   1   7   87.5%  6.993000e-04    0    1     6     1  \n",
       "\n",
       "[47669 rows x 20 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/Dell/Downloads/KHHSPTGPTTL34_final_Depth.vcf', comment=\"#\", sep=\"\\t\", header=0, low_memory=False)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT',\n",
       "       'GQ', 'SDP', 'DP', 'RD', 'AD', 'RDF', 'RDR', 'ADF', 'ADR', 'FREQ',\n",
       "       'PVAL'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>Raw Read Depth as reported by SAMtools</th>\n",
       "      <th>Quality Read Depth of bases with Phred score &gt;= 15</th>\n",
       "      <th>Depth of reference-supporting bases (reads1)</th>\n",
       "      <th>Depth of variant-supporting bases (reads2)</th>\n",
       "      <th>Depth of reference-supporting bases on forward strand (reads1plus)</th>\n",
       "      <th>Depth of reference-supporting bases on reverse strand (reads1minus)</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>5.609500e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14907</td>\n",
       "      <td>rs6682375</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>76</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>65.52%</td>\n",
       "      <td>2.114400e-08</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>87</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>76.92%</td>\n",
       "      <td>1.827300e-09</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17365</td>\n",
       "      <td>rs369606208</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>46</td>\n",
       "      <td>16</td>\n",
       "      <td>25.81%</td>\n",
       "      <td>5.029900e-06</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47664</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208581</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>60%</td>\n",
       "      <td>2.254800e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47665</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209184</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>23</td>\n",
       "      <td>10</td>\n",
       "      <td>30.3%</td>\n",
       "      <td>4.387200e-04</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47666</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209591</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>9.661800e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47667</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>6.993000e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47669 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS           ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752   rs17855866   G   A    .   PASS  \\\n",
       "1      chr1     14907    rs6682375   A   G    .   PASS   \n",
       "2      chr1     14930    rs6682385   A   G    .   PASS   \n",
       "3      chr1     16495    rs3210724   G   C    .   PASS   \n",
       "4      chr1     17365  rs369606208   C   G    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "47664  chrY  57208581            .   G   C    .   PASS   \n",
       "47665  chrY  57209184            .   T   C    .   PASS   \n",
       "47666  chrY  57209591            .   A   G    .   PASS   \n",
       "47667  chrY  57209613            .   A   G    .   PASS   \n",
       "47668  chrY  57210016            .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....      1/1   \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...      1/1   \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47664  ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47665  ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47666  ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47667  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  Raw Read Depth as reported by SAMtools   \n",
       "0                    22                                      35  \\\n",
       "1                    76                                      29   \n",
       "2                    87                                      26   \n",
       "3                    71                                      17   \n",
       "4                    52                                      62   \n",
       "...                 ...                                     ...   \n",
       "47664                46                                      20   \n",
       "47665                33                                      33   \n",
       "47666                30                                      14   \n",
       "47667                29                                      15   \n",
       "47668                31                                       8   \n",
       "\n",
       "       Quality Read Depth of bases with Phred score >= 15   \n",
       "0                                                     35   \\\n",
       "1                                                     29    \n",
       "2                                                     26    \n",
       "3                                                     17    \n",
       "4                                                     62    \n",
       "...                                                  ...    \n",
       "47664                                                 20    \n",
       "47665                                                 33    \n",
       "47666                                                 14    \n",
       "47667                                                 15    \n",
       "47668                                                  8    \n",
       "\n",
       "       Depth of reference-supporting bases (reads1)   \n",
       "0                                                28  \\\n",
       "1                                                10   \n",
       "2                                                 6   \n",
       "3                                                 2   \n",
       "4                                                46   \n",
       "...                                             ...   \n",
       "47664                                             8   \n",
       "47665                                            23   \n",
       "47666                                             6   \n",
       "47667                                             7   \n",
       "47668                                             1   \n",
       "\n",
       "       Depth of variant-supporting bases (reads2)   \n",
       "0                                               7  \\\n",
       "1                                              19   \n",
       "2                                              20   \n",
       "3                                              15   \n",
       "4                                              16   \n",
       "...                                           ...   \n",
       "47664                                          12   \n",
       "47665                                          10   \n",
       "47666                                           8   \n",
       "47667                                           8   \n",
       "47668                                           7   \n",
       "\n",
       "      Depth of reference-supporting bases on forward strand (reads1plus)   \n",
       "0                                                    20%                  \\\n",
       "1                                                 65.52%                   \n",
       "2                                                 76.92%                   \n",
       "3                                                 88.24%                   \n",
       "4                                                 25.81%                   \n",
       "...                                                  ...                   \n",
       "47664                                                60%                   \n",
       "47665                                              30.3%                   \n",
       "47666                                             57.14%                   \n",
       "47667                                             53.33%                   \n",
       "47668                                              87.5%                   \n",
       "\n",
       "       Depth of reference-supporting bases on reverse strand (reads1minus)   \n",
       "0                                           5.609500e-03                    \\\n",
       "1                                           2.114400e-08                     \n",
       "2                                           1.827300e-09                     \n",
       "3                                           7.327700e-08                     \n",
       "4                                           5.029900e-06                     \n",
       "...                                                  ...                     \n",
       "47664                                       2.254800e-05                     \n",
       "47665                                       4.387200e-04                     \n",
       "47666                                       9.661800e-04                     \n",
       "47667                                       1.099500e-03                     \n",
       "47668                                       6.993000e-04                     \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "1                                                      5                  \n",
       "2                                                      4                  \n",
       "3                                                      2                  \n",
       "4                                                     28                  \n",
       "...                                                  ...                  \n",
       "47664                                                  8                  \n",
       "47665                                                 23                  \n",
       "47666                                                  4                  \n",
       "47667                                                  5                  \n",
       "47668                                                  0                  \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "1                                                      5                   \n",
       "2                                                      2                   \n",
       "3                                                      0                   \n",
       "4                                                     18                   \n",
       "...                                                  ...                   \n",
       "47664                                                  0                   \n",
       "47665                                                  0                   \n",
       "47666                                                  2                   \n",
       "47667                                                  2                   \n",
       "47668                                                  1                   \n",
       "\n",
       "       Variant allele frequency  P-value from Fishers Exact Test  \n",
       "0                             7                                0  \n",
       "1                            16                                3  \n",
       "2                            15                                5  \n",
       "3                            15                                0  \n",
       "4                            15                                1  \n",
       "...                         ...                              ...  \n",
       "47664                        12                                0  \n",
       "47665                        10                                0  \n",
       "47666                         7                                1  \n",
       "47667                         7                                1  \n",
       "47668                         6                                1  \n",
       "\n",
       "[47669 rows x 20 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO',\n",
    "               'Genotype', 'Genotype Quality', 'Raw Read Depth as reported by SAMtools', \n",
    "               'Quality Read Depth of bases with Phred score >= 15', \n",
    "               'Depth of reference-supporting bases (reads1)', 'Depth of variant-supporting bases (reads2)', \n",
    "               'Depth of reference-supporting bases on forward strand (reads1plus)',\n",
    "               'Depth of reference-supporting bases on reverse strand (reads1minus)', \n",
    "               'Depth of variant-supporting bases on forward strand (reads2plus)', \n",
    "               'Depth of variant-supporting bases on reverse strand (reads2minus)', \n",
    "               'Variant allele frequency', 'P-value from Fishers Exact Test']\n",
    "vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO',\n",
       "       'Genotype', 'Genotype Quality',\n",
       "       'Raw Read Depth as reported by SAMtools',\n",
       "       'Quality Read Depth of bases with Phred score >= 15',\n",
       "       'Depth of reference-supporting bases (reads1)',\n",
       "       'Depth of variant-supporting bases (reads2)',\n",
       "       'Depth of reference-supporting bases on forward strand (reads1plus)',\n",
       "       'Depth of reference-supporting bases on reverse strand (reads1minus)',\n",
       "       'Depth of variant-supporting bases on forward strand (reads2plus)',\n",
       "       'Depth of variant-supporting bases on reverse strand (reads2minus)',\n",
       "       'Variant allele frequency', 'P-value from Fishers Exact Test'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Quality Read Depth of bases with Phred score &gt;= 15</th>\n",
       "      <th>Depth of reference-supporting bases (reads1)</th>\n",
       "      <th>Depth of variant-supporting bases (reads2)</th>\n",
       "      <th>Depth of reference-supporting bases on forward strand (reads1plus)</th>\n",
       "      <th>Depth of reference-supporting bases on reverse strand (reads1minus)</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>New_Column</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>5.609500e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14907</td>\n",
       "      <td>rs6682375</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>76</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>65.52%</td>\n",
       "      <td>2.114400e-08</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>87</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>76.92%</td>\n",
       "      <td>1.827300e-09</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>...</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17365</td>\n",
       "      <td>rs369606208</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>...</td>\n",
       "      <td>62</td>\n",
       "      <td>46</td>\n",
       "      <td>16</td>\n",
       "      <td>25.81%</td>\n",
       "      <td>5.029900e-06</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47664</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208581</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>60%</td>\n",
       "      <td>2.254800e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47665</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209184</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>23</td>\n",
       "      <td>10</td>\n",
       "      <td>30.3%</td>\n",
       "      <td>4.387200e-04</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47666</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209591</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>9.661800e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47667</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>6.993000e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47669 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS           ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752   rs17855866   G   A    .   PASS  \\\n",
       "1      chr1     14907    rs6682375   A   G    .   PASS   \n",
       "2      chr1     14930    rs6682385   A   G    .   PASS   \n",
       "3      chr1     16495    rs3210724   G   C    .   PASS   \n",
       "4      chr1     17365  rs369606208   C   G    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "47664  chrY  57208581            .   G   C    .   PASS   \n",
       "47665  chrY  57209184            .   T   C    .   PASS   \n",
       "47666  chrY  57209591            .   A   G    .   PASS   \n",
       "47667  chrY  57209613            .   A   G    .   PASS   \n",
       "47668  chrY  57210016            .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....      1/1   \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...      1/1   \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47664  ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47665  ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47666  ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47667  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...   \n",
       "0                    22  ...  \\\n",
       "1                    76  ...   \n",
       "2                    87  ...   \n",
       "3                    71  ...   \n",
       "4                    52  ...   \n",
       "...                 ...  ...   \n",
       "47664                46  ...   \n",
       "47665                33  ...   \n",
       "47666                30  ...   \n",
       "47667                29  ...   \n",
       "47668                31  ...   \n",
       "\n",
       "       Quality Read Depth of bases with Phred score >= 15   \n",
       "0                                                     35   \\\n",
       "1                                                     29    \n",
       "2                                                     26    \n",
       "3                                                     17    \n",
       "4                                                     62    \n",
       "...                                                  ...    \n",
       "47664                                                 20    \n",
       "47665                                                 33    \n",
       "47666                                                 14    \n",
       "47667                                                 15    \n",
       "47668                                                  8    \n",
       "\n",
       "       Depth of reference-supporting bases (reads1)   \n",
       "0                                                28  \\\n",
       "1                                                10   \n",
       "2                                                 6   \n",
       "3                                                 2   \n",
       "4                                                46   \n",
       "...                                             ...   \n",
       "47664                                             8   \n",
       "47665                                            23   \n",
       "47666                                             6   \n",
       "47667                                             7   \n",
       "47668                                             1   \n",
       "\n",
       "       Depth of variant-supporting bases (reads2)   \n",
       "0                                               7  \\\n",
       "1                                              19   \n",
       "2                                              20   \n",
       "3                                              15   \n",
       "4                                              16   \n",
       "...                                           ...   \n",
       "47664                                          12   \n",
       "47665                                          10   \n",
       "47666                                           8   \n",
       "47667                                           8   \n",
       "47668                                           7   \n",
       "\n",
       "       Depth of reference-supporting bases on forward strand (reads1plus)   \n",
       "0                                                    20%                   \\\n",
       "1                                                 65.52%                    \n",
       "2                                                 76.92%                    \n",
       "3                                                 88.24%                    \n",
       "4                                                 25.81%                    \n",
       "...                                                  ...                    \n",
       "47664                                                60%                    \n",
       "47665                                              30.3%                    \n",
       "47666                                             57.14%                    \n",
       "47667                                             53.33%                    \n",
       "47668                                              87.5%                    \n",
       "\n",
       "      Depth of reference-supporting bases on reverse strand (reads1minus)   \n",
       "0                                           5.609500e-03                   \\\n",
       "1                                           2.114400e-08                    \n",
       "2                                           1.827300e-09                    \n",
       "3                                           7.327700e-08                    \n",
       "4                                           5.029900e-06                    \n",
       "...                                                  ...                    \n",
       "47664                                       2.254800e-05                    \n",
       "47665                                       4.387200e-04                    \n",
       "47666                                       9.661800e-04                    \n",
       "47667                                       1.099500e-03                    \n",
       "47668                                       6.993000e-04                    \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "1                                                      5                  \n",
       "2                                                      4                  \n",
       "3                                                      2                  \n",
       "4                                                     28                  \n",
       "...                                                  ...                  \n",
       "47664                                                  8                  \n",
       "47665                                                 23                  \n",
       "47666                                                  4                  \n",
       "47667                                                  5                  \n",
       "47668                                                  0                  \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "1                                                      5                   \n",
       "2                                                      2                   \n",
       "3                                                      0                   \n",
       "4                                                     18                   \n",
       "...                                                  ...                   \n",
       "47664                                                  0                   \n",
       "47665                                                  0                   \n",
       "47666                                                  2                   \n",
       "47667                                                  2                   \n",
       "47668                                                  1                   \n",
       "\n",
       "       Variant allele frequency  P-value from Fishers Exact Test   \n",
       "0                             7                                0  \\\n",
       "1                            16                                3   \n",
       "2                            15                                5   \n",
       "3                            15                                0   \n",
       "4                            15                                1   \n",
       "...                         ...                              ...   \n",
       "47664                        12                                0   \n",
       "47665                        10                                0   \n",
       "47666                         7                                1   \n",
       "47667                         7                                1   \n",
       "47668                         6                                1   \n",
       "\n",
       "                                              New_Column  \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  \n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....  \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...  \n",
       "...                                                  ...  \n",
       "47664                       ADP=20;WT=0;HET=1;HOM=0;NC=0  \n",
       "47665                       ADP=33;WT=0;HET=1;HOM=0;NC=0  \n",
       "47666                       ADP=14;WT=0;HET=1;HOM=0;NC=0  \n",
       "47667                       ADP=15;WT=0;HET=1;HOM=0;NC=0  \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  \n",
       "\n",
       "[47669 rows x 21 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"New_Column\"] = vcf[\"INFO\"].str.extract(r'^(.*?)(?=;?CSQ)')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Depth of reference-supporting bases on forward strand (reads1plus)</th>\n",
       "      <th>Depth of reference-supporting bases on reverse strand (reads1minus)</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>New_Column</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>20%</td>\n",
       "      <td>5.609500e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14907</td>\n",
       "      <td>rs6682375</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>76</td>\n",
       "      <td>...</td>\n",
       "      <td>65.52%</td>\n",
       "      <td>2.114400e-08</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>87</td>\n",
       "      <td>...</td>\n",
       "      <td>76.92%</td>\n",
       "      <td>1.827300e-09</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>...</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17365</td>\n",
       "      <td>rs369606208</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>...</td>\n",
       "      <td>25.81%</td>\n",
       "      <td>5.029900e-06</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47664</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208581</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>60%</td>\n",
       "      <td>2.254800e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47665</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209184</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>...</td>\n",
       "      <td>30.3%</td>\n",
       "      <td>4.387200e-04</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47666</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209591</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>...</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>9.661800e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47667</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>...</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>6.993000e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47669 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS           ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752   rs17855866   G   A    .   PASS  \\\n",
       "1      chr1     14907    rs6682375   A   G    .   PASS   \n",
       "2      chr1     14930    rs6682385   A   G    .   PASS   \n",
       "3      chr1     16495    rs3210724   G   C    .   PASS   \n",
       "4      chr1     17365  rs369606208   C   G    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "47664  chrY  57208581            .   G   C    .   PASS   \n",
       "47665  chrY  57209184            .   T   C    .   PASS   \n",
       "47666  chrY  57209591            .   A   G    .   PASS   \n",
       "47667  chrY  57209613            .   A   G    .   PASS   \n",
       "47668  chrY  57210016            .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....      1/1   \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...      1/1   \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47664  ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47665  ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47666  ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47667  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...   \n",
       "0                    22  ...  \\\n",
       "1                    76  ...   \n",
       "2                    87  ...   \n",
       "3                    71  ...   \n",
       "4                    52  ...   \n",
       "...                 ...  ...   \n",
       "47664                46  ...   \n",
       "47665                33  ...   \n",
       "47666                30  ...   \n",
       "47667                29  ...   \n",
       "47668                31  ...   \n",
       "\n",
       "       Depth of reference-supporting bases on forward strand (reads1plus)   \n",
       "0                                                    20%                   \\\n",
       "1                                                 65.52%                    \n",
       "2                                                 76.92%                    \n",
       "3                                                 88.24%                    \n",
       "4                                                 25.81%                    \n",
       "...                                                  ...                    \n",
       "47664                                                60%                    \n",
       "47665                                              30.3%                    \n",
       "47666                                             57.14%                    \n",
       "47667                                             53.33%                    \n",
       "47668                                              87.5%                    \n",
       "\n",
       "       Depth of reference-supporting bases on reverse strand (reads1minus)   \n",
       "0                                           5.609500e-03                    \\\n",
       "1                                           2.114400e-08                     \n",
       "2                                           1.827300e-09                     \n",
       "3                                           7.327700e-08                     \n",
       "4                                           5.029900e-06                     \n",
       "...                                                  ...                     \n",
       "47664                                       2.254800e-05                     \n",
       "47665                                       4.387200e-04                     \n",
       "47666                                       9.661800e-04                     \n",
       "47667                                       1.099500e-03                     \n",
       "47668                                       6.993000e-04                     \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "1                                                      5                  \n",
       "2                                                      4                  \n",
       "3                                                      2                  \n",
       "4                                                     28                  \n",
       "...                                                  ...                  \n",
       "47664                                                  8                  \n",
       "47665                                                 23                  \n",
       "47666                                                  4                  \n",
       "47667                                                  5                  \n",
       "47668                                                  0                  \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "1                                                      5                   \n",
       "2                                                      2                   \n",
       "3                                                      0                   \n",
       "4                                                     18                   \n",
       "...                                                  ...                   \n",
       "47664                                                  0                   \n",
       "47665                                                  0                   \n",
       "47666                                                  2                   \n",
       "47667                                                  2                   \n",
       "47668                                                  1                   \n",
       "\n",
       "      Variant allele frequency  P-value from Fishers Exact Test   \n",
       "0                            7                                0  \\\n",
       "1                           16                                3   \n",
       "2                           15                                5   \n",
       "3                           15                                0   \n",
       "4                           15                                1   \n",
       "...                        ...                              ...   \n",
       "47664                       12                                0   \n",
       "47665                       10                                0   \n",
       "47666                        7                                1   \n",
       "47667                        7                                1   \n",
       "47668                        6                                1   \n",
       "\n",
       "                                              New_Column  HET  HOM   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0  \\\n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0   \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....    0    1   \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...    0    1   \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...    1    0   \n",
       "...                                                  ...  ...  ...   \n",
       "47664                       ADP=20;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47665                       ADP=33;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47666                       ADP=14;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47667                       ADP=15;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0    0    1   \n",
       "\n",
       "           Zygosity  \n",
       "0      Heterozygous  \n",
       "1      Heterozygous  \n",
       "2        Homozygous  \n",
       "3        Homozygous  \n",
       "4      Heterozygous  \n",
       "...             ...  \n",
       "47664  Heterozygous  \n",
       "47665  Heterozygous  \n",
       "47666  Heterozygous  \n",
       "47667  Heterozygous  \n",
       "47668    Homozygous  \n",
       "\n",
       "[47669 rows x 24 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Depth of reference-supporting bases on reverse strand (reads1minus)</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>New_Column</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>5.609500e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14907</td>\n",
       "      <td>rs6682375</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>76</td>\n",
       "      <td>...</td>\n",
       "      <td>2.114400e-08</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "      <td>ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14930</td>\n",
       "      <td>rs6682385</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>87</td>\n",
       "      <td>...</td>\n",
       "      <td>1.827300e-09</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16495</td>\n",
       "      <td>rs3210724</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>...</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17365</td>\n",
       "      <td>rs369606208</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>...</td>\n",
       "      <td>5.029900e-06</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47664</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208581</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>2.254800e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47665</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209184</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>33</td>\n",
       "      <td>...</td>\n",
       "      <td>4.387200e-04</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=33;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47666</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209591</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>...</td>\n",
       "      <td>9.661800e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=14;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>G|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47667</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>...</td>\n",
       "      <td>1.099500e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>G|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>6.993000e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47669 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS           ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752   rs17855866   G   A    .   PASS  \\\n",
       "1      chr1     14907    rs6682375   A   G    .   PASS   \n",
       "2      chr1     14930    rs6682385   A   G    .   PASS   \n",
       "3      chr1     16495    rs3210724   G   C    .   PASS   \n",
       "4      chr1     17365  rs369606208   C   G    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "47664  chrY  57208581            .   G   C    .   PASS   \n",
       "47665  chrY  57209184            .   T   C    .   PASS   \n",
       "47666  chrY  57209591            .   A   G    .   PASS   \n",
       "47667  chrY  57209613            .   A   G    .   PASS   \n",
       "47668  chrY  57210016            .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....      1/1   \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...      1/1   \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47664  ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47665  ADP=33;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...      0/1   \n",
       "47666  ADP=14;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47667  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...      0/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...   \n",
       "0                    22  ...  \\\n",
       "1                    76  ...   \n",
       "2                    87  ...   \n",
       "3                    71  ...   \n",
       "4                    52  ...   \n",
       "...                 ...  ...   \n",
       "47664                46  ...   \n",
       "47665                33  ...   \n",
       "47666                30  ...   \n",
       "47667                29  ...   \n",
       "47668                31  ...   \n",
       "\n",
       "       Depth of reference-supporting bases on reverse strand (reads1minus)   \n",
       "0                                           5.609500e-03                    \\\n",
       "1                                           2.114400e-08                     \n",
       "2                                           1.827300e-09                     \n",
       "3                                           7.327700e-08                     \n",
       "4                                           5.029900e-06                     \n",
       "...                                                  ...                     \n",
       "47664                                       2.254800e-05                     \n",
       "47665                                       4.387200e-04                     \n",
       "47666                                       9.661800e-04                     \n",
       "47667                                       1.099500e-03                     \n",
       "47668                                       6.993000e-04                     \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "1                                                      5                  \n",
       "2                                                      4                  \n",
       "3                                                      2                  \n",
       "4                                                     28                  \n",
       "...                                                  ...                  \n",
       "47664                                                  8                  \n",
       "47665                                                 23                  \n",
       "47666                                                  4                  \n",
       "47667                                                  5                  \n",
       "47668                                                  0                  \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "1                                                      5                   \n",
       "2                                                      2                   \n",
       "3                                                      0                   \n",
       "4                                                     18                   \n",
       "...                                                  ...                   \n",
       "47664                                                  0                   \n",
       "47665                                                  0                   \n",
       "47666                                                  2                   \n",
       "47667                                                  2                   \n",
       "47668                                                  1                   \n",
       "\n",
       "       Variant allele frequency P-value from Fishers Exact Test   \n",
       "0                             7                               0  \\\n",
       "1                            16                               3   \n",
       "2                            15                               5   \n",
       "3                            15                               0   \n",
       "4                            15                               1   \n",
       "...                         ...                             ...   \n",
       "47664                        12                               0   \n",
       "47665                        10                               0   \n",
       "47666                         7                               1   \n",
       "47667                         7                               1   \n",
       "47668                         6                               1   \n",
       "\n",
       "                                              New_Column  HET  HOM   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0  \\\n",
       "1      ADP=29;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0   \n",
       "2      ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5178,0....    0    1   \n",
       "3      ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...    0    1   \n",
       "4      ADP=62;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEIN...    1    0   \n",
       "...                                                  ...  ...  ...   \n",
       "47664                       ADP=20;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47665                       ADP=33;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47666                       ADP=14;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47667                       ADP=15;WT=0;HET=1;HOM=0;NC=0    1    0   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0    0    1   \n",
       "\n",
       "           Zygosity                                                CSQ  \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "1      Heterozygous  G|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "2        Homozygous  G|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "3        Homozygous  C|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "4      Heterozygous  G|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "...             ...                                                ...  \n",
       "47664  Heterozygous  C|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "47665  Heterozygous  C|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "47666  Heterozygous  G|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "47667  Heterozygous  G|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...  \n",
       "\n",
       "[47669 rows x 25 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>New_Column</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...   \n",
       "0                    22  ...  \\\n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "...                 ...  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "...                                                  ...                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "...                                                  ...                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "\n",
       "       Variant allele frequency  P-value from Fishers Exact Test   \n",
       "0                             7                                0  \\\n",
       "0                             7                                0   \n",
       "0                             7                                0   \n",
       "0                             7                                0   \n",
       "0                             7                                0   \n",
       "...                         ...                              ...   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "\n",
       "                                              New_Column  HET  HOM   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...    1    0   \n",
       "...                                                  ...  ...  ...   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0    0    1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0    0    1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0    0    1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0    0    1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0    0    1   \n",
       "\n",
       "           Zygosity                                                CSQ   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...             ...                                                ...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq  \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...  \n",
       "0      A|intron_variant&non_coding_transcript_variant...  \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...  \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...  \n",
       "...                                                  ...  \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...  \n",
       "47668  T|intron_variant&non_coding_transcript_variant...  \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...  \n",
       "\n",
       "[371550 rows x 26 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  PUBMED  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS   \n",
       "0                    22  ...                                               \\\n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "...                 ...  ...     ...         ...        ...           ...   \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "\n",
       "      MOTIF_SCORE_CHANGE  TRANSCRIPTION_FACTORS  ClinVar  ClinVar_CLNSIG   \n",
       "0                                                                         \\\n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "...                  ...                    ...      ...             ...   \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar_CLNDN  \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "...                   ...            ...  \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "\n",
       "[371550 rows x 109 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Allele'] = vcf['csq'].str.split('|').str[0]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SYMBOL'] = vcf['csq'].str.split('|').str[3]\n",
    "vcf['Gene'] = vcf['csq'].str.split('|').str[4]\n",
    "vcf['Feature_type'] = vcf['csq'].str.split('|').str[5]\n",
    "vcf['Feature'] = vcf['csq'].str.split('|').str[6]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['HGVSc'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSp'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['cDNA_position'] = vcf['csq'].str.split('|').str[12]\n",
    "vcf['CDS_position'] = vcf['csq'].str.split('|').str[13]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['Existing_variation'] = vcf['csq'].str.split('|').str[17]\n",
    "vcf['DISTANCE'] = vcf['csq'].str.split('|').str[18]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['FLAGS'] = vcf['csq'].str.split('|').str[20]\n",
    "vcf['VARIANT_CLASS'] = vcf['csq'].str.split('|').str[21]\n",
    "vcf['SYMBOL_SOURCE'] = vcf['csq'].str.split('|').str[22]\n",
    "vcf['HGNC_ID'] = vcf['csq'].str.split('|').str[23]\n",
    "vcf['CANONICAL'] = vcf['csq'].str.split('|').str[24]\n",
    "vcf['MANE_SELECT'] = vcf['csq'].str.split('|').str[25]\n",
    "vcf['MANE_PLUS_CLINICAL'] = vcf['csq'].str.split('|').str[26]\n",
    "vcf['TSL'] = vcf['csq'].str.split('|').str[27]\n",
    "vcf['APPRIS'] = vcf['csq'].str.split('|').str[28]\n",
    "vcf['CCDS'] = vcf['csq'].str.split('|').str[29]\n",
    "vcf['ENSP'] = vcf['csq'].str.split('|').str[30]\n",
    "vcf['SWISSPROT'] = vcf['csq'].str.split('|').str[31]\n",
    "vcf['TREMBL'] = vcf['csq'].str.split('|').str[32]\n",
    "vcf['UNIPARC'] = vcf['csq'].str.split('|').str[33]\n",
    "vcf['UNIPROT_ISOFORM'] = vcf['csq'].str.split('|').str[34]\n",
    "vcf['SOURCE'] = vcf['csq'].str.split('|').str[35]\n",
    "vcf['GENE_PHENO'] = vcf['csq'].str.split('|').str[36]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['DOMAINS'] = vcf['csq'].str.split('|').str[39]\n",
    "vcf['miRNA'] = vcf['csq'].str.split('|').str[40]\n",
    "vcf['HGVS_OFFSET'] = vcf['csq'].str.split('|').str[41]\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['SOMATIC'] = vcf['csq'].str.split('|').str[71]\n",
    "vcf['PHENO'] = vcf['csq'].str.split('|').str[72]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['MOTIF_NAME'] = vcf['csq'].str.split('|').str[74]\n",
    "vcf['MOTIF_POS'] = vcf['csq'].str.split('|').str[75]\n",
    "vcf['HIGH_INF_POS'] = vcf['csq'].str.split('|').str[76]\n",
    "vcf['MOTIF_SCORE_CHANGE'] = vcf['csq'].str.split('|').str[77]\n",
    "vcf['TRANSCRIPTION_FACTORS'] = vcf['csq'].str.split('|').str[78]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['ClinVar_CLNSIG'] = vcf['csq'].str.split('|').str[80]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  PUBMED  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS   \n",
       "0                    22  ...                                               \\\n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "...                 ...  ...     ...         ...        ...           ...   \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "\n",
       "      MOTIF_SCORE_CHANGE  TRANSCRIPTION_FACTORS  ClinVar  ClinVar_CLNSIG   \n",
       "0                                                                         \\\n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "...                  ...                    ...      ...             ...   \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar_CLNDN  \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "...                   ...            ...  \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "\n",
       "[371550 rows x 109 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1 = vcf.copy()\n",
    "vcf_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>SAO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 159 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866       A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...    SAO                                  GENEINFO   \n",
       "0                    22  ...  SAO=0  GENEINFO=DDX11L1:100287102|WASH7P:653635  \\\n",
       "0                    22  ...  SAO=0  GENEINFO=DDX11L1:100287102|WASH7P:653635   \n",
       "0                    22  ...  SAO=0  GENEINFO=DDX11L1:100287102|WASH7P:653635   \n",
       "0                    22  ...  SAO=0  GENEINFO=DDX11L1:100287102|WASH7P:653635   \n",
       "0                    22  ...  SAO=0  GENEINFO=DDX11L1:100287102|WASH7P:653635   \n",
       "...                 ...  ...    ...                                       ...   \n",
       "47668                31  ...                                                    \n",
       "47668                31  ...                                                    \n",
       "47668                31  ...                                                    \n",
       "47668                31  ...                                                    \n",
       "47668                31  ...                                                    \n",
       "\n",
       "       INT  G5 OM  PMC    SSR        RSPOS  HD  PM  \n",
       "0                       SSR=0  RSPOS=14752          \n",
       "0                       SSR=0  RSPOS=14752          \n",
       "0                       SSR=0  RSPOS=14752          \n",
       "0                       SSR=0  RSPOS=14752          \n",
       "0                       SSR=0  RSPOS=14752          \n",
       "...    ...  .. ..  ...    ...          ...  ..  ..  \n",
       "47668                                               \n",
       "47668                                               \n",
       "47668                                               \n",
       "47668                                               \n",
       "47668                                               \n",
       "\n",
       "[371550 rows x 159 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create empty columns\n",
    "columns = ['ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID', 'SLO',\n",
    "           'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV', 'TPA', 'CFL', 'GNO',\n",
    "           'VLD', 'ASP', 'ASS', 'REF', 'U3', 'U5', 'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC',\n",
    "           'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO',\n",
    "           'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar',\n",
    "           'ClinVar_CLNSIG']\n",
    "\n",
    "for col in columns:\n",
    "    vcf_1[col] = ''\n",
    "\n",
    "# Populate columns based on 'info' values\n",
    "for i, row in vcf_1.iterrows():\n",
    "    info = row['New_Column']\n",
    "    items = info.split(';')\n",
    "    for item in items:\n",
    "        key_value = item.split('=')\n",
    "        key = key_value[0]\n",
    "        if key in columns:\n",
    "            if len(key_value) > 1:\n",
    "                value = key_value[1]\n",
    "                vcf_1.at[i, key] = f\"{key}={value}\"\n",
    "            else:\n",
    "                vcf_1.at[i, key] = key\n",
    "        else:\n",
    "            vcf_1.at[i, key] = 'null'\n",
    "\n",
    "# Print the resulting DataFrame\n",
    "vcf_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID\n",
       ".              6677\n",
       "rs28529458      149\n",
       "rs12624152      145\n",
       "rs958           122\n",
       "rs183956        112\n",
       "               ... \n",
       "rs4768269         1\n",
       "rs749770969       1\n",
       "rs4768275         1\n",
       "rs2933356         1\n",
       "rs6568353         1\n",
       "Name: count, Length: 44733, dtype: int64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1.ID.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>Raw Read Depth as reported by SAMtools</th>\n",
       "      <th>Quality Read Depth of bases with Phred score &gt;= 15</th>\n",
       "      <th>Depth of reference-supporting bases (reads1)</th>\n",
       "      <th>Depth of variant-supporting bases (reads2)</th>\n",
       "      <th>Depth of reference-supporting bases on forward strand (reads1plus)</th>\n",
       "      <th>Depth of reference-supporting bases on reverse strand (reads1minus)</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>New_Column</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE_SELECT</th>\n",
       "      <th>MANE_PLUS_CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT_ISOFORM</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450305</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>1082</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37102</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.00162</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001186</td>\n",
       "      <td>0.0002987</td>\n",
       "      <td>0.0001935</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6.055e-05</td>\n",
       "      <td>0.00203</td>\n",
       "      <td>0.0002195</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=123</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.99442533129459734,0.00553484964322120...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000040005000002000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>ENSG00000290825</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>343</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td></td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.00162</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001186</td>\n",
       "      <td>0.0002987</td>\n",
       "      <td>0.0001935</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6.055e-05</td>\n",
       "      <td>0.00203</td>\n",
       "      <td>0.0002195</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=123</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.99442533129459734,0.00553484964322120...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000040005000002000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000488147</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td>10/10</td>\n",
       "      <td>ENST00000488147.1:n.1254-251C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:38034</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.00162</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001186</td>\n",
       "      <td>0.0002987</td>\n",
       "      <td>0.0001935</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6.055e-05</td>\n",
       "      <td>0.00203</td>\n",
       "      <td>0.0002195</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=123</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.99442533129459734,0.00553484964322120...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000040005000002000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619216</td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>2617</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:50039</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.00162</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001186</td>\n",
       "      <td>0.0002987</td>\n",
       "      <td>0.0001935</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6.055e-05</td>\n",
       "      <td>0.00203</td>\n",
       "      <td>0.0002195</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=123</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.99442533129459734,0.00553484964322120...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000040005000002000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001164745</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.00162</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001186</td>\n",
       "      <td>0.0002987</td>\n",
       "      <td>0.0001935</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6.055e-05</td>\n",
       "      <td>0.00203</td>\n",
       "      <td>0.0002195</td>\n",
       "      <td>0.005487</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=123</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.99442533129459734,0.00553484964322120...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000040005000002000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=DDX11L1:100287102|WASH7P:653635</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=14752</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>T</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000484415</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.813C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>813</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>T</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000492963</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/6</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.1157C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>1157</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000496011</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>289</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000496301</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>ENST00000496301.6:n.94+36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>ENSG00000227159</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000507418</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2168</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37115</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 160 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866       A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "0      chr1     14752  rs17855866       A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "47668  chrY  57210016           .       T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  Raw Read Depth as reported by SAMtools   \n",
       "0                    22                                      35  \\\n",
       "0                    22                                      35   \n",
       "0                    22                                      35   \n",
       "0                    22                                      35   \n",
       "0                    22                                      35   \n",
       "...                 ...                                     ...   \n",
       "47668                31                                       8   \n",
       "47668                31                                       8   \n",
       "47668                31                                       8   \n",
       "47668                31                                       8   \n",
       "47668                31                                       8   \n",
       "\n",
       "       Quality Read Depth of bases with Phred score >= 15   \n",
       "0                                                     35   \\\n",
       "0                                                     35    \n",
       "0                                                     35    \n",
       "0                                                     35    \n",
       "0                                                     35    \n",
       "...                                                  ...    \n",
       "47668                                                  8    \n",
       "47668                                                  8    \n",
       "47668                                                  8    \n",
       "47668                                                  8    \n",
       "47668                                                  8    \n",
       "\n",
       "       Depth of reference-supporting bases (reads1)   \n",
       "0                                                28  \\\n",
       "0                                                28   \n",
       "0                                                28   \n",
       "0                                                28   \n",
       "0                                                28   \n",
       "...                                             ...   \n",
       "47668                                             1   \n",
       "47668                                             1   \n",
       "47668                                             1   \n",
       "47668                                             1   \n",
       "47668                                             1   \n",
       "\n",
       "       Depth of variant-supporting bases (reads2)   \n",
       "0                                               7  \\\n",
       "0                                               7   \n",
       "0                                               7   \n",
       "0                                               7   \n",
       "0                                               7   \n",
       "...                                           ...   \n",
       "47668                                           7   \n",
       "47668                                           7   \n",
       "47668                                           7   \n",
       "47668                                           7   \n",
       "47668                                           7   \n",
       "\n",
       "      Depth of reference-supporting bases on forward strand (reads1plus)   \n",
       "0                                                    20%                  \\\n",
       "0                                                    20%                   \n",
       "0                                                    20%                   \n",
       "0                                                    20%                   \n",
       "0                                                    20%                   \n",
       "...                                                  ...                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "\n",
       "       Depth of reference-supporting bases on reverse strand (reads1minus)   \n",
       "0                                               0.005609                    \\\n",
       "0                                               0.005609                     \n",
       "0                                               0.005609                     \n",
       "0                                               0.005609                     \n",
       "0                                               0.005609                     \n",
       "...                                                  ...                     \n",
       "47668                                           0.000699                     \n",
       "47668                                           0.000699                     \n",
       "47668                                           0.000699                     \n",
       "47668                                           0.000699                     \n",
       "47668                                           0.000699                     \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "...                                                  ...                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "...                                                  ...                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "\n",
       "       Variant allele frequency  P-value from Fishers Exact Test   \n",
       "0                             7                                0  \\\n",
       "0                             7                                0   \n",
       "0                             7                                0   \n",
       "0                             7                                0   \n",
       "0                             7                                0   \n",
       "...                         ...                              ...   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "47668                         6                                1   \n",
       "\n",
       "                                              New_Column    HET    HOM   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  HET=1  HOM=0  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  HET=1  HOM=0   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  HET=1  HOM=0   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  HET=1  HOM=0   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  HET=1  HOM=0   \n",
       "...                                                  ...    ...    ...   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  HET=0  HOM=1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  HET=0  HOM=1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  HET=0  HOM=1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  HET=0  HOM=1   \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  HET=0  HOM=1   \n",
       "\n",
       "           Zygosity                                                CSQ   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...             ...                                                ...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq Allele   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...      A  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...      A   \n",
       "0      A|intron_variant&non_coding_transcript_variant...      A   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...      A   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...      A   \n",
       "...                                                  ...    ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...      T   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...      T   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...      T   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...      T   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...      T   \n",
       "\n",
       "                                        Consequence    IMPACT     SYMBOL   \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L1  \\\n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L2   \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                           downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                         regulatory_region_variant  MODIFIER              \n",
       "...                                             ...       ...        ...   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER     WASH6P   \n",
       "47668  intron_variant&non_coding_transcript_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER   DDX11L16   \n",
       "\n",
       "                  Gene       Feature_type          Feature   \n",
       "0      ENSG00000223972         Transcript  ENST00000450305  \\\n",
       "0      ENSG00000290825         Transcript  ENST00000456328   \n",
       "0      ENSG00000227232         Transcript  ENST00000488147   \n",
       "0      ENSG00000278267         Transcript  ENST00000619216   \n",
       "0                       RegulatoryFeature  ENSR00001164745   \n",
       "...                ...                ...              ...   \n",
       "47668  ENSG00000182484         Transcript  ENST00000484415   \n",
       "47668  ENSG00000182484         Transcript  ENST00000492963   \n",
       "47668  ENSG00000182484         Transcript  ENST00000496011   \n",
       "47668  ENSG00000182484         Transcript  ENST00000496301   \n",
       "47668  ENSG00000227159         Transcript  ENST00000507418   \n",
       "\n",
       "                                  BIOTYPE EXON INTRON   \n",
       "0      transcribed_unprocessed_pseudogene              \\\n",
       "0                                  lncRNA               \n",
       "0                  unprocessed_pseudogene       10/10   \n",
       "0                                   miRNA               \n",
       "0                                enhancer               \n",
       "...                                   ...  ...    ...   \n",
       "47668                     retained_intron  5/5          \n",
       "47668                     retained_intron  5/6          \n",
       "47668                     retained_intron               \n",
       "47668                     retained_intron         1/1   \n",
       "47668              unprocessed_pseudogene               \n",
       "\n",
       "                                 HGVSc HGVSp cDNA_position CDS_position   \n",
       "0                                                                        \\\n",
       "0                                                                         \n",
       "0      ENST00000488147.1:n.1254-251C>T                                    \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "...                                ...   ...           ...          ...   \n",
       "47668       ENST00000484415.6:n.813C>T                 813                \n",
       "47668      ENST00000492963.6:n.1157C>T                1157                \n",
       "47668                                                                     \n",
       "47668     ENST00000496301.6:n.94+36C>T                                    \n",
       "47668                                                                     \n",
       "\n",
       "      Protein_position Amino_acids Codons Existing_variation DISTANCE STRAND   \n",
       "0                                                 rs17855866     1082      1  \\\n",
       "0                                                 rs17855866      343      1   \n",
       "0                                                 rs17855866              -1   \n",
       "0                                                 rs17855866     2617     -1   \n",
       "0                                                 rs17855866                   \n",
       "...                ...         ...    ...                ...      ...    ...   \n",
       "47668                                                                      1   \n",
       "47668                                                                      1   \n",
       "47668                                                             289      1   \n",
       "47668                                                                      1   \n",
       "47668                                                            2168     -1   \n",
       "\n",
       "      FLAGS VARIANT_CLASS SYMBOL_SOURCE     HGNC_ID CANONICAL MANE_SELECT   \n",
       "0                     SNV          HGNC  HGNC:37102       YES              \\\n",
       "0                     SNV    EntrezGene                   YES               \n",
       "0                     SNV          HGNC  HGNC:38034       YES               \n",
       "0                     SNV          HGNC  HGNC:50039       YES               \n",
       "0                     SNV                                                   \n",
       "...     ...           ...           ...         ...       ...         ...   \n",
       "47668                 SNV          HGNC  HGNC:31685                         \n",
       "47668                 SNV          HGNC  HGNC:31685                         \n",
       "47668                 SNV          HGNC  HGNC:31685                         \n",
       "47668                 SNV          HGNC  HGNC:31685                         \n",
       "47668                 SNV          HGNC  HGNC:37115       YES               \n",
       "\n",
       "      MANE_PLUS_CLINICAL TSL APPRIS CCDS ENSP SWISSPROT TREMBL UNIPARC   \n",
       "0                                                                       \\\n",
       "0                          1                                             \n",
       "0                                                                        \n",
       "0                                                                        \n",
       "0                                                                        \n",
       "...                  ...  ..    ...  ...  ...       ...    ...     ...   \n",
       "47668                      5                                             \n",
       "47668                      5                                             \n",
       "47668                      5                                             \n",
       "47668                      2                                             \n",
       "47668                                                                    \n",
       "\n",
       "      UNIPROT_ISOFORM SOURCE GENE_PHENO SIFT PolyPhen DOMAINS miRNA   \n",
       "0                                                                    \\\n",
       "0                                                                     \n",
       "0                                                                     \n",
       "0                                                                     \n",
       "0                                                                     \n",
       "...               ...    ...        ...  ...      ...     ...   ...   \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "\n",
       "      HGVS_OFFSET AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF   \n",
       "0                                                                   \\\n",
       "0                                                                    \n",
       "0                                                                    \n",
       "0                                                                    \n",
       "0                                                                    \n",
       "...           ... ..    ...    ...    ...    ...    ...        ...   \n",
       "47668                                                                \n",
       "47668                                                                \n",
       "47668                                                                \n",
       "47668                                                                \n",
       "47668                                                                \n",
       "\n",
       "      gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF   \n",
       "0                                                                  \\\n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF   \n",
       "0                                                                    0.00162  \\\n",
       "0                                                                    0.00162   \n",
       "0                                                                    0.00162   \n",
       "0                                                                    0.00162   \n",
       "0                                                                    0.00162   \n",
       "...              ...            ...            ...            ...        ...   \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF   \n",
       "0           0.005487              0       0.001186      0.0002987  \\\n",
       "0           0.005487              0       0.001186      0.0002987   \n",
       "0           0.005487              0       0.001186      0.0002987   \n",
       "0           0.005487              0       0.001186      0.0002987   \n",
       "0           0.005487              0       0.001186      0.0002987   \n",
       "...              ...            ...            ...            ...   \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF   \n",
       "0          0.0001935              0              0      6.055e-05  \\\n",
       "0          0.0001935              0              0      6.055e-05   \n",
       "0          0.0001935              0              0      6.055e-05   \n",
       "0          0.0001935              0              0      6.055e-05   \n",
       "0          0.0001935              0              0      6.055e-05   \n",
       "...              ...            ...            ...            ...   \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "47668                                                               \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF    MAX_AF  MAX_AF_POPS CLIN_SIG SOMATIC   \n",
       "0            0.00203      0.0002195  0.005487  gnomADg_AFR                   \\\n",
       "0            0.00203      0.0002195  0.005487  gnomADg_AFR                    \n",
       "0            0.00203      0.0002195  0.005487  gnomADg_AFR                    \n",
       "0            0.00203      0.0002195  0.005487  gnomADg_AFR                    \n",
       "0            0.00203      0.0002195  0.005487  gnomADg_AFR                    \n",
       "...              ...            ...       ...          ...      ...     ...   \n",
       "47668                                                                         \n",
       "47668                                                                         \n",
       "47668                                                                         \n",
       "47668                                                                         \n",
       "47668                                                                         \n",
       "\n",
       "      PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE   \n",
       "0                                                                        \\\n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "...     ...    ...        ...       ...          ...                ...   \n",
       "47668                                                                     \n",
       "47668                                                                     \n",
       "47668                                                                     \n",
       "47668                                                                     \n",
       "47668                                                                     \n",
       "\n",
       "      TRANSCRIPTION_FACTORS ClinVar ClinVar_CLNSIG ClinVar_CLNREVSTAT   \n",
       "0                                                                      \\\n",
       "0                                                                       \n",
       "0                                                                       \n",
       "0                                                                       \n",
       "0                                                                       \n",
       "...                     ...     ...            ...                ...   \n",
       "47668                                                                   \n",
       "47668                                                                   \n",
       "47668                                                                   \n",
       "47668                                                                   \n",
       "47668                                                                   \n",
       "\n",
       "      ClinVar_CLNDN     ADP    WT    NC CDA OTH S3D WTD      dbSNPBuildID SLO   \n",
       "0                    ADP=35  WT=0  NC=0                  dbSNPBuildID=123      \\\n",
       "0                    ADP=35  WT=0  NC=0                  dbSNPBuildID=123       \n",
       "0                    ADP=35  WT=0  NC=0                  dbSNPBuildID=123       \n",
       "0                    ADP=35  WT=0  NC=0                  dbSNPBuildID=123       \n",
       "0                    ADP=35  WT=0  NC=0                  dbSNPBuildID=123       \n",
       "...             ...     ...   ...   ...  ..  ..  ..  ..               ...  ..   \n",
       "47668                 ADP=8  WT=0  NC=0                                         \n",
       "47668                 ADP=8  WT=0  NC=0                                         \n",
       "47668                 ADP=8  WT=0  NC=0                                         \n",
       "47668                 ADP=8  WT=0  NC=0                                         \n",
       "47668                 ADP=8  WT=0  NC=0                                         \n",
       "\n",
       "      NSF  R3 R5 NSN NSM G5A COMMON           RS RV TPA CFL GNO VLD  ASP ASS   \n",
       "0          R3                        RS=17855866                     ASP      \\\n",
       "0          R3                        RS=17855866                     ASP       \n",
       "0          R3                        RS=17855866                     ASP       \n",
       "0          R3                        RS=17855866                     ASP       \n",
       "0          R3                        RS=17855866                     ASP       \n",
       "...    ..  .. ..  ..  ..  ..    ...          ... ..  ..  ..  ..  ..  ...  ..   \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "47668                                                                          \n",
       "\n",
       "      U3 U5                                             TOPMED    WGT MTP LSD   \n",
       "0            TOPMED=0.99442533129459734,0.00553484964322120...  WGT=1          \\\n",
       "0            TOPMED=0.99442533129459734,0.00553484964322120...  WGT=1           \n",
       "0            TOPMED=0.99442533129459734,0.00553484964322120...  WGT=1           \n",
       "0            TOPMED=0.99442533129459734,0.00553484964322120...  WGT=1           \n",
       "0            TOPMED=0.99442533129459734,0.00553484964322120...  WGT=1           \n",
       "...   .. ..                                                ...    ...  ..  ..   \n",
       "47668                                                                           \n",
       "47668                                                                           \n",
       "47668                                                                           \n",
       "47668                                                                           \n",
       "47668                                                                           \n",
       "\n",
       "      NOC DSS SYN KGPhase3 CAF      VC MUT KGPhase1 NOV   \n",
       "0                               VC=SNV                   \\\n",
       "0                               VC=SNV                    \n",
       "0                               VC=SNV                    \n",
       "0                               VC=SNV                    \n",
       "0                               VC=SNV                    \n",
       "...    ..  ..  ..      ...  ..     ...  ..      ...  ..   \n",
       "47668                                                     \n",
       "47668                                                     \n",
       "47668                                                     \n",
       "47668                                                     \n",
       "47668                                                     \n",
       "\n",
       "                                  VP    SAO   \n",
       "0      VP=0x050000040005000002000100  SAO=0  \\\n",
       "0      VP=0x050000040005000002000100  SAO=0   \n",
       "0      VP=0x050000040005000002000100  SAO=0   \n",
       "0      VP=0x050000040005000002000100  SAO=0   \n",
       "0      VP=0x050000040005000002000100  SAO=0   \n",
       "...                              ...    ...   \n",
       "47668                                         \n",
       "47668                                         \n",
       "47668                                         \n",
       "47668                                         \n",
       "47668                                         \n",
       "\n",
       "                                       GENEINFO INT G5 OM PMC    SSR   \n",
       "0      GENEINFO=DDX11L1:100287102|WASH7P:653635                SSR=0  \\\n",
       "0      GENEINFO=DDX11L1:100287102|WASH7P:653635                SSR=0   \n",
       "0      GENEINFO=DDX11L1:100287102|WASH7P:653635                SSR=0   \n",
       "0      GENEINFO=DDX11L1:100287102|WASH7P:653635                SSR=0   \n",
       "0      GENEINFO=DDX11L1:100287102|WASH7P:653635                SSR=0   \n",
       "...                                         ...  .. .. ..  ..    ...   \n",
       "47668                                                                  \n",
       "47668                                                                  \n",
       "47668                                                                  \n",
       "47668                                                                  \n",
       "47668                                                                  \n",
       "\n",
       "             RSPOS HD PM Protein Position and Amino Acid  \n",
       "0      RSPOS=14752                                   NaN  \n",
       "0      RSPOS=14752                                   NaN  \n",
       "0      RSPOS=14752                                   NaN  \n",
       "0      RSPOS=14752                                   NaN  \n",
       "0      RSPOS=14752                                   NaN  \n",
       "...            ... .. ..                             ...  \n",
       "47668                                                NaN  \n",
       "47668                                                NaN  \n",
       "47668                                                NaN  \n",
       "47668                                                NaN  \n",
       "47668                                                NaN  \n",
       "\n",
       "[371550 rows x 160 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1['Protein Position and Amino Acid'] = vcf_1['Amino_acids'].str[0] + vcf_1['Protein_position'] + np.where(vcf_1['Amino_acids'].str[-1] == vcf_1['Amino_acids'].str[0], '', vcf_1['Amino_acids'].str[-1])\n",
    "vcf_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>Raw Read Depth as reported by SAMtools</th>\n",
       "      <th>Quality Read Depth of bases with Phred score &gt;= 15</th>\n",
       "      <th>Depth of reference-supporting bases (reads1)</th>\n",
       "      <th>Depth of variant-supporting bases (reads2)</th>\n",
       "      <th>Depth of reference-supporting bases on forward strand (reads1plus)</th>\n",
       "      <th>Depth of reference-supporting bases on reverse strand (reads1minus)</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>New_Column</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE_SELECT</th>\n",
       "      <th>MANE_PLUS_CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT_ISOFORM</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18541</th>\n",
       "      <td>chr17</td>\n",
       "      <td>21007891</td>\n",
       "      <td>rs11541311</td>\n",
       "      <td>REF</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8442,.,...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8442,.,...</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|synonymous_variant|LOW|USP22|ENSG00000124422...</td>\n",
       "      <td>T|synonymous_variant|LOW|USP22|ENSG00000124422...</td>\n",
       "      <td>T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>USP22</td>\n",
       "      <td>ENSG00000124422</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000537526</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000537526.6:c.1173C&gt;A</td>\n",
       "      <td>ENSP00000440950.2:p.Ile391%3D</td>\n",
       "      <td>1173</td>\n",
       "      <td>1173</td>\n",
       "      <td>391</td>\n",
       "      <td>I</td>\n",
       "      <td>atC/atA</td>\n",
       "      <td>rs11541311&amp;COSV54945859</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12621</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENSP00000440950</td>\n",
       "      <td>Q9UPT9.179</td>\n",
       "      <td></td>\n",
       "      <td>UPI00016397EF</td>\n",
       "      <td>Q9UPT9-2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Gene3D:3.90.70.10&amp;Pfam:PF00443&amp;PROSITE_profile...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1558</td>\n",
       "      <td>0.0159</td>\n",
       "      <td>0.1326</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.2465</td>\n",
       "      <td>0.2566</td>\n",
       "      <td>0.1974</td>\n",
       "      <td>0.04178</td>\n",
       "      <td>0.1105</td>\n",
       "      <td>0.176</td>\n",
       "      <td>0.1724</td>\n",
       "      <td>0.3131</td>\n",
       "      <td>0.2181</td>\n",
       "      <td>0.1977</td>\n",
       "      <td>0.2381</td>\n",
       "      <td>0.1705</td>\n",
       "      <td>0.04537</td>\n",
       "      <td>0.2303</td>\n",
       "      <td>0.1462</td>\n",
       "      <td>0.1723</td>\n",
       "      <td>0.1723</td>\n",
       "      <td>0.3116</td>\n",
       "      <td>0.1487</td>\n",
       "      <td>0.224</td>\n",
       "      <td>0.1644</td>\n",
       "      <td>0.2474</td>\n",
       "      <td>0.3131</td>\n",
       "      <td>gnomADe_FIN</td>\n",
       "      <td></td>\n",
       "      <td>0&amp;1</td>\n",
       "      <td>0&amp;1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=11</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>S3D</td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=120</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NSM</td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=11541311</td>\n",
       "      <td>RV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.84234040519877675,0.00013538481141692...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SYN</td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.8442,.,.,0.1558</td>\n",
       "      <td>VC=SNV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050300000b05150536000100</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=USP22:23326</td>\n",
       "      <td></td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=21007891</td>\n",
       "      <td>HD</td>\n",
       "      <td></td>\n",
       "      <td>I391</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS          ID  REF ALT QUAL FILTER   \n",
       "18541  chr17  21007891  rs11541311  REF   T    .   PASS  \\\n",
       "\n",
       "                                                    INFO Genotype   \n",
       "18541  ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8442,.,...      1/1  \\\n",
       "\n",
       "       Genotype Quality  Raw Read Depth as reported by SAMtools   \n",
       "18541                58                                      11  \\\n",
       "\n",
       "       Quality Read Depth of bases with Phred score >= 15   \n",
       "18541                                                 11   \\\n",
       "\n",
       "       Depth of reference-supporting bases (reads1)   \n",
       "18541                                             0  \\\n",
       "\n",
       "       Depth of variant-supporting bases (reads2)   \n",
       "18541                                          11  \\\n",
       "\n",
       "      Depth of reference-supporting bases on forward strand (reads1plus)   \n",
       "18541                                               100%                  \\\n",
       "\n",
       "       Depth of reference-supporting bases on reverse strand (reads1minus)   \n",
       "18541                                           0.000001                    \\\n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "18541                                                  0                 \\\n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "18541                                                  0                  \\\n",
       "\n",
       "       Variant allele frequency  P-value from Fishers Exact Test   \n",
       "18541                         7                                4  \\\n",
       "\n",
       "                                              New_Column    HET    HOM   \n",
       "18541  ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8442,.,...  HET=0  HOM=1  \\\n",
       "\n",
       "         Zygosity                                                CSQ   \n",
       "18541  Homozygous  T|synonymous_variant|LOW|USP22|ENSG00000124422...  \\\n",
       "\n",
       "                                                     csq Allele   \n",
       "18541  T|synonymous_variant|LOW|USP22|ENSG00000124422...      T  \\\n",
       "\n",
       "              Consequence IMPACT SYMBOL             Gene Feature_type   \n",
       "18541  synonymous_variant    LOW  USP22  ENSG00000124422   Transcript  \\\n",
       "\n",
       "               Feature         BIOTYPE  EXON INTRON   \n",
       "18541  ENST00000537526  protein_coding  9/13         \\\n",
       "\n",
       "                             HGVSc                          HGVSp   \n",
       "18541  ENST00000537526.6:c.1173C>A  ENSP00000440950.2:p.Ile391%3D  \\\n",
       "\n",
       "      cDNA_position CDS_position Protein_position Amino_acids   Codons   \n",
       "18541          1173         1173              391           I  atC/atA  \\\n",
       "\n",
       "            Existing_variation DISTANCE STRAND FLAGS VARIANT_CLASS   \n",
       "18541  rs11541311&COSV54945859              -1                 SNV  \\\n",
       "\n",
       "      SYMBOL_SOURCE     HGNC_ID CANONICAL MANE_SELECT MANE_PLUS_CLINICAL TSL   \n",
       "18541          HGNC  HGNC:12621                                            1  \\\n",
       "\n",
       "      APPRIS CCDS             ENSP   SWISSPROT TREMBL        UNIPARC   \n",
       "18541              ENSP00000440950  Q9UPT9.179         UPI00016397EF  \\\n",
       "\n",
       "      UNIPROT_ISOFORM SOURCE GENE_PHENO SIFT PolyPhen   \n",
       "18541        Q9UPT9-2                                  \\\n",
       "\n",
       "                                                 DOMAINS miRNA HGVS_OFFSET   \n",
       "18541  Gene3D:3.90.70.10&Pfam:PF00443&PROSITE_profile...                    \\\n",
       "\n",
       "           AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF   \n",
       "18541  0.1558  0.0159  0.1326  0.1667  0.2465  0.2566     0.1974  \\\n",
       "\n",
       "      gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF   \n",
       "18541        0.04178         0.1105          0.176         0.1724  \\\n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF   \n",
       "18541         0.3131         0.2181         0.1977         0.2381     0.1705  \\\n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF   \n",
       "18541        0.04537         0.2303         0.1462         0.1723  \\\n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF   \n",
       "18541         0.1723         0.3116         0.1487          0.224  \\\n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS CLIN_SIG SOMATIC   \n",
       "18541         0.1644         0.2474  0.3131  gnomADe_FIN              0&1  \\\n",
       "\n",
       "      PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE   \n",
       "18541   0&1                                                              \\\n",
       "\n",
       "      TRANSCRIPTION_FACTORS ClinVar ClinVar_CLNSIG ClinVar_CLNREVSTAT   \n",
       "18541                                                                  \\\n",
       "\n",
       "      ClinVar_CLNDN     ADP    WT    NC CDA OTH  S3D WTD      dbSNPBuildID   \n",
       "18541                ADP=11  WT=0  NC=0          S3D      dbSNPBuildID=120  \\\n",
       "\n",
       "       SLO NSF R3 R5 NSN  NSM G5A    COMMON           RS  RV TPA CFL  GNO   \n",
       "18541  SLO                NSM      COMMON=1  RS=11541311  RV          GNO  \\\n",
       "\n",
       "       VLD  ASP ASS U3 U5                                             TOPMED   \n",
       "18541  VLD  ASP            TOPMED=0.84234040519877675,0.00013538481141692...  \\\n",
       "\n",
       "         WGT MTP LSD NOC DSS  SYN  KGPhase3                    CAF      VC   \n",
       "18541  WGT=1                  SYN  KGPhase3  CAF=0.8442,.,.,0.1558  VC=SNV  \\\n",
       "\n",
       "      MUT  KGPhase1 NOV                             VP    SAO   \n",
       "18541      KGPhase1      VP=0x050300000b05150536000100  SAO=0  \\\n",
       "\n",
       "                   GENEINFO INT  G5 OM PMC    SSR           RSPOS  HD PM   \n",
       "18541  GENEINFO=USP22:23326      G5         SSR=0  RSPOS=21007891  HD     \\\n",
       "\n",
       "      Protein Position and Amino Acid  \n",
       "18541                            I391  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1[vcf_1['Protein Position and Amino Acid'] == 'I391']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein Position and Amino Acid\n",
       "A28T     58\n",
       "L26V     58\n",
       "A15      57\n",
       "M13V     52\n",
       "S38      50\n",
       "         ..\n",
       "A84V      1\n",
       "A17P      1\n",
       "Y107      1\n",
       "I391      1\n",
       "S268G     1\n",
       "Name: count, Length: 27378, dtype: int64"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1['Protein Position and Amino Acid'].value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.01278,0.9872;COMMON=1;GENEINFO=CRELD1:78987|IL17RC:84818;GNO;KGPhase1;KGPhase3;R5;REF;RS=183956;RSPOS=9933563;RV;SAO=0;SLO;SSR=0;SYN;TOPMED=0.01415965851172273,0.98584034148827726;VC=SNV;VLD;VP=0x050100020305140136000100;WGT=1;dbSNPBuildID=79;CSQ=G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000295981|protein_coding|19/19||ENST00000295981.7:c.2346C>G|ENSP00000295981.3:p.Gly782%3D|2564|2346|782|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358||||2|A2|CCDS2590.1|ENSP00000295981|Q8NAC3.156||UPI000013E2E3|Q8NAC3-1||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite&AFDB-ENSP_mappings:AF-Q8NAC3-F1.A|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000326434|protein_coding||||||||||rs183956|293|1||SNV|HGNC|HGNC:14630||||1||CCDS33693.1|ENSP00000321856|Q96HD1.171||UPI000049DF8B|Q96HD1-2||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000383811|protein_coding||||||||||rs183956|271|1||SNV|HGNC|HGNC:14630||||1|P1|CCDS2593.1|ENSP00000373322|Q96HD1.171||UPI000013E2E5|Q96HD1-1||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000383812|protein_coding|18/18||ENST00000383812.9:c.2088C>G|ENSP00000373323.4:p.Gly696%3D|2330|2088|696|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358||||1||CCDS2591.2|ENSP00000373323|Q8NAC3.156||UPI00001A8668|Q8NAC3-3||1|||MobiDB_lite:mobidb-lite&Low_complexity_(Seg):seg|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000397170|protein_coding||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630||||5|||ENSP00000380355||A0A8C8KJP3.3|UPI001B89579C|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000403601|protein_coding|19/19||ENST00000403601.8:c.2133C>G|ENSP00000384969.3:p.Gly711%3D|2351|2133|711|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|YES|NM_153460.4||1|A2|CCDS46746.1|ENSP00000384969|Q8NAC3.156||UPI000173A2C3|Q8NAC3-2||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000412901|nonsense_mediated_decay||||||||||rs183956|330|1||SNV|HGNC|HGNC:18358||||2|||ENSP00000399396||B4DI24.88|UPI00017A7186|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000413608|protein_coding|18/18||ENST00000413608.2:c.2094C>G|ENSP00000396064.1:p.Gly698%3D|2094|2094|698|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358||||1|A2|CCDS56240.1|ENSP00000396064|Q8NAC3.156||UPI000198C927|Q8NAC3-5||1|||MobiDB_lite:mobidb-lite&Low_complexity_(Seg):seg|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000414117|protein_coding||||||||||rs183956|857|1|cds_end_NF|SNV|HGNC|HGNC:14630||||5|||ENSP00000411021||F8WBY3.51|UPI001B88E9B7|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000416074|protein_coding|17/17||ENST00000416074.6:c.2043C>G|ENSP00000395315.3:p.Gly681%3D|2285|2043|681|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358||||1||CCDS74898.1|ENSP00000395315|Q8NAC3.156||UPI0004E4CB92|Q8NAC3-7||1|||MobiDB_lite:mobidb-lite&Low_complexity_(Seg):seg|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000436503|protein_coding||||||||||rs183956|860|1|cds_end_NF|SNV|HGNC|HGNC:18358||||1|||ENSP00000401128||C9K0M1.70|UPI001FC3A156|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000438091|protein_coding|16/16||ENST00000438091.6:c.1977C>G|ENSP00000414609.2:p.Gly659%3D|2195|1977|659|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358||||5|||ENSP00000414609||C9JSZ3.69|UPI001FB1720B|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000451165|protein_coding|16/16||ENST00000451165.7:c.1929C>G|ENSP00000387675.3:p.Gly643%3D|2129|1929|643|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358||||2|||ENSP00000387675||F6X4Z5.18|UPI001FC248B7|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|3_prime_UTR_variant&NMD_transcript_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000451231|nonsense_mediated_decay|15/15||ENST00000451231.5:c.*1516C>G||2106|||||rs183956||1||SNV|HGNC|HGNC:18358||||2|||ENSP00000410834||F8WB43.53|UPI000198C928|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|3_prime_UTR_variant&NMD_transcript_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000451271|nonsense_mediated_decay|17/17||ENST00000451271.5:c.*1826C>G||2230|||||rs183956||1||SNV|HGNC|HGNC:18358||||1|||ENSP00000412777||F8WC09.42|UPI000198C924|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000452070|protein_coding||||||||||rs183956|271|1||SNV|HGNC|HGNC:14630|YES|NM_001077415.3||2|P1|CCDS2593.1|ENSP00000393643|Q96HD1.171||UPI000013E2E5|Q96HD1-1||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000455057|protein_coding|17/17||ENST00000455057.5:c.2037C>G|ENSP00000407894.1:p.Gly679%3D|2190|2037|679|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358||||1||CCDS56241.1|ENSP00000407894|Q8NAC3.156||UPI000198C93E|Q8NAC3-6||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000464406|retained_intron||||||||||rs183956|862|1||SNV|HGNC|HGNC:18358||||2|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000465716|retained_intron||||||||||rs183956|276|1||SNV|HGNC|HGNC:14630||||2|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000465794|retained_intron||||||||||rs183956|605|1||SNV|HGNC|HGNC:18358||||3|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|non_coding_transcript_exon_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000466046|retained_intron|17/17||ENST00000466046.5:n.2517C>G||2517|||||rs183956||1||SNV|HGNC|HGNC:18358||||2|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000466712|retained_intron||||||||||rs183956|860|1||SNV|HGNC|HGNC:18358||||3|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000467713|retained_intron||||||||||rs183956|271|1||SNV|HGNC|HGNC:14630||||2|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000469686|retained_intron||||||||||rs183956|170|1||SNV|HGNC|HGNC:18358||||1|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|3_prime_UTR_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000483582|protein_coding|17/17||ENST00000483582.5:c.*565C>G||2400|||||rs183956||1||SNV|HGNC|HGNC:18358||||1|||ENSP00000512844|||UPI0000141B34|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000489674|retained_intron||||||||||rs183956|4508|1||SNV|HGNC|HGNC:14630||||5|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000491527|protein_coding||||||||||rs183956|281|1||SNV|HGNC|HGNC:14630||||2|||ENSP00000508342||A0A804HLG6.4|UPI001B89C238|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|non_coding_transcript_exon_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000494365|retained_intron|16/16||ENST00000494365.5:n.2446C>G||2446|||||rs183956||1||SNV|HGNC|HGNC:18358||||2|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|non_coding_transcript_exon_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000497387|retained_intron|6/6||ENST00000497387.5:n.1808C>G||1808|||||rs183956||1||SNV|HGNC|HGNC:18358||||2|||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|3_prime_UTR_variant&NMD_transcript_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000498214|nonsense_mediated_decay|16/16||ENST00000498214.6:c.*1809C>G||2303|||||rs183956||1||SNV|HGNC|HGNC:18358||||2|||ENSP00000497013||A8K3Q5.93|UPI0000161642|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000602411|protein_coding_CDS_not_defined||||||||||rs183956|276|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000673635|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000501186||A0A669KB87.8|UPI00005795E9|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000673677|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000500985||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000673737|nonsense_mediated_decay||||||||||rs183956|262|1||SNV|HGNC|HGNC:14630|||||||ENSP00000501208||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000673935|protein_coding||||||||||rs183956|247|1||SNV|HGNC|HGNC:14630|||||||ENSP00000500961||A0A669KAV0.9|UPI00122F6BE2|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000674057|nonsense_mediated_decay||||||||||rs183956|272|1||SNV|HGNC|HGNC:14630|||||||ENSP00000501128||A0A669KB87.8|UPI00005795E9|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000674067|nonsense_mediated_decay||||||||||rs183956|281|1||SNV|HGNC|HGNC:14630|||||||ENSP00000501009||A0A669KAY9.5|UPI00004DEF49|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682120|retained_intron||||||||||rs183956|275|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682200|nonsense_mediated_decay||||||||||rs183956|292|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508321||A0A804HL19.4|UPI001B8938AB|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682318|nonsense_mediated_decay||||||||||rs183956|292|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507946||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682355|protein_coding_CDS_not_defined||||||||||rs183956|296|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682397|nonsense_mediated_decay||||||||||rs183956|320|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507177||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682403|nonsense_mediated_decay||||||||||rs183956|262|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507865||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682570|protein_coding||||||||||rs183956|300|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507973||A0A804HKL1.4|UPI001B88F283|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682642|protein_coding||||||||||rs183956|275|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507251||A0A804HKP7.4|UPI000387D83A|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682771|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507114||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682783|protein_coding||||||||||rs183956|294|1||SNV|HGNC|HGNC:14630|||||P1|CCDS2593.1|ENSP00000508358|Q96HD1.171||UPI000013E2E5|Q96HD1-1||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682790|nonsense_mediated_decay||||||||||rs183956|294|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507904||A0A669KB87.8|UPI00005795E9|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682798|retained_intron||||||||||rs183956|271|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682866|retained_intron||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682884|retained_intron||||||||||rs183956|382|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682906|nonsense_mediated_decay||||||||||rs183956|284|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507674||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682907|nonsense_mediated_decay||||||||||rs183956|282|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508209||A0A804HL57.4|UPI001B896A17|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000682929|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507930||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|intron_variant&non_coding_transcript_variant|MODIFIER||ENSG00000288550|Transcript|ENST00000683061|retained_intron||15/22|ENST00000683061.1:n.1849+860C>G|||||||rs183956||1||SNV|||||||||||||||||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683083|retained_intron||||||||||rs183956|269|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683122|protein_coding_CDS_not_defined||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683127|nonsense_mediated_decay||||||||||rs183956|276|1||SNV|HGNC|HGNC:14630|||||||ENSP00000506906||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683189|protein_coding||||||||||rs183956|271|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508018||A0A804HKP9.4|UPI0003EB03F8|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683232|nonsense_mediated_decay||||||||||rs183956|262|1||SNV|HGNC|HGNC:14630|||||||ENSP00000506842||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683239|retained_intron||||||||||rs183956|247|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683279|retained_intron||||||||||rs183956|230|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683301|nonsense_mediated_decay||||||||||rs183956|327|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507718||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683316|nonsense_mediated_decay||||||||||rs183956|267|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507076||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683388|retained_intron||||||||||rs183956|4723|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683507|protein_coding_CDS_not_defined||||||||||rs183956|513|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683547|nonsense_mediated_decay||||||||||rs183956|267|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507258||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683601|nonsense_mediated_decay||||||||||rs183956|313|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507737||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683603|nonsense_mediated_decay||||||||||rs183956|275|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508016||A0A804HKP7.4|UPI000387D83A|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683736|protein_coding||||||||||rs183956|857|1|cds_end_NF|SNV|HGNC|HGNC:14630|||||||ENSP00000507761||A0A804HK40.4|UPI001B896128|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683835|protein_coding||||||||||rs183956|294|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508148||A0A804HL07.4|UPI001B892DCB|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683849|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000506901||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683888|retained_intron||||||||||rs183956|271|1||SNV|HGNC|HGNC:14630|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683891|nonsense_mediated_decay||||||||||rs183956|287|1||SNV|HGNC|HGNC:14630|||||||ENSP00000506972||A0A804HL19.4|UPI001B8938AB|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683921|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507395||A0A804HJ85.4|UPI001B88FA94|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683970|nonsense_mediated_decay||||||||||rs183956|267|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507906||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000683982|protein_coding||||||||||rs183956|765|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507437||A0A804HJB9.4|UPI001B89934B|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684181|protein_coding||||||||||rs183956|271|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507522||A0A804HJJ0.4|UPI0005D02EE8|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684212|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507565||A0A804HJM5.4|UPI001B898FB6|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684243|nonsense_mediated_decay||||||||||rs183956|292|1||SNV|HGNC|HGNC:14630|||||||ENSP00000506959||A0A669KB87.8|UPI00005795E9|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684291|nonsense_mediated_decay||||||||||rs183956|276|1||SNV|HGNC|HGNC:14630|||||||ENSP00000506927||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684318|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508180||A0A804HL34.4|UPI001B8991EB|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684493|protein_coding||||||||||rs183956|242|1||SNV|HGNC|HGNC:14630|||||P1|CCDS2593.1|ENSP00000507127|Q96HD1.171||UPI000013E2E5|Q96HD1-1||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684526|nonsense_mediated_decay||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508163||A0A804HL19.4|UPI001B8938AB|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684532|protein_coding||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508212||A0A804HL60.4|UPI001B89D350|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684601|protein_coding||||||||||rs183956|857|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507886||A0A804HKD9.4|UPI001B898978|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684613|nonsense_mediated_decay||||||||||rs183956|305|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508218||A0A669KBB3.9|UPI000387D7A3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684629|nonsense_mediated_decay||||||||||rs183956|292|1||SNV|HGNC|HGNC:14630|||||||ENSP00000506879||A0A804HI35.4|UPI000198C942|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684648|nonsense_mediated_decay||||||||||rs183956|276|1||SNV|HGNC|HGNC:14630|||||||ENSP00000507931||A0A804HL19.4|UPI001B8938AB|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|upstream_gene_variant|MODIFIER|CRELD1|ENSG00000163703|Transcript|ENST00000684659|protein_coding||||||||||rs183956|276|1||SNV|HGNC|HGNC:14630|||||||ENSP00000508386||A0A804HLG6.4|UPI001B89C238|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696815|protein_coding||||||||||rs183956|860|1|cds_end_NF|SNV|HGNC|HGNC:18358|||||||ENSP00000512896|||UPI001FC37E3B|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000696816|protein_coding|17/17||ENST00000696816.1:c.2049C>G|ENSP00000512897.1:p.Gly683%3D|2251|2049|683|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512897|||UPI00076F1BA8|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696817|retained_intron||||||||||rs183956|483|1||SNV|HGNC|HGNC:18358|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696818|nonsense_mediated_decay||||||||||rs183956|170|1||SNV|HGNC|HGNC:18358|||||||ENSP00000512898||Q8N2D7.60|UPI000003B0F3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696819|nonsense_mediated_decay||||||||||rs183956|170|1||SNV|HGNC|HGNC:18358|||||||ENSP00000512899||Q8N2D7.60|UPI000003B0F3|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696820|retained_intron||||||||||rs183956|170|1||SNV|HGNC|HGNC:18358|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696821|retained_intron||||||||||rs183956|170|1||SNV|HGNC|HGNC:18358|||||||||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696822|nonsense_mediated_decay||||||||||rs183956|160|1||SNV|HGNC|HGNC:18358|||||||ENSP00000512900|||UPI001FC26FA2|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000696823|protein_coding|16/16||ENST00000696823.1:c.2001C>G|ENSP00000512901.1:p.Gly667%3D|2201|2001|667|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512901|||UPI001FC30D11|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000696824|protein_coding|18/18||ENST00000696824.1:c.2055C>G|ENSP00000512902.1:p.Gly685%3D|2255|2055|685|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512902|||UPI0007DC7B3C|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000696825|protein_coding|17/17||ENST00000696825.1:c.2046C>G|ENSP00000512903.1:p.Gly682%3D|2246|2046|682|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512903|||UPI001FC2CD07|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000696826|protein_coding|16/16||ENST00000696826.1:c.1998C>G|ENSP00000512904.1:p.Gly666%3D|2198|1998|666|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512904|||UPI001FB0DF6D|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|3_prime_UTR_variant&NMD_transcript_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696827|nonsense_mediated_decay|16/16||ENST00000696827.1:c.*742C>G||2127|||||rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512905|||UPI001FC37CD1|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000696828|protein_coding|17/17||ENST00000696828.1:c.2034C>G|ENSP00000512906.1:p.Gly678%3D|2234|2034|678|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512906|||UPI001FC23E36|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|3_prime_UTR_variant&NMD_transcript_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696829|nonsense_mediated_decay|15/15||ENST00000696829.1:c.*1487C>G||1983|||||rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512907|||UPI000198C925|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696830|protein_coding||||||||||rs183956|14|1||SNV|HGNC|HGNC:18358|||||||ENSP00000512908|||UPI001FC2ED8A|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|synonymous_variant|LOW|IL17RC|ENSG00000163702|Transcript|ENST00000696832|protein_coding|15/15||ENST00000696832.1:c.1917C>G|ENSP00000512910.1:p.Gly639%3D|2056|1917|639|G|ggC/ggG|rs183956||1||SNV|HGNC|HGNC:18358|||||||ENSP00000512910|||UPI001FC311B8|||1|||Low_complexity_(Seg):seg&MobiDB_lite:mobidb-lite|||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696833|nonsense_mediated_decay||||||||||rs183956|859|1|cds_start_NF|SNV|HGNC|HGNC:18358|||||||ENSP00000512911|||UPI001FC22ABF|||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|downstream_gene_variant|MODIFIER|IL17RC|ENSG00000163702|Transcript|ENST00000696834|protein_coding||||||||||rs183956|880|1|cds_start_NF&cds_end_NF|SNV|HGNC|HGNC:18358|||||||ENSP00000512912||||||1||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000676867|promoter||||||||||rs183956||||SNV|||||||||||||||||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1|||||||1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000212702|||||||||||rs183956||1||SNV|||||||||||||||||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1||ENSPFM0239|20|Y|-0.004|GCM1::MAX|1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided,G|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00525202403|||||||||||rs183956||1||SNV|||||||||||||||||||||0.9872|0.9728|0.9942|0.999|0.9831|0.9939|0.9894|0.9779|0.9915|0.9899|0.9969|0.9979|0.9877|0.9916|0.9874|0.9859|0.977|1|0.9884|0.9885|0.9975|0.998|0.9842|0.9875|0.9842|0.9897|1|gnomADg_AMI|benign||1||ENSPFM0068|3|N|-0.041|E2F1::ELK1|1164864|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Candidiasis&_familial&_9&not_provided'"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_1[vcf_1['ID'] == 'rs183956'].INFO.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_1.to_csv(r'C:/Users/Dell/Downloads/VCF_all_columns_split.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO',\n",
      "       'Genotype', 'Genotype Quality',\n",
      "       ...\n",
      "       'SAO', 'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD',\n",
      "       'PM'],\n",
      "      dtype='object', length=159)\n"
     ]
    }
   ],
   "source": [
    "print(vcf_1.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = vcf[vcf['ID'] == 'rs6672356']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "x.to_excel(r'C:/Users/Dell/Downloads/rs6672356.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...   \n",
       "0                    22  ...  \\\n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "...                 ...  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "...                                                  ...                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "...                                                  ...                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "\n",
       "       Variant allele frequency  P-value from Fishers Exact Test HET  HOM   \n",
       "0                             7                                0   1    0  \\\n",
       "0                             7                                0   1    0   \n",
       "0                             7                                0   1    0   \n",
       "0                             7                                0   1    0   \n",
       "0                             7                                0   1    0   \n",
       "...                         ...                              ...  ..  ...   \n",
       "47668                         6                                1   0    1   \n",
       "47668                         6                                1   0    1   \n",
       "47668                         6                                1   0    1   \n",
       "47668                         6                                1   0    1   \n",
       "47668                         6                                1   0    1   \n",
       "\n",
       "           Zygosity                                                CSQ   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...             ...                                                ...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...   \n",
       "0      A|intron_variant&non_coding_transcript_variant...   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "...                                                  ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...   \n",
       "\n",
       "                                        Consequence  \n",
       "0                           downstream_gene_variant  \n",
       "0                           downstream_gene_variant  \n",
       "0      intron_variant&non_coding_transcript_variant  \n",
       "0                           downstream_gene_variant  \n",
       "0                         regulatory_region_variant  \n",
       "...                                             ...  \n",
       "47668            non_coding_transcript_exon_variant  \n",
       "47668            non_coding_transcript_exon_variant  \n",
       "47668                       downstream_gene_variant  \n",
       "47668  intron_variant&non_coding_transcript_variant  \n",
       "47668                       downstream_gene_variant  \n",
       "\n",
       "[371550 rows x 26 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Consequence'] = vcf['csq'].str.extract(r'\\|([^|]+)\\|')\n",
    "vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.Consequence.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...   \n",
       "0                    22  ...  \\\n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "...                 ...  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "\n",
       "       Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                  \\\n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "0                                                      0                   \n",
       "...                                                  ...                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "47668                                                  1                   \n",
       "\n",
       "       Variant allele frequency  P-value from Fishers Exact Test  HET HOM   \n",
       "0                             7                                0    1   0  \\\n",
       "0                             7                                0    1   0   \n",
       "0                             7                                0    1   0   \n",
       "0                             7                                0    1   0   \n",
       "0                             7                                0    1   0   \n",
       "...                         ...                              ...  ...  ..   \n",
       "47668                         6                                1    0   1   \n",
       "47668                         6                                1    0   1   \n",
       "47668                         6                                1    0   1   \n",
       "47668                         6                                1    0   1   \n",
       "47668                         6                                1    0   1   \n",
       "\n",
       "           Zygosity                                                CSQ   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...             ...                                                ...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...   \n",
       "0      A|intron_variant&non_coding_transcript_variant...   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "...                                                  ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...   \n",
       "\n",
       "                                        Consequence    Impact  \n",
       "0                           downstream_gene_variant  MODIFIER  \n",
       "0                           downstream_gene_variant  MODIFIER  \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER  \n",
       "0                           downstream_gene_variant  MODIFIER  \n",
       "0                         regulatory_region_variant  MODIFIER  \n",
       "...                                             ...       ...  \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER  \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER  \n",
       "47668                       downstream_gene_variant  MODIFIER  \n",
       "47668  intron_variant&non_coding_transcript_variant  MODIFIER  \n",
       "47668                       downstream_gene_variant  MODIFIER  \n",
       "\n",
       "[371550 rows x 27 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Impact'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  Variant allele frequency   \n",
       "0                    22  ...                         7  \\\n",
       "0                    22  ...                         7   \n",
       "0                    22  ...                         7   \n",
       "0                    22  ...                         7   \n",
       "0                    22  ...                         7   \n",
       "...                 ...  ...                       ...   \n",
       "47668                31  ...                         6   \n",
       "47668                31  ...                         6   \n",
       "47668                31  ...                         6   \n",
       "47668                31  ...                         6   \n",
       "47668                31  ...                         6   \n",
       "\n",
       "       P-value from Fishers Exact Test  HET  HOM      Zygosity   \n",
       "0                                    0    1    0  Heterozygous  \\\n",
       "0                                    0    1    0  Heterozygous   \n",
       "0                                    0    1    0  Heterozygous   \n",
       "0                                    0    1    0  Heterozygous   \n",
       "0                                    0    1    0  Heterozygous   \n",
       "...                                ...  ...  ...           ...   \n",
       "47668                                1    0    1    Homozygous   \n",
       "47668                                1    0    1    Homozygous   \n",
       "47668                                1    0    1    Homozygous   \n",
       "47668                                1    0    1    Homozygous   \n",
       "47668                                1    0    1    Homozygous   \n",
       "\n",
       "                                                     CSQ   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...   \n",
       "0      A|intron_variant&non_coding_transcript_variant...   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "...                                                  ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...   \n",
       "\n",
       "                                        Consequence    Impact  Gene_Name  \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L1  \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L2  \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P  \n",
       "0                           downstream_gene_variant  MODIFIER  MIR6859-1  \n",
       "0                         regulatory_region_variant  MODIFIER             \n",
       "...                                             ...       ...        ...  \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P  \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P  \n",
       "47668                       downstream_gene_variant  MODIFIER     WASH6P  \n",
       "47668  intron_variant&non_coding_transcript_variant  MODIFIER     WASH6P  \n",
       "47668                       downstream_gene_variant  MODIFIER   DDX11L16  \n",
       "\n",
       "[371550 rows x 28 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Gene_Name'] = vcf['csq'].str.split('|').str[3]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  P-value from Fishers Exact Test  HET  HOM   \n",
       "0                    22  ...                                0    1    0  \\\n",
       "0                    22  ...                                0    1    0   \n",
       "0                    22  ...                                0    1    0   \n",
       "0                    22  ...                                0    1    0   \n",
       "0                    22  ...                                0    1    0   \n",
       "...                 ...  ...                              ...  ...  ...   \n",
       "47668                31  ...                                1    0    1   \n",
       "47668                31  ...                                1    0    1   \n",
       "47668                31  ...                                1    0    1   \n",
       "47668                31  ...                                1    0    1   \n",
       "47668                31  ...                                1    0    1   \n",
       "\n",
       "           Zygosity                                                CSQ   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      Heterozygous  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...             ...                                                ...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668    Homozygous  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...   \n",
       "0      A|intron_variant&non_coding_transcript_variant...   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "...                                                  ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...   \n",
       "\n",
       "                                        Consequence    Impact  Gene_Name   \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L1  \\\n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L2   \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                           downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                         regulatory_region_variant  MODIFIER              \n",
       "...                                             ...       ...        ...   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER     WASH6P   \n",
       "47668  intron_variant&non_coding_transcript_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER   DDX11L16   \n",
       "\n",
       "                                  Biotype  \n",
       "0      transcribed_unprocessed_pseudogene  \n",
       "0                                  lncRNA  \n",
       "0                  unprocessed_pseudogene  \n",
       "0                                   miRNA  \n",
       "0                                enhancer  \n",
       "...                                   ...  \n",
       "47668                     retained_intron  \n",
       "47668                     retained_intron  \n",
       "47668                     retained_intron  \n",
       "47668                     retained_intron  \n",
       "47668              unprocessed_pseudogene  \n",
       "\n",
       "[371550 rows x 29 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Biotype'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  HET  HOM      Zygosity   \n",
       "0                    22  ...    1    0  Heterozygous  \\\n",
       "0                    22  ...    1    0  Heterozygous   \n",
       "0                    22  ...    1    0  Heterozygous   \n",
       "0                    22  ...    1    0  Heterozygous   \n",
       "0                    22  ...    1    0  Heterozygous   \n",
       "...                 ...  ...  ...  ...           ...   \n",
       "47668                31  ...    0    1    Homozygous   \n",
       "47668                31  ...    0    1    Homozygous   \n",
       "47668                31  ...    0    1    Homozygous   \n",
       "47668                31  ...    0    1    Homozygous   \n",
       "47668                31  ...    0    1    Homozygous   \n",
       "\n",
       "                                                     CSQ   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...   \n",
       "0      A|intron_variant&non_coding_transcript_variant...   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "...                                                  ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...   \n",
       "\n",
       "                                        Consequence    Impact  Gene_Name   \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L1  \\\n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L2   \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                           downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                         regulatory_region_variant  MODIFIER              \n",
       "...                                             ...       ...        ...   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER     WASH6P   \n",
       "47668  intron_variant&non_coding_transcript_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER   DDX11L16   \n",
       "\n",
       "                                  Biotype  Strand  \n",
       "0      transcribed_unprocessed_pseudogene       1  \n",
       "0                                  lncRNA       1  \n",
       "0                  unprocessed_pseudogene      -1  \n",
       "0                                   miRNA      -1  \n",
       "0                                enhancer          \n",
       "...                                   ...     ...  \n",
       "47668                     retained_intron       1  \n",
       "47668                     retained_intron       1  \n",
       "47668                     retained_intron       1  \n",
       "47668                     retained_intron       1  \n",
       "47668              unprocessed_pseudogene      -1  \n",
       "\n",
       "[371550 rows x 30 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Strand'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Clinical Significance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  HOM      Zygosity   \n",
       "0                    22  ...    0  Heterozygous  \\\n",
       "0                    22  ...    0  Heterozygous   \n",
       "0                    22  ...    0  Heterozygous   \n",
       "0                    22  ...    0  Heterozygous   \n",
       "0                    22  ...    0  Heterozygous   \n",
       "...                 ...  ...  ...           ...   \n",
       "47668                31  ...    1    Homozygous   \n",
       "47668                31  ...    1    Homozygous   \n",
       "47668                31  ...    1    Homozygous   \n",
       "47668                31  ...    1    Homozygous   \n",
       "47668                31  ...    1    Homozygous   \n",
       "\n",
       "                                                     CSQ   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...   \n",
       "0      A|intron_variant&non_coding_transcript_variant...   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "...                                                  ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...   \n",
       "\n",
       "                                        Consequence    Impact  Gene_Name   \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L1  \\\n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L2   \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                           downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                         regulatory_region_variant  MODIFIER              \n",
       "...                                             ...       ...        ...   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER     WASH6P   \n",
       "47668  intron_variant&non_coding_transcript_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER   DDX11L16   \n",
       "\n",
       "                                  Biotype  Strand  Clinical Significance  \n",
       "0      transcribed_unprocessed_pseudogene       1                         \n",
       "0                                  lncRNA       1                         \n",
       "0                  unprocessed_pseudogene      -1                         \n",
       "0                                   miRNA      -1                         \n",
       "0                                enhancer                                 \n",
       "...                                   ...     ...                    ...  \n",
       "47668                     retained_intron       1                         \n",
       "47668                     retained_intron       1                         \n",
       "47668                     retained_intron       1                         \n",
       "47668                     retained_intron       1                         \n",
       "47668              unprocessed_pseudogene      -1                         \n",
       "\n",
       "[371550 rows x 31 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Clinical Significance'] = vcf['csq'].str.split('|').str[80]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Clinical Significance\n",
       "                                                            261106\n",
       "Benign                                                      101428\n",
       "Benign/Likely_benign                                          4386\n",
       "Likely_benign                                                 1806\n",
       "Uncertain_significance                                        1110\n",
       "Conflicting_interpretations_of_pathogenicity                   822\n",
       "risk_factor                                                    216\n",
       "association                                                    216\n",
       "drug_response                                                  180\n",
       "not_provided                                                    60\n",
       "Uncertain_significance&drug_response                            28\n",
       "protective                                                      19\n",
       "Conflicting_interpretations_of_pathogenicity&other              19\n",
       "Benign&drug_response                                            18\n",
       "Benign&association                                              17\n",
       "Pathogenic&protective                                           16\n",
       "Conflicting_interpretations_of_pathogenicity&risk_factor        16\n",
       "Affects                                                         13\n",
       "Pathogenic&risk_factor                                          12\n",
       "confers_sensitivity                                             12\n",
       "Uncertain_risk_allele&protective                                11\n",
       "Benign&other                                                     9\n",
       "Likely_pathogenic                                                8\n",
       "Benign&risk_factor                                               6\n",
       "Conflicting_interpretations_of_pathogenicity&protective          5\n",
       "Uncertain_significance&association                               4\n",
       "protective&risk_factor                                           4\n",
       "association&drug_response                                        2\n",
       "Likely_risk_allele                                               1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Clinical Significance'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>csq</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Clinical Significance</th>\n",
       "      <th>Codons_position</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000488147.1:n.1254-251C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.813C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.1157C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.94+36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...   \n",
       "0                    22  ...  \\\n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "0                    22  ...   \n",
       "...                 ...  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "47668                31  ...   \n",
       "\n",
       "                                                     csq   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...   \n",
       "0      A|intron_variant&non_coding_transcript_variant...   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "...                                                  ...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|WASH6P|ENSG...   \n",
       "47668  T|intron_variant&non_coding_transcript_variant...   \n",
       "47668  T|downstream_gene_variant|MODIFIER|DDX11L16|EN...   \n",
       "\n",
       "                                        Consequence    Impact  Gene_Name   \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L1  \\\n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L2   \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                           downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                         regulatory_region_variant  MODIFIER              \n",
       "...                                             ...       ...        ...   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER     WASH6P   \n",
       "47668  intron_variant&non_coding_transcript_variant  MODIFIER     WASH6P   \n",
       "47668                       downstream_gene_variant  MODIFIER   DDX11L16   \n",
       "\n",
       "                                  Biotype  Strand  Clinical Significance   \n",
       "0      transcribed_unprocessed_pseudogene       1                         \\\n",
       "0                                  lncRNA       1                          \n",
       "0                  unprocessed_pseudogene      -1                          \n",
       "0                                   miRNA      -1                          \n",
       "0                                enhancer                                  \n",
       "...                                   ...     ...                    ...   \n",
       "47668                     retained_intron       1                          \n",
       "47668                     retained_intron       1                          \n",
       "47668                     retained_intron       1                          \n",
       "47668                     retained_intron       1                          \n",
       "47668              unprocessed_pseudogene      -1                          \n",
       "\n",
       "                       Codons_position  Protein  Codon  \n",
       "0                                                       \n",
       "0                                                       \n",
       "0      ENST00000488147.1:n.1254-251C>T                  \n",
       "0                                                       \n",
       "0                                                       \n",
       "...                                ...      ...    ...  \n",
       "47668       ENST00000484415.6:n.813C>T                  \n",
       "47668      ENST00000492963.6:n.1157C>T                  \n",
       "47668                                                   \n",
       "47668     ENST00000496301.6:n.94+36C>T                  \n",
       "47668                                                   \n",
       "\n",
       "[371550 rows x 34 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Codons_position'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['Protein'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['Codon'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Clinical Significance</th>\n",
       "      <th>Codons_position</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "      <th>Clinvar ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000488147.1:n.1254-251C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.813C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.1157C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.94+36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...                                   Consequence   \n",
       "0                    22  ...                       downstream_gene_variant  \\\n",
       "0                    22  ...                       downstream_gene_variant   \n",
       "0                    22  ...  intron_variant&non_coding_transcript_variant   \n",
       "0                    22  ...                       downstream_gene_variant   \n",
       "0                    22  ...                     regulatory_region_variant   \n",
       "...                 ...  ...                                           ...   \n",
       "47668                31  ...            non_coding_transcript_exon_variant   \n",
       "47668                31  ...            non_coding_transcript_exon_variant   \n",
       "47668                31  ...                       downstream_gene_variant   \n",
       "47668                31  ...  intron_variant&non_coding_transcript_variant   \n",
       "47668                31  ...                       downstream_gene_variant   \n",
       "\n",
       "         Impact  Gene_Name                             Biotype Strand   \n",
       "0      MODIFIER    DDX11L1  transcribed_unprocessed_pseudogene      1  \\\n",
       "0      MODIFIER    DDX11L2                              lncRNA      1   \n",
       "0      MODIFIER     WASH7P              unprocessed_pseudogene     -1   \n",
       "0      MODIFIER  MIR6859-1                               miRNA     -1   \n",
       "0      MODIFIER                                       enhancer          \n",
       "...         ...        ...                                 ...    ...   \n",
       "47668  MODIFIER     WASH6P                     retained_intron      1   \n",
       "47668  MODIFIER     WASH6P                     retained_intron      1   \n",
       "47668  MODIFIER     WASH6P                     retained_intron      1   \n",
       "47668  MODIFIER     WASH6P                     retained_intron      1   \n",
       "47668  MODIFIER   DDX11L16              unprocessed_pseudogene     -1   \n",
       "\n",
       "       Clinical Significance                  Codons_position  Protein  Codon   \n",
       "0                                                                              \\\n",
       "0                                                                               \n",
       "0                             ENST00000488147.1:n.1254-251C>T                   \n",
       "0                                                                               \n",
       "0                                                                               \n",
       "...                      ...                              ...      ...    ...   \n",
       "47668                              ENST00000484415.6:n.813C>T                   \n",
       "47668                             ENST00000492963.6:n.1157C>T                   \n",
       "47668                                                                           \n",
       "47668                            ENST00000496301.6:n.94+36C>T                   \n",
       "47668                                                                           \n",
       "\n",
       "       Clinvar ID  \n",
       "0                  \n",
       "0                  \n",
       "0                  \n",
       "0                  \n",
       "0                  \n",
       "...           ...  \n",
       "47668              \n",
       "47668              \n",
       "47668              \n",
       "47668              \n",
       "47668              \n",
       "\n",
       "[371550 rows x 35 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Clinvar ID'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Clinical Significance</th>\n",
       "      <th>Codons_position</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "      <th>Clinvar ID</th>\n",
       "      <th>Exon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000488147.1:n.1254-251C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.813C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.1157C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.94+36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...    Impact  Gene_Name   \n",
       "0                    22  ...  MODIFIER    DDX11L1  \\\n",
       "0                    22  ...  MODIFIER    DDX11L2   \n",
       "0                    22  ...  MODIFIER     WASH7P   \n",
       "0                    22  ...  MODIFIER  MIR6859-1   \n",
       "0                    22  ...  MODIFIER              \n",
       "...                 ...  ...       ...        ...   \n",
       "47668                31  ...  MODIFIER     WASH6P   \n",
       "47668                31  ...  MODIFIER     WASH6P   \n",
       "47668                31  ...  MODIFIER     WASH6P   \n",
       "47668                31  ...  MODIFIER     WASH6P   \n",
       "47668                31  ...  MODIFIER   DDX11L16   \n",
       "\n",
       "                                  Biotype  Strand Clinical Significance   \n",
       "0      transcribed_unprocessed_pseudogene       1                        \\\n",
       "0                                  lncRNA       1                         \n",
       "0                  unprocessed_pseudogene      -1                         \n",
       "0                                   miRNA      -1                         \n",
       "0                                enhancer                                 \n",
       "...                                   ...     ...                   ...   \n",
       "47668                     retained_intron       1                         \n",
       "47668                     retained_intron       1                         \n",
       "47668                     retained_intron       1                         \n",
       "47668                     retained_intron       1                         \n",
       "47668              unprocessed_pseudogene      -1                         \n",
       "\n",
       "                       Codons_position  Protein  Codon  Clinvar ID  Exon  \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0      ENST00000488147.1:n.1254-251C>T                                    \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "...                                ...      ...    ...         ...   ...  \n",
       "47668       ENST00000484415.6:n.813C>T                               5/5  \n",
       "47668      ENST00000492963.6:n.1157C>T                               5/6  \n",
       "47668                                                                     \n",
       "47668     ENST00000496301.6:n.94+36C>T                                    \n",
       "47668                                                                     \n",
       "\n",
       "[371550 rows x 36 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Exon'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Clinical Significance</th>\n",
       "      <th>Codons_position</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "      <th>Clinvar ID</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Existing variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000488147.1:n.1254-251C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.813C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.1157C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/6</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.94+36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  Gene_Name                             Biotype   \n",
       "0                    22  ...    DDX11L1  transcribed_unprocessed_pseudogene  \\\n",
       "0                    22  ...    DDX11L2                              lncRNA   \n",
       "0                    22  ...     WASH7P              unprocessed_pseudogene   \n",
       "0                    22  ...  MIR6859-1                               miRNA   \n",
       "0                    22  ...                                       enhancer   \n",
       "...                 ...  ...        ...                                 ...   \n",
       "47668                31  ...     WASH6P                     retained_intron   \n",
       "47668                31  ...     WASH6P                     retained_intron   \n",
       "47668                31  ...     WASH6P                     retained_intron   \n",
       "47668                31  ...     WASH6P                     retained_intron   \n",
       "47668                31  ...   DDX11L16              unprocessed_pseudogene   \n",
       "\n",
       "       Strand  Clinical Significance                  Codons_position   \n",
       "0           1                                                          \\\n",
       "0           1                                                           \n",
       "0          -1                         ENST00000488147.1:n.1254-251C>T   \n",
       "0          -1                                                           \n",
       "0                                                                       \n",
       "...       ...                    ...                              ...   \n",
       "47668       1                              ENST00000484415.6:n.813C>T   \n",
       "47668       1                             ENST00000492963.6:n.1157C>T   \n",
       "47668       1                                                           \n",
       "47668       1                            ENST00000496301.6:n.94+36C>T   \n",
       "47668      -1                                                           \n",
       "\n",
       "       Protein  Codon  Clinvar ID  Exon  Existing variant  \n",
       "0                                              rs17855866  \n",
       "0                                              rs17855866  \n",
       "0                                              rs17855866  \n",
       "0                                              rs17855866  \n",
       "0                                              rs17855866  \n",
       "...        ...    ...         ...   ...               ...  \n",
       "47668                               5/5                    \n",
       "47668                               5/6                    \n",
       "47668                                                      \n",
       "47668                                                      \n",
       "47668                                                      \n",
       "\n",
       "[371550 rows x 37 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Existing variant'] = vcf['csq'].str.split('|').str[17]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Clinical Significance</th>\n",
       "      <th>Codons_position</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "      <th>Clinvar ID</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Existing variant</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000488147.1:n.1254-251C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.813C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.1157C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.94+36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  Strand  Clinical Significance   \n",
       "0                    22  ...       1                         \\\n",
       "0                    22  ...       1                          \n",
       "0                    22  ...      -1                          \n",
       "0                    22  ...      -1                          \n",
       "0                    22  ...                                  \n",
       "...                 ...  ...     ...                    ...   \n",
       "47668                31  ...       1                          \n",
       "47668                31  ...       1                          \n",
       "47668                31  ...       1                          \n",
       "47668                31  ...       1                          \n",
       "47668                31  ...      -1                          \n",
       "\n",
       "                       Codons_position  Protein Codon  Clinvar ID  Exon   \n",
       "0                                                                        \\\n",
       "0                                                                         \n",
       "0      ENST00000488147.1:n.1254-251C>T                                    \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "...                                ...      ...   ...         ...   ...   \n",
       "47668       ENST00000484415.6:n.813C>T                              5/5   \n",
       "47668      ENST00000492963.6:n.1157C>T                              5/6   \n",
       "47668                                                                     \n",
       "47668     ENST00000496301.6:n.94+36C>T                                    \n",
       "47668                                                                     \n",
       "\n",
       "       Existing variant  CLNREVSTAT  CLNDN  \n",
       "0            rs17855866                     \n",
       "0            rs17855866                     \n",
       "0            rs17855866                     \n",
       "0            rs17855866                     \n",
       "0            rs17855866                     \n",
       "...                 ...         ...    ...  \n",
       "47668                                       \n",
       "47668                                       \n",
       "47668                                       \n",
       "47668                                       \n",
       "47668                                       \n",
       "\n",
       "[371550 rows x 39 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Clinical Significance</th>\n",
       "      <th>Codons_position</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "      <th>Clinvar ID</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Existing variant</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>Position_14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000488147.1:n.1254-251C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.813C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.1157C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.94+36C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  Clinical Significance   \n",
       "0                    22  ...                         \\\n",
       "0                    22  ...                          \n",
       "0                    22  ...                          \n",
       "0                    22  ...                          \n",
       "0                    22  ...                          \n",
       "...                 ...  ...                    ...   \n",
       "47668                31  ...                          \n",
       "47668                31  ...                          \n",
       "47668                31  ...                          \n",
       "47668                31  ...                          \n",
       "47668                31  ...                          \n",
       "\n",
       "                       Codons_position  Protein  Codon Clinvar ID  Exon   \n",
       "0                                                                        \\\n",
       "0                                                                         \n",
       "0      ENST00000488147.1:n.1254-251C>T                                    \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "...                                ...      ...    ...        ...   ...   \n",
       "47668       ENST00000484415.6:n.813C>T                              5/5   \n",
       "47668      ENST00000492963.6:n.1157C>T                              5/6   \n",
       "47668                                                                     \n",
       "47668     ENST00000496301.6:n.94+36C>T                                    \n",
       "47668                                                                     \n",
       "\n",
       "       Existing variant  CLNREVSTAT  CLNDN  Position_14  \n",
       "0            rs17855866                                  \n",
       "0            rs17855866                                  \n",
       "0            rs17855866                                  \n",
       "0            rs17855866                                  \n",
       "0            rs17855866                                  \n",
       "...                 ...         ...    ...          ...  \n",
       "47668                                                    \n",
       "47668                                                    \n",
       "47668                                                    \n",
       "47668                                                    \n",
       "47668                                                    \n",
       "\n",
       "[371550 rows x 40 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Position_14'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "      <th>Clinvar ID</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Existing variant</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>Position_14</th>\n",
       "      <th>Position_15</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  Protein  Codon  Clinvar ID  Exon   \n",
       "0                    22  ...                                    \\\n",
       "0                    22  ...                                     \n",
       "0                    22  ...                                     \n",
       "0                    22  ...                                     \n",
       "0                    22  ...                                     \n",
       "...                 ...  ...      ...    ...         ...   ...   \n",
       "47668                31  ...                               5/5   \n",
       "47668                31  ...                               5/6   \n",
       "47668                31  ...                                     \n",
       "47668                31  ...                                     \n",
       "47668                31  ...                                     \n",
       "\n",
       "      Existing variant  CLNREVSTAT  CLNDN  Position_14  Position_15   \n",
       "0           rs17855866                                               \\\n",
       "0           rs17855866                                                \n",
       "0           rs17855866                                                \n",
       "0           rs17855866                                                \n",
       "0           rs17855866                                                \n",
       "...                ...         ...    ...          ...          ...   \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "47668                                                                 \n",
       "\n",
       "       Protein Position and Amino Acid  \n",
       "0                                  NaN  \n",
       "0                                  NaN  \n",
       "0                                  NaN  \n",
       "0                                  NaN  \n",
       "0                                  NaN  \n",
       "...                                ...  \n",
       "47668                              NaN  \n",
       "47668                              NaN  \n",
       "47668                              NaN  \n",
       "47668                              NaN  \n",
       "47668                              NaN  \n",
       "\n",
       "[371550 rows x 42 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Position_15'] = vcf['csq'].str.split('|').str[15]\n",
    "\n",
    "# Format the values as \"W170R\"\n",
    "vcf['Protein Position and Amino Acid'] = vcf['Position_15'].str[0] + vcf['Position_14'] + vcf['Position_15'].str[-1]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO',\n",
       "       'Genotype', 'Genotype Quality',\n",
       "       'Raw Read Depth as reported by SAMtools',\n",
       "       'Quality Read Depth of bases with Phred score >= 15',\n",
       "       'Depth of reference-supporting bases (reads1)',\n",
       "       'Depth of variant-supporting bases (reads2)',\n",
       "       'Depth of reference-supporting bases on forward strand (reads1plus)',\n",
       "       'Depth of reference-supporting bases on reverse strand (reads1minus)',\n",
       "       'Depth of variant-supporting bases on forward strand (reads2plus)',\n",
       "       'Depth of variant-supporting bases on reverse strand (reads2minus)',\n",
       "       'Variant allele frequency', 'P-value from Fishers Exact Test', 'HET',\n",
       "       'HOM', 'Zygosity', 'CSQ', 'csq', 'Consequence', 'Impact', 'Gene_Name',\n",
       "       'Biotype', 'Strand', 'Clinical Significance', 'Codons_position',\n",
       "       'Protein', 'Codon', 'Clinvar ID', 'Exon', 'Existing variant',\n",
       "       'CLNREVSTAT', 'CLNDN', 'Position_14', 'Position_15',\n",
       "       'Protein Position and Amino Acid'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>Clinical Significance</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>Clinvar ID</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Strand</th>\n",
       "      <th>...</th>\n",
       "      <th>Raw Read Depth as reported by SAMtools</th>\n",
       "      <th>Quality Read Depth of bases with Phred score &gt;= 15</th>\n",
       "      <th>Depth of reference-supporting bases (reads1)</th>\n",
       "      <th>Depth of variant-supporting bases (reads2)</th>\n",
       "      <th>Depth of reference-supporting bases on forward strand (reads1plus)</th>\n",
       "      <th>Depth of reference-supporting bases on reverse strand (reads1minus)</th>\n",
       "      <th>Depth of variant-supporting bases on forward strand (reads2plus)</th>\n",
       "      <th>Depth of variant-supporting bases on reverse strand (reads2minus)</th>\n",
       "      <th>Variant allele frequency</th>\n",
       "      <th>P-value from Fishers Exact Test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WASH7P</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>28</td>\n",
       "      <td>7</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.005609</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>WASH6P</td>\n",
       "      <td>.</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>WASH6P</td>\n",
       "      <td>.</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>WASH6P</td>\n",
       "      <td>.</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>WASH6P</td>\n",
       "      <td>.</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>.</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>0.000699</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene_Name          ID                                   Consequence   \n",
       "0        DDX11L1  rs17855866                       downstream_gene_variant  \\\n",
       "0        DDX11L2  rs17855866                       downstream_gene_variant   \n",
       "0         WASH7P  rs17855866  intron_variant&non_coding_transcript_variant   \n",
       "0      MIR6859-1  rs17855866                       downstream_gene_variant   \n",
       "0                 rs17855866                     regulatory_region_variant   \n",
       "...          ...         ...                                           ...   \n",
       "47668     WASH6P           .            non_coding_transcript_exon_variant   \n",
       "47668     WASH6P           .            non_coding_transcript_exon_variant   \n",
       "47668     WASH6P           .                       downstream_gene_variant   \n",
       "47668     WASH6P           .  intron_variant&non_coding_transcript_variant   \n",
       "47668   DDX11L16           .                       downstream_gene_variant   \n",
       "\n",
       "           Zygosity CLNDN Clinical Significance CLNREVSTAT Clinvar ID   \n",
       "0      Heterozygous                                                    \\\n",
       "0      Heterozygous                                                     \n",
       "0      Heterozygous                                                     \n",
       "0      Heterozygous                                                     \n",
       "0      Heterozygous                                                     \n",
       "...             ...   ...                   ...        ...        ...   \n",
       "47668    Homozygous                                                     \n",
       "47668    Homozygous                                                     \n",
       "47668    Homozygous                                                     \n",
       "47668    Homozygous                                                     \n",
       "47668    Homozygous                                                     \n",
       "\n",
       "                                  Biotype Strand  ...   \n",
       "0      transcribed_unprocessed_pseudogene      1  ...  \\\n",
       "0                                  lncRNA      1  ...   \n",
       "0                  unprocessed_pseudogene     -1  ...   \n",
       "0                                   miRNA     -1  ...   \n",
       "0                                enhancer         ...   \n",
       "...                                   ...    ...  ...   \n",
       "47668                     retained_intron      1  ...   \n",
       "47668                     retained_intron      1  ...   \n",
       "47668                     retained_intron      1  ...   \n",
       "47668                     retained_intron      1  ...   \n",
       "47668              unprocessed_pseudogene     -1  ...   \n",
       "\n",
       "      Raw Read Depth as reported by SAMtools   \n",
       "0                                         35  \\\n",
       "0                                         35   \n",
       "0                                         35   \n",
       "0                                         35   \n",
       "0                                         35   \n",
       "...                                      ...   \n",
       "47668                                      8   \n",
       "47668                                      8   \n",
       "47668                                      8   \n",
       "47668                                      8   \n",
       "47668                                      8   \n",
       "\n",
       "      Quality Read Depth of bases with Phred score >= 15   \n",
       "0                                                     35  \\\n",
       "0                                                     35   \n",
       "0                                                     35   \n",
       "0                                                     35   \n",
       "0                                                     35   \n",
       "...                                                  ...   \n",
       "47668                                                  8   \n",
       "47668                                                  8   \n",
       "47668                                                  8   \n",
       "47668                                                  8   \n",
       "47668                                                  8   \n",
       "\n",
       "      Depth of reference-supporting bases (reads1)   \n",
       "0                                               28  \\\n",
       "0                                               28   \n",
       "0                                               28   \n",
       "0                                               28   \n",
       "0                                               28   \n",
       "...                                            ...   \n",
       "47668                                            1   \n",
       "47668                                            1   \n",
       "47668                                            1   \n",
       "47668                                            1   \n",
       "47668                                            1   \n",
       "\n",
       "      Depth of variant-supporting bases (reads2)   \n",
       "0                                              7  \\\n",
       "0                                              7   \n",
       "0                                              7   \n",
       "0                                              7   \n",
       "0                                              7   \n",
       "...                                          ...   \n",
       "47668                                          7   \n",
       "47668                                          7   \n",
       "47668                                          7   \n",
       "47668                                          7   \n",
       "47668                                          7   \n",
       "\n",
       "      Depth of reference-supporting bases on forward strand (reads1plus)   \n",
       "0                                                    20%                  \\\n",
       "0                                                    20%                   \n",
       "0                                                    20%                   \n",
       "0                                                    20%                   \n",
       "0                                                    20%                   \n",
       "...                                                  ...                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "47668                                              87.5%                   \n",
       "\n",
       "      Depth of reference-supporting bases on reverse strand (reads1minus)   \n",
       "0                                               0.005609                   \\\n",
       "0                                               0.005609                    \n",
       "0                                               0.005609                    \n",
       "0                                               0.005609                    \n",
       "0                                               0.005609                    \n",
       "...                                                  ...                    \n",
       "47668                                           0.000699                    \n",
       "47668                                           0.000699                    \n",
       "47668                                           0.000699                    \n",
       "47668                                           0.000699                    \n",
       "47668                                           0.000699                    \n",
       "\n",
       "       Depth of variant-supporting bases on forward strand (reads2plus)   \n",
       "0                                                     28                 \\\n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "0                                                     28                  \n",
       "...                                                  ...                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "47668                                                  0                  \n",
       "\n",
       "      Depth of variant-supporting bases on reverse strand (reads2minus)   \n",
       "0                                                      0                 \\\n",
       "0                                                      0                  \n",
       "0                                                      0                  \n",
       "0                                                      0                  \n",
       "0                                                      0                  \n",
       "...                                                  ...                  \n",
       "47668                                                  1                  \n",
       "47668                                                  1                  \n",
       "47668                                                  1                  \n",
       "47668                                                  1                  \n",
       "47668                                                  1                  \n",
       "\n",
       "      Variant allele frequency P-value from Fishers Exact Test  \n",
       "0                            7                               0  \n",
       "0                            7                               0  \n",
       "0                            7                               0  \n",
       "0                            7                               0  \n",
       "0                            7                               0  \n",
       "...                        ...                             ...  \n",
       "47668                        6                               1  \n",
       "47668                        6                               1  \n",
       "47668                        6                               1  \n",
       "47668                        6                               1  \n",
       "47668                        6                               1  \n",
       "\n",
       "[371550 rows x 33 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = vcf[['Gene_Name', 'ID', 'Consequence', 'Zygosity', \n",
    "          'CLNDN', 'Clinical Significance', 'CLNREVSTAT', \n",
    "          'Clinvar ID', 'Biotype', 'Strand', 'Protein Position and Amino Acid', \n",
    "          'Protein', 'Codon', 'Codons_position', 'CHROM', 'Exon', 'POS', \n",
    "          'REF', 'ALT', 'Impact', 'Existing variant', 'Genotype', 'Genotype Quality',\n",
    "       'Raw Read Depth as reported by SAMtools',\n",
    "       'Quality Read Depth of bases with Phred score >= 15',\n",
    "       'Depth of reference-supporting bases (reads1)',\n",
    "       'Depth of variant-supporting bases (reads2)',\n",
    "       'Depth of reference-supporting bases on forward strand (reads1plus)',\n",
    "       'Depth of reference-supporting bases on reverse strand (reads1minus)',\n",
    "       'Depth of variant-supporting bases on forward strand (reads2plus)',\n",
    "       'Depth of variant-supporting bases on reverse strand (reads2minus)',\n",
    "       'Variant allele frequency', 'P-value from Fishers Exact Test']]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/Dell/Downloads/KHHSPTGPTTL34_final_vcf.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Codon</th>\n",
       "      <th>Clinvar ID</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Existing variant</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>Position_14</th>\n",
       "      <th>Position_15</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000349216.4:p.Trp251Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>8/12</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>251</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W251R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000342313.3:p.Trp343Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>10/14</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>343</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W343R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000412228.1:p.Trp170Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>4/7</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>170</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W170R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td>1/2</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td>2/2</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td>3/4</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000478421.2:p.Trp506Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>10/14</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>506</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W506R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000484643.1:p.Trp235Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>7/11</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>235</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W235R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000482090.2:p.Trp344Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>9/13</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>344</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W344R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000480870.1:p.Trp218Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>6/10</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>218</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W218R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000480678.2:p.Trp507Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>10/14</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>507</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W507R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000484256.1:p.Trp281Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>8/12</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>281</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W281R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSP00000482138.1:p.Trp344Arg</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1166513</td>\n",
       "      <td>9/13</td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>344</td>\n",
       "      <td>W/R</td>\n",
       "      <td>W344R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>rs6672356&amp;COSV58985794</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>not_provided</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19 rows Ã— 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     POS         ID REF ALT QUAL FILTER   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS  \\\n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "30  chr1  942451  rs6672356   T   C    .   PASS   \n",
       "\n",
       "                                                 INFO Genotype   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1  \\\n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "30  ADP=26;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0,1;COMMO...      1/1   \n",
       "\n",
       "    Genotype Quality  ...                        Protein    Codon  Clinvar ID   \n",
       "30               146  ...                                             1166513  \\\n",
       "30               146  ...  ENSP00000349216.4:p.Trp251Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000342313.3:p.Trp343Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000412228.1:p.Trp170Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...                                             1166513   \n",
       "30               146  ...                                             1166513   \n",
       "30               146  ...                                             1166513   \n",
       "30               146  ...                                             1166513   \n",
       "30               146  ...                                             1166513   \n",
       "30               146  ...                                             1166513   \n",
       "30               146  ...                                             1166513   \n",
       "30               146  ...  ENSP00000478421.2:p.Trp506Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000484643.1:p.Trp235Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000482090.2:p.Trp344Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000480870.1:p.Trp218Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000480678.2:p.Trp507Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000484256.1:p.Trp281Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...  ENSP00000482138.1:p.Trp344Arg  Tgg/Cgg     1166513   \n",
       "30               146  ...                                             1166513   \n",
       "\n",
       "     Exon        Existing variant                           CLNREVSTAT   \n",
       "30         rs6672356&COSV58985794  criteria_provided&_single_submitter  \\\n",
       "30   8/12  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30  10/14  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30    4/7  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30    1/2  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30    2/2  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30    3/4  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30         rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30         rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30         rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30         rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30  10/14  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30   7/11  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30   9/13  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30   6/10  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30  10/14  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30   8/12  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30   9/13  rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "30         rs6672356&COSV58985794  criteria_provided&_single_submitter   \n",
       "\n",
       "           CLNDN  Position_14  Position_15  Protein Position and Amino Acid  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided          251          W/R                            W251R  \n",
       "30  not_provided          343          W/R                            W343R  \n",
       "30  not_provided          170          W/R                            W170R  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided                                                        NaN  \n",
       "30  not_provided          506          W/R                            W506R  \n",
       "30  not_provided          235          W/R                            W235R  \n",
       "30  not_provided          344          W/R                            W344R  \n",
       "30  not_provided          218          W/R                            W218R  \n",
       "30  not_provided          507          W/R                            W507R  \n",
       "30  not_provided          281          W/R                            W281R  \n",
       "30  not_provided          344          W/R                            W344R  \n",
       "30  not_provided                                                        NaN  \n",
       "\n",
       "[19 rows x 42 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = vcf[vcf['ID'] == 'rs6672356']\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO',\n",
       "       'Genotype', 'Genotype Quality',\n",
       "       'Raw Read Depth as reported by SAMtools',\n",
       "       'Quality Read Depth of bases with Phred score >= 15',\n",
       "       'Depth of reference-supporting bases (reads1)',\n",
       "       'Depth of variant-supporting bases (reads2)',\n",
       "       'Depth of reference-supporting bases on forward strand (reads1plus)',\n",
       "       'Depth of reference-supporting bases on reverse strand (reads1minus)',\n",
       "       'Depth of variant-supporting bases on forward strand (reads2plus)',\n",
       "       'Depth of variant-supporting bases on reverse strand (reads2minus)',\n",
       "       'Variant allele frequency', 'P-value from Fishers Exact Test', 'HET',\n",
       "       'HOM', 'Zygosity', 'CSQ', 'csq', 'Consequence', 'Impact', 'Gene_Name',\n",
       "       'Biotype', 'Strand', 'Clinical Significance', 'Codons_position',\n",
       "       'Protein', 'Codon', 'Clinvar ID', 'Exon', 'Existing variant'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  PUBMED  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS   \n",
       "0                    22  ...                                               \\\n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "0                    22  ...                                                \n",
       "...                 ...  ...     ...         ...        ...           ...   \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "47668                31  ...                                                \n",
       "\n",
       "      MOTIF_SCORE_CHANGE  TRANSCRIPTION_FACTORS  ClinVar  ClinVar_CLNSIG   \n",
       "0                                                                         \\\n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "...                  ...                    ...      ...             ...   \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar_CLNDN  \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "0                                         \n",
       "...                   ...            ...  \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "47668                                     \n",
       "\n",
       "[371550 rows x 108 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_adp = vcf.copy()\n",
    "df_adp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype Quality</th>\n",
       "      <th>...</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>New_Column</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47668</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210016</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371550 rows Ã— 109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS  \\\n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "0      chr1     14752  rs17855866   G   A    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "47668  chrY  57210016           .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO Genotype   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1  \\\n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...      0/1   \n",
       "...                                                  ...      ...   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "47668  ADP=8;WT=0;HET=0;HOM=1;NC=0;CSQ=T|intron_varia...      1/1   \n",
       "\n",
       "       Genotype Quality  ...  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS   \n",
       "0                    22  ...                                       \\\n",
       "0                    22  ...                                        \n",
       "0                    22  ...                                        \n",
       "0                    22  ...                                        \n",
       "0                    22  ...                                        \n",
       "...                 ...  ...         ...        ...           ...   \n",
       "47668                31  ...                                        \n",
       "47668                31  ...                                        \n",
       "47668                31  ...                                        \n",
       "47668                31  ...                                        \n",
       "47668                31  ...                                        \n",
       "\n",
       "       MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  ClinVar  ClinVar_CLNSIG   \n",
       "0                                                                         \\\n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "0                                                                          \n",
       "...                   ...                   ...      ...             ...   \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "47668                                                                      \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar_CLNDN   \n",
       "0                                         \\\n",
       "0                                          \n",
       "0                                          \n",
       "0                                          \n",
       "0                                          \n",
       "...                   ...            ...   \n",
       "47668                                      \n",
       "47668                                      \n",
       "47668                                      \n",
       "47668                                      \n",
       "47668                                      \n",
       "\n",
       "                                              New_Column  \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  \n",
       "0      ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  \n",
       "...                                                  ...  \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  \n",
       "47668                        ADP=8;WT=0;HET=0;HOM=1;NC=0  \n",
       "\n",
       "[371550 rows x 109 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_adp[\"New_Column\"] = df_adp[\"INFO\"].str.extract(r'^(.*?)(?=;?CSQ)')\n",
    "df_adp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ADP=35;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX11L1:100287102|WASH7P:653635;R3;RS=17855866;RSPOS=14752;SAO=0;SSR=0;TOPMED=0.99442533129459734,0.00553484964322120,0.00003981906218144;VC=SNV;VP=0x050000040005000002000100;WGT=1;dbSNPBuildID=123'"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_adp.New_Column.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'A|intron_variant&non_coding_transcript_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000488147|unprocessed_pseudogene||10/10|ENST00000488147.1:n.1254-251C>T|||||||rs17855866||-1||SNV|HGNC|HGNC:38034|YES|||||||||||||||||||||||||||||||||0.00162|0.005487|0|0.001186|0.0002987|0.0001935|0|0|6.055e-05|0.00203|0.0002195|0.005487|gnomADg_AFR|||||||||||||'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.csq.iloc[2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'A|downstream_gene_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328|lncRNA||||||||||rs17855866|343|1||SNV|EntrezGene||YES|||1||||||||||||||||||||||||||||||0.00162|0.005487|0|0.001186|0.0002987|0.0001935|0|0|6.055e-05|0.00203|0.0002195|0.005487|gnomADg_AFR|||||||||||||'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.csq.iloc[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'T|synonymous_variant|LOW|AGRN|ENSG00000188157|Transcript|ENST00000620552|protein_coding|39/39||ENST00000620552.4:c.5712C>T|ENSP00000484607.1:p.Asp1904%3D|6176|5712|1904|D|gaC/gaT|rs4275402&COSV65070964||1||SNV|HGNC|HGNC:329||||5|||ENSP00000484607||A0A087X208.61|UPI0004E4CB7F|||1|||Gene3D:2.60.120.200&Pfam:PF00054&PROSITE_profiles:PS50025&SMART:SM00282&Superfamily:SSF49899&CDD:cd00110|||0.5853|0.2617|0.7262|0.8214|0.6779|0.5838|0.6764|0.3114|0.7722|0.6575|0.8138|0.641|0.698|0.6758|0.6133|0.5913|0.3227|0.6733|0.7196|0.6622|0.8119|0.6343|0.6392|0.6961|0.5995|0.6072|0.8214|EAS|benign|0&1|1&1|25741868&18414213||||||128319|Benign|criteria_provided&_multiple_submitters&_no_conflicts|Congenital_myasthenic_syndrome_8&not_specified'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.csq.iloc[500]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'T|downstream_gene_variant|MODIFIER|RER1|ENSG00000157916|Transcript|ENST00000378518|protein_coding||||||||||rs1129171&COSV56579903|3268|1||SNV|HGNC|HGNC:30309||||3|||ENSP00000367779||Q5T093.117|UPI0000470C2F|||||||||0.3065|0.1815|0.268|0.4544|0.3062|0.3507||||||||||0.2822|0.1869|0.3169|0.2834|0.4237|0.4289|0.2518|0.269|0.3216|0.2989|0.334|0.4544|EAS|benign|0&1|1&1|||||||296268|Benign|criteria_provided&_multiple_submitters&_no_conflicts|not_provided&Peroxisome_biogenesis_disorder_6A_(Zellweger)'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.csq.iloc[2234]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Final_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15461640</td>\n",
       "      <td>rs372947070</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=372947070;dbSNPBuildID=138;SSR=0;GENEINFO=C...</td>\n",
       "      <td>CELA2B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15461640</td>\n",
       "      <td>rs372947070</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=372947070;dbSNPBuildID=138;SSR=0;GENEINFO=C...</td>\n",
       "      <td>CELA2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65572326</td>\n",
       "      <td>rs35573508</td>\n",
       "      <td>A</td>\n",
       "      <td>G,T</td>\n",
       "      <td>RS=35573508;dbSNPBuildID=126;SSR=0;GENEINFO=LE...</td>\n",
       "      <td>LEPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65616108</td>\n",
       "      <td>rs34499590</td>\n",
       "      <td>C</td>\n",
       "      <td>G,T</td>\n",
       "      <td>RS=34499590;dbSNPBuildID=126;SSR=0;GENEINFO=LE...</td>\n",
       "      <td>LEPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65636215</td>\n",
       "      <td>rs142647085</td>\n",
       "      <td>A</td>\n",
       "      <td>G,T</td>\n",
       "      <td>RS=142647085;dbSNPBuildID=134;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LEPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65636536</td>\n",
       "      <td>rs145651189</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=145651189;dbSNPBuildID=134;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LEPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65636996</td>\n",
       "      <td>rs145685060</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>RS=145685060;dbSNPBuildID=134;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LEPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>112918054</td>\n",
       "      <td>rs149491709</td>\n",
       "      <td>CAATAAATAAATAAATAAATAAATA</td>\n",
       "      <td>C,CAATA,CAATAAATA,CAATAAATAAATA,CAATAAATAAATAA...</td>\n",
       "      <td>RS=149491709;dbSNPBuildID=134;SSR=0;GENEINFO=S...</td>\n",
       "      <td>SLC16A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>112918057</td>\n",
       "      <td>rs773587590</td>\n",
       "      <td>T</td>\n",
       "      <td>TAAAG</td>\n",
       "      <td>RS=773587590;dbSNPBuildID=144;SSR=0;GENEINFO=S...</td>\n",
       "      <td>SLC16A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>112956232</td>\n",
       "      <td>rs58047463</td>\n",
       "      <td>CG</td>\n",
       "      <td>C,CGG</td>\n",
       "      <td>RS=58047463;dbSNPBuildID=129;SSR=0;GENEINFO=SL...</td>\n",
       "      <td>SLC16A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>112956232</td>\n",
       "      <td>rs58047463</td>\n",
       "      <td>CG</td>\n",
       "      <td>C,CGG</td>\n",
       "      <td>RS=58047463;dbSNPBuildID=129;SSR=0;GENEINFO=SL...</td>\n",
       "      <td>SLC16A1-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156115212</td>\n",
       "      <td>rs886045363</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>RS=886045363;dbSNPBuildID=149;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156115213</td>\n",
       "      <td>rs886045364</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "      <td>RS=886045364;dbSNPBuildID=149;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156115268</td>\n",
       "      <td>rs397517901</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>RS=397517901;dbSNPBuildID=138;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156130617</td>\n",
       "      <td>rs41313880</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=41313880;dbSNPBuildID=127;SSR=0;GENEINFO=LM...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156130658</td>\n",
       "      <td>rs60864230</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C,T</td>\n",
       "      <td>RS=60864230;dbSNPBuildID=133;SSR=0;GENEINFO=LM...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156130731</td>\n",
       "      <td>rs150645079</td>\n",
       "      <td>G</td>\n",
       "      <td>A,T</td>\n",
       "      <td>RS=150645079;dbSNPBuildID=134;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156134914</td>\n",
       "      <td>rs397517907</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=397517907;dbSNPBuildID=138;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156135893</td>\n",
       "      <td>rs751707982</td>\n",
       "      <td>C</td>\n",
       "      <td>A,G,T</td>\n",
       "      <td>RS=751707982;dbSNPBuildID=144;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156135917</td>\n",
       "      <td>rs1212920276</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=1212920276;dbSNPBuildID=151;SSR=0;GENEINFO=...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156135949</td>\n",
       "      <td>rs775159300</td>\n",
       "      <td>C</td>\n",
       "      <td>A,G,T</td>\n",
       "      <td>RS=775159300;dbSNPBuildID=144;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156135991</td>\n",
       "      <td>rs749784223</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "      <td>RS=749784223;dbSNPBuildID=144;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156136113</td>\n",
       "      <td>rs267607603</td>\n",
       "      <td>G</td>\n",
       "      <td>A,T</td>\n",
       "      <td>RS=267607603;dbSNPBuildID=137;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156136283</td>\n",
       "      <td>rs762130433</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>RS=762130433;dbSNPBuildID=144;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156136380</td>\n",
       "      <td>rs368542816</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>RS=368542816;dbSNPBuildID=138;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156136414</td>\n",
       "      <td>rs267607598</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>RS=267607598;dbSNPBuildID=137;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156136985</td>\n",
       "      <td>rs11575937</td>\n",
       "      <td>G</td>\n",
       "      <td>A,T</td>\n",
       "      <td>RS=11575937;dbSNPBuildID=120;SSR=0;GENEINFO=LM...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156137027</td>\n",
       "      <td>rs200466188</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=200466188;dbSNPBuildID=137;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156137028</td>\n",
       "      <td>rs375516745</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>RS=375516745;dbSNPBuildID=138;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156137042</td>\n",
       "      <td>rs377700689</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=377700689;dbSNPBuildID=138;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156137141</td>\n",
       "      <td>rs878855233</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>RS=878855233;dbSNPBuildID=147;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156137175</td>\n",
       "      <td>rs41314035</td>\n",
       "      <td>G</td>\n",
       "      <td>A,T</td>\n",
       "      <td>RS=41314035;dbSNPBuildID=127;SSR=0;GENEINFO=LM...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156137190</td>\n",
       "      <td>rs149339264</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "      <td>RS=149339264;dbSNPBuildID=134;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156137678</td>\n",
       "      <td>rs267607613</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RS=267607613;dbSNPBuildID=137;SSR=0;GENEINFO=L...</td>\n",
       "      <td>LMNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM        POS          rsID                        REF   \n",
       "0   chr1   15461640   rs372947070                          C  \\\n",
       "1   chr1   15461640   rs372947070                          C   \n",
       "2   chr1   65572326    rs35573508                          A   \n",
       "3   chr1   65616108    rs34499590                          C   \n",
       "4   chr1   65636215   rs142647085                          A   \n",
       "5   chr1   65636536   rs145651189                          A   \n",
       "6   chr1   65636996   rs145685060                          T   \n",
       "7   chr1  112918054   rs149491709  CAATAAATAAATAAATAAATAAATA   \n",
       "8   chr1  112918057   rs773587590                          T   \n",
       "9   chr1  112956232    rs58047463                         CG   \n",
       "10  chr1  112956232    rs58047463                         CG   \n",
       "11  chr1  156115212   rs886045363                          G   \n",
       "12  chr1  156115213   rs886045364                          C   \n",
       "13  chr1  156115268   rs397517901                          A   \n",
       "14  chr1  156130617    rs41313880                          C   \n",
       "15  chr1  156130658    rs60864230                          G   \n",
       "16  chr1  156130731   rs150645079                          G   \n",
       "17  chr1  156134914   rs397517907                          C   \n",
       "18  chr1  156135893   rs751707982                          C   \n",
       "19  chr1  156135917  rs1212920276                          C   \n",
       "20  chr1  156135949   rs775159300                          C   \n",
       "21  chr1  156135991   rs749784223                          C   \n",
       "22  chr1  156136113   rs267607603                          G   \n",
       "23  chr1  156136283   rs762130433                          A   \n",
       "24  chr1  156136380   rs368542816                          G   \n",
       "25  chr1  156136414   rs267607598                          G   \n",
       "26  chr1  156136985    rs11575937                          G   \n",
       "27  chr1  156137027   rs200466188                          C   \n",
       "28  chr1  156137028   rs375516745                          G   \n",
       "29  chr1  156137042   rs377700689                          C   \n",
       "30  chr1  156137141   rs878855233                          A   \n",
       "31  chr1  156137175    rs41314035                          G   \n",
       "32  chr1  156137190   rs149339264                          C   \n",
       "33  chr1  156137678   rs267607613                          C   \n",
       "\n",
       "                                                  ALT   \n",
       "0                                                   T  \\\n",
       "1                                                   T   \n",
       "2                                                 G,T   \n",
       "3                                                 G,T   \n",
       "4                                                 G,T   \n",
       "5                                                   T   \n",
       "6                                                   C   \n",
       "7   C,CAATA,CAATAAATA,CAATAAATAAATA,CAATAAATAAATAA...   \n",
       "8                                               TAAAG   \n",
       "9                                               C,CGG   \n",
       "10                                              C,CGG   \n",
       "11                                                  A   \n",
       "12                                                A,T   \n",
       "13                                                  G   \n",
       "14                                                  T   \n",
       "15                                              A,C,T   \n",
       "16                                                A,T   \n",
       "17                                                  T   \n",
       "18                                              A,G,T   \n",
       "19                                                  T   \n",
       "20                                              A,G,T   \n",
       "21                                                A,T   \n",
       "22                                                A,T   \n",
       "23                                                  G   \n",
       "24                                                A,C   \n",
       "25                                                A,C   \n",
       "26                                                A,T   \n",
       "27                                                  T   \n",
       "28                                                A,C   \n",
       "29                                                  T   \n",
       "30                                                  C   \n",
       "31                                                A,T   \n",
       "32                                                A,T   \n",
       "33                                                  T   \n",
       "\n",
       "                                                 INFO   Final_Gene  \n",
       "0   RS=372947070;dbSNPBuildID=138;SSR=0;GENEINFO=C...       CELA2B  \n",
       "1   RS=372947070;dbSNPBuildID=138;SSR=0;GENEINFO=C...       CELA2A  \n",
       "2   RS=35573508;dbSNPBuildID=126;SSR=0;GENEINFO=LE...         LEPR  \n",
       "3   RS=34499590;dbSNPBuildID=126;SSR=0;GENEINFO=LE...         LEPR  \n",
       "4   RS=142647085;dbSNPBuildID=134;SSR=0;GENEINFO=L...         LEPR  \n",
       "5   RS=145651189;dbSNPBuildID=134;SSR=0;GENEINFO=L...         LEPR  \n",
       "6   RS=145685060;dbSNPBuildID=134;SSR=0;GENEINFO=L...         LEPR  \n",
       "7   RS=149491709;dbSNPBuildID=134;SSR=0;GENEINFO=S...      SLC16A1  \n",
       "8   RS=773587590;dbSNPBuildID=144;SSR=0;GENEINFO=S...      SLC16A1  \n",
       "9   RS=58047463;dbSNPBuildID=129;SSR=0;GENEINFO=SL...      SLC16A1  \n",
       "10  RS=58047463;dbSNPBuildID=129;SSR=0;GENEINFO=SL...  SLC16A1-AS1  \n",
       "11  RS=886045363;dbSNPBuildID=149;SSR=0;GENEINFO=L...         LMNA  \n",
       "12  RS=886045364;dbSNPBuildID=149;SSR=0;GENEINFO=L...         LMNA  \n",
       "13  RS=397517901;dbSNPBuildID=138;SSR=0;GENEINFO=L...         LMNA  \n",
       "14  RS=41313880;dbSNPBuildID=127;SSR=0;GENEINFO=LM...         LMNA  \n",
       "15  RS=60864230;dbSNPBuildID=133;SSR=0;GENEINFO=LM...         LMNA  \n",
       "16  RS=150645079;dbSNPBuildID=134;SSR=0;GENEINFO=L...         LMNA  \n",
       "17  RS=397517907;dbSNPBuildID=138;SSR=0;GENEINFO=L...         LMNA  \n",
       "18  RS=751707982;dbSNPBuildID=144;SSR=0;GENEINFO=L...         LMNA  \n",
       "19  RS=1212920276;dbSNPBuildID=151;SSR=0;GENEINFO=...         LMNA  \n",
       "20  RS=775159300;dbSNPBuildID=144;SSR=0;GENEINFO=L...         LMNA  \n",
       "21  RS=749784223;dbSNPBuildID=144;SSR=0;GENEINFO=L...         LMNA  \n",
       "22  RS=267607603;dbSNPBuildID=137;SSR=0;GENEINFO=L...         LMNA  \n",
       "23  RS=762130433;dbSNPBuildID=144;SSR=0;GENEINFO=L...         LMNA  \n",
       "24  RS=368542816;dbSNPBuildID=138;SSR=0;GENEINFO=L...         LMNA  \n",
       "25  RS=267607598;dbSNPBuildID=137;SSR=0;GENEINFO=L...         LMNA  \n",
       "26  RS=11575937;dbSNPBuildID=120;SSR=0;GENEINFO=LM...         LMNA  \n",
       "27  RS=200466188;dbSNPBuildID=137;SSR=0;GENEINFO=L...         LMNA  \n",
       "28  RS=375516745;dbSNPBuildID=138;SSR=0;GENEINFO=L...         LMNA  \n",
       "29  RS=377700689;dbSNPBuildID=138;SSR=0;GENEINFO=L...         LMNA  \n",
       "30  RS=878855233;dbSNPBuildID=147;SSR=0;GENEINFO=L...         LMNA  \n",
       "31  RS=41314035;dbSNPBuildID=127;SSR=0;GENEINFO=LM...         LMNA  \n",
       "32  RS=149339264;dbSNPBuildID=134;SSR=0;GENEINFO=L...         LMNA  \n",
       "33  RS=267607613;dbSNPBuildID=137;SSR=0;GENEINFO=L...         LMNA  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_csv(r'C:/Users/Dell/Downloads/test_file.csv', sep = '\\t')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'RS=372947070;dbSNPBuildID=138;SSR=0;GENEINFO=CELA2A:63036;VC=SNV;NSM;GNO;FREQ=1000Genomes:0.9997,0.0003123|ExAC:0.9998,0.0002224|GnomAD:0.9999,7.128e-05|GnomAD_exomes:0.9998,0.0002108|GoESP:0.9999,7.689e-05|TOMMO:1,3.539e-05|TOPMED:0.9999,6.423e-05|dbGaP_PopFreq:0.9999,5.645e-05;CLNVI=.,;CLNORIGIN=.,1;CLNSIG=.,5|5;CLNDISDB=.,Office_of_Rare_Diseases:11944/MONDO:MONDO:0005010/MeSH:D003324/MedGen:C1956346/MONDO:MONDO:0005347/MedGen:C0813230/Human_Phenotype_Ontology:HP:0002155/Human_Phenotype_Ontology:HP:0003082/Human_Phenotype_Ontology:HP:0008174/Human_Phenotype_Ontology:HP:0008332/MONDO:MONDO:0005044/MedGen:C0020538/Human_Phenotype_Ontology:HP:0000822/Human_Phenotype_Ontology:HP:0004949/Human_Phenotype_Ontology:HP:0005126/MedGen:C1320657|MONDO:MONDO:0032837/MedGen:C5231430/OMIM:618620;CLNDN=.,Coronary_artery_disorder/Diabetes/Hypertensive_disorder/Hypertriglyceridemia|Abdominal_obesity-metabolic_syndrome_4;CLNREVSTAT=.,no_criteria|single;CLNACC=.,RCV000853074.1|RCV002249481.1;CLNHGVS=NC_000001.11:g.15461640=,NC_000001.11:g.15461640C>T;CSQ=T|missense_variant|MODERATE|CELA2A|ENSG00000142615|Transcript|ENST00000359621|protein_coding|3/8||ENST00000359621.5:c.209C>T|ENSP00000352639.4:p.Thr70Met|231|209|70|T/M|aCg/aTg|rs372947070&COSV62747322||1||SNV|HGNC|HGNC:24609|YES|NM_033440.3||1|P1|CCDS157.1|ENSP00000352639|P08217.184||UPI0000129E36|||1|deleterious(0)||CDD:cd00190&Gene3D:2.40.10.10&Gene3D:2.40.10.10&Pfam:PF00089&SMART:SM00020&Superfamily:SSF50494&PROSITE_profiles:PS50240&PANTHER:PTHR24257:SF23&PANTHER:PTHR24257&PROSITE_patterns:PS00134&Prints:PR00722&AFDB-ENSP_mappings:AF-P08217-F1.A|||0.0004|0|0|0.001|0|0.001|0.0002108|0|5.782e-05|0|0.0009786|0|1.758e-05|0|0.001013|0.0001117|0|0|0.0002618|0|0.0007689|0|0|4.409e-05|0|0.001241|0.001241|gnomADg_SAS|pathogenic|0&1|1&1|31358993||||||633595|Pathogenic|criteria_provided&_single_submitter|Abdominal_obesity-metabolic_syndrome_4&Coronary_artery_disorder&Hypertensive_disorder&Hypertriglyceridemia&Diabetes,T|non_coding_transcript_exon_variant|MODIFIER|CELA2A|ENSG00000142615|Transcript|ENST00000459653|protein_coding_CDS_not_defined|3/4||ENST00000459653.1:n.235C>T||235|||||rs372947070&COSV62747322||1||SNV|HGNC|HGNC:24609||||2|||||||||1||||||0.0004|0|0|0.001|0|0.001|0.0002108|0|5.782e-05|0|0.0009786|0|1.758e-05|0|0.001013|0.0001117|0|0|0.0002618|0|0.0007689|0|0|4.409e-05|0|0.001241|0.001241|gnomADg_SAS|pathogenic|0&1|1&1|31358993||||||633595|Pathogenic|criteria_provided&_single_submitter|Abdominal_obesity-metabolic_syndrome_4&Coronary_artery_disorder&Hypertensive_disorder&Hypertriglyceridemia&Diabetes,T|upstream_gene_variant|MODIFIER|CELA2B|ENSG00000215704|Transcript|ENST00000494280|protein_coding_CDS_not_defined||||||||||rs372947070&COSV62747322|4269|1||SNV|HGNC|HGNC:29995||||5|||||||||||||||0.0004|0|0|0.001|0|0.001|0.0002108|0|5.782e-05|0|0.0009786|0|1.758e-05|0|0.001013|0.0001117|0|0|0.0002618|0|0.0007689|0|0|4.409e-05|0|0.001241|0.001241|gnomADg_SAS|pathogenic|0&1|1&1|31358993||||||633595|Pathogenic|criteria_provided&_single_submitter|Abdominal_obesity-metabolic_syndrome_4&Coronary_artery_disorder&Hypertensive_disorder&Hypertriglyceridemia&Diabetes,T|downstream_gene_variant|MODIFIER|CELA2A|ENSG00000142615|Transcript|ENST00000497590|protein_coding_CDS_not_defined||||||||||rs372947070&COSV62747322|4017|1||SNV|HGNC|HGNC:24609||||2|||||||||1||||||0.0004|0|0|0.001|0|0.001|0.0002108|0|5.782e-05|0|0.0009786|0|1.758e-05|0|0.001013|0.0001117|0|0|0.0002618|0|0.0007689|0|0|4.409e-05|0|0.001241|0.001241|gnomADg_SAS|pathogenic|0&1|1&1|31358993||||||633595|Pathogenic|criteria_provided&_single_submitter|Abdominal_obesity-metabolic_syndrome_4&Coronary_artery_disorder&Hypertensive_disorder&Hypertriglyceridemia&Diabetes'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1.INFO.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
