{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "47fb07cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a1c22e34",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/kalyani_mam_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHGLBS448_final.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHGLBS448_final.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "158a1c2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>58</td>\n",
       "      <td>89</td>\n",
       "      <td>89</td>\n",
       "      <td>71</td>\n",
       "      <td>18</td>\n",
       "      <td>20.22%</td>\n",
       "      <td>1.4677E-6</td>\n",
       "      <td>58</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33484</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155997799</td>\n",
       "      <td>rs6642401</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>48.15%</td>\n",
       "      <td>1.81E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33485</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0.5...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>1.1312E-3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33486</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2789135</td>\n",
       "      <td>rs2534636</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9278,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33488</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33489 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1     685694            .   T   C    .   PASS   \n",
       "4      chr1     686266            .   A   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "33484  chrX  155997799    rs6642401   C   T    .   PASS   \n",
       "33485  chrX  156003433    rs2037999   T   C    .   PASS   \n",
       "33486  chrY    2789135    rs2534636   C   T    .   PASS   \n",
       "33487  chrY    9467257  rs777840135   G   A    .   PASS   \n",
       "33488  chrY   21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1      ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "4      ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   58   89   89   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "33484  ADP=27;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....  0/1   47   27   27   \n",
       "33485  ADP=9;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0.5...  1/1   29    9    9   \n",
       "33486  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9278,0....  1/1  255   55   54   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33488  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "        RD   AD    FREQ         PVAL  RDF RDR  ADF ADR  \n",
       "0      102   41  28.67%   1.5868E-14   99   3   33   8  \n",
       "1        0  168    100%  1.6424E-100    0   0  145  23  \n",
       "2       66   32  32.65%   1.1206E-11   52  14   25   7  \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0  \n",
       "4       71   18  20.22%    1.4677E-6   58  13   14   4  \n",
       "...    ...  ...     ...          ...  ...  ..  ...  ..  \n",
       "33484   14   13  48.15%      1.81E-5   12   2   12   1  \n",
       "33485    2    7  77.78%    1.1312E-3    2   0    6   1  \n",
       "33486    0   54    100%   4.0229E-32    0   0   47   7  \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9  \n",
       "33488    0   19    100%   2.8292E-11    0   0   17   2  \n",
       "\n",
       "[33489 rows x 20 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHGLBS448_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "9d235f02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>58</td>\n",
       "      <td>89</td>\n",
       "      <td>89</td>\n",
       "      <td>71</td>\n",
       "      <td>18</td>\n",
       "      <td>20.22%</td>\n",
       "      <td>1.4677E-6</td>\n",
       "      <td>58</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33484</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155997799</td>\n",
       "      <td>rs6642401</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>48.15%</td>\n",
       "      <td>1.81E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33485</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0.5...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>1.1312E-3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33486</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2789135</td>\n",
       "      <td>rs2534636</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9278,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33488</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33489 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1     685694            .   T   C    .   PASS   \n",
       "4      chr1     686266            .   A   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "33484  chrX  155997799    rs6642401   C   T    .   PASS   \n",
       "33485  chrX  156003433    rs2037999   T   C    .   PASS   \n",
       "33486  chrY    2789135    rs2534636   C   T    .   PASS   \n",
       "33487  chrY    9467257  rs777840135   G   A    .   PASS   \n",
       "33488  chrY   21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1      ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "4      ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   58   89   89   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "33484  ADP=27;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....  0/1   47   27   27   \n",
       "33485  ADP=9;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0.5...  1/1   29    9    9   \n",
       "33486  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9278,0....  1/1  255   55   54   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33488  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "        RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \n",
       "0      102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous  \n",
       "1        0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous  \n",
       "2       66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous  \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous  \n",
       "4       71   18  20.22%    1.4677E-6   58  13   14   4   1   0  Heterozygous  \n",
       "...    ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...  \n",
       "33484   14   13  48.15%      1.81E-5   12   2   12   1   1   0  Heterozygous  \n",
       "33485    2    7  77.78%    1.1312E-3    2   0    6   1   0   1    Homozygous  \n",
       "33486    0   54    100%   4.0229E-32    0   0   47   7   0   1    Homozygous  \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous  \n",
       "33488    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous  \n",
       "\n",
       "[33489 rows x 23 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "d1436913",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>58</td>\n",
       "      <td>89</td>\n",
       "      <td>89</td>\n",
       "      <td>71</td>\n",
       "      <td>18</td>\n",
       "      <td>20.22%</td>\n",
       "      <td>1.4677E-6</td>\n",
       "      <td>58</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33484</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155997799</td>\n",
       "      <td>rs6642401</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=27;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>47</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>48.15%</td>\n",
       "      <td>1.81E-5</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33485</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156003433</td>\n",
       "      <td>rs2037999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0.5...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>1.1312E-3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33486</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2789135</td>\n",
       "      <td>rs2534636</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9278,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SRY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33488</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33489 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1      69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1      69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1      69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1     685694            .   T   C    .   PASS   \n",
       "4      chr1     686266            .   A   C    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "33484  chrX  155997799    rs6642401   C   T    .   PASS   \n",
       "33485  chrX  156003433    rs2037999   T   C    .   PASS   \n",
       "33486  chrY    2789135    rs2534636   C   T    .   PASS   \n",
       "33487  chrY    9467257  rs777840135   G   A    .   PASS   \n",
       "33488  chrY   21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1      ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "4      ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   58   89   89   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "33484  ADP=27;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9483,0....  0/1   47   27   27   \n",
       "33485  ADP=9;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.4079,0.5...  1/1   29    9    9   \n",
       "33486  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9278,0....  1/1  255   55   54   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33488  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "        RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0      102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1        0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2       66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "4       71   18  20.22%    1.4677E-6   58  13   14   4   1   0  Heterozygous   \n",
       "...    ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "33484   14   13  48.15%      1.81E-5   12   2   12   1   1   0  Heterozygous   \n",
       "33485    2    7  77.78%    1.1312E-3    2   0    6   1   0   1    Homozygous   \n",
       "33486    0   54    100%   4.0229E-32    0   0   47   7   0   1    Homozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33488    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "      Gene Name  \n",
       "0         OR4F5  \n",
       "1         OR4F5  \n",
       "2         OR4F5  \n",
       "3                \n",
       "4                \n",
       "...         ...  \n",
       "33484      IL9R  \n",
       "33485      IL9R  \n",
       "33486       SRY  \n",
       "33487     TSPY1  \n",
       "33488            \n",
       "\n",
       "[33489 rows x 24 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1561107f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33488</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>275979 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1     69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33488  chrY  21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1      ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33488  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "        RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0      102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1        0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2       66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "...    ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33488    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0         OR4F5  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1         OR4F5  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2         OR4F5  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...         ...                                                ...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33488            A|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                     csq  \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...  \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...  \n",
       "2      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...  \n",
       "3      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...  \n",
       "3      C|intron_variant&non_coding_transcript_variant...  \n",
       "...                                                  ...  \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...  \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...  \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...  \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...  \n",
       "33488  A|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "\n",
       "[275979 rows x 26 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "da032507",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>243</td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>544</td>\n",
       "      <td>484</td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>930</td>\n",
       "      <td>870</td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:15079</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000419394</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000423647</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000451548</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000615605</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2900</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619815</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2912</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33488</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000226918</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000437359</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>48</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>275979 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1     69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33488  chrY  21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1      ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33488  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "        RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0      102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1        0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2       66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "...    ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33488    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0         OR4F5  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1         OR4F5  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2         OR4F5  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...         ...                                                ...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33488            A|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                     csq Allele  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...      G   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...      G   \n",
       "2      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...      C   \n",
       "3      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...      C   \n",
       "3      C|intron_variant&non_coding_transcript_variant...      C   \n",
       "...                                                  ...    ...   \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "33488  A|non_coding_transcript_exon_variant|MODIFIER|...      A   \n",
       "\n",
       "                                        Consequence    IMPACT    SYMBOL  \\\n",
       "0                                synonymous_variant       LOW     OR4F5   \n",
       "1                                  missense_variant  MODERATE     OR4F5   \n",
       "2                                synonymous_variant       LOW     OR4F5   \n",
       "3                           downstream_gene_variant  MODIFIER    OR4F16   \n",
       "3      intron_variant&non_coding_transcript_variant  MODIFIER             \n",
       "...                                             ...       ...       ...   \n",
       "33487                              missense_variant  MODERATE     TSPY1   \n",
       "33487                              missense_variant  MODERATE     TSPY1   \n",
       "33487                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "33487                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "33488            non_coding_transcript_exon_variant  MODIFIER             \n",
       "\n",
       "                  Gene Feature_type          Feature                 BIOTYPE  \\\n",
       "0      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "1      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "2      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "3      ENSG00000284662   Transcript  ENST00000332831          protein_coding   \n",
       "3      ENSG00000230021   Transcript  ENST00000419394    processed_transcript   \n",
       "...                ...          ...              ...                     ...   \n",
       "33487  ENSG00000258992   Transcript  ENST00000423647          protein_coding   \n",
       "33487  ENSG00000258992   Transcript  ENST00000451548          protein_coding   \n",
       "33487  ENSG00000225516   Transcript  ENST00000615605    processed_transcript   \n",
       "33487  ENSG00000225516   Transcript  ENST00000619815    processed_transcript   \n",
       "33488  ENSG00000226918   Transcript  ENST00000437359  unprocessed_pseudogene   \n",
       "\n",
       "      EXON INTRON HGVSc HGVSp cDNA_position CDS_position Protein_position  \\\n",
       "0      3/3                              303          243               81   \n",
       "1      3/3                              544          484              162   \n",
       "2      3/3                              930          870              290   \n",
       "3                                                                           \n",
       "3             3/3                                                           \n",
       "...    ...    ...   ...   ...           ...          ...              ...   \n",
       "33487  1/6                              303          257               86   \n",
       "33487  1/6                              303          257               86   \n",
       "33487                                                                       \n",
       "33487                                                                       \n",
       "33488  1/9                               48                                 \n",
       "\n",
       "      Amino_acids   Codons Existing_variation DISTANCE STRAND FLAGS  \\\n",
       "0               S  tcA/tcG                                  1         \n",
       "1             T/A  Aca/Gca                                  1         \n",
       "2               S  tcT/tcC                                  1         \n",
       "3                                                   22     -1         \n",
       "3                                                          -1         \n",
       "...           ...      ...                ...      ...    ...   ...   \n",
       "33487         R/Q  cGg/cAg                                  1         \n",
       "33487         R/Q  cGg/cAg                                  1         \n",
       "33487                                             2900     -1         \n",
       "33487                                             2912     -1         \n",
       "33488                                                       1         \n",
       "\n",
       "      SYMBOL_SOURCE     HGNC_ID SOURCE ClinVar ClinVar_CLNSIG  \\\n",
       "0              HGNC  HGNC:14825                                 \n",
       "1              HGNC  HGNC:14825                                 \n",
       "2              HGNC  HGNC:14825                                 \n",
       "3              HGNC  HGNC:15079                                 \n",
       "3                                                               \n",
       "...             ...         ...    ...     ...            ...   \n",
       "33487          HGNC  HGNC:12381                                 \n",
       "33487          HGNC  HGNC:12381                                 \n",
       "33487          HGNC  HGNC:37467                                 \n",
       "33487          HGNC  HGNC:37467                                 \n",
       "33488                                                           \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar_CLNDN  \n",
       "0                                       \n",
       "1                                       \n",
       "2                                       \n",
       "3                                       \n",
       "3                                       \n",
       "...                  ...           ...  \n",
       "33487                                   \n",
       "33487                                   \n",
       "33487                                   \n",
       "33487                                   \n",
       "33488                                   \n",
       "\n",
       "[275979 rows x 54 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "########################################################### Required columns extraction from the CSQ column ####################\n",
    "vcf['Allele'] = vcf['csq'].str.split('|').str[0]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SYMBOL'] = vcf['csq'].str.split('|').str[3]\n",
    "vcf['Gene'] = vcf['csq'].str.split('|').str[4]\n",
    "vcf['Feature_type'] = vcf['csq'].str.split('|').str[5]\n",
    "vcf['Feature'] = vcf['csq'].str.split('|').str[6]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['HGVSc'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSp'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['cDNA_position'] = vcf['csq'].str.split('|').str[12]\n",
    "vcf['CDS_position'] = vcf['csq'].str.split('|').str[13]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['Existing_variation'] = vcf['csq'].str.split('|').str[17]\n",
    "vcf['DISTANCE'] = vcf['csq'].str.split('|').str[18]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['FLAGS'] = vcf['csq'].str.split('|').str[20]\n",
    "vcf['SYMBOL_SOURCE'] = vcf['csq'].str.split('|').str[21]\n",
    "vcf['HGNC_ID'] = vcf['csq'].str.split('|').str[22]\n",
    "vcf['SOURCE'] = vcf['csq'].str.split('|').str[23]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[24]\n",
    "vcf['ClinVar_CLNSIG'] = vcf['csq'].str.split('|').str[25]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[26]\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[27]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "b9243be3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>243</td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>544</td>\n",
       "      <td>484</td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>930</td>\n",
       "      <td>870</td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:15079</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000419394</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000423647</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000451548</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000615605</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2900</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619815</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2912</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33488</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000226918</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000437359</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>48</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>275979 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1     69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33488  chrY  21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1      ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33488  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "        RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0      102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1        0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2       66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "...    ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33488    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0         OR4F5  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1         OR4F5  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2         OR4F5  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...         ...                                                ...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33488            A|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                     csq Allele  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...      G   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...      G   \n",
       "2      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...      C   \n",
       "3      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...      C   \n",
       "3      C|intron_variant&non_coding_transcript_variant...      C   \n",
       "...                                                  ...    ...   \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "33488  A|non_coding_transcript_exon_variant|MODIFIER|...      A   \n",
       "\n",
       "                                        Consequence    IMPACT    SYMBOL  \\\n",
       "0                                synonymous_variant       LOW     OR4F5   \n",
       "1                                  missense_variant  MODERATE     OR4F5   \n",
       "2                                synonymous_variant       LOW     OR4F5   \n",
       "3                           downstream_gene_variant  MODIFIER    OR4F16   \n",
       "3      intron_variant&non_coding_transcript_variant  MODIFIER             \n",
       "...                                             ...       ...       ...   \n",
       "33487                              missense_variant  MODERATE     TSPY1   \n",
       "33487                              missense_variant  MODERATE     TSPY1   \n",
       "33487                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "33487                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "33488            non_coding_transcript_exon_variant  MODIFIER             \n",
       "\n",
       "                  Gene Feature_type          Feature                 BIOTYPE  \\\n",
       "0      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "1      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "2      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "3      ENSG00000284662   Transcript  ENST00000332831          protein_coding   \n",
       "3      ENSG00000230021   Transcript  ENST00000419394    processed_transcript   \n",
       "...                ...          ...              ...                     ...   \n",
       "33487  ENSG00000258992   Transcript  ENST00000423647          protein_coding   \n",
       "33487  ENSG00000258992   Transcript  ENST00000451548          protein_coding   \n",
       "33487  ENSG00000225516   Transcript  ENST00000615605    processed_transcript   \n",
       "33487  ENSG00000225516   Transcript  ENST00000619815    processed_transcript   \n",
       "33488  ENSG00000226918   Transcript  ENST00000437359  unprocessed_pseudogene   \n",
       "\n",
       "      EXON INTRON HGVSc HGVSp cDNA_position CDS_position Protein_position  \\\n",
       "0      3/3                              303          243               81   \n",
       "1      3/3                              544          484              162   \n",
       "2      3/3                              930          870              290   \n",
       "3                                                                           \n",
       "3             3/3                                                           \n",
       "...    ...    ...   ...   ...           ...          ...              ...   \n",
       "33487  1/6                              303          257               86   \n",
       "33487  1/6                              303          257               86   \n",
       "33487                                                                       \n",
       "33487                                                                       \n",
       "33488  1/9                               48                                 \n",
       "\n",
       "      Amino_acids   Codons Existing_variation DISTANCE STRAND FLAGS  \\\n",
       "0               S  tcA/tcG                                  1         \n",
       "1             T/A  Aca/Gca                                  1         \n",
       "2               S  tcT/tcC                                  1         \n",
       "3                                                   22     -1         \n",
       "3                                                          -1         \n",
       "...           ...      ...                ...      ...    ...   ...   \n",
       "33487         R/Q  cGg/cAg                                  1         \n",
       "33487         R/Q  cGg/cAg                                  1         \n",
       "33487                                             2900     -1         \n",
       "33487                                             2912     -1         \n",
       "33488                                                       1         \n",
       "\n",
       "      SYMBOL_SOURCE     HGNC_ID SOURCE ClinVar ClinVar_CLNSIG  \\\n",
       "0              HGNC  HGNC:14825                                 \n",
       "1              HGNC  HGNC:14825                                 \n",
       "2              HGNC  HGNC:14825                                 \n",
       "3              HGNC  HGNC:15079                                 \n",
       "3                                                               \n",
       "...             ...         ...    ...     ...            ...   \n",
       "33487          HGNC  HGNC:12381                                 \n",
       "33487          HGNC  HGNC:12381                                 \n",
       "33487          HGNC  HGNC:37467                                 \n",
       "33487          HGNC  HGNC:37467                                 \n",
       "33488                                                           \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar_CLNDN                         consequence  \n",
       "0                                                       synonymous_variant  \n",
       "1                                                         missense_variant  \n",
       "2                                                       synonymous_variant  \n",
       "3                                                  downstream_gene_variant  \n",
       "3                                                           intron_variant  \n",
       "...                  ...           ...                                 ...  \n",
       "33487                                                     missense_variant  \n",
       "33487                                                     missense_variant  \n",
       "33487                                                upstream_gene_variant  \n",
       "33487                                                upstream_gene_variant  \n",
       "33488                                   non_coding_transcript_exon_variant  \n",
       "\n",
       "[275979 rows x 55 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['consequence'] = vcf['Consequence'].apply(lambda x: x.split('&')[0] if isinstance(x, str) else x)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "8714c420",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>243</td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>544</td>\n",
       "      <td>484</td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>930</td>\n",
       "      <td>870</td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:15079</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000419394</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000423647</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000451548</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000615605</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2900</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33487</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619815</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2912</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33488</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000226918</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000437359</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>48</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>275979 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1     69897  rs200676709   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "3      chr1    685694            .   T   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33487  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "33488  chrY  21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1      ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2      ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "3      ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33487  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "33488  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "        RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0      102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1        0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2       66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "3       12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "...    ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33487  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "33488    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "      Gene Name                                                CSQ  \\\n",
       "0         OR4F5  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1         OR4F5  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2         OR4F5  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "3                C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...         ...                                                ...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33487     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "33488            A|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                     csq Allele  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...      G   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...      G   \n",
       "2      C|synonymous_variant|LOW|OR4F5|ENSG00000186092...      C   \n",
       "3      C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...      C   \n",
       "3      C|intron_variant&non_coding_transcript_variant...      C   \n",
       "...                                                  ...    ...   \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "33487  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "33487  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "33488  A|non_coding_transcript_exon_variant|MODIFIER|...      A   \n",
       "\n",
       "                                        Consequence    IMPACT    SYMBOL  \\\n",
       "0                                synonymous_variant       LOW     OR4F5   \n",
       "1                                  missense_variant  MODERATE     OR4F5   \n",
       "2                                synonymous_variant       LOW     OR4F5   \n",
       "3                           downstream_gene_variant  MODIFIER    OR4F16   \n",
       "3      intron_variant&non_coding_transcript_variant  MODIFIER             \n",
       "...                                             ...       ...       ...   \n",
       "33487                              missense_variant  MODERATE     TSPY1   \n",
       "33487                              missense_variant  MODERATE     TSPY1   \n",
       "33487                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "33487                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "33488            non_coding_transcript_exon_variant  MODIFIER             \n",
       "\n",
       "                  Gene Feature_type          Feature                 BIOTYPE  \\\n",
       "0      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "1      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "2      ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "3      ENSG00000284662   Transcript  ENST00000332831          protein_coding   \n",
       "3      ENSG00000230021   Transcript  ENST00000419394    processed_transcript   \n",
       "...                ...          ...              ...                     ...   \n",
       "33487  ENSG00000258992   Transcript  ENST00000423647          protein_coding   \n",
       "33487  ENSG00000258992   Transcript  ENST00000451548          protein_coding   \n",
       "33487  ENSG00000225516   Transcript  ENST00000615605    processed_transcript   \n",
       "33487  ENSG00000225516   Transcript  ENST00000619815    processed_transcript   \n",
       "33488  ENSG00000226918   Transcript  ENST00000437359  unprocessed_pseudogene   \n",
       "\n",
       "      EXON INTRON HGVSc HGVSp cDNA_position CDS_position Protein_position  \\\n",
       "0      3/3                              303          243               81   \n",
       "1      3/3                              544          484              162   \n",
       "2      3/3                              930          870              290   \n",
       "3                                                                           \n",
       "3             3/3                                                           \n",
       "...    ...    ...   ...   ...           ...          ...              ...   \n",
       "33487  1/6                              303          257               86   \n",
       "33487  1/6                              303          257               86   \n",
       "33487                                                                       \n",
       "33487                                                                       \n",
       "33488  1/9                               48                                 \n",
       "\n",
       "      Amino_acids   Codons Existing_variation DISTANCE STRAND FLAGS  \\\n",
       "0               S  tcA/tcG                                  1         \n",
       "1             T/A  Aca/Gca                                  1         \n",
       "2               S  tcT/tcC                                  1         \n",
       "3                                                   22     -1         \n",
       "3                                                          -1         \n",
       "...           ...      ...                ...      ...    ...   ...   \n",
       "33487         R/Q  cGg/cAg                                  1         \n",
       "33487         R/Q  cGg/cAg                                  1         \n",
       "33487                                             2900     -1         \n",
       "33487                                             2912     -1         \n",
       "33488                                                       1         \n",
       "\n",
       "      SYMBOL_SOURCE     HGNC_ID SOURCE ClinVar ClinVar_CLNSIG  \\\n",
       "0              HGNC  HGNC:14825                                 \n",
       "1              HGNC  HGNC:14825                                 \n",
       "2              HGNC  HGNC:14825                                 \n",
       "3              HGNC  HGNC:15079                                 \n",
       "3                                                               \n",
       "...             ...         ...    ...     ...            ...   \n",
       "33487          HGNC  HGNC:12381                                 \n",
       "33487          HGNC  HGNC:12381                                 \n",
       "33487          HGNC  HGNC:37467                                 \n",
       "33487          HGNC  HGNC:37467                                 \n",
       "33488                                                           \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar_CLNDN                         consequence  \n",
       "0                                                       synonymous variant  \n",
       "1                                                         missense variant  \n",
       "2                                                       synonymous variant  \n",
       "3                                                  downstream gene variant  \n",
       "3                                                           intron variant  \n",
       "...                  ...           ...                                 ...  \n",
       "33487                                                     missense variant  \n",
       "33487                                                     missense variant  \n",
       "33487                                                upstream gene variant  \n",
       "33487                                                upstream gene variant  \n",
       "33488                                   non coding transcript exon variant  \n",
       "\n",
       "[275979 rows x 55 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['consequence'] = vcf['consequence'].str.split(',').str[0]\n",
    "vcf['consequence'] = vcf['consequence'].str.replace('_', ' ')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "04cfade7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "7c6f54e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>243</td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>544</td>\n",
       "      <td>484</td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>930</td>\n",
       "      <td>870</td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:15079</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000419394</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275974</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000423647</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275975</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000451548</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275976</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000615605</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2900</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275977</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619815</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2912</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275978</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000226918</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000437359</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>48</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>275979 rows × 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0       chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1       chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2       chr1     69897  rs200676709   T   C    .   PASS   \n",
       "3       chr1    685694            .   T   C    .   PASS   \n",
       "4       chr1    685694            .   T   C    .   PASS   \n",
       "...      ...       ...          ...  ..  ..  ...    ...   \n",
       "275974  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275975  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275976  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275977  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275978  chrY  21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                     INFO   GT   GQ  SDP   DP  \\\n",
       "0       ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1       ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2       ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3       ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "4       ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "...                                                   ...  ...  ...  ...  ...   \n",
       "275974  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275975  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275976  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275977  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275978  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "         RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0       102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1         0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2        66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3        12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "4        12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "...     ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "275974  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275975  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275976  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275977  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275978    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "       Gene Name                                                CSQ  \\\n",
       "0          OR4F5  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1          OR4F5  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2          OR4F5  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3                 C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "4                 C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...          ...                                                ...   \n",
       "275974     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275975     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275976     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275977     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275978            A|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                      csq Allele  \\\n",
       "0       G|synonymous_variant|LOW|OR4F5|ENSG00000186092...      G   \n",
       "1       G|missense_variant|MODERATE|OR4F5|ENSG00000186...      G   \n",
       "2       C|synonymous_variant|LOW|OR4F5|ENSG00000186092...      C   \n",
       "3       C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...      C   \n",
       "4       C|intron_variant&non_coding_transcript_variant...      C   \n",
       "...                                                   ...    ...   \n",
       "275974  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "275975  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "275976  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "275977  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "275978  A|non_coding_transcript_exon_variant|MODIFIER|...      A   \n",
       "\n",
       "                                         Consequence    IMPACT    SYMBOL  \\\n",
       "0                                 synonymous_variant       LOW     OR4F5   \n",
       "1                                   missense_variant  MODERATE     OR4F5   \n",
       "2                                 synonymous_variant       LOW     OR4F5   \n",
       "3                            downstream_gene_variant  MODIFIER    OR4F16   \n",
       "4       intron_variant&non_coding_transcript_variant  MODIFIER             \n",
       "...                                              ...       ...       ...   \n",
       "275974                              missense_variant  MODERATE     TSPY1   \n",
       "275975                              missense_variant  MODERATE     TSPY1   \n",
       "275976                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "275977                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "275978            non_coding_transcript_exon_variant  MODIFIER             \n",
       "\n",
       "                   Gene Feature_type          Feature                 BIOTYPE  \\\n",
       "0       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "1       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "2       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "3       ENSG00000284662   Transcript  ENST00000332831          protein_coding   \n",
       "4       ENSG00000230021   Transcript  ENST00000419394    processed_transcript   \n",
       "...                 ...          ...              ...                     ...   \n",
       "275974  ENSG00000258992   Transcript  ENST00000423647          protein_coding   \n",
       "275975  ENSG00000258992   Transcript  ENST00000451548          protein_coding   \n",
       "275976  ENSG00000225516   Transcript  ENST00000615605    processed_transcript   \n",
       "275977  ENSG00000225516   Transcript  ENST00000619815    processed_transcript   \n",
       "275978  ENSG00000226918   Transcript  ENST00000437359  unprocessed_pseudogene   \n",
       "\n",
       "       EXON INTRON HGVSc HGVSp cDNA_position CDS_position Protein_position  \\\n",
       "0       3/3                              303          243               81   \n",
       "1       3/3                              544          484              162   \n",
       "2       3/3                              930          870              290   \n",
       "3                                                                            \n",
       "4              3/3                                                           \n",
       "...     ...    ...   ...   ...           ...          ...              ...   \n",
       "275974  1/6                              303          257               86   \n",
       "275975  1/6                              303          257               86   \n",
       "275976                                                                       \n",
       "275977                                                                       \n",
       "275978  1/9                               48                                 \n",
       "\n",
       "       Amino_acids   Codons Existing_variation DISTANCE STRAND FLAGS  \\\n",
       "0                S  tcA/tcG                                  1         \n",
       "1              T/A  Aca/Gca                                  1         \n",
       "2                S  tcT/tcC                                  1         \n",
       "3                                                    22     -1         \n",
       "4                                                           -1         \n",
       "...            ...      ...                ...      ...    ...   ...   \n",
       "275974         R/Q  cGg/cAg                                  1         \n",
       "275975         R/Q  cGg/cAg                                  1         \n",
       "275976                                             2900     -1         \n",
       "275977                                             2912     -1         \n",
       "275978                                                       1         \n",
       "\n",
       "       SYMBOL_SOURCE     HGNC_ID SOURCE ClinVar ClinVar_CLNSIG  \\\n",
       "0               HGNC  HGNC:14825                                 \n",
       "1               HGNC  HGNC:14825                                 \n",
       "2               HGNC  HGNC:14825                                 \n",
       "3               HGNC  HGNC:15079                                 \n",
       "4                                                                \n",
       "...              ...         ...    ...     ...            ...   \n",
       "275974          HGNC  HGNC:12381                                 \n",
       "275975          HGNC  HGNC:12381                                 \n",
       "275976          HGNC  HGNC:37467                                 \n",
       "275977          HGNC  HGNC:37467                                 \n",
       "275978                                                           \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar_CLNDN                         consequence  \\\n",
       "0                                                        synonymous variant   \n",
       "1                                                          missense variant   \n",
       "2                                                        synonymous variant   \n",
       "3                                                   downstream gene variant   \n",
       "4                                                            intron variant   \n",
       "...                   ...           ...                                 ...   \n",
       "275974                                                     missense variant   \n",
       "275975                                                     missense variant   \n",
       "275976                                                upstream gene variant   \n",
       "275977                                                upstream gene variant   \n",
       "275978                                   non coding transcript exon variant   \n",
       "\n",
       "       Consequence_score  \n",
       "0                   3/10  \n",
       "1                   7/10  \n",
       "2                   3/10  \n",
       "3                   2/10  \n",
       "4                   2/10  \n",
       "...                  ...  \n",
       "275974              7/10  \n",
       "275975              7/10  \n",
       "275976              2/10  \n",
       "275977              2/10  \n",
       "275978              2/10  \n",
       "\n",
       "[275979 rows x 56 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(vcf, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "e5feaff9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     IMPACT  IMPACT_score\n",
       "0      HIGH          10.0\n",
       "1  MODERATE           5.0\n",
       "2       LOW           2.5\n",
       "3  MODIFIER           1.5"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "be4f1ce4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>243</td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>544</td>\n",
       "      <td>484</td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>930</td>\n",
       "      <td>870</td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:15079</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000419394</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275974</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000423647</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275975</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000451548</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275976</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000615605</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2900</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275977</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619815</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2912</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275978</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000226918</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000437359</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>48</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>275979 rows × 57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0       chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1       chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2       chr1     69897  rs200676709   T   C    .   PASS   \n",
       "3       chr1    685694            .   T   C    .   PASS   \n",
       "4       chr1    685694            .   T   C    .   PASS   \n",
       "...      ...       ...          ...  ..  ..  ...    ...   \n",
       "275974  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275975  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275976  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275977  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275978  chrY  21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                     INFO   GT   GQ  SDP   DP  \\\n",
       "0       ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1       ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2       ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3       ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "4       ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "...                                                   ...  ...  ...  ...  ...   \n",
       "275974  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275975  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275976  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275977  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275978  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "         RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0       102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1         0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2        66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3        12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "4        12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "...     ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "275974  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275975  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275976  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275977  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275978    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "       Gene Name                                                CSQ  \\\n",
       "0          OR4F5  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1          OR4F5  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2          OR4F5  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3                 C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "4                 C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...          ...                                                ...   \n",
       "275974     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275975     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275976     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275977     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275978            A|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                      csq Allele  \\\n",
       "0       G|synonymous_variant|LOW|OR4F5|ENSG00000186092...      G   \n",
       "1       G|missense_variant|MODERATE|OR4F5|ENSG00000186...      G   \n",
       "2       C|synonymous_variant|LOW|OR4F5|ENSG00000186092...      C   \n",
       "3       C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...      C   \n",
       "4       C|intron_variant&non_coding_transcript_variant...      C   \n",
       "...                                                   ...    ...   \n",
       "275974  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "275975  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "275976  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "275977  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "275978  A|non_coding_transcript_exon_variant|MODIFIER|...      A   \n",
       "\n",
       "                                         Consequence    IMPACT    SYMBOL  \\\n",
       "0                                 synonymous_variant       LOW     OR4F5   \n",
       "1                                   missense_variant  MODERATE     OR4F5   \n",
       "2                                 synonymous_variant       LOW     OR4F5   \n",
       "3                            downstream_gene_variant  MODIFIER    OR4F16   \n",
       "4       intron_variant&non_coding_transcript_variant  MODIFIER             \n",
       "...                                              ...       ...       ...   \n",
       "275974                              missense_variant  MODERATE     TSPY1   \n",
       "275975                              missense_variant  MODERATE     TSPY1   \n",
       "275976                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "275977                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "275978            non_coding_transcript_exon_variant  MODIFIER             \n",
       "\n",
       "                   Gene Feature_type          Feature                 BIOTYPE  \\\n",
       "0       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "1       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "2       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "3       ENSG00000284662   Transcript  ENST00000332831          protein_coding   \n",
       "4       ENSG00000230021   Transcript  ENST00000419394    processed_transcript   \n",
       "...                 ...          ...              ...                     ...   \n",
       "275974  ENSG00000258992   Transcript  ENST00000423647          protein_coding   \n",
       "275975  ENSG00000258992   Transcript  ENST00000451548          protein_coding   \n",
       "275976  ENSG00000225516   Transcript  ENST00000615605    processed_transcript   \n",
       "275977  ENSG00000225516   Transcript  ENST00000619815    processed_transcript   \n",
       "275978  ENSG00000226918   Transcript  ENST00000437359  unprocessed_pseudogene   \n",
       "\n",
       "       EXON INTRON HGVSc HGVSp cDNA_position CDS_position Protein_position  \\\n",
       "0       3/3                              303          243               81   \n",
       "1       3/3                              544          484              162   \n",
       "2       3/3                              930          870              290   \n",
       "3                                                                            \n",
       "4              3/3                                                           \n",
       "...     ...    ...   ...   ...           ...          ...              ...   \n",
       "275974  1/6                              303          257               86   \n",
       "275975  1/6                              303          257               86   \n",
       "275976                                                                       \n",
       "275977                                                                       \n",
       "275978  1/9                               48                                 \n",
       "\n",
       "       Amino_acids   Codons Existing_variation DISTANCE STRAND FLAGS  \\\n",
       "0                S  tcA/tcG                                  1         \n",
       "1              T/A  Aca/Gca                                  1         \n",
       "2                S  tcT/tcC                                  1         \n",
       "3                                                    22     -1         \n",
       "4                                                           -1         \n",
       "...            ...      ...                ...      ...    ...   ...   \n",
       "275974         R/Q  cGg/cAg                                  1         \n",
       "275975         R/Q  cGg/cAg                                  1         \n",
       "275976                                             2900     -1         \n",
       "275977                                             2912     -1         \n",
       "275978                                                       1         \n",
       "\n",
       "       SYMBOL_SOURCE     HGNC_ID SOURCE ClinVar ClinVar_CLNSIG  \\\n",
       "0               HGNC  HGNC:14825                                 \n",
       "1               HGNC  HGNC:14825                                 \n",
       "2               HGNC  HGNC:14825                                 \n",
       "3               HGNC  HGNC:15079                                 \n",
       "4                                                                \n",
       "...              ...         ...    ...     ...            ...   \n",
       "275974          HGNC  HGNC:12381                                 \n",
       "275975          HGNC  HGNC:12381                                 \n",
       "275976          HGNC  HGNC:37467                                 \n",
       "275977          HGNC  HGNC:37467                                 \n",
       "275978                                                           \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar_CLNDN                         consequence  \\\n",
       "0                                                        synonymous variant   \n",
       "1                                                          missense variant   \n",
       "2                                                        synonymous variant   \n",
       "3                                                   downstream gene variant   \n",
       "4                                                            intron variant   \n",
       "...                   ...           ...                                 ...   \n",
       "275974                                                     missense variant   \n",
       "275975                                                     missense variant   \n",
       "275976                                                upstream gene variant   \n",
       "275977                                                upstream gene variant   \n",
       "275978                                   non coding transcript exon variant   \n",
       "\n",
       "       Consequence_score  IMPACT_score  \n",
       "0                   3/10           2.5  \n",
       "1                   7/10           5.0  \n",
       "2                   3/10           2.5  \n",
       "3                   2/10           1.5  \n",
       "4                   2/10           1.5  \n",
       "...                  ...           ...  \n",
       "275974              7/10           5.0  \n",
       "275975              7/10           5.0  \n",
       "275976              2/10           1.5  \n",
       "275977              2/10           1.5  \n",
       "275978              2/10           1.5  \n",
       "\n",
       "[275979 rows x 57 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "0ba7c174",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BARD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRIP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CDH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CDKN2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CHEK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MSH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MSH6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PMS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>EPCAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PTEN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>RAD51C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>RAD51D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PALB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>STK11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>RRSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>FANCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MRE11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MUTYH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NBN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>RAD50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>RECQL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>RINT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SLX4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>SMARCA4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>XRCC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>HOXB13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name\n",
       "0        ATM\n",
       "1      BARD1\n",
       "2      BRCA1\n",
       "3      BRCA2\n",
       "4      BRIP1\n",
       "5       CDH1\n",
       "6     CDKN2A\n",
       "7      CHEK2\n",
       "8       MSH2\n",
       "9       MSH6\n",
       "10      PMS2\n",
       "11     EPCAM\n",
       "12      PTEN\n",
       "13    RAD51C\n",
       "14    RAD51D\n",
       "15       NF1\n",
       "16     PALB2\n",
       "17     STK11\n",
       "18      TP53\n",
       "19       MRI\n",
       "20      RRSO\n",
       "21       RRM\n",
       "22     FANCC\n",
       "23     MRE11\n",
       "24     MUTYH\n",
       "25       NBN\n",
       "26     RAD50\n",
       "27     RECQL\n",
       "28     RINT1\n",
       "29      SLX4\n",
       "30   SMARCA4\n",
       "31     XRCC2\n",
       "32    HOXB13"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Hereditary_Cancer_New_genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "b0798e26",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>137</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>102</td>\n",
       "      <td>41</td>\n",
       "      <td>28.67%</td>\n",
       "      <td>1.5868E-14</td>\n",
       "      <td>99</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>243</td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>168</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.6424E-100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>145</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>544</td>\n",
       "      <td>484</td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "      <td>66</td>\n",
       "      <td>32</td>\n",
       "      <td>32.65%</td>\n",
       "      <td>1.1206E-11</td>\n",
       "      <td>52</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>930</td>\n",
       "      <td>870</td>\n",
       "      <td>290</td>\n",
       "      <td>S</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:14825</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>22</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:15079</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>9.5308E-3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td></td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000419394</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275974</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000423647</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275975</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|missense_variant|MODERATE|TSPY1|ENSG00000258...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>ENSG00000258992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000451548</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>303</td>\n",
       "      <td>257</td>\n",
       "      <td>86</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12381</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275976</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000615605</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2900</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275977</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>224</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>174</td>\n",
       "      <td>67</td>\n",
       "      <td>27.8%</td>\n",
       "      <td>3.236E-23</td>\n",
       "      <td>132</td>\n",
       "      <td>42</td>\n",
       "      <td>58</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY1</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...</td>\n",
       "      <td>A</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FAM197Y5</td>\n",
       "      <td>ENSG00000225516</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619815</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2912</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37467</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275978</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.8292E-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td></td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>A</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000226918</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000437359</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>48</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>275979 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0       chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1       chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2       chr1     69897  rs200676709   T   C    .   PASS   \n",
       "3       chr1    685694            .   T   C    .   PASS   \n",
       "4       chr1    685694            .   T   C    .   PASS   \n",
       "...      ...       ...          ...  ..  ..  ...    ...   \n",
       "275974  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275975  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275976  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275977  chrY   9467257  rs777840135   G   A    .   PASS   \n",
       "275978  chrY  21311315   rs17250535   T   A    .   PASS   \n",
       "\n",
       "                                                     INFO   GT   GQ  SDP   DP  \\\n",
       "0       ADP=143;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  137  143  143   \n",
       "1       ADP=168;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  168  168   \n",
       "2       ADP=98;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  109   98   98   \n",
       "3       ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "4       ADP=18;WT=0;HET=1;HOM=0;NC=0;CSQ=C|downstream_...  0/1   20   18   18   \n",
       "...                                                   ...  ...  ...  ...  ...   \n",
       "275974  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275975  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275976  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275977  ADP=241;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  224  241  241   \n",
       "275978  ADP=19;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.9319,0....  1/1  105   19   19   \n",
       "\n",
       "         RD   AD    FREQ         PVAL  RDF RDR  ADF ADR HET HOM      Zygosity  \\\n",
       "0       102   41  28.67%   1.5868E-14   99   3   33   8   1   0  Heterozygous   \n",
       "1         0  168    100%  1.6424E-100    0   0  145  23   0   1    Homozygous   \n",
       "2        66   32  32.65%   1.1206E-11   52  14   25   7   1   0  Heterozygous   \n",
       "3        12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "4        12    6  33.33%    9.5308E-3   12   0    6   0   1   0  Heterozygous   \n",
       "...     ...  ...     ...          ...  ...  ..  ...  ..  ..  ..           ...   \n",
       "275974  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275975  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275976  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275977  174   67   27.8%    3.236E-23  132  42   58   9   1   0  Heterozygous   \n",
       "275978    0   19    100%   2.8292E-11    0   0   17   2   0   1    Homozygous   \n",
       "\n",
       "       Gene Name                                                CSQ  \\\n",
       "0          OR4F5  G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1          OR4F5  G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2          OR4F5  C|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "3                 C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "4                 C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...   \n",
       "...          ...                                                ...   \n",
       "275974     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275975     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275976     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275977     TSPY1  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...   \n",
       "275978            A|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                      csq Allele  \\\n",
       "0       G|synonymous_variant|LOW|OR4F5|ENSG00000186092...      G   \n",
       "1       G|missense_variant|MODERATE|OR4F5|ENSG00000186...      G   \n",
       "2       C|synonymous_variant|LOW|OR4F5|ENSG00000186092...      C   \n",
       "3       C|downstream_gene_variant|MODIFIER|OR4F16|ENSG...      C   \n",
       "4       C|intron_variant&non_coding_transcript_variant...      C   \n",
       "...                                                   ...    ...   \n",
       "275974  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "275975  A|missense_variant|MODERATE|TSPY1|ENSG00000258...      A   \n",
       "275976  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "275977  A|upstream_gene_variant|MODIFIER|FAM197Y5|ENSG...      A   \n",
       "275978  A|non_coding_transcript_exon_variant|MODIFIER|...      A   \n",
       "\n",
       "                                         Consequence    IMPACT    SYMBOL  \\\n",
       "0                                 synonymous_variant       LOW     OR4F5   \n",
       "1                                   missense_variant  MODERATE     OR4F5   \n",
       "2                                 synonymous_variant       LOW     OR4F5   \n",
       "3                            downstream_gene_variant  MODIFIER    OR4F16   \n",
       "4       intron_variant&non_coding_transcript_variant  MODIFIER             \n",
       "...                                              ...       ...       ...   \n",
       "275974                              missense_variant  MODERATE     TSPY1   \n",
       "275975                              missense_variant  MODERATE     TSPY1   \n",
       "275976                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "275977                         upstream_gene_variant  MODIFIER  FAM197Y5   \n",
       "275978            non_coding_transcript_exon_variant  MODIFIER             \n",
       "\n",
       "                   Gene Feature_type          Feature                 BIOTYPE  \\\n",
       "0       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "1       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "2       ENSG00000186092   Transcript  ENST00000641515          protein_coding   \n",
       "3       ENSG00000284662   Transcript  ENST00000332831          protein_coding   \n",
       "4       ENSG00000230021   Transcript  ENST00000419394    processed_transcript   \n",
       "...                 ...          ...              ...                     ...   \n",
       "275974  ENSG00000258992   Transcript  ENST00000423647          protein_coding   \n",
       "275975  ENSG00000258992   Transcript  ENST00000451548          protein_coding   \n",
       "275976  ENSG00000225516   Transcript  ENST00000615605    processed_transcript   \n",
       "275977  ENSG00000225516   Transcript  ENST00000619815    processed_transcript   \n",
       "275978  ENSG00000226918   Transcript  ENST00000437359  unprocessed_pseudogene   \n",
       "\n",
       "       EXON INTRON HGVSc HGVSp cDNA_position CDS_position Protein_position  \\\n",
       "0       3/3                              303          243               81   \n",
       "1       3/3                              544          484              162   \n",
       "2       3/3                              930          870              290   \n",
       "3                                                                            \n",
       "4              3/3                                                           \n",
       "...     ...    ...   ...   ...           ...          ...              ...   \n",
       "275974  1/6                              303          257               86   \n",
       "275975  1/6                              303          257               86   \n",
       "275976                                                                       \n",
       "275977                                                                       \n",
       "275978  1/9                               48                                 \n",
       "\n",
       "       Amino_acids   Codons Existing_variation DISTANCE STRAND FLAGS  \\\n",
       "0                S  tcA/tcG                                  1         \n",
       "1              T/A  Aca/Gca                                  1         \n",
       "2                S  tcT/tcC                                  1         \n",
       "3                                                    22     -1         \n",
       "4                                                           -1         \n",
       "...            ...      ...                ...      ...    ...   ...   \n",
       "275974         R/Q  cGg/cAg                                  1         \n",
       "275975         R/Q  cGg/cAg                                  1         \n",
       "275976                                             2900     -1         \n",
       "275977                                             2912     -1         \n",
       "275978                                                       1         \n",
       "\n",
       "       SYMBOL_SOURCE     HGNC_ID SOURCE ClinVar ClinVar_CLNSIG  \\\n",
       "0               HGNC  HGNC:14825                                 \n",
       "1               HGNC  HGNC:14825                                 \n",
       "2               HGNC  HGNC:14825                                 \n",
       "3               HGNC  HGNC:15079                                 \n",
       "4                                                                \n",
       "...              ...         ...    ...     ...            ...   \n",
       "275974          HGNC  HGNC:12381                                 \n",
       "275975          HGNC  HGNC:12381                                 \n",
       "275976          HGNC  HGNC:37467                                 \n",
       "275977          HGNC  HGNC:37467                                 \n",
       "275978                                                           \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar_CLNDN                         consequence  \\\n",
       "0                                                        synonymous variant   \n",
       "1                                                          missense variant   \n",
       "2                                                        synonymous variant   \n",
       "3                                                   downstream gene variant   \n",
       "4                                                            intron variant   \n",
       "...                   ...           ...                                 ...   \n",
       "275974                                                     missense variant   \n",
       "275975                                                     missense variant   \n",
       "275976                                                upstream gene variant   \n",
       "275977                                                upstream gene variant   \n",
       "275978                                   non coding transcript exon variant   \n",
       "\n",
       "       Consequence_score  IMPACT_score Gene_Match  \n",
       "0                   3/10           2.5         No  \n",
       "1                   7/10           5.0         No  \n",
       "2                   3/10           2.5         No  \n",
       "3                   2/10           1.5         No  \n",
       "4                   2/10           1.5         No  \n",
       "...                  ...           ...        ...  \n",
       "275974              7/10           5.0         No  \n",
       "275975              7/10           5.0         No  \n",
       "275976              2/10           1.5         No  \n",
       "275977              2/10           1.5         No  \n",
       "275978              2/10           1.5         No  \n",
       "\n",
       "[275979 rows x 58 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in merged_2['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            merged_2.loc[merged_2['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "4cf9ecff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10581</th>\n",
       "      <td>chr1</td>\n",
       "      <td>45331833</td>\n",
       "      <td>rs3219489</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>53.12%</td>\n",
       "      <td>4.1013E-7</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...</td>\n",
       "      <td>G</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>HPDL</td>\n",
       "      <td>ENSG00000186603</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000334815</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3123</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:28242</td>\n",
       "      <td></td>\n",
       "      <td>41767</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>Carcinoma_of_colon&amp;MUTYH-related_attenuated_fa...</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10582</th>\n",
       "      <td>chr1</td>\n",
       "      <td>45331833</td>\n",
       "      <td>rs3219489</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>53.12%</td>\n",
       "      <td>4.1013E-7</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...</td>\n",
       "      <td>G|missense_variant|MODERATE|MUTYH|ENSG00000132...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>ENSG00000132781</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000354383</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/16</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1044</td>\n",
       "      <td>933</td>\n",
       "      <td>311</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caG/caC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:7527</td>\n",
       "      <td></td>\n",
       "      <td>41767</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>Carcinoma_of_colon&amp;MUTYH-related_attenuated_fa...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10583</th>\n",
       "      <td>chr1</td>\n",
       "      <td>45331833</td>\n",
       "      <td>rs3219489</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>53.12%</td>\n",
       "      <td>4.1013E-7</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...</td>\n",
       "      <td>G|missense_variant|MODERATE|MUTYH|ENSG00000132...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>ENSG00000132781</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000355498</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/16</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1076</td>\n",
       "      <td>930</td>\n",
       "      <td>310</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caG/caC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:7527</td>\n",
       "      <td></td>\n",
       "      <td>41767</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>Carcinoma_of_colon&amp;MUTYH-related_attenuated_fa...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10584</th>\n",
       "      <td>chr1</td>\n",
       "      <td>45331833</td>\n",
       "      <td>rs3219489</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>53.12%</td>\n",
       "      <td>4.1013E-7</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...</td>\n",
       "      <td>G|missense_variant|MODERATE|MUTYH|ENSG00000132...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>ENSG00000132781</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000372098</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/16</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1139</td>\n",
       "      <td>1005</td>\n",
       "      <td>335</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caG/caC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:7527</td>\n",
       "      <td></td>\n",
       "      <td>41767</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>Carcinoma_of_colon&amp;MUTYH-related_attenuated_fa...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10585</th>\n",
       "      <td>chr1</td>\n",
       "      <td>45331833</td>\n",
       "      <td>rs3219489</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>63</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>53.12%</td>\n",
       "      <td>4.1013E-7</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...</td>\n",
       "      <td>G|missense_variant|MODERATE|MUTYH|ENSG00000132...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>ENSG00000132781</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000372104</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13/17</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1123</td>\n",
       "      <td>930</td>\n",
       "      <td>310</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caG/caC</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:7527</td>\n",
       "      <td></td>\n",
       "      <td>41767</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>Carcinoma_of_colon&amp;MUTYH-related_attenuated_fa...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266041</th>\n",
       "      <td>chr9</td>\n",
       "      <td>95099836</td>\n",
       "      <td>rs4647558</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=59;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5865,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>98</td>\n",
       "      <td>59</td>\n",
       "      <td>59</td>\n",
       "      <td>32</td>\n",
       "      <td>27</td>\n",
       "      <td>45.76%</td>\n",
       "      <td>1.5211E-10</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCC,C9orf3</td>\n",
       "      <td>T|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>T</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FANCC</td>\n",
       "      <td>ENSG00000158169</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000696260</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>4363</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:3584</td>\n",
       "      <td></td>\n",
       "      <td>367581</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>Fanconi_anemia_complementation_group_C</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266042</th>\n",
       "      <td>chr9</td>\n",
       "      <td>95100419</td>\n",
       "      <td>rs4647554</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>45</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>40.62%</td>\n",
       "      <td>2.6443E-5</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCC,C9orf3</td>\n",
       "      <td>G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...</td>\n",
       "      <td>G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...</td>\n",
       "      <td>G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FANCC</td>\n",
       "      <td>ENSG00000158169</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000289081</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/15</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:3584</td>\n",
       "      <td></td>\n",
       "      <td>367589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>Fanconi_anemia_complementation_group_C</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266043</th>\n",
       "      <td>chr9</td>\n",
       "      <td>95100419</td>\n",
       "      <td>rs4647554</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>45</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>40.62%</td>\n",
       "      <td>2.6443E-5</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCC,C9orf3</td>\n",
       "      <td>G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...</td>\n",
       "      <td>G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...</td>\n",
       "      <td>G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FANCC</td>\n",
       "      <td>ENSG00000158169</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000375305</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/15</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3169</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:3584</td>\n",
       "      <td></td>\n",
       "      <td>367589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>Fanconi_anemia_complementation_group_C</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266044</th>\n",
       "      <td>chr9</td>\n",
       "      <td>95100419</td>\n",
       "      <td>rs4647554</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>45</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>40.62%</td>\n",
       "      <td>2.6443E-5</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCC,C9orf3</td>\n",
       "      <td>G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|FANCC|ENSG0...</td>\n",
       "      <td>G</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FANCC</td>\n",
       "      <td>ENSG00000158169</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000649334</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1291</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:3584</td>\n",
       "      <td></td>\n",
       "      <td>367589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>Fanconi_anemia_complementation_group_C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266045</th>\n",
       "      <td>chr9</td>\n",
       "      <td>95100419</td>\n",
       "      <td>rs4647554</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>45</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>40.62%</td>\n",
       "      <td>2.6443E-5</td>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCC,C9orf3</td>\n",
       "      <td>G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...</td>\n",
       "      <td>G|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>G</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>FANCC</td>\n",
       "      <td>ENSG00000158169</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000696260</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3780</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:3584</td>\n",
       "      <td></td>\n",
       "      <td>367589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria_provided&amp;_single_submitter</td>\n",
       "      <td>Fanconi_anemia_complementation_group_C</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1726 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "10581   chr1  45331833  rs3219489   C   G    .   PASS   \n",
       "10582   chr1  45331833  rs3219489   C   G    .   PASS   \n",
       "10583   chr1  45331833  rs3219489   C   G    .   PASS   \n",
       "10584   chr1  45331833  rs3219489   C   G    .   PASS   \n",
       "10585   chr1  45331833  rs3219489   C   G    .   PASS   \n",
       "...      ...       ...        ...  ..  ..  ...    ...   \n",
       "266041  chr9  95099836  rs4647558   C   T    .   PASS   \n",
       "266042  chr9  95100419  rs4647554   A   G    .   PASS   \n",
       "266043  chr9  95100419  rs4647554   A   G    .   PASS   \n",
       "266044  chr9  95100419  rs4647554   A   G    .   PASS   \n",
       "266045  chr9  95100419  rs4647554   A   G    .   PASS   \n",
       "\n",
       "                                                     INFO   GT  GQ SDP  DP  \\\n",
       "10581   ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....  0/1  63  32  32   \n",
       "10582   ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....  0/1  63  32  32   \n",
       "10583   ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....  0/1  63  32  32   \n",
       "10584   ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....  0/1  63  32  32   \n",
       "10585   ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.6865,0....  0/1  63  32  32   \n",
       "...                                                   ...  ...  ..  ..  ..   \n",
       "266041  ADP=59;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.5865,0....  0/1  98  59  59   \n",
       "266042  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....  0/1  45  32  32   \n",
       "266043  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....  0/1  45  32  32   \n",
       "266044  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....  0/1  45  32  32   \n",
       "266045  ADP=32;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.4972,0....  0/1  45  32  32   \n",
       "\n",
       "        RD  AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "10581   15  17  53.12%   4.1013E-7  13   2  15   2   1   0  Heterozygous   \n",
       "10582   15  17  53.12%   4.1013E-7  13   2  15   2   1   0  Heterozygous   \n",
       "10583   15  17  53.12%   4.1013E-7  13   2  15   2   1   0  Heterozygous   \n",
       "10584   15  17  53.12%   4.1013E-7  13   2  15   2   1   0  Heterozygous   \n",
       "10585   15  17  53.12%   4.1013E-7  13   2  15   2   1   0  Heterozygous   \n",
       "...     ..  ..     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "266041  32  27  45.76%  1.5211E-10  24   8  23   4   1   0  Heterozygous   \n",
       "266042  19  13  40.62%   2.6443E-5  15   4  11   2   1   0  Heterozygous   \n",
       "266043  19  13  40.62%   2.6443E-5  15   4  11   2   1   0  Heterozygous   \n",
       "266044  19  13  40.62%   2.6443E-5  15   4  11   2   1   0  Heterozygous   \n",
       "266045  19  13  40.62%   2.6443E-5  15   4  11   2   1   0  Heterozygous   \n",
       "\n",
       "           Gene Name                                                CSQ  \\\n",
       "10581          MUTYH  G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...   \n",
       "10582          MUTYH  G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...   \n",
       "10583          MUTYH  G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...   \n",
       "10584          MUTYH  G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...   \n",
       "10585          MUTYH  G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...   \n",
       "...              ...                                                ...   \n",
       "266041  FANCC,C9orf3  T|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...   \n",
       "266042  FANCC,C9orf3  G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...   \n",
       "266043  FANCC,C9orf3  G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...   \n",
       "266044  FANCC,C9orf3  G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...   \n",
       "266045  FANCC,C9orf3  G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...   \n",
       "\n",
       "                                                      csq Allele  \\\n",
       "10581   G|downstream_gene_variant|MODIFIER|HPDL|ENSG00...      G   \n",
       "10582   G|missense_variant|MODERATE|MUTYH|ENSG00000132...      G   \n",
       "10583   G|missense_variant|MODERATE|MUTYH|ENSG00000132...      G   \n",
       "10584   G|missense_variant|MODERATE|MUTYH|ENSG00000132...      G   \n",
       "10585   G|missense_variant|MODERATE|MUTYH|ENSG00000132...      G   \n",
       "...                                                   ...    ...   \n",
       "266041  T|non_coding_transcript_exon_variant|MODIFIER|...      T   \n",
       "266042  G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...      G   \n",
       "266043  G|3_prime_UTR_variant|MODIFIER|FANCC|ENSG00000...      G   \n",
       "266044  G|downstream_gene_variant|MODIFIER|FANCC|ENSG0...      G   \n",
       "266045  G|non_coding_transcript_exon_variant|MODIFIER|...      G   \n",
       "\n",
       "                               Consequence    IMPACT SYMBOL             Gene  \\\n",
       "10581              downstream_gene_variant  MODIFIER   HPDL  ENSG00000186603   \n",
       "10582                     missense_variant  MODERATE  MUTYH  ENSG00000132781   \n",
       "10583                     missense_variant  MODERATE  MUTYH  ENSG00000132781   \n",
       "10584                     missense_variant  MODERATE  MUTYH  ENSG00000132781   \n",
       "10585                     missense_variant  MODERATE  MUTYH  ENSG00000132781   \n",
       "...                                    ...       ...    ...              ...   \n",
       "266041  non_coding_transcript_exon_variant  MODIFIER  FANCC  ENSG00000158169   \n",
       "266042                 3_prime_UTR_variant  MODIFIER  FANCC  ENSG00000158169   \n",
       "266043                 3_prime_UTR_variant  MODIFIER  FANCC  ENSG00000158169   \n",
       "266044             downstream_gene_variant  MODIFIER  FANCC  ENSG00000158169   \n",
       "266045  non_coding_transcript_exon_variant  MODIFIER  FANCC  ENSG00000158169   \n",
       "\n",
       "       Feature_type          Feature                  BIOTYPE   EXON INTRON  \\\n",
       "10581    Transcript  ENST00000334815           protein_coding                 \n",
       "10582    Transcript  ENST00000354383           protein_coding  12/16          \n",
       "10583    Transcript  ENST00000355498           protein_coding  12/16          \n",
       "10584    Transcript  ENST00000372098           protein_coding  12/16          \n",
       "10585    Transcript  ENST00000372104           protein_coding  13/17          \n",
       "...             ...              ...                      ...    ...    ...   \n",
       "266041   Transcript  ENST00000696260          retained_intron    3/3          \n",
       "266042   Transcript  ENST00000289081           protein_coding  15/15          \n",
       "266043   Transcript  ENST00000375305           protein_coding  15/15          \n",
       "266044   Transcript  ENST00000649334  nonsense_mediated_decay                 \n",
       "266045   Transcript  ENST00000696260          retained_intron    3/3          \n",
       "\n",
       "       HGVSc HGVSp cDNA_position CDS_position Protein_position Amino_acids  \\\n",
       "10581                                                                        \n",
       "10582                       1044          933              311         Q/H   \n",
       "10583                       1076          930              310         Q/H   \n",
       "10584                       1139         1005              335         Q/H   \n",
       "10585                       1123          930              310         Q/H   \n",
       "...      ...   ...           ...          ...              ...         ...   \n",
       "266041                      4363                                             \n",
       "266042                      3227                                             \n",
       "266043                      3169                                             \n",
       "266044                                                                       \n",
       "266045                      3780                                             \n",
       "\n",
       "         Codons Existing_variation DISTANCE STRAND FLAGS SYMBOL_SOURCE  \\\n",
       "10581                                  3123      1                HGNC   \n",
       "10582   caG/caC                                 -1                HGNC   \n",
       "10583   caG/caC                                 -1                HGNC   \n",
       "10584   caG/caC                                 -1                HGNC   \n",
       "10585   caG/caC                                 -1                HGNC   \n",
       "...         ...                ...      ...    ...   ...           ...   \n",
       "266041                                          -1                HGNC   \n",
       "266042                                          -1                HGNC   \n",
       "266043                                          -1                HGNC   \n",
       "266044                                 1291     -1                HGNC   \n",
       "266045                                          -1                HGNC   \n",
       "\n",
       "           HGNC_ID SOURCE ClinVar ClinVar_CLNSIG  \\\n",
       "10581   HGNC:28242          41767         Benign   \n",
       "10582    HGNC:7527          41767         Benign   \n",
       "10583    HGNC:7527          41767         Benign   \n",
       "10584    HGNC:7527          41767         Benign   \n",
       "10585    HGNC:7527          41767         Benign   \n",
       "...            ...    ...     ...            ...   \n",
       "266041   HGNC:3584         367581         Benign   \n",
       "266042   HGNC:3584         367589         Benign   \n",
       "266043   HGNC:3584         367589         Benign   \n",
       "266044   HGNC:3584         367589         Benign   \n",
       "266045   HGNC:3584         367589         Benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT  \\\n",
       "10581   criteria_provided&_multiple_submitters&_no_con...   \n",
       "10582   criteria_provided&_multiple_submitters&_no_con...   \n",
       "10583   criteria_provided&_multiple_submitters&_no_con...   \n",
       "10584   criteria_provided&_multiple_submitters&_no_con...   \n",
       "10585   criteria_provided&_multiple_submitters&_no_con...   \n",
       "...                                                   ...   \n",
       "266041                criteria_provided&_single_submitter   \n",
       "266042                criteria_provided&_single_submitter   \n",
       "266043                criteria_provided&_single_submitter   \n",
       "266044                criteria_provided&_single_submitter   \n",
       "266045                criteria_provided&_single_submitter   \n",
       "\n",
       "                                            ClinVar_CLNDN  \\\n",
       "10581   Carcinoma_of_colon&MUTYH-related_attenuated_fa...   \n",
       "10582   Carcinoma_of_colon&MUTYH-related_attenuated_fa...   \n",
       "10583   Carcinoma_of_colon&MUTYH-related_attenuated_fa...   \n",
       "10584   Carcinoma_of_colon&MUTYH-related_attenuated_fa...   \n",
       "10585   Carcinoma_of_colon&MUTYH-related_attenuated_fa...   \n",
       "...                                                   ...   \n",
       "266041             Fanconi_anemia_complementation_group_C   \n",
       "266042             Fanconi_anemia_complementation_group_C   \n",
       "266043             Fanconi_anemia_complementation_group_C   \n",
       "266044             Fanconi_anemia_complementation_group_C   \n",
       "266045             Fanconi_anemia_complementation_group_C   \n",
       "\n",
       "                               consequence Consequence_score  IMPACT_score  \\\n",
       "10581              downstream gene variant              2/10           1.5   \n",
       "10582                     missense variant              7/10           5.0   \n",
       "10583                     missense variant              7/10           5.0   \n",
       "10584                     missense variant              7/10           5.0   \n",
       "10585                     missense variant              7/10           5.0   \n",
       "...                                    ...               ...           ...   \n",
       "266041  non coding transcript exon variant              2/10           1.5   \n",
       "266042                 3 prime UTR variant              3/10           1.5   \n",
       "266043                 3 prime UTR variant              3/10           1.5   \n",
       "266044             downstream gene variant              2/10           1.5   \n",
       "266045  non coding transcript exon variant              2/10           1.5   \n",
       "\n",
       "       Gene_Match  \n",
       "10581         Yes  \n",
       "10582         Yes  \n",
       "10583         Yes  \n",
       "10584         Yes  \n",
       "10585         Yes  \n",
       "...           ...  \n",
       "266041        Yes  \n",
       "266042        Yes  \n",
       "266043        Yes  \n",
       "266044        Yes  \n",
       "266045        Yes  \n",
       "\n",
       "[1726 rows x 58 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2[merged_2['Gene_Match'] == 'Yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "f11fb7c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_2.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHGLBS448_final_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d322346",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
