{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "6a1cb0e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "0845e0c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Downloads/BED_files/kalyani_mam_covered.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/Schizophrenia_Annotations.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/Schizophrenia_Annotations.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b7a25dba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27518370</td>\n",
       "      <td>rs780094</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=780094;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27519736</td>\n",
       "      <td>rs780093</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=780093;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>87310240</td>\n",
       "      <td>rs72613567</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=72613567;dbSNPBuildID=130;SSR=0;GENEINFO=HS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>87310277</td>\n",
       "      <td>rs62305723</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=62305723;dbSNPBuildID=129;SSR=0;GENEINFO=HS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr19</td>\n",
       "      <td>19268740</td>\n",
       "      <td>rs58542926</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=58542926;dbSNPBuildID=129;SSR=0;GENEINFO=TM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43928847</td>\n",
       "      <td>rs738409</td>\n",
       "      <td>C</td>\n",
       "      <td>A,G,T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=738409;dbSNPBuildID=86;SSR=0;GENEINFO=PNPLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43930116</td>\n",
       "      <td>rs12483959</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C,T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=12483959;dbSNPBuildID=120;SSR=0;GENEINFO=PN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43932850</td>\n",
       "      <td>rs4823173</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=4823173;dbSNPBuildID=111;SSR=0;GENEINFO=PNP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43933050</td>\n",
       "      <td>rs143392071</td>\n",
       "      <td>A</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=143392071;dbSNPBuildID=134;SSR=0;GENEINFO=P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43936690</td>\n",
       "      <td>rs2281135</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2281135;dbSNPBuildID=100;SSR=0;GENEINFO=PNP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43937814</td>\n",
       "      <td>rs2896019</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2896019;dbSNPBuildID=101;SSR=0;GENEINFO=PNP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43942254</td>\n",
       "      <td>rs3810622</td>\n",
       "      <td>T</td>\n",
       "      <td>A,C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=3810622;dbSNPBuildID=107;SSR=0;GENEINFO=PNP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS         rsID REF    ALT QUAL FILTER  \\\n",
       "0    chr2  27508073    rs1260326   T    C,G    .      .   \n",
       "1    chr2  27518370     rs780094   T      C    .      .   \n",
       "2    chr2  27519736     rs780093   T      C    .      .   \n",
       "3    chr4  87310240   rs72613567   T     TA    .      .   \n",
       "4    chr4  87310277   rs62305723   G      A    .      .   \n",
       "5   chr19  19268740   rs58542926   C    A,T    .      .   \n",
       "6   chr22  43928847     rs738409   C  A,G,T    .      .   \n",
       "7   chr22  43930116   rs12483959   G  A,C,T    .      .   \n",
       "8   chr22  43932850    rs4823173   G      A    .      .   \n",
       "9   chr22  43933050  rs143392071   A    C,G    .      .   \n",
       "10  chr22  43936690    rs2281135   G    A,C    .      .   \n",
       "11  chr22  43937814    rs2896019   T      G    .      .   \n",
       "12  chr22  43942254    rs3810622   T  A,C,G    .      .   \n",
       "13  chr22  43995806    rs2143571   G    A,C    .      .   \n",
       "\n",
       "                                                 INFO  \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...  \n",
       "1   RS=780094;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...  \n",
       "2   RS=780093;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...  \n",
       "3   RS=72613567;dbSNPBuildID=130;SSR=0;GENEINFO=HS...  \n",
       "4   RS=62305723;dbSNPBuildID=129;SSR=0;GENEINFO=HS...  \n",
       "5   RS=58542926;dbSNPBuildID=129;SSR=0;GENEINFO=TM...  \n",
       "6   RS=738409;dbSNPBuildID=86;SSR=0;GENEINFO=PNPLA...  \n",
       "7   RS=12483959;dbSNPBuildID=120;SSR=0;GENEINFO=PN...  \n",
       "8   RS=4823173;dbSNPBuildID=111;SSR=0;GENEINFO=PNP...  \n",
       "9   RS=143392071;dbSNPBuildID=134;SSR=0;GENEINFO=P...  \n",
       "10  RS=2281135;dbSNPBuildID=100;SSR=0;GENEINFO=PNP...  \n",
       "11  RS=2896019;dbSNPBuildID=101;SSR=0;GENEINFO=PNP...  \n",
       "12  RS=3810622;dbSNPBuildID=107;SSR=0;GENEINFO=PNP...  \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/Fatty_liver_Annotations.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO']#, 'FORMAT', 'SAMPLE']\n",
    "\n",
    "#sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "#sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "#vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "#vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "351e51e6",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'GT'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3802\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3801\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:138\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\index.pyx:165\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5745\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5753\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'GT'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[5], line 9\u001b[0m\n\u001b[0;32m      7\u001b[0m vcf\u001b[38;5;241m.\u001b[39mloc[vcf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mHOM\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m1\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mZygosity\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mHomozygous\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m      8\u001b[0m vcf\u001b[38;5;241m.\u001b[39mloc[vcf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mHET\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m1\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mZygosity\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mHeterozygous\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m----> 9\u001b[0m vcf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGT\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mvcf\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mGT\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241m.\u001b[39mastype(\u001b[38;5;28mstr\u001b[39m)\n\u001b[0;32m     10\u001b[0m vcf\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py:3807\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   3805\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[0;32m   3806\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[1;32m-> 3807\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   3808\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[0;32m   3809\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py:3804\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m-> 3804\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[0;32m   3805\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[0;32m   3806\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[0;32m   3807\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[0;32m   3808\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[0;32m   3809\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[1;31mKeyError\u001b[0m: 'GT'"
     ]
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c4e53043",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27518370</td>\n",
       "      <td>rs780094</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=780094;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27519736</td>\n",
       "      <td>rs780093</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=780093;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>87310240</td>\n",
       "      <td>rs72613567</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=72613567;dbSNPBuildID=130;SSR=0;GENEINFO=HS...</td>\n",
       "      <td>HSD17B13:345275</td>\n",
       "      <td>HSD17B13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>87310277</td>\n",
       "      <td>rs62305723</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=62305723;dbSNPBuildID=129;SSR=0;GENEINFO=HS...</td>\n",
       "      <td>HSD17B13:345275</td>\n",
       "      <td>HSD17B13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr19</td>\n",
       "      <td>19268740</td>\n",
       "      <td>rs58542926</td>\n",
       "      <td>C</td>\n",
       "      <td>A,T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=58542926;dbSNPBuildID=129;SSR=0;GENEINFO=TM...</td>\n",
       "      <td>TM6SF2:53345</td>\n",
       "      <td>TM6SF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43928847</td>\n",
       "      <td>rs738409</td>\n",
       "      <td>C</td>\n",
       "      <td>A,G,T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=738409;dbSNPBuildID=86;SSR=0;GENEINFO=PNPLA...</td>\n",
       "      <td>PNPLA3:80339</td>\n",
       "      <td>PNPLA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43930116</td>\n",
       "      <td>rs12483959</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C,T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=12483959;dbSNPBuildID=120;SSR=0;GENEINFO=PN...</td>\n",
       "      <td>PNPLA3:80339</td>\n",
       "      <td>PNPLA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43932850</td>\n",
       "      <td>rs4823173</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=4823173;dbSNPBuildID=111;SSR=0;GENEINFO=PNP...</td>\n",
       "      <td>PNPLA3:80339</td>\n",
       "      <td>PNPLA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43933050</td>\n",
       "      <td>rs143392071</td>\n",
       "      <td>A</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=143392071;dbSNPBuildID=134;SSR=0;GENEINFO=P...</td>\n",
       "      <td>PNPLA3:80339</td>\n",
       "      <td>PNPLA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43936690</td>\n",
       "      <td>rs2281135</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2281135;dbSNPBuildID=100;SSR=0;GENEINFO=PNP...</td>\n",
       "      <td>PNPLA3:80339</td>\n",
       "      <td>PNPLA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43937814</td>\n",
       "      <td>rs2896019</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2896019;dbSNPBuildID=101;SSR=0;GENEINFO=PNP...</td>\n",
       "      <td>PNPLA3:80339</td>\n",
       "      <td>PNPLA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43942254</td>\n",
       "      <td>rs3810622</td>\n",
       "      <td>T</td>\n",
       "      <td>A,C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=3810622;dbSNPBuildID=107;SSR=0;GENEINFO=PNP...</td>\n",
       "      <td>PNPLA3:80339</td>\n",
       "      <td>PNPLA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS         rsID REF    ALT QUAL FILTER  \\\n",
       "0    chr2  27508073    rs1260326   T    C,G    .      .   \n",
       "1    chr2  27518370     rs780094   T      C    .      .   \n",
       "2    chr2  27519736     rs780093   T      C    .      .   \n",
       "3    chr4  87310240   rs72613567   T     TA    .      .   \n",
       "4    chr4  87310277   rs62305723   G      A    .      .   \n",
       "5   chr19  19268740   rs58542926   C    A,T    .      .   \n",
       "6   chr22  43928847     rs738409   C  A,G,T    .      .   \n",
       "7   chr22  43930116   rs12483959   G  A,C,T    .      .   \n",
       "8   chr22  43932850    rs4823173   G      A    .      .   \n",
       "9   chr22  43933050  rs143392071   A    C,G    .      .   \n",
       "10  chr22  43936690    rs2281135   G    A,C    .      .   \n",
       "11  chr22  43937814    rs2896019   T      G    .      .   \n",
       "12  chr22  43942254    rs3810622   T  A,C,G    .      .   \n",
       "13  chr22  43995806    rs2143571   G    A,C    .      .   \n",
       "\n",
       "                                                 INFO        Gene_Name  \\\n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...        GCKR:2646   \n",
       "1   RS=780094;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...        GCKR:2646   \n",
       "2   RS=780093;dbSNPBuildID=86;SSR=0;GENEINFO=GCKR:...        GCKR:2646   \n",
       "3   RS=72613567;dbSNPBuildID=130;SSR=0;GENEINFO=HS...  HSD17B13:345275   \n",
       "4   RS=62305723;dbSNPBuildID=129;SSR=0;GENEINFO=HS...  HSD17B13:345275   \n",
       "5   RS=58542926;dbSNPBuildID=129;SSR=0;GENEINFO=TM...     TM6SF2:53345   \n",
       "6   RS=738409;dbSNPBuildID=86;SSR=0;GENEINFO=PNPLA...     PNPLA3:80339   \n",
       "7   RS=12483959;dbSNPBuildID=120;SSR=0;GENEINFO=PN...     PNPLA3:80339   \n",
       "8   RS=4823173;dbSNPBuildID=111;SSR=0;GENEINFO=PNP...     PNPLA3:80339   \n",
       "9   RS=143392071;dbSNPBuildID=134;SSR=0;GENEINFO=P...     PNPLA3:80339   \n",
       "10  RS=2281135;dbSNPBuildID=100;SSR=0;GENEINFO=PNP...     PNPLA3:80339   \n",
       "11  RS=2896019;dbSNPBuildID=101;SSR=0;GENEINFO=PNP...     PNPLA3:80339   \n",
       "12  RS=3810622;dbSNPBuildID=107;SSR=0;GENEINFO=PNP...     PNPLA3:80339   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...     SAMM50:25813   \n",
       "\n",
       "   Gene Name  \n",
       "0       GCKR  \n",
       "1       GCKR  \n",
       "2       GCKR  \n",
       "3   HSD17B13  \n",
       "4   HSD17B13  \n",
       "5     TM6SF2  \n",
       "6     PNPLA3  \n",
       "7     PNPLA3  \n",
       "8     PNPLA3  \n",
       "9     PNPLA3  \n",
       "10    PNPLA3  \n",
       "11    PNPLA3  \n",
       "12    PNPLA3  \n",
       "13    SAMM50  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene_Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "#vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join(set([segment.split(':')[0] for segment in x.split('|')])) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8c3ae216",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS       rsID REF  ALT QUAL FILTER  \\\n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "..    ...       ...        ...  ..  ...  ...    ...   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "\n",
       "                                                 INFO     Gene_Name Gene Name  \\\n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "..                                                ...           ...       ...   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "\n",
       "                                                  CSQ  \\\n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "..                                                ...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "\n",
       "                                                  csq  \n",
       "0   C|missense_variant&splice_region_variant|MODER...  \n",
       "0   G|missense_variant&splice_region_variant|MODER...  \n",
       "0   C|missense_variant&splice_region_variant|MODER...  \n",
       "0   G|missense_variant&splice_region_variant|MODER...  \n",
       "0   C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...  \n",
       "..                                                ...  \n",
       "13  C|intron_variant&non_coding_transcript_variant...  \n",
       "13  A|intron_variant&non_coding_transcript_variant...  \n",
       "13  C|intron_variant&non_coding_transcript_variant...  \n",
       "13  A|regulatory_region_variant|MODIFIER|||Regulat...  \n",
       "13  C|regulatory_region_variant|MODIFIER|||Regulat...  \n",
       "\n",
       "[110 rows x 12 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f2ab4d36",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7:c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2:p.Leu446Pro</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>446</td>\n",
       "      <td>L/P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7:c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2:p.Leu446Arg</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>446</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1:c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1:p.Leu147Pro</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>147</td>\n",
       "      <td>L/P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1:c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1:p.Leu147Arg</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>147</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1:n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1:n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1:n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS       rsID REF  ALT QUAL FILTER  \\\n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "..    ...       ...        ...  ..  ...  ...    ...   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "\n",
       "                                                 INFO     Gene_Name Gene Name  \\\n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "..                                                ...           ...       ...   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "\n",
       "                                                  CSQ  \\\n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "..                                                ...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "\n",
       "                                                  csq  \\\n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   G|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   G|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...   \n",
       "..                                                ...   \n",
       "13  C|intron_variant&non_coding_transcript_variant...   \n",
       "13  A|intron_variant&non_coding_transcript_variant...   \n",
       "13  C|intron_variant&non_coding_transcript_variant...   \n",
       "13  A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "13  C|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "\n",
       "                                        ClinVar_CLNDN Allele  \\\n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "0                                                          G   \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "0                                                          G   \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "..                                                ...    ...   \n",
       "13                                                         C   \n",
       "13                                                         A   \n",
       "13                                                         C   \n",
       "13                                                         A   \n",
       "13                                                         C   \n",
       "\n",
       "              CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "0                                                                           \n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "0                                                                           \n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "..                 ...                                                ...   \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "\n",
       "   ClinVar                            HGVSC                          HGVSP  \\\n",
       "0     8751      ENST00000264717.7:c.1337T>C  ENSP00000264717.2:p.Leu446Pro   \n",
       "0               ENST00000264717.7:c.1337T>G  ENSP00000264717.2:p.Leu446Arg   \n",
       "0     8751       ENST00000411584.1:c.440T>C  ENSP00000416917.1:p.Leu147Pro   \n",
       "0                ENST00000411584.1:c.440T>G  ENSP00000416917.1:p.Leu147Arg   \n",
       "0     8751                                                                   \n",
       "..     ...                              ...                            ...   \n",
       "13           ENST00000493621.1:n.100+316G>C                                  \n",
       "13          ENST00000494795.1:n.3057-532G>A                                  \n",
       "13          ENST00000494795.1:n.3057-532G>C                                  \n",
       "13                                                                           \n",
       "13                                                                           \n",
       "\n",
       "         PolyPhen                         BIOTYPE   EXON INTRON  \\\n",
       "0       benign(0)                  protein_coding  15/19          \n",
       "0       benign(0)                  protein_coding  15/19          \n",
       "0   benign(0.001)                  protein_coding    5/7          \n",
       "0   benign(0.286)                  protein_coding    5/7          \n",
       "0                                 retained_intron                 \n",
       "..            ...                             ...    ...    ...   \n",
       "13                 protein_coding_CDS_not_defined           1/1   \n",
       "13                                retained_intron           2/2   \n",
       "13                                retained_intron           2/2   \n",
       "13                                       promoter                 \n",
       "13                                       promoter                 \n",
       "\n",
       "   Protein_position Amino_acids   Codons STRAND  \\\n",
       "0               446         L/P  cTg/cCg      1   \n",
       "0               446         L/R  cTg/cGg      1   \n",
       "0               147         L/P  cTg/cCg      1   \n",
       "0               147         L/R  cTg/cGg      1   \n",
       "0                                             1   \n",
       "..              ...         ...      ...    ...   \n",
       "13                                            1   \n",
       "13                                            1   \n",
       "13                                            1   \n",
       "13                                                \n",
       "13                                                \n",
       "\n",
       "                                               PUBMED  \\\n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "..                                                ...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "\n",
       "                                     Consequence    IMPACT             SIFT  \\\n",
       "0         missense_variant&splice_region_variant  MODERATE     tolerated(1)   \n",
       "0         missense_variant&splice_region_variant  MODERATE  tolerated(0.06)   \n",
       "0         missense_variant&splice_region_variant  MODERATE     tolerated(1)   \n",
       "0         missense_variant&splice_region_variant  MODERATE  tolerated(0.06)   \n",
       "0                        downstream_gene_variant  MODIFIER                    \n",
       "..                                           ...       ...              ...   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13                     regulatory_region_variant  MODIFIER                    \n",
       "13                     regulatory_region_variant  MODIFIER                    \n",
       "\n",
       "        AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "0                                                   3.982e-06              0   \n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "0                                                   3.982e-06              0   \n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "..     ...     ...     ...     ...     ...     ...        ...            ...   \n",
       "13                                                                             \n",
       "13  0.3075  0.3797  0.3415  0.3661  0.2087   0.227                             \n",
       "13                                                                             \n",
       "13  0.3075  0.3797  0.3415  0.3661  0.2087   0.227                             \n",
       "13                                                                             \n",
       "\n",
       "   gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "0               0              0      5.444e-05              0              0   \n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "0               0              0      5.444e-05              0              0   \n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "\n",
       "   gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "0               0              0                                            \n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "0               0              0                                            \n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "..            ...            ...        ...            ...            ...   \n",
       "13                                                                          \n",
       "13                                   0.2519         0.3536         0.2325   \n",
       "13                                                                          \n",
       "13                                   0.2519         0.3536         0.2325   \n",
       "13                                                                          \n",
       "\n",
       "   gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "0                                                                               \n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "0                                                                               \n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13         0.3173         0.1317         0.3875         0.2268         0.2184   \n",
       "13                                                                              \n",
       "13         0.3173         0.1317         0.3875         0.2268         0.2184   \n",
       "13                                                                              \n",
       "\n",
       "   gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  MAX_AF_POPS  \n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR  \n",
       "0                                                5.444e-05  gnomADe_EAS  \n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR  \n",
       "0                                                5.444e-05  gnomADe_EAS  \n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR  \n",
       "..            ...            ...            ...        ...          ...  \n",
       "13                                                                       \n",
       "13         0.1789         0.2555         0.2035     0.3875  gnomADg_EAS  \n",
       "13                                                                       \n",
       "13         0.1789         0.2555         0.2035     0.3875  gnomADg_EAS  \n",
       "13                                                                       \n",
       "\n",
       "[110 rows x 59 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "########################################################### Required columns extraction from the CSQ column ####################\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['Allele'] = vcf['csq'].str.split('|').str[0]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "################################################## Frequency columns extraction ################################################\n",
    "vcf['AF'] = vcf['csq'].str.split('|').str[42]\n",
    "vcf['AFR_AF'] = vcf['csq'].str.split('|').str[43]\n",
    "vcf['AMR_AF'] = vcf['csq'].str.split('|').str[44]\n",
    "vcf['EAS_AF'] = vcf['csq'].str.split('|').str[45]\n",
    "vcf['EUR_AF'] = vcf['csq'].str.split('|').str[46]\n",
    "vcf['SAS_AF'] = vcf['csq'].str.split('|').str[47]\n",
    "vcf['gnomADe_AF'] = vcf['csq'].str.split('|').str[48]\n",
    "vcf['gnomADe_AFR_AF'] = vcf['csq'].str.split('|').str[49]\n",
    "vcf['gnomADe_AMR_AF'] = vcf['csq'].str.split('|').str[50]\n",
    "vcf['gnomADe_ASJ_AF'] = vcf['csq'].str.split('|').str[51]\n",
    "vcf['gnomADe_EAS_AF'] = vcf['csq'].str.split('|').str[52]\n",
    "vcf['gnomADe_FIN_AF'] = vcf['csq'].str.split('|').str[53]\n",
    "vcf['gnomADe_NFE_AF'] = vcf['csq'].str.split('|').str[54]\n",
    "vcf['gnomADe_OTH_AF'] = vcf['csq'].str.split('|').str[55]\n",
    "vcf['gnomADe_SAS_AF'] = vcf['csq'].str.split('|').str[56]\n",
    "vcf['gnomADg_AF'] = vcf['csq'].str.split('|').str[57]\n",
    "vcf['gnomADg_AFR_AF'] = vcf['csq'].str.split('|').str[58]\n",
    "vcf['gnomADg_AMI_AF'] = vcf['csq'].str.split('|').str[59]\n",
    "vcf['gnomADg_AMR_AF'] = vcf['csq'].str.split('|').str[60]\n",
    "vcf['gnomADg_ASJ_AF'] = vcf['csq'].str.split('|').str[61]\n",
    "vcf['gnomADg_EAS_AF'] = vcf['csq'].str.split('|').str[62]\n",
    "vcf['gnomADg_FIN_AF'] = vcf['csq'].str.split('|').str[63]\n",
    "vcf['gnomADg_MID_AF'] = vcf['csq'].str.split('|').str[64]\n",
    "vcf['gnomADg_NFE_AF'] = vcf['csq'].str.split('|').str[65]\n",
    "vcf['gnomADg_OTH_AF'] = vcf['csq'].str.split('|').str[66]\n",
    "vcf['gnomADg_SAS_AF'] = vcf['csq'].str.split('|').str[67]\n",
    "vcf['MAX_AF'] = vcf['csq'].str.split('|').str[68]\n",
    "vcf['MAX_AF_POPS'] = vcf['csq'].str.split('|').str[69]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "789e468c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7:c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2:p.Leu446Pro</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>446</td>\n",
       "      <td>L/P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>L446P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7:c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2:p.Leu446Arg</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>446</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>L446R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1:c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1:p.Leu147Pro</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>147</td>\n",
       "      <td>L/P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>L147P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1:c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1:p.Leu147Arg</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>147</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>L147R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1:n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1:n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1:n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS       rsID REF  ALT QUAL FILTER  \\\n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "..    ...       ...        ...  ..  ...  ...    ...   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "\n",
       "                                                 INFO     Gene_Name Gene Name  \\\n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "..                                                ...           ...       ...   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "\n",
       "                                                  CSQ  \\\n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "..                                                ...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "\n",
       "                                                  csq  \\\n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   G|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   G|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...   \n",
       "..                                                ...   \n",
       "13  C|intron_variant&non_coding_transcript_variant...   \n",
       "13  A|intron_variant&non_coding_transcript_variant...   \n",
       "13  C|intron_variant&non_coding_transcript_variant...   \n",
       "13  A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "13  C|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "\n",
       "                                        ClinVar_CLNDN Allele  \\\n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "0                                                          G   \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "0                                                          G   \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "..                                                ...    ...   \n",
       "13                                                         C   \n",
       "13                                                         A   \n",
       "13                                                         C   \n",
       "13                                                         A   \n",
       "13                                                         C   \n",
       "\n",
       "              CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "0                                                                           \n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "0                                                                           \n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "..                 ...                                                ...   \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "\n",
       "   ClinVar                            HGVSC                          HGVSP  \\\n",
       "0     8751      ENST00000264717.7:c.1337T>C  ENSP00000264717.2:p.Leu446Pro   \n",
       "0               ENST00000264717.7:c.1337T>G  ENSP00000264717.2:p.Leu446Arg   \n",
       "0     8751       ENST00000411584.1:c.440T>C  ENSP00000416917.1:p.Leu147Pro   \n",
       "0                ENST00000411584.1:c.440T>G  ENSP00000416917.1:p.Leu147Arg   \n",
       "0     8751                                                                   \n",
       "..     ...                              ...                            ...   \n",
       "13           ENST00000493621.1:n.100+316G>C                                  \n",
       "13          ENST00000494795.1:n.3057-532G>A                                  \n",
       "13          ENST00000494795.1:n.3057-532G>C                                  \n",
       "13                                                                           \n",
       "13                                                                           \n",
       "\n",
       "         PolyPhen                         BIOTYPE   EXON INTRON  \\\n",
       "0       benign(0)                  protein_coding  15/19          \n",
       "0       benign(0)                  protein_coding  15/19          \n",
       "0   benign(0.001)                  protein_coding    5/7          \n",
       "0   benign(0.286)                  protein_coding    5/7          \n",
       "0                                 retained_intron                 \n",
       "..            ...                             ...    ...    ...   \n",
       "13                 protein_coding_CDS_not_defined           1/1   \n",
       "13                                retained_intron           2/2   \n",
       "13                                retained_intron           2/2   \n",
       "13                                       promoter                 \n",
       "13                                       promoter                 \n",
       "\n",
       "   Protein_position Amino_acids   Codons STRAND  \\\n",
       "0               446         L/P  cTg/cCg      1   \n",
       "0               446         L/R  cTg/cGg      1   \n",
       "0               147         L/P  cTg/cCg      1   \n",
       "0               147         L/R  cTg/cGg      1   \n",
       "0                                             1   \n",
       "..              ...         ...      ...    ...   \n",
       "13                                            1   \n",
       "13                                            1   \n",
       "13                                            1   \n",
       "13                                                \n",
       "13                                                \n",
       "\n",
       "                                               PUBMED  \\\n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "..                                                ...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "\n",
       "                                     Consequence    IMPACT             SIFT  \\\n",
       "0         missense_variant&splice_region_variant  MODERATE     tolerated(1)   \n",
       "0         missense_variant&splice_region_variant  MODERATE  tolerated(0.06)   \n",
       "0         missense_variant&splice_region_variant  MODERATE     tolerated(1)   \n",
       "0         missense_variant&splice_region_variant  MODERATE  tolerated(0.06)   \n",
       "0                        downstream_gene_variant  MODIFIER                    \n",
       "..                                           ...       ...              ...   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13                     regulatory_region_variant  MODIFIER                    \n",
       "13                     regulatory_region_variant  MODIFIER                    \n",
       "\n",
       "        AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "0                                                   3.982e-06              0   \n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "0                                                   3.982e-06              0   \n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "..     ...     ...     ...     ...     ...     ...        ...            ...   \n",
       "13                                                                             \n",
       "13  0.3075  0.3797  0.3415  0.3661  0.2087   0.227                             \n",
       "13                                                                             \n",
       "13  0.3075  0.3797  0.3415  0.3661  0.2087   0.227                             \n",
       "13                                                                             \n",
       "\n",
       "   gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "0               0              0      5.444e-05              0              0   \n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "0               0              0      5.444e-05              0              0   \n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "\n",
       "   gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "0               0              0                                            \n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "0               0              0                                            \n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "..            ...            ...        ...            ...            ...   \n",
       "13                                                                          \n",
       "13                                   0.2519         0.3536         0.2325   \n",
       "13                                                                          \n",
       "13                                   0.2519         0.3536         0.2325   \n",
       "13                                                                          \n",
       "\n",
       "   gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "0                                                                               \n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "0                                                                               \n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13         0.3173         0.1317         0.3875         0.2268         0.2184   \n",
       "13                                                                              \n",
       "13         0.3173         0.1317         0.3875         0.2268         0.2184   \n",
       "13                                                                              \n",
       "\n",
       "   gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  MAX_AF_POPS  \\\n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR   \n",
       "0                                                5.444e-05  gnomADe_EAS   \n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR   \n",
       "0                                                5.444e-05  gnomADe_EAS   \n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR   \n",
       "..            ...            ...            ...        ...          ...   \n",
       "13                                                                        \n",
       "13         0.1789         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                                        \n",
       "13         0.1789         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                                        \n",
       "\n",
       "   Protein Position and Amino Acid  \n",
       "0                            L446P  \n",
       "0                            L446R  \n",
       "0                            L147P  \n",
       "0                            L147R  \n",
       "0                              NaN  \n",
       "..                             ...  \n",
       "13                             NaN  \n",
       "13                             NaN  \n",
       "13                             NaN  \n",
       "13                             NaN  \n",
       "13                             NaN  \n",
       "\n",
       "[110 rows x 60 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "155ca646",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7:c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2:p.Leu446Pro</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>446</td>\n",
       "      <td>L/P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>L446P</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7:c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2:p.Leu446Arg</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>446</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>L446R</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1:c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1:p.Leu147Pro</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>147</td>\n",
       "      <td>L/P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>L147P</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>G|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td></td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1:c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1:p.Leu147Arg</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>147</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>L147R</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...</td>\n",
       "      <td>GCKR:2646</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>C|missense_variant&amp;splice_region_variant|MODER...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>C</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1:n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1:n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1:n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...</td>\n",
       "      <td>SAMM50:25813</td>\n",
       "      <td>SAMM50</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMM50|ENSG000001003...</td>\n",
       "      <td>C|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS       rsID REF  ALT QUAL FILTER  \\\n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "0    chr2  27508073  rs1260326   T  C,G    .      .   \n",
       "..    ...       ...        ...  ..  ...  ...    ...   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "13  chr22  43995806  rs2143571   G  A,C    .      .   \n",
       "\n",
       "                                                 INFO     Gene_Name Gene Name  \\\n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "0   RS=1260326;dbSNPBuildID=87;SSR=0;GENEINFO=GCKR...     GCKR:2646      GCKR   \n",
       "..                                                ...           ...       ...   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "13  RS=2143571;dbSNPBuildID=96;SSR=0;GENEINFO=SAMM...  SAMM50:25813    SAMM50   \n",
       "\n",
       "                                                  CSQ  \\\n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "..                                                ...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "13  A|intron_variant|MODIFIER|SAMM50|ENSG000001003...   \n",
       "\n",
       "                                                  csq  \\\n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   G|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|missense_variant&splice_region_variant|MODER...   \n",
       "0   G|missense_variant&splice_region_variant|MODER...   \n",
       "0   C|downstream_gene_variant|MODIFIER|GCKR|ENSG00...   \n",
       "..                                                ...   \n",
       "13  C|intron_variant&non_coding_transcript_variant...   \n",
       "13  A|intron_variant&non_coding_transcript_variant...   \n",
       "13  C|intron_variant&non_coding_transcript_variant...   \n",
       "13  A|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "13  C|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "\n",
       "                                        ClinVar_CLNDN Allele  \\\n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "0                                                          G   \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "0                                                          G   \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...      C   \n",
       "..                                                ...    ...   \n",
       "13                                                         C   \n",
       "13                                                         A   \n",
       "13                                                         C   \n",
       "13                                                         A   \n",
       "13                                                         C   \n",
       "\n",
       "              CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "0                                                                           \n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "0                                                                           \n",
       "0   benign&association  criteria_provided&_multiple_submitters&_no_con...   \n",
       "..                 ...                                                ...   \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "\n",
       "   ClinVar                            HGVSC                          HGVSP  \\\n",
       "0     8751      ENST00000264717.7:c.1337T>C  ENSP00000264717.2:p.Leu446Pro   \n",
       "0               ENST00000264717.7:c.1337T>G  ENSP00000264717.2:p.Leu446Arg   \n",
       "0     8751       ENST00000411584.1:c.440T>C  ENSP00000416917.1:p.Leu147Pro   \n",
       "0                ENST00000411584.1:c.440T>G  ENSP00000416917.1:p.Leu147Arg   \n",
       "0     8751                                                                   \n",
       "..     ...                              ...                            ...   \n",
       "13           ENST00000493621.1:n.100+316G>C                                  \n",
       "13          ENST00000494795.1:n.3057-532G>A                                  \n",
       "13          ENST00000494795.1:n.3057-532G>C                                  \n",
       "13                                                                           \n",
       "13                                                                           \n",
       "\n",
       "         PolyPhen                         BIOTYPE   EXON INTRON  \\\n",
       "0       benign(0)                  protein_coding  15/19          \n",
       "0       benign(0)                  protein_coding  15/19          \n",
       "0   benign(0.001)                  protein_coding    5/7          \n",
       "0   benign(0.286)                  protein_coding    5/7          \n",
       "0                                 retained_intron                 \n",
       "..            ...                             ...    ...    ...   \n",
       "13                 protein_coding_CDS_not_defined           1/1   \n",
       "13                                retained_intron           2/2   \n",
       "13                                retained_intron           2/2   \n",
       "13                                       promoter                 \n",
       "13                                       promoter                 \n",
       "\n",
       "   Protein_position Amino_acids   Codons STRAND  \\\n",
       "0               446         L/P  cTg/cCg      1   \n",
       "0               446         L/R  cTg/cGg      1   \n",
       "0               147         L/P  cTg/cCg      1   \n",
       "0               147         L/R  cTg/cGg      1   \n",
       "0                                             1   \n",
       "..              ...         ...      ...    ...   \n",
       "13                                            1   \n",
       "13                                            1   \n",
       "13                                            1   \n",
       "13                                                \n",
       "13                                                \n",
       "\n",
       "                                               PUBMED  \\\n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...   \n",
       "..                                                ...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...   \n",
       "\n",
       "                                     Consequence    IMPACT             SIFT  \\\n",
       "0         missense_variant&splice_region_variant  MODERATE     tolerated(1)   \n",
       "0         missense_variant&splice_region_variant  MODERATE  tolerated(0.06)   \n",
       "0         missense_variant&splice_region_variant  MODERATE     tolerated(1)   \n",
       "0         missense_variant&splice_region_variant  MODERATE  tolerated(0.06)   \n",
       "0                        downstream_gene_variant  MODIFIER                    \n",
       "..                                           ...       ...              ...   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER                    \n",
       "13                     regulatory_region_variant  MODIFIER                    \n",
       "13                     regulatory_region_variant  MODIFIER                    \n",
       "\n",
       "        AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "0                                                   3.982e-06              0   \n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "0                                                   3.982e-06              0   \n",
       "0   0.7067  0.9062  0.6383  0.5188  0.5895  0.7996     0.6331         0.8679   \n",
       "..     ...     ...     ...     ...     ...     ...        ...            ...   \n",
       "13                                                                             \n",
       "13  0.3075  0.3797  0.3415  0.3661  0.2087   0.227                             \n",
       "13                                                                             \n",
       "13  0.3075  0.3797  0.3415  0.3661  0.2087   0.227                             \n",
       "13                                                                             \n",
       "\n",
       "   gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "0               0              0      5.444e-05              0              0   \n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "0               0              0      5.444e-05              0              0   \n",
       "0          0.6662         0.4668         0.5108         0.6426         0.5902   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "\n",
       "   gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "0               0              0                                            \n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "0               0              0                                            \n",
       "0          0.6005          0.758       0.67         0.8582         0.7083   \n",
       "..            ...            ...        ...            ...            ...   \n",
       "13                                                                          \n",
       "13                                   0.2519         0.3536         0.2325   \n",
       "13                                                                          \n",
       "13                                   0.2519         0.3536         0.2325   \n",
       "13                                                                          \n",
       "\n",
       "   gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF  \\\n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "0                                                                               \n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "0                                                                               \n",
       "0           0.626         0.4585         0.5014         0.6378         0.4904   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13         0.3173         0.1317         0.3875         0.2268         0.2184   \n",
       "13                                                                              \n",
       "13         0.3173         0.1317         0.3875         0.2268         0.2184   \n",
       "13                                                                              \n",
       "\n",
       "   gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  MAX_AF_POPS  \\\n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR   \n",
       "0                                                5.444e-05  gnomADe_EAS   \n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR   \n",
       "0                                                5.444e-05  gnomADe_EAS   \n",
       "0          0.5898         0.6106         0.7566     0.9062          AFR   \n",
       "..            ...            ...            ...        ...          ...   \n",
       "13                                                                        \n",
       "13         0.1789         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                                        \n",
       "13         0.1789         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                                        \n",
       "\n",
       "   Protein Position and Amino Acid              HGVSc HGVSc (Transcript)  \\\n",
       "0                            L446P  ENST00000264717.7          c.1337T>C   \n",
       "0                            L446R  ENST00000264717.7          c.1337T>G   \n",
       "0                            L147P  ENST00000411584.1           c.440T>C   \n",
       "0                            L147R  ENST00000411584.1           c.440T>G   \n",
       "0                              NaN                                  None   \n",
       "..                             ...                ...                ...   \n",
       "13                             NaN  ENST00000493621.1       n.100+316G>C   \n",
       "13                             NaN  ENST00000494795.1      n.3057-532G>A   \n",
       "13                             NaN  ENST00000494795.1      n.3057-532G>C   \n",
       "13                             NaN                                  None   \n",
       "13                             NaN                                  None   \n",
       "\n",
       "                HGVSp HGVSp (Transcript)  \n",
       "0   ENSP00000264717.2        p.Leu446Pro  \n",
       "0   ENSP00000264717.2        p.Leu446Arg  \n",
       "0   ENSP00000416917.1        p.Leu147Pro  \n",
       "0   ENSP00000416917.1        p.Leu147Arg  \n",
       "0                                   None  \n",
       "..                ...                ...  \n",
       "13                                  None  \n",
       "13                                  None  \n",
       "13                                  None  \n",
       "13                                  None  \n",
       "13                                  None  \n",
       "\n",
       "[110 rows x 64 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f3047b52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not_provided&amp;Fasting_plasma_glucose_level_quan...</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name       rsID  CHROM       POS REF  ALT Allele  \\\n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "..       ...        ...    ...       ...  ..  ...    ...   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "\n",
       "                                     Consequence    IMPACT  \\\n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0                        downstream_gene_variant  MODIFIER   \n",
       "..                                           ...       ...   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "13                     regulatory_region_variant  MODIFIER   \n",
       "13                     regulatory_region_variant  MODIFIER   \n",
       "\n",
       "                                        ClinVar_CLNDN            CLIN_SIG  \\\n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...  benign&association   \n",
       "0                                                                           \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...  benign&association   \n",
       "0                                                                           \n",
       "0   not_provided&Fasting_plasma_glucose_level_quan...  benign&association   \n",
       "..                                                ...                 ...   \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "\n",
       "                                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0   criteria_provided&_multiple_submitters&_no_con...    8751   \n",
       "0                                                               \n",
       "0   criteria_provided&_multiple_submitters&_no_con...    8751   \n",
       "0                                                               \n",
       "0   criteria_provided&_multiple_submitters&_no_con...    8751   \n",
       "..                                                ...     ...   \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "\n",
       "                HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0   ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "0   ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "0   ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "0   ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "0                                   None                      \n",
       "..                ...                ...                ...   \n",
       "13  ENST00000493621.1       n.100+316G>C                      \n",
       "13  ENST00000494795.1      n.3057-532G>A                      \n",
       "13  ENST00000494795.1      n.3057-532G>C                      \n",
       "13                                  None                      \n",
       "13                                  None                      \n",
       "\n",
       "   HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0         p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu446Arg  tolerated(0.06)      benign(0)                           \n",
       "0         p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu147Arg  tolerated(0.06)  benign(0.286)                           \n",
       "0                None                                  0.7067  0.9062  0.6383   \n",
       "..                ...              ...            ...     ...     ...     ...   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "\n",
       "    EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "..     ...     ...     ...        ...            ...            ...   \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "\n",
       "   gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "\n",
       "   gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "..            ...        ...            ...            ...            ...   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "\n",
       "   gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "\n",
       "   gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  MAX_AF_POPS  \\\n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe_EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe_EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "..            ...            ...        ...          ...   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                         \n",
       "\n",
       "                           BIOTYPE   EXON INTRON  \\\n",
       "0                   protein_coding  15/19          \n",
       "0                   protein_coding  15/19          \n",
       "0                   protein_coding    5/7          \n",
       "0                   protein_coding    5/7          \n",
       "0                  retained_intron                 \n",
       "..                             ...    ...    ...   \n",
       "13  protein_coding_CDS_not_defined           1/1   \n",
       "13                 retained_intron           2/2   \n",
       "13                 retained_intron           2/2   \n",
       "13                        promoter                 \n",
       "13                        promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                            L446P  cTg/cCg      1   \n",
       "0                            L446R  cTg/cGg      1   \n",
       "0                            L147P  cTg/cCg      1   \n",
       "0                            L147R  cTg/cGg      1   \n",
       "0                              NaN               1   \n",
       "..                             ...      ...    ...   \n",
       "13                             NaN               1   \n",
       "13                             NaN               1   \n",
       "13                             NaN               1   \n",
       "13                             NaN                   \n",
       "13                             NaN                   \n",
       "\n",
       "                                               PUBMED  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "..                                                ...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "\n",
       "[110 rows x 54 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf[['Gene Name', 'rsID','CHROM', 'POS', 'REF', 'ALT', 'Allele', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
    "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
    "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
    "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
    "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
    "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
    "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f4d5a049",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting_plasma_glucose_level_quantitative_trai...</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting_plasma_glucose_level_quantitative_trai...</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant&amp;splice_region_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Fasting_plasma_glucose_level_quantitative_trai...</td>\n",
       "      <td>benign&amp;association</td>\n",
       "      <td>criteria_provided&amp;_multiple_submitters&amp;_no_con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728&amp;30409984&amp;18439548&amp;21054877&amp;21647738&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding_CDS_not_defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889&amp;18940312&amp;25018854&amp;27458502&amp;22719876&amp;2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name       rsID  CHROM       POS REF  ALT Allele  \\\n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "..       ...        ...    ...       ...  ..  ...    ...   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "\n",
       "                                     Consequence    IMPACT  \\\n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0         missense_variant&splice_region_variant  MODERATE   \n",
       "0                        downstream_gene_variant  MODIFIER   \n",
       "..                                           ...       ...   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "13  intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "13                     regulatory_region_variant  MODIFIER   \n",
       "13                     regulatory_region_variant  MODIFIER   \n",
       "\n",
       "                                        ClinVar_CLNDN            CLIN_SIG  \\\n",
       "0   Fasting_plasma_glucose_level_quantitative_trai...  benign&association   \n",
       "0                                                                           \n",
       "0   Fasting_plasma_glucose_level_quantitative_trai...  benign&association   \n",
       "0                                                                           \n",
       "0   Fasting_plasma_glucose_level_quantitative_trai...  benign&association   \n",
       "..                                                ...                 ...   \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "\n",
       "                                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0   criteria_provided&_multiple_submitters&_no_con...    8751   \n",
       "0                                                               \n",
       "0   criteria_provided&_multiple_submitters&_no_con...    8751   \n",
       "0                                                               \n",
       "0   criteria_provided&_multiple_submitters&_no_con...    8751   \n",
       "..                                                ...     ...   \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "\n",
       "                HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0   ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "0   ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "0   ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "0   ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "0                                   None                      \n",
       "..                ...                ...                ...   \n",
       "13  ENST00000493621.1       n.100+316G>C                      \n",
       "13  ENST00000494795.1      n.3057-532G>A                      \n",
       "13  ENST00000494795.1      n.3057-532G>C                      \n",
       "13                                  None                      \n",
       "13                                  None                      \n",
       "\n",
       "   HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0         p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu446Arg  tolerated(0.06)      benign(0)                           \n",
       "0         p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu147Arg  tolerated(0.06)  benign(0.286)                           \n",
       "0                None                                  0.7067  0.9062  0.6383   \n",
       "..                ...              ...            ...     ...     ...     ...   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "\n",
       "    EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "..     ...     ...     ...        ...            ...            ...   \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "\n",
       "   gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "\n",
       "   gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "..            ...        ...            ...            ...            ...   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "\n",
       "   gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "\n",
       "   gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  MAX_AF_POPS  \\\n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe_EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe_EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "..            ...            ...        ...          ...   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg_EAS   \n",
       "13                                                         \n",
       "\n",
       "                           BIOTYPE   EXON INTRON  \\\n",
       "0                   protein_coding  15/19          \n",
       "0                   protein_coding  15/19          \n",
       "0                   protein_coding    5/7          \n",
       "0                   protein_coding    5/7          \n",
       "0                  retained_intron                 \n",
       "..                             ...    ...    ...   \n",
       "13  protein_coding_CDS_not_defined           1/1   \n",
       "13                 retained_intron           2/2   \n",
       "13                 retained_intron           2/2   \n",
       "13                        promoter                 \n",
       "13                        promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                            L446P  cTg/cCg      1   \n",
       "0                            L446R  cTg/cGg      1   \n",
       "0                            L147P  cTg/cCg      1   \n",
       "0                            L147R  cTg/cGg      1   \n",
       "0                              NaN               1   \n",
       "..                             ...      ...    ...   \n",
       "13                             NaN               1   \n",
       "13                             NaN               1   \n",
       "13                             NaN               1   \n",
       "13                             NaN                   \n",
       "13                             NaN                   \n",
       "\n",
       "                                               PUBMED  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "0   21467728&30409984&18439548&21054877&21647738&2...  \n",
       "..                                                ...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "13  26780889&18940312&25018854&27458502&22719876&2...  \n",
       "\n",
       "[110 rows x 54 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "# Print the modified DataFrame\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0e361f9b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name       rsID  CHROM       POS REF  ALT Allele  \\\n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "..       ...        ...    ...       ...  ..  ...    ...   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "\n",
       "                                     Consequence    IMPACT  \\\n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0                        downstream gene variant  MODIFIER   \n",
       "..                                           ...       ...   \n",
       "13  intron variant,non coding transcript variant  MODIFIER   \n",
       "13  intron variant,non coding transcript variant  MODIFIER   \n",
       "13  intron variant,non coding transcript variant  MODIFIER   \n",
       "13                     regulatory region variant  MODIFIER   \n",
       "13                     regulatory region variant  MODIFIER   \n",
       "\n",
       "                                        ClinVar_CLNDN            CLIN_SIG  \\\n",
       "0   Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "0                                                                           \n",
       "0   Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "0                                                                           \n",
       "0   Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "..                                                ...                 ...   \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "\n",
       "                                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0   criteria provided, multiple submitters, no con...    8751   \n",
       "0                                                               \n",
       "0   criteria provided, multiple submitters, no con...    8751   \n",
       "0                                                               \n",
       "0   criteria provided, multiple submitters, no con...    8751   \n",
       "..                                                ...     ...   \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "\n",
       "                HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0   ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "0   ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "0   ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "0   ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "0                                   None                      \n",
       "..                ...                ...                ...   \n",
       "13  ENST00000493621.1       n.100+316G>C                      \n",
       "13  ENST00000494795.1      n.3057-532G>A                      \n",
       "13  ENST00000494795.1      n.3057-532G>C                      \n",
       "13                                  None                      \n",
       "13                                  None                      \n",
       "\n",
       "   HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0         p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu446Arg  tolerated(0.06)      benign(0)                           \n",
       "0         p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu147Arg  tolerated(0.06)  benign(0.286)                           \n",
       "0                None                                  0.7067  0.9062  0.6383   \n",
       "..                ...              ...            ...     ...     ...     ...   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "\n",
       "    EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "..     ...     ...     ...        ...            ...            ...   \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "\n",
       "   gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "\n",
       "   gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "..            ...        ...            ...            ...            ...   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "\n",
       "   gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "\n",
       "   gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  MAX_AF_POPS  \\\n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "..            ...            ...        ...          ...   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg EAS   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg EAS   \n",
       "13                                                         \n",
       "\n",
       "                           BIOTYPE   EXON INTRON  \\\n",
       "0                   protein coding  15/19          \n",
       "0                   protein coding  15/19          \n",
       "0                   protein coding    5/7          \n",
       "0                   protein coding    5/7          \n",
       "0                  retained intron                 \n",
       "..                             ...    ...    ...   \n",
       "13  protein coding CDS not defined           1/1   \n",
       "13                 retained intron           2/2   \n",
       "13                 retained intron           2/2   \n",
       "13                        promoter                 \n",
       "13                        promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                            L446P  cTg/cCg      1   \n",
       "0                            L446R  cTg/cGg      1   \n",
       "0                            L147P  cTg/cCg      1   \n",
       "0                            L147R  cTg/cGg      1   \n",
       "0                              nan               1   \n",
       "..                             ...      ...    ...   \n",
       "13                             nan               1   \n",
       "13                             nan               1   \n",
       "13                             nan               1   \n",
       "13                             nan                   \n",
       "13                             nan                   \n",
       "\n",
       "                                               PUBMED  \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...  \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...  \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...  \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...  \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...  \n",
       "..                                                ...  \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "\n",
       "[110 rows x 54 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e41eeca4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>regulatory region variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>regulatory region variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name       rsID  CHROM       POS REF  ALT Allele  \\\n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "0       GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "..       ...        ...    ...       ...  ..  ...    ...   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "13    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "\n",
       "                                     Consequence    IMPACT  \\\n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0         missense variant,splice region variant  MODERATE   \n",
       "0                        downstream gene variant  MODIFIER   \n",
       "..                                           ...       ...   \n",
       "13  intron variant,non coding transcript variant  MODIFIER   \n",
       "13  intron variant,non coding transcript variant  MODIFIER   \n",
       "13  intron variant,non coding transcript variant  MODIFIER   \n",
       "13                     regulatory region variant  MODIFIER   \n",
       "13                     regulatory region variant  MODIFIER   \n",
       "\n",
       "                                        ClinVar_CLNDN            CLIN_SIG  \\\n",
       "0   Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "0                                                                           \n",
       "0   Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "0                                                                           \n",
       "0   Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "..                                                ...                 ...   \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "13                                                                          \n",
       "\n",
       "                                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0   criteria provided, multiple submitters, no con...    8751   \n",
       "0                                                               \n",
       "0   criteria provided, multiple submitters, no con...    8751   \n",
       "0                                                               \n",
       "0   criteria provided, multiple submitters, no con...    8751   \n",
       "..                                                ...     ...   \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "13                                                              \n",
       "\n",
       "                HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0   ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "0   ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "0   ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "0   ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "0                                   None                      \n",
       "..                ...                ...                ...   \n",
       "13  ENST00000493621.1       n.100+316G>C                      \n",
       "13  ENST00000494795.1      n.3057-532G>A                      \n",
       "13  ENST00000494795.1      n.3057-532G>C                      \n",
       "13                                  None                      \n",
       "13                                  None                      \n",
       "\n",
       "   HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0         p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu446Arg  tolerated(0.06)      benign(0)                           \n",
       "0         p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062  0.6383   \n",
       "0         p.Leu147Arg  tolerated(0.06)  benign(0.286)                           \n",
       "0                None                                  0.7067  0.9062  0.6383   \n",
       "..                ...              ...            ...     ...     ...     ...   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "13               None                                  0.3075  0.3797  0.3415   \n",
       "13               None                                                           \n",
       "\n",
       "    EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "0                           3.982e-06              0              0   \n",
       "0   0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "..     ...     ...     ...        ...            ...            ...   \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "13  0.3661  0.2087   0.227                                            \n",
       "13                                                                    \n",
       "\n",
       "   gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF  \\\n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "0               0      5.444e-05              0              0              0   \n",
       "0          0.4668         0.5108         0.6426         0.5902         0.6005   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "13                                                                              \n",
       "\n",
       "   gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF  \\\n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "0               0                                                           \n",
       "0           0.758       0.67         0.8582         0.7083          0.626   \n",
       "..            ...        ...            ...            ...            ...   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "13                    0.2519         0.3536         0.2325         0.3173   \n",
       "13                                                                          \n",
       "\n",
       "   gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "0                                                                               \n",
       "0          0.4585         0.5014         0.6378         0.4904         0.5898   \n",
       "..            ...            ...            ...            ...            ...   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "13         0.1317         0.3875         0.2268         0.2184         0.1789   \n",
       "13                                                                              \n",
       "\n",
       "   gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  MAX_AF_POPS  \\\n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "0                                 5.444e-05  gnomADe EAS   \n",
       "0          0.6106         0.7566     0.9062          AFR   \n",
       "..            ...            ...        ...          ...   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg EAS   \n",
       "13                                                         \n",
       "13         0.2555         0.2035     0.3875  gnomADg EAS   \n",
       "13                                                         \n",
       "\n",
       "                           BIOTYPE   EXON INTRON  \\\n",
       "0                   protein coding  15/19          \n",
       "0                   protein coding  15/19          \n",
       "0                   protein coding    5/7          \n",
       "0                   protein coding    5/7          \n",
       "0                  retained intron                 \n",
       "..                             ...    ...    ...   \n",
       "13  protein coding CDS not defined           1/1   \n",
       "13                 retained intron           2/2   \n",
       "13                 retained intron           2/2   \n",
       "13                        promoter                 \n",
       "13                        promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                            L446P  cTg/cCg      1   \n",
       "0                            L446R  cTg/cGg      1   \n",
       "0                            L147P  cTg/cCg      1   \n",
       "0                            L147R  cTg/cGg      1   \n",
       "0                              nan               1   \n",
       "..                             ...      ...    ...   \n",
       "13                             nan               1   \n",
       "13                             nan               1   \n",
       "13                             nan               1   \n",
       "13                             nan                   \n",
       "13                             nan                   \n",
       "\n",
       "                                               PUBMED  \\\n",
       "0   21467728,30409984,18439548,21054877,21647738,2...   \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...   \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...   \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...   \n",
       "0   21467728,30409984,18439548,21054877,21647738,2...   \n",
       "..                                                ...   \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "13  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "\n",
       "                  consequence  \n",
       "0            missense variant  \n",
       "0            missense variant  \n",
       "0            missense variant  \n",
       "0            missense variant  \n",
       "0     downstream gene variant  \n",
       "..                        ...  \n",
       "13             intron variant  \n",
       "13             intron variant  \n",
       "13             intron variant  \n",
       "13  regulatory region variant  \n",
       "13  regulatory region variant  \n",
       "\n",
       "[110 rows x 55 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "cde87401",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4a0b6309",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name       rsID  CHROM       POS REF  ALT Allele  \\\n",
       "0        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "1        GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "2        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "3        GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "4        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "..        ...        ...    ...       ...  ..  ...    ...   \n",
       "105    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "106    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "107    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "108    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "109    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "\n",
       "                                      Consequence    IMPACT  \\\n",
       "0          missense variant,splice region variant  MODERATE   \n",
       "1          missense variant,splice region variant  MODERATE   \n",
       "2          missense variant,splice region variant  MODERATE   \n",
       "3          missense variant,splice region variant  MODERATE   \n",
       "4                         downstream gene variant  MODIFIER   \n",
       "..                                            ...       ...   \n",
       "105  intron variant,non coding transcript variant  MODIFIER   \n",
       "106  intron variant,non coding transcript variant  MODIFIER   \n",
       "107  intron variant,non coding transcript variant  MODIFIER   \n",
       "108                     regulatory region variant  MODIFIER   \n",
       "109                     regulatory region variant  MODIFIER   \n",
       "\n",
       "                                         ClinVar_CLNDN            CLIN_SIG  \\\n",
       "0    Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "1                                                                            \n",
       "2    Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "3                                                                            \n",
       "4    Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "..                                                 ...                 ...   \n",
       "105                                                                          \n",
       "106                                                                          \n",
       "107                                                                          \n",
       "108                                                                          \n",
       "109                                                                          \n",
       "\n",
       "                                    ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0    criteria provided, multiple submitters, no con...    8751   \n",
       "1                                                                \n",
       "2    criteria provided, multiple submitters, no con...    8751   \n",
       "3                                                                \n",
       "4    criteria provided, multiple submitters, no con...    8751   \n",
       "..                                                 ...     ...   \n",
       "105                                                              \n",
       "106                                                              \n",
       "107                                                              \n",
       "108                                                              \n",
       "109                                                              \n",
       "\n",
       "                 HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0    ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "1    ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "2    ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "3    ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "4                                    None                      \n",
       "..                 ...                ...                ...   \n",
       "105  ENST00000493621.1       n.100+316G>C                      \n",
       "106  ENST00000494795.1      n.3057-532G>A                      \n",
       "107  ENST00000494795.1      n.3057-532G>C                      \n",
       "108                                  None                      \n",
       "109                                  None                      \n",
       "\n",
       "    HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  \\\n",
       "0          p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062   \n",
       "1          p.Leu446Arg  tolerated(0.06)      benign(0)                   \n",
       "2          p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062   \n",
       "3          p.Leu147Arg  tolerated(0.06)  benign(0.286)                   \n",
       "4                 None                                  0.7067  0.9062   \n",
       "..                 ...              ...            ...     ...     ...   \n",
       "105               None                                                   \n",
       "106               None                                  0.3075  0.3797   \n",
       "107               None                                                   \n",
       "108               None                                  0.3075  0.3797   \n",
       "109               None                                                   \n",
       "\n",
       "     AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "1                                    3.982e-06              0              0   \n",
       "2    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "3                                    3.982e-06              0              0   \n",
       "4    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "..      ...     ...     ...     ...        ...            ...            ...   \n",
       "105                                                                            \n",
       "106  0.3415  0.3661  0.2087   0.227                                            \n",
       "107                                                                            \n",
       "108  0.3415  0.3661  0.2087   0.227                                            \n",
       "109                                                                            \n",
       "\n",
       "    gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0           0.4668         0.5108         0.6426         0.5902   \n",
       "1                0      5.444e-05              0              0   \n",
       "2           0.4668         0.5108         0.6426         0.5902   \n",
       "3                0      5.444e-05              0              0   \n",
       "4           0.4668         0.5108         0.6426         0.5902   \n",
       "..             ...            ...            ...            ...   \n",
       "105                                                               \n",
       "106                                                               \n",
       "107                                                               \n",
       "108                                                               \n",
       "109                                                               \n",
       "\n",
       "    gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "1                0              0                                            \n",
       "2           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "3                0              0                                            \n",
       "4           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "..             ...            ...        ...            ...            ...   \n",
       "105                                                                          \n",
       "106                                   0.2519         0.3536         0.2325   \n",
       "107                                                                          \n",
       "108                                   0.2519         0.3536         0.2325   \n",
       "109                                                                          \n",
       "\n",
       "    gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0            0.626         0.4585         0.5014         0.6378   \n",
       "1                                                                 \n",
       "2            0.626         0.4585         0.5014         0.6378   \n",
       "3                                                                 \n",
       "4            0.626         0.4585         0.5014         0.6378   \n",
       "..             ...            ...            ...            ...   \n",
       "105                                                               \n",
       "106         0.3173         0.1317         0.3875         0.2268   \n",
       "107                                                               \n",
       "108         0.3173         0.1317         0.3875         0.2268   \n",
       "109                                                               \n",
       "\n",
       "    gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  \\\n",
       "0           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "1                                                                5.444e-05   \n",
       "2           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "3                                                                5.444e-05   \n",
       "4           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "..             ...            ...            ...            ...        ...   \n",
       "105                                                                          \n",
       "106         0.2184         0.1789         0.2555         0.2035     0.3875   \n",
       "107                                                                          \n",
       "108         0.2184         0.1789         0.2555         0.2035     0.3875   \n",
       "109                                                                          \n",
       "\n",
       "     MAX_AF_POPS                         BIOTYPE   EXON INTRON  \\\n",
       "0            AFR                  protein coding  15/19          \n",
       "1    gnomADe EAS                  protein coding  15/19          \n",
       "2            AFR                  protein coding    5/7          \n",
       "3    gnomADe EAS                  protein coding    5/7          \n",
       "4            AFR                 retained intron                 \n",
       "..           ...                             ...    ...    ...   \n",
       "105               protein coding CDS not defined           1/1   \n",
       "106  gnomADg EAS                 retained intron           2/2   \n",
       "107                              retained intron           2/2   \n",
       "108  gnomADg EAS                        promoter                 \n",
       "109                                     promoter                 \n",
       "\n",
       "    Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                             L446P  cTg/cCg      1   \n",
       "1                             L446R  cTg/cGg      1   \n",
       "2                             L147P  cTg/cCg      1   \n",
       "3                             L147R  cTg/cGg      1   \n",
       "4                               nan               1   \n",
       "..                              ...      ...    ...   \n",
       "105                             nan               1   \n",
       "106                             nan               1   \n",
       "107                             nan               1   \n",
       "108                             nan                   \n",
       "109                             nan                   \n",
       "\n",
       "                                                PUBMED  \\\n",
       "0    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "1    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "2    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "3    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "4    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "..                                                 ...   \n",
       "105  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "106  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "107  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "108  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "109  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "\n",
       "                   consequence Consequence_score  \n",
       "0             missense variant              7/10  \n",
       "1             missense variant              7/10  \n",
       "2             missense variant              7/10  \n",
       "3             missense variant              7/10  \n",
       "4      downstream gene variant              2/10  \n",
       "..                         ...               ...  \n",
       "105             intron variant              2/10  \n",
       "106             intron variant              2/10  \n",
       "107             intron variant              2/10  \n",
       "108  regulatory region variant              2/10  \n",
       "109  regulatory region variant              2/10  \n",
       "\n",
       "[110 rows x 56 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "49fd9b0e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     IMPACT  IMPACT_score\n",
       "0      HIGH          10.0\n",
       "1  MODERATE           5.0\n",
       "2       LOW           2.5\n",
       "3  MODIFIER           1.5"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "dd5cc9ec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name       rsID  CHROM       POS REF  ALT Allele  \\\n",
       "0        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "1        GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "2        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "3        GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "4        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "..        ...        ...    ...       ...  ..  ...    ...   \n",
       "105    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "106    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "107    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "108    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "109    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "\n",
       "                                      Consequence    IMPACT  \\\n",
       "0          missense variant,splice region variant  MODERATE   \n",
       "1          missense variant,splice region variant  MODERATE   \n",
       "2          missense variant,splice region variant  MODERATE   \n",
       "3          missense variant,splice region variant  MODERATE   \n",
       "4                         downstream gene variant  MODIFIER   \n",
       "..                                            ...       ...   \n",
       "105  intron variant,non coding transcript variant  MODIFIER   \n",
       "106  intron variant,non coding transcript variant  MODIFIER   \n",
       "107  intron variant,non coding transcript variant  MODIFIER   \n",
       "108                     regulatory region variant  MODIFIER   \n",
       "109                     regulatory region variant  MODIFIER   \n",
       "\n",
       "                                         ClinVar_CLNDN            CLIN_SIG  \\\n",
       "0    Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "1                                                                            \n",
       "2    Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "3                                                                            \n",
       "4    Fasting plasma glucose level quantitative trai...  benign,association   \n",
       "..                                                 ...                 ...   \n",
       "105                                                                          \n",
       "106                                                                          \n",
       "107                                                                          \n",
       "108                                                                          \n",
       "109                                                                          \n",
       "\n",
       "                                    ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0    criteria provided, multiple submitters, no con...    8751   \n",
       "1                                                                \n",
       "2    criteria provided, multiple submitters, no con...    8751   \n",
       "3                                                                \n",
       "4    criteria provided, multiple submitters, no con...    8751   \n",
       "..                                                 ...     ...   \n",
       "105                                                              \n",
       "106                                                              \n",
       "107                                                              \n",
       "108                                                              \n",
       "109                                                              \n",
       "\n",
       "                 HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0    ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "1    ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "2    ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "3    ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "4                                    None                      \n",
       "..                 ...                ...                ...   \n",
       "105  ENST00000493621.1       n.100+316G>C                      \n",
       "106  ENST00000494795.1      n.3057-532G>A                      \n",
       "107  ENST00000494795.1      n.3057-532G>C                      \n",
       "108                                  None                      \n",
       "109                                  None                      \n",
       "\n",
       "    HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  \\\n",
       "0          p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062   \n",
       "1          p.Leu446Arg  tolerated(0.06)      benign(0)                   \n",
       "2          p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062   \n",
       "3          p.Leu147Arg  tolerated(0.06)  benign(0.286)                   \n",
       "4                 None                                  0.7067  0.9062   \n",
       "..                 ...              ...            ...     ...     ...   \n",
       "105               None                                                   \n",
       "106               None                                  0.3075  0.3797   \n",
       "107               None                                                   \n",
       "108               None                                  0.3075  0.3797   \n",
       "109               None                                                   \n",
       "\n",
       "     AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "1                                    3.982e-06              0              0   \n",
       "2    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "3                                    3.982e-06              0              0   \n",
       "4    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "..      ...     ...     ...     ...        ...            ...            ...   \n",
       "105                                                                            \n",
       "106  0.3415  0.3661  0.2087   0.227                                            \n",
       "107                                                                            \n",
       "108  0.3415  0.3661  0.2087   0.227                                            \n",
       "109                                                                            \n",
       "\n",
       "    gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0           0.4668         0.5108         0.6426         0.5902   \n",
       "1                0      5.444e-05              0              0   \n",
       "2           0.4668         0.5108         0.6426         0.5902   \n",
       "3                0      5.444e-05              0              0   \n",
       "4           0.4668         0.5108         0.6426         0.5902   \n",
       "..             ...            ...            ...            ...   \n",
       "105                                                               \n",
       "106                                                               \n",
       "107                                                               \n",
       "108                                                               \n",
       "109                                                               \n",
       "\n",
       "    gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "1                0              0                                            \n",
       "2           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "3                0              0                                            \n",
       "4           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "..             ...            ...        ...            ...            ...   \n",
       "105                                                                          \n",
       "106                                   0.2519         0.3536         0.2325   \n",
       "107                                                                          \n",
       "108                                   0.2519         0.3536         0.2325   \n",
       "109                                                                          \n",
       "\n",
       "    gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0            0.626         0.4585         0.5014         0.6378   \n",
       "1                                                                 \n",
       "2            0.626         0.4585         0.5014         0.6378   \n",
       "3                                                                 \n",
       "4            0.626         0.4585         0.5014         0.6378   \n",
       "..             ...            ...            ...            ...   \n",
       "105                                                               \n",
       "106         0.3173         0.1317         0.3875         0.2268   \n",
       "107                                                               \n",
       "108         0.3173         0.1317         0.3875         0.2268   \n",
       "109                                                               \n",
       "\n",
       "    gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  \\\n",
       "0           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "1                                                                5.444e-05   \n",
       "2           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "3                                                                5.444e-05   \n",
       "4           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "..             ...            ...            ...            ...        ...   \n",
       "105                                                                          \n",
       "106         0.2184         0.1789         0.2555         0.2035     0.3875   \n",
       "107                                                                          \n",
       "108         0.2184         0.1789         0.2555         0.2035     0.3875   \n",
       "109                                                                          \n",
       "\n",
       "     MAX_AF_POPS                         BIOTYPE   EXON INTRON  \\\n",
       "0            AFR                  protein coding  15/19          \n",
       "1    gnomADe EAS                  protein coding  15/19          \n",
       "2            AFR                  protein coding    5/7          \n",
       "3    gnomADe EAS                  protein coding    5/7          \n",
       "4            AFR                 retained intron                 \n",
       "..           ...                             ...    ...    ...   \n",
       "105               protein coding CDS not defined           1/1   \n",
       "106  gnomADg EAS                 retained intron           2/2   \n",
       "107                              retained intron           2/2   \n",
       "108  gnomADg EAS                        promoter                 \n",
       "109                                     promoter                 \n",
       "\n",
       "    Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                             L446P  cTg/cCg      1   \n",
       "1                             L446R  cTg/cGg      1   \n",
       "2                             L147P  cTg/cCg      1   \n",
       "3                             L147R  cTg/cGg      1   \n",
       "4                               nan               1   \n",
       "..                              ...      ...    ...   \n",
       "105                             nan               1   \n",
       "106                             nan               1   \n",
       "107                             nan               1   \n",
       "108                             nan                   \n",
       "109                             nan                   \n",
       "\n",
       "                                                PUBMED  \\\n",
       "0    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "1    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "2    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "3    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "4    21467728,30409984,18439548,21054877,21647738,2...   \n",
       "..                                                 ...   \n",
       "105  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "106  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "107  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "108  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "109  26780889,18940312,25018854,27458502,22719876,2...   \n",
       "\n",
       "                   consequence Consequence_score  IMPACT_score  \n",
       "0             missense variant              7/10           5.0  \n",
       "1             missense variant              7/10           5.0  \n",
       "2             missense variant              7/10           5.0  \n",
       "3             missense variant              7/10           5.0  \n",
       "4      downstream gene variant              2/10           1.5  \n",
       "..                         ...               ...           ...  \n",
       "105             intron variant              2/10           1.5  \n",
       "106             intron variant              2/10           1.5  \n",
       "107             intron variant              2/10           1.5  \n",
       "108  regulatory region variant              2/10           1.5  \n",
       "109  regulatory region variant              2/10           1.5  \n",
       "\n",
       "[110 rows x 57 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "8a8a6316",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td></td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.982e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5.444e-05</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td></td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C,G</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.6331</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.5108</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>21467728,30409984,18439548,21054877,21647738,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.227</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A,C</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>26780889,18940312,25018854,27458502,22719876,2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name       rsID  CHROM       POS REF  ALT Allele  \\\n",
       "0        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "1        GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "2        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "3        GCKR  rs1260326   chr2  27508073   T  C,G      G   \n",
       "4        GCKR  rs1260326   chr2  27508073   T  C,G      C   \n",
       "..        ...        ...    ...       ...  ..  ...    ...   \n",
       "105    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "106    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "107    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "108    SAMM50  rs2143571  chr22  43995806   G  A,C      A   \n",
       "109    SAMM50  rs2143571  chr22  43995806   G  A,C      C   \n",
       "\n",
       "                                      Consequence Consequence_score    IMPACT  \\\n",
       "0          missense variant,splice region variant              7/10  MODERATE   \n",
       "1          missense variant,splice region variant              7/10  MODERATE   \n",
       "2          missense variant,splice region variant              7/10  MODERATE   \n",
       "3          missense variant,splice region variant              7/10  MODERATE   \n",
       "4                         downstream gene variant              2/10  MODIFIER   \n",
       "..                                            ...               ...       ...   \n",
       "105  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "106  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "107  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "108                     regulatory region variant              2/10  MODIFIER   \n",
       "109                     regulatory region variant              2/10  MODIFIER   \n",
       "\n",
       "     IMPACT_score                                      ClinVar_CLNDN  \\\n",
       "0             5.0  Fasting plasma glucose level quantitative trai...   \n",
       "1             5.0                                                      \n",
       "2             5.0  Fasting plasma glucose level quantitative trai...   \n",
       "3             5.0                                                      \n",
       "4             1.5  Fasting plasma glucose level quantitative trai...   \n",
       "..            ...                                                ...   \n",
       "105           1.5                                                      \n",
       "106           1.5                                                      \n",
       "107           1.5                                                      \n",
       "108           1.5                                                      \n",
       "109           1.5                                                      \n",
       "\n",
       "               CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "0    benign,association  criteria provided, multiple submitters, no con...   \n",
       "1                                                                            \n",
       "2    benign,association  criteria provided, multiple submitters, no con...   \n",
       "3                                                                            \n",
       "4    benign,association  criteria provided, multiple submitters, no con...   \n",
       "..                  ...                                                ...   \n",
       "105                                                                          \n",
       "106                                                                          \n",
       "107                                                                          \n",
       "108                                                                          \n",
       "109                                                                          \n",
       "\n",
       "    ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0      8751  ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "1            ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "2      8751  ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "3            ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "4      8751                                  None                      \n",
       "..      ...                ...                ...                ...   \n",
       "105          ENST00000493621.1       n.100+316G>C                      \n",
       "106          ENST00000494795.1      n.3057-532G>A                      \n",
       "107          ENST00000494795.1      n.3057-532G>C                      \n",
       "108                                          None                      \n",
       "109                                          None                      \n",
       "\n",
       "    HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  \\\n",
       "0          p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062   \n",
       "1          p.Leu446Arg  tolerated(0.06)      benign(0)                   \n",
       "2          p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062   \n",
       "3          p.Leu147Arg  tolerated(0.06)  benign(0.286)                   \n",
       "4                 None                                  0.7067  0.9062   \n",
       "..                 ...              ...            ...     ...     ...   \n",
       "105               None                                                   \n",
       "106               None                                  0.3075  0.3797   \n",
       "107               None                                                   \n",
       "108               None                                  0.3075  0.3797   \n",
       "109               None                                                   \n",
       "\n",
       "     AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "1                                    3.982e-06              0              0   \n",
       "2    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "3                                    3.982e-06              0              0   \n",
       "4    0.6383  0.5188  0.5895  0.7996     0.6331         0.8679         0.6662   \n",
       "..      ...     ...     ...     ...        ...            ...            ...   \n",
       "105                                                                            \n",
       "106  0.3415  0.3661  0.2087   0.227                                            \n",
       "107                                                                            \n",
       "108  0.3415  0.3661  0.2087   0.227                                            \n",
       "109                                                                            \n",
       "\n",
       "    gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0           0.4668         0.5108         0.6426         0.5902   \n",
       "1                0      5.444e-05              0              0   \n",
       "2           0.4668         0.5108         0.6426         0.5902   \n",
       "3                0      5.444e-05              0              0   \n",
       "4           0.4668         0.5108         0.6426         0.5902   \n",
       "..             ...            ...            ...            ...   \n",
       "105                                                               \n",
       "106                                                               \n",
       "107                                                               \n",
       "108                                                               \n",
       "109                                                               \n",
       "\n",
       "    gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "1                0              0                                            \n",
       "2           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "3                0              0                                            \n",
       "4           0.6005          0.758       0.67         0.8582         0.7083   \n",
       "..             ...            ...        ...            ...            ...   \n",
       "105                                                                          \n",
       "106                                   0.2519         0.3536         0.2325   \n",
       "107                                                                          \n",
       "108                                   0.2519         0.3536         0.2325   \n",
       "109                                                                          \n",
       "\n",
       "    gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0            0.626         0.4585         0.5014         0.6378   \n",
       "1                                                                 \n",
       "2            0.626         0.4585         0.5014         0.6378   \n",
       "3                                                                 \n",
       "4            0.626         0.4585         0.5014         0.6378   \n",
       "..             ...            ...            ...            ...   \n",
       "105                                                               \n",
       "106         0.3173         0.1317         0.3875         0.2268   \n",
       "107                                                               \n",
       "108         0.3173         0.1317         0.3875         0.2268   \n",
       "109                                                               \n",
       "\n",
       "    gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF     MAX_AF  \\\n",
       "0           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "1                                                                5.444e-05   \n",
       "2           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "3                                                                5.444e-05   \n",
       "4           0.4904         0.5898         0.6106         0.7566     0.9062   \n",
       "..             ...            ...            ...            ...        ...   \n",
       "105                                                                          \n",
       "106         0.2184         0.1789         0.2555         0.2035     0.3875   \n",
       "107                                                                          \n",
       "108         0.2184         0.1789         0.2555         0.2035     0.3875   \n",
       "109                                                                          \n",
       "\n",
       "     MAX_AF_POPS                         BIOTYPE   EXON INTRON  \\\n",
       "0            AFR                  protein coding  15/19          \n",
       "1    gnomADe EAS                  protein coding  15/19          \n",
       "2            AFR                  protein coding    5/7          \n",
       "3    gnomADe EAS                  protein coding    5/7          \n",
       "4            AFR                 retained intron                 \n",
       "..           ...                             ...    ...    ...   \n",
       "105               protein coding CDS not defined           1/1   \n",
       "106  gnomADg EAS                 retained intron           2/2   \n",
       "107                              retained intron           2/2   \n",
       "108  gnomADg EAS                        promoter                 \n",
       "109                                     promoter                 \n",
       "\n",
       "    Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                             L446P  cTg/cCg      1   \n",
       "1                             L446R  cTg/cGg      1   \n",
       "2                             L147P  cTg/cCg      1   \n",
       "3                             L147R  cTg/cGg      1   \n",
       "4                               nan               1   \n",
       "..                              ...      ...    ...   \n",
       "105                             nan               1   \n",
       "106                             nan               1   \n",
       "107                             nan               1   \n",
       "108                             nan                   \n",
       "109                             nan                   \n",
       "\n",
       "                                                PUBMED  \n",
       "0    21467728,30409984,18439548,21054877,21647738,2...  \n",
       "1    21467728,30409984,18439548,21054877,21647738,2...  \n",
       "2    21467728,30409984,18439548,21054877,21647738,2...  \n",
       "3    21467728,30409984,18439548,21054877,21647738,2...  \n",
       "4    21467728,30409984,18439548,21054877,21647738,2...  \n",
       "..                                                 ...  \n",
       "105  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "106  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "107  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "108  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "109  26780889,18940312,25018854,27458502,22719876,2...  \n",
       "\n",
       "[110 rows x 56 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Allele',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "2daa3863",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS</td>\n",
       "      <td>Type1Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CEL</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HLADQ</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HLADR</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PTPN22</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IL2</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ERBB3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BACH2</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>C1QTNF6</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PRKQ</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>RGS1</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CCR5</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TAGAP</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>STAT3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FOXP3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>IRS1</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SLC2A2</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>WFS1</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCK</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PAX4</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>WRN</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ABCC8</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PDX1</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>AKT2</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HNF4A</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>KLF11</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ENPP1</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PPP1R3A</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>ACADS</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>LIPC</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>INSR</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>MLKL</td>\n",
       "      <td>MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NEUROD1</td>\n",
       "      <td>MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>SELP</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CDKAL1</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>TGFBR2</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>FABP2</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>HNF1B</td>\n",
       "      <td>Diabetic_Retinopathy, Diabetic_Nephropathy, Di...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>AQP2</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>AVPR2</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>AVP</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>PRKAG2</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>SCN9A</td>\n",
       "      <td>Diabetic_Neuropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>Diabetic_Neuropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name                                       Matched_Gene\n",
       "0      CTLA4                                      Type1Diabetes\n",
       "1        INS                                Type1Diabetes, MODY\n",
       "2      IL2RA                                      Type1Diabetes\n",
       "3      HNF1A                 Type1Diabetes, Type2Diabetes, MODY\n",
       "4        CEL                                      Type1Diabetes\n",
       "5      HLADQ                                      Type1Diabetes\n",
       "6      HLADR                                      Type1Diabetes\n",
       "7     PTPN22                                      Type1Diabetes\n",
       "8        IL2                                      Type1Diabetes\n",
       "9      ERBB3                                      Type1Diabetes\n",
       "10     BACH2                                      Type1Diabetes\n",
       "11   C1QTNF6                                      Type1Diabetes\n",
       "12      PRKQ                                      Type1Diabetes\n",
       "13      RGS1                                      Type1Diabetes\n",
       "14   IL18RAP                                      Type1Diabetes\n",
       "15      CCR5                                      Type1Diabetes\n",
       "16     TAGAP                                      Type1Diabetes\n",
       "17     SH2B3                                      Type1Diabetes\n",
       "18     STAT3                                      Type1Diabetes\n",
       "19     FOXP3                                      Type1Diabetes\n",
       "20      IRS1                                      Type2Diabetes\n",
       "21    SLC2A2                                      Type2Diabetes\n",
       "22      WFS1                                      Type2Diabetes\n",
       "23       GCK                                Type2Diabetes, MODY\n",
       "24      PAX4                                Type2Diabetes, MODY\n",
       "25       WRN                                      Type2Diabetes\n",
       "26    KCNJ11                                Type2Diabetes, MODY\n",
       "27     ABCC8                                Type2Diabetes, MODY\n",
       "28      PDX1                                Type2Diabetes, MODY\n",
       "29      AKT2                                      Type2Diabetes\n",
       "30     HNF4A                                Type2Diabetes, MODY\n",
       "31     KLF11                                Type2Diabetes, MODY\n",
       "32     ENPP1                                      Type2Diabetes\n",
       "33   PPP1R3A                                      Type2Diabetes\n",
       "34     ACADS                                      Type2Diabetes\n",
       "35      LIPC                                      Type2Diabetes\n",
       "36      INSR                                      Type2Diabetes\n",
       "37    TCF7L2                                      Type2Diabetes\n",
       "38      MLKL                                               MODY\n",
       "39   NEUROD1                                               MODY\n",
       "40      SELP                               Diabetic_Retinopathy\n",
       "41    CDKAL1                               Diabetic_Retinopathy\n",
       "42    TGFBR2                               Diabetic_Retinopathy\n",
       "43      HHEX                               Diabetic_Retinopathy\n",
       "44     FABP2                               Diabetic_Retinopathy\n",
       "45     HNF1B  Diabetic_Retinopathy, Diabetic_Nephropathy, Di...\n",
       "46      AQP2                               Diabetic_Nephropathy\n",
       "47     AVPR2                               Diabetic_Nephropathy\n",
       "48       AVP                               Diabetic_Nephropathy\n",
       "49     PARK7                               Diabetic_Nephropathy\n",
       "50    PRKAG2                               Diabetic_Nephropathy\n",
       "51     SCN9A                                Diabetic_Neuropathy\n",
       "52   ZSCAN20                                Diabetic_Neuropathy\n",
       "53      LMNA                              Risk of Heart Disease"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Diab_all_genes_conditions.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "94a1f63a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "1                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "2                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "3          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "4          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "...          ...           ...   ...        ...  ..  ..           ...   \n",
       "306164                       .  chrX  155459837   A   G  Heterozygous   \n",
       "306165      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306166      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306167      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306168      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1                intron variant              2/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3            synonymous variant              3/10       LOW           2.5   \n",
       "4       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "306164    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "306165           intron variant              2/10  MODIFIER           1.5   \n",
       "306166           intron variant              2/10  MODIFIER           1.5   \n",
       "306167  downstream gene variant              2/10  MODIFIER           1.5   \n",
       "306168    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                                              \n",
       "1                                                          ENST00000641515.2   \n",
       "2                                                                              \n",
       "3                                                          ENST00000641515.2   \n",
       "4                                                                              \n",
       "...              ...      ...                ...     ...                 ...   \n",
       "306164                                                                         \n",
       "306165                                                    ENST00000244174.11   \n",
       "306166                                                     ENST00000369423.7   \n",
       "306167                                                                         \n",
       "306168                                                                         \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                                  None  0/1  22   31   \n",
       "1               c.9+18C>T                                  None  0/1  22   31   \n",
       "2                    None                                  None  0/1  22   31   \n",
       "3                 c.36T>C  ENSP00000493376.2         p.Asn12%3D  0/1  67   45   \n",
       "4                    None                                  None  0/1  67   45   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "306164               None                                  None  0/1  71  103   \n",
       "306165        c.580-26C>T                                  None  0/1  31   12   \n",
       "306166        c.685-26C>T                                  None  0/1  31   12   \n",
       "306167               None                                  None  0/1  31   12   \n",
       "306168               None                                  None  0/1  31   12   \n",
       "\n",
       "         DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT PolyPhen AF  \\\n",
       "0        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "1        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "2        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "3        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "4        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "...     ...  ..  ..     ...        ...  ..  ..  ..  ..  ...      ... ..   \n",
       "306164  103  81  22  21.36%  6.7976E-8  54  27  14   8                    \n",
       "306165   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306166   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306167   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306168   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "\n",
       "       AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                     \n",
       "1                                                                     \n",
       "2                                                                     \n",
       "3                                                                     \n",
       "4                                                                     \n",
       "...       ...    ...    ...    ...    ...        ...            ...   \n",
       "306164                                                                \n",
       "306165                                       0.02802       0.001789   \n",
       "306166                                       0.02802       0.001789   \n",
       "306167                                       0.02802       0.001789   \n",
       "306168                                       0.02802       0.001789   \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.0527      9.934e-05         0.2012              0   \n",
       "306166         0.0527      9.934e-05         0.2012              0   \n",
       "306167         0.0527      9.934e-05         0.2012              0   \n",
       "306168         0.0527      9.934e-05         0.2012              0   \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03416        0.02306   \n",
       "1                                                      0.03416        0.02306   \n",
       "2                                                      0.03416        0.02306   \n",
       "3                                                      0.05175        0.04058   \n",
       "4                                                      0.05175        0.04058   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "306164                                                                          \n",
       "306165      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306166      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306167      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306168      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0              0.1667        0.02711        0.03333       0.001639   \n",
       "1              0.1667        0.02711        0.03333       0.001639   \n",
       "2              0.1667        0.02711        0.03333       0.001639   \n",
       "3             0.02941        0.04766        0.04545        0.06778   \n",
       "4             0.02941        0.04766        0.04545        0.06778   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0        0.02902       0.000288         0.2012   \n",
       "306166              0        0.02902       0.000288         0.2012   \n",
       "306167              0        0.02902       0.000288         0.2012   \n",
       "306168              0        0.02902       0.000288         0.2012   \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0           0.05        0.09167        0.05357   \n",
       "1                   0           0.05        0.09167        0.05357   \n",
       "2                   0           0.05        0.09167        0.05357   \n",
       "3              0.0303         0.1538        0.06103        0.05789   \n",
       "4              0.0303         0.1538        0.06103        0.05789   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0              0      0.0003382         0.0134   \n",
       "306166              0              0      0.0003382         0.0134   \n",
       "306167              0              0      0.0003382         0.0134   \n",
       "306168              0              0      0.0003382         0.0134   \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.02083  0.1667              gnomADg AMI   \n",
       "1             0.02083  0.1667              gnomADg AMI   \n",
       "2             0.02083  0.1667              gnomADg AMI   \n",
       "3              0.1089  0.1538              gnomADg MID   \n",
       "4              0.1089  0.1538              gnomADg MID   \n",
       "...               ...     ...                      ...   \n",
       "306164                                                   \n",
       "306165        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Gene Match  \\\n",
       "0                                  nan               1                No   \n",
       "1                                  nan               1                No   \n",
       "2                                  nan               1                No   \n",
       "3                                  N12  aaT/aaC      1                No   \n",
       "4                                  nan               1                No   \n",
       "...                                ...      ...    ...    ...        ...   \n",
       "306164                             nan              -1                No   \n",
       "306165                             nan               1                No   \n",
       "306166                             nan               1                No   \n",
       "306167                             nan               1                No   \n",
       "306168                             nan               1                No   \n",
       "\n",
       "       Matched_Gene  \n",
       "0                    \n",
       "1                    \n",
       "2                    \n",
       "3                    \n",
       "4                    \n",
       "...             ...  \n",
       "306164               \n",
       "306165               \n",
       "306166               \n",
       "306167               \n",
       "306168               \n",
       "\n",
       "[306169 rows x 70 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "1f5aaeb4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS</td>\n",
       "      <td>Type1Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CEL</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HLADQ</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HLADR</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PTPN22</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IL2</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ERBB3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BACH2</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>C1QTNF6</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PRKQ</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>RGS1</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CCR5</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TAGAP</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>STAT3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FOXP3</td>\n",
       "      <td>Type1Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>IRS1</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SLC2A2</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>WFS1</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCK</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PAX4</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>WRN</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ABCC8</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PDX1</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>AKT2</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HNF4A</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>KLF11</td>\n",
       "      <td>Type2Diabetes, MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ENPP1</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PPP1R3A</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>ACADS</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>LIPC</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>INSR</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>Type2Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>MLKL</td>\n",
       "      <td>MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NEUROD1</td>\n",
       "      <td>MODY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>SELP</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CDKAL1</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>TGFBR2</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>FABP2</td>\n",
       "      <td>Diabetic_Retinopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>HNF1B</td>\n",
       "      <td>Diabetic_Retinopathy, Diabetic_Nephropathy, Di...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>AQP2</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>AVPR2</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>AVP</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>PARK7</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>PRKAG2</td>\n",
       "      <td>Diabetic_Nephropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>SCN9A</td>\n",
       "      <td>Diabetic_Neuropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>Diabetic_Neuropathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>LMNA</td>\n",
       "      <td>Risk of Heart Disease</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Matched_Gene                                         Gene Match\n",
       "0         CTLA4                                      Type1Diabetes\n",
       "1           INS                                Type1Diabetes, MODY\n",
       "2         IL2RA                                      Type1Diabetes\n",
       "3         HNF1A                 Type1Diabetes, Type2Diabetes, MODY\n",
       "4           CEL                                      Type1Diabetes\n",
       "5         HLADQ                                      Type1Diabetes\n",
       "6         HLADR                                      Type1Diabetes\n",
       "7        PTPN22                                      Type1Diabetes\n",
       "8           IL2                                      Type1Diabetes\n",
       "9         ERBB3                                      Type1Diabetes\n",
       "10        BACH2                                      Type1Diabetes\n",
       "11      C1QTNF6                                      Type1Diabetes\n",
       "12         PRKQ                                      Type1Diabetes\n",
       "13         RGS1                                      Type1Diabetes\n",
       "14      IL18RAP                                      Type1Diabetes\n",
       "15         CCR5                                      Type1Diabetes\n",
       "16        TAGAP                                      Type1Diabetes\n",
       "17        SH2B3                                      Type1Diabetes\n",
       "18        STAT3                                      Type1Diabetes\n",
       "19        FOXP3                                      Type1Diabetes\n",
       "20         IRS1                                      Type2Diabetes\n",
       "21       SLC2A2                                      Type2Diabetes\n",
       "22         WFS1                                      Type2Diabetes\n",
       "23          GCK                                Type2Diabetes, MODY\n",
       "24         PAX4                                Type2Diabetes, MODY\n",
       "25          WRN                                      Type2Diabetes\n",
       "26       KCNJ11                                Type2Diabetes, MODY\n",
       "27        ABCC8                                Type2Diabetes, MODY\n",
       "28         PDX1                                Type2Diabetes, MODY\n",
       "29         AKT2                                      Type2Diabetes\n",
       "30        HNF4A                                Type2Diabetes, MODY\n",
       "31        KLF11                                Type2Diabetes, MODY\n",
       "32        ENPP1                                      Type2Diabetes\n",
       "33      PPP1R3A                                      Type2Diabetes\n",
       "34        ACADS                                      Type2Diabetes\n",
       "35         LIPC                                      Type2Diabetes\n",
       "36         INSR                                      Type2Diabetes\n",
       "37       TCF7L2                                      Type2Diabetes\n",
       "38         MLKL                                               MODY\n",
       "39      NEUROD1                                               MODY\n",
       "40         SELP                               Diabetic_Retinopathy\n",
       "41       CDKAL1                               Diabetic_Retinopathy\n",
       "42       TGFBR2                               Diabetic_Retinopathy\n",
       "43         HHEX                               Diabetic_Retinopathy\n",
       "44        FABP2                               Diabetic_Retinopathy\n",
       "45        HNF1B  Diabetic_Retinopathy, Diabetic_Nephropathy, Di...\n",
       "46         AQP2                               Diabetic_Nephropathy\n",
       "47        AVPR2                               Diabetic_Nephropathy\n",
       "48          AVP                               Diabetic_Nephropathy\n",
       "49        PARK7                               Diabetic_Nephropathy\n",
       "50       PRKAG2                               Diabetic_Nephropathy\n",
       "51        SCN9A                                Diabetic_Neuropathy\n",
       "52      ZSCAN20                                Diabetic_Neuropathy\n",
       "53         LMNA                              Risk of Heart Disease"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = df_gene.rename({'Gene Name':'Matched_Gene', 'Matched_Gene':'Gene Match'}, axis=1)\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "41b52aeb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "1                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "2                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "3          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "4          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "...          ...           ...   ...        ...  ..  ..           ...   \n",
       "306164                       .  chrX  155459837   A   G  Heterozygous   \n",
       "306165      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306166      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306167      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306168      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1                intron variant              2/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3            synonymous variant              3/10       LOW           2.5   \n",
       "4       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "306164    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "306165           intron variant              2/10  MODIFIER           1.5   \n",
       "306166           intron variant              2/10  MODIFIER           1.5   \n",
       "306167  downstream gene variant              2/10  MODIFIER           1.5   \n",
       "306168    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                                              \n",
       "1                                                          ENST00000641515.2   \n",
       "2                                                                              \n",
       "3                                                          ENST00000641515.2   \n",
       "4                                                                              \n",
       "...              ...      ...                ...     ...                 ...   \n",
       "306164                                                                         \n",
       "306165                                                    ENST00000244174.11   \n",
       "306166                                                     ENST00000369423.7   \n",
       "306167                                                                         \n",
       "306168                                                                         \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                                  None  0/1  22   31   \n",
       "1               c.9+18C>T                                  None  0/1  22   31   \n",
       "2                    None                                  None  0/1  22   31   \n",
       "3                 c.36T>C  ENSP00000493376.2         p.Asn12%3D  0/1  67   45   \n",
       "4                    None                                  None  0/1  67   45   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "306164               None                                  None  0/1  71  103   \n",
       "306165        c.580-26C>T                                  None  0/1  31   12   \n",
       "306166        c.685-26C>T                                  None  0/1  31   12   \n",
       "306167               None                                  None  0/1  31   12   \n",
       "306168               None                                  None  0/1  31   12   \n",
       "\n",
       "         DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT PolyPhen AF  \\\n",
       "0        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "1        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "2        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "3        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "4        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "...     ...  ..  ..     ...        ...  ..  ..  ..  ..  ...      ... ..   \n",
       "306164  103  81  22  21.36%  6.7976E-8  54  27  14   8                    \n",
       "306165   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306166   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306167   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306168   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "\n",
       "       AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                     \n",
       "1                                                                     \n",
       "2                                                                     \n",
       "3                                                                     \n",
       "4                                                                     \n",
       "...       ...    ...    ...    ...    ...        ...            ...   \n",
       "306164                                                                \n",
       "306165                                       0.02802       0.001789   \n",
       "306166                                       0.02802       0.001789   \n",
       "306167                                       0.02802       0.001789   \n",
       "306168                                       0.02802       0.001789   \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.0527      9.934e-05         0.2012              0   \n",
       "306166         0.0527      9.934e-05         0.2012              0   \n",
       "306167         0.0527      9.934e-05         0.2012              0   \n",
       "306168         0.0527      9.934e-05         0.2012              0   \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03416        0.02306   \n",
       "1                                                      0.03416        0.02306   \n",
       "2                                                      0.03416        0.02306   \n",
       "3                                                      0.05175        0.04058   \n",
       "4                                                      0.05175        0.04058   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "306164                                                                          \n",
       "306165      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306166      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306167      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306168      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0              0.1667        0.02711        0.03333       0.001639   \n",
       "1              0.1667        0.02711        0.03333       0.001639   \n",
       "2              0.1667        0.02711        0.03333       0.001639   \n",
       "3             0.02941        0.04766        0.04545        0.06778   \n",
       "4             0.02941        0.04766        0.04545        0.06778   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0        0.02902       0.000288         0.2012   \n",
       "306166              0        0.02902       0.000288         0.2012   \n",
       "306167              0        0.02902       0.000288         0.2012   \n",
       "306168              0        0.02902       0.000288         0.2012   \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0           0.05        0.09167        0.05357   \n",
       "1                   0           0.05        0.09167        0.05357   \n",
       "2                   0           0.05        0.09167        0.05357   \n",
       "3              0.0303         0.1538        0.06103        0.05789   \n",
       "4              0.0303         0.1538        0.06103        0.05789   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0              0      0.0003382         0.0134   \n",
       "306166              0              0      0.0003382         0.0134   \n",
       "306167              0              0      0.0003382         0.0134   \n",
       "306168              0              0      0.0003382         0.0134   \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.02083  0.1667              gnomADg AMI   \n",
       "1             0.02083  0.1667              gnomADg AMI   \n",
       "2             0.02083  0.1667              gnomADg AMI   \n",
       "3              0.1089  0.1538              gnomADg MID   \n",
       "4              0.1089  0.1538              gnomADg MID   \n",
       "...               ...     ...                      ...   \n",
       "306164                                                   \n",
       "306165        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \n",
       "0                                  nan               1                      \n",
       "1                                  nan               1                      \n",
       "2                                  nan               1                      \n",
       "3                                  N12  aaT/aaC      1                      \n",
       "4                                  nan               1                      \n",
       "...                                ...      ...    ...    ...          ...  \n",
       "306164                             nan              -1                      \n",
       "306165                             nan               1                      \n",
       "306166                             nan               1                      \n",
       "306167                             nan               1                      \n",
       "306168                             nan               1                      \n",
       "\n",
       "[306169 rows x 69 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2.drop(columns=['Gene Match'], axis=1)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "61e99d95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Gene Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "1                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "2                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "3          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "4          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "...          ...           ...   ...        ...  ..  ..           ...   \n",
       "306164                       .  chrX  155459837   A   G  Heterozygous   \n",
       "306165      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306166      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306167      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306168      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1                intron variant              2/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3            synonymous variant              3/10       LOW           2.5   \n",
       "4       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "306164    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "306165           intron variant              2/10  MODIFIER           1.5   \n",
       "306166           intron variant              2/10  MODIFIER           1.5   \n",
       "306167  downstream gene variant              2/10  MODIFIER           1.5   \n",
       "306168    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                                              \n",
       "1                                                          ENST00000641515.2   \n",
       "2                                                                              \n",
       "3                                                          ENST00000641515.2   \n",
       "4                                                                              \n",
       "...              ...      ...                ...     ...                 ...   \n",
       "306164                                                                         \n",
       "306165                                                    ENST00000244174.11   \n",
       "306166                                                     ENST00000369423.7   \n",
       "306167                                                                         \n",
       "306168                                                                         \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                                  None  0/1  22   31   \n",
       "1               c.9+18C>T                                  None  0/1  22   31   \n",
       "2                    None                                  None  0/1  22   31   \n",
       "3                 c.36T>C  ENSP00000493376.2         p.Asn12%3D  0/1  67   45   \n",
       "4                    None                                  None  0/1  67   45   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "306164               None                                  None  0/1  71  103   \n",
       "306165        c.580-26C>T                                  None  0/1  31   12   \n",
       "306166        c.685-26C>T                                  None  0/1  31   12   \n",
       "306167               None                                  None  0/1  31   12   \n",
       "306168               None                                  None  0/1  31   12   \n",
       "\n",
       "         DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT PolyPhen AF  \\\n",
       "0        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "1        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "2        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "3        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "4        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "...     ...  ..  ..     ...        ...  ..  ..  ..  ..  ...      ... ..   \n",
       "306164  103  81  22  21.36%  6.7976E-8  54  27  14   8                    \n",
       "306165   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306166   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306167   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306168   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "\n",
       "       AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                     \n",
       "1                                                                     \n",
       "2                                                                     \n",
       "3                                                                     \n",
       "4                                                                     \n",
       "...       ...    ...    ...    ...    ...        ...            ...   \n",
       "306164                                                                \n",
       "306165                                       0.02802       0.001789   \n",
       "306166                                       0.02802       0.001789   \n",
       "306167                                       0.02802       0.001789   \n",
       "306168                                       0.02802       0.001789   \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.0527      9.934e-05         0.2012              0   \n",
       "306166         0.0527      9.934e-05         0.2012              0   \n",
       "306167         0.0527      9.934e-05         0.2012              0   \n",
       "306168         0.0527      9.934e-05         0.2012              0   \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03416        0.02306   \n",
       "1                                                      0.03416        0.02306   \n",
       "2                                                      0.03416        0.02306   \n",
       "3                                                      0.05175        0.04058   \n",
       "4                                                      0.05175        0.04058   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "306164                                                                          \n",
       "306165      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306166      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306167      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306168      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0              0.1667        0.02711        0.03333       0.001639   \n",
       "1              0.1667        0.02711        0.03333       0.001639   \n",
       "2              0.1667        0.02711        0.03333       0.001639   \n",
       "3             0.02941        0.04766        0.04545        0.06778   \n",
       "4             0.02941        0.04766        0.04545        0.06778   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0        0.02902       0.000288         0.2012   \n",
       "306166              0        0.02902       0.000288         0.2012   \n",
       "306167              0        0.02902       0.000288         0.2012   \n",
       "306168              0        0.02902       0.000288         0.2012   \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0           0.05        0.09167        0.05357   \n",
       "1                   0           0.05        0.09167        0.05357   \n",
       "2                   0           0.05        0.09167        0.05357   \n",
       "3              0.0303         0.1538        0.06103        0.05789   \n",
       "4              0.0303         0.1538        0.06103        0.05789   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0              0      0.0003382         0.0134   \n",
       "306166              0              0      0.0003382         0.0134   \n",
       "306167              0              0      0.0003382         0.0134   \n",
       "306168              0              0      0.0003382         0.0134   \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.02083  0.1667              gnomADg AMI   \n",
       "1             0.02083  0.1667              gnomADg AMI   \n",
       "2             0.02083  0.1667              gnomADg AMI   \n",
       "3              0.1089  0.1538              gnomADg MID   \n",
       "4              0.1089  0.1538              gnomADg MID   \n",
       "...               ...     ...                      ...   \n",
       "306164                                                   \n",
       "306165        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Matched_Gene  \\\n",
       "0                                  nan               1                       \n",
       "1                                  nan               1                       \n",
       "2                                  nan               1                       \n",
       "3                                  N12  aaT/aaC      1                       \n",
       "4                                  nan               1                       \n",
       "...                                ...      ...    ...    ...          ...   \n",
       "306164                             nan              -1                       \n",
       "306165                             nan               1                       \n",
       "306166                             nan               1                       \n",
       "306167                             nan               1                       \n",
       "306168                             nan               1                       \n",
       "\n",
       "       Gene Match  \n",
       "0              No  \n",
       "1              No  \n",
       "2              No  \n",
       "3              No  \n",
       "4              No  \n",
       "...           ...  \n",
       "306164         No  \n",
       "306165         No  \n",
       "306166         No  \n",
       "306167         No  \n",
       "306168         No  \n",
       "\n",
       "[306169 rows x 70 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_2, df_gene, on= 'Matched_Gene', how = 'left', sort = False)\n",
    "merged_2['Gene Match'] = merged_2['Gene Match'].fillna('No')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "ee9d1255",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No                                    304214\n",
       "Type2Diabetes, MODY                      589\n",
       "Type2Diabetes                            416\n",
       "Type1Diabetes                            389\n",
       "Type1Diabetes, Type2Diabetes, MODY       206\n",
       "Diabetic_Nephropathy                     124\n",
       "Diabetic_Neuropathy                      117\n",
       "Type1Diabetes, MODY                       65\n",
       "MODY                                      34\n",
       "Diabetic_Retinopathy                      15\n",
       "Name: Gene Match, dtype: int64"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene Match'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "994da10d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
       "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
       "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
       "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
       "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
       "       'PolyPhen', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',\n",
       "       'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF',\n",
       "       'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF',\n",
       "       'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF',\n",
       "       'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF',\n",
       "       'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF',\n",
       "       'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Matched_Gene', 'Gene Match'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "b6a62b84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene Match          rsID CHROM        POS REF ALT  \\\n",
       "0                        No  rs1416222198  chr1      65591   C   T   \n",
       "1                        No  rs1416222198  chr1      65591   C   T   \n",
       "2                        No  rs1416222198  chr1      65591   C   T   \n",
       "3          OR4F5         No  rs1260343719  chr1      69063   T   C   \n",
       "4          OR4F5         No  rs1260343719  chr1      69063   T   C   \n",
       "...          ...        ...           ...   ...        ...  ..  ..   \n",
       "306164                   No             .  chrX  155459837   A   G   \n",
       "306165      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306166      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306167      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306168      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous           intron variant              2/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3       Heterozygous       synonymous variant              3/10       LOW   \n",
       "4       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "306164  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "306165  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306166  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306167  Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "306168  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                2.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "306164           1.5                                                     \n",
       "306165           1.5                                                     \n",
       "306166           1.5                                                     \n",
       "306167           1.5                                                     \n",
       "306168           1.5                                                     \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                                        None                      \n",
       "1        ENST00000641515.2          c.9+18C>T                      \n",
       "2                                        None                      \n",
       "3        ENST00000641515.2            c.36T>C  ENSP00000493376.2   \n",
       "4                                        None                      \n",
       "...                    ...                ...                ...   \n",
       "306164                                   None                      \n",
       "306165  ENST00000244174.11        c.580-26C>T                      \n",
       "306166   ENST00000369423.7        c.685-26C>T                      \n",
       "306167                                   None                      \n",
       "306168                                   None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT  GQ  SDP   DP  RD  AD    FREQ       PVAL RDF  \\\n",
       "0                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "1                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "2                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "3              p.Asn12%3D  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "4                    None  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "...                   ...  ...  ..  ...  ...  ..  ..     ...        ...  ..   \n",
       "306164               None  0/1  71  103  103  81  22  21.36%  6.7976E-8  54   \n",
       "306165               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306166               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306167               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306168               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "\n",
       "       RDR ADF ADR SIFT PolyPhen AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF  \\\n",
       "0        7   3   4                                                       \n",
       "1        7   3   4                                                       \n",
       "2        7   3   4                                                       \n",
       "3        3  18   1                                                       \n",
       "4        3  18   1                                                       \n",
       "...     ..  ..  ..  ...      ... ..    ...    ...    ...    ...    ...   \n",
       "306164  27  14   8                                                       \n",
       "306165   0   8   0                                                       \n",
       "306166   0   8   0                                                       \n",
       "306167   0   8   0                                                       \n",
       "306168   0   8   0                                                       \n",
       "\n",
       "       gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                               \n",
       "1                                                                               \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...           ...            ...            ...            ...            ...   \n",
       "306164                                                                          \n",
       "306165    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306166    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306167    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306168    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.05175   \n",
       "4                                                                     0.05175   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "306164                                                                          \n",
       "306165              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306166              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306167              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306168              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.04058        0.02941        0.04766        0.04545   \n",
       "4             0.04058        0.02941        0.04766        0.04545   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165       0.001739              0        0.02902       0.000288   \n",
       "306166       0.001739              0        0.02902       0.000288   \n",
       "306167       0.001739              0        0.02902       0.000288   \n",
       "306168       0.001739              0        0.02902       0.000288   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3             0.06778         0.0303         0.1538        0.06103   \n",
       "4             0.06778         0.0303         0.1538        0.06103   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.2012              0              0      0.0003382   \n",
       "306166         0.2012              0              0      0.0003382   \n",
       "306167         0.2012              0              0      0.0003382   \n",
       "306168         0.2012              0              0      0.0003382   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "1             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "2             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "3             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "4             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "...               ...            ...     ...                      ...   \n",
       "306164                                                                  \n",
       "306165         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  nan               1         \n",
       "1                                  nan               1         \n",
       "2                                  nan               1         \n",
       "3                                  N12  aaT/aaC      1         \n",
       "4                                  nan               1         \n",
       "...                                ...      ...    ...    ...  \n",
       "306164                             nan              -1         \n",
       "306165                             nan               1         \n",
       "306166                             nan               1         \n",
       "306167                             nan               1         \n",
       "306168                             nan               1         \n",
       "\n",
       "[306169 rows x 69 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'Gene Match', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "274b8615",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd2d9793",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd5c3858",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fce8c05",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81c0b252",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0067af31",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4de75dd6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6f9325e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3e8ef59a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "0                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "1                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "2                 rs1416222198  chr1      65591   C   T  Heterozygous   \n",
       "3          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "4          OR4F5  rs1260343719  chr1      69063   T   C  Heterozygous   \n",
       "...          ...           ...   ...        ...  ..  ..           ...   \n",
       "306164                       .  chrX  155459837   A   G  Heterozygous   \n",
       "306165      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306166      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306167      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "306168      IL9R     rs2073467  chrX  156005252   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1                intron variant              2/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3            synonymous variant              3/10       LOW           2.5   \n",
       "4       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "306164    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "306165           intron variant              2/10  MODIFIER           1.5   \n",
       "306166           intron variant              2/10  MODIFIER           1.5   \n",
       "306167  downstream gene variant              2/10  MODIFIER           1.5   \n",
       "306168    upstream gene variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "0                                                                              \n",
       "1                                                          ENST00000641515.2   \n",
       "2                                                                              \n",
       "3                                                          ENST00000641515.2   \n",
       "4                                                                              \n",
       "...              ...      ...                ...     ...                 ...   \n",
       "306164                                                                         \n",
       "306165                                                    ENST00000244174.11   \n",
       "306166                                                     ENST00000369423.7   \n",
       "306167                                                                         \n",
       "306168                                                                         \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                                  None  0/1  22   31   \n",
       "1               c.9+18C>T                                  None  0/1  22   31   \n",
       "2                    None                                  None  0/1  22   31   \n",
       "3                 c.36T>C  ENSP00000493376.2         p.Asn12%3D  0/1  67   45   \n",
       "4                    None                                  None  0/1  67   45   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "306164               None                                  None  0/1  71  103   \n",
       "306165        c.580-26C>T                                  None  0/1  31   12   \n",
       "306166        c.685-26C>T                                  None  0/1  31   12   \n",
       "306167               None                                  None  0/1  31   12   \n",
       "306168               None                                  None  0/1  31   12   \n",
       "\n",
       "         DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT PolyPhen AF  \\\n",
       "0        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "1        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "2        31  24   7  22.58%  5.3469E-3  17   7   3   4                    \n",
       "3        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "4        45  26  19  42.22%  1.6979E-7  23   3  18   1                    \n",
       "...     ...  ..  ..     ...        ...  ..  ..  ..  ..  ...      ... ..   \n",
       "306164  103  81  22  21.36%  6.7976E-8  54  27  14   8                    \n",
       "306165   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306166   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306167   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "306168   12   4   8  66.67%  6.7304E-4   4   0   8   0                    \n",
       "\n",
       "       AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "0                                                                     \n",
       "1                                                                     \n",
       "2                                                                     \n",
       "3                                                                     \n",
       "4                                                                     \n",
       "...       ...    ...    ...    ...    ...        ...            ...   \n",
       "306164                                                                \n",
       "306165                                       0.02802       0.001789   \n",
       "306166                                       0.02802       0.001789   \n",
       "306167                                       0.02802       0.001789   \n",
       "306168                                       0.02802       0.001789   \n",
       "\n",
       "       gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.0527      9.934e-05         0.2012              0   \n",
       "306166         0.0527      9.934e-05         0.2012              0   \n",
       "306167         0.0527      9.934e-05         0.2012              0   \n",
       "306168         0.0527      9.934e-05         0.2012              0   \n",
       "\n",
       "       gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "0                                                      0.03416        0.02306   \n",
       "1                                                      0.03416        0.02306   \n",
       "2                                                      0.03416        0.02306   \n",
       "3                                                      0.05175        0.04058   \n",
       "4                                                      0.05175        0.04058   \n",
       "...               ...            ...            ...        ...            ...   \n",
       "306164                                                                          \n",
       "306165      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306166      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306167      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "306168      0.0005125        0.01736        0.04299    0.01208       0.001739   \n",
       "\n",
       "       gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "0              0.1667        0.02711        0.03333       0.001639   \n",
       "1              0.1667        0.02711        0.03333       0.001639   \n",
       "2              0.1667        0.02711        0.03333       0.001639   \n",
       "3             0.02941        0.04766        0.04545        0.06778   \n",
       "4             0.02941        0.04766        0.04545        0.06778   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0        0.02902       0.000288         0.2012   \n",
       "306166              0        0.02902       0.000288         0.2012   \n",
       "306167              0        0.02902       0.000288         0.2012   \n",
       "306168              0        0.02902       0.000288         0.2012   \n",
       "\n",
       "       gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "0                   0           0.05        0.09167        0.05357   \n",
       "1                   0           0.05        0.09167        0.05357   \n",
       "2                   0           0.05        0.09167        0.05357   \n",
       "3              0.0303         0.1538        0.06103        0.05789   \n",
       "4              0.0303         0.1538        0.06103        0.05789   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165              0              0      0.0003382         0.0134   \n",
       "306166              0              0      0.0003382         0.0134   \n",
       "306167              0              0      0.0003382         0.0134   \n",
       "306168              0              0      0.0003382         0.0134   \n",
       "\n",
       "       gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.02083  0.1667              gnomADg AMI   \n",
       "1             0.02083  0.1667              gnomADg AMI   \n",
       "2             0.02083  0.1667              gnomADg AMI   \n",
       "3              0.1089  0.1538              gnomADg MID   \n",
       "4              0.1089  0.1538              gnomADg MID   \n",
       "...               ...     ...                      ...   \n",
       "306164                                                   \n",
       "306165        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Gene_Match  \n",
       "0                                  nan               1                No  \n",
       "1                                  nan               1                No  \n",
       "2                                  nan               1                No  \n",
       "3                                  N12  aaT/aaC      1                No  \n",
       "4                                  nan               1                No  \n",
       "...                                ...      ...    ...    ...        ...  \n",
       "306164                             nan              -1                No  \n",
       "306165                             nan               1                No  \n",
       "306166                             nan               1                No  \n",
       "306167                             nan               1                No  \n",
       "306168                             nan               1                No  \n",
       "\n",
       "[306169 rows x 69 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in merged_2['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            merged_2.loc[merged_2['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "aad67a3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     303375\n",
       "Yes      2794\n",
       "Name: Gene_Match, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.Gene_Match.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "5eed38f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match          rsID CHROM        POS REF ALT  \\\n",
       "0                        No  rs1416222198  chr1      65591   C   T   \n",
       "1                        No  rs1416222198  chr1      65591   C   T   \n",
       "2                        No  rs1416222198  chr1      65591   C   T   \n",
       "3          OR4F5         No  rs1260343719  chr1      69063   T   C   \n",
       "4          OR4F5         No  rs1260343719  chr1      69063   T   C   \n",
       "...          ...        ...           ...   ...        ...  ..  ..   \n",
       "306164                   No             .  chrX  155459837   A   G   \n",
       "306165      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306166      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306167      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306168      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous           intron variant              2/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3       Heterozygous       synonymous variant              3/10       LOW   \n",
       "4       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "306164  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "306165  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306166  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306167  Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "306168  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                2.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "306164           1.5                                                     \n",
       "306165           1.5                                                     \n",
       "306166           1.5                                                     \n",
       "306167           1.5                                                     \n",
       "306168           1.5                                                     \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                                        None                      \n",
       "1        ENST00000641515.2          c.9+18C>T                      \n",
       "2                                        None                      \n",
       "3        ENST00000641515.2            c.36T>C  ENSP00000493376.2   \n",
       "4                                        None                      \n",
       "...                    ...                ...                ...   \n",
       "306164                                   None                      \n",
       "306165  ENST00000244174.11        c.580-26C>T                      \n",
       "306166   ENST00000369423.7        c.685-26C>T                      \n",
       "306167                                   None                      \n",
       "306168                                   None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT  GQ  SDP   DP  RD  AD    FREQ       PVAL RDF  \\\n",
       "0                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "1                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "2                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "3              p.Asn12%3D  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "4                    None  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "...                   ...  ...  ..  ...  ...  ..  ..     ...        ...  ..   \n",
       "306164               None  0/1  71  103  103  81  22  21.36%  6.7976E-8  54   \n",
       "306165               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306166               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306167               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306168               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "\n",
       "       RDR ADF ADR SIFT PolyPhen AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF  \\\n",
       "0        7   3   4                                                       \n",
       "1        7   3   4                                                       \n",
       "2        7   3   4                                                       \n",
       "3        3  18   1                                                       \n",
       "4        3  18   1                                                       \n",
       "...     ..  ..  ..  ...      ... ..    ...    ...    ...    ...    ...   \n",
       "306164  27  14   8                                                       \n",
       "306165   0   8   0                                                       \n",
       "306166   0   8   0                                                       \n",
       "306167   0   8   0                                                       \n",
       "306168   0   8   0                                                       \n",
       "\n",
       "       gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                               \n",
       "1                                                                               \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...           ...            ...            ...            ...            ...   \n",
       "306164                                                                          \n",
       "306165    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306166    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306167    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306168    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.05175   \n",
       "4                                                                     0.05175   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "306164                                                                          \n",
       "306165              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306166              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306167              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306168              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.04058        0.02941        0.04766        0.04545   \n",
       "4             0.04058        0.02941        0.04766        0.04545   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165       0.001739              0        0.02902       0.000288   \n",
       "306166       0.001739              0        0.02902       0.000288   \n",
       "306167       0.001739              0        0.02902       0.000288   \n",
       "306168       0.001739              0        0.02902       0.000288   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3             0.06778         0.0303         0.1538        0.06103   \n",
       "4             0.06778         0.0303         0.1538        0.06103   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.2012              0              0      0.0003382   \n",
       "306166         0.2012              0              0      0.0003382   \n",
       "306167         0.2012              0              0      0.0003382   \n",
       "306168         0.2012              0              0      0.0003382   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "1             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "2             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "3             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "4             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "...               ...            ...     ...                      ...   \n",
       "306164                                                                  \n",
       "306165         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  nan               1         \n",
       "1                                  nan               1         \n",
       "2                                  nan               1         \n",
       "3                                  N12  aaT/aaC      1         \n",
       "4                                  nan               1         \n",
       "...                                ...      ...    ...    ...  \n",
       "306164                             nan              -1         \n",
       "306165                             nan               1         \n",
       "306166                             nan               1         \n",
       "306167                             nan               1         \n",
       "306168                             nan               1         \n",
       "\n",
       "[306169 rows x 69 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "ec428ef8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Diab_lit_variants_all.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "df_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHGLBS578/Hypoglycemia_lit_variants.xlsx', index = False) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "beb8471d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978691</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978908</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120997590</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120996531</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978928</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4761</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17408513</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4798</th>\n",
       "      <td>chr11</td>\n",
       "      <td>66616700</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4803</th>\n",
       "      <td>chr11</td>\n",
       "      <td>66616749</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4806</th>\n",
       "      <td>chr11</td>\n",
       "      <td>66616734</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4809</th>\n",
       "      <td>chr11</td>\n",
       "      <td>66625360</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1777 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS Literature Covered/Not_Covered\n",
       "0     chr12  120978691        Yes             Covered\n",
       "1     chr12  120978908        Yes             Covered\n",
       "2     chr12  120997590        Yes             Covered\n",
       "3     chr12  120996531        Yes             Covered\n",
       "4     chr12  120978928        Yes             Covered\n",
       "...     ...        ...        ...                 ...\n",
       "4761  chr11   17408513        Yes             Covered\n",
       "4798  chr11   66616700        Yes             Covered\n",
       "4803  chr11   66616749        Yes             Covered\n",
       "4806  chr11   66616734        Yes             Covered\n",
       "4809  chr11   66625360        Yes             Covered\n",
       "\n",
       "[1777 rows x 4 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "150d0781",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene Match          rsID CHROM        POS REF ALT  \\\n",
       "0                        No  rs1416222198  chr1      65591   C   T   \n",
       "1                        No  rs1416222198  chr1      65591   C   T   \n",
       "2                        No  rs1416222198  chr1      65591   C   T   \n",
       "3          OR4F5         No  rs1260343719  chr1      69063   T   C   \n",
       "4          OR4F5         No  rs1260343719  chr1      69063   T   C   \n",
       "...          ...        ...           ...   ...        ...  ..  ..   \n",
       "306164                   No             .  chrX  155459837   A   G   \n",
       "306165      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306166      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306167      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "306168      IL9R         No     rs2073467  chrX  156005252   C   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous           intron variant              2/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3       Heterozygous       synonymous variant              3/10       LOW   \n",
       "4       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "306164  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "306165  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306166  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306167  Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "306168  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                2.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "306164           1.5                                                     \n",
       "306165           1.5                                                     \n",
       "306166           1.5                                                     \n",
       "306167           1.5                                                     \n",
       "306168           1.5                                                     \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                                        None                      \n",
       "1        ENST00000641515.2          c.9+18C>T                      \n",
       "2                                        None                      \n",
       "3        ENST00000641515.2            c.36T>C  ENSP00000493376.2   \n",
       "4                                        None                      \n",
       "...                    ...                ...                ...   \n",
       "306164                                   None                      \n",
       "306165  ENST00000244174.11        c.580-26C>T                      \n",
       "306166   ENST00000369423.7        c.685-26C>T                      \n",
       "306167                                   None                      \n",
       "306168                                   None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT  GQ  SDP   DP  RD  AD    FREQ       PVAL RDF  \\\n",
       "0                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "1                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "2                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "3              p.Asn12%3D  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "4                    None  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "...                   ...  ...  ..  ...  ...  ..  ..     ...        ...  ..   \n",
       "306164               None  0/1  71  103  103  81  22  21.36%  6.7976E-8  54   \n",
       "306165               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306166               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306167               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306168               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "\n",
       "       RDR ADF ADR SIFT PolyPhen AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF  \\\n",
       "0        7   3   4                                                       \n",
       "1        7   3   4                                                       \n",
       "2        7   3   4                                                       \n",
       "3        3  18   1                                                       \n",
       "4        3  18   1                                                       \n",
       "...     ..  ..  ..  ...      ... ..    ...    ...    ...    ...    ...   \n",
       "306164  27  14   8                                                       \n",
       "306165   0   8   0                                                       \n",
       "306166   0   8   0                                                       \n",
       "306167   0   8   0                                                       \n",
       "306168   0   8   0                                                       \n",
       "\n",
       "       gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                               \n",
       "1                                                                               \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...           ...            ...            ...            ...            ...   \n",
       "306164                                                                          \n",
       "306165    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306166    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306167    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306168    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.05175   \n",
       "4                                                                     0.05175   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "306164                                                                          \n",
       "306165              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306166              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306167              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306168              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.04058        0.02941        0.04766        0.04545   \n",
       "4             0.04058        0.02941        0.04766        0.04545   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165       0.001739              0        0.02902       0.000288   \n",
       "306166       0.001739              0        0.02902       0.000288   \n",
       "306167       0.001739              0        0.02902       0.000288   \n",
       "306168       0.001739              0        0.02902       0.000288   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3             0.06778         0.0303         0.1538        0.06103   \n",
       "4             0.06778         0.0303         0.1538        0.06103   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.2012              0              0      0.0003382   \n",
       "306166         0.2012              0              0      0.0003382   \n",
       "306167         0.2012              0              0      0.0003382   \n",
       "306168         0.2012              0              0      0.0003382   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "1             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "2             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "3             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "4             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "...               ...            ...     ...                      ...   \n",
       "306164                                                                  \n",
       "306165         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Literature  \\\n",
       "0                                  nan               1                No   \n",
       "1                                  nan               1                No   \n",
       "2                                  nan               1                No   \n",
       "3                                  N12  aaT/aaC      1                No   \n",
       "4                                  nan               1                No   \n",
       "...                                ...      ...    ...    ...        ...   \n",
       "306164                             nan              -1                No   \n",
       "306165                             nan               1                No   \n",
       "306166                             nan               1                No   \n",
       "306167                             nan               1                No   \n",
       "306168                             nan               1                No   \n",
       "\n",
       "       Covered/Not_Covered  \n",
       "0                      NaN  \n",
       "1                      NaN  \n",
       "2                      NaN  \n",
       "3                      NaN  \n",
       "4                      NaN  \n",
       "...                    ...  \n",
       "306164                 NaN  \n",
       "306165                 NaN  \n",
       "306166                 NaN  \n",
       "306167                 NaN  \n",
       "306168                 NaN  \n",
       "\n",
       "[306169 rows x 71 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "b12cbc55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     306097\n",
       "Yes        72\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "1d6dd43d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69781</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Maturity onset diabetes mellitus in young,Nonp...</td>\n",
       "      <td>risk factor,benign</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>14937</td>\n",
       "      <td>ENST00000257555.11</td>\n",
       "      <td>c.79A&gt;C</td>\n",
       "      <td>ENSP00000257555.5</td>\n",
       "      <td>p.Ile27Leu</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>45%</td>\n",
       "      <td>3.1919E-7</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>tolerated(0.11)</td>\n",
       "      <td>benign(0.282)</td>\n",
       "      <td>0.2985</td>\n",
       "      <td>0.0832</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.339</td>\n",
       "      <td>0.4325</td>\n",
       "      <td>0.3549</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>0.3666</td>\n",
       "      <td>0.4666</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.3992</td>\n",
       "      <td>0.3365</td>\n",
       "      <td>0.3613</td>\n",
       "      <td>0.4268</td>\n",
       "      <td>0.2888</td>\n",
       "      <td>0.1248</td>\n",
       "      <td>0.3998</td>\n",
       "      <td>0.3444</td>\n",
       "      <td>0.4579</td>\n",
       "      <td>0.4113</td>\n",
       "      <td>0.4038</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>0.3279</td>\n",
       "      <td>0.3109</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/10</td>\n",
       "      <td></td>\n",
       "      <td>I27L</td>\n",
       "      <td>Atc/Ctc</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,21094359,21647738,21931794,21937998,2...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69782</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Maturity onset diabetes mellitus in young,Nonp...</td>\n",
       "      <td>risk factor,benign</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>14937</td>\n",
       "      <td>ENST00000400024.6</td>\n",
       "      <td>c.79A&gt;C</td>\n",
       "      <td>ENSP00000476181.1</td>\n",
       "      <td>p.Ile27Leu</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>45%</td>\n",
       "      <td>3.1919E-7</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>tolerated(0.11)</td>\n",
       "      <td>probably damaging(0.938)</td>\n",
       "      <td>0.2985</td>\n",
       "      <td>0.0832</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.339</td>\n",
       "      <td>0.4325</td>\n",
       "      <td>0.3549</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>0.3666</td>\n",
       "      <td>0.4666</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.3992</td>\n",
       "      <td>0.3365</td>\n",
       "      <td>0.3613</td>\n",
       "      <td>0.4268</td>\n",
       "      <td>0.2888</td>\n",
       "      <td>0.1248</td>\n",
       "      <td>0.3998</td>\n",
       "      <td>0.3444</td>\n",
       "      <td>0.4579</td>\n",
       "      <td>0.4113</td>\n",
       "      <td>0.4038</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>0.3279</td>\n",
       "      <td>0.3109</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/7</td>\n",
       "      <td></td>\n",
       "      <td>I27L</td>\n",
       "      <td>Atc/Ctc</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,21094359,21647738,21931794,21937998,2...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69783</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Maturity onset diabetes mellitus in young,Nonp...</td>\n",
       "      <td>risk factor,benign</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>14937</td>\n",
       "      <td>ENST00000402929.5</td>\n",
       "      <td>n.214A&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>45%</td>\n",
       "      <td>3.1919E-7</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2985</td>\n",
       "      <td>0.0832</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.339</td>\n",
       "      <td>0.4325</td>\n",
       "      <td>0.3549</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>0.3666</td>\n",
       "      <td>0.4666</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.3992</td>\n",
       "      <td>0.3365</td>\n",
       "      <td>0.3613</td>\n",
       "      <td>0.4268</td>\n",
       "      <td>0.2888</td>\n",
       "      <td>0.1248</td>\n",
       "      <td>0.3998</td>\n",
       "      <td>0.3444</td>\n",
       "      <td>0.4579</td>\n",
       "      <td>0.4113</td>\n",
       "      <td>0.4038</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>0.3279</td>\n",
       "      <td>0.3109</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,21094359,21647738,21931794,21937998,2...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69784</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Maturity onset diabetes mellitus in young,Nonp...</td>\n",
       "      <td>risk factor,benign</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>14937</td>\n",
       "      <td>ENST00000433033.3</td>\n",
       "      <td>n.134+767T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>45%</td>\n",
       "      <td>3.1919E-7</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2985</td>\n",
       "      <td>0.0832</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.339</td>\n",
       "      <td>0.4325</td>\n",
       "      <td>0.3549</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>0.3666</td>\n",
       "      <td>0.4666</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.3992</td>\n",
       "      <td>0.3365</td>\n",
       "      <td>0.3613</td>\n",
       "      <td>0.4268</td>\n",
       "      <td>0.2888</td>\n",
       "      <td>0.1248</td>\n",
       "      <td>0.3998</td>\n",
       "      <td>0.3444</td>\n",
       "      <td>0.4579</td>\n",
       "      <td>0.4113</td>\n",
       "      <td>0.4038</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>0.3279</td>\n",
       "      <td>0.3109</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/3</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21094359,21647738,21931794,21937998,2...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69785</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs1169288</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120978847</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Maturity onset diabetes mellitus in young,Nonp...</td>\n",
       "      <td>risk factor,benign</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>14937</td>\n",
       "      <td>ENST00000535301.2</td>\n",
       "      <td>n.322+1797T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>40</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>45%</td>\n",
       "      <td>3.1919E-7</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2985</td>\n",
       "      <td>0.0832</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.3889</td>\n",
       "      <td>0.339</td>\n",
       "      <td>0.4325</td>\n",
       "      <td>0.3549</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>0.3666</td>\n",
       "      <td>0.4666</td>\n",
       "      <td>0.4147</td>\n",
       "      <td>0.3992</td>\n",
       "      <td>0.3365</td>\n",
       "      <td>0.3613</td>\n",
       "      <td>0.4268</td>\n",
       "      <td>0.2888</td>\n",
       "      <td>0.1248</td>\n",
       "      <td>0.3998</td>\n",
       "      <td>0.3444</td>\n",
       "      <td>0.4579</td>\n",
       "      <td>0.4113</td>\n",
       "      <td>0.4038</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>0.3279</td>\n",
       "      <td>0.3109</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>0.5222</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21094359,21647738,21931794,21937998,2...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69860</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>3 prime UTR variant,NMD transcript variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Nonpapillary renal cell carcinoma,Maturity ons...</td>\n",
       "      <td>benign,uncertain significance</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>129231</td>\n",
       "      <td>ENST00000541924.5</td>\n",
       "      <td>c.*522G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>206</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.2598E-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3596</td>\n",
       "      <td>0.1263</td>\n",
       "      <td>0.3963</td>\n",
       "      <td>0.4802</td>\n",
       "      <td>0.3777</td>\n",
       "      <td>0.5061</td>\n",
       "      <td>0.3898</td>\n",
       "      <td>0.1331</td>\n",
       "      <td>0.3995</td>\n",
       "      <td>0.5007</td>\n",
       "      <td>0.4853</td>\n",
       "      <td>0.3716</td>\n",
       "      <td>0.3637</td>\n",
       "      <td>0.3934</td>\n",
       "      <td>0.5206</td>\n",
       "      <td>0.3176</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.3786</td>\n",
       "      <td>0.3923</td>\n",
       "      <td>0.5026</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.3844</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>0.3557</td>\n",
       "      <td>0.3654</td>\n",
       "      <td>0.5137</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24763700,24728327,18599530,18439552,1...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69861</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Nonpapillary renal cell carcinoma,Maturity ons...</td>\n",
       "      <td>benign,uncertain significance</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>129231</td>\n",
       "      <td>ENST00000543255.1</td>\n",
       "      <td>n.552G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>206</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.2598E-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3596</td>\n",
       "      <td>0.1263</td>\n",
       "      <td>0.3963</td>\n",
       "      <td>0.4802</td>\n",
       "      <td>0.3777</td>\n",
       "      <td>0.5061</td>\n",
       "      <td>0.3898</td>\n",
       "      <td>0.1331</td>\n",
       "      <td>0.3995</td>\n",
       "      <td>0.5007</td>\n",
       "      <td>0.4853</td>\n",
       "      <td>0.3716</td>\n",
       "      <td>0.3637</td>\n",
       "      <td>0.3934</td>\n",
       "      <td>0.5206</td>\n",
       "      <td>0.3176</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.3786</td>\n",
       "      <td>0.3923</td>\n",
       "      <td>0.5026</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.3844</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>0.3557</td>\n",
       "      <td>0.3654</td>\n",
       "      <td>0.5137</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24763700,24728327,18599530,18439552,1...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69862</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>splice region variant,intron variant</td>\n",
       "      <td>4/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Nonpapillary renal cell carcinoma,Maturity ons...</td>\n",
       "      <td>benign,uncertain significance</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>129231</td>\n",
       "      <td>ENST00000544413.2</td>\n",
       "      <td>c.1501+7G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>206</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.2598E-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3596</td>\n",
       "      <td>0.1263</td>\n",
       "      <td>0.3963</td>\n",
       "      <td>0.4802</td>\n",
       "      <td>0.3777</td>\n",
       "      <td>0.5061</td>\n",
       "      <td>0.3898</td>\n",
       "      <td>0.1331</td>\n",
       "      <td>0.3995</td>\n",
       "      <td>0.5007</td>\n",
       "      <td>0.4853</td>\n",
       "      <td>0.3716</td>\n",
       "      <td>0.3637</td>\n",
       "      <td>0.3934</td>\n",
       "      <td>0.5206</td>\n",
       "      <td>0.3176</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.3786</td>\n",
       "      <td>0.3923</td>\n",
       "      <td>0.5026</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.3844</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>0.3557</td>\n",
       "      <td>0.3654</td>\n",
       "      <td>0.5137</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>7/9</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24763700,24728327,18599530,18439552,1...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69863</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>3 prime UTR variant,NMD transcript variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Nonpapillary renal cell carcinoma,Maturity ons...</td>\n",
       "      <td>benign,uncertain significance</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>129231</td>\n",
       "      <td>ENST00000544574.5</td>\n",
       "      <td>c.*271G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>206</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.2598E-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3596</td>\n",
       "      <td>0.1263</td>\n",
       "      <td>0.3963</td>\n",
       "      <td>0.4802</td>\n",
       "      <td>0.3777</td>\n",
       "      <td>0.5061</td>\n",
       "      <td>0.3898</td>\n",
       "      <td>0.1331</td>\n",
       "      <td>0.3995</td>\n",
       "      <td>0.5007</td>\n",
       "      <td>0.4853</td>\n",
       "      <td>0.3716</td>\n",
       "      <td>0.3637</td>\n",
       "      <td>0.3934</td>\n",
       "      <td>0.5206</td>\n",
       "      <td>0.3176</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.3786</td>\n",
       "      <td>0.3923</td>\n",
       "      <td>0.5026</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.3844</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>0.3557</td>\n",
       "      <td>0.3654</td>\n",
       "      <td>0.5137</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24763700,24728327,18599530,18439552,1...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69864</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1Diabetes, Type2Diabetes, MODY</td>\n",
       "      <td>rs2464195</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120997672</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>splice region variant,intron variant,NMD trans...</td>\n",
       "      <td>4/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Nonpapillary renal cell carcinoma,Maturity ons...</td>\n",
       "      <td>benign,uncertain significance</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>129231</td>\n",
       "      <td>ENST00000560968.5</td>\n",
       "      <td>c.*248+7G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>206</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.2598E-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.3596</td>\n",
       "      <td>0.1263</td>\n",
       "      <td>0.3963</td>\n",
       "      <td>0.4802</td>\n",
       "      <td>0.3777</td>\n",
       "      <td>0.5061</td>\n",
       "      <td>0.3898</td>\n",
       "      <td>0.1331</td>\n",
       "      <td>0.3995</td>\n",
       "      <td>0.5007</td>\n",
       "      <td>0.4853</td>\n",
       "      <td>0.3716</td>\n",
       "      <td>0.3637</td>\n",
       "      <td>0.3934</td>\n",
       "      <td>0.5206</td>\n",
       "      <td>0.3176</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.3786</td>\n",
       "      <td>0.3923</td>\n",
       "      <td>0.5026</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.3844</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>0.3557</td>\n",
       "      <td>0.3654</td>\n",
       "      <td>0.5137</td>\n",
       "      <td>0.5854</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td></td>\n",
       "      <td>7/9</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,24763700,24728327,18599530,18439552,1...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows Ã— 71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name                          Gene Match       rsID  CHROM  \\\n",
       "69781     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs1169288  chr12   \n",
       "69782     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs1169288  chr12   \n",
       "69783     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs1169288  chr12   \n",
       "69784     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs1169288  chr12   \n",
       "69785     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs1169288  chr12   \n",
       "...         ...                                 ...        ...    ...   \n",
       "69860     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs2464195  chr12   \n",
       "69861     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs2464195  chr12   \n",
       "69862     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs2464195  chr12   \n",
       "69863     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs2464195  chr12   \n",
       "69864     HNF1A  Type1Diabetes, Type2Diabetes, MODY  rs2464195  chr12   \n",
       "\n",
       "             POS REF ALT      Zygosity  \\\n",
       "69781  120978847   A   C  Heterozygous   \n",
       "69782  120978847   A   C  Heterozygous   \n",
       "69783  120978847   A   C  Heterozygous   \n",
       "69784  120978847   A   C  Heterozygous   \n",
       "69785  120978847   A   C  Heterozygous   \n",
       "...          ...  ..  ..           ...   \n",
       "69860  120997672   G   A    Homozygous   \n",
       "69861  120997672   G   A    Homozygous   \n",
       "69862  120997672   G   A    Homozygous   \n",
       "69863  120997672   G   A    Homozygous   \n",
       "69864  120997672   G   A    Homozygous   \n",
       "\n",
       "                                             Consequence Consequence_score  \\\n",
       "69781                                   missense variant              7/10   \n",
       "69782                                   missense variant              7/10   \n",
       "69783                 non coding transcript exon variant              2/10   \n",
       "69784       intron variant,non coding transcript variant              2/10   \n",
       "69785       intron variant,non coding transcript variant              2/10   \n",
       "...                                                  ...               ...   \n",
       "69860         3 prime UTR variant,NMD transcript variant              3/10   \n",
       "69861                 non coding transcript exon variant              2/10   \n",
       "69862               splice region variant,intron variant              4/10   \n",
       "69863         3 prime UTR variant,NMD transcript variant              3/10   \n",
       "69864  splice region variant,intron variant,NMD trans...              4/10   \n",
       "\n",
       "         IMPACT  IMPACT_score  \\\n",
       "69781  MODERATE           5.0   \n",
       "69782  MODERATE           5.0   \n",
       "69783  MODIFIER           1.5   \n",
       "69784  MODIFIER           1.5   \n",
       "69785  MODIFIER           1.5   \n",
       "...         ...           ...   \n",
       "69860  MODIFIER           1.5   \n",
       "69861  MODIFIER           1.5   \n",
       "69862       LOW           2.5   \n",
       "69863  MODIFIER           1.5   \n",
       "69864       LOW           2.5   \n",
       "\n",
       "                                           ClinVar_CLNDN  \\\n",
       "69781  Maturity onset diabetes mellitus in young,Nonp...   \n",
       "69782  Maturity onset diabetes mellitus in young,Nonp...   \n",
       "69783  Maturity onset diabetes mellitus in young,Nonp...   \n",
       "69784  Maturity onset diabetes mellitus in young,Nonp...   \n",
       "69785  Maturity onset diabetes mellitus in young,Nonp...   \n",
       "...                                                  ...   \n",
       "69860  Nonpapillary renal cell carcinoma,Maturity ons...   \n",
       "69861  Nonpapillary renal cell carcinoma,Maturity ons...   \n",
       "69862  Nonpapillary renal cell carcinoma,Maturity ons...   \n",
       "69863  Nonpapillary renal cell carcinoma,Maturity ons...   \n",
       "69864  Nonpapillary renal cell carcinoma,Maturity ons...   \n",
       "\n",
       "                            CLIN_SIG  \\\n",
       "69781             risk factor,benign   \n",
       "69782             risk factor,benign   \n",
       "69783             risk factor,benign   \n",
       "69784             risk factor,benign   \n",
       "69785             risk factor,benign   \n",
       "...                              ...   \n",
       "69860  benign,uncertain significance   \n",
       "69861  benign,uncertain significance   \n",
       "69862  benign,uncertain significance   \n",
       "69863  benign,uncertain significance   \n",
       "69864  benign,uncertain significance   \n",
       "\n",
       "                                   ClinVar_CLNREVSTAT ClinVar  \\\n",
       "69781  criteria provided, conflicting interpretations   14937   \n",
       "69782  criteria provided, conflicting interpretations   14937   \n",
       "69783  criteria provided, conflicting interpretations   14937   \n",
       "69784  criteria provided, conflicting interpretations   14937   \n",
       "69785  criteria provided, conflicting interpretations   14937   \n",
       "...                                               ...     ...   \n",
       "69860  criteria provided, conflicting interpretations  129231   \n",
       "69861  criteria provided, conflicting interpretations  129231   \n",
       "69862  criteria provided, conflicting interpretations  129231   \n",
       "69863  criteria provided, conflicting interpretations  129231   \n",
       "69864  criteria provided, conflicting interpretations  129231   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "69781  ENST00000257555.11            c.79A>C  ENSP00000257555.5   \n",
       "69782   ENST00000400024.6            c.79A>C  ENSP00000476181.1   \n",
       "69783   ENST00000402929.5           n.214A>C                      \n",
       "69784   ENST00000433033.3       n.134+767T>G                      \n",
       "69785   ENST00000535301.2      n.322+1797T>G                      \n",
       "...                   ...                ...                ...   \n",
       "69860   ENST00000541924.5          c.*522G>A                      \n",
       "69861   ENST00000543255.1           n.552G>A                      \n",
       "69862   ENST00000544413.2        c.1501+7G>A                      \n",
       "69863   ENST00000544574.5          c.*271G>A                      \n",
       "69864   ENST00000560968.5        c.*248+7G>A                      \n",
       "\n",
       "      HGVSp (Transcript)   GT   GQ SDP  DP  RD  AD  FREQ        PVAL RDF RDR  \\\n",
       "69781         p.Ile27Leu  0/1   64  40  40  22  18   45%   3.1919E-7  18   4   \n",
       "69782         p.Ile27Leu  0/1   64  40  40  22  18   45%   3.1919E-7  18   4   \n",
       "69783               None  0/1   64  40  40  22  18   45%   3.1919E-7  18   4   \n",
       "69784               None  0/1   64  40  40  22  18   45%   3.1919E-7  18   4   \n",
       "69785               None  0/1   64  40  40  22  18   45%   3.1919E-7  18   4   \n",
       "...                  ...  ...  ...  ..  ..  ..  ..   ...         ...  ..  ..   \n",
       "69860               None  1/1  206  36  36   0  36  100%  2.2598E-21   0   0   \n",
       "69861               None  1/1  206  36  36   0  36  100%  2.2598E-21   0   0   \n",
       "69862               None  1/1  206  36  36   0  36  100%  2.2598E-21   0   0   \n",
       "69863               None  1/1  206  36  36   0  36  100%  2.2598E-21   0   0   \n",
       "69864               None  1/1  206  36  36   0  36  100%  2.2598E-21   0   0   \n",
       "\n",
       "      ADF ADR             SIFT                  PolyPhen      AF  AFR_AF  \\\n",
       "69781  11   7  tolerated(0.11)             benign(0.282)  0.2985  0.0832   \n",
       "69782  11   7  tolerated(0.11)  probably damaging(0.938)  0.2985  0.0832   \n",
       "69783  11   7                                             0.2985  0.0832   \n",
       "69784  11   7                                             0.2985  0.0832   \n",
       "69785  11   7                                             0.2985  0.0832   \n",
       "...    ..  ..              ...                       ...     ...     ...   \n",
       "69860  30   6                                             0.3596  0.1263   \n",
       "69861  30   6                                             0.3596  0.1263   \n",
       "69862  30   6                                             0.3596  0.1263   \n",
       "69863  30   6                                             0.3596  0.1263   \n",
       "69864  30   6                                             0.3596  0.1263   \n",
       "\n",
       "       AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF  \\\n",
       "69781    0.33  0.3889   0.339  0.4325     0.3549         0.1179   \n",
       "69782    0.33  0.3889   0.339  0.4325     0.3549         0.1179   \n",
       "69783    0.33  0.3889   0.339  0.4325     0.3549         0.1179   \n",
       "69784    0.33  0.3889   0.339  0.4325     0.3549         0.1179   \n",
       "69785    0.33  0.3889   0.339  0.4325     0.3549         0.1179   \n",
       "...       ...     ...     ...     ...        ...            ...   \n",
       "69860  0.3963  0.4802  0.3777  0.5061     0.3898         0.1331   \n",
       "69861  0.3963  0.4802  0.3777  0.5061     0.3898         0.1331   \n",
       "69862  0.3963  0.4802  0.3777  0.5061     0.3898         0.1331   \n",
       "69863  0.3963  0.4802  0.3777  0.5061     0.3898         0.1331   \n",
       "69864  0.3963  0.4802  0.3777  0.5061     0.3898         0.1331   \n",
       "\n",
       "      gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF  \\\n",
       "69781         0.3666         0.4666         0.4147         0.3992   \n",
       "69782         0.3666         0.4666         0.4147         0.3992   \n",
       "69783         0.3666         0.4666         0.4147         0.3992   \n",
       "69784         0.3666         0.4666         0.4147         0.3992   \n",
       "69785         0.3666         0.4666         0.4147         0.3992   \n",
       "...              ...            ...            ...            ...   \n",
       "69860         0.3995         0.5007         0.4853         0.3716   \n",
       "69861         0.3995         0.5007         0.4853         0.3716   \n",
       "69862         0.3995         0.5007         0.4853         0.3716   \n",
       "69863         0.3995         0.5007         0.4853         0.3716   \n",
       "69864         0.3995         0.5007         0.4853         0.3716   \n",
       "\n",
       "      gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF  \\\n",
       "69781         0.3365         0.3613         0.4268     0.2888         0.1248   \n",
       "69782         0.3365         0.3613         0.4268     0.2888         0.1248   \n",
       "69783         0.3365         0.3613         0.4268     0.2888         0.1248   \n",
       "69784         0.3365         0.3613         0.4268     0.2888         0.1248   \n",
       "69785         0.3365         0.3613         0.4268     0.2888         0.1248   \n",
       "...              ...            ...            ...        ...            ...   \n",
       "69860         0.3637         0.3934         0.5206     0.3176         0.1446   \n",
       "69861         0.3637         0.3934         0.5206     0.3176         0.1446   \n",
       "69862         0.3637         0.3934         0.5206     0.3176         0.1446   \n",
       "69863         0.3637         0.3934         0.5206     0.3176         0.1446   \n",
       "69864         0.3637         0.3934         0.5206     0.3176         0.1446   \n",
       "\n",
       "      gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF  \\\n",
       "69781         0.3998         0.3444         0.4579         0.4113   \n",
       "69782         0.3998         0.3444         0.4579         0.4113   \n",
       "69783         0.3998         0.3444         0.4579         0.4113   \n",
       "69784         0.3998         0.3444         0.4579         0.4113   \n",
       "69785         0.3998         0.3444         0.4579         0.4113   \n",
       "...              ...            ...            ...            ...   \n",
       "69860         0.3786         0.3923         0.5026         0.4909   \n",
       "69861         0.3786         0.3923         0.5026         0.4909   \n",
       "69862         0.3786         0.3923         0.5026         0.4909   \n",
       "69863         0.3786         0.3923         0.5026         0.4909   \n",
       "69864         0.3786         0.3923         0.5026         0.4909   \n",
       "\n",
       "      gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF  \\\n",
       "69781         0.4038         0.5222         0.3279         0.3109   \n",
       "69782         0.4038         0.5222         0.3279         0.3109   \n",
       "69783         0.4038         0.5222         0.3279         0.3109   \n",
       "69784         0.4038         0.5222         0.3279         0.3109   \n",
       "69785         0.4038         0.5222         0.3279         0.3109   \n",
       "...              ...            ...            ...            ...   \n",
       "69860         0.3844         0.5854         0.3557         0.3654   \n",
       "69861         0.3844         0.5854         0.3557         0.3654   \n",
       "69862         0.3844         0.5854         0.3557         0.3654   \n",
       "69863         0.3844         0.5854         0.3557         0.3654   \n",
       "69864         0.3844         0.5854         0.3557         0.3654   \n",
       "\n",
       "      gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS                  BIOTYPE  EXON  \\\n",
       "69781         0.4168  0.5222  gnomADg MID           protein coding  1/10   \n",
       "69782         0.4168  0.5222  gnomADg MID           protein coding   1/7   \n",
       "69783         0.4168  0.5222  gnomADg MID          retained intron   1/6   \n",
       "69784         0.4168  0.5222  gnomADg MID                   lncRNA         \n",
       "69785         0.4168  0.5222  gnomADg MID                   lncRNA         \n",
       "...              ...     ...          ...                      ...   ...   \n",
       "69860         0.5137  0.5854  gnomADg MID  nonsense mediated decay   6/6   \n",
       "69861         0.5137  0.5854  gnomADg MID          retained intron   2/2   \n",
       "69862         0.5137  0.5854  gnomADg MID           protein coding         \n",
       "69863         0.5137  0.5854  gnomADg MID  nonsense mediated decay   3/3   \n",
       "69864         0.5137  0.5854  gnomADg MID  nonsense mediated decay         \n",
       "\n",
       "      INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "69781                                   I27L  Atc/Ctc      1   \n",
       "69782                                   I27L  Atc/Ctc      1   \n",
       "69783                                    nan               1   \n",
       "69784    1/3                             nan              -1   \n",
       "69785    1/1                             nan              -1   \n",
       "...      ...                             ...      ...    ...   \n",
       "69860                                    nan               1   \n",
       "69861                                    nan               1   \n",
       "69862    7/9                             nan               1   \n",
       "69863                                    nan               1   \n",
       "69864    7/9                             nan               1   \n",
       "\n",
       "                                                  PUBMED Literature  \\\n",
       "69781  25741868,21094359,21647738,21931794,21937998,2...        Yes   \n",
       "69782  25741868,21094359,21647738,21931794,21937998,2...        Yes   \n",
       "69783  25741868,21094359,21647738,21931794,21937998,2...        Yes   \n",
       "69784  25741868,21094359,21647738,21931794,21937998,2...        Yes   \n",
       "69785  25741868,21094359,21647738,21931794,21937998,2...        Yes   \n",
       "...                                                  ...        ...   \n",
       "69860  25741868,24763700,24728327,18599530,18439552,1...        Yes   \n",
       "69861  25741868,24763700,24728327,18599530,18439552,1...        Yes   \n",
       "69862  25741868,24763700,24728327,18599530,18439552,1...        Yes   \n",
       "69863  25741868,24763700,24728327,18599530,18439552,1...        Yes   \n",
       "69864  25741868,24763700,24728327,18599530,18439552,1...        Yes   \n",
       "\n",
       "      Covered/Not_Covered  \n",
       "69781             Covered  \n",
       "69782             Covered  \n",
       "69783             Covered  \n",
       "69784             Covered  \n",
       "69785             Covered  \n",
       "...                   ...  \n",
       "69860             Covered  \n",
       "69861             Covered  \n",
       "69862             Covered  \n",
       "69863             Covered  \n",
       "69864             Covered  \n",
       "\n",
       "[72 rows x 71 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3[merged_3['Literature'] == 'Yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "c4179a6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>22.58%</td>\n",
       "      <td>5.3469E-3</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.36T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Asn12%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>N12</td>\n",
       "      <td>aaT/aaC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1260343719</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69063</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>67</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>42.22%</td>\n",
       "      <td>1.6979E-7</td>\n",
       "      <td>23</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.05175</td>\n",
       "      <td>0.04058</td>\n",
       "      <td>0.02941</td>\n",
       "      <td>0.04766</td>\n",
       "      <td>0.04545</td>\n",
       "      <td>0.06778</td>\n",
       "      <td>0.0303</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>0.06103</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1089</td>\n",
       "      <td>0.1538</td>\n",
       "      <td>gnomADg MID</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306164</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155459837</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>71</td>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "      <td>81</td>\n",
       "      <td>22</td>\n",
       "      <td>21.36%</td>\n",
       "      <td>6.7976E-8</td>\n",
       "      <td>54</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306165</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000244174.11</td>\n",
       "      <td>c.580-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>5/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306166</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000369423.7</td>\n",
       "      <td>c.685-26C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/8</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306167</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306168</th>\n",
       "      <td>IL9R</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2073467</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156005252</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.7304E-4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.02802</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>0.0527</td>\n",
       "      <td>9.934e-05</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0005125</td>\n",
       "      <td>0.01736</td>\n",
       "      <td>0.04299</td>\n",
       "      <td>0.01208</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02902</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003382</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>0.04662</td>\n",
       "      <td>0.2012</td>\n",
       "      <td>gnomADe EAS,gnomADg EAS</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306169 rows Ã— 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene Match          rsID CHROM        POS Literature REF ALT  \\\n",
       "0                        No  rs1416222198  chr1      65591         No   C   T   \n",
       "1                        No  rs1416222198  chr1      65591         No   C   T   \n",
       "2                        No  rs1416222198  chr1      65591         No   C   T   \n",
       "3          OR4F5         No  rs1260343719  chr1      69063         No   T   C   \n",
       "4          OR4F5         No  rs1260343719  chr1      69063         No   T   C   \n",
       "...          ...        ...           ...   ...        ...        ...  ..  ..   \n",
       "306164                   No             .  chrX  155459837         No   A   G   \n",
       "306165      IL9R         No     rs2073467  chrX  156005252         No   C   T   \n",
       "306166      IL9R         No     rs2073467  chrX  156005252         No   C   T   \n",
       "306167      IL9R         No     rs2073467  chrX  156005252         No   C   T   \n",
       "306168      IL9R         No     rs2073467  chrX  156005252         No   C   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous           intron variant              2/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3       Heterozygous       synonymous variant              3/10       LOW   \n",
       "4       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "306164  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "306165  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306166  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "306167  Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "306168  Heterozygous    upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                2.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "306164           1.5                                                     \n",
       "306165           1.5                                                     \n",
       "306166           1.5                                                     \n",
       "306167           1.5                                                     \n",
       "306168           1.5                                                     \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                                        None                      \n",
       "1        ENST00000641515.2          c.9+18C>T                      \n",
       "2                                        None                      \n",
       "3        ENST00000641515.2            c.36T>C  ENSP00000493376.2   \n",
       "4                                        None                      \n",
       "...                    ...                ...                ...   \n",
       "306164                                   None                      \n",
       "306165  ENST00000244174.11        c.580-26C>T                      \n",
       "306166   ENST00000369423.7        c.685-26C>T                      \n",
       "306167                                   None                      \n",
       "306168                                   None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT  GQ  SDP   DP  RD  AD    FREQ       PVAL RDF  \\\n",
       "0                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "1                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "2                    None  0/1  22   31   31  24   7  22.58%  5.3469E-3  17   \n",
       "3              p.Asn12%3D  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "4                    None  0/1  67   45   45  26  19  42.22%  1.6979E-7  23   \n",
       "...                   ...  ...  ..  ...  ...  ..  ..     ...        ...  ..   \n",
       "306164               None  0/1  71  103  103  81  22  21.36%  6.7976E-8  54   \n",
       "306165               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306166               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306167               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "306168               None  0/1  31   12   12   4   8  66.67%  6.7304E-4   4   \n",
       "\n",
       "       RDR ADF ADR SIFT PolyPhen AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF  \\\n",
       "0        7   3   4                                                       \n",
       "1        7   3   4                                                       \n",
       "2        7   3   4                                                       \n",
       "3        3  18   1                                                       \n",
       "4        3  18   1                                                       \n",
       "...     ..  ..  ..  ...      ... ..    ...    ...    ...    ...    ...   \n",
       "306164  27  14   8                                                       \n",
       "306165   0   8   0                                                       \n",
       "306166   0   8   0                                                       \n",
       "306167   0   8   0                                                       \n",
       "306168   0   8   0                                                       \n",
       "\n",
       "       gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                               \n",
       "1                                                                               \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...           ...            ...            ...            ...            ...   \n",
       "306164                                                                          \n",
       "306165    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306166    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306167    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "306168    0.02802       0.001789         0.0527      9.934e-05         0.2012   \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.05175   \n",
       "4                                                                     0.05175   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "306164                                                                          \n",
       "306165              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306166              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306167              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "306168              0      0.0005125        0.01736        0.04299    0.01208   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.04058        0.02941        0.04766        0.04545   \n",
       "4             0.04058        0.02941        0.04766        0.04545   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165       0.001739              0        0.02902       0.000288   \n",
       "306166       0.001739              0        0.02902       0.000288   \n",
       "306167       0.001739              0        0.02902       0.000288   \n",
       "306168       0.001739              0        0.02902       0.000288   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3             0.06778         0.0303         0.1538        0.06103   \n",
       "4             0.06778         0.0303         0.1538        0.06103   \n",
       "...               ...            ...            ...            ...   \n",
       "306164                                                               \n",
       "306165         0.2012              0              0      0.0003382   \n",
       "306166         0.2012              0              0      0.0003382   \n",
       "306167         0.2012              0              0      0.0003382   \n",
       "306168         0.2012              0              0      0.0003382   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF              MAX_AF_POPS  \\\n",
       "0             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "1             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "2             0.05357        0.02083  0.1667              gnomADg AMI   \n",
       "3             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "4             0.05789         0.1089  0.1538              gnomADg MID   \n",
       "...               ...            ...     ...                      ...   \n",
       "306164                                                                  \n",
       "306165         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306166         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306167         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "306168         0.0134        0.04662  0.2012  gnomADe EAS,gnomADg EAS   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3                           protein coding  3/3          \n",
       "4                                   lncRNA               \n",
       "...                                    ...  ...    ...   \n",
       "306164                      protein coding               \n",
       "306165                      protein coding         5/8   \n",
       "306166                      protein coding         6/8   \n",
       "306167      protein coding CDS not defined               \n",
       "306168      protein coding CDS not defined               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  nan               1         \n",
       "1                                  nan               1         \n",
       "2                                  nan               1         \n",
       "3                                  N12  aaT/aaC      1         \n",
       "4                                  nan               1         \n",
       "...                                ...      ...    ...    ...  \n",
       "306164                             nan              -1         \n",
       "306165                             nan               1         \n",
       "306166                             nan               1         \n",
       "306167                             nan               1         \n",
       "306168                             nan               1         \n",
       "\n",
       "[306169 rows x 70 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = merged_3[['Gene Name', 'Gene Match', 'rsID', 'CHROM', 'POS', 'Literature', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "84691a43",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_2.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/Fatty_liver_data.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77986f42",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "760c9e34",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc04bc87",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d185d7f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "122ec63f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "78fb3080",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>251</td>\n",
       "      <td>251</td>\n",
       "      <td>0</td>\n",
       "      <td>250</td>\n",
       "      <td>99.6%</td>\n",
       "      <td>8.5657E-150</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>194</td>\n",
       "      <td>56</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>171</td>\n",
       "      <td>179</td>\n",
       "      <td>179</td>\n",
       "      <td>128</td>\n",
       "      <td>51</td>\n",
       "      <td>28.49%</td>\n",
       "      <td>6.8954E-18</td>\n",
       "      <td>86</td>\n",
       "      <td>42</td>\n",
       "      <td>35</td>\n",
       "      <td>16</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>rs2808347</td>\n",
       "      <td>chr1</td>\n",
       "      <td>450718</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>34.62%</td>\n",
       "      <td>8.4928E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>rs2808347</td>\n",
       "      <td>chr1</td>\n",
       "      <td>450718</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+34322A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>34.62%</td>\n",
       "      <td>8.4928E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>36%</td>\n",
       "      <td>8.1542E-4</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336023</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336024</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336025</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336026</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336027</th>\n",
       "      <td></td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000437359.1</td>\n",
       "      <td>n.48T&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>135</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.101E-14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0681</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.2731</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2731</td>\n",
       "      <td>SAS</td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>34424929,35176104</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>336028 rows Ã— 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1     69897   T   C  Heterozygous   \n",
       "2         OR4F29    rs2808347  chr1    450718   T   C  Heterozygous   \n",
       "3         OR4F29    rs2808347  chr1    450718   T   C  Heterozygous   \n",
       "4                           .  chr1    685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...       ...  ..  ..           ...   \n",
       "336023    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336024    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336025    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336026    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336027             rs17250535  chrY  21311315   T   A    Homozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4                            downstream gene variant              2/10   \n",
       "...                                              ...               ...   \n",
       "336023                         upstream gene variant              2/10   \n",
       "336024                         upstream gene variant              2/10   \n",
       "336025                         upstream gene variant              2/10   \n",
       "336026                         upstream gene variant              2/10   \n",
       "336027            non coding transcript exon variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0       MODERATE           5.0                                             \n",
       "1            LOW           2.5                                             \n",
       "2       MODIFIER           1.5                                             \n",
       "3       MODIFIER           1.5                                             \n",
       "4       MODIFIER           1.5                                             \n",
       "...          ...           ...           ...      ...                ...   \n",
       "336023  MODIFIER           1.5                                             \n",
       "336024  MODIFIER           1.5                                             \n",
       "336025  MODIFIER           1.5                                             \n",
       "336026  MODIFIER           1.5                                             \n",
       "336027  MODIFIER           1.5                                             \n",
       "\n",
       "       ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0               ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1               ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                               None                      \n",
       "3               ENST00000455207.5     n.169+34322A>G                      \n",
       "4                                               None                      \n",
       "...        ...                ...                ...                ...   \n",
       "336023                                          None                      \n",
       "336024                                          None                      \n",
       "336025                                          None                      \n",
       "336026                                          None                      \n",
       "336027          ENST00000437359.1            n.48T>A                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ         PVAL  \\\n",
       "0             p.Thr162Ala  1/1  255  251  251    0  250   99.6%  8.5657E-150   \n",
       "1             p.Ser290%3D  0/1  171  179  179  128   51  28.49%   6.8954E-18   \n",
       "2                    None  0/1   30   26   26   17    9  34.62%    8.4928E-4   \n",
       "3                    None  0/1   30   26   26   17    9  34.62%    8.4928E-4   \n",
       "4                    None  0/1   30   25   25   16    9     36%    8.1542E-4   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...          ...   \n",
       "336023               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336024               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336025               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336026               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336027               None  1/1  135   24   24    0   24    100%    3.101E-14   \n",
       "\n",
       "       RDF RDR  ADF ADR             SIFT   PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0        0   0  194  56  tolerated(0.92)  benign(0)                           \n",
       "1       86  42   35  16                              0.6881   0.407  0.6254   \n",
       "2       17   0    9   0                                                       \n",
       "3       17   0    9   0                                                       \n",
       "4       16   0    9   0                                                       \n",
       "...     ..  ..  ...  ..              ...        ...     ...     ...     ...   \n",
       "336023   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336024   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336025   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336026   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336027   0   0   19   5                              0.0681       0  0.0059   \n",
       "\n",
       "        EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                                  0.9497         0.6075         0.9514   \n",
       "1        0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "2                                                                         \n",
       "3                                                                         \n",
       "4                                                                         \n",
       "...        ...     ...     ...        ...            ...            ...   \n",
       "336023  0.9959  0.4667  0.9962                                            \n",
       "336024  0.9959  0.4667  0.9962                                            \n",
       "336025  0.9959  0.4667  0.9962                                            \n",
       "336026  0.9959  0.4667  0.9962                                            \n",
       "336027       0    0.05  0.2731                                            \n",
       "\n",
       "       gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0              0.9767         0.9994         0.9916         0.9726   \n",
       "1              0.8131         0.7793         0.8652         0.8438   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "336023                                                               \n",
       "336024                                                               \n",
       "336025                                                               \n",
       "336026                                                               \n",
       "336027                                                               \n",
       "\n",
       "       gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0              0.9506         0.9854      0.846         0.5948          0.998   \n",
       "1              0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...            ...        ...            ...            ...   \n",
       "336023                                   0.6593         0.8119         0.2864   \n",
       "336024                                   0.6593         0.8119         0.2864   \n",
       "336025                                   0.6593         0.8119         0.2864   \n",
       "336026                                   0.6593         0.8119         0.2864   \n",
       "336027                                                                          \n",
       "\n",
       "       gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0              0.8951         0.9784         0.9998         0.9907   \n",
       "1              0.4782         0.6714         0.6047         0.6987   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "336023         0.5121         0.9004         0.9992         0.9714   \n",
       "336024         0.5121         0.9004         0.9992         0.9714   \n",
       "336025         0.5121         0.9004         0.9992         0.9714   \n",
       "336026         0.5121         0.9004         0.9992         0.9714   \n",
       "336027                                                               \n",
       "\n",
       "       gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  \\\n",
       "0                 0.9         0.9674         0.8624         0.9772  0.9998   \n",
       "1              0.6441         0.6751         0.4659          0.622   0.876   \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...               ...            ...            ...            ...     ...   \n",
       "336023         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336024         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336025         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336026         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336027                                                              0.2731   \n",
       "\n",
       "        MAX_AF_POPS                 BIOTYPE EXON INTRON  \\\n",
       "0       gnomADg EAS          protein coding  3/3          \n",
       "1               EAS          protein coding  3/3          \n",
       "2                            protein coding               \n",
       "3                                    lncRNA         1/2   \n",
       "4                            protein coding               \n",
       "...             ...                     ...  ...    ...   \n",
       "336023  gnomADg EAS                  lncRNA               \n",
       "336024  gnomADg EAS                  lncRNA               \n",
       "336025  gnomADg EAS                  lncRNA               \n",
       "336026  gnomADg EAS                  lncRNA               \n",
       "336027          SAS  unprocessed pseudogene  1/9          \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                T162A  Aca/Gca      1   \n",
       "1                                 S290  tcT/tcC      1   \n",
       "2                                  nan              -1   \n",
       "3                                  nan              -1   \n",
       "4                                  nan              -1   \n",
       "...                                ...      ...    ...   \n",
       "336023                             nan              -1   \n",
       "336024                             nan              -1   \n",
       "336025                             nan              -1   \n",
       "336026                             nan              -1   \n",
       "336027                             nan               1   \n",
       "\n",
       "                                                   PUBMED Matched_Gene  \n",
       "0                                                                       \n",
       "1                                                                       \n",
       "2                                                                       \n",
       "3                                                                       \n",
       "4                                                                       \n",
       "...                                                   ...          ...  \n",
       "336023  22271044,29391530,24262073,15896936,32996047,3...               \n",
       "336024  22271044,29391530,24262073,15896936,32996047,3...               \n",
       "336025  22271044,29391530,24262073,15896936,32996047,3...               \n",
       "336026  22271044,29391530,24262073,15896936,32996047,3...               \n",
       "336027                                  34424929,35176104               \n",
       "\n",
       "[336028 rows x 69 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene Match'] = 'No'\n",
    "merged_2['Matched_Gene'] = ''\n",
    "    \n",
    "# Iterate through each gene in vcf['Gene']\n",
    "for index, genes in merged_2['Gene Name'].iteritems():\n",
    "    if isinstance(genes, str):\n",
    "        gene_list = genes.split(',')\n",
    "        for gene in gene_list:\n",
    "            if gene in df_gene['Gene Name'].values:\n",
    "                merged_2.at[index, 'Gene Match'] = 'Yes'\n",
    "                merged_2.at[index, 'Matched_Gene'] = gene\n",
    "                break\n",
    "    \n",
    "df_gene = df_gene.rename({'Gene Name':'Matched_Gene', 'Gene_Match':'Gene Match'}, axis=1)\n",
    "\n",
    "merged_2 = merged_2.drop(columns=['Gene Match'], axis=1)\n",
    "\n",
    "merged_2 = pd.merge(merged_2, df_gene, on= 'Matched_Gene', how = 'left', sort = False)\n",
    "#merged_2['Gene Match'] = merged_2['Gene Match'].fillna('No')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "dfe64bee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           333579\n",
       "PMS2          446\n",
       "BRCA1         421\n",
       "NBN           302\n",
       "TP53          211\n",
       "SMARCA4       180\n",
       "NF1           162\n",
       "BARD1         133\n",
       "ATM           116\n",
       "MUTYH          96\n",
       "BRCA2          74\n",
       "BRIP1          72\n",
       "RECQL          38\n",
       "PTEN           30\n",
       "RAD50          26\n",
       "RAD51D         24\n",
       "PALB2          22\n",
       "MSH2           21\n",
       "MSH6           18\n",
       "CDKN2A         17\n",
       "SLX4           14\n",
       "STK11          10\n",
       "CDH1           10\n",
       "MRE11           6\n",
       "Name: Matched_Gene, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.Matched_Gene.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "68bfefd8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A1742P</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A2062V</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>A2067D</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1650</th>\n",
       "      <td>TP53</td>\n",
       "      <td>A347T</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1651</th>\n",
       "      <td>TP53</td>\n",
       "      <td>L348F</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1652</th>\n",
       "      <td>TP53</td>\n",
       "      <td>L348S</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1653</th>\n",
       "      <td>TP53</td>\n",
       "      <td>G356A</td>\n",
       "      <td>Inconclusive</td>\n",
       "      <td>Inconclusive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1654</th>\n",
       "      <td>TP53</td>\n",
       "      <td>G360V</td>\n",
       "      <td>Likely Neutral</td>\n",
       "      <td>Inconclusive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1655 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Matched_Gene Protein Position and Amino Acid         Oncogenic  \\\n",
       "0             ATM                        Deletion  Likely Oncogenic   \n",
       "1             ATM            Truncating Mutations  Likely Oncogenic   \n",
       "2             ATM                          A1742P  Likely Oncogenic   \n",
       "3             ATM                          A2062V  Likely Oncogenic   \n",
       "4             ATM                          A2067D  Likely Oncogenic   \n",
       "...           ...                             ...               ...   \n",
       "1650         TP53                           A347T  Likely Oncogenic   \n",
       "1651         TP53                           L348F  Likely Oncogenic   \n",
       "1652         TP53                           L348S  Likely Oncogenic   \n",
       "1653         TP53                           G356A      Inconclusive   \n",
       "1654         TP53                           G360V    Likely Neutral   \n",
       "\n",
       "              Mutation effect  \n",
       "0     Likely Loss-of-function  \n",
       "1     Likely Loss-of-function  \n",
       "2     Likely Loss-of-function  \n",
       "3     Likely Loss-of-function  \n",
       "4     Likely Loss-of-function  \n",
       "...                       ...  \n",
       "1650  Likely Loss-of-function  \n",
       "1651  Likely Loss-of-function  \n",
       "1652  Likely Loss-of-function  \n",
       "1653             Inconclusive  \n",
       "1654             Inconclusive  \n",
       "\n",
       "[1655 rows x 4 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Hereditary_Cancer_risk_Oncokb.xlsx')\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "8c50a036",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Matched_Gene</th>\n",
       "      <th>Oncogenic</th>\n",
       "      <th>Mutation effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>251</td>\n",
       "      <td>251</td>\n",
       "      <td>0</td>\n",
       "      <td>250</td>\n",
       "      <td>99.6%</td>\n",
       "      <td>8.5657E-150</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>194</td>\n",
       "      <td>56</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.9497</td>\n",
       "      <td>0.6075</td>\n",
       "      <td>0.9514</td>\n",
       "      <td>0.9767</td>\n",
       "      <td>0.9994</td>\n",
       "      <td>0.9916</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9506</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.8951</td>\n",
       "      <td>0.9784</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.8624</td>\n",
       "      <td>0.9772</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>171</td>\n",
       "      <td>179</td>\n",
       "      <td>179</td>\n",
       "      <td>128</td>\n",
       "      <td>51</td>\n",
       "      <td>28.49%</td>\n",
       "      <td>6.8954E-18</td>\n",
       "      <td>86</td>\n",
       "      <td>42</td>\n",
       "      <td>35</td>\n",
       "      <td>16</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6881</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.6254</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.7942</td>\n",
       "      <td>0.8098</td>\n",
       "      <td>0.7209</td>\n",
       "      <td>0.307</td>\n",
       "      <td>0.6749</td>\n",
       "      <td>0.8131</td>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.8652</td>\n",
       "      <td>0.8438</td>\n",
       "      <td>0.7932</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>0.6511</td>\n",
       "      <td>0.4782</td>\n",
       "      <td>0.6714</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>0.6987</td>\n",
       "      <td>0.6441</td>\n",
       "      <td>0.6751</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>0.622</td>\n",
       "      <td>0.876</td>\n",
       "      <td>EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>rs2808347</td>\n",
       "      <td>chr1</td>\n",
       "      <td>450718</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>34.62%</td>\n",
       "      <td>8.4928E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>rs2808347</td>\n",
       "      <td>chr1</td>\n",
       "      <td>450718</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+34322A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>34.62%</td>\n",
       "      <td>8.4928E-4</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>685694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>30</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>36%</td>\n",
       "      <td>8.1542E-4</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336023</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336024</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336025</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336026</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336027</th>\n",
       "      <td></td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000437359.1</td>\n",
       "      <td>n.48T&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>135</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.101E-14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0681</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.2731</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2731</td>\n",
       "      <td>SAS</td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>34424929,35176104</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>336028 rows Ã— 71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0          OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "1          OR4F5  rs200676709  chr1     69897   T   C  Heterozygous   \n",
       "2         OR4F29    rs2808347  chr1    450718   T   C  Heterozygous   \n",
       "3         OR4F29    rs2808347  chr1    450718   T   C  Heterozygous   \n",
       "4                           .  chr1    685694   T   C  Heterozygous   \n",
       "...          ...          ...   ...       ...  ..  ..           ...   \n",
       "336023    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336024    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336025    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336026    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "336027             rs17250535  chrY  21311315   T   A    Homozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                   missense variant              7/10   \n",
       "1                                 synonymous variant              3/10   \n",
       "2                            downstream gene variant              2/10   \n",
       "3       intron variant,non coding transcript variant              2/10   \n",
       "4                            downstream gene variant              2/10   \n",
       "...                                              ...               ...   \n",
       "336023                         upstream gene variant              2/10   \n",
       "336024                         upstream gene variant              2/10   \n",
       "336025                         upstream gene variant              2/10   \n",
       "336026                         upstream gene variant              2/10   \n",
       "336027            non coding transcript exon variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0       MODERATE           5.0                                             \n",
       "1            LOW           2.5                                             \n",
       "2       MODIFIER           1.5                                             \n",
       "3       MODIFIER           1.5                                             \n",
       "4       MODIFIER           1.5                                             \n",
       "...          ...           ...           ...      ...                ...   \n",
       "336023  MODIFIER           1.5                                             \n",
       "336024  MODIFIER           1.5                                             \n",
       "336025  MODIFIER           1.5                                             \n",
       "336026  MODIFIER           1.5                                             \n",
       "336027  MODIFIER           1.5                                             \n",
       "\n",
       "       ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0               ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "1               ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "2                                               None                      \n",
       "3               ENST00000455207.5     n.169+34322A>G                      \n",
       "4                                               None                      \n",
       "...        ...                ...                ...                ...   \n",
       "336023                                          None                      \n",
       "336024                                          None                      \n",
       "336025                                          None                      \n",
       "336026                                          None                      \n",
       "336027          ENST00000437359.1            n.48T>A                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ         PVAL  \\\n",
       "0             p.Thr162Ala  1/1  255  251  251    0  250   99.6%  8.5657E-150   \n",
       "1             p.Ser290%3D  0/1  171  179  179  128   51  28.49%   6.8954E-18   \n",
       "2                    None  0/1   30   26   26   17    9  34.62%    8.4928E-4   \n",
       "3                    None  0/1   30   26   26   17    9  34.62%    8.4928E-4   \n",
       "4                    None  0/1   30   25   25   16    9     36%    8.1542E-4   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...          ...   \n",
       "336023               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336024               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336025               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336026               None  1/1  188   33   33    0   33    100%   1.3852E-19   \n",
       "336027               None  1/1  135   24   24    0   24    100%    3.101E-14   \n",
       "\n",
       "       RDF RDR  ADF ADR             SIFT   PolyPhen      AF  AFR_AF  AMR_AF  \\\n",
       "0        0   0  194  56  tolerated(0.92)  benign(0)                           \n",
       "1       86  42   35  16                              0.6881   0.407  0.6254   \n",
       "2       17   0    9   0                                                       \n",
       "3       17   0    9   0                                                       \n",
       "4       16   0    9   0                                                       \n",
       "...     ..  ..  ...  ..              ...        ...     ...     ...     ...   \n",
       "336023   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336024   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336025   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336026   0   0   24   9                              0.8273  0.9624  0.5824   \n",
       "336027   0   0   19   5                              0.0681       0  0.0059   \n",
       "\n",
       "        EAS_AF  EUR_AF  SAS_AF gnomADe_AF gnomADe_AFR_AF gnomADe_AMR_AF  \\\n",
       "0                                  0.9497         0.6075         0.9514   \n",
       "1        0.876  0.7942  0.8098     0.7209          0.307         0.6749   \n",
       "2                                                                         \n",
       "3                                                                         \n",
       "4                                                                         \n",
       "...        ...     ...     ...        ...            ...            ...   \n",
       "336023  0.9959  0.4667  0.9962                                            \n",
       "336024  0.9959  0.4667  0.9962                                            \n",
       "336025  0.9959  0.4667  0.9962                                            \n",
       "336026  0.9959  0.4667  0.9962                                            \n",
       "336027       0    0.05  0.2731                                            \n",
       "\n",
       "       gnomADe_ASJ_AF gnomADe_EAS_AF gnomADe_FIN_AF gnomADe_NFE_AF  \\\n",
       "0              0.9767         0.9994         0.9916         0.9726   \n",
       "1              0.8131         0.7793         0.8652         0.8438   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "336023                                                               \n",
       "336024                                                               \n",
       "336025                                                               \n",
       "336026                                                               \n",
       "336027                                                               \n",
       "\n",
       "       gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF  \\\n",
       "0              0.9506         0.9854      0.846         0.5948          0.998   \n",
       "1              0.7932         0.8049     0.4864         0.2916         0.6511   \n",
       "2                                                                               \n",
       "3                                                                               \n",
       "4                                                                               \n",
       "...               ...            ...        ...            ...            ...   \n",
       "336023                                   0.6593         0.8119         0.2864   \n",
       "336024                                   0.6593         0.8119         0.2864   \n",
       "336025                                   0.6593         0.8119         0.2864   \n",
       "336026                                   0.6593         0.8119         0.2864   \n",
       "336027                                                                          \n",
       "\n",
       "       gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF  \\\n",
       "0              0.8951         0.9784         0.9998         0.9907   \n",
       "1              0.4782         0.6714         0.6047         0.6987   \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "336023         0.5121         0.9004         0.9992         0.9714   \n",
       "336024         0.5121         0.9004         0.9992         0.9714   \n",
       "336025         0.5121         0.9004         0.9992         0.9714   \n",
       "336026         0.5121         0.9004         0.9992         0.9714   \n",
       "336027                                                               \n",
       "\n",
       "       gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  \\\n",
       "0                 0.9         0.9674         0.8624         0.9772  0.9998   \n",
       "1              0.6441         0.6751         0.4659          0.622   0.876   \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...               ...            ...            ...            ...     ...   \n",
       "336023         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336024         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336025         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336026         0.9865         0.4599         0.6294         0.9906  0.9992   \n",
       "336027                                                              0.2731   \n",
       "\n",
       "        MAX_AF_POPS                 BIOTYPE EXON INTRON  \\\n",
       "0       gnomADg EAS          protein coding  3/3          \n",
       "1               EAS          protein coding  3/3          \n",
       "2                            protein coding               \n",
       "3                                    lncRNA         1/2   \n",
       "4                            protein coding               \n",
       "...             ...                     ...  ...    ...   \n",
       "336023  gnomADg EAS                  lncRNA               \n",
       "336024  gnomADg EAS                  lncRNA               \n",
       "336025  gnomADg EAS                  lncRNA               \n",
       "336026  gnomADg EAS                  lncRNA               \n",
       "336027          SAS  unprocessed pseudogene  1/9          \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                T162A  Aca/Gca      1   \n",
       "1                                 S290  tcT/tcC      1   \n",
       "2                                  nan              -1   \n",
       "3                                  nan              -1   \n",
       "4                                  nan              -1   \n",
       "...                                ...      ...    ...   \n",
       "336023                             nan              -1   \n",
       "336024                             nan              -1   \n",
       "336025                             nan              -1   \n",
       "336026                             nan              -1   \n",
       "336027                             nan               1   \n",
       "\n",
       "                                                   PUBMED Matched_Gene  \\\n",
       "0                                                                        \n",
       "1                                                                        \n",
       "2                                                                        \n",
       "3                                                                        \n",
       "4                                                                        \n",
       "...                                                   ...          ...   \n",
       "336023  22271044,29391530,24262073,15896936,32996047,3...                \n",
       "336024  22271044,29391530,24262073,15896936,32996047,3...                \n",
       "336025  22271044,29391530,24262073,15896936,32996047,3...                \n",
       "336026  22271044,29391530,24262073,15896936,32996047,3...                \n",
       "336027                                  34424929,35176104                \n",
       "\n",
       "       Oncogenic Mutation effect  \n",
       "0            NaN             NaN  \n",
       "1            NaN             NaN  \n",
       "2            NaN             NaN  \n",
       "3            NaN             NaN  \n",
       "4            NaN             NaN  \n",
       "...          ...             ...  \n",
       "336023       NaN             NaN  \n",
       "336024       NaN             NaN  \n",
       "336025       NaN             NaN  \n",
       "336026       NaN             NaN  \n",
       "336027       NaN             NaN  \n",
       "\n",
       "[336028 rows x 71 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(merged_2, df_3, on=['Matched_Gene', 'Protein Position and Amino Acid'], how='left', sort=False)\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "092ffdf1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Likely Neutral    21\n",
       "Inconclusive      11\n",
       "Name: Oncogenic, dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Oncogenic.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "09d972a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Likely- Neutral    12\n",
       "Inconclusive       11\n",
       "Likely Neutral      9\n",
       "Name: Mutation effect, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3['Mutation effect'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "de59f3af",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHHSPTGPTTL60_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be414f9c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1bdd7f77",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "2cf6d7c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+76G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297877</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>52</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4125E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297878</th>\n",
       "      <td></td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000437359.1</td>\n",
       "      <td>n.48T&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698E-7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0681</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.2731</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2731</td>\n",
       "      <td>SAS</td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>34424929,35176104</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297879</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000315357.9</td>\n",
       "      <td>c.714+2253G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297880</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000382365.7</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/18</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297881</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000446723.4</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>297882 rows Ã— 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name          rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0                 rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "1                 rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "2                 rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "3                 rs1166163776  chr1     65649   G   A  Heterozygous   \n",
       "4                 rs1166163776  chr1     65649   G   A  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "297877    EIF1AY     rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "297878              rs17250535  chrY  21311315   T   A    Homozygous   \n",
       "297879      DAZ3  rs1156813394  chrY  24796170   C   G    Homozygous   \n",
       "297880      DAZ3  rs1156813394  chrY  24796170   C   G    Homozygous   \n",
       "297881      DAZ3  rs1156813394  chrY  24796170   C   G    Homozygous   \n",
       "\n",
       "                               Consequence Consequence_score    IMPACT  \\\n",
       "0                  downstream gene variant              2/10  MODIFIER   \n",
       "1                           intron variant              2/10  MODIFIER   \n",
       "2                  downstream gene variant              2/10  MODIFIER   \n",
       "3                  downstream gene variant              2/10  MODIFIER   \n",
       "4                           intron variant              2/10  MODIFIER   \n",
       "...                                    ...               ...       ...   \n",
       "297877               upstream gene variant              2/10  MODIFIER   \n",
       "297878  non coding transcript exon variant              2/10  MODIFIER   \n",
       "297879                      intron variant              2/10  MODIFIER   \n",
       "297880                      intron variant              2/10  MODIFIER   \n",
       "297881                      intron variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                1.5                                                     \n",
       "1                1.5                                                     \n",
       "2                1.5                                                     \n",
       "3                1.5                                                     \n",
       "4                1.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "297877           1.5                                                     \n",
       "297878           1.5                                                     \n",
       "297879           1.5                                                     \n",
       "297880           1.5                                                     \n",
       "297881           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript) HGVSp HGVSp (Transcript)   GT  \\\n",
       "0                                       None                     None  0/1   \n",
       "1       ENST00000641515.2          c.9+18C>T                     None  0/1   \n",
       "2                                       None                     None  0/1   \n",
       "3                                       None                     None  0/1   \n",
       "4       ENST00000641515.2          c.9+76G>A                     None  0/1   \n",
       "...                   ...                ...   ...                ...  ...   \n",
       "297877                                  None                     None  1/1   \n",
       "297878  ENST00000437359.1            n.48T>A                     None  1/1   \n",
       "297879  ENST00000315357.9      c.714+2253G>C                     None  1/1   \n",
       "297880  ENST00000382365.7        c.715-72G>C                     None  1/1   \n",
       "297881  ENST00000446723.4        c.715-72G>C                     None  1/1   \n",
       "\n",
       "        GQ SDP  DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT PolyPhen  \\\n",
       "0       23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3                 \n",
       "1       23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3                 \n",
       "2       23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3                 \n",
       "3       20  19  19  13   6  31.58%   9.828E-3   1  12   2   4                 \n",
       "4       20  19  19  13   6  31.58%   9.828E-3   1  12   2   4                 \n",
       "...     ..  ..  ..  ..  ..     ...        ...  ..  ..  ..  ..  ...      ...   \n",
       "297877  52  10  10   0  10    100%  5.4125E-6   0   0   3   7                 \n",
       "297878  64  12  12   0  12    100%   3.698E-7   0   0   6   6                 \n",
       "297879  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6                 \n",
       "297880  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6                 \n",
       "297881  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6                 \n",
       "\n",
       "            AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF  \\\n",
       "0                                                                   \n",
       "1                                                                   \n",
       "2                                                                   \n",
       "3                                                                   \n",
       "4                                                                   \n",
       "...        ...     ...     ...     ...     ...     ...        ...   \n",
       "297877  0.8273  0.9624  0.5824  0.9959  0.4667  0.9962              \n",
       "297878  0.0681       0  0.0059       0    0.05  0.2731              \n",
       "297879                                                              \n",
       "297880                                                              \n",
       "297881                                                              \n",
       "\n",
       "       gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "297877                                                               \n",
       "297878                                                               \n",
       "297879                                                               \n",
       "297880                                                               \n",
       "297881                                                               \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.03432   \n",
       "4                                                                     0.03432   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "297877                                                                 0.6593   \n",
       "297878                                                                          \n",
       "297879                                                                      0   \n",
       "297880                                                                      0   \n",
       "297881                                                                      0   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.01435        0.05263        0.03176        0.09639   \n",
       "4             0.01435        0.05263        0.03176        0.09639   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.8119         0.2864         0.5121         0.9004   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3            0.001832         0.0122        0.07895        0.09393   \n",
       "4            0.001832         0.0122        0.07895        0.09393   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.9992         0.9714         0.9865         0.4599   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF   MAX_AF  \\\n",
       "0             0.05357        0.02083   0.1667   \n",
       "1             0.05357        0.02083   0.1667   \n",
       "2             0.05357        0.02083   0.1667   \n",
       "3             0.02439        0.03321  0.09639   \n",
       "4             0.02439        0.03321  0.09639   \n",
       "...               ...            ...      ...   \n",
       "297877         0.6294         0.9906   0.9992   \n",
       "297878                                 0.2731   \n",
       "297879              0              0        0   \n",
       "297880              0              0        0   \n",
       "297881              0              0        0   \n",
       "\n",
       "                                              MAX_AF_POPS  \\\n",
       "0                                             gnomADg AMI   \n",
       "1                                             gnomADg AMI   \n",
       "2                                             gnomADg AMI   \n",
       "3                                             gnomADg ASJ   \n",
       "4                                             gnomADg ASJ   \n",
       "...                                                   ...   \n",
       "297877                                        gnomADg EAS   \n",
       "297878                                                SAS   \n",
       "297879  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297880  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297881  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3       transcribed unprocessed pseudogene               \n",
       "4                           protein coding         2/2   \n",
       "...                                    ...  ...    ...   \n",
       "297877                              lncRNA               \n",
       "297878              unprocessed pseudogene  1/9          \n",
       "297879                      protein coding        9/15   \n",
       "297880                      protein coding        9/18   \n",
       "297881                      protein coding        9/16   \n",
       "\n",
       "       Protein Position and Amino Acid Codons STRAND  \\\n",
       "0                                  nan             1   \n",
       "1                                  nan             1   \n",
       "2                                  nan             1   \n",
       "3                                  nan             1   \n",
       "4                                  nan             1   \n",
       "...                                ...    ...    ...   \n",
       "297877                             nan            -1   \n",
       "297878                             nan             1   \n",
       "297879                             nan            -1   \n",
       "297880                             nan            -1   \n",
       "297881                             nan            -1   \n",
       "\n",
       "                                                   PUBMED Gene_Match  \n",
       "0                                                                 No  \n",
       "1                                                                 No  \n",
       "2                                                                 No  \n",
       "3                                                                 No  \n",
       "4                                                                 No  \n",
       "...                                                   ...        ...  \n",
       "297877  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "297878                                  34424929,35176104         No  \n",
       "297879                                                            No  \n",
       "297880                                                            No  \n",
       "297881                                                            No  \n",
       "\n",
       "[297882 rows x 69 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in merged_2['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            merged_2.loc[merged_2['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "18368899",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     296225\n",
       "Yes      1657\n",
       "Name: Gene_Match, dtype: int64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "44475d3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+76G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297877</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>52</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4125E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297878</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000437359.1</td>\n",
       "      <td>n.48T&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698E-7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0681</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.2731</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2731</td>\n",
       "      <td>SAS</td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>34424929,35176104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297879</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000315357.9</td>\n",
       "      <td>c.714+2253G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297880</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000382365.7</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/18</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297881</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000446723.4</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>297882 rows Ã— 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match          rsID CHROM       POS REF ALT  \\\n",
       "0                        No  rs1416222198  chr1     65591   C   T   \n",
       "1                        No  rs1416222198  chr1     65591   C   T   \n",
       "2                        No  rs1416222198  chr1     65591   C   T   \n",
       "3                        No  rs1166163776  chr1     65649   G   A   \n",
       "4                        No  rs1166163776  chr1     65649   G   A   \n",
       "...          ...        ...           ...   ...       ...  ..  ..   \n",
       "297877    EIF1AY         No     rs9786153  chrY  20577481   C   T   \n",
       "297878                   No    rs17250535  chrY  21311315   T   A   \n",
       "297879      DAZ3         No  rs1156813394  chrY  24796170   C   G   \n",
       "297880      DAZ3         No  rs1156813394  chrY  24796170   C   G   \n",
       "297881      DAZ3         No  rs1156813394  chrY  24796170   C   G   \n",
       "\n",
       "            Zygosity                         Consequence Consequence_score  \\\n",
       "0       Heterozygous             downstream gene variant              2/10   \n",
       "1       Heterozygous                      intron variant              2/10   \n",
       "2       Heterozygous             downstream gene variant              2/10   \n",
       "3       Heterozygous             downstream gene variant              2/10   \n",
       "4       Heterozygous                      intron variant              2/10   \n",
       "...              ...                                 ...               ...   \n",
       "297877    Homozygous               upstream gene variant              2/10   \n",
       "297878    Homozygous  non coding transcript exon variant              2/10   \n",
       "297879    Homozygous                      intron variant              2/10   \n",
       "297880    Homozygous                      intron variant              2/10   \n",
       "297881    Homozygous                      intron variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0       MODIFIER           1.5                                             \n",
       "1       MODIFIER           1.5                                             \n",
       "2       MODIFIER           1.5                                             \n",
       "3       MODIFIER           1.5                                             \n",
       "4       MODIFIER           1.5                                             \n",
       "...          ...           ...           ...      ...                ...   \n",
       "297877  MODIFIER           1.5                                             \n",
       "297878  MODIFIER           1.5                                             \n",
       "297879  MODIFIER           1.5                                             \n",
       "297880  MODIFIER           1.5                                             \n",
       "297881  MODIFIER           1.5                                             \n",
       "\n",
       "       ClinVar              HGVSc HGVSc (Transcript) HGVSp HGVSp (Transcript)  \\\n",
       "0                                               None                     None   \n",
       "1               ENST00000641515.2          c.9+18C>T                     None   \n",
       "2                                               None                     None   \n",
       "3                                               None                     None   \n",
       "4               ENST00000641515.2          c.9+76G>A                     None   \n",
       "...        ...                ...                ...   ...                ...   \n",
       "297877                                          None                     None   \n",
       "297878          ENST00000437359.1            n.48T>A                     None   \n",
       "297879          ENST00000315357.9      c.714+2253G>C                     None   \n",
       "297880          ENST00000382365.7        c.715-72G>C                     None   \n",
       "297881          ENST00000446723.4        c.715-72G>C                     None   \n",
       "\n",
       "         GT  GQ SDP  DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT  \\\n",
       "0       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "1       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "2       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "3       0/1  20  19  19  13   6  31.58%   9.828E-3   1  12   2   4        \n",
       "4       0/1  20  19  19  13   6  31.58%   9.828E-3   1  12   2   4        \n",
       "...     ...  ..  ..  ..  ..  ..     ...        ...  ..  ..  ..  ..  ...   \n",
       "297877  1/1  52  10  10   0  10    100%  5.4125E-6   0   0   3   7        \n",
       "297878  1/1  64  12  12   0  12    100%   3.698E-7   0   0   6   6        \n",
       "297879  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "297880  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "297881  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "\n",
       "       PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF  \\\n",
       "0                                                                            \n",
       "1                                                                            \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...         ...     ...     ...     ...     ...     ...     ...        ...   \n",
       "297877           0.8273  0.9624  0.5824  0.9959  0.4667  0.9962              \n",
       "297878           0.0681       0  0.0059       0    0.05  0.2731              \n",
       "297879                                                                       \n",
       "297880                                                                       \n",
       "297881                                                                       \n",
       "\n",
       "       gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "297877                                                               \n",
       "297878                                                               \n",
       "297879                                                               \n",
       "297880                                                               \n",
       "297881                                                               \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.03432   \n",
       "4                                                                     0.03432   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "297877                                                                 0.6593   \n",
       "297878                                                                          \n",
       "297879                                                                      0   \n",
       "297880                                                                      0   \n",
       "297881                                                                      0   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.01435        0.05263        0.03176        0.09639   \n",
       "4             0.01435        0.05263        0.03176        0.09639   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.8119         0.2864         0.5121         0.9004   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3            0.001832         0.0122        0.07895        0.09393   \n",
       "4            0.001832         0.0122        0.07895        0.09393   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.9992         0.9714         0.9865         0.4599   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF   MAX_AF  \\\n",
       "0             0.05357        0.02083   0.1667   \n",
       "1             0.05357        0.02083   0.1667   \n",
       "2             0.05357        0.02083   0.1667   \n",
       "3             0.02439        0.03321  0.09639   \n",
       "4             0.02439        0.03321  0.09639   \n",
       "...               ...            ...      ...   \n",
       "297877         0.6294         0.9906   0.9992   \n",
       "297878                                 0.2731   \n",
       "297879              0              0        0   \n",
       "297880              0              0        0   \n",
       "297881              0              0        0   \n",
       "\n",
       "                                              MAX_AF_POPS  \\\n",
       "0                                             gnomADg AMI   \n",
       "1                                             gnomADg AMI   \n",
       "2                                             gnomADg AMI   \n",
       "3                                             gnomADg ASJ   \n",
       "4                                             gnomADg ASJ   \n",
       "...                                                   ...   \n",
       "297877                                        gnomADg EAS   \n",
       "297878                                                SAS   \n",
       "297879  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297880  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297881  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3       transcribed unprocessed pseudogene               \n",
       "4                           protein coding         2/2   \n",
       "...                                    ...  ...    ...   \n",
       "297877                              lncRNA               \n",
       "297878              unprocessed pseudogene  1/9          \n",
       "297879                      protein coding        9/15   \n",
       "297880                      protein coding        9/18   \n",
       "297881                      protein coding        9/16   \n",
       "\n",
       "       Protein Position and Amino Acid Codons STRAND  \\\n",
       "0                                  nan             1   \n",
       "1                                  nan             1   \n",
       "2                                  nan             1   \n",
       "3                                  nan             1   \n",
       "4                                  nan             1   \n",
       "...                                ...    ...    ...   \n",
       "297877                             nan            -1   \n",
       "297878                             nan             1   \n",
       "297879                             nan            -1   \n",
       "297880                             nan            -1   \n",
       "297881                             nan            -1   \n",
       "\n",
       "                                                   PUBMED  \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "3                                                          \n",
       "4                                                          \n",
       "...                                                   ...  \n",
       "297877  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "297878                                  34424929,35176104  \n",
       "297879                                                     \n",
       "297880                                                     \n",
       "297881                                                     \n",
       "\n",
       "[297882 rows x 69 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2['POS'] = merged_2['POS'].astype('int64')\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "04079642",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101728759</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101729212</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101728656</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101729109</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr19</td>\n",
       "      <td>57232128</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>chr11</td>\n",
       "      <td>32396328</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>chr11</td>\n",
       "      <td>32400021</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>chr11</td>\n",
       "      <td>32396277</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>chr11</td>\n",
       "      <td>32392036</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451</th>\n",
       "      <td>chr11</td>\n",
       "      <td>32392670</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>339 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS Literature Covered/Not_Covered\n",
       "0    chr12  101728759        Yes             Covered\n",
       "1    chr12  101729212        Yes             Covered\n",
       "3    chr12  101728656        Yes             Covered\n",
       "4    chr12  101729109        Yes             Covered\n",
       "6    chr19   57232128        Yes             Covered\n",
       "..     ...        ...        ...                 ...\n",
       "438  chr11   32396328        Yes             Covered\n",
       "439  chr11   32400021        Yes             Covered\n",
       "441  chr11   32396277        Yes             Covered\n",
       "450  chr11   32392036        Yes             Covered\n",
       "451  chr11   32392670        Yes             Covered\n",
       "\n",
       "[339 rows x 4 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Reproductivehealth_Male_Lit_final_Positions_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "385a356c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+76G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297877</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>52</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4125E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297878</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000437359.1</td>\n",
       "      <td>n.48T&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698E-7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0681</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.2731</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2731</td>\n",
       "      <td>SAS</td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>34424929,35176104</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297879</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000315357.9</td>\n",
       "      <td>c.714+2253G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297880</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000382365.7</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/18</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297881</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000446723.4</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>297882 rows Ã— 71 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match          rsID CHROM       POS REF ALT  \\\n",
       "0                        No  rs1416222198  chr1     65591   C   T   \n",
       "1                        No  rs1416222198  chr1     65591   C   T   \n",
       "2                        No  rs1416222198  chr1     65591   C   T   \n",
       "3                        No  rs1166163776  chr1     65649   G   A   \n",
       "4                        No  rs1166163776  chr1     65649   G   A   \n",
       "...          ...        ...           ...   ...       ...  ..  ..   \n",
       "297877    EIF1AY         No     rs9786153  chrY  20577481   C   T   \n",
       "297878                   No    rs17250535  chrY  21311315   T   A   \n",
       "297879      DAZ3         No  rs1156813394  chrY  24796170   C   G   \n",
       "297880      DAZ3         No  rs1156813394  chrY  24796170   C   G   \n",
       "297881      DAZ3         No  rs1156813394  chrY  24796170   C   G   \n",
       "\n",
       "            Zygosity                         Consequence Consequence_score  \\\n",
       "0       Heterozygous             downstream gene variant              2/10   \n",
       "1       Heterozygous                      intron variant              2/10   \n",
       "2       Heterozygous             downstream gene variant              2/10   \n",
       "3       Heterozygous             downstream gene variant              2/10   \n",
       "4       Heterozygous                      intron variant              2/10   \n",
       "...              ...                                 ...               ...   \n",
       "297877    Homozygous               upstream gene variant              2/10   \n",
       "297878    Homozygous  non coding transcript exon variant              2/10   \n",
       "297879    Homozygous                      intron variant              2/10   \n",
       "297880    Homozygous                      intron variant              2/10   \n",
       "297881    Homozygous                      intron variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0       MODIFIER           1.5                                             \n",
       "1       MODIFIER           1.5                                             \n",
       "2       MODIFIER           1.5                                             \n",
       "3       MODIFIER           1.5                                             \n",
       "4       MODIFIER           1.5                                             \n",
       "...          ...           ...           ...      ...                ...   \n",
       "297877  MODIFIER           1.5                                             \n",
       "297878  MODIFIER           1.5                                             \n",
       "297879  MODIFIER           1.5                                             \n",
       "297880  MODIFIER           1.5                                             \n",
       "297881  MODIFIER           1.5                                             \n",
       "\n",
       "       ClinVar              HGVSc HGVSc (Transcript) HGVSp HGVSp (Transcript)  \\\n",
       "0                                               None                     None   \n",
       "1               ENST00000641515.2          c.9+18C>T                     None   \n",
       "2                                               None                     None   \n",
       "3                                               None                     None   \n",
       "4               ENST00000641515.2          c.9+76G>A                     None   \n",
       "...        ...                ...                ...   ...                ...   \n",
       "297877                                          None                     None   \n",
       "297878          ENST00000437359.1            n.48T>A                     None   \n",
       "297879          ENST00000315357.9      c.714+2253G>C                     None   \n",
       "297880          ENST00000382365.7        c.715-72G>C                     None   \n",
       "297881          ENST00000446723.4        c.715-72G>C                     None   \n",
       "\n",
       "         GT  GQ SDP  DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT  \\\n",
       "0       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "1       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "2       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "3       0/1  20  19  19  13   6  31.58%   9.828E-3   1  12   2   4        \n",
       "4       0/1  20  19  19  13   6  31.58%   9.828E-3   1  12   2   4        \n",
       "...     ...  ..  ..  ..  ..  ..     ...        ...  ..  ..  ..  ..  ...   \n",
       "297877  1/1  52  10  10   0  10    100%  5.4125E-6   0   0   3   7        \n",
       "297878  1/1  64  12  12   0  12    100%   3.698E-7   0   0   6   6        \n",
       "297879  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "297880  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "297881  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "\n",
       "       PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF  \\\n",
       "0                                                                            \n",
       "1                                                                            \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...         ...     ...     ...     ...     ...     ...     ...        ...   \n",
       "297877           0.8273  0.9624  0.5824  0.9959  0.4667  0.9962              \n",
       "297878           0.0681       0  0.0059       0    0.05  0.2731              \n",
       "297879                                                                       \n",
       "297880                                                                       \n",
       "297881                                                                       \n",
       "\n",
       "       gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "297877                                                               \n",
       "297878                                                               \n",
       "297879                                                               \n",
       "297880                                                               \n",
       "297881                                                               \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.03432   \n",
       "4                                                                     0.03432   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "297877                                                                 0.6593   \n",
       "297878                                                                          \n",
       "297879                                                                      0   \n",
       "297880                                                                      0   \n",
       "297881                                                                      0   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.01435        0.05263        0.03176        0.09639   \n",
       "4             0.01435        0.05263        0.03176        0.09639   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.8119         0.2864         0.5121         0.9004   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3            0.001832         0.0122        0.07895        0.09393   \n",
       "4            0.001832         0.0122        0.07895        0.09393   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.9992         0.9714         0.9865         0.4599   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF   MAX_AF  \\\n",
       "0             0.05357        0.02083   0.1667   \n",
       "1             0.05357        0.02083   0.1667   \n",
       "2             0.05357        0.02083   0.1667   \n",
       "3             0.02439        0.03321  0.09639   \n",
       "4             0.02439        0.03321  0.09639   \n",
       "...               ...            ...      ...   \n",
       "297877         0.6294         0.9906   0.9992   \n",
       "297878                                 0.2731   \n",
       "297879              0              0        0   \n",
       "297880              0              0        0   \n",
       "297881              0              0        0   \n",
       "\n",
       "                                              MAX_AF_POPS  \\\n",
       "0                                             gnomADg AMI   \n",
       "1                                             gnomADg AMI   \n",
       "2                                             gnomADg AMI   \n",
       "3                                             gnomADg ASJ   \n",
       "4                                             gnomADg ASJ   \n",
       "...                                                   ...   \n",
       "297877                                        gnomADg EAS   \n",
       "297878                                                SAS   \n",
       "297879  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297880  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297881  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3       transcribed unprocessed pseudogene               \n",
       "4                           protein coding         2/2   \n",
       "...                                    ...  ...    ...   \n",
       "297877                              lncRNA               \n",
       "297878              unprocessed pseudogene  1/9          \n",
       "297879                      protein coding        9/15   \n",
       "297880                      protein coding        9/18   \n",
       "297881                      protein coding        9/16   \n",
       "\n",
       "       Protein Position and Amino Acid Codons STRAND  \\\n",
       "0                                  nan             1   \n",
       "1                                  nan             1   \n",
       "2                                  nan             1   \n",
       "3                                  nan             1   \n",
       "4                                  nan             1   \n",
       "...                                ...    ...    ...   \n",
       "297877                             nan            -1   \n",
       "297878                             nan             1   \n",
       "297879                             nan            -1   \n",
       "297880                             nan            -1   \n",
       "297881                             nan            -1   \n",
       "\n",
       "                                                   PUBMED Literature  \\\n",
       "0                                                                 No   \n",
       "1                                                                 No   \n",
       "2                                                                 No   \n",
       "3                                                                 No   \n",
       "4                                                                 No   \n",
       "...                                                   ...        ...   \n",
       "297877  22271044,29391530,24262073,15896936,32996047,3...         No   \n",
       "297878                                  34424929,35176104         No   \n",
       "297879                                                            No   \n",
       "297880                                                            No   \n",
       "297881                                                            No   \n",
       "\n",
       "       Covered/Not_Covered  \n",
       "0                      NaN  \n",
       "1                      NaN  \n",
       "2                      NaN  \n",
       "3                      NaN  \n",
       "4                      NaN  \n",
       "...                    ...  \n",
       "297877                 NaN  \n",
       "297878                 NaN  \n",
       "297879                 NaN  \n",
       "297880                 NaN  \n",
       "297881                 NaN  \n",
       "\n",
       "[297882 rows x 71 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(merged_2, df_3, on=['CHROM', 'POS'], how='left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "58dcebad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No    297882\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "0bd49452",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>4.9169E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03416</td>\n",
       "      <td>0.02306</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>0.02711</td>\n",
       "      <td>0.03333</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09167</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.02083</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>gnomADg AMI</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>No</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs1166163776</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65649</td>\n",
       "      <td>No</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+76G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>31.58%</td>\n",
       "      <td>9.828E-3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.03432</td>\n",
       "      <td>0.01435</td>\n",
       "      <td>0.05263</td>\n",
       "      <td>0.03176</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>0.07895</td>\n",
       "      <td>0.09393</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>0.03321</td>\n",
       "      <td>0.09639</td>\n",
       "      <td>gnomADg ASJ</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297877</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>52</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4125E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.8273</td>\n",
       "      <td>0.9624</td>\n",
       "      <td>0.5824</td>\n",
       "      <td>0.9959</td>\n",
       "      <td>0.4667</td>\n",
       "      <td>0.9962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.6593</td>\n",
       "      <td>0.8119</td>\n",
       "      <td>0.2864</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.9004</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9714</td>\n",
       "      <td>0.9865</td>\n",
       "      <td>0.4599</td>\n",
       "      <td>0.6294</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297878</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>rs17250535</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21311315</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000437359.1</td>\n",
       "      <td>n.48T&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698E-7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.0681</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.2731</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.2731</td>\n",
       "      <td>SAS</td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td>1/9</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>34424929,35176104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297879</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000315357.9</td>\n",
       "      <td>c.714+2253G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/15</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297880</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000382365.7</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/18</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297881</th>\n",
       "      <td>DAZ3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1156813394</td>\n",
       "      <td>chrY</td>\n",
       "      <td>24796170</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000446723.4</td>\n",
       "      <td>c.715-72G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>74</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>77.27%</td>\n",
       "      <td>3.8368E-8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...</td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>9/16</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>297882 rows Ã— 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match          rsID CHROM       POS Literature REF ALT  \\\n",
       "0                        No  rs1416222198  chr1     65591         No   C   T   \n",
       "1                        No  rs1416222198  chr1     65591         No   C   T   \n",
       "2                        No  rs1416222198  chr1     65591         No   C   T   \n",
       "3                        No  rs1166163776  chr1     65649         No   G   A   \n",
       "4                        No  rs1166163776  chr1     65649         No   G   A   \n",
       "...          ...        ...           ...   ...       ...        ...  ..  ..   \n",
       "297877    EIF1AY         No     rs9786153  chrY  20577481         No   C   T   \n",
       "297878                   No    rs17250535  chrY  21311315         No   T   A   \n",
       "297879      DAZ3         No  rs1156813394  chrY  24796170         No   C   G   \n",
       "297880      DAZ3         No  rs1156813394  chrY  24796170         No   C   G   \n",
       "297881      DAZ3         No  rs1156813394  chrY  24796170         No   C   G   \n",
       "\n",
       "            Zygosity                         Consequence Consequence_score  \\\n",
       "0       Heterozygous             downstream gene variant              2/10   \n",
       "1       Heterozygous                      intron variant              2/10   \n",
       "2       Heterozygous             downstream gene variant              2/10   \n",
       "3       Heterozygous             downstream gene variant              2/10   \n",
       "4       Heterozygous                      intron variant              2/10   \n",
       "...              ...                                 ...               ...   \n",
       "297877    Homozygous               upstream gene variant              2/10   \n",
       "297878    Homozygous  non coding transcript exon variant              2/10   \n",
       "297879    Homozygous                      intron variant              2/10   \n",
       "297880    Homozygous                      intron variant              2/10   \n",
       "297881    Homozygous                      intron variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0       MODIFIER           1.5                                             \n",
       "1       MODIFIER           1.5                                             \n",
       "2       MODIFIER           1.5                                             \n",
       "3       MODIFIER           1.5                                             \n",
       "4       MODIFIER           1.5                                             \n",
       "...          ...           ...           ...      ...                ...   \n",
       "297877  MODIFIER           1.5                                             \n",
       "297878  MODIFIER           1.5                                             \n",
       "297879  MODIFIER           1.5                                             \n",
       "297880  MODIFIER           1.5                                             \n",
       "297881  MODIFIER           1.5                                             \n",
       "\n",
       "       ClinVar              HGVSc HGVSc (Transcript) HGVSp HGVSp (Transcript)  \\\n",
       "0                                               None                     None   \n",
       "1               ENST00000641515.2          c.9+18C>T                     None   \n",
       "2                                               None                     None   \n",
       "3                                               None                     None   \n",
       "4               ENST00000641515.2          c.9+76G>A                     None   \n",
       "...        ...                ...                ...   ...                ...   \n",
       "297877                                          None                     None   \n",
       "297878          ENST00000437359.1            n.48T>A                     None   \n",
       "297879          ENST00000315357.9      c.714+2253G>C                     None   \n",
       "297880          ENST00000382365.7        c.715-72G>C                     None   \n",
       "297881          ENST00000446723.4        c.715-72G>C                     None   \n",
       "\n",
       "         GT  GQ SDP  DP  RD  AD    FREQ       PVAL RDF RDR ADF ADR SIFT  \\\n",
       "0       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "1       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "2       0/1  23  26  26  19   7  26.92%  4.9169E-3  10   9   4   3        \n",
       "3       0/1  20  19  19  13   6  31.58%   9.828E-3   1  12   2   4        \n",
       "4       0/1  20  19  19  13   6  31.58%   9.828E-3   1  12   2   4        \n",
       "...     ...  ..  ..  ..  ..  ..     ...        ...  ..  ..  ..  ..  ...   \n",
       "297877  1/1  52  10  10   0  10    100%  5.4125E-6   0   0   3   7        \n",
       "297878  1/1  64  12  12   0  12    100%   3.698E-7   0   0   6   6        \n",
       "297879  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "297880  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "297881  1/1  74  22  22   5  17  77.27%  3.8368E-8   1   4  11   6        \n",
       "\n",
       "       PolyPhen      AF  AFR_AF  AMR_AF  EAS_AF  EUR_AF  SAS_AF gnomADe_AF  \\\n",
       "0                                                                            \n",
       "1                                                                            \n",
       "2                                                                            \n",
       "3                                                                            \n",
       "4                                                                            \n",
       "...         ...     ...     ...     ...     ...     ...     ...        ...   \n",
       "297877           0.8273  0.9624  0.5824  0.9959  0.4667  0.9962              \n",
       "297878           0.0681       0  0.0059       0    0.05  0.2731              \n",
       "297879                                                                       \n",
       "297880                                                                       \n",
       "297881                                                                       \n",
       "\n",
       "       gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                    \n",
       "1                                                                    \n",
       "2                                                                    \n",
       "3                                                                    \n",
       "4                                                                    \n",
       "...               ...            ...            ...            ...   \n",
       "297877                                                               \n",
       "297878                                                               \n",
       "297879                                                               \n",
       "297880                                                               \n",
       "297881                                                               \n",
       "\n",
       "       gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.03416   \n",
       "1                                                                     0.03416   \n",
       "2                                                                     0.03416   \n",
       "3                                                                     0.03432   \n",
       "4                                                                     0.03432   \n",
       "...               ...            ...            ...            ...        ...   \n",
       "297877                                                                 0.6593   \n",
       "297878                                                                          \n",
       "297879                                                                      0   \n",
       "297880                                                                      0   \n",
       "297881                                                                      0   \n",
       "\n",
       "       gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0             0.02306         0.1667        0.02711        0.03333   \n",
       "1             0.02306         0.1667        0.02711        0.03333   \n",
       "2             0.02306         0.1667        0.02711        0.03333   \n",
       "3             0.01435        0.05263        0.03176        0.09639   \n",
       "4             0.01435        0.05263        0.03176        0.09639   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.8119         0.2864         0.5121         0.9004   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.001639              0           0.05        0.09167   \n",
       "1            0.001639              0           0.05        0.09167   \n",
       "2            0.001639              0           0.05        0.09167   \n",
       "3            0.001832         0.0122        0.07895        0.09393   \n",
       "4            0.001832         0.0122        0.07895        0.09393   \n",
       "...               ...            ...            ...            ...   \n",
       "297877         0.9992         0.9714         0.9865         0.4599   \n",
       "297878                                                               \n",
       "297879              0              0              0              0   \n",
       "297880              0              0              0              0   \n",
       "297881              0              0              0              0   \n",
       "\n",
       "       gnomADg_OTH_AF gnomADg_SAS_AF   MAX_AF  \\\n",
       "0             0.05357        0.02083   0.1667   \n",
       "1             0.05357        0.02083   0.1667   \n",
       "2             0.05357        0.02083   0.1667   \n",
       "3             0.02439        0.03321  0.09639   \n",
       "4             0.02439        0.03321  0.09639   \n",
       "...               ...            ...      ...   \n",
       "297877         0.6294         0.9906   0.9992   \n",
       "297878                                 0.2731   \n",
       "297879              0              0        0   \n",
       "297880              0              0        0   \n",
       "297881              0              0        0   \n",
       "\n",
       "                                              MAX_AF_POPS  \\\n",
       "0                                             gnomADg AMI   \n",
       "1                                             gnomADg AMI   \n",
       "2                                             gnomADg AMI   \n",
       "3                                             gnomADg ASJ   \n",
       "4                                             gnomADg ASJ   \n",
       "...                                                   ...   \n",
       "297877                                        gnomADg EAS   \n",
       "297878                                                SAS   \n",
       "297879  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297880  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "297881  gnomADg AFR,gnomADg AMI,gnomADg AMR,gnomADg AS...   \n",
       "\n",
       "                                   BIOTYPE EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene               \n",
       "1                           protein coding         2/2   \n",
       "2                                   lncRNA               \n",
       "3       transcribed unprocessed pseudogene               \n",
       "4                           protein coding         2/2   \n",
       "...                                    ...  ...    ...   \n",
       "297877                              lncRNA               \n",
       "297878              unprocessed pseudogene  1/9          \n",
       "297879                      protein coding        9/15   \n",
       "297880                      protein coding        9/18   \n",
       "297881                      protein coding        9/16   \n",
       "\n",
       "       Protein Position and Amino Acid Codons STRAND  \\\n",
       "0                                  nan             1   \n",
       "1                                  nan             1   \n",
       "2                                  nan             1   \n",
       "3                                  nan             1   \n",
       "4                                  nan             1   \n",
       "...                                ...    ...    ...   \n",
       "297877                             nan            -1   \n",
       "297878                             nan             1   \n",
       "297879                             nan            -1   \n",
       "297880                             nan            -1   \n",
       "297881                             nan            -1   \n",
       "\n",
       "                                                   PUBMED  \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "3                                                          \n",
       "4                                                          \n",
       "...                                                   ...  \n",
       "297877  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "297878                                  34424929,35176104  \n",
       "297879                                                     \n",
       "297880                                                     \n",
       "297881                                                     \n",
       "\n",
       "[297882 rows x 70 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = merged_3[['Gene Name', 'Gene_Match', 'rsID', 'CHROM', 'POS', 'Literature', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'AF', 'AFR_AF',\n",
    "       'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF',\n",
    "       'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF',\n",
    "       'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF',\n",
    "       'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF',\n",
    "       'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF',\n",
    "       'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "14571e77",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHHSPTGPTTL59_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db05da5e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c987b188",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7ecb11d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d517494",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01ae5cfe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7894e447",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0162137f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89a9e989",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97e7193e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d7dec95",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9ea4569a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4225143d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7bea7d89",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da7f308a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "092080ea",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ba17891",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab07d6ee",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9722338",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1abf95ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03b6cce4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a8f3ee5b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "178a78c0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3373f62b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c543250a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50364c84",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d329b99",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb2d7da8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "910157b2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1928a96",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d480c27",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f02385e8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88bd541c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f812a05d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "745ada77",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f0c8870",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c77db29",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df76f08e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "ef2d2672",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Covered/Not_Covered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978691</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978908</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120997590</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120996531</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978928</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2206</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17394331</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2207</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17396955</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2208</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17395249</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2209</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17413396</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17408487</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Covered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1685 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS Literature Covered/Not_Covered\n",
       "0     chr12  120978691        Yes             Covered\n",
       "1     chr12  120978908        Yes             Covered\n",
       "2     chr12  120997590        Yes             Covered\n",
       "3     chr12  120996531        Yes             Covered\n",
       "4     chr12  120978928        Yes             Covered\n",
       "...     ...        ...        ...                 ...\n",
       "2206  chr11   17394331        Yes             Covered\n",
       "2207  chr11   17396955        Yes             Covered\n",
       "2208  chr11   17395249        Yes             Covered\n",
       "2209  chr11   17413396        Yes             Covered\n",
       "2210  chr11   17408487        Yes             Covered\n",
       "\n",
       "[1685 rows x 4 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Mody_Lit_final_Positions_hg38_hg37 (2).xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chrom):\n",
    "    if pd.notnull(chrom) and chrom.strip() != '':\n",
    "        return 'chr' + str(chrom)\n",
    "    else:\n",
    "        return chrom\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "df_3 = x[['CHROM', 'POS']]\n",
    "df_3['Literature'] = 'Yes'\n",
    "df_3.drop_duplicates(subset='POS', inplace=True)\n",
    "df_3['POS'] = df_3['POS'].astype('int64')\n",
    "df_3 = df_3.reset_index()\n",
    "df_3 = df_3[['CHROM', 'POS', 'Literature']]\n",
    "\n",
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/KAPA HyperExome_hg38_capture_targets (1).bed', sep = '\\t', header = None)\n",
    "df.columns = ['chromosome', 'Start_pos', 'End_pos', 'INFO']\n",
    "\n",
    "df['Extended_Start_pos'] = df['Start_pos'] - 20\n",
    "df['Extended_End_pos'] = df['End_pos'] + 20\n",
    "\n",
    "df['gene_symbol'] = df['INFO'].str.extract(r'gene_symbol=([^;]+)')\n",
    "df = df[['chromosome', 'Extended_Start_pos', 'Extended_End_pos', 'INFO', 'gene_symbol']]\n",
    "\n",
    "\n",
    "# Step 1: Create a dictionary from the df DataFrame\n",
    "chromosome_dict = {}\n",
    "for _, row in df.iterrows():\n",
    "    chromosome = row['chromosome']\n",
    "    start_pos = row['Extended_Start_pos']\n",
    "    end_pos = row['Extended_End_pos']\n",
    "    if chromosome not in chromosome_dict:\n",
    "        chromosome_dict[chromosome] = []\n",
    "    chromosome_dict[chromosome].append((start_pos, end_pos))\n",
    "\n",
    "# Step 2: Define a function to check coverage\n",
    "def check_coverage(row):\n",
    "    pos = row['POS']\n",
    "    chromosome = row['CHROM']\n",
    "    if chromosome in chromosome_dict:\n",
    "        ranges = chromosome_dict[chromosome]\n",
    "        for start, end in ranges:\n",
    "            if start <= pos <= end:\n",
    "                return 'Covered'\n",
    "    return 'Not_Covered'\n",
    "\n",
    "# Step 3: Apply the function to create the new column in dataset2\n",
    "df_3['Covered/Not_Covered'] = df_3.apply(check_coverage, axis=1)\n",
    "\n",
    "df_3 = df_3[df_3['Covered/Not_Covered'] == 'Covered']\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "796bab84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978691</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978908</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120997590</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120996531</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>120978928</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2206</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17394331</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2207</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17396955</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2208</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17395249</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2209</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17413396</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17408487</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1685 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS Literature\n",
       "0     chr12  120978691        Yes\n",
       "1     chr12  120978908        Yes\n",
       "2     chr12  120997590        Yes\n",
       "3     chr12  120996531        Yes\n",
       "4     chr12  120978928        Yes\n",
       "...     ...        ...        ...\n",
       "2206  chr11   17394331        Yes\n",
       "2207  chr11   17396955        Yes\n",
       "2208  chr11   17395249        Yes\n",
       "2209  chr11   17413396        Yes\n",
       "2210  chr11   17408487        Yes\n",
       "\n",
       "[1685 rows x 3 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = df_3[['CHROM', 'POS', 'Literature']]\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "34f1abbe",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Mody_lit_variants.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2ca3c5c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57840f10",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bec6e6fb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6b957460",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d7532880",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14574</td>\n",
       "      <td>rs28503599</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=DDX11...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>75%</td>\n",
       "      <td>3.4965E-3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=DDX11...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>75%</td>\n",
       "      <td>3.4965E-3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14599</td>\n",
       "      <td>rs707680</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=10;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>60%</td>\n",
       "      <td>5.418E-3</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14604</td>\n",
       "      <td>rs541940975</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,0...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>4.5249E-3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14610</td>\n",
       "      <td>rs878986575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=9;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX11...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>4.5249E-3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61971</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57196682</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>20</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>37.5%</td>\n",
       "      <td>8.837E-3</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61972</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208367</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>69</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>76.19%</td>\n",
       "      <td>1.2221E-7</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61973</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208581</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>28</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>42.11%</td>\n",
       "      <td>1.5455E-3</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61974</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209184</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>18</td>\n",
       "      <td>13</td>\n",
       "      <td>41.94%</td>\n",
       "      <td>2.4825E-5</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61975</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211636</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>40</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>50%</td>\n",
       "      <td>9.1981E-5</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61976 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14574   rs28503599   A   G    .   PASS   \n",
       "1      chr1     14590     rs707679   G   A    .   PASS   \n",
       "2      chr1     14599     rs707680   T   A    .   PASS   \n",
       "3      chr1     14604  rs541940975   A   G    .   PASS   \n",
       "4      chr1     14610  rs878986575   T   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "61971  chrY  57196682            .   A   G    .   PASS   \n",
       "61972  chrY  57208367            .   A   G    .   PASS   \n",
       "61973  chrY  57208581            .   G   C    .   PASS   \n",
       "61974  chrY  57209184            .   T   C    .   PASS   \n",
       "61975  chrY  57211636            .   G   A    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ SDP  DP  RD  \\\n",
       "0      ADP=8;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=DDX11...  1/1  24   8   8   2   \n",
       "1      ADP=8;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=DDX11...  1/1  24   8   8   2   \n",
       "2      ADP=10;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....  0/1  22  10  10   4   \n",
       "3      ADP=9;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,0...  0/1  23   9   9   3   \n",
       "4      ADP=9;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX11...  0/1  23   9   9   3   \n",
       "...                                                  ...  ...  ..  ..  ..  ..   \n",
       "61971  ADP=16;WT=0;HET=1;HOM=0;NC=0;CSQ=G|missense_va...  0/1  20  16  16  10   \n",
       "61972  ADP=21;WT=0;HET=0;HOM=1;NC=0;CSQ=G|intron_vari...  1/1  69  21  21   5   \n",
       "61973  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  28  19  19  11   \n",
       "61974  ADP=31;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  46  31  31  18   \n",
       "61975  ADP=22;WT=0;HET=1;HOM=0;NC=0;CSQ=A|intron_vari...  0/1  40  22  22  11   \n",
       "\n",
       "       AD    FREQ       PVAL RDF RDR ADF ADR  \n",
       "0       6     75%  3.4965E-3   2   0   6   0  \n",
       "1       6     75%  3.4965E-3   2   0   6   0  \n",
       "2       6     60%   5.418E-3   4   0   6   0  \n",
       "3       6  66.67%  4.5249E-3   3   0   6   0  \n",
       "4       6  66.67%  4.5249E-3   3   0   6   0  \n",
       "...    ..     ...        ...  ..  ..  ..  ..  \n",
       "61971   6   37.5%   8.837E-3  10   0   4   2  \n",
       "61972  16  76.19%  1.2221E-7   4   1   7   9  \n",
       "61973   8  42.11%  1.5455E-3  11   0   8   0  \n",
       "61974  13  41.94%  2.4825E-5  15   3  11   2  \n",
       "61975  11     50%  9.1981E-5   6   5   9   2  \n",
       "\n",
       "[61976 rows x 20 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_data_df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/220_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False, encoding='ISO-8859-1')\n",
    "vcf_data_df.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf_data_df['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf_data_df = pd.concat([vcf_data_df, sample_cols], axis=1)\n",
    "vcf_data_df = vcf_data_df[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf_data_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2f4b3668",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "127\n"
     ]
    }
   ],
   "source": [
    "gene_list = [\"CSF1\", \"FGF1\", \"FGF2\", \"ELK1\", \"FGF3\", \"IGF1R\", \"FGF4\", \"FGF5\", \"FGF6\", \"FGF7\", \"FGF8\", \"FGF9\", \"PRKACG\", \"AKT2\", \"AKT3\", \"KDR\", \"AKT1\", \"PRKACA\", \"PRKACB\", \"PDGFRB\", \"PRKCG\", \"PDGFRA\", \"MAP2K1\", \"MAP2K2\", \"PRKCB\", \"HGF\", \"FLT3LG\", \n",
    "    \"PRKCA\", \"PGF\", \"RAF1\", \"EPHA2\", \"CSF1R\", \"PDGFB\", \"PDGFBA\", \"TGFA\", \"STK4\", \n",
    "    \"EFNA5\", \"RASGRP2\", \"RASGRP1\", \"EFNA4\", \"RASGRP4\", \"RASGRP3\", \"RAP1B\", \"RAP1A\", \n",
    "    \"RRAS\", \"PDGFD\", \"PDGFC\", \"FGF20\", \"FGF23\", \"FGF22\", \"FGF21\", \"NGFR\", \"BDNF\", \n",
    "    \"INSR\", \"IGF2\", \"IGF1\", \"GNG12\", \"NFKB1\", \"FGF17\", \"EFNA1\", \"FGF16\", \"EFNA3\", \n",
    "    \"EFNA2\", \"FGF19\", \"FGF18\", \"NF1\", \"GRB2\", \"FGFR4\", \"FGFR3\", \"FGFR2\", \"FGF10\", \n",
    "    \"FGFR1\", \"FLT1\", \"FLT3\", \"FLT4\", \"RASGRF2\", \"RASGRF1\", \"FASLG\", \"IKBKB\", \"RAC2\", \n",
    "    \"RAC3\", \"IKBKG\", \"RAC1\", \"HRAS\", \"PLA2G4F\", \"PLA2G4D\", \"CHUK\", \"PLA2G4E\", \n",
    "    \"PLA2G4B\", \"PLA2G4C\", \"RRAS2\", \"PLA2G4A\", \"NGF\", \"MRAS\", \"RASA1\", \"RASA2\", \n",
    "    \"KIT\", \"SOS1\", \"MET\", \"SOS2\", \"RELA\", \"EGFR\", \"INS\", \"NTF4\", \"CDC42\", \"MAPK9\", \n",
    "    \"NRAS\", \"PAK1\", \"MAPK8\", \"NTF3\", \"MAPK1\", \"PAK2\", \"MAPK3\", \"NTRK1\", \"NTRK2\", \n",
    "    \"ANGPT4\", \"ANGPT2\", \"ANGPT1\", \"EGF\", \"VEGFB\", \"VEGFC\", \"VEGFD\", \"VEGFA\", \n",
    "    \"MAPK10\", \"KITLG\", \"KRAS\", \"TEK\"]\n",
    "print(len(gene_list))\n",
    "\n",
    "\n",
    "\n",
    "[\"CSF1\", \"FGF1\", \"FGF2\", \"ELK1\", \"FGF3\", \"IGF1R\", \"FGF4\", \"FGF5\", \"FGF6\", \"FGF7\", \n",
    "\"FGF8\", \"FGF9\", \"PRKACG\", \"AKT2\", \"AKT3\", \"KDR\", \"AKT1\", \"PRKACA\", \"PRKACB\", \n",
    "\"PDGFRB\", \"PRKCG\", \"PDGFRA\", \"MAP2K1\", \"MAP2K2\", \"PRKCB\", \"HGF\", \"FLT3LG\", \n",
    "\"PRKCA\", \"PGF\", \"RAF1\", \"EPHA2\", \"CSF1R\", \"PDGFB\", \"PDGFA\", \"TGFA\", \"STK4\", \n",
    "\"EFNA5\", \"RASGRP2\", \"RASGRP1\", \"EFNA4\", \"RASGRP4\", \"RASGRP3\", \"RAP1B\", \"RAP1A\", \n",
    "\"RRAS\", \"PDGFD\", \"PDGFC\", \"FGF20\", \"FGF23\", \"FGF22\", \"FGF21\", \"NGFR\", \"BDNF\", \n",
    "\"INSR\", \"IGF2\", \"IGF1\", \"GNG12\", \"NFKB1\", \"FGF17\", \"EFNA1\", \"FGF16\", \"EFNA3\", \n",
    "\"EFNA2\", \"FGF19\", \"FGF18\", \"NF1\", \"GRB2\", \"FGFR4\", \"FGFR3\", \"FGFR2\", \"FGF10\", \n",
    "\"FGFR1\", \"FLT1\", \"FLT3\", \"FLT4\", \"RASGRF2\", \"RASGRF1\", \"FASLG\", \"IKBKB\", \"RAC2\", \n",
    "\"RAC3\", \"IKBKG\", \"RAC1\", \"HRAS\", \"PLA2G4F\", \"PLA2G4D\", \"CHUK\", \"PLA2G4E\", \n",
    "\"PLA2G4B\", \"PLA2G4C\", \"RRAS2\", \"PLA2G4A\", \"NGF\", \"MRAS\", \"RASA1\", \"RASA2\", \n",
    "\"KIT\", \"SOS1\", \"MET\", \"SOS2\", \"RELA\", \"EGFR\", \"INS\", \"NTF4\", \"CDC42\", \"MAPK9\", \n",
    "\"NRAS\", \"PAK1\", \"MAPK8\", \"NTF3\", \"MAPK1\", \"PAK2\", \"MAPK3\", \"NTRK1\", \"NTRK2\", \n",
    "\"ANGPT4\", \"ANGPT2\", \"ANGPT1\", \"EGF\", \"VEGFB\", \"VEGFC\", \"VEGFD\", \"VEGFA\", \n",
    "\"MAPK10\", \"KITLG\", \"KRAS\", \"TEK\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8bf65a87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>Matched_Genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EPHA2, EPHA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16124918</td>\n",
       "      <td>rs1803527</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=23;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.09425,0...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>129</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.2146E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EPHA2, EPHA2, EPHA2, EPHA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16132319</td>\n",
       "      <td>rs2291805</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8726,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>30</td>\n",
       "      <td>8</td>\n",
       "      <td>21.05%</td>\n",
       "      <td>2.5935E-3</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EPHA2, EPHA2, EPHA2, EPHA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61808</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154556836</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>40%</td>\n",
       "      <td>1.638E-3</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61809</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61809</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154558768</td>\n",
       "      <td>rs55810978</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=8;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=IKBKG...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>41</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>100%</td>\n",
       "      <td>7.77E-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61810</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61810</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154563953</td>\n",
       "      <td>rs201709278;rs5945206</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=13;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148E-8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>758 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS                   rsID  REF  ALT QUAL FILTER  \\\n",
       "866     NaN        NaN                    NaN  NaN  NaN  NaN    NaN   \n",
       "866    chr1   16124918              rs1803527    C    T    .   PASS   \n",
       "867     NaN        NaN                    NaN  NaN  NaN  NaN    NaN   \n",
       "867    chr1   16132319              rs2291805    C    T    .   PASS   \n",
       "868     NaN        NaN                    NaN  NaN  NaN  NaN    NaN   \n",
       "...     ...        ...                    ...  ...  ...  ...    ...   \n",
       "61808  chrX  154556836                      .    C    T    .   PASS   \n",
       "61809   NaN        NaN                    NaN  NaN  NaN  NaN    NaN   \n",
       "61809  chrX  154558768             rs55810978    G    C    .   PASS   \n",
       "61810   NaN        NaN                    NaN  NaN  NaN  NaN    NaN   \n",
       "61810  chrX  154563953  rs201709278;rs5945206    C    T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "866                                                  NaN  NaN  NaN  NaN  NaN   \n",
       "866    ADP=23;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.09425,0...  1/1  129   23   23   \n",
       "867                                                  NaN  NaN  NaN  NaN  NaN   \n",
       "867    ADP=38;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8726,0....  0/1   25   38   38   \n",
       "868                                                  NaN  NaN  NaN  NaN  NaN   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "61808  ADP=20;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...  0/1   27   20   20   \n",
       "61809                                                NaN  NaN  NaN  NaN  NaN   \n",
       "61809  ADP=8;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=IKBKG...  1/1   41    8    8   \n",
       "61810                                                NaN  NaN  NaN  NaN  NaN   \n",
       "61810  ADP=13;WT=0;HET=0;HOM=1;NC=0;ASP;G5;G5A;GENEIN...  1/1   70   13   13   \n",
       "\n",
       "        RD   AD    FREQ        PVAL  RDF  RDR  ADF  ADR  \\\n",
       "866    NaN  NaN     NaN         NaN  NaN  NaN  NaN  NaN   \n",
       "866      0   23    100%  1.2146E-13    0    0   16    7   \n",
       "867    NaN  NaN     NaN         NaN  NaN  NaN  NaN  NaN   \n",
       "867     30    8  21.05%   2.5935E-3   18   12    7    1   \n",
       "868    NaN  NaN     NaN         NaN  NaN  NaN  NaN  NaN   \n",
       "...    ...  ...     ...         ...  ...  ...  ...  ...   \n",
       "61808   12    8     40%    1.638E-3    9    3    4    4   \n",
       "61809  NaN  NaN     NaN         NaN  NaN  NaN  NaN  NaN   \n",
       "61809    0    8    100%     7.77E-5    0    0    1    7   \n",
       "61810  NaN  NaN     NaN         NaN  NaN  NaN  NaN  NaN   \n",
       "61810    0   13    100%   9.6148E-8    0    0    9    4   \n",
       "\n",
       "                                           Matched_Genes  \n",
       "866                                         EPHA2, EPHA2  \n",
       "866                                                  NaN  \n",
       "867                           EPHA2, EPHA2, EPHA2, EPHA2  \n",
       "867                                                  NaN  \n",
       "868                           EPHA2, EPHA2, EPHA2, EPHA2  \n",
       "...                                                  ...  \n",
       "61808                                                NaN  \n",
       "61809  IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBK...  \n",
       "61809                                                NaN  \n",
       "61810  IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBKG, IKBK...  \n",
       "61810                                                NaN  \n",
       "\n",
       "[758 rows x 21 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create a list of genes you want to search for\n",
    "gene_list = [\"CSF1\", \"FGF1\", \"FGF2\", \"ELK1\", \"FGF3\", \"IGF1R\", \"FGF4\", \"FGF5\", \"FGF6\", \"FGF7\", \n",
    "             \"FGF8\", \"FGF9\", \"PRKACG\", \"AKT2\", \"AKT3\", \"KDR\", \"AKT1\", \"PRKACA\", \"PRKACB\", \n",
    "             \"PDGFRB\", \"PRKCG\", \"PDGFRA\", \"MAP2K1\", \"MAP2K2\", \"PRKCB\", \"HGF\", \"FLT3LG\", \n",
    "             \"PRKCA\", \"PGF\", \"RAF1\", \"EPHA2\", \"CSF1R\", \"PDGFB\", \"PDGFA\", \"TGFA\", \"STK4\", \n",
    "             \"EFNA5\", \"RASGRP2\", \"RASGRP1\", \"EFNA4\", \"RASGRP4\", \"RASGRP3\", \"RAP1B\", \"RAP1A\", \n",
    "             \"RRAS\", \"PDGFD\", \"PDGFC\", \"FGF20\", \"FGF23\", \"FGF22\", \"FGF21\", \"NGFR\", \"BDNF\", \n",
    "             \"INSR\", \"IGF2\", \"IGF1\", \"GNG12\", \"NFKB1\", \"FGF17\", \"EFNA1\", \"FGF16\", \"EFNA3\", \n",
    "             \"EFNA2\", \"FGF19\", \"FGF18\", \"NF1\", \"GRB2\", \"FGFR4\", \"FGFR3\", \"FGFR2\", \"FGF10\", \n",
    "             \"FGFR1\", \"FLT1\", \"FLT3\", \"FLT4\", \"RASGRF2\", \"RASGRF1\", \"FASLG\", \"IKBKB\", \"RAC2\", \n",
    "             \"RAC3\", \"IKBKG\", \"RAC1\", \"HRAS\", \"PLA2G4F\", \"PLA2G4D\", \"CHUK\", \"PLA2G4E\", \n",
    "             \"PLA2G4B\", \"PLA2G4C\", \"RRAS2\", \"PLA2G4A\", \"NGF\", \"MRAS\", \"RASA1\", \"RASA2\", \n",
    "             \"KIT\", \"SOS1\", \"MET\", \"SOS2\", \"RELA\", \"EGFR\", \"INS\", \"NTF4\", \"CDC42\", \"MAPK9\", \n",
    "             \"NRAS\", \"PAK1\", \"MAPK8\", \"NTF3\", \"MAPK1\", \"PAK2\", \"MAPK3\", \"NTRK1\", \"NTRK2\", \n",
    "             \"ANGPT4\", \"ANGPT2\", \"ANGPT1\", \"EGF\", \"VEGFB\", \"VEGFC\", \"VEGFD\", \"VEGFA\", \n",
    "             \"MAPK10\", \"KITLG\", \"KRAS\", \"TEK\"]\n",
    "\n",
    "# Create a new DataFrame to store the matching rows\n",
    "matched_rows = pd.DataFrame(columns=vcf_data_df.columns)  # Create an empty DataFrame with the same columns\n",
    "\n",
    "# Create a new column to store matched genes\n",
    "matched_rows[\"Matched_Genes\"] = \"\"\n",
    "\n",
    "# Create a regex pattern to match genes within any separator\n",
    "gene_pattern = r'\\b(' + '|'.join(gene_list) + r')\\b'\n",
    "\n",
    "# Iterate through each row in the data\n",
    "for index, row in vcf_data_df.iterrows():\n",
    "    info = row[\"INFO\"]\n",
    "    \n",
    "    # Use regex to find all matched genes within the INFO column\n",
    "    matched_genes = re.findall(gene_pattern, info, re.IGNORECASE)\n",
    "    \n",
    "    if matched_genes:\n",
    "        # Update the \"Matched_Genes\" column with matched genes\n",
    "        matched_rows.at[index, \"Matched_Genes\"] = ', '.join(matched_genes)\n",
    "        matched_rows = pd.concat([matched_rows, vcf_data_df.loc[[index]]])\n",
    "        \n",
    "matched_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2b16a0db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "379.0"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "758/2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "8f3555d6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ADP=9;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.003195,0.9968;COMMON=1;GENEINFO=CSF1:1435;GNO;HD;INT;KGPhase1;KGPhase3;RS=333972;RSPOS=109925123;SAO=0;SLO;SSR=0;TOPMED=0.00305810397553516,0.99694189602446483;VC=SNV;VLD;VP=0x05010008000504053e000100;WGT=1;dbSNPBuildID=79;CSQ=G|intron_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000329608|protein_coding||7/8||||||||||1||HGNC|HGNC:2432|||||,G|intron_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000369801|protein_coding||8/8||||||||||1||HGNC|HGNC:2432|||||,G|intron_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000369802|protein_coding||7/8||||||||||1||HGNC|HGNC:2432|||||,G|intron_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000420111|protein_coding||7/8||||||||||1||HGNC|HGNC:2432|||||,G|downstream_gene_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000488198|protein_coding|||||||||||1524|1|cds_end_NF|HGNC|HGNC:2432|||||,G|downstream_gene_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000525659|protein_coding|||||||||||3210|1|cds_end_NF|HGNC|HGNC:2432|||||,G|downstream_gene_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000526001|processed_transcript|||||||||||2574|1||HGNC|HGNC:2432|||||,G|downstream_gene_variant|MODIFIER|CSF1|ENSG00000184371|Transcript|ENST00000527192|protein_coding|||||||||||3243|1|cds_end_NF|HGNC|HGNC:2432|||||'"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matched_rows.INFO.iloc[3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "b9ffb9c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]    61976\n",
       "Name: Matching_Genes, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_data_df.Matching_Genes.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "12f6720d",
   "metadata": {},
   "outputs": [],
   "source": [
    "matched_rows.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/220_output.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01536a2c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "7ae7c8fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Citatation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs738409</td>\n",
       "      <td>PMID: 32333362, PMID: 27278285, PMID: 26409295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs4823173</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs2896019</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs3810622</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs2281135</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs12483959</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs143392071</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TM6SF2</td>\n",
       "      <td>rs58542926</td>\n",
       "      <td>PMID: 26807382, PMID: 36353245, PMID: 31713012...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>PMID: 32383295,PMID: 34535985, PMID: 34841290,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs780094</td>\n",
       "      <td>PMID: 37711901, PMID: 33243135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs780093</td>\n",
       "      <td>PMID: 28298293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HSD17B13</td>\n",
       "      <td>rs72613567</td>\n",
       "      <td>PMID: 32145256, PMID: 33363067, PMID: 33102799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HSD17B13</td>\n",
       "      <td>rs62305723</td>\n",
       "      <td>PMID: 32145256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HSD17B13</td>\n",
       "      <td>rs143404524</td>\n",
       "      <td>PMID: 32145256</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene         rsID                                         Citatation\n",
       "0     PNPLA3     rs738409     PMID: 32333362, PMID: 27278285, PMID: 26409295\n",
       "1     PNPLA3    rs4823173                                     PMID: 36000237\n",
       "2     PNPLA3    rs2896019                                     PMID: 36000237\n",
       "3     PNPLA3    rs3810622                                     PMID: 36000237\n",
       "4     PNPLA3    rs2281135                                     PMID: 36000237\n",
       "5     PNPLA3   rs12483959                                     PMID: 36000237\n",
       "6     PNPLA3  rs143392071                                     PMID: 36000237\n",
       "7     PNPLA3    rs2143571                                     PMID: 36000237\n",
       "8     TM6SF2   rs58542926  PMID: 26807382, PMID: 36353245, PMID: 31713012...\n",
       "9       GCKR    rs1260326  PMID: 32383295,PMID: 34535985, PMID: 34841290,...\n",
       "10      GCKR     rs780094                     PMID: 37711901, PMID: 33243135\n",
       "11      GCKR     rs780093                                     PMID: 28298293\n",
       "12  HSD17B13   rs72613567     PMID: 32145256, PMID: 33363067, PMID: 33102799\n",
       "13  HSD17B13   rs62305723                                     PMID: 32145256\n",
       "14  HSD17B13  rs143404524                                     PMID: 32145256"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Fatty_Liver.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2c0b654d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Covered/Not_covered</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751.0</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.633100</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.510800</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.6260</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.906200</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751.0</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.633100</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.510800</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.6260</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.906200</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.633100</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.510800</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.6260</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.906200</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.387500</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.387500</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name       rsID Covered/Not_covered  CHROM       POS REF ALT  \\\n",
       "0        GCKR  rs1260326             Covered   chr2  27508073   T   C   \n",
       "1        GCKR  rs1260326             Covered   chr2  27508073   T   G   \n",
       "2        GCKR  rs1260326             Covered   chr2  27508073   T   C   \n",
       "3        GCKR  rs1260326             Covered   chr2  27508073   T   G   \n",
       "4        GCKR  rs1260326             Covered   chr2  27508073   T   C   \n",
       "..        ...        ...                 ...    ...       ...  ..  ..   \n",
       "105    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   C   \n",
       "106    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   A   \n",
       "107    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   C   \n",
       "108    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   A   \n",
       "109    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   C   \n",
       "\n",
       "                                      Consequence Consequence_score    IMPACT  \\\n",
       "0          missense variant,splice region variant              7/10  MODERATE   \n",
       "1          missense variant,splice region variant              7/10  MODERATE   \n",
       "2          missense variant,splice region variant              7/10  MODERATE   \n",
       "3          missense variant,splice region variant              7/10  MODERATE   \n",
       "4                         downstream gene variant              2/10  MODIFIER   \n",
       "..                                            ...               ...       ...   \n",
       "105  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "106  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "107  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "108                     regulatory region variant              2/10  MODIFIER   \n",
       "109                     regulatory region variant              2/10  MODIFIER   \n",
       "\n",
       "     IMPACT_score                                      ClinVar_CLNDN  \\\n",
       "0             5.0  Fasting plasma glucose level quantitative trai...   \n",
       "1             5.0                                                NaN   \n",
       "2             5.0  Fasting plasma glucose level quantitative trai...   \n",
       "3             5.0                                                NaN   \n",
       "4             1.5  Fasting plasma glucose level quantitative trai...   \n",
       "..            ...                                                ...   \n",
       "105           1.5                                                NaN   \n",
       "106           1.5                                                NaN   \n",
       "107           1.5                                                NaN   \n",
       "108           1.5                                                NaN   \n",
       "109           1.5                                                NaN   \n",
       "\n",
       "               CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "0    benign,association  criteria provided, multiple submitters, no con...   \n",
       "1                   NaN                                                NaN   \n",
       "2    benign,association  criteria provided, multiple submitters, no con...   \n",
       "3                   NaN                                                NaN   \n",
       "4    benign,association  criteria provided, multiple submitters, no con...   \n",
       "..                  ...                                                ...   \n",
       "105                 NaN                                                NaN   \n",
       "106                 NaN                                                NaN   \n",
       "107                 NaN                                                NaN   \n",
       "108                 NaN                                                NaN   \n",
       "109                 NaN                                                NaN   \n",
       "\n",
       "     ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0     8751.0  ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "1        NaN  ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "2     8751.0  ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "3        NaN  ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "4     8751.0                NaN               None                NaN   \n",
       "..       ...                ...                ...                ...   \n",
       "105      NaN  ENST00000493621.1       n.100+316G>C                NaN   \n",
       "106      NaN  ENST00000494795.1      n.3057-532G>A                NaN   \n",
       "107      NaN  ENST00000494795.1      n.3057-532G>C                NaN   \n",
       "108      NaN                NaN               None                NaN   \n",
       "109      NaN                NaN               None                NaN   \n",
       "\n",
       "    HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  \\\n",
       "0          p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062   \n",
       "1          p.Leu446Arg  tolerated(0.06)      benign(0)     NaN     NaN   \n",
       "2          p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062   \n",
       "3          p.Leu147Arg  tolerated(0.06)  benign(0.286)     NaN     NaN   \n",
       "4                 None              NaN            NaN  0.7067  0.9062   \n",
       "..                 ...              ...            ...     ...     ...   \n",
       "105               None              NaN            NaN     NaN     NaN   \n",
       "106               None              NaN            NaN  0.3075  0.3797   \n",
       "107               None              NaN            NaN     NaN     NaN   \n",
       "108               None              NaN            NaN  0.3075  0.3797   \n",
       "109               None              NaN            NaN     NaN     NaN   \n",
       "\n",
       "     AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  gnomADe_AFR_AF  \\\n",
       "0    0.6383  0.5188  0.5895  0.7996    0.633100          0.8679   \n",
       "1       NaN     NaN     NaN     NaN    0.000004          0.0000   \n",
       "2    0.6383  0.5188  0.5895  0.7996    0.633100          0.8679   \n",
       "3       NaN     NaN     NaN     NaN    0.000004          0.0000   \n",
       "4    0.6383  0.5188  0.5895  0.7996    0.633100          0.8679   \n",
       "..      ...     ...     ...     ...         ...             ...   \n",
       "105     NaN     NaN     NaN     NaN         NaN             NaN   \n",
       "106  0.3415  0.3661  0.2087  0.2270         NaN             NaN   \n",
       "107     NaN     NaN     NaN     NaN         NaN             NaN   \n",
       "108  0.3415  0.3661  0.2087  0.2270         NaN             NaN   \n",
       "109     NaN     NaN     NaN     NaN         NaN             NaN   \n",
       "\n",
       "     gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  \\\n",
       "0            0.6662          0.4668        0.510800          0.6426   \n",
       "1            0.0000          0.0000        0.000054          0.0000   \n",
       "2            0.6662          0.4668        0.510800          0.6426   \n",
       "3            0.0000          0.0000        0.000054          0.0000   \n",
       "4            0.6662          0.4668        0.510800          0.6426   \n",
       "..              ...             ...             ...             ...   \n",
       "105             NaN             NaN             NaN             NaN   \n",
       "106             NaN             NaN             NaN             NaN   \n",
       "107             NaN             NaN             NaN             NaN   \n",
       "108             NaN             NaN             NaN             NaN   \n",
       "109             NaN             NaN             NaN             NaN   \n",
       "\n",
       "     gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  \\\n",
       "0            0.5902          0.6005           0.758      0.6700   \n",
       "1            0.0000          0.0000           0.000         NaN   \n",
       "2            0.5902          0.6005           0.758      0.6700   \n",
       "3            0.0000          0.0000           0.000         NaN   \n",
       "4            0.5902          0.6005           0.758      0.6700   \n",
       "..              ...             ...             ...         ...   \n",
       "105             NaN             NaN             NaN         NaN   \n",
       "106             NaN             NaN             NaN      0.2519   \n",
       "107             NaN             NaN             NaN         NaN   \n",
       "108             NaN             NaN             NaN      0.2519   \n",
       "109             NaN             NaN             NaN         NaN   \n",
       "\n",
       "     gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  \\\n",
       "0            0.8582          0.7083          0.6260          0.4585   \n",
       "1               NaN             NaN             NaN             NaN   \n",
       "2            0.8582          0.7083          0.6260          0.4585   \n",
       "3               NaN             NaN             NaN             NaN   \n",
       "4            0.8582          0.7083          0.6260          0.4585   \n",
       "..              ...             ...             ...             ...   \n",
       "105             NaN             NaN             NaN             NaN   \n",
       "106          0.3536          0.2325          0.3173          0.1317   \n",
       "107             NaN             NaN             NaN             NaN   \n",
       "108          0.3536          0.2325          0.3173          0.1317   \n",
       "109             NaN             NaN             NaN             NaN   \n",
       "\n",
       "     gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  \\\n",
       "0            0.5014          0.6378          0.4904          0.5898   \n",
       "1               NaN             NaN             NaN             NaN   \n",
       "2            0.5014          0.6378          0.4904          0.5898   \n",
       "3               NaN             NaN             NaN             NaN   \n",
       "4            0.5014          0.6378          0.4904          0.5898   \n",
       "..              ...             ...             ...             ...   \n",
       "105             NaN             NaN             NaN             NaN   \n",
       "106          0.3875          0.2268          0.2184          0.1789   \n",
       "107             NaN             NaN             NaN             NaN   \n",
       "108          0.3875          0.2268          0.2184          0.1789   \n",
       "109             NaN             NaN             NaN             NaN   \n",
       "\n",
       "     gnomADg_OTH_AF  gnomADg_SAS_AF    MAX_AF  MAX_AF_POPS  \\\n",
       "0            0.6106          0.7566  0.906200          AFR   \n",
       "1               NaN             NaN  0.000054  gnomADe EAS   \n",
       "2            0.6106          0.7566  0.906200          AFR   \n",
       "3               NaN             NaN  0.000054  gnomADe EAS   \n",
       "4            0.6106          0.7566  0.906200          AFR   \n",
       "..              ...             ...       ...          ...   \n",
       "105             NaN             NaN       NaN          NaN   \n",
       "106          0.2555          0.2035  0.387500  gnomADg EAS   \n",
       "107             NaN             NaN       NaN          NaN   \n",
       "108          0.2555          0.2035  0.387500  gnomADg EAS   \n",
       "109             NaN             NaN       NaN          NaN   \n",
       "\n",
       "                            BIOTYPE   EXON INTRON  \\\n",
       "0                    protein coding  15/19    NaN   \n",
       "1                    protein coding  15/19    NaN   \n",
       "2                    protein coding    5/7    NaN   \n",
       "3                    protein coding    5/7    NaN   \n",
       "4                   retained intron    NaN    NaN   \n",
       "..                              ...    ...    ...   \n",
       "105  protein coding CDS not defined    NaN    1/1   \n",
       "106                 retained intron    NaN    2/2   \n",
       "107                 retained intron    NaN    2/2   \n",
       "108                        promoter    NaN    NaN   \n",
       "109                        promoter    NaN    NaN   \n",
       "\n",
       "    Protein Position and Amino Acid   Codons  STRAND  \n",
       "0                             L446P  cTg/cCg     1.0  \n",
       "1                             L446R  cTg/cGg     1.0  \n",
       "2                             L147P  cTg/cCg     1.0  \n",
       "3                             L147R  cTg/cGg     1.0  \n",
       "4                               NaN      NaN     1.0  \n",
       "..                              ...      ...     ...  \n",
       "105                             NaN      NaN     1.0  \n",
       "106                             NaN      NaN     1.0  \n",
       "107                             NaN      NaN     1.0  \n",
       "108                             NaN      NaN     NaN  \n",
       "109                             NaN      NaN     NaN  \n",
       "\n",
       "[110 rows x 55 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/Fatty_liver_data.xlsx')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "a612219e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Covered/Not_covered</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Citatation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751.0</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;C</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.633100</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.510800</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.6260</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.906200</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L446P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>PMID: 32383295,PMID: 34535985, PMID: 34841290,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000264717.7</td>\n",
       "      <td>c.1337T&gt;G</td>\n",
       "      <td>ENSP00000264717.2</td>\n",
       "      <td>p.Leu446Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>15/19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L446R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>PMID: 32383295,PMID: 34535985, PMID: 34841290,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751.0</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;C</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Pro</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.633100</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.510800</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.6260</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.906200</td>\n",
       "      <td>AFR</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L147P</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>PMID: 32383295,PMID: 34535985, PMID: 34841290,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>missense variant,splice region variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000411584.1</td>\n",
       "      <td>c.440T&gt;G</td>\n",
       "      <td>ENSP00000416917.1</td>\n",
       "      <td>p.Leu147Arg</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.286)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>gnomADe EAS</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L147R</td>\n",
       "      <td>cTg/cGg</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>PMID: 32383295,PMID: 34535985, PMID: 34841290,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCKR</td>\n",
       "      <td>rs1260326</td>\n",
       "      <td>Covered</td>\n",
       "      <td>chr2</td>\n",
       "      <td>27508073</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Fasting plasma glucose level quantitative trai...</td>\n",
       "      <td>benign,association</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>8751.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.9062</td>\n",
       "      <td>0.6383</td>\n",
       "      <td>0.5188</td>\n",
       "      <td>0.5895</td>\n",
       "      <td>0.7996</td>\n",
       "      <td>0.633100</td>\n",
       "      <td>0.8679</td>\n",
       "      <td>0.6662</td>\n",
       "      <td>0.4668</td>\n",
       "      <td>0.510800</td>\n",
       "      <td>0.6426</td>\n",
       "      <td>0.5902</td>\n",
       "      <td>0.6005</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.8582</td>\n",
       "      <td>0.7083</td>\n",
       "      <td>0.6260</td>\n",
       "      <td>0.4585</td>\n",
       "      <td>0.5014</td>\n",
       "      <td>0.6378</td>\n",
       "      <td>0.4904</td>\n",
       "      <td>0.5898</td>\n",
       "      <td>0.6106</td>\n",
       "      <td>0.7566</td>\n",
       "      <td>0.906200</td>\n",
       "      <td>AFR</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCKR</td>\n",
       "      <td>PMID: 32383295,PMID: 34535985, PMID: 34841290,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000493621.1</td>\n",
       "      <td>n.100+316G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.387500</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000494795.1</td>\n",
       "      <td>n.3057-532G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3075</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.3661</td>\n",
       "      <td>0.2087</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2519</td>\n",
       "      <td>0.3536</td>\n",
       "      <td>0.2325</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.1317</td>\n",
       "      <td>0.3875</td>\n",
       "      <td>0.2268</td>\n",
       "      <td>0.2184</td>\n",
       "      <td>0.1789</td>\n",
       "      <td>0.2555</td>\n",
       "      <td>0.2035</td>\n",
       "      <td>0.387500</td>\n",
       "      <td>gnomADg EAS</td>\n",
       "      <td>promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>SAMM50</td>\n",
       "      <td>rs2143571</td>\n",
       "      <td>Not_Covered</td>\n",
       "      <td>chr22</td>\n",
       "      <td>43995806</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>PMID: 36000237</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows Ã— 57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name       rsID Covered/Not_covered  CHROM       POS REF ALT  \\\n",
       "0        GCKR  rs1260326             Covered   chr2  27508073   T   C   \n",
       "1        GCKR  rs1260326             Covered   chr2  27508073   T   G   \n",
       "2        GCKR  rs1260326             Covered   chr2  27508073   T   C   \n",
       "3        GCKR  rs1260326             Covered   chr2  27508073   T   G   \n",
       "4        GCKR  rs1260326             Covered   chr2  27508073   T   C   \n",
       "..        ...        ...                 ...    ...       ...  ..  ..   \n",
       "105    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   C   \n",
       "106    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   A   \n",
       "107    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   C   \n",
       "108    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   A   \n",
       "109    SAMM50  rs2143571         Not_Covered  chr22  43995806   G   C   \n",
       "\n",
       "                                      Consequence Consequence_score    IMPACT  \\\n",
       "0          missense variant,splice region variant              7/10  MODERATE   \n",
       "1          missense variant,splice region variant              7/10  MODERATE   \n",
       "2          missense variant,splice region variant              7/10  MODERATE   \n",
       "3          missense variant,splice region variant              7/10  MODERATE   \n",
       "4                         downstream gene variant              2/10  MODIFIER   \n",
       "..                                            ...               ...       ...   \n",
       "105  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "106  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "107  intron variant,non coding transcript variant              2/10  MODIFIER   \n",
       "108                     regulatory region variant              2/10  MODIFIER   \n",
       "109                     regulatory region variant              2/10  MODIFIER   \n",
       "\n",
       "     IMPACT_score                                      ClinVar_CLNDN  \\\n",
       "0             5.0  Fasting plasma glucose level quantitative trai...   \n",
       "1             5.0                                                NaN   \n",
       "2             5.0  Fasting plasma glucose level quantitative trai...   \n",
       "3             5.0                                                NaN   \n",
       "4             1.5  Fasting plasma glucose level quantitative trai...   \n",
       "..            ...                                                ...   \n",
       "105           1.5                                                NaN   \n",
       "106           1.5                                                NaN   \n",
       "107           1.5                                                NaN   \n",
       "108           1.5                                                NaN   \n",
       "109           1.5                                                NaN   \n",
       "\n",
       "               CLIN_SIG                                 ClinVar_CLNREVSTAT  \\\n",
       "0    benign,association  criteria provided, multiple submitters, no con...   \n",
       "1                   NaN                                                NaN   \n",
       "2    benign,association  criteria provided, multiple submitters, no con...   \n",
       "3                   NaN                                                NaN   \n",
       "4    benign,association  criteria provided, multiple submitters, no con...   \n",
       "..                  ...                                                ...   \n",
       "105                 NaN                                                NaN   \n",
       "106                 NaN                                                NaN   \n",
       "107                 NaN                                                NaN   \n",
       "108                 NaN                                                NaN   \n",
       "109                 NaN                                                NaN   \n",
       "\n",
       "     ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0     8751.0  ENST00000264717.7          c.1337T>C  ENSP00000264717.2   \n",
       "1        NaN  ENST00000264717.7          c.1337T>G  ENSP00000264717.2   \n",
       "2     8751.0  ENST00000411584.1           c.440T>C  ENSP00000416917.1   \n",
       "3        NaN  ENST00000411584.1           c.440T>G  ENSP00000416917.1   \n",
       "4     8751.0                NaN               None                NaN   \n",
       "..       ...                ...                ...                ...   \n",
       "105      NaN  ENST00000493621.1       n.100+316G>C                NaN   \n",
       "106      NaN  ENST00000494795.1      n.3057-532G>A                NaN   \n",
       "107      NaN  ENST00000494795.1      n.3057-532G>C                NaN   \n",
       "108      NaN                NaN               None                NaN   \n",
       "109      NaN                NaN               None                NaN   \n",
       "\n",
       "    HGVSp (Transcript)             SIFT       PolyPhen      AF  AFR_AF  \\\n",
       "0          p.Leu446Pro     tolerated(1)      benign(0)  0.7067  0.9062   \n",
       "1          p.Leu446Arg  tolerated(0.06)      benign(0)     NaN     NaN   \n",
       "2          p.Leu147Pro     tolerated(1)  benign(0.001)  0.7067  0.9062   \n",
       "3          p.Leu147Arg  tolerated(0.06)  benign(0.286)     NaN     NaN   \n",
       "4                 None              NaN            NaN  0.7067  0.9062   \n",
       "..                 ...              ...            ...     ...     ...   \n",
       "105               None              NaN            NaN     NaN     NaN   \n",
       "106               None              NaN            NaN  0.3075  0.3797   \n",
       "107               None              NaN            NaN     NaN     NaN   \n",
       "108               None              NaN            NaN  0.3075  0.3797   \n",
       "109               None              NaN            NaN     NaN     NaN   \n",
       "\n",
       "     AMR_AF  EAS_AF  EUR_AF  SAS_AF  gnomADe_AF  gnomADe_AFR_AF  \\\n",
       "0    0.6383  0.5188  0.5895  0.7996    0.633100          0.8679   \n",
       "1       NaN     NaN     NaN     NaN    0.000004          0.0000   \n",
       "2    0.6383  0.5188  0.5895  0.7996    0.633100          0.8679   \n",
       "3       NaN     NaN     NaN     NaN    0.000004          0.0000   \n",
       "4    0.6383  0.5188  0.5895  0.7996    0.633100          0.8679   \n",
       "..      ...     ...     ...     ...         ...             ...   \n",
       "105     NaN     NaN     NaN     NaN         NaN             NaN   \n",
       "106  0.3415  0.3661  0.2087  0.2270         NaN             NaN   \n",
       "107     NaN     NaN     NaN     NaN         NaN             NaN   \n",
       "108  0.3415  0.3661  0.2087  0.2270         NaN             NaN   \n",
       "109     NaN     NaN     NaN     NaN         NaN             NaN   \n",
       "\n",
       "     gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  \\\n",
       "0            0.6662          0.4668        0.510800          0.6426   \n",
       "1            0.0000          0.0000        0.000054          0.0000   \n",
       "2            0.6662          0.4668        0.510800          0.6426   \n",
       "3            0.0000          0.0000        0.000054          0.0000   \n",
       "4            0.6662          0.4668        0.510800          0.6426   \n",
       "..              ...             ...             ...             ...   \n",
       "105             NaN             NaN             NaN             NaN   \n",
       "106             NaN             NaN             NaN             NaN   \n",
       "107             NaN             NaN             NaN             NaN   \n",
       "108             NaN             NaN             NaN             NaN   \n",
       "109             NaN             NaN             NaN             NaN   \n",
       "\n",
       "     gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  \\\n",
       "0            0.5902          0.6005           0.758      0.6700   \n",
       "1            0.0000          0.0000           0.000         NaN   \n",
       "2            0.5902          0.6005           0.758      0.6700   \n",
       "3            0.0000          0.0000           0.000         NaN   \n",
       "4            0.5902          0.6005           0.758      0.6700   \n",
       "..              ...             ...             ...         ...   \n",
       "105             NaN             NaN             NaN         NaN   \n",
       "106             NaN             NaN             NaN      0.2519   \n",
       "107             NaN             NaN             NaN         NaN   \n",
       "108             NaN             NaN             NaN      0.2519   \n",
       "109             NaN             NaN             NaN         NaN   \n",
       "\n",
       "     gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  \\\n",
       "0            0.8582          0.7083          0.6260          0.4585   \n",
       "1               NaN             NaN             NaN             NaN   \n",
       "2            0.8582          0.7083          0.6260          0.4585   \n",
       "3               NaN             NaN             NaN             NaN   \n",
       "4            0.8582          0.7083          0.6260          0.4585   \n",
       "..              ...             ...             ...             ...   \n",
       "105             NaN             NaN             NaN             NaN   \n",
       "106          0.3536          0.2325          0.3173          0.1317   \n",
       "107             NaN             NaN             NaN             NaN   \n",
       "108          0.3536          0.2325          0.3173          0.1317   \n",
       "109             NaN             NaN             NaN             NaN   \n",
       "\n",
       "     gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  \\\n",
       "0            0.5014          0.6378          0.4904          0.5898   \n",
       "1               NaN             NaN             NaN             NaN   \n",
       "2            0.5014          0.6378          0.4904          0.5898   \n",
       "3               NaN             NaN             NaN             NaN   \n",
       "4            0.5014          0.6378          0.4904          0.5898   \n",
       "..              ...             ...             ...             ...   \n",
       "105             NaN             NaN             NaN             NaN   \n",
       "106          0.3875          0.2268          0.2184          0.1789   \n",
       "107             NaN             NaN             NaN             NaN   \n",
       "108          0.3875          0.2268          0.2184          0.1789   \n",
       "109             NaN             NaN             NaN             NaN   \n",
       "\n",
       "     gnomADg_OTH_AF  gnomADg_SAS_AF    MAX_AF  MAX_AF_POPS  \\\n",
       "0            0.6106          0.7566  0.906200          AFR   \n",
       "1               NaN             NaN  0.000054  gnomADe EAS   \n",
       "2            0.6106          0.7566  0.906200          AFR   \n",
       "3               NaN             NaN  0.000054  gnomADe EAS   \n",
       "4            0.6106          0.7566  0.906200          AFR   \n",
       "..              ...             ...       ...          ...   \n",
       "105             NaN             NaN       NaN          NaN   \n",
       "106          0.2555          0.2035  0.387500  gnomADg EAS   \n",
       "107             NaN             NaN       NaN          NaN   \n",
       "108          0.2555          0.2035  0.387500  gnomADg EAS   \n",
       "109             NaN             NaN       NaN          NaN   \n",
       "\n",
       "                            BIOTYPE   EXON INTRON  \\\n",
       "0                    protein coding  15/19    NaN   \n",
       "1                    protein coding  15/19    NaN   \n",
       "2                    protein coding    5/7    NaN   \n",
       "3                    protein coding    5/7    NaN   \n",
       "4                   retained intron    NaN    NaN   \n",
       "..                              ...    ...    ...   \n",
       "105  protein coding CDS not defined    NaN    1/1   \n",
       "106                 retained intron    NaN    2/2   \n",
       "107                 retained intron    NaN    2/2   \n",
       "108                        promoter    NaN    NaN   \n",
       "109                        promoter    NaN    NaN   \n",
       "\n",
       "    Protein Position and Amino Acid   Codons  STRAND    Gene  \\\n",
       "0                             L446P  cTg/cCg     1.0    GCKR   \n",
       "1                             L446R  cTg/cGg     1.0    GCKR   \n",
       "2                             L147P  cTg/cCg     1.0    GCKR   \n",
       "3                             L147R  cTg/cGg     1.0    GCKR   \n",
       "4                               NaN      NaN     1.0    GCKR   \n",
       "..                              ...      ...     ...     ...   \n",
       "105                             NaN      NaN     1.0  PNPLA3   \n",
       "106                             NaN      NaN     1.0  PNPLA3   \n",
       "107                             NaN      NaN     1.0  PNPLA3   \n",
       "108                             NaN      NaN     NaN  PNPLA3   \n",
       "109                             NaN      NaN     NaN  PNPLA3   \n",
       "\n",
       "                                            Citatation  \n",
       "0    PMID: 32383295,PMID: 34535985, PMID: 34841290,...  \n",
       "1    PMID: 32383295,PMID: 34535985, PMID: 34841290,...  \n",
       "2    PMID: 32383295,PMID: 34535985, PMID: 34841290,...  \n",
       "3    PMID: 32383295,PMID: 34535985, PMID: 34841290,...  \n",
       "4    PMID: 32383295,PMID: 34535985, PMID: 34841290,...  \n",
       "..                                                 ...  \n",
       "105                                     PMID: 36000237  \n",
       "106                                     PMID: 36000237  \n",
       "107                                     PMID: 36000237  \n",
       "108                                     PMID: 36000237  \n",
       "109                                     PMID: 36000237  \n",
       "\n",
       "[110 rows x 57 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.merge(data, df, on = 'rsID', how='left', sort=False)\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "b1c8bd8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/Fatty_liver_data.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "60a1f6c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Matched Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CTLA4</td>\n",
       "      <td>Type1 Daibetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INS</td>\n",
       "      <td>Type1 Daibetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>Type1 Daibetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HNF1A</td>\n",
       "      <td>Type1 Daibetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CEL</td>\n",
       "      <td>Type1 Daibetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>TRDN</td>\n",
       "      <td>Cardiac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>Cardiac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>PCSK9</td>\n",
       "      <td>Cardiac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>APOB</td>\n",
       "      <td>Cardiac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>STAP1</td>\n",
       "      <td>Cardiac</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>283 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name    Matched Gene\n",
       "0       CTLA4  Type1 Daibetes\n",
       "1         INS  Type1 Daibetes\n",
       "2       IL2RA  Type1 Daibetes\n",
       "3       HNF1A  Type1 Daibetes\n",
       "4         CEL  Type1 Daibetes\n",
       "..        ...             ...\n",
       "278      TRDN         Cardiac\n",
       "279   LDLRAP1         Cardiac\n",
       "280     PCSK9         Cardiac\n",
       "281      APOB         Cardiac\n",
       "282     STAP1         Cardiac\n",
       "\n",
       "[283 rows x 2 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/KHHSPTGPCSP22_genes.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "66c54585",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Matched Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Cholesterol, Cardiac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCB4</td>\n",
       "      <td>Cholesterol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCC8</td>\n",
       "      <td>Type2 Diabetes, MODY, Obesity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCC9</td>\n",
       "      <td>Cardiac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCG5</td>\n",
       "      <td>Cholesterol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>WNK1</td>\n",
       "      <td>Hypertension</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>WNK4</td>\n",
       "      <td>Hypertension</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>WRN</td>\n",
       "      <td>Type2 Diabetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>ZMPSTE24</td>\n",
       "      <td>Cholesterol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>Diabetic Neuropathy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>246 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Name                   Matched Gene\n",
       "0       ABCA1           Cholesterol, Cardiac\n",
       "1       ABCB4                    Cholesterol\n",
       "2       ABCC8  Type2 Diabetes, MODY, Obesity\n",
       "3       ABCC9                        Cardiac\n",
       "4       ABCG5                    Cholesterol\n",
       "..        ...                            ...\n",
       "241      WNK1                   Hypertension\n",
       "242      WNK4                   Hypertension\n",
       "243       WRN                 Type2 Diabetes\n",
       "244  ZMPSTE24                    Cholesterol\n",
       "245   ZSCAN20            Diabetic Neuropathy\n",
       "\n",
       "[246 rows x 2 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result = df.groupby('Gene Name')['Matched Gene'].agg(lambda x: ', '.join(x.unique())).reset_index()\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "91f2ae3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "result.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/KHHSPTGPCSP22_genes.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69b1b251",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
