{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f43cb518",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2c6ed7e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant_classification</th>\n",
       "      <th>Tool</th>\n",
       "      <th>Sample</th>\n",
       "      <th>VAF_Filter</th>\n",
       "      <th>DP_Filter</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Benign</td>\n",
       "      <td>BCFTOOL</td>\n",
       "      <td>12652705</td>\n",
       "      <td>0.100</td>\n",
       "      <td>10</td>\n",
       "      <td>25425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>BCFTOOL</td>\n",
       "      <td>12652705</td>\n",
       "      <td>0.100</td>\n",
       "      <td>10</td>\n",
       "      <td>3281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>BCFTOOL</td>\n",
       "      <td>12652705</td>\n",
       "      <td>0.100</td>\n",
       "      <td>10</td>\n",
       "      <td>673</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>BCFTOOL</td>\n",
       "      <td>12652705</td>\n",
       "      <td>0.100</td>\n",
       "      <td>10</td>\n",
       "      <td>480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VUS/Weak Benign</td>\n",
       "      <td>BCFTOOL</td>\n",
       "      <td>12652705</td>\n",
       "      <td>0.100</td>\n",
       "      <td>10</td>\n",
       "      <td>252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2995</th>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>DEEPVARIANT</td>\n",
       "      <td>12652712</td>\n",
       "      <td>0.001</td>\n",
       "      <td>30</td>\n",
       "      <td>825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2996</th>\n",
       "      <td>VUS/Weak Pathogenic</td>\n",
       "      <td>DEEPVARIANT</td>\n",
       "      <td>12652712</td>\n",
       "      <td>0.001</td>\n",
       "      <td>30</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2997</th>\n",
       "      <td>VUS/Weak Benign</td>\n",
       "      <td>DEEPVARIANT</td>\n",
       "      <td>12652712</td>\n",
       "      <td>0.001</td>\n",
       "      <td>30</td>\n",
       "      <td>227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2998</th>\n",
       "      <td>VUS</td>\n",
       "      <td>DEEPVARIANT</td>\n",
       "      <td>12652712</td>\n",
       "      <td>0.001</td>\n",
       "      <td>30</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2999</th>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>DEEPVARIANT</td>\n",
       "      <td>12652712</td>\n",
       "      <td>0.001</td>\n",
       "      <td>30</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3000 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Variant_classification         Tool    Sample  VAF_Filter DP_Filter  \\\n",
       "0                    Benign      BCFTOOL  12652705       0.100        10   \n",
       "1                      None      BCFTOOL  12652705       0.100        10   \n",
       "2             Likely Benign      BCFTOOL  12652705       0.100        10   \n",
       "3           VUS/Conflicting      BCFTOOL  12652705       0.100        10   \n",
       "4           VUS/Weak Benign      BCFTOOL  12652705       0.100        10   \n",
       "...                     ...          ...       ...         ...       ...   \n",
       "2995          Likely Benign  DEEPVARIANT  12652712       0.001        30   \n",
       "2996    VUS/Weak Pathogenic  DEEPVARIANT  12652712       0.001        30   \n",
       "2997        VUS/Weak Benign  DEEPVARIANT  12652712       0.001        30   \n",
       "2998                    VUS  DEEPVARIANT  12652712       0.001        30   \n",
       "2999      Likely Pathogenic  DEEPVARIANT  12652712       0.001        30   \n",
       "\n",
       "     Counts  \n",
       "0     25425  \n",
       "1      3281  \n",
       "2       673  \n",
       "3       480  \n",
       "4       252  \n",
       "...     ...  \n",
       "2995    825  \n",
       "2996    245  \n",
       "2997    227  \n",
       "2998     28  \n",
       "2999     13  \n",
       "\n",
       "[3000 rows x 6 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "truth_set_path = r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/Truthset_CHROM_POS_REF_ALT.tsv'\n",
    "truth_set = pd.read_csv(truth_set_path, sep='\\t', header=0, low_memory=False, encoding='latin-1')\n",
    "\n",
    "# Read your main data file\n",
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_new_05_10_2023/All_5samples.xlsx')\n",
    "df['CHROM'] = 'chr' + df['Chr:Pos'].str.split(':').str[0]\n",
    "df['POS'] = df['Chr:Pos'].str.split(':').str[1].astype(int)\n",
    "df['REF'] = df['Ref/Alt'].str.split('/').str[0]\n",
    "df['ALT'] = df['Ref/Alt'].str.split('/').str[1]\n",
    "columns_to_check = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "# Use drop_duplicates to remove duplicated rows based on the specified columns\n",
    "df = df.drop_duplicates(subset=columns_to_check)\n",
    "\n",
    "# Define the sample, tool, VAF filter, and DP filter combinations\n",
    "samples = ['12652705', '12652707', '12652709', '12652710', '12652712']\n",
    "tools = ['BCFTOOL', 'VARSCAN2', 'MUTECT2', 'HAPLOTYPECALLER', 'DEEPVARIANT']\n",
    "vaf_filters = [0.1, 0.01, 0.001]\n",
    "dp_filters = [10, 18, 20, 22, 30]\n",
    "\n",
    "# Create an empty DataFrame to store the results\n",
    "result_df = pd.DataFrame(columns=['Variant_classification', 'Tool', 'Sample', 'VAF_Filter', 'DP_Filter', 'Counts'])\n",
    "\n",
    "# Iterate through combinations and calculate counts\n",
    "for sample in samples:\n",
    "    for tool in tools:\n",
    "        for vaf_filter in vaf_filters:\n",
    "            for dp_filter in dp_filters:\n",
    "                # Read the corresponding VCF file for the tool\n",
    "                vcf_file_path = f'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/{sample}_{tool}.vcf'\n",
    "                vcf = pd.read_csv(vcf_file_path, comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "                vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "                # Extract VAF based on the tool\n",
    "                if tool == 'BCFTOOL':\n",
    "                    # Customize VAF extraction for BCFTOOL\n",
    "                    vcf['AD'] = vcf['SAMPLE'].str.split(':').str[6]\n",
    "                    vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "                    vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "                    vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "                    # For BCFTOOL, you can choose not to filter based on 'DP'\n",
    "                    filtered_data = vcf[vcf['VAF'] >= vaf_filter]\n",
    "                elif tool == 'VARSCAN2':\n",
    "                    # Customize VAF extraction for VARSCAN2\n",
    "                    vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "                    vcf['RD'] = vcf['SAMPLE'].str.split(':').str[4].fillna('0').astype(int)\n",
    "                    vcf['AD'] = vcf['SAMPLE'].str.split(':').str[5].fillna('0').astype(int)\n",
    "                    vcf['VAF'] = vcf['AD'] / (vcf['RD'] + vcf['AD'])\n",
    "                    # Apply VAF and DP filters\n",
    "                    filtered_data = vcf[(vcf['VAF'] >= vaf_filter) & (vcf['DP'] >= dp_filter)]\n",
    "                elif tool == 'MUTECT2':\n",
    "                    # Customize VAF extraction for MUTECT2\n",
    "                    vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "                    vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "                    vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "                    vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "                    vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "                    # For MUTECT2, you can choose not to filter based on 'DP'\n",
    "                    filtered_data = vcf[vcf['VAF'] >= vaf_filter]\n",
    "                elif tool == 'HAPLOTYPECALLER':\n",
    "                    # Customize VAF extraction for HAPLOTYPECALLER\n",
    "                    vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2].fillna('0').astype(int)\n",
    "                    vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "                    vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "                    vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "                    vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "                    # For HAPLOTYPECALLER, you can choose not to filter based on 'DP'\n",
    "                    filtered_data = vcf[vcf['VAF'] >= vaf_filter]\n",
    "                elif tool == 'DEEPVARIANT':\n",
    "                    # Customize VAF extraction for DEEPVARIANT\n",
    "                    vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2].fillna('0').astype(int)\n",
    "                    vcf['AD'] = vcf['SAMPLE'].str.split(':').str[3]\n",
    "                    vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "                    vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "                    vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "                    # For DEEPVARIANT, you can choose not to filter based on 'DP'\n",
    "                    filtered_data = vcf[vcf['VAF'] >= vaf_filter]\n",
    "\n",
    "                    \n",
    "                filtered_data = pd.merge(truth_set, filtered_data, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort = False)\n",
    "\n",
    "                # Merge the VCF data with the main data (df)\n",
    "                merged_data = pd.merge(filtered_data, df, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "                merged_data = merged_data.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "\n",
    "                # To fill NaN values in the 'Classification' column with 'NA', you need to assign the result back to the column.\n",
    "                merged_data['Classification'] = merged_data['Classification'].fillna('NA')\n",
    "\n",
    "                # List of your specified priorities\n",
    "                priorities = ['Pathogenic', 'Likely Pathogenic', 'VUS/Weak Pathogenic', 'VUS/Conflicting', 'VUS', 'VUS/Weak Benign', 'Likely Benign', 'Benign']\n",
    "\n",
    "                # Function to select the highest ranked value from Original_Column\n",
    "                def select_highest_ranked(row):\n",
    "                    values = row['Classification'].split(',')\n",
    "                    for rank in priorities:\n",
    "                        if rank in values:\n",
    "                            return rank\n",
    "                    return None  # Return None if no rank found\n",
    "\n",
    "                # Apply the function to create the final selected rank column\n",
    "                merged_data['Classification'] = merged_data.apply(select_highest_ranked, axis=1)\n",
    "                \n",
    "                # Calculate counts for each variant classification\n",
    "                counts = merged_data['Classification'].value_counts(dropna=False)\n",
    "\n",
    "                # Append the results to the result_df\n",
    "                for classification, count in counts.items():\n",
    "                    result_df = result_df.append({\n",
    "                        'Variant_classification': classification,\n",
    "                        'Tool': tool,\n",
    "                        'Sample': sample,\n",
    "                        'VAF_Filter': vaf_filter,\n",
    "                        'DP_Filter': dp_filter,\n",
    "                        'Counts': count\n",
    "                    }, ignore_index=True)\n",
    "\n",
    "# Reset index for the final result DataFrame\n",
    "result_df.reset_index(drop=True, inplace=True)\n",
    "\n",
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/True_positives_data_5samples_new_27_10_2023.xlsx', index = False)\n",
    "# Display the final result\n",
    "print(result_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "097bdf6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/True_positives_data_5samples_new_27_10_2023.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32bb2ce6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ebf2967",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be934180",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a8eec419",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65490</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65532</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65552</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65556</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761809</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194235</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761810</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194236</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761811</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194239</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761812</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194242</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761813</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194243</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23761814 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM       POS REF ALT\n",
       "0         chr1     65490   G   A\n",
       "1         chr1     65529   C   T\n",
       "2         chr1     65532   A   T\n",
       "3         chr1     65552   G   A\n",
       "4         chr1     65556   C   T\n",
       "...        ...       ...  ..  ..\n",
       "23761809  chrY  57194235   T   C\n",
       "23761810  chrY  57194236   G   C\n",
       "23761811  chrY  57194239   A   C\n",
       "23761812  chrY  57194242   G   T\n",
       "23761813  chrY  57194243   C   G\n",
       "\n",
       "[23761814 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "truth_set_path = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/output.tsv'\n",
    "truth_set = pd.read_csv(truth_set_path, sep='\\t', header=0, low_memory=False, encoding='latin-1')\n",
    "truth_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7e7729d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>118.3520</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:145,255,0:230:0:0,201:1,28:1,229:0</td>\n",
       "      <td>1,229</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.995652</td>\n",
       "      <td>248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,255,0:93:0:0,38:0,55:0,93:0</td>\n",
       "      <td>0,93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,178,0:59:0:0,53:0,6:0,59:0</td>\n",
       "      <td>0,59</td>\n",
       "      <td>0</td>\n",
       "      <td>59</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>205.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:235,51,0:17:0:0,15:0,2:0,17:0</td>\n",
       "      <td>0,17</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=14;ADF=0,7;ADR=0,5;AD=0,12;SCR=0;VDB=0.4033...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,36,0:12:0:0,7:0,5:0,12:0</td>\n",
       "      <td>0,12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,166,0:55:0:0,14:0,41:0,55:0</td>\n",
       "      <td>0,55</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79443</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1418109</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>25.4267</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:55,255,0:99:0:0,26:0,73:0,99:0</td>\n",
       "      <td>0,99</td>\n",
       "      <td>0</td>\n",
       "      <td>99</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>26.4242</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:56,208,0:69:0:0,17:0,52:0,69:0</td>\n",
       "      <td>0,69</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1428198</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,148,0:49:0:0,12:0,37:0,49:0</td>\n",
       "      <td>0,49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2221159</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>28.4205</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:58,255,0:153:0:0,42:0,111:0,153:0</td>\n",
       "      <td>0,153</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74079 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM      POS       rsID REF ALT      QUAL FILTER  \\\n",
       "0      chr1   817514  rs3131971   T   C  118.3520      .   \n",
       "1      chr1   826893  rs3115849   G   A   34.4159      .   \n",
       "2      chr1   930939  rs9988021   G   A  225.4170      .   \n",
       "3      chr1   941119  rs4372192   A   G  205.4170      .   \n",
       "7      chr1   944858  rs3748592   A   G  225.4170      .   \n",
       "...     ...      ...        ...  ..  ..       ...    ...   \n",
       "79442  chrY  1387425          .   C   G   23.4340      .   \n",
       "79443  chrY  1418109          .   C   G   25.4267      .   \n",
       "79444  chrY  1425778          .   G   A   26.4242      .   \n",
       "79445  chrY  1428198          .   C   A   23.4340      .   \n",
       "79446  chrY  2221159          .   T   C   28.4205      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...   \n",
       "1      DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...   \n",
       "2      DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...   \n",
       "3      DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...   \n",
       "7      DP=14;ADF=0,7;ADR=0,5;AD=0,12;SCR=0;VDB=0.4033...   \n",
       "...                                                  ...   \n",
       "79442  DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...   \n",
       "79443  DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...   \n",
       "79444  DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...   \n",
       "79445  DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...   \n",
       "79446  DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...   \n",
       "\n",
       "                           FORMAT                                  SAMPLE  \\\n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR  1/1:145,255,0:230:0:0,201:1,28:1,229:0   \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:64,255,0:93:0:0,38:0,55:0,93:0   \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,178,0:59:0:0,53:0,6:0,59:0   \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR       1/1:235,51,0:17:0:0,15:0,2:0,17:0   \n",
       "7      GT:PL:DP:SP:ADF:ADR:AD:SCR        1/1:255,36,0:12:0:0,7:0,5:0,12:0   \n",
       "...                           ...                                     ...   \n",
       "79442  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,166,0:55:0:0,14:0,41:0,55:0   \n",
       "79443  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:55,255,0:99:0:0,26:0,73:0,99:0   \n",
       "79444  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:56,208,0:69:0:0,17:0,52:0,69:0   \n",
       "79445  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,148,0:49:0:0,12:0,37:0,49:0   \n",
       "79446  GT:PL:DP:SP:ADF:ADR:AD:SCR   1/1:58,255,0:153:0:0,42:0,111:0,153:0   \n",
       "\n",
       "          AD  RD  A_D       VAF   DP  \n",
       "0      1,229   1  229  0.995652  248  \n",
       "1       0,93   0   93  1.000000   97  \n",
       "2       0,59   0   59  1.000000   67  \n",
       "3       0,17   0   17  1.000000   23  \n",
       "7       0,12   0   12  1.000000   14  \n",
       "...      ...  ..  ...       ...  ...  \n",
       "79442   0,55   0   55  1.000000   65  \n",
       "79443   0,99   0   99  1.000000  111  \n",
       "79444   0,69   0   69  1.000000   79  \n",
       "79445   0,49   0   49  1.000000   57  \n",
       "79446  0,153   0  153  1.000000  174  \n",
       "\n",
       "[74079 rows x 15 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_BCFTOOL.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[6]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "vcf['DP'] = vcf['INFO'].str.extract(r'DP=(\\d+)')[0].fillna('0').astype(int)\n",
    "filter_var = vcf[vcf['VAF'] >= 0.1]\n",
    "filter_var_DP = filter_var[filter_var['DP'] >= 10]\n",
    "#filter_var_DP = vcf.copy()\n",
    "filter_var_DP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fcacbe1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>948711</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>rs7522415</td>\n",
       "      <td>41.4010</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=17;ADF=1,0;ADR=11,4;AD=12,4;SCR=0;VDB=0.051...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:75,0,176:16:0:1,0:11,4:12,4:0</td>\n",
       "      <td>12,4</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>948721</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs4970455</td>\n",
       "      <td>120.0800</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=18;ADF=0,1;ADR=6,8;AD=6,9;SCR=0;VDB=0.00108...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:153,0,82:15:0:0,1:6,8:6,9:0</td>\n",
       "      <td>6,9</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>953779</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs13302945</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=57;ADF=0,36;ADR=0,10;AD=0,46;SCR=0;VDB=0.44...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,138,0:46:0:0,36:0,10:0,46:0</td>\n",
       "      <td>0,46</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962933</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs375277853</td>\n",
       "      <td>152.3590</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=56;ADF=0,0;ADR=24,28;AD=24,28;SCR=0;VDB=0.2...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:185,0,169:52:0:0,0:24,28:24,28:0</td>\n",
       "      <td>24,28</td>\n",
       "      <td>24</td>\n",
       "      <td>28</td>\n",
       "      <td>0.538462</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973929</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs3829738</td>\n",
       "      <td>206.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=10;ADF=0,6;ADR=0,3;AD=0,9;SCR=0;VDB=0.72764...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:236,27,0:9:0:0,6:0,3:0,9:0</td>\n",
       "      <td>0,9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28374</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154901948</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs4074307</td>\n",
       "      <td>43.9289</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=18;ADF=14,4;ADR=0,0;AD=14,4;SCR=0;VDB=0.372...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:78,0,176:18:0:14,4:0,0:14,4:0</td>\n",
       "      <td>14,4</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>0.222222</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28375</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154903815</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>rs4898352</td>\n",
       "      <td>130.1590</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=44;ADF=27,17;ADR=0,0;AD=27,17;SCR=0;VDB=0.8...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:163,0,188:44:0:27,17:0,0:27,17:0</td>\n",
       "      <td>27,17</td>\n",
       "      <td>27</td>\n",
       "      <td>17</td>\n",
       "      <td>0.386364</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28376</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154929926</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>rs1800292</td>\n",
       "      <td>222.3760</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=86;ADF=22,21;ADR=21,16;AD=43,37;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:255,0,255:80:2:22,21:21,16:43,37:0</td>\n",
       "      <td>43,37</td>\n",
       "      <td>43</td>\n",
       "      <td>37</td>\n",
       "      <td>0.462500</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28377</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>26.4242</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:56,208,0:69:0:0,17:0,52:0,69:0</td>\n",
       "      <td>0,69</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28378</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2221159</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>28.4205</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:58,255,0:153:0:0,42:0,111:0,153:0</td>\n",
       "      <td>0,153</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28379 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT         rsID      QUAL FILTER  \\\n",
       "0      chr1     948711   C   G    rs7522415   41.4010      .   \n",
       "1      chr1     948721   A   C    rs4970455  120.0800      .   \n",
       "2      chr1     953779   A   C   rs13302945  225.4170      .   \n",
       "3      chr1     962933   C   T  rs375277853  152.3590      .   \n",
       "4      chr1     973929   T   C    rs3829738  206.4170      .   \n",
       "...     ...        ...  ..  ..          ...       ...    ...   \n",
       "28374  chrX  154901948   A   G    rs4074307   43.9289      .   \n",
       "28375  chrX  154903815   A   T    rs4898352  130.1590      .   \n",
       "28376  chrX  154929926   T   G    rs1800292  222.3760      .   \n",
       "28377  chrY    1425778   G   A            .   26.4242      .   \n",
       "28378  chrY    2221159   T   C            .   28.4205      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=17;ADF=1,0;ADR=11,4;AD=12,4;SCR=0;VDB=0.051...   \n",
       "1      DP=18;ADF=0,1;ADR=6,8;AD=6,9;SCR=0;VDB=0.00108...   \n",
       "2      DP=57;ADF=0,36;ADR=0,10;AD=0,46;SCR=0;VDB=0.44...   \n",
       "3      DP=56;ADF=0,0;ADR=24,28;AD=24,28;SCR=0;VDB=0.2...   \n",
       "4      DP=10;ADF=0,6;ADR=0,3;AD=0,9;SCR=0;VDB=0.72764...   \n",
       "...                                                  ...   \n",
       "28374  DP=18;ADF=14,4;ADR=0,0;AD=14,4;SCR=0;VDB=0.372...   \n",
       "28375  DP=44;ADF=27,17;ADR=0,0;AD=27,17;SCR=0;VDB=0.8...   \n",
       "28376  DP=86;ADF=22,21;ADR=21,16;AD=43,37;SCR=0;VDB=0...   \n",
       "28377  DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...   \n",
       "28378  DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...   \n",
       "\n",
       "                           FORMAT                                  SAMPLE  \\\n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR       0/1:75,0,176:16:0:1,0:11,4:12,4:0   \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR         0/1:153,0,82:15:0:0,1:6,8:6,9:0   \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR     1/1:255,138,0:46:0:0,36:0,10:0,46:0   \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR    0/1:185,0,169:52:0:0,0:24,28:24,28:0   \n",
       "4      GT:PL:DP:SP:ADF:ADR:AD:SCR          1/1:236,27,0:9:0:0,6:0,3:0,9:0   \n",
       "...                           ...                                     ...   \n",
       "28374  GT:PL:DP:SP:ADF:ADR:AD:SCR       0/1:78,0,176:18:0:14,4:0,0:14,4:0   \n",
       "28375  GT:PL:DP:SP:ADF:ADR:AD:SCR    0/1:163,0,188:44:0:27,17:0,0:27,17:0   \n",
       "28376  GT:PL:DP:SP:ADF:ADR:AD:SCR  0/1:255,0,255:80:2:22,21:21,16:43,37:0   \n",
       "28377  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:56,208,0:69:0:0,17:0,52:0,69:0   \n",
       "28378  GT:PL:DP:SP:ADF:ADR:AD:SCR   1/1:58,255,0:153:0:0,42:0,111:0,153:0   \n",
       "\n",
       "          AD  RD  A_D       VAF   DP  \n",
       "0       12,4  12    4  0.250000   17  \n",
       "1        6,9   6    9  0.600000   18  \n",
       "2       0,46   0   46  1.000000   57  \n",
       "3      24,28  24   28  0.538462   56  \n",
       "4        0,9   0    9  1.000000   10  \n",
       "...      ...  ..  ...       ...  ...  \n",
       "28374   14,4  14    4  0.222222   18  \n",
       "28375  27,17  27   17  0.386364   44  \n",
       "28376  43,37  43   37  0.462500   86  \n",
       "28377   0,69   0   69  1.000000   79  \n",
       "28378  0,153   0  153  1.000000  174  \n",
       "\n",
       "[28379 rows x 15 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mer = pd.merge(truth_set, filter_var_DP, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'inner', sort=False)\n",
    "mer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f0402fa4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAtYAAAIhCAYAAABqh/1nAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAACNsUlEQVR4nOzdd1gUV9sG8HtZelc6ioANRI0FGxjFir1Eo9iwodHERBGN3ViSiDEaS+wJRU0UYjexxN6xI9Fo1ETsIFYWBaSd7w8/9nXdBXZxYQHv33XN9WbPPHPOM8Pq+zCeOSMRQggQEREREdE70dN1AkREREREZQELayIiIiIiLWBhTURERESkBSysiYiIiIi0gIU1EREREZEWsLAmIiIiItICFtZERERERFrAwpqIiIiISAtYWBMRERERaQELa3qvpKSkYMKECfD394ednR0kEglmzpypFJednY0ffvgB7du3R8WKFWFqaooaNWpg0qRJeP78ucq+b9++jaFDh8LZ2RlGRkaoUKECPvroI4WYyMhISCQSlVtiYqI8TiaT4dtvv0WLFi3g6OgIc3Nz1K5dG9999x3S09MV+pw5c2aefUokEkRFRclj//77b3z22Wfw8fGBmZkZJBIJDh8+rHQuCQkJmDZtGnx8fGBrawtLS0t4e3tj9erVyM7OVvt6Hzt2DEZGRrh9+zYePXoEQ0ND9OnTJ894mUwGU1NTdO3aVaE9JCQEEokEnTt3VnncrVu38jz/Bg0aqJ0vAGRmZmLWrFlwc3ODkZERPD098eOPP6p1rLrfr4KsWbMGtWrVgomJCSpUqIBevXrhzp07ah//9vfM2NgYjo6OaNmyJUJDQ5GUlKR0zNvfI0NDQ7i7u2PMmDF5fudLmrVr16JPnz7w8PCAnp4e3Nzc8oyNjY1F9+7d4ezsDFNTU3h6emL27NlITU1ViGvevDmCg4PVzuHJkyeYPHkyvLy8YGZmBisrK3h6eiIwMBB//fVXIc+seBw+fDjPvxMKK68/l7a2tvIYdb97b3+v9fX14eTkhD59+uDGjRtay7kwcq+dOlte51KxYkUMGTIE9+/fz7NfqVQKOzs7dOnSBefOndPV6VI+9HWdAFFxevLkCVavXo06deqge/fu+Pnnn1XGpaWlYebMmejbty+GDRsGW1tbXLhwAd988w1+//13nDt3DiYmJvL4y5cvo0WLFqhcuTLmz5+PihUrIiEhAX/++afK/iMiIuDp6anQZmNjI//vO3fuYNGiRQgMDERISAjMzc1x7NgxzJw5E/v27cO+ffvkf0EPGzYM7du3Vxpj+PDh+O+//xT2nTt3Dtu2bUO9evXQunVr/P777yrzO3/+PNauXYuBAwdi+vTpMDAwwO7du/Hpp5/i1KlTCA8Pz+MK/48QAsHBwRg+fDhcXV0BAF27dsW2bdvw7NkzlCtXTumYqKgopKWlISgoSN6WmZmJX375BQCwZ88e3L9/HxUqVFA55hdffIF+/foptJmbmxeY65s+++wzrFu3Dl9//TUaNmyIP//8E2PGjEFKSgqmTJmS77Hqfr/ys2XLFgwePBiDBw/G4sWL8fDhQ0RHR+PWrVuoVKmSRn3lfs8yMzORlJSE48eP47vvvsP8+fMRHR2NNm3aKB2zZ88eWFlZISUlBbt27cLixYtx5swZnDx5Uv6dK6nWrVuHxMRENGrUCDk5OcjMzFQZd+XKFfj6+sLDwwOLFi2Cra0tjh49itmzZ+P8+fPYvn27PPbrr79G27Zt8emnn8LDwyPf8V+8eIEmTZrgxYsX+PLLL1GnTh2kpaXh+vXr2LJlCy5evIgPPvhAq+dcGnz88ccYN26cQpuBgYFSnLrfvdzvdXp6Ok6cOIFvv/0Whw4dwj///KPy75XiUL9+fcTExCi0ffTRR6hSpQrmz5+f53G555KWloajR48iNDQUR44cwaVLl2BmZiaPmzNnDlq2bInMzEzExsZi1qxZ8PPzw8WLF1GtWrUiOy8qBEH0HsnJyRE5OTlCCCEePXokAIgZM2YoxWVlZYnHjx8rtW/cuFEAEOvWrVPos27duqJu3boiPT093/EjIiIEAHH27Nl84168eCFevHih1P79998LAOLYsWP5Hh8fHy8kEokYMGCAQnt2drbSuRw6dEjp+KdPn4qMjAyl9lGjRgkA4s6dO/mOL4QQu3btEgDEP//8o9T2448/qjymcePGwsHBQWRmZirl2alTJwFAfPvttyrPF4D4/vvvC8wrP5cvXxYSiUTMmTNHoX348OHCxMREPHnyJN/j1f1+5ad3797CyclJ3k9h5Pc9u337tnBxcREWFhYiMTFR3j5jxgwBQDx69EghPjAwUAAQx48fz3fMN3/OuvLm97tTp07C1dVVZdzUqVMFAPHvv/8qtH/yyScCgHj69KlCe61atcTw4cMLHD88PFwAEAcPHiwwv5Lo0KFDef6dUFgAxKhRo/KNUfe7l9f3etasWQKACA8P11re2uDq6io6deqkcl9e5zJ9+nQBQPzyyy9CiP/9TDZu3KgQt2bNGgFAfPXVV0WTPBUap4LQe+XNf4rLj1QqVbiDnKtRo0YAgLt378rbjh49iosXLyI4OBhGRkZaydPMzEzhbkV+46sSHh4OIQSGDRum0K6np94f+XLlyqm8o5Q7/r179wrsY8WKFWjYsKHCXb527dqhYsWKiIiIUIq/evUqTp8+jYEDB0Jf/3//mBYWFgZDQ0NERETAxcUFEREREEKodR6a2rZtG4QQGDJkiEL7kCFDkJaWhj179uR7vLrfr/xIpVI8fvwYjx8/fqd+8lKpUiUsWLAAKSkpWLVqVYHxTZo0AfB6qlN+PD094e3tjfnz56v1/SgK6n6/c7/bVlZWCu3W1tbQ09ODoaGhQntgYCDWr1+PlJSUfPt98uQJAMDJyanA/P79918MGTIE1apVg6mpKSpUqIAuXbrg0qVLCsfkTgVYv349Jk6cCCcnJ5ibm6NLly54+PAhUlJS8Mknn8DW1ha2trYYMmQIXrx4odCHRCLB559/jlWrVqF69eowMjKCl5eXwjSx/Jw7dw5du3ZF+fLlYWxsjHr16uG3335T69h3oe53L3e618OHD/ONW7RoESQSCf7991+lfRMnToShoaH8z11sbCw6d+4Me3t7GBkZwdnZGZ06dSry77a2z5mKHwtrIg0cPHgQAFCzZk1529GjRwEAFhYW6NixI4yNjWFubo7OnTvjn3/+UdlP586dIZVKUb58efTo0QOXL18u9Phvy8nJQWRkJKpWrQo/Pz+1+lXXwYMHoa+vj+rVq+cbl5GRgf3796Nly5YK7Xp6ehg8eDAuXLiAuLg4hX25xfbQoUPlbffu3cPevXvRrVs32NnZYdCgQfj333/l1/xtOTk5yMrKUtg0KcIvX74MOzs7ODo6KrTn/vO9uj+nd/HJJ58gMzMTPXv2VJrvqy0dO3aEVCrN8zq+KbcIsbOzyzdu3759qFevHkJDQ1GpUiX4+flh5cqVBf6C8PbPK69Nm79MDRo0CNbW1vj0009x8+ZNpKSk4I8//sCqVaswatQopV9qW7RogZcvXxY499jHxwcAMHDgQGzbtk1eaKvy4MED2NjYYO7cudizZw+WLVsGfX19NG7cGNeuXVOKnzJlCpKSkhAZGYkFCxbg8OHD6Nu3L3r27AkrKyts2LABEyZMwLp161ROWdqxYweWLFmC2bNnY9OmTXB1dUXfvn2xadOmfM/p0KFDaNq0KZ4/f46VK1di+/btqFu3LgICAhAZGZnvsbmEEIX6ear73YuPjweAAv9eGjBgAAwNDZXyzs7Oxi+//IIuXbrA1tYWL1++RNu2bfHw4UMsW7YM+/btw6JFi1CpUqUCf7l6V9o+Z9IBHd4tJ9IpTf+p/t69e8LBwUE0aNBA4Z90R4wYIQAIS0tLERQUJPbv3y/WrVsnXF1dha2trXjw4IE8dvfu3WLq1Kni999/F0eOHBFLly4VFStWFGZmZuLixYv5jh8XFydMTEzERx99lG/c7t27BQARGhqab1x+U0FU+fPPP4Wenp4YO3ZsgbGnT58WAERUVJTSvps3bwqJRCJGjx4tb8vMzBSOjo6iadOmCrGzZ88WAMSePXsUjg0MDFSIy50Komrbt2+fWucnhBBt27YVHh4eKvcZGhqKTz75RO2+CjsVZObMmcLV1VWYmJiI1q1bi9TUVI2OF0K9KUcODg6iRo0a8s+5/xyfmJgoMjMzxbNnz8Qvv/wiTExMhIuLi0hLS1Nr7IyMDPHHH3+IAQMGCAsLC6Gvry86duwo1q1bJ1JSUpTi8/q5vb1FRERodA3ymwoihBBXr14Vnp6eCmOMHj1a5RScjIwMIZFIxMSJEwscd/bs2cLQ0FDep7u7uxg5cqSIi4vL97isrCyRkZEhqlWrpvBnLHcqQJcuXRTig4OD5Tm/qXv37qJ8+fIKbQCEiYmJwtSfrKws4enpKapWrao01pt/J3h6eop69eopTM8SQojOnTsLJyenAqe35PXz/Omnn+Qx6n73cr/Xp06dEpmZmSIlJUXs2bNHODo6iubNmyvlqEqPHj1ExYoVFfLOnaL2+++/CyGEOHfunAAgtm3bVmB/+VFnKsib5/LHH38IOzs7hWlauT+T6OhokZmZKVJTU8WJEyeEh4eH8PLyEs+ePXunHEn7WFjTe0uTwufJkyfigw8+EPb29uK///5T2Dd8+HABQLRr106hPTY2VgAQU6dOzbfv+Ph4YW5uLrp27ZpvjIuLi6hevXqB83w//vhjoa+vLxISEvKN06SwPn/+vLCyshK+vr4FziMXQoitW7fmO9e0VatWwsbGRrx69UoIIcSOHTsEABEWFiaPycnJEe7u7sLFxUXh/wRbtmwpTE1NRXJysrwtt7AeM2aMOHv2rMImk8kKzDdX27Zthaenp8p9hoaGYsSIEWr3VZjCet68ecLS0lLcvHlT7Nu3T5iYmIi2bdsqFLVVqlQRAwcOzLcfdQpre3t7lYX121vTpk3F33//rfY5vCktLU1s2rRJ9OrVS5iYmAgTExPx22+/KcS8/fPKa1P1zEN+8ius4+PjRdWqVUXTpk3Fpk2bxJEjR+TXfujQoSqPKVeunNIzC3lJTEwU4eHhYsSIEaJ27doCgNDX1xfr16+Xx2RmZopvv/1W1KhRQxgYGChc8/bt28vjcgurVatWKYyxatUqAUD8+eefCu2TJ08WABR+iQEgOnfurJRn7s/87t27CmPl/p1w48YNAUDMnz9fZGZmKmzLly8XAMSVK1fyvRYARO/evZV+nm/Op1b3u5f7vX57q1GjhtoF5u+//6503Xr16iUcHR1FVlaWEEKI58+fi3LlygkPDw+xYsWKQn//1Sms395q166t8DxD7s/k7c3JyUnEx8cXKi8qWlwVhKgAz549Q9u2bXH//n0cPHgQlStXVtifOxe7Xbt2Cu1169aFk5MTLly4kG//bm5u+PDDD3Hq1CmV+2/fvo2WLVtCX18fBw4cQPny5fPs6/Hjx9ixYwc6deqkNJ2hsGJjY9G2bVtUq1YNu3btUmseeVpaGgDA2NhY5f6goCD0798fO3bswMcff4yIiAiYm5ujd+/e8piDBw8iPj4eISEhkMlk8vbevXvj0KFD2LBhA0aMGKHQb8WKFTVeXu9NNjY2uHjxolL7y5cvkZGRke+1f1dZWVn45ptvMHDgQLi7u8Pd3R2///47unTpgu7du2P79u1ISkrCzZs3MWfOnHca6+XLl3jy5Alq166ttG///v2wsrKCgYEBKlasqPJZA3WlpaUhOTkZycnJyMzMhLW1tdJ3om7dumr1JZVKC53H2yZNmgSZTIaLFy/Kp300b94ctra2GDp0KAYOHKg0jcrY2Fj+vS6Ig4MDhgwZIp+rf/ToUXTo0AFjxoxB3759AbxeQnLZsmWYOHEi/Pz8UK5cOejp6WHYsGEqx3n7u5c7Dzyv9vT0dIUVcVT9fZDb9uTJE1SsWFFpf+783fHjx2P8+PEqz1WdZwHs7OzU+nOp7ndv7dq1qFGjBlJSUhAdHY1Vq1ahb9++2L17d4FjdOjQAU5OToiIiIC/vz+ePXuGHTt2YMyYMfLvmJWVFY4cOYJvv/0WU6ZMwbNnz+Dk5IThw4dj2rRpKp8/Kazcc9HX14eDg0Oe8/O/++47tGrVCqmpqdi7dy9CQ0PRvXt3nD59WmvP9pB2sLAmysezZ8/Qpk0bxMfH48CBAyqXyspv+SwhhFoPVOUVd/v2bbRo0QJCCBw+fFjl//m9ad26dcjIyFB6aLGwYmNj0aZNG7i6umLv3r1KD3vlJXeN2qdPn6rc36NHD5QrVw7h4eHw8/PDH3/8gYEDByoUAmFhYQCAH374AT/88INSH2FhYUqF9buqXbs2oqKikJiYqFCI5D5QVqtWLa2O96bHjx9DJpPB0tJS3ta6dWvs3LkTnTt3Ro8ePWBpaQlPT0/06NHjncbauXMnsrOz0aJFC6V9derUUVhjWFMpKSnYvn07oqKisHfvXhgZGaFbt27Yvn07/P39FR5MBVQvu6ZKREQEBg8eXOi83nTx4kX5OtNvatiwIYDXc+nfLqyfPXtW6OvSvHlz+Pv7Y9u2bUhKSoK9vT1++eUXDBw4UOmXpMePH8Pa2rpQ4+TnzXXy327Lq4DNPd/Jkyfn+Z0raAlCTaj73atRo4a8UG/ZsiWys7Px888/Y9OmTfj444/zPVYqlSIwMBBLlizB8+fPsX79erx69UrpgeXcvwuEEPjrr78QGRmJ2bNnw8TEBJMmTSr8SeZzLvmpXLmyPK558+YwMTHBtGnT8OOPP+b5Sw/pBh9eJMpDblF98+ZN7N27F/Xq1VMZ16FDB5iamirdLblw4QISExPlT3nnJT4+HidOnFCKu3PnDlq0aIHs7GwcPHhQvhZ0fsLCwuDs7IwOHToUGFuQixcvok2bNqhYsSL27dun0fqwNWrUAAD8999/KvcbGxujX79+2Lt3L7777jtkZmYqPLT47NkzbN26FU2bNsWhQ4eUtv79++Ps2bNaf5iwW7dukEgkWLNmjUJ7ZGQkTExMVK4Xri12dnawt7fH5s2b8fLlS3l7y5YtsXPnTuzbtw9RUVFYvny5UnGqiTt37mD8+PGwsrLS6i8m0dHR6NGjB+zt7REUFASpVIp169YhKSkJv/zyCzp27Kgy77Nnz6q1denSRWu5Ojs74++//1ZaPSN3HeK3f4F98OAB0tPT4eXllW+/Dx8+RE5OjlJ7dnY2bty4AVNTU3nRLJFIlO407ty5U+HlINp04MABhRUksrOzER0djSpVquT5C7uHhweqVauGuLg4NGjQQOVmYWFRJPlqYt68eShXrhy++uorldf/bUOGDEF6ejo2bNiAyMhI+Pj4KL1XIJdEIkGdOnWwcOFCWFtbF/gvkMVlwoQJqFq1KubOnVvkD1SSZnjHmt47u3fvxsuXL+V/GV25ckX+ZHzHjh1hamqKtLQ0tGvXDrGxsVi0aBGysrIUpmrY2dmhSpUqAF4v0TV79myMHz8egwcPRt++fZGYmIjp06ejUqVK+Oyzz+THtWnTBs2bN8cHH3wAS0tLXLp0CfPmzYNEIsHXX38tj0tKSkLLli2RkJCAsLAwJCUlKbwtr2LFikr/Z3j69Gn8/fffmDJlSp7/bJ6amopdu3YBgPx8jhw5gsePH8PMzExekF+7dk3+8pBvv/0WN27cUHizWZUqVfJ9ar1ixYqoXLkyTp06hdGjR6uMCQoKwrJly/DDDz/A09MTvr6+8n2//vor0tPTMXr0aJV3VW1sbPDrr78iLCwMCxcuzDMPTdWsWRNBQUGYMWMGpFIpGjZsiL1792L16tX45ptvFP7Zffbs2Zg9ezYOHDigcHdTne+XKlKpFIsXL0a/fv3g4+ODsWPHws3NDbdv30Z4eDiMjY1hZmaGKVOmYO/evWq9+Oby5cvyVRiSkpJw7NgxREREQCqVYuvWrQWuPKCJfv36wc/PDz/++CM+/vhjte+6vsvUnbdduXIFV65cAfD6bmxqaqr82nt5eckL4+DgYHTv3h1t27bF2LFjYWtri1OnTiE0NBReXl5Kv5jm/ll5e5Wbt61btw6rVq1Cv3790LBhQ1hZWeHevXv4+eef8ffff+Orr76ST9Xo3LkzIiMj4enpiQ8++ADnz5/H999/X+C/ShWWra0tWrVqhenTp8PMzAzLly/HP//8U+CSe6tWrUKHDh3Qrl07DB48GBUqVMDTp09x9epVXLhwARs3biySfDVRrlw5TJ48GRMmTMD69esxYMCAfOM9PT3h4+OD0NBQ3L17F6tXr1bY/8cff2D58uXo3r07KleuDCEEtmzZgufPn6Nt27ZFeSpqMzAwwJw5c9C7d28sXrwY06ZN03VKlEunM7yJdMDV1TXPJ9VzHwbJb5UJAGLQoEFK/f7000+iVq1awtDQUNjY2Ij+/fvLHwrKFRwcLLy8vOSrJTg7O4sBAwaIa9euKcTl9cBK7qbqgbjhw4cLiUSi9HDlm/I7rzcf9MrrwZrcTZ0VGqZPny7KlSuX78OO9erVEwDEvHnzFNrr1q0r7O3t5Q83qtKkSRNha2srXr16pbUXxAjxegWIGTNmiEqVKglDQ0NRvXp1sWTJEqW43Aeu3n74U53vV36OHDkiOnToIKytrYWBgYGoXLmy+OKLL8SdO3fE8ePHhbGxsWjWrJnKFwjlevvnZ2hoKOzt7YWfn5+YM2eOSEpKyvN83n5Jh7reXP1GV/J6CE7Vn5mDBw8Kf39/4ejoKExMTET16tXFuHHjVD4kGRgYKGrXrl3g+FeuXBHjxo0TDRo0EHZ2dkJfX1+UK1dO+Pn5KbxUSgghnj17JoKCgoS9vb0wNTUVH374oTh27Jjw8/MTfn5+8ri8XhCS1wOqqn6OwOuXtCxfvlxUqVJFGBgYCE9PT/Hrr78qHJvXC2Li4uJE7969hb29vTAwMBCOjo6iVatWYuXKlQVek9yx86Pudy+/h3LT0tJEpUqVRLVq1eQPIeZn9erV8tVS3nwQWojXLzvq27evqFKlijAxMRFWVlaiUaNGIjIyssB+31SYF8S8La+ff67GjRuLcuXKiefPn2uUGxUdiRBF9KYFInqvPXjwAO7u7li7di0CAgJ0nQ5RochkMjg7O2PhwoUYPny4rtMpFIlEglGjRmHp0qW6ToWozOMcayIqEs7OzggODsa3336r1rxHopJo4cKFqFSpktLDbUREqnCONREVmWnTpsHU1BT379+Hi4uLzvIQQiA7OzvfGKlU+s6vI6eyx9LSEpGRke/0wCgRvT84FYSIyrzIyMgC7zgeOnRI5YOSRERE6mJhTURl3pMnTxAfH59vjIeHR4lYOoyIiEovFtZERERERFrAhxeJiIiIiLSAT2PoWE5ODh48eAALCws+OEVERERUAgkhkJKSAmdnZ+jp5X1fmoW1jj148ECnqyUQERERkXru3r2b7xtSWVjrWO7DUnfv3oWlpaWOsyEiIiKit8lkMri4uBT4kDsLax3Lnf5haWnJwpqIiIioBCto2i4fXiQiIiIi0gIW1kREREREWsDCmoiIiIhICzjHuhQQQiArKwvZ2dm6ToVIK6RSKfT19bnEJBERlSksrEu4jIwMJCQkIDU1VdepEGmVqakpnJycYGhoqOtUiIiItIKFdQmWk5OD+Ph4SKVSODs7w9DQkHf4qNQTQiAjIwOPHj1CfHw8qlWrlu9i+0RERKUFC+sSLCMjAzk5OXBxcYGpqamu0yHSGhMTExgYGOD27dvIyMiAsbGxrlMiIiJ6Z7xNVArwbh6VRfxeExFRWcP/ZyMiIiIi0gIW1kREREREWsA51u+B7ByBM/FPkZSSDnsLYzRyLw+pHh+CJCIiItIm3rEu4/ZcTsCH3x1E359OYUzURfT96RQ+/O4g9lxOKNJxBw8eDIlEIt9sbGzQvn17/PXXX/IYIQRWr16Nxo0bw9zcHNbW1mjQoAEWLVokX15w5syZCv3kbj///LPK9je3mTNnysdas2YNGjVqBDMzM1hYWKB58+b4448/lPLOzs7GwoUL8cEHH8DY2BjW1tbo0KEDTpw4oRAXGRkJa2vrIrl2REREVDqxsC7D9lxOwKe/XEBCcrpCe2JyOj795UKRF9ft27dHQkICEhIScODAAejr66Nz587y/YGBgQgODka3bt1w6NAhXLx4EdOnT8f27duxd+9eeVzNmjXl/eRugYGBCp/HjRunFDd+/HgAwPjx4zFixAj07t0bcXFxOHPmDJo1a4Zu3bph6dKl8nGEEOjTpw9mz56N0aNH4+rVqzhy5AhcXFzQokULbNu2rUivFxEREZVunApSigghkJap3tsXs3MEZuz4G0JVPwAkAGbuuIKmVW3VmhZiYiDVeA1tIyMjODo6AgAcHR0xceJENG/eHI8ePcKhQ4fw66+/Ytu2bejWrZv8GDc3N3Tt2hUymUzepq+vL+/nTW+2mZubq4w7deoUFixYgCVLluCLL76Qt3/77bdIT09HSEgIunXrBhcXF/z222/YtGkTduzYgS5dushjV69ejSdPnmDYsGFo27YtzMzMNLoOREREpEU52cDtk8CLh4C5A+DqC+hJdZ0VABbWpUpaZja8vvpTK30JAImydNSeubfAWAC4MrsdTA0L/3V58eIFfv31V1StWhU2Njb49ddf4eHhoVBU55JIJLCysir0WG/asGEDzM3NMWLECKV948aNww8//IDNmzcjODgY69evR/Xq1RWK6jdjt2zZgn379qF79+5ayY2IiIg0dGUHsGciIHvwvzZLZ6D9d4BXV93l9f9YWFOR+eOPP2Bubg4AePnyJZycnPDHH39AT08PN27cgIeHh1r9XLp0Sd4PAHh5eeHMmTNqHXv9+nVUqVJF5WuznZ2dYWVlhevXr8tja9SoobKf3PbcWCIiIipmV3YAvw0E3v73eFnC6/bea3VeXLOwLkVMDKS4MrudWrFn4p9icMTZAuMihzREI/fyao2tqZYtW2LFihUAgKdPn2L58uXo0KEDzpw5AyGE2lNLPDw8sGPHDvlnIyMjjXPJiyZ5AOAr5YmIiHQhJ/v1ner8JrnumQR4dtLptBAW1qWIRCJRezpGs2p2cLIyRmJyusqvoASAo5UxmlWzK7Kl98zMzFC1alX5Z29vb1hZWeGnn35C9erVcfXqVbX6MTQ0VOhHE9WrV8fx48eRkZGhdNf6wYMHkMlkqFatmjz2ypUrKvvJzTU3loiIiIrR7ZOK0z+UCEB2/3Wce7NiS+ttXBWkjJLqSTCjixeA10X0m3I/z+jiVazrWUskEujp6SEtLQ39+vXD9evXsX37dqU4IQSSk5O1MmafPn3w4sULrFq1Smnf/PnzYWBggJ49e8pjb9y4gd9//10pdsGCBbCxsUHbtm21khcRERFp4MVD7cYVEd6xLsPa13LCigH1Mev3KwpL7jlaGWNGFy+0r+VUpOO/evUKiYmJAIBnz55h6dKlePHiBbp06QI/Pz9s3boVffv2xfTp09G2bVvY2dnh0qVLWLhwIb744gutPCTo4+ODMWPG4Msvv0RGRga6d++OzMxM/PLLL1i8eDEWLVoEFxcXAK8L640bN2LQoEH4/vvv0bp1a8hkMixbtgw7duzAxo0bFVYEyc7OxsWLFxXGMzQ0hJeX1zvnTURERG8wd9BuXBFhYV3Gta/lhLZejjp58+KePXvg5PS6eLewsICnpyc2btyIFi1aAADWr1+P1atXIzw8HN988w309fVRrVo1DBw4EO3aqTeXXB2LFi3CBx98gBUrVmD69OmQSCSoX78+tm3bprACiEQiwW+//YbFixdj4cKFGDVqFIyMjODj44NDhw7hww8/VOj3xYsXqFevnkKbq6srbt26pbXciYiICK+X1LN0fv2gYl6TXC2dX8fpkEQIoSo7KiYymQxWVlZITk6GpaWlwr709HTEx8fD3d0dxsbGOsqQqGjw+01ERBqRrwoCKBbX/3+zsAhXBcmvXnsT51gTERERUcnn1fV18Wz51lRWS+cSsdQewKkgRERERFRaeHV9vaQe37xIRERERPSO9KQ6XVIvP5wKQkRERESkBSysiYiIiIi0gIU1EREREZEWsLAmIiIiItICFtZERERERFrAwpqIiIiISAu43N77ICe7xK73SERERFRW8I51WXdlB7CoFrCmM7A56PX/Lqr1ur0IDR48GBKJBBKJBAYGBnBwcEDbtm0RHh6OnJwceZybm5s87s1t7ty5AIBbt26p3C+RSHDq1CkAQGRkpEK7k5MTevfujfj4eGRkZMDW1hbffPONyjxDQ0Nha2uLjIwMeZuHhwcMDQ1x//59pfgWLVrIxzE0NESVKlUwefJkvHr1SiFOIpFg27ZtSsd/8sknkEqliIqKUto3c+ZMSCQSjBw5UqH94sWLkEgkuHXrljwmv+3WrVuqfyhERERUpFhYl2VXdgC/DQRkDxTbZQmv24u4uG7fvj0SEhJw69Yt7N69Gy1btsSYMWPQuXNnZGVlyeNmz56NhIQEhe2LL75Q6Gv//v1KMd7e3vL9lpaWSEhIwIMHD7B+/XpcvHgRXbt2hVQqxYABAxAZGQkhhFKOERERCAwMhKGhIQDg+PHjSE9PR69evRAZGanyvIYPH46EhAT8+++/mDdvHpYtW4aZM2cWeD1SU1MRHR2NL7/8EmFhYSpjjI2NERYWhuvXr6vcP378eIVrULFiRaXr5+LiUmAuREREpH2cClKaCAFkpqoXm5MN7J4AQLmYfN0mAfZMBCq3UG9aiIEpIJGonysAIyMjODo6AgAqVKiA+vXro0mTJmjdujUiIyMxbNgwAICFhYU8Li82Njb5xkgkEvl+JycnzJgxAwMGDMC///6LoKAgLF68GEePHoWfn5/8mGPHjuHGjRsICgqSt4WFhaFfv37w8/PDqFGjMGXKFEjeOm9TU1P5WJUqVcL69euxd+9ehIaG5nsOGzduhJeXFyZPngwnJyfcunULbm5uCjEeHh6wt7fHtGnT8Ntvvyn1YW5uDnNzc/lnqVSq1vUjIiKiosfCujTJTAXmOGupM/H6TvZcNe9uTnkAGJq986itWrVCnTp1sGXLFnlhXRRMTEwAAJmZmahduzYaNmyIiIgIhcI6PDwcjRo1Qq1atQAAKSkp2LhxI06fPg1PT0+8fPkShw8fRsuWLfMcJy4uDidOnFAqkFUJCwvDgAEDYGVlhY4dOyIiIgKzZs1Sips7dy4aNmyIs2fPomHDhhqeOREREekKp4JQsfP09FSYBzxx4kT5ndjc7fDhwwrH+Pr6KsVkZ2er7P/evXv4/vvvUbFiRVSvXh0AMHToUGzatAkvXrwAALx48QIbN25UuFsdFRWFatWqoWbNmpBKpejTp4/KKRvLly+Hubk5jIyMULduXTx69Ahffvllvud848YNnDp1CgEBAQCAAQMGICIiQmG+ea769eujd+/emDRpUr59EhERUcnCO9aliYHp6zvH6rh9Evj144Lj+m96vUqIOmNriRBCYXrFl19+icGDByvEVKhQQeFzdHQ0atSoodAmlf5vCktycjLMzc0hhEBqairq16+PLVu2yOdO9+3bFyEhIYiOjkZQUBCio6MhhECfPn3kfeTeUc41YMAANG/eHM+fP4e1tbW8vX///pg6dSpkMhm+++47WFpaomfPnvmec1hYGNq1awdbW1sAQMeOHREUFIT9+/fD399fKf6bb75BjRo1sHfvXtjb2+fbNxEREZUMLKxLE4lE/ekYVVoBls6vH1RUOc9a8np/lVbFvvTe1atX4e7uLv9sa2uLqlWr5nuMi4tLvjEWFha4cOEC9PT04ODgADMzxetkZWWFjz/+GBEREQgKCkJERAQ+/vhjWFpaAgCuXLmC06dP4+zZs5g4caL8uOzsbGzYsAGffvqpQl+5ufzyyy+oWbMmwsLCFO5+vyk7Oxtr165FYmIi9PX1FdrDwsJUFtZVqlTB8OHDMWnSpDwfdCQiIqKShYV1WaUnBdp/93r1D0igWFz//93i9nOLvag+ePAgLl26hLFjx2q1Xz09vQKL86CgILRo0QJ//PEHTpw4gTlz5sj3hYWFoXnz5li2bJnCMevWrUNYWJhCYf0mAwMDTJkyBZMnT0bfvn1haqp8Z3/Xrl1ISUlBbGyswl32f/75B/3798eTJ09gY2OjdNxXX32FKlWqqFyaj4iIiEoezrEuy7y6Ar3XApZOiu2Wzq/bvboW6fCvXr1CYmIi7t+/jwsXLmDOnDno1q0bOnfujIEDB8rjUlJSkJiYqLDJZDKFvp48eaIUk56erlE+fn5+qFq1KgYOHIiqVauiefPmAF4/4Lhu3Tr07dsXtWrVUtiGDRuG8+fPIy4uLs9++/XrB4lEguXLl6vcHxYWhk6dOqFOnToKfffs2RN2dnb45ZdfVB7n4OCAkJAQLFmyRKPzJCIiIt1gYV3WeXUFgi8Dg/4Aeoa9/t/gS0VeVAPAnj174OTkBDc3N7Rv3x6HDh3CkiVLsH37doU7t1999RWcnJwUtgkTJij01aZNG6UYVS9gKcjQoUPx7NkzDB06VN62Y8cOPHnyBB999JFSfLVq1VC7du18p2MYGhri888/x7x58+QPR+Z6+PAhdu7cqXIOtkQiQY8ePfLt+8svv1RYXo+IiIhKLolQ9dYMKjYymQxWVlZITk6Wz/fNlZ6ejvj4eLi7u8PY2FhHGRIVDX6/iYiotMivXnsT71gTEREREWkBC2siIiIiIi1gYU1EREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWmBvq4ToKKXnZONC0kX8Cj1EexM7VDfvj6ketKCDyQiIiIitfGOdRm3//Z+tNvcDkP/HIqJxyZi6J9D0W5zO+y/vb9IxuvSpQvatGmjcl9MTAwkEgkuXLgAAPjkk08glUoRFRWlFDtz5kxIJBJIJBLo6enB2dkZ/fv3x927dxXibt68ib59+8LZ2RnGxsaoWLEiunXrhuvXryvEHTp0CB07doSNjQ1MTU3h5eWFcePG4f79+0pje3h4wNDQUOW+Fi1aQCKRKOW8aNEiuLm5yT9v2bIFbdu2hZ2dHSwtLeHj44M///xT9UUjIiKiMoGFdRm2//Z+hBwOwcPUhwrtSalJCDkcUiTFdVBQEA4ePIjbt28r7QsPD0fdunVRv359pKamIjo6Gl9++SXCwsJU9lWzZk0kJCTg3r17iI6OxqVLl9C7d2/5/oyMDLRt2xYymQxbtmzBtWvXEB0djVq1aiE5OVket2rVKrRp0waOjo7YvHkzrly5gpUrVyI5ORkLFixQGPP48eNIT09Hr169EBkZqTIvY2NjTJs2DZmZmXleh6NHj6Jt27bYtWsXzp8/j5YtW6JLly6IjY3N7/IRERFRKcapIKWIEAJpWWlqxWbnZCP0TCgEhHI//98298xcNHZsrNa0EBN9E0gkkgLjOnfuDHt7e0RGRmLGjBny9txCes6cOQCAjRs3wsvLC5MnT4aTkxNu3bqlcMcXAPT19eHo6AgAcHZ2xvDhwzF69GjIZDJYWlriypUruHnzJg4ePAhXV1cAgKurK5o2bSrv4969exg9ejRGjx6NhQsXytvd3NzQvHlzPH/+XGHMsLAw9OvXD35+fhg1ahSmTJmidN59+/bF77//jp9++gmfffaZyuuwaNEihc9z5szB9u3b8fvvv6NevXoFXkciIiIqfVhYlyJpWWlovL6x1vp7mPoQvlG+asWe7ncapgamBcbp6+tj4MCBiIyMxFdffSUvSjdu3IiMjAz0798fwOsCdsCAAbCyskLHjh0RERGBWbNm5dlvYmIitmzZAqlUCqn09S8CdnZ20NPTw6ZNmxAcHCxvf1PuuBMmTFDZr7W1tfy/U1JSsHHjRpw+fRqenp54+fIlDh8+jJYtWyocY2lpiSlTpmD27NkYNGgQzMzMCrwuOTk5SElJQfny5QuMJSIiotKJU0FI64YOHYpbt27h8OHD8rbw8HD06NED5cqVw40bN3Dq1CkEBAQAAAYMGICIiAjk5OQo9HPp0iWYm5vD1NQUTk5OOHz4MEaNGiUvZCtUqIAlS5bgq6++Qrly5dCqVSt8/fXXuHnzpryPGzduwNLSEk5OTgXmHRUVhWrVqqFmzZqQSqXo06dPntNUPvvsMxgbG+OHH35Q65osWLAAL1++VJjKQkRERGUL71iXIib6Jjjd77RasecfnsdnB1RPU3jT8tbL4e3grdbY6vL09ISvry/Cw8PRsmVL/Pfffzh27Bj27t0L4PXd6nbt2sHW1hYA0LFjRwQFBWH//v3w9/eX9+Ph4YEdO3bg1atX2L59OzZu3Ihvv/1WYaxRo0Zh4MCBOHToEE6fPo2NGzdizpw52LFjB9q2bQshhFpTWHLzGjBggPzzgAED5NNF3ryzDQBGRkaYPXs2Pv/8c3z66af59rthwwbMnDkT27dvh729vVq5EBERUenDO9aliEQigamBqVqbr7MvHEwdIIHqolICCRxNHeHr7KtWf+oWp7mCgoKwefNmyGQyREREwNXVFa1bt0Z2djbWrl2LnTt3Ql9fH/r6+jA1NcXTp0+V7g4bGhqiatWqqFmzJqZMmYK6deuqLGItLCzQtWtXfPvtt4iLi0OzZs3wzTffAACqV6+O5ORkJCQk5JvvlStXcPr0aUyYMEGeV5MmTZCWloYNGzaoPGbAgAFwc3OTj6VKdHQ0goKC8Ntvv+W5WgoRERGVDSysyyipnhSTGk0CAKXiOvfzxEYTi2w96969e0MqlWL9+vVYs2YNhgwZAolEgl27diElJQWxsbG4ePGifNu4cSO2bduGJ0+e5Nnn9OnTsWHDBvlyfapIJBL5/GgA+Pjjj2FoaIh58+apjM99eDEsLAzNmzdHXFycQl4TJkzIczqInp4eQkNDsWLFCty6dUtp/4YNGzB48GCsX78enTp1yjNnIiIiKhtYWJdhbVzb4IcWP8DeVHH6gYOpA35o8QPauBbdHVRzc3MEBARgypQpePDgAQYPHgzgdQHbqVMn1KlTB7Vq1ZJvPXv2hJ2dHX755Zc8+6xcuTK6deuGr776CgBw8eJFdOvWDZs2bcKVK1fw77//IiwsDOHh4ejWrRsAwMXFBQsXLsTixYsRFBSEI0eO4Pbt2zhx4gRGjBiBr7/+GpmZmVi3bh369u2rkFOtWrUwbNgwnD9/HnFxcSpz6tSpExo3boxVq1YptG/YsAEDBw7EggUL0KRJEyQmJiIxMVFhGUAiIiIqYwTpVHJysgAgkpOTlfalpaWJK1euiLS0tHcaIys7S5xJOCN2/rdTnEk4I7Kys96pP3WdPHlSABD+/v5CCCESExOFvr6++O2331TGf/HFF6J27dpCCCFmzJgh6tSpoxRz4sQJAUCcOnVKPHr0SIwePVrUqlVLmJubCwsLC1G7dm0xf/58kZ2drXDcvn37RLt27US5cuWEsbGx8PT0FOPHjxcPHjwQmzZtEnp6eiIxMVFlXrVr1xZffPGFEEIIPz8/MWbMGJXn6erqKm/z8/MTAJS2QYMGqXHl3g/a+n4TEREVtfzqtTfpvLBetmyZcHNzE0ZGRqJ+/fri6NGj+cYfPnxY1K9fXxgZGQl3d3exYsUKpZhNmzaJGjVqCENDQ1GjRg2xZcsWjcdVVRQBEPPmzVOIO3nypGjZsqUwNTUVVlZWws/PT6Smpqp9/sVRWBOVRPx+ExFRaaFuYa3TqSDR0dEIDg7G1KlTERsbi2bNmqFDhw64c+eOyvj4+Hh07NgRzZo1Q2xsLKZMmYLRo0dj8+bN8piYmBgEBAQgMDAQcXFxCAwMRO/evXH69P9W01Bn3ISEBIUtPDwcEokEPXv2VBirffv28Pf3x5kzZ3D27Fl8/vnn0NPjDBsiIiKi941ECKH8ar5i0rhxY9SvXx8rVqyQt9WoUQPdu3dHaGioUvzEiROxY8cOXL16Vd42cuRIxMXFISYmBgAQEBAAmUyG3bt3y2Pat2+PcuXKyVd30HRcAOjevTtSUlJw4MABeVuTJk3Qtm1bfP3114W8AoBMJoOVlRWSk5NhaWmpsC89PR3x8fFwd3eHsbFxoccgKon4/SYiotIiv3rtTTq7tZqRkYHz588rrFsMAP7+/jh58qTKY2JiYpTi27Vrh3PnziEzMzPfmNw+CzPuw4cPsXPnTgQFBcnbkpKScPr0adjb28PX1xcODg7w8/PD8ePH8z3vV69eQSaTKWxEREREVPrprLB+/PgxsrOz4eDgoNDu4OCAxMRElcckJiaqjM/KysLjx4/zjcntszDjrlmzBhYWFujRo4e8LfftfjNnzsTw4cOxZ88e1K9fH61bt8aNGzfyPO/Q0FBYWVnJNxcXlzxjiYiIiKj00Plk4LdfPCIKeFOeqvi329XpU5Nxw8PD0b9/f4V/rs59/faIESMwZMgQ1KtXDwsXLoSHhwfCw8PzzH/y5MlITk6Wb3fv3s0zloiIiIhKD5290tzW1hZSqVTpLnFSUpLS3eRcjo6OKuP19fVhY2OTb0xun5qOe+zYMVy7dg3R0dEK7U5OTgAALy8vhfYaNWrk+fAl8PpV2EZGRnnuJyIiIqLSSWd3rA0NDeHt7Y19+/YptO/btw++vr4qj/Hx8VGK37t3Lxo0aAADA4N8Y3L71HTcsLAweHt7o06dOgrtbm5ucHZ2xrVr1xTar1+/DldX17xOm4iIiIjKKJ3dsQaAkJAQBAYGokGDBvDx8cHq1atx584djBw5EsDraRP379/H2rVrAbxeAWTp0qUICQnB8OHDERMTg7CwMPlqHwAwZswYNG/eHN999x26deuG7du3Y//+/QoPFRY0bi6ZTIaNGzdiwYIFSrlLJBJ8+eWXmDFjBurUqYO6detizZo1+Oeff7Bp06aiuFxEREREVILptLAOCAjAkydPMHv2bCQkJKBWrVrYtWuX/I5vQkKCwrQKd3d37Nq1C2PHjsWyZcvg7OyMJUuWKKwt7evri6ioKEybNg3Tp09HlSpVEB0djcaNG6s9bq6oqCgIIdC3b1+V+QcHByM9PR1jx47F06dPUadOHezbtw9VqlTR5mUiIiIiolJAp+tYU/GsYy2ys5F67jyyHj2Cvp0dTBt4QyKVvmvqRO+E61gTEVFpUeLXsabiIdu7F/+2boM7gwbhwfjxuDNoEP5t3QayvXuLdNzBgweje/fuSu2HDx+GRCLB8+fPFdo9PDxgaGiI+/fvKx3TokULSCQSSCQSGBkZoXr16pgzZw6ys7Pz7fNNf/zxB1q0aAELCwuYmpqiYcOGiIyMlO+fOXOmfIy8ttatW6N27drIyMhQ6HvXrl0wMDDAuXPncOvWLYVjypUrh+bNm+PIkSMK10ZV/+3bt1foNzY2Fr169YKDgwOMjY1RvXp1DB8+HNevX1c6P39/f0ilUpw6dUppX14/i1xubm5YtGiRyn1vn8+bW+5YkZGRCu0ODg7o0qUL/v777zzHJCIiKotYWJdhsr17cX9MMLLeWgEl6+FD3B8TXOTFtbqOHz+O9PR09OrVS6HYfdPw4cORkJCAa9euYfTo0Zg2bRrmz5+vVv8//vgjunXrBl9fX5w+fRp//fUX+vTpg5EjR2L8+PEAgPHjxyu8wr5ixYryqUK525YtW5CSkoIZM2bI+37+/Dk++eQTTJ06FQ0aNJC379+/HwkJCThy5AgsLS3RsWNHxMfHy/e3b99eoe+EhASFZwX++OMPNGnSBK9evcKvv/6Kq1evYt26dbCyssL06dMVzu/OnTuIiYnB559/jrCwMLWuiaZyz+fNzdvbW77f0tISCQkJePDgAXbu3ImXL1+iU6dOSr+EEBERlWU6nWNNmhFCQKSlqRebnY2H33wLqJrpIwQgAR5+OwdmPj5qTQuRmJjku774uwgLC0O/fv3g5+eHUaNGYcqUKUpjmZqawtHREQDw+eefY/v27di2bRsmTpyYb993797FuHHjEBwcjDlz5sjbx40bB0NDQ4wePRq9evVC48aNYW5uLt8vlUphYWEhHzNXZGQk/P390b17dzRu3BjBwcFwcnLCtGnTFOJsbGzg6OgIR0dHrFq1ChUrVsTevXsxYsQIAK+XXXy771ypqakYMmQIOnbsiK1bt8rb3d3d0bhxY6U78xEREejcuTM+/fRTNGrUCIsWLYKZmVm+10VTueeTF4lEIt/v5OSEsWPHomvXrrh27Rpq166t1VyIiIhKKhbWpYhIS8O1+t4FB6rV2es719cbNlIr3OPCeUhMTbUz9htSUlKwceNGnD59Gp6ennj58iUOHz6Mli1b5nuciYkJnj17VmD/mzZtQmZmpvzO9JtGjBiBKVOmYMOGDQoPt+anRYsW+OyzzzBo0CB8/fXX+O2333D+/Hno6+f9R8n0/69bZmamWmP8+eefePz4MSZMmKByv7W1tfy/hRCIiIjAsmXL4OnpierVq+O3337DkCFD1BqrKDx//hzr168HAPkymERERO8DTgWhIvPHH3/A3NxcYevQoYNCTFRUFKpVq4aaNWtCKpWiT58++U5nyMnJwZ49e/Dnn3+idevWBeZw/fp1WFlZyV/o8yZDQ0NUrlxZ5Zzl/ISGhkIikaBPnz6YM2cOatSokWfsy5cvMXnyZEilUvj5+cnbVV2br7/+GgBw48YNAICnp2eBuezfvx+pqalo164dAGDAgAFFMh3E19dXKd/cOe4AkJycDHNzc5iZmaFcuXKIiopC165d1ToHIiKisoJ3rEsRiYkJPC6cVys29dw53P1kRIFxLqtXwfSNucH5ja2pli1bYsWKFQptp0+fxoABA+Sfw8LCFD4PGDAAzZs3x/PnzxXuzC5fvhw///yzfM5uYGCgwlznwsrvVfZ5MTExwbhx4zB27FiMGTNGZYyvry/09PSQmpoKJycnREZGKkyJUHVtypcvL89JXWFhYQgICJDfMe/bty++/PJLXLt2DR4eHhqdV36io6OVfoGQvjGFyMLCAhcuXEBWVhaOHDmC77//HitXrtTa+ERERKUBC+tSRCKRqD0dw6xpU+g7OiLr4UPV86wlEug7OMCsadMiW3rPzMwMVatWVWi7d++e/L+vXLmC06dP4+zZswpzpbOzs7FhwwZ8+umn8rb+/ftj6tSpMDIygrOzs0JRl5/q1asjOTkZDx48gLOzs8K+jIwM3Lx5E61atdL43PT19SGVSvMsyqOjo+Hl5QVra2vY2Ngo7Vd1bd7MGQD++ecf+Pj45JnD06dPsW3bNmRmZioU6dnZ2QgPD8d3332nySnly8XFJc98AUBPT0++39PTE4mJiQgICMDRo0e1lgMREVFJx6kgZZREKoXDlMn//+Gt4u//PztMmazT9azDwsLQvHlzxMXF4eLFi/JtwoQJStMZrKysULVqVbi4uKhdVANAz549oa+vr/LtmStXrsTLly/zfAHQu3BxcUGVKlVUFtUF8ff3h62tLebNm6dyf+7Di7/++isqVqyodP0WLVqENWvWICsr611O4Z2MHTsWcXFxCg9fEhERlXW8Y12GWfr7A4sX4eGcUIUl9/QdHOAwZfLr/TqSmZmJdevWYfbs2ahVq5bCvmHDhmHevHmIi4tDnTp11O7z0qVLsLCwUGirW7cu5s2bh/Hjx8PY2BiBgYEwMDDA9u3bMWXKFIwbN07tBxe16dWrV0h8axlEfX192NrawszMDD///DN69eqFrl27YvTo0ahatSoeP36M3377DXfu3EFUVBTCwsLw8ccfK10/V1dXTJw4ETt37kS3bt0AvJ4DffHiRYW48uXLo1KlSgCA+/fvK+3P3QcAT548UcrX2to6zxe7WFpaYtiwYZgxYwa6d+9eZCvKEBERlSQsrMs4S39/WLRuXeLevHj06FE8efIEH330kdK+atWqoXbt2ggLC8OSJUvU7rN58+ZKbUIIjB07FlWqVMH8+fOxePFiZGdno2bNmlixYoXOVs/Ys2eP0gOVHh4e+OeffwAA3bp1w8mTJxEaGop+/fpBJpPBxcUFrVq1wjfffIPz588jLi4OP/30k1LfFhYW8Pf3R1hYmLywPnz4MOrVq6cQN2jQIPm64fPnz1daFzwiIgItWrQAALRp00ZpnA0bNqBPnz55nuOYMWOwZMkSbNy4Eb17987/ghAREZUBfKW5jhXHK82JSiJ+v4mIqLTgK82JiIiIiIoRC2siIiIiIi1gYU1EREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWkBX2n+HsjJEUi48RwvZa9gZmkEp2rW0NOT6DotIiIiojKFd6zLuP9ik7B2yklsWxiLfWFXsG1hLNZOOYn/YpOKdNzBgwdDIpFg5MiRSvs+++wzSCQSDB48GADQokULBAcHK8Vt27YNEolEHiORSPLc3Nzc8o17O49Dhw6hY8eOsLGxgampKby8vDBu3Djcv39fnnt+GwCEhoaiYcOGsLCwgL29Pbp3745r165p7yISERFRqcLCugz7LzYJe1ZdxsvnrxTaXz5/hT2rLhd5ce3i4oKoqCikpaXJ29LT07FhwwZUqlRJo762bNmChIQEJCQk4MyZMwCA/fv3y9vOnj0rjx0+fLi8PXebN2+efP+qVavQpk0bODo6YvPmzbhy5QpWrlyJ5ORkLFiwAIsXL1Y4FgAiIiKU2o4cOYJRo0bh1KlT2LdvH7KysuDv74+XL18W+poRERFR6cWpIKWIEAJZGTlqxebkCByLvp5vzLHoG6joWV6taSH6hnryO7Xqql+/Pm7evIktW7agf//+AF4XyC4uLqhcubJGfZUvX17+3+np6QAAGxsbODo6KsWampqqbAeAe/fuYfTo0Rg9ejQWLlwob3dzc0Pz5s3x/PlzWFlZwcrKSuE4a2trpT737Nmj8DkiIgL29vY4f/48mjdvrtH5ERERUenHwroUycrIweoxR7TW38vnr/Dz2KNqxX6y2A8GRlKNxxgyZAgiIiLkhXV4eDiGDh2Kw4cPa9yXNmzcuBEZGRmYMGGCyv3W1taF7js5ORmA4i8BRERE9P7gVBAqUoGBgTh+/Dhu3bqF27dv48SJExgwYECRjrl8+XKYm5srbGvWrAEA3LhxA5aWlnByctLqmEIIhISE4MMPP0StWrW02jcRERGVDrxjXYroG+rhk8V+asU+uPEcfyyNKzCu8+d14FzNWq2xC8PW1hadOnXCmjVrIIRAp06dYGtrW6i+1NW/f39MnTpVoc3e3h7A6wJY0ykt6vj888/x119/4fjx41rvm4iIiEoHFtaliEQiUXs6hotXeZhZGyk9uPgm83JGcPFSb471uxg6dCg+//xzAMCyZcuU9ltaWsqnUbzp+fPnsLS01Hg8KysrVK1aVeW+6tWrIzk5GQkJCVq7a/3FF19gx44dOHr0KCpWrKiVPomIiKj04VSQMkpPT4JmAdXyjfmwd7ViWc+6ffv2yMjIQEZGBtq1a6e039PTE+fOnVNqP3v2LDw8PLSay8cffwxDQ0OFVULe9Pz5c7X7EkLg888/x5YtW3Dw4EG4u7trKUsiIiIqjXjHugyrUs8e7UfUwrHoGwp3rs3LGeHD3tVQpZ59seQhlUpx9epV+X+/7bPPPsPSpUsxatQofPLJJzAxMcG+ffsQFhaGdevWaTxeamoqEhMTFdqMjIxQrlw5uLi4YOHChfj8888hk8kwcOBAuLm54d69e1i7di3Mzc2xYMECtcYZNWoU1q9fj+3bt8PCwkI+ppWVFUxMTDTOm4iIiEo3FtZlXJV69nCvY6fzNy/mN6XDzc0Nx44dw9SpU+Hv74/09HRUr14dkZGR6NWrl8Zj/fTTT/jpp58U2tq1aydfHu+zzz5D9erVMX/+fHz00UdIS0uDm5sbOnfujJCQELXHWbFiBYDXL6V5U0REhPzlN0RERPT+kAghhK6TeJ/JZDJYWVkhOTlZqfhMT09HfHw83N3dYWxsrKMMiYoGv99ERFRa5FevvYlzrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMK6FODzpVQW8XtNRERlDQvrEszAwADA63WZicqa3O917veciIiotOM61iWYVCqFtbU1kpKSAACmpqaQSIp3/WkibRNCIDU1FUlJSbC2tlb50iAiIqLSiIV1Cefo6AgA8uKaqKywtraWf7+JiIjKAhbWJZxEIoGTkxPs7e2RmZmp63SItMLAwIB3qomIqMxhYV1KSKVSFiJEREREJRgfXiQiIiIi0gIW1kREREREWsDCmoiIiIhIC1hYExERERFpAQtrIiIiIiItYGFNRERERKQFLKyJiIiIiLSAhTURERERkRawsCYiIiIi0gIW1kREREREWsDCmoiIiIhIC1hYExERERFpAQtrIiIiIiItYGFNRERERKQFLKyJiIiIiLSAhTURERERkRawsCYiIiIi0gIW1kREREREWsDCmoiIiIhIC1hYExERERFpAQtrIiIiIiItYGFNRERERKQFOi+sly9fDnd3dxgbG8Pb2xvHjh3LN/7IkSPw9vaGsbExKleujJUrVyrFbN68GV5eXjAyMoKXlxe2bt2q8bgSiUTl9v333yv1JYRAhw4dIJFIsG3bNs0uABERERGVCTotrKOjoxEcHIypU6ciNjYWzZo1Q4cOHXDnzh2V8fHx8ejYsSOaNWuG2NhYTJkyBaNHj8bmzZvlMTExMQgICEBgYCDi4uIQGBiI3r174/Tp0xqNm5CQoLCFh4dDIpGgZ8+eSnktWrQIEolEi1eGiIiIiEobiRBC6Grwxo0bo379+lixYoW8rUaNGujevTtCQ0OV4idOnIgdO3bg6tWr8raRI0ciLi4OMTExAICAgADIZDLs3r1bHtO+fXuUK1cOGzZsKNS4ANC9e3ekpKTgwIEDCu1xcXHo3Lkzzp49CycnJ2zduhXdu3dX+xrIZDJYWVkhOTkZlpaWah9HRERERMVD3XpNZ3esMzIycP78efj7+yu0+/v74+TJkyqPiYmJUYpv164dzp07h8zMzHxjcvsszLgPHz7Ezp07ERQUpNCempqKvn37YunSpXB0dCzgjF979eoVZDKZwkZEREREpZ/OCuvHjx8jOzsbDg4OCu0ODg5ITExUeUxiYqLK+KysLDx+/DjfmNw+CzPumjVrYGFhgR49eii0jx07Fr6+vujWrVsBZ/s/oaGhsLKykm8uLi5qH0tEREREJZfOH158e26yECLf+cqq4t9uV6dPTcYNDw9H//79YWxsLG/bsWMHDh48iEWLFuWZqyqTJ09GcnKyfLt7965GxxMRERFRyaSvq4FtbW0hlUqV7hInJSUp3U3O5ejoqDJeX18fNjY2+cbk9qnpuMeOHcO1a9cQHR2t0H7w4EH8999/sLa2Vmjv2bMnmjVrhsOHD6s8ByMjIxgZGancR0RERESll87uWBsaGsLb2xv79u1TaN+3bx98fX1VHuPj46MUv3fvXjRo0AAGBgb5xuT2qem4YWFh8Pb2Rp06dRTaJ02ahL/++gsXL16UbwCwcOFCREREFHD2RERERFTWvNMd61evXr3T3deQkBAEBgaiQYMG8PHxwerVq3Hnzh2MHDkSwOtpE/fv38fatWsBvF4BZOnSpQgJCcHw4cMRExODsLAw+WofADBmzBg0b94c3333Hbp164bt27dj//79OH78uNrj5pLJZNi4cSMWLFiglLujo6PKBxYrVaoEd3f3Ql8TIiIiIiqdNCqs//zzT2zYsAHHjh3DnTt3kJOTA1NTU9SvXx/+/v4YMmQInJ2d1e4vICAAT548wezZs5GQkIBatWph165dcHV1BfB6Lek315Z2d3fHrl27MHbsWCxbtgzOzs5YsmSJwtrSvr6+iIqKwrRp0zB9+nRUqVIF0dHRaNy4sdrj5oqKioIQAn379tXkMhERERHRe0itday3bduGiRMnIjk5GR07dkSjRo1QoUIFmJiY4OnTp7h8+TKOHTuGmJgYDB48GF9//TXs7OyKI/9Sj+tYExEREZVs6tZrahXWjRo1wvTp09GpUyfo6eU9Lfv+/ftYvHgxHBwcMG7cuMJl/p5hYU1ERERUsmm1sKaiw8KaiIiIqGTTyZsXz549q83uiIiIiIhKDY0L6xcvXiAtLU2h7eLFi+jSpQuaNGmitcSIiIiIiEoTtQvre/fuoWnTpvJXcYeEhCA1NRUDBw5Ew4YNYWRkpLCkHRERERHR+0Tt5fYmTZqEFy9eYPHixdi8eTMWL16MI0eOoE6dOrh+/TrXbiYiIiKi95rahfWhQ4fw22+/oWnTpvj444/h7OyMXr16YdKkSUWZHxERERFRqaD2VJDExERUqVIFwOu3DpqYmKBbt25FlhgRERERUWmi0cOLUqn0fwfq6cHY2FjrCRERERERlUZqTwURQqB169bQ1399SFpaGrp06QJDQ0OFuAsXLmg3QyIiIiKiUkDtwnrGjBkKnzkNhIiIiIjof/jmRR3jmxeJiIiISjZ16zW171gDwOnTp7Fjxw5kZmaiTZs28Pf3f+dEiYiIiIjKArUL661bt6JXr14wNjaGvr4+FixYgAULFiA4OLgI0yMiIiIiKh3UXhVkzpw5GDx4MJ4/f47nz59j1qxZ+Oabb4oyNyIiIiKiUkPtOdaWlpY4d+4cqlevDgB49eoVzMzMkJiYCFtb2yJNsizjHGsiIiKikk3dek3tO9YvXryAtbW1/LORkRFMTEwgk8neKVEiIiIiorJAo4cX//zzT1hZWck/5+Tk4MCBA7h8+bK8rWvXrtrLjoiIiIiolFB7KoieXsE3tyUSCbKzs985qfcJp4IQERERlWxaX24vJydHK4kREREREZVFas+xHjp0KFJSUooyFyIiIiKiUkvtwnrNmjVIS0srylyIiIiIiEottQtrvvmciIiIiChvahfWwOuHE4mIiIiISJlGy+1Vr169wOL66dOn75QQEREREVFppFFhPWvWLIV1rImIiIiI6DWNCus+ffrA3t6+qHIhIiIiIiq11J5jzfnVRERERER546ogRERERERawDcvEhERERFpgVp3rEeOHIm7d++q1WF0dDR+/fXXd0qKiIiIiKi0UeuOtZ2dHWrVqgVfX1907doVDRo0gLOzM4yNjfHs2TNcuXIFx48fR1RUFCpUqIDVq1cXdd5ERERERCWKRKg5eTopKQlhYWGIiorC5cuXFfZZWFigTZs2+OSTT+Dv718kiZZVMpkMVlZWSE5OhqWlpa7TISIiIqK3qFuvqV1Yv+n58+e4ffs20tLSYGtriypVqnDVkEJiYU1ERERUsqlbr2m0jnUua2trWFtbFzY3IiIiIqIyR+3l9oiIiIiIKG8srImIiIiItICFNRERERGRFrCwJiIiIiLSgkIV1llZWdi/fz9WrVqFlJQUAMCDBw/w4sULrSZHRERERFRaaLwqyO3bt9G+fXvcuXMHr169Qtu2bWFhYYF58+YhPT0dK1euLIo8iYiIiIhKNI3vWI8ZMwYNGjTAs2fPYGJiIm//6KOPcODAAa0mR0RERERUWmh8x/r48eM4ceIEDA0NFdpdXV1x//59rSVGRERERFSaaHzHOicnB9nZ2Urt9+7dg4WFhVaSIiIiIiIqbTQurNu2bYtFixbJP0skErx48QIzZsxAx44dtZkbEREREVGpIRFCCE0OePDgAVq2bAmpVIobN26gQYMGuHHjBmxtbXH06FHY29sXVa5lkrrvniciIiIi3VC3XtN4jrWzszMuXryIqKgonD9/Hjk5OQgKCkL//v0VHmYkIiIiInqfaHzH+ujRo/D19YW+vmJNnpWVhZMnT6J58+ZaTbCs4x1rIiIiopJN3XpN4znWLVu2xNOnT5Xak5OT0bJlS027IyIiIiIqEzQurIUQkEgkSu1PnjyBmZmZVpIiIiIiIipt1J5j3aNHDwCvVwEZPHgwjIyM5Puys7Px119/wdfXV/sZEhERERGVAmoX1lZWVgBe37G2sLBQeFDR0NAQTZo0wfDhw7WfIRERERFRKaB2YR0REQEAcHNzw/jx4zntg4iIiIjoDRqvCkLaxVVBiIiIiEq2IlvHGgA2bdqE3377DXfu3EFGRobCvgsXLhSmSyIiIiKiUk3jVUGWLFmCIUOGwN7eHrGxsWjUqBFsbGxw8+ZNdOjQoShyJCIiIiIq8TQurJcvX47Vq1dj6dKlMDQ0xIQJE7Bv3z6MHj0aycnJRZEjEREREVGJp3FhfefOHfmyeiYmJkhJSQEABAYGYsOGDdrNjoiIiIiolNC4sHZ0dMSTJ08AAK6urjh16hQAID4+HnwOkoiIiIjeVxoX1q1atcLvv/8OAAgKCsLYsWPRtm1bBAQE4KOPPtJ6gkREREREpYHGy+3l5OQgJycH+vqvFxT57bffcPz4cVStWhUjR46EoaFhkSRaVnG5PSIiIqKSTd16TavrWN+/fx8VKlTQVnfvBRbWRERERCWbuvWaxlNBVElMTMQXX3yBqlWraqM7IiIiIqJSR+3C+vnz5+jfvz/s7Ozg7OyMJUuWICcnB1999RUqV66MU6dOITw8vChzJSIiIiIqsdQurKdMmYKjR49i0KBBKF++PMaOHYvOnTvj+PHj2L17N86ePYu+fftqnMDy5cvh7u4OY2NjeHt749ixY/nGHzlyBN7e3jA2NkblypWxcuVKpZjNmzfDy8sLRkZG8PLywtatWzUeVyKRqNy+//57AMDTp0/xxRdfwMPDA6ampqhUqRLX8iYiIiJ6j6ldWO/cuRMRERGYP38+duzYASEEqlevjoMHD8LPz69Qg0dHRyM4OBhTp05FbGwsmjVrhg4dOuDOnTsq4+Pj49GxY0c0a9YMsbGxmDJlCkaPHo3NmzfLY2JiYhAQEIDAwEDExcUhMDAQvXv3xunTpzUaNyEhQWELDw+HRCJBz549AQAPHjzAgwcPMH/+fFy6dAmRkZHYs2cPgoKCCnUtiIiIiKh0U/vhRQMDA9y+fRvOzs4AAFNTU5w5cwa1atUq9OCNGzdG/fr1sWLFCnlbjRo10L17d4SGhirFT5w4ETt27MDVq1flbSNHjkRcXBxiYmIAAAEBAZDJZNi9e7c8pn379ihXrpz8BTaajgsA3bt3R0pKCg4cOJDn+WzcuBEDBgzAy5cv5aumFIQPLxIRERGVbFp/eDEnJwcGBgbyz1KpFGZmZoVOMCMjA+fPn4e/v79Cu7+/P06ePKnymJiYGKX4du3a4dy5c8jMzMw3JrfPwoz78OFD7Ny5s8C70bkXO7+i+tWrV5DJZAobEREREZV+6t1WBSCEwODBg2FkZAQASE9Px8iRI5WK6y1btqjV3+PHj5GdnQ0HBweFdgcHByQmJqo8JjExUWV8VlYWHj9+DCcnpzxjcvsszLhr1qyBhYUFevTokef5PHnyBF9//TVGjBiRZwwAhIaGYtasWfnGEBEREVHpo3ZhPWjQIIXPAwYM0EoCEolE4bMQQqmtoPi329XpU5Nxw8PD0b9/fxgbG6vcL5PJ0KlTJ3h5eWHGjBl55g4AkydPRkhIiMKxLi4u+R5DRERERCWf2oV1RESEVge2tbWFVCpVukuclJSkdDc5l6Ojo8p4fX192NjY5BuT26em4x47dgzXrl1DdHS0ypxSUlLQvn17mJubY+vWrQrTZVQxMjKS3/UnIiIiorJDKy+IKQxDQ0N4e3tj3759Cu379u2Dr6+vymN8fHyU4vfu3YsGDRrIC9q8YnL71HTcsLAweHt7o06dOkr7ZDIZ/P39YWhoiB07duR5R5uIiIiIyj6171gXhZCQEAQGBqJBgwbw8fHB6tWrcefOHYwcORLA62kT9+/fx9q1awG8XgFk6dKlCAkJwfDhwxETE4OwsDD5ah8AMGbMGDRv3hzfffcdunXrhu3bt2P//v04fvy42uPmkslk2LhxIxYsWKCUe0pKCvz9/ZGamopffvlF4UFEOzs7SKVSrV8vIiIiIiq5dFpYBwQE4MmTJ5g9ezYSEhJQq1Yt7Nq1C66urgBeryX95trS7u7u2LVrF8aOHYtly5bJ3wCZu7Y0APj6+iIqKgrTpk3D9OnTUaVKFURHR6Nx48Zqj5srKioKQgiVL745f/68fG3st1/lHh8fDzc3t3e+PkRERERUeqi9jjUVDa5jTURERFSyaX0dayIiIiIiyluhCut169ahadOmcHZ2xu3btwEAixYtwvbt27WaHBERERFRaaFxYb1ixQqEhISgY8eOeP78ObKzswEA1tbWWLRokbbzIyIiIiIqFTQurH/88Uf89NNPmDp1qsLKFw0aNMClS5e0mhwRERERUWmhcWEdHx+PevXqKbUbGRnh5cuXWkmKiIiIiKi00biwdnd3x8WLF5Xad+/eDS8vL23kRERERERU6mi8jvWXX36JUaNGIT09HUIInDlzBhs2bEBoaCh+/vnnosiRiIiIiKjE07iwHjJkCLKysjBhwgSkpqaiX79+qFChAhYvXow+ffoURY5ERERERCXeO70g5vHjx8jJyYG9vb02c3qv8AUxRERERCVbkb0gZtasWfjvv/8AALa2tiyqiYiIiIhQiMJ68+bNqF69Opo0aYKlS5fi0aNHRZEXEREREVGponFh/ddff+Gvv/5Cq1at8MMPP6BChQro2LEj1q9fj9TU1KLIkYiIiIioxHunOdYAcOLECaxfvx4bN25Eeno6ZDKZtnJ7L3CONREREVHJVmRzrN9mZmYGExMTGBoaIjMz8127IyIiIiIqlQpVWMfHx+Pbb7+Fl5cXGjRogAsXLmDmzJlITEzUdn5ERERERKWCxutY+/j44MyZM6hduzaGDBkiX8eaiIiIiOh9pnFh3bJlS/z888+oWbNmUeRDRERERFQqvfPDi/Ru+PAiERERUcmmbr2m1h3rkJAQfP311zAzM0NISEi+sT/88INmmRIRERERlQFqFdaxsbHyFT9iY2OLNCEiIiIiotKIU0F0jFNBiIiIiEq2IlvHeujQoUhJSVFqf/nyJYYOHappd0REREREZYLGhfWaNWuQlpam1J6Wloa1a9dqJSkiIiIiotJG7eX2ZDIZhBAQQiAlJQXGxsbyfdnZ2di1axfs7e2LJEkiIiIiopJO7cLa2toaEokEEokE1atXV9ovkUgwa9YsrSZHRERERFRaqF1YHzp0CEIItGrVCps3b0b58uXl+wwNDeHq6gpnZ+ciSZKIiIiIqKRTu7D28/MDAMTHx6NSpUqQSCRFlhQRERERUWmjVmH9119/oVatWtDT00NycjIuXbqUZ+wHH3ygteSIiIiIiEoLtQrrunXrIjExEfb29qhbty4kEglULX8tkUiQnZ2t9SSJiIiIiEo6tQrr+Ph42NnZyf+biIiIiIgUqVVYu7q6qvxvIiIiIiJ6rVAviNm5c6f884QJE2BtbQ1fX1/cvn1bq8kREREREZUWGhfWc+bMgYmJCQAgJiYGS5cuxbx582Bra4uxY8dqPUEiIiIiotJA7eX2ct29exdVq1YFAGzbtg0ff/wxPvnkEzRt2hQtWrTQdn5ERERERKWCxneszc3N8eTJEwDA3r170aZNGwCAsbEx0tLStJsdEREREVEpofEd67Zt22LYsGGoV68erl+/jk6dOgEA/v77b7i5uWk7PyIiIiKiUkHjO9bLli2Dj48PHj16hM2bN8PGxgYAcP78efTt21frCRIRERERlQYSoepNL1RsZDIZrKyskJycDEtLS12nQ0RERERvUbde03gqCAA8f/4cYWFhuHr1KiQSCWrUqIGgoCBYWVkVOmEiIiIiotJM46kg586dQ5UqVbBw4UI8ffoUjx8/xsKFC1GlShVcuHChKHIkIiIiIirxNJ4K0qxZM1StWhU//fQT9PVf3/DOysrCsGHDcPPmTRw9erRIEi2rOBWEiIiIqGRTt17TuLA2MTFBbGwsPD09FdqvXLmCBg0aIDU1tXAZv6dYWBMRERGVbOrWaxpPBbG0tMSdO3eU2u/evQsLCwtNuyMiIiIiKhM0LqwDAgIQFBSE6Oho3L17F/fu3UNUVBSGDRvG5faIiIiI6L2l8aog8+fPh0QiwcCBA5GVlQUAMDAwwKeffoq5c+dqPUEiIiIiotKg0OtYp6am4r///oMQAlWrVoWpqam2c3svcI41ERERUcmm9TnWqampGDVqFCpUqAB7e3sMGzYMTk5O+OCDD1hUExEREdF7T+3CesaMGYiMjESnTp3Qp08f7Nu3D59++mlR5kZEREREVGqoPcd6y5YtCAsLQ58+fQAAAwYMQNOmTZGdnQ2pVFpkCRIRERERlQZq37G+e/cumjVrJv/cqFEj6Ovr48GDB0WSGBERERFRaaJ2YZ2dnQ1DQ0OFNn19ffnKIERERERE7zO1p4IIITB48GAYGRnJ29LT0zFy5EiYmZnJ27Zs2aLdDImIiIiISgG1C+tBgwYptQ0YMECryRARERERlVZqF9YRERFFmQcRERERUamm8SvNiYiIiIhIGQtrIiIiIiItYGFNRERERKQFLKyJiIiIiLSAhTURERERkRYUqrBet24dmjZtCmdnZ9y+fRsAsGjRImzfvl2ryRERERERlRYaF9YrVqxASEgIOnbsiOfPnyM7OxsAYG1tjUWLFmk7PyIiIiKiUkHjwvrHH3/ETz/9hKlTp0IqlcrbGzRogEuXLmk1OSIiIiKi0kLjwjo+Ph716tVTajcyMsLLly+1khQRERERUWmjcWHt7u6OixcvKrXv3r0bXl5e2siJiIiIiKjUUfuV5rm+/PJLjBo1Cunp6RBC4MyZM9iwYQNCQ0Px888/F0WOREREREQlnsZ3rIcMGYIZM2ZgwoQJSE1NRb9+/bBy5UosXrwYffr00TiB5cuXw93dHcbGxvD29saxY8fyjT9y5Ai8vb1hbGyMypUrY+XKlUoxmzdvhpeXF4yMjODl5YWtW7dqPK5EIlG5ff/99/KYV69e4YsvvoCtrS3MzMzQtWtX3Lt3T+NrQERERESlX6GW2xs+fDhu376NpKQkJCYm4u7duwgKCtK4n+joaAQHB2Pq1KmIjY1Fs2bN0KFDB9y5c0dlfHx8PDp27IhmzZohNjYWU6ZMwejRo7F582Z5TExMDAICAhAYGIi4uDgEBgaid+/eOH36tEbjJiQkKGzh4eGQSCTo2bOnPCY4OBhbt25FVFQUjh8/jhcvXqBz587ylVKIiIiI6P0hEUIIXQ3euHFj1K9fHytWrJC31ahRA927d0doaKhS/MSJE7Fjxw5cvXpV3jZy5EjExcUhJiYGABAQEACZTIbdu3fLY9q3b49y5cphw4YNhRoXALp3746UlBQcOHAAAJCcnAw7OzusW7cOAQEBAIAHDx7AxcUFu3btQrt27VT28+rVK7x69Ur+WSaTwcXFBcnJybC0tMz/ghERERFRsZPJZLCysiqwXivUw4uVK1fOc1NXRkYGzp8/D39/f4V2f39/nDx5UuUxMTExSvHt2rXDuXPnkJmZmW9Mbp+FGffhw4fYuXOnwl358+fPIzMzU6EfZ2dn1KpVK89+ACA0NBRWVlbyzcXFJc9YIiIiIio9NH54MTg4WOFzZmYmYmNjsWfPHnz55Zdq9/P48WNkZ2fDwcFBod3BwQGJiYkqj0lMTFQZn5WVhcePH8PJySnPmNw+CzPumjVrYGFhgR49eijkYmhoiHLlyqndDwBMnjwZISEh8s+5d6yJiIiIqHTTuLAeM2aMyvZly5bh3LlzGicgkUgUPgshlNoKin+7XZ0+NRk3PDwc/fv3h7GxcZ55qZu/kZERjIyMCuyHiIiIiEqXQj28qEqHDh0UHiIsiK2tLaRSqdLd3aSkJKW7ybkcHR1Vxuvr68PGxibfmNw+NR332LFjuHbtGoYNG6aUS0ZGBp49e6Z2/kRERERUdmmtsN60aRPKly+vdryhoSG8vb2xb98+hfZ9+/bB19dX5TE+Pj5K8Xv37kWDBg1gYGCQb0xun5qOGxYWBm9vb9SpU0eh3dvbGwYGBgr9JCQk4PLly3nmT0RERERll8ZTQerVq6cw1UEIgcTERDx69AjLly/XqK+QkBAEBgaiQYMG8PHxwerVq3Hnzh2MHDkSwOv5yPfv38fatWsBvF4BZOnSpQgJCcHw4cMRExODsLAw+WofwOupKs2bN8d3332Hbt26Yfv27di/fz+OHz+u9ri5ZDIZNm7ciAULFijlbmVlhaCgIIwbNw42NjYoX748xo8fj9q1a6NNmzYaXQciIiIiKv00Lqy7d++u8FlPTw92dnZo0aIFPD09NeorICAAT548wezZs5GQkIBatWph165dcHV1BfD6DvCba0u7u7tj165dGDt2LJYtWwZnZ2csWbJEYW1pX19fREVFYdq0aZg+fTqqVKmC6OhoNG7cWO1xc0VFRUEIgb59+6rMf+HChdDX10fv3r2RlpaG1q1bIzIyElKpVKPrQERERESln0brWGdlZeHXX39Fu3bt4OjoWJR5vTfUXReRiIiIiHSjSNax1tfXx6effqrwghMiIiIiIirEw4uNGzdGbGxsUeRCRERERFRqaTzH+rPPPsO4ceNw7949eHt7w8zMTGH/Bx98oLXkiIiIiIhKC7XnWA8dOhSLFi2CtbW1cicSifzFKNnZ2drOsUzjHGsiIiKikk3dek3twloqlSIhIQFpaWn5xr29sgblj4U1ERERUcmmbr2m9lSQ3PqbhTMRERERkTKNHl5888UwRERERET0Pxo9vFi9evUCi+unT5++U0JERERERKWRRoX1rFmzYGVlVVS5EBERERGVWhoV1n369IG9vX1R5UJEREREVGqpPcea86uJiIiIiPKmdmGt5qp8RERERETvJbWnguTk5BRlHkREREREpZpGy+0REREREZFqLKyJiIiIiLSAhTURERERkRawsCYiIiIi0gIW1kREREREWsDCmoiIiIhIC1hYExERERFpAQtrIiIiIiItYGFNRERERKQFLKyJiIiIiLSAhTURERERkRawsCYiIiIi0gIW1kREREREWsDCmoiIiIhIC1hYExERERFpAQtrIiIiIiItYGFNRERERKQF+rpOgIiIXsvOETgT/xRJKemwtzBGI/fykOpJdJ0WERGpiYU1EVEJsOdyAmb9fgUJyenyNicrY8zo4oX2tZx0mBkREamLU0GIiHRsz+UEfPrLBYWiGgASk9Px6S8XsOdygo4yIyIiTbCwJiLSoewcgVm/X4FQsS+3bdbvV5CdoyqCiIhKEhbWREQ6dCb+qdKd6jcJAAnJ6TgT/7T4kiIiokJhYU1EpENJKXkX1YWJIyIi3WFhTUSkQ/YWxlqNIyIi3WFhTUSkQ43cy8PJyhh5LaonwevVQRq5ly/OtIiIqBBYWBMR6ZBUT4IZXbwAQKm4zv08o4sX17MmIioFWFgTEelY+1pOWDGgPhytFKd7OFoZY8WA+lzHmoiolOALYoiISoD2tZzQ1suRb14kIirFWFgTEZUQUj0JfKrY6DoNIiIqJE4FISIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWkBC2siIiIiIi1gYU1EREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWkBC2siIiIiIi1gYU1EREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWkBC2siIiIiIi1gYU1EREREpAU6L6yXL18Od3d3GBsbw9vbG8eOHcs3/siRI/D29oaxsTEqV66MlStXKsVs3rwZXl5eMDIygpeXF7Zu3Vqoca9evYquXbvCysoKFhYWaNKkCe7cuSPfn5iYiMDAQDg6OsLMzAz169fHpk2bCnEViIiIiKi002lhHR0djeDgYEydOhWxsbFo1qwZOnTooFC8vik+Ph4dO3ZEs2bNEBsbiylTpmD06NHYvHmzPCYmJgYBAQEIDAxEXFwcAgMD0bt3b5w+fVqjcf/77z98+OGH8PT0xOHDhxEXF4fp06fD2NhYHhMYGIhr165hx44duHTpEnr06IGAgADExsYWwdUiIiIiopJMIoQQuhq8cePGqF+/PlasWCFvq1GjBrp3747Q0FCl+IkTJ2LHjh24evWqvG3kyJGIi4tDTEwMACAgIAAymQy7d++Wx7Rv3x7lypXDhg0b1B63T58+MDAwwLp16/LM39zcHCtWrEBgYKC8zcbGBvPmzUNQUJBa10Amk8HKygrJycmwtLRU6xgiIiIiKj7q1ms6u2OdkZGB8+fPw9/fX6Hd398fJ0+eVHlMTEyMUny7du1w7tw5ZGZm5huT26c64+bk5GDnzp2oXr062rVrB3t7ezRu3Bjbtm1TOObDDz9EdHQ0nj59ipycHERFReHVq1do0aJFnuf96tUryGQyhY2IiIiISj+dFdaPHz9GdnY2HBwcFNodHByQmJio8pjExESV8VlZWXj8+HG+Mbl9qjNuUlISXrx4gblz56J9+/bYu3cvPvroI/To0QNHjhyRHxMdHY2srCzY2NjAyMgII0aMwNatW1GlSpU8zzs0NBRWVlbyzcXFJb/LRERERESlhM4fXpRIJAqfhRBKbQXFv92uTp/5xeTk5AAAunXrhrFjx6Ju3bqYNGkSOnfurPCw5LRp0/Ds2TPs378f586dQ0hICHr16oVLly7lmf/kyZORnJws3+7evZtnLBERERGVHvq6GtjW1hZSqVTp7nRSUpLS3eRcjo6OKuP19fVhY2OTb0xun+qMa2trC319fXh5eSnE1KhRA8ePHwfw+uHGpUuX4vLly6hZsyYAoE6dOjh27BiWLVumcrUSADAyMoKRkZHqi0JEREREpZbO7lgbGhrC29sb+/btU2jft28ffH19VR7j4+OjFL937140aNAABgYG+cbk9qnOuIaGhmjYsCGuXbumEHP9+nW4uroCAFJTUwEAenqKl1AqlcrveBMRERHRe0ToUFRUlDAwMBBhYWHiypUrIjg4WJiZmYlbt24JIYSYNGmSCAwMlMffvHlTmJqairFjx4orV66IsLAwYWBgIDZt2iSPOXHihJBKpWLu3Lni6tWrYu7cuUJfX1+cOnVK7XGFEGLLli3CwMBArF69Wty4cUP8+OOPQiqVimPHjgkhhMjIyBBVq1YVzZo1E6dPnxb//vuvmD9/vpBIJGLnzp1qX4Pk5GQBQCQnJxf6OhIRERFR0VG3XtNpYS2EEMuWLROurq7C0NBQ1K9fXxw5ckS+b9CgQcLPz08h/vDhw6JevXrC0NBQuLm5iRUrVij1uXHjRuHh4SEMDAyEp6en2Lx5s0bj5goLCxNVq1YVxsbGok6dOmLbtm0K+69fvy569Ogh7O3thampqfjggw/E2rVrNTp/FtZEREREJZu69ZpO17EmrmNNREREVNKV+HWsiYiIiIjKEhbWRERERERawMKaiIiIiEgLWFgTEREREWkBC2siIiIiIi1gYU1EREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWkBC2siIiIiIi1gYU1EREREpAUsrImIiIiItEBf1wkQke6I7GyknjuPrEePoG9nB9MG3pBIpbpOi4iIqFRiYU30npLt3YuHc0KRlZgob9N3dITDlMmw9PfXYWZERESlE6eCEL2HZHv34v6YYIWiGgCyHj7E/THBkO3dq6PMiIiISi8W1kTvGZGdjYdzQgEhVOx83fZwTihEdnYxZ0ZERFS6sbAmes+knjuvdKdagRDISkxE6rnzxZcUERFRGcDCmug9k/XokVbjiIiI6DUW1kTvGX07O63GERER0WssrIneM6YNvKHv6AhIJKoDJBLoOzrCtIF38SZGRERUyrGwJnrPSKRSOEyZ/P8f3iqu//+zw5TJXM+aiIhIQyysicqgnByB+9ee4frZRNy/9gw5OYorgFj6+6PC4kXQd3BQaNd3cECFxYu4jjUREVEh8AUxRGXMf7FJOBZ9Ay+fv5K3mVkboVlANVSpZy9vs/T3h0Xr1nzzIhERkZZIhFC1mC0VF5lMBisrKyQnJ8PS0lLX6VAp919sEvasupzn/vYjaikU10RERFQwdes1TgUhKiNycgSORd/IN+b4bzeUpoUQERGRdrCwJiojEm48V5j+ocqLZ6+QcON58SRERET0nmFhTVRGvJTlX1RrGkdERESaYWFNVEaYWRppNY6IiIg0w8KaqIxwqmYNM+v8i2bzckZwqmZdPAkRERG9Z1hYE5URenoSNAuolm/Mh72rQU8vjzcuEhER0TthYU1UhlSpZ4/2I2op3bk2L2fEpfaIiIiKGF8QQ1TGVKlnD/c6dq9XCZG9gpnl6+kfvFNNRERUtFhYE5VBenoSVPAop+s0iIiI3iucCkJEREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWkBC2siIiIiIi1gYU1EREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgLWFgTEREREWkBC2siIiIiIi1gYU1EREREpAUsrImIiIiItICFNRERERGRFrCwJiIiIiLSAhbWRERERERawMKaiIiIiEgL9HWdABWf7JxsXEi6gEepj2Bnaof69vUh1ZPqOi0iIiKiMoGF9Xti/+39mHtmLh6mPpS3OZg6YFKjSWjj2kaHmRERERGVDZwK8h7Yf3s/Qg6HKBTVAJCUmoSQwyHYf3u/jjIjIiIiKjtYWJdx2TnZmHtmLgSE0r7ctu/OfIfsnOziTo2IiIioTGFhXcZdSLqgdKf6TQICiamJuJB0oRizIiIiIip7WFiXcY9SH2k1joiIiIhU03lhvXz5cri7u8PY2Bje3t44duxYvvFHjhyBt7c3jI2NUblyZaxcuVIpZvPmzfDy8oKRkRG8vLywdevWQo179epVdO3aFVZWVrCwsECTJk1w584dhZiYmBi0atUKZmZmsLa2RosWLZCWlqbhVSg6dqZ2Wo0jIiIiItV0WlhHR0cjODgYU6dORWxsLJo1a4YOHTooFa+54uPj0bFjRzRr1gyxsbGYMmUKRo8ejc2bN8tjYmJiEBAQgMDAQMTFxSEwMBC9e/fG6dOnNRr3v//+w4cffghPT08cPnwYcXFxmD59OoyNjRXGat++Pfz9/XHmzBmcPXsWn3/+OfT0dP77ilx9+/pwMHWABBKV+yWQwNHUEfXt6xdzZkRERERli0QIofxUWzFp3Lgx6tevjxUrVsjbatSoge7duyM0NFQpfuLEidixYweuXr0qbxs5ciTi4uIQExMDAAgICIBMJsPu3bvlMe3bt0e5cuWwYcMGtcft06cPDAwMsG7dujzzb9KkCdq2bYuvv/66kFcAkMlksLKyQnJyMiwtLQvdT35yVwUBoPAQY26x/UOLH7jkHhEREVEe1K3XdHZrNSMjA+fPn4e/v79Cu7+/P06ePKnymJiYGKX4du3a4dy5c8jMzMw3JrdPdcbNycnBzp07Ub16dbRr1w729vZo3Lgxtm3bJo9PSkrC6dOnYW9vD19fXzg4OMDPzw/Hjx/P97xfvXoFmUymsBW1Nq5t8EOLH2Bvaq/Q7mDqwKKaiIiISEt0Vlg/fvwY2dnZcHBwUGh3cHBAYmKiymMSExNVxmdlZeHx48f5xuT2qc64SUlJePHiBebOnYv27dtj7969+Oijj9CjRw8cOXIEAHDz5k0AwMyZMzF8+HDs2bMH9evXR+vWrXHjxo08zzs0NBRWVlbyzcXFJd/rpC1tXNvgz55/IrxdOL5r9h3C24VjT889LKqJiIiItETnb16USBTn/gohlNoKin+7XZ0+84vJyckBAHTr1g1jx44FANStWxcnT57EypUr4efnJ48ZMWIEhgwZAgCoV68eDhw4gPDwcJVTWQBg8uTJCAkJkX+WyWTFVlxL9aRo6NiwWMYiIiIiet/orLC2tbWFVCpVujudlJSkdDc5l6Ojo8p4fX192NjY5BuT26c649ra2kJfXx9eXl4KMTVq1JBP9XBycgIAlTF5PXwJAEZGRjAyMspzPxERERGVTjqbCmJoaAhvb2/s27dPoX3fvn3w9fVVeYyPj49S/N69e9GgQQMYGBjkG5PbpzrjGhoaomHDhrh27ZpCzPXr1+Hq6goAcHNzg7Ozc74xRERERPQeEToUFRUlDAwMRFhYmLhy5YoIDg4WZmZm4tatW0IIISZNmiQCAwPl8Tdv3hSmpqZi7Nix4sqVKyIsLEwYGBiITZs2yWNOnDghpFKpmDt3rrh69aqYO3eu0NfXF6dOnVJ7XCGE2LJlizAwMBCrV68WN27cED/++KOQSqXi2LFj8piFCxcKS0tLsXHjRnHjxg0xbdo0YWxsLP7991+1r0FycrIAIJKTkwt1DYmIiIioaKlbr+m0sBZCiGXLlglXV1dhaGgo6tevL44cOSLfN2jQIOHn56cQf/jwYVGvXj1haGgo3NzcxIoVK5T63Lhxo/Dw8BAGBgbC09NTbN68WaNxc4WFhYmqVasKY2NjUadOHbFt2zalmNDQUFGxYkVhamoqfHx8FApvdbCwJiIiIirZ1K3XdLqONRXPOtZEREREVHglfh1rIiIiIqKyhIU1EREREZEWsLAmIiIiItICFtZERERERFrAwpqIiIiISAtYWBMRERERaQELayIiIiIiLWBhTURERESkBfq6TuB9l/t+HplMpuNMiIiIiEiV3DqtoPcqsrDWsZSUFACAi4uLjjMhIiIiovykpKTAysoqz/18pbmO5eTk4MGDB7CwsIBEItF1Olojk8ng4uKCu3fv8lXteeA1KhivUf54fQrGa1QwXqP88foU7H24RkIIpKSkwNnZGXp6ec+k5h1rHdPT00PFihV1nUaRsbS0LLN/yLSF16hgvEb54/UpGK9RwXiN8sfrU7Cyfo3yu1Odiw8vEhERERFpAQtrIiIiIiItYGFNRcLIyAgzZsyAkZGRrlMpsXiNCsZrlD9en4LxGhWM1yh/vD4F4zX6Hz68SERERESkBbxjTURERESkBSysiYiIiIi0gIU1EREREZEWsLAmIiIiItICFtaksRUrVuCDDz6QLwTv4+OD3bt353vMr7/+ijp16sDU1BROTk4YMmQInjx5UkwZF7/CXKNly5ahRo0aMDExgYeHB9auXVtM2epeaGgoJBIJgoOD8407cuQIvL29YWxsjMqVK2PlypXFk2AJoM41SkhIQL9+/eDh4QE9Pb0Cr2dZo8412rJlC9q2bQs7Ozv5n80///yz+JLUIXWuz/Hjx9G0aVPY2NjAxMQEnp6eWLhwYfElqWPq/l2U68SJE9DX10fdunWLNK+SQp3rc/jwYUgkEqXtn3/+Kb5EdYiFNWmsYsWKmDt3Ls6dO4dz586hVatW6NatG/7++2+V8cePH8fAgQMRFBSEv//+Gxs3bsTZs2cxbNiwYs68+Gh6jVasWIHJkydj5syZ+PvvvzFr1iyMGjUKv//+ezFnXvzOnj2L1atX44MPPsg3Lj4+Hh07dkSzZs0QGxuLKVOmYPTo0di8eXMxZao76l6jV69ewc7ODlOnTkWdOnWKKbuSQd1rdPToUbRt2xa7du3C+fPn0bJlS3Tp0gWxsbHFlKluqHt9zMzM8Pnnn+Po0aO4evUqpk2bhmnTpmH16tXFlKnuqHuNciUnJ2PgwIFo3bp1EWdWMmh6fa5du4aEhAT5Vq1atSLOsIQQRFpQrlw58fPPP6vc9/3334vKlSsrtC1ZskRUrFixOFIrMfK7Rj4+PmL8+PEKbWPGjBFNmzYtjtR0JiUlRVSrVk3s27dP+Pn5iTFjxuQZO2HCBOHp6anQNmLECNGkSZMizlK3NLlGb9IktrQr7DXK5eXlJWbNmlU0yZUA73p9PvroIzFgwICiSa6EKMw1CggIENOmTRMzZswQderUKfIcdUmT63Po0CEBQDx79qzY8itJeMea3kl2djaioqLw8uVL+Pj4qIzx9fXFvXv3sGvXLggh8PDhQ2zatAmdOnUq5mx1Q51r9OrVKxgbGyu0mZiY4MyZM8jMzCyONHVi1KhR6NSpE9q0aVNgbExMDPz9/RXa2rVrh3PnzvEavefe5Rrl5OQgJSUF5cuXL4LMSoZ3uT6xsbE4efIk/Pz8iiCzkkPTaxQREYH//vsPM2bMKOLMSobCfIfq1asHJycntG7dGocOHSrC7EoWfV0nQKXTpUuX4OPjg/T0dJibm2Pr1q3w8vJSGevr64tff/0VAQEBSE9PR1ZWFrp27Yoff/yxmLMuXppco3bt2uHnn39G9+7dUb9+fZw/fx7h4eHIzMzE48eP4eTkVMzZF72oqChcuHABZ8+eVSs+MTERDg4OCm0ODg7IysriNXqPves1WrBgAV6+fInevXtrObOSobDXp2LFinj06BGysrIwc+bMMj11T9NrdOPGDUyaNAnHjh2Dvn7ZL6M0vT5OTk5YvXo1vL298erVK6xbtw6tW7fG4cOH0bx58yLOVvfK/jeCioSHhwcuXryI58+fY/PmzRg0aBCOHDmisnC8cuUKRo8eja+++grt2rVDQkICvvzyS4wcORJhYWE6yL54aHKNpk+fjsTERDRp0gRCCDg4OGDw4MGYN28epFKpDrIvWnfv3sWYMWOwd+9epTv1+ZFIJAqfxf+/OPbt9rKgsNfoffKu12jDhg2YOXMmtm/fDnt7+yLIULfe5focO3YML168wKlTpzBp0iRUrVoVffv2LaJMdUfTa5SdnY1+/fph1qxZqF69ejFkqFuF+Q55eHjAw8ND/tnHxwd3797F/Pnz34vCmnOsSStat24tPvnkE5X7BgwYID7++GOFtmPHjgkA4sGDB8WRXomQ3zXKlZGRIe7evSuysrLE8uXLhYWFhcjOzi6mDIvP1q1bBQAhlUrlGwAhkUiEVCoVWVlZSsc0a9ZMjB49WqFty5YtQl9fX2RkZBRX6sWmMNfoTe/DHOt3uUZRUVHCxMRE/PHHH8WYcfF61+9Qrq+//lpUr169iLPVDU2v0bNnz5TiJRKJvO3AgQM6OpOioa3v0DfffKP0jExZxTvWpBVCCLx69UrlvtTUVKV/Lsu9Cyv+/47j+yC/a5TLwMAAFStWBPD6n986d+4MPb2y9yhE69atcenSJYW2IUOGwNPTExMnTlR5l97Hx0dplZS9e/eiQYMGMDAwKNJ8daEw1+h9U9hrtGHDBgwdOhQbNmwo0896aOs7pM7fXaWVptfI0tJSKX758uU4ePAgNm3aBHd39yLPuThp6zsUGxtbJqfrqcLCmjQ2ZcoUdOjQAS4uLkhJSUFUVBQOHz6MPXv2AAAmT56M+/fvy9dh7tKlC4YPH44VK1bIp4IEBwejUaNGcHZ21uWpFBlNr9H169dx5swZNG7cGM+ePcMPP/yAy5cvY82aNbo8jSJjYWGBWrVqKbSZmZnBxsZG3v72NRo5ciSWLl2KkJAQDB8+HDExMQgLC8OGDRuKPf/iUJhrBAAXL14EALx48QKPHj3CxYsXYWhomOf8/tKsMNdow4YNGDhwIBYvXowmTZogMTERwOuHha2srIr3BIpYYa7PsmXLUKlSJXh6egJ4vVzq/Pnz8cUXXxRv8sVE02ukp6enFG9vbw9jY2Ol9rKgMN+hRYsWwc3NDTVr1kRGRgZ++eUXbN68+b1YGhVgYU2F8PDhQwQGBiIhIQFWVlb44IMPsGfPHrRt2xbA65dU3LlzRx4/ePBgpKSkYOnSpRg3bhysra3RqlUrfPfdd7o6hSKn6TXKzs7GggULcO3aNRgYGKBly5Y4efIk3NzcdHQGuvf2NXJ3d8euXbswduxYLFu2DM7OzliyZAl69uypwyx16+1rBLx+Ej/X+fPnsX79eri6uuLWrVvFnF3J8PY1WrVqFbKysjBq1CiMGjVK3j5o0CBERkbqIEPdevv65OTkYPLkyYiPj4e+vj6qVKmCuXPnYsSIETrMUrdU/Tmj/3n7+mRkZGD8+PG4f/8+TExMULNmTezcuRMdO3bUYZbFRyLep3+LJyIiIiIqImVv8iYRERERkQ6wsCYiIiIi0gIW1kREREREWsDCmoiIiIhIC1hYExERERFpAQtrIiIiIiItYGFNRERERKQFLKyJiIiIqMQ5evQounTpAmdnZ0gkEmzbtq1Ix5s5cyYkEonC5ujoqFEfLKyJiN5BZGQkrK2tdZ1Gobm5uWHRokX5xsycORN169YtlnxKmoMHD8LT0xM5OTnFOu6lS5dQsWJFvHz5sljHJSpJXr58iTp16mDp0qXFNmbNmjWRkJAg3y5duqTR8Sysiei9N3jwYKW7FBKJBP/++6+uU0NkZKRCTk5OTujduzfi4+O10v/Zs2fxySefyD+ruis0fvx4HDhwQCvj5eXt83RwcECXLl3w999/a9yPNn/RmTBhAqZOnQo9PT2VeeZuP//8s8r9qn5ebm5u8v0mJibw9PTE999/jzdfhFy7dm00atQICxcu1Nq5EJU2HTp0wDfffIMePXqo3J+RkYEJEyagQoUKMDMzQ+PGjXH48OF3GlNfXx+Ojo7yzc7OTqPjWVgTEQFo3769wl2KhIQEuLu76zotAIClpSUSEhLw4MEDrF+/HhcvXkTXrl2RnZ39zn3b2dnB1NQ03xhzc3PY2Ni881gFefM8d+7ciZcvX6JTp07IyMgo8rFVOXnyJG7cuIFevXqpzPPNrX///kr78/t5zZ49GwkJCbh69SrGjx+PKVOmYPXq1QrjDBkyBCtWrNDKz5moLBoyZAhOnDiBqKgo/PXXX+jVqxfat2+PGzduFLrPGzduwNnZGe7u7ujTpw9u3ryp0fEsrImIABgZGSncpXB0dIRUKsUPP/yA2rVrw8zMDC4uLvjss8/w4sWLPPuJi4tDy5YtYWFhAUtLS3h7e+PcuXPy/SdPnkTz5s1hYmICFxcXjB49usB/7s+d5+fk5ISWLVtixowZuHz5svyO+ooVK1ClShUYGhrCw8MD69atUzh+5syZqFSpEoyMjODs7IzRo0fL9705FcTNzQ0A8NFHH0Eikcg/vzkV5M8//4SxsTGeP3+uMMbo0aPh5+entfNs0KABxo4di9u3b+PatWvymPx+HocPH8aQIUOQnJwsvyM8c+ZMAIW7sxUVFQV/f38YGxurzPPNzcTEROV5qPp54f/au/+YqOs/gONP7uIWAfkrZjUQFnhhBXasH3jEpnaLvCaTgpySYIuSQnMKWo01clnNGKj1RzLX7tQhhAGtZs0M9RT7MYScEZc7DVkhzMDSGQSh7+8ffvmMD9wh0H37rnw9ts8f78+9P+/3633v7Xjdm/fnc0BoaCi33norUVFR5OTkEB8fz+eff67rJyUlhe7ublwu16hxCnE9On36NBUVFezZs4fk5GSio6MpKCjgoYcewuFwTKjNBx98kJ07d7Jv3z62b99OZ2cnVquV7u7uMbchibUQQozCYDDwzjvv0NzczI4dOzhw4ADr16/3WT8zM5Pw8HAaGhpobGzk5ZdfJjAwELi6bzYlJYXHH3+cEydO8MEHH1BfX8/KlSvHFdNgEvfnn39SW1vL6tWryc/Pp7m5mRUrVvD0009z8OBBAD788EM2b95MWVkZHo+Hjz76iLi4OK/tNjQ0AOBwOOjo6NDKQ9lsNiZPnkx1dbV27vLly1RVVWmrtv4Y52+//cbu3bsBtPcPRp8Pq9XKli1bdCvKBQUFwMRWtg4fPsx999035ph9GTpfwymlOHToEG63WzdOAJPJxOzZszly5MhfjkGIf5umpiaUUpjNZkJCQrTD5XJx+vRpAM6cOeN169bQY+jn0oIFC3jiiSeIi4vDZrOxd+9eAHbs2DH2wJQQQlznsrOzldFoVMHBwdqRnp7utW5VVZWaNm2aVnY4HGrSpElaOTQ0VDmdTq/XLlu2TD333HO6c0eOHFEGg0H19vZ6vWZ4+z/99JNKTExU4eHhqq+vT1mtVvXss8/qrsnIyFB2u10ppVRJSYkym82qv7/fa/uRkZFq8+bNWhlQtbW1ujpFRUVq9uzZWvnFF19U8+fP18r79u1TJpNJnT9//i+NE1DBwcHqpptuUoACVGpqqtf6g641H0opderUKRUQEKDa29t15x9++GH1yiuv+Gx70qRJaufOnT7jHDymT5/us//h86XU1ffcZDKp4OBgFRgYqAB14403qqNHj46IIS0tTS1fvnzU90CI68Hwz6bKykplNBrVDz/8oDwej+7o6OhQSinV39+v3G73qEdnZ+eo/dpsNpWbmzvmOG/wz/cGIYT4Z5s3bx7vvfeeVg4ODgbg4MGDvPnmm7S0tHDx4kUGBgb4448/+P3337U6Q61du5acnBx27dqFzWYjIyOD6OhoABobGzl16hTl5eVafaUUV65cobW1lVmzZnmN7cKFC4SEhKCUoqenh4SEBGpqajCZTLjdbt3NhwBJSUls3boVgIyMDLZs2cIdd9zBo48+it1uZ+HChdxww8Q//jMzM5kzZw5nz57l9ttvp7y8HLvdzpQpU/7SOENDQ2lqamJgYACXy0VxcTHbtm3T1RnvfIB+ZWuovr6+UfeO9/b2jtgGMjTOQYM3Ng4abb4GrVu3juXLl/PLL79QWFjI/PnzsVqtI/oKCgqip6fHZ4xCXK8sFguXL1/m3LlzJCcne60TGBhIbGzshPvo6+vD7Xb7bN8bSayFEIKriXRMTIzuXFtbG3a7ndzcXF5//XWmTp1KfX09zzzzjNd/68PV/chLly5l7969fPbZZxQVFVFZWUlaWhpXrlxhxYoVuj3Og2bMmOEztsFEzmAwMH369BEJZEBAgK6slNLORUREcPLkSfbv388XX3zBCy+8QHFxMS6Xa8TWg7F64IEHiI6OprKykueff57a2lrdnsaJjtNgMGhzEBsbS2dnJ4sXL+bw4cPAxOZjMB6j0UhjYyNGo1H3WkhIiM/rbrnlFn799ddR4/TmWvM12HZMTAwxMTFUV1cTExNDYmIiNptNV+/8+fPaFzMhrjeXLl3S3ZvQ2trK8ePHmTp1KmazmczMTLKysigpKcFisdDV1cWBAweIi4vDbrePu7+CggIWLlzIjBkzOHfuHBs3buTixYtkZ2ePuQ1JrIUQwodjx44xMDBASUmJtipZVVV1zevMZjNms5k1a9awZMkSHA4HaWlpJCQk8P3334+alHkzWiI3a9Ys6uvrycrK0s59+eWXulXhoKAgUlNTSU1NJS8vj9jYWL777jsSEhJGtBcYGDimp1AsXbqU8vJywsPDMRgMPPbYY9prEx3ncGvWrKG0tJTa2lrS0tLGNB8mk2lE/GNZ2fLGYrHQ0tIy7rivlXgPN2XKFFatWkVBQQHffvut7otSc3Mz6enp445BiH+DY8eOMW/ePK28du1aALKzs3E6nTgcDjZu3Eh+fj7t7e1MmzaNOXPmTCipBvj5559ZsmQJXV1dhIWFkZiYyNdff01kZOSY25CbF4UQwofo6GgGBgZ49913+fHHH9m1a9eIrQlD9fb2snLlSg4dOkRbWxtHjx6loaFBS3JfeuklvvrqK/Ly8jh+/Dgej4ePP/6YVatWTTjGdevW4XQ62bZtGx6Ph9LSUmpqarSb9pxOJ++//z7Nzc3aGIKCgnz+oYiKiqKuro7Ozk6vq7WDMjMzaWpq4o033iA9PV23ZcJf47z55pvJycmhqKgIpdSY5iMqKopLly5RV1dHV1cXPT09upWtmpoaWltbaWhoYNOmTXz66ac++09JSaG+vn5cMU9UXl4eJ0+e1N0UeubMGdrb20esYgtxvZg7dy5KqRGH0+kEri4EbNiwgdbWVvr7++no6KCmpsbnDdrXUllZydmzZ+nv76e9vZ3q6mruuuuucbUhibUQQvhw7733UlpayqZNm7jnnnsoLy/nrbfe8lnfaDTS3d1NVlYWZrOZJ598kgULFrBhwwYA4uPjcblceDwekpOTsVgsvPrqq9x2220TjnHRokVs3bqV4uJi7r77bsrKynA4HMydOxeAyZMns337dpKSkoiPj6euro5PPvnE597ikpIS9u/fT0REBBaLxWe/M2fO5P777+fEiRO6Zzj7e5yrV6/G7XazZ8+eMc2H1WolNzeXxYsXExYWxttvvw1cfdJJVlYW+fn53HnnnaSmpvLNN98QERHhs++nnnqKlpYW3eP+/lfCwsJYtmwZr732mvYrjxUVFTzyyCPjWi0TQvx/Bfz3TkshhBBCDLN+/XouXLhAWVnZ39pvX18fM2fOpKKigqSkpL+1byHExMmKtRBCCOFDYWEhkZGRf/uvH7a1tVFYWChJtRD/MLJiLYQQQgghhB/IirUQQgghhBB+IIm1EEIIIYQQfiCJtRBCCCGEEH4gibUQQgghhBB+IIm1EEIIIYQQfiCJtRBCCCGEEH4gibUQQgghhBB+IIm1EEIIIYQQfiCJtRBCCCGEEH7wHy1qI1nCfy9tAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 800x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Data\n",
    "tools = [\"BCFTOOL\", \"DEEPVARIANT\", \"VARSCAN2\", \"HAPLOTYPECALLER\", \"MUTECT2\"]\n",
    "TP = [25641, 28790, 24625, 25050, 24962]\n",
    "TN = [1008250088, 1008254674, 1008246847, 1008248698, 1008248555]\n",
    "FP = [41190, 45776, 37949, 39800, 39657]\n",
    "FN = [37210814, 37207665, 37211830, 37211405, 37211493]\n",
    "\n",
    "# Initialize a figure for ROC curves\n",
    "plt.figure(figsize=(8, 6))\n",
    "\n",
    "# Iterate over tools\n",
    "for i in range(len(tools)):\n",
    "    # Calculate True Positive Rate (TPR) and False Positive Rate (FPR)\n",
    "    tpr = TP[i] / (TP[i] + FN[i])\n",
    "    fpr = FP[i] / (TN[i] + FP[i])\n",
    "\n",
    "    # Plot ROC curve\n",
    "    plt.plot(fpr, tpr, marker='o', label=tools[i])\n",
    "\n",
    "# Set axis labels and title\n",
    "plt.xlabel('False Positive Rate (FPR)')\n",
    "plt.ylabel('True Positive Rate (TPR)')\n",
    "plt.title('12652712 (VAF_0.1 & DP >= 18) Sample FPR vs TPR')\n",
    "\n",
    "# Add legend\n",
    "plt.legend()\n",
    "\n",
    "plt.savefig(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/new_output_files/12652712_VAF0.1_DP18_FPR_TPR.svg', dpi=450, bbox_inches='tight')\n",
    "\n",
    "# Show the ROC curves\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d1ccd12e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Specify the directory where your Excel files are located\n",
    "directory = r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/new_output_files/'  # Change to your directory path\n",
    "\n",
    "# Create an empty DataFrame to store the combined data\n",
    "combined_data = pd.DataFrame()\n",
    "\n",
    "# Iterate through the files in the directory\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.startswith(\"snp_0.1VAF\") and filename.endswith(\".xlsx\"):\n",
    "        file_path = os.path.join(directory, filename)\n",
    "        data = pd.read_excel(file_path)\n",
    "        combined_data = combined_data.append(data, ignore_index=True)\n",
    "\n",
    "# Save the combined data to a new Excel file\n",
    "combined_data.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/new_output_files/0.1_combined_data.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65208d95",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4417fce",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6da09c49",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5bab8f0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a4353ea",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aed5535b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c2a5626e",
   "metadata": {},
   "source": [
    "# All 5 samples ACMG data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f17eae23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:827209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.314C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894115</td>\n",
       "      <td>7952</td>\n",
       "      <td>7110</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:827212</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.311G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894327</td>\n",
       "      <td>7826</td>\n",
       "      <td>6999</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481215</th>\n",
       "      <td>X:154641461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154641461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481219</th>\n",
       "      <td>X:154773313</td>\n",
       "      <td>A/T</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>1736</td>\n",
       "      <td>NM_001363.5</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001363.5:c.1155+64A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>PM2,BP7</td>\n",
       "      <td>DKC1:Variant is missing from all sub populatio...</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154773313</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481223</th>\n",
       "      <td>X:154785208</td>\n",
       "      <td>-/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.678-51_678-50insG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.0492715</td>\n",
       "      <td>3775</td>\n",
       "      <td>186</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154785208</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481224</th>\n",
       "      <td>X:154785208</td>\n",
       "      <td>A/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.678-51T&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.132715</td>\n",
       "      <td>3775</td>\n",
       "      <td>501</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154785208</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481239</th>\n",
       "      <td>Y:11153779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11153779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194203 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Chr:Pos Ref/Alt  Gene Name     Gene Inheritance Entrez Gene ID  \\\n",
       "0           1:69511     A/G      OR4F5  Default (Recessive)          79501   \n",
       "1          1:817514     T/C     FAM87B  Default (Recessive)         400728   \n",
       "2          1:826893     G/A  LINC00115  Default (Recessive)          79854   \n",
       "3          1:827209     G/C  LINC00115  Default (Recessive)          79854   \n",
       "4          1:827212     C/G  LINC00115  Default (Recessive)          79854   \n",
       "...             ...     ...        ...                  ...            ...   \n",
       "481215  X:154641461     NaN        NaN                  NaN            NaN   \n",
       "481219  X:154773313     A/T       DKC1            Recessive           1736   \n",
       "481223  X:154785208     -/C       MPP1  Default (Recessive)           4354   \n",
       "481224  X:154785208     A/C       MPP1  Default (Recessive)           4354   \n",
       "481239   Y:11153779     NaN        NaN                  NaN            NaN   \n",
       "\n",
       "       Transcript Name Transcript Strand                        HGVS cDot  \\\n",
       "0       NM_001005484.2                 +          NM_001005484.2:c.484A>G   \n",
       "1          NR_103536.1                 +             NR_103536.1:n.144T>C   \n",
       "2          NR_024321.1                 -             NR_024321.1:n.630C>T   \n",
       "3          NR_024321.1                 -             NR_024321.1:n.314C>G   \n",
       "4          NR_024321.1                 -             NR_024321.1:n.311G>C   \n",
       "...                ...               ...                              ...   \n",
       "481215             NaN               NaN                              NaN   \n",
       "481219     NM_001363.5                 +         NM_001363.5:c.1155+64A>T   \n",
       "481223     NM_002436.4                 -  NM_002436.4:c.678-51_678-50insG   \n",
       "481224     NM_002436.4                 -          NM_002436.4:c.678-51T>G   \n",
       "481239             NaN               NaN                              NaN   \n",
       "\n",
       "                         HGVS pDot        Sequence Ontology  \\\n",
       "0       NP_001005484.2:p.Thr162Ala         missense_variant   \n",
       "1                              NaN  non_coding_exon_variant   \n",
       "2                              NaN  non_coding_exon_variant   \n",
       "3                              NaN  non_coding_exon_variant   \n",
       "4                              NaN  non_coding_exon_variant   \n",
       "...                            ...                      ...   \n",
       "481215                         NaN                      NaN   \n",
       "481219                         NaN           intron_variant   \n",
       "481223                         NaN           intron_variant   \n",
       "481224                         NaN           intron_variant   \n",
       "481239                         NaN                      NaN   \n",
       "\n",
       "                  Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                   Single Transcript Gene                      BA1,BP4   \n",
       "1                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "3                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "4                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                                    ...                          ...   \n",
       "481215                                 NaN                          NaN   \n",
       "481219  Other Transcripts Have Same Effect                      PM2,BP7   \n",
       "481223  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "481224  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "481239                                 NaN                          NaN   \n",
       "\n",
       "                 ACMG Classification Criteria Description   Classification  \\\n",
       "0       OR4F5:Allele frequency is above 0.01 recessive...           Benign   \n",
       "1       FAM87B:Allele frequency is above 0.01 recessiv...           Benign   \n",
       "2       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "3       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "4       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "...                                                   ...              ...   \n",
       "481215                                                NaN              NaN   \n",
       "481219  DKC1:Variant is missing from all sub populatio...  VUS/Conflicting   \n",
       "481223  MPP1:Allele frequency is above 0.01 recessive ...           Benign   \n",
       "481224  MPP1:Allele frequency is above 0.01 recessive ...           Benign   \n",
       "481239                                                NaN              NaN   \n",
       "\n",
       "       Auto Classification Previous Classification Count  \\\n",
       "0                   Benign                           NaN   \n",
       "1                   Benign                           NaN   \n",
       "2                   Benign                           NaN   \n",
       "3                   Benign                           NaN   \n",
       "4                   Benign                           NaN   \n",
       "...                    ...                           ...   \n",
       "481215                 NaN                           NaN   \n",
       "481219     VUS/Conflicting                           NaN   \n",
       "481223              Benign                           NaN   \n",
       "481224              Benign                           NaN   \n",
       "481239                 NaN                           NaN   \n",
       "\n",
       "       Previous Classification Last Classification Date  \\\n",
       "0                          NaN                      NaN   \n",
       "1                          NaN                      NaN   \n",
       "2                          NaN                      NaN   \n",
       "3                          NaN                      NaN   \n",
       "4                          NaN                      NaN   \n",
       "...                        ...                      ...   \n",
       "481215                     NaN                      NaN   \n",
       "481219                     NaN                      NaN   \n",
       "481223                     NaN                      NaN   \n",
       "481224                     NaN                      NaN   \n",
       "481239                     NaN                      NaN   \n",
       "\n",
       "       Max Sub Population Freq Group Name    Max MAF Max Allele Number  \\\n",
       "0             Annotated gnomAD East Asian   0.999461             16698   \n",
       "1                       Annotated 1kG All   0.753195              5008   \n",
       "2             Annotated gnomAD East Asian   0.896992             10106   \n",
       "3             Annotated gnomAD East Asian   0.894115              7952   \n",
       "4             Annotated gnomAD East Asian   0.894327              7826   \n",
       "...                                   ...        ...               ...   \n",
       "481215                                NaN        NaN               NaN   \n",
       "481219                                NaN        NaN               NaN   \n",
       "481223                  Annotated 1kG All  0.0492715              3775   \n",
       "481224                  Annotated 1kG All   0.132715              3775   \n",
       "481239                                NaN        NaN               NaN   \n",
       "\n",
       "       Max Allele Count CHROM        POS  REF  ALT  \n",
       "0                 16689  chr1      69511    A    G  \n",
       "1                  3772  chr1     817514    T    C  \n",
       "2                  9065  chr1     826893    G    A  \n",
       "3                  7110  chr1     827209    G    C  \n",
       "4                  6999  chr1     827212    C    G  \n",
       "...                 ...   ...        ...  ...  ...  \n",
       "481215              NaN  chrX  154641461  NaN  NaN  \n",
       "481219              NaN  chrX  154773313    A    T  \n",
       "481223              186  chrX  154785208    -    C  \n",
       "481224              501  chrX  154785208    A    C  \n",
       "481239              NaN  chrY   11153779  NaN  NaN  \n",
       "\n",
       "[194203 rows x 26 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_new_05_10_2023/All_5samples.xlsx')\n",
    "df['CHROM'] = 'chr' + df['Chr:Pos'].str.split(':').str[0]\n",
    "df['POS'] = df['Chr:Pos'].str.split(':').str[1].astype(int)\n",
    "df['REF'] = df['Ref/Alt'].str.split('/').str[0]\n",
    "df['ALT'] = df['Ref/Alt'].str.split('/').str[1]\n",
    "columns_to_check = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "# Use drop_duplicates to remove duplicated rows based on the specified columns\n",
    "df = df.drop_duplicates(subset=columns_to_check)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0d0dd8eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_new_05_10_2023/All_samples_dup_data.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7dec1b0b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dd9c413",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67bf6f85",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "42c521b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65490</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65532</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65552</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65556</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761809</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194235</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761810</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194236</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761811</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194239</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761812</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194242</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23761813</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57194243</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28790922 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM       POS REF ALT\n",
       "0         chr1     65490   G   A\n",
       "1         chr1     65529   C   T\n",
       "2         chr1     65532   A   T\n",
       "3         chr1     65552   G   A\n",
       "4         chr1     65556   C   T\n",
       "...        ...       ...  ..  ..\n",
       "23761809  chrY  57194235   T   C\n",
       "23761810  chrY  57194236   G   C\n",
       "23761811  chrY  57194239   A   C\n",
       "23761812  chrY  57194242   G   T\n",
       "23761813  chrY  57194243   C   G\n",
       "\n",
       "[28790922 rows x 4 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "truth_set_path = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/output.tsv'\n",
    "truth_set = pd.read_csv(truth_set_path, sep='\\t', header=0, low_memory=False, encoding='latin-1')\n",
    "truth_1 = truth_set.copy()\n",
    "truth_1['ALT'] = truth_1['ALT'].str.split(',')\n",
    "truth_1 = truth_1.explode('ALT')\n",
    "truth_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "dce4a2be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:827209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.314C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894115</td>\n",
       "      <td>7952</td>\n",
       "      <td>7110</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:827212</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.311G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894327</td>\n",
       "      <td>7826</td>\n",
       "      <td>6999</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481215</th>\n",
       "      <td>X:154641461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154641461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481219</th>\n",
       "      <td>X:154773313</td>\n",
       "      <td>A/T</td>\n",
       "      <td>DKC1</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>1736</td>\n",
       "      <td>NM_001363.5</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001363.5:c.1155+64A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>PM2,BP7</td>\n",
       "      <td>DKC1:Variant is missing from all sub populatio...</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154773313</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481223</th>\n",
       "      <td>X:154785208</td>\n",
       "      <td>-/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.678-51_678-50insG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.0492715</td>\n",
       "      <td>3775</td>\n",
       "      <td>186</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154785208</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481224</th>\n",
       "      <td>X:154785208</td>\n",
       "      <td>A/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.678-51T&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.132715</td>\n",
       "      <td>3775</td>\n",
       "      <td>501</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154785208</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481239</th>\n",
       "      <td>Y:11153779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11153779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194203 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Chr:Pos Ref/Alt  Gene Name     Gene Inheritance Entrez Gene ID  \\\n",
       "0           1:69511     A/G      OR4F5  Default (Recessive)          79501   \n",
       "1          1:817514     T/C     FAM87B  Default (Recessive)         400728   \n",
       "2          1:826893     G/A  LINC00115  Default (Recessive)          79854   \n",
       "3          1:827209     G/C  LINC00115  Default (Recessive)          79854   \n",
       "4          1:827212     C/G  LINC00115  Default (Recessive)          79854   \n",
       "...             ...     ...        ...                  ...            ...   \n",
       "481215  X:154641461     NaN        NaN                  NaN            NaN   \n",
       "481219  X:154773313     A/T       DKC1            Recessive           1736   \n",
       "481223  X:154785208     -/C       MPP1  Default (Recessive)           4354   \n",
       "481224  X:154785208     A/C       MPP1  Default (Recessive)           4354   \n",
       "481239   Y:11153779     NaN        NaN                  NaN            NaN   \n",
       "\n",
       "       Transcript Name Transcript Strand                        HGVS cDot  \\\n",
       "0       NM_001005484.2                 +          NM_001005484.2:c.484A>G   \n",
       "1          NR_103536.1                 +             NR_103536.1:n.144T>C   \n",
       "2          NR_024321.1                 -             NR_024321.1:n.630C>T   \n",
       "3          NR_024321.1                 -             NR_024321.1:n.314C>G   \n",
       "4          NR_024321.1                 -             NR_024321.1:n.311G>C   \n",
       "...                ...               ...                              ...   \n",
       "481215             NaN               NaN                              NaN   \n",
       "481219     NM_001363.5                 +         NM_001363.5:c.1155+64A>T   \n",
       "481223     NM_002436.4                 -  NM_002436.4:c.678-51_678-50insG   \n",
       "481224     NM_002436.4                 -          NM_002436.4:c.678-51T>G   \n",
       "481239             NaN               NaN                              NaN   \n",
       "\n",
       "                         HGVS pDot        Sequence Ontology  \\\n",
       "0       NP_001005484.2:p.Thr162Ala         missense_variant   \n",
       "1                              NaN  non_coding_exon_variant   \n",
       "2                              NaN  non_coding_exon_variant   \n",
       "3                              NaN  non_coding_exon_variant   \n",
       "4                              NaN  non_coding_exon_variant   \n",
       "...                            ...                      ...   \n",
       "481215                         NaN                      NaN   \n",
       "481219                         NaN           intron_variant   \n",
       "481223                         NaN           intron_variant   \n",
       "481224                         NaN           intron_variant   \n",
       "481239                         NaN                      NaN   \n",
       "\n",
       "                  Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                   Single Transcript Gene                      BA1,BP4   \n",
       "1                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "3                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "4                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                                    ...                          ...   \n",
       "481215                                 NaN                          NaN   \n",
       "481219  Other Transcripts Have Same Effect                      PM2,BP7   \n",
       "481223  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "481224  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "481239                                 NaN                          NaN   \n",
       "\n",
       "                 ACMG Classification Criteria Description   Classification  \\\n",
       "0       OR4F5:Allele frequency is above 0.01 recessive...           Benign   \n",
       "1       FAM87B:Allele frequency is above 0.01 recessiv...           Benign   \n",
       "2       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "3       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "4       LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "...                                                   ...              ...   \n",
       "481215                                                NaN              NaN   \n",
       "481219  DKC1:Variant is missing from all sub populatio...  VUS/Conflicting   \n",
       "481223  MPP1:Allele frequency is above 0.01 recessive ...           Benign   \n",
       "481224  MPP1:Allele frequency is above 0.01 recessive ...           Benign   \n",
       "481239                                                NaN              NaN   \n",
       "\n",
       "       Auto Classification Previous Classification Count  \\\n",
       "0                   Benign                           NaN   \n",
       "1                   Benign                           NaN   \n",
       "2                   Benign                           NaN   \n",
       "3                   Benign                           NaN   \n",
       "4                   Benign                           NaN   \n",
       "...                    ...                           ...   \n",
       "481215                 NaN                           NaN   \n",
       "481219     VUS/Conflicting                           NaN   \n",
       "481223              Benign                           NaN   \n",
       "481224              Benign                           NaN   \n",
       "481239                 NaN                           NaN   \n",
       "\n",
       "       Previous Classification Last Classification Date  \\\n",
       "0                          NaN                      NaN   \n",
       "1                          NaN                      NaN   \n",
       "2                          NaN                      NaN   \n",
       "3                          NaN                      NaN   \n",
       "4                          NaN                      NaN   \n",
       "...                        ...                      ...   \n",
       "481215                     NaN                      NaN   \n",
       "481219                     NaN                      NaN   \n",
       "481223                     NaN                      NaN   \n",
       "481224                     NaN                      NaN   \n",
       "481239                     NaN                      NaN   \n",
       "\n",
       "       Max Sub Population Freq Group Name    Max MAF Max Allele Number  \\\n",
       "0             Annotated gnomAD East Asian   0.999461             16698   \n",
       "1                       Annotated 1kG All   0.753195              5008   \n",
       "2             Annotated gnomAD East Asian   0.896992             10106   \n",
       "3             Annotated gnomAD East Asian   0.894115              7952   \n",
       "4             Annotated gnomAD East Asian   0.894327              7826   \n",
       "...                                   ...        ...               ...   \n",
       "481215                                NaN        NaN               NaN   \n",
       "481219                                NaN        NaN               NaN   \n",
       "481223                  Annotated 1kG All  0.0492715              3775   \n",
       "481224                  Annotated 1kG All   0.132715              3775   \n",
       "481239                                NaN        NaN               NaN   \n",
       "\n",
       "       Max Allele Count CHROM        POS  REF  ALT  \n",
       "0                 16689  chr1      69511    A    G  \n",
       "1                  3772  chr1     817514    T    C  \n",
       "2                  9065  chr1     826893    G    A  \n",
       "3                  7110  chr1     827209    G    C  \n",
       "4                  6999  chr1     827212    C    G  \n",
       "...                 ...   ...        ...  ...  ...  \n",
       "481215              NaN  chrX  154641461  NaN  NaN  \n",
       "481219              NaN  chrX  154773313    A    T  \n",
       "481223              186  chrX  154785208    -    C  \n",
       "481224              501  chrX  154785208    A    C  \n",
       "481239              NaN  chrY   11153779  NaN  NaN  \n",
       "\n",
       "[194203 rows x 26 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_new_05_10_2023/All_5samples.xlsx')\n",
    "df['CHROM'] = 'chr' + df['Chr:Pos'].str.split(':').str[0]\n",
    "df['POS'] = df['Chr:Pos'].str.split(':').str[1].astype(int)\n",
    "df['REF'] = df['Ref/Alt'].str.split('/').str[0]\n",
    "df['ALT'] = df['Ref/Alt'].str.split('/').str[1]\n",
    "columns_to_check = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "# Use drop_duplicates to remove duplicated rows based on the specified columns\n",
    "df = df.drop_duplicates(subset=columns_to_check)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a78fbf82",
   "metadata": {},
   "source": [
    "# BCFTOOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "7d27bd75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=71;ADF=0,56;ADR=0,1;AD=0,57;DPR=0,57;SCR=0;...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>1/1:255,172,0:57:57:0:0,0,56,1:0,56:0,1:0,57:0...</td>\n",
       "      <td>71</td>\n",
       "      <td>0,56</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMD11|ENSG000001876...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=26;ADF=0,19;ADR=0,6;AD=0,25;DPR=0,25;SCR=0;...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>1/1:255,75,0:25:25:0:0,0,19,6:0,19:0,6:0,25:0,...</td>\n",
       "      <td>26</td>\n",
       "      <td>0,19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945086</td>\n",
       "      <td>rs138652036</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>222.2460</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=71;ADF=19,14;ADR=16,11;AD=35,25;DPR=35,25;S...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>0/1:255,0,255:60:25:0:19,16,14,11:19,14:16,11:...</td>\n",
       "      <td>71</td>\n",
       "      <td>19,14</td>\n",
       "      <td>19</td>\n",
       "      <td>14</td>\n",
       "      <td>0.424242</td>\n",
       "      <td>C|missense_variant|MODERATE|NOC2L|ENSG00000188...</td>\n",
       "      <td>0.009679</td>\n",
       "      <td>0.001378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>948711</td>\n",
       "      <td>rs7522415</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>118.2420</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=22;ADF=0,1;ADR=12,8;AD=12,9;DPR=12,9;SCR=13...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>0/1:151,0,163:21:9:0:0,12,1,8:0,1:12,8:12,9:12...</td>\n",
       "      <td>22</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>965125</td>\n",
       "      <td>rs28705211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>222.4060</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=234;ADF=42,49;ADR=60,57;AD=102,106;DPR=102,...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>0/1:255,0,255:208:106:3:42,60,49,57:42,49:60,5...</td>\n",
       "      <td>234</td>\n",
       "      <td>42,49</td>\n",
       "      <td>42</td>\n",
       "      <td>49</td>\n",
       "      <td>0.538462</td>\n",
       "      <td>C|synonymous_variant|LOW|KLHL17|ENSG0000018796...</td>\n",
       "      <td>0.301600</td>\n",
       "      <td>0.241100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45617</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1304050</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>41.4148</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=100;ADF=0,14;ADR=0,79;AD=0,93;DPR=0,93;SCR=...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>1/1:71,255,0:93:93:0:0,0,14,79:0,14:0,79:0,93:...</td>\n",
       "      <td>100</td>\n",
       "      <td>0,14</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|intron_variant|MODIFIER|CSF2RA|ENSG000001982...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45618</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>27.4222</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=58;ADF=0,8;ADR=0,40;AD=0,48;DPR=0,48;SCR=1;...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>1/1:57,144,0:48:48:0:0,0,8,40:0,8:0,40:0,48:0,...</td>\n",
       "      <td>58</td>\n",
       "      <td>0,8</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>G|downstream_gene_variant|MODIFIER|IL3RA|ENSG0...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45619</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1439179</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>37.4152</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=84;ADF=0,13;ADR=0,65;AD=0,78;DPR=0,78;SCR=1...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>1/1:67,235,0:78:78:0:0,0,13,65:0,13:0,65:0,78:...</td>\n",
       "      <td>84</td>\n",
       "      <td>0,13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>A|intron_variant|MODIFIER|ASMTL|ENSG0000016909...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45621</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>rs9650861</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>102.4150</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=41;ADF=0,38;ADR=0,0;AD=0,38;DPR=0,38;SCR=1;...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>1/1:132,114,0:38:38:0:0,0,38,0:0,38:0,0:0,38:0...</td>\n",
       "      <td>41</td>\n",
       "      <td>0,38</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45623</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986732</td>\n",
       "      <td>rs867343413</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>123.3690</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=19;ADF=1,5;ADR=0,12;AD=1,17;DPR=1,17;SCR=0;...</td>\n",
       "      <td>GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR</td>\n",
       "      <td>1/1:150,42,0:18:17:0:1,0,5,12:1,5:0,12:1,17:1,...</td>\n",
       "      <td>19</td>\n",
       "      <td>1,5</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25822 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT      QUAL FILTER  \\\n",
       "2      chr1    930939    rs9988021   G   A  225.4170      .   \n",
       "4      chr1    944858    rs3748592   A   G  225.4170      .   \n",
       "5      chr1    945086  rs138652036   T   C  222.2460      .   \n",
       "8      chr1    948711    rs7522415   C   G  118.2420      .   \n",
       "18     chr1    965125   rs28705211   G   C  222.4060      .   \n",
       "...     ...       ...          ...  ..  ..       ...    ...   \n",
       "45617  chrY   1304050            .   T   C   41.4148      .   \n",
       "45618  chrY   1387425            .   C   G   27.4222      .   \n",
       "45619  chrY   1439179            .   G   A   37.4152      .   \n",
       "45621  chrY  11986362    rs9650861   A   C  102.4150      .   \n",
       "45623  chrY  11986732  rs867343413   C   T  123.3690      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "2      DP=71;ADF=0,56;ADR=0,1;AD=0,57;DPR=0,57;SCR=0;...   \n",
       "4      DP=26;ADF=0,19;ADR=0,6;AD=0,25;DPR=0,25;SCR=0;...   \n",
       "5      DP=71;ADF=19,14;ADR=16,11;AD=35,25;DPR=35,25;S...   \n",
       "8      DP=22;ADF=0,1;ADR=12,8;AD=12,9;DPR=12,9;SCR=13...   \n",
       "18     DP=234;ADF=42,49;ADR=60,57;AD=102,106;DPR=102,...   \n",
       "...                                                  ...   \n",
       "45617  DP=100;ADF=0,14;ADR=0,79;AD=0,93;DPR=0,93;SCR=...   \n",
       "45618  DP=58;ADF=0,8;ADR=0,40;AD=0,48;DPR=0,48;SCR=1;...   \n",
       "45619  DP=84;ADF=0,13;ADR=0,65;AD=0,78;DPR=0,78;SCR=1...   \n",
       "45621  DP=41;ADF=0,38;ADR=0,0;AD=0,38;DPR=0,38;SCR=1;...   \n",
       "45623  DP=19;ADF=1,5;ADR=0,12;AD=1,17;DPR=1,17;SCR=0;...   \n",
       "\n",
       "                                      FORMAT  \\\n",
       "2      GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "4      GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "5      GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "8      GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "18     GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "...                                      ...   \n",
       "45617  GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "45618  GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "45619  GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "45621  GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "45623  GT:PL:DP:DV:SP:DP4:ADF:ADR:AD:DPR:SCR   \n",
       "\n",
       "                                                  SAMPLE   DP     AD  RD  A_D  \\\n",
       "2      1/1:255,172,0:57:57:0:0,0,56,1:0,56:0,1:0,57:0...   71   0,56   0   56   \n",
       "4      1/1:255,75,0:25:25:0:0,0,19,6:0,19:0,6:0,25:0,...   26   0,19   0   19   \n",
       "5      0/1:255,0,255:60:25:0:19,16,14,11:19,14:16,11:...   71  19,14  19   14   \n",
       "8      0/1:151,0,163:21:9:0:0,12,1,8:0,1:12,8:12,9:12...   22    0,1   0    1   \n",
       "18     0/1:255,0,255:208:106:3:42,60,49,57:42,49:60,5...  234  42,49  42   49   \n",
       "...                                                  ...  ...    ...  ..  ...   \n",
       "45617  1/1:71,255,0:93:93:0:0,0,14,79:0,14:0,79:0,93:...  100   0,14   0   14   \n",
       "45618  1/1:57,144,0:48:48:0:0,0,8,40:0,8:0,40:0,48:0,...   58    0,8   0    8   \n",
       "45619  1/1:67,235,0:78:78:0:0,0,13,65:0,13:0,65:0,78:...   84   0,13   0   13   \n",
       "45621  1/1:132,114,0:38:38:0:0,0,38,0:0,38:0,0:0,38:0...   41   0,38   0   38   \n",
       "45623  1/1:150,42,0:18:17:0:1,0,5,12:1,5:0,12:1,17:1,...   19    1,5   1    5   \n",
       "\n",
       "            VAF                                                CSQ  \\\n",
       "2      1.000000  A|intron_variant|MODIFIER|SAMD11|ENSG000001876...   \n",
       "4      1.000000  G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...   \n",
       "5      0.424242  C|missense_variant|MODERATE|NOC2L|ENSG00000188...   \n",
       "8      1.000000  G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...   \n",
       "18     0.538462  C|synonymous_variant|LOW|KLHL17|ENSG0000018796...   \n",
       "...         ...                                                ...   \n",
       "45617  1.000000  C|intron_variant|MODIFIER|CSF2RA|ENSG000001982...   \n",
       "45618  1.000000  G|downstream_gene_variant|MODIFIER|IL3RA|ENSG0...   \n",
       "45619  1.000000  A|intron_variant|MODIFIER|ASMTL|ENSG0000016909...   \n",
       "45621  1.000000  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "45623  0.833333  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "       gnomADe_SAS_AF  gnomADe_AF  \n",
       "2            0.000000    0.000000  \n",
       "4            0.000000    0.000000  \n",
       "5            0.009679    0.001378  \n",
       "8            0.000000    0.000000  \n",
       "18           0.301600    0.241100  \n",
       "...               ...         ...  \n",
       "45617        0.000000    0.000000  \n",
       "45618        0.000000    0.000000  \n",
       "45619        0.000000    0.000000  \n",
       "45621        0.000000    0.000000  \n",
       "45623        0.000000    0.000000  \n",
       "\n",
       "[25822 rows x 18 columns]"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/VCC_NEW/12652709_BCFTOOL.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['INFO'].str.extract(r'DP=(\\d+)')[0].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[6]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['CSQ'].str.split('|').str[56]\n",
    "vcf['gnomADe_AF'] = vcf['CSQ'].str.split('|').str[48]\n",
    "vcf['gnomADe_AF'] = vcf['gnomADe_AF'].replace('', 0).astype('float')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].replace('', 0)\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].astype('float')\n",
    "# For BCFTOOL, you can choose not to filter based on 'DP'\n",
    "vcf = vcf[~vcf['ALT'].str.contains(',')]\n",
    "filtered_data = vcf[vcf['VAF'] >= 0.1]\n",
    "filtered_data = filtered_data[filtered_data['DP'] >= 15]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_AF'] <= 0.5]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_SAS_AF'] <= 0.5]\n",
    "filtered_data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09a0951a",
   "metadata": {},
   "source": [
    "# VARSCAN2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "901cf1ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786070</td>\n",
       "      <td>rs2977675</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=249;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=LOC...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:249:249:167:82:32.93%:6.7642E-29:43:38:...</td>\n",
       "      <td>249</td>\n",
       "      <td>167</td>\n",
       "      <td>82</td>\n",
       "      <td>0.329317</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.01013</td>\n",
       "      <td>0.01256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826729</td>\n",
       "      <td>rs540936498</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=210;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=LIN...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:211:210:149:61:29.05%:2.6055E-21:44:38:...</td>\n",
       "      <td>210</td>\n",
       "      <td>149</td>\n",
       "      <td>61</td>\n",
       "      <td>0.290476</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.03310</td>\n",
       "      <td>0.02253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827092</td>\n",
       "      <td>rs145493205</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=83;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=LINC...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:83:83:49:34:40.96%:8.056E-13:39:32:27:2...</td>\n",
       "      <td>83</td>\n",
       "      <td>49</td>\n",
       "      <td>34</td>\n",
       "      <td>0.409639</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11850</td>\n",
       "      <td>0.11010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827105</td>\n",
       "      <td>rs12095200</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=86;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8339,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:87:86:48:38:44.19%:1.8221E-14:38:36:29:...</td>\n",
       "      <td>86</td>\n",
       "      <td>48</td>\n",
       "      <td>38</td>\n",
       "      <td>0.441860</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11770</td>\n",
       "      <td>0.11670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930314</td>\n",
       "      <td>rs9988179</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9475,.,...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:77:76:40:36:47.37%:5.9795E-14:42:35:15:...</td>\n",
       "      <td>76</td>\n",
       "      <td>40</td>\n",
       "      <td>36</td>\n",
       "      <td>0.473684</td>\n",
       "      <td>T|missense_variant|MODERATE|SAMD11|ENSG0000018...</td>\n",
       "      <td>0.02636</td>\n",
       "      <td>0.02893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42548</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2738242</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=205;WT=0;HET=1;HOM=0;NC=0;CSQ=T|3_prime_UT...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:212:205:102:103:50.24%:2.6979E-39:42:39...</td>\n",
       "      <td>205</td>\n",
       "      <td>102</td>\n",
       "      <td>103</td>\n",
       "      <td>0.502439</td>\n",
       "      <td>T|3_prime_UTR_variant&amp;NMD_transcript_variant|M...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42549</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2774669</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=142;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_var...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:99:142:142:73:69:48.59%:2.6187E-26:45:45:5...</td>\n",
       "      <td>142</td>\n",
       "      <td>73</td>\n",
       "      <td>69</td>\n",
       "      <td>0.485915</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42550</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>rs879016544</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;RS=879016544;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:99:30:29:0:29:100%:3.3259E-17:0:35:0:0:0:29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42551</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986732</td>\n",
       "      <td>rs867343413</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;RS=867343413;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:93:17:17:0:17:100%:4.2852E-10:0:34:0:0:3:14</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56961295</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:74:52:52:31:21:40.38%:3.7578E-8:39:45:19:1...</td>\n",
       "      <td>52</td>\n",
       "      <td>31</td>\n",
       "      <td>21</td>\n",
       "      <td>0.403846</td>\n",
       "      <td>T|intergenic_variant|MODIFIER|||||||||||||||||...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25836 rows Ã— 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "1      chr1    786070    rs2977675   G   A    .   PASS   \n",
       "2      chr1    826729  rs540936498   C   T    .   PASS   \n",
       "4      chr1    827092  rs145493205   C   T    .   PASS   \n",
       "5      chr1    827105   rs12095200   C   A    .   PASS   \n",
       "6      chr1    930314    rs9988179   C   T    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "42548  chrY   2738242            .   A   T    .   PASS   \n",
       "42549  chrY   2774669            .   A   G    .   PASS   \n",
       "42550  chrY  11986608  rs879016544   T   C    .   PASS   \n",
       "42551  chrY  11986732  rs867343413   C   T    .   PASS   \n",
       "42553  chrY  56961295            .   G   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "1      ADP=249;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=LOC...   \n",
       "2      ADP=210;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=LIN...   \n",
       "4      ADP=83;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=LINC...   \n",
       "5      ADP=86;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8339,0....   \n",
       "6      ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.9475,.,...   \n",
       "...                                                  ...   \n",
       "42548  ADP=205;WT=0;HET=1;HOM=0;NC=0;CSQ=T|3_prime_UT...   \n",
       "42549  ADP=142;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_var...   \n",
       "42550  ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;RS=879016544;...   \n",
       "42551  ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;RS=867343413;...   \n",
       "42553  ADP=52;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intergenic_...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "5      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "6      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "42548  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "42549  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "42550  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "42551  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "42553  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   DP   RD   AD  \\\n",
       "1      0/1:99:249:249:167:82:32.93%:6.7642E-29:43:38:...  249  167   82   \n",
       "2      0/1:99:211:210:149:61:29.05%:2.6055E-21:44:38:...  210  149   61   \n",
       "4      0/1:99:83:83:49:34:40.96%:8.056E-13:39:32:27:2...   83   49   34   \n",
       "5      0/1:99:87:86:48:38:44.19%:1.8221E-14:38:36:29:...   86   48   38   \n",
       "6      0/1:99:77:76:40:36:47.37%:5.9795E-14:42:35:15:...   76   40   36   \n",
       "...                                                  ...  ...  ...  ...   \n",
       "42548  0/1:99:212:205:102:103:50.24%:2.6979E-39:42:39...  205  102  103   \n",
       "42549  0/1:99:142:142:73:69:48.59%:2.6187E-26:45:45:5...  142   73   69   \n",
       "42550    1/1:99:30:29:0:29:100%:3.3259E-17:0:35:0:0:0:29   29    0   29   \n",
       "42551    1/1:93:17:17:0:17:100%:4.2852E-10:0:34:0:0:3:14   17    0   17   \n",
       "42553  0/1:74:52:52:31:21:40.38%:3.7578E-8:39:45:19:1...   52   31   21   \n",
       "\n",
       "            VAF                                                CSQ  \\\n",
       "1      0.329317  A|intron_variant&non_coding_transcript_variant...   \n",
       "2      0.290476  T|upstream_gene_variant|MODIFIER|LINC01128|ENS...   \n",
       "4      0.409639  T|upstream_gene_variant|MODIFIER|LINC01128|ENS...   \n",
       "5      0.441860  A|upstream_gene_variant|MODIFIER|LINC01128|ENS...   \n",
       "6      0.473684  T|missense_variant|MODERATE|SAMD11|ENSG0000018...   \n",
       "...         ...                                                ...   \n",
       "42548  0.502439  T|3_prime_UTR_variant&NMD_transcript_variant|M...   \n",
       "42549  0.485915  G|intron_variant&non_coding_transcript_variant...   \n",
       "42550  1.000000  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "42551  1.000000  T|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "42553  0.403846  T|intergenic_variant|MODIFIER|||||||||||||||||...   \n",
       "\n",
       "       gnomADe_SAS_AF  gnomADe_AF  \n",
       "1             0.01013     0.01256  \n",
       "2             0.03310     0.02253  \n",
       "4             0.11850     0.11010  \n",
       "5             0.11770     0.11670  \n",
       "6             0.02636     0.02893  \n",
       "...               ...         ...  \n",
       "42548         0.00000     0.00000  \n",
       "42549         0.00000     0.00000  \n",
       "42550         0.00000     0.00000  \n",
       "42551         0.00000     0.00000  \n",
       "42553         0.00000     0.00000  \n",
       "\n",
       "[25836 rows x 17 columns]"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/VCC_NEW/12652712_VARSCAN2.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "vcf['RD'] = vcf['SAMPLE'].str.split(':').str[4].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[5].fillna('0').astype(int)\n",
    "vcf['VAF'] = vcf['AD'] / (vcf['RD'] + vcf['AD'])\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['CSQ'].str.split('|').str[56]\n",
    "vcf['gnomADe_AF'] = vcf['CSQ'].str.split('|').str[48]\n",
    "vcf['gnomADe_AF'] = vcf['gnomADe_AF'].replace('', 0).astype('float')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].replace('', 0)\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].astype('float')\n",
    "# Apply VAF and DP filters\n",
    "vcf = vcf[~vcf['ALT'].str.contains(',')]\n",
    "filtered_data = vcf[vcf['VAF'] >= 0.1]\n",
    "filtered_data = filtered_data[filtered_data['DP'] >= 15]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_AF'] <= 0.5]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_SAS_AF'] <= 0.5]\n",
    "filtered_data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6ab7829",
   "metadata": {},
   "source": [
    "# MUTECT2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "43284c46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786070</td>\n",
       "      <td>rs2977675</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AS_FilterStatus=SITE;AS_SB_TABLE=75,77|11,29;D...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:152,40:0.233:192:65,13:84,27:75,77,11,29</td>\n",
       "      <td>192</td>\n",
       "      <td>152,40</td>\n",
       "      <td>152</td>\n",
       "      <td>40</td>\n",
       "      <td>0.208333</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.01013</td>\n",
       "      <td>0.01256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786377</td>\n",
       "      <td>rs189147642</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AS_FilterStatus=SITE;AS_SB_TABLE=34,62|7,19;DP...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:96,26:0.219:122:50,11:46,14:34,62,7,19</td>\n",
       "      <td>122</td>\n",
       "      <td>96,26</td>\n",
       "      <td>96</td>\n",
       "      <td>26</td>\n",
       "      <td>0.213115</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.04863</td>\n",
       "      <td>0.03993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826729</td>\n",
       "      <td>rs540936498</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AS_FilterStatus=SITE;AS_SB_TABLE=63,55|17,13;D...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:118,30:0.22:148:49,17:69,13:63,55,17,13</td>\n",
       "      <td>148</td>\n",
       "      <td>118,30</td>\n",
       "      <td>118</td>\n",
       "      <td>30</td>\n",
       "      <td>0.202703</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.03310</td>\n",
       "      <td>0.02253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827092</td>\n",
       "      <td>rs145493205</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>map_qual</td>\n",
       "      <td>AS_FilterStatus=map_qual;AS_SB_TABLE=12,17|2,6...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:29,8:0.256:37:14,6:15,2:12,17,2,6</td>\n",
       "      <td>37</td>\n",
       "      <td>29,8</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>0.216216</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11850</td>\n",
       "      <td>0.11010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827105</td>\n",
       "      <td>rs12095200</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>map_qual</td>\n",
       "      <td>AS_FilterStatus=map_qual;AS_SB_TABLE=13,14|9,6...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:27,15:0.39:42:13,8:14,7:13,14,9,6</td>\n",
       "      <td>42</td>\n",
       "      <td>27,15</td>\n",
       "      <td>27</td>\n",
       "      <td>15</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11770</td>\n",
       "      <td>0.11670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48824</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154929926</td>\n",
       "      <td>rs1800292</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AS_FilterStatus=SITE;AS_SB_TABLE=27,29|40,29;D...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:56,69:0.553:125:30,23:26,46:27,29,40,29</td>\n",
       "      <td>125</td>\n",
       "      <td>56,69</td>\n",
       "      <td>56</td>\n",
       "      <td>69</td>\n",
       "      <td>0.552000</td>\n",
       "      <td>G|synonymous_variant|LOW|F8|ENSG00000185010|Tr...</td>\n",
       "      <td>0.33770</td>\n",
       "      <td>0.14700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48825</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AS_FilterStatus=SITE;AS_SB_TABLE=19,10|28,18;D...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:29,46:0.606:75:8,19:20,27:19,10,28,18</td>\n",
       "      <td>75</td>\n",
       "      <td>29,46</td>\n",
       "      <td>29</td>\n",
       "      <td>46</td>\n",
       "      <td>0.613333</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48826</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155349551</td>\n",
       "      <td>rs563910</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AS_FilterStatus=SITE;AS_SB_TABLE=11,0|9,2;DP=2...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:11,11:0.5:22:8,4:3,7:11,0,9,2</td>\n",
       "      <td>22</td>\n",
       "      <td>11,11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48831</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>rs9650861</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>map_qual</td>\n",
       "      <td>AS_FilterStatus=map_qual;AS_SB_TABLE=0,0|27,0;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,27:0.966:27:0,14:0,13:0,0,27,0</td>\n",
       "      <td>27</td>\n",
       "      <td>0,27</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48832</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>rs879016544</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>map_qual</td>\n",
       "      <td>AS_FilterStatus=map_qual;AS_SB_TABLE=0,0|0,21;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,21:0.957:21:0,14:0,6:0,0,0,21</td>\n",
       "      <td>21</td>\n",
       "      <td>0,21</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27042 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL    FILTER  \\\n",
       "1      chr1     786070    rs2977675   G   A    .      PASS   \n",
       "2      chr1     786377  rs189147642   T   A    .      PASS   \n",
       "3      chr1     826729  rs540936498   C   T    .      PASS   \n",
       "5      chr1     827092  rs145493205   C   T    .  map_qual   \n",
       "6      chr1     827105   rs12095200   C   A    .  map_qual   \n",
       "...     ...        ...          ...  ..  ..  ...       ...   \n",
       "48824  chrX  154929926    rs1800292   T   G    .      PASS   \n",
       "48825  chrX  155277884     rs559165   G   T    .      PASS   \n",
       "48826  chrX  155349551     rs563910   C   G    .      PASS   \n",
       "48831  chrY   11986362    rs9650861   A   C    .  map_qual   \n",
       "48832  chrY   11986608  rs879016544   T   C    .  map_qual   \n",
       "\n",
       "                                                    INFO  \\\n",
       "1      AS_FilterStatus=SITE;AS_SB_TABLE=75,77|11,29;D...   \n",
       "2      AS_FilterStatus=SITE;AS_SB_TABLE=34,62|7,19;DP...   \n",
       "3      AS_FilterStatus=SITE;AS_SB_TABLE=63,55|17,13;D...   \n",
       "5      AS_FilterStatus=map_qual;AS_SB_TABLE=12,17|2,6...   \n",
       "6      AS_FilterStatus=map_qual;AS_SB_TABLE=13,14|9,6...   \n",
       "...                                                  ...   \n",
       "48824  AS_FilterStatus=SITE;AS_SB_TABLE=27,29|40,29;D...   \n",
       "48825  AS_FilterStatus=SITE;AS_SB_TABLE=19,10|28,18;D...   \n",
       "48826  AS_FilterStatus=SITE;AS_SB_TABLE=11,0|9,2;DP=2...   \n",
       "48831  AS_FilterStatus=map_qual;AS_SB_TABLE=0,0|27,0;...   \n",
       "48832  AS_FilterStatus=map_qual;AS_SB_TABLE=0,0|0,21;...   \n",
       "\n",
       "                         FORMAT                                        SAMPLE  \\\n",
       "1      GT:AD:AF:DP:F1R2:F2R1:SB  0/1:152,40:0.233:192:65,13:84,27:75,77,11,29   \n",
       "2      GT:AD:AF:DP:F1R2:F2R1:SB    0/1:96,26:0.219:122:50,11:46,14:34,62,7,19   \n",
       "3      GT:AD:AF:DP:F1R2:F2R1:SB   0/1:118,30:0.22:148:49,17:69,13:63,55,17,13   \n",
       "5      GT:AD:AF:DP:F1R2:F2R1:SB         0/1:29,8:0.256:37:14,6:15,2:12,17,2,6   \n",
       "6      GT:AD:AF:DP:F1R2:F2R1:SB         0/1:27,15:0.39:42:13,8:14,7:13,14,9,6   \n",
       "...                         ...                                           ...   \n",
       "48824  GT:AD:AF:DP:F1R2:F2R1:SB   0/1:56,69:0.553:125:30,23:26,46:27,29,40,29   \n",
       "48825  GT:AD:AF:DP:F1R2:F2R1:SB     0/1:29,46:0.606:75:8,19:20,27:19,10,28,18   \n",
       "48826  GT:AD:AF:DP:F1R2:F2R1:SB             0/1:11,11:0.5:22:8,4:3,7:11,0,9,2   \n",
       "48831  GT:AD:AF:DP:F1R2:F2R1:SB          0/1:0,27:0.966:27:0,14:0,13:0,0,27,0   \n",
       "48832  GT:AD:AF:DP:F1R2:F2R1:SB           0/1:0,21:0.957:21:0,14:0,6:0,0,0,21   \n",
       "\n",
       "        DP      AD   RD  A_D       VAF  \\\n",
       "1      192  152,40  152   40  0.208333   \n",
       "2      122   96,26   96   26  0.213115   \n",
       "3      148  118,30  118   30  0.202703   \n",
       "5       37    29,8   29    8  0.216216   \n",
       "6       42   27,15   27   15  0.357143   \n",
       "...    ...     ...  ...  ...       ...   \n",
       "48824  125   56,69   56   69  0.552000   \n",
       "48825   75   29,46   29   46  0.613333   \n",
       "48826   22   11,11   11   11  0.500000   \n",
       "48831   27    0,27    0   27  1.000000   \n",
       "48832   21    0,21    0   21  1.000000   \n",
       "\n",
       "                                                     CSQ  gnomADe_SAS_AF  \\\n",
       "1      A|intron_variant&non_coding_transcript_variant...         0.01013   \n",
       "2      A|intron_variant&non_coding_transcript_variant...         0.04863   \n",
       "3      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.03310   \n",
       "5      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11850   \n",
       "6      A|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11770   \n",
       "...                                                  ...             ...   \n",
       "48824  G|synonymous_variant|LOW|F8|ENSG00000185010|Tr...         0.33770   \n",
       "48825  T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...         0.00000   \n",
       "48826  G|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "48831  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "48832  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "\n",
       "       gnomADe_AF  \n",
       "1         0.01256  \n",
       "2         0.03993  \n",
       "3         0.02253  \n",
       "5         0.11010  \n",
       "6         0.11670  \n",
       "...           ...  \n",
       "48824     0.14700  \n",
       "48825     0.00000  \n",
       "48826     0.00000  \n",
       "48831     0.00000  \n",
       "48832     0.00000  \n",
       "\n",
       "[27042 rows x 18 columns]"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/VCC_NEW/12652712_MUTECT2.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['CSQ'].str.split('|').str[56]\n",
    "vcf['gnomADe_AF'] = vcf['CSQ'].str.split('|').str[48]\n",
    "vcf['gnomADe_AF'] = vcf['gnomADe_AF'].replace('', 0).astype('float')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].replace('', 0)\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].astype('float')\n",
    "# For MUTECT2, you can choose not to filter based on 'DP'\n",
    "filtered_data = vcf[vcf['VAF'] >= 0.1]\n",
    "filtered_data = filtered_data[filtered_data['DP'] >= 15]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_AF'] <= 0.5]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_SAS_AF'] <= 0.5]\n",
    "filtered_data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c27cdaa4",
   "metadata": {},
   "source": [
    "# HAPLOTYPECALLER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "a391cd7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786070</td>\n",
       "      <td>rs2977675</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>781.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-1.955;DB;DP=1...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:152,40:192:99:789,0,4530</td>\n",
       "      <td>192</td>\n",
       "      <td>152,40</td>\n",
       "      <td>152</td>\n",
       "      <td>40</td>\n",
       "      <td>0.208333</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.01013</td>\n",
       "      <td>0.01256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786377</td>\n",
       "      <td>rs189147642</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>485.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-2.995;DB;DP=1...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:96,26:122:99:493,0,2909</td>\n",
       "      <td>122</td>\n",
       "      <td>96,26</td>\n",
       "      <td>96</td>\n",
       "      <td>26</td>\n",
       "      <td>0.213115</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.04863</td>\n",
       "      <td>0.03993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826729</td>\n",
       "      <td>rs540936498</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>574.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-0.910;DP=160;...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:119,31:150:99:582,0,3486</td>\n",
       "      <td>150</td>\n",
       "      <td>119,31</td>\n",
       "      <td>119</td>\n",
       "      <td>31</td>\n",
       "      <td>0.206667</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.03310</td>\n",
       "      <td>0.02253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827092</td>\n",
       "      <td>rs145493205</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>134.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=3.306;DB;DP=38...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:29,8:37:99:142,0,747</td>\n",
       "      <td>37</td>\n",
       "      <td>29,8</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>0.216216</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11850</td>\n",
       "      <td>0.11010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827105</td>\n",
       "      <td>rs148989274;rs12095200</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>360.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-1.809;DB;DP=4...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:27,15:42:99:368,0,761</td>\n",
       "      <td>42</td>\n",
       "      <td>27,15</td>\n",
       "      <td>27</td>\n",
       "      <td>15</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11770</td>\n",
       "      <td>0.11670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46162</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154929926</td>\n",
       "      <td>rs1800292</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2037.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=2.629;DB;DP=13...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:56,69:125:99:2045,0,1512</td>\n",
       "      <td>125</td>\n",
       "      <td>56,69</td>\n",
       "      <td>56</td>\n",
       "      <td>69</td>\n",
       "      <td>0.552000</td>\n",
       "      <td>G|synonymous_variant|LOW|F8|ENSG00000185010|Tr...</td>\n",
       "      <td>0.33770</td>\n",
       "      <td>0.14700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46163</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1340.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-1.922;DP=78;E...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:30,45:75:99:1348,0,800</td>\n",
       "      <td>75</td>\n",
       "      <td>30,45</td>\n",
       "      <td>30</td>\n",
       "      <td>45</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46164</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155349551</td>\n",
       "      <td>rs563910</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>324.64</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;BaseQRankSum=-0.560;DP=22;E...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:11,11:22:99:332,0,353</td>\n",
       "      <td>22</td>\n",
       "      <td>11,11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46166</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>rs9650861</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>845.06</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=2;AF=1.00;AN=2;DB;DP=29;ExcessHet=3.0103;FS...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,27:27:81:859,81,0</td>\n",
       "      <td>27</td>\n",
       "      <td>0,27</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46167</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>rs879016544</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>583.06</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=2;AF=1.00;AN=2;DP=22;ExcessHet=3.0103;FS=0....</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,21:21:63:597,63,0</td>\n",
       "      <td>21</td>\n",
       "      <td>0,21</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27061 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS                    rsID REF ALT     QUAL FILTER  \\\n",
       "1      chr1     786070               rs2977675   G   A   781.64   PASS   \n",
       "2      chr1     786377             rs189147642   T   A   485.64   PASS   \n",
       "3      chr1     826729             rs540936498   C   T   574.64   PASS   \n",
       "5      chr1     827092             rs145493205   C   T   134.64   PASS   \n",
       "6      chr1     827105  rs148989274;rs12095200   C   A   360.64   PASS   \n",
       "...     ...        ...                     ...  ..  ..      ...    ...   \n",
       "46162  chrX  154929926               rs1800292   T   G  2037.64   PASS   \n",
       "46163  chrX  155277884                rs559165   G   T  1340.64   PASS   \n",
       "46164  chrX  155349551                rs563910   C   G   324.64   PASS   \n",
       "46166  chrY   11986362               rs9650861   A   C   845.06   PASS   \n",
       "46167  chrY   11986608             rs879016544   T   C   583.06   PASS   \n",
       "\n",
       "                                                    INFO          FORMAT  \\\n",
       "1      AC=1;AF=0.500;AN=2;BaseQRankSum=-1.955;DB;DP=1...  GT:AD:DP:GQ:PL   \n",
       "2      AC=1;AF=0.500;AN=2;BaseQRankSum=-2.995;DB;DP=1...  GT:AD:DP:GQ:PL   \n",
       "3      AC=1;AF=0.500;AN=2;BaseQRankSum=-0.910;DP=160;...  GT:AD:DP:GQ:PL   \n",
       "5      AC=1;AF=0.500;AN=2;BaseQRankSum=3.306;DB;DP=38...  GT:AD:DP:GQ:PL   \n",
       "6      AC=1;AF=0.500;AN=2;BaseQRankSum=-1.809;DB;DP=4...  GT:AD:DP:GQ:PL   \n",
       "...                                                  ...             ...   \n",
       "46162  AC=1;AF=0.500;AN=2;BaseQRankSum=2.629;DB;DP=13...  GT:AD:DP:GQ:PL   \n",
       "46163  AC=1;AF=0.500;AN=2;BaseQRankSum=-1.922;DP=78;E...  GT:AD:DP:GQ:PL   \n",
       "46164  AC=1;AF=0.500;AN=2;BaseQRankSum=-0.560;DP=22;E...  GT:AD:DP:GQ:PL   \n",
       "46166  AC=2;AF=1.00;AN=2;DB;DP=29;ExcessHet=3.0103;FS...  GT:AD:DP:GQ:PL   \n",
       "46167  AC=2;AF=1.00;AN=2;DP=22;ExcessHet=3.0103;FS=0....  GT:AD:DP:GQ:PL   \n",
       "\n",
       "                             SAMPLE   DP      AD   RD  A_D       VAF  \\\n",
       "1      0/1:152,40:192:99:789,0,4530  192  152,40  152   40  0.208333   \n",
       "2       0/1:96,26:122:99:493,0,2909  122   96,26   96   26  0.213115   \n",
       "3      0/1:119,31:150:99:582,0,3486  150  119,31  119   31  0.206667   \n",
       "5          0/1:29,8:37:99:142,0,747   37    29,8   29    8  0.216216   \n",
       "6         0/1:27,15:42:99:368,0,761   42   27,15   27   15  0.357143   \n",
       "...                             ...  ...     ...  ...  ...       ...   \n",
       "46162  0/1:56,69:125:99:2045,0,1512  125   56,69   56   69  0.552000   \n",
       "46163    0/1:30,45:75:99:1348,0,800   75   30,45   30   45  0.600000   \n",
       "46164     0/1:11,11:22:99:332,0,353   22   11,11   11   11  0.500000   \n",
       "46166       1/1:0,27:27:81:859,81,0   27    0,27    0   27  1.000000   \n",
       "46167       1/1:0,21:21:63:597,63,0   21    0,21    0   21  1.000000   \n",
       "\n",
       "                                                     CSQ  gnomADe_SAS_AF  \\\n",
       "1      A|intron_variant&non_coding_transcript_variant...         0.01013   \n",
       "2      A|intron_variant&non_coding_transcript_variant...         0.04863   \n",
       "3      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.03310   \n",
       "5      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11850   \n",
       "6      A|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11770   \n",
       "...                                                  ...             ...   \n",
       "46162  G|synonymous_variant|LOW|F8|ENSG00000185010|Tr...         0.33770   \n",
       "46163  T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...         0.00000   \n",
       "46164  G|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "46166  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "46167  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "\n",
       "       gnomADe_AF  \n",
       "1         0.01256  \n",
       "2         0.03993  \n",
       "3         0.02253  \n",
       "5         0.11010  \n",
       "6         0.11670  \n",
       "...           ...  \n",
       "46162     0.14700  \n",
       "46163     0.00000  \n",
       "46164     0.00000  \n",
       "46166     0.00000  \n",
       "46167     0.00000  \n",
       "\n",
       "[27061 rows x 18 columns]"
      ]
     },
     "execution_count": 289,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/VCC_NEW/12652712_HAPLOTYPECALLER.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['CSQ'].str.split('|').str[56]\n",
    "vcf['gnomADe_AF'] = vcf['CSQ'].str.split('|').str[48]\n",
    "vcf['gnomADe_AF'] = vcf['gnomADe_AF'].replace('', 0).astype('float')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].replace('', 0)\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].astype('float')\n",
    "# For HAPLOTYPECALLER, you can choose not to filter based on 'DP'\n",
    "filtered_data = vcf[vcf['VAF'] >= 0.1]\n",
    "filtered_data = filtered_data[filtered_data['DP'] >= 15]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_AF'] <= 0.5]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_SAS_AF'] <= 0.5]\n",
    "filtered_data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37faeea3",
   "metadata": {},
   "source": [
    "# DEEPVARIANT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "id": "f42dfcb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/VCC_OLD/12652712_DEEPVARIANT.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[3]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['CSQ'].str.split('|').str[56]\n",
    "vcf['gnomADe_AF'] = vcf['CSQ'].str.split('|').str[48]\n",
    "vcf['gnomADe_AF'] = vcf['gnomADe_AF'].replace('', 0).astype('float')\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].replace('', 0)\n",
    "vcf['gnomADe_SAS_AF'] = vcf['gnomADe_SAS_AF'].astype('float')\n",
    "# For DEEPVARIANT, you can choose not to filter based on 'DP'\n",
    "filtered_data = vcf[vcf['VAF'] >= 0.1]\n",
    "filtered_data = filtered_data[filtered_data['DP'] >= 15]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_AF'] <= 0.5]\n",
    "filtered_data = filtered_data[filtered_data['gnomADe_SAS_AF'] <= 0.5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "id": "0b8ddd03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786070</td>\n",
       "      <td>rs2977675</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CSQ=A|intron_variant&amp;non_coding_transcript...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:7:209:144,65:0.311005:0,5,31</td>\n",
       "      <td>209</td>\n",
       "      <td>144,65</td>\n",
       "      <td>144</td>\n",
       "      <td>65</td>\n",
       "      <td>0.311005</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.01013</td>\n",
       "      <td>0.01256</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786377</td>\n",
       "      <td>rs189147642</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CAF=0.9718,0.02815;COMMON=1;CSQ=A|intron_v...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:22:111:89,22:0.198198:0,22,39</td>\n",
       "      <td>111</td>\n",
       "      <td>89,22</td>\n",
       "      <td>89</td>\n",
       "      <td>22</td>\n",
       "      <td>0.198198</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.04863</td>\n",
       "      <td>0.03993</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826729</td>\n",
       "      <td>rs540936498</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4.4</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/1:4:244:176,67:0.27459:2,0,31</td>\n",
       "      <td>244</td>\n",
       "      <td>176,67</td>\n",
       "      <td>176</td>\n",
       "      <td>67</td>\n",
       "      <td>0.275720</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.03310</td>\n",
       "      <td>0.02253</td>\n",
       "      <td>1:826729</td>\n",
       "      <td>C/T</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.794G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD Ashkenazi Jewish</td>\n",
       "      <td>0.0533499</td>\n",
       "      <td>7254</td>\n",
       "      <td>387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827092</td>\n",
       "      <td>rs145493205</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6.2</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/1:6:58:46,12:0.206897:4,0,23</td>\n",
       "      <td>58</td>\n",
       "      <td>46,12</td>\n",
       "      <td>46</td>\n",
       "      <td>12</td>\n",
       "      <td>0.206897</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11850</td>\n",
       "      <td>0.11010</td>\n",
       "      <td>1:827092</td>\n",
       "      <td>C/T</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.431G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD African</td>\n",
       "      <td>0.296989</td>\n",
       "      <td>6044</td>\n",
       "      <td>1795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827105</td>\n",
       "      <td>rs12095200</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CAF=0.8339,0.1661;COMMON=1;CSQ=A|upstream_...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:4:66:44,22:0.333333:0,2,19</td>\n",
       "      <td>66</td>\n",
       "      <td>44,22</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11770</td>\n",
       "      <td>0.11670</td>\n",
       "      <td>1:827105</td>\n",
       "      <td>C/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.418G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD African</td>\n",
       "      <td>0.356906</td>\n",
       "      <td>6038</td>\n",
       "      <td>2155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28453</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155349551</td>\n",
       "      <td>rs563910</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.5</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CSQ=G|intron_variant&amp;non_coding_transcript...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:10:63:45,15:0.238095:0,11,13</td>\n",
       "      <td>63</td>\n",
       "      <td>45,15</td>\n",
       "      <td>45</td>\n",
       "      <td>15</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>X:155349551</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LOC101927830</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>101927830</td>\n",
       "      <td>NR_109985.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_109985.1:n.472-1184G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>PM2,BP7</td>\n",
       "      <td>LOC101927830:Variant is missing from all sub p...</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28454</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155349553</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2.3</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>CSQ=G|intron_variant&amp;non_coding_transcript_var...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:4:66:44,20:0.30303:0,14,1</td>\n",
       "      <td>66</td>\n",
       "      <td>44,20</td>\n",
       "      <td>44</td>\n",
       "      <td>20</td>\n",
       "      <td>0.312500</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28455</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5615222</td>\n",
       "      <td>rs2557028</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>31.9</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=A|intron_variant|MODIFIER|PCDH11Y|ENSG...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:31:22:1,21:0.954545:31,37,0</td>\n",
       "      <td>22</td>\n",
       "      <td>1,21</td>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>0.954545</td>\n",
       "      <td>A|intron_variant|MODIFIER|PCDH11Y|ENSG00000099...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28456</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>rs9650861</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>36.9</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=C|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:35:37:2,35:0.945946:36,40,0</td>\n",
       "      <td>37</td>\n",
       "      <td>2,35</td>\n",
       "      <td>2</td>\n",
       "      <td>35</td>\n",
       "      <td>0.945946</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>Y:11986362</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28457</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>rs879016544</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>35.5</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=C|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:34:28:0,28:1:35,38,0</td>\n",
       "      <td>28</td>\n",
       "      <td>0,28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>Y:11986608</td>\n",
       "      <td>T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28458 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT  QUAL   FILTER  \\\n",
       "0      chr1     786070    rs2977675   G   A   1.0  RefCall   \n",
       "1      chr1     786377  rs189147642   T   A   0.0  RefCall   \n",
       "2      chr1     826729  rs540936498   C   T   4.4     PASS   \n",
       "3      chr1     827092  rs145493205   C   T   6.2     PASS   \n",
       "4      chr1     827105   rs12095200   C   A   2.0  RefCall   \n",
       "...     ...        ...          ...  ..  ..   ...      ...   \n",
       "28453  chrX  155349551     rs563910   C   G   0.5  RefCall   \n",
       "28454  chrX  155349553            .   T   G   2.3  RefCall   \n",
       "28455  chrY    5615222    rs2557028   T   A  31.9     PASS   \n",
       "28456  chrY   11986362    rs9650861   A   C  36.9     PASS   \n",
       "28457  chrY   11986608  rs879016544   T   C  35.5     PASS   \n",
       "\n",
       "                                                    INFO              FORMAT  \\\n",
       "0      ASP;CSQ=A|intron_variant&non_coding_transcript...  GT:GQ:DP:AD:VAF:PL   \n",
       "1      ASP;CAF=0.9718,0.02815;COMMON=1;CSQ=A|intron_v...  GT:GQ:DP:AD:VAF:PL   \n",
       "2      ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...  GT:GQ:DP:AD:VAF:PL   \n",
       "3      ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...  GT:GQ:DP:AD:VAF:PL   \n",
       "4      ASP;CAF=0.8339,0.1661;COMMON=1;CSQ=A|upstream_...  GT:GQ:DP:AD:VAF:PL   \n",
       "...                                                  ...                 ...   \n",
       "28453  ASP;CSQ=G|intron_variant&non_coding_transcript...  GT:GQ:DP:AD:VAF:PL   \n",
       "28454  CSQ=G|intron_variant&non_coding_transcript_var...  GT:GQ:DP:AD:VAF:PL   \n",
       "28455  ASP;CSQ=A|intron_variant|MODIFIER|PCDH11Y|ENSG...  GT:GQ:DP:AD:VAF:PL   \n",
       "28456  ASP;CSQ=C|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "28457  ASP;CSQ=C|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "\n",
       "                                  SAMPLE   DP      AD   RD  A_D       VAF  \\\n",
       "0       ./.:7:209:144,65:0.311005:0,5,31  209  144,65  144   65  0.311005   \n",
       "1      0/0:22:111:89,22:0.198198:0,22,39  111   89,22   89   22  0.198198   \n",
       "2        0/1:4:244:176,67:0.27459:2,0,31  244  176,67  176   67  0.275720   \n",
       "3         0/1:6:58:46,12:0.206897:4,0,23   58   46,12   46   12  0.206897   \n",
       "4         ./.:4:66:44,22:0.333333:0,2,19   66   44,22   44   22  0.333333   \n",
       "...                                  ...  ...     ...  ...  ...       ...   \n",
       "28453   ./.:10:63:45,15:0.238095:0,11,13   63   45,15   45   15  0.250000   \n",
       "28454      ./.:4:66:44,20:0.30303:0,14,1   66   44,20   44   20  0.312500   \n",
       "28455    1/1:31:22:1,21:0.954545:31,37,0   22    1,21    1   21  0.954545   \n",
       "28456    1/1:35:37:2,35:0.945946:36,40,0   37    2,35    2   35  0.945946   \n",
       "28457           1/1:34:28:0,28:1:35,38,0   28    0,28    0   28  1.000000   \n",
       "\n",
       "                                                     CSQ  gnomADe_SAS_AF  \\\n",
       "0      A|intron_variant&non_coding_transcript_variant...         0.01013   \n",
       "1      A|intron_variant&non_coding_transcript_variant...         0.04863   \n",
       "2      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.03310   \n",
       "3      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11850   \n",
       "4      A|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11770   \n",
       "...                                                  ...             ...   \n",
       "28453  G|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "28454  G|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "28455  A|intron_variant|MODIFIER|PCDH11Y|ENSG00000099...         0.00000   \n",
       "28456  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "28457  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "\n",
       "       gnomADe_AF      Chr:Pos Ref/Alt     Gene Name     Gene Inheritance  \\\n",
       "0         0.01256          NaN     NaN           NaN                  NaN   \n",
       "1         0.03993          NaN     NaN           NaN                  NaN   \n",
       "2         0.02253     1:826729     C/T     LINC00115  Default (Recessive)   \n",
       "3         0.11010     1:827092     C/T     LINC00115  Default (Recessive)   \n",
       "4         0.11670     1:827105     C/A     LINC00115  Default (Recessive)   \n",
       "...           ...          ...     ...           ...                  ...   \n",
       "28453     0.00000  X:155349551     C/G  LOC101927830  Default (Recessive)   \n",
       "28454     0.00000          NaN     NaN           NaN                  NaN   \n",
       "28455     0.00000          NaN     NaN           NaN                  NaN   \n",
       "28456     0.00000   Y:11986362     A/C           NaN                  NaN   \n",
       "28457     0.00000   Y:11986608     T/C           NaN                  NaN   \n",
       "\n",
       "      Entrez Gene ID Transcript Name Transcript Strand  \\\n",
       "0                NaN             NaN               NaN   \n",
       "1                NaN             NaN               NaN   \n",
       "2              79854     NR_024321.1                 -   \n",
       "3              79854     NR_024321.1                 -   \n",
       "4              79854     NR_024321.1                 -   \n",
       "...              ...             ...               ...   \n",
       "28453      101927830     NR_109985.1                 -   \n",
       "28454            NaN             NaN               NaN   \n",
       "28455            NaN             NaN               NaN   \n",
       "28456            NaN             NaN               NaN   \n",
       "28457            NaN             NaN               NaN   \n",
       "\n",
       "                       HGVS cDot HGVS pDot        Sequence Ontology  \\\n",
       "0                            NaN       NaN                      NaN   \n",
       "1                            NaN       NaN                      NaN   \n",
       "2           NR_024321.1:n.794G>A       NaN  non_coding_exon_variant   \n",
       "3           NR_024321.1:n.431G>A       NaN  non_coding_exon_variant   \n",
       "4           NR_024321.1:n.418G>T       NaN  non_coding_exon_variant   \n",
       "...                          ...       ...                      ...   \n",
       "28453  NR_109985.1:n.472-1184G>C       NaN           intron_variant   \n",
       "28454                        NaN       NaN                      NaN   \n",
       "28455                        NaN       NaN                      NaN   \n",
       "28456                        NaN       NaN                      NaN   \n",
       "28457                        NaN       NaN                      NaN   \n",
       "\n",
       "      Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                          NaN                          NaN   \n",
       "1                          NaN                          NaN   \n",
       "2       Single Transcript Gene                      BA1,BP7   \n",
       "3       Single Transcript Gene                      BA1,BP7   \n",
       "4       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                        ...                          ...   \n",
       "28453   Single Transcript Gene                      PM2,BP7   \n",
       "28454                      NaN                          NaN   \n",
       "28455                      NaN                          NaN   \n",
       "28456                      NaN                          NaN   \n",
       "28457                      NaN                          NaN   \n",
       "\n",
       "                ACMG Classification Criteria Description   Classification  \\\n",
       "0                                                    NaN               NA   \n",
       "1                                                    NaN               NA   \n",
       "2      LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "3      LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "4      LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "...                                                  ...              ...   \n",
       "28453  LOC101927830:Variant is missing from all sub p...  VUS/Conflicting   \n",
       "28454                                                NaN               NA   \n",
       "28455                                                NaN               NA   \n",
       "28456                                                NaN               NA   \n",
       "28457                                                NaN               NA   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                     NaN                           NaN   \n",
       "1                     NaN                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                  Benign                           NaN   \n",
       "4                  Benign                           NaN   \n",
       "...                   ...                           ...   \n",
       "28453     VUS/Conflicting                           NaN   \n",
       "28454                 NaN                           NaN   \n",
       "28455                 NaN                           NaN   \n",
       "28456                 NaN                           NaN   \n",
       "28457                 NaN                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "28453                     NaN                      NaN   \n",
       "28454                     NaN                      NaN   \n",
       "28455                     NaN                      NaN   \n",
       "28456                     NaN                      NaN   \n",
       "28457                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name    Max MAF Max Allele Number  \\\n",
       "0                                    NaN        NaN               NaN   \n",
       "1                                    NaN        NaN               NaN   \n",
       "2      Annotated gnomAD Ashkenazi Jewish  0.0533499              7254   \n",
       "3               Annotated gnomAD African   0.296989              6044   \n",
       "4               Annotated gnomAD African   0.356906              6038   \n",
       "...                                  ...        ...               ...   \n",
       "28453                                NaN        NaN               NaN   \n",
       "28454                                NaN        NaN               NaN   \n",
       "28455                                NaN        NaN               NaN   \n",
       "28456                                NaN        NaN               NaN   \n",
       "28457                                NaN        NaN               NaN   \n",
       "\n",
       "      Max Allele Count  \n",
       "0                  NaN  \n",
       "1                  NaN  \n",
       "2                  387  \n",
       "3                 1795  \n",
       "4                 2155  \n",
       "...                ...  \n",
       "28453              NaN  \n",
       "28454              NaN  \n",
       "28455              NaN  \n",
       "28456              NaN  \n",
       "28457              NaN  \n",
       "\n",
       "[28458 rows x 40 columns]"
      ]
     },
     "execution_count": 353,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_data = pd.merge(filtered_data, truth_1, on = ['CHROM', 'POS', 'REF', 'ALT'])\n",
    "filtered_data = filtered_data.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "# Merge the VCF data with the main data (df)\n",
    "merged_data = pd.merge(filtered_data, df, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_data = merged_data.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "# To fill NaN values in the 'Classification' column with 'NA', you need to assign the result back to the column.\n",
    "merged_data['Classification'] = merged_data['Classification'].fillna('NA')\n",
    "merged_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "id": "16b5f725",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786070</td>\n",
       "      <td>rs2977675</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CSQ=A|intron_variant&amp;non_coding_transcript...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:7:209:144,65:0.311005:0,5,31</td>\n",
       "      <td>209</td>\n",
       "      <td>144,65</td>\n",
       "      <td>144</td>\n",
       "      <td>65</td>\n",
       "      <td>0.311005</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.01013</td>\n",
       "      <td>0.01256</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>786377</td>\n",
       "      <td>rs189147642</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CAF=0.9718,0.02815;COMMON=1;CSQ=A|intron_v...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/0:22:111:89,22:0.198198:0,22,39</td>\n",
       "      <td>111</td>\n",
       "      <td>89,22</td>\n",
       "      <td>89</td>\n",
       "      <td>22</td>\n",
       "      <td>0.198198</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.04863</td>\n",
       "      <td>0.03993</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826729</td>\n",
       "      <td>rs540936498</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4.4</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/1:4:244:176,67:0.27459:2,0,31</td>\n",
       "      <td>244</td>\n",
       "      <td>176,67</td>\n",
       "      <td>176</td>\n",
       "      <td>67</td>\n",
       "      <td>0.275720</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.03310</td>\n",
       "      <td>0.02253</td>\n",
       "      <td>1:826729</td>\n",
       "      <td>C/T</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.794G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD Ashkenazi Jewish</td>\n",
       "      <td>0.0533499</td>\n",
       "      <td>7254</td>\n",
       "      <td>387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827092</td>\n",
       "      <td>rs145493205</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6.2</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>0/1:6:58:46,12:0.206897:4,0,23</td>\n",
       "      <td>58</td>\n",
       "      <td>46,12</td>\n",
       "      <td>46</td>\n",
       "      <td>12</td>\n",
       "      <td>0.206897</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11850</td>\n",
       "      <td>0.11010</td>\n",
       "      <td>1:827092</td>\n",
       "      <td>C/T</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.431G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD African</td>\n",
       "      <td>0.296989</td>\n",
       "      <td>6044</td>\n",
       "      <td>1795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827105</td>\n",
       "      <td>rs12095200</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CAF=0.8339,0.1661;COMMON=1;CSQ=A|upstream_...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:4:66:44,22:0.333333:0,2,19</td>\n",
       "      <td>66</td>\n",
       "      <td>44,22</td>\n",
       "      <td>44</td>\n",
       "      <td>22</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>A|upstream_gene_variant|MODIFIER|LINC01128|ENS...</td>\n",
       "      <td>0.11770</td>\n",
       "      <td>0.11670</td>\n",
       "      <td>1:827105</td>\n",
       "      <td>C/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.418G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD African</td>\n",
       "      <td>0.356906</td>\n",
       "      <td>6038</td>\n",
       "      <td>2155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28453</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155349551</td>\n",
       "      <td>rs563910</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.5</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>ASP;CSQ=G|intron_variant&amp;non_coding_transcript...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:10:63:45,15:0.238095:0,11,13</td>\n",
       "      <td>63</td>\n",
       "      <td>45,15</td>\n",
       "      <td>45</td>\n",
       "      <td>15</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>X:155349551</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LOC101927830</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>101927830</td>\n",
       "      <td>NR_109985.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_109985.1:n.472-1184G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>PM2,BP7</td>\n",
       "      <td>LOC101927830:Variant is missing from all sub p...</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28454</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155349553</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2.3</td>\n",
       "      <td>RefCall</td>\n",
       "      <td>CSQ=G|intron_variant&amp;non_coding_transcript_var...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>./.:4:66:44,20:0.30303:0,14,1</td>\n",
       "      <td>66</td>\n",
       "      <td>44,20</td>\n",
       "      <td>44</td>\n",
       "      <td>20</td>\n",
       "      <td>0.312500</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28455</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5615222</td>\n",
       "      <td>rs2557028</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>31.9</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=A|intron_variant|MODIFIER|PCDH11Y|ENSG...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:31:22:1,21:0.954545:31,37,0</td>\n",
       "      <td>22</td>\n",
       "      <td>1,21</td>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>0.954545</td>\n",
       "      <td>A|intron_variant|MODIFIER|PCDH11Y|ENSG00000099...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28456</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>rs9650861</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>36.9</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=C|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:35:37:2,35:0.945946:36,40,0</td>\n",
       "      <td>37</td>\n",
       "      <td>2,35</td>\n",
       "      <td>2</td>\n",
       "      <td>35</td>\n",
       "      <td>0.945946</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>Y:11986362</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28457</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>rs879016544</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>35.5</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=C|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:34:28:0,28:1:35,38,0</td>\n",
       "      <td>28</td>\n",
       "      <td>0,28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>Y:11986608</td>\n",
       "      <td>T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28458 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT  QUAL   FILTER  \\\n",
       "0      chr1     786070    rs2977675   G   A   1.0  RefCall   \n",
       "1      chr1     786377  rs189147642   T   A   0.0  RefCall   \n",
       "2      chr1     826729  rs540936498   C   T   4.4     PASS   \n",
       "3      chr1     827092  rs145493205   C   T   6.2     PASS   \n",
       "4      chr1     827105   rs12095200   C   A   2.0  RefCall   \n",
       "...     ...        ...          ...  ..  ..   ...      ...   \n",
       "28453  chrX  155349551     rs563910   C   G   0.5  RefCall   \n",
       "28454  chrX  155349553            .   T   G   2.3  RefCall   \n",
       "28455  chrY    5615222    rs2557028   T   A  31.9     PASS   \n",
       "28456  chrY   11986362    rs9650861   A   C  36.9     PASS   \n",
       "28457  chrY   11986608  rs879016544   T   C  35.5     PASS   \n",
       "\n",
       "                                                    INFO              FORMAT  \\\n",
       "0      ASP;CSQ=A|intron_variant&non_coding_transcript...  GT:GQ:DP:AD:VAF:PL   \n",
       "1      ASP;CAF=0.9718,0.02815;COMMON=1;CSQ=A|intron_v...  GT:GQ:DP:AD:VAF:PL   \n",
       "2      ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...  GT:GQ:DP:AD:VAF:PL   \n",
       "3      ASP;CSQ=T|upstream_gene_variant|MODIFIER|LINC0...  GT:GQ:DP:AD:VAF:PL   \n",
       "4      ASP;CAF=0.8339,0.1661;COMMON=1;CSQ=A|upstream_...  GT:GQ:DP:AD:VAF:PL   \n",
       "...                                                  ...                 ...   \n",
       "28453  ASP;CSQ=G|intron_variant&non_coding_transcript...  GT:GQ:DP:AD:VAF:PL   \n",
       "28454  CSQ=G|intron_variant&non_coding_transcript_var...  GT:GQ:DP:AD:VAF:PL   \n",
       "28455  ASP;CSQ=A|intron_variant|MODIFIER|PCDH11Y|ENSG...  GT:GQ:DP:AD:VAF:PL   \n",
       "28456  ASP;CSQ=C|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "28457  ASP;CSQ=C|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "\n",
       "                                  SAMPLE   DP      AD   RD  A_D       VAF  \\\n",
       "0       ./.:7:209:144,65:0.311005:0,5,31  209  144,65  144   65  0.311005   \n",
       "1      0/0:22:111:89,22:0.198198:0,22,39  111   89,22   89   22  0.198198   \n",
       "2        0/1:4:244:176,67:0.27459:2,0,31  244  176,67  176   67  0.275720   \n",
       "3         0/1:6:58:46,12:0.206897:4,0,23   58   46,12   46   12  0.206897   \n",
       "4         ./.:4:66:44,22:0.333333:0,2,19   66   44,22   44   22  0.333333   \n",
       "...                                  ...  ...     ...  ...  ...       ...   \n",
       "28453   ./.:10:63:45,15:0.238095:0,11,13   63   45,15   45   15  0.250000   \n",
       "28454      ./.:4:66:44,20:0.30303:0,14,1   66   44,20   44   20  0.312500   \n",
       "28455    1/1:31:22:1,21:0.954545:31,37,0   22    1,21    1   21  0.954545   \n",
       "28456    1/1:35:37:2,35:0.945946:36,40,0   37    2,35    2   35  0.945946   \n",
       "28457           1/1:34:28:0,28:1:35,38,0   28    0,28    0   28  1.000000   \n",
       "\n",
       "                                                     CSQ  gnomADe_SAS_AF  \\\n",
       "0      A|intron_variant&non_coding_transcript_variant...         0.01013   \n",
       "1      A|intron_variant&non_coding_transcript_variant...         0.04863   \n",
       "2      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.03310   \n",
       "3      T|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11850   \n",
       "4      A|upstream_gene_variant|MODIFIER|LINC01128|ENS...         0.11770   \n",
       "...                                                  ...             ...   \n",
       "28453  G|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "28454  G|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "28455  A|intron_variant|MODIFIER|PCDH11Y|ENSG00000099...         0.00000   \n",
       "28456  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "28457  C|non_coding_transcript_exon_variant|MODIFIER|...         0.00000   \n",
       "\n",
       "       gnomADe_AF      Chr:Pos Ref/Alt     Gene Name     Gene Inheritance  \\\n",
       "0         0.01256          NaN     NaN           NaN                  NaN   \n",
       "1         0.03993          NaN     NaN           NaN                  NaN   \n",
       "2         0.02253     1:826729     C/T     LINC00115  Default (Recessive)   \n",
       "3         0.11010     1:827092     C/T     LINC00115  Default (Recessive)   \n",
       "4         0.11670     1:827105     C/A     LINC00115  Default (Recessive)   \n",
       "...           ...          ...     ...           ...                  ...   \n",
       "28453     0.00000  X:155349551     C/G  LOC101927830  Default (Recessive)   \n",
       "28454     0.00000          NaN     NaN           NaN                  NaN   \n",
       "28455     0.00000          NaN     NaN           NaN                  NaN   \n",
       "28456     0.00000   Y:11986362     A/C           NaN                  NaN   \n",
       "28457     0.00000   Y:11986608     T/C           NaN                  NaN   \n",
       "\n",
       "      Entrez Gene ID Transcript Name Transcript Strand  \\\n",
       "0                NaN             NaN               NaN   \n",
       "1                NaN             NaN               NaN   \n",
       "2              79854     NR_024321.1                 -   \n",
       "3              79854     NR_024321.1                 -   \n",
       "4              79854     NR_024321.1                 -   \n",
       "...              ...             ...               ...   \n",
       "28453      101927830     NR_109985.1                 -   \n",
       "28454            NaN             NaN               NaN   \n",
       "28455            NaN             NaN               NaN   \n",
       "28456            NaN             NaN               NaN   \n",
       "28457            NaN             NaN               NaN   \n",
       "\n",
       "                       HGVS cDot HGVS pDot        Sequence Ontology  \\\n",
       "0                            NaN       NaN                      NaN   \n",
       "1                            NaN       NaN                      NaN   \n",
       "2           NR_024321.1:n.794G>A       NaN  non_coding_exon_variant   \n",
       "3           NR_024321.1:n.431G>A       NaN  non_coding_exon_variant   \n",
       "4           NR_024321.1:n.418G>T       NaN  non_coding_exon_variant   \n",
       "...                          ...       ...                      ...   \n",
       "28453  NR_109985.1:n.472-1184G>C       NaN           intron_variant   \n",
       "28454                        NaN       NaN                      NaN   \n",
       "28455                        NaN       NaN                      NaN   \n",
       "28456                        NaN       NaN                      NaN   \n",
       "28457                        NaN       NaN                      NaN   \n",
       "\n",
       "      Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                          NaN                          NaN   \n",
       "1                          NaN                          NaN   \n",
       "2       Single Transcript Gene                      BA1,BP7   \n",
       "3       Single Transcript Gene                      BA1,BP7   \n",
       "4       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                        ...                          ...   \n",
       "28453   Single Transcript Gene                      PM2,BP7   \n",
       "28454                      NaN                          NaN   \n",
       "28455                      NaN                          NaN   \n",
       "28456                      NaN                          NaN   \n",
       "28457                      NaN                          NaN   \n",
       "\n",
       "                ACMG Classification Criteria Description   Classification  \\\n",
       "0                                                    NaN             None   \n",
       "1                                                    NaN             None   \n",
       "2      LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "3      LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "4      LINC00115:Allele frequency is above 0.01 reces...           Benign   \n",
       "...                                                  ...              ...   \n",
       "28453  LOC101927830:Variant is missing from all sub p...  VUS/Conflicting   \n",
       "28454                                                NaN             None   \n",
       "28455                                                NaN             None   \n",
       "28456                                                NaN             None   \n",
       "28457                                                NaN             None   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                     NaN                           NaN   \n",
       "1                     NaN                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                  Benign                           NaN   \n",
       "4                  Benign                           NaN   \n",
       "...                   ...                           ...   \n",
       "28453     VUS/Conflicting                           NaN   \n",
       "28454                 NaN                           NaN   \n",
       "28455                 NaN                           NaN   \n",
       "28456                 NaN                           NaN   \n",
       "28457                 NaN                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "28453                     NaN                      NaN   \n",
       "28454                     NaN                      NaN   \n",
       "28455                     NaN                      NaN   \n",
       "28456                     NaN                      NaN   \n",
       "28457                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name    Max MAF Max Allele Number  \\\n",
       "0                                    NaN        NaN               NaN   \n",
       "1                                    NaN        NaN               NaN   \n",
       "2      Annotated gnomAD Ashkenazi Jewish  0.0533499              7254   \n",
       "3               Annotated gnomAD African   0.296989              6044   \n",
       "4               Annotated gnomAD African   0.356906              6038   \n",
       "...                                  ...        ...               ...   \n",
       "28453                                NaN        NaN               NaN   \n",
       "28454                                NaN        NaN               NaN   \n",
       "28455                                NaN        NaN               NaN   \n",
       "28456                                NaN        NaN               NaN   \n",
       "28457                                NaN        NaN               NaN   \n",
       "\n",
       "      Max Allele Count  \n",
       "0                  NaN  \n",
       "1                  NaN  \n",
       "2                  387  \n",
       "3                 1795  \n",
       "4                 2155  \n",
       "...                ...  \n",
       "28453              NaN  \n",
       "28454              NaN  \n",
       "28455              NaN  \n",
       "28456              NaN  \n",
       "28457              NaN  \n",
       "\n",
       "[28458 rows x 40 columns]"
      ]
     },
     "execution_count": 354,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of your specified priorities\n",
    "priorities = ['Pathogenic', 'Likely Pathogenic', 'VUS/Weak Pathogenic', 'VUS/Conflicting', 'VUS', 'VUS/Weak Benign', 'Likely Benign', 'Benign']\n",
    "\n",
    "# Function to select the highest ranked value from Original_Column\n",
    "def select_highest_ranked(row):\n",
    "    values = row['Classification'].split(',')\n",
    "    for rank in priorities:\n",
    "        if rank in values:\n",
    "            return rank\n",
    "    return None  # Return None if no rank found\n",
    "\n",
    "# Apply the function to create the final selected rank column\n",
    "merged_data['Classification'] = merged_data.apply(select_highest_ranked, axis=1)\n",
    "merged_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "id": "bb62db42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Benign                 23320\n",
       "None                    2906\n",
       "VUS/Conflicting          883\n",
       "Likely Benign            819\n",
       "VUS/Weak Pathogenic      299\n",
       "VUS/Weak Benign          189\n",
       "VUS                       28\n",
       "Likely Pathogenic         14\n",
       "Name: Classification, dtype: int64"
      ]
     },
     "execution_count": 355,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_data.Classification.value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "id": "86c5a593",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28458"
      ]
     },
     "execution_count": 356,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_data.Classification.value_counts(dropna=False).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "id": "d18c97e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1850</th>\n",
       "      <td>chr1</td>\n",
       "      <td>160136997</td>\n",
       "      <td>rs201048446</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7119</th>\n",
       "      <td>chr12</td>\n",
       "      <td>113105712</td>\n",
       "      <td>rs202143463</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11365</th>\n",
       "      <td>chr17</td>\n",
       "      <td>21300898</td>\n",
       "      <td>rs55796947</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11464</th>\n",
       "      <td>chr17</td>\n",
       "      <td>29097221</td>\n",
       "      <td>rs758984673</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14104</th>\n",
       "      <td>chr19</td>\n",
       "      <td>38734140</td>\n",
       "      <td>rs759651004</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15442</th>\n",
       "      <td>chr2</td>\n",
       "      <td>60961461</td>\n",
       "      <td>rs200065642</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16201</th>\n",
       "      <td>chr2</td>\n",
       "      <td>160062293</td>\n",
       "      <td>rs778403226</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17579</th>\n",
       "      <td>chr20</td>\n",
       "      <td>46128329</td>\n",
       "      <td>rs78782129</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17841</th>\n",
       "      <td>chr21</td>\n",
       "      <td>10569701</td>\n",
       "      <td>rs1810540</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19130</th>\n",
       "      <td>chr3</td>\n",
       "      <td>14540267</td>\n",
       "      <td>rs746394100</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21299</th>\n",
       "      <td>chr4</td>\n",
       "      <td>109828135</td>\n",
       "      <td>rs749547068</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24907</th>\n",
       "      <td>chr7</td>\n",
       "      <td>93294638</td>\n",
       "      <td>rs75893203</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25072</th>\n",
       "      <td>chr7</td>\n",
       "      <td>100960609</td>\n",
       "      <td>rs79874934</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25637</th>\n",
       "      <td>chr7</td>\n",
       "      <td>154889456</td>\n",
       "      <td>rs753126853</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         rsID REF ALT\n",
       "1850    chr1  160136997  rs201048446   A   C\n",
       "7119   chr12  113105712  rs202143463   A   C\n",
       "11365  chr17   21300898   rs55796947   C   T\n",
       "11464  chr17   29097221  rs758984673   A   C\n",
       "14104  chr19   38734140  rs759651004   A   C\n",
       "15442   chr2   60961461  rs200065642   A   C\n",
       "16201   chr2  160062293  rs778403226   A   C\n",
       "17579  chr20   46128329   rs78782129   G   A\n",
       "17841  chr21   10569701    rs1810540   C   T\n",
       "19130   chr3   14540267  rs746394100   A   C\n",
       "21299   chr4  109828135  rs749547068   T   G\n",
       "24907   chr7   93294638   rs75893203   T   G\n",
       "25072   chr7  100960609   rs79874934   C   T\n",
       "25637   chr7  154889456  rs753126853   G   A"
      ]
     },
     "execution_count": 357,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = merged_data[merged_data['Classification'] == 'Likely Pathogenic']\n",
    "#x = merged_data[merged_data['Classification'].isin(['Likely Pathogenic', 'Pathogenic'])]\n",
    "x = x[['CHROM', 'POS', 'rsID', 'REF', 'ALT']]\n",
    "x.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/VCC_OLD/TP_variants/12652712_DEEPVARIANT.xlsx', index=False)\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50897e64",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17f78bfb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "id": "5aca9f8b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23447</th>\n",
       "      <td>chr6</td>\n",
       "      <td>49457801</td>\n",
       "      <td>rs121918258</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT\n",
       "23447  chr6  49457801  rs121918258   C   A"
      ]
     },
     "execution_count": 345,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = merged_data[merged_data['Classification'] == 'Pathogenic']\n",
    "#x = merged_data[merged_data['Classification'].isin(['Likely Pathogenic', 'Pathogenic'])]\n",
    "x = x[['CHROM', 'POS', 'rsID', 'REF', 'ALT']]\n",
    "x.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/VCC_OLD/TP_variants/12652709_DEEPVARIANT_Pathogenic.xlsx', index=False)\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4677eb18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=71;ADF=0,54;ADR=0,11;AD=0,65;SCR=0;VDB=0.49...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,196,0:65:0:0,54:0,11:0,65:0</td>\n",
       "      <td>71</td>\n",
       "      <td>0,65</td>\n",
       "      <td>0</td>\n",
       "      <td>65</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMD11|ENSG000001876...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1:930939</td>\n",
       "      <td>G/A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.792-100G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>None</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.953474</td>\n",
       "      <td>5008</td>\n",
       "      <td>4775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=20;ADF=0,11;ADR=0,7;AD=0,18;SCR=0;VDB=0.165...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,54,0:18:0:0,11:0,7:0,18:0</td>\n",
       "      <td>20</td>\n",
       "      <td>0,18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1:944858</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>26155</td>\n",
       "      <td>NM_015658.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_015658.4:c.2144-58T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>NOC2L:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>None</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.920927</td>\n",
       "      <td>5008</td>\n",
       "      <td>4612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946538</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>222.4010</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=54;ADF=9,11;ADR=13,12;AD=22,23;SCR=0;VDB=0....</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:255,0,255:45:1:9,11:13,12:22,23:0</td>\n",
       "      <td>54</td>\n",
       "      <td>22,23</td>\n",
       "      <td>22</td>\n",
       "      <td>23</td>\n",
       "      <td>0.511111</td>\n",
       "      <td>A|missense_variant|MODERATE|NOC2L|ENSG00000188...</td>\n",
       "      <td>0.06653</td>\n",
       "      <td>0.04557</td>\n",
       "      <td>1:946538</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>26155</td>\n",
       "      <td>NM_015658.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_015658.4:c.1667C&gt;T</td>\n",
       "      <td>NP_056473.3:p.Ser556Leu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,PP3</td>\n",
       "      <td>NOC2L:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>None</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD South Asian</td>\n",
       "      <td>0.066289</td>\n",
       "      <td>30714</td>\n",
       "      <td>2036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>948711</td>\n",
       "      <td>rs7522415</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>40.5469</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=15;ADF=0,1;ADR=10,4;AD=10,5;SCR=0;VDB=0.240...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:74,0,148:15:0:0,1:10,4:10,5:0</td>\n",
       "      <td>15</td>\n",
       "      <td>10,5</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>957353</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>174.4050</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=34;ADF=0,1;ADR=16,14;AD=16,15;SCR=0;VDB=0.0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:207,0,172:31:0:0,1:16,14:16,15:0</td>\n",
       "      <td>34</td>\n",
       "      <td>16,15</td>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "      <td>0.483871</td>\n",
       "      <td>T|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1:957353</td>\n",
       "      <td>C/T</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>26155</td>\n",
       "      <td>NM_015658.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_015658.4:c.180-80G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>PM2,BP7</td>\n",
       "      <td>NOC2L:Variant is missing from all sub populati...</td>\n",
       "      <td>None</td>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24462</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=27;ADF=0,16;ADR=0,8;AD=0,24;SCR=0;VDB=0.740...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,72,0:24:0:0,16:0,8:0,24:0</td>\n",
       "      <td>27</td>\n",
       "      <td>0,24</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>X:155277884</td>\n",
       "      <td>G/T</td>\n",
       "      <td>CLIC2</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>1193</td>\n",
       "      <td>NM_001289.6</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_001289.6:c.*19C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP4,BP7</td>\n",
       "      <td>CLIC2:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>None</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD Ashkenazi Jewish</td>\n",
       "      <td>0.772765</td>\n",
       "      <td>7248</td>\n",
       "      <td>5601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24463</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1304050</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>38.4150</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=83;ADF=0,9;ADR=0,65;AD=0,74;SCR=0;VDB=0.369...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:68,223,0:74:0:0,9:0,65:0,74:0</td>\n",
       "      <td>83</td>\n",
       "      <td>0,74</td>\n",
       "      <td>0</td>\n",
       "      <td>74</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>C|intron_variant|MODIFIER|CSF2RA|ENSG000001982...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24464</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1418109</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>20.4535</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=95;ADF=0,25;ADR=0,59;AD=0,84;SCR=0;VDB=0.08...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:50,253,0:84:0:0,25:0,59:0,84:0</td>\n",
       "      <td>95</td>\n",
       "      <td>0,84</td>\n",
       "      <td>0</td>\n",
       "      <td>84</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>G|synonymous_variant|LOW|ASMTL|ENSG00000169093...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24465</th>\n",
       "      <td>chrY</td>\n",
       "      <td>6872663</td>\n",
       "      <td>rs34551924</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>214.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=20;ADF=0,1;ADR=0,18;AD=0,19;SCR=0;VDB=0.031...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:244,57,0:19:0:0,1:0,18:0,19:0</td>\n",
       "      <td>20</td>\n",
       "      <td>0,19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>G|upstream_gene_variant|MODIFIER|AMELY|ENSG000...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>Y:6872663</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AMELY</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>266</td>\n",
       "      <td>NM_001143.2</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_001143.2:c.-12-43T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BS1,BS2,BP7</td>\n",
       "      <td>AMELY:Allele frequency for Annotate-1kG Phase3...</td>\n",
       "      <td>None</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.00768212</td>\n",
       "      <td>3775</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24466</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12340004</td>\n",
       "      <td>rs377316799</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>200.4160</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=52;ADF=0,24;ADR=0,27;AD=0,51;SCR=0;VDB=0.17...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:230,154,0:51:0:0,24:0,27:0,51:0</td>\n",
       "      <td>52</td>\n",
       "      <td>0,51</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24467 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT      QUAL FILTER  \\\n",
       "0      chr1     930939    rs9988021   G   A  225.4170      .   \n",
       "1      chr1     944858    rs3748592   A   G  225.4170      .   \n",
       "2      chr1     946538   rs35471880   G   A  222.4010      .   \n",
       "3      chr1     948711    rs7522415   C   G   40.5469      .   \n",
       "4      chr1     957353            .   C   T  174.4050      .   \n",
       "...     ...        ...          ...  ..  ..       ...    ...   \n",
       "24462  chrX  155277884     rs559165   G   T  225.4170      .   \n",
       "24463  chrY    1304050            .   T   C   38.4150      .   \n",
       "24464  chrY    1418109            .   C   G   20.4535      .   \n",
       "24465  chrY    6872663   rs34551924   A   G  214.4170      .   \n",
       "24466  chrY   12340004  rs377316799   G   T  200.4160      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=71;ADF=0,54;ADR=0,11;AD=0,65;SCR=0;VDB=0.49...   \n",
       "1      DP=20;ADF=0,11;ADR=0,7;AD=0,18;SCR=0;VDB=0.165...   \n",
       "2      DP=54;ADF=9,11;ADR=13,12;AD=22,23;SCR=0;VDB=0....   \n",
       "3      DP=15;ADF=0,1;ADR=10,4;AD=10,5;SCR=0;VDB=0.240...   \n",
       "4      DP=34;ADF=0,1;ADR=16,14;AD=16,15;SCR=0;VDB=0.0...   \n",
       "...                                                  ...   \n",
       "24462  DP=27;ADF=0,16;ADR=0,8;AD=0,24;SCR=0;VDB=0.740...   \n",
       "24463  DP=83;ADF=0,9;ADR=0,65;AD=0,74;SCR=0;VDB=0.369...   \n",
       "24464  DP=95;ADF=0,25;ADR=0,59;AD=0,84;SCR=0;VDB=0.08...   \n",
       "24465  DP=20;ADF=0,1;ADR=0,18;AD=0,19;SCR=0;VDB=0.031...   \n",
       "24466  DP=52;ADF=0,24;ADR=0,27;AD=0,51;SCR=0;VDB=0.17...   \n",
       "\n",
       "                           FORMAT                                 SAMPLE  DP  \\\n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR    1/1:255,196,0:65:0:0,54:0,11:0,65:0  71   \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,54,0:18:0:0,11:0,7:0,18:0  20   \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR  0/1:255,0,255:45:1:9,11:13,12:22,23:0  54   \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR      0/1:74,0,148:15:0:0,1:10,4:10,5:0  15   \n",
       "4      GT:PL:DP:SP:ADF:ADR:AD:SCR   0/1:207,0,172:31:0:0,1:16,14:16,15:0  34   \n",
       "...                           ...                                    ...  ..   \n",
       "24462  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,72,0:24:0:0,16:0,8:0,24:0  27   \n",
       "24463  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:68,223,0:74:0:0,9:0,65:0,74:0  83   \n",
       "24464  GT:PL:DP:SP:ADF:ADR:AD:SCR     1/1:50,253,0:84:0:0,25:0,59:0,84:0  95   \n",
       "24465  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:244,57,0:19:0:0,1:0,18:0,19:0  20   \n",
       "24466  GT:PL:DP:SP:ADF:ADR:AD:SCR    1/1:230,154,0:51:0:0,24:0,27:0,51:0  52   \n",
       "\n",
       "          AD  RD  A_D       VAF  \\\n",
       "0       0,65   0   65  1.000000   \n",
       "1       0,18   0   18  1.000000   \n",
       "2      22,23  22   23  0.511111   \n",
       "3       10,5  10    5  0.333333   \n",
       "4      16,15  16   15  0.483871   \n",
       "...      ...  ..  ...       ...   \n",
       "24462   0,24   0   24  1.000000   \n",
       "24463   0,74   0   74  1.000000   \n",
       "24464   0,84   0   84  1.000000   \n",
       "24465   0,19   0   19  1.000000   \n",
       "24466   0,51   0   51  1.000000   \n",
       "\n",
       "                                                     CSQ  gnomADe_SAS_AF  \\\n",
       "0      A|intron_variant|MODIFIER|SAMD11|ENSG000001876...         0.00000   \n",
       "1      G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...         0.00000   \n",
       "2      A|missense_variant|MODERATE|NOC2L|ENSG00000188...         0.06653   \n",
       "3      G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...         0.00000   \n",
       "4      T|intron_variant|MODIFIER|NOC2L|ENSG0000018897...         0.00000   \n",
       "...                                                  ...             ...   \n",
       "24462  T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...         0.00000   \n",
       "24463  C|intron_variant|MODIFIER|CSF2RA|ENSG000001982...         0.00000   \n",
       "24464  G|synonymous_variant|LOW|ASMTL|ENSG00000169093...         0.00000   \n",
       "24465  G|upstream_gene_variant|MODIFIER|AMELY|ENSG000...         0.00000   \n",
       "24466  T|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "\n",
       "       gnomADe_AF      Chr:Pos Ref/Alt Gene Name     Gene Inheritance  \\\n",
       "0         0.00000     1:930939     G/A    SAMD11  Default (Recessive)   \n",
       "1         0.00000     1:944858     A/G     NOC2L  Default (Recessive)   \n",
       "2         0.04557     1:946538     G/A     NOC2L  Default (Recessive)   \n",
       "3         0.00000          NaN     NaN       NaN                  NaN   \n",
       "4         0.00000     1:957353     C/T     NOC2L  Default (Recessive)   \n",
       "...           ...          ...     ...       ...                  ...   \n",
       "24462     0.00000  X:155277884     G/T     CLIC2            Recessive   \n",
       "24463     0.00000          NaN     NaN       NaN                  NaN   \n",
       "24464     0.00000          NaN     NaN       NaN                  NaN   \n",
       "24465     0.00000    Y:6872663     A/G     AMELY  Default (Recessive)   \n",
       "24466     0.00000          NaN     NaN       NaN                  NaN   \n",
       "\n",
       "      Entrez Gene ID Transcript Name Transcript Strand  \\\n",
       "0             148398  NM_001385641.1                 +   \n",
       "1              26155     NM_015658.4                 -   \n",
       "2              26155     NM_015658.4                 -   \n",
       "3                NaN             NaN               NaN   \n",
       "4              26155     NM_015658.4                 -   \n",
       "...              ...             ...               ...   \n",
       "24462           1193     NM_001289.6                 -   \n",
       "24463            NaN             NaN               NaN   \n",
       "24464            NaN             NaN               NaN   \n",
       "24465            266     NM_001143.2                 -   \n",
       "24466            NaN             NaN               NaN   \n",
       "\n",
       "                         HGVS cDot                HGVS pDot  \\\n",
       "0      NM_001385641.1:c.792-100G>A                      NaN   \n",
       "1         NM_015658.4:c.2144-58T>C                      NaN   \n",
       "2            NM_015658.4:c.1667C>T  NP_056473.3:p.Ser556Leu   \n",
       "3                              NaN                      NaN   \n",
       "4          NM_015658.4:c.180-80G>A                      NaN   \n",
       "...                            ...                      ...   \n",
       "24462         NM_001289.6:c.*19C>A                      NaN   \n",
       "24463                          NaN                      NaN   \n",
       "24464                          NaN                      NaN   \n",
       "24465      NM_001143.2:c.-12-43T>C                      NaN   \n",
       "24466                          NaN                      NaN   \n",
       "\n",
       "         Sequence Ontology            Other Transcript Effects  \\\n",
       "0           intron_variant  Other Transcripts Have Same Effect   \n",
       "1           intron_variant              Single Transcript Gene   \n",
       "2         missense_variant              Single Transcript Gene   \n",
       "3                      NaN                                 NaN   \n",
       "4           intron_variant              Single Transcript Gene   \n",
       "...                    ...                                 ...   \n",
       "24462  3_prime_UTR_variant              Single Transcript Gene   \n",
       "24463                  NaN                                 NaN   \n",
       "24464                  NaN                                 NaN   \n",
       "24465       intron_variant  Other Transcripts Have Same Effect   \n",
       "24466                  NaN                                 NaN   \n",
       "\n",
       "      ACMG Classification Criteria  \\\n",
       "0                      BA1,BS2,BP7   \n",
       "1                      BA1,BS2,BP7   \n",
       "2                      BA1,BS2,PP3   \n",
       "3                              NaN   \n",
       "4                          PM2,BP7   \n",
       "...                            ...   \n",
       "24462              BA1,BS2,BP4,BP7   \n",
       "24463                          NaN   \n",
       "24464                          NaN   \n",
       "24465                  BS1,BS2,BP7   \n",
       "24466                          NaN   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0      SAMD11:Allele frequency is above 0.01 recessiv...           None   \n",
       "1      NOC2L:Allele frequency is above 0.01 recessive...           None   \n",
       "2      NOC2L:Allele frequency is above 0.01 recessive...           None   \n",
       "3                                                    NaN           None   \n",
       "4      NOC2L:Variant is missing from all sub populati...           None   \n",
       "...                                                  ...            ...   \n",
       "24462  CLIC2:Allele frequency is above 0.01 recessive...           None   \n",
       "24463                                                NaN           None   \n",
       "24464                                                NaN           None   \n",
       "24465  AMELY:Allele frequency for Annotate-1kG Phase3...           None   \n",
       "24466                                                NaN           None   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                  Benign                           NaN   \n",
       "1                  Benign                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                     NaN                           NaN   \n",
       "4         VUS/Conflicting                           NaN   \n",
       "...                   ...                           ...   \n",
       "24462              Benign                           NaN   \n",
       "24463                 NaN                           NaN   \n",
       "24464                 NaN                           NaN   \n",
       "24465              Benign                           NaN   \n",
       "24466                 NaN                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "24462                     NaN                      NaN   \n",
       "24463                     NaN                      NaN   \n",
       "24464                     NaN                      NaN   \n",
       "24465                     NaN                      NaN   \n",
       "24466                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name     Max MAF Max Allele Number  \\\n",
       "0                      Annotated 1kG All    0.953474              5008   \n",
       "1                      Annotated 1kG All    0.920927              5008   \n",
       "2           Annotated gnomAD South Asian    0.066289             30714   \n",
       "3                                    NaN         NaN               NaN   \n",
       "4                                    NaN         NaN               NaN   \n",
       "...                                  ...         ...               ...   \n",
       "24462  Annotated gnomAD Ashkenazi Jewish    0.772765              7248   \n",
       "24463                                NaN         NaN               NaN   \n",
       "24464                                NaN         NaN               NaN   \n",
       "24465                  Annotated 1kG All  0.00768212              3775   \n",
       "24466                                NaN         NaN               NaN   \n",
       "\n",
       "      Max Allele Count  \n",
       "0                 4775  \n",
       "1                 4612  \n",
       "2                 2036  \n",
       "3                  NaN  \n",
       "4                  NaN  \n",
       "...                ...  \n",
       "24462             5601  \n",
       "24463              NaN  \n",
       "24464              NaN  \n",
       "24465               29  \n",
       "24466              NaN  \n",
       "\n",
       "[24467 rows x 40 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_data = pd.merge(filtered_data, truth_1, on = ['CHROM', 'POS', 'REF', 'ALT'])\n",
    "filtered_data = filtered_data.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "# Merge the VCF data with the main data (df)\n",
    "merged_data = pd.merge(filtered_data, df, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged_data = merged_data.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "# To fill NaN values in the 'Classification' column with 'NA', you need to assign the result back to the column.\n",
    "merged_data['Classification'] = merged_data['Classification'].fillna('NA')\n",
    "# List of your specified priorities\n",
    "priorities = ['Pathogenic', 'Likely Pathogenic', 'VUS/Weak Pathogenic', 'VUS/Conflicting', 'VUS', 'VUS/Weak Benign', 'Likely Benign', 'Benign']\n",
    "# Function to select the highest ranked value from Original_Column\n",
    "def select_highest_ranked(row):\n",
    "    values = row['Classification'].split(',')\n",
    "    for rank in priorities:\n",
    "        if rank in values:\n",
    "            return rank\n",
    "        return None  # Return None if no rank found\n",
    "# Apply the function to create the final selected rank column\n",
    "merged_data['Classification'] = merged_data.apply(select_highest_ranked, axis=1)\n",
    "merged_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "d57e5e20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CHROM, POS, rsID, REF, ALT, QUAL, FILTER, INFO, FORMAT, SAMPLE, DP, AD, RD, A_D, VAF, CSQ, gnomADe_SAS_AF, gnomADe_AF, Chr:Pos, Ref/Alt, Gene Name, Gene Inheritance, Entrez Gene ID, Transcript Name, Transcript Strand, HGVS cDot, HGVS pDot, Sequence Ontology, Other Transcript Effects, ACMG Classification Criteria, ACMG Classification Criteria Description, Classification, Auto Classification, Previous Classification Count, Previous Classification, Last Classification Date, Max Sub Population Freq Group Name, Max MAF, Max Allele Number, Max Allele Count]\n",
       "Index: []"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = merged_data[merged_data['Classification'] == 'Likely Pathogenic']\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe32f6fa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "daffe558",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=71;ADF=0,54;ADR=0,11;AD=0,65;SCR=0;VDB=0.49...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,196,0:65:0:0,54:0,11:0,65:0</td>\n",
       "      <td>0,65</td>\n",
       "      <td>0</td>\n",
       "      <td>65</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>71</td>\n",
       "      <td>A|intron_variant|MODIFIER|SAMD11|ENSG000001876...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=20;ADF=0,11;ADR=0,7;AD=0,18;SCR=0;VDB=0.165...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,54,0:18:0:0,11:0,7:0,18:0</td>\n",
       "      <td>0,18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>20</td>\n",
       "      <td>G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946538</td>\n",
       "      <td>rs35471880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>222.4010</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=54;ADF=9,11;ADR=13,12;AD=22,23;SCR=0;VDB=0....</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:255,0,255:45:1:9,11:13,12:22,23:0</td>\n",
       "      <td>22,23</td>\n",
       "      <td>22</td>\n",
       "      <td>23</td>\n",
       "      <td>0.511111</td>\n",
       "      <td>54</td>\n",
       "      <td>A|missense_variant|MODERATE|NOC2L|ENSG00000188...</td>\n",
       "      <td>0.06653</td>\n",
       "      <td>0.04557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>948711</td>\n",
       "      <td>rs7522415</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>40.5469</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=15;ADF=0,1;ADR=10,4;AD=10,5;SCR=0;VDB=0.240...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:74,0,148:15:0:0,1:10,4:10,5:0</td>\n",
       "      <td>10,5</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>15</td>\n",
       "      <td>G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>957353</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>174.4050</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=34;ADF=0,1;ADR=16,14;AD=16,15;SCR=0;VDB=0.0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>0/1:207,0,172:31:0:0,1:16,14:16,15:0</td>\n",
       "      <td>16,15</td>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "      <td>0.483871</td>\n",
       "      <td>34</td>\n",
       "      <td>T|intron_variant|MODIFIER|NOC2L|ENSG0000018897...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26706</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155277884</td>\n",
       "      <td>rs559165</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=27;ADF=0,16;ADR=0,8;AD=0,24;SCR=0;VDB=0.740...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,72,0:24:0:0,16:0,8:0,24:0</td>\n",
       "      <td>0,24</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>27</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26707</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1304050</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>38.4150</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=83;ADF=0,9;ADR=0,65;AD=0,74;SCR=0;VDB=0.369...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:68,223,0:74:0:0,9:0,65:0,74:0</td>\n",
       "      <td>0,74</td>\n",
       "      <td>0</td>\n",
       "      <td>74</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>83</td>\n",
       "      <td>C|intron_variant|MODIFIER|CSF2RA|ENSG000001982...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26708</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1418109</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>20.4535</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=95;ADF=0,25;ADR=0,59;AD=0,84;SCR=0;VDB=0.08...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:50,253,0:84:0:0,25:0,59:0,84:0</td>\n",
       "      <td>0,84</td>\n",
       "      <td>0</td>\n",
       "      <td>84</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>95</td>\n",
       "      <td>G|synonymous_variant|LOW|ASMTL|ENSG00000169093...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26709</th>\n",
       "      <td>chrY</td>\n",
       "      <td>6872663</td>\n",
       "      <td>rs34551924</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>214.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=20;ADF=0,1;ADR=0,18;AD=0,19;SCR=0;VDB=0.031...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:244,57,0:19:0:0,1:0,18:0,19:0</td>\n",
       "      <td>0,19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>20</td>\n",
       "      <td>G|upstream_gene_variant|MODIFIER|AMELY|ENSG000...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26710</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12340004</td>\n",
       "      <td>rs377316799</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>200.4160</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=52;ADF=0,24;ADR=0,27;AD=0,51;SCR=0;VDB=0.17...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:230,154,0:51:0:0,24:0,27:0,51:0</td>\n",
       "      <td>0,51</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>52</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26668 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT      QUAL FILTER  \\\n",
       "0      chr1     930939    rs9988021   G   A  225.4170      .   \n",
       "1      chr1     944858    rs3748592   A   G  225.4170      .   \n",
       "2      chr1     946538   rs35471880   G   A  222.4010      .   \n",
       "3      chr1     948711    rs7522415   C   G   40.5469      .   \n",
       "4      chr1     957353            .   C   T  174.4050      .   \n",
       "...     ...        ...          ...  ..  ..       ...    ...   \n",
       "26706  chrX  155277884     rs559165   G   T  225.4170      .   \n",
       "26707  chrY    1304050            .   T   C   38.4150      .   \n",
       "26708  chrY    1418109            .   C   G   20.4535      .   \n",
       "26709  chrY    6872663   rs34551924   A   G  214.4170      .   \n",
       "26710  chrY   12340004  rs377316799   G   T  200.4160      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=71;ADF=0,54;ADR=0,11;AD=0,65;SCR=0;VDB=0.49...   \n",
       "1      DP=20;ADF=0,11;ADR=0,7;AD=0,18;SCR=0;VDB=0.165...   \n",
       "2      DP=54;ADF=9,11;ADR=13,12;AD=22,23;SCR=0;VDB=0....   \n",
       "3      DP=15;ADF=0,1;ADR=10,4;AD=10,5;SCR=0;VDB=0.240...   \n",
       "4      DP=34;ADF=0,1;ADR=16,14;AD=16,15;SCR=0;VDB=0.0...   \n",
       "...                                                  ...   \n",
       "26706  DP=27;ADF=0,16;ADR=0,8;AD=0,24;SCR=0;VDB=0.740...   \n",
       "26707  DP=83;ADF=0,9;ADR=0,65;AD=0,74;SCR=0;VDB=0.369...   \n",
       "26708  DP=95;ADF=0,25;ADR=0,59;AD=0,84;SCR=0;VDB=0.08...   \n",
       "26709  DP=20;ADF=0,1;ADR=0,18;AD=0,19;SCR=0;VDB=0.031...   \n",
       "26710  DP=52;ADF=0,24;ADR=0,27;AD=0,51;SCR=0;VDB=0.17...   \n",
       "\n",
       "                           FORMAT                                 SAMPLE  \\\n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR    1/1:255,196,0:65:0:0,54:0,11:0,65:0   \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,54,0:18:0:0,11:0,7:0,18:0   \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR  0/1:255,0,255:45:1:9,11:13,12:22,23:0   \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR      0/1:74,0,148:15:0:0,1:10,4:10,5:0   \n",
       "4      GT:PL:DP:SP:ADF:ADR:AD:SCR   0/1:207,0,172:31:0:0,1:16,14:16,15:0   \n",
       "...                           ...                                    ...   \n",
       "26706  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,72,0:24:0:0,16:0,8:0,24:0   \n",
       "26707  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:68,223,0:74:0:0,9:0,65:0,74:0   \n",
       "26708  GT:PL:DP:SP:ADF:ADR:AD:SCR     1/1:50,253,0:84:0:0,25:0,59:0,84:0   \n",
       "26709  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:244,57,0:19:0:0,1:0,18:0,19:0   \n",
       "26710  GT:PL:DP:SP:ADF:ADR:AD:SCR    1/1:230,154,0:51:0:0,24:0,27:0,51:0   \n",
       "\n",
       "          AD  RD  A_D       VAF  DP  \\\n",
       "0       0,65   0   65  1.000000  71   \n",
       "1       0,18   0   18  1.000000  20   \n",
       "2      22,23  22   23  0.511111  54   \n",
       "3       10,5  10    5  0.333333  15   \n",
       "4      16,15  16   15  0.483871  34   \n",
       "...      ...  ..  ...       ...  ..   \n",
       "26706   0,24   0   24  1.000000  27   \n",
       "26707   0,74   0   74  1.000000  83   \n",
       "26708   0,84   0   84  1.000000  95   \n",
       "26709   0,19   0   19  1.000000  20   \n",
       "26710   0,51   0   51  1.000000  52   \n",
       "\n",
       "                                                     CSQ  gnomADe_SAS_AF  \\\n",
       "0      A|intron_variant|MODIFIER|SAMD11|ENSG000001876...         0.00000   \n",
       "1      G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...         0.00000   \n",
       "2      A|missense_variant|MODERATE|NOC2L|ENSG00000188...         0.06653   \n",
       "3      G|intron_variant|MODIFIER|NOC2L|ENSG0000018897...         0.00000   \n",
       "4      T|intron_variant|MODIFIER|NOC2L|ENSG0000018897...         0.00000   \n",
       "...                                                  ...             ...   \n",
       "26706  T|downstream_gene_variant|MODIFIER|CLIC2|ENSG0...         0.00000   \n",
       "26707  C|intron_variant|MODIFIER|CSF2RA|ENSG000001982...         0.00000   \n",
       "26708  G|synonymous_variant|LOW|ASMTL|ENSG00000169093...         0.00000   \n",
       "26709  G|upstream_gene_variant|MODIFIER|AMELY|ENSG000...         0.00000   \n",
       "26710  T|intron_variant&non_coding_transcript_variant...         0.00000   \n",
       "\n",
       "       gnomADe_AF  \n",
       "0         0.00000  \n",
       "1         0.00000  \n",
       "2         0.04557  \n",
       "3         0.00000  \n",
       "4         0.00000  \n",
       "...           ...  \n",
       "26706     0.00000  \n",
       "26707     0.00000  \n",
       "26708     0.00000  \n",
       "26709     0.00000  \n",
       "26710     0.00000  \n",
       "\n",
       "[26668 rows x 18 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Compare with the truth set\n",
    "truth_set['CHROM'] = truth_set['CHROM'].astype(str)\n",
    "truth_set['POS'] = truth_set['POS'].astype(int)\n",
    "truth_set['REF'] = truth_set['REF'].astype(str)\n",
    "truth_set['ALT'] = truth_set['ALT'].astype(str)\n",
    "truth_1 = truth_set.copy()\n",
    "truth_1['ALT'] = truth_1['ALT'].str.split(',')\n",
    "truth_1 = truth_1.explode('ALT') \n",
    "intersection_ab = pd.merge(filter_var_DP, truth_1, on=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "intersection_ab = intersection_ab.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "intersection_ab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4edabc3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of your specified priorities\n",
    "priorities = ['Pathogenic', 'Likely Pathogenic', 'VUS/Weak Pathogenic', 'VUS/Conflicting', 'VUS', 'VUS/Weak Benign', 'Likely Benign', 'Benign']\n",
    "\n",
    "# Function to select the highest ranked value from Original_Column\n",
    "def select_highest_ranked(row):\n",
    "    values = row['Classification'].split(',')\n",
    "    for rank in priorities:\n",
    "        if rank in values:\n",
    "            return rank\n",
    "    return None  # Return None if no rank found\n",
    "\n",
    "# Apply the function to create the final selected rank column\n",
    "intersection_ab['Classification'] = intersection_ab.apply(select_highest_ranked, axis=1)\n",
    "intersection_ab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9e80215",
   "metadata": {},
   "outputs": [],
   "source": [
    "25581"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e060436",
   "metadata": {},
   "source": [
    "# VARSCAN2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "032d7a47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=370;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.2468,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:371:370:192:178:48.11%:1.106E-66:38:37...</td>\n",
       "      <td>370</td>\n",
       "      <td>192</td>\n",
       "      <td>178</td>\n",
       "      <td>0.481081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:61:60:0:60:100%:1.035E-35:0:36:0:0:26:34</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "      <td>60</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=144;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:144:144:27:117:81.25%:8.7897E-55:36:36...</td>\n",
       "      <td>144</td>\n",
       "      <td>27</td>\n",
       "      <td>117</td>\n",
       "      <td>0.812500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=148;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:148:148:29:119:80.41%:2.627E-55:36:36:...</td>\n",
       "      <td>148</td>\n",
       "      <td>29</td>\n",
       "      <td>119</td>\n",
       "      <td>0.804054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827221</td>\n",
       "      <td>rs3131949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=148;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:148:148:31:117:79.05%:9.0003E-54:36:36...</td>\n",
       "      <td>148</td>\n",
       "      <td>31</td>\n",
       "      <td>117</td>\n",
       "      <td>0.790541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66857</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57084531</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:105:65:64:35:29:45.31%:2.9719E-11:37:36:30...</td>\n",
       "      <td>64</td>\n",
       "      <td>35</td>\n",
       "      <td>29</td>\n",
       "      <td>0.453125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66858</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57128393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=238;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:238:238:120:118:49.58%:9.1094E-45:39:3...</td>\n",
       "      <td>238</td>\n",
       "      <td>120</td>\n",
       "      <td>118</td>\n",
       "      <td>0.495798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66859</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184331</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=139;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_var...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:139:139:63:76:54.68%:7.2335E-30:37:38:...</td>\n",
       "      <td>139</td>\n",
       "      <td>63</td>\n",
       "      <td>76</td>\n",
       "      <td>0.546763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66860</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184462</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=48;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:89:49:48:24:24:50%:1.2355E-9:38:37:0:24:1:23</td>\n",
       "      <td>48</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66861</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57189953</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=49;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:93:49:49:24:25:51.02%:4.6489E-10:37:40:14:...</td>\n",
       "      <td>49</td>\n",
       "      <td>24</td>\n",
       "      <td>25</td>\n",
       "      <td>0.510204</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65773 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1    817514  rs3131971   T   C    .   PASS   \n",
       "1      chr1    826893  rs3115849   G   A    .   PASS   \n",
       "2      chr1    827209  rs3115848   G   C    .   PASS   \n",
       "3      chr1    827212  rs3131950   C   G    .   PASS   \n",
       "4      chr1    827221  rs3131949   T   C    .   PASS   \n",
       "...     ...       ...        ...  ..  ..  ...    ...   \n",
       "66857  chrY  57084531          .   A   G    .   PASS   \n",
       "66858  chrY  57128393          .   G   C    .   PASS   \n",
       "66859  chrY  57184331          .   C   G    .   PASS   \n",
       "66860  chrY  57184462          .   T   G    .   PASS   \n",
       "66861  chrY  57189953          .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=370;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.2468,0...   \n",
       "1      ADP=60;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....   \n",
       "2      ADP=144;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "3      ADP=148;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "4      ADP=148;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,....   \n",
       "...                                                  ...   \n",
       "66857  ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "66858  ADP=238;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...   \n",
       "66859  ADP=139;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_var...   \n",
       "66860  ADP=48;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "66861  ADP=49;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "66857  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "66858  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "66859  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "66860  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "66861  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   DP   RD   AD  \\\n",
       "0      0/1:255:371:370:192:178:48.11%:1.106E-66:38:37...  370  192  178   \n",
       "1       1/1:255:61:60:0:60:100%:1.035E-35:0:36:0:0:26:34   60    0   60   \n",
       "2      1/1:255:144:144:27:117:81.25%:8.7897E-55:36:36...  144   27  117   \n",
       "3      1/1:255:148:148:29:119:80.41%:2.627E-55:36:36:...  148   29  119   \n",
       "4      1/1:255:148:148:31:117:79.05%:9.0003E-54:36:36...  148   31  117   \n",
       "...                                                  ...  ...  ...  ...   \n",
       "66857  0/1:105:65:64:35:29:45.31%:2.9719E-11:37:36:30...   64   35   29   \n",
       "66858  0/1:255:238:238:120:118:49.58%:9.1094E-45:39:3...  238  120  118   \n",
       "66859  0/1:255:139:139:63:76:54.68%:7.2335E-30:37:38:...  139   63   76   \n",
       "66860   0/1:89:49:48:24:24:50%:1.2355E-9:38:37:0:24:1:23   48   24   24   \n",
       "66861  0/1:93:49:49:24:25:51.02%:4.6489E-10:37:40:14:...   49   24   25   \n",
       "\n",
       "            VAF  \n",
       "0      0.481081  \n",
       "1      1.000000  \n",
       "2      0.812500  \n",
       "3      0.804054  \n",
       "4      0.790541  \n",
       "...         ...  \n",
       "66857  0.453125  \n",
       "66858  0.495798  \n",
       "66859  0.546763  \n",
       "66860  0.500000  \n",
       "66861  0.510204  \n",
       "\n",
       "[65773 rows x 14 columns]"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652707_VARSCAN2.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "vcf['RD'] = vcf['SAMPLE'].str.split(':').str[4].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[5].fillna('0').astype(int)\n",
    "vcf['VAF'] = vcf['AD'] / (vcf['RD'] + vcf['AD'])\n",
    "filter_var = vcf[vcf['VAF'] >= 0.01]\n",
    "filter_var_DP = filter_var[filter_var['DP'] >= 10]\n",
    "#filter_var_DP = vcf.copy()\n",
    "filter_var_DP"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3db8022",
   "metadata": {},
   "source": [
    "# MUTECT2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "8ad11c81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,31;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,31:0.970:31:0,13:0,9:0,0,0,31</td>\n",
       "      <td>31</td>\n",
       "      <td>0,31</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,14...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,62:0.984:62:0,26:0,35:0,0,48,14</td>\n",
       "      <td>62</td>\n",
       "      <td>0,62</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>935954</td>\n",
       "      <td>rs4072383</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|2,14;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,16:0.941:16:0,2:0,13:0,0,2,14</td>\n",
       "      <td>16</td>\n",
       "      <td>0,16</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|17,4;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,21:0.955:21:0,10:0,11:0,0,17,4</td>\n",
       "      <td>21</td>\n",
       "      <td>0,21</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|9,6;C...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,15:0.941:15:0,6:0,9:0,0,9,6</td>\n",
       "      <td>15</td>\n",
       "      <td>0,15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81087</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12738369</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;map_qual;weak_evidence</td>\n",
       "      <td>AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB</td>\n",
       "      <td>0|1:36,1:0.059:37:13,1:21,0:0|1:12738359_G_T:1...</td>\n",
       "      <td>37</td>\n",
       "      <td>36,1</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>0.027027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81088</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12738371</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;map_qual;weak_evidence</td>\n",
       "      <td>AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB</td>\n",
       "      <td>0|1:35,1:0.059:36:13,1:20,0:0|1:12738359_G_T:1...</td>\n",
       "      <td>36</td>\n",
       "      <td>35,1</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>0.027778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81089</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12914512</td>\n",
       "      <td>rs2032624</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|21,7;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,28:0.967:28:0,13:0,15:0,0,21,7</td>\n",
       "      <td>28</td>\n",
       "      <td>0,28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81090</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13250864</td>\n",
       "      <td>rs376039285</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|27,0;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,27:0.966:27:0,12:0,15:0,0,27,0</td>\n",
       "      <td>27</td>\n",
       "      <td>0,27</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81091</th>\n",
       "      <td>chrY</td>\n",
       "      <td>19732200</td>\n",
       "      <td>rs750047037</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,10;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,10:0.916:10:0,6:0,3:0,0,0,10</td>\n",
       "      <td>10</td>\n",
       "      <td>0,10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73294 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL  \\\n",
       "0      chr1    817514    rs3131971   T   C    .   \n",
       "5      chr1    930939    rs9988021   G   A    .   \n",
       "6      chr1    935954    rs4072383   G   T    .   \n",
       "7      chr1    941119    rs4372192   A   G    .   \n",
       "8      chr1    944858    rs3748592   A   G    .   \n",
       "...     ...       ...          ...  ..  ..  ...   \n",
       "81087  chrY  12738369            .   G   C    .   \n",
       "81088  chrY  12738371            .   T   G    .   \n",
       "81089  chrY  12914512    rs2032624   C   A    .   \n",
       "81090  chrY  13250864  rs376039285   C   T    .   \n",
       "81091  chrY  19732200  rs750047037   A   C    .   \n",
       "\n",
       "                                                  FILTER  \\\n",
       "0                                                   PASS   \n",
       "5                                                   PASS   \n",
       "6                                                   PASS   \n",
       "7                                                   PASS   \n",
       "8                                                   PASS   \n",
       "...                                                  ...   \n",
       "81087  clustered_events;haplotype;map_qual;weak_evidence   \n",
       "81088  clustered_events;haplotype;map_qual;weak_evidence   \n",
       "81089                                               PASS   \n",
       "81090                                               PASS   \n",
       "81091                                               PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,31;...   \n",
       "5      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,14...   \n",
       "6      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|2,14;...   \n",
       "7      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|17,4;...   \n",
       "8      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|9,6;C...   \n",
       "...                                                  ...   \n",
       "81087  AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...   \n",
       "81088  AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...   \n",
       "81089  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|21,7;...   \n",
       "81090  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|27,0;...   \n",
       "81091  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,10;...   \n",
       "\n",
       "                                    FORMAT  \\\n",
       "0                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "5                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "6                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "7                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "8                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "...                                    ...   \n",
       "81087  GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB   \n",
       "81088  GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB   \n",
       "81089             GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "81090             GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "81091             GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "\n",
       "                                                  SAMPLE  DP    AD  RD  A_D  \\\n",
       "0                    0/1:0,31:0.970:31:0,13:0,9:0,0,0,31  31  0,31   0   31   \n",
       "5                  0/1:0,62:0.984:62:0,26:0,35:0,0,48,14  62  0,62   0   62   \n",
       "6                    0/1:0,16:0.941:16:0,2:0,13:0,0,2,14  16  0,16   0   16   \n",
       "7                   0/1:0,21:0.955:21:0,10:0,11:0,0,17,4  21  0,21   0   21   \n",
       "8                      0/1:0,15:0.941:15:0,6:0,9:0,0,9,6  15  0,15   0   15   \n",
       "...                                                  ...  ..   ...  ..  ...   \n",
       "81087  0|1:36,1:0.059:37:13,1:21,0:0|1:12738359_G_T:1...  37  36,1  36    1   \n",
       "81088  0|1:35,1:0.059:36:13,1:20,0:0|1:12738359_G_T:1...  36  35,1  35    1   \n",
       "81089               0/1:0,28:0.967:28:0,13:0,15:0,0,21,7  28  0,28   0   28   \n",
       "81090               0/1:0,27:0.966:27:0,12:0,15:0,0,27,0  27  0,27   0   27   \n",
       "81091                 0/1:0,10:0.916:10:0,6:0,3:0,0,0,10  10  0,10   0   10   \n",
       "\n",
       "            VAF  \n",
       "0      1.000000  \n",
       "5      1.000000  \n",
       "6      1.000000  \n",
       "7      1.000000  \n",
       "8      1.000000  \n",
       "...         ...  \n",
       "81087  0.027027  \n",
       "81088  0.027778  \n",
       "81089  1.000000  \n",
       "81090  1.000000  \n",
       "81091  1.000000  \n",
       "\n",
       "[73294 rows x 15 columns]"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652707_MUTECT2.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "filter_var = vcf[vcf['VAF'] >= 0.001]\n",
    "filter_var_DP = filter_var[filter_var['DP'] >= 10]\n",
    "#filter_var_DP = vcf.copy()\n",
    "filter_var_DP"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3dc6ce8a",
   "metadata": {},
   "source": [
    "# HAPLOTYPECALLER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "2121394b",
   "metadata": {},
   "outputs": [
    {
     "ename": "ParserError",
     "evalue": "Error tokenizing data. C error: out of memory",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mParserError\u001b[0m                               Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[248], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mpd\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m vcf \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mr\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mC:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652707_HAPLOTYPECALLER.vcf\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcomment\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m#\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;130;43;01m\\t\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheader\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlow_memory\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mlatin-1\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m      3\u001b[0m vcf\u001b[38;5;241m.\u001b[39mcolumns \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCHROM\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPOS\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mrsID\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mREF\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mALT\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mQUAL\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mFILTER\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mINFO\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mFORMAT\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSAMPLE\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m      4\u001b[0m vcf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mDP\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m vcf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSAMPLE\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m)\u001b[38;5;241m.\u001b[39mstr[\u001b[38;5;241m2\u001b[39m]\u001b[38;5;241m.\u001b[39mfillna(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m0\u001b[39m\u001b[38;5;124m'\u001b[39m)\u001b[38;5;241m.\u001b[39mastype(\u001b[38;5;28mint\u001b[39m)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\util\\_decorators.py:211\u001b[0m, in \u001b[0;36mdeprecate_kwarg.<locals>._deprecate_kwarg.<locals>.wrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    209\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    210\u001b[0m         kwargs[new_arg_name] \u001b[38;5;241m=\u001b[39m new_arg_value\n\u001b[1;32m--> 211\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m func(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\util\\_decorators.py:331\u001b[0m, in \u001b[0;36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    325\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(args) \u001b[38;5;241m>\u001b[39m num_allow_args:\n\u001b[0;32m    326\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[0;32m    327\u001b[0m         msg\u001b[38;5;241m.\u001b[39mformat(arguments\u001b[38;5;241m=\u001b[39m_format_argument_list(allow_args)),\n\u001b[0;32m    328\u001b[0m         \u001b[38;5;167;01mFutureWarning\u001b[39;00m,\n\u001b[0;32m    329\u001b[0m         stacklevel\u001b[38;5;241m=\u001b[39mfind_stack_level(),\n\u001b[0;32m    330\u001b[0m     )\n\u001b[1;32m--> 331\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m func(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:950\u001b[0m, in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[0;32m    935\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[0;32m    936\u001b[0m     dialect,\n\u001b[0;32m    937\u001b[0m     delimiter,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    946\u001b[0m     defaults\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdelimiter\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m\"\u001b[39m},\n\u001b[0;32m    947\u001b[0m )\n\u001b[0;32m    948\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[1;32m--> 950\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:611\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    608\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n\u001b[0;32m    610\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m parser:\n\u001b[1;32m--> 611\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mparser\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnrows\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1778\u001b[0m, in \u001b[0;36mTextFileReader.read\u001b[1;34m(self, nrows)\u001b[0m\n\u001b[0;32m   1771\u001b[0m nrows \u001b[38;5;241m=\u001b[39m validate_integer(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnrows\u001b[39m\u001b[38;5;124m\"\u001b[39m, nrows)\n\u001b[0;32m   1772\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m   1773\u001b[0m     \u001b[38;5;66;03m# error: \"ParserBase\" has no attribute \"read\"\u001b[39;00m\n\u001b[0;32m   1774\u001b[0m     (\n\u001b[0;32m   1775\u001b[0m         index,\n\u001b[0;32m   1776\u001b[0m         columns,\n\u001b[0;32m   1777\u001b[0m         col_dict,\n\u001b[1;32m-> 1778\u001b[0m     ) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43m  \u001b[49m\u001b[38;5;66;43;03m# type: ignore[attr-defined]\u001b[39;49;00m\n\u001b[0;32m   1779\u001b[0m \u001b[43m        \u001b[49m\u001b[43mnrows\u001b[49m\n\u001b[0;32m   1780\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1781\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m:\n\u001b[0;32m   1782\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mclose()\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\io\\parsers\\c_parser_wrapper.py:235\u001b[0m, in \u001b[0;36mCParserWrapper.read\u001b[1;34m(self, nrows)\u001b[0m\n\u001b[0;32m    232\u001b[0m         data \u001b[38;5;241m=\u001b[39m _concatenate_chunks(chunks)\n\u001b[0;32m    234\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 235\u001b[0m         data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_reader\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnrows\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    236\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mStopIteration\u001b[39;00m:\n\u001b[0;32m    237\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_first_chunk:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:790\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader.read\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:883\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader._read_rows\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:1973\u001b[0m, in \u001b[0;36mpandas._libs.parsers.raise_parser_error\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mParserError\u001b[0m: Error tokenizing data. C error: out of memory"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652707_HAPLOTYPECALLER.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "#filter_var = vcf[vcf['VAF'] >= 0.001]\n",
    "#filter_var_DP = filter_var[filter_var['DP'] >= 10]\n",
    "filter_var_DP = vcf.copy()\n",
    "filter_var_DP"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f683121",
   "metadata": {},
   "source": [
    "# DEEPVARIANT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "2a2be794",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>32.1</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=G|missense_variant|MODERATE|OR4F5|ENSG...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:32:222:0,221:0.995495:32,42,0</td>\n",
       "      <td>222</td>\n",
       "      <td>0,221</td>\n",
       "      <td>0</td>\n",
       "      <td>221</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.1</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CAF=0.2468,0.7532;COMMON=1;CSQ=C|non_codin...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:40:61:1,60:0.983607:47,40,0</td>\n",
       "      <td>61</td>\n",
       "      <td>1,60</td>\n",
       "      <td>1</td>\n",
       "      <td>60</td>\n",
       "      <td>0.983607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>31.8</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CAF=0.2682,0.7318;COMMON=1;CSQ=A|upstream_...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:31:90:0,90:1:31,37,0</td>\n",
       "      <td>90</td>\n",
       "      <td>0,90</td>\n",
       "      <td>0</td>\n",
       "      <td>90</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>47.6</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CAF=0.2484,0.7516;COMMON=1;CSQ=C|upstream_...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:47:129:4,125:0.968992:47,55,0</td>\n",
       "      <td>129</td>\n",
       "      <td>4,125</td>\n",
       "      <td>4</td>\n",
       "      <td>125</td>\n",
       "      <td>0.968992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>51.7</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CAF=0.2484,0.7516,.;COMMON=1;CSQ=G|upstrea...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:48:126:4,122:0.968254:51,50,0</td>\n",
       "      <td>126</td>\n",
       "      <td>4,122</td>\n",
       "      <td>4</td>\n",
       "      <td>122</td>\n",
       "      <td>0.968254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92127</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11153722</td>\n",
       "      <td>rs77624695</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>20.1</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=T|downstream_gene_variant|MODIFIER||EN...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:20:11:0,11:1:20,38,0</td>\n",
       "      <td>11</td>\n",
       "      <td>0,11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92134</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986256</td>\n",
       "      <td>rs9650860</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>31.4</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=C|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:31:13:0,13:1:31,40,0</td>\n",
       "      <td>13</td>\n",
       "      <td>0,13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92135</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>rs9650861</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>31.2</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=C|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:28:63:0,63:1:31,31,0</td>\n",
       "      <td>63</td>\n",
       "      <td>0,63</td>\n",
       "      <td>0</td>\n",
       "      <td>63</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92136</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>rs879016544</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>33.8</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=C|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:33:52:0,52:1:33,38,0</td>\n",
       "      <td>52</td>\n",
       "      <td>0,52</td>\n",
       "      <td>0</td>\n",
       "      <td>52</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92137</th>\n",
       "      <td>chrY</td>\n",
       "      <td>11986732</td>\n",
       "      <td>rs867343413</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>16.5</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;CSQ=T|non_coding_transcript_exon_variant|M...</td>\n",
       "      <td>GT:GQ:DP:AD:VAF:PL</td>\n",
       "      <td>1/1:15:13:0,13:1:16,20,0</td>\n",
       "      <td>13</td>\n",
       "      <td>0,13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>83656 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT  QUAL FILTER  \\\n",
       "0      chr1     69511    rs2691305   A   G  32.1   PASS   \n",
       "1      chr1    817514    rs3131971   T   C  47.1   PASS   \n",
       "2      chr1    826893    rs3115849   G   A  31.8   PASS   \n",
       "3      chr1    827209    rs3115848   G   C  47.6   PASS   \n",
       "4      chr1    827212    rs3131950   C   G  51.7   PASS   \n",
       "...     ...       ...          ...  ..  ..   ...    ...   \n",
       "92127  chrY  11153722   rs77624695   C   T  20.1   PASS   \n",
       "92134  chrY  11986256    rs9650860   T   C  31.4   PASS   \n",
       "92135  chrY  11986362    rs9650861   A   C  31.2   PASS   \n",
       "92136  chrY  11986608  rs879016544   T   C  33.8   PASS   \n",
       "92137  chrY  11986732  rs867343413   C   T  16.5   PASS   \n",
       "\n",
       "                                                    INFO              FORMAT  \\\n",
       "0      ASP;CSQ=G|missense_variant|MODERATE|OR4F5|ENSG...  GT:GQ:DP:AD:VAF:PL   \n",
       "1      ASP;CAF=0.2468,0.7532;COMMON=1;CSQ=C|non_codin...  GT:GQ:DP:AD:VAF:PL   \n",
       "2      ASP;CAF=0.2682,0.7318;COMMON=1;CSQ=A|upstream_...  GT:GQ:DP:AD:VAF:PL   \n",
       "3      ASP;CAF=0.2484,0.7516;COMMON=1;CSQ=C|upstream_...  GT:GQ:DP:AD:VAF:PL   \n",
       "4      ASP;CAF=0.2484,0.7516,.;COMMON=1;CSQ=G|upstrea...  GT:GQ:DP:AD:VAF:PL   \n",
       "...                                                  ...                 ...   \n",
       "92127  ASP;CSQ=T|downstream_gene_variant|MODIFIER||EN...  GT:GQ:DP:AD:VAF:PL   \n",
       "92134  ASP;CSQ=C|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "92135  ASP;CSQ=C|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "92136  ASP;CSQ=C|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "92137  ASP;CSQ=T|non_coding_transcript_exon_variant|M...  GT:GQ:DP:AD:VAF:PL   \n",
       "\n",
       "                                  SAMPLE   DP     AD  RD  A_D       VAF  \n",
       "0      1/1:32:222:0,221:0.995495:32,42,0  222  0,221   0  221  1.000000  \n",
       "1        1/1:40:61:1,60:0.983607:47,40,0   61   1,60   1   60  0.983607  \n",
       "2               1/1:31:90:0,90:1:31,37,0   90   0,90   0   90  1.000000  \n",
       "3      1/1:47:129:4,125:0.968992:47,55,0  129  4,125   4  125  0.968992  \n",
       "4      1/1:48:126:4,122:0.968254:51,50,0  126  4,122   4  122  0.968254  \n",
       "...                                  ...  ...    ...  ..  ...       ...  \n",
       "92127           1/1:20:11:0,11:1:20,38,0   11   0,11   0   11  1.000000  \n",
       "92134           1/1:31:13:0,13:1:31,40,0   13   0,13   0   13  1.000000  \n",
       "92135           1/1:28:63:0,63:1:31,31,0   63   0,63   0   63  1.000000  \n",
       "92136           1/1:33:52:0,52:1:33,38,0   52   0,52   0   52  1.000000  \n",
       "92137           1/1:15:13:0,13:1:16,20,0   13   0,13   0   13  1.000000  \n",
       "\n",
       "[83656 rows x 15 columns]"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_DEEPVARIANT.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[2].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[3]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "filter_var = vcf[vcf['VAF'] >= 0.001]\n",
    "filter_var_DP = filter_var[filter_var['DP'] >= 10]\n",
    "#filter_var_DP = vcf.copy()\n",
    "filter_var_DP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8adc566",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter_var_DP['POS'] = filter_var_DP['POS'].astype(int)\n",
    "\n",
    "merged = pd.merge(filter_var_DP, df, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged = merged.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "\n",
    "# To fill NaN values in the 'Classification' column with 'NA', you need to assign the result back to the column.\n",
    "merged['Classification'] = merged['Classification'].fillna('NA')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "f84b67a1",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,31;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,31:0.970:31:0,13:0,9:0,0,0,31</td>\n",
       "      <td>31</td>\n",
       "      <td>0,31</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,14...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,62:0.984:62:0,26:0,35:0,0,48,14</td>\n",
       "      <td>62</td>\n",
       "      <td>0,62</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:930939</td>\n",
       "      <td>G/A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.792-100G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.953474</td>\n",
       "      <td>5008</td>\n",
       "      <td>4775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>935954</td>\n",
       "      <td>rs4072383</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|2,14;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,16:0.941:16:0,2:0,13:0,0,2,14</td>\n",
       "      <td>16</td>\n",
       "      <td>0,16</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:935954</td>\n",
       "      <td>G/T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.967+58G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.497804</td>\n",
       "      <td>5008</td>\n",
       "      <td>2493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|17,4;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,21:0.955:21:0,10:0,11:0,0,17,4</td>\n",
       "      <td>21</td>\n",
       "      <td>0,21</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:941119</td>\n",
       "      <td>A/G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.1196-25A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD European Finnish</td>\n",
       "      <td>0.960537</td>\n",
       "      <td>16902</td>\n",
       "      <td>16235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944858</td>\n",
       "      <td>rs3748592</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|9,6;C...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,15:0.941:15:0,6:0,9:0,0,9,6</td>\n",
       "      <td>15</td>\n",
       "      <td>0,15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:944858</td>\n",
       "      <td>A/G</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>26155</td>\n",
       "      <td>NM_015658.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_015658.4:c.2144-58T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>NOC2L:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.920927</td>\n",
       "      <td>5008</td>\n",
       "      <td>4612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73289</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12738369</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;map_qual;weak_evidence</td>\n",
       "      <td>AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB</td>\n",
       "      <td>0|1:36,1:0.059:37:13,1:21,0:0|1:12738359_G_T:1...</td>\n",
       "      <td>37</td>\n",
       "      <td>36,1</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>0.027027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73290</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12738371</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;map_qual;weak_evidence</td>\n",
       "      <td>AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB</td>\n",
       "      <td>0|1:35,1:0.059:36:13,1:20,0:0|1:12738359_G_T:1...</td>\n",
       "      <td>36</td>\n",
       "      <td>35,1</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>0.027778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73291</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12914512</td>\n",
       "      <td>rs2032624</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|21,7;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,28:0.967:28:0,13:0,15:0,0,21,7</td>\n",
       "      <td>28</td>\n",
       "      <td>0,28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Y:12914512</td>\n",
       "      <td>C/A</td>\n",
       "      <td>DDX3Y</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>8653</td>\n",
       "      <td>NM_004660.5</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_004660.5:c.674-52C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>DDX3Y:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.920795</td>\n",
       "      <td>3775</td>\n",
       "      <td>3476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73292</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13250864</td>\n",
       "      <td>rs376039285</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|27,0;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,27:0.966:27:0,12:0,15:0,0,27,0</td>\n",
       "      <td>27</td>\n",
       "      <td>0,27</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73293</th>\n",
       "      <td>chrY</td>\n",
       "      <td>19732200</td>\n",
       "      <td>rs750047037</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,10;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,10:0.916:10:0,6:0,3:0,0,0,10</td>\n",
       "      <td>10</td>\n",
       "      <td>0,10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Y:19732200</td>\n",
       "      <td>A/C</td>\n",
       "      <td>KDM5D</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>8284</td>\n",
       "      <td>NM_004653.5</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_004653.5:c.1093-35T&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>PP3</td>\n",
       "      <td>KDM5D:3 out of 4 novel splice site algorithms ...</td>\n",
       "      <td>VUS/Weak Pathogenic</td>\n",
       "      <td>VUS/Weak Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD South Asian</td>\n",
       "      <td>0.000447477</td>\n",
       "      <td>8939</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73294 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL  \\\n",
       "0      chr1    817514    rs3131971   T   C    .   \n",
       "1      chr1    930939    rs9988021   G   A    .   \n",
       "2      chr1    935954    rs4072383   G   T    .   \n",
       "3      chr1    941119    rs4372192   A   G    .   \n",
       "4      chr1    944858    rs3748592   A   G    .   \n",
       "...     ...       ...          ...  ..  ..  ...   \n",
       "73289  chrY  12738369            .   G   C    .   \n",
       "73290  chrY  12738371            .   T   G    .   \n",
       "73291  chrY  12914512    rs2032624   C   A    .   \n",
       "73292  chrY  13250864  rs376039285   C   T    .   \n",
       "73293  chrY  19732200  rs750047037   A   C    .   \n",
       "\n",
       "                                                  FILTER  \\\n",
       "0                                                   PASS   \n",
       "1                                                   PASS   \n",
       "2                                                   PASS   \n",
       "3                                                   PASS   \n",
       "4                                                   PASS   \n",
       "...                                                  ...   \n",
       "73289  clustered_events;haplotype;map_qual;weak_evidence   \n",
       "73290  clustered_events;haplotype;map_qual;weak_evidence   \n",
       "73291                                               PASS   \n",
       "73292                                               PASS   \n",
       "73293                                               PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,31;...   \n",
       "1      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,14...   \n",
       "2      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|2,14;...   \n",
       "3      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|17,4;...   \n",
       "4      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|9,6;C...   \n",
       "...                                                  ...   \n",
       "73289  AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...   \n",
       "73290  AS_FilterStatus=weak_evidence,map_qual;AS_SB_T...   \n",
       "73291  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|21,7;...   \n",
       "73292  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|27,0;...   \n",
       "73293  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,10;...   \n",
       "\n",
       "                                    FORMAT  \\\n",
       "0                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "1                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "2                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "3                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "4                 GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "...                                    ...   \n",
       "73289  GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB   \n",
       "73290  GT:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB   \n",
       "73291             GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "73292             GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "73293             GT:AD:AF:DP:F1R2:F2R1:SB   \n",
       "\n",
       "                                                  SAMPLE  DP    AD  RD  A_D  \\\n",
       "0                    0/1:0,31:0.970:31:0,13:0,9:0,0,0,31  31  0,31   0   31   \n",
       "1                  0/1:0,62:0.984:62:0,26:0,35:0,0,48,14  62  0,62   0   62   \n",
       "2                    0/1:0,16:0.941:16:0,2:0,13:0,0,2,14  16  0,16   0   16   \n",
       "3                   0/1:0,21:0.955:21:0,10:0,11:0,0,17,4  21  0,21   0   21   \n",
       "4                      0/1:0,15:0.941:15:0,6:0,9:0,0,9,6  15  0,15   0   15   \n",
       "...                                                  ...  ..   ...  ..  ...   \n",
       "73289  0|1:36,1:0.059:37:13,1:21,0:0|1:12738359_G_T:1...  37  36,1  36    1   \n",
       "73290  0|1:35,1:0.059:36:13,1:20,0:0|1:12738359_G_T:1...  36  35,1  35    1   \n",
       "73291               0/1:0,28:0.967:28:0,13:0,15:0,0,21,7  28  0,28   0   28   \n",
       "73292               0/1:0,27:0.966:27:0,12:0,15:0,0,27,0  27  0,27   0   27   \n",
       "73293                 0/1:0,10:0.916:10:0,6:0,3:0,0,0,10  10  0,10   0   10   \n",
       "\n",
       "            VAF     Chr:Pos Ref/Alt Gene Name     Gene Inheritance  \\\n",
       "0      1.000000    1:817514     T/C    FAM87B  Default (Recessive)   \n",
       "1      1.000000    1:930939     G/A    SAMD11  Default (Recessive)   \n",
       "2      1.000000    1:935954     G/T    SAMD11  Default (Recessive)   \n",
       "3      1.000000    1:941119     A/G    SAMD11  Default (Recessive)   \n",
       "4      1.000000    1:944858     A/G     NOC2L  Default (Recessive)   \n",
       "...         ...         ...     ...       ...                  ...   \n",
       "73289  0.027027         NaN     NaN       NaN                  NaN   \n",
       "73290  0.027778         NaN     NaN       NaN                  NaN   \n",
       "73291  1.000000  Y:12914512     C/A     DDX3Y  Default (Recessive)   \n",
       "73292  1.000000         NaN     NaN       NaN                  NaN   \n",
       "73293  1.000000  Y:19732200     A/C     KDM5D  Default (Recessive)   \n",
       "\n",
       "      Entrez Gene ID Transcript Name Transcript Strand  \\\n",
       "0             400728     NR_103536.1                 +   \n",
       "1             148398  NM_001385641.1                 +   \n",
       "2             148398  NM_001385641.1                 +   \n",
       "3             148398  NM_001385641.1                 +   \n",
       "4              26155     NM_015658.4                 -   \n",
       "...              ...             ...               ...   \n",
       "73289            NaN             NaN               NaN   \n",
       "73290            NaN             NaN               NaN   \n",
       "73291           8653     NM_004660.5                 +   \n",
       "73292            NaN             NaN               NaN   \n",
       "73293           8284     NM_004653.5                 -   \n",
       "\n",
       "                         HGVS cDot HGVS pDot        Sequence Ontology  \\\n",
       "0             NR_103536.1:n.144T>C       NaN  non_coding_exon_variant   \n",
       "1      NM_001385641.1:c.792-100G>A       NaN           intron_variant   \n",
       "2       NM_001385641.1:c.967+58G>T       NaN           intron_variant   \n",
       "3      NM_001385641.1:c.1196-25A>G       NaN           intron_variant   \n",
       "4         NM_015658.4:c.2144-58T>C       NaN           intron_variant   \n",
       "...                            ...       ...                      ...   \n",
       "73289                          NaN       NaN                      NaN   \n",
       "73290                          NaN       NaN                      NaN   \n",
       "73291      NM_004660.5:c.674-52C>A       NaN           intron_variant   \n",
       "73292                          NaN       NaN                      NaN   \n",
       "73293     NM_004653.5:c.1093-35T>G       NaN           intron_variant   \n",
       "\n",
       "                 Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                  Single Transcript Gene                  BA1,BS2,BP7   \n",
       "1      Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "2      Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "3      Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "4                  Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                                   ...                          ...   \n",
       "73289                                 NaN                          NaN   \n",
       "73290                                 NaN                          NaN   \n",
       "73291  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "73292                                 NaN                          NaN   \n",
       "73293  Other Transcripts Have Same Effect                          PP3   \n",
       "\n",
       "                ACMG Classification Criteria Description       Classification  \\\n",
       "0      FAM87B:Allele frequency is above 0.01 recessiv...               Benign   \n",
       "1      SAMD11:Allele frequency is above 0.01 recessiv...               Benign   \n",
       "2      SAMD11:Allele frequency is above 0.01 recessiv...               Benign   \n",
       "3      SAMD11:Allele frequency is above 0.01 recessiv...               Benign   \n",
       "4      NOC2L:Allele frequency is above 0.01 recessive...               Benign   \n",
       "...                                                  ...                  ...   \n",
       "73289                                                NaN                 None   \n",
       "73290                                                NaN                 None   \n",
       "73291  DDX3Y:Allele frequency is above 0.01 recessive...               Benign   \n",
       "73292                                                NaN                 None   \n",
       "73293  KDM5D:3 out of 4 novel splice site algorithms ...  VUS/Weak Pathogenic   \n",
       "\n",
       "       Auto Classification Previous Classification Count  \\\n",
       "0                   Benign                           NaN   \n",
       "1                   Benign                           NaN   \n",
       "2                   Benign                           NaN   \n",
       "3                   Benign                           NaN   \n",
       "4                   Benign                           NaN   \n",
       "...                    ...                           ...   \n",
       "73289                  NaN                           NaN   \n",
       "73290                  NaN                           NaN   \n",
       "73291               Benign                           NaN   \n",
       "73292                  NaN                           NaN   \n",
       "73293  VUS/Weak Pathogenic                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "73289                     NaN                      NaN   \n",
       "73290                     NaN                      NaN   \n",
       "73291                     NaN                      NaN   \n",
       "73292                     NaN                      NaN   \n",
       "73293                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name      Max MAF Max Allele Number  \\\n",
       "0                      Annotated 1kG All     0.753195              5008   \n",
       "1                      Annotated 1kG All     0.953474              5008   \n",
       "2                      Annotated 1kG All     0.497804              5008   \n",
       "3      Annotated gnomAD European Finnish     0.960537             16902   \n",
       "4                      Annotated 1kG All     0.920927              5008   \n",
       "...                                  ...          ...               ...   \n",
       "73289                                NaN          NaN               NaN   \n",
       "73290                                NaN          NaN               NaN   \n",
       "73291                  Annotated 1kG All     0.920795              3775   \n",
       "73292                                NaN          NaN               NaN   \n",
       "73293       Annotated gnomAD South Asian  0.000447477              8939   \n",
       "\n",
       "      Max Allele Count  \n",
       "0                 3772  \n",
       "1                 4775  \n",
       "2                 2493  \n",
       "3                16235  \n",
       "4                 4612  \n",
       "...                ...  \n",
       "73289              NaN  \n",
       "73290              NaN  \n",
       "73291             3476  \n",
       "73292              NaN  \n",
       "73293                4  \n",
       "\n",
       "[73294 rows x 37 columns]"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of your specified priorities\n",
    "priorities = ['Pathogenic', 'Likely Pathogenic', 'VUS/Weak Pathogenic', 'VUS/Conflicting', 'VUS', 'VUS/Weak Benign', 'Likely Benign', 'Benign']\n",
    "\n",
    "# Function to select the highest ranked value from Original_Column\n",
    "def select_highest_ranked(row):\n",
    "    values = row['Classification'].split(',')\n",
    "    for rank in priorities:\n",
    "        if rank in values:\n",
    "            return rank\n",
    "    return None  # Return None if no rank found\n",
    "\n",
    "# Apply the function to create the final selected rank column\n",
    "merged['Classification'] = merged.apply(select_highest_ranked, axis=1)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "eefb2dec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant_classification</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Benign</td>\n",
       "      <td>60945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>9764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>1044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VUS/Weak Benign</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>VUS/Weak Pathogenic</td>\n",
       "      <td>266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>VUS</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Variant_classification  Counts\n",
       "0                 Benign   60945\n",
       "1                   None    9764\n",
       "2        VUS/Conflicting    1044\n",
       "3          Likely Benign     963\n",
       "4        VUS/Weak Benign     268\n",
       "5    VUS/Weak Pathogenic     266\n",
       "6                    VUS      37\n",
       "7      Likely Pathogenic       7"
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "value_counts_with_null_bcftools = merged['Classification'].value_counts(dropna=False)\n",
    "#result_df = pd.DataFrame({'Counts': value_counts_with_null_bcftools})\n",
    "result_df = pd.DataFrame({'Variant_classification': value_counts_with_null_bcftools.index, 'Counts': value_counts_with_null_bcftools.values})\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "eb1c5a63",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/new_output_files/ACMG_outputs/ACMG_new/0.001_DP10_ACMG_MUTECT2_12652707.xlsx', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ad81ce03",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3157c49f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "bec59bf3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:827209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.314C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894115</td>\n",
       "      <td>7952</td>\n",
       "      <td>7110</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:827212</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.311G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894327</td>\n",
       "      <td>7826</td>\n",
       "      <td>6999</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481238</th>\n",
       "      <td>X:156007686</td>\n",
       "      <td>C/-</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>3581</td>\n",
       "      <td>NM_002186.3</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_002186.3:c.972+79delC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>IL9R:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.997604</td>\n",
       "      <td>5008</td>\n",
       "      <td>4996</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156007686</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481240</th>\n",
       "      <td>Y:11986362</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481241</th>\n",
       "      <td>Y:11986608</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481242</th>\n",
       "      <td>Y:11986732</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986732</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481243</th>\n",
       "      <td>Y:11986765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>403159 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Chr:Pos Ref/Alt  Gene Name     Gene Inheritance Entrez Gene ID  \\\n",
       "0           1:69511     A/G      OR4F5  Default (Recessive)          79501   \n",
       "1          1:817514     T/C     FAM87B  Default (Recessive)         400728   \n",
       "2          1:826893     G/A  LINC00115  Default (Recessive)          79854   \n",
       "3          1:827209     G/C  LINC00115  Default (Recessive)          79854   \n",
       "4          1:827212     C/G  LINC00115  Default (Recessive)          79854   \n",
       "...             ...     ...        ...                  ...            ...   \n",
       "481238  X:156007686     C/-       IL9R  Default (Recessive)           3581   \n",
       "481240   Y:11986362     NaN        NaN                  NaN            NaN   \n",
       "481241   Y:11986608     NaN        NaN                  NaN            NaN   \n",
       "481242   Y:11986732     NaN        NaN                  NaN            NaN   \n",
       "481243   Y:11986765     NaN        NaN                  NaN            NaN   \n",
       "\n",
       "       Transcript Name Transcript Strand                 HGVS cDot  \\\n",
       "0       NM_001005484.2                 +   NM_001005484.2:c.484A>G   \n",
       "1          NR_103536.1                 +      NR_103536.1:n.144T>C   \n",
       "2          NR_024321.1                 -      NR_024321.1:n.630C>T   \n",
       "3          NR_024321.1                 -      NR_024321.1:n.314C>G   \n",
       "4          NR_024321.1                 -      NR_024321.1:n.311G>C   \n",
       "...                ...               ...                       ...   \n",
       "481238     NM_002186.3                 +  NM_002186.3:c.972+79delC   \n",
       "481240             NaN               NaN                       NaN   \n",
       "481241             NaN               NaN                       NaN   \n",
       "481242             NaN               NaN                       NaN   \n",
       "481243             NaN               NaN                       NaN   \n",
       "\n",
       "                         HGVS pDot        Sequence Ontology  \\\n",
       "0       NP_001005484.2:p.Thr162Ala         missense_variant   \n",
       "1                              NaN  non_coding_exon_variant   \n",
       "2                              NaN  non_coding_exon_variant   \n",
       "3                              NaN  non_coding_exon_variant   \n",
       "4                              NaN  non_coding_exon_variant   \n",
       "...                            ...                      ...   \n",
       "481238                         NaN           intron_variant   \n",
       "481240                         NaN                      NaN   \n",
       "481241                         NaN                      NaN   \n",
       "481242                         NaN                      NaN   \n",
       "481243                         NaN                      NaN   \n",
       "\n",
       "                  Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                   Single Transcript Gene                      BA1,BP4   \n",
       "1                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "3                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "4                   Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                                    ...                          ...   \n",
       "481238  Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "481240                                 NaN                          NaN   \n",
       "481241                                 NaN                          NaN   \n",
       "481242                                 NaN                          NaN   \n",
       "481243                                 NaN                          NaN   \n",
       "\n",
       "                 ACMG Classification Criteria Description Classification  \\\n",
       "0       OR4F5:Allele frequency is above 0.01 recessive...         Benign   \n",
       "1       FAM87B:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "2       LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "3       LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "4       LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "...                                                   ...            ...   \n",
       "481238  IL9R:Allele frequency is above 0.01 recessive ...         Benign   \n",
       "481240                                                NaN            NaN   \n",
       "481241                                                NaN            NaN   \n",
       "481242                                                NaN            NaN   \n",
       "481243                                                NaN            NaN   \n",
       "\n",
       "       Auto Classification Previous Classification Count  \\\n",
       "0                   Benign                           NaN   \n",
       "1                   Benign                           NaN   \n",
       "2                   Benign                           NaN   \n",
       "3                   Benign                           NaN   \n",
       "4                   Benign                           NaN   \n",
       "...                    ...                           ...   \n",
       "481238              Benign                           NaN   \n",
       "481240                 NaN                           NaN   \n",
       "481241                 NaN                           NaN   \n",
       "481242                 NaN                           NaN   \n",
       "481243                 NaN                           NaN   \n",
       "\n",
       "       Previous Classification Last Classification Date  \\\n",
       "0                          NaN                      NaN   \n",
       "1                          NaN                      NaN   \n",
       "2                          NaN                      NaN   \n",
       "3                          NaN                      NaN   \n",
       "4                          NaN                      NaN   \n",
       "...                        ...                      ...   \n",
       "481238                     NaN                      NaN   \n",
       "481240                     NaN                      NaN   \n",
       "481241                     NaN                      NaN   \n",
       "481242                     NaN                      NaN   \n",
       "481243                     NaN                      NaN   \n",
       "\n",
       "       Max Sub Population Freq Group Name   Max MAF Max Allele Number  \\\n",
       "0             Annotated gnomAD East Asian  0.999461             16698   \n",
       "1                       Annotated 1kG All  0.753195              5008   \n",
       "2             Annotated gnomAD East Asian  0.896992             10106   \n",
       "3             Annotated gnomAD East Asian  0.894115              7952   \n",
       "4             Annotated gnomAD East Asian  0.894327              7826   \n",
       "...                                   ...       ...               ...   \n",
       "481238                  Annotated 1kG All  0.997604              5008   \n",
       "481240                                NaN       NaN               NaN   \n",
       "481241                                NaN       NaN               NaN   \n",
       "481242                                NaN       NaN               NaN   \n",
       "481243                                NaN       NaN               NaN   \n",
       "\n",
       "       Max Allele Count CHROM        POS  REF  ALT  \n",
       "0                 16689  chr1      69511    A    G  \n",
       "1                  3772  chr1     817514    T    C  \n",
       "2                  9065  chr1     826893    G    A  \n",
       "3                  7110  chr1     827209    G    C  \n",
       "4                  6999  chr1     827212    C    G  \n",
       "...                 ...   ...        ...  ...  ...  \n",
       "481238             4996  chrX  156007686    C    -  \n",
       "481240              NaN  chrY   11986362  NaN  NaN  \n",
       "481241              NaN  chrY   11986608  NaN  NaN  \n",
       "481242              NaN  chrY   11986732  NaN  NaN  \n",
       "481243              NaN  chrY   11986765  NaN  NaN  \n",
       "\n",
       "[403159 rows x 26 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns_to_check = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "\n",
    "# Use the duplicated() method to find duplicated rows\n",
    "duplicates = df[df.duplicated(subset=columns_to_check, keep=False)]\n",
    "duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "b45f1049",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr:Pos Ref/Alt Gene Name     Gene Inheritance Entrez Gene ID  \\\n",
       "0  1:69511     A/G     OR4F5  Default (Recessive)          79501   \n",
       "\n",
       "  Transcript Name Transcript Strand                HGVS cDot  \\\n",
       "0  NM_001005484.2                 +  NM_001005484.2:c.484A>G   \n",
       "\n",
       "                    HGVS pDot Sequence Ontology Other Transcript Effects  \\\n",
       "0  NP_001005484.2:p.Thr162Ala  missense_variant   Single Transcript Gene   \n",
       "\n",
       "  ACMG Classification Criteria  \\\n",
       "0                      BA1,BP4   \n",
       "\n",
       "            ACMG Classification Criteria Description Classification  \\\n",
       "0  OR4F5:Allele frequency is above 0.01 recessive...         Benign   \n",
       "\n",
       "  Auto Classification Previous Classification Count Previous Classification  \\\n",
       "0              Benign                           NaN                     NaN   \n",
       "\n",
       "  Last Classification Date Max Sub Population Freq Group Name   Max MAF  \\\n",
       "0                      NaN        Annotated gnomAD East Asian  0.999461   \n",
       "\n",
       "  Max Allele Number Max Allele Count CHROM    POS REF ALT  \n",
       "0             16698            16689  chr1  69511   A   G  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['Chr:Pos'] == '1:69511']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37943943",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "b1917c15",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_BCFTOOL.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[6]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "vcf['DP'] = vcf['INFO'].str.extract(r'DP=(\\d+)')[0].fillna('0').astype(int)\n",
    "#filter_var = vcf[vcf['VAF'] >= 0.1]\n",
    "#filter_var_DP = filter_var[filter_var['DP'] >= 10]\n",
    "filter_var_DP = vcf.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "9b8d678f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>118.3520</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:145,255,0:230:0:0,201:1,28:1,229:0</td>\n",
       "      <td>1,229</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.995652</td>\n",
       "      <td>248</td>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,255,0:93:0:0,38:0,55:0,93:0</td>\n",
       "      <td>0,93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>97</td>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,178,0:59:0:0,53:0,6:0,59:0</td>\n",
       "      <td>0,59</td>\n",
       "      <td>0</td>\n",
       "      <td>59</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>67</td>\n",
       "      <td>1:930939</td>\n",
       "      <td>G/A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.792-100G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.953474</td>\n",
       "      <td>5008</td>\n",
       "      <td>4775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>205.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:235,51,0:17:0:0,15:0,2:0,17:0</td>\n",
       "      <td>0,17</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>23</td>\n",
       "      <td>1:941119</td>\n",
       "      <td>A/G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.1196-25A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD European Finnish</td>\n",
       "      <td>0.960537</td>\n",
       "      <td>16902</td>\n",
       "      <td>16235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,6,0:2:0:0,2:0,0:0,2:0</td>\n",
       "      <td>0,2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,166,0:55:0:0,14:0,41:0,55:0</td>\n",
       "      <td>0,55</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79447</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1418109</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>25.4267</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:55,255,0:99:0:0,26:0,73:0,99:0</td>\n",
       "      <td>0,99</td>\n",
       "      <td>0</td>\n",
       "      <td>99</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79448</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>26.4242</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:56,208,0:69:0:0,17:0,52:0,69:0</td>\n",
       "      <td>0,69</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>79</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79449</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1428198</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,148,0:49:0:0,12:0,37:0,49:0</td>\n",
       "      <td>0,49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>57</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79450</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2221159</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>28.4205</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:58,255,0:153:0:0,42:0,111:0,153:0</td>\n",
       "      <td>0,153</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79447 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM      POS       rsID REF ALT      QUAL FILTER  \\\n",
       "0      chr1   817514  rs3131971   T   C  118.3520      .   \n",
       "1      chr1   826893  rs3115849   G   A   34.4159      .   \n",
       "2      chr1   930939  rs9988021   G   A  225.4170      .   \n",
       "3      chr1   941119  rs4372192   A   G  205.4170      .   \n",
       "4      chr1   942451  rs6672356   T   C   34.4159      .   \n",
       "...     ...      ...        ...  ..  ..       ...    ...   \n",
       "79446  chrY  1387425          .   C   G   23.4340      .   \n",
       "79447  chrY  1418109          .   C   G   25.4267      .   \n",
       "79448  chrY  1425778          .   G   A   26.4242      .   \n",
       "79449  chrY  1428198          .   C   A   23.4340      .   \n",
       "79450  chrY  2221159          .   T   C   28.4205      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...   \n",
       "1      DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...   \n",
       "2      DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...   \n",
       "3      DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...   \n",
       "4      DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...   \n",
       "...                                                  ...   \n",
       "79446  DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...   \n",
       "79447  DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...   \n",
       "79448  DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...   \n",
       "79449  DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...   \n",
       "79450  DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...   \n",
       "\n",
       "                           FORMAT                                  SAMPLE  \\\n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR  1/1:145,255,0:230:0:0,201:1,28:1,229:0   \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:64,255,0:93:0:0,38:0,55:0,93:0   \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,178,0:59:0:0,53:0,6:0,59:0   \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR       1/1:235,51,0:17:0:0,15:0,2:0,17:0   \n",
       "4      GT:PL:DP:SP:ADF:ADR:AD:SCR            1/1:64,6,0:2:0:0,2:0,0:0,2:0   \n",
       "...                           ...                                     ...   \n",
       "79446  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,166,0:55:0:0,14:0,41:0,55:0   \n",
       "79447  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:55,255,0:99:0:0,26:0,73:0,99:0   \n",
       "79448  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:56,208,0:69:0:0,17:0,52:0,69:0   \n",
       "79449  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,148,0:49:0:0,12:0,37:0,49:0   \n",
       "79450  GT:PL:DP:SP:ADF:ADR:AD:SCR   1/1:58,255,0:153:0:0,42:0,111:0,153:0   \n",
       "\n",
       "          AD  RD  A_D       VAF   DP   Chr:Pos Ref/Alt  Gene Name  \\\n",
       "0      1,229   1  229  0.995652  248  1:817514     T/C     FAM87B   \n",
       "1       0,93   0   93  1.000000   97  1:826893     G/A  LINC00115   \n",
       "2       0,59   0   59  1.000000   67  1:930939     G/A     SAMD11   \n",
       "3       0,17   0   17  1.000000   23  1:941119     A/G     SAMD11   \n",
       "4        0,2   0    2  1.000000    2       NaN     NaN        NaN   \n",
       "...      ...  ..  ...       ...  ...       ...     ...        ...   \n",
       "79446   0,55   0   55  1.000000   65       NaN     NaN        NaN   \n",
       "79447   0,99   0   99  1.000000  111       NaN     NaN        NaN   \n",
       "79448   0,69   0   69  1.000000   79       NaN     NaN        NaN   \n",
       "79449   0,49   0   49  1.000000   57       NaN     NaN        NaN   \n",
       "79450  0,153   0  153  1.000000  174       NaN     NaN        NaN   \n",
       "\n",
       "          Gene Inheritance Entrez Gene ID Transcript Name Transcript Strand  \\\n",
       "0      Default (Recessive)         400728     NR_103536.1                 +   \n",
       "1      Default (Recessive)          79854     NR_024321.1                 -   \n",
       "2      Default (Recessive)         148398  NM_001385641.1                 +   \n",
       "3      Default (Recessive)         148398  NM_001385641.1                 +   \n",
       "4                      NaN            NaN             NaN               NaN   \n",
       "...                    ...            ...             ...               ...   \n",
       "79446                  NaN            NaN             NaN               NaN   \n",
       "79447                  NaN            NaN             NaN               NaN   \n",
       "79448                  NaN            NaN             NaN               NaN   \n",
       "79449                  NaN            NaN             NaN               NaN   \n",
       "79450                  NaN            NaN             NaN               NaN   \n",
       "\n",
       "                         HGVS cDot HGVS pDot        Sequence Ontology  \\\n",
       "0             NR_103536.1:n.144T>C       NaN  non_coding_exon_variant   \n",
       "1             NR_024321.1:n.630C>T       NaN  non_coding_exon_variant   \n",
       "2      NM_001385641.1:c.792-100G>A       NaN           intron_variant   \n",
       "3      NM_001385641.1:c.1196-25A>G       NaN           intron_variant   \n",
       "4                              NaN       NaN                      NaN   \n",
       "...                            ...       ...                      ...   \n",
       "79446                          NaN       NaN                      NaN   \n",
       "79447                          NaN       NaN                      NaN   \n",
       "79448                          NaN       NaN                      NaN   \n",
       "79449                          NaN       NaN                      NaN   \n",
       "79450                          NaN       NaN                      NaN   \n",
       "\n",
       "                 Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                  Single Transcript Gene                  BA1,BS2,BP7   \n",
       "1                  Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2      Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "3      Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "4                                     NaN                          NaN   \n",
       "...                                   ...                          ...   \n",
       "79446                                 NaN                          NaN   \n",
       "79447                                 NaN                          NaN   \n",
       "79448                                 NaN                          NaN   \n",
       "79449                                 NaN                          NaN   \n",
       "79450                                 NaN                          NaN   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0      FAM87B:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "1      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "2      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "3      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "4                                                    NaN             NA   \n",
       "...                                                  ...            ...   \n",
       "79446                                                NaN             NA   \n",
       "79447                                                NaN             NA   \n",
       "79448                                                NaN             NA   \n",
       "79449                                                NaN             NA   \n",
       "79450                                                NaN             NA   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                  Benign                           NaN   \n",
       "1                  Benign                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                  Benign                           NaN   \n",
       "4                     NaN                           NaN   \n",
       "...                   ...                           ...   \n",
       "79446                 NaN                           NaN   \n",
       "79447                 NaN                           NaN   \n",
       "79448                 NaN                           NaN   \n",
       "79449                 NaN                           NaN   \n",
       "79450                 NaN                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "79446                     NaN                      NaN   \n",
       "79447                     NaN                      NaN   \n",
       "79448                     NaN                      NaN   \n",
       "79449                     NaN                      NaN   \n",
       "79450                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name   Max MAF Max Allele Number  \\\n",
       "0                      Annotated 1kG All  0.753195              5008   \n",
       "1            Annotated gnomAD East Asian  0.896992             10106   \n",
       "2                      Annotated 1kG All  0.953474              5008   \n",
       "3      Annotated gnomAD European Finnish  0.960537             16902   \n",
       "4                                    NaN       NaN               NaN   \n",
       "...                                  ...       ...               ...   \n",
       "79446                                NaN       NaN               NaN   \n",
       "79447                                NaN       NaN               NaN   \n",
       "79448                                NaN       NaN               NaN   \n",
       "79449                                NaN       NaN               NaN   \n",
       "79450                                NaN       NaN               NaN   \n",
       "\n",
       "      Max Allele Count  \n",
       "0                 3772  \n",
       "1                 9065  \n",
       "2                 4775  \n",
       "3                16235  \n",
       "4                  NaN  \n",
       "...                ...  \n",
       "79446              NaN  \n",
       "79447              NaN  \n",
       "79448              NaN  \n",
       "79449              NaN  \n",
       "79450              NaN  \n",
       "\n",
       "[79447 rows x 37 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filter_var_DP['POS'] = filter_var_DP['POS'].astype(int)\n",
    "\n",
    "merged = pd.merge(filter_var_DP, df, on=['CHROM', 'POS', 'REF', 'ALT'], how='left', sort=False)\n",
    "merged = merged.drop_duplicates(subset=['CHROM', 'POS', 'REF', 'ALT'])\n",
    "\n",
    "# To fill NaN values in the 'Classification' column with 'NA', you need to assign the result back to the column.\n",
    "merged['Classification'] = merged['Classification'].fillna('NA')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "ac90fa56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "      <th>new_Classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>118.3520</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:145,255,0:230:0:0,201:1,28:1,229:0</td>\n",
       "      <td>1,229</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.995652</td>\n",
       "      <td>248</td>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,255,0:93:0:0,38:0,55:0,93:0</td>\n",
       "      <td>0,93</td>\n",
       "      <td>0</td>\n",
       "      <td>93</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>97</td>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:255,178,0:59:0:0,53:0,6:0,59:0</td>\n",
       "      <td>0,59</td>\n",
       "      <td>0</td>\n",
       "      <td>59</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>67</td>\n",
       "      <td>1:930939</td>\n",
       "      <td>G/A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.792-100G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.953474</td>\n",
       "      <td>5008</td>\n",
       "      <td>4775</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>205.4170</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:235,51,0:17:0:0,15:0,2:0,17:0</td>\n",
       "      <td>0,17</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>23</td>\n",
       "      <td>1:941119</td>\n",
       "      <td>A/G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.1196-25A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD European Finnish</td>\n",
       "      <td>0.960537</td>\n",
       "      <td>16902</td>\n",
       "      <td>16235</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>34.4159</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:64,6,0:2:0:0,2:0,0:0,2:0</td>\n",
       "      <td>0,2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1387425</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,166,0:55:0:0,14:0,41:0,55:0</td>\n",
       "      <td>0,55</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79447</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1418109</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>25.4267</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:55,255,0:99:0:0,26:0,73:0,99:0</td>\n",
       "      <td>0,99</td>\n",
       "      <td>0</td>\n",
       "      <td>99</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79448</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1425778</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>26.4242</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:56,208,0:69:0:0,17:0,52:0,69:0</td>\n",
       "      <td>0,69</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>79</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79449</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1428198</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>23.4340</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:53,148,0:49:0:0,12:0,37:0,49:0</td>\n",
       "      <td>0,49</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>57</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79450</th>\n",
       "      <td>chrY</td>\n",
       "      <td>2221159</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>28.4205</td>\n",
       "      <td>.</td>\n",
       "      <td>DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...</td>\n",
       "      <td>GT:PL:DP:SP:ADF:ADR:AD:SCR</td>\n",
       "      <td>1/1:58,255,0:153:0:0,42:0,111:0,153:0</td>\n",
       "      <td>0,153</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79447 rows Ã— 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM      POS       rsID REF ALT      QUAL FILTER  \\\n",
       "0      chr1   817514  rs3131971   T   C  118.3520      .   \n",
       "1      chr1   826893  rs3115849   G   A   34.4159      .   \n",
       "2      chr1   930939  rs9988021   G   A  225.4170      .   \n",
       "3      chr1   941119  rs4372192   A   G  205.4170      .   \n",
       "4      chr1   942451  rs6672356   T   C   34.4159      .   \n",
       "...     ...      ...        ...  ..  ..       ...    ...   \n",
       "79446  chrY  1387425          .   C   G   23.4340      .   \n",
       "79447  chrY  1418109          .   C   G   25.4267      .   \n",
       "79448  chrY  1425778          .   G   A   26.4242      .   \n",
       "79449  chrY  1428198          .   C   A   23.4340      .   \n",
       "79450  chrY  2221159          .   T   C   28.4205      .   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      DP=248;ADF=0,201;ADR=1,28;AD=1,229;SCR=0;VDB=0...   \n",
       "1      DP=97;ADF=0,38;ADR=0,55;AD=0,93;SCR=0;VDB=0.66...   \n",
       "2      DP=67;ADF=0,53;ADR=0,6;AD=0,59;SCR=0;VDB=0.002...   \n",
       "3      DP=23;ADF=0,15;ADR=0,2;AD=0,17;SCR=0;VDB=0.124...   \n",
       "4      DP=2;ADF=0,2;ADR=0,0;AD=0,2;SCR=0;VDB=0.02;SGB...   \n",
       "...                                                  ...   \n",
       "79446  DP=65;ADF=0,14;ADR=0,41;AD=0,55;SCR=0;VDB=0.59...   \n",
       "79447  DP=111;ADF=0,26;ADR=0,73;AD=0,99;SCR=0;VDB=0.3...   \n",
       "79448  DP=79;ADF=0,17;ADR=0,52;AD=0,69;SCR=0;VDB=0.61...   \n",
       "79449  DP=57;ADF=0,12;ADR=0,37;AD=0,49;SCR=0;VDB=0.03...   \n",
       "79450  DP=174;ADF=0,42;ADR=0,111;AD=0,153;SCR=0;VDB=0...   \n",
       "\n",
       "                           FORMAT                                  SAMPLE  \\\n",
       "0      GT:PL:DP:SP:ADF:ADR:AD:SCR  1/1:145,255,0:230:0:0,201:1,28:1,229:0   \n",
       "1      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:64,255,0:93:0:0,38:0,55:0,93:0   \n",
       "2      GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:255,178,0:59:0:0,53:0,6:0,59:0   \n",
       "3      GT:PL:DP:SP:ADF:ADR:AD:SCR       1/1:235,51,0:17:0:0,15:0,2:0,17:0   \n",
       "4      GT:PL:DP:SP:ADF:ADR:AD:SCR            1/1:64,6,0:2:0:0,2:0,0:0,2:0   \n",
       "...                           ...                                     ...   \n",
       "79446  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,166,0:55:0:0,14:0,41:0,55:0   \n",
       "79447  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:55,255,0:99:0:0,26:0,73:0,99:0   \n",
       "79448  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:56,208,0:69:0:0,17:0,52:0,69:0   \n",
       "79449  GT:PL:DP:SP:ADF:ADR:AD:SCR      1/1:53,148,0:49:0:0,12:0,37:0,49:0   \n",
       "79450  GT:PL:DP:SP:ADF:ADR:AD:SCR   1/1:58,255,0:153:0:0,42:0,111:0,153:0   \n",
       "\n",
       "          AD  RD  A_D       VAF   DP   Chr:Pos Ref/Alt  Gene Name  \\\n",
       "0      1,229   1  229  0.995652  248  1:817514     T/C     FAM87B   \n",
       "1       0,93   0   93  1.000000   97  1:826893     G/A  LINC00115   \n",
       "2       0,59   0   59  1.000000   67  1:930939     G/A     SAMD11   \n",
       "3       0,17   0   17  1.000000   23  1:941119     A/G     SAMD11   \n",
       "4        0,2   0    2  1.000000    2       NaN     NaN        NaN   \n",
       "...      ...  ..  ...       ...  ...       ...     ...        ...   \n",
       "79446   0,55   0   55  1.000000   65       NaN     NaN        NaN   \n",
       "79447   0,99   0   99  1.000000  111       NaN     NaN        NaN   \n",
       "79448   0,69   0   69  1.000000   79       NaN     NaN        NaN   \n",
       "79449   0,49   0   49  1.000000   57       NaN     NaN        NaN   \n",
       "79450  0,153   0  153  1.000000  174       NaN     NaN        NaN   \n",
       "\n",
       "          Gene Inheritance Entrez Gene ID Transcript Name Transcript Strand  \\\n",
       "0      Default (Recessive)         400728     NR_103536.1                 +   \n",
       "1      Default (Recessive)          79854     NR_024321.1                 -   \n",
       "2      Default (Recessive)         148398  NM_001385641.1                 +   \n",
       "3      Default (Recessive)         148398  NM_001385641.1                 +   \n",
       "4                      NaN            NaN             NaN               NaN   \n",
       "...                    ...            ...             ...               ...   \n",
       "79446                  NaN            NaN             NaN               NaN   \n",
       "79447                  NaN            NaN             NaN               NaN   \n",
       "79448                  NaN            NaN             NaN               NaN   \n",
       "79449                  NaN            NaN             NaN               NaN   \n",
       "79450                  NaN            NaN             NaN               NaN   \n",
       "\n",
       "                         HGVS cDot HGVS pDot        Sequence Ontology  \\\n",
       "0             NR_103536.1:n.144T>C       NaN  non_coding_exon_variant   \n",
       "1             NR_024321.1:n.630C>T       NaN  non_coding_exon_variant   \n",
       "2      NM_001385641.1:c.792-100G>A       NaN           intron_variant   \n",
       "3      NM_001385641.1:c.1196-25A>G       NaN           intron_variant   \n",
       "4                              NaN       NaN                      NaN   \n",
       "...                            ...       ...                      ...   \n",
       "79446                          NaN       NaN                      NaN   \n",
       "79447                          NaN       NaN                      NaN   \n",
       "79448                          NaN       NaN                      NaN   \n",
       "79449                          NaN       NaN                      NaN   \n",
       "79450                          NaN       NaN                      NaN   \n",
       "\n",
       "                 Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0                  Single Transcript Gene                  BA1,BS2,BP7   \n",
       "1                  Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2      Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "3      Other Transcripts Have Same Effect                  BA1,BS2,BP7   \n",
       "4                                     NaN                          NaN   \n",
       "...                                   ...                          ...   \n",
       "79446                                 NaN                          NaN   \n",
       "79447                                 NaN                          NaN   \n",
       "79448                                 NaN                          NaN   \n",
       "79449                                 NaN                          NaN   \n",
       "79450                                 NaN                          NaN   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0      FAM87B:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "1      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "2      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "3      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "4                                                    NaN           None   \n",
       "...                                                  ...            ...   \n",
       "79446                                                NaN           None   \n",
       "79447                                                NaN           None   \n",
       "79448                                                NaN           None   \n",
       "79449                                                NaN           None   \n",
       "79450                                                NaN           None   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                  Benign                           NaN   \n",
       "1                  Benign                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                  Benign                           NaN   \n",
       "4                     NaN                           NaN   \n",
       "...                   ...                           ...   \n",
       "79446                 NaN                           NaN   \n",
       "79447                 NaN                           NaN   \n",
       "79448                 NaN                           NaN   \n",
       "79449                 NaN                           NaN   \n",
       "79450                 NaN                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "79446                     NaN                      NaN   \n",
       "79447                     NaN                      NaN   \n",
       "79448                     NaN                      NaN   \n",
       "79449                     NaN                      NaN   \n",
       "79450                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name   Max MAF Max Allele Number  \\\n",
       "0                      Annotated 1kG All  0.753195              5008   \n",
       "1            Annotated gnomAD East Asian  0.896992             10106   \n",
       "2                      Annotated 1kG All  0.953474              5008   \n",
       "3      Annotated gnomAD European Finnish  0.960537             16902   \n",
       "4                                    NaN       NaN               NaN   \n",
       "...                                  ...       ...               ...   \n",
       "79446                                NaN       NaN               NaN   \n",
       "79447                                NaN       NaN               NaN   \n",
       "79448                                NaN       NaN               NaN   \n",
       "79449                                NaN       NaN               NaN   \n",
       "79450                                NaN       NaN               NaN   \n",
       "\n",
       "      Max Allele Count new_Classification  \n",
       "0                 3772             Benign  \n",
       "1                 9065             Benign  \n",
       "2                 4775             Benign  \n",
       "3                16235             Benign  \n",
       "4                  NaN               None  \n",
       "...                ...                ...  \n",
       "79446              NaN               None  \n",
       "79447              NaN               None  \n",
       "79448              NaN               None  \n",
       "79449              NaN               None  \n",
       "79450              NaN               None  \n",
       "\n",
       "[79447 rows x 38 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of your specified priorities\n",
    "priorities = ['Pathogenic', 'Likely Pathogenic', 'VUS/Weak Pathogenic', 'VUS/Conflicting', 'VUS', 'VUS/Weak Benign', 'Likely Benign', 'Benign']\n",
    "\n",
    "# Function to select the highest ranked value from Original_Column\n",
    "def select_highest_ranked(row):\n",
    "    values = row['Classification'].split(',')\n",
    "    for rank in priorities:\n",
    "        if rank in values:\n",
    "            return rank\n",
    "    return None  # Return None if no rank found\n",
    "\n",
    "# Apply the function to create the final selected rank column\n",
    "merged['Classification'] = merged.apply(select_highest_ranked, axis=1)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "84be3fb0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Benign                 68221\n",
       "Likely Benign           1078\n",
       "VUS/Conflicting          958\n",
       "VUS/Weak Benign          366\n",
       "VUS/Weak Pathogenic      239\n",
       "VUS                       57\n",
       "Likely Pathogenic          7\n",
       "Name: Classification, dtype: int64"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.Classification.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "bf3d9eee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tool</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Benign</td>\n",
       "      <td>68221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>8521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>1078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VUS/Weak Benign</td>\n",
       "      <td>366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>VUS/Weak Pathogenic</td>\n",
       "      <td>239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>VUS</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Tool  Counts\n",
       "0               Benign   68221\n",
       "1                 None    8521\n",
       "2        Likely Benign    1078\n",
       "3      VUS/Conflicting     958\n",
       "4      VUS/Weak Benign     366\n",
       "5  VUS/Weak Pathogenic     239\n",
       "6                  VUS      57\n",
       "7    Likely Pathogenic       7"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "value_counts_with_null_bcftools = merged['Classification'].value_counts(dropna=False)\n",
    "value_counts_with_null_bcftools\n",
    "#result_df = pd.DataFrame({'Counts': value_counts_with_null_bcftools})\n",
    "result_df = pd.DataFrame({'Tool': value_counts_with_null_bcftools.index, 'Counts': value_counts_with_null_bcftools.values})\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "77455922",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/new_output_files/ACMG_outputs/ACMG_new/ACMG_BCFTOOL.xlsx', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7024342",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1d23e93",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39e3107a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcd5dca6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7fae34e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc72a11e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e45e9c3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0470adbf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "691bd7d4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e745b8f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "575bbd46",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a25e142",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ef8c3af",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af5c844a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "3ab85c87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...</td>\n",
       "      <td>211</td>\n",
       "      <td>0</td>\n",
       "      <td>210</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>367</td>\n",
       "      <td>0.997283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...</td>\n",
       "      <td>125</td>\n",
       "      <td>15</td>\n",
       "      <td>110</td>\n",
       "      <td>0.880000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...</td>\n",
       "      <td>128</td>\n",
       "      <td>15</td>\n",
       "      <td>113</td>\n",
       "      <td>0.882812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69181</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57084531</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...</td>\n",
       "      <td>62</td>\n",
       "      <td>37</td>\n",
       "      <td>25</td>\n",
       "      <td>0.403226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69182</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57128393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...</td>\n",
       "      <td>223</td>\n",
       "      <td>112</td>\n",
       "      <td>111</td>\n",
       "      <td>0.497758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69183</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184462</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...</td>\n",
       "      <td>43</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>0.465116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69184</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57189953</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9</td>\n",
       "      <td>50</td>\n",
       "      <td>30</td>\n",
       "      <td>20</td>\n",
       "      <td>0.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69185</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191772</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...</td>\n",
       "      <td>76</td>\n",
       "      <td>33</td>\n",
       "      <td>43</td>\n",
       "      <td>0.565789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53006 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69511  rs2691305   A   G    .   PASS   \n",
       "1      chr1    817514  rs3131971   T   C    .   PASS   \n",
       "2      chr1    826893  rs3115849   G   A    .   PASS   \n",
       "3      chr1    827209  rs3115848   G   C    .   PASS   \n",
       "4      chr1    827212  rs3131950   C   G    .   PASS   \n",
       "...     ...       ...        ...  ..  ..  ...    ...   \n",
       "69181  chrY  57084531          .   A   G    .   PASS   \n",
       "69182  chrY  57128393          .   G   C    .   PASS   \n",
       "69183  chrY  57184462          .   T   G    .   PASS   \n",
       "69184  chrY  57189953          .   T   C    .   PASS   \n",
       "69185  chrY  57191772          .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "1      ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...   \n",
       "2      ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....   \n",
       "3      ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "4      ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "...                                                  ...   \n",
       "69181  ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69182  ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...   \n",
       "69183  ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69184  ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...   \n",
       "69185  ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "69181  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69182  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69183  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69184  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69185  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   DP   RD   AD  \\\n",
       "0      1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...  211    0  210   \n",
       "1      1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...  368    1  367   \n",
       "2      1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55   91    0   91   \n",
       "3      1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...  125   15  110   \n",
       "4      1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...  128   15  113   \n",
       "...                                                  ...  ...  ...  ...   \n",
       "69181  0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...   62   37   25   \n",
       "69182  0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...  223  112  111   \n",
       "69183  0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...   43   23   20   \n",
       "69184   0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9   50   30   20   \n",
       "69185  0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...   76   33   43   \n",
       "\n",
       "            VAF  \n",
       "0      1.000000  \n",
       "1      0.997283  \n",
       "2      1.000000  \n",
       "3      0.880000  \n",
       "4      0.882812  \n",
       "...         ...  \n",
       "69181  0.403226  \n",
       "69182  0.497758  \n",
       "69183  0.465116  \n",
       "69184  0.400000  \n",
       "69185  0.565789  \n",
       "\n",
       "[53006 rows x 14 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_VARSCAN2.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "vcf['RD'] = vcf['SAMPLE'].str.split(':').str[4].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[5].fillna('0').astype(int)\n",
    "vcf['VAF'] = vcf['AD'] / (vcf['RD'] + vcf['AD'])\n",
    "filter_var = vcf[vcf['VAF'] >= 0.1]\n",
    "filter_var_DP = filter_var[filter_var['DP'] >= 30]\n",
    "filter_var_DP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "533fbb5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_MUTECT2.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[1]\n",
    "vcf['RD'] = vcf['AD'].str.split(',').str[0].astype(int)\n",
    "vcf['A_D'] = vcf['AD'].str.split(',').str[1].astype(int)\n",
    "vcf['VAF'] = vcf['A_D'] / (vcf['RD'] + vcf['A_D'])\n",
    "#filter_var = vcf[vcf['VAF'] >= 0.1]\n",
    "#filter_var_DP = filter_var[filter_var['DP'] >= 10]\n",
    "filter_var_DP = vcf.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7aa585b1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3da7d1db",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21e3d589",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a6d4003",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2d66b8d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8578e483",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:827209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.314C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894115</td>\n",
       "      <td>7952</td>\n",
       "      <td>7110</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:827212</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.311G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894327</td>\n",
       "      <td>7826</td>\n",
       "      <td>6999</td>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96894</th>\n",
       "      <td>Y:11154005</td>\n",
       "      <td>C/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11154005</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96895</th>\n",
       "      <td>Y:11986362</td>\n",
       "      <td>A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986362</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96896</th>\n",
       "      <td>Y:11986608</td>\n",
       "      <td>T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986608</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96897</th>\n",
       "      <td>Y:11986732</td>\n",
       "      <td>C/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986732</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96898</th>\n",
       "      <td>Y:11986765</td>\n",
       "      <td>G/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrY</td>\n",
       "      <td>11986765</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>96899 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr:Pos Ref/Alt  Gene Name     Gene Inheritance Entrez Gene ID  \\\n",
       "0         1:69511     A/G      OR4F5  Default (Recessive)          79501   \n",
       "1        1:817514     T/C     FAM87B  Default (Recessive)         400728   \n",
       "2        1:826893     G/A  LINC00115  Default (Recessive)          79854   \n",
       "3        1:827209     G/C  LINC00115  Default (Recessive)          79854   \n",
       "4        1:827212     C/G  LINC00115  Default (Recessive)          79854   \n",
       "...           ...     ...        ...                  ...            ...   \n",
       "96894  Y:11154005     C/T        NaN                  NaN            NaN   \n",
       "96895  Y:11986362     A/C        NaN                  NaN            NaN   \n",
       "96896  Y:11986608     T/C        NaN                  NaN            NaN   \n",
       "96897  Y:11986732     C/T        NaN                  NaN            NaN   \n",
       "96898  Y:11986765     G/T        NaN                  NaN            NaN   \n",
       "\n",
       "      Transcript Name Transcript Strand                HGVS cDot  \\\n",
       "0      NM_001005484.2                 +  NM_001005484.2:c.484A>G   \n",
       "1         NR_103536.1                 +     NR_103536.1:n.144T>C   \n",
       "2         NR_024321.1                 -     NR_024321.1:n.630C>T   \n",
       "3         NR_024321.1                 -     NR_024321.1:n.314C>G   \n",
       "4         NR_024321.1                 -     NR_024321.1:n.311G>C   \n",
       "...               ...               ...                      ...   \n",
       "96894             NaN               NaN                      NaN   \n",
       "96895             NaN               NaN                      NaN   \n",
       "96896             NaN               NaN                      NaN   \n",
       "96897             NaN               NaN                      NaN   \n",
       "96898             NaN               NaN                      NaN   \n",
       "\n",
       "                        HGVS pDot        Sequence Ontology  \\\n",
       "0      NP_001005484.2:p.Thr162Ala         missense_variant   \n",
       "1                             NaN  non_coding_exon_variant   \n",
       "2                             NaN  non_coding_exon_variant   \n",
       "3                             NaN  non_coding_exon_variant   \n",
       "4                             NaN  non_coding_exon_variant   \n",
       "...                           ...                      ...   \n",
       "96894                         NaN                      NaN   \n",
       "96895                         NaN                      NaN   \n",
       "96896                         NaN                      NaN   \n",
       "96897                         NaN                      NaN   \n",
       "96898                         NaN                      NaN   \n",
       "\n",
       "      Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0       Single Transcript Gene                      BA1,BP4   \n",
       "1       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "3       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "4       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                        ...                          ...   \n",
       "96894                      NaN                          NaN   \n",
       "96895                      NaN                          NaN   \n",
       "96896                      NaN                          NaN   \n",
       "96897                      NaN                          NaN   \n",
       "96898                      NaN                          NaN   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0      OR4F5:Allele frequency is above 0.01 recessive...         Benign   \n",
       "1      FAM87B:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "2      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "3      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "4      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "...                                                  ...            ...   \n",
       "96894                                                NaN            NaN   \n",
       "96895                                                NaN            NaN   \n",
       "96896                                                NaN            NaN   \n",
       "96897                                                NaN            NaN   \n",
       "96898                                                NaN            NaN   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                  Benign                           NaN   \n",
       "1                  Benign                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                  Benign                           NaN   \n",
       "4                  Benign                           NaN   \n",
       "...                   ...                           ...   \n",
       "96894                 NaN                           NaN   \n",
       "96895                 NaN                           NaN   \n",
       "96896                 NaN                           NaN   \n",
       "96897                 NaN                           NaN   \n",
       "96898                 NaN                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "96894                     NaN                      NaN   \n",
       "96895                     NaN                      NaN   \n",
       "96896                     NaN                      NaN   \n",
       "96897                     NaN                      NaN   \n",
       "96898                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name   Max MAF Max Allele Number  \\\n",
       "0            Annotated gnomAD East Asian  0.999461             16698   \n",
       "1                      Annotated 1kG All  0.753195              5008   \n",
       "2            Annotated gnomAD East Asian  0.896992             10106   \n",
       "3            Annotated gnomAD East Asian  0.894115              7952   \n",
       "4            Annotated gnomAD East Asian  0.894327              7826   \n",
       "...                                  ...       ...               ...   \n",
       "96894                                NaN       NaN               NaN   \n",
       "96895                                NaN       NaN               NaN   \n",
       "96896                                NaN       NaN               NaN   \n",
       "96897                                NaN       NaN               NaN   \n",
       "96898                                NaN       NaN               NaN   \n",
       "\n",
       "      Max Allele Count CHROM       POS REF ALT  \n",
       "0                16689  chr1     69511   A   G  \n",
       "1                 3772  chr1    817514   T   C  \n",
       "2                 9065  chr1    826893   G   A  \n",
       "3                 7110  chr1    827209   G   C  \n",
       "4                 6999  chr1    827212   C   G  \n",
       "...                ...   ...       ...  ..  ..  \n",
       "96894              NaN  chrY  11154005   C   T  \n",
       "96895              NaN  chrY  11986362   A   C  \n",
       "96896              NaN  chrY  11986608   T   C  \n",
       "96897              NaN  chrY  11986732   C   T  \n",
       "96898              NaN  chrY  11986765   G   T  \n",
       "\n",
       "[96899 rows x 26 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/MODY_new_05_10_2023/12652705.xlsx')\n",
    "df['CHROM'] = 'chr' + df['Chr:Pos'].str.split(':').str[0]\n",
    "df['POS'] = df['Chr:Pos'].str.split(':').str[1].astype(int)\n",
    "df['REF'] = df['Ref/Alt'].str.split('/').str[0]\n",
    "df['ALT'] = df['Ref/Alt'].str.split('/').str[1]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "9c6c2b1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,1|0,40;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:1,40:0.955:41:1,21:0,17:0,1,0,40</td>\n",
       "      <td>41</td>\n",
       "      <td>1,40</td>\n",
       "      <td>1</td>\n",
       "      <td>40</td>\n",
       "      <td>0.975610</td>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,9;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,57:0.983:57:0,24:0,32:0,0,48,9</td>\n",
       "      <td>57</td>\n",
       "      <td>0,57</td>\n",
       "      <td>0</td>\n",
       "      <td>57</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:930939</td>\n",
       "      <td>G/A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.792-100G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.953474</td>\n",
       "      <td>5008</td>\n",
       "      <td>4775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|20,3;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,23:0.961:23:0,14:0,8:0,0,20,3</td>\n",
       "      <td>23</td>\n",
       "      <td>0,23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:941119</td>\n",
       "      <td>A/G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.1196-25A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD European Finnish</td>\n",
       "      <td>0.960537</td>\n",
       "      <td>16902</td>\n",
       "      <td>16235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,0;C...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,1:0.667:1:0,1:0,0:0,0,1,0</td>\n",
       "      <td>1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944296</td>\n",
       "      <td>rs6605067</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,2;C...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,3:0.800:3:0,1:0,2:0,0,1,2</td>\n",
       "      <td>3</td>\n",
       "      <td>0,3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:944296</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NOC2L,SAMD11</td>\n",
       "      <td>Default (Recessive),Default (Recessive)</td>\n",
       "      <td>26155,148398</td>\n",
       "      <td>NM_015658.4,NM_001385641.1</td>\n",
       "      <td>-,+</td>\n",
       "      <td>NM_015658.4:c.*398C&gt;T,NM_001385641.1:c.*143G&gt;A</td>\n",
       "      <td>,</td>\n",
       "      <td>3_prime_UTR_variant,3_prime_UTR_variant</td>\n",
       "      <td>Single Transcript Gene,Other Transcripts Have ...</td>\n",
       "      <td>BA1,BS2,BP4,BP7,BA1,BS2,BP4,BP7</td>\n",
       "      <td>NOC2L:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>,</td>\n",
       "      <td>,</td>\n",
       "      <td>,</td>\n",
       "      <td>Annotated 1kG All,Annotated 1kG All</td>\n",
       "      <td>0.885982,0.885982</td>\n",
       "      <td>5008,5008</td>\n",
       "      <td>4437,4437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82911</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154781346</td>\n",
       "      <td>rs782421732</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=3,11|2,19...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:14,21:0.603:35:11,5:3,10:3,11,2,19</td>\n",
       "      <td>35</td>\n",
       "      <td>14,21</td>\n",
       "      <td>14</td>\n",
       "      <td>21</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>X:154781346</td>\n",
       "      <td>G/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.1150-33C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BS1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency for Annotate-gnomAD Exom...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD South Asian</td>\n",
       "      <td>0.00518672</td>\n",
       "      <td>15424</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82912</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154784234</td>\n",
       "      <td>rs5945238</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,26;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,27:0.966:27:0,10:0,17:0,0,1,26</td>\n",
       "      <td>27</td>\n",
       "      <td>0,27</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>X:154784234</td>\n",
       "      <td>C/T</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.785-126G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.161325</td>\n",
       "      <td>3775</td>\n",
       "      <td>609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82913</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154901948</td>\n",
       "      <td>rs4074307</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=13,0|5,0;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:13,5:0.300:18:4,4:9,1:13,0,5,0</td>\n",
       "      <td>18</td>\n",
       "      <td>13,5</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>0.277778</td>\n",
       "      <td>X:154901948</td>\n",
       "      <td>A/G</td>\n",
       "      <td>F8</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>2157</td>\n",
       "      <td>NM_000132.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_000132.4:c.6115+103T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7,BP6</td>\n",
       "      <td>F8:Allele frequency is above 0.01 recessive th...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.440795</td>\n",
       "      <td>3775</td>\n",
       "      <td>1664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82914</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154903815</td>\n",
       "      <td>rs4898352</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=26,0|17,0...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:26,17:0.400:43:11,7:14,10:26,0,17,0</td>\n",
       "      <td>43</td>\n",
       "      <td>26,17</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>0.395349</td>\n",
       "      <td>X:154903815</td>\n",
       "      <td>A/T</td>\n",
       "      <td>F8</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>2157</td>\n",
       "      <td>NM_000132.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_000132.4:c.5998+91T&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7,BP6</td>\n",
       "      <td>F8:Allele frequency is above 0.01 recessive th...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.441325</td>\n",
       "      <td>3775</td>\n",
       "      <td>1666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82915</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154929926</td>\n",
       "      <td>rs1800292</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=19,22|20,...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:41,38:0.487:79:15,19:26,19:19,22,20,18</td>\n",
       "      <td>79</td>\n",
       "      <td>41,38</td>\n",
       "      <td>41</td>\n",
       "      <td>38</td>\n",
       "      <td>0.481013</td>\n",
       "      <td>X:154929926</td>\n",
       "      <td>T/G</td>\n",
       "      <td>F8</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>2157</td>\n",
       "      <td>NM_000132.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_000132.4:c.3864A&gt;C</td>\n",
       "      <td>NP_000123.1:p.Ser1288=</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP4,BP7,BP6</td>\n",
       "      <td>F8:Allele frequency is above 0.01 recessive th...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD South Asian</td>\n",
       "      <td>0.336505</td>\n",
       "      <td>19126</td>\n",
       "      <td>6436</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>82915 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     817514    rs3131971   T   C    .   PASS   \n",
       "1      chr1     930939    rs9988021   G   A    .   PASS   \n",
       "2      chr1     941119    rs4372192   A   G    .   PASS   \n",
       "3      chr1     942451    rs6672356   T   C    .   PASS   \n",
       "4      chr1     944296    rs6605067   G   A    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "82911  chrX  154781346  rs782421732   G   C    .   PASS   \n",
       "82912  chrX  154784234    rs5945238   C   T    .   PASS   \n",
       "82913  chrX  154901948    rs4074307   A   G    .   PASS   \n",
       "82914  chrX  154903815    rs4898352   A   T    .   PASS   \n",
       "82915  chrX  154929926    rs1800292   T   G    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,1|0,40;...   \n",
       "1      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,9;...   \n",
       "2      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|20,3;...   \n",
       "3      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,0;C...   \n",
       "4      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,2;C...   \n",
       "...                                                  ...   \n",
       "82911  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=3,11|2,19...   \n",
       "82912  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,26;...   \n",
       "82913  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=13,0|5,0;...   \n",
       "82914  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=26,0|17,0...   \n",
       "82915  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=19,22|20,...   \n",
       "\n",
       "                         FORMAT                                      SAMPLE  \\\n",
       "0      GT:AD:AF:DP:F1R2:F2R1:SB        0/1:1,40:0.955:41:1,21:0,17:0,1,0,40   \n",
       "1      GT:AD:AF:DP:F1R2:F2R1:SB        0/1:0,57:0.983:57:0,24:0,32:0,0,48,9   \n",
       "2      GT:AD:AF:DP:F1R2:F2R1:SB         0/1:0,23:0.961:23:0,14:0,8:0,0,20,3   \n",
       "3      GT:AD:AF:DP:F1R2:F2R1:SB             0/1:0,1:0.667:1:0,1:0,0:0,0,1,0   \n",
       "4      GT:AD:AF:DP:F1R2:F2R1:SB             0/1:0,3:0.800:3:0,1:0,2:0,0,1,2   \n",
       "...                         ...                                         ...   \n",
       "82911  GT:AD:AF:DP:F1R2:F2R1:SB      0/1:14,21:0.603:35:11,5:3,10:3,11,2,19   \n",
       "82912  GT:AD:AF:DP:F1R2:F2R1:SB        0/1:0,27:0.966:27:0,10:0,17:0,0,1,26   \n",
       "82913  GT:AD:AF:DP:F1R2:F2R1:SB          0/1:13,5:0.300:18:4,4:9,1:13,0,5,0   \n",
       "82914  GT:AD:AF:DP:F1R2:F2R1:SB     0/1:26,17:0.400:43:11,7:14,10:26,0,17,0   \n",
       "82915  GT:AD:AF:DP:F1R2:F2R1:SB  0/1:41,38:0.487:79:15,19:26,19:19,22,20,18   \n",
       "\n",
       "       DP     AD  RD  A_D       VAF      Chr:Pos Ref/Alt     Gene Name  \\\n",
       "0      41   1,40   1   40  0.975610     1:817514     T/C        FAM87B   \n",
       "1      57   0,57   0   57  1.000000     1:930939     G/A        SAMD11   \n",
       "2      23   0,23   0   23  1.000000     1:941119     A/G        SAMD11   \n",
       "3       1    0,1   0    1  1.000000          NaN     NaN           NaN   \n",
       "4       3    0,3   0    3  1.000000     1:944296     G/A  NOC2L,SAMD11   \n",
       "...    ..    ...  ..  ...       ...          ...     ...           ...   \n",
       "82911  35  14,21  14   21  0.600000  X:154781346     G/C          MPP1   \n",
       "82912  27   0,27   0   27  1.000000  X:154784234     C/T          MPP1   \n",
       "82913  18   13,5  13    5  0.277778  X:154901948     A/G            F8   \n",
       "82914  43  26,17  26   17  0.395349  X:154903815     A/T            F8   \n",
       "82915  79  41,38  41   38  0.481013  X:154929926     T/G            F8   \n",
       "\n",
       "                              Gene Inheritance Entrez Gene ID  \\\n",
       "0                          Default (Recessive)         400728   \n",
       "1                          Default (Recessive)         148398   \n",
       "2                          Default (Recessive)         148398   \n",
       "3                                          NaN            NaN   \n",
       "4      Default (Recessive),Default (Recessive)   26155,148398   \n",
       "...                                        ...            ...   \n",
       "82911                      Default (Recessive)           4354   \n",
       "82912                      Default (Recessive)           4354   \n",
       "82913                                Recessive           2157   \n",
       "82914                                Recessive           2157   \n",
       "82915                                Recessive           2157   \n",
       "\n",
       "                  Transcript Name Transcript Strand  \\\n",
       "0                     NR_103536.1                 +   \n",
       "1                  NM_001385641.1                 +   \n",
       "2                  NM_001385641.1                 +   \n",
       "3                             NaN               NaN   \n",
       "4      NM_015658.4,NM_001385641.1               -,+   \n",
       "...                           ...               ...   \n",
       "82911                 NM_002436.4                 -   \n",
       "82912                 NM_002436.4                 -   \n",
       "82913                 NM_000132.4                 -   \n",
       "82914                 NM_000132.4                 -   \n",
       "82915                 NM_000132.4                 -   \n",
       "\n",
       "                                            HGVS cDot               HGVS pDot  \\\n",
       "0                                NR_103536.1:n.144T>C                     NaN   \n",
       "1                         NM_001385641.1:c.792-100G>A                     NaN   \n",
       "2                         NM_001385641.1:c.1196-25A>G                     NaN   \n",
       "3                                                 NaN                     NaN   \n",
       "4      NM_015658.4:c.*398C>T,NM_001385641.1:c.*143G>A                       ,   \n",
       "...                                               ...                     ...   \n",
       "82911                        NM_002436.4:c.1150-33C>G                     NaN   \n",
       "82912                        NM_002436.4:c.785-126G>A                     NaN   \n",
       "82913                       NM_000132.4:c.6115+103T>C                     NaN   \n",
       "82914                        NM_000132.4:c.5998+91T>A                     NaN   \n",
       "82915                           NM_000132.4:c.3864A>C  NP_000123.1:p.Ser1288=   \n",
       "\n",
       "                             Sequence Ontology  \\\n",
       "0                      non_coding_exon_variant   \n",
       "1                               intron_variant   \n",
       "2                               intron_variant   \n",
       "3                                          NaN   \n",
       "4      3_prime_UTR_variant,3_prime_UTR_variant   \n",
       "...                                        ...   \n",
       "82911                           intron_variant   \n",
       "82912                           intron_variant   \n",
       "82913                           intron_variant   \n",
       "82914                           intron_variant   \n",
       "82915                       synonymous_variant   \n",
       "\n",
       "                                Other Transcript Effects  \\\n",
       "0                                 Single Transcript Gene   \n",
       "1                     Other Transcripts Have Same Effect   \n",
       "2                     Other Transcripts Have Same Effect   \n",
       "3                                                    NaN   \n",
       "4      Single Transcript Gene,Other Transcripts Have ...   \n",
       "...                                                  ...   \n",
       "82911                 Other Transcripts Have Same Effect   \n",
       "82912                 Other Transcripts Have Same Effect   \n",
       "82913                             Single Transcript Gene   \n",
       "82914                             Single Transcript Gene   \n",
       "82915                             Single Transcript Gene   \n",
       "\n",
       "          ACMG Classification Criteria  \\\n",
       "0                          BA1,BS2,BP7   \n",
       "1                          BA1,BS2,BP7   \n",
       "2                          BA1,BS2,BP7   \n",
       "3                                  NaN   \n",
       "4      BA1,BS2,BP4,BP7,BA1,BS2,BP4,BP7   \n",
       "...                                ...   \n",
       "82911                      BS1,BS2,BP7   \n",
       "82912                      BA1,BS2,BP7   \n",
       "82913                  BA1,BS2,BP7,BP6   \n",
       "82914                  BA1,BS2,BP7,BP6   \n",
       "82915              BA1,BS2,BP4,BP7,BP6   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0      FAM87B:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "1      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "2      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "3                                                    NaN            NaN   \n",
       "4      NOC2L:Allele frequency is above 0.01 recessive...  Benign,Benign   \n",
       "...                                                  ...            ...   \n",
       "82911  MPP1:Allele frequency for Annotate-gnomAD Exom...         Benign   \n",
       "82912  MPP1:Allele frequency is above 0.01 recessive ...         Benign   \n",
       "82913  F8:Allele frequency is above 0.01 recessive th...         Benign   \n",
       "82914  F8:Allele frequency is above 0.01 recessive th...         Benign   \n",
       "82915  F8:Allele frequency is above 0.01 recessive th...         Benign   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                  Benign                           NaN   \n",
       "1                  Benign                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                     NaN                           NaN   \n",
       "4           Benign,Benign                             ,   \n",
       "...                   ...                           ...   \n",
       "82911              Benign                           NaN   \n",
       "82912              Benign                           NaN   \n",
       "82913              Benign                           NaN   \n",
       "82914              Benign                           NaN   \n",
       "82915              Benign                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                           ,                        ,   \n",
       "...                       ...                      ...   \n",
       "82911                     NaN                      NaN   \n",
       "82912                     NaN                      NaN   \n",
       "82913                     NaN                      NaN   \n",
       "82914                     NaN                      NaN   \n",
       "82915                     NaN                      NaN   \n",
       "\n",
       "        Max Sub Population Freq Group Name            Max MAF  \\\n",
       "0                        Annotated 1kG All           0.753195   \n",
       "1                        Annotated 1kG All           0.953474   \n",
       "2        Annotated gnomAD European Finnish           0.960537   \n",
       "3                                      NaN                NaN   \n",
       "4      Annotated 1kG All,Annotated 1kG All  0.885982,0.885982   \n",
       "...                                    ...                ...   \n",
       "82911         Annotated gnomAD South Asian         0.00518672   \n",
       "82912                    Annotated 1kG All           0.161325   \n",
       "82913                    Annotated 1kG All           0.440795   \n",
       "82914                    Annotated 1kG All           0.441325   \n",
       "82915         Annotated gnomAD South Asian           0.336505   \n",
       "\n",
       "      Max Allele Number Max Allele Count  \n",
       "0                  5008             3772  \n",
       "1                  5008             4775  \n",
       "2                 16902            16235  \n",
       "3                   NaN              NaN  \n",
       "4             5008,5008        4437,4437  \n",
       "...                 ...              ...  \n",
       "82911             15424               80  \n",
       "82912              3775              609  \n",
       "82913              3775             1664  \n",
       "82914              3775             1666  \n",
       "82915             19126             6436  \n",
       "\n",
       "[82915 rows x 37 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filter_var_DP['POS'] = filter_var_DP['POS'].astype(int)\n",
    "merged = pd.merge(filter_var_DP, df, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "merged = merged.drop_duplicates(subset = ['CHROM', 'POS', 'REF', 'ALT'])\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "ee7ce2fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>AD</th>\n",
       "      <th>RD</th>\n",
       "      <th>A_D</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,1|0,40;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:1,40:0.955:41:1,21:0,17:0,1,0,40</td>\n",
       "      <td>41</td>\n",
       "      <td>1,40</td>\n",
       "      <td>1</td>\n",
       "      <td>40</td>\n",
       "      <td>0.975610</td>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930939</td>\n",
       "      <td>rs9988021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,9;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,57:0.983:57:0,24:0,32:0,0,48,9</td>\n",
       "      <td>57</td>\n",
       "      <td>0,57</td>\n",
       "      <td>0</td>\n",
       "      <td>57</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:930939</td>\n",
       "      <td>G/A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.792-100G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.953474</td>\n",
       "      <td>5008</td>\n",
       "      <td>4775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941119</td>\n",
       "      <td>rs4372192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|20,3;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,23:0.961:23:0,14:0,8:0,0,20,3</td>\n",
       "      <td>23</td>\n",
       "      <td>0,23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:941119</td>\n",
       "      <td>A/G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001385641.1:c.1196-25A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>SAMD11:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD European Finnish</td>\n",
       "      <td>0.960537</td>\n",
       "      <td>16902</td>\n",
       "      <td>16235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,0;C...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,1:0.667:1:0,1:0,0:0,0,1,0</td>\n",
       "      <td>1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944296</td>\n",
       "      <td>rs6605067</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,2;C...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,3:0.800:3:0,1:0,2:0,0,1,2</td>\n",
       "      <td>3</td>\n",
       "      <td>0,3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1:944296</td>\n",
       "      <td>G/A</td>\n",
       "      <td>NOC2L,SAMD11</td>\n",
       "      <td>Default (Recessive),Default (Recessive)</td>\n",
       "      <td>26155,148398</td>\n",
       "      <td>NM_015658.4,NM_001385641.1</td>\n",
       "      <td>-,+</td>\n",
       "      <td>NM_015658.4:c.*398C&gt;T,NM_001385641.1:c.*143G&gt;A</td>\n",
       "      <td>,</td>\n",
       "      <td>3_prime_UTR_variant,3_prime_UTR_variant</td>\n",
       "      <td>Single Transcript Gene,Other Transcripts Have ...</td>\n",
       "      <td>BA1,BS2,BP4,BP7,BA1,BS2,BP4,BP7</td>\n",
       "      <td>NOC2L:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign,Benign</td>\n",
       "      <td>,</td>\n",
       "      <td>,</td>\n",
       "      <td>,</td>\n",
       "      <td>Annotated 1kG All,Annotated 1kG All</td>\n",
       "      <td>0.885982,0.885982</td>\n",
       "      <td>5008,5008</td>\n",
       "      <td>4437,4437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82911</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154781346</td>\n",
       "      <td>rs782421732</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=3,11|2,19...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:14,21:0.603:35:11,5:3,10:3,11,2,19</td>\n",
       "      <td>35</td>\n",
       "      <td>14,21</td>\n",
       "      <td>14</td>\n",
       "      <td>21</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>X:154781346</td>\n",
       "      <td>G/C</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.1150-33C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BS1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency for Annotate-gnomAD Exom...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD South Asian</td>\n",
       "      <td>0.00518672</td>\n",
       "      <td>15424</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82912</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154784234</td>\n",
       "      <td>rs5945238</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,26;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:0,27:0.966:27:0,10:0,17:0,0,1,26</td>\n",
       "      <td>27</td>\n",
       "      <td>0,27</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>X:154784234</td>\n",
       "      <td>C/T</td>\n",
       "      <td>MPP1</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>4354</td>\n",
       "      <td>NM_002436.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_002436.4:c.785-126G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Other Transcripts Have Same Effect</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>MPP1:Allele frequency is above 0.01 recessive ...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.161325</td>\n",
       "      <td>3775</td>\n",
       "      <td>609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82913</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154901948</td>\n",
       "      <td>rs4074307</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=13,0|5,0;...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:13,5:0.300:18:4,4:9,1:13,0,5,0</td>\n",
       "      <td>18</td>\n",
       "      <td>13,5</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>0.277778</td>\n",
       "      <td>X:154901948</td>\n",
       "      <td>A/G</td>\n",
       "      <td>F8</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>2157</td>\n",
       "      <td>NM_000132.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_000132.4:c.6115+103T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7,BP6</td>\n",
       "      <td>F8:Allele frequency is above 0.01 recessive th...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.440795</td>\n",
       "      <td>3775</td>\n",
       "      <td>1664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82914</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154903815</td>\n",
       "      <td>rs4898352</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=26,0|17,0...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:26,17:0.400:43:11,7:14,10:26,0,17,0</td>\n",
       "      <td>43</td>\n",
       "      <td>26,17</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>0.395349</td>\n",
       "      <td>X:154903815</td>\n",
       "      <td>A/T</td>\n",
       "      <td>F8</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>2157</td>\n",
       "      <td>NM_000132.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_000132.4:c.5998+91T&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7,BP6</td>\n",
       "      <td>F8:Allele frequency is above 0.01 recessive th...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.441325</td>\n",
       "      <td>3775</td>\n",
       "      <td>1666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82915</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154929926</td>\n",
       "      <td>rs1800292</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ASP;AS_FilterStatus=SITE;AS_SB_TABLE=19,22|20,...</td>\n",
       "      <td>GT:AD:AF:DP:F1R2:F2R1:SB</td>\n",
       "      <td>0/1:41,38:0.487:79:15,19:26,19:19,22,20,18</td>\n",
       "      <td>79</td>\n",
       "      <td>41,38</td>\n",
       "      <td>41</td>\n",
       "      <td>38</td>\n",
       "      <td>0.481013</td>\n",
       "      <td>X:154929926</td>\n",
       "      <td>T/G</td>\n",
       "      <td>F8</td>\n",
       "      <td>Recessive</td>\n",
       "      <td>2157</td>\n",
       "      <td>NM_000132.4</td>\n",
       "      <td>-</td>\n",
       "      <td>NM_000132.4:c.3864A&gt;C</td>\n",
       "      <td>NP_000123.1:p.Ser1288=</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP4,BP7,BP6</td>\n",
       "      <td>F8:Allele frequency is above 0.01 recessive th...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD South Asian</td>\n",
       "      <td>0.336505</td>\n",
       "      <td>19126</td>\n",
       "      <td>6436</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>82915 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     817514    rs3131971   T   C    .   PASS   \n",
       "1      chr1     930939    rs9988021   G   A    .   PASS   \n",
       "2      chr1     941119    rs4372192   A   G    .   PASS   \n",
       "3      chr1     942451    rs6672356   T   C    .   PASS   \n",
       "4      chr1     944296    rs6605067   G   A    .   PASS   \n",
       "...     ...        ...          ...  ..  ..  ...    ...   \n",
       "82911  chrX  154781346  rs782421732   G   C    .   PASS   \n",
       "82912  chrX  154784234    rs5945238   C   T    .   PASS   \n",
       "82913  chrX  154901948    rs4074307   A   G    .   PASS   \n",
       "82914  chrX  154903815    rs4898352   A   T    .   PASS   \n",
       "82915  chrX  154929926    rs1800292   T   G    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,1|0,40;...   \n",
       "1      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|48,9;...   \n",
       "2      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|20,3;...   \n",
       "3      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|0,0;C...   \n",
       "4      ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,2;C...   \n",
       "...                                                  ...   \n",
       "82911  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=3,11|2,19...   \n",
       "82912  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=0,0|1,26;...   \n",
       "82913  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=13,0|5,0;...   \n",
       "82914  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=26,0|17,0...   \n",
       "82915  ASP;AS_FilterStatus=SITE;AS_SB_TABLE=19,22|20,...   \n",
       "\n",
       "                         FORMAT                                      SAMPLE  \\\n",
       "0      GT:AD:AF:DP:F1R2:F2R1:SB        0/1:1,40:0.955:41:1,21:0,17:0,1,0,40   \n",
       "1      GT:AD:AF:DP:F1R2:F2R1:SB        0/1:0,57:0.983:57:0,24:0,32:0,0,48,9   \n",
       "2      GT:AD:AF:DP:F1R2:F2R1:SB         0/1:0,23:0.961:23:0,14:0,8:0,0,20,3   \n",
       "3      GT:AD:AF:DP:F1R2:F2R1:SB             0/1:0,1:0.667:1:0,1:0,0:0,0,1,0   \n",
       "4      GT:AD:AF:DP:F1R2:F2R1:SB             0/1:0,3:0.800:3:0,1:0,2:0,0,1,2   \n",
       "...                         ...                                         ...   \n",
       "82911  GT:AD:AF:DP:F1R2:F2R1:SB      0/1:14,21:0.603:35:11,5:3,10:3,11,2,19   \n",
       "82912  GT:AD:AF:DP:F1R2:F2R1:SB        0/1:0,27:0.966:27:0,10:0,17:0,0,1,26   \n",
       "82913  GT:AD:AF:DP:F1R2:F2R1:SB          0/1:13,5:0.300:18:4,4:9,1:13,0,5,0   \n",
       "82914  GT:AD:AF:DP:F1R2:F2R1:SB     0/1:26,17:0.400:43:11,7:14,10:26,0,17,0   \n",
       "82915  GT:AD:AF:DP:F1R2:F2R1:SB  0/1:41,38:0.487:79:15,19:26,19:19,22,20,18   \n",
       "\n",
       "       DP     AD  RD  A_D       VAF      Chr:Pos Ref/Alt     Gene Name  \\\n",
       "0      41   1,40   1   40  0.975610     1:817514     T/C        FAM87B   \n",
       "1      57   0,57   0   57  1.000000     1:930939     G/A        SAMD11   \n",
       "2      23   0,23   0   23  1.000000     1:941119     A/G        SAMD11   \n",
       "3       1    0,1   0    1  1.000000          NaN     NaN           NaN   \n",
       "4       3    0,3   0    3  1.000000     1:944296     G/A  NOC2L,SAMD11   \n",
       "...    ..    ...  ..  ...       ...          ...     ...           ...   \n",
       "82911  35  14,21  14   21  0.600000  X:154781346     G/C          MPP1   \n",
       "82912  27   0,27   0   27  1.000000  X:154784234     C/T          MPP1   \n",
       "82913  18   13,5  13    5  0.277778  X:154901948     A/G            F8   \n",
       "82914  43  26,17  26   17  0.395349  X:154903815     A/T            F8   \n",
       "82915  79  41,38  41   38  0.481013  X:154929926     T/G            F8   \n",
       "\n",
       "                              Gene Inheritance Entrez Gene ID  \\\n",
       "0                          Default (Recessive)         400728   \n",
       "1                          Default (Recessive)         148398   \n",
       "2                          Default (Recessive)         148398   \n",
       "3                                          NaN            NaN   \n",
       "4      Default (Recessive),Default (Recessive)   26155,148398   \n",
       "...                                        ...            ...   \n",
       "82911                      Default (Recessive)           4354   \n",
       "82912                      Default (Recessive)           4354   \n",
       "82913                                Recessive           2157   \n",
       "82914                                Recessive           2157   \n",
       "82915                                Recessive           2157   \n",
       "\n",
       "                  Transcript Name Transcript Strand  \\\n",
       "0                     NR_103536.1                 +   \n",
       "1                  NM_001385641.1                 +   \n",
       "2                  NM_001385641.1                 +   \n",
       "3                             NaN               NaN   \n",
       "4      NM_015658.4,NM_001385641.1               -,+   \n",
       "...                           ...               ...   \n",
       "82911                 NM_002436.4                 -   \n",
       "82912                 NM_002436.4                 -   \n",
       "82913                 NM_000132.4                 -   \n",
       "82914                 NM_000132.4                 -   \n",
       "82915                 NM_000132.4                 -   \n",
       "\n",
       "                                            HGVS cDot               HGVS pDot  \\\n",
       "0                                NR_103536.1:n.144T>C                     NaN   \n",
       "1                         NM_001385641.1:c.792-100G>A                     NaN   \n",
       "2                         NM_001385641.1:c.1196-25A>G                     NaN   \n",
       "3                                                 NaN                     NaN   \n",
       "4      NM_015658.4:c.*398C>T,NM_001385641.1:c.*143G>A                       ,   \n",
       "...                                               ...                     ...   \n",
       "82911                        NM_002436.4:c.1150-33C>G                     NaN   \n",
       "82912                        NM_002436.4:c.785-126G>A                     NaN   \n",
       "82913                       NM_000132.4:c.6115+103T>C                     NaN   \n",
       "82914                        NM_000132.4:c.5998+91T>A                     NaN   \n",
       "82915                           NM_000132.4:c.3864A>C  NP_000123.1:p.Ser1288=   \n",
       "\n",
       "                             Sequence Ontology  \\\n",
       "0                      non_coding_exon_variant   \n",
       "1                               intron_variant   \n",
       "2                               intron_variant   \n",
       "3                                          NaN   \n",
       "4      3_prime_UTR_variant,3_prime_UTR_variant   \n",
       "...                                        ...   \n",
       "82911                           intron_variant   \n",
       "82912                           intron_variant   \n",
       "82913                           intron_variant   \n",
       "82914                           intron_variant   \n",
       "82915                       synonymous_variant   \n",
       "\n",
       "                                Other Transcript Effects  \\\n",
       "0                                 Single Transcript Gene   \n",
       "1                     Other Transcripts Have Same Effect   \n",
       "2                     Other Transcripts Have Same Effect   \n",
       "3                                                    NaN   \n",
       "4      Single Transcript Gene,Other Transcripts Have ...   \n",
       "...                                                  ...   \n",
       "82911                 Other Transcripts Have Same Effect   \n",
       "82912                 Other Transcripts Have Same Effect   \n",
       "82913                             Single Transcript Gene   \n",
       "82914                             Single Transcript Gene   \n",
       "82915                             Single Transcript Gene   \n",
       "\n",
       "          ACMG Classification Criteria  \\\n",
       "0                          BA1,BS2,BP7   \n",
       "1                          BA1,BS2,BP7   \n",
       "2                          BA1,BS2,BP7   \n",
       "3                                  NaN   \n",
       "4      BA1,BS2,BP4,BP7,BA1,BS2,BP4,BP7   \n",
       "...                                ...   \n",
       "82911                      BS1,BS2,BP7   \n",
       "82912                      BA1,BS2,BP7   \n",
       "82913                  BA1,BS2,BP7,BP6   \n",
       "82914                  BA1,BS2,BP7,BP6   \n",
       "82915              BA1,BS2,BP4,BP7,BP6   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0      FAM87B:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "1      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "2      SAMD11:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "3                                                    NaN            NaN   \n",
       "4      NOC2L:Allele frequency is above 0.01 recessive...         Benign   \n",
       "...                                                  ...            ...   \n",
       "82911  MPP1:Allele frequency for Annotate-gnomAD Exom...         Benign   \n",
       "82912  MPP1:Allele frequency is above 0.01 recessive ...         Benign   \n",
       "82913  F8:Allele frequency is above 0.01 recessive th...         Benign   \n",
       "82914  F8:Allele frequency is above 0.01 recessive th...         Benign   \n",
       "82915  F8:Allele frequency is above 0.01 recessive th...         Benign   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                  Benign                           NaN   \n",
       "1                  Benign                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                     NaN                           NaN   \n",
       "4           Benign,Benign                             ,   \n",
       "...                   ...                           ...   \n",
       "82911              Benign                           NaN   \n",
       "82912              Benign                           NaN   \n",
       "82913              Benign                           NaN   \n",
       "82914              Benign                           NaN   \n",
       "82915              Benign                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                           ,                        ,   \n",
       "...                       ...                      ...   \n",
       "82911                     NaN                      NaN   \n",
       "82912                     NaN                      NaN   \n",
       "82913                     NaN                      NaN   \n",
       "82914                     NaN                      NaN   \n",
       "82915                     NaN                      NaN   \n",
       "\n",
       "        Max Sub Population Freq Group Name            Max MAF  \\\n",
       "0                        Annotated 1kG All           0.753195   \n",
       "1                        Annotated 1kG All           0.953474   \n",
       "2        Annotated gnomAD European Finnish           0.960537   \n",
       "3                                      NaN                NaN   \n",
       "4      Annotated 1kG All,Annotated 1kG All  0.885982,0.885982   \n",
       "...                                    ...                ...   \n",
       "82911         Annotated gnomAD South Asian         0.00518672   \n",
       "82912                    Annotated 1kG All           0.161325   \n",
       "82913                    Annotated 1kG All           0.440795   \n",
       "82914                    Annotated 1kG All           0.441325   \n",
       "82915         Annotated gnomAD South Asian           0.336505   \n",
       "\n",
       "      Max Allele Number Max Allele Count  \n",
       "0                  5008             3772  \n",
       "1                  5008             4775  \n",
       "2                 16902            16235  \n",
       "3                   NaN              NaN  \n",
       "4             5008,5008        4437,4437  \n",
       "...                 ...              ...  \n",
       "82911             15424               80  \n",
       "82912              3775              609  \n",
       "82913              3775             1664  \n",
       "82914              3775             1666  \n",
       "82915             19126             6436  \n",
       "\n",
       "[82915 rows x 37 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged['Classification'] = merged['Classification'].str.split(',').str[0]\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "ccd49b79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tool</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Benign</td>\n",
       "      <td>68003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>11725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS/Conflicting</td>\n",
       "      <td>1280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>1156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VUS/Weak Benign</td>\n",
       "      <td>374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>VUS/Weak Pathogenic</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>VUS</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Tool  Counts\n",
       "0               Benign   68003\n",
       "1                  NaN   11725\n",
       "2      VUS/Conflicting    1280\n",
       "3        Likely Benign    1156\n",
       "4      VUS/Weak Benign     374\n",
       "5  VUS/Weak Pathogenic     311\n",
       "6                  VUS      58\n",
       "7    Likely Pathogenic       8"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "value_counts_with_null_bcftools = merged['Classification'].value_counts(dropna=False)\n",
    "value_counts_with_null_bcftools\n",
    "#result_df = pd.DataFrame({'Counts': value_counts_with_null_bcftools})\n",
    "result_df = pd.DataFrame({'Tool': value_counts_with_null_bcftools.index, 'Counts': value_counts_with_null_bcftools.values})\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "fc7981d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/new_output_files/ACMG_outputs/ACMG_MUTECT2.xlsx', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61aa64b4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48a1f2ff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3d33056e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...</td>\n",
       "      <td>0</td>\n",
       "      <td>210</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...</td>\n",
       "      <td>1</td>\n",
       "      <td>367</td>\n",
       "      <td>0.997283</td>\n",
       "      <td>368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...</td>\n",
       "      <td>15</td>\n",
       "      <td>110</td>\n",
       "      <td>0.880000</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...</td>\n",
       "      <td>15</td>\n",
       "      <td>113</td>\n",
       "      <td>0.882812</td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69181</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57084531</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...</td>\n",
       "      <td>37</td>\n",
       "      <td>25</td>\n",
       "      <td>0.403226</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69182</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57128393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...</td>\n",
       "      <td>112</td>\n",
       "      <td>111</td>\n",
       "      <td>0.497758</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69183</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184462</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>0.465116</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69184</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57189953</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9</td>\n",
       "      <td>30</td>\n",
       "      <td>20</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69185</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191772</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...</td>\n",
       "      <td>33</td>\n",
       "      <td>43</td>\n",
       "      <td>0.565789</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67996 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69511  rs2691305   A   G    .   PASS   \n",
       "1      chr1    817514  rs3131971   T   C    .   PASS   \n",
       "2      chr1    826893  rs3115849   G   A    .   PASS   \n",
       "3      chr1    827209  rs3115848   G   C    .   PASS   \n",
       "4      chr1    827212  rs3131950   C   G    .   PASS   \n",
       "...     ...       ...        ...  ..  ..  ...    ...   \n",
       "69181  chrY  57084531          .   A   G    .   PASS   \n",
       "69182  chrY  57128393          .   G   C    .   PASS   \n",
       "69183  chrY  57184462          .   T   G    .   PASS   \n",
       "69184  chrY  57189953          .   T   C    .   PASS   \n",
       "69185  chrY  57191772          .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "1      ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...   \n",
       "2      ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....   \n",
       "3      ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "4      ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "...                                                  ...   \n",
       "69181  ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69182  ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...   \n",
       "69183  ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "69184  ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...   \n",
       "69185  ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "69181  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69182  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69183  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69184  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "69185  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   RD   AD       VAF  \\\n",
       "0      1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...    0  210  1.000000   \n",
       "1      1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...    1  367  0.997283   \n",
       "2      1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55    0   91  1.000000   \n",
       "3      1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...   15  110  0.880000   \n",
       "4      1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...   15  113  0.882812   \n",
       "...                                                  ...  ...  ...       ...   \n",
       "69181  0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...   37   25  0.403226   \n",
       "69182  0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...  112  111  0.497758   \n",
       "69183  0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...   23   20  0.465116   \n",
       "69184   0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9   30   20  0.400000   \n",
       "69185  0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...   33   43  0.565789   \n",
       "\n",
       "        DP  \n",
       "0      211  \n",
       "1      368  \n",
       "2       91  \n",
       "3      125  \n",
       "4      128  \n",
       "...    ...  \n",
       "69181   62  \n",
       "69182  223  \n",
       "69183   43  \n",
       "69184   50  \n",
       "69185   76  \n",
       "\n",
       "[67996 rows x 14 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/12652705_VARSCAN2.vcf', comment='#', sep='\\t', header=None, low_memory=False, encoding='latin-1')\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf['RD'] = vcf['SAMPLE'].str.split(':').str[4].fillna('0').astype(int)\n",
    "vcf['AD'] = vcf['SAMPLE'].str.split(':').str[5].fillna('0').astype(int)\n",
    "vcf['VAF'] = vcf['AD'] / (vcf['RD'] + vcf['AD'])\n",
    "vcf['DP'] = vcf['SAMPLE'].str.split(':').str[3].fillna('0').astype(int)\n",
    "filter_var = vcf[vcf['VAF'] >= 0.1]\n",
    "filter_var_DP_var = filter_var[filter_var['DP'] >= 10]\n",
    "filter_var_DP_var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "910e842a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>VAF</th>\n",
       "      <th>DP</th>\n",
       "      <th>Chr:Pos</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Inheritance</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>Transcript Name</th>\n",
       "      <th>Transcript Strand</th>\n",
       "      <th>HGVS cDot</th>\n",
       "      <th>HGVS pDot</th>\n",
       "      <th>Sequence Ontology</th>\n",
       "      <th>Other Transcript Effects</th>\n",
       "      <th>ACMG Classification Criteria</th>\n",
       "      <th>ACMG Classification Criteria Description</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Auto Classification</th>\n",
       "      <th>Previous Classification Count</th>\n",
       "      <th>Previous Classification</th>\n",
       "      <th>Last Classification Date</th>\n",
       "      <th>Max Sub Population Freq Group Name</th>\n",
       "      <th>Max MAF</th>\n",
       "      <th>Max Allele Number</th>\n",
       "      <th>Max Allele Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...</td>\n",
       "      <td>0</td>\n",
       "      <td>210</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>211</td>\n",
       "      <td>1:69511</td>\n",
       "      <td>A/G</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79501</td>\n",
       "      <td>NM_001005484.2</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_001005484.2:c.484A&gt;G</td>\n",
       "      <td>NP_001005484.2:p.Thr162Ala</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BP4</td>\n",
       "      <td>OR4F5:Allele frequency is above 0.01 recessive...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.999461</td>\n",
       "      <td>16698</td>\n",
       "      <td>16689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>817514</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...</td>\n",
       "      <td>1</td>\n",
       "      <td>367</td>\n",
       "      <td>0.997283</td>\n",
       "      <td>368</td>\n",
       "      <td>1:817514</td>\n",
       "      <td>T/C</td>\n",
       "      <td>FAM87B</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>400728</td>\n",
       "      <td>NR_103536.1</td>\n",
       "      <td>+</td>\n",
       "      <td>NR_103536.1:n.144T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>FAM87B:Allele frequency is above 0.01 recessiv...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated 1kG All</td>\n",
       "      <td>0.753195</td>\n",
       "      <td>5008</td>\n",
       "      <td>3772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826893</td>\n",
       "      <td>rs3115849</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>91</td>\n",
       "      <td>1:826893</td>\n",
       "      <td>G/A</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.630C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.896992</td>\n",
       "      <td>10106</td>\n",
       "      <td>9065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827209</td>\n",
       "      <td>rs3115848</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...</td>\n",
       "      <td>15</td>\n",
       "      <td>110</td>\n",
       "      <td>0.880000</td>\n",
       "      <td>125</td>\n",
       "      <td>1:827209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.314C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894115</td>\n",
       "      <td>7952</td>\n",
       "      <td>7110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827212</td>\n",
       "      <td>rs3131950</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...</td>\n",
       "      <td>15</td>\n",
       "      <td>113</td>\n",
       "      <td>0.882812</td>\n",
       "      <td>128</td>\n",
       "      <td>1:827212</td>\n",
       "      <td>C/G</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>Default (Recessive)</td>\n",
       "      <td>79854</td>\n",
       "      <td>NR_024321.1</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024321.1:n.311G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding_exon_variant</td>\n",
       "      <td>Single Transcript Gene</td>\n",
       "      <td>BA1,BS2,BP7</td>\n",
       "      <td>LINC00115:Allele frequency is above 0.01 reces...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotated gnomAD East Asian</td>\n",
       "      <td>0.894327</td>\n",
       "      <td>7826</td>\n",
       "      <td>6999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68002</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57084531</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...</td>\n",
       "      <td>37</td>\n",
       "      <td>25</td>\n",
       "      <td>0.403226</td>\n",
       "      <td>62</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68003</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57128393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...</td>\n",
       "      <td>112</td>\n",
       "      <td>111</td>\n",
       "      <td>0.497758</td>\n",
       "      <td>223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68004</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184462</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>0.465116</td>\n",
       "      <td>43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68005</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57189953</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9</td>\n",
       "      <td>30</td>\n",
       "      <td>20</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68006</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57191772</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...</td>\n",
       "      <td>33</td>\n",
       "      <td>43</td>\n",
       "      <td>0.565789</td>\n",
       "      <td>76</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67996 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69511  rs2691305   A   G    .   PASS   \n",
       "1      chr1    817514  rs3131971   T   C    .   PASS   \n",
       "2      chr1    826893  rs3115849   G   A    .   PASS   \n",
       "3      chr1    827209  rs3115848   G   C    .   PASS   \n",
       "4      chr1    827212  rs3131950   C   G    .   PASS   \n",
       "...     ...       ...        ...  ..  ..  ...    ...   \n",
       "68002  chrY  57084531          .   A   G    .   PASS   \n",
       "68003  chrY  57128393          .   G   C    .   PASS   \n",
       "68004  chrY  57184462          .   T   G    .   PASS   \n",
       "68005  chrY  57189953          .   T   C    .   PASS   \n",
       "68006  chrY  57191772          .   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=211;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "1      ADP=368;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2468,0...   \n",
       "2      ADP=91;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2682,0....   \n",
       "3      ADP=125;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "4      ADP=128;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2484,0...   \n",
       "...                                                  ...   \n",
       "68002  ADP=62;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "68003  ADP=223;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_pol...   \n",
       "68004  ADP=43;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...   \n",
       "68005  ADP=50;WT=0;HET=1;HOM=0;NC=0;CSQ=C|splice_poly...   \n",
       "68006  ADP=76;WT=0;HET=1;HOM=0;NC=0;CSQ=T|intron_vari...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "68002  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "68003  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "68004  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "68005  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "68006  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE   RD   AD       VAF  \\\n",
       "0      1/1:255:212:211:0:210:99.53%:9.4917E-126:0:39:...    0  210  1.000000   \n",
       "1      1/1:255:369:368:1:367:99.73%:3.4721E-218:25:38...    1  367  0.997283   \n",
       "2      1/1:255:93:91:0:91:100%:2.7621E-54:0:36:0:0:36:55    0   91  1.000000   \n",
       "3      1/1:255:127:125:15:110:88%:5.9951E-55:36:38:0:...   15  110  0.880000   \n",
       "4      1/1:255:129:128:15:113:88.28%:1.325E-56:36:38:...   15  113  0.882812   \n",
       "...                                                  ...  ...  ...       ...   \n",
       "68002  0/1:88:62:62:37:25:40.32%:1.4168E-9:38:40:36:1...   37   25  0.403226   \n",
       "68003  0/1:255:223:223:112:111:49.78%:3.4158E-42:38:4...  112  111  0.497758   \n",
       "68004  0/1:72:43:43:23:20:46.51%:5.2508E-8:38:37:0:23...   23   20  0.465116   \n",
       "68005   0/1:70:50:50:30:20:40%:8.793E-8:37:37:15:15:11:9   30   20  0.400000   \n",
       "68006  0/1:166:78:76:33:43:56.58%:2.3908E-17:40:39:24...   33   43  0.565789   \n",
       "\n",
       "        DP   Chr:Pos Ref/Alt  Gene Name     Gene Inheritance Entrez Gene ID  \\\n",
       "0      211   1:69511     A/G      OR4F5  Default (Recessive)          79501   \n",
       "1      368  1:817514     T/C     FAM87B  Default (Recessive)         400728   \n",
       "2       91  1:826893     G/A  LINC00115  Default (Recessive)          79854   \n",
       "3      125  1:827209     G/C  LINC00115  Default (Recessive)          79854   \n",
       "4      128  1:827212     C/G  LINC00115  Default (Recessive)          79854   \n",
       "...    ...       ...     ...        ...                  ...            ...   \n",
       "68002   62       NaN     NaN        NaN                  NaN            NaN   \n",
       "68003  223       NaN     NaN        NaN                  NaN            NaN   \n",
       "68004   43       NaN     NaN        NaN                  NaN            NaN   \n",
       "68005   50       NaN     NaN        NaN                  NaN            NaN   \n",
       "68006   76       NaN     NaN        NaN                  NaN            NaN   \n",
       "\n",
       "      Transcript Name Transcript Strand                HGVS cDot  \\\n",
       "0      NM_001005484.2                 +  NM_001005484.2:c.484A>G   \n",
       "1         NR_103536.1                 +     NR_103536.1:n.144T>C   \n",
       "2         NR_024321.1                 -     NR_024321.1:n.630C>T   \n",
       "3         NR_024321.1                 -     NR_024321.1:n.314C>G   \n",
       "4         NR_024321.1                 -     NR_024321.1:n.311G>C   \n",
       "...               ...               ...                      ...   \n",
       "68002             NaN               NaN                      NaN   \n",
       "68003             NaN               NaN                      NaN   \n",
       "68004             NaN               NaN                      NaN   \n",
       "68005             NaN               NaN                      NaN   \n",
       "68006             NaN               NaN                      NaN   \n",
       "\n",
       "                        HGVS pDot        Sequence Ontology  \\\n",
       "0      NP_001005484.2:p.Thr162Ala         missense_variant   \n",
       "1                             NaN  non_coding_exon_variant   \n",
       "2                             NaN  non_coding_exon_variant   \n",
       "3                             NaN  non_coding_exon_variant   \n",
       "4                             NaN  non_coding_exon_variant   \n",
       "...                           ...                      ...   \n",
       "68002                         NaN                      NaN   \n",
       "68003                         NaN                      NaN   \n",
       "68004                         NaN                      NaN   \n",
       "68005                         NaN                      NaN   \n",
       "68006                         NaN                      NaN   \n",
       "\n",
       "      Other Transcript Effects ACMG Classification Criteria  \\\n",
       "0       Single Transcript Gene                      BA1,BP4   \n",
       "1       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "2       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "3       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "4       Single Transcript Gene                  BA1,BS2,BP7   \n",
       "...                        ...                          ...   \n",
       "68002                      NaN                          NaN   \n",
       "68003                      NaN                          NaN   \n",
       "68004                      NaN                          NaN   \n",
       "68005                      NaN                          NaN   \n",
       "68006                      NaN                          NaN   \n",
       "\n",
       "                ACMG Classification Criteria Description Classification  \\\n",
       "0      OR4F5:Allele frequency is above 0.01 recessive...         Benign   \n",
       "1      FAM87B:Allele frequency is above 0.01 recessiv...         Benign   \n",
       "2      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "3      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "4      LINC00115:Allele frequency is above 0.01 reces...         Benign   \n",
       "...                                                  ...            ...   \n",
       "68002                                                NaN            NaN   \n",
       "68003                                                NaN            NaN   \n",
       "68004                                                NaN            NaN   \n",
       "68005                                                NaN            NaN   \n",
       "68006                                                NaN            NaN   \n",
       "\n",
       "      Auto Classification Previous Classification Count  \\\n",
       "0                  Benign                           NaN   \n",
       "1                  Benign                           NaN   \n",
       "2                  Benign                           NaN   \n",
       "3                  Benign                           NaN   \n",
       "4                  Benign                           NaN   \n",
       "...                   ...                           ...   \n",
       "68002                 NaN                           NaN   \n",
       "68003                 NaN                           NaN   \n",
       "68004                 NaN                           NaN   \n",
       "68005                 NaN                           NaN   \n",
       "68006                 NaN                           NaN   \n",
       "\n",
       "      Previous Classification Last Classification Date  \\\n",
       "0                         NaN                      NaN   \n",
       "1                         NaN                      NaN   \n",
       "2                         NaN                      NaN   \n",
       "3                         NaN                      NaN   \n",
       "4                         NaN                      NaN   \n",
       "...                       ...                      ...   \n",
       "68002                     NaN                      NaN   \n",
       "68003                     NaN                      NaN   \n",
       "68004                     NaN                      NaN   \n",
       "68005                     NaN                      NaN   \n",
       "68006                     NaN                      NaN   \n",
       "\n",
       "      Max Sub Population Freq Group Name   Max MAF Max Allele Number  \\\n",
       "0            Annotated gnomAD East Asian  0.999461             16698   \n",
       "1                      Annotated 1kG All  0.753195              5008   \n",
       "2            Annotated gnomAD East Asian  0.896992             10106   \n",
       "3            Annotated gnomAD East Asian  0.894115              7952   \n",
       "4            Annotated gnomAD East Asian  0.894327              7826   \n",
       "...                                  ...       ...               ...   \n",
       "68002                                NaN       NaN               NaN   \n",
       "68003                                NaN       NaN               NaN   \n",
       "68004                                NaN       NaN               NaN   \n",
       "68005                                NaN       NaN               NaN   \n",
       "68006                                NaN       NaN               NaN   \n",
       "\n",
       "      Max Allele Count  \n",
       "0                16689  \n",
       "1                 3772  \n",
       "2                 9065  \n",
       "3                 7110  \n",
       "4                 6999  \n",
       "...                ...  \n",
       "68002              NaN  \n",
       "68003              NaN  \n",
       "68004              NaN  \n",
       "68005              NaN  \n",
       "68006              NaN  \n",
       "\n",
       "[67996 rows x 36 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filter_var_DP_var['POS'] = filter_var_DP_var['POS'].astype(int)\n",
    "merged_var = pd.merge(filter_var_DP_var, df, on = ['CHROM', 'POS', 'REF', 'ALT'], how = 'left', sort = False)\n",
    "merged_var = merged_var.drop_duplicates(subset = ['CHROM', 'POS', 'REF', 'ALT'])\n",
    "merged_var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a437fb3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Benign                                                                                                                                                                       54561\n",
       "Benign,Benign                                                                                                                                                                 3212\n",
       "VUS/Conflicting                                                                                                                                                               1364\n",
       "Likely Benign                                                                                                                                                                 1122\n",
       "VUS/Weak Pathogenic                                                                                                                                                            331\n",
       "VUS/Weak Benign                                                                                                                                                                320\n",
       "Benign,Benign,Benign                                                                                                                                                            88\n",
       "Likely Benign,Likely Benign                                                                                                                                                     70\n",
       "VUS                                                                                                                                                                             54\n",
       "VUS/Conflicting,VUS/Conflicting                                                                                                                                                 39\n",
       "VUS/Weak Benign,VUS/Weak Benign                                                                                                                                                 14\n",
       "VUS/Weak Pathogenic,VUS/Conflicting                                                                                                                                              9\n",
       "Benign,Benign,Benign,Benign                                                                                                                                                      9\n",
       "VUS/Conflicting,Likely Benign                                                                                                                                                    8\n",
       "Likely Pathogenic                                                                                                                                                                7\n",
       "VUS/Conflicting,VUS/Weak Pathogenic                                                                                                                                              6\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                                                                          6\n",
       "Likely Benign,VUS/Conflicting                                                                                                                                                    6\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                                                                   6\n",
       "Benign,Likely Benign                                                                                                                                                             5\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                        5\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                                              5\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                                                            5\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                    4\n",
       "Benign,Benign,Benign,Benign,Benign                                                                                                                                               4\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                      3\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                                3\n",
       "VUS,VUS/Weak Benign                                                                                                                                                              3\n",
       "VUS/Weak Pathogenic,VUS/Weak Pathogenic                                                                                                                                          3\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                                                                                 2\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                                       2\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                             2\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                           2\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                         2\n",
       "Likely Benign,VUS/Weak Benign                                                                                                                                                    2\n",
       "VUS/Weak Benign,Likely Benign                                                                                                                                                    2\n",
       "Likely Benign,VUS/Weak Pathogenic                                                                                                                                                2\n",
       "Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,VUS/Conflicting        1\n",
       "Benign,Benign,Benign,Benign,Benign,Benign                                                                                                                                        1\n",
       "VUS,VUS                                                                                                                                                                          1\n",
       "Benign,VUS/Weak Benign                                                                                                                                                           1\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                                                                  1\n",
       "Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign,Likely Benign                                                                                              1\n",
       "Likely Benign,Likely Benign,Likely Benign,VUS/Conflicting                                                                                                                        1\n",
       "Likely Benign,Likely Benign,Likely Benign,VUS/Weak Benign                                                                                                                        1\n",
       "Likely Benign,Likely Benign,Likely Benign                                                                                                                                        1\n",
       "Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign,Benign                               1\n",
       "VUS/Conflicting,VUS/Conflicting,VUS/Conflicting                                                                                                                                  1\n",
       "Name: Classification, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_var.Classification.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "996c0a42",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "518564b9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "5a81645c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAApkAAAMWCAYAAAC6Pp3rAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAABWGklEQVR4nO3deVxV1f7/8fdhHoQTQkCY8zXS1AbsKjaoqWg5NX3NSNKumV1Ks+xafpvIfuJN07xpo7fSK5oNpreyCC21zDGSCiUr07AUUUOcAXH9/ujh+XoEVHApB3g9H4/zuJ29P3vvdfY+W9537b32cRhjjAAAAACLvKq7AQAAAKh9CJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZQA3w3XffaciQIWrevLkCAwMVGBioFi1aaNiwYfr666+ru3nnxODBg9WkSZNT1nXu3FkOh8P18vX1VZMmTTRkyBD9+uuvZ7+hx9m7d6/GjRundu3aKTQ0VP7+/mrSpIn+9re/6ZtvvnHVpaSkyOFwaNeuXeekXS+99JJmzJhx2vV//PGHBgwYoMjISDkcDt14441npV0zZsxwO3YVvU7ne3C6tmzZIofDcdr7Y8eOHXr00UfVpk0b1atXTwEBAWrRooUeeOAB/fTTT9badSZWrFihlJQU7dmzp7qbgjrOp7obAODkXn31Vd1///2KjY3VAw88oEsuuUQOh0M5OTl66623dOWVV+rnn39W8+bNq7upHqNZs2aaPXu2JKm4uFjZ2dl6+umntWjRIv3www8KCgo6623YtGmTEhISlJ+fr3vvvVdPP/206tWrpy1btuidd95RXFyc9uzZI6fTedbbcqKXXnpJERERGjx48GnVP/PMM5o/f77eeOMNNW/eXPXr1z8r7erVq5dWrlzpNi0+Pl633nqrRo0a5Zrm7+9/VrZ/KmvWrFHv3r1ljNH999+v+Ph4+fn5aePGjUpLS9Nf//pXFRQUVEvbjrdixQo9/fTTGjx4sM4777zqbg7qMEIm4MG++uorJScnq1evXnrvvffk5+fnmnfdddfpvvvu07vvvqvAwMCTrufgwYPnJFh5isDAQHXo0MH1/tprr1VAQICGDBmi5cuXKyEh4Yy3UVJSIofDIR+fsv+MlpaW6qabbtKuXbu0cuVKtW7d2jWvU6dOGjRokD755BP5+vqecTsqo6rfg+zsbDVv3lx33HGHlXYYY3T48OEy39vzzz9f559/fpn6qKgot+NZHfbu3at+/fopICBAK1as0IUXXuia17lzZw0bNkzvvfdeNbYQ8DxcLgc8WGpqqry9vfXqq6+6Bczj/c///I9iYmJc7wcPHqx69erp+++/V0JCgkJCQtS1a1dJf172TE5OVoMGDeTn56dmzZrpscceU1FRkWv5k10+dDgcSklJcb0/dpl33bp1uvnmmxUaGiqn06mBAwdq586dZZZ/++23FR8fr+DgYNWrV089evTQunXrytTNmDFDsbGx8vf3V8uWLfWf//zndHdZhY71GB4f7H7++WfdddddatGihYKCgtSgQQP16dNH33//vduyS5culcPh0KxZszRq1Cg1aNBA/v7++vnnn8vd1oIFC/T9999rzJgxbgHzeNdff32ZwLdjxw7dfvvtcjqdioqK0t/+9jcVFha61bz44ou69tprFRkZqeDgYLVp00YTJkxQSUmJW13nzp3VunVrffHFF+rYsaOCgoL0t7/9TU2aNNH69eu1bNmyU15+PvZdWLx4sXJyclz1S5culXR63yfpz+/N/fffr1deeUUtW7aUv7+/Zs6cWe42T8fy5cvVtWtXhYSEKCgoSB07dtTChQvL1GVnZ6tfv34KCwtTQECALrvssipvd/r06crLy9OECRPcAubxbr31Vrf3H3zwgeLj4xUUFKSQkBB17969TE9tRbeBHDu3jndsP86aNUstW7ZUUFCQLr30Un300Uduy/3jH/+QJDVt2rTMMfv888/VuXNnhYeHKzAwUI0aNdItt9yigwcPVnaXAKdETybgoUpLS7VkyRK1a9dOF1xwQaWWLS4uVt++fTVs2DA9+uijOnLkiA4fPqwuXbpo06ZNevrpp9W2bVt9+eWXGj9+vLKyssr9I326brrpJvXv31/33nuv1q9fryeeeEIbNmzQ6tWrXaEuNTVVjz/+uO666y49/vjjKi4u1sSJE3XNNddozZo1atWqlaQ/A+Zdd92lfv36adKkSSosLFRKSoqKiork5XX6/7/4yJEjrn2RnZ2tsWPHqlmzZurYsaOrZtu2bQoPD9c///lPnX/++frjjz80c+ZMtW/fXuvWrVNsbKzbOseMGaP4+Hi98sor8vLyUmRkZLnbzsjIkKRK37t4yy236LbbbtOQIUNcIVWS3njjDVfNpk2blJiYqKZNm8rPz0/ffvutxo0bpx9++MGtTpK2b9+ugQMHavTo0UpNTZWXl5ceeeQR3XrrrXI6nXrppZckVXz5+YILLtDKlSuVnJyswsJC1y0IrVq1qvT3acGCBfryyy/15JNPKjo6usJ9dyrLli1T9+7d1bZtW73++uvy9/fXSy+9pD59+uitt97SbbfdJknauHGjOnbsqMjISL3wwgsKDw9XWlqaBg8erB07dmj06NGV2m5GRoa8vb3Vp0+f06qfM2eO7rjjDiUkJOitt95SUVGRJkyYoM6dO+uzzz7T1VdfXenPLkkLFy7U2rVrNXbsWNWrV08TJkzQTTfdpI0bN6pZs2a6++679ccff2jq1Kl6//33Xf92tGrVSlu2bFGvXr10zTXX6I033tB5552n33//Xenp6SouLq5TVztwjhgAHikvL89IMgMGDCgz78iRI6akpMT1Onr0qGveoEGDjCTzxhtvuC3zyiuvGEnmnXfecZv+7LPPGkkmIyPDGGPM5s2bjSTz5ptvltmuJPPUU0+53j/11FNGknnwwQfd6mbPnm0kmbS0NGOMMbm5ucbHx8cMHz7crW7fvn0mOjra9O/f3xhjTGlpqYmJiTFXXHGF22fasmWL8fX1NY0bN65gb/2fTp06GUllXhdddJHJyck56bJHjhwxxcXFpkWLFm6facmSJUaSufbaa0+5fWOM6dmzp5FkDh8+fFr1x/bjhAkT3KYnJyebgIAAt31xvNLSUlNSUmL+85//GG9vb/PHH3+45h3bD5999lmZ5S655BLTqVOn02rbsXVdcsklbtNO9/tkzJ/fG6fT6da+0yXJ3Hfffa73HTp0MJGRkWbfvn2uaUeOHDGtW7c2F154oWtfDRgwwPj7+5vc3Fy39V1//fUmKCjI7Nmzxxhz8u/78S6++GITHR19Wm0+9j1u06aNKS0tdU3ft2+fiYyMNB07dnRNGzRoULnf62PfieNJMlFRUWbv3r2uaXl5ecbLy8uMHz/eNW3ixIlGktm8ebPb8u+9956RZLKysk7rcwBnisvlQA0UFxcnX19f12vSpEllam655Ra3959//rmCg4PLXNI7Nvjjs88+q3J7TrxXr3///vLx8dGSJUskSZ9++qmOHDmiO++8U0eOHHG9AgIC1KlTJ9elvI0bN2rbtm1KTEx0u1TYuHFjtx7IU2nevLnWrl2rtWvXauXKlZozZ44CAwPVtWtXtxHAR44cUWpqqlq1aiU/Pz/5+PjIz89PP/30k3Jycsqs98R9alvfvn3d3rdt21aHDx9Wfn6+a9q6devUt29fhYeHy9vbW76+vrrzzjtVWlqqH3/80W35sLAwXXfddWelrZX9Pl133XUKCws7o20eOHBAq1ev1q233qp69eq5pnt7eyspKUm//fabNm7c6Gpf165d1bBhwzLtO3jwYJnL1jYd+x4nJSW59b7Xq1dPt9xyi1atWlXly9NdunRRSEiI631UVJQiIyNP68kJl112mfz8/HTPPfdo5syZ+uWXX6rUBuB0ETIBDxUREaHAwMBy/3jMmTNHa9eu1QcffFDuskFBQQoNDXWbtnv3bkVHR5e5zysyMlI+Pj7avXt3ldsaHR3t9t7Hx0fh4eGude7YsUOSdOWVV7qFY19fX7399tuuR/ccqz9xfRVNq0hAQIDatWundu3aqUOHDrr99tv1ySefaPv27XryySdddQ899JCeeOIJ3Xjjjfrwww+1evVqrV27VpdeeqkOHTpUZr2ne9tCo0aNJEmbN28+7TZLUnh4uNv7Y5exj7UlNzdX11xzjX7//Xf961//0pdffqm1a9fqxRdfdKurbHurorLfJxttKSgokDGm3HUduy/52HZ37959WnWnq1GjRtq5c6cOHDhwytpj665o+0ePHq3yKPQTvyPSn9+T8r6vJ2revLkWL16syMhI3XfffWrevLmaN2+uf/3rX1VqC3Aq3JMJeChvb29dd911ysjI0Pbt293+YB27f3HLli3lLnviH37pzz9Oq1evljHGbX5+fr6OHDmiiIgISX8GNEllBm+c7I9yXl6eGjRo4Hp/5MgR7d692/UH8di633vvPTVu3LjC9Ryrz8vLK3cbZ+KCCy5QRESEvv32W9e0tLQ03XnnnUpNTXWr3bVrV7mPfilvv5anR48eeu2117RgwQI9+uijZ9Tu4y1YsEAHDhzQ+++/77Yfs7Kyyq0/3fZWxel+n2y2JSwsTF5eXtq+fXuZedu2bZP0f9+18PDw06o7XT169FBGRoY+/PBDDRgw4KS1x77HFW3fy8vL1asbEBBQ5lyTdNaemXrNNdfommuuUWlpqb7++mtNnTpVI0eOVFRU1Ck/F1BZ9GQCHmzMmDEqLS3VvffeW2b0cGV17dpV+/fv14IFC9ymHxu5fWwEelRUlAICAvTdd9+51f33v/+tcN3HBoQc88477+jIkSPq3LmzpD//QPv4+GjTpk2uHsYTX5IUGxurCy64QG+99ZaMMa71/frrr1qxYkWVPvcxv/32m3bt2uU24MThcJQZ9LJw4UL9/vvvZ7Stfv36qU2bNho/fryys7PLrfn0008rfcn0WFA7vs3GGE2fPr1S6zndnq+TOd3vk03BwcFq37693n//fbf2Hz16VGlpabrwwgt10UUXubb/+eefu0Ll8e0LCgqq9CORhgwZoujoaI0ePbrC78f7778v6c/vcYMGDTRnzhy37/GBAwc0b94814hzSWrSpIny8/Ndvf3Sn4PVPv3000q173gn9oCXx9vbW+3bt3f1gh//4wCALfRkAh7sqquu0osvvqjhw4friiuu0D333KNLLrnE1Zszb948SSpzabw8d955p1588UUNGjRIW7ZsUZs2bbR8+XKlpqbqhhtuULdu3ST9GWQGDhzoevD2pZdeqjVr1mjOnDkVrvv999+Xj4+Punfv7hpdfumll6p///6S/vxDOnbsWD322GP65Zdf1LNnT4WFhWnHjh1as2aNgoOD9fTTT8vLy0vPPPOM7r77bt10000aOnSo9uzZo5SUlEpdLj906JBWrVol6c9R+ps3b9aECRMkSSNHjnTV9e7dWzNmzNDFF1+stm3bKjMzUxMnTqzwETWny9vbW/Pnz1dCQoLi4+P197//XV26dFFwcLB+/fVXvffee/rwww8rfcm0e/fu8vPz0+23367Ro0fr8OHDevnllyu9njZt2mju3Ll6++231axZMwUEBKhNmzaVWsfpfp9sGz9+vLp3764uXbro4Ycflp+fn1566SVlZ2frrbfecgXxp556Sh999JG6dOmiJ598UvXr19fs2bO1cOFCTZgwodIPwXc6nfrvf/+r3r176/LLL3d7GPtPP/2ktLQ0ffvtt7r55pvl5eWlCRMm6I477lDv3r01bNgwFRUVaeLEidqzZ4/++c9/utZ722236cknn9SAAQP0j3/8Q4cPH9YLL7yg0tLSKu+jY8fyX//6lwYNGiRfX1/FxsZq9uzZ+vzzz9WrVy81atRIhw8fdj2R4GwdL9Rx1TnqCMDpycrKMnfddZdp2rSp8ff3NwEBAeYvf/mLufPOO8uMHh40aJAJDg4udz27d+829957r7nggguMj4+Pady4sRkzZkyZUdCFhYXm7rvvNlFRUSY4ONj06dPHbNmypcLR5ZmZmaZPnz6mXr16JiQkxNx+++1mx44dZba/YMEC06VLFxMaGmr8/f1N48aNza233moWL17sVvfvf//btGjRwvj5+ZmLLrrIvPHGGxWOwj3RiaPLvby8TExMjLn++uvN0qVL3WoLCgrMkCFDTGRkpAkKCjJXX321+fLLL02nTp3cRl8fG13+7rvvnnL7x9uzZ4955plnzBVXXGHq1atnfH19TaNGjczAgQPNV199VWY/7ty50235N998s8wo4Q8//NBceumlJiAgwDRo0MD84x//MJ988omRZJYsWeK2H04cEX7Mli1bTEJCggkJCTGSTrlfK1rX6X6fdMII8coob9kvv/zSXHfddSY4ONgEBgaaDh06mA8//LDMst9//73p06ePcTqdxs/Pz1x66aVlRpGf7ujyY/Ly8swjjzxiLrnkEhMUFGT8/f3NX/7yFzNs2DDz/fffu9UuWLDAtG/f3gQEBJjg4GDTtWtXt+N+zMcff2wuu+wyExgYaJo1a2amTZtW4ejy8vZj48aNzaBBg9ymjRkzxsTExBgvLy/Xd2PlypXmpptuMo0bNzb+/v4mPDzcdOrUyXzwwQen9dmBynIYc1xfPgBUQkpKip5++mnt3Lmz0ve4AQBqN+7JBAAAgHWETAAAAFjH5XIAAABYR08mAAAArCNkAgAAwDpCJgAAAKzjYezn2NGjR7Vt2zaFhISc1Z98AwAAsM0Yo3379ikmJkZeXifvqyRknmPbtm1Tw4YNq7sZAAAAVbZ169ZT/joaIfMcCwkJkfTnwTmdnwIEAADwFHv37lXDhg1deeZkCJnn2LFL5KGhoYRMAABQI53OLX8M/AEAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFjnU90NAAAAZ09ubq527dpV5eUjIiLUqFEjiy1CXUHIBACglsrNzVXsxS11+NDBKq8jIDBIG3/IIWii0giZAADUUrt27dLhQwfVtleagsNbVnr5A7tz9N3Cgdq1axchE5VGyAQAoJYLDm8pZ/QV1d0M1DEM/AEAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHXVGjK/+OIL9enTRzExMXI4HFqwYIHbfGOMUlJSFBMTo8DAQHXu3Fnr1693qykqKtLw4cMVERGh4OBg9e3bV7/99ptbTUFBgZKSkuR0OuV0OpWUlKQ9e/a41eTm5qpPnz4KDg5WRESERowYoeLiYrea77//Xp06dVJgYKAaNGigsWPHyhhjbX8AAADUFtUaMg8cOKBLL71U06ZNK3f+hAkTNHnyZE2bNk1r165VdHS0unfvrn379rlqRo4cqfnz52vu3Llavny59u/fr969e6u0tNRVk5iYqKysLKWnpys9PV1ZWVlKSkpyzS8tLVWvXr104MABLV++XHPnztW8efM0atQoV83evXvVvXt3xcTEaO3atZo6daqee+45TZ48+SzsGQAAgJrNpzo3fv311+v6668vd54xRlOmTNFjjz2mm2++WZI0c+ZMRUVFac6cORo2bJgKCwv1+uuva9asWerWrZskKS0tTQ0bNtTixYvVo0cP5eTkKD09XatWrVL79u0lSdOnT1d8fLw2btyo2NhYZWRkaMOGDdq6datiYmIkSZMmTdLgwYM1btw4hYaGavbs2Tp8+LBmzJghf39/tW7dWj/++KMmT56shx56SA6H4xzsMQAAgJrBY+/J3Lx5s/Ly8pSQkOCa5u/vr06dOmnFihWSpMzMTJWUlLjVxMTEqHXr1q6alStXyul0ugKmJHXo0EFOp9OtpnXr1q6AKUk9evRQUVGRMjMzXTWdOnWSv7+/W822bdu0ZcuWCj9HUVGR9u7d6/YCAACo7Tw2ZObl5UmSoqKi3KZHRUW55uXl5cnPz09hYWEnrYmMjCyz/sjISLeaE7cTFhYmPz+/k9Yce3+spjzjx4933QvqdDrVsGHDk39wAACAWsBjQ+YxJ16GNsac8tL0iTXl1duoOTbo52TtGTNmjAoLC12vrVu3nrTtAAAAtYHHhszo6GhJZXsJ8/PzXT2I0dHRKi4uVkFBwUlrduzYUWb9O3fudKs5cTsFBQUqKSk5aU1+fr6ksr2tx/P391doaKjbCwAAoLbz2JDZtGlTRUdHa9GiRa5pxcXFWrZsmTp27ChJiouLk6+vr1vN9u3blZ2d7aqJj49XYWGh1qxZ46pZvXq1CgsL3Wqys7O1fft2V01GRob8/f0VFxfnqvniiy/cHmuUkZGhmJgYNWnSxP4OAAAAqMGqNWTu379fWVlZysrKkvTnYJ+srCzl5ubK4XBo5MiRSk1N1fz585Wdna3BgwcrKChIiYmJkiSn06khQ4Zo1KhR+uyzz7Ru3ToNHDhQbdq0cY02b9mypXr27KmhQ4dq1apVWrVqlYYOHarevXsrNjZWkpSQkKBWrVopKSlJ69at02effaaHH35YQ4cOdfU8JiYmyt/fX4MHD1Z2drbmz5+v1NRURpYDAACUo1ofYfT111+rS5curvcPPfSQJGnQoEGaMWOGRo8erUOHDik5OVkFBQVq3769MjIyFBIS4lrm+eefl4+Pj/r3769Dhw6pa9eumjFjhry9vV01s2fP1ogRI1yj0Pv27ev2bE5vb28tXLhQycnJuuqqqxQYGKjExEQ999xzrhqn06lFixbpvvvuU7t27RQWFqaHHnrI1WYAAAD8H4fhJ2vOqb1798rpdKqwsJD7MwEAZ9U333yjuLg4xd+ZKWf0FZVevjDvG638T5wyMzN1xRWVXx61T2VyjMfekwkAAICai5AJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrfKq7AQAAAJByc3O1a9euM1pHRESEGjVqZKlFZ4aQCQAAUM1yc3MVe3FLHT508IzWExAYpI0/5HhE0CRkAgAAVLNdu3bp8KGDatsrTcHhLau0jgO7c/TdwoHatWsXIRMAAAD/Jzi8pZzRV1R3M6xg4A8AAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6HmEEAKiUM/1VEk/6RRIAZw8hEwBw2mz8Kokn/SIJgLOHkAkAOG1n+qsknvaLJADOHkImAKDSatOvkgA4Oxj4AwAAAOsImQAAALCOkAkAAADrCJkAAACwjoE/qLV4lh8AANWHkIlaiWf5AQBQvQiZqJV4lh8AANWLkIlajWf5AQBQPRj4AwAAAOvoyQRQ7c50kJbEQC0A8DSETADVysYgLYmBWgDgaQiZAKrVmQ7SkhioBQCeiJAJwCMwSAsAahcG/gAAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDqPDplHjhzR448/rqZNmyowMFDNmjXT2LFjdfToUVeNMUYpKSmKiYlRYGCgOnfurPXr17utp6ioSMOHD1dERISCg4PVt29f/fbbb241BQUFSkpKktPplNPpVFJSkvbs2eNWk5ubqz59+ig4OFgREREaMWKEiouLz9rnBwAAqKk8OmQ+++yzeuWVVzRt2jTl5ORowoQJmjhxoqZOneqqmTBhgiZPnqxp06Zp7dq1io6OVvfu3bVv3z5XzciRIzV//nzNnTtXy5cv1/79+9W7d2+Vlpa6ahITE5WVlaX09HSlp6crKytLSUlJrvmlpaXq1auXDhw4oOXLl2vu3LmaN2+eRo0adW52BgAAQA3iU90NOJmVK1eqX79+6tWrlySpSZMmeuutt/T1119L+rMXc8qUKXrsscd08803S5JmzpypqKgozZkzR8OGDVNhYaFef/11zZo1S926dZMkpaWlqWHDhlq8eLF69OihnJwcpaena9WqVWrfvr0kafr06YqPj9fGjRsVGxurjIwMbdiwQVu3blVMTIwkadKkSRo8eLDGjRun0NDQc717AAAAPJZH92ReffXV+uyzz/Tjjz9Kkr799lstX75cN9xwgyRp8+bNysvLU0JCgmsZf39/derUSStWrJAkZWZmqqSkxK0mJiZGrVu3dtWsXLlSTqfTFTAlqUOHDnI6nW41rVu3dgVMSerRo4eKioqUmZl5lvYAAABAzeTRPZmPPPKICgsLdfHFF8vb21ulpaUaN26cbr/9dklSXl6eJCkqKsptuaioKP3666+uGj8/P4WFhZWpObZ8Xl6eIiMjy2w/MjLSrebE7YSFhcnPz89VU56ioiIVFRW53u/du/e0PjsAAEBN5tE9mW+//bbS0tI0Z84cffPNN5o5c6aee+45zZw5063O4XC4vTfGlJl2ohNryquvSs2Jxo8f7xpM5HQ61bBhw5O2CwAAoDbw6JD5j3/8Q48++qgGDBigNm3aKCkpSQ8++KDGjx8vSYqOjpakMj2J+fn5rl7H6OhoFRcXq6Cg4KQ1O3bsKLP9nTt3utWcuJ2CggKVlJSU6eE83pgxY1RYWOh6bd26tTK7AAAAoEby6JB58OBBeXm5N9Hb29v1CKOmTZsqOjpaixYtcs0vLi7WsmXL1LFjR0lSXFycfH193Wq2b9+u7OxsV018fLwKCwu1Zs0aV83q1atVWFjoVpOdna3t27e7ajIyMuTv76+4uLgKP4O/v79CQ0PdXgAAALWdR9+T2adPH40bN06NGjXSJZdconXr1mny5Mn629/+JunPy9cjR45UamqqWrRooRYtWig1NVVBQUFKTEyUJDmdTg0ZMkSjRo1SeHi46tevr4cfflht2rRxjTZv2bKlevbsqaFDh+rVV1+VJN1zzz3q3bu3YmNjJUkJCQlq1aqVkpKSNHHiRP3xxx96+OGHNXToUIIjAADACTw6ZE6dOlVPPPGEkpOTlZ+fr5iYGA0bNkxPPvmkq2b06NE6dOiQkpOTVVBQoPbt2ysjI0MhISGumueff14+Pj7q37+/Dh06pK5du2rGjBny9vZ21cyePVsjRoxwjULv27evpk2b5prv7e2thQsXKjk5WVdddZUCAwOVmJio55577hzsCQAAgJrFo0NmSEiIpkyZoilTplRY43A4lJKSopSUlAprAgICNHXqVLeHuJ+ofv36SktLO2l7GjVqpI8++uhUzQYAAKjzPPqeTAAAANRMhEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWOdT3Q0AAOBsyc3N1a5du6q8fEREhBo1amSxRUDdQcgEANRKubm5ir24pQ4fOljldQQEBmnjDzkETaAKCJkAgFpp165dOnzooNr2SlNweMtKL39gd46+WzhQu3btImQCVUDIBADUasHhLeWMvqK6m1FlZ3LJPycnx3JrgNNHyAQA4CxZu3atfvzxxyovv2vXLj3yyBgVFR06o3Yc2F21sFnV5QCJkAkAwFmRm5urjh3jdeRI6Rmvq6qX/PN//kCbVj6j7xYOrPK2HQ4vFRUVVXl51F2ETAAAzoJdu3bpyJFSPdNFahpWtXV8lSu9/HXVL/nv350jmaNVbsPmAumJJUfl7+9f+YVR5xEyAQA4i25oIV0RU/XlX/66+trwzTbpiSVnvn3UTYTMWopnwwFnx5meWxLnF4C6gZBZC/FsOODssHFuSZxfAOoGQmYtxLPhgLPjTM8tifMLOBmuwtUuhMxarKY/Gw7wVDX93OK5i/BEXIWrfQiZAFDDnMmzF209dxGwjatwtQ8hEx6L3hagLFvPXqzqH/Kdv3ysn5c/cUbbBk6mpl8pwP8hZMIj2RpgAdQ2Z/rsRSvPXQSA00DIxFlxpj+ltnnz5jO6bEJvC2q7M3n2oo3nLgIoiytw7giZsM7mT6nR2wIAOFe439kuQiass/lTagAAnAvc72wfIRNnjSf8lBoAAKeD+53tI2QCHoAHEAOAZ+B+Z3sImUA14wHEAIDaiJAJVDMeQAwAqI0ImYCH4AHEAIDaxKu6GwAAAIDah55MAGf88PzQ0FA1aNCgSsvWxgcQAwAImUCdZ+XZcA4vyRy11ygAQI1HyATqODvPhjvKA4gBAG4ImQAknfmz4XgAMQDgeAz8AQAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHU+1d0AAAA8WU5OzjldDqgtCJkAAJSjaP92yeGlgQMHVndTgBqJkAkAQDlKivZI5qja9kpTcHjLSi+/85eP9fPyJ+w3DKghCJkAAJxEcHhLOaOvqPRy+3dzuRx1GwN/AAAAYB0hEwAAANZxuRwVYkQlAKCm4W+X5yBkogxGVAIAahr+dnkeQibKYEQlAKCm4W+X5yFkokKMqAQA1DT87fIcDPwBAACAdfRkAkA1YHACgNqOkAkA5xCDEwDUFYRMADiHGJwAoK4gZAJANWBwAoDajoE/AAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjtHlQC3Bw70BAJ6EkAnUcDzcGwDgiQiZQA3Hw70BAJ6IkAnUEjzcGwDgSRj4AwAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6zw+ZP7+++8aOHCgwsPDFRQUpMsuu0yZmZmu+cYYpaSkKCYmRoGBgercubPWr1/vto6ioiINHz5cERERCg4OVt++ffXbb7+51RQUFCgpKUlOp1NOp1NJSUnas2ePW01ubq769Omj4OBgRUREaMSIESouLj5rnx0AAKCm8uiQWVBQoKuuukq+vr765JNPtGHDBk2aNEnnnXeeq2bChAmaPHmypk2bprVr1yo6Olrdu3fXvn37XDUjR47U/PnzNXfuXC1fvlz79+9X7969VVpa6qpJTExUVlaW0tPTlZ6erqysLCUlJbnml5aWqlevXjpw4ICWL1+uuXPnat68eRo1atQ52RcAAAA1iU91N+Bknn32WTVs2FBvvvmma1qTJk1c/22M0ZQpU/TYY4/p5ptvliTNnDlTUVFRmjNnjoYNG6bCwkK9/vrrmjVrlrp16yZJSktLU8OGDbV48WL16NFDOTk5Sk9P16pVq9S+fXtJ0vTp0xUfH6+NGzcqNjZWGRkZ2rBhg7Zu3aqYmBhJ0qRJkzR48GCNGzdOoaGh52ivAAAAeD6P7sn84IMP1K5dO/3P//yPIiMjdfnll2v69Omu+Zs3b1ZeXp4SEhJc0/z9/dWpUyetWLFCkpSZmamSkhK3mpiYGLVu3dpVs3LlSjmdTlfAlKQOHTrI6XS61bRu3doVMCWpR48eKioqcrt8DwAAAA8Pmb/88otefvlltWjRQp9++qnuvfdejRgxQv/5z38kSXl5eZKkqKgot+WioqJc8/Ly8uTn56ewsLCT1kRGRpbZfmRkpFvNidsJCwuTn5+fq6Y8RUVF2rt3r9sLAACgtvPoy+VHjx5Vu3btlJqaKkm6/PLLtX79er388su68847XXUOh8NtOWNMmWknOrGmvPqq1Jxo/Pjxevrpp0/aFgAAgNrGo3syL7jgArVq1cptWsuWLZWbmytJio6OlqQyPYn5+fmuXsfo6GgVFxeroKDgpDU7duwos/2dO3e61Zy4nYKCApWUlJTp4TzemDFjVFhY6Hpt3br1lJ8bAACgpvPokHnVVVdp48aNbtN+/PFHNW7cWJLUtGlTRUdHa9GiRa75xcXFWrZsmTp27ChJiouLk6+vr1vN9u3blZ2d7aqJj49XYWGh1qxZ46pZvXq1CgsL3Wqys7O1fft2V01GRob8/f0VFxdX4Wfw9/dXaGio2wsAAKC28+jL5Q8++KA6duyo1NRU9e/fX2vWrNFrr72m1157TdKfl69Hjhyp1NRUtWjRQi1atFBqaqqCgoKUmJgoSXI6nRoyZIhGjRql8PBw1a9fXw8//LDatGnjGm3esmVL9ezZU0OHDtWrr74qSbrnnnvUu3dvxcbGSpISEhLUqlUrJSUlaeLEifrjjz/08MMPa+jQoQRHAACAE3h0yLzyyis1f/58jRkzRmPHjlXTpk01ZcoU3XHHHa6a0aNH69ChQ0pOTlZBQYHat2+vjIwMhYSEuGqef/55+fj4qH///jp06JC6du2qGTNmyNvb21Uze/ZsjRgxwjUKvW/fvpo2bZprvre3txYuXKjk5GRdddVVCgwMVGJiop577rlzsCcAAABqFo8OmZLUu3dv9e7du8L5DodDKSkpSklJqbAmICBAU6dO1dSpUyusqV+/vtLS0k7alkaNGumjjz46ZZsBAADqOo++JxMAAAA1EyETAAAA1hEyAQAAYB0hEwAAANZVKWQ2a9ZMu3fvLjN9z549atas2Rk3CgAAADVblULmli1bVFpaWmZ6UVGRfv/99zNuFAAAAGq2Sj3C6IMPPnD996effiqn0+l6X1paqs8++0xNmjSx1jgAAIC65MDunGpZ9myoVMi88cYbJf35bMpBgwa5zfP19VWTJk00adIka40DAACoC0xpseTw0ncLB57RehwOLxUVFVlq1ZmpVMg8evSopD9/M3zt2rWKiIg4K40CAACoSxzefpI5qme6SE3DqraOzQXSE0uOyt/f327jqqhKv/izefNm2+0AAACo825oIV0RU7Vlv9kmPbHEbnvORJV/VvKzzz7TZ599pvz8fFcP5zFvvPHGGTcMAAAANVeVQubTTz+tsWPHql27drrgggvkcDhstwsAAAA1WJVC5iuvvKIZM2YoKSnJdnsAAABQC1TpOZnFxcXq2LGj7bYAAACglqhSyLz77rs1Z84c220BAABALVGly+WHDx/Wa6+9psWLF6tt27by9fV1mz958mQrjQMAAEDNVKWQ+d133+myyy6TJGVnZ7vNYxAQAAAAqhQylyzxoIcwAQAAwONU6Z5MAAAA4GSq1JPZpUuXk14W//zzz6vcIACA5zuwO+ecLgeg5qlSyDx2P+YxJSUlysrKUnZ2tgYNGmSjXQAAD2RKiyWHl75bOLDK63A4vFRUVGSxVQA8UZVC5vPPP1/u9JSUFO3fv/+MGgQA8FwObz/JHNUzXaSmYZVffnOB9MSSo/L397ffOAAepcq/XV6egQMH6q9//auee+45m6sFAHiYG1pIV8RUfrlvtklPMHYUqBOsDvxZuXKlAgICbK4SAAAANVCVejJvvvlmt/fGGG3fvl1ff/21nnjiCSsNAwAAQM1VpZDpdDrd3nt5eSk2NlZjx45VQkKClYYBAACg5qpSyHzzzTdttwMAAAC1yBkN/MnMzFROTo4cDodatWqlyy+/3Fa7AAAAUINVKWTm5+drwIABWrp0qc477zwZY1RYWKguXbpo7ty5Ov/88223E6gSHhgNAED1qFLIHD58uPbu3av169erZcuWkqQNGzZo0KBBGjFihN566y2rjQQqiwdGAwBQvaoUMtPT07V48WJXwJSkVq1a6cUXX2TgDzwCD4wGAKB6VSlkHj16VL6+vmWm+/r66ujRo2fcKMAWHhgNAED1qNLD2K+77jo98MAD2rZtm2va77//rgcffFBdu3a11jgAAADUTFUKmdOmTdO+ffvUpEkTNW/eXH/5y1/UtGlT7du3T1OnTrXdRgAAANQwVbpc3rBhQ33zzTdatGiRfvjhBxlj1KpVK3Xr1s12+wDUEWcyop+nAQCA56lUyPz88891//33a9WqVQoNDVX37t3VvXt3SVJhYaEuueQSvfLKK7rmmmvOSmMB1D42ngQg8TQAAPA0lQqZU6ZM0dChQxUaGlpmntPp1LBhwzR58mRCJoDTdqZPApB4GgAAeKJKhcxvv/1Wzz77bIXzExIS9Nxzz51xowDUPVV9EoDE0wAAwBNVauDPjh07yn100TE+Pj7auXPnGTcKAAAANVulQmaDBg30/fffVzj/u+++0wUXXHDGjQIAAEDNVqmQecMNN+jJJ5/U4cOHy8w7dOiQnnrqKfXu3dta4wAAAFAzVeqezMcff1zvv/++LrroIt1///2KjY2Vw+FQTk6OXnzxRZWWluqxxx47W20FAABADVGpkBkVFaUVK1bo73//u8aMGSNjjCTJ4XCoR48eeumllxQVFXVWGgoAAICao9IPY2/cuLE+/vhjFRQU6Oeff5YxRi1atFBYWBWfPQIAAIBap0q/+CNJYWFhuvLKK222BQAAALVElX67HAAAADgZQiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKzzqe4GAADOvQO7c6q03KHCzZZbAtQunFv/h5AJj8bJCthVXCrJ4aXvFg6s7qYAtQrnVlmETHgkTlbg7PDzlmSO6pkuUtOwyi//Va708tfWmwW41NTOBc6tsgiZtVhNPVElTlZ4tqqeW5JnnF+SdEML6YqYqi3LuYWzoaioSI5a0LnAufV/CJm1UG05USVOVniW2nRuoWapyZ0Gp8vf31+GzoVahZBZC3GiAmfHmZ5bEucXKqcu3jpE50LtQcisxThRgbPjTM4tifMLp49bh1CTETIBAPBwdBqgJiJkAgBqtbpwPyP+D8fbcxAyAQC1EgO16pa6eP+qpyNkAgBqJVuDIOkZqxm4f9XzEDJRIf5hBc4ezq9zp6r3MxaXSi9n0jNW03D/qucgZKIMLjkAZw/nV81BzxhwZgiZKIN/WIGzh/Or5qFnDKgaQiYqxD+swNnD+QWgtiNkArCCewwBAMcjZAIeoqaGNO4xBACUh5AJVLOa/iw/7jEEAJSHkAlUM1vP8qtu3GMIADgeIRPwEIQ0AEBt4lXdDQAAAEDtQ8gEAACAdYRMAAAAWMc9mUAtUVMfgQQAqJ0ImUANx3MqAQCeiJAJ1HA8pxIA4IkImUAtwSOQAACehIE/AAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCuRoXM8ePHy+FwaOTIka5pxhilpKQoJiZGgYGB6ty5s9avX++2XFFRkYYPH66IiAgFBwerb9+++u2339xqCgoKlJSUJKfTKafTqaSkJO3Zs8etJjc3V3369FFwcLAiIiI0YsQIFRcXn62PCwAAUGPVmJC5du1avfbaa2rbtq3b9AkTJmjy5MmaNm2a1q5dq+joaHXv3l379u1z1YwcOVLz58/X3LlztXz5cu3fv1+9e/dWaWmpqyYxMVFZWVlKT09Xenq6srKylJSU5JpfWlqqXr166cCBA1q+fLnmzp2refPmadSoUWf/wwMAANQwNSJk7t+/X3fccYemT5+usLAw13RjjKZMmaLHHntMN998s1q3bq2ZM2fq4MGDmjNnjiSpsLBQr7/+uiZNmqRu3brp8ssvV1pamr7//nstXrxYkpSTk6P09HT9+9//Vnx8vOLj4zV9+nR99NFH2rhxoyQpIyNDGzZsUFpami6//HJ169ZNkyZN0vTp07V3795zv1MAAAA8WI0Imffdd5969eqlbt26uU3fvHmz8vLylJCQ4Jrm7++vTp06acWKFZKkzMxMlZSUuNXExMSodevWrpqVK1fK6XSqffv2rpoOHTrI6XS61bRu3VoxMTGumh49eqioqEiZmZkVtr2oqEh79+51ewEAANR2PtXdgFOZO3euvvnmG61du7bMvLy8PElSVFSU2/SoqCj9+uuvrho/Pz+3HtBjNceWz8vLU2RkZJn1R0ZGutWcuJ2wsDD5+fm5asozfvx4Pf3006f6mAAAALWKR/dkbt26VQ888IDS0tIUEBBQYZ3D4XB7b4wpM+1EJ9aUV1+VmhONGTNGhYWFrtfWrVtP2i4AAIDawKNDZmZmpvLz8xUXFycfHx/5+Pho2bJleuGFF+Tj4+PqWTyxJzE/P981Lzo6WsXFxSooKDhpzY4dO8psf+fOnW41J26noKBAJSUlZXo4j+fv76/Q0FC3FwAAQG3n0SGza9eu+v7775WVleV6tWvXTnfccYeysrLUrFkzRUdHa9GiRa5liouLtWzZMnXs2FGSFBcXJ19fX7ea7du3Kzs721UTHx+vwsJCrVmzxlWzevVqFRYWutVkZ2dr+/btrpqMjAz5+/srLi7urO4HAACAmsaj78kMCQlR69at3aYFBwcrPDzcNX3kyJFKTU1VixYt1KJFC6WmpiooKEiJiYmSJKfTqSFDhmjUqFEKDw9X/fr19fDDD6tNmzaugUQtW7ZUz549NXToUL366quSpHvuuUe9e/dWbGysJCkhIUGtWrVSUlKSJk6cqD/++EMPP/ywhg4dSu8kAADACTw6ZJ6O0aNH69ChQ0pOTlZBQYHat2+vjIwMhYSEuGqef/55+fj4qH///jp06JC6du2qGTNmyNvb21Uze/ZsjRgxwjUKvW/fvpo2bZprvre3txYuXKjk5GRdddVVCgwMVGJiop577rlz92EBAABqiBoXMpcuXer23uFwKCUlRSkpKRUuExAQoKlTp2rq1KkV1tSvX19paWkn3XajRo300UcfVaa5AAAAdZJH35MJAACAmomQCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOs8OmSOHz9eV155pUJCQhQZGakbb7xRGzdudKsxxiglJUUxMTEKDAxU586dtX79ereaoqIiDR8+XBEREQoODlbfvn3122+/udUUFBQoKSlJTqdTTqdTSUlJ2rNnj1tNbm6u+vTpo+DgYEVERGjEiBEqLi4+K58dAACgJvPokLls2TLdd999WrVqlRYtWqQjR44oISFBBw4ccNVMmDBBkydP1rRp07R27VpFR0ere/fu2rdvn6tm5MiRmj9/vubOnavly5dr//796t27t0pLS101iYmJysrKUnp6utLT05WVlaWkpCTX/NLSUvXq1UsHDhzQ8uXLNXfuXM2bN0+jRo06NzsDAACgBvGp7gacTHp6utv7N998U5GRkcrMzNS1114rY4ymTJmixx57TDfffLMkaebMmYqKitKcOXM0bNgwFRYW6vXXX9esWbPUrVs3SVJaWpoaNmyoxYsXq0ePHsrJyVF6erpWrVql9u3bS5KmT5+u+Ph4bdy4UbGxscrIyNCGDRu0detWxcTESJImTZqkwYMHa9y4cQoNDT2HewYAAMCzeXRP5okKCwslSfXr15ckbd68WXl5eUpISHDV+Pv7q1OnTlqxYoUkKTMzUyUlJW41MTExat26tatm5cqVcjqdroApSR06dJDT6XSrad26tStgSlKPHj1UVFSkzMzMCttcVFSkvXv3ur0AAABquxoTMo0xeuihh3T11VerdevWkqS8vDxJUlRUlFttVFSUa15eXp78/PwUFhZ20prIyMgy24yMjHSrOXE7YWFh8vPzc9WUZ/z48a77PJ1Opxo2bFiZjw0AAFAj1ZiQef/99+u7777TW2+9VWaew+Fwe2+MKTPtRCfWlFdflZoTjRkzRoWFha7X1q1bT9ouAACA2qBGhMzhw4frgw8+0JIlS3ThhRe6pkdHR0tSmZ7E/Px8V69jdHS0iouLVVBQcNKaHTt2lNnuzp073WpO3E5BQYFKSkrK9HAez9/fX6GhoW4vAACA2s6jQ6YxRvfff7/ef/99ff7552ratKnb/KZNmyo6OlqLFi1yTSsuLtayZcvUsWNHSVJcXJx8fX3darZv367s7GxXTXx8vAoLC7VmzRpXzerVq1VYWOhWk52dre3bt7tqMjIy5O/vr7i4OPsfHgAAoAbz6NHl9913n+bMmaP//ve/CgkJcfUkOp1OBQYGyuFwaOTIkUpNTVWLFi3UokULpaamKigoSImJia7aIUOGaNSoUQoPD1f9+vX18MMPq02bNq7R5i1btlTPnj01dOhQvfrqq5Kke+65R71791ZsbKwkKSEhQa1atVJSUpImTpyoP/74Qw8//LCGDh1K7yQAAMAJPDpkvvzyy5Kkzp07u01/8803NXjwYEnS6NGjdejQISUnJ6ugoEDt27dXRkaGQkJCXPXPP/+8fHx81L9/fx06dEhdu3bVjBkz5O3t7aqZPXu2RowY4RqF3rdvX02bNs0139vbWwsXLlRycrKuuuoqBQYGKjExUc8999xZ+vQAAAA1l0eHTGPMKWscDodSUlKUkpJSYU1AQICmTp2qqVOnVlhTv359paWlnXRbjRo10kcffXTKNgEAANR1Hn1PJgAAAGomQiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2QCAADAOkImAAAArCNkAgAAwDpCJgAAAKwjZAIAAMA6QiYAAACsI2RWwUsvvaSmTZsqICBAcXFx+vLLL6u7SQAAAB6FkFlJb7/9tkaOHKnHHntM69at0zXXXKPrr79eubm51d00AAAAj0HIrKTJkydryJAhuvvuu9WyZUtNmTJFDRs21Msvv1zdTQMAAPAYhMxKKC4uVmZmphISEtymJyQkaMWKFdXUKgAAAM/jU90NqEl27dql0tJSRUVFuU2PiopSXl5eucsUFRWpqKjI9b6wsFCStHfv3rPWzv3790uSMrdL+4srv3zOTlXr8p7Qho27//zf/fv3n9VjdWwbUs09XtW9vHTujteZHiup+vdXdS/PuVWz2sDxqjnLS+fmeB1brzHm1MUGp+333383ksyKFSvcpv+///f/TGxsbLnLPPXUU0YSL168ePHixYtXrXlt3br1lLmJnsxKiIiIkLe3d5ley/z8/DK9m8eMGTNGDz30kOv90aNH9ccffyg8PFwOh+OsttdT7d27Vw0bNtTWrVsVGhpa3c3BKXC8ahaOV83C8ao5OFZ/MsZo3759iomJOWUtIbMS/Pz8FBcXp0WLFummm25yTV+0aJH69etX7jL+/v7y9/d3m3beeeedzWbWGKGhoXX6RK1pOF41C8erZuF41RwcK8npdJ5WHSGzkh566CElJSWpXbt2io+P12uvvabc3Fzde++91d00AAAAj0HIrKTbbrtNu3fv1tixY7V9+3a1bt1aH3/8sRo3blzdTQMAAPAYhMwqSE5OVnJycnU3o8by9/fXU089VeY2AngmjlfNwvGqWTheNQfHqvIcxpzOGHQAAADg9PEwdgAAAFhHyAQAAIB1hEwAAABYR8isg7744gv16dNHMTExcjgcWrBggWteSUmJHnnkEbVp00bBwcGKiYnRnXfeqW3btpVZz8qVK3XdddcpODhY5513njp37qxDhw655jdp0kQOh8Pt9eijj7rmf/vtt7r99tvVsGFDBQYGqmXLlvrXv/7lto2UlJQy63A4HAoODnarW7ZsmeLi4hQQEKBmzZrplVdecZs/Y8aMctdz+PDhM9mVZ11dPFadO3cudz29evU6k115TtTF41VSUqKxY8eqefPmCggI0KWXXqr09PQz2Y3nTG07Xtu3b1diYqJiY2Pl5eWlkSNHlvu5582bp1atWsnf31+tWrXS/Pnzq7gHz526eKymT5+ua665RmFhYQoLC1O3bt20Zs2aM9iL1eDMf2wRNc3HH39sHnvsMTNv3jwjycyfP981b8+ePaZbt27m7bffNj/88INZuXKlad++vYmLi3Nbx4oVK0xoaKgZP368yc7ONj/++KN59913zeHDh101jRs3NmPHjjXbt293vfbt2+ea//rrr5vhw4ebpUuXmk2bNplZs2aZwMBAM3XqVFfNvn373Jbfvn27adWqlRk0aJCr5pdffjFBQUHmgQceMBs2bDDTp083vr6+5r333nPVvPnmmyY0NLTMujxdXTxWu3fvdltHdna28fb2Nm+++aa9HXuW1MXjNXr0aBMTE2MWLlxoNm3aZF566SUTEBBgvvnmG4t79uyobcdr8+bNZsSIEWbmzJnmsssuMw888ECZz7xixQrj7e1tUlNTTU5OjklNTTU+Pj5m1apVZ75Dz6K6eKwSExPNiy++aNatW2dycnLMXXfdZZxOp/ntt9/OfIeeI4TMOu7Ek7U8a9asMZLMr7/+6prWvn178/jjj590ucaNG5vnn3++Uu1JTk42Xbp0qXB+VlaWkWS++OIL17TRo0ebiy++2K1u2LBhpkOHDq73b775pnE6nZVqi6epK8fqRM8//7wJCQkx+/fvr1T7qltdOV4XXHCBmTZtmltNv379zB133FGp9lW32nC8jtepU6dyg0v//v1Nz5493ab16NHDDBgwoFLtq0515Vid6MiRIyYkJMTMnDmzUu2rTlwuxykVFhbK4XC4fg4zPz9fq1evVmRkpDp27KioqCh16tRJy5cvL7Pss88+q/DwcF122WUaN26ciouLT7mt+vXrVzj/3//+ty666CJdc801rmkrV65UQkKCW12PHj309ddfq6SkxDVt//79aty4sS688EL17t1b69atO52PX6PUlmN1vNdff10DBgwocxm3NqgNx6uoqEgBAQFuNYGBgeW2uabz9ON1Oio6pitWrKjUejxdbThWJzp48KBKSkpOui2PU90pF9VLp/h/hIcOHTJxcXFuvRIrV640kkz9+vXNG2+8Yb755hszcuRI4+fnZ3788UdX3eTJk83SpUvNt99+a6ZPn24iIiLMkCFDKtzWihUrjK+vr8nIyCh3/uHDh01YWJh59tln3aa3aNHCjBs3zm3aV199ZSSZbdu2udo8a9Ysk5WVZb744gtzyy23mMDAQLf2erq6cqyOt3r1aiPJrF69usK2eKq6crxuv/1206pVK/Pjjz+a0tJSk5GRYQIDA42fn1+F7fFEteF4Ha+i3jFfX18ze/Zst2mzZ8+uUcerrhyrEyUnJ5vmzZubQ4cOnbLWUxAy67iTnazFxcWmX79+5vLLLzeFhYWu6cf+yIwZM8atvk2bNubRRx+tcFvvvfeekWR27dpVZl52drY5//zzzTPPPFPh8nPmzDE+Pj5l7qVs0aKFSU1NdZu2fPlyI6nC+y5LS0vNpZdeaoYPH17h9jxNXTxW99xzj2ndunWF2/FkdeV45efnm379+hkvLy/j7e1tLrroIpOcnGwCAwMr3J4nqg3H63gnC5lz5sxxm5aWlmb8/f0rXJenqSvH6njPPvusCQsLM99+++1J6zwNl8tRrpKSEvXv31+bN2/WokWLFBoa6pp3wQUXSJJatWrltkzLli2Vm5tb4To7dOggSfr555/dpm/YsEHXXXedhg4dqscff7zC5f/973+rd+/eio6OdpseHR2tvLw8t2n5+fny8fFReHh4uevy8vLSlVdeqZ9++qnC7dUUtfVYHTx4UHPnztXdd99d4XZqotp2vM4//3wtWLBABw4c0K+//qoffvhB9erVU9OmTSvcXk1Sk47X6ajomEZFRVV6XZ6mth2rY5577jmlpqYqIyNDbdu2rfJ6qgMhE2UcO1F/+uknLV68uMwf/yZNmigmJkYbN250m/7jjz+qcePGFa732D2Qx052SVq/fr26dOmiQYMGady4cRUuu3nzZi1ZskRDhgwpMy8+Pl6LFi1ym5aRkaF27drJ19e33PUZY5SVleXWlpqoNh+rd955R0VFRRo4cGCF26ppavPxCggIUIMGDXTkyBHNmzdP/fr1q3CbNUVNO16no6Jj2rFjxyqtz1PUxmMlSRMnTtQzzzyj9PR0tWvXrsrrqTbV3ZWKc2/fvn1m3bp1Zt26dUaSmTx5slm3bp359ddfTUlJienbt6+58MILTVZWltsjGIqKilzreP75501oaKh59913zU8//WQef/xxExAQYH7++WdjzJ/3qRxb7y+//GLefvttExMTY/r27etax7FLDXfccYfbdvLz88u0+fHHHzcxMTHmyJEjZeYde8zKgw8+aDZs2GBef/31Mo9ZSUlJMenp6WbTpk1m3bp15q677jI+Pj4ef69fXTxWx1x99dXmtttus7Ebz5m6eLxWrVpl5s2bZzZt2mS++OILc91115mmTZuagoICi3v27Khtx8sY4/o8cXFxJjEx0axbt86sX7/eNf+rr74y3t7e5p///KfJyckx//znP2vEI4zq4rF69tlnjZ+fn3nvvfcqfKSSpyNk1kFLliwxksq8Bg0aZDZv3lzuPElmyZIlbusZP368ufDCC01QUJCJj483X375pWteZmamad++vXE6nSYgIMDExsaap556yhw4cMBV89RTT5W7ncaNG7ttp7S01Fx44YXmf//3fyv8TEuXLjWXX3658fPzM02aNDEvv/yy2/yRI0eaRo0aGT8/P3P++eebhIQEs2LFiqrvxHOkLh4rY4zZuHGjkVThzfSeqi4er6VLl5qWLVsaf39/Ex4ebpKSkszvv/9e9Z14DtXG43U663n33XdNbGys8fX1NRdffLGZN29elffhuVIXj1Xjxo3LrXnqqafOZFeeUw5jjDlpVycAAABQSdyTCQAAAOsImQAAALCOkAkAAADrCJkAAACwjpAJAAAA6wiZAAAAsI6QCQAAAOsImQAAALCOkAkAcHE4HFqwYEF1NwNALUDIBIBzLD8/X8OGDVOjRo3k7++v6Oho9ejRQytXrqzupgGANT7V3QAAqGtuueUWlZSUaObMmWrWrJl27Nihzz77TH/88Ud1Nw0ArKEnEwDOoT179mj58uV69tln1aVLFzVu3Fh//etfNWbMGPXq1UuSNHnyZLVp00bBwcFq2LChkpOTtX//ftc6ZsyYofPOO08fffSRYmNjFRQUpFtvvVUHDhzQzJkz1aRJE4WFhWn48OEqLS11LdekSRM988wzSkxMVL169RQTE6OpU6eetL2///67brvtNoWFhSk8PFz9+vXTli1bzsq+AVC7EDIB4ByqV6+e6tWrpwULFqioqKjcGi8vL73wwgvKzs7WzJkz9fnnn2v06NFuNQcPHtQLL7yguXPnKj09XUuXLtXNN9+sjz/+WB9//LFmzZql1157Te+9957bchMnTlTbtm31zTffaMyYMXrwwQe1aNGicttx8OBBdenSRfXq1dMXX3yh5cuXq169eurZs6eKi4vt7BAAtZbDGGOquxEAUJfMmzdPQ4cO1aFDh3TFFVeoU6dOGjBggNq2bVtu/bvvvqu///3v2rVrl6Q/ezLvuusu/fzzz2revLkk6d5779WsWbO0Y8cO1atXT5LUs2dPNWnSRK+88oqkP3syW7ZsqU8++cS17gEDBmjv3r36+OOPJf058Gf+/Pm68cYb9cYbb2jChAnKycmRw+GQJBUXF+u8887TggULlJCQcHZ2EIBagZ5MADjHbrnlFm3btk0ffPCBevTooaVLl+qKK67QjBkzJElLlixR9+7d1aBBA4WEhOjOO+/U7t27deDAAdc6goKCXAFTkqKiotSkSRNXwDw2LT8/323b8fHxZd7n5OSU287MzEz9/PPPCgkJcfXA1q9fX4cPH9amTZvOdDcAqOUY+AMA1SAgIEDdu3dX9+7d9eSTT+ruu+/WU089pS5duuiGG27Qvffeq2eeeUb169fX8uXLNWTIEJWUlLiW9/X1dVufw+Eod9rRo0dP2ZZjvZQnOnr0qOLi4jR79uwy884///zT+ZgA6jBCJgB4gFatWmnBggX6+uuvdeTIEU2aNEleXn9ebHrnnXesbWfVqlVl3l988cXl1l5xxRV6++23FRkZqdDQUGttAFA3cLkcAM6h3bt367rrrlNaWpq+++47bd68We+++64mTJigfv36qXnz5jpy5IimTp2qX375RbNmzXLdU2nDV199pQkTJujHH3/Uiy++qHfffVcPPPBAubV33HGHIiIi1K9fP3355ZfavHmzli1bpgceeEC//fabtTYBqJ3oyQSAc6hevXpq3769nn/+eW3atEklJSVq2LChhg4dqv/93/9VYGCgJk+erGeffVZjxozRtddeq/Hjx+vOO++0sv1Ro0YpMzNTTz/9tEJCQjRp0iT16NGj3NqgoCB98cUXeuSRR3TzzTdr3759atCggbp27UrPJoBTYnQ5ANQRTZo00ciRIzVy5MjqbgqAOoDL5QAAALCOkAkAAADruFwOAAAA6+jJBAAAgHWETAAAAFhHyAQAAIB1hEwAAABYR8gEAACAdYRMAAAAWEfIBAAAgHWETAAAAFhHyAQAAIB1/x/qQ/DxKjCnYgAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 1000x800 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "# Your data\n",
    "data = {\n",
    "    \"Sample\": [12652705, 12652705, 12652705, 12652705, 12652705,\n",
    "               12652707, 12652707, 12652707, 12652707, 12652707,\n",
    "               12652709, 12652709, 12652709, 12652709, 12652709,\n",
    "               12652710, 12652710, 12652710, 12652710, 12652710,\n",
    "               12652712, 12652712, 12652712, 12652712, 12652712],\n",
    "    \"Tool\": [\"BCFTOOLS\", \"VARSCAN2\", \"MUTECT2\", \"HAPLOTYPECALLER\", \"DEEPVARIANT\",\n",
    "             \"BCFTOOLS\", \"VARSCAN2\", \"MUTECT2\", \"HAPLOTYPECALLER\", \"DEEPVARIANT\",\n",
    "             \"BCFTOOLS\", \"VARSCAN2\", \"MUTECT2\", \"HAPLOTYPECALLER\", \"DEEPVARIANT\",\n",
    "             \"BCFTOOLS\", \"VARSCAN2\", \"MUTECT2\", \"HAPLOTYPECALLER\", \"DEEPVARIANT\",\n",
    "             \"BCFTOOLS\", \"VARSCAN2\", \"MUTECT2\", \"HAPLOTYPECALLER\", \"DEEPVARIANT\"],\n",
    "    \"HET_count\": [45612, 44437, 73089, 45939, 54559,\n",
    "                  43594, 42539, 71549, 43821, 53536,\n",
    "                  45702, 44351, 73480, 46137, 55884,\n",
    "                  44941, 44631, 74705, 45461, 74705,\n",
    "                  45785, 44899, 72893, 46192, 57871],\n",
    "    \"HOM_count\": [28467, 23559, 0, 27418, 29097,\n",
    "                  28846, 23234, 0, 27825, 29417,\n",
    "                  28462, 22844, 0, 27468, 29095,\n",
    "                  30400, 24262, 0, 29405, 31016,\n",
    "                  28006, 23161, 0, 26853, 28620]\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Create unique x-values for each sample\n",
    "samples = df['Sample'].unique()\n",
    "x = np.arange(len(samples))\n",
    "\n",
    "# Width of each bar\n",
    "bar_width = 0.15  # Adjust the bar width as needed\n",
    "\n",
    "# Create a figure and axis\n",
    "fig, ax = plt.subplots(figsize=(10, 8))\n",
    "\n",
    "# Define custom colors for HET_count and HOM_count\n",
    "het_color = '#FF8000'\n",
    "hom_color = '#4876FF'\n",
    "\n",
    "# Grouped bar chart for each tool (HET_count in blue and HOM_count in orange)\n",
    "tools = df['Tool'].unique()\n",
    "legend_labels = []  # To store legend labels in the correct order\n",
    "\n",
    "for i, tool in enumerate(tools):\n",
    "    het_counts = df[df['Tool'] == tool]['HET_count']\n",
    "    hom_counts = df[df['Tool'] == tool]['HOM_count']\n",
    "\n",
    "    # Comment out the line that creates labels for each bar\n",
    "    # labels = [f'{sample}' for sample in samples]\n",
    "\n",
    "    legend_labels.append(f'{tool} (HET_count)')\n",
    "    legend_labels.append(f'{tool} (HOM_count')\n",
    "\n",
    "    ax.bar(x - bar_width * (len(tools) - 1) / 2 + i * bar_width, het_counts, bar_width, label=f'{tool} (HET_count)', color=het_color, edgecolor='black')\n",
    "    ax.bar(x - bar_width * (len(tools) - 1) / 2 + i * bar_width, hom_counts, bar_width, label=f'{tool} (HOM_count)', bottom=het_counts, color=hom_color, edgecolor='black')\n",
    "\n",
    "    # Comment out the lines that add labels for the overall counts on the bars (rotated)\n",
    "    # for xi, overall_count in zip(x, het_counts + hom_counts):\n",
    "    #     ax.text(xi - bar_width * (len(tools) - 1) / 2 + i * bar_width, overall_count, f'{overall_count}', ha='center', va='bottom', fontsize=10, rotation=90)\n",
    "\n",
    "# Customize the plot\n",
    "ax.set_xlabel('Sample')\n",
    "ax.set_ylabel('Count')\n",
    "ax.set_title('Grouped Bar Chart for Tool Counts')\n",
    "ax.set_xticks(x)\n",
    "ax.set_xticklabels(samples)\n",
    "\n",
    "# Rotate x-axis labels for better readability if needed\n",
    "plt.xticks(rotation=0)\n",
    "\n",
    "# Place both legends outside the plot to the right\n",
    "ax.legend(title='Tool', labels=legend_labels, bbox_to_anchor=(1.15, 1), loc='upper left')\n",
    "custom_legend_labels = [\"1. BCFTOOLS\", \"2. VARSCAN2\", \"3. MUTECT2\", \"4. HAPLOTYPECALLER\", \"5. DEEPVARIANT\"]\n",
    "custom_legend = plt.legend(custom_legend_labels, title='Tools', bbox_to_anchor=(1.15, 0.5), loc='center left')\n",
    "\n",
    "# Show the plot\n",
    "plt.tight_layout()\n",
    "\n",
    "# Combine both legends\n",
    "ax.add_artist(custom_legend)\n",
    "plt.savefig(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/Result/ALL_samples_bargraph.svg', dpi=450, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "fedb6501",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAsUAAAIhCAYAAACmHseMAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAAB8s0lEQVR4nO3dd1hUx8IG8HdZ+tISlaaICCIqaFSUiAUrtlgTY8WuqGDDBoqRaBSTWDCJWBLEEgsa602MEY2NcG2oUYOJGokFwRpARUCW+f7wcj7XBWSRvu/vec5z3Tmzc+YM5O7rODtHJoQQICIiIiLSYjpl3QEiIiIiorLGUExEREREWo+hmIiIiIi0HkMxEREREWk9hmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdZjKCYiIiIircdQTERUQtavXw+ZTJbnMX36dKlerVq1VM6ZmJjAw8MDGzduVGmvbdu2KvUMDQ1Rv359fPbZZ8jKyirt21MxfPjwfO/11WP48OF53ouRkREaNWqEsLAw5OTk5Nuuvr4+HB0dMX36dKSlpZXR3RJRZaRb1h0gIqrsIiMj4eLiolJma2ur8rply5ZYsmQJAODOnTtYsmQJhg0bhmfPnmH8+PFSvdq1a2Pz5s0AgAcPHuC7777D3LlzcevWLaxdu7aE7yR/c+fOxbhx46TX586dg5+fHxYtWoR27dpJ5dWqVZP+/Oq93L9/H6tXr8bUqVORlJSEzz//XKpnZGSEX3/9FQCQkpKCH374AUuXLsXFixdx8ODBkr41ItISMiGEKOtOEBFVRuvXr8eIESNw5swZuLu751uvVq1acHV1xY8//iiVpaSkwN7eHpaWlrh27RqAl7OrDx8+xOXLl6V62dnZqF+/Pm7evInU1FQYGhqW3A1p4OjRo2jXrh127NiBjz76SO18Xvfy4sULuLi4IDk5GSkpKdDT08Pw4cPxww8/4OnTpyrvb9++PY4cOYIbN27AwcGhxO+HiCo/Lp8gIiqHLCwsULduXdy8ebPAerq6unjvvfeQlZWFlJSUAus2btwYrVu3VitXKpWoXr06+vbtK5WtWrUKjRo1gomJCUxNTeHi4oLZs2cX6V4KS09PD02bNkV6ejoePHhQYN3cv2Tcu3evRPtERNqDoZiIqIQplUpkZ2erHG/y4sUL3Lx5U2W5QX4SEhJgYWHxxrojRoxATEyMNPOc6+DBg7h79y5GjBgBANi2bRsmTJgALy8v7N69G3v27MHUqVPx7NmzN/blbf3999/Q1dXFO++8U2C9hIQE6Orqonbt2iXeJyLSDlxTTERUwt5//321shcvXkBX9///L1gIIYXlO3fuICQkBPfv38eMGTPU3ptb7+HDh1i1ahXOnj2L1atXQy6XF9iPwYMHY8aMGVi/fj0WLlwola9fvx5WVlbo2rUrAOC3336DhYUFvvrqK6lOhw4dNLjjwsu9lwcPHuCrr77CuXPn0K9fPxgZGeVZLzU1FTt27MCuXbsQGBgIS0vLEukXEWkfhmIiohK2ceNG1KtXT6Xs1UAMAPv374eenp702sjICBMnTsRnn32mUu+PP/5QqQcAQUFB8PX1fWM/qlSpgh49emDDhg1YsGABdHR08O+//2Lv3r2YNGmS1KfmzZvjm2++wcCBAzFgwAC0bNkSVatW1eieC+P1e9HT08PgwYOxcuVKlXrPnj1Tu+eBAweqBHsiorfFUExEVMLq1atX4BftAKBVq1ZYvnw5ZDIZjI2N4ejoCH19fbV6jo6O2LZtG4QQuHnzJj777DOEhoaiYcOGGDBgwBv7MnLkSOzcuRPR0dHo3Lkztm7diszMTGmrNADw8fFBdnY2vv32W3z44YfIyclBs2bN8Nlnn6FTp04a339+cu8ld3s5BwcHGBsbq9UzMjLC8ePHAQDJyclYunQptm7dioYNGyIwMLDY+kNE2o1riomIygFzc3O4u7ujadOmqFevXp6BGAAMDQ3h7u6OZs2a4aOPPsLhw4dhZWWFKVOmqO3QkJfOnTvD1tYWkZGRAF5uF+fh4YH69eur1BsxYgRiY2ORmpqKn376CUIIfPDBB2/84p8mcu+ladOmaNCgQZ6BGAB0dHTg7u4Od3d3fPDBBzhw4AAaNGiATz/9FLdv3y62/hCRdmMoJiKqwKpUqYLFixfj3r17+Prrr99YXy6Xw8fHB3v27MGJEydw9uxZjBw5Mt/6CoUCXbt2xZw5c5CVlYU//vijOLtfJAYGBli5ciUyMjLUlpcQERUVQzERUQU3dOhQuLm5YcmSJYV6ytvIkSORmZmJQYMGwcjICP3791c5P2bMGEyaNAlRUVE4fvw4tm/fjpCQEJibm6NZs2YldRsa8fLyQrdu3RAZGYmEhISy7g4RVQIMxUREFZyOjg4WL16Mx48fIyws7I31nZ2d4enpiTt37qBv374wNzdXOd+6dWtcvnwZkydPRqdOnTB16lQ4OzvjxIkThdoirrR8/vnnUCqVWLBgQVl3hYgqAT7RjoiIiIi0HmeKiYiIiEjrMRQTERERkdZjKCYiIiIircdQTERERERaj6GYiIiIiLQeQzERERERaT3dsu5ARZWTk4O7d+/C1NQUMpmsrLtDRERERK8RQuDJkyewtbWFjk7Bc8EMxUV09+5d2NnZlXU3iIiIiOgNbt++jRo1ahRYh6G4iExNTQG8HGQzM7My7g0RERERvS4tLQ12dnZSbisIQ3ER5S6ZMDMzYygmIiIiKscKs9SVX7QjIiIiIq3HUExEREREWo+hmIiIiIi0HtcUlyAhBLKzs6FUKsu6K0TFQi6XQ1dXl9sQEhFRpcNQXEKysrKQlJSE9PT0su4KUbEyNjaGjY0N9PX1y7orRERExYahuATk5OQgISEBcrkctra20NfX58waVXhCCGRlZeHBgwdISEhAnTp13rgROhERUUXBUFwCsrKykJOTAzs7OxgbG5d1d4iKjZGREfT09HDz5k1kZWXB0NCwrLtERERULMp8mic8PBwODg4wNDRE06ZNceLEiXzrxsTEoGXLlqhSpQqMjIzg4uKC5cuXq9R58eIF5s+fD0dHRxgaGqJRo0Y4cOCASp3Q0FA0a9YMpqamsLS0RO/evfHXX38V+71xFo0qI/5eExFRZVSmn25RUVGYMmUK5syZg/Pnz6N169bo2rUrbt26lWd9hUIBf39/HD9+HFeuXEFwcDCCg4Oxdu1aqU5wcDDWrFmDr7/+GvHx8Rg3bhz69OmD8+fPS3WOHTsGPz8/nDx5EtHR0cjOzoa3tzeePXtW4vdMREREROWPTAghyuriHh4eaNKkCVatWiWV1atXD71790ZoaGih2ujbty8UCgU2bdoEALC1tcWcOXPg5+cn1enduzdMTEzw/fff59nGgwcPYGlpiWPHjqFNmzaFum5aWhrMzc2Rmpqq9kS7jIwMJCQkSDPgRJUJf7+JiKiiKCivva7MZoqzsrIQFxcHb29vlXJvb2/ExsYWqo3z588jNjYWXl5eUllmZqbaB7WRkRFiYmLybSc1NRUA8O677+ZbJzMzE2lpaSpHaVDmCPz370fYeyER//37EZQ5ZfZ3GCIiIqJKq8xC8cOHD6FUKmFlZaVSbmVlheTk5ALfW6NGDRgYGMDd3R1+fn4YPXq0dK5z585YtmwZrl27hpycHERHR2Pv3r1ISkrKsy0hBAICAtCqVSu4urrme83Q0FCYm5tLh52dnQZ3WzQHLieh1ee/YuC3JzF52wUM/PYkWn3+Kw5czvteisPw4cMhk8mko0qVKujSpQsuXrwo1RFCYO3atfDw8ICJiQksLCzg7u6OsLAwaQu6kJAQlXZyj++++y7P8lePkJAQ6VobNmxA8+bNoVAoYGpqijZt2uDHH39U67dSqcTy5cvRsGFDGBoawsLCAl27dsVvv/2mUm/9+vWwsLAokbEjIiKiiqvMvzHz+lZlQog3bl924sQJnD17FqtXr0ZYWBi2bt0qnVuxYgXq1KkDFxcX6Ovrw9/fHyNGjIBcLs+zLX9/f1y8eFGljbwEBQUhNTVVOm7fvl3IOyyaA5eTMP77c0hKzVApT07NwPjvz5VoMO7SpQuSkpKQlJSEw4cPQ1dXFx988IF03sfHB1OmTEGvXr1w5MgRXLhwAXPnzsXevXtx8OBBqV6DBg2kdnIPHx8fldfTpk1Tqzd9+nQAwPTp0+Hr64uPP/4Yv//+O06fPo3WrVujV69e+Oabb6TrCCEwYMAAzJ8/H5MmTcKVK1dw7Ngx2NnZoW3bttizZ0+JjRURERFVDmW2JVvVqlUhl8vVZoXv37+vNnv8OgcHBwCAm5sb7t27h5CQEAwcOBAAUK1aNezZswcZGRl49OgRbG1tERgYKL3nVRMnTsS+fftw/Phx1KhRo8BrGhgYwMDAQJNbVCGEwPMXhXuynTJHYN6+P5DXQgkBQAYgZF88WjpVhVyn4L9AGOnJNd4j2cDAANbW1gAAa2trzJo1C23atMGDBw9w5MgRbN68GXv27EGvXr2k99SqVQs9e/ZUWVaiq6srtfOqV8tMTEzyrHfy5EksXboUX331FSZOnCiVL1y4EBkZGQgICECvXr1gZ2eH7du344cffsC+ffvQo0cPqe7atWvx6NEjjB49Gp06dYJCodBoHIiIiKj4KHOUOHf/HB6kP0A142poYtkEcp28Jy3LQpmFYn19fTRt2hTR0dHo06ePVB4dHa0Stt5ECIHMzEy1ckNDQ1SvXh0vXrzAzp078fHHH6u8Z+LEidi9ezeOHj2aZ2Aubs9fKFH/k1+KpS0BIDktA24hB99YN35+ZxjrF/3H/PTpU2zevBlOTk6oUqUKNm/ejLp16+b5M5LJZDA3Ny/ytV61detWmJiYwNfXV+3ctGnTsGzZMuzcuRNTpkzBli1b4OzsrBKIX627a9cuREdHo3fv3sXSNyIiItLMoZuHsPj0YtxLvyeVWRlbIbB5IDradyzDnv2/Mn14R0BAAHx8fODu7o4WLVpg7dq1uHXrFsaNGwfg5ZKFxMREbNy4EQCwcuVK1KxZEy4uLgBe7lu8ZMkSlZnEU6dOITExEe+99x4SExMREhKCnJwczJw5U6rj5+eHLVu2YO/evTA1NZVmq83NzWFkZFRat19u/fjjjzAxMQEAPHv2DDY2Nvjxxx+ho6ODa9euoW7duoVq59KlS1I7AFC/fn2cPn26UO+9evUqHB0d83yUsK2tLczNzXH16lWpbr169fJsJ7c8ty4RERGVrkM3DyHgaADEa/8Gfj/9PgKOBmBZ22XlIhiXaSju378/Hj16hPnz5yMpKQmurq7Yv38/7O3tAQBJSUkqexbn5OQgKCgICQkJ0NXVhaOjIxYvXqwym5iRkYHg4GDcuHEDJiYm6NatGzZt2qTy5arcLeDatm2r0p/IyEgMHz68RO7VSE+O+PmdC1X3dMJjDI8888Z660c0Q3OH/HfMyL2uptq1ayeN0ePHjxEeHo6uXbvi9OnThVrznatu3brYt2+f9Pptlp+8TpN+AOpr14mIiKjkKXOUWHx6sVogBgABARlk+Pz052hn167Ml1KU+WOeJ0yYgAkTJuR5bv369SqvJ06cqDIrnBcvLy/Ex8cXWKcstmaWyWSFXsbQuk412JgbIjk1I891xTIA1uaGaF2n2hvXFBeFQqGAk5OT9Lpp06YwNzfHt99+C2dnZ1y5cqVQ7ejr66u0owlnZ2fExMQgKytLbbb47t27SEtLQ506daS6+f3Mc/uaW5eIiIhKz7n751SWTLxOQCA5PRnn7p9DM+tmpdgzdWW++wSpk+vIMK9HfQAvA/Crcl/P61G/RAJxXmQyGXR0dPD8+XMMGjQIV69exd69e9XqCSGkPZ/f1oABA/D06VOsWbNG7dySJUugp6eHDz/8UKp77do1/Oc//1Gru3TpUlSpUgWdOnUqln4RERFR4T1If1Cs9UpSmc8UU966uNpg1ZAm+PQ/8SrbslmbG2Jej/ro4mpTYtfOzMyU1ln/+++/+Oabb/D06VP06NEDXl5e2L17NwYOHIi5c+eiU6dOqFatGi5duoTly5dj4sSJxfKFthYtWmDy5MmYMWMGsrKy0Lt3b7x48QLff/89VqxYgbCwMGmv6AEDBmDHjh0YNmwYvvzyS3To0AFpaWlYuXIl9u3bhx07dqjsPKFUKnHhwgWV6+nr66N+/fpv3W8iIiL6f9WMqxVrvZLEUFyOdXG1Qaf61jid8Bj3n2TA0tQQzR3eLfEZ4gMHDsDG5mXoNjU1hYuLC3bs2CGtwd6yZQvWrl2LdevW4bPPPoOuri7q1KmDoUOHonPnwq2bLoywsDA0bNgQq1atwty5cyGTydCkSRPs2bNHZacJmUyG7du3Y8WKFVi+fDn8/PxgYGCAFi1a4MiRI2jVqpVKu0+fPkXjxo1Vyuzt7fHPP/8UW9+JiIgIaGLZBFbGVriffj/PdcUyyGBlbIUmlk3KoHev9UWUxQLbSqCgZ2lnZGQgISEBDg4Oao+cJqro+PtNRESayN19AoBKMJb9b1FoSe4+UVBeex3XFBMRERFRielo3xHL2i6DpbGlSrmVsVW52Y4N4PIJIiIiIiphHe07op1dOz7RjoiIiIi0m1xHXubbrhWEyyeIiIiISOsxFBMRERGR1mMoJiIiIiKtx1BMRERERFqPoZiIiIiItB5DMRERERFpPW7JVt7lKIGbscDTe4CJFWDvCZSjPf2IiIiIKgPOFJdn8fuAMFdgwwfAzlEv/zfM9WV5CRk+fDhkMhlkMhn09PRgZWWFTp06Yd26dcjJyZHq1apVS6r36rF48WIAwD///JPneZlMhpMnTwIA1q9fr1JuY2ODjz/+GAkJCcjKykLVqlXx2Wef5dnP0NBQVK1aFVlZWVJZ3bp1oa+vj8TERLX6bdu2la6jr68PR0dHBAUFITMzU6WeTCbDnj171N4/duxYyOVybNu2Te1cSEgIZDIZxo0bp1J+4cIFyGQy/PPPP1Kdgo5//vkn7x8KERERlTiG4vIqfh+wfSiQdle1PC3pZXkJBuMuXbogKSkJ//zzD37++We0a9cOkydPxgcffIDs7Gyp3vz585GUlKRyTJw4UaWtQ4cOqdVp2rSpdN7MzAxJSUm4e/cutmzZggsXLqBnz56Qy+UYMmQI1q9fDyEEXhcZGQkfHx/o6+sDAGJiYpCRkYF+/fph/fr1ed7XmDFjkJSUhOvXr+OLL77AypUrERIS8sbxSE9PR1RUFGbMmIGIiIg86xgaGiIiIgJXr17N8/z06dNVxqBGjRpq42dnZ/fGvhAREVHJ4PKJ0iIE8CK9cHVzlMDPMwGoh8GXZTLgwCygdts3L6XQMwZkMo26amBgAGtrawBA9erV0aRJE7z//vvo0KED1q9fj9GjRwMATE1NpXr5qVKlSoF1ZDKZdN7Gxgbz5s3DkCFDcP36dYwaNQorVqzA8ePH4eXlJb3nxIkTuHbtGkaNGiWVRUREYNCgQfDy8oKfnx9mz54N2Wv3bWxsLF2rZs2a2LJlCw4ePIjQ0NAC72HHjh2oX78+goKCYGNjg3/++Qe1atVSqVO3bl1YWloiODgY27dvV2vDxMQEJiYm0mu5XF6o8SMiIqLSwVBcWl6kA4tsi6kx8XIGeXEhZhZn3wX0FW99xfbt26NRo0bYtWuXFIpLgpGREQDgxYsXcHNzQ7NmzRAZGakSitetW4fmzZvD1dUVAPDkyRPs2LEDp06dgouLC549e4ajR4+iXbt2+V7n999/x2+//aYWbvMSERGBIUOGwNzcHN26dUNkZCQ+/fRTtXqLFy9Gs2bNcObMGTRrVn4fY0lERETquHyCCs3FxUVl3eusWbOkGdDc4+jRoyrv8fT0VKujVCrzbP/OnTv48ssvUaNGDTg7OwMARo4ciR9++AFPnz4FADx9+hQ7duxQmSXetm0b6tSpgwYNGkAul2PAgAF5LnMIDw+HiYkJDAwM8N577+HBgweYMWNGgfd87do1nDx5Ev379wcADBkyBJGRkSrrq3M1adIEH3/8MQIDAwtsk4iIiMofzhSXFj3jl7O2hXEzFtj80ZvrDf7h5W4Ub7puMRFCqCxJmDFjBoYPH65Sp3r16iqvo6KiUK9ePZUyufz/l3ykpqbCxMQEQgikp6ejSZMm2LVrl7RWeODAgQgICEBUVBRGjRqFqKgoCCEwYMAAqY3cmdxcQ4YMQZs2bZCSkgILCwupfPDgwZgzZw7S0tLw+eefw8zMDB9++GGB9xwREYHOnTujatWqAIBu3bph1KhROHToELy9vdXqf/bZZ6hXrx4OHjwIS0vLAtsmIiKi8oOhuLTIZIVfxuDYHjCzffmlujzXFctenndsX6rbs125cgUODg7S66pVq8LJyanA99jZ2RVYx9TUFOfOnYOOjg6srKygUKiOkbm5OT766CNERkZi1KhRiIyMxEcffQQzMzMAQHx8PE6dOoUzZ85g1qxZ0vuUSiW2bt2K8ePHq7SV25fvv/8eDRo0QEREhMqs86uUSiU2btyI5ORk6OrqqpRHRETkGYodHR0xZswYBAYG5vulPCIiIip/GIrLIx050OXzl7tMQAbVYPy/mdoui0s1EP/666+4dOkSpk6dWqzt6ujovDFYjxo1Cm3btsWPP/6I3377DYsWLZLORUREoE2bNli5cqXKezZt2oSIiAiVUPwqPT09zJ49G0FBQRg4cCCMjdVn1Pfv348nT57g/PnzKrPbf/75JwYPHoxHjx6hSpUqau/75JNP4OjomOf2bURERFQ+cU1xeVW/J/DxRsDMRrXczPZlef2eJXbpzMxMJCcnIzExEefOncOiRYvQq1cvfPDBBxg6dKhU78mTJ0hOTlY50tLSVNp69OiRWp2MjAyN+uPl5QUnJycMHToUTk5OaNOmDYCXX8bbtGkTBg4cCFdXV5Vj9OjRiIuLw++//55vu4MGDYJMJkN4eHie5yMiItC9e3c0atRIpe0PP/wQ1apVw/fff5/n+6ysrBAQEICvvvpKo/skIiKissNQXJ7V7wlMuQwM+xH4MOLl/065VKKBGAAOHDgAGxsb1KpVC126dMGRI0fw1VdfYe/evSozpp988glsbGxUjpkzZ6q01bFjR7U6eT0c401GjhyJf//9FyNHjpTK9u3bh0ePHqFPnz5q9evUqQM3N7cClzDo6+vD398fX3zxhfRFvlz37t3DTz/9lOeaY5lMhr59+xbY9owZM1S2YCMiIqLyTSbyejICvVFaWhrMzc2RmpoqrW/NlZGRgYSEBDg4OMDQ0LCMekhUMvj7TUREFUVBee11nCkmIiIiIq3HUExEREREWo+hmIiIiIi0HkMxEREREWk9hmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdZjKCYiIiIiradb1h2ggilzlDh3/xwepD9ANeNqaGLZBHId+ZvfSERERESFxpnicuzQzUPovLMzRv4yErNOzMLIX0ai887OOHTzUIlcr0ePHujYsWOe5/773/9CJpPh3LlzAICxY8dCLpdj27ZtanVDQkIgk8kgk8mgo6MDW1tbDB48GLdv31apd+PGDQwcOBC2trYwNDREjRo10KtXL1y9elWl3pEjR9CtWzdUqVIFxsbGqF+/PqZNm4bExES1a9etWxf6+vp5nmvbti1kMplan8PCwlCrVi3p9a5du9CpUydUq1YNZmZmaNGiBX755Ze8B42IiIgqBYbicurQzUMIOBqAe+n3VMrvp99HwNGAEgnGo0aNwq+//oqbN2+qnVu3bh3ee+89NGnSBOnp6YiKisKMGTMQERGRZ1sNGjRAUlIS7ty5g6ioKFy6dAkff/yxdD4rKwudOnVCWloadu3ahb/++gtRUVFwdXVFamqqVG/NmjXo2LEjrK2tsXPnTsTHx2P16tVITU3F0qVLVa4ZExODjIwM9OvXD+vXr8+zX4aGhggODsaLFy/yHYfjx4+jU6dO2L9/P+Li4tCuXTv06NED58+fL2j4iIiIqALj8olSIoTA8+znhaqrzFEi9HQoBIR6O/8rW3x6MTysPd64lMJI1wgymaxQ1/3ggw9gaWmJ9evXY968eVJ5bghetGgRAGDHjh2oX78+goKCYGNjg3/++UdlphUAdHV1YW1tDQCwtbXFmDFjMGnSJKSlpcHMzAzx8fG4ceMGfv31V9jb2wMA7O3t0bJlS6mNO3fuYNKkSZg0aRKWL18uldeqVQtt2rRBSkqKyjUjIiIwaNAgeHl5wc/PD7Nnz1a794EDB+I///kPvv32W0yYMCHPcQgLC1N5vWjRIuzduxf/+c9/0Lhx4zcPJBEREVU4DMWl5Hn2c3hs8Si29u6l34PnNs831js16BSM9YwL1aauri6GDh2K9evX45NPPpEC5Y4dO5CVlYXBgwcDeBk+hwwZAnNzc3Tr1g2RkZH49NNP8203OTkZu3btglwuh1z+MsRXq1YNOjo6+OGHHzBlyhSp/FW51505c2ae7VpYWEh/fvLkCXbs2IFTp07BxcUFz549w9GjR9GuXTuV95iZmWH27NmYP38+hg0bBoVC8cZxycnJwZMnT/Duu+++sS4RERFVTFw+QSpGjhyJf/75B0ePHpXK1q1bh759++Kdd97BtWvXcPLkSfTv3x8AMGTIEERGRiInJ0elnUuXLsHExATGxsawsbHB0aNH4efnJ4XQ6tWr46uvvsInn3yCd955B+3bt8eCBQtw48YNqY1r167BzMwMNjY2b+z3tm3bUKdOHTRo0AByuRwDBgzId2nHhAkTYGhoiGXLlhVqTJYuXYpnz56pLP8gIiKiyoUzxaXESNcIpwadKlTduHtxmHA473/af1V4h3A0tWr6xutqwsXFBZ6enli3bh3atWuHv//+GydOnMDBgwcBvJwl7ty5M6pWrQoA6NatG0aNGoVDhw7B29tbaqdu3brYt28fMjMzsXfvXuzYsQMLFy5UuZafnx+GDh2KI0eO4NSpU9ixYwcWLVqEffv2oVOnThBCFHrpR+7sda4hQ4ZISyxenVEGAAMDA8yfPx/+/v4YP358ge1u3boVISEh2Lt3LywtLQvVFyIiIqp4OFNcSmQyGYz1jAt1eNp6wsrYCjLkHQhlkMHa2Bqetp5vbKuwofJVo0aNws6dO5GWlobIyEjY29ujQ4cOUCqV2LhxI3766Sfo6upCV1cXxsbGePz4sdqsrL6+PpycnNCgQQPMnj0b7733Xp4B1NTUFD179sTChQvx+++/o3Xr1vjss88AAM7OzkhNTUVSUlKB/Y2Pj8epU6cwc+ZMqV/vv/8+nj9/jq1bt+b5niFDhqBWrVrStfISFRWFUaNGYfv27fnuykFERESVA0NxOSTXkSOweSAAqAXj3Nezms8qsf2KP/74Y8jlcmzZsgUbNmzAiBEjIJPJsH//fjx58gTnz5/HhQsXpGPHjh3Ys2cPHj16lG+bc+fOxdatW6Ut3fIik8mk9cAA8NFHH0FfXx9ffPFFnvVzv2gXERGBNm3a4Pfff1fp18yZM/NdQqGjo4PQ0FCsWrUK//zzj9r5rVu3Yvjw4diyZQu6d++eb5+JiIiocmAoLqc62nfEsrbLYGms+k/2VsZWWNZ2GTral9zMpYmJCfr374/Zs2fj7t27GD58OICX4bN79+5o1KgRXF1dpePDDz9EtWrV8P333+fbZu3atdGrVy988sknAIALFy6gV69e+OGHHxAfH4/r168jIiIC69atQ69evQAAdnZ2WL58OVasWIFRo0bh2LFjuHnzJn777Tf4+vpiwYIFePHiBTZt2oSBAweq9MnV1RWjR49GXFwcfv/99zz71L17d3h4eGDNmjUq5Vu3bsXQoUOxdOlSvP/++0hOTkZycrLKVnFERERUyQgqktTUVAFApKamqp17/vy5iI+PF8+fP3/r62Qrs8XppNPip79/EqeTTotsZfZbt1kYsbGxAoDw9vYWQgiRnJwsdHV1xfbt2/OsP3HiROHm5iaEEGLevHmiUaNGanV+++03AUCcPHlSPHjwQEyaNEm4uroKExMTYWpqKtzc3MSSJUuEUqlUeV90dLTo3LmzeOedd4ShoaFwcXER06dPF3fv3hU//PCD0NHREcnJyXn2y83NTUycOFEIIYSXl5eYPHlynvdpb28vlXl5eQkAasewYcMKMXKVX3H+fhMREZWkgvLa62RCCPXNcEtReHg4vvzySyQlJaFBgwYICwtD69at86wbExODWbNm4c8//0R6ejrs7e3h6+uLqVOnSnVevHiB0NBQbNiwAYmJiahbty4+//xzdOnSpcjXzUtaWhrMzc2RmpoKMzMzlXMZGRlISEiAg4MDDA0NNRgNovKPv99ERFRRFJTXXlemyyeioqIwZcoUzJkzB+fPn0fr1q3RtWtX3Lp1K8/6CoUC/v7+OH78OK5cuYLg4GAEBwdj7dq1Up3g4GCsWbMGX3/9NeLj4zFu3Dj06dNH5Wlkml6XiIiIiCq3Mp0p9vDwQJMmTbBq1SqprF69eujduzdCQ0ML1Ubfvn2hUCiwadMmAC+fnjZnzhz4+flJdXr37g0TExNpzWtxXJczxaSt+PtNREQVRYWYKc7KykJcXJzK3rYA4O3tjdjY2EK1cf78ecTGxsLLy0sqy8zMVPugNjIyQkxMzFtdNzMzE2lpaSoHEREREVUOZRaKHz58CKVSCSsrK5VyKysrJCcnF/jeGjVqwMDAAO7u7vDz88Po0aOlc507d8ayZctw7do15OTkIDo6Gnv37pX2ui3qdUNDQ2Fubi4ddnZ2mt4yEREREZVTZb4l2+sPlxCFeIrZiRMncPbsWaxevRphYWEqD2hYsWIF6tSpAxcXF+jr68Pf3x8jRoyAXK66p6+m1w0KCkJqaqp03L59u7C3SERERETlXJk95rlq1aqQy+Vqs7P3799Xm8V9nYODAwDAzc0N9+7dQ0hICAYOHAgAqFatGvbs2YOMjAw8evQItra2CAwMlN5T1OsaGBjAwMBA4/skIiIiovKvzGaK9fX10bRpU0RHR6uUR0dHw9PTs9DtCCGQmZmpVm5oaIjq1asjOzsbO3fulB4IUVzXJSIiIqLKo8xmigEgICAAPj4+cHd3R4sWLbB27VrcunUL48aNA/ByyUJiYiI2btwIAFi5ciVq1qwJFxcXAC/3LV6yZAkmTpwotXnq1CkkJibivffeQ2JiIkJCQpCTk4OZM2cW+rpEREREpF3KNBT3798fjx49wvz585GUlARXV1fs378f9vb2AICkpCSVvYNzcnIQFBSEhIQE6OrqwtHREYsXL4avr69UJyMjA8HBwbhx4wZMTEzQrVs3bNq0CRYWFoW+LhERERFplzJ/ol1FVVr7FAulEuln45D94AF0q1WDsXtTyF770iBRaeI+xUREVFFUiH2K6c3SDh7E9Q4dcWvYMNydPh23hg3D9Q4dkXbwYIldc/jw4ejdu7da+dGjRyGTyZCSkqJSXrduXejr6yMxMVHtPW3btoVMJoNMJoOBgQGcnZ2xaNEiKJXKAtt81Y8//oi2bdvC1NQUxsbGaNasGdavXy+dDwkJka6R39GhQwe4ubkhKytLpe39+/dDT08PZ8+exT///KPynnfeeQdt2rTBsWPHVMYmr/Zff4T4+fPn0a9fP1hZWcHQ0BDOzs4YM2YMrl69qnZ/3t7ekMvlOHnypNq5/H4WuWrVqoWwsLA8z71+P68euddav369SrmVlRV69OiBP/74I99rEhERVVYMxeVU2sGDSJw8Bdmv7ZKRfe8eEidPKdFgXFgxMTHIyMhAv379VILqq8aMGYOkpCT89ddfmDRpEoKDg7FkyZJCtf/111+jV69e8PT0xKlTp3Dx4kUMGDAA48aNw/Tp0wEA06dPR1JSknTUqFFDWhaTe+zatQtPnjzBvHnzpLZTUlIwduxYzJkzB+7u7lL5oUOHkJSUhGPHjsHMzAzdunVDQkKCdL5Lly4qbSclJalsCfjjjz/i/fffR2ZmJjZv3owrV65g06ZNMDc3x9y5c1Xu79atW/jvf/8Lf39/REREFGpMNJV7P68eTZs2lc6bmZkhKSkJd+/exU8//YRnz56he/fuan+BICIiquzKdE2xNhFCQDx/Xri6SiXufbYQyGtlixCADLi3cBEULVq8cSmFzMjojfs+F1VERAQGDRoELy8v+Pn5Yfbs2WrXMjY2hrW1NQDA398fe/fuxZ49ezBr1qwC2759+zamTZuGKVOmYNGiRVL5tGnToK+vj0mTJqFfv37w8PCAiYmJdF4ul8PU1FS6Zq7169fD29sbvXv3hoeHB6ZMmQIbGxsEBwer1KtSpQqsra1hbW2NNWvWoEaNGjh48KC0bt3AwECt7Vzp6ekYMWIEunXrht27d0vlDg4O8PDwUJsRj4yMxAcffIDx48ejefPmCAsLg0KhKHBcNJV7P/mRyWTSeRsbG0ydOhU9e/bEX3/9BTc3t2LtCxERUXnGUFxKxPPn+KtJ0zdXLFRjL2eMrzZr/saqdc/FQWZsXDzXfcWTJ0+wY8cOnDp1Ci4uLnj27BmOHj2Kdu3aFfg+IyMj/Pvvv29s/4cffsCLFy+kGeFX+fr6Yvbs2di6dSs8PDwK1d+2bdtiwoQJGDZsGBYsWIDt27cjLi4Ourr5/ydg/L9xe/HiRaGu8csvv+Dhw4cqO5286tUvewohEBkZiZUrV8LFxQXOzs7Yvn07RowYUahrlYSUlBRs2bIFAKCnp1dm/SAiIioLXD5Ban788UeYmJioHF27dlWps23bNtSpUwcNGjSAXC7HgAEDClwCkJOTgwMHDuCXX35Bhw4d3tiHq1evwtzcHDY2Nmrn9PX1Ubt27TzX6BYkNDQUMpkMAwYMwKJFi1CvXr186z579gxBQUGQy+Xw8vKSyvMamwULFgAArl27BgDSloEFOXToENLT09G5c2cAwJAhQ0pkCYWnp6daf3PXdANAamoqTExMoFAo8M4772Dbtm3o2bNnoe6BiIioMuFMcSmRGRmh7rm4QtVNP3sWt8f6vrGe3do1MH5lPWx+19VUu3btsGrVKpWyU6dOYciQIdLriIgIlddDhgxBmzZtkJKSojIjGh4eju+++05ao+rj46OytreoCvM48NcZGRlh2rRpmDp1KiZPnpxnHU9PT+jo6CA9PR02NjZYv369yjKCvMbm3XfflfpUWBEREejfv780Uz1w4EDMmDEDf/31F+rWravRfRUkKipKLfy/+shzU1NTnDt3DtnZ2Th27Bi+/PJLrF69utiuT0REVFEwFJcSmUxW6GUMipYtoWttjex79/JeVyyTQdfKCoqWLUtkezaFQgEnJyeVsjt37kh/jo+Px6lTp3DmzBmVtcFKpRJbt27F+PHjpbLBgwdjzpw5MDAwgK2trUogK4izszNSU1Nx9+5d2NraqpzLysrCjRs30L59e43vTVdXF3K5PN9AHRUVhfr168PCwgJVqlRRO5/X2LzaZwD4888/0aJFi3z78PjxY+zZswcvXrxQCdhKpRLr1q3D559/rsktFcjOzi7f/gKAjo6OdN7FxQXJycno378/jh8/Xmx9ICIiqgi4fKIcksnlsJod9L8Xr4W3/722mh1UZvsVR0REoE2bNvj9999x4cIF6Zg5c6baEgBzc3M4OTnBzs6u0IEYAD788EPo6upi6dKlaudWr16NZ8+eYeDAgW99L6+zs7ODo6NjnoH4Tby9vVG1alV88cUXeZ7P/aLd5s2bUaNGDbXxCwsLw4YNG5Cdnf02t/BWpk6dit9//13li4JERETagDPF5ZSZtzewIgz3FoWqbMuma2UFq9lBL8+XgRcvXmDTpk2YP38+XF1dVc6NHj0aX3zxBX7//Xc0atSo0G1eunQJpqamKmXvvfcevvjiC0yfPh2Ghobw8fGBnp4e9u7di9mzZ2PatGmF/pJdccrMzETya9vk6erqomrVqlAoFPjuu+/Qr18/9OzZE5MmTYKTkxMePnyI7du349atW9i2bRsiIiLw0UcfqY2fvb09Zs2ahZ9++gm9evUC8HLN74ULF1Tqvfvuu6hZsyYAIDExUe187jkAePTokVp/LSws8n3ohpmZGUaPHo158+ahd+/eJbZzCRERUXnDUFyOmXl7w7RDh3L1RLvjx4/j0aNH6NOnj9q5OnXqwM3NDREREfjqq68K3WabNm3UyoQQmDp1KhwdHbFkyRKsWLECSqUSDRo0wKpVq8psl4YDBw6offmvbt26+PPPPwEAvXr1QmxsLEJDQzFo0CCkpaXBzs4O7du3x2effYa4uDj8/vvv+Pbbb9XaNjU1hbe3NyIiIqRQfPToUTRu3Fil3rBhw6R9oZcsWaK273NkZCTatm0LAOjYsaPadbZu3YoBAwbke4+TJ0/GV199hR07duDjjz8ueECIiIgqCT7muYhK6zHPROUNf7+JiKii4GOeiYiIiIg0wFBMRERERFqPoZiIiIiItB5DMRERERFpPYZiIiIiItJ6DMVEREREpPUYiomIiIhI6zEUExEREZHWYygmIiIiIq3HxzyXczk5AknXUvAsLRMKMwPY1LGAjo6srLtFREREVKlwprgc+/v8fWycHYs9y88jOiIee5afx8bZsfj7/P0Su+bw4cMhk8kwbtw4tXMTJkyATCbD8OHDAQBt27bFlClT1Ort2bMHMplMqiOTyfI9atWqVWC91/tx5MgRdOvWDVWqVIGxsTHq16+PadOmITExUep7QQcAhIaGolmzZjA1NYWlpSV69+6Nv/76q/gGkYiIiCochuJy6u/z93FgzWU8S8lUKX+WkokDay6XaDC2s7PDtm3b8Pz5c6ksIyMDW7duRc2aNTVqa9euXUhKSkJSUhJOnz4NADh06JBUdubMGanumDFjpPLc44svvpDOr1mzBh07doS1tTV27tyJ+Ph4rF69GqmpqVi6dClWrFih8l4AiIyMVCs7duwY/Pz8cPLkSURHRyM7Oxve3t549uxZkceMiIiIKjYunyglQghkZ+UUqm5OjsCJqKsF1jkRdQ01XN5941IKXX0daYa0sJo0aYIbN25g165dGDx4MICX4dbOzg61a9fWqK13331X+nNGRgYAoEqVKrC2tlara2xsnGc5ANy5cweTJk3CpEmTsHz5cqm8Vq1aaNOmDVJSUmBubg5zc3OV91lYWKi1eeDAAZXXkZGRsLS0RFxcHNq0aaPR/REREVHlwFBcSrKzcrB28rFia+9ZSia+m3r8jfXGrvCCnoFc4/ZHjBiByMhIKRSvW7cOI0eOxNGjRzVuqzjs2LEDWVlZmDlzZp7nLSwsitx2amoqANUAT0RERNqFyycoTz4+PoiJicE///yDmzdv4rfffsOQIUNK9Jrh4eEwMTFROTZs2AAAuHbtGszMzGBjY1Os1xRCICAgAK1atYKrq2uxtk1EREQVB2eKS4muvg7GrvAqVN2711Lw4ze/v7HeB/6NYFvH4o3XLYqqVauie/fu2LBhA4QQ6N69O6pWrVqktgpr8ODBmDNnjkqZpaUlgJfhVdNlIIXh7++PixcvIiYmptjbJiIiooqDobiUyGSyQi9jsKv/LhQWBmpfsnuVyTsGsKv/5jXFb2PkyJHw9/cHAKxcuVLtvJmZmbT04FUpKSkwMzPT+Hrm5uZwcnLK85yzszNSU1ORlJRUbLPFEydOxL59+3D8+HHUqFGjWNokIiKiionLJ8ohHR0ZWvevU2CdVh/XKfH9irt06YKsrCxkZWWhc+fOauddXFxw9uxZtfIzZ86gbt26xdqXjz76CPr6+iq7UbwqJSWl0G0JIeDv749du3bh119/hYODQzH1koiIiCoqzhSXU46NLdHF1xUnoq6pzBibvGOAVh/XgWNjyxLvg1wux5UrV6Q/v27ChAn45ptv4Ofnh7Fjx8LIyAjR0dGIiIjApk2bNL5eeno6kpOTVcoMDAzwzjvvwM7ODsuXL4e/vz/S0tIwdOhQ1KpVC3fu3MHGjRthYmKCpUuXFuo6fn5+2LJlC/bu3QtTU1Ppmubm5jAyMtK430RERFTxMRSXY46NLeHQqFqZPtGuoGUQtWrVwokTJzBnzhx4e3sjIyMDzs7OWL9+Pfr166fxtb799lt8++23KmWdO3eWtlCbMGECnJ2dsWTJEvTp0wfPnz9HrVq18MEHHyAgIKDQ11m1ahWAlw8MeVVkZKT0YBIiIiLSLjIhhCjrTlREaWlpMDc3R2pqqlpwzMjIQEJCAhwcHGBoaFhGPSQqGfz9JiKiiqKgvPY6rikmIiIiIq3HUExEREREWo+hmIiIiIi0HkMxEREREWk9huISxO8wUmXE32siIqqMGIpLgJ6eHoCX++4SVTa5v9e5v+dERESVAfcpLgFyuRwWFha4f/8+AMDY2BgyWentLUxUEoQQSE9Px/3792FhYZHnA12IiIgqKobiEmJtbQ0AUjAmqiwsLCyk328iIqLKgqG4hMhkMtjY2MDS0hIvXrwo6+4QFQs9PT3OEBMRUaXEUFzC5HI5QwQRERFROccv2hERERGR1mMoJiIiIiKtx1BMRERERFqvzENxeHg4HBwcYGhoiKZNm+LEiRP51o2JiUHLli1RpUoVGBkZwcXFBcuXL1erFxYWhrp168LIyAh2dnaYOnUqMjIypPPZ2dkIDg6Gg4MDjIyMULt2bcyfPx85OTklco9EREREVL6V6RftoqKiMGXKFISHh6Nly5ZYs2YNunbtivj4eNSsWVOtvkKhgL+/Pxo2bAiFQoGYmBj4+vpCoVBg7NixAIDNmzcjMDAQ69atg6enJ65evYrhw4cDgBSgP//8c6xevRobNmxAgwYNcPbsWYwYMQLm5uaYPHlyqd0/EREREZUPMlGGz2z18PBAkyZNsGrVKqmsXr166N27N0JDQwvVRt++faFQKLBp0yYAgL+/P65cuYLDhw9LdaZNm4bTp09Ls9AffPABrKysEBERIdX58MMPYWxsLLXzJmlpaTA3N0dqairMzMwK9R4iIiIiKj2a5LUyWz6RlZWFuLg4eHt7q5R7e3sjNja2UG2cP38esbGx8PLykspatWqFuLg4nD59GgBw48YN7N+/H927d1epc/jwYVy9ehUA8PvvvyMmJgbdunXL91qZmZlIS0tTOYiIiIiociiz5RMPHz6EUqmElZWVSrmVlRWSk5MLfG+NGjXw4MEDZGdnIyQkBKNHj5bODRgwAA8ePECrVq0ghEB2djbGjx+PwMBAqc6sWbOQmpoKFxcXyOVyKJVKLFy4EAMHDsz3mqGhofj000+LeLdEREREVJ6V+RftZDKZymshhFrZ606cOIGzZ89i9erVCAsLw9atW6VzR48excKFCxEeHo5z585h165d+PHHH7FgwQKpTlRUFL7//nts2bIF586dw4YNG7BkyRJs2LAh32sGBQUhNTVVOm7fvl3EOyYiIiKi8qbMZoqrVq0KuVyuNit8//59tdnj1zk4OAAA3NzccO/ePYSEhEizvHPnzoWPj480e+zm5oZnz55h7NixmDNnDnR0dDBjxgwEBgZiwIABUp2bN28iNDQUw4YNy/OaBgYGMDAweKt7JiIiIqLyqcxmivX19dG0aVNER0erlEdHR8PT07PQ7QghkJmZKb1OT0+Hjo7qbcnlcgghkPudwvzqcEs2IiIiIu1UpluyBQQEwMfHB+7u7mjRogXWrl2LW7duYdy4cQBeLllITEzExo0bAQArV65EzZo14eLiAuDlvsVLlizBxIkTpTZ79OiBZcuWoXHjxvDw8MD169cxd+5c9OzZE3K5XKqzcOFC1KxZEw0aNMD58+exbNkyjBw5spRHgIiIiIjKgzINxf3798ejR48wf/58JCUlwdXVFfv374e9vT0AICkpCbdu3ZLq5+TkICgoCAkJCdDV1YWjoyMWL14MX19fqU5wcDBkMhmCg4ORmJiIatWqSSE419dff425c+diwoQJuH//PmxtbeHr64tPPvmk9G6eiIiIiMqNMt2nuCLjPsVERERE5VuF2KeYiIiIiKi8YCgmIiIiIq3HUExEREREWo+hmIiIiIi0HkMxEREREWk9hmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdZjKCYiIiIircdQTERERERaj6GYiIiIiLQeQzERERERaT2GYiIiIiLSegzFRERERKT1GIqJiIiISOu9dSjOzMwsjn4QEREREZUZjUPxL7/8guHDh8PR0RF6enowNjaGqakpvLy8sHDhQty9e7ck+klEREREVGIKHYr37NmDunXrYtiwYdDR0cGMGTOwa9cu/PLLL4iIiICXlxcOHTqE2rVrY9y4cXjw4EFJ9puIiIiIqNjIhBCiMBWbN2+OuXPnonv37tDRyT9LJyYmYsWKFbCyssK0adOKraPlTVpaGszNzZGamgozM7Oy7g4RERERvUaTvFboUEyqGIqJiIiIyjdN8lqx7z5x5syZ4m6SiIiIiKhEFSkUP336FM+fP1cpu3DhAnr06IH333+/WDpGRERERFRaNArFd+7cQcuWLWFubg5zc3MEBAQgPT0dQ4cORbNmzWBgYICYmJiS6isRERERUYnQ1aRyYGAgnj59ihUrVmDnzp1YsWIFjh07hkaNGuHq1atwcHAoqX4SEREREZUYjULxkSNHsH37drRs2RIfffQRbG1t0a9fPwQGBpZU/4iIiIiISpxGyyeSk5Ph6OgIALC2toaRkRF69epVIh0jIiIiIiotGn/RTi6X//+bdXRgaGhYrB0iIiIiIiptGi2fEEKgQ4cO0NV9+bbnz5+jR48e0NfXV6l37ty54ushEREREVEJ0ygUz5s3T+U1l04QERERUWXAJ9oVEZ9oR0RERFS+aZLXNJopBoBTp05h3759ePHiBTp27Ahvb+8id5SIiIiIqDzQKBTv3r0b/fr1g6GhIXR1dbF06VIsXboUU6ZMKaHuERERERGVPI12n1i0aBGGDx+OlJQUpKSk4NNPP8Vnn31WUn0jIiIiIioVGq0pNjMzw9mzZ+Hs7AwAyMzMhEKhQHJyMqpWrVpinSyPuKaYiIiIqHzTJK9pNFP89OlTWFhYSK8NDAxgZGSEtLS0InWUiIiIiKg80PiLdr/88gvMzc2l1zk5OTh8+DAuX74slfXs2bN4ekdEREREVAo0Wj6ho/PmiWWZTAalUvlWnaoIuHyCiIiIqHwrsS3ZcnJy3qpjRERERETlkUZrikeOHIknT56UVF+IiIiIiMqERqF4w4YNeP78eUn1hYiIiIioTGgUivlEaCIiIiKqjDQKxcDLL9IREREREVUmGm/J5uzs/MZg/Pjx4yJ3iIiIiIiotGkcij/99FOVfYqJiIiIiCo6jUPxgAEDYGlpWWwdCA8Px5dffomkpCQ0aNAAYWFhaN26dZ51Y2JiMGvWLPz5559IT0+Hvb09fH19MXXqVJV6YWFhWLVqFW7duoWqVavio48+QmhoKAwNDaU6iYmJmDVrFn7++Wc8f/4czs7OiIiIQNOmTYvt3oiIiCoSoVQi/Wwcsh88gG61ajB2bwqZXF7W3SIqFRqF4uJeTxwVFYUpU6YgPDwcLVu2xJo1a9C1a1fEx8ejZs2aavUVCgX8/f3RsGFDKBQKxMTEwNfXFwqFAmPHjgUAbN68GYGBgVi3bh08PT1x9epVDB8+HACwfPlyAMC///6Lli1bol27dvj5559haWmJv//+W+UR1kRERNok7eBB3FsUiuzkZKlM19oaVrODYObtXYY9IyodGj/RLjk5udhmij08PNCkSROsWrVKKqtXrx569+6N0NDQQrXRt29fKBQKbNq0CQDg7++PK1eu4PDhw1KdadOm4fTp0zhx4gQAIDAwEL/99pv0uij4RDsiIqos0g4eROLkKcDrkeB/k2HVV4QxGFOFpEle02j3iZycnGILxFlZWYiLi4P3a/+ReXt7IzY2tlBtnD9/HrGxsfDy8pLKWrVqhbi4OJw+fRoAcOPGDezfvx/du3eX6uzbtw/u7u7o168fLC0t0bhxY3z77bcFXiszMxNpaWkqBxERUUUnlErcWxSqHogBqezeolAIpbKUe0ZUugodiseNG4fbt28Xqm5UVBQ2b95cYJ2HDx9CqVTCyspKpdzKygrJr/zTTV5q1KgBAwMDuLu7w8/PD6NHj5bODRgwAAsWLECrVq2gp6cHR0dHtGvXDoGBgVKdGzduYNWqVahTpw5++eUXjBs3DpMmTcLGjRvzvWZoaCjMzc2lw87OrsA+EhERVQTpZ+NUlkyoEQLZyclIPxtXep0iKgOFXlNcrVo1uLq6wtPTEz179oS7uztsbW1haGiIf//9F/Hx8YiJicG2bdtQvXp1rF27tlDtvr5OWQjxxrXLJ06cwNOnT3Hy5EkEBgbCyckJAwcOBAAcPXoUCxcuRHh4ODw8PHD9+nVMnjwZNjY2mDt3LoCXM97u7u5YtGgRAKBx48b4448/sGrVKgwdOjTPawYFBSEgIEB6nZaWxmBMREQVXvaDB8Vaj6iiKnQoXrBgASZOnIiIiAisXr0aly9fVjlvamqKjh074rvvvlNbEpGXqlWrQi6Xq80K379/X232+HUODg4AADc3N9y7dw8hISFSKJ47dy58fHyk2WM3Nzc8e/YMY8eOxZw5c6CjowMbGxvUr19fpc169eph586d+V7TwMAABgYGb7wvIiKiikS3WrVirUdUUWm0+4SlpSWCgoIQFBSElJQU3Lx5E8+fP0fVqlXh6Oio0e4U+vr6aNq0KaKjo9GnTx+pPDo6Gr169Sp0O0IIZGZmSq/T09Oho6O6KkQul0MIIT2mumXLlvjrr79U6ly9ehX29vaFvi4REVFlYOzeFLrW1si+dy/vdcUyGXStrGDszi1LqXLTeJ/iXBYWFm+9hVlAQAB8fHzg7u6OFi1aYO3atbh16xbGjRsH4OWShcTERGmt78qVK1GzZk24uLgAeLlv8ZIlSzBx4kSpzR49emDZsmVo3LixtHxi7ty56NmzJ+T/22tx6tSp8PT0xKJFi/Dxxx/j9OnTWLt2baGXfBAREVUWMrkcVrODXu4+IZOpBuP/TXZZzQ7ifsVU6RU5FBeH/v3749GjR5g/fz6SkpLg6uqK/fv3SzO2SUlJuHXrllQ/JycHQUFBSEhIgK6uLhwdHbF48WL4+vpKdYKDgyGTyRAcHIzExERUq1YNPXr0wMKFC6U6zZo1w+7duxEUFIT58+fDwcEBYWFhGDx4cOndPBERUTlh5u0NrAhT36fYyor7FJPW0GifYvp/3KeYiIgqGz7RjiobTfJamc4UExERUfkhk8uh8Ghe1t0gKhMaPbyDiIiIiKgyKnIozs7OxqFDh7BmzRo8efIEAHD37l08ffq02DpHRERERFQairR84ubNm+jSpQtu3bqFzMxMdOrUCaampvjiiy+QkZGB1atXF3c/iYiIiIhKTJFmiidPngx3d3f8+++/MDIyksr79OmDw4cPF1vniIiIiIhKQ5FmimNiYvDbb79BX19fpdze3h6JiYnF0jEiIiIiotJSpJninJwcKJVKtfI7d+7A1NT0rTtFRERERFSaihSKO3XqhLCwMOm1TCbD06dPMW/ePHTr1q24+kZERERlKCdHIPGvf3H1TDIS//oXOTl8tAFVXkV6eMfdu3fRrl07yOVyXLt2De7u7rh27RqqVq2K48ePw9LSsiT6Wq7w4R1ERFSZ/X3+Pk5EXcOzlEypTGFhgNb968CxceX/nKfKQZO8VuQn2j1//hzbtm1DXFwccnJy0KRJEwwePFjli3eVGUMxERFVVn+fv48Day7ne76LryuDMVUIJR6Kjx8/Dk9PT+jqqn5PLzs7G7GxsWjTpo2mTVY4DMVERFQZ5eQIbJwdqzJD/DqTdwzgs9ATOjqyUuwZkeY0yWtFWlPcrl07PH78WK08NTUV7dq1K0qTREREVA4kXUspMBADwNN/M5F0LaV0OkRUSooUioUQkMnU/3b46NEjKBSKt+4UERERlY1naQUHYk3rEVUUGu1T3LdvXwAvd5sYPnw4DAwMpHNKpRIXL16Ep6dn8faQiIiISo3CzODNlTSoR1RRaBSKzc3NAbycKTY1NVX5Up2+vj7ef/99jBkzpnh7SERERKXGpo4FFBYGb1xTbFPHovQ6RVQKNArFkZGRAIBatWph+vTpXCpBRERUyejoyNC6f50Cd59o9XEdfsmOKp0ib8mm7bj7BBERVWZ57VNs8o4BWn3MfYqp4tAkr2k0U/yqH374Adu3b8etW7eQlZWlcu7cuXNFbZaIiIjKAcfGlnBoVO3lbhRpmVCYvVwywRliqqyKtPvEV199hREjRsDS0hLnz59H8+bNUaVKFdy4cQNdu3Yt7j4SERFRGdDRkaF63Xfg3Mwa1eu+w0BMlVqRQnF4eDjWrl2Lb775Bvr6+pg5cyaio6MxadIkpKamFncfiYiIiIhKVJFC8a1bt6St14yMjPDkyRMAgI+PD7Zu3Vp8vSMiIiIiKgVFCsXW1tZ49OgRAMDe3h4nT54EACQkJIDf2yMiIiKiiqZIobh9+/b4z3/+AwAYNWoUpk6dik6dOqF///7o06dPsXaQiIiIiKikFWlLtpycHOTk5EBX9+XmFdu3b0dMTAycnJwwbtw46OvrF3tHyxtuyUZERERUvmmS14p9n+LExERUr169OJsslxiKiYiIiMo3TfJakZZP5CU5ORkTJ06Ek5NTcTVJRERERFQqNArFKSkpGDx4MKpVqwZbW1t89dVXyMnJwSeffILatWvj5MmTWLduXUn1lYiIiIioRGj0RLvZs2fj+PHjGDZsGA4cOICpU6fiwIEDyMjIwM8//wwvL6+S6icRERERUYnRKBT/9NNPiIyMRMeOHTFhwgQ4OTnB2dkZYWFhJdQ9IiIiIqKSp9Hyibt376J+/foAgNq1a8PQ0BCjR48ukY4REREREZUWjUJxTk4O9PT0pNdyuRwKhaLYO0VEREREVJo0Wj4hhMDw4cNhYGAAAMjIyMC4cePUgvGuXbuKr4dERERERCVMo1A8bNgwlddDhgwp1s4QEREREZUFjUJxZGRkSfWDiIiIiKjMFNvDO4iIiIiIKiqGYiIiIiLSegzFRERERKT1GIqJiIiISOsxFBMRERGR1ityKN60aRNatmwJW1tb3Lx5EwAQFhaGvXv3FlvniIiIiIhKQ5FC8apVqxAQEIBu3bohJSUFSqUSAGBhYYGwsLDi7B8RERERUYkrUij++uuv8e2332LOnDmQy+VSubu7Oy5dulRsnSMiIiIiKg1FCsUJCQlo3LixWrmBgQGePXv21p0iIiIiIipNRQrFDg4OuHDhglr5zz//jPr1679tn4iIiIiISpVGj3nONWPGDPj5+SEjIwNCCJw+fRpbt25FaGgovvvuu+LuIxERERFRiSrSTPGIESMwb948zJw5E+np6Rg0aBBWr16NFStWYMCAARq1FR4eDgcHBxgaGqJp06Y4ceJEvnVjYmLQsmVLVKlSBUZGRnBxccHy5cvV6oWFhaFu3bowMjKCnZ0dpk6dioyMjDzbDA0NhUwmw5QpUzTqNxERERFVHkWaKQaAMWPGYMyYMXj48CFycnJgaWmpcRtRUVGYMmUKwsPD0bJlS6xZswZdu3ZFfHw8atasqVZfoVDA398fDRs2hEKhQExMDHx9faFQKDB27FgAwObNmxEYGIh169bB09MTV69exfDhwwFALUCfOXMGa9euRcOGDTUfACIiIiKqNGRCCKHpmz799FMMGTIEjo6Ob3VxDw8PNGnSBKtWrZLK6tWrh969eyM0NLRQbfTt2xcKhQKbNm0CAPj7++PKlSs4fPiwVGfatGk4ffq0yiz006dP0aRJE4SHh+Ozzz7De++9p9F2cmlpaTA3N0dqairMzMwK/T4iIiIiKh2a5LUiLZ/YuXMnnJ2d8f777+Obb77BgwcPNG4jKysLcXFx8Pb2Vin39vZGbGxsodo4f/48YmNj4eXlJZW1atUKcXFxOH36NADgxo0b2L9/P7p3767yXj8/P3Tv3h0dO3Ys1LUyMzORlpamchARERFR5VCkUHzx4kVcvHgR7du3x7Jly1C9enV069YNW7ZsQXp6eqHaePjwIZRKJaysrFTKrayskJycXOB7a9SoAQMDA7i7u8PPzw+jR4+Wzg0YMAALFixAq1atoKenB0dHR7Rr1w6BgYFSnW3btuHcuXOFno0GXq49Njc3lw47O7tCv5eIiIiIyrciP+a5QYMGWLRoEW7cuIEjR47AwcEBU6ZMgbW1tUbtyGQylddCCLWy1504cQJnz57F6tWrERYWhq1bt0rnjh49ioULFyI8PBznzp3Drl278OOPP2LBggUAgNu3b2Py5Mn4/vvvYWhoWOh+BgUFITU1VTpu376twV0SERERUXlW5C/avUqhUMDIyAj6+vp48uRJod5TtWpVyOVytVnh+/fvq80ev87BwQEA4Obmhnv37iEkJAQDBw4EAMydOxc+Pj7S7LGbmxuePXuGsWPHYs6cOYiLi8P9+/fRtGlTqT2lUonjx4/jm2++QWZmpspT+nIZGBjAwMCgUPdGRERERBVLkWeKExISsHDhQtSvXx/u7u44d+4cQkJC3rj0IZe+vj6aNm2K6OholfLo6Gh4enoWuh9CCGRmZkqv09PToaOjeltyuRxCCAgh0KFDB1y6dAkXLlyQDnd3dwwePBgXLlzIMxATERERUeVWpJniFi1a4PTp03Bzc8OIESMwaNAgVK9eXeN2AgIC4OPjA3d3d7Ro0QJr167FrVu3MG7cOAAvlywkJiZi48aNAICVK1eiZs2acHFxAfBy3+IlS5Zg4sSJUps9evTAsmXL0LhxY3h4eOD69euYO3cuevbsCblcDlNTU7i6uqr0Q6FQoEqVKmrlRERERKQdihSK27Vrh++++w4NGjR4q4v3798fjx49wvz585GUlARXV1fs378f9vb2AICkpCTcunVLqp+Tk4OgoCAkJCRAV1cXjo6OWLx4MXx9faU6wcHBkMlkCA4ORmJiIqpVq4YePXpg4cKFb9VXIiIiIqq8irRPMXGfYiIiIqLyTpO8VuiZ4oCAACxYsAAKhQIBAQEF1l22bFlhmyUiIiIiKnOFDsXnz5/HixcvpD8TEREREVUWXD5RRFw+QURERFS+lfhjnkeOHJnnfsTPnj3DyJEji9IkEREREVGZKVIo3rBhA54/f65W/vz5c2n7NCIiIiKiikKjLdnS0tKkh2A8efJE5THJSqUS+/fvh6WlZbF3koiIiIioJGkUii0sLCCTySCTyeDs7Kx2XiaT4dNPPy22zhERERERlQaNQvGRI0cghED79u2xc+dOvPvuu9I5fX192Nvbw9bWttg7SURERERUkjQKxV5eXgCAhIQE1KxZEzKZrEQ6RURERERUmgodii9evAhXV1fo6OggNTUVly5dyrduw4YNi6VzRERERESlodCh+L333kNycjIsLS3x3nvvQSaTIa8tjmUyGZRKZbF2koiIiF5S5gicTniM+08yYGlqiOYO70Kuw3+5JXpbhQ7FCQkJqFatmvRnIiIiKl0HLifh0//EIyk1QyqzMTfEvB710cXVpgx7RlTx8Yl2RcQn2hERUWk6cDkJ478/h9c/tHPniFcNacJgTPSaEn+i3YYNG/DTTz9Jr2fOnAkLCwt4enri5s2bRWmSiIiI8qHMEfj0P/FqgRiAVPbpf+KhzOE8F1FRFSkUL1q0CEZGRgCA//73v/jmm2/wxRdfoGrVqpg6dWqxdpCIiEjbnU54rLJk4nUCQFJqBk4nPC69ThFVMhptyZbr9u3bcHJyAgDs2bMHH330EcaOHYuWLVuibdu2xdk/IiIirXf/Sf6BuCj1iEhdkWaKTUxM8OjRIwDAwYMH0bFjRwCAoaEhnj9/Xny9IyIiIliaGhZrPSJSV6SZ4k6dOmH06NFo3Lgxrl69iu7duwMA/vjjD9SqVas4+0dERKT1mju8CxtzQySnZuS5rlgGwNr85fZsRFQ0RZopXrlyJVq0aIEHDx5g586dqFKlCgAgLi4OAwcOLNYOEhERaTu5jgzzetQH8P+7TeTKfT2vR33uV0z0FrglWxFxSzYiIipt3KeYSDOa5LUiLZ8AgJSUFERERODKlSuQyWSoV68eRo0aBXNz86I2SURERAXo4mqDTvWt+UQ7ohJQpJnis2fPonPnzjAyMkLz5s0hhMDZs2fx/PlzHDx4EE2aNCmJvpYrnCkmIiIiKt80yWtFCsWtW7eGk5MTvv32W+jqvpxszs7OxujRo3Hjxg0cP368aD2vQBiKiYiIiMq3Eg/FRkZGOH/+PFxcXFTK4+Pj4e7ujvT0dE2brHAYiomIiIjKtxJ/zLOZmRlu3bqlVn779m2YmpoWpUkiIiIiojJTpFDcv39/jBo1ClFRUbh9+zbu3LmDbdu2YfTo0dySjYiIiIgqnCLtPrFkyRLIZDIMHToU2dnZAAA9PT2MHz8eixcvLtYOEhERERGVtLfapzg9PR1///03hBBwcnKCsbFxcfatXOOaYiIiIqLyrcTWFKenp8PPzw/Vq1eHpaUlRo8eDRsbGzRs2FCrAjERERERVS4aheJ58+Zh/fr16N69OwYMGIDo6GiMHz++pPpGRERERFQqNFpTvGvXLkRERGDAgAEAgCFDhqBly5ZQKpWQy+Ul0kEiIiIiopKm0Uzx7du30bp1a+l18+bNoauri7t37xZ7x4iIiIiISotGoVipVEJfX1+lTFdXV9qBgoiIiIioItJo+YQQAsOHD4eBgYFUlpGRgXHjxkGhUEhlu3btKr4eEhERERGVMI1C8bBhw9TKhgwZUmydISIiIiIqCxqF4sjIyJLqBxERERFRmSnSY56JiIiIiCoThmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdYrcijetGkTWrZsCVtbW9y8eRMAEBYWhr179xZb54iIiIiISkORQvGqVasQEBCAbt26ISUlBUqlEgBgYWGBsLCw4uwfEREREVGJK1Io/vrrr/Htt99izpw5kMvlUrm7uzsuXbpUbJ0jIiIiIioNRQrFCQkJaNy4sVq5gYEBnj179tadIiIiIiIqTUUKxQ4ODrhw4YJa+c8//4z69etr1FZ4eDgcHBxgaGiIpk2b4sSJE/nWjYmJQcuWLVGlShUYGRnBxcUFy5cvV6sXFhaGunXrwsjICHZ2dpg6dSoyMjKk86GhoWjWrBlMTU1haWmJ3r1746+//tKo30RERERUeWj0mOdcM2bMgJ+fHzIyMiCEwOnTp7F161aEhobiu+++K3Q7UVFRmDJlCsLDw9GyZUusWbMGXbt2RXx8PGrWrKlWX6FQwN/fHw0bNoRCoUBMTAx8fX2hUCgwduxYAMDmzZsRGBiIdevWwdPTE1evXsXw4cMBQArQx44dg5+fH5o1a4bs7GzMmTMH3t7eiI+Ph0KhKMqQEBEREVEFJhNCiKK88dtvv8Vnn32G27dvAwCqV6+OkJAQjBo1qtBteHh4oEmTJli1apVUVq9ePfTu3RuhoaGFaqNv375QKBTYtGkTAMDf3x9XrlzB4cOHpTrTpk3D6dOn852FfvDgASwtLXHs2DG0adOmUNdNS0uDubk5UlNTYWZmVqj3EBEREVHp0SSvFXlLtjFjxuDmzZu4f/8+kpOTcfv2bY0CcVZWFuLi4uDt7a1S7u3tjdjY2EK1cf78ecTGxsLLy0sqa9WqFeLi4nD69GkAwI0bN7B//350794933ZSU1MBAO+++26+dTIzM5GWlqZyEBEREVHlUKTlE6+qWrVqkd738OFDKJVKWFlZqZRbWVkhOTm5wPfWqFEDDx48QHZ2NkJCQjB69Gjp3IABA/DgwQO0atUKQghkZ2dj/PjxCAwMzLMtIQQCAgLQqlUruLq65nvN0NBQfPrppxrcIRERERFVFEUKxQ4ODpDJZPmev3HjRqHber0dIUSBbQPAiRMn8PTpU5w8eRKBgYFwcnLCwIEDAQBHjx7FwoULER4eDg8PD1y/fh2TJ0+GjY0N5s6dq9aWv78/Ll68iJiYmAKvGRQUhICAAOl1Wloa7OzsCnubRERERFSOFSkUT5kyReX1ixcvcP78eRw4cAAzZswoVBtVq1aFXC5XmxW+f/++2uzx6xwcHAAAbm5uuHfvHkJCQqRQPHfuXPj4+Eizx25ubnj27BnGjh2LOXPmQEfn/1eMTJw4Efv27cPx48dRo0aNAq9pYGAAAwODQt0bEREREVUsRQrFkydPzrN85cqVOHv2bKHa0NfXR9OmTREdHY0+ffpI5dHR0ejVq1eh+yKEQGZmpvQ6PT1dJfgCgFwuhxACud8pFEJg4sSJ2L17N44ePSqFbCIiIiLSTm+9pvhVXbt2RVBQECIjIwtVPyAgAD4+PnB3d0eLFi2wdu1a3Lp1C+PGjQPwcslCYmIiNm7cCOBl6K5ZsyZcXFwAvNy3eMmSJZg4caLUZo8ePbBs2TI0btxYWj4xd+5c9OzZU3r6np+fH7Zs2YK9e/fC1NRUmq02NzeHkZFRsY0HEREREVUMxRqKf/jhhwJ3cHhd//798ejRI8yfPx9JSUlwdXXF/v37YW9vDwBISkrCrVu3pPo5OTkICgpCQkICdHV14ejoiMWLF8PX11eqExwcDJlMhuDgYCQmJqJatWro0aMHFi5cKNXJ3QKubdu2Kv2JjIyU9jQmIiIiIu1RpH2KGzdurPJlOCEEkpOT8eDBA4SHh0sP0qjMuE8xERERUfmmSV4r0kxx7969VV7r6OigWrVqaNu2rbS0gYiIiIiootA4FGdnZ6NWrVro3LkzrK2tS6JPRERERESlSuMn2unq6mL8+PEqOz4QEREREVVkRXrMs4eHB86fP1/cfSEiIiIiKhNFWlM8YcIETJs2DXfu3EHTpk2hUChUzjds2LBYOkdEREREVBo02n1i5MiRCAsLg4WFhXpDMpn0iGalUlmcfSyXuPsEERERUfmmSV7TKBTL5XIkJSXh+fPnBdbL3We4MmMoJiIiIirfSmxLttz8rA2hl4iIiIi0h8ZftHv1oR1ERERERJWBxl+0c3Z2fmMwfvz4cZE7RERERERU2jQOxZ9++inMzc1Loi9ERERERGVC41A8YMAAWFpalkRfiIiIiIjKhEZrirmemIiIiIgqI41CsQa7txERERERVRgaLZ/IyckpqX4QEREREZUZjbdkIyIiIiKqbBiKiYiIiEjrMRQTERERkdZjKCYiIiIircdQTERERERaj6GYiIiIiLQeQzERERERaT2GYiIiIiLSegzFRERERKT1GIqJiIiISOsxFBMRERGR1mMoJiIiIiKtx1BMRERERFqPoZiIiIiItB5DMRERERFpPYZiIiIiItJ6DMVEREREpPUYiomIiIhI6zEUExEREZHWYygmIiIiIq3HUExEREREWo+hmIiIiIi0HkMxEREREWk9hmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdZjKCYiIiIircdQTERERERaj6GYiIiIiLQeQzERERERab0yD8Xh4eFwcHCAoaEhmjZtihMnTuRbNyYmBi1btkSVKlVgZGQEFxcXLF++XK1eWFgY6tatCyMjI9jZ2WHq1KnIyMgo8nWJiIiIqHLTLcuLR0VFYcqUKQgPD0fLli2xZs0adO3aFfHx8ahZs6ZafYVCAX9/fzRs2BAKhQIxMTHw9fWFQqHA2LFjAQCbN29GYGAg1q1bB09PT1y9ehXDhw8HAClAa3pdIiIiIqrcZEIIUVYX9/DwQJMmTbBq1SqprF69eujduzdCQ0ML1Ubfvn2hUCiwadMmAIC/vz+uXLmCw4cPS3WmTZuG06dPS7PBRbluZmYmMjMzpddpaWmws7NDamoqzMzMCn/TRERERFQq0tLSYG5uXqi8VmbLJ7KyshAXFwdvb2+Vcm9vb8TGxhaqjfPnzyM2NhZeXl5SWatWrRAXF4fTp08DAG7cuIH9+/eje/fub3Xd0NBQmJubS4ednV2h+khERERE5V+ZLZ94+PAhlEolrKysVMqtrKyQnJxc4Htr1KiBBw8eIDs7GyEhIRg9erR0bsCAAXjw4AFatWoFIQSys7Mxfvx4BAYGvtV1g4KCEBAQIL3OnSkmIiIiooqvTNcUA4BMJlN5LYRQK3vdiRMn8PTpU5w8eRKBgYFwcnLCwIEDAQBHjx7FwoULER4eDg8PD1y/fh2TJ0+GjY0N5s6dW+TrGhgYwMDAQNPbIyIiIqIKoMxCcdWqVSGXy9VmZ+/fv682i/s6BwcHAICbmxvu3buHkJAQKRTPnTsXPj4+0uyxm5sbnj17hrFjx2LOnDlvdV0iIiIiqpzKbE2xvr4+mjZtiujoaJXy6OhoeHp6FrodIYTKF+DS09Oho6N6W3K5HEIICCGK7bpEREREVHmU6fKJgIAA+Pj4wN3dHS1atMDatWtx69YtjBs3DsDLdbyJiYnYuHEjAGDlypWoWbMmXFxcALzct3jJkiWYOHGi1GaPHj2wbNkyNG7cWFo+MXfuXPTs2RNyubxQ1yUiIiIi7VKmobh///549OgR5s+fj6SkJLi6umL//v2wt7cHACQlJeHWrVtS/ZycHAQFBSEhIQG6urpwdHTE4sWL4evrK9UJDg6GTCZDcHAwEhMTUa1aNfTo0QMLFy4s9HWJiIiISLuU6T7FFZkm+94RERERUemrEPsUExERERGVFwzFRERERKT1GIqJiIiISOsxFBMRERGR1mMoJiIiIiKtx1BMRERERFqPoZiIiIiItB5DMRERERFpPYZiIiIiItJ6DMVEREREpPUYiomIiIhI6zEUExEREZHWYygmIiIiIq3HUExEREREWo+hmIiIiIi0HkMxEREREWk9hmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdZjKCYiIiIircdQTERERERaj6GYiIiIiLQeQzERERERaT2GYiIiIiLSegzFRERERKT1GIqJiIiISOsxFBMRERGR1mMoJiIiIiKtx1BMRERERFqPoZiIiIiItB5DMRERERFpPYZiIiIiItJ6DMVEREREpPUYiomIiIhI6zEUExEREZHWYygmIiIiIq3HUExEREREWo+hmIiIiIi0HkMxEREREWk9hmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdZjKCYiIiIirVfmoTg8PBwODg4wNDRE06ZNceLEiXzrxsTEoGXLlqhSpQqMjIzg4uKC5cuXq9Rp27YtZDKZ2tG9e3epTnZ2NoKDg+Hg4AAjIyPUrl0b8+fPR05OTondJxERERGVX7plefGoqChMmTIF4eHhaNmyJdasWYOuXbsiPj4eNWvWVKuvUCjg7++Phg0bQqFQICYmBr6+vlAoFBg7diwAYNeuXcjKypLe8+jRIzRq1Aj9+vWTyj7//HOsXr0aGzZsQIMGDXD27FmMGDEC5ubmmDx5csnfOBERERGVKzIhhCiri3t4eKBJkyZYtWqVVFavXj307t0boaGhhWqjb9++UCgU2LRpU57nw8LC8MknnyApKQkKhQIA8MEHH8DKygoRERFSvQ8//BDGxsb5tpOZmYnMzEzpdVpaGuzs7JCamgozM7NC9ZWIiIiISk9aWhrMzc0LldfKbPlEVlYW4uLi4O3trVLu7e2N2NjYQrVx/vx5xMbGwsvLK986ERERGDBggBSIAaBVq1Y4fPgwrl69CgD4/fffERMTg27duuXbTmhoKMzNzaXDzs6uUH0kIiIiovKvzJZPPHz4EEqlElZWVirlVlZWSE5OLvC9NWrUwIMHD5CdnY2QkBCMHj06z3qnT5/G5cuXVWaEAWDWrFlITU2Fi4sL5HI5lEolFi5ciIEDB+Z7zaCgIAQEBEivc2eKiYiIiKjiK9M1xQAgk8lUXgsh1Mped+LECTx9+hQnT55EYGAgnJyc8gy0ERERcHV1RfPmzVXKo6Ki8P3332PLli1o0KABLly4gClTpsDW1hbDhg3L85oGBgYwMDDQ8O6IiIiIqCIos1BctWpVyOVytVnh+/fvq80ev87BwQEA4Obmhnv37iEkJEQtFKenp2Pbtm2YP3++2vtnzJiBwMBADBgwQGrn5s2bCA0NzTcUExEREVHlVWZrivX19dG0aVNER0erlEdHR8PT07PQ7QghVL4Al2v79u3IzMzEkCFD1M6lp6dDR0f11uVyObdkIyIiItJSZbp8IiAgAD4+PnB3d0eLFi2wdu1a3Lp1C+PGjQPwch1vYmIiNm7cCABYuXIlatasCRcXFwAv9y1esmQJJk6cqNZ2REQEevfujSpVqqid69GjBxYuXIiaNWuiQYMGOH/+PJYtW4aRI0eW4N0SERERUXlVpqG4f//+ePToEebPn4+kpCS4urpi//79sLe3BwAkJSXh1q1bUv2cnBwEBQUhISEBurq6cHR0xOLFi+Hr66vS7tWrVxETE4ODBw/med2vv/4ac+fOxYQJE3D//n3Y2trC19cXn3zyScndLBERERGVW2W6T3FFpsm+d0RERERU+irEPsVEREREROUFQzERERERaT2GYiIiIiLSegzFRERERKT1GIqJiIiISOsxFBMRERGR1mMoJiIiIiKtx1BMRERERFqPoZiIiIiItB5DMRERERFpPYZiIiIiItJ6DMVEREREpPUYiomIiIhI6zEUExEREZHWYygmIiIiIq3HUExEREREWo+hmIiIiIi0HkMxEREREWk9hmIiIiIi0noMxURERESk9RiKiYiIiEjrMRQTERERkdbTLesOUCHkKIGbscDTe4CJFWDvCejIy7pXRERERJUGQ3F5F78PODALSLv7/2VmtkCXz4H6PcuuX0RERESVCJdPlGfx+4DtQ1UDMQCkJb0sj99XNv0iIiIiqmQYisurHOXLGWKIPE7+r+xA4Mt6RERERPRWGIrLq5ux6jPEKgSQlviyHhERERG9FYbi8urpveKtR0RERET5Yigur0ysirceEREREeWLobi8svd8ucsEZPlUkAFm1V/WIyIiIqK3wlBcXunIX267BkA9GP/vdZfF3K+YiIiIqBgwFJdn9XsCH28EzGxUy81sX5Zzn2IiIiKiYsGHd5R39XsCLt35RDsiIiKiEsRQXBHoyAGH1mXdCyIiIqJKi8sniIiIiEjrMRQTERERkdZjKCYiIiIircdQTERERERaj6GYiIiIiLQeQzERERERaT2GYiIiIiLSegzFRERERKT1GIqJiIiISOsxFBMRERGR1mMoJiIiIiKtx1BMRERERFqPoZiIiIiItJ5uWXegohJCAADS0tLKuCdERERElJfcnJab2wrCUFxET548AQDY2dmVcU+IiIiIqCBPnjyBubl5gXVkojDRmdTk5OTg7t27MDU1hUwmK9Vrp6Wlwc7ODrdv34aZmVmpXlvbcKxLD8e69HCsSw/HuvRwrEtXRRlvIQSePHkCW1tb6OgUvGqYM8VFpKOjgxo1apRpH8zMzMr1L2JlwrEuPRzr0sOxLj0c69LDsS5dFWG83zRDnItftCMiIiIircdQTERERERaj6G4AjIwMMC8efNgYGBQ1l2p9DjWpYdjXXo41qWHY116ONalqzKON79oR0RERERajzPFRERERKT1GIqJiIiISOsxFBMRERGR1mMoJiIiIiKtx1BczMLDw+Hg4ABDQ0M0bdoUJ06cKNT7fvvtN+jq6uK9995TKX/x4gXmz58PR0dHGBoaolGjRjhw4IBKnSdPnmDKlCmwt7eHkZERPD09cebMmXyv5evrC5lMhrCwMKns8ePHmDhxIurWrQtjY2PUrFkTkyZNQmpqaqHvvbRV1LF+lRACXbt2hUwmw549ewrV/7JQ0cf6v//9L9q3bw+FQgELCwu0bdsWz58/L9Q9lLaKPNbJycnw8fGBtbU1FAoFmjRpgh9++KFQ/S8L5XWshw8fDplMpnK8//77KnUyMzMxceJEVK1aFQqFAj179sSdO3c0H4RSVFHHm5+Ppfu7natMPh8FFZtt27YJPT098e2334r4+HgxefJkoVAoxM2bNwt8X0pKiqhdu7bw9vYWjRo1Ujk3c+ZMYWtrK3766Sfx999/i/DwcGFoaCjOnTsn1fn4449F/fr1xbFjx8S1a9fEvHnzhJmZmbhz547atXbv3i0aNWokbG1txfLly6XyS5cuib59+4p9+/aJ69evi8OHD4s6deqIDz/88K3GpKRU5LF+1bJly0TXrl0FALF7925Nh6FUVPSxjo2NFWZmZiI0NFRcvnxZXL16VezYsUNkZGQUeUxKSkUf644dO4pmzZqJU6dOib///lssWLBA6OjoqFyrvCjPYz1s2DDRpUsXkZSUJB2PHj1Suda4ceNE9erVRXR0tDh37pxo166daNSokcjOzn77wSkBFXm8+flYur/bucri85GhuBg1b95cjBs3TqXMxcVFBAYGFvi+/v37i+DgYDFv3jy1X0QbGxvxzTffqJT16tVLDB48WAghRHp6upDL5eLHH39UqdOoUSMxZ84clbI7d+6I6tWri8uXLwt7e/t8g1qu7du3C319ffHixYsC65WFyjDWFy5cEDVq1BBJSUnlOhRX9LH28PAQwcHBb7rNcqGij7VCoRAbN25UKXv33XfFd999V2D/y0J5Huthw4aJXr165duHlJQUoaenJ7Zt2yaVJSYmCh0dHXHgwIEC+19WKvJ454WfjyU71mX1+cjlE8UkKysLcXFx8Pb2Vin39vZGbGxsvu+LjIzE33//jXnz5uV5PjMzE4aGhiplRkZGiImJAQBkZ2dDqVQWWAcAcnJy4OPjgxkzZqBBgwaFuqfU1FSYmZlBV1e3UPVLS2UY6/T0dAwcOBDffPMNrK2t87/ZMlbRx/r+/fs4deoULC0t4enpCSsrK3h5eam0UV5U9LEGgFatWiEqKgqPHz9GTk4Otm3bhszMTLRt2zbf/peF8j7WAHD06FFYWlrC2dkZY8aMwf3796VzcXFxePHihUr/bW1t4erqWmD/y0pFH++88POx5Ma6LD8fGYqLycOHD6FUKmFlZaVSbmVlheTk5Dzfc+3aNQQGBmLz5s35/ofVuXNnLFu2DNeuXUNOTg6io6Oxd+9eJCUlAQBMTU3RokULLFiwAHfv3oVSqcT333+PU6dOSXUA4PPPP4euri4mTZpUqPt59OgRFixYAF9f30LVL02VYaynTp0KT09P9OrVS9PbL1UVfaxv3LgBAAgJCcGYMWNw4MABNGnSBB06dMC1a9c0Ho+SVNHHGgCioqKQnZ2NKlWqwMDAAL6+vti9ezccHR01HY4SVd7HumvXrti8eTN+/fVXLF26FGfOnEH79u2RmZkJ4OXabX19fbzzzjuF7n9Zqujj/Tp+PpbsWJfl5yNDcTGTyWQqr4UQamUAoFQqMWjQIHz66adwdnbOt70VK1agTp06cHFxgb6+Pvz9/TFixAjI5XKpzqZNmyCEQPXq1WFgYICvvvoKgwYNkurExcVhxYoVWL9+fZ59eV1aWhq6d++O+vXr5/u3xvKgoo71vn378Ouvv+b75bvyqKKOdU5ODoCXXwwbMWIEGjdujOXLl6Nu3bpYt26dxuNQGirqWANAcHAw/v33Xxw6dAhnz55FQEAA+vXrh0uXLmk6DKWiPI41APTv3x/du3eHq6srevTogZ9//hlXr17FTz/9VOD95Nf/8qIyjDc/H0t2rMv887FUFmlogczMTCGXy8WuXbtUyidNmiTatGmjVv/ff/8VAIRcLpcOmUwmlR0+fFil/vPnz8WdO3dETk6OmDlzpqhfv75am0+fPhV3794VQrxc8N6tWzchhBDLly8XMplM5VoAhI6OjrC3t1dpIy0tTbRo0UJ06NBBPH/+/G2GpMRU9LGePHlyvnW8vLyKYYSKT0Uf6xs3bggAYtOmTSptfvzxx2LQoEFFHpeSUNHH+vr16wKAuHz5skqbHTp0EL6+vkUel5JQnsc6P05OTmLx4sVCCCEOHz4sAIjHjx+r1GnYsKH45JNP3jwApayij3cufj6W/FiX9ecjQ3Exat68uRg/frxKWb169fJc3K5UKsWlS5dUjvHjx4u6deuKS5cuiadPn+Z5jaysLOHo6CiCgoLy7cfjx4+Fubm5WLNmjRBCiIcPH6pdy9bWVsyaNUv8+eef0vtSU1PF+++/L7y8vMSzZ8+KMgSlpiKPdVJSklodAGLFihXixo0bRR2SElORxzonJ0fY2tqqfdHuvffeK/BaZaUij/XFixcFABEfH6/Slre3txgzZoxG41AayutY5+Xhw4fCwMBAbNiwQQjx/1+0i4qKkurcvXu33H/RrqKOtxD8fHxdSY11WX8+MhQXo9xtUCIiIkR8fLyYMmWKUCgU4p9//hFCCBEYGCh8fHzyfX9e3/g8efKk2Llzp/j777/F8ePHRfv27YWDg4P4999/pToHDhwQP//8s7hx44Y4ePCgaNSokWjevLnIysrK91qvf3M8LS1NeHh4CDc3N3H9+nWV7VLK4xY/FXms84JyvPtERR/r5cuXCzMzM7Fjxw5x7do1ERwcLAwNDcX169c1HouSVpHHOisrSzg5OYnWrVuLU6dOievXr4slS5YImUwmfvrppyKNR0kqr2P95MkTMW3aNBEbGysSEhLEkSNHRIsWLUT16tVFWlqa1M64ceNEjRo1xKFDh8S5c+dE+/btK8SWbBVxvPn5WLq/268rzc9HhuJitnLlSmFvby/09fVFkyZNxLFjx6Rzw4YNK3D6P69fxKNHj4p69eoJAwMDUaVKFeHj4yMSExNV6kRFRYnatWsLfX19YW1tLfz8/ERKSkqB/Xz9A+3IkSMCQJ5HQkJCYW+/VFXUsc5LeQ7FQlT8sQ4NDRU1atQQxsbGokWLFuLEiRNvvOeyUpHH+urVq6Jv377C0tJSGBsbi4YNG6pt0VaelMexTk9PF97e3qJatWpCT09P1KxZUwwbNkzcunVLpZ3nz58Lf39/8e677wojIyPxwQcfqNUpbyrqePPzsXR/t19Xmp+Psv9dkIiIiIhIa3H3CSIiIiLSegzFRERERKT1GIqJiIiISOsxFBMRERGR1mMoJiIiIiKtx1BMRERERFqPoZiIiIiItB5DMRERERFpPYZiItJa69evh4WFRVl3o8hq1aqFsLCwAuuEhITgvffeK5X+lDe//vorXFxckJOTU6rXvXTpEmrUqIFnz56V6nWJ6O0wFBNRhTZ8+HDIZDK14/r162XdNaxfv16lTzY2Nvj444+RkJBQLO2fOXMGY8eOlV7LZDLs2bNHpc706dNx+PDhYrlefl6/TysrK/To0QN//PGHxu0U519SZs6ciTlz5kBHRyfPfuYe3333XZ7n8/p51apVSzpvZGQEFxcXfPnll3j14bBubm5o3rw5li9fXmz3QkQlj6GYiCq8Ll26ICkpSeVwcHAo624BAMzMzJCUlIS7d+9iy5YtuHDhAnr27AmlUvnWbVerVg3GxsYF1jExMUGVKlXe+lpv8up9/vTTT3j27Bm6d++OrKysEr92XmJjY3Ht2jX069cvz36+egwePFjtfEE/r/nz5yMpKQlXrlzB9OnTMXv2bKxdu1blOiNGjMCqVauK5edMRKWDoZiIKjwDAwNYW1urHHK5HMuWLYObmxsUCgXs7OwwYcIEPH36NN92fv/9d7Rr1w6mpqYwMzND06ZNcfbsWel8bGws2rRpAyMjI9jZ2WHSpElv/CdymUwGa2tr2NjYoF27dpg3bx4uX74szWSvWrUKjo6O0NfXR926dbFp0yaV94eEhKBmzZowMDCAra0tJk2aJJ17dflErVq1AAB9+vSBTCaTXr+6fOKXX36BoaEhUlJSVK4xadIkeHl5Fdt9uru7Y+rUqbh58yb++usvqU5BP4+jR49ixIgRSE1NlWZiQ0JCAABZWVmYOXMmqlevDoVCAQ8PDxw9erTA/mzbtg3e3t4wNDTMs5+vHkZGRnneR14/LwAwNTWFtbU1atWqhdGjR6Nhw4Y4ePCgynU6d+6MR48e4dixYwX2k4jKD4ZiIqq0dHR08NVXX+Hy5cvYsGEDfv31V8ycOTPf+oMHD0aNGjVw5swZxMXFITAwEHp6egBerhPt3Lkz+vbti4sXLyIqKgoxMTHw9/fXqE+5AezFixfYvXs3Jk+ejGnTpuHy5cvw9fXFiBEjcOTIEQDADz/8gOXLl2PNmjW4du0a9uzZAzc3tzzbPXPmDAAgMjISSUlJ0utXdezYERYWFti5c6dUplQqsX37dmm2tDjuMyUlBVu2bAEAafyAgn8enp6eCAsLU5nJnT59OoCXs66//fYbtm3bhosXL6Jfv37o0qULrl27lm8fjh8/Dnd390L3OT+v/rxeJ4TA0aNHceXKFZX7BAB9fX00atQIJ06ceOs+EFEpEUREFdiwYcOEXC4XCoVCOj766KM8627fvl1UqVJFeh0ZGSnMzc2l16ampmL9+vV5vtfHx0eMHTtWpezEiRNCR0dHPH/+PM/3vN7+7du3xfvvvy9q1KghMjMzhaenpxgzZozKe/r16ye6desmhBBi6dKlwtnZWWRlZeXZvr29vVi+fLn0GoDYvXu3Sp158+aJRo0aSa8nTZok2rdvL73+5ZdfhL6+vnj8+PFb3ScAoVAohLGxsQAgAIiePXvmWT/Xm34eQghx/fp1IZPJRGJiokp5hw4dRFBQUL5tm5ubi40bN+bbz9zDysoq3+u//vMS4uWY6+vrC4VCIfT09AQAYWhoKH777Te1PvTp00cMHz68wDEgovJDtywDORFRcWjXrh1WrVolvVYoFACAI0eOYNGiRYiPj0daWhqys7ORkZGBZ8+eSXVeFRAQgNGjR2PTpk3o2LEj+vXrB0dHRwBAXFwcrl+/js2bN0v1hRDIyclBQkIC6tWrl2ffUlNTYWJiAiEE0tPT0aRJE+zatQv6+vq4cuWKyhflAKBly5ZYsWIFAKBfv34ICwtD7dq10aVLF3Tr1g09evSArm7R/6978ODBaNGiBe7evQtbW1ts3rwZ3bp1wzvvvPNW92lqaopz584hOzsbx44dw5dffonVq1er1NH05wEA586dgxACzs7OKuWZmZkFrpV+/vy52tKJV/uZK/dLeLkK+nnlmjFjBoYPH44HDx5gzpw5aN++PTw9PdWuZWRkhPT09Hz7SETlC0MxEVV4CoUCTk5OKmU3b95Et27dMG7cOCxYsADvvvsuYmJiMGrUqDz/KRx4uf520KBB+Omnn/Dzzz9j3rx52LZtG/r06YOcnBz4+vqqrOnNVbNmzXz7lhvCdHR0YGVlpRb+ZDKZymshhFRmZ2eHv/76C9HR0Th06BAmTJiAL7/8EseOHVP75/rCat68ORwdHbFt2zaMHz8eu3fvRmRkpHS+qPepo6Mj/QxcXFyQnJyM/v374/jx4wCK9vPI7Y9cLkdcXBzkcrnKORMTk3zfV7VqVfz7778F9jMvb/p55bbt5OQEJycn7Ny5E05OTnj//ffRsWNHlXqPHz+W/lJFROUfQzERVUpnz55FdnY2li5dKs0Gbt++/Y3vc3Z2hrOzM6ZOnYqBAwciMjISffr0QZMmTfDHH38UGKjyUlAIq1evHmJiYjB06FCpLDY2VmU21sjICD179kTPnj3h5+cHFxcXXLp0CU2aNFFrT09Pr1C7HQwaNAibN29GjRo1oKOjg+7du0vninqfr5s6dSqWLVuG3bt3o0+fPoX6eejr66v1v3HjxlAqlbh//z5at25d6Os3btwY8fHxGvf7TaH5de+88w4mTpyI6dOn4/z58yp/ybl8+TI++ugjjftARGWDX7QjokrJ0dER2dnZ+Prrr3Hjxg1s2rRJ7Z/zX/X8+XP4+/vj6NGjuHnzJn777TecOXNGCqizZs3Cf//7X/j5+eHChQu4du0a9u3bh4kTJxa5jzNmzMD69euxevVqXLt2DcuWLcOuXbukL5itX78eERERuHz5snQPRkZGsLe3z7O9WrVq4fDhw0hOTs5zljTX4MGDce7cOSxcuBAfffSRyjKD4rpPMzMzjB49GvPmzYMQolA/j1q1auHp06c4fPgwHj58iPT0dDg7O2Pw4MEYOnQodu3ahYSEBJw5cwaff/459u/fn+/1O3fujJiYGI36XFR+fn7466+/VL7A+M8//yAxMVFt9piIyrEyXM9MRPTWhg0bJnr16pXnuWXLlgkbGxthZGQkOnfuLDZu3CgAiH///VcIofrFqszMTDFgwABhZ2cn9PX1ha2trfD391f5ctnp06dFp06dhImJiVAoFKJhw4Zi4cKF+fYtry+OvS48PFzUrl1b6OnpCWdnZ5Uvh+3evVt4eHgIMzMzoVAoxPvvvy8OHToknX/9i3b79u0TTk5OQldXV9jb2wsh1L9ol6tZs2YCgPj111/VzhXXfd68eVPo6uqKqKgoIcSbfx5CCDFu3DhRpUoVAUDMmzdPCCFEVlaW+OSTT0StWrWEnp6esLa2Fn369BEXL17Mt0+PHz8WRkZG4s8//3xjPwt7Xgj1Mc81ZswY0aBBA6FUKoUQQixatEh07ty5wLaIqHyRCfHKY3iIiIgqiZkzZyI1NRVr1qwp1etmZmaiTp062Lp1K1q2bFmq1yaiouPyCSIiqpTmzJkDe3v7Un+q3M2bNzFnzhwGYqIKhjPFRERERKT1OFNMRERERFqPoZiIiIiItB5DMRERERFpPYZiIiIiItJ6DMVEREREpPUYiomIiIhI6zEUExEREZHWYygmIiIiIq3HUExEREREWu//AKmsTD0CGubJAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 800x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Data\n",
    "tools = [\"BCFTOOL\", \"DEEPVARIANT\", \"VARSCAN2\", \"HAPLOTYPECALLER\", \"MUTECT2\"]\n",
    "TP = [28278, 32588, 26663, 28192, 28089]\n",
    "TN = [37281968, 37290358, 37277852, 37281308, 37281259]\n",
    "FP = [37208177, 37203867, 37209792, 37208263, 37208366]\n",
    "FN = [45513, 53903, 41397, 44853, 44804]\n",
    "\n",
    "# Initialize a figure for ROC curves\n",
    "plt.figure(figsize=(8, 6))\n",
    "\n",
    "# Iterate over tools\n",
    "for i in range(len(tools)):\n",
    "    # Calculate True Positive Rate (TPR) and False Positive Rate (FPR)\n",
    "    tpr = TP[i] / (TP[i] + FN[i])\n",
    "    fpr = FP[i] / (TN[i] + FP[i])\n",
    "\n",
    "    # Plot ROC curve\n",
    "    plt.plot(fpr, tpr, marker='o', label=tools[i])\n",
    "\n",
    "# Set axis labels and title\n",
    "plt.xlabel('False Positive Rate (FPR)')\n",
    "plt.ylabel('True Positive Rate (TPR)')\n",
    "plt.title('FPR vs TPR')\n",
    "\n",
    "# Add legend\n",
    "plt.legend()\n",
    "\n",
    "plt.savefig(r'C:/Users/GenepoweRx_Madhu/Downloads/sen_spe_files_07_09_2023/VCC_input_files/Result/12652712_samples_FPR_TPR.svg', dpi=450, bbox_inches='tight')\n",
    "\n",
    "# Show the ROC curves\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a061051f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjMAAAHFCAYAAAAHcXhbAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAABb6UlEQVR4nO3deXhM59sH8O8kM1klEbEksYTEGnvR2GMLJbVFtFrV6L7Y11ZVhWqpqr3lZy1tlVprKRJbqK0Upah9J2JPBJHlef943hkZWWRiZs6cyfdzXXM5c+bMmfuZE3I797NohBACRERERCrloHQARERERM+DyQwRERGpGpMZIiIiUjUmM0RERKRqTGaIiIhI1ZjMEBERkaoxmSEiIiJVYzJDREREqsZkhoiIiFSNyQyRlWk0mjw9tm3bZrbPbNasGZo1a2bSe1544QVoNBpMmDDBbHEUJDt27MArr7yCkiVLwsnJCV5eXmjYsCFmzJiB5ORkpcMjsitapQMgKmh2795t9PzLL7/E1q1bsWXLFqP9wcHB1gzLyKFDh3Dw4EEAwNy5czF48GDFYlGjkSNHYvTo0WjYsCG+/PJLBAUF4cGDB9i1axeio6Nx8uRJTJo0SekwiewGkxkiK6tfv77R82LFisHBwSHLfiXNmTMHABAeHo5169Zh165daNiwocJRZSWEwKNHj+Dq6qp0KAZLly7F6NGj8c4772D27NnQaDSG19q2bYuhQ4dmSWjz68GDB3BzczPLuYjUjGUmIht0+/ZtfPzxx4YSRWBgIIYPH46UlBSj4x49eoRhw4ahXLlycHJyQsmSJdGrVy/cvXs335/96NEjLFq0CHXq1DHcPZg3b162x27YsAEtW7aEl5cX3NzcUKVKFYwdO9bomL1796J9+/bw8fGBi4sLgoKC0L9/f8PrPXv2RNmyZbOcOzo62igRAGSJrnfv3pg5cyaqVKkCZ2dnLFiwAAAwatQohISEoEiRIvD09MQLL7yAuXPnIru1dBctWoQGDRqgUKFCKFSoEGrVqoW5c+cCkHfKtFotLl26lOV9b7/9Nnx8fPDo0aMcv7/Ro0fD29sbU6dOzRI/AHh4eKB169YAgPPnz0Oj0eDHH3/McpxGo0F0dHSW7+PAgQOIjIyEt7c3goKCMHnyZGg0Gpw+fTrLOT755BM4OTnh5s2bhn2bNm1Cy5Yt4enpCTc3NzRq1AibN2/OsT1EasBkhsjGPHr0CM2bN8fChQsxcOBArFu3Dm+88QbGjx+PiIgIw3FCCHTq1AkTJkxAjx49sG7dOgwcOBALFixAixYtsiQ+ebVixQrcuXMHb7/9NipUqIDGjRtjyZIluH//vtFxc+fORbt27ZCRkYGZM2dizZo16Nu3Ly5fvmw4ZuPGjWjSpAkuXryIiRMnYv369fj8889x/fr1/H05AFatWoUZM2bgiy++MJwfkInBBx98gN9++w0rVqxAREQE+vTpgy+//NLo/V988QW6d+8Of39//Pjjj1i5ciWioqJw4cIFAMAHH3wArVaL//3vf0bvu337NhYvXox33nkHLi4u2cZ27do1/Pvvv2jdurXF7phERESgfPnyWLp0KWbOnIk33ngDTk5OWRKi9PR0/Pzzz2jfvj2KFi0KAPj555/RunVreHp6YsGCBfjtt99QpEgRtGnThgkNqZsgIkVFRUUJd3d3w/OZM2cKAOK3334zOu6bb74RAERMTIwQQogNGzYIAGL8+PFGxy1ZskQAELNmzTLsCw0NFaGhoXmKp0WLFsLFxUXcuXNHCCHE/PnzBQAxd+5cwzFJSUnC09NTNG7cWGRkZOR4rqCgIBEUFCQePnyY4zFRUVEiICAgy/6RI0eKp/+JAiC8vLzE7du3c21Denq6SE1NFaNHjxY+Pj6GGM+ePSscHR1F9+7dc31/VFSUKF68uEhJSTHs++abb4SDg4M4d+5cju/bs2ePACA+/fTTXM+vd+7cOQFAzJ8/P8trAMTIkSMNz/XfxxdffJHl2IiICFGqVCmRnp5u2PfHH38IAGLNmjVCCCGSk5NFkSJFRPv27Y3em56eLmrWrClefPHFPMVMZIt4Z4bIxmzZsgXu7u6IjIw02t+zZ08AMPwPWt9hWL9fr2vXrnB3d8/X/7TPnTuHrVu3IiIiAoULFzacz8PDw6jUtGvXLiQmJuLjjz/OtpQCACdPnsSZM2dyvZORHy1atIC3t3eW/Vu2bEGrVq3g5eUFR0dH6HQ6fPHFF7h16xYSEhIAALGxsUhPT0evXr1y/Yx+/fohISEBS5cuBQBkZGRgxowZCA8Pz7YkZk1dunTJsu+tt97C5cuXsWnTJsO++fPnw9fXF23btgUgr9nt27cRFRWFtLQ0wyMjIwMvvfQS9u3bx1FWpFpMZohszK1bt+Dr65slSShevDi0Wi1u3bplOE6r1aJYsWJGx2k0Gvj6+hqOM8W8efMghEBkZCTu3r2Lu3fvIjU1FR06dMDOnTvx33//AQBu3LgBAChVqlSO58rLMfnh5+eXZd9ff/1l6Icye/Zs7Ny5E/v27cPw4cMBAA8fPjQpptq1a6NJkyb4/vvvAQBr167F+fPn0bt371zfV6ZMGQAyKbSU7Nrftm1b+Pn5Yf78+QCAO3fuYPXq1XjzzTfh6OgIAIbSXmRkJHQ6ndHjm2++gRACt2/ftljcRJbE0UxENsbHxwd79+6FEMIooUlISEBaWpqh/4OPjw/S0tJw48YNo4RGCIH4+HjUq1fPpM/NyMgw9LvI3Dcns3nz5mH8+PGGz8vcP+ZpeTkGAFxcXLLt35O502pm2d0JWrx4MXQ6HdauXWt0F2jVqlU5xlS6dOlc4+rbty+6du2KAwcOYPr06ahYsSLCwsJyfY+fnx+qV6+OmJiYPI000sf6dPtzS0Sza7+joyN69OiBqVOn4u7du1i0aBFSUlLw1ltvGY7R/9xMmzYtx5FzJUqUyDVeIlvFOzNENqZly5a4f/9+ll/ECxcuNLye+c+ff/7Z6Ljly5cjOTnZ8Hpebdy4EZcvX0avXr2wdevWLI+qVati4cKFSEtLQ8OGDeHl5YWZM2dmO1oIACpWrIigoCDMmzcv187IZcuWRUJCglGn4MePH2Pjxo15jl2j0UCr1RruQgDybsxPP/1kdFzr1q3h6OiIGTNmPPOcnTt3RpkyZTBo0CBs2rQp15JaZiNGjMCdO3fQt2/fbL+b+/fvIyYmBoBMHlxcXHD48GGjY37//fdnfs7T3nrrLTx69Ai//vorfvzxRzRo0ACVK1c2vN6oUSMULlwYx44dQ926dbN9ODk5mfy5RDZByQ47RJS1A/DDhw9FjRo1hIeHh5g4caKIjY0VI0eOFDqdTrRr185wXEZGhmjTpo3Q6XQiOjpaxMbGiu+++04UKlRI1K5dWzx69MhwbF46AHfp0kVotVpx5cqVbF+fOnWqACBWrVolhBBizpw5AoBo0aKF+PXXX8WWLVvErFmzRK9evQzv2bBhg9DpdKJWrVpiwYIFYuvWrWLBggXi9ddfNxxz9uxZodPpRLNmzcS6devE8uXLRWhoqChXrly2HYAzn19v8+bNAoCIjIwUMTEx4tdffxV16tQRFSpUEACMOu2OGDHCcOzy5cvFpk2bxNSpU7PtWKvvdO3u7i7u3r2b6/eXmf4zGjVqJObNmyfi4uLE+vXrRXR0tPDz8xP9+/c3HPvuu+8KFxcX8d1334lNmzaJr7/+WlSrVi3HDsA3btzI8XMbNGggSpcunaUDuN5PP/0kHBwcxKuvviqWLl0q4uLixLJly8SIESPEhx9+mOf2EdkaJjNECns6mRFCiFu3bokPP/xQ+Pn5Ca1WKwICAsSwYcOMEhQhZOLzySefiICAAKHT6YSfn5/46KOPDCOR9J6VzNy4cUM4OTmJTp065XjMnTt3hKurq9FomD/++EOEhoYKd3d34ebmJoKDg8U333xj9L7du3eLtm3bCi8vL+Hs7CyCgoLEgAEDjI75448/RK1atYSrq6sIDAwU06dPz3E0U3bJjBBCzJs3T1SqVEk4OzuLwMBAMXbsWDF37twsyYwQQixcuFDUq1dPuLi4GJK/7EYUnT9/XgDI1y/6uLg4ERkZKfz8/IROpxOenp6iQYMG4ttvvxWJiYmG4+7duyfeffddUaJECeHu7i7at29v+FxTk5lZs2YJAMLV1VXcu3cvx7jCw8NFkSJFhE6nEyVLlhTh4eFi6dKlJreRyFZohMjhHjERUQE3bdo09O3bF//++y+qVq2qdDhElAMmM0RETzl48CDOnTuHDz74AI0aNcrSf4mIbAuTGSKip5QtWxbx8fFo0qQJfvrpJ/j6+iodEhHlgskMERERqRqHZhMREZGqMZkhIiIiVWMyQ0RERKpm98sZZGRk4OrVq/Dw8MjT7J1ERESkPCEEkpKS4O/vDweH3O+92H0yc/Xq1WeuwUJERES26dKlS89cHNbukxkPDw8A8svw9PQ067lTU1MRExOD1q1bQ6fTmfXctoDtUz97byPbp3723ka2L/8SExNRunRpw+/x3Nh9MqMvLXl6elokmXFzc4Onp6fd/pCyfepm721k+9TP3tvI9j2/vHQRYQdgIiIiUjUmM0RERKRqTGaIiIhI1ZjMEBERkaoxmSEiIiJVYzJDREREqsZkhoiIiFSNyQwRERGpGpMZIiIiUjUmM0SkSunpQFycBtu3l0RcnAbp6UpHRERKYTJDRKqzYgVQtiwQFqbFxIl1ERamRdmycj8RFTxMZohIVVasACIjgcuXjfdfuSL3M6EhKniYzBCRaqSnA/36AUJkfU2/r39/sOREVMAwmSEi1dixI+sdmcyEAC5dkscRUcHBZIaIVOPaNfMeR0T2gckMEamGn595jyMi+8BkhohUo0kToFQpQKPJ+ZjSpeVxRFRwMJkhItVwdASmTMm+A7BekybyOCIqOJjMEJGqREQALVtm3e/tLf9ctAhYssS6MRGRspjMEJGqpKYCBw7I7W+/TcfAgfsRG5uGGzeAAQPk/qgoYNcu5WIkIutiMkNEqrJ5M3DnDlC8ONC7dwaaNr2C0FABR0fg22+Bjh2BlBT555kzSkdLRNbAZIaIVOW33+SfXbpk7Rvj6Aj88gtQpw5w8yYQHg7cvm39GInIupjMEJFqPH4MrFwpt195Jftj3N2BNWvkqKYTJ2Qfm8ePrRcjEVkfkxkiUo1Nm4C7d4ESJXIffu3nB6xbB3h4AHFxwHvv5T4CiojUjckMEamGvsQUGfns4dfVqwNLl8rjFi4ExoyxfHxEpAwmM0SkCo8fA6tWye2cSkxPa9MG+OEHuf3FF7I/DRHZHyYzRKQKsbHAvXuyhNSoUd7f9/77wJAhcvvtt7kIJZE9YjJDRKpgSonpaePGydFPjx8DnToBp06ZPTwiUhCTGSKyeSkpppeYMnNwkP1mXnxRDtVu1w64dcusIRKRgpjMEJHNi4kBEhMBf3+gYcP8ncPNDVi9GggIAE6flndoUlLMGiYRKYTJDBHZvMwlJofn+FerRAngjz8ALy/gzz9lHxoO2SZSPyYzRGTTHj0Cfv9dbuenxPS04GBg2TJAq5WLUkZHP/85iUhZTGaIyKZt3AgkJQElSwINGpjnnK1aATNnyu3Ro2V/GiJSLyYzRGTT9CWmrl2fr8T0tHfeAYYNk9vvvgts22a+cxORdTGZISKb9fCh7LQLmKfE9LQxY+R5U1OBzp2B//4z/2cQkeUxmSEim7VxI3D/vlw0MiTE/Od3cAB+/FGWr+7elats37hh/s8hIstiMkNENstSJabMXF1lB+PAQODsWaBjR9npmIjUg8kMEdkkS5eYMitWTK6yXbgwsHs30LMnkJFh2c8kIvNRNJmJjo6GRqMxevj6+mZ77AcffACNRoPJkydbN0giUsT69UByMlCmjJy519IqVwZWrgR0OmDJEmDECMt/JhGZh+J3ZqpWrYpr164ZHkeOHMlyzKpVq7B37174+/srECERKSFziUmjsc5nNmsGzJ4tt7/+Gpg3zzqfS0TPR/FkRqvVwtfX1/AoVqyY0etXrlxB79698csvv0Cn0ykUJRFZ04MHwJo1ctvSJaanRUU9uSvzwQfA5s3W/XwiMp1W6QBOnToFf39/ODs7IyQkBF9//TUCAwMBABkZGejRoweGDBmCqlWr5ul8KSkpSMm04EpiYiIAIDU1FampqWaNXX8+c5/XVrB96qfWNq5ercGDB1oEBAjUqpWGnMK3VPs+/xw4dcoRixc7oEsXgbi4NAQHm/Uj8kSt188U9t5Gtu/5z50XGiGUW5lk/fr1ePDgASpWrIjr169jzJgx+O+//3D06FH4+Phg7Nix2Lp1KzZu3AiNRoOyZcuif//+6N+/f47njI6OxqhRo7LsX7RoEdzc3CzYGiIyl/Hj62LXrpLo1OkUevY8pkgMjx87YOTIhjh+3AfFiydj/PgdKFyYK1MSWcuDBw/w+uuv4969e/D09Mz1WEWTmaclJycjKCgIQ4cORWhoKMLDw3HgwAFDX5m8JDPZ3ZkpXbo0bt68+cwvw1SpqamIjY1FWFiYXZbA2D71U2Mbk5OBkiW1ePBAg92701CnTs7/RFm6fTdvAk2banH6tAb16mUgNjYd1vw/kRqvn6nsvY1sX/4lJiaiaNGieUpmFC8zZebu7o7q1avj1KlTcHBwQEJCAsqUKWN4PT09HYMGDcLkyZNx/vz5bM/h7OwMZ2fnLPt1Op3FfpAseW5bwPapn5raGBsr+8yUKweEhGjz1PnXUu3z85OrbNevD+zb54B33nHAb79Zbs6bnKjp+uWXvbeR7cvfOfNK8Q7AmaWkpOD48ePw8/NDjx49cPjwYRw6dMjw8Pf3x5AhQ7Bx40alQyUiC9GPYnrlFeuNYspNhQrAqlWAkxOwfPmT9ZyIyHYoemdm8ODBaN++PcqUKYOEhASMGTMGiYmJiIqKgo+PD3x8fIyO1+l08PX1RaVKlRSKmIgs6f59OXkdYP1RTLlp0kQO037jDWD8eCAoCHj/faWjIiI9Re/MXL58Ga+99hoqVaqEiIgIODk5Yc+ePQgICFAyLCJSyLp1cubfoCCgdm2lozHWvTugH1vw8cdATIyy8RDRE4remVm8eLFJx+fUT4aI7IMSE+WZYsQI4PRp4KefgMhIYNcuoFo1paMiIpvqM0NEBVdSkuxsC9hWiSkzjUbOEBwaKuMNDweuXVM6KiJiMkNENmHtWrladfnyQK1aSkeTM2dnYMUKoGJF4OJFoEMHOZyciJTDZIaIbIKtjWLKTZEi8i5S0aLA/v2yP016utJRERVcTGaISHFJSXKVbMB2S0xPCwqSQ7adnYHffweGDlU6IqKCi8kMESluzRogJUWWbmrUUDqavGvUCPjxR7k9cSLwww+KhkNUYDGZISLFqanE9LRu3YCvvpLbffo86cRMRNbDZIaIFJWYqL4S09OGDQPeegvIyABefRX45x+lIyIqWJjMEJGiVq8GHj8GKldW75wtGg0wcybQooWcxTg8HLhyRemoiAoOJjNEpChbnygvr/RrN1WpIhOZ9u1lYkNElsdkhogUc/cuoF83Vq0lpswKF5ZLMhQrBhw8CLz2GodsE1kDkxkiUoy+xFSlClC1qtLRmEe5crJdLi5yIsCBA5WOiMj+MZkhIsWoeRRTburXl+s3AcDUqfJBRJbDZIaIFHH37pOVp7t2VTQUi4iMBL75Rm4PGCDn0iEiy2AyQ0SK+P13IDVVlpfspcT0tCFDgPfek0O2u3UDDhxQOiIi+8RkhogUkbnEZK80GuD774GwMODBA+Dll4FLl5SOisj+MJkhIqu7c8e+S0yZ6XTA0qXy7tO1azKhSUpSOioi+8JkhoisbtUqIC0NqF5djmSyd15ecsh2iRLA4cNyluC0NKWjIrIfTGaIyOoyT5RXUAQEyE7Arq5y+Ya+fQEhlI6KyD4wmSEiq7p1C9i0SW4XpGQGAOrVA375RfalmTEDmDxZ6YiI7AOTGSKyKn2JqUYNuR5TQdO5MzBhgtweNEh+H0T0fJjMEJFVFYRRTM8yYADw0UeyzPT668D+/UpHRKRuTGaIyGpu3QI2b5bbBa3ElJlGI2cFfukl4OFDuSjlhQtKR0WkXkxmiMhqVq6UCy/WqgVUrKh0NMrSaoElS2S5LT5eDtm+d0/pqIjUickMEVkNS0zGPD3lYpR+fsC//8q7VampSkdFpD5MZojIKm7cALZskdsFucT0tNKlZULj5gbExgK9enHINpGpmMwQkVXoS0wvvACUL690NLblhReAxYtlX5rZs5+MdiKivGEyQ0RWURAnyjNF+/ZP5p0ZOhRYtkzRcIhUhckMEVlcQgKwdavcZjKTs759gT595HaPHsBff2mUDYhIJZjMEJHFrVgBZGQAdeoAQUFKR2PbJk2SI5sePQIiIhxx/bqb0iER2TwmM0RkcRzFlHeOjsCvvwK1awMJCRp8+WV93L2rdFREto3JDBFZ1PXrQFyc3GaJKW8KFZKLUpYsKXD5sgdefdURjx8rHRWR7WIyQ0QWpS8x1asHlCundDTqUbIksGpVGlxc0rB1q4Nh+QMiyorJDBFZFEtM+VezJjB48D44OAjMmweMG6d0RES2ickMEVlMfDxLTM+rbt0ETJ6cAQD47DO5BAIRGWMyQ0QWs3y5LI2EhAABAUpHo14ffpiBAQPkdlQUsGuXsvEQ2RomM0RkMZwoz3y+/Rbo2BFISZF/njmjdEREtoPJDBFZxNWrwI4dcjsyUtlY7IGjI/DLL3Kunps3gfBw4PZtpaMisg1MZojIIvQlpvr1WWIyF3d3OWS7dGngxAkgIgIcsk0EJjNEZCEcxWQZfn7AunWAh4fsXP3eexyyTcRkhojM7soVYOdOuc0Sk/lVrw4sXSpLTwsXAmPGKB0RkbKYzBCR2elLTA0bypIImV+bNsAPP8jtL76Q/WmICiomM0RkdiwxWcf77wNDhsjtt99+0uGaqKBhMkNEZnX5MktM1jRuHNCli+wI3KkTcOqU0hERWR+TGSIyq2XL5J+NG8v1hciyHBxkv5kXX5RDtdu1A27dUjoqIutiMkNEZsWJ8qzPzQ1YvVoOgT99Wt6hSUlROioi62EyQ0Rmc/EisHs3oNHI0gdZT4kSwB9/AF5ewJ9/yj40HLJNBQWTGSIyG5aYlBUcLK+BVgssWgRERysdEZF1MJkhIrNZulT+yVFMymnVCpg5U26PHi370xDZOyYzRGQWFy4Ae/awxGQL3nkHGDZMbr/7LrBtm6LhEFkckxkiMgt9ialpUznlPilrzBh5hyw1FejcGfjvP6UjIrIcJjNEZBacKM+2ODgAP/4INGgA3L0rV9m+cUPpqIgsg8kMET238+eBv/6Sv0AjIpSOhvRcXYHffwcCA4GzZ4GOHYFHj5SOisj8mMwQ0XPTd/wNDQV8fZWNhYwVKyZX2S5cWA6b79kTyMhQOioi82IyQ0TPjRPl2bbKlYGVKwGdDliyBBgxQumIiMxL0WQmOjoaGo3G6OH7//+tS01NxSeffILq1avD3d0d/v7+ePPNN3H16lUlQyaip5w9C+zfzxKTrWvWDJg9W25//TUwb56i4RCZleJ3ZqpWrYpr164ZHkeOHAEAPHjwAAcOHMCIESNw4MABrFixAidPnkSHDh0UjpiIMtOXmJo1k7PQku2KinpyV+aDD4DNm5WNh8hctIoHoNUa7sZk5uXlhdjYWKN906ZNw4svvoiLFy+iTJky1gqRiHLBifLUZdQo4MwZOUNwly7Arl1y5mAiNVP8zsypU6fg7++PcuXKoVu3bjh79myOx967dw8ajQaFCxe2XoBElKMzZ4C//2aJSU00GmDuXLnkxL17csj29etKR0X0fBS9MxMSEoKFCxeiYsWKuH79OsaMGYOGDRvi6NGj8PHxMTr20aNH+PTTT/H666/D09Mzx3OmpKQgJdNysYmJiQBkH5zU1FSzxq8/n7nPayvYPvWzdBsXL3YA4IjmzTNQuHA6rP1V2vs1tFT7HB1lp+2mTbU4fVqD9u0zEBubDjc3s35MnvAaqpsl22fKOTVC2M66qsnJyQgKCsLQoUMxcOBAw/7U1FR07doVFy9exLZt23JNZqKjozFq1Kgs+xctWgQ3Jf6mEtmxgQNDcfZsYXz88SG0bn1B6XDIRFevuuOTT5oiKckJDRpcxZAh++Cg+P16IunBgwd4/fXXce/evVx/7wM2lswAQFhYGMqXL48ZM2YAkInMK6+8grNnz2LLli1Z7tg8Lbs7M6VLl8bNmzef+WWYKjU1FbGxsQgLC4NOpzPruW0B26d+lmzjqVNA1ao6ODoKXLqUhqJFzXr6PLH3a2iN9v35pwYvveSIx481GDQoHWPHWncSGl5DdbNk+xITE1G0aNE8JTOKdwDOLCUlBcePH0eTJk0APElkTp06ha1btz4zkQEAZ2dnODs7Z9mv0+ks9oNkyXPbArZP/SzRxlWr5J8tW2rg56fs92fv19CS7WveXA7TfuMN4LvvHFGxoiPef98iH5UrXkN1s0T7TDmfojcUBw8ejLi4OJw7dw579+5FZGQkEhMTERUVhbS0NERGRmL//v345ZdfkJ6ejvj4eMTHx+Px48dKhk1E4ER59qR7dznKCQA+/hiIiVE2HiJTKZrMXL58Ga+99hoqVaqEiIgIODk5Yc+ePQgICMDly5exevVqXL58GbVq1YKfn5/hsWvXLiXDJirwTpwA/vlHdiTt3FnpaMgcRowAevQA0tOByEjg33+Vjogo7xQtMy1evDjH18qWLQsb685DRP9PP7dMq1ZAHqq/pAIajZwh+OJFIC5ODtneswfw81M6MqJnY791IjIZJ8qzT87OwIoVQMWKMqnp0AFITlY6KqJnYzJDRCb57z/g8GFAqwU6dVI6GjK3IkWAP/4AihaVa2517y5LT0S2jMkMEZlEf1cmLEz+4iP7ExQkR6s5OwO//w4MHap0RES5YzJDRCbRj2Jiicm+NWoE/Pij3J44EfjhB0XDIcoVkxkiyrNjx+QoF50O6NhR6WjI0rp1A776Sm736SPLT0S2iMkMEeWZvsTUujXg7a1sLGQdw4YBb70FZGQAr74qh+QT2RomM0SUZ5wor+DRaICZM4EWLYD79+WQ7StXlI6KyBiTGSLKk6NHZZmJJaaCx8kJWL4cqFJFJjLt28vEhshWMJkhojzR35Vp0wYoXFjRUEgBhQsD69YBxYoBBw8Cr73GIdtkO5jMENEzCcGJ8ggoVw5YvRpwcQHWrgUGDlQ6IiKJyQwRPdPRo8Dx47Lc0KGD0tGQkurXB376SW5PnSofREpjMkNEz6QvMb30EuDlpWwspLzISOCbb+T2gAHAmjXKxkPEZIaIciUEJ8qjrIYMAd57Tw7Z7tYNOHBA6YioIGMyQ0S5OnIEOHFCTm3fvr3S0ZCt0GiA77+Xy1o8eAC8/DJw6ZLSUVFBxWSGiHKVucTk6alsLGRbdDrZMbxqVeDaNZnQJCUpHRUVRExmiChHLDHRs3h5ySHbJUrI1dRffRVIS1M6KipomMwQUY7++Qc4dYolJspdQIDsBOzqCqxfD/TtKxNhImthMkNEOdLflWnXDvDwUDYWsm316gG//CL70syYAUyerHREVJAwmSGibHGiPDJV587AhAlye9AgYNUqRcOhAoTJDBFl69Ah4PRpOdvryy8rHQ2pxYABwEcfyWT49deB/fuVjogKAiYzRJQtfYkpPBwoVEjZWEg9NBo5K/BLLwEPH8q+VhcuKB0VWUJ6OhAXp8H27SURF6dRdK0uJjNElAVHMdHz0GqBJUuAGjWA+Hh5Z+/ePaWjInNasQIoWxYIC9Ni4sS6CAvTomxZuV8JTGaIKIsDB4CzZ+XolPBwpaMhNfL0lItR+vkB//4LdO0KpKYqHRWZw4oVckmLy5eN91+5IvcrkdAwmSGiLDKXmNzdlY2F1Kt0aZnQuLkBsbFAr14csq126elAv37ZX0f9vv79YfWSk9a6H0dEto4lJjKnF14AFi8GOnYEZs8GKlSQ6zqRsh4/BhITnzySkoyf5/Ta5ctZ78hkJoRc1mLHDqBZM6s1h8kMERnbvx84f17+b7pdO6WjIXvQvr2cd6ZfP2DoUKBcOVmOINNkZADJyXlPPHJ7LSXFsrFeu2bZ8z+NyQwRGdHflXn5ZZaYyHz69pVD/adNA3r0kCWounWfjIZxd9egeXPA0VHpSM0vNVWDW7fk6K78Jh/65+Yu07m5yf5N+oeHh/Hzp1+7cAEYMeLZ5/XzM2+cz8JkhogMOFEeWdKkScC5c7IfTevW8hdpfLwWQF1MnAiUKgVMmQJERCgdqfy7kJyc/8TjyUOLlJQOZo3N0THvyUdu+z085MgzU6SnA//7n+zsm11ipdHI69ikiXnamldMZojIYN8++T8vd3egbVuloyF74+gI/PqrHLJ97pz8ZZ+ZfjTMsmX5T2hSU58v+dDvT0qSZZ3npzFsubqannhk95qrq0walODoKBPOyEgZQ+aERh/T5MnWv8PGZIaIDPQlpvbt5f+aiczN1RV49Cj714SQvxA//lhO1Ki/M2JKUpLTufPLwSF/dz2eJB6p2Ls3FhERYXB11Zk3OIVERMiEs18/487ApUrJREaJO2tMZogIAEcxkXXs2JF751AhgOvXgTZtnu9zMt8FyU8pRv+am9vz3QVJTQWOHk01uZxj6yIi5Ai1rVvTsH79IbRtWwvNm2sV6/NkZ18vEeXX3r1ySGWhQnIqeiJLyOsol1KlZCfh/NwR8fAAdPZxE8SmOToCoaECyclXEBpaU9HO20xmiAiAcYnJ1VXZWMh+5XWUy08/WXeeElI3zgBMRMjI4Cgmso4mTeRdl5xKNxqNvCNj7dEwpG5MZogIe/bIjnwsMZGl6UfDAFkTGiVHw5C6MZkhIkOJqWNHwMVF2VjI/ulHw5Qsaby/VKnnG5ZNBRf7zBAVcBkZ8hcIwBITWY+tjYYhdWMyQ1TA7d4tJyvz9JSzshJZiy2NhiF1Y5mJqIBjiYmI1M7kZKZs2bIYPXo0Ll68aIl4iMiKOIqJiOyBycnMoEGD8PvvvyMwMBBhYWFYvHgxUiy9ljgRWcTOnXISMy8vICxM6WiIiPLH5GSmT58++Pvvv/H3338jODgYffv2hZ+fH3r37o0DBw5YIkYispDMJSZnZ2VjISLKr3z3malZsyamTJmCK1euYOTIkZgzZw7q1auHmjVrYt68eRDZrQ1uR9LTgbg4DbZvL4m4OA3S05WOiMg06ekcxURE9iHfyUxqaip+++03dOjQAYMGDULdunUxZ84cvPLKKxg+fDi6d+9uzjhtyooVQNmyQFiYFhMn1kVYmBZly8r9RGrx559AfDxLTESkfiYPzT5w4ADmz5+PX3/9FY6OjujRowcmTZqEypUrG45p3bo1mjZtatZAbcWKFUBkpFzZNbMrV+R+TvhEaqEvMXXuDDg5KRsLEdHzMDmZqVevHsLCwjBjxgx06tQJumyWJg0ODka3bt3MEqAtSU8H+vXLmsgAcp9GA/TvL/sfcL4EsmXp6cDy5XKbJSYiUjuTk5mzZ88iICAg12Pc3d0xf/78fAdlq3bskOvX5EQI4NIleRxXeyVbtmMHcP064O0NtGypdDRERM/H5D4zCQkJ2Lt3b5b9e/fuxf79+80SlK26ds28xxEphSUmIrInJiczvXr1wqVLl7Lsv3LlCnr16mWWoGyVn595jyNSQloaS0xEZF9MTmaOHTuGF154Icv+2rVr49ixY2YJylY1aSJXdX162Xo9jQYoXVoeR2Srtm8HEhKAIkWAFi2UjoaI6PmZnMw4Ozvj+vXrWfZfu3YNWq19r1vp6AhMmSK3c0poJk9m51+ybZlLTNn03yciUh2Tk5mwsDAMGzYM9+7dM+y7e/cuPvvsM4QVgMkqIiLk8OuSJbO+NmwYh2WTbWOJiYjskcnJzHfffYdLly4hICAAzZs3R/PmzVGuXDnEx8fju+++s0SMNiciAjh/HoiNTcPAgfvRtauc/nfJEuDRI2VjI8rNtm3AzZuAjw/QvLnS0RARmYfJyUzJkiVx+PBhjB8/HsHBwahTpw6mTJmCI0eOoHTp0paI0SY5OgKhoQJNm17BzJkZ8PcHzpwBJkxQOjKinOlLTBERLDERkf3IVycXd3d3vP/+++aORbU8PGQS8/rrwNdfAz16AM+YiofI6tLSniy5wRITEdmTfK/NdOzYMWzYsAGrV682epgiOjoaGo3G6OHr62t4XQiB6Oho+Pv7w9XVFc2aNcPRo0fzG7JFdesGhIYCDx8CAwYoHQ1RVlu3ArduAUWLclJHIrIv+ZoBuHPnzjhy5Ag0Go1hdWzN/w/vSTdx+eiqVati06ZNhueOmYYCjR8/HhMnTsSPP/6IihUrYsyYMQgLC8OJEyfg4eFhaugWpdEA06cDtWoBK1cCGzcCbdooHRXRE/oSU5cugJ0PPCSiAsbkOzP9+vVDuXLlcP36dbi5ueHo0aPYvn076tati23btpkcgFarha+vr+FRrFgxAPKuzOTJkzF8+HBERESgWrVqWLBgAR48eIBFixaZ/DnWUK0a0Lev3O7TB0hJUTYeIr3UVJaYiMh+mfz/s927d2PLli0oVqwYHBwc4ODggMaNG2Ps2LHo27cvDh48aNL5Tp06BX9/fzg7OyMkJARff/01AgMDce7cOcTHx6N169aGY52dnREaGopdu3bhgw8+yPZ8KSkpSMmURSQmJgIAUlNTkZqaampzc6U/X+bzDh8O/PqrFqdOafDtt+n45JMMs36mNWXXPnti7+0DnrQtNjYdt2/rULy4QIMGabCXJtv7NbT39gH230a27/nPnRcmJzPp6ekoVKgQAKBo0aK4evUqKlWqhICAAJw4ccKkc4WEhGDhwoWoWLEirl+/jjFjxqBhw4Y4evQo4uPjAQAlSpQwek+JEiVw4cKFHM85duxYjBo1Ksv+mJgYuLm5mRRfXsXGxho979atFCZProMxYwR8fbeiWLGHFvlca3m6ffbG3tsHANOmXQcQgBdeOI+YmMNKh2N29n4N7b19gP23ke0z3YMHD/J8rMnJTLVq1XD48GEEBgYiJCQE48ePh5OTE2bNmoXAwECTztW2bVvDdvXq1dGgQQMEBQVhwYIFqF+/PoAnfXH0hBBZ9mU2bNgwDBw40PA8MTERpUuXRuvWreHp6WlSfM+SmpqK2NhYhIWFQZdpnGvbtsD+/Rn4808t1q9vhcWLTetHZCtyap+9sPf2AbKNf/yxCQcOlAEADBxYGs2alVI4KvOx92to7+0D7L+NbF/+6SsreWFyMvP5558jOTkZADBmzBi8/PLLaNKkCXx8fLBkyRJTT2fE3d0d1atXx6lTp9CpUycAQHx8PPwyrdyYkJCQ5W5NZs7OznB2ds6yX6fTWewHKbtzf/898MILwIoVDoiLc0CrVhb5aKuw5HdnC+y9fYcPF8OdOxoULw60aKG1y+U27P0a2nv7APtvI9uXv3PmlckdgNu0aYOI/5+zPzAwEMeOHcPNmzeRkJCAFs+5al1KSgqOHz8OPz8/lCtXDr6+vka3rh4/foy4uDg0bNjwuT7HGmrUAPSLiPfuDTx+rGw8VHDt2uUPAIiM5LphRGSfTEpm0tLSoNVq8e+//xrtL1KkSK6ln5wMHjwYcXFxOHfuHPbu3YvIyEgkJiYiKioKGo0G/fv3x9dff42VK1fi33//Rc+ePeHm5obXX3/d5M9SwqhRQPHiwIkTcgFKImt7/BjYs0fe2eQoJiKyVyaVmbRaLQICAkyeSyYnly9fxmuvvYabN2+iWLFiqF+/Pvbs2YOA/58+d+jQoXj48CE+/vhj3LlzByEhIYiJibG5OWZyUrgwMH480LMnMHq0nCG4lP10VyAV2LxZg+RkHXx9BRo3Nv0/HEREamBymenzzz/HsGHDcPv27ef+8MWLF+Pq1at4/Pgxrly5guXLlyM4ONjwukajQXR0NK5du4ZHjx4hLi4O1apVe+7PtaYePYCGDYHkZGDwYKWjoYJm2TL5VzwiIoMlJiKyWyZ3AJ46dSpOnz4Nf39/BAQEwN3d3ej1AwcOmC04e+DgIDsD16kjV9X+4AOuVkzWkZICrF4t78ZERgqFoyEishyTkxn9KCPKu1q1gI8+kklN797AoUNcsZgsLzYWuHdPA2/vR2jYkLdliMh+mZzMjBw50hJx2L0vv5Rr4xw7BkydCgwapHREZO/0azE1bHgVDg6llQ2GiMiC8r1qNpnG2xsYN05uR0cDV68qGg7ZuUePgN9/l9uNGl1RNhgiIgszOZlxcHCAo6Njjg/KWc+eQEgIcP8+MGSI0tGQPYuJARITAX9/gcqVn7+zPhGRLTO5zLRy5Uqj56mpqTh48CAWLFiQ7ZpI9IS+M3C9esCiRcD77wOhoUpHRfZo6VL5Z5cuGXDg/VcisnMmJzMdO3bMsi8yMhJVq1bFkiVL8M4775glMHtVp44c0TRzpuwMfOAAOwOTeWUuMUVGCty5o2w8RESWZrb/s4WEhGDTpk3mOp1d++orwMcH+PdfeaeGyJw2bgSSkuQEjSEhHJJNRPbPLMnMw4cPMW3aNJTi9LZ5UqQIMHas3B45EoiPVzYesi/6UUxdu4IlJiIqEEwuM3l7exutwySEQFJSEtzc3PDzzz+bNTh79s47wOzZwL59wNChwMKFSkdE9uDhQ2D1arnNtZiIqKAwOZmZNGmSUTLj4OCAYsWKISQkBN7e3mYNzp7pOwOHhAA//SQ7AzdurHRUpHYbNsjRcmXKyJ+ttDSlIyIisjyTk5mePXtaIIyCqV494N135R2aXr2Av/8GtCZfEaIn9CWmyEggHwvZExGpkskV9fnz52OpftxnJkuXLsWCBQvMElRB8vXXsg/N4cPAjBlKR0Nq9uABsGaN3GaJiYgKEpOTmXHjxqFo0aJZ9hcvXhxff/21WYIqSIoWlaObAGDECOD6dWXjIfVav16uzl6mDPDii0pHQ0RkPSYnMxcuXEC5cuWy7A8ICMDFixfNElRB8957wAsvAPfuAZ9+qnQ0pFb6G6avvMISExEVLCYnM8WLF8fhw4ez7P/nn3/g4+NjlqAKGkfHJ/PN/PgjsHu3ouGQCrHEREQFmcnJTLdu3dC3b19s3boV6enpSE9Px5YtW9CvXz9069bNEjEWCPXrA2+/Lbd79QLS05WNh9Tljz9kQlO2LFC3rtLREBFZl8nJzJgxYxASEoKWLVvC1dUVrq6uaN26NVq0aME+M89p3DigcGHg4EHgf/9TOhpSE/0oJpaYiKggMjmZcXJywpIlS3DixAn88ssvWLFiBc6cOYN58+bBycnJEjEWGMWKAWPGyO3hw4EbN5SNh9QhORlYu1Zus8RERAVRvmc1qVChAipUqGDOWAjAhx8Cc+YAhw4Bw4bJbaLcrFsnZ/4NDJQdyYmIChqT78xERkZi3LhxWfZ/++236Nq1q1mCKsgydwaeOxfYu1fZeMj2ZV6LiSUmIiqITE5m4uLiEB4enmX/Sy+9hO3bt5slqIKuYUMgKkpuszMw5eb+fXlnBmCJiYgKLpOTmfv372fbN0an0yExMdEsQRHwzTeAl5dc4oClJsrJ2rXAo0dAUBBQu7bS0RARKcPkZKZatWpYsmRJlv2LFy9GcHCwWYIioEQJYPRouf3ZZ8CtW8rGQ7aJE+UREeWjA/CIESPQpUsXnDlzBi1atAAAbN68GYsWLcKyZcvMHmBB9vHHst/M4cMyoeFwbcosKUnOLwOwxEREBZvJd2Y6dOiAVatW4fTp0/j4448xaNAgXLlyBVu2bEHZsmUtEGLBpdUC06fL7dmzgf37lY2HbIu+xFShAlCzptLREBEpx+RkBgDCw8Oxc+dOJCcn4/Tp04iIiED//v1Rp04dc8dX4DVpArzxBiCE7AyckaF0RGQrOFEeEZGUr2QGALZs2YI33ngD/v7+mD59Otq1a4f9vHVgEePHAx4ewF9/AfPmKR0N2YLERLlKNsASExGRScnM5cuXMWbMGAQGBuK1116Dt7c3UlNTsXz5cowZMwa1OZzCIvz8gFGj5PannwK3bysbDylvzRogJQWoVAmoXl3paIiIlJXnZKZdu3YIDg7GsWPHMG3aNFy9ehXTpk2zZGyUSe/eQNWqclTT558rHQ0pjRPlERE9kedkJiYmBu+++y5GjRqF8PBwODo6WjIueopO92Rm4JkzgQMHlI2HlHPvHrBhg9xmiYmIyIRkZseOHUhKSkLdunUREhKC6dOn4wZXQrSq0FDgtdfYGbigW70aePwYqFwZqFZN6WiIiJSX52SmQYMGmD17Nq5du4YPPvgAixcvRsmSJZGRkYHY2FgkJSVZMk76fxMmAIUKAXv2AAsWKB0NKYET5RERGTN5NJObmxvefvtt/Pnnnzhy5AgGDRqEcePGoXjx4ujQoYMlYqRM/P2BkSPl9iefAHfuKBsPWdfdu8DGjXKbJSYiIinfQ7MBoFKlShg/fjwuX76MX3/91Vwx0TP06wdUqQLcuAF88YXS0ZA16UtMwcGyQzgRET1nMqPn6OiITp06YfXq1eY4HT2DTvdkZuAffgAOHVI0HLKizBPlERGRZJZkhqyvRQv5Cy0jQw7bFkLpiMjS7twBYmLkdteuysZCRGRLmMyo2HffAe7uwM6dwE8/KR0NWdrvvwOpqXIEExeoJyJ6gsmMipUqBYwYIbeHDpXzj5D9yjxRHhERPcFkRuUGDJBT2l+//mSUE9mf27eB2Fi5zWSGiMgYkxmVc3IC9KtKTJ8OHDmibDxkGatWAWlpch2mKlWUjoaIyLYwmbEDYWFAly5AerqcGZidge1P5onyiIjIGJMZOzFxIuDmBuzYASxapHQ0ZE63bgGbNsltlpiIiLJiMmMnypQBhg+X24MHA4mJysZD5qMvMdWsKftHERGRMSYzdmTQIKBCBSA+Hhg1SuloyFw4UR4RUe6YzNgRZ2dg6lS5PWUKcPSosvHQ87t5E9i8WW6zxERElD0mM3bmpZeATp1kZ2DODKx+K1fKa1mrlrzrRkREWTGZsUOTJgEuLsC2bcCSJUpHQ8+DJSYiomdjMmOHypYFPvtMbg8aBCQlKRoO5dONG8CWLXKbJSYiopwxmbFTQ4YAQUHA1avAl18qHQ3lx4oVciHRF14AypdXOhoiItvFZMZOubjITsCALDsdP65sPGQ6TpRHRJQ3TGbsWHg40L69nKOkTx92BlaThARg61a5zRITEVHumMzYucmT5ZDtzZuBZcuUjobySl9iqlsXCAxUOhoiItvGZMbOBQYCn34qtwcOBO7fVzYeyhuOYiIiyjsmMwXAJ58A5coBly8DX32ldDT0LPHxQFyc3GaJiYjo2WwmmRk7diw0Gg369+9v2Hf//n307t0bpUqVgqurK6pUqYIZM2YoF6RKubrKchMAfPcdcOKEouHQM+hLTPXqyWH2RESUO5tIZvbt24dZs2ahRo0aRvsHDBiADRs24Oeff8bx48cxYMAA9OnTB7///rtCkapX+/ZAu3ZAairQty87A9sylpiIiEyjeDJz//59dO/eHbNnz4a3t7fRa7t370ZUVBSaNWuGsmXL4v3330fNmjWxf/9+haJVL41GDtV2cgJiYuQ0+WR7rl0Dtm+X2ywxERHljVbpAHr16oXw8HC0atUKY8aMMXqtcePGWL16Nd5++234+/tj27ZtOHnyJKboJ1DJRkpKClJSUgzPExMTAQCpqalITU01a+z685n7vJYSEAAMGuSAsWMdMWCAQMuWaXBzy/l4tbXPVLbYvt9+c4AQjnjxxQz4+6fjeUOzxTaaE9unfvbeRrbv+c+dFxohlCs4LF68GF999RX27dsHFxcXNGvWDLVq1cLk/+/g8fjxY7z33ntYuHAhtFotHBwcMGfOHPTo0SPHc0ZHR2PUqFFZ9i9atAhuuf3mLiBSUhzRu3cL3Ljhhq5dT6B79/+UDokyGT68EY4eLYq33voXHTueUTocIiLFPHjwAK+//jru3bsHT0/PXI9VLJm5dOkS6tati5iYGNSsWRMAsiQzEyZMwOzZszFhwgQEBARg+/btGDZsGFauXIlWrVple97s7syULl0aN2/efOaXYarU1FTExsYiLCwMOp3OrOe2pFWrNHjlFS2cnAQOHkzLcTVmtbYvr2ytfVevAuXKaSGEBqdPp6JMmec/p6210dzYPvWz9zayffmXmJiIokWL5imZUazM9PfffyMhIQF16tQx7EtPT8f27dsxffp03Lt3D5999hlWrlyJ8PBwAECNGjVw6NAhTJgwIcdkxtnZGc7Ozln263Q6i/0gWfLclhAZCbRpA2zcqMHgwTqsWyf71OREbe0zla20b/Vq2TG7QQMgKMi88dhKGy2F7VM/e28j25e/c+aVYslMy5YtceTIEaN9b731FipXroxPPvkE6enpSE1NhYODcR9lR0dHZGRkWDNUu6PRAFOnAtWqAevXy1+iHTsqHRVxFBMRUf4olsx4eHigWrVqRvvc3d3h4+Nj2B8aGoohQ4bA1dUVAQEBiIuLw8KFCzFx4kQlQrYrFSsCgwcDY8cC/fsDrVvL+WhIGVeuAH/+KbcjI5WNhYhIbRQfmp2bxYsXo169eujevTuCg4Mxbtw4fPXVV/jwww+VDs0uDB8OlC4NnD8PjBundDQFm37drIYNgVKllI2FiEhtFB+andm2bduMnvv6+mL+/PnKBFMAuLsDEyfK+Uy++QZ4800gKEjpqAomlpiIiPLPpu/MkOV16QK0agWkpMhyE1nfpUvArl1ymyUmIiLTMZkp4DQaYNo0QKcD1q6VD7IufYmpcWOgZEllYyEiUiMmM4TKlYEBA+R2v37Ao0fKxlPQLF0q/2SJiYgof5jMEABgxAh5V+DsWWD8eKWjKTguXgR275Z3yLp0UToaIiJ1YjJDAIBChYDvvpPbY8cC584pG09BoS8xNWkC+PsrGwsRkVoxmSGDV14BWrSQZSZ92Yksi6OYiIieH5MZMtB3BtZqgd9/BzZsyGWNA3pu588De/eyxERE9LyYzJCR4GDZCRgABg50RGoqf0QsRV9iatoU8PVVNhYiIjXjbyrKYuRIwM8POH1ag1Wryisdjt1iiYmIyDyYzFAWHh7AhAlye+nSCrhwQdl47NG5c8C+fYCDAxARoXQ0RETqxmSGsvXaa0DTphl4/FiLIUMclQ7H7uhLTKGhLDERET0vJjOULY0GmDw5HQ4OGVi1ygExMUpHZF9YYiIiMh8mM5SjatWA8HA54UyfPnL9Jnp+Z88C+/ezxEREZC5MZihX3br9hxIlBE6eBCZNUjoa+6BfvqB5c6B4cWVjISKyB0xmKFfu7mkYOzYdAPDll3KFZ3o+LDEREZkXkxl6pu7dBRo3Bh48AAYNUjoadTt9GjhwAHB0BDp3VjoaIiL7wGSGnkmjAaZPl308li4FNm1SOiL1ylxiKlZM2ViIiOwFkxnKk5o1gV695HafPsDjx8rGo1YsMRERmR+TGcqz0aNlh9X//gOmTFE6GvU5dQo4dIglJiIic2MyQ3lWuDDwzTdye9Qo4MoVRcNRHX2JqWVLoGhRZWMhIrInTGbIJG++CTRoACQnA4MHKx2NurDERERkGUxmyCQODsD338s/Fy8Gtm5VOiJ1OHEC+OcfQKsFOnVSOhoiIvvCZIZMVrs28OGHcrt3byA1Vdl41EBfYmrVCvDxUTYWIiJ7w2SG8mXMGNnv49gxYNo0paOxfSwxERFZDpMZyhdvb2DcOLkdHQ1cu6ZoODbt+HHgyBFAp2OJiYjIEpjMUL699RYQEgIkJQFDhigdje3KXGLy9lY2FiIie8RkhvLNwUHODKzRAL/8AmzfrnREtoklJiIiy2IyQ8+lbl3g/ffldu/eQFqasvHYmmPHgKNHZYmpY0eloyEisk9MZui5ffUVUKSI7Bfy/fdKR2Nb9CWm1q1ZYiIishQmM/TcfHyAsWPl9hdfAPHxysZjS1hiIiKyPCYzZBbvvCNLTomJwCefKB2NbTh6VJaZnJyADh2UjoaIyH4xmSGzcHSUJSaNBli4ENi5U+mIlKe/K9OmjVzXioiILIPJDJnNiy/KOzQA0KtXwe4MLARLTERE1sJkhsxq7FjZ0fWff4CZM5WORjn//gv89x/g7MwSExGRpTGZIbMqWlSObgKAzz8HEhKUjUcpmUtMnp7KxkJEZO+YzJDZvf8+8MILwL17wKefKh2N9bHERERkXUxmyOwcHeXMwAAwfz6we7ey8Vjb4cPAyZOyxNS+vdLREBHZPyYzZBENGsi1mwA5M3B6urLxWJN+ory2bVliIiKyBiYzZDHjxskhyQcOALNmKR2NdbDERERkfUxmyGKKFwe+/FJuDx8O3LypbDzW8M8/wKlTgIsL8PLLSkdDRFQwMJkhi/rwQ6BmTeDOHWDYMKWjsTz9XZl27QAPD2VjISIqKJjMkEVptU8Wn5w7F/jrL2XjsSSWmIiIlMFkhiyuUSPgzTflL/tevey3M/DBg8CZM4CrKxAernQ0REQFB5MZsorx4+XInv375R0ae5S5xFSokLKxEBEVJExmyCpKlABGj5bbw4YBt24pG4+5scRERKQcJjNkNb16AdWrA7dvy9FN9uTvv4Fz51hiIiJSApMZshqt9snMwLNmyZKTvdBPlPfyy4C7u7KxEBEVNExmyKqaNgW6d5dlmd69gYwMpSN6fiwxEREpi8kMWd2338o5WPbulWs3qd3+/cD584Cbm+z8S0RE1sVkhqzOzw+Ijpbbn34q+9Comf6uTPv2MqEhIiLrYjJDiujTB6haVS5xMGKE0tHkH0tMRETKYzJDitDpnnQGnjlTTjinRn/9BVy8KDv9tm2rdDRERAUTkxlSTLNmQLdushNwr17q7AycucTk6qpsLEREBRWTGVLUhAlyttzdu4GFC5WOxjQZGU+GZLPERESkHJtJZsaOHQuNRoP+/fsb7T9+/Dg6dOgALy8veHh4oH79+rh48aIyQZLZlSwJfPGF3B46FLh7V9FwTLJ3L3DpkkzGXnpJ6WiIiAoum0hm9u3bh1mzZqFGjRpG+8+cOYPGjRujcuXK2LZtG/755x+MGDECLi4uCkVKltCvH1C5MnDjxpPERg30d2U6dGCJiYhISYonM/fv30f37t0xe/ZseHt7G702fPhwtGvXDuPHj0ft2rURGBiI8PBwFC9eXKFoyRKcnJ50Bv7+e+Cff5SNJy9YYiIish1apQPo1asXwsPD0apVK4wZM8awPyMjA+vWrcPQoUPRpk0bHDx4EOXKlcOwYcPQqVOnHM+XkpKClJQUw/PExEQAQGpqKlJTU80au/585j6vrbBm+5o2Bbp0ccTy5Q7o1SsDW7akQ6Ox7Gc+T/t279bg8mUtPDwEWrRIg63+CPBnVN3svX2A/beR7Xv+c+eFRgghzB5BHi1evBhfffUV9u3bBxcXFzRr1gy1atXC5MmTER8fDz8/P7i5uWHMmDFo3rw5NmzYgM8++wxbt25FaGhotueMjo7GqFGjsuxftGgR3DijmU27ccMFvXu3REqKFv36/Y3mzS8rHVKO5syphrVrgxAaegkDBhxQOhwiIrvz4MEDvP7667h37x48PT1zPVaxZObSpUuoW7cuYmJiULNmTQAwSmauXr2KkiVL4rXXXsOiRYsM7+vQoQPc3d3x66+/Znve7O7MlC5dGjdv3nzml2Gq1NRUxMbGIiwsDDqdzqzntgVKtG/8eAd8/rkjSpQQ+PffNHh5We6z8tu+jAwgMFCLq1c1WLEiDS+/rNj/B56JP6PqZu/tA+y/jWxf/iUmJqJo0aJ5SmYUKzP9/fffSEhIQJ06dQz70tPTsX37dkyfPh3JycnQarUIDg42el+VKlXw559/5nheZ2dnODs7Z9mv0+ks9oNkyXPbAmu2b8gQOUT75EkNvvpKh0mTLP+Zprbvzz+Bq1cBT0+gXTst1HDp+TOqbvbePsD+28j25e+ceaVYB+CWLVviyJEjOHTokOFRt25ddO/eHYcOHYKzszPq1auHEydOGL3v5MmTCAgIUChqsjQnJ2DaNLk9bRpw5Iiy8WRHP1Fex45ANnkzERFZmWJ3Zjw8PFCtWjWjfe7u7vDx8THsHzJkCF599VU0bdrU0GdmzZo12LZtmwIRk7W0bg1ERAArVgC9ewPbtsHinYHzKj0dWLZMbnMUExGRbVB8aHZuOnfujJkzZ2L8+PGoXr065syZg+XLl6Nx48ZKh0YWNmmSnLtl+3Ygh+5Riti5E7h2DfDyAsLClI6GiIgAGxianVl2d1zefvttvP3229YPhhRVpgwwfDjw+efA4MHAyy/LPipK088t06kTS0xERLbCpu/MUME2eDBQvry8EzJ6tNLRsMRERGSrmMyQzXJ2BqZOldtTpgDHjikbz59/AvHxQOHCQKtWysZCRERPMJkhm9a2rRw1lJYmOwMrN8Xjk1FMnTvLUVdERGQbmMyQzZs8GXBxAbZufZJQWBtLTEREtovJDNm8smWBYcPk9qBBwP371o9h+3YgIQHw9gZatrT+5xMRUc6YzJAqDB0KBAYCV64AX35p/c/PXGKy40k8iYhUickMqYKLi+wEDAATJwL//We9z05LA5Yvl9ssMRER2R4mM6QaL78sH2lpQJ8+1usMHBcH3LgBFCkCtGhhnc8kIqK8YzJDqjJlihyyvWnTk7sllqafKC8igiUmIiJbxGSGVCUwEPjkE7k9cCCQnGzZz2OJiYjI9jGZIdX59FM5wunSJeCrryz7Wdu2ATdvAj4+QPPmlv0sIiLKHyYzpDqurnLuGQCYMAE4edJyn6UfxdSlC6C1qZXMiIhIj8kMqVKHDnJ24NRUoG9fy3QGTk0FVqyQ2ywxERHZLiYzpEoajVy3yckJ2LgRWLXK/J+xdStw6xZQtCgQGmr+8xMRkXkwmSHVKl8eGDJEbvfvDzx4YN7zs8RERKQOTGZI1T77DChTBrh4ERg71nznZYmJiEg9mMyQqrm5AZMmye3x44HTp81z3s2bgTt3gOLFgaZNzXNOIiKyDCYzpHqdOwOtWwOPHwP9+pmnM7B+ojyWmIiIbB+TGVI9jQaYNk3OzvvHH8CaNc93vsePgZUr5TZLTEREto/JDNmFihWBQYPkdr9+wMOH+T+XvsRUogTQpIl54iMiIsthMkN24/PPgVKlgPPngW++yf959KOYIiMBR0ezhEZERBbEZIbshrs7MHGi3B43Djh71vRzsMRERKQ+TGbIrkRGAi1bAikpcu4ZU8XGAvfuAb6+QKNGZg+PiIgsgMkM2RWNBpg+XXYGXrMGWLfOtPezxEREpD5MZsjuVK4MDBggt/v2BR49ytv7UlKeLIvAEhMRkXowmSG7NGIEULKk7Dfz7bd5e09sLJCYCPj5scRERKQmTGbILhUqBHz3ndz++ms5wulZ9CWmrl0BB/7NICJSDf6TTXbrlVeAFi1kmUlfdsrJo0fA778/eR8REakHkxmyW/qZgbVa2Rdmw4acj42JkSWmkiWBBg2sFiIREZkBkxmya8HBckZgAOjTR3byzQ5LTERE6sV/tsnujRwpO/WePv2kH01mDx+yxEREpGZMZsjueXgAEybI7TFjgIsXjV+PidHg/n25FEJIiPXjIyKi58NkhgqE114DQkPlXZiBA41fW7ZM/jVgiYmISJ34TzcVCPqZgR0dgeXL5ZwyAJCS4oC1azUAWGIiIlIrJjNUYFSrJjsBA0Dv3sCmTRr88ksVJCdrULo0S0xERGrFZIYKlOhowMsLOHkSaNdOi9WrywMA7tx5slo2ERGpC5MZKlA2b5arYj/t/n25uOSKFdaPiYiIng+TGSow0tOfzDmTk/795XFERKQeTGaowNixA7h8OefXhQAuXZLHERGRejCZoQLj2jXzHkdERLaByQwVGH5+5j2OiIhsA5MZKjCaNJGz/Go02b+u0QClS8vjiIhIPZjMUIHh6AhMmSK3n05o9M8nT5bHERGRejCZoQIlIgJYtgwoWdJ4f6lScn9EhDJxERFR/mmVDoDI2iIigI4dga1b07B+/SG0bVsLzZtreUeGiEilmMxQgeToCISGCiQnX0FoaE0mMkREKsYyExEREakakxkiIiJSNSYzREREpGpMZoiIiEjVmMwQERGRqjGZISIiIlVjMkNERESqxmSGiIiIVI3JDBEREama3c8ALIQAACQmJpr93KmpqXjw4AESExOh0+nMfn6lsX3qZ+9tZPvUz97byPbln/73tv73eG7sPplJSkoCAJQuXVrhSIiIiMhUSUlJ8PLyyvUYjchLyqNiGRkZuHr1Kjw8PKDRaMx67sTERJQuXRqXLl2Cp6enWc9tC9g+9bP3NrJ96mfvbWT78k8IgaSkJPj7+8PBIfdeMXZ/Z8bBwQGlSpWy6Gd4enra5Q+pHtunfvbeRrZP/ey9jWxf/jzrjoweOwATERGRqjGZISIiIlVjMvMcnJ2dMXLkSDg7OysdikWwfepn721k+9TP3tvI9lmH3XcAJiIiIvvGOzNERESkakxmiIiISNWYzBAREZGqMZkhIiIiVWMyk4Pt27ejffv28Pf3h0ajwapVq575nri4ONSpUwcuLi4IDAzEzJkzLR9oPpnavm3btkGj0WR5/Pfff9YJ2ERjx45FvXr14OHhgeLFi6NTp044ceLEM9+npmuYnzaq6TrOmDEDNWrUMEzG1aBBA6xfvz7X96jp+pnaPjVdu+yMHTsWGo0G/fv3z/U4NV3Dp+WljWq6jtHR0Vni9PX1zfU9Sl0/JjM5SE5ORs2aNTF9+vQ8HX/u3Dm0a9cOTZo0wcGDB/HZZ5+hb9++WL58uYUjzR9T26d34sQJXLt2zfCoUKGChSJ8PnFxcejVqxf27NmD2NhYpKWloXXr1khOTs7xPWq7hvlpo54armOpUqUwbtw47N+/H/v370eLFi3QsWNHHD16NNvj1Xb9TG2fnhqu3dP27duHWbNmoUaNGrkep7ZrmFle26inlutYtWpVoziPHDmS47GKXj9BzwRArFy5Mtdjhg4dKipXrmy074MPPhD169e3YGTmkZf2bd26VQAQd+7csUpM5paQkCAAiLi4uByPUfM1FCJvbVT7dfT29hZz5szJ9jW1Xz8hcm+fWq9dUlKSqFChgoiNjRWhoaGiX79+OR6r1mtoShvVdB1Hjhwpatasmefjlbx+vDNjJrt370br1q2N9rVp0wb79+9HamqqQlGZX+3ateHn54eWLVti69atSoeTZ/fu3QMAFClSJMdj1H4N89JGPbVdx/T0dCxevBjJyclo0KBBtseo+frlpX16art2vXr1Qnh4OFq1avXMY9V6DU1po55aruOpU6fg7++PcuXKoVu3bjh79myOxyp5/ex+oUlriY+PR4kSJYz2lShRAmlpabh58yb8/PwUisw8/Pz8MGvWLNSpUwcpKSn46aef0LJlS2zbtg1NmzZVOrxcCSEwcOBANG7cGNWqVcvxODVfw7y2UW3X8ciRI2jQoAEePXqEQoUKYeXKlQgODs72WDVeP1Pap7ZrBwCLFy/GgQMHsG/fvjwdr8ZraGob1XQdQ0JCsHDhQlSsWBHXr1/HmDFj0LBhQxw9ehQ+Pj5Zjlfy+jGZMSONRmP0XPz/5MpP71ejSpUqoVKlSobnDRo0wKVLlzBhwgSb+wv4tN69e+Pw4cP4888/n3msWq9hXtuotutYqVIlHDp0CHfv3sXy5csRFRWFuLi4HH/hq+36mdI+tV27S5cuoV+/foiJiYGLi0ue36ema5ifNqrpOrZt29awXb16dTRo0ABBQUFYsGABBg4cmO17lLp+LDOZia+vL+Lj4432JSQkQKvVZpvB2oP69evj1KlTSoeRqz59+mD16tXYunUrSpUqleuxar2GprQxO7Z8HZ2cnFC+fHnUrVsXY8eORc2aNTFlypRsj1Xj9TOlfdmx5Wv3999/IyEhAXXq1IFWq4VWq0VcXBymTp0KrVaL9PT0LO9R2zXMTxuzY8vXMTN3d3dUr149x1iVvH68M2MmDRo0wJo1a4z2xcTEoG7dutDpdApFZVkHDx60ydu+gPzfQJ8+fbBy5Ups27YN5cqVe+Z71HYN89PG7NjydXyaEAIpKSnZvqa265ed3NqXHVu+di1btswy8uWtt95C5cqV8cknn8DR0THLe9R2DfPTxuzY8nXMLCUlBcePH0eTJk2yfV3R62fxLsYqlZSUJA4ePCgOHjwoAIiJEyeKgwcPigsXLgghhPj0009Fjx49DMefPXtWuLm5iQEDBohjx46JuXPnCp1OJ5YtW6ZUE3JlavsmTZokVq5cKU6ePCn+/fdf8emnnwoAYvny5Uo1IVcfffSR8PLyEtu2bRPXrl0zPB48eGA4Ru3XMD9tVNN1HDZsmNi+fbs4d+6cOHz4sPjss8+Eg4ODiImJEUKo//qZ2j41XbucPD3SR+3XMDvPaqOaruOgQYPEtm3bxNmzZ8WePXvEyy+/LDw8PMT58+eFELZ1/ZjM5EA/fO7pR1RUlBBCiKioKBEaGmr0nm3btonatWsLJycnUbZsWTFjxgzrB55Hprbvm2++EUFBQcLFxUV4e3uLxo0bi3Xr1ikTfB5k1zYAYv78+YZj1H4N89NGNV3Ht99+WwQEBAgnJydRrFgx0bJlS8MveiHUf/1MbZ+arl1Onv5Fr/ZrmJ1ntVFN1/HVV18Vfn5+QqfTCX9/fxERESGOHj1qeN2Wrp9GiP/vnUNERESkQuwATERERKrGZIaIiIhUjckMERERqRqTGSIiIlI1JjNERESkakxmiIiISNWYzBAREZGqMZkhIrPTaDRYtWqV0mHYlB9//BGFCxdWOgwiu8RkhsiO9OzZExqNJsvjpZdeUjo0qzl//jw0Gg0OHToEANi2bRs0Gg3u3r1rtRjKli2LyZMnG+179dVXcfLkSavFQFSQcKFJIjvz0ksvYf78+Ub7nJ2dFYrGctLT06HRaODgYJ3/kwkhkJ6eDq02f/9surq6wtXV1cxRERHAOzNEdsfZ2Rm+vr5GD29vb8PrGo0GM2bMQNu2beHq6opy5cph6dKlRuc4cuQIWrRoAVdXV/j4+OD999/H/fv3jY6ZN28eqlatCmdnZ/j5+aF3795Gr9+8eROdO3eGm5sbKlSogNWrV+ca9507d/Dmm2/C29sbbm5uaNu2LU6dOmV4XV+mWbt2LYKDg+Hs7IwLFy7kes7z58+jefPmAABvb29oNBr07NkTgExOxo8fj8DAQLi6uqJmzZpYtmyZ4b36OzobN25E3bp14ezsjB07duDMmTPo2LEjSpQogUKFCqFevXrYtGmT4X3NmjXDhQsXMGDAAMOdsczxZzZjxgwEBQXByckJlSpVwk8//WT0ukajwZw5c0z6HokKJKusAEVEVhEVFSU6duyY6zEAhI+Pj5g9e7Y4ceKE+Pzzz4Wjo6M4duyYEEKI5ORkw6JyR44cEZs3bxblypUzLEIqhBA//PCDcHFxEZMnTxYnTpwQf/31l5g0aZLRZ5QqVUosWrRInDp1SvTt21cUKlRI3Lp1K8e4OnToIKpUqSK2b98uDh06JNq0aSPKly8vHj9+LIQQYv78+UKn04mGDRuKnTt3iv/++0/cv38/y3nOnTsnAIiDBw+KtLQ0sXz5cgFAnDhxQly7dk3cvXtXCCHEZ599JipXriw2bNggzpw5I+bPny+cnZ3Ftm3bhBBPFmOtUaOGiImJEadPnxY3b94Uhw4dEjNnzhSHDx8WJ0+eFMOHDxcuLi6GFedv3bolSpUqJUaPHm1YyVwfv5eXlyHOFStWCJ1OJ77//ntx4sQJ8d133wlHR0exZcuW5/oeiQoiJjNEdiQqKko4OjoKd3d3o8fo0aMNxwAQH374odH7QkJCxEcffSSEEGLWrFnC29vbKFFYt26dcHBwEPHx8UIIIfz9/cXw4cNzjAOA+Pzzzw3P79+/LzQajVi/fn22x588eVIAEDt37jTsu3nzpnB1dRW//fabEEImAwDEoUOHcv0OMiczQjxJSu7cuWMUj4uLi9i1a5fRe9955x3x2muvGb1v1apVuX6eEEIEBweLadOmGZ4HBAQYJXf6+DMnMw0bNhTvvfee0TFdu3YV7dq1Mzw39XskKqjYZ4bIzjRv3hwzZsww2lekSBGj5w0aNMjyXN9h9vjx46hZsybc3d0Nrzdq1AgZGRk4ceIENBoNrl69ipYtW+YaR40aNQzb7u7u8PDwQEJCQrbHHj9+HFqtFiEhIYZ9Pj4+qFSpEo4fP27Y5+TkZHTe/Dp27BgePXqEsLAwo/2PHz9G7dq1jfbVrVvX6HlycjJGjRqFtWvX4urVq0hLS8PDhw9x8eJFk2I4fvw43n//faN9jRo1wpQpU4z2mfI9EhVUTGaI7Iy7uzvKly9v8vv0fTuEEIbt7I7JaydWnU6X5b0ZGRnZHiuEyHF/5lhcXV1zjM0U+jjWrVuHkiVLGr32dGfpzEkdAAwZMgQbN27EhAkTUL58ebi6uiIyMhKPHz82OY6n25Ldd2/K90hUULEDMFEBtGfPnizPK1euDAAIDg7GoUOHkJycbHh9586dcHBwQMWKFeHh4YGyZcti8+bNZosnODgYaWlp2Lt3r2HfrVu3cPLkSVSpUuW5zu3k5ARAjn7K/HnOzs64ePEiypcvb/QoXbp0rufbsWMHevbsic6dO6N69erw9fXF+fPns3xm5s/LTpUqVfDnn38a7du1a9dzt5eoIOKdGSI7k5KSgvj4eKN9Wq0WRYsWNTxfunQp6tati8aNG+OXX37BX3/9hblz5wIAunfvjpEjRyIqKgrR0dG4ceMG+vTpgx49eqBEiRIAgOjoaHz44YcoXrw42rZti6SkJOzcuRN9+vTJV8wVKlRAx44d8d577+F///sfPDw88Omnn6JkyZLo2LFjPr8JKSAgABqNBmvXrkW7du3g6uoKDw8PDB48GAMGDEBGRgYaN26MxMRE7Nq1C4UKFUJUVFSO5ytfvjxWrFiB9u3bQ6PRYMSIEVnulJQtWxbbt29Ht27d4OzsbPTd6w0ZMgSvvPIKXnjhBbRs2RJr1qzBihUrjEZGEVEeKdtlh4jMKSoqSgDI8qhUqZLhGADi+++/F2FhYcLZ2VkEBASIX3/91eg8hw8fFs2bNxcuLi6iSJEi4r333hNJSUlGx8ycOVNUqlRJ6HQ64efnJ/r06WP0GStXrjQ63svLS8yfPz/H2G/fvi169OghvLy8hKurq2jTpo04efKk4fWnO9Dm5OkOwEIIMXr0aOHr6ys0Go1hVFZGRoaYMmWKoQ3FihUTbdq0EXFxcUKI7DsO68/fvHlz4erqKkqXLi2mT58uQkNDRb9+/QzH7N69W9SoUUM4OzsL/T+z2cX/ww8/iMDAQKHT6UTFihXFwoULjV7Pz/dIVBBphMihWE1Edkmj0WDlypXo1KmT0qEQEZkF+8wQERGRqjGZISIiIlVjB2CiAoaVZSKyN7wzQ0RERKrGZIaIiIhUjckMERERqRqTGSIiIlI1JjNERESkakxmiIiISNWYzBAREZGqMZkhIiIiVWMyQ0RERKr2f9c0WP4BF1CbAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "# Replace with your actual accuracy values\n",
    "accuracy_values = [50.36587157, 44.61139787, 54.86271398, 50.86060226, 51.04672933]\n",
    "epoch_numbers = np.arange(1, len(accuracy_values) + 1)\n",
    "\n",
    "# Create the accuracy curve\n",
    "plt.plot(epoch_numbers, accuracy_values, marker='o', linestyle='-', color='b')\n",
    "plt.title('Tool Accuracy Curve')\n",
    "plt.xlabel('Epoch or Iteration')\n",
    "plt.ylabel('Accuracy')\n",
    "plt.grid(True)\n",
    "\n",
    "# Display the accuracy curve\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa4e55ee",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
