{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4597b62f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d91e8361",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "1           5 prime UTR variant  MODIFIER           NaN      NaN   \n",
       "2       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "3       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "4                intron variant  MODIFIER           NaN      NaN   \n",
       "...                         ...       ...           ...      ...   \n",
       "316574       synonymous variant       LOW           NaN      NaN   \n",
       "316575       synonymous variant       LOW           NaN      NaN   \n",
       "316576           intron variant  MODIFIER           NaN      NaN   \n",
       "316577           intron variant  MODIFIER           NaN      NaN   \n",
       "316578           intron variant  MODIFIER           NaN      NaN   \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                     NaN      NaN                NaN               None   \n",
       "1                     NaN      NaN  ENST00000641515.2           c.-36C>T   \n",
       "2                     NaN      NaN                NaN               None   \n",
       "3                     NaN      NaN                NaN               None   \n",
       "4                     NaN      NaN  ENST00000641515.2          c.9+18C>T   \n",
       "...                   ...      ...                ...                ...   \n",
       "316574                NaN      NaN  ENST00000426000.6           c.351A>G   \n",
       "316575                NaN      NaN  ENST00000540248.5           c.351A>G   \n",
       "316576                NaN      NaN  ENST00000405239.6        c.243-46G>A   \n",
       "316577                NaN      NaN  ENST00000426000.6        c.243-46G>A   \n",
       "316578                NaN      NaN  ENST00000540248.5        c.243-46G>A   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD  \\\n",
       "0                     NaN               None  0/1  23   26  26  19   7   \n",
       "1                     NaN               None  0/1  23   26  26  19   7   \n",
       "2                     NaN               None  0/1  23   26  26  19   7   \n",
       "3                     NaN               None  0/1  22   28  28  21   7   \n",
       "4                     NaN               None  0/1  22   28  28  21   7   \n",
       "...                   ...                ...  ...  ..  ...  ..  ..  ..   \n",
       "316574  ENSP00000400716.3        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316575  ENSP00000444407.1        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316576                NaN               None  0/1  26   33  33  25   8   \n",
       "316577                NaN               None  0/1  26   33  33  25   8   \n",
       "316578                NaN               None  0/1  26   33  33  25   8   \n",
       "\n",
       "          FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...        ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED  \n",
       "0                                  NaN      NaN     1.0    NaN  \n",
       "1                                  NaN      NaN     1.0    NaN  \n",
       "2                                  NaN      NaN     1.0    NaN  \n",
       "3                                  NaN      NaN     1.0    NaN  \n",
       "4                                  NaN      NaN     1.0    NaN  \n",
       "...                                ...      ...     ...    ...  \n",
       "316574                            Q117  caA/caG    -1.0    NaN  \n",
       "316575                            Q117  caA/caG    -1.0    NaN  \n",
       "316576                             NaN      NaN    -1.0    NaN  \n",
       "316577                             NaN      NaN    -1.0    NaN  \n",
       "316578                             NaN      NaN    -1.0    NaN  \n",
       "\n",
       "[316579 rows x 38 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL4_processed_vcf.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "530d0546",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "1           5 prime UTR variant  MODIFIER           NaN      NaN   \n",
       "2       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "3       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "4                intron variant  MODIFIER           NaN      NaN   \n",
       "...                         ...       ...           ...      ...   \n",
       "316574       synonymous variant       LOW           NaN      NaN   \n",
       "316575       synonymous variant       LOW           NaN      NaN   \n",
       "316576           intron variant  MODIFIER           NaN      NaN   \n",
       "316577           intron variant  MODIFIER           NaN      NaN   \n",
       "316578           intron variant  MODIFIER           NaN      NaN   \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                     NaN      NaN                NaN               None   \n",
       "1                     NaN      NaN  ENST00000641515.2           c.-36C>T   \n",
       "2                     NaN      NaN                NaN               None   \n",
       "3                     NaN      NaN                NaN               None   \n",
       "4                     NaN      NaN  ENST00000641515.2          c.9+18C>T   \n",
       "...                   ...      ...                ...                ...   \n",
       "316574                NaN      NaN  ENST00000426000.6           c.351A>G   \n",
       "316575                NaN      NaN  ENST00000540248.5           c.351A>G   \n",
       "316576                NaN      NaN  ENST00000405239.6        c.243-46G>A   \n",
       "316577                NaN      NaN  ENST00000426000.6        c.243-46G>A   \n",
       "316578                NaN      NaN  ENST00000540248.5        c.243-46G>A   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD  \\\n",
       "0                     NaN               None  0/1  23   26  26  19   7   \n",
       "1                     NaN               None  0/1  23   26  26  19   7   \n",
       "2                     NaN               None  0/1  23   26  26  19   7   \n",
       "3                     NaN               None  0/1  22   28  28  21   7   \n",
       "4                     NaN               None  0/1  22   28  28  21   7   \n",
       "...                   ...                ...  ...  ..  ...  ..  ..  ..   \n",
       "316574  ENSP00000400716.3        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316575  ENSP00000444407.1        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316576                NaN               None  0/1  26   33  33  25   8   \n",
       "316577                NaN               None  0/1  26   33  33  25   8   \n",
       "316578                NaN               None  0/1  26   33  33  25   8   \n",
       "\n",
       "          FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...        ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED  \\\n",
       "0                                  NaN      NaN     1.0    NaN   \n",
       "1                                  NaN      NaN     1.0    NaN   \n",
       "2                                  NaN      NaN     1.0    NaN   \n",
       "3                                  NaN      NaN     1.0    NaN   \n",
       "4                                  NaN      NaN     1.0    NaN   \n",
       "...                                ...      ...     ...    ...   \n",
       "316574                            Q117  caA/caG    -1.0    NaN   \n",
       "316575                            Q117  caA/caG    -1.0    NaN   \n",
       "316576                             NaN      NaN    -1.0    NaN   \n",
       "316577                             NaN      NaN    -1.0    NaN   \n",
       "316578                             NaN      NaN    -1.0    NaN   \n",
       "\n",
       "                    consequence  \n",
       "0       downstream gene variant  \n",
       "1           5 prime UTR variant  \n",
       "2       downstream gene variant  \n",
       "3       downstream gene variant  \n",
       "4                intron variant  \n",
       "...                         ...  \n",
       "316574       synonymous variant  \n",
       "316575       synonymous variant  \n",
       "316576           intron variant  \n",
       "316577           intron variant  \n",
       "316578           intron variant  \n",
       "\n",
       "[316579 rows x 39 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['consequence'] = df['Consequence'].str.split(',').str[0]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1a0e5a6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "78e4c9c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "1           5 prime UTR variant  MODIFIER           NaN      NaN   \n",
       "2       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "3       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "4                intron variant  MODIFIER           NaN      NaN   \n",
       "...                         ...       ...           ...      ...   \n",
       "316574       synonymous variant       LOW           NaN      NaN   \n",
       "316575       synonymous variant       LOW           NaN      NaN   \n",
       "316576           intron variant  MODIFIER           NaN      NaN   \n",
       "316577           intron variant  MODIFIER           NaN      NaN   \n",
       "316578           intron variant  MODIFIER           NaN      NaN   \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                     NaN      NaN                NaN               None   \n",
       "1                     NaN      NaN  ENST00000641515.2           c.-36C>T   \n",
       "2                     NaN      NaN                NaN               None   \n",
       "3                     NaN      NaN                NaN               None   \n",
       "4                     NaN      NaN  ENST00000641515.2          c.9+18C>T   \n",
       "...                   ...      ...                ...                ...   \n",
       "316574                NaN      NaN  ENST00000426000.6           c.351A>G   \n",
       "316575                NaN      NaN  ENST00000540248.5           c.351A>G   \n",
       "316576                NaN      NaN  ENST00000405239.6        c.243-46G>A   \n",
       "316577                NaN      NaN  ENST00000426000.6        c.243-46G>A   \n",
       "316578                NaN      NaN  ENST00000540248.5        c.243-46G>A   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD  \\\n",
       "0                     NaN               None  0/1  23   26  26  19   7   \n",
       "1                     NaN               None  0/1  23   26  26  19   7   \n",
       "2                     NaN               None  0/1  23   26  26  19   7   \n",
       "3                     NaN               None  0/1  22   28  28  21   7   \n",
       "4                     NaN               None  0/1  22   28  28  21   7   \n",
       "...                   ...                ...  ...  ..  ...  ..  ..  ..   \n",
       "316574  ENSP00000400716.3        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316575  ENSP00000444407.1        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316576                NaN               None  0/1  26   33  33  25   8   \n",
       "316577                NaN               None  0/1  26   33  33  25   8   \n",
       "316578                NaN               None  0/1  26   33  33  25   8   \n",
       "\n",
       "          FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...        ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED  \\\n",
       "0                                  NaN      NaN     1.0    NaN   \n",
       "1                                  NaN      NaN     1.0    NaN   \n",
       "2                                  NaN      NaN     1.0    NaN   \n",
       "3                                  NaN      NaN     1.0    NaN   \n",
       "4                                  NaN      NaN     1.0    NaN   \n",
       "...                                ...      ...     ...    ...   \n",
       "316574                            Q117  caA/caG    -1.0    NaN   \n",
       "316575                            Q117  caA/caG    -1.0    NaN   \n",
       "316576                             NaN      NaN    -1.0    NaN   \n",
       "316577                             NaN      NaN    -1.0    NaN   \n",
       "316578                             NaN      NaN    -1.0    NaN   \n",
       "\n",
       "                    consequence Consequence_score  \n",
       "0       downstream gene variant              2/10  \n",
       "1           5 prime UTR variant              3/10  \n",
       "2       downstream gene variant              2/10  \n",
       "3       downstream gene variant              2/10  \n",
       "4                intron variant              2/10  \n",
       "...                         ...               ...  \n",
       "316574       synonymous variant              3/10  \n",
       "316575       synonymous variant              3/10  \n",
       "316576           intron variant              2/10  \n",
       "316577           intron variant              2/10  \n",
       "316578           intron variant              2/10  \n",
       "\n",
       "[316579 rows x 40 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(df, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "10fb1af3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000616016.5</td>\n",
       "      <td>c.102A&gt;G</td>\n",
       "      <td>ENSP00000478421.2</td>\n",
       "      <td>p.Pro34%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P34</td>\n",
       "      <td>ccA/ccG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000618323.5</td>\n",
       "      <td>c.102A&gt;G</td>\n",
       "      <td>ENSP00000480678.2</td>\n",
       "      <td>p.Pro34%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P34</td>\n",
       "      <td>ccA/ccG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>200</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>100%</td>\n",
       "      <td>8.913700e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene Name           ID CHROM     POS REF ALT    Zygosity  \\\n",
       "16  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "17  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "18  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "19  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "20  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "21  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "22  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "23  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "24  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "25  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "26  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "27  LOC107985728,SAMD11  rs112703963  chr1  924533   A   G  Homozygous   \n",
       "\n",
       "                  Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "16      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "17      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "18      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "19      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "20         synonymous variant       LOW           NaN      NaN   \n",
       "21      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "22      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "23      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "24         synonymous variant       LOW           NaN      NaN   \n",
       "25      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "26      upstream gene variant  MODIFIER           NaN      NaN   \n",
       "27  regulatory region variant  MODIFIER           NaN      NaN   \n",
       "\n",
       "   ClinVar_CLNREVSTAT  ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "16                NaN      NaN                NaN               None   \n",
       "17                NaN      NaN                NaN               None   \n",
       "18                NaN      NaN                NaN               None   \n",
       "19                NaN      NaN                NaN               None   \n",
       "20                NaN      NaN  ENST00000616016.5           c.102A>G   \n",
       "21                NaN      NaN                NaN               None   \n",
       "22                NaN      NaN                NaN               None   \n",
       "23                NaN      NaN                NaN               None   \n",
       "24                NaN      NaN  ENST00000618323.5           c.102A>G   \n",
       "25                NaN      NaN                NaN               None   \n",
       "26                NaN      NaN                NaN               None   \n",
       "27                NaN      NaN                NaN               None   \n",
       "\n",
       "                HGVSp HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD  FREQ  \\\n",
       "16                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "17                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "18                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "19                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "20  ENSP00000478421.2         p.Pro34%3D  1/1  200   35  35   0  35  100%   \n",
       "21                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "22                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "23                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "24  ENSP00000480678.2         p.Pro34%3D  1/1  200   35  35   0  35  100%   \n",
       "25                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "26                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "27                NaN               None  1/1  200   35  35   0  35  100%   \n",
       "\n",
       "            PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen          BIOTYPE  EXON  \\\n",
       "16  8.913700e-21    0    0   22   13  NaN      NaN   protein coding   NaN   \n",
       "17  8.913700e-21    0    0   22   13  NaN      NaN           lncRNA   NaN   \n",
       "18  8.913700e-21    0    0   22   13  NaN      NaN   protein coding   NaN   \n",
       "19  8.913700e-21    0    0   22   13  NaN      NaN  retained intron   NaN   \n",
       "20  8.913700e-21    0    0   22   13  NaN      NaN   protein coding  1/14   \n",
       "21  8.913700e-21    0    0   22   13  NaN      NaN   protein coding   NaN   \n",
       "22  8.913700e-21    0    0   22   13  NaN      NaN   protein coding   NaN   \n",
       "23  8.913700e-21    0    0   22   13  NaN      NaN   protein coding   NaN   \n",
       "24  8.913700e-21    0    0   22   13  NaN      NaN   protein coding  1/14   \n",
       "25  8.913700e-21    0    0   22   13  NaN      NaN   protein coding   NaN   \n",
       "26  8.913700e-21    0    0   22   13  NaN      NaN   protein coding   NaN   \n",
       "27  8.913700e-21    0    0   22   13  NaN      NaN         promoter   NaN   \n",
       "\n",
       "   INTRON Protein Position and Amino Acid   Codons  STRAND PUBMED  \\\n",
       "16    NaN                             NaN      NaN     1.0    NaN   \n",
       "17    NaN                             NaN      NaN    -1.0    NaN   \n",
       "18    NaN                             NaN      NaN     1.0    NaN   \n",
       "19    NaN                             NaN      NaN    -1.0    NaN   \n",
       "20    NaN                             P34  ccA/ccG     1.0    NaN   \n",
       "21    NaN                             NaN      NaN     1.0    NaN   \n",
       "22    NaN                             NaN      NaN     1.0    NaN   \n",
       "23    NaN                             NaN      NaN     1.0    NaN   \n",
       "24    NaN                             P34  ccA/ccG     1.0    NaN   \n",
       "25    NaN                             NaN      NaN     1.0    NaN   \n",
       "26    NaN                             NaN      NaN     1.0    NaN   \n",
       "27    NaN                             NaN      NaN     NaN    NaN   \n",
       "\n",
       "                  consequence Consequence_score  \n",
       "16      upstream gene variant              2/10  \n",
       "17      upstream gene variant              2/10  \n",
       "18      upstream gene variant              2/10  \n",
       "19      upstream gene variant              2/10  \n",
       "20         synonymous variant              3/10  \n",
       "21      upstream gene variant              2/10  \n",
       "22      upstream gene variant              2/10  \n",
       "23      upstream gene variant              2/10  \n",
       "24         synonymous variant              3/10  \n",
       "25      upstream gene variant              2/10  \n",
       "26      upstream gene variant              2/10  \n",
       "27  regulatory region variant              2/10  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1[merged_1['ID'] == 'rs112703963']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "06a464e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     IMPACT  IMPACT_score\n",
       "0      HIGH          10.0\n",
       "1  MODERATE           5.0\n",
       "2       LOW           2.5\n",
       "3  MODIFIER           1.5"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "4ace5182",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "1           5 prime UTR variant  MODIFIER           NaN      NaN   \n",
       "2       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "3       downstream gene variant  MODIFIER           NaN      NaN   \n",
       "4                intron variant  MODIFIER           NaN      NaN   \n",
       "...                         ...       ...           ...      ...   \n",
       "316574       synonymous variant       LOW           NaN      NaN   \n",
       "316575       synonymous variant       LOW           NaN      NaN   \n",
       "316576           intron variant  MODIFIER           NaN      NaN   \n",
       "316577           intron variant  MODIFIER           NaN      NaN   \n",
       "316578           intron variant  MODIFIER           NaN      NaN   \n",
       "\n",
       "       ClinVar_CLNREVSTAT  ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                     NaN      NaN                NaN               None   \n",
       "1                     NaN      NaN  ENST00000641515.2           c.-36C>T   \n",
       "2                     NaN      NaN                NaN               None   \n",
       "3                     NaN      NaN                NaN               None   \n",
       "4                     NaN      NaN  ENST00000641515.2          c.9+18C>T   \n",
       "...                   ...      ...                ...                ...   \n",
       "316574                NaN      NaN  ENST00000426000.6           c.351A>G   \n",
       "316575                NaN      NaN  ENST00000540248.5           c.351A>G   \n",
       "316576                NaN      NaN  ENST00000405239.6        c.243-46G>A   \n",
       "316577                NaN      NaN  ENST00000426000.6        c.243-46G>A   \n",
       "316578                NaN      NaN  ENST00000540248.5        c.243-46G>A   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD  \\\n",
       "0                     NaN               None  0/1  23   26  26  19   7   \n",
       "1                     NaN               None  0/1  23   26  26  19   7   \n",
       "2                     NaN               None  0/1  23   26  26  19   7   \n",
       "3                     NaN               None  0/1  22   28  28  21   7   \n",
       "4                     NaN               None  0/1  22   28  28  21   7   \n",
       "...                   ...                ...  ...  ..  ...  ..  ..  ..   \n",
       "316574  ENSP00000400716.3        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316575  ENSP00000444407.1        p.Gln117%3D  0/1  48   73  73  58  15   \n",
       "316576                NaN               None  0/1  26   33  33  25   8   \n",
       "316577                NaN               None  0/1  26   33  33  25   8   \n",
       "316578                NaN               None  0/1  26   33  33  25   8   \n",
       "\n",
       "          FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4          25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...        ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED  \\\n",
       "0                                  NaN      NaN     1.0    NaN   \n",
       "1                                  NaN      NaN     1.0    NaN   \n",
       "2                                  NaN      NaN     1.0    NaN   \n",
       "3                                  NaN      NaN     1.0    NaN   \n",
       "4                                  NaN      NaN     1.0    NaN   \n",
       "...                                ...      ...     ...    ...   \n",
       "316574                            Q117  caA/caG    -1.0    NaN   \n",
       "316575                            Q117  caA/caG    -1.0    NaN   \n",
       "316576                             NaN      NaN    -1.0    NaN   \n",
       "316577                             NaN      NaN    -1.0    NaN   \n",
       "316578                             NaN      NaN    -1.0    NaN   \n",
       "\n",
       "                    consequence Consequence_score  IMPACT_score  \n",
       "0       downstream gene variant              2/10           1.5  \n",
       "1           5 prime UTR variant              3/10           1.5  \n",
       "2       downstream gene variant              2/10           1.5  \n",
       "3       downstream gene variant              2/10           1.5  \n",
       "4                intron variant              2/10           1.5  \n",
       "...                         ...               ...           ...  \n",
       "316574       synonymous variant              3/10           2.5  \n",
       "316575       synonymous variant              3/10           2.5  \n",
       "316576           intron variant              2/10           1.5  \n",
       "316577           intron variant              2/10           1.5  \n",
       "316578           intron variant              2/10           1.5  \n",
       "\n",
       "[316579 rows x 41 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "d703bbe8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1           5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "4                intron variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "316574       synonymous variant              3/10       LOW           2.5   \n",
       "316575       synonymous variant              3/10       LOW           2.5   \n",
       "316576           intron variant              2/10  MODIFIER           1.5   \n",
       "316577           intron variant              2/10  MODIFIER           1.5   \n",
       "316578           intron variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "0                NaN      NaN                NaN      NaN                NaN   \n",
       "1                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "2                NaN      NaN                NaN      NaN                NaN   \n",
       "3                NaN      NaN                NaN      NaN                NaN   \n",
       "4                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "...              ...      ...                ...      ...                ...   \n",
       "316574           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316575           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "316576           NaN      NaN                NaN      NaN  ENST00000405239.6   \n",
       "316577           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316578           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                NaN               None  0/1  23   26   \n",
       "1                c.-36C>T                NaN               None  0/1  23   26   \n",
       "2                    None                NaN               None  0/1  23   26   \n",
       "3                    None                NaN               None  0/1  22   28   \n",
       "4               c.9+18C>T                NaN               None  0/1  22   28   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "316574           c.351A>G  ENSP00000400716.3        p.Gln117%3D  0/1  48   73   \n",
       "316575           c.351A>G  ENSP00000444407.1        p.Gln117%3D  0/1  48   73   \n",
       "316576        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316577        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316578        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "\n",
       "        DP  RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...     ..  ..  ..     ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED  \n",
       "0                                  NaN      NaN     1.0    NaN  \n",
       "1                                  NaN      NaN     1.0    NaN  \n",
       "2                                  NaN      NaN     1.0    NaN  \n",
       "3                                  NaN      NaN     1.0    NaN  \n",
       "4                                  NaN      NaN     1.0    NaN  \n",
       "...                                ...      ...     ...    ...  \n",
       "316574                            Q117  caA/caG    -1.0    NaN  \n",
       "316575                            Q117  caA/caG    -1.0    NaN  \n",
       "316576                             NaN      NaN    -1.0    NaN  \n",
       "316577                             NaN      NaN    -1.0    NaN  \n",
       "316578                             NaN      NaN    -1.0    NaN  \n",
       "\n",
       "[316579 rows x 40 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'ID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "d35473a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".             2424\n",
       "rs12624152     145\n",
       "rs848291       144\n",
       "rs2589516      123\n",
       "rs183956       112\n",
       "              ... \n",
       "rs1197683        1\n",
       "rs2275443        1\n",
       "rs61746215       1\n",
       "rs12411843       1\n",
       "rs2900373        1\n",
       "Name: ID, Length: 35804, dtype: int64"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.ID.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "a3cdf63f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>182712</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000233741.9</td>\n",
       "      <td>c.692-27A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182713</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182714</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000402135.8</td>\n",
       "      <td>c.707-27A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182715</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000403295.8</td>\n",
       "      <td>c.692-470A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182716</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000403676.6</td>\n",
       "      <td>c.*255-27A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182852</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000696635.1</td>\n",
       "      <td>c.692-51A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182853</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000696636.1</td>\n",
       "      <td>n.1477-27A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182854</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000696637.1</td>\n",
       "      <td>n.7298A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>3/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182855</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000696638.1</td>\n",
       "      <td>n.707-27A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182856</th>\n",
       "      <td>VRK2,FANCL</td>\n",
       "      <td>rs12624152</td>\n",
       "      <td>chr2</td>\n",
       "      <td>58163544</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1277221.0</td>\n",
       "      <td>ENST00000696639.1</td>\n",
       "      <td>n.2303A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>61.9%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>retained intron</td>\n",
       "      <td>1/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>145 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene Name          ID CHROM       POS REF ALT      Zygosity  \\\n",
       "182712  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182713  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182714  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182715  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182716  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "...            ...         ...   ...       ...  ..  ..           ...   \n",
       "182852  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182853  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182854  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182855  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "182856  VRK2,FANCL  rs12624152  chr2  58163544   T   G  Heterozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "182712                                intron variant              2/10   \n",
       "182713                       downstream gene variant              2/10   \n",
       "182714                                intron variant              2/10   \n",
       "182715                                intron variant              2/10   \n",
       "182716         intron variant,NMD transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "182852                                intron variant              2/10   \n",
       "182853  intron variant,non coding transcript variant              2/10   \n",
       "182854            non coding transcript exon variant              2/10   \n",
       "182855  intron variant,non coding transcript variant              2/10   \n",
       "182856            non coding transcript exon variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "182712  MODIFIER           1.5  not provided   benign   \n",
       "182713  MODIFIER           1.5  not provided   benign   \n",
       "182714  MODIFIER           1.5  not provided   benign   \n",
       "182715  MODIFIER           1.5  not provided   benign   \n",
       "182716  MODIFIER           1.5  not provided   benign   \n",
       "...          ...           ...           ...      ...   \n",
       "182852  MODIFIER           1.5  not provided   benign   \n",
       "182853  MODIFIER           1.5  not provided   benign   \n",
       "182854  MODIFIER           1.5  not provided   benign   \n",
       "182855  MODIFIER           1.5  not provided   benign   \n",
       "182856  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                         ClinVar_CLNREVSTAT    ClinVar              HGVSc  \\\n",
       "182712  criteria provided, single submitter  1277221.0  ENST00000233741.9   \n",
       "182713  criteria provided, single submitter  1277221.0                NaN   \n",
       "182714  criteria provided, single submitter  1277221.0  ENST00000402135.8   \n",
       "182715  criteria provided, single submitter  1277221.0  ENST00000403295.8   \n",
       "182716  criteria provided, single submitter  1277221.0  ENST00000403676.6   \n",
       "...                                     ...        ...                ...   \n",
       "182852  criteria provided, single submitter  1277221.0  ENST00000696635.1   \n",
       "182853  criteria provided, single submitter  1277221.0  ENST00000696636.1   \n",
       "182854  criteria provided, single submitter  1277221.0  ENST00000696637.1   \n",
       "182855  criteria provided, single submitter  1277221.0  ENST00000696638.1   \n",
       "182856  criteria provided, single submitter  1277221.0  ENST00000696639.1   \n",
       "\n",
       "       HGVSc (Transcript) HGVSp HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD  \\\n",
       "182712        c.692-27A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "182713               None   NaN               None  0/1  50   21  21   8  13   \n",
       "182714        c.707-27A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "182715       c.692-470A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "182716       c.*255-27A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "...                   ...   ...                ...  ...  ..  ...  ..  ..  ..   \n",
       "182852        c.692-51A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "182853       n.1477-27A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "182854          n.7298A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "182855        n.707-27A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "182856          n.2303A>C   NaN               None  0/1  50   21  21   8  13   \n",
       "\n",
       "         FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "182712  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182713  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182714  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182715  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182716  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "...       ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "182852  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182853  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182854  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182855  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "182856  61.9%  0.000008    4    4    9    4  NaN      NaN   \n",
       "\n",
       "                        BIOTYPE EXON INTRON Protein Position and Amino Acid  \\\n",
       "182712           protein coding  NaN   8/13                             NaN   \n",
       "182713           protein coding  NaN    NaN                             NaN   \n",
       "182714           protein coding  NaN   8/13                             NaN   \n",
       "182715           protein coding  NaN   8/12                             NaN   \n",
       "182716  nonsense mediated decay  NaN    4/9                             NaN   \n",
       "...                         ...  ...    ...                             ...   \n",
       "182852           protein coding  NaN   8/13                             NaN   \n",
       "182853          retained intron  NaN    1/2                             NaN   \n",
       "182854          retained intron  3/6    NaN                             NaN   \n",
       "182855          retained intron  NaN    4/9                             NaN   \n",
       "182856          retained intron  1/3    NaN                             NaN   \n",
       "\n",
       "       Codons  STRAND PUBMED  \n",
       "182712    NaN    -1.0    NaN  \n",
       "182713    NaN     1.0    NaN  \n",
       "182714    NaN    -1.0    NaN  \n",
       "182715    NaN    -1.0    NaN  \n",
       "182716    NaN    -1.0    NaN  \n",
       "...       ...     ...    ...  \n",
       "182852    NaN    -1.0    NaN  \n",
       "182853    NaN    -1.0    NaN  \n",
       "182854    NaN    -1.0    NaN  \n",
       "182855    NaN    -1.0    NaN  \n",
       "182856    NaN    -1.0    NaN  \n",
       "\n",
       "[145 rows x 40 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2[merged_2['ID'] == 'rs12624152']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7afe991",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc00e307",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "68dcf8e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_2.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL4_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1800ce3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39eb2c1b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f701cac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "752637e4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fed19e0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "ca82732e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1           5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "4                intron variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "316574       synonymous variant              3/10       LOW           2.5   \n",
       "316575       synonymous variant              3/10       LOW           2.5   \n",
       "316576           intron variant              2/10  MODIFIER           1.5   \n",
       "316577           intron variant              2/10  MODIFIER           1.5   \n",
       "316578           intron variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "0                NaN      NaN                NaN      NaN                NaN   \n",
       "1                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "2                NaN      NaN                NaN      NaN                NaN   \n",
       "3                NaN      NaN                NaN      NaN                NaN   \n",
       "4                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "...              ...      ...                ...      ...                ...   \n",
       "316574           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316575           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "316576           NaN      NaN                NaN      NaN  ENST00000405239.6   \n",
       "316577           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316578           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                NaN               None  0/1  23   26   \n",
       "1                c.-36C>T                NaN               None  0/1  23   26   \n",
       "2                    None                NaN               None  0/1  23   26   \n",
       "3                    None                NaN               None  0/1  22   28   \n",
       "4               c.9+18C>T                NaN               None  0/1  22   28   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "316574           c.351A>G  ENSP00000400716.3        p.Gln117%3D  0/1  48   73   \n",
       "316575           c.351A>G  ENSP00000444407.1        p.Gln117%3D  0/1  48   73   \n",
       "316576        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316577        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316578        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "\n",
       "        DP  RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...     ..  ..  ..     ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED  \n",
       "0                                  NaN      NaN     1.0    NaN  \n",
       "1                                  NaN      NaN     1.0    NaN  \n",
       "2                                  NaN      NaN     1.0    NaN  \n",
       "3                                  NaN      NaN     1.0    NaN  \n",
       "4                                  NaN      NaN     1.0    NaN  \n",
       "...                                ...      ...     ...    ...  \n",
       "316574                            Q117  caA/caG    -1.0    NaN  \n",
       "316575                            Q117  caA/caG    -1.0    NaN  \n",
       "316576                             NaN      NaN    -1.0    NaN  \n",
       "316577                             NaN      NaN    -1.0    NaN  \n",
       "316578                             NaN      NaN    -1.0    NaN  \n",
       "\n",
       "[316579 rows x 40 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL4_depth_vcf_processed.xlsx')\n",
    "df['POS'] = df['POS'].astype('int64')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4dd01d4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>104858586</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>104826974</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>104824472</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>104858554</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>104794495</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1861</th>\n",
       "      <td>31592980</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1862</th>\n",
       "      <td>31595223</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1863</th>\n",
       "      <td>31593004</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1864</th>\n",
       "      <td>31598691</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1865</th>\n",
       "      <td>31595256</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1866 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            POS Literature\n",
       "0     104858586        Yes\n",
       "1     104826974        Yes\n",
       "2     104824472        Yes\n",
       "3     104858554        Yes\n",
       "4     104794495        Yes\n",
       "...         ...        ...\n",
       "1861   31592980        Yes\n",
       "1862   31595223        Yes\n",
       "1863   31593004        Yes\n",
       "1864   31598691        Yes\n",
       "1865   31595256        Yes\n",
       "\n",
       "[1866 rows x 2 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/Cardiac_pos.xlsx')\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "e59484b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1           5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "4                intron variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "316574       synonymous variant              3/10       LOW           2.5   \n",
       "316575       synonymous variant              3/10       LOW           2.5   \n",
       "316576           intron variant              2/10  MODIFIER           1.5   \n",
       "316577           intron variant              2/10  MODIFIER           1.5   \n",
       "316578           intron variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "0                NaN      NaN                NaN      NaN                NaN   \n",
       "1                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "2                NaN      NaN                NaN      NaN                NaN   \n",
       "3                NaN      NaN                NaN      NaN                NaN   \n",
       "4                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "...              ...      ...                ...      ...                ...   \n",
       "316574           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316575           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "316576           NaN      NaN                NaN      NaN  ENST00000405239.6   \n",
       "316577           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316578           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                NaN               None  0/1  23   26   \n",
       "1                c.-36C>T                NaN               None  0/1  23   26   \n",
       "2                    None                NaN               None  0/1  23   26   \n",
       "3                    None                NaN               None  0/1  22   28   \n",
       "4               c.9+18C>T                NaN               None  0/1  22   28   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "316574           c.351A>G  ENSP00000400716.3        p.Gln117%3D  0/1  48   73   \n",
       "316575           c.351A>G  ENSP00000444407.1        p.Gln117%3D  0/1  48   73   \n",
       "316576        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316577        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316578        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "\n",
       "        DP  RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...     ..  ..  ..     ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED Literature  \n",
       "0                                  NaN      NaN     1.0    NaN        NaN  \n",
       "1                                  NaN      NaN     1.0    NaN        NaN  \n",
       "2                                  NaN      NaN     1.0    NaN        NaN  \n",
       "3                                  NaN      NaN     1.0    NaN        NaN  \n",
       "4                                  NaN      NaN     1.0    NaN        NaN  \n",
       "...                                ...      ...     ...    ...        ...  \n",
       "316574                            Q117  caA/caG    -1.0    NaN        NaN  \n",
       "316575                            Q117  caA/caG    -1.0    NaN        NaN  \n",
       "316576                             NaN      NaN    -1.0    NaN        NaN  \n",
       "316577                             NaN      NaN    -1.0    NaN        NaN  \n",
       "316578                             NaN      NaN    -1.0    NaN        NaN  \n",
       "\n",
       "[316579 rows x 41 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(df, df_3, on = 'POS', how = 'left', sort=False)\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "af1b48e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "1            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "2            NaN  rs1473003496  chr1     65529   C   T  Heterozygous   \n",
       "3            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "4            NaN  rs1416222198  chr1     65591   C   T  Heterozygous   \n",
       "...          ...           ...   ...       ...  ..  ..           ...   \n",
       "316574       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316575       NaN             .  chrY  23190581   T   C  Heterozygous   \n",
       "316576       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316577       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "316578       NaN             .  chrY  23191171   C   T  Heterozygous   \n",
       "\n",
       "                    Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "1           5 prime UTR variant              3/10  MODIFIER           1.5   \n",
       "2       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "3       downstream gene variant              2/10  MODIFIER           1.5   \n",
       "4                intron variant              2/10  MODIFIER           1.5   \n",
       "...                         ...               ...       ...           ...   \n",
       "316574       synonymous variant              3/10       LOW           2.5   \n",
       "316575       synonymous variant              3/10       LOW           2.5   \n",
       "316576           intron variant              2/10  MODIFIER           1.5   \n",
       "316577           intron variant              2/10  MODIFIER           1.5   \n",
       "316578           intron variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "0                NaN      NaN                NaN      NaN                NaN   \n",
       "1                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "2                NaN      NaN                NaN      NaN                NaN   \n",
       "3                NaN      NaN                NaN      NaN                NaN   \n",
       "4                NaN      NaN                NaN      NaN  ENST00000641515.2   \n",
       "...              ...      ...                ...      ...                ...   \n",
       "316574           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316575           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "316576           NaN      NaN                NaN      NaN  ENST00000405239.6   \n",
       "316577           NaN      NaN                NaN      NaN  ENST00000426000.6   \n",
       "316578           NaN      NaN                NaN      NaN  ENST00000540248.5   \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ  SDP  \\\n",
       "0                    None                NaN               None  0/1  23   26   \n",
       "1                c.-36C>T                NaN               None  0/1  23   26   \n",
       "2                    None                NaN               None  0/1  23   26   \n",
       "3                    None                NaN               None  0/1  22   28   \n",
       "4               c.9+18C>T                NaN               None  0/1  22   28   \n",
       "...                   ...                ...                ...  ...  ..  ...   \n",
       "316574           c.351A>G  ENSP00000400716.3        p.Gln117%3D  0/1  48   73   \n",
       "316575           c.351A>G  ENSP00000444407.1        p.Gln117%3D  0/1  48   73   \n",
       "316576        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316577        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "316578        c.243-46G>A                NaN               None  0/1  26   33   \n",
       "\n",
       "        DP  RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR SIFT PolyPhen  \\\n",
       "0       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "1       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "2       26  19   7  26.92%  0.004917   18    1    5    2  NaN      NaN   \n",
       "3       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "4       28  21   7     25%  0.005105   17    4    6    1  NaN      NaN   \n",
       "...     ..  ..  ..     ...       ...  ...  ...  ...  ...  ...      ...   \n",
       "316574  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316575  73  58  15  20.55%  0.000014   38   20    9    6  NaN      NaN   \n",
       "316576  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316577  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "316578  33  25   8  24.24%  0.002417   17    8    7    1  NaN      NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND PUBMED Literature  \n",
       "0                                  NaN      NaN     1.0    NaN         No  \n",
       "1                                  NaN      NaN     1.0    NaN         No  \n",
       "2                                  NaN      NaN     1.0    NaN         No  \n",
       "3                                  NaN      NaN     1.0    NaN         No  \n",
       "4                                  NaN      NaN     1.0    NaN         No  \n",
       "...                                ...      ...     ...    ...        ...  \n",
       "316574                            Q117  caA/caG    -1.0    NaN         No  \n",
       "316575                            Q117  caA/caG    -1.0    NaN         No  \n",
       "316576                             NaN      NaN    -1.0    NaN         No  \n",
       "316577                             NaN      NaN    -1.0    NaN         No  \n",
       "316578                             NaN      NaN    -1.0    NaN         No  \n",
       "\n",
       "[316579 rows x 41 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "565f28c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     316485\n",
       "Yes        94\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b6ad0171",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "rs1071646     39\n",
       "rs1805127      8\n",
       "rs1799983      5\n",
       "rs5882         4\n",
       "rs1800777      4\n",
       "rs6771157      4\n",
       "rs59468016     4\n",
       "rs7374804      4\n",
       "rs2230806      4\n",
       "rs7521023      3\n",
       "rs6599242      3\n",
       "rs78425180     3\n",
       "rs7630989      3\n",
       "rs7617919      3\n",
       "rs74717885     3\n",
       "Name: ID, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3[merged_3['Literature'] == 'Yes'].ID.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "4a6cb2a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.-36C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>26.92%</td>\n",
       "      <td>0.004917</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.9+18C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.005105</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000400716.3</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.351A&gt;G</td>\n",
       "      <td>ENSP00000444407.1</td>\n",
       "      <td>p.Gln117%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>73</td>\n",
       "      <td>73</td>\n",
       "      <td>58</td>\n",
       "      <td>15</td>\n",
       "      <td>20.55%</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000405239.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000426000.6</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000540248.5</td>\n",
       "      <td>c.243-46G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>24.24%</td>\n",
       "      <td>0.002417</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS Literature REF ALT  \\\n",
       "0            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "1            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "2            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "3            NaN  rs1416222198  chr1     65591         No   C   T   \n",
       "4            NaN  rs1416222198  chr1     65591         No   C   T   \n",
       "...          ...           ...   ...       ...        ...  ..  ..   \n",
       "316574       NaN             .  chrY  23190581         No   T   C   \n",
       "316575       NaN             .  chrY  23190581         No   T   C   \n",
       "316576       NaN             .  chrY  23191171         No   C   T   \n",
       "316577       NaN             .  chrY  23191171         No   C   T   \n",
       "316578       NaN             .  chrY  23191171         No   C   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "1       Heterozygous      5 prime UTR variant              3/10  MODIFIER   \n",
       "2       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "3       Heterozygous  downstream gene variant              2/10  MODIFIER   \n",
       "4       Heterozygous           intron variant              2/10  MODIFIER   \n",
       "...              ...                      ...               ...       ...   \n",
       "316574  Heterozygous       synonymous variant              3/10       LOW   \n",
       "316575  Heterozygous       synonymous variant              3/10       LOW   \n",
       "316576  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "316577  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "316578  Heterozygous           intron variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  ClinVar  \\\n",
       "0                1.5           NaN      NaN                NaN      NaN   \n",
       "1                1.5           NaN      NaN                NaN      NaN   \n",
       "2                1.5           NaN      NaN                NaN      NaN   \n",
       "3                1.5           NaN      NaN                NaN      NaN   \n",
       "4                1.5           NaN      NaN                NaN      NaN   \n",
       "...              ...           ...      ...                ...      ...   \n",
       "316574           2.5           NaN      NaN                NaN      NaN   \n",
       "316575           2.5           NaN      NaN                NaN      NaN   \n",
       "316576           1.5           NaN      NaN                NaN      NaN   \n",
       "316577           1.5           NaN      NaN                NaN      NaN   \n",
       "316578           1.5           NaN      NaN                NaN      NaN   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0                     NaN               None                NaN   \n",
       "1       ENST00000641515.2           c.-36C>T                NaN   \n",
       "2                     NaN               None                NaN   \n",
       "3                     NaN               None                NaN   \n",
       "4       ENST00000641515.2          c.9+18C>T                NaN   \n",
       "...                   ...                ...                ...   \n",
       "316574  ENST00000426000.6           c.351A>G  ENSP00000400716.3   \n",
       "316575  ENST00000540248.5           c.351A>G  ENSP00000444407.1   \n",
       "316576  ENST00000405239.6        c.243-46G>A                NaN   \n",
       "316577  ENST00000426000.6        c.243-46G>A                NaN   \n",
       "316578  ENST00000540248.5        c.243-46G>A                NaN   \n",
       "\n",
       "       HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD    FREQ      PVAL  RDF  \\\n",
       "0                    None  0/1  23   26  26  19   7  26.92%  0.004917   18   \n",
       "1                    None  0/1  23   26  26  19   7  26.92%  0.004917   18   \n",
       "2                    None  0/1  23   26  26  19   7  26.92%  0.004917   18   \n",
       "3                    None  0/1  22   28  28  21   7     25%  0.005105   17   \n",
       "4                    None  0/1  22   28  28  21   7     25%  0.005105   17   \n",
       "...                   ...  ...  ..  ...  ..  ..  ..     ...       ...  ...   \n",
       "316574        p.Gln117%3D  0/1  48   73  73  58  15  20.55%  0.000014   38   \n",
       "316575        p.Gln117%3D  0/1  48   73  73  58  15  20.55%  0.000014   38   \n",
       "316576               None  0/1  26   33  33  25   8  24.24%  0.002417   17   \n",
       "316577               None  0/1  26   33  33  25   8  24.24%  0.002417   17   \n",
       "316578               None  0/1  26   33  33  25   8  24.24%  0.002417   17   \n",
       "\n",
       "        RDR  ADF  ADR SIFT PolyPhen                             BIOTYPE  EXON  \\\n",
       "0         1    5    2  NaN      NaN  transcribed unprocessed pseudogene   NaN   \n",
       "1         1    5    2  NaN      NaN                      protein coding   2/3   \n",
       "2         1    5    2  NaN      NaN                              lncRNA   NaN   \n",
       "3         4    6    1  NaN      NaN  transcribed unprocessed pseudogene   NaN   \n",
       "4         4    6    1  NaN      NaN                      protein coding   NaN   \n",
       "...     ...  ...  ...  ...      ...                                 ...   ...   \n",
       "316574   20    9    6  NaN      NaN                      protein coding  5/16   \n",
       "316575   20    9    6  NaN      NaN                      protein coding  5/17   \n",
       "316576    8    7    1  NaN      NaN                      protein coding   NaN   \n",
       "316577    8    7    1  NaN      NaN                      protein coding   NaN   \n",
       "316578    8    7    1  NaN      NaN                      protein coding   NaN   \n",
       "\n",
       "       INTRON Protein Position and Amino Acid   Codons  STRAND PUBMED  \n",
       "0         NaN                             NaN      NaN     1.0    NaN  \n",
       "1         NaN                             NaN      NaN     1.0    NaN  \n",
       "2         NaN                             NaN      NaN     1.0    NaN  \n",
       "3         NaN                             NaN      NaN     1.0    NaN  \n",
       "4         2/2                             NaN      NaN     1.0    NaN  \n",
       "...       ...                             ...      ...     ...    ...  \n",
       "316574    NaN                            Q117  caA/caG    -1.0    NaN  \n",
       "316575    NaN                            Q117  caA/caG    -1.0    NaN  \n",
       "316576   3/27                             NaN      NaN    -1.0    NaN  \n",
       "316577   3/15                             NaN      NaN    -1.0    NaN  \n",
       "316578   3/16                             NaN      NaN    -1.0    NaN  \n",
       "\n",
       "[316579 rows x 41 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = merged_3[['Gene Name', 'ID', 'CHROM', 'POS', 'Literature', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f1789224",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL4_depth_vcf_processed.xlsx', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39d9e31c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a9cbd5b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create sample data\n",
    "data1 = {'Gene': ['OR4F29', 'LOC107985728,SAMD11', 'ABCD1,XYZ']}\n",
    "df1 = pd.DataFrame(data1)\n",
    "\n",
    "data2 = {'Gene': ['ABCD1', 'XYZ', 'EFGH', 'IJKL']}\n",
    "df2 = pd.DataFrame(data2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "52999944",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>XYZ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EFGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IJKL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene\n",
       "0  ABCD1\n",
       "1    XYZ\n",
       "2   EFGH\n",
       "3   IJKL"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24647315",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7301213a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "307b8549",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>...</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS Literature REF ALT  \\\n",
       "0            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "1            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "2            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "3            NaN  rs1416222198  chr1     65591         No   C   T   \n",
       "4            NaN  rs1416222198  chr1     65591         No   C   T   \n",
       "...          ...           ...   ...       ...        ...  ..  ..   \n",
       "316574       NaN             .  chrY  23190581         No   T   C   \n",
       "316575       NaN             .  chrY  23190581         No   T   C   \n",
       "316576       NaN             .  chrY  23191171         No   C   T   \n",
       "316577       NaN             .  chrY  23191171         No   C   T   \n",
       "316578       NaN             .  chrY  23191171         No   C   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score  ... ADR  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  ...   2   \n",
       "1       Heterozygous      5 prime UTR variant              3/10  ...   2   \n",
       "2       Heterozygous  downstream gene variant              2/10  ...   2   \n",
       "3       Heterozygous  downstream gene variant              2/10  ...   1   \n",
       "4       Heterozygous           intron variant              2/10  ...   1   \n",
       "...              ...                      ...               ...  ...  ..   \n",
       "316574  Heterozygous       synonymous variant              3/10  ...   6   \n",
       "316575  Heterozygous       synonymous variant              3/10  ...   6   \n",
       "316576  Heterozygous           intron variant              2/10  ...   1   \n",
       "316577  Heterozygous           intron variant              2/10  ...   1   \n",
       "316578  Heterozygous           intron variant              2/10  ...   1   \n",
       "\n",
       "        SIFT PolyPhen                             BIOTYPE  EXON  INTRON  \\\n",
       "0        NaN      NaN  transcribed unprocessed pseudogene   NaN     NaN   \n",
       "1        NaN      NaN                      protein coding   2/3     NaN   \n",
       "2        NaN      NaN                              lncRNA   NaN     NaN   \n",
       "3        NaN      NaN  transcribed unprocessed pseudogene   NaN     NaN   \n",
       "4        NaN      NaN                      protein coding   NaN     2/2   \n",
       "...      ...      ...                                 ...   ...     ...   \n",
       "316574   NaN      NaN                      protein coding  5/16     NaN   \n",
       "316575   NaN      NaN                      protein coding  5/17     NaN   \n",
       "316576   NaN      NaN                      protein coding   NaN    3/27   \n",
       "316577   NaN      NaN                      protein coding   NaN    3/15   \n",
       "316578   NaN      NaN                      protein coding   NaN    3/16   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  NaN      NaN    1.0    NaN  \n",
       "1                                  NaN      NaN    1.0    NaN  \n",
       "2                                  NaN      NaN    1.0    NaN  \n",
       "3                                  NaN      NaN    1.0    NaN  \n",
       "4                                  NaN      NaN    1.0    NaN  \n",
       "...                                ...      ...    ...    ...  \n",
       "316574                            Q117  caA/caG   -1.0    NaN  \n",
       "316575                            Q117  caA/caG   -1.0    NaN  \n",
       "316576                             NaN      NaN   -1.0    NaN  \n",
       "316577                             NaN      NaN   -1.0    NaN  \n",
       "316578                             NaN      NaN   -1.0    NaN  \n",
       "\n",
       "[316579 rows x 41 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL4_depth_vcf_processed.xlsx')\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "10481422",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LDLR-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NOS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRKAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>TRDN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>LDLRAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>PCSK9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>APOB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>STAP1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name\n",
       "0       LDLR\n",
       "1   LDLR-AS1\n",
       "2       NOS3\n",
       "3     PRKAG2\n",
       "4       LRP6\n",
       "..       ...\n",
       "75      TRDN\n",
       "76   LDLRAP1\n",
       "77     PCSK9\n",
       "78      APOB\n",
       "79     STAP1\n",
       "\n",
       "[80 rows x 1 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/cardiac genes.xlsx')\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "64bb0ab3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>...</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM       POS Literature REF ALT  \\\n",
       "0            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "1            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "2            NaN  rs1473003496  chr1     65529         No   C   T   \n",
       "3            NaN  rs1416222198  chr1     65591         No   C   T   \n",
       "4            NaN  rs1416222198  chr1     65591         No   C   T   \n",
       "...          ...           ...   ...       ...        ...  ..  ..   \n",
       "316574       NaN             .  chrY  23190581         No   T   C   \n",
       "316575       NaN             .  chrY  23190581         No   T   C   \n",
       "316576       NaN             .  chrY  23191171         No   C   T   \n",
       "316577       NaN             .  chrY  23191171         No   C   T   \n",
       "316578       NaN             .  chrY  23191171         No   C   T   \n",
       "\n",
       "            Zygosity              Consequence Consequence_score  ... SIFT  \\\n",
       "0       Heterozygous  downstream gene variant              2/10  ...  NaN   \n",
       "1       Heterozygous      5 prime UTR variant              3/10  ...  NaN   \n",
       "2       Heterozygous  downstream gene variant              2/10  ...  NaN   \n",
       "3       Heterozygous  downstream gene variant              2/10  ...  NaN   \n",
       "4       Heterozygous           intron variant              2/10  ...  NaN   \n",
       "...              ...                      ...               ...  ...  ...   \n",
       "316574  Heterozygous       synonymous variant              3/10  ...  NaN   \n",
       "316575  Heterozygous       synonymous variant              3/10  ...  NaN   \n",
       "316576  Heterozygous           intron variant              2/10  ...  NaN   \n",
       "316577  Heterozygous           intron variant              2/10  ...  NaN   \n",
       "316578  Heterozygous           intron variant              2/10  ...  NaN   \n",
       "\n",
       "        PolyPhen                             BIOTYPE  EXON INTRON  \\\n",
       "0            NaN  transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1            NaN                      protein coding   2/3    NaN   \n",
       "2            NaN                              lncRNA   NaN    NaN   \n",
       "3            NaN  transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4            NaN                      protein coding   NaN    2/2   \n",
       "...          ...                                 ...   ...    ...   \n",
       "316574       NaN                      protein coding  5/16    NaN   \n",
       "316575       NaN                      protein coding  5/17    NaN   \n",
       "316576       NaN                      protein coding   NaN   3/27   \n",
       "316577       NaN                      protein coding   NaN   3/15   \n",
       "316578       NaN                      protein coding   NaN   3/16   \n",
       "\n",
       "        Protein Position and Amino Acid   Codons STRAND PUBMED Gene_Match  \n",
       "0                                   NaN      NaN    1.0    NaN         No  \n",
       "1                                   NaN      NaN    1.0    NaN         No  \n",
       "2                                   NaN      NaN    1.0    NaN         No  \n",
       "3                                   NaN      NaN    1.0    NaN         No  \n",
       "4                                   NaN      NaN    1.0    NaN         No  \n",
       "...                                 ...      ...    ...    ...        ...  \n",
       "316574                             Q117  caA/caG   -1.0    NaN         No  \n",
       "316575                             Q117  caA/caG   -1.0    NaN         No  \n",
       "316576                              NaN      NaN   -1.0    NaN         No  \n",
       "316577                              NaN      NaN   -1.0    NaN         No  \n",
       "316578                              NaN      NaN   -1.0    NaN         No  \n",
       "\n",
       "[316579 rows x 42 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in df1['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df2['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            df1.loc[df1['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a3d01725",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>...</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7713</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7714</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7715</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7716</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>1/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7717</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>5/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304958</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>Yes</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>...</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304959</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L98</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304960</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304961</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304962</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4380 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         ID CHROM        POS Literature REF ALT      Zygosity  \\\n",
       "7713     LDLRAP1  rs6687605  chr1   25563141         No   T   C  Heterozygous   \n",
       "7714     LDLRAP1  rs6687605  chr1   25563141         No   T   C  Heterozygous   \n",
       "7715     LDLRAP1  rs6687605  chr1   25563141         No   T   C  Heterozygous   \n",
       "7716     LDLRAP1  rs6687605  chr1   25563141         No   T   C  Heterozygous   \n",
       "7717     LDLRAP1  rs6687605  chr1   25563141         No   T   C  Heterozygous   \n",
       "...          ...        ...   ...        ...        ...  ..  ..           ...   \n",
       "304958     ABCA1  rs2230806  chr9  104858586        Yes   C   T    Homozygous   \n",
       "304959     ABCA1  rs2230805  chr9  104861748         No   C   T    Homozygous   \n",
       "304960     ABCA1  rs2230805  chr9  104861748         No   C   T    Homozygous   \n",
       "304961     ABCA1  rs2230805  chr9  104861748         No   C   T    Homozygous   \n",
       "304962     ABCA1  rs2230805  chr9  104861748         No   C   T    Homozygous   \n",
       "\n",
       "                               Consequence Consequence_score  ...  \\\n",
       "7713                      missense variant              7/10  ...   \n",
       "7714                 upstream gene variant              2/10  ...   \n",
       "7715                 upstream gene variant              2/10  ...   \n",
       "7716    non coding transcript exon variant              2/10  ...   \n",
       "7717    non coding transcript exon variant              2/10  ...   \n",
       "...                                    ...               ...  ...   \n",
       "304958                    missense variant              7/10  ...   \n",
       "304959                  synonymous variant              3/10  ...   \n",
       "304960                  synonymous variant              3/10  ...   \n",
       "304961                  synonymous variant              3/10  ...   \n",
       "304962                  synonymous variant              3/10  ...   \n",
       "\n",
       "                               SIFT       PolyPhen  \\\n",
       "7713              deleterious(0.04)  benign(0.037)   \n",
       "7714                            NaN            NaN   \n",
       "7715                            NaN            NaN   \n",
       "7716                            NaN            NaN   \n",
       "7717                            NaN            NaN   \n",
       "...                             ...            ...   \n",
       "304958  tolerated low confidence(1)      benign(0)   \n",
       "304959                          NaN            NaN   \n",
       "304960                          NaN            NaN   \n",
       "304961                          NaN            NaN   \n",
       "304962                          NaN            NaN   \n",
       "\n",
       "                               BIOTYPE  EXON INTRON  \\\n",
       "7713                    protein coding   6/9    NaN   \n",
       "7714    protein coding CDS not defined   NaN    NaN   \n",
       "7715    protein coding CDS not defined   NaN    NaN   \n",
       "7716    protein coding CDS not defined   1/4    NaN   \n",
       "7717    protein coding CDS not defined   5/7    NaN   \n",
       "...                                ...   ...    ...   \n",
       "304958                  protein coding  7/50    NaN   \n",
       "304959                  protein coding   5/5    NaN   \n",
       "304960                  protein coding  6/50    NaN   \n",
       "304961                  protein coding   6/8    NaN   \n",
       "304962                  protein coding  6/50    NaN   \n",
       "\n",
       "        Protein Position and Amino Acid   Codons STRAND  \\\n",
       "7713                              S202P  Tcc/Ccc    1.0   \n",
       "7714                                NaN      NaN    1.0   \n",
       "7715                                NaN      NaN    1.0   \n",
       "7716                                NaN      NaN    1.0   \n",
       "7717                                NaN      NaN    1.0   \n",
       "...                                 ...      ...    ...   \n",
       "304958                            R219K  aGg/aAg   -1.0   \n",
       "304959                              L98  ctG/ctA   -1.0   \n",
       "304960                             L158  ctG/ctA   -1.0   \n",
       "304961                             L158  ctG/ctA   -1.0   \n",
       "304962                             L158  ctG/ctA   -1.0   \n",
       "\n",
       "                                                   PUBMED Gene_Match  \n",
       "7713         31227787,24906453,31397093,26129832,24067191        Yes  \n",
       "7714         31227787,24906453,31397093,26129832,24067191        Yes  \n",
       "7715         31227787,24906453,31397093,26129832,24067191        Yes  \n",
       "7716         31227787,24906453,31397093,26129832,24067191        Yes  \n",
       "7717         31227787,24906453,31397093,26129832,24067191        Yes  \n",
       "...                                                   ...        ...  \n",
       "304958  23021345,23497168,28690685,20185793,30584432,2...        Yes  \n",
       "304959  20167577,22675311,29156910,19606474,22369239,1...        Yes  \n",
       "304960  20167577,22675311,29156910,19606474,22369239,1...        Yes  \n",
       "304961  20167577,22675311,29156910,19606474,22369239,1...        Yes  \n",
       "304962  20167577,22675311,29156910,19606474,22369239,1...        Yes  \n",
       "\n",
       "[4380 rows x 42 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1[df1['Gene_Match'] == 'Yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "258d52f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'ID', 'CHROM', 'POS', 'Literature', 'REF', 'ALT',\n",
       "       'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT',\n",
       "       'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
       "       'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)',\n",
       "       'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR',\n",
       "       'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Gene_Match'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "ee48dcad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>2/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1416222198</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65591</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316574</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316575</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q117</td>\n",
       "      <td>caA/caG</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316576</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316577</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316578</th>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>23191171</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>316579 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match            ID CHROM       POS Literature REF ALT  \\\n",
       "0            NaN         No  rs1473003496  chr1     65529         No   C   T   \n",
       "1            NaN         No  rs1473003496  chr1     65529         No   C   T   \n",
       "2            NaN         No  rs1473003496  chr1     65529         No   C   T   \n",
       "3            NaN         No  rs1416222198  chr1     65591         No   C   T   \n",
       "4            NaN         No  rs1416222198  chr1     65591         No   C   T   \n",
       "...          ...        ...           ...   ...       ...        ...  ..  ..   \n",
       "316574       NaN         No             .  chrY  23190581         No   T   C   \n",
       "316575       NaN         No             .  chrY  23190581         No   T   C   \n",
       "316576       NaN         No             .  chrY  23191171         No   C   T   \n",
       "316577       NaN         No             .  chrY  23191171         No   C   T   \n",
       "316578       NaN         No             .  chrY  23191171         No   C   T   \n",
       "\n",
       "            Zygosity              Consequence  ... ADR SIFT  PolyPhen  \\\n",
       "0       Heterozygous  downstream gene variant  ...   2  NaN       NaN   \n",
       "1       Heterozygous      5 prime UTR variant  ...   2  NaN       NaN   \n",
       "2       Heterozygous  downstream gene variant  ...   2  NaN       NaN   \n",
       "3       Heterozygous  downstream gene variant  ...   1  NaN       NaN   \n",
       "4       Heterozygous           intron variant  ...   1  NaN       NaN   \n",
       "...              ...                      ...  ...  ..  ...       ...   \n",
       "316574  Heterozygous       synonymous variant  ...   6  NaN       NaN   \n",
       "316575  Heterozygous       synonymous variant  ...   6  NaN       NaN   \n",
       "316576  Heterozygous           intron variant  ...   1  NaN       NaN   \n",
       "316577  Heterozygous           intron variant  ...   1  NaN       NaN   \n",
       "316578  Heterozygous           intron variant  ...   1  NaN       NaN   \n",
       "\n",
       "                                   BIOTYPE  EXON INTRON  \\\n",
       "0       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "1                           protein coding   2/3    NaN   \n",
       "2                                   lncRNA   NaN    NaN   \n",
       "3       transcribed unprocessed pseudogene   NaN    NaN   \n",
       "4                           protein coding   NaN    2/2   \n",
       "...                                    ...   ...    ...   \n",
       "316574                      protein coding  5/16    NaN   \n",
       "316575                      protein coding  5/17    NaN   \n",
       "316576                      protein coding   NaN   3/27   \n",
       "316577                      protein coding   NaN   3/15   \n",
       "316578                      protein coding   NaN   3/16   \n",
       "\n",
       "        Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                   NaN      NaN    1.0    NaN  \n",
       "1                                   NaN      NaN    1.0    NaN  \n",
       "2                                   NaN      NaN    1.0    NaN  \n",
       "3                                   NaN      NaN    1.0    NaN  \n",
       "4                                   NaN      NaN    1.0    NaN  \n",
       "...                                 ...      ...    ...    ...  \n",
       "316574                             Q117  caA/caG   -1.0    NaN  \n",
       "316575                             Q117  caA/caG   -1.0    NaN  \n",
       "316576                              NaN      NaN   -1.0    NaN  \n",
       "316577                              NaN      NaN   -1.0    NaN  \n",
       "316578                              NaN      NaN   -1.0    NaN  \n",
       "\n",
       "[316579 rows x 42 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = df1[['Gene Name', 'Gene_Match', 'ID', 'CHROM', 'POS', 'Literature', 'REF', 'ALT',\n",
    "       'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT',\n",
    "       'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "       'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)',\n",
    "       'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR',\n",
    "       'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "be254a7a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7713</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Hypercholesterolemia, familial, 4,Cardiovascul...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000374338.5</td>\n",
       "      <td>c.604T&gt;C</td>\n",
       "      <td>ENSP00000363458.4</td>\n",
       "      <td>p.Ser202Pro</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>3.160900e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>benign(0.037)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S202P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7714</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Hypercholesterolemia, familial, 4,Cardiovascul...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>3.160900e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7715</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Hypercholesterolemia, familial, 4,Cardiovascul...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>3.160900e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7716</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Hypercholesterolemia, familial, 4,Cardiovascul...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000484476.5</td>\n",
       "      <td>n.326T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>3.160900e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>1/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7717</th>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs6687605</td>\n",
       "      <td>chr1</td>\n",
       "      <td>25563141</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Hypercholesterolemia, familial, 4,Cardiovascul...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>296982.0</td>\n",
       "      <td>ENST00000488127.1</td>\n",
       "      <td>n.1074T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>3.160900e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td>5/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31227787,24906453,31397093,26129832,24067191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304958</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2230806</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104858586</td>\n",
       "      <td>Yes</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Cardiovascular phenotype,Hypoalphalipoproteine...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>9506.0</td>\n",
       "      <td>ENST00000678995.1</td>\n",
       "      <td>c.656G&gt;A</td>\n",
       "      <td>ENSP00000504612.1</td>\n",
       "      <td>p.Arg219Lys</td>\n",
       "      <td>1/1</td>\n",
       "      <td>206</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.259800e-21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R219K</td>\n",
       "      <td>aGg/aAg</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>23021345,23497168,28690685,20185793,30584432,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304959</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Cardiovascular phenotype,Hypoalphalipoproteine...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>364460.0</td>\n",
       "      <td>ENST00000374733.1</td>\n",
       "      <td>c.294G&gt;A</td>\n",
       "      <td>ENSP00000363865.1</td>\n",
       "      <td>p.Leu98%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698000e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>5/5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L98</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304960</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Cardiovascular phenotype,Hypoalphalipoproteine...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>364460.0</td>\n",
       "      <td>ENST00000374736.8</td>\n",
       "      <td>c.474G&gt;A</td>\n",
       "      <td>ENSP00000363868.3</td>\n",
       "      <td>p.Leu158%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698000e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304961</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Cardiovascular phenotype,Hypoalphalipoproteine...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>364460.0</td>\n",
       "      <td>ENST00000423487.6</td>\n",
       "      <td>c.474G&gt;A</td>\n",
       "      <td>ENSP00000416623.2</td>\n",
       "      <td>p.Leu158%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698000e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304962</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2230805</td>\n",
       "      <td>chr9</td>\n",
       "      <td>104861748</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Cardiovascular phenotype,Hypoalphalipoproteine...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>364460.0</td>\n",
       "      <td>ENST00000678995.1</td>\n",
       "      <td>c.474G&gt;A</td>\n",
       "      <td>ENSP00000504612.1</td>\n",
       "      <td>p.Leu158%3D</td>\n",
       "      <td>1/1</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698000e-07</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L158</td>\n",
       "      <td>ctG/ctA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>20167577,22675311,29156910,19606474,22369239,1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4380 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match         ID CHROM        POS Literature REF ALT  \\\n",
       "7713     LDLRAP1        Yes  rs6687605  chr1   25563141         No   T   C   \n",
       "7714     LDLRAP1        Yes  rs6687605  chr1   25563141         No   T   C   \n",
       "7715     LDLRAP1        Yes  rs6687605  chr1   25563141         No   T   C   \n",
       "7716     LDLRAP1        Yes  rs6687605  chr1   25563141         No   T   C   \n",
       "7717     LDLRAP1        Yes  rs6687605  chr1   25563141         No   T   C   \n",
       "...          ...        ...        ...   ...        ...        ...  ..  ..   \n",
       "304958     ABCA1        Yes  rs2230806  chr9  104858586        Yes   C   T   \n",
       "304959     ABCA1        Yes  rs2230805  chr9  104861748         No   C   T   \n",
       "304960     ABCA1        Yes  rs2230805  chr9  104861748         No   C   T   \n",
       "304961     ABCA1        Yes  rs2230805  chr9  104861748         No   C   T   \n",
       "304962     ABCA1        Yes  rs2230805  chr9  104861748         No   C   T   \n",
       "\n",
       "            Zygosity                         Consequence Consequence_score  \\\n",
       "7713    Heterozygous                    missense variant              7/10   \n",
       "7714    Heterozygous               upstream gene variant              2/10   \n",
       "7715    Heterozygous               upstream gene variant              2/10   \n",
       "7716    Heterozygous  non coding transcript exon variant              2/10   \n",
       "7717    Heterozygous  non coding transcript exon variant              2/10   \n",
       "...              ...                                 ...               ...   \n",
       "304958    Homozygous                    missense variant              7/10   \n",
       "304959    Homozygous                  synonymous variant              3/10   \n",
       "304960    Homozygous                  synonymous variant              3/10   \n",
       "304961    Homozygous                  synonymous variant              3/10   \n",
       "304962    Homozygous                  synonymous variant              3/10   \n",
       "\n",
       "          IMPACT  IMPACT_score  \\\n",
       "7713    MODERATE           5.0   \n",
       "7714    MODIFIER           1.5   \n",
       "7715    MODIFIER           1.5   \n",
       "7716    MODIFIER           1.5   \n",
       "7717    MODIFIER           1.5   \n",
       "...          ...           ...   \n",
       "304958  MODERATE           5.0   \n",
       "304959       LOW           2.5   \n",
       "304960       LOW           2.5   \n",
       "304961       LOW           2.5   \n",
       "304962       LOW           2.5   \n",
       "\n",
       "                                            ClinVar_CLNDN CLIN_SIG  \\\n",
       "7713    Hypercholesterolemia, familial, 4,Cardiovascul...   benign   \n",
       "7714    Hypercholesterolemia, familial, 4,Cardiovascul...   benign   \n",
       "7715    Hypercholesterolemia, familial, 4,Cardiovascul...   benign   \n",
       "7716    Hypercholesterolemia, familial, 4,Cardiovascul...   benign   \n",
       "7717    Hypercholesterolemia, familial, 4,Cardiovascul...   benign   \n",
       "...                                                   ...      ...   \n",
       "304958  Cardiovascular phenotype,Hypoalphalipoproteine...   benign   \n",
       "304959  Cardiovascular phenotype,Hypoalphalipoproteine...   benign   \n",
       "304960  Cardiovascular phenotype,Hypoalphalipoproteine...   benign   \n",
       "304961  Cardiovascular phenotype,Hypoalphalipoproteine...   benign   \n",
       "304962  Cardiovascular phenotype,Hypoalphalipoproteine...   benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT   ClinVar  \\\n",
       "7713    criteria provided, multiple submitters, no con...  296982.0   \n",
       "7714    criteria provided, multiple submitters, no con...  296982.0   \n",
       "7715    criteria provided, multiple submitters, no con...  296982.0   \n",
       "7716    criteria provided, multiple submitters, no con...  296982.0   \n",
       "7717    criteria provided, multiple submitters, no con...  296982.0   \n",
       "...                                                   ...       ...   \n",
       "304958  criteria provided, multiple submitters, no con...    9506.0   \n",
       "304959  criteria provided, multiple submitters, no con...  364460.0   \n",
       "304960  criteria provided, multiple submitters, no con...  364460.0   \n",
       "304961  criteria provided, multiple submitters, no con...  364460.0   \n",
       "304962  criteria provided, multiple submitters, no con...  364460.0   \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "7713    ENST00000374338.5           c.604T>C  ENSP00000363458.4   \n",
       "7714                  NaN               None                NaN   \n",
       "7715                  NaN               None                NaN   \n",
       "7716    ENST00000484476.5           n.326T>C                NaN   \n",
       "7717    ENST00000488127.1          n.1074T>C                NaN   \n",
       "...                   ...                ...                ...   \n",
       "304958  ENST00000678995.1           c.656G>A  ENSP00000504612.1   \n",
       "304959  ENST00000374733.1           c.294G>A  ENSP00000363865.1   \n",
       "304960  ENST00000374736.8           c.474G>A  ENSP00000363868.3   \n",
       "304961  ENST00000423487.6           c.474G>A  ENSP00000416623.2   \n",
       "304962  ENST00000678995.1           c.474G>A  ENSP00000504612.1   \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP  DP  RD  AD    FREQ          PVAL  \\\n",
       "7713          p.Ser202Pro  0/1   25   15  15   8   7  46.67%  3.160900e-03   \n",
       "7714                 None  0/1   25   15  15   8   7  46.67%  3.160900e-03   \n",
       "7715                 None  0/1   25   15  15   8   7  46.67%  3.160900e-03   \n",
       "7716                 None  0/1   25   15  15   8   7  46.67%  3.160900e-03   \n",
       "7717                 None  0/1   25   15  15   8   7  46.67%  3.160900e-03   \n",
       "...                   ...  ...  ...  ...  ..  ..  ..     ...           ...   \n",
       "304958        p.Arg219Lys  1/1  206   36  36   0  36    100%  2.259800e-21   \n",
       "304959         p.Leu98%3D  1/1   64   12  12   0  12    100%  3.698000e-07   \n",
       "304960        p.Leu158%3D  1/1   64   12  12   0  12    100%  3.698000e-07   \n",
       "304961        p.Leu158%3D  1/1   64   12  12   0  12    100%  3.698000e-07   \n",
       "304962        p.Leu158%3D  1/1   64   12  12   0  12    100%  3.698000e-07   \n",
       "\n",
       "        RDF  RDR  ADF  ADR                         SIFT       PolyPhen  \\\n",
       "7713      6    2    5    2            deleterious(0.04)  benign(0.037)   \n",
       "7714      6    2    5    2                          NaN            NaN   \n",
       "7715      6    2    5    2                          NaN            NaN   \n",
       "7716      6    2    5    2                          NaN            NaN   \n",
       "7717      6    2    5    2                          NaN            NaN   \n",
       "...     ...  ...  ...  ...                          ...            ...   \n",
       "304958    0    0   24   12  tolerated low confidence(1)      benign(0)   \n",
       "304959    0    0   10    2                          NaN            NaN   \n",
       "304960    0    0   10    2                          NaN            NaN   \n",
       "304961    0    0   10    2                          NaN            NaN   \n",
       "304962    0    0   10    2                          NaN            NaN   \n",
       "\n",
       "                               BIOTYPE  EXON INTRON  \\\n",
       "7713                    protein coding   6/9    NaN   \n",
       "7714    protein coding CDS not defined   NaN    NaN   \n",
       "7715    protein coding CDS not defined   NaN    NaN   \n",
       "7716    protein coding CDS not defined   1/4    NaN   \n",
       "7717    protein coding CDS not defined   5/7    NaN   \n",
       "...                                ...   ...    ...   \n",
       "304958                  protein coding  7/50    NaN   \n",
       "304959                  protein coding   5/5    NaN   \n",
       "304960                  protein coding  6/50    NaN   \n",
       "304961                  protein coding   6/8    NaN   \n",
       "304962                  protein coding  6/50    NaN   \n",
       "\n",
       "       Protein Position and Amino Acid   Codons  STRAND  \\\n",
       "7713                             S202P  Tcc/Ccc     1.0   \n",
       "7714                               NaN      NaN     1.0   \n",
       "7715                               NaN      NaN     1.0   \n",
       "7716                               NaN      NaN     1.0   \n",
       "7717                               NaN      NaN     1.0   \n",
       "...                                ...      ...     ...   \n",
       "304958                           R219K  aGg/aAg    -1.0   \n",
       "304959                             L98  ctG/ctA    -1.0   \n",
       "304960                            L158  ctG/ctA    -1.0   \n",
       "304961                            L158  ctG/ctA    -1.0   \n",
       "304962                            L158  ctG/ctA    -1.0   \n",
       "\n",
       "                                                   PUBMED  \n",
       "7713         31227787,24906453,31397093,26129832,24067191  \n",
       "7714         31227787,24906453,31397093,26129832,24067191  \n",
       "7715         31227787,24906453,31397093,26129832,24067191  \n",
       "7716         31227787,24906453,31397093,26129832,24067191  \n",
       "7717         31227787,24906453,31397093,26129832,24067191  \n",
       "...                                                   ...  \n",
       "304958  23021345,23497168,28690685,20185793,30584432,2...  \n",
       "304959  20167577,22675311,29156910,19606474,22369239,1...  \n",
       "304960  20167577,22675311,29156910,19606474,22369239,1...  \n",
       "304961  20167577,22675311,29156910,19606474,22369239,1...  \n",
       "304962  20167577,22675311,29156910,19606474,22369239,1...  \n",
       "\n",
       "[4380 rows x 42 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1[df1['Gene_Match'] == 'Yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "98d31ffe",
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df1.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL4_Gene_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d38dcb56",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "7b899edb",
   "metadata": {},
   "source": [
    "## InDel files Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d16f0855",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5bf885e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "d282ba61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CTGTG', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ELN</td>\n",
       "      <td>rs10579871</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCD2</td>\n",
       "      <td>rs797045572</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>rs8190483</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['C', CG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GAA</td>\n",
       "      <td>rs865903736</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Glycogen_storage_disease', '_type_II|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>581 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity gene_name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3        ['CAT', C]    Homozygous    PTGES3   rs10579382   \n",
       "4      ['CTGTG', C]  Heterozygous       ELN   rs10579871   \n",
       "..              ...           ...       ...          ...   \n",
       "576       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "577       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "578       ['CT', C]  Heterozygous    FANCD2  rs797045572   \n",
       "579       ['C', CA]  Heterozygous   HSD17B3    rs8190483   \n",
       "580       ['C', CG]  Heterozygous       GAA  rs865903736   \n",
       "\n",
       "                                consequence  \\\n",
       "0                       5_prime_UTR_variant   \n",
       "1                       5_prime_UTR_variant   \n",
       "2                       3_prime_UTR_variant   \n",
       "3    initiatior_codon_variant', 'SO:0001589   \n",
       "4                            intron_variant   \n",
       "..                                      ...   \n",
       "576                     3_prime_UTR_variant   \n",
       "577                     3_prime_UTR_variant   \n",
       "578                          intron_variant   \n",
       "579                          intron_variant   \n",
       "580                     3_prime_UTR_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4    Conflicting_interpretations_of_pathogenicity   \n",
       "..                                            ...   \n",
       "576                                        Benign   \n",
       "577                        Uncertain_significance   \n",
       "578                        Uncertain_significance   \n",
       "579                                        Benign   \n",
       "580                          Benign/Likely_benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3                                         not_provided   \n",
       "4    Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...   \n",
       "..                                                 ...   \n",
       "576                      Cayman_type_cerebellar_ataxia   \n",
       "577                      Cayman_type_cerebellar_ataxia   \n",
       "578                                      not_specified   \n",
       "579                         not_specified|not_provided   \n",
       "580  Glycogen_storage_disease', '_type_II|not_provided   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "576             criteria_provided', '_single_submitter  germline        indel   \n",
       "577             criteria_provided', '_single_submitter  germline        indel   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "580  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \n",
       "0         Insertion           MayBe No          No                  No  \n",
       "1          Deletion           MayBe No          No                  No  \n",
       "2          Deletion           MayBe No          No                  No  \n",
       "3          Deletion           MayBe No          No                  No  \n",
       "4         Insertion           MayBe No          No                 Yes  \n",
       "..              ...                ...         ...                 ...  \n",
       "576        Deletion           MayBe No          No                  No  \n",
       "577        Deletion           MayBe No          No                  No  \n",
       "578        Deletion           MayBe No          No                 Yes  \n",
       "579       Insertion           MayBe No          No                 Yes  \n",
       "580       Insertion           MayBe No          No                  No  \n",
       "\n",
       "[581 rows x 14 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHGLBS535/KHGLBS535_indel_clinical_significance.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "6c436cc7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HLADR4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HLADR7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HLADQ7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HLADR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLAB13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HLADQW3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HLACW6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HLAA30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AIS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>FOXD3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NALP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CXCL10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HLAA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PTPN22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>LPP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name\n",
       "0     HLADR4\n",
       "1     HLADR7\n",
       "2     HLADQ7\n",
       "3     HLADR1\n",
       "4     HLAB13\n",
       "5    HLADQW3\n",
       "6     HLACW6\n",
       "7     HLAA30\n",
       "8       AIS1\n",
       "9      FOXD3\n",
       "10     NALP1\n",
       "11    CXCL10\n",
       "12     HLAA2\n",
       "13    PTPN22\n",
       "14       LPP"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Vitiligo_genes.xlsx')\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "d55c606a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CTGTG', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ELN</td>\n",
       "      <td>rs10579871</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCD2</td>\n",
       "      <td>rs797045572</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>rs8190483</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['C', CG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GAA</td>\n",
       "      <td>rs865903736</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Glycogen_storage_disease', '_type_II|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>581 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3        ['CAT', C]    Homozygous    PTGES3   rs10579382   \n",
       "4      ['CTGTG', C]  Heterozygous       ELN   rs10579871   \n",
       "..              ...           ...       ...          ...   \n",
       "576       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "577       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "578       ['CT', C]  Heterozygous    FANCD2  rs797045572   \n",
       "579       ['C', CA]  Heterozygous   HSD17B3    rs8190483   \n",
       "580       ['C', CG]  Heterozygous       GAA  rs865903736   \n",
       "\n",
       "                                consequence  \\\n",
       "0                       5_prime_UTR_variant   \n",
       "1                       5_prime_UTR_variant   \n",
       "2                       3_prime_UTR_variant   \n",
       "3    initiatior_codon_variant', 'SO:0001589   \n",
       "4                            intron_variant   \n",
       "..                                      ...   \n",
       "576                     3_prime_UTR_variant   \n",
       "577                     3_prime_UTR_variant   \n",
       "578                          intron_variant   \n",
       "579                          intron_variant   \n",
       "580                     3_prime_UTR_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4    Conflicting_interpretations_of_pathogenicity   \n",
       "..                                            ...   \n",
       "576                                        Benign   \n",
       "577                        Uncertain_significance   \n",
       "578                        Uncertain_significance   \n",
       "579                                        Benign   \n",
       "580                          Benign/Likely_benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3                                         not_provided   \n",
       "4    Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...   \n",
       "..                                                 ...   \n",
       "576                      Cayman_type_cerebellar_ataxia   \n",
       "577                      Cayman_type_cerebellar_ataxia   \n",
       "578                                      not_specified   \n",
       "579                         not_specified|not_provided   \n",
       "580  Glycogen_storage_disease', '_type_II|not_provided   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "576             criteria_provided', '_single_submitter  germline        indel   \n",
       "577             criteria_provided', '_single_submitter  germline        indel   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "580  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \n",
       "0         Insertion           MayBe No          No                  No  \n",
       "1          Deletion           MayBe No          No                  No  \n",
       "2          Deletion           MayBe No          No                  No  \n",
       "3          Deletion           MayBe No          No                  No  \n",
       "4         Insertion           MayBe No          No                 Yes  \n",
       "..              ...                ...         ...                 ...  \n",
       "576        Deletion           MayBe No          No                  No  \n",
       "577        Deletion           MayBe No          No                  No  \n",
       "578        Deletion           MayBe No          No                 Yes  \n",
       "579       Insertion           MayBe No          No                 Yes  \n",
       "580       Insertion           MayBe No          No                  No  \n",
       "\n",
       "[581 rows x 14 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.rename({'gene_name':'Gene Name'}, axis = 1)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "a06fe7f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CTGTG', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ELN</td>\n",
       "      <td>rs10579871</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCD2</td>\n",
       "      <td>rs797045572</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>rs8190483</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['C', CG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GAA</td>\n",
       "      <td>rs865903736</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Glycogen_storage_disease', '_type_II|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>581 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3        ['CAT', C]    Homozygous    PTGES3   rs10579382   \n",
       "4      ['CTGTG', C]  Heterozygous       ELN   rs10579871   \n",
       "..              ...           ...       ...          ...   \n",
       "576       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "577       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "578       ['CT', C]  Heterozygous    FANCD2  rs797045572   \n",
       "579       ['C', CA]  Heterozygous   HSD17B3    rs8190483   \n",
       "580       ['C', CG]  Heterozygous       GAA  rs865903736   \n",
       "\n",
       "                                consequence  \\\n",
       "0                       5_prime_UTR_variant   \n",
       "1                       5_prime_UTR_variant   \n",
       "2                       3_prime_UTR_variant   \n",
       "3    initiatior_codon_variant', 'SO:0001589   \n",
       "4                            intron_variant   \n",
       "..                                      ...   \n",
       "576                     3_prime_UTR_variant   \n",
       "577                     3_prime_UTR_variant   \n",
       "578                          intron_variant   \n",
       "579                          intron_variant   \n",
       "580                     3_prime_UTR_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4    Conflicting_interpretations_of_pathogenicity   \n",
       "..                                            ...   \n",
       "576                                        Benign   \n",
       "577                        Uncertain_significance   \n",
       "578                        Uncertain_significance   \n",
       "579                                        Benign   \n",
       "580                          Benign/Likely_benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3                                         not_provided   \n",
       "4    Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...   \n",
       "..                                                 ...   \n",
       "576                      Cayman_type_cerebellar_ataxia   \n",
       "577                      Cayman_type_cerebellar_ataxia   \n",
       "578                                      not_specified   \n",
       "579                         not_specified|not_provided   \n",
       "580  Glycogen_storage_disease', '_type_II|not_provided   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "576             criteria_provided', '_single_submitter  germline        indel   \n",
       "577             criteria_provided', '_single_submitter  germline        indel   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "580  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \\\n",
       "0         Insertion           MayBe No          No                  No   \n",
       "1          Deletion           MayBe No          No                  No   \n",
       "2          Deletion           MayBe No          No                  No   \n",
       "3          Deletion           MayBe No          No                  No   \n",
       "4         Insertion           MayBe No          No                 Yes   \n",
       "..              ...                ...         ...                 ...   \n",
       "576        Deletion           MayBe No          No                  No   \n",
       "577        Deletion           MayBe No          No                  No   \n",
       "578        Deletion           MayBe No          No                 Yes   \n",
       "579       Insertion           MayBe No          No                 Yes   \n",
       "580       Insertion           MayBe No          No                  No   \n",
       "\n",
       "    Gene_Match  \n",
       "0           No  \n",
       "1           No  \n",
       "2           No  \n",
       "3           No  \n",
       "4           No  \n",
       "..         ...  \n",
       "576         No  \n",
       "577         No  \n",
       "578         No  \n",
       "579         No  \n",
       "580         No  \n",
       "\n",
       "[581 rows x 15 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in df['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df2['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            df.loc[df['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "8795cd1b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No    581\n",
       "Name: Gene_Match, dtype: int64"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.Gene_Match.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "f21907b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CTGTG', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ELN</td>\n",
       "      <td>rs10579871</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCD2</td>\n",
       "      <td>rs797045572</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>rs8190483</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['C', CG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GAA</td>\n",
       "      <td>rs865903736</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Glycogen_storage_disease', '_type_II|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>581 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3        ['CAT', C]    Homozygous    PTGES3   rs10579382   \n",
       "4      ['CTGTG', C]  Heterozygous       ELN   rs10579871   \n",
       "..              ...           ...       ...          ...   \n",
       "576       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "577       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "578       ['CT', C]  Heterozygous    FANCD2  rs797045572   \n",
       "579       ['C', CA]  Heterozygous   HSD17B3    rs8190483   \n",
       "580       ['C', CG]  Heterozygous       GAA  rs865903736   \n",
       "\n",
       "                                consequence  \\\n",
       "0                       5_prime_UTR_variant   \n",
       "1                       5_prime_UTR_variant   \n",
       "2                       3_prime_UTR_variant   \n",
       "3    initiatior_codon_variant', 'SO:0001589   \n",
       "4                            intron_variant   \n",
       "..                                      ...   \n",
       "576                     3_prime_UTR_variant   \n",
       "577                     3_prime_UTR_variant   \n",
       "578                          intron_variant   \n",
       "579                          intron_variant   \n",
       "580                     3_prime_UTR_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4    Conflicting_interpretations_of_pathogenicity   \n",
       "..                                            ...   \n",
       "576                                        Benign   \n",
       "577                        Uncertain_significance   \n",
       "578                        Uncertain_significance   \n",
       "579                                        Benign   \n",
       "580                          Benign/Likely_benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3                                         not_provided   \n",
       "4    Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...   \n",
       "..                                                 ...   \n",
       "576                      Cayman_type_cerebellar_ataxia   \n",
       "577                      Cayman_type_cerebellar_ataxia   \n",
       "578                                      not_specified   \n",
       "579                         not_specified|not_provided   \n",
       "580  Glycogen_storage_disease', '_type_II|not_provided   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "576             criteria_provided', '_single_submitter  germline        indel   \n",
       "577             criteria_provided', '_single_submitter  germline        indel   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "580  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \\\n",
       "0         Insertion           MayBe No          No                  No   \n",
       "1          Deletion           MayBe No          No                  No   \n",
       "2          Deletion           MayBe No          No                  No   \n",
       "3          Deletion           MayBe No          No                  No   \n",
       "4         Insertion           MayBe No          No                 Yes   \n",
       "..              ...                ...         ...                 ...   \n",
       "576        Deletion           MayBe No          No                  No   \n",
       "577        Deletion           MayBe No          No                  No   \n",
       "578        Deletion           MayBe No          No                 Yes   \n",
       "579       Insertion           MayBe No          No                 Yes   \n",
       "580       Insertion           MayBe No          No                  No   \n",
       "\n",
       "    Gene_Match  \n",
       "0           No  \n",
       "1           No  \n",
       "2           No  \n",
       "3           No  \n",
       "4           No  \n",
       "..         ...  \n",
       "576         No  \n",
       "577         No  \n",
       "578         No  \n",
       "579         No  \n",
       "580         No  \n",
       "\n",
       "[581 rows x 15 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.consequence.fillna(\"NA\", inplace=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "5618f728",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.consequence.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "78d9e7ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>Consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>initiatior codon variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CTGTG', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ELN</td>\n",
       "      <td>rs10579871</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCD2</td>\n",
       "      <td>rs797045572</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>rs8190483</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['C', CG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GAA</td>\n",
       "      <td>rs865903736</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Glycogen_storage_disease', '_type_II|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>581 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3        ['CAT', C]    Homozygous    PTGES3   rs10579382   \n",
       "4      ['CTGTG', C]  Heterozygous       ELN   rs10579871   \n",
       "..              ...           ...       ...          ...   \n",
       "576       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "577       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "578       ['CT', C]  Heterozygous    FANCD2  rs797045572   \n",
       "579       ['C', CA]  Heterozygous   HSD17B3    rs8190483   \n",
       "580       ['C', CG]  Heterozygous       GAA  rs865903736   \n",
       "\n",
       "                                consequence  \\\n",
       "0                       5_prime_UTR_variant   \n",
       "1                       5_prime_UTR_variant   \n",
       "2                       3_prime_UTR_variant   \n",
       "3    initiatior_codon_variant', 'SO:0001589   \n",
       "4                            intron_variant   \n",
       "..                                      ...   \n",
       "576                     3_prime_UTR_variant   \n",
       "577                     3_prime_UTR_variant   \n",
       "578                          intron_variant   \n",
       "579                          intron_variant   \n",
       "580                     3_prime_UTR_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4    Conflicting_interpretations_of_pathogenicity   \n",
       "..                                            ...   \n",
       "576                                        Benign   \n",
       "577                        Uncertain_significance   \n",
       "578                        Uncertain_significance   \n",
       "579                                        Benign   \n",
       "580                          Benign/Likely_benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3                                         not_provided   \n",
       "4    Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...   \n",
       "..                                                 ...   \n",
       "576                      Cayman_type_cerebellar_ataxia   \n",
       "577                      Cayman_type_cerebellar_ataxia   \n",
       "578                                      not_specified   \n",
       "579                         not_specified|not_provided   \n",
       "580  Glycogen_storage_disease', '_type_II|not_provided   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "576             criteria_provided', '_single_submitter  germline        indel   \n",
       "577             criteria_provided', '_single_submitter  germline        indel   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "580  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \\\n",
       "0         Insertion           MayBe No          No                  No   \n",
       "1          Deletion           MayBe No          No                  No   \n",
       "2          Deletion           MayBe No          No                  No   \n",
       "3          Deletion           MayBe No          No                  No   \n",
       "4         Insertion           MayBe No          No                 Yes   \n",
       "..              ...                ...         ...                 ...   \n",
       "576        Deletion           MayBe No          No                  No   \n",
       "577        Deletion           MayBe No          No                  No   \n",
       "578        Deletion           MayBe No          No                 Yes   \n",
       "579       Insertion           MayBe No          No                 Yes   \n",
       "580       Insertion           MayBe No          No                  No   \n",
       "\n",
       "    Gene_Match               Consequence  \n",
       "0           No       5 prime UTR variant  \n",
       "1           No       5 prime UTR variant  \n",
       "2           No       3 prime UTR variant  \n",
       "3           No  initiatior codon variant  \n",
       "4           No            intron variant  \n",
       "..         ...                       ...  \n",
       "576         No       3 prime UTR variant  \n",
       "577         No       3 prime UTR variant  \n",
       "578         No            intron variant  \n",
       "579         No            intron variant  \n",
       "580         No       3 prime UTR variant  \n",
       "\n",
       "[581 rows x 16 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Consequence'] = df['consequence'].astype(str).apply(lambda x: x.replace('&', ',').replace('_', ' ').replace(\"'\", '').replace(\"-\", ' '))\n",
    "df['Consequence'] = df['Consequence'].str.split(',').str[0]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "850d81f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1 = df_1.rename({'consequence':'Consequence'}, axis =1)\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "d2d81270",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>initiatior codon variant</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CTGTG', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ELN</td>\n",
       "      <td>rs10579871</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCD2</td>\n",
       "      <td>rs797045572</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>rs8190483</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['C', CG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GAA</td>\n",
       "      <td>rs865903736</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Glycogen_storage_disease', '_type_II|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>581 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3        ['CAT', C]    Homozygous    PTGES3   rs10579382   \n",
       "4      ['CTGTG', C]  Heterozygous       ELN   rs10579871   \n",
       "..              ...           ...       ...          ...   \n",
       "576       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "577       ['AT', A]    Homozygous     ATCAY  rs796133913   \n",
       "578       ['CT', C]  Heterozygous    FANCD2  rs797045572   \n",
       "579       ['C', CA]  Heterozygous   HSD17B3    rs8190483   \n",
       "580       ['C', CG]  Heterozygous       GAA  rs865903736   \n",
       "\n",
       "                                consequence  \\\n",
       "0                       5_prime_UTR_variant   \n",
       "1                       5_prime_UTR_variant   \n",
       "2                       3_prime_UTR_variant   \n",
       "3    initiatior_codon_variant', 'SO:0001589   \n",
       "4                            intron_variant   \n",
       "..                                      ...   \n",
       "576                     3_prime_UTR_variant   \n",
       "577                     3_prime_UTR_variant   \n",
       "578                          intron_variant   \n",
       "579                          intron_variant   \n",
       "580                     3_prime_UTR_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4    Conflicting_interpretations_of_pathogenicity   \n",
       "..                                            ...   \n",
       "576                                        Benign   \n",
       "577                        Uncertain_significance   \n",
       "578                        Uncertain_significance   \n",
       "579                                        Benign   \n",
       "580                          Benign/Likely_benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3                                         not_provided   \n",
       "4    Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...   \n",
       "..                                                 ...   \n",
       "576                      Cayman_type_cerebellar_ataxia   \n",
       "577                      Cayman_type_cerebellar_ataxia   \n",
       "578                                      not_specified   \n",
       "579                         not_specified|not_provided   \n",
       "580  Glycogen_storage_disease', '_type_II|not_provided   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "576             criteria_provided', '_single_submitter  germline        indel   \n",
       "577             criteria_provided', '_single_submitter  germline        indel   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "580  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \\\n",
       "0         Insertion           MayBe No          No                  No   \n",
       "1          Deletion           MayBe No          No                  No   \n",
       "2          Deletion           MayBe No          No                  No   \n",
       "3          Deletion           MayBe No          No                  No   \n",
       "4         Insertion           MayBe No          No                 Yes   \n",
       "..              ...                ...         ...                 ...   \n",
       "576        Deletion           MayBe No          No                  No   \n",
       "577        Deletion           MayBe No          No                  No   \n",
       "578        Deletion           MayBe No          No                 Yes   \n",
       "579       Insertion           MayBe No          No                 Yes   \n",
       "580       Insertion           MayBe No          No                  No   \n",
       "\n",
       "    Gene_Match               Consequence Consequence_score  \n",
       "0           No       5 prime UTR variant              3/10  \n",
       "1           No       5 prime UTR variant              3/10  \n",
       "2           No       3 prime UTR variant              3/10  \n",
       "3           No  initiatior codon variant               NaN  \n",
       "4           No            intron variant              2/10  \n",
       "..         ...                       ...               ...  \n",
       "576         No       3 prime UTR variant              3/10  \n",
       "577         No       3 prime UTR variant              3/10  \n",
       "578         No            intron variant              2/10  \n",
       "579         No            intron variant              2/10  \n",
       "580         No       3 prime UTR variant              3/10  \n",
       "\n",
       "[581 rows x 17 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(df, df_1, on='Consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "fe3d3660",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['allele', 'zygocity', 'Gene Name', 'rsid', 'consequence',\n",
       "       'clinical_significance', 'associated_diseases', 'review_status',\n",
       "       'origin', 'variant_type', 'variant_subtype', 'Phargkb_ann_exists',\n",
       "       'is_mutation', 'Variant_is_precious', 'Gene_Match', 'Consequence',\n",
       "       'Consequence_score'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "c8eb6e34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CTGTG', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ELN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10579871</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>FANCD2</td>\n",
       "      <td>No</td>\n",
       "      <td>rs797045572</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>HSD17B3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs8190483</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['C', CG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GAA</td>\n",
       "      <td>No</td>\n",
       "      <td>rs865903736</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Glycogen_storage_disease', '_type_II|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>581 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name Gene_Match         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1         No   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1         No   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17         No   rs10549565   \n",
       "3        ['CAT', C]    Homozygous    PTGES3         No   rs10579382   \n",
       "4      ['CTGTG', C]  Heterozygous       ELN         No   rs10579871   \n",
       "..              ...           ...       ...        ...          ...   \n",
       "576       ['AT', A]    Homozygous     ATCAY         No  rs796133913   \n",
       "577       ['AT', A]    Homozygous     ATCAY         No  rs796133913   \n",
       "578       ['CT', C]  Heterozygous    FANCD2         No  rs797045572   \n",
       "579       ['C', CA]  Heterozygous   HSD17B3         No    rs8190483   \n",
       "580       ['C', CG]  Heterozygous       GAA         No  rs865903736   \n",
       "\n",
       "                                consequence Consequence_score  \\\n",
       "0                       5_prime_UTR_variant              3/10   \n",
       "1                       5_prime_UTR_variant              3/10   \n",
       "2                       3_prime_UTR_variant              3/10   \n",
       "3    initiatior_codon_variant', 'SO:0001589               NaN   \n",
       "4                            intron_variant              2/10   \n",
       "..                                      ...               ...   \n",
       "576                     3_prime_UTR_variant              3/10   \n",
       "577                     3_prime_UTR_variant              3/10   \n",
       "578                          intron_variant              2/10   \n",
       "579                          intron_variant              2/10   \n",
       "580                     3_prime_UTR_variant              3/10   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4    Conflicting_interpretations_of_pathogenicity   \n",
       "..                                            ...   \n",
       "576                                        Benign   \n",
       "577                        Uncertain_significance   \n",
       "578                        Uncertain_significance   \n",
       "579                                        Benign   \n",
       "580                          Benign/Likely_benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3                                         not_provided   \n",
       "4    Supravalvar_aortic_stenosis|Cutis_laxa', '_aut...   \n",
       "..                                                 ...   \n",
       "576                      Cayman_type_cerebellar_ataxia   \n",
       "577                      Cayman_type_cerebellar_ataxia   \n",
       "578                                      not_specified   \n",
       "579                         not_specified|not_provided   \n",
       "580  Glycogen_storage_disease', '_type_II|not_provided   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "576             criteria_provided', '_single_submitter  germline        indel   \n",
       "577             criteria_provided', '_single_submitter  germline        indel   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "580  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \n",
       "0         Insertion           MayBe No          No                  No  \n",
       "1          Deletion           MayBe No          No                  No  \n",
       "2          Deletion           MayBe No          No                  No  \n",
       "3          Deletion           MayBe No          No                  No  \n",
       "4         Insertion           MayBe No          No                 Yes  \n",
       "..              ...                ...         ...                 ...  \n",
       "576        Deletion           MayBe No          No                  No  \n",
       "577        Deletion           MayBe No          No                  No  \n",
       "578        Deletion           MayBe No          No                 Yes  \n",
       "579       Insertion           MayBe No          No                 Yes  \n",
       "580       Insertion           MayBe No          No                  No  \n",
       "\n",
       "[581 rows x 16 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = merged_1[['allele', 'zygocity', 'Gene Name', 'Gene_Match', 'rsid', 'consequence', 'Consequence_score', 'clinical_significance', 'associated_diseases', 'review_status',\n",
    "       'origin', 'variant_type', 'variant_subtype', 'Phargkb_ann_exists',\n",
    "       'is_mutation', 'Variant_is_precious']]\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "a4b4bfce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>['G', GT]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>MYOZ2</td>\n",
       "      <td>No</td>\n",
       "      <td>rs112369914</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>['A', AGGCCTTGCGTCGT]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SPIRE2</td>\n",
       "      <td>No</td>\n",
       "      <td>rs11275235</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CDH3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs11440411</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>EEM_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>['AG', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>MYBPC3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs11570050</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Cardiomyopathy|Hypertrophic_cardiomyopathy|Lef...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>['C', CTG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ARSB</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1255033691</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Mucopolysaccharidosis_type_6</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>['GCCCACCCCAC', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>VSX1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs144366921</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Polymorphous_corneal_dystrophy</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>['GCCCACCCCAC', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>VSX1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs144366921</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Polymorphous_corneal_dystrophy</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>['T', TCGGCG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ALG11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs148013251</td>\n",
       "      <td>genic_upstream_transcript_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Wilson_disease|ALG11-CDG|Congenital_disorder_o...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>['G', GT]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KMT2C</td>\n",
       "      <td>No</td>\n",
       "      <td>rs150073007</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Intellectual_disability|not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>LOC101928076</td>\n",
       "      <td>No</td>\n",
       "      <td>rs17133512</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Cardiomyopathy|Arrhythmogenic_right_ventricula...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>['CTG', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ZNF480</td>\n",
       "      <td>No</td>\n",
       "      <td>rs3217319</td>\n",
       "      <td>nonsense', 'SO:0001623</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>['A', AGTGTGC]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TMEM37</td>\n",
       "      <td>No</td>\n",
       "      <td>rs3217464</td>\n",
       "      <td>inframe_indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>['A', AT]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TMPO</td>\n",
       "      <td>No</td>\n",
       "      <td>rs34449077</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>['CA', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>GNRHR</td>\n",
       "      <td>No</td>\n",
       "      <td>rs34993359</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Isolated_GnRH_Deficiency|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>['T', TCCG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLC40A1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs3833570</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Hereditary_hemochromatosis|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>['T', TCCG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLC40A1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs3833570</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Hereditary_hemochromatosis</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>['T', TCCG]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SLC40A1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs3833570</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Hereditary_hemochromatosis</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>['T', TA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TTN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs397517532</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>Tibial_muscular_dystrophy|Myopathy', '_myofibr...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>['A', AC]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>CYP27A1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs397795841</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cholestanol_storage_disease|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366</th>\n",
       "      <td>['CG', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>C21orf58</td>\n",
       "      <td>No</td>\n",
       "      <td>rs397802954</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microcephalic_osteodysplastic_primordial_dwarfism</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>['CG', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>C21orf58</td>\n",
       "      <td>No</td>\n",
       "      <td>rs397802954</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microcephalic_osteodysplastic_primordial_dwarfism</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>['AC', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>SLC37A4</td>\n",
       "      <td>No</td>\n",
       "      <td>rs56966114</td>\n",
       "      <td>no_sequence_alteration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Glucose-6-phosphate_transport_defect|not_provided</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>['GCCT', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>SNORA100</td>\n",
       "      <td>No</td>\n",
       "      <td>rs575582638</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Developmental_and_epileptic_encephalopathy', '...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>['GT', G]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>GHSR</td>\n",
       "      <td>No</td>\n",
       "      <td>rs5854471</td>\n",
       "      <td>genic_downstream_transcript_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Short_stature_due_to_partial_GHR_deficiency</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>['CT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>KLLN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs71022512</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>['T', TC]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TIPIN</td>\n",
       "      <td>No</td>\n",
       "      <td>rs730880340</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    allele      zygocity     Gene Name Gene_Match  \\\n",
       "3               ['CAT', C]    Homozygous        PTGES3         No   \n",
       "33               ['G', GT]  Heterozygous         MYOZ2         No   \n",
       "36   ['A', AGGCCTTGCGTCGT]  Heterozygous        SPIRE2         No   \n",
       "110              ['C', CA]  Heterozygous          CDH3         No   \n",
       "116              ['AG', A]    Homozygous        MYBPC3         No   \n",
       "118             ['C', CTG]  Heterozygous          ARSB         No   \n",
       "154     ['GCCCACCCCAC', G]  Heterozygous          VSX1         No   \n",
       "155     ['GCCCACCCCAC', G]  Heterozygous          VSX1         No   \n",
       "165          ['T', TCGGCG]  Heterozygous         ALG11         No   \n",
       "171              ['G', GT]  Heterozygous         KMT2C         No   \n",
       "180              ['C', CA]  Heterozygous  LOC101928076         No   \n",
       "236             ['CTG', C]    Homozygous        ZNF480         No   \n",
       "238         ['A', AGTGTGC]  Heterozygous        TMEM37         No   \n",
       "256              ['A', AT]  Heterozygous          TMPO         No   \n",
       "272              ['CA', C]    Homozygous         GNRHR         No   \n",
       "332            ['T', TCCG]  Heterozygous       SLC40A1         No   \n",
       "333            ['T', TCCG]  Heterozygous       SLC40A1         No   \n",
       "334            ['T', TCCG]  Heterozygous       SLC40A1         No   \n",
       "353              ['T', TA]  Heterozygous           TTN         No   \n",
       "365              ['A', AC]    Homozygous       CYP27A1         No   \n",
       "366              ['CG', C]    Homozygous      C21orf58         No   \n",
       "367              ['CG', C]    Homozygous      C21orf58         No   \n",
       "438              ['AC', A]    Homozygous       SLC37A4         No   \n",
       "444            ['GCCT', G]  Heterozygous      SNORA100         No   \n",
       "466              ['GT', G]    Homozygous          GHSR         No   \n",
       "535              ['CT', C]    Homozygous          KLLN         No   \n",
       "560              ['T', TC]  Heterozygous         TIPIN         No   \n",
       "\n",
       "             rsid                             consequence Consequence_score  \\\n",
       "3      rs10579382  initiatior_codon_variant', 'SO:0001589               NaN   \n",
       "33    rs112369914                                      NA               NaN   \n",
       "36     rs11275235                                      NA               NaN   \n",
       "110    rs11440411                                      NA               NaN   \n",
       "116    rs11570050                                      NA               NaN   \n",
       "118  rs1255033691                                      NA               NaN   \n",
       "154   rs144366921                                      NA               NaN   \n",
       "155   rs144366921                                      NA               NaN   \n",
       "165   rs148013251       genic_upstream_transcript_variant               NaN   \n",
       "171   rs150073007                                nonsense               NaN   \n",
       "180    rs17133512                                      NA               NaN   \n",
       "236     rs3217319                  nonsense', 'SO:0001623               NaN   \n",
       "238     rs3217464                           inframe_indel               NaN   \n",
       "256    rs34449077                                      NA               NaN   \n",
       "272    rs34993359                                      NA               NaN   \n",
       "332     rs3833570                                      NA               NaN   \n",
       "333     rs3833570                                      NA               NaN   \n",
       "334     rs3833570                                      NA               NaN   \n",
       "353   rs397517532                                      NA               NaN   \n",
       "365   rs397795841                                      NA               NaN   \n",
       "366   rs397802954                                      NA               NaN   \n",
       "367   rs397802954                                      NA               NaN   \n",
       "438    rs56966114                  no_sequence_alteration               NaN   \n",
       "444   rs575582638                                      NA               NaN   \n",
       "466     rs5854471     genic_downstream_transcript_variant               NaN   \n",
       "535    rs71022512                                      NA               NaN   \n",
       "560   rs730880340                                      NA               NaN   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "3                                          Benign   \n",
       "33                                         Benign   \n",
       "36   Conflicting_interpretations_of_pathogenicity   \n",
       "110                                        Benign   \n",
       "116                          Benign/Likely_benign   \n",
       "118                                        Benign   \n",
       "154                        Uncertain_significance   \n",
       "155                                        Benign   \n",
       "165                          Benign/Likely_benign   \n",
       "171                                        Benign   \n",
       "180  Conflicting_interpretations_of_pathogenicity   \n",
       "236                                        Benign   \n",
       "238                                        Benign   \n",
       "256                                        Benign   \n",
       "272                          Benign/Likely_benign   \n",
       "332                                        Benign   \n",
       "333                                        Benign   \n",
       "334                        Uncertain_significance   \n",
       "353                          Benign/Likely_benign   \n",
       "365                                        Benign   \n",
       "366                        Uncertain_significance   \n",
       "367                                        Benign   \n",
       "438                                        Benign   \n",
       "444  Conflicting_interpretations_of_pathogenicity   \n",
       "466                                        Benign   \n",
       "535                                        Benign   \n",
       "560                                        Benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "3                                         not_provided   \n",
       "33                          not_specified|not_provided   \n",
       "36                          not_specified|not_provided   \n",
       "110                                       EEM_syndrome   \n",
       "116  Cardiomyopathy|Hypertrophic_cardiomyopathy|Lef...   \n",
       "118                       Mucopolysaccharidosis_type_6   \n",
       "154                     Polymorphous_corneal_dystrophy   \n",
       "155                     Polymorphous_corneal_dystrophy   \n",
       "165  Wilson_disease|ALG11-CDG|Congenital_disorder_o...   \n",
       "171              Intellectual_disability|not_specified   \n",
       "180  Cardiomyopathy|Arrhythmogenic_right_ventricula...   \n",
       "236                                      not_specified   \n",
       "238                                       not_provided   \n",
       "256                         not_specified|not_provided   \n",
       "272              Isolated_GnRH_Deficiency|not_provided   \n",
       "332            Hereditary_hemochromatosis|not_provided   \n",
       "333                         Hereditary_hemochromatosis   \n",
       "334                         Hereditary_hemochromatosis   \n",
       "353  Tibial_muscular_dystrophy|Myopathy', '_myofibr...   \n",
       "365           Cholestanol_storage_disease|not_provided   \n",
       "366  Microcephalic_osteodysplastic_primordial_dwarfism   \n",
       "367  Microcephalic_osteodysplastic_primordial_dwarfism   \n",
       "438  Glucose-6-phosphate_transport_defect|not_provided   \n",
       "444  Developmental_and_epileptic_encephalopathy', '...   \n",
       "466        Short_stature_due_to_partial_GHR_deficiency   \n",
       "535                         not_specified|not_provided   \n",
       "560                                      not_specified   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "33   criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "36   criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "110             criteria_provided', '_single_submitter  germline        indel   \n",
       "116  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "118             criteria_provided', '_single_submitter  germline        indel   \n",
       "154             criteria_provided', '_single_submitter  germline        indel   \n",
       "155             criteria_provided', '_single_submitter  germline        indel   \n",
       "165  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "171             criteria_provided', '_single_submitter       NaN        indel   \n",
       "180  criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "236             criteria_provided', '_single_submitter  germline        indel   \n",
       "238                     no_assertion_criteria_provided  germline        indel   \n",
       "256  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "272  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "332  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "333             criteria_provided', '_single_submitter  germline        indel   \n",
       "334             criteria_provided', '_single_submitter  germline        indel   \n",
       "353  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "365  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "366             criteria_provided', '_single_submitter  germline        indel   \n",
       "367             criteria_provided', '_single_submitter  germline        indel   \n",
       "438             criteria_provided', '_single_submitter  germline        indel   \n",
       "444  criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "466             criteria_provided', '_single_submitter  germline        indel   \n",
       "535             criteria_provided', '_single_submitter  germline        indel   \n",
       "560  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \n",
       "3          Deletion           MayBe No          No                  No  \n",
       "33        Insertion           MayBe No          No                 Yes  \n",
       "36        Insertion           MayBe No          No                 Yes  \n",
       "110       Insertion           MayBe No          No                  No  \n",
       "116        Deletion           MayBe No          No                 Yes  \n",
       "118       Insertion           MayBe No          No                  No  \n",
       "154        Deletion           MayBe No          No                  No  \n",
       "155        Deletion           MayBe No          No                  No  \n",
       "165       Insertion           MayBe No          No                 Yes  \n",
       "171       Insertion           MayBe No          No                 Yes  \n",
       "180       Insertion           MayBe No          No                 Yes  \n",
       "236        Deletion           MayBe No          No                 Yes  \n",
       "238       Insertion           MayBe No          No                  No  \n",
       "256       Insertion           MayBe No          No                 Yes  \n",
       "272        Deletion           MayBe No          No                  No  \n",
       "332       Insertion           MayBe No          No                 Yes  \n",
       "333       Insertion           MayBe No          No                 Yes  \n",
       "334       Insertion           MayBe No          No                 Yes  \n",
       "353       Insertion           MayBe No          No                 Yes  \n",
       "365       Insertion           MayBe No          No                  No  \n",
       "366       Insertion           MayBe No          No                  No  \n",
       "367        Deletion           MayBe No          No                  No  \n",
       "438        Deletion           MayBe No          No                  No  \n",
       "444        Deletion           MayBe No          No                  No  \n",
       "466        Deletion           MayBe No          No                  No  \n",
       "535        Deletion           MayBe No          No                  No  \n",
       "560       Insertion           MayBe No          No                 Yes  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "null_rows = merged_1[merged_1['Consequence_score'].isnull()]\n",
    "null_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "4f9725b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_1.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHGLBS535/KHGLBS535_InDel.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a3db09c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0cc6b5c4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0e441bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load datasets\n",
    "dataset1 = pd.read_csv('dataset1.csv')  # Replace with your file path\n",
    "dataset2 = pd.read_csv('dataset2.csv')  # Replace with your file path\n",
    "\n",
    "# Preprocess dataset1 to handle comma-separated gene names\n",
    "dataset1['Gene Name'] = dataset1['Gene Name'].str.replace(',', ' ').str.split()\n",
    "\n",
    "# Create a dictionary to map gene names to conditions from dataset1\n",
    "gene_to_condition = {}\n",
    "for _, row in dataset1.iterrows():\n",
    "    for gene in row['Gene Name']:\n",
    "        gene_to_condition[gene] = row['Condition']\n",
    "\n",
    "# Populate the condition column in dataset2\n",
    "dataset2['Condition'] = dataset2['Gene Name'].map(gene_to_condition)\n",
    "\n",
    "# Save the updated dataset2 with the new Condition column\n",
    "dataset2.to_csv('updated_dataset2.csv', index=False)  # Replace with your desired file name\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c2a1997b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ab8c349c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "      <th>Chrom-pos-Ref-Alt_38</th>\n",
       "      <th>rsID_hg38</th>\n",
       "      <th>Chrom-pos-Ref-Alt_37</th>\n",
       "      <th>rsID_hg37</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OTC</td>\n",
       "      <td>rs200016637</td>\n",
       "      <td>rs200016637</td>\n",
       "      <td>rs200016637</td>\n",
       "      <td>nan, X-38352177-CTTC-C,</td>\n",
       "      <td>rs200016637</td>\n",
       "      <td>nan, X-38211430-CTTC-C</td>\n",
       "      <td>rs200016637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OTC</td>\n",
       "      <td>rs201351170</td>\n",
       "      <td>rs201351170</td>\n",
       "      <td>rs201351170</td>\n",
       "      <td>X-38352183-TCTAT-T,</td>\n",
       "      <td>rs201351170</td>\n",
       "      <td>X-38211436-TCTAT-T</td>\n",
       "      <td>rs201351170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OTC</td>\n",
       "      <td>rs57752938</td>\n",
       "      <td>rs57752938</td>\n",
       "      <td>rs57752938</td>\n",
       "      <td>X-38352181-TTTCT-T,</td>\n",
       "      <td>rs57752938</td>\n",
       "      <td>X-38211434-TTTCT-T</td>\n",
       "      <td>rs57752938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OTC</td>\n",
       "      <td>rs5917572</td>\n",
       "      <td>rs5917572</td>\n",
       "      <td>rs5917572</td>\n",
       "      <td>X-38352256-A-G, X-38352256-A-T,</td>\n",
       "      <td>rs5917572</td>\n",
       "      <td>X-38211509-A-G, X-38211509-A-T</td>\n",
       "      <td>rs5917572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OTC</td>\n",
       "      <td>rs5963030</td>\n",
       "      <td>rs5963030</td>\n",
       "      <td>rs5963030</td>\n",
       "      <td>X-38352332-G-A,</td>\n",
       "      <td>rs5963030</td>\n",
       "      <td>X-38211585-G-A</td>\n",
       "      <td>rs5963030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>764</th>\n",
       "      <td>PDP1</td>\n",
       "      <td>c.575dupT</td>\n",
       "      <td>PDP1:c.575dupT</td>\n",
       "      <td>PDP1:c.575dupT</td>\n",
       "      <td>8-93922634-G-GG, 8-93922559-T-TT, nan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8-94934787-T-TT, 8-94934862-G-GG, nan</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>765</th>\n",
       "      <td>PDP1</td>\n",
       "      <td>c.277G&gt;T</td>\n",
       "      <td>PDP1:c.277G&gt;T</td>\n",
       "      <td>PDP1:c.277G&gt;T</td>\n",
       "      <td>8-93858083-G-T, 8-93922336-G-T, 8-93922261-G-T...</td>\n",
       "      <td>rs267606938</td>\n",
       "      <td>8-94870311-G-T, 8-94934564-G-T, 8-94934489-G-T...</td>\n",
       "      <td>rs267606938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>766</th>\n",
       "      <td>DLD</td>\n",
       "      <td>c.259C&gt;T</td>\n",
       "      <td>DLD:c.259C&gt;T</td>\n",
       "      <td>DLD:c.259C&gt;T</td>\n",
       "      <td>7-107902385-C-T, nan</td>\n",
       "      <td>rs925493461</td>\n",
       "      <td>7-107542830-C-T, nan</td>\n",
       "      <td>rs925493461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>767</th>\n",
       "      <td>DLD</td>\n",
       "      <td>c.803_804delAG</td>\n",
       "      <td>DLD:c.803_804delAG</td>\n",
       "      <td>DLD:c.803_804delAG</td>\n",
       "      <td>7-107916869-CTT-C, 7-107915623-CAG-C, 7-107915...</td>\n",
       "      <td>rs764611160</td>\n",
       "      <td>7-107556068-CAG-C, 7-107557314-CTT-C, 7-107556...</td>\n",
       "      <td>rs764611160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>768</th>\n",
       "      <td>DLD</td>\n",
       "      <td>c.1123G&gt;A</td>\n",
       "      <td>DLD:c.1123G&gt;A</td>\n",
       "      <td>DLD:c.1123G&gt;A</td>\n",
       "      <td>7-107917349-G-A, 7-107917418-G-ASLC25A15:p.Arg...</td>\n",
       "      <td>rs121964992</td>\n",
       "      <td>7-107557794-G-A, 7-107557863-G-A, 7-107559500-...</td>\n",
       "      <td>rs121964992</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>769 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene          Change Uploaded_variant_hg38 Uploaded_variant_hg37  \\\n",
       "0     OTC     rs200016637           rs200016637           rs200016637   \n",
       "1     OTC     rs201351170           rs201351170           rs201351170   \n",
       "2     OTC      rs57752938            rs57752938            rs57752938   \n",
       "3     OTC       rs5917572             rs5917572             rs5917572   \n",
       "4     OTC       rs5963030             rs5963030             rs5963030   \n",
       "..    ...             ...                   ...                   ...   \n",
       "764  PDP1       c.575dupT        PDP1:c.575dupT        PDP1:c.575dupT   \n",
       "765  PDP1        c.277G>T         PDP1:c.277G>T         PDP1:c.277G>T   \n",
       "766   DLD        c.259C>T          DLD:c.259C>T          DLD:c.259C>T   \n",
       "767   DLD  c.803_804delAG    DLD:c.803_804delAG    DLD:c.803_804delAG   \n",
       "768   DLD       c.1123G>A         DLD:c.1123G>A         DLD:c.1123G>A   \n",
       "\n",
       "                                  Chrom-pos-Ref-Alt_38    rsID_hg38  \\\n",
       "0                             nan, X-38352177-CTTC-C,   rs200016637   \n",
       "1                                 X-38352183-TCTAT-T,   rs201351170   \n",
       "2                                 X-38352181-TTTCT-T,    rs57752938   \n",
       "3                     X-38352256-A-G, X-38352256-A-T,     rs5917572   \n",
       "4                                     X-38352332-G-A,     rs5963030   \n",
       "..                                                 ...          ...   \n",
       "764              8-93922634-G-GG, 8-93922559-T-TT, nan          NaN   \n",
       "765  8-93858083-G-T, 8-93922336-G-T, 8-93922261-G-T...  rs267606938   \n",
       "766                               7-107902385-C-T, nan  rs925493461   \n",
       "767  7-107916869-CTT-C, 7-107915623-CAG-C, 7-107915...  rs764611160   \n",
       "768  7-107917349-G-A, 7-107917418-G-ASLC25A15:p.Arg...  rs121964992   \n",
       "\n",
       "                                  Chrom-pos-Ref-Alt_37    rsID_hg37  \n",
       "0                               nan, X-38211430-CTTC-C  rs200016637  \n",
       "1                                   X-38211436-TCTAT-T  rs201351170  \n",
       "2                                   X-38211434-TTTCT-T   rs57752938  \n",
       "3                       X-38211509-A-G, X-38211509-A-T    rs5917572  \n",
       "4                                       X-38211585-G-A    rs5963030  \n",
       "..                                                 ...          ...  \n",
       "764              8-94934787-T-TT, 8-94934862-G-GG, nan          NaN  \n",
       "765  8-94870311-G-T, 8-94934564-G-T, 8-94934489-G-T...  rs267606938  \n",
       "766                               7-107542830-C-T, nan  rs925493461  \n",
       "767  7-107556068-CAG-C, 7-107557314-CTT-C, 7-107556...  rs764611160  \n",
       "768  7-107557794-G-A, 7-107557863-G-A, 7-107559500-...  rs121964992  \n",
       "\n",
       "[769 rows x 8 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/Cerebellarataxia_Lit_final_Positions_hg38_hg37.csv', sep = '\\t')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4e2a9096",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Cerebellarataxia_Lit_final_Positions_hg38_hg37.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "010cb036",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a833b755",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>1.949000e-03</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14599</td>\n",
       "      <td>rs707680</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>37</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>36.67%</td>\n",
       "      <td>1.594000e-04</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14604</td>\n",
       "      <td>rs541940975</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>37</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>20</td>\n",
       "      <td>11</td>\n",
       "      <td>35.48%</td>\n",
       "      <td>1.665900e-04</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14610</td>\n",
       "      <td>rs878986575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>37</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>20</td>\n",
       "      <td>11</td>\n",
       "      <td>35.48%</td>\n",
       "      <td>1.665900e-04</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=120;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>84</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>94</td>\n",
       "      <td>26</td>\n",
       "      <td>21.67%</td>\n",
       "      <td>3.263200e-09</td>\n",
       "      <td>69</td>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82332</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>62</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>71.43%</td>\n",
       "      <td>5.499400e-07</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82333</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210759</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>77</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>35</td>\n",
       "      <td>22</td>\n",
       "      <td>37.93%</td>\n",
       "      <td>1.917900e-08</td>\n",
       "      <td>21</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82334</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210867</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>28%</td>\n",
       "      <td>4.700600e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82335</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211440</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>36.84%</td>\n",
       "      <td>3.992600e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>1.144200e-03</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>82337 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS           ID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14590     rs707679   G   A    .   PASS   \n",
       "1      chr1     14599     rs707680   T   A    .   PASS   \n",
       "2      chr1     14604  rs541940975   A   G    .   PASS   \n",
       "3      chr1     14610  rs878986575   T   C    .   PASS   \n",
       "4      chr1     14752   rs17855866   G   A    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "82332  chrY  57209613            .   A   G    .   PASS   \n",
       "82333  chrY  57210759            .   C   G    .   PASS   \n",
       "82334  chrY  57210867            .   T   C    .   PASS   \n",
       "82335  chrY  57211440            .   A   G    .   PASS   \n",
       "82336  chrY  57211716            .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP   DP  \\\n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24   24   \n",
       "1      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....  0/1  37   30   30   \n",
       "2      ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...  0/1  37   31   31   \n",
       "3      ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  37   31   31   \n",
       "4      ADP=120;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX...  0/1  84  120  120   \n",
       "...                                                  ...  ...  ..  ...  ...   \n",
       "82332  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...  0/1  62   21   21   \n",
       "82333  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...  0/1  77   58   58   \n",
       "82334  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  23   25   25   \n",
       "82335  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...  0/1  23   19   19   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38   38   \n",
       "\n",
       "       RD  AD    FREQ          PVAL  RDF  RDR  ADF  ADR  \n",
       "0      16   8  33.33%  1.949000e-03   16    0    8    0  \n",
       "1      19  11  36.67%  1.594000e-04   19    0   11    0  \n",
       "2      20  11  35.48%  1.665900e-04   20    0   11    0  \n",
       "3      20  11  35.48%  1.665900e-04   20    0   11    0  \n",
       "4      94  26  21.67%  3.263200e-09   69   25   24    2  \n",
       "...    ..  ..     ...           ...  ...  ...  ...  ...  \n",
       "82332   6  15  71.43%  5.499400e-07    2    4   10    5  \n",
       "82333  35  22  37.93%  1.917900e-08   21   14   14    8  \n",
       "82334  17   7     28%  4.700600e-03    4   13    1    6  \n",
       "82335  12   7  36.84%  3.992600e-03   11    1    6    1  \n",
       "82336  29   9  23.68%  1.144200e-03   23    6    3    6  \n",
       "\n",
       "[82337 rows x 20 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_4/KHBSTLGPTTL2/KHBSTLGPTTL2_final_depth.tsv', sep=\"\\t\", low_memory=False)\n",
    "#vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5f9343d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>1.949000e-03</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14599</td>\n",
       "      <td>rs707680</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>37</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>36.67%</td>\n",
       "      <td>1.594000e-04</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14604</td>\n",
       "      <td>rs541940975</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>37</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>20</td>\n",
       "      <td>11</td>\n",
       "      <td>35.48%</td>\n",
       "      <td>1.665900e-04</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14610</td>\n",
       "      <td>rs878986575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>37</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>20</td>\n",
       "      <td>11</td>\n",
       "      <td>35.48%</td>\n",
       "      <td>1.665900e-04</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14752</td>\n",
       "      <td>rs17855866</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=120;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>84</td>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "      <td>94</td>\n",
       "      <td>26</td>\n",
       "      <td>21.67%</td>\n",
       "      <td>3.263200e-09</td>\n",
       "      <td>69</td>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82332</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209613</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>62</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>71.43%</td>\n",
       "      <td>5.499400e-07</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82333</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210759</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>77</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>35</td>\n",
       "      <td>22</td>\n",
       "      <td>37.93%</td>\n",
       "      <td>1.917900e-08</td>\n",
       "      <td>21</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82334</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210867</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>28%</td>\n",
       "      <td>4.700600e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82335</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211440</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>36.84%</td>\n",
       "      <td>3.992600e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>1.144200e-03</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>82337 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS           ID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14590     rs707679   G   A    .   PASS   \n",
       "1      chr1     14599     rs707680   T   A    .   PASS   \n",
       "2      chr1     14604  rs541940975   A   G    .   PASS   \n",
       "3      chr1     14610  rs878986575   T   C    .   PASS   \n",
       "4      chr1     14752   rs17855866   G   A    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "82332  chrY  57209613            .   A   G    .   PASS   \n",
       "82333  chrY  57210759            .   C   G    .   PASS   \n",
       "82334  chrY  57210867            .   T   C    .   PASS   \n",
       "82335  chrY  57211440            .   A   G    .   PASS   \n",
       "82336  chrY  57211716            .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP   DP  \\\n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24   24   \n",
       "1      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,0....  0/1  37   30   30   \n",
       "2      ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8524,.,...  0/1  37   31   31   \n",
       "3      ADP=31;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  37   31   31   \n",
       "4      ADP=120;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX...  0/1  84  120  120   \n",
       "...                                                  ...  ...  ..  ...  ...   \n",
       "82332  ADP=21;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...  0/1  62   21   21   \n",
       "82333  ADP=58;WT=0;HET=1;HOM=0;NC=0;CSQ=G|non_coding_...  0/1  77   58   58   \n",
       "82334  ADP=25;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  23   25   25   \n",
       "82335  ADP=19;WT=0;HET=1;HOM=0;NC=0;CSQ=G|intron_vari...  0/1  23   19   19   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38   38   \n",
       "\n",
       "       RD  AD    FREQ          PVAL  RDF  RDR  ADF  ADR HET HOM      Zygosity  \n",
       "0      16   8  33.33%  1.949000e-03   16    0    8    0   1   0  Heterozygous  \n",
       "1      19  11  36.67%  1.594000e-04   19    0   11    0   1   0  Heterozygous  \n",
       "2      20  11  35.48%  1.665900e-04   20    0   11    0   1   0  Heterozygous  \n",
       "3      20  11  35.48%  1.665900e-04   20    0   11    0   1   0  Heterozygous  \n",
       "4      94  26  21.67%  3.263200e-09   69   25   24    2   1   0  Heterozygous  \n",
       "...    ..  ..     ...           ...  ...  ...  ...  ...  ..  ..           ...  \n",
       "82332   6  15  71.43%  5.499400e-07    2    4   10    5   1   0  Heterozygous  \n",
       "82333  35  22  37.93%  1.917900e-08   21   14   14    8   1   0  Heterozygous  \n",
       "82334  17   7     28%  4.700600e-03    4   13    1    6   1   0  Heterozygous  \n",
       "82335  12   7  36.84%  3.992600e-03   11    1    6    1   1   0  Heterozygous  \n",
       "82336  29   9  23.68%  1.144200e-03   23    6    3    6   1   0  Heterozygous  \n",
       "\n",
       "[82337 rows x 23 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cd458d11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>652066 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS        ID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "...     ...       ...       ...  ..  ..  ...    ...   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP  DP  \\\n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "...                                                  ...  ...  ..  ...  ..   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "\n",
       "       RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR HET HOM      Zygosity  \\\n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "...    ..  ..     ...       ...  ...  ...  ...  ...  ..  ..           ...   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq  \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...  \n",
       "0      A|intron_variant&non_coding_transcript_variant...  \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...  \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...  \n",
       "...                                                  ...  \n",
       "82336  C|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "82336  C|intron_variant&non_coding_transcript_variant...  \n",
       "82336  C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...  \n",
       "82336  C|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "82336  C|downstream_gene_variant|MODIFIER|DDX11L16|EN...  \n",
       "\n",
       "[652066 rows x 25 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0af3841f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE_SELECT</th>\n",
       "      <th>MANE_PLUS_CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT_ISOFORM</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>AF</th>\n",
       "      <th>AFR_AF</th>\n",
       "      <th>AMR_AF</th>\n",
       "      <th>EAS_AF</th>\n",
       "      <th>EUR_AF</th>\n",
       "      <th>SAS_AF</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450305</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs707679</td>\n",
       "      <td>920</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37102</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1311</td>\n",
       "      <td>0.09045</td>\n",
       "      <td>0.1524</td>\n",
       "      <td>0.1382</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>0.06325</td>\n",
       "      <td>0.1167</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1594</td>\n",
       "      <td>0.1394</td>\n",
       "      <td>0.1409</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>ENSG00000290825</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs707679</td>\n",
       "      <td>181</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td></td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1311</td>\n",
       "      <td>0.09045</td>\n",
       "      <td>0.1524</td>\n",
       "      <td>0.1382</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>0.06325</td>\n",
       "      <td>0.1167</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1594</td>\n",
       "      <td>0.1394</td>\n",
       "      <td>0.1409</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000488147</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td>10/10</td>\n",
       "      <td>ENST00000488147.1:n.1254-89C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs707679</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:38034</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1311</td>\n",
       "      <td>0.09045</td>\n",
       "      <td>0.1524</td>\n",
       "      <td>0.1382</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>0.06325</td>\n",
       "      <td>0.1167</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1594</td>\n",
       "      <td>0.1394</td>\n",
       "      <td>0.1409</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000619216</td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs707679</td>\n",
       "      <td>2779</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:50039</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1311</td>\n",
       "      <td>0.09045</td>\n",
       "      <td>0.1524</td>\n",
       "      <td>0.1382</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>0.06325</td>\n",
       "      <td>0.1167</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1594</td>\n",
       "      <td>0.1394</td>\n",
       "      <td>0.1409</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14590</td>\n",
       "      <td>rs707679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>A|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>A</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001164745</td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>rs707679</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.1311</td>\n",
       "      <td>0.09045</td>\n",
       "      <td>0.1524</td>\n",
       "      <td>0.1382</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>0.06325</td>\n",
       "      <td>0.1167</td>\n",
       "      <td>0.05789</td>\n",
       "      <td>0.1594</td>\n",
       "      <td>0.1394</td>\n",
       "      <td>0.1409</td>\n",
       "      <td>0.2095</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000484415</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.2513T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>2513</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000492963</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "      <td>ENST00000492963.6:n.2762-45T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000496011</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1989</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000496301</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.1171T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>1171</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82336</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211716</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>23.68%</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>ENSG00000227159</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000507418</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>468</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:37115</td>\n",
       "      <td>YES</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>652066 rows × 108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS        ID REF ALT QUAL FILTER  \\\n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "0      chr1     14590  rs707679   G   A    .   PASS   \n",
       "...     ...       ...       ...  ..  ..  ...    ...   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "82336  chrY  57211716         .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP  DP  \\\n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "0      ADP=24;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  27   24  24   \n",
       "...                                                  ...  ...  ..  ...  ..   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "82336  ADP=38;WT=0;HET=1;HOM=0;NC=0;CSQ=C|intron_vari...  0/1  29   38  38   \n",
       "\n",
       "       RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR HET HOM      Zygosity  \\\n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "0      16   8  33.33%  0.001949   16    0    8    0   1   0  Heterozygous   \n",
       "...    ..  ..     ...       ...  ...  ...  ...  ...  ..  ..           ...   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "82336  29   9  23.68%  0.001144   23    6    3    6   1   0  Heterozygous   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                     csq Allele  \\\n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...      A   \n",
       "0      A|downstream_gene_variant|MODIFIER|DDX11L2|ENS...      A   \n",
       "0      A|intron_variant&non_coding_transcript_variant...      A   \n",
       "0      A|downstream_gene_variant|MODIFIER|MIR6859-1|E...      A   \n",
       "0      A|regulatory_region_variant|MODIFIER|||Regulat...      A   \n",
       "...                                                  ...    ...   \n",
       "82336  C|non_coding_transcript_exon_variant|MODIFIER|...      C   \n",
       "82336  C|intron_variant&non_coding_transcript_variant...      C   \n",
       "82336  C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...      C   \n",
       "82336  C|non_coding_transcript_exon_variant|MODIFIER|...      C   \n",
       "82336  C|downstream_gene_variant|MODIFIER|DDX11L16|EN...      C   \n",
       "\n",
       "                                        Consequence    IMPACT     SYMBOL  \\\n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L1   \n",
       "0                           downstream_gene_variant  MODIFIER    DDX11L2   \n",
       "0      intron_variant&non_coding_transcript_variant  MODIFIER     WASH7P   \n",
       "0                           downstream_gene_variant  MODIFIER  MIR6859-1   \n",
       "0                         regulatory_region_variant  MODIFIER              \n",
       "...                                             ...       ...        ...   \n",
       "82336            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "82336  intron_variant&non_coding_transcript_variant  MODIFIER     WASH6P   \n",
       "82336                       downstream_gene_variant  MODIFIER     WASH6P   \n",
       "82336            non_coding_transcript_exon_variant  MODIFIER     WASH6P   \n",
       "82336                       downstream_gene_variant  MODIFIER   DDX11L16   \n",
       "\n",
       "                  Gene       Feature_type          Feature  \\\n",
       "0      ENSG00000223972         Transcript  ENST00000450305   \n",
       "0      ENSG00000290825         Transcript  ENST00000456328   \n",
       "0      ENSG00000227232         Transcript  ENST00000488147   \n",
       "0      ENSG00000278267         Transcript  ENST00000619216   \n",
       "0                       RegulatoryFeature  ENSR00001164745   \n",
       "...                ...                ...              ...   \n",
       "82336  ENSG00000182484         Transcript  ENST00000484415   \n",
       "82336  ENSG00000182484         Transcript  ENST00000492963   \n",
       "82336  ENSG00000182484         Transcript  ENST00000496011   \n",
       "82336  ENSG00000182484         Transcript  ENST00000496301   \n",
       "82336  ENSG00000227159         Transcript  ENST00000507418   \n",
       "\n",
       "                                  BIOTYPE EXON INTRON  \\\n",
       "0      transcribed_unprocessed_pseudogene               \n",
       "0                                  lncRNA               \n",
       "0                  unprocessed_pseudogene       10/10   \n",
       "0                                   miRNA               \n",
       "0                                enhancer               \n",
       "...                                   ...  ...    ...   \n",
       "82336                     retained_intron  5/5          \n",
       "82336                     retained_intron         5/5   \n",
       "82336                     retained_intron               \n",
       "82336                     retained_intron  2/2          \n",
       "82336              unprocessed_pseudogene               \n",
       "\n",
       "                                HGVSc HGVSp cDNA_position CDS_position  \\\n",
       "0                                                                        \n",
       "0                                                                        \n",
       "0      ENST00000488147.1:n.1254-89C>T                                    \n",
       "0                                                                        \n",
       "0                                                                        \n",
       "...                               ...   ...           ...          ...   \n",
       "82336     ENST00000484415.6:n.2513T>C                2513                \n",
       "82336  ENST00000492963.6:n.2762-45T>C                                    \n",
       "82336                                                                    \n",
       "82336     ENST00000496301.6:n.1171T>C                1171                \n",
       "82336                                                                    \n",
       "\n",
       "      Protein_position Amino_acids Codons Existing_variation DISTANCE STRAND  \\\n",
       "0                                                   rs707679      920      1   \n",
       "0                                                   rs707679      181      1   \n",
       "0                                                   rs707679              -1   \n",
       "0                                                   rs707679     2779     -1   \n",
       "0                                                   rs707679                   \n",
       "...                ...         ...    ...                ...      ...    ...   \n",
       "82336                                                                      1   \n",
       "82336                                                                      1   \n",
       "82336                                                            1989      1   \n",
       "82336                                                                      1   \n",
       "82336                                                             468     -1   \n",
       "\n",
       "      FLAGS VARIANT_CLASS SYMBOL_SOURCE     HGNC_ID CANONICAL MANE_SELECT  \\\n",
       "0                     SNV          HGNC  HGNC:37102       YES               \n",
       "0                     SNV    EntrezGene                   YES               \n",
       "0                     SNV          HGNC  HGNC:38034       YES               \n",
       "0                     SNV          HGNC  HGNC:50039       YES               \n",
       "0                     SNV                                                   \n",
       "...     ...           ...           ...         ...       ...         ...   \n",
       "82336                 SNV          HGNC  HGNC:31685                         \n",
       "82336                 SNV          HGNC  HGNC:31685                         \n",
       "82336                 SNV          HGNC  HGNC:31685                         \n",
       "82336                 SNV          HGNC  HGNC:31685                         \n",
       "82336                 SNV          HGNC  HGNC:37115       YES               \n",
       "\n",
       "      MANE_PLUS_CLINICAL TSL APPRIS CCDS ENSP SWISSPROT TREMBL UNIPARC  \\\n",
       "0                                                                        \n",
       "0                          1                                             \n",
       "0                                                                        \n",
       "0                                                                        \n",
       "0                                                                        \n",
       "...                  ...  ..    ...  ...  ...       ...    ...     ...   \n",
       "82336                      5                                             \n",
       "82336                      5                                             \n",
       "82336                      5                                             \n",
       "82336                      2                                             \n",
       "82336                                                                    \n",
       "\n",
       "      UNIPROT_ISOFORM SOURCE GENE_PHENO SIFT PolyPhen DOMAINS miRNA  \\\n",
       "0                                                                     \n",
       "0                                                                     \n",
       "0                                                                     \n",
       "0                                                                     \n",
       "0                                                                     \n",
       "...               ...    ...        ...  ...      ...     ...   ...   \n",
       "82336                                                                 \n",
       "82336                                                                 \n",
       "82336                                                                 \n",
       "82336                                                                 \n",
       "82336                                                                 \n",
       "\n",
       "      HGVS_OFFSET AF AFR_AF AMR_AF EAS_AF EUR_AF SAS_AF gnomADe_AF  \\\n",
       "0                                                                    \n",
       "0                                                                    \n",
       "0                                                                    \n",
       "0                                                                    \n",
       "0                                                                    \n",
       "...           ... ..    ...    ...    ...    ...    ...        ...   \n",
       "82336                                                                \n",
       "82336                                                                \n",
       "82336                                                                \n",
       "82336                                                                \n",
       "82336                                                                \n",
       "\n",
       "      gnomADe_AFR_AF gnomADe_AMR_AF gnomADe_ASJ_AF gnomADe_EAS_AF  \\\n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "0                                                                   \n",
       "...              ...            ...            ...            ...   \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "\n",
       "      gnomADe_FIN_AF gnomADe_NFE_AF gnomADe_OTH_AF gnomADe_SAS_AF gnomADg_AF  \\\n",
       "0                                                                     0.1311   \n",
       "0                                                                     0.1311   \n",
       "0                                                                     0.1311   \n",
       "0                                                                     0.1311   \n",
       "0                                                                     0.1311   \n",
       "...              ...            ...            ...            ...        ...   \n",
       "82336                                                                          \n",
       "82336                                                                          \n",
       "82336                                                                          \n",
       "82336                                                                          \n",
       "82336                                                                          \n",
       "\n",
       "      gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF  \\\n",
       "0            0.09045         0.1524         0.1382         0.2095   \n",
       "0            0.09045         0.1524         0.1382         0.2095   \n",
       "0            0.09045         0.1524         0.1382         0.2095   \n",
       "0            0.09045         0.1524         0.1382         0.2095   \n",
       "0            0.09045         0.1524         0.1382         0.2095   \n",
       "...              ...            ...            ...            ...   \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "\n",
       "      gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF  \\\n",
       "0            0.06325         0.1167        0.05789         0.1594   \n",
       "0            0.06325         0.1167        0.05789         0.1594   \n",
       "0            0.06325         0.1167        0.05789         0.1594   \n",
       "0            0.06325         0.1167        0.05789         0.1594   \n",
       "0            0.06325         0.1167        0.05789         0.1594   \n",
       "...              ...            ...            ...            ...   \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "82336                                                               \n",
       "\n",
       "      gnomADg_OTH_AF gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS CLIN_SIG SOMATIC  \\\n",
       "0             0.1394         0.1409  0.2095  gnomADg_ASJ                    \n",
       "0             0.1394         0.1409  0.2095  gnomADg_ASJ                    \n",
       "0             0.1394         0.1409  0.2095  gnomADg_ASJ                    \n",
       "0             0.1394         0.1409  0.2095  gnomADg_ASJ                    \n",
       "0             0.1394         0.1409  0.2095  gnomADg_ASJ                    \n",
       "...              ...            ...     ...          ...      ...     ...   \n",
       "82336                                                                       \n",
       "82336                                                                       \n",
       "82336                                                                       \n",
       "82336                                                                       \n",
       "82336                                                                       \n",
       "\n",
       "      PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE  \\\n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "0                                                                         \n",
       "...     ...    ...        ...       ...          ...                ...   \n",
       "82336                                                                     \n",
       "82336                                                                     \n",
       "82336                                                                     \n",
       "82336                                                                     \n",
       "82336                                                                     \n",
       "\n",
       "      TRANSCRIPTION_FACTORS ClinVar ClinVar_CLNSIG ClinVar_CLNREVSTAT  \\\n",
       "0                                                                       \n",
       "0                                                                       \n",
       "0                                                                       \n",
       "0                                                                       \n",
       "0                                                                       \n",
       "...                     ...     ...            ...                ...   \n",
       "82336                                                                   \n",
       "82336                                                                   \n",
       "82336                                                                   \n",
       "82336                                                                   \n",
       "82336                                                                   \n",
       "\n",
       "      ClinVar_CLNDN  \n",
       "0                    \n",
       "0                    \n",
       "0                    \n",
       "0                    \n",
       "0                    \n",
       "...             ...  \n",
       "82336                \n",
       "82336                \n",
       "82336                \n",
       "82336                \n",
       "82336                \n",
       "\n",
       "[652066 rows x 108 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[['Allele', 'Consequence', 'IMPACT', 'SYMBOL', 'Gene', 'Feature_type', 'Feature', 'BIOTYPE', 'EXON', 'INTRON', 'HGVSc', 'HGVSp', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL_SOURCE', 'HGNC_ID', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'SOURCE', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'DOMAINS', 'miRNA', 'HGVS_OFFSET', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE', 'TRANSCRIPTION_FACTORS', 'ClinVar', 'ClinVar_CLNSIG', 'ClinVar_CLNREVSTAT', 'ClinVar_CLNDN']] = vcf['csq'].str.split('|', expand=True)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50037354",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "43c28991",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TNIP1</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg62Gln</td>\n",
       "      <td>CARD14:p.Arg62Gln</td>\n",
       "      <td>CARD14:p.Arg62Gln</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg69Trp</td>\n",
       "      <td>CARD14:p.Arg69Trp</td>\n",
       "      <td>CARD14:p.Arg69Trp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs281875215</td>\n",
       "      <td>rs281875215</td>\n",
       "      <td>rs281875215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Met119Val</td>\n",
       "      <td>CARD14:p.Met119Val</td>\n",
       "      <td>CARD14:p.Met119Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Leu124Pro</td>\n",
       "      <td>CARD14:p.Leu124Pro</td>\n",
       "      <td>CARD14:p.Leu124Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Glu138Ala</td>\n",
       "      <td>CARD14:p.Glu138Ala</td>\n",
       "      <td>CARD14:p.Glu138Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Glu138Lys</td>\n",
       "      <td>CARD14:p.Glu138Lys</td>\n",
       "      <td>CARD14:p.Glu138Lys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Glu138del</td>\n",
       "      <td>CARD14:p.Glu138del</td>\n",
       "      <td>CARD14:p.Glu138del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Glu142Lys</td>\n",
       "      <td>CARD14:p.Glu142Lys</td>\n",
       "      <td>CARD14:p.Glu142Lys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Glu142Gly</td>\n",
       "      <td>CARD14:p.Glu142Gly</td>\n",
       "      <td>CARD14:p.Glu142Gly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Leu150Arg</td>\n",
       "      <td>CARD14:p.Leu150Arg</td>\n",
       "      <td>CARD14:p.Leu150Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg151Gln</td>\n",
       "      <td>CARD14:p.Arg151Gln</td>\n",
       "      <td>CARD14:p.Arg151Gln</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg151Trp</td>\n",
       "      <td>CARD14:p.Arg151Trp</td>\n",
       "      <td>CARD14:p.Arg151Trp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Leu156Pro</td>\n",
       "      <td>CARD14:p.Leu156Pro</td>\n",
       "      <td>CARD14:p.Leu156Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg166His</td>\n",
       "      <td>CARD14:p.Arg166His</td>\n",
       "      <td>CARD14:p.Arg166His</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.His171Asn</td>\n",
       "      <td>CARD14:p.His171Asn</td>\n",
       "      <td>CARD14:p.His171Asn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Asp176His</td>\n",
       "      <td>CARD14:p.Asp176His</td>\n",
       "      <td>CARD14:p.Asp176His</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg179His</td>\n",
       "      <td>CARD14:p.Arg179His</td>\n",
       "      <td>CARD14:p.Arg179His</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Val191Leu</td>\n",
       "      <td>CARD14:p.Val191Leu</td>\n",
       "      <td>CARD14:p.Val191Leu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Glu197Lys</td>\n",
       "      <td>CARD14:p.Glu197Lys</td>\n",
       "      <td>CARD14:p.Glu197Lys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Ser200Asn</td>\n",
       "      <td>CARD14:p.Ser200Asn</td>\n",
       "      <td>CARD14:p.Ser200Asn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Leu209Pro</td>\n",
       "      <td>CARD14:p.Leu209Pro</td>\n",
       "      <td>CARD14:p.Leu209Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Ala216Thr</td>\n",
       "      <td>CARD14:p.Ala216Thr</td>\n",
       "      <td>CARD14:p.Ala216Thr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Asp285Gly</td>\n",
       "      <td>CARD14:p.Asp285Gly</td>\n",
       "      <td>CARD14:p.Asp285Gly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Met338Val</td>\n",
       "      <td>CARD14:p.Met338Val</td>\n",
       "      <td>CARD14:p.Met338Val</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Thr420Ala</td>\n",
       "      <td>CARD14:p.Thr420Ala</td>\n",
       "      <td>CARD14:p.Thr420Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Thr591Met</td>\n",
       "      <td>CARD14:p.Thr591Met</td>\n",
       "      <td>CARD14:p.Thr591Met</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Ile593</td>\n",
       "      <td>CARD14:p.Ile593</td>\n",
       "      <td>CARD14:p.Ile593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Ser602Leu</td>\n",
       "      <td>CARD14:p.Ser602Leu</td>\n",
       "      <td>CARD14:p.Ser602Leu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg682Trp</td>\n",
       "      <td>CARD14:p.Arg682Trp</td>\n",
       "      <td>CARD14:p.Arg682Trp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Gly714Ser</td>\n",
       "      <td>CARD14:p.Gly714Ser</td>\n",
       "      <td>CARD14:p.Gly714Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg820Trp</td>\n",
       "      <td>CARD14:p.Arg820Trp</td>\n",
       "      <td>CARD14:p.Arg820Trp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Asp973Glu</td>\n",
       "      <td>CARD14:p.Asp973Glu</td>\n",
       "      <td>CARD14:p.Asp973Glu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>c.412G&gt;A</td>\n",
       "      <td>CARD14:c.412G&gt;A</td>\n",
       "      <td>CARD14:c.412G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs281875214</td>\n",
       "      <td>rs281875214</td>\n",
       "      <td>rs281875214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>c.412_414delGAG</td>\n",
       "      <td>CARD14:c.412_414delGAG</td>\n",
       "      <td>CARD14:c.412_414delGAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs281875212</td>\n",
       "      <td>rs281875212</td>\n",
       "      <td>rs281875212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs281875213</td>\n",
       "      <td>rs281875213</td>\n",
       "      <td>rs281875213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs387907240</td>\n",
       "      <td>rs387907240</td>\n",
       "      <td>rs387907240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs281875216</td>\n",
       "      <td>rs281875216</td>\n",
       "      <td>rs281875216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs144475004</td>\n",
       "      <td>rs144475004</td>\n",
       "      <td>rs144475004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene            Change   Uploaded_variant_hg38   Uploaded_variant_hg37\n",
       "0     TNIP1       rs17728338              rs17728338              rs17728338\n",
       "1   TNFAIP3         rs610604                rs610604                rs610604\n",
       "2    CARD14       p.Arg38Cys       CARD14:p.Arg38Cys       CARD14:p.Arg38Cys\n",
       "3    CARD14       p.Arg62Gln       CARD14:p.Arg62Gln       CARD14:p.Arg62Gln\n",
       "4    CARD14       p.Arg69Trp       CARD14:p.Arg69Trp       CARD14:p.Arg69Trp\n",
       "5    CARD14      rs281875215             rs281875215             rs281875215\n",
       "6    CARD14      p.Met119Val      CARD14:p.Met119Val      CARD14:p.Met119Val\n",
       "7    CARD14      p.Leu124Pro      CARD14:p.Leu124Pro      CARD14:p.Leu124Pro\n",
       "8    CARD14      p.Glu138Ala      CARD14:p.Glu138Ala      CARD14:p.Glu138Ala\n",
       "9    CARD14      p.Glu138Lys      CARD14:p.Glu138Lys      CARD14:p.Glu138Lys\n",
       "10   CARD14      p.Glu138del      CARD14:p.Glu138del      CARD14:p.Glu138del\n",
       "11   CARD14      p.Glu142Lys      CARD14:p.Glu142Lys      CARD14:p.Glu142Lys\n",
       "12   CARD14      p.Glu142Gly      CARD14:p.Glu142Gly      CARD14:p.Glu142Gly\n",
       "13   CARD14      p.Leu150Arg      CARD14:p.Leu150Arg      CARD14:p.Leu150Arg\n",
       "14   CARD14      p.Arg151Gln      CARD14:p.Arg151Gln      CARD14:p.Arg151Gln\n",
       "15   CARD14      p.Arg151Trp      CARD14:p.Arg151Trp      CARD14:p.Arg151Trp\n",
       "16   CARD14      p.Leu156Pro      CARD14:p.Leu156Pro      CARD14:p.Leu156Pro\n",
       "17   CARD14      p.Arg166His      CARD14:p.Arg166His      CARD14:p.Arg166His\n",
       "18   CARD14      p.His171Asn      CARD14:p.His171Asn      CARD14:p.His171Asn\n",
       "19   CARD14      p.Asp176His      CARD14:p.Asp176His      CARD14:p.Asp176His\n",
       "20   CARD14      p.Arg179His      CARD14:p.Arg179His      CARD14:p.Arg179His\n",
       "21   CARD14      p.Val191Leu      CARD14:p.Val191Leu      CARD14:p.Val191Leu\n",
       "22   CARD14      p.Glu197Lys      CARD14:p.Glu197Lys      CARD14:p.Glu197Lys\n",
       "23   CARD14      p.Ser200Asn      CARD14:p.Ser200Asn      CARD14:p.Ser200Asn\n",
       "24   CARD14      p.Leu209Pro      CARD14:p.Leu209Pro      CARD14:p.Leu209Pro\n",
       "25   CARD14      p.Ala216Thr      CARD14:p.Ala216Thr      CARD14:p.Ala216Thr\n",
       "26   CARD14      p.Asp285Gly      CARD14:p.Asp285Gly      CARD14:p.Asp285Gly\n",
       "27   CARD14      p.Met338Val      CARD14:p.Met338Val      CARD14:p.Met338Val\n",
       "28   CARD14      p.Thr420Ala      CARD14:p.Thr420Ala      CARD14:p.Thr420Ala\n",
       "29   CARD14      p.Thr591Met      CARD14:p.Thr591Met      CARD14:p.Thr591Met\n",
       "30   CARD14         p.Ile593         CARD14:p.Ile593         CARD14:p.Ile593\n",
       "31   CARD14      p.Ser602Leu      CARD14:p.Ser602Leu      CARD14:p.Ser602Leu\n",
       "32   CARD14      p.Arg682Trp      CARD14:p.Arg682Trp      CARD14:p.Arg682Trp\n",
       "33   CARD14      p.Gly714Ser      CARD14:p.Gly714Ser      CARD14:p.Gly714Ser\n",
       "34   CARD14      p.Arg820Trp      CARD14:p.Arg820Trp      CARD14:p.Arg820Trp\n",
       "35   CARD14      p.Asp973Glu      CARD14:p.Asp973Glu      CARD14:p.Asp973Glu\n",
       "36   CARD14         c.412G>A         CARD14:c.412G>A         CARD14:c.412G>A\n",
       "37   CARD14      rs281875214             rs281875214             rs281875214\n",
       "38   CARD14  c.412_414delGAG  CARD14:c.412_414delGAG  CARD14:c.412_414delGAG\n",
       "39   CARD14      rs281875212             rs281875212             rs281875212\n",
       "40   CARD14      rs281875213             rs281875213             rs281875213\n",
       "41   CARD14      rs387907240             rs387907240             rs387907240\n",
       "42   CARD14      rs281875216             rs281875216             rs281875216\n",
       "43   CARD14      rs144475004             rs144475004             rs144475004\n",
       "44   CARD14      rs587777763             rs587777763             rs587777763\n",
       "45   CARD14      rs886041402             rs886041402             rs886041402"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/Psoriasis_Lit_variants.csv', sep = '\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "82fd43fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Chrom-pos-Ref-Alt</th>\n",
       "      <th>rsID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>17-80181550-C-T,</td>\n",
       "      <td>rs281875217,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CARD14:p.Arg62Gln</td>\n",
       "      <td>17-80188471-AG-CA,</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CARD14:p.Arg62Gln</td>\n",
       "      <td>17-80181623-G-A,</td>\n",
       "      <td>rs115582620,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CARD14:p.Arg69Trp</td>\n",
       "      <td>17-80192583-AGA-TGG,</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CARD14:p.Arg69Trp</td>\n",
       "      <td>17-80181643-C-T,</td>\n",
       "      <td>rs375624435,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>rs281875216</td>\n",
       "      <td>17-80184074-C-T,</td>\n",
       "      <td>rs281875216,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>rs144475004</td>\n",
       "      <td>17-80184089-G-C,</td>\n",
       "      <td>rs144475004,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-C,</td>\n",
       "      <td>rs587777763,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-A,</td>\n",
       "      <td>rs587777763,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>rs886041402</td>\n",
       "      <td>17-80182791-G-A,</td>\n",
       "      <td>rs886041402,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Uploaded_variant_hg38      Chrom-pos-Ref-Alt           rsID\n",
       "0      CARD14:p.Arg38Cys      17-80181550-C-T,   rs281875217, \n",
       "1      CARD14:p.Arg62Gln    17-80188471-AG-CA,               -\n",
       "2      CARD14:p.Arg62Gln      17-80181623-G-A,   rs115582620, \n",
       "3      CARD14:p.Arg69Trp  17-80192583-AGA-TGG,               -\n",
       "4      CARD14:p.Arg69Trp      17-80181643-C-T,   rs375624435, \n",
       "..                   ...                    ...            ...\n",
       "57           rs281875216      17-80184074-C-T,   rs281875216, \n",
       "58           rs144475004      17-80184089-G-C,   rs144475004, \n",
       "59           rs587777763      17-80182795-G-C,   rs587777763, \n",
       "60           rs587777763      17-80182795-G-A,   rs587777763, \n",
       "61           rs886041402      17-80182791-G-A,   rs886041402, \n",
       "\n",
       "[62 rows x 3 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/hg38_results_psoriasis.tsv', sep = '\\t')\n",
    "data = data.rename({'Uploaded_variant': 'Uploaded_variant_hg38'}, axis=1)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "5f1a8689",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "      <th>Chrom-pos-Ref-Alt</th>\n",
       "      <th>rsID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TNIP1</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>5-151098757-G-A</td>\n",
       "      <td>rs17728338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-T</td>\n",
       "      <td>rs610604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-A</td>\n",
       "      <td>rs610604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-C</td>\n",
       "      <td>rs610604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>17-80181550-C-T,</td>\n",
       "      <td>rs281875217,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs281875216</td>\n",
       "      <td>rs281875216</td>\n",
       "      <td>rs281875216</td>\n",
       "      <td>17-80184074-C-T,</td>\n",
       "      <td>rs281875216,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs144475004</td>\n",
       "      <td>rs144475004</td>\n",
       "      <td>rs144475004</td>\n",
       "      <td>17-80184089-G-C,</td>\n",
       "      <td>rs144475004,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-C,</td>\n",
       "      <td>rs587777763,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-A,</td>\n",
       "      <td>rs587777763,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>17-80182791-G-A,</td>\n",
       "      <td>rs886041402,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene        Change Uploaded_variant_hg38 Uploaded_variant_hg37  \\\n",
       "0     TNIP1   rs17728338            rs17728338            rs17728338   \n",
       "1   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "2   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "3   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "4    CARD14   p.Arg38Cys     CARD14:p.Arg38Cys     CARD14:p.Arg38Cys   \n",
       "..      ...          ...                   ...                   ...   \n",
       "57   CARD14  rs281875216           rs281875216           rs281875216   \n",
       "58   CARD14  rs144475004           rs144475004           rs144475004   \n",
       "59   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "60   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "61   CARD14  rs886041402           rs886041402           rs886041402   \n",
       "\n",
       "    Chrom-pos-Ref-Alt           rsID  \n",
       "0     5-151098757-G-A     rs17728338  \n",
       "1     6-137878280-G-T       rs610604  \n",
       "2     6-137878280-G-A       rs610604  \n",
       "3     6-137878280-G-C       rs610604  \n",
       "4   17-80181550-C-T,   rs281875217,   \n",
       "..                ...            ...  \n",
       "57  17-80184074-C-T,   rs281875216,   \n",
       "58  17-80184089-G-C,   rs144475004,   \n",
       "59  17-80182795-G-C,   rs587777763,   \n",
       "60  17-80182795-G-A,   rs587777763,   \n",
       "61  17-80182791-G-A,   rs886041402,   \n",
       "\n",
       "[62 rows x 6 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.merge(df, data, on = 'Uploaded_variant_hg38', how = 'left')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "c578c3d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "      <th>Chrom-pos-Ref-Alt</th>\n",
       "      <th>rsID</th>\n",
       "      <th>chrom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TNIP1</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>5-151098757-G-A</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>5-151098757-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-T</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-A</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-C</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>17-80181550-C-T,</td>\n",
       "      <td>rs281875217,</td>\n",
       "      <td>17-80181550-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-C,</td>\n",
       "      <td>rs587777763,</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-A,</td>\n",
       "      <td>rs587777763,</td>\n",
       "      <td>17-80182795-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-A,</td>\n",
       "      <td>rs587777763,</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>17-80182791-G-A,</td>\n",
       "      <td>rs886041402,</td>\n",
       "      <td>17-80182791-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>17-80182791-G-A,</td>\n",
       "      <td>rs886041402,</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>163 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene        Change Uploaded_variant_hg38 Uploaded_variant_hg37  \\\n",
       "0     TNIP1   rs17728338            rs17728338            rs17728338   \n",
       "1   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "2   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "3   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "4    CARD14   p.Arg38Cys     CARD14:p.Arg38Cys     CARD14:p.Arg38Cys   \n",
       "..      ...          ...                   ...                   ...   \n",
       "59   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "60   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "60   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "61   CARD14  rs886041402           rs886041402           rs886041402   \n",
       "61   CARD14  rs886041402           rs886041402           rs886041402   \n",
       "\n",
       "    Chrom-pos-Ref-Alt           rsID            chrom  \n",
       "0     5-151098757-G-A     rs17728338  5-151098757-G-A  \n",
       "1     6-137878280-G-T       rs610604  6-137878280-G-T  \n",
       "2     6-137878280-G-A       rs610604  6-137878280-G-A  \n",
       "3     6-137878280-G-C       rs610604  6-137878280-G-C  \n",
       "4   17-80181550-C-T,   rs281875217,   17-80181550-C-T  \n",
       "..                ...            ...              ...  \n",
       "59  17-80182795-G-C,   rs587777763,                    \n",
       "60  17-80182795-G-A,   rs587777763,   17-80182795-G-A  \n",
       "60  17-80182795-G-A,   rs587777763,                    \n",
       "61  17-80182791-G-A,   rs886041402,   17-80182791-G-A  \n",
       "61  17-80182791-G-A,   rs886041402,                    \n",
       "\n",
       "[163 rows x 7 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged['chrom'] = merged['Chrom-pos-Ref-Alt'].str.split(',')\n",
    "merged = merged.explode('chrom')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "e287dbea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "      <th>Chrom-pos-Ref-Alt</th>\n",
       "      <th>rsID</th>\n",
       "      <th>chrom</th>\n",
       "      <th>POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TNIP1</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>5-151098757-G-A</td>\n",
       "      <td>rs17728338</td>\n",
       "      <td>5-151098757-G-A</td>\n",
       "      <td>151098757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-T</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-T</td>\n",
       "      <td>137878280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-A</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-A</td>\n",
       "      <td>137878280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-C</td>\n",
       "      <td>rs610604</td>\n",
       "      <td>6-137878280-G-C</td>\n",
       "      <td>137878280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>CARD14:p.Arg38Cys</td>\n",
       "      <td>17-80181550-C-T,</td>\n",
       "      <td>rs281875217,</td>\n",
       "      <td>17-80181550-C-T</td>\n",
       "      <td>80181550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-C,</td>\n",
       "      <td>rs587777763,</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-A,</td>\n",
       "      <td>rs587777763,</td>\n",
       "      <td>17-80182795-G-A</td>\n",
       "      <td>80182795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>rs587777763</td>\n",
       "      <td>17-80182795-G-A,</td>\n",
       "      <td>rs587777763,</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>17-80182791-G-A,</td>\n",
       "      <td>rs886041402,</td>\n",
       "      <td>17-80182791-G-A</td>\n",
       "      <td>80182791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CARD14</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>rs886041402</td>\n",
       "      <td>17-80182791-G-A,</td>\n",
       "      <td>rs886041402,</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>163 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene        Change Uploaded_variant_hg38 Uploaded_variant_hg37  \\\n",
       "0     TNIP1   rs17728338            rs17728338            rs17728338   \n",
       "1   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "2   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "3   TNFAIP3     rs610604              rs610604              rs610604   \n",
       "4    CARD14   p.Arg38Cys     CARD14:p.Arg38Cys     CARD14:p.Arg38Cys   \n",
       "..      ...          ...                   ...                   ...   \n",
       "59   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "60   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "60   CARD14  rs587777763           rs587777763           rs587777763   \n",
       "61   CARD14  rs886041402           rs886041402           rs886041402   \n",
       "61   CARD14  rs886041402           rs886041402           rs886041402   \n",
       "\n",
       "    Chrom-pos-Ref-Alt           rsID            chrom        POS  \n",
       "0     5-151098757-G-A     rs17728338  5-151098757-G-A  151098757  \n",
       "1     6-137878280-G-T       rs610604  6-137878280-G-T  137878280  \n",
       "2     6-137878280-G-A       rs610604  6-137878280-G-A  137878280  \n",
       "3     6-137878280-G-C       rs610604  6-137878280-G-C  137878280  \n",
       "4   17-80181550-C-T,   rs281875217,   17-80181550-C-T   80181550  \n",
       "..                ...            ...              ...        ...  \n",
       "59  17-80182795-G-C,   rs587777763,                          NaN  \n",
       "60  17-80182795-G-A,   rs587777763,   17-80182795-G-A   80182795  \n",
       "60  17-80182795-G-A,   rs587777763,                          NaN  \n",
       "61  17-80182791-G-A,   rs886041402,   17-80182791-G-A   80182791  \n",
       "61  17-80182791-G-A,   rs886041402,                          NaN  \n",
       "\n",
       "[163 rows x 8 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged['POS'] = merged['chrom'].str.split('-').str[1]\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "15a62e59",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene                     0\n",
       "Change                   0\n",
       "Uploaded_variant_hg38    0\n",
       "Uploaded_variant_hg37    0\n",
       "Chrom-pos-Ref-Alt        2\n",
       "rsID                     4\n",
       "dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "36a56f45",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/psy.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b068c2a7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07a4fb55",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "e78a0143",
   "metadata": {},
   "source": [
    "# POS and CHROM extraction from the condition final sheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "d25511dc",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>100407579</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>100407852</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>100406643</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>166046970</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>165998075</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>chr6</td>\n",
       "      <td>145627532</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>chr6</td>\n",
       "      <td>145640407</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>chr6</td>\n",
       "      <td>145635288</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>chr6</td>\n",
       "      <td>145627510</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>chr6</td>\n",
       "      <td>145631422</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>742 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS Literature\n",
       "0    chrX  100407579        Yes\n",
       "2    chrX  100407852        Yes\n",
       "3    chrX  100406643        Yes\n",
       "6    chr2  166046970        Yes\n",
       "7    chr2  165998075        Yes\n",
       "..    ...        ...        ...\n",
       "299  chr6  145627532        Yes\n",
       "299  chr6  145640407        Yes\n",
       "300  chr6  145635288        Yes\n",
       "300  chr6  145627510        Yes\n",
       "300  chr6  145631422        Yes\n",
       "\n",
       "[742 rows x 3 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Epilepsy_Lit_final_Positions_hg38_hg37 (1).xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[0]\n",
    "\n",
    "# Function to add 'chr' prefix conditionally\n",
    "def add_chr_prefix(chromosome):\n",
    "    if pd.notnull(chromosome) and chromosome.strip() != '':\n",
    "        return 'chr' + str(chromosome)\n",
    "    else:\n",
    "        return chromosome\n",
    "\n",
    "# Applying the function to the 'chromosome' column\n",
    "x['CHROM'] = x['CHROM'].apply(add_chr_prefix)\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['CHROM'] = x['CHROM'].str.replace(r'\\s+', '')\n",
    "x['POS'] = x['chrom'].str.split('-').str[1]\n",
    "\n",
    "x.dropna(subset=['CHROM'], inplace=True)\n",
    "# Drop rows with empty cells after removing leading and trailing whitespaces\n",
    "x['CHROM'] = x['CHROM'].str.strip()\n",
    "x['POS'] = x['POS'].str.strip()\n",
    "# Dropping rows with empty cells and NaN values in both 'chromosome' and 'position' columns\n",
    "x.dropna(subset=['CHROM', 'POS'], inplace=True)\n",
    "x = x[['CHROM', 'POS']]\n",
    "x['Literature'] = 'Yes'\n",
    "x.drop_duplicates(subset='POS', inplace=True)\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "4b2010fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "x.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Condition_pos/Epilepsy_pos_new.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1b3f07e",
   "metadata": {},
   "source": [
    "# ---------------------------------------------------------------------------------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83bcc886",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d3c90d2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "      <th>Chrom-pos-Ref-Alt_38</th>\n",
       "      <th>rsID_hg38</th>\n",
       "      <th>Chrom-pos-Ref-Alt_37</th>\n",
       "      <th>rsID_hg37</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COMT</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>22-19963748-G-A,</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>22-19951271-G-A</td>\n",
       "      <td>rs4680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>19-13228314-G-A, 19-13228314-G-C,</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>19-13339128-G-C, 19-13339128-G-A,</td>\n",
       "      <td>rs10405121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>c.653C &gt; T</td>\n",
       "      <td>CACNA1A:c.653C&gt;T</td>\n",
       "      <td>CACNA1A:c.653C&gt;T</td>\n",
       "      <td>19-13365448-G-A, 19-13214557-G-A, 19-13230151-...</td>\n",
       "      <td>rs121908225, rs1483005213, rs2056673232</td>\n",
       "      <td>19-13476262-G-A, 19-13366069-G-A,</td>\n",
       "      <td>rs121908225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>c.1088T &gt; C</td>\n",
       "      <td>CACNA1A:c.1088T&gt;C</td>\n",
       "      <td>CACNA1A:c.1088T&gt;C</td>\n",
       "      <td>19-13334488-A-G, 19-13214246-A-G, 19-13359691-...</td>\n",
       "      <td>rs959681605, rs1291528882</td>\n",
       "      <td>19-13445302-A-G, 19-13346465-A-G,</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>c.4037G &gt; A</td>\n",
       "      <td>CACNA1A:c.4037G&gt;A</td>\n",
       "      <td>CACNA1A:c.4037G&gt;A</td>\n",
       "      <td>19-13262789-C-T,</td>\n",
       "      <td>rs121908230</td>\n",
       "      <td>19-13373603-C-T,</td>\n",
       "      <td>rs121908230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>KCNK18</td>\n",
       "      <td>p.Phe139Trp</td>\n",
       "      <td>KCNK18:p.Phe139Trp</td>\n",
       "      <td>KCNK18:p.Phe139Trp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.649dupC</td>\n",
       "      <td>PRRT2:c.649dupC</td>\n",
       "      <td>PRRT2:c.649dupC</td>\n",
       "      <td>16-29814394-A-AA, 16-29813703-C-CC, 16-2981481...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16-29825024-C-CC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.649delC</td>\n",
       "      <td>PRRT2:c.649delC</td>\n",
       "      <td>PRRT2:c.649delC</td>\n",
       "      <td>16-29813702-CC-C, 16-29814393-AA-A, 16-2981481...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16-29825023-CC-C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.843C&gt;G</td>\n",
       "      <td>PRRT2:c.843C&gt;G</td>\n",
       "      <td>PRRT2:c.843C&gt;G</td>\n",
       "      <td>16-29815009-C-G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.649_650insC</td>\n",
       "      <td>PRRT2:c.649_650insC</td>\n",
       "      <td>PRRT2:c.649_650insC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene         Change Uploaded_variant_hg38 Uploaded_variant_hg37  \\\n",
       "0      COMT         rs4680                rs4680                rs4680   \n",
       "1   CACNA1A     rs10405121            rs10405121            rs10405121   \n",
       "2   CACNA1A     c.653C > T      CACNA1A:c.653C>T      CACNA1A:c.653C>T   \n",
       "3   CACNA1A    c.1088T > C     CACNA1A:c.1088T>C     CACNA1A:c.1088T>C   \n",
       "4   CACNA1A    c.4037G > A     CACNA1A:c.4037G>A     CACNA1A:c.4037G>A   \n",
       "..      ...            ...                   ...                   ...   \n",
       "62   KCNK18    p.Phe139Trp    KCNK18:p.Phe139Trp    KCNK18:p.Phe139Trp   \n",
       "63    PRRT2      c.649dupC       PRRT2:c.649dupC       PRRT2:c.649dupC   \n",
       "64    PRRT2      c.649delC       PRRT2:c.649delC       PRRT2:c.649delC   \n",
       "65    PRRT2       c.843C>G        PRRT2:c.843C>G        PRRT2:c.843C>G   \n",
       "66    PRRT2  c.649_650insC   PRRT2:c.649_650insC   PRRT2:c.649_650insC   \n",
       "\n",
       "                                 Chrom-pos-Ref-Alt_38  \\\n",
       "0                                   22-19963748-G-A,    \n",
       "1                  19-13228314-G-A, 19-13228314-G-C,    \n",
       "2   19-13365448-G-A, 19-13214557-G-A, 19-13230151-...   \n",
       "3   19-13334488-A-G, 19-13214246-A-G, 19-13359691-...   \n",
       "4                                   19-13262789-C-T,    \n",
       "..                                                ...   \n",
       "62                                                NaN   \n",
       "63  16-29814394-A-AA, 16-29813703-C-CC, 16-2981481...   \n",
       "64  16-29813702-CC-C, 16-29814393-AA-A, 16-2981481...   \n",
       "65                                    16-29815009-C-G   \n",
       "66                                                NaN   \n",
       "\n",
       "                                  rsID_hg38  \\\n",
       "0                                    rs4680   \n",
       "1                                rs10405121   \n",
       "2   rs121908225, rs1483005213, rs2056673232   \n",
       "3                 rs959681605, rs1291528882   \n",
       "4                               rs121908230   \n",
       "..                                      ...   \n",
       "62                                      NaN   \n",
       "63                                      NaN   \n",
       "64                                      NaN   \n",
       "65                                      NaN   \n",
       "66                                      NaN   \n",
       "\n",
       "                  Chrom-pos-Ref-Alt_37    rsID_hg37  \n",
       "0                      22-19951271-G-A       rs4680  \n",
       "1   19-13339128-G-C, 19-13339128-G-A,    rs10405121  \n",
       "2   19-13476262-G-A, 19-13366069-G-A,   rs121908225  \n",
       "3   19-13445302-A-G, 19-13346465-A-G,           NaN  \n",
       "4                    19-13373603-C-T,   rs121908230  \n",
       "..                                 ...          ...  \n",
       "62                                 NaN          NaN  \n",
       "63                    16-29825024-C-CC          NaN  \n",
       "64                    16-29825023-CC-C          NaN  \n",
       "65                                 NaN          NaN  \n",
       "66                                 NaN          NaN  \n",
       "\n",
       "[67 rows x 8 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Migranes_Lit_final_Positions_hg38_hg37.xlsx')\n",
    "y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bdb60863",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "      <th>Chrom-pos-Ref-Alt_38</th>\n",
       "      <th>rsID_hg38</th>\n",
       "      <th>Chrom-pos-Ref-Alt_37</th>\n",
       "      <th>rsID_hg37</th>\n",
       "      <th>CHROM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COMT</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>22-19963748-G-A,</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>22-19951271-G-A</td>\n",
       "      <td>rs4680</td>\n",
       "      <td>19963748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>19-13228314-G-A, 19-13228314-G-C,</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>19-13339128-G-C, 19-13339128-G-A,</td>\n",
       "      <td>rs10405121</td>\n",
       "      <td>13228314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>c.653C &gt; T</td>\n",
       "      <td>CACNA1A:c.653C&gt;T</td>\n",
       "      <td>CACNA1A:c.653C&gt;T</td>\n",
       "      <td>19-13365448-G-A, 19-13214557-G-A, 19-13230151-...</td>\n",
       "      <td>rs121908225, rs1483005213, rs2056673232</td>\n",
       "      <td>19-13476262-G-A, 19-13366069-G-A,</td>\n",
       "      <td>rs121908225</td>\n",
       "      <td>13365448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>c.1088T &gt; C</td>\n",
       "      <td>CACNA1A:c.1088T&gt;C</td>\n",
       "      <td>CACNA1A:c.1088T&gt;C</td>\n",
       "      <td>19-13334488-A-G, 19-13214246-A-G, 19-13359691-...</td>\n",
       "      <td>rs959681605, rs1291528882</td>\n",
       "      <td>19-13445302-A-G, 19-13346465-A-G,</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13334488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>c.4037G &gt; A</td>\n",
       "      <td>CACNA1A:c.4037G&gt;A</td>\n",
       "      <td>CACNA1A:c.4037G&gt;A</td>\n",
       "      <td>19-13262789-C-T,</td>\n",
       "      <td>rs121908230</td>\n",
       "      <td>19-13373603-C-T,</td>\n",
       "      <td>rs121908230</td>\n",
       "      <td>13262789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>KCNK18</td>\n",
       "      <td>p.Phe139Trp</td>\n",
       "      <td>KCNK18:p.Phe139Trp</td>\n",
       "      <td>KCNK18:p.Phe139Trp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.649dupC</td>\n",
       "      <td>PRRT2:c.649dupC</td>\n",
       "      <td>PRRT2:c.649dupC</td>\n",
       "      <td>16-29814394-A-AA, 16-29813703-C-CC, 16-2981481...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16-29825024-C-CC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29814394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.649delC</td>\n",
       "      <td>PRRT2:c.649delC</td>\n",
       "      <td>PRRT2:c.649delC</td>\n",
       "      <td>16-29813702-CC-C, 16-29814393-AA-A, 16-2981481...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16-29825023-CC-C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29813702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.843C&gt;G</td>\n",
       "      <td>PRRT2:c.843C&gt;G</td>\n",
       "      <td>PRRT2:c.843C&gt;G</td>\n",
       "      <td>16-29815009-C-G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29815009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>PRRT2</td>\n",
       "      <td>c.649_650insC</td>\n",
       "      <td>PRRT2:c.649_650insC</td>\n",
       "      <td>PRRT2:c.649_650insC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene         Change Uploaded_variant_hg38 Uploaded_variant_hg37  \\\n",
       "0      COMT         rs4680                rs4680                rs4680   \n",
       "1   CACNA1A     rs10405121            rs10405121            rs10405121   \n",
       "2   CACNA1A     c.653C > T      CACNA1A:c.653C>T      CACNA1A:c.653C>T   \n",
       "3   CACNA1A    c.1088T > C     CACNA1A:c.1088T>C     CACNA1A:c.1088T>C   \n",
       "4   CACNA1A    c.4037G > A     CACNA1A:c.4037G>A     CACNA1A:c.4037G>A   \n",
       "..      ...            ...                   ...                   ...   \n",
       "62   KCNK18    p.Phe139Trp    KCNK18:p.Phe139Trp    KCNK18:p.Phe139Trp   \n",
       "63    PRRT2      c.649dupC       PRRT2:c.649dupC       PRRT2:c.649dupC   \n",
       "64    PRRT2      c.649delC       PRRT2:c.649delC       PRRT2:c.649delC   \n",
       "65    PRRT2       c.843C>G        PRRT2:c.843C>G        PRRT2:c.843C>G   \n",
       "66    PRRT2  c.649_650insC   PRRT2:c.649_650insC   PRRT2:c.649_650insC   \n",
       "\n",
       "                                 Chrom-pos-Ref-Alt_38  \\\n",
       "0                                   22-19963748-G-A,    \n",
       "1                  19-13228314-G-A, 19-13228314-G-C,    \n",
       "2   19-13365448-G-A, 19-13214557-G-A, 19-13230151-...   \n",
       "3   19-13334488-A-G, 19-13214246-A-G, 19-13359691-...   \n",
       "4                                   19-13262789-C-T,    \n",
       "..                                                ...   \n",
       "62                                                NaN   \n",
       "63  16-29814394-A-AA, 16-29813703-C-CC, 16-2981481...   \n",
       "64  16-29813702-CC-C, 16-29814393-AA-A, 16-2981481...   \n",
       "65                                    16-29815009-C-G   \n",
       "66                                                NaN   \n",
       "\n",
       "                                  rsID_hg38  \\\n",
       "0                                    rs4680   \n",
       "1                                rs10405121   \n",
       "2   rs121908225, rs1483005213, rs2056673232   \n",
       "3                 rs959681605, rs1291528882   \n",
       "4                               rs121908230   \n",
       "..                                      ...   \n",
       "62                                      NaN   \n",
       "63                                      NaN   \n",
       "64                                      NaN   \n",
       "65                                      NaN   \n",
       "66                                      NaN   \n",
       "\n",
       "                  Chrom-pos-Ref-Alt_37    rsID_hg37     CHROM  \n",
       "0                      22-19951271-G-A       rs4680  19963748  \n",
       "1   19-13339128-G-C, 19-13339128-G-A,    rs10405121  13228314  \n",
       "2   19-13476262-G-A, 19-13366069-G-A,   rs121908225  13365448  \n",
       "3   19-13445302-A-G, 19-13346465-A-G,           NaN  13334488  \n",
       "4                    19-13373603-C-T,   rs121908230  13262789  \n",
       "..                                 ...          ...       ...  \n",
       "62                                 NaN          NaN       NaN  \n",
       "63                    16-29825024-C-CC          NaN  29814394  \n",
       "64                    16-29825023-CC-C          NaN  29813702  \n",
       "65                                 NaN          NaN  29815009  \n",
       "66                                 NaN          NaN       NaN  \n",
       "\n",
       "[67 rows x 9 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y['CHROM'] = y['Chrom-pos-Ref-Alt_38'].str.split('-').str[1]\n",
    "y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "f3b06c8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "y['POS'] = y['Chr-Position-Ref-Alt'].str.split('-').str[1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "d7a3cb27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Amino acid change</th>\n",
       "      <th>Uploaded_variant</th>\n",
       "      <th>Chr-Position-Ref-Alt</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Gene Info</th>\n",
       "      <th>Publication</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G551D</td>\n",
       "      <td>CFTR:p.Gly551Asp</td>\n",
       "      <td>7-117587806-G-A</td>\n",
       "      <td>rs75527207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117587806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.N1303K</td>\n",
       "      <td>CFTR:p.Asn1303Lys</td>\n",
       "      <td>7-117652877-C-G</td>\n",
       "      <td>rs80034486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117652877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R117H</td>\n",
       "      <td>CFTR:p.Arg117His</td>\n",
       "      <td>7-117710365-GA-AT</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117710365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G85E</td>\n",
       "      <td>CFTR:p.Gly85Glu</td>\n",
       "      <td>7-117509123-G-A</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117509123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R334W</td>\n",
       "      <td>CFTR:p.Arg334Trp</td>\n",
       "      <td>7-117540230-C-T</td>\n",
       "      <td>rs121909011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117540230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>TRPV6</td>\n",
       "      <td>p.A626P</td>\n",
       "      <td>TRPV6:p.Ala626Pro</td>\n",
       "      <td>7-142873480-C-G</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Transient receptor potential vanilloid subfami...</td>\n",
       "      <td>PMID: 33610740</td>\n",
       "      <td>7</td>\n",
       "      <td>142873480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>CLDN2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLDN2 gene product belongs to the claudin prot...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>CEL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The enzyme carboxyl ester lipase (CEL), hydrol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>CASR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The CASR allows regulation of parathyroid horm...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>GGT1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGT1 encodes for the membrane-bound enzyme gam...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>214 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Amino acid change   Uploaded_variant Chr-Position-Ref-Alt  \\\n",
       "0     CFTR           p.G551D   CFTR:p.Gly551Asp      7-117587806-G-A   \n",
       "1     CFTR          p.N1303K  CFTR:p.Asn1303Lys      7-117652877-C-G   \n",
       "2     CFTR           p.R117H   CFTR:p.Arg117His    7-117710365-GA-AT   \n",
       "3     CFTR            p.G85E    CFTR:p.Gly85Glu      7-117509123-G-A   \n",
       "4     CFTR           p.R334W   CFTR:p.Arg334Trp      7-117540230-C-T   \n",
       "..     ...               ...                ...                  ...   \n",
       "209  TRPV6           p.A626P  TRPV6:p.Ala626Pro      7-142873480-C-G   \n",
       "210  CLDN2                 -                  -                    -   \n",
       "211    CEL                 -                  -                    -   \n",
       "212   CASR                 -                  -                    -   \n",
       "213   GGT1                 -                  -                    -   \n",
       "\n",
       "            rsID Variant mechanism  \\\n",
       "0     rs75527207               NaN   \n",
       "1     rs80034486               NaN   \n",
       "2     rs78655421               NaN   \n",
       "3     rs75961395               NaN   \n",
       "4    rs121909011               NaN   \n",
       "..           ...               ...   \n",
       "209            -               NaN   \n",
       "210            -               NaN   \n",
       "211            -               NaN   \n",
       "212            -               NaN   \n",
       "213            -               NaN   \n",
       "\n",
       "                                             Gene Info  \\\n",
       "0    Cystic fibrosis transmembrane conductance regu...   \n",
       "1    Cystic fibrosis transmembrane conductance regu...   \n",
       "2    Cystic fibrosis transmembrane conductance regu...   \n",
       "3    Cystic fibrosis transmembrane conductance regu...   \n",
       "4    Cystic fibrosis transmembrane conductance regu...   \n",
       "..                                                 ...   \n",
       "209  Transient receptor potential vanilloid subfami...   \n",
       "210  CLDN2 gene product belongs to the claudin prot...   \n",
       "211  The enzyme carboxyl ester lipase (CEL), hydrol...   \n",
       "212  The CASR allows regulation of parathyroid horm...   \n",
       "213  GGT1 encodes for the membrane-bound enzyme gam...   \n",
       "\n",
       "                                        Publication CHROM        POS  \n",
       "0    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117587806  \n",
       "1    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117652877  \n",
       "2    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117710365  \n",
       "3    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117509123  \n",
       "4    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117540230  \n",
       "..                                              ...   ...        ...  \n",
       "209                                  PMID: 33610740     7  142873480  \n",
       "210                                             NaN                   \n",
       "211                                             NaN                   \n",
       "212                                             NaN                   \n",
       "213                                             NaN                   \n",
       "\n",
       "[214 rows x 10 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "946d0cb3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Amino acid change</th>\n",
       "      <th>Uploaded_variant</th>\n",
       "      <th>Chr-Position-Ref-Alt</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Gene Info</th>\n",
       "      <th>Publication</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF_ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G551D</td>\n",
       "      <td>CFTR:p.Gly551Asp</td>\n",
       "      <td>7-117587806-G-A</td>\n",
       "      <td>rs75527207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117587806</td>\n",
       "      <td>G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.N1303K</td>\n",
       "      <td>CFTR:p.Asn1303Lys</td>\n",
       "      <td>7-117652877-C-G</td>\n",
       "      <td>rs80034486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117652877</td>\n",
       "      <td>C-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R117H</td>\n",
       "      <td>CFTR:p.Arg117His</td>\n",
       "      <td>7-117710365-GA-AT</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117710365</td>\n",
       "      <td>GA-AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G85E</td>\n",
       "      <td>CFTR:p.Gly85Glu</td>\n",
       "      <td>7-117509123-G-A</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R334W</td>\n",
       "      <td>CFTR:p.Arg334Trp</td>\n",
       "      <td>7-117540230-C-T</td>\n",
       "      <td>rs121909011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "      <td>7</td>\n",
       "      <td>117540230</td>\n",
       "      <td>C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>TRPV6</td>\n",
       "      <td>p.A626P</td>\n",
       "      <td>TRPV6:p.Ala626Pro</td>\n",
       "      <td>7-142873480-C-G</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Transient receptor potential vanilloid subfami...</td>\n",
       "      <td>PMID: 33610740</td>\n",
       "      <td>7</td>\n",
       "      <td>142873480</td>\n",
       "      <td>C-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>CLDN2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLDN2 gene product belongs to the claudin prot...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>CEL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The enzyme carboxyl ester lipase (CEL), hydrol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>CASR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The CASR allows regulation of parathyroid horm...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>GGT1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGT1 encodes for the membrane-bound enzyme gam...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>214 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Amino acid change   Uploaded_variant Chr-Position-Ref-Alt  \\\n",
       "0     CFTR           p.G551D   CFTR:p.Gly551Asp      7-117587806-G-A   \n",
       "1     CFTR          p.N1303K  CFTR:p.Asn1303Lys      7-117652877-C-G   \n",
       "2     CFTR           p.R117H   CFTR:p.Arg117His    7-117710365-GA-AT   \n",
       "3     CFTR            p.G85E    CFTR:p.Gly85Glu      7-117509123-G-A   \n",
       "4     CFTR           p.R334W   CFTR:p.Arg334Trp      7-117540230-C-T   \n",
       "..     ...               ...                ...                  ...   \n",
       "209  TRPV6           p.A626P  TRPV6:p.Ala626Pro      7-142873480-C-G   \n",
       "210  CLDN2                 -                  -                    -   \n",
       "211    CEL                 -                  -                    -   \n",
       "212   CASR                 -                  -                    -   \n",
       "213   GGT1                 -                  -                    -   \n",
       "\n",
       "            rsID Variant mechanism  \\\n",
       "0     rs75527207               NaN   \n",
       "1     rs80034486               NaN   \n",
       "2     rs78655421               NaN   \n",
       "3     rs75961395               NaN   \n",
       "4    rs121909011               NaN   \n",
       "..           ...               ...   \n",
       "209            -               NaN   \n",
       "210            -               NaN   \n",
       "211            -               NaN   \n",
       "212            -               NaN   \n",
       "213            -               NaN   \n",
       "\n",
       "                                             Gene Info  \\\n",
       "0    Cystic fibrosis transmembrane conductance regu...   \n",
       "1    Cystic fibrosis transmembrane conductance regu...   \n",
       "2    Cystic fibrosis transmembrane conductance regu...   \n",
       "3    Cystic fibrosis transmembrane conductance regu...   \n",
       "4    Cystic fibrosis transmembrane conductance regu...   \n",
       "..                                                 ...   \n",
       "209  Transient receptor potential vanilloid subfami...   \n",
       "210  CLDN2 gene product belongs to the claudin prot...   \n",
       "211  The enzyme carboxyl ester lipase (CEL), hydrol...   \n",
       "212  The CASR allows regulation of parathyroid horm...   \n",
       "213  GGT1 encodes for the membrane-bound enzyme gam...   \n",
       "\n",
       "                                        Publication CHROM        POS REF_ALT  \n",
       "0    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117587806     G-A  \n",
       "1    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117652877     C-G  \n",
       "2    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117710365   GA-AT  \n",
       "3    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117509123     G-A  \n",
       "4    PMID: 19092437, PMID: 34557648, PMID: 32153386     7  117540230     C-T  \n",
       "..                                              ...   ...        ...     ...  \n",
       "209                                  PMID: 33610740     7  142873480     C-G  \n",
       "210                                             NaN                           \n",
       "211                                             NaN                           \n",
       "212                                             NaN                           \n",
       "213                                             NaN                           \n",
       "\n",
       "[214 rows x 11 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y['REF_ALT'] = y['Chr-Position-Ref-Alt'].str.split('-').str[2:5].str.join('-')\n",
    "y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "1d8445ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene', 'Amino acid change', 'Uploaded_variant', 'Chr-Position-Ref-Alt',\n",
       "       'rsID', 'Variant mechanism', 'Gene Info', 'Publication', 'CHROM', 'POS',\n",
       "       'REF_ALT'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "2a8082ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Amino acid change</th>\n",
       "      <th>Uploaded_variant</th>\n",
       "      <th>Chr-Position-Ref-Alt</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF_ALT</th>\n",
       "      <th>rsID</th>\n",
       "      <th>Variant mechanism</th>\n",
       "      <th>Gene Info</th>\n",
       "      <th>Publication</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G551D</td>\n",
       "      <td>CFTR:p.Gly551Asp</td>\n",
       "      <td>7-117587806-G-A</td>\n",
       "      <td>7</td>\n",
       "      <td>117587806</td>\n",
       "      <td>G-A</td>\n",
       "      <td>rs75527207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.N1303K</td>\n",
       "      <td>CFTR:p.Asn1303Lys</td>\n",
       "      <td>7-117652877-C-G</td>\n",
       "      <td>7</td>\n",
       "      <td>117652877</td>\n",
       "      <td>C-G</td>\n",
       "      <td>rs80034486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R117H</td>\n",
       "      <td>CFTR:p.Arg117His</td>\n",
       "      <td>7-117710365-GA-AT</td>\n",
       "      <td>7</td>\n",
       "      <td>117710365</td>\n",
       "      <td>GA-AT</td>\n",
       "      <td>rs78655421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.G85E</td>\n",
       "      <td>CFTR:p.Gly85Glu</td>\n",
       "      <td>7-117509123-G-A</td>\n",
       "      <td>7</td>\n",
       "      <td>117509123</td>\n",
       "      <td>G-A</td>\n",
       "      <td>rs75961395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>p.R334W</td>\n",
       "      <td>CFTR:p.Arg334Trp</td>\n",
       "      <td>7-117540230-C-T</td>\n",
       "      <td>7</td>\n",
       "      <td>117540230</td>\n",
       "      <td>C-T</td>\n",
       "      <td>rs121909011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cystic fibrosis transmembrane conductance regu...</td>\n",
       "      <td>PMID: 19092437, PMID: 34557648, PMID: 32153386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>TRPV6</td>\n",
       "      <td>p.A626P</td>\n",
       "      <td>TRPV6:p.Ala626Pro</td>\n",
       "      <td>7-142873480-C-G</td>\n",
       "      <td>7</td>\n",
       "      <td>142873480</td>\n",
       "      <td>C-G</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Transient receptor potential vanilloid subfami...</td>\n",
       "      <td>PMID: 33610740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>CLDN2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLDN2 gene product belongs to the claudin prot...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>CEL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The enzyme carboxyl ester lipase (CEL), hydrol...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>CASR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The CASR allows regulation of parathyroid horm...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>GGT1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGT1 encodes for the membrane-bound enzyme gam...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>214 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Amino acid change   Uploaded_variant Chr-Position-Ref-Alt CHROM  \\\n",
       "0     CFTR           p.G551D   CFTR:p.Gly551Asp      7-117587806-G-A     7   \n",
       "1     CFTR          p.N1303K  CFTR:p.Asn1303Lys      7-117652877-C-G     7   \n",
       "2     CFTR           p.R117H   CFTR:p.Arg117His    7-117710365-GA-AT     7   \n",
       "3     CFTR            p.G85E    CFTR:p.Gly85Glu      7-117509123-G-A     7   \n",
       "4     CFTR           p.R334W   CFTR:p.Arg334Trp      7-117540230-C-T     7   \n",
       "..     ...               ...                ...                  ...   ...   \n",
       "209  TRPV6           p.A626P  TRPV6:p.Ala626Pro      7-142873480-C-G     7   \n",
       "210  CLDN2                 -                  -                    -         \n",
       "211    CEL                 -                  -                    -         \n",
       "212   CASR                 -                  -                    -         \n",
       "213   GGT1                 -                  -                    -         \n",
       "\n",
       "           POS REF_ALT         rsID Variant mechanism  \\\n",
       "0    117587806     G-A   rs75527207               NaN   \n",
       "1    117652877     C-G   rs80034486               NaN   \n",
       "2    117710365   GA-AT   rs78655421               NaN   \n",
       "3    117509123     G-A   rs75961395               NaN   \n",
       "4    117540230     C-T  rs121909011               NaN   \n",
       "..         ...     ...          ...               ...   \n",
       "209  142873480     C-G            -               NaN   \n",
       "210                               -               NaN   \n",
       "211                               -               NaN   \n",
       "212                               -               NaN   \n",
       "213                               -               NaN   \n",
       "\n",
       "                                             Gene Info  \\\n",
       "0    Cystic fibrosis transmembrane conductance regu...   \n",
       "1    Cystic fibrosis transmembrane conductance regu...   \n",
       "2    Cystic fibrosis transmembrane conductance regu...   \n",
       "3    Cystic fibrosis transmembrane conductance regu...   \n",
       "4    Cystic fibrosis transmembrane conductance regu...   \n",
       "..                                                 ...   \n",
       "209  Transient receptor potential vanilloid subfami...   \n",
       "210  CLDN2 gene product belongs to the claudin prot...   \n",
       "211  The enzyme carboxyl ester lipase (CEL), hydrol...   \n",
       "212  The CASR allows regulation of parathyroid horm...   \n",
       "213  GGT1 encodes for the membrane-bound enzyme gam...   \n",
       "\n",
       "                                        Publication  \n",
       "0    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "1    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "2    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "3    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "4    PMID: 19092437, PMID: 34557648, PMID: 32153386  \n",
       "..                                              ...  \n",
       "209                                  PMID: 33610740  \n",
       "210                                             NaN  \n",
       "211                                             NaN  \n",
       "212                                             NaN  \n",
       "213                                             NaN  \n",
       "\n",
       "[214 rows x 11 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y = y[['Gene', 'Amino acid change', 'Uploaded_variant', 'Chr-Position-Ref-Alt', 'CHROM', 'POS', 'REF_ALT', 'rsID', \n",
    "      'Variant mechanism', 'Gene Info', 'Publication']]\n",
    "y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "e7095201",
   "metadata": {},
   "outputs": [],
   "source": [
    "y.to_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Final_Pancreatitis_data.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7be49703",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Change</th>\n",
       "      <th>Uploaded_variant_hg38</th>\n",
       "      <th>Uploaded_variant_hg37</th>\n",
       "      <th>Chrom-pos-Ref-Alt_38</th>\n",
       "      <th>rsID_hg38</th>\n",
       "      <th>Chrom-pos-Ref-Alt_37</th>\n",
       "      <th>rsID_hg37</th>\n",
       "      <th>chrom</th>\n",
       "      <th>CHROM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>c.78C&gt;A</td>\n",
       "      <td>KCNJ11:c.78C&gt;A</td>\n",
       "      <td>KCNJ11:c.78C&gt;A</td>\n",
       "      <td>11-17387753-G-T, 11-17388014-G-T, 11-17387786-G-T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11-17409561-G-T, 11-17409300-G-T, 11-17409578-G-T</td>\n",
       "      <td>rs1393796559</td>\n",
       "      <td>11-17387753-G-T</td>\n",
       "      <td>17387753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>c.78C&gt;A</td>\n",
       "      <td>KCNJ11:c.78C&gt;A</td>\n",
       "      <td>KCNJ11:c.78C&gt;A</td>\n",
       "      <td>11-17387753-G-T, 11-17388014-G-T, 11-17387786-G-T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11-17409561-G-T, 11-17409300-G-T, 11-17409578-G-T</td>\n",
       "      <td>rs1393796559</td>\n",
       "      <td>11-17388014-G-T</td>\n",
       "      <td>17388014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>c.78C&gt;A</td>\n",
       "      <td>KCNJ11:c.78C&gt;A</td>\n",
       "      <td>KCNJ11:c.78C&gt;A</td>\n",
       "      <td>11-17387753-G-T, 11-17388014-G-T, 11-17387786-G-T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11-17409561-G-T, 11-17409300-G-T, 11-17409578-G-T</td>\n",
       "      <td>rs1393796559</td>\n",
       "      <td>11-17387786-G-T</td>\n",
       "      <td>17387786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>c.79C&gt;T</td>\n",
       "      <td>KCNJ11:c.79C&gt;T</td>\n",
       "      <td>KCNJ11:c.79C&gt;T</td>\n",
       "      <td>11-17388013-G-A</td>\n",
       "      <td>rs752507753</td>\n",
       "      <td>11-17409560-G-A, 11-17409577-G-A</td>\n",
       "      <td>rs752507753</td>\n",
       "      <td>11-17388013-G-A</td>\n",
       "      <td>17388013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>c.112A&gt;G</td>\n",
       "      <td>KCNJ11:c.112A&gt;G</td>\n",
       "      <td>KCNJ11:c.112A&gt;G</td>\n",
       "      <td>11-17387719-T-C, 11-17387980-T-C, 11-17386639-...</td>\n",
       "      <td>rs914289214</td>\n",
       "      <td>11-17409527-T-C, 11-17409266-T-C, 11-17408186-T-C</td>\n",
       "      <td>rs914289214</td>\n",
       "      <td>11-17387719-T-C</td>\n",
       "      <td>17387719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>ALDOB</td>\n",
       "      <td>c.841_842delAC</td>\n",
       "      <td>ALDOB:c.841_842delAC</td>\n",
       "      <td>ALDOB:c.841_842delAC</td>\n",
       "      <td>9-101424999-AGT-A, 9-101429312-CAC-C, 9-101429...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9-104187281-AGT-A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9-101429406-GCC-G</td>\n",
       "      <td>101429406</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>ALDOB</td>\n",
       "      <td>c.841_842delAC</td>\n",
       "      <td>ALDOB:c.841_842delAC</td>\n",
       "      <td>ALDOB:c.841_842delAC</td>\n",
       "      <td>9-101424999-AGT-A, 9-101429312-CAC-C, 9-101429...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9-104187281-AGT-A</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>ALDOB</td>\n",
       "      <td>c.851T&gt;C</td>\n",
       "      <td>ALDOB:c.851T&gt;C</td>\n",
       "      <td>ALDOB:c.851T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>ALDOB</td>\n",
       "      <td>c.1005C&gt;G</td>\n",
       "      <td>ALDOB:c.1005C&gt;G</td>\n",
       "      <td>ALDOB:c.1005C&gt;G</td>\n",
       "      <td>9-101421899-G-C,</td>\n",
       "      <td>rs78340951</td>\n",
       "      <td>9-104184181-G-C</td>\n",
       "      <td>rs78340951</td>\n",
       "      <td>9-101421899-G-C</td>\n",
       "      <td>101421899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>ALDOB</td>\n",
       "      <td>c.1005C&gt;G</td>\n",
       "      <td>ALDOB:c.1005C&gt;G</td>\n",
       "      <td>ALDOB:c.1005C&gt;G</td>\n",
       "      <td>9-101421899-G-C,</td>\n",
       "      <td>rs78340951</td>\n",
       "      <td>9-104184181-G-C</td>\n",
       "      <td>rs78340951</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>462 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene          Change Uploaded_variant_hg38 Uploaded_variant_hg37  \\\n",
       "0    KCNJ11         c.78C>A        KCNJ11:c.78C>A        KCNJ11:c.78C>A   \n",
       "0    KCNJ11         c.78C>A        KCNJ11:c.78C>A        KCNJ11:c.78C>A   \n",
       "0    KCNJ11         c.78C>A        KCNJ11:c.78C>A        KCNJ11:c.78C>A   \n",
       "1    KCNJ11         c.79C>T        KCNJ11:c.79C>T        KCNJ11:c.79C>T   \n",
       "2    KCNJ11        c.112A>G       KCNJ11:c.112A>G       KCNJ11:c.112A>G   \n",
       "..      ...             ...                   ...                   ...   \n",
       "189   ALDOB  c.841_842delAC  ALDOB:c.841_842delAC  ALDOB:c.841_842delAC   \n",
       "189   ALDOB  c.841_842delAC  ALDOB:c.841_842delAC  ALDOB:c.841_842delAC   \n",
       "190   ALDOB        c.851T>C        ALDOB:c.851T>C        ALDOB:c.851T>C   \n",
       "191   ALDOB       c.1005C>G       ALDOB:c.1005C>G       ALDOB:c.1005C>G   \n",
       "191   ALDOB       c.1005C>G       ALDOB:c.1005C>G       ALDOB:c.1005C>G   \n",
       "\n",
       "                                  Chrom-pos-Ref-Alt_38    rsID_hg38  \\\n",
       "0    11-17387753-G-T, 11-17388014-G-T, 11-17387786-G-T          NaN   \n",
       "0    11-17387753-G-T, 11-17388014-G-T, 11-17387786-G-T          NaN   \n",
       "0    11-17387753-G-T, 11-17388014-G-T, 11-17387786-G-T          NaN   \n",
       "1                                      11-17388013-G-A  rs752507753   \n",
       "2    11-17387719-T-C, 11-17387980-T-C, 11-17386639-...  rs914289214   \n",
       "..                                                 ...          ...   \n",
       "189  9-101424999-AGT-A, 9-101429312-CAC-C, 9-101429...          NaN   \n",
       "189  9-101424999-AGT-A, 9-101429312-CAC-C, 9-101429...          NaN   \n",
       "190                                                NaN          NaN   \n",
       "191                                  9-101421899-G-C,    rs78340951   \n",
       "191                                  9-101421899-G-C,    rs78340951   \n",
       "\n",
       "                                  Chrom-pos-Ref-Alt_37     rsID_hg37  \\\n",
       "0    11-17409561-G-T, 11-17409300-G-T, 11-17409578-G-T  rs1393796559   \n",
       "0    11-17409561-G-T, 11-17409300-G-T, 11-17409578-G-T  rs1393796559   \n",
       "0    11-17409561-G-T, 11-17409300-G-T, 11-17409578-G-T  rs1393796559   \n",
       "1                     11-17409560-G-A, 11-17409577-G-A   rs752507753   \n",
       "2    11-17409527-T-C, 11-17409266-T-C, 11-17408186-T-C   rs914289214   \n",
       "..                                                 ...           ...   \n",
       "189                                  9-104187281-AGT-A           NaN   \n",
       "189                                  9-104187281-AGT-A           NaN   \n",
       "190                                                NaN           NaN   \n",
       "191                                    9-104184181-G-C    rs78340951   \n",
       "191                                    9-104184181-G-C    rs78340951   \n",
       "\n",
       "                  chrom      CHROM  \n",
       "0       11-17387753-G-T   17387753  \n",
       "0       11-17388014-G-T   17388014  \n",
       "0       11-17387786-G-T   17387786  \n",
       "1       11-17388013-G-A   17388013  \n",
       "2       11-17387719-T-C   17387719  \n",
       "..                  ...        ...  \n",
       "189   9-101429406-GCC-G  101429406  \n",
       "189                            NaN  \n",
       "190                 NaN        NaN  \n",
       "191     9-101421899-G-C  101421899  \n",
       "191                            NaN  \n",
       "\n",
       "[462 rows x 10 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "x = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Hypoglycemia_Lit_final_Positions_hg38_hg37.xlsx')\n",
    "x['chrom'] = x['Chrom-pos-Ref-Alt_38'].str.split(',')\n",
    "x = x.explode('chrom')\n",
    "\n",
    "x['CHROM'] = x['chrom'].str.split('-').str[1]\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6e384f82",
   "metadata": {},
   "outputs": [],
   "source": [
    "x.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/hypogly.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21360d59",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
